--- com/mojang/blaze3d/platform/DebugMemoryUntracker.java
+++ com/mojang/blaze3d/platform/DebugMemoryUntracker.java
@@ -11,20 +11,20 @@
     public com.mojang.blaze3d.platform.DebugMemoryUntracker() {
     }
 
-    public static void untrack(long memAddr) {
+    public static void untrack(long $$0) {
         if (UNTRACK == null) {
             return;
         }
         try {
-            UNTRACK.invoke(memAddr);
+            UNTRACK.invoke($$0);
         }
-        catch (java.lang.Throwable throwable) {
-            throw new java.lang.RuntimeException(throwable);
+        catch (java.lang.Throwable $$1) {
+            throw new java.lang.RuntimeException($$1);
         }
     }
 
-    public static void untrack(org.lwjgl.system.Pointer pointer) {
-        com.mojang.blaze3d.platform.DebugMemoryUntracker.untrack((long)pointer.address());
+    public static void untrack(org.lwjgl.system.Pointer $$0) {
+        com.mojang.blaze3d.platform.DebugMemoryUntracker.untrack((long)$$0.address());
     }
 
     public static /* synthetic */ java.lang.invoke.MethodHandle lambda$static$0() {