--- com/mojang/blaze3d/platform/DisplayData.java
+++ com/mojang/blaze3d/platform/DisplayData.java
@@ -11,12 +11,12 @@
     public java.util.OptionalInt fullscreenHeight;
     public boolean isFullscreen;
 
-    public com.mojang.blaze3d.platform.DisplayData(int width, int height, java.util.OptionalInt fullscreenWidth, java.util.OptionalInt fullscreenHeight, boolean isFullscreen) {
-        this.width = width;
-        this.height = height;
-        this.fullscreenWidth = fullscreenWidth;
-        this.fullscreenHeight = fullscreenHeight;
-        this.isFullscreen = isFullscreen;
+    public com.mojang.blaze3d.platform.DisplayData(int $$0, int $$1, java.util.OptionalInt $$2, java.util.OptionalInt $$3, boolean $$4) {
+        this.width = $$0;
+        this.height = $$1;
+        this.fullscreenWidth = $$2;
+        this.fullscreenHeight = $$3;
+        this.isFullscreen = $$4;
     }
 }
 