--- com/mojang/blaze3d/platform/GlStateManager$DestFactor.java
+++ com/mojang/blaze3d/platform/GlStateManager$DestFactor.java
@@ -32,9 +32,9 @@
         return (com.mojang.blaze3d.platform.GlStateManager$DestFactor)java.lang.Enum.valueOf(com.mojang.blaze3d.platform.GlStateManager$DestFactor.class, (java.lang.String)$$0);
     }
 
-    public com.mojang.blaze3d.platform.GlStateManager$DestFactor(int value) {
+    public com.mojang.blaze3d.platform.GlStateManager$DestFactor(int $$0) {
         super(string, n);
-        this.value = value;
+        this.value = $$0;
     }
 
     public static /* synthetic */ com.mojang.blaze3d.platform.GlStateManager$DestFactor[] $values() {