--- com/mojang/blaze3d/platform/GlStateManager$BooleanState.java
+++ com/mojang/blaze3d/platform/GlStateManager$BooleanState.java
@@ -8,8 +8,8 @@
     public int state;
     public boolean enabled;
 
-    public com.mojang.blaze3d.platform.GlStateManager$BooleanState(int state) {
-        this.state = state;
+    public com.mojang.blaze3d.platform.GlStateManager$BooleanState(int $$0) {
+        this.state = $$0;
     }
 
     public void disable() {
@@ -20,11 +20,11 @@
         this.setEnabled(true);
     }
 
-    public void setEnabled(boolean enabled) {
+    public void setEnabled(boolean $$0) {
         com.mojang.blaze3d.systems.RenderSystem.assertOnRenderThreadOrInit();
-        if (enabled != this.enabled) {
-            this.enabled = enabled;
-            if (enabled) {
+        if ($$0 != this.enabled) {
+            this.enabled = $$0;
+            if ($$0) {
                 org.lwjgl.opengl.GL11.glEnable((int)this.state);
             } else {
                 org.lwjgl.opengl.GL11.glDisable((int)this.state);