--- com/mojang/blaze3d/platform/GlStateManager$LogicOp.java
+++ com/mojang/blaze3d/platform/GlStateManager$LogicOp.java
@@ -33,9 +33,9 @@
         return (com.mojang.blaze3d.platform.GlStateManager$LogicOp)java.lang.Enum.valueOf(com.mojang.blaze3d.platform.GlStateManager$LogicOp.class, (java.lang.String)$$0);
     }
 
-    public com.mojang.blaze3d.platform.GlStateManager$LogicOp(int value) {
+    public com.mojang.blaze3d.platform.GlStateManager$LogicOp(int $$0) {
         super(string, n);
-        this.value = value;
+        this.value = $$0;
     }
 
     public static /* synthetic */ com.mojang.blaze3d.platform.GlStateManager$LogicOp[] $values() {