--- com/mojang/blaze3d/platform/InputConstants$Key.java
+++ com/mojang/blaze3d/platform/InputConstants$Key.java
@@ -11,12 +11,12 @@
     public net.minecraft.util.LazyLoadedValue<net.minecraft.network.chat.Component> displayName;
     public static java.util.Map<java.lang.String, com.mojang.blaze3d.platform.InputConstants$Key> NAME_MAP;
 
-    public com.mojang.blaze3d.platform.InputConstants$Key(java.lang.String name, com.mojang.blaze3d.platform.InputConstants$Type type, int value) {
-        this.name = name;
-        this.type = type;
-        this.value = value;
-        this.displayName = new net.minecraft.util.LazyLoadedValue(() -> (net.minecraft.network.chat.Component)$$0.displayTextSupplier.apply((java.lang.Object)value, (java.lang.Object)name));
-        NAME_MAP.put((java.lang.Object)name, (java.lang.Object)this);
+    public com.mojang.blaze3d.platform.InputConstants$Key(java.lang.String $$0, com.mojang.blaze3d.platform.InputConstants$Type $$1, int $$2) {
+        this.name = $$0;
+        this.type = $$1;
+        this.value = $$2;
+        this.displayName = new net.minecraft.util.LazyLoadedValue(() -> (net.minecraft.network.chat.Component)$$0.displayTextSupplier.apply((java.lang.Object)$$2, (java.lang.Object)$$0));
+        NAME_MAP.put((java.lang.Object)$$0, (java.lang.Object)this);
     }
 
     public com.mojang.blaze3d.platform.InputConstants$Type getType() {
@@ -45,15 +45,15 @@
         return java.util.OptionalInt.empty();
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other == null || this.getClass() != other.getClass()) {
+        if ($$0 == null || this.getClass() != $$0.getClass()) {
             return false;
         }
-        com.mojang.blaze3d.platform.InputConstants$Key key = (com.mojang.blaze3d.platform.InputConstants$Key)other;
-        return this.value == key.value && this.type == key.type;
+        com.mojang.blaze3d.platform.InputConstants$Key $$1 = (com.mojang.blaze3d.platform.InputConstants$Key)$$0;
+        return this.value == $$1.value && this.type == $$1.type;
     }
 
     public int hashCode() {