--- com/mojang/blaze3d/platform/InputConstants$Type.java
+++ com/mojang/blaze3d/platform/InputConstants$Type.java
@@ -22,20 +22,20 @@
         return (com.mojang.blaze3d.platform.InputConstants$Type)java.lang.Enum.valueOf(com.mojang.blaze3d.platform.InputConstants$Type.class, (java.lang.String)$$0);
     }
 
-    public static void addKey(com.mojang.blaze3d.platform.InputConstants$Type type, java.lang.String name, int keyCode) {
-        com.mojang.blaze3d.platform.InputConstants$Key key = new com.mojang.blaze3d.platform.InputConstants$Key(name, type, keyCode);
-        type.map.put(keyCode, (java.lang.Object)key);
+    public static void addKey(com.mojang.blaze3d.platform.InputConstants$Type $$0, java.lang.String $$1, int $$2) {
+        com.mojang.blaze3d.platform.InputConstants$Key $$3 = new com.mojang.blaze3d.platform.InputConstants$Key($$1, $$0, $$2);
+        $$0.map.put($$2, (java.lang.Object)$$3);
     }
 
-    public com.mojang.blaze3d.platform.InputConstants$Type(java.lang.String defaultPrefix, java.util.function.BiFunction<java.lang.Integer, java.lang.String, net.minecraft.network.chat.Component> displayTextSupplier) {
+    public com.mojang.blaze3d.platform.InputConstants$Type(java.lang.String $$0, java.util.function.BiFunction<java.lang.Integer, java.lang.String, net.minecraft.network.chat.Component> $$1) {
         super(string, n);
         this.map = new it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap();
-        this.defaultPrefix = defaultPrefix;
-        this.displayTextSupplier = displayTextSupplier;
+        this.defaultPrefix = $$0;
+        this.displayTextSupplier = $$1;
     }
 
-    public com.mojang.blaze3d.platform.InputConstants$Key getOrCreate(int keyCode) {
-        return (com.mojang.blaze3d.platform.InputConstants$Key)this.map.computeIfAbsent(keyCode, $$0 -> {
+    public com.mojang.blaze3d.platform.InputConstants$Key getOrCreate(int $$02) {
+        return (com.mojang.blaze3d.platform.InputConstants$Key)this.map.computeIfAbsent($$02, $$0 -> {
             int $$1 = $$0;
             if (this == MOUSE) {
                 ++$$1;
@@ -45,18 +45,18 @@
         });
     }
 
-    public static /* synthetic */ net.minecraft.network.chat.Component lambda$static$2(java.lang.Integer key, java.lang.String translationKey) {
-        return net.minecraft.locale.Language.getInstance().has(translationKey) ? net.minecraft.network.chat.Component.translatable((java.lang.String)translationKey) : net.minecraft.network.chat.Component.translatable((java.lang.String)"key.mouse", (java.lang.Object[])new java.lang.Object[]{key + 1});
+    public static /* synthetic */ net.minecraft.network.chat.Component lambda$static$2(java.lang.Integer $$0, java.lang.String $$1) {
+        return net.minecraft.locale.Language.getInstance().has($$1) ? net.minecraft.network.chat.Component.translatable((java.lang.String)$$1) : net.minecraft.network.chat.Component.translatable((java.lang.String)"key.mouse", (java.lang.Object[])new java.lang.Object[]{$$0 + 1});
     }
 
-    public static /* synthetic */ net.minecraft.network.chat.Component lambda$static$1(java.lang.Integer key, java.lang.String translationKey) {
-        java.lang.String string = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)-1, (int)key);
-        return string != null ? net.minecraft.network.chat.Component.literal((java.lang.String)string) : net.minecraft.network.chat.Component.translatable((java.lang.String)translationKey);
+    public static /* synthetic */ net.minecraft.network.chat.Component lambda$static$1(java.lang.Integer $$0, java.lang.String $$1) {
+        java.lang.String $$2 = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)-1, (int)$$0);
+        return $$2 != null ? net.minecraft.network.chat.Component.literal((java.lang.String)$$2) : net.minecraft.network.chat.Component.translatable((java.lang.String)$$1);
     }
 
-    public static /* synthetic */ net.minecraft.network.chat.Component lambda$static$0(java.lang.Integer key, java.lang.String translationKey) {
-        java.lang.String string = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)key, (int)-1);
-        return string != null ? net.minecraft.network.chat.Component.literal((java.lang.String)string) : net.minecraft.network.chat.Component.translatable((java.lang.String)translationKey);
+    public static /* synthetic */ net.minecraft.network.chat.Component lambda$static$0(java.lang.Integer $$0, java.lang.String $$1) {
+        java.lang.String $$2 = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)$$0, (int)-1);
+        return $$2 != null ? net.minecraft.network.chat.Component.literal((java.lang.String)$$2) : net.minecraft.network.chat.Component.translatable((java.lang.String)$$1);
     }
 
     public static /* synthetic */ com.mojang.blaze3d.platform.InputConstants$Type[] $values() {
@@ -64,15 +64,15 @@
     }
 
     static {
-        KEYSYM = new com.mojang.blaze3d.platform.InputConstants$Type("KEYSYM", 0, "key.keyboard", (key, translationKey) -> {
-            java.lang.String string = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)key, (int)-1);
-            return string != null ? net.minecraft.network.chat.Component.literal((java.lang.String)string) : net.minecraft.network.chat.Component.translatable((java.lang.String)translationKey);
+        KEYSYM = new com.mojang.blaze3d.platform.InputConstants$Type("KEYSYM", 0, "key.keyboard", ($$0, $$1) -> {
+            java.lang.String $$2 = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)$$0, (int)-1);
+            return $$2 != null ? net.minecraft.network.chat.Component.literal((java.lang.String)$$2) : net.minecraft.network.chat.Component.translatable((java.lang.String)$$1);
         });
-        SCANCODE = new com.mojang.blaze3d.platform.InputConstants$Type("SCANCODE", 1, "scancode", (key, translationKey) -> {
-            java.lang.String string = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)-1, (int)key);
-            return string != null ? net.minecraft.network.chat.Component.literal((java.lang.String)string) : net.minecraft.network.chat.Component.translatable((java.lang.String)translationKey);
+        SCANCODE = new com.mojang.blaze3d.platform.InputConstants$Type("SCANCODE", 1, "scancode", ($$0, $$1) -> {
+            java.lang.String $$2 = org.lwjgl.glfw.GLFW.glfwGetKeyName((int)-1, (int)$$0);
+            return $$2 != null ? net.minecraft.network.chat.Component.literal((java.lang.String)$$2) : net.minecraft.network.chat.Component.translatable((java.lang.String)$$1);
         });
-        MOUSE = new com.mojang.blaze3d.platform.InputConstants$Type("MOUSE", 2, "key.mouse", (key, translationKey) -> net.minecraft.locale.Language.getInstance().has(translationKey) ? net.minecraft.network.chat.Component.translatable((java.lang.String)translationKey) : net.minecraft.network.chat.Component.translatable((java.lang.String)"key.mouse", (java.lang.Object[])new java.lang.Object[]{key + 1}));
+        MOUSE = new com.mojang.blaze3d.platform.InputConstants$Type("MOUSE", 2, "key.mouse", ($$0, $$1) -> net.minecraft.locale.Language.getInstance().has($$1) ? net.minecraft.network.chat.Component.translatable((java.lang.String)$$1) : net.minecraft.network.chat.Component.translatable((java.lang.String)"key.mouse", (java.lang.Object[])new java.lang.Object[]{$$0 + 1}));
         $VALUES = com.mojang.blaze3d.platform.InputConstants$Type.$values();
         com.mojang.blaze3d.platform.InputConstants$Type.addKey((com.mojang.blaze3d.platform.InputConstants$Type)KEYSYM, (java.lang.String)"key.keyboard.unknown", (int)-1);
         com.mojang.blaze3d.platform.InputConstants$Type.addKey((com.mojang.blaze3d.platform.InputConstants$Type)MOUSE, (java.lang.String)"key.mouse.left", (int)0);