--- com/mojang/blaze3d/platform/Lighting.java
+++ com/mojang/blaze3d/platform/Lighting.java
@@ -15,12 +15,12 @@
     public com.mojang.blaze3d.platform.Lighting() {
     }
 
-    public static void setupNetherLevel(com.mojang.math.Matrix4f matrix) {
-        com.mojang.blaze3d.systems.RenderSystem.setupLevelDiffuseLighting((com.mojang.math.Vector3f)NETHER_DIFFUSE_LIGHT_0, (com.mojang.math.Vector3f)NETHER_DIFFUSE_LIGHT_1, (com.mojang.math.Matrix4f)matrix);
+    public static void setupNetherLevel(com.mojang.math.Matrix4f $$0) {
+        com.mojang.blaze3d.systems.RenderSystem.setupLevelDiffuseLighting((com.mojang.math.Vector3f)NETHER_DIFFUSE_LIGHT_0, (com.mojang.math.Vector3f)NETHER_DIFFUSE_LIGHT_1, (com.mojang.math.Matrix4f)$$0);
     }
 
-    public static void setupLevel(com.mojang.math.Matrix4f matrix) {
-        com.mojang.blaze3d.systems.RenderSystem.setupLevelDiffuseLighting((com.mojang.math.Vector3f)DIFFUSE_LIGHT_0, (com.mojang.math.Vector3f)DIFFUSE_LIGHT_1, (com.mojang.math.Matrix4f)matrix);
+    public static void setupLevel(com.mojang.math.Matrix4f $$0) {
+        com.mojang.blaze3d.systems.RenderSystem.setupLevelDiffuseLighting((com.mojang.math.Vector3f)DIFFUSE_LIGHT_0, (com.mojang.math.Vector3f)DIFFUSE_LIGHT_1, (com.mojang.math.Matrix4f)$$0);
     }
 
     public static void setupForFlatItems() {