--- com/mojang/blaze3d/platform/MacosUtil.java
+++ com/mojang/blaze3d/platform/MacosUtil.java
@@ -30,12 +30,15 @@
         $$0.send("toggleFullScreen:", new java.lang.Object[]{com.sun.jna.Pointer.NULL});
     }
 
-    public static void loadIcon(java.io.InputStream $$0) throws java.io.IOException {
-        java.lang.String $$1 = java.util.Base64.getEncoder().encodeToString($$0.readAllBytes());
-        ca.weblite.objc.Client $$2 = ca.weblite.objc.Client.getInstance();
-        java.lang.Object $$3 = $$2.sendProxy("NSData", "alloc", new java.lang.Object[0]).send("initWithBase64Encoding:", new java.lang.Object[]{$$1});
-        java.lang.Object $$4 = $$2.sendProxy("NSImage", "alloc", new java.lang.Object[0]).send("initWithData:", new java.lang.Object[]{$$3});
-        $$2.sendProxy("NSApplication", "sharedApplication", new java.lang.Object[0]).send("setApplicationIconImage:", new java.lang.Object[]{$$4});
+    public static void loadIcon(net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream> $$0) throws java.io.IOException {
+        try (java.io.InputStream $$1 = (java.io.InputStream)$$0.get();
+){
+            java.lang.String $$2 = java.util.Base64.getEncoder().encodeToString($$1.readAllBytes());
+            ca.weblite.objc.Client $$3 = ca.weblite.objc.Client.getInstance();
+            java.lang.Object $$4 = $$3.sendProxy("NSData", "alloc", new java.lang.Object[0]).send("initWithBase64Encoding:", new java.lang.Object[]{$$2});
+            java.lang.Object $$5 = $$3.sendProxy("NSImage", "alloc", new java.lang.Object[0]).send("initWithData:", new java.lang.Object[]{$$4});
+            $$3.sendProxy("NSApplication", "sharedApplication", new java.lang.Object[0]).send("setApplicationIconImage:", new java.lang.Object[]{$$5});
+        }
     }
 }
 