--- com/mojang/blaze3d/platform/Monitor.java
+++ com/mojang/blaze3d/platform/Monitor.java
@@ -11,8 +11,8 @@
     public int x;
     public int y;
 
-    public com.mojang.blaze3d.platform.Monitor(long monitor) {
-        this.monitor = monitor;
+    public com.mojang.blaze3d.platform.Monitor(long $$0) {
+        this.monitor = $$0;
         this.videoModes = com.google.common.collect.Lists.newArrayList();
         this.refreshVideoModes();
     }
@@ -36,21 +36,21 @@
         this.currentMode = new com.mojang.blaze3d.platform.VideoMode($$5);
     }
 
-    public com.mojang.blaze3d.platform.VideoMode getPreferredVidMode(java.util.Optional<com.mojang.blaze3d.platform.VideoMode> videoMode) {
+    public com.mojang.blaze3d.platform.VideoMode getPreferredVidMode(java.util.Optional<com.mojang.blaze3d.platform.VideoMode> $$0) {
         com.mojang.blaze3d.systems.RenderSystem.assertInInitPhase();
-        if (videoMode.isPresent()) {
-            com.mojang.blaze3d.platform.VideoMode videoMode2 = (com.mojang.blaze3d.platform.VideoMode)videoMode.get();
-            for (com.mojang.blaze3d.platform.VideoMode videoMode3 : this.videoModes) {
-                if (!videoMode3.equals((java.lang.Object)videoMode2)) continue;
-                return videoMode3;
+        if ($$0.isPresent()) {
+            com.mojang.blaze3d.platform.VideoMode $$1 = (com.mojang.blaze3d.platform.VideoMode)$$0.get();
+            for (com.mojang.blaze3d.platform.VideoMode $$2 : this.videoModes) {
+                if (!$$2.equals((java.lang.Object)$$1)) continue;
+                return $$2;
             }
         }
         return this.getCurrentMode();
     }
 
-    public int getVideoModeIndex(com.mojang.blaze3d.platform.VideoMode videoMode) {
+    public int getVideoModeIndex(com.mojang.blaze3d.platform.VideoMode $$0) {
         com.mojang.blaze3d.systems.RenderSystem.assertInInitPhase();
-        return this.videoModes.indexOf((java.lang.Object)videoMode);
+        return this.videoModes.indexOf((java.lang.Object)$$0);
     }
 
     public com.mojang.blaze3d.platform.VideoMode getCurrentMode() {
@@ -65,8 +65,8 @@
         return this.y;
     }
 
-    public com.mojang.blaze3d.platform.VideoMode getMode(int index) {
-        return (com.mojang.blaze3d.platform.VideoMode)this.videoModes.get(index);
+    public com.mojang.blaze3d.platform.VideoMode getMode(int $$0) {
+        return (com.mojang.blaze3d.platform.VideoMode)this.videoModes.get($$0);
     }
 
     public int getModeCount() {