--- com/mojang/blaze3d/platform/NativeImage$InternalGlFormat.java
+++ com/mojang/blaze3d/platform/NativeImage$InternalGlFormat.java
@@ -21,9 +21,9 @@
         return (com.mojang.blaze3d.platform.NativeImage$InternalGlFormat)java.lang.Enum.valueOf(com.mojang.blaze3d.platform.NativeImage$InternalGlFormat.class, (java.lang.String)$$0);
     }
 
-    public com.mojang.blaze3d.platform.NativeImage$InternalGlFormat(int glFormat) {
+    public com.mojang.blaze3d.platform.NativeImage$InternalGlFormat(int $$0) {
         super(string, n);
-        this.glFormat = glFormat;
+        this.glFormat = $$0;
     }
 
     public int glFormat() {