--- com/mojang/blaze3d/platform/NativeImage$Format.java
+++ com/mojang/blaze3d/platform/NativeImage$Format.java
@@ -33,21 +33,21 @@
         return (com.mojang.blaze3d.platform.NativeImage$Format)java.lang.Enum.valueOf(com.mojang.blaze3d.platform.NativeImage$Format.class, (java.lang.String)$$0);
     }
 
-    public com.mojang.blaze3d.platform.NativeImage$Format(int components, int glFormat, boolean hasRed, boolean hasGreen, boolean hasBlue, boolean hasLuminance, boolean hasAlpha, int redOffset, int greenOffset, int blueOffset, int luminanceOffset, int alphaOffset, boolean supportedByStb) {
+    public com.mojang.blaze3d.platform.NativeImage$Format(int $$0, int $$1, boolean $$2, boolean $$3, boolean $$4, boolean $$5, boolean $$6, int $$7, int $$8, int $$9, int $$10, int $$11, boolean $$12) {
         super(string, n);
-        this.components = components;
-        this.glFormat = glFormat;
-        this.hasRed = hasRed;
-        this.hasGreen = hasGreen;
-        this.hasBlue = hasBlue;
-        this.hasLuminance = hasLuminance;
-        this.hasAlpha = hasAlpha;
-        this.redOffset = redOffset;
-        this.greenOffset = greenOffset;
-        this.blueOffset = blueOffset;
-        this.luminanceOffset = luminanceOffset;
-        this.alphaOffset = alphaOffset;
-        this.supportedByStb = supportedByStb;
+        this.components = $$0;
+        this.glFormat = $$1;
+        this.hasRed = $$2;
+        this.hasGreen = $$3;
+        this.hasBlue = $$4;
+        this.hasLuminance = $$5;
+        this.hasAlpha = $$6;
+        this.redOffset = $$7;
+        this.greenOffset = $$8;
+        this.blueOffset = $$9;
+        this.luminanceOffset = $$10;
+        this.alphaOffset = $$11;
+        this.supportedByStb = $$12;
     }
 
     public int components() {
@@ -144,8 +144,8 @@
         return this.supportedByStb;
     }
 
-    public static com.mojang.blaze3d.platform.NativeImage$Format getStbFormat(int channels) {
-        switch (channels) {
+    public static com.mojang.blaze3d.platform.NativeImage$Format getStbFormat(int $$0) {
+        switch ($$0) {
             case 1: {
                 return LUMINANCE;
             }