--- com/mojang/blaze3d/platform/NativeImage$WriteCallback.java
+++ com/mojang/blaze3d/platform/NativeImage$WriteCallback.java
@@ -10,8 +10,8 @@
     @javax.annotation.Nullable
     public java.io.IOException exception;
 
-    public com.mojang.blaze3d.platform.NativeImage$WriteCallback(java.nio.channels.WritableByteChannel output) {
-        this.output = output;
+    public com.mojang.blaze3d.platform.NativeImage$WriteCallback(java.nio.channels.WritableByteChannel $$0) {
+        this.output = $$0;
     }
 
     public void invoke(long $$0, long $$1, int $$2) {