--- com/mojang/blaze3d/platform/VideoMode.java
+++ com/mojang/blaze3d/platform/VideoMode.java
@@ -13,31 +13,31 @@
     public int refreshRate;
     public static java.util.regex.Pattern PATTERN;
 
-    public com.mojang.blaze3d.platform.VideoMode(int width, int height, int redBits, int greenBits, int blueBits, int refreshRate) {
-        this.width = width;
-        this.height = height;
-        this.redBits = redBits;
-        this.greenBits = greenBits;
-        this.blueBits = blueBits;
-        this.refreshRate = refreshRate;
+    public com.mojang.blaze3d.platform.VideoMode(int $$0, int $$1, int $$2, int $$3, int $$4, int $$5) {
+        this.width = $$0;
+        this.height = $$1;
+        this.redBits = $$2;
+        this.greenBits = $$3;
+        this.blueBits = $$4;
+        this.refreshRate = $$5;
     }
 
-    public com.mojang.blaze3d.platform.VideoMode(org.lwjgl.glfw.GLFWVidMode$Buffer bufferVideoMode) {
-        this.width = bufferVideoMode.width();
-        this.height = bufferVideoMode.height();
-        this.redBits = bufferVideoMode.redBits();
-        this.greenBits = bufferVideoMode.greenBits();
-        this.blueBits = bufferVideoMode.blueBits();
-        this.refreshRate = bufferVideoMode.refreshRate();
+    public com.mojang.blaze3d.platform.VideoMode(org.lwjgl.glfw.GLFWVidMode$Buffer $$0) {
+        this.width = $$0.width();
+        this.height = $$0.height();
+        this.redBits = $$0.redBits();
+        this.greenBits = $$0.greenBits();
+        this.blueBits = $$0.blueBits();
+        this.refreshRate = $$0.refreshRate();
     }
 
-    public com.mojang.blaze3d.platform.VideoMode(org.lwjgl.glfw.GLFWVidMode glfwVideoMode) {
-        this.width = glfwVideoMode.width();
-        this.height = glfwVideoMode.height();
-        this.redBits = glfwVideoMode.redBits();
-        this.greenBits = glfwVideoMode.greenBits();
-        this.blueBits = glfwVideoMode.blueBits();
-        this.refreshRate = glfwVideoMode.refreshRate();
+    public com.mojang.blaze3d.platform.VideoMode(org.lwjgl.glfw.GLFWVidMode $$0) {
+        this.width = $$0.width();
+        this.height = $$0.height();
+        this.redBits = $$0.redBits();
+        this.greenBits = $$0.greenBits();
+        this.blueBits = $$0.blueBits();
+        this.refreshRate = $$0.refreshRate();
     }
 
     public int getWidth() {
@@ -64,15 +64,15 @@
         return this.refreshRate;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other == null || this.getClass() != other.getClass()) {
+        if ($$0 == null || this.getClass() != $$0.getClass()) {
             return false;
         }
-        com.mojang.blaze3d.platform.VideoMode videoMode = (com.mojang.blaze3d.platform.VideoMode)other;
-        return this.width == videoMode.width && this.height == videoMode.height && this.redBits == videoMode.redBits && this.greenBits == videoMode.greenBits && this.blueBits == videoMode.blueBits && this.refreshRate == videoMode.refreshRate;
+        com.mojang.blaze3d.platform.VideoMode $$1 = (com.mojang.blaze3d.platform.VideoMode)$$0;
+        return this.width == $$1.width && this.height == $$1.height && this.redBits == $$1.redBits && this.greenBits == $$1.greenBits && this.blueBits == $$1.blueBits && this.refreshRate == $$1.refreshRate;
     }
 
     public int hashCode() {
@@ -83,21 +83,31 @@
         return java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"%sx%s@%s (%sbit)", (java.lang.Object[])new java.lang.Object[]{this.width, this.height, this.refreshRate, this.redBits + this.greenBits + this.blueBits});
     }
 
-    public static java.util.Optional<com.mojang.blaze3d.platform.VideoMode> read(@javax.annotation.Nullable java.lang.String videoMode) {
-        if (videoMode == null) {
+    public static java.util.Optional<com.mojang.blaze3d.platform.VideoMode> read(@javax.annotation.Nullable java.lang.String $$0) {
+        if ($$0 == null) {
             return java.util.Optional.empty();
         }
         try {
-            java.util.regex.Matcher matcher = PATTERN.matcher((java.lang.CharSequence)videoMode);
-            if (matcher.matches()) {
-                int n = java.lang.Integer.parseInt((java.lang.String)matcher.group(1));
-                int n2 = java.lang.Integer.parseInt((java.lang.String)matcher.group(2));
-                java.lang.String string = matcher.group(3);
-                int n3 = string == null ? 60 : java.lang.Integer.parseInt((java.lang.String)string);
-                java.lang.String string2 = matcher.group(4);
-                int n4 = string2 == null ? 24 : java.lang.Integer.parseInt((java.lang.String)string2);
-                int n5 = n4 / 3;
-                return java.util.Optional.of((java.lang.Object)new com.mojang.blaze3d.platform.VideoMode(n, n2, n5, n5, n5, n3));
+            java.util.regex.Matcher $$1 = PATTERN.matcher((java.lang.CharSequence)$$0);
+            if ($$1.matches()) {
+                int $$9;
+                int $$6;
+                int $$2 = java.lang.Integer.parseInt((java.lang.String)$$1.group(1));
+                int $$3 = java.lang.Integer.parseInt((java.lang.String)$$1.group(2));
+                java.lang.String $$4 = $$1.group(3);
+                if ($$4 == null) {
+                    int $$5 = 60;
+                } else {
+                    $$6 = java.lang.Integer.parseInt((java.lang.String)$$4);
+                }
+                java.lang.String $$7 = $$1.group(4);
+                if ($$7 == null) {
+                    int $$8 = 24;
+                } else {
+                    $$9 = java.lang.Integer.parseInt((java.lang.String)$$7);
+                }
+                int $$10 = $$9 / 3;
+                return java.util.Optional.of((java.lang.Object)new com.mojang.blaze3d.platform.VideoMode($$2, $$3, $$10, $$10, $$10, $$6));
             }
         }
         catch (java.lang.Exception exception) {