--- net/minecraft/world/level/block/state/pattern/BlockPattern$BlockPatternMatch.java
+++ net/minecraft/world/level/block/state/pattern/BlockPattern$BlockPatternMatch.java
@@ -13,14 +13,14 @@
     public int height;
     public int depth;
 
-    public net.minecraft.world.level.block.state.pattern.BlockPattern$BlockPatternMatch(net.minecraft.core.BlockPos frontTopLeft, net.minecraft.core.Direction forwards, net.minecraft.core.Direction up, com.google.common.cache.LoadingCache<net.minecraft.core.BlockPos, net.minecraft.world.level.block.state.pattern.BlockInWorld> cache, int width, int height, int depth) {
-        this.frontTopLeft = frontTopLeft;
-        this.forwards = forwards;
-        this.up = up;
-        this.cache = cache;
-        this.width = width;
-        this.height = height;
-        this.depth = depth;
+    public net.minecraft.world.level.block.state.pattern.BlockPattern$BlockPatternMatch(net.minecraft.core.BlockPos $$0, net.minecraft.core.Direction $$1, net.minecraft.core.Direction $$2, com.google.common.cache.LoadingCache<net.minecraft.core.BlockPos, net.minecraft.world.level.block.state.pattern.BlockInWorld> $$3, int $$4, int $$5, int $$6) {
+        this.frontTopLeft = $$0;
+        this.forwards = $$1;
+        this.up = $$2;
+        this.cache = $$3;
+        this.width = $$4;
+        this.height = $$5;
+        this.depth = $$6;
     }
 
     public net.minecraft.core.BlockPos getFrontTopLeft() {
@@ -47,8 +47,8 @@
         return this.depth;
     }
 
-    public net.minecraft.world.level.block.state.pattern.BlockInWorld getBlock(int palmOffset, int thumbOffset, int fingerOffset) {
-        return (net.minecraft.world.level.block.state.pattern.BlockInWorld)this.cache.getUnchecked((java.lang.Object)net.minecraft.world.level.block.state.pattern.BlockPattern.translateAndRotate((net.minecraft.core.BlockPos)this.frontTopLeft, (net.minecraft.core.Direction)this.getForwards(), (net.minecraft.core.Direction)this.getUp(), (int)palmOffset, (int)thumbOffset, (int)fingerOffset));
+    public net.minecraft.world.level.block.state.pattern.BlockInWorld getBlock(int $$0, int $$1, int $$2) {
+        return (net.minecraft.world.level.block.state.pattern.BlockInWorld)this.cache.getUnchecked((java.lang.Object)net.minecraft.world.level.block.state.pattern.BlockPattern.translateAndRotate((net.minecraft.core.BlockPos)this.frontTopLeft, (net.minecraft.core.Direction)this.getForwards(), (net.minecraft.core.Direction)this.getUp(), (int)$$0, (int)$$1, (int)$$2));
     }
 
     public java.lang.String toString() {