--- net/minecraft/world/level/block/state/pattern/BlockInWorld.java
+++ net/minecraft/world/level/block/state/pattern/BlockInWorld.java
@@ -14,10 +14,10 @@
     public net.minecraft.world.level.block.entity.BlockEntity entity;
     public boolean cachedEntity;
 
-    public net.minecraft.world.level.block.state.pattern.BlockInWorld(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos, boolean loadChunks) {
-        this.level = level;
-        this.pos = pos.immutable();
-        this.loadChunks = loadChunks;
+    public net.minecraft.world.level.block.state.pattern.BlockInWorld(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1, boolean $$2) {
+        this.level = $$0;
+        this.pos = $$1.immutable();
+        this.loadChunks = $$2;
     }
 
     public net.minecraft.world.level.block.state.BlockState getState() {
@@ -44,8 +44,8 @@
         return this.pos;
     }
 
-    public static java.util.function.Predicate<net.minecraft.world.level.block.state.pattern.BlockInWorld> hasState(java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> predicates) {
-        return $$1 -> $$1 != null && predicates.test((java.lang.Object)$$1.getState());
+    public static java.util.function.Predicate<net.minecraft.world.level.block.state.pattern.BlockInWorld> hasState(java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> $$0) {
+        return $$1 -> $$1 != null && $$0.test((java.lang.Object)$$1.getState());
     }
 }
 