--- net/minecraft/world/level/block/state/pattern/BlockPattern$BlockCacheLoader.java
+++ net/minecraft/world/level/block/state/pattern/BlockPattern$BlockCacheLoader.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.level.LevelReader level;
     public boolean loadChunks;
 
-    public net.minecraft.world.level.block.state.pattern.BlockPattern$BlockCacheLoader(net.minecraft.world.level.LevelReader level, boolean loadChunks) {
-        this.level = level;
-        this.loadChunks = loadChunks;
+    public net.minecraft.world.level.block.state.pattern.BlockPattern$BlockCacheLoader(net.minecraft.world.level.LevelReader $$0, boolean $$1) {
+        this.level = $$0;
+        this.loadChunks = $$1;
     }
 
     public net.minecraft.world.level.block.state.pattern.BlockInWorld load(net.minecraft.core.BlockPos $$0) {