--- net/minecraft/world/level/material/EmptyFluid.java
+++ net/minecraft/world/level/material/EmptyFluid.java
@@ -13,11 +13,11 @@
         return net.minecraft.world.item.Items.AIR;
     }
 
-    public boolean canBeReplacedWith(net.minecraft.world.level.material.FluidState fluidState, net.minecraft.world.level.BlockGetter blockReader, net.minecraft.core.BlockPos pos, net.minecraft.world.level.material.Fluid fluid, net.minecraft.core.Direction direction) {
+    public boolean canBeReplacedWith(net.minecraft.world.level.material.FluidState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.material.Fluid $$3, net.minecraft.core.Direction $$4) {
         return true;
     }
 
-    public net.minecraft.world.phys.Vec3 getFlow(net.minecraft.world.level.BlockGetter blockReader, net.minecraft.core.BlockPos pos, net.minecraft.world.level.material.FluidState fluidState) {
+    public net.minecraft.world.phys.Vec3 getFlow(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.material.FluidState $$2) {
         return net.minecraft.world.phys.Vec3.ZERO;
     }
 
@@ -41,15 +41,15 @@
         return 0.0f;
     }
 
-    public net.minecraft.world.level.block.state.BlockState createLegacyBlock(net.minecraft.world.level.material.FluidState state) {
+    public net.minecraft.world.level.block.state.BlockState createLegacyBlock(net.minecraft.world.level.material.FluidState $$0) {
         return net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
     }
 
-    public boolean isSource(net.minecraft.world.level.material.FluidState state) {
+    public boolean isSource(net.minecraft.world.level.material.FluidState $$0) {
         return false;
     }
 
-    public int getAmount(net.minecraft.world.level.material.FluidState state) {
+    public int getAmount(net.minecraft.world.level.material.FluidState $$0) {
         return 0;
     }
 