--- net/minecraft/world/level/material/FluidState.java
+++ net/minecraft/world/level/material/FluidState.java
@@ -42,20 +42,20 @@
         return this.getType().getAmount(this);
     }
 
-    public boolean shouldRenderBackwardUpFace(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        for (int i = -1; i <= 1; ++i) {
-            for (int j = -1; j <= 1; ++j) {
-                net.minecraft.core.BlockPos blockPos = pos.offset(i, 0, j);
-                net.minecraft.world.level.material.FluidState fluidState = level.getFluidState(blockPos);
-                if (fluidState.getType().isSame(this.getType()) || level.getBlockState(blockPos).isSolidRender(level, blockPos)) continue;
+    public boolean shouldRenderBackwardUpFace(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1) {
+        for (int $$2 = -1; $$2 <= 1; ++$$2) {
+            for (int $$3 = -1; $$3 <= 1; ++$$3) {
+                net.minecraft.core.BlockPos $$4 = $$1.offset($$2, 0, $$3);
+                net.minecraft.world.level.material.FluidState $$5 = $$0.getFluidState($$4);
+                if ($$5.getType().isSame(this.getType()) || $$0.getBlockState($$4).isSolidRender($$0, $$4)) continue;
                 return true;
             }
         }
         return false;
     }
 
-    public void tick(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        this.getType().tick(level, pos, this);
+    public void tick(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1) {
+        this.getType().tick($$0, $$1, this);
     }
 
     public void animateTick(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.util.RandomSource $$2) {