--- net/minecraft/world/level/material/LavaFluid$Source.java
+++ net/minecraft/world/level/material/LavaFluid$Source.java
@@ -9,11 +9,11 @@
     public net.minecraft.world.level.material.LavaFluid$Source() {
     }
 
-    public int getAmount(net.minecraft.world.level.material.FluidState state) {
+    public int getAmount(net.minecraft.world.level.material.FluidState $$0) {
         return 8;
     }
 
-    public boolean isSource(net.minecraft.world.level.material.FluidState state) {
+    public boolean isSource(net.minecraft.world.level.material.FluidState $$0) {
         return true;
     }
 }