--- net/minecraft/world/level/material/MaterialColor.java
+++ net/minecraft/world/level/material/MaterialColor.java
@@ -71,13 +71,13 @@
     public int col;
     public int id;
 
-    public net.minecraft.world.level.material.MaterialColor(int id, int color) {
-        if (id < 0 || id > 63) {
+    public net.minecraft.world.level.material.MaterialColor(int $$0, int $$1) {
+        if ($$0 < 0 || $$0 > 63) {
             throw new java.lang.IndexOutOfBoundsException("Map colour ID must be between 0 and 63 (inclusive)");
         }
-        this.id = id;
-        this.col = color;
-        net.minecraft.world.level.material.MaterialColor.MATERIAL_COLORS[id] = this;
+        this.id = $$0;
+        this.col = $$1;
+        net.minecraft.world.level.material.MaterialColor.MATERIAL_COLORS[$$0] = this;
     }
 
     public int calculateRGBColor(net.minecraft.world.level.material.MaterialColor$Brightness $$0) {