--- net/minecraft/client/gui/components/spectator/SpectatorGui.java
+++ net/minecraft/client/gui/components/spectator/SpectatorGui.java
@@ -16,14 +16,14 @@
     @javax.annotation.Nullable
     public net.minecraft.client.gui.spectator.SpectatorMenu menu;
 
-    public net.minecraft.client.gui.components.spectator.SpectatorGui(net.minecraft.client.Minecraft minecraft) {
-        this.minecraft = minecraft;
+    public net.minecraft.client.gui.components.spectator.SpectatorGui(net.minecraft.client.Minecraft $$0) {
+        this.minecraft = $$0;
     }
 
-    public void onHotbarSelected(int slot) {
+    public void onHotbarSelected(int $$0) {
         this.lastSelectionTime = net.minecraft.Util.getMillis();
         if (this.menu != null) {
-            this.menu.selectSlot(slot);
+            this.menu.selectSlot($$0);
         } else {
             this.menu = new net.minecraft.client.gui.spectator.SpectatorMenu((net.minecraft.client.gui.spectator.SpectatorMenuListener)this);
         }
@@ -52,56 +52,56 @@
         this.setBlitOffset($$3);
     }
 
-    public void renderPage(com.mojang.blaze3d.vertex.PoseStack poseStack, float f, int n, int n2, net.minecraft.client.gui.spectator.categories.SpectatorPage spectatorPage) {
+    public void renderPage(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3, net.minecraft.client.gui.spectator.categories.SpectatorPage $$4) {
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
-        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)f);
+        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)$$1);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)WIDGETS_LOCATION);
-        this.blit(poseStack, n - 91, n2, 0, 0, 182, 22);
-        if (spectatorPage.getSelectedSlot() >= 0) {
-            this.blit(poseStack, n - 91 - 1 + spectatorPage.getSelectedSlot() * 20, n2 - 1, 0, 22, 24, 22);
+        this.blit($$0, $$2 - 91, $$3, 0, 0, 182, 22);
+        if ($$4.getSelectedSlot() >= 0) {
+            this.blit($$0, $$2 - 91 - 1 + $$4.getSelectedSlot() * 20, $$3 - 1, 0, 22, 24, 22);
         }
-        for (int i = 0; i < 9; ++i) {
-            this.renderSlot(poseStack, i, this.minecraft.getWindow().getGuiScaledWidth() / 2 - 90 + i * 20 + 2, (float)(n2 + 3), f, spectatorPage.getItem(i));
+        for (int $$5 = 0; $$5 < 9; ++$$5) {
+            this.renderSlot($$0, $$5, this.minecraft.getWindow().getGuiScaledWidth() / 2 - 90 + $$5 * 20 + 2, (float)($$3 + 3), $$1, $$4.getItem($$5));
         }
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();
     }
 
-    public void renderSlot(com.mojang.blaze3d.vertex.PoseStack poseStack, int n, int n2, float f, float f2, net.minecraft.client.gui.spectator.SpectatorMenuItem spectatorMenuItem) {
+    public void renderSlot(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3, float $$4, net.minecraft.client.gui.spectator.SpectatorMenuItem $$5) {
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)SPECTATOR_LOCATION);
-        if (spectatorMenuItem != net.minecraft.client.gui.spectator.SpectatorMenu.EMPTY_SLOT) {
-            int n3 = (int)(f2 * 255.0f);
-            poseStack.pushPose();
-            poseStack.translate((double)n2, (double)f, 0.0);
-            float f3 = spectatorMenuItem.isEnabled() ? 1.0f : 0.25f;
-            com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)f3, (float)f3, (float)f3, (float)f2);
-            spectatorMenuItem.renderIcon(poseStack, f3, n3);
-            poseStack.popPose();
-            if (n3 > 3 && spectatorMenuItem.isEnabled()) {
-                net.minecraft.network.chat.Component component = this.minecraft.options.keyHotbarSlots[n].getTranslatedKeyMessage();
-                this.minecraft.font.drawShadow(poseStack, component, (float)(n2 + 19 - 2 - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)component)), f + 6.0f + 3.0f, 0xFFFFFF + (n3 << 24));
+        if ($$5 != net.minecraft.client.gui.spectator.SpectatorMenu.EMPTY_SLOT) {
+            int $$6 = (int)($$4 * 255.0f);
+            $$0.pushPose();
+            $$0.translate((double)$$2, (double)$$3, 0.0);
+            float $$7 = $$5.isEnabled() ? 1.0f : 0.25f;
+            com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)$$7, (float)$$7, (float)$$7, (float)$$4);
+            $$5.renderIcon($$0, $$7, $$6);
+            $$0.popPose();
+            if ($$6 > 3 && $$5.isEnabled()) {
+                net.minecraft.network.chat.Component $$8 = this.minecraft.options.keyHotbarSlots[$$1].getTranslatedKeyMessage();
+                this.minecraft.font.drawShadow($$0, $$8, (float)($$2 + 19 - 2 - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)$$8)), $$3 + 6.0f + 3.0f, 0xFFFFFF + ($$6 << 24));
             }
         }
     }
 
-    public void renderTooltip(com.mojang.blaze3d.vertex.PoseStack poseStack) {
-        int n = (int)(this.getHotbarAlpha() * 255.0f);
-        if (n > 3 && this.menu != null) {
-            net.minecraft.network.chat.Component component;
-            net.minecraft.client.gui.spectator.SpectatorMenuItem spectatorMenuItem = this.menu.getSelectedItem();
-            net.minecraft.network.chat.Component component2 = component = spectatorMenuItem == net.minecraft.client.gui.spectator.SpectatorMenu.EMPTY_SLOT ? this.menu.getSelectedCategory().getPrompt() : spectatorMenuItem.getName();
-            if (component != null) {
-                int n2 = (this.minecraft.getWindow().getGuiScaledWidth() - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)component)) / 2;
-                int n3 = this.minecraft.getWindow().getGuiScaledHeight() - 35;
+    public void renderTooltip(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        int $$1 = (int)(this.getHotbarAlpha() * 255.0f);
+        if ($$1 > 3 && this.menu != null) {
+            net.minecraft.network.chat.Component $$3;
+            net.minecraft.client.gui.spectator.SpectatorMenuItem $$2 = this.menu.getSelectedItem();
+            net.minecraft.network.chat.Component component = $$3 = $$2 == net.minecraft.client.gui.spectator.SpectatorMenu.EMPTY_SLOT ? this.menu.getSelectedCategory().getPrompt() : $$2.getName();
+            if ($$3 != null) {
+                int $$4 = (this.minecraft.getWindow().getGuiScaledWidth() - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)$$3)) / 2;
+                int $$5 = this.minecraft.getWindow().getGuiScaledHeight() - 35;
                 com.mojang.blaze3d.systems.RenderSystem.enableBlend();
                 com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
-                this.minecraft.font.drawShadow(poseStack, component, (float)n2, (float)n3, 0xFFFFFF + (n << 24));
+                this.minecraft.font.drawShadow($$0, $$3, (float)$$4, (float)$$5, 0xFFFFFF + ($$1 << 24));
                 com.mojang.blaze3d.systems.RenderSystem.disableBlend();
             }
         }
     }
 
-    public void onSpectatorMenuClosed(net.minecraft.client.gui.spectator.SpectatorMenu menu) {
+    public void onSpectatorMenuClosed(net.minecraft.client.gui.spectator.SpectatorMenu $$0) {
         this.menu = null;
         this.lastSelectionTime = 0L;
     }