--- net/minecraft/client/gui/spectator/categories/SpectatorPage.java
+++ net/minecraft/client/gui/spectator/categories/SpectatorPage.java
@@ -14,11 +14,11 @@
         this.selection = $$1;
     }
 
-    public net.minecraft.client.gui.spectator.SpectatorMenuItem getItem(int index) {
-        if (index < 0 || index >= this.items.size()) {
+    public net.minecraft.client.gui.spectator.SpectatorMenuItem getItem(int $$0) {
+        if ($$0 < 0 || $$0 >= this.items.size()) {
             return net.minecraft.client.gui.spectator.SpectatorMenu.EMPTY_SLOT;
         }
-        return (net.minecraft.client.gui.spectator.SpectatorMenuItem)com.google.common.base.MoreObjects.firstNonNull((java.lang.Object)((net.minecraft.client.gui.spectator.SpectatorMenuItem)this.items.get(index)), (java.lang.Object)net.minecraft.client.gui.spectator.SpectatorMenu.EMPTY_SLOT);
+        return (net.minecraft.client.gui.spectator.SpectatorMenuItem)com.google.common.base.MoreObjects.firstNonNull((java.lang.Object)((net.minecraft.client.gui.spectator.SpectatorMenuItem)this.items.get($$0)), (java.lang.Object)net.minecraft.client.gui.spectator.SpectatorMenu.EMPTY_SLOT);
     }
 
     public int getSelectedSlot() {