--- net/minecraft/client/gui/spectator/categories/TeleportToTeamMenuCategory.java
+++ net/minecraft/client/gui/spectator/categories/TeleportToTeamMenuCategory.java
@@ -27,8 +27,8 @@
         return TELEPORT_PROMPT;
     }
 
-    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu menu) {
-        menu.selectCategory((net.minecraft.client.gui.spectator.SpectatorMenuCategory)this);
+    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu $$0) {
+        $$0.selectCategory((net.minecraft.client.gui.spectator.SpectatorMenuCategory)this);
     }
 
     public net.minecraft.network.chat.Component getName() {