--- net/minecraft/world/entity/animal/allay/Allay$AllayVibrationListenerConfig.java
+++ net/minecraft/world/entity/animal/allay/Allay$AllayVibrationListenerConfig.java
@@ -13,7 +13,7 @@
     }
 
     public boolean shouldListen(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.gameevent.GameEventListener $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.gameevent.GameEvent $$3, net.minecraft.world.level.gameevent.GameEvent$Context $$4) {
-        if (this.this$0.getLevel() != $$0 || this.this$0.isRemoved() || this.this$0.isNoAi()) {
+        if (this.this$0.isNoAi()) {
             return false;
         }
         java.util.Optional $$5 = this.this$0.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.LIKED_NOTEBLOCK_POSITION);