--- net/minecraft/client/gui/spectator/PlayerMenuItem.java
+++ net/minecraft/client/gui/spectator/PlayerMenuItem.java
@@ -17,7 +17,7 @@
         this.name = net.minecraft.network.chat.Component.literal((java.lang.String)$$0.getName());
     }
 
-    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu menu) {
+    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu $$0) {
         net.minecraft.client.Minecraft.getInstance().getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundTeleportToEntityPacket(this.profile.getId()));
     }
 