--- net/minecraft/client/gui/spectator/SpectatorMenu.java
+++ net/minecraft/client/gui/spectator/SpectatorMenu.java
@@ -25,24 +25,24 @@
         this.listener = $$0;
     }
 
-    public net.minecraft.client.gui.spectator.SpectatorMenuItem getItem(int index) {
-        int n = index + this.page * 6;
-        if (this.page > 0 && index == 0) {
+    public net.minecraft.client.gui.spectator.SpectatorMenuItem getItem(int $$0) {
+        int $$1 = $$0 + this.page * 6;
+        if (this.page > 0 && $$0 == 0) {
             return SCROLL_LEFT;
         }
-        if (index == 7) {
-            if (n < this.category.getItems().size()) {
+        if ($$0 == 7) {
+            if ($$1 < this.category.getItems().size()) {
                 return SCROLL_RIGHT_ENABLED;
             }
             return SCROLL_RIGHT_DISABLED;
         }
-        if (index == 8) {
+        if ($$0 == 8) {
             return CLOSE_ITEM;
         }
-        if (n < 0 || n >= this.category.getItems().size()) {
+        if ($$1 < 0 || $$1 >= this.category.getItems().size()) {
             return EMPTY_SLOT;
         }
-        return (net.minecraft.client.gui.spectator.SpectatorMenuItem)com.google.common.base.MoreObjects.firstNonNull((java.lang.Object)((net.minecraft.client.gui.spectator.SpectatorMenuItem)this.category.getItems().get(n)), (java.lang.Object)EMPTY_SLOT);
+        return (net.minecraft.client.gui.spectator.SpectatorMenuItem)com.google.common.base.MoreObjects.firstNonNull((java.lang.Object)((net.minecraft.client.gui.spectator.SpectatorMenuItem)this.category.getItems().get($$1)), (java.lang.Object)EMPTY_SLOT);
     }
 
     public java.util.List<net.minecraft.client.gui.spectator.SpectatorMenuItem> getItems() {
@@ -61,13 +61,13 @@
         return this.category;
     }
 
-    public void selectSlot(int slot) {
-        net.minecraft.client.gui.spectator.SpectatorMenuItem spectatorMenuItem = this.getItem(slot);
-        if (spectatorMenuItem != EMPTY_SLOT) {
-            if (this.selectedSlot == slot && spectatorMenuItem.isEnabled()) {
-                spectatorMenuItem.selectItem(this);
+    public void selectSlot(int $$0) {
+        net.minecraft.client.gui.spectator.SpectatorMenuItem $$1 = this.getItem($$0);
+        if ($$1 != EMPTY_SLOT) {
+            if (this.selectedSlot == $$0 && $$1.isEnabled()) {
+                $$1.selectItem(this);
             } else {
-                this.selectedSlot = slot;
+                this.selectedSlot = $$0;
             }
         }
     }
@@ -80,8 +80,8 @@
         return this.selectedSlot;
     }
 
-    public void selectCategory(net.minecraft.client.gui.spectator.SpectatorMenuCategory menuView) {
-        this.category = menuView;
+    public void selectCategory(net.minecraft.client.gui.spectator.SpectatorMenuCategory $$0) {
+        this.category = $$0;
         this.selectedSlot = -1;
         this.page = 0;
     }