--- net/minecraft/client/gui/spectator/SpectatorMenu$1.java
+++ net/minecraft/client/gui/spectator/SpectatorMenu$1.java
@@ -9,7 +9,7 @@
     public net.minecraft.client.gui.spectator.SpectatorMenu$1() {
     }
 
-    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu menu) {
+    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu $$0) {
     }
 
     public net.minecraft.network.chat.Component getName() {