--- net/minecraft/client/gui/spectator/SpectatorMenu$CloseSpectatorItem.java
+++ net/minecraft/client/gui/spectator/SpectatorMenu$CloseSpectatorItem.java
@@ -9,8 +9,8 @@
     public net.minecraft.client.gui.spectator.SpectatorMenu$CloseSpectatorItem() {
     }
 
-    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu menu) {
-        menu.exit();
+    public void selectItem(net.minecraft.client.gui.spectator.SpectatorMenu $$0) {
+        $$0.exit();
     }
 
     public net.minecraft.network.chat.Component getName() {