--- net/minecraft/server/packs/resources/FallbackResourceManager$EntryStack.java
+++ net/minecraft/server/packs/resources/FallbackResourceManager$EntryStack.java
@@ -6,36 +6,48 @@
 public class net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack
 extends java.lang.Record
  {
+    public net.minecraft.resources.ResourceLocation fileLocation;
     public net.minecraft.resources.ResourceLocation metadataLocation;
-    public java.util.List<net.minecraft.server.packs.resources.FallbackResourceManager$SinglePackResourceThunkSupplier> entries;
+    public java.util.List<net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource> fileSources;
+    public java.util.Map<net.minecraft.server.packs.PackResources, net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream>> metaSources;
 
-    public net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack(net.minecraft.resources.ResourceLocation $$0, java.util.List<net.minecraft.server.packs.resources.FallbackResourceManager$SinglePackResourceThunkSupplier> $$1) {
-        this.metadataLocation = $$0;
-        this.entries = $$1;
+    public net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack(net.minecraft.resources.ResourceLocation $$0) {
+        this($$0, net.minecraft.server.packs.resources.FallbackResourceManager.getMetadataLocation((net.minecraft.resources.ResourceLocation)$$0), (java.util.List)new java.util.ArrayList(), (java.util.Map)new it.unimi.dsi.fastutil.objects.Object2ObjectArrayMap());
     }
 
-    public java.util.List<net.minecraft.server.packs.resources.Resource> createThunks() {
-        return this.entries().stream().map(net.minecraft.server.packs.resources.FallbackResourceManager$SinglePackResourceThunkSupplier::create).toList();
+    public net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack(net.minecraft.resources.ResourceLocation $$0, net.minecraft.resources.ResourceLocation $$1, java.util.List<net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource> $$2, java.util.Map<net.minecraft.server.packs.PackResources, net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream>> $$3) {
+        this.fileLocation = $$0;
+        this.metadataLocation = $$1;
+        this.fileSources = $$2;
+        this.metaSources = $$3;
     }
 
     public java.lang.String toString() {
-        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack.class, "metadataLocation;entries", "metadataLocation", "entries"}, this);
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack.class, "fileLocation;metadataLocation;fileSources;metaSources", "fileLocation", "metadataLocation", "fileSources", "metaSources"}, this);
     }
 
     public int hashCode() {
-        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack.class, "metadataLocation;entries", "metadataLocation", "entries"}, this);
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack.class, "fileLocation;metadataLocation;fileSources;metaSources", "fileLocation", "metadataLocation", "fileSources", "metaSources"}, this);
     }
 
     public boolean equals(java.lang.Object $$0) {
-        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack.class, "metadataLocation;entries", "metadataLocation", "entries"}, this, $$0);
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$EntryStack.class, "fileLocation;metadataLocation;fileSources;metaSources", "fileLocation", "metadataLocation", "fileSources", "metaSources"}, this, $$0);
     }
 
+    public net.minecraft.resources.ResourceLocation fileLocation() {
+        return this.fileLocation;
+    }
+
     public net.minecraft.resources.ResourceLocation metadataLocation() {
         return this.metadataLocation;
     }
 
-    public java.util.List<net.minecraft.server.packs.resources.FallbackResourceManager$SinglePackResourceThunkSupplier> entries() {
-        return this.entries;
+    public java.util.List<net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource> fileSources() {
+        return this.fileSources;
     }
+
+    public java.util.Map<net.minecraft.server.packs.PackResources, net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream>> metaSources() {
+        return this.metaSources;
+    }
 }
 