--- net/minecraft/server/packs/resources/FallbackResourceManager$1ResourceWithSourceAndIndex.java
+++ net/minecraft/server/packs/resources/FallbackResourceManager$1ResourceWithSourceAndIndex.java
@@ -1,0 +1,43 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.resources;
+
+public class net.minecraft.server.packs.resources.FallbackResourceManager$1ResourceWithSourceAndIndex
+extends java.lang.Record
+ {
+    public net.minecraft.server.packs.PackResources packResources;
+    public net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream> resource;
+    public int packIndex;
+
+    public net.minecraft.server.packs.resources.FallbackResourceManager$1ResourceWithSourceAndIndex(net.minecraft.server.packs.PackResources $$0, net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream> $$1, int $$2) {
+        this.packResources = $$0;
+        this.resource = $$1;
+        this.packIndex = $$2;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$1ResourceWithSourceAndIndex.class, "packResources;resource;packIndex", "packResources", "resource", "packIndex"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$1ResourceWithSourceAndIndex.class, "packResources;resource;packIndex", "packResources", "resource", "packIndex"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$1ResourceWithSourceAndIndex.class, "packResources;resource;packIndex", "packResources", "resource", "packIndex"}, this, $$0);
+    }
+
+    public net.minecraft.server.packs.PackResources packResources() {
+        return this.packResources;
+    }
+
+    public net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream> resource() {
+        return this.resource;
+    }
+
+    public int packIndex() {
+        return this.packIndex;
+    }
+}
+