--- net/minecraft/server/packs/resources/FallbackResourceManager$ResourceWithSource.java
+++ net/minecraft/server/packs/resources/FallbackResourceManager$ResourceWithSource.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.resources;
+
+public class net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource
+extends java.lang.Record
+ {
+    public net.minecraft.server.packs.PackResources source;
+    public net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream> resource;
+
+    public net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource(net.minecraft.server.packs.PackResources $$0, net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream> $$1) {
+        this.source = $$0;
+        this.resource = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource.class, "source;resource", "source", "resource"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource.class, "source;resource", "source", "resource"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.resources.FallbackResourceManager$ResourceWithSource.class, "source;resource", "source", "resource"}, this, $$0);
+    }
+
+    public net.minecraft.server.packs.PackResources source() {
+        return this.source;
+    }
+
+    public net.minecraft.server.packs.resources.IoSupplier<java.io.InputStream> resource() {
+        return this.resource;
+    }
+}
+