--- net/minecraft/server/packs/resources/SimpleReloadInstance.java
+++ net/minecraft/server/packs/resources/SimpleReloadInstance.java
@@ -18,8 +18,8 @@
     public java.util.concurrent.atomic.AtomicInteger startedTaskCounter;
     public java.util.concurrent.atomic.AtomicInteger doneTaskCounter;
 
-    public static net.minecraft.server.packs.resources.SimpleReloadInstance<java.lang.Void> of(net.minecraft.server.packs.resources.ResourceManager resourceManager, java.util.List<net.minecraft.server.packs.resources.PreparableReloadListener> listeners, java.util.concurrent.Executor backgroundExecutor, java.util.concurrent.Executor gameExecutor, java.util.concurrent.CompletableFuture<net.minecraft.util.Unit> alsoWaitedFor) {
-        return new net.minecraft.server.packs.resources.SimpleReloadInstance(backgroundExecutor, gameExecutor, resourceManager, listeners, ($$1, $$2, $$3, $$4, $$5) -> $$3.reload($$1, $$2, (net.minecraft.util.profiling.ProfilerFiller)net.minecraft.util.profiling.InactiveProfiler.INSTANCE, (net.minecraft.util.profiling.ProfilerFiller)net.minecraft.util.profiling.InactiveProfiler.INSTANCE, backgroundExecutor, $$5), alsoWaitedFor);
+    public static net.minecraft.server.packs.resources.SimpleReloadInstance<java.lang.Void> of(net.minecraft.server.packs.resources.ResourceManager $$0, java.util.List<net.minecraft.server.packs.resources.PreparableReloadListener> $$12, java.util.concurrent.Executor $$22, java.util.concurrent.Executor $$32, java.util.concurrent.CompletableFuture<net.minecraft.util.Unit> $$42) {
+        return new net.minecraft.server.packs.resources.SimpleReloadInstance($$22, $$32, $$0, $$12, ($$1, $$2, $$3, $$4, $$5) -> $$3.reload($$1, $$2, (net.minecraft.util.profiling.ProfilerFiller)net.minecraft.util.profiling.InactiveProfiler.INSTANCE, (net.minecraft.util.profiling.ProfilerFiller)net.minecraft.util.profiling.InactiveProfiler.INSTANCE, $$22, $$5), $$42);
     }
 
     public net.minecraft.server.packs.resources.SimpleReloadInstance(java.util.concurrent.Executor $$0, java.util.concurrent.Executor $$12, net.minecraft.server.packs.resources.ResourceManager $$2, java.util.List<net.minecraft.server.packs.resources.PreparableReloadListener> $$3, net.minecraft.server.packs.resources.SimpleReloadInstance$StateFactory<S> $$4, java.util.concurrent.CompletableFuture<net.minecraft.util.Unit> $$5) {