--- net/minecraft/server/packs/resources/SimpleReloadInstance$1.java
+++ net/minecraft/server/packs/resources/SimpleReloadInstance$1.java
@@ -18,14 +18,14 @@
         this.val$previousTask = completableFuture;
     }
 
-    public <T> java.util.concurrent.CompletableFuture<T> wait(T backgroundResult) {
+    public <T> java.util.concurrent.CompletableFuture<T> wait(T $$0) {
         this.val$mainThreadExecutor.execute(() -> {
             this.this$0.preparingListeners.remove((java.lang.Object)this.val$listener);
             if (this.this$0.preparingListeners.isEmpty()) {
                 this.this$0.allPreparations.complete((java.lang.Object)net.minecraft.util.Unit.INSTANCE);
             }
         });
-        return this.this$0.allPreparations.thenCombine((java.util.concurrent.CompletionStage)this.val$previousTask, ($$1, $$2) -> backgroundResult);
+        return this.this$0.allPreparations.thenCombine((java.util.concurrent.CompletionStage)this.val$previousTask, ($$1, $$2) -> $$0);
     }
 }
 