--- net/minecraft/server/rcon/thread/RconClient.java
+++ net/minecraft/server/rcon/thread/RconClient.java
@@ -96,17 +96,17 @@
         }
     }
 
-    public void send(int n, int n2, java.lang.String message) throws java.io.IOException {
-        java.io.ByteArrayOutputStream byteArrayOutputStream = new java.io.ByteArrayOutputStream(1248);
-        java.io.DataOutputStream dataOutputStream = new java.io.DataOutputStream((java.io.OutputStream)byteArrayOutputStream);
-        byte[] byArray = message.getBytes(java.nio.charset.StandardCharsets.UTF_8);
-        dataOutputStream.writeInt(java.lang.Integer.reverseBytes((int)(byArray.length + 10)));
-        dataOutputStream.writeInt(java.lang.Integer.reverseBytes((int)n));
-        dataOutputStream.writeInt(java.lang.Integer.reverseBytes((int)n2));
-        dataOutputStream.write(byArray);
-        dataOutputStream.write(0);
-        dataOutputStream.write(0);
-        this.client.getOutputStream().write(byteArrayOutputStream.toByteArray());
+    public void send(int $$0, int $$1, java.lang.String $$2) throws java.io.IOException {
+        java.io.ByteArrayOutputStream $$3 = new java.io.ByteArrayOutputStream(1248);
+        java.io.DataOutputStream $$4 = new java.io.DataOutputStream((java.io.OutputStream)$$3);
+        byte[] $$5 = $$2.getBytes(java.nio.charset.StandardCharsets.UTF_8);
+        $$4.writeInt(java.lang.Integer.reverseBytes((int)($$5.length + 10)));
+        $$4.writeInt(java.lang.Integer.reverseBytes((int)$$0));
+        $$4.writeInt(java.lang.Integer.reverseBytes((int)$$1));
+        $$4.write($$5);
+        $$4.write(0);
+        $$4.write(0);
+        this.client.getOutputStream().write($$3.toByteArray());
     }
 
     public void sendAuthFailure() throws java.io.IOException {