--- net/minecraft/client/resources/metadata/animation/AnimationFrame.java
+++ net/minecraft/client/resources/metadata/animation/AnimationFrame.java
@@ -9,13 +9,13 @@
     public int index;
     public int time;
 
-    public net.minecraft.client.resources.metadata.animation.AnimationFrame(int index) {
-        this(index, -1);
+    public net.minecraft.client.resources.metadata.animation.AnimationFrame(int $$0) {
+        this($$0, -1);
     }
 
-    public net.minecraft.client.resources.metadata.animation.AnimationFrame(int index, int time) {
-        this.index = index;
-        this.time = time;
+    public net.minecraft.client.resources.metadata.animation.AnimationFrame(int $$0, int $$1) {
+        this.index = $$0;
+        this.time = $$1;
     }
 
     public int getTime(int $$0) {