--- net/minecraft/client/resources/metadata/animation/AnimationMetadataSection$1.java
+++ net/minecraft/client/resources/metadata/animation/AnimationMetadataSection$1.java
@@ -10,8 +10,8 @@
         super($$0, $$1, $$2, $$3, $$4);
     }
 
-    public com.mojang.datafixers.util.Pair<java.lang.Integer, java.lang.Integer> getFrameSize(int width, int height) {
-        return com.mojang.datafixers.util.Pair.of((java.lang.Object)width, (java.lang.Object)height);
+    public net.minecraft.client.resources.metadata.animation.FrameSize calculateFrameSize(int $$0, int $$1) {
+        return new net.minecraft.client.resources.metadata.animation.FrameSize($$0, $$1);
     }
 }
 