--- net/minecraft/client/resources/metadata/animation/FrameSize.java
+++ net/minecraft/client/resources/metadata/animation/FrameSize.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.resources.metadata.animation;
+
+public class net.minecraft.client.resources.metadata.animation.FrameSize
+extends java.lang.Record
+ {
+    public int width;
+    public int height;
+
+    public net.minecraft.client.resources.metadata.animation.FrameSize(int $$0, int $$1) {
+        this.width = $$0;
+        this.height = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.resources.metadata.animation.FrameSize.class, "width;height", "width", "height"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.resources.metadata.animation.FrameSize.class, "width;height", "width", "height"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.resources.metadata.animation.FrameSize.class, "width;height", "width", "height"}, this, $$0);
+    }
+
+    public int width() {
+        return this.width;
+    }
+
+    public int height() {
+        return this.height;
+    }
+}
+