--- net/minecraft/client/resources/metadata/animation/VillagerMetaDataSection$Hat.java
+++ net/minecraft/client/resources/metadata/animation/VillagerMetaDataSection$Hat.java
@@ -21,17 +21,17 @@
         return (net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat)java.lang.Enum.valueOf(net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat(java.lang.String name) {
+    public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
+        this.name = $$0;
     }
 
     public java.lang.String getName() {
         return this.name;
     }
 
-    public static net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat getByName(java.lang.String name) {
-        return (net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat)BY_NAME.getOrDefault((java.lang.Object)name, (java.lang.Object)NONE);
+    public static net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat getByName(java.lang.String $$0) {
+        return (net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat)BY_NAME.getOrDefault((java.lang.Object)$$0, (java.lang.Object)NONE);
     }
 
     public static /* synthetic */ net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat lambda$static$0(net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat $$0) {