--- net/minecraft/client/resources/metadata/animation/VillagerMetaDataSection.java
+++ net/minecraft/client/resources/metadata/animation/VillagerMetaDataSection.java
@@ -9,8 +9,8 @@
     public static java.lang.String SECTION_NAME = "villager";
     public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat hat;
 
-    public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection(net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat hat) {
-        this.hat = hat;
+    public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection(net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat $$0) {
+        this.hat = $$0;
     }
 
     public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat getHat() {