--- net/minecraft/client/resources/metadata/animation/VillagerMetadataSectionSerializer.java
+++ net/minecraft/client/resources/metadata/animation/VillagerMetadataSectionSerializer.java
@@ -9,8 +9,8 @@
     public net.minecraft.client.resources.metadata.animation.VillagerMetadataSectionSerializer() {
     }
 
-    public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection fromJson(com.google.gson.JsonObject json) {
-        return new net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection(net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat.getByName((java.lang.String)net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)json, (java.lang.String)"hat", (java.lang.String)"none")));
+    public net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection fromJson(com.google.gson.JsonObject $$0) {
+        return new net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection(net.minecraft.client.resources.metadata.animation.VillagerMetaDataSection$Hat.getByName((java.lang.String)net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"hat", (java.lang.String)"none")));
     }
 
     public java.lang.String getMetadataSectionName() {