--- net/minecraft/client/resources/sounds/AbstractSoundInstance.java
+++ net/minecraft/client/resources/sounds/AbstractSoundInstance.java
@@ -37,10 +37,10 @@
         return this.location;
     }
 
-    public net.minecraft.client.sounds.WeighedSoundEvents resolve(net.minecraft.client.sounds.SoundManager handler) {
-        net.minecraft.client.sounds.WeighedSoundEvents weighedSoundEvents = handler.getSoundEvent(this.location);
-        this.sound = weighedSoundEvents == null ? net.minecraft.client.sounds.SoundManager.EMPTY_SOUND : weighedSoundEvents.getSound(this.random);
-        return weighedSoundEvents;
+    public net.minecraft.client.sounds.WeighedSoundEvents resolve(net.minecraft.client.sounds.SoundManager $$0) {
+        net.minecraft.client.sounds.WeighedSoundEvents $$1 = $$0.getSoundEvent(this.location);
+        this.sound = $$1 == null ? net.minecraft.client.sounds.SoundManager.EMPTY_SOUND : $$1.getSound(this.random);
+        return $$1;
     }
 
     public net.minecraft.client.resources.sounds.Sound getSound() {