--- net/minecraft/client/resources/sounds/SimpleSoundInstance.java
+++ net/minecraft/client/resources/sounds/SimpleSoundInstance.java
@@ -10,28 +10,28 @@
         this($$0, $$1, $$2, $$3, $$4, (double)$$5.getX() + 0.5, (double)$$5.getY() + 0.5, (double)$$5.getZ() + 0.5);
     }
 
-    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forUI(net.minecraft.sounds.SoundEvent sound, float pitch) {
-        return net.minecraft.client.resources.sounds.SimpleSoundInstance.forUI((net.minecraft.sounds.SoundEvent)sound, (float)pitch, (float)0.25f);
+    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forUI(net.minecraft.sounds.SoundEvent $$0, float $$1) {
+        return net.minecraft.client.resources.sounds.SimpleSoundInstance.forUI((net.minecraft.sounds.SoundEvent)$$0, (float)$$1, (float)0.25f);
     }
 
-    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forUI(net.minecraft.sounds.SoundEvent sound, float pitch, float volume) {
-        return new net.minecraft.client.resources.sounds.SimpleSoundInstance(sound.getLocation(), net.minecraft.sounds.SoundSource.MASTER, volume, pitch, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.NONE, 0.0, 0.0, 0.0, true);
+    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forUI(net.minecraft.sounds.SoundEvent $$0, float $$1, float $$2) {
+        return new net.minecraft.client.resources.sounds.SimpleSoundInstance($$0.getLocation(), net.minecraft.sounds.SoundSource.MASTER, $$2, $$1, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.NONE, 0.0, 0.0, 0.0, true);
     }
 
-    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forMusic(net.minecraft.sounds.SoundEvent sound) {
-        return new net.minecraft.client.resources.sounds.SimpleSoundInstance(sound.getLocation(), net.minecraft.sounds.SoundSource.MUSIC, 1.0f, 1.0f, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.NONE, 0.0, 0.0, 0.0, true);
+    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forMusic(net.minecraft.sounds.SoundEvent $$0) {
+        return new net.minecraft.client.resources.sounds.SimpleSoundInstance($$0.getLocation(), net.minecraft.sounds.SoundSource.MUSIC, 1.0f, 1.0f, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.NONE, 0.0, 0.0, 0.0, true);
     }
 
-    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forRecord(net.minecraft.sounds.SoundEvent sound, double x, double y, double z) {
-        return new net.minecraft.client.resources.sounds.SimpleSoundInstance(sound, net.minecraft.sounds.SoundSource.RECORDS, 4.0f, 1.0f, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.LINEAR, x, y, z);
+    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forRecord(net.minecraft.sounds.SoundEvent $$0, net.minecraft.world.phys.Vec3 $$1) {
+        return new net.minecraft.client.resources.sounds.SimpleSoundInstance($$0, net.minecraft.sounds.SoundSource.RECORDS, 4.0f, 1.0f, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.LINEAR, $$1.x, $$1.y, $$1.z);
     }
 
-    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forLocalAmbience(net.minecraft.sounds.SoundEvent sound, float volume, float pitch) {
-        return new net.minecraft.client.resources.sounds.SimpleSoundInstance(sound.getLocation(), net.minecraft.sounds.SoundSource.AMBIENT, pitch, volume, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.NONE, 0.0, 0.0, 0.0, true);
+    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forLocalAmbience(net.minecraft.sounds.SoundEvent $$0, float $$1, float $$2) {
+        return new net.minecraft.client.resources.sounds.SimpleSoundInstance($$0.getLocation(), net.minecraft.sounds.SoundSource.AMBIENT, $$2, $$1, net.minecraft.client.resources.sounds.SoundInstance.createUnseededRandom(), false, 0, net.minecraft.client.resources.sounds.SoundInstance$Attenuation.NONE, 0.0, 0.0, 0.0, true);
     }
 
-    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forAmbientAddition(net.minecraft.sounds.SoundEvent sound) {
-        return net.minecraft.client.resources.sounds.SimpleSoundInstance.forLocalAmbience((net.minecraft.sounds.SoundEvent)sound, (float)1.0f, (float)1.0f);
+    public static net.minecraft.client.resources.sounds.SimpleSoundInstance forAmbientAddition(net.minecraft.sounds.SoundEvent $$0) {
+        return net.minecraft.client.resources.sounds.SimpleSoundInstance.forLocalAmbience((net.minecraft.sounds.SoundEvent)$$0, (float)1.0f, (float)1.0f);
     }
 
     public static net.minecraft.client.resources.sounds.SimpleSoundInstance forAmbientMood(net.minecraft.sounds.SoundEvent $$0, net.minecraft.util.RandomSource $$1, double $$2, double $$3, double $$4) {