--- net/minecraft/client/resources/sounds/Sound$Type.java
+++ net/minecraft/client/resources/sounds/Sound$Type.java
@@ -25,10 +25,10 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.client.resources.sounds.Sound$Type getByName(java.lang.String name) {
-        for (net.minecraft.client.resources.sounds.Sound$Type type : net.minecraft.client.resources.sounds.Sound$Type.values()) {
-            if (!type.name.equals((java.lang.Object)name)) continue;
-            return type;
+    public static net.minecraft.client.resources.sounds.Sound$Type getByName(java.lang.String $$0) {
+        for (net.minecraft.client.resources.sounds.Sound$Type $$1 : net.minecraft.client.resources.sounds.Sound$Type.values()) {
+            if (!$$1.name.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
         return null;
     }