--- net/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java
+++ net/minecraft/world/level/levelgen/structure/PoolElementStructurePiece.java
@@ -62,9 +62,9 @@
         this.element.place(this.structureTemplateManager, $$0, $$1, $$2, this.position, $$5, this.rotation, $$4, $$3, $$6);
     }
 
-    public void move(int x, int y, int z) {
-        super.move(x, y, z);
-        this.position = this.position.offset(x, y, z);
+    public void move(int $$0, int $$1, int $$2) {
+        super.move($$0, $$1, $$2);
+        this.position = this.position.offset($$0, $$1, $$2);
     }
 
     public net.minecraft.world.level.block.Rotation getRotation() {