--- net/minecraft/world/level/levelgen/structure/StructureFeatureIndexSavedData.java
+++ net/minecraft/world/level/levelgen/structure/StructureFeatureIndexSavedData.java
@@ -24,27 +24,27 @@
         return new net.minecraft.world.level.levelgen.structure.StructureFeatureIndexSavedData((it.unimi.dsi.fastutil.longs.LongSet)new it.unimi.dsi.fastutil.longs.LongOpenHashSet($$0.getLongArray("All")), (it.unimi.dsi.fastutil.longs.LongSet)new it.unimi.dsi.fastutil.longs.LongOpenHashSet($$0.getLongArray("Remaining")));
     }
 
-    public net.minecraft.nbt.CompoundTag save(net.minecraft.nbt.CompoundTag compound) {
-        compound.putLongArray("All", this.all.toLongArray());
-        compound.putLongArray("Remaining", this.remaining.toLongArray());
-        return compound;
+    public net.minecraft.nbt.CompoundTag save(net.minecraft.nbt.CompoundTag $$0) {
+        $$0.putLongArray("All", this.all.toLongArray());
+        $$0.putLongArray("Remaining", this.remaining.toLongArray());
+        return $$0;
     }
 
-    public void addIndex(long chunkPos) {
-        this.all.add(chunkPos);
-        this.remaining.add(chunkPos);
+    public void addIndex(long $$0) {
+        this.all.add($$0);
+        this.remaining.add($$0);
     }
 
-    public boolean hasStartIndex(long chunkPos) {
-        return this.all.contains(chunkPos);
+    public boolean hasStartIndex(long $$0) {
+        return this.all.contains($$0);
     }
 
-    public boolean hasUnhandledIndex(long chunkPos) {
-        return this.remaining.contains(chunkPos);
+    public boolean hasUnhandledIndex(long $$0) {
+        return this.remaining.contains($$0);
     }
 
-    public void removeIndex(long chunkPos) {
-        this.remaining.remove(chunkPos);
+    public void removeIndex(long $$0) {
+        this.remaining.remove($$0);
     }
 
     public it.unimi.dsi.fastutil.longs.LongSet getAll() {