--- net/minecraft/world/level/levelgen/structure/StructureCheck.java
+++ net/minecraft/world/level/levelgen/structure/StructureCheck.java
@@ -35,8 +35,8 @@
         this.biomeSource = $$7;
         this.seed = $$8;
         this.fixerUpper = $$9;
-        this.biomes = $$1.ownedRegistryOrThrow(net.minecraft.core.Registry.BIOME_REGISTRY);
-        this.structureConfigs = $$1.ownedRegistryOrThrow(net.minecraft.core.Registry.STRUCTURE_REGISTRY);
+        this.biomes = $$1.registryOrThrow(net.minecraft.core.Registry.BIOME_REGISTRY);
+        this.structureConfigs = $$1.registryOrThrow(net.minecraft.core.Registry.STRUCTURE_REGISTRY);
     }
 
     public net.minecraft.world.level.levelgen.structure.StructureCheckResult checkStart(net.minecraft.world.level.ChunkPos $$02, net.minecraft.world.level.levelgen.structure.Structure $$1, boolean $$22) {