--- net/minecraft/world/level/block/state/BlockBehaviour$BlockStateBase$Cache.java
+++ net/minecraft/world/level/block/state/BlockBehaviour$BlockStateBase$Cache.java
@@ -48,12 +48,12 @@
         this.isCollisionShapeFullBlock = net.minecraft.world.level.block.Block.isShapeFullBlock((net.minecraft.world.phys.shapes.VoxelShape)$$02.getCollisionShape((net.minecraft.world.level.BlockGetter)net.minecraft.world.level.EmptyBlockGetter.INSTANCE, net.minecraft.core.BlockPos.ZERO));
     }
 
-    public boolean isFaceSturdy(net.minecraft.core.Direction direction, net.minecraft.world.level.block.SupportType supportType) {
-        return this.faceSturdy[net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase$Cache.getFaceSupportIndex((net.minecraft.core.Direction)direction, (net.minecraft.world.level.block.SupportType)supportType)];
+    public boolean isFaceSturdy(net.minecraft.core.Direction $$0, net.minecraft.world.level.block.SupportType $$1) {
+        return this.faceSturdy[net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase$Cache.getFaceSupportIndex((net.minecraft.core.Direction)$$0, (net.minecraft.world.level.block.SupportType)$$1)];
     }
 
-    public static int getFaceSupportIndex(net.minecraft.core.Direction direction, net.minecraft.world.level.block.SupportType supportType) {
-        return direction.ordinal() * SUPPORT_TYPE_COUNT + supportType.ordinal();
+    public static int getFaceSupportIndex(net.minecraft.core.Direction $$0, net.minecraft.world.level.block.SupportType $$1) {
+        return $$0.ordinal() * SUPPORT_TYPE_COUNT + $$1.ordinal();
     }
 
     static {