--- net/minecraft/server/commands/BanListCommands.java
+++ net/minecraft/server/commands/BanListCommands.java
@@ -8,23 +8,23 @@
     public net.minecraft.server.commands.BanListCommands() {
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        dispatcher.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"banlist").requires($$0 -> $$0.hasPermission(3))).executes($$0 -> {
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$02) {
+        $$02.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"banlist").requires($$0 -> $$0.hasPermission(3))).executes($$0 -> {
             net.minecraft.server.players.PlayerList $$1 = ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().getPlayerList();
             return net.minecraft.server.commands.BanListCommands.showList((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$0.getSource()), (java.util.Collection)com.google.common.collect.Lists.newArrayList((java.lang.Iterable)com.google.common.collect.Iterables.concat((java.lang.Iterable)$$1.getBans().getEntries(), (java.lang.Iterable)$$1.getIpBans().getEntries())));
         })).then(net.minecraft.commands.Commands.literal((java.lang.String)"ips").executes($$0 -> net.minecraft.server.commands.BanListCommands.showList((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$0.getSource()), (java.util.Collection)((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().getPlayerList().getIpBans().getEntries())))).then(net.minecraft.commands.Commands.literal((java.lang.String)"players").executes($$0 -> net.minecraft.server.commands.BanListCommands.showList((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$0.getSource()), (java.util.Collection)((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().getPlayerList().getBans().getEntries()))));
     }
 
-    public static int showList(net.minecraft.commands.CommandSourceStack source, java.util.Collection<? extends net.minecraft.server.players.BanListEntry<?>> bannedPlayerList) {
-        if (bannedPlayerList.isEmpty()) {
-            source.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.banlist.none"), false);
+    public static int showList(net.minecraft.commands.CommandSourceStack $$0, java.util.Collection<? extends net.minecraft.server.players.BanListEntry<?>> $$1) {
+        if ($$1.isEmpty()) {
+            $$0.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.banlist.none"), false);
         } else {
-            source.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.banlist.list", (java.lang.Object[])new java.lang.Object[]{bannedPlayerList.size()}), false);
-            for (net.minecraft.server.players.BanListEntry banListEntry : bannedPlayerList) {
-                source.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.banlist.entry", (java.lang.Object[])new java.lang.Object[]{banListEntry.getDisplayName(), banListEntry.getSource(), banListEntry.getReason()}), false);
+            $$0.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.banlist.list", (java.lang.Object[])new java.lang.Object[]{$$1.size()}), false);
+            for (net.minecraft.server.players.BanListEntry $$2 : $$1) {
+                $$0.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.banlist.entry", (java.lang.Object[])new java.lang.Object[]{$$2.getDisplayName(), $$2.getSource(), $$2.getReason()}), false);
             }
         }
-        return bannedPlayerList.size();
+        return $$1.size();
     }
 }
 