--- net/minecraft/server/commands/DefaultGameModeCommands.java
+++ net/minecraft/server/commands/DefaultGameModeCommands.java
@@ -8,27 +8,27 @@
     public net.minecraft.server.commands.DefaultGameModeCommands() {
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        com.mojang.brigadier.builder.LiteralArgumentBuilder literalArgumentBuilder = (com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"defaultgamemode").requires($$0 -> $$0.hasPermission(2));
-        for (net.minecraft.world.level.GameType gameType : net.minecraft.world.level.GameType.values()) {
-            literalArgumentBuilder.then(net.minecraft.commands.Commands.literal((java.lang.String)gameType.getName()).executes($$1 -> net.minecraft.server.commands.DefaultGameModeCommands.setMode((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$1.getSource()), (net.minecraft.world.level.GameType)gameType)));
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$02) {
+        com.mojang.brigadier.builder.LiteralArgumentBuilder $$12 = (com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"defaultgamemode").requires($$0 -> $$0.hasPermission(2));
+        for (net.minecraft.world.level.GameType $$2 : net.minecraft.world.level.GameType.values()) {
+            $$12.then(net.minecraft.commands.Commands.literal((java.lang.String)$$2.getName()).executes($$1 -> net.minecraft.server.commands.DefaultGameModeCommands.setMode((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$1.getSource()), (net.minecraft.world.level.GameType)$$2)));
         }
-        dispatcher.register(literalArgumentBuilder);
+        $$02.register($$12);
     }
 
-    public static int setMode(net.minecraft.commands.CommandSourceStack commandSource, net.minecraft.world.level.GameType gamemode) {
-        int n = 0;
-        net.minecraft.server.MinecraftServer minecraftServer = commandSource.getServer();
-        minecraftServer.setDefaultGameType(gamemode);
-        net.minecraft.world.level.GameType gameType = minecraftServer.getForcedGameType();
-        if (gameType != null) {
-            for (net.minecraft.server.level.ServerPlayer serverPlayer : minecraftServer.getPlayerList().getPlayers()) {
-                if (!serverPlayer.setGameMode(gameType)) continue;
-                ++n;
+    public static int setMode(net.minecraft.commands.CommandSourceStack $$0, net.minecraft.world.level.GameType $$1) {
+        int $$2 = 0;
+        net.minecraft.server.MinecraftServer $$3 = $$0.getServer();
+        $$3.setDefaultGameType($$1);
+        net.minecraft.world.level.GameType $$4 = $$3.getForcedGameType();
+        if ($$4 != null) {
+            for (net.minecraft.server.level.ServerPlayer $$5 : $$3.getPlayerList().getPlayers()) {
+                if (!$$5.setGameMode($$4)) continue;
+                ++$$2;
             }
         }
-        commandSource.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.defaultgamemode.success", (java.lang.Object[])new java.lang.Object[]{gamemode.getLongDisplayName()}), true);
-        return n;
+        $$0.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.defaultgamemode.success", (java.lang.Object[])new java.lang.Object[]{$$1.getLongDisplayName()}), true);
+        return $$2;
     }
 }
 