--- net/minecraft/server/commands/EmoteCommands.java
+++ net/minecraft/server/commands/EmoteCommands.java
@@ -8,12 +8,13 @@
     public net.minecraft.server.commands.EmoteCommands() {
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        dispatcher.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"me").then(net.minecraft.commands.Commands.argument((java.lang.String)"action", (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.MessageArgument.message()).executes($$0 -> {
-            net.minecraft.commands.arguments.MessageArgument$ChatMessage $$1 = net.minecraft.commands.arguments.MessageArgument.getChatMessage((com.mojang.brigadier.context.CommandContext)$$0, (java.lang.String)"action");
-            net.minecraft.commands.CommandSourceStack $$22 = (net.minecraft.commands.CommandSourceStack)$$0.getSource();
-            net.minecraft.server.players.PlayerList $$3 = $$22.getServer().getPlayerList();
-            $$1.resolve($$22, $$2 -> $$3.broadcastChatMessage($$2, $$22, net.minecraft.network.chat.ChatType.bind((net.minecraft.resources.ResourceKey)net.minecraft.network.chat.ChatType.EMOTE_COMMAND, (net.minecraft.commands.CommandSourceStack)$$22)));
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$02) {
+        $$02.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"me").then(net.minecraft.commands.Commands.argument((java.lang.String)"action", (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.MessageArgument.message()).executes($$0 -> {
+            net.minecraft.commands.arguments.MessageArgument.resolveChatMessage((com.mojang.brigadier.context.CommandContext)$$0, (java.lang.String)"action", $$1 -> {
+                net.minecraft.commands.CommandSourceStack $$2 = (net.minecraft.commands.CommandSourceStack)$$0.getSource();
+                net.minecraft.server.players.PlayerList $$3 = $$2.getServer().getPlayerList();
+                $$3.broadcastChatMessage($$1, $$2, net.minecraft.network.chat.ChatType.bind((net.minecraft.resources.ResourceKey)net.minecraft.network.chat.ChatType.EMOTE_COMMAND, (net.minecraft.commands.CommandSourceStack)$$2));
+            });
             return 1;
         })));
     }