--- net/minecraft/server/commands/GameRuleCommand.java
+++ net/minecraft/server/commands/GameRuleCommand.java
@@ -8,10 +8,10 @@
     public net.minecraft.server.commands.GameRuleCommand() {
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        com.mojang.brigadier.builder.LiteralArgumentBuilder literalArgumentBuilder = (com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"gamerule").requires($$0 -> $$0.hasPermission(2));
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$02) {
+        com.mojang.brigadier.builder.LiteralArgumentBuilder $$1 = (com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"gamerule").requires($$0 -> $$0.hasPermission(2));
         net.minecraft.world.level.GameRules.visitGameRuleTypes((net.minecraft.world.level.GameRules$GameRuleTypeVisitor)new /* Unavailable Anonymous Inner Class!! */);
-        dispatcher.register(literalArgumentBuilder);
+        $$02.register($$1);
     }
 
     public static <T extends net.minecraft.world.level.GameRules$Value<T>> int setRule(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, net.minecraft.world.level.GameRules$Key<T> $$1) {