--- net/minecraft/server/commands/GameRuleCommand$1.java
+++ net/minecraft/server/commands/GameRuleCommand$1.java
@@ -12,8 +12,8 @@
         this.val$base = literalArgumentBuilder;
     }
 
-    public <T extends net.minecraft.world.level.GameRules$Value<T>> void visit(net.minecraft.world.level.GameRules$Key<T> key, net.minecraft.world.level.GameRules$Type<T> type) {
-        this.val$base.then(((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)key.getId()).executes($$1 -> net.minecraft.server.commands.GameRuleCommand.queryRule((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$1.getSource()), (net.minecraft.world.level.GameRules$Key)key))).then(type.createArgument("value").executes($$1 -> net.minecraft.server.commands.GameRuleCommand.setRule((com.mojang.brigadier.context.CommandContext)$$1, (net.minecraft.world.level.GameRules$Key)key))));
+    public <T extends net.minecraft.world.level.GameRules$Value<T>> void visit(net.minecraft.world.level.GameRules$Key<T> $$0, net.minecraft.world.level.GameRules$Type<T> $$12) {
+        this.val$base.then(((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)$$0.getId()).executes($$1 -> net.minecraft.server.commands.GameRuleCommand.queryRule((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$1.getSource()), (net.minecraft.world.level.GameRules$Key)$$0))).then($$12.createArgument("value").executes($$1 -> net.minecraft.server.commands.GameRuleCommand.setRule((com.mojang.brigadier.context.CommandContext)$$1, (net.minecraft.world.level.GameRules$Key)$$0))));
     }
 }
 