--- net/minecraft/server/commands/HelpCommand.java
+++ net/minecraft/server/commands/HelpCommand.java
@@ -10,19 +10,19 @@
     public net.minecraft.server.commands.HelpCommand() {
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        dispatcher.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"help").executes($$1 -> {
-            java.util.Map $$2 = dispatcher.getSmartUsage((com.mojang.brigadier.tree.CommandNode)dispatcher.getRoot(), (java.lang.Object)((net.minecraft.commands.CommandSourceStack)$$1.getSource()));
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$0) {
+        $$0.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"help").executes($$1 -> {
+            java.util.Map $$2 = $$0.getSmartUsage((com.mojang.brigadier.tree.CommandNode)$$0.getRoot(), (java.lang.Object)((net.minecraft.commands.CommandSourceStack)$$1.getSource()));
             for (java.lang.String $$3 : $$2.values()) {
                 ((net.minecraft.commands.CommandSourceStack)$$1.getSource()).sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)("/" + $$3)), false);
             }
             return $$2.size();
         })).then(net.minecraft.commands.Commands.argument((java.lang.String)"command", (com.mojang.brigadier.arguments.ArgumentType)com.mojang.brigadier.arguments.StringArgumentType.greedyString()).executes($$1 -> {
-            com.mojang.brigadier.ParseResults $$2 = dispatcher.parse(com.mojang.brigadier.arguments.StringArgumentType.getString((com.mojang.brigadier.context.CommandContext)$$1, (java.lang.String)"command"), (java.lang.Object)((net.minecraft.commands.CommandSourceStack)$$1.getSource()));
+            com.mojang.brigadier.ParseResults $$2 = $$0.parse(com.mojang.brigadier.arguments.StringArgumentType.getString((com.mojang.brigadier.context.CommandContext)$$1, (java.lang.String)"command"), (java.lang.Object)((net.minecraft.commands.CommandSourceStack)$$1.getSource()));
             if ($$2.getContext().getNodes().isEmpty()) {
                 throw ERROR_FAILED.create();
             }
-            java.util.Map $$3 = dispatcher.getSmartUsage(((com.mojang.brigadier.context.ParsedCommandNode)com.google.common.collect.Iterables.getLast((java.lang.Iterable)$$2.getContext().getNodes())).getNode(), (java.lang.Object)((net.minecraft.commands.CommandSourceStack)$$1.getSource()));
+            java.util.Map $$3 = $$0.getSmartUsage(((com.mojang.brigadier.context.ParsedCommandNode)com.google.common.collect.Iterables.getLast((java.lang.Iterable)$$2.getContext().getNodes())).getNode(), (java.lang.Object)((net.minecraft.commands.CommandSourceStack)$$1.getSource()));
             for (java.lang.String $$4 : $$3.values()) {
                 ((net.minecraft.commands.CommandSourceStack)$$1.getSource()).sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)("/" + $$2.getReader().getString() + " " + $$4)), false);
             }