--- net/minecraft/server/commands/ReloadCommand.java
+++ net/minecraft/server/commands/ReloadCommand.java
@@ -21,7 +21,7 @@
     public static java.util.Collection<java.lang.String> discoverNewPacks(net.minecraft.server.packs.repository.PackRepository $$0, net.minecraft.world.level.storage.WorldData $$1, java.util.Collection<java.lang.String> $$2) {
         $$0.reload();
         java.util.ArrayList $$3 = com.google.common.collect.Lists.newArrayList($$2);
-        java.util.List $$4 = $$1.getDataPackConfig().getDisabled();
+        java.util.List $$4 = $$1.getDataConfiguration().dataPacks().getDisabled();
         for (java.lang.String $$5 : $$0.getAvailableIds()) {
             if ($$4.contains((java.lang.Object)$$5) || $$3.contains((java.lang.Object)$$5)) continue;
             $$3.add((java.lang.Object)$$5);
@@ -29,8 +29,8 @@
         return $$3;
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        dispatcher.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"reload").requires($$0 -> $$0.hasPermission(2))).executes($$0 -> {
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$02) {
+        $$02.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"reload").requires($$0 -> $$0.hasPermission(2))).executes($$0 -> {
             net.minecraft.commands.CommandSourceStack $$1 = (net.minecraft.commands.CommandSourceStack)$$0.getSource();
             net.minecraft.server.MinecraftServer $$2 = $$1.getServer();
             net.minecraft.server.packs.repository.PackRepository $$3 = $$2.getPackRepository();