--- net/minecraft/server/commands/SaveAllCommand.java
+++ net/minecraft/server/commands/SaveAllCommand.java
@@ -10,18 +10,18 @@
     public net.minecraft.server.commands.SaveAllCommand() {
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        dispatcher.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"save-all").requires($$0 -> $$0.hasPermission(4))).executes($$0 -> net.minecraft.server.commands.SaveAllCommand.saveAll((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$0.getSource()), (boolean)false))).then(net.minecraft.commands.Commands.literal((java.lang.String)"flush").executes($$0 -> net.minecraft.server.commands.SaveAllCommand.saveAll((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$0.getSource()), (boolean)true))));
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$02) {
+        $$02.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"save-all").requires($$0 -> $$0.hasPermission(4))).executes($$0 -> net.minecraft.server.commands.SaveAllCommand.saveAll((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$0.getSource()), (boolean)false))).then(net.minecraft.commands.Commands.literal((java.lang.String)"flush").executes($$0 -> net.minecraft.server.commands.SaveAllCommand.saveAll((net.minecraft.commands.CommandSourceStack)((net.minecraft.commands.CommandSourceStack)$$0.getSource()), (boolean)true))));
     }
 
-    public static int saveAll(net.minecraft.commands.CommandSourceStack source, boolean flush) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        source.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.save.saving"), false);
-        net.minecraft.server.MinecraftServer minecraftServer = source.getServer();
-        boolean bl = minecraftServer.saveEverything(true, flush, true);
-        if (!bl) {
+    public static int saveAll(net.minecraft.commands.CommandSourceStack $$0, boolean $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        $$0.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.save.saving"), false);
+        net.minecraft.server.MinecraftServer $$2 = $$0.getServer();
+        boolean $$3 = $$2.saveEverything(true, $$1, true);
+        if (!$$3) {
             throw ERROR_FAILED.create();
         }
-        source.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.save.success"), true);
+        $$0.sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.save.success"), true);
         return 1;
     }
 