--- net/minecraft/server/commands/StopCommand.java
+++ net/minecraft/server/commands/StopCommand.java
@@ -8,8 +8,8 @@
     public net.minecraft.server.commands.StopCommand() {
     }
 
-    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> dispatcher) {
-        dispatcher.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"stop").requires($$0 -> $$0.hasPermission(4))).executes($$0 -> {
+    public static void register(com.mojang.brigadier.CommandDispatcher<net.minecraft.commands.CommandSourceStack> $$02) {
+        $$02.register((com.mojang.brigadier.builder.LiteralArgumentBuilder)((com.mojang.brigadier.builder.LiteralArgumentBuilder)net.minecraft.commands.Commands.literal((java.lang.String)"stop").requires($$0 -> $$0.hasPermission(4))).executes($$0 -> {
             ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).sendSuccess((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.stop.stopping"), true);
             ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().halt(false);
             return 1;