--- net/minecraft/server/commands/TeleportCommand$LookAt.java
+++ net/minecraft/server/commands/TeleportCommand$LookAt.java
@@ -21,15 +21,15 @@
         this.anchor = null;
     }
 
-    public void perform(net.minecraft.commands.CommandSourceStack source, net.minecraft.world.entity.Entity entity) {
+    public void perform(net.minecraft.commands.CommandSourceStack $$0, net.minecraft.world.entity.Entity $$1) {
         if (this.entity != null) {
-            if (entity instanceof net.minecraft.server.level.ServerPlayer) {
-                ((net.minecraft.server.level.ServerPlayer)entity).lookAt(source.getAnchor(), this.entity, this.anchor);
+            if ($$1 instanceof net.minecraft.server.level.ServerPlayer) {
+                ((net.minecraft.server.level.ServerPlayer)$$1).lookAt($$0.getAnchor(), this.entity, this.anchor);
             } else {
-                entity.lookAt(source.getAnchor(), this.position);
+                $$1.lookAt($$0.getAnchor(), this.position);
             }
         } else {
-            entity.lookAt(source.getAnchor(), this.position);
+            $$1.lookAt($$0.getAnchor(), this.position);
         }
     }
 }