--- net/minecraft/client/renderer/entity/layers/CapeLayer.java
+++ net/minecraft/client/renderer/entity/layers/CapeLayer.java
@@ -10,42 +10,42 @@
         super($$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.client.player.AbstractClientPlayer livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (!livingEntity.isCapeLoaded() || livingEntity.isInvisible() || !livingEntity.isModelPartShown(net.minecraft.world.entity.player.PlayerModelPart.CAPE) || livingEntity.getCloakTextureLocation() == null) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.client.player.AbstractClientPlayer $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        if (!$$3.isCapeLoaded() || $$3.isInvisible() || !$$3.isModelPartShown(net.minecraft.world.entity.player.PlayerModelPart.CAPE) || $$3.getCloakTextureLocation() == null) {
             return;
         }
-        net.minecraft.world.item.ItemStack itemStack = livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.CHEST);
-        if (itemStack.is(net.minecraft.world.item.Items.ELYTRA)) {
+        net.minecraft.world.item.ItemStack $$10 = $$3.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.CHEST);
+        if ($$10.is(net.minecraft.world.item.Items.ELYTRA)) {
             return;
         }
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.0, 0.125);
-        double d = net.minecraft.util.Mth.lerp((double)partialTicks, (double)livingEntity.xCloakO, (double)livingEntity.xCloak) - net.minecraft.util.Mth.lerp((double)partialTicks, (double)livingEntity.xo, (double)livingEntity.getX());
-        double d2 = net.minecraft.util.Mth.lerp((double)partialTicks, (double)livingEntity.yCloakO, (double)livingEntity.yCloak) - net.minecraft.util.Mth.lerp((double)partialTicks, (double)livingEntity.yo, (double)livingEntity.getY());
-        double d3 = net.minecraft.util.Mth.lerp((double)partialTicks, (double)livingEntity.zCloakO, (double)livingEntity.zCloak) - net.minecraft.util.Mth.lerp((double)partialTicks, (double)livingEntity.zo, (double)livingEntity.getZ());
-        float f = livingEntity.yBodyRotO + (livingEntity.yBodyRot - livingEntity.yBodyRotO);
-        double d4 = net.minecraft.util.Mth.sin((float)(f * ((float)java.lang.Math.PI / 180)));
-        double d5 = -net.minecraft.util.Mth.cos((float)(f * ((float)java.lang.Math.PI / 180)));
-        float f2 = (float)d2 * 10.0f;
-        f2 = net.minecraft.util.Mth.clamp((float)f2, (float)-6.0f, (float)32.0f);
-        float f3 = (float)(d * d4 + d3 * d5) * 100.0f;
-        f3 = net.minecraft.util.Mth.clamp((float)f3, (float)0.0f, (float)150.0f);
-        float f4 = (float)(d * d5 - d3 * d4) * 100.0f;
-        f4 = net.minecraft.util.Mth.clamp((float)f4, (float)-20.0f, (float)20.0f);
-        if (f3 < 0.0f) {
-            f3 = 0.0f;
+        $$0.pushPose();
+        $$0.translate(0.0, 0.0, 0.125);
+        double $$11 = net.minecraft.util.Mth.lerp((double)$$6, (double)$$3.xCloakO, (double)$$3.xCloak) - net.minecraft.util.Mth.lerp((double)$$6, (double)$$3.xo, (double)$$3.getX());
+        double $$12 = net.minecraft.util.Mth.lerp((double)$$6, (double)$$3.yCloakO, (double)$$3.yCloak) - net.minecraft.util.Mth.lerp((double)$$6, (double)$$3.yo, (double)$$3.getY());
+        double $$13 = net.minecraft.util.Mth.lerp((double)$$6, (double)$$3.zCloakO, (double)$$3.zCloak) - net.minecraft.util.Mth.lerp((double)$$6, (double)$$3.zo, (double)$$3.getZ());
+        float $$14 = $$3.yBodyRotO + ($$3.yBodyRot - $$3.yBodyRotO);
+        double $$15 = net.minecraft.util.Mth.sin((float)($$14 * ((float)java.lang.Math.PI / 180)));
+        double $$16 = -net.minecraft.util.Mth.cos((float)($$14 * ((float)java.lang.Math.PI / 180)));
+        float $$17 = (float)$$12 * 10.0f;
+        $$17 = net.minecraft.util.Mth.clamp((float)$$17, (float)-6.0f, (float)32.0f);
+        float $$18 = (float)($$11 * $$15 + $$13 * $$16) * 100.0f;
+        $$18 = net.minecraft.util.Mth.clamp((float)$$18, (float)0.0f, (float)150.0f);
+        float $$19 = (float)($$11 * $$16 - $$13 * $$15) * 100.0f;
+        $$19 = net.minecraft.util.Mth.clamp((float)$$19, (float)-20.0f, (float)20.0f);
+        if ($$18 < 0.0f) {
+            $$18 = 0.0f;
         }
-        float f5 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)livingEntity.oBob, (float)livingEntity.bob);
-        f2 += net.minecraft.util.Mth.sin((float)(net.minecraft.util.Mth.lerp((float)partialTicks, (float)livingEntity.walkDistO, (float)livingEntity.walkDist) * 6.0f)) * 32.0f * f5;
-        if (livingEntity.isCrouching()) {
-            f2 += 25.0f;
+        float $$20 = net.minecraft.util.Mth.lerp((float)$$6, (float)$$3.oBob, (float)$$3.bob);
+        $$17 += net.minecraft.util.Mth.sin((float)(net.minecraft.util.Mth.lerp((float)$$6, (float)$$3.walkDistO, (float)$$3.walkDist) * 6.0f)) * 32.0f * $$20;
+        if ($$3.isCrouching()) {
+            $$17 += 25.0f;
         }
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(6.0f + f3 / 2.0f + f2));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(f4 / 2.0f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - f4 / 2.0f));
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)livingEntity.getCloakTextureLocation()));
-        ((net.minecraft.client.model.PlayerModel)this.getParentModel()).renderCloak(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
-        matrixStack.popPose();
+        $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(6.0f + $$18 / 2.0f + $$17));
+        $$0.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees($$19 / 2.0f));
+        $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - $$19 / 2.0f));
+        com.mojang.blaze3d.vertex.VertexConsumer $$21 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)$$3.getCloakTextureLocation()));
+        ((net.minecraft.client.model.PlayerModel)this.getParentModel()).renderCloak($$0, $$21, $$2, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
+        $$0.popPose();
     }
 }
 