--- net/minecraft/client/renderer/entity/layers/CarriedBlockLayer.java
+++ net/minecraft/client/renderer/entity/layers/CarriedBlockLayer.java
@@ -13,21 +13,21 @@
         this.blockRenderer = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.monster.EnderMan livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.world.level.block.state.BlockState blockState = livingEntity.getCarriedBlock();
-        if (blockState == null) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.monster.EnderMan $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        net.minecraft.world.level.block.state.BlockState $$10 = $$3.getCarriedBlock();
+        if ($$10 == null) {
             return;
         }
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.6875, -0.75);
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(20.0f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(45.0f));
-        matrixStack.translate(0.25, 0.1875, 0.25);
-        float f = 0.5f;
-        matrixStack.scale(-0.5f, -0.5f, 0.5f);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f));
-        this.blockRenderer.renderSingleBlock(blockState, matrixStack, buffer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
-        matrixStack.popPose();
+        $$0.pushPose();
+        $$0.translate(0.0, 0.6875, -0.75);
+        $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(20.0f));
+        $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(45.0f));
+        $$0.translate(0.25, 0.1875, 0.25);
+        float $$11 = 0.5f;
+        $$0.scale(-0.5f, -0.5f, 0.5f);
+        $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f));
+        this.blockRenderer.renderSingleBlock($$10, $$0, $$1, $$2, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
+        $$0.popPose();
     }
 }
 