--- net/minecraft/client/renderer/entity/layers/CatCollarLayer.java
+++ net/minecraft/client/renderer/entity/layers/CatCollarLayer.java
@@ -14,12 +14,12 @@
         this.catModel = new net.minecraft.client.model.CatModel($$1.bakeLayer(net.minecraft.client.model.geom.ModelLayers.CAT_COLLAR));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.Cat livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (!livingEntity.isTame()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.Cat $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        if (!$$3.isTame()) {
             return;
         }
-        float[] fArray = livingEntity.getCollarColor().getTextureDiffuseColors();
-        net.minecraft.client.renderer.entity.layers.CatCollarLayer.coloredCutoutModelCopyLayerRender((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.client.model.EntityModel)this.catModel, (net.minecraft.resources.ResourceLocation)CAT_COLLAR_LOCATION, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight, (net.minecraft.world.entity.LivingEntity)livingEntity, (float)limbSwing, (float)limbSwingAmount, (float)ageInTicks, (float)netHeadYaw, (float)headPitch, (float)partialTicks, (float)fArray[0], (float)fArray[1], (float)fArray[2]);
+        float[] $$10 = $$3.getCollarColor().getTextureDiffuseColors();
+        net.minecraft.client.renderer.entity.layers.CatCollarLayer.coloredCutoutModelCopyLayerRender((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.client.model.EntityModel)this.catModel, (net.minecraft.resources.ResourceLocation)CAT_COLLAR_LOCATION, (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.renderer.MultiBufferSource)$$1, (int)$$2, (net.minecraft.world.entity.LivingEntity)$$3, (float)$$4, (float)$$5, (float)$$7, (float)$$8, (float)$$9, (float)$$6, (float)$$10[0], (float)$$10[1], (float)$$10[2]);
     }
 
     static {