--- net/minecraft/client/renderer/entity/layers/CrossedArmsItemLayer.java
+++ net/minecraft/client/renderer/entity/layers/CrossedArmsItemLayer.java
@@ -13,13 +13,13 @@
         this.itemInHandRenderer = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, (double)0.4f, (double)-0.4f);
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(180.0f));
-        net.minecraft.world.item.ItemStack itemStack = livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND);
-        this.itemInHandRenderer.renderItem(livingEntity, itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, matrixStack, buffer, packedLight);
-        matrixStack.popPose();
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, T $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        $$0.pushPose();
+        $$0.translate(0.0, (double)0.4f, (double)-0.4f);
+        $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(180.0f));
+        net.minecraft.world.item.ItemStack $$10 = $$3.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND);
+        this.itemInHandRenderer.renderItem($$3, $$10, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, $$0, $$1, $$2);
+        $$0.popPose();
     }
 }
 