--- net/minecraft/client/renderer/entity/layers/CustomHeadLayer.java
+++ net/minecraft/client/renderer/entity/layers/CustomHeadLayer.java
@@ -25,46 +25,45 @@
         this.itemInHandRenderer = $$5;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f;
-        boolean bl;
-        net.minecraft.world.item.ItemStack itemStack = livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD);
-        if (itemStack.isEmpty()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, T $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        boolean $$12;
+        net.minecraft.world.item.ItemStack $$10 = $$3.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD);
+        if ($$10.isEmpty()) {
             return;
         }
-        net.minecraft.world.item.Item item = itemStack.getItem();
-        matrixStack.pushPose();
-        matrixStack.scale(this.scaleX, this.scaleY, this.scaleZ);
-        boolean bl2 = bl = livingEntity instanceof net.minecraft.world.entity.npc.Villager || livingEntity instanceof net.minecraft.world.entity.monster.ZombieVillager;
-        if (livingEntity.isBaby() && !(livingEntity instanceof net.minecraft.world.entity.npc.Villager)) {
-            f = 2.0f;
-            float f2 = 1.4f;
-            matrixStack.translate(0.0, 0.03125, 0.0);
-            matrixStack.scale(0.7f, 0.7f, 0.7f);
-            matrixStack.translate(0.0, 1.0, 0.0);
+        net.minecraft.world.item.Item $$11 = $$10.getItem();
+        $$0.pushPose();
+        $$0.scale(this.scaleX, this.scaleY, this.scaleZ);
+        boolean bl = $$12 = $$3 instanceof net.minecraft.world.entity.npc.Villager || $$3 instanceof net.minecraft.world.entity.monster.ZombieVillager;
+        if ($$3.isBaby() && !($$3 instanceof net.minecraft.world.entity.npc.Villager)) {
+            float $$13 = 2.0f;
+            float $$14 = 1.4f;
+            $$0.translate(0.0, 0.03125, 0.0);
+            $$0.scale(0.7f, 0.7f, 0.7f);
+            $$0.translate(0.0, 1.0, 0.0);
         }
-        ((net.minecraft.client.model.HeadedModel)this.getParentModel()).getHead().translateAndRotate(matrixStack);
-        if (item instanceof net.minecraft.world.item.BlockItem && ((net.minecraft.world.item.BlockItem)item).getBlock() instanceof net.minecraft.world.level.block.AbstractSkullBlock) {
-            net.minecraft.nbt.CompoundTag compoundTag;
-            f = 1.1875f;
-            matrixStack.scale(1.1875f, -1.1875f, -1.1875f);
-            if (bl) {
-                matrixStack.translate(0.0, 0.0625, 0.0);
+        ((net.minecraft.client.model.HeadedModel)this.getParentModel()).getHead().translateAndRotate($$0);
+        if ($$11 instanceof net.minecraft.world.item.BlockItem && ((net.minecraft.world.item.BlockItem)$$11).getBlock() instanceof net.minecraft.world.level.block.AbstractSkullBlock) {
+            net.minecraft.nbt.CompoundTag $$17;
+            float $$15 = 1.1875f;
+            $$0.scale(1.1875f, -1.1875f, -1.1875f);
+            if ($$12) {
+                $$0.translate(0.0, 0.0625, 0.0);
             }
-            com.mojang.authlib.GameProfile gameProfile = null;
-            if (itemStack.hasTag() && (compoundTag = itemStack.getTag()).contains("SkullOwner", 10)) {
-                gameProfile = net.minecraft.nbt.NbtUtils.readGameProfile((net.minecraft.nbt.CompoundTag)compoundTag.getCompound("SkullOwner"));
+            com.mojang.authlib.GameProfile $$16 = null;
+            if ($$10.hasTag() && ($$17 = $$10.getTag()).contains("SkullOwner", 10)) {
+                $$16 = net.minecraft.nbt.NbtUtils.readGameProfile((net.minecraft.nbt.CompoundTag)$$17.getCompound("SkullOwner"));
             }
-            matrixStack.translate(-0.5, 0.0, -0.5);
-            compoundTag = ((net.minecraft.world.level.block.AbstractSkullBlock)((net.minecraft.world.item.BlockItem)item).getBlock()).getType();
-            net.minecraft.client.model.SkullModelBase skullModelBase = (net.minecraft.client.model.SkullModelBase)this.skullModels.get((java.lang.Object)compoundTag);
-            net.minecraft.client.renderer.RenderType renderType = net.minecraft.client.renderer.blockentity.SkullBlockRenderer.getRenderType((net.minecraft.world.level.block.SkullBlock$Type)compoundTag, (com.mojang.authlib.GameProfile)gameProfile);
-            net.minecraft.client.renderer.blockentity.SkullBlockRenderer.renderSkull(null, (float)180.0f, (float)limbSwing, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight, (net.minecraft.client.model.SkullModelBase)skullModelBase, (net.minecraft.client.renderer.RenderType)renderType);
-        } else if (!(item instanceof net.minecraft.world.item.ArmorItem) || ((net.minecraft.world.item.ArmorItem)item).getSlot() != net.minecraft.world.entity.EquipmentSlot.HEAD) {
-            net.minecraft.client.renderer.entity.layers.CustomHeadLayer.translateToHead((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (boolean)bl);
-            this.itemInHandRenderer.renderItem(livingEntity, itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.HEAD, false, matrixStack, buffer, packedLight);
+            $$0.translate(-0.5, 0.0, -0.5);
+            net.minecraft.world.level.block.SkullBlock$Type $$18 = ((net.minecraft.world.level.block.AbstractSkullBlock)((net.minecraft.world.item.BlockItem)$$11).getBlock()).getType();
+            net.minecraft.client.model.SkullModelBase $$19 = (net.minecraft.client.model.SkullModelBase)this.skullModels.get((java.lang.Object)$$18);
+            net.minecraft.client.renderer.RenderType $$20 = net.minecraft.client.renderer.blockentity.SkullBlockRenderer.getRenderType((net.minecraft.world.level.block.SkullBlock$Type)$$18, (com.mojang.authlib.GameProfile)$$16);
+            net.minecraft.client.renderer.blockentity.SkullBlockRenderer.renderSkull(null, (float)180.0f, (float)$$4, (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.renderer.MultiBufferSource)$$1, (int)$$2, (net.minecraft.client.model.SkullModelBase)$$19, (net.minecraft.client.renderer.RenderType)$$20);
+        } else if (!($$11 instanceof net.minecraft.world.item.ArmorItem) || ((net.minecraft.world.item.ArmorItem)$$11).getSlot() != net.minecraft.world.entity.EquipmentSlot.HEAD) {
+            net.minecraft.client.renderer.entity.layers.CustomHeadLayer.translateToHead((com.mojang.blaze3d.vertex.PoseStack)$$0, (boolean)$$12);
+            this.itemInHandRenderer.renderItem($$3, $$10, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.HEAD, false, $$0, $$1, $$2);
         }
-        matrixStack.popPose();
+        $$0.popPose();
     }
 
     public static void translateToHead(com.mojang.blaze3d.vertex.PoseStack $$0, boolean $$1) {