--- net/minecraft/client/renderer/entity/layers/Deadmau5EarsLayer.java
+++ net/minecraft/client/renderer/entity/layers/Deadmau5EarsLayer.java
@@ -10,26 +10,26 @@
         super($$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.client.player.AbstractClientPlayer livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (!"deadmau5".equals((java.lang.Object)livingEntity.getName().getString()) || !livingEntity.isSkinLoaded() || livingEntity.isInvisible()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.client.player.AbstractClientPlayer $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        if (!"deadmau5".equals((java.lang.Object)$$3.getName().getString()) || !$$3.isSkinLoaded() || $$3.isInvisible()) {
             return;
         }
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)livingEntity.getSkinTextureLocation()));
-        int n = net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)livingEntity, (float)0.0f);
-        for (int i = 0; i < 2; ++i) {
-            float f = net.minecraft.util.Mth.lerp((float)partialTicks, (float)livingEntity.yRotO, (float)livingEntity.getYRot()) - net.minecraft.util.Mth.lerp((float)partialTicks, (float)livingEntity.yBodyRotO, (float)livingEntity.yBodyRot);
-            float f2 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)livingEntity.xRotO, (float)livingEntity.getXRot());
-            matrixStack.pushPose();
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f));
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(f2));
-            matrixStack.translate((double)(0.375f * (float)(i * 2 - 1)), 0.0, 0.0);
-            matrixStack.translate(0.0, -0.375, 0.0);
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-f2));
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-f));
-            float f3 = 1.3333334f;
-            matrixStack.scale(1.3333334f, 1.3333334f, 1.3333334f);
-            ((net.minecraft.client.model.PlayerModel)this.getParentModel()).renderEars(matrixStack, vertexConsumer, packedLight, n);
-            matrixStack.popPose();
+        com.mojang.blaze3d.vertex.VertexConsumer $$10 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)$$3.getSkinTextureLocation()));
+        int $$11 = net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)$$3, (float)0.0f);
+        for (int $$12 = 0; $$12 < 2; ++$$12) {
+            float $$13 = net.minecraft.util.Mth.lerp((float)$$6, (float)$$3.yRotO, (float)$$3.getYRot()) - net.minecraft.util.Mth.lerp((float)$$6, (float)$$3.yBodyRotO, (float)$$3.yBodyRot);
+            float $$14 = net.minecraft.util.Mth.lerp((float)$$6, (float)$$3.xRotO, (float)$$3.getXRot());
+            $$0.pushPose();
+            $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$13));
+            $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$14));
+            $$0.translate((double)(0.375f * (float)($$12 * 2 - 1)), 0.0, 0.0);
+            $$0.translate(0.0, -0.375, 0.0);
+            $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-$$14));
+            $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-$$13));
+            float $$15 = 1.3333334f;
+            $$0.scale(1.3333334f, 1.3333334f, 1.3333334f);
+            ((net.minecraft.client.model.PlayerModel)this.getParentModel()).renderEars($$0, $$10, $$2, $$11);
+            $$0.popPose();
         }
     }
 }