--- net/minecraft/client/renderer/entity/layers/DolphinCarryingItemLayer.java
+++ net/minecraft/client/renderer/entity/layers/DolphinCarryingItemLayer.java
@@ -13,20 +13,20 @@
         this.itemInHandRenderer = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.Dolphin livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        boolean bl = livingEntity.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT;
-        matrixStack.pushPose();
-        float f = 1.0f;
-        float f2 = -1.0f;
-        float f3 = net.minecraft.util.Mth.abs((float)livingEntity.getXRot()) / 60.0f;
-        if (livingEntity.getXRot() < 0.0f) {
-            matrixStack.translate(0.0, (double)(1.0f - f3 * 0.5f), (double)(-1.0f + f3 * 0.5f));
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.Dolphin $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        boolean $$10 = $$3.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT;
+        $$0.pushPose();
+        float $$11 = 1.0f;
+        float $$12 = -1.0f;
+        float $$13 = net.minecraft.util.Mth.abs((float)$$3.getXRot()) / 60.0f;
+        if ($$3.getXRot() < 0.0f) {
+            $$0.translate(0.0, (double)(1.0f - $$13 * 0.5f), (double)(-1.0f + $$13 * 0.5f));
         } else {
-            matrixStack.translate(0.0, (double)(1.0f + f3 * 0.8f), (double)(-1.0f + f3 * 0.2f));
+            $$0.translate(0.0, (double)(1.0f + $$13 * 0.8f), (double)(-1.0f + $$13 * 0.2f));
         }
-        net.minecraft.world.item.ItemStack itemStack = bl ? livingEntity.getMainHandItem() : livingEntity.getOffhandItem();
-        this.itemInHandRenderer.renderItem((net.minecraft.world.entity.LivingEntity)livingEntity, itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, matrixStack, buffer, packedLight);
-        matrixStack.popPose();
+        net.minecraft.world.item.ItemStack $$14 = $$10 ? $$3.getMainHandItem() : $$3.getOffhandItem();
+        this.itemInHandRenderer.renderItem((net.minecraft.world.entity.LivingEntity)$$3, $$14, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, $$0, $$1, $$2);
+        $$0.popPose();
     }
 }
 