--- net/minecraft/client/renderer/entity/layers/ElytraLayer.java
+++ net/minecraft/client/renderer/entity/layers/ElytraLayer.java
@@ -14,20 +14,31 @@
         this.elytraModel = new net.minecraft.client.model.ElytraModel($$1.bakeLayer(net.minecraft.client.model.geom.ModelLayers.ELYTRA));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.client.player.AbstractClientPlayer abstractClientPlayer;
-        net.minecraft.world.item.ItemStack itemStack = livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.CHEST);
-        if (!itemStack.is(net.minecraft.world.item.Items.ELYTRA)) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, T $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        net.minecraft.resources.ResourceLocation $$15;
+        net.minecraft.world.item.ItemStack $$10 = $$3.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.CHEST);
+        if (!$$10.is(net.minecraft.world.item.Items.ELYTRA)) {
             return;
         }
-        net.minecraft.resources.ResourceLocation resourceLocation = livingEntity instanceof net.minecraft.client.player.AbstractClientPlayer ? ((abstractClientPlayer = (net.minecraft.client.player.AbstractClientPlayer)livingEntity).isElytraLoaded() && abstractClientPlayer.getElytraTextureLocation() != null ? abstractClientPlayer.getElytraTextureLocation() : (abstractClientPlayer.isCapeLoaded() && abstractClientPlayer.getCloakTextureLocation() != null && abstractClientPlayer.isModelPartShown(net.minecraft.world.entity.player.PlayerModelPart.CAPE) ? abstractClientPlayer.getCloakTextureLocation() : WINGS_LOCATION)) : WINGS_LOCATION;
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.0, 0.125);
+        if ($$3 instanceof net.minecraft.client.player.AbstractClientPlayer) {
+            net.minecraft.client.player.AbstractClientPlayer $$11 = (net.minecraft.client.player.AbstractClientPlayer)$$3;
+            if ($$11.isElytraLoaded() && $$11.getElytraTextureLocation() != null) {
+                net.minecraft.resources.ResourceLocation $$12 = $$11.getElytraTextureLocation();
+            } else if ($$11.isCapeLoaded() && $$11.getCloakTextureLocation() != null && $$11.isModelPartShown(net.minecraft.world.entity.player.PlayerModelPart.CAPE)) {
+                net.minecraft.resources.ResourceLocation $$13 = $$11.getCloakTextureLocation();
+            } else {
+                net.minecraft.resources.ResourceLocation $$14 = WINGS_LOCATION;
+            }
+        } else {
+            $$15 = WINGS_LOCATION;
+        }
+        $$0.pushPose();
+        $$0.translate(0.0, 0.0, 0.125);
         this.getParentModel().copyPropertiesTo((net.minecraft.client.model.EntityModel)this.elytraModel);
-        this.elytraModel.setupAnim(livingEntity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        abstractClientPlayer = net.minecraft.client.renderer.entity.ItemRenderer.getArmorFoilBuffer((net.minecraft.client.renderer.MultiBufferSource)buffer, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.RenderType.armorCutoutNoCull((net.minecraft.resources.ResourceLocation)resourceLocation), (boolean)false, (boolean)itemStack.hasFoil());
-        this.elytraModel.renderToBuffer(matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)abstractClientPlayer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
+        this.elytraModel.setupAnim($$3, $$4, $$5, $$7, $$8, $$9);
+        com.mojang.blaze3d.vertex.VertexConsumer $$16 = net.minecraft.client.renderer.entity.ItemRenderer.getArmorFoilBuffer((net.minecraft.client.renderer.MultiBufferSource)$$1, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.RenderType.armorCutoutNoCull((net.minecraft.resources.ResourceLocation)$$15), (boolean)false, (boolean)$$10.hasFoil());
+        this.elytraModel.renderToBuffer($$0, $$16, $$2, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$0.popPose();
     }
 
     static {