--- net/minecraft/client/renderer/entity/layers/FoxHeldItemLayer.java
+++ net/minecraft/client/renderer/entity/layers/FoxHeldItemLayer.java
@@ -13,39 +13,38 @@
         this.itemInHandRenderer = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.Fox livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f;
-        boolean bl = livingEntity.isSleeping();
-        boolean bl2 = livingEntity.isBaby();
-        matrixStack.pushPose();
-        if (bl2) {
-            f = 0.75f;
-            matrixStack.scale(0.75f, 0.75f, 0.75f);
-            matrixStack.translate(0.0, 0.5, (double)0.209375f);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.Fox $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        boolean $$10 = $$3.isSleeping();
+        boolean $$11 = $$3.isBaby();
+        $$0.pushPose();
+        if ($$11) {
+            float $$12 = 0.75f;
+            $$0.scale(0.75f, 0.75f, 0.75f);
+            $$0.translate(0.0, 0.5, (double)0.209375f);
         }
-        matrixStack.translate((double)(((net.minecraft.client.model.FoxModel)this.getParentModel()).head.x / 16.0f), (double)(((net.minecraft.client.model.FoxModel)this.getParentModel()).head.y / 16.0f), (double)(((net.minecraft.client.model.FoxModel)this.getParentModel()).head.z / 16.0f));
-        f = livingEntity.getHeadRollAngle(partialTicks);
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotation(f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(netHeadYaw));
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(headPitch));
-        if (livingEntity.isBaby()) {
-            if (bl) {
-                matrixStack.translate((double)0.4f, (double)0.26f, (double)0.15f);
+        $$0.translate((double)(((net.minecraft.client.model.FoxModel)this.getParentModel()).head.x / 16.0f), (double)(((net.minecraft.client.model.FoxModel)this.getParentModel()).head.y / 16.0f), (double)(((net.minecraft.client.model.FoxModel)this.getParentModel()).head.z / 16.0f));
+        float $$13 = $$3.getHeadRollAngle($$6);
+        $$0.mulPose(com.mojang.math.Vector3f.ZP.rotation($$13));
+        $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$8));
+        $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$9));
+        if ($$3.isBaby()) {
+            if ($$10) {
+                $$0.translate((double)0.4f, (double)0.26f, (double)0.15f);
             } else {
-                matrixStack.translate((double)0.06f, (double)0.26f, -0.5);
+                $$0.translate((double)0.06f, (double)0.26f, -0.5);
             }
-        } else if (bl) {
-            matrixStack.translate((double)0.46f, (double)0.26f, (double)0.22f);
+        } else if ($$10) {
+            $$0.translate((double)0.46f, (double)0.26f, (double)0.22f);
         } else {
-            matrixStack.translate((double)0.06f, (double)0.27f, -0.5);
+            $$0.translate((double)0.06f, (double)0.27f, -0.5);
         }
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
-        if (bl) {
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
+        $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
+        if ($$10) {
+            $$0.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
         }
-        net.minecraft.world.item.ItemStack itemStack = livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND);
-        this.itemInHandRenderer.renderItem((net.minecraft.world.entity.LivingEntity)livingEntity, itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, matrixStack, buffer, packedLight);
-        matrixStack.popPose();
+        net.minecraft.world.item.ItemStack $$14 = $$3.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND);
+        this.itemInHandRenderer.renderItem((net.minecraft.world.entity.LivingEntity)$$3, $$14, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, $$0, $$1, $$2);
+        $$0.popPose();
     }
 }
 