--- net/minecraft/client/renderer/entity/layers/HorseArmorLayer.java
+++ net/minecraft/client/renderer/entity/layers/HorseArmorLayer.java
@@ -13,30 +13,30 @@
         this.model = new net.minecraft.client.model.HorseModel($$1.bakeLayer(net.minecraft.client.model.geom.ModelLayers.HORSE_ARMOR));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.horse.Horse livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f;
-        float f2;
-        float f3;
-        net.minecraft.world.item.ItemStack itemStack = livingEntity.getArmor();
-        if (!(itemStack.getItem() instanceof net.minecraft.world.item.HorseArmorItem)) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.horse.Horse $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        float $$18;
+        float $$17;
+        float $$16;
+        net.minecraft.world.item.ItemStack $$10 = $$3.getArmor();
+        if (!($$10.getItem() instanceof net.minecraft.world.item.HorseArmorItem)) {
             return;
         }
-        net.minecraft.world.item.HorseArmorItem horseArmorItem = (net.minecraft.world.item.HorseArmorItem)itemStack.getItem();
+        net.minecraft.world.item.HorseArmorItem $$11 = (net.minecraft.world.item.HorseArmorItem)$$10.getItem();
         ((net.minecraft.client.model.HorseModel)this.getParentModel()).copyPropertiesTo((net.minecraft.client.model.EntityModel)this.model);
-        this.model.prepareMobModel((net.minecraft.world.entity.animal.horse.AbstractHorse)livingEntity, limbSwing, limbSwingAmount, partialTicks);
-        this.model.setupAnim((net.minecraft.world.entity.animal.horse.AbstractHorse)livingEntity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        if (horseArmorItem instanceof net.minecraft.world.item.DyeableHorseArmorItem) {
-            int n = ((net.minecraft.world.item.DyeableHorseArmorItem)horseArmorItem).getColor(itemStack);
-            f3 = (float)(n >> 16 & 0xFF) / 255.0f;
-            f2 = (float)(n >> 8 & 0xFF) / 255.0f;
-            f = (float)(n & 0xFF) / 255.0f;
+        this.model.prepareMobModel((net.minecraft.world.entity.animal.horse.AbstractHorse)$$3, $$4, $$5, $$6);
+        this.model.setupAnim((net.minecraft.world.entity.animal.horse.AbstractHorse)$$3, $$4, $$5, $$7, $$8, $$9);
+        if ($$11 instanceof net.minecraft.world.item.DyeableHorseArmorItem) {
+            int $$12 = ((net.minecraft.world.item.DyeableHorseArmorItem)$$11).getColor($$10);
+            float $$13 = (float)($$12 >> 16 & 0xFF) / 255.0f;
+            float $$14 = (float)($$12 >> 8 & 0xFF) / 255.0f;
+            float $$15 = (float)($$12 & 0xFF) / 255.0f;
         } else {
-            f3 = 1.0f;
-            f2 = 1.0f;
-            f = 1.0f;
+            $$16 = 1.0f;
+            $$17 = 1.0f;
+            $$18 = 1.0f;
         }
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entityCutoutNoCull((net.minecraft.resources.ResourceLocation)horseArmorItem.getTexture()));
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, f3, f2, f, 1.0f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$19 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.entityCutoutNoCull((net.minecraft.resources.ResourceLocation)$$11.getTexture()));
+        this.model.renderToBuffer($$0, $$19, $$2, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, $$16, $$17, $$18, 1.0f);
     }
 }
 