--- net/minecraft/client/renderer/entity/layers/HorseMarkingLayer.java
+++ net/minecraft/client/renderer/entity/layers/HorseMarkingLayer.java
@@ -12,13 +12,13 @@
         super($$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.horse.Horse livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.resources.ResourceLocation resourceLocation = (net.minecraft.resources.ResourceLocation)LOCATION_BY_MARKINGS.get((java.lang.Object)livingEntity.getMarkings());
-        if (resourceLocation == null || livingEntity.isInvisible()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.horse.Horse $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        net.minecraft.resources.ResourceLocation $$10 = (net.minecraft.resources.ResourceLocation)LOCATION_BY_MARKINGS.get((java.lang.Object)$$3.getMarkings());
+        if ($$10 == null || $$3.isInvisible()) {
             return;
         }
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entityTranslucent((net.minecraft.resources.ResourceLocation)resourceLocation));
-        ((net.minecraft.client.model.HorseModel)this.getParentModel()).renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)livingEntity, (float)0.0f), 1.0f, 1.0f, 1.0f, 1.0f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$11 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.entityTranslucent((net.minecraft.resources.ResourceLocation)$$10));
+        ((net.minecraft.client.model.HorseModel)this.getParentModel()).renderToBuffer($$0, $$11, $$2, net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)$$3, (float)0.0f), 1.0f, 1.0f, 1.0f, 1.0f);
     }
 
     public static /* synthetic */ void lambda$static$0(java.util.EnumMap $$0) {