--- net/minecraft/client/renderer/entity/layers/HumanoidArmorLayer.java
+++ net/minecraft/client/renderer/entity/layers/HumanoidArmorLayer.java
@@ -16,11 +16,11 @@
         this.outerModel = $$2;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.renderArmorPiece(matrixStack, buffer, livingEntity, net.minecraft.world.entity.EquipmentSlot.CHEST, packedLight, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.CHEST));
-        this.renderArmorPiece(matrixStack, buffer, livingEntity, net.minecraft.world.entity.EquipmentSlot.LEGS, packedLight, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.LEGS));
-        this.renderArmorPiece(matrixStack, buffer, livingEntity, net.minecraft.world.entity.EquipmentSlot.FEET, packedLight, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.FEET));
-        this.renderArmorPiece(matrixStack, buffer, livingEntity, net.minecraft.world.entity.EquipmentSlot.HEAD, packedLight, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.HEAD));
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, T $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        this.renderArmorPiece($$0, $$1, $$3, net.minecraft.world.entity.EquipmentSlot.CHEST, $$2, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.CHEST));
+        this.renderArmorPiece($$0, $$1, $$3, net.minecraft.world.entity.EquipmentSlot.LEGS, $$2, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.LEGS));
+        this.renderArmorPiece($$0, $$1, $$3, net.minecraft.world.entity.EquipmentSlot.FEET, $$2, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.FEET));
+        this.renderArmorPiece($$0, $$1, $$3, net.minecraft.world.entity.EquipmentSlot.HEAD, $$2, this.getArmorModel(net.minecraft.world.entity.EquipmentSlot.HEAD));
     }
 
     public void renderArmorPiece(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, T $$2, net.minecraft.world.entity.EquipmentSlot $$3, int $$4, A $$5) {
@@ -48,29 +48,29 @@
         }
     }
 
-    public void setPartVisibility(A model, net.minecraft.world.entity.EquipmentSlot slot) {
-        model.setAllVisible(false);
-        switch (net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer$1.$SwitchMap$net$minecraft$world$entity$EquipmentSlot[slot.ordinal()]) {
+    public void setPartVisibility(A $$0, net.minecraft.world.entity.EquipmentSlot $$1) {
+        $$0.setAllVisible(false);
+        switch (net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer$1.$SwitchMap$net$minecraft$world$entity$EquipmentSlot[$$1.ordinal()]) {
             case 1: {
-                model.head.visible = true;
-                model.hat.visible = true;
+                $$0.head.visible = true;
+                $$0.hat.visible = true;
                 break;
             }
             case 2: {
-                model.body.visible = true;
-                model.rightArm.visible = true;
-                model.leftArm.visible = true;
+                $$0.body.visible = true;
+                $$0.rightArm.visible = true;
+                $$0.leftArm.visible = true;
                 break;
             }
             case 3: {
-                model.body.visible = true;
-                model.rightLeg.visible = true;
-                model.leftLeg.visible = true;
+                $$0.body.visible = true;
+                $$0.rightLeg.visible = true;
+                $$0.leftLeg.visible = true;
                 break;
             }
             case 4: {
-                model.rightLeg.visible = true;
-                model.leftLeg.visible = true;
+                $$0.rightLeg.visible = true;
+                $$0.leftLeg.visible = true;
             }
         }
     }
@@ -84,8 +84,8 @@
         return (A)(this.usesInnerModel($$0) ? this.innerModel : this.outerModel);
     }
 
-    public boolean usesInnerModel(net.minecraft.world.entity.EquipmentSlot slot) {
-        return slot == net.minecraft.world.entity.EquipmentSlot.LEGS;
+    public boolean usesInnerModel(net.minecraft.world.entity.EquipmentSlot $$0) {
+        return $$0 == net.minecraft.world.entity.EquipmentSlot.LEGS;
     }
 
     public net.minecraft.resources.ResourceLocation getArmorLocation(net.minecraft.world.item.ArmorItem $$0, boolean $$1, @javax.annotation.Nullable java.lang.String $$2) {