--- net/minecraft/client/renderer/entity/layers/IronGolemCrackinessLayer.java
+++ net/minecraft/client/renderer/entity/layers/IronGolemCrackinessLayer.java
@@ -12,16 +12,16 @@
         super($$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.IronGolem livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (livingEntity.isInvisible()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.IronGolem $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        if ($$3.isInvisible()) {
             return;
         }
-        net.minecraft.world.entity.animal.IronGolem$Crackiness crackiness = livingEntity.getCrackiness();
-        if (crackiness == net.minecraft.world.entity.animal.IronGolem$Crackiness.NONE) {
+        net.minecraft.world.entity.animal.IronGolem$Crackiness $$10 = $$3.getCrackiness();
+        if ($$10 == net.minecraft.world.entity.animal.IronGolem$Crackiness.NONE) {
             return;
         }
-        net.minecraft.resources.ResourceLocation resourceLocation = (net.minecraft.resources.ResourceLocation)resourceLocations.get((java.lang.Object)crackiness);
-        net.minecraft.client.renderer.entity.layers.IronGolemCrackinessLayer.renderColoredCutoutModel((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.resources.ResourceLocation)resourceLocation, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight, (net.minecraft.world.entity.LivingEntity)livingEntity, (float)1.0f, (float)1.0f, (float)1.0f);
+        net.minecraft.resources.ResourceLocation $$11 = (net.minecraft.resources.ResourceLocation)resourceLocations.get((java.lang.Object)$$10);
+        net.minecraft.client.renderer.entity.layers.IronGolemCrackinessLayer.renderColoredCutoutModel((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.resources.ResourceLocation)$$11, (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.renderer.MultiBufferSource)$$1, (int)$$2, (net.minecraft.world.entity.LivingEntity)$$3, (float)1.0f, (float)1.0f, (float)1.0f);
     }
 
     static {