--- net/minecraft/client/renderer/entity/layers/ItemInHandLayer.java
+++ net/minecraft/client/renderer/entity/layers/ItemInHandLayer.java
@@ -13,23 +13,23 @@
         this.itemInHandRenderer = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.world.item.ItemStack itemStack;
-        boolean bl = livingEntity.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT;
-        net.minecraft.world.item.ItemStack itemStack2 = bl ? livingEntity.getOffhandItem() : livingEntity.getMainHandItem();
-        net.minecraft.world.item.ItemStack itemStack3 = itemStack = bl ? livingEntity.getMainHandItem() : livingEntity.getOffhandItem();
-        if (itemStack2.isEmpty() && itemStack.isEmpty()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, T $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        net.minecraft.world.item.ItemStack $$12;
+        boolean $$10 = $$3.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT;
+        net.minecraft.world.item.ItemStack $$11 = $$10 ? $$3.getOffhandItem() : $$3.getMainHandItem();
+        net.minecraft.world.item.ItemStack itemStack = $$12 = $$10 ? $$3.getMainHandItem() : $$3.getOffhandItem();
+        if ($$11.isEmpty() && $$12.isEmpty()) {
             return;
         }
-        matrixStack.pushPose();
+        $$0.pushPose();
         if (this.getParentModel().young) {
-            float f = 0.5f;
-            matrixStack.translate(0.0, 0.75, 0.0);
-            matrixStack.scale(0.5f, 0.5f, 0.5f);
+            float $$13 = 0.5f;
+            $$0.translate(0.0, 0.75, 0.0);
+            $$0.scale(0.5f, 0.5f, 0.5f);
         }
-        this.renderArmWithItem(livingEntity, itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.THIRD_PERSON_RIGHT_HAND, net.minecraft.world.entity.HumanoidArm.RIGHT, matrixStack, buffer, packedLight);
-        this.renderArmWithItem(livingEntity, itemStack2, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.THIRD_PERSON_LEFT_HAND, net.minecraft.world.entity.HumanoidArm.LEFT, matrixStack, buffer, packedLight);
-        matrixStack.popPose();
+        this.renderArmWithItem($$3, $$12, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.THIRD_PERSON_RIGHT_HAND, net.minecraft.world.entity.HumanoidArm.RIGHT, $$0, $$1, $$2);
+        this.renderArmWithItem($$3, $$11, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.THIRD_PERSON_LEFT_HAND, net.minecraft.world.entity.HumanoidArm.LEFT, $$0, $$1, $$2);
+        $$0.popPose();
     }
 
     public void renderArmWithItem(net.minecraft.world.entity.LivingEntity $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType $$2, net.minecraft.world.entity.HumanoidArm $$3, com.mojang.blaze3d.vertex.PoseStack $$4, net.minecraft.client.renderer.MultiBufferSource $$5, int $$6) {