--- net/minecraft/client/renderer/entity/layers/LlamaDecorLayer.java
+++ net/minecraft/client/renderer/entity/layers/LlamaDecorLayer.java
@@ -15,20 +15,23 @@
         this.model = new net.minecraft.client.model.LlamaModel($$1.bakeLayer(net.minecraft.client.model.geom.ModelLayers.LLAMA_DECOR));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.horse.Llama livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.resources.ResourceLocation resourceLocation;
-        net.minecraft.world.item.DyeColor dyeColor = livingEntity.getSwag();
-        if (dyeColor != null) {
-            resourceLocation = TEXTURE_LOCATION[dyeColor.getId()];
-        } else if (livingEntity.isTraderLlama()) {
-            resourceLocation = TRADER_LLAMA;
+    /*
+     * WARNING - void declaration
+     */
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.horse.Llama $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        void $$13;
+        net.minecraft.world.item.DyeColor $$10 = $$3.getSwag();
+        if ($$10 != null) {
+            net.minecraft.resources.ResourceLocation $$11 = TEXTURE_LOCATION[$$10.getId()];
+        } else if ($$3.isTraderLlama()) {
+            net.minecraft.resources.ResourceLocation $$12 = TRADER_LLAMA;
         } else {
             return;
         }
         ((net.minecraft.client.model.LlamaModel)this.getParentModel()).copyPropertiesTo((net.minecraft.client.model.EntityModel)this.model);
-        this.model.setupAnim((net.minecraft.world.entity.animal.horse.AbstractChestedHorse)livingEntity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entityCutoutNoCull((net.minecraft.resources.ResourceLocation)resourceLocation));
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        this.model.setupAnim((net.minecraft.world.entity.animal.horse.AbstractChestedHorse)$$3, $$4, $$5, $$7, $$8, $$9);
+        com.mojang.blaze3d.vertex.VertexConsumer $$14 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.entityCutoutNoCull((net.minecraft.resources.ResourceLocation)$$13));
+        this.model.renderToBuffer($$0, $$14, $$2, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
     }
 
     static {