--- net/minecraft/client/renderer/entity/layers/ShulkerHeadLayer.java
+++ net/minecraft/client/renderer/entity/layers/ShulkerHeadLayer.java
@@ -10,10 +10,10 @@
         super($$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.monster.Shulker livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.client.renderer.entity.ShulkerRenderer.getTextureLocation((net.minecraft.world.item.DyeColor)livingEntity.getColor());
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)resourceLocation));
-        ((net.minecraft.client.model.ShulkerModel)this.getParentModel()).getHead().render(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)livingEntity, (float)0.0f));
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.monster.Shulker $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        net.minecraft.resources.ResourceLocation $$10 = net.minecraft.client.renderer.entity.ShulkerRenderer.getTextureLocation((net.minecraft.world.item.DyeColor)$$3.getColor());
+        com.mojang.blaze3d.vertex.VertexConsumer $$11 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)$$10));
+        ((net.minecraft.client.model.ShulkerModel)this.getParentModel()).getHead().render($$0, $$11, $$2, net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)$$3, (float)0.0f));
     }
 }
 