--- net/minecraft/client/renderer/entity/layers/SnowGolemHeadLayer.java
+++ net/minecraft/client/renderer/entity/layers/SnowGolemHeadLayer.java
@@ -15,32 +15,32 @@
         this.itemRenderer = $$2;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.SnowGolem livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        boolean bl;
-        if (!livingEntity.hasPumpkin()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.SnowGolem $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        boolean $$10;
+        if (!$$3.hasPumpkin()) {
             return;
         }
-        boolean bl2 = bl = net.minecraft.client.Minecraft.getInstance().shouldEntityAppearGlowing((net.minecraft.world.entity.Entity)livingEntity) && livingEntity.isInvisible();
-        if (livingEntity.isInvisible() && !bl) {
+        boolean bl = $$10 = net.minecraft.client.Minecraft.getInstance().shouldEntityAppearGlowing((net.minecraft.world.entity.Entity)$$3) && $$3.isInvisible();
+        if ($$3.isInvisible() && !$$10) {
             return;
         }
-        matrixStack.pushPose();
-        ((net.minecraft.client.model.SnowGolemModel)this.getParentModel()).getHead().translateAndRotate(matrixStack);
-        float f = 0.625f;
-        matrixStack.translate(0.0, -0.34375, 0.0);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
-        matrixStack.scale(0.625f, -0.625f, -0.625f);
-        net.minecraft.world.item.ItemStack itemStack = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN);
-        if (bl) {
-            net.minecraft.world.level.block.state.BlockState blockState = net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN.defaultBlockState();
-            net.minecraft.client.resources.model.BakedModel bakedModel = this.blockRenderer.getBlockModel(blockState);
-            int n = net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)livingEntity, (float)0.0f);
-            matrixStack.translate(-0.5, -0.5, -0.5);
-            this.blockRenderer.getModelRenderer().renderModel(matrixStack.last(), buffer.getBuffer(net.minecraft.client.renderer.RenderType.outline((net.minecraft.resources.ResourceLocation)net.minecraft.client.renderer.texture.TextureAtlas.LOCATION_BLOCKS)), blockState, bakedModel, 0.0f, 0.0f, 0.0f, packedLight, n);
+        $$0.pushPose();
+        ((net.minecraft.client.model.SnowGolemModel)this.getParentModel()).getHead().translateAndRotate($$0);
+        float $$11 = 0.625f;
+        $$0.translate(0.0, -0.34375, 0.0);
+        $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
+        $$0.scale(0.625f, -0.625f, -0.625f);
+        net.minecraft.world.item.ItemStack $$12 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN);
+        if ($$10) {
+            net.minecraft.world.level.block.state.BlockState $$13 = net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN.defaultBlockState();
+            net.minecraft.client.resources.model.BakedModel $$14 = this.blockRenderer.getBlockModel($$13);
+            int $$15 = net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)$$3, (float)0.0f);
+            $$0.translate(-0.5, -0.5, -0.5);
+            this.blockRenderer.getModelRenderer().renderModel($$0.last(), $$1.getBuffer(net.minecraft.client.renderer.RenderType.outline((net.minecraft.resources.ResourceLocation)net.minecraft.client.renderer.texture.TextureAtlas.LOCATION_BLOCKS)), $$13, $$14, 0.0f, 0.0f, 0.0f, $$2, $$15);
         } else {
-            this.itemRenderer.renderStatic((net.minecraft.world.entity.LivingEntity)livingEntity, itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.HEAD, false, matrixStack, buffer, livingEntity.level, packedLight, net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)livingEntity, (float)0.0f), livingEntity.getId());
+            this.itemRenderer.renderStatic((net.minecraft.world.entity.LivingEntity)$$3, $$12, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.HEAD, false, $$0, $$1, $$3.level, $$2, net.minecraft.client.renderer.entity.LivingEntityRenderer.getOverlayCoords((net.minecraft.world.entity.LivingEntity)$$3, (float)0.0f), $$3.getId());
         }
-        matrixStack.popPose();
+        $$0.popPose();
     }
 }
 