--- net/minecraft/client/renderer/entity/layers/SpinAttackEffectLayer.java
+++ net/minecraft/client/renderer/entity/layers/SpinAttackEffectLayer.java
@@ -23,20 +23,20 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)64);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (!livingEntity.isAutoSpinAttack()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, T $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        if (!$$3.isAutoSpinAttack()) {
             return;
         }
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entityCutoutNoCull((net.minecraft.resources.ResourceLocation)TEXTURE));
-        for (int i = 0; i < 3; ++i) {
-            matrixStack.pushPose();
-            float f = ageInTicks * (float)(-(45 + i * 5));
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f));
-            float f2 = 0.75f * (float)i;
-            matrixStack.scale(f2, f2, f2);
-            matrixStack.translate(0.0, (double)(-0.2f + 0.6f * (float)i), 0.0);
-            this.box.render(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
-            matrixStack.popPose();
+        com.mojang.blaze3d.vertex.VertexConsumer $$10 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.entityCutoutNoCull((net.minecraft.resources.ResourceLocation)TEXTURE));
+        for (int $$11 = 0; $$11 < 3; ++$$11) {
+            $$0.pushPose();
+            float $$12 = $$7 * (float)(-(45 + $$11 * 5));
+            $$0.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$12));
+            float $$13 = 0.75f * (float)$$11;
+            $$0.scale($$13, $$13, $$13);
+            $$0.translate(0.0, (double)(-0.2f + 0.6f * (float)$$11), 0.0);
+            this.box.render($$0, $$10, $$2, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
+            $$0.popPose();
         }
     }
 