--- net/minecraft/client/renderer/entity/layers/TropicalFishPatternLayer.java
+++ net/minecraft/client/renderer/entity/layers/TropicalFishPatternLayer.java
@@ -15,10 +15,10 @@
         this.modelB = new net.minecraft.client.model.TropicalFishModelB($$1.bakeLayer(net.minecraft.client.model.geom.ModelLayers.TROPICAL_FISH_LARGE_PATTERN));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.TropicalFish livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.client.model.TropicalFishModelA tropicalFishModelA = livingEntity.getBaseVariant() == 0 ? this.modelA : this.modelB;
-        float[] fArray = livingEntity.getPatternColor();
-        net.minecraft.client.renderer.entity.layers.TropicalFishPatternLayer.coloredCutoutModelCopyLayerRender((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.client.model.EntityModel)tropicalFishModelA, (net.minecraft.resources.ResourceLocation)livingEntity.getPatternTextureLocation(), (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight, (net.minecraft.world.entity.LivingEntity)livingEntity, (float)limbSwing, (float)limbSwingAmount, (float)ageInTicks, (float)netHeadYaw, (float)headPitch, (float)partialTicks, (float)fArray[0], (float)fArray[1], (float)fArray[2]);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.TropicalFish $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        net.minecraft.client.model.TropicalFishModelA $$10 = $$3.getBaseVariant() == 0 ? this.modelA : this.modelB;
+        float[] $$11 = $$3.getPatternColor();
+        net.minecraft.client.renderer.entity.layers.TropicalFishPatternLayer.coloredCutoutModelCopyLayerRender((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.client.model.EntityModel)$$10, (net.minecraft.resources.ResourceLocation)$$3.getPatternTextureLocation(), (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.renderer.MultiBufferSource)$$1, (int)$$2, (net.minecraft.world.entity.LivingEntity)$$3, (float)$$4, (float)$$5, (float)$$7, (float)$$8, (float)$$9, (float)$$6, (float)$$11[0], (float)$$11[1], (float)$$11[2]);
     }
 }
 