--- net/minecraft/client/renderer/entity/layers/WitchItemLayer.java
+++ net/minecraft/client/renderer/entity/layers/WitchItemLayer.java
@@ -10,20 +10,20 @@
         super($$0, $$1);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, T livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.world.item.ItemStack itemStack = livingEntity.getMainHandItem();
-        matrixStack.pushPose();
-        if (itemStack.is(net.minecraft.world.item.Items.POTION)) {
-            ((net.minecraft.client.model.WitchModel)this.getParentModel()).getHead().translateAndRotate(matrixStack);
-            ((net.minecraft.client.model.WitchModel)this.getParentModel()).getNose().translateAndRotate(matrixStack);
-            matrixStack.translate(0.0625, 0.25, 0.0);
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(140.0f));
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(10.0f));
-            matrixStack.translate(0.0, (double)-0.4f, (double)0.4f);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, T $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        net.minecraft.world.item.ItemStack $$10 = $$3.getMainHandItem();
+        $$0.pushPose();
+        if ($$10.is(net.minecraft.world.item.Items.POTION)) {
+            ((net.minecraft.client.model.WitchModel)this.getParentModel()).getHead().translateAndRotate($$0);
+            ((net.minecraft.client.model.WitchModel)this.getParentModel()).getNose().translateAndRotate($$0);
+            $$0.translate(0.0625, 0.25, 0.0);
+            $$0.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
+            $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(140.0f));
+            $$0.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(10.0f));
+            $$0.translate(0.0, (double)-0.4f, (double)0.4f);
         }
-        super.render(matrixStack, buffer, packedLight, livingEntity, limbSwing, limbSwingAmount, partialTicks, ageInTicks, netHeadYaw, headPitch);
-        matrixStack.popPose();
+        super.render($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7, $$8, $$9);
+        $$0.popPose();
     }
 }
 