--- net/minecraft/client/renderer/entity/layers/WolfCollarLayer.java
+++ net/minecraft/client/renderer/entity/layers/WolfCollarLayer.java
@@ -12,12 +12,12 @@
         super($$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.animal.Wolf livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (!livingEntity.isTame() || livingEntity.isInvisible()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.animal.Wolf $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        if (!$$3.isTame() || $$3.isInvisible()) {
             return;
         }
-        float[] fArray = livingEntity.getCollarColor().getTextureDiffuseColors();
-        net.minecraft.client.renderer.entity.layers.WolfCollarLayer.renderColoredCutoutModel((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.resources.ResourceLocation)WOLF_COLLAR_LOCATION, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight, (net.minecraft.world.entity.LivingEntity)livingEntity, (float)fArray[0], (float)fArray[1], (float)fArray[2]);
+        float[] $$10 = $$3.getCollarColor().getTextureDiffuseColors();
+        net.minecraft.client.renderer.entity.layers.WolfCollarLayer.renderColoredCutoutModel((net.minecraft.client.model.EntityModel)this.getParentModel(), (net.minecraft.resources.ResourceLocation)WOLF_COLLAR_LOCATION, (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.renderer.MultiBufferSource)$$1, (int)$$2, (net.minecraft.world.entity.LivingEntity)$$3, (float)$$10[0], (float)$$10[1], (float)$$10[2]);
     }
 
     static {