--- net/minecraft/world/entity/ai/behavior/BecomePassiveIfMemoryPresent.java
+++ net/minecraft/world/entity/ai/behavior/BecomePassiveIfMemoryPresent.java
@@ -13,9 +13,9 @@
         this.pacifyDuration = $$1;
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity, long gameTime) {
-        entity.getBrain().setMemoryWithExpiry(net.minecraft.world.entity.ai.memory.MemoryModuleType.PACIFIED, (java.lang.Object)true, (long)this.pacifyDuration);
-        entity.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.ATTACK_TARGET);
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1, long $$2) {
+        $$1.getBrain().setMemoryWithExpiry(net.minecraft.world.entity.ai.memory.MemoryModuleType.PACIFIED, (java.lang.Object)true, (long)this.pacifyDuration);
+        $$1.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.ATTACK_TARGET);
     }
 }
 