--- net/minecraft/world/entity/ai/behavior/BlockPosTracker.java
+++ net/minecraft/world/entity/ai/behavior/BlockPosTracker.java
@@ -14,6 +14,11 @@
         this.centerPosition = net.minecraft.world.phys.Vec3.atCenterOf((net.minecraft.core.Vec3i)$$0);
     }
 
+    public net.minecraft.world.entity.ai.behavior.BlockPosTracker(net.minecraft.world.phys.Vec3 $$0) {
+        this.blockPos = new net.minecraft.core.BlockPos($$0);
+        this.centerPosition = $$0;
+    }
+
     public net.minecraft.world.phys.Vec3 currentPosition() {
         return this.centerPosition;
     }
@@ -22,7 +27,7 @@
         return this.blockPos;
     }
 
-    public boolean isVisibleBy(net.minecraft.world.entity.LivingEntity entity) {
+    public boolean isVisibleBy(net.minecraft.world.entity.LivingEntity $$0) {
         return true;
     }
 