--- net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
+++ net/minecraft/world/entity/ai/behavior/CopyMemoryWithExpiry.java
@@ -19,13 +19,13 @@
         this.durationOfCopy = $$3;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        return this.predicate.test(owner);
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        return this.predicate.test($$1);
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        net.minecraft.world.entity.ai.Brain brain = entity.getBrain();
-        brain.setMemoryWithExpiry(this.targetMemory, brain.getMemory(this.sourceMemory).get(), (long)this.durationOfCopy.sample(level.random));
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        net.minecraft.world.entity.ai.Brain $$3 = $$1.getBrain();
+        $$3.setMemoryWithExpiry(this.targetMemory, $$3.getMemory(this.sourceMemory).get(), (long)this.durationOfCopy.sample($$0.random));
     }
 }
 