--- net/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java
+++ net/minecraft/world/entity/ai/behavior/DismountOrSkipMounting.java
@@ -15,23 +15,23 @@
         this.dontRideIf = $$1;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        net.minecraft.world.entity.Entity entity = owner.getVehicle();
-        net.minecraft.world.entity.Entity entity2 = (net.minecraft.world.entity.Entity)owner.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.RIDE_TARGET).orElse(null);
-        if (entity == null && entity2 == null) {
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        net.minecraft.world.entity.Entity $$2 = $$1.getVehicle();
+        net.minecraft.world.entity.Entity $$3 = (net.minecraft.world.entity.Entity)$$1.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.RIDE_TARGET).orElse(null);
+        if ($$2 == null && $$3 == null) {
             return false;
         }
-        net.minecraft.world.entity.Entity entity3 = entity == null ? entity2 : entity;
-        return !this.isVehicleValid(owner, entity3) || this.dontRideIf.test(owner, (java.lang.Object)entity3);
+        net.minecraft.world.entity.Entity $$4 = $$2 == null ? $$3 : $$2;
+        return !this.isVehicleValid($$1, $$4) || this.dontRideIf.test($$1, (java.lang.Object)$$4);
     }
 
     public boolean isVehicleValid(E $$0, net.minecraft.world.entity.Entity $$1) {
         return $$1.isAlive() && $$1.closerThan($$0, (double)this.maxWalkDistToRideTarget) && $$1.level == ((net.minecraft.world.entity.LivingEntity)$$0).level;
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        entity.stopRiding();
-        entity.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.RIDE_TARGET);
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        $$1.stopRiding();
+        $$1.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.RIDE_TARGET);
     }
 }
 