--- net/minecraft/world/entity/ai/behavior/DoNothing.java
+++ net/minecraft/world/entity/ai/behavior/DoNothing.java
@@ -10,7 +10,7 @@
         super((java.util.Map)com.google.common.collect.ImmutableMap.of(), $$0, $$1);
     }
 
-    public boolean canStillUse(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity, long gameTime) {
+    public boolean canStillUse(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1, long $$2) {
         return true;
     }
 }