--- net/minecraft/world/entity/ai/behavior/EntityTracker.java
+++ net/minecraft/world/entity/ai/behavior/EntityTracker.java
@@ -22,17 +22,21 @@
         return this.entity.blockPosition();
     }
 
-    public boolean isVisibleBy(net.minecraft.world.entity.LivingEntity entity) {
-        net.minecraft.world.entity.Entity entity2 = this.entity;
-        if (!(entity2 instanceof net.minecraft.world.entity.LivingEntity)) {
+    /*
+     * WARNING - void declaration
+     */
+    public boolean isVisibleBy(net.minecraft.world.entity.LivingEntity $$0) {
+        void $$2;
+        net.minecraft.world.entity.Entity entity = this.entity;
+        if (!(entity instanceof net.minecraft.world.entity.LivingEntity)) {
             return true;
         }
-        net.minecraft.world.entity.LivingEntity livingEntity = (net.minecraft.world.entity.LivingEntity)entity2;
-        if (!livingEntity.isAlive()) {
+        net.minecraft.world.entity.LivingEntity $$1 = (net.minecraft.world.entity.LivingEntity)entity;
+        if (!$$2.isAlive()) {
             return false;
         }
-        entity2 = entity.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES);
-        return entity2.isPresent() && ((net.minecraft.world.entity.ai.memory.NearestVisibleLivingEntities)entity2.get()).contains(livingEntity);
+        java.util.Optional $$3 = $$0.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES);
+        return $$3.isPresent() && ((net.minecraft.world.entity.ai.memory.NearestVisibleLivingEntities)$$3.get()).contains((net.minecraft.world.entity.LivingEntity)$$2);
     }
 
     public net.minecraft.world.entity.Entity getEntity() {