--- net/minecraft/world/entity/ai/behavior/EraseMemoryIf.java
+++ net/minecraft/world/entity/ai/behavior/EraseMemoryIf.java
@@ -15,12 +15,12 @@
         this.memoryType = $$1;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        return this.predicate.test(owner);
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        return this.predicate.test($$1);
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        entity.getBrain().eraseMemory(this.memoryType);
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        $$1.getBrain().eraseMemory(this.memoryType);
     }
 }
 