--- net/minecraft/world/entity/ai/behavior/GoAndGiveItemsToTarget.java
+++ net/minecraft/world/entity/ai/behavior/GoAndGiveItemsToTarget.java
@@ -11,8 +11,8 @@
     public java.util.function.Function<net.minecraft.world.entity.LivingEntity, java.util.Optional<net.minecraft.world.entity.ai.behavior.PositionTracker>> targetPositionGetter;
     public float speedModifier;
 
-    public net.minecraft.world.entity.ai.behavior.GoAndGiveItemsToTarget(java.util.function.Function<net.minecraft.world.entity.LivingEntity, java.util.Optional<net.minecraft.world.entity.ai.behavior.PositionTracker>> $$0, float $$1) {
-        super(java.util.Map.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.LOOK_TARGET, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.WALK_TARGET, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.ITEM_PICKUP_COOLDOWN_TICKS, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED));
+    public net.minecraft.world.entity.ai.behavior.GoAndGiveItemsToTarget(java.util.function.Function<net.minecraft.world.entity.LivingEntity, java.util.Optional<net.minecraft.world.entity.ai.behavior.PositionTracker>> $$0, float $$1, int $$2) {
+        super(java.util.Map.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.LOOK_TARGET, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.WALK_TARGET, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.ITEM_PICKUP_COOLDOWN_TICKS, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED), $$2);
         this.targetPositionGetter = $$0;
         this.speedModifier = $$1;
     }