--- net/minecraft/world/entity/ai/behavior/GoToWantedItem.java
+++ net/minecraft/world/entity/ai/behavior/GoToWantedItem.java
@@ -21,12 +21,13 @@
         this.speedModifier = $$1;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        return !this.isOnPickupCooldown(owner) && this.predicate.test(owner) && this.getClosestLovedItem(owner).closerThan(owner, (double)this.maxDistToWalk);
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        net.minecraft.world.entity.item.ItemEntity $$2 = this.getClosestLovedItem($$1);
+        return !this.isOnPickupCooldown($$1) && this.predicate.test($$1) && $$2.closerThan($$1, (double)this.maxDistToWalk) && ((net.minecraft.world.entity.LivingEntity)$$1).level.getWorldBorder().isWithinBounds($$2.blockPosition());
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        net.minecraft.world.entity.ai.behavior.BehaviorUtils.setWalkAndLookTargetMemories(entity, (net.minecraft.world.entity.Entity)this.getClosestLovedItem(entity), (float)this.speedModifier, (int)0);
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        net.minecraft.world.entity.ai.behavior.BehaviorUtils.setWalkAndLookTargetMemories($$1, (net.minecraft.world.entity.Entity)this.getClosestLovedItem($$1), (float)this.speedModifier, (int)0);
     }
 
     public boolean isOnPickupCooldown(E $$0) {