--- net/minecraft/world/entity/ai/behavior/LocateHidingPlaceDuringRaid.java
+++ net/minecraft/world/entity/ai/behavior/LocateHidingPlaceDuringRaid.java
@@ -10,9 +10,9 @@
         super($$0, $$1, 1);
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity owner) {
-        net.minecraft.world.entity.raid.Raid raid = level.getRaidAt(owner.blockPosition());
-        return super.checkExtraStartConditions(level, owner) && raid != null && raid.isActive() && !raid.isVictory() && !raid.isLoss();
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        net.minecraft.world.entity.raid.Raid $$2 = $$0.getRaidAt($$1.blockPosition());
+        return super.checkExtraStartConditions($$0, $$1) && $$2 != null && $$2.isActive() && !$$2.isVictory() && !$$2.isLoss();
     }
 }
 