--- net/minecraft/world/entity/ai/behavior/Mount.java
+++ net/minecraft/world/entity/ai/behavior/Mount.java
@@ -14,15 +14,15 @@
         this.speedModifier = $$0;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        return !owner.isPassenger();
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        return !$$1.isPassenger();
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        if (this.isCloseEnoughToStartRiding(entity)) {
-            entity.startRiding(this.getRidableEntity(entity));
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        if (this.isCloseEnoughToStartRiding($$1)) {
+            $$1.startRiding(this.getRidableEntity($$1));
         } else {
-            net.minecraft.world.entity.ai.behavior.BehaviorUtils.setWalkAndLookTargetMemories(entity, (net.minecraft.world.entity.Entity)this.getRidableEntity(entity), (float)this.speedModifier, (int)1);
+            net.minecraft.world.entity.ai.behavior.BehaviorUtils.setWalkAndLookTargetMemories($$1, (net.minecraft.world.entity.Entity)this.getRidableEntity($$1), (float)this.speedModifier, (int)1);
         }
     }
 