--- net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
+++ net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.java
@@ -17,22 +17,22 @@
         super((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.VISIBLE_VILLAGER_BABIES, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_PRESENT, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.WALK_TARGET, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_ABSENT, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.LOOK_TARGET, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.INTERACTION_TARGET, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED));
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.PathfinderMob owner) {
-        return level.getRandom().nextInt(10) == 0 && this.hasFriendsNearby(owner);
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.PathfinderMob $$1) {
+        return $$0.getRandom().nextInt(10) == 0 && this.hasFriendsNearby($$1);
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.PathfinderMob entity, long gameTime) {
-        net.minecraft.world.entity.LivingEntity livingEntity = this.seeIfSomeoneIsChasingMe((net.minecraft.world.entity.LivingEntity)entity);
-        if (livingEntity != null) {
-            this.fleeFromChaser(level, entity, livingEntity);
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.PathfinderMob $$12, long $$2) {
+        net.minecraft.world.entity.LivingEntity $$3 = this.seeIfSomeoneIsChasingMe((net.minecraft.world.entity.LivingEntity)$$12);
+        if ($$3 != null) {
+            this.fleeFromChaser($$0, $$12, $$3);
             return;
         }
-        java.util.Optional optional = this.findSomeoneBeingChased(entity);
-        if (optional.isPresent()) {
-            net.minecraft.world.entity.ai.behavior.PlayTagWithOtherKids.chaseKid((net.minecraft.world.entity.PathfinderMob)entity, (net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)optional.get()));
+        java.util.Optional $$4 = this.findSomeoneBeingChased($$12);
+        if ($$4.isPresent()) {
+            net.minecraft.world.entity.ai.behavior.PlayTagWithOtherKids.chaseKid((net.minecraft.world.entity.PathfinderMob)$$12, (net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)$$4.get()));
             return;
         }
-        this.findSomeoneToChase(entity).ifPresent($$1 -> net.minecraft.world.entity.ai.behavior.PlayTagWithOtherKids.chaseKid((net.minecraft.world.entity.PathfinderMob)entity, (net.minecraft.world.entity.LivingEntity)$$1));
+        this.findSomeoneToChase($$12).ifPresent($$1 -> net.minecraft.world.entity.ai.behavior.PlayTagWithOtherKids.chaseKid((net.minecraft.world.entity.PathfinderMob)$$12, (net.minecraft.world.entity.LivingEntity)$$1));
     }
 
     public void fleeFromChaser(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.PathfinderMob $$1, net.minecraft.world.entity.LivingEntity $$2) {