--- net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
+++ net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
@@ -13,9 +13,9 @@
         this.profession = $$0;
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.npc.Villager entity, long gameTime) {
-        net.minecraft.core.GlobalPos globalPos = (net.minecraft.core.GlobalPos)entity.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.JOB_SITE).get();
-        level.getPoiManager().getType(globalPos.pos()).ifPresent($$22 -> net.minecraft.world.entity.ai.behavior.BehaviorUtils.getNearbyVillagersWithCondition((net.minecraft.world.entity.npc.Villager)entity, $$2 -> this.competesForSameJobsite(globalPos, $$22, $$2)).reduce((java.lang.Object)entity, net.minecraft.world.entity.ai.behavior.PoiCompetitorScan::selectWinner));
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.npc.Villager $$1, long $$2) {
+        net.minecraft.core.GlobalPos $$3 = (net.minecraft.core.GlobalPos)$$1.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.JOB_SITE).get();
+        $$0.getPoiManager().getType($$3.pos()).ifPresent($$22 -> net.minecraft.world.entity.ai.behavior.BehaviorUtils.getNearbyVillagersWithCondition((net.minecraft.world.entity.npc.Villager)$$1, $$2 -> this.competesForSameJobsite($$3, $$22, $$2)).reduce((java.lang.Object)$$1, net.minecraft.world.entity.ai.behavior.PoiCompetitorScan::selectWinner));
     }
 
     public static net.minecraft.world.entity.npc.Villager selectWinner(net.minecraft.world.entity.npc.Villager $$0, net.minecraft.world.entity.npc.Villager $$1) {