--- net/minecraft/world/entity/ai/behavior/PrepareRamNearestTarget.java
+++ net/minecraft/world/entity/ai/behavior/PrepareRamNearestTarget.java
@@ -49,7 +49,7 @@
 
     public void tick(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
         boolean $$3;
-        if (!this.ramCandidate.isPresent()) {
+        if (this.ramCandidate.isEmpty()) {
             return;
         }
         $$1.getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.WALK_TARGET, (java.lang.Object)new net.minecraft.world.entity.ai.memory.WalkTarget(((net.minecraft.world.entity.ai.behavior.PrepareRamNearestTarget$RamCandidate)this.ramCandidate.get()).getStartPosition(), this.walkSpeed, 0));
@@ -63,7 +63,7 @@
             net.minecraft.core.BlockPos $$4 = $$1.blockPosition();
             if ($$4.equals((java.lang.Object)((net.minecraft.world.entity.ai.behavior.PrepareRamNearestTarget$RamCandidate)this.ramCandidate.get()).getStartPosition())) {
                 $$0.broadcastEntityEvent($$1, (byte)58);
-                if (!this.reachedRamPositionTimestamp.isPresent()) {
+                if (this.reachedRamPositionTimestamp.isEmpty()) {
                     this.reachedRamPositionTimestamp = java.util.Optional.of((java.lang.Object)$$2);
                 }
                 if ($$2 - (java.lang.Long)this.reachedRamPositionTimestamp.get() >= (long)this.ramPrepareTime) {