--- net/minecraft/world/entity/ai/behavior/RandomStroll.java
+++ net/minecraft/world/entity/ai/behavior/RandomStroll.java
@@ -37,9 +37,9 @@
         return this.mayStrollFromWater || !$$1.isInWaterOrBubble();
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.PathfinderMob entity, long gameTime) {
-        java.util.Optional optional = java.util.Optional.ofNullable((java.lang.Object)this.getTargetPos(entity));
-        entity.getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.WALK_TARGET, optional.map($$0 -> new net.minecraft.world.entity.ai.memory.WalkTarget($$0, this.speedModifier, 0)));
+    public void start(net.minecraft.server.level.ServerLevel $$02, net.minecraft.world.entity.PathfinderMob $$1, long $$2) {
+        java.util.Optional $$3 = java.util.Optional.ofNullable((java.lang.Object)this.getTargetPos($$1));
+        $$1.getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.WALK_TARGET, $$3.map($$0 -> new net.minecraft.world.entity.ai.memory.WalkTarget($$0, this.speedModifier, 0)));
     }
 
     @javax.annotation.Nullable