--- net/minecraft/world/entity/ai/behavior/ResetProfession.java
+++ net/minecraft/world/entity/ai/behavior/ResetProfession.java
@@ -10,14 +10,14 @@
         super((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.JOB_SITE, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_ABSENT));
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.npc.Villager owner) {
-        net.minecraft.world.entity.npc.VillagerData villagerData = owner.getVillagerData();
-        return villagerData.getProfession() != net.minecraft.world.entity.npc.VillagerProfession.NONE && villagerData.getProfession() != net.minecraft.world.entity.npc.VillagerProfession.NITWIT && owner.getVillagerXp() == 0 && villagerData.getLevel() <= 1;
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.npc.Villager $$1) {
+        net.minecraft.world.entity.npc.VillagerData $$2 = $$1.getVillagerData();
+        return $$2.getProfession() != net.minecraft.world.entity.npc.VillagerProfession.NONE && $$2.getProfession() != net.minecraft.world.entity.npc.VillagerProfession.NITWIT && $$1.getVillagerXp() == 0 && $$2.getLevel() <= 1;
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.npc.Villager entity, long gameTime) {
-        entity.setVillagerData(entity.getVillagerData().setProfession(net.minecraft.world.entity.npc.VillagerProfession.NONE));
-        entity.refreshBrain(level);
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.npc.Villager $$1, long $$2) {
+        $$1.setVillagerData($$1.getVillagerData().setProfession(net.minecraft.world.entity.npc.VillagerProfession.NONE));
+        $$1.refreshBrain($$0);
     }
 }
 