--- net/minecraft/world/entity/ai/behavior/RingBell.java
+++ net/minecraft/world/entity/ai/behavior/RingBell.java
@@ -13,17 +13,17 @@
         super((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.MEETING_POINT, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_PRESENT));
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity owner) {
-        return level.random.nextFloat() > 0.95f;
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        return $$0.random.nextFloat() > 0.95f;
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity, long gameTime) {
-        net.minecraft.world.level.block.state.BlockState blockState;
-        net.minecraft.world.entity.ai.Brain brain = entity.getBrain();
-        net.minecraft.core.BlockPos blockPos = ((net.minecraft.core.GlobalPos)brain.getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.MEETING_POINT).get()).pos();
-        if (blockPos.closerThan((net.minecraft.core.Vec3i)entity.blockPosition(), 3.0) && (blockState = level.getBlockState(blockPos)).is(net.minecraft.world.level.block.Blocks.BELL)) {
-            net.minecraft.world.level.block.BellBlock bellBlock = (net.minecraft.world.level.block.BellBlock)blockState.getBlock();
-            bellBlock.attemptToRing((net.minecraft.world.entity.Entity)entity, (net.minecraft.world.level.Level)level, blockPos, null);
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1, long $$2) {
+        net.minecraft.world.level.block.state.BlockState $$5;
+        net.minecraft.world.entity.ai.Brain $$3 = $$1.getBrain();
+        net.minecraft.core.BlockPos $$4 = ((net.minecraft.core.GlobalPos)$$3.getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.MEETING_POINT).get()).pos();
+        if ($$4.closerThan((net.minecraft.core.Vec3i)$$1.blockPosition(), 3.0) && ($$5 = $$0.getBlockState($$4)).is(net.minecraft.world.level.block.Blocks.BELL)) {
+            net.minecraft.world.level.block.BellBlock $$6 = (net.minecraft.world.level.block.BellBlock)$$5.getBlock();
+            $$6.attemptToRing((net.minecraft.world.entity.Entity)$$1, (net.minecraft.world.level.Level)$$0, $$4, null);
         }
     }
 }