--- net/minecraft/world/entity/ai/behavior/SetRaidStatus.java
+++ net/minecraft/world/entity/ai/behavior/SetRaidStatus.java
@@ -10,20 +10,20 @@
         super((java.util.Map)com.google.common.collect.ImmutableMap.of());
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity owner) {
-        return level.random.nextInt(20) == 0;
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        return $$0.random.nextInt(20) == 0;
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity, long gameTime) {
-        net.minecraft.world.entity.ai.Brain brain = entity.getBrain();
-        net.minecraft.world.entity.raid.Raid raid = level.getRaidAt(entity.blockPosition());
-        if (raid != null) {
-            if (!raid.hasFirstWaveSpawned() || raid.isBetweenWaves()) {
-                brain.setDefaultActivity(net.minecraft.world.entity.schedule.Activity.PRE_RAID);
-                brain.setActiveActivityIfPossible(net.minecraft.world.entity.schedule.Activity.PRE_RAID);
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1, long $$2) {
+        net.minecraft.world.entity.ai.Brain $$3 = $$1.getBrain();
+        net.minecraft.world.entity.raid.Raid $$4 = $$0.getRaidAt($$1.blockPosition());
+        if ($$4 != null) {
+            if (!$$4.hasFirstWaveSpawned() || $$4.isBetweenWaves()) {
+                $$3.setDefaultActivity(net.minecraft.world.entity.schedule.Activity.PRE_RAID);
+                $$3.setActiveActivityIfPossible(net.minecraft.world.entity.schedule.Activity.PRE_RAID);
             } else {
-                brain.setDefaultActivity(net.minecraft.world.entity.schedule.Activity.RAID);
-                brain.setActiveActivityIfPossible(net.minecraft.world.entity.schedule.Activity.RAID);
+                $$3.setDefaultActivity(net.minecraft.world.entity.schedule.Activity.RAID);
+                $$3.setActiveActivityIfPossible(net.minecraft.world.entity.schedule.Activity.RAID);
             }
         }
     }