--- net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
+++ net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.java
@@ -16,22 +16,22 @@
         this.memoryType = $$1;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity owner) {
-        net.minecraft.core.GlobalPos globalPos = (net.minecraft.core.GlobalPos)owner.getBrain().getMemory(this.memoryType).get();
-        return level.dimension() == globalPos.dimension() && globalPos.pos().closerToCenterThan((net.minecraft.core.Position)owner.position(), 16.0);
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        net.minecraft.core.GlobalPos $$2 = (net.minecraft.core.GlobalPos)$$1.getBrain().getMemory(this.memoryType).get();
+        return $$0.dimension() == $$2.dimension() && $$2.pos().closerToCenterThan((net.minecraft.core.Position)$$1.position(), 16.0);
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity, long gameTime) {
-        net.minecraft.world.entity.ai.Brain brain = entity.getBrain();
-        net.minecraft.core.GlobalPos globalPos = (net.minecraft.core.GlobalPos)brain.getMemory(this.memoryType).get();
-        net.minecraft.core.BlockPos blockPos = globalPos.pos();
-        net.minecraft.server.level.ServerLevel serverLevel = level.getServer().getLevel(globalPos.dimension());
-        if (serverLevel == null || this.poiDoesntExist(serverLevel, blockPos)) {
-            brain.eraseMemory(this.memoryType);
-        } else if (this.bedIsOccupied(serverLevel, blockPos, entity)) {
-            brain.eraseMemory(this.memoryType);
-            level.getPoiManager().release(blockPos);
-            net.minecraft.network.protocol.game.DebugPackets.sendPoiTicketCountPacket((net.minecraft.server.level.ServerLevel)level, (net.minecraft.core.BlockPos)blockPos);
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1, long $$2) {
+        net.minecraft.world.entity.ai.Brain $$3 = $$1.getBrain();
+        net.minecraft.core.GlobalPos $$4 = (net.minecraft.core.GlobalPos)$$3.getMemory(this.memoryType).get();
+        net.minecraft.core.BlockPos $$5 = $$4.pos();
+        net.minecraft.server.level.ServerLevel $$6 = $$0.getServer().getLevel($$4.dimension());
+        if ($$6 == null || this.poiDoesntExist($$6, $$5)) {
+            $$3.eraseMemory(this.memoryType);
+        } else if (this.bedIsOccupied($$6, $$5, $$1)) {
+            $$3.eraseMemory(this.memoryType);
+            $$0.getPoiManager().release($$5);
+            net.minecraft.network.protocol.game.DebugPackets.sendPoiTicketCountPacket((net.minecraft.server.level.ServerLevel)$$0, (net.minecraft.core.BlockPos)$$5);
         }
     }
 