--- net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
+++ net/minecraft/world/entity/ai/behavior/VillagerCalmDown.java
@@ -12,18 +12,18 @@
         super((java.util.Map)com.google.common.collect.ImmutableMap.of());
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.npc.Villager entity, long gameTime) {
-        boolean bl;
-        boolean bl2 = bl = net.minecraft.world.entity.ai.behavior.VillagerPanicTrigger.isHurt((net.minecraft.world.entity.LivingEntity)entity) || net.minecraft.world.entity.ai.behavior.VillagerPanicTrigger.hasHostile((net.minecraft.world.entity.LivingEntity)entity) || net.minecraft.world.entity.ai.behavior.VillagerCalmDown.isCloseToEntityThatHurtMe((net.minecraft.world.entity.npc.Villager)entity);
-        if (!bl) {
-            entity.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY);
-            entity.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY);
-            entity.getBrain().updateActivityFromSchedule(level.getDayTime(), level.getGameTime());
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.npc.Villager $$1, long $$2) {
+        boolean $$3;
+        boolean bl = $$3 = net.minecraft.world.entity.ai.behavior.VillagerPanicTrigger.isHurt((net.minecraft.world.entity.LivingEntity)$$1) || net.minecraft.world.entity.ai.behavior.VillagerPanicTrigger.hasHostile((net.minecraft.world.entity.LivingEntity)$$1) || net.minecraft.world.entity.ai.behavior.VillagerCalmDown.isCloseToEntityThatHurtMe((net.minecraft.world.entity.npc.Villager)$$1);
+        if (!$$3) {
+            $$1.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY);
+            $$1.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY);
+            $$1.getBrain().updateActivityFromSchedule($$0.getDayTime(), $$0.getGameTime());
         }
     }
 
-    public static boolean isCloseToEntityThatHurtMe(net.minecraft.world.entity.npc.Villager villager) {
-        return villager.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY).filter($$1 -> $$1.distanceToSqr((net.minecraft.world.entity.Entity)villager) <= 36.0).isPresent();
+    public static boolean isCloseToEntityThatHurtMe(net.minecraft.world.entity.npc.Villager $$0) {
+        return $$0.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY).filter($$1 -> $$1.distanceToSqr((net.minecraft.world.entity.Entity)$$0) <= 36.0).isPresent();
     }
 }
 