--- net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
+++ net/minecraft/world/entity/ai/behavior/VillageBoundRandomStroll.java
@@ -23,17 +23,17 @@
         this.maxYDist = $$2;
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.PathfinderMob entity, long gameTime) {
-        net.minecraft.core.BlockPos blockPos = entity.blockPosition();
-        if (level.isVillage(blockPos)) {
-            this.setRandomPos(entity);
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.PathfinderMob $$1, long $$2) {
+        net.minecraft.core.BlockPos $$3 = $$1.blockPosition();
+        if ($$0.isVillage($$3)) {
+            this.setRandomPos($$1);
         } else {
-            net.minecraft.core.SectionPos sectionPos = net.minecraft.core.SectionPos.of((net.minecraft.core.BlockPos)blockPos);
-            net.minecraft.core.SectionPos sectionPos2 = net.minecraft.world.entity.ai.behavior.BehaviorUtils.findSectionClosestToVillage((net.minecraft.server.level.ServerLevel)level, (net.minecraft.core.SectionPos)sectionPos, (int)2);
-            if (sectionPos2 != sectionPos) {
-                this.setTargetedPos(entity, sectionPos2);
+            net.minecraft.core.SectionPos $$4 = net.minecraft.core.SectionPos.of((net.minecraft.core.BlockPos)$$3);
+            net.minecraft.core.SectionPos $$5 = net.minecraft.world.entity.ai.behavior.BehaviorUtils.findSectionClosestToVillage((net.minecraft.server.level.ServerLevel)$$0, (net.minecraft.core.SectionPos)$$4, (int)2);
+            if ($$5 != $$4) {
+                this.setTargetedPos($$1, $$5);
             } else {
-                this.setRandomPos(entity);
+                this.setRandomPos($$1);
             }
         }
     }