--- com/mojang/blaze3d/shaders/AbstractUniform.java
+++ com/mojang/blaze3d/shaders/AbstractUniform.java
@@ -8,22 +8,22 @@
     public com.mojang.blaze3d.shaders.AbstractUniform() {
     }
 
-    public void set(float x) {
+    public void set(float $$0) {
     }
 
-    public void set(float x, float y) {
+    public void set(float $$0, float $$1) {
     }
 
-    public void set(float x, float y, float z) {
+    public void set(float $$0, float $$1, float $$2) {
     }
 
-    public void set(float x, float y, float z, float w) {
+    public void set(float $$0, float $$1, float $$2, float $$3) {
     }
 
-    public void setSafe(float x, float y, float z, float w) {
+    public void setSafe(float $$0, float $$1, float $$2, float $$3) {
     }
 
-    public void setSafe(int x, int y, int z, int w) {
+    public void setSafe(int $$0, int $$1, int $$2, int $$3) {
     }
 
     public void set(int $$0) {
@@ -38,7 +38,7 @@
     public void set(int $$0, int $$1, int $$2, int $$3) {
     }
 
-    public void set(float[] valueArray) {
+    public void set(float[] $$0) {
     }
 
     public void set(com.mojang.math.Vector3f $$0) {
@@ -74,7 +74,7 @@
     public void setMat4x4(float $$0, float $$1, float $$2, float $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9, float $$10, float $$11, float $$12, float $$13, float $$14, float $$15) {
     }
 
-    public void set(com.mojang.math.Matrix4f matrix) {
+    public void set(com.mojang.math.Matrix4f $$0) {
     }
 
     public void set(com.mojang.math.Matrix3f $$0) {