--- com/mojang/blaze3d/shaders/Program.java
+++ com/mojang/blaze3d/shaders/Program.java
@@ -10,10 +10,10 @@
     public java.lang.String name;
     public int id;
 
-    public com.mojang.blaze3d.shaders.Program(com.mojang.blaze3d.shaders.Program$Type type, int id, java.lang.String name) {
-        this.type = type;
-        this.id = id;
-        this.name = name;
+    public com.mojang.blaze3d.shaders.Program(com.mojang.blaze3d.shaders.Program$Type $$0, int $$1, java.lang.String $$2) {
+        this.type = $$0;
+        this.id = $$1;
+        this.name = $$2;
     }
 
     public void attachToShader(com.mojang.blaze3d.shaders.Shader $$0) {