--- com/mojang/blaze3d/shaders/Program$Type.java
+++ com/mojang/blaze3d/shaders/Program$Type.java
@@ -22,12 +22,12 @@
         return (com.mojang.blaze3d.shaders.Program$Type)java.lang.Enum.valueOf(com.mojang.blaze3d.shaders.Program$Type.class, (java.lang.String)$$0);
     }
 
-    public com.mojang.blaze3d.shaders.Program$Type(java.lang.String name, java.lang.String extension, int glType) {
+    public com.mojang.blaze3d.shaders.Program$Type(java.lang.String $$0, java.lang.String $$1, int $$2) {
         super(string, n);
         this.programs = com.google.common.collect.Maps.newHashMap();
-        this.name = name;
-        this.extension = extension;
-        this.glType = glType;
+        this.name = $$0;
+        this.extension = $$1;
+        this.glType = $$2;
     }
 
     public java.lang.String getName() {