--- net/minecraft/core/AxisCycle$3.java
+++ net/minecraft/core/AxisCycle$3.java
@@ -10,16 +10,16 @@
         super(string, n);
     }
 
-    public int cycle(int x, int y, int z, net.minecraft.core.Direction$Axis axis) {
-        return axis.choose(y, z, x);
+    public int cycle(int $$0, int $$1, int $$2, net.minecraft.core.Direction$Axis $$3) {
+        return $$3.choose($$1, $$2, $$0);
     }
 
     public double cycle(double $$0, double $$1, double $$2, net.minecraft.core.Direction$Axis $$3) {
         return $$3.choose($$1, $$2, $$0);
     }
 
-    public net.minecraft.core.Direction$Axis cycle(net.minecraft.core.Direction$Axis axis) {
-        return AXIS_VALUES[java.lang.Math.floorMod((int)(axis.ordinal() - 1), (int)3)];
+    public net.minecraft.core.Direction$Axis cycle(net.minecraft.core.Direction$Axis $$0) {
+        return AXIS_VALUES[java.lang.Math.floorMod((int)($$0.ordinal() - 1), (int)3)];
     }
 
     public net.minecraft.core.AxisCycle inverse() {