--- net/minecraft/core/AxisCycle$1.java
+++ net/minecraft/core/AxisCycle$1.java
@@ -10,16 +10,16 @@
         super(string, n);
     }
 
-    public int cycle(int x, int y, int z, net.minecraft.core.Direction$Axis axis) {
-        return axis.choose(x, y, z);
+    public int cycle(int $$0, int $$1, int $$2, net.minecraft.core.Direction$Axis $$3) {
+        return $$3.choose($$0, $$1, $$2);
     }
 
     public double cycle(double $$0, double $$1, double $$2, net.minecraft.core.Direction$Axis $$3) {
         return $$3.choose($$0, $$1, $$2);
     }
 
-    public net.minecraft.core.Direction$Axis cycle(net.minecraft.core.Direction$Axis axis) {
-        return axis;
+    public net.minecraft.core.Direction$Axis cycle(net.minecraft.core.Direction$Axis $$0) {
+        return $$0;
     }
 
     public net.minecraft.core.AxisCycle inverse() {