--- net/minecraft/core/Direction$AxisDirection.java
+++ net/minecraft/core/Direction$AxisDirection.java
@@ -20,10 +20,10 @@
         return (net.minecraft.core.Direction$AxisDirection)java.lang.Enum.valueOf(net.minecraft.core.Direction$AxisDirection.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.core.Direction$AxisDirection(int step, java.lang.String name) {
+    public net.minecraft.core.Direction$AxisDirection(int $$0, java.lang.String $$1) {
         super(string, n);
-        this.step = step;
-        this.name = name;
+        this.step = $$0;
+        this.name = $$1;
     }
 
     public int getStep() {