--- net/minecraft/core/Direction8.java
+++ net/minecraft/core/Direction8.java
@@ -26,12 +26,12 @@
         return (net.minecraft.core.Direction8)java.lang.Enum.valueOf(net.minecraft.core.Direction8.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.core.Direction8(net.minecraft.core.Direction ... directions) {
+    public net.minecraft.core.Direction8(net.minecraft.core.Direction ... $$0) {
         super(string, n);
-        this.directions = com.google.common.collect.Sets.immutableEnumSet((java.lang.Iterable)java.util.Arrays.asList((java.lang.Object[])directions));
+        this.directions = com.google.common.collect.Sets.immutableEnumSet((java.lang.Iterable)java.util.Arrays.asList((java.lang.Object[])$$0));
         this.step = new net.minecraft.core.Vec3i(0, 0, 0);
-        for (net.minecraft.core.Direction direction : directions) {
-            this.step.setX(this.step.getX() + direction.getStepX()).setY(this.step.getY() + direction.getStepY()).setZ(this.step.getZ() + direction.getStepZ());
+        for (net.minecraft.core.Direction $$1 : $$0) {
+            this.step.setX(this.step.getX() + $$1.getStepX()).setY(this.step.getY() + $$1.getStepY()).setZ(this.step.getZ() + $$1.getStepZ());
         }
     }
 