--- net/minecraft/core/Direction$Plane.java
+++ net/minecraft/core/Direction$Plane.java
@@ -22,10 +22,10 @@
         return (net.minecraft.core.Direction$Plane)java.lang.Enum.valueOf(net.minecraft.core.Direction$Plane.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.core.Direction$Plane(net.minecraft.core.Direction[] faces, net.minecraft.core.Direction$Axis[] axis) {
+    public net.minecraft.core.Direction$Plane(net.minecraft.core.Direction[] $$0, net.minecraft.core.Direction$Axis[] $$1) {
         super(string, n);
-        this.faces = faces;
-        this.axis = axis;
+        this.faces = $$0;
+        this.axis = $$1;
     }
 
     public net.minecraft.core.Direction getRandomDirection(net.minecraft.util.RandomSource $$0) {