--- net/minecraft/core/FrontAndTop.java
+++ net/minecraft/core/FrontAndTop.java
@@ -33,24 +33,24 @@
         return (net.minecraft.core.FrontAndTop)java.lang.Enum.valueOf(net.minecraft.core.FrontAndTop.class, (java.lang.String)$$0);
     }
 
-    public static int lookupKey(net.minecraft.core.Direction front, net.minecraft.core.Direction top) {
-        return top.ordinal() << 3 | front.ordinal();
+    public static int lookupKey(net.minecraft.core.Direction $$0, net.minecraft.core.Direction $$1) {
+        return $$1.ordinal() << 3 | $$0.ordinal();
     }
 
-    public net.minecraft.core.FrontAndTop(java.lang.String name, net.minecraft.core.Direction front, net.minecraft.core.Direction top) {
+    public net.minecraft.core.FrontAndTop(java.lang.String $$0, net.minecraft.core.Direction $$1, net.minecraft.core.Direction $$2) {
         super(string, n);
-        this.name = name;
-        this.front = front;
-        this.top = top;
+        this.name = $$0;
+        this.front = $$1;
+        this.top = $$2;
     }
 
     public java.lang.String getSerializedName() {
         return this.name;
     }
 
-    public static net.minecraft.core.FrontAndTop fromFrontAndTop(net.minecraft.core.Direction front, net.minecraft.core.Direction top) {
-        int n = net.minecraft.core.FrontAndTop.lookupKey((net.minecraft.core.Direction)front, (net.minecraft.core.Direction)top);
-        return (net.minecraft.core.FrontAndTop)LOOKUP_TOP_FRONT.get(n);
+    public static net.minecraft.core.FrontAndTop fromFrontAndTop(net.minecraft.core.Direction $$0, net.minecraft.core.Direction $$1) {
+        int $$2 = net.minecraft.core.FrontAndTop.lookupKey((net.minecraft.core.Direction)$$0, (net.minecraft.core.Direction)$$1);
+        return (net.minecraft.core.FrontAndTop)LOOKUP_TOP_FRONT.get($$2);
     }
 
     public net.minecraft.core.Direction front() {