--- net/minecraft/core/GlobalPos.java
+++ net/minecraft/core/GlobalPos.java
@@ -9,13 +9,13 @@
     public net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> dimension;
     public net.minecraft.core.BlockPos pos;
 
-    public net.minecraft.core.GlobalPos(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> dimension, net.minecraft.core.BlockPos pos) {
-        this.dimension = dimension;
-        this.pos = pos;
+    public net.minecraft.core.GlobalPos(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$0, net.minecraft.core.BlockPos $$1) {
+        this.dimension = $$0;
+        this.pos = $$1;
     }
 
-    public static net.minecraft.core.GlobalPos of(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> dimension, net.minecraft.core.BlockPos pos) {
-        return new net.minecraft.core.GlobalPos(dimension, pos);
+    public static net.minecraft.core.GlobalPos of(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$0, net.minecraft.core.BlockPos $$1) {
+        return new net.minecraft.core.GlobalPos($$0, $$1);
     }
 
     public net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> dimension() {
@@ -26,15 +26,15 @@
         return this.pos;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other == null || this.getClass() != other.getClass()) {
+        if ($$0 == null || this.getClass() != $$0.getClass()) {
             return false;
         }
-        net.minecraft.core.GlobalPos globalPos = (net.minecraft.core.GlobalPos)other;
-        return java.util.Objects.equals((java.lang.Object)this.dimension, (java.lang.Object)globalPos.dimension) && java.util.Objects.equals((java.lang.Object)this.pos, (java.lang.Object)globalPos.pos);
+        net.minecraft.core.GlobalPos $$1 = (net.minecraft.core.GlobalPos)$$0;
+        return java.util.Objects.equals((java.lang.Object)this.dimension, (java.lang.Object)$$1.dimension) && java.util.Objects.equals((java.lang.Object)this.pos, (java.lang.Object)$$1.pos);
     }
 
     public int hashCode() {