--- net/minecraft/core/HolderSet$Direct.java
+++ net/minecraft/core/HolderSet$Direct.java
@@ -22,6 +22,10 @@
         return com.mojang.datafixers.util.Either.right((java.lang.Object)this.contents);
     }
 
+    public java.util.Optional<net.minecraft.tags.TagKey<T>> unwrapKey() {
+        return java.util.Optional.empty();
+    }
+
     public boolean contains(net.minecraft.core.Holder<T> $$0) {
         if (this.contentsSet == null) {
             this.contentsSet = java.util.Set.copyOf((java.util.Collection)this.contents);