--- net/minecraft/core/HolderSet$Named.java
+++ net/minecraft/core/HolderSet$Named.java
@@ -32,6 +32,10 @@
         return com.mojang.datafixers.util.Either.left((java.lang.Object)this.key);
     }
 
+    public java.util.Optional<net.minecraft.tags.TagKey<T>> unwrapKey() {
+        return java.util.Optional.of((java.lang.Object)this.key);
+    }
+
     public boolean contains(net.minecraft.core.Holder<T> $$0) {
         return $$0.is(this.key);
     }