--- net/minecraft/core/Holder$Reference.java
+++ net/minecraft/core/Holder$Reference.java
@@ -92,6 +92,20 @@
         this.value = $$1;
     }
 
+    public void bindKey(net.minecraft.resources.ResourceKey<T> $$0) {
+        if (this.key != null && $$0 != this.key) {
+            throw new java.lang.IllegalStateException("Can't change holder key: existing=" + this.key + ", new=" + $$0);
+        }
+        this.key = $$0;
+    }
+
+    public void bindValue(T $$0) {
+        if (this.type == net.minecraft.core.Holder$Reference$Type.INTRUSIVE && this.value != $$0) {
+            throw new java.lang.IllegalStateException("Can't change holder " + this.key + " value: existing=" + this.value + ", new=" + $$0);
+        }
+        this.value = $$0;
+    }
+
     public void bindTags(java.util.Collection<net.minecraft.tags.TagKey<T>> $$0) {
         this.tags = java.util.Set.copyOf($$0);
     }