--- net/minecraft/core/PositionImpl.java
+++ net/minecraft/core/PositionImpl.java
@@ -10,10 +10,10 @@
     public double y;
     public double z;
 
-    public net.minecraft.core.PositionImpl(double x, double y, double z) {
-        this.x = x;
-        this.y = y;
-        this.z = z;
+    public net.minecraft.core.PositionImpl(double $$0, double $$1, double $$2) {
+        this.x = $$0;
+        this.y = $$1;
+        this.z = $$2;
     }
 
     public double x() {