--- net/minecraft/core/RegistryAccess$RegistryEntry.java
+++ net/minecraft/core/RegistryAccess$RegistryEntry.java
@@ -18,10 +18,6 @@
         return net.minecraft.core.RegistryAccess$RegistryEntry.fromUntyped((net.minecraft.resources.ResourceKey)((net.minecraft.resources.ResourceKey)$$0.getKey()), (net.minecraft.core.Registry)((net.minecraft.core.Registry)$$0.getValue()));
     }
 
-    public static <T> net.minecraft.core.RegistryAccess$RegistryEntry<T> fromHolder(net.minecraft.core.Holder$Reference<? extends net.minecraft.core.Registry<? extends T>> $$0) {
-        return net.minecraft.core.RegistryAccess$RegistryEntry.fromUntyped((net.minecraft.resources.ResourceKey)$$0.key(), (net.minecraft.core.Registry)((net.minecraft.core.Registry)$$0.value()));
-    }
-
     public static <T> net.minecraft.core.RegistryAccess$RegistryEntry<T> fromUntyped(net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<?>> $$0, net.minecraft.core.Registry<?> $$1) {
         return new net.minecraft.core.RegistryAccess$RegistryEntry($$0, $$1);
     }