--- net/minecraft/core/RegistryAccess$1.java
+++ net/minecraft/core/RegistryAccess$1.java
@@ -12,13 +12,17 @@
         this.val$registries = registry;
     }
 
-    public <T> java.util.Optional<net.minecraft.core.Registry<T>> ownedRegistry(net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<? extends T>> $$0) {
+    public <T> java.util.Optional<net.minecraft.core.Registry<T>> registry(net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<? extends T>> $$0) {
         net.minecraft.core.Registry $$1 = this.val$registries;
         return $$1.getOptional($$0);
     }
 
-    public java.util.stream.Stream<net.minecraft.core.RegistryAccess$RegistryEntry<?>> ownedRegistries() {
+    public java.util.stream.Stream<net.minecraft.core.RegistryAccess$RegistryEntry<?>> registries() {
         return this.val$registries.entrySet().stream().map(net.minecraft.core.RegistryAccess$RegistryEntry::fromMapEntry);
     }
+
+    public net.minecraft.core.RegistryAccess$Frozen freeze() {
+        return this;
+    }
 }
 