--- net/minecraft/core/RegistrySynchronization$NetworkedRegistryData.java
+++ net/minecraft/core/RegistrySynchronization$NetworkedRegistryData.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.core;
+
+public class net.minecraft.core.RegistrySynchronization$NetworkedRegistryData<E>
+extends java.lang.Record
+ {
+    public net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<E>> key;
+    public com.mojang.serialization.Codec<E> networkCodec;
+
+    public net.minecraft.core.RegistrySynchronization$NetworkedRegistryData(net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<E>> $$0, com.mojang.serialization.Codec<E> $$1) {
+        this.key = $$0;
+        this.networkCodec = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.core.RegistrySynchronization$NetworkedRegistryData.class, "key;networkCodec", "key", "networkCodec"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.core.RegistrySynchronization$NetworkedRegistryData.class, "key;networkCodec", "key", "networkCodec"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.core.RegistrySynchronization$NetworkedRegistryData.class, "key;networkCodec", "key", "networkCodec"}, this, $$0);
+    }
+
+    public net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<E>> key() {
+        return this.key;
+    }
+
+    public com.mojang.serialization.Codec<E> networkCodec() {
+        return this.networkCodec;
+    }
+}
+