--- net/minecraft/data/BlockFamily$Builder.java
+++ net/minecraft/data/BlockFamily$Builder.java
@@ -25,6 +25,11 @@
         return this;
     }
 
+    public net.minecraft.data.BlockFamily$Builder mosaic(net.minecraft.world.level.block.Block $$0) {
+        this.family.variants.put((java.lang.Object)net.minecraft.data.BlockFamily$Variant.MOSAIC, (java.lang.Object)$$0);
+        return this;
+    }
+
     public net.minecraft.data.BlockFamily$Builder cracked(net.minecraft.world.level.block.Block $$0) {
         this.family.variants.put((java.lang.Object)net.minecraft.data.BlockFamily$Variant.CRACKED, (java.lang.Object)$$0);
         return this;
@@ -40,11 +45,21 @@
         return this;
     }
 
+    public net.minecraft.data.BlockFamily$Builder customFence(net.minecraft.world.level.block.Block $$0) {
+        this.family.variants.put((java.lang.Object)net.minecraft.data.BlockFamily$Variant.CUSTOM_FENCE, (java.lang.Object)$$0);
+        return this;
+    }
+
     public net.minecraft.data.BlockFamily$Builder fence(net.minecraft.world.level.block.Block $$0) {
         this.family.variants.put((java.lang.Object)net.minecraft.data.BlockFamily$Variant.FENCE, (java.lang.Object)$$0);
         return this;
     }
 
+    public net.minecraft.data.BlockFamily$Builder customFenceGate(net.minecraft.world.level.block.Block $$0) {
+        this.family.variants.put((java.lang.Object)net.minecraft.data.BlockFamily$Variant.CUSTOM_FENCE_GATE, (java.lang.Object)$$0);
+        return this;
+    }
+
     public net.minecraft.data.BlockFamily$Builder fenceGate(net.minecraft.world.level.block.Block $$0) {
         this.family.variants.put((java.lang.Object)net.minecraft.data.BlockFamily$Variant.FENCE_GATE, (java.lang.Object)$$0);
         return this;
@@ -96,6 +111,11 @@
         return this;
     }
 
+    public net.minecraft.data.BlockFamily$Builder featureLockedBehind(net.minecraft.world.flag.FeatureFlag ... $$0) {
+        this.family.requiredFeatures = net.minecraft.world.flag.FeatureFlags.REGISTRY.subset($$0);
+        return this;
+    }
+
     public net.minecraft.data.BlockFamily$Builder recipeGroupPrefix(java.lang.String $$0) {
         this.family.recipeGroupPrefix = $$0;
         return this;