--- net/minecraft/data/BlockFamily$Variant.java
+++ net/minecraft/data/BlockFamily$Variant.java
@@ -11,8 +11,11 @@
     public static /* enum */ net.minecraft.data.BlockFamily$Variant CRACKED;
     public static /* enum */ net.minecraft.data.BlockFamily$Variant CUT;
     public static /* enum */ net.minecraft.data.BlockFamily$Variant DOOR;
+    public static /* enum */ net.minecraft.data.BlockFamily$Variant CUSTOM_FENCE;
     public static /* enum */ net.minecraft.data.BlockFamily$Variant FENCE;
+    public static /* enum */ net.minecraft.data.BlockFamily$Variant CUSTOM_FENCE_GATE;
     public static /* enum */ net.minecraft.data.BlockFamily$Variant FENCE_GATE;
+    public static /* enum */ net.minecraft.data.BlockFamily$Variant MOSAIC;
     public static /* enum */ net.minecraft.data.BlockFamily$Variant SIGN;
     public static /* enum */ net.minecraft.data.BlockFamily$Variant SLAB;
     public static /* enum */ net.minecraft.data.BlockFamily$Variant STAIRS;
@@ -42,7 +45,7 @@
     }
 
     public static /* synthetic */ net.minecraft.data.BlockFamily$Variant[] $values() {
-        return new net.minecraft.data.BlockFamily$Variant[]{BUTTON, CHISELED, CRACKED, CUT, DOOR, FENCE, FENCE_GATE, SIGN, SLAB, STAIRS, PRESSURE_PLATE, POLISHED, TRAPDOOR, WALL, WALL_SIGN};
+        return new net.minecraft.data.BlockFamily$Variant[]{BUTTON, CHISELED, CRACKED, CUT, DOOR, CUSTOM_FENCE, FENCE, CUSTOM_FENCE_GATE, FENCE_GATE, MOSAIC, SIGN, SLAB, STAIRS, PRESSURE_PLATE, POLISHED, TRAPDOOR, WALL, WALL_SIGN};
     }
 
     static {
@@ -51,16 +54,19 @@
         CRACKED = new net.minecraft.data.BlockFamily$Variant("CRACKED", 2, "cracked");
         CUT = new net.minecraft.data.BlockFamily$Variant("CUT", 3, "cut");
         DOOR = new net.minecraft.data.BlockFamily$Variant("DOOR", 4, "door");
-        FENCE = new net.minecraft.data.BlockFamily$Variant("FENCE", 5, "fence");
-        FENCE_GATE = new net.minecraft.data.BlockFamily$Variant("FENCE_GATE", 6, "fence_gate");
-        SIGN = new net.minecraft.data.BlockFamily$Variant("SIGN", 7, "sign");
-        SLAB = new net.minecraft.data.BlockFamily$Variant("SLAB", 8, "slab");
-        STAIRS = new net.minecraft.data.BlockFamily$Variant("STAIRS", 9, "stairs");
-        PRESSURE_PLATE = new net.minecraft.data.BlockFamily$Variant("PRESSURE_PLATE", 10, "pressure_plate");
-        POLISHED = new net.minecraft.data.BlockFamily$Variant("POLISHED", 11, "polished");
-        TRAPDOOR = new net.minecraft.data.BlockFamily$Variant("TRAPDOOR", 12, "trapdoor");
-        WALL = new net.minecraft.data.BlockFamily$Variant("WALL", 13, "wall");
-        WALL_SIGN = new net.minecraft.data.BlockFamily$Variant("WALL_SIGN", 14, "wall_sign");
+        CUSTOM_FENCE = new net.minecraft.data.BlockFamily$Variant("CUSTOM_FENCE", 5, "custom_fence");
+        FENCE = new net.minecraft.data.BlockFamily$Variant("FENCE", 6, "fence");
+        CUSTOM_FENCE_GATE = new net.minecraft.data.BlockFamily$Variant("CUSTOM_FENCE_GATE", 7, "custom_fence_gate");
+        FENCE_GATE = new net.minecraft.data.BlockFamily$Variant("FENCE_GATE", 8, "fence_gate");
+        MOSAIC = new net.minecraft.data.BlockFamily$Variant("MOSAIC", 9, "mosaic");
+        SIGN = new net.minecraft.data.BlockFamily$Variant("SIGN", 10, "sign");
+        SLAB = new net.minecraft.data.BlockFamily$Variant("SLAB", 11, "slab");
+        STAIRS = new net.minecraft.data.BlockFamily$Variant("STAIRS", 12, "stairs");
+        PRESSURE_PLATE = new net.minecraft.data.BlockFamily$Variant("PRESSURE_PLATE", 13, "pressure_plate");
+        POLISHED = new net.minecraft.data.BlockFamily$Variant("POLISHED", 14, "polished");
+        TRAPDOOR = new net.minecraft.data.BlockFamily$Variant("TRAPDOOR", 15, "trapdoor");
+        WALL = new net.minecraft.data.BlockFamily$Variant("WALL", 16, "wall");
+        WALL_SIGN = new net.minecraft.data.BlockFamily$Variant("WALL_SIGN", 17, "wall_sign");
         $VALUES = net.minecraft.data.BlockFamily$Variant.$values();
     }
 }