--- net/minecraft/data/Main.java
+++ net/minecraft/data/Main.java
@@ -9,62 +9,73 @@
     }
 
     @net.minecraft.obfuscate.DontObfuscate
-    public static void main(java.lang.String[] args) throws java.io.IOException {
+    public static void main(java.lang.String[] $$02) throws java.io.IOException {
         net.minecraft.SharedConstants.tryDetectVersion();
-        joptsimple.OptionParser optionParser = new joptsimple.OptionParser();
-        joptsimple.AbstractOptionSpec abstractOptionSpec = optionParser.accepts("help", "Show the help menu").forHelp();
-        joptsimple.OptionSpecBuilder optionSpecBuilder = optionParser.accepts("server", "Include server generators");
-        joptsimple.OptionSpecBuilder optionSpecBuilder2 = optionParser.accepts("client", "Include client generators");
-        joptsimple.OptionSpecBuilder optionSpecBuilder3 = optionParser.accepts("dev", "Include development tools");
-        joptsimple.OptionSpecBuilder optionSpecBuilder4 = optionParser.accepts("reports", "Include data reports");
-        joptsimple.OptionSpecBuilder optionSpecBuilder5 = optionParser.accepts("validate", "Validate inputs");
-        joptsimple.OptionSpecBuilder optionSpecBuilder6 = optionParser.accepts("all", "Include all generators");
-        joptsimple.ArgumentAcceptingOptionSpec argumentAcceptingOptionSpec = optionParser.accepts("output", "Output folder").withRequiredArg().defaultsTo((java.lang.Object)"generated", (java.lang.Object[])new java.lang.String[0]);
-        joptsimple.ArgumentAcceptingOptionSpec argumentAcceptingOptionSpec2 = optionParser.accepts("input", "Input folder").withRequiredArg();
-        joptsimple.OptionSet optionSet = optionParser.parse(args);
-        if (optionSet.has((joptsimple.OptionSpec)abstractOptionSpec) || !optionSet.hasOptions()) {
-            optionParser.printHelpOn((java.io.OutputStream)java.lang.System.out);
+        joptsimple.OptionParser $$1 = new joptsimple.OptionParser();
+        joptsimple.AbstractOptionSpec $$2 = $$1.accepts("help", "Show the help menu").forHelp();
+        joptsimple.OptionSpecBuilder $$3 = $$1.accepts("server", "Include server generators");
+        joptsimple.OptionSpecBuilder $$4 = $$1.accepts("client", "Include client generators");
+        joptsimple.OptionSpecBuilder $$5 = $$1.accepts("dev", "Include development tools");
+        joptsimple.OptionSpecBuilder $$6 = $$1.accepts("reports", "Include data reports");
+        joptsimple.OptionSpecBuilder $$7 = $$1.accepts("validate", "Validate inputs");
+        joptsimple.OptionSpecBuilder $$8 = $$1.accepts("all", "Include all generators");
+        joptsimple.ArgumentAcceptingOptionSpec $$9 = $$1.accepts("output", "Output folder").withRequiredArg().defaultsTo((java.lang.Object)"generated", (java.lang.Object[])new java.lang.String[0]);
+        joptsimple.ArgumentAcceptingOptionSpec $$10 = $$1.accepts("input", "Input folder").withRequiredArg();
+        joptsimple.OptionSet $$11 = $$1.parse($$02);
+        if ($$11.has((joptsimple.OptionSpec)$$2) || !$$11.hasOptions()) {
+            $$1.printHelpOn((java.io.OutputStream)java.lang.System.out);
             return;
         }
-        java.nio.file.Path path = java.nio.file.Paths.get((java.lang.String)((java.lang.String)argumentAcceptingOptionSpec.value(optionSet)), (java.lang.String[])new java.lang.String[0]);
-        boolean bl = optionSet.has((joptsimple.OptionSpec)optionSpecBuilder6);
-        boolean bl2 = bl || optionSet.has((joptsimple.OptionSpec)optionSpecBuilder2);
-        boolean bl3 = bl || optionSet.has((joptsimple.OptionSpec)optionSpecBuilder);
-        boolean bl4 = bl || optionSet.has((joptsimple.OptionSpec)optionSpecBuilder3);
-        boolean bl5 = bl || optionSet.has((joptsimple.OptionSpec)optionSpecBuilder4);
-        boolean bl6 = bl || optionSet.has((joptsimple.OptionSpec)optionSpecBuilder5);
-        net.minecraft.data.DataGenerator dataGenerator = net.minecraft.data.Main.createStandardGenerator((java.nio.file.Path)path, (java.util.Collection)((java.util.Collection)optionSet.valuesOf((joptsimple.OptionSpec)argumentAcceptingOptionSpec2).stream().map($$0 -> java.nio.file.Paths.get((java.lang.String)$$0, (java.lang.String[])new java.lang.String[0])).collect(java.util.stream.Collectors.toList())), (boolean)bl2, (boolean)bl3, (boolean)bl4, (boolean)bl5, (boolean)bl6, (net.minecraft.WorldVersion)net.minecraft.SharedConstants.getCurrentVersion(), (boolean)true);
-        dataGenerator.run();
+        java.nio.file.Path $$12 = java.nio.file.Paths.get((java.lang.String)((java.lang.String)$$9.value($$11)), (java.lang.String[])new java.lang.String[0]);
+        boolean $$13 = $$11.has((joptsimple.OptionSpec)$$8);
+        boolean $$14 = $$13 || $$11.has((joptsimple.OptionSpec)$$4);
+        boolean $$15 = $$13 || $$11.has((joptsimple.OptionSpec)$$3);
+        boolean $$16 = $$13 || $$11.has((joptsimple.OptionSpec)$$5);
+        boolean $$17 = $$13 || $$11.has((joptsimple.OptionSpec)$$6);
+        boolean $$18 = $$13 || $$11.has((joptsimple.OptionSpec)$$7);
+        net.minecraft.data.DataGenerator $$19 = net.minecraft.data.Main.createStandardGenerator((java.nio.file.Path)$$12, (java.util.Collection)((java.util.Collection)$$11.valuesOf((joptsimple.OptionSpec)$$10).stream().map($$0 -> java.nio.file.Paths.get((java.lang.String)$$0, (java.lang.String[])new java.lang.String[0])).collect(java.util.stream.Collectors.toList())), (boolean)$$14, (boolean)$$15, (boolean)$$16, (boolean)$$17, (boolean)$$18, (net.minecraft.WorldVersion)net.minecraft.SharedConstants.getCurrentVersion(), (boolean)true);
+        $$19.run();
     }
 
     public static net.minecraft.data.DataGenerator createStandardGenerator(java.nio.file.Path $$0, java.util.Collection<java.nio.file.Path> $$1, boolean $$2, boolean $$3, boolean $$4, boolean $$5, boolean $$6, net.minecraft.WorldVersion $$7, boolean $$8) {
-        net.minecraft.data.DataGenerator $$9 = new net.minecraft.data.DataGenerator($$0, $$1, $$7, $$8);
-        $$9.addProvider($$2 || $$3, (net.minecraft.data.DataProvider)new net.minecraft.data.structures.SnbtToNbt($$9).addFilter((net.minecraft.data.structures.SnbtToNbt$Filter)new net.minecraft.data.structures.StructureUpdater()));
-        $$9.addProvider($$2, (net.minecraft.data.DataProvider)new net.minecraft.data.models.ModelProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.advancements.AdvancementProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.loot.LootTableProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.recipes.RecipeProvider($$9));
-        net.minecraft.data.tags.BlockTagsProvider $$10 = new net.minecraft.data.tags.BlockTagsProvider($$9);
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)$$10);
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.ItemTagsProvider($$9, $$10));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.BannerPatternTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.BiomeTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.CatVariantTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.EntityTypeTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.FlatLevelGeneratorPresetTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.FluidTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.GameEventTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.InstrumentTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.PaintingVariantTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.PoiTypeTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.StructureTagsProvider($$9));
-        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.WorldPresetTagsProvider($$9));
-        $$9.addProvider($$4, (net.minecraft.data.DataProvider)new net.minecraft.data.structures.NbtToSnbt($$9));
-        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.BiomeParametersDumpReport($$9));
-        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.BlockListReport($$9));
-        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.CommandsReport($$9));
-        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.RegistryDumpReport($$9));
-        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.WorldgenRegistryDumpReport($$9));
+        net.minecraft.data.DataGenerator $$9 = new net.minecraft.data.DataGenerator($$0, $$7, $$8);
+        net.minecraft.data.PackOutput $$10 = $$9.getVanillaPackOutput();
+        $$9.addProvider($$2 || $$3, (net.minecraft.data.DataProvider)new net.minecraft.data.structures.SnbtToNbt($$10, $$1).addFilter((net.minecraft.data.structures.SnbtToNbt$Filter)new net.minecraft.data.structures.StructureUpdater()));
+        $$9.addProvider($$2, (net.minecraft.data.DataProvider)new net.minecraft.data.models.ModelProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.worldgen.BuiltinRegistriesDatapackGenerator($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)net.minecraft.data.advancements.packs.VanillaAdvancementProvider.create((net.minecraft.data.PackOutput)$$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)net.minecraft.data.loot.packs.VanillaLootTableProvider.create((net.minecraft.data.PackOutput)$$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.recipes.packs.VanillaRecipeProvider($$10));
+        net.minecraft.data.tags.VanillaBlockTagsProvider $$11 = new net.minecraft.data.tags.VanillaBlockTagsProvider($$10);
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)$$11);
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.VanillaItemTagsProvider($$10, (net.minecraft.data.tags.TagsProvider)$$11));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.BannerPatternTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.BiomeTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.CatVariantTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.EntityTypeTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.FlatLevelGeneratorPresetTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.FluidTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.GameEventTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.InstrumentTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.PaintingVariantTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.PoiTypeTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.StructureTagsProvider($$10));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.WorldPresetTagsProvider($$10));
+        $$9.addProvider($$4, (net.minecraft.data.DataProvider)new net.minecraft.data.structures.NbtToSnbt($$10, $$1));
+        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.BiomeParametersDumpReport($$10));
+        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.BlockListReport($$10));
+        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.CommandsReport($$10));
+        $$9.addProvider($$5, (net.minecraft.data.DataProvider)new net.minecraft.data.info.RegistryDumpReport($$10));
+        net.minecraft.data.PackOutput $$12 = $$9.createBuiltinDatapackOutput("bundle");
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.recipes.packs.BundleRecipeProvider($$12));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)net.minecraft.data.metadata.PackMetadataGenerator.forFeaturePack((net.minecraft.data.PackOutput)$$12, (java.lang.String)"bundle", (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"dataPack.bundle.description"), (net.minecraft.world.flag.FeatureFlagSet)net.minecraft.world.flag.FeatureFlagSet.of((net.minecraft.world.flag.FeatureFlag)net.minecraft.world.flag.FeatureFlags.BUNDLE)));
+        net.minecraft.data.PackOutput $$13 = $$9.createBuiltinDatapackOutput("update_1_20");
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.recipes.packs.UpdateOneTwentyRecipeProvider($$13));
+        net.minecraft.data.tags.UpdateOneTwentyBlockTagsProvider $$14 = new net.minecraft.data.tags.UpdateOneTwentyBlockTagsProvider($$13);
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)$$14);
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)new net.minecraft.data.tags.UpdateOneTwentyItemTagsProvider($$13, (net.minecraft.data.tags.TagsProvider)$$14));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)net.minecraft.data.loot.packs.UpdateOneTwentyLootTableProvider.create((net.minecraft.data.PackOutput)$$13));
+        $$9.addProvider($$3, (net.minecraft.data.DataProvider)net.minecraft.data.metadata.PackMetadataGenerator.forFeaturePack((net.minecraft.data.PackOutput)$$13, (java.lang.String)"update_1_20", (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"dataPack.update_1_20.description"), (net.minecraft.world.flag.FeatureFlagSet)net.minecraft.world.flag.FeatureFlagSet.of((net.minecraft.world.flag.FeatureFlag)net.minecraft.world.flag.FeatureFlags.UPDATE_1_20)));
         return $$9;
     }
 }