--- net/minecraft/data/PackOutput$Target.java
+++ net/minecraft/data/PackOutput$Target.java
@@ -1,0 +1,39 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data;
+
+public class net.minecraft.data.PackOutput$Target
+extends java.lang.Enum<net.minecraft.data.PackOutput$Target>
+ {
+    public static /* enum */ net.minecraft.data.PackOutput$Target DATA_PACK;
+    public static /* enum */ net.minecraft.data.PackOutput$Target RESOURCE_PACK;
+    public static /* enum */ net.minecraft.data.PackOutput$Target REPORTS;
+    public java.lang.String directory;
+    public static /* synthetic */ net.minecraft.data.PackOutput$Target[] $VALUES;
+
+    public static net.minecraft.data.PackOutput$Target[] values() {
+        return (net.minecraft.data.PackOutput$Target[])$VALUES.clone();
+    }
+
+    public static net.minecraft.data.PackOutput$Target valueOf(java.lang.String $$0) {
+        return (net.minecraft.data.PackOutput$Target)java.lang.Enum.valueOf(net.minecraft.data.PackOutput$Target.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.data.PackOutput$Target(java.lang.String $$0) {
+        super(string, n);
+        this.directory = $$0;
+    }
+
+    public static /* synthetic */ net.minecraft.data.PackOutput$Target[] $values() {
+        return new net.minecraft.data.PackOutput$Target[]{DATA_PACK, RESOURCE_PACK, REPORTS};
+    }
+
+    static {
+        DATA_PACK = new net.minecraft.data.PackOutput$Target("DATA_PACK", 0, "data");
+        RESOURCE_PACK = new net.minecraft.data.PackOutput$Target("RESOURCE_PACK", 1, "assets");
+        REPORTS = new net.minecraft.data.PackOutput$Target("REPORTS", 2, "reports");
+        $VALUES = net.minecraft.data.PackOutput$Target.$values();
+    }
+}
+