--- net/minecraft/data/PackOutput$PathProvider.java
+++ net/minecraft/data/PackOutput$PathProvider.java
@@ -1,0 +1,24 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data;
+
+public class net.minecraft.data.PackOutput$PathProvider
+ {
+    public java.nio.file.Path root;
+    public java.lang.String kind;
+
+    public net.minecraft.data.PackOutput$PathProvider(net.minecraft.data.PackOutput $$0, net.minecraft.data.PackOutput$Target $$1, java.lang.String $$2) {
+        this.root = $$0.getOutputFolder($$1);
+        this.kind = $$2;
+    }
+
+    public java.nio.file.Path file(net.minecraft.resources.ResourceLocation $$0, java.lang.String $$1) {
+        return this.root.resolve($$0.getNamespace()).resolve(this.kind).resolve($$0.getPath() + "." + $$1);
+    }
+
+    public java.nio.file.Path json(net.minecraft.resources.ResourceLocation $$0) {
+        return this.root.resolve($$0.getNamespace()).resolve(this.kind).resolve($$0.getPath() + ".json");
+    }
+}
+