--- com/mojang/blaze3d/font/TrueTypeGlyphProvider$Glyph.java
+++ com/mojang/blaze3d/font/TrueTypeGlyphProvider$Glyph.java
@@ -14,14 +14,14 @@
     public int index;
     public /* synthetic */ com.mojang.blaze3d.font.TrueTypeGlyphProvider this$0;
 
-    public com.mojang.blaze3d.font.TrueTypeGlyphProvider$Glyph(com.mojang.blaze3d.font.TrueTypeGlyphProvider trueTypeGlyphProvider, int minWidth, int maxWidth, int minHeight, int maxHeight, float advance, float bearingX, int index) {
+    public com.mojang.blaze3d.font.TrueTypeGlyphProvider$Glyph(com.mojang.blaze3d.font.TrueTypeGlyphProvider trueTypeGlyphProvider, int $$0, int $$1, int $$2, int $$3, float $$4, float $$5, int $$6) {
         this.this$0 = trueTypeGlyphProvider;
-        this.width = maxWidth - minWidth;
-        this.height = minHeight - maxHeight;
-        this.advance = advance / trueTypeGlyphProvider.oversample;
-        this.bearingX = (bearingX + (float)minWidth + trueTypeGlyphProvider.shiftX) / trueTypeGlyphProvider.oversample;
-        this.bearingY = (trueTypeGlyphProvider.ascent - (float)minHeight + trueTypeGlyphProvider.shiftY) / trueTypeGlyphProvider.oversample;
-        this.index = index;
+        this.width = $$1 - $$0;
+        this.height = $$2 - $$3;
+        this.advance = $$4 / trueTypeGlyphProvider.oversample;
+        this.bearingX = ($$5 + (float)$$0 + trueTypeGlyphProvider.shiftX) / trueTypeGlyphProvider.oversample;
+        this.bearingY = (trueTypeGlyphProvider.ascent - (float)$$2 + trueTypeGlyphProvider.shiftY) / trueTypeGlyphProvider.oversample;
+        this.index = $$6;
     }
 
     public float getAdvance() {