--- net/minecraft/world/entity/vehicle/Boat$1.java
+++ net/minecraft/world/entity/vehicle/Boat$1.java
@@ -42,43 +42,43 @@
         }
         $SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type = new int[net.minecraft.world.entity.vehicle.Boat$Type.values().length];
         try {
-            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.OAK.ordinal()] = 1;
+            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.SPRUCE.ordinal()] = 1;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
         try {
-            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.SPRUCE.ordinal()] = 2;
+            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.BIRCH.ordinal()] = 2;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
         try {
-            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.BIRCH.ordinal()] = 3;
+            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.JUNGLE.ordinal()] = 3;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
         try {
-            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.JUNGLE.ordinal()] = 4;
+            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.ACACIA.ordinal()] = 4;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
         try {
-            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.ACACIA.ordinal()] = 5;
+            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.DARK_OAK.ordinal()] = 5;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
         try {
-            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.DARK_OAK.ordinal()] = 6;
+            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.MANGROVE.ordinal()] = 6;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
         try {
-            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.MANGROVE.ordinal()] = 7;
+            net.minecraft.world.entity.vehicle.Boat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.BAMBOO.ordinal()] = 7;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block