--- net/minecraft/world/entity/vehicle/Boat$Type.java
+++ net/minecraft/world/entity/vehicle/Boat$Type.java
@@ -13,6 +13,7 @@
     public static /* enum */ net.minecraft.world.entity.vehicle.Boat$Type ACACIA;
     public static /* enum */ net.minecraft.world.entity.vehicle.Boat$Type DARK_OAK;
     public static /* enum */ net.minecraft.world.entity.vehicle.Boat$Type MANGROVE;
+    public static /* enum */ net.minecraft.world.entity.vehicle.Boat$Type BAMBOO;
     public java.lang.String name;
     public net.minecraft.world.level.block.Block planks;
     public static /* synthetic */ net.minecraft.world.entity.vehicle.Boat$Type[] $VALUES;
@@ -25,10 +26,10 @@
         return (net.minecraft.world.entity.vehicle.Boat$Type)java.lang.Enum.valueOf(net.minecraft.world.entity.vehicle.Boat$Type.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.vehicle.Boat$Type(net.minecraft.world.level.block.Block name, java.lang.String planks) {
+    public net.minecraft.world.entity.vehicle.Boat$Type(net.minecraft.world.level.block.Block $$0, java.lang.String $$1) {
         super(string, n);
-        this.name = planks;
-        this.planks = name;
+        this.name = $$1;
+        this.planks = $$0;
     }
 
     public java.lang.String getName() {
@@ -43,25 +44,25 @@
         return this.name;
     }
 
-    public static net.minecraft.world.entity.vehicle.Boat$Type byId(int id) {
-        net.minecraft.world.entity.vehicle.Boat$Type[] typeArray = net.minecraft.world.entity.vehicle.Boat$Type.values();
-        if (id < 0 || id >= typeArray.length) {
-            id = 0;
+    public static net.minecraft.world.entity.vehicle.Boat$Type byId(int $$0) {
+        net.minecraft.world.entity.vehicle.Boat$Type[] $$1 = net.minecraft.world.entity.vehicle.Boat$Type.values();
+        if ($$0 < 0 || $$0 >= $$1.length) {
+            $$0 = 0;
         }
-        return typeArray[id];
+        return $$1[$$0];
     }
 
-    public static net.minecraft.world.entity.vehicle.Boat$Type byName(java.lang.String name) {
-        net.minecraft.world.entity.vehicle.Boat$Type[] typeArray = net.minecraft.world.entity.vehicle.Boat$Type.values();
-        for (int i = 0; i < typeArray.length; ++i) {
-            if (!typeArray[i].getName().equals((java.lang.Object)name)) continue;
-            return typeArray[i];
+    public static net.minecraft.world.entity.vehicle.Boat$Type byName(java.lang.String $$0) {
+        net.minecraft.world.entity.vehicle.Boat$Type[] $$1 = net.minecraft.world.entity.vehicle.Boat$Type.values();
+        for (int $$2 = 0; $$2 < $$1.length; ++$$2) {
+            if (!$$1[$$2].getName().equals((java.lang.Object)$$0)) continue;
+            return $$1[$$2];
         }
-        return typeArray[0];
+        return $$1[0];
     }
 
     public static /* synthetic */ net.minecraft.world.entity.vehicle.Boat$Type[] $values() {
-        return new net.minecraft.world.entity.vehicle.Boat$Type[]{OAK, SPRUCE, BIRCH, JUNGLE, ACACIA, DARK_OAK, MANGROVE};
+        return new net.minecraft.world.entity.vehicle.Boat$Type[]{OAK, SPRUCE, BIRCH, JUNGLE, ACACIA, DARK_OAK, MANGROVE, BAMBOO};
     }
 
     static {
@@ -72,6 +73,7 @@
         ACACIA = new net.minecraft.world.entity.vehicle.Boat$Type("ACACIA", 4, net.minecraft.world.level.block.Blocks.ACACIA_PLANKS, "acacia");
         DARK_OAK = new net.minecraft.world.entity.vehicle.Boat$Type("DARK_OAK", 5, net.minecraft.world.level.block.Blocks.DARK_OAK_PLANKS, "dark_oak");
         MANGROVE = new net.minecraft.world.entity.vehicle.Boat$Type("MANGROVE", 6, net.minecraft.world.level.block.Blocks.MANGROVE_PLANKS, "mangrove");
+        BAMBOO = new net.minecraft.world.entity.vehicle.Boat$Type("BAMBOO", 7, net.minecraft.world.level.block.Blocks.BAMBOO_PLANKS, "bamboo");
         $VALUES = net.minecraft.world.entity.vehicle.Boat$Type.$values();
     }
 }