--- net/minecraft/world/entity/vehicle/ChestBoat$1.java
+++ net/minecraft/world/entity/vehicle/ChestBoat$1.java
@@ -45,6 +45,12 @@
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
+        try {
+            net.minecraft.world.entity.vehicle.ChestBoat$1.$SwitchMap$net$minecraft$world$entity$vehicle$Boat$Type[net.minecraft.world.entity.vehicle.Boat$Type.BAMBOO.ordinal()] = 7;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
     }
 }
 