--- net/minecraft/world/entity/vehicle/ChestBoat.java
+++ net/minecraft/world/entity/vehicle/ChestBoat.java
@@ -80,6 +80,7 @@
             case 4 -> net.minecraft.world.item.Items.ACACIA_CHEST_BOAT;
             case 5 -> net.minecraft.world.item.Items.DARK_OAK_CHEST_BOAT;
             case 6 -> net.minecraft.world.item.Items.MANGROVE_CHEST_BOAT;
+            case 7 -> net.minecraft.world.item.Items.BAMBOO_CHEST_RAFT;
             default -> net.minecraft.world.item.Items.OAK_CHEST_BOAT;
         };
     }