--- net/minecraft/world/entity/vehicle/MinecartCommandBlock.java
+++ net/minecraft/world/entity/vehicle/MinecartCommandBlock.java
@@ -17,8 +17,8 @@
         this.commandBlock = new net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase(this);
     }
 
-    public net.minecraft.world.entity.vehicle.MinecartCommandBlock(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.COMMAND_BLOCK_MINECART, level, x, y, z);
+    public net.minecraft.world.entity.vehicle.MinecartCommandBlock(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.COMMAND_BLOCK_MINECART, $$0, $$1, $$2, $$3);
         this.commandBlock = new net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase(this);
     }
 
@@ -32,16 +32,16 @@
         this.getEntityData().define(DATA_ID_LAST_OUTPUT, (java.lang.Object)net.minecraft.network.chat.CommonComponents.EMPTY);
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.commandBlock.load(compound);
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.commandBlock.load($$0);
         this.getEntityData().set(DATA_ID_COMMAND_NAME, (java.lang.Object)this.getCommandBlock().getCommand());
         this.getEntityData().set(DATA_ID_LAST_OUTPUT, (java.lang.Object)this.getCommandBlock().getLastOutput());
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        this.commandBlock.save(compound);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        this.commandBlock.save($$0);
     }
 
     public net.minecraft.world.entity.vehicle.AbstractMinecart$Type getMinecartType() {
@@ -56,25 +56,25 @@
         return this.commandBlock;
     }
 
-    public void activateMinecart(int x, int y, int z, boolean receivingPower) {
-        if (receivingPower && this.tickCount - this.lastActivated >= 4) {
+    public void activateMinecart(int $$0, int $$1, int $$2, boolean $$3) {
+        if ($$3 && this.tickCount - this.lastActivated >= 4) {
             this.getCommandBlock().performCommand(this.level);
             this.lastActivated = this.tickCount;
         }
     }
 
-    public net.minecraft.world.InteractionResult interact(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        return this.commandBlock.usedBy(player);
+    public net.minecraft.world.InteractionResult interact(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$1) {
+        return this.commandBlock.usedBy($$0);
     }
 
-    public void onSyncedDataUpdated(net.minecraft.network.syncher.EntityDataAccessor<?> key) {
-        super.onSyncedDataUpdated(key);
-        if (DATA_ID_LAST_OUTPUT.equals(key)) {
+    public void onSyncedDataUpdated(net.minecraft.network.syncher.EntityDataAccessor<?> $$0) {
+        super.onSyncedDataUpdated($$0);
+        if (DATA_ID_LAST_OUTPUT.equals($$0)) {
             try {
                 this.commandBlock.setLastOutput((net.minecraft.network.chat.Component)this.getEntityData().get(DATA_ID_LAST_OUTPUT));
             }
             catch (java.lang.Throwable throwable) {}
-        } else if (DATA_ID_COMMAND_NAME.equals(key)) {
+        } else if (DATA_ID_COMMAND_NAME.equals($$0)) {
             this.commandBlock.setCommand((java.lang.String)this.getEntityData().get(DATA_ID_COMMAND_NAME));
         }
     }