--- net/minecraft/world/entity/vehicle/Minecart.java
+++ net/minecraft/world/entity/vehicle/Minecart.java
@@ -10,19 +10,19 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.vehicle.Minecart(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.MINECART, level, x, y, z);
+    public net.minecraft.world.entity.vehicle.Minecart(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.MINECART, $$0, $$1, $$2, $$3);
     }
 
-    public net.minecraft.world.InteractionResult interact(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        if (player.isSecondaryUseActive()) {
+    public net.minecraft.world.InteractionResult interact(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$1) {
+        if ($$0.isSecondaryUseActive()) {
             return net.minecraft.world.InteractionResult.PASS;
         }
         if (this.isVehicle()) {
             return net.minecraft.world.InteractionResult.PASS;
         }
         if (!this.level.isClientSide) {
-            return player.startRiding((net.minecraft.world.entity.Entity)this) ? net.minecraft.world.InteractionResult.CONSUME : net.minecraft.world.InteractionResult.PASS;
+            return $$0.startRiding((net.minecraft.world.entity.Entity)this) ? net.minecraft.world.InteractionResult.CONSUME : net.minecraft.world.InteractionResult.PASS;
         }
         return net.minecraft.world.InteractionResult.SUCCESS;
     }
@@ -31,8 +31,8 @@
         return net.minecraft.world.item.Items.MINECART;
     }
 
-    public void activateMinecart(int x, int y, int z, boolean receivingPower) {
-        if (receivingPower) {
+    public void activateMinecart(int $$0, int $$1, int $$2, boolean $$3) {
+        if ($$3) {
             if (this.isVehicle()) {
                 this.ejectPassengers();
             }