--- net/minecraft/world/entity/vehicle/MinecartChest.java
+++ net/minecraft/world/entity/vehicle/MinecartChest.java
@@ -10,8 +10,8 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.vehicle.MinecartChest(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.CHEST_MINECART, x, y, z, level);
+    public net.minecraft.world.entity.vehicle.MinecartChest(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.CHEST_MINECART, $$1, $$2, $$3, $$0);
     }
 
     public net.minecraft.world.item.Item getDropItem() {
@@ -34,8 +34,8 @@
         return 8;
     }
 
-    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int id, net.minecraft.world.entity.player.Inventory playerInventory) {
-        return net.minecraft.world.inventory.ChestMenu.threeRows((int)id, (net.minecraft.world.entity.player.Inventory)playerInventory, (net.minecraft.world.Container)this);
+    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return net.minecraft.world.inventory.ChestMenu.threeRows((int)$$0, (net.minecraft.world.entity.player.Inventory)$$1, (net.minecraft.world.Container)this);
     }
 }
 