--- net/minecraft/world/entity/vehicle/MinecartHopper.java
+++ net/minecraft/world/entity/vehicle/MinecartHopper.java
@@ -19,8 +19,8 @@
         this.lastPosition = net.minecraft.core.BlockPos.ZERO;
     }
 
-    public net.minecraft.world.entity.vehicle.MinecartHopper(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.HOPPER_MINECART, x, y, z, level);
+    public net.minecraft.world.entity.vehicle.MinecartHopper(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.HOPPER_MINECART, $$1, $$2, $$3, $$0);
         this.enabled = true;
         this.cooldownTime = -1;
         this.lastPosition = net.minecraft.core.BlockPos.ZERO;
@@ -42,11 +42,11 @@
         return 5;
     }
 
-    public void activateMinecart(int x, int y, int z, boolean receivingPower) {
-        boolean bl;
-        boolean bl2 = bl = !receivingPower;
-        if (bl != this.isEnabled()) {
-            this.setEnabled(bl);
+    public void activateMinecart(int $$0, int $$1, int $$2, boolean $$3) {
+        boolean $$4;
+        boolean bl = $$4 = !$$3;
+        if ($$4 != this.isEnabled()) {
+            this.setEnabled($$4);
         }
     }
 
@@ -54,8 +54,8 @@
         return this.enabled;
     }
 
-    public void setEnabled(boolean enabled) {
-        this.enabled = enabled;
+    public void setEnabled(boolean $$0) {
+        this.enabled = $$0;
     }
 
     public double getLevelX() {
@@ -104,28 +104,28 @@
         return net.minecraft.world.item.Items.HOPPER_MINECART;
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("TransferCooldown", this.cooldownTime);
-        compound.putBoolean("Enabled", this.enabled);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("TransferCooldown", this.cooldownTime);
+        $$0.putBoolean("Enabled", this.enabled);
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.cooldownTime = compound.getInt("TransferCooldown");
-        this.enabled = compound.contains("Enabled") ? compound.getBoolean("Enabled") : true;
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.cooldownTime = $$0.getInt("TransferCooldown");
+        this.enabled = $$0.contains("Enabled") ? $$0.getBoolean("Enabled") : true;
     }
 
-    public void setCooldown(int cooldownTime) {
-        this.cooldownTime = cooldownTime;
+    public void setCooldown(int $$0) {
+        this.cooldownTime = $$0;
     }
 
     public boolean isOnCooldown() {
         return this.cooldownTime > 0;
     }
 
-    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int id, net.minecraft.world.entity.player.Inventory playerInventory) {
-        return new net.minecraft.world.inventory.HopperMenu(id, playerInventory, (net.minecraft.world.Container)this);
+    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.HopperMenu($$0, $$1, (net.minecraft.world.Container)this);
     }
 }
 