--- net/minecraft/client/gui/screens/packs/PackSelectionScreen$Watcher.java
+++ net/minecraft/client/gui/screens/packs/PackSelectionScreen$Watcher.java
@@ -9,12 +9,12 @@
     public java.nio.file.WatchService watcher;
     public java.nio.file.Path packPath;
 
-    public net.minecraft.client.gui.screens.packs.PackSelectionScreen$Watcher(java.io.File $$0) throws java.io.IOException {
-        this.packPath = $$0.toPath();
-        this.watcher = this.packPath.getFileSystem().newWatchService();
+    public net.minecraft.client.gui.screens.packs.PackSelectionScreen$Watcher(java.nio.file.Path $$0) throws java.io.IOException {
+        this.packPath = $$0;
+        this.watcher = $$0.getFileSystem().newWatchService();
         try {
-            this.watchDir(this.packPath);
-            try (java.nio.file.DirectoryStream $$1 = java.nio.file.Files.newDirectoryStream((java.nio.file.Path)this.packPath);
+            this.watchDir($$0);
+            try (java.nio.file.DirectoryStream $$1 = java.nio.file.Files.newDirectoryStream((java.nio.file.Path)$$0);
 ){
                 for (java.nio.file.Path $$2 : $$1) {
                     if (!java.nio.file.Files.isDirectory((java.nio.file.Path)$$2, (java.nio.file.LinkOption[])new java.nio.file.LinkOption[]{java.nio.file.LinkOption.NOFOLLOW_LINKS})) continue;
@@ -29,7 +29,7 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.client.gui.screens.packs.PackSelectionScreen$Watcher create(java.io.File $$0) {
+    public static net.minecraft.client.gui.screens.packs.PackSelectionScreen$Watcher create(java.nio.file.Path $$0) {
         try {
             return new net.minecraft.client.gui.screens.packs.PackSelectionScreen$Watcher($$0);
         }