--- net/minecraft/client/gui/screens/packs/TransferableSelectionList.java
+++ net/minecraft/client/gui/screens/packs/TransferableSelectionList.java
@@ -19,9 +19,9 @@
         this.setRenderHeader(true, (int)(9.0f * 1.5f));
     }
 
-    public void renderHeader(com.mojang.blaze3d.vertex.PoseStack matrixStack, int x, int y, com.mojang.blaze3d.vertex.Tesselator tessellator) {
-        net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component.empty().append(this.title).withStyle(new net.minecraft.ChatFormatting[]{net.minecraft.ChatFormatting.UNDERLINE, net.minecraft.ChatFormatting.BOLD});
-        this.minecraft.font.draw(matrixStack, (net.minecraft.network.chat.Component)mutableComponent, (float)(x + this.width / 2 - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)mutableComponent) / 2), (float)java.lang.Math.min((int)(this.y0 + 3), (int)y), 0xFFFFFF);
+    public void renderHeader(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, com.mojang.blaze3d.vertex.Tesselator $$3) {
+        net.minecraft.network.chat.MutableComponent $$4 = net.minecraft.network.chat.Component.empty().append(this.title).withStyle(new net.minecraft.ChatFormatting[]{net.minecraft.ChatFormatting.UNDERLINE, net.minecraft.ChatFormatting.BOLD});
+        this.minecraft.font.draw($$0, (net.minecraft.network.chat.Component)$$4, (float)($$1 + this.width / 2 - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)$$4) / 2), (float)java.lang.Math.min((int)(this.y0 + 3), (int)$$2), 0xFFFFFF);
     }
 
     public int getRowWidth() {