--- net/minecraft/client/gui/screens/packs/TransferableSelectionList$PackEntry.java
+++ net/minecraft/client/gui/screens/packs/TransferableSelectionList$PackEntry.java
@@ -52,95 +52,95 @@
         return net.minecraft.network.chat.Component.translatable((java.lang.String)"narrator.select", (java.lang.Object[])new java.lang.Object[]{this.pack.getTitle()});
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        net.minecraft.server.packs.repository.PackCompatibility packCompatibility = this.pack.getCompatibility();
-        if (!packCompatibility.isCompatible()) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        net.minecraft.server.packs.repository.PackCompatibility $$10 = this.pack.getCompatibility();
+        if (!$$10.isCompatible()) {
             com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)(left - 1), (int)(top - 1), (int)(left + width - 9), (int)(top + height + 1), (int)-8978432);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$3 - 1), (int)($$2 - 1), (int)($$3 + $$4 - 9), (int)($$2 + $$5 + 1), (int)-8978432);
         }
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)this.pack.getIconTexture());
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)32, (int)32);
-        net.minecraft.util.FormattedCharSequence formattedCharSequence = this.nameDisplayCache;
-        net.minecraft.client.gui.components.MultiLineLabel multiLineLabel = this.descriptionDisplayCache;
-        if (this.showHoverOverlay() && (((java.lang.Boolean)this.minecraft.options.touchscreen().get()).booleanValue() || isMouseOver)) {
+        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)32, (int)32);
+        net.minecraft.util.FormattedCharSequence $$11 = this.nameDisplayCache;
+        net.minecraft.client.gui.components.MultiLineLabel $$12 = this.descriptionDisplayCache;
+        if (this.showHoverOverlay() && (((java.lang.Boolean)this.minecraft.options.touchscreen().get()).booleanValue() || $$8)) {
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)net.minecraft.client.gui.screens.packs.TransferableSelectionList.ICON_OVERLAY_LOCATION);
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (int)(left + 32), (int)(top + 32), (int)-1601138544);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (int)($$3 + 32), (int)($$2 + 32), (int)-1601138544);
             com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
             com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-            int n = mouseX - left;
-            int n2 = mouseY - top;
+            int $$13 = $$6 - $$3;
+            int $$14 = $$7 - $$2;
             if (!this.pack.getCompatibility().isCompatible()) {
-                formattedCharSequence = this.incompatibleNameDisplayCache;
-                multiLineLabel = this.incompatibleDescriptionDisplayCache;
+                $$11 = this.incompatibleNameDisplayCache;
+                $$12 = this.incompatibleDescriptionDisplayCache;
             }
             if (this.pack.canSelect()) {
-                if (n < 32) {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)0.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
+                if ($$13 < 32) {
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)0.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
                 } else {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
                 }
             } else {
                 if (this.pack.canUnselect()) {
-                    if (n < 16) {
-                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)32.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
+                    if ($$13 < 16) {
+                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)32.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
                     } else {
-                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)32.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
+                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)32.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
                     }
                 }
                 if (this.pack.canMoveUp()) {
-                    if (n < 32 && n > 16 && n2 < 16) {
-                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)96.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
+                    if ($$13 < 32 && $$13 > 16 && $$14 < 16) {
+                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)96.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
                     } else {
-                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)96.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
+                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)96.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
                     }
                 }
                 if (this.pack.canMoveDown()) {
-                    if (n < 32 && n > 16 && n2 > 16) {
-                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)64.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
+                    if ($$13 < 32 && $$13 > 16 && $$14 > 16) {
+                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)64.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
                     } else {
-                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)64.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
+                        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)64.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
                     }
                 }
             }
         }
-        this.minecraft.font.drawShadow(matrixStack, formattedCharSequence, (float)(left + 32 + 2), (float)(top + 1), 0xFFFFFF);
-        multiLineLabel.renderLeftAligned(matrixStack, left + 32 + 2, top + 12, 10, 0x808080);
+        this.minecraft.font.drawShadow($$0, $$11, (float)($$3 + 32 + 2), (float)($$2 + 1), 0xFFFFFF);
+        $$12.renderLeftAligned($$0, $$3 + 32 + 2, $$2 + 12, 10, 0x808080);
     }
 
     public boolean showHoverOverlay() {
         return !this.pack.isFixedPosition() || !this.pack.isRequired();
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        double d = mouseX - (double)this.parent.getRowLeft();
-        double d2 = mouseY - (double)net.minecraft.client.gui.screens.packs.TransferableSelectionList.access$000((net.minecraft.client.gui.screens.packs.TransferableSelectionList)this.parent, (int)this.parent.children().indexOf((java.lang.Object)this));
-        if (this.showHoverOverlay() && d <= 32.0) {
+    public boolean mouseClicked(double $$02, double $$1, int $$2) {
+        double $$3 = $$02 - (double)this.parent.getRowLeft();
+        double $$4 = $$1 - (double)net.minecraft.client.gui.screens.packs.TransferableSelectionList.access$000((net.minecraft.client.gui.screens.packs.TransferableSelectionList)this.parent, (int)this.parent.children().indexOf((java.lang.Object)this));
+        if (this.showHoverOverlay() && $$3 <= 32.0) {
             if (this.pack.canSelect()) {
-                net.minecraft.server.packs.repository.PackCompatibility packCompatibility = this.pack.getCompatibility();
-                if (packCompatibility.isCompatible()) {
+                net.minecraft.server.packs.repository.PackCompatibility $$5 = this.pack.getCompatibility();
+                if ($$5.isCompatible()) {
                     this.pack.select();
                 } else {
-                    net.minecraft.network.chat.Component component = packCompatibility.getConfirmation();
+                    net.minecraft.network.chat.Component $$6 = $$5.getConfirmation();
                     this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.ConfirmScreen($$0 -> {
                         this.minecraft.setScreen(this.screen);
                         if ($$0) {
                             this.pack.select();
                         }
-                    }, net.minecraft.client.gui.screens.packs.TransferableSelectionList.INCOMPATIBLE_CONFIRM_TITLE, component));
+                    }, net.minecraft.client.gui.screens.packs.TransferableSelectionList.INCOMPATIBLE_CONFIRM_TITLE, $$6));
                 }
                 return true;
             }
-            if (d < 16.0 && this.pack.canUnselect()) {
+            if ($$3 < 16.0 && this.pack.canUnselect()) {
                 this.pack.unselect();
                 return true;
             }
-            if (d > 16.0 && d2 < 16.0 && this.pack.canMoveUp()) {
+            if ($$3 > 16.0 && $$4 < 16.0 && this.pack.canMoveUp()) {
                 this.pack.moveUp();
                 return true;
             }
-            if (d > 16.0 && d2 > 16.0 && this.pack.canMoveDown()) {
+            if ($$3 > 16.0 && $$4 > 16.0 && this.pack.canMoveDown()) {
                 this.pack.moveDown();
                 return true;
             }