--- net/minecraft/client/renderer/blockentity/BannerRenderer.java
+++ net/minecraft/client/renderer/blockentity/BannerRenderer.java
@@ -32,45 +32,44 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)64);
     }
 
-    public void render(net.minecraft.world.level.block.entity.BannerBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        net.minecraft.world.level.block.state.BlockState blockState;
-        long l;
-        java.util.List list = blockEntity.getPatterns();
-        float f = 0.6666667f;
-        boolean bl = blockEntity.getLevel() == null;
-        matrixStack.pushPose();
-        if (bl) {
-            l = 0L;
-            matrixStack.translate(0.5, 0.5, 0.5);
+    public void render(net.minecraft.world.level.block.entity.BannerBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        long $$10;
+        java.util.List $$6 = $$0.getPatterns();
+        float $$7 = 0.6666667f;
+        boolean $$8 = $$0.getLevel() == null;
+        $$2.pushPose();
+        if ($$8) {
+            long $$9 = 0L;
+            $$2.translate(0.5, 0.5, 0.5);
             this.pole.visible = true;
         } else {
-            l = blockEntity.getLevel().getGameTime();
-            blockState = blockEntity.getBlockState();
-            if (blockState.getBlock() instanceof net.minecraft.world.level.block.BannerBlock) {
-                matrixStack.translate(0.5, 0.5, 0.5);
-                var13_12 = (float)(-((java.lang.Integer)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BannerBlock.ROTATION)).intValue() * 360) / 16.0f;
-                matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(var13_12));
+            $$10 = $$0.getLevel().getGameTime();
+            net.minecraft.world.level.block.state.BlockState $$11 = $$0.getBlockState();
+            if ($$11.getBlock() instanceof net.minecraft.world.level.block.BannerBlock) {
+                $$2.translate(0.5, 0.5, 0.5);
+                float $$12 = -net.minecraft.world.level.block.state.properties.RotationSegment.convertToDegrees((int)((java.lang.Integer)$$11.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BannerBlock.ROTATION)));
+                $$2.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$12));
                 this.pole.visible = true;
             } else {
-                matrixStack.translate(0.5, -0.1666666716337204, 0.5);
-                var13_12 = -((net.minecraft.core.Direction)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.WallBannerBlock.FACING)).toYRot();
-                matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(var13_12));
-                matrixStack.translate(0.0, -0.3125, -0.4375);
+                $$2.translate(0.5, -0.1666666716337204, 0.5);
+                float $$13 = -((net.minecraft.core.Direction)$$11.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.WallBannerBlock.FACING)).toYRot();
+                $$2.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$13));
+                $$2.translate(0.0, -0.3125, -0.4375);
                 this.pole.visible = false;
             }
         }
-        matrixStack.pushPose();
-        matrixStack.scale(0.6666667f, -0.6666667f, -0.6666667f);
-        blockState = net.minecraft.client.resources.model.ModelBakery.BANNER_BASE.buffer(buffer, net.minecraft.client.renderer.RenderType::entitySolid);
-        this.pole.render(matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)blockState, combinedLight, combinedOverlay);
-        this.bar.render(matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)blockState, combinedLight, combinedOverlay);
-        net.minecraft.core.BlockPos blockPos = blockEntity.getBlockPos();
-        float f2 = ((float)java.lang.Math.floorMod((long)((long)(blockPos.getX() * 7 + blockPos.getY() * 9 + blockPos.getZ() * 13) + l), (long)100L) + partialTicks) / 100.0f;
-        this.flag.xRot = (-0.0125f + 0.01f * net.minecraft.util.Mth.cos((float)((float)java.lang.Math.PI * 2 * f2))) * (float)java.lang.Math.PI;
+        $$2.pushPose();
+        $$2.scale(0.6666667f, -0.6666667f, -0.6666667f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$14 = net.minecraft.client.resources.model.ModelBakery.BANNER_BASE.buffer($$3, net.minecraft.client.renderer.RenderType::entitySolid);
+        this.pole.render($$2, $$14, $$4, $$5);
+        this.bar.render($$2, $$14, $$4, $$5);
+        net.minecraft.core.BlockPos $$15 = $$0.getBlockPos();
+        float $$16 = ((float)java.lang.Math.floorMod((long)((long)($$15.getX() * 7 + $$15.getY() * 9 + $$15.getZ() * 13) + $$10), (long)100L) + $$1) / 100.0f;
+        this.flag.xRot = (-0.0125f + 0.01f * net.minecraft.util.Mth.cos((float)((float)java.lang.Math.PI * 2 * $$16))) * (float)java.lang.Math.PI;
         this.flag.y = -32.0f;
-        net.minecraft.client.renderer.blockentity.BannerRenderer.renderPatterns((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)combinedLight, (int)combinedOverlay, (net.minecraft.client.model.geom.ModelPart)this.flag, (net.minecraft.client.resources.model.Material)net.minecraft.client.resources.model.ModelBakery.BANNER_BASE, (boolean)true, (java.util.List)list);
-        matrixStack.popPose();
-        matrixStack.popPose();
+        net.minecraft.client.renderer.blockentity.BannerRenderer.renderPatterns((com.mojang.blaze3d.vertex.PoseStack)$$2, (net.minecraft.client.renderer.MultiBufferSource)$$3, (int)$$4, (int)$$5, (net.minecraft.client.model.geom.ModelPart)this.flag, (net.minecraft.client.resources.model.Material)net.minecraft.client.resources.model.ModelBakery.BANNER_BASE, (boolean)true, (java.util.List)$$6);
+        $$2.popPose();
+        $$2.popPose();
     }
 
     public static void renderPatterns(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, int $$3, net.minecraft.client.model.geom.ModelPart $$4, net.minecraft.client.resources.model.Material $$5, boolean $$6, java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.core.Holder<net.minecraft.world.level.block.entity.BannerPattern>, net.minecraft.world.item.DyeColor>> $$7) {