--- net/minecraft/client/renderer/blockentity/BellRenderer.java
+++ net/minecraft/client/renderer/blockentity/BellRenderer.java
@@ -23,26 +23,26 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)32, (int)32);
     }
 
-    public void render(net.minecraft.world.level.block.entity.BellBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        float f = (float)blockEntity.ticks + partialTicks;
-        float f2 = 0.0f;
-        float f3 = 0.0f;
-        if (blockEntity.shaking) {
-            float f4 = net.minecraft.util.Mth.sin((float)(f / (float)java.lang.Math.PI)) / (4.0f + f / 3.0f);
-            if (blockEntity.clickDirection == net.minecraft.core.Direction.NORTH) {
-                f2 = -f4;
-            } else if (blockEntity.clickDirection == net.minecraft.core.Direction.SOUTH) {
-                f2 = f4;
-            } else if (blockEntity.clickDirection == net.minecraft.core.Direction.EAST) {
-                f3 = -f4;
-            } else if (blockEntity.clickDirection == net.minecraft.core.Direction.WEST) {
-                f3 = f4;
+    public void render(net.minecraft.world.level.block.entity.BellBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        float $$6 = (float)$$0.ticks + $$1;
+        float $$7 = 0.0f;
+        float $$8 = 0.0f;
+        if ($$0.shaking) {
+            float $$9 = net.minecraft.util.Mth.sin((float)($$6 / (float)java.lang.Math.PI)) / (4.0f + $$6 / 3.0f);
+            if ($$0.clickDirection == net.minecraft.core.Direction.NORTH) {
+                $$7 = -$$9;
+            } else if ($$0.clickDirection == net.minecraft.core.Direction.SOUTH) {
+                $$7 = $$9;
+            } else if ($$0.clickDirection == net.minecraft.core.Direction.EAST) {
+                $$8 = -$$9;
+            } else if ($$0.clickDirection == net.minecraft.core.Direction.WEST) {
+                $$8 = $$9;
             }
         }
-        this.bellBody.xRot = f2;
-        this.bellBody.zRot = f3;
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = BELL_RESOURCE_LOCATION.buffer(buffer, net.minecraft.client.renderer.RenderType::entitySolid);
-        this.bellBody.render(matrixStack, vertexConsumer, combinedLight, combinedOverlay);
+        this.bellBody.xRot = $$7;
+        this.bellBody.zRot = $$8;
+        com.mojang.blaze3d.vertex.VertexConsumer $$10 = BELL_RESOURCE_LOCATION.buffer($$3, net.minecraft.client.renderer.RenderType::entitySolid);
+        this.bellBody.render($$2, $$10, $$4, $$5);
     }
 
     static {