--- net/minecraft/client/renderer/blockentity/BlockEntityRenderers.java
+++ net/minecraft/client/renderer/blockentity/BlockEntityRenderers.java
@@ -30,6 +30,7 @@
     static {
         PROVIDERS = com.google.common.collect.Maps.newHashMap();
         net.minecraft.client.renderer.blockentity.BlockEntityRenderers.register((net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.SIGN, net.minecraft.client.renderer.blockentity.SignRenderer::new);
+        net.minecraft.client.renderer.blockentity.BlockEntityRenderers.register((net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.HANGING_SIGN, net.minecraft.client.renderer.blockentity.HangingSignRenderer::new);
         net.minecraft.client.renderer.blockentity.BlockEntityRenderers.register((net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.MOB_SPAWNER, net.minecraft.client.renderer.blockentity.SpawnerRenderer::new);
         net.minecraft.client.renderer.blockentity.BlockEntityRenderers.register((net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.PISTON, net.minecraft.client.renderer.blockentity.PistonHeadRenderer::new);
         net.minecraft.client.renderer.blockentity.BlockEntityRenderers.register((net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.CHEST, net.minecraft.client.renderer.blockentity.ChestRenderer::new);