--- net/minecraft/client/renderer/blockentity/CampfireRenderer.java
+++ net/minecraft/client/renderer/blockentity/CampfireRenderer.java
@@ -13,23 +13,23 @@
         this.itemRenderer = $$0.getItemRenderer();
     }
 
-    public void render(net.minecraft.world.level.block.entity.CampfireBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        net.minecraft.core.Direction direction = (net.minecraft.core.Direction)blockEntity.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.CampfireBlock.FACING);
-        net.minecraft.core.NonNullList nonNullList = blockEntity.getItems();
-        int n = (int)blockEntity.getBlockPos().asLong();
-        for (int i = 0; i < nonNullList.size(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack = (net.minecraft.world.item.ItemStack)nonNullList.get(i);
-            if (itemStack == net.minecraft.world.item.ItemStack.EMPTY) continue;
-            matrixStack.pushPose();
-            matrixStack.translate(0.5, 0.44921875, 0.5);
-            net.minecraft.core.Direction direction2 = net.minecraft.core.Direction.from2DDataValue((int)((i + direction.get2DDataValue()) % 4));
-            float f = -direction2.toYRot();
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f));
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
-            matrixStack.translate(-0.3125, -0.3125, 0.0);
-            matrixStack.scale(0.375f, 0.375f, 0.375f);
-            this.itemRenderer.renderStatic(itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.FIXED, combinedLight, combinedOverlay, matrixStack, buffer, n + i);
-            matrixStack.popPose();
+    public void render(net.minecraft.world.level.block.entity.CampfireBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        net.minecraft.core.Direction $$6 = (net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.CampfireBlock.FACING);
+        net.minecraft.core.NonNullList $$7 = $$0.getItems();
+        int $$8 = (int)$$0.getBlockPos().asLong();
+        for (int $$9 = 0; $$9 < $$7.size(); ++$$9) {
+            net.minecraft.world.item.ItemStack $$10 = (net.minecraft.world.item.ItemStack)$$7.get($$9);
+            if ($$10 == net.minecraft.world.item.ItemStack.EMPTY) continue;
+            $$2.pushPose();
+            $$2.translate(0.5, 0.44921875, 0.5);
+            net.minecraft.core.Direction $$11 = net.minecraft.core.Direction.from2DDataValue((int)(($$9 + $$6.get2DDataValue()) % 4));
+            float $$12 = -$$11.toYRot();
+            $$2.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$12));
+            $$2.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
+            $$2.translate(-0.3125, -0.3125, 0.0);
+            $$2.scale(0.375f, 0.375f, 0.375f);
+            this.itemRenderer.renderStatic($$10, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.FIXED, $$4, $$5, $$2, $$3, $$8 + $$9);
+            $$2.popPose();
         }
     }
 }