--- net/minecraft/client/renderer/blockentity/ConduitRenderer.java
+++ net/minecraft/client/renderer/blockentity/ConduitRenderer.java
@@ -54,58 +54,58 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)32, (int)16);
     }
 
-    public void render(net.minecraft.world.level.block.entity.ConduitBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        float f = (float)blockEntity.tickCount + partialTicks;
-        if (!blockEntity.isActive()) {
-            float f2 = blockEntity.getActiveRotation(0.0f);
-            com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = SHELL_TEXTURE.buffer(buffer, net.minecraft.client.renderer.RenderType::entitySolid);
-            matrixStack.pushPose();
-            matrixStack.translate(0.5, 0.5, 0.5);
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f2));
-            this.shell.render(matrixStack, vertexConsumer, combinedLight, combinedOverlay);
-            matrixStack.popPose();
+    public void render(net.minecraft.world.level.block.entity.ConduitBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        float $$6 = (float)$$0.tickCount + $$1;
+        if (!$$0.isActive()) {
+            float $$7 = $$0.getActiveRotation(0.0f);
+            com.mojang.blaze3d.vertex.VertexConsumer $$8 = SHELL_TEXTURE.buffer($$3, net.minecraft.client.renderer.RenderType::entitySolid);
+            $$2.pushPose();
+            $$2.translate(0.5, 0.5, 0.5);
+            $$2.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$7));
+            this.shell.render($$2, $$8, $$4, $$5);
+            $$2.popPose();
             return;
         }
-        float f3 = blockEntity.getActiveRotation(partialTicks) * 57.295776f;
-        float f4 = net.minecraft.util.Mth.sin((float)(f * 0.1f)) / 2.0f + 0.5f;
-        f4 = f4 * f4 + f4;
-        matrixStack.pushPose();
-        matrixStack.translate(0.5, (double)(0.3f + f4 * 0.2f), 0.5);
-        com.mojang.math.Vector3f vector3f = new com.mojang.math.Vector3f(0.5f, 1.0f, 0.5f);
-        vector3f.normalize();
-        matrixStack.mulPose(vector3f.rotationDegrees(f3));
-        this.cage.render(matrixStack, ACTIVE_SHELL_TEXTURE.buffer(buffer, net.minecraft.client.renderer.RenderType::entityCutoutNoCull), combinedLight, combinedOverlay);
-        matrixStack.popPose();
-        int n = blockEntity.tickCount / 66 % 3;
-        matrixStack.pushPose();
-        matrixStack.translate(0.5, 0.5, 0.5);
-        if (n == 1) {
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
-        } else if (n == 2) {
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
+        float $$9 = $$0.getActiveRotation($$1) * 57.295776f;
+        float $$10 = net.minecraft.util.Mth.sin((float)($$6 * 0.1f)) / 2.0f + 0.5f;
+        $$10 = $$10 * $$10 + $$10;
+        $$2.pushPose();
+        $$2.translate(0.5, (double)(0.3f + $$10 * 0.2f), 0.5);
+        com.mojang.math.Vector3f $$11 = new com.mojang.math.Vector3f(0.5f, 1.0f, 0.5f);
+        $$11.normalize();
+        $$2.mulPose($$11.rotationDegrees($$9));
+        this.cage.render($$2, ACTIVE_SHELL_TEXTURE.buffer($$3, net.minecraft.client.renderer.RenderType::entityCutoutNoCull), $$4, $$5);
+        $$2.popPose();
+        int $$12 = $$0.tickCount / 66 % 3;
+        $$2.pushPose();
+        $$2.translate(0.5, 0.5, 0.5);
+        if ($$12 == 1) {
+            $$2.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
+        } else if ($$12 == 2) {
+            $$2.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
         }
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = (n == 1 ? VERTICAL_WIND_TEXTURE : WIND_TEXTURE).buffer(buffer, net.minecraft.client.renderer.RenderType::entityCutoutNoCull);
-        this.wind.render(matrixStack, vertexConsumer, combinedLight, combinedOverlay);
-        matrixStack.popPose();
-        matrixStack.pushPose();
-        matrixStack.translate(0.5, 0.5, 0.5);
-        matrixStack.scale(0.875f, 0.875f, 0.875f);
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(180.0f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
-        this.wind.render(matrixStack, vertexConsumer, combinedLight, combinedOverlay);
-        matrixStack.popPose();
-        net.minecraft.client.Camera camera = this.renderer.camera;
-        matrixStack.pushPose();
-        matrixStack.translate(0.5, (double)(0.3f + f4 * 0.2f), 0.5);
-        matrixStack.scale(0.5f, 0.5f, 0.5f);
-        float f5 = -camera.getYRot();
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f5));
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(camera.getXRot()));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
-        float f6 = 1.3333334f;
-        matrixStack.scale(1.3333334f, 1.3333334f, 1.3333334f);
-        this.eye.render(matrixStack, (blockEntity.isHunting() ? OPEN_EYE_TEXTURE : CLOSED_EYE_TEXTURE).buffer(buffer, net.minecraft.client.renderer.RenderType::entityCutoutNoCull), combinedLight, combinedOverlay);
-        matrixStack.popPose();
+        com.mojang.blaze3d.vertex.VertexConsumer $$13 = ($$12 == 1 ? VERTICAL_WIND_TEXTURE : WIND_TEXTURE).buffer($$3, net.minecraft.client.renderer.RenderType::entityCutoutNoCull);
+        this.wind.render($$2, $$13, $$4, $$5);
+        $$2.popPose();
+        $$2.pushPose();
+        $$2.translate(0.5, 0.5, 0.5);
+        $$2.scale(0.875f, 0.875f, 0.875f);
+        $$2.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(180.0f));
+        $$2.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
+        this.wind.render($$2, $$13, $$4, $$5);
+        $$2.popPose();
+        net.minecraft.client.Camera $$14 = this.renderer.camera;
+        $$2.pushPose();
+        $$2.translate(0.5, (double)(0.3f + $$10 * 0.2f), 0.5);
+        $$2.scale(0.5f, 0.5f, 0.5f);
+        float $$15 = -$$14.getYRot();
+        $$2.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$15));
+        $$2.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$14.getXRot()));
+        $$2.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
+        float $$16 = 1.3333334f;
+        $$2.scale(1.3333334f, 1.3333334f, 1.3333334f);
+        this.eye.render($$2, ($$0.isHunting() ? OPEN_EYE_TEXTURE : CLOSED_EYE_TEXTURE).buffer($$3, net.minecraft.client.renderer.RenderType::entityCutoutNoCull), $$4, $$5);
+        $$2.popPose();
     }
 
     static {