--- net/minecraft/client/renderer/blockentity/EnchantTableRenderer.java
+++ net/minecraft/client/renderer/blockentity/EnchantTableRenderer.java
@@ -13,29 +13,29 @@
         this.bookModel = new net.minecraft.client.model.BookModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.BOOK));
     }
 
-    public void render(net.minecraft.world.level.block.entity.EnchantmentTableBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        float f;
-        matrixStack.pushPose();
-        matrixStack.translate(0.5, 0.75, 0.5);
-        float f2 = (float)blockEntity.time + partialTicks;
-        matrixStack.translate(0.0, (double)(0.1f + net.minecraft.util.Mth.sin((float)(f2 * 0.1f)) * 0.01f), 0.0);
-        for (f = blockEntity.rot - blockEntity.oRot;
- f >= (float)java.lang.Math.PI; f -= (float)java.lang.Math.PI * 2) {
+    public void render(net.minecraft.world.level.block.entity.EnchantmentTableBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        float $$7;
+        $$2.pushPose();
+        $$2.translate(0.5, 0.75, 0.5);
+        float $$6 = (float)$$0.time + $$1;
+        $$2.translate(0.0, (double)(0.1f + net.minecraft.util.Mth.sin((float)($$6 * 0.1f)) * 0.01f), 0.0);
+        for ($$7 = $$0.rot - $$0.oRot;
+ $$7 >= (float)java.lang.Math.PI; $$7 -= (float)java.lang.Math.PI * 2) {
         }
-        while (f < (float)(-java.lang.Math.PI)) {
-            f += (float)java.lang.Math.PI * 2;
+        while ($$7 < (float)(-java.lang.Math.PI)) {
+            $$7 += (float)java.lang.Math.PI * 2;
         }
-        float f3 = blockEntity.oRot + f * partialTicks;
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotation(-f3));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(80.0f));
-        float f4 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)blockEntity.oFlip, (float)blockEntity.flip);
-        float f5 = net.minecraft.util.Mth.frac((float)(f4 + 0.25f)) * 1.6f - 0.3f;
-        float f6 = net.minecraft.util.Mth.frac((float)(f4 + 0.75f)) * 1.6f - 0.3f;
-        float f7 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)blockEntity.oOpen, (float)blockEntity.open);
-        this.bookModel.setupAnim(f2, net.minecraft.util.Mth.clamp((float)f5, (float)0.0f, (float)1.0f), net.minecraft.util.Mth.clamp((float)f6, (float)0.0f, (float)1.0f), f7);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = BOOK_LOCATION.buffer(buffer, net.minecraft.client.renderer.RenderType::entitySolid);
-        this.bookModel.render(matrixStack, vertexConsumer, combinedLight, combinedOverlay, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
+        float $$8 = $$0.oRot + $$7 * $$1;
+        $$2.mulPose(com.mojang.math.Vector3f.YP.rotation(-$$8));
+        $$2.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(80.0f));
+        float $$9 = net.minecraft.util.Mth.lerp((float)$$1, (float)$$0.oFlip, (float)$$0.flip);
+        float $$10 = net.minecraft.util.Mth.frac((float)($$9 + 0.25f)) * 1.6f - 0.3f;
+        float $$11 = net.minecraft.util.Mth.frac((float)($$9 + 0.75f)) * 1.6f - 0.3f;
+        float $$12 = net.minecraft.util.Mth.lerp((float)$$1, (float)$$0.oOpen, (float)$$0.open);
+        this.bookModel.setupAnim($$6, net.minecraft.util.Mth.clamp((float)$$10, (float)0.0f, (float)1.0f), net.minecraft.util.Mth.clamp((float)$$11, (float)0.0f, (float)1.0f), $$12);
+        com.mojang.blaze3d.vertex.VertexConsumer $$13 = BOOK_LOCATION.buffer($$3, net.minecraft.client.renderer.RenderType::entitySolid);
+        this.bookModel.render($$2, $$13, $$4, $$5, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$2.popPose();
     }
 
     static {