--- net/minecraft/client/renderer/blockentity/LecternRenderer.java
+++ net/minecraft/client/renderer/blockentity/LecternRenderer.java
@@ -12,21 +12,21 @@
         this.bookModel = new net.minecraft.client.model.BookModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.BOOK));
     }
 
-    public void render(net.minecraft.world.level.block.entity.LecternBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        net.minecraft.world.level.block.state.BlockState blockState = blockEntity.getBlockState();
-        if (!((java.lang.Boolean)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.LecternBlock.HAS_BOOK)).booleanValue()) {
+    public void render(net.minecraft.world.level.block.entity.LecternBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        net.minecraft.world.level.block.state.BlockState $$6 = $$0.getBlockState();
+        if (!((java.lang.Boolean)$$6.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.LecternBlock.HAS_BOOK)).booleanValue()) {
             return;
         }
-        matrixStack.pushPose();
-        matrixStack.translate(0.5, 1.0625, 0.5);
-        float f = ((net.minecraft.core.Direction)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.LecternBlock.FACING)).getClockWise().toYRot();
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(67.5f));
-        matrixStack.translate(0.0, -0.125, 0.0);
+        $$2.pushPose();
+        $$2.translate(0.5, 1.0625, 0.5);
+        float $$7 = ((net.minecraft.core.Direction)$$6.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.LecternBlock.FACING)).getClockWise().toYRot();
+        $$2.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-$$7));
+        $$2.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(67.5f));
+        $$2.translate(0.0, -0.125, 0.0);
         this.bookModel.setupAnim(0.0f, 0.1f, 0.9f, 1.2f);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = net.minecraft.client.renderer.blockentity.EnchantTableRenderer.BOOK_LOCATION.buffer(buffer, net.minecraft.client.renderer.RenderType::entitySolid);
-        this.bookModel.render(matrixStack, vertexConsumer, combinedLight, combinedOverlay, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
+        com.mojang.blaze3d.vertex.VertexConsumer $$8 = net.minecraft.client.renderer.blockentity.EnchantTableRenderer.BOOK_LOCATION.buffer($$3, net.minecraft.client.renderer.RenderType::entitySolid);
+        this.bookModel.render($$2, $$8, $$4, $$5, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$2.popPose();
     }
 }
 