--- net/minecraft/client/renderer/blockentity/ShulkerBoxRenderer.java
+++ net/minecraft/client/renderer/blockentity/ShulkerBoxRenderer.java
@@ -12,27 +12,32 @@
         this.model = new net.minecraft.client.model.ShulkerModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.SHULKER));
     }
 
-    public void render(net.minecraft.world.level.block.entity.ShulkerBoxBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        net.minecraft.world.item.DyeColor dyeColor;
-        net.minecraft.world.level.block.state.BlockState blockState;
-        net.minecraft.core.Direction direction = net.minecraft.core.Direction.UP;
-        if (blockEntity.hasLevel() && (blockState = blockEntity.getLevel().getBlockState(blockEntity.getBlockPos())).getBlock() instanceof net.minecraft.world.level.block.ShulkerBoxBlock) {
-            direction = (net.minecraft.core.Direction)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.ShulkerBoxBlock.FACING);
+    public void render(net.minecraft.world.level.block.entity.ShulkerBoxBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        net.minecraft.client.resources.model.Material $$10;
+        net.minecraft.world.item.DyeColor $$8;
+        net.minecraft.world.level.block.state.BlockState $$7;
+        net.minecraft.core.Direction $$6 = net.minecraft.core.Direction.UP;
+        if ($$0.hasLevel() && ($$7 = $$0.getLevel().getBlockState($$0.getBlockPos())).getBlock() instanceof net.minecraft.world.level.block.ShulkerBoxBlock) {
+            $$6 = (net.minecraft.core.Direction)$$7.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.ShulkerBoxBlock.FACING);
         }
-        blockState = (dyeColor = blockEntity.getColor()) == null ? net.minecraft.client.renderer.Sheets.DEFAULT_SHULKER_TEXTURE_LOCATION : (net.minecraft.client.resources.model.Material)net.minecraft.client.renderer.Sheets.SHULKER_TEXTURE_LOCATION.get(dyeColor.getId());
-        matrixStack.pushPose();
-        matrixStack.translate(0.5, 0.5, 0.5);
-        float f = 0.9995f;
-        matrixStack.scale(0.9995f, 0.9995f, 0.9995f);
-        matrixStack.mulPose(direction.getRotation());
-        matrixStack.scale(1.0f, -1.0f, -1.0f);
-        matrixStack.translate(0.0, -1.0, 0.0);
-        net.minecraft.client.model.geom.ModelPart modelPart = this.model.getLid();
-        modelPart.setPos(0.0f, 24.0f - blockEntity.getProgress(partialTicks) * 0.5f * 16.0f, 0.0f);
-        modelPart.yRot = 270.0f * blockEntity.getProgress(partialTicks) * ((float)java.lang.Math.PI / 180);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = blockState.buffer(buffer, net.minecraft.client.renderer.RenderType::entityCutoutNoCull);
-        this.model.renderToBuffer(matrixStack, vertexConsumer, combinedLight, combinedOverlay, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
+        if (($$8 = $$0.getColor()) == null) {
+            net.minecraft.client.resources.model.Material $$9 = net.minecraft.client.renderer.Sheets.DEFAULT_SHULKER_TEXTURE_LOCATION;
+        } else {
+            $$10 = (net.minecraft.client.resources.model.Material)net.minecraft.client.renderer.Sheets.SHULKER_TEXTURE_LOCATION.get($$8.getId());
+        }
+        $$2.pushPose();
+        $$2.translate(0.5, 0.5, 0.5);
+        float $$11 = 0.9995f;
+        $$2.scale(0.9995f, 0.9995f, 0.9995f);
+        $$2.mulPose($$6.getRotation());
+        $$2.scale(1.0f, -1.0f, -1.0f);
+        $$2.translate(0.0, -1.0, 0.0);
+        net.minecraft.client.model.geom.ModelPart $$12 = this.model.getLid();
+        $$12.setPos(0.0f, 24.0f - $$0.getProgress($$1) * 0.5f * 16.0f, 0.0f);
+        $$12.yRot = 270.0f * $$0.getProgress($$1) * ((float)java.lang.Math.PI / 180);
+        com.mojang.blaze3d.vertex.VertexConsumer $$13 = $$10.buffer($$3, net.minecraft.client.renderer.RenderType::entityCutoutNoCull);
+        this.model.renderToBuffer($$2, $$13, $$4, $$5, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$2.popPose();
     }
 }
 