--- net/minecraft/client/renderer/blockentity/StructureBlockRenderer.java
+++ net/minecraft/client/renderer/blockentity/StructureBlockRenderer.java
@@ -9,76 +9,83 @@
     public net.minecraft.client.renderer.blockentity.StructureBlockRenderer(net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider$Context $$0) {
     }
 
-    public void render(net.minecraft.world.level.block.entity.StructureBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        double d;
-        double d2;
-        double d3;
-        double d4;
+    public void render(net.minecraft.world.level.block.entity.StructureBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        double $$33;
+        double $$32;
+        double $$31;
+        double $$30;
+        double $$17;
+        double $$16;
         if (!net.minecraft.client.Minecraft.getInstance().player.canUseGameMasterBlocks() && !net.minecraft.client.Minecraft.getInstance().player.isSpectator()) {
             return;
         }
-        net.minecraft.core.BlockPos blockPos = blockEntity.getStructurePos();
-        net.minecraft.core.Vec3i vec3i = blockEntity.getStructureSize();
-        if (vec3i.getX() < 1 || vec3i.getY() < 1 || vec3i.getZ() < 1) {
+        net.minecraft.core.BlockPos $$6 = $$0.getStructurePos();
+        net.minecraft.core.Vec3i $$7 = $$0.getStructureSize();
+        if ($$7.getX() < 1 || $$7.getY() < 1 || $$7.getZ() < 1) {
             return;
         }
-        if (blockEntity.getMode() != net.minecraft.world.level.block.state.properties.StructureMode.SAVE && blockEntity.getMode() != net.minecraft.world.level.block.state.properties.StructureMode.LOAD) {
+        if ($$0.getMode() != net.minecraft.world.level.block.state.properties.StructureMode.SAVE && $$0.getMode() != net.minecraft.world.level.block.state.properties.StructureMode.LOAD) {
             return;
         }
-        double d5 = blockPos.getX();
-        double d6 = blockPos.getZ();
-        double d7 = blockPos.getY();
-        double d8 = d7 + (double)vec3i.getY();
-        double d9 = switch (net.minecraft.client.renderer.blockentity.StructureBlockRenderer$1.$SwitchMap$net$minecraft$world$level$block$Mirror[blockEntity.getMirror().ordinal()]) {
-            case 1 -> {
-                d4 = vec3i.getX();
-                yield -vec3i.getZ();
+        double $$8 = $$6.getX();
+        double $$9 = $$6.getZ();
+        double $$10 = $$6.getY();
+        double $$11 = $$10 + (double)$$7.getY();
+        switch (net.minecraft.client.renderer.blockentity.StructureBlockRenderer$1.$SwitchMap$net$minecraft$world$level$block$Mirror[$$0.getMirror().ordinal()]) {
+            case 1: {
+                double $$12 = $$7.getX();
+                double $$13 = -$$7.getZ();
+                break;
             }
-            case 2 -> {
-                d4 = -vec3i.getX();
-                yield vec3i.getZ();
+            case 2: {
+                double $$14 = -$$7.getX();
+                double $$15 = $$7.getZ();
+                break;
             }
-            default -> {
-                d4 = vec3i.getX();
-                yield vec3i.getZ();
+            default: {
+                $$16 = $$7.getX();
+                $$17 = $$7.getZ();
             }
-        };
-        double d10 = switch (net.minecraft.client.renderer.blockentity.StructureBlockRenderer$1.$SwitchMap$net$minecraft$world$level$block$Rotation[blockEntity.getRotation().ordinal()]) {
-            case 1 -> {
-                d3 = d9 < 0.0 ? d5 : d5 + 1.0;
-                d2 = d4 < 0.0 ? d6 + 1.0 : d6;
-                d = d3 - d9;
-                yield d2 + d4;
+        }
+        switch (net.minecraft.client.renderer.blockentity.StructureBlockRenderer$1.$SwitchMap$net$minecraft$world$level$block$Rotation[$$0.getRotation().ordinal()]) {
+            case 1: {
+                double $$18 = $$17 < 0.0 ? $$8 : $$8 + 1.0;
+                double $$19 = $$16 < 0.0 ? $$9 + 1.0 : $$9;
+                double $$20 = $$18 - $$17;
+                double $$21 = $$19 + $$16;
+                break;
             }
-            case 2 -> {
-                d3 = d4 < 0.0 ? d5 : d5 + 1.0;
-                d2 = d9 < 0.0 ? d6 : d6 + 1.0;
-                d = d3 - d4;
-                yield d2 - d9;
+            case 2: {
+                double $$22 = $$16 < 0.0 ? $$8 : $$8 + 1.0;
+                double $$23 = $$17 < 0.0 ? $$9 : $$9 + 1.0;
+                double $$24 = $$22 - $$16;
+                double $$25 = $$23 - $$17;
+                break;
             }
-            case 3 -> {
-                d3 = d9 < 0.0 ? d5 + 1.0 : d5;
-                d2 = d4 < 0.0 ? d6 : d6 + 1.0;
-                d = d3 + d9;
-                yield d2 - d4;
+            case 3: {
+                double $$26 = $$17 < 0.0 ? $$8 + 1.0 : $$8;
+                double $$27 = $$16 < 0.0 ? $$9 : $$9 + 1.0;
+                double $$28 = $$26 + $$17;
+                double $$29 = $$27 - $$16;
+                break;
             }
-            default -> {
-                d3 = d4 < 0.0 ? d5 + 1.0 : d5;
-                d2 = d9 < 0.0 ? d6 + 1.0 : d6;
-                d = d3 + d4;
-                yield d2 + d9;
+            default: {
+                $$30 = $$16 < 0.0 ? $$8 + 1.0 : $$8;
+                $$31 = $$17 < 0.0 ? $$9 + 1.0 : $$9;
+                $$32 = $$30 + $$16;
+                $$33 = $$31 + $$17;
             }
-        };
-        float f = 1.0f;
-        float f2 = 0.9f;
-        float f3 = 0.5f;
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.lines());
-        if (blockEntity.getMode() == net.minecraft.world.level.block.state.properties.StructureMode.SAVE || blockEntity.getShowBoundingBox()) {
-            net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (double)d3, (double)d7, (double)d2, (double)d, (double)d8, (double)d10, (float)0.9f, (float)0.9f, (float)0.9f, (float)1.0f, (float)0.5f, (float)0.5f, (float)0.5f);
         }
-        if (blockEntity.getMode() == net.minecraft.world.level.block.state.properties.StructureMode.SAVE && blockEntity.getShowAir()) {
-            this.renderInvisibleBlocks(blockEntity, vertexConsumer, blockPos, matrixStack);
+        float $$34 = 1.0f;
+        float $$35 = 0.9f;
+        float $$36 = 0.5f;
+        com.mojang.blaze3d.vertex.VertexConsumer $$37 = $$3.getBuffer(net.minecraft.client.renderer.RenderType.lines());
+        if ($$0.getMode() == net.minecraft.world.level.block.state.properties.StructureMode.SAVE || $$0.getShowBoundingBox()) {
+            net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)$$2, (com.mojang.blaze3d.vertex.VertexConsumer)$$37, (double)$$30, (double)$$10, (double)$$31, (double)$$32, (double)$$11, (double)$$33, (float)0.9f, (float)0.9f, (float)0.9f, (float)1.0f, (float)0.5f, (float)0.5f, (float)0.5f);
         }
+        if ($$0.getMode() == net.minecraft.world.level.block.state.properties.StructureMode.SAVE && $$0.getShowAir()) {
+            this.renderInvisibleBlocks($$0, $$37, $$6, $$2);
+        }
     }
 
     public void renderInvisibleBlocks(net.minecraft.world.level.block.entity.StructureBlockEntity $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, net.minecraft.core.BlockPos $$2, com.mojang.blaze3d.vertex.PoseStack $$3) {
@@ -118,7 +125,7 @@
         }
     }
 
-    public boolean shouldRenderOffScreen(net.minecraft.world.level.block.entity.StructureBlockEntity te) {
+    public boolean shouldRenderOffScreen(net.minecraft.world.level.block.entity.StructureBlockEntity $$0) {
         return true;
     }
 