--- net/minecraft/client/renderer/blockentity/TheEndGatewayRenderer.java
+++ net/minecraft/client/renderer/blockentity/TheEndGatewayRenderer.java
@@ -12,17 +12,17 @@
         super($$0);
     }
 
-    public void render(net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        if (blockEntity.isSpawning() || blockEntity.isCoolingDown()) {
-            float f = blockEntity.isSpawning() ? blockEntity.getSpawnPercent(partialTicks) : blockEntity.getCooldownPercent(partialTicks);
-            double d = blockEntity.isSpawning() ? (double)blockEntity.getLevel().getMaxBuildHeight() : 50.0;
-            f = net.minecraft.util.Mth.sin((float)(f * (float)java.lang.Math.PI));
-            int n = net.minecraft.util.Mth.floor((double)((double)f * d));
-            float[] fArray = blockEntity.isSpawning() ? net.minecraft.world.item.DyeColor.MAGENTA.getTextureDiffuseColors() : net.minecraft.world.item.DyeColor.PURPLE.getTextureDiffuseColors();
-            long l = blockEntity.getLevel().getGameTime();
-            net.minecraft.client.renderer.blockentity.BeaconRenderer.renderBeaconBeam((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (net.minecraft.resources.ResourceLocation)BEAM_LOCATION, (float)partialTicks, (float)f, (long)l, (int)(-n), (int)(n * 2), (float[])fArray, (float)0.15f, (float)0.175f);
+    public void render(net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        if ($$0.isSpawning() || $$0.isCoolingDown()) {
+            float $$6 = $$0.isSpawning() ? $$0.getSpawnPercent($$1) : $$0.getCooldownPercent($$1);
+            double $$7 = $$0.isSpawning() ? (double)$$0.getLevel().getMaxBuildHeight() : 50.0;
+            $$6 = net.minecraft.util.Mth.sin((float)($$6 * (float)java.lang.Math.PI));
+            int $$8 = net.minecraft.util.Mth.floor((double)((double)$$6 * $$7));
+            float[] $$9 = $$0.isSpawning() ? net.minecraft.world.item.DyeColor.MAGENTA.getTextureDiffuseColors() : net.minecraft.world.item.DyeColor.PURPLE.getTextureDiffuseColors();
+            long $$10 = $$0.getLevel().getGameTime();
+            net.minecraft.client.renderer.blockentity.BeaconRenderer.renderBeaconBeam((com.mojang.blaze3d.vertex.PoseStack)$$2, (net.minecraft.client.renderer.MultiBufferSource)$$3, (net.minecraft.resources.ResourceLocation)BEAM_LOCATION, (float)$$1, (float)$$6, (long)$$10, (int)(-$$8), (int)($$8 * 2), (float[])$$9, (float)0.15f, (float)0.175f);
         }
-        super.render((net.minecraft.world.level.block.entity.TheEndPortalBlockEntity)blockEntity, partialTicks, matrixStack, buffer, combinedLight, combinedOverlay);
+        super.render((net.minecraft.world.level.block.entity.TheEndPortalBlockEntity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
     public float getOffsetUp() {