--- net/minecraft/client/renderer/blockentity/TheEndPortalRenderer.java
+++ net/minecraft/client/renderer/blockentity/TheEndPortalRenderer.java
@@ -12,9 +12,9 @@
     public net.minecraft.client.renderer.blockentity.TheEndPortalRenderer(net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider$Context $$0) {
     }
 
-    public void render(T blockEntity, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int combinedLight, int combinedOverlay) {
-        com.mojang.math.Matrix4f matrix4f = matrixStack.last().pose();
-        this.renderCube(blockEntity, matrix4f, buffer.getBuffer(this.renderType()));
+    public void render(T $$0, float $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, int $$5) {
+        com.mojang.math.Matrix4f $$6 = $$2.last().pose();
+        this.renderCube($$0, $$6, $$3.getBuffer(this.renderType()));
     }
 
     public void renderCube(T $$0, com.mojang.math.Matrix4f $$1, com.mojang.blaze3d.vertex.VertexConsumer $$2) {