--- net/minecraft/server/packs/metadata/MetadataSectionType$1.java
+++ net/minecraft/server/packs/metadata/MetadataSectionType$1.java
@@ -1,0 +1,31 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.metadata;
+
+public class net.minecraft.server.packs.metadata.MetadataSectionType$1
+implements net.minecraft.server.packs.metadata.MetadataSectionType<T>
+ {
+    public /* synthetic */ java.lang.String val$name;
+    public /* synthetic */ com.mojang.serialization.Codec val$codec;
+
+    public net.minecraft.server.packs.metadata.MetadataSectionType$1(java.lang.String string, com.mojang.serialization.Codec codec) {
+        this.val$name = string;
+        this.val$codec = codec;
+    }
+
+    public java.lang.String getMetadataSectionName() {
+        return this.val$name;
+    }
+
+    public T fromJson(com.google.gson.JsonObject $$02) {
+        return (T)this.val$codec.parse((com.mojang.serialization.DynamicOps)com.mojang.serialization.JsonOps.INSTANCE, (java.lang.Object)$$02).getOrThrow(false, $$0 -> {}
+);
+    }
+
+    public com.google.gson.JsonObject toJson(T $$02) {
+        return ((com.google.gson.JsonElement)this.val$codec.encodeStart((com.mojang.serialization.DynamicOps)com.mojang.serialization.JsonOps.INSTANCE, $$02).getOrThrow(false, $$0 -> {}
+)).getAsJsonObject();
+    }
+}
+