--- net/minecraft/world/item/crafting/AbstractCookingRecipe.java
+++ net/minecraft/world/item/crafting/AbstractCookingRecipe.java
@@ -8,31 +8,33 @@
  {
     public net.minecraft.world.item.crafting.RecipeType<?> type;
     public net.minecraft.resources.ResourceLocation id;
+    public net.minecraft.world.item.crafting.CookingBookCategory category;
     public java.lang.String group;
     public net.minecraft.world.item.crafting.Ingredient ingredient;
     public net.minecraft.world.item.ItemStack result;
     public float experience;
     public int cookingTime;
 
-    public net.minecraft.world.item.crafting.AbstractCookingRecipe(net.minecraft.world.item.crafting.RecipeType<?> type, net.minecraft.resources.ResourceLocation id, java.lang.String group, net.minecraft.world.item.crafting.Ingredient ingredient, net.minecraft.world.item.ItemStack result, float experience, int cookingTime) {
-        this.type = type;
-        this.id = id;
-        this.group = group;
-        this.ingredient = ingredient;
-        this.result = result;
-        this.experience = experience;
-        this.cookingTime = cookingTime;
+    public net.minecraft.world.item.crafting.AbstractCookingRecipe(net.minecraft.world.item.crafting.RecipeType<?> $$0, net.minecraft.resources.ResourceLocation $$1, java.lang.String $$2, net.minecraft.world.item.crafting.CookingBookCategory $$3, net.minecraft.world.item.crafting.Ingredient $$4, net.minecraft.world.item.ItemStack $$5, float $$6, int $$7) {
+        this.type = $$0;
+        this.category = $$3;
+        this.id = $$1;
+        this.group = $$2;
+        this.ingredient = $$4;
+        this.result = $$5;
+        this.experience = $$6;
+        this.cookingTime = $$7;
     }
 
-    public boolean matches(net.minecraft.world.Container inv, net.minecraft.world.level.Level level) {
-        return this.ingredient.test(inv.getItem(0));
+    public boolean matches(net.minecraft.world.Container $$0, net.minecraft.world.level.Level $$1) {
+        return this.ingredient.test($$0.getItem(0));
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.Container inv) {
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.Container $$0) {
         return this.result.copy();
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
+    public boolean canCraftInDimensions(int $$0, int $$1) {
         return true;
     }
 
@@ -65,5 +67,9 @@
     public net.minecraft.world.item.crafting.RecipeType<?> getType() {
         return this.type;
     }
+
+    public net.minecraft.world.item.crafting.CookingBookCategory category() {
+        return this.category;
+    }
 }
 