--- net/minecraft/world/item/crafting/ArmorDyeRecipe.java
+++ net/minecraft/world/item/crafting/ArmorDyeRecipe.java
@@ -6,60 +6,60 @@
 public class net.minecraft.world.item.crafting.ArmorDyeRecipe
 extends net.minecraft.world.item.crafting.CustomRecipe
  {
-    public net.minecraft.world.item.crafting.ArmorDyeRecipe(net.minecraft.resources.ResourceLocation $$0) {
-        super($$0);
+    public net.minecraft.world.item.crafting.ArmorDyeRecipe(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1) {
+        super($$0, $$1);
     }
 
-    public boolean matches(net.minecraft.world.inventory.CraftingContainer inv, net.minecraft.world.level.Level level) {
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.EMPTY;
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack2 = inv.getItem(i);
-            if (itemStack2.isEmpty()) continue;
-            if (itemStack2.getItem() instanceof net.minecraft.world.item.DyeableLeatherItem) {
-                if (!itemStack.isEmpty()) {
+    public boolean matches(net.minecraft.world.inventory.CraftingContainer $$0, net.minecraft.world.level.Level $$1) {
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        java.util.ArrayList $$3 = com.google.common.collect.Lists.newArrayList();
+        for (int $$4 = 0; $$4 < $$0.getContainerSize(); ++$$4) {
+            net.minecraft.world.item.ItemStack $$5 = $$0.getItem($$4);
+            if ($$5.isEmpty()) continue;
+            if ($$5.getItem() instanceof net.minecraft.world.item.DyeableLeatherItem) {
+                if (!$$2.isEmpty()) {
                     return false;
                 }
-                itemStack = itemStack2;
+                $$2 = $$5;
                 continue;
             }
-            if (itemStack2.getItem() instanceof net.minecraft.world.item.DyeItem) {
-                arrayList.add((java.lang.Object)itemStack2);
+            if ($$5.getItem() instanceof net.minecraft.world.item.DyeItem) {
+                $$3.add((java.lang.Object)$$5);
                 continue;
             }
             return false;
         }
-        return !itemStack.isEmpty() && !arrayList.isEmpty();
+        return !$$2.isEmpty() && !$$3.isEmpty();
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer inv) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.EMPTY;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack2 = inv.getItem(i);
-            if (itemStack2.isEmpty()) continue;
-            net.minecraft.world.item.Item item = itemStack2.getItem();
-            if (item instanceof net.minecraft.world.item.DyeableLeatherItem) {
-                if (!itemStack.isEmpty()) {
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer $$0) {
+        java.util.ArrayList $$1 = com.google.common.collect.Lists.newArrayList();
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        for (int $$3 = 0; $$3 < $$0.getContainerSize(); ++$$3) {
+            net.minecraft.world.item.ItemStack $$4 = $$0.getItem($$3);
+            if ($$4.isEmpty()) continue;
+            net.minecraft.world.item.Item $$5 = $$4.getItem();
+            if ($$5 instanceof net.minecraft.world.item.DyeableLeatherItem) {
+                if (!$$2.isEmpty()) {
                     return net.minecraft.world.item.ItemStack.EMPTY;
                 }
-                itemStack = itemStack2.copy();
+                $$2 = $$4.copy();
                 continue;
             }
-            if (item instanceof net.minecraft.world.item.DyeItem) {
-                arrayList.add((java.lang.Object)((net.minecraft.world.item.DyeItem)item));
+            if ($$5 instanceof net.minecraft.world.item.DyeItem) {
+                $$1.add((java.lang.Object)((net.minecraft.world.item.DyeItem)$$5));
                 continue;
             }
             return net.minecraft.world.item.ItemStack.EMPTY;
         }
-        if (itemStack.isEmpty() || arrayList.isEmpty()) {
+        if ($$2.isEmpty() || $$1.isEmpty()) {
             return net.minecraft.world.item.ItemStack.EMPTY;
         }
-        return net.minecraft.world.item.DyeableLeatherItem.dyeArmor((net.minecraft.world.item.ItemStack)itemStack, (java.util.List)arrayList);
+        return net.minecraft.world.item.DyeableLeatherItem.dyeArmor((net.minecraft.world.item.ItemStack)$$2, (java.util.List)$$1);
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
-        return width * height >= 2;
+    public boolean canCraftInDimensions(int $$0, int $$1) {
+        return $$0 * $$1 >= 2;
     }
 
     public net.minecraft.world.item.crafting.RecipeSerializer<?> getSerializer() {