--- net/minecraft/world/item/crafting/BannerDuplicateRecipe.java
+++ net/minecraft/world/item/crafting/BannerDuplicateRecipe.java
@@ -6,82 +6,82 @@
 public class net.minecraft.world.item.crafting.BannerDuplicateRecipe
 extends net.minecraft.world.item.crafting.CustomRecipe
  {
-    public net.minecraft.world.item.crafting.BannerDuplicateRecipe(net.minecraft.resources.ResourceLocation $$0) {
-        super($$0);
+    public net.minecraft.world.item.crafting.BannerDuplicateRecipe(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1) {
+        super($$0, $$1);
     }
 
-    public boolean matches(net.minecraft.world.inventory.CraftingContainer inv, net.minecraft.world.level.Level level) {
-        net.minecraft.world.item.DyeColor dyeColor = null;
-        net.minecraft.world.item.ItemStack itemStack = null;
-        net.minecraft.world.item.ItemStack itemStack2 = null;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack3 = inv.getItem(i);
-            if (itemStack3.isEmpty()) continue;
-            net.minecraft.world.item.Item item = itemStack3.getItem();
-            if (!(item instanceof net.minecraft.world.item.BannerItem)) {
+    public boolean matches(net.minecraft.world.inventory.CraftingContainer $$0, net.minecraft.world.level.Level $$1) {
+        net.minecraft.world.item.DyeColor $$2 = null;
+        net.minecraft.world.item.ItemStack $$3 = null;
+        net.minecraft.world.item.ItemStack $$4 = null;
+        for (int $$5 = 0; $$5 < $$0.getContainerSize(); ++$$5) {
+            net.minecraft.world.item.ItemStack $$6 = $$0.getItem($$5);
+            if ($$6.isEmpty()) continue;
+            net.minecraft.world.item.Item $$7 = $$6.getItem();
+            if (!($$7 instanceof net.minecraft.world.item.BannerItem)) {
                 return false;
             }
-            net.minecraft.world.item.BannerItem bannerItem = (net.minecraft.world.item.BannerItem)item;
-            if (dyeColor == null) {
-                dyeColor = bannerItem.getColor();
-            } else if (dyeColor != bannerItem.getColor()) {
+            net.minecraft.world.item.BannerItem $$8 = (net.minecraft.world.item.BannerItem)$$7;
+            if ($$2 == null) {
+                $$2 = $$8.getColor();
+            } else if ($$2 != $$8.getColor()) {
                 return false;
             }
-            int n = net.minecraft.world.level.block.entity.BannerBlockEntity.getPatternCount((net.minecraft.world.item.ItemStack)itemStack3);
-            if (n > 6) {
+            int $$9 = net.minecraft.world.level.block.entity.BannerBlockEntity.getPatternCount((net.minecraft.world.item.ItemStack)$$6);
+            if ($$9 > 6) {
                 return false;
             }
-            if (n > 0) {
-                if (itemStack == null) {
-                    itemStack = itemStack3;
+            if ($$9 > 0) {
+                if ($$3 == null) {
+                    $$3 = $$6;
                     continue;
                 }
                 return false;
             }
-            if (itemStack2 == null) {
-                itemStack2 = itemStack3;
+            if ($$4 == null) {
+                $$4 = $$6;
                 continue;
             }
             return false;
         }
-        return itemStack != null && itemStack2 != null;
+        return $$3 != null && $$4 != null;
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer inv) {
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            int n;
-            net.minecraft.world.item.ItemStack itemStack = inv.getItem(i);
-            if (itemStack.isEmpty() || (n = net.minecraft.world.level.block.entity.BannerBlockEntity.getPatternCount((net.minecraft.world.item.ItemStack)itemStack)) <= 0 || n > 6) continue;
-            net.minecraft.world.item.ItemStack itemStack2 = itemStack.copy();
-            itemStack2.setCount(1);
-            return itemStack2;
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer $$0) {
+        for (int $$1 = 0; $$1 < $$0.getContainerSize(); ++$$1) {
+            int $$3;
+            net.minecraft.world.item.ItemStack $$2 = $$0.getItem($$1);
+            if ($$2.isEmpty() || ($$3 = net.minecraft.world.level.block.entity.BannerBlockEntity.getPatternCount((net.minecraft.world.item.ItemStack)$$2)) <= 0 || $$3 > 6) continue;
+            net.minecraft.world.item.ItemStack $$4 = $$2.copy();
+            $$4.setCount(1);
+            return $$4;
         }
         return net.minecraft.world.item.ItemStack.EMPTY;
     }
 
-    public net.minecraft.core.NonNullList<net.minecraft.world.item.ItemStack> getRemainingItems(net.minecraft.world.inventory.CraftingContainer inv) {
-        net.minecraft.core.NonNullList nonNullList = net.minecraft.core.NonNullList.withSize((int)inv.getContainerSize(), (java.lang.Object)net.minecraft.world.item.ItemStack.EMPTY);
-        for (int i = 0; i < nonNullList.size(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack = inv.getItem(i);
-            if (itemStack.isEmpty()) continue;
-            if (itemStack.getItem().hasCraftingRemainingItem()) {
-                nonNullList.set(i, (java.lang.Object)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)itemStack.getItem().getCraftingRemainingItem()));
+    public net.minecraft.core.NonNullList<net.minecraft.world.item.ItemStack> getRemainingItems(net.minecraft.world.inventory.CraftingContainer $$0) {
+        net.minecraft.core.NonNullList $$1 = net.minecraft.core.NonNullList.withSize((int)$$0.getContainerSize(), (java.lang.Object)net.minecraft.world.item.ItemStack.EMPTY);
+        for (int $$2 = 0; $$2 < $$1.size(); ++$$2) {
+            net.minecraft.world.item.ItemStack $$3 = $$0.getItem($$2);
+            if ($$3.isEmpty()) continue;
+            if ($$3.getItem().hasCraftingRemainingItem()) {
+                $$1.set($$2, (java.lang.Object)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$3.getItem().getCraftingRemainingItem()));
                 continue;
             }
-            if (!itemStack.hasTag() || net.minecraft.world.level.block.entity.BannerBlockEntity.getPatternCount((net.minecraft.world.item.ItemStack)itemStack) <= 0) continue;
-            net.minecraft.world.item.ItemStack itemStack2 = itemStack.copy();
-            itemStack2.setCount(1);
-            nonNullList.set(i, (java.lang.Object)itemStack2);
+            if (!$$3.hasTag() || net.minecraft.world.level.block.entity.BannerBlockEntity.getPatternCount((net.minecraft.world.item.ItemStack)$$3) <= 0) continue;
+            net.minecraft.world.item.ItemStack $$4 = $$3.copy();
+            $$4.setCount(1);
+            $$1.set($$2, (java.lang.Object)$$4);
         }
-        return nonNullList;
+        return $$1;
     }
 
     public net.minecraft.world.item.crafting.RecipeSerializer<?> getSerializer() {
         return net.minecraft.world.item.crafting.RecipeSerializer.BANNER_DUPLICATE;
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
-        return width * height >= 2;
+    public boolean canCraftInDimensions(int $$0, int $$1) {
+        return $$0 * $$1 >= 2;
     }
 }
 