--- net/minecraft/world/item/crafting/CookingBookCategory.java
+++ net/minecraft/world/item/crafting/CookingBookCategory.java
@@ -1,0 +1,46 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item.crafting;
+
+public class net.minecraft.world.item.crafting.CookingBookCategory
+extends java.lang.Enum<net.minecraft.world.item.crafting.CookingBookCategory>
+implements net.minecraft.util.StringRepresentable
+ {
+    public static /* enum */ net.minecraft.world.item.crafting.CookingBookCategory FOOD;
+    public static /* enum */ net.minecraft.world.item.crafting.CookingBookCategory BLOCKS;
+    public static /* enum */ net.minecraft.world.item.crafting.CookingBookCategory MISC;
+    public static net.minecraft.util.StringRepresentable$EnumCodec<net.minecraft.world.item.crafting.CookingBookCategory> CODEC;
+    public java.lang.String name;
+    public static /* synthetic */ net.minecraft.world.item.crafting.CookingBookCategory[] $VALUES;
+
+    public static net.minecraft.world.item.crafting.CookingBookCategory[] values() {
+        return (net.minecraft.world.item.crafting.CookingBookCategory[])$VALUES.clone();
+    }
+
+    public static net.minecraft.world.item.crafting.CookingBookCategory valueOf(java.lang.String $$0) {
+        return (net.minecraft.world.item.crafting.CookingBookCategory)java.lang.Enum.valueOf(net.minecraft.world.item.crafting.CookingBookCategory.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.world.item.crafting.CookingBookCategory(java.lang.String $$0) {
+        super(string, n);
+        this.name = $$0;
+    }
+
+    public java.lang.String getSerializedName() {
+        return this.name;
+    }
+
+    public static /* synthetic */ net.minecraft.world.item.crafting.CookingBookCategory[] $values() {
+        return new net.minecraft.world.item.crafting.CookingBookCategory[]{FOOD, BLOCKS, MISC};
+    }
+
+    static {
+        FOOD = new net.minecraft.world.item.crafting.CookingBookCategory("FOOD", 0, "food");
+        BLOCKS = new net.minecraft.world.item.crafting.CookingBookCategory("BLOCKS", 1, "blocks");
+        MISC = new net.minecraft.world.item.crafting.CookingBookCategory("MISC", 2, "misc");
+        $VALUES = net.minecraft.world.item.crafting.CookingBookCategory.$values();
+        CODEC = net.minecraft.util.StringRepresentable.fromEnum(net.minecraft.world.item.crafting.CookingBookCategory::values);
+    }
+}
+