--- net/minecraft/world/item/crafting/CraftingRecipe.java
+++ net/minecraft/world/item/crafting/CraftingRecipe.java
@@ -1,13 +1,1 @@
-/*
- * Decompiled with CFR 0.151.
- */
-package net.minecraft.world.item.crafting;
-
-public interface net.minecraft.world.item.crafting.CraftingRecipe
-extends net.minecraft.world.item.crafting.Recipe<net.minecraft.world.inventory.CraftingContainer>
- {
-    default public net.minecraft.world.item.crafting.RecipeType<?> getType() {
-        return net.minecraft.world.item.crafting.RecipeType.CRAFTING;
-    }
-}
-
+// Failed to decompile net/minecraft/world/item/crafting/CraftingRecipe.class