--- net/minecraft/world/item/crafting/CraftingBookCategory.java
+++ net/minecraft/world/item/crafting/CraftingBookCategory.java
@@ -1,0 +1,48 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item.crafting;
+
+public class net.minecraft.world.item.crafting.CraftingBookCategory
+extends java.lang.Enum<net.minecraft.world.item.crafting.CraftingBookCategory>
+implements net.minecraft.util.StringRepresentable
+ {
+    public static /* enum */ net.minecraft.world.item.crafting.CraftingBookCategory BUILDING;
+    public static /* enum */ net.minecraft.world.item.crafting.CraftingBookCategory REDSTONE;
+    public static /* enum */ net.minecraft.world.item.crafting.CraftingBookCategory EQUIPMENT;
+    public static /* enum */ net.minecraft.world.item.crafting.CraftingBookCategory MISC;
+    public static net.minecraft.util.StringRepresentable$EnumCodec<net.minecraft.world.item.crafting.CraftingBookCategory> CODEC;
+    public java.lang.String name;
+    public static /* synthetic */ net.minecraft.world.item.crafting.CraftingBookCategory[] $VALUES;
+
+    public static net.minecraft.world.item.crafting.CraftingBookCategory[] values() {
+        return (net.minecraft.world.item.crafting.CraftingBookCategory[])$VALUES.clone();
+    }
+
+    public static net.minecraft.world.item.crafting.CraftingBookCategory valueOf(java.lang.String $$0) {
+        return (net.minecraft.world.item.crafting.CraftingBookCategory)java.lang.Enum.valueOf(net.minecraft.world.item.crafting.CraftingBookCategory.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.world.item.crafting.CraftingBookCategory(java.lang.String $$0) {
+        super(string, n);
+        this.name = $$0;
+    }
+
+    public java.lang.String getSerializedName() {
+        return this.name;
+    }
+
+    public static /* synthetic */ net.minecraft.world.item.crafting.CraftingBookCategory[] $values() {
+        return new net.minecraft.world.item.crafting.CraftingBookCategory[]{BUILDING, REDSTONE, EQUIPMENT, MISC};
+    }
+
+    static {
+        BUILDING = new net.minecraft.world.item.crafting.CraftingBookCategory("BUILDING", 0, "building");
+        REDSTONE = new net.minecraft.world.item.crafting.CraftingBookCategory("REDSTONE", 1, "redstone");
+        EQUIPMENT = new net.minecraft.world.item.crafting.CraftingBookCategory("EQUIPMENT", 2, "equipment");
+        MISC = new net.minecraft.world.item.crafting.CraftingBookCategory("MISC", 3, "misc");
+        $VALUES = net.minecraft.world.item.crafting.CraftingBookCategory.$values();
+        CODEC = net.minecraft.util.StringRepresentable.fromEnum(net.minecraft.world.item.crafting.CraftingBookCategory::values);
+    }
+}
+