--- net/minecraft/world/item/crafting/FireworkStarFadeRecipe.java
+++ net/minecraft/world/item/crafting/FireworkStarFadeRecipe.java
@@ -8,55 +8,55 @@
  {
     public static net.minecraft.world.item.crafting.Ingredient STAR_INGREDIENT;
 
-    public net.minecraft.world.item.crafting.FireworkStarFadeRecipe(net.minecraft.resources.ResourceLocation $$0) {
-        super($$0);
+    public net.minecraft.world.item.crafting.FireworkStarFadeRecipe(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1) {
+        super($$0, $$1);
     }
 
-    public boolean matches(net.minecraft.world.inventory.CraftingContainer inv, net.minecraft.world.level.Level level) {
-        boolean bl = false;
-        boolean bl2 = false;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack = inv.getItem(i);
-            if (itemStack.isEmpty()) continue;
-            if (itemStack.getItem() instanceof net.minecraft.world.item.DyeItem) {
-                bl = true;
+    public boolean matches(net.minecraft.world.inventory.CraftingContainer $$0, net.minecraft.world.level.Level $$1) {
+        boolean $$2 = false;
+        boolean $$3 = false;
+        for (int $$4 = 0; $$4 < $$0.getContainerSize(); ++$$4) {
+            net.minecraft.world.item.ItemStack $$5 = $$0.getItem($$4);
+            if ($$5.isEmpty()) continue;
+            if ($$5.getItem() instanceof net.minecraft.world.item.DyeItem) {
+                $$2 = true;
                 continue;
             }
-            if (STAR_INGREDIENT.test(itemStack)) {
-                if (bl2) {
+            if (STAR_INGREDIENT.test($$5)) {
+                if ($$3) {
                     return false;
                 }
-                bl2 = true;
+                $$3 = true;
                 continue;
             }
             return false;
         }
-        return bl2 && bl;
+        return $$3 && $$2;
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer inv) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        net.minecraft.world.item.ItemStack itemStack = null;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack2 = inv.getItem(i);
-            net.minecraft.world.item.Item item = itemStack2.getItem();
-            if (item instanceof net.minecraft.world.item.DyeItem) {
-                arrayList.add((java.lang.Object)((net.minecraft.world.item.DyeItem)item).getDyeColor().getFireworkColor());
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer $$0) {
+        java.util.ArrayList $$1 = com.google.common.collect.Lists.newArrayList();
+        net.minecraft.world.item.ItemStack $$2 = null;
+        for (int $$3 = 0; $$3 < $$0.getContainerSize(); ++$$3) {
+            net.minecraft.world.item.ItemStack $$4 = $$0.getItem($$3);
+            net.minecraft.world.item.Item $$5 = $$4.getItem();
+            if ($$5 instanceof net.minecraft.world.item.DyeItem) {
+                $$1.add((java.lang.Object)((net.minecraft.world.item.DyeItem)$$5).getDyeColor().getFireworkColor());
                 continue;
             }
-            if (!STAR_INGREDIENT.test(itemStack2)) continue;
-            itemStack = itemStack2.copy();
-            itemStack.setCount(1);
+            if (!STAR_INGREDIENT.test($$4)) continue;
+            $$2 = $$4.copy();
+            $$2.setCount(1);
         }
-        if (itemStack == null || arrayList.isEmpty()) {
+        if ($$2 == null || $$1.isEmpty()) {
             return net.minecraft.world.item.ItemStack.EMPTY;
         }
-        itemStack.getOrCreateTagElement("Explosion").putIntArray("FadeColors", (java.util.List)arrayList);
-        return itemStack;
+        $$2.getOrCreateTagElement("Explosion").putIntArray("FadeColors", (java.util.List)$$1);
+        return $$2;
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
-        return width * height >= 2;
+    public boolean canCraftInDimensions(int $$0, int $$1) {
+        return $$0 * $$1 >= 2;
     }
 
     public net.minecraft.world.item.crafting.RecipeSerializer<?> getSerializer() {