--- net/minecraft/world/item/crafting/FireworkRocketRecipe.java
+++ net/minecraft/world/item/crafting/FireworkRocketRecipe.java
@@ -10,54 +10,54 @@
     public static net.minecraft.world.item.crafting.Ingredient GUNPOWDER_INGREDIENT;
     public static net.minecraft.world.item.crafting.Ingredient STAR_INGREDIENT;
 
-    public net.minecraft.world.item.crafting.FireworkRocketRecipe(net.minecraft.resources.ResourceLocation $$0) {
-        super($$0);
+    public net.minecraft.world.item.crafting.FireworkRocketRecipe(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1) {
+        super($$0, $$1);
     }
 
-    public boolean matches(net.minecraft.world.inventory.CraftingContainer inv, net.minecraft.world.level.Level level) {
-        boolean bl = false;
-        int n = 0;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack = inv.getItem(i);
-            if (itemStack.isEmpty()) continue;
-            if (PAPER_INGREDIENT.test(itemStack)) {
-                if (bl) {
+    public boolean matches(net.minecraft.world.inventory.CraftingContainer $$0, net.minecraft.world.level.Level $$1) {
+        boolean $$2 = false;
+        int $$3 = 0;
+        for (int $$4 = 0; $$4 < $$0.getContainerSize(); ++$$4) {
+            net.minecraft.world.item.ItemStack $$5 = $$0.getItem($$4);
+            if ($$5.isEmpty()) continue;
+            if (PAPER_INGREDIENT.test($$5)) {
+                if ($$2) {
                     return false;
                 }
-                bl = true;
+                $$2 = true;
                 continue;
             }
-            if (!(GUNPOWDER_INGREDIENT.test(itemStack) ? ++n > 3 : !STAR_INGREDIENT.test(itemStack))) continue;
+            if (!(GUNPOWDER_INGREDIENT.test($$5) ? ++$$3 > 3 : !STAR_INGREDIENT.test($$5))) continue;
             return false;
         }
-        return bl && n >= 1;
+        return $$2 && $$3 >= 1;
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer inv) {
-        net.minecraft.world.item.ItemStack itemStack = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FIREWORK_ROCKET, 3);
-        net.minecraft.nbt.CompoundTag compoundTag = itemStack.getOrCreateTagElement("Fireworks");
-        net.minecraft.nbt.ListTag listTag = new net.minecraft.nbt.ListTag();
-        int n = 0;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.nbt.CompoundTag compoundTag2;
-            net.minecraft.world.item.ItemStack itemStack2 = inv.getItem(i);
-            if (itemStack2.isEmpty()) continue;
-            if (GUNPOWDER_INGREDIENT.test(itemStack2)) {
-                ++n;
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer $$0) {
+        net.minecraft.world.item.ItemStack $$1 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FIREWORK_ROCKET, 3);
+        net.minecraft.nbt.CompoundTag $$2 = $$1.getOrCreateTagElement("Fireworks");
+        net.minecraft.nbt.ListTag $$3 = new net.minecraft.nbt.ListTag();
+        int $$4 = 0;
+        for (int $$5 = 0; $$5 < $$0.getContainerSize(); ++$$5) {
+            net.minecraft.nbt.CompoundTag $$7;
+            net.minecraft.world.item.ItemStack $$6 = $$0.getItem($$5);
+            if ($$6.isEmpty()) continue;
+            if (GUNPOWDER_INGREDIENT.test($$6)) {
+                ++$$4;
                 continue;
             }
-            if (!STAR_INGREDIENT.test(itemStack2) || (compoundTag2 = itemStack2.getTagElement("Explosion")) == null) continue;
-            listTag.add((java.lang.Object)compoundTag2);
+            if (!STAR_INGREDIENT.test($$6) || ($$7 = $$6.getTagElement("Explosion")) == null) continue;
+            $$3.add((java.lang.Object)$$7);
         }
-        compoundTag.putByte("Flight", (byte)n);
-        if (!listTag.isEmpty()) {
-            compoundTag.put("Explosions", (net.minecraft.nbt.Tag)listTag);
+        $$2.putByte("Flight", (byte)$$4);
+        if (!$$3.isEmpty()) {
+            $$2.put("Explosions", (net.minecraft.nbt.Tag)$$3);
         }
-        return itemStack;
+        return $$1;
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
-        return width * height >= 2;
+    public boolean canCraftInDimensions(int $$0, int $$1) {
+        return $$0 * $$1 >= 2;
     }
 
     public net.minecraft.world.item.ItemStack getResultItem() {