--- net/minecraft/world/item/crafting/FireworkStarRecipe.java
+++ net/minecraft/world/item/crafting/FireworkStarRecipe.java
@@ -12,86 +12,86 @@
     public static java.util.Map<net.minecraft.world.item.Item, net.minecraft.world.item.FireworkRocketItem$Shape> SHAPE_BY_ITEM;
     public static net.minecraft.world.item.crafting.Ingredient GUNPOWDER_INGREDIENT;
 
-    public net.minecraft.world.item.crafting.FireworkStarRecipe(net.minecraft.resources.ResourceLocation $$0) {
-        super($$0);
+    public net.minecraft.world.item.crafting.FireworkStarRecipe(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1) {
+        super($$0, $$1);
     }
 
-    public boolean matches(net.minecraft.world.inventory.CraftingContainer inv, net.minecraft.world.level.Level level) {
-        boolean bl = false;
-        boolean bl2 = false;
-        boolean bl3 = false;
-        boolean bl4 = false;
-        boolean bl5 = false;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack = inv.getItem(i);
-            if (itemStack.isEmpty()) continue;
-            if (SHAPE_INGREDIENT.test(itemStack)) {
-                if (bl3) {
+    public boolean matches(net.minecraft.world.inventory.CraftingContainer $$0, net.minecraft.world.level.Level $$1) {
+        boolean $$2 = false;
+        boolean $$3 = false;
+        boolean $$4 = false;
+        boolean $$5 = false;
+        boolean $$6 = false;
+        for (int $$7 = 0; $$7 < $$0.getContainerSize(); ++$$7) {
+            net.minecraft.world.item.ItemStack $$8 = $$0.getItem($$7);
+            if ($$8.isEmpty()) continue;
+            if (SHAPE_INGREDIENT.test($$8)) {
+                if ($$4) {
                     return false;
                 }
-                bl3 = true;
+                $$4 = true;
                 continue;
             }
-            if (FLICKER_INGREDIENT.test(itemStack)) {
-                if (bl5) {
+            if (FLICKER_INGREDIENT.test($$8)) {
+                if ($$6) {
                     return false;
                 }
-                bl5 = true;
+                $$6 = true;
                 continue;
             }
-            if (TRAIL_INGREDIENT.test(itemStack)) {
-                if (bl4) {
+            if (TRAIL_INGREDIENT.test($$8)) {
+                if ($$5) {
                     return false;
                 }
-                bl4 = true;
+                $$5 = true;
                 continue;
             }
-            if (GUNPOWDER_INGREDIENT.test(itemStack)) {
-                if (bl) {
+            if (GUNPOWDER_INGREDIENT.test($$8)) {
+                if ($$2) {
                     return false;
                 }
-                bl = true;
+                $$2 = true;
                 continue;
             }
-            if (itemStack.getItem() instanceof net.minecraft.world.item.DyeItem) {
-                bl2 = true;
+            if ($$8.getItem() instanceof net.minecraft.world.item.DyeItem) {
+                $$3 = true;
                 continue;
             }
             return false;
         }
-        return bl && bl2;
+        return $$2 && $$3;
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer inv) {
-        net.minecraft.world.item.ItemStack itemStack = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FIREWORK_STAR);
-        net.minecraft.nbt.CompoundTag compoundTag = itemStack.getOrCreateTagElement("Explosion");
-        net.minecraft.world.item.FireworkRocketItem$Shape shape = net.minecraft.world.item.FireworkRocketItem$Shape.SMALL_BALL;
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack2 = inv.getItem(i);
-            if (itemStack2.isEmpty()) continue;
-            if (SHAPE_INGREDIENT.test(itemStack2)) {
-                shape = (net.minecraft.world.item.FireworkRocketItem$Shape)SHAPE_BY_ITEM.get((java.lang.Object)itemStack2.getItem());
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer $$0) {
+        net.minecraft.world.item.ItemStack $$1 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FIREWORK_STAR);
+        net.minecraft.nbt.CompoundTag $$2 = $$1.getOrCreateTagElement("Explosion");
+        net.minecraft.world.item.FireworkRocketItem$Shape $$3 = net.minecraft.world.item.FireworkRocketItem$Shape.SMALL_BALL;
+        java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayList();
+        for (int $$5 = 0; $$5 < $$0.getContainerSize(); ++$$5) {
+            net.minecraft.world.item.ItemStack $$6 = $$0.getItem($$5);
+            if ($$6.isEmpty()) continue;
+            if (SHAPE_INGREDIENT.test($$6)) {
+                $$3 = (net.minecraft.world.item.FireworkRocketItem$Shape)SHAPE_BY_ITEM.get((java.lang.Object)$$6.getItem());
                 continue;
             }
-            if (FLICKER_INGREDIENT.test(itemStack2)) {
-                compoundTag.putBoolean("Flicker", true);
+            if (FLICKER_INGREDIENT.test($$6)) {
+                $$2.putBoolean("Flicker", true);
                 continue;
             }
-            if (TRAIL_INGREDIENT.test(itemStack2)) {
-                compoundTag.putBoolean("Trail", true);
+            if (TRAIL_INGREDIENT.test($$6)) {
+                $$2.putBoolean("Trail", true);
                 continue;
             }
-            if (!(itemStack2.getItem() instanceof net.minecraft.world.item.DyeItem)) continue;
-            arrayList.add((java.lang.Object)((net.minecraft.world.item.DyeItem)itemStack2.getItem()).getDyeColor().getFireworkColor());
+            if (!($$6.getItem() instanceof net.minecraft.world.item.DyeItem)) continue;
+            $$4.add((java.lang.Object)((net.minecraft.world.item.DyeItem)$$6.getItem()).getDyeColor().getFireworkColor());
         }
-        compoundTag.putIntArray("Colors", (java.util.List)arrayList);
-        compoundTag.putByte("Type", (byte)shape.getId());
-        return itemStack;
+        $$2.putIntArray("Colors", (java.util.List)$$4);
+        $$2.putByte("Type", (byte)$$3.getId());
+        return $$1;
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
-        return width * height >= 2;
+    public boolean canCraftInDimensions(int $$0, int $$1) {
+        return $$0 * $$1 >= 2;
     }
 
     public net.minecraft.world.item.ItemStack getResultItem() {
@@ -102,32 +102,32 @@
         return net.minecraft.world.item.crafting.RecipeSerializer.FIREWORK_STAR;
     }
 
-    public static /* synthetic */ void lambda$static$0(java.util.HashMap shapeMap) {
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.FIRE_CHARGE, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.LARGE_BALL);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.FEATHER, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.BURST);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.GOLD_NUGGET, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.STAR);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.WITHER_SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.CREEPER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.PLAYER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.DRAGON_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-        shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.ZOMBIE_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+    public static /* synthetic */ void lambda$static$0(java.util.HashMap $$0) {
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.FIRE_CHARGE, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.LARGE_BALL);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.FEATHER, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.BURST);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.GOLD_NUGGET, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.STAR);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.WITHER_SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.CREEPER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.PLAYER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.DRAGON_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+        $$0.put((java.lang.Object)net.minecraft.world.item.Items.ZOMBIE_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
     }
 
     static {
         SHAPE_INGREDIENT = net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.FIRE_CHARGE, net.minecraft.world.item.Items.FEATHER, net.minecraft.world.item.Items.GOLD_NUGGET, net.minecraft.world.item.Items.SKELETON_SKULL, net.minecraft.world.item.Items.WITHER_SKELETON_SKULL, net.minecraft.world.item.Items.CREEPER_HEAD, net.minecraft.world.item.Items.PLAYER_HEAD, net.minecraft.world.item.Items.DRAGON_HEAD, net.minecraft.world.item.Items.ZOMBIE_HEAD});
         TRAIL_INGREDIENT = net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.DIAMOND});
         FLICKER_INGREDIENT = net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.GLOWSTONE_DUST});
-        SHAPE_BY_ITEM = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), shapeMap -> {
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.FIRE_CHARGE, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.LARGE_BALL);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.FEATHER, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.BURST);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.GOLD_NUGGET, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.STAR);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.WITHER_SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.CREEPER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.PLAYER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.DRAGON_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
-            shapeMap.put((java.lang.Object)net.minecraft.world.item.Items.ZOMBIE_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+        SHAPE_BY_ITEM = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), $$0 -> {
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.FIRE_CHARGE, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.LARGE_BALL);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.FEATHER, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.BURST);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.GOLD_NUGGET, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.STAR);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.WITHER_SKELETON_SKULL, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.CREEPER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.PLAYER_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.DRAGON_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
+            $$0.put((java.lang.Object)net.minecraft.world.item.Items.ZOMBIE_HEAD, (java.lang.Object)net.minecraft.world.item.FireworkRocketItem$Shape.CREEPER);
         });
         GUNPOWDER_INGREDIENT = net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.GUNPOWDER});
     }