--- net/minecraft/world/item/crafting/RecipeType.java
+++ net/minecraft/world/item/crafting/RecipeType.java
@@ -13,8 +13,8 @@
     public static final net.minecraft.world.item.crafting.RecipeType<net.minecraft.world.item.crafting.StonecutterRecipe> STONECUTTING;
     public static final net.minecraft.world.item.crafting.RecipeType<net.minecraft.world.item.crafting.UpgradeRecipe> SMITHING;
 
-    public static <T extends net.minecraft.world.item.crafting.Recipe<?>> net.minecraft.world.item.crafting.RecipeType<T> register(java.lang.String identifier) {
-        return (net.minecraft.world.item.crafting.RecipeType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.RECIPE_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(identifier), (java.lang.Object)new /* Unavailable Anonymous Inner Class!! */);
+    public static <T extends net.minecraft.world.item.crafting.Recipe<?>> net.minecraft.world.item.crafting.RecipeType<T> register(java.lang.String $$0) {
+        return (net.minecraft.world.item.crafting.RecipeType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.RECIPE_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0), (java.lang.Object)new /* Unavailable Anonymous Inner Class!! */);
     }
 
     static {