--- net/minecraft/world/item/crafting/ShieldDecorationRecipe.java
+++ net/minecraft/world/item/crafting/ShieldDecorationRecipe.java
@@ -6,64 +6,63 @@
 public class net.minecraft.world.item.crafting.ShieldDecorationRecipe
 extends net.minecraft.world.item.crafting.CustomRecipe
  {
-    public net.minecraft.world.item.crafting.ShieldDecorationRecipe(net.minecraft.resources.ResourceLocation $$0) {
-        super($$0);
+    public net.minecraft.world.item.crafting.ShieldDecorationRecipe(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1) {
+        super($$0, $$1);
     }
 
-    public boolean matches(net.minecraft.world.inventory.CraftingContainer inv, net.minecraft.world.level.Level level) {
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.EMPTY;
-        net.minecraft.world.item.ItemStack itemStack2 = net.minecraft.world.item.ItemStack.EMPTY;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            net.minecraft.world.item.ItemStack itemStack3 = inv.getItem(i);
-            if (itemStack3.isEmpty()) continue;
-            if (itemStack3.getItem() instanceof net.minecraft.world.item.BannerItem) {
-                if (!itemStack2.isEmpty()) {
+    public boolean matches(net.minecraft.world.inventory.CraftingContainer $$0, net.minecraft.world.level.Level $$1) {
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        net.minecraft.world.item.ItemStack $$3 = net.minecraft.world.item.ItemStack.EMPTY;
+        for (int $$4 = 0; $$4 < $$0.getContainerSize(); ++$$4) {
+            net.minecraft.world.item.ItemStack $$5 = $$0.getItem($$4);
+            if ($$5.isEmpty()) continue;
+            if ($$5.getItem() instanceof net.minecraft.world.item.BannerItem) {
+                if (!$$3.isEmpty()) {
                     return false;
                 }
-                itemStack2 = itemStack3;
+                $$3 = $$5;
                 continue;
             }
-            if (itemStack3.is(net.minecraft.world.item.Items.SHIELD)) {
-                if (!itemStack.isEmpty()) {
+            if ($$5.is(net.minecraft.world.item.Items.SHIELD)) {
+                if (!$$2.isEmpty()) {
                     return false;
                 }
-                if (net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)itemStack3) != null) {
+                if (net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$5) != null) {
                     return false;
                 }
-                itemStack = itemStack3;
+                $$2 = $$5;
                 continue;
             }
             return false;
         }
-        return !itemStack.isEmpty() && !itemStack2.isEmpty();
+        return !$$2.isEmpty() && !$$3.isEmpty();
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer inv) {
-        net.minecraft.world.item.ItemStack itemStack;
-        net.minecraft.world.item.ItemStack itemStack2 = net.minecraft.world.item.ItemStack.EMPTY;
-        net.minecraft.world.item.ItemStack itemStack3 = net.minecraft.world.item.ItemStack.EMPTY;
-        for (int i = 0; i < inv.getContainerSize(); ++i) {
-            itemStack = inv.getItem(i);
-            if (itemStack.isEmpty()) continue;
-            if (itemStack.getItem() instanceof net.minecraft.world.item.BannerItem) {
-                itemStack2 = itemStack;
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer $$0) {
+        net.minecraft.world.item.ItemStack $$1 = net.minecraft.world.item.ItemStack.EMPTY;
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        for (int $$3 = 0; $$3 < $$0.getContainerSize(); ++$$3) {
+            net.minecraft.world.item.ItemStack $$4 = $$0.getItem($$3);
+            if ($$4.isEmpty()) continue;
+            if ($$4.getItem() instanceof net.minecraft.world.item.BannerItem) {
+                $$1 = $$4;
                 continue;
             }
-            if (!itemStack.is(net.minecraft.world.item.Items.SHIELD)) continue;
-            itemStack3 = itemStack.copy();
+            if (!$$4.is(net.minecraft.world.item.Items.SHIELD)) continue;
+            $$2 = $$4.copy();
         }
-        if (itemStack3.isEmpty()) {
-            return itemStack3;
+        if ($$2.isEmpty()) {
+            return $$2;
         }
-        net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)itemStack2);
-        itemStack = compoundTag == null ? new net.minecraft.nbt.CompoundTag() : compoundTag.copy();
-        itemStack.putInt("Base", ((net.minecraft.world.item.BannerItem)itemStack2.getItem()).getColor().getId());
-        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)itemStack3, (net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.BANNER, (net.minecraft.nbt.CompoundTag)itemStack);
-        return itemStack3;
+        net.minecraft.nbt.CompoundTag $$5 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$1);
+        net.minecraft.nbt.CompoundTag $$6 = $$5 == null ? new net.minecraft.nbt.CompoundTag() : $$5.copy();
+        $$6.putInt("Base", ((net.minecraft.world.item.BannerItem)$$1.getItem()).getColor().getId());
+        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)$$2, (net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.BANNER, (net.minecraft.nbt.CompoundTag)$$6);
+        return $$2;
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
-        return width * height >= 2;
+    public boolean canCraftInDimensions(int $$0, int $$1) {
+        return $$0 * $$1 >= 2;
     }
 
     public net.minecraft.world.item.crafting.RecipeSerializer<?> getSerializer() {