--- net/minecraft/world/item/crafting/SimpleCraftingRecipeSerializer.java
+++ net/minecraft/world/item/crafting/SimpleCraftingRecipeSerializer.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item.crafting;
+
+public class net.minecraft.world.item.crafting.SimpleCraftingRecipeSerializer<T extends net.minecraft.world.item.crafting.CraftingRecipe>
+implements net.minecraft.world.item.crafting.RecipeSerializer<T>
+ {
+    public net.minecraft.world.item.crafting.SimpleCraftingRecipeSerializer$Factory<T> constructor;
+
+    public net.minecraft.world.item.crafting.SimpleCraftingRecipeSerializer(net.minecraft.world.item.crafting.SimpleCraftingRecipeSerializer$Factory<T> $$0) {
+        this.constructor = $$0;
+    }
+
+    public T fromJson(net.minecraft.resources.ResourceLocation $$0, com.google.gson.JsonObject $$1) {
+        net.minecraft.world.item.crafting.CraftingBookCategory $$2 = (net.minecraft.world.item.crafting.CraftingBookCategory)java.util.Objects.requireNonNullElse((java.lang.Object)((net.minecraft.world.item.crafting.CraftingBookCategory)net.minecraft.world.item.crafting.CraftingBookCategory.CODEC.byName(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$1, (java.lang.String)"category", null))), (java.lang.Object)net.minecraft.world.item.crafting.CraftingBookCategory.MISC);
+        return (T)this.constructor.create($$0, $$2);
+    }
+
+    public T fromNetwork(net.minecraft.resources.ResourceLocation $$0, net.minecraft.network.FriendlyByteBuf $$1) {
+        net.minecraft.world.item.crafting.CraftingBookCategory $$2 = (net.minecraft.world.item.crafting.CraftingBookCategory)$$1.readEnum(net.minecraft.world.item.crafting.CraftingBookCategory.class);
+        return (T)this.constructor.create($$0, $$2);
+    }
+
+    public void toNetwork(net.minecraft.network.FriendlyByteBuf $$0, T $$1) {
+        $$0.writeEnum((java.lang.Enum)$$1.category());
+    }
+
+    public /* synthetic */ net.minecraft.world.item.crafting.Recipe fromNetwork(net.minecraft.resources.ResourceLocation resourceLocation, net.minecraft.network.FriendlyByteBuf friendlyByteBuf) {
+        return this.fromNetwork(resourceLocation, friendlyByteBuf);
+    }
+
+    public /* synthetic */ net.minecraft.world.item.crafting.Recipe fromJson(net.minecraft.resources.ResourceLocation resourceLocation, com.google.gson.JsonObject jsonObject) {
+        return this.fromJson(resourceLocation, jsonObject);
+    }
+}
+