--- net/minecraft/world/item/crafting/SmeltingRecipe.java
+++ net/minecraft/world/item/crafting/SmeltingRecipe.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.item.crafting.SmeltingRecipe
 extends net.minecraft.world.item.crafting.AbstractCookingRecipe
  {
-    public net.minecraft.world.item.crafting.SmeltingRecipe(net.minecraft.resources.ResourceLocation id, java.lang.String group, net.minecraft.world.item.crafting.Ingredient ingredient, net.minecraft.world.item.ItemStack result, float experience, int cookingTime) {
-        super(net.minecraft.world.item.crafting.RecipeType.SMELTING, id, group, ingredient, result, experience, cookingTime);
+    public net.minecraft.world.item.crafting.SmeltingRecipe(net.minecraft.resources.ResourceLocation $$0, java.lang.String $$1, net.minecraft.world.item.crafting.CookingBookCategory $$2, net.minecraft.world.item.crafting.Ingredient $$3, net.minecraft.world.item.ItemStack $$4, float $$5, int $$6) {
+        super(net.minecraft.world.item.crafting.RecipeType.SMELTING, $$0, $$1, $$2, $$3, $$4, $$5, $$6);
     }
 
     public net.minecraft.world.item.ItemStack getToastSymbol() {