--- net/minecraft/world/item/crafting/TippedArrowRecipe.java
+++ net/minecraft/world/item/crafting/TippedArrowRecipe.java
@@ -6,40 +6,40 @@
 public class net.minecraft.world.item.crafting.TippedArrowRecipe
 extends net.minecraft.world.item.crafting.CustomRecipe
  {
-    public net.minecraft.world.item.crafting.TippedArrowRecipe(net.minecraft.resources.ResourceLocation $$0) {
-        super($$0);
+    public net.minecraft.world.item.crafting.TippedArrowRecipe(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1) {
+        super($$0, $$1);
     }
 
-    public boolean matches(net.minecraft.world.inventory.CraftingContainer inv, net.minecraft.world.level.Level level) {
-        if (inv.getWidth() != 3 || inv.getHeight() != 3) {
+    public boolean matches(net.minecraft.world.inventory.CraftingContainer $$0, net.minecraft.world.level.Level $$1) {
+        if ($$0.getWidth() != 3 || $$0.getHeight() != 3) {
             return false;
         }
-        for (int i = 0; i < inv.getWidth(); ++i) {
-            for (int j = 0; j < inv.getHeight(); ++j) {
-                net.minecraft.world.item.ItemStack itemStack = inv.getItem(i + j * inv.getWidth());
-                if (itemStack.isEmpty()) {
+        for (int $$2 = 0; $$2 < $$0.getWidth(); ++$$2) {
+            for (int $$3 = 0; $$3 < $$0.getHeight(); ++$$3) {
+                net.minecraft.world.item.ItemStack $$4 = $$0.getItem($$2 + $$3 * $$0.getWidth());
+                if ($$4.isEmpty()) {
                     return false;
                 }
-                if (!(i == 1 && j == 1 ? !itemStack.is(net.minecraft.world.item.Items.LINGERING_POTION) : !itemStack.is(net.minecraft.world.item.Items.ARROW))) continue;
+                if (!($$2 == 1 && $$3 == 1 ? !$$4.is(net.minecraft.world.item.Items.LINGERING_POTION) : !$$4.is(net.minecraft.world.item.Items.ARROW))) continue;
                 return false;
             }
         }
         return true;
     }
 
-    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer inv) {
-        net.minecraft.world.item.ItemStack itemStack = inv.getItem(1 + inv.getWidth());
-        if (!itemStack.is(net.minecraft.world.item.Items.LINGERING_POTION)) {
+    public net.minecraft.world.item.ItemStack assemble(net.minecraft.world.inventory.CraftingContainer $$0) {
+        net.minecraft.world.item.ItemStack $$1 = $$0.getItem(1 + $$0.getWidth());
+        if (!$$1.is(net.minecraft.world.item.Items.LINGERING_POTION)) {
             return net.minecraft.world.item.ItemStack.EMPTY;
         }
-        net.minecraft.world.item.ItemStack itemStack2 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TIPPED_ARROW, 8);
-        net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)itemStack2, (net.minecraft.world.item.alchemy.Potion)net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)itemStack));
-        net.minecraft.world.item.alchemy.PotionUtils.setCustomEffects((net.minecraft.world.item.ItemStack)itemStack2, (java.util.Collection)net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.world.item.ItemStack)itemStack));
-        return itemStack2;
+        net.minecraft.world.item.ItemStack $$2 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TIPPED_ARROW, 8);
+        net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)$$2, (net.minecraft.world.item.alchemy.Potion)net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)$$1));
+        net.minecraft.world.item.alchemy.PotionUtils.setCustomEffects((net.minecraft.world.item.ItemStack)$$2, (java.util.Collection)net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.world.item.ItemStack)$$1));
+        return $$2;
     }
 
-    public boolean canCraftInDimensions(int width, int height) {
-        return width >= 2 && height >= 2;
+    public boolean canCraftInDimensions(int $$0, int $$1) {
+        return $$0 >= 2 && $$1 >= 2;
     }
 
     public net.minecraft.world.item.crafting.RecipeSerializer<?> getSerializer() {