--- net/minecraft/commands/arguments/item/ItemInput.java
+++ net/minecraft/commands/arguments/item/ItemInput.java
@@ -20,19 +20,19 @@
         return (net.minecraft.world.item.Item)this.item.value();
     }
 
-    public boolean test(net.minecraft.world.item.ItemStack stack) {
-        return stack.is(this.item) && net.minecraft.nbt.NbtUtils.compareNbt((net.minecraft.nbt.Tag)this.tag, (net.minecraft.nbt.Tag)stack.getTag(), (boolean)true);
+    public boolean test(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(this.item) && net.minecraft.nbt.NbtUtils.compareNbt((net.minecraft.nbt.Tag)this.tag, (net.minecraft.nbt.Tag)$$0.getTag(), (boolean)true);
     }
 
-    public net.minecraft.world.item.ItemStack createItemStack(int count, boolean allowOversizedStacks) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        net.minecraft.world.item.ItemStack itemStack = new net.minecraft.world.item.ItemStack(this.item, count);
+    public net.minecraft.world.item.ItemStack createItemStack(int $$0, boolean $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        net.minecraft.world.item.ItemStack $$2 = new net.minecraft.world.item.ItemStack(this.item, $$0);
         if (this.tag != null) {
-            itemStack.setTag(this.tag);
+            $$2.setTag(this.tag);
         }
-        if (allowOversizedStacks && count > itemStack.getMaxStackSize()) {
-            throw ERROR_STACK_TOO_BIG.create((java.lang.Object)this.getItemName(), (java.lang.Object)itemStack.getMaxStackSize());
+        if ($$1 && $$0 > $$2.getMaxStackSize()) {
+            throw ERROR_STACK_TOO_BIG.create((java.lang.Object)this.getItemName(), (java.lang.Object)$$2.getMaxStackSize());
         }
-        return itemStack;
+        return $$2;
     }
 
     public java.lang.String serialize() {
@@ -51,12 +51,12 @@
         return this.test((net.minecraft.world.item.ItemStack)object);
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object item, java.lang.Object quantity) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"arguments.item.overstacked", (java.lang.Object[])new java.lang.Object[]{item, quantity});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object $$0, java.lang.Object $$1) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"arguments.item.overstacked", (java.lang.Object[])new java.lang.Object[]{$$0, $$1});
     }
 
     static {
-        ERROR_STACK_TOO_BIG = new com.mojang.brigadier.exceptions.Dynamic2CommandExceptionType((item, quantity) -> net.minecraft.network.chat.Component.translatable((java.lang.String)"arguments.item.overstacked", (java.lang.Object[])new java.lang.Object[]{item, quantity}));
+        ERROR_STACK_TOO_BIG = new com.mojang.brigadier.exceptions.Dynamic2CommandExceptionType(($$0, $$1) -> net.minecraft.network.chat.Component.translatable((java.lang.String)"arguments.item.overstacked", (java.lang.Object[])new java.lang.Object[]{$$0, $$1}));
     }
 }
 