--- net/minecraft/commands/arguments/item/ItemArgument.java
+++ net/minecraft/commands/arguments/item/ItemArgument.java
@@ -17,17 +17,17 @@
         return new net.minecraft.commands.arguments.item.ItemArgument($$0);
     }
 
-    public net.minecraft.commands.arguments.item.ItemInput parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        net.minecraft.commands.arguments.item.ItemParser$ItemResult itemResult = net.minecraft.commands.arguments.item.ItemParser.parseForItem((net.minecraft.core.HolderLookup)this.items, (com.mojang.brigadier.StringReader)reader);
-        return new net.minecraft.commands.arguments.item.ItemInput(itemResult.item(), itemResult.nbt());
+    public net.minecraft.commands.arguments.item.ItemInput parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        net.minecraft.commands.arguments.item.ItemParser$ItemResult $$1 = net.minecraft.commands.arguments.item.ItemParser.parseForItem((net.minecraft.core.HolderLookup)this.items, (com.mojang.brigadier.StringReader)$$0);
+        return new net.minecraft.commands.arguments.item.ItemInput($$1.item(), $$1.nbt());
     }
 
-    public static <S> net.minecraft.commands.arguments.item.ItemInput getItem(com.mojang.brigadier.context.CommandContext<S> context, java.lang.String name) {
-        return (net.minecraft.commands.arguments.item.ItemInput)context.getArgument(name, net.minecraft.commands.arguments.item.ItemInput.class);
+    public static <S> net.minecraft.commands.arguments.item.ItemInput getItem(com.mojang.brigadier.context.CommandContext<S> $$0, java.lang.String $$1) {
+        return (net.minecraft.commands.arguments.item.ItemInput)$$0.getArgument($$1, net.minecraft.commands.arguments.item.ItemInput.class);
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        return net.minecraft.commands.arguments.item.ItemParser.fillSuggestions((net.minecraft.core.HolderLookup)this.items, (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder, (boolean)false);
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        return net.minecraft.commands.arguments.item.ItemParser.fillSuggestions((net.minecraft.core.HolderLookup)this.items, (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1, (boolean)false);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {