--- net/minecraft/client/server/IntegratedPlayerList.java
+++ net/minecraft/client/server/IntegratedPlayerList.java
@@ -8,23 +8,23 @@
  {
     public net.minecraft.nbt.CompoundTag playerData;
 
-    public net.minecraft.client.server.IntegratedPlayerList(net.minecraft.client.server.IntegratedServer $$0, net.minecraft.core.RegistryAccess$Frozen $$1, net.minecraft.world.level.storage.PlayerDataStorage $$2) {
+    public net.minecraft.client.server.IntegratedPlayerList(net.minecraft.client.server.IntegratedServer $$0, net.minecraft.core.LayeredRegistryAccess<net.minecraft.server.RegistryLayer> $$1, net.minecraft.world.level.storage.PlayerDataStorage $$2) {
         super((net.minecraft.server.MinecraftServer)$$0, $$1, $$2, 8);
         this.setViewDistance(10);
     }
 
-    public void save(net.minecraft.server.level.ServerPlayer player) {
-        if (this.getServer().isSingleplayerOwner(player.getGameProfile())) {
-            this.playerData = player.saveWithoutId(new net.minecraft.nbt.CompoundTag());
+    public void save(net.minecraft.server.level.ServerPlayer $$0) {
+        if (this.getServer().isSingleplayerOwner($$0.getGameProfile())) {
+            this.playerData = $$0.saveWithoutId(new net.minecraft.nbt.CompoundTag());
         }
-        super.save(player);
+        super.save($$0);
     }
 
-    public net.minecraft.network.chat.Component canPlayerLogin(java.net.SocketAddress address, com.mojang.authlib.GameProfile profile) {
-        if (this.getServer().isSingleplayerOwner(profile) && this.getPlayerByName(profile.getName()) != null) {
+    public net.minecraft.network.chat.Component canPlayerLogin(java.net.SocketAddress $$0, com.mojang.authlib.GameProfile $$1) {
+        if (this.getServer().isSingleplayerOwner($$1) && this.getPlayerByName($$1.getName()) != null) {
             return net.minecraft.network.chat.Component.translatable((java.lang.String)"multiplayer.disconnect.name_taken");
         }
-        return super.canPlayerLogin(address, profile);
+        return super.canPlayerLogin($$0, $$1);
     }
 
     public net.minecraft.client.server.IntegratedServer getServer() {