--- net/minecraft/client/server/LanServerPinger.java
+++ net/minecraft/client/server/LanServerPinger.java
@@ -16,11 +16,11 @@
     public boolean isRunning;
     public java.lang.String serverAddress;
 
-    public net.minecraft.client.server.LanServerPinger(java.lang.String motd, java.lang.String serverAddress) throws java.io.IOException {
+    public net.minecraft.client.server.LanServerPinger(java.lang.String $$0, java.lang.String $$1) throws java.io.IOException {
         super("LanServerPinger #" + UNIQUE_THREAD_ID.incrementAndGet());
         this.isRunning = true;
-        this.motd = motd;
-        this.serverAddress = serverAddress;
+        this.motd = $$0;
+        this.serverAddress = $$1;
         this.setDaemon(true);
         this.setUncaughtExceptionHandler((java.lang.Thread$UncaughtExceptionHandler)new net.minecraft.DefaultUncaughtExceptionHandler(LOGGER));
         this.socket = new java.net.DatagramSocket();
@@ -51,40 +51,40 @@
         this.isRunning = false;
     }
 
-    public static java.lang.String createPingString(java.lang.String motdMessage, java.lang.String adMessage) {
-        return "[MOTD]" + motdMessage + "[/MOTD][AD]" + adMessage + "[/AD]";
+    public static java.lang.String createPingString(java.lang.String $$0, java.lang.String $$1) {
+        return "[MOTD]" + $$0 + "[/MOTD][AD]" + $$1 + "[/AD]";
     }
 
-    public static java.lang.String parseMotd(java.lang.String pingResponse) {
-        int n = pingResponse.indexOf("[MOTD]");
-        if (n < 0) {
+    public static java.lang.String parseMotd(java.lang.String $$0) {
+        int $$1 = $$0.indexOf("[MOTD]");
+        if ($$1 < 0) {
             return "missing no";
         }
-        int n2 = pingResponse.indexOf("[/MOTD]", n + "[MOTD]".length());
-        if (n2 < n) {
+        int $$2 = $$0.indexOf("[/MOTD]", $$1 + "[MOTD]".length());
+        if ($$2 < $$1) {
             return "missing no";
         }
-        return pingResponse.substring(n + "[MOTD]".length(), n2);
+        return $$0.substring($$1 + "[MOTD]".length(), $$2);
     }
 
-    public static java.lang.String parseAddress(java.lang.String pingResponse) {
-        int n = pingResponse.indexOf("[/MOTD]");
-        if (n < 0) {
+    public static java.lang.String parseAddress(java.lang.String $$0) {
+        int $$1 = $$0.indexOf("[/MOTD]");
+        if ($$1 < 0) {
             return null;
         }
-        int n2 = pingResponse.indexOf("[/MOTD]", n + "[/MOTD]".length());
-        if (n2 >= 0) {
+        int $$2 = $$0.indexOf("[/MOTD]", $$1 + "[/MOTD]".length());
+        if ($$2 >= 0) {
             return null;
         }
-        int n3 = pingResponse.indexOf("[AD]", n + "[/MOTD]".length());
-        if (n3 < 0) {
+        int $$3 = $$0.indexOf("[AD]", $$1 + "[/MOTD]".length());
+        if ($$3 < 0) {
             return null;
         }
-        int n4 = pingResponse.indexOf("[/AD]", n3 + "[AD]".length());
-        if (n4 < n3) {
+        int $$4 = $$0.indexOf("[/AD]", $$3 + "[AD]".length());
+        if ($$4 < $$3) {
             return null;
         }
-        return pingResponse.substring(n3 + "[AD]".length(), n4);
+        return $$0.substring($$3 + "[AD]".length(), $$4);
     }
 
     static {