--- net/minecraft/client/server/LanServerDetection$LanServerDetector.java
+++ net/minecraft/client/server/LanServerDetection$LanServerDetector.java
@@ -10,9 +10,9 @@
     public java.net.InetAddress pingGroup;
     public java.net.MulticastSocket socket;
 
-    public net.minecraft.client.server.LanServerDetection$LanServerDetector(net.minecraft.client.server.LanServerDetection$LanServerList serverList) throws java.io.IOException {
+    public net.minecraft.client.server.LanServerDetection$LanServerDetector(net.minecraft.client.server.LanServerDetection$LanServerList $$0) throws java.io.IOException {
         super("LanServerDetector #" + net.minecraft.client.server.LanServerDetection.UNIQUE_THREAD_ID.incrementAndGet());
-        this.serverList = serverList;
+        this.serverList = $$0;
         this.setDaemon(true);
         this.setUncaughtExceptionHandler((java.lang.Thread$UncaughtExceptionHandler)new net.minecraft.DefaultUncaughtExceptionHandler(net.minecraft.client.server.LanServerDetection.LOGGER));
         this.socket = new java.net.MulticastSocket(4445);