--- net/minecraft/world/entity/boss/enderdragon/phases/AbstractDragonPhaseInstance.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/AbstractDragonPhaseInstance.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.entity.boss.enderdragon.EnderDragon dragon;
 
-    public net.minecraft.world.entity.boss.enderdragon.phases.AbstractDragonPhaseInstance(net.minecraft.world.entity.boss.enderdragon.EnderDragon dragon) {
-        this.dragon = dragon;
+    public net.minecraft.world.entity.boss.enderdragon.phases.AbstractDragonPhaseInstance(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0) {
+        this.dragon = $$0;
     }
 
     public boolean isSitting() {
@@ -22,7 +22,7 @@
     public void doServerTick() {
     }
 
-    public void onCrystalDestroyed(net.minecraft.world.entity.boss.enderdragon.EndCrystal crystal, net.minecraft.core.BlockPos pos, net.minecraft.world.damagesource.DamageSource dmgSrc, @javax.annotation.Nullable net.minecraft.world.entity.player.Player plyr) {
+    public void onCrystalDestroyed(net.minecraft.world.entity.boss.enderdragon.EndCrystal $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.damagesource.DamageSource $$2, @javax.annotation.Nullable net.minecraft.world.entity.player.Player $$3) {
     }
 
     public void begin() {
@@ -40,8 +40,8 @@
         return null;
     }
 
-    public float onHurt(net.minecraft.world.damagesource.DamageSource damageSource, float amount) {
-        return amount;
+    public float onHurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
+        return $$1;
     }
 
     public float getTurnSpeed() {