--- net/minecraft/world/entity/boss/enderdragon/phases/DragonChargePlayerPhase.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/DragonChargePlayerPhase.java
@@ -37,8 +37,8 @@
         this.timeSinceCharge = 0;
     }
 
-    public void setTarget(net.minecraft.world.phys.Vec3 targetLocation) {
-        this.targetLocation = targetLocation;
+    public void setTarget(net.minecraft.world.phys.Vec3 $$0) {
+        this.targetLocation = $$0;
     }
 
     public float getFlySpeed() {