--- net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingScanningPhase.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/DragonSittingScanningPhase.java
@@ -16,7 +16,7 @@
 
     public net.minecraft.world.entity.boss.enderdragon.phases.DragonSittingScanningPhase(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0) {
         super($$0);
-        this.scanTargeting = net.minecraft.world.entity.ai.targeting.TargetingConditions.forCombat().range(20.0).selector(entity -> java.lang.Math.abs((double)(entity.getY() - $$0.getY())) <= 10.0);
+        this.scanTargeting = net.minecraft.world.entity.ai.targeting.TargetingConditions.forCombat().range(20.0).selector($$1 -> java.lang.Math.abs((double)($$1.getY() - $$0.getY())) <= 10.0);
     }
 
     public void doServerTick() {