--- net/minecraft/world/entity/boss/enderdragon/phases/DragonHoldingPatternPhase.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/DragonHoldingPatternPhase.java
@@ -81,9 +81,9 @@
         this.navigateToNextPathNode();
     }
 
-    public void strafePlayer(net.minecraft.world.entity.player.Player player) {
+    public void strafePlayer(net.minecraft.world.entity.player.Player $$0) {
         this.dragon.getPhaseManager().setPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase.STRAFE_PLAYER);
-        ((net.minecraft.world.entity.boss.enderdragon.phases.DragonStrafePlayerPhase)this.dragon.getPhaseManager().getPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase.STRAFE_PLAYER)).setTarget((net.minecraft.world.entity.LivingEntity)player);
+        ((net.minecraft.world.entity.boss.enderdragon.phases.DragonStrafePlayerPhase)this.dragon.getPhaseManager().getPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase.STRAFE_PLAYER)).setTarget((net.minecraft.world.entity.LivingEntity)$$0);
     }
 
     public void navigateToNextPathNode() {
@@ -99,9 +99,9 @@
         }
     }
 
-    public void onCrystalDestroyed(net.minecraft.world.entity.boss.enderdragon.EndCrystal crystal, net.minecraft.core.BlockPos pos, net.minecraft.world.damagesource.DamageSource dmgSrc, @javax.annotation.Nullable net.minecraft.world.entity.player.Player plyr) {
-        if (plyr != null && this.dragon.canAttack((net.minecraft.world.entity.LivingEntity)plyr)) {
-            this.strafePlayer(plyr);
+    public void onCrystalDestroyed(net.minecraft.world.entity.boss.enderdragon.EndCrystal $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.damagesource.DamageSource $$2, @javax.annotation.Nullable net.minecraft.world.entity.player.Player $$3) {
+        if ($$3 != null && this.dragon.canAttack((net.minecraft.world.entity.LivingEntity)$$3)) {
+            this.strafePlayer($$3);
         }
     }
 