--- net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhaseManager.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhaseManager.java
@@ -11,24 +11,24 @@
     @javax.annotation.Nullable
     public net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance currentPhase;
 
-    public net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhaseManager(net.minecraft.world.entity.boss.enderdragon.EnderDragon dragon) {
+    public net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhaseManager(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0) {
         this.phases = new net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance[net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase.getCount()];
-        this.dragon = dragon;
+        this.dragon = $$0;
         this.setPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase.HOVERING);
     }
 
-    public void setPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<?> phase) {
-        if (this.currentPhase != null && phase == this.currentPhase.getPhase()) {
+    public void setPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<?> $$0) {
+        if (this.currentPhase != null && $$0 == this.currentPhase.getPhase()) {
             return;
         }
         if (this.currentPhase != null) {
             this.currentPhase.end();
         }
-        this.currentPhase = this.getPhase(phase);
+        this.currentPhase = this.getPhase($$0);
         if (!this.dragon.level.isClientSide) {
-            this.dragon.getEntityData().set(net.minecraft.world.entity.boss.enderdragon.EnderDragon.DATA_PHASE, (java.lang.Object)phase.getId());
+            this.dragon.getEntityData().set(net.minecraft.world.entity.boss.enderdragon.EnderDragon.DATA_PHASE, (java.lang.Object)$$0.getId());
         }
-        LOGGER.debug("Dragon is now in phase {} on the {}", phase, (java.lang.Object)(this.dragon.level.isClientSide ? "client" : "server"));
+        LOGGER.debug("Dragon is now in phase {} on the {}", $$0, (java.lang.Object)(this.dragon.level.isClientSide ? "client" : "server"));
         this.currentPhase.begin();
     }
 
@@ -36,12 +36,12 @@
         return this.currentPhase;
     }
 
-    public <T extends net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance> T getPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<T> phase) {
-        int n = phase.getId();
-        if (this.phases[n] == null) {
-            this.phases[n] = phase.createInstance(this.dragon);
+    public <T extends net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance> T getPhase(net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<T> $$0) {
+        int $$1 = $$0.getId();
+        if (this.phases[$$1] == null) {
+            this.phases[$$1] = $$0.createInstance(this.dragon);
         }
-        return (T)this.phases[n];
+        return (T)this.phases[$$1];
     }
 
     static {