--- net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhase.java
+++ net/minecraft/world/entity/boss/enderdragon/phases/EnderDragonPhase.java
@@ -21,19 +21,19 @@
     public int id;
     public java.lang.String name;
 
-    public net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase(int id, java.lang.Class<? extends net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance> instanceClass, java.lang.String name) {
-        this.id = id;
-        this.instanceClass = instanceClass;
-        this.name = name;
+    public net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase(int $$0, java.lang.Class<? extends net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance> $$1, java.lang.String $$2) {
+        this.id = $$0;
+        this.instanceClass = $$1;
+        this.name = $$2;
     }
 
-    public net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance createInstance(net.minecraft.world.entity.boss.enderdragon.EnderDragon dragon) {
+    public net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance createInstance(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0) {
         try {
-            java.lang.reflect.Constructor constructor = this.getConstructor();
-            return (net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance)constructor.newInstance(new java.lang.Object[]{dragon});
+            java.lang.reflect.Constructor $$1 = this.getConstructor();
+            return (net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance)$$1.newInstance(new java.lang.Object[]{$$0});
         }
-        catch (java.lang.Exception exception) {
-            throw new java.lang.Error((java.lang.Throwable)exception);
+        catch (java.lang.Exception $$2) {
+            throw new java.lang.Error((java.lang.Throwable)$$2);
         }
     }
 
@@ -49,22 +49,22 @@
         return this.name + " (#" + this.id + ")";
     }
 
-    public static net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<?> getById(int id) {
-        if (id < 0 || id >= phases.length) {
+    public static net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<?> getById(int $$0) {
+        if ($$0 < 0 || $$0 >= phases.length) {
             return HOLDING_PATTERN;
         }
-        return phases[id];
+        return phases[$$0];
     }
 
     public static int getCount() {
         return phases.length;
     }
 
-    public static <T extends net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance> net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<T> create(java.lang.Class<T> phase, java.lang.String name) {
-        net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase enderDragonPhase = new net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase(phases.length, phase, name);
+    public static <T extends net.minecraft.world.entity.boss.enderdragon.phases.DragonPhaseInstance> net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase<T> create(java.lang.Class<T> $$0, java.lang.String $$1) {
+        net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase $$2 = new net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase(phases.length, $$0, $$1);
         phases = (net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase[])java.util.Arrays.copyOf((java.lang.Object[])phases, (int)(phases.length + 1));
-        net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase.phases[enderDragonPhase.getId()] = enderDragonPhase;
-        return enderDragonPhase;
+        net.minecraft.world.entity.boss.enderdragon.phases.EnderDragonPhase.phases[$$2.getId()] = $$2;
+        return $$2;
     }
 
     static {