--- net/minecraft/world/entity/boss/EnderDragonPart.java
+++ net/minecraft/world/entity/boss/EnderDragonPart.java
@@ -10,43 +10,43 @@
     public java.lang.String name;
     public net.minecraft.world.entity.EntityDimensions size;
 
-    public net.minecraft.world.entity.boss.EnderDragonPart(net.minecraft.world.entity.boss.enderdragon.EnderDragon parentMob, java.lang.String name, float width, float height) {
-        super(parentMob.getType(), parentMob.level);
-        this.size = net.minecraft.world.entity.EntityDimensions.scalable((float)width, (float)height);
+    public net.minecraft.world.entity.boss.EnderDragonPart(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0, java.lang.String $$1, float $$2, float $$3) {
+        super($$0.getType(), $$0.level);
+        this.size = net.minecraft.world.entity.EntityDimensions.scalable((float)$$2, (float)$$3);
         this.refreshDimensions();
-        this.parentMob = parentMob;
-        this.name = name;
+        this.parentMob = $$0;
+        this.name = $$1;
     }
 
     public void defineSynchedData() {
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
     }
 
     public boolean isPickable() {
         return true;
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
-        if (this.isInvulnerableTo(source)) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
+        if (this.isInvulnerableTo($$0)) {
             return false;
         }
-        return this.parentMob.hurt(this, source, amount);
+        return this.parentMob.hurt(this, $$0, $$1);
     }
 
-    public boolean is(net.minecraft.world.entity.Entity entity) {
-        return this == entity || this.parentMob == entity;
+    public boolean is(net.minecraft.world.entity.Entity $$0) {
+        return this == $$0 || this.parentMob == $$0;
     }
 
-    public net.minecraft.network.protocol.Packet<?> getAddEntityPacket() {
+    public net.minecraft.network.protocol.Packet<net.minecraft.network.protocol.game.ClientGamePacketListener> getAddEntityPacket() {
         throw new java.lang.UnsupportedOperationException();
     }
 
-    public net.minecraft.world.entity.EntityDimensions getDimensions(net.minecraft.world.entity.Pose pose) {
+    public net.minecraft.world.entity.EntityDimensions getDimensions(net.minecraft.world.entity.Pose $$0) {
         return this.size;
     }
 