--- net/minecraft/world/entity/monster/piglin/StartAdmiringItemIfSeen.java
+++ net/minecraft/world/entity/monster/piglin/StartAdmiringItemIfSeen.java
@@ -8,18 +8,18 @@
  {
     public int admireDuration;
 
-    public net.minecraft.world.entity.monster.piglin.StartAdmiringItemIfSeen(int admireDuration) {
+    public net.minecraft.world.entity.monster.piglin.StartAdmiringItemIfSeen(int $$0) {
         super((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_PRESENT, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_ITEM, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_ABSENT, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_DISABLED, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_ABSENT, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.DISABLE_WALK_TO_ADMIRE_ITEM, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_ABSENT));
-        this.admireDuration = admireDuration;
+        this.admireDuration = $$0;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        net.minecraft.world.entity.item.ItemEntity itemEntity = (net.minecraft.world.entity.item.ItemEntity)owner.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM).get();
-        return net.minecraft.world.entity.monster.piglin.PiglinAi.isLovedItem((net.minecraft.world.item.ItemStack)itemEntity.getItem());
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        net.minecraft.world.entity.item.ItemEntity $$2 = (net.minecraft.world.entity.item.ItemEntity)$$1.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM).get();
+        return net.minecraft.world.entity.monster.piglin.PiglinAi.isLovedItem((net.minecraft.world.item.ItemStack)$$2.getItem());
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        entity.getBrain().setMemoryWithExpiry(net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_ITEM, (java.lang.Object)true, (long)this.admireDuration);
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        $$1.getBrain().setMemoryWithExpiry(net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_ITEM, (java.lang.Object)true, (long)this.admireDuration);
     }
 }
 