--- net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
+++ net/minecraft/world/entity/monster/piglin/StopHoldingItemIfNoLongerAdmiring.java
@@ -10,12 +10,12 @@
         super((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_ITEM, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_ABSENT));
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        return !owner.getOffhandItem().isEmpty() && !owner.getOffhandItem().is(net.minecraft.world.item.Items.SHIELD);
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        return !$$1.getOffhandItem().isEmpty() && !$$1.getOffhandItem().is(net.minecraft.world.item.Items.SHIELD);
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        net.minecraft.world.entity.monster.piglin.PiglinAi.stopHoldingOffHandItem(entity, (boolean)true);
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        net.minecraft.world.entity.monster.piglin.PiglinAi.stopHoldingOffHandItem($$1, (boolean)true);
     }
 }
 