--- net/minecraft/world/entity/monster/piglin/StopAdmiringIfItemTooFarAway.java
+++ net/minecraft/world/entity/monster/piglin/StopAdmiringIfItemTooFarAway.java
@@ -8,24 +8,24 @@
  {
     public int maxDistanceToItem;
 
-    public net.minecraft.world.entity.monster.piglin.StopAdmiringIfItemTooFarAway(int maxDistanceToItem) {
+    public net.minecraft.world.entity.monster.piglin.StopAdmiringIfItemTooFarAway(int $$0) {
         super((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_ITEM, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.VALUE_PRESENT, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryStatus.REGISTERED));
-        this.maxDistanceToItem = maxDistanceToItem;
+        this.maxDistanceToItem = $$0;
     }
 
-    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel level, E owner) {
-        if (!owner.getOffhandItem().isEmpty()) {
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, E $$1) {
+        if (!$$1.getOffhandItem().isEmpty()) {
             return false;
         }
-        java.util.Optional optional = owner.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM);
-        if (!optional.isPresent()) {
+        java.util.Optional $$2 = $$1.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_WANTED_ITEM);
+        if (!$$2.isPresent()) {
             return true;
         }
-        return !((net.minecraft.world.entity.item.ItemEntity)optional.get()).closerThan(owner, (double)this.maxDistanceToItem);
+        return !((net.minecraft.world.entity.item.ItemEntity)$$2.get()).closerThan($$1, (double)this.maxDistanceToItem);
     }
 
-    public void start(net.minecraft.server.level.ServerLevel level, E entity, long gameTime) {
-        entity.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_ITEM);
+    public void start(net.minecraft.server.level.ServerLevel $$0, E $$1, long $$2) {
+        $$1.getBrain().eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.ADMIRING_ITEM);
     }
 }
 