--- net/minecraft/core/particles/BlockParticleOption.java
+++ net/minecraft/core/particles/BlockParticleOption.java
@@ -19,8 +19,8 @@
         this.state = $$1;
     }
 
-    public void writeToNetwork(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.world.level.block.Block.BLOCK_STATE_REGISTRY, (java.lang.Object)this.state);
+    public void writeToNetwork(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.world.level.block.Block.BLOCK_STATE_REGISTRY, (java.lang.Object)this.state);
     }
 
     public java.lang.String writeToString() {