--- net/minecraft/core/particles/DustParticleOptions$1.java
+++ net/minecraft/core/particles/DustParticleOptions$1.java
@@ -9,15 +9,15 @@
     public net.minecraft.core.particles.DustParticleOptions$1() {
     }
 
-    public net.minecraft.core.particles.DustParticleOptions fromCommand(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.DustParticleOptions> particleType, com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        com.mojang.math.Vector3f vector3f = net.minecraft.core.particles.DustParticleOptionsBase.readVector3f((com.mojang.brigadier.StringReader)reader);
-        reader.expect(' ');
-        float f = reader.readFloat();
-        return new net.minecraft.core.particles.DustParticleOptions(vector3f, f);
+    public net.minecraft.core.particles.DustParticleOptions fromCommand(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.DustParticleOptions> $$0, com.mojang.brigadier.StringReader $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        com.mojang.math.Vector3f $$2 = net.minecraft.core.particles.DustParticleOptionsBase.readVector3f((com.mojang.brigadier.StringReader)$$1);
+        $$1.expect(' ');
+        float $$3 = $$1.readFloat();
+        return new net.minecraft.core.particles.DustParticleOptions($$2, $$3);
     }
 
-    public net.minecraft.core.particles.DustParticleOptions fromNetwork(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.DustParticleOptions> particleType, net.minecraft.network.FriendlyByteBuf buffer) {
-        return new net.minecraft.core.particles.DustParticleOptions(net.minecraft.core.particles.DustParticleOptionsBase.readVector3f((net.minecraft.network.FriendlyByteBuf)buffer), buffer.readFloat());
+    public net.minecraft.core.particles.DustParticleOptions fromNetwork(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.DustParticleOptions> $$0, net.minecraft.network.FriendlyByteBuf $$1) {
+        return new net.minecraft.core.particles.DustParticleOptions(net.minecraft.core.particles.DustParticleOptionsBase.readVector3f((net.minecraft.network.FriendlyByteBuf)$$1), $$1.readFloat());
     }
 
     public /* synthetic */ net.minecraft.core.particles.ParticleOptions fromNetwork(net.minecraft.core.particles.ParticleType particleType, net.minecraft.network.FriendlyByteBuf friendlyByteBuf) {