--- net/minecraft/core/particles/ItemParticleOption$1.java
+++ net/minecraft/core/particles/ItemParticleOption$1.java
@@ -9,15 +9,15 @@
     public net.minecraft.core.particles.ItemParticleOption$1() {
     }
 
-    public net.minecraft.core.particles.ItemParticleOption fromCommand(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.ItemParticleOption> particleType, com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        reader.expect(' ');
-        net.minecraft.commands.arguments.item.ItemParser$ItemResult itemResult = net.minecraft.commands.arguments.item.ItemParser.parseForItem((net.minecraft.core.HolderLookup)net.minecraft.core.HolderLookup.forRegistry((net.minecraft.core.Registry)net.minecraft.core.Registry.ITEM), (com.mojang.brigadier.StringReader)reader);
-        net.minecraft.world.item.ItemStack itemStack = new net.minecraft.commands.arguments.item.ItemInput(itemResult.item(), itemResult.nbt()).createItemStack(1, false);
-        return new net.minecraft.core.particles.ItemParticleOption(particleType, itemStack);
+    public net.minecraft.core.particles.ItemParticleOption fromCommand(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.ItemParticleOption> $$0, com.mojang.brigadier.StringReader $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        $$1.expect(' ');
+        net.minecraft.commands.arguments.item.ItemParser$ItemResult $$2 = net.minecraft.commands.arguments.item.ItemParser.parseForItem((net.minecraft.core.HolderLookup)net.minecraft.core.HolderLookup.forRegistry((net.minecraft.core.Registry)net.minecraft.core.Registry.ITEM), (com.mojang.brigadier.StringReader)$$1);
+        net.minecraft.world.item.ItemStack $$3 = new net.minecraft.commands.arguments.item.ItemInput($$2.item(), $$2.nbt()).createItemStack(1, false);
+        return new net.minecraft.core.particles.ItemParticleOption($$0, $$3);
     }
 
-    public net.minecraft.core.particles.ItemParticleOption fromNetwork(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.ItemParticleOption> particleType, net.minecraft.network.FriendlyByteBuf buffer) {
-        return new net.minecraft.core.particles.ItemParticleOption(particleType, buffer.readItem());
+    public net.minecraft.core.particles.ItemParticleOption fromNetwork(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.ItemParticleOption> $$0, net.minecraft.network.FriendlyByteBuf $$1) {
+        return new net.minecraft.core.particles.ItemParticleOption($$0, $$1.readItem());
     }
 
     public /* synthetic */ net.minecraft.core.particles.ParticleOptions fromNetwork(net.minecraft.core.particles.ParticleType particleType, net.minecraft.network.FriendlyByteBuf friendlyByteBuf) {