--- net/minecraft/core/particles/ItemParticleOption.java
+++ net/minecraft/core/particles/ItemParticleOption.java
@@ -19,8 +19,8 @@
         this.itemStack = $$1;
     }
 
-    public void writeToNetwork(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeItem(this.itemStack);
+    public void writeToNetwork(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeItem(this.itemStack);
     }
 
     public java.lang.String writeToString() {