--- net/minecraft/core/particles/ParticleTypes.java
+++ net/minecraft/core/particles/ParticleTypes.java
@@ -103,8 +103,8 @@
     public net.minecraft.core.particles.ParticleTypes() {
     }
 
-    public static net.minecraft.core.particles.SimpleParticleType register(java.lang.String key, boolean alwaysShow) {
-        return (net.minecraft.core.particles.SimpleParticleType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.PARTICLE_TYPE, (java.lang.String)key, (java.lang.Object)new net.minecraft.core.particles.SimpleParticleType(alwaysShow));
+    public static net.minecraft.core.particles.SimpleParticleType register(java.lang.String $$0, boolean $$1) {
+        return (net.minecraft.core.particles.SimpleParticleType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.PARTICLE_TYPE, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.core.particles.SimpleParticleType($$1));
     }
 
     public static <T extends net.minecraft.core.particles.ParticleOptions> net.minecraft.core.particles.ParticleType<T> register(java.lang.String $$0, boolean $$1, net.minecraft.core.particles.ParticleOptions$Deserializer<T> $$2, java.util.function.Function<net.minecraft.core.particles.ParticleType<T>, com.mojang.serialization.Codec<T>> $$3) {