--- net/minecraft/core/particles/SimpleParticleType$1.java
+++ net/minecraft/core/particles/SimpleParticleType$1.java
@@ -9,12 +9,12 @@
     public net.minecraft.core.particles.SimpleParticleType$1() {
     }
 
-    public net.minecraft.core.particles.SimpleParticleType fromCommand(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.SimpleParticleType> particleType, com.mojang.brigadier.StringReader reader) {
-        return (net.minecraft.core.particles.SimpleParticleType)particleType;
+    public net.minecraft.core.particles.SimpleParticleType fromCommand(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.SimpleParticleType> $$0, com.mojang.brigadier.StringReader $$1) {
+        return (net.minecraft.core.particles.SimpleParticleType)$$0;
     }
 
-    public net.minecraft.core.particles.SimpleParticleType fromNetwork(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.SimpleParticleType> particleType, net.minecraft.network.FriendlyByteBuf buffer) {
-        return (net.minecraft.core.particles.SimpleParticleType)particleType;
+    public net.minecraft.core.particles.SimpleParticleType fromNetwork(net.minecraft.core.particles.ParticleType<net.minecraft.core.particles.SimpleParticleType> $$0, net.minecraft.network.FriendlyByteBuf $$1) {
+        return (net.minecraft.core.particles.SimpleParticleType)$$0;
     }
 
     public /* synthetic */ net.minecraft.core.particles.ParticleOptions fromNetwork(net.minecraft.core.particles.ParticleType particleType, net.minecraft.network.FriendlyByteBuf friendlyByteBuf) {