--- net/minecraft/data/models/model/DelegatedModel.java
+++ net/minecraft/data/models/model/DelegatedModel.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.resources.ResourceLocation parent;
 
-    public net.minecraft.data.models.model.DelegatedModel(net.minecraft.resources.ResourceLocation parent) {
-        this.parent = parent;
+    public net.minecraft.data.models.model.DelegatedModel(net.minecraft.resources.ResourceLocation $$0) {
+        this.parent = $$0;
     }
 
     public com.google.gson.JsonElement get() {