--- net/minecraft/data/models/model/TexturedModel.java
+++ net/minecraft/data/models/model/TexturedModel.java
@@ -29,9 +29,9 @@
     public net.minecraft.data.models.model.TextureMapping mapping;
     public net.minecraft.data.models.model.ModelTemplate template;
 
-    public net.minecraft.data.models.model.TexturedModel(net.minecraft.data.models.model.TextureMapping mapping, net.minecraft.data.models.model.ModelTemplate template) {
-        this.mapping = mapping;
-        this.template = template;
+    public net.minecraft.data.models.model.TexturedModel(net.minecraft.data.models.model.TextureMapping $$0, net.minecraft.data.models.model.ModelTemplate $$1) {
+        this.mapping = $$0;
+        this.template = $$1;
     }
 
     public net.minecraft.data.models.model.ModelTemplate getTemplate() {
@@ -42,25 +42,25 @@
         return this.mapping;
     }
 
-    public net.minecraft.data.models.model.TexturedModel updateTextures(java.util.function.Consumer<net.minecraft.data.models.model.TextureMapping> textureMappingConsumer) {
-        textureMappingConsumer.accept((java.lang.Object)this.mapping);
+    public net.minecraft.data.models.model.TexturedModel updateTextures(java.util.function.Consumer<net.minecraft.data.models.model.TextureMapping> $$0) {
+        $$0.accept((java.lang.Object)this.mapping);
         return this;
     }
 
-    public net.minecraft.resources.ResourceLocation create(net.minecraft.world.level.block.Block modelBlock, java.util.function.BiConsumer<net.minecraft.resources.ResourceLocation, java.util.function.Supplier<com.google.gson.JsonElement>> modelOutput) {
-        return this.template.create(modelBlock, this.mapping, modelOutput);
+    public net.minecraft.resources.ResourceLocation create(net.minecraft.world.level.block.Block $$0, java.util.function.BiConsumer<net.minecraft.resources.ResourceLocation, java.util.function.Supplier<com.google.gson.JsonElement>> $$1) {
+        return this.template.create($$0, this.mapping, $$1);
     }
 
-    public net.minecraft.resources.ResourceLocation createWithSuffix(net.minecraft.world.level.block.Block modelBlock, java.lang.String modelLocationSuffix, java.util.function.BiConsumer<net.minecraft.resources.ResourceLocation, java.util.function.Supplier<com.google.gson.JsonElement>> modelOutput) {
-        return this.template.createWithSuffix(modelBlock, modelLocationSuffix, this.mapping, modelOutput);
+    public net.minecraft.resources.ResourceLocation createWithSuffix(net.minecraft.world.level.block.Block $$0, java.lang.String $$1, java.util.function.BiConsumer<net.minecraft.resources.ResourceLocation, java.util.function.Supplier<com.google.gson.JsonElement>> $$2) {
+        return this.template.createWithSuffix($$0, $$1, this.mapping, $$2);
     }
 
-    public static net.minecraft.data.models.model.TexturedModel$Provider createDefault(java.util.function.Function<net.minecraft.world.level.block.Block, net.minecraft.data.models.model.TextureMapping> blockToTextureMapping, net.minecraft.data.models.model.ModelTemplate modelTemplate) {
-        return $$2 -> new net.minecraft.data.models.model.TexturedModel((net.minecraft.data.models.model.TextureMapping)blockToTextureMapping.apply((java.lang.Object)$$2), modelTemplate);
+    public static net.minecraft.data.models.model.TexturedModel$Provider createDefault(java.util.function.Function<net.minecraft.world.level.block.Block, net.minecraft.data.models.model.TextureMapping> $$0, net.minecraft.data.models.model.ModelTemplate $$1) {
+        return $$2 -> new net.minecraft.data.models.model.TexturedModel((net.minecraft.data.models.model.TextureMapping)$$0.apply((java.lang.Object)$$2), $$1);
     }
 
-    public static net.minecraft.data.models.model.TexturedModel createAllSame(net.minecraft.resources.ResourceLocation allTextureLocation) {
-        return new net.minecraft.data.models.model.TexturedModel(net.minecraft.data.models.model.TextureMapping.cube((net.minecraft.resources.ResourceLocation)allTextureLocation), net.minecraft.data.models.model.ModelTemplates.CUBE_ALL);
+    public static net.minecraft.data.models.model.TexturedModel createAllSame(net.minecraft.resources.ResourceLocation $$0) {
+        return new net.minecraft.data.models.model.TexturedModel(net.minecraft.data.models.model.TextureMapping.cube((net.minecraft.resources.ResourceLocation)$$0), net.minecraft.data.models.model.ModelTemplates.CUBE_ALL);
     }
 
     static {