--- net/minecraft/data/models/model/TextureSlot.java
+++ net/minecraft/data/models/model/TextureSlot.java
@@ -48,17 +48,17 @@
     @javax.annotation.Nullable
     public net.minecraft.data.models.model.TextureSlot parent;
 
-    public static net.minecraft.data.models.model.TextureSlot create(java.lang.String id) {
-        return new net.minecraft.data.models.model.TextureSlot(id, null);
+    public static net.minecraft.data.models.model.TextureSlot create(java.lang.String $$0) {
+        return new net.minecraft.data.models.model.TextureSlot($$0, null);
     }
 
-    public static net.minecraft.data.models.model.TextureSlot create(java.lang.String id, net.minecraft.data.models.model.TextureSlot parent) {
-        return new net.minecraft.data.models.model.TextureSlot(id, parent);
+    public static net.minecraft.data.models.model.TextureSlot create(java.lang.String $$0, net.minecraft.data.models.model.TextureSlot $$1) {
+        return new net.minecraft.data.models.model.TextureSlot($$0, $$1);
     }
 
-    public net.minecraft.data.models.model.TextureSlot(java.lang.String id, @javax.annotation.Nullable net.minecraft.data.models.model.TextureSlot parent) {
-        this.id = id;
-        this.parent = parent;
+    public net.minecraft.data.models.model.TextureSlot(java.lang.String $$0, @javax.annotation.Nullable net.minecraft.data.models.model.TextureSlot $$1) {
+        this.id = $$0;
+        this.parent = $$1;
     }
 
     public java.lang.String getId() {