--- net/minecraft/server/packs/metadata/pack/PackMetadataSectionSerializer.java
+++ net/minecraft/server/packs/metadata/pack/PackMetadataSectionSerializer.java
@@ -4,20 +4,27 @@
 package net.minecraft.server.packs.metadata.pack;
 
 public class net.minecraft.server.packs.metadata.pack.PackMetadataSectionSerializer
-implements net.minecraft.server.packs.metadata.MetadataSectionSerializer<net.minecraft.server.packs.metadata.pack.PackMetadataSection>
+implements net.minecraft.server.packs.metadata.MetadataSectionType<net.minecraft.server.packs.metadata.pack.PackMetadataSection>
  {
     public net.minecraft.server.packs.metadata.pack.PackMetadataSectionSerializer() {
     }
 
-    public net.minecraft.server.packs.metadata.pack.PackMetadataSection fromJson(com.google.gson.JsonObject json) {
-        net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)json.get("description"));
-        if (mutableComponent == null) {
+    public net.minecraft.server.packs.metadata.pack.PackMetadataSection fromJson(com.google.gson.JsonObject $$0) {
+        net.minecraft.network.chat.MutableComponent $$1 = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)$$0.get("description"));
+        if ($$1 == null) {
             throw new com.google.gson.JsonParseException("Invalid/missing description!");
         }
-        int n = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)json, (java.lang.String)"pack_format");
-        return new net.minecraft.server.packs.metadata.pack.PackMetadataSection((net.minecraft.network.chat.Component)mutableComponent, n);
+        int $$2 = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)$$0, (java.lang.String)"pack_format");
+        return new net.minecraft.server.packs.metadata.pack.PackMetadataSection((net.minecraft.network.chat.Component)$$1, $$2);
     }
 
+    public com.google.gson.JsonObject toJson(net.minecraft.server.packs.metadata.pack.PackMetadataSection $$0) {
+        com.google.gson.JsonObject $$1 = new com.google.gson.JsonObject();
+        $$1.add("description", net.minecraft.network.chat.Component$Serializer.toJsonTree((net.minecraft.network.chat.Component)$$0.getDescription()));
+        $$1.addProperty("pack_format", (java.lang.Number)java.lang.Integer.valueOf((int)$$0.getPackFormat()));
+        return $$1;
+    }
+
     public java.lang.String getMetadataSectionName() {
         return "pack";
     }