--- net/minecraft/server/packs/metadata/pack/PackMetadataSection.java
+++ net/minecraft/server/packs/metadata/pack/PackMetadataSection.java
@@ -5,13 +5,13 @@
 
 public class net.minecraft.server.packs.metadata.pack.PackMetadataSection
  {
-    public static net.minecraft.server.packs.metadata.pack.PackMetadataSectionSerializer SERIALIZER;
+    public static net.minecraft.server.packs.metadata.MetadataSectionType<net.minecraft.server.packs.metadata.pack.PackMetadataSection> TYPE;
     public net.minecraft.network.chat.Component description;
     public int packFormat;
 
-    public net.minecraft.server.packs.metadata.pack.PackMetadataSection(net.minecraft.network.chat.Component description, int packFormat) {
-        this.description = description;
-        this.packFormat = packFormat;
+    public net.minecraft.server.packs.metadata.pack.PackMetadataSection(net.minecraft.network.chat.Component $$0, int $$1) {
+        this.description = $$0;
+        this.packFormat = $$1;
     }
 
     public net.minecraft.network.chat.Component getDescription() {
@@ -23,7 +23,7 @@
     }
 
     static {
-        SERIALIZER = new net.minecraft.server.packs.metadata.pack.PackMetadataSectionSerializer();
+        TYPE = new net.minecraft.server.packs.metadata.pack.PackMetadataSectionSerializer();
     }
 }
 