--- net/minecraft/client/AmbientOcclusionStatus.java
+++ net/minecraft/client/AmbientOcclusionStatus.java
@@ -37,8 +37,8 @@
         return this.key;
     }
 
-    public static net.minecraft.client.AmbientOcclusionStatus byId(int value) {
-        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)value, (int)BY_ID.length)];
+    public static net.minecraft.client.AmbientOcclusionStatus byId(int $$0) {
+        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)$$0, (int)BY_ID.length)];
     }
 
     public static /* synthetic */ net.minecraft.client.AmbientOcclusionStatus[] lambda$static$0(int $$0) {