--- net/minecraft/client/AttackIndicatorStatus.java
+++ net/minecraft/client/AttackIndicatorStatus.java
@@ -37,8 +37,8 @@
         return this.key;
     }
 
-    public static net.minecraft.client.AttackIndicatorStatus byId(int id) {
-        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)id, (int)BY_ID.length)];
+    public static net.minecraft.client.AttackIndicatorStatus byId(int $$0) {
+        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)$$0, (int)BY_ID.length)];
     }
 
     public static /* synthetic */ net.minecraft.client.AttackIndicatorStatus[] lambda$static$0(int $$0) {