--- net/minecraft/client/CameraType.java
+++ net/minecraft/client/CameraType.java
@@ -22,10 +22,10 @@
         return (net.minecraft.client.CameraType)java.lang.Enum.valueOf(net.minecraft.client.CameraType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.client.CameraType(boolean firstPerson, boolean mirrored) {
+    public net.minecraft.client.CameraType(boolean $$0, boolean $$1) {
         super(string, n);
-        this.firstPerson = firstPerson;
-        this.mirrored = mirrored;
+        this.firstPerson = $$0;
+        this.mirrored = $$1;
     }
 
     public boolean isFirstPerson() {