--- net/minecraft/client/ClientRecipeBook$1.java
+++ net/minecraft/client/ClientRecipeBook$1.java
@@ -1,0 +1,58 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client;
+
+public class net.minecraft.client.ClientRecipeBook$1
+ {
+    public static /* synthetic */ int[] $SwitchMap$net$minecraft$world$item$crafting$CraftingBookCategory;
+    public static /* synthetic */ int[] $SwitchMap$net$minecraft$world$item$crafting$CookingBookCategory;
+
+    static {
+        $SwitchMap$net$minecraft$world$item$crafting$CookingBookCategory = new int[net.minecraft.world.item.crafting.CookingBookCategory.values().length];
+        try {
+            net.minecraft.client.ClientRecipeBook$1.$SwitchMap$net$minecraft$world$item$crafting$CookingBookCategory[net.minecraft.world.item.crafting.CookingBookCategory.BLOCKS.ordinal()] = 1;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.client.ClientRecipeBook$1.$SwitchMap$net$minecraft$world$item$crafting$CookingBookCategory[net.minecraft.world.item.crafting.CookingBookCategory.FOOD.ordinal()] = 2;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.client.ClientRecipeBook$1.$SwitchMap$net$minecraft$world$item$crafting$CookingBookCategory[net.minecraft.world.item.crafting.CookingBookCategory.MISC.ordinal()] = 3;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        $SwitchMap$net$minecraft$world$item$crafting$CraftingBookCategory = new int[net.minecraft.world.item.crafting.CraftingBookCategory.values().length];
+        try {
+            net.minecraft.client.ClientRecipeBook$1.$SwitchMap$net$minecraft$world$item$crafting$CraftingBookCategory[net.minecraft.world.item.crafting.CraftingBookCategory.BUILDING.ordinal()] = 1;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.client.ClientRecipeBook$1.$SwitchMap$net$minecraft$world$item$crafting$CraftingBookCategory[net.minecraft.world.item.crafting.CraftingBookCategory.EQUIPMENT.ordinal()] = 2;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.client.ClientRecipeBook$1.$SwitchMap$net$minecraft$world$item$crafting$CraftingBookCategory[net.minecraft.world.item.crafting.CraftingBookCategory.REDSTONE.ordinal()] = 3;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.client.ClientRecipeBook$1.$SwitchMap$net$minecraft$world$item$crafting$CraftingBookCategory[net.minecraft.world.item.crafting.CraftingBookCategory.MISC.ordinal()] = 4;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+    }
+}
+