--- net/minecraft/client/DebugQueryHandler.java
+++ net/minecraft/client/DebugQueryHandler.java
@@ -29,9 +29,9 @@
         return ++this.transactionId;
     }
 
-    public void queryEntityTag(int entId, java.util.function.Consumer<net.minecraft.nbt.CompoundTag> consumer) {
-        int n = this.startTransaction(consumer);
-        this.connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundEntityTagQuery(n, entId));
+    public void queryEntityTag(int $$0, java.util.function.Consumer<net.minecraft.nbt.CompoundTag> $$1) {
+        int $$2 = this.startTransaction($$1);
+        this.connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundEntityTagQuery($$2, $$0));
     }
 
     public void queryBlockEntityTag(net.minecraft.core.BlockPos $$0, java.util.function.Consumer<net.minecraft.nbt.CompoundTag> $$1) {