--- net/minecraft/client/Game.java
+++ net/minecraft/client/Game.java
@@ -11,9 +11,9 @@
     public com.mojang.bridge.launcher.Launcher launcher;
     public com.mojang.bridge.launcher.SessionEventListener listener;
 
-    public net.minecraft.client.Game(net.minecraft.client.Minecraft minecraft) {
+    public net.minecraft.client.Game(net.minecraft.client.Minecraft $$0) {
         this.listener = com.mojang.bridge.launcher.SessionEventListener.NONE;
-        this.minecraft = minecraft;
+        this.minecraft = $$0;
         this.launcher = com.mojang.bridge.Bridge.getLauncher();
         if (this.launcher != null) {
             this.launcher.registerGame((com.mojang.bridge.game.RunningGame)this);
@@ -47,8 +47,8 @@
         return new net.minecraft.client.Game$Metrics((int)$$1, (int)$$2, (int)($$3 / (long)$$0.getLog().length), $$0.getLog().length);
     }
 
-    public void setSessionEventListener(com.mojang.bridge.launcher.SessionEventListener listener) {
-        this.listener = listener;
+    public void setSessionEventListener(com.mojang.bridge.launcher.SessionEventListener $$0) {
+        this.listener = $$0;
     }
 
     public void onStartGameSession() {