--- net/minecraft/client/Game$Metrics.java
+++ net/minecraft/client/Game$Metrics.java
@@ -11,11 +11,11 @@
     public int average;
     public int samples;
 
-    public net.minecraft.client.Game$Metrics(int min, int max, int average, int samples) {
-        this.min = min;
-        this.max = max;
-        this.average = average;
-        this.samples = samples;
+    public net.minecraft.client.Game$Metrics(int $$0, int $$1, int $$2, int $$3) {
+        this.min = $$0;
+        this.max = $$1;
+        this.average = $$2;
+        this.samples = $$3;
     }
 
     public int getMinTime() {