--- net/minecraft/client/GuiMessage.java
+++ net/minecraft/client/GuiMessage.java
@@ -9,27 +9,27 @@
     public int addedTime;
     public net.minecraft.network.chat.Component content;
     @javax.annotation.Nullable
-    public net.minecraft.network.chat.MessageSignature headerSignature;
+    public net.minecraft.network.chat.MessageSignature signature;
     @javax.annotation.Nullable
     public net.minecraft.client.GuiMessageTag tag;
 
     public net.minecraft.client.GuiMessage(int $$0, net.minecraft.network.chat.Component $$1, @javax.annotation.Nullable net.minecraft.network.chat.MessageSignature $$2, @javax.annotation.Nullable net.minecraft.client.GuiMessageTag $$3) {
         this.addedTime = $$0;
         this.content = $$1;
-        this.headerSignature = $$2;
+        this.signature = $$2;
         this.tag = $$3;
     }
 
     public java.lang.String toString() {
-        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.GuiMessage.class, "addedTime;content;headerSignature;tag", "addedTime", "content", "headerSignature", "tag"}, this);
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.GuiMessage.class, "addedTime;content;signature;tag", "addedTime", "content", "signature", "tag"}, this);
     }
 
     public int hashCode() {
-        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.GuiMessage.class, "addedTime;content;headerSignature;tag", "addedTime", "content", "headerSignature", "tag"}, this);
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.GuiMessage.class, "addedTime;content;signature;tag", "addedTime", "content", "signature", "tag"}, this);
     }
 
     public boolean equals(java.lang.Object $$0) {
-        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.GuiMessage.class, "addedTime;content;headerSignature;tag", "addedTime", "content", "headerSignature", "tag"}, this, $$0);
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.GuiMessage.class, "addedTime;content;signature;tag", "addedTime", "content", "signature", "tag"}, this, $$0);
     }
 
     public int addedTime() {
@@ -41,8 +41,8 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.network.chat.MessageSignature headerSignature() {
-        return this.headerSignature;
+    public net.minecraft.network.chat.MessageSignature signature() {
+        return this.signature;
     }
 
     @javax.annotation.Nullable