--- net/minecraft/client/GuiMessageTag.java
+++ net/minecraft/client/GuiMessageTag.java
@@ -13,15 +13,13 @@
     public net.minecraft.network.chat.Component text;
     @javax.annotation.Nullable
     public java.lang.String logTag;
+    public static net.minecraft.network.chat.Component SYSTEM_TEXT;
     public static net.minecraft.network.chat.Component CHAT_NOT_SECURE_TEXT;
     public static net.minecraft.network.chat.Component CHAT_MODIFIED_TEXT;
-    public static net.minecraft.network.chat.Component CHAT_FILTERED_TEXT;
-    public static int SYSTEM_INDICATOR_COLOR = 0xA0A0A0;
-    public static int CHAT_NOT_SECURE_INDICATOR_COLOR = 15224664;
-    public static int CHAT_MODIFIED_INDICATOR_COLOR = 15386724;
+    public static int CHAT_NOT_SECURE_INDICATOR_COLOR = 0xD0D0D0;
+    public static int CHAT_MODIFIED_INDICATOR_COLOR = 0x606060;
     public static net.minecraft.client.GuiMessageTag SYSTEM;
     public static net.minecraft.client.GuiMessageTag CHAT_NOT_SECURE;
-    public static net.minecraft.client.GuiMessageTag CHAT_FILTERED;
     public static net.minecraft.resources.ResourceLocation TEXTURE_LOCATION;
 
     public net.minecraft.client.GuiMessageTag(int $$0, @javax.annotation.Nullable net.minecraft.client.GuiMessageTag$Icon $$1, @javax.annotation.Nullable net.minecraft.network.chat.Component $$2, @javax.annotation.Nullable java.lang.String $$3) {
@@ -40,15 +38,11 @@
     }
 
     public static net.minecraft.client.GuiMessageTag chatModified(java.lang.String $$0) {
-        net.minecraft.network.chat.MutableComponent $$1 = net.minecraft.network.chat.Component.translatable((java.lang.String)"chat.tag.modified.original", (java.lang.Object[])new java.lang.Object[]{$$0});
+        net.minecraft.network.chat.MutableComponent $$1 = net.minecraft.network.chat.Component.literal((java.lang.String)$$0).withStyle(net.minecraft.ChatFormatting.GRAY);
         net.minecraft.network.chat.MutableComponent $$2 = net.minecraft.network.chat.Component.empty().append(CHAT_MODIFIED_TEXT).append(net.minecraft.network.chat.CommonComponents.NEW_LINE).append((net.minecraft.network.chat.Component)$$1);
-        return new net.minecraft.client.GuiMessageTag(15386724, net.minecraft.client.GuiMessageTag$Icon.CHAT_MODIFIED, (net.minecraft.network.chat.Component)$$2, "Modified");
+        return new net.minecraft.client.GuiMessageTag(0x606060, net.minecraft.client.GuiMessageTag$Icon.CHAT_MODIFIED, (net.minecraft.network.chat.Component)$$2, "Modified");
     }
 
-    public static net.minecraft.client.GuiMessageTag chatFiltered() {
-        return CHAT_FILTERED;
-    }
-
     public java.lang.String toString() {
         return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.GuiMessageTag.class, "indicatorColor;icon;text;logTag", "indicatorColor", "icon", "text", "logTag"}, this);
     }
@@ -81,12 +75,11 @@
     }
 
     static {
-        CHAT_NOT_SECURE_TEXT = net.minecraft.network.chat.Component.translatable((java.lang.String)"chat.tag.not_secure").withStyle(net.minecraft.ChatFormatting.UNDERLINE);
-        CHAT_MODIFIED_TEXT = net.minecraft.network.chat.Component.translatable((java.lang.String)"chat.tag.modified").withStyle(net.minecraft.ChatFormatting.UNDERLINE);
-        CHAT_FILTERED_TEXT = net.minecraft.network.chat.Component.translatable((java.lang.String)"chat.tag.filtered").withStyle(net.minecraft.ChatFormatting.UNDERLINE);
-        SYSTEM = new net.minecraft.client.GuiMessageTag(0xA0A0A0, null, null, "System");
-        CHAT_NOT_SECURE = new net.minecraft.client.GuiMessageTag(15224664, net.minecraft.client.GuiMessageTag$Icon.CHAT_NOT_SECURE, CHAT_NOT_SECURE_TEXT, "Not Secure");
-        CHAT_FILTERED = new net.minecraft.client.GuiMessageTag(15386724, net.minecraft.client.GuiMessageTag$Icon.CHAT_MODIFIED, CHAT_FILTERED_TEXT, "Filtered");
+        SYSTEM_TEXT = net.minecraft.network.chat.Component.translatable((java.lang.String)"chat.tag.system");
+        CHAT_NOT_SECURE_TEXT = net.minecraft.network.chat.Component.translatable((java.lang.String)"chat.tag.not_secure");
+        CHAT_MODIFIED_TEXT = net.minecraft.network.chat.Component.translatable((java.lang.String)"chat.tag.modified");
+        SYSTEM = new net.minecraft.client.GuiMessageTag(0xD0D0D0, null, SYSTEM_TEXT, "System");
+        CHAT_NOT_SECURE = new net.minecraft.client.GuiMessageTag(0xD0D0D0, null, CHAT_NOT_SECURE_TEXT, "Not Secure");
         TEXTURE_LOCATION = new net.minecraft.resources.ResourceLocation("textures/gui/chat_tags.png");
     }
 }