--- net/minecraft/client/GuiMessageTag$Icon.java
+++ net/minecraft/client/GuiMessageTag$Icon.java
@@ -6,7 +6,6 @@
 public class net.minecraft.client.GuiMessageTag$Icon
 extends java.lang.Enum<net.minecraft.client.GuiMessageTag$Icon>
  {
-    public static /* enum */ net.minecraft.client.GuiMessageTag$Icon CHAT_NOT_SECURE;
     public static /* enum */ net.minecraft.client.GuiMessageTag$Icon CHAT_MODIFIED;
     public int u;
     public int v;
@@ -36,12 +35,11 @@
     }
 
     public static /* synthetic */ net.minecraft.client.GuiMessageTag$Icon[] $values() {
-        return new net.minecraft.client.GuiMessageTag$Icon[]{CHAT_NOT_SECURE, CHAT_MODIFIED};
+        return new net.minecraft.client.GuiMessageTag$Icon[]{CHAT_MODIFIED};
     }
 
     static {
-        CHAT_NOT_SECURE = new net.minecraft.client.GuiMessageTag$Icon("CHAT_NOT_SECURE", 0, 0, 0, 9, 9);
-        CHAT_MODIFIED = new net.minecraft.client.GuiMessageTag$Icon("CHAT_MODIFIED", 1, 9, 0, 9, 9);
+        CHAT_MODIFIED = new net.minecraft.client.GuiMessageTag$Icon("CHAT_MODIFIED", 0, 0, 0, 9, 9);
         $VALUES = net.minecraft.client.GuiMessageTag$Icon.$values();
     }
 }