--- net/minecraft/client/HotbarManager.java
+++ net/minecraft/client/HotbarManager.java
@@ -54,12 +54,12 @@
         }
     }
 
-    public net.minecraft.client.player.inventory.Hotbar get(int index) {
+    public net.minecraft.client.player.inventory.Hotbar get(int $$0) {
         if (!this.loaded) {
             this.load();
             this.loaded = true;
         }
-        return this.hotbars[index];
+        return this.hotbars[$$0];
     }
 
     static {