--- net/minecraft/client/KeyMapping.java
+++ net/minecraft/client/KeyMapping.java
@@ -24,17 +24,17 @@
     public boolean isDown;
     public int clickCount;
 
-    public static void click(com.mojang.blaze3d.platform.InputConstants$Key key) {
-        net.minecraft.client.KeyMapping keyMapping = (net.minecraft.client.KeyMapping)MAP.get((java.lang.Object)key);
-        if (keyMapping != null) {
-            ++keyMapping.clickCount;
+    public static void click(com.mojang.blaze3d.platform.InputConstants$Key $$0) {
+        net.minecraft.client.KeyMapping $$1 = (net.minecraft.client.KeyMapping)MAP.get((java.lang.Object)$$0);
+        if ($$1 != null) {
+            ++$$1.clickCount;
         }
     }
 
-    public static void set(com.mojang.blaze3d.platform.InputConstants$Key key, boolean held) {
-        net.minecraft.client.KeyMapping keyMapping = (net.minecraft.client.KeyMapping)MAP.get((java.lang.Object)key);
-        if (keyMapping != null) {
-            keyMapping.setDown(held);
+    public static void set(com.mojang.blaze3d.platform.InputConstants$Key $$0, boolean $$1) {
+        net.minecraft.client.KeyMapping $$2 = (net.minecraft.client.KeyMapping)MAP.get((java.lang.Object)$$0);
+        if ($$2 != null) {
+            $$2.setDown($$1);
         }
     }
 
@@ -100,8 +100,8 @@
         return this.defaultKey;
     }
 
-    public void setKey(com.mojang.blaze3d.platform.InputConstants$Key key) {
-        this.key = key;
+    public void setKey(com.mojang.blaze3d.platform.InputConstants$Key $$0) {
+        this.key = $$0;
     }
 
     public int compareTo(net.minecraft.client.KeyMapping $$0) {
@@ -111,31 +111,31 @@
         return ((java.lang.Integer)CATEGORY_SORT_ORDER.get((java.lang.Object)this.category)).compareTo((java.lang.Integer)CATEGORY_SORT_ORDER.get((java.lang.Object)$$0.category));
     }
 
-    public static java.util.function.Supplier<net.minecraft.network.chat.Component> createNameSupplier(java.lang.String key) {
-        net.minecraft.client.KeyMapping keyMapping = (net.minecraft.client.KeyMapping)ALL.get((java.lang.Object)key);
-        if (keyMapping == null) {
-            return () -> net.minecraft.network.chat.Component.translatable((java.lang.String)key);
+    public static java.util.function.Supplier<net.minecraft.network.chat.Component> createNameSupplier(java.lang.String $$0) {
+        net.minecraft.client.KeyMapping $$1 = (net.minecraft.client.KeyMapping)ALL.get((java.lang.Object)$$0);
+        if ($$1 == null) {
+            return () -> net.minecraft.network.chat.Component.translatable((java.lang.String)$$0);
         }
-        return () -> keyMapping.getTranslatedKeyMessage();
+        return () -> $$1.getTranslatedKeyMessage();
     }
 
-    public boolean same(net.minecraft.client.KeyMapping binding) {
-        return this.key.equals((java.lang.Object)binding.key);
+    public boolean same(net.minecraft.client.KeyMapping $$0) {
+        return this.key.equals((java.lang.Object)$$0.key);
     }
 
     public boolean isUnbound() {
         return this.key.equals((java.lang.Object)com.mojang.blaze3d.platform.InputConstants.UNKNOWN);
     }
 
-    public boolean matches(int keysym, int scancode) {
-        if (keysym == com.mojang.blaze3d.platform.InputConstants.UNKNOWN.getValue()) {
-            return this.key.getType() == com.mojang.blaze3d.platform.InputConstants$Type.SCANCODE && this.key.getValue() == scancode;
+    public boolean matches(int $$0, int $$1) {
+        if ($$0 == com.mojang.blaze3d.platform.InputConstants.UNKNOWN.getValue()) {
+            return this.key.getType() == com.mojang.blaze3d.platform.InputConstants$Type.SCANCODE && this.key.getValue() == $$1;
         }
-        return this.key.getType() == com.mojang.blaze3d.platform.InputConstants$Type.KEYSYM && this.key.getValue() == keysym;
+        return this.key.getType() == com.mojang.blaze3d.platform.InputConstants$Type.KEYSYM && this.key.getValue() == $$0;
     }
 
-    public boolean matchesMouse(int key) {
-        return this.key.getType() == com.mojang.blaze3d.platform.InputConstants$Type.MOUSE && this.key.getValue() == key;
+    public boolean matchesMouse(int $$0) {
+        return this.key.getType() == com.mojang.blaze3d.platform.InputConstants$Type.MOUSE && this.key.getValue() == $$0;
     }
 
     public net.minecraft.network.chat.Component getTranslatedKeyMessage() {
@@ -150,8 +150,8 @@
         return this.key.getName();
     }
 
-    public void setDown(boolean value) {
-        this.isDown = value;
+    public void setDown(boolean $$0) {
+        this.isDown = $$0;
     }
 
     public /* synthetic */ int compareTo(java.lang.Object object) {