--- net/minecraft/client/NarratorStatus.java
+++ net/minecraft/client/NarratorStatus.java
@@ -37,8 +37,8 @@
         return this.name;
     }
 
-    public static net.minecraft.client.NarratorStatus byId(int id) {
-        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)id, (int)BY_ID.length)];
+    public static net.minecraft.client.NarratorStatus byId(int $$0) {
+        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)$$0, (int)BY_ID.length)];
     }
 
     public boolean shouldNarrateChat() {