--- net/minecraft/client/OptionInstance$AltEnum.java
+++ net/minecraft/client/OptionInstance$AltEnum.java
@@ -53,11 +53,11 @@
         return this.altCondition;
     }
 
-    public net.minecraft.client.OptionInstance$CycleableValueSet$ValueSetter<T> valueSetterVanilla() {
+    public net.minecraft.client.OptionInstance$CycleableValueSet$ValueSetter<T> valueSetter() {
         return this.valueSetter;
     }
 
-    public com.mojang.serialization.Codec<T> codecVanilla() {
+    public com.mojang.serialization.Codec<T> codec() {
         return this.codec;
     }
 }