--- net/minecraft/client/OptionInstance$UnitDouble$1.java
+++ net/minecraft/client/OptionInstance$UnitDouble$1.java
@@ -28,8 +28,8 @@
         return (R)this.val$to.apply(this.this$0.fromSliderValue($$0).doubleValue());
     }
 
-    public com.mojang.serialization.Codec<R> codecVanilla() {
-        return this.this$0.codecVanilla().xmap(arg_0 -> ((java.util.function.DoubleFunction)this.val$to).apply(arg_0), arg_0 -> ((java.util.function.ToDoubleFunction)this.val$from).applyAsDouble(arg_0));
+    public com.mojang.serialization.Codec<R> codec() {
+        return this.this$0.codec().xmap(arg_0 -> ((java.util.function.DoubleFunction)this.val$to).apply(arg_0), arg_0 -> ((java.util.function.ToDoubleFunction)this.val$from).applyAsDouble(arg_0));
     }
 }
 