--- net/minecraft/client/Options$2.java
+++ net/minecraft/client/Options$2.java
@@ -24,7 +24,7 @@
         if ($$22 != null) {
             com.google.gson.stream.JsonReader $$3 = new com.google.gson.stream.JsonReader((java.io.Reader)new java.io.StringReader($$22.isEmpty() ? "\"\"" : $$22));
             com.google.gson.JsonElement $$4 = com.google.gson.JsonParser.parseReader((com.google.gson.stream.JsonReader)$$3);
-            com.mojang.serialization.DataResult $$5 = $$1.codecVanilla().parse((com.mojang.serialization.DynamicOps)com.mojang.serialization.JsonOps.INSTANCE, (java.lang.Object)$$4);
+            com.mojang.serialization.DataResult $$5 = $$1.codec().parse((com.mojang.serialization.DynamicOps)com.mojang.serialization.JsonOps.INSTANCE, (java.lang.Object)$$4);
             $$5.error().ifPresent($$2 -> net.minecraft.client.Options.LOGGER.error("Error parsing option value " + $$22 + " for option " + $$1 + ": " + $$2.message()));
             $$5.result().ifPresent(arg_0 -> $$1.set(arg_0));
         }