--- net/minecraft/client/Options$4.java
+++ net/minecraft/client/Options$4.java
@@ -5,7 +5,6 @@
 
 public class net.minecraft.client.Options$4
  {
-    public static /* synthetic */ int[] $SwitchMap$net$minecraft$client$multiplayer$chat$ChatPreviewStatus;
     public static /* synthetic */ int[] $SwitchMap$net$minecraft$client$PrioritizeChunkUpdates;
     public static /* synthetic */ int[] $SwitchMap$net$minecraft$client$GraphicsStatus;
     public static /* synthetic */ int[] $SwitchMap$net$minecraft$client$CloudStatus;
@@ -68,25 +67,6 @@
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
-        $SwitchMap$net$minecraft$client$multiplayer$chat$ChatPreviewStatus = new int[net.minecraft.client.multiplayer.chat.ChatPreviewStatus.values().length];
-        try {
-            net.minecraft.client.Options$4.$SwitchMap$net$minecraft$client$multiplayer$chat$ChatPreviewStatus[net.minecraft.client.multiplayer.chat.ChatPreviewStatus.OFF.ordinal()] = 1;
-        }
-        catch (java.lang.NoSuchFieldError noSuchFieldError) {
-            // empty catch block
-        }
-        try {
-            net.minecraft.client.Options$4.$SwitchMap$net$minecraft$client$multiplayer$chat$ChatPreviewStatus[net.minecraft.client.multiplayer.chat.ChatPreviewStatus.LIVE.ordinal()] = 2;
-        }
-        catch (java.lang.NoSuchFieldError noSuchFieldError) {
-            // empty catch block
-        }
-        try {
-            net.minecraft.client.Options$4.$SwitchMap$net$minecraft$client$multiplayer$chat$ChatPreviewStatus[net.minecraft.client.multiplayer.chat.ChatPreviewStatus.CONFIRM.ordinal()] = 3;
-        }
-        catch (java.lang.NoSuchFieldError noSuchFieldError) {
-            // empty catch block
-        }
     }
 }
 