--- net/minecraft/client/Options$3.java
+++ net/minecraft/client/Options$3.java
@@ -20,7 +20,7 @@
     }
 
     public <T> void process(java.lang.String $$0, net.minecraft.client.OptionInstance<T> $$12) {
-        com.mojang.serialization.DataResult $$22 = $$12.codecVanilla().encodeStart((com.mojang.serialization.DynamicOps)com.mojang.serialization.JsonOps.INSTANCE, $$12.get());
+        com.mojang.serialization.DataResult $$22 = $$12.codec().encodeStart((com.mojang.serialization.DynamicOps)com.mojang.serialization.JsonOps.INSTANCE, $$12.get());
         $$22.error().ifPresent($$1 -> net.minecraft.client.Options.LOGGER.error("Error saving option " + $$12 + ": " + $$1));
         $$22.result().ifPresent($$2 -> {
             this.writePrefix($$0);