--- net/minecraft/client/OptionInstance$UnitDouble.java
+++ net/minecraft/client/OptionInstance$UnitDouble.java
@@ -38,7 +38,7 @@
         return new /* Unavailable Anonymous Inner Class!! */;
     }
 
-    public com.mojang.serialization.Codec<java.lang.Double> codecVanilla() {
+    public com.mojang.serialization.Codec<java.lang.Double> codec() {
         return com.mojang.serialization.Codec.either((com.mojang.serialization.Codec)com.mojang.serialization.Codec.doubleRange((double)0.0, (double)1.0), (com.mojang.serialization.Codec)com.mojang.serialization.Codec.BOOL).xmap($$02 -> (java.lang.Double)$$02.map($$0 -> $$0, $$0 -> $$0 != false ? 1.0 : 0.0), com.mojang.datafixers.util.Either::left);
     }
 