--- net/minecraft/client/OptionInstance$ClampingLazyMaxIntRange.java
+++ net/minecraft/client/OptionInstance$ClampingLazyMaxIntRange.java
@@ -24,7 +24,7 @@
         return this.maxSupplier.getAsInt();
     }
 
-    public com.mojang.serialization.Codec<java.lang.Integer> codecVanilla() {
+    public com.mojang.serialization.Codec<java.lang.Integer> codec() {
         java.util.function.Function $$02 = $$0 -> {
             int $$1 = this.maxSupplier.getAsInt() + 1;
             if ($$0.compareTo(java.lang.Integer.valueOf((int)this.minInclusive)) >= 0 && $$0.compareTo(java.lang.Integer.valueOf((int)$$1)) <= 0) {