--- net/minecraft/client/RecipeBookCategories.java
+++ net/minecraft/client/RecipeBookCategories.java
@@ -46,21 +46,13 @@
     }
 
     public static java.util.List<net.minecraft.client.RecipeBookCategories> getCategories(net.minecraft.world.inventory.RecipeBookType $$0) {
-        switch (net.minecraft.client.RecipeBookCategories$1.$SwitchMap$net$minecraft$world$inventory$RecipeBookType[$$0.ordinal()]) {
-            case 1: {
-                return CRAFTING_CATEGORIES;
-            }
-            case 2: {
-                return FURNACE_CATEGORIES;
-            }
-            case 3: {
-                return BLAST_FURNACE_CATEGORIES;
-            }
-            case 4: {
-                return SMOKER_CATEGORIES;
-            }
-        }
-        return com.google.common.collect.ImmutableList.of();
+        return switch (net.minecraft.client.RecipeBookCategories$1.$SwitchMap$net$minecraft$world$inventory$RecipeBookType[$$0.ordinal()]) {
+            default -> throw new java.lang.IncompatibleClassChangeError();
+            case 1 -> CRAFTING_CATEGORIES;
+            case 2 -> FURNACE_CATEGORIES;
+            case 3 -> BLAST_FURNACE_CATEGORIES;
+            case 4 -> SMOKER_CATEGORIES;
+        };
     }
 
     public java.util.List<net.minecraft.world.item.ItemStack> getIconItems() {