--- net/minecraft/client/ResourceLoadStateTracker.java
+++ net/minecraft/client/ResourceLoadStateTracker.java
@@ -18,7 +18,7 @@
         if (this.reloadState != null && !this.reloadState.finished) {
             LOGGER.warn("Reload already ongoing, replacing");
         }
-        this.reloadState = new net.minecraft.client.ResourceLoadStateTracker$ReloadState($$0, (java.util.List)$$1.stream().map(net.minecraft.server.packs.PackResources::getName).collect(com.google.common.collect.ImmutableList.toImmutableList()));
+        this.reloadState = new net.minecraft.client.ResourceLoadStateTracker$ReloadState($$0, (java.util.List)$$1.stream().map(net.minecraft.server.packs.PackResources::packId).collect(com.google.common.collect.ImmutableList.toImmutableList()));
     }
 
     public void startRecovery(java.lang.Throwable $$0) {