--- net/minecraft/client/Screenshot.java
+++ net/minecraft/client/Screenshot.java
@@ -61,14 +61,14 @@
         return $$3;
     }
 
-    public static java.io.File getFile(java.io.File gameDirectory) {
-        java.lang.String string = net.minecraft.Util.getFilenameFormattedDateTime();
-        int n = 1;
-        java.io.File file;
-        while ((file = new java.io.File(gameDirectory, string + (n == 1 ? "" : "_" + n) + ".png")).exists()) {
-            ++n;
+    public static java.io.File getFile(java.io.File $$0) {
+        java.lang.String $$1 = net.minecraft.Util.getFilenameFormattedDateTime();
+        int $$2 = 1;
+        java.io.File $$3;
+        while (($$3 = new java.io.File($$0, $$1 + ($$2 == 1 ? "" : "_" + $$2) + ".png")).exists()) {
+            ++$$2;
         }
-        return file;
+        return $$3;
     }
 
     public net.minecraft.client.Screenshot(java.io.File $$0, int $$1, int $$2, int $$3) throws java.io.IOException {