--- net/minecraft/client/StringSplitter.java
+++ net/minecraft/client/StringSplitter.java
@@ -100,12 +100,6 @@
         return (net.minecraft.network.chat.FormattedText)$$0.visit((net.minecraft.network.chat.FormattedText$StyledContentConsumer)new /* Unavailable Anonymous Inner Class!! */, $$2).orElse((java.lang.Object)$$0);
     }
 
-    public java.util.List<net.minecraft.client.StringSplitter$Span> findSpans(net.minecraft.util.FormattedCharSequence $$0, java.util.function.Predicate<net.minecraft.network.chat.Style> $$1) {
-        net.minecraft.client.StringSplitter$SpanBuilder $$2 = new net.minecraft.client.StringSplitter$SpanBuilder(this, $$1);
-        $$0.accept((net.minecraft.util.FormattedCharSink)$$2);
-        return $$2.build();
-    }
-
     public int findLineBreak(java.lang.String $$0, int $$1, net.minecraft.network.chat.Style $$2) {
         net.minecraft.client.StringSplitter$LineBreakFinder $$3 = new net.minecraft.client.StringSplitter$LineBreakFinder(this, (float)$$1);
         net.minecraft.util.StringDecomposer.iterateFormatted((java.lang.String)$$0, (net.minecraft.network.chat.Style)$$2, (net.minecraft.util.FormattedCharSink)$$3);