--- net/minecraft/client/StringSplitter$LineComponent.java
+++ net/minecraft/client/StringSplitter$LineComponent.java
@@ -14,12 +14,12 @@
         this.style = $$1;
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> acceptor) {
-        return acceptor.accept(this.contents);
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> $$0) {
+        return $$0.accept(this.contents);
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> acceptor, net.minecraft.network.chat.Style style) {
-        return acceptor.accept(this.style.applyTo(style), this.contents);
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> $$0, net.minecraft.network.chat.Style $$1) {
+        return $$0.accept(this.style.applyTo($$1), this.contents);
     }
 }
 