--- net/minecraft/client/Timer.java
+++ net/minecraft/client/Timer.java
@@ -10,18 +10,18 @@
     public long lastMs;
     public float msPerTick;
 
-    public net.minecraft.client.Timer(float ticksPerSecond, long lastMs) {
-        this.msPerTick = 1000.0f / ticksPerSecond;
-        this.lastMs = lastMs;
+    public net.minecraft.client.Timer(float $$0, long $$1) {
+        this.msPerTick = 1000.0f / $$0;
+        this.lastMs = $$1;
     }
 
-    public int advanceTime(long gameTime) {
-        this.tickDelta = (float)(gameTime - this.lastMs) / this.msPerTick;
-        this.lastMs = gameTime;
+    public int advanceTime(long $$0) {
+        this.tickDelta = (float)($$0 - this.lastMs) / this.msPerTick;
+        this.lastMs = $$0;
         this.partialTick += this.tickDelta;
-        int n = (int)this.partialTick;
-        this.partialTick -= (float)n;
-        return n;
+        int $$1 = (int)this.partialTick;
+        this.partialTick -= (float)$$1;
+        return $$1;
     }
 }
 