--- net/minecraft/client/User$Type.java
+++ net/minecraft/client/User$Type.java
@@ -21,14 +21,14 @@
         return (net.minecraft.client.User$Type)java.lang.Enum.valueOf(net.minecraft.client.User$Type.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.client.User$Type(java.lang.String name) {
+    public net.minecraft.client.User$Type(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
+        this.name = $$0;
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.client.User$Type byName(java.lang.String typeName) {
-        return (net.minecraft.client.User$Type)BY_NAME.get((java.lang.Object)typeName.toLowerCase(java.util.Locale.ROOT));
+    public static net.minecraft.client.User$Type byName(java.lang.String $$0) {
+        return (net.minecraft.client.User$Type)BY_NAME.get((java.lang.Object)$$0.toLowerCase(java.util.Locale.ROOT));
     }
 
     public java.lang.String getName() {