--- net/minecraft/client/gui/screens/AccessibilityOptionsScreen.java
+++ net/minecraft/client/gui/screens/AccessibilityOptionsScreen.java
@@ -9,7 +9,7 @@
     public static java.lang.String GUIDE_LINK = "https://aka.ms/MinecraftJavaAccessibility";
 
     public static net.minecraft.client.OptionInstance<?>[] options(net.minecraft.client.Options $$0) {
-        return new net.minecraft.client.OptionInstance[]{$$0.narrator(), $$0.showSubtitles(), $$0.textBackgroundOpacity(), $$0.backgroundForChatOnly(), $$0.chatOpacity(), $$0.chatLineSpacing(), $$0.chatDelay(), $$0.autoJump(), $$0.toggleCrouch(), $$0.toggleSprint(), $$0.screenEffectScale(), $$0.fovEffectScale(), $$0.darkMojangStudiosBackground(), $$0.hideLightningFlash(), $$0.darknessEffectScale()};
+        return new net.minecraft.client.OptionInstance[]{$$0.narrator(), $$0.showSubtitles(), $$0.textBackgroundOpacity(), $$0.backgroundForChatOnly(), $$0.chatOpacity(), $$0.chatLineSpacing(), $$0.chatDelay(), $$0.autoJump(), $$0.toggleCrouch(), $$0.toggleSprint(), $$0.screenEffectScale(), $$0.fovEffectScale(), $$0.darkMojangStudiosBackground(), $$0.hideLightningFlash(), $$0.darknessEffectScale(), $$0.panoramaSpeed()};
     }
 
     public net.minecraft.client.gui.screens.AccessibilityOptionsScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.client.Options $$1) {