--- net/minecraft/client/gui/screens/AlertScreen.java
+++ net/minecraft/client/gui/screens/AlertScreen.java
@@ -13,8 +13,8 @@
     public net.minecraft.network.chat.Component okButton;
     public boolean shouldCloseOnEsc;
 
-    public net.minecraft.client.gui.screens.AlertScreen(java.lang.Runnable callback, net.minecraft.network.chat.Component title, net.minecraft.network.chat.Component text) {
-        this(callback, title, text, net.minecraft.network.chat.CommonComponents.GUI_BACK, true);
+    public net.minecraft.client.gui.screens.AlertScreen(java.lang.Runnable $$0, net.minecraft.network.chat.Component $$1, net.minecraft.network.chat.Component $$2) {
+        this($$0, $$1, $$2, net.minecraft.network.chat.CommonComponents.GUI_BACK, true);
     }
 
     public net.minecraft.client.gui.screens.AlertScreen(java.lang.Runnable $$0, net.minecraft.network.chat.Component $$1, net.minecraft.network.chat.Component $$2, net.minecraft.network.chat.Component $$3, boolean $$4) {
@@ -41,11 +41,11 @@
         this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button((this.width - 150) / 2, $$1, 150, 20, this.okButton, $$0 -> this.callback.run()));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.client.gui.screens.AlertScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)70, (int)0xFFFFFF);
-        this.message.renderCentered(matrixStack, this.width / 2, 90);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.client.gui.screens.AlertScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)70, (int)0xFFFFFF);
+        this.message.renderCentered($$0, this.width / 2, 90);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     public boolean shouldCloseOnEsc() {