--- net/minecraft/client/gui/screens/ChatOptionsScreen.java
+++ net/minecraft/client/gui/screens/ChatOptionsScreen.java
@@ -7,7 +7,7 @@
 extends net.minecraft.client.gui.screens.SimpleOptionsSubScreen
  {
     public net.minecraft.client.gui.screens.ChatOptionsScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.client.Options $$1) {
-        super($$0, $$1, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.chat.title"), new net.minecraft.client.OptionInstance[]{$$1.chatVisibility(), $$1.chatColors(), $$1.chatLinks(), $$1.chatLinksPrompt(), $$1.chatOpacity(), $$1.textBackgroundOpacity(), $$1.chatScale(), $$1.chatLineSpacing(), $$1.chatDelay(), $$1.chatWidth(), $$1.chatHeightFocused(), $$1.chatHeightUnfocused(), $$1.narrator(), $$1.autoSuggestions(), $$1.hideMatchedNames(), $$1.reducedDebugInfo(), $$1.chatPreview(), $$1.onlyShowSecureChat()});
+        super($$0, $$1, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.chat.title"), new net.minecraft.client.OptionInstance[]{$$1.chatVisibility(), $$1.chatColors(), $$1.chatLinks(), $$1.chatLinksPrompt(), $$1.chatOpacity(), $$1.textBackgroundOpacity(), $$1.chatScale(), $$1.chatLineSpacing(), $$1.chatDelay(), $$1.chatWidth(), $$1.chatHeightFocused(), $$1.chatHeightUnfocused(), $$1.narrator(), $$1.autoSuggestions(), $$1.hideMatchedNames(), $$1.reducedDebugInfo(), $$1.onlyShowSecureChat()});
     }
 }
 