--- net/minecraft/client/gui/screens/ConnectScreen.java
+++ net/minecraft/client/gui/screens/ConnectScreen.java
@@ -28,17 +28,17 @@
         net.minecraft.client.gui.screens.ConnectScreen $$4 = new net.minecraft.client.gui.screens.ConnectScreen($$0);
         $$1.clearLevel();
         $$1.prepareForMultiplayer();
-        $$1.setCurrentServer($$3);
+        $$1.updateReportEnvironment(net.minecraft.client.multiplayer.chat.report.ReportEnvironment.thirdParty((java.lang.String)($$3 != null ? $$3.ip : $$2.getHost())));
         $$1.setScreen((net.minecraft.client.gui.screens.Screen)$$4);
-        $$4.connect($$1, $$2);
+        $$4.connect($$1, $$2, $$3);
     }
 
-    public void connect(net.minecraft.client.Minecraft $$0, net.minecraft.client.multiplayer.resolver.ServerAddress $$1) {
-        java.util.concurrent.CompletableFuture $$2 = $$0.getProfileKeyPairManager().preparePublicKey();
+    public void connect(net.minecraft.client.Minecraft $$0, net.minecraft.client.multiplayer.resolver.ServerAddress $$1, @javax.annotation.Nullable net.minecraft.client.multiplayer.ServerData $$2) {
+        java.util.concurrent.CompletableFuture $$3 = $$0.getProfileKeyPairManager().prepareChatSession();
         LOGGER.info("Connecting to {}, {}", (java.lang.Object)$$1.getHost(), (java.lang.Object)$$1.getPort());
-        net.minecraft.client.gui.screens.ConnectScreen$1 $$3 = new /* Unavailable Anonymous Inner Class!! */;
-        $$3.setUncaughtExceptionHandler((java.lang.Thread$UncaughtExceptionHandler)new net.minecraft.DefaultUncaughtExceptionHandler(LOGGER));
-        $$3.start();
+        net.minecraft.client.gui.screens.ConnectScreen$1 $$4 = new /* Unavailable Anonymous Inner Class!! */;
+        $$4.setUncaughtExceptionHandler((java.lang.Thread$UncaughtExceptionHandler)new net.minecraft.DefaultUncaughtExceptionHandler(LOGGER));
+        $$4.start();
     }
 
     public void updateStatus(net.minecraft.network.chat.Component $$0) {
@@ -69,15 +69,15 @@
         }));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        long l = net.minecraft.Util.getMillis();
-        if (l - this.lastNarration > 2000L) {
-            this.lastNarration = l;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        long $$4 = net.minecraft.Util.getMillis();
+        if ($$4 - this.lastNarration > 2000L) {
+            this.lastNarration = $$4;
             this.minecraft.getNarrator().sayNow((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"narrator.joining"));
         }
-        net.minecraft.client.gui.screens.ConnectScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.status, (int)(this.width / 2), (int)(this.height / 2 - 50), (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        net.minecraft.client.gui.screens.ConnectScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.status, (int)(this.width / 2), (int)(this.height / 2 - 50), (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {