--- net/minecraft/client/gui/screens/CreateFlatWorldScreen$DetailsList.java
+++ net/minecraft/client/gui/screens/CreateFlatWorldScreen$DetailsList.java
@@ -16,8 +16,8 @@
         }
     }
 
-    public void setSelected(@javax.annotation.Nullable net.minecraft.client.gui.screens.CreateFlatWorldScreen$DetailsList$Entry entry) {
-        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)entry);
+    public void setSelected(@javax.annotation.Nullable net.minecraft.client.gui.screens.CreateFlatWorldScreen$DetailsList$Entry $$0) {
+        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)$$0);
         this.this$0.updateButtonValidity();
     }
 