--- net/minecraft/client/gui/screens/CreateFlatWorldScreen.java
+++ net/minecraft/client/gui/screens/CreateFlatWorldScreen.java
@@ -84,14 +84,14 @@
         this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)this.parent);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        this.list.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.CreateFlatWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)8, (int)0xFFFFFF);
-        int n = this.width / 2 - 92 - 16;
-        net.minecraft.client.gui.screens.CreateFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.columnType, (int)n, (int)32, (int)0xFFFFFF);
-        net.minecraft.client.gui.screens.CreateFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.columnHeight, (int)(n + 2 + 213 - this.font.width((net.minecraft.network.chat.FormattedText)this.columnHeight)), (int)32, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        this.list.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.CreateFlatWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)8, (int)0xFFFFFF);
+        int $$4 = this.width / 2 - 92 - 16;
+        net.minecraft.client.gui.screens.CreateFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.columnType, (int)$$4, (int)32, (int)0xFFFFFF);
+        net.minecraft.client.gui.screens.CreateFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.columnHeight, (int)($$4 + 2 + 213 - this.font.width((net.minecraft.network.chat.FormattedText)this.columnHeight)), (int)32, (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 