--- net/minecraft/client/gui/screens/CreateFlatWorldScreen$DetailsList$Entry.java
+++ net/minecraft/client/gui/screens/CreateFlatWorldScreen$DetailsList$Entry.java
@@ -12,14 +12,21 @@
         this.this$1 = detailsList;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        net.minecraft.world.level.levelgen.flat.FlatLayerInfo flatLayerInfo = (net.minecraft.world.level.levelgen.flat.FlatLayerInfo)this.this$1.this$0.generator.getLayersInfo().get(this.this$1.this$0.generator.getLayersInfo().size() - index - 1);
-        net.minecraft.world.level.block.state.BlockState blockState = flatLayerInfo.getBlockState();
-        net.minecraft.world.item.ItemStack itemStack = this.getDisplayItem(blockState);
-        this.blitSlot(matrixStack, left, top, itemStack);
-        this.this$1.this$0.font.draw(matrixStack, itemStack.getHoverName(), (float)(left + 18 + 5), (float)(top + 3), 0xFFFFFF);
-        net.minecraft.network.chat.MutableComponent mutableComponent = index == 0 ? net.minecraft.network.chat.Component.translatable((java.lang.String)"createWorld.customize.flat.layer.top", (java.lang.Object[])new java.lang.Object[]{flatLayerInfo.getHeight()}) : (index == this.this$1.this$0.generator.getLayersInfo().size() - 1 ? net.minecraft.network.chat.Component.translatable((java.lang.String)"createWorld.customize.flat.layer.bottom", (java.lang.Object[])new java.lang.Object[]{flatLayerInfo.getHeight()}) : net.minecraft.network.chat.Component.translatable((java.lang.String)"createWorld.customize.flat.layer", (java.lang.Object[])new java.lang.Object[]{flatLayerInfo.getHeight()}));
-        this.this$1.this$0.font.draw(matrixStack, (net.minecraft.network.chat.Component)mutableComponent, (float)(left + 2 + 213 - this.this$1.this$0.font.width((net.minecraft.network.chat.FormattedText)mutableComponent)), (float)(top + 3), 0xFFFFFF);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        net.minecraft.network.chat.MutableComponent $$15;
+        net.minecraft.world.level.levelgen.flat.FlatLayerInfo $$10 = (net.minecraft.world.level.levelgen.flat.FlatLayerInfo)this.this$1.this$0.generator.getLayersInfo().get(this.this$1.this$0.generator.getLayersInfo().size() - $$1 - 1);
+        net.minecraft.world.level.block.state.BlockState $$11 = $$10.getBlockState();
+        net.minecraft.world.item.ItemStack $$12 = this.getDisplayItem($$11);
+        this.blitSlot($$0, $$3, $$2, $$12);
+        this.this$1.this$0.font.draw($$0, $$12.getHoverName(), (float)($$3 + 18 + 5), (float)($$2 + 3), 0xFFFFFF);
+        if ($$1 == 0) {
+            net.minecraft.network.chat.MutableComponent $$13 = net.minecraft.network.chat.Component.translatable((java.lang.String)"createWorld.customize.flat.layer.top", (java.lang.Object[])new java.lang.Object[]{$$10.getHeight()});
+        } else if ($$1 == this.this$1.this$0.generator.getLayersInfo().size() - 1) {
+            net.minecraft.network.chat.MutableComponent $$14 = net.minecraft.network.chat.Component.translatable((java.lang.String)"createWorld.customize.flat.layer.bottom", (java.lang.Object[])new java.lang.Object[]{$$10.getHeight()});
+        } else {
+            $$15 = net.minecraft.network.chat.Component.translatable((java.lang.String)"createWorld.customize.flat.layer", (java.lang.Object[])new java.lang.Object[]{$$10.getHeight()});
+        }
+        this.this$1.this$0.font.draw($$0, (net.minecraft.network.chat.Component)$$15, (float)($$3 + 2 + 213 - this.this$1.this$0.font.width((net.minecraft.network.chat.FormattedText)$$15)), (float)($$2 + 3), 0xFFFFFF);
     }
 
     public net.minecraft.world.item.ItemStack getDisplayItem(net.minecraft.world.level.block.state.BlockState $$0) {
@@ -43,8 +50,8 @@
         return net.minecraft.network.chat.CommonComponents.EMPTY;
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (button == 0) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if ($$2 == 0) {
             this.this$1.setSelected(this);
             return true;
         }