--- net/minecraft/client/gui/screens/DeathScreen.java
+++ net/minecraft/client/gui/screens/DeathScreen.java
@@ -62,24 +62,24 @@
         this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.TitleScreen());
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.fillGradient(matrixStack, 0, 0, this.width, this.height, 0x60500000, -1602211792);
-        matrixStack.pushPose();
-        matrixStack.scale(2.0f, 2.0f, 2.0f);
-        net.minecraft.client.gui.screens.DeathScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2 / 2), (int)30, (int)0xFFFFFF);
-        matrixStack.popPose();
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.fillGradient($$0, 0, 0, this.width, this.height, 0x60500000, -1602211792);
+        $$0.pushPose();
+        $$0.scale(2.0f, 2.0f, 2.0f);
+        net.minecraft.client.gui.screens.DeathScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2 / 2), (int)30, (int)0xFFFFFF);
+        $$0.popPose();
         if (this.causeOfDeath != null) {
-            net.minecraft.client.gui.screens.DeathScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.causeOfDeath, (int)(this.width / 2), (int)85, (int)0xFFFFFF);
+            net.minecraft.client.gui.screens.DeathScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.causeOfDeath, (int)(this.width / 2), (int)85, (int)0xFFFFFF);
         }
-        net.minecraft.client.gui.screens.DeathScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.deathScore, (int)(this.width / 2), (int)100, (int)0xFFFFFF);
-        if (this.causeOfDeath != null && mouseY > 85) {
+        net.minecraft.client.gui.screens.DeathScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.deathScore, (int)(this.width / 2), (int)100, (int)0xFFFFFF);
+        if (this.causeOfDeath != null && $$2 > 85) {
             java.util.Objects.requireNonNull((java.lang.Object)this.font);
-            if (mouseY < 85 + 9) {
-                net.minecraft.network.chat.Style style = this.getClickedComponentStyleAt(mouseX);
-                this.renderComponentHoverEffect(matrixStack, style, mouseX, mouseY);
+            if ($$2 < 85 + 9) {
+                net.minecraft.network.chat.Style $$4 = this.getClickedComponentStyleAt($$1);
+                this.renderComponentHoverEffect($$0, $$4, $$1, $$2);
             }
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     @javax.annotation.Nullable
@@ -96,16 +96,16 @@
         return this.minecraft.font.getSplitter().componentStyleAtWidth((net.minecraft.network.chat.FormattedText)this.causeOfDeath, $$0 - $$2);
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (this.causeOfDeath != null && mouseY > 85.0) {
-            net.minecraft.network.chat.Style style;
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if (this.causeOfDeath != null && $$1 > 85.0) {
+            net.minecraft.network.chat.Style $$3;
             java.util.Objects.requireNonNull((java.lang.Object)this.font);
-            if (mouseY < (double)(85 + 9) && (style = this.getClickedComponentStyleAt((int)mouseX)) != null && style.getClickEvent() != null && style.getClickEvent().getAction() == net.minecraft.network.chat.ClickEvent$Action.OPEN_URL) {
-                this.handleComponentClicked(style);
+            if ($$1 < (double)(85 + 9) && ($$3 = this.getClickedComponentStyleAt((int)$$0)) != null && $$3.getClickEvent() != null && $$3.getClickEvent().getAction() == net.minecraft.network.chat.ClickEvent$Action.OPEN_URL) {
+                this.handleComponentClicked($$3);
                 return false;
             }
         }
-        return super.mouseClicked(mouseX, mouseY, button);
+        return super.mouseClicked($$0, $$1, $$2);
     }
 
     public boolean isPauseScreen() {