--- net/minecraft/client/gui/screens/DemoIntroScreen.java
+++ net/minecraft/client/gui/screens/DemoIntroScreen.java
@@ -31,24 +31,24 @@
         this.durationMessage = net.minecraft.client.gui.components.MultiLineLabel.create((net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.FormattedText)net.minecraft.network.chat.Component.translatable((java.lang.String)"demo.help.fullWrapped"), (int)218);
     }
 
-    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        super.renderBackground(matrixStack);
+    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        super.renderBackground($$0);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)DEMO_BACKGROUND_LOCATION);
-        int n = (this.width - 248) / 2;
-        int n2 = (this.height - 166) / 2;
-        this.blit(matrixStack, n, n2, 0, 0, 248, 166);
+        int $$1 = (this.width - 248) / 2;
+        int $$2 = (this.height - 166) / 2;
+        this.blit($$0, $$1, $$2, 0, 0, 248, 166);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        int n = (this.width - 248) / 2 + 10;
-        int n2 = (this.height - 166) / 2 + 8;
-        this.font.draw(matrixStack, this.title, (float)n, (float)n2, 0x1F1F1F);
-        n2 = this.movementMessage.renderLeftAlignedNoShadow(matrixStack, n, n2 + 12, 12, 0x4F4F4F);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        int $$4 = (this.width - 248) / 2 + 10;
+        int $$5 = (this.height - 166) / 2 + 8;
+        this.font.draw($$0, this.title, (float)$$4, (float)$$5, 0x1F1F1F);
+        $$5 = this.movementMessage.renderLeftAlignedNoShadow($$0, $$4, $$5 + 12, 12, 0x4F4F4F);
         java.util.Objects.requireNonNull((java.lang.Object)this.font);
-        this.durationMessage.renderLeftAlignedNoShadow(matrixStack, n, n2 + 20, 9, 0x1F1F1F);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        this.durationMessage.renderLeftAlignedNoShadow($$0, $$4, $$5 + 20, 9, 0x1F1F1F);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {