--- net/minecraft/client/gui/screens/DirectJoinServerScreen.java
+++ net/minecraft/client/gui/screens/DirectJoinServerScreen.java
@@ -24,12 +24,12 @@
         this.ipEdit.tick();
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (this.selectButton.active && this.getFocused() == this.ipEdit && (keyCode == 257 || keyCode == 335)) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if (this.selectButton.active && this.getFocused() == this.ipEdit && ($$0 == 257 || $$0 == 335)) {
             this.onSelect();
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
     public void init() {
@@ -46,10 +46,10 @@
         this.updateSelectButtonStatus();
     }
 
-    public void resize(net.minecraft.client.Minecraft minecraft, int width, int height) {
-        java.lang.String string = this.ipEdit.getValue();
-        this.init(minecraft, width, height);
-        this.ipEdit.setValue(string);
+    public void resize(net.minecraft.client.Minecraft $$0, int $$1, int $$2) {
+        java.lang.String $$3 = this.ipEdit.getValue();
+        this.init($$0, $$1, $$2);
+        this.ipEdit.setValue($$3);
     }
 
     public void onSelect() {
@@ -71,12 +71,12 @@
         this.selectButton.active = net.minecraft.client.multiplayer.resolver.ServerAddress.isValidAddress((java.lang.String)this.ipEdit.getValue());
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.client.gui.screens.DirectJoinServerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
-        net.minecraft.client.gui.screens.DirectJoinServerScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)ENTER_IP_LABEL, (int)(this.width / 2 - 100), (int)100, (int)0xA0A0A0);
-        this.ipEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.client.gui.screens.DirectJoinServerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
+        net.minecraft.client.gui.screens.DirectJoinServerScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)ENTER_IP_LABEL, (int)(this.width / 2 - 100), (int)100, (int)0xA0A0A0);
+        this.ipEdit.render($$0, $$1, $$2, $$3);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {