--- net/minecraft/client/gui/screens/DisconnectedScreen.java
+++ net/minecraft/client/gui/screens/DisconnectedScreen.java
@@ -33,14 +33,14 @@
         this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(n2, java.lang.Math.min((int)(n3 + 9), (int)(this.height - 30)), 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"gui.toMenu"), $$0 -> this.minecraft.setScreen(this.parent)));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
         int n = this.width / 2;
         int n2 = this.height / 2 - this.textHeight / 2;
         java.util.Objects.requireNonNull((java.lang.Object)this.font);
-        net.minecraft.client.gui.screens.DisconnectedScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)n, (int)(n2 - 9 * 2), (int)0xAAAAAA);
-        this.message.renderCentered(matrixStack, this.width / 2, this.height / 2 - this.textHeight / 2);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        net.minecraft.client.gui.screens.DisconnectedScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)n, (int)(n2 - 9 * 2), (int)0xAAAAAA);
+        this.message.renderCentered($$0, this.width / 2, this.height / 2 - this.textHeight / 2);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 