--- net/minecraft/client/gui/screens/EditServerScreen.java
+++ net/minecraft/client/gui/screens/EditServerScreen.java
@@ -45,12 +45,12 @@
         this.updateAddButtonStatus();
     }
 
-    public void resize(net.minecraft.client.Minecraft minecraft, int width, int height) {
-        java.lang.String string = this.ipEdit.getValue();
-        java.lang.String string2 = this.nameEdit.getValue();
-        this.init(minecraft, width, height);
-        this.ipEdit.setValue(string);
-        this.nameEdit.setValue(string2);
+    public void resize(net.minecraft.client.Minecraft $$0, int $$1, int $$2) {
+        java.lang.String $$3 = this.ipEdit.getValue();
+        java.lang.String $$4 = this.nameEdit.getValue();
+        this.init($$0, $$1, $$2);
+        this.ipEdit.setValue($$3);
+        this.nameEdit.setValue($$4);
     }
 
     public void removed() {
@@ -71,14 +71,14 @@
         this.addButton.active = net.minecraft.client.multiplayer.resolver.ServerAddress.isValidAddress((java.lang.String)this.ipEdit.getValue()) && !this.nameEdit.getValue().isEmpty();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.client.gui.screens.EditServerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)17, (int)0xFFFFFF);
-        net.minecraft.client.gui.screens.EditServerScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)NAME_LABEL, (int)(this.width / 2 - 100), (int)53, (int)0xA0A0A0);
-        net.minecraft.client.gui.screens.EditServerScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)IP_LABEL, (int)(this.width / 2 - 100), (int)94, (int)0xA0A0A0);
-        this.nameEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.ipEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.client.gui.screens.EditServerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)17, (int)0xFFFFFF);
+        net.minecraft.client.gui.screens.EditServerScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)NAME_LABEL, (int)(this.width / 2 - 100), (int)53, (int)0xA0A0A0);
+        net.minecraft.client.gui.screens.EditServerScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)IP_LABEL, (int)(this.width / 2 - 100), (int)94, (int)0xA0A0A0);
+        this.nameEdit.render($$0, $$1, $$2, $$3);
+        this.ipEdit.render($$0, $$1, $$2, $$3);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {