--- net/minecraft/client/gui/screens/InBedChatScreen.java
+++ net/minecraft/client/gui/screens/InBedChatScreen.java
@@ -6,30 +6,23 @@
 public class net.minecraft.client.gui.screens.InBedChatScreen
 extends net.minecraft.client.gui.screens.ChatScreen
  {
-    public net.minecraft.client.gui.components.Button leaveBedButton;
-
     public net.minecraft.client.gui.screens.InBedChatScreen() {
         super("");
     }
 
     public void init() {
         super.init();
-        this.leaveBedButton = (net.minecraft.client.gui.components.Button)this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, this.height - 40, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"multiplayer.stopSleeping"), $$0 -> this.sendWakeUp()));
+        this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, this.height - 40, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"multiplayer.stopSleeping"), $$0 -> this.sendWakeUp()));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
-        this.leaveBedButton.visible = this.getDisplayedPreviewText() == null;
-        super.render($$0, $$1, $$2, $$3);
-    }
-
     public void onClose() {
         this.sendWakeUp();
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.sendWakeUp();
-        } else if (keyCode == 257 || keyCode == 335) {
+        } else if ($$0 == 257 || $$0 == 335) {
             if (this.handleChatInput(this.input.getValue(), true)) {
                 this.minecraft.setScreen(null);
                 this.input.setValue("");
@@ -37,7 +30,7 @@
             }
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
     public void sendWakeUp() {