--- net/minecraft/client/gui/screens/LanguageSelectScreen$LanguageSelectionList$Entry.java
+++ net/minecraft/client/gui/screens/LanguageSelectScreen$LanguageSelectionList$Entry.java
@@ -9,18 +9,18 @@
     public net.minecraft.client.resources.language.LanguageInfo language;
     public /* synthetic */ net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList this$1;
 
-    public net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList$Entry(net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList languageSelectionList, net.minecraft.client.resources.language.LanguageInfo language) {
-        this.this$1 = languageSelectionList;
-        this.language = language;
+    public net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList$Entry(net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList $$0, net.minecraft.client.resources.language.LanguageInfo $$1) {
+        this.this$1 = $$0;
+        this.language = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        java.lang.String string = this.language.toString();
-        this.this$1.this$0.font.drawShadow(matrixStack, string, (float)(net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList.access$000((net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList)this.this$1) / 2 - this.this$1.this$0.font.width(string) / 2), (float)(top + 1), 0xFFFFFF, true);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        java.lang.String $$10 = this.language.toString();
+        this.this$1.this$0.font.drawShadow($$0, $$10, (float)(net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList.access$000((net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList)this.this$1) / 2 - this.this$1.this$0.font.width($$10) / 2), (float)($$2 + 1), 0xFFFFFF, true);
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (button == 0) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if ($$2 == 0) {
             this.select();
             return true;
         }