--- net/minecraft/client/gui/screens/LanguageSelectScreen$LanguageSelectionList.java
+++ net/minecraft/client/gui/screens/LanguageSelectScreen$LanguageSelectionList.java
@@ -30,8 +30,8 @@
         return super.getRowWidth() + 50;
     }
 
-    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        this.this$0.renderBackground(matrixStack);
+    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        this.this$0.renderBackground($$0);
     }
 
     public boolean isFocused() {