--- net/minecraft/client/gui/screens/LanguageSelectScreen.java
+++ net/minecraft/client/gui/screens/LanguageSelectScreen.java
@@ -10,9 +10,9 @@
     public net.minecraft.client.gui.screens.LanguageSelectScreen$LanguageSelectionList packSelectionList;
     public net.minecraft.client.resources.language.LanguageManager languageManager;
 
-    public net.minecraft.client.gui.screens.LanguageSelectScreen(net.minecraft.client.gui.screens.Screen screen, net.minecraft.client.Options options, net.minecraft.client.resources.language.LanguageManager languageManager) {
-        super(screen, options, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.language"));
-        this.languageManager = languageManager;
+    public net.minecraft.client.gui.screens.LanguageSelectScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.client.Options $$1, net.minecraft.client.resources.language.LanguageManager $$2) {
+        super($$0, $$1, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.language"));
+        this.languageManager = $$2;
     }
 
     public void init() {
@@ -32,11 +32,11 @@
         super.init();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.packSelectionList.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.LanguageSelectScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)16, (int)0xFFFFFF);
-        net.minecraft.client.gui.screens.LanguageSelectScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)WARNING_LABEL, (int)(this.width / 2), (int)(this.height - 56), (int)0x808080);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.packSelectionList.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.LanguageSelectScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)16, (int)0xFFFFFF);
+        net.minecraft.client.gui.screens.LanguageSelectScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)WARNING_LABEL, (int)(this.width / 2), (int)(this.height - 56), (int)0x808080);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {