--- net/minecraft/client/gui/screens/LevelLoadingScreen.java
+++ net/minecraft/client/gui/screens/LevelLoadingScreen.java
@@ -12,10 +12,10 @@
     public boolean done;
     public static it.unimi.dsi.fastutil.objects.Object2IntMap<net.minecraft.world.level.chunk.ChunkStatus> COLORS;
 
-    public net.minecraft.client.gui.screens.LevelLoadingScreen(net.minecraft.server.level.progress.StoringChunkProgressListener progressListener) {
+    public net.minecraft.client.gui.screens.LevelLoadingScreen(net.minecraft.server.level.progress.StoringChunkProgressListener $$0) {
         super(net.minecraft.client.GameNarrator.NO_TITLE);
         this.lastNarration = -1L;
-        this.progressListener = progressListener;
+        this.progressListener = $$0;
     }
 
     public boolean shouldCloseOnEsc() {
@@ -40,44 +40,44 @@
         return net.minecraft.util.Mth.clamp((int)this.progressListener.getProgress(), (int)0, (int)100) + "%";
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        long l = net.minecraft.Util.getMillis();
-        if (l - this.lastNarration > 2000L) {
-            this.lastNarration = l;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        long $$4 = net.minecraft.Util.getMillis();
+        if ($$4 - this.lastNarration > 2000L) {
+            this.lastNarration = $$4;
             this.triggerImmediateNarration(true);
         }
-        int n = this.width / 2;
-        int n2 = this.height / 2;
-        int n3 = 30;
-        net.minecraft.client.gui.screens.LevelLoadingScreen.renderChunks((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.server.level.progress.StoringChunkProgressListener)this.progressListener, (int)n, (int)(n2 + 30), (int)2, (int)0);
+        int $$5 = this.width / 2;
+        int $$6 = this.height / 2;
+        int $$7 = 30;
+        net.minecraft.client.gui.screens.LevelLoadingScreen.renderChunks((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.server.level.progress.StoringChunkProgressListener)this.progressListener, (int)$$5, (int)($$6 + 30), (int)2, (int)0);
         java.lang.String string = this.getFormattedProgress();
         java.util.Objects.requireNonNull((java.lang.Object)this.font);
-        net.minecraft.client.gui.screens.LevelLoadingScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (java.lang.String)string, (int)n, (int)(n2 - 9 / 2 - 30), (int)0xFFFFFF);
+        net.minecraft.client.gui.screens.LevelLoadingScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (java.lang.String)string, (int)$$5, (int)($$6 - 9 / 2 - 30), (int)0xFFFFFF);
     }
 
-    public static void renderChunks(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.server.level.progress.StoringChunkProgressListener progressListener, int n, int n2, int n3, int n4) {
-        int n5 = n3 + n4;
-        int n6 = progressListener.getFullDiameter();
-        int n7 = n6 * n5 - n4;
-        int n8 = progressListener.getDiameter();
-        int n9 = n8 * n5 - n4;
-        int n10 = n - n9 / 2;
-        int n11 = n2 - n9 / 2;
-        int n12 = n7 / 2 + 1;
-        int n13 = -16772609;
-        if (n4 != 0) {
-            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(n - n12), (int)(n2 - n12), (int)(n - n12 + 1), (int)(n2 + n12), (int)-16772609);
-            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(n + n12 - 1), (int)(n2 - n12), (int)(n + n12), (int)(n2 + n12), (int)-16772609);
-            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(n - n12), (int)(n2 - n12), (int)(n + n12), (int)(n2 - n12 + 1), (int)-16772609);
-            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(n - n12), (int)(n2 + n12 - 1), (int)(n + n12), (int)(n2 + n12), (int)-16772609);
+    public static void renderChunks(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.server.level.progress.StoringChunkProgressListener $$1, int $$2, int $$3, int $$4, int $$5) {
+        int $$6 = $$4 + $$5;
+        int $$7 = $$1.getFullDiameter();
+        int $$8 = $$7 * $$6 - $$5;
+        int $$9 = $$1.getDiameter();
+        int $$10 = $$9 * $$6 - $$5;
+        int $$11 = $$2 - $$10 / 2;
+        int $$12 = $$3 - $$10 / 2;
+        int $$13 = $$8 / 2 + 1;
+        int $$14 = -16772609;
+        if ($$5 != 0) {
+            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$2 - $$13), (int)($$3 - $$13), (int)($$2 - $$13 + 1), (int)($$3 + $$13), (int)-16772609);
+            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$2 + $$13 - 1), (int)($$3 - $$13), (int)($$2 + $$13), (int)($$3 + $$13), (int)-16772609);
+            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$2 - $$13), (int)($$3 - $$13), (int)($$2 + $$13), (int)($$3 - $$13 + 1), (int)-16772609);
+            net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$2 - $$13), (int)($$3 + $$13 - 1), (int)($$2 + $$13), (int)($$3 + $$13), (int)-16772609);
         }
-        for (int i = 0; i < n8; ++i) {
-            for (int j = 0; j < n8; ++j) {
-                net.minecraft.world.level.chunk.ChunkStatus chunkStatus = progressListener.getStatus(i, j);
-                int n14 = n10 + i * n5;
-                int n15 = n11 + j * n5;
-                net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)n14, (int)n15, (int)(n14 + n3), (int)(n15 + n3), (int)(COLORS.getInt((java.lang.Object)chunkStatus) | 0xFF000000));
+        for (int $$15 = 0; $$15 < $$9; ++$$15) {
+            for (int $$16 = 0; $$16 < $$9; ++$$16) {
+                net.minecraft.world.level.chunk.ChunkStatus $$17 = $$1.getStatus($$15, $$16);
+                int $$18 = $$11 + $$15 * $$6;
+                int $$19 = $$12 + $$16 * $$6;
+                net.minecraft.client.gui.screens.LevelLoadingScreen.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$18, (int)$$19, (int)($$18 + $$4), (int)($$19 + $$4), (int)(COLORS.getInt((java.lang.Object)$$17) | 0xFF000000));
             }
         }
     }