--- net/minecraft/client/gui/screens/LoadingOverlay$LogoTexture.java
+++ net/minecraft/client/gui/screens/LoadingOverlay$LogoTexture.java
@@ -10,21 +10,24 @@
         super(net.minecraft.client.gui.screens.LoadingOverlay.MOJANG_STUDIOS_LOGO_LOCATION);
     }
 
-    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage getTextureImage(net.minecraft.server.packs.resources.ResourceManager resourceManager) {
+    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage getTextureImage(net.minecraft.server.packs.resources.ResourceManager $$0) {
         net.minecraft.client.renderer.texture.SimpleTexture$TextureImage textureImage;
-block8:        {
-            net.minecraft.client.Minecraft minecraft = net.minecraft.client.Minecraft.getInstance();
-            net.minecraft.server.packs.VanillaPackResources vanillaPackResources = minecraft.getClientPackSource().getVanillaPack();
-            java.io.InputStream inputStream = vanillaPackResources.getResource(net.minecraft.server.packs.PackType.CLIENT_RESOURCES, net.minecraft.client.gui.screens.LoadingOverlay.MOJANG_STUDIOS_LOGO_LOCATION);
+block9:        {
+            net.minecraft.server.packs.VanillaPackResources $$1 = net.minecraft.client.Minecraft.getInstance().getVanillaPackResources();
+            net.minecraft.server.packs.resources.IoSupplier $$2 = $$1.getResource(net.minecraft.server.packs.PackType.CLIENT_RESOURCES, net.minecraft.client.gui.screens.LoadingOverlay.MOJANG_STUDIOS_LOGO_LOCATION);
+            if ($$2 == null) {
+                return new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage((java.io.IOException)new java.io.FileNotFoundException(net.minecraft.client.gui.screens.LoadingOverlay.MOJANG_STUDIOS_LOGO_LOCATION.toString()));
+            }
+            java.io.InputStream $$3 = (java.io.InputStream)$$2.get();
             try {
-                textureImage = new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(new net.minecraft.client.resources.metadata.texture.TextureMetadataSection(true, true), com.mojang.blaze3d.platform.NativeImage.read((java.io.InputStream)inputStream));
-                if (inputStream == null) break block8;
+                textureImage = new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(new net.minecraft.client.resources.metadata.texture.TextureMetadataSection(true, true), com.mojang.blaze3d.platform.NativeImage.read((java.io.InputStream)$$3));
+                if ($$3 == null) break block9;
             }
             catch (java.lang.Throwable throwable) {
                 try {
-                    if (inputStream != null) {
+                    if ($$3 != null) {
                         try {
-                            inputStream.close();
+                            $$3.close();
                         }
                         catch (java.lang.Throwable throwable2) {
                             throwable.addSuppressed(throwable2);
@@ -32,11 +35,11 @@
                     }
                     throw throwable;
                 }
-                catch (java.io.IOException iOException) {
-                    return new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(iOException);
+                catch (java.io.IOException $$4) {
+                    return new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage($$4);
                 }
             }
-            inputStream.close();
+            $$3.close();
         }
         return textureImage;
     }