--- net/minecraft/client/gui/screens/LoadingOverlay.java
+++ net/minecraft/client/gui/screens/LoadingOverlay.java
@@ -26,90 +26,89 @@
     public long fadeOutStart;
     public long fadeInStart;
 
-    public net.minecraft.client.gui.screens.LoadingOverlay(net.minecraft.client.Minecraft minecraft, net.minecraft.server.packs.resources.ReloadInstance reload, java.util.function.Consumer<java.util.Optional<java.lang.Throwable>> onFinish, boolean fadeIn) {
+    public net.minecraft.client.gui.screens.LoadingOverlay(net.minecraft.client.Minecraft $$0, net.minecraft.server.packs.resources.ReloadInstance $$1, java.util.function.Consumer<java.util.Optional<java.lang.Throwable>> $$2, boolean $$3) {
         this.fadeOutStart = -1L;
         this.fadeInStart = -1L;
-        this.minecraft = minecraft;
-        this.reload = reload;
-        this.onFinish = onFinish;
-        this.fadeIn = fadeIn;
+        this.minecraft = $$0;
+        this.reload = $$1;
+        this.onFinish = $$2;
+        this.fadeIn = $$3;
     }
 
-    public static void registerTextures(net.minecraft.client.Minecraft mc) {
-        mc.getTextureManager().register(MOJANG_STUDIOS_LOGO_LOCATION, (net.minecraft.client.renderer.texture.AbstractTexture)new net.minecraft.client.gui.screens.LoadingOverlay$LogoTexture());
+    public static void registerTextures(net.minecraft.client.Minecraft $$0) {
+        $$0.getTextureManager().register(MOJANG_STUDIOS_LOGO_LOCATION, (net.minecraft.client.renderer.texture.AbstractTexture)new net.minecraft.client.gui.screens.LoadingOverlay$LogoTexture());
     }
 
     public static int replaceAlpha(int $$0, int $$1) {
         return $$0 & 0xFFFFFF | $$1 << 24;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        float f;
-        int n;
-        float f2;
-        int n2 = this.minecraft.getWindow().getGuiScaledWidth();
-        int n3 = this.minecraft.getWindow().getGuiScaledHeight();
-        long l = net.minecraft.Util.getMillis();
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        float $$17;
+        float $$8;
+        int $$4 = this.minecraft.getWindow().getGuiScaledWidth();
+        int $$5 = this.minecraft.getWindow().getGuiScaledHeight();
+        long $$6 = net.minecraft.Util.getMillis();
         if (this.fadeIn && this.fadeInStart == -1L) {
-            this.fadeInStart = l;
+            this.fadeInStart = $$6;
         }
-        float f3 = this.fadeOutStart > -1L ? (float)(l - this.fadeOutStart) / 1000.0f : -1.0f;
-        float f4 = f2 = this.fadeInStart > -1L ? (float)(l - this.fadeInStart) / 500.0f : -1.0f;
-        if (f3 >= 1.0f) {
+        float $$7 = this.fadeOutStart > -1L ? (float)($$6 - this.fadeOutStart) / 1000.0f : -1.0f;
+        float f = $$8 = this.fadeInStart > -1L ? (float)($$6 - this.fadeInStart) / 500.0f : -1.0f;
+        if ($$7 >= 1.0f) {
             if (this.minecraft.screen != null) {
-                this.minecraft.screen.render(matrixStack, 0, 0, partialTicks);
+                this.minecraft.screen.render($$0, 0, 0, $$3);
             }
-            n = net.minecraft.util.Mth.ceil((float)((1.0f - net.minecraft.util.Mth.clamp((float)(f3 - 1.0f), (float)0.0f, (float)1.0f)) * 255.0f));
-            net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)0, (int)0, (int)n2, (int)n3, (int)net.minecraft.client.gui.screens.LoadingOverlay.replaceAlpha((int)BRAND_BACKGROUND.getAsInt(), (int)n));
-            f = 1.0f - net.minecraft.util.Mth.clamp((float)(f3 - 1.0f), (float)0.0f, (float)1.0f);
+            int $$9 = net.minecraft.util.Mth.ceil((float)((1.0f - net.minecraft.util.Mth.clamp((float)($$7 - 1.0f), (float)0.0f, (float)1.0f)) * 255.0f));
+            net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)0, (int)0, (int)$$4, (int)$$5, (int)net.minecraft.client.gui.screens.LoadingOverlay.replaceAlpha((int)BRAND_BACKGROUND.getAsInt(), (int)$$9));
+            float $$10 = 1.0f - net.minecraft.util.Mth.clamp((float)($$7 - 1.0f), (float)0.0f, (float)1.0f);
         } else if (this.fadeIn) {
-            if (this.minecraft.screen != null && f2 < 1.0f) {
-                this.minecraft.screen.render(matrixStack, mouseX, mouseY, partialTicks);
+            if (this.minecraft.screen != null && $$8 < 1.0f) {
+                this.minecraft.screen.render($$0, $$1, $$2, $$3);
             }
-            n = net.minecraft.util.Mth.ceil((double)(net.minecraft.util.Mth.clamp((double)f2, (double)0.15, (double)1.0) * 255.0));
-            net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)0, (int)0, (int)n2, (int)n3, (int)net.minecraft.client.gui.screens.LoadingOverlay.replaceAlpha((int)BRAND_BACKGROUND.getAsInt(), (int)n));
-            f = net.minecraft.util.Mth.clamp((float)f2, (float)0.0f, (float)1.0f);
+            int $$11 = net.minecraft.util.Mth.ceil((double)(net.minecraft.util.Mth.clamp((double)$$8, (double)0.15, (double)1.0) * 255.0));
+            net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)0, (int)0, (int)$$4, (int)$$5, (int)net.minecraft.client.gui.screens.LoadingOverlay.replaceAlpha((int)BRAND_BACKGROUND.getAsInt(), (int)$$11));
+            float $$12 = net.minecraft.util.Mth.clamp((float)$$8, (float)0.0f, (float)1.0f);
         } else {
-            n = BRAND_BACKGROUND.getAsInt();
-            float f5 = (float)(n >> 16 & 0xFF) / 255.0f;
-            float f6 = (float)(n >> 8 & 0xFF) / 255.0f;
-            float f7 = (float)(n & 0xFF) / 255.0f;
-            com.mojang.blaze3d.platform.GlStateManager._clearColor((float)f5, (float)f6, (float)f7, (float)1.0f);
+            int $$13 = BRAND_BACKGROUND.getAsInt();
+            float $$14 = (float)($$13 >> 16 & 0xFF) / 255.0f;
+            float $$15 = (float)($$13 >> 8 & 0xFF) / 255.0f;
+            float $$16 = (float)($$13 & 0xFF) / 255.0f;
+            com.mojang.blaze3d.platform.GlStateManager._clearColor((float)$$14, (float)$$15, (float)$$16, (float)1.0f);
             com.mojang.blaze3d.platform.GlStateManager._clear((int)16384, (boolean)net.minecraft.client.Minecraft.ON_OSX);
-            f = 1.0f;
+            $$17 = 1.0f;
         }
-        n = (int)((double)this.minecraft.getWindow().getGuiScaledWidth() * 0.5);
-        int n4 = (int)((double)this.minecraft.getWindow().getGuiScaledHeight() * 0.5);
-        double d = java.lang.Math.min((double)((double)this.minecraft.getWindow().getGuiScaledWidth() * 0.75), (double)this.minecraft.getWindow().getGuiScaledHeight()) * 0.25;
-        int n5 = (int)(d * 0.5);
-        double d2 = d * 4.0;
-        int n6 = (int)(d2 * 0.5);
+        int $$18 = (int)((double)this.minecraft.getWindow().getGuiScaledWidth() * 0.5);
+        int $$19 = (int)((double)this.minecraft.getWindow().getGuiScaledHeight() * 0.5);
+        double $$20 = java.lang.Math.min((double)((double)this.minecraft.getWindow().getGuiScaledWidth() * 0.75), (double)this.minecraft.getWindow().getGuiScaledHeight()) * 0.25;
+        int $$21 = (int)($$20 * 0.5);
+        double $$22 = $$20 * 4.0;
+        int $$23 = (int)($$22 * 0.5);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)MOJANG_STUDIOS_LOGO_LOCATION);
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.blendEquation((int)32774);
         com.mojang.blaze3d.systems.RenderSystem.blendFunc((int)770, (int)1);
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
-        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)f);
-        net.minecraft.client.gui.screens.LoadingOverlay.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)(n - n6), (int)(n4 - n5), (int)n6, (int)((int)d), (float)-0.0625f, (float)0.0f, (int)120, (int)60, (int)120, (int)120);
-        net.minecraft.client.gui.screens.LoadingOverlay.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)n, (int)(n4 - n5), (int)n6, (int)((int)d), (float)0.0625f, (float)60.0f, (int)120, (int)60, (int)120, (int)120);
+        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)$$17);
+        net.minecraft.client.gui.screens.LoadingOverlay.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$18 - $$23), (int)($$19 - $$21), (int)$$23, (int)((int)$$20), (float)-0.0625f, (float)0.0f, (int)120, (int)60, (int)120, (int)120);
+        net.minecraft.client.gui.screens.LoadingOverlay.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$18, (int)($$19 - $$21), (int)$$23, (int)((int)$$20), (float)0.0625f, (float)60.0f, (int)120, (int)60, (int)120, (int)120);
         com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();
-        int n7 = (int)((double)this.minecraft.getWindow().getGuiScaledHeight() * 0.8325);
-        float f8 = this.reload.getActualProgress();
-        this.currentProgress = net.minecraft.util.Mth.clamp((float)(this.currentProgress * 0.95f + f8 * 0.050000012f), (float)0.0f, (float)1.0f);
-        if (f3 < 1.0f) {
-            this.drawProgressBar(matrixStack, n2 / 2 - n6, n7 - 5, n2 / 2 + n6, n7 + 5, 1.0f - net.minecraft.util.Mth.clamp((float)f3, (float)0.0f, (float)1.0f));
+        int $$24 = (int)((double)this.minecraft.getWindow().getGuiScaledHeight() * 0.8325);
+        float $$25 = this.reload.getActualProgress();
+        this.currentProgress = net.minecraft.util.Mth.clamp((float)(this.currentProgress * 0.95f + $$25 * 0.050000012f), (float)0.0f, (float)1.0f);
+        if ($$7 < 1.0f) {
+            this.drawProgressBar($$0, $$4 / 2 - $$23, $$24 - 5, $$4 / 2 + $$23, $$24 + 5, 1.0f - net.minecraft.util.Mth.clamp((float)$$7, (float)0.0f, (float)1.0f));
         }
-        if (f3 >= 2.0f) {
+        if ($$7 >= 2.0f) {
             this.minecraft.setOverlay(null);
         }
-        if (this.fadeOutStart == -1L && this.reload.isDone() && (!this.fadeIn || f2 >= 2.0f)) {
+        if (this.fadeOutStart == -1L && this.reload.isDone() && (!this.fadeIn || $$8 >= 2.0f)) {
             try {
                 this.reload.checkExceptions();
                 this.onFinish.accept((java.lang.Object)java.util.Optional.empty());
             }
-            catch (java.lang.Throwable throwable) {
-                this.onFinish.accept((java.lang.Object)java.util.Optional.of((java.lang.Object)((java.lang.Object)throwable)));
+            catch (java.lang.Throwable $$26) {
+                this.onFinish.accept((java.lang.Object)java.util.Optional.of((java.lang.Object)((java.lang.Object)$$26)));
             }
             this.fadeOutStart = net.minecraft.Util.getMillis();
             if (this.minecraft.screen != null) {
@@ -118,15 +117,15 @@
         }
     }
 
-    public void drawProgressBar(com.mojang.blaze3d.vertex.PoseStack poseStack, int n, int n2, int n3, int n4, float partialTicks) {
-        int n5 = net.minecraft.util.Mth.ceil((float)((float)(n3 - n - 2) * this.currentProgress));
-        int n6 = java.lang.Math.round((float)(partialTicks * 255.0f));
-        int n7 = net.minecraft.util.FastColor$ARGB32.color((int)n6, (int)255, (int)255, (int)255);
-        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(n + 2), (int)(n2 + 2), (int)(n + n5), (int)(n4 - 2), (int)n7);
-        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(n + 1), (int)n2, (int)(n3 - 1), (int)(n2 + 1), (int)n7);
-        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(n + 1), (int)n4, (int)(n3 - 1), (int)(n4 - 1), (int)n7);
-        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)n, (int)n2, (int)(n + 1), (int)n4, (int)n7);
-        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)n3, (int)n2, (int)(n3 - 1), (int)n4, (int)n7);
+    public void drawProgressBar(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, float $$5) {
+        int $$6 = net.minecraft.util.Mth.ceil((float)((float)($$3 - $$1 - 2) * this.currentProgress));
+        int $$7 = java.lang.Math.round((float)($$5 * 255.0f));
+        int $$8 = net.minecraft.util.FastColor$ARGB32.color((int)$$7, (int)255, (int)255, (int)255);
+        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 2), (int)($$2 + 2), (int)($$1 + $$6), (int)($$4 - 2), (int)$$8);
+        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 1), (int)$$2, (int)($$3 - 1), (int)($$2 + 1), (int)$$8);
+        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 1), (int)$$4, (int)($$3 - 1), (int)($$4 - 1), (int)$$8);
+        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$1, (int)$$2, (int)($$1 + 1), (int)$$4, (int)$$8);
+        net.minecraft.client.gui.screens.LoadingOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (int)($$3 - 1), (int)$$4, (int)$$8);
     }
 
     public boolean isPauseScreen() {