--- net/minecraft/client/gui/screens/MenuScreens.java
+++ net/minecraft/client/gui/screens/MenuScreens.java
@@ -11,28 +11,28 @@
     public net.minecraft.client.gui.screens.MenuScreens() {
     }
 
-    public static <T extends net.minecraft.world.inventory.AbstractContainerMenu> void create(@javax.annotation.Nullable net.minecraft.world.inventory.MenuType<T> type, net.minecraft.client.Minecraft mc, int windowId, net.minecraft.network.chat.Component title) {
-        if (type == null) {
-            LOGGER.warn("Trying to open invalid screen with name: {}", (java.lang.Object)title.getString());
+    public static <T extends net.minecraft.world.inventory.AbstractContainerMenu> void create(@javax.annotation.Nullable net.minecraft.world.inventory.MenuType<T> $$0, net.minecraft.client.Minecraft $$1, int $$2, net.minecraft.network.chat.Component $$3) {
+        if ($$0 == null) {
+            LOGGER.warn("Trying to open invalid screen with name: {}", (java.lang.Object)$$3.getString());
             return;
         }
-        net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor screenConstructor = net.minecraft.client.gui.screens.MenuScreens.getConstructor(type);
-        if (screenConstructor == null) {
-            LOGGER.warn("Failed to create screen for menu type: {}", (java.lang.Object)net.minecraft.core.Registry.MENU.getKey(type));
+        net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor $$4 = net.minecraft.client.gui.screens.MenuScreens.getConstructor($$0);
+        if ($$4 == null) {
+            LOGGER.warn("Failed to create screen for menu type: {}", (java.lang.Object)net.minecraft.core.Registry.MENU.getKey($$0));
             return;
         }
-        screenConstructor.fromPacket(title, type, mc, windowId);
+        $$4.fromPacket($$3, $$0, $$1, $$2);
     }
 
     @javax.annotation.Nullable
-    public static <T extends net.minecraft.world.inventory.AbstractContainerMenu> net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor<T, ?> getConstructor(net.minecraft.world.inventory.MenuType<T> type) {
-        return (net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor)SCREENS.get(type);
+    public static <T extends net.minecraft.world.inventory.AbstractContainerMenu> net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor<T, ?> getConstructor(net.minecraft.world.inventory.MenuType<T> $$0) {
+        return (net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor)SCREENS.get($$0);
     }
 
-    public static <M extends net.minecraft.world.inventory.AbstractContainerMenu, U extends net.minecraft.client.gui.screens.Screen> void register(net.minecraft.world.inventory.MenuType<? extends M> type, net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor<M, U> factory) {
-        net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor screenConstructor = (net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor)SCREENS.put(type, factory);
-        if (screenConstructor != null) {
-            throw new java.lang.IllegalStateException("Duplicate registration for " + net.minecraft.core.Registry.MENU.getKey(type));
+    public static <M extends net.minecraft.world.inventory.AbstractContainerMenu, U extends net.minecraft.client.gui.screens.Screen> void register(net.minecraft.world.inventory.MenuType<? extends M> $$0, net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor<M, U> $$1) {
+        net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor $$2 = (net.minecraft.client.gui.screens.MenuScreens$ScreenConstructor)SCREENS.put($$0, $$1);
+        if ($$2 != null) {
+            throw new java.lang.IllegalStateException("Duplicate registration for " + net.minecraft.core.Registry.MENU.getKey($$0));
         }
     }
 