--- net/minecraft/client/gui/screens/OptionsScreen.java
+++ net/minecraft/client/gui/screens/OptionsScreen.java
@@ -11,10 +11,10 @@
     public net.minecraft.client.gui.components.CycleButton<net.minecraft.world.Difficulty> difficultyButton;
     public net.minecraft.client.gui.components.LockIconButton lockButton;
 
-    public net.minecraft.client.gui.screens.OptionsScreen(net.minecraft.client.gui.screens.Screen lastScreen, net.minecraft.client.Options options) {
+    public net.minecraft.client.gui.screens.OptionsScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.client.Options $$1) {
         super((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.title"));
-        this.lastScreen = lastScreen;
-        this.options = options;
+        this.lastScreen = $$0;
+        this.options = $$1;
     }
 
     public void init() {
@@ -54,26 +54,26 @@
         return net.minecraft.client.gui.components.CycleButton.builder(net.minecraft.world.Difficulty::getDisplayName).withValues((java.lang.Object[])net.minecraft.world.Difficulty.values()).withInitialValue((java.lang.Object)$$4.level.getDifficulty()).create($$12 / 2 - 155 + $$0 % 2 * 160, $$22 / 6 - 12 + 24 * ($$0 >> 1), 150, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)$$3), ($$1, $$2) -> $$4.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundChangeDifficultyPacket($$2)));
     }
 
-    public void updatePackList(net.minecraft.server.packs.repository.PackRepository packRepository) {
-        com.google.common.collect.ImmutableList immutableList = com.google.common.collect.ImmutableList.copyOf((java.util.Collection)this.options.resourcePacks);
+    public void updatePackList(net.minecraft.server.packs.repository.PackRepository $$0) {
+        com.google.common.collect.ImmutableList $$1 = com.google.common.collect.ImmutableList.copyOf((java.util.Collection)this.options.resourcePacks);
         this.options.resourcePacks.clear();
         this.options.incompatibleResourcePacks.clear();
-        for (net.minecraft.server.packs.repository.Pack pack : packRepository.getSelectedPacks()) {
-            if (pack.isFixedPosition()) continue;
-            this.options.resourcePacks.add((java.lang.Object)pack.getId());
-            if (pack.getCompatibility().isCompatible()) continue;
-            this.options.incompatibleResourcePacks.add((java.lang.Object)pack.getId());
+        for (net.minecraft.server.packs.repository.Pack $$2 : $$0.getSelectedPacks()) {
+            if ($$2.isFixedPosition()) continue;
+            this.options.resourcePacks.add((java.lang.Object)$$2.getId());
+            if ($$2.getCompatibility().isCompatible()) continue;
+            this.options.incompatibleResourcePacks.add((java.lang.Object)$$2.getId());
         }
         this.options.save();
-        java.util.Iterator iterator = com.google.common.collect.ImmutableList.copyOf((java.util.Collection)this.options.resourcePacks);
-        if (!iterator.equals((java.lang.Object)immutableList)) {
+        com.google.common.collect.ImmutableList $$3 = com.google.common.collect.ImmutableList.copyOf((java.util.Collection)this.options.resourcePacks);
+        if (!$$3.equals((java.lang.Object)$$1)) {
             this.minecraft.reloadResourcePacks();
         }
     }
 
-    public void lockCallback(boolean value) {
+    public void lockCallback(boolean $$0) {
         this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)this);
-        if (value && this.minecraft.level != null) {
+        if ($$0 && this.minecraft.level != null) {
             this.minecraft.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundLockDifficultyPacket(true));
             this.lockButton.setLocked(true);
             this.lockButton.active = false;
@@ -85,10 +85,10 @@
         this.options.save();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.client.gui.screens.OptionsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)15, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.client.gui.screens.OptionsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)15, (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 