--- net/minecraft/client/gui/screens/OptionsSubScreen.java
+++ net/minecraft/client/gui/screens/OptionsSubScreen.java
@@ -9,10 +9,10 @@
     public net.minecraft.client.gui.screens.Screen lastScreen;
     public net.minecraft.client.Options options;
 
-    public net.minecraft.client.gui.screens.OptionsSubScreen(net.minecraft.client.gui.screens.Screen lastScreen, net.minecraft.client.Options options, net.minecraft.network.chat.Component title) {
-        super(title);
-        this.lastScreen = lastScreen;
-        this.options = options;
+    public net.minecraft.client.gui.screens.OptionsSubScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.client.Options $$1, net.minecraft.network.chat.Component $$2) {
+        super($$2);
+        this.lastScreen = $$0;
+        this.options = $$1;
     }
 
     public void removed() {
@@ -23,12 +23,21 @@
         this.minecraft.setScreen(this.lastScreen);
     }
 
-    public static java.util.List<net.minecraft.util.FormattedCharSequence> tooltipAt(net.minecraft.client.gui.components.OptionsList options, int x, int y) {
-        java.util.Optional optional = options.getMouseOver((double)x, (double)y);
-        if (optional.isPresent() && optional.get() instanceof net.minecraft.client.gui.components.TooltipAccessor) {
-            return ((net.minecraft.client.gui.components.TooltipAccessor)optional.get()).getTooltip();
+    public static java.util.List<net.minecraft.util.FormattedCharSequence> tooltipAt(net.minecraft.client.gui.components.OptionsList $$0, int $$1, int $$2) {
+        java.util.Optional $$3 = $$0.getMouseOver((double)$$1, (double)$$2);
+        if ($$3.isPresent() && $$3.get() instanceof net.minecraft.client.gui.components.TooltipAccessor) {
+            return ((net.minecraft.client.gui.components.TooltipAccessor)$$3.get()).getTooltip();
         }
         return com.google.common.collect.ImmutableList.of();
     }
+
+    public void basicListRender(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.gui.components.OptionsList $$1, int $$2, int $$3, float $$4) {
+        this.renderBackground($$0);
+        $$1.render($$0, $$2, $$3, $$4);
+        net.minecraft.client.gui.screens.OptionsSubScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
+        super.render($$0, $$2, $$3, $$4);
+        java.util.List $$5 = net.minecraft.client.gui.screens.OptionsSubScreen.tooltipAt((net.minecraft.client.gui.components.OptionsList)$$1, (int)$$2, (int)$$3);
+        this.renderTooltip($$0, $$5, $$2, $$3);
+    }
 }
 