--- net/minecraft/client/gui/screens/PauseScreen.java
+++ net/minecraft/client/gui/screens/PauseScreen.java
@@ -11,9 +11,9 @@
     public static java.lang.String URL_BUGS = "https://aka.ms/snapshotbugs?ref=game";
     public boolean showPauseMenu;
 
-    public net.minecraft.client.gui.screens.PauseScreen(boolean showPauseMenu) {
-        super((net.minecraft.network.chat.Component)(showPauseMenu ? net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.game") : net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.paused")));
-        this.showPauseMenu = showPauseMenu;
+    public net.minecraft.client.gui.screens.PauseScreen(boolean $$0) {
+        super((net.minecraft.network.chat.Component)($$0 ? net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.game") : net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.paused")));
+        this.showPauseMenu = $$0;
     }
 
     public void init() {
@@ -77,14 +77,14 @@
         super.tick();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         if (this.showPauseMenu) {
-            this.renderBackground(matrixStack);
-            net.minecraft.client.gui.screens.PauseScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)40, (int)0xFFFFFF);
+            this.renderBackground($$0);
+            net.minecraft.client.gui.screens.PauseScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)40, (int)0xFFFFFF);
         } else {
-            net.minecraft.client.gui.screens.PauseScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)10, (int)0xFFFFFF);
+            net.minecraft.client.gui.screens.PauseScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)10, (int)0xFFFFFF);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 