--- net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
+++ net/minecraft/client/gui/screens/PresetFlatWorldScreen.java
@@ -24,9 +24,9 @@
     public net.minecraft.client.gui.components.EditBox export;
     public net.minecraft.world.level.levelgen.flat.FlatLevelGeneratorSettings settings;
 
-    public net.minecraft.client.gui.screens.PresetFlatWorldScreen(net.minecraft.client.gui.screens.CreateFlatWorldScreen parent) {
+    public net.minecraft.client.gui.screens.PresetFlatWorldScreen(net.minecraft.client.gui.screens.CreateFlatWorldScreen $$0) {
         super((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"createWorld.customize.presets.title"));
-        this.parent = parent;
+        this.parent = $$0;
     }
 
     /*
@@ -102,7 +102,7 @@
                 $$7 = DEFAULT_BIOME;
             }
         }
-        $$6.setBiome($$0.getOrCreateHolderOrThrow($$7));
+        $$6.setBiome((net.minecraft.core.Holder)$$0.getOrCreateHolderOrThrow($$7));
         return $$6;
     }
 
@@ -142,14 +142,14 @@
         this.updateButtonValidity(this.list.getSelected() != null);
     }
 
-    public boolean mouseScrolled(double mouseX, double mouseY, double delta) {
-        return this.list.mouseScrolled(mouseX, mouseY, delta);
+    public boolean mouseScrolled(double $$0, double $$1, double $$2) {
+        return this.list.mouseScrolled($$0, $$1, $$2);
     }
 
-    public void resize(net.minecraft.client.Minecraft minecraft, int width, int height) {
-        java.lang.String string = this.export.getValue();
-        this.init(minecraft, width, height);
-        this.export.setValue(string);
+    public void resize(net.minecraft.client.Minecraft $$0, int $$1, int $$2) {
+        java.lang.String $$3 = this.export.getValue();
+        this.init($$0, $$1, $$2);
+        this.export.setValue($$3);
     }
 
     public void onClose() {
@@ -160,17 +160,17 @@
         this.minecraft.keyboardHandler.setSendRepeatsToGui(false);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        this.list.render(matrixStack, mouseX, mouseY, partialTicks);
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.0, 400.0);
-        net.minecraft.client.gui.screens.PresetFlatWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)8, (int)0xFFFFFF);
-        net.minecraft.client.gui.screens.PresetFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.shareText, (int)50, (int)30, (int)0xA0A0A0);
-        net.minecraft.client.gui.screens.PresetFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.listText, (int)50, (int)70, (int)0xA0A0A0);
-        matrixStack.popPose();
-        this.export.render(matrixStack, mouseX, mouseY, partialTicks);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        this.list.render($$0, $$1, $$2, $$3);
+        $$0.pushPose();
+        $$0.translate(0.0, 0.0, 400.0);
+        net.minecraft.client.gui.screens.PresetFlatWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)8, (int)0xFFFFFF);
+        net.minecraft.client.gui.screens.PresetFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.shareText, (int)50, (int)30, (int)0xA0A0A0);
+        net.minecraft.client.gui.screens.PresetFlatWorldScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.listText, (int)50, (int)70, (int)0xA0A0A0);
+        $$0.popPose();
+        this.export.render($$0, $$1, $$2, $$3);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     public void tick() {