--- net/minecraft/client/gui/screens/ProgressScreen.java
+++ net/minecraft/client/gui/screens/ProgressScreen.java
@@ -24,43 +24,43 @@
         return false;
     }
 
-    public void progressStartNoAbort(net.minecraft.network.chat.Component component) {
-        this.progressStart(component);
+    public void progressStartNoAbort(net.minecraft.network.chat.Component $$0) {
+        this.progressStart($$0);
     }
 
-    public void progressStart(net.minecraft.network.chat.Component component) {
-        this.header = component;
+    public void progressStart(net.minecraft.network.chat.Component $$0) {
+        this.header = $$0;
         this.progressStage((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"progress.working"));
     }
 
-    public void progressStage(net.minecraft.network.chat.Component component) {
-        this.stage = component;
+    public void progressStage(net.minecraft.network.chat.Component $$0) {
+        this.stage = $$0;
         this.progressStagePercentage(0);
     }
 
-    public void progressStagePercentage(int progress) {
-        this.progress = progress;
+    public void progressStagePercentage(int $$0) {
+        this.progress = $$0;
     }
 
     public void stop() {
         this.stop = true;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         if (this.stop) {
             if (this.clearScreenAfterStop) {
                 this.minecraft.setScreen(null);
             }
             return;
         }
-        this.renderBackground(matrixStack);
+        this.renderBackground($$0);
         if (this.header != null) {
-            net.minecraft.client.gui.screens.ProgressScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.header, (int)(this.width / 2), (int)70, (int)0xFFFFFF);
+            net.minecraft.client.gui.screens.ProgressScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.header, (int)(this.width / 2), (int)70, (int)0xFFFFFF);
         }
         if (this.stage != null && this.progress != 0) {
-            net.minecraft.client.gui.screens.ProgressScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.empty().append(this.stage).append(" " + this.progress + "%"), (int)(this.width / 2), (int)90, (int)0xFFFFFF);
+            net.minecraft.client.gui.screens.ProgressScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.empty().append(this.stage).append(" " + this.progress + "%"), (int)(this.width / 2), (int)90, (int)0xFFFFFF);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 