--- net/minecraft/client/gui/screens/ShareToLanScreen.java
+++ net/minecraft/client/gui/screens/ShareToLanScreen.java
@@ -13,10 +13,10 @@
     public net.minecraft.world.level.GameType gameMode;
     public boolean commands;
 
-    public net.minecraft.client.gui.screens.ShareToLanScreen(net.minecraft.client.gui.screens.Screen lastScreen) {
+    public net.minecraft.client.gui.screens.ShareToLanScreen(net.minecraft.client.gui.screens.Screen $$0) {
         super((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"lanServer.title"));
         this.gameMode = net.minecraft.world.level.GameType.SURVIVAL;
-        this.lastScreen = lastScreen;
+        this.lastScreen = $$0;
     }
 
     public void init() {
@@ -41,11 +41,11 @@
         this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 + 5, this.height - 28, 150, 20, net.minecraft.network.chat.CommonComponents.GUI_CANCEL, $$0 -> this.minecraft.setScreen(this.lastScreen)));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.client.gui.screens.ShareToLanScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)50, (int)0xFFFFFF);
-        net.minecraft.client.gui.screens.ShareToLanScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)INFO_TEXT, (int)(this.width / 2), (int)82, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.client.gui.screens.ShareToLanScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)50, (int)0xFFFFFF);
+        net.minecraft.client.gui.screens.ShareToLanScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)INFO_TEXT, (int)(this.width / 2), (int)82, (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {