--- net/minecraft/client/gui/screens/SimpleOptionsSubScreen.java
+++ net/minecraft/client/gui/screens/SimpleOptionsSubScreen.java
@@ -31,13 +31,8 @@
         this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, this.height - 27, 200, 20, net.minecraft.network.chat.CommonComponents.GUI_DONE, $$0 -> this.minecraft.setScreen(this.lastScreen)));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        this.list.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.SimpleOptionsSubScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        java.util.List list = net.minecraft.client.gui.screens.SimpleOptionsSubScreen.tooltipAt((net.minecraft.client.gui.components.OptionsList)this.list, (int)mouseX, (int)mouseY);
-        this.renderTooltip(matrixStack, list, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.basicListRender($$0, this.list, $$1, $$2, $$3);
     }
 
     public void updateNarratorButton() {