--- net/minecraft/client/gui/screens/TitleScreen.java
+++ net/minecraft/client/gui/screens/TitleScreen.java
@@ -48,8 +48,8 @@
         this.minecraft.getRealms32BitWarningStatus().showRealms32BitWarningIfNeeded((net.minecraft.client.gui.screens.Screen)this);
     }
 
-    public static java.util.concurrent.CompletableFuture<java.lang.Void> preloadResources(net.minecraft.client.renderer.texture.TextureManager texMngr, java.util.concurrent.Executor backgroundExecutor) {
-        return java.util.concurrent.CompletableFuture.allOf((java.util.concurrent.CompletableFuture[])new java.util.concurrent.CompletableFuture[]{texMngr.preload(MINECRAFT_LOGO, backgroundExecutor), texMngr.preload(MINECRAFT_EDITION, backgroundExecutor), texMngr.preload(PANORAMA_OVERLAY, backgroundExecutor), CUBE_MAP.preload(texMngr, backgroundExecutor)});
+    public static java.util.concurrent.CompletableFuture<java.lang.Void> preloadResources(net.minecraft.client.renderer.texture.TextureManager $$0, java.util.concurrent.Executor $$1) {
+        return java.util.concurrent.CompletableFuture.allOf((java.util.concurrent.CompletableFuture[])new java.util.concurrent.CompletableFuture[]{$$0.preload(MINECRAFT_LOGO, $$1), $$0.preload(MINECRAFT_EDITION, $$1), $$0.preload(PANORAMA_OVERLAY, $$1), CUBE_MAP.preload($$0, $$1)});
     }
 
     public boolean isPauseScreen() {
@@ -90,13 +90,13 @@
         }
     }
 
-    public void createNormalMenuOptions(int y, int rowHeight) {
-        this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, y, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.singleplayer"), $$0 -> this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.worldselection.SelectWorldScreen((net.minecraft.client.gui.screens.Screen)this))));
-        net.minecraft.network.chat.Component component = this.getMultiplayerDisabledReason();
-        boolean bl = component == null;
-        net.minecraft.client.gui.components.Button$OnTooltip onTooltip = component == null ? net.minecraft.client.gui.components.Button.NO_TOOLTIP : new /* Unavailable Anonymous Inner Class!! */;
-        ((net.minecraft.client.gui.components.Button)this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button((int)(this.width / 2 - 100), (int)(y + rowHeight * 1), (int)200, (int)20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.multiplayer"), (net.minecraft.client.gui.components.Button$OnPress)(net.minecraft.client.gui.components.Button$OnPress)java.lang.invoke.LambdaMetafactory.metafactory(null, null, null, (Lnet/minecraft/client/gui/components/Button;)V, lambda$createNormalMenuOptions$6(net.minecraft.client.gui.components.Button ), (Lnet/minecraft/client/gui/components/Button;)V)((net.minecraft.client.gui.screens.TitleScreen)this), (net.minecraft.client.gui.components.Button$OnTooltip)onTooltip))).active = bl;
-        ((net.minecraft.client.gui.components.Button)this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button((int)(this.width / 2 - 100), (int)(y + rowHeight * 2), (int)200, (int)20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.online"), (net.minecraft.client.gui.components.Button$OnPress)(net.minecraft.client.gui.components.Button$OnPress)java.lang.invoke.LambdaMetafactory.metafactory(null, null, null, (Lnet/minecraft/client/gui/components/Button;)V, lambda$createNormalMenuOptions$7(net.minecraft.client.gui.components.Button ), (Lnet/minecraft/client/gui/components/Button;)V)((net.minecraft.client.gui.screens.TitleScreen)this), (net.minecraft.client.gui.components.Button$OnTooltip)onTooltip))).active = bl;
+    public void createNormalMenuOptions(int $$02, int $$1) {
+        this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, $$02, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.singleplayer"), $$0 -> this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.worldselection.SelectWorldScreen((net.minecraft.client.gui.screens.Screen)this))));
+        net.minecraft.network.chat.Component $$2 = this.getMultiplayerDisabledReason();
+        boolean $$3 = $$2 == null;
+        net.minecraft.client.gui.components.Button$OnTooltip $$4 = $$2 == null ? net.minecraft.client.gui.components.Button.NO_TOOLTIP : new /* Unavailable Anonymous Inner Class!! */;
+        ((net.minecraft.client.gui.components.Button)this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button((int)(this.width / 2 - 100), (int)($$02 + $$1 * 1), (int)200, (int)20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.multiplayer"), (net.minecraft.client.gui.components.Button$OnPress)(net.minecraft.client.gui.components.Button$OnPress)java.lang.invoke.LambdaMetafactory.metafactory(null, null, null, (Lnet/minecraft/client/gui/components/Button;)V, lambda$createNormalMenuOptions$6(net.minecraft.client.gui.components.Button ), (Lnet/minecraft/client/gui/components/Button;)V)((net.minecraft.client.gui.screens.TitleScreen)this), (net.minecraft.client.gui.components.Button$OnTooltip)$$4))).active = $$3;
+        ((net.minecraft.client.gui.components.Button)this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button((int)(this.width / 2 - 100), (int)($$02 + $$1 * 2), (int)200, (int)20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.online"), (net.minecraft.client.gui.components.Button$OnPress)(net.minecraft.client.gui.components.Button$OnPress)java.lang.invoke.LambdaMetafactory.metafactory(null, null, null, (Lnet/minecraft/client/gui/components/Button;)V, lambda$createNormalMenuOptions$7(net.minecraft.client.gui.components.Button ), (Lnet/minecraft/client/gui/components/Button;)V)((net.minecraft.client.gui.screens.TitleScreen)this), (net.minecraft.client.gui.components.Button$OnTooltip)$$4))).active = $$3;
     }
 
     @javax.annotation.Nullable
@@ -114,17 +114,16 @@
         return net.minecraft.network.chat.Component.translatable((java.lang.String)"title.multiplayer.disabled");
     }
 
-    public void createDemoMenuOptions(int y, int rowHeight) {
-        boolean bl = this.checkDemoWorldPresence();
-        this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, y, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.playdemo"), $$1 -> {
-            if (bl) {
+    public void createDemoMenuOptions(int $$02, int $$12) {
+        boolean $$2 = this.checkDemoWorldPresence();
+        this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, $$02, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.playdemo"), $$1 -> {
+            if ($$2) {
                 this.minecraft.createWorldOpenFlows().loadLevel((net.minecraft.client.gui.screens.Screen)this, "Demo_World");
             } else {
-                net.minecraft.core.RegistryAccess$Frozen $$2 = net.minecraft.core.RegistryAccess.builtinCopy().freeze();
-                this.minecraft.createWorldOpenFlows().createFreshLevel("Demo_World", net.minecraft.server.MinecraftServer.DEMO_SETTINGS, (net.minecraft.core.RegistryAccess)$$2, net.minecraft.world.level.levelgen.presets.WorldPresets.demoSettings((net.minecraft.core.RegistryAccess)$$2));
+                this.minecraft.createWorldOpenFlows().createFreshLevel("Demo_World", net.minecraft.server.MinecraftServer.DEMO_SETTINGS, net.minecraft.world.level.levelgen.WorldOptions.DEMO_OPTIONS, net.minecraft.world.level.levelgen.presets.WorldPresets::createNormalWorldDimensions);
             }
         }));
-        this.resetDemoButton = (net.minecraft.client.gui.components.Button)this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, y + rowHeight * 1, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.resetdemo"), $$0 -> {
+        this.resetDemoButton = (net.minecraft.client.gui.components.Button)this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, $$02 + $$12 * 1, 200, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"menu.resetdemo"), $$0 -> {
             net.minecraft.world.level.storage.LevelStorageSource $$1 = this.minecraft.getLevelSource();
             try (net.minecraft.world.level.storage.LevelStorageSource$LevelStorageAccess $$2 = $$1.createAccess("Demo_World");
 ){
@@ -138,7 +137,7 @@
                 LOGGER.warn("Failed to access demo world", (java.lang.Throwable)$$4);
             }
         }));
-        this.resetDemoButton.active = bl;
+        this.resetDemoButton.active = $$2;
     }
 
     public boolean checkDemoWorldPresence() {
@@ -176,80 +175,80 @@
         this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new com.mojang.realmsclient.RealmsMainScreen((net.minecraft.client.gui.screens.Screen)this));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$12, int $$22, float $$3) {
         if (this.fadeInStart == 0L && this.fading) {
             this.fadeInStart = net.minecraft.Util.getMillis();
         }
-        float f = this.fading ? (float)(net.minecraft.Util.getMillis() - this.fadeInStart) / 1000.0f : 1.0f;
-        this.panorama.render(partialTicks, net.minecraft.util.Mth.clamp((float)f, (float)0.0f, (float)1.0f));
-        int n = 274;
-        int n2 = this.width / 2 - 137;
-        int n3 = 30;
+        float $$4 = this.fading ? (float)(net.minecraft.Util.getMillis() - this.fadeInStart) / 1000.0f : 1.0f;
+        this.panorama.render($$3, net.minecraft.util.Mth.clamp((float)$$4, (float)0.0f, (float)1.0f));
+        int $$5 = 274;
+        int $$6 = this.width / 2 - 137;
+        int $$7 = 30;
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)PANORAMA_OVERLAY);
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.blendFunc((com.mojang.blaze3d.platform.GlStateManager$SourceFactor)com.mojang.blaze3d.platform.GlStateManager$SourceFactor.SRC_ALPHA, (com.mojang.blaze3d.platform.GlStateManager$DestFactor)com.mojang.blaze3d.platform.GlStateManager$DestFactor.ONE_MINUS_SRC_ALPHA);
-        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)(this.fading ? (float)net.minecraft.util.Mth.ceil((float)net.minecraft.util.Mth.clamp((float)f, (float)0.0f, (float)1.0f)) : 1.0f));
-        net.minecraft.client.gui.screens.TitleScreen.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)0, (int)0, (int)this.width, (int)this.height, (float)0.0f, (float)0.0f, (int)16, (int)128, (int)16, (int)128);
-        float f2 = this.fading ? net.minecraft.util.Mth.clamp((float)(f - 1.0f), (float)0.0f, (float)1.0f) : 1.0f;
-        int n4 = net.minecraft.util.Mth.ceil((float)(f2 * 255.0f)) << 24;
-        if ((n4 & 0xFC000000) == 0) {
+        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)(this.fading ? (float)net.minecraft.util.Mth.ceil((float)net.minecraft.util.Mth.clamp((float)$$4, (float)0.0f, (float)1.0f)) : 1.0f));
+        net.minecraft.client.gui.screens.TitleScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)0, (int)0, (int)this.width, (int)this.height, (float)0.0f, (float)0.0f, (int)16, (int)128, (int)16, (int)128);
+        float $$8 = this.fading ? net.minecraft.util.Mth.clamp((float)($$4 - 1.0f), (float)0.0f, (float)1.0f) : 1.0f;
+        int $$9 = net.minecraft.util.Mth.ceil((float)($$8 * 255.0f)) << 24;
+        if (($$9 & 0xFC000000) == 0) {
             return;
         }
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)MINECRAFT_LOGO);
-        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)f2);
+        com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)$$8);
         if (this.minceraftEasterEgg) {
-            this.blitOutlineBlack(n2, 30, ($$1, $$2) -> {
-                this.blit(matrixStack, $$1 + 0, $$2.intValue(), 0, 0, 99, 44);
-                this.blit(matrixStack, $$1 + 99, $$2.intValue(), 129, 0, 27, 44);
-                this.blit(matrixStack, $$1 + 99 + 26, $$2.intValue(), 126, 0, 3, 44);
-                this.blit(matrixStack, $$1 + 99 + 26 + 3, $$2.intValue(), 99, 0, 26, 44);
-                this.blit(matrixStack, $$1 + 155, $$2.intValue(), 0, 45, 155, 44);
+            this.blitOutlineBlack($$6, 30, ($$1, $$2) -> {
+                this.blit($$0, $$1 + 0, $$2.intValue(), 0, 0, 99, 44);
+                this.blit($$0, $$1 + 99, $$2.intValue(), 129, 0, 27, 44);
+                this.blit($$0, $$1 + 99 + 26, $$2.intValue(), 126, 0, 3, 44);
+                this.blit($$0, $$1 + 99 + 26 + 3, $$2.intValue(), 99, 0, 26, 44);
+                this.blit($$0, $$1 + 155, $$2.intValue(), 0, 45, 155, 44);
             });
         } else {
-            this.blitOutlineBlack(n2, 30, ($$1, $$2) -> {
-                this.blit(matrixStack, $$1 + 0, $$2.intValue(), 0, 0, 155, 44);
-                this.blit(matrixStack, $$1 + 155, $$2.intValue(), 0, 45, 155, 44);
+            this.blitOutlineBlack($$6, 30, ($$1, $$2) -> {
+                this.blit($$0, $$1 + 0, $$2.intValue(), 0, 0, 155, 44);
+                this.blit($$0, $$1 + 155, $$2.intValue(), 0, 45, 155, 44);
             });
         }
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)MINECRAFT_EDITION);
-        net.minecraft.client.gui.screens.TitleScreen.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)(n2 + 88), (int)67, (float)0.0f, (float)0.0f, (int)98, (int)14, (int)128, (int)16);
+        net.minecraft.client.gui.screens.TitleScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$6 + 88), (int)67, (float)0.0f, (float)0.0f, (int)98, (int)14, (int)128, (int)16);
         if (this.warningLabel != null) {
-            this.warningLabel.render(matrixStack, n4);
+            this.warningLabel.render($$0, $$9);
         }
         if (this.splash != null) {
-            matrixStack.pushPose();
-            matrixStack.translate((double)(this.width / 2 + 90), 70.0, 0.0);
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(-20.0f));
-            float f3 = 1.8f - net.minecraft.util.Mth.abs((float)(net.minecraft.util.Mth.sin((float)((float)(net.minecraft.Util.getMillis() % 1000L) / 1000.0f * ((float)java.lang.Math.PI * 2))) * 0.1f));
-            f3 = f3 * 100.0f / (float)(this.font.width(this.splash) + 32);
-            matrixStack.scale(f3, f3, f3);
-            net.minecraft.client.gui.screens.TitleScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (java.lang.String)this.splash, (int)0, (int)-8, (int)(0xFFFF00 | n4));
-            matrixStack.popPose();
+            $$0.pushPose();
+            $$0.translate((double)(this.width / 2 + 90), 70.0, 0.0);
+            $$0.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(-20.0f));
+            float $$10 = 1.8f - net.minecraft.util.Mth.abs((float)(net.minecraft.util.Mth.sin((float)((float)(net.minecraft.Util.getMillis() % 1000L) / 1000.0f * ((float)java.lang.Math.PI * 2))) * 0.1f));
+            $$10 = $$10 * 100.0f / (float)(this.font.width(this.splash) + 32);
+            $$0.scale($$10, $$10, $$10);
+            net.minecraft.client.gui.screens.TitleScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (java.lang.String)this.splash, (int)0, (int)-8, (int)(0xFFFF00 | $$9));
+            $$0.popPose();
         }
-        java.lang.String string = "Minecraft " + net.minecraft.SharedConstants.getCurrentVersion().getName();
-        string = this.minecraft.isDemo() ? string + " Demo" : string + ("release".equalsIgnoreCase(this.minecraft.getVersionType()) ? "" : "/" + this.minecraft.getVersionType());
+        java.lang.String $$11 = "Minecraft " + net.minecraft.SharedConstants.getCurrentVersion().getName();
+        $$11 = this.minecraft.isDemo() ? $$11 + " Demo" : $$11 + ("release".equalsIgnoreCase(this.minecraft.getVersionType()) ? "" : "/" + this.minecraft.getVersionType());
         if (net.minecraft.client.Minecraft.checkModStatus().shouldReportAsModified()) {
-            string = string + net.minecraft.client.resources.language.I18n.get((java.lang.String)"menu.modded", (java.lang.Object[])new java.lang.Object[0]);
+            $$11 = $$11 + net.minecraft.client.resources.language.I18n.get((java.lang.String)"menu.modded", (java.lang.Object[])new java.lang.Object[0]);
         }
-        net.minecraft.client.gui.screens.TitleScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (java.lang.String)string, (int)2, (int)(this.height - 10), (int)(0xFFFFFF | n4));
-        for (net.minecraft.client.gui.components.events.GuiEventListener guiEventListener : this.children()) {
-            if (!(guiEventListener instanceof net.minecraft.client.gui.components.AbstractWidget)) continue;
-            ((net.minecraft.client.gui.components.AbstractWidget)guiEventListener).setAlpha(f2);
+        net.minecraft.client.gui.screens.TitleScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (java.lang.String)$$11, (int)2, (int)(this.height - 10), (int)(0xFFFFFF | $$9));
+        for (net.minecraft.client.gui.components.events.GuiEventListener $$122 : this.children()) {
+            if (!($$122 instanceof net.minecraft.client.gui.components.AbstractWidget)) continue;
+            ((net.minecraft.client.gui.components.AbstractWidget)$$122).setAlpha($$8);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        if (this.realmsNotificationsEnabled() && f2 >= 1.0f) {
+        super.render($$0, $$12, $$22, $$3);
+        if (this.realmsNotificationsEnabled() && $$8 >= 1.0f) {
             com.mojang.blaze3d.systems.RenderSystem.enableDepthTest();
-            this.realmsNotificationsScreen.render(matrixStack, mouseX, mouseY, partialTicks);
+            this.realmsNotificationsScreen.render($$0, $$12, $$22, $$3);
         }
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (super.mouseClicked(mouseX, mouseY, button)) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if (super.mouseClicked($$0, $$1, $$2)) {
             return true;
         }
-        return this.realmsNotificationsEnabled() && this.realmsNotificationsScreen.mouseClicked(mouseX, mouseY, button);
+        return this.realmsNotificationsEnabled() && this.realmsNotificationsScreen.mouseClicked($$0, $$1, $$2);
     }
 
     public void removed() {