--- net/minecraft/client/gui/screens/VideoSettingsScreen.java
+++ net/minecraft/client/gui/screens/VideoSettingsScreen.java
@@ -19,14 +19,14 @@
         return new net.minecraft.client.OptionInstance[]{$$0.graphicsMode(), $$0.renderDistance(), $$0.prioritizeChunkUpdates(), $$0.simulationDistance(), $$0.ambientOcclusion(), $$0.framerateLimit(), $$0.enableVsync(), $$0.bobView(), $$0.guiScale(), $$0.attackIndicator(), $$0.gamma(), $$0.cloudStatus(), $$0.fullscreen(), $$0.particles(), $$0.mipmapLevels(), $$0.entityShadows(), $$0.screenEffectScale(), $$0.entityDistanceScaling(), $$0.fovEffectScale(), $$0.showAutosaveIndicator()};
     }
 
-    public net.minecraft.client.gui.screens.VideoSettingsScreen(net.minecraft.client.gui.screens.Screen lastScreen, net.minecraft.client.Options options) {
-        super(lastScreen, options, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.videoTitle"));
-        this.gpuWarnlistManager = lastScreen.minecraft.getGpuWarnlistManager();
+    public net.minecraft.client.gui.screens.VideoSettingsScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.client.Options $$1) {
+        super($$0, $$1, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.videoTitle"));
+        this.gpuWarnlistManager = $$0.minecraft.getGpuWarnlistManager();
         this.gpuWarnlistManager.resetWarnings();
-        if (options.graphicsMode().get() == net.minecraft.client.GraphicsStatus.FABULOUS) {
+        if ($$1.graphicsMode().get() == net.minecraft.client.GraphicsStatus.FABULOUS) {
             this.gpuWarnlistManager.dismissWarning();
         }
-        this.oldMipmaps = (java.lang.Integer)options.mipmapLevels().get();
+        this.oldMipmaps = (java.lang.Integer)$$1.mipmapLevels().get();
     }
 
     public void init() {
@@ -74,30 +74,30 @@
         super.removed();
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        int n = (java.lang.Integer)this.options.guiScale().get();
-        if (super.mouseClicked(mouseX, mouseY, button)) {
-            if ((java.lang.Integer)this.options.guiScale().get() != n) {
+    public boolean mouseClicked(double $$02, double $$1, int $$2) {
+        int $$3 = (java.lang.Integer)this.options.guiScale().get();
+        if (super.mouseClicked($$02, $$1, $$2)) {
+            if ((java.lang.Integer)this.options.guiScale().get() != $$3) {
                 this.minecraft.resizeDisplay();
             }
             if (this.gpuWarnlistManager.isShowingWarning()) {
-                java.lang.String string;
-                java.lang.String string2;
-                java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList((java.lang.Object[])new net.minecraft.network.chat.Component[]{WARNING_MESSAGE, net.minecraft.network.chat.CommonComponents.NEW_LINE});
-                java.lang.String string3 = this.gpuWarnlistManager.getRendererWarnings();
-                if (string3 != null) {
-                    arrayList.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.NEW_LINE);
-                    arrayList.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.warning.renderer", (java.lang.Object[])new java.lang.Object[]{string3}).withStyle(net.minecraft.ChatFormatting.GRAY));
+                java.lang.String $$7;
+                java.lang.String $$6;
+                java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayList((java.lang.Object[])new net.minecraft.network.chat.Component[]{WARNING_MESSAGE, net.minecraft.network.chat.CommonComponents.NEW_LINE});
+                java.lang.String $$5 = this.gpuWarnlistManager.getRendererWarnings();
+                if ($$5 != null) {
+                    $$4.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.NEW_LINE);
+                    $$4.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.warning.renderer", (java.lang.Object[])new java.lang.Object[]{$$5}).withStyle(net.minecraft.ChatFormatting.GRAY));
                 }
-                if ((string2 = this.gpuWarnlistManager.getVendorWarnings()) != null) {
-                    arrayList.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.NEW_LINE);
-                    arrayList.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.warning.vendor", (java.lang.Object[])new java.lang.Object[]{string2}).withStyle(net.minecraft.ChatFormatting.GRAY));
+                if (($$6 = this.gpuWarnlistManager.getVendorWarnings()) != null) {
+                    $$4.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.NEW_LINE);
+                    $$4.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.warning.vendor", (java.lang.Object[])new java.lang.Object[]{$$6}).withStyle(net.minecraft.ChatFormatting.GRAY));
                 }
-                if ((string = this.gpuWarnlistManager.getVersionWarnings()) != null) {
-                    arrayList.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.NEW_LINE);
-                    arrayList.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.warning.version", (java.lang.Object[])new java.lang.Object[]{string}).withStyle(net.minecraft.ChatFormatting.GRAY));
+                if (($$7 = this.gpuWarnlistManager.getVersionWarnings()) != null) {
+                    $$4.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.NEW_LINE);
+                    $$4.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.warning.version", (java.lang.Object[])new java.lang.Object[]{$$7}).withStyle(net.minecraft.ChatFormatting.GRAY));
                 }
-                this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.PopupScreen(WARNING_TITLE, (java.util.List)arrayList, com.google.common.collect.ImmutableList.of((java.lang.Object)new net.minecraft.client.gui.screens.PopupScreen$ButtonOption(BUTTON_ACCEPT, $$0 -> {
+                this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.PopupScreen(WARNING_TITLE, (java.util.List)$$4, com.google.common.collect.ImmutableList.of((java.lang.Object)new net.minecraft.client.gui.screens.PopupScreen$ButtonOption(BUTTON_ACCEPT, $$0 -> {
                     this.options.graphicsMode().set((java.lang.Object)net.minecraft.client.GraphicsStatus.FABULOUS);
                     net.minecraft.client.Minecraft.getInstance().levelRenderer.allChanged();
                     this.gpuWarnlistManager.dismissWarning();
@@ -112,29 +112,10 @@
         return false;
     }
 
-    public boolean mouseReleased(double mouseX, double mouseY, int button) {
-        int n = (java.lang.Integer)this.options.guiScale().get();
-        if (super.mouseReleased(mouseX, mouseY, button)) {
-            return true;
-        }
-        if (this.list.mouseReleased(mouseX, mouseY, button)) {
-            if ((java.lang.Integer)this.options.guiScale().get() != n) {
-                this.minecraft.resizeDisplay();
-            }
-            return true;
-        }
-        return false;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.basicListRender($$0, this.list, $$1, $$2, $$3);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        this.list.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.VideoSettingsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)5, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        java.util.List list = net.minecraft.client.gui.screens.VideoSettingsScreen.tooltipAt((net.minecraft.client.gui.components.OptionsList)this.list, (int)mouseX, (int)mouseY);
-        this.renderTooltip(matrixStack, list, mouseX, mouseY);
-    }
-
     static {
         FABULOUS = net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.fabulous").withStyle(net.minecraft.ChatFormatting.ITALIC);
         WARNING_MESSAGE = net.minecraft.network.chat.Component.translatable((java.lang.String)"options.graphics.warning.message", (java.lang.Object[])new java.lang.Object[]{FABULOUS, FABULOUS});