--- net/minecraft/world/entity/ai/goal/target/HurtByTargetGoal.java
+++ net/minecraft/world/entity/ai/goal/target/HurtByTargetGoal.java
@@ -36,9 +36,9 @@
         return this.canAttack($$1, HURT_BY_TARGETING);
     }
 
-    public net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal setAlertOthers(java.lang.Class<?> ... reinforcementTypes) {
+    public net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal setAlertOthers(java.lang.Class<?> ... $$0) {
         this.alertSameType = true;
-        this.toIgnoreAlert = reinforcementTypes;
+        this.toIgnoreAlert = $$0;
         return this;
     }
 
@@ -72,8 +72,8 @@
         }
     }
 
-    public void alertOther(net.minecraft.world.entity.Mob mob, net.minecraft.world.entity.LivingEntity target) {
-        mob.setTarget(target);
+    public void alertOther(net.minecraft.world.entity.Mob $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        $$0.setTarget($$1);
     }
 
     static {