--- net/minecraft/world/entity/ai/goal/target/NearestAttackableWitchTargetGoal.java
+++ net/minecraft/world/entity/ai/goal/target/NearestAttackableWitchTargetGoal.java
@@ -13,8 +13,8 @@
         this.canAttack = true;
     }
 
-    public void setCanAttack(boolean active) {
-        this.canAttack = active;
+    public void setCanAttack(boolean $$0) {
+        this.canAttack = $$0;
     }
 
     public boolean canUse() {