--- net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
+++ net/minecraft/world/entity/ai/goal/target/NearestAttackableTargetGoal.java
@@ -41,8 +41,8 @@
         return this.target != null;
     }
 
-    public net.minecraft.world.phys.AABB getTargetSearchArea(double targetDistance) {
-        return this.mob.getBoundingBox().inflate(targetDistance, 4.0, targetDistance);
+    public net.minecraft.world.phys.AABB getTargetSearchArea(double $$0) {
+        return this.mob.getBoundingBox().inflate($$0, 4.0, $$0);
     }
 
     public void findTarget() {
@@ -54,8 +54,8 @@
         super.start();
     }
 
-    public void setTarget(@javax.annotation.Nullable net.minecraft.world.entity.LivingEntity target) {
-        this.target = target;
+    public void setTarget(@javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$0) {
+        this.target = $$0;
     }
 }
 