--- net/minecraft/world/entity/ai/goal/target/TargetGoal.java
+++ net/minecraft/world/entity/ai/goal/target/TargetGoal.java
@@ -76,14 +76,14 @@
         this.targetMob = null;
     }
 
-    public boolean canAttack(@javax.annotation.Nullable net.minecraft.world.entity.LivingEntity potentialTarget, net.minecraft.world.entity.ai.targeting.TargetingConditions targetPredicate) {
-        if (potentialTarget == null) {
+    public boolean canAttack(@javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$0, net.minecraft.world.entity.ai.targeting.TargetingConditions $$1) {
+        if ($$0 == null) {
             return false;
         }
-        if (!targetPredicate.test((net.minecraft.world.entity.LivingEntity)this.mob, potentialTarget)) {
+        if (!$$1.test((net.minecraft.world.entity.LivingEntity)this.mob, $$0)) {
             return false;
         }
-        if (!this.mob.isWithinRestriction(potentialTarget.blockPosition())) {
+        if (!this.mob.isWithinRestriction($$0.blockPosition())) {
             return false;
         }
         if (this.mustReach) {
@@ -91,7 +91,7 @@
                 this.reachCache = 0;
             }
             if (this.reachCache == 0) {
-                int n = this.reachCache = this.canReach(potentialTarget) ? 1 : 2;
+                int n = this.reachCache = this.canReach($$0) ? 1 : 2;
             }
             if (this.reachCache == 2) {
                 return false;
@@ -100,23 +100,23 @@
         return true;
     }
 
-    public boolean canReach(net.minecraft.world.entity.LivingEntity target) {
-        int n;
+    public boolean canReach(net.minecraft.world.entity.LivingEntity $$0) {
+        int $$4;
         this.reachCacheTime = net.minecraft.world.entity.ai.goal.target.TargetGoal.reducedTickDelay((int)(10 + this.mob.getRandom().nextInt(5)));
-        net.minecraft.world.level.pathfinder.Path path = this.mob.getNavigation().createPath((net.minecraft.world.entity.Entity)target, 0);
-        if (path == null) {
+        net.minecraft.world.level.pathfinder.Path $$1 = this.mob.getNavigation().createPath((net.minecraft.world.entity.Entity)$$0, 0);
+        if ($$1 == null) {
             return false;
         }
-        net.minecraft.world.level.pathfinder.Node node = path.getEndNode();
-        if (node == null) {
+        net.minecraft.world.level.pathfinder.Node $$2 = $$1.getEndNode();
+        if ($$2 == null) {
             return false;
         }
-        int n2 = node.x - target.getBlockX();
-        return (double)(n2 * n2 + (n = node.z - target.getBlockZ()) * n) <= 2.25;
+        int $$3 = $$2.x - $$0.getBlockX();
+        return (double)($$3 * $$3 + ($$4 = $$2.z - $$0.getBlockZ()) * $$4) <= 2.25;
     }
 
-    public net.minecraft.world.entity.ai.goal.target.TargetGoal setUnseenMemoryTicks(int unseenMemoryTicks) {
-        this.unseenMemoryTicks = unseenMemoryTicks;
+    public net.minecraft.world.entity.ai.goal.target.TargetGoal setUnseenMemoryTicks(int $$0) {
+        this.unseenMemoryTicks = $$0;
         return this;
     }
 }