--- net/minecraft/client/gui/screens/inventory/AbstractFurnaceScreen.java
+++ net/minecraft/client/gui/screens/inventory/AbstractFurnaceScreen.java
@@ -12,10 +12,10 @@
     public boolean widthTooNarrow;
     public net.minecraft.resources.ResourceLocation texture;
 
-    public net.minecraft.client.gui.screens.inventory.AbstractFurnaceScreen(T abstractFurnaceMenu, net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent recipeBookComponent, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title, net.minecraft.resources.ResourceLocation texture) {
-        super(abstractFurnaceMenu, playerInventory, title);
-        this.recipeBookComponent = recipeBookComponent;
-        this.texture = texture;
+    public net.minecraft.client.gui.screens.inventory.AbstractFurnaceScreen(T $$0, net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent $$1, net.minecraft.world.entity.player.Inventory $$2, net.minecraft.network.chat.Component $$3, net.minecraft.resources.ResourceLocation $$4) {
+        super($$0, $$2, $$3);
+        this.recipeBookComponent = $$1;
+        this.texture = $$4;
     }
 
     public void init() {
@@ -36,68 +36,67 @@
         this.recipeBookComponent.tick();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
         if (this.recipeBookComponent.isVisible() && this.widthTooNarrow) {
-            this.renderBg(matrixStack, partialTicks, mouseX, mouseY);
-            this.recipeBookComponent.render(matrixStack, mouseX, mouseY, partialTicks);
+            this.renderBg($$0, $$3, $$1, $$2);
+            this.recipeBookComponent.render($$0, $$1, $$2, $$3);
         } else {
-            this.recipeBookComponent.render(matrixStack, mouseX, mouseY, partialTicks);
-            super.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.recipeBookComponent.renderGhostRecipe(matrixStack, this.leftPos, this.topPos, true, partialTicks);
+            this.recipeBookComponent.render($$0, $$1, $$2, $$3);
+            super.render($$0, $$1, $$2, $$3);
+            this.recipeBookComponent.renderGhostRecipe($$0, this.leftPos, this.topPos, true, $$3);
         }
-        this.renderTooltip(matrixStack, mouseX, mouseY);
-        this.recipeBookComponent.renderTooltip(matrixStack, this.leftPos, this.topPos, mouseX, mouseY);
+        this.renderTooltip($$0, $$1, $$2);
+        this.recipeBookComponent.renderTooltip($$0, this.leftPos, this.topPos, $$1, $$2);
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
-        int n;
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)this.texture);
-        int n2 = this.leftPos;
-        int n3 = this.topPos;
-        this.blit(matrixStack, n2, n3, 0, 0, this.imageWidth, this.imageHeight);
+        int $$4 = this.leftPos;
+        int $$5 = this.topPos;
+        this.blit($$0, $$4, $$5, 0, 0, this.imageWidth, this.imageHeight);
         if (((net.minecraft.world.inventory.AbstractFurnaceMenu)this.menu).isLit()) {
-            n = ((net.minecraft.world.inventory.AbstractFurnaceMenu)this.menu).getLitProgress();
-            this.blit(matrixStack, n2 + 56, n3 + 36 + 12 - n, 176, 12 - n, 14, n + 1);
+            int $$6 = ((net.minecraft.world.inventory.AbstractFurnaceMenu)this.menu).getLitProgress();
+            this.blit($$0, $$4 + 56, $$5 + 36 + 12 - $$6, 176, 12 - $$6, 14, $$6 + 1);
         }
-        n = ((net.minecraft.world.inventory.AbstractFurnaceMenu)this.menu).getBurnProgress();
-        this.blit(matrixStack, n2 + 79, n3 + 34, 176, 14, n + 1, 16);
+        int $$7 = ((net.minecraft.world.inventory.AbstractFurnaceMenu)this.menu).getBurnProgress();
+        this.blit($$0, $$4 + 79, $$5 + 34, 176, 14, $$7 + 1, 16);
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (this.recipeBookComponent.mouseClicked(mouseX, mouseY, button)) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if (this.recipeBookComponent.mouseClicked($$0, $$1, $$2)) {
             return true;
         }
         if (this.widthTooNarrow && this.recipeBookComponent.isVisible()) {
             return true;
         }
-        return super.mouseClicked(mouseX, mouseY, button);
+        return super.mouseClicked($$0, $$1, $$2);
     }
 
-    public void slotClicked(net.minecraft.world.inventory.Slot slot, int slotId, int mouseButton, net.minecraft.world.inventory.ClickType type) {
-        super.slotClicked(slot, slotId, mouseButton, type);
-        this.recipeBookComponent.slotClicked(slot);
+    public void slotClicked(net.minecraft.world.inventory.Slot $$0, int $$1, int $$2, net.minecraft.world.inventory.ClickType $$3) {
+        super.slotClicked($$0, $$1, $$2, $$3);
+        this.recipeBookComponent.slotClicked($$0);
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (this.recipeBookComponent.keyPressed(keyCode, scanCode, modifiers)) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if (this.recipeBookComponent.keyPressed($$0, $$1, $$2)) {
             return false;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
-    public boolean hasClickedOutside(double mouseX, double mouseY, int guiLeft, int guiTop, int mouseButton) {
-        boolean bl = mouseX < (double)guiLeft || mouseY < (double)guiTop || mouseX >= (double)(guiLeft + this.imageWidth) || mouseY >= (double)(guiTop + this.imageHeight);
-        return this.recipeBookComponent.hasClickedOutside(mouseX, mouseY, this.leftPos, this.topPos, this.imageWidth, this.imageHeight, mouseButton) && bl;
+    public boolean hasClickedOutside(double $$0, double $$1, int $$2, int $$3, int $$4) {
+        boolean $$5 = $$0 < (double)$$2 || $$1 < (double)$$3 || $$0 >= (double)($$2 + this.imageWidth) || $$1 >= (double)($$3 + this.imageHeight);
+        return this.recipeBookComponent.hasClickedOutside($$0, $$1, this.leftPos, this.topPos, this.imageWidth, this.imageHeight, $$4) && $$5;
     }
 
-    public boolean charTyped(char codePoint, int modifiers) {
-        if (this.recipeBookComponent.charTyped(codePoint, modifiers)) {
+    public boolean charTyped(char $$0, int $$1) {
+        if (this.recipeBookComponent.charTyped($$0, $$1)) {
             return true;
         }
-        return super.charTyped(codePoint, modifiers);
+        return super.charTyped($$0, $$1);
     }
 
     public void recipesUpdated() {
@@ -108,11 +107,6 @@
         return this.recipeBookComponent;
     }
 
-    public void removed() {
-        this.recipeBookComponent.removed();
-        super.removed();
-    }
-
     static {
         RECIPE_BUTTON_LOCATION = new net.minecraft.resources.ResourceLocation("textures/gui/recipe_button.png");
     }