--- net/minecraft/client/gui/screens/inventory/BeaconScreen.java
+++ net/minecraft/client/gui/screens/inventory/BeaconScreen.java
@@ -15,12 +15,12 @@
     @javax.annotation.Nullable
     public net.minecraft.world.effect.MobEffect secondary;
 
-    public net.minecraft.client.gui.screens.inventory.BeaconScreen(net.minecraft.world.inventory.BeaconMenu beaconMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractContainerMenu)beaconMenu, playerInventory, title);
+    public net.minecraft.client.gui.screens.inventory.BeaconScreen(net.minecraft.world.inventory.BeaconMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractContainerMenu)$$0, $$1, $$2);
         this.beaconButtons = com.google.common.collect.Lists.newArrayList();
         this.imageWidth = 230;
         this.imageHeight = 219;
-        beaconMenu.addSlotListener((net.minecraft.world.inventory.ContainerListener)new /* Unavailable Anonymous Inner Class!! */);
+        $$0.addSlotListener((net.minecraft.world.inventory.ContainerListener)new /* Unavailable Anonymous Inner Class!! */);
     }
 
     public <T extends net.minecraft.client.gui.components.AbstractWidget> void addBeaconButton(T $$0) {
@@ -67,36 +67,36 @@
         this.beaconButtons.forEach($$1 -> $$1.updateStatus($$0));
     }
 
-    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack matrixStack, int x, int y) {
-        net.minecraft.client.gui.screens.inventory.BeaconScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)PRIMARY_EFFECT_LABEL, (int)62, (int)10, (int)0xE0E0E0);
-        net.minecraft.client.gui.screens.inventory.BeaconScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SECONDARY_EFFECT_LABEL, (int)169, (int)10, (int)0xE0E0E0);
-        for (net.minecraft.client.gui.screens.inventory.BeaconScreen$BeaconButton beaconButton : this.beaconButtons) {
-            if (!beaconButton.isShowingTooltip()) continue;
-            beaconButton.renderToolTip(matrixStack, x - this.leftPos, y - this.topPos);
+    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
+        net.minecraft.client.gui.screens.inventory.BeaconScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)PRIMARY_EFFECT_LABEL, (int)62, (int)10, (int)0xE0E0E0);
+        net.minecraft.client.gui.screens.inventory.BeaconScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SECONDARY_EFFECT_LABEL, (int)169, (int)10, (int)0xE0E0E0);
+        for (net.minecraft.client.gui.screens.inventory.BeaconScreen$BeaconButton $$3 : this.beaconButtons) {
+            if (!$$3.isShowingTooltip()) continue;
+            $$3.renderToolTip($$0, $$1 - this.leftPos, $$2 - this.topPos);
             break;
         }
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)BEACON_LOCATION);
-        int n = (this.width - this.imageWidth) / 2;
-        int n2 = (this.height - this.imageHeight) / 2;
-        this.blit(matrixStack, n, n2, 0, 0, this.imageWidth, this.imageHeight);
+        int $$4 = (this.width - this.imageWidth) / 2;
+        int $$5 = (this.height - this.imageHeight) / 2;
+        this.blit($$0, $$4, $$5, 0, 0, this.imageWidth, this.imageHeight);
         this.itemRenderer.blitOffset = 100.0f;
-        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NETHERITE_INGOT), n + 20, n2 + 109);
-        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.EMERALD), n + 41, n2 + 109);
-        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DIAMOND), n + 41 + 22, n2 + 109);
-        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLD_INGOT), n + 42 + 44, n2 + 109);
-        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_INGOT), n + 42 + 66, n2 + 109);
+        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NETHERITE_INGOT), $$4 + 20, $$5 + 109);
+        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.EMERALD), $$4 + 41, $$5 + 109);
+        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DIAMOND), $$4 + 41 + 22, $$5 + 109);
+        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLD_INGOT), $$4 + 42 + 44, $$5 + 109);
+        this.itemRenderer.renderAndDecorateItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_INGOT), $$4 + 42 + 66, $$5 + 109);
         this.itemRenderer.blitOffset = 0.0f;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderTooltip(matrixStack, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        super.render($$0, $$1, $$2, $$3);
+        this.renderTooltip($$0, $$1, $$2);
     }
 
     public static /* synthetic */ net.minecraft.network.chat.Component access$000(net.minecraft.client.gui.screens.inventory.BeaconScreen $$0) {