--- net/minecraft/client/gui/screens/inventory/BeaconScreen$1.java
+++ net/minecraft/client/gui/screens/inventory/BeaconScreen$1.java
@@ -14,7 +14,7 @@
         this.val$menu = beaconMenu;
     }
 
-    public void slotChanged(net.minecraft.world.inventory.AbstractContainerMenu containerToSend, int slotInd, net.minecraft.world.item.ItemStack stack) {
+    public void slotChanged(net.minecraft.world.inventory.AbstractContainerMenu $$0, int $$1, net.minecraft.world.item.ItemStack $$2) {
     }
 
     public void dataChanged(net.minecraft.world.inventory.AbstractContainerMenu $$0, int $$1, int $$2) {