--- net/minecraft/client/gui/screens/inventory/BeaconScreen$BeaconPowerButton.java
+++ net/minecraft/client/gui/screens/inventory/BeaconScreen$BeaconPowerButton.java
@@ -43,13 +43,13 @@
         this.this$0.updateButtons();
     }
 
-    public void renderToolTip(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY) {
-        this.this$0.renderTooltip(matrixStack, this.tooltip, mouseX, mouseY);
+    public void renderToolTip(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
+        this.this$0.renderTooltip($$0, this.tooltip, $$1, $$2);
     }
 
-    public void renderIcon(com.mojang.blaze3d.vertex.PoseStack poseStack) {
-        com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)this.sprite.atlas().location());
-        net.minecraft.client.gui.screens.inventory.BeaconScreen$BeaconPowerButton.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(this.x + 2), (int)(this.y + 2), (int)this.getBlitOffset(), (int)18, (int)18, (net.minecraft.client.renderer.texture.TextureAtlasSprite)this.sprite);
+    public void renderIcon(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)this.sprite.atlasLocation());
+        net.minecraft.client.gui.screens.inventory.BeaconScreen$BeaconPowerButton.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)(this.x + 2), (int)(this.y + 2), (int)this.getBlitOffset(), (int)18, (int)18, (net.minecraft.client.renderer.texture.TextureAtlasSprite)this.sprite);
     }
 
     public void updateStatus(int $$0) {