--- net/minecraft/client/gui/screens/inventory/BeaconScreen$BeaconConfirmButton.java
+++ net/minecraft/client/gui/screens/inventory/BeaconScreen$BeaconConfirmButton.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.client.gui.screens.inventory.BeaconScreen this$0;
 
-    public net.minecraft.client.gui.screens.inventory.BeaconScreen$BeaconConfirmButton(net.minecraft.client.gui.screens.inventory.BeaconScreen beaconScreen, int x, int y) {
+    public net.minecraft.client.gui.screens.inventory.BeaconScreen$BeaconConfirmButton(net.minecraft.client.gui.screens.inventory.BeaconScreen beaconScreen, int $$0, int $$1) {
         this.this$0 = beaconScreen;
-        super(beaconScreen, x, y, 90, 220, net.minecraft.network.chat.CommonComponents.GUI_DONE);
+        super(beaconScreen, $$0, $$1, 90, 220, net.minecraft.network.chat.CommonComponents.GUI_DONE);
     }
 
     public void onPress() {