--- net/minecraft/client/gui/screens/inventory/BlastFurnaceScreen.java
+++ net/minecraft/client/gui/screens/inventory/BlastFurnaceScreen.java
@@ -8,8 +8,8 @@
  {
     public static net.minecraft.resources.ResourceLocation TEXTURE;
 
-    public net.minecraft.client.gui.screens.inventory.BlastFurnaceScreen(net.minecraft.world.inventory.BlastFurnaceMenu blastFurnaceMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractFurnaceMenu)blastFurnaceMenu, (net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent)new net.minecraft.client.gui.screens.recipebook.BlastingRecipeBookComponent(), playerInventory, title, TEXTURE);
+    public net.minecraft.client.gui.screens.inventory.BlastFurnaceScreen(net.minecraft.world.inventory.BlastFurnaceMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractFurnaceMenu)$$0, (net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent)new net.minecraft.client.gui.screens.recipebook.BlastingRecipeBookComponent(), $$1, $$2, TEXTURE);
     }
 
     static {