--- net/minecraft/client/gui/screens/inventory/BookEditScreen$Pos2i.java
+++ net/minecraft/client/gui/screens/inventory/BookEditScreen$Pos2i.java
@@ -8,9 +8,9 @@
     public int x;
     public int y;
 
-    public net.minecraft.client.gui.screens.inventory.BookEditScreen$Pos2i(int x, int y) {
-        this.x = x;
-        this.y = y;
+    public net.minecraft.client.gui.screens.inventory.BookEditScreen$Pos2i(int $$0, int $$1) {
+        this.x = $$0;
+        this.y = $$1;
     }
 }
 