--- net/minecraft/client/gui/screens/inventory/BookEditScreen$LineInfo.java
+++ net/minecraft/client/gui/screens/inventory/BookEditScreen$LineInfo.java
@@ -11,12 +11,12 @@
     public int x;
     public int y;
 
-    public net.minecraft.client.gui.screens.inventory.BookEditScreen$LineInfo(net.minecraft.network.chat.Style style, java.lang.String contents, int x, int y) {
-        this.style = style;
-        this.contents = contents;
-        this.x = x;
-        this.y = y;
-        this.asComponent = net.minecraft.network.chat.Component.literal((java.lang.String)contents).setStyle(style);
+    public net.minecraft.client.gui.screens.inventory.BookEditScreen$LineInfo(net.minecraft.network.chat.Style $$0, java.lang.String $$1, int $$2, int $$3) {
+        this.style = $$0;
+        this.contents = $$1;
+        this.x = $$2;
+        this.y = $$3;
+        this.asComponent = net.minecraft.network.chat.Component.literal((java.lang.String)$$1).setStyle($$0);
     }
 }
 