--- net/minecraft/client/gui/screens/inventory/BrewingStandScreen.java
+++ net/minecraft/client/gui/screens/inventory/BrewingStandScreen.java
@@ -9,8 +9,8 @@
     public static net.minecraft.resources.ResourceLocation BREWING_STAND_LOCATION;
     public static int[] BUBBLELENGTHS;
 
-    public net.minecraft.client.gui.screens.inventory.BrewingStandScreen(net.minecraft.world.inventory.BrewingStandMenu brewingStandMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractContainerMenu)brewingStandMenu, playerInventory, title);
+    public net.minecraft.client.gui.screens.inventory.BrewingStandScreen(net.minecraft.world.inventory.BrewingStandMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractContainerMenu)$$0, $$1, $$2);
     }
 
     public void init() {
@@ -18,32 +18,32 @@
         this.titleLabelX = (this.imageWidth - this.font.width((net.minecraft.network.chat.FormattedText)this.title)) / 2;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderTooltip(matrixStack, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        super.render($$0, $$1, $$2, $$3);
+        this.renderTooltip($$0, $$1, $$2);
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
-        int n;
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
+        int $$8;
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)BREWING_STAND_LOCATION);
-        int n2 = (this.width - this.imageWidth) / 2;
-        int n3 = (this.height - this.imageHeight) / 2;
-        this.blit(matrixStack, n2, n3, 0, 0, this.imageWidth, this.imageHeight);
-        int n4 = ((net.minecraft.world.inventory.BrewingStandMenu)this.menu).getFuel();
-        int n5 = net.minecraft.util.Mth.clamp((int)((18 * n4 + 20 - 1) / 20), (int)0, (int)18);
-        if (n5 > 0) {
-            this.blit(matrixStack, n2 + 60, n3 + 44, 176, 29, n5, 4);
+        int $$4 = (this.width - this.imageWidth) / 2;
+        int $$5 = (this.height - this.imageHeight) / 2;
+        this.blit($$0, $$4, $$5, 0, 0, this.imageWidth, this.imageHeight);
+        int $$6 = ((net.minecraft.world.inventory.BrewingStandMenu)this.menu).getFuel();
+        int $$7 = net.minecraft.util.Mth.clamp((int)((18 * $$6 + 20 - 1) / 20), (int)0, (int)18);
+        if ($$7 > 0) {
+            this.blit($$0, $$4 + 60, $$5 + 44, 176, 29, $$7, 4);
         }
-        if ((n = ((net.minecraft.world.inventory.BrewingStandMenu)this.menu).getBrewingTicks()) > 0) {
-            int n6 = (int)(28.0f * (1.0f - (float)n / 400.0f));
-            if (n6 > 0) {
-                this.blit(matrixStack, n2 + 97, n3 + 16, 176, 0, 9, n6);
+        if (($$8 = ((net.minecraft.world.inventory.BrewingStandMenu)this.menu).getBrewingTicks()) > 0) {
+            int $$9 = (int)(28.0f * (1.0f - (float)$$8 / 400.0f));
+            if ($$9 > 0) {
+                this.blit($$0, $$4 + 97, $$5 + 16, 176, 0, 9, $$9);
             }
-            if ((n6 = BUBBLELENGTHS[n / 2 % 7]) > 0) {
-                this.blit(matrixStack, n2 + 63, n3 + 14 + 29 - n6, 185, 29 - n6, 12, n6);
+            if (($$9 = BUBBLELENGTHS[$$8 / 2 % 7]) > 0) {
+                this.blit($$0, $$4 + 63, $$5 + 14 + 29 - $$9, 185, 29 - $$9, 12, $$9);
             }
         }
     }