--- net/minecraft/client/gui/screens/inventory/CartographyTableScreen.java
+++ net/minecraft/client/gui/screens/inventory/CartographyTableScreen.java
@@ -8,52 +8,52 @@
  {
     public static net.minecraft.resources.ResourceLocation BG_LOCATION;
 
-    public net.minecraft.client.gui.screens.inventory.CartographyTableScreen(net.minecraft.world.inventory.CartographyTableMenu cartographyTableMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractContainerMenu)cartographyTableMenu, playerInventory, title);
+    public net.minecraft.client.gui.screens.inventory.CartographyTableScreen(net.minecraft.world.inventory.CartographyTableMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractContainerMenu)$$0, $$1, $$2);
         this.titleLabelY -= 2;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderTooltip(matrixStack, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        super.render($$0, $$1, $$2, $$3);
+        this.renderTooltip($$0, $$1, $$2);
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
-        net.minecraft.world.level.saveddata.maps.MapItemSavedData mapItemSavedData;
-        java.lang.Integer n;
-        this.renderBackground(matrixStack);
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
+        net.minecraft.world.level.saveddata.maps.MapItemSavedData $$15;
+        java.lang.Integer $$14;
+        this.renderBackground($$0);
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)BG_LOCATION);
-        int n2 = this.leftPos;
-        int n3 = this.topPos;
-        this.blit(matrixStack, n2, n3, 0, 0, this.imageWidth, this.imageHeight);
-        net.minecraft.world.item.ItemStack itemStack = ((net.minecraft.world.inventory.CartographyTableMenu)this.menu).getSlot(1).getItem();
-        boolean bl = itemStack.is(net.minecraft.world.item.Items.MAP);
-        boolean bl2 = itemStack.is(net.minecraft.world.item.Items.PAPER);
-        boolean bl3 = itemStack.is(net.minecraft.world.item.Items.GLASS_PANE);
-        net.minecraft.world.item.ItemStack itemStack2 = ((net.minecraft.world.inventory.CartographyTableMenu)this.menu).getSlot(0).getItem();
-        boolean bl4 = false;
-        if (itemStack2.is(net.minecraft.world.item.Items.FILLED_MAP)) {
-            n = net.minecraft.world.item.MapItem.getMapId((net.minecraft.world.item.ItemStack)itemStack2);
-            mapItemSavedData = net.minecraft.world.item.MapItem.getSavedData((java.lang.Integer)n, (net.minecraft.world.level.Level)this.minecraft.level);
-            if (mapItemSavedData != null) {
-                if (mapItemSavedData.locked) {
-                    bl4 = true;
-                    if (bl2 || bl3) {
-                        this.blit(matrixStack, n2 + 35, n3 + 31, this.imageWidth + 50, 132, 28, 21);
+        int $$4 = this.leftPos;
+        int $$5 = this.topPos;
+        this.blit($$0, $$4, $$5, 0, 0, this.imageWidth, this.imageHeight);
+        net.minecraft.world.item.ItemStack $$6 = ((net.minecraft.world.inventory.CartographyTableMenu)this.menu).getSlot(1).getItem();
+        boolean $$7 = $$6.is(net.minecraft.world.item.Items.MAP);
+        boolean $$8 = $$6.is(net.minecraft.world.item.Items.PAPER);
+        boolean $$9 = $$6.is(net.minecraft.world.item.Items.GLASS_PANE);
+        net.minecraft.world.item.ItemStack $$10 = ((net.minecraft.world.inventory.CartographyTableMenu)this.menu).getSlot(0).getItem();
+        boolean $$11 = false;
+        if ($$10.is(net.minecraft.world.item.Items.FILLED_MAP)) {
+            java.lang.Integer $$12 = net.minecraft.world.item.MapItem.getMapId((net.minecraft.world.item.ItemStack)$$10);
+            net.minecraft.world.level.saveddata.maps.MapItemSavedData $$13 = net.minecraft.world.item.MapItem.getSavedData((java.lang.Integer)$$12, (net.minecraft.world.level.Level)this.minecraft.level);
+            if ($$13 != null) {
+                if ($$13.locked) {
+                    $$11 = true;
+                    if ($$8 || $$9) {
+                        this.blit($$0, $$4 + 35, $$5 + 31, this.imageWidth + 50, 132, 28, 21);
                     }
                 }
-                if (bl2 && mapItemSavedData.scale >= 4) {
-                    bl4 = true;
-                    this.blit(matrixStack, n2 + 35, n3 + 31, this.imageWidth + 50, 132, 28, 21);
+                if ($$8 && $$13.scale >= 4) {
+                    $$11 = true;
+                    this.blit($$0, $$4 + 35, $$5 + 31, this.imageWidth + 50, 132, 28, 21);
                 }
             }
         } else {
-            n = null;
-            mapItemSavedData = null;
+            $$14 = null;
+            $$15 = null;
         }
-        this.renderResultingMap(matrixStack, n, mapItemSavedData, bl, bl2, bl3, bl4);
+        this.renderResultingMap($$0, $$14, $$15, $$7, $$8, $$9, $$11);
     }
 
     public void renderResultingMap(com.mojang.blaze3d.vertex.PoseStack $$0, @javax.annotation.Nullable java.lang.Integer $$1, @javax.annotation.Nullable net.minecraft.world.level.saveddata.maps.MapItemSavedData $$2, boolean $$3, boolean $$4, boolean $$5, boolean $$6) {