--- net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
+++ net/minecraft/client/gui/screens/inventory/CommandBlockEditScreen.java
@@ -14,9 +14,9 @@
     public boolean conditional;
     public boolean autoexec;
 
-    public net.minecraft.client.gui.screens.inventory.CommandBlockEditScreen(net.minecraft.world.level.block.entity.CommandBlockEntity commandBlockEntity) {
+    public net.minecraft.client.gui.screens.inventory.CommandBlockEditScreen(net.minecraft.world.level.block.entity.CommandBlockEntity $$0) {
         this.mode = net.minecraft.world.level.block.entity.CommandBlockEntity$Mode.REDSTONE;
-        this.autoCommandBlock = commandBlockEntity;
+        this.autoCommandBlock = $$0;
     }
 
     public net.minecraft.world.level.BaseCommandBlock getCommandBlock() {
@@ -74,13 +74,13 @@
         this.enableControls(true);
     }
 
-    public void resize(net.minecraft.client.Minecraft minecraft, int width, int height) {
-        super.resize(minecraft, width, height);
+    public void resize(net.minecraft.client.Minecraft $$0, int $$1, int $$2) {
+        super.resize($$0, $$1, $$2);
         this.enableControls(true);
     }
 
-    public void populateAndSendPacket(net.minecraft.world.level.BaseCommandBlock commandBlock) {
-        this.minecraft.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundSetCommandBlockPacket(new net.minecraft.core.BlockPos(commandBlock.getPosition()), this.commandEdit.getValue(), this.mode, commandBlock.isTrackOutput(), this.conditional, this.autoexec));
+    public void populateAndSendPacket(net.minecraft.world.level.BaseCommandBlock $$0) {
+        this.minecraft.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundSetCommandBlockPacket(new net.minecraft.core.BlockPos($$0.getPosition()), this.commandEdit.getValue(), this.mode, $$0.isTrackOutput(), this.conditional, this.autoexec));
     }
 }
 