--- net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen$CustomCreativeSlot.java
+++ net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen$CustomCreativeSlot.java
@@ -10,8 +10,8 @@
         super($$0, $$1, $$2, $$3);
     }
 
-    public boolean mayPickup(net.minecraft.world.entity.player.Player player) {
-        if (super.mayPickup(player) && this.hasItem()) {
+    public boolean mayPickup(net.minecraft.world.entity.player.Player $$0) {
+        if (super.mayPickup($$0) && this.hasItem()) {
             return this.getItem().getTagElement("CustomCreativeLock") == null;
         }
         return !this.hasItem();