--- net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen$SlotWrapper.java
+++ net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen$SlotWrapper.java
@@ -8,17 +8,17 @@
  {
     public net.minecraft.world.inventory.Slot target;
 
-    public net.minecraft.client.gui.screens.inventory.CreativeModeInventoryScreen$SlotWrapper(net.minecraft.world.inventory.Slot slot, int index, int x, int y) {
-        super(slot.container, index, x, y);
-        this.target = slot;
+    public net.minecraft.client.gui.screens.inventory.CreativeModeInventoryScreen$SlotWrapper(net.minecraft.world.inventory.Slot $$0, int $$1, int $$2, int $$3) {
+        super($$0.container, $$1, $$2, $$3);
+        this.target = $$0;
     }
 
     public void onTake(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.item.ItemStack $$1) {
         this.target.onTake($$0, $$1);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return this.target.mayPlace(stack);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return this.target.mayPlace($$0);
     }
 
     public net.minecraft.world.item.ItemStack getItem() {
@@ -29,8 +29,8 @@
         return this.target.hasItem();
     }
 
-    public void set(net.minecraft.world.item.ItemStack stack) {
-        this.target.set(stack);
+    public void set(net.minecraft.world.item.ItemStack $$0) {
+        this.target.set($$0);
     }
 
     public void setChanged() {
@@ -41,8 +41,8 @@
         return this.target.getMaxStackSize();
     }
 
-    public int getMaxStackSize(net.minecraft.world.item.ItemStack stack) {
-        return this.target.getMaxStackSize(stack);
+    public int getMaxStackSize(net.minecraft.world.item.ItemStack $$0) {
+        return this.target.getMaxStackSize($$0);
     }
 
     @javax.annotation.Nullable
@@ -50,16 +50,16 @@
         return this.target.getNoItemIcon();
     }
 
-    public net.minecraft.world.item.ItemStack remove(int amount) {
-        return this.target.remove(amount);
+    public net.minecraft.world.item.ItemStack remove(int $$0) {
+        return this.target.remove($$0);
     }
 
     public boolean isActive() {
         return this.target.isActive();
     }
 
-    public boolean mayPickup(net.minecraft.world.entity.player.Player player) {
-        return this.target.mayPickup(player);
+    public boolean mayPickup(net.minecraft.world.entity.player.Player $$0) {
+        return this.target.mayPickup($$0);
     }
 }
 