--- net/minecraft/client/gui/screens/inventory/CreativeInventoryListener.java
+++ net/minecraft/client/gui/screens/inventory/CreativeInventoryListener.java
@@ -8,12 +8,12 @@
  {
     public net.minecraft.client.Minecraft minecraft;
 
-    public net.minecraft.client.gui.screens.inventory.CreativeInventoryListener(net.minecraft.client.Minecraft minecraft) {
-        this.minecraft = minecraft;
+    public net.minecraft.client.gui.screens.inventory.CreativeInventoryListener(net.minecraft.client.Minecraft $$0) {
+        this.minecraft = $$0;
     }
 
-    public void slotChanged(net.minecraft.world.inventory.AbstractContainerMenu containerToSend, int slotInd, net.minecraft.world.item.ItemStack stack) {
-        this.minecraft.gameMode.handleCreativeModeItemAdd(stack, slotInd);
+    public void slotChanged(net.minecraft.world.inventory.AbstractContainerMenu $$0, int $$1, net.minecraft.world.item.ItemStack $$2) {
+        this.minecraft.gameMode.handleCreativeModeItemAdd($$2, $$1);
     }
 
     public void dataChanged(net.minecraft.world.inventory.AbstractContainerMenu $$0, int $$1, int $$2) {