--- net/minecraft/client/gui/screens/inventory/DispenserScreen.java
+++ net/minecraft/client/gui/screens/inventory/DispenserScreen.java
@@ -8,8 +8,8 @@
  {
     public static net.minecraft.resources.ResourceLocation CONTAINER_LOCATION;
 
-    public net.minecraft.client.gui.screens.inventory.DispenserScreen(net.minecraft.world.inventory.DispenserMenu dispenserMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractContainerMenu)dispenserMenu, playerInventory, title);
+    public net.minecraft.client.gui.screens.inventory.DispenserScreen(net.minecraft.world.inventory.DispenserMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractContainerMenu)$$0, $$1, $$2);
     }
 
     public void init() {
@@ -17,19 +17,19 @@
         this.titleLabelX = (this.imageWidth - this.font.width((net.minecraft.network.chat.FormattedText)this.title)) / 2;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderTooltip(matrixStack, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        super.render($$0, $$1, $$2, $$3);
+        this.renderTooltip($$0, $$1, $$2);
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)CONTAINER_LOCATION);
-        int n = (this.width - this.imageWidth) / 2;
-        int n2 = (this.height - this.imageHeight) / 2;
-        this.blit(matrixStack, n, n2, 0, 0, this.imageWidth, this.imageHeight);
+        int $$4 = (this.width - this.imageWidth) / 2;
+        int $$5 = (this.height - this.imageHeight) / 2;
+        this.blit($$0, $$4, $$5, 0, 0, this.imageWidth, this.imageHeight);
     }
 
     static {