--- net/minecraft/client/gui/screens/inventory/EffectRenderingInventoryScreen.java
+++ net/minecraft/client/gui/screens/inventory/EffectRenderingInventoryScreen.java
@@ -10,9 +10,9 @@
         super($$0, $$1, $$2);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderEffects(matrixStack, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        super.render($$0, $$1, $$2, $$3);
+        this.renderEffects($$0, $$1, $$2);
     }
 
     public boolean canSeeEffects() {
@@ -75,20 +75,20 @@
         for (net.minecraft.world.effect.MobEffectInstance $$7 : $$3) {
             net.minecraft.world.effect.MobEffect $$8 = $$7.getEffect();
             net.minecraft.client.renderer.texture.TextureAtlasSprite $$9 = $$5.get($$8);
-            com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)$$9.atlas().location());
+            com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)$$9.atlasLocation());
             net.minecraft.client.gui.screens.inventory.EffectRenderingInventoryScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + ($$4 ? 6 : 7)), (int)($$6 + 7), (int)this.getBlitOffset(), (int)18, (int)18, (net.minecraft.client.renderer.texture.TextureAtlasSprite)$$9);
             $$6 += $$2;
         }
     }
 
-    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack poseStack, int renderX, int yOffset, java.lang.Iterable<net.minecraft.world.effect.MobEffectInstance> effects) {
-        int n = this.topPos;
-        for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : effects) {
-            net.minecraft.network.chat.Component component = this.getEffectName(mobEffectInstance);
-            this.font.drawShadow(poseStack, component, (float)(renderX + 10 + 18), (float)(n + 6), 0xFFFFFF);
-            java.lang.String string = net.minecraft.world.effect.MobEffectUtil.formatDuration((net.minecraft.world.effect.MobEffectInstance)mobEffectInstance, (float)1.0f);
-            this.font.drawShadow(poseStack, string, (float)(renderX + 10 + 18), (float)(n + 6 + 10), 0x7F7F7F);
-            n += yOffset;
+    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, java.lang.Iterable<net.minecraft.world.effect.MobEffectInstance> $$3) {
+        int $$4 = this.topPos;
+        for (net.minecraft.world.effect.MobEffectInstance $$5 : $$3) {
+            net.minecraft.network.chat.Component $$6 = this.getEffectName($$5);
+            this.font.drawShadow($$0, $$6, (float)($$1 + 10 + 18), (float)($$4 + 6), 0xFFFFFF);
+            java.lang.String $$7 = net.minecraft.world.effect.MobEffectUtil.formatDuration((net.minecraft.world.effect.MobEffectInstance)$$5, (float)1.0f);
+            this.font.drawShadow($$0, $$7, (float)($$1 + 10 + 18), (float)($$4 + 6 + 10), 0x7F7F7F);
+            $$4 += $$2;
         }
     }
 