--- net/minecraft/client/gui/screens/inventory/EnchantmentNames.java
+++ net/minecraft/client/gui/screens/inventory/EnchantmentNames.java
@@ -20,20 +20,20 @@
         return INSTANCE;
     }
 
-    public net.minecraft.network.chat.FormattedText getRandomName(net.minecraft.client.gui.Font fontRenderer, int maxWidth) {
-        java.lang.StringBuilder stringBuilder = new java.lang.StringBuilder();
-        int n = this.random.nextInt(2) + 3;
-        for (int i = 0; i < n; ++i) {
-            if (i != 0) {
-                stringBuilder.append(" ");
+    public net.minecraft.network.chat.FormattedText getRandomName(net.minecraft.client.gui.Font $$0, int $$1) {
+        java.lang.StringBuilder $$2 = new java.lang.StringBuilder();
+        int $$3 = this.random.nextInt(2) + 3;
+        for (int $$4 = 0; $$4 < $$3; ++$$4) {
+            if ($$4 != 0) {
+                $$2.append(" ");
             }
-            stringBuilder.append((java.lang.String)net.minecraft.Util.getRandom((java.lang.Object[])this.words, (net.minecraft.util.RandomSource)this.random));
+            $$2.append((java.lang.String)net.minecraft.Util.getRandom((java.lang.Object[])this.words, (net.minecraft.util.RandomSource)this.random));
         }
-        return fontRenderer.getSplitter().headByWidth((net.minecraft.network.chat.FormattedText)net.minecraft.network.chat.Component.literal((java.lang.String)stringBuilder.toString()).withStyle(ROOT_STYLE), maxWidth, net.minecraft.network.chat.Style.EMPTY);
+        return $$0.getSplitter().headByWidth((net.minecraft.network.chat.FormattedText)net.minecraft.network.chat.Component.literal((java.lang.String)$$2.toString()).withStyle(ROOT_STYLE), $$1, net.minecraft.network.chat.Style.EMPTY);
     }
 
-    public void initSeed(long seed) {
-        this.random.setSeed(seed);
+    public void initSeed(long $$0) {
+        this.random.setSeed($$0);
     }
 
     static {