--- net/minecraft/client/gui/screens/inventory/FurnaceScreen.java
+++ net/minecraft/client/gui/screens/inventory/FurnaceScreen.java
@@ -8,8 +8,8 @@
  {
     public static net.minecraft.resources.ResourceLocation TEXTURE;
 
-    public net.minecraft.client.gui.screens.inventory.FurnaceScreen(net.minecraft.world.inventory.FurnaceMenu furnaceMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractFurnaceMenu)furnaceMenu, (net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent)new net.minecraft.client.gui.screens.recipebook.SmeltingRecipeBookComponent(), playerInventory, title, TEXTURE);
+    public net.minecraft.client.gui.screens.inventory.FurnaceScreen(net.minecraft.world.inventory.FurnaceMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractFurnaceMenu)$$0, (net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent)new net.minecraft.client.gui.screens.recipebook.SmeltingRecipeBookComponent(), $$1, $$2, TEXTURE);
     }
 
     static {