--- net/minecraft/client/gui/screens/inventory/GrindstoneScreen.java
+++ net/minecraft/client/gui/screens/inventory/GrindstoneScreen.java
@@ -8,26 +8,26 @@
  {
     public static net.minecraft.resources.ResourceLocation GRINDSTONE_LOCATION;
 
-    public net.minecraft.client.gui.screens.inventory.GrindstoneScreen(net.minecraft.world.inventory.GrindstoneMenu grindstoneMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractContainerMenu)grindstoneMenu, playerInventory, title);
+    public net.minecraft.client.gui.screens.inventory.GrindstoneScreen(net.minecraft.world.inventory.GrindstoneMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractContainerMenu)$$0, $$1, $$2);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        this.renderBg(matrixStack, partialTicks, mouseX, mouseY);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderTooltip(matrixStack, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        this.renderBg($$0, $$3, $$1, $$2);
+        super.render($$0, $$1, $$2, $$3);
+        this.renderTooltip($$0, $$1, $$2);
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)GRINDSTONE_LOCATION);
-        int n = (this.width - this.imageWidth) / 2;
-        int n2 = (this.height - this.imageHeight) / 2;
-        this.blit(matrixStack, n, n2, 0, 0, this.imageWidth, this.imageHeight);
+        int $$4 = (this.width - this.imageWidth) / 2;
+        int $$5 = (this.height - this.imageHeight) / 2;
+        this.blit($$0, $$4, $$5, 0, 0, this.imageWidth, this.imageHeight);
         if ((((net.minecraft.world.inventory.GrindstoneMenu)this.menu).getSlot(0).hasItem() || ((net.minecraft.world.inventory.GrindstoneMenu)this.menu).getSlot(1).hasItem()) && !((net.minecraft.world.inventory.GrindstoneMenu)this.menu).getSlot(2).hasItem()) {
-            this.blit(matrixStack, n + 92, n2 + 31, this.imageWidth, 0, 28, 21);
+            this.blit($$0, $$4 + 92, $$5 + 31, this.imageWidth, 0, 28, 21);
         }
     }
 