--- net/minecraft/client/gui/screens/inventory/ItemCombinerScreen.java
+++ net/minecraft/client/gui/screens/inventory/ItemCombinerScreen.java
@@ -9,9 +9,9 @@
  {
     public net.minecraft.resources.ResourceLocation menuResource;
 
-    public net.minecraft.client.gui.screens.inventory.ItemCombinerScreen(T itemCombinerMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title, net.minecraft.resources.ResourceLocation menuTexture) {
-        super(itemCombinerMenu, playerInventory, title);
-        this.menuResource = menuTexture;
+    public net.minecraft.client.gui.screens.inventory.ItemCombinerScreen(T $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2, net.minecraft.resources.ResourceLocation $$3) {
+        super($$0, $$1, $$2);
+        this.menuResource = $$3;
     }
 
     public void subInit() {
@@ -28,34 +28,34 @@
         ((net.minecraft.world.inventory.ItemCombinerMenu)this.menu).removeSlotListener((net.minecraft.world.inventory.ContainerListener)this);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        super.render($$0, $$1, $$2, $$3);
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();
-        this.renderFg(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderTooltip(matrixStack, mouseX, mouseY);
+        this.renderFg($$0, $$1, $$2, $$3);
+        this.renderTooltip($$0, $$1, $$2);
     }
 
-    public void renderFg(com.mojang.blaze3d.vertex.PoseStack poseStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderFg(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)this.menuResource);
-        int n = (this.width - this.imageWidth) / 2;
-        int n2 = (this.height - this.imageHeight) / 2;
-        this.blit(matrixStack, n, n2, 0, 0, this.imageWidth, this.imageHeight);
-        this.blit(matrixStack, n + 59, n2 + 20, 0, this.imageHeight + (((net.minecraft.world.inventory.ItemCombinerMenu)this.menu).getSlot(0).hasItem() ? 0 : 16), 110, 16);
+        int $$4 = (this.width - this.imageWidth) / 2;
+        int $$5 = (this.height - this.imageHeight) / 2;
+        this.blit($$0, $$4, $$5, 0, 0, this.imageWidth, this.imageHeight);
+        this.blit($$0, $$4 + 59, $$5 + 20, 0, this.imageHeight + (((net.minecraft.world.inventory.ItemCombinerMenu)this.menu).getSlot(0).hasItem() ? 0 : 16), 110, 16);
         if ((((net.minecraft.world.inventory.ItemCombinerMenu)this.menu).getSlot(0).hasItem() || ((net.minecraft.world.inventory.ItemCombinerMenu)this.menu).getSlot(1).hasItem()) && !((net.minecraft.world.inventory.ItemCombinerMenu)this.menu).getSlot(2).hasItem()) {
-            this.blit(matrixStack, n + 99, n2 + 45, this.imageWidth, 0, 28, 21);
+            this.blit($$0, $$4 + 99, $$5 + 45, this.imageWidth, 0, 28, 21);
         }
     }
 
     public void dataChanged(net.minecraft.world.inventory.AbstractContainerMenu $$0, int $$1, int $$2) {
     }
 
-    public void slotChanged(net.minecraft.world.inventory.AbstractContainerMenu containerToSend, int slotInd, net.minecraft.world.item.ItemStack stack) {
+    public void slotChanged(net.minecraft.world.inventory.AbstractContainerMenu $$0, int $$1, net.minecraft.world.item.ItemStack $$2) {
     }
 }
 