--- net/minecraft/client/gui/screens/inventory/JigsawBlockEditScreen.java
+++ net/minecraft/client/gui/screens/inventory/JigsawBlockEditScreen.java
@@ -24,10 +24,10 @@
     public net.minecraft.client.gui.components.Button generateButton;
     public net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType joint;
 
-    public net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen(net.minecraft.world.level.block.entity.JigsawBlockEntity jigsawBlockEntity) {
+    public net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen(net.minecraft.world.level.block.entity.JigsawBlockEntity $$0) {
         super(net.minecraft.client.GameNarrator.NO_TITLE);
         this.keepJigsaws = true;
-        this.jigsawEntity = jigsawBlockEntity;
+        this.jigsawEntity = $$0;
     }
 
     public void tick() {
@@ -107,52 +107,52 @@
         this.generateButton.active = $$0;
     }
 
-    public void resize(net.minecraft.client.Minecraft minecraft, int width, int height) {
-        java.lang.String string = this.nameEdit.getValue();
-        java.lang.String string2 = this.targetEdit.getValue();
-        java.lang.String string3 = this.poolEdit.getValue();
-        java.lang.String string4 = this.finalStateEdit.getValue();
-        int n = this.levels;
-        net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType jointType = this.joint;
-        this.init(minecraft, width, height);
-        this.nameEdit.setValue(string);
-        this.targetEdit.setValue(string2);
-        this.poolEdit.setValue(string3);
-        this.finalStateEdit.setValue(string4);
-        this.levels = n;
-        this.joint = jointType;
-        this.jointButton.setValue((java.lang.Object)jointType);
+    public void resize(net.minecraft.client.Minecraft $$0, int $$1, int $$2) {
+        java.lang.String $$3 = this.nameEdit.getValue();
+        java.lang.String $$4 = this.targetEdit.getValue();
+        java.lang.String $$5 = this.poolEdit.getValue();
+        java.lang.String $$6 = this.finalStateEdit.getValue();
+        int $$7 = this.levels;
+        net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType $$8 = this.joint;
+        this.init($$0, $$1, $$2);
+        this.nameEdit.setValue($$3);
+        this.targetEdit.setValue($$4);
+        this.poolEdit.setValue($$5);
+        this.finalStateEdit.setValue($$6);
+        this.levels = $$7;
+        this.joint = $$8;
+        this.jointButton.setValue((java.lang.Object)$$8);
     }
 
     public void removed() {
         this.minecraft.keyboardHandler.setSendRepeatsToGui(false);
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (super.keyPressed(keyCode, scanCode, modifiers)) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if (super.keyPressed($$0, $$1, $$2)) {
             return true;
         }
-        if (this.doneButton.active && (keyCode == 257 || keyCode == 335)) {
+        if (this.doneButton.active && ($$0 == 257 || $$0 == 335)) {
             this.onDone();
             return true;
         }
         return false;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)POOL_LABEL, (int)(this.width / 2 - 153), (int)10, (int)0xA0A0A0);
-        this.poolEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)NAME_LABEL, (int)(this.width / 2 - 153), (int)45, (int)0xA0A0A0);
-        this.nameEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)TARGET_LABEL, (int)(this.width / 2 - 153), (int)80, (int)0xA0A0A0);
-        this.targetEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)FINAL_STATE_LABEL, (int)(this.width / 2 - 153), (int)115, (int)0xA0A0A0);
-        this.finalStateEdit.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)POOL_LABEL, (int)(this.width / 2 - 153), (int)10, (int)0xA0A0A0);
+        this.poolEdit.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)NAME_LABEL, (int)(this.width / 2 - 153), (int)45, (int)0xA0A0A0);
+        this.nameEdit.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)TARGET_LABEL, (int)(this.width / 2 - 153), (int)80, (int)0xA0A0A0);
+        this.targetEdit.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)FINAL_STATE_LABEL, (int)(this.width / 2 - 153), (int)115, (int)0xA0A0A0);
+        this.finalStateEdit.render($$0, $$1, $$2, $$3);
         if (net.minecraft.world.level.block.JigsawBlock.getFrontFacing((net.minecraft.world.level.block.state.BlockState)this.jigsawEntity.getBlockState()).getAxis().isVertical()) {
-            net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)JOINT_LABEL, (int)(this.width / 2 - 153), (int)156, (int)0xFFFFFF);
+            net.minecraft.client.gui.screens.inventory.JigsawBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)JOINT_LABEL, (int)(this.width / 2 - 153), (int)156, (int)0xFFFFFF);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {