--- net/minecraft/client/gui/screens/inventory/LecternScreen.java
+++ net/minecraft/client/gui/screens/inventory/LecternScreen.java
@@ -10,9 +10,9 @@
     public net.minecraft.world.inventory.LecternMenu menu;
     public net.minecraft.world.inventory.ContainerListener listener;
 
-    public net.minecraft.client.gui.screens.inventory.LecternScreen(net.minecraft.world.inventory.LecternMenu lecternMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
+    public net.minecraft.client.gui.screens.inventory.LecternScreen(net.minecraft.world.inventory.LecternMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
         this.listener = new /* Unavailable Anonymous Inner Class!! */;
-        this.menu = lecternMenu;
+        this.menu = $$0;
     }
 
     public net.minecraft.world.inventory.LecternMenu getMenu() {
@@ -51,16 +51,16 @@
         this.sendButtonClick(2);
     }
 
-    public boolean forcePage(int pageNum) {
-        if (pageNum != this.menu.getPage()) {
-            this.sendButtonClick(100 + pageNum);
+    public boolean forcePage(int $$0) {
+        if ($$0 != this.menu.getPage()) {
+            this.sendButtonClick(100 + $$0);
             return true;
         }
         return false;
     }
 
-    public void sendButtonClick(int pageData) {
-        this.minecraft.gameMode.handleInventoryButtonClick(this.menu.containerId, pageData);
+    public void sendButtonClick(int $$0) {
+        this.minecraft.gameMode.handleInventoryButtonClick(this.menu.containerId, $$0);
     }
 
     public boolean isPauseScreen() {