--- net/minecraft/client/gui/screens/inventory/MerchantScreen$TradeOfferButton.java
+++ net/minecraft/client/gui/screens/inventory/MerchantScreen$TradeOfferButton.java
@@ -9,10 +9,10 @@
     public int index;
     public /* synthetic */ net.minecraft.client.gui.screens.inventory.MerchantScreen this$0;
 
-    public net.minecraft.client.gui.screens.inventory.MerchantScreen$TradeOfferButton(net.minecraft.client.gui.screens.inventory.MerchantScreen merchantScreen, int x, int y, int index, net.minecraft.client.gui.components.Button$OnPress onPress) {
+    public net.minecraft.client.gui.screens.inventory.MerchantScreen$TradeOfferButton(net.minecraft.client.gui.screens.inventory.MerchantScreen merchantScreen, int $$0, int $$1, int $$2, net.minecraft.client.gui.components.Button$OnPress $$3) {
         this.this$0 = merchantScreen;
-        super(x, y, 89, 20, net.minecraft.network.chat.CommonComponents.EMPTY, onPress);
-        this.index = index;
+        super($$0, $$1, 89, 20, net.minecraft.network.chat.CommonComponents.EMPTY, $$3);
+        this.index = $$2;
         this.visible = false;
     }
 
@@ -20,19 +20,19 @@
         return this.index;
     }
 
-    public void renderToolTip(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY) {
+    public void renderToolTip(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
         if (this.isHovered && ((net.minecraft.world.inventory.MerchantMenu)this.this$0.menu).getOffers().size() > this.index + this.this$0.scrollOff) {
-            if (mouseX < this.x + 20) {
-                net.minecraft.world.item.ItemStack itemStack = ((net.minecraft.world.item.trading.MerchantOffer)((net.minecraft.world.inventory.MerchantMenu)this.this$0.menu).getOffers().get(this.index + this.this$0.scrollOff)).getCostA();
-                net.minecraft.client.gui.screens.inventory.MerchantScreen.access$000((net.minecraft.client.gui.screens.inventory.MerchantScreen)this.this$0, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.world.item.ItemStack)itemStack, (int)mouseX, (int)mouseY);
-            } else if (mouseX < this.x + 50 && mouseX > this.x + 30) {
-                net.minecraft.world.item.ItemStack itemStack = ((net.minecraft.world.item.trading.MerchantOffer)((net.minecraft.world.inventory.MerchantMenu)this.this$0.menu).getOffers().get(this.index + this.this$0.scrollOff)).getCostB();
-                if (!itemStack.isEmpty()) {
-                    net.minecraft.client.gui.screens.inventory.MerchantScreen.access$100((net.minecraft.client.gui.screens.inventory.MerchantScreen)this.this$0, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.world.item.ItemStack)itemStack, (int)mouseX, (int)mouseY);
+            if ($$1 < this.x + 20) {
+                net.minecraft.world.item.ItemStack $$3 = ((net.minecraft.world.item.trading.MerchantOffer)((net.minecraft.world.inventory.MerchantMenu)this.this$0.menu).getOffers().get(this.index + this.this$0.scrollOff)).getCostA();
+                net.minecraft.client.gui.screens.inventory.MerchantScreen.access$000((net.minecraft.client.gui.screens.inventory.MerchantScreen)this.this$0, (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.world.item.ItemStack)$$3, (int)$$1, (int)$$2);
+            } else if ($$1 < this.x + 50 && $$1 > this.x + 30) {
+                net.minecraft.world.item.ItemStack $$4 = ((net.minecraft.world.item.trading.MerchantOffer)((net.minecraft.world.inventory.MerchantMenu)this.this$0.menu).getOffers().get(this.index + this.this$0.scrollOff)).getCostB();
+                if (!$$4.isEmpty()) {
+                    net.minecraft.client.gui.screens.inventory.MerchantScreen.access$100((net.minecraft.client.gui.screens.inventory.MerchantScreen)this.this$0, (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.world.item.ItemStack)$$4, (int)$$1, (int)$$2);
                 }
-            } else if (mouseX > this.x + 65) {
-                net.minecraft.world.item.ItemStack itemStack = ((net.minecraft.world.item.trading.MerchantOffer)((net.minecraft.world.inventory.MerchantMenu)this.this$0.menu).getOffers().get(this.index + this.this$0.scrollOff)).getResult();
-                net.minecraft.client.gui.screens.inventory.MerchantScreen.access$200((net.minecraft.client.gui.screens.inventory.MerchantScreen)this.this$0, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.world.item.ItemStack)itemStack, (int)mouseX, (int)mouseY);
+            } else if ($$1 > this.x + 65) {
+                net.minecraft.world.item.ItemStack $$5 = ((net.minecraft.world.item.trading.MerchantOffer)((net.minecraft.world.inventory.MerchantMenu)this.this$0.menu).getOffers().get(this.index + this.this$0.scrollOff)).getResult();
+                net.minecraft.client.gui.screens.inventory.MerchantScreen.access$200((net.minecraft.client.gui.screens.inventory.MerchantScreen)this.this$0, (com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.world.item.ItemStack)$$5, (int)$$1, (int)$$2);
             }
         }
     }