--- net/minecraft/client/gui/screens/inventory/MerchantScreen.java
+++ net/minecraft/client/gui/screens/inventory/MerchantScreen.java
@@ -33,8 +33,8 @@
     public int scrollOff;
     public boolean isDragging;
 
-    public net.minecraft.client.gui.screens.inventory.MerchantScreen(net.minecraft.world.inventory.MerchantMenu merchantMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractContainerMenu)merchantMenu, playerInventory, title);
+    public net.minecraft.client.gui.screens.inventory.MerchantScreen(net.minecraft.world.inventory.MerchantMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractContainerMenu)$$0, $$1, $$2);
         this.tradeOfferButtons = new net.minecraft.client.gui.screens.inventory.MerchantScreen$TradeOfferButton[7];
         this.imageWidth = 276;
         this.inventoryLabelX = 107;
@@ -62,200 +62,199 @@
         }
     }
 
-    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack matrixStack, int x, int y) {
-        int n = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getTraderLevel();
-        if (n > 0 && n <= 5 && ((net.minecraft.world.inventory.MerchantMenu)this.menu).showProgressBar()) {
-            net.minecraft.network.chat.MutableComponent mutableComponent = this.title.copy().append(LEVEL_SEPARATOR).append((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)("merchant.level." + n)));
-            int n2 = this.font.width((net.minecraft.network.chat.FormattedText)mutableComponent);
-            int n3 = 49 + this.imageWidth / 2 - n2 / 2;
-            this.font.draw(matrixStack, (net.minecraft.network.chat.Component)mutableComponent, (float)n3, 6.0f, 0x404040);
+    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
+        int $$3 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getTraderLevel();
+        if ($$3 > 0 && $$3 <= 5 && ((net.minecraft.world.inventory.MerchantMenu)this.menu).showProgressBar()) {
+            net.minecraft.network.chat.MutableComponent $$4 = this.title.copy().append(LEVEL_SEPARATOR).append((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)("merchant.level." + $$3)));
+            int $$5 = this.font.width((net.minecraft.network.chat.FormattedText)$$4);
+            int $$6 = 49 + this.imageWidth / 2 - $$5 / 2;
+            this.font.draw($$0, (net.minecraft.network.chat.Component)$$4, (float)$$6, 6.0f, 0x404040);
         } else {
-            this.font.draw(matrixStack, this.title, (float)(49 + this.imageWidth / 2 - this.font.width((net.minecraft.network.chat.FormattedText)this.title) / 2), 6.0f, 0x404040);
+            this.font.draw($$0, this.title, (float)(49 + this.imageWidth / 2 - this.font.width((net.minecraft.network.chat.FormattedText)this.title) / 2), 6.0f, 0x404040);
         }
-        this.font.draw(matrixStack, this.playerInventoryTitle, (float)this.inventoryLabelX, (float)this.inventoryLabelY, 0x404040);
-        int n4 = this.font.width((net.minecraft.network.chat.FormattedText)TRADES_LABEL);
-        this.font.draw(matrixStack, TRADES_LABEL, (float)(5 - n4 / 2 + 48), 6.0f, 0x404040);
+        this.font.draw($$0, this.playerInventoryTitle, (float)this.inventoryLabelX, (float)this.inventoryLabelY, 0x404040);
+        int $$7 = this.font.width((net.minecraft.network.chat.FormattedText)TRADES_LABEL);
+        this.font.draw($$0, TRADES_LABEL, (float)(5 - $$7 / 2 + 48), 6.0f, 0x404040);
     }
 
-    public void renderBg(com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTicks, int x, int y) {
+    public void renderBg(com.mojang.blaze3d.vertex.PoseStack $$0, float $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)VILLAGER_LOCATION);
-        int n = (this.width - this.imageWidth) / 2;
-        int n2 = (this.height - this.imageHeight) / 2;
-        net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)n, (int)n2, (int)this.getBlitOffset(), (float)0.0f, (float)0.0f, (int)this.imageWidth, (int)this.imageHeight, (int)512, (int)256);
-        net.minecraft.world.item.trading.MerchantOffers merchantOffers = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers();
-        if (!merchantOffers.isEmpty()) {
-            int n3 = this.shopItem;
-            if (n3 < 0 || n3 >= merchantOffers.size()) {
+        int $$4 = (this.width - this.imageWidth) / 2;
+        int $$5 = (this.height - this.imageHeight) / 2;
+        net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$4, (int)$$5, (int)this.getBlitOffset(), (float)0.0f, (float)0.0f, (int)this.imageWidth, (int)this.imageHeight, (int)512, (int)256);
+        net.minecraft.world.item.trading.MerchantOffers $$6 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers();
+        if (!$$6.isEmpty()) {
+            int $$7 = this.shopItem;
+            if ($$7 < 0 || $$7 >= $$6.size()) {
                 return;
             }
-            net.minecraft.world.item.trading.MerchantOffer merchantOffer = (net.minecraft.world.item.trading.MerchantOffer)merchantOffers.get(n3);
-            if (merchantOffer.isOutOfStock()) {
+            net.minecraft.world.item.trading.MerchantOffer $$8 = (net.minecraft.world.item.trading.MerchantOffer)$$6.get($$7);
+            if ($$8.isOutOfStock()) {
                 com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)VILLAGER_LOCATION);
                 com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-                net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)(this.leftPos + 83 + 99), (int)(this.topPos + 35), (int)this.getBlitOffset(), (float)311.0f, (float)0.0f, (int)28, (int)21, (int)512, (int)256);
+                net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)(this.leftPos + 83 + 99), (int)(this.topPos + 35), (int)this.getBlitOffset(), (float)311.0f, (float)0.0f, (int)28, (int)21, (int)512, (int)256);
             }
         }
     }
 
-    public void renderProgressBar(com.mojang.blaze3d.vertex.PoseStack poseStack, int posX, int posY, net.minecraft.world.item.trading.MerchantOffer merchantOffer) {
+    public void renderProgressBar(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, net.minecraft.world.item.trading.MerchantOffer $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)VILLAGER_LOCATION);
-        int n = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getTraderLevel();
-        int n2 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getTraderXp();
-        if (n >= 5) {
+        int $$4 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getTraderLevel();
+        int $$5 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getTraderXp();
+        if ($$4 >= 5) {
             return;
         }
-        net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(posX + 136), (int)(posY + 16), (int)this.getBlitOffset(), (float)0.0f, (float)186.0f, (int)102, (int)5, (int)512, (int)256);
-        int n3 = net.minecraft.world.entity.npc.VillagerData.getMinXpPerLevel((int)n);
-        if (n2 < n3 || !net.minecraft.world.entity.npc.VillagerData.canLevelUp((int)n)) {
+        net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 136), (int)($$2 + 16), (int)this.getBlitOffset(), (float)0.0f, (float)186.0f, (int)102, (int)5, (int)512, (int)256);
+        int $$6 = net.minecraft.world.entity.npc.VillagerData.getMinXpPerLevel((int)$$4);
+        if ($$5 < $$6 || !net.minecraft.world.entity.npc.VillagerData.canLevelUp((int)$$4)) {
             return;
         }
-        int n4 = 100;
-        float f = 100.0f / (float)(net.minecraft.world.entity.npc.VillagerData.getMaxXpPerLevel((int)n) - n3);
-        int n5 = java.lang.Math.min((int)net.minecraft.util.Mth.floor((float)(f * (float)(n2 - n3))), (int)100);
-        net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(posX + 136), (int)(posY + 16), (int)this.getBlitOffset(), (float)0.0f, (float)191.0f, (int)(n5 + 1), (int)5, (int)512, (int)256);
-        int n6 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getFutureTraderXp();
-        if (n6 > 0) {
-            int n7 = java.lang.Math.min((int)net.minecraft.util.Mth.floor((float)((float)n6 * f)), (int)(100 - n5));
-            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(posX + 136 + n5 + 1), (int)(posY + 16 + 1), (int)this.getBlitOffset(), (float)2.0f, (float)182.0f, (int)n7, (int)3, (int)512, (int)256);
+        int $$7 = 100;
+        float $$8 = 100.0f / (float)(net.minecraft.world.entity.npc.VillagerData.getMaxXpPerLevel((int)$$4) - $$6);
+        int $$9 = java.lang.Math.min((int)net.minecraft.util.Mth.floor((float)($$8 * (float)($$5 - $$6))), (int)100);
+        net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 136), (int)($$2 + 16), (int)this.getBlitOffset(), (float)0.0f, (float)191.0f, (int)($$9 + 1), (int)5, (int)512, (int)256);
+        int $$10 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getFutureTraderXp();
+        if ($$10 > 0) {
+            int $$11 = java.lang.Math.min((int)net.minecraft.util.Mth.floor((float)((float)$$10 * $$8)), (int)(100 - $$9));
+            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 136 + $$9 + 1), (int)($$2 + 16 + 1), (int)this.getBlitOffset(), (float)2.0f, (float)182.0f, (int)$$11, (int)3, (int)512, (int)256);
         }
     }
 
-    public void renderScroller(com.mojang.blaze3d.vertex.PoseStack poseStack, int posX, int posY, net.minecraft.world.item.trading.MerchantOffers merchantOffers) {
-        int n = merchantOffers.size() + 1 - 7;
-        if (n > 1) {
-            int n2 = 139 - (27 + (n - 1) * 139 / n);
-            int n3 = 1 + n2 / n + 139 / n;
-            int n4 = 113;
-            int n5 = java.lang.Math.min((int)113, (int)(this.scrollOff * n3));
-            if (this.scrollOff == n - 1) {
-                n5 = 113;
+    public void renderScroller(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, net.minecraft.world.item.trading.MerchantOffers $$3) {
+        int $$4 = $$3.size() + 1 - 7;
+        if ($$4 > 1) {
+            int $$5 = 139 - (27 + ($$4 - 1) * 139 / $$4);
+            int $$6 = 1 + $$5 / $$4 + 139 / $$4;
+            int $$7 = 113;
+            int $$8 = java.lang.Math.min((int)113, (int)(this.scrollOff * $$6));
+            if (this.scrollOff == $$4 - 1) {
+                $$8 = 113;
             }
-            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(posX + 94), (int)(posY + 18 + n5), (int)this.getBlitOffset(), (float)0.0f, (float)199.0f, (int)6, (int)27, (int)512, (int)256);
+            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 94), (int)($$2 + 18 + $$8), (int)this.getBlitOffset(), (float)0.0f, (float)199.0f, (int)6, (int)27, (int)512, (int)256);
         } else {
-            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(posX + 94), (int)(posY + 18), (int)this.getBlitOffset(), (float)6.0f, (float)199.0f, (int)6, (int)27, (int)512, (int)256);
+            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$1 + 94), (int)($$2 + 18), (int)this.getBlitOffset(), (float)6.0f, (float)199.0f, (int)6, (int)27, (int)512, (int)256);
         }
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.world.item.trading.MerchantOffers merchantOffers = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers();
-        if (!merchantOffers.isEmpty()) {
-            net.minecraft.world.item.trading.MerchantOffer merchantOffer22;
-            int n = (this.width - this.imageWidth) / 2;
-            int n2 = (this.height - this.imageHeight) / 2;
-            int n3 = n2 + 16 + 1;
-            int n4 = n + 5 + 5;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        super.render($$0, $$1, $$2, $$3);
+        net.minecraft.world.item.trading.MerchantOffers $$4 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers();
+        if (!$$4.isEmpty()) {
+            int $$5 = (this.width - this.imageWidth) / 2;
+            int $$6 = (this.height - this.imageHeight) / 2;
+            int $$7 = $$6 + 16 + 1;
+            int $$8 = $$5 + 5 + 5;
             com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)VILLAGER_LOCATION);
-            this.renderScroller(matrixStack, n, n2, merchantOffers);
-            int n5 = 0;
-            for (net.minecraft.world.item.trading.MerchantOffer merchantOffer22 : merchantOffers) {
-                if (this.canScroll(merchantOffers.size()) && (n5 < this.scrollOff || n5 >= 7 + this.scrollOff)) {
-                    ++n5;
+            this.renderScroller($$0, $$5, $$6, $$4);
+            int $$9 = 0;
+            for (net.minecraft.world.item.trading.MerchantOffer $$10 : $$4) {
+                if (this.canScroll($$4.size()) && ($$9 < this.scrollOff || $$9 >= 7 + this.scrollOff)) {
+                    ++$$9;
                     continue;
                 }
-                net.minecraft.client.gui.screens.inventory.MerchantScreen$TradeOfferButton[] tradeOfferButtonArray = merchantOffer22.getBaseCostA();
-                net.minecraft.world.item.ItemStack itemStack = merchantOffer22.getCostA();
-                net.minecraft.world.item.ItemStack itemStack2 = merchantOffer22.getCostB();
-                net.minecraft.world.item.ItemStack itemStack3 = merchantOffer22.getResult();
+                net.minecraft.world.item.ItemStack $$11 = $$10.getBaseCostA();
+                net.minecraft.world.item.ItemStack $$12 = $$10.getCostA();
+                net.minecraft.world.item.ItemStack $$13 = $$10.getCostB();
+                net.minecraft.world.item.ItemStack $$14 = $$10.getResult();
                 this.itemRenderer.blitOffset = 100.0f;
-                int n6 = n3 + 2;
-                this.renderAndDecorateCostA(matrixStack, itemStack, (net.minecraft.world.item.ItemStack)tradeOfferButtonArray, n4, n6);
-                if (!itemStack2.isEmpty()) {
-                    this.itemRenderer.renderAndDecorateFakeItem(itemStack2, n + 5 + 35, n6);
-                    this.itemRenderer.renderGuiItemDecorations(this.font, itemStack2, n + 5 + 35, n6);
+                int $$15 = $$7 + 2;
+                this.renderAndDecorateCostA($$0, $$12, $$11, $$8, $$15);
+                if (!$$13.isEmpty()) {
+                    this.itemRenderer.renderAndDecorateFakeItem($$13, $$5 + 5 + 35, $$15);
+                    this.itemRenderer.renderGuiItemDecorations(this.font, $$13, $$5 + 5 + 35, $$15);
                 }
-                this.renderButtonArrows(matrixStack, merchantOffer22, n, n6);
-                this.itemRenderer.renderAndDecorateFakeItem(itemStack3, n + 5 + 68, n6);
-                this.itemRenderer.renderGuiItemDecorations(this.font, itemStack3, n + 5 + 68, n6);
+                this.renderButtonArrows($$0, $$10, $$5, $$15);
+                this.itemRenderer.renderAndDecorateFakeItem($$14, $$5 + 5 + 68, $$15);
+                this.itemRenderer.renderGuiItemDecorations(this.font, $$14, $$5 + 5 + 68, $$15);
                 this.itemRenderer.blitOffset = 0.0f;
-                n3 += 20;
-                ++n5;
+                $$7 += 20;
+                ++$$9;
             }
-            int n7 = this.shopItem;
-            merchantOffer22 = (net.minecraft.world.item.trading.MerchantOffer)merchantOffers.get(n7);
+            int $$16 = this.shopItem;
+            net.minecraft.world.item.trading.MerchantOffer $$17 = (net.minecraft.world.item.trading.MerchantOffer)$$4.get($$16);
             if (((net.minecraft.world.inventory.MerchantMenu)this.menu).showProgressBar()) {
-                this.renderProgressBar(matrixStack, n, n2, merchantOffer22);
+                this.renderProgressBar($$0, $$5, $$6, $$17);
             }
-            if (merchantOffer22.isOutOfStock() && this.isHovering(186, 35, 22, 21, (double)mouseX, (double)mouseY) && ((net.minecraft.world.inventory.MerchantMenu)this.menu).canRestock()) {
-                this.renderTooltip(matrixStack, DEPRECATED_TOOLTIP, mouseX, mouseY);
+            if ($$17.isOutOfStock() && this.isHovering(186, 35, 22, 21, (double)$$1, (double)$$2) && ((net.minecraft.world.inventory.MerchantMenu)this.menu).canRestock()) {
+                this.renderTooltip($$0, DEPRECATED_TOOLTIP, $$1, $$2);
             }
-            for (net.minecraft.client.gui.screens.inventory.MerchantScreen$TradeOfferButton tradeOfferButton : this.tradeOfferButtons) {
-                if (tradeOfferButton.isHoveredOrFocused()) {
-                    tradeOfferButton.renderToolTip(matrixStack, mouseX, mouseY);
+            for (net.minecraft.client.gui.screens.inventory.MerchantScreen$TradeOfferButton $$18 : this.tradeOfferButtons) {
+                if ($$18.isHoveredOrFocused()) {
+                    $$18.renderToolTip($$0, $$1, $$2);
                 }
-                tradeOfferButton.visible = tradeOfferButton.index < ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size();
+                $$18.visible = $$18.index < ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size();
             }
             com.mojang.blaze3d.systems.RenderSystem.enableDepthTest();
         }
-        this.renderTooltip(matrixStack, mouseX, mouseY);
+        this.renderTooltip($$0, $$1, $$2);
     }
 
-    public void renderButtonArrows(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.world.item.trading.MerchantOffer merchantOffer, int posX, int posY) {
+    public void renderButtonArrows(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.world.item.trading.MerchantOffer $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)VILLAGER_LOCATION);
-        if (merchantOffer.isOutOfStock()) {
-            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(posX + 5 + 35 + 20), (int)(posY + 3), (int)this.getBlitOffset(), (float)25.0f, (float)171.0f, (int)10, (int)9, (int)512, (int)256);
+        if ($$1.isOutOfStock()) {
+            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$2 + 5 + 35 + 20), (int)($$3 + 3), (int)this.getBlitOffset(), (float)25.0f, (float)171.0f, (int)10, (int)9, (int)512, (int)256);
         } else {
-            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(posX + 5 + 35 + 20), (int)(posY + 3), (int)this.getBlitOffset(), (float)15.0f, (float)171.0f, (int)10, (int)9, (int)512, (int)256);
+            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$2 + 5 + 35 + 20), (int)($$3 + 3), (int)this.getBlitOffset(), (float)15.0f, (float)171.0f, (int)10, (int)9, (int)512, (int)256);
         }
     }
 
-    public void renderAndDecorateCostA(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.world.item.ItemStack realCost, net.minecraft.world.item.ItemStack baseCost, int x, int y) {
-        this.itemRenderer.renderAndDecorateFakeItem(realCost, x, y);
-        if (baseCost.getCount() == realCost.getCount()) {
-            this.itemRenderer.renderGuiItemDecorations(this.font, realCost, x, y);
+    public void renderAndDecorateCostA(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.world.item.ItemStack $$2, int $$3, int $$4) {
+        this.itemRenderer.renderAndDecorateFakeItem($$1, $$3, $$4);
+        if ($$2.getCount() == $$1.getCount()) {
+            this.itemRenderer.renderGuiItemDecorations(this.font, $$1, $$3, $$4);
         } else {
-            this.itemRenderer.renderGuiItemDecorations(this.font, baseCost, x, y, baseCost.getCount() == 1 ? "1" : null);
-            this.itemRenderer.renderGuiItemDecorations(this.font, realCost, x + 14, y, realCost.getCount() == 1 ? "1" : null);
+            this.itemRenderer.renderGuiItemDecorations(this.font, $$2, $$3, $$4, $$2.getCount() == 1 ? "1" : null);
+            this.itemRenderer.renderGuiItemDecorations(this.font, $$1, $$3 + 14, $$4, $$1.getCount() == 1 ? "1" : null);
             com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)VILLAGER_LOCATION);
             this.setBlitOffset(this.getBlitOffset() + 300);
-            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(x + 7), (int)(y + 12), (int)this.getBlitOffset(), (float)0.0f, (float)176.0f, (int)9, (int)2, (int)512, (int)256);
+            net.minecraft.client.gui.screens.inventory.MerchantScreen.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$3 + 7), (int)($$4 + 12), (int)this.getBlitOffset(), (float)0.0f, (float)176.0f, (int)9, (int)2, (int)512, (int)256);
             this.setBlitOffset(this.getBlitOffset() - 300);
         }
     }
 
-    public boolean canScroll(int numOffers) {
-        return numOffers > 7;
+    public boolean canScroll(int $$0) {
+        return $$0 > 7;
     }
 
-    public boolean mouseScrolled(double mouseX, double mouseY, double delta) {
-        int n = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size();
-        if (this.canScroll(n)) {
-            int n2 = n - 7;
-            this.scrollOff = net.minecraft.util.Mth.clamp((int)((int)((double)this.scrollOff - delta)), (int)0, (int)n2);
+    public boolean mouseScrolled(double $$0, double $$1, double $$2) {
+        int $$3 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size();
+        if (this.canScroll($$3)) {
+            int $$4 = $$3 - 7;
+            this.scrollOff = net.minecraft.util.Mth.clamp((int)((int)((double)this.scrollOff - $$2)), (int)0, (int)$$4);
         }
         return true;
     }
 
-    public boolean mouseDragged(double mouseX, double mouseY, int button, double dragX, double dragY) {
-        int n = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size();
+    public boolean mouseDragged(double $$0, double $$1, int $$2, double $$3, double $$4) {
+        int $$5 = ((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size();
         if (this.isDragging) {
-            int n2 = this.topPos + 18;
-            int n3 = n2 + 139;
-            int n4 = n - 7;
-            float f = ((float)mouseY - (float)n2 - 13.5f) / ((float)(n3 - n2) - 27.0f);
-            f = f * (float)n4 + 0.5f;
-            this.scrollOff = net.minecraft.util.Mth.clamp((int)((int)f), (int)0, (int)n4);
+            int $$6 = this.topPos + 18;
+            int $$7 = $$6 + 139;
+            int $$8 = $$5 - 7;
+            float $$9 = ((float)$$1 - (float)$$6 - 13.5f) / ((float)($$7 - $$6) - 27.0f);
+            $$9 = $$9 * (float)$$8 + 0.5f;
+            this.scrollOff = net.minecraft.util.Mth.clamp((int)((int)$$9), (int)0, (int)$$8);
             return true;
         }
-        return super.mouseDragged(mouseX, mouseY, button, dragX, dragY);
+        return super.mouseDragged($$0, $$1, $$2, $$3, $$4);
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
         this.isDragging = false;
-        int n = (this.width - this.imageWidth) / 2;
-        int n2 = (this.height - this.imageHeight) / 2;
-        if (this.canScroll(((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size()) && mouseX > (double)(n + 94) && mouseX < (double)(n + 94 + 6) && mouseY > (double)(n2 + 18) && mouseY <= (double)(n2 + 18 + 139 + 1)) {
+        int $$3 = (this.width - this.imageWidth) / 2;
+        int $$4 = (this.height - this.imageHeight) / 2;
+        if (this.canScroll(((net.minecraft.world.inventory.MerchantMenu)this.menu).getOffers().size()) && $$0 > (double)($$3 + 94) && $$0 < (double)($$3 + 94 + 6) && $$1 > (double)($$4 + 18) && $$1 <= (double)($$4 + 18 + 139 + 1)) {
             this.isDragging = true;
         }
-        return super.mouseClicked(mouseX, mouseY, button);
+        return super.mouseClicked($$0, $$1, $$2);
     }
 
     public static /* synthetic */ void access$000(net.minecraft.client.gui.screens.inventory.MerchantScreen $$0, com.mojang.blaze3d.vertex.PoseStack $$1, net.minecraft.world.item.ItemStack $$2, int $$3, int $$4) {