--- net/minecraft/client/gui/screens/inventory/MinecartCommandBlockEditScreen.java
+++ net/minecraft/client/gui/screens/inventory/MinecartCommandBlockEditScreen.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.level.BaseCommandBlock commandBlock;
 
-    public net.minecraft.client.gui.screens.inventory.MinecartCommandBlockEditScreen(net.minecraft.world.level.BaseCommandBlock commandBlock) {
-        this.commandBlock = commandBlock;
+    public net.minecraft.client.gui.screens.inventory.MinecartCommandBlockEditScreen(net.minecraft.world.level.BaseCommandBlock $$0) {
+        this.commandBlock = $$0;
     }
 
     public net.minecraft.world.level.BaseCommandBlock getCommandBlock() {
@@ -25,10 +25,10 @@
         this.commandEdit.setValue(this.getCommandBlock().getCommand());
     }
 
-    public void populateAndSendPacket(net.minecraft.world.level.BaseCommandBlock commandBlock) {
-        if (commandBlock instanceof net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase) {
-            net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase minecartCommandBase = (net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase)commandBlock;
-            this.minecraft.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundSetCommandMinecartPacket(minecartCommandBase.getMinecart().getId(), this.commandEdit.getValue(), commandBlock.isTrackOutput()));
+    public void populateAndSendPacket(net.minecraft.world.level.BaseCommandBlock $$0) {
+        if ($$0 instanceof net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase) {
+            net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase $$1 = (net.minecraft.world.entity.vehicle.MinecartCommandBlock$MinecartCommandBase)$$0;
+            this.minecraft.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundSetCommandMinecartPacket($$1.getMinecart().getId(), this.commandEdit.getValue(), $$0.isTrackOutput()));
         }
     }
 }