--- net/minecraft/client/gui/screens/inventory/PageButton.java
+++ net/minecraft/client/gui/screens/inventory/PageButton.java
@@ -9,30 +9,30 @@
     public boolean isForward;
     public boolean playTurnSound;
 
-    public net.minecraft.client.gui.screens.inventory.PageButton(int x, int y, boolean isForward, net.minecraft.client.gui.components.Button$OnPress onPress, boolean playTurnSound) {
-        super(x, y, 23, 13, net.minecraft.network.chat.CommonComponents.EMPTY, onPress);
-        this.isForward = isForward;
-        this.playTurnSound = playTurnSound;
+    public net.minecraft.client.gui.screens.inventory.PageButton(int $$0, int $$1, boolean $$2, net.minecraft.client.gui.components.Button$OnPress $$3, boolean $$4) {
+        super($$0, $$1, 23, 13, net.minecraft.network.chat.CommonComponents.EMPTY, $$3);
+        this.isForward = $$2;
+        this.playTurnSound = $$4;
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)net.minecraft.client.gui.screens.inventory.BookViewScreen.BOOK_LOCATION);
-        int n = 0;
-        int n2 = 192;
+        int $$4 = 0;
+        int $$5 = 192;
         if (this.isHoveredOrFocused()) {
-            n += 23;
+            $$4 += 23;
         }
         if (!this.isForward) {
-            n2 += 13;
+            $$5 += 13;
         }
-        this.blit(matrixStack, this.x, this.y, n, n2, 23, 13);
+        this.blit($$0, this.x, this.y, $$4, $$5, 23, 13);
     }
 
-    public void playDownSound(net.minecraft.client.sounds.SoundManager handler) {
+    public void playDownSound(net.minecraft.client.sounds.SoundManager $$0) {
         if (this.playTurnSound) {
-            handler.play((net.minecraft.client.resources.sounds.SoundInstance)net.minecraft.client.resources.sounds.SimpleSoundInstance.forUI((net.minecraft.sounds.SoundEvent)net.minecraft.sounds.SoundEvents.BOOK_PAGE_TURN, (float)1.0f));
+            $$0.play((net.minecraft.client.resources.sounds.SoundInstance)net.minecraft.client.resources.sounds.SimpleSoundInstance.forUI((net.minecraft.sounds.SoundEvent)net.minecraft.sounds.SoundEvents.BOOK_PAGE_TURN, (float)1.0f));
         }
     }
 }