--- net/minecraft/client/gui/screens/inventory/SmithingScreen.java
+++ net/minecraft/client/gui/screens/inventory/SmithingScreen.java
@@ -8,15 +8,15 @@
  {
     public static net.minecraft.resources.ResourceLocation SMITHING_LOCATION;
 
-    public net.minecraft.client.gui.screens.inventory.SmithingScreen(net.minecraft.world.inventory.SmithingMenu smithingMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.ItemCombinerMenu)smithingMenu, playerInventory, title, SMITHING_LOCATION);
+    public net.minecraft.client.gui.screens.inventory.SmithingScreen(net.minecraft.world.inventory.SmithingMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.ItemCombinerMenu)$$0, $$1, $$2, SMITHING_LOCATION);
         this.titleLabelX = 60;
         this.titleLabelY = 18;
     }
 
-    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack matrixStack, int x, int y) {
+    public void renderLabels(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();
-        super.renderLabels(matrixStack, x, y);
+        super.renderLabels($$0, $$1, $$2);
     }
 
     static {