--- net/minecraft/client/gui/screens/inventory/SmokerScreen.java
+++ net/minecraft/client/gui/screens/inventory/SmokerScreen.java
@@ -8,8 +8,8 @@
  {
     public static net.minecraft.resources.ResourceLocation TEXTURE;
 
-    public net.minecraft.client.gui.screens.inventory.SmokerScreen(net.minecraft.world.inventory.SmokerMenu smokerMenu, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.network.chat.Component title) {
-        super((net.minecraft.world.inventory.AbstractFurnaceMenu)smokerMenu, (net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent)new net.minecraft.client.gui.screens.recipebook.SmokingRecipeBookComponent(), playerInventory, title, TEXTURE);
+    public net.minecraft.client.gui.screens.inventory.SmokerScreen(net.minecraft.world.inventory.SmokerMenu $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.network.chat.Component $$2) {
+        super((net.minecraft.world.inventory.AbstractFurnaceMenu)$$0, (net.minecraft.client.gui.screens.recipebook.AbstractFurnaceRecipeBookComponent)new net.minecraft.client.gui.screens.recipebook.SmokingRecipeBookComponent(), $$1, $$2, TEXTURE);
     }
 
     static {