--- net/minecraft/client/gui/screens/inventory/StructureBlockEditScreen.java
+++ net/minecraft/client/gui/screens/inventory/StructureBlockEditScreen.java
@@ -47,13 +47,13 @@
     public net.minecraft.client.gui.components.CycleButton<java.lang.Boolean> toggleBoundingBox;
     public java.text.DecimalFormat decimalFormat;
 
-    public net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen(net.minecraft.world.level.block.entity.StructureBlockEntity structureBlockEntity) {
+    public net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen(net.minecraft.world.level.block.entity.StructureBlockEntity $$0) {
         super((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)net.minecraft.world.level.block.Blocks.STRUCTURE_BLOCK.getDescriptionId()));
         this.initialMirror = net.minecraft.world.level.block.Mirror.NONE;
         this.initialRotation = net.minecraft.world.level.block.Rotation.NONE;
         this.initialMode = net.minecraft.world.level.block.state.properties.StructureMode.DATA;
         this.decimalFormat = new java.text.DecimalFormat("0.0###");
-        this.structure = structureBlockEntity;
+        this.structure = $$0;
         this.decimalFormat.setDecimalFormatSymbols(java.text.DecimalFormatSymbols.getInstance((java.util.Locale)java.util.Locale.ROOT));
     }
 
@@ -185,28 +185,28 @@
         this.setInitialFocus((net.minecraft.client.gui.components.events.GuiEventListener)this.nameEdit);
     }
 
-    public void resize(net.minecraft.client.Minecraft minecraft, int width, int height) {
-        java.lang.String string = this.nameEdit.getValue();
-        java.lang.String string2 = this.posXEdit.getValue();
-        java.lang.String string3 = this.posYEdit.getValue();
-        java.lang.String string4 = this.posZEdit.getValue();
-        java.lang.String string5 = this.sizeXEdit.getValue();
-        java.lang.String string6 = this.sizeYEdit.getValue();
-        java.lang.String string7 = this.sizeZEdit.getValue();
-        java.lang.String string8 = this.integrityEdit.getValue();
-        java.lang.String string9 = this.seedEdit.getValue();
-        java.lang.String string10 = this.dataEdit.getValue();
-        this.init(minecraft, width, height);
-        this.nameEdit.setValue(string);
-        this.posXEdit.setValue(string2);
-        this.posYEdit.setValue(string3);
-        this.posZEdit.setValue(string4);
-        this.sizeXEdit.setValue(string5);
-        this.sizeYEdit.setValue(string6);
-        this.sizeZEdit.setValue(string7);
-        this.integrityEdit.setValue(string8);
-        this.seedEdit.setValue(string9);
-        this.dataEdit.setValue(string10);
+    public void resize(net.minecraft.client.Minecraft $$0, int $$1, int $$2) {
+        java.lang.String $$3 = this.nameEdit.getValue();
+        java.lang.String $$4 = this.posXEdit.getValue();
+        java.lang.String $$5 = this.posYEdit.getValue();
+        java.lang.String $$6 = this.posZEdit.getValue();
+        java.lang.String $$7 = this.sizeXEdit.getValue();
+        java.lang.String $$8 = this.sizeYEdit.getValue();
+        java.lang.String $$9 = this.sizeZEdit.getValue();
+        java.lang.String $$10 = this.integrityEdit.getValue();
+        java.lang.String $$11 = this.seedEdit.getValue();
+        java.lang.String $$12 = this.dataEdit.getValue();
+        this.init($$0, $$1, $$2);
+        this.nameEdit.setValue($$3);
+        this.posXEdit.setValue($$4);
+        this.posYEdit.setValue($$5);
+        this.posZEdit.setValue($$6);
+        this.sizeXEdit.setValue($$7);
+        this.sizeYEdit.setValue($$8);
+        this.sizeZEdit.setValue($$9);
+        this.integrityEdit.setValue($$10);
+        this.seedEdit.setValue($$11);
+        this.dataEdit.setValue($$12);
     }
 
     public void removed() {
@@ -302,38 +302,38 @@
         }
     }
 
-    public boolean sendToServer(net.minecraft.world.level.block.entity.StructureBlockEntity$UpdateType updateType) {
-        net.minecraft.core.BlockPos blockPos = new net.minecraft.core.BlockPos(this.parseCoordinate(this.posXEdit.getValue()), this.parseCoordinate(this.posYEdit.getValue()), this.parseCoordinate(this.posZEdit.getValue()));
-        net.minecraft.core.Vec3i vec3i = new net.minecraft.core.Vec3i(this.parseCoordinate(this.sizeXEdit.getValue()), this.parseCoordinate(this.sizeYEdit.getValue()), this.parseCoordinate(this.sizeZEdit.getValue()));
-        float f = this.parseIntegrity(this.integrityEdit.getValue());
-        long l = this.parseSeed(this.seedEdit.getValue());
-        this.minecraft.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundSetStructureBlockPacket(this.structure.getBlockPos(), updateType, this.structure.getMode(), this.nameEdit.getValue(), blockPos, vec3i, this.structure.getMirror(), this.structure.getRotation(), this.dataEdit.getValue(), this.structure.isIgnoreEntities(), this.structure.getShowAir(), this.structure.getShowBoundingBox(), f, l));
+    public boolean sendToServer(net.minecraft.world.level.block.entity.StructureBlockEntity$UpdateType $$0) {
+        net.minecraft.core.BlockPos $$1 = new net.minecraft.core.BlockPos(this.parseCoordinate(this.posXEdit.getValue()), this.parseCoordinate(this.posYEdit.getValue()), this.parseCoordinate(this.posZEdit.getValue()));
+        net.minecraft.core.Vec3i $$2 = new net.minecraft.core.Vec3i(this.parseCoordinate(this.sizeXEdit.getValue()), this.parseCoordinate(this.sizeYEdit.getValue()), this.parseCoordinate(this.sizeZEdit.getValue()));
+        float $$3 = this.parseIntegrity(this.integrityEdit.getValue());
+        long $$4 = this.parseSeed(this.seedEdit.getValue());
+        this.minecraft.getConnection().send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ServerboundSetStructureBlockPacket(this.structure.getBlockPos(), $$0, this.structure.getMode(), this.nameEdit.getValue(), $$1, $$2, this.structure.getMirror(), this.structure.getRotation(), this.dataEdit.getValue(), this.structure.isIgnoreEntities(), this.structure.getShowAir(), this.structure.getShowBoundingBox(), $$3, $$4));
         return true;
     }
 
-    public long parseSeed(java.lang.String seed) {
+    public long parseSeed(java.lang.String $$0) {
         try {
-            return java.lang.Long.valueOf((java.lang.String)seed);
+            return java.lang.Long.valueOf((java.lang.String)$$0);
         }
-        catch (java.lang.NumberFormatException numberFormatException) {
+        catch (java.lang.NumberFormatException $$1) {
             return 0L;
         }
     }
 
-    public float parseIntegrity(java.lang.String integrity) {
+    public float parseIntegrity(java.lang.String $$0) {
         try {
-            return java.lang.Float.valueOf((java.lang.String)integrity).floatValue();
+            return java.lang.Float.valueOf((java.lang.String)$$0).floatValue();
         }
-        catch (java.lang.NumberFormatException numberFormatException) {
+        catch (java.lang.NumberFormatException $$1) {
             return 1.0f;
         }
     }
 
-    public int parseCoordinate(java.lang.String coordinate) {
+    public int parseCoordinate(java.lang.String $$0) {
         try {
-            return java.lang.Integer.parseInt((java.lang.String)coordinate);
+            return java.lang.Integer.parseInt((java.lang.String)$$0);
         }
-        catch (java.lang.NumberFormatException numberFormatException) {
+        catch (java.lang.NumberFormatException $$1) {
             return 0;
         }
     }
@@ -342,52 +342,52 @@
         this.onCancel();
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (super.keyPressed(keyCode, scanCode, modifiers)) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if (super.keyPressed($$0, $$1, $$2)) {
             return true;
         }
-        if (keyCode == 257 || keyCode == 335) {
+        if ($$0 == 257 || $$0 == 335) {
             this.onDone();
             return true;
         }
         return false;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.world.level.block.state.properties.StructureMode structureMode = this.structure.getMode();
-        net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)10, (int)0xFFFFFF);
-        if (structureMode != net.minecraft.world.level.block.state.properties.StructureMode.DATA) {
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)NAME_LABEL, (int)(this.width / 2 - 153), (int)30, (int)0xA0A0A0);
-            this.nameEdit.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.world.level.block.state.properties.StructureMode $$4 = this.structure.getMode();
+        net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)10, (int)0xFFFFFF);
+        if ($$4 != net.minecraft.world.level.block.state.properties.StructureMode.DATA) {
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)NAME_LABEL, (int)(this.width / 2 - 153), (int)30, (int)0xA0A0A0);
+            this.nameEdit.render($$0, $$1, $$2, $$3);
         }
-        if (structureMode == net.minecraft.world.level.block.state.properties.StructureMode.LOAD || structureMode == net.minecraft.world.level.block.state.properties.StructureMode.SAVE) {
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)POSITION_LABEL, (int)(this.width / 2 - 153), (int)70, (int)0xA0A0A0);
-            this.posXEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.posYEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.posZEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)INCLUDE_ENTITIES_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)INCLUDE_ENTITIES_LABEL)), (int)150, (int)0xA0A0A0);
+        if ($$4 == net.minecraft.world.level.block.state.properties.StructureMode.LOAD || $$4 == net.minecraft.world.level.block.state.properties.StructureMode.SAVE) {
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)POSITION_LABEL, (int)(this.width / 2 - 153), (int)70, (int)0xA0A0A0);
+            this.posXEdit.render($$0, $$1, $$2, $$3);
+            this.posYEdit.render($$0, $$1, $$2, $$3);
+            this.posZEdit.render($$0, $$1, $$2, $$3);
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)INCLUDE_ENTITIES_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)INCLUDE_ENTITIES_LABEL)), (int)150, (int)0xA0A0A0);
         }
-        if (structureMode == net.minecraft.world.level.block.state.properties.StructureMode.SAVE) {
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SIZE_LABEL, (int)(this.width / 2 - 153), (int)110, (int)0xA0A0A0);
-            this.sizeXEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.sizeYEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.sizeZEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)DETECT_SIZE_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)DETECT_SIZE_LABEL)), (int)110, (int)0xA0A0A0);
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SHOW_AIR_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)SHOW_AIR_LABEL)), (int)70, (int)0xA0A0A0);
+        if ($$4 == net.minecraft.world.level.block.state.properties.StructureMode.SAVE) {
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SIZE_LABEL, (int)(this.width / 2 - 153), (int)110, (int)0xA0A0A0);
+            this.sizeXEdit.render($$0, $$1, $$2, $$3);
+            this.sizeYEdit.render($$0, $$1, $$2, $$3);
+            this.sizeZEdit.render($$0, $$1, $$2, $$3);
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)DETECT_SIZE_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)DETECT_SIZE_LABEL)), (int)110, (int)0xA0A0A0);
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SHOW_AIR_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)SHOW_AIR_LABEL)), (int)70, (int)0xA0A0A0);
         }
-        if (structureMode == net.minecraft.world.level.block.state.properties.StructureMode.LOAD) {
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)INTEGRITY_LABEL, (int)(this.width / 2 - 153), (int)110, (int)0xA0A0A0);
-            this.integrityEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.seedEdit.render(matrixStack, mouseX, mouseY, partialTicks);
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SHOW_BOUNDING_BOX_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)SHOW_BOUNDING_BOX_LABEL)), (int)70, (int)0xA0A0A0);
+        if ($$4 == net.minecraft.world.level.block.state.properties.StructureMode.LOAD) {
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)INTEGRITY_LABEL, (int)(this.width / 2 - 153), (int)110, (int)0xA0A0A0);
+            this.integrityEdit.render($$0, $$1, $$2, $$3);
+            this.seedEdit.render($$0, $$1, $$2, $$3);
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)SHOW_BOUNDING_BOX_LABEL, (int)(this.width / 2 + 154 - this.font.width((net.minecraft.network.chat.FormattedText)SHOW_BOUNDING_BOX_LABEL)), (int)70, (int)0xA0A0A0);
         }
-        if (structureMode == net.minecraft.world.level.block.state.properties.StructureMode.DATA) {
-            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)CUSTOM_DATA_LABEL, (int)(this.width / 2 - 153), (int)110, (int)0xA0A0A0);
-            this.dataEdit.render(matrixStack, mouseX, mouseY, partialTicks);
+        if ($$4 == net.minecraft.world.level.block.state.properties.StructureMode.DATA) {
+            net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)CUSTOM_DATA_LABEL, (int)(this.width / 2 - 153), (int)110, (int)0xA0A0A0);
+            this.dataEdit.render($$0, $$1, $$2, $$3);
         }
-        net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)structureMode.getDisplayName(), (int)(this.width / 2 - 153), (int)174, (int)0xA0A0A0);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        net.minecraft.client.gui.screens.inventory.StructureBlockEditScreen.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)$$4.getDisplayName(), (int)(this.width / 2 - 153), (int)174, (int)0xA0A0A0);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     public boolean isPauseScreen() {