--- net/minecraft/util/profiling/EmptyProfileResults.java
+++ net/minecraft/util/profiling/EmptyProfileResults.java
@@ -11,7 +11,7 @@
     public net.minecraft.util.profiling.EmptyProfileResults() {
     }
 
-    public java.util.List<net.minecraft.util.profiling.ResultField> getTimes(java.lang.String sectionPath) {
+    public java.util.List<net.minecraft.util.profiling.ResultField> getTimes(java.lang.String $$0) {
         return java.util.Collections.emptyList();
     }
 