--- net/minecraft/util/profiling/InactiveProfiler.java
+++ net/minecraft/util/profiling/InactiveProfiler.java
@@ -17,10 +17,10 @@
     public void endTick() {
     }
 
-    public void push(java.lang.String name) {
+    public void push(java.lang.String $$0) {
     }
 
-    public void push(java.util.function.Supplier<java.lang.String> nameSupplier) {
+    public void push(java.util.function.Supplier<java.lang.String> $$0) {
     }
 
     public void markForCharting(net.minecraft.util.profiling.metrics.MetricCategory $$0) {
@@ -29,10 +29,10 @@
     public void pop() {
     }
 
-    public void popPush(java.lang.String name) {
+    public void popPush(java.lang.String $$0) {
     }
 
-    public void popPush(java.util.function.Supplier<java.lang.String> nameSupplier) {
+    public void popPush(java.util.function.Supplier<java.lang.String> $$0) {
     }
 
     public void incrementCounter(java.lang.String $$0, int $$1) {