--- net/minecraft/util/profiling/ProfilerFiller$1.java
+++ net/minecraft/util/profiling/ProfilerFiller$1.java
@@ -24,14 +24,14 @@
         this.val$second.endTick();
     }
 
-    public void push(java.lang.String name) {
-        this.val$first.push(name);
-        this.val$second.push(name);
+    public void push(java.lang.String $$0) {
+        this.val$first.push($$0);
+        this.val$second.push($$0);
     }
 
-    public void push(java.util.function.Supplier<java.lang.String> nameSupplier) {
-        this.val$first.push(nameSupplier);
-        this.val$second.push(nameSupplier);
+    public void push(java.util.function.Supplier<java.lang.String> $$0) {
+        this.val$first.push($$0);
+        this.val$second.push($$0);
     }
 
     public void markForCharting(net.minecraft.util.profiling.metrics.MetricCategory $$0) {
@@ -44,14 +44,14 @@
         this.val$second.pop();
     }
 
-    public void popPush(java.lang.String name) {
-        this.val$first.popPush(name);
-        this.val$second.popPush(name);
+    public void popPush(java.lang.String $$0) {
+        this.val$first.popPush($$0);
+        this.val$second.popPush($$0);
     }
 
-    public void popPush(java.util.function.Supplier<java.lang.String> nameSupplier) {
-        this.val$first.popPush(nameSupplier);
-        this.val$second.popPush(nameSupplier);
+    public void popPush(java.util.function.Supplier<java.lang.String> $$0) {
+        this.val$first.popPush($$0);
+        this.val$second.popPush($$0);
     }
 
     public void incrementCounter(java.lang.String $$0, int $$1) {