--- net/minecraft/util/profiling/ResultField.java
+++ net/minecraft/util/profiling/ResultField.java
@@ -11,11 +11,11 @@
     public long count;
     public java.lang.String name;
 
-    public net.minecraft.util.profiling.ResultField(java.lang.String name, double percentage, double globalPercentage, long count) {
-        this.name = name;
-        this.percentage = percentage;
-        this.globalPercentage = globalPercentage;
-        this.count = count;
+    public net.minecraft.util.profiling.ResultField(java.lang.String $$0, double $$1, double $$2, long $$3) {
+        this.name = $$0;
+        this.percentage = $$1;
+        this.globalPercentage = $$2;
+        this.count = $$3;
     }
 
     public int compareTo(net.minecraft.util.profiling.ResultField $$0) {