--- com/mojang/blaze3d/audio/Library$CountingChannelPool.java
+++ com/mojang/blaze3d/audio/Library$CountingChannelPool.java
@@ -9,9 +9,9 @@
     public int limit;
     public java.util.Set<com.mojang.blaze3d.audio.Channel> activeChannels;
 
-    public com.mojang.blaze3d.audio.Library$CountingChannelPool(int limit) {
+    public com.mojang.blaze3d.audio.Library$CountingChannelPool(int $$0) {
         this.activeChannels = com.google.common.collect.Sets.newIdentityHashSet();
-        this.limit = limit;
+        this.limit = $$0;
     }
 
     @javax.annotation.Nullable
@@ -29,11 +29,11 @@
         return $$0;
     }
 
-    public boolean release(com.mojang.blaze3d.audio.Channel source) {
-        if (!this.activeChannels.remove((java.lang.Object)source)) {
+    public boolean release(com.mojang.blaze3d.audio.Channel $$0) {
+        if (!this.activeChannels.remove((java.lang.Object)$$0)) {
             return false;
         }
-        source.destroy();
+        $$0.destroy();
         return true;
     }
 