--- com/mojang/blaze3d/audio/Listener.java
+++ com/mojang/blaze3d/audio/Listener.java
@@ -13,22 +13,22 @@
         this.position = net.minecraft.world.phys.Vec3.ZERO;
     }
 
-    public void setListenerPosition(net.minecraft.world.phys.Vec3 position) {
-        this.position = position;
-        org.lwjgl.openal.AL10.alListener3f((int)4100, (float)((float)position.x), (float)((float)position.y), (float)((float)position.z));
+    public void setListenerPosition(net.minecraft.world.phys.Vec3 $$0) {
+        this.position = $$0;
+        org.lwjgl.openal.AL10.alListener3f((int)4100, (float)((float)$$0.x), (float)((float)$$0.y), (float)((float)$$0.z));
     }
 
     public net.minecraft.world.phys.Vec3 getListenerPosition() {
         return this.position;
     }
 
-    public void setListenerOrientation(com.mojang.math.Vector3f clientViewVector, com.mojang.math.Vector3f viewVectorRaised) {
-        org.lwjgl.openal.AL10.alListenerfv((int)4111, (float[])new float[]{clientViewVector.x(), clientViewVector.y(), clientViewVector.z(), viewVectorRaised.x(), viewVectorRaised.y(), viewVectorRaised.z()});
+    public void setListenerOrientation(com.mojang.math.Vector3f $$0, com.mojang.math.Vector3f $$1) {
+        org.lwjgl.openal.AL10.alListenerfv((int)4111, (float[])new float[]{$$0.x(), $$0.y(), $$0.z(), $$1.x(), $$1.y(), $$1.z()});
     }
 
-    public void setGain(float gain) {
-        org.lwjgl.openal.AL10.alListenerf((int)4106, (float)gain);
-        this.gain = gain;
+    public void setGain(float $$0) {
+        org.lwjgl.openal.AL10.alListenerf((int)4106, (float)$$0);
+        this.gain = $$0;
     }
 
     public float getGain() {