--- com/mojang/blaze3d/audio/OggAudioStream$OutputConcat.java
+++ com/mojang/blaze3d/audio/OggAudioStream$OutputConcat.java
@@ -10,9 +10,9 @@
     public int byteCount;
     public java.nio.ByteBuffer currentBuffer;
 
-    public com.mojang.blaze3d.audio.OggAudioStream$OutputConcat(int size) {
+    public com.mojang.blaze3d.audio.OggAudioStream$OutputConcat(int $$0) {
         this.buffers = com.google.common.collect.Lists.newArrayList();
-        this.bufferSize = size + 1 & 0xFFFFFFFE;
+        this.bufferSize = $$0 + 1 & 0xFFFFFFFE;
         this.createNewBuffer();
     }
 
@@ -20,14 +20,14 @@
         this.currentBuffer = org.lwjgl.BufferUtils.createByteBuffer((int)this.bufferSize);
     }
 
-    public void put(float sample) {
+    public void put(float $$0) {
         if (this.currentBuffer.remaining() == 0) {
             this.currentBuffer.flip();
             this.buffers.add((java.lang.Object)this.currentBuffer);
             this.createNewBuffer();
         }
-        int n = net.minecraft.util.Mth.clamp((int)((int)(sample * 32767.5f - 0.5f)), (int)java.lang.Short.MIN_VALUE, (int)java.lang.Short.MAX_VALUE);
-        this.currentBuffer.putShort((short)n);
+        int $$1 = net.minecraft.util.Mth.clamp((int)((int)($$0 * 32767.5f - 0.5f)), (int)java.lang.Short.MIN_VALUE, (int)java.lang.Short.MAX_VALUE);
+        this.currentBuffer.putShort((short)$$1);
         this.byteCount += 2;
     }
 