--- com/mojang/blaze3d/audio/OpenAlUtil.java
+++ com/mojang/blaze3d/audio/OpenAlUtil.java
@@ -10,8 +10,8 @@
     public com.mojang.blaze3d.audio.OpenAlUtil() {
     }
 
-    public static java.lang.String alErrorToString(int errorCode) {
-        switch (errorCode) {
+    public static java.lang.String alErrorToString(int $$0) {
+        switch ($$0) {
             case 40961: {
                 return "Invalid name parameter.";
             }
@@ -31,17 +31,17 @@
         return "An unrecognized error occurred.";
     }
 
-    public static boolean checkALError(java.lang.String operationState) {
-        int n = org.lwjgl.openal.AL10.alGetError();
-        if (n != 0) {
-            LOGGER.error("{}: {}", (java.lang.Object)operationState, (java.lang.Object)com.mojang.blaze3d.audio.OpenAlUtil.alErrorToString((int)n));
+    public static boolean checkALError(java.lang.String $$0) {
+        int $$1 = org.lwjgl.openal.AL10.alGetError();
+        if ($$1 != 0) {
+            LOGGER.error("{}: {}", (java.lang.Object)$$0, (java.lang.Object)com.mojang.blaze3d.audio.OpenAlUtil.alErrorToString((int)$$1));
             return true;
         }
         return false;
     }
 
-    public static java.lang.String alcErrorToString(int errorCode) {
-        switch (errorCode) {
+    public static java.lang.String alcErrorToString(int $$0) {
+        switch ($$0) {
             case 40961: {
                 return "Invalid device.";
             }
@@ -61,37 +61,37 @@
         return "An unrecognized error occurred.";
     }
 
-    public static boolean checkALCError(long deviceHandle, java.lang.String operationState) {
-        int n = org.lwjgl.openal.ALC10.alcGetError((long)deviceHandle);
-        if (n != 0) {
-            LOGGER.error("{}{}: {}", new java.lang.Object[]{operationState, deviceHandle, com.mojang.blaze3d.audio.OpenAlUtil.alcErrorToString((int)n)});
+    public static boolean checkALCError(long $$0, java.lang.String $$1) {
+        int $$2 = org.lwjgl.openal.ALC10.alcGetError((long)$$0);
+        if ($$2 != 0) {
+            LOGGER.error("{}{}: {}", new java.lang.Object[]{$$1, $$0, com.mojang.blaze3d.audio.OpenAlUtil.alcErrorToString((int)$$2)});
             return true;
         }
         return false;
     }
 
-    public static int audioFormatToOpenAl(javax.sound.sampled.AudioFormat format) {
-        javax.sound.sampled.AudioFormat$Encoding encoding = format.getEncoding();
-        int n = format.getChannels();
-        int n2 = format.getSampleSizeInBits();
-        if (encoding.equals((java.lang.Object)javax.sound.sampled.AudioFormat$Encoding.PCM_UNSIGNED) || encoding.equals((java.lang.Object)javax.sound.sampled.AudioFormat$Encoding.PCM_SIGNED)) {
-            if (n == 1) {
-                if (n2 == 8) {
+    public static int audioFormatToOpenAl(javax.sound.sampled.AudioFormat $$0) {
+        javax.sound.sampled.AudioFormat$Encoding $$1 = $$0.getEncoding();
+        int $$2 = $$0.getChannels();
+        int $$3 = $$0.getSampleSizeInBits();
+        if ($$1.equals((java.lang.Object)javax.sound.sampled.AudioFormat$Encoding.PCM_UNSIGNED) || $$1.equals((java.lang.Object)javax.sound.sampled.AudioFormat$Encoding.PCM_SIGNED)) {
+            if ($$2 == 1) {
+                if ($$3 == 8) {
                     return 4352;
                 }
-                if (n2 == 16) {
+                if ($$3 == 16) {
                     return 4353;
                 }
-            } else if (n == 2) {
-                if (n2 == 8) {
+            } else if ($$2 == 2) {
+                if ($$3 == 8) {
                     return 4354;
                 }
-                if (n2 == 16) {
+                if ($$3 == 16) {
                     return 4355;
                 }
             }
         }
-        throw new java.lang.IllegalArgumentException("Invalid audio format: " + format);
+        throw new java.lang.IllegalArgumentException("Invalid audio format: " + $$0);
     }
 
     static {