--- com/mojang/blaze3d/audio/SoundBuffer.java
+++ com/mojang/blaze3d/audio/SoundBuffer.java
@@ -11,9 +11,9 @@
     public boolean hasAlBuffer;
     public int alBuffer;
 
-    public com.mojang.blaze3d.audio.SoundBuffer(java.nio.ByteBuffer data, javax.sound.sampled.AudioFormat format) {
-        this.data = data;
-        this.format = format;
+    public com.mojang.blaze3d.audio.SoundBuffer(java.nio.ByteBuffer $$0, javax.sound.sampled.AudioFormat $$1) {
+        this.data = $$0;
+        this.format = $$1;
     }
 
     public java.util.OptionalInt getAlBuffer() {