--- net/minecraft/world/level/levelgen/structure/templatesystem/AxisAlignedLinearPosTest.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/AxisAlignedLinearPosTest.java
@@ -13,15 +13,15 @@
     public int maxDist;
     public net.minecraft.core.Direction$Axis axis;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.AxisAlignedLinearPosTest(float minChance, float maxChance, int minDist, int maxDist, net.minecraft.core.Direction$Axis axis) {
-        if (minDist >= maxDist) {
-            throw new java.lang.IllegalArgumentException("Invalid range: [" + minDist + "," + maxDist + "]");
+    public net.minecraft.world.level.levelgen.structure.templatesystem.AxisAlignedLinearPosTest(float $$0, float $$1, int $$2, int $$3, net.minecraft.core.Direction$Axis $$4) {
+        if ($$2 >= $$3) {
+            throw new java.lang.IllegalArgumentException("Invalid range: [" + $$2 + "," + $$3 + "]");
         }
-        this.minChance = minChance;
-        this.maxChance = maxChance;
-        this.minDist = minDist;
-        this.maxDist = maxDist;
-        this.axis = axis;
+        this.minChance = $$0;
+        this.maxChance = $$1;
+        this.minDist = $$2;
+        this.maxDist = $$3;
+        this.axis = $$4;
     }
 
     public boolean test(net.minecraft.core.BlockPos $$0, net.minecraft.core.BlockPos $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {