--- net/minecraft/world/level/levelgen/structure/templatesystem/BlockMatchTest.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/BlockMatchTest.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.structure.templatesystem.BlockMatchTest> CODEC;
     public net.minecraft.world.level.block.Block block;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.BlockMatchTest(net.minecraft.world.level.block.Block block) {
-        this.block = block;
+    public net.minecraft.world.level.levelgen.structure.templatesystem.BlockMatchTest(net.minecraft.world.level.block.Block $$0) {
+        this.block = $$0;
     }
 
     public boolean test(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.util.RandomSource $$1) {