--- net/minecraft/world/level/levelgen/structure/templatesystem/BlockIgnoreProcessor.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/BlockIgnoreProcessor.java
@@ -12,16 +12,16 @@
     public static net.minecraft.world.level.levelgen.structure.templatesystem.BlockIgnoreProcessor STRUCTURE_AND_AIR;
     public com.google.common.collect.ImmutableList<net.minecraft.world.level.block.Block> toIgnore;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.BlockIgnoreProcessor(java.util.List<net.minecraft.world.level.block.Block> ignoredBlocks) {
-        this.toIgnore = com.google.common.collect.ImmutableList.copyOf(ignoredBlocks);
+    public net.minecraft.world.level.levelgen.structure.templatesystem.BlockIgnoreProcessor(java.util.List<net.minecraft.world.level.block.Block> $$0) {
+        this.toIgnore = com.google.common.collect.ImmutableList.copyOf($$0);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo processBlock(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos blockPos, net.minecraft.core.BlockPos blockPos2, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo structureBlockInfo, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo structureBlockInfo2, net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings settings) {
-        if (this.toIgnore.contains((java.lang.Object)structureBlockInfo2.state.getBlock())) {
+    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo processBlock(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo $$3, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo $$4, net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings $$5) {
+        if (this.toIgnore.contains((java.lang.Object)$$4.state.getBlock())) {
             return null;
         }
-        return structureBlockInfo2;
+        return $$4;
     }
 
     public net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessorType<?> getType() {