--- net/minecraft/world/level/levelgen/structure/templatesystem/BlockAgeProcessor.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/BlockAgeProcessor.java
@@ -13,31 +13,31 @@
     public static net.minecraft.world.level.block.state.BlockState[] NON_MOSSY_REPLACEMENTS;
     public float mossiness;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.BlockAgeProcessor(float mossiness) {
-        this.mossiness = mossiness;
+    public net.minecraft.world.level.levelgen.structure.templatesystem.BlockAgeProcessor(float $$0) {
+        this.mossiness = $$0;
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo processBlock(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos blockPos, net.minecraft.core.BlockPos blockPos2, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo structureBlockInfo, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo structureBlockInfo2, net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings settings) {
-        net.minecraft.util.RandomSource randomSource = settings.getRandom(structureBlockInfo2.pos);
-        net.minecraft.world.level.block.state.BlockState blockState = structureBlockInfo2.state;
-        net.minecraft.core.BlockPos blockPos3 = structureBlockInfo2.pos;
-        net.minecraft.world.level.block.state.BlockState blockState2 = null;
-        if (blockState.is(net.minecraft.world.level.block.Blocks.STONE_BRICKS) || blockState.is(net.minecraft.world.level.block.Blocks.STONE) || blockState.is(net.minecraft.world.level.block.Blocks.CHISELED_STONE_BRICKS)) {
-            blockState2 = this.maybeReplaceFullStoneBlock(randomSource);
-        } else if (blockState.is(net.minecraft.tags.BlockTags.STAIRS)) {
-            blockState2 = this.maybeReplaceStairs(randomSource, structureBlockInfo2.state);
-        } else if (blockState.is(net.minecraft.tags.BlockTags.SLABS)) {
-            blockState2 = this.maybeReplaceSlab(randomSource);
-        } else if (blockState.is(net.minecraft.tags.BlockTags.WALLS)) {
-            blockState2 = this.maybeReplaceWall(randomSource);
-        } else if (blockState.is(net.minecraft.world.level.block.Blocks.OBSIDIAN)) {
-            blockState2 = this.maybeReplaceObsidian(randomSource);
+    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo processBlock(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo $$3, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo $$4, net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings $$5) {
+        net.minecraft.util.RandomSource $$6 = $$5.getRandom($$4.pos);
+        net.minecraft.world.level.block.state.BlockState $$7 = $$4.state;
+        net.minecraft.core.BlockPos $$8 = $$4.pos;
+        net.minecraft.world.level.block.state.BlockState $$9 = null;
+        if ($$7.is(net.minecraft.world.level.block.Blocks.STONE_BRICKS) || $$7.is(net.minecraft.world.level.block.Blocks.STONE) || $$7.is(net.minecraft.world.level.block.Blocks.CHISELED_STONE_BRICKS)) {
+            $$9 = this.maybeReplaceFullStoneBlock($$6);
+        } else if ($$7.is(net.minecraft.tags.BlockTags.STAIRS)) {
+            $$9 = this.maybeReplaceStairs($$6, $$4.state);
+        } else if ($$7.is(net.minecraft.tags.BlockTags.SLABS)) {
+            $$9 = this.maybeReplaceSlab($$6);
+        } else if ($$7.is(net.minecraft.tags.BlockTags.WALLS)) {
+            $$9 = this.maybeReplaceWall($$6);
+        } else if ($$7.is(net.minecraft.world.level.block.Blocks.OBSIDIAN)) {
+            $$9 = this.maybeReplaceObsidian($$6);
         }
-        if (blockState2 != null) {
-            return new net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo(blockPos3, blockState2, structureBlockInfo2.nbt);
+        if ($$9 != null) {
+            return new net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo($$8, $$9, $$4.nbt);
         }
-        return structureBlockInfo2;
+        return $$4;
     }
 
     @javax.annotation.Nullable