--- net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/GravityProcessor.java
@@ -10,17 +10,28 @@
     public net.minecraft.world.level.levelgen.Heightmap$Types heightmap;
     public int offset;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.GravityProcessor(net.minecraft.world.level.levelgen.Heightmap$Types heightmap, int offset) {
-        this.heightmap = heightmap;
-        this.offset = offset;
+    public net.minecraft.world.level.levelgen.structure.templatesystem.GravityProcessor(net.minecraft.world.level.levelgen.Heightmap$Types $$0, int $$1) {
+        this.heightmap = $$0;
+        this.offset = $$1;
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo processBlock(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos blockPos, net.minecraft.core.BlockPos blockPos2, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo structureBlockInfo, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo structureBlockInfo2, net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings settings) {
-        net.minecraft.world.level.levelgen.Heightmap$Types types = level instanceof net.minecraft.server.level.ServerLevel ? (this.heightmap == net.minecraft.world.level.levelgen.Heightmap$Types.WORLD_SURFACE_WG ? net.minecraft.world.level.levelgen.Heightmap$Types.WORLD_SURFACE : (this.heightmap == net.minecraft.world.level.levelgen.Heightmap$Types.OCEAN_FLOOR_WG ? net.minecraft.world.level.levelgen.Heightmap$Types.OCEAN_FLOOR : this.heightmap)) : this.heightmap;
-        int n = level.getHeight(types, structureBlockInfo2.pos.getX(), structureBlockInfo2.pos.getZ()) + this.offset;
-        int n2 = structureBlockInfo.pos.getY();
-        return new net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo(new net.minecraft.core.BlockPos(structureBlockInfo2.pos.getX(), n + n2, structureBlockInfo2.pos.getZ()), structureBlockInfo2.state, structureBlockInfo2.nbt);
+    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo processBlock(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo $$3, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo $$4, net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings $$5) {
+        net.minecraft.world.level.levelgen.Heightmap$Types $$9;
+        if ($$0 instanceof net.minecraft.server.level.ServerLevel) {
+            if (this.heightmap == net.minecraft.world.level.levelgen.Heightmap$Types.WORLD_SURFACE_WG) {
+                net.minecraft.world.level.levelgen.Heightmap$Types $$6 = net.minecraft.world.level.levelgen.Heightmap$Types.WORLD_SURFACE;
+            } else if (this.heightmap == net.minecraft.world.level.levelgen.Heightmap$Types.OCEAN_FLOOR_WG) {
+                net.minecraft.world.level.levelgen.Heightmap$Types $$7 = net.minecraft.world.level.levelgen.Heightmap$Types.OCEAN_FLOOR;
+            } else {
+                net.minecraft.world.level.levelgen.Heightmap$Types $$8 = this.heightmap;
+            }
+        } else {
+            $$9 = this.heightmap;
+        }
+        int $$10 = $$0.getHeight($$9, $$4.pos.getX(), $$4.pos.getZ()) + this.offset;
+        int $$11 = $$3.pos.getY();
+        return new net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo(new net.minecraft.core.BlockPos($$4.pos.getX(), $$10 + $$11, $$4.pos.getZ()), $$4.state, $$4.nbt);
     }
 
     public net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessorType<?> getType() {