--- net/minecraft/world/level/levelgen/structure/templatesystem/JigsawReplacementProcessor.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/JigsawReplacementProcessor.java
@@ -24,7 +24,7 @@
         }
         java.lang.String $$7 = $$4.nbt.getString("final_state");
         try {
-            net.minecraft.commands.arguments.blocks.BlockStateParser$BlockResult $$8 = net.minecraft.commands.arguments.blocks.BlockStateParser.parseForBlock((net.minecraft.core.Registry)net.minecraft.core.Registry.BLOCK, (java.lang.String)$$7, (boolean)true);
+            net.minecraft.commands.arguments.blocks.BlockStateParser$BlockResult $$8 = net.minecraft.commands.arguments.blocks.BlockStateParser.parseForBlock((net.minecraft.core.HolderLookup)$$0.holderLookup(net.minecraft.core.Registry.BLOCK_REGISTRY), (java.lang.String)$$7, (boolean)true);
             net.minecraft.world.level.block.state.BlockState $$9 = $$8.blockState();
         }
         catch (com.mojang.brigadier.exceptions.CommandSyntaxException $$10) {