--- net/minecraft/world/level/levelgen/structure/templatesystem/ProcessorRule.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/ProcessorRule.java
@@ -13,20 +13,20 @@
     @javax.annotation.Nullable
     public net.minecraft.nbt.CompoundTag outputTag;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.ProcessorRule(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest inputPredicate, net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest locPredicate, net.minecraft.world.level.block.state.BlockState outputState) {
-        this(inputPredicate, locPredicate, (net.minecraft.world.level.levelgen.structure.templatesystem.PosRuleTest)net.minecraft.world.level.levelgen.structure.templatesystem.PosAlwaysTrueTest.INSTANCE, outputState, java.util.Optional.empty());
+    public net.minecraft.world.level.levelgen.structure.templatesystem.ProcessorRule(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$0, net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$1, net.minecraft.world.level.block.state.BlockState $$2) {
+        this($$0, $$1, (net.minecraft.world.level.levelgen.structure.templatesystem.PosRuleTest)net.minecraft.world.level.levelgen.structure.templatesystem.PosAlwaysTrueTest.INSTANCE, $$2, java.util.Optional.empty());
     }
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.ProcessorRule(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest inputPredicate, net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest locPredicate, net.minecraft.world.level.levelgen.structure.templatesystem.PosRuleTest posPredicate, net.minecraft.world.level.block.state.BlockState outputState) {
-        this(inputPredicate, locPredicate, posPredicate, outputState, java.util.Optional.empty());
+    public net.minecraft.world.level.levelgen.structure.templatesystem.ProcessorRule(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$0, net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$1, net.minecraft.world.level.levelgen.structure.templatesystem.PosRuleTest $$2, net.minecraft.world.level.block.state.BlockState $$3) {
+        this($$0, $$1, $$2, $$3, java.util.Optional.empty());
     }
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.ProcessorRule(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest inputPredicate, net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest locPredicate, net.minecraft.world.level.levelgen.structure.templatesystem.PosRuleTest posPredicate, net.minecraft.world.level.block.state.BlockState outputState, java.util.Optional<net.minecraft.nbt.CompoundTag> outputTag) {
-        this.inputPredicate = inputPredicate;
-        this.locPredicate = locPredicate;
-        this.posPredicate = posPredicate;
-        this.outputState = outputState;
-        this.outputTag = (net.minecraft.nbt.CompoundTag)outputTag.orElse(null);
+    public net.minecraft.world.level.levelgen.structure.templatesystem.ProcessorRule(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$0, net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$1, net.minecraft.world.level.levelgen.structure.templatesystem.PosRuleTest $$2, net.minecraft.world.level.block.state.BlockState $$3, java.util.Optional<net.minecraft.nbt.CompoundTag> $$4) {
+        this.inputPredicate = $$0;
+        this.locPredicate = $$1;
+        this.posPredicate = $$2;
+        this.outputState = $$3;
+        this.outputTag = (net.minecraft.nbt.CompoundTag)$$4.orElse(null);
     }
 
     public boolean test(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.state.BlockState $$1, net.minecraft.core.BlockPos $$2, net.minecraft.core.BlockPos $$3, net.minecraft.core.BlockPos $$4, net.minecraft.util.RandomSource $$5) {