--- net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate$StructureBlockInfo.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate$StructureBlockInfo.java
@@ -9,10 +9,10 @@
     public net.minecraft.world.level.block.state.BlockState state;
     public net.minecraft.nbt.CompoundTag nbt;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag nbt) {
-        this.pos = pos;
-        this.state = state;
-        this.nbt = nbt;
+    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate$StructureBlockInfo(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$2) {
+        this.pos = $$0;
+        this.state = $$1;
+        this.nbt = $$2;
     }
 
     public java.lang.String toString() {