--- net/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorList.java
+++ net/minecraft/world/level/levelgen/structure/templatesystem/StructureProcessorList.java
@@ -7,8 +7,8 @@
  {
     public java.util.List<net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor> list;
 
-    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessorList(java.util.List<net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor> list) {
-        this.list = list;
+    public net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessorList(java.util.List<net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor> $$0) {
+        this.list = $$0;
     }
 
     public java.util.List<net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor> list() {