--- net/minecraft/world/entity/ai/attributes/AttributeSupplier$Builder.java
+++ net/minecraft/world/entity/ai/attributes/AttributeSupplier$Builder.java
@@ -12,24 +12,24 @@
         this.builder = com.google.common.collect.Maps.newHashMap();
     }
 
-    public net.minecraft.world.entity.ai.attributes.AttributeInstance create(net.minecraft.world.entity.ai.attributes.Attribute attribute) {
-        net.minecraft.world.entity.ai.attributes.AttributeInstance attributeInstance = new net.minecraft.world.entity.ai.attributes.AttributeInstance(attribute, $$1 -> {
+    public net.minecraft.world.entity.ai.attributes.AttributeInstance create(net.minecraft.world.entity.ai.attributes.Attribute $$0) {
+        net.minecraft.world.entity.ai.attributes.AttributeInstance $$12 = new net.minecraft.world.entity.ai.attributes.AttributeInstance($$0, $$1 -> {
             if (this.instanceFrozen) {
-                throw new java.lang.UnsupportedOperationException("Tried to change value for default attribute instance: " + net.minecraft.core.Registry.ATTRIBUTE.getKey((java.lang.Object)attribute));
+                throw new java.lang.UnsupportedOperationException("Tried to change value for default attribute instance: " + net.minecraft.core.Registry.ATTRIBUTE.getKey((java.lang.Object)$$0));
             }
         });
-        this.builder.put((java.lang.Object)attribute, (java.lang.Object)attributeInstance);
-        return attributeInstance;
+        this.builder.put((java.lang.Object)$$0, (java.lang.Object)$$12);
+        return $$12;
     }
 
-    public net.minecraft.world.entity.ai.attributes.AttributeSupplier$Builder add(net.minecraft.world.entity.ai.attributes.Attribute attribute) {
-        this.create(attribute);
+    public net.minecraft.world.entity.ai.attributes.AttributeSupplier$Builder add(net.minecraft.world.entity.ai.attributes.Attribute $$0) {
+        this.create($$0);
         return this;
     }
 
-    public net.minecraft.world.entity.ai.attributes.AttributeSupplier$Builder add(net.minecraft.world.entity.ai.attributes.Attribute attribute, double value) {
-        net.minecraft.world.entity.ai.attributes.AttributeInstance attributeInstance = this.create(attribute);
-        attributeInstance.setBaseValue(value);
+    public net.minecraft.world.entity.ai.attributes.AttributeSupplier$Builder add(net.minecraft.world.entity.ai.attributes.Attribute $$0, double $$1) {
+        net.minecraft.world.entity.ai.attributes.AttributeInstance $$2 = this.create($$0);
+        $$2.setBaseValue($$1);
         return this;
     }
 