--- net/minecraft/world/entity/ai/attributes/AttributeModifier.java
+++ net/minecraft/world/entity/ai/attributes/AttributeModifier.java
@@ -71,14 +71,14 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.entity.ai.attributes.AttributeModifier load(net.minecraft.nbt.CompoundTag nbt) {
+    public static net.minecraft.world.entity.ai.attributes.AttributeModifier load(net.minecraft.nbt.CompoundTag $$0) {
         try {
-            java.util.UUID uUID = nbt.getUUID("UUID");
-            net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation operation = net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation.fromValue((int)nbt.getInt("Operation"));
-            return new net.minecraft.world.entity.ai.attributes.AttributeModifier(uUID, nbt.getString("Name"), nbt.getDouble("Amount"), operation);
+            java.util.UUID $$1 = $$0.getUUID("UUID");
+            net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation $$2 = net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation.fromValue((int)$$0.getInt("Operation"));
+            return new net.minecraft.world.entity.ai.attributes.AttributeModifier($$1, $$0.getString("Name"), $$0.getDouble("Amount"), $$2);
         }
-        catch (java.lang.Exception exception) {
-            LOGGER.warn("Unable to create attribute: {}", (java.lang.Object)exception.getMessage());
+        catch (java.lang.Exception $$3) {
+            LOGGER.warn("Unable to create attribute: {}", (java.lang.Object)$$3.getMessage());
             return null;
         }
     }