--- net/minecraft/world/entity/ai/attributes/Attribute.java
+++ net/minecraft/world/entity/ai/attributes/Attribute.java
@@ -10,9 +10,9 @@
     public boolean syncable;
     public java.lang.String descriptionId;
 
-    public net.minecraft.world.entity.ai.attributes.Attribute(java.lang.String descriptionId, double defaultValue) {
-        this.defaultValue = defaultValue;
-        this.descriptionId = descriptionId;
+    public net.minecraft.world.entity.ai.attributes.Attribute(java.lang.String $$0, double $$1) {
+        this.defaultValue = $$1;
+        this.descriptionId = $$0;
     }
 
     public double getDefaultValue() {
@@ -23,13 +23,13 @@
         return this.syncable;
     }
 
-    public net.minecraft.world.entity.ai.attributes.Attribute setSyncable(boolean watch) {
-        this.syncable = watch;
+    public net.minecraft.world.entity.ai.attributes.Attribute setSyncable(boolean $$0) {
+        this.syncable = $$0;
         return this;
     }
 
-    public double sanitizeValue(double value) {
-        return value;
+    public double sanitizeValue(double $$0) {
+        return $$0;
     }
 
     public java.lang.String getDescriptionId() {