--- net/minecraft/world/entity/ai/attributes/AttributeModifier$Operation.java
+++ net/minecraft/world/entity/ai/attributes/AttributeModifier$Operation.java
@@ -30,11 +30,11 @@
         return this.value;
     }
 
-    public static net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation fromValue(int id) {
-        if (id < 0 || id >= OPERATIONS.length) {
-            throw new java.lang.IllegalArgumentException("No operation with value " + id);
+    public static net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation fromValue(int $$0) {
+        if ($$0 < 0 || $$0 >= OPERATIONS.length) {
+            throw new java.lang.IllegalArgumentException("No operation with value " + $$0);
         }
-        return OPERATIONS[id];
+        return OPERATIONS[$$0];
     }
 
     public static /* synthetic */ net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation[] $values() {