--- net/minecraft/world/phys/Vec2.java
+++ net/minecraft/world/phys/Vec2.java
@@ -16,9 +16,9 @@
     public float x;
     public float y;
 
-    public net.minecraft.world.phys.Vec2(float x, float y) {
-        this.x = x;
-        this.y = y;
+    public net.minecraft.world.phys.Vec2(float $$0, float $$1) {
+        this.x = $$0;
+        this.y = $$1;
     }
 
     public net.minecraft.world.phys.Vec2 scale(float $$0) {
@@ -37,8 +37,8 @@
         return new net.minecraft.world.phys.Vec2(this.x + $$0, this.y + $$0);
     }
 
-    public boolean equals(net.minecraft.world.phys.Vec2 other) {
-        return this.x == other.x && this.y == other.y;
+    public boolean equals(net.minecraft.world.phys.Vec2 $$0) {
+        return this.x == $$0.x && this.y == $$0.y;
     }
 
     public net.minecraft.world.phys.Vec2 normalized() {