--- net/minecraft/util/datafix/fixes/EntityRavagerRenameFix.java
+++ net/minecraft/util/datafix/fixes/EntityRavagerRenameFix.java
@@ -12,8 +12,8 @@
         super("EntityRavagerRenameFix", $$0, $$1);
     }
 
-    public java.lang.String rename(java.lang.String name) {
-        return java.util.Objects.equals((java.lang.Object)"minecraft:illager_beast", (java.lang.Object)name) ? "minecraft:ravager" : name;
+    public java.lang.String rename(java.lang.String $$0) {
+        return java.util.Objects.equals((java.lang.Object)"minecraft:illager_beast", (java.lang.Object)$$0) ? "minecraft:ravager" : $$0;
     }
 
     static {