--- net/minecraft/util/datafix/fixes/EntityCatSplitFix.java
+++ net/minecraft/util/datafix/fixes/EntityCatSplitFix.java
@@ -10,22 +10,22 @@
         super("EntityCatSplitFix", $$0, $$1);
     }
 
-    public com.mojang.datafixers.util.Pair<java.lang.String, com.mojang.serialization.Dynamic<?>> getNewNameAndTag(java.lang.String name, com.mojang.serialization.Dynamic<?> tag) {
-        if (java.util.Objects.equals((java.lang.Object)"minecraft:ocelot", (java.lang.Object)name)) {
-            int n = tag.get("CatType").asInt(0);
-            if (n == 0) {
-                java.lang.String string = tag.get("Owner").asString("");
-                java.lang.String string2 = tag.get("OwnerUUID").asString("");
-                if (string.length() > 0 || string2.length() > 0) {
-                    tag.set("Trusting", tag.createBoolean(true));
+    public com.mojang.datafixers.util.Pair<java.lang.String, com.mojang.serialization.Dynamic<?>> getNewNameAndTag(java.lang.String $$0, com.mojang.serialization.Dynamic<?> $$1) {
+        if (java.util.Objects.equals((java.lang.Object)"minecraft:ocelot", (java.lang.Object)$$0)) {
+            int $$2 = $$1.get("CatType").asInt(0);
+            if ($$2 == 0) {
+                java.lang.String $$3 = $$1.get("Owner").asString("");
+                java.lang.String $$4 = $$1.get("OwnerUUID").asString("");
+                if ($$3.length() > 0 || $$4.length() > 0) {
+                    $$1.set("Trusting", $$1.createBoolean(true));
                 }
-            } else if (n > 0 && n < 4) {
-                tag = tag.set("CatType", tag.createInt(n));
-                tag = tag.set("OwnerUUID", tag.createString(tag.get("OwnerUUID").asString("")));
-                return com.mojang.datafixers.util.Pair.of((java.lang.Object)"minecraft:cat", (java.lang.Object)tag);
+            } else if ($$2 > 0 && $$2 < 4) {
+                $$1 = $$1.set("CatType", $$1.createInt($$2));
+                $$1 = $$1.set("OwnerUUID", $$1.createString($$1.get("OwnerUUID").asString("")));
+                return com.mojang.datafixers.util.Pair.of((java.lang.Object)"minecraft:cat", (java.lang.Object)$$1);
             }
         }
-        return com.mojang.datafixers.util.Pair.of((java.lang.Object)name, tag);
+        return com.mojang.datafixers.util.Pair.of((java.lang.Object)$$0, $$1);
     }
 }
 