--- net/minecraft/util/datafix/fixes/EntityElderGuardianSplitFix.java
+++ net/minecraft/util/datafix/fixes/EntityElderGuardianSplitFix.java
@@ -10,8 +10,8 @@
         super("EntityElderGuardianSplitFix", $$0, $$1);
     }
 
-    public com.mojang.datafixers.util.Pair<java.lang.String, com.mojang.serialization.Dynamic<?>> getNewNameAndTag(java.lang.String name, com.mojang.serialization.Dynamic<?> tag) {
-        return com.mojang.datafixers.util.Pair.of((java.lang.Object)(java.util.Objects.equals((java.lang.Object)name, (java.lang.Object)"Guardian") && tag.get("Elder").asBoolean(false) ? "ElderGuardian" : name), tag);
+    public com.mojang.datafixers.util.Pair<java.lang.String, com.mojang.serialization.Dynamic<?>> getNewNameAndTag(java.lang.String $$0, com.mojang.serialization.Dynamic<?> $$1) {
+        return com.mojang.datafixers.util.Pair.of((java.lang.Object)(java.util.Objects.equals((java.lang.Object)$$0, (java.lang.Object)"Guardian") && $$1.get("Elder").asBoolean(false) ? "ElderGuardian" : $$0), $$1);
     }
 }
 