--- net/minecraft/util/datafix/fixes/EntityCodSalmonFix.java
+++ net/minecraft/util/datafix/fixes/EntityCodSalmonFix.java
@@ -13,8 +13,8 @@
         super("EntityCodSalmonFix", $$0, $$1);
     }
 
-    public java.lang.String rename(java.lang.String name) {
-        return (java.lang.String)RENAMED_IDS.getOrDefault((java.lang.Object)name, (java.lang.Object)name);
+    public java.lang.String rename(java.lang.String $$0) {
+        return (java.lang.String)RENAMED_IDS.getOrDefault((java.lang.Object)$$0, (java.lang.Object)$$0);
     }
 
     static {