--- net/minecraft/util/datafix/fixes/EntityZombifiedPiglinRenameFix.java
+++ net/minecraft/util/datafix/fixes/EntityZombifiedPiglinRenameFix.java
@@ -12,8 +12,8 @@
         super("EntityZombifiedPiglinRenameFix", $$0, true);
     }
 
-    public java.lang.String rename(java.lang.String name) {
-        return java.util.Objects.equals((java.lang.Object)"minecraft:zombie_pigman", (java.lang.Object)name) ? "minecraft:zombified_piglin" : name;
+    public java.lang.String rename(java.lang.String $$0) {
+        return java.util.Objects.equals((java.lang.Object)"minecraft:zombie_pigman", (java.lang.Object)$$0) ? "minecraft:zombified_piglin" : $$0;
     }
 
     static {