--- net/minecraft/util/datafix/fixes/EntityTheRenameningFix.java
+++ net/minecraft/util/datafix/fixes/EntityTheRenameningFix.java
@@ -15,11 +15,11 @@
         super("EntityTheRenameningBlock", $$0, $$1);
     }
 
-    public java.lang.String rename(java.lang.String name) {
-        if (name.startsWith("minecraft:bred_")) {
-            name = "minecraft:" + name.substring("minecraft:bred_".length());
+    public java.lang.String rename(java.lang.String $$0) {
+        if ($$0.startsWith("minecraft:bred_")) {
+            $$0 = "minecraft:" + $$0.substring("minecraft:bred_".length());
         }
-        return (java.lang.String)RENAMED_IDS.getOrDefault((java.lang.Object)name, (java.lang.Object)name);
+        return (java.lang.String)RENAMED_IDS.getOrDefault((java.lang.Object)$$0, (java.lang.Object)$$0);
     }
 
     static {