--- net/minecraft/util/datafix/fixes/EntityPufferfishRenameFix.java
+++ net/minecraft/util/datafix/fixes/EntityPufferfishRenameFix.java
@@ -12,8 +12,8 @@
         super("EntityPufferfishRenameFix", $$0, $$1);
     }
 
-    public java.lang.String rename(java.lang.String name) {
-        return java.util.Objects.equals((java.lang.Object)"minecraft:puffer_fish", (java.lang.Object)name) ? "minecraft:pufferfish" : name;
+    public java.lang.String rename(java.lang.String $$0) {
+        return java.util.Objects.equals((java.lang.Object)"minecraft:puffer_fish", (java.lang.Object)$$0) ? "minecraft:pufferfish" : $$0;
     }
 
     static {