--- net/minecraft/util/datafix/fixes/OptionsProgrammerArtFix.java
+++ net/minecraft/util/datafix/fixes/OptionsProgrammerArtFix.java
@@ -1,0 +1,21 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.util.datafix.fixes;
+
+public class net.minecraft.util.datafix.fixes.OptionsProgrammerArtFix
+extends com.mojang.datafixers.DataFix
+ {
+    public net.minecraft.util.datafix.fixes.OptionsProgrammerArtFix(com.mojang.datafixers.schemas.Schema $$0) {
+        super($$0, false);
+    }
+
+    public com.mojang.datafixers.TypeRewriteRule makeRule() {
+        return this.fixTypeEverywhereTyped("OptionsProgrammerArtFix", this.getInputSchema().getType(net.minecraft.util.datafix.fixes.References.OPTIONS), $$02 -> $$02.update(com.mojang.datafixers.DSL.remainderFinder(), $$0 -> $$0.update("resourcePacks", arg_0 -> this.fixList(arg_0)).update("incompatibleResourcePacks", arg_0 -> this.fixList(arg_0))));
+    }
+
+    public <T> com.mojang.serialization.Dynamic<T> fixList(com.mojang.serialization.Dynamic<T> $$0) {
+        return (com.mojang.serialization.Dynamic)$$0.asString().result().map($$1 -> $$0.createString($$1.replace((java.lang.CharSequence)"\"programer_art\"", (java.lang.CharSequence)"\"programmer_art\""))).orElse($$0);
+    }
+}
+