--- net/minecraft/world/entity/animal/AbstractFish$FishSwimGoal.java
+++ net/minecraft/world/entity/animal/AbstractFish$FishSwimGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.AbstractFish fish;
 
-    public net.minecraft.world.entity.animal.AbstractFish$FishSwimGoal(net.minecraft.world.entity.animal.AbstractFish fish) {
-        super((net.minecraft.world.entity.PathfinderMob)fish, 1.0, 40);
-        this.fish = fish;
+    public net.minecraft.world.entity.animal.AbstractFish$FishSwimGoal(net.minecraft.world.entity.animal.AbstractFish $$0) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, 1.0, 40);
+        this.fish = $$0;
     }
 
     public boolean canUse() {