--- net/minecraft/world/entity/animal/AbstractGolem.java
+++ net/minecraft/world/entity/animal/AbstractGolem.java
@@ -20,7 +20,7 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return null;
     }
 
@@ -33,7 +33,7 @@
         return 120;
     }
 
-    public boolean removeWhenFarAway(double distanceToClosestPlayer) {
+    public boolean removeWhenFarAway(double $$0) {
         return false;
     }
 }