--- net/minecraft/world/entity/animal/AbstractSchoolingFish$SchoolSpawnGroupData.java
+++ net/minecraft/world/entity/animal/AbstractSchoolingFish$SchoolSpawnGroupData.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.entity.animal.AbstractSchoolingFish leader;
 
-    public net.minecraft.world.entity.animal.AbstractSchoolingFish$SchoolSpawnGroupData(net.minecraft.world.entity.animal.AbstractSchoolingFish leader) {
-        this.leader = leader;
+    public net.minecraft.world.entity.animal.AbstractSchoolingFish$SchoolSpawnGroupData(net.minecraft.world.entity.animal.AbstractSchoolingFish $$0) {
+        this.leader = $$0;
     }
 }
 