--- net/minecraft/world/entity/animal/Bee$BeeGoToHiveGoal.java
+++ net/minecraft/world/entity/animal/Bee$BeeGoToHiveGoal.java
@@ -80,18 +80,18 @@
         this.this$0.pathfindRandomlyTowards(this.this$0.hivePos);
     }
 
-    public boolean pathfindDirectlyTowards(net.minecraft.core.BlockPos pos) {
+    public boolean pathfindDirectlyTowards(net.minecraft.core.BlockPos $$0) {
         net.minecraft.world.entity.animal.Bee.access$1000((net.minecraft.world.entity.animal.Bee)this.this$0).setMaxVisitedNodesMultiplier(10.0f);
-        net.minecraft.world.entity.animal.Bee.access$1100((net.minecraft.world.entity.animal.Bee)this.this$0).moveTo((double)pos.getX(), (double)pos.getY(), (double)pos.getZ(), 1.0);
+        net.minecraft.world.entity.animal.Bee.access$1100((net.minecraft.world.entity.animal.Bee)this.this$0).moveTo((double)$$0.getX(), (double)$$0.getY(), (double)$$0.getZ(), 1.0);
         return net.minecraft.world.entity.animal.Bee.access$1200((net.minecraft.world.entity.animal.Bee)this.this$0).getPath() != null && net.minecraft.world.entity.animal.Bee.access$1300((net.minecraft.world.entity.animal.Bee)this.this$0).getPath().canReach();
     }
 
-    public boolean isTargetBlacklisted(net.minecraft.core.BlockPos pos) {
-        return this.blacklistedTargets.contains((java.lang.Object)pos);
+    public boolean isTargetBlacklisted(net.minecraft.core.BlockPos $$0) {
+        return this.blacklistedTargets.contains((java.lang.Object)$$0);
     }
 
-    public void blacklistTarget(net.minecraft.core.BlockPos pos) {
-        this.blacklistedTargets.add((java.lang.Object)pos);
+    public void blacklistTarget(net.minecraft.core.BlockPos $$0) {
+        this.blacklistedTargets.add((java.lang.Object)$$0);
         while (this.blacklistedTargets.size() > 3) {
             this.blacklistedTargets.remove(0);
         }
@@ -113,12 +113,12 @@
         this.this$0.remainingCooldownBeforeLocatingNewHive = 200;
     }
 
-    public boolean hasReachedTarget(net.minecraft.core.BlockPos pos) {
-        if (this.this$0.closerThan(pos, 2)) {
+    public boolean hasReachedTarget(net.minecraft.core.BlockPos $$0) {
+        if (this.this$0.closerThan($$0, 2)) {
             return true;
         }
-        net.minecraft.world.level.pathfinder.Path path = net.minecraft.world.entity.animal.Bee.access$1400((net.minecraft.world.entity.animal.Bee)this.this$0).getPath();
-        return path != null && path.getTarget().equals((java.lang.Object)pos) && path.canReach() && path.isDone();
+        net.minecraft.world.level.pathfinder.Path $$1 = net.minecraft.world.entity.animal.Bee.access$1400((net.minecraft.world.entity.animal.Bee)this.this$0).getPath();
+        return $$1 != null && $$1.getTarget().equals((java.lang.Object)$$0) && $$1.canReach() && $$1.isDone();
     }
 }
 