--- net/minecraft/world/entity/animal/Bee$BeeBecomeAngryTargetGoal.java
+++ net/minecraft/world/entity/animal/Bee$BeeBecomeAngryTargetGoal.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.entity.animal.Bee$BeeBecomeAngryTargetGoal
 extends net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal<net.minecraft.world.entity.player.Player>
  {
-    public net.minecraft.world.entity.animal.Bee$BeeBecomeAngryTargetGoal(net.minecraft.world.entity.animal.Bee mob) {
-        super((net.minecraft.world.entity.Mob)mob, net.minecraft.world.entity.player.Player.class, 10, true, false, arg_0 -> ((net.minecraft.world.entity.animal.Bee)mob).isAngryAt(arg_0));
+    public net.minecraft.world.entity.animal.Bee$BeeBecomeAngryTargetGoal(net.minecraft.world.entity.animal.Bee $$0) {
+        super((net.minecraft.world.entity.Mob)$$0, net.minecraft.world.entity.player.Player.class, 10, true, false, arg_0 -> ((net.minecraft.world.entity.animal.Bee)$$0).isAngryAt(arg_0));
     }
 
     public boolean canUse() {