--- net/minecraft/world/entity/animal/Bee$BeeAttackGoal.java
+++ net/minecraft/world/entity/animal/Bee$BeeAttackGoal.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.animal.Bee this$0;
 
-    public net.minecraft.world.entity.animal.Bee$BeeAttackGoal(net.minecraft.world.entity.animal.Bee bee, net.minecraft.world.entity.PathfinderMob mob, double speedModifier, boolean followingTargetEvenIfNotSeed) {
+    public net.minecraft.world.entity.animal.Bee$BeeAttackGoal(net.minecraft.world.entity.animal.Bee bee, net.minecraft.world.entity.PathfinderMob $$0, double $$1, boolean $$2) {
         this.this$0 = bee;
-        super(mob, speedModifier, followingTargetEvenIfNotSeed);
+        super($$0, $$1, $$2);
     }
 
     public boolean canUse() {