--- net/minecraft/world/entity/animal/Bee$1.java
+++ net/minecraft/world/entity/animal/Bee$1.java
@@ -13,8 +13,8 @@
         super($$1, $$2);
     }
 
-    public boolean isStableDestination(net.minecraft.core.BlockPos pos) {
-        return !this.level.getBlockState(pos.below()).isAir();
+    public boolean isStableDestination(net.minecraft.core.BlockPos $$0) {
+        return !this.level.getBlockState($$0.below()).isAir();
     }
 
     public void tick() {