--- net/minecraft/world/entity/animal/Bee$BeeHurtByOtherGoal.java
+++ net/minecraft/world/entity/animal/Bee$BeeHurtByOtherGoal.java
@@ -8,18 +8,18 @@
  {
     public /* synthetic */ net.minecraft.world.entity.animal.Bee this$0;
 
-    public net.minecraft.world.entity.animal.Bee$BeeHurtByOtherGoal(net.minecraft.world.entity.animal.Bee bee, net.minecraft.world.entity.animal.Bee mob) {
+    public net.minecraft.world.entity.animal.Bee$BeeHurtByOtherGoal(net.minecraft.world.entity.animal.Bee bee, net.minecraft.world.entity.animal.Bee $$0) {
         this.this$0 = bee;
-        super((net.minecraft.world.entity.PathfinderMob)mob, new java.lang.Class[0]);
+        super((net.minecraft.world.entity.PathfinderMob)$$0, new java.lang.Class[0]);
     }
 
     public boolean canContinueToUse() {
         return this.this$0.isAngry() && super.canContinueToUse();
     }
 
-    public void alertOther(net.minecraft.world.entity.Mob mob, net.minecraft.world.entity.LivingEntity target) {
-        if (mob instanceof net.minecraft.world.entity.animal.Bee && this.mob.hasLineOfSight((net.minecraft.world.entity.Entity)target)) {
-            mob.setTarget(target);
+    public void alertOther(net.minecraft.world.entity.Mob $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        if ($$0 instanceof net.minecraft.world.entity.animal.Bee && this.mob.hasLineOfSight((net.minecraft.world.entity.Entity)$$1)) {
+            $$0.setTarget($$1);
         }
     }
 }