--- net/minecraft/world/entity/animal/Cat$CatRelaxOnOwnerGoal.java
+++ net/minecraft/world/entity/animal/Cat$CatRelaxOnOwnerGoal.java
@@ -13,8 +13,8 @@
     public net.minecraft.core.BlockPos goalPos;
     public int onBedTicks;
 
-    public net.minecraft.world.entity.animal.Cat$CatRelaxOnOwnerGoal(net.minecraft.world.entity.animal.Cat cat) {
-        this.cat = cat;
+    public net.minecraft.world.entity.animal.Cat$CatRelaxOnOwnerGoal(net.minecraft.world.entity.animal.Cat $$0) {
+        this.cat = $$0;
     }
 
     public boolean canUse() {
@@ -33,10 +33,10 @@
             if (this.cat.distanceToSqr((net.minecraft.world.entity.Entity)this.ownerPlayer) > 100.0) {
                 return false;
             }
-            net.minecraft.core.BlockPos $$1 = this.ownerPlayer.blockPosition();
-            net.minecraft.world.level.block.state.BlockState $$2 = this.cat.level.getBlockState($$1);
+            net.minecraft.core.BlockPos $$12 = this.ownerPlayer.blockPosition();
+            net.minecraft.world.level.block.state.BlockState $$2 = this.cat.level.getBlockState($$12);
             if ($$2.is(net.minecraft.tags.BlockTags.BEDS)) {
-                this.goalPos = (net.minecraft.core.BlockPos)$$2.getOptionalValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BedBlock.FACING).map(direction -> $$1.relative(direction.getOpposite())).orElseGet(() -> new net.minecraft.core.BlockPos((net.minecraft.core.Vec3i)$$1));
+                this.goalPos = (net.minecraft.core.BlockPos)$$2.getOptionalValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BedBlock.FACING).map($$1 -> $$12.relative($$1.getOpposite())).orElseGet(() -> new net.minecraft.core.BlockPos((net.minecraft.core.Vec3i)$$12));
                 return !this.spaceIsOccupied();
             }
         }
@@ -77,7 +77,7 @@
     public void giveMorningGift() {
         net.minecraft.util.RandomSource $$0 = this.cat.getRandom();
         net.minecraft.core.BlockPos$MutableBlockPos $$1 = new net.minecraft.core.BlockPos$MutableBlockPos();
-        $$1.set((net.minecraft.core.Vec3i)this.cat.blockPosition());
+        $$1.set((net.minecraft.core.Vec3i)(this.cat.isLeashed() ? this.cat.getLeashHolder().blockPosition() : this.cat.blockPosition()));
         this.cat.randomTeleport((double)($$1.getX() + $$0.nextInt(11) - 5), (double)($$1.getY() + $$0.nextInt(5) - 2), (double)($$1.getZ() + $$0.nextInt(11) - 5), false);
         $$1.set((net.minecraft.core.Vec3i)this.cat.blockPosition());
         net.minecraft.world.level.storage.loot.LootTable $$2 = this.cat.level.getServer().getLootTables().get(net.minecraft.world.level.storage.loot.BuiltInLootTables.CAT_MORNING_GIFT);