--- net/minecraft/world/entity/animal/Cod.java
+++ net/minecraft/world/entity/animal/Cod.java
@@ -22,7 +22,7 @@
         return net.minecraft.sounds.SoundEvents.COD_DEATH;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.COD_HURT;
     }
 