--- net/minecraft/world/entity/animal/Dolphin$DolphinSwimToTreasureGoal.java
+++ net/minecraft/world/entity/animal/Dolphin$DolphinSwimToTreasureGoal.java
@@ -9,8 +9,8 @@
     public net.minecraft.world.entity.animal.Dolphin dolphin;
     public boolean stuck;
 
-    public net.minecraft.world.entity.animal.Dolphin$DolphinSwimToTreasureGoal(net.minecraft.world.entity.animal.Dolphin dolphin) {
-        this.dolphin = dolphin;
+    public net.minecraft.world.entity.animal.Dolphin$DolphinSwimToTreasureGoal(net.minecraft.world.entity.animal.Dolphin $$0) {
+        this.dolphin = $$0;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE, (java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.LOOK));
     }
 