--- net/minecraft/world/entity/animal/Dolphin$PlayWithItemsGoal.java
+++ net/minecraft/world/entity/animal/Dolphin$PlayWithItemsGoal.java
@@ -50,19 +50,19 @@
         }
     }
 
-    public void drop(net.minecraft.world.item.ItemStack stack) {
-        if (stack.isEmpty()) {
+    public void drop(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0.isEmpty()) {
             return;
         }
-        double d = this.this$0.getEyeY() - (double)0.3f;
-        net.minecraft.world.entity.item.ItemEntity itemEntity = new net.minecraft.world.entity.item.ItemEntity(this.this$0.level, this.this$0.getX(), d, this.this$0.getZ(), stack);
-        itemEntity.setPickUpDelay(40);
-        itemEntity.setThrower(this.this$0.getUUID());
-        float f = 0.3f;
-        float f2 = net.minecraft.world.entity.animal.Dolphin.access$100((net.minecraft.world.entity.animal.Dolphin)this.this$0).nextFloat() * ((float)java.lang.Math.PI * 2);
-        float f3 = 0.02f * net.minecraft.world.entity.animal.Dolphin.access$200((net.minecraft.world.entity.animal.Dolphin)this.this$0).nextFloat();
-        itemEntity.setDeltaMovement((double)(0.3f * -net.minecraft.util.Mth.sin((float)(this.this$0.getYRot() * ((float)java.lang.Math.PI / 180))) * net.minecraft.util.Mth.cos((float)(this.this$0.getXRot() * ((float)java.lang.Math.PI / 180))) + net.minecraft.util.Mth.cos((float)f2) * f3), (double)(0.3f * net.minecraft.util.Mth.sin((float)(this.this$0.getXRot() * ((float)java.lang.Math.PI / 180))) * 1.5f), (double)(0.3f * net.minecraft.util.Mth.cos((float)(this.this$0.getYRot() * ((float)java.lang.Math.PI / 180))) * net.minecraft.util.Mth.cos((float)(this.this$0.getXRot() * ((float)java.lang.Math.PI / 180))) + net.minecraft.util.Mth.sin((float)f2) * f3));
-        this.this$0.level.addFreshEntity((net.minecraft.world.entity.Entity)itemEntity);
+        double $$1 = this.this$0.getEyeY() - (double)0.3f;
+        net.minecraft.world.entity.item.ItemEntity $$2 = new net.minecraft.world.entity.item.ItemEntity(this.this$0.level, this.this$0.getX(), $$1, this.this$0.getZ(), $$0);
+        $$2.setPickUpDelay(40);
+        $$2.setThrower(this.this$0.getUUID());
+        float $$3 = 0.3f;
+        float $$4 = net.minecraft.world.entity.animal.Dolphin.access$100((net.minecraft.world.entity.animal.Dolphin)this.this$0).nextFloat() * ((float)java.lang.Math.PI * 2);
+        float $$5 = 0.02f * net.minecraft.world.entity.animal.Dolphin.access$200((net.minecraft.world.entity.animal.Dolphin)this.this$0).nextFloat();
+        $$2.setDeltaMovement((double)(0.3f * -net.minecraft.util.Mth.sin((float)(this.this$0.getYRot() * ((float)java.lang.Math.PI / 180))) * net.minecraft.util.Mth.cos((float)(this.this$0.getXRot() * ((float)java.lang.Math.PI / 180))) + net.minecraft.util.Mth.cos((float)$$4) * $$5), (double)(0.3f * net.minecraft.util.Mth.sin((float)(this.this$0.getXRot() * ((float)java.lang.Math.PI / 180))) * 1.5f), (double)(0.3f * net.minecraft.util.Mth.cos((float)(this.this$0.getYRot() * ((float)java.lang.Math.PI / 180))) * net.minecraft.util.Mth.cos((float)(this.this$0.getXRot() * ((float)java.lang.Math.PI / 180))) + net.minecraft.util.Mth.sin((float)$$4) * $$5));
+        this.this$0.level.addFreshEntity((net.minecraft.world.entity.Entity)$$2);
     }
 }
 