--- net/minecraft/world/entity/animal/Fox$FoxBreedGoal.java
+++ net/minecraft/world/entity/animal/Fox$FoxBreedGoal.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$FoxBreedGoal(net.minecraft.world.entity.animal.Fox fox, double speedModifier) {
+    public net.minecraft.world.entity.animal.Fox$FoxBreedGoal(net.minecraft.world.entity.animal.Fox fox, double $$0) {
         this.this$0 = fox;
-        super((net.minecraft.world.entity.animal.Animal)fox, speedModifier);
+        super((net.minecraft.world.entity.animal.Animal)fox, $$0);
     }
 
     public void start() {