--- net/minecraft/world/entity/animal/Fox$DefendTrustedTargetGoal.java
+++ net/minecraft/world/entity/animal/Fox$DefendTrustedTargetGoal.java
@@ -13,9 +13,9 @@
     public int timestamp;
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$DefendTrustedTargetGoal(net.minecraft.world.entity.animal.Fox fox, java.lang.Class<net.minecraft.world.entity.LivingEntity> targetType, @javax.annotation.Nullable boolean mustSee, boolean mustReach, java.util.function.Predicate<net.minecraft.world.entity.LivingEntity> predicate) {
+    public net.minecraft.world.entity.animal.Fox$DefendTrustedTargetGoal(net.minecraft.world.entity.animal.Fox fox, java.lang.Class<net.minecraft.world.entity.LivingEntity> $$0, @javax.annotation.Nullable boolean $$1, boolean $$2, java.util.function.Predicate<net.minecraft.world.entity.LivingEntity> $$3) {
         this.this$0 = fox;
-        super((net.minecraft.world.entity.Mob)fox, targetType, 10, mustSee, mustReach, predicate);
+        super((net.minecraft.world.entity.Mob)fox, $$0, 10, $$1, $$2, $$3);
     }
 
     public boolean canUse() {