--- net/minecraft/world/entity/animal/Fox$FoxFollowParentGoal.java
+++ net/minecraft/world/entity/animal/Fox$FoxFollowParentGoal.java
@@ -9,10 +9,10 @@
     public net.minecraft.world.entity.animal.Fox fox;
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$FoxFollowParentGoal(net.minecraft.world.entity.animal.Fox fox, net.minecraft.world.entity.animal.Fox fox2, double speedModifier) {
+    public net.minecraft.world.entity.animal.Fox$FoxFollowParentGoal(net.minecraft.world.entity.animal.Fox fox, net.minecraft.world.entity.animal.Fox $$0, double $$1) {
         this.this$0 = fox;
-        super((net.minecraft.world.entity.animal.Animal)fox2, speedModifier);
-        this.fox = fox2;
+        super((net.minecraft.world.entity.animal.Animal)$$0, $$1);
+        this.fox = $$0;
     }
 
     public boolean canUse() {