--- net/minecraft/world/entity/animal/Fox$SeekShelterGoal.java
+++ net/minecraft/world/entity/animal/Fox$SeekShelterGoal.java
@@ -9,9 +9,9 @@
     public int interval;
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$SeekShelterGoal(net.minecraft.world.entity.animal.Fox fox, double speedModifier) {
+    public net.minecraft.world.entity.animal.Fox$SeekShelterGoal(net.minecraft.world.entity.animal.Fox fox, double $$0) {
         this.this$0 = fox;
-        super((net.minecraft.world.entity.PathfinderMob)fox, speedModifier);
+        super((net.minecraft.world.entity.PathfinderMob)fox, $$0);
         this.interval = net.minecraft.world.entity.animal.Fox$SeekShelterGoal.reducedTickDelay((int)100);
     }
 