--- net/minecraft/world/entity/animal/Fox$FoxEatBerriesGoal.java
+++ net/minecraft/world/entity/animal/Fox$FoxEatBerriesGoal.java
@@ -10,9 +10,9 @@
     public int ticksWaited;
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$FoxEatBerriesGoal(net.minecraft.world.entity.animal.Fox fox, double speedModifier, int searchRange, int verticalSearchRange) {
-        this.this$0 = fox;
-        super((net.minecraft.world.entity.PathfinderMob)fox, speedModifier, searchRange, verticalSearchRange);
+    public net.minecraft.world.entity.animal.Fox$FoxEatBerriesGoal(net.minecraft.world.entity.animal.Fox $$0, double $$1, int $$2, int $$3) {
+        this.this$0 = $$0;
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, $$2, $$3);
     }
 
     public double acceptedDistance() {
@@ -23,9 +23,9 @@
         return this.tryTicks % 100 == 0;
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(pos);
-        return blockState.is(net.minecraft.world.level.block.Blocks.SWEET_BERRY_BUSH) && (java.lang.Integer)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.SweetBerryBushBlock.AGE) >= 2 || net.minecraft.world.level.block.CaveVines.hasGlowBerries((net.minecraft.world.level.block.state.BlockState)blockState);
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        net.minecraft.world.level.block.state.BlockState $$2 = $$0.getBlockState($$1);
+        return $$2.is(net.minecraft.world.level.block.Blocks.SWEET_BERRY_BUSH) && (java.lang.Integer)$$2.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.SweetBerryBushBlock.AGE) >= 2 || net.minecraft.world.level.block.CaveVines.hasGlowBerries((net.minecraft.world.level.block.state.BlockState)$$2);
     }
 
     public void tick() {