--- net/minecraft/world/entity/animal/Fox$FoxLookAtPlayerGoal.java
+++ net/minecraft/world/entity/animal/Fox$FoxLookAtPlayerGoal.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$FoxLookAtPlayerGoal(net.minecraft.world.entity.animal.Fox fox, net.minecraft.world.entity.Mob mob, java.lang.Class<? extends net.minecraft.world.entity.LivingEntity> lookAtType, float lookDistance) {
+    public net.minecraft.world.entity.animal.Fox$FoxLookAtPlayerGoal(net.minecraft.world.entity.animal.Fox fox, net.minecraft.world.entity.Mob $$0, java.lang.Class<? extends net.minecraft.world.entity.LivingEntity> $$1, float $$2) {
         this.this$0 = fox;
-        super(mob, lookAtType, lookDistance);
+        super($$0, $$1, $$2);
     }
 
     public boolean canUse() {