--- net/minecraft/world/entity/animal/Fox$Type.java
+++ net/minecraft/world/entity/animal/Fox$Type.java
@@ -36,23 +36,23 @@
         return this.id;
     }
 
-    public static net.minecraft.world.entity.animal.Fox$Type byName(java.lang.String name) {
-        return (net.minecraft.world.entity.animal.Fox$Type)BY_NAME.getOrDefault((java.lang.Object)name, (java.lang.Object)RED);
+    public static net.minecraft.world.entity.animal.Fox$Type byName(java.lang.String $$0) {
+        return (net.minecraft.world.entity.animal.Fox$Type)BY_NAME.getOrDefault((java.lang.Object)$$0, (java.lang.Object)RED);
     }
 
-    public static net.minecraft.world.entity.animal.Fox$Type byId(int index) {
-        if (index < 0 || index > BY_ID.length) {
-            index = 0;
+    public static net.minecraft.world.entity.animal.Fox$Type byId(int $$0) {
+        if ($$0 < 0 || $$0 > BY_ID.length) {
+            $$0 = 0;
         }
-        return BY_ID[index];
+        return BY_ID[$$0];
     }
 
     public static net.minecraft.world.entity.animal.Fox$Type byBiome(net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> $$0) {
         return ((net.minecraft.world.level.biome.Biome)$$0.value()).getPrecipitation() == net.minecraft.world.level.biome.Biome$Precipitation.SNOW ? SNOW : RED;
     }
 
-    public static /* synthetic */ net.minecraft.world.entity.animal.Fox$Type lambda$static$1(net.minecraft.world.entity.animal.Fox$Type type) {
-        return type;
+    public static /* synthetic */ net.minecraft.world.entity.animal.Fox$Type lambda$static$1(net.minecraft.world.entity.animal.Fox$Type $$0) {
+        return $$0;
     }
 
     public static /* synthetic */ net.minecraft.world.entity.animal.Fox$Type[] lambda$static$0(int $$0) {
@@ -68,7 +68,7 @@
         SNOW = new net.minecraft.world.entity.animal.Fox$Type("SNOW", 1, 1, "snow");
         $VALUES = net.minecraft.world.entity.animal.Fox$Type.$values();
         BY_ID = (net.minecraft.world.entity.animal.Fox$Type[])java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.entity.animal.Fox$Type.values()).sorted(java.util.Comparator.comparingInt(net.minecraft.world.entity.animal.Fox$Type::getId)).toArray(net.minecraft.world.entity.animal.Fox$Type[]::new);
-        BY_NAME = (java.util.Map)java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.entity.animal.Fox$Type.values()).collect(java.util.stream.Collectors.toMap(net.minecraft.world.entity.animal.Fox$Type::getName, type -> type));
+        BY_NAME = (java.util.Map)java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.entity.animal.Fox$Type.values()).collect(java.util.stream.Collectors.toMap(net.minecraft.world.entity.animal.Fox$Type::getName, $$0 -> $$0));
     }
 }
 