--- net/minecraft/world/entity/animal/Fox$FoxStrollThroughVillageGoal.java
+++ net/minecraft/world/entity/animal/Fox$FoxStrollThroughVillageGoal.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$FoxStrollThroughVillageGoal(net.minecraft.world.entity.animal.Fox fox, int unused32, int interval) {
+    public net.minecraft.world.entity.animal.Fox$FoxStrollThroughVillageGoal(net.minecraft.world.entity.animal.Fox fox, int $$0, int $$1) {
         this.this$0 = fox;
-        super((net.minecraft.world.entity.PathfinderMob)fox, interval);
+        super((net.minecraft.world.entity.PathfinderMob)fox, $$1);
     }
 
     public void start() {