--- net/minecraft/world/entity/animal/Fox$FoxMeleeAttackGoal.java
+++ net/minecraft/world/entity/animal/Fox$FoxMeleeAttackGoal.java
@@ -8,16 +8,16 @@
  {
     public /* synthetic */ net.minecraft.world.entity.animal.Fox this$0;
 
-    public net.minecraft.world.entity.animal.Fox$FoxMeleeAttackGoal(net.minecraft.world.entity.animal.Fox fox, double speedModifier, boolean followingTargetEvenIfNotSeen) {
+    public net.minecraft.world.entity.animal.Fox$FoxMeleeAttackGoal(net.minecraft.world.entity.animal.Fox fox, double $$0, boolean $$1) {
         this.this$0 = fox;
-        super((net.minecraft.world.entity.PathfinderMob)fox, speedModifier, followingTargetEvenIfNotSeen);
+        super((net.minecraft.world.entity.PathfinderMob)fox, $$0, $$1);
     }
 
-    public void checkAndPerformAttack(net.minecraft.world.entity.LivingEntity enemy, double distToEnemySqr) {
-        double d = this.getAttackReachSqr(enemy);
-        if (distToEnemySqr <= d && this.isTimeToAttack()) {
+    public void checkAndPerformAttack(net.minecraft.world.entity.LivingEntity $$0, double $$1) {
+        double $$2 = this.getAttackReachSqr($$0);
+        if ($$1 <= $$2 && this.isTimeToAttack()) {
             this.resetAttackCooldown();
-            this.mob.doHurtTarget((net.minecraft.world.entity.Entity)enemy);
+            this.mob.doHurtTarget((net.minecraft.world.entity.Entity)$$0);
             this.this$0.playSound(net.minecraft.sounds.SoundEvents.FOX_BITE, 1.0f, 1.0f);
         }
     }