--- net/minecraft/world/entity/animal/Fox$FoxGroupData.java
+++ net/minecraft/world/entity/animal/Fox$FoxGroupData.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Fox$Type type;
 
-    public net.minecraft.world.entity.animal.Fox$FoxGroupData(net.minecraft.world.entity.animal.Fox$Type type) {
+    public net.minecraft.world.entity.animal.Fox$FoxGroupData(net.minecraft.world.entity.animal.Fox$Type $$0) {
         super(false);
-        this.type = type;
+        this.type = $$0;
     }
 }
 