--- net/minecraft/world/entity/animal/IronGolem$Crackiness.java
+++ net/minecraft/world/entity/animal/IronGolem$Crackiness.java
@@ -22,21 +22,21 @@
         return (net.minecraft.world.entity.animal.IronGolem$Crackiness)java.lang.Enum.valueOf(net.minecraft.world.entity.animal.IronGolem$Crackiness.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.animal.IronGolem$Crackiness(float fraction) {
+    public net.minecraft.world.entity.animal.IronGolem$Crackiness(float $$0) {
         super(string, n);
-        this.fraction = fraction;
+        this.fraction = $$0;
     }
 
-    public static net.minecraft.world.entity.animal.IronGolem$Crackiness byFraction(float fraction) {
-        for (net.minecraft.world.entity.animal.IronGolem$Crackiness crackiness : BY_DAMAGE) {
-            if (!(fraction < crackiness.fraction)) continue;
-            return crackiness;
+    public static net.minecraft.world.entity.animal.IronGolem$Crackiness byFraction(float $$0) {
+        for (net.minecraft.world.entity.animal.IronGolem$Crackiness $$1 : BY_DAMAGE) {
+            if (!($$0 < $$1.fraction)) continue;
+            return $$1;
         }
         return NONE;
     }
 
-    public static /* synthetic */ double lambda$static$0(net.minecraft.world.entity.animal.IronGolem$Crackiness crack) {
-        return crack.fraction;
+    public static /* synthetic */ double lambda$static$0(net.minecraft.world.entity.animal.IronGolem$Crackiness $$0) {
+        return $$0.fraction;
     }
 
     public static /* synthetic */ net.minecraft.world.entity.animal.IronGolem$Crackiness[] $values() {
@@ -49,7 +49,7 @@
         MEDIUM = new net.minecraft.world.entity.animal.IronGolem$Crackiness("MEDIUM", 2, 0.5f);
         HIGH = new net.minecraft.world.entity.animal.IronGolem$Crackiness("HIGH", 3, 0.25f);
         $VALUES = net.minecraft.world.entity.animal.IronGolem$Crackiness.$values();
-        BY_DAMAGE = (java.util.List)java.util.stream.Stream.of((java.lang.Object[])net.minecraft.world.entity.animal.IronGolem$Crackiness.values()).sorted(java.util.Comparator.comparingDouble(crack -> crack.fraction)).collect(com.google.common.collect.ImmutableList.toImmutableList());
+        BY_DAMAGE = (java.util.List)java.util.stream.Stream.of((java.lang.Object[])net.minecraft.world.entity.animal.IronGolem$Crackiness.values()).sorted(java.util.Comparator.comparingDouble($$0 -> $$0.fraction)).collect(com.google.common.collect.ImmutableList.toImmutableList());
     }
 }
 