--- net/minecraft/world/entity/animal/IronGolem.java
+++ net/minecraft/world/entity/animal/IronGolem.java
@@ -32,7 +32,7 @@
         this.targetSelector.addGoal(1, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.DefendVillageTargetGoal(this));
         this.targetSelector.addGoal(2, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal((net.minecraft.world.entity.PathfinderMob)this, new java.lang.Class[0]));
         this.targetSelector.addGoal(3, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal((net.minecraft.world.entity.Mob)this, net.minecraft.world.entity.player.Player.class, 10, true, false, arg_0 -> ((net.minecraft.world.entity.animal.IronGolem)this).isAngryAt(arg_0)));
-        this.targetSelector.addGoal(3, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal((net.minecraft.world.entity.Mob)this, net.minecraft.world.entity.Mob.class, 5, false, false, entity -> entity instanceof net.minecraft.world.entity.monster.Enemy && !(entity instanceof net.minecraft.world.entity.monster.Creeper)));
+        this.targetSelector.addGoal(3, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal((net.minecraft.world.entity.Mob)this, net.minecraft.world.entity.Mob.class, 5, false, false, $$0 -> $$0 instanceof net.minecraft.world.entity.monster.Enemy && !($$0 instanceof net.minecraft.world.entity.monster.Creeper)));
         this.targetSelector.addGoal(4, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.ResetUniversalAngerTargetGoal((net.minecraft.world.entity.Mob)this, false));
     }
 
@@ -45,15 +45,15 @@
         return net.minecraft.world.entity.Mob.createMobAttributes().add(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, 100.0).add(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED, 0.25).add(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE, 1.0).add(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE, 15.0);
     }
 
-    public int decreaseAirSupply(int air) {
-        return air;
+    public int decreaseAirSupply(int $$0) {
+        return $$0;
     }
 
-    public void doPush(net.minecraft.world.entity.Entity entity) {
-        if (entity instanceof net.minecraft.world.entity.monster.Enemy && !(entity instanceof net.minecraft.world.entity.monster.Creeper) && this.getRandom().nextInt(20) == 0) {
-            this.setTarget((net.minecraft.world.entity.LivingEntity)entity);
+    public void doPush(net.minecraft.world.entity.Entity $$0) {
+        if ($$0 instanceof net.minecraft.world.entity.monster.Enemy && !($$0 instanceof net.minecraft.world.entity.monster.Creeper) && this.getRandom().nextInt(20) == 0) {
+            this.setTarget((net.minecraft.world.entity.LivingEntity)$$0);
         }
-        super.doPush(entity);
+        super.doPush($$0);
     }
 
     public void aiStep() {
@@ -76,42 +76,42 @@
         }
     }
 
-    public boolean canAttackType(net.minecraft.world.entity.EntityType<?> type) {
-        if (this.isPlayerCreated() && type == net.minecraft.world.entity.EntityType.PLAYER) {
+    public boolean canAttackType(net.minecraft.world.entity.EntityType<?> $$0) {
+        if (this.isPlayerCreated() && $$0 == net.minecraft.world.entity.EntityType.PLAYER) {
             return false;
         }
-        if (type == net.minecraft.world.entity.EntityType.CREEPER) {
+        if ($$0 == net.minecraft.world.entity.EntityType.CREEPER) {
             return false;
         }
-        return super.canAttackType(type);
+        return super.canAttackType($$0);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putBoolean("PlayerCreated", this.isPlayerCreated());
-        this.addPersistentAngerSaveData(compound);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putBoolean("PlayerCreated", this.isPlayerCreated());
+        this.addPersistentAngerSaveData($$0);
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.setPlayerCreated(compound.getBoolean("PlayerCreated"));
-        this.readPersistentAngerSaveData(this.level, compound);
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.setPlayerCreated($$0.getBoolean("PlayerCreated"));
+        this.readPersistentAngerSaveData(this.level, $$0);
     }
 
     public void startPersistentAngerTimer() {
         this.setRemainingPersistentAngerTime(PERSISTENT_ANGER_TIME.sample(this.random));
     }
 
-    public void setRemainingPersistentAngerTime(int time) {
-        this.remainingPersistentAngerTime = time;
+    public void setRemainingPersistentAngerTime(int $$0) {
+        this.remainingPersistentAngerTime = $$0;
     }
 
     public int getRemainingPersistentAngerTime() {
         return this.remainingPersistentAngerTime;
     }
 
-    public void setPersistentAngerTarget(@javax.annotation.Nullable java.util.UUID target) {
-        this.persistentAngerTarget = target;
+    public void setPersistentAngerTarget(@javax.annotation.Nullable java.util.UUID $$0) {
+        this.persistentAngerTarget = $$0;
     }
 
     @javax.annotation.Nullable
@@ -123,52 +123,52 @@
         return (float)this.getAttributeValue(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE);
     }
 
-    public boolean doHurtTarget(net.minecraft.world.entity.Entity entity) {
+    public boolean doHurtTarget(net.minecraft.world.entity.Entity $$0) {
         this.attackAnimationTick = 10;
         this.level.broadcastEntityEvent((net.minecraft.world.entity.Entity)this, (byte)4);
-        float f = this.getAttackDamage();
-        float f2 = (int)f > 0 ? f / 2.0f + (float)this.random.nextInt((int)f) : f;
-        boolean bl = entity.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), f2);
-        if (bl) {
+        float $$1 = this.getAttackDamage();
+        float $$2 = (int)$$1 > 0 ? $$1 / 2.0f + (float)this.random.nextInt((int)$$1) : $$1;
+        boolean $$3 = $$0.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), $$2);
+        if ($$3) {
             double d;
-            if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-                net.minecraft.world.entity.LivingEntity livingEntity = (net.minecraft.world.entity.LivingEntity)entity;
-                d = livingEntity.getAttributeValue(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE);
+            if ($$0 instanceof net.minecraft.world.entity.LivingEntity) {
+                net.minecraft.world.entity.LivingEntity $$4 = (net.minecraft.world.entity.LivingEntity)$$0;
+                d = $$4.getAttributeValue(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE);
             } else {
                 d = 0.0;
             }
-            double d2 = d;
-            double d3 = java.lang.Math.max((double)0.0, (double)(1.0 - d2));
-            entity.setDeltaMovement(entity.getDeltaMovement().add(0.0, (double)0.4f * d3, 0.0));
-            this.doEnchantDamageEffects((net.minecraft.world.entity.LivingEntity)this, entity);
+            double $$5 = d;
+            double $$6 = java.lang.Math.max((double)0.0, (double)(1.0 - $$5));
+            $$0.setDeltaMovement($$0.getDeltaMovement().add(0.0, (double)0.4f * $$6, 0.0));
+            this.doEnchantDamageEffects((net.minecraft.world.entity.LivingEntity)this, $$0);
         }
         this.playSound(net.minecraft.sounds.SoundEvents.IRON_GOLEM_ATTACK, 1.0f, 1.0f);
-        return bl;
+        return $$3;
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
-        net.minecraft.world.entity.animal.IronGolem$Crackiness crackiness = this.getCrackiness();
-        boolean bl = super.hurt(source, amount);
-        if (bl && this.getCrackiness() != crackiness) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
+        net.minecraft.world.entity.animal.IronGolem$Crackiness $$2 = this.getCrackiness();
+        boolean $$3 = super.hurt($$0, $$1);
+        if ($$3 && this.getCrackiness() != $$2) {
             this.playSound(net.minecraft.sounds.SoundEvents.IRON_GOLEM_DAMAGE, 1.0f, 1.0f);
         }
-        return bl;
+        return $$3;
     }
 
     public net.minecraft.world.entity.animal.IronGolem$Crackiness getCrackiness() {
         return net.minecraft.world.entity.animal.IronGolem$Crackiness.byFraction((float)(this.getHealth() / this.getMaxHealth()));
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 4) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 4) {
             this.attackAnimationTick = 10;
             this.playSound(net.minecraft.sounds.SoundEvents.IRON_GOLEM_ATTACK, 1.0f, 1.0f);
-        } else if (id == 11) {
+        } else if ($$0 == 11) {
             this.offerFlowerTick = 400;
-        } else if (id == 34) {
+        } else if ($$0 == 34) {
             this.offerFlowerTick = 0;
         } else {
-            super.handleEntityEvent(id);
+            super.handleEntityEvent($$0);
         }
     }
 
@@ -176,8 +176,8 @@
         return this.attackAnimationTick;
     }
 
-    public void offerFlower(boolean offeringFlower) {
-        if (offeringFlower) {
+    public void offerFlower(boolean $$0) {
+        if ($$0) {
             this.offerFlowerTick = 400;
             this.level.broadcastEntityEvent((net.minecraft.world.entity.Entity)this, (byte)11);
         } else {
@@ -186,7 +186,7 @@
         }
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.IRON_GOLEM_HURT;
     }
 
@@ -194,25 +194,25 @@
         return net.minecraft.sounds.SoundEvents.IRON_GOLEM_DEATH;
     }
 
-    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        if (!itemStack.is(net.minecraft.world.item.Items.IRON_INGOT)) {
+    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$1) {
+        net.minecraft.world.item.ItemStack $$2 = $$0.getItemInHand($$1);
+        if (!$$2.is(net.minecraft.world.item.Items.IRON_INGOT)) {
             return net.minecraft.world.InteractionResult.PASS;
         }
-        float f = this.getHealth();
+        float $$3 = this.getHealth();
         this.heal(25.0f);
-        if (this.getHealth() == f) {
+        if (this.getHealth() == $$3) {
             return net.minecraft.world.InteractionResult.PASS;
         }
-        float f2 = 1.0f + (this.random.nextFloat() - this.random.nextFloat()) * 0.2f;
-        this.playSound(net.minecraft.sounds.SoundEvents.IRON_GOLEM_REPAIR, 1.0f, f2);
-        if (!player.getAbilities().instabuild) {
-            itemStack.shrink(1);
+        float $$4 = 1.0f + (this.random.nextFloat() - this.random.nextFloat()) * 0.2f;
+        this.playSound(net.minecraft.sounds.SoundEvents.IRON_GOLEM_REPAIR, 1.0f, $$4);
+        if (!$$0.getAbilities().instabuild) {
+            $$2.shrink(1);
         }
         return net.minecraft.world.InteractionResult.sidedSuccess((boolean)this.level.isClientSide);
     }
 
-    public void playStepSound(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState block) {
+    public void playStepSound(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         this.playSound(net.minecraft.sounds.SoundEvents.IRON_GOLEM_STEP, 1.0f, 1.0f);
     }
 
@@ -224,31 +224,31 @@
         return ((java.lang.Byte)this.entityData.get(DATA_FLAGS_ID) & 1) != 0;
     }
 
-    public void setPlayerCreated(boolean playerCreated) {
-        byte by = (java.lang.Byte)this.entityData.get(DATA_FLAGS_ID);
-        if (playerCreated) {
-            this.entityData.set(DATA_FLAGS_ID, (java.lang.Object)((byte)(by | 1)));
+    public void setPlayerCreated(boolean $$0) {
+        byte $$1 = (java.lang.Byte)this.entityData.get(DATA_FLAGS_ID);
+        if ($$0) {
+            this.entityData.set(DATA_FLAGS_ID, (java.lang.Object)((byte)($$1 | 1)));
         } else {
-            this.entityData.set(DATA_FLAGS_ID, (java.lang.Object)((byte)(by & 0xFFFFFFFE)));
+            this.entityData.set(DATA_FLAGS_ID, (java.lang.Object)((byte)($$1 & 0xFFFFFFFE)));
         }
     }
 
-    public void die(net.minecraft.world.damagesource.DamageSource cause) {
-        super.die(cause);
+    public void die(net.minecraft.world.damagesource.DamageSource $$0) {
+        super.die($$0);
     }
 
-    public boolean checkSpawnObstruction(net.minecraft.world.level.LevelReader level) {
-        net.minecraft.core.BlockPos blockPos = this.blockPosition();
-        net.minecraft.core.BlockPos blockPos2 = blockPos.below();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos2);
-        if (blockState.entityCanStandOn((net.minecraft.world.level.BlockGetter)level, blockPos2, (net.minecraft.world.entity.Entity)this)) {
-            for (int i = 1; i < 3; ++i) {
-                net.minecraft.world.level.block.state.BlockState blockState2;
-                net.minecraft.core.BlockPos blockPos3 = blockPos.above(i);
-                if (net.minecraft.world.level.NaturalSpawner.isValidEmptySpawnBlock((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)blockPos3, (net.minecraft.world.level.block.state.BlockState)(blockState2 = level.getBlockState(blockPos3)), (net.minecraft.world.level.material.FluidState)blockState2.getFluidState(), (net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.IRON_GOLEM)) continue;
+    public boolean checkSpawnObstruction(net.minecraft.world.level.LevelReader $$0) {
+        net.minecraft.core.BlockPos $$1 = this.blockPosition();
+        net.minecraft.core.BlockPos $$2 = $$1.below();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$0.getBlockState($$2);
+        if ($$3.entityCanStandOn((net.minecraft.world.level.BlockGetter)$$0, $$2, (net.minecraft.world.entity.Entity)this)) {
+            for (int $$4 = 1; $$4 < 3; ++$$4) {
+                net.minecraft.world.level.block.state.BlockState $$6;
+                net.minecraft.core.BlockPos $$5 = $$1.above($$4);
+                if (net.minecraft.world.level.NaturalSpawner.isValidEmptySpawnBlock((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$5, (net.minecraft.world.level.block.state.BlockState)($$6 = $$0.getBlockState($$5)), (net.minecraft.world.level.material.FluidState)$$6.getFluidState(), (net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.IRON_GOLEM)) continue;
                 return false;
             }
-            return net.minecraft.world.level.NaturalSpawner.isValidEmptySpawnBlock((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.world.level.block.state.BlockState)level.getBlockState(blockPos), (net.minecraft.world.level.material.FluidState)net.minecraft.world.level.material.Fluids.EMPTY.defaultFluidState(), (net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.IRON_GOLEM) && level.isUnobstructed((net.minecraft.world.entity.Entity)this);
+            return net.minecraft.world.level.NaturalSpawner.isValidEmptySpawnBlock((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$1, (net.minecraft.world.level.block.state.BlockState)$$0.getBlockState($$1), (net.minecraft.world.level.material.FluidState)net.minecraft.world.level.material.Fluids.EMPTY.defaultFluidState(), (net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.IRON_GOLEM) && $$0.isUnobstructed((net.minecraft.world.entity.Entity)this);
         }
         return false;
     }