--- net/minecraft/world/entity/animal/MushroomCow$MushroomType.java
+++ net/minecraft/world/entity/animal/MushroomCow$MushroomType.java
@@ -20,20 +20,20 @@
         return (net.minecraft.world.entity.animal.MushroomCow$MushroomType)java.lang.Enum.valueOf(net.minecraft.world.entity.animal.MushroomCow$MushroomType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.animal.MushroomCow$MushroomType(java.lang.String type, net.minecraft.world.level.block.state.BlockState blockState) {
+    public net.minecraft.world.entity.animal.MushroomCow$MushroomType(java.lang.String $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         super(string, n);
-        this.type = type;
-        this.blockState = blockState;
+        this.type = $$0;
+        this.blockState = $$1;
     }
 
     public net.minecraft.world.level.block.state.BlockState getBlockState() {
         return this.blockState;
     }
 
-    public static net.minecraft.world.entity.animal.MushroomCow$MushroomType byType(java.lang.String name) {
-        for (net.minecraft.world.entity.animal.MushroomCow$MushroomType mushroomType : net.minecraft.world.entity.animal.MushroomCow$MushroomType.values()) {
-            if (!mushroomType.type.equals((java.lang.Object)name)) continue;
-            return mushroomType;
+    public static net.minecraft.world.entity.animal.MushroomCow$MushroomType byType(java.lang.String $$0) {
+        for (net.minecraft.world.entity.animal.MushroomCow$MushroomType $$1 : net.minecraft.world.entity.animal.MushroomCow$MushroomType.values()) {
+            if (!$$1.type.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
         return RED;
     }