--- net/minecraft/world/entity/animal/Ocelot$OcelotTemptGoal.java
+++ net/minecraft/world/entity/animal/Ocelot$OcelotTemptGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Ocelot ocelot;
 
-    public net.minecraft.world.entity.animal.Ocelot$OcelotTemptGoal(net.minecraft.world.entity.animal.Ocelot ocelot, double speedModifier, net.minecraft.world.item.crafting.Ingredient items, boolean canScare) {
-        super((net.minecraft.world.entity.PathfinderMob)ocelot, speedModifier, items, canScare);
-        this.ocelot = ocelot;
+    public net.minecraft.world.entity.animal.Ocelot$OcelotTemptGoal(net.minecraft.world.entity.animal.Ocelot $$0, double $$1, net.minecraft.world.item.crafting.Ingredient $$2, boolean $$3) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, $$2, $$3);
+        this.ocelot = $$0;
     }
 
     public boolean canScare() {