--- net/minecraft/world/entity/animal/Ocelot$OcelotAvoidEntityGoal.java
+++ net/minecraft/world/entity/animal/Ocelot$OcelotAvoidEntityGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Ocelot ocelot;
 
-    public net.minecraft.world.entity.animal.Ocelot$OcelotAvoidEntityGoal(net.minecraft.world.entity.animal.Ocelot ocelot, java.lang.Class<T> entityClassToAvoid, float maxDist, double walkSpeedModifier, double sprintSpeedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)ocelot, entityClassToAvoid, maxDist, walkSpeedModifier, sprintSpeedModifier, arg_0 -> ((java.util.function.Predicate)net.minecraft.world.entity.EntitySelector.NO_CREATIVE_OR_SPECTATOR).test(arg_0));
-        this.ocelot = ocelot;
+    public net.minecraft.world.entity.animal.Ocelot$OcelotAvoidEntityGoal(net.minecraft.world.entity.animal.Ocelot $$0, java.lang.Class<T> $$1, float $$2, double $$3, double $$4) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, $$2, $$3, $$4, arg_0 -> ((java.util.function.Predicate)net.minecraft.world.entity.EntitySelector.NO_CREATIVE_OR_SPECTATOR).test(arg_0));
+        this.ocelot = $$0;
     }
 
     public boolean canUse() {