--- net/minecraft/world/entity/animal/Panda$PandaHurtByTargetGoal.java
+++ net/minecraft/world/entity/animal/Panda$PandaHurtByTargetGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Panda panda;
 
-    public net.minecraft.world.entity.animal.Panda$PandaHurtByTargetGoal(net.minecraft.world.entity.animal.Panda panda, java.lang.Class<?> ... entityClassToIgnoreDamage) {
-        super((net.minecraft.world.entity.PathfinderMob)panda, entityClassToIgnoreDamage);
-        this.panda = panda;
+    public net.minecraft.world.entity.animal.Panda$PandaHurtByTargetGoal(net.minecraft.world.entity.animal.Panda $$0, java.lang.Class<?> ... $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1);
+        this.panda = $$0;
     }
 
     public boolean canContinueToUse() {
@@ -21,9 +21,9 @@
         return super.canContinueToUse();
     }
 
-    public void alertOther(net.minecraft.world.entity.Mob mob, net.minecraft.world.entity.LivingEntity target) {
-        if (mob instanceof net.minecraft.world.entity.animal.Panda && ((net.minecraft.world.entity.animal.Panda)mob).isAggressive()) {
-            mob.setTarget(target);
+    public void alertOther(net.minecraft.world.entity.Mob $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        if ($$0 instanceof net.minecraft.world.entity.animal.Panda && ((net.minecraft.world.entity.animal.Panda)$$0).isAggressive()) {
+            $$0.setTarget($$1);
         }
     }
 }