--- net/minecraft/world/entity/animal/Panda$PandaAvoidGoal.java
+++ net/minecraft/world/entity/animal/Panda$PandaAvoidGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Panda panda;
 
-    public net.minecraft.world.entity.animal.Panda$PandaAvoidGoal(net.minecraft.world.entity.animal.Panda panda, java.lang.Class<T> entityClassToAvoid, float maxDist, double walkSpeedModifier, double sprintSpeedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)panda, entityClassToAvoid, maxDist, walkSpeedModifier, sprintSpeedModifier, arg_0 -> ((java.util.function.Predicate)net.minecraft.world.entity.EntitySelector.NO_SPECTATORS).test(arg_0));
-        this.panda = panda;
+    public net.minecraft.world.entity.animal.Panda$PandaAvoidGoal(net.minecraft.world.entity.animal.Panda $$0, java.lang.Class<T> $$1, float $$2, double $$3, double $$4) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, $$2, $$3, $$4, arg_0 -> ((java.util.function.Predicate)net.minecraft.world.entity.EntitySelector.NO_SPECTATORS).test(arg_0));
+        this.panda = $$0;
     }
 
     public boolean canUse() {