--- net/minecraft/world/entity/animal/Panda$PandaPanicGoal.java
+++ net/minecraft/world/entity/animal/Panda$PandaPanicGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Panda panda;
 
-    public net.minecraft.world.entity.animal.Panda$PandaPanicGoal(net.minecraft.world.entity.animal.Panda panda, double speedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)panda, speedModifier);
-        this.panda = panda;
+    public net.minecraft.world.entity.animal.Panda$PandaPanicGoal(net.minecraft.world.entity.animal.Panda $$0, double $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1);
+        this.panda = $$0;
     }
 
     public boolean shouldPanic() {