--- net/minecraft/world/entity/animal/Panda$PandaLookAtPlayerGoal.java
+++ net/minecraft/world/entity/animal/Panda$PandaLookAtPlayerGoal.java
@@ -8,13 +8,13 @@
  {
     public net.minecraft.world.entity.animal.Panda panda;
 
-    public net.minecraft.world.entity.animal.Panda$PandaLookAtPlayerGoal(net.minecraft.world.entity.animal.Panda panda, java.lang.Class<? extends net.minecraft.world.entity.LivingEntity> lookAtType, float lookDistance) {
-        super((net.minecraft.world.entity.Mob)panda, lookAtType, lookDistance);
-        this.panda = panda;
+    public net.minecraft.world.entity.animal.Panda$PandaLookAtPlayerGoal(net.minecraft.world.entity.animal.Panda $$0, java.lang.Class<? extends net.minecraft.world.entity.LivingEntity> $$1, float $$2) {
+        super((net.minecraft.world.entity.Mob)$$0, $$1, $$2);
+        this.panda = $$0;
     }
 
-    public void setTarget(net.minecraft.world.entity.LivingEntity lookAt) {
-        this.lookAt = lookAt;
+    public void setTarget(net.minecraft.world.entity.LivingEntity $$0) {
+        this.lookAt = $$0;
     }
 
     public boolean canContinueToUse() {