--- net/minecraft/world/entity/animal/Panda$PandaMoveControl.java
+++ net/minecraft/world/entity/animal/Panda$PandaMoveControl.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Panda panda;
 
-    public net.minecraft.world.entity.animal.Panda$PandaMoveControl(net.minecraft.world.entity.animal.Panda panda) {
-        super((net.minecraft.world.entity.Mob)panda);
-        this.panda = panda;
+    public net.minecraft.world.entity.animal.Panda$PandaMoveControl(net.minecraft.world.entity.animal.Panda $$0) {
+        super((net.minecraft.world.entity.Mob)$$0);
+        this.panda = $$0;
     }
 
     public void tick() {