--- net/minecraft/world/entity/animal/Panda.java
+++ net/minecraft/world/entity/animal/Panda.java
@@ -41,20 +41,20 @@
         }
     }
 
-    public boolean canTakeItem(net.minecraft.world.item.ItemStack itemstack) {
-        net.minecraft.world.entity.EquipmentSlot equipmentSlot = net.minecraft.world.entity.Mob.getEquipmentSlotForItem((net.minecraft.world.item.ItemStack)itemstack);
-        if (!this.getItemBySlot(equipmentSlot).isEmpty()) {
+    public boolean canTakeItem(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.world.entity.EquipmentSlot $$1 = net.minecraft.world.entity.Mob.getEquipmentSlotForItem((net.minecraft.world.item.ItemStack)$$0);
+        if (!this.getItemBySlot($$1).isEmpty()) {
             return false;
         }
-        return equipmentSlot == net.minecraft.world.entity.EquipmentSlot.MAINHAND && super.canTakeItem(itemstack);
+        return $$1 == net.minecraft.world.entity.EquipmentSlot.MAINHAND && super.canTakeItem($$0);
     }
 
     public int getUnhappyCounter() {
         return (java.lang.Integer)this.entityData.get(UNHAPPY_COUNTER);
     }
 
-    public void setUnhappyCounter(int unhappyCounter) {
-        this.entityData.set(UNHAPPY_COUNTER, (java.lang.Object)unhappyCounter);
+    public void setUnhappyCounter(int $$0) {
+        this.entityData.set(UNHAPPY_COUNTER, (java.lang.Object)$$0);
     }
 
     public boolean isSneezing() {
@@ -65,37 +65,37 @@
         return this.getFlag(8);
     }
 
-    public void sit(boolean sitting) {
-        this.setFlag(8, sitting);
+    public void sit(boolean $$0) {
+        this.setFlag(8, $$0);
     }
 
     public boolean isOnBack() {
         return this.getFlag(16);
     }
 
-    public void setOnBack(boolean onBack) {
-        this.setFlag(16, onBack);
+    public void setOnBack(boolean $$0) {
+        this.setFlag(16, $$0);
     }
 
     public boolean isEating() {
         return (java.lang.Integer)this.entityData.get(EAT_COUNTER) > 0;
     }
 
-    public void eat(boolean eating) {
-        this.entityData.set(EAT_COUNTER, (java.lang.Object)(eating ? 1 : 0));
+    public void eat(boolean $$0) {
+        this.entityData.set(EAT_COUNTER, (java.lang.Object)($$0 ? 1 : 0));
     }
 
     public int getEatCounter() {
         return (java.lang.Integer)this.entityData.get(EAT_COUNTER);
     }
 
-    public void setEatCounter(int eatCounter) {
-        this.entityData.set(EAT_COUNTER, (java.lang.Object)eatCounter);
+    public void setEatCounter(int $$0) {
+        this.entityData.set(EAT_COUNTER, (java.lang.Object)$$0);
     }
 
-    public void sneeze(boolean sneezing) {
-        this.setFlag(2, sneezing);
-        if (!sneezing) {
+    public void sneeze(boolean $$0) {
+        this.setFlag(2, $$0);
+        if (!$$0) {
             this.setSneezeCounter(0);
         }
     }
@@ -104,38 +104,38 @@
         return (java.lang.Integer)this.entityData.get(SNEEZE_COUNTER);
     }
 
-    public void setSneezeCounter(int sneezeCounter) {
-        this.entityData.set(SNEEZE_COUNTER, (java.lang.Object)sneezeCounter);
+    public void setSneezeCounter(int $$0) {
+        this.entityData.set(SNEEZE_COUNTER, (java.lang.Object)$$0);
     }
 
     public net.minecraft.world.entity.animal.Panda$Gene getMainGene() {
         return net.minecraft.world.entity.animal.Panda$Gene.byId((int)((java.lang.Byte)this.entityData.get(MAIN_GENE_ID)).byteValue());
     }
 
-    public void setMainGene(net.minecraft.world.entity.animal.Panda$Gene pandaType) {
-        if (pandaType.getId() > 6) {
-            pandaType = net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)this.random);
+    public void setMainGene(net.minecraft.world.entity.animal.Panda$Gene $$0) {
+        if ($$0.getId() > 6) {
+            $$0 = net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)this.random);
         }
-        this.entityData.set(MAIN_GENE_ID, (java.lang.Object)((byte)pandaType.getId()));
+        this.entityData.set(MAIN_GENE_ID, (java.lang.Object)((byte)$$0.getId()));
     }
 
     public net.minecraft.world.entity.animal.Panda$Gene getHiddenGene() {
         return net.minecraft.world.entity.animal.Panda$Gene.byId((int)((java.lang.Byte)this.entityData.get(HIDDEN_GENE_ID)).byteValue());
     }
 
-    public void setHiddenGene(net.minecraft.world.entity.animal.Panda$Gene pandaType) {
-        if (pandaType.getId() > 6) {
-            pandaType = net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)this.random);
+    public void setHiddenGene(net.minecraft.world.entity.animal.Panda$Gene $$0) {
+        if ($$0.getId() > 6) {
+            $$0 = net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)this.random);
         }
-        this.entityData.set(HIDDEN_GENE_ID, (java.lang.Object)((byte)pandaType.getId()));
+        this.entityData.set(HIDDEN_GENE_ID, (java.lang.Object)((byte)$$0.getId()));
     }
 
     public boolean isRolling() {
         return this.getFlag(4);
     }
 
-    public void roll(boolean rolling) {
-        this.setFlag(4, rolling);
+    public void roll(boolean $$0) {
+        this.setFlag(4, $$0);
     }
 
     public void defineSynchedData() {
@@ -148,38 +148,41 @@
         this.entityData.define(EAT_COUNTER, (java.lang.Object)0);
     }
 
-    public boolean getFlag(int flagId) {
-        return ((java.lang.Byte)this.entityData.get(DATA_ID_FLAGS) & flagId) != 0;
+    public boolean getFlag(int $$0) {
+        return ((java.lang.Byte)this.entityData.get(DATA_ID_FLAGS) & $$0) != 0;
     }
 
-    public void setFlag(int flagId, boolean value) {
-        byte by = (java.lang.Byte)this.entityData.get(DATA_ID_FLAGS);
-        if (value) {
-            this.entityData.set(DATA_ID_FLAGS, (java.lang.Object)((byte)(by | flagId)));
+    public void setFlag(int $$0, boolean $$1) {
+        byte $$2 = (java.lang.Byte)this.entityData.get(DATA_ID_FLAGS);
+        if ($$1) {
+            this.entityData.set(DATA_ID_FLAGS, (java.lang.Object)((byte)($$2 | $$0)));
         } else {
-            this.entityData.set(DATA_ID_FLAGS, (java.lang.Object)((byte)(by & ~flagId)));
+            this.entityData.set(DATA_ID_FLAGS, (java.lang.Object)((byte)($$2 & ~$$0)));
         }
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putString("MainGene", this.getMainGene().getName());
-        compound.putString("HiddenGene", this.getHiddenGene().getName());
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putString("MainGene", this.getMainGene().getName());
+        $$0.putString("HiddenGene", this.getHiddenGene().getName());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.setMainGene(net.minecraft.world.entity.animal.Panda$Gene.byName((java.lang.String)compound.getString("MainGene")));
-        this.setHiddenGene(net.minecraft.world.entity.animal.Panda$Gene.byName((java.lang.String)compound.getString("HiddenGene")));
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.setMainGene(net.minecraft.world.entity.animal.Panda$Gene.byName((java.lang.String)$$0.getString("MainGene")));
+        this.setHiddenGene(net.minecraft.world.entity.animal.Panda$Gene.byName((java.lang.String)$$0.getString("HiddenGene")));
     }
 
     @javax.annotation.Nullable
     public net.minecraft.world.entity.AgeableMob getBreedOffspring(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.AgeableMob $$1) {
         net.minecraft.world.entity.animal.Panda $$2 = (net.minecraft.world.entity.animal.Panda)net.minecraft.world.entity.EntityType.PANDA.create((net.minecraft.world.level.Level)$$0);
-        if ($$1 instanceof net.minecraft.world.entity.animal.Panda) {
-            $$2.setGeneFromParents(this, (net.minecraft.world.entity.animal.Panda)$$1);
+        if ($$2 != null) {
+            if ($$1 instanceof net.minecraft.world.entity.animal.Panda) {
+                net.minecraft.world.entity.animal.Panda $$3 = (net.minecraft.world.entity.animal.Panda)$$1;
+                $$2.setGeneFromParents(this, $$3);
+            }
+            $$2.setAttributes();
         }
-        $$2.setAttributes();
         return $$2;
     }
 
@@ -235,16 +238,16 @@
         return this.getVariant() == net.minecraft.world.entity.animal.Panda$Gene.AGGRESSIVE;
     }
 
-    public boolean canBeLeashed(net.minecraft.world.entity.player.Player player) {
+    public boolean canBeLeashed(net.minecraft.world.entity.player.Player $$0) {
         return false;
     }
 
-    public boolean doHurtTarget(net.minecraft.world.entity.Entity entity) {
+    public boolean doHurtTarget(net.minecraft.world.entity.Entity $$0) {
         this.playSound(net.minecraft.sounds.SoundEvents.PANDA_BITE, 1.0f, 1.0f);
         if (!this.isAggressive()) {
             this.didBite = true;
         }
-        return super.doHurtTarget(entity);
+        return super.doHurtTarget($$0);
     }
 
     public void tick() {
@@ -352,16 +355,16 @@
         this.rollAmount = this.isRolling() ? java.lang.Math.min((float)1.0f, (float)(this.rollAmount + 0.15f)) : java.lang.Math.max((float)0.0f, (float)(this.rollAmount - 0.19f));
     }
 
-    public float getSitAmount(float partialTick) {
-        return net.minecraft.util.Mth.lerp((float)partialTick, (float)this.sitAmountO, (float)this.sitAmount);
+    public float getSitAmount(float $$0) {
+        return net.minecraft.util.Mth.lerp((float)$$0, (float)this.sitAmountO, (float)this.sitAmount);
     }
 
-    public float getLieOnBackAmount(float partialTick) {
-        return net.minecraft.util.Mth.lerp((float)partialTick, (float)this.onBackAmountO, (float)this.onBackAmount);
+    public float getLieOnBackAmount(float $$0) {
+        return net.minecraft.util.Mth.lerp((float)$$0, (float)this.onBackAmountO, (float)this.onBackAmount);
     }
 
-    public float getRollAmount(float partialTick) {
-        return net.minecraft.util.Mth.lerp((float)partialTick, (float)this.rollAmountO, (float)this.rollAmount);
+    public float getRollAmount(float $$0) {
+        return net.minecraft.util.Mth.lerp((float)$$0, (float)this.rollAmountO, (float)this.rollAmount);
     }
 
     public void handleRoll() {
@@ -399,51 +402,51 @@
         }
     }
 
-    public void pickUpItem(net.minecraft.world.entity.item.ItemEntity itemEntity) {
-        if (this.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND).isEmpty() && PANDA_ITEMS.test((java.lang.Object)itemEntity)) {
-            this.onItemPickup(itemEntity);
-            net.minecraft.world.item.ItemStack itemStack = itemEntity.getItem();
-            this.setItemSlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND, itemStack);
+    public void pickUpItem(net.minecraft.world.entity.item.ItemEntity $$0) {
+        if (this.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND).isEmpty() && PANDA_ITEMS.test((java.lang.Object)$$0)) {
+            this.onItemPickup($$0);
+            net.minecraft.world.item.ItemStack $$1 = $$0.getItem();
+            this.setItemSlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND, $$1);
             this.setGuaranteedDrop(net.minecraft.world.entity.EquipmentSlot.MAINHAND);
-            this.take((net.minecraft.world.entity.Entity)itemEntity, itemStack.getCount());
-            itemEntity.discard();
+            this.take((net.minecraft.world.entity.Entity)$$0, $$1.getCount());
+            $$0.discard();
         }
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
         if (!this.level.isClientSide) {
             this.sit(false);
         }
-        return super.hurt(source, amount);
+        return super.hurt($$0, $$1);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor level, net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.MobSpawnType reason, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData spawnData, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag dataTag) {
-        net.minecraft.util.RandomSource randomSource = level.getRandom();
-        this.setMainGene(net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)randomSource));
-        this.setHiddenGene(net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)randomSource));
+    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.world.DifficultyInstance $$1, net.minecraft.world.entity.MobSpawnType $$2, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData $$3, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$4) {
+        net.minecraft.util.RandomSource $$5 = $$0.getRandom();
+        this.setMainGene(net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)$$5));
+        this.setHiddenGene(net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)$$5));
         this.setAttributes();
-        if (spawnData == null) {
-            spawnData = new net.minecraft.world.entity.AgeableMob$AgeableMobGroupData(0.2f);
+        if ($$3 == null) {
+            $$3 = new net.minecraft.world.entity.AgeableMob$AgeableMobGroupData(0.2f);
         }
-        return super.finalizeSpawn(level, difficulty, reason, spawnData, dataTag);
+        return super.finalizeSpawn($$0, $$1, $$2, $$3, $$4);
     }
 
-    public void setGeneFromParents(net.minecraft.world.entity.animal.Panda father, @javax.annotation.Nullable net.minecraft.world.entity.animal.Panda mother) {
-        if (mother == null) {
+    public void setGeneFromParents(net.minecraft.world.entity.animal.Panda $$0, @javax.annotation.Nullable net.minecraft.world.entity.animal.Panda $$1) {
+        if ($$1 == null) {
             if (this.random.nextBoolean()) {
-                this.setMainGene(father.getOneOfGenesRandomly());
+                this.setMainGene($$0.getOneOfGenesRandomly());
                 this.setHiddenGene(net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)this.random));
             } else {
                 this.setMainGene(net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)this.random));
-                this.setHiddenGene(father.getOneOfGenesRandomly());
+                this.setHiddenGene($$0.getOneOfGenesRandomly());
             }
         } else if (this.random.nextBoolean()) {
-            this.setMainGene(father.getOneOfGenesRandomly());
-            this.setHiddenGene(mother.getOneOfGenesRandomly());
+            this.setMainGene($$0.getOneOfGenesRandomly());
+            this.setHiddenGene($$1.getOneOfGenesRandomly());
         } else {
-            this.setMainGene(mother.getOneOfGenesRandomly());
-            this.setHiddenGene(father.getOneOfGenesRandomly());
+            this.setMainGene($$1.getOneOfGenesRandomly());
+            this.setHiddenGene($$0.getOneOfGenesRandomly());
         }
         if (this.random.nextInt(32) == 0) {
             this.setMainGene(net.minecraft.world.entity.animal.Panda$Gene.getRandom((net.minecraft.util.RandomSource)this.random));
@@ -477,8 +480,8 @@
         }
     }
 
-    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
+    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$1) {
+        net.minecraft.world.item.ItemStack $$2 = $$0.getItemInHand($$1);
         if (this.isScared()) {
             return net.minecraft.world.InteractionResult.PASS;
         }
@@ -486,25 +489,25 @@
             this.setOnBack(false);
             return net.minecraft.world.InteractionResult.sidedSuccess((boolean)this.level.isClientSide);
         }
-        if (this.isFood(itemStack)) {
+        if (this.isFood($$2)) {
             if (this.getTarget() != null) {
                 this.gotBamboo = true;
             }
             if (this.isBaby()) {
-                this.usePlayerItem(player, hand, itemStack);
+                this.usePlayerItem($$0, $$1, $$2);
                 this.ageUp((int)((float)(-this.getAge() / 20) * 0.1f), true);
             } else if (!this.level.isClientSide && this.getAge() == 0 && this.canFallInLove()) {
-                this.usePlayerItem(player, hand, itemStack);
-                this.setInLove(player);
+                this.usePlayerItem($$0, $$1, $$2);
+                this.setInLove($$0);
             } else if (!(this.level.isClientSide || this.isSitting() || this.isInWater())) {
                 this.tryToSit();
                 this.eat(true);
-                net.minecraft.world.item.ItemStack itemStack2 = this.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND);
-                if (!itemStack2.isEmpty() && !player.getAbilities().instabuild) {
-                    this.spawnAtLocation(itemStack2);
+                net.minecraft.world.item.ItemStack $$3 = this.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND);
+                if (!$$3.isEmpty() && !$$0.getAbilities().instabuild) {
+                    this.spawnAtLocation($$3);
                 }
-                this.setItemSlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)itemStack.getItem(), 1));
-                this.usePlayerItem(player, hand, itemStack);
+                this.setItemSlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$2.getItem(), 1));
+                this.usePlayerItem($$0, $$1, $$2);
             } else {
                 return net.minecraft.world.InteractionResult.PASS;
             }
@@ -524,16 +527,16 @@
         return net.minecraft.sounds.SoundEvents.PANDA_AMBIENT;
     }
 
-    public void playStepSound(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState block) {
+    public void playStepSound(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         this.playSound(net.minecraft.sounds.SoundEvents.PANDA_STEP, 0.15f, 1.0f);
     }
 
-    public boolean isFood(net.minecraft.world.item.ItemStack stack) {
-        return stack.is(net.minecraft.world.level.block.Blocks.BAMBOO.asItem());
+    public boolean isFood(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(net.minecraft.world.level.block.Blocks.BAMBOO.asItem());
     }
 
-    public boolean isFoodOrCake(net.minecraft.world.item.ItemStack stack) {
-        return this.isFood(stack) || stack.is(net.minecraft.world.level.block.Blocks.CAKE.asItem());
+    public boolean isFoodOrCake(net.minecraft.world.item.ItemStack $$0) {
+        return this.isFood($$0) || $$0.is(net.minecraft.world.level.block.Blocks.CAKE.asItem());
     }
 
     @javax.annotation.Nullable
@@ -542,7 +545,7 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.PANDA_HURT;
     }
 
@@ -550,9 +553,9 @@
         return !this.isOnBack() && !this.isScared() && !this.isEating() && !this.isRolling() && !this.isSitting();
     }
 
-    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.entity.item.ItemEntity itemEntity) {
-        net.minecraft.world.item.ItemStack itemStack = itemEntity.getItem();
-        return (itemStack.is(net.minecraft.world.level.block.Blocks.BAMBOO.asItem()) || itemStack.is(net.minecraft.world.level.block.Blocks.CAKE.asItem())) && itemEntity.isAlive() && !itemEntity.hasPickUpDelay();
+    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.entity.item.ItemEntity $$0) {
+        net.minecraft.world.item.ItemStack $$1 = $$0.getItem();
+        return ($$1.is(net.minecraft.world.level.block.Blocks.BAMBOO.asItem()) || $$1.is(net.minecraft.world.level.block.Blocks.CAKE.asItem())) && $$0.isAlive() && !$$0.hasPickUpDelay();
     }
 
     public static /* synthetic */ boolean access$000(net.minecraft.world.entity.animal.Panda $$0) {
@@ -611,9 +614,9 @@
         HIDDEN_GENE_ID = net.minecraft.network.syncher.SynchedEntityData.defineId(net.minecraft.world.entity.animal.Panda.class, (net.minecraft.network.syncher.EntityDataSerializer)net.minecraft.network.syncher.EntityDataSerializers.BYTE);
         DATA_ID_FLAGS = net.minecraft.network.syncher.SynchedEntityData.defineId(net.minecraft.world.entity.animal.Panda.class, (net.minecraft.network.syncher.EntityDataSerializer)net.minecraft.network.syncher.EntityDataSerializers.BYTE);
         BREED_TARGETING = net.minecraft.world.entity.ai.targeting.TargetingConditions.forNonCombat().range(8.0);
-        PANDA_ITEMS = itemEntity -> {
-            net.minecraft.world.item.ItemStack itemStack = itemEntity.getItem();
-            return (itemStack.is(net.minecraft.world.level.block.Blocks.BAMBOO.asItem()) || itemStack.is(net.minecraft.world.level.block.Blocks.CAKE.asItem())) && itemEntity.isAlive() && !itemEntity.hasPickUpDelay();
+        PANDA_ITEMS = $$0 -> {
+            net.minecraft.world.item.ItemStack $$1 = $$0.getItem();
+            return ($$1.is(net.minecraft.world.level.block.Blocks.BAMBOO.asItem()) || $$1.is(net.minecraft.world.level.block.Blocks.CAKE.asItem())) && $$0.isAlive() && !$$0.hasPickUpDelay();
         };
     }
 }