--- net/minecraft/world/entity/animal/PolarBear$PolarBearMeleeAttackGoal.java
+++ net/minecraft/world/entity/animal/PolarBear$PolarBearMeleeAttackGoal.java
@@ -13,13 +13,13 @@
         super((net.minecraft.world.entity.PathfinderMob)polarBear, 1.25, true);
     }
 
-    public void checkAndPerformAttack(net.minecraft.world.entity.LivingEntity enemy, double distToEnemySqr) {
-        double d = this.getAttackReachSqr(enemy);
-        if (distToEnemySqr <= d && this.isTimeToAttack()) {
+    public void checkAndPerformAttack(net.minecraft.world.entity.LivingEntity $$0, double $$1) {
+        double $$2 = this.getAttackReachSqr($$0);
+        if ($$1 <= $$2 && this.isTimeToAttack()) {
             this.resetAttackCooldown();
-            this.mob.doHurtTarget((net.minecraft.world.entity.Entity)enemy);
+            this.mob.doHurtTarget((net.minecraft.world.entity.Entity)$$0);
             this.this$0.setStanding(false);
-        } else if (distToEnemySqr <= d * 2.0) {
+        } else if ($$1 <= $$2 * 2.0) {
             if (this.isTimeToAttack()) {
                 this.this$0.setStanding(false);
                 this.resetAttackCooldown();
@@ -39,8 +39,8 @@
         super.stop();
     }
 
-    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity attackTarget) {
-        return 4.0f + attackTarget.getBbWidth();
+    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity $$0) {
+        return 4.0f + $$0.getBbWidth();
     }
 }
 