--- net/minecraft/world/entity/animal/Pufferfish.java
+++ net/minecraft/world/entity/animal/Pufferfish.java
@@ -29,25 +29,25 @@
         return (java.lang.Integer)this.entityData.get(PUFF_STATE);
     }
 
-    public void setPuffState(int puffState) {
-        this.entityData.set(PUFF_STATE, (java.lang.Object)puffState);
+    public void setPuffState(int $$0) {
+        this.entityData.set(PUFF_STATE, (java.lang.Object)$$0);
     }
 
-    public void onSyncedDataUpdated(net.minecraft.network.syncher.EntityDataAccessor<?> key) {
-        if (PUFF_STATE.equals(key)) {
+    public void onSyncedDataUpdated(net.minecraft.network.syncher.EntityDataAccessor<?> $$0) {
+        if (PUFF_STATE.equals($$0)) {
             this.refreshDimensions();
         }
-        super.onSyncedDataUpdated(key);
+        super.onSyncedDataUpdated($$0);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("PuffState", this.getPuffState());
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("PuffState", this.getPuffState());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.setPuffState(java.lang.Math.min((int)compound.getInt("PuffState"), (int)2));
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.setPuffState(java.lang.Math.min((int)$$0.getInt("PuffState"), (int)2));
     }
 
     public net.minecraft.world.item.ItemStack getBucketItemStack() {
@@ -95,21 +95,21 @@
         }
     }
 
-    public void touch(net.minecraft.world.entity.Mob mob) {
-        int n = this.getPuffState();
-        if (mob.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), (float)(1 + n))) {
-            mob.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.POISON, 60 * n, 0), (net.minecraft.world.entity.Entity)this);
+    public void touch(net.minecraft.world.entity.Mob $$0) {
+        int $$1 = this.getPuffState();
+        if ($$0.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), (float)(1 + $$1))) {
+            $$0.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.POISON, 60 * $$1, 0), (net.minecraft.world.entity.Entity)this);
             this.playSound(net.minecraft.sounds.SoundEvents.PUFFER_FISH_STING, 1.0f, 1.0f);
         }
     }
 
-    public void playerTouch(net.minecraft.world.entity.player.Player entity) {
-        int n = this.getPuffState();
-        if (entity instanceof net.minecraft.server.level.ServerPlayer && n > 0 && entity.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), (float)(1 + n))) {
+    public void playerTouch(net.minecraft.world.entity.player.Player $$0) {
+        int $$1 = this.getPuffState();
+        if ($$0 instanceof net.minecraft.server.level.ServerPlayer && $$1 > 0 && $$0.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), (float)(1 + $$1))) {
             if (!this.isSilent()) {
-                ((net.minecraft.server.level.ServerPlayer)entity).connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundGameEventPacket(net.minecraft.network.protocol.game.ClientboundGameEventPacket.PUFFER_FISH_STING, 0.0f));
+                ((net.minecraft.server.level.ServerPlayer)$$0).connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundGameEventPacket(net.minecraft.network.protocol.game.ClientboundGameEventPacket.PUFFER_FISH_STING, 0.0f));
             }
-            entity.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.POISON, 60 * n, 0), (net.minecraft.world.entity.Entity)this);
+            $$0.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.POISON, 60 * $$1, 0), (net.minecraft.world.entity.Entity)this);
         }
     }
 
@@ -121,7 +121,7 @@
         return net.minecraft.sounds.SoundEvents.PUFFER_FISH_DEATH;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.PUFFER_FISH_HURT;
     }
 
@@ -129,12 +129,12 @@
         return net.minecraft.sounds.SoundEvents.PUFFER_FISH_FLOP;
     }
 
-    public net.minecraft.world.entity.EntityDimensions getDimensions(net.minecraft.world.entity.Pose pose) {
-        return super.getDimensions(pose).scale(net.minecraft.world.entity.animal.Pufferfish.getScale((int)this.getPuffState()));
+    public net.minecraft.world.entity.EntityDimensions getDimensions(net.minecraft.world.entity.Pose $$0) {
+        return super.getDimensions($$0).scale(net.minecraft.world.entity.animal.Pufferfish.getScale((int)this.getPuffState()));
     }
 
-    public static float getScale(int puffState) {
-        switch (puffState) {
+    public static float getScale(int $$0) {
+        switch ($$0) {
             case 1: {
                 return 0.7f;
             }
@@ -145,20 +145,20 @@
         return 1.0f;
     }
 
-    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.entity.LivingEntity entity) {
-        if (entity instanceof net.minecraft.world.entity.player.Player && ((net.minecraft.world.entity.player.Player)entity).isCreative()) {
+    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.entity.LivingEntity $$0) {
+        if ($$0 instanceof net.minecraft.world.entity.player.Player && ((net.minecraft.world.entity.player.Player)$$0).isCreative()) {
             return false;
         }
-        return entity.getType() == net.minecraft.world.entity.EntityType.AXOLOTL || entity.getMobType() != net.minecraft.world.entity.MobType.WATER;
+        return $$0.getType() == net.minecraft.world.entity.EntityType.AXOLOTL || $$0.getMobType() != net.minecraft.world.entity.MobType.WATER;
     }
 
     static {
         PUFF_STATE = net.minecraft.network.syncher.SynchedEntityData.defineId(net.minecraft.world.entity.animal.Pufferfish.class, (net.minecraft.network.syncher.EntityDataSerializer)net.minecraft.network.syncher.EntityDataSerializers.INT);
-        SCARY_MOB = entity -> {
-            if (entity instanceof net.minecraft.world.entity.player.Player && ((net.minecraft.world.entity.player.Player)entity).isCreative()) {
+        SCARY_MOB = $$0 -> {
+            if ($$0 instanceof net.minecraft.world.entity.player.Player && ((net.minecraft.world.entity.player.Player)$$0).isCreative()) {
                 return false;
             }
-            return entity.getType() == net.minecraft.world.entity.EntityType.AXOLOTL || entity.getMobType() != net.minecraft.world.entity.MobType.WATER;
+            return $$0.getType() == net.minecraft.world.entity.EntityType.AXOLOTL || $$0.getMobType() != net.minecraft.world.entity.MobType.WATER;
         };
         targetingConditions = net.minecraft.world.entity.ai.targeting.TargetingConditions.forNonCombat().ignoreInvisibilityTesting().ignoreLineOfSight().selector(SCARY_MOB);
     }