--- net/minecraft/world/entity/animal/Pufferfish$PufferfishPuffGoal.java
+++ net/minecraft/world/entity/animal/Pufferfish$PufferfishPuffGoal.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.entity.animal.Pufferfish fish;
 
-    public net.minecraft.world.entity.animal.Pufferfish$PufferfishPuffGoal(net.minecraft.world.entity.animal.Pufferfish fish) {
-        this.fish = fish;
+    public net.minecraft.world.entity.animal.Pufferfish$PufferfishPuffGoal(net.minecraft.world.entity.animal.Pufferfish $$0) {
+        this.fish = $$0;
     }
 
     public boolean canUse() {