--- net/minecraft/world/entity/animal/Rabbit$RabbitMoveControl.java
+++ net/minecraft/world/entity/animal/Rabbit$RabbitMoveControl.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.entity.animal.Rabbit rabbit;
     public double nextJumpSpeed;
 
-    public net.minecraft.world.entity.animal.Rabbit$RabbitMoveControl(net.minecraft.world.entity.animal.Rabbit rabbit) {
-        super((net.minecraft.world.entity.Mob)rabbit);
-        this.rabbit = rabbit;
+    public net.minecraft.world.entity.animal.Rabbit$RabbitMoveControl(net.minecraft.world.entity.animal.Rabbit $$0) {
+        super((net.minecraft.world.entity.Mob)$$0);
+        this.rabbit = $$0;
     }
 
     public void tick() {
@@ -23,13 +23,13 @@
         super.tick();
     }
 
-    public void setWantedPosition(double x, double y, double z, double speed) {
+    public void setWantedPosition(double $$0, double $$1, double $$2, double $$3) {
         if (this.rabbit.isInWater()) {
-            speed = 1.5;
+            $$3 = 1.5;
         }
-        super.setWantedPosition(x, y, z, speed);
-        if (speed > 0.0) {
-            this.nextJumpSpeed = speed;
+        super.setWantedPosition($$0, $$1, $$2, $$3);
+        if ($$3 > 0.0) {
+            this.nextJumpSpeed = $$3;
         }
     }
 }