--- net/minecraft/world/entity/animal/Rabbit$EvilRabbitAttackGoal.java
+++ net/minecraft/world/entity/animal/Rabbit$EvilRabbitAttackGoal.java
@@ -6,12 +6,12 @@
 public class net.minecraft.world.entity.animal.Rabbit$EvilRabbitAttackGoal
 extends net.minecraft.world.entity.ai.goal.MeleeAttackGoal
  {
-    public net.minecraft.world.entity.animal.Rabbit$EvilRabbitAttackGoal(net.minecraft.world.entity.animal.Rabbit rabbit) {
-        super((net.minecraft.world.entity.PathfinderMob)rabbit, 1.4, true);
+    public net.minecraft.world.entity.animal.Rabbit$EvilRabbitAttackGoal(net.minecraft.world.entity.animal.Rabbit $$0) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, 1.4, true);
     }
 
-    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity attackTarget) {
-        return 4.0f + attackTarget.getBbWidth();
+    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity $$0) {
+        return 4.0f + $$0.getBbWidth();
     }
 }
 