--- net/minecraft/world/entity/animal/Rabbit$RabbitGroupData.java
+++ net/minecraft/world/entity/animal/Rabbit$RabbitGroupData.java
@@ -8,9 +8,9 @@
  {
     public int rabbitType;
 
-    public net.minecraft.world.entity.animal.Rabbit$RabbitGroupData(int rabbitType) {
+    public net.minecraft.world.entity.animal.Rabbit$RabbitGroupData(int $$0) {
         super(1.0f);
-        this.rabbitType = rabbitType;
+        this.rabbitType = $$0;
     }
 }
 