--- net/minecraft/world/entity/animal/Rabbit.java
+++ net/minecraft/world/entity/animal/Rabbit.java
@@ -79,21 +79,21 @@
         }
     }
 
-    public float getJumpCompletion(float partialTick) {
+    public float getJumpCompletion(float $$0) {
         if (this.jumpDuration == 0) {
             return 0.0f;
         }
-        return ((float)this.jumpTicks + partialTick) / (float)this.jumpDuration;
+        return ((float)this.jumpTicks + $$0) / (float)this.jumpDuration;
     }
 
-    public void setSpeedModifier(double speedModifier) {
-        this.getNavigation().setSpeedModifier(speedModifier);
-        this.moveControl.setWantedPosition(this.moveControl.getWantedX(), this.moveControl.getWantedY(), this.moveControl.getWantedZ(), speedModifier);
+    public void setSpeedModifier(double $$0) {
+        this.getNavigation().setSpeedModifier($$0);
+        this.moveControl.setWantedPosition(this.moveControl.getWantedX(), this.moveControl.getWantedY(), this.moveControl.getWantedZ(), $$0);
     }
 
-    public void setJumping(boolean jumping) {
-        super.setJumping(jumping);
-        if (jumping) {
+    public void setJumping(boolean $$0) {
+        super.setJumping($$0);
+        if ($$0) {
             this.playSound(this.getJumpSound(), this.getSoundVolume(), ((this.random.nextFloat() - this.random.nextFloat()) * 0.2f + 1.0f) * 0.8f);
         }
     }
@@ -153,8 +153,8 @@
         return false;
     }
 
-    public void facePoint(double x, double z) {
-        this.setYRot((float)(net.minecraft.util.Mth.atan2((double)(z - this.getZ()), (double)(x - this.getX())) * 57.2957763671875) - 90.0f);
+    public void facePoint(double $$0, double $$1) {
+        this.setYRot((float)(net.minecraft.util.Mth.atan2((double)($$1 - this.getZ()), (double)($$0 - this.getX())) * 57.2957763671875) - 90.0f);
     }
 
     public void enableJumpControl() {
@@ -189,16 +189,16 @@
         return net.minecraft.world.entity.Mob.createMobAttributes().add(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, 3.0).add(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED, (double)0.3f);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("RabbitType", this.getRabbitType());
-        compound.putInt("MoreCarrotTicks", this.moreCarrotTicks);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("RabbitType", this.getRabbitType());
+        $$0.putInt("MoreCarrotTicks", this.moreCarrotTicks);
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.setRabbitType(compound.getInt("RabbitType"));
-        this.moreCarrotTicks = compound.getInt("MoreCarrotTicks");
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.setRabbitType($$0.getInt("RabbitType"));
+        this.moreCarrotTicks = $$0.getInt("MoreCarrotTicks");
     }
 
     public net.minecraft.sounds.SoundEvent getJumpSound() {
@@ -209,7 +209,7 @@
         return net.minecraft.sounds.SoundEvents.RABBIT_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.RABBIT_HURT;
     }
 
@@ -217,12 +217,12 @@
         return net.minecraft.sounds.SoundEvents.RABBIT_DEATH;
     }
 
-    public boolean doHurtTarget(net.minecraft.world.entity.Entity entity) {
+    public boolean doHurtTarget(net.minecraft.world.entity.Entity $$0) {
         if (this.getRabbitType() == 99) {
             this.playSound(net.minecraft.sounds.SoundEvents.RABBIT_ATTACK, 1.0f, (this.random.nextFloat() - this.random.nextFloat()) * 0.2f + 1.0f);
-            return entity.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), 8.0f);
+            return $$0.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), 8.0f);
         }
-        return entity.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), 3.0f);
+        return $$0.hurt(net.minecraft.world.damagesource.DamageSource.mobAttack((net.minecraft.world.entity.LivingEntity)this), 3.0f);
     }
 
     public net.minecraft.sounds.SoundSource getSoundSource() {
@@ -233,26 +233,42 @@
         return $$0.is(net.minecraft.world.item.Items.CARROT) || $$0.is(net.minecraft.world.item.Items.GOLDEN_CARROT) || $$0.is(net.minecraft.world.level.block.Blocks.DANDELION.asItem());
     }
 
+    /*
+     * Unable to fully structure code
+     */
+    @javax.annotation.Nullable
     public net.minecraft.world.entity.animal.Rabbit getBreedOffspring(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.AgeableMob $$1) {
-        net.minecraft.world.entity.animal.Rabbit $$2 = (net.minecraft.world.entity.animal.Rabbit)net.minecraft.world.entity.EntityType.RABBIT.create((net.minecraft.world.level.Level)$$0);
-        int $$3 = this.getRandomRabbitType((net.minecraft.world.level.LevelAccessor)$$0);
-        if (this.random.nextInt(20) != 0) {
-            $$3 = $$1 instanceof net.minecraft.world.entity.animal.Rabbit && this.random.nextBoolean() ? ((net.minecraft.world.entity.animal.Rabbit)$$1).getRabbitType() : this.getRabbitType();
+block2:        {
+block3:            {
+                $$2 = (net.minecraft.world.entity.animal.Rabbit)net.minecraft.world.entity.EntityType.RABBIT.create((net.minecraft.world.level.Level)$$0);
+                if ($$2 == null) break block2;
+                $$3 = this.getRandomRabbitType((net.minecraft.world.level.LevelAccessor)$$0);
+                if (this.random.nextInt(20) == 0) break block3;
+                if (!($$1 instanceof net.minecraft.world.entity.animal.Rabbit)) ** GOTO lbl-1000
+                $$4 = (net.minecraft.world.entity.animal.Rabbit)$$1;
+                if (this.random.nextBoolean()) {
+                    $$3 = $$4.getRabbitType();
+                } else lbl-1000:// 2 sources
+
+                {
+                    $$3 = this.getRabbitType();
+                }
+            }
+            $$2.setRabbitType($$3);
         }
-        $$2.setRabbitType($$3);
         return $$2;
     }
 
-    public boolean isFood(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.entity.animal.Rabbit.isTemptingItem((net.minecraft.world.item.ItemStack)stack);
+    public boolean isFood(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.entity.animal.Rabbit.isTemptingItem((net.minecraft.world.item.ItemStack)$$0);
     }
 
     public int getRabbitType() {
         return (java.lang.Integer)this.entityData.get(DATA_TYPE_ID);
     }
 
-    public void setRabbitType(int rabbitTypeId) {
-        if (rabbitTypeId == 99) {
+    public void setRabbitType(int $$0) {
+        if ($$0 == 99) {
             this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ARMOR).setBaseValue(8.0);
             this.goalSelector.addGoal(4, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.animal.Rabbit$EvilRabbitAttackGoal(this));
             this.targetSelector.addGoal(1, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal((net.minecraft.world.entity.PathfinderMob)this, new java.lang.Class[0]).setAlertOthers(new java.lang.Class[0]));
@@ -262,31 +278,31 @@
                 this.setCustomName((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)net.minecraft.Util.makeDescriptionId((java.lang.String)"entity", (net.minecraft.resources.ResourceLocation)KILLER_BUNNY)));
             }
         }
-        this.entityData.set(DATA_TYPE_ID, (java.lang.Object)rabbitTypeId);
+        this.entityData.set(DATA_TYPE_ID, (java.lang.Object)$$0);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor level, net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.MobSpawnType reason, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData spawnData, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag dataTag) {
-        int n = this.getRandomRabbitType((net.minecraft.world.level.LevelAccessor)level);
-        if (spawnData instanceof net.minecraft.world.entity.animal.Rabbit$RabbitGroupData) {
-            n = ((net.minecraft.world.entity.animal.Rabbit$RabbitGroupData)spawnData).rabbitType;
+    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.world.DifficultyInstance $$1, net.minecraft.world.entity.MobSpawnType $$2, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData $$3, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$4) {
+        int $$5 = this.getRandomRabbitType((net.minecraft.world.level.LevelAccessor)$$0);
+        if ($$3 instanceof net.minecraft.world.entity.animal.Rabbit$RabbitGroupData) {
+            $$5 = ((net.minecraft.world.entity.animal.Rabbit$RabbitGroupData)$$3).rabbitType;
         } else {
-            spawnData = new net.minecraft.world.entity.animal.Rabbit$RabbitGroupData(n);
+            $$3 = new net.minecraft.world.entity.animal.Rabbit$RabbitGroupData($$5);
         }
-        this.setRabbitType(n);
-        return super.finalizeSpawn(level, difficulty, reason, spawnData, dataTag);
+        this.setRabbitType($$5);
+        return super.finalizeSpawn($$0, $$1, $$2, $$3, $$4);
     }
 
-    public int getRandomRabbitType(net.minecraft.world.level.LevelAccessor level) {
-        net.minecraft.core.Holder holder = level.getBiome(this.blockPosition());
-        int n = level.getRandom().nextInt(100);
-        if (((net.minecraft.world.level.biome.Biome)holder.value()).getPrecipitation() == net.minecraft.world.level.biome.Biome$Precipitation.SNOW) {
-            return n < 80 ? 1 : 3;
+    public int getRandomRabbitType(net.minecraft.world.level.LevelAccessor $$0) {
+        net.minecraft.core.Holder $$1 = $$0.getBiome(this.blockPosition());
+        int $$2 = $$0.getRandom().nextInt(100);
+        if (((net.minecraft.world.level.biome.Biome)$$1.value()).getPrecipitation() == net.minecraft.world.level.biome.Biome$Precipitation.SNOW) {
+            return $$2 < 80 ? 1 : 3;
         }
-        if (holder.is(net.minecraft.tags.BiomeTags.ONLY_ALLOWS_SNOW_AND_GOLD_RABBITS)) {
+        if ($$1.is(net.minecraft.tags.BiomeTags.ONLY_ALLOWS_SNOW_AND_GOLD_RABBITS)) {
             return 4;
         }
-        return n < 50 ? 0 : (n < 90 ? 5 : 2);
+        return $$2 < 50 ? 0 : ($$2 < 90 ? 5 : 2);
     }
 
     public static boolean checkRabbitSpawnRules(net.minecraft.world.entity.EntityType<net.minecraft.world.entity.animal.Rabbit> $$0, net.minecraft.world.level.LevelAccessor $$1, net.minecraft.world.entity.MobSpawnType $$2, net.minecraft.core.BlockPos $$3, net.minecraft.util.RandomSource $$4) {
@@ -294,16 +310,16 @@
     }
 
     public boolean wantsMoreFood() {
-        return this.moreCarrotTicks == 0;
+        return this.moreCarrotTicks <= 0;
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 1) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 1) {
             this.spawnSprintParticle();
             this.jumpDuration = 10;
             this.jumpTicks = 0;
         } else {
-            super.handleEntityEvent(id);
+            super.handleEntityEvent($$0);
         }
     }
 
@@ -311,6 +327,7 @@
         return new net.minecraft.world.phys.Vec3(0.0, (double)(0.6f * this.getEyeHeight()), (double)(this.getBbWidth() * 0.4f));
     }
 
+    @javax.annotation.Nullable
     public /* synthetic */ net.minecraft.world.entity.AgeableMob getBreedOffspring(net.minecraft.server.level.ServerLevel serverLevel, net.minecraft.world.entity.AgeableMob ageableMob) {
         return this.getBreedOffspring(serverLevel, ageableMob);
     }