--- net/minecraft/world/entity/animal/Rabbit$RaidGardenGoal.java
+++ net/minecraft/world/entity/animal/Rabbit$RaidGardenGoal.java
@@ -10,9 +10,9 @@
     public boolean wantsToRaid;
     public boolean canRaid;
 
-    public net.minecraft.world.entity.animal.Rabbit$RaidGardenGoal(net.minecraft.world.entity.animal.Rabbit rabbit) {
-        super((net.minecraft.world.entity.PathfinderMob)rabbit, (double)0.7f, 16);
-        this.rabbit = rabbit;
+    public net.minecraft.world.entity.animal.Rabbit$RaidGardenGoal(net.minecraft.world.entity.animal.Rabbit $$0) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, (double)0.7f, 16);
+        this.rabbit = $$0;
     }
 
     public boolean canUse() {
@@ -22,7 +22,6 @@
             }
             this.canRaid = false;
             this.wantsToRaid = this.rabbit.wantsMoreFood();
-            this.wantsToRaid = true;
         }
         return super.canUse();
     }
@@ -55,9 +54,9 @@
         }
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(pos);
-        if (blockState.is(net.minecraft.world.level.block.Blocks.FARMLAND) && this.wantsToRaid && !this.canRaid && (blockState = level.getBlockState(pos.above())).getBlock() instanceof net.minecraft.world.level.block.CarrotBlock && ((net.minecraft.world.level.block.CarrotBlock)blockState.getBlock()).isMaxAge(blockState)) {
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        net.minecraft.world.level.block.state.BlockState $$2 = $$0.getBlockState($$1);
+        if ($$2.is(net.minecraft.world.level.block.Blocks.FARMLAND) && this.wantsToRaid && !this.canRaid && ($$2 = $$0.getBlockState($$1.above())).getBlock() instanceof net.minecraft.world.level.block.CarrotBlock && ((net.minecraft.world.level.block.CarrotBlock)$$2.getBlock()).isMaxAge($$2)) {
             this.canRaid = true;
             return true;
         }