--- net/minecraft/world/entity/animal/Rabbit$RabbitPanicGoal.java
+++ net/minecraft/world/entity/animal/Rabbit$RabbitPanicGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Rabbit rabbit;
 
-    public net.minecraft.world.entity.animal.Rabbit$RabbitPanicGoal(net.minecraft.world.entity.animal.Rabbit rabbit, double speedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)rabbit, speedModifier);
-        this.rabbit = rabbit;
+    public net.minecraft.world.entity.animal.Rabbit$RabbitPanicGoal(net.minecraft.world.entity.animal.Rabbit $$0, double $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1);
+        this.rabbit = $$0;
     }
 
     public void tick() {