--- net/minecraft/world/entity/animal/Salmon.java
+++ net/minecraft/world/entity/animal/Salmon.java
@@ -26,7 +26,7 @@
         return net.minecraft.sounds.SoundEvents.SALMON_DEATH;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.SALMON_HURT;
     }
 