--- net/minecraft/world/entity/animal/Sheep.java
+++ net/minecraft/world/entity/animal/Sheep.java
@@ -14,17 +14,17 @@
     public int eatAnimationTick;
     public net.minecraft.world.entity.ai.goal.EatBlockGoal eatBlockGoal;
 
-    public static float[] createSheepColor(net.minecraft.world.item.DyeColor dyeColor) {
-        if (dyeColor == net.minecraft.world.item.DyeColor.WHITE) {
+    public static float[] createSheepColor(net.minecraft.world.item.DyeColor $$0) {
+        if ($$0 == net.minecraft.world.item.DyeColor.WHITE) {
             return new float[]{0.9019608f, 0.9019608f, 0.9019608f};
         }
-        float[] fArray = dyeColor.getTextureDiffuseColors();
-        float f = 0.75f;
-        return new float[]{fArray[0] * 0.75f, fArray[1] * 0.75f, fArray[2] * 0.75f};
+        float[] $$1 = $$0.getTextureDiffuseColors();
+        float $$2 = 0.75f;
+        return new float[]{$$1[0] * 0.75f, $$1[1] * 0.75f, $$1[2] * 0.75f};
     }
 
-    public static float[] getColorArray(net.minecraft.world.item.DyeColor dyeColor) {
-        return (float[])COLORARRAY_BY_COLOR.get((java.lang.Object)dyeColor);
+    public static float[] getColorArray(net.minecraft.world.item.DyeColor $$0) {
+        return (float[])COLORARRAY_BY_COLOR.get((java.lang.Object)$$0);
     }
 
     public net.minecraft.world.entity.animal.Sheep(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.animal.Sheep> $$0, net.minecraft.world.level.Level $$1) {
@@ -69,66 +69,36 @@
         if (this.isSheared()) {
             return this.getType().getDefaultLootTable();
         }
-        switch (net.minecraft.world.entity.animal.Sheep$2.$SwitchMap$net$minecraft$world$item$DyeColor[this.getColor().ordinal()]) {
-            default: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_WHITE;
-            }
-            case 2: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_ORANGE;
-            }
-            case 3: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_MAGENTA;
-            }
-            case 4: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_LIGHT_BLUE;
-            }
-            case 5: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_YELLOW;
-            }
-            case 6: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_LIME;
-            }
-            case 7: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_PINK;
-            }
-            case 8: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_GRAY;
-            }
-            case 9: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_LIGHT_GRAY;
-            }
-            case 10: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_CYAN;
-            }
-            case 11: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_PURPLE;
-            }
-            case 12: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_BLUE;
-            }
-            case 13: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_BROWN;
-            }
-            case 14: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_GREEN;
-            }
-            case 15: {
-                return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_RED;
-            }
-            case 16: 
-        }
-        return net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_BLACK;
+        return switch (net.minecraft.world.entity.animal.Sheep$2.$SwitchMap$net$minecraft$world$item$DyeColor[this.getColor().ordinal()]) {
+            default -> throw new java.lang.IncompatibleClassChangeError();
+            case 1 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_WHITE;
+            case 2 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_ORANGE;
+            case 3 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_MAGENTA;
+            case 4 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_LIGHT_BLUE;
+            case 5 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_YELLOW;
+            case 6 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_LIME;
+            case 7 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_PINK;
+            case 8 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_GRAY;
+            case 9 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_LIGHT_GRAY;
+            case 10 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_CYAN;
+            case 11 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_PURPLE;
+            case 12 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_BLUE;
+            case 13 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_BROWN;
+            case 14 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_GREEN;
+            case 15 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_RED;
+            case 16 -> net.minecraft.world.level.storage.loot.BuiltInLootTables.SHEEP_BLACK;
+        };
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 10) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 10) {
             this.eatAnimationTick = 40;
         } else {
-            super.handleEntityEvent(id);
+            super.handleEntityEvent($$0);
         }
     }
 
-    public float getHeadEatPositionScale(float partialTick) {
+    public float getHeadEatPositionScale(float $$0) {
         if (this.eatAnimationTick <= 0) {
             return 0.0f;
         }
@@ -136,15 +106,15 @@
             return 1.0f;
         }
         if (this.eatAnimationTick < 4) {
-            return ((float)this.eatAnimationTick - partialTick) / 4.0f;
+            return ((float)this.eatAnimationTick - $$0) / 4.0f;
         }
-        return -((float)(this.eatAnimationTick - 40) - partialTick) / 4.0f;
+        return -((float)(this.eatAnimationTick - 40) - $$0) / 4.0f;
     }
 
-    public float getHeadEatAngleScale(float partialTick) {
+    public float getHeadEatAngleScale(float $$0) {
         if (this.eatAnimationTick > 4 && this.eatAnimationTick <= 36) {
-            float f = ((float)(this.eatAnimationTick - 4) - partialTick) / 32.0f;
-            return 0.62831855f + 0.21991149f * net.minecraft.util.Mth.sin((float)(f * 28.7f));
+            float $$1 = ((float)(this.eatAnimationTick - 4) - $$0) / 32.0f;
+            return 0.62831855f + 0.21991149f * net.minecraft.util.Mth.sin((float)($$1 * 28.7f));
         }
         if (this.eatAnimationTick > 0) {
             return 0.62831855f;
@@ -152,28 +122,28 @@
         return this.getXRot() * ((float)java.lang.Math.PI / 180);
     }
 
-    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        if (itemStack.is(net.minecraft.world.item.Items.SHEARS)) {
+    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$12) {
+        net.minecraft.world.item.ItemStack $$2 = $$0.getItemInHand($$12);
+        if ($$2.is(net.minecraft.world.item.Items.SHEARS)) {
             if (!this.level.isClientSide && this.readyForShearing()) {
                 this.shear(net.minecraft.sounds.SoundSource.PLAYERS);
-                this.gameEvent(net.minecraft.world.level.gameevent.GameEvent.SHEAR, (net.minecraft.world.entity.Entity)player);
-                itemStack.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)player, entity -> entity.broadcastBreakEvent(hand));
+                this.gameEvent(net.minecraft.world.level.gameevent.GameEvent.SHEAR, (net.minecraft.world.entity.Entity)$$0);
+                $$2.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)$$0, $$1 -> $$1.broadcastBreakEvent($$12));
                 return net.minecraft.world.InteractionResult.SUCCESS;
             }
             return net.minecraft.world.InteractionResult.CONSUME;
         }
-        return super.mobInteract(player, hand);
+        return super.mobInteract($$0, $$12);
     }
 
-    public void shear(net.minecraft.sounds.SoundSource category) {
-        this.level.playSound(null, (net.minecraft.world.entity.Entity)this, net.minecraft.sounds.SoundEvents.SHEEP_SHEAR, category, 1.0f, 1.0f);
+    public void shear(net.minecraft.sounds.SoundSource $$0) {
+        this.level.playSound(null, (net.minecraft.world.entity.Entity)this, net.minecraft.sounds.SoundEvents.SHEEP_SHEAR, $$0, 1.0f, 1.0f);
         this.setSheared(true);
-        int n = 1 + this.random.nextInt(3);
-        for (int i = 0; i < n; ++i) {
-            net.minecraft.world.entity.item.ItemEntity itemEntity = this.spawnAtLocation((net.minecraft.world.level.ItemLike)ITEM_BY_DYE.get((java.lang.Object)this.getColor()), 1);
-            if (itemEntity == null) continue;
-            itemEntity.setDeltaMovement(itemEntity.getDeltaMovement().add((double)((this.random.nextFloat() - this.random.nextFloat()) * 0.1f), (double)(this.random.nextFloat() * 0.05f), (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.1f)));
+        int $$1 = 1 + this.random.nextInt(3);
+        for (int $$2 = 0; $$2 < $$1; ++$$2) {
+            net.minecraft.world.entity.item.ItemEntity $$3 = this.spawnAtLocation((net.minecraft.world.level.ItemLike)ITEM_BY_DYE.get((java.lang.Object)this.getColor()), 1);
+            if ($$3 == null) continue;
+            $$3.setDeltaMovement($$3.getDeltaMovement().add((double)((this.random.nextFloat() - this.random.nextFloat()) * 0.1f), (double)(this.random.nextFloat() * 0.05f), (double)((this.random.nextFloat() - this.random.nextFloat()) * 0.1f)));
         }
     }
 
@@ -181,23 +151,23 @@
         return this.isAlive() && !this.isSheared() && !this.isBaby();
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putBoolean("Sheared", this.isSheared());
-        compound.putByte("Color", (byte)this.getColor().getId());
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putBoolean("Sheared", this.isSheared());
+        $$0.putByte("Color", (byte)this.getColor().getId());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.setSheared(compound.getBoolean("Sheared"));
-        this.setColor(net.minecraft.world.item.DyeColor.byId((int)compound.getByte("Color")));
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.setSheared($$0.getBoolean("Sheared"));
+        this.setColor(net.minecraft.world.item.DyeColor.byId((int)$$0.getByte("Color")));
     }
 
     public net.minecraft.sounds.SoundEvent getAmbientSound() {
         return net.minecraft.sounds.SoundEvents.SHEEP_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.SHEEP_HURT;
     }
 
@@ -205,7 +175,7 @@
         return net.minecraft.sounds.SoundEvents.SHEEP_DEATH;
     }
 
-    public void playStepSound(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState block) {
+    public void playStepSound(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         this.playSound(net.minecraft.sounds.SoundEvents.SHEEP_STEP, 0.15f, 1.0f);
     }
 
@@ -213,21 +183,21 @@
         return net.minecraft.world.item.DyeColor.byId((int)((java.lang.Byte)this.entityData.get(DATA_WOOL_ID) & 0xF));
     }
 
-    public void setColor(net.minecraft.world.item.DyeColor dyeColor) {
-        byte by = (java.lang.Byte)this.entityData.get(DATA_WOOL_ID);
-        this.entityData.set(DATA_WOOL_ID, (java.lang.Object)((byte)(by & 0xF0 | dyeColor.getId() & 0xF)));
+    public void setColor(net.minecraft.world.item.DyeColor $$0) {
+        byte $$1 = (java.lang.Byte)this.entityData.get(DATA_WOOL_ID);
+        this.entityData.set(DATA_WOOL_ID, (java.lang.Object)((byte)($$1 & 0xF0 | $$0.getId() & 0xF)));
     }
 
     public boolean isSheared() {
         return ((java.lang.Byte)this.entityData.get(DATA_WOOL_ID) & 0x10) != 0;
     }
 
-    public void setSheared(boolean sheared) {
-        byte by = (java.lang.Byte)this.entityData.get(DATA_WOOL_ID);
-        if (sheared) {
-            this.entityData.set(DATA_WOOL_ID, (java.lang.Object)((byte)(by | 0x10)));
+    public void setSheared(boolean $$0) {
+        byte $$1 = (java.lang.Byte)this.entityData.get(DATA_WOOL_ID);
+        if ($$0) {
+            this.entityData.set(DATA_WOOL_ID, (java.lang.Object)((byte)($$1 | 0x10)));
         } else {
-            this.entityData.set(DATA_WOOL_ID, (java.lang.Object)((byte)(by & 0xFFFFFFEF)));
+            this.entityData.set(DATA_WOOL_ID, (java.lang.Object)((byte)($$1 & 0xFFFFFFEF)));
         }
     }
 
@@ -251,11 +221,13 @@
         return net.minecraft.world.item.DyeColor.WHITE;
     }
 
+    @javax.annotation.Nullable
     public net.minecraft.world.entity.animal.Sheep getBreedOffspring(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.AgeableMob $$1) {
-        net.minecraft.world.entity.animal.Sheep $$2 = (net.minecraft.world.entity.animal.Sheep)$$1;
-        net.minecraft.world.entity.animal.Sheep $$3 = (net.minecraft.world.entity.animal.Sheep)net.minecraft.world.entity.EntityType.SHEEP.create((net.minecraft.world.level.Level)$$0);
-        $$3.setColor(this.getOffspringColor((net.minecraft.world.entity.animal.Animal)this, (net.minecraft.world.entity.animal.Animal)$$2));
-        return $$3;
+        net.minecraft.world.entity.animal.Sheep $$2 = (net.minecraft.world.entity.animal.Sheep)net.minecraft.world.entity.EntityType.SHEEP.create((net.minecraft.world.level.Level)$$0);
+        if ($$2 != null) {
+            $$2.setColor(this.getOffspringColor((net.minecraft.world.entity.animal.Animal)this, (net.minecraft.world.entity.animal.Animal)((net.minecraft.world.entity.animal.Sheep)$$1)));
+        }
+        return $$2;
     }
 
     public void ate() {
@@ -267,77 +239,78 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor level, net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.MobSpawnType reason, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData spawnData, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag dataTag) {
-        this.setColor(net.minecraft.world.entity.animal.Sheep.getRandomSheepColor((net.minecraft.util.RandomSource)level.getRandom()));
-        return super.finalizeSpawn(level, difficulty, reason, spawnData, dataTag);
+    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.world.DifficultyInstance $$1, net.minecraft.world.entity.MobSpawnType $$2, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData $$3, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$4) {
+        this.setColor(net.minecraft.world.entity.animal.Sheep.getRandomSheepColor((net.minecraft.util.RandomSource)$$0.getRandom()));
+        return super.finalizeSpawn($$0, $$1, $$2, $$3, $$4);
     }
 
-    public net.minecraft.world.item.DyeColor getOffspringColor(net.minecraft.world.entity.animal.Animal father, net.minecraft.world.entity.animal.Animal mother) {
-        net.minecraft.world.item.DyeColor dyeColor = ((net.minecraft.world.entity.animal.Sheep)father).getColor();
-        net.minecraft.world.item.DyeColor dyeColor2 = ((net.minecraft.world.entity.animal.Sheep)mother).getColor();
-        net.minecraft.world.inventory.CraftingContainer craftingContainer = net.minecraft.world.entity.animal.Sheep.makeContainer((net.minecraft.world.item.DyeColor)dyeColor, (net.minecraft.world.item.DyeColor)dyeColor2);
-        return (net.minecraft.world.item.DyeColor)this.level.getRecipeManager().getRecipeFor(net.minecraft.world.item.crafting.RecipeType.CRAFTING, (net.minecraft.world.Container)craftingContainer, this.level).map(craftingRecipe -> craftingRecipe.assemble((net.minecraft.world.Container)craftingContainer)).map(net.minecraft.world.item.ItemStack::getItem).filter(arg_0 -> net.minecraft.world.item.DyeItem.class.isInstance(arg_0)).map(arg_0 -> net.minecraft.world.item.DyeItem.class.cast(arg_0)).map(net.minecraft.world.item.DyeItem::getDyeColor).orElseGet(() -> this.level.random.nextBoolean() ? dyeColor : dyeColor2);
+    public net.minecraft.world.item.DyeColor getOffspringColor(net.minecraft.world.entity.animal.Animal $$0, net.minecraft.world.entity.animal.Animal $$12) {
+        net.minecraft.world.item.DyeColor $$2 = ((net.minecraft.world.entity.animal.Sheep)$$0).getColor();
+        net.minecraft.world.item.DyeColor $$3 = ((net.minecraft.world.entity.animal.Sheep)$$12).getColor();
+        net.minecraft.world.inventory.CraftingContainer $$4 = net.minecraft.world.entity.animal.Sheep.makeContainer((net.minecraft.world.item.DyeColor)$$2, (net.minecraft.world.item.DyeColor)$$3);
+        return (net.minecraft.world.item.DyeColor)this.level.getRecipeManager().getRecipeFor(net.minecraft.world.item.crafting.RecipeType.CRAFTING, (net.minecraft.world.Container)$$4, this.level).map($$1 -> $$1.assemble((net.minecraft.world.Container)$$4)).map(net.minecraft.world.item.ItemStack::getItem).filter(arg_0 -> net.minecraft.world.item.DyeItem.class.isInstance(arg_0)).map(arg_0 -> net.minecraft.world.item.DyeItem.class.cast(arg_0)).map(net.minecraft.world.item.DyeItem::getDyeColor).orElseGet(() -> this.level.random.nextBoolean() ? $$2 : $$3);
     }
 
-    public static net.minecraft.world.inventory.CraftingContainer makeContainer(net.minecraft.world.item.DyeColor fatherColor, net.minecraft.world.item.DyeColor motherColor) {
-        net.minecraft.world.inventory.CraftingContainer craftingContainer = new net.minecraft.world.inventory.CraftingContainer((net.minecraft.world.inventory.AbstractContainerMenu)new /* Unavailable Anonymous Inner Class!! */, 2, 1);
-        craftingContainer.setItem(0, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.DyeItem.byColor((net.minecraft.world.item.DyeColor)fatherColor)));
-        craftingContainer.setItem(1, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.DyeItem.byColor((net.minecraft.world.item.DyeColor)motherColor)));
-        return craftingContainer;
+    public static net.minecraft.world.inventory.CraftingContainer makeContainer(net.minecraft.world.item.DyeColor $$0, net.minecraft.world.item.DyeColor $$1) {
+        net.minecraft.world.inventory.CraftingContainer $$2 = new net.minecraft.world.inventory.CraftingContainer((net.minecraft.world.inventory.AbstractContainerMenu)new /* Unavailable Anonymous Inner Class!! */, 2, 1);
+        $$2.setItem(0, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.DyeItem.byColor((net.minecraft.world.item.DyeColor)$$0)));
+        $$2.setItem(1, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.DyeItem.byColor((net.minecraft.world.item.DyeColor)$$1)));
+        return $$2;
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
-        return 0.95f * size.height;
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
+        return 0.95f * $$1.height;
     }
 
+    @javax.annotation.Nullable
     public /* synthetic */ net.minecraft.world.entity.AgeableMob getBreedOffspring(net.minecraft.server.level.ServerLevel serverLevel, net.minecraft.world.entity.AgeableMob ageableMob) {
         return this.getBreedOffspring(serverLevel, ageableMob);
     }
 
-    public static /* synthetic */ net.minecraft.world.item.DyeColor lambda$static$1(net.minecraft.world.item.DyeColor color) {
-        return color;
+    public static /* synthetic */ net.minecraft.world.item.DyeColor lambda$static$1(net.minecraft.world.item.DyeColor $$0) {
+        return $$0;
     }
 
-    public static /* synthetic */ void lambda$static$0(java.util.EnumMap map) {
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.WHITE, (java.lang.Object)net.minecraft.world.level.block.Blocks.WHITE_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.ORANGE, (java.lang.Object)net.minecraft.world.level.block.Blocks.ORANGE_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.MAGENTA, (java.lang.Object)net.minecraft.world.level.block.Blocks.MAGENTA_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_BLUE_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.YELLOW, (java.lang.Object)net.minecraft.world.level.block.Blocks.YELLOW_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIME, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIME_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PINK, (java.lang.Object)net.minecraft.world.level.block.Blocks.PINK_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.GRAY_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_GRAY_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.CYAN, (java.lang.Object)net.minecraft.world.level.block.Blocks.CYAN_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PURPLE, (java.lang.Object)net.minecraft.world.level.block.Blocks.PURPLE_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLUE_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BROWN, (java.lang.Object)net.minecraft.world.level.block.Blocks.BROWN_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GREEN, (java.lang.Object)net.minecraft.world.level.block.Blocks.GREEN_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.RED, (java.lang.Object)net.minecraft.world.level.block.Blocks.RED_WOOL);
-        map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLACK, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLACK_WOOL);
+    public static /* synthetic */ void lambda$static$0(java.util.EnumMap $$0) {
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.WHITE, (java.lang.Object)net.minecraft.world.level.block.Blocks.WHITE_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.ORANGE, (java.lang.Object)net.minecraft.world.level.block.Blocks.ORANGE_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.MAGENTA, (java.lang.Object)net.minecraft.world.level.block.Blocks.MAGENTA_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_BLUE_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.YELLOW, (java.lang.Object)net.minecraft.world.level.block.Blocks.YELLOW_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIME, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIME_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PINK, (java.lang.Object)net.minecraft.world.level.block.Blocks.PINK_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.GRAY_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_GRAY_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.CYAN, (java.lang.Object)net.minecraft.world.level.block.Blocks.CYAN_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PURPLE, (java.lang.Object)net.minecraft.world.level.block.Blocks.PURPLE_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLUE_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BROWN, (java.lang.Object)net.minecraft.world.level.block.Blocks.BROWN_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GREEN, (java.lang.Object)net.minecraft.world.level.block.Blocks.GREEN_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.RED, (java.lang.Object)net.minecraft.world.level.block.Blocks.RED_WOOL);
+        $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLACK, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLACK_WOOL);
     }
 
     static {
         DATA_WOOL_ID = net.minecraft.network.syncher.SynchedEntityData.defineId(net.minecraft.world.entity.animal.Sheep.class, (net.minecraft.network.syncher.EntityDataSerializer)net.minecraft.network.syncher.EntityDataSerializers.BYTE);
-        ITEM_BY_DYE = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newEnumMap(net.minecraft.world.item.DyeColor.class), map -> {
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.WHITE, (java.lang.Object)net.minecraft.world.level.block.Blocks.WHITE_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.ORANGE, (java.lang.Object)net.minecraft.world.level.block.Blocks.ORANGE_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.MAGENTA, (java.lang.Object)net.minecraft.world.level.block.Blocks.MAGENTA_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_BLUE_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.YELLOW, (java.lang.Object)net.minecraft.world.level.block.Blocks.YELLOW_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIME, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIME_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PINK, (java.lang.Object)net.minecraft.world.level.block.Blocks.PINK_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.GRAY_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_GRAY_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.CYAN, (java.lang.Object)net.minecraft.world.level.block.Blocks.CYAN_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PURPLE, (java.lang.Object)net.minecraft.world.level.block.Blocks.PURPLE_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLUE_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BROWN, (java.lang.Object)net.minecraft.world.level.block.Blocks.BROWN_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GREEN, (java.lang.Object)net.minecraft.world.level.block.Blocks.GREEN_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.RED, (java.lang.Object)net.minecraft.world.level.block.Blocks.RED_WOOL);
-            map.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLACK, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLACK_WOOL);
+        ITEM_BY_DYE = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newEnumMap(net.minecraft.world.item.DyeColor.class), $$0 -> {
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.WHITE, (java.lang.Object)net.minecraft.world.level.block.Blocks.WHITE_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.ORANGE, (java.lang.Object)net.minecraft.world.level.block.Blocks.ORANGE_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.MAGENTA, (java.lang.Object)net.minecraft.world.level.block.Blocks.MAGENTA_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_BLUE_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.YELLOW, (java.lang.Object)net.minecraft.world.level.block.Blocks.YELLOW_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIME, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIME_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PINK, (java.lang.Object)net.minecraft.world.level.block.Blocks.PINK_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.GRAY_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.LIGHT_GRAY, (java.lang.Object)net.minecraft.world.level.block.Blocks.LIGHT_GRAY_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.CYAN, (java.lang.Object)net.minecraft.world.level.block.Blocks.CYAN_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.PURPLE, (java.lang.Object)net.minecraft.world.level.block.Blocks.PURPLE_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLUE, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLUE_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BROWN, (java.lang.Object)net.minecraft.world.level.block.Blocks.BROWN_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.GREEN, (java.lang.Object)net.minecraft.world.level.block.Blocks.GREEN_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.RED, (java.lang.Object)net.minecraft.world.level.block.Blocks.RED_WOOL);
+            $$0.put((java.lang.Enum)net.minecraft.world.item.DyeColor.BLACK, (java.lang.Object)net.minecraft.world.level.block.Blocks.BLACK_WOOL);
         });
-        COLORARRAY_BY_COLOR = com.google.common.collect.Maps.newEnumMap((java.util.Map)((java.util.Map)java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.item.DyeColor.values()).collect(java.util.stream.Collectors.toMap(color -> color, net.minecraft.world.entity.animal.Sheep::createSheepColor))));
+        COLORARRAY_BY_COLOR = com.google.common.collect.Maps.newEnumMap((java.util.Map)((java.util.Map)java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.item.DyeColor.values()).collect(java.util.stream.Collectors.toMap($$0 -> $$0, net.minecraft.world.entity.animal.Sheep::createSheepColor))));
     }
 }
 