--- net/minecraft/world/entity/animal/ShoulderRidingEntity.java
+++ net/minecraft/world/entity/animal/ShoulderRidingEntity.java
@@ -13,11 +13,11 @@
         super($$0, $$1);
     }
 
-    public boolean setEntityOnShoulder(net.minecraft.server.level.ServerPlayer player) {
-        net.minecraft.nbt.CompoundTag compoundTag = new net.minecraft.nbt.CompoundTag();
-        compoundTag.putString("id", this.getEncodeId());
-        this.saveWithoutId(compoundTag);
-        if (player.setEntityOnShoulder(compoundTag)) {
+    public boolean setEntityOnShoulder(net.minecraft.server.level.ServerPlayer $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = new net.minecraft.nbt.CompoundTag();
+        $$1.putString("id", this.getEncodeId());
+        this.saveWithoutId($$1);
+        if ($$0.setEntityOnShoulder($$1)) {
             this.discard();
             return true;
         }