--- net/minecraft/world/entity/animal/SnowGolem.java
+++ net/minecraft/world/entity/animal/SnowGolem.java
@@ -21,7 +21,7 @@
         this.goalSelector.addGoal(2, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.WaterAvoidingRandomStrollGoal((net.minecraft.world.entity.PathfinderMob)this, 1.0, 1.0000001E-5f));
         this.goalSelector.addGoal(3, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.LookAtPlayerGoal((net.minecraft.world.entity.Mob)this, net.minecraft.world.entity.player.Player.class, 6.0f));
         this.goalSelector.addGoal(4, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.RandomLookAroundGoal((net.minecraft.world.entity.Mob)this));
-        this.targetSelector.addGoal(1, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal((net.minecraft.world.entity.Mob)this, net.minecraft.world.entity.Mob.class, 10, true, false, entity -> entity instanceof net.minecraft.world.entity.monster.Enemy));
+        this.targetSelector.addGoal(1, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal((net.minecraft.world.entity.Mob)this, net.minecraft.world.entity.Mob.class, 10, true, false, $$0 -> $$0 instanceof net.minecraft.world.entity.monster.Enemy));
     }
 
     public static net.minecraft.world.entity.ai.attributes.AttributeSupplier$Builder createAttributes() {
@@ -33,15 +33,15 @@
         this.entityData.define(DATA_PUMPKIN_ID, (java.lang.Object)16);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putBoolean("Pumpkin", this.hasPumpkin());
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putBoolean("Pumpkin", this.hasPumpkin());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        if (compound.contains("Pumpkin")) {
-            this.setPumpkin(compound.getBoolean("Pumpkin"));
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        if ($$0.contains("Pumpkin")) {
+            this.setPumpkin($$0.getBoolean("Pumpkin"));
         }
     }
 
@@ -74,37 +74,37 @@
         }
     }
 
-    public void performRangedAttack(net.minecraft.world.entity.LivingEntity target, float distanceFactor) {
-        net.minecraft.world.entity.projectile.Snowball snowball = new net.minecraft.world.entity.projectile.Snowball(this.level, (net.minecraft.world.entity.LivingEntity)this);
-        double d = target.getEyeY() - (double)1.1f;
-        double d2 = target.getX() - this.getX();
-        double d3 = d - snowball.getY();
-        double d4 = target.getZ() - this.getZ();
-        double d5 = java.lang.Math.sqrt((double)(d2 * d2 + d4 * d4)) * (double)0.2f;
-        snowball.shoot(d2, d3 + d5, d4, 1.6f, 12.0f);
+    public void performRangedAttack(net.minecraft.world.entity.LivingEntity $$0, float $$1) {
+        net.minecraft.world.entity.projectile.Snowball $$2 = new net.minecraft.world.entity.projectile.Snowball(this.level, (net.minecraft.world.entity.LivingEntity)this);
+        double $$3 = $$0.getEyeY() - (double)1.1f;
+        double $$4 = $$0.getX() - this.getX();
+        double $$5 = $$3 - $$2.getY();
+        double $$6 = $$0.getZ() - this.getZ();
+        double $$7 = java.lang.Math.sqrt((double)($$4 * $$4 + $$6 * $$6)) * (double)0.2f;
+        $$2.shoot($$4, $$5 + $$7, $$6, 1.6f, 12.0f);
         this.playSound(net.minecraft.sounds.SoundEvents.SNOW_GOLEM_SHOOT, 1.0f, 0.4f / (this.getRandom().nextFloat() * 0.4f + 0.8f));
-        this.level.addFreshEntity((net.minecraft.world.entity.Entity)snowball);
+        this.level.addFreshEntity((net.minecraft.world.entity.Entity)$$2);
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
         return 1.7f;
     }
 
-    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        if (itemStack.is(net.minecraft.world.item.Items.SHEARS) && this.readyForShearing()) {
+    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$12) {
+        net.minecraft.world.item.ItemStack $$2 = $$0.getItemInHand($$12);
+        if ($$2.is(net.minecraft.world.item.Items.SHEARS) && this.readyForShearing()) {
             this.shear(net.minecraft.sounds.SoundSource.PLAYERS);
-            this.gameEvent(net.minecraft.world.level.gameevent.GameEvent.SHEAR, (net.minecraft.world.entity.Entity)player);
+            this.gameEvent(net.minecraft.world.level.gameevent.GameEvent.SHEAR, (net.minecraft.world.entity.Entity)$$0);
             if (!this.level.isClientSide) {
-                itemStack.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)player, entity -> entity.broadcastBreakEvent(hand));
+                $$2.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)$$0, $$1 -> $$1.broadcastBreakEvent($$12));
             }
             return net.minecraft.world.InteractionResult.sidedSuccess((boolean)this.level.isClientSide);
         }
         return net.minecraft.world.InteractionResult.PASS;
     }
 
-    public void shear(net.minecraft.sounds.SoundSource category) {
-        this.level.playSound(null, (net.minecraft.world.entity.Entity)this, net.minecraft.sounds.SoundEvents.SNOW_GOLEM_SHEAR, category, 1.0f, 1.0f);
+    public void shear(net.minecraft.sounds.SoundSource $$0) {
+        this.level.playSound(null, (net.minecraft.world.entity.Entity)this, net.minecraft.sounds.SoundEvents.SNOW_GOLEM_SHEAR, $$0, 1.0f, 1.0f);
         if (!this.level.isClientSide()) {
             this.setPumpkin(false);
             this.spawnAtLocation(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CARVED_PUMPKIN), 1.7f);
@@ -119,12 +119,12 @@
         return ((java.lang.Byte)this.entityData.get(DATA_PUMPKIN_ID) & 0x10) != 0;
     }
 
-    public void setPumpkin(boolean pumpkinEquipped) {
-        byte by = (java.lang.Byte)this.entityData.get(DATA_PUMPKIN_ID);
-        if (pumpkinEquipped) {
-            this.entityData.set(DATA_PUMPKIN_ID, (java.lang.Object)((byte)(by | 0x10)));
+    public void setPumpkin(boolean $$0) {
+        byte $$1 = (java.lang.Byte)this.entityData.get(DATA_PUMPKIN_ID);
+        if ($$0) {
+            this.entityData.set(DATA_PUMPKIN_ID, (java.lang.Object)((byte)($$1 | 0x10)));
         } else {
-            this.entityData.set(DATA_PUMPKIN_ID, (java.lang.Object)((byte)(by & 0xFFFFFFEF)));
+            this.entityData.set(DATA_PUMPKIN_ID, (java.lang.Object)((byte)($$1 & 0xFFFFFFEF)));
         }
     }
 
@@ -134,7 +134,7 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.SNOW_GOLEM_HURT;
     }
 