--- net/minecraft/world/entity/animal/Squid.java
+++ net/minecraft/world/entity/animal/Squid.java
@@ -36,15 +36,15 @@
         return net.minecraft.world.entity.Mob.createMobAttributes().add(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, 10.0);
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
-        return size.height * 0.5f;
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
+        return $$1.height * 0.5f;
     }
 
     public net.minecraft.sounds.SoundEvent getAmbientSound() {
         return net.minecraft.sounds.SoundEvents.SQUID_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.SQUID_HURT;
     }
 
@@ -125,8 +125,8 @@
         }
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
-        if (super.hurt(source, amount) && this.getLastHurtByMob() != null) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
+        if (super.hurt($$0, $$1) && this.getLastHurtByMob() != null) {
             if (!this.level.isClientSide) {
                 this.spawnInk();
             }
@@ -135,10 +135,10 @@
         return false;
     }
 
-    public net.minecraft.world.phys.Vec3 rotateVector(net.minecraft.world.phys.Vec3 vector) {
-        net.minecraft.world.phys.Vec3 vec3 = vector.xRot(this.xBodyRotO * ((float)java.lang.Math.PI / 180));
-        vec3 = vec3.yRot(-this.yBodyRotO * ((float)java.lang.Math.PI / 180));
-        return vec3;
+    public net.minecraft.world.phys.Vec3 rotateVector(net.minecraft.world.phys.Vec3 $$0) {
+        net.minecraft.world.phys.Vec3 $$1 = $$0.xRot(this.xBodyRotO * ((float)java.lang.Math.PI / 180));
+        $$1 = $$1.yRot(-this.yBodyRotO * ((float)java.lang.Math.PI / 180));
+        return $$1;
     }
 
     public void spawnInk() {
@@ -155,22 +155,22 @@
         return net.minecraft.core.particles.ParticleTypes.SQUID_INK;
     }
 
-    public void travel(net.minecraft.world.phys.Vec3 travelVector) {
+    public void travel(net.minecraft.world.phys.Vec3 $$0) {
         this.move(net.minecraft.world.entity.MoverType.SELF, this.getDeltaMovement());
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 19) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 19) {
             this.tentacleMovement = 0.0f;
         } else {
-            super.handleEntityEvent(id);
+            super.handleEntityEvent($$0);
         }
     }
 
-    public void setMovementVector(float tx, float ty, float tz) {
-        this.tx = tx;
-        this.ty = ty;
-        this.tz = tz;
+    public void setMovementVector(float $$0, float $$1, float $$2) {
+        this.tx = $$0;
+        this.ty = $$1;
+        this.tz = $$2;
     }
 
     public boolean hasMovementVector() {