--- net/minecraft/world/entity/animal/TropicalFish$TropicalFishGroupData.java
+++ net/minecraft/world/entity/animal/TropicalFish$TropicalFishGroupData.java
@@ -11,12 +11,12 @@
     public int baseColor;
     public int patternColor;
 
-    public net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData(net.minecraft.world.entity.animal.TropicalFish fish, int base, int pattern, int baseColor, int patternColor) {
-        super((net.minecraft.world.entity.animal.AbstractSchoolingFish)fish);
-        this.base = base;
-        this.pattern = pattern;
-        this.baseColor = baseColor;
-        this.patternColor = patternColor;
+    public net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData(net.minecraft.world.entity.animal.TropicalFish $$0, int $$1, int $$2, int $$3, int $$4) {
+        super((net.minecraft.world.entity.animal.AbstractSchoolingFish)$$0);
+        this.base = $$1;
+        this.pattern = $$2;
+        this.baseColor = $$3;
+        this.patternColor = $$4;
     }
 }
 