--- net/minecraft/world/entity/animal/TropicalFish$Pattern.java
+++ net/minecraft/world/entity/animal/TropicalFish$Pattern.java
@@ -31,10 +31,10 @@
         return (net.minecraft.world.entity.animal.TropicalFish$Pattern)java.lang.Enum.valueOf(net.minecraft.world.entity.animal.TropicalFish$Pattern.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.animal.TropicalFish$Pattern(int base, int index) {
+    public net.minecraft.world.entity.animal.TropicalFish$Pattern(int $$0, int $$1) {
         super(string, n);
-        this.base = base;
-        this.index = index;
+        this.base = $$0;
+        this.index = $$1;
     }
 
     public int getBase() {
@@ -45,8 +45,8 @@
         return this.index;
     }
 
-    public static java.lang.String getPatternName(int baseVariant, int patternVariant) {
-        return VALUES[patternVariant + 6 * baseVariant].getName();
+    public static java.lang.String getPatternName(int $$0, int $$1) {
+        return VALUES[$$1 + 6 * $$0].getName();
     }
 
     public java.lang.String getName() {