--- net/minecraft/world/entity/animal/TropicalFish.java
+++ net/minecraft/world/entity/animal/TropicalFish.java
@@ -19,8 +19,8 @@
     public static int[] COMMON_VARIANTS;
     public boolean isSchool;
 
-    public static int calculateVariant(net.minecraft.world.entity.animal.TropicalFish$Pattern pattern, net.minecraft.world.item.DyeColor baseColor, net.minecraft.world.item.DyeColor patternColor) {
-        return pattern.getBase() & 0xFF | (pattern.getIndex() & 0xFF) << 8 | (baseColor.getId() & 0xFF) << 16 | (patternColor.getId() & 0xFF) << 24;
+    public static int calculateVariant(net.minecraft.world.entity.animal.TropicalFish$Pattern $$0, net.minecraft.world.item.DyeColor $$1, net.minecraft.world.item.DyeColor $$2) {
+        return $$0.getBase() & 0xFF | ($$0.getIndex() & 0xFF) << 8 | ($$1.getId() & 0xFF) << 16 | ($$2.getId() & 0xFF) << 24;
     }
 
     public net.minecraft.world.entity.animal.TropicalFish(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.animal.TropicalFish> $$0, net.minecraft.world.level.Level $$1) {
@@ -28,22 +28,22 @@
         this.isSchool = true;
     }
 
-    public static java.lang.String getPredefinedName(int variantId) {
-        return "entity.minecraft.tropical_fish.predefined." + variantId;
+    public static java.lang.String getPredefinedName(int $$0) {
+        return "entity.minecraft.tropical_fish.predefined." + $$0;
     }
 
-    public static net.minecraft.world.item.DyeColor getBaseColor(int variantId) {
-        return net.minecraft.world.item.DyeColor.byId((int)net.minecraft.world.entity.animal.TropicalFish.getBaseColorIdx((int)variantId));
+    public static net.minecraft.world.item.DyeColor getBaseColor(int $$0) {
+        return net.minecraft.world.item.DyeColor.byId((int)net.minecraft.world.entity.animal.TropicalFish.getBaseColorIdx((int)$$0));
     }
 
-    public static net.minecraft.world.item.DyeColor getPatternColor(int variantId) {
-        return net.minecraft.world.item.DyeColor.byId((int)net.minecraft.world.entity.animal.TropicalFish.getPatternColorIdx((int)variantId));
+    public static net.minecraft.world.item.DyeColor getPatternColor(int $$0) {
+        return net.minecraft.world.item.DyeColor.byId((int)net.minecraft.world.entity.animal.TropicalFish.getPatternColorIdx((int)$$0));
     }
 
-    public static java.lang.String getFishTypeName(int variantId) {
-        int n = net.minecraft.world.entity.animal.TropicalFish.getBaseVariant((int)variantId);
-        int n2 = net.minecraft.world.entity.animal.TropicalFish.getPatternVariant((int)variantId);
-        return "entity.minecraft.tropical_fish.type." + net.minecraft.world.entity.animal.TropicalFish$Pattern.getPatternName((int)n, (int)n2);
+    public static java.lang.String getFishTypeName(int $$0) {
+        int $$1 = net.minecraft.world.entity.animal.TropicalFish.getBaseVariant((int)$$0);
+        int $$2 = net.minecraft.world.entity.animal.TropicalFish.getPatternVariant((int)$$0);
+        return "entity.minecraft.tropical_fish.type." + net.minecraft.world.entity.animal.TropicalFish$Pattern.getPatternName((int)$$1, (int)$$2);
     }
 
     public void defineSynchedData() {
@@ -51,21 +51,21 @@
         this.entityData.define(DATA_ID_TYPE_VARIANT, (java.lang.Object)0);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("Variant", this.getVariant());
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("Variant", this.getVariant());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.setVariant(compound.getInt("Variant"));
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.setVariant($$0.getInt("Variant"));
     }
 
-    public void setVariant(int variantId) {
-        this.entityData.set(DATA_ID_TYPE_VARIANT, (java.lang.Object)variantId);
+    public void setVariant(int $$0) {
+        this.entityData.set(DATA_ID_TYPE_VARIANT, (java.lang.Object)$$0);
     }
 
-    public boolean isMaxGroupSizeReached(int size) {
+    public boolean isMaxGroupSizeReached(int $$0) {
         return !this.isSchool;
     }
 
@@ -73,10 +73,10 @@
         return (java.lang.Integer)this.entityData.get(DATA_ID_TYPE_VARIANT);
     }
 
-    public void saveToBucketTag(net.minecraft.world.item.ItemStack bucketStack) {
-        super.saveToBucketTag(bucketStack);
-        net.minecraft.nbt.CompoundTag compoundTag = bucketStack.getOrCreateTag();
-        compoundTag.putInt("BucketVariantTag", this.getVariant());
+    public void saveToBucketTag(net.minecraft.world.item.ItemStack $$0) {
+        super.saveToBucketTag($$0);
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getOrCreateTag();
+        $$1.putInt("BucketVariantTag", this.getVariant());
     }
 
     public net.minecraft.world.item.ItemStack getBucketItemStack() {
@@ -91,7 +91,7 @@
         return net.minecraft.sounds.SoundEvents.TROPICAL_FISH_DEATH;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.TROPICAL_FISH_HURT;
     }
 
@@ -99,32 +99,32 @@
         return net.minecraft.sounds.SoundEvents.TROPICAL_FISH_FLOP;
     }
 
-    public static int getBaseColorIdx(int variantId) {
-        return (variantId & 0xFF0000) >> 16;
+    public static int getBaseColorIdx(int $$0) {
+        return ($$0 & 0xFF0000) >> 16;
     }
 
     public float[] getBaseColor() {
         return net.minecraft.world.item.DyeColor.byId((int)net.minecraft.world.entity.animal.TropicalFish.getBaseColorIdx((int)this.getVariant())).getTextureDiffuseColors();
     }
 
-    public static int getPatternColorIdx(int variantId) {
-        return (variantId & 0xFF000000) >> 24;
+    public static int getPatternColorIdx(int $$0) {
+        return ($$0 & 0xFF000000) >> 24;
     }
 
     public float[] getPatternColor() {
         return net.minecraft.world.item.DyeColor.byId((int)net.minecraft.world.entity.animal.TropicalFish.getPatternColorIdx((int)this.getVariant())).getTextureDiffuseColors();
     }
 
-    public static int getBaseVariant(int variantId) {
-        return java.lang.Math.min((int)(variantId & 0xFF), (int)1);
+    public static int getBaseVariant(int $$0) {
+        return java.lang.Math.min((int)($$0 & 0xFF), (int)1);
     }
 
     public int getBaseVariant() {
         return net.minecraft.world.entity.animal.TropicalFish.getBaseVariant((int)this.getVariant());
     }
 
-    public static int getPatternVariant(int variantId) {
-        return java.lang.Math.min((int)((variantId & 0xFF00) >> 8), (int)5);
+    public static int getPatternVariant(int $$0) {
+        return java.lang.Math.min((int)(($$0 & 0xFF00) >> 8), (int)5);
     }
 
     public net.minecraft.resources.ResourceLocation getPatternTextureLocation() {
@@ -139,39 +139,39 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor level, net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.MobSpawnType reason, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData spawnData, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag dataTag) {
-        int n;
-        int n2;
-        int n3;
-        int n4;
-        spawnData = super.finalizeSpawn(level, difficulty, reason, spawnData, dataTag);
-        if (reason == net.minecraft.world.entity.MobSpawnType.BUCKET && dataTag != null && dataTag.contains("BucketVariantTag", 3)) {
-            this.setVariant(dataTag.getInt("BucketVariantTag"));
-            return spawnData;
+    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.world.DifficultyInstance $$1, net.minecraft.world.entity.MobSpawnType $$2, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData $$3, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$4) {
+        int $$19;
+        int $$18;
+        int $$17;
+        int $$16;
+        $$3 = super.finalizeSpawn($$0, $$1, $$2, $$3, $$4);
+        if ($$2 == net.minecraft.world.entity.MobSpawnType.BUCKET && $$4 != null && $$4.contains("BucketVariantTag", 3)) {
+            this.setVariant($$4.getInt("BucketVariantTag"));
+            return $$3;
         }
-        net.minecraft.util.RandomSource randomSource = level.getRandom();
-        if (spawnData instanceof net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData) {
-            net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData tropicalFishGroupData = (net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData)spawnData;
-            n4 = tropicalFishGroupData.base;
-            n3 = tropicalFishGroupData.pattern;
-            n2 = tropicalFishGroupData.baseColor;
-            n = tropicalFishGroupData.patternColor;
-        } else if ((double)randomSource.nextFloat() < 0.9) {
-            int n5 = net.minecraft.Util.getRandom((int[])COMMON_VARIANTS, (net.minecraft.util.RandomSource)randomSource);
-            n4 = n5 & 0xFF;
-            n3 = (n5 & 0xFF00) >> 8;
-            n2 = (n5 & 0xFF0000) >> 16;
-            n = (n5 & 0xFF000000) >> 24;
-            spawnData = new net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData(this, n4, n3, n2, n);
+        net.minecraft.util.RandomSource $$5 = $$0.getRandom();
+        if ($$3 instanceof net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData) {
+            net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData $$6 = (net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData)$$3;
+            int $$7 = $$6.base;
+            int $$8 = $$6.pattern;
+            int $$9 = $$6.baseColor;
+            int $$10 = $$6.patternColor;
+        } else if ((double)$$5.nextFloat() < 0.9) {
+            int $$11 = net.minecraft.Util.getRandom((int[])COMMON_VARIANTS, (net.minecraft.util.RandomSource)$$5);
+            int $$12 = $$11 & 0xFF;
+            int $$13 = ($$11 & 0xFF00) >> 8;
+            int $$14 = ($$11 & 0xFF0000) >> 16;
+            int $$15 = ($$11 & 0xFF000000) >> 24;
+            $$3 = new net.minecraft.world.entity.animal.TropicalFish$TropicalFishGroupData(this, $$12, $$13, $$14, $$15);
         } else {
             this.isSchool = false;
-            n4 = randomSource.nextInt(2);
-            n3 = randomSource.nextInt(6);
-            n2 = randomSource.nextInt(15);
-            n = randomSource.nextInt(15);
+            $$16 = $$5.nextInt(2);
+            $$17 = $$5.nextInt(6);
+            $$18 = $$5.nextInt(15);
+            $$19 = $$5.nextInt(15);
         }
-        this.setVariant(n4 | n3 << 8 | n2 << 16 | n << 24);
-        return spawnData;
+        this.setVariant($$16 | $$17 << 8 | $$18 << 16 | $$19 << 24);
+        return $$3;
     }
 
     public static boolean checkTropicalFishSpawnRules(net.minecraft.world.entity.EntityType<net.minecraft.world.entity.animal.TropicalFish> $$0, net.minecraft.world.level.LevelAccessor $$1, net.minecraft.world.entity.MobSpawnType $$2, net.minecraft.core.BlockPos $$3, net.minecraft.util.RandomSource $$4) {