--- net/minecraft/world/entity/animal/Turtle$TurtleGoHomeGoal.java
+++ net/minecraft/world/entity/animal/Turtle$TurtleGoHomeGoal.java
@@ -12,9 +12,9 @@
     public int closeToHomeTryTicks;
     public static int GIVE_UP_TICKS = 600;
 
-    public net.minecraft.world.entity.animal.Turtle$TurtleGoHomeGoal(net.minecraft.world.entity.animal.Turtle turtle, double speedModifier) {
-        this.turtle = turtle;
-        this.speedModifier = speedModifier;
+    public net.minecraft.world.entity.animal.Turtle$TurtleGoHomeGoal(net.minecraft.world.entity.animal.Turtle $$0, double $$1) {
+        this.turtle = $$0;
+        this.speedModifier = $$1;
     }
 
     public boolean canUse() {