--- net/minecraft/world/entity/animal/Turtle$TurtleBreedGoal.java
+++ net/minecraft/world/entity/animal/Turtle$TurtleBreedGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Turtle turtle;
 
-    public net.minecraft.world.entity.animal.Turtle$TurtleBreedGoal(net.minecraft.world.entity.animal.Turtle turtle, double speedModifier) {
-        super((net.minecraft.world.entity.animal.Animal)turtle, speedModifier);
-        this.turtle = turtle;
+    public net.minecraft.world.entity.animal.Turtle$TurtleBreedGoal(net.minecraft.world.entity.animal.Turtle $$0, double $$1) {
+        super((net.minecraft.world.entity.animal.Animal)$$0, $$1);
+        this.turtle = $$0;
     }
 
     public boolean canUse() {