--- net/minecraft/world/entity/animal/Turtle$TurtleRandomStrollGoal.java
+++ net/minecraft/world/entity/animal/Turtle$TurtleRandomStrollGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Turtle turtle;
 
-    public net.minecraft.world.entity.animal.Turtle$TurtleRandomStrollGoal(net.minecraft.world.entity.animal.Turtle turtle, double speedModifier, int interval) {
-        super((net.minecraft.world.entity.PathfinderMob)turtle, speedModifier, interval);
-        this.turtle = turtle;
+    public net.minecraft.world.entity.animal.Turtle$TurtleRandomStrollGoal(net.minecraft.world.entity.animal.Turtle $$0, double $$1, int $$2) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, $$2);
+        this.turtle = $$0;
     }
 
     public boolean canUse() {