--- net/minecraft/world/entity/animal/Turtle$TurtleGoToWaterGoal.java
+++ net/minecraft/world/entity/animal/Turtle$TurtleGoToWaterGoal.java
@@ -9,9 +9,9 @@
     public static int GIVE_UP_TICKS = 1200;
     public net.minecraft.world.entity.animal.Turtle turtle;
 
-    public net.minecraft.world.entity.animal.Turtle$TurtleGoToWaterGoal(net.minecraft.world.entity.animal.Turtle turtle, double speedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)turtle, turtle.isBaby() ? 2.0 : speedModifier, 24);
-        this.turtle = turtle;
+    public net.minecraft.world.entity.animal.Turtle$TurtleGoToWaterGoal(net.minecraft.world.entity.animal.Turtle $$0, double $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$0.isBaby() ? 2.0 : $$1, 24);
+        this.turtle = $$0;
         this.verticalSearchStart = -1;
     }
 
@@ -33,8 +33,8 @@
         return this.tryTicks % 160 == 0;
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        return level.getBlockState(pos).is(net.minecraft.world.level.block.Blocks.WATER);
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        return $$0.getBlockState($$1).is(net.minecraft.world.level.block.Blocks.WATER);
     }
 }
 