--- net/minecraft/world/entity/animal/Turtle$TurtlePanicGoal.java
+++ net/minecraft/world/entity/animal/Turtle$TurtlePanicGoal.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.entity.animal.Turtle$TurtlePanicGoal
 extends net.minecraft.world.entity.ai.goal.PanicGoal
  {
-    public net.minecraft.world.entity.animal.Turtle$TurtlePanicGoal(net.minecraft.world.entity.animal.Turtle turtle, double speedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)turtle, speedModifier);
+    public net.minecraft.world.entity.animal.Turtle$TurtlePanicGoal(net.minecraft.world.entity.animal.Turtle $$0, double $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1);
     }
 
     public boolean canUse() {