--- net/minecraft/world/entity/animal/Turtle$TurtleLayEggGoal.java
+++ net/minecraft/world/entity/animal/Turtle$TurtleLayEggGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.Turtle turtle;
 
-    public net.minecraft.world.entity.animal.Turtle$TurtleLayEggGoal(net.minecraft.world.entity.animal.Turtle turtle, double speedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)turtle, speedModifier, 16);
-        this.turtle = turtle;
+    public net.minecraft.world.entity.animal.Turtle$TurtleLayEggGoal(net.minecraft.world.entity.animal.Turtle $$0, double $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, 16);
+        this.turtle = $$0;
     }
 
     public boolean canUse() {
@@ -44,11 +44,11 @@
         }
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        if (!level.isEmptyBlock(pos.above())) {
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        if (!$$0.isEmptyBlock($$1.above())) {
             return false;
         }
-        return net.minecraft.world.level.block.TurtleEggBlock.isSand((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)pos);
+        return net.minecraft.world.level.block.TurtleEggBlock.isSand((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$1);
     }
 }
 