--- net/minecraft/world/entity/animal/Turtle$TurtlePathNavigation.java
+++ net/minecraft/world/entity/animal/Turtle$TurtlePathNavigation.java
@@ -6,17 +6,17 @@
 public class net.minecraft.world.entity.animal.Turtle$TurtlePathNavigation
 extends net.minecraft.world.entity.ai.navigation.AmphibiousPathNavigation
  {
-    public net.minecraft.world.entity.animal.Turtle$TurtlePathNavigation(net.minecraft.world.entity.animal.Turtle turtle, net.minecraft.world.level.Level level) {
-        super((net.minecraft.world.entity.Mob)turtle, level);
+    public net.minecraft.world.entity.animal.Turtle$TurtlePathNavigation(net.minecraft.world.entity.animal.Turtle $$0, net.minecraft.world.level.Level $$1) {
+        super((net.minecraft.world.entity.Mob)$$0, $$1);
     }
 
-    public boolean isStableDestination(net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.entity.animal.Turtle turtle;
+    public boolean isStableDestination(net.minecraft.core.BlockPos $$0) {
+        net.minecraft.world.entity.animal.Turtle $$1;
         net.minecraft.world.entity.Mob mob = this.mob;
-        if (mob instanceof net.minecraft.world.entity.animal.Turtle && (turtle = (net.minecraft.world.entity.animal.Turtle)mob).isTravelling()) {
-            return this.level.getBlockState(pos).is(net.minecraft.world.level.block.Blocks.WATER);
+        if (mob instanceof net.minecraft.world.entity.animal.Turtle && ($$1 = (net.minecraft.world.entity.animal.Turtle)mob).isTravelling()) {
+            return this.level.getBlockState($$0).is(net.minecraft.world.level.block.Blocks.WATER);
         }
-        return !this.level.getBlockState(pos.below()).isAir();
+        return !this.level.getBlockState($$0.below()).isAir();
     }
 }
 