--- net/minecraft/world/entity/animal/Turtle$TurtleTravelGoal.java
+++ net/minecraft/world/entity/animal/Turtle$TurtleTravelGoal.java
@@ -10,9 +10,9 @@
     public double speedModifier;
     public boolean stuck;
 
-    public net.minecraft.world.entity.animal.Turtle$TurtleTravelGoal(net.minecraft.world.entity.animal.Turtle turtle, double speedModifier) {
-        this.turtle = turtle;
-        this.speedModifier = speedModifier;
+    public net.minecraft.world.entity.animal.Turtle$TurtleTravelGoal(net.minecraft.world.entity.animal.Turtle $$0, double $$1) {
+        this.turtle = $$0;
+        this.speedModifier = $$1;
     }
 
     public boolean canUse() {