--- net/minecraft/world/entity/animal/WaterAnimal.java
+++ net/minecraft/world/entity/animal/WaterAnimal.java
@@ -19,8 +19,8 @@
         return net.minecraft.world.entity.MobType.WATER;
     }
 
-    public boolean checkSpawnObstruction(net.minecraft.world.level.LevelReader level) {
-        return level.isUnobstructed((net.minecraft.world.entity.Entity)this);
+    public boolean checkSpawnObstruction(net.minecraft.world.level.LevelReader $$0) {
+        return $$0.isUnobstructed((net.minecraft.world.entity.Entity)this);
     }
 
     public int getAmbientSoundInterval() {
@@ -31,9 +31,9 @@
         return 1 + this.level.random.nextInt(3);
     }
 
-    public void handleAirSupply(int airSupply) {
+    public void handleAirSupply(int $$0) {
         if (this.isAlive() && !this.isInWaterOrBubble()) {
-            this.setAirSupply(airSupply - 1);
+            this.setAirSupply($$0 - 1);
             if (this.getAirSupply() == -20) {
                 this.setAirSupply(0);
                 this.hurt(net.minecraft.world.damagesource.DamageSource.DROWN, 2.0f);
@@ -53,7 +53,7 @@
         return false;
     }
 
-    public boolean canBeLeashed(net.minecraft.world.entity.player.Player player) {
+    public boolean canBeLeashed(net.minecraft.world.entity.player.Player $$0) {
         return false;
     }
 