--- net/minecraft/data/advancements/packs/VanillaAdvancementProvider.java
+++ net/minecraft/data/advancements/packs/VanillaAdvancementProvider.java
@@ -1,0 +1,15 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data.advancements.packs;
+
+public class net.minecraft.data.advancements.packs.VanillaAdvancementProvider
+ {
+    public net.minecraft.data.advancements.packs.VanillaAdvancementProvider() {
+    }
+
+    public static net.minecraft.data.advancements.AdvancementProvider create(net.minecraft.data.PackOutput $$0) {
+        return new net.minecraft.data.advancements.AdvancementProvider("Vanilla Advancements", $$0, java.util.List.of((java.lang.Object)new net.minecraft.data.advancements.packs.VanillaTheEndAdvancements(), (java.lang.Object)new net.minecraft.data.advancements.packs.VanillaHusbandryAdvancements(), (java.lang.Object)new net.minecraft.data.advancements.packs.VanillaAdventureAdvancements(), (java.lang.Object)new net.minecraft.data.advancements.packs.VanillaNetherAdvancements(), (java.lang.Object)new net.minecraft.data.advancements.packs.VanillaStoryAdvancements()));
+    }
+}
+