--- net/minecraft/world/level/storage/loot/BuiltInLootTables.java
+++ net/minecraft/world/level/storage/loot/BuiltInLootTables.java
@@ -90,15 +90,15 @@
     public net.minecraft.world.level.storage.loot.BuiltInLootTables() {
     }
 
-    public static net.minecraft.resources.ResourceLocation register(java.lang.String id) {
-        return net.minecraft.world.level.storage.loot.BuiltInLootTables.register((net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(id));
+    public static net.minecraft.resources.ResourceLocation register(java.lang.String $$0) {
+        return net.minecraft.world.level.storage.loot.BuiltInLootTables.register((net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0));
     }
 
-    public static net.minecraft.resources.ResourceLocation register(net.minecraft.resources.ResourceLocation id) {
-        if (LOCATIONS.add((java.lang.Object)id)) {
-            return id;
+    public static net.minecraft.resources.ResourceLocation register(net.minecraft.resources.ResourceLocation $$0) {
+        if (LOCATIONS.add((java.lang.Object)$$0)) {
+            return $$0;
         }
-        throw new java.lang.IllegalArgumentException(id + " is already a registered built-in loot table");
+        throw new java.lang.IllegalArgumentException($$0 + " is already a registered built-in loot table");
     }
 
     public static java.util.Set<net.minecraft.resources.ResourceLocation> all() {