--- net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
+++ net/minecraft/world/level/storage/loot/GsonAdapterFactory.java
@@ -8,8 +8,8 @@
     public net.minecraft.world.level.storage.loot.GsonAdapterFactory() {
     }
 
-    public static <E, T extends net.minecraft.world.level.storage.loot.SerializerType<E>> net.minecraft.world.level.storage.loot.GsonAdapterFactory$Builder<E, T> builder(net.minecraft.core.Registry<T> registry, java.lang.String id, java.lang.String name, java.util.function.Function<E, T> typeFunction) {
-        return new net.minecraft.world.level.storage.loot.GsonAdapterFactory$Builder(registry, id, name, typeFunction);
+    public static <E, T extends net.minecraft.world.level.storage.loot.SerializerType<E>> net.minecraft.world.level.storage.loot.GsonAdapterFactory$Builder<E, T> builder(net.minecraft.core.Registry<T> $$0, java.lang.String $$1, java.lang.String $$2, java.util.function.Function<E, T> $$3) {
+        return new net.minecraft.world.level.storage.loot.GsonAdapterFactory$Builder($$0, $$1, $$2, $$3);
     }
 }
 