--- net/minecraft/world/level/storage/loot/GsonAdapterFactory$Builder.java
+++ net/minecraft/world/level/storage/loot/GsonAdapterFactory$Builder.java
@@ -14,11 +14,11 @@
     @javax.annotation.Nullable
     public T defaultType;
 
-    public net.minecraft.world.level.storage.loot.GsonAdapterFactory$Builder(net.minecraft.core.Registry<T> registry, java.lang.String elementName, java.lang.String typeKey, java.util.function.Function<E, T> typeGetter) {
-        this.registry = registry;
-        this.elementName = elementName;
-        this.typeKey = typeKey;
-        this.typeGetter = typeGetter;
+    public net.minecraft.world.level.storage.loot.GsonAdapterFactory$Builder(net.minecraft.core.Registry<T> $$0, java.lang.String $$1, java.lang.String $$2, java.util.function.Function<E, T> $$3) {
+        this.registry = $$0;
+        this.elementName = $$1;
+        this.typeKey = $$2;
+        this.typeGetter = $$3;
     }
 
     public net.minecraft.world.level.storage.loot.GsonAdapterFactory$Builder<E, T> withInlineSerializer(T $$0, net.minecraft.world.level.storage.loot.GsonAdapterFactory$InlineSerializer<? extends E> $$1) {