--- net/minecraft/world/level/storage/loot/LootContext$EntityTarget.java
+++ net/minecraft/world/level/storage/loot/LootContext$EntityTarget.java
@@ -22,22 +22,22 @@
         return (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)java.lang.Enum.valueOf(net.minecraft.world.level.storage.loot.LootContext$EntityTarget.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.storage.loot.LootContext$EntityTarget(java.lang.String name, net.minecraft.world.level.storage.loot.parameters.LootContextParam<? extends net.minecraft.world.entity.Entity> param) {
+    public net.minecraft.world.level.storage.loot.LootContext$EntityTarget(java.lang.String $$0, net.minecraft.world.level.storage.loot.parameters.LootContextParam<? extends net.minecraft.world.entity.Entity> $$1) {
         super(string, n);
-        this.name = name;
-        this.param = param;
+        this.name = $$0;
+        this.param = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.parameters.LootContextParam<? extends net.minecraft.world.entity.Entity> getParam() {
         return this.param;
     }
 
-    public static net.minecraft.world.level.storage.loot.LootContext$EntityTarget getByName(java.lang.String name) {
-        for (net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget : net.minecraft.world.level.storage.loot.LootContext$EntityTarget.values()) {
-            if (!entityTarget.name.equals((java.lang.Object)name)) continue;
-            return entityTarget;
+    public static net.minecraft.world.level.storage.loot.LootContext$EntityTarget getByName(java.lang.String $$0) {
+        for (net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$1 : net.minecraft.world.level.storage.loot.LootContext$EntityTarget.values()) {
+            if (!$$1.name.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
-        throw new java.lang.IllegalArgumentException("Invalid entity target " + name);
+        throw new java.lang.IllegalArgumentException("Invalid entity target " + $$0);
     }
 
     public static /* synthetic */ net.minecraft.world.level.storage.loot.LootContext$EntityTarget[] $values() {