--- net/minecraft/world/level/storage/loot/LootTable$Builder.java
+++ net/minecraft/world/level/storage/loot/LootTable$Builder.java
@@ -16,18 +16,18 @@
         this.paramSet = net.minecraft.world.level.storage.loot.LootTable.DEFAULT_PARAM_SET;
     }
 
-    public net.minecraft.world.level.storage.loot.LootTable$Builder withPool(net.minecraft.world.level.storage.loot.LootPool$Builder lootPool) {
-        this.pools.add((java.lang.Object)lootPool.build());
+    public net.minecraft.world.level.storage.loot.LootTable$Builder withPool(net.minecraft.world.level.storage.loot.LootPool$Builder $$0) {
+        this.pools.add((java.lang.Object)$$0.build());
         return this;
     }
 
-    public net.minecraft.world.level.storage.loot.LootTable$Builder setParamSet(net.minecraft.world.level.storage.loot.parameters.LootContextParamSet parameterSet) {
-        this.paramSet = parameterSet;
+    public net.minecraft.world.level.storage.loot.LootTable$Builder setParamSet(net.minecraft.world.level.storage.loot.parameters.LootContextParamSet $$0) {
+        this.paramSet = $$0;
         return this;
     }
 
-    public net.minecraft.world.level.storage.loot.LootTable$Builder apply(net.minecraft.world.level.storage.loot.functions.LootItemFunction$Builder functionBuilder) {
-        this.functions.add((java.lang.Object)functionBuilder.build());
+    public net.minecraft.world.level.storage.loot.LootTable$Builder apply(net.minecraft.world.level.storage.loot.functions.LootItemFunction$Builder $$0) {
+        this.functions.add((java.lang.Object)$$0.build());
         return this;
     }
 