--- net/minecraft/world/level/storage/loot/SerializerType.java
+++ net/minecraft/world/level/storage/loot/SerializerType.java
@@ -7,8 +7,8 @@
  {
     public net.minecraft.world.level.storage.loot.Serializer<? extends T> serializer;
 
-    public net.minecraft.world.level.storage.loot.SerializerType(net.minecraft.world.level.storage.loot.Serializer<? extends T> serializer) {
-        this.serializer = serializer;
+    public net.minecraft.world.level.storage.loot.SerializerType(net.minecraft.world.level.storage.loot.Serializer<? extends T> $$0) {
+        this.serializer = $$0;
     }
 
     public net.minecraft.world.level.storage.loot.Serializer<? extends T> getSerializer() {