--- net/minecraft/server/dedicated/DedicatedPlayerList.java
+++ net/minecraft/server/dedicated/DedicatedPlayerList.java
@@ -8,7 +8,7 @@
  {
     public static org.slf4j.Logger LOGGER;
 
-    public net.minecraft.server.dedicated.DedicatedPlayerList(net.minecraft.server.dedicated.DedicatedServer $$0, net.minecraft.core.RegistryAccess$Frozen $$1, net.minecraft.world.level.storage.PlayerDataStorage $$2) {
+    public net.minecraft.server.dedicated.DedicatedPlayerList(net.minecraft.server.dedicated.DedicatedServer $$0, net.minecraft.core.LayeredRegistryAccess<net.minecraft.server.RegistryLayer> $$1, net.minecraft.world.level.storage.PlayerDataStorage $$2) {
         super((net.minecraft.server.MinecraftServer)$$0, $$1, $$2, $$0.getProperties().maxPlayers);
         net.minecraft.server.dedicated.DedicatedServerProperties $$3 = $$0.getProperties();
         this.setViewDistance($$3.viewDistance);
@@ -26,18 +26,18 @@
         }
     }
 
-    public void setUsingWhiteList(boolean whitelistEnabled) {
-        super.setUsingWhiteList(whitelistEnabled);
-        this.getServer().storeUsingWhiteList(whitelistEnabled);
+    public void setUsingWhiteList(boolean $$0) {
+        super.setUsingWhiteList($$0);
+        this.getServer().storeUsingWhiteList($$0);
     }
 
-    public void op(com.mojang.authlib.GameProfile profile) {
-        super.op(profile);
+    public void op(com.mojang.authlib.GameProfile $$0) {
+        super.op($$0);
         this.saveOps();
     }
 
-    public void deop(com.mojang.authlib.GameProfile profile) {
-        super.deop(profile);
+    public void deop(com.mojang.authlib.GameProfile $$0) {
+        super.deop($$0);
         this.saveOps();
     }
 
@@ -117,16 +117,16 @@
         }
     }
 
-    public boolean isWhiteListed(com.mojang.authlib.GameProfile profile) {
-        return !this.isUsingWhitelist() || this.isOp(profile) || this.getWhiteList().isWhiteListed(profile);
+    public boolean isWhiteListed(com.mojang.authlib.GameProfile $$0) {
+        return !this.isUsingWhitelist() || this.isOp($$0) || this.getWhiteList().isWhiteListed($$0);
     }
 
     public net.minecraft.server.dedicated.DedicatedServer getServer() {
         return (net.minecraft.server.dedicated.DedicatedServer)super.getServer();
     }
 
-    public boolean canBypassPlayerLimit(com.mojang.authlib.GameProfile profile) {
-        return this.getOps().canBypassPlayerLimit(profile);
+    public boolean canBypassPlayerLimit(com.mojang.authlib.GameProfile $$0) {
+        return this.getOps().canBypassPlayerLimit($$0);
     }
 
     public /* synthetic */ net.minecraft.server.MinecraftServer getServer() {