--- net/minecraft/data/info/BiomeParametersDumpReport.java
+++ net/minecraft/data/info/BiomeParametersDumpReport.java
@@ -9,12 +9,12 @@
     public static org.slf4j.Logger LOGGER;
     public java.nio.file.Path topPath;
 
-    public net.minecraft.data.info.BiomeParametersDumpReport(net.minecraft.data.DataGenerator $$0) {
-        this.topPath = $$0.getOutputFolder(net.minecraft.data.DataGenerator$Target.REPORTS).resolve("biome_parameters");
+    public net.minecraft.data.info.BiomeParametersDumpReport(net.minecraft.data.PackOutput $$0) {
+        this.topPath = $$0.getOutputFolder(net.minecraft.data.PackOutput$Target.REPORTS).resolve("biome_parameters");
     }
 
     public void run(net.minecraft.data.CachedOutput $$0) {
-        net.minecraft.core.RegistryAccess$Frozen $$1 = (net.minecraft.core.RegistryAccess$Frozen)net.minecraft.core.RegistryAccess.BUILTIN.get();
+        net.minecraft.core.RegistryAccess$Frozen $$1 = net.minecraft.data.BuiltinRegistries.createAccess();
         net.minecraft.resources.RegistryOps $$2 = net.minecraft.resources.RegistryOps.create((com.mojang.serialization.DynamicOps)com.mojang.serialization.JsonOps.INSTANCE, (net.minecraft.core.RegistryAccess)$$1);
         net.minecraft.core.Registry $$3 = $$1.registryOrThrow(net.minecraft.core.Registry.BIOME_REGISTRY);
         net.minecraft.world.level.biome.MultiNoiseBiomeSource$Preset.getPresets().forEach(arg_0 -> this.lambda$run$0($$3, $$0, (com.mojang.serialization.DynamicOps)$$2, arg_0));