--- net/minecraft/data/info/BlockListReport.java
+++ net/minecraft/data/info/BlockListReport.java
@@ -6,10 +6,10 @@
 public class net.minecraft.data.info.BlockListReport
 implements net.minecraft.data.DataProvider
  {
-    public net.minecraft.data.DataGenerator generator;
+    public net.minecraft.data.PackOutput output;
 
-    public net.minecraft.data.info.BlockListReport(net.minecraft.data.DataGenerator generator) {
-        this.generator = generator;
+    public net.minecraft.data.info.BlockListReport(net.minecraft.data.PackOutput $$0) {
+        this.output = $$0;
     }
 
     public void run(net.minecraft.data.CachedOutput $$0) throws java.io.IOException {
@@ -48,7 +48,7 @@
             $$4.add("states", (com.google.gson.JsonElement)$$10);
             $$1.add($$3.toString(), (com.google.gson.JsonElement)$$4);
         }
-        java.nio.file.Path $$15 = this.generator.getOutputFolder(net.minecraft.data.DataGenerator$Target.REPORTS).resolve("blocks.json");
+        java.nio.file.Path $$15 = this.output.getOutputFolder(net.minecraft.data.PackOutput$Target.REPORTS).resolve("blocks.json");
         net.minecraft.data.DataProvider.saveStable((net.minecraft.data.CachedOutput)$$0, (com.google.gson.JsonElement)$$1, (java.nio.file.Path)$$15);
     }
 