--- net/minecraft/client/tutorial/CraftPlanksTutorialStep.java
+++ net/minecraft/client/tutorial/CraftPlanksTutorialStep.java
@@ -13,8 +13,8 @@
     public net.minecraft.client.gui.components.toasts.TutorialToast toast;
     public int timeWaiting;
 
-    public net.minecraft.client.tutorial.CraftPlanksTutorialStep(net.minecraft.client.tutorial.Tutorial tutorial) {
-        this.tutorial = tutorial;
+    public net.minecraft.client.tutorial.CraftPlanksTutorialStep(net.minecraft.client.tutorial.Tutorial $$0) {
+        this.tutorial = $$0;
     }
 
     public void tick() {
@@ -47,8 +47,8 @@
         }
     }
 
-    public void onGetItem(net.minecraft.world.item.ItemStack stack) {
-        if (stack.is(net.minecraft.tags.ItemTags.PLANKS)) {
+    public void onGetItem(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0.is(net.minecraft.tags.ItemTags.PLANKS)) {
             this.tutorial.setStep(net.minecraft.client.tutorial.TutorialSteps.NONE);
         }
     }