--- net/minecraft/client/tutorial/FindTreeTutorialStepInstance.java
+++ net/minecraft/client/tutorial/FindTreeTutorialStepInstance.java
@@ -13,8 +13,8 @@
     public net.minecraft.client.gui.components.toasts.TutorialToast toast;
     public int timeWaiting;
 
-    public net.minecraft.client.tutorial.FindTreeTutorialStepInstance(net.minecraft.client.tutorial.Tutorial tutorial) {
-        this.tutorial = tutorial;
+    public net.minecraft.client.tutorial.FindTreeTutorialStepInstance(net.minecraft.client.tutorial.Tutorial $$0) {
+        this.tutorial = $$0;
     }
 
     public void tick() {
@@ -41,15 +41,15 @@
         }
     }
 
-    public void onLookAt(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.world.phys.HitResult result) {
-        net.minecraft.world.level.block.state.BlockState blockState;
-        if (result.getType() == net.minecraft.world.phys.HitResult$Type.BLOCK && (blockState = level.getBlockState(((net.minecraft.world.phys.BlockHitResult)result).getBlockPos())).is(net.minecraft.tags.BlockTags.COMPLETES_FIND_TREE_TUTORIAL)) {
+    public void onLookAt(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.world.phys.HitResult $$1) {
+        net.minecraft.world.level.block.state.BlockState $$2;
+        if ($$1.getType() == net.minecraft.world.phys.HitResult$Type.BLOCK && ($$2 = $$0.getBlockState(((net.minecraft.world.phys.BlockHitResult)$$1).getBlockPos())).is(net.minecraft.tags.BlockTags.COMPLETES_FIND_TREE_TUTORIAL)) {
             this.tutorial.setStep(net.minecraft.client.tutorial.TutorialSteps.PUNCH_TREE);
         }
     }
 
-    public void onGetItem(net.minecraft.world.item.ItemStack stack) {
-        if (stack.is(net.minecraft.tags.ItemTags.COMPLETES_FIND_TREE_TUTORIAL)) {
+    public void onGetItem(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0.is(net.minecraft.tags.ItemTags.COMPLETES_FIND_TREE_TUTORIAL)) {
             this.tutorial.setStep(net.minecraft.client.tutorial.TutorialSteps.CRAFT_PLANKS);
         }
     }
@@ -58,10 +58,10 @@
         return $$02.getInventory().hasAnyMatching($$0 -> $$0.is(net.minecraft.tags.ItemTags.COMPLETES_FIND_TREE_TUTORIAL));
     }
 
-    public static boolean hasPunchedTreesPreviously(net.minecraft.client.player.LocalPlayer player) {
-        for (net.minecraft.core.Holder holder : net.minecraft.core.Registry.BLOCK.getTagOrEmpty(net.minecraft.tags.BlockTags.COMPLETES_FIND_TREE_TUTORIAL)) {
-            net.minecraft.world.level.block.Block block = (net.minecraft.world.level.block.Block)holder.value();
-            if (player.getStats().getValue(net.minecraft.stats.Stats.BLOCK_MINED.get((java.lang.Object)block)) <= 0) continue;
+    public static boolean hasPunchedTreesPreviously(net.minecraft.client.player.LocalPlayer $$0) {
+        for (net.minecraft.core.Holder $$1 : net.minecraft.core.Registry.BLOCK.getTagOrEmpty(net.minecraft.tags.BlockTags.COMPLETES_FIND_TREE_TUTORIAL)) {
+            net.minecraft.world.level.block.Block $$2 = (net.minecraft.world.level.block.Block)$$1.value();
+            if ($$0.getStats().getValue(net.minecraft.stats.Stats.BLOCK_MINED.get((java.lang.Object)$$2)) <= 0) continue;
             return true;
         }
         return false;