--- net/minecraft/client/tutorial/MovementTutorialStepInstance.java
+++ net/minecraft/client/tutorial/MovementTutorialStepInstance.java
@@ -26,10 +26,10 @@
     public int moveCompleted;
     public int lookCompleted;
 
-    public net.minecraft.client.tutorial.MovementTutorialStepInstance(net.minecraft.client.tutorial.Tutorial tutorial) {
+    public net.minecraft.client.tutorial.MovementTutorialStepInstance(net.minecraft.client.tutorial.Tutorial $$0) {
         this.moveCompleted = -1;
         this.lookCompleted = -1;
-        this.tutorial = tutorial;
+        this.tutorial = $$0;
     }
 
     public void tick() {
@@ -91,14 +91,14 @@
         }
     }
 
-    public void onInput(net.minecraft.client.player.Input input) {
-        if (input.up || input.down || input.left || input.right || input.jumping) {
+    public void onInput(net.minecraft.client.player.Input $$0) {
+        if ($$0.up || $$0.down || $$0.left || $$0.right || $$0.jumping) {
             this.moved = true;
         }
     }
 
-    public void onMouse(double velocityX, double velocityY) {
-        if (java.lang.Math.abs((double)velocityX) > 0.01 || java.lang.Math.abs((double)velocityY) > 0.01) {
+    public void onMouse(double $$0, double $$1) {
+        if (java.lang.Math.abs((double)$$0) > 0.01 || java.lang.Math.abs((double)$$1) > 0.01) {
             this.turned = true;
         }
     }