--- net/minecraft/client/tutorial/OpenInventoryTutorialStep.java
+++ net/minecraft/client/tutorial/OpenInventoryTutorialStep.java
@@ -13,8 +13,8 @@
     public net.minecraft.client.gui.components.toasts.TutorialToast toast;
     public int timeWaiting;
 
-    public net.minecraft.client.tutorial.OpenInventoryTutorialStep(net.minecraft.client.tutorial.Tutorial tutorial) {
-        this.tutorial = tutorial;
+    public net.minecraft.client.tutorial.OpenInventoryTutorialStep(net.minecraft.client.tutorial.Tutorial $$0) {
+        this.tutorial = $$0;
     }
 
     public void tick() {