--- net/minecraft/client/tutorial/PunchTreeTutorialStepInstance.java
+++ net/minecraft/client/tutorial/PunchTreeTutorialStepInstance.java
@@ -14,8 +14,8 @@
     public int timeWaiting;
     public int resetCount;
 
-    public net.minecraft.client.tutorial.PunchTreeTutorialStepInstance(net.minecraft.client.tutorial.Tutorial tutorial) {
-        this.tutorial = tutorial;
+    public net.minecraft.client.tutorial.PunchTreeTutorialStepInstance(net.minecraft.client.tutorial.Tutorial $$0) {
+        this.tutorial = $$0;
     }
 
     public void tick() {
@@ -48,24 +48,24 @@
         }
     }
 
-    public void onDestroyBlock(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, float diggingStage) {
-        boolean bl = state.is(net.minecraft.tags.BlockTags.LOGS);
-        if (bl && diggingStage > 0.0f) {
+    public void onDestroyBlock(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, float $$3) {
+        boolean $$4 = $$2.is(net.minecraft.tags.BlockTags.LOGS);
+        if ($$4 && $$3 > 0.0f) {
             if (this.toast != null) {
-                this.toast.updateProgress(diggingStage);
+                this.toast.updateProgress($$3);
             }
-            if (diggingStage >= 1.0f) {
+            if ($$3 >= 1.0f) {
                 this.tutorial.setStep(net.minecraft.client.tutorial.TutorialSteps.OPEN_INVENTORY);
             }
         } else if (this.toast != null) {
             this.toast.updateProgress(0.0f);
-        } else if (bl) {
+        } else if ($$4) {
             ++this.resetCount;
         }
     }
 
-    public void onGetItem(net.minecraft.world.item.ItemStack stack) {
-        if (stack.is(net.minecraft.tags.ItemTags.LOGS)) {
+    public void onGetItem(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0.is(net.minecraft.tags.ItemTags.LOGS)) {
             this.tutorial.setStep(net.minecraft.client.tutorial.TutorialSteps.CRAFT_PLANKS);
             return;
         }