--- net/minecraft/client/tutorial/TutorialStepInstance.java
+++ net/minecraft/client/tutorial/TutorialStepInstance.java
@@ -11,22 +11,22 @@
     default public void tick() {
     }
 
-    default public void onInput(net.minecraft.client.player.Input input) {
+    default public void onInput(net.minecraft.client.player.Input $$0) {
     }
 
-    default public void onMouse(double velocityX, double velocityY) {
+    default public void onMouse(double $$0, double $$1) {
     }
 
-    default public void onLookAt(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.world.phys.HitResult result) {
+    default public void onLookAt(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.world.phys.HitResult $$1) {
     }
 
-    default public void onDestroyBlock(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, float diggingStage) {
+    default public void onDestroyBlock(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, float $$3) {
     }
 
     default public void onOpenInventory() {
     }
 
-    default public void onGetItem(net.minecraft.world.item.ItemStack stack) {
+    default public void onGetItem(net.minecraft.world.item.ItemStack $$0) {
     }
 }
 