--- net/minecraft/client/tutorial/Tutorial$TimedToast.java
+++ net/minecraft/client/tutorial/Tutorial$TimedToast.java
@@ -9,9 +9,9 @@
     public int durationTicks;
     public int progress;
 
-    public net.minecraft.client.tutorial.Tutorial$TimedToast(net.minecraft.client.gui.components.toasts.TutorialToast toast, int durationTicks) {
-        this.toast = toast;
-        this.durationTicks = durationTicks;
+    public net.minecraft.client.tutorial.Tutorial$TimedToast(net.minecraft.client.gui.components.toasts.TutorialToast $$0, int $$1) {
+        this.toast = $$0;
+        this.durationTicks = $$1;
     }
 
     public boolean updateProgress() {