--- net/minecraft/client/tutorial/TutorialSteps.java
+++ net/minecraft/client/tutorial/TutorialSteps.java
@@ -24,24 +24,24 @@
         return (net.minecraft.client.tutorial.TutorialSteps)java.lang.Enum.valueOf(net.minecraft.client.tutorial.TutorialSteps.class, (java.lang.String)$$0);
     }
 
-    public <T extends net.minecraft.client.tutorial.TutorialStepInstance> net.minecraft.client.tutorial.TutorialSteps(java.lang.String name, java.util.function.Function<net.minecraft.client.tutorial.Tutorial, T> constructor) {
+    public <T extends net.minecraft.client.tutorial.TutorialStepInstance> net.minecraft.client.tutorial.TutorialSteps(java.lang.String $$0, java.util.function.Function<net.minecraft.client.tutorial.Tutorial, T> $$1) {
         super(string, n);
-        this.name = name;
-        this.constructor = constructor;
+        this.name = $$0;
+        this.constructor = $$1;
     }
 
-    public net.minecraft.client.tutorial.TutorialStepInstance create(net.minecraft.client.tutorial.Tutorial tutorial) {
-        return (net.minecraft.client.tutorial.TutorialStepInstance)this.constructor.apply((java.lang.Object)tutorial);
+    public net.minecraft.client.tutorial.TutorialStepInstance create(net.minecraft.client.tutorial.Tutorial $$0) {
+        return (net.minecraft.client.tutorial.TutorialStepInstance)this.constructor.apply((java.lang.Object)$$0);
     }
 
     public java.lang.String getName() {
         return this.name;
     }
 
-    public static net.minecraft.client.tutorial.TutorialSteps getByName(java.lang.String name) {
-        for (net.minecraft.client.tutorial.TutorialSteps tutorialSteps : net.minecraft.client.tutorial.TutorialSteps.values()) {
-            if (!tutorialSteps.name.equals((java.lang.Object)name)) continue;
-            return tutorialSteps;
+    public static net.minecraft.client.tutorial.TutorialSteps getByName(java.lang.String $$0) {
+        for (net.minecraft.client.tutorial.TutorialSteps $$1 : net.minecraft.client.tutorial.TutorialSteps.values()) {
+            if (!$$1.name.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
         return NONE;
     }