--- net/minecraft/client/tutorial/Tutorial.java
+++ net/minecraft/client/tutorial/Tutorial.java
@@ -17,27 +17,27 @@
         this.bundleTutorial = new net.minecraft.client.tutorial.BundleTutorial(this, $$1);
     }
 
-    public void onInput(net.minecraft.client.player.Input input) {
+    public void onInput(net.minecraft.client.player.Input $$0) {
         if (this.instance != null) {
-            this.instance.onInput(input);
+            this.instance.onInput($$0);
         }
     }
 
-    public void onMouse(double velocityX, double velocityY) {
+    public void onMouse(double $$0, double $$1) {
         if (this.instance != null) {
-            this.instance.onMouse(velocityX, velocityY);
+            this.instance.onMouse($$0, $$1);
         }
     }
 
-    public void onLookAt(@javax.annotation.Nullable net.minecraft.client.multiplayer.ClientLevel level, @javax.annotation.Nullable net.minecraft.world.phys.HitResult result) {
-        if (this.instance != null && result != null && level != null) {
-            this.instance.onLookAt(level, result);
+    public void onLookAt(@javax.annotation.Nullable net.minecraft.client.multiplayer.ClientLevel $$0, @javax.annotation.Nullable net.minecraft.world.phys.HitResult $$1) {
+        if (this.instance != null && $$1 != null && $$0 != null) {
+            this.instance.onLookAt($$0, $$1);
         }
     }
 
-    public void onDestroyBlock(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, float diggingStage) {
+    public void onDestroyBlock(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, float $$3) {
         if (this.instance != null) {
-            this.instance.onDestroyBlock(level, pos, state, diggingStage);
+            this.instance.onDestroyBlock($$0, $$1, $$2, $$3);
         }
     }
 
@@ -47,9 +47,9 @@
         }
     }
 
-    public void onGetItem(net.minecraft.world.item.ItemStack stack) {
+    public void onGetItem(net.minecraft.world.item.ItemStack $$0) {
         if (this.instance != null) {
-            this.instance.onGetItem(stack);
+            this.instance.onGetItem($$0);
         }
     }
 
@@ -68,14 +68,14 @@
         this.instance = this.minecraft.options.tutorialStep.create(this);
     }
 
-    public void addTimedToast(net.minecraft.client.gui.components.toasts.TutorialToast toast, int durationTicks) {
-        this.timedToasts.add((java.lang.Object)new net.minecraft.client.tutorial.Tutorial$TimedToast(toast, durationTicks));
-        this.minecraft.getToasts().addToast((net.minecraft.client.gui.components.toasts.Toast)toast);
+    public void addTimedToast(net.minecraft.client.gui.components.toasts.TutorialToast $$0, int $$1) {
+        this.timedToasts.add((java.lang.Object)new net.minecraft.client.tutorial.Tutorial$TimedToast($$0, $$1));
+        this.minecraft.getToasts().addToast((net.minecraft.client.gui.components.toasts.Toast)$$0);
     }
 
-    public void removeTimedToast(net.minecraft.client.gui.components.toasts.TutorialToast toast) {
-        this.timedToasts.removeIf($$1 -> $$1.toast == toast);
-        toast.hide();
+    public void removeTimedToast(net.minecraft.client.gui.components.toasts.TutorialToast $$0) {
+        this.timedToasts.removeIf($$1 -> $$1.toast == $$0);
+        $$0.hide();
     }
 
     public void tick() {
@@ -91,12 +91,12 @@
         }
     }
 
-    public void setStep(net.minecraft.client.tutorial.TutorialSteps step) {
-        this.minecraft.options.tutorialStep = step;
+    public void setStep(net.minecraft.client.tutorial.TutorialSteps $$0) {
+        this.minecraft.options.tutorialStep = $$0;
         this.minecraft.options.save();
         if (this.instance != null) {
             this.instance.clear();
-            this.instance = step.create(this);
+            this.instance = $$0.create(this);
         }
     }
 
@@ -111,8 +111,8 @@
         return this.minecraft.gameMode.getPlayerMode() == net.minecraft.world.level.GameType.SURVIVAL;
     }
 
-    public static net.minecraft.network.chat.Component key(java.lang.String keybind) {
-        return net.minecraft.network.chat.Component.keybind((java.lang.String)("key." + keybind)).withStyle(net.minecraft.ChatFormatting.BOLD);
+    public static net.minecraft.network.chat.Component key(java.lang.String $$0) {
+        return net.minecraft.network.chat.Component.keybind((java.lang.String)("key." + $$0)).withStyle(net.minecraft.ChatFormatting.BOLD);
     }
 
     public void onInventoryAction(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.world.inventory.ClickAction $$2) {