--- net/minecraft/world/item/ArmorItem$1.java
+++ net/minecraft/world/item/ArmorItem$1.java
@@ -9,8 +9,8 @@
     public net.minecraft.world.item.ArmorItem$1() {
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.item.ArmorItem.dispenseArmor((net.minecraft.core.BlockSource)source, (net.minecraft.world.item.ItemStack)stack) ? stack : super.execute(source, stack);
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        return net.minecraft.world.item.ArmorItem.dispenseArmor((net.minecraft.core.BlockSource)$$0, (net.minecraft.world.item.ItemStack)$$1) ? $$1 : super.execute($$0, $$1);
     }
 }
 