--- net/minecraft/world/item/ArmorMaterials.java
+++ net/minecraft/world/item/ArmorMaterials.java
@@ -33,24 +33,24 @@
         return (net.minecraft.world.item.ArmorMaterials)java.lang.Enum.valueOf(net.minecraft.world.item.ArmorMaterials.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.item.ArmorMaterials(java.lang.String name, int durabilityMultiplier, int[] slotProtections, int enchantmentValue, net.minecraft.sounds.SoundEvent sound, float toughness, float knockbackResistance, java.util.function.Supplier<net.minecraft.world.item.crafting.Ingredient> repairIngredient) {
+    public net.minecraft.world.item.ArmorMaterials(java.lang.String $$0, int $$1, int[] $$2, int $$3, net.minecraft.sounds.SoundEvent $$4, float $$5, float $$6, java.util.function.Supplier<net.minecraft.world.item.crafting.Ingredient> $$7) {
         super(string, n);
-        this.name = name;
-        this.durabilityMultiplier = durabilityMultiplier;
-        this.slotProtections = slotProtections;
-        this.enchantmentValue = enchantmentValue;
-        this.sound = sound;
-        this.toughness = toughness;
-        this.knockbackResistance = knockbackResistance;
-        this.repairIngredient = new net.minecraft.util.LazyLoadedValue(repairIngredient);
+        this.name = $$0;
+        this.durabilityMultiplier = $$1;
+        this.slotProtections = $$2;
+        this.enchantmentValue = $$3;
+        this.sound = $$4;
+        this.toughness = $$5;
+        this.knockbackResistance = $$6;
+        this.repairIngredient = new net.minecraft.util.LazyLoadedValue($$7);
     }
 
-    public int getDurabilityForSlot(net.minecraft.world.entity.EquipmentSlot slot) {
-        return HEALTH_PER_SLOT[slot.getIndex()] * this.durabilityMultiplier;
+    public int getDurabilityForSlot(net.minecraft.world.entity.EquipmentSlot $$0) {
+        return HEALTH_PER_SLOT[$$0.getIndex()] * this.durabilityMultiplier;
     }
 
-    public int getDefenseForSlot(net.minecraft.world.entity.EquipmentSlot slot) {
-        return this.slotProtections[slot.getIndex()];
+    public int getDefenseForSlot(net.minecraft.world.entity.EquipmentSlot $$0) {
+        return this.slotProtections[$$0.getIndex()];
     }
 
     public int getEnchantmentValue() {