--- net/minecraft/world/item/ArmorStandItem.java
+++ net/minecraft/world/item/ArmorStandItem.java
@@ -10,35 +10,35 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.core.Direction direction = context.getClickedFace();
-        if (direction == net.minecraft.core.Direction.DOWN) {
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.core.Direction $$1 = $$0.getClickedFace();
+        if ($$1 == net.minecraft.core.Direction.DOWN) {
             return net.minecraft.world.InteractionResult.FAIL;
         }
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.item.context.BlockPlaceContext blockPlaceContext = new net.minecraft.world.item.context.BlockPlaceContext(context);
-        net.minecraft.core.BlockPos blockPos = blockPlaceContext.getClickedPos();
-        net.minecraft.world.item.ItemStack itemStack = context.getItemInHand();
-        net.minecraft.world.phys.Vec3 vec3 = net.minecraft.world.phys.Vec3.atBottomCenterOf((net.minecraft.core.Vec3i)blockPos);
-        net.minecraft.world.phys.AABB aABB = net.minecraft.world.entity.EntityType.ARMOR_STAND.getDimensions().makeBoundingBox(vec3.x(), vec3.y(), vec3.z());
-        if (!level.noCollision(null, aABB) || !level.getEntities(null, aABB).isEmpty()) {
+        net.minecraft.world.level.Level $$2 = $$0.getLevel();
+        net.minecraft.world.item.context.BlockPlaceContext $$3 = new net.minecraft.world.item.context.BlockPlaceContext($$0);
+        net.minecraft.core.BlockPos $$4 = $$3.getClickedPos();
+        net.minecraft.world.item.ItemStack $$5 = $$0.getItemInHand();
+        net.minecraft.world.phys.Vec3 $$6 = net.minecraft.world.phys.Vec3.atBottomCenterOf((net.minecraft.core.Vec3i)$$4);
+        net.minecraft.world.phys.AABB $$7 = net.minecraft.world.entity.EntityType.ARMOR_STAND.getDimensions().makeBoundingBox($$6.x(), $$6.y(), $$6.z());
+        if (!$$2.noCollision(null, $$7) || !$$2.getEntities(null, $$7).isEmpty()) {
             return net.minecraft.world.InteractionResult.FAIL;
         }
-        if (level instanceof net.minecraft.server.level.ServerLevel) {
-            net.minecraft.server.level.ServerLevel serverLevel = (net.minecraft.server.level.ServerLevel)level;
-            net.minecraft.world.entity.decoration.ArmorStand armorStand = (net.minecraft.world.entity.decoration.ArmorStand)net.minecraft.world.entity.EntityType.ARMOR_STAND.create(serverLevel, itemStack.getTag(), null, context.getPlayer(), blockPos, net.minecraft.world.entity.MobSpawnType.SPAWN_EGG, true, true);
-            if (armorStand == null) {
+        if ($$2 instanceof net.minecraft.server.level.ServerLevel) {
+            net.minecraft.server.level.ServerLevel $$8 = (net.minecraft.server.level.ServerLevel)$$2;
+            net.minecraft.world.entity.decoration.ArmorStand $$9 = (net.minecraft.world.entity.decoration.ArmorStand)net.minecraft.world.entity.EntityType.ARMOR_STAND.create($$8, $$5.getTag(), null, $$0.getPlayer(), $$4, net.minecraft.world.entity.MobSpawnType.SPAWN_EGG, true, true);
+            if ($$9 == null) {
                 return net.minecraft.world.InteractionResult.FAIL;
             }
-            float f = (float)net.minecraft.util.Mth.floor((float)((net.minecraft.util.Mth.wrapDegrees((float)(context.getRotation() - 180.0f)) + 22.5f) / 45.0f)) * 45.0f;
-            armorStand.moveTo(armorStand.getX(), armorStand.getY(), armorStand.getZ(), f, 0.0f);
-            this.randomizePose(armorStand, level.random);
-            serverLevel.addFreshEntityWithPassengers((net.minecraft.world.entity.Entity)armorStand);
-            level.playSound(null, armorStand.getX(), armorStand.getY(), armorStand.getZ(), net.minecraft.sounds.SoundEvents.ARMOR_STAND_PLACE, net.minecraft.sounds.SoundSource.BLOCKS, 0.75f, 0.8f);
-            armorStand.gameEvent(net.minecraft.world.level.gameevent.GameEvent.ENTITY_PLACE, (net.minecraft.world.entity.Entity)context.getPlayer());
+            float $$10 = (float)net.minecraft.util.Mth.floor((float)((net.minecraft.util.Mth.wrapDegrees((float)($$0.getRotation() - 180.0f)) + 22.5f) / 45.0f)) * 45.0f;
+            $$9.moveTo($$9.getX(), $$9.getY(), $$9.getZ(), $$10, 0.0f);
+            this.randomizePose($$9, $$2.random);
+            $$8.addFreshEntityWithPassengers((net.minecraft.world.entity.Entity)$$9);
+            $$2.playSound(null, $$9.getX(), $$9.getY(), $$9.getZ(), net.minecraft.sounds.SoundEvents.ARMOR_STAND_PLACE, net.minecraft.sounds.SoundSource.BLOCKS, 0.75f, 0.8f);
+            $$9.gameEvent(net.minecraft.world.level.gameevent.GameEvent.ENTITY_PLACE, (net.minecraft.world.entity.Entity)$$0.getPlayer());
         }
-        itemStack.shrink(1);
-        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+        $$5.shrink(1);
+        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$2.isClientSide);
     }
 
     public void randomizePose(net.minecraft.world.entity.decoration.ArmorStand $$0, net.minecraft.util.RandomSource $$1) {