--- net/minecraft/world/item/ArrowItem.java
+++ net/minecraft/world/item/ArrowItem.java
@@ -10,10 +10,10 @@
         super($$0);
     }
 
-    public net.minecraft.world.entity.projectile.AbstractArrow createArrow(net.minecraft.world.level.Level level, net.minecraft.world.item.ItemStack stack, net.minecraft.world.entity.LivingEntity shooter) {
-        net.minecraft.world.entity.projectile.Arrow arrow = new net.minecraft.world.entity.projectile.Arrow(level, shooter);
-        arrow.setEffectsFromItem(stack);
-        return arrow;
+    public net.minecraft.world.entity.projectile.AbstractArrow createArrow(net.minecraft.world.level.Level $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.world.entity.LivingEntity $$2) {
+        net.minecraft.world.entity.projectile.Arrow $$3 = new net.minecraft.world.entity.projectile.Arrow($$0, $$2);
+        $$3.setEffectsFromItem($$1);
+        return $$3;
     }
 }
 