--- net/minecraft/world/item/BannerItem.java
+++ net/minecraft/world/item/BannerItem.java
@@ -9,23 +9,23 @@
     public static java.lang.String PATTERN_PREFIX = "block.minecraft.banner.";
 
     public net.minecraft.world.item.BannerItem(net.minecraft.world.level.block.Block $$0, net.minecraft.world.level.block.Block $$1, net.minecraft.world.item.Item$Properties $$2) {
-        super($$0, $$1, $$2);
+        super($$0, $$1, $$2, net.minecraft.core.Direction.DOWN);
         org.apache.commons.lang3.Validate.isInstanceOf(net.minecraft.world.level.block.AbstractBannerBlock.class, (java.lang.Object)$$0);
         org.apache.commons.lang3.Validate.isInstanceOf(net.minecraft.world.level.block.AbstractBannerBlock.class, (java.lang.Object)$$1);
     }
 
-    public static void appendHoverTextFromBannerBlockEntityTag(net.minecraft.world.item.ItemStack stack, java.util.List<net.minecraft.network.chat.Component> tooltipComponents) {
-        net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack);
-        if (compoundTag == null || !compoundTag.contains("Patterns")) {
+    public static void appendHoverTextFromBannerBlockEntityTag(net.minecraft.world.item.ItemStack $$02, java.util.List<net.minecraft.network.chat.Component> $$1) {
+        net.minecraft.nbt.CompoundTag $$22 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$02);
+        if ($$22 == null || !$$22.contains("Patterns")) {
             return;
         }
-        net.minecraft.nbt.ListTag listTag = compoundTag.getList("Patterns", 10);
-        for (int i = 0; i < listTag.size() && i < 6; ++i) {
-            net.minecraft.nbt.CompoundTag compoundTag2 = listTag.getCompound(i);
-            net.minecraft.world.item.DyeColor dyeColor = net.minecraft.world.item.DyeColor.byId((int)compoundTag2.getInt("Color"));
-            net.minecraft.core.Holder holder = net.minecraft.world.level.block.entity.BannerPattern.byHash((java.lang.String)compoundTag2.getString("Pattern"));
-            if (holder == null) continue;
-            holder.unwrapKey().map($$0 -> $$0.location().toShortLanguageKey()).ifPresent($$2 -> tooltipComponents.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("block.minecraft.banner." + $$2 + "." + dyeColor.getName())).withStyle(net.minecraft.ChatFormatting.GRAY)));
+        net.minecraft.nbt.ListTag $$3 = $$22.getList("Patterns", 10);
+        for (int $$4 = 0; $$4 < $$3.size() && $$4 < 6; ++$$4) {
+            net.minecraft.nbt.CompoundTag $$5 = $$3.getCompound($$4);
+            net.minecraft.world.item.DyeColor $$6 = net.minecraft.world.item.DyeColor.byId((int)$$5.getInt("Color"));
+            net.minecraft.core.Holder $$7 = net.minecraft.world.level.block.entity.BannerPattern.byHash((java.lang.String)$$5.getString("Pattern"));
+            if ($$7 == null) continue;
+            $$7.unwrapKey().map($$0 -> $$0.location().toShortLanguageKey()).ifPresent($$2 -> $$1.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("block.minecraft.banner." + $$2 + "." + $$6.getName())).withStyle(net.minecraft.ChatFormatting.GRAY)));
         }
     }
 
@@ -33,8 +33,8 @@
         return ((net.minecraft.world.level.block.AbstractBannerBlock)this.getBlock()).getColor();
     }
 
-    public void appendHoverText(net.minecraft.world.item.ItemStack stack, @javax.annotation.Nullable net.minecraft.world.level.Level level, java.util.List<net.minecraft.network.chat.Component> tooltip, net.minecraft.world.item.TooltipFlag flag) {
-        net.minecraft.world.item.BannerItem.appendHoverTextFromBannerBlockEntityTag((net.minecraft.world.item.ItemStack)stack, tooltip);
+    public void appendHoverText(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.world.level.Level $$1, java.util.List<net.minecraft.network.chat.Component> $$2, net.minecraft.world.item.TooltipFlag $$3) {
+        net.minecraft.world.item.BannerItem.appendHoverTextFromBannerBlockEntityTag((net.minecraft.world.item.ItemStack)$$0, $$2);
     }
 }
 