--- net/minecraft/world/item/BannerPatternItem.java
+++ net/minecraft/world/item/BannerPatternItem.java
@@ -17,8 +17,8 @@
         return this.bannerPattern;
     }
 
-    public void appendHoverText(net.minecraft.world.item.ItemStack stack, @javax.annotation.Nullable net.minecraft.world.level.Level level, java.util.List<net.minecraft.network.chat.Component> tooltip, net.minecraft.world.item.TooltipFlag flag) {
-        tooltip.add((java.lang.Object)this.getDisplayName().withStyle(net.minecraft.ChatFormatting.GRAY));
+    public void appendHoverText(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.world.level.Level $$1, java.util.List<net.minecraft.network.chat.Component> $$2, net.minecraft.world.item.TooltipFlag $$3) {
+        $$2.add((java.lang.Object)this.getDisplayName().withStyle(net.minecraft.ChatFormatting.GRAY));
     }
 
     public net.minecraft.network.chat.MutableComponent getDisplayName() {