--- net/minecraft/world/item/BookItem.java
+++ net/minecraft/world/item/BookItem.java
@@ -10,8 +10,8 @@
         super($$0);
     }
 
-    public boolean isEnchantable(net.minecraft.world.item.ItemStack stack) {
-        return stack.getCount() == 1;
+    public boolean isEnchantable(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.getCount() == 1;
     }
 
     public int getEnchantmentValue() {