--- net/minecraft/world/item/BowItem.java
+++ net/minecraft/world/item/BowItem.java
@@ -14,85 +14,85 @@
         super($$0);
     }
 
-    public void releaseUsing(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity entityLiving, int timeLeft) {
-        boolean bl;
-        int n;
-        float f;
-        if (!(entityLiving instanceof net.minecraft.world.entity.player.Player)) {
+    public void releaseUsing(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.Level $$12, net.minecraft.world.entity.LivingEntity $$2, int $$3) {
+        boolean $$9;
+        int $$7;
+        float $$8;
+        if (!($$2 instanceof net.minecraft.world.entity.player.Player)) {
             return;
         }
-        net.minecraft.world.entity.player.Player player = (net.minecraft.world.entity.player.Player)entityLiving;
-        boolean bl2 = player.getAbilities().instabuild || net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.INFINITY_ARROWS, (net.minecraft.world.item.ItemStack)stack) > 0;
-        net.minecraft.world.item.ItemStack itemStack = player.getProjectile(stack);
-        if (itemStack.isEmpty() && !bl2) {
+        net.minecraft.world.entity.player.Player $$4 = (net.minecraft.world.entity.player.Player)$$2;
+        boolean $$5 = $$4.getAbilities().instabuild || net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.INFINITY_ARROWS, (net.minecraft.world.item.ItemStack)$$0) > 0;
+        net.minecraft.world.item.ItemStack $$6 = $$4.getProjectile($$0);
+        if ($$6.isEmpty() && !$$5) {
             return;
         }
-        if (itemStack.isEmpty()) {
-            itemStack = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ARROW);
+        if ($$6.isEmpty()) {
+            $$6 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ARROW);
         }
-        if ((double)(f = net.minecraft.world.item.BowItem.getPowerForTime((int)(n = this.getUseDuration(stack) - timeLeft))) < 0.1) {
+        if ((double)($$8 = net.minecraft.world.item.BowItem.getPowerForTime((int)($$7 = this.getUseDuration($$0) - $$3))) < 0.1) {
             return;
         }
-        boolean bl3 = bl = bl2 && itemStack.is(net.minecraft.world.item.Items.ARROW);
-        if (!level.isClientSide) {
-            int n2;
-            int n3;
-            net.minecraft.world.item.ArrowItem arrowItem = (net.minecraft.world.item.ArrowItem)(itemStack.getItem() instanceof net.minecraft.world.item.ArrowItem ? itemStack.getItem() : net.minecraft.world.item.Items.ARROW);
-            net.minecraft.world.entity.projectile.AbstractArrow abstractArrow = arrowItem.createArrow(level, itemStack, (net.minecraft.world.entity.LivingEntity)player);
-            abstractArrow.shootFromRotation((net.minecraft.world.entity.Entity)player, player.getXRot(), player.getYRot(), 0.0f, f * 3.0f, 1.0f);
-            if (f == 1.0f) {
-                abstractArrow.setCritArrow(true);
+        boolean bl = $$9 = $$5 && $$6.is(net.minecraft.world.item.Items.ARROW);
+        if (!$$12.isClientSide) {
+            int $$13;
+            int $$122;
+            net.minecraft.world.item.ArrowItem $$10 = (net.minecraft.world.item.ArrowItem)($$6.getItem() instanceof net.minecraft.world.item.ArrowItem ? $$6.getItem() : net.minecraft.world.item.Items.ARROW);
+            net.minecraft.world.entity.projectile.AbstractArrow $$11 = $$10.createArrow($$12, $$6, (net.minecraft.world.entity.LivingEntity)$$4);
+            $$11.shootFromRotation((net.minecraft.world.entity.Entity)$$4, $$4.getXRot(), $$4.getYRot(), 0.0f, $$8 * 3.0f, 1.0f);
+            if ($$8 == 1.0f) {
+                $$11.setCritArrow(true);
             }
-            if ((n3 = net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.POWER_ARROWS, (net.minecraft.world.item.ItemStack)stack)) > 0) {
-                abstractArrow.setBaseDamage(abstractArrow.getBaseDamage() + (double)n3 * 0.5 + 0.5);
+            if (($$122 = net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.POWER_ARROWS, (net.minecraft.world.item.ItemStack)$$0)) > 0) {
+                $$11.setBaseDamage($$11.getBaseDamage() + (double)$$122 * 0.5 + 0.5);
             }
-            if ((n2 = net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.PUNCH_ARROWS, (net.minecraft.world.item.ItemStack)stack)) > 0) {
-                abstractArrow.setKnockback(n2);
+            if (($$13 = net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.PUNCH_ARROWS, (net.minecraft.world.item.ItemStack)$$0)) > 0) {
+                $$11.setKnockback($$13);
             }
-            if (net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.FLAMING_ARROWS, (net.minecraft.world.item.ItemStack)stack) > 0) {
-                abstractArrow.setSecondsOnFire(100);
+            if (net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.FLAMING_ARROWS, (net.minecraft.world.item.ItemStack)$$0) > 0) {
+                $$11.setSecondsOnFire(100);
             }
-            stack.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)player, contextPlayer -> contextPlayer.broadcastBreakEvent(player.getUsedItemHand()));
-            if (bl || player.getAbilities().instabuild && (itemStack.is(net.minecraft.world.item.Items.SPECTRAL_ARROW) || itemStack.is(net.minecraft.world.item.Items.TIPPED_ARROW))) {
-                abstractArrow.pickup = net.minecraft.world.entity.projectile.AbstractArrow$Pickup.CREATIVE_ONLY;
+            $$0.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)$$4, $$1 -> $$1.broadcastBreakEvent($$4.getUsedItemHand()));
+            if ($$9 || $$4.getAbilities().instabuild && ($$6.is(net.minecraft.world.item.Items.SPECTRAL_ARROW) || $$6.is(net.minecraft.world.item.Items.TIPPED_ARROW))) {
+                $$11.pickup = net.minecraft.world.entity.projectile.AbstractArrow$Pickup.CREATIVE_ONLY;
             }
-            level.addFreshEntity((net.minecraft.world.entity.Entity)abstractArrow);
+            $$12.addFreshEntity((net.minecraft.world.entity.Entity)$$11);
         }
-        level.playSound(null, player.getX(), player.getY(), player.getZ(), net.minecraft.sounds.SoundEvents.ARROW_SHOOT, net.minecraft.sounds.SoundSource.PLAYERS, 1.0f, 1.0f / (level.getRandom().nextFloat() * 0.4f + 1.2f) + f * 0.5f);
-        if (!bl && !player.getAbilities().instabuild) {
-            itemStack.shrink(1);
-            if (itemStack.isEmpty()) {
-                player.getInventory().removeItem(itemStack);
+        $$12.playSound(null, $$4.getX(), $$4.getY(), $$4.getZ(), net.minecraft.sounds.SoundEvents.ARROW_SHOOT, net.minecraft.sounds.SoundSource.PLAYERS, 1.0f, 1.0f / ($$12.getRandom().nextFloat() * 0.4f + 1.2f) + $$8 * 0.5f);
+        if (!$$9 && !$$4.getAbilities().instabuild) {
+            $$6.shrink(1);
+            if ($$6.isEmpty()) {
+                $$4.getInventory().removeItem($$6);
             }
         }
-        player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+        $$4.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
     }
 
-    public static float getPowerForTime(int charge) {
-        float f = (float)charge / 20.0f;
-        if ((f = (f * f + f * 2.0f) / 3.0f) > 1.0f) {
-            f = 1.0f;
+    public static float getPowerForTime(int $$0) {
+        float $$1 = (float)$$0 / 20.0f;
+        if (($$1 = ($$1 * $$1 + $$1 * 2.0f) / 3.0f) > 1.0f) {
+            $$1 = 1.0f;
         }
-        return f;
+        return $$1;
     }
 
-    public int getUseDuration(net.minecraft.world.item.ItemStack stack) {
+    public int getUseDuration(net.minecraft.world.item.ItemStack $$0) {
         return 72000;
     }
 
-    public net.minecraft.world.item.UseAnim getUseAnimation(net.minecraft.world.item.ItemStack stack) {
+    public net.minecraft.world.item.UseAnim getUseAnimation(net.minecraft.world.item.ItemStack $$0) {
         return net.minecraft.world.item.UseAnim.BOW;
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        boolean bl;
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        boolean bl2 = bl = !player.getProjectile(itemStack).isEmpty();
-        if (player.getAbilities().instabuild || bl) {
-            player.startUsingItem(hand);
-            return net.minecraft.world.InteractionResultHolder.consume((java.lang.Object)itemStack);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        boolean $$4;
+        net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+        boolean bl = $$4 = !$$1.getProjectile($$3).isEmpty();
+        if ($$1.getAbilities().instabuild || $$4) {
+            $$1.startUsingItem($$2);
+            return net.minecraft.world.InteractionResultHolder.consume((java.lang.Object)$$3);
         }
-        return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)itemStack);
+        return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)$$3);
     }
 
     public java.util.function.Predicate<net.minecraft.world.item.ItemStack> getAllSupportedProjectiles() {