--- net/minecraft/world/item/BucketItem.java
+++ net/minecraft/world/item/BucketItem.java
@@ -9,61 +9,61 @@
  {
     public net.minecraft.world.level.material.Fluid content;
 
-    public net.minecraft.world.item.BucketItem(net.minecraft.world.level.material.Fluid content, net.minecraft.world.item.Item$Properties properties) {
-        super(properties);
-        this.content = content;
+    public net.minecraft.world.item.BucketItem(net.minecraft.world.level.material.Fluid $$0, net.minecraft.world.item.Item$Properties $$1) {
+        super($$1);
+        this.content = $$0;
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        net.minecraft.world.phys.BlockHitResult blockHitResult = net.minecraft.world.item.BucketItem.getPlayerPOVHitResult((net.minecraft.world.level.Level)level, (net.minecraft.world.entity.player.Player)player, (net.minecraft.world.level.ClipContext$Fluid)(this.content == net.minecraft.world.level.material.Fluids.EMPTY ? net.minecraft.world.level.ClipContext$Fluid.SOURCE_ONLY : net.minecraft.world.level.ClipContext$Fluid.NONE));
-        if (blockHitResult.getType() == net.minecraft.world.phys.HitResult$Type.MISS) {
-            return net.minecraft.world.InteractionResultHolder.pass((java.lang.Object)itemStack);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$12, net.minecraft.world.InteractionHand $$2) {
+        net.minecraft.world.item.ItemStack $$3 = $$12.getItemInHand($$2);
+        net.minecraft.world.phys.BlockHitResult $$4 = net.minecraft.world.item.BucketItem.getPlayerPOVHitResult((net.minecraft.world.level.Level)$$0, (net.minecraft.world.entity.player.Player)$$12, (net.minecraft.world.level.ClipContext$Fluid)(this.content == net.minecraft.world.level.material.Fluids.EMPTY ? net.minecraft.world.level.ClipContext$Fluid.SOURCE_ONLY : net.minecraft.world.level.ClipContext$Fluid.NONE));
+        if ($$4.getType() == net.minecraft.world.phys.HitResult$Type.MISS) {
+            return net.minecraft.world.InteractionResultHolder.pass((java.lang.Object)$$3);
         }
-        if (blockHitResult.getType() == net.minecraft.world.phys.HitResult$Type.BLOCK) {
-            net.minecraft.core.BlockPos blockPos;
-            net.minecraft.core.BlockPos blockPos2 = blockHitResult.getBlockPos();
-            net.minecraft.core.Direction direction = blockHitResult.getDirection();
-            net.minecraft.core.BlockPos blockPos3 = blockPos2.relative(direction);
-            if (!level.mayInteract(player, blockPos2) || !player.mayUseItemAt(blockPos3, direction, itemStack)) {
-                return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)itemStack);
+        if ($$4.getType() == net.minecraft.world.phys.HitResult$Type.BLOCK) {
+            net.minecraft.core.BlockPos $$13;
+            net.minecraft.core.BlockPos $$5 = $$4.getBlockPos();
+            net.minecraft.core.Direction $$6 = $$4.getDirection();
+            net.minecraft.core.BlockPos $$7 = $$5.relative($$6);
+            if (!$$0.mayInteract($$12, $$5) || !$$12.mayUseItemAt($$7, $$6, $$3)) {
+                return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)$$3);
             }
             if (this.content == net.minecraft.world.level.material.Fluids.EMPTY) {
-                net.minecraft.world.level.block.BucketPickup bucketPickup;
-                net.minecraft.world.item.ItemStack itemStack2;
-                net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos2);
-                if (blockState.getBlock() instanceof net.minecraft.world.level.block.BucketPickup && !(itemStack2 = (bucketPickup = (net.minecraft.world.level.block.BucketPickup)blockState.getBlock()).pickupBlock((net.minecraft.world.level.LevelAccessor)level, blockPos2, blockState)).isEmpty()) {
-                    player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
-                    bucketPickup.getPickupSound().ifPresent($$1 -> player.playSound($$1, 1.0f, 1.0f));
-                    level.gameEvent((net.minecraft.world.entity.Entity)player, net.minecraft.world.level.gameevent.GameEvent.FLUID_PICKUP, blockPos2);
-                    net.minecraft.world.item.ItemStack itemStack3 = net.minecraft.world.item.ItemUtils.createFilledResult((net.minecraft.world.item.ItemStack)itemStack, (net.minecraft.world.entity.player.Player)player, (net.minecraft.world.item.ItemStack)itemStack2);
-                    if (!level.isClientSide) {
-                        net.minecraft.advancements.CriteriaTriggers.FILLED_BUCKET.trigger((net.minecraft.server.level.ServerPlayer)player, itemStack2);
+                net.minecraft.world.level.block.BucketPickup $$9;
+                net.minecraft.world.item.ItemStack $$10;
+                net.minecraft.world.level.block.state.BlockState $$8 = $$0.getBlockState($$5);
+                if ($$8.getBlock() instanceof net.minecraft.world.level.block.BucketPickup && !($$10 = ($$9 = (net.minecraft.world.level.block.BucketPickup)$$8.getBlock()).pickupBlock((net.minecraft.world.level.LevelAccessor)$$0, $$5, $$8)).isEmpty()) {
+                    $$12.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+                    $$9.getPickupSound().ifPresent($$1 -> $$12.playSound($$1, 1.0f, 1.0f));
+                    $$0.gameEvent((net.minecraft.world.entity.Entity)$$12, net.minecraft.world.level.gameevent.GameEvent.FLUID_PICKUP, $$5);
+                    net.minecraft.world.item.ItemStack $$11 = net.minecraft.world.item.ItemUtils.createFilledResult((net.minecraft.world.item.ItemStack)$$3, (net.minecraft.world.entity.player.Player)$$12, (net.minecraft.world.item.ItemStack)$$10);
+                    if (!$$0.isClientSide) {
+                        net.minecraft.advancements.CriteriaTriggers.FILLED_BUCKET.trigger((net.minecraft.server.level.ServerPlayer)$$12, $$10);
                     }
-                    return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)itemStack3, (boolean)level.isClientSide());
+                    return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)$$11, (boolean)$$0.isClientSide());
                 }
-                return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)itemStack);
+                return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)$$3);
             }
-            net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos2);
-            net.minecraft.core.BlockPos blockPos4 = blockPos = blockState.getBlock() instanceof net.minecraft.world.level.block.LiquidBlockContainer && this.content == net.minecraft.world.level.material.Fluids.WATER ? blockPos2 : blockPos3;
-            if (this.emptyContents(player, level, blockPos, blockHitResult)) {
-                this.checkExtraContent(player, level, itemStack, blockPos);
-                if (player instanceof net.minecraft.server.level.ServerPlayer) {
-                    net.minecraft.advancements.CriteriaTriggers.PLACED_BLOCK.trigger((net.minecraft.server.level.ServerPlayer)player, blockPos, itemStack);
+            net.minecraft.world.level.block.state.BlockState $$122 = $$0.getBlockState($$5);
+            net.minecraft.core.BlockPos blockPos = $$13 = $$122.getBlock() instanceof net.minecraft.world.level.block.LiquidBlockContainer && this.content == net.minecraft.world.level.material.Fluids.WATER ? $$5 : $$7;
+            if (this.emptyContents($$12, $$0, $$13, $$4)) {
+                this.checkExtraContent($$12, $$0, $$3, $$13);
+                if ($$12 instanceof net.minecraft.server.level.ServerPlayer) {
+                    net.minecraft.advancements.CriteriaTriggers.PLACED_BLOCK.trigger((net.minecraft.server.level.ServerPlayer)$$12, $$13, $$3);
                 }
-                player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
-                return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)net.minecraft.world.item.BucketItem.getEmptySuccessItem((net.minecraft.world.item.ItemStack)itemStack, (net.minecraft.world.entity.player.Player)player), (boolean)level.isClientSide());
+                $$12.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+                return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)net.minecraft.world.item.BucketItem.getEmptySuccessItem((net.minecraft.world.item.ItemStack)$$3, (net.minecraft.world.entity.player.Player)$$12), (boolean)$$0.isClientSide());
             }
-            return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)itemStack);
+            return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)$$3);
         }
-        return net.minecraft.world.InteractionResultHolder.pass((java.lang.Object)itemStack);
+        return net.minecraft.world.InteractionResultHolder.pass((java.lang.Object)$$3);
     }
 
-    public static net.minecraft.world.item.ItemStack getEmptySuccessItem(net.minecraft.world.item.ItemStack itemStack, net.minecraft.world.entity.player.Player stack) {
-        if (!stack.getAbilities().instabuild) {
+    public static net.minecraft.world.item.ItemStack getEmptySuccessItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.entity.player.Player $$1) {
+        if (!$$1.getAbilities().instabuild) {
             return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BUCKET);
         }
-        return itemStack;
+        return $$0;
     }
 
     public void checkExtraContent(@javax.annotation.Nullable net.minecraft.world.entity.player.Player $$0, net.minecraft.world.level.Level $$1, net.minecraft.world.item.ItemStack $$2, net.minecraft.core.BlockPos $$3) {
@@ -107,10 +107,10 @@
         return false;
     }
 
-    public void playEmptySound(@javax.annotation.Nullable net.minecraft.world.entity.player.Player player, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.sounds.SoundEvent soundEvent = this.content.is(net.minecraft.tags.FluidTags.LAVA) ? net.minecraft.sounds.SoundEvents.BUCKET_EMPTY_LAVA : net.minecraft.sounds.SoundEvents.BUCKET_EMPTY;
-        level.playSound(player, pos, soundEvent, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
-        level.gameEvent((net.minecraft.world.entity.Entity)player, net.minecraft.world.level.gameevent.GameEvent.FLUID_PLACE, pos);
+    public void playEmptySound(@javax.annotation.Nullable net.minecraft.world.entity.player.Player $$0, net.minecraft.world.level.LevelAccessor $$1, net.minecraft.core.BlockPos $$2) {
+        net.minecraft.sounds.SoundEvent $$3 = this.content.is(net.minecraft.tags.FluidTags.LAVA) ? net.minecraft.sounds.SoundEvents.BUCKET_EMPTY_LAVA : net.minecraft.sounds.SoundEvents.BUCKET_EMPTY;
+        $$1.playSound($$0, $$2, $$3, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
+        $$1.gameEvent((net.minecraft.world.entity.Entity)$$0, net.minecraft.world.level.gameevent.GameEvent.FLUID_PLACE, $$2);
     }
 }
 