--- net/minecraft/world/item/ChorusFruitItem.java
+++ net/minecraft/world/item/ChorusFruitItem.java
@@ -10,32 +10,32 @@
         super($$0);
     }
 
-    public net.minecraft.world.item.ItemStack finishUsingItem(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity entityLiving) {
-        net.minecraft.world.item.ItemStack itemStack = super.finishUsingItem(stack, level, entityLiving);
-        if (!level.isClientSide) {
-            double d = entityLiving.getX();
-            double d2 = entityLiving.getY();
-            double d3 = entityLiving.getZ();
-            for (int i = 0; i < 16; ++i) {
-                double d4 = entityLiving.getX() + (entityLiving.getRandom().nextDouble() - 0.5) * 16.0;
-                double d5 = net.minecraft.util.Mth.clamp((double)(entityLiving.getY() + (double)(entityLiving.getRandom().nextInt(16) - 8)), (double)level.getMinBuildHeight(), (double)(level.getMinBuildHeight() + ((net.minecraft.server.level.ServerLevel)level).getLogicalHeight() - 1));
-                double d6 = entityLiving.getZ() + (entityLiving.getRandom().nextDouble() - 0.5) * 16.0;
-                if (entityLiving.isPassenger()) {
-                    entityLiving.stopRiding();
+    public net.minecraft.world.item.ItemStack finishUsingItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.Level $$1, net.minecraft.world.entity.LivingEntity $$2) {
+        net.minecraft.world.item.ItemStack $$3 = super.finishUsingItem($$0, $$1, $$2);
+        if (!$$1.isClientSide) {
+            double $$4 = $$2.getX();
+            double $$5 = $$2.getY();
+            double $$6 = $$2.getZ();
+            for (int $$7 = 0; $$7 < 16; ++$$7) {
+                double $$8 = $$2.getX() + ($$2.getRandom().nextDouble() - 0.5) * 16.0;
+                double $$9 = net.minecraft.util.Mth.clamp((double)($$2.getY() + (double)($$2.getRandom().nextInt(16) - 8)), (double)$$1.getMinBuildHeight(), (double)($$1.getMinBuildHeight() + ((net.minecraft.server.level.ServerLevel)$$1).getLogicalHeight() - 1));
+                double $$10 = $$2.getZ() + ($$2.getRandom().nextDouble() - 0.5) * 16.0;
+                if ($$2.isPassenger()) {
+                    $$2.stopRiding();
                 }
-                net.minecraft.world.phys.Vec3 vec3 = entityLiving.position();
-                if (!entityLiving.randomTeleport(d4, d5, d6, true)) continue;
-                level.gameEvent(net.minecraft.world.level.gameevent.GameEvent.TELEPORT, vec3, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)entityLiving));
-                net.minecraft.sounds.SoundEvent soundEvent = entityLiving instanceof net.minecraft.world.entity.animal.Fox ? net.minecraft.sounds.SoundEvents.FOX_TELEPORT : net.minecraft.sounds.SoundEvents.CHORUS_FRUIT_TELEPORT;
-                level.playSound(null, d, d2, d3, soundEvent, net.minecraft.sounds.SoundSource.PLAYERS, 1.0f, 1.0f);
-                entityLiving.playSound(soundEvent, 1.0f, 1.0f);
+                net.minecraft.world.phys.Vec3 $$11 = $$2.position();
+                if (!$$2.randomTeleport($$8, $$9, $$10, true)) continue;
+                $$1.gameEvent(net.minecraft.world.level.gameevent.GameEvent.TELEPORT, $$11, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)$$2));
+                net.minecraft.sounds.SoundEvent $$12 = $$2 instanceof net.minecraft.world.entity.animal.Fox ? net.minecraft.sounds.SoundEvents.FOX_TELEPORT : net.minecraft.sounds.SoundEvents.CHORUS_FRUIT_TELEPORT;
+                $$1.playSound(null, $$4, $$5, $$6, $$12, net.minecraft.sounds.SoundSource.PLAYERS, 1.0f, 1.0f);
+                $$2.playSound($$12, 1.0f, 1.0f);
                 break;
             }
-            if (entityLiving instanceof net.minecraft.world.entity.player.Player) {
-                ((net.minecraft.world.entity.player.Player)entityLiving).getCooldowns().addCooldown((net.minecraft.world.item.Item)this, 20);
+            if ($$2 instanceof net.minecraft.world.entity.player.Player) {
+                ((net.minecraft.world.entity.player.Player)$$2).getCooldowns().addCooldown((net.minecraft.world.item.Item)this, 20);
             }
         }
-        return itemStack;
+        return $$3;
     }
 }
 