--- net/minecraft/world/item/CreativeModeTabs$9.java
+++ net/minecraft/world/item/CreativeModeTabs$9.java
@@ -1,0 +1,26 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.CreativeModeTabs$9
+extends net.minecraft.world.item.CreativeModeTab
+ {
+    public net.minecraft.world.item.CreativeModeTabs$9(int $$0, net.minecraft.network.chat.Component $$1) {
+        super($$0, $$1);
+    }
+
+    public net.minecraft.world.item.ItemStack makeIcon() {
+        return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COMPASS);
+    }
+
+    public void generateDisplayItems(net.minecraft.world.flag.FeatureFlagSet $$0, net.minecraft.world.item.CreativeModeTab$Output $$1) {
+        net.minecraft.world.item.ItemStackLinkedSet $$2 = new net.minecraft.world.item.ItemStackLinkedSet();
+        for (net.minecraft.world.item.CreativeModeTab $$3 : net.minecraft.world.item.CreativeModeTabs.TABS) {
+            if ($$3 == this) continue;
+            $$2.addAll((java.util.Collection)$$3.getSearchTabDisplayItems($$0));
+        }
+        $$1.acceptAll((java.util.Collection)$$2);
+    }
+}
+