--- net/minecraft/world/item/CreativeModeTabs$7.java
+++ net/minecraft/world/item/CreativeModeTabs$7.java
@@ -1,0 +1,111 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.CreativeModeTabs$7
+extends net.minecraft.world.item.CreativeModeTab
+ {
+    public net.minecraft.world.item.CreativeModeTabs$7(int $$0, net.minecraft.network.chat.Component $$1) {
+        super($$0, $$1);
+    }
+
+    public net.minecraft.world.item.ItemStack makeIcon() {
+        return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.REDSTONE);
+    }
+
+    public void generateDisplayItems(net.minecraft.world.flag.FeatureFlagSet $$0, net.minecraft.world.item.CreativeModeTab$Output $$1) {
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.REDSTONE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.REDSTONE_TORCH);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.REDSTONE_BLOCK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.REPEATER);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COMPARATOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PISTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STICKY_PISTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SLIME_BLOCK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HONEY_BLOCK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OBSERVER);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HOPPER);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DISPENSER);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DROPPER);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.LECTERN);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TARGET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.LEVER);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.LIGHTNING_ROD);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DAYLIGHT_DETECTOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SCULK_SENSOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TRIPWIRE_HOOK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TRAPPED_CHEST);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TNT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.REDSTONE_LAMP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NOTE_BLOCK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POWERED_RAIL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DETECTOR_RAIL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.RAIL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACTIVATOR_RAIL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CHEST_MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FURNACE_MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TNT_MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HOPPER_MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STONE_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POLISHED_BLACKSTONE_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OAK_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPRUCE_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BIRCH_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.JUNGLE_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACACIA_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DARK_OAK_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MANGROVE_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CRIMSON_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WARPED_BUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STONE_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POLISHED_BLACKSTONE_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.LIGHT_WEIGHTED_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HEAVY_WEIGHTED_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OAK_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPRUCE_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BIRCH_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.JUNGLE_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACACIA_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DARK_OAK_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MANGROVE_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CRIMSON_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WARPED_PRESSURE_PLATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OAK_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPRUCE_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BIRCH_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.JUNGLE_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACACIA_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DARK_OAK_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MANGROVE_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CRIMSON_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WARPED_DOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OAK_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPRUCE_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BIRCH_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.JUNGLE_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACACIA_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DARK_OAK_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MANGROVE_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CRIMSON_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WARPED_TRAPDOOR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OAK_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPRUCE_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BIRCH_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.JUNGLE_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACACIA_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DARK_OAK_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MANGROVE_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CRIMSON_FENCE_GATE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WARPED_FENCE_GATE);
+    }
+}
+