--- net/minecraft/world/item/CreativeModeTabs$8.java
+++ net/minecraft/world/item/CreativeModeTabs$8.java
@@ -1,0 +1,24 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.CreativeModeTabs$8
+extends net.minecraft.world.item.CreativeModeTab
+ {
+    public net.minecraft.world.item.CreativeModeTabs$8(int $$0, net.minecraft.network.chat.Component $$1) {
+        super($$0, $$1);
+    }
+
+    public net.minecraft.world.item.ItemStack makeIcon() {
+        return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.level.block.Blocks.BOOKSHELF);
+    }
+
+    public void generateDisplayItems(net.minecraft.world.flag.FeatureFlagSet $$0, net.minecraft.world.item.CreativeModeTab$Output $$1) {
+    }
+
+    public boolean isAlignedRight() {
+        return true;
+    }
+}
+