--- net/minecraft/world/item/CreativeModeTabs$10.java
+++ net/minecraft/world/item/CreativeModeTabs$10.java
@@ -1,0 +1,115 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.CreativeModeTabs$10
+extends net.minecraft.world.item.CreativeModeTab
+ {
+    public net.minecraft.world.item.CreativeModeTabs$10(int $$0, net.minecraft.network.chat.Component $$1) {
+        super($$0, $$1);
+    }
+
+    public net.minecraft.world.item.ItemStack makeIcon() {
+        return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_AXE);
+    }
+
+    public void generateDisplayItems(net.minecraft.world.flag.FeatureFlagSet $$0, net.minecraft.world.item.CreativeModeTab$Output $$1) {
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WOODEN_SHOVEL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WOODEN_PICKAXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WOODEN_AXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WOODEN_HOE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STONE_SHOVEL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STONE_PICKAXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STONE_AXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STONE_HOE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLDEN_SHOVEL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLDEN_PICKAXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLDEN_AXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLDEN_HOE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_SHOVEL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_PICKAXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_AXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_HOE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DIAMOND_SHOVEL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DIAMOND_PICKAXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DIAMOND_AXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DIAMOND_HOE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NETHERITE_SHOVEL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NETHERITE_PICKAXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NETHERITE_AXE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NETHERITE_HOE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WATER_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PUFFERFISH_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SALMON_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COD_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TROPICAL_FISH_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.AXOLOTL_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TADPOLE_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.LAVA_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POWDER_SNOW_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MILK_BUCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FISHING_ROD);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FLINT_AND_STEEL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FIRE_CHARGE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SHEARS);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.NAME_TAG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.LEAD);
+        if ($$0.contains(net.minecraft.world.flag.FeatureFlags.BUNDLE)) {
+            $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BUNDLE);
+        }
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COMPASS);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.RECOVERY_COMPASS);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CLOCK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPYGLASS);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MAP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WRITABLE_BOOK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ENDER_PEARL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ENDER_EYE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ELYTRA);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FIREWORK_ROCKET);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SADDLE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CARROT_ON_A_STICK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WARPED_FUNGUS_ON_A_STICK);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OAK_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OAK_CHEST_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPRUCE_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPRUCE_CHEST_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BIRCH_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BIRCH_CHEST_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.JUNGLE_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.JUNGLE_CHEST_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACACIA_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ACACIA_CHEST_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DARK_OAK_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DARK_OAK_CHEST_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MANGROVE_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MANGROVE_CHEST_BOAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO_RAFT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO_CHEST_RAFT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CHEST_MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FURNACE_MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TNT_MINECART);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HOPPER_MINECART);
+        net.minecraft.world.item.CreativeModeTabs.generateInstrumentTypes((net.minecraft.world.item.CreativeModeTab$Output)$$1, (net.minecraft.world.item.Item)net.minecraft.world.item.Items.GOAT_HORN, (net.minecraft.tags.TagKey)net.minecraft.tags.InstrumentTags.GOAT_HORNS, (net.minecraft.world.item.CreativeModeTab$TabVisibility)net.minecraft.world.item.CreativeModeTab$TabVisibility.PARENT_AND_SEARCH_TABS);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_13);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_CAT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_BLOCKS);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_CHIRP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_FAR);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_MALL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_MELLOHI);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_STAL);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_STRAD);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_WARD);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_11);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_WAIT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_OTHERSIDE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_5);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSIC_DISC_PIGSTEP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.EXPERIENCE_BOTTLE);
+    }
+}
+