--- net/minecraft/world/item/CreativeModeTabs$12.java
+++ net/minecraft/world/item/CreativeModeTabs$12.java
@@ -1,0 +1,63 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.CreativeModeTabs$12
+extends net.minecraft.world.item.CreativeModeTab
+ {
+    public net.minecraft.world.item.CreativeModeTabs$12(int $$0, net.minecraft.network.chat.Component $$1) {
+        super($$0, $$1);
+    }
+
+    public net.minecraft.world.item.ItemStack makeIcon() {
+        return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CARROT);
+    }
+
+    public void generateDisplayItems(net.minecraft.world.flag.FeatureFlagSet $$0, net.minecraft.world.item.CreativeModeTab$Output $$1) {
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.APPLE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUSHROOM_STEW);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BREAD);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PORKCHOP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKED_PORKCHOP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLDEN_APPLE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ENCHANTED_GOLDEN_APPLE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COD);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SALMON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TROPICAL_FISH);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PUFFERFISH);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKED_COD);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKED_SALMON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CAKE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKIE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MELON_SLICE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DRIED_KELP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BEEF);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKED_BEEF);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CHICKEN);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKED_CHICKEN);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CARROT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOLDEN_CARROT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POTATO);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAKED_POTATO);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PUMPKIN_PIE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.RABBIT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKED_RABBIT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.RABBIT_STEW);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COOKED_MUTTON);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BEETROOT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BEETROOT_SOUP);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SWEET_BERRIES);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GLOW_BERRIES);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CHORUS_FRUIT);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ROTTEN_FLESH);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POISONOUS_POTATO);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPIDER_EYE);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HONEY_BOTTLE);
+        net.minecraft.world.item.CreativeModeTabs.generatePotionEffectTypes((net.minecraft.world.item.CreativeModeTab$Output)$$1, (net.minecraft.world.item.Item)net.minecraft.world.item.Items.POTION, (net.minecraft.world.item.CreativeModeTab$TabVisibility)net.minecraft.world.item.CreativeModeTab$TabVisibility.PARENT_AND_SEARCH_TABS);
+        net.minecraft.world.item.CreativeModeTabs.generatePotionEffectTypes((net.minecraft.world.item.CreativeModeTab$Output)$$1, (net.minecraft.world.item.Item)net.minecraft.world.item.Items.SPLASH_POTION, (net.minecraft.world.item.CreativeModeTab$TabVisibility)net.minecraft.world.item.CreativeModeTab$TabVisibility.PARENT_AND_SEARCH_TABS);
+        net.minecraft.world.item.CreativeModeTabs.generatePotionEffectTypes((net.minecraft.world.item.CreativeModeTab$Output)$$1, (net.minecraft.world.item.Item)net.minecraft.world.item.Items.LINGERING_POTION, (net.minecraft.world.item.CreativeModeTab$TabVisibility)net.minecraft.world.item.CreativeModeTab$TabVisibility.PARENT_AND_SEARCH_TABS);
+    }
+}
+