--- net/minecraft/world/item/CreativeModeTab$ItemDisplayBuilder.java
+++ net/minecraft/world/item/CreativeModeTab$ItemDisplayBuilder.java
@@ -1,0 +1,53 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.CreativeModeTab$ItemDisplayBuilder
+implements net.minecraft.world.item.CreativeModeTab$Output
+ {
+    public net.minecraft.world.item.ItemStackLinkedSet tabContents;
+    public net.minecraft.world.item.ItemStackLinkedSet searchTabContents;
+    public net.minecraft.world.item.CreativeModeTab tab;
+    public net.minecraft.world.flag.FeatureFlagSet featureFlagSet;
+
+    public net.minecraft.world.item.CreativeModeTab$ItemDisplayBuilder(net.minecraft.world.item.CreativeModeTab $$0, net.minecraft.world.flag.FeatureFlagSet $$1) {
+        this.tabContents = new net.minecraft.world.item.ItemStackLinkedSet();
+        this.searchTabContents = new net.minecraft.world.item.ItemStackLinkedSet();
+        this.tab = $$0;
+        this.featureFlagSet = $$1;
+    }
+
+    public void accept(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.CreativeModeTab$TabVisibility $$1) {
+        boolean $$2;
+        boolean bl = $$2 = this.tabContents.contains((java.lang.Object)$$0) && $$1 != net.minecraft.world.item.CreativeModeTab$TabVisibility.SEARCH_TAB_ONLY;
+        if ($$2) {
+            throw new java.lang.IllegalStateException("Accidentally adding the same item stack twice " + $$0.getDisplayName().getString() + " to a Creative Mode Tab: " + this.tab.getDisplayName().getString());
+        }
+        if ($$0.getItem().isEnabled(this.featureFlagSet)) {
+            switch (net.minecraft.world.item.CreativeModeTab$1.$SwitchMap$net$minecraft$world$item$CreativeModeTab$TabVisibility[$$1.ordinal()]) {
+                case 1: {
+                    this.tabContents.add((java.lang.Object)$$0);
+                    this.searchTabContents.add((java.lang.Object)$$0);
+                    break;
+                }
+                case 2: {
+                    this.tabContents.add((java.lang.Object)$$0);
+                    break;
+                }
+                case 3: {
+                    this.searchTabContents.add((java.lang.Object)$$0);
+                }
+            }
+        }
+    }
+
+    public net.minecraft.world.item.ItemStackLinkedSet getTabContents() {
+        return this.tabContents;
+    }
+
+    public net.minecraft.world.item.ItemStackLinkedSet getSearchTabContents() {
+        return this.searchTabContents;
+    }
+}
+