--- net/minecraft/world/item/CreativeModeTabs$3.java
+++ net/minecraft/world/item/CreativeModeTabs$3.java
@@ -1,0 +1,92 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.CreativeModeTabs$3
+extends net.minecraft.world.item.CreativeModeTab
+ {
+    public net.minecraft.world.item.CreativeModeTabs$3(int $$0, net.minecraft.network.chat.Component $$1) {
+        super($$0, $$1);
+    }
+
+    public net.minecraft.world.item.ItemStack makeIcon() {
+        return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SHULKER_SPAWN_EGG);
+    }
+
+    public void generateDisplayItems(net.minecraft.world.flag.FeatureFlagSet $$0, net.minecraft.world.item.CreativeModeTab$Output $$1) {
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ALLAY_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.AXOLOTL_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAT_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BEE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BLAZE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CAMEL_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CAT_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CAVE_SPIDER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CHICKEN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COD_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.COW_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.CREEPER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DOLPHIN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DONKEY_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.DROWNED_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ELDER_GUARDIAN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ENDERMAN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ENDERMITE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.EVOKER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FOX_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.FROG_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GHAST_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GLOW_SQUID_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GOAT_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GUARDIAN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HOGLIN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HORSE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HUSK_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.LLAMA_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MAGMA_CUBE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MOOSHROOM_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.MULE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.OCELOT_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PANDA_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PARROT_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PHANTOM_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PIG_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PIGLIN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PIGLIN_BRUTE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PILLAGER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POLAR_BEAR_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PUFFERFISH_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.RABBIT_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.RAVAGER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SALMON_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SHEEP_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SHULKER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SILVERFISH_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SKELETON_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SKELETON_HORSE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SLIME_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPIDER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SQUID_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STRAY_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.STRIDER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TADPOLE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TRADER_LLAMA_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TROPICAL_FISH_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.TURTLE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.VEX_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.VILLAGER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.VINDICATOR_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WANDERING_TRADER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WARDEN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WITCH_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WITHER_SKELETON_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WOLF_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ZOGLIN_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ZOMBIE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ZOMBIE_HORSE_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ZOMBIE_VILLAGER_SPAWN_EGG);
+        $$1.accept((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ZOMBIFIED_PIGLIN_SPAWN_EGG);
+    }
+}
+