--- net/minecraft/world/item/DoubleHighBlockItem.java
+++ net/minecraft/world/item/DoubleHighBlockItem.java
@@ -10,12 +10,12 @@
         super($$0, $$1);
     }
 
-    public boolean placeBlock(net.minecraft.world.item.context.BlockPlaceContext context, net.minecraft.world.level.block.state.BlockState state) {
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = level.isWaterAt(blockPos = context.getClickedPos().above()) ? net.minecraft.world.level.block.Blocks.WATER.defaultBlockState() : net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
-        level.setBlock(blockPos, blockState, 27);
-        return super.placeBlock(context, state);
+    public boolean placeBlock(net.minecraft.world.item.context.BlockPlaceContext $$0, net.minecraft.world.level.block.state.BlockState $$1) {
+        net.minecraft.core.BlockPos $$3;
+        net.minecraft.world.level.Level $$2 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$4 = $$2.isWaterAt($$3 = $$0.getClickedPos().above()) ? net.minecraft.world.level.block.Blocks.WATER.defaultBlockState() : net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
+        $$2.setBlock($$3, $$4, 27);
+        return super.placeBlock($$0, $$1);
     }
 }
 