--- net/minecraft/world/item/DyeableLeatherItem.java
+++ net/minecraft/world/item/DyeableLeatherItem.java
@@ -9,82 +9,80 @@
     public static final java.lang.String TAG_DISPLAY = "display";
     public static final int DEFAULT_LEATHER_COLOR = 10511680;
 
-    default public boolean hasCustomColor(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTagElement("display");
-        return compoundTag != null && compoundTag.contains("color", 99);
+    default public boolean hasCustomColor(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTagElement("display");
+        return $$1 != null && $$1.contains("color", 99);
     }
 
-    default public int getColor(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTagElement("display");
-        if (compoundTag != null && compoundTag.contains("color", 99)) {
-            return compoundTag.getInt("color");
+    default public int getColor(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTagElement("display");
+        if ($$1 != null && $$1.contains("color", 99)) {
+            return $$1.getInt("color");
         }
         return 10511680;
     }
 
-    default public void clearColor(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTagElement("display");
-        if (compoundTag != null && compoundTag.contains("color")) {
-            compoundTag.remove("color");
+    default public void clearColor(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTagElement("display");
+        if ($$1 != null && $$1.contains("color")) {
+            $$1.remove("color");
         }
     }
 
-    default public void setColor(net.minecraft.world.item.ItemStack stack, int color) {
-        stack.getOrCreateTagElement("display").putInt("color", color);
+    default public void setColor(net.minecraft.world.item.ItemStack $$0, int $$1) {
+        $$0.getOrCreateTagElement("display").putInt("color", $$1);
     }
 
-    public static net.minecraft.world.item.ItemStack dyeArmor(net.minecraft.world.item.ItemStack stack, java.util.List<net.minecraft.world.item.DyeItem> dyes) {
-        int n;
-        float f;
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.EMPTY;
-        int[] nArray = new int[3];
-        int n2 = 0;
-        int n3 = 0;
-        net.minecraft.world.item.DyeableLeatherItem dyeableLeatherItem = null;
-        net.minecraft.world.item.Item item = stack.getItem();
-        if (item instanceof net.minecraft.world.item.DyeableLeatherItem) {
-            dyeableLeatherItem = (net.minecraft.world.item.DyeableLeatherItem)item;
-            itemStack = stack.copy();
-            itemStack.setCount(1);
-            if (dyeableLeatherItem.hasCustomColor(stack)) {
-                int n4 = dyeableLeatherItem.getColor(itemStack);
-                float f2 = (float)(n4 >> 16 & 0xFF) / 255.0f;
-                float f3 = (float)(n4 >> 8 & 0xFF) / 255.0f;
-                f = (float)(n4 & 0xFF) / 255.0f;
-                n2 += (int)(java.lang.Math.max((float)f2, (float)java.lang.Math.max((float)f3, (float)f)) * 255.0f);
-                nArray[0] = nArray[0] + (int)(f2 * 255.0f);
-                nArray[1] = nArray[1] + (int)(f3 * 255.0f);
-                nArray[2] = nArray[2] + (int)(f * 255.0f);
-                ++n3;
+    public static net.minecraft.world.item.ItemStack dyeArmor(net.minecraft.world.item.ItemStack $$0, java.util.List<net.minecraft.world.item.DyeItem> $$1) {
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        int[] $$3 = new int[3];
+        int $$4 = 0;
+        int $$5 = 0;
+        net.minecraft.world.item.DyeableLeatherItem $$6 = null;
+        net.minecraft.world.item.Item $$7 = $$0.getItem();
+        if ($$7 instanceof net.minecraft.world.item.DyeableLeatherItem) {
+            $$6 = (net.minecraft.world.item.DyeableLeatherItem)$$7;
+            $$2 = $$0.copy();
+            $$2.setCount(1);
+            if ($$6.hasCustomColor($$0)) {
+                int $$8 = $$6.getColor($$2);
+                float $$9 = (float)($$8 >> 16 & 0xFF) / 255.0f;
+                float $$10 = (float)($$8 >> 8 & 0xFF) / 255.0f;
+                float $$11 = (float)($$8 & 0xFF) / 255.0f;
+                $$4 += (int)(java.lang.Math.max((float)$$9, (float)java.lang.Math.max((float)$$10, (float)$$11)) * 255.0f);
+                $$3[0] = $$3[0] + (int)($$9 * 255.0f);
+                $$3[1] = $$3[1] + (int)($$10 * 255.0f);
+                $$3[2] = $$3[2] + (int)($$11 * 255.0f);
+                ++$$5;
             }
-            for (net.minecraft.world.item.DyeItem dyeItem : dyes) {
-                float[] fArray = dyeItem.getDyeColor().getTextureDiffuseColors();
-                int n5 = (int)(fArray[0] * 255.0f);
-                int n6 = (int)(fArray[1] * 255.0f);
-                n = (int)(fArray[2] * 255.0f);
-                n2 += java.lang.Math.max((int)n5, (int)java.lang.Math.max((int)n6, (int)n));
-                nArray[0] = nArray[0] + n5;
-                nArray[1] = nArray[1] + n6;
-                nArray[2] = nArray[2] + n;
-                ++n3;
+            for (net.minecraft.world.item.DyeItem $$12 : $$1) {
+                float[] $$13 = $$12.getDyeColor().getTextureDiffuseColors();
+                int $$14 = (int)($$13[0] * 255.0f);
+                int $$15 = (int)($$13[1] * 255.0f);
+                int $$16 = (int)($$13[2] * 255.0f);
+                $$4 += java.lang.Math.max((int)$$14, (int)java.lang.Math.max((int)$$15, (int)$$16));
+                $$3[0] = $$3[0] + $$14;
+                $$3[1] = $$3[1] + $$15;
+                $$3[2] = $$3[2] + $$16;
+                ++$$5;
             }
         }
-        if (dyeableLeatherItem == null) {
+        if ($$6 == null) {
             return net.minecraft.world.item.ItemStack.EMPTY;
         }
-        int n7 = nArray[0] / n3;
-        int n8 = nArray[1] / n3;
-        int n9 = nArray[2] / n3;
-        f = (float)n2 / (float)n3;
-        float f4 = java.lang.Math.max((int)n7, (int)java.lang.Math.max((int)n8, (int)n9));
-        n7 = (int)((float)n7 * f / f4);
-        n8 = (int)((float)n8 * f / f4);
-        n9 = (int)((float)n9 * f / f4);
-        n = n7;
-        n = (n << 8) + n8;
-        n = (n << 8) + n9;
-        dyeableLeatherItem.setColor(itemStack, n);
-        return itemStack;
+        int $$17 = $$3[0] / $$5;
+        int $$18 = $$3[1] / $$5;
+        int $$19 = $$3[2] / $$5;
+        float $$20 = (float)$$4 / (float)$$5;
+        float $$21 = java.lang.Math.max((int)$$17, (int)java.lang.Math.max((int)$$18, (int)$$19));
+        $$17 = (int)((float)$$17 * $$20 / $$21);
+        $$18 = (int)((float)$$18 * $$20 / $$21);
+        $$19 = (int)((float)$$19 * $$20 / $$21);
+        int $$22 = $$17;
+        $$22 = ($$22 << 8) + $$18;
+        $$22 = ($$22 << 8) + $$19;
+        $$6.setColor($$2, $$22);
+        return $$2;
     }
 }
 