--- net/minecraft/world/item/DyeableHorseArmorItem.java
+++ net/minecraft/world/item/DyeableHorseArmorItem.java
@@ -7,8 +7,8 @@
 extends net.minecraft.world.item.HorseArmorItem
 implements net.minecraft.world.item.DyeableLeatherItem
  {
-    public net.minecraft.world.item.DyeableHorseArmorItem(int protection, java.lang.String identifier, net.minecraft.world.item.Item$Properties properties) {
-        super(protection, identifier, properties);
+    public net.minecraft.world.item.DyeableHorseArmorItem(int $$0, java.lang.String $$1, net.minecraft.world.item.Item$Properties $$2) {
+        super($$0, $$1, $$2);
     }
 }
 