--- net/minecraft/world/item/DyeColor.java
+++ net/minecraft/world/item/DyeColor.java
@@ -41,17 +41,17 @@
         return (net.minecraft.world.item.DyeColor)java.lang.Enum.valueOf(net.minecraft.world.item.DyeColor.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.item.DyeColor(int id, java.lang.String name, int textureColor, net.minecraft.world.level.material.MaterialColor color, int fireworkColor, int textColor) {
+    public net.minecraft.world.item.DyeColor(int $$0, java.lang.String $$1, int $$2, net.minecraft.world.level.material.MaterialColor $$3, int $$4, int $$5) {
         super(string, n);
-        this.id = id;
-        this.name = name;
-        this.color = color;
-        this.textColor = textColor;
-        int n2 = (textureColor & 0xFF0000) >> 16;
-        int n3 = (textureColor & 0xFF00) >> 8;
-        int n4 = (textureColor & 0xFF) >> 0;
-        this.textureDiffuseColors = new float[]{(float)n2 / 255.0f, (float)n3 / 255.0f, (float)n4 / 255.0f};
-        this.fireworkColor = fireworkColor;
+        this.id = $$0;
+        this.name = $$1;
+        this.color = $$3;
+        this.textColor = $$5;
+        int $$6 = ($$2 & 0xFF0000) >> 16;
+        int $$7 = ($$2 & 0xFF00) >> 8;
+        int $$8 = ($$2 & 0xFF) >> 0;
+        this.textureDiffuseColors = new float[]{(float)$$6 / 255.0f, (float)$$7 / 255.0f, (float)$$8 / 255.0f};
+        this.fireworkColor = $$4;
     }
 
     public int getId() {
@@ -78,26 +78,26 @@
         return this.textColor;
     }
 
-    public static net.minecraft.world.item.DyeColor byId(int colorId) {
-        if (colorId < 0 || colorId >= BY_ID.length) {
-            colorId = 0;
+    public static net.minecraft.world.item.DyeColor byId(int $$0) {
+        if ($$0 < 0 || $$0 >= BY_ID.length) {
+            $$0 = 0;
         }
-        return BY_ID[colorId];
+        return BY_ID[$$0];
     }
 
     @javax.annotation.Nullable
     @org.jetbrains.annotations.Contract(value="_,!null->!null;_,null->_")
-    public static net.minecraft.world.item.DyeColor byName(java.lang.String translationKey, @javax.annotation.Nullable net.minecraft.world.item.DyeColor fallback) {
-        for (net.minecraft.world.item.DyeColor dyeColor : net.minecraft.world.item.DyeColor.values()) {
-            if (!dyeColor.name.equals((java.lang.Object)translationKey)) continue;
-            return dyeColor;
+    public static net.minecraft.world.item.DyeColor byName(java.lang.String $$0, @javax.annotation.Nullable net.minecraft.world.item.DyeColor $$1) {
+        for (net.minecraft.world.item.DyeColor $$2 : net.minecraft.world.item.DyeColor.values()) {
+            if (!$$2.name.equals((java.lang.Object)$$0)) continue;
+            return $$2;
         }
-        return fallback;
+        return $$1;
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.item.DyeColor byFireworkColor(int fireworkColor) {
-        return (net.minecraft.world.item.DyeColor)BY_FIREWORK_COLOR.get(fireworkColor);
+    public static net.minecraft.world.item.DyeColor byFireworkColor(int $$0) {
+        return (net.minecraft.world.item.DyeColor)BY_FIREWORK_COLOR.get($$0);
     }
 
     public java.lang.String toString() {
@@ -108,12 +108,12 @@
         return this.name;
     }
 
-    public static /* synthetic */ net.minecraft.world.item.DyeColor lambda$static$2(net.minecraft.world.item.DyeColor dyeColor) {
-        return dyeColor;
+    public static /* synthetic */ net.minecraft.world.item.DyeColor lambda$static$2(net.minecraft.world.item.DyeColor $$0) {
+        return $$0;
     }
 
-    public static /* synthetic */ java.lang.Integer lambda$static$1(net.minecraft.world.item.DyeColor fireworkColor) {
-        return fireworkColor.fireworkColor;
+    public static /* synthetic */ java.lang.Integer lambda$static$1(net.minecraft.world.item.DyeColor $$0) {
+        return $$0.fireworkColor;
     }
 
     public static /* synthetic */ net.minecraft.world.item.DyeColor[] lambda$static$0(int $$0) {
@@ -143,7 +143,7 @@
         BLACK = new net.minecraft.world.item.DyeColor("BLACK", 15, 15, "black", 0x1D1D21, net.minecraft.world.level.material.MaterialColor.COLOR_BLACK, 0x1E1B1B, 0);
         $VALUES = net.minecraft.world.item.DyeColor.$values();
         BY_ID = (net.minecraft.world.item.DyeColor[])java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.item.DyeColor.values()).sorted(java.util.Comparator.comparingInt(net.minecraft.world.item.DyeColor::getId)).toArray(net.minecraft.world.item.DyeColor[]::new);
-        BY_FIREWORK_COLOR = new it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap((java.util.Map)java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.item.DyeColor.values()).collect(java.util.stream.Collectors.toMap(fireworkColor -> fireworkColor.fireworkColor, dyeColor -> dyeColor)));
+        BY_FIREWORK_COLOR = new it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap((java.util.Map)java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.item.DyeColor.values()).collect(java.util.stream.Collectors.toMap($$0 -> $$0.fireworkColor, $$0 -> $$0)));
     }
 }
 