--- net/minecraft/world/item/DyeItem.java
+++ net/minecraft/world/item/DyeItem.java
@@ -9,21 +9,21 @@
     public static java.util.Map<net.minecraft.world.item.DyeColor, net.minecraft.world.item.DyeItem> ITEM_BY_COLOR;
     public net.minecraft.world.item.DyeColor dyeColor;
 
-    public net.minecraft.world.item.DyeItem(net.minecraft.world.item.DyeColor dyeColor, net.minecraft.world.item.Item$Properties properties) {
-        super(properties);
-        this.dyeColor = dyeColor;
-        ITEM_BY_COLOR.put((java.lang.Object)dyeColor, (java.lang.Object)this);
+    public net.minecraft.world.item.DyeItem(net.minecraft.world.item.DyeColor $$0, net.minecraft.world.item.Item$Properties $$1) {
+        super($$1);
+        this.dyeColor = $$0;
+        ITEM_BY_COLOR.put((java.lang.Object)$$0, (java.lang.Object)this);
     }
 
-    public net.minecraft.world.InteractionResult interactLivingEntity(net.minecraft.world.item.ItemStack stack, net.minecraft.world.entity.player.Player player, net.minecraft.world.entity.LivingEntity target, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.entity.animal.Sheep sheep;
-        if (target instanceof net.minecraft.world.entity.animal.Sheep && (sheep = (net.minecraft.world.entity.animal.Sheep)target).isAlive() && !sheep.isSheared() && sheep.getColor() != this.dyeColor) {
-            sheep.level.playSound(player, (net.minecraft.world.entity.Entity)sheep, net.minecraft.sounds.SoundEvents.DYE_USE, net.minecraft.sounds.SoundSource.PLAYERS, 1.0f, 1.0f);
-            if (!player.level.isClientSide) {
-                sheep.setColor(this.dyeColor);
-                stack.shrink(1);
+    public net.minecraft.world.InteractionResult interactLivingEntity(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.entity.LivingEntity $$2, net.minecraft.world.InteractionHand $$3) {
+        net.minecraft.world.entity.animal.Sheep $$4;
+        if ($$2 instanceof net.minecraft.world.entity.animal.Sheep && ($$4 = (net.minecraft.world.entity.animal.Sheep)$$2).isAlive() && !$$4.isSheared() && $$4.getColor() != this.dyeColor) {
+            $$4.level.playSound($$1, (net.minecraft.world.entity.Entity)$$4, net.minecraft.sounds.SoundEvents.DYE_USE, net.minecraft.sounds.SoundSource.PLAYERS, 1.0f, 1.0f);
+            if (!$$1.level.isClientSide) {
+                $$4.setColor(this.dyeColor);
+                $$0.shrink(1);
             }
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)player.level.isClientSide);
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.level.isClientSide);
         }
         return net.minecraft.world.InteractionResult.PASS;
     }
@@ -32,8 +32,8 @@
         return this.dyeColor;
     }
 
-    public static net.minecraft.world.item.DyeItem byColor(net.minecraft.world.item.DyeColor color) {
-        return (net.minecraft.world.item.DyeItem)ITEM_BY_COLOR.get((java.lang.Object)color);
+    public static net.minecraft.world.item.DyeItem byColor(net.minecraft.world.item.DyeColor $$0) {
+        return (net.minecraft.world.item.DyeItem)ITEM_BY_COLOR.get((java.lang.Object)$$0);
     }
 
     static {