--- net/minecraft/world/item/ElytraItem.java
+++ net/minecraft/world/item/ElytraItem.java
@@ -12,27 +12,27 @@
         net.minecraft.world.level.block.DispenserBlock.registerBehavior((net.minecraft.world.level.ItemLike)this, (net.minecraft.core.dispenser.DispenseItemBehavior)net.minecraft.world.item.ArmorItem.DISPENSE_ITEM_BEHAVIOR);
     }
 
-    public static boolean isFlyEnabled(net.minecraft.world.item.ItemStack elytraStack) {
-        return elytraStack.getDamageValue() < elytraStack.getMaxDamage() - 1;
+    public static boolean isFlyEnabled(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.getDamageValue() < $$0.getMaxDamage() - 1;
     }
 
-    public boolean isValidRepairItem(net.minecraft.world.item.ItemStack toRepair, net.minecraft.world.item.ItemStack repair) {
-        return repair.is(net.minecraft.world.item.Items.PHANTOM_MEMBRANE);
+    public boolean isValidRepairItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        return $$1.is(net.minecraft.world.item.Items.PHANTOM_MEMBRANE);
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        net.minecraft.world.entity.EquipmentSlot equipmentSlot = net.minecraft.world.entity.Mob.getEquipmentSlotForItem((net.minecraft.world.item.ItemStack)itemStack);
-        net.minecraft.world.item.ItemStack itemStack2 = player.getItemBySlot(equipmentSlot);
-        if (itemStack2.isEmpty()) {
-            player.setItemSlot(equipmentSlot, itemStack.copy());
-            if (!level.isClientSide()) {
-                player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+        net.minecraft.world.entity.EquipmentSlot $$4 = net.minecraft.world.entity.Mob.getEquipmentSlotForItem((net.minecraft.world.item.ItemStack)$$3);
+        net.minecraft.world.item.ItemStack $$5 = $$1.getItemBySlot($$4);
+        if ($$5.isEmpty()) {
+            $$1.setItemSlot($$4, $$3.copy());
+            if (!$$0.isClientSide()) {
+                $$1.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
             }
-            itemStack.setCount(0);
-            return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)itemStack, (boolean)level.isClientSide());
+            $$3.setCount(0);
+            return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)$$3, (boolean)$$0.isClientSide());
         }
-        return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)itemStack);
+        return net.minecraft.world.InteractionResultHolder.fail((java.lang.Object)$$3);
     }
 
     @javax.annotation.Nullable