--- net/minecraft/world/item/EnchantedBookItem.java
+++ net/minecraft/world/item/EnchantedBookItem.java
@@ -12,71 +12,51 @@
         super($$0);
     }
 
-    public boolean isFoil(net.minecraft.world.item.ItemStack stack) {
+    public boolean isFoil(net.minecraft.world.item.ItemStack $$0) {
         return true;
     }
 
-    public boolean isEnchantable(net.minecraft.world.item.ItemStack stack) {
+    public boolean isEnchantable(net.minecraft.world.item.ItemStack $$0) {
         return false;
     }
 
-    public static net.minecraft.nbt.ListTag getEnchantments(net.minecraft.world.item.ItemStack enchantedBookStack) {
-        net.minecraft.nbt.CompoundTag compoundTag = enchantedBookStack.getTag();
-        if (compoundTag != null) {
-            return compoundTag.getList("StoredEnchantments", 10);
+    public static net.minecraft.nbt.ListTag getEnchantments(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTag();
+        if ($$1 != null) {
+            return $$1.getList("StoredEnchantments", 10);
         }
         return new net.minecraft.nbt.ListTag();
     }
 
-    public void appendHoverText(net.minecraft.world.item.ItemStack stack, @javax.annotation.Nullable net.minecraft.world.level.Level level, java.util.List<net.minecraft.network.chat.Component> tooltip, net.minecraft.world.item.TooltipFlag flag) {
-        super.appendHoverText(stack, level, tooltip, flag);
-        net.minecraft.world.item.ItemStack.appendEnchantmentNames(tooltip, (net.minecraft.nbt.ListTag)net.minecraft.world.item.EnchantedBookItem.getEnchantments((net.minecraft.world.item.ItemStack)stack));
+    public void appendHoverText(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.world.level.Level $$1, java.util.List<net.minecraft.network.chat.Component> $$2, net.minecraft.world.item.TooltipFlag $$3) {
+        super.appendHoverText($$0, $$1, $$2, $$3);
+        net.minecraft.world.item.ItemStack.appendEnchantmentNames($$2, (net.minecraft.nbt.ListTag)net.minecraft.world.item.EnchantedBookItem.getEnchantments((net.minecraft.world.item.ItemStack)$$0));
     }
 
-    public static void addEnchantment(net.minecraft.world.item.ItemStack stack, net.minecraft.world.item.enchantment.EnchantmentInstance instance) {
-        net.minecraft.nbt.ListTag listTag = net.minecraft.world.item.EnchantedBookItem.getEnchantments((net.minecraft.world.item.ItemStack)stack);
-        boolean bl = true;
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentId((net.minecraft.world.item.enchantment.Enchantment)instance.enchantment);
-        for (int i = 0; i < listTag.size(); ++i) {
-            net.minecraft.nbt.CompoundTag compoundTag = listTag.getCompound(i);
-            net.minecraft.resources.ResourceLocation resourceLocation2 = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentId((net.minecraft.nbt.CompoundTag)compoundTag);
-            if (resourceLocation2 == null || !resourceLocation2.equals((java.lang.Object)resourceLocation)) continue;
-            if (net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentLevel((net.minecraft.nbt.CompoundTag)compoundTag) < instance.level) {
-                net.minecraft.world.item.enchantment.EnchantmentHelper.setEnchantmentLevel((net.minecraft.nbt.CompoundTag)compoundTag, (int)instance.level);
+    public static void addEnchantment(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.enchantment.EnchantmentInstance $$1) {
+        net.minecraft.nbt.ListTag $$2 = net.minecraft.world.item.EnchantedBookItem.getEnchantments((net.minecraft.world.item.ItemStack)$$0);
+        boolean $$3 = true;
+        net.minecraft.resources.ResourceLocation $$4 = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentId((net.minecraft.world.item.enchantment.Enchantment)$$1.enchantment);
+        for (int $$5 = 0; $$5 < $$2.size(); ++$$5) {
+            net.minecraft.nbt.CompoundTag $$6 = $$2.getCompound($$5);
+            net.minecraft.resources.ResourceLocation $$7 = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentId((net.minecraft.nbt.CompoundTag)$$6);
+            if ($$7 == null || !$$7.equals((java.lang.Object)$$4)) continue;
+            if (net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentLevel((net.minecraft.nbt.CompoundTag)$$6) < $$1.level) {
+                net.minecraft.world.item.enchantment.EnchantmentHelper.setEnchantmentLevel((net.minecraft.nbt.CompoundTag)$$6, (int)$$1.level);
             }
-            bl = false;
+            $$3 = false;
             break;
         }
-        if (bl) {
-            listTag.add((java.lang.Object)net.minecraft.world.item.enchantment.EnchantmentHelper.storeEnchantment((net.minecraft.resources.ResourceLocation)resourceLocation, (int)instance.level));
+        if ($$3) {
+            $$2.add((java.lang.Object)net.minecraft.world.item.enchantment.EnchantmentHelper.storeEnchantment((net.minecraft.resources.ResourceLocation)$$4, (int)$$1.level));
         }
-        stack.getOrCreateTag().put("StoredEnchantments", (net.minecraft.nbt.Tag)listTag);
+        $$0.getOrCreateTag().put("StoredEnchantments", (net.minecraft.nbt.Tag)$$2);
     }
 
-    public static net.minecraft.world.item.ItemStack createForEnchantment(net.minecraft.world.item.enchantment.EnchantmentInstance instance) {
-        net.minecraft.world.item.ItemStack itemStack = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ENCHANTED_BOOK);
-        net.minecraft.world.item.EnchantedBookItem.addEnchantment((net.minecraft.world.item.ItemStack)itemStack, (net.minecraft.world.item.enchantment.EnchantmentInstance)instance);
-        return itemStack;
+    public static net.minecraft.world.item.ItemStack createForEnchantment(net.minecraft.world.item.enchantment.EnchantmentInstance $$0) {
+        net.minecraft.world.item.ItemStack $$1 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ENCHANTED_BOOK);
+        net.minecraft.world.item.EnchantedBookItem.addEnchantment((net.minecraft.world.item.ItemStack)$$1, (net.minecraft.world.item.enchantment.EnchantmentInstance)$$0);
+        return $$1;
     }
-
-    public void fillItemCategory(net.minecraft.world.item.CreativeModeTab group, net.minecraft.core.NonNullList<net.minecraft.world.item.ItemStack> items) {
-block4:        {
-block3:            {
-                if (group != net.minecraft.world.item.CreativeModeTab.TAB_SEARCH) break block3;
-                for (net.minecraft.world.item.enchantment.Enchantment enchantment : net.minecraft.core.Registry.ENCHANTMENT) {
-                    if (enchantment.category == null) continue;
-                    for (int i = enchantment.getMinLevel(); i <= enchantment.getMaxLevel(); ++i) {
-                        items.add((java.lang.Object)net.minecraft.world.item.EnchantedBookItem.createForEnchantment((net.minecraft.world.item.enchantment.EnchantmentInstance)new net.minecraft.world.item.enchantment.EnchantmentInstance(enchantment, i)));
-                    }
-                }
-                break block4;
-            }
-            if (group.getEnchantmentCategories().length == 0) break block4;
-            for (net.minecraft.world.item.enchantment.Enchantment enchantment : net.minecraft.core.Registry.ENCHANTMENT) {
-                if (!group.hasEnchantmentCategory(enchantment.category)) continue;
-                items.add((java.lang.Object)net.minecraft.world.item.EnchantedBookItem.createForEnchantment((net.minecraft.world.item.enchantment.EnchantmentInstance)new net.minecraft.world.item.enchantment.EnchantmentInstance(enchantment, enchantment.getMaxLevel())));
-            }
-        }
-    }
 }
 