--- net/minecraft/world/item/EndCrystalItem.java
+++ net/minecraft/world/item/EndCrystalItem.java
@@ -10,39 +10,39 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        double d;
-        double d2;
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos = context.getClickedPos());
-        if (!blockState.is(net.minecraft.world.level.block.Blocks.OBSIDIAN) && !blockState.is(net.minecraft.world.level.block.Blocks.BEDROCK)) {
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        double $$7;
+        double $$6;
+        net.minecraft.core.BlockPos $$2;
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$1.getBlockState($$2 = $$0.getClickedPos());
+        if (!$$3.is(net.minecraft.world.level.block.Blocks.OBSIDIAN) && !$$3.is(net.minecraft.world.level.block.Blocks.BEDROCK)) {
             return net.minecraft.world.InteractionResult.FAIL;
         }
-        net.minecraft.core.BlockPos blockPos2 = blockPos.above();
-        if (!level.isEmptyBlock(blockPos2)) {
+        net.minecraft.core.BlockPos $$4 = $$2.above();
+        if (!$$1.isEmptyBlock($$4)) {
             return net.minecraft.world.InteractionResult.FAIL;
         }
-        double d3 = blockPos2.getX();
-        java.util.List list = level.getEntities(null, new net.minecraft.world.phys.AABB(d3, d2 = (double)blockPos2.getY(), d = (double)blockPos2.getZ(), d3 + 1.0, d2 + 2.0, d + 1.0));
-        if (!list.isEmpty()) {
+        double $$5 = $$4.getX();
+        java.util.List $$8 = $$1.getEntities(null, new net.minecraft.world.phys.AABB($$5, $$6 = (double)$$4.getY(), $$7 = (double)$$4.getZ(), $$5 + 1.0, $$6 + 2.0, $$7 + 1.0));
+        if (!$$8.isEmpty()) {
             return net.minecraft.world.InteractionResult.FAIL;
         }
-        if (level instanceof net.minecraft.server.level.ServerLevel) {
-            net.minecraft.world.entity.boss.enderdragon.EndCrystal endCrystal = new net.minecraft.world.entity.boss.enderdragon.EndCrystal(level, d3 + 0.5, d2, d + 0.5);
-            endCrystal.setShowBottom(false);
-            level.addFreshEntity((net.minecraft.world.entity.Entity)endCrystal);
-            level.gameEvent((net.minecraft.world.entity.Entity)context.getPlayer(), net.minecraft.world.level.gameevent.GameEvent.ENTITY_PLACE, blockPos2);
-            net.minecraft.world.level.dimension.end.EndDragonFight endDragonFight = ((net.minecraft.server.level.ServerLevel)level).dragonFight();
-            if (endDragonFight != null) {
-                endDragonFight.tryRespawn();
+        if ($$1 instanceof net.minecraft.server.level.ServerLevel) {
+            net.minecraft.world.entity.boss.enderdragon.EndCrystal $$9 = new net.minecraft.world.entity.boss.enderdragon.EndCrystal($$1, $$5 + 0.5, $$6, $$7 + 0.5);
+            $$9.setShowBottom(false);
+            $$1.addFreshEntity((net.minecraft.world.entity.Entity)$$9);
+            $$1.gameEvent((net.minecraft.world.entity.Entity)$$0.getPlayer(), net.minecraft.world.level.gameevent.GameEvent.ENTITY_PLACE, $$4);
+            net.minecraft.world.level.dimension.end.EndDragonFight $$10 = ((net.minecraft.server.level.ServerLevel)$$1).dragonFight();
+            if ($$10 != null) {
+                $$10.tryRespawn();
             }
         }
-        context.getItemInHand().shrink(1);
-        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+        $$0.getItemInHand().shrink(1);
+        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
     }
 
-    public boolean isFoil(net.minecraft.world.item.ItemStack stack) {
+    public boolean isFoil(net.minecraft.world.item.ItemStack $$0) {
         return true;
     }
 }