--- net/minecraft/world/item/FireworkRocketItem$Shape.java
+++ net/minecraft/world/item/FireworkRocketItem$Shape.java
@@ -24,10 +24,10 @@
         return (net.minecraft.world.item.FireworkRocketItem$Shape)java.lang.Enum.valueOf(net.minecraft.world.item.FireworkRocketItem$Shape.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.item.FireworkRocketItem$Shape(int id, java.lang.String name) {
+    public net.minecraft.world.item.FireworkRocketItem$Shape(int $$0, java.lang.String $$1) {
         super(string, n);
-        this.id = id;
-        this.name = name;
+        this.id = $$0;
+        this.name = $$1;
     }
 
     public int getId() {
@@ -38,19 +38,19 @@
         return this.name;
     }
 
-    public static net.minecraft.world.item.FireworkRocketItem$Shape byId(int index) {
-        if (index < 0 || index >= BY_ID.length) {
+    public static net.minecraft.world.item.FireworkRocketItem$Shape byId(int $$0) {
+        if ($$0 < 0 || $$0 >= BY_ID.length) {
             return SMALL_BALL;
         }
-        return BY_ID[index];
+        return BY_ID[$$0];
     }
 
     public static /* synthetic */ net.minecraft.world.item.FireworkRocketItem$Shape[] lambda$static$1(int $$0) {
         return new net.minecraft.world.item.FireworkRocketItem$Shape[$$0];
     }
 
-    public static /* synthetic */ int lambda$static$0(net.minecraft.world.item.FireworkRocketItem$Shape shape) {
-        return shape.id;
+    public static /* synthetic */ int lambda$static$0(net.minecraft.world.item.FireworkRocketItem$Shape $$0) {
+        return $$0.id;
     }
 
     public static /* synthetic */ net.minecraft.world.item.FireworkRocketItem$Shape[] $values() {
@@ -64,7 +64,7 @@
         CREEPER = new net.minecraft.world.item.FireworkRocketItem$Shape("CREEPER", 3, 3, "creeper");
         BURST = new net.minecraft.world.item.FireworkRocketItem$Shape("BURST", 4, 4, "burst");
         $VALUES = net.minecraft.world.item.FireworkRocketItem$Shape.$values();
-        BY_ID = (net.minecraft.world.item.FireworkRocketItem$Shape[])java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.item.FireworkRocketItem$Shape.values()).sorted(java.util.Comparator.comparingInt(shape -> shape.id)).toArray(net.minecraft.world.item.FireworkRocketItem$Shape[]::new);
+        BY_ID = (net.minecraft.world.item.FireworkRocketItem$Shape[])java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.item.FireworkRocketItem$Shape.values()).sorted(java.util.Comparator.comparingInt($$0 -> $$0.id)).toArray(net.minecraft.world.item.FireworkRocketItem$Shape[]::new);
     }
 }
 