--- net/minecraft/world/item/FireworkStarItem.java
+++ net/minecraft/world/item/FireworkStarItem.java
@@ -10,48 +10,48 @@
         super($$0);
     }
 
-    public void appendHoverText(net.minecraft.world.item.ItemStack stack, @javax.annotation.Nullable net.minecraft.world.level.Level level, java.util.List<net.minecraft.network.chat.Component> tooltip, net.minecraft.world.item.TooltipFlag flag) {
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTagElement("Explosion");
-        if (compoundTag != null) {
-            net.minecraft.world.item.FireworkStarItem.appendHoverText((net.minecraft.nbt.CompoundTag)compoundTag, tooltip);
+    public void appendHoverText(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.world.level.Level $$1, java.util.List<net.minecraft.network.chat.Component> $$2, net.minecraft.world.item.TooltipFlag $$3) {
+        net.minecraft.nbt.CompoundTag $$4 = $$0.getTagElement("Explosion");
+        if ($$4 != null) {
+            net.minecraft.world.item.FireworkStarItem.appendHoverText((net.minecraft.nbt.CompoundTag)$$4, $$2);
         }
     }
 
-    public static void appendHoverText(net.minecraft.nbt.CompoundTag compound, java.util.List<net.minecraft.network.chat.Component> tooltipComponents) {
-        int[] nArray;
-        net.minecraft.world.item.FireworkRocketItem$Shape shape = net.minecraft.world.item.FireworkRocketItem$Shape.byId((int)compound.getByte("Type"));
-        tooltipComponents.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("item.minecraft.firework_star.shape." + shape.getName())).withStyle(net.minecraft.ChatFormatting.GRAY));
-        int[] nArray2 = compound.getIntArray("Colors");
-        if (nArray2.length > 0) {
-            tooltipComponents.add((java.lang.Object)net.minecraft.world.item.FireworkStarItem.appendColors((net.minecraft.network.chat.MutableComponent)net.minecraft.network.chat.Component.empty().withStyle(net.minecraft.ChatFormatting.GRAY), (int[])nArray2));
+    public static void appendHoverText(net.minecraft.nbt.CompoundTag $$0, java.util.List<net.minecraft.network.chat.Component> $$1) {
+        int[] $$4;
+        net.minecraft.world.item.FireworkRocketItem$Shape $$2 = net.minecraft.world.item.FireworkRocketItem$Shape.byId((int)$$0.getByte("Type"));
+        $$1.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("item.minecraft.firework_star.shape." + $$2.getName())).withStyle(net.minecraft.ChatFormatting.GRAY));
+        int[] $$3 = $$0.getIntArray("Colors");
+        if ($$3.length > 0) {
+            $$1.add((java.lang.Object)net.minecraft.world.item.FireworkStarItem.appendColors((net.minecraft.network.chat.MutableComponent)net.minecraft.network.chat.Component.empty().withStyle(net.minecraft.ChatFormatting.GRAY), (int[])$$3));
         }
-        if ((nArray = compound.getIntArray("FadeColors")).length > 0) {
-            tooltipComponents.add((java.lang.Object)net.minecraft.world.item.FireworkStarItem.appendColors((net.minecraft.network.chat.MutableComponent)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_star.fade_to").append(" ").withStyle(net.minecraft.ChatFormatting.GRAY), (int[])nArray));
+        if (($$4 = $$0.getIntArray("FadeColors")).length > 0) {
+            $$1.add((java.lang.Object)net.minecraft.world.item.FireworkStarItem.appendColors((net.minecraft.network.chat.MutableComponent)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_star.fade_to").append(" ").withStyle(net.minecraft.ChatFormatting.GRAY), (int[])$$4));
         }
-        if (compound.getBoolean("Trail")) {
-            tooltipComponents.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_star.trail").withStyle(net.minecraft.ChatFormatting.GRAY));
+        if ($$0.getBoolean("Trail")) {
+            $$1.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_star.trail").withStyle(net.minecraft.ChatFormatting.GRAY));
         }
-        if (compound.getBoolean("Flicker")) {
-            tooltipComponents.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_star.flicker").withStyle(net.minecraft.ChatFormatting.GRAY));
+        if ($$0.getBoolean("Flicker")) {
+            $$1.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_star.flicker").withStyle(net.minecraft.ChatFormatting.GRAY));
         }
     }
 
-    public static net.minecraft.network.chat.Component appendColors(net.minecraft.network.chat.MutableComponent tooltipComponent, int[] colors) {
-        for (int i = 0; i < colors.length; ++i) {
-            if (i > 0) {
-                tooltipComponent.append(", ");
+    public static net.minecraft.network.chat.Component appendColors(net.minecraft.network.chat.MutableComponent $$0, int[] $$1) {
+        for (int $$2 = 0; $$2 < $$1.length; ++$$2) {
+            if ($$2 > 0) {
+                $$0.append(", ");
             }
-            tooltipComponent.append(net.minecraft.world.item.FireworkStarItem.getColorName((int)colors[i]));
+            $$0.append(net.minecraft.world.item.FireworkStarItem.getColorName((int)$$1[$$2]));
         }
-        return tooltipComponent;
+        return $$0;
     }
 
-    public static net.minecraft.network.chat.Component getColorName(int color) {
-        net.minecraft.world.item.DyeColor dyeColor = net.minecraft.world.item.DyeColor.byFireworkColor((int)color);
-        if (dyeColor == null) {
+    public static net.minecraft.network.chat.Component getColorName(int $$0) {
+        net.minecraft.world.item.DyeColor $$1 = net.minecraft.world.item.DyeColor.byFireworkColor((int)$$0);
+        if ($$1 == null) {
             return net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_star.custom_color");
         }
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)("item.minecraft.firework_star." + dyeColor.getName()));
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)("item.minecraft.firework_star." + $$1.getName()));
     }
 }
 