--- net/minecraft/world/item/FireworkRocketItem.java
+++ net/minecraft/world/item/FireworkRocketItem.java
@@ -21,54 +21,54 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.world.level.Level level = context.getLevel();
-        if (!level.isClientSide) {
-            net.minecraft.world.item.ItemStack itemStack = context.getItemInHand();
-            net.minecraft.world.phys.Vec3 vec3 = context.getClickLocation();
-            net.minecraft.core.Direction direction = context.getClickedFace();
-            net.minecraft.world.entity.projectile.FireworkRocketEntity fireworkRocketEntity = new net.minecraft.world.entity.projectile.FireworkRocketEntity(level, (net.minecraft.world.entity.Entity)context.getPlayer(), vec3.x + (double)direction.getStepX() * 0.15, vec3.y + (double)direction.getStepY() * 0.15, vec3.z + (double)direction.getStepZ() * 0.15, itemStack);
-            level.addFreshEntity((net.minecraft.world.entity.Entity)fireworkRocketEntity);
-            itemStack.shrink(1);
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        if (!$$1.isClientSide) {
+            net.minecraft.world.item.ItemStack $$2 = $$0.getItemInHand();
+            net.minecraft.world.phys.Vec3 $$3 = $$0.getClickLocation();
+            net.minecraft.core.Direction $$4 = $$0.getClickedFace();
+            net.minecraft.world.entity.projectile.FireworkRocketEntity $$5 = new net.minecraft.world.entity.projectile.FireworkRocketEntity($$1, (net.minecraft.world.entity.Entity)$$0.getPlayer(), $$3.x + (double)$$4.getStepX() * 0.15, $$3.y + (double)$$4.getStepY() * 0.15, $$3.z + (double)$$4.getStepZ() * 0.15, $$2);
+            $$1.addFreshEntity((net.minecraft.world.entity.Entity)$$5);
+            $$2.shrink(1);
         }
-        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        if (player.isFallFlying()) {
-            net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-            if (!level.isClientSide) {
-                net.minecraft.world.entity.projectile.FireworkRocketEntity fireworkRocketEntity = new net.minecraft.world.entity.projectile.FireworkRocketEntity(level, itemStack, (net.minecraft.world.entity.LivingEntity)player);
-                level.addFreshEntity((net.minecraft.world.entity.Entity)fireworkRocketEntity);
-                if (!player.getAbilities().instabuild) {
-                    itemStack.shrink(1);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        if ($$1.isFallFlying()) {
+            net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+            if (!$$0.isClientSide) {
+                net.minecraft.world.entity.projectile.FireworkRocketEntity $$4 = new net.minecraft.world.entity.projectile.FireworkRocketEntity($$0, $$3, (net.minecraft.world.entity.LivingEntity)$$1);
+                $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$4);
+                if (!$$1.getAbilities().instabuild) {
+                    $$3.shrink(1);
                 }
-                player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+                $$1.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
             }
-            return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)player.getItemInHand(hand), (boolean)level.isClientSide());
+            return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)$$1.getItemInHand($$2), (boolean)$$0.isClientSide());
         }
-        return net.minecraft.world.InteractionResultHolder.pass((java.lang.Object)player.getItemInHand(hand));
+        return net.minecraft.world.InteractionResultHolder.pass((java.lang.Object)$$1.getItemInHand($$2));
     }
 
-    public void appendHoverText(net.minecraft.world.item.ItemStack stack, @javax.annotation.Nullable net.minecraft.world.level.Level level, java.util.List<net.minecraft.network.chat.Component> tooltip, net.minecraft.world.item.TooltipFlag flag) {
-        net.minecraft.nbt.ListTag listTag;
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTagElement("Fireworks");
-        if (compoundTag == null) {
+    public void appendHoverText(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.world.level.Level $$1, java.util.List<net.minecraft.network.chat.Component> $$2, net.minecraft.world.item.TooltipFlag $$3) {
+        net.minecraft.nbt.ListTag $$5;
+        net.minecraft.nbt.CompoundTag $$4 = $$0.getTagElement("Fireworks");
+        if ($$4 == null) {
             return;
         }
-        if (compoundTag.contains("Flight", 99)) {
-            tooltip.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_rocket.flight").append(" ").append(java.lang.String.valueOf((int)compoundTag.getByte("Flight"))).withStyle(net.minecraft.ChatFormatting.GRAY));
+        if ($$4.contains("Flight", 99)) {
+            $$2.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"item.minecraft.firework_rocket.flight").append(" ").append(java.lang.String.valueOf((int)$$4.getByte("Flight"))).withStyle(net.minecraft.ChatFormatting.GRAY));
         }
-        if (!(listTag = compoundTag.getList("Explosions", 10)).isEmpty()) {
-            for (int i = 0; i < listTag.size(); ++i) {
-                net.minecraft.nbt.CompoundTag compoundTag2 = listTag.getCompound(i);
-                java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-                net.minecraft.world.item.FireworkStarItem.appendHoverText((net.minecraft.nbt.CompoundTag)compoundTag2, (java.util.List)arrayList);
-                if (arrayList.isEmpty()) continue;
-                for (int j = 1; j < arrayList.size(); ++j) {
-                    arrayList.set(j, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)"  ").append((net.minecraft.network.chat.Component)arrayList.get(j)).withStyle(net.minecraft.ChatFormatting.GRAY));
+        if (!($$5 = $$4.getList("Explosions", 10)).isEmpty()) {
+            for (int $$6 = 0; $$6 < $$5.size(); ++$$6) {
+                net.minecraft.nbt.CompoundTag $$7 = $$5.getCompound($$6);
+                java.util.ArrayList $$8 = com.google.common.collect.Lists.newArrayList();
+                net.minecraft.world.item.FireworkStarItem.appendHoverText((net.minecraft.nbt.CompoundTag)$$7, (java.util.List)$$8);
+                if ($$8.isEmpty()) continue;
+                for (int $$9 = 1; $$9 < $$8.size(); ++$$9) {
+                    $$8.set($$9, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)"  ").append((net.minecraft.network.chat.Component)$$8.get($$9)).withStyle(net.minecraft.ChatFormatting.GRAY));
                 }
-                tooltip.addAll((java.util.Collection)arrayList);
+                $$2.addAll((java.util.Collection)$$8);
             }
         }
     }