--- net/minecraft/world/item/FireChargeItem.java
+++ net/minecraft/world/item/FireChargeItem.java
@@ -10,32 +10,32 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.core.BlockPos blockPos = context.getClickedPos();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos);
-        boolean bl = false;
-        if (net.minecraft.world.level.block.CampfireBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState) || net.minecraft.world.level.block.CandleBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState) || net.minecraft.world.level.block.CandleCakeBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState)) {
-            this.playSound(level, blockPos);
-            level.setBlockAndUpdate(blockPos, (net.minecraft.world.level.block.state.BlockState)blockState.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.state.properties.BlockStateProperties.LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)));
-            level.gameEvent((net.minecraft.world.entity.Entity)context.getPlayer(), net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, blockPos);
-            bl = true;
-        } else if (net.minecraft.world.level.block.BaseFireBlock.canBePlacedAt((net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)(blockPos = blockPos.relative(context.getClickedFace())), (net.minecraft.core.Direction)context.getHorizontalDirection())) {
-            this.playSound(level, blockPos);
-            level.setBlockAndUpdate(blockPos, net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)blockPos));
-            level.gameEvent((net.minecraft.world.entity.Entity)context.getPlayer(), net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, blockPos);
-            bl = true;
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        net.minecraft.core.BlockPos $$2 = $$0.getClickedPos();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$1.getBlockState($$2);
+        boolean $$4 = false;
+        if (net.minecraft.world.level.block.CampfireBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$3) || net.minecraft.world.level.block.CandleBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$3) || net.minecraft.world.level.block.CandleCakeBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$3)) {
+            this.playSound($$1, $$2);
+            $$1.setBlockAndUpdate($$2, (net.minecraft.world.level.block.state.BlockState)$$3.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.state.properties.BlockStateProperties.LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)));
+            $$1.gameEvent((net.minecraft.world.entity.Entity)$$0.getPlayer(), net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, $$2);
+            $$4 = true;
+        } else if (net.minecraft.world.level.block.BaseFireBlock.canBePlacedAt((net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)($$2 = $$2.relative($$0.getClickedFace())), (net.minecraft.core.Direction)$$0.getHorizontalDirection())) {
+            this.playSound($$1, $$2);
+            $$1.setBlockAndUpdate($$2, net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)$$1, (net.minecraft.core.BlockPos)$$2));
+            $$1.gameEvent((net.minecraft.world.entity.Entity)$$0.getPlayer(), net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, $$2);
+            $$4 = true;
         }
-        if (bl) {
-            context.getItemInHand().shrink(1);
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+        if ($$4) {
+            $$0.getItemInHand().shrink(1);
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
         }
         return net.minecraft.world.InteractionResult.FAIL;
     }
 
-    public void playSound(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.util.RandomSource randomSource = level.getRandom();
-        level.playSound(null, pos, net.minecraft.sounds.SoundEvents.FIRECHARGE_USE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, (randomSource.nextFloat() - randomSource.nextFloat()) * 0.2f + 1.0f);
+    public void playSound(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1) {
+        net.minecraft.util.RandomSource $$2 = $$0.getRandom();
+        $$0.playSound(null, $$1, net.minecraft.sounds.SoundEvents.FIRECHARGE_USE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, ($$2.nextFloat() - $$2.nextFloat()) * 0.2f + 1.0f);
     }
 }
 