--- net/minecraft/world/item/FlintAndSteelItem.java
+++ net/minecraft/world/item/FlintAndSteelItem.java
@@ -10,32 +10,32 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.entity.player.Player player = context.getPlayer();
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos = context.getClickedPos());
-        if (net.minecraft.world.level.block.CampfireBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState) || net.minecraft.world.level.block.CandleBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState) || net.minecraft.world.level.block.CandleCakeBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState)) {
-            level.playSound(player, blockPos, net.minecraft.sounds.SoundEvents.FLINTANDSTEEL_USE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, level.getRandom().nextFloat() * 0.4f + 0.8f);
-            level.setBlock(blockPos, (net.minecraft.world.level.block.state.BlockState)blockState.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.state.properties.BlockStateProperties.LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)), 11);
-            level.gameEvent((net.minecraft.world.entity.Entity)player, net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, blockPos);
-            if (player != null) {
-                context.getItemInHand().hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)player, contextPlayer -> contextPlayer.broadcastBreakEvent(context.getHand()));
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.core.BlockPos $$3;
+        net.minecraft.world.entity.player.Player $$12 = $$0.getPlayer();
+        net.minecraft.world.level.Level $$2 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$4 = $$2.getBlockState($$3 = $$0.getClickedPos());
+        if (net.minecraft.world.level.block.CampfireBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$4) || net.minecraft.world.level.block.CandleBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$4) || net.minecraft.world.level.block.CandleCakeBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$4)) {
+            $$2.playSound($$12, $$3, net.minecraft.sounds.SoundEvents.FLINTANDSTEEL_USE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, $$2.getRandom().nextFloat() * 0.4f + 0.8f);
+            $$2.setBlock($$3, (net.minecraft.world.level.block.state.BlockState)$$4.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.state.properties.BlockStateProperties.LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)), 11);
+            $$2.gameEvent((net.minecraft.world.entity.Entity)$$12, net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, $$3);
+            if ($$12 != null) {
+                $$0.getItemInHand().hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)$$12, $$1 -> $$1.broadcastBreakEvent($$0.getHand()));
             }
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide());
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$2.isClientSide());
         }
-        net.minecraft.core.BlockPos blockPos2 = blockPos.relative(context.getClickedFace());
-        if (net.minecraft.world.level.block.BaseFireBlock.canBePlacedAt((net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)blockPos2, (net.minecraft.core.Direction)context.getHorizontalDirection())) {
-            level.playSound(player, blockPos2, net.minecraft.sounds.SoundEvents.FLINTANDSTEEL_USE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, level.getRandom().nextFloat() * 0.4f + 0.8f);
-            net.minecraft.world.level.block.state.BlockState blockState2 = net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)blockPos2);
-            level.setBlock(blockPos2, blockState2, 11);
-            level.gameEvent((net.minecraft.world.entity.Entity)player, net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, blockPos);
-            net.minecraft.world.item.ItemStack itemStack = context.getItemInHand();
-            if (player instanceof net.minecraft.server.level.ServerPlayer) {
-                net.minecraft.advancements.CriteriaTriggers.PLACED_BLOCK.trigger((net.minecraft.server.level.ServerPlayer)player, blockPos2, itemStack);
-                itemStack.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)player, contextPlayer -> contextPlayer.broadcastBreakEvent(context.getHand()));
+        net.minecraft.core.BlockPos $$5 = $$3.relative($$0.getClickedFace());
+        if (net.minecraft.world.level.block.BaseFireBlock.canBePlacedAt((net.minecraft.world.level.Level)$$2, (net.minecraft.core.BlockPos)$$5, (net.minecraft.core.Direction)$$0.getHorizontalDirection())) {
+            $$2.playSound($$12, $$5, net.minecraft.sounds.SoundEvents.FLINTANDSTEEL_USE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, $$2.getRandom().nextFloat() * 0.4f + 0.8f);
+            net.minecraft.world.level.block.state.BlockState $$6 = net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)$$2, (net.minecraft.core.BlockPos)$$5);
+            $$2.setBlock($$5, $$6, 11);
+            $$2.gameEvent((net.minecraft.world.entity.Entity)$$12, net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, $$3);
+            net.minecraft.world.item.ItemStack $$7 = $$0.getItemInHand();
+            if ($$12 instanceof net.minecraft.server.level.ServerPlayer) {
+                net.minecraft.advancements.CriteriaTriggers.PLACED_BLOCK.trigger((net.minecraft.server.level.ServerPlayer)$$12, $$5, $$7);
+                $$7.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)$$12, $$1 -> $$1.broadcastBreakEvent($$0.getHand()));
             }
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide());
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$2.isClientSide());
         }
         return net.minecraft.world.InteractionResult.FAIL;
     }