--- net/minecraft/world/item/GameMasterBlockItem.java
+++ net/minecraft/world/item/GameMasterBlockItem.java
@@ -11,9 +11,9 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.block.state.BlockState getPlacementState(net.minecraft.world.item.context.BlockPlaceContext context) {
-        net.minecraft.world.entity.player.Player player = context.getPlayer();
-        return player == null || player.canUseGameMasterBlocks() ? super.getPlacementState(context) : null;
+    public net.minecraft.world.level.block.state.BlockState getPlacementState(net.minecraft.world.item.context.BlockPlaceContext $$0) {
+        net.minecraft.world.entity.player.Player $$1 = $$0.getPlayer();
+        return $$1 == null || $$1.canUseGameMasterBlocks() ? super.getPlacementState($$0) : null;
     }
 }
 