--- net/minecraft/world/item/HangingSignItem.java
+++ net/minecraft/world/item/HangingSignItem.java
@@ -1,0 +1,32 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.HangingSignItem
+extends net.minecraft.world.item.StandingAndWallBlockItem
+ {
+    public net.minecraft.world.item.HangingSignItem(net.minecraft.world.level.block.Block $$0, net.minecraft.world.level.block.Block $$1, net.minecraft.world.item.Item$Properties $$2) {
+        super($$0, $$1, $$2, net.minecraft.core.Direction.UP);
+    }
+
+    public boolean canPlace(net.minecraft.world.level.LevelReader $$0, net.minecraft.world.level.block.state.BlockState $$1, net.minecraft.core.BlockPos $$2) {
+        net.minecraft.world.level.block.WallHangingSignBlock $$3;
+        net.minecraft.world.level.block.Block block = $$1.getBlock();
+        if (block instanceof net.minecraft.world.level.block.WallHangingSignBlock && !($$3 = (net.minecraft.world.level.block.WallHangingSignBlock)block).canPlace($$1, $$0, $$2)) {
+            return false;
+        }
+        return super.canPlace($$0, $$1, $$2);
+    }
+
+    public boolean updateCustomBlockEntityTag(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.Level $$1, @javax.annotation.Nullable net.minecraft.world.entity.player.Player $$2, net.minecraft.world.item.ItemStack $$3, net.minecraft.world.level.block.state.BlockState $$4) {
+        net.minecraft.world.level.block.entity.BlockEntity blockEntity;
+        boolean $$5 = super.updateCustomBlockEntityTag($$0, $$1, $$2, $$3, $$4);
+        if (!$$1.isClientSide && !$$5 && $$2 != null && (blockEntity = $$1.getBlockEntity($$0)) instanceof net.minecraft.world.level.block.entity.SignBlockEntity) {
+            net.minecraft.world.level.block.entity.SignBlockEntity $$6 = (net.minecraft.world.level.block.entity.SignBlockEntity)blockEntity;
+            $$2.openTextEdit($$6);
+        }
+        return $$5;
+    }
+}
+