--- net/minecraft/world/item/HoeItem.java
+++ net/minecraft/world/item/HoeItem.java
@@ -8,29 +8,29 @@
  {
     public static java.util.Map<net.minecraft.world.level.block.Block, com.mojang.datafixers.util.Pair<java.util.function.Predicate<net.minecraft.world.item.context.UseOnContext>, java.util.function.Consumer<net.minecraft.world.item.context.UseOnContext>>> TILLABLES;
 
-    public net.minecraft.world.item.HoeItem(net.minecraft.world.item.Tier tier, int attackDamageModifier, float attackSpeedModifier, net.minecraft.world.item.Item$Properties properties) {
-        super((float)attackDamageModifier, attackSpeedModifier, tier, net.minecraft.tags.BlockTags.MINEABLE_WITH_HOE, properties);
+    public net.minecraft.world.item.HoeItem(net.minecraft.world.item.Tier $$0, int $$1, float $$2, net.minecraft.world.item.Item$Properties $$3) {
+        super((float)$$1, $$2, $$0, net.minecraft.tags.BlockTags.MINEABLE_WITH_HOE, $$3);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        com.mojang.datafixers.util.Pair pair = (com.mojang.datafixers.util.Pair)TILLABLES.get((java.lang.Object)level.getBlockState(blockPos = context.getClickedPos()).getBlock());
-        if (pair == null) {
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.core.BlockPos $$2;
+        net.minecraft.world.level.Level $$12 = $$0.getLevel();
+        com.mojang.datafixers.util.Pair $$3 = (com.mojang.datafixers.util.Pair)TILLABLES.get((java.lang.Object)$$12.getBlockState($$2 = $$0.getClickedPos()).getBlock());
+        if ($$3 == null) {
             return net.minecraft.world.InteractionResult.PASS;
         }
-        java.util.function.Predicate predicate = (java.util.function.Predicate)pair.getFirst();
-        java.util.function.Consumer consumer = (java.util.function.Consumer)pair.getSecond();
-        if (predicate.test((java.lang.Object)context)) {
-            net.minecraft.world.entity.player.Player player = context.getPlayer();
-            level.playSound(player, blockPos, net.minecraft.sounds.SoundEvents.HOE_TILL, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
-            if (!level.isClientSide) {
-                consumer.accept((java.lang.Object)context);
-                if (player != null) {
-                    context.getItemInHand().hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)player, $$1 -> $$1.broadcastBreakEvent(context.getHand()));
+        java.util.function.Predicate $$4 = (java.util.function.Predicate)$$3.getFirst();
+        java.util.function.Consumer $$5 = (java.util.function.Consumer)$$3.getSecond();
+        if ($$4.test((java.lang.Object)$$0)) {
+            net.minecraft.world.entity.player.Player $$6 = $$0.getPlayer();
+            $$12.playSound($$6, $$2, net.minecraft.sounds.SoundEvents.HOE_TILL, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
+            if (!$$12.isClientSide) {
+                $$5.accept((java.lang.Object)$$0);
+                if ($$6 != null) {
+                    $$0.getItemInHand().hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)$$6, $$1 -> $$1.broadcastBreakEvent($$0.getHand()));
                 }
             }
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$12.isClientSide);
         }
         return net.minecraft.world.InteractionResult.PASS;
     }