--- net/minecraft/world/item/HoneyBottleItem.java
+++ net/minecraft/world/item/HoneyBottleItem.java
@@ -12,35 +12,34 @@
         super($$0);
     }
 
-    public net.minecraft.world.item.ItemStack finishUsingItem(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity entityLiving) {
-        net.minecraft.server.level.ServerPlayer serverPlayer;
-        super.finishUsingItem(stack, level, entityLiving);
-        if (entityLiving instanceof net.minecraft.server.level.ServerPlayer) {
-            serverPlayer = (net.minecraft.server.level.ServerPlayer)entityLiving;
-            net.minecraft.advancements.CriteriaTriggers.CONSUME_ITEM.trigger(serverPlayer, stack);
-            serverPlayer.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+    public net.minecraft.world.item.ItemStack finishUsingItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.Level $$1, net.minecraft.world.entity.LivingEntity $$2) {
+        super.finishUsingItem($$0, $$1, $$2);
+        if ($$2 instanceof net.minecraft.server.level.ServerPlayer) {
+            net.minecraft.server.level.ServerPlayer $$3 = (net.minecraft.server.level.ServerPlayer)$$2;
+            net.minecraft.advancements.CriteriaTriggers.CONSUME_ITEM.trigger($$3, $$0);
+            $$3.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
         }
-        if (!level.isClientSide) {
-            entityLiving.removeEffect(net.minecraft.world.effect.MobEffects.POISON);
+        if (!$$1.isClientSide) {
+            $$2.removeEffect(net.minecraft.world.effect.MobEffects.POISON);
         }
-        if (stack.isEmpty()) {
+        if ($$0.isEmpty()) {
             return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GLASS_BOTTLE);
         }
-        if (entityLiving instanceof net.minecraft.world.entity.player.Player && !((net.minecraft.world.entity.player.Player)entityLiving).getAbilities().instabuild) {
-            serverPlayer = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GLASS_BOTTLE);
-            net.minecraft.world.entity.player.Player player = (net.minecraft.world.entity.player.Player)entityLiving;
-            if (!player.getInventory().add((net.minecraft.world.item.ItemStack)serverPlayer)) {
-                player.drop((net.minecraft.world.item.ItemStack)serverPlayer, false);
+        if ($$2 instanceof net.minecraft.world.entity.player.Player && !((net.minecraft.world.entity.player.Player)$$2).getAbilities().instabuild) {
+            net.minecraft.world.item.ItemStack $$4 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GLASS_BOTTLE);
+            net.minecraft.world.entity.player.Player $$5 = (net.minecraft.world.entity.player.Player)$$2;
+            if (!$$5.getInventory().add($$4)) {
+                $$5.drop($$4, false);
             }
         }
-        return stack;
+        return $$0;
     }
 
-    public int getUseDuration(net.minecraft.world.item.ItemStack stack) {
+    public int getUseDuration(net.minecraft.world.item.ItemStack $$0) {
         return 40;
     }
 
-    public net.minecraft.world.item.UseAnim getUseAnimation(net.minecraft.world.item.ItemStack stack) {
+    public net.minecraft.world.item.UseAnim getUseAnimation(net.minecraft.world.item.ItemStack $$0) {
         return net.minecraft.world.item.UseAnim.DRINK;
     }
 
@@ -52,8 +51,8 @@
         return net.minecraft.sounds.SoundEvents.HONEY_DRINK;
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        return net.minecraft.world.item.ItemUtils.startUsingInstantly((net.minecraft.world.level.Level)level, (net.minecraft.world.entity.player.Player)player, (net.minecraft.world.InteractionHand)hand);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        return net.minecraft.world.item.ItemUtils.startUsingInstantly((net.minecraft.world.level.Level)$$0, (net.minecraft.world.entity.player.Player)$$1, (net.minecraft.world.InteractionHand)$$2);
     }
 }
 