--- net/minecraft/world/item/HorseArmorItem.java
+++ net/minecraft/world/item/HorseArmorItem.java
@@ -10,10 +10,10 @@
     public int protection;
     public java.lang.String texture;
 
-    public net.minecraft.world.item.HorseArmorItem(int protection, java.lang.String identifier, net.minecraft.world.item.Item$Properties properties) {
-        super(properties);
-        this.protection = protection;
-        this.texture = "textures/entity/horse/armor/horse_armor_" + identifier + ".png";
+    public net.minecraft.world.item.HorseArmorItem(int $$0, java.lang.String $$1, net.minecraft.world.item.Item$Properties $$2) {
+        super($$2);
+        this.protection = $$0;
+        this.texture = "textures/entity/horse/armor/horse_armor_" + $$1 + ".png";
     }
 
     public net.minecraft.resources.ResourceLocation getTexture() {