--- net/minecraft/world/item/ItemUtils.java
+++ net/minecraft/world/item/ItemUtils.java
@@ -13,28 +13,28 @@
         return net.minecraft.world.InteractionResultHolder.consume((java.lang.Object)$$1.getItemInHand($$2));
     }
 
-    public static net.minecraft.world.item.ItemStack createFilledResult(net.minecraft.world.item.ItemStack emptyStack, net.minecraft.world.entity.player.Player player, net.minecraft.world.item.ItemStack filledStack, boolean preventDuplicates) {
-        boolean bl = player.getAbilities().instabuild;
-        if (preventDuplicates && bl) {
-            if (!player.getInventory().contains(filledStack)) {
-                player.getInventory().add(filledStack);
+    public static net.minecraft.world.item.ItemStack createFilledResult(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.item.ItemStack $$2, boolean $$3) {
+        boolean $$4 = $$1.getAbilities().instabuild;
+        if ($$3 && $$4) {
+            if (!$$1.getInventory().contains($$2)) {
+                $$1.getInventory().add($$2);
             }
-            return emptyStack;
+            return $$0;
         }
-        if (!bl) {
-            emptyStack.shrink(1);
+        if (!$$4) {
+            $$0.shrink(1);
         }
-        if (emptyStack.isEmpty()) {
-            return filledStack;
+        if ($$0.isEmpty()) {
+            return $$2;
         }
-        if (!player.getInventory().add(filledStack)) {
-            player.drop(filledStack, false);
+        if (!$$1.getInventory().add($$2)) {
+            $$1.drop($$2, false);
         }
-        return emptyStack;
+        return $$0;
     }
 
-    public static net.minecraft.world.item.ItemStack createFilledResult(net.minecraft.world.item.ItemStack emptyStack, net.minecraft.world.entity.player.Player player, net.minecraft.world.item.ItemStack filledStack) {
-        return net.minecraft.world.item.ItemUtils.createFilledResult((net.minecraft.world.item.ItemStack)emptyStack, (net.minecraft.world.entity.player.Player)player, (net.minecraft.world.item.ItemStack)filledStack, (boolean)true);
+    public static net.minecraft.world.item.ItemStack createFilledResult(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.item.ItemStack $$2) {
+        return net.minecraft.world.item.ItemUtils.createFilledResult((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.entity.player.Player)$$1, (net.minecraft.world.item.ItemStack)$$2, (boolean)true);
     }
 
     public static void onContainerDestroyed(net.minecraft.world.entity.item.ItemEntity $$0, java.util.stream.Stream<net.minecraft.world.item.ItemStack> $$1) {