--- net/minecraft/world/item/ItemCooldowns.java
+++ net/minecraft/world/item/ItemCooldowns.java
@@ -12,16 +12,16 @@
         this.cooldowns = com.google.common.collect.Maps.newHashMap();
     }
 
-    public boolean isOnCooldown(net.minecraft.world.item.Item item) {
-        return this.getCooldownPercent(item, 0.0f) > 0.0f;
+    public boolean isOnCooldown(net.minecraft.world.item.Item $$0) {
+        return this.getCooldownPercent($$0, 0.0f) > 0.0f;
     }
 
-    public float getCooldownPercent(net.minecraft.world.item.Item item, float partialTicks) {
-        net.minecraft.world.item.ItemCooldowns$CooldownInstance cooldownInstance = (net.minecraft.world.item.ItemCooldowns$CooldownInstance)this.cooldowns.get((java.lang.Object)item);
-        if (cooldownInstance != null) {
-            float f = cooldownInstance.endTime - cooldownInstance.startTime;
-            float f2 = (float)cooldownInstance.endTime - ((float)this.tickCount + partialTicks);
-            return net.minecraft.util.Mth.clamp((float)(f2 / f), (float)0.0f, (float)1.0f);
+    public float getCooldownPercent(net.minecraft.world.item.Item $$0, float $$1) {
+        net.minecraft.world.item.ItemCooldowns$CooldownInstance $$2 = (net.minecraft.world.item.ItemCooldowns$CooldownInstance)this.cooldowns.get((java.lang.Object)$$0);
+        if ($$2 != null) {
+            float $$3 = $$2.endTime - $$2.startTime;
+            float $$4 = (float)$$2.endTime - ((float)this.tickCount + $$1);
+            return net.minecraft.util.Mth.clamp((float)($$4 / $$3), (float)0.0f, (float)1.0f);
         }
         return 0.0f;
     }
@@ -39,20 +39,20 @@
         }
     }
 
-    public void addCooldown(net.minecraft.world.item.Item item, int ticks) {
-        this.cooldowns.put((java.lang.Object)item, (java.lang.Object)new net.minecraft.world.item.ItemCooldowns$CooldownInstance(this.tickCount, this.tickCount + ticks));
-        this.onCooldownStarted(item, ticks);
+    public void addCooldown(net.minecraft.world.item.Item $$0, int $$1) {
+        this.cooldowns.put((java.lang.Object)$$0, (java.lang.Object)new net.minecraft.world.item.ItemCooldowns$CooldownInstance(this.tickCount, this.tickCount + $$1));
+        this.onCooldownStarted($$0, $$1);
     }
 
-    public void removeCooldown(net.minecraft.world.item.Item item) {
-        this.cooldowns.remove((java.lang.Object)item);
-        this.onCooldownEnded(item);
+    public void removeCooldown(net.minecraft.world.item.Item $$0) {
+        this.cooldowns.remove((java.lang.Object)$$0);
+        this.onCooldownEnded($$0);
     }
 
-    public void onCooldownStarted(net.minecraft.world.item.Item item, int ticks) {
+    public void onCooldownStarted(net.minecraft.world.item.Item $$0, int $$1) {
     }
 
-    public void onCooldownEnded(net.minecraft.world.item.Item item) {
+    public void onCooldownEnded(net.minecraft.world.item.Item $$0) {
     }
 }
 