--- net/minecraft/world/item/ItemStackLinkedSet$1.java
+++ net/minecraft/world/item/ItemStackLinkedSet$1.java
@@ -1,0 +1,31 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.item;
+
+public class net.minecraft.world.item.ItemStackLinkedSet$1
+implements it.unimi.dsi.fastutil.Hash$Strategy<net.minecraft.world.item.ItemStack>
+ {
+    public net.minecraft.world.item.ItemStackLinkedSet$1() {
+    }
+
+    public int hashCode(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0 != null) {
+            return java.util.Objects.hash((java.lang.Object[])new java.lang.Object[]{$$0.getItem(), $$0.getTag()});
+        }
+        return 0;
+    }
+
+    public boolean equals(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        return $$0 == $$1 || $$0 != null && $$1 != null && net.minecraft.world.item.ItemStack.matches((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.item.ItemStack)$$1);
+    }
+
+    public /* synthetic */ boolean equals(java.lang.Object object, java.lang.Object object2) {
+        return this.equals((net.minecraft.world.item.ItemStack)object, (net.minecraft.world.item.ItemStack)object2);
+    }
+
+    public /* synthetic */ int hashCode(java.lang.Object object) {
+        return this.hashCode((net.minecraft.world.item.ItemStack)object);
+    }
+}
+