--- net/minecraft/world/item/ItemFrameItem.java
+++ net/minecraft/world/item/ItemFrameItem.java
@@ -10,8 +10,8 @@
         super($$0, $$1);
     }
 
-    public boolean mayPlace(net.minecraft.world.entity.player.Player player, net.minecraft.core.Direction direction, net.minecraft.world.item.ItemStack itemStack, net.minecraft.core.BlockPos pos) {
-        return !player.level.isOutsideBuildHeight(pos) && player.mayUseItemAt(pos, direction, itemStack);
+    public boolean mayPlace(net.minecraft.world.entity.player.Player $$0, net.minecraft.core.Direction $$1, net.minecraft.world.item.ItemStack $$2, net.minecraft.core.BlockPos $$3) {
+        return !$$0.level.isOutsideBuildHeight($$3) && $$0.mayUseItemAt($$3, $$1, $$2);
     }
 }
 