--- net/minecraft/world/item/LeadItem.java
+++ net/minecraft/world/item/LeadItem.java
@@ -10,39 +10,39 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos = context.getClickedPos());
-        if (blockState.is(net.minecraft.tags.BlockTags.FENCES)) {
-            net.minecraft.world.entity.player.Player player = context.getPlayer();
-            if (!level.isClientSide && player != null) {
-                net.minecraft.world.item.LeadItem.bindPlayerMobs((net.minecraft.world.entity.player.Player)player, (net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)blockPos);
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.core.BlockPos $$2;
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$1.getBlockState($$2 = $$0.getClickedPos());
+        if ($$3.is(net.minecraft.tags.BlockTags.FENCES)) {
+            net.minecraft.world.entity.player.Player $$4 = $$0.getPlayer();
+            if (!$$1.isClientSide && $$4 != null) {
+                net.minecraft.world.item.LeadItem.bindPlayerMobs((net.minecraft.world.entity.player.Player)$$4, (net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2);
             }
-            level.gameEvent(net.minecraft.world.level.gameevent.GameEvent.BLOCK_ATTACH, blockPos, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)player));
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+            $$1.gameEvent(net.minecraft.world.level.gameevent.GameEvent.BLOCK_ATTACH, $$2, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)$$4));
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
         }
         return net.minecraft.world.InteractionResult.PASS;
     }
 
-    public static net.minecraft.world.InteractionResult bindPlayerMobs(net.minecraft.world.entity.player.Player player, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.entity.decoration.LeashFenceKnotEntity leashFenceKnotEntity = null;
-        boolean bl = false;
-        double d = 7.0;
-        int n = pos.getX();
-        int n2 = pos.getY();
-        int n3 = pos.getZ();
-        java.util.List list = level.getEntitiesOfClass(net.minecraft.world.entity.Mob.class, new net.minecraft.world.phys.AABB((double)n - 7.0, (double)n2 - 7.0, (double)n3 - 7.0, (double)n + 7.0, (double)n2 + 7.0, (double)n3 + 7.0));
-        for (net.minecraft.world.entity.Mob mob : list) {
-            if (mob.getLeashHolder() != player) continue;
-            if (leashFenceKnotEntity == null) {
-                leashFenceKnotEntity = net.minecraft.world.entity.decoration.LeashFenceKnotEntity.getOrCreateKnot((net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)pos);
-                leashFenceKnotEntity.playPlacementSound();
+    public static net.minecraft.world.InteractionResult bindPlayerMobs(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2) {
+        net.minecraft.world.entity.decoration.LeashFenceKnotEntity $$3 = null;
+        boolean $$4 = false;
+        double $$5 = 7.0;
+        int $$6 = $$2.getX();
+        int $$7 = $$2.getY();
+        int $$8 = $$2.getZ();
+        java.util.List $$9 = $$1.getEntitiesOfClass(net.minecraft.world.entity.Mob.class, new net.minecraft.world.phys.AABB((double)$$6 - 7.0, (double)$$7 - 7.0, (double)$$8 - 7.0, (double)$$6 + 7.0, (double)$$7 + 7.0, (double)$$8 + 7.0));
+        for (net.minecraft.world.entity.Mob $$10 : $$9) {
+            if ($$10.getLeashHolder() != $$0) continue;
+            if ($$3 == null) {
+                $$3 = net.minecraft.world.entity.decoration.LeashFenceKnotEntity.getOrCreateKnot((net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2);
+                $$3.playPlacementSound();
             }
-            mob.setLeashedTo((net.minecraft.world.entity.Entity)leashFenceKnotEntity, true);
-            bl = true;
+            $$10.setLeashedTo((net.minecraft.world.entity.Entity)$$3, true);
+            $$4 = true;
         }
-        return bl ? net.minecraft.world.InteractionResult.SUCCESS : net.minecraft.world.InteractionResult.PASS;
+        return $$4 ? net.minecraft.world.InteractionResult.SUCCESS : net.minecraft.world.InteractionResult.PASS;
     }
 }
 