--- net/minecraft/world/item/MinecartItem$1.java
+++ net/minecraft/world/item/MinecartItem$1.java
@@ -12,39 +12,49 @@
         this.defaultDispenseItemBehavior = new net.minecraft.core.dispenser.DefaultDispenseItemBehavior();
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.world.level.block.state.BlockState blockState;
-        double d;
-        net.minecraft.world.level.block.state.properties.RailShape railShape;
-        net.minecraft.core.Direction direction = (net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        double d2 = source.x() + (double)direction.getStepX() * 1.125;
-        double d3 = java.lang.Math.floor((double)source.y()) + (double)direction.getStepY();
-        double d4 = source.z() + (double)direction.getStepZ() * 1.125;
-        net.minecraft.core.BlockPos blockPos = source.getPos().relative(direction);
-        net.minecraft.world.level.block.state.BlockState blockState2 = serverLevel.getBlockState(blockPos);
-        net.minecraft.world.level.block.state.properties.RailShape railShape2 = railShape = blockState2.getBlock() instanceof net.minecraft.world.level.block.BaseRailBlock ? (net.minecraft.world.level.block.state.properties.RailShape)blockState2.getValue(((net.minecraft.world.level.block.BaseRailBlock)blockState2.getBlock()).getShapeProperty()) : net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
-        if (blockState2.is(net.minecraft.tags.BlockTags.RAILS)) {
-            d = railShape.isAscending() ? 0.6 : 0.1;
-        } else if (blockState2.isAir() && serverLevel.getBlockState(blockPos.below()).is(net.minecraft.tags.BlockTags.RAILS)) {
-            net.minecraft.world.level.block.state.properties.RailShape railShape3;
-            blockState = serverLevel.getBlockState(blockPos.below());
-            net.minecraft.world.level.block.state.properties.RailShape railShape4 = railShape3 = blockState.getBlock() instanceof net.minecraft.world.level.block.BaseRailBlock ? (net.minecraft.world.level.block.state.properties.RailShape)blockState.getValue(((net.minecraft.world.level.block.BaseRailBlock)blockState.getBlock()).getShapeProperty()) : net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
-            d = direction == net.minecraft.core.Direction.DOWN || !railShape3.isAscending() ? -0.9 : -0.4;
+    /*
+     * WARNING - void declaration
+     */
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        void $$16;
+        net.minecraft.world.level.block.state.properties.RailShape $$9;
+        net.minecraft.core.Direction $$2 = (net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
+        net.minecraft.server.level.ServerLevel $$3 = $$0.getLevel();
+        double $$4 = $$0.x() + (double)$$2.getStepX() * 1.125;
+        double $$5 = java.lang.Math.floor((double)$$0.y()) + (double)$$2.getStepY();
+        double $$6 = $$0.z() + (double)$$2.getStepZ() * 1.125;
+        net.minecraft.core.BlockPos $$7 = $$0.getPos().relative($$2);
+        net.minecraft.world.level.block.state.BlockState $$8 = $$3.getBlockState($$7);
+        net.minecraft.world.level.block.state.properties.RailShape railShape = $$9 = $$8.getBlock() instanceof net.minecraft.world.level.block.BaseRailBlock ? (net.minecraft.world.level.block.state.properties.RailShape)$$8.getValue(((net.minecraft.world.level.block.BaseRailBlock)$$8.getBlock()).getShapeProperty()) : net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+        if ($$8.is(net.minecraft.tags.BlockTags.RAILS)) {
+            if ($$9.isAscending()) {
+                double $$10 = 0.6;
+            } else {
+                double $$11 = 0.1;
+            }
+        } else if ($$8.isAir() && $$3.getBlockState($$7.below()).is(net.minecraft.tags.BlockTags.RAILS)) {
+            net.minecraft.world.level.block.state.properties.RailShape $$13;
+            net.minecraft.world.level.block.state.BlockState $$12 = $$3.getBlockState($$7.below());
+            net.minecraft.world.level.block.state.properties.RailShape railShape2 = $$13 = $$12.getBlock() instanceof net.minecraft.world.level.block.BaseRailBlock ? (net.minecraft.world.level.block.state.properties.RailShape)$$12.getValue(((net.minecraft.world.level.block.BaseRailBlock)$$12.getBlock()).getShapeProperty()) : net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+            if ($$2 == net.minecraft.core.Direction.DOWN || !$$13.isAscending()) {
+                double $$14 = -0.9;
+            } else {
+                double $$15 = -0.4;
+            }
         } else {
-            return this.defaultDispenseItemBehavior.dispense(source, stack);
+            return this.defaultDispenseItemBehavior.dispense($$0, $$1);
         }
-        blockState = net.minecraft.world.entity.vehicle.AbstractMinecart.createMinecart((net.minecraft.world.level.Level)serverLevel, (double)d2, (double)(d3 + d), (double)d4, (net.minecraft.world.entity.vehicle.AbstractMinecart$Type)((net.minecraft.world.item.MinecartItem)stack.getItem()).type);
-        if (stack.hasCustomHoverName()) {
-            blockState.setCustomName(stack.getHoverName());
+        net.minecraft.world.entity.vehicle.AbstractMinecart $$17 = net.minecraft.world.entity.vehicle.AbstractMinecart.createMinecart((net.minecraft.world.level.Level)$$3, (double)$$4, (double)($$5 + $$16), (double)$$6, (net.minecraft.world.entity.vehicle.AbstractMinecart$Type)((net.minecraft.world.item.MinecartItem)$$1.getItem()).type);
+        if ($$1.hasCustomHoverName()) {
+            $$17.setCustomName($$1.getHoverName());
         }
-        serverLevel.addFreshEntity((net.minecraft.world.entity.Entity)blockState);
-        stack.shrink(1);
-        return stack;
+        $$3.addFreshEntity((net.minecraft.world.entity.Entity)$$17);
+        $$1.shrink(1);
+        return $$1;
     }
 
-    public void playSound(net.minecraft.core.BlockSource source) {
-        source.getLevel().levelEvent(1000, source.getPos(), 0);
+    public void playSound(net.minecraft.core.BlockSource $$0) {
+        $$0.getLevel().levelEvent(1000, $$0.getPos(), 0);
     }
 }
 