--- net/minecraft/world/item/MinecartItem.java
+++ net/minecraft/world/item/MinecartItem.java
@@ -9,35 +9,35 @@
     public static net.minecraft.core.dispenser.DispenseItemBehavior DISPENSE_ITEM_BEHAVIOR;
     public net.minecraft.world.entity.vehicle.AbstractMinecart$Type type;
 
-    public net.minecraft.world.item.MinecartItem(net.minecraft.world.entity.vehicle.AbstractMinecart$Type type, net.minecraft.world.item.Item$Properties properties) {
-        super(properties);
-        this.type = type;
+    public net.minecraft.world.item.MinecartItem(net.minecraft.world.entity.vehicle.AbstractMinecart$Type $$0, net.minecraft.world.item.Item$Properties $$1) {
+        super($$1);
+        this.type = $$0;
         net.minecraft.world.level.block.DispenserBlock.registerBehavior((net.minecraft.world.level.ItemLike)this, (net.minecraft.core.dispenser.DispenseItemBehavior)DISPENSE_ITEM_BEHAVIOR);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos = context.getClickedPos());
-        if (!blockState.is(net.minecraft.tags.BlockTags.RAILS)) {
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.core.BlockPos $$2;
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$1.getBlockState($$2 = $$0.getClickedPos());
+        if (!$$3.is(net.minecraft.tags.BlockTags.RAILS)) {
             return net.minecraft.world.InteractionResult.FAIL;
         }
-        net.minecraft.world.item.ItemStack itemStack = context.getItemInHand();
-        if (!level.isClientSide) {
-            net.minecraft.world.level.block.state.properties.RailShape railShape = blockState.getBlock() instanceof net.minecraft.world.level.block.BaseRailBlock ? (net.minecraft.world.level.block.state.properties.RailShape)blockState.getValue(((net.minecraft.world.level.block.BaseRailBlock)blockState.getBlock()).getShapeProperty()) : net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
-            double d = 0.0;
-            if (railShape.isAscending()) {
-                d = 0.5;
+        net.minecraft.world.item.ItemStack $$4 = $$0.getItemInHand();
+        if (!$$1.isClientSide) {
+            net.minecraft.world.level.block.state.properties.RailShape $$5 = $$3.getBlock() instanceof net.minecraft.world.level.block.BaseRailBlock ? (net.minecraft.world.level.block.state.properties.RailShape)$$3.getValue(((net.minecraft.world.level.block.BaseRailBlock)$$3.getBlock()).getShapeProperty()) : net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+            double $$6 = 0.0;
+            if ($$5.isAscending()) {
+                $$6 = 0.5;
             }
-            net.minecraft.world.entity.vehicle.AbstractMinecart abstractMinecart = net.minecraft.world.entity.vehicle.AbstractMinecart.createMinecart((net.minecraft.world.level.Level)level, (double)((double)blockPos.getX() + 0.5), (double)((double)blockPos.getY() + 0.0625 + d), (double)((double)blockPos.getZ() + 0.5), (net.minecraft.world.entity.vehicle.AbstractMinecart$Type)this.type);
-            if (itemStack.hasCustomHoverName()) {
-                abstractMinecart.setCustomName(itemStack.getHoverName());
+            net.minecraft.world.entity.vehicle.AbstractMinecart $$7 = net.minecraft.world.entity.vehicle.AbstractMinecart.createMinecart((net.minecraft.world.level.Level)$$1, (double)((double)$$2.getX() + 0.5), (double)((double)$$2.getY() + 0.0625 + $$6), (double)((double)$$2.getZ() + 0.5), (net.minecraft.world.entity.vehicle.AbstractMinecart$Type)this.type);
+            if ($$4.hasCustomHoverName()) {
+                $$7.setCustomName($$4.getHoverName());
             }
-            level.addFreshEntity((net.minecraft.world.entity.Entity)abstractMinecart);
-            level.gameEvent(net.minecraft.world.level.gameevent.GameEvent.ENTITY_PLACE, blockPos, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)context.getPlayer(), (net.minecraft.world.level.block.state.BlockState)level.getBlockState(blockPos.below())));
+            $$1.addFreshEntity((net.minecraft.world.entity.Entity)$$7);
+            $$1.gameEvent(net.minecraft.world.level.gameevent.GameEvent.ENTITY_PLACE, $$2, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)$$0.getPlayer(), (net.minecraft.world.level.block.state.BlockState)$$1.getBlockState($$2.below())));
         }
-        itemStack.shrink(1);
-        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+        $$4.shrink(1);
+        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
     }
 
     static {