--- net/minecraft/world/item/NameTagItem.java
+++ net/minecraft/world/item/NameTagItem.java
@@ -10,16 +10,16 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult interactLivingEntity(net.minecraft.world.item.ItemStack stack, net.minecraft.world.entity.player.Player player, net.minecraft.world.entity.LivingEntity target, net.minecraft.world.InteractionHand hand) {
-        if (stack.hasCustomHoverName() && !(target instanceof net.minecraft.world.entity.player.Player)) {
-            if (!player.level.isClientSide && target.isAlive()) {
-                target.setCustomName(stack.getHoverName());
-                if (target instanceof net.minecraft.world.entity.Mob) {
-                    ((net.minecraft.world.entity.Mob)target).setPersistenceRequired();
+    public net.minecraft.world.InteractionResult interactLivingEntity(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.entity.LivingEntity $$2, net.minecraft.world.InteractionHand $$3) {
+        if ($$0.hasCustomHoverName() && !($$2 instanceof net.minecraft.world.entity.player.Player)) {
+            if (!$$1.level.isClientSide && $$2.isAlive()) {
+                $$2.setCustomName($$0.getHoverName());
+                if ($$2 instanceof net.minecraft.world.entity.Mob) {
+                    ((net.minecraft.world.entity.Mob)$$2).setPersistenceRequired();
                 }
-                stack.shrink(1);
+                $$0.shrink(1);
             }
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)player.level.isClientSide);
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.level.isClientSide);
         }
         return net.minecraft.world.InteractionResult.PASS;
     }