--- net/minecraft/world/item/PickaxeItem.java
+++ net/minecraft/world/item/PickaxeItem.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.item.PickaxeItem
 extends net.minecraft.world.item.DiggerItem
  {
-    public net.minecraft.world.item.PickaxeItem(net.minecraft.world.item.Tier tier, int attackDamageModifier, float attackSpeedModifier, net.minecraft.world.item.Item$Properties properties) {
-        super((float)attackDamageModifier, attackSpeedModifier, tier, net.minecraft.tags.BlockTags.MINEABLE_WITH_PICKAXE, properties);
+    public net.minecraft.world.item.PickaxeItem(net.minecraft.world.item.Tier $$0, int $$1, float $$2, net.minecraft.world.item.Item$Properties $$3) {
+        super((float)$$1, $$2, $$0, net.minecraft.tags.BlockTags.MINEABLE_WITH_PICKAXE, $$3);
     }
 }
 