--- net/minecraft/world/item/PlayerHeadItem.java
+++ net/minecraft/world/item/PlayerHeadItem.java
@@ -9,24 +9,24 @@
     public static java.lang.String TAG_SKULL_OWNER = "SkullOwner";
 
     public net.minecraft.world.item.PlayerHeadItem(net.minecraft.world.level.block.Block $$0, net.minecraft.world.level.block.Block $$1, net.minecraft.world.item.Item$Properties $$2) {
-        super($$0, $$1, $$2);
+        super($$0, $$1, $$2, net.minecraft.core.Direction.DOWN);
     }
 
-    public net.minecraft.network.chat.Component getName(net.minecraft.world.item.ItemStack stack) {
-        if (stack.is(net.minecraft.world.item.Items.PLAYER_HEAD) && stack.hasTag()) {
-            net.minecraft.nbt.CompoundTag compoundTag;
-            java.lang.String string = null;
-            net.minecraft.nbt.CompoundTag compoundTag2 = stack.getTag();
-            if (compoundTag2.contains("SkullOwner", 8)) {
-                string = compoundTag2.getString("SkullOwner");
-            } else if (compoundTag2.contains("SkullOwner", 10) && (compoundTag = compoundTag2.getCompound("SkullOwner")).contains("Name", 8)) {
-                string = compoundTag.getString("Name");
+    public net.minecraft.network.chat.Component getName(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0.is(net.minecraft.world.item.Items.PLAYER_HEAD) && $$0.hasTag()) {
+            net.minecraft.nbt.CompoundTag $$3;
+            java.lang.String $$1 = null;
+            net.minecraft.nbt.CompoundTag $$2 = $$0.getTag();
+            if ($$2.contains("SkullOwner", 8)) {
+                $$1 = $$2.getString("SkullOwner");
+            } else if ($$2.contains("SkullOwner", 10) && ($$3 = $$2.getCompound("SkullOwner")).contains("Name", 8)) {
+                $$1 = $$3.getString("Name");
             }
-            if (string != null) {
-                return net.minecraft.network.chat.Component.translatable((java.lang.String)(this.getDescriptionId() + ".named"), (java.lang.Object[])new java.lang.Object[]{string});
+            if ($$1 != null) {
+                return net.minecraft.network.chat.Component.translatable((java.lang.String)(this.getDescriptionId() + ".named"), (java.lang.Object[])new java.lang.Object[]{$$1});
             }
         }
-        return super.getName(stack);
+        return super.getName($$0);
     }
 
     public void verifyTagAfterLoad(net.minecraft.nbt.CompoundTag $$0) {