--- net/minecraft/world/item/Rarity.java
+++ net/minecraft/world/item/Rarity.java
@@ -21,9 +21,9 @@
         return (net.minecraft.world.item.Rarity)java.lang.Enum.valueOf(net.minecraft.world.item.Rarity.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.item.Rarity(net.minecraft.ChatFormatting color) {
+    public net.minecraft.world.item.Rarity(net.minecraft.ChatFormatting $$0) {
         super(string, n);
-        this.color = color;
+        this.color = $$0;
     }
 
     public static /* synthetic */ net.minecraft.world.item.Rarity[] $values() {