--- net/minecraft/world/item/SaddleItem.java
+++ net/minecraft/world/item/SaddleItem.java
@@ -10,15 +10,15 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult interactLivingEntity(net.minecraft.world.item.ItemStack stack, net.minecraft.world.entity.player.Player player, net.minecraft.world.entity.LivingEntity target, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.entity.Saddleable saddleable;
-        if (target instanceof net.minecraft.world.entity.Saddleable && target.isAlive() && !(saddleable = (net.minecraft.world.entity.Saddleable)target).isSaddled() && saddleable.isSaddleable()) {
-            if (!player.level.isClientSide) {
-                saddleable.equipSaddle(net.minecraft.sounds.SoundSource.NEUTRAL);
-                target.level.gameEvent((net.minecraft.world.entity.Entity)target, net.minecraft.world.level.gameevent.GameEvent.EQUIP, target.position());
-                stack.shrink(1);
+    public net.minecraft.world.InteractionResult interactLivingEntity(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.entity.LivingEntity $$2, net.minecraft.world.InteractionHand $$3) {
+        net.minecraft.world.entity.Saddleable $$4;
+        if ($$2 instanceof net.minecraft.world.entity.Saddleable && $$2.isAlive() && !($$4 = (net.minecraft.world.entity.Saddleable)$$2).isSaddled() && $$4.isSaddleable()) {
+            if (!$$1.level.isClientSide) {
+                $$4.equipSaddle(net.minecraft.sounds.SoundSource.NEUTRAL);
+                $$2.level.gameEvent((net.minecraft.world.entity.Entity)$$2, net.minecraft.world.level.gameevent.GameEvent.EQUIP, $$2.position());
+                $$0.shrink(1);
             }
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)player.level.isClientSide);
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.level.isClientSide);
         }
         return net.minecraft.world.InteractionResult.PASS;
     }