--- net/minecraft/world/item/ShieldItem.java
+++ net/minecraft/world/item/ShieldItem.java
@@ -15,38 +15,38 @@
         net.minecraft.world.level.block.DispenserBlock.registerBehavior((net.minecraft.world.level.ItemLike)this, (net.minecraft.core.dispenser.DispenseItemBehavior)net.minecraft.world.item.ArmorItem.DISPENSE_ITEM_BEHAVIOR);
     }
 
-    public java.lang.String getDescriptionId(net.minecraft.world.item.ItemStack stack) {
-        if (net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack) != null) {
-            return this.getDescriptionId() + "." + net.minecraft.world.item.ShieldItem.getColor((net.minecraft.world.item.ItemStack)stack).getName();
+    public java.lang.String getDescriptionId(net.minecraft.world.item.ItemStack $$0) {
+        if (net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$0) != null) {
+            return this.getDescriptionId() + "." + net.minecraft.world.item.ShieldItem.getColor((net.minecraft.world.item.ItemStack)$$0).getName();
         }
-        return super.getDescriptionId(stack);
+        return super.getDescriptionId($$0);
     }
 
-    public void appendHoverText(net.minecraft.world.item.ItemStack stack, @javax.annotation.Nullable net.minecraft.world.level.Level level, java.util.List<net.minecraft.network.chat.Component> tooltip, net.minecraft.world.item.TooltipFlag flag) {
-        net.minecraft.world.item.BannerItem.appendHoverTextFromBannerBlockEntityTag((net.minecraft.world.item.ItemStack)stack, tooltip);
+    public void appendHoverText(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.world.level.Level $$1, java.util.List<net.minecraft.network.chat.Component> $$2, net.minecraft.world.item.TooltipFlag $$3) {
+        net.minecraft.world.item.BannerItem.appendHoverTextFromBannerBlockEntityTag((net.minecraft.world.item.ItemStack)$$0, $$2);
     }
 
-    public net.minecraft.world.item.UseAnim getUseAnimation(net.minecraft.world.item.ItemStack stack) {
+    public net.minecraft.world.item.UseAnim getUseAnimation(net.minecraft.world.item.ItemStack $$0) {
         return net.minecraft.world.item.UseAnim.BLOCK;
     }
 
-    public int getUseDuration(net.minecraft.world.item.ItemStack stack) {
+    public int getUseDuration(net.minecraft.world.item.ItemStack $$0) {
         return 72000;
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        player.startUsingItem(hand);
-        return net.minecraft.world.InteractionResultHolder.consume((java.lang.Object)itemStack);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+        $$1.startUsingItem($$2);
+        return net.minecraft.world.InteractionResultHolder.consume((java.lang.Object)$$3);
     }
 
-    public boolean isValidRepairItem(net.minecraft.world.item.ItemStack toRepair, net.minecraft.world.item.ItemStack repair) {
-        return repair.is(net.minecraft.tags.ItemTags.PLANKS) || super.isValidRepairItem(toRepair, repair);
+    public boolean isValidRepairItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        return $$1.is(net.minecraft.tags.ItemTags.PLANKS) || super.isValidRepairItem($$0, $$1);
     }
 
-    public static net.minecraft.world.item.DyeColor getColor(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack);
-        return compoundTag != null ? net.minecraft.world.item.DyeColor.byId((int)compoundTag.getInt("Base")) : net.minecraft.world.item.DyeColor.WHITE;
+    public static net.minecraft.world.item.DyeColor getColor(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$0);
+        return $$1 != null ? net.minecraft.world.item.DyeColor.byId((int)$$1.getInt("Base")) : net.minecraft.world.item.DyeColor.WHITE;
     }
 }
 