--- net/minecraft/world/item/SnowballItem.java
+++ net/minecraft/world/item/SnowballItem.java
@@ -10,20 +10,20 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        level.playSound(null, player.getX(), player.getY(), player.getZ(), net.minecraft.sounds.SoundEvents.SNOWBALL_THROW, net.minecraft.sounds.SoundSource.NEUTRAL, 0.5f, 0.4f / (level.getRandom().nextFloat() * 0.4f + 0.8f));
-        if (!level.isClientSide) {
-            net.minecraft.world.entity.projectile.Snowball snowball = new net.minecraft.world.entity.projectile.Snowball(level, (net.minecraft.world.entity.LivingEntity)player);
-            snowball.setItem(itemStack);
-            snowball.shootFromRotation((net.minecraft.world.entity.Entity)player, player.getXRot(), player.getYRot(), 0.0f, 1.5f, 1.0f);
-            level.addFreshEntity((net.minecraft.world.entity.Entity)snowball);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+        $$0.playSound(null, $$1.getX(), $$1.getY(), $$1.getZ(), net.minecraft.sounds.SoundEvents.SNOWBALL_THROW, net.minecraft.sounds.SoundSource.NEUTRAL, 0.5f, 0.4f / ($$0.getRandom().nextFloat() * 0.4f + 0.8f));
+        if (!$$0.isClientSide) {
+            net.minecraft.world.entity.projectile.Snowball $$4 = new net.minecraft.world.entity.projectile.Snowball($$0, (net.minecraft.world.entity.LivingEntity)$$1);
+            $$4.setItem($$3);
+            $$4.shootFromRotation((net.minecraft.world.entity.Entity)$$1, $$1.getXRot(), $$1.getYRot(), 0.0f, 1.5f, 1.0f);
+            $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$4);
         }
-        player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
-        if (!player.getAbilities().instabuild) {
-            itemStack.shrink(1);
+        $$1.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+        if (!$$1.getAbilities().instabuild) {
+            $$3.shrink(1);
         }
-        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)itemStack, (boolean)level.isClientSide());
+        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)$$3, (boolean)$$0.isClientSide());
     }
 }
 