--- net/minecraft/world/item/SpectralArrowItem.java
+++ net/minecraft/world/item/SpectralArrowItem.java
@@ -10,8 +10,8 @@
         super($$0);
     }
 
-    public net.minecraft.world.entity.projectile.AbstractArrow createArrow(net.minecraft.world.level.Level level, net.minecraft.world.item.ItemStack stack, net.minecraft.world.entity.LivingEntity shooter) {
-        return new net.minecraft.world.entity.projectile.SpectralArrow(level, shooter);
+    public net.minecraft.world.entity.projectile.AbstractArrow createArrow(net.minecraft.world.level.Level $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.world.entity.LivingEntity $$2) {
+        return new net.minecraft.world.entity.projectile.SpectralArrow($$0, $$2);
     }
 }
 