--- net/minecraft/world/item/SplashPotionItem.java
+++ net/minecraft/world/item/SplashPotionItem.java
@@ -10,9 +10,9 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        level.playSound(null, player.getX(), player.getY(), player.getZ(), net.minecraft.sounds.SoundEvents.SPLASH_POTION_THROW, net.minecraft.sounds.SoundSource.PLAYERS, 0.5f, 0.4f / (level.getRandom().nextFloat() * 0.4f + 0.8f));
-        return super.use(level, player, hand);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        $$0.playSound(null, $$1.getX(), $$1.getY(), $$1.getZ(), net.minecraft.sounds.SoundEvents.SPLASH_POTION_THROW, net.minecraft.sounds.SoundSource.PLAYERS, 0.5f, 0.4f / ($$0.getRandom().nextFloat() * 0.4f + 0.8f));
+        return super.use($$0, $$1, $$2);
     }
 }
 