--- net/minecraft/world/item/SuspiciousStewItem.java
+++ net/minecraft/world/item/SuspiciousStewItem.java
@@ -14,34 +14,34 @@
         super($$0);
     }
 
-    public static void saveMobEffect(net.minecraft.world.item.ItemStack bowlStack, net.minecraft.world.effect.MobEffect effect, int effectDuration) {
-        net.minecraft.nbt.CompoundTag compoundTag = bowlStack.getOrCreateTag();
-        net.minecraft.nbt.ListTag listTag = compoundTag.getList("Effects", 9);
-        net.minecraft.nbt.CompoundTag compoundTag2 = new net.minecraft.nbt.CompoundTag();
-        compoundTag2.putInt("EffectId", net.minecraft.world.effect.MobEffect.getId((net.minecraft.world.effect.MobEffect)effect));
-        compoundTag2.putInt("EffectDuration", effectDuration);
-        listTag.add((java.lang.Object)compoundTag2);
-        compoundTag.put("Effects", (net.minecraft.nbt.Tag)listTag);
+    public static void saveMobEffect(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.effect.MobEffect $$1, int $$2) {
+        net.minecraft.nbt.CompoundTag $$3 = $$0.getOrCreateTag();
+        net.minecraft.nbt.ListTag $$4 = $$3.getList("Effects", 9);
+        net.minecraft.nbt.CompoundTag $$5 = new net.minecraft.nbt.CompoundTag();
+        $$5.putInt("EffectId", net.minecraft.world.effect.MobEffect.getId((net.minecraft.world.effect.MobEffect)$$1));
+        $$5.putInt("EffectDuration", $$2);
+        $$4.add((java.lang.Object)$$5);
+        $$3.put("Effects", (net.minecraft.nbt.Tag)$$4);
     }
 
-    public net.minecraft.world.item.ItemStack finishUsingItem(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity entityLiving) {
-        net.minecraft.world.item.ItemStack itemStack = super.finishUsingItem(stack, level, entityLiving);
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTag();
-        if (compoundTag != null && compoundTag.contains("Effects", 9)) {
-            net.minecraft.nbt.ListTag listTag = compoundTag.getList("Effects", 10);
-            for (int i = 0; i < listTag.size(); ++i) {
-                net.minecraft.world.effect.MobEffect mobEffect;
-                int n = 160;
-                net.minecraft.nbt.CompoundTag compoundTag2 = listTag.getCompound(i);
-                if (compoundTag2.contains("EffectDuration", 3)) {
-                    n = compoundTag2.getInt("EffectDuration");
+    public net.minecraft.world.item.ItemStack finishUsingItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.Level $$1, net.minecraft.world.entity.LivingEntity $$2) {
+        net.minecraft.world.item.ItemStack $$3 = super.finishUsingItem($$0, $$1, $$2);
+        net.minecraft.nbt.CompoundTag $$4 = $$0.getTag();
+        if ($$4 != null && $$4.contains("Effects", 9)) {
+            net.minecraft.nbt.ListTag $$5 = $$4.getList("Effects", 10);
+            for (int $$6 = 0; $$6 < $$5.size(); ++$$6) {
+                net.minecraft.world.effect.MobEffect $$9;
+                int $$7 = 160;
+                net.minecraft.nbt.CompoundTag $$8 = $$5.getCompound($$6);
+                if ($$8.contains("EffectDuration", 3)) {
+                    $$7 = $$8.getInt("EffectDuration");
                 }
-                if ((mobEffect = net.minecraft.world.effect.MobEffect.byId((int)compoundTag2.getInt("EffectId"))) == null) continue;
-                entityLiving.addEffect(new net.minecraft.world.effect.MobEffectInstance(mobEffect, n));
+                if (($$9 = net.minecraft.world.effect.MobEffect.byId((int)$$8.getInt("EffectId"))) == null) continue;
+                $$2.addEffect(new net.minecraft.world.effect.MobEffectInstance($$9, $$7));
             }
         }
-        if (entityLiving instanceof net.minecraft.world.entity.player.Player && ((net.minecraft.world.entity.player.Player)entityLiving).getAbilities().instabuild) {
-            return itemStack;
+        if ($$2 instanceof net.minecraft.world.entity.player.Player && ((net.minecraft.world.entity.player.Player)$$2).getAbilities().instabuild) {
+            return $$3;
         }
         return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BOWL);
     }