--- net/minecraft/world/item/ThrowablePotionItem.java
+++ net/minecraft/world/item/ThrowablePotionItem.java
@@ -10,19 +10,19 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        if (!level.isClientSide) {
-            net.minecraft.world.entity.projectile.ThrownPotion thrownPotion = new net.minecraft.world.entity.projectile.ThrownPotion(level, (net.minecraft.world.entity.LivingEntity)player);
-            thrownPotion.setItem(itemStack);
-            thrownPotion.shootFromRotation((net.minecraft.world.entity.Entity)player, player.getXRot(), player.getYRot(), -20.0f, 0.5f, 1.0f);
-            level.addFreshEntity((net.minecraft.world.entity.Entity)thrownPotion);
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+        if (!$$0.isClientSide) {
+            net.minecraft.world.entity.projectile.ThrownPotion $$4 = new net.minecraft.world.entity.projectile.ThrownPotion($$0, (net.minecraft.world.entity.LivingEntity)$$1);
+            $$4.setItem($$3);
+            $$4.shootFromRotation((net.minecraft.world.entity.Entity)$$1, $$1.getXRot(), $$1.getYRot(), -20.0f, 0.5f, 1.0f);
+            $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$4);
         }
-        player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
-        if (!player.getAbilities().instabuild) {
-            itemStack.shrink(1);
+        $$1.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+        if (!$$1.getAbilities().instabuild) {
+            $$3.shrink(1);
         }
-        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)itemStack, (boolean)level.isClientSide());
+        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)$$3, (boolean)$$0.isClientSide());
     }
 }
 