--- net/minecraft/world/item/Tiers.java
+++ net/minecraft/world/item/Tiers.java
@@ -29,14 +29,14 @@
         return (net.minecraft.world.item.Tiers)java.lang.Enum.valueOf(net.minecraft.world.item.Tiers.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.item.Tiers(int level, int uses, float speed, float damage, int enchantmentValue, java.util.function.Supplier<net.minecraft.world.item.crafting.Ingredient> repairIngredient) {
+    public net.minecraft.world.item.Tiers(int $$0, int $$1, float $$2, float $$3, int $$4, java.util.function.Supplier<net.minecraft.world.item.crafting.Ingredient> $$5) {
         super(string, n);
-        this.level = level;
-        this.uses = uses;
-        this.speed = speed;
-        this.damage = damage;
-        this.enchantmentValue = enchantmentValue;
-        this.repairIngredient = new net.minecraft.util.LazyLoadedValue(repairIngredient);
+        this.level = $$0;
+        this.uses = $$1;
+        this.speed = $$2;
+        this.damage = $$3;
+        this.enchantmentValue = $$4;
+        this.repairIngredient = new net.minecraft.util.LazyLoadedValue($$5);
     }
 
     public int getUses() {