--- net/minecraft/world/item/TieredItem.java
+++ net/minecraft/world/item/TieredItem.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.item.Tier tier;
 
-    public net.minecraft.world.item.TieredItem(net.minecraft.world.item.Tier tier, net.minecraft.world.item.Item$Properties properties) {
-        super(properties.defaultDurability(tier.getUses()));
-        this.tier = tier;
+    public net.minecraft.world.item.TieredItem(net.minecraft.world.item.Tier $$0, net.minecraft.world.item.Item$Properties $$1) {
+        super($$1.defaultDurability($$0.getUses()));
+        this.tier = $$0;
     }
 
     public net.minecraft.world.item.Tier getTier() {
@@ -21,8 +21,8 @@
         return this.tier.getEnchantmentValue();
     }
 
-    public boolean isValidRepairItem(net.minecraft.world.item.ItemStack toRepair, net.minecraft.world.item.ItemStack repair) {
-        return this.tier.getRepairIngredient().test(repair) || super.isValidRepairItem(toRepair, repair);
+    public boolean isValidRepairItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        return this.tier.getRepairIngredient().test($$1) || super.isValidRepairItem($$0, $$1);
     }
 }
 