--- net/minecraft/world/item/TooltipFlag$Default.java
+++ net/minecraft/world/item/TooltipFlag$Default.java
@@ -20,9 +20,9 @@
         return (net.minecraft.world.item.TooltipFlag$Default)java.lang.Enum.valueOf(net.minecraft.world.item.TooltipFlag$Default.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.item.TooltipFlag$Default(boolean advanced) {
+    public net.minecraft.world.item.TooltipFlag$Default(boolean $$0) {
         super(string, n);
-        this.advanced = advanced;
+        this.advanced = $$0;
     }
 
     public boolean isAdvanced() {