--- net/minecraft/world/item/WrittenBookItem.java
+++ net/minecraft/world/item/WrittenBookItem.java
@@ -24,98 +24,97 @@
         super($$0);
     }
 
-    public static boolean makeSureTagIsValid(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag compoundTag) {
-        if (!net.minecraft.world.item.WritableBookItem.makeSureTagIsValid((net.minecraft.nbt.CompoundTag)compoundTag)) {
+    public static boolean makeSureTagIsValid(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$0) {
+        if (!net.minecraft.world.item.WritableBookItem.makeSureTagIsValid((net.minecraft.nbt.CompoundTag)$$0)) {
             return false;
         }
-        if (!compoundTag.contains("title", 8)) {
+        if (!$$0.contains("title", 8)) {
             return false;
         }
-        java.lang.String string = compoundTag.getString("title");
-        if (string.length() > 32) {
+        java.lang.String $$1 = $$0.getString("title");
+        if ($$1.length() > 32) {
             return false;
         }
-        return compoundTag.contains("author", 8);
+        return $$0.contains("author", 8);
     }
 
-    public static int getGeneration(net.minecraft.world.item.ItemStack bookStack) {
-        return bookStack.getTag().getInt("generation");
+    public static int getGeneration(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.getTag().getInt("generation");
     }
 
-    public static int getPageCount(net.minecraft.world.item.ItemStack bookSTack) {
-        net.minecraft.nbt.CompoundTag compoundTag = bookSTack.getTag();
-        return compoundTag != null ? compoundTag.getList("pages", 8).size() : 0;
+    public static int getPageCount(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTag();
+        return $$1 != null ? $$1.getList("pages", 8).size() : 0;
     }
 
-    public net.minecraft.network.chat.Component getName(net.minecraft.world.item.ItemStack stack) {
-        java.lang.String string;
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTag();
-        if (compoundTag != null && !net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)(string = compoundTag.getString("title")))) {
-            return net.minecraft.network.chat.Component.literal((java.lang.String)string);
+    public net.minecraft.network.chat.Component getName(net.minecraft.world.item.ItemStack $$0) {
+        java.lang.String $$2;
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTag();
+        if ($$1 != null && !net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)($$2 = $$1.getString("title")))) {
+            return net.minecraft.network.chat.Component.literal((java.lang.String)$$2);
         }
-        return super.getName(stack);
+        return super.getName($$0);
     }
 
-    public void appendHoverText(net.minecraft.world.item.ItemStack stack, @javax.annotation.Nullable net.minecraft.world.level.Level level, java.util.List<net.minecraft.network.chat.Component> tooltip, net.minecraft.world.item.TooltipFlag flag) {
-        if (stack.hasTag()) {
-            net.minecraft.nbt.CompoundTag compoundTag = stack.getTag();
-            java.lang.String string = compoundTag.getString("author");
-            if (!net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)string)) {
-                tooltip.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"book.byAuthor", (java.lang.Object[])new java.lang.Object[]{string}).withStyle(net.minecraft.ChatFormatting.GRAY));
+    public void appendHoverText(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.world.level.Level $$1, java.util.List<net.minecraft.network.chat.Component> $$2, net.minecraft.world.item.TooltipFlag $$3) {
+        if ($$0.hasTag()) {
+            net.minecraft.nbt.CompoundTag $$4 = $$0.getTag();
+            java.lang.String $$5 = $$4.getString("author");
+            if (!net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)$$5)) {
+                $$2.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"book.byAuthor", (java.lang.Object[])new java.lang.Object[]{$$5}).withStyle(net.minecraft.ChatFormatting.GRAY));
             }
-            tooltip.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("book.generation." + compoundTag.getInt("generation"))).withStyle(net.minecraft.ChatFormatting.GRAY));
+            $$2.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("book.generation." + $$4.getInt("generation"))).withStyle(net.minecraft.ChatFormatting.GRAY));
         }
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos = context.getClickedPos());
-        if (blockState.is(net.minecraft.world.level.block.Blocks.LECTERN)) {
-            return net.minecraft.world.level.block.LecternBlock.tryPlaceBook((net.minecraft.world.entity.player.Player)context.getPlayer(), (net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.world.level.block.state.BlockState)blockState, (net.minecraft.world.item.ItemStack)context.getItemInHand()) ? net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide) : net.minecraft.world.InteractionResult.PASS;
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.core.BlockPos $$2;
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$1.getBlockState($$2 = $$0.getClickedPos());
+        if ($$3.is(net.minecraft.world.level.block.Blocks.LECTERN)) {
+            return net.minecraft.world.level.block.LecternBlock.tryPlaceBook((net.minecraft.world.entity.player.Player)$$0.getPlayer(), (net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2, (net.minecraft.world.level.block.state.BlockState)$$3, (net.minecraft.world.item.ItemStack)$$0.getItemInHand()) ? net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide) : net.minecraft.world.InteractionResult.PASS;
         }
         return net.minecraft.world.InteractionResult.PASS;
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        player.openItemGui(itemStack, hand);
-        player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
-        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)itemStack, (boolean)level.isClientSide());
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+        $$1.openItemGui($$3, $$2);
+        $$1.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)$$3, (boolean)$$0.isClientSide());
     }
 
-    public static boolean resolveBookComponents(net.minecraft.world.item.ItemStack bookStack, @javax.annotation.Nullable net.minecraft.commands.CommandSourceStack resolvingSource, @javax.annotation.Nullable net.minecraft.world.entity.player.Player resolvingPlayer) {
-        java.lang.String string;
-        net.minecraft.nbt.CompoundTag compoundTag = bookStack.getTag();
-        if (compoundTag == null || compoundTag.getBoolean("resolved")) {
+    public static boolean resolveBookComponents(net.minecraft.world.item.ItemStack $$0, @javax.annotation.Nullable net.minecraft.commands.CommandSourceStack $$1, @javax.annotation.Nullable net.minecraft.world.entity.player.Player $$2) {
+        net.minecraft.nbt.CompoundTag $$3 = $$0.getTag();
+        if ($$3 == null || $$3.getBoolean("resolved")) {
             return false;
         }
-        compoundTag.putBoolean("resolved", true);
-        if (!net.minecraft.world.item.WrittenBookItem.makeSureTagIsValid((net.minecraft.nbt.CompoundTag)compoundTag)) {
+        $$3.putBoolean("resolved", true);
+        if (!net.minecraft.world.item.WrittenBookItem.makeSureTagIsValid((net.minecraft.nbt.CompoundTag)$$3)) {
             return false;
         }
-        net.minecraft.nbt.ListTag listTag = compoundTag.getList("pages", 8);
-        net.minecraft.nbt.ListTag listTag2 = new net.minecraft.nbt.ListTag();
-        for (int i = 0; i < listTag.size(); ++i) {
-            string = net.minecraft.world.item.WrittenBookItem.resolvePage((net.minecraft.commands.CommandSourceStack)resolvingSource, (net.minecraft.world.entity.player.Player)resolvingPlayer, (java.lang.String)listTag.getString(i));
-            if (string.length() > java.lang.Short.MAX_VALUE) {
+        net.minecraft.nbt.ListTag $$4 = $$3.getList("pages", 8);
+        net.minecraft.nbt.ListTag $$5 = new net.minecraft.nbt.ListTag();
+        for (int $$6 = 0; $$6 < $$4.size(); ++$$6) {
+            java.lang.String $$7 = net.minecraft.world.item.WrittenBookItem.resolvePage((net.minecraft.commands.CommandSourceStack)$$1, (net.minecraft.world.entity.player.Player)$$2, (java.lang.String)$$4.getString($$6));
+            if ($$7.length() > java.lang.Short.MAX_VALUE) {
                 return false;
             }
-            listTag2.add(i, (net.minecraft.nbt.Tag)net.minecraft.nbt.StringTag.valueOf((java.lang.String)string));
+            $$5.add($$6, (net.minecraft.nbt.Tag)net.minecraft.nbt.StringTag.valueOf((java.lang.String)$$7));
         }
-        if (compoundTag.contains("filtered_pages", 10)) {
-            net.minecraft.nbt.CompoundTag compoundTag2 = compoundTag.getCompound("filtered_pages");
-            string = new net.minecraft.nbt.CompoundTag();
-            for (java.lang.String string2 : compoundTag2.getAllKeys()) {
-                java.lang.String string3 = net.minecraft.world.item.WrittenBookItem.resolvePage((net.minecraft.commands.CommandSourceStack)resolvingSource, (net.minecraft.world.entity.player.Player)resolvingPlayer, (java.lang.String)compoundTag2.getString(string2));
-                if (string3.length() > java.lang.Short.MAX_VALUE) {
+        if ($$3.contains("filtered_pages", 10)) {
+            net.minecraft.nbt.CompoundTag $$8 = $$3.getCompound("filtered_pages");
+            net.minecraft.nbt.CompoundTag $$9 = new net.minecraft.nbt.CompoundTag();
+            for (java.lang.String $$10 : $$8.getAllKeys()) {
+                java.lang.String $$11 = net.minecraft.world.item.WrittenBookItem.resolvePage((net.minecraft.commands.CommandSourceStack)$$1, (net.minecraft.world.entity.player.Player)$$2, (java.lang.String)$$8.getString($$10));
+                if ($$11.length() > java.lang.Short.MAX_VALUE) {
                     return false;
                 }
-                string.putString(string2, string3);
+                $$9.putString($$10, $$11);
             }
-            compoundTag.put("filtered_pages", (net.minecraft.nbt.Tag)string);
+            $$3.put("filtered_pages", (net.minecraft.nbt.Tag)$$9);
         }
-        compoundTag.put("pages", (net.minecraft.nbt.Tag)listTag2);
+        $$3.put("pages", (net.minecraft.nbt.Tag)$$5);
         return true;
     }
 
@@ -131,7 +130,7 @@
         return net.minecraft.network.chat.Component$Serializer.toJson((net.minecraft.network.chat.Component)$$5);
     }
 
-    public boolean isFoil(net.minecraft.world.item.ItemStack stack) {
+    public boolean isFoil(net.minecraft.world.item.ItemStack $$0) {
         return true;
     }
 }