--- net/minecraft/world/item/WritableBookItem.java
+++ net/minecraft/world/item/WritableBookItem.java
@@ -10,34 +10,34 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext context) {
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos = context.getClickedPos());
-        if (blockState.is(net.minecraft.world.level.block.Blocks.LECTERN)) {
-            return net.minecraft.world.level.block.LecternBlock.tryPlaceBook((net.minecraft.world.entity.player.Player)context.getPlayer(), (net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.world.level.block.state.BlockState)blockState, (net.minecraft.world.item.ItemStack)context.getItemInHand()) ? net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide) : net.minecraft.world.InteractionResult.PASS;
+    public net.minecraft.world.InteractionResult useOn(net.minecraft.world.item.context.UseOnContext $$0) {
+        net.minecraft.core.BlockPos $$2;
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$1.getBlockState($$2 = $$0.getClickedPos());
+        if ($$3.is(net.minecraft.world.level.block.Blocks.LECTERN)) {
+            return net.minecraft.world.level.block.LecternBlock.tryPlaceBook((net.minecraft.world.entity.player.Player)$$0.getPlayer(), (net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2, (net.minecraft.world.level.block.state.BlockState)$$3, (net.minecraft.world.item.ItemStack)$$0.getItemInHand()) ? net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide) : net.minecraft.world.InteractionResult.PASS;
         }
         return net.minecraft.world.InteractionResult.PASS;
     }
 
-    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        player.openItemGui(itemStack, hand);
-        player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
-        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)itemStack, (boolean)level.isClientSide());
+    public net.minecraft.world.InteractionResultHolder<net.minecraft.world.item.ItemStack> use(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2) {
+        net.minecraft.world.item.ItemStack $$3 = $$1.getItemInHand($$2);
+        $$1.openItemGui($$3, $$2);
+        $$1.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)this));
+        return net.minecraft.world.InteractionResultHolder.sidedSuccess((java.lang.Object)$$3, (boolean)$$0.isClientSide());
     }
 
-    public static boolean makeSureTagIsValid(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag compoundTag) {
-        if (compoundTag == null) {
+    public static boolean makeSureTagIsValid(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$0) {
+        if ($$0 == null) {
             return false;
         }
-        if (!compoundTag.contains("pages", 9)) {
+        if (!$$0.contains("pages", 9)) {
             return false;
         }
-        net.minecraft.nbt.ListTag listTag = compoundTag.getList("pages", 8);
-        for (int i = 0; i < listTag.size(); ++i) {
-            java.lang.String string = listTag.getString(i);
-            if (string.length() <= java.lang.Short.MAX_VALUE) continue;
+        net.minecraft.nbt.ListTag $$1 = $$0.getList("pages", 8);
+        for (int $$2 = 0; $$2 < $$1.size(); ++$$2) {
+            java.lang.String $$3 = $$1.getString($$2);
+            if ($$3.length() <= java.lang.Short.MAX_VALUE) continue;
             return false;
         }
         return true;