--- net/minecraft/util/datafix/schemas/NamespacedSchema.java
+++ net/minecraft/util/datafix/schemas/NamespacedSchema.java
@@ -13,12 +13,12 @@
         super($$0, $$1);
     }
 
-    public static java.lang.String ensureNamespaced(java.lang.String string) {
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.resources.ResourceLocation.tryParse((java.lang.String)string);
-        if (resourceLocation != null) {
-            return resourceLocation.toString();
+    public static java.lang.String ensureNamespaced(java.lang.String $$0) {
+        net.minecraft.resources.ResourceLocation $$1 = net.minecraft.resources.ResourceLocation.tryParse((java.lang.String)$$0);
+        if ($$1 != null) {
+            return $$1.toString();
         }
-        return string;
+        return $$0;
     }
 
     public static com.mojang.datafixers.types.Type<java.lang.String> namespacedString() {