--- net/minecraft/util/datafix/schemas/V3204.java
+++ net/minecraft/util/datafix/schemas/V3204.java
@@ -1,0 +1,19 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.util.datafix.schemas;
+
+public class net.minecraft.util.datafix.schemas.V3204
+extends net.minecraft.util.datafix.schemas.NamespacedSchema
+ {
+    public net.minecraft.util.datafix.schemas.V3204(int $$0, com.mojang.datafixers.schemas.Schema $$1) {
+        super($$0, $$1);
+    }
+
+    public java.util.Map<java.lang.String, java.util.function.Supplier<com.mojang.datafixers.types.templates.TypeTemplate>> registerBlockEntities(com.mojang.datafixers.schemas.Schema $$0) {
+        java.util.Map $$1 = super.registerBlockEntities($$0);
+        $$0.registerSimple($$1, "minecraft:chiseled_bookshelf");
+        return $$1;
+    }
+}
+