--- net/minecraft/util/datafix/schemas/V704.java
+++ net/minecraft/util/datafix/schemas/V704.java
@@ -13,8 +13,8 @@
         super($$0, $$1);
     }
 
-    public static void registerInventory(com.mojang.datafixers.schemas.Schema schema, java.util.Map<java.lang.String, java.util.function.Supplier<com.mojang.datafixers.types.templates.TypeTemplate>> map, java.lang.String name) {
-        schema.register(map, name, () -> com.mojang.datafixers.DSL.optionalFields((java.lang.String)"Items", (com.mojang.datafixers.types.templates.TypeTemplate)com.mojang.datafixers.DSL.list((com.mojang.datafixers.types.templates.TypeTemplate)net.minecraft.util.datafix.fixes.References.ITEM_STACK.in(schema))));
+    public static void registerInventory(com.mojang.datafixers.schemas.Schema $$0, java.util.Map<java.lang.String, java.util.function.Supplier<com.mojang.datafixers.types.templates.TypeTemplate>> $$1, java.lang.String $$2) {
+        $$0.register($$1, $$2, () -> com.mojang.datafixers.DSL.optionalFields((java.lang.String)"Items", (com.mojang.datafixers.types.templates.TypeTemplate)com.mojang.datafixers.DSL.list((com.mojang.datafixers.types.templates.TypeTemplate)net.minecraft.util.datafix.fixes.References.ITEM_STACK.in($$0))));
     }
 
     public com.mojang.datafixers.types.Type<?> getChoiceType(com.mojang.datafixers.DSL$TypeReference $$0, java.lang.String $$1) {