--- net/minecraft/world/entity/animal/frog/Tadpole.java
+++ net/minecraft/world/entity/animal/frog/Tadpole.java
@@ -164,17 +164,19 @@
         if (level instanceof net.minecraft.server.level.ServerLevel) {
             net.minecraft.server.level.ServerLevel $$0 = (net.minecraft.server.level.ServerLevel)level;
             net.minecraft.world.entity.animal.frog.Frog $$1 = (net.minecraft.world.entity.animal.frog.Frog)net.minecraft.world.entity.EntityType.FROG.create(this.level);
-            $$1.moveTo(this.getX(), this.getY(), this.getZ(), this.getYRot(), this.getXRot());
-            $$1.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)$$0, this.level.getCurrentDifficultyAt($$1.blockPosition()), net.minecraft.world.entity.MobSpawnType.CONVERSION, null, null);
-            $$1.setNoAi(this.isNoAi());
-            if (this.hasCustomName()) {
-                $$1.setCustomName(this.getCustomName());
-                $$1.setCustomNameVisible(this.isCustomNameVisible());
+            if ($$1 != null) {
+                $$1.moveTo(this.getX(), this.getY(), this.getZ(), this.getYRot(), this.getXRot());
+                $$1.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)$$0, this.level.getCurrentDifficultyAt($$1.blockPosition()), net.minecraft.world.entity.MobSpawnType.CONVERSION, null, null);
+                $$1.setNoAi(this.isNoAi());
+                if (this.hasCustomName()) {
+                    $$1.setCustomName(this.getCustomName());
+                    $$1.setCustomNameVisible(this.isCustomNameVisible());
+                }
+                $$1.setPersistenceRequired();
+                this.playSound(net.minecraft.sounds.SoundEvents.TADPOLE_GROW_UP, 0.15f, 1.0f);
+                $$0.addFreshEntityWithPassengers((net.minecraft.world.entity.Entity)$$1);
+                this.discard();
             }
-            $$1.setPersistenceRequired();
-            this.playSound(net.minecraft.sounds.SoundEvents.TADPOLE_GROW_UP, 0.15f, 1.0f);
-            $$0.addFreshEntityWithPassengers((net.minecraft.world.entity.Entity)$$1);
-            this.discard();
         }
     }
 