--- net/minecraft/network/chat/ClickEvent.java
+++ net/minecraft/network/chat/ClickEvent.java
@@ -8,9 +8,9 @@
     public net.minecraft.network.chat.ClickEvent$Action action;
     public java.lang.String value;
 
-    public net.minecraft.network.chat.ClickEvent(net.minecraft.network.chat.ClickEvent$Action action, java.lang.String value) {
-        this.action = action;
-        this.value = value;
+    public net.minecraft.network.chat.ClickEvent(net.minecraft.network.chat.ClickEvent$Action $$0, java.lang.String $$1) {
+        this.action = $$0;
+        this.value = $$1;
     }
 
     public net.minecraft.network.chat.ClickEvent$Action getAction() {