--- net/minecraft/network/chat/CommonComponents.java
+++ net/minecraft/network/chat/CommonComponents.java
@@ -35,12 +35,12 @@
         return net.minecraft.network.chat.Component.translatable((java.lang.String)"gui.minutes", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
-    public static net.minecraft.network.chat.Component optionStatus(boolean isEnabled) {
-        return isEnabled ? OPTION_ON : OPTION_OFF;
+    public static net.minecraft.network.chat.Component optionStatus(boolean $$0) {
+        return $$0 ? OPTION_ON : OPTION_OFF;
     }
 
-    public static net.minecraft.network.chat.MutableComponent optionStatus(net.minecraft.network.chat.Component message, boolean composed) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)(composed ? "options.on.composed" : "options.off.composed"), (java.lang.Object[])new java.lang.Object[]{message});
+    public static net.minecraft.network.chat.MutableComponent optionStatus(net.minecraft.network.chat.Component $$0, boolean $$1) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)($$1 ? "options.on.composed" : "options.off.composed"), (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
     public static net.minecraft.network.chat.MutableComponent optionNameValue(net.minecraft.network.chat.Component $$0, net.minecraft.network.chat.Component $$1) {