--- net/minecraft/network/chat/FormattedText$1.java
+++ net/minecraft/network/chat/FormattedText$1.java
@@ -9,11 +9,11 @@
     public net.minecraft.network.chat.FormattedText$1() {
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> acceptor) {
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> $$0) {
         return java.util.Optional.empty();
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> acceptor, net.minecraft.network.chat.Style style) {
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> $$0, net.minecraft.network.chat.Style $$1) {
         return java.util.Optional.empty();
     }
 }