--- net/minecraft/network/chat/FormattedText$2.java
+++ net/minecraft/network/chat/FormattedText$2.java
@@ -12,12 +12,12 @@
         this.val$text = string;
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> acceptor) {
-        return acceptor.accept(this.val$text);
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> $$0) {
+        return $$0.accept(this.val$text);
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> acceptor, net.minecraft.network.chat.Style style) {
-        return acceptor.accept(style, this.val$text);
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> $$0, net.minecraft.network.chat.Style $$1) {
+        return $$0.accept($$1, this.val$text);
     }
 }
 