--- net/minecraft/network/chat/FormattedText$4.java
+++ net/minecraft/network/chat/FormattedText$4.java
@@ -12,20 +12,20 @@
         this.val$parts = list;
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> acceptor) {
-        for (net.minecraft.network.chat.FormattedText formattedText : this.val$parts) {
-            java.util.Optional optional = formattedText.visit(acceptor);
-            if (!optional.isPresent()) continue;
-            return optional;
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$ContentConsumer<T> $$0) {
+        for (net.minecraft.network.chat.FormattedText $$1 : this.val$parts) {
+            java.util.Optional $$2 = $$1.visit($$0);
+            if (!$$2.isPresent()) continue;
+            return $$2;
         }
         return java.util.Optional.empty();
     }
 
-    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> acceptor, net.minecraft.network.chat.Style style) {
-        for (net.minecraft.network.chat.FormattedText formattedText : this.val$parts) {
-            java.util.Optional optional = formattedText.visit(acceptor, style);
-            if (!optional.isPresent()) continue;
-            return optional;
+    public <T> java.util.Optional<T> visit(net.minecraft.network.chat.FormattedText$StyledContentConsumer<T> $$0, net.minecraft.network.chat.Style $$1) {
+        for (net.minecraft.network.chat.FormattedText $$2 : this.val$parts) {
+            java.util.Optional $$3 = $$2.visit($$0, $$1);
+            if (!$$3.isPresent()) continue;
+            return $$3;
         }
         return java.util.Optional.empty();
     }