--- net/minecraft/network/chat/HoverEvent.java
+++ net/minecraft/network/chat/HoverEvent.java
@@ -9,9 +9,9 @@
     public net.minecraft.network.chat.HoverEvent$Action<?> action;
     public java.lang.Object value;
 
-    public <T> net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action<T> action, T value) {
-        this.action = action;
-        this.value = value;
+    public <T> net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action<T> $$0, T $$1) {
+        this.action = $$0;
+        this.value = $$1;
     }
 
     public net.minecraft.network.chat.HoverEvent$Action<?> getAction() {
@@ -19,9 +19,9 @@
     }
 
     @javax.annotation.Nullable
-    public <T> T getValue(net.minecraft.network.chat.HoverEvent$Action<T> actionType) {
-        if (this.action == actionType) {
-            return (T)actionType.cast(this.value);
+    public <T> T getValue(net.minecraft.network.chat.HoverEvent$Action<T> $$0) {
+        if (this.action == $$0) {
+            return (T)$$0.cast(this.value);
         }
         return null;
     }
@@ -48,22 +48,22 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.network.chat.HoverEvent deserialize(com.google.gson.JsonObject json) {
-        java.lang.String string = net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)json, (java.lang.String)"action", null);
-        if (string == null) {
+    public static net.minecraft.network.chat.HoverEvent deserialize(com.google.gson.JsonObject $$0) {
+        java.lang.String $$1 = net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"action", null);
+        if ($$1 == null) {
             return null;
         }
-        net.minecraft.network.chat.HoverEvent$Action action = net.minecraft.network.chat.HoverEvent$Action.getByName((java.lang.String)string);
-        if (action == null) {
+        net.minecraft.network.chat.HoverEvent$Action $$2 = net.minecraft.network.chat.HoverEvent$Action.getByName((java.lang.String)$$1);
+        if ($$2 == null) {
             return null;
         }
-        com.google.gson.JsonElement jsonElement = json.get("contents");
-        if (jsonElement != null) {
-            return action.deserialize(jsonElement);
+        com.google.gson.JsonElement $$3 = $$0.get("contents");
+        if ($$3 != null) {
+            return $$2.deserialize($$3);
         }
-        net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)json.get("value"));
-        if (mutableComponent != null) {
-            return action.deserializeFromLegacy((net.minecraft.network.chat.Component)mutableComponent);
+        net.minecraft.network.chat.MutableComponent $$4 = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)$$0.get("value"));
+        if ($$4 != null) {
+            return $$2.deserializeFromLegacy((net.minecraft.network.chat.Component)$$4);
         }
         return null;
     }