--- net/minecraft/network/chat/HoverEvent$EntityTooltipInfo.java
+++ net/minecraft/network/chat/HoverEvent$EntityTooltipInfo.java
@@ -12,34 +12,34 @@
     @javax.annotation.Nullable
     public java.util.List<net.minecraft.network.chat.Component> linesCache;
 
-    public net.minecraft.network.chat.HoverEvent$EntityTooltipInfo(net.minecraft.world.entity.EntityType<?> type, java.util.UUID id, @javax.annotation.Nullable net.minecraft.network.chat.Component name) {
-        this.type = type;
-        this.id = id;
-        this.name = name;
+    public net.minecraft.network.chat.HoverEvent$EntityTooltipInfo(net.minecraft.world.entity.EntityType<?> $$0, java.util.UUID $$1, @javax.annotation.Nullable net.minecraft.network.chat.Component $$2) {
+        this.type = $$0;
+        this.id = $$1;
+        this.name = $$2;
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.network.chat.HoverEvent$EntityTooltipInfo create(com.google.gson.JsonElement element) {
-        if (!element.isJsonObject()) {
+    public static net.minecraft.network.chat.HoverEvent$EntityTooltipInfo create(com.google.gson.JsonElement $$0) {
+        if (!$$0.isJsonObject()) {
             return null;
         }
-        com.google.gson.JsonObject jsonObject = element.getAsJsonObject();
-        net.minecraft.world.entity.EntityType entityType = (net.minecraft.world.entity.EntityType)net.minecraft.core.Registry.ENTITY_TYPE.get(new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)jsonObject, (java.lang.String)"type")));
-        java.util.UUID uUID = java.util.UUID.fromString((java.lang.String)net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)jsonObject, (java.lang.String)"id"));
-        net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)jsonObject.get("name"));
-        return new net.minecraft.network.chat.HoverEvent$EntityTooltipInfo(entityType, uUID, (net.minecraft.network.chat.Component)mutableComponent);
+        com.google.gson.JsonObject $$1 = $$0.getAsJsonObject();
+        net.minecraft.world.entity.EntityType $$2 = (net.minecraft.world.entity.EntityType)net.minecraft.core.Registry.ENTITY_TYPE.get(new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$1, (java.lang.String)"type")));
+        java.util.UUID $$3 = java.util.UUID.fromString((java.lang.String)net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$1, (java.lang.String)"id"));
+        net.minecraft.network.chat.MutableComponent $$4 = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)$$1.get("name"));
+        return new net.minecraft.network.chat.HoverEvent$EntityTooltipInfo($$2, $$3, (net.minecraft.network.chat.Component)$$4);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.network.chat.HoverEvent$EntityTooltipInfo create(net.minecraft.network.chat.Component component) {
+    public static net.minecraft.network.chat.HoverEvent$EntityTooltipInfo create(net.minecraft.network.chat.Component $$0) {
         try {
-            net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.nbt.TagParser.parseTag((java.lang.String)component.getString());
-            net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component$Serializer.fromJson((java.lang.String)compoundTag.getString("name"));
-            net.minecraft.world.entity.EntityType entityType = (net.minecraft.world.entity.EntityType)net.minecraft.core.Registry.ENTITY_TYPE.get(new net.minecraft.resources.ResourceLocation(compoundTag.getString("type")));
-            java.util.UUID uUID = java.util.UUID.fromString((java.lang.String)compoundTag.getString("id"));
-            return new net.minecraft.network.chat.HoverEvent$EntityTooltipInfo(entityType, uUID, (net.minecraft.network.chat.Component)mutableComponent);
+            net.minecraft.nbt.CompoundTag $$1 = net.minecraft.nbt.TagParser.parseTag((java.lang.String)$$0.getString());
+            net.minecraft.network.chat.MutableComponent $$2 = net.minecraft.network.chat.Component$Serializer.fromJson((java.lang.String)$$1.getString("name"));
+            net.minecraft.world.entity.EntityType $$3 = (net.minecraft.world.entity.EntityType)net.minecraft.core.Registry.ENTITY_TYPE.get(new net.minecraft.resources.ResourceLocation($$1.getString("type")));
+            java.util.UUID $$4 = java.util.UUID.fromString((java.lang.String)$$1.getString("id"));
+            return new net.minecraft.network.chat.HoverEvent$EntityTooltipInfo($$3, $$4, (net.minecraft.network.chat.Component)$$2);
         }
-        catch (java.lang.Exception exception) {
+        catch (java.lang.Exception $$5) {
             return null;
         }
     }