--- net/minecraft/network/chat/LastSeenMessages$Update.java
+++ net/minecraft/network/chat/LastSeenMessages$Update.java
@@ -6,41 +6,41 @@
 public class net.minecraft.network.chat.LastSeenMessages$Update
 extends java.lang.Record
  {
-    public net.minecraft.network.chat.LastSeenMessages lastSeen;
-    public java.util.Optional<net.minecraft.network.chat.LastSeenMessages$Entry> lastReceived;
+    public int offset;
+    public java.util.BitSet acknowledged;
 
     public net.minecraft.network.chat.LastSeenMessages$Update(net.minecraft.network.FriendlyByteBuf $$0) {
-        this(new net.minecraft.network.chat.LastSeenMessages($$0), $$0.readOptional(net.minecraft.network.chat.LastSeenMessages$Entry::new));
+        this($$0.readVarInt(), $$0.readFixedBitSet(20));
     }
 
-    public net.minecraft.network.chat.LastSeenMessages$Update(net.minecraft.network.chat.LastSeenMessages $$0, java.util.Optional<net.minecraft.network.chat.LastSeenMessages$Entry> $$1) {
-        this.lastSeen = $$0;
-        this.lastReceived = $$1;
+    public net.minecraft.network.chat.LastSeenMessages$Update(int $$0, java.util.BitSet $$1) {
+        this.offset = $$0;
+        this.acknowledged = $$1;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf $$02) {
-        this.lastSeen.write($$02);
-        $$02.writeOptional(this.lastReceived, ($$0, $$1) -> $$1.write($$0));
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.offset);
+        $$0.writeFixedBitSet(this.acknowledged, 20);
     }
 
     public java.lang.String toString() {
-        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Update.class, "lastSeen;lastReceived", "lastSeen", "lastReceived"}, this);
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Update.class, "offset;acknowledged", "offset", "acknowledged"}, this);
     }
 
     public int hashCode() {
-        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Update.class, "lastSeen;lastReceived", "lastSeen", "lastReceived"}, this);
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Update.class, "offset;acknowledged", "offset", "acknowledged"}, this);
     }
 
     public boolean equals(java.lang.Object $$0) {
-        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Update.class, "lastSeen;lastReceived", "lastSeen", "lastReceived"}, this, $$0);
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Update.class, "offset;acknowledged", "offset", "acknowledged"}, this, $$0);
     }
 
-    public net.minecraft.network.chat.LastSeenMessages lastSeen() {
-        return this.lastSeen;
+    public int offset() {
+        return this.offset;
     }
 
-    public java.util.Optional<net.minecraft.network.chat.LastSeenMessages$Entry> lastReceived() {
-        return this.lastReceived;
+    public java.util.BitSet acknowledged() {
+        return this.acknowledged;
     }
 }
 