--- net/minecraft/network/chat/LastSeenMessagesTracker$Update.java
+++ net/minecraft/network/chat/LastSeenMessagesTracker$Update.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.LastSeenMessagesTracker$Update
+extends java.lang.Record
+ {
+    public net.minecraft.network.chat.LastSeenMessages lastSeen;
+    public net.minecraft.network.chat.LastSeenMessages$Update update;
+
+    public net.minecraft.network.chat.LastSeenMessagesTracker$Update(net.minecraft.network.chat.LastSeenMessages $$0, net.minecraft.network.chat.LastSeenMessages$Update $$1) {
+        this.lastSeen = $$0;
+        this.update = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessagesTracker$Update.class, "lastSeen;update", "lastSeen", "update"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessagesTracker$Update.class, "lastSeen;update", "lastSeen", "update"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessagesTracker$Update.class, "lastSeen;update", "lastSeen", "update"}, this, $$0);
+    }
+
+    public net.minecraft.network.chat.LastSeenMessages lastSeen() {
+        return this.lastSeen;
+    }
+
+    public net.minecraft.network.chat.LastSeenMessages$Update update() {
+        return this.update;
+    }
+}
+