--- net/minecraft/network/chat/LastSeenTrackedEntry.java
+++ net/minecraft/network/chat/LastSeenTrackedEntry.java
@@ -1,0 +1,41 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.LastSeenTrackedEntry
+extends java.lang.Record
+ {
+    public net.minecraft.network.chat.MessageSignature signature;
+    public boolean pending;
+
+    public net.minecraft.network.chat.LastSeenTrackedEntry(net.minecraft.network.chat.MessageSignature $$0, boolean $$1) {
+        this.signature = $$0;
+        this.pending = $$1;
+    }
+
+    public net.minecraft.network.chat.LastSeenTrackedEntry acknowledge() {
+        return this.pending ? new net.minecraft.network.chat.LastSeenTrackedEntry(this.signature, false) : this;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenTrackedEntry.class, "signature;pending", "signature", "pending"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenTrackedEntry.class, "signature;pending", "signature", "pending"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenTrackedEntry.class, "signature;pending", "signature", "pending"}, this, $$0);
+    }
+
+    public net.minecraft.network.chat.MessageSignature signature() {
+        return this.signature;
+    }
+
+    public boolean pending() {
+        return this.pending;
+    }
+}
+