--- net/minecraft/network/chat/LastSeenMessages$Packed.java
+++ net/minecraft/network/chat/LastSeenMessages$Packed.java
@@ -1,0 +1,56 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.LastSeenMessages$Packed
+extends java.lang.Record
+ {
+    public java.util.List<net.minecraft.network.chat.MessageSignature$Packed> entries;
+    public static net.minecraft.network.chat.LastSeenMessages$Packed EMPTY;
+
+    public net.minecraft.network.chat.LastSeenMessages$Packed(net.minecraft.network.FriendlyByteBuf $$0) {
+        this((java.util.List)$$0.readCollection(net.minecraft.network.FriendlyByteBuf.limitValue(java.util.ArrayList::new, (int)20), net.minecraft.network.chat.MessageSignature$Packed::read));
+    }
+
+    public net.minecraft.network.chat.LastSeenMessages$Packed(java.util.List<net.minecraft.network.chat.MessageSignature$Packed> $$0) {
+        this.entries = $$0;
+    }
+
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeCollection((java.util.Collection)this.entries, net.minecraft.network.chat.MessageSignature$Packed::write);
+    }
+
+    public java.util.Optional<net.minecraft.network.chat.LastSeenMessages> unpack(net.minecraft.network.chat.MessageSignature$Unpacker $$0) {
+        java.util.ArrayList $$1 = new java.util.ArrayList(this.entries.size());
+        for (net.minecraft.network.chat.MessageSignature$Packed $$2 : this.entries) {
+            java.util.Optional $$3 = $$2.unpack($$0);
+            if ($$3.isEmpty()) {
+                return java.util.Optional.empty();
+            }
+            $$1.add((java.lang.Object)((net.minecraft.network.chat.MessageSignature)$$3.get()));
+        }
+        return java.util.Optional.of((java.lang.Object)new net.minecraft.network.chat.LastSeenMessages((java.util.List)$$1));
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Packed.class, "entries", "entries"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Packed.class, "entries", "entries"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.LastSeenMessages$Packed.class, "entries", "entries"}, this, $$0);
+    }
+
+    public java.util.List<net.minecraft.network.chat.MessageSignature$Packed> entries() {
+        return this.entries;
+    }
+
+    static {
+        EMPTY = new net.minecraft.network.chat.LastSeenMessages$Packed(java.util.List.of());
+    }
+}
+