--- net/minecraft/network/chat/MessageSignature.java
+++ net/minecraft/network/chat/MessageSignature.java
@@ -7,45 +7,29 @@
 extends java.lang.Record
  {
     public byte[] bytes;
-    public static net.minecraft.network.chat.MessageSignature EMPTY;
+    public static int BYTES = 256;
 
-    public net.minecraft.network.chat.MessageSignature(net.minecraft.network.FriendlyByteBuf $$0) {
-        this($$0.readByteArray());
-    }
-
     public net.minecraft.network.chat.MessageSignature(byte[] $$0) {
+        com.google.common.base.Preconditions.checkState(($$0.length == 256 ? 1 : 0) != 0, (java.lang.Object)"Invalid message signature size");
         this.bytes = $$0;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
-        $$0.writeByteArray(this.bytes);
+    public static net.minecraft.network.chat.MessageSignature read(net.minecraft.network.FriendlyByteBuf $$0) {
+        byte[] $$1 = new byte[256];
+        $$0.readBytes($$1);
+        return new net.minecraft.network.chat.MessageSignature($$1);
     }
 
-    public boolean verify(net.minecraft.util.SignatureValidator $$0, net.minecraft.network.chat.SignedMessageHeader $$1, net.minecraft.network.chat.SignedMessageBody $$22) {
-        if (!this.isEmpty()) {
-            byte[] $$3 = $$22.hash().asBytes();
-            return $$0.validate($$2 -> $$1.updateSignature($$2, $$3), this.bytes);
-        }
-        return false;
+    public static void write(net.minecraft.network.FriendlyByteBuf $$0, net.minecraft.network.chat.MessageSignature $$1) {
+        $$0.writeBytes($$1.bytes);
     }
 
-    public boolean verify(net.minecraft.util.SignatureValidator $$0, net.minecraft.network.chat.SignedMessageHeader $$1, byte[] $$22) {
-        if (!this.isEmpty()) {
-            return $$0.validate($$2 -> $$1.updateSignature($$2, $$22), this.bytes);
-        }
-        return false;
+    public boolean verify(net.minecraft.util.SignatureValidator $$0, net.minecraft.util.SignatureUpdater $$1) {
+        return $$0.validate($$1, this.bytes);
     }
 
-    public boolean isEmpty() {
-        return this.bytes.length == 0;
-    }
-
-    @javax.annotation.Nullable
     public java.nio.ByteBuffer asByteBuffer() {
-        if (!this.isEmpty()) {
-            return java.nio.ByteBuffer.wrap((byte[])this.bytes);
-        }
-        return null;
+        return java.nio.ByteBuffer.wrap((byte[])this.bytes);
     }
 
     /*
@@ -79,18 +63,16 @@
     }
 
     public java.lang.String toString() {
-        if (!this.isEmpty()) {
-            return java.util.Base64.getEncoder().encodeToString(this.bytes);
-        }
-        return "empty";
+        return java.util.Base64.getEncoder().encodeToString(this.bytes);
     }
 
+    public net.minecraft.network.chat.MessageSignature$Packed pack(net.minecraft.network.chat.MessageSignature$Packer $$0) {
+        int $$1 = $$0.pack(this);
+        return $$1 != -1 ? new net.minecraft.network.chat.MessageSignature$Packed($$1) : new net.minecraft.network.chat.MessageSignature$Packed(this);
+    }
+
     public byte[] bytes() {
         return this.bytes;
     }
-
-    static {
-        EMPTY = new net.minecraft.network.chat.MessageSignature(it.unimi.dsi.fastutil.bytes.ByteArrays.EMPTY_ARRAY);
-    }
 }
 