--- net/minecraft/network/chat/MessageSignature$Packed.java
+++ net/minecraft/network/chat/MessageSignature$Packed.java
@@ -1,0 +1,70 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.MessageSignature$Packed
+extends java.lang.Record
+ {
+    public int id;
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.MessageSignature fullSignature;
+    public static int FULL_SIGNATURE = -1;
+
+    public net.minecraft.network.chat.MessageSignature$Packed(net.minecraft.network.chat.MessageSignature $$0) {
+        this(-1, $$0);
+    }
+
+    public net.minecraft.network.chat.MessageSignature$Packed(int $$0) {
+        this($$0, null);
+    }
+
+    public net.minecraft.network.chat.MessageSignature$Packed(int $$0, @javax.annotation.Nullable net.minecraft.network.chat.MessageSignature $$1) {
+        this.id = $$0;
+        this.fullSignature = $$1;
+    }
+
+    public static net.minecraft.network.chat.MessageSignature$Packed read(net.minecraft.network.FriendlyByteBuf $$0) {
+        int $$1 = $$0.readVarInt() - 1;
+        if ($$1 == -1) {
+            return new net.minecraft.network.chat.MessageSignature$Packed(net.minecraft.network.chat.MessageSignature.read((net.minecraft.network.FriendlyByteBuf)$$0));
+        }
+        return new net.minecraft.network.chat.MessageSignature$Packed($$1);
+    }
+
+    public static void write(net.minecraft.network.FriendlyByteBuf $$0, net.minecraft.network.chat.MessageSignature$Packed $$1) {
+        $$0.writeVarInt($$1.id() + 1);
+        if ($$1.fullSignature() != null) {
+            net.minecraft.network.chat.MessageSignature.write((net.minecraft.network.FriendlyByteBuf)$$0, (net.minecraft.network.chat.MessageSignature)$$1.fullSignature());
+        }
+    }
+
+    public java.util.Optional<net.minecraft.network.chat.MessageSignature> unpack(net.minecraft.network.chat.MessageSignature$Unpacker $$0) {
+        if (this.fullSignature != null) {
+            return java.util.Optional.of((java.lang.Object)this.fullSignature);
+        }
+        return java.util.Optional.ofNullable((java.lang.Object)$$0.unpack(this.id));
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.MessageSignature$Packed.class, "id;fullSignature", "id", "fullSignature"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.MessageSignature$Packed.class, "id;fullSignature", "id", "fullSignature"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.MessageSignature$Packed.class, "id;fullSignature", "id", "fullSignature"}, this, $$0);
+    }
+
+    public int id() {
+        return this.id;
+    }
+
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.MessageSignature fullSignature() {
+        return this.fullSignature;
+    }
+}
+