--- net/minecraft/network/chat/OutgoingChatMessage$Disguised.java
+++ net/minecraft/network/chat/OutgoingChatMessage$Disguised.java
@@ -1,0 +1,36 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.OutgoingChatMessage$Disguised
+extends java.lang.Record
+implements net.minecraft.network.chat.OutgoingChatMessage
+ {
+    public net.minecraft.network.chat.Component content;
+
+    public net.minecraft.network.chat.OutgoingChatMessage$Disguised(net.minecraft.network.chat.Component $$0) {
+        this.content = $$0;
+    }
+
+    public net.minecraft.network.chat.Component content() {
+        return this.content;
+    }
+
+    public void sendToPlayer(net.minecraft.server.level.ServerPlayer $$0, boolean $$1, net.minecraft.network.chat.ChatType$Bound $$2) {
+        $$0.connection.sendDisguisedChatMessage(this.content, $$2);
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.OutgoingChatMessage$Disguised.class, "content", "content"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.OutgoingChatMessage$Disguised.class, "content", "content"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.OutgoingChatMessage$Disguised.class, "content", "content"}, this, $$0);
+    }
+}
+