--- net/minecraft/network/chat/OutgoingChatMessage$Player.java
+++ net/minecraft/network/chat/OutgoingChatMessage$Player.java
@@ -1,0 +1,43 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.OutgoingChatMessage$Player
+extends java.lang.Record
+implements net.minecraft.network.chat.OutgoingChatMessage
+ {
+    public net.minecraft.network.chat.PlayerChatMessage message;
+
+    public net.minecraft.network.chat.OutgoingChatMessage$Player(net.minecraft.network.chat.PlayerChatMessage $$0) {
+        this.message = $$0;
+    }
+
+    public net.minecraft.network.chat.Component content() {
+        return this.message.decoratedContent();
+    }
+
+    public void sendToPlayer(net.minecraft.server.level.ServerPlayer $$0, boolean $$1, net.minecraft.network.chat.ChatType$Bound $$2) {
+        net.minecraft.network.chat.PlayerChatMessage $$3 = this.message.filter($$1);
+        if (!$$3.isFullyFiltered()) {
+            $$0.connection.sendPlayerChatMessage($$3, $$2);
+        }
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.OutgoingChatMessage$Player.class, "message", "message"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.OutgoingChatMessage$Player.class, "message", "message"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.OutgoingChatMessage$Player.class, "message", "message"}, this, $$0);
+    }
+
+    public net.minecraft.network.chat.PlayerChatMessage message() {
+        return this.message;
+    }
+}
+