--- net/minecraft/network/chat/RemoteChatSession.java
+++ net/minecraft/network/chat/RemoteChatSession.java
@@ -1,0 +1,66 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.RemoteChatSession
+extends java.lang.Record
+ {
+    public java.util.UUID sessionId;
+    @javax.annotation.Nullable
+    public net.minecraft.world.entity.player.ProfilePublicKey profilePublicKey;
+    public static net.minecraft.network.chat.RemoteChatSession UNVERIFIED;
+
+    public net.minecraft.network.chat.RemoteChatSession(java.util.UUID $$0, @javax.annotation.Nullable net.minecraft.world.entity.player.ProfilePublicKey $$1) {
+        this.sessionId = $$0;
+        this.profilePublicKey = $$1;
+    }
+
+    public net.minecraft.network.chat.SignedMessageValidator createMessageValidator() {
+        if (this.profilePublicKey != null) {
+            return new net.minecraft.network.chat.SignedMessageValidator$KeyBased(this.profilePublicKey.createSignatureValidator());
+        }
+        return net.minecraft.network.chat.SignedMessageValidator.ACCEPT_UNSIGNED;
+    }
+
+    public net.minecraft.network.chat.SignedMessageChain$Decoder createMessageDecoder(java.util.UUID $$0) {
+        if (this.profilePublicKey != null) {
+            return new net.minecraft.network.chat.SignedMessageChain($$0, this.sessionId).decoder(this.profilePublicKey);
+        }
+        return net.minecraft.network.chat.SignedMessageChain$Decoder.unsigned((java.util.UUID)$$0);
+    }
+
+    public net.minecraft.network.chat.RemoteChatSession$Data asData() {
+        return new net.minecraft.network.chat.RemoteChatSession$Data(this.sessionId, (net.minecraft.world.entity.player.ProfilePublicKey$Data)net.minecraft.Util.mapNullable((java.lang.Object)this.profilePublicKey, net.minecraft.world.entity.player.ProfilePublicKey::data));
+    }
+
+    public boolean verifiable() {
+        return this.profilePublicKey != null;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.RemoteChatSession.class, "sessionId;profilePublicKey", "sessionId", "profilePublicKey"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.RemoteChatSession.class, "sessionId;profilePublicKey", "sessionId", "profilePublicKey"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.RemoteChatSession.class, "sessionId;profilePublicKey", "sessionId", "profilePublicKey"}, this, $$0);
+    }
+
+    public java.util.UUID sessionId() {
+        return this.sessionId;
+    }
+
+    @javax.annotation.Nullable
+    public net.minecraft.world.entity.player.ProfilePublicKey profilePublicKey() {
+        return this.profilePublicKey;
+    }
+
+    static {
+        UNVERIFIED = new net.minecraft.network.chat.RemoteChatSession(net.minecraft.Util.NIL_UUID, null);
+    }
+}
+