--- net/minecraft/network/chat/SignedMessageLink.java
+++ net/minecraft/network/chat/SignedMessageLink.java
@@ -1,0 +1,69 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.SignedMessageLink
+extends java.lang.Record
+ {
+    public int index;
+    public java.util.UUID sender;
+    public java.util.UUID sessionId;
+
+    public net.minecraft.network.chat.SignedMessageLink(int $$0, java.util.UUID $$1, java.util.UUID $$2) {
+        this.index = $$0;
+        this.sender = $$1;
+        this.sessionId = $$2;
+    }
+
+    public static net.minecraft.network.chat.SignedMessageLink unsigned(java.util.UUID $$0) {
+        return net.minecraft.network.chat.SignedMessageLink.root((java.util.UUID)$$0, (java.util.UUID)net.minecraft.Util.NIL_UUID);
+    }
+
+    public static net.minecraft.network.chat.SignedMessageLink root(java.util.UUID $$0, java.util.UUID $$1) {
+        return new net.minecraft.network.chat.SignedMessageLink(0, $$0, $$1);
+    }
+
+    public void updateSignature(net.minecraft.util.SignatureUpdater$Output $$0) throws java.security.SignatureException {
+        $$0.update(net.minecraft.core.UUIDUtil.uuidToByteArray((java.util.UUID)this.sender));
+        $$0.update(net.minecraft.core.UUIDUtil.uuidToByteArray((java.util.UUID)this.sessionId));
+        $$0.update(com.google.common.primitives.Ints.toByteArray((int)this.index));
+    }
+
+    public boolean isDescendantOf(net.minecraft.network.chat.SignedMessageLink $$0) {
+        return this.index > $$0.index() && this.sender.equals((java.lang.Object)$$0.sender()) && this.sessionId.equals((java.lang.Object)$$0.sessionId());
+    }
+
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.SignedMessageLink advance() {
+        if (this.index == java.lang.Integer.MAX_VALUE) {
+            return null;
+        }
+        return new net.minecraft.network.chat.SignedMessageLink(this.index + 1, this.sender, this.sessionId);
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignedMessageLink.class, "index;sender;sessionId", "index", "sender", "sessionId"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignedMessageLink.class, "index;sender;sessionId", "index", "sender", "sessionId"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignedMessageLink.class, "index;sender;sessionId", "index", "sender", "sessionId"}, this, $$0);
+    }
+
+    public int index() {
+        return this.index;
+    }
+
+    public java.util.UUID sender() {
+        return this.sender;
+    }
+
+    public java.util.UUID sessionId() {
+        return this.sessionId;
+    }
+}
+