--- net/minecraft/network/chat/SignableCommand$Argument.java
+++ net/minecraft/network/chat/SignableCommand$Argument.java
@@ -1,0 +1,41 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.SignableCommand$Argument<S>
+extends java.lang.Record
+ {
+    public com.mojang.brigadier.tree.ArgumentCommandNode<S, ?> node;
+    public java.lang.String value;
+
+    public net.minecraft.network.chat.SignableCommand$Argument(com.mojang.brigadier.tree.ArgumentCommandNode<S, ?> $$0, java.lang.String $$1) {
+        this.node = $$0;
+        this.value = $$1;
+    }
+
+    public java.lang.String name() {
+        return this.node.getName();
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignableCommand$Argument.class, "node;value", "node", "value"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignableCommand$Argument.class, "node;value", "node", "value"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignableCommand$Argument.class, "node;value", "node", "value"}, this, $$0);
+    }
+
+    public com.mojang.brigadier.tree.ArgumentCommandNode<S, ?> node() {
+        return this.node;
+    }
+
+    public java.lang.String value() {
+        return this.value;
+    }
+}
+