--- net/minecraft/network/chat/SignedMessageBody$Packed.java
+++ net/minecraft/network/chat/SignedMessageBody$Packed.java
@@ -1,0 +1,64 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.SignedMessageBody$Packed
+extends java.lang.Record
+ {
+    public java.lang.String content;
+    public java.time.Instant timeStamp;
+    public long salt;
+    public net.minecraft.network.chat.LastSeenMessages$Packed lastSeen;
+
+    public net.minecraft.network.chat.SignedMessageBody$Packed(net.minecraft.network.FriendlyByteBuf $$0) {
+        this($$0.readUtf(256), $$0.readInstant(), $$0.readLong(), new net.minecraft.network.chat.LastSeenMessages$Packed($$0));
+    }
+
+    public net.minecraft.network.chat.SignedMessageBody$Packed(java.lang.String $$0, java.time.Instant $$1, long $$2, net.minecraft.network.chat.LastSeenMessages$Packed $$3) {
+        this.content = $$0;
+        this.timeStamp = $$1;
+        this.salt = $$2;
+        this.lastSeen = $$3;
+    }
+
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUtf(this.content, 256);
+        $$0.writeInstant(this.timeStamp);
+        $$0.writeLong(this.salt);
+        this.lastSeen.write($$0);
+    }
+
+    public java.util.Optional<net.minecraft.network.chat.SignedMessageBody> unpack(net.minecraft.network.chat.MessageSignature$Unpacker $$02) {
+        return this.lastSeen.unpack($$02).map($$0 -> new net.minecraft.network.chat.SignedMessageBody(this.content, this.timeStamp, this.salt, $$0));
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignedMessageBody$Packed.class, "content;timeStamp;salt;lastSeen", "content", "timeStamp", "salt", "lastSeen"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignedMessageBody$Packed.class, "content;timeStamp;salt;lastSeen", "content", "timeStamp", "salt", "lastSeen"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.chat.SignedMessageBody$Packed.class, "content;timeStamp;salt;lastSeen", "content", "timeStamp", "salt", "lastSeen"}, this, $$0);
+    }
+
+    public java.lang.String content() {
+        return this.content;
+    }
+
+    public java.time.Instant timeStamp() {
+        return this.timeStamp;
+    }
+
+    public long salt() {
+        return this.salt;
+    }
+
+    public net.minecraft.network.chat.LastSeenMessages$Packed lastSeen() {
+        return this.lastSeen;
+    }
+}
+