--- net/minecraft/network/chat/SignedMessageChain$DecodeException.java
+++ net/minecraft/network/chat/SignedMessageChain$DecodeException.java
@@ -1,0 +1,20 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.chat;
+
+public class net.minecraft.network.chat.SignedMessageChain$DecodeException
+extends net.minecraft.network.chat.ThrowingComponent
+ {
+    public boolean shouldDisconnect;
+
+    public net.minecraft.network.chat.SignedMessageChain$DecodeException(net.minecraft.network.chat.Component $$0, boolean $$1) {
+        super($$0);
+        this.shouldDisconnect = $$1;
+    }
+
+    public boolean shouldDisconnect() {
+        return this.shouldDisconnect;
+    }
+}
+