--- net/minecraft/network/chat/Style.java
+++ net/minecraft/network/chat/Style.java
@@ -33,17 +33,17 @@
         return new net.minecraft.network.chat.Style((net.minecraft.network.chat.TextColor)$$0.orElse(null), (java.lang.Boolean)$$1.orElse(null), (java.lang.Boolean)$$2.orElse(null), (java.lang.Boolean)$$3.orElse(null), (java.lang.Boolean)$$4.orElse(null), (java.lang.Boolean)$$5.orElse(null), null, null, (java.lang.String)$$6.orElse(null), (net.minecraft.resources.ResourceLocation)$$7.orElse(null));
     }
 
-    public net.minecraft.network.chat.Style(@javax.annotation.Nullable net.minecraft.network.chat.TextColor color, @javax.annotation.Nullable java.lang.Boolean bold, @javax.annotation.Nullable java.lang.Boolean italic, @javax.annotation.Nullable java.lang.Boolean underlined, @javax.annotation.Nullable java.lang.Boolean strikethrough, @javax.annotation.Nullable java.lang.Boolean obfuscated, @javax.annotation.Nullable net.minecraft.network.chat.ClickEvent clickEvent, @javax.annotation.Nullable net.minecraft.network.chat.HoverEvent hoverEvent, @javax.annotation.Nullable java.lang.String insertion, @javax.annotation.Nullable net.minecraft.resources.ResourceLocation font) {
-        this.color = color;
-        this.bold = bold;
-        this.italic = italic;
-        this.underlined = underlined;
-        this.strikethrough = strikethrough;
-        this.obfuscated = obfuscated;
-        this.clickEvent = clickEvent;
-        this.hoverEvent = hoverEvent;
-        this.insertion = insertion;
-        this.font = font;
+    public net.minecraft.network.chat.Style(@javax.annotation.Nullable net.minecraft.network.chat.TextColor $$0, @javax.annotation.Nullable java.lang.Boolean $$1, @javax.annotation.Nullable java.lang.Boolean $$2, @javax.annotation.Nullable java.lang.Boolean $$3, @javax.annotation.Nullable java.lang.Boolean $$4, @javax.annotation.Nullable java.lang.Boolean $$5, @javax.annotation.Nullable net.minecraft.network.chat.ClickEvent $$6, @javax.annotation.Nullable net.minecraft.network.chat.HoverEvent $$7, @javax.annotation.Nullable java.lang.String $$8, @javax.annotation.Nullable net.minecraft.resources.ResourceLocation $$9) {
+        this.color = $$0;
+        this.bold = $$1;
+        this.italic = $$2;
+        this.underlined = $$3;
+        this.strikethrough = $$4;
+        this.obfuscated = $$5;
+        this.clickEvent = $$6;
+        this.hoverEvent = $$7;
+        this.insertion = $$8;
+        this.font = $$9;
     }
 
     @javax.annotation.Nullable
@@ -94,28 +94,28 @@
         return this.font != null ? this.font : DEFAULT_FONT;
     }
 
-    public net.minecraft.network.chat.Style withColor(@javax.annotation.Nullable net.minecraft.network.chat.TextColor color) {
-        return new net.minecraft.network.chat.Style(color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
+    public net.minecraft.network.chat.Style withColor(@javax.annotation.Nullable net.minecraft.network.chat.TextColor $$0) {
+        return new net.minecraft.network.chat.Style($$0, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style withColor(@javax.annotation.Nullable net.minecraft.ChatFormatting formatting) {
-        return this.withColor(formatting != null ? net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)formatting) : null);
+    public net.minecraft.network.chat.Style withColor(@javax.annotation.Nullable net.minecraft.ChatFormatting $$0) {
+        return this.withColor($$0 != null ? net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)$$0) : null);
     }
 
     public net.minecraft.network.chat.Style withColor(int $$0) {
         return this.withColor(net.minecraft.network.chat.TextColor.fromRgb((int)$$0));
     }
 
-    public net.minecraft.network.chat.Style withBold(@javax.annotation.Nullable java.lang.Boolean bold) {
-        return new net.minecraft.network.chat.Style(this.color, bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
+    public net.minecraft.network.chat.Style withBold(@javax.annotation.Nullable java.lang.Boolean $$0) {
+        return new net.minecraft.network.chat.Style(this.color, $$0, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style withItalic(@javax.annotation.Nullable java.lang.Boolean italic) {
-        return new net.minecraft.network.chat.Style(this.color, this.bold, italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
+    public net.minecraft.network.chat.Style withItalic(@javax.annotation.Nullable java.lang.Boolean $$0) {
+        return new net.minecraft.network.chat.Style(this.color, this.bold, $$0, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style withUnderlined(@javax.annotation.Nullable java.lang.Boolean underlined) {
-        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
+    public net.minecraft.network.chat.Style withUnderlined(@javax.annotation.Nullable java.lang.Boolean $$0) {
+        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, $$0, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
     public net.minecraft.network.chat.Style withStrikethrough(@javax.annotation.Nullable java.lang.Boolean $$0) {
@@ -126,151 +126,151 @@
         return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, $$0, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style withClickEvent(@javax.annotation.Nullable net.minecraft.network.chat.ClickEvent clickEvent) {
-        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, clickEvent, this.hoverEvent, this.insertion, this.font);
+    public net.minecraft.network.chat.Style withClickEvent(@javax.annotation.Nullable net.minecraft.network.chat.ClickEvent $$0) {
+        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, $$0, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style withHoverEvent(@javax.annotation.Nullable net.minecraft.network.chat.HoverEvent hoverEvent) {
-        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, hoverEvent, this.insertion, this.font);
+    public net.minecraft.network.chat.Style withHoverEvent(@javax.annotation.Nullable net.minecraft.network.chat.HoverEvent $$0) {
+        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, $$0, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style withInsertion(@javax.annotation.Nullable java.lang.String insertion) {
-        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, insertion, this.font);
+    public net.minecraft.network.chat.Style withInsertion(@javax.annotation.Nullable java.lang.String $$0) {
+        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, $$0, this.font);
     }
 
-    public net.minecraft.network.chat.Style withFont(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation fontId) {
-        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, fontId);
+    public net.minecraft.network.chat.Style withFont(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation $$0) {
+        return new net.minecraft.network.chat.Style(this.color, this.bold, this.italic, this.underlined, this.strikethrough, this.obfuscated, this.clickEvent, this.hoverEvent, this.insertion, $$0);
     }
 
-    public net.minecraft.network.chat.Style applyFormat(net.minecraft.ChatFormatting formatting) {
-        net.minecraft.network.chat.TextColor textColor = this.color;
-        java.lang.Boolean bl = this.bold;
-        java.lang.Boolean bl2 = this.italic;
-        java.lang.Boolean bl3 = this.strikethrough;
-        java.lang.Boolean bl4 = this.underlined;
-        java.lang.Boolean bl5 = this.obfuscated;
-        switch (net.minecraft.network.chat.Style$1.$SwitchMap$net$minecraft$ChatFormatting[formatting.ordinal()]) {
+    public net.minecraft.network.chat.Style applyFormat(net.minecraft.ChatFormatting $$0) {
+        net.minecraft.network.chat.TextColor $$1 = this.color;
+        java.lang.Boolean $$2 = this.bold;
+        java.lang.Boolean $$3 = this.italic;
+        java.lang.Boolean $$4 = this.strikethrough;
+        java.lang.Boolean $$5 = this.underlined;
+        java.lang.Boolean $$6 = this.obfuscated;
+        switch (net.minecraft.network.chat.Style$1.$SwitchMap$net$minecraft$ChatFormatting[$$0.ordinal()]) {
             case 1: {
-                bl5 = true;
+                $$6 = true;
                 break;
             }
             case 2: {
-                bl = true;
+                $$2 = true;
                 break;
             }
             case 3: {
-                bl3 = true;
+                $$4 = true;
                 break;
             }
             case 4: {
-                bl4 = true;
+                $$5 = true;
                 break;
             }
             case 5: {
-                bl2 = true;
+                $$3 = true;
                 break;
             }
             case 6: {
                 return EMPTY;
             }
             default: {
-                textColor = net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)formatting);
+                $$1 = net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)$$0);
             }
         }
-        return new net.minecraft.network.chat.Style(textColor, bl, bl2, bl4, bl3, bl5, this.clickEvent, this.hoverEvent, this.insertion, this.font);
+        return new net.minecraft.network.chat.Style($$1, $$2, $$3, $$5, $$4, $$6, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style applyLegacyFormat(net.minecraft.ChatFormatting formatting) {
-        net.minecraft.network.chat.TextColor textColor = this.color;
-        java.lang.Boolean bl = this.bold;
-        java.lang.Boolean bl2 = this.italic;
-        java.lang.Boolean bl3 = this.strikethrough;
-        java.lang.Boolean bl4 = this.underlined;
-        java.lang.Boolean bl5 = this.obfuscated;
-        switch (net.minecraft.network.chat.Style$1.$SwitchMap$net$minecraft$ChatFormatting[formatting.ordinal()]) {
+    public net.minecraft.network.chat.Style applyLegacyFormat(net.minecraft.ChatFormatting $$0) {
+        net.minecraft.network.chat.TextColor $$1 = this.color;
+        java.lang.Boolean $$2 = this.bold;
+        java.lang.Boolean $$3 = this.italic;
+        java.lang.Boolean $$4 = this.strikethrough;
+        java.lang.Boolean $$5 = this.underlined;
+        java.lang.Boolean $$6 = this.obfuscated;
+        switch (net.minecraft.network.chat.Style$1.$SwitchMap$net$minecraft$ChatFormatting[$$0.ordinal()]) {
             case 1: {
-                bl5 = true;
+                $$6 = true;
                 break;
             }
             case 2: {
-                bl = true;
+                $$2 = true;
                 break;
             }
             case 3: {
-                bl3 = true;
+                $$4 = true;
                 break;
             }
             case 4: {
-                bl4 = true;
+                $$5 = true;
                 break;
             }
             case 5: {
-                bl2 = true;
+                $$3 = true;
                 break;
             }
             case 6: {
                 return EMPTY;
             }
             default: {
-                bl5 = false;
-                bl = false;
-                bl3 = false;
-                bl4 = false;
-                bl2 = false;
-                textColor = net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)formatting);
+                $$6 = false;
+                $$2 = false;
+                $$4 = false;
+                $$5 = false;
+                $$3 = false;
+                $$1 = net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)$$0);
             }
         }
-        return new net.minecraft.network.chat.Style(textColor, bl, bl2, bl4, bl3, bl5, this.clickEvent, this.hoverEvent, this.insertion, this.font);
+        return new net.minecraft.network.chat.Style($$1, $$2, $$3, $$5, $$4, $$6, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style applyFormats(net.minecraft.ChatFormatting ... formats) {
-        net.minecraft.network.chat.TextColor textColor = this.color;
-        java.lang.Boolean bl = this.bold;
-        java.lang.Boolean bl2 = this.italic;
-        java.lang.Boolean bl3 = this.strikethrough;
-        java.lang.Boolean bl4 = this.underlined;
-        java.lang.Boolean bl5 = this.obfuscated;
-block8:        for (net.minecraft.ChatFormatting chatFormatting : formats) {
-            switch (net.minecraft.network.chat.Style$1.$SwitchMap$net$minecraft$ChatFormatting[chatFormatting.ordinal()]) {
+    public net.minecraft.network.chat.Style applyFormats(net.minecraft.ChatFormatting ... $$0) {
+        net.minecraft.network.chat.TextColor $$1 = this.color;
+        java.lang.Boolean $$2 = this.bold;
+        java.lang.Boolean $$3 = this.italic;
+        java.lang.Boolean $$4 = this.strikethrough;
+        java.lang.Boolean $$5 = this.underlined;
+        java.lang.Boolean $$6 = this.obfuscated;
+block8:        for (net.minecraft.ChatFormatting $$7 : $$0) {
+            switch (net.minecraft.network.chat.Style$1.$SwitchMap$net$minecraft$ChatFormatting[$$7.ordinal()]) {
                 case 1: {
-                    bl5 = true;
+                    $$6 = true;
                     continue block8;
                 }
                 case 2: {
-                    bl = true;
+                    $$2 = true;
                     continue block8;
                 }
                 case 3: {
-                    bl3 = true;
+                    $$4 = true;
                     continue block8;
                 }
                 case 4: {
-                    bl4 = true;
+                    $$5 = true;
                     continue block8;
                 }
                 case 5: {
-                    bl2 = true;
+                    $$3 = true;
                     continue block8;
                 }
                 case 6: {
                     return EMPTY;
                 }
                 default: {
-                    textColor = net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)chatFormatting);
+                    $$1 = net.minecraft.network.chat.TextColor.fromLegacyFormat((net.minecraft.ChatFormatting)$$7);
                 }
             }
         }
-        return new net.minecraft.network.chat.Style(textColor, bl, bl2, bl4, bl3, bl5, this.clickEvent, this.hoverEvent, this.insertion, this.font);
+        return new net.minecraft.network.chat.Style($$1, $$2, $$3, $$5, $$4, $$6, this.clickEvent, this.hoverEvent, this.insertion, this.font);
     }
 
-    public net.minecraft.network.chat.Style applyTo(net.minecraft.network.chat.Style style) {
+    public net.minecraft.network.chat.Style applyTo(net.minecraft.network.chat.Style $$0) {
         if (this == EMPTY) {
-            return style;
+            return $$0;
         }
-        if (style == EMPTY) {
+        if ($$0 == EMPTY) {
             return this;
         }
-        return new net.minecraft.network.chat.Style(this.color != null ? this.color : style.color, this.bold != null ? this.bold : style.bold, this.italic != null ? this.italic : style.italic, this.underlined != null ? this.underlined : style.underlined, this.strikethrough != null ? this.strikethrough : style.strikethrough, this.obfuscated != null ? this.obfuscated : style.obfuscated, this.clickEvent != null ? this.clickEvent : style.clickEvent, this.hoverEvent != null ? this.hoverEvent : style.hoverEvent, this.insertion != null ? this.insertion : style.insertion, this.font != null ? this.font : style.font);
+        return new net.minecraft.network.chat.Style(this.color != null ? this.color : $$0.color, this.bold != null ? this.bold : $$0.bold, this.italic != null ? this.italic : $$0.italic, this.underlined != null ? this.underlined : $$0.underlined, this.strikethrough != null ? this.strikethrough : $$0.strikethrough, this.obfuscated != null ? this.obfuscated : $$0.obfuscated, this.clickEvent != null ? this.clickEvent : $$0.clickEvent, this.hoverEvent != null ? this.hoverEvent : $$0.hoverEvent, this.insertion != null ? this.insertion : $$0.insertion, this.font != null ? this.font : $$0.font);
     }
 
     public java.lang.String toString() {