--- net/minecraft/network/chat/SubStringSource.java
+++ net/minecraft/network/chat/SubStringSource.java
@@ -9,48 +9,48 @@
     public java.util.List<net.minecraft.network.chat.Style> charStyles;
     public it.unimi.dsi.fastutil.ints.Int2IntFunction reverseCharModifier;
 
-    public net.minecraft.network.chat.SubStringSource(java.lang.String plainText, java.util.List<net.minecraft.network.chat.Style> charStyles, it.unimi.dsi.fastutil.ints.Int2IntFunction reverseCharModifier) {
-        this.plainText = plainText;
-        this.charStyles = com.google.common.collect.ImmutableList.copyOf(charStyles);
-        this.reverseCharModifier = reverseCharModifier;
+    public net.minecraft.network.chat.SubStringSource(java.lang.String $$0, java.util.List<net.minecraft.network.chat.Style> $$1, it.unimi.dsi.fastutil.ints.Int2IntFunction $$2) {
+        this.plainText = $$0;
+        this.charStyles = com.google.common.collect.ImmutableList.copyOf($$1);
+        this.reverseCharModifier = $$2;
     }
 
     public java.lang.String getPlainText() {
         return this.plainText;
     }
 
-    public java.util.List<net.minecraft.util.FormattedCharSequence> substring(int fromIndex, int toIndex, boolean reversed) {
-        if (toIndex == 0) {
+    public java.util.List<net.minecraft.util.FormattedCharSequence> substring(int $$0, int $$1, boolean $$2) {
+        if ($$1 == 0) {
             return com.google.common.collect.ImmutableList.of();
         }
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        net.minecraft.network.chat.Style style = (net.minecraft.network.chat.Style)this.charStyles.get(fromIndex);
-        int n = fromIndex;
-        for (int i = 1; i < toIndex; ++i) {
-            int n2 = fromIndex + i;
-            net.minecraft.network.chat.Style style2 = (net.minecraft.network.chat.Style)this.charStyles.get(n2);
-            if (style2.equals((java.lang.Object)style)) continue;
-            java.lang.String string = this.plainText.substring(n, n2);
-            arrayList.add((java.lang.Object)(reversed ? net.minecraft.util.FormattedCharSequence.backward((java.lang.String)string, (net.minecraft.network.chat.Style)style, (it.unimi.dsi.fastutil.ints.Int2IntFunction)this.reverseCharModifier) : net.minecraft.util.FormattedCharSequence.forward((java.lang.String)string, (net.minecraft.network.chat.Style)style)));
-            style = style2;
-            n = n2;
+        java.util.ArrayList $$3 = com.google.common.collect.Lists.newArrayList();
+        net.minecraft.network.chat.Style $$4 = (net.minecraft.network.chat.Style)this.charStyles.get($$0);
+        int $$5 = $$0;
+        for (int $$6 = 1; $$6 < $$1; ++$$6) {
+            int $$7 = $$0 + $$6;
+            net.minecraft.network.chat.Style $$8 = (net.minecraft.network.chat.Style)this.charStyles.get($$7);
+            if ($$8.equals((java.lang.Object)$$4)) continue;
+            java.lang.String $$9 = this.plainText.substring($$5, $$7);
+            $$3.add((java.lang.Object)($$2 ? net.minecraft.util.FormattedCharSequence.backward((java.lang.String)$$9, (net.minecraft.network.chat.Style)$$4, (it.unimi.dsi.fastutil.ints.Int2IntFunction)this.reverseCharModifier) : net.minecraft.util.FormattedCharSequence.forward((java.lang.String)$$9, (net.minecraft.network.chat.Style)$$4)));
+            $$4 = $$8;
+            $$5 = $$7;
         }
-        if (n < fromIndex + toIndex) {
-            java.lang.String string = this.plainText.substring(n, fromIndex + toIndex);
-            arrayList.add((java.lang.Object)(reversed ? net.minecraft.util.FormattedCharSequence.backward((java.lang.String)string, (net.minecraft.network.chat.Style)style, (it.unimi.dsi.fastutil.ints.Int2IntFunction)this.reverseCharModifier) : net.minecraft.util.FormattedCharSequence.forward((java.lang.String)string, (net.minecraft.network.chat.Style)style)));
+        if ($$5 < $$0 + $$1) {
+            java.lang.String $$10 = this.plainText.substring($$5, $$0 + $$1);
+            $$3.add((java.lang.Object)($$2 ? net.minecraft.util.FormattedCharSequence.backward((java.lang.String)$$10, (net.minecraft.network.chat.Style)$$4, (it.unimi.dsi.fastutil.ints.Int2IntFunction)this.reverseCharModifier) : net.minecraft.util.FormattedCharSequence.forward((java.lang.String)$$10, (net.minecraft.network.chat.Style)$$4)));
         }
-        return reversed ? com.google.common.collect.Lists.reverse((java.util.List)arrayList) : arrayList;
+        return $$2 ? com.google.common.collect.Lists.reverse((java.util.List)$$3) : $$3;
     }
 
     public static net.minecraft.network.chat.SubStringSource create(net.minecraft.network.chat.FormattedText $$02) {
         return net.minecraft.network.chat.SubStringSource.create((net.minecraft.network.chat.FormattedText)$$02, $$0 -> $$0, $$0 -> $$0);
     }
 
-    public static net.minecraft.network.chat.SubStringSource create(net.minecraft.network.chat.FormattedText formattedText, it.unimi.dsi.fastutil.ints.Int2IntFunction reverseCharModifier, java.util.function.UnaryOperator<java.lang.String> textTransformer) {
-        java.lang.StringBuilder stringBuilder = new java.lang.StringBuilder();
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        formattedText.visit((arg_0, arg_1) -> net.minecraft.network.chat.SubStringSource.lambda$create$3(stringBuilder, (java.util.List)arrayList, arg_0, arg_1), net.minecraft.network.chat.Style.EMPTY);
-        return new net.minecraft.network.chat.SubStringSource((java.lang.String)textTransformer.apply((java.lang.Object)stringBuilder.toString()), (java.util.List)arrayList, reverseCharModifier);
+    public static net.minecraft.network.chat.SubStringSource create(net.minecraft.network.chat.FormattedText $$0, it.unimi.dsi.fastutil.ints.Int2IntFunction $$1, java.util.function.UnaryOperator<java.lang.String> $$2) {
+        java.lang.StringBuilder $$3 = new java.lang.StringBuilder();
+        java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayList();
+        $$0.visit((arg_0, arg_1) -> net.minecraft.network.chat.SubStringSource.lambda$create$3($$3, (java.util.List)$$4, arg_0, arg_1), net.minecraft.network.chat.Style.EMPTY);
+        return new net.minecraft.network.chat.SubStringSource((java.lang.String)$$2.apply((java.lang.Object)$$3.toString()), (java.util.List)$$4, $$1);
     }
 
     public static /* synthetic */ java.util.Optional lambda$create$3(java.lang.StringBuilder $$0, java.util.List $$1, net.minecraft.network.chat.Style $$22, java.lang.String $$32) {