--- net/minecraft/network/chat/TextColor.java
+++ net/minecraft/network/chat/TextColor.java
@@ -13,13 +13,13 @@
     @javax.annotation.Nullable
     public java.lang.String name;
 
-    public net.minecraft.network.chat.TextColor(int value, java.lang.String name) {
-        this.value = value;
-        this.name = name;
+    public net.minecraft.network.chat.TextColor(int $$0, java.lang.String $$1) {
+        this.value = $$0;
+        this.name = $$1;
     }
 
-    public net.minecraft.network.chat.TextColor(int value) {
-        this.value = value;
+    public net.minecraft.network.chat.TextColor(int $$0) {
+        this.value = $$0;
         this.name = null;
     }
 
@@ -58,26 +58,26 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.network.chat.TextColor fromLegacyFormat(net.minecraft.ChatFormatting formatting) {
-        return (net.minecraft.network.chat.TextColor)LEGACY_FORMAT_TO_COLOR.get((java.lang.Object)formatting);
+    public static net.minecraft.network.chat.TextColor fromLegacyFormat(net.minecraft.ChatFormatting $$0) {
+        return (net.minecraft.network.chat.TextColor)LEGACY_FORMAT_TO_COLOR.get((java.lang.Object)$$0);
     }
 
-    public static net.minecraft.network.chat.TextColor fromRgb(int color) {
-        return new net.minecraft.network.chat.TextColor(color);
+    public static net.minecraft.network.chat.TextColor fromRgb(int $$0) {
+        return new net.minecraft.network.chat.TextColor($$0);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.network.chat.TextColor parseColor(java.lang.String hexString) {
-        if (hexString.startsWith("#")) {
+    public static net.minecraft.network.chat.TextColor parseColor(java.lang.String $$0) {
+        if ($$0.startsWith("#")) {
             try {
-                int n = java.lang.Integer.parseInt((java.lang.String)hexString.substring(1), (int)16);
-                return net.minecraft.network.chat.TextColor.fromRgb((int)n);
+                int $$1 = java.lang.Integer.parseInt((java.lang.String)$$0.substring(1), (int)16);
+                return net.minecraft.network.chat.TextColor.fromRgb((int)$$1);
             }
-            catch (java.lang.NumberFormatException numberFormatException) {
+            catch (java.lang.NumberFormatException $$2) {
                 return null;
             }
         }
-        return (net.minecraft.network.chat.TextColor)NAMED_COLORS.get((java.lang.Object)hexString);
+        return (net.minecraft.network.chat.TextColor)NAMED_COLORS.get((java.lang.Object)$$0);
     }
 
     public static /* synthetic */ java.lang.String lambda$static$2(net.minecraft.network.chat.TextColor $$0) {