--- net/minecraft/client/gui/components/toasts/AdvancementToast.java
+++ net/minecraft/client/gui/components/toasts/AdvancementToast.java
@@ -9,50 +9,50 @@
     public net.minecraft.advancements.Advancement advancement;
     public boolean playedSound;
 
-    public net.minecraft.client.gui.components.toasts.AdvancementToast(net.minecraft.advancements.Advancement advancement) {
-        this.advancement = advancement;
+    public net.minecraft.client.gui.components.toasts.AdvancementToast(net.minecraft.advancements.Advancement $$0) {
+        this.advancement = $$0;
     }
 
-    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.client.gui.components.toasts.ToastComponent toastComponent, long l) {
+    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.gui.components.toasts.ToastComponent $$1, long $$2) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)TEXTURE);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        net.minecraft.advancements.DisplayInfo displayInfo = this.advancement.getDisplay();
-        toastComponent.blit(poseStack, 0, 0, 0, 0, this.width(), this.height());
-        if (displayInfo != null) {
-            int n;
-            java.util.List list = toastComponent.getMinecraft().font.split((net.minecraft.network.chat.FormattedText)displayInfo.getTitle(), 125);
-            int n2 = n = displayInfo.getFrame() == net.minecraft.advancements.FrameType.CHALLENGE ? 0xFF88FF : 0xFFFF00;
-            if (list.size() == 1) {
-                toastComponent.getMinecraft().font.draw(poseStack, displayInfo.getFrame().getDisplayName(), 30.0f, 7.0f, n | 0xFF000000);
-                toastComponent.getMinecraft().font.draw(poseStack, (net.minecraft.util.FormattedCharSequence)list.get(0), 30.0f, 18.0f, -1);
+        net.minecraft.advancements.DisplayInfo $$3 = this.advancement.getDisplay();
+        $$1.blit($$0, 0, 0, 0, 0, this.width(), this.height());
+        if ($$3 != null) {
+            int $$5;
+            java.util.List $$4 = $$1.getMinecraft().font.split((net.minecraft.network.chat.FormattedText)$$3.getTitle(), 125);
+            int n = $$5 = $$3.getFrame() == net.minecraft.advancements.FrameType.CHALLENGE ? 0xFF88FF : 0xFFFF00;
+            if ($$4.size() == 1) {
+                $$1.getMinecraft().font.draw($$0, $$3.getFrame().getDisplayName(), 30.0f, 7.0f, $$5 | 0xFF000000);
+                $$1.getMinecraft().font.draw($$0, (net.minecraft.util.FormattedCharSequence)$$4.get(0), 30.0f, 18.0f, -1);
             } else {
-                int n3 = 1500;
-                float f = 300.0f;
-                if (l < 1500L) {
-                    int n4 = net.minecraft.util.Mth.floor((float)(net.minecraft.util.Mth.clamp((float)((float)(1500L - l) / 300.0f), (float)0.0f, (float)1.0f) * 255.0f)) << 24 | 0x4000000;
-                    toastComponent.getMinecraft().font.draw(poseStack, displayInfo.getFrame().getDisplayName(), 30.0f, 11.0f, n | n4);
+                int $$6 = 1500;
+                float $$7 = 300.0f;
+                if ($$2 < 1500L) {
+                    int $$8 = net.minecraft.util.Mth.floor((float)(net.minecraft.util.Mth.clamp((float)((float)(1500L - $$2) / 300.0f), (float)0.0f, (float)1.0f) * 255.0f)) << 24 | 0x4000000;
+                    $$1.getMinecraft().font.draw($$0, $$3.getFrame().getDisplayName(), 30.0f, 11.0f, $$5 | $$8);
                 } else {
-                    int n5 = net.minecraft.util.Mth.floor((float)(net.minecraft.util.Mth.clamp((float)((float)(l - 1500L) / 300.0f), (float)0.0f, (float)1.0f) * 252.0f)) << 24 | 0x4000000;
-                    int n6 = this.height() / 2;
-                    int n7 = list.size();
-                    java.util.Objects.requireNonNull((java.lang.Object)toastComponent.getMinecraft().font);
-                    int n8 = n6 - n7 * 9 / 2;
-                    for (net.minecraft.util.FormattedCharSequence formattedCharSequence : list) {
-                        toastComponent.getMinecraft().font.draw(poseStack, formattedCharSequence, 30.0f, (float)n8, 0xFFFFFF | n5);
-                        java.util.Objects.requireNonNull((java.lang.Object)toastComponent.getMinecraft().font);
-                        n8 += 9;
+                    int $$9 = net.minecraft.util.Mth.floor((float)(net.minecraft.util.Mth.clamp((float)((float)($$2 - 1500L) / 300.0f), (float)0.0f, (float)1.0f) * 252.0f)) << 24 | 0x4000000;
+                    int n2 = this.height() / 2;
+                    int n3 = $$4.size();
+                    java.util.Objects.requireNonNull((java.lang.Object)$$1.getMinecraft().font);
+                    int $$10 = n2 - n3 * 9 / 2;
+                    for (net.minecraft.util.FormattedCharSequence $$11 : $$4) {
+                        $$1.getMinecraft().font.draw($$0, $$11, 30.0f, (float)$$10, 0xFFFFFF | $$9);
+                        java.util.Objects.requireNonNull((java.lang.Object)$$1.getMinecraft().font);
+                        $$10 += 9;
                     }
                 }
             }
-            if (!this.playedSound && l > 0L) {
+            if (!this.playedSound && $$2 > 0L) {
                 this.playedSound = true;
-                if (displayInfo.getFrame() == net.minecraft.advancements.FrameType.CHALLENGE) {
-                    toastComponent.getMinecraft().getSoundManager().play((net.minecraft.client.resources.sounds.SoundInstance)net.minecraft.client.resources.sounds.SimpleSoundInstance.forUI((net.minecraft.sounds.SoundEvent)net.minecraft.sounds.SoundEvents.UI_TOAST_CHALLENGE_COMPLETE, (float)1.0f, (float)1.0f));
+                if ($$3.getFrame() == net.minecraft.advancements.FrameType.CHALLENGE) {
+                    $$1.getMinecraft().getSoundManager().play((net.minecraft.client.resources.sounds.SoundInstance)net.minecraft.client.resources.sounds.SimpleSoundInstance.forUI((net.minecraft.sounds.SoundEvent)net.minecraft.sounds.SoundEvents.UI_TOAST_CHALLENGE_COMPLETE, (float)1.0f, (float)1.0f));
                 }
             }
-            toastComponent.getMinecraft().getItemRenderer().renderAndDecorateFakeItem(displayInfo.getIcon(), 8, 8);
-            return l >= 5000L ? net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE : net.minecraft.client.gui.components.toasts.Toast$Visibility.SHOW;
+            $$1.getMinecraft().getItemRenderer().renderAndDecorateFakeItem($$3.getIcon(), 8, 8);
+            return $$2 >= 5000L ? net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE : net.minecraft.client.gui.components.toasts.Toast$Visibility.SHOW;
         }
         return net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE;
     }