--- net/minecraft/client/gui/components/toasts/RecipeToast.java
+++ net/minecraft/client/gui/components/toasts/RecipeToast.java
@@ -13,14 +13,14 @@
     public long lastChanged;
     public boolean changed;
 
-    public net.minecraft.client.gui.components.toasts.RecipeToast(net.minecraft.world.item.crafting.Recipe<?> recipe) {
+    public net.minecraft.client.gui.components.toasts.RecipeToast(net.minecraft.world.item.crafting.Recipe<?> $$0) {
         this.recipes = com.google.common.collect.Lists.newArrayList();
-        this.recipes.add(recipe);
+        this.recipes.add($$0);
     }
 
-    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.client.gui.components.toasts.ToastComponent toastComponent, long l) {
+    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.gui.components.toasts.ToastComponent $$1, long $$2) {
         if (this.changed) {
-            this.lastChanged = l;
+            this.lastChanged = $$2;
             this.changed = false;
         }
         if (this.recipes.isEmpty()) {
@@ -29,33 +29,33 @@
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)TEXTURE);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        toastComponent.blit(poseStack, 0, 0, 0, 32, this.width(), this.height());
-        toastComponent.getMinecraft().font.draw(poseStack, TITLE_TEXT, 30.0f, 7.0f, -11534256);
-        toastComponent.getMinecraft().font.draw(poseStack, DESCRIPTION_TEXT, 30.0f, 18.0f, -16777216);
-        net.minecraft.world.item.crafting.Recipe recipe = (net.minecraft.world.item.crafting.Recipe)this.recipes.get((int)(l / java.lang.Math.max((long)1L, (long)(5000L / (long)this.recipes.size())) % (long)this.recipes.size()));
-        net.minecraft.world.item.ItemStack itemStack = recipe.getToastSymbol();
-        com.mojang.blaze3d.vertex.PoseStack poseStack2 = com.mojang.blaze3d.systems.RenderSystem.getModelViewStack();
-        poseStack2.pushPose();
-        poseStack2.scale(0.6f, 0.6f, 1.0f);
+        $$1.blit($$0, 0, 0, 0, 32, this.width(), this.height());
+        $$1.getMinecraft().font.draw($$0, TITLE_TEXT, 30.0f, 7.0f, -11534256);
+        $$1.getMinecraft().font.draw($$0, DESCRIPTION_TEXT, 30.0f, 18.0f, -16777216);
+        net.minecraft.world.item.crafting.Recipe $$3 = (net.minecraft.world.item.crafting.Recipe)this.recipes.get((int)($$2 / java.lang.Math.max((long)1L, (long)(5000L / (long)this.recipes.size())) % (long)this.recipes.size()));
+        net.minecraft.world.item.ItemStack $$4 = $$3.getToastSymbol();
+        com.mojang.blaze3d.vertex.PoseStack $$5 = com.mojang.blaze3d.systems.RenderSystem.getModelViewStack();
+        $$5.pushPose();
+        $$5.scale(0.6f, 0.6f, 1.0f);
         com.mojang.blaze3d.systems.RenderSystem.applyModelViewMatrix();
-        toastComponent.getMinecraft().getItemRenderer().renderAndDecorateFakeItem(itemStack, 3, 3);
-        poseStack2.popPose();
+        $$1.getMinecraft().getItemRenderer().renderAndDecorateFakeItem($$4, 3, 3);
+        $$5.popPose();
         com.mojang.blaze3d.systems.RenderSystem.applyModelViewMatrix();
-        toastComponent.getMinecraft().getItemRenderer().renderAndDecorateFakeItem(recipe.getResultItem(), 8, 8);
-        return l - this.lastChanged >= 5000L ? net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE : net.minecraft.client.gui.components.toasts.Toast$Visibility.SHOW;
+        $$1.getMinecraft().getItemRenderer().renderAndDecorateFakeItem($$3.getResultItem(), 8, 8);
+        return $$2 - this.lastChanged >= 5000L ? net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE : net.minecraft.client.gui.components.toasts.Toast$Visibility.SHOW;
     }
 
-    public void addItem(net.minecraft.world.item.crafting.Recipe<?> recipe) {
-        this.recipes.add(recipe);
+    public void addItem(net.minecraft.world.item.crafting.Recipe<?> $$0) {
+        this.recipes.add($$0);
         this.changed = true;
     }
 
-    public static void addOrUpdate(net.minecraft.client.gui.components.toasts.ToastComponent toastGui, net.minecraft.world.item.crafting.Recipe<?> recipe) {
-        net.minecraft.client.gui.components.toasts.RecipeToast recipeToast = (net.minecraft.client.gui.components.toasts.RecipeToast)toastGui.getToast(net.minecraft.client.gui.components.toasts.RecipeToast.class, NO_TOKEN);
-        if (recipeToast == null) {
-            toastGui.addToast((net.minecraft.client.gui.components.toasts.Toast)new net.minecraft.client.gui.components.toasts.RecipeToast(recipe));
+    public static void addOrUpdate(net.minecraft.client.gui.components.toasts.ToastComponent $$0, net.minecraft.world.item.crafting.Recipe<?> $$1) {
+        net.minecraft.client.gui.components.toasts.RecipeToast $$2 = (net.minecraft.client.gui.components.toasts.RecipeToast)$$0.getToast(net.minecraft.client.gui.components.toasts.RecipeToast.class, NO_TOKEN);
+        if ($$2 == null) {
+            $$0.addToast((net.minecraft.client.gui.components.toasts.Toast)new net.minecraft.client.gui.components.toasts.RecipeToast($$1));
         } else {
-            recipeToast.addItem(recipe);
+            $$2.addItem($$1);
         }
     }
 