--- net/minecraft/client/gui/components/toasts/SystemToast.java
+++ net/minecraft/client/gui/components/toasts/SystemToast.java
@@ -16,26 +16,26 @@
     public boolean changed;
     public int width;
 
-    public net.minecraft.client.gui.components.toasts.SystemToast(net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds id, net.minecraft.network.chat.Component title, @javax.annotation.Nullable net.minecraft.network.chat.Component message) {
-        this(id, title, (java.util.List)net.minecraft.client.gui.components.toasts.SystemToast.nullToEmpty((net.minecraft.network.chat.Component)message), java.lang.Math.max((int)160, (int)(30 + java.lang.Math.max((int)net.minecraft.client.Minecraft.getInstance().font.width((net.minecraft.network.chat.FormattedText)title), (int)(message == null ? 0 : net.minecraft.client.Minecraft.getInstance().font.width((net.minecraft.network.chat.FormattedText)message))))));
+    public net.minecraft.client.gui.components.toasts.SystemToast(net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds $$0, net.minecraft.network.chat.Component $$1, @javax.annotation.Nullable net.minecraft.network.chat.Component $$2) {
+        this($$0, $$1, (java.util.List)net.minecraft.client.gui.components.toasts.SystemToast.nullToEmpty((net.minecraft.network.chat.Component)$$2), java.lang.Math.max((int)160, (int)(30 + java.lang.Math.max((int)net.minecraft.client.Minecraft.getInstance().font.width((net.minecraft.network.chat.FormattedText)$$1), (int)($$2 == null ? 0 : net.minecraft.client.Minecraft.getInstance().font.width((net.minecraft.network.chat.FormattedText)$$2))))));
     }
 
-    public static net.minecraft.client.gui.components.toasts.SystemToast multiline(net.minecraft.client.Minecraft minecraft, net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds id, net.minecraft.network.chat.Component title, net.minecraft.network.chat.Component message) {
-        net.minecraft.client.gui.Font font = minecraft.font;
-        java.util.List list = font.split((net.minecraft.network.chat.FormattedText)message, 200);
-        int n = java.lang.Math.max((int)200, (int)list.stream().mapToInt(arg_0 -> ((net.minecraft.client.gui.Font)font).width(arg_0)).max().orElse(200));
-        return new net.minecraft.client.gui.components.toasts.SystemToast(id, title, list, n + 30);
+    public static net.minecraft.client.gui.components.toasts.SystemToast multiline(net.minecraft.client.Minecraft $$0, net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds $$1, net.minecraft.network.chat.Component $$2, net.minecraft.network.chat.Component $$3) {
+        net.minecraft.client.gui.Font $$4 = $$0.font;
+        java.util.List $$5 = $$4.split((net.minecraft.network.chat.FormattedText)$$3, 200);
+        int $$6 = java.lang.Math.max((int)200, (int)$$5.stream().mapToInt(arg_0 -> ((net.minecraft.client.gui.Font)$$4).width(arg_0)).max().orElse(200));
+        return new net.minecraft.client.gui.components.toasts.SystemToast($$1, $$2, $$5, $$6 + 30);
     }
 
-    public net.minecraft.client.gui.components.toasts.SystemToast(net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds id, net.minecraft.network.chat.Component title, java.util.List<net.minecraft.util.FormattedCharSequence> messageLines, int width) {
-        this.id = id;
-        this.title = title;
-        this.messageLines = messageLines;
-        this.width = width;
+    public net.minecraft.client.gui.components.toasts.SystemToast(net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds $$0, net.minecraft.network.chat.Component $$1, java.util.List<net.minecraft.util.FormattedCharSequence> $$2, int $$3) {
+        this.id = $$0;
+        this.title = $$1;
+        this.messageLines = $$2;
+        this.width = $$3;
     }
 
-    public static com.google.common.collect.ImmutableList<net.minecraft.util.FormattedCharSequence> nullToEmpty(@javax.annotation.Nullable net.minecraft.network.chat.Component message) {
-        return message == null ? com.google.common.collect.ImmutableList.of() : com.google.common.collect.ImmutableList.of((java.lang.Object)message.getVisualOrderText());
+    public static com.google.common.collect.ImmutableList<net.minecraft.util.FormattedCharSequence> nullToEmpty(@javax.annotation.Nullable net.minecraft.network.chat.Component $$0) {
+        return $$0 == null ? com.google.common.collect.ImmutableList.of() : com.google.common.collect.ImmutableList.of((java.lang.Object)$$0.getVisualOrderText());
     }
 
     public int width() {
@@ -43,55 +43,53 @@
     }
 
     public int height() {
-        return 20 + this.messageLines.size() * 12;
+        return 20 + java.lang.Math.max((int)this.messageLines.size(), (int)1) * 12;
     }
 
-    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.client.gui.components.toasts.ToastComponent toastComponent, long l) {
-        int n;
+    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.gui.components.toasts.ToastComponent $$1, long $$2) {
         if (this.changed) {
-            this.lastChanged = l;
+            this.lastChanged = $$2;
             this.changed = false;
         }
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)TEXTURE);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        int n2 = this.width();
-        if (n2 == 160 && this.messageLines.size() <= 1) {
-            toastComponent.blit(poseStack, 0, 0, 0, 64, n2, this.height());
+        int $$3 = this.width();
+        if ($$3 == 160 && this.messageLines.size() <= 1) {
+            $$1.blit($$0, 0, 0, 0, 64, $$3, this.height());
         } else {
-            n = this.height();
-            int n3 = 28;
-            int n4 = java.lang.Math.min((int)4, (int)(n - 28));
-            this.renderBackgroundRow(poseStack, toastComponent, n2, 0, 0, 28);
-            for (int i = 28; i < n - n4; i += 10) {
-                this.renderBackgroundRow(poseStack, toastComponent, n2, 16, i, java.lang.Math.min((int)16, (int)(n - i - n4)));
+            int $$4 = this.height();
+            int $$5 = 28;
+            int $$6 = java.lang.Math.min((int)4, (int)($$4 - 28));
+            this.renderBackgroundRow($$0, $$1, $$3, 0, 0, 28);
+            for (int $$7 = 28; $$7 < $$4 - $$6; $$7 += 10) {
+                this.renderBackgroundRow($$0, $$1, $$3, 16, $$7, java.lang.Math.min((int)16, (int)($$4 - $$7 - $$6)));
             }
-            this.renderBackgroundRow(poseStack, toastComponent, n2, 32 - n4, n - n4, n4);
+            this.renderBackgroundRow($$0, $$1, $$3, 32 - $$6, $$4 - $$6, $$6);
         }
         if (this.messageLines == null) {
-            toastComponent.getMinecraft().font.draw(poseStack, this.title, 18.0f, 12.0f, -256);
+            $$1.getMinecraft().font.draw($$0, this.title, 18.0f, 12.0f, -256);
         } else {
-            toastComponent.getMinecraft().font.draw(poseStack, this.title, 18.0f, 7.0f, -256);
-            for (n = 0;
- n < this.messageLines.size(); ++n) {
-                toastComponent.getMinecraft().font.draw(poseStack, (net.minecraft.util.FormattedCharSequence)this.messageLines.get(n), 18.0f, (float)(18 + n * 12), -1);
+            $$1.getMinecraft().font.draw($$0, this.title, 18.0f, 7.0f, -256);
+            for (int $$8 = 0; $$8 < this.messageLines.size(); ++$$8) {
+                $$1.getMinecraft().font.draw($$0, (net.minecraft.util.FormattedCharSequence)this.messageLines.get($$8), 18.0f, (float)(18 + $$8 * 12), -1);
             }
         }
-        return l - this.lastChanged < this.id.displayTime ? net.minecraft.client.gui.components.toasts.Toast$Visibility.SHOW : net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE;
+        return $$2 - this.lastChanged < this.id.displayTime ? net.minecraft.client.gui.components.toasts.Toast$Visibility.SHOW : net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE;
     }
 
-    public void renderBackgroundRow(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.client.gui.components.toasts.ToastComponent toastComponent, int n, int n2, int n3, int n4) {
-        int n5 = n2 == 0 ? 20 : 5;
-        int n6 = java.lang.Math.min((int)60, (int)(n - n5));
-        toastComponent.blit(poseStack, 0, n3, 0, 64 + n2, n5, n4);
-        for (int i = n5; i < n - n6; i += 64) {
-            toastComponent.blit(poseStack, i, n3, 32, 64 + n2, java.lang.Math.min((int)64, (int)(n - i - n6)), n4);
+    public void renderBackgroundRow(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.gui.components.toasts.ToastComponent $$1, int $$2, int $$3, int $$4, int $$5) {
+        int $$6 = $$3 == 0 ? 20 : 5;
+        int $$7 = java.lang.Math.min((int)60, (int)($$2 - $$6));
+        $$1.blit($$0, 0, $$4, 0, 64 + $$3, $$6, $$5);
+        for (int $$8 = $$6; $$8 < $$2 - $$7; $$8 += 64) {
+            $$1.blit($$0, $$8, $$4, 32, 64 + $$3, java.lang.Math.min((int)64, (int)($$2 - $$8 - $$7)), $$5);
         }
-        toastComponent.blit(poseStack, n - n6, n3, 160 - n6, 64 + n2, n6, n4);
+        $$1.blit($$0, $$2 - $$7, $$4, 160 - $$7, 64 + $$3, $$7, $$5);
     }
 
-    public void reset(net.minecraft.network.chat.Component title, @javax.annotation.Nullable net.minecraft.network.chat.Component message) {
-        this.title = title;
-        this.messageLines = net.minecraft.client.gui.components.toasts.SystemToast.nullToEmpty((net.minecraft.network.chat.Component)message);
+    public void reset(net.minecraft.network.chat.Component $$0, @javax.annotation.Nullable net.minecraft.network.chat.Component $$1) {
+        this.title = $$0;
+        this.messageLines = net.minecraft.client.gui.components.toasts.SystemToast.nullToEmpty((net.minecraft.network.chat.Component)$$1);
         this.changed = true;
     }
 
@@ -99,29 +97,29 @@
         return this.id;
     }
 
-    public static void add(net.minecraft.client.gui.components.toasts.ToastComponent toastComponent, net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds id, net.minecraft.network.chat.Component title, @javax.annotation.Nullable net.minecraft.network.chat.Component message) {
-        toastComponent.addToast((net.minecraft.client.gui.components.toasts.Toast)new net.minecraft.client.gui.components.toasts.SystemToast(id, title, message));
+    public static void add(net.minecraft.client.gui.components.toasts.ToastComponent $$0, net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds $$1, net.minecraft.network.chat.Component $$2, @javax.annotation.Nullable net.minecraft.network.chat.Component $$3) {
+        $$0.addToast((net.minecraft.client.gui.components.toasts.Toast)new net.minecraft.client.gui.components.toasts.SystemToast($$1, $$2, $$3));
     }
 
-    public static void addOrUpdate(net.minecraft.client.gui.components.toasts.ToastComponent toastComponent, net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds id, net.minecraft.network.chat.Component title, @javax.annotation.Nullable net.minecraft.network.chat.Component message) {
-        net.minecraft.client.gui.components.toasts.SystemToast systemToast = (net.minecraft.client.gui.components.toasts.SystemToast)toastComponent.getToast(net.minecraft.client.gui.components.toasts.SystemToast.class, (java.lang.Object)id);
-        if (systemToast == null) {
-            net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)toastComponent, (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)id, (net.minecraft.network.chat.Component)title, (net.minecraft.network.chat.Component)message);
+    public static void addOrUpdate(net.minecraft.client.gui.components.toasts.ToastComponent $$0, net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds $$1, net.minecraft.network.chat.Component $$2, @javax.annotation.Nullable net.minecraft.network.chat.Component $$3) {
+        net.minecraft.client.gui.components.toasts.SystemToast $$4 = (net.minecraft.client.gui.components.toasts.SystemToast)$$0.getToast(net.minecraft.client.gui.components.toasts.SystemToast.class, (java.lang.Object)$$1);
+        if ($$4 == null) {
+            net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)$$0, (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)$$1, (net.minecraft.network.chat.Component)$$2, (net.minecraft.network.chat.Component)$$3);
         } else {
-            systemToast.reset(title, message);
+            $$4.reset($$2, $$3);
         }
     }
 
-    public static void onWorldAccessFailure(net.minecraft.client.Minecraft minecraft, java.lang.String message) {
-        net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)minecraft.getToasts(), (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds.WORLD_ACCESS_FAILURE, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.access_failure"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)message));
+    public static void onWorldAccessFailure(net.minecraft.client.Minecraft $$0, java.lang.String $$1) {
+        net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)$$0.getToasts(), (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds.WORLD_ACCESS_FAILURE, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.access_failure"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)$$1));
     }
 
-    public static void onWorldDeleteFailure(net.minecraft.client.Minecraft minecraft, java.lang.String message) {
-        net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)minecraft.getToasts(), (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds.WORLD_ACCESS_FAILURE, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.delete_failure"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)message));
+    public static void onWorldDeleteFailure(net.minecraft.client.Minecraft $$0, java.lang.String $$1) {
+        net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)$$0.getToasts(), (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds.WORLD_ACCESS_FAILURE, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.delete_failure"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)$$1));
     }
 
-    public static void onPackCopyFailure(net.minecraft.client.Minecraft minecraft, java.lang.String message) {
-        net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)minecraft.getToasts(), (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds.PACK_COPY_FAILURE, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"pack.copyFailure"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)message));
+    public static void onPackCopyFailure(net.minecraft.client.Minecraft $$0, java.lang.String $$1) {
+        net.minecraft.client.gui.components.toasts.SystemToast.add((net.minecraft.client.gui.components.toasts.ToastComponent)$$0.getToasts(), (net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds)net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds.PACK_COPY_FAILURE, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"pack.copyFailure"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.literal((java.lang.String)$$1));
     }
 
     public /* synthetic */ java.lang.Object getToken() {