--- net/minecraft/client/gui/components/toasts/SystemToast$SystemToastIds.java
+++ net/minecraft/client/gui/components/toasts/SystemToast$SystemToastIds.java
@@ -14,7 +14,6 @@
     public static /* enum */ net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds WORLD_ACCESS_FAILURE;
     public static /* enum */ net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds PACK_COPY_FAILURE;
     public static /* enum */ net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds PERIODIC_NOTIFICATION;
-    public static /* enum */ net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds CHAT_PREVIEW_WARNING;
     public static /* enum */ net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds UNSECURE_SERVER_WARNING;
     public long displayTime;
     public static /* synthetic */ net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds[] $VALUES;
@@ -37,7 +36,7 @@
     }
 
     public static /* synthetic */ net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds[] $values() {
-        return new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds[]{TUTORIAL_HINT, NARRATOR_TOGGLE, WORLD_BACKUP, WORLD_GEN_SETTINGS_TRANSFER, PACK_LOAD_FAILURE, WORLD_ACCESS_FAILURE, PACK_COPY_FAILURE, PERIODIC_NOTIFICATION, CHAT_PREVIEW_WARNING, UNSECURE_SERVER_WARNING};
+        return new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds[]{TUTORIAL_HINT, NARRATOR_TOGGLE, WORLD_BACKUP, WORLD_GEN_SETTINGS_TRANSFER, PACK_LOAD_FAILURE, WORLD_ACCESS_FAILURE, PACK_COPY_FAILURE, PERIODIC_NOTIFICATION, UNSECURE_SERVER_WARNING};
     }
 
     static {
@@ -49,8 +48,7 @@
         WORLD_ACCESS_FAILURE = new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds("WORLD_ACCESS_FAILURE", 5);
         PACK_COPY_FAILURE = new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds("PACK_COPY_FAILURE", 6);
         PERIODIC_NOTIFICATION = new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds("PERIODIC_NOTIFICATION", 7);
-        CHAT_PREVIEW_WARNING = new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds("CHAT_PREVIEW_WARNING", 8, 10000L);
-        UNSECURE_SERVER_WARNING = new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds("UNSECURE_SERVER_WARNING", 9, 10000L);
+        UNSECURE_SERVER_WARNING = new net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds("UNSECURE_SERVER_WARNING", 8, 10000L);
         $VALUES = net.minecraft.client.gui.components.toasts.SystemToast$SystemToastIds.$values();
     }
 }