--- net/minecraft/client/gui/components/toasts/TutorialToast.java
+++ net/minecraft/client/gui/components/toasts/TutorialToast.java
@@ -20,32 +20,37 @@
     public float progress;
     public boolean progressable;
 
-    public net.minecraft.client.gui.components.toasts.TutorialToast(net.minecraft.client.gui.components.toasts.TutorialToast$Icons icon, net.minecraft.network.chat.Component title, @javax.annotation.Nullable net.minecraft.network.chat.Component message, boolean progressable) {
+    public net.minecraft.client.gui.components.toasts.TutorialToast(net.minecraft.client.gui.components.toasts.TutorialToast$Icons $$0, net.minecraft.network.chat.Component $$1, @javax.annotation.Nullable net.minecraft.network.chat.Component $$2, boolean $$3) {
         this.visibility = net.minecraft.client.gui.components.toasts.Toast$Visibility.SHOW;
-        this.icon = icon;
-        this.title = title;
-        this.message = message;
-        this.progressable = progressable;
+        this.icon = $$0;
+        this.title = $$1;
+        this.message = $$2;
+        this.progressable = $$3;
     }
 
-    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.client.gui.components.toasts.ToastComponent toastComponent, long l) {
+    public net.minecraft.client.gui.components.toasts.Toast$Visibility render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.gui.components.toasts.ToastComponent $$1, long $$2) {
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)TEXTURE);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        toastComponent.blit(poseStack, 0, 0, 0, 96, this.width(), this.height());
-        this.icon.render(poseStack, (net.minecraft.client.gui.GuiComponent)toastComponent, 6, 6);
+        $$1.blit($$0, 0, 0, 0, 96, this.width(), this.height());
+        this.icon.render($$0, (net.minecraft.client.gui.GuiComponent)$$1, 6, 6);
         if (this.message == null) {
-            toastComponent.getMinecraft().font.draw(poseStack, this.title, 30.0f, 12.0f, -11534256);
+            $$1.getMinecraft().font.draw($$0, this.title, 30.0f, 12.0f, -11534256);
         } else {
-            toastComponent.getMinecraft().font.draw(poseStack, this.title, 30.0f, 7.0f, -11534256);
-            toastComponent.getMinecraft().font.draw(poseStack, this.message, 30.0f, 18.0f, -16777216);
+            $$1.getMinecraft().font.draw($$0, this.title, 30.0f, 7.0f, -11534256);
+            $$1.getMinecraft().font.draw($$0, this.message, 30.0f, 18.0f, -16777216);
         }
         if (this.progressable) {
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)3, (int)28, (int)157, (int)29, (int)-1);
-            float f = net.minecraft.util.Mth.clampedLerp((float)this.lastProgress, (float)this.progress, (float)((float)(l - this.lastProgressTime) / 100.0f));
-            int n = this.progress >= this.lastProgress ? -16755456 : -11206656;
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)3, (int)28, (int)((int)(3.0f + 154.0f * f)), (int)29, (int)n);
-            this.lastProgress = f;
-            this.lastProgressTime = l;
+            int $$5;
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)3, (int)28, (int)157, (int)29, (int)-1);
+            float $$3 = net.minecraft.util.Mth.clampedLerp((float)this.lastProgress, (float)this.progress, (float)((float)($$2 - this.lastProgressTime) / 100.0f));
+            if (this.progress >= this.lastProgress) {
+                int $$4 = -16755456;
+            } else {
+                $$5 = -11206656;
+            }
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)3, (int)28, (int)((int)(3.0f + 154.0f * $$3)), (int)29, (int)$$5);
+            this.lastProgress = $$3;
+            this.lastProgressTime = $$2;
         }
         return this.visibility;
     }
@@ -54,8 +59,8 @@
         this.visibility = net.minecraft.client.gui.components.toasts.Toast$Visibility.HIDE;
     }
 
-    public void updateProgress(float progress) {
-        this.progress = progress;
+    public void updateProgress(float $$0) {
+        this.progress = $$0;
     }
 }
 