--- net/minecraft/client/gui/components/toasts/TutorialToast$Icons.java
+++ net/minecraft/client/gui/components/toasts/TutorialToast$Icons.java
@@ -25,15 +25,15 @@
         return (net.minecraft.client.gui.components.toasts.TutorialToast$Icons)java.lang.Enum.valueOf(net.minecraft.client.gui.components.toasts.TutorialToast$Icons.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.client.gui.components.toasts.TutorialToast$Icons(int x, int y) {
+    public net.minecraft.client.gui.components.toasts.TutorialToast$Icons(int $$0, int $$1) {
         super(string, n);
-        this.x = x;
-        this.y = y;
+        this.x = $$0;
+        this.y = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack poseStack, net.minecraft.client.gui.GuiComponent guiComponent, int x, int y) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.gui.GuiComponent $$1, int $$2, int $$3) {
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
-        guiComponent.blit(poseStack, x, y, 176 + this.x * 20, this.y * 20, 20, 20);
+        $$1.blit($$0, $$2, $$3, 176 + this.x * 20, this.y * 20, 20, 20);
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
     }
 