--- net/minecraft/world/level/border/BorderChangeListener$DelegateBorderChangeListener.java
+++ net/minecraft/world/level/border/BorderChangeListener$DelegateBorderChangeListener.java
@@ -8,36 +8,36 @@
  {
     public net.minecraft.world.level.border.WorldBorder worldBorder;
 
-    public net.minecraft.world.level.border.BorderChangeListener$DelegateBorderChangeListener(net.minecraft.world.level.border.WorldBorder worldBorder) {
-        this.worldBorder = worldBorder;
+    public net.minecraft.world.level.border.BorderChangeListener$DelegateBorderChangeListener(net.minecraft.world.level.border.WorldBorder $$0) {
+        this.worldBorder = $$0;
     }
 
-    public void onBorderSizeSet(net.minecraft.world.level.border.WorldBorder border, double newSize) {
-        this.worldBorder.setSize(newSize);
+    public void onBorderSizeSet(net.minecraft.world.level.border.WorldBorder $$0, double $$1) {
+        this.worldBorder.setSize($$1);
     }
 
-    public void onBorderSizeLerping(net.minecraft.world.level.border.WorldBorder border, double oldSize, double newSize, long time) {
-        this.worldBorder.lerpSizeBetween(oldSize, newSize, time);
+    public void onBorderSizeLerping(net.minecraft.world.level.border.WorldBorder $$0, double $$1, double $$2, long $$3) {
+        this.worldBorder.lerpSizeBetween($$1, $$2, $$3);
     }
 
-    public void onBorderCenterSet(net.minecraft.world.level.border.WorldBorder border, double x, double z) {
-        this.worldBorder.setCenter(x, z);
+    public void onBorderCenterSet(net.minecraft.world.level.border.WorldBorder $$0, double $$1, double $$2) {
+        this.worldBorder.setCenter($$1, $$2);
     }
 
-    public void onBorderSetWarningTime(net.minecraft.world.level.border.WorldBorder border, int newTime) {
-        this.worldBorder.setWarningTime(newTime);
+    public void onBorderSetWarningTime(net.minecraft.world.level.border.WorldBorder $$0, int $$1) {
+        this.worldBorder.setWarningTime($$1);
     }
 
-    public void onBorderSetWarningBlocks(net.minecraft.world.level.border.WorldBorder border, int newDistance) {
-        this.worldBorder.setWarningBlocks(newDistance);
+    public void onBorderSetWarningBlocks(net.minecraft.world.level.border.WorldBorder $$0, int $$1) {
+        this.worldBorder.setWarningBlocks($$1);
     }
 
-    public void onBorderSetDamagePerBlock(net.minecraft.world.level.border.WorldBorder border, double newAmount) {
-        this.worldBorder.setDamagePerBlock(newAmount);
+    public void onBorderSetDamagePerBlock(net.minecraft.world.level.border.WorldBorder $$0, double $$1) {
+        this.worldBorder.setDamagePerBlock($$1);
     }
 
-    public void onBorderSetDamageSafeZOne(net.minecraft.world.level.border.WorldBorder border, double newSize) {
-        this.worldBorder.setDamageSafeZone(newSize);
+    public void onBorderSetDamageSafeZOne(net.minecraft.world.level.border.WorldBorder $$0, double $$1) {
+        this.worldBorder.setDamageSafeZone($$1);
     }
 }
 