--- net/minecraft/world/level/border/WorldBorder$Settings.java
+++ net/minecraft/world/level/border/WorldBorder$Settings.java
@@ -15,28 +15,28 @@
     public long sizeLerpTime;
     public double sizeLerpTarget;
 
-    public net.minecraft.world.level.border.WorldBorder$Settings(double centerX, double centerZ, double damagePerBlock, double safeZone, int warningBlocks, int warningTime, double size, long sizeLerpTime, double sizeLerpTarget) {
-        this.centerX = centerX;
-        this.centerZ = centerZ;
-        this.damagePerBlock = damagePerBlock;
-        this.safeZone = safeZone;
-        this.warningBlocks = warningBlocks;
-        this.warningTime = warningTime;
-        this.size = size;
-        this.sizeLerpTime = sizeLerpTime;
-        this.sizeLerpTarget = sizeLerpTarget;
+    public net.minecraft.world.level.border.WorldBorder$Settings(double $$0, double $$1, double $$2, double $$3, int $$4, int $$5, double $$6, long $$7, double $$8) {
+        this.centerX = $$0;
+        this.centerZ = $$1;
+        this.damagePerBlock = $$2;
+        this.safeZone = $$3;
+        this.warningBlocks = $$4;
+        this.warningTime = $$5;
+        this.size = $$6;
+        this.sizeLerpTime = $$7;
+        this.sizeLerpTarget = $$8;
     }
 
-    public net.minecraft.world.level.border.WorldBorder$Settings(net.minecraft.world.level.border.WorldBorder border) {
-        this.centerX = border.getCenterX();
-        this.centerZ = border.getCenterZ();
-        this.damagePerBlock = border.getDamagePerBlock();
-        this.safeZone = border.getDamageSafeZone();
-        this.warningBlocks = border.getWarningBlocks();
-        this.warningTime = border.getWarningTime();
-        this.size = border.getSize();
-        this.sizeLerpTime = border.getLerpRemainingTime();
-        this.sizeLerpTarget = border.getLerpTarget();
+    public net.minecraft.world.level.border.WorldBorder$Settings(net.minecraft.world.level.border.WorldBorder $$0) {
+        this.centerX = $$0.getCenterX();
+        this.centerZ = $$0.getCenterZ();
+        this.damagePerBlock = $$0.getDamagePerBlock();
+        this.safeZone = $$0.getDamageSafeZone();
+        this.warningBlocks = $$0.getWarningBlocks();
+        this.warningTime = $$0.getWarningTime();
+        this.size = $$0.getSize();
+        this.sizeLerpTime = $$0.getLerpRemainingTime();
+        this.sizeLerpTarget = $$0.getLerpTarget();
     }
 
     public double getCenterX() {
@@ -75,29 +75,29 @@
         return this.sizeLerpTarget;
     }
 
-    public static net.minecraft.world.level.border.WorldBorder$Settings read(com.mojang.serialization.DynamicLike<?> dynamic, net.minecraft.world.level.border.WorldBorder$Settings defaultValue) {
-        double d = net.minecraft.util.Mth.clamp((double)dynamic.get("BorderCenterX").asDouble(defaultValue.centerX), (double)-2.9999984E7, (double)2.9999984E7);
-        double d2 = net.minecraft.util.Mth.clamp((double)dynamic.get("BorderCenterZ").asDouble(defaultValue.centerZ), (double)-2.9999984E7, (double)2.9999984E7);
-        double d3 = dynamic.get("BorderSize").asDouble(defaultValue.size);
-        long l = dynamic.get("BorderSizeLerpTime").asLong(defaultValue.sizeLerpTime);
-        double d4 = dynamic.get("BorderSizeLerpTarget").asDouble(defaultValue.sizeLerpTarget);
-        double d5 = dynamic.get("BorderSafeZone").asDouble(defaultValue.safeZone);
-        double d6 = dynamic.get("BorderDamagePerBlock").asDouble(defaultValue.damagePerBlock);
-        int n = dynamic.get("BorderWarningBlocks").asInt(defaultValue.warningBlocks);
-        int n2 = dynamic.get("BorderWarningTime").asInt(defaultValue.warningTime);
-        return new net.minecraft.world.level.border.WorldBorder$Settings(d, d2, d6, d5, n, n2, d3, l, d4);
+    public static net.minecraft.world.level.border.WorldBorder$Settings read(com.mojang.serialization.DynamicLike<?> $$0, net.minecraft.world.level.border.WorldBorder$Settings $$1) {
+        double $$2 = net.minecraft.util.Mth.clamp((double)$$0.get("BorderCenterX").asDouble($$1.centerX), (double)-2.9999984E7, (double)2.9999984E7);
+        double $$3 = net.minecraft.util.Mth.clamp((double)$$0.get("BorderCenterZ").asDouble($$1.centerZ), (double)-2.9999984E7, (double)2.9999984E7);
+        double $$4 = $$0.get("BorderSize").asDouble($$1.size);
+        long $$5 = $$0.get("BorderSizeLerpTime").asLong($$1.sizeLerpTime);
+        double $$6 = $$0.get("BorderSizeLerpTarget").asDouble($$1.sizeLerpTarget);
+        double $$7 = $$0.get("BorderSafeZone").asDouble($$1.safeZone);
+        double $$8 = $$0.get("BorderDamagePerBlock").asDouble($$1.damagePerBlock);
+        int $$9 = $$0.get("BorderWarningBlocks").asInt($$1.warningBlocks);
+        int $$10 = $$0.get("BorderWarningTime").asInt($$1.warningTime);
+        return new net.minecraft.world.level.border.WorldBorder$Settings($$2, $$3, $$8, $$7, $$9, $$10, $$4, $$5, $$6);
     }
 
-    public void write(net.minecraft.nbt.CompoundTag nbt) {
-        nbt.putDouble("BorderCenterX", this.centerX);
-        nbt.putDouble("BorderCenterZ", this.centerZ);
-        nbt.putDouble("BorderSize", this.size);
-        nbt.putLong("BorderSizeLerpTime", this.sizeLerpTime);
-        nbt.putDouble("BorderSafeZone", this.safeZone);
-        nbt.putDouble("BorderDamagePerBlock", this.damagePerBlock);
-        nbt.putDouble("BorderSizeLerpTarget", this.sizeLerpTarget);
-        nbt.putDouble("BorderWarningBlocks", (double)this.warningBlocks);
-        nbt.putDouble("BorderWarningTime", (double)this.warningTime);
+    public void write(net.minecraft.nbt.CompoundTag $$0) {
+        $$0.putDouble("BorderCenterX", this.centerX);
+        $$0.putDouble("BorderCenterZ", this.centerZ);
+        $$0.putDouble("BorderSize", this.size);
+        $$0.putLong("BorderSizeLerpTime", this.sizeLerpTime);
+        $$0.putDouble("BorderSafeZone", this.safeZone);
+        $$0.putDouble("BorderDamagePerBlock", this.damagePerBlock);
+        $$0.putDouble("BorderSizeLerpTarget", this.sizeLerpTarget);
+        $$0.putDouble("BorderWarningBlocks", (double)this.warningBlocks);
+        $$0.putDouble("BorderWarningTime", (double)this.warningTime);
     }
 }
 