--- net/minecraft/world/level/border/WorldBorder$StaticBorderExtent.java
+++ net/minecraft/world/level/border/WorldBorder$StaticBorderExtent.java
@@ -14,9 +14,9 @@
     public net.minecraft.world.phys.shapes.VoxelShape shape;
     public /* synthetic */ net.minecraft.world.level.border.WorldBorder this$0;
 
-    public net.minecraft.world.level.border.WorldBorder$StaticBorderExtent(net.minecraft.world.level.border.WorldBorder worldBorder, double size) {
+    public net.minecraft.world.level.border.WorldBorder$StaticBorderExtent(net.minecraft.world.level.border.WorldBorder worldBorder, double $$0) {
         this.this$0 = worldBorder;
-        this.size = size;
+        this.size = $$0;
         this.updateBox();
     }
 