--- net/minecraft/client/gui/components/events/AbstractContainerEventHandler.java
+++ net/minecraft/client/gui/components/events/AbstractContainerEventHandler.java
@@ -18,8 +18,8 @@
         return this.isDragging;
     }
 
-    public void setDragging(boolean dragging) {
-        this.isDragging = dragging;
+    public void setDragging(boolean $$0) {
+        this.isDragging = $$0;
     }
 
     @javax.annotation.Nullable
@@ -27,8 +27,8 @@
         return this.focused;
     }
 
-    public void setFocused(@javax.annotation.Nullable net.minecraft.client.gui.components.events.GuiEventListener listener) {
-        this.focused = listener;
+    public void setFocused(@javax.annotation.Nullable net.minecraft.client.gui.components.events.GuiEventListener $$0) {
+        this.focused = $$0;
     }
 }
 