--- net/minecraft/world/level/levelgen/feature/configurations/BlockPileConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/BlockPileConfiguration.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.configurations.BlockPileConfiguration> CODEC;
     public net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider stateProvider;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.BlockPileConfiguration(net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider stateProvider) {
-        this.stateProvider = stateProvider;
+    public net.minecraft.world.level.levelgen.feature.configurations.BlockPileConfiguration(net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider $$0) {
+        this.stateProvider = $$0;
     }
 
     public static /* synthetic */ net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider lambda$static$0(net.minecraft.world.level.levelgen.feature.configurations.BlockPileConfiguration $$0) {