--- net/minecraft/world/level/levelgen/feature/configurations/BlockStateConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/BlockStateConfiguration.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.configurations.BlockStateConfiguration> CODEC;
     public net.minecraft.world.level.block.state.BlockState state;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.BlockStateConfiguration(net.minecraft.world.level.block.state.BlockState state) {
-        this.state = state;
+    public net.minecraft.world.level.levelgen.feature.configurations.BlockStateConfiguration(net.minecraft.world.level.block.state.BlockState $$0) {
+        this.state = $$0;
     }
 
     public static /* synthetic */ net.minecraft.world.level.block.state.BlockState lambda$static$0(net.minecraft.world.level.levelgen.feature.configurations.BlockStateConfiguration $$0) {