--- net/minecraft/world/level/levelgen/feature/configurations/CountConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/CountConfiguration.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.configurations.CountConfiguration> CODEC;
     public net.minecraft.util.valueproviders.IntProvider count;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.CountConfiguration(int count) {
-        this.count = net.minecraft.util.valueproviders.ConstantInt.of((int)count);
+    public net.minecraft.world.level.levelgen.feature.configurations.CountConfiguration(int $$0) {
+        this.count = net.minecraft.util.valueproviders.ConstantInt.of((int)$$0);
     }
 
     public net.minecraft.world.level.levelgen.feature.configurations.CountConfiguration(net.minecraft.util.valueproviders.IntProvider $$0) {