--- net/minecraft/world/level/levelgen/feature/configurations/EndGatewayConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/EndGatewayConfiguration.java
@@ -10,13 +10,13 @@
     public java.util.Optional<net.minecraft.core.BlockPos> exit;
     public boolean exact;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration(java.util.Optional<net.minecraft.core.BlockPos> exit, boolean exact) {
-        this.exit = exit;
-        this.exact = exact;
+    public net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration(java.util.Optional<net.minecraft.core.BlockPos> $$0, boolean $$1) {
+        this.exit = $$0;
+        this.exact = $$1;
     }
 
-    public static net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration knownExit(net.minecraft.core.BlockPos exit, boolean exact) {
-        return new net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration(java.util.Optional.of((java.lang.Object)exit), exact);
+    public static net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration knownExit(net.minecraft.core.BlockPos $$0, boolean $$1) {
+        return new net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration(java.util.Optional.of((java.lang.Object)$$0), $$1);
     }
 
     public static net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration delayedExitSearch() {