--- net/minecraft/world/level/levelgen/feature/configurations/HugeMushroomFeatureConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/HugeMushroomFeatureConfiguration.java
@@ -11,10 +11,10 @@
     public net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider stemProvider;
     public int foliageRadius;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.HugeMushroomFeatureConfiguration(net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider capProvider, net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider stemProvider, int foliageRadius) {
-        this.capProvider = capProvider;
-        this.stemProvider = stemProvider;
-        this.foliageRadius = foliageRadius;
+    public net.minecraft.world.level.levelgen.feature.configurations.HugeMushroomFeatureConfiguration(net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider $$0, net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider $$1, int $$2) {
+        this.capProvider = $$0;
+        this.stemProvider = $$1;
+        this.foliageRadius = $$2;
     }
 
     public static /* synthetic */ com.mojang.datafixers.kinds.App lambda$static$3(com.mojang.serialization.codecs.RecordCodecBuilder$Instance $$02) {