--- net/minecraft/world/level/levelgen/feature/configurations/LayerConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/LayerConfiguration.java
@@ -10,9 +10,9 @@
     public int height;
     public net.minecraft.world.level.block.state.BlockState state;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.LayerConfiguration(int height, net.minecraft.world.level.block.state.BlockState state) {
-        this.height = height;
-        this.state = state;
+    public net.minecraft.world.level.levelgen.feature.configurations.LayerConfiguration(int $$0, net.minecraft.world.level.block.state.BlockState $$1) {
+        this.height = $$0;
+        this.state = $$1;
     }
 
     public static /* synthetic */ com.mojang.datafixers.kinds.App lambda$static$2(com.mojang.serialization.codecs.RecordCodecBuilder$Instance $$02) {