--- net/minecraft/world/level/levelgen/feature/configurations/OreConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/OreConfiguration.java
@@ -25,8 +25,8 @@
         this((java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)new net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration$TargetBlockState($$0, $$1)), $$2, $$3);
     }
 
-    public net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest target, net.minecraft.world.level.block.state.BlockState state, int size) {
-        this((java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)new net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration$TargetBlockState(target, state)), size, 0.0f);
+    public net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$0, net.minecraft.world.level.block.state.BlockState $$1, int $$2) {
+        this((java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)new net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration$TargetBlockState($$0, $$1)), $$2, 0.0f);
     }
 
     public static net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration$TargetBlockState target(net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest $$0, net.minecraft.world.level.block.state.BlockState $$1) {