--- net/minecraft/world/level/levelgen/feature/configurations/ProbabilityFeatureConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/ProbabilityFeatureConfiguration.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.configurations.ProbabilityFeatureConfiguration> CODEC;
     public float probability;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.ProbabilityFeatureConfiguration(float probability) {
-        this.probability = probability;
+    public net.minecraft.world.level.levelgen.feature.configurations.ProbabilityFeatureConfiguration(float $$0) {
+        this.probability = $$0;
     }
 
     public static /* synthetic */ com.mojang.datafixers.kinds.App lambda$static$1(com.mojang.serialization.codecs.RecordCodecBuilder$Instance $$02) {