--- net/minecraft/world/level/levelgen/feature/configurations/ReplaceBlockConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/ReplaceBlockConfiguration.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.configurations.ReplaceBlockConfiguration> CODEC;
     public java.util.List<net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration$TargetBlockState> targetStates;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.ReplaceBlockConfiguration(net.minecraft.world.level.block.state.BlockState targetState, net.minecraft.world.level.block.state.BlockState state) {
-        this((java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration.target((net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest)new net.minecraft.world.level.levelgen.structure.templatesystem.BlockStateMatchTest(targetState), (net.minecraft.world.level.block.state.BlockState)state)));
+    public net.minecraft.world.level.levelgen.feature.configurations.ReplaceBlockConfiguration(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.state.BlockState $$1) {
+        this((java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration.target((net.minecraft.world.level.levelgen.structure.templatesystem.RuleTest)new net.minecraft.world.level.levelgen.structure.templatesystem.BlockStateMatchTest($$0), (net.minecraft.world.level.block.state.BlockState)$$1)));
     }
 
     public net.minecraft.world.level.levelgen.feature.configurations.ReplaceBlockConfiguration(java.util.List<net.minecraft.world.level.levelgen.feature.configurations.OreConfiguration$TargetBlockState> $$0) {