--- net/minecraft/world/level/levelgen/feature/configurations/SpikeConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/configurations/SpikeConfiguration.java
@@ -12,14 +12,14 @@
     @javax.annotation.Nullable
     public net.minecraft.core.BlockPos crystalBeamTarget;
 
-    public net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration(boolean crystalInvulnerable, java.util.List<net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike> spikes, @javax.annotation.Nullable net.minecraft.core.BlockPos crystalBeamTarget) {
-        this(crystalInvulnerable, spikes, java.util.Optional.ofNullable((java.lang.Object)crystalBeamTarget));
+    public net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration(boolean $$0, java.util.List<net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike> $$1, @javax.annotation.Nullable net.minecraft.core.BlockPos $$2) {
+        this($$0, $$1, java.util.Optional.ofNullable((java.lang.Object)$$2));
     }
 
-    public net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration(boolean crystalInvulnerable, java.util.List<net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike> spikes, java.util.Optional<net.minecraft.core.BlockPos> crystalBeamTarget) {
-        this.crystalInvulnerable = crystalInvulnerable;
-        this.spikes = spikes;
-        this.crystalBeamTarget = (net.minecraft.core.BlockPos)crystalBeamTarget.orElse(null);
+    public net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration(boolean $$0, java.util.List<net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike> $$1, java.util.Optional<net.minecraft.core.BlockPos> $$2) {
+        this.crystalInvulnerable = $$0;
+        this.spikes = $$1;
+        this.crystalBeamTarget = (net.minecraft.core.BlockPos)$$2.orElse(null);
     }
 
     public boolean isCrystalInvulnerable() {