--- net/minecraft/network/protocol/status/ClientboundStatusResponsePacket.java
+++ net/minecraft/network/protocol/status/ClientboundStatusResponsePacket.java
@@ -9,20 +9,20 @@
     public static com.google.gson.Gson GSON;
     public net.minecraft.network.protocol.status.ServerStatus status;
 
-    public net.minecraft.network.protocol.status.ClientboundStatusResponsePacket(net.minecraft.network.protocol.status.ServerStatus status) {
-        this.status = status;
+    public net.minecraft.network.protocol.status.ClientboundStatusResponsePacket(net.minecraft.network.protocol.status.ServerStatus $$0) {
+        this.status = $$0;
     }
 
     public net.minecraft.network.protocol.status.ClientboundStatusResponsePacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.status = (net.minecraft.network.protocol.status.ServerStatus)net.minecraft.util.GsonHelper.fromJson((com.google.gson.Gson)GSON, (java.lang.String)$$0.readUtf(java.lang.Short.MAX_VALUE), net.minecraft.network.protocol.status.ServerStatus.class);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUtf(GSON.toJson((java.lang.Object)this.status));
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUtf(GSON.toJson((java.lang.Object)this.status));
     }
 
-    public void handle(net.minecraft.network.protocol.status.ClientStatusPacketListener handler) {
-        handler.handleStatusResponse(this);
+    public void handle(net.minecraft.network.protocol.status.ClientStatusPacketListener $$0) {
+        $$0.handleStatusResponse(this);
     }
 
     public net.minecraft.network.protocol.status.ServerStatus getStatus() {