--- net/minecraft/network/protocol/status/ServerStatus$Serializer.java
+++ net/minecraft/network/protocol/status/ServerStatus$Serializer.java
@@ -25,9 +25,6 @@
         if ($$3.has("favicon")) {
             $$4.setFavicon(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$3, (java.lang.String)"favicon"));
         }
-        if ($$3.has("previewsChat")) {
-            $$4.setPreviewsChat(net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)$$3, (java.lang.String)"previewsChat"));
-        }
         if ($$3.has("enforcesSecureChat")) {
             $$4.setEnforcesSecureChat(net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)$$3, (java.lang.String)"enforcesSecureChat"));
         }
@@ -36,7 +33,6 @@
 
     public com.google.gson.JsonElement serialize(net.minecraft.network.protocol.status.ServerStatus $$0, java.lang.reflect.Type $$1, com.google.gson.JsonSerializationContext $$2) {
         com.google.gson.JsonObject $$3 = new com.google.gson.JsonObject();
-        $$3.addProperty("previewsChat", java.lang.Boolean.valueOf((boolean)$$0.previewsChat()));
         $$3.addProperty("enforcesSecureChat", java.lang.Boolean.valueOf((boolean)$$0.enforcesSecureChat()));
         if ($$0.getDescription() != null) {
             $$3.add("description", $$2.serialize((java.lang.Object)$$0.getDescription()));