--- net/minecraft/network/protocol/status/ServerboundStatusRequestPacket.java
+++ net/minecraft/network/protocol/status/ServerboundStatusRequestPacket.java
@@ -12,11 +12,11 @@
     public net.minecraft.network.protocol.status.ServerboundStatusRequestPacket(net.minecraft.network.FriendlyByteBuf $$0) {
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
     }
 
-    public void handle(net.minecraft.network.protocol.status.ServerStatusPacketListener handler) {
-        handler.handleStatusRequest(this);
+    public void handle(net.minecraft.network.protocol.status.ServerStatusPacketListener $$0) {
+        $$0.handleStatusRequest(this);
     }
 }
 