--- net/minecraft/network/protocol/status/ServerStatus$Players.java
+++ net/minecraft/network/protocol/status/ServerStatus$Players.java
@@ -10,9 +10,9 @@
     @javax.annotation.Nullable
     public com.mojang.authlib.GameProfile[] sample;
 
-    public net.minecraft.network.protocol.status.ServerStatus$Players(int maxPlayers, int numPlayers) {
-        this.maxPlayers = maxPlayers;
-        this.numPlayers = numPlayers;
+    public net.minecraft.network.protocol.status.ServerStatus$Players(int $$0, int $$1) {
+        this.maxPlayers = $$0;
+        this.numPlayers = $$1;
     }
 
     public int getMaxPlayers() {
@@ -28,8 +28,8 @@
         return this.sample;
     }
 
-    public void setSample(com.mojang.authlib.GameProfile[] players) {
-        this.sample = players;
+    public void setSample(com.mojang.authlib.GameProfile[] $$0) {
+        this.sample = $$0;
     }
 }
 