--- net/minecraft/network/protocol/status/ServerStatus$Version.java
+++ net/minecraft/network/protocol/status/ServerStatus$Version.java
@@ -8,9 +8,9 @@
     public java.lang.String name;
     public int protocol;
 
-    public net.minecraft.network.protocol.status.ServerStatus$Version(java.lang.String name, int protocol) {
-        this.name = name;
-        this.protocol = protocol;
+    public net.minecraft.network.protocol.status.ServerStatus$Version(java.lang.String $$0, int $$1) {
+        this.name = $$0;
+        this.protocol = $$1;
     }
 
     public java.lang.String getName() {