--- net/minecraft/network/protocol/status/ServerStatus.java
+++ net/minecraft/network/protocol/status/ServerStatus.java
@@ -15,7 +15,6 @@
     public net.minecraft.network.protocol.status.ServerStatus$Version version;
     @javax.annotation.Nullable
     public java.lang.String favicon;
-    public boolean previewsChat;
     public boolean enforcesSecureChat;
 
     public net.minecraft.network.protocol.status.ServerStatus() {
@@ -26,8 +25,8 @@
         return this.description;
     }
 
-    public void setDescription(net.minecraft.network.chat.Component description) {
-        this.description = description;
+    public void setDescription(net.minecraft.network.chat.Component $$0) {
+        this.description = $$0;
     }
 
     @javax.annotation.Nullable
@@ -35,8 +34,8 @@
         return this.players;
     }
 
-    public void setPlayers(net.minecraft.network.protocol.status.ServerStatus$Players players) {
-        this.players = players;
+    public void setPlayers(net.minecraft.network.protocol.status.ServerStatus$Players $$0) {
+        this.players = $$0;
     }
 
     @javax.annotation.Nullable
@@ -44,12 +43,12 @@
         return this.version;
     }
 
-    public void setVersion(net.minecraft.network.protocol.status.ServerStatus$Version version) {
-        this.version = version;
+    public void setVersion(net.minecraft.network.protocol.status.ServerStatus$Version $$0) {
+        this.version = $$0;
     }
 
-    public void setFavicon(java.lang.String faviconBlob) {
-        this.favicon = faviconBlob;
+    public void setFavicon(java.lang.String $$0) {
+        this.favicon = $$0;
     }
 
     @javax.annotation.Nullable
@@ -57,14 +56,6 @@
         return this.favicon;
     }
 
-    public void setPreviewsChat(boolean $$0) {
-        this.previewsChat = $$0;
-    }
-
-    public boolean previewsChat() {
-        return this.previewsChat;
-    }
-
     public void setEnforcesSecureChat(boolean $$0) {
         this.enforcesSecureChat = $$0;
     }