--- net/minecraft/realms/DisconnectedRealmsScreen.java
+++ net/minecraft/realms/DisconnectedRealmsScreen.java
@@ -11,17 +11,17 @@
     public net.minecraft.client.gui.screens.Screen parent;
     public int textHeight;
 
-    public net.minecraft.realms.DisconnectedRealmsScreen(net.minecraft.client.gui.screens.Screen parent, net.minecraft.network.chat.Component title, net.minecraft.network.chat.Component reason) {
-        super(title);
+    public net.minecraft.realms.DisconnectedRealmsScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.network.chat.Component $$1, net.minecraft.network.chat.Component $$2) {
+        super($$1);
         this.message = net.minecraft.client.gui.components.MultiLineLabel.EMPTY;
-        this.parent = parent;
-        this.reason = reason;
+        this.parent = $$0;
+        this.reason = $$2;
     }
 
     public void init() {
         net.minecraft.client.Minecraft $$0 = net.minecraft.client.Minecraft.getInstance();
         $$0.setConnectedToRealms(false);
-        $$0.getClientPackSource().clearServerPack();
+        $$0.getDownloadedPackSource().clearServerPack();
         this.message = net.minecraft.client.gui.components.MultiLineLabel.create((net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.FormattedText)this.reason, (int)(this.width - 50));
         int n = this.message.getLineCount();
         java.util.Objects.requireNonNull((java.lang.Object)this.font);
@@ -40,14 +40,14 @@
         net.minecraft.client.Minecraft.getInstance().setScreen(this.parent);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
         int n = this.width / 2;
         int n2 = this.height / 2 - this.textHeight / 2;
         java.util.Objects.requireNonNull((java.lang.Object)this.font);
-        net.minecraft.realms.DisconnectedRealmsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)n, (int)(n2 - 9 * 2), (int)0xAAAAAA);
-        this.message.renderCentered(matrixStack, this.width / 2, this.height / 2 - this.textHeight / 2);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        net.minecraft.realms.DisconnectedRealmsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)n, (int)(n2 - 9 * 2), (int)0xAAAAAA);
+        this.message.renderCentered($$0, this.width / 2, this.height / 2 - this.textHeight / 2);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 