--- net/minecraft/realms/RealmsLabel.java
+++ net/minecraft/realms/RealmsLabel.java
@@ -11,11 +11,11 @@
     public int y;
     public int color;
 
-    public net.minecraft.realms.RealmsLabel(net.minecraft.network.chat.Component text, int x, int y, int color) {
-        this.text = text;
-        this.x = x;
-        this.y = y;
-        this.color = color;
+    public net.minecraft.realms.RealmsLabel(net.minecraft.network.chat.Component $$0, int $$1, int $$2, int $$3) {
+        this.text = $$0;
+        this.x = $$1;
+        this.y = $$2;
+        this.color = $$3;
     }
 
     public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {