--- net/minecraft/realms/RealmsConnect.java
+++ net/minecraft/realms/RealmsConnect.java
@@ -11,8 +11,8 @@
     @javax.annotation.Nullable
     public net.minecraft.network.Connection connection;
 
-    public net.minecraft.realms.RealmsConnect(net.minecraft.client.gui.screens.Screen onlineScreen) {
-        this.onlineScreen = onlineScreen;
+    public net.minecraft.realms.RealmsConnect(net.minecraft.client.gui.screens.Screen $$0) {
+        this.onlineScreen = $$0;
     }
 
     public void connect(com.mojang.realmsclient.dto.RealmsServer $$0, net.minecraft.client.multiplayer.resolver.ServerAddress $$1) {
@@ -20,7 +20,7 @@
         $$2.setConnectedToRealms(true);
         $$2.prepareForMultiplayer();
         $$2.getNarrator().sayNow((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"mco.connect.success"));
-        java.util.concurrent.CompletableFuture $$3 = $$2.getProfileKeyPairManager().preparePublicKey();
+        java.util.concurrent.CompletableFuture $$3 = $$2.getProfileKeyPairManager().prepareChatSession();
         java.lang.String $$4 = $$1.getHost();
         int $$5 = $$1.getPort();
         new /* Unavailable Anonymous Inner Class!! */.start();