--- net/minecraft/realms/RealmsConnect$1.java
+++ net/minecraft/realms/RealmsConnect$1.java
@@ -9,7 +9,7 @@
     public /* synthetic */ java.lang.String val$hostname;
     public /* synthetic */ int val$port;
     public /* synthetic */ net.minecraft.client.Minecraft val$minecraft;
-    public /* synthetic */ java.util.concurrent.CompletableFuture val$publicKey;
+    public /* synthetic */ java.util.concurrent.CompletableFuture val$chatSessionFuture;
     public /* synthetic */ com.mojang.realmsclient.dto.RealmsServer val$server;
     public /* synthetic */ net.minecraft.realms.RealmsConnect this$0;
 
@@ -18,7 +18,7 @@
         this.val$hostname = string;
         this.val$port = n;
         this.val$minecraft = minecraft;
-        this.val$publicKey = completableFuture;
+        this.val$chatSessionFuture = completableFuture;
         this.val$server = realmsServer;
         super($$1);
     }
@@ -34,7 +34,8 @@
             if (this.this$0.aborted) {
                 return;
             }
-            this.this$0.connection.setListener((net.minecraft.network.PacketListener)new net.minecraft.client.multiplayer.ClientHandshakePacketListenerImpl(this.this$0.connection, this.val$minecraft, this.this$0.onlineScreen, $$0 -> {}
+            net.minecraft.network.chat.LocalChatSession $$1 = (net.minecraft.network.chat.LocalChatSession)this.val$chatSessionFuture.join();
+            this.this$0.connection.setListener((net.minecraft.network.PacketListener)new net.minecraft.client.multiplayer.ClientHandshakePacketListenerImpl(this.this$0.connection, this.val$minecraft, $$1, this.val$server.toServerData(this.val$hostname), this.this$0.onlineScreen, $$0 -> {}
 ));
             if (this.this$0.aborted) {
                 return;
@@ -43,24 +44,24 @@
             if (this.this$0.aborted) {
                 return;
             }
-            java.lang.String $$1 = this.val$minecraft.getUser().getName();
-            java.util.UUID $$2 = this.val$minecraft.getUser().getProfileId();
-            this.this$0.connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.login.ServerboundHelloPacket($$1, (java.util.Optional)this.val$publicKey.join(), java.util.Optional.ofNullable((java.lang.Object)$$2)));
-            this.val$minecraft.setCurrentServer(this.val$server, this.val$hostname);
+            java.lang.String $$2 = this.val$minecraft.getUser().getName();
+            java.util.UUID $$3 = this.val$minecraft.getUser().getProfileId();
+            this.this$0.connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.login.ServerboundHelloPacket($$2, $$1.asRemote().asData(), java.util.Optional.ofNullable((java.lang.Object)$$3)));
+            this.val$minecraft.updateReportEnvironment(net.minecraft.client.multiplayer.chat.report.ReportEnvironment.realm((com.mojang.realmsclient.dto.RealmsServer)this.val$server));
         }
-        catch (java.lang.Exception $$3) {
-            this.val$minecraft.getClientPackSource().clearServerPack();
+        catch (java.lang.Exception $$4) {
+            this.val$minecraft.getDownloadedPackSource().clearServerPack();
             if (this.this$0.aborted) {
                 return;
             }
-            net.minecraft.realms.RealmsConnect.LOGGER.error("Couldn't connect to world", (java.lang.Throwable)$$3);
-            java.lang.String $$4 = $$3.toString();
+            net.minecraft.realms.RealmsConnect.LOGGER.error("Couldn't connect to world", (java.lang.Throwable)$$4);
+            java.lang.String $$5 = $$4.toString();
             if ($$02 != null) {
-                java.lang.String $$5 = $$02 + ":" + this.val$port;
-                $$4 = $$4.replaceAll($$5, "");
+                java.lang.String $$6 = $$02 + ":" + this.val$port;
+                $$5 = $$5.replaceAll($$6, "");
             }
-            net.minecraft.realms.DisconnectedRealmsScreen $$6 = new net.minecraft.realms.DisconnectedRealmsScreen(this.this$0.onlineScreen, net.minecraft.network.chat.CommonComponents.CONNECT_FAILED, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"disconnect.genericReason", (java.lang.Object[])new java.lang.Object[]{$$4}));
-            this.val$minecraft.execute(() -> this.val$minecraft.setScreen((net.minecraft.client.gui.screens.Screen)$$6));
+            net.minecraft.realms.DisconnectedRealmsScreen $$7 = new net.minecraft.realms.DisconnectedRealmsScreen(this.this$0.onlineScreen, net.minecraft.network.chat.CommonComponents.CONNECT_FAILED, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"disconnect.genericReason", (java.lang.Object[])new java.lang.Object[]{$$5}));
+            this.val$minecraft.execute(() -> this.val$minecraft.setScreen((net.minecraft.client.gui.screens.Screen)$$7));
         }
     }
 }