--- net/minecraft/realms/RealmsScreen.java
+++ net/minecraft/realms/RealmsScreen.java
@@ -34,8 +34,8 @@
         this.labels = com.google.common.collect.Lists.newArrayList();
     }
 
-    public static int row(int position) {
-        return 40 + position * 13;
+    public static int row(int $$0) {
+        return 40 + $$0 * 13;
     }
 
     public net.minecraft.realms.RealmsLabel addLabel(net.minecraft.realms.RealmsLabel $$0) {