--- net/minecraft/realms/RepeatedNarrator.java
+++ net/minecraft/realms/RepeatedNarrator.java
@@ -8,9 +8,9 @@
     public float permitsPerSecond;
     public java.util.concurrent.atomic.AtomicReference<net.minecraft.realms.RepeatedNarrator$Params> params;
 
-    public net.minecraft.realms.RepeatedNarrator(java.time.Duration duration) {
+    public net.minecraft.realms.RepeatedNarrator(java.time.Duration $$0) {
         this.params = new java.util.concurrent.atomic.AtomicReference();
-        this.permitsPerSecond = 1000.0f / (float)duration.toMillis();
+        this.permitsPerSecond = 1000.0f / (float)$$0.toMillis();
     }
 
     public void narrate(net.minecraft.client.GameNarrator $$0, net.minecraft.network.chat.Component $$12) {