--- net/minecraft/server/network/MemoryServerHandshakePacketListenerImpl.java
+++ net/minecraft/server/network/MemoryServerHandshakePacketListenerImpl.java
@@ -14,12 +14,12 @@
         this.connection = $$1;
     }
 
-    public void handleIntention(net.minecraft.network.protocol.handshake.ClientIntentionPacket packet) {
-        this.connection.setProtocol(packet.getIntention());
+    public void handleIntention(net.minecraft.network.protocol.handshake.ClientIntentionPacket $$0) {
+        this.connection.setProtocol($$0.getIntention());
         this.connection.setListener((net.minecraft.network.PacketListener)new net.minecraft.server.network.ServerLoginPacketListenerImpl(this.server, this.connection));
     }
 
-    public void onDisconnect(net.minecraft.network.chat.Component reason) {
+    public void onDisconnect(net.minecraft.network.chat.Component $$0) {
     }
 
     public net.minecraft.network.Connection getConnection() {