--- net/minecraft/server/network/ServerStatusPacketListenerImpl.java
+++ net/minecraft/server/network/ServerStatusPacketListenerImpl.java
@@ -16,14 +16,14 @@
         this.connection = $$1;
     }
 
-    public void onDisconnect(net.minecraft.network.chat.Component reason) {
+    public void onDisconnect(net.minecraft.network.chat.Component $$0) {
     }
 
     public net.minecraft.network.Connection getConnection() {
         return this.connection;
     }
 
-    public void handleStatusRequest(net.minecraft.network.protocol.status.ServerboundStatusRequestPacket packet) {
+    public void handleStatusRequest(net.minecraft.network.protocol.status.ServerboundStatusRequestPacket $$0) {
         if (this.hasRequestedStatus) {
             this.connection.disconnect(DISCONNECT_REASON);
             return;
@@ -32,8 +32,8 @@
         this.connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.status.ClientboundStatusResponsePacket(this.server.getStatus()));
     }
 
-    public void handlePingRequest(net.minecraft.network.protocol.status.ServerboundPingRequestPacket packet) {
-        this.connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.status.ClientboundPongResponsePacket(packet.getTime()));
+    public void handlePingRequest(net.minecraft.network.protocol.status.ServerboundPingRequestPacket $$0) {
+        this.connection.send((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.status.ClientboundPongResponsePacket($$0.getTime()));
         this.connection.disconnect(DISCONNECT_REASON);
     }
 