--- net/minecraft/client/gui/font/providers/GlyphProviderBuilderType.java
+++ net/minecraft/client/gui/font/providers/GlyphProviderBuilderType.java
@@ -29,16 +29,16 @@
         this.factory = $$1;
     }
 
-    public static net.minecraft.client.gui.font.providers.GlyphProviderBuilderType byName(java.lang.String type) {
-        net.minecraft.client.gui.font.providers.GlyphProviderBuilderType glyphProviderBuilderType = (net.minecraft.client.gui.font.providers.GlyphProviderBuilderType)BY_NAME.get((java.lang.Object)type);
-        if (glyphProviderBuilderType == null) {
-            throw new java.lang.IllegalArgumentException("Invalid type: " + type);
+    public static net.minecraft.client.gui.font.providers.GlyphProviderBuilderType byName(java.lang.String $$0) {
+        net.minecraft.client.gui.font.providers.GlyphProviderBuilderType $$1 = (net.minecraft.client.gui.font.providers.GlyphProviderBuilderType)BY_NAME.get((java.lang.Object)$$0);
+        if ($$1 == null) {
+            throw new java.lang.IllegalArgumentException("Invalid type: " + $$0);
         }
-        return glyphProviderBuilderType;
+        return $$1;
     }
 
-    public net.minecraft.client.gui.font.providers.GlyphProviderBuilder create(com.google.gson.JsonObject json) {
-        return (net.minecraft.client.gui.font.providers.GlyphProviderBuilder)this.factory.apply((java.lang.Object)json);
+    public net.minecraft.client.gui.font.providers.GlyphProviderBuilder create(com.google.gson.JsonObject $$0) {
+        return (net.minecraft.client.gui.font.providers.GlyphProviderBuilder)this.factory.apply((java.lang.Object)$$0);
     }
 
     public static /* synthetic */ void lambda$static$0(java.util.HashMap $$0) {