--- net/minecraft/client/gui/font/providers/LegacyUnicodeBitmapsProvider$Sheet.java
+++ net/minecraft/client/gui/font/providers/LegacyUnicodeBitmapsProvider$Sheet.java
@@ -1,0 +1,31 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.gui.font.providers;
+
+public class net.minecraft.client.gui.font.providers.LegacyUnicodeBitmapsProvider$Sheet
+implements java.lang.AutoCloseable
+ {
+    public byte[] sizes;
+    public com.mojang.blaze3d.platform.NativeImage source;
+
+    public net.minecraft.client.gui.font.providers.LegacyUnicodeBitmapsProvider$Sheet(byte[] $$0, com.mojang.blaze3d.platform.NativeImage $$1) {
+        this.sizes = $$0;
+        this.source = $$1;
+    }
+
+    public void close() {
+        this.source.close();
+    }
+
+    @javax.annotation.Nullable
+    public com.mojang.blaze3d.font.GlyphInfo getGlyph(int $$0) {
+        byte $$1 = this.sizes[$$0];
+        if ($$1 != 0) {
+            int $$2 = net.minecraft.client.gui.font.providers.LegacyUnicodeBitmapsProvider.getLeft((byte)$$1);
+            return new net.minecraft.client.gui.font.providers.LegacyUnicodeBitmapsProvider$Glyph($$0 % 16 * 16 + $$2, ($$0 & 0xFF) / 16 * 16, net.minecraft.client.gui.font.providers.LegacyUnicodeBitmapsProvider.getRight((byte)$$1) - $$2, 16, this.source);
+        }
+        return null;
+    }
+}
+