--- net/minecraft/client/resources/model/AtlasSet$ResourceLister.java
+++ net/minecraft/client/resources/model/AtlasSet$ResourceLister.java
@@ -1,0 +1,11 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.resources.model;
+
+@java.lang.FunctionalInterface
+public interface net.minecraft.client.resources.model.AtlasSet$ResourceLister
+extends java.util.function.Function<net.minecraft.server.packs.resources.ResourceManager, java.util.Map<net.minecraft.resources.ResourceLocation, net.minecraft.server.packs.resources.Resource>>
+ {
+}
+