--- net/minecraft/client/resources/model/AtlasSet$StitchResult.java
+++ net/minecraft/client/resources/model/AtlasSet$StitchResult.java
@@ -1,0 +1,33 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.resources.model;
+
+public class net.minecraft.client.resources.model.AtlasSet$StitchResult
+ {
+    public net.minecraft.client.renderer.texture.TextureAtlas atlas;
+    public net.minecraft.client.renderer.texture.SpriteLoader$Preparations preparations;
+
+    public net.minecraft.client.resources.model.AtlasSet$StitchResult(net.minecraft.client.renderer.texture.TextureAtlas $$0, net.minecraft.client.renderer.texture.SpriteLoader$Preparations $$1) {
+        this.atlas = $$0;
+        this.preparations = $$1;
+    }
+
+    @javax.annotation.Nullable
+    public net.minecraft.client.renderer.texture.TextureAtlasSprite getSprite(net.minecraft.resources.ResourceLocation $$0) {
+        return (net.minecraft.client.renderer.texture.TextureAtlasSprite)this.preparations.regions().get((java.lang.Object)$$0);
+    }
+
+    public net.minecraft.client.renderer.texture.TextureAtlasSprite missing() {
+        return this.preparations.missing();
+    }
+
+    public java.util.concurrent.CompletableFuture<java.lang.Void> readyForUpload() {
+        return this.preparations.readyForUpload();
+    }
+
+    public void upload() {
+        this.atlas.upload(this.preparations);
+    }
+}
+