--- net/minecraft/client/resources/model/BlockModelRotation.java
+++ net/minecraft/client/resources/model/BlockModelRotation.java
@@ -38,8 +38,8 @@
         return (net.minecraft.client.resources.model.BlockModelRotation)java.lang.Enum.valueOf(net.minecraft.client.resources.model.BlockModelRotation.class, (java.lang.String)$$0);
     }
 
-    public static int getIndex(int x, int y) {
-        return x * 360 + y;
+    public static int getIndex(int $$0, int $$1) {
+        return $$0 * 360 + $$1;
     }
 
     public net.minecraft.client.resources.model.BlockModelRotation(int $$0, int $$1) {
@@ -62,8 +62,8 @@
         return this.transformation;
     }
 
-    public static net.minecraft.client.resources.model.BlockModelRotation by(int x, int y) {
-        return (net.minecraft.client.resources.model.BlockModelRotation)BY_INDEX.get((java.lang.Object)net.minecraft.client.resources.model.BlockModelRotation.getIndex((int)net.minecraft.util.Mth.positiveModulo((int)x, (int)360), (int)net.minecraft.util.Mth.positiveModulo((int)y, (int)360)));
+    public static net.minecraft.client.resources.model.BlockModelRotation by(int $$0, int $$1) {
+        return (net.minecraft.client.resources.model.BlockModelRotation)BY_INDEX.get((java.lang.Object)net.minecraft.client.resources.model.BlockModelRotation.getIndex((int)net.minecraft.util.Mth.positiveModulo((int)$$0, (int)360), (int)net.minecraft.util.Mth.positiveModulo((int)$$1, (int)360)));
     }
 
     public com.mojang.math.OctahedralGroup actualRotation() {