--- net/minecraft/client/resources/model/ModelResourceLocation.java
+++ net/minecraft/client/resources/model/ModelResourceLocation.java
@@ -10,39 +10,26 @@
     public static char VARIANT_SEPARATOR = '#';
     public java.lang.String variant;
 
-    public net.minecraft.client.resources.model.ModelResourceLocation(java.lang.String[] $$0) {
-        super($$0);
-        this.variant = $$0[2].toLowerCase(java.util.Locale.ROOT);
+    public net.minecraft.client.resources.model.ModelResourceLocation(java.lang.String $$0, java.lang.String $$1, java.lang.String $$2, @javax.annotation.Nullable net.minecraft.resources.ResourceLocation$Dummy $$3) {
+        super($$0, $$1, $$3);
+        this.variant = $$2;
     }
 
     public net.minecraft.client.resources.model.ModelResourceLocation(java.lang.String $$0, java.lang.String $$1, java.lang.String $$2) {
-        this(new java.lang.String[]{$$0, $$1, $$2});
+        super($$0, $$1);
+        this.variant = net.minecraft.client.resources.model.ModelResourceLocation.lowercaseVariant((java.lang.String)$$2);
     }
 
-    public net.minecraft.client.resources.model.ModelResourceLocation(java.lang.String $$0) {
-        this(net.minecraft.client.resources.model.ModelResourceLocation.decompose((java.lang.String)$$0));
-    }
-
     public net.minecraft.client.resources.model.ModelResourceLocation(net.minecraft.resources.ResourceLocation $$0, java.lang.String $$1) {
-        this($$0.toString(), $$1);
+        this($$0.getNamespace(), $$0.getPath(), net.minecraft.client.resources.model.ModelResourceLocation.lowercaseVariant((java.lang.String)$$1), null);
     }
 
-    public net.minecraft.client.resources.model.ModelResourceLocation(java.lang.String $$0, java.lang.String $$1) {
-        this(net.minecraft.client.resources.model.ModelResourceLocation.decompose((java.lang.String)($$0 + "#" + $$1)));
+    public static net.minecraft.client.resources.model.ModelResourceLocation vanilla(java.lang.String $$0, java.lang.String $$1) {
+        return new net.minecraft.client.resources.model.ModelResourceLocation("minecraft", $$0, $$1);
     }
 
-    public static java.lang.String[] decompose(java.lang.String path) {
-        java.lang.String[] stringArray = new java.lang.String[]{null, path, ""};
-        int n = path.indexOf(35);
-        java.lang.String string = path;
-        if (n >= 0) {
-            stringArray[2] = path.substring(n + 1, path.length());
-            if (n > 1) {
-                string = path.substring(0, n);
-            }
-        }
-        java.lang.System.arraycopy((java.lang.Object)net.minecraft.resources.ResourceLocation.decompose((java.lang.String)string, (char)':'), (int)0, (java.lang.Object)stringArray, (int)0, (int)2);
-        return stringArray;
+    public static java.lang.String lowercaseVariant(java.lang.String $$0) {
+        return $$0.toLowerCase(java.util.Locale.ROOT);
     }
 
     public java.lang.String getVariant() {