--- net/minecraft/client/resources/model/ModelBakery$ModelBakerImpl.java
+++ net/minecraft/client/resources/model/ModelBakery$ModelBakerImpl.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.resources.model;
+
+public class net.minecraft.client.resources.model.ModelBakery$ModelBakerImpl
+implements net.minecraft.client.resources.model.ModelBaker
+ {
+    public java.util.function.Function<net.minecraft.client.resources.model.Material, net.minecraft.client.renderer.texture.TextureAtlasSprite> modelTextureGetter;
+    public /* synthetic */ net.minecraft.client.resources.model.ModelBakery this$0;
+
+    public net.minecraft.client.resources.model.ModelBakery$ModelBakerImpl(net.minecraft.client.resources.model.ModelBakery modelBakery, java.util.function.BiFunction<net.minecraft.resources.ResourceLocation, net.minecraft.client.resources.model.Material, net.minecraft.client.renderer.texture.TextureAtlasSprite> $$0, net.minecraft.resources.ResourceLocation $$1) {
+        this.this$0 = modelBakery;
+        this.modelTextureGetter = $$2 -> (net.minecraft.client.renderer.texture.TextureAtlasSprite)$$0.apply((java.lang.Object)$$1, $$2);
+    }
+
+    public net.minecraft.client.resources.model.UnbakedModel getModel(net.minecraft.resources.ResourceLocation $$0) {
+        return this.this$0.getModel($$0);
+    }
+
+    public net.minecraft.client.resources.model.BakedModel bake(net.minecraft.resources.ResourceLocation $$0, net.minecraft.client.resources.model.ModelState $$1) {
+        net.minecraft.client.renderer.block.model.BlockModel $$5;
+        net.minecraft.client.resources.model.ModelBakery$BakedCacheKey $$2 = new net.minecraft.client.resources.model.ModelBakery$BakedCacheKey($$0, $$1.getRotation(), $$1.isUvLocked());
+        net.minecraft.client.resources.model.BakedModel $$3 = (net.minecraft.client.resources.model.BakedModel)this.this$0.bakedCache.get((java.lang.Object)$$2);
+        if ($$3 != null) {
+            return $$3;
+        }
+        net.minecraft.client.resources.model.UnbakedModel $$4 = this.getModel($$0);
+        if ($$4 instanceof net.minecraft.client.renderer.block.model.BlockModel && ($$5 = (net.minecraft.client.renderer.block.model.BlockModel)$$4).getRootModel() == net.minecraft.client.resources.model.ModelBakery.GENERATION_MARKER) {
+            return net.minecraft.client.resources.model.ModelBakery.ITEM_MODEL_GENERATOR.generateBlockModel(this.modelTextureGetter, $$5).bake((net.minecraft.client.resources.model.ModelBaker)this, $$5, this.modelTextureGetter, $$1, $$0, false);
+        }
+        net.minecraft.client.resources.model.BakedModel $$6 = $$4.bake((net.minecraft.client.resources.model.ModelBaker)this, this.modelTextureGetter, $$1, $$0);
+        this.this$0.bakedCache.put((java.lang.Object)$$2, (java.lang.Object)$$6);
+        return $$6;
+    }
+}
+