--- net/minecraft/client/resources/model/ModelBakery$LoadedJson.java
+++ net/minecraft/client/resources/model/ModelBakery$LoadedJson.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.resources.model;
+
+public class net.minecraft.client.resources.model.ModelBakery$LoadedJson
+extends java.lang.Record
+ {
+    public java.lang.String source;
+    public com.google.gson.JsonElement data;
+
+    public net.minecraft.client.resources.model.ModelBakery$LoadedJson(java.lang.String $$0, com.google.gson.JsonElement $$1) {
+        this.source = $$0;
+        this.data = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.resources.model.ModelBakery$LoadedJson.class, "source;data", "source", "data"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.resources.model.ModelBakery$LoadedJson.class, "source;data", "source", "data"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.resources.model.ModelBakery$LoadedJson.class, "source;data", "source", "data"}, this, $$0);
+    }
+
+    public java.lang.String source() {
+        return this.source;
+    }
+
+    public com.google.gson.JsonElement data() {
+        return this.data;
+    }
+}
+