--- net/minecraft/client/resources/model/MultiPartBakedModel$Builder.java
+++ net/minecraft/client/resources/model/MultiPartBakedModel$Builder.java
@@ -11,8 +11,8 @@
         this.selectors = com.google.common.collect.Lists.newArrayList();
     }
 
-    public void add(java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> predicate, net.minecraft.client.resources.model.BakedModel model) {
-        this.selectors.add((java.lang.Object)org.apache.commons.lang3.tuple.Pair.of(predicate, (java.lang.Object)model));
+    public void add(java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> $$0, net.minecraft.client.resources.model.BakedModel $$1) {
+        this.selectors.add((java.lang.Object)org.apache.commons.lang3.tuple.Pair.of($$0, (java.lang.Object)$$1));
     }
 
     public net.minecraft.client.resources.model.BakedModel build() {