--- net/minecraft/client/resources/model/SimpleBakedModel$Builder.java
+++ net/minecraft/client/resources/model/SimpleBakedModel$Builder.java
@@ -31,18 +31,18 @@
         this.transforms = $$3;
     }
 
-    public net.minecraft.client.resources.model.SimpleBakedModel$Builder addCulledFace(net.minecraft.core.Direction facing, net.minecraft.client.renderer.block.model.BakedQuad quad) {
-        ((java.util.List)this.culledFaces.get((java.lang.Object)facing)).add((java.lang.Object)quad);
+    public net.minecraft.client.resources.model.SimpleBakedModel$Builder addCulledFace(net.minecraft.core.Direction $$0, net.minecraft.client.renderer.block.model.BakedQuad $$1) {
+        ((java.util.List)this.culledFaces.get((java.lang.Object)$$0)).add((java.lang.Object)$$1);
         return this;
     }
 
-    public net.minecraft.client.resources.model.SimpleBakedModel$Builder addUnculledFace(net.minecraft.client.renderer.block.model.BakedQuad quad) {
-        this.unculledFaces.add((java.lang.Object)quad);
+    public net.minecraft.client.resources.model.SimpleBakedModel$Builder addUnculledFace(net.minecraft.client.renderer.block.model.BakedQuad $$0) {
+        this.unculledFaces.add((java.lang.Object)$$0);
         return this;
     }
 
-    public net.minecraft.client.resources.model.SimpleBakedModel$Builder particle(net.minecraft.client.renderer.texture.TextureAtlasSprite texture) {
-        this.particleIcon = texture;
+    public net.minecraft.client.resources.model.SimpleBakedModel$Builder particle(net.minecraft.client.renderer.texture.TextureAtlasSprite $$0) {
+        this.particleIcon = $$0;
         return this;
     }
 