--- net/minecraft/client/searchtree/SearchRegistry.java
+++ net/minecraft/client/searchtree/SearchRegistry.java
@@ -15,9 +15,9 @@
         this.searchTrees = new java.util.HashMap();
     }
 
-    public void onResourceManagerReload(net.minecraft.server.packs.resources.ResourceManager resourceManager) {
-        for (net.minecraft.client.searchtree.SearchRegistry$TreeEntry treeEntry : this.searchTrees.values()) {
-            treeEntry.refresh();
+    public void onResourceManagerReload(net.minecraft.server.packs.resources.ResourceManager $$0) {
+        for (net.minecraft.client.searchtree.SearchRegistry$TreeEntry $$1 : this.searchTrees.values()) {
+            $$1.refresh();
         }
     }
 