--- net/minecraft/client/searchtree/SuffixArray.java
+++ net/minecraft/client/searchtree/SuffixArray.java
@@ -25,18 +25,18 @@
         this.offsets = new it.unimi.dsi.fastutil.ints.IntArrayList();
     }
 
-    public void add(T object, java.lang.String string) {
-        this.maxStringLength = java.lang.Math.max((int)this.maxStringLength, (int)string.length());
-        int n = this.list.size();
-        this.list.add(object);
+    public void add(T $$0, java.lang.String $$1) {
+        this.maxStringLength = java.lang.Math.max((int)this.maxStringLength, (int)$$1.length());
+        int $$2 = this.list.size();
+        this.list.add($$0);
         this.wordStarts.add(this.chars.size());
-        for (int i = 0; i < string.length(); ++i) {
-            this.suffixToT.add(n);
-            this.offsets.add(i);
-            this.chars.add((int)string.charAt(i));
+        for (int $$3 = 0; $$3 < $$1.length(); ++$$3) {
+            this.suffixToT.add($$2);
+            this.offsets.add($$3);
+            this.chars.add((int)$$1.charAt($$3));
         }
-        this.suffixToT.add(n);
-        this.offsets.add(string.length());
+        this.suffixToT.add($$2);
+        this.offsets.add($$1.length());
         this.chars.add(-1);
     }
 