--- net/minecraft/client/gui/Font$StringRenderOutput.java
+++ net/minecraft/client/gui/Font$StringRenderOutput.java
@@ -22,11 +22,11 @@
     public java.util.List<net.minecraft.client.gui.font.glyphs.BakedGlyph$Effect> effects;
     public /* synthetic */ net.minecraft.client.gui.Font this$0;
 
-    public void addEffect(net.minecraft.client.gui.font.glyphs.BakedGlyph$Effect effect) {
+    public void addEffect(net.minecraft.client.gui.font.glyphs.BakedGlyph$Effect $$0) {
         if (this.effects == null) {
             this.effects = com.google.common.collect.Lists.newArrayList();
         }
-        this.effects.add((java.lang.Object)effect);
+        this.effects.add((java.lang.Object)$$0);
     }
 
     public net.minecraft.client.gui.Font$StringRenderOutput(net.minecraft.client.gui.Font font, net.minecraft.client.renderer.MultiBufferSource $$0, float $$1, float $$2, int $$3, boolean $$4, com.mojang.math.Matrix4f $$5, boolean $$6, int $$7) {