--- net/minecraft/server/rcon/NetworkDataOutputStream.java
+++ net/minecraft/server/rcon/NetworkDataOutputStream.java
@@ -13,21 +13,21 @@
         this.dataOutputStream = new java.io.DataOutputStream((java.io.OutputStream)this.outputStream);
     }
 
-    public void writeBytes(byte[] data) throws java.io.IOException {
-        this.dataOutputStream.write(data, 0, data.length);
+    public void writeBytes(byte[] $$0) throws java.io.IOException {
+        this.dataOutputStream.write($$0, 0, $$0.length);
     }
 
-    public void writeString(java.lang.String data) throws java.io.IOException {
-        this.dataOutputStream.writeBytes(data);
+    public void writeString(java.lang.String $$0) throws java.io.IOException {
+        this.dataOutputStream.writeBytes($$0);
         this.dataOutputStream.write(0);
     }
 
-    public void write(int data) throws java.io.IOException {
-        this.dataOutputStream.write(data);
+    public void write(int $$0) throws java.io.IOException {
+        this.dataOutputStream.write($$0);
     }
 
-    public void writeShort(short data) throws java.io.IOException {
-        this.dataOutputStream.writeShort((int)java.lang.Short.reverseBytes((short)data));
+    public void writeShort(short $$0) throws java.io.IOException {
+        this.dataOutputStream.writeShort((int)java.lang.Short.reverseBytes((short)$$0));
     }
 
     public void writeInt(int $$0) throws java.io.IOException {