--- net/minecraft/server/rcon/PktUtils.java
+++ net/minecraft/server/rcon/PktUtils.java
@@ -39,8 +39,8 @@
         return $$0[$$1] << 24 | ($$0[$$1 + 1] & 0xFF) << 16 | ($$0[$$1 + 2] & 0xFF) << 8 | $$0[$$1 + 3] & 0xFF;
     }
 
-    public static java.lang.String toHexString(byte input) {
-        return "" + HEX_CHAR[(input & 0xF0) >>> 4] + HEX_CHAR[input & 0xF];
+    public static java.lang.String toHexString(byte $$0) {
+        return "" + HEX_CHAR[($$0 & 0xF0) >>> 4] + HEX_CHAR[$$0 & 0xF];
     }
 
     static {