--- net/minecraft/client/renderer/block/model/BlockElementFace$Deserializer.java
+++ net/minecraft/client/renderer/block/model/BlockElementFace$Deserializer.java
@@ -20,18 +20,18 @@
         return new net.minecraft.client.renderer.block.model.BlockElementFace($$4, $$5, $$6, $$7);
     }
 
-    public int getTintIndex(com.google.gson.JsonObject object) {
-        return net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)object, (java.lang.String)"tintindex", (int)-1);
+    public int getTintIndex(com.google.gson.JsonObject $$0) {
+        return net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)$$0, (java.lang.String)"tintindex", (int)-1);
     }
 
-    public java.lang.String getTexture(com.google.gson.JsonObject object) {
-        return net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)object, (java.lang.String)"texture");
+    public java.lang.String getTexture(com.google.gson.JsonObject $$0) {
+        return net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"texture");
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.core.Direction getCullFacing(com.google.gson.JsonObject object) {
-        java.lang.String string = net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)object, (java.lang.String)"cullface", (java.lang.String)"");
-        return net.minecraft.core.Direction.byName((java.lang.String)string);
+    public net.minecraft.core.Direction getCullFacing(com.google.gson.JsonObject $$0) {
+        java.lang.String $$1 = net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"cullface", (java.lang.String)"");
+        return net.minecraft.core.Direction.byName((java.lang.String)$$1);
     }
 
     public /* synthetic */ java.lang.Object deserialize(com.google.gson.JsonElement jsonElement, java.lang.reflect.Type type, com.google.gson.JsonDeserializationContext jsonDeserializationContext) throws com.google.gson.JsonParseException {