--- net/minecraft/client/renderer/block/model/BlockElement.java
+++ net/minecraft/client/renderer/block/model/BlockElement.java
@@ -30,8 +30,8 @@
         }
     }
 
-    public float[] uvsByFace(net.minecraft.core.Direction facing) {
-        switch (net.minecraft.client.renderer.block.model.BlockElement$1.$SwitchMap$net$minecraft$core$Direction[facing.ordinal()]) {
+    public float[] uvsByFace(net.minecraft.core.Direction $$0) {
+        switch (net.minecraft.client.renderer.block.model.BlockElement$1.$SwitchMap$net$minecraft$core$Direction[$$0.ordinal()]) {
             case 1: {
                 return new float[]{this.from.x(), 16.0f - this.to.z(), this.to.x(), 16.0f - this.from.z()};
             }