--- net/minecraft/client/renderer/block/model/BlockModel$GuiLight.java
+++ net/minecraft/client/renderer/block/model/BlockModel$GuiLight.java
@@ -24,12 +24,12 @@
         this.name = $$0;
     }
 
-    public static net.minecraft.client.renderer.block.model.BlockModel$GuiLight getByName(java.lang.String name) {
-        for (net.minecraft.client.renderer.block.model.BlockModel$GuiLight guiLight : net.minecraft.client.renderer.block.model.BlockModel$GuiLight.values()) {
-            if (!guiLight.name.equals((java.lang.Object)name)) continue;
-            return guiLight;
+    public static net.minecraft.client.renderer.block.model.BlockModel$GuiLight getByName(java.lang.String $$0) {
+        for (net.minecraft.client.renderer.block.model.BlockModel$GuiLight $$1 : net.minecraft.client.renderer.block.model.BlockModel$GuiLight.values()) {
+            if (!$$1.name.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
-        throw new java.lang.IllegalArgumentException("Invalid gui light: " + name);
+        throw new java.lang.IllegalArgumentException("Invalid gui light: " + $$0);
     }
 
     public boolean lightLikeBlock() {