--- net/minecraft/client/renderer/block/model/BlockFaceUV.java
+++ net/minecraft/client/renderer/block/model/BlockFaceUV.java
@@ -13,33 +13,33 @@
         this.rotation = $$1;
     }
 
-    public float getU(int index) {
+    public float getU(int $$0) {
         if (this.uvs == null) {
             throw new java.lang.NullPointerException("uvs");
         }
-        int n = this.getShiftedIndex(index);
-        return this.uvs[n == 0 || n == 1 ? 0 : 2];
+        int $$1 = this.getShiftedIndex($$0);
+        return this.uvs[$$1 == 0 || $$1 == 1 ? 0 : 2];
     }
 
-    public float getV(int index) {
+    public float getV(int $$0) {
         if (this.uvs == null) {
             throw new java.lang.NullPointerException("uvs");
         }
-        int n = this.getShiftedIndex(index);
-        return this.uvs[n == 0 || n == 3 ? 1 : 3];
+        int $$1 = this.getShiftedIndex($$0);
+        return this.uvs[$$1 == 0 || $$1 == 3 ? 1 : 3];
     }
 
-    public int getShiftedIndex(int index) {
-        return (index + this.rotation / 90) % 4;
+    public int getShiftedIndex(int $$0) {
+        return ($$0 + this.rotation / 90) % 4;
     }
 
-    public int getReverseIndex(int index) {
-        return (index + 4 - this.rotation / 90) % 4;
+    public int getReverseIndex(int $$0) {
+        return ($$0 + 4 - this.rotation / 90) % 4;
     }
 
-    public void setMissingUv(float[] uvs) {
+    public void setMissingUv(float[] $$0) {
         if (this.uvs == null) {
-            this.uvs = uvs;
+            this.uvs = $$0;
         }
     }
 }