--- net/minecraft/client/renderer/block/model/BlockModelDefinition.java
+++ net/minecraft/client/renderer/block/model/BlockModelDefinition.java
@@ -8,10 +8,14 @@
     public java.util.Map<java.lang.String, net.minecraft.client.renderer.block.model.MultiVariant> variants;
     public net.minecraft.client.renderer.block.model.multipart.MultiPart multiPart;
 
-    public static net.minecraft.client.renderer.block.model.BlockModelDefinition fromStream(net.minecraft.client.renderer.block.model.BlockModelDefinition$Context containerHolder, java.io.Reader reader) {
-        return (net.minecraft.client.renderer.block.model.BlockModelDefinition)net.minecraft.util.GsonHelper.fromJson((com.google.gson.Gson)containerHolder.gson, (java.io.Reader)reader, net.minecraft.client.renderer.block.model.BlockModelDefinition.class);
+    public static net.minecraft.client.renderer.block.model.BlockModelDefinition fromStream(net.minecraft.client.renderer.block.model.BlockModelDefinition$Context $$0, java.io.Reader $$1) {
+        return (net.minecraft.client.renderer.block.model.BlockModelDefinition)net.minecraft.util.GsonHelper.fromJson((com.google.gson.Gson)$$0.gson, (java.io.Reader)$$1, net.minecraft.client.renderer.block.model.BlockModelDefinition.class);
     }
 
+    public static net.minecraft.client.renderer.block.model.BlockModelDefinition fromJsonElement(net.minecraft.client.renderer.block.model.BlockModelDefinition$Context $$0, com.google.gson.JsonElement $$1) {
+        return (net.minecraft.client.renderer.block.model.BlockModelDefinition)$$0.gson.fromJson($$1, net.minecraft.client.renderer.block.model.BlockModelDefinition.class);
+    }
+
     public net.minecraft.client.renderer.block.model.BlockModelDefinition(java.util.Map<java.lang.String, net.minecraft.client.renderer.block.model.MultiVariant> $$0, net.minecraft.client.renderer.block.model.multipart.MultiPart $$1) {
         this.variants = com.google.common.collect.Maps.newLinkedHashMap();
         this.multiPart = $$1;