--- net/minecraft/client/renderer/block/model/BlockFaceUV$Deserializer.java
+++ net/minecraft/client/renderer/block/model/BlockFaceUV$Deserializer.java
@@ -18,28 +18,28 @@
         return new net.minecraft.client.renderer.block.model.BlockFaceUV($$4, $$5);
     }
 
-    public int getRotation(com.google.gson.JsonObject object) {
-        int n = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)object, (java.lang.String)"rotation", (int)0);
-        if (n < 0 || n % 90 != 0 || n / 90 > 3) {
-            throw new com.google.gson.JsonParseException("Invalid rotation " + n + " found, only 0/90/180/270 allowed");
+    public int getRotation(com.google.gson.JsonObject $$0) {
+        int $$1 = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)$$0, (java.lang.String)"rotation", (int)0);
+        if ($$1 < 0 || $$1 % 90 != 0 || $$1 / 90 > 3) {
+            throw new com.google.gson.JsonParseException("Invalid rotation " + $$1 + " found, only 0/90/180/270 allowed");
         }
-        return n;
+        return $$1;
     }
 
     @javax.annotation.Nullable
-    public float[] getUVs(com.google.gson.JsonObject object) {
-        if (!object.has("uv")) {
+    public float[] getUVs(com.google.gson.JsonObject $$0) {
+        if (!$$0.has("uv")) {
             return null;
         }
-        com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)object, (java.lang.String)"uv");
-        if (jsonArray.size() != 4) {
-            throw new com.google.gson.JsonParseException("Expected 4 uv values, found: " + jsonArray.size());
+        com.google.gson.JsonArray $$1 = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)$$0, (java.lang.String)"uv");
+        if ($$1.size() != 4) {
+            throw new com.google.gson.JsonParseException("Expected 4 uv values, found: " + $$1.size());
         }
-        float[] fArray = new float[4];
-        for (int i = 0; i < fArray.length; ++i) {
-            fArray[i] = net.minecraft.util.GsonHelper.convertToFloat((com.google.gson.JsonElement)jsonArray.get(i), (java.lang.String)("uv[" + i + "]"));
+        float[] $$2 = new float[4];
+        for (int $$3 = 0; $$3 < $$2.length; ++$$3) {
+            $$2[$$3] = net.minecraft.util.GsonHelper.convertToFloat((com.google.gson.JsonElement)$$1.get($$3), (java.lang.String)("uv[" + $$3 + "]"));
         }
-        return fArray;
+        return $$2;
     }
 
     public /* synthetic */ java.lang.Object deserialize(com.google.gson.JsonElement jsonElement, java.lang.reflect.Type type, com.google.gson.JsonDeserializationContext jsonDeserializationContext) throws com.google.gson.JsonParseException {