--- net/minecraft/client/renderer/block/model/BlockModelDefinition$Deserializer.java
+++ net/minecraft/client/renderer/block/model/BlockModelDefinition$Deserializer.java
@@ -19,24 +19,24 @@
         return new net.minecraft.client.renderer.block.model.BlockModelDefinition($$4, $$5);
     }
 
-    public java.util.Map<java.lang.String, net.minecraft.client.renderer.block.model.MultiVariant> getVariants(com.google.gson.JsonDeserializationContext deserializationContext, com.google.gson.JsonObject object) {
-        java.util.HashMap hashMap = com.google.common.collect.Maps.newHashMap();
-        if (object.has("variants")) {
-            com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.getAsJsonObject((com.google.gson.JsonObject)object, (java.lang.String)"variants");
-            for (java.util.Map$Entry entry : jsonObject.entrySet()) {
-                hashMap.put((java.lang.Object)((java.lang.String)entry.getKey()), (java.lang.Object)((net.minecraft.client.renderer.block.model.MultiVariant)deserializationContext.deserialize((com.google.gson.JsonElement)entry.getValue(), net.minecraft.client.renderer.block.model.MultiVariant.class)));
+    public java.util.Map<java.lang.String, net.minecraft.client.renderer.block.model.MultiVariant> getVariants(com.google.gson.JsonDeserializationContext $$0, com.google.gson.JsonObject $$1) {
+        java.util.HashMap $$2 = com.google.common.collect.Maps.newHashMap();
+        if ($$1.has("variants")) {
+            com.google.gson.JsonObject $$3 = net.minecraft.util.GsonHelper.getAsJsonObject((com.google.gson.JsonObject)$$1, (java.lang.String)"variants");
+            for (java.util.Map$Entry $$4 : $$3.entrySet()) {
+                $$2.put((java.lang.Object)((java.lang.String)$$4.getKey()), (java.lang.Object)((net.minecraft.client.renderer.block.model.MultiVariant)$$0.deserialize((com.google.gson.JsonElement)$$4.getValue(), net.minecraft.client.renderer.block.model.MultiVariant.class)));
             }
         }
-        return hashMap;
+        return $$2;
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.client.renderer.block.model.multipart.MultiPart getMultiPart(com.google.gson.JsonDeserializationContext deserializationContext, com.google.gson.JsonObject object) {
-        if (!object.has("multipart")) {
+    public net.minecraft.client.renderer.block.model.multipart.MultiPart getMultiPart(com.google.gson.JsonDeserializationContext $$0, com.google.gson.JsonObject $$1) {
+        if (!$$1.has("multipart")) {
             return null;
         }
-        com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)object, (java.lang.String)"multipart");
-        return (net.minecraft.client.renderer.block.model.multipart.MultiPart)deserializationContext.deserialize((com.google.gson.JsonElement)jsonArray, net.minecraft.client.renderer.block.model.multipart.MultiPart.class);
+        com.google.gson.JsonArray $$2 = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)$$1, (java.lang.String)"multipart");
+        return (net.minecraft.client.renderer.block.model.multipart.MultiPart)$$0.deserialize((com.google.gson.JsonElement)$$2, net.minecraft.client.renderer.block.model.multipart.MultiPart.class);
     }
 
     public /* synthetic */ java.lang.Object deserialize(com.google.gson.JsonElement jsonElement, java.lang.reflect.Type type, com.google.gson.JsonDeserializationContext jsonDeserializationContext) throws com.google.gson.JsonParseException {