--- net/minecraft/client/renderer/block/model/BlockModelDefinition$Context.java
+++ net/minecraft/client/renderer/block/model/BlockModelDefinition$Context.java
@@ -16,8 +16,8 @@
         return this.definition;
     }
 
-    public void setDefinition(net.minecraft.world.level.block.state.StateDefinition<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> stateContainer) {
-        this.definition = stateContainer;
+    public void setDefinition(net.minecraft.world.level.block.state.StateDefinition<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        this.definition = $$0;
     }
 }
 