--- net/minecraft/client/renderer/block/model/ItemTransforms.java
+++ net/minecraft/client/renderer/block/model/ItemTransforms.java
@@ -41,8 +41,8 @@
         this.fixed = $$7;
     }
 
-    public net.minecraft.client.renderer.block.model.ItemTransform getTransform(net.minecraft.client.renderer.block.model.ItemTransforms$TransformType type) {
-        switch (net.minecraft.client.renderer.block.model.ItemTransforms$1.$SwitchMap$net$minecraft$client$renderer$block$model$ItemTransforms$TransformType[type.ordinal()]) {
+    public net.minecraft.client.renderer.block.model.ItemTransform getTransform(net.minecraft.client.renderer.block.model.ItemTransforms$TransformType $$0) {
+        switch (net.minecraft.client.renderer.block.model.ItemTransforms$1.$SwitchMap$net$minecraft$client$renderer$block$model$ItemTransforms$TransformType[$$0.ordinal()]) {
             case 1: {
                 return this.thirdPersonLeftHand;
             }
@@ -71,8 +71,8 @@
         return net.minecraft.client.renderer.block.model.ItemTransform.NO_TRANSFORM;
     }
 
-    public boolean hasTransform(net.minecraft.client.renderer.block.model.ItemTransforms$TransformType type) {
-        return this.getTransform(type) != net.minecraft.client.renderer.block.model.ItemTransform.NO_TRANSFORM;
+    public boolean hasTransform(net.minecraft.client.renderer.block.model.ItemTransforms$TransformType $$0) {
+        return this.getTransform($$0) != net.minecraft.client.renderer.block.model.ItemTransform.NO_TRANSFORM;
     }
 
     static {