--- net/minecraft/client/gui/screens/reporting/ChatSelectionScreen.java
+++ net/minecraft/client/gui/screens/reporting/ChatSelectionScreen.java
@@ -17,7 +17,7 @@
     public net.minecraft.client.gui.screens.reporting.ChatSelectionScreen$ChatSelectionList chatSelectionList;
     public net.minecraft.client.multiplayer.chat.report.ChatReportBuilder report;
     public java.util.function.Consumer<net.minecraft.client.multiplayer.chat.report.ChatReportBuilder> onSelected;
-    public net.minecraft.client.gui.screens.reporting.ChatSelectionLogFiller<net.minecraft.client.multiplayer.chat.LoggedChatMessage$Player> chatLogFiller;
+    public net.minecraft.client.gui.screens.reporting.ChatSelectionLogFiller chatLogFiller;
     @javax.annotation.Nullable
     public java.util.List<net.minecraft.util.FormattedCharSequence> tooltip;
 
@@ -30,7 +30,7 @@
     }
 
     public void init() {
-        this.chatLogFiller = new net.minecraft.client.gui.screens.reporting.ChatSelectionLogFiller(this.reportingContext.chatLog(), arg_0 -> this.canReport(arg_0), net.minecraft.client.multiplayer.chat.LoggedChatMessage$Player.class);
+        this.chatLogFiller = new net.minecraft.client.gui.screens.reporting.ChatSelectionLogFiller(this.reportingContext, arg_0 -> this.canReport(arg_0));
         this.contextInfoLabel = net.minecraft.client.gui.components.MultiLineLabel.create((net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.FormattedText)CONTEXT_INFO, (int)(this.width - 16));
         int n = this.contextInfoLabel.getLineCount() + 1;
         java.util.Objects.requireNonNull((java.lang.Object)this.font);