--- net/minecraft/client/gui/screens/reporting/ChatSelectionScreen$ChatSelectionList$MessageEntry.java
+++ net/minecraft/client/gui/screens/reporting/ChatSelectionScreen$ChatSelectionList$MessageEntry.java
@@ -127,7 +127,7 @@
     }
 
     static {
-        CHECKMARK_TEXTURE = new net.minecraft.resources.ResourceLocation("realms", "textures/gui/realms/checkmark.png");
+        CHECKMARK_TEXTURE = new net.minecraft.resources.ResourceLocation("minecraft", "textures/gui/checkmark.png");
     }
 }
 