--- net/minecraft/client/gui/screens/reporting/ChatSelectionScreen$ChatSelectionList.java
+++ net/minecraft/client/gui/screens/reporting/ChatSelectionScreen$ChatSelectionList.java
@@ -5,7 +5,7 @@
 
 public class net.minecraft.client.gui.screens.reporting.ChatSelectionScreen$ChatSelectionList
 extends net.minecraft.client.gui.components.ObjectSelectionList<net.minecraft.client.gui.screens.reporting.ChatSelectionScreen$ChatSelectionList$Entry>
-implements net.minecraft.client.gui.screens.reporting.ChatSelectionLogFiller$Output<net.minecraft.client.multiplayer.chat.LoggedChatMessage$Player>
+implements net.minecraft.client.gui.screens.reporting.ChatSelectionLogFiller$Output
  {
     @javax.annotation.Nullable
     public net.minecraft.client.gui.screens.reporting.ChatSelectionScreen$ChatSelectionList$Heading previousHeading;