--- net/minecraft/server/packs/linkfs/DummyFileAttributes.java
+++ net/minecraft/server/packs/linkfs/DummyFileAttributes.java
@@ -1,0 +1,47 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.linkfs;
+
+public abstract class net.minecraft.server.packs.linkfs.DummyFileAttributes
+implements java.nio.file.attribute.BasicFileAttributes
+ {
+    public static java.nio.file.attribute.FileTime EPOCH;
+
+    public net.minecraft.server.packs.linkfs.DummyFileAttributes() {
+    }
+
+    public java.nio.file.attribute.FileTime lastModifiedTime() {
+        return EPOCH;
+    }
+
+    public java.nio.file.attribute.FileTime lastAccessTime() {
+        return EPOCH;
+    }
+
+    public java.nio.file.attribute.FileTime creationTime() {
+        return EPOCH;
+    }
+
+    public boolean isSymbolicLink() {
+        return false;
+    }
+
+    public boolean isOther() {
+        return false;
+    }
+
+    public long size() {
+        return 0L;
+    }
+
+    @javax.annotation.Nullable
+    public java.lang.Object fileKey() {
+        return null;
+    }
+
+    static {
+        EPOCH = java.nio.file.attribute.FileTime.fromMillis((long)0L);
+    }
+}
+