--- net/minecraft/server/packs/linkfs/LinkFSPath$1.java
+++ net/minecraft/server/packs/linkfs/LinkFSPath$1.java
@@ -1,0 +1,20 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.linkfs;
+
+public class net.minecraft.server.packs.linkfs.LinkFSPath$1
+extends net.minecraft.server.packs.linkfs.DummyFileAttributes
+ {
+    public net.minecraft.server.packs.linkfs.LinkFSPath$1() {
+    }
+
+    public boolean isRegularFile() {
+        return false;
+    }
+
+    public boolean isDirectory() {
+        return true;
+    }
+}
+