--- net/minecraft/server/packs/linkfs/LinkFSProvider$2.java
+++ net/minecraft/server/packs/linkfs/LinkFSProvider$2.java
@@ -1,0 +1,32 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.linkfs;
+
+public class net.minecraft.server.packs.linkfs.LinkFSProvider$2
+ {
+    public static /* synthetic */ int[] $SwitchMap$java$nio$file$AccessMode;
+
+    static {
+        $SwitchMap$java$nio$file$AccessMode = new int[java.nio.file.AccessMode.values().length];
+        try {
+            net.minecraft.server.packs.linkfs.LinkFSProvider$2.$SwitchMap$java$nio$file$AccessMode[java.nio.file.AccessMode.READ.ordinal()] = 1;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.server.packs.linkfs.LinkFSProvider$2.$SwitchMap$java$nio$file$AccessMode[java.nio.file.AccessMode.EXECUTE.ordinal()] = 2;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.server.packs.linkfs.LinkFSProvider$2.$SwitchMap$java$nio$file$AccessMode[java.nio.file.AccessMode.WRITE.ordinal()] = 3;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+    }
+}
+