--- net/minecraft/server/packs/linkfs/LinkFSFileStore.java
+++ net/minecraft/server/packs/linkfs/LinkFSFileStore.java
@@ -1,0 +1,56 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.linkfs;
+
+public class net.minecraft.server.packs.linkfs.LinkFSFileStore
+extends java.nio.file.FileStore
+ {
+    public java.lang.String name;
+
+    public net.minecraft.server.packs.linkfs.LinkFSFileStore(java.lang.String $$0) {
+        this.name = $$0;
+    }
+
+    public java.lang.String name() {
+        return this.name;
+    }
+
+    public java.lang.String type() {
+        return "index";
+    }
+
+    public boolean isReadOnly() {
+        return true;
+    }
+
+    public long getTotalSpace() {
+        return 0L;
+    }
+
+    public long getUsableSpace() {
+        return 0L;
+    }
+
+    public long getUnallocatedSpace() {
+        return 0L;
+    }
+
+    public boolean supportsFileAttributeView(java.lang.Class<? extends java.nio.file.attribute.FileAttributeView> $$0) {
+        return $$0 == java.nio.file.attribute.BasicFileAttributeView.class;
+    }
+
+    public boolean supportsFileAttributeView(java.lang.String $$0) {
+        return "basic".equals((java.lang.Object)$$0);
+    }
+
+    @javax.annotation.Nullable
+    public <V extends java.nio.file.attribute.FileStoreAttributeView> V getFileStoreAttributeView(java.lang.Class<V> $$0) {
+        return null;
+    }
+
+    public java.lang.Object getAttribute(java.lang.String $$0) throws java.io.IOException {
+        throw new java.lang.UnsupportedOperationException();
+    }
+}
+