--- net/minecraft/server/packs/linkfs/PathContents.java
+++ net/minecraft/server/packs/linkfs/PathContents.java
@@ -1,0 +1,16 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.linkfs;
+
+public interface net.minecraft.server.packs.linkfs.PathContents
+ {
+    public static final net.minecraft.server.packs.linkfs.PathContents MISSING;
+    public static final net.minecraft.server.packs.linkfs.PathContents RELATIVE;
+
+    static {
+        MISSING = new /* Unavailable Anonymous Inner Class!! */;
+        RELATIVE = new /* Unavailable Anonymous Inner Class!! */;
+    }
+}
+