--- net/minecraft/server/packs/linkfs/PathContents$2.java
+++ net/minecraft/server/packs/linkfs/PathContents$2.java
@@ -1,0 +1,16 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.linkfs;
+
+public class net.minecraft.server.packs.linkfs.PathContents$2
+implements net.minecraft.server.packs.linkfs.PathContents
+ {
+    public net.minecraft.server.packs.linkfs.PathContents$2() {
+    }
+
+    public java.lang.String toString() {
+        return "relative";
+    }
+}
+