--- net/minecraft/world/level/levelgen/feature/treedecorators/AlterGroundDecorator.java
+++ net/minecraft/world/level/levelgen/feature/treedecorators/AlterGroundDecorator.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.treedecorators.AlterGroundDecorator> CODEC;
     public net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider provider;
 
-    public net.minecraft.world.level.levelgen.feature.treedecorators.AlterGroundDecorator(net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider provider) {
-        this.provider = provider;
+    public net.minecraft.world.level.levelgen.feature.treedecorators.AlterGroundDecorator(net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider $$0) {
+        this.provider = $$0;
     }
 
     public net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType<?> type() {