--- net/minecraft/world/level/levelgen/feature/treedecorators/BeehiveDecorator.java
+++ net/minecraft/world/level/levelgen/feature/treedecorators/BeehiveDecorator.java
@@ -11,8 +11,8 @@
     public static net.minecraft.core.Direction[] SPAWN_DIRECTIONS;
     public float probability;
 
-    public net.minecraft.world.level.levelgen.feature.treedecorators.BeehiveDecorator(float probability) {
-        this.probability = probability;
+    public net.minecraft.world.level.levelgen.feature.treedecorators.BeehiveDecorator(float $$0) {
+        this.probability = $$0;
     }
 
     public net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType<?> type() {