--- net/minecraft/world/level/levelgen/feature/treedecorators/CocoaDecorator.java
+++ net/minecraft/world/level/levelgen/feature/treedecorators/CocoaDecorator.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.treedecorators.CocoaDecorator> CODEC;
     public float probability;
 
-    public net.minecraft.world.level.levelgen.feature.treedecorators.CocoaDecorator(float probability) {
-        this.probability = probability;
+    public net.minecraft.world.level.levelgen.feature.treedecorators.CocoaDecorator(float $$0) {
+        this.probability = $$0;
     }
 
     public net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType<?> type() {