--- net/minecraft/world/level/levelgen/feature/treedecorators/TreeDecoratorType.java
+++ net/minecraft/world/level/levelgen/feature/treedecorators/TreeDecoratorType.java
@@ -13,12 +13,12 @@
     public static net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType<net.minecraft.world.level.levelgen.feature.treedecorators.AttachedToLeavesDecorator> ATTACHED_TO_LEAVES;
     public com.mojang.serialization.Codec<P> codec;
 
-    public static <P extends net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecorator> net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType<P> register(java.lang.String name, com.mojang.serialization.Codec<P> codec) {
-        return (net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.TREE_DECORATOR_TYPES, (java.lang.String)name, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType(codec));
+    public static <P extends net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecorator> net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType<P> register(java.lang.String $$0, com.mojang.serialization.Codec<P> $$1) {
+        return (net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.TREE_DECORATOR_TYPES, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType($$1));
     }
 
-    public net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType(com.mojang.serialization.Codec<P> codec) {
-        this.codec = codec;
+    public net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecoratorType(com.mojang.serialization.Codec<P> $$0) {
+        this.codec = $$0;
     }
 
     public com.mojang.serialization.Codec<P> codec() {