--- net/minecraft/world/entity/animal/horse/Horse$HorseGroupData.java
+++ net/minecraft/world/entity/animal/horse/Horse$HorseGroupData.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.animal.horse.Variant variant;
 
-    public net.minecraft.world.entity.animal.horse.Horse$HorseGroupData(net.minecraft.world.entity.animal.horse.Variant variant) {
+    public net.minecraft.world.entity.animal.horse.Horse$HorseGroupData(net.minecraft.world.entity.animal.horse.Variant $$0) {
         super(true);
-        this.variant = variant;
+        this.variant = $$0;
     }
 }
 