--- net/minecraft/world/entity/animal/horse/Llama$LlamaAttackWolfGoal.java
+++ net/minecraft/world/entity/animal/horse/Llama$LlamaAttackWolfGoal.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.entity.animal.horse.Llama$LlamaAttackWolfGoal
 extends net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal<net.minecraft.world.entity.animal.Wolf>
  {
-    public net.minecraft.world.entity.animal.horse.Llama$LlamaAttackWolfGoal(net.minecraft.world.entity.animal.horse.Llama llama) {
-        super((net.minecraft.world.entity.Mob)llama, net.minecraft.world.entity.animal.Wolf.class, 16, false, true, entity -> !((net.minecraft.world.entity.animal.Wolf)entity).isTame());
+    public net.minecraft.world.entity.animal.horse.Llama$LlamaAttackWolfGoal(net.minecraft.world.entity.animal.horse.Llama $$02) {
+        super((net.minecraft.world.entity.Mob)$$02, net.minecraft.world.entity.animal.Wolf.class, 16, false, true, $$0 -> !((net.minecraft.world.entity.animal.Wolf)$$0).isTame());
     }
 
     public double getFollowDistance() {