--- net/minecraft/world/entity/animal/horse/Llama$LlamaGroupData.java
+++ net/minecraft/world/entity/animal/horse/Llama$LlamaGroupData.java
@@ -8,9 +8,9 @@
  {
     public int variant;
 
-    public net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData(int variant) {
+    public net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData(int $$0) {
         super(true);
-        this.variant = variant;
+        this.variant = $$0;
     }
 }
 