--- net/minecraft/world/entity/animal/horse/Llama$LlamaHurtByTargetGoal.java
+++ net/minecraft/world/entity/animal/horse/Llama$LlamaHurtByTargetGoal.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.entity.animal.horse.Llama$LlamaHurtByTargetGoal
 extends net.minecraft.world.entity.ai.goal.target.HurtByTargetGoal
  {
-    public net.minecraft.world.entity.animal.horse.Llama$LlamaHurtByTargetGoal(net.minecraft.world.entity.animal.horse.Llama llama) {
-        super((net.minecraft.world.entity.PathfinderMob)llama, new java.lang.Class[0]);
+    public net.minecraft.world.entity.animal.horse.Llama$LlamaHurtByTargetGoal(net.minecraft.world.entity.animal.horse.Llama $$0) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, new java.lang.Class[0]);
     }
 
     public boolean canContinueToUse() {