--- net/minecraft/world/entity/animal/horse/Llama.java
+++ net/minecraft/world/entity/animal/horse/Llama.java
@@ -27,8 +27,8 @@
         return false;
     }
 
-    public void setStrength(int strength) {
-        this.entityData.set(DATA_STRENGTH_ID, (java.lang.Object)java.lang.Math.max((int)1, (int)java.lang.Math.min((int)5, (int)strength)));
+    public void setStrength(int $$0) {
+        this.entityData.set(DATA_STRENGTH_ID, (java.lang.Object)java.lang.Math.max((int)1, (int)java.lang.Math.min((int)5, (int)$$0)));
     }
 
     public void setRandomStrength(net.minecraft.util.RandomSource $$0) {
@@ -40,21 +40,21 @@
         return (java.lang.Integer)this.entityData.get(DATA_STRENGTH_ID);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("Variant", this.getVariant());
-        compound.putInt("Strength", this.getStrength());
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("Variant", this.getVariant());
+        $$0.putInt("Strength", this.getStrength());
         if (!this.inventory.getItem(1).isEmpty()) {
-            compound.put("DecorItem", (net.minecraft.nbt.Tag)this.inventory.getItem(1).save(new net.minecraft.nbt.CompoundTag()));
+            $$0.put("DecorItem", (net.minecraft.nbt.Tag)this.inventory.getItem(1).save(new net.minecraft.nbt.CompoundTag()));
         }
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        this.setStrength(compound.getInt("Strength"));
-        super.readAdditionalSaveData(compound);
-        this.setVariant(compound.getInt("Variant"));
-        if (compound.contains("DecorItem", 10)) {
-            this.inventory.setItem(1, net.minecraft.world.item.ItemStack.of((net.minecraft.nbt.CompoundTag)compound.getCompound("DecorItem")));
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        this.setStrength($$0.getInt("Strength"));
+        super.readAdditionalSaveData($$0);
+        this.setVariant($$0.getInt("Variant"));
+        if ($$0.contains("DecorItem", 10)) {
+            this.inventory.setItem(1, net.minecraft.world.item.ItemStack.of((net.minecraft.nbt.CompoundTag)$$0.getCompound("DecorItem")));
         }
         this.updateContainerEquipment();
     }
@@ -90,8 +90,8 @@
         return net.minecraft.util.Mth.clamp((int)((java.lang.Integer)this.entityData.get(DATA_VARIANT_ID)), (int)0, (int)3);
     }
 
-    public void setVariant(int variant) {
-        this.entityData.set(DATA_VARIANT_ID, (java.lang.Object)variant);
+    public void setVariant(int $$0) {
+        this.entityData.set(DATA_VARIANT_ID, (java.lang.Object)$$0);
     }
 
     public int getInventorySize() {
@@ -101,14 +101,14 @@
         return super.getInventorySize();
     }
 
-    public void positionRider(net.minecraft.world.entity.Entity passenger) {
-        if (!this.hasPassenger(passenger)) {
+    public void positionRider(net.minecraft.world.entity.Entity $$0) {
+        if (!this.hasPassenger($$0)) {
             return;
         }
-        float f = net.minecraft.util.Mth.cos((float)(this.yBodyRot * ((float)java.lang.Math.PI / 180)));
-        float f2 = net.minecraft.util.Mth.sin((float)(this.yBodyRot * ((float)java.lang.Math.PI / 180)));
-        float f3 = 0.3f;
-        passenger.setPos(this.getX() + (double)(0.3f * f2), this.getY() + this.getPassengersRidingOffset() + passenger.getMyRidingOffset(), this.getZ() - (double)(0.3f * f));
+        float $$1 = net.minecraft.util.Mth.cos((float)(this.yBodyRot * ((float)java.lang.Math.PI / 180)));
+        float $$2 = net.minecraft.util.Mth.sin((float)(this.yBodyRot * ((float)java.lang.Math.PI / 180)));
+        float $$3 = 0.3f;
+        $$0.setPos(this.getX() + (double)(0.3f * $$2), this.getY() + this.getPassengersRidingOffset() + $$0.getMyRidingOffset(), this.getZ() - (double)(0.3f * $$1));
     }
 
     public double getPassengersRidingOffset() {
@@ -120,50 +120,50 @@
         return null;
     }
 
-    public boolean isFood(net.minecraft.world.item.ItemStack stack) {
-        return FOOD_ITEMS.test(stack);
+    public boolean isFood(net.minecraft.world.item.ItemStack $$0) {
+        return FOOD_ITEMS.test($$0);
     }
 
-    public boolean handleEating(net.minecraft.world.entity.player.Player player, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.sounds.SoundEvent soundEvent;
-        int n = 0;
-        int n2 = 0;
-        float f = 0.0f;
-        boolean bl = false;
-        if (stack.is(net.minecraft.world.item.Items.WHEAT)) {
-            n = 10;
-            n2 = 3;
-            f = 2.0f;
-        } else if (stack.is(net.minecraft.world.level.block.Blocks.HAY_BLOCK.asItem())) {
-            n = 90;
-            n2 = 6;
-            f = 10.0f;
+    public boolean handleEating(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.sounds.SoundEvent $$6;
+        int $$2 = 0;
+        int $$3 = 0;
+        float $$4 = 0.0f;
+        boolean $$5 = false;
+        if ($$1.is(net.minecraft.world.item.Items.WHEAT)) {
+            $$2 = 10;
+            $$3 = 3;
+            $$4 = 2.0f;
+        } else if ($$1.is(net.minecraft.world.level.block.Blocks.HAY_BLOCK.asItem())) {
+            $$2 = 90;
+            $$3 = 6;
+            $$4 = 10.0f;
             if (this.isTamed() && this.getAge() == 0 && this.canFallInLove()) {
-                bl = true;
-                this.setInLove(player);
+                $$5 = true;
+                this.setInLove($$0);
             }
         }
-        if (this.getHealth() < this.getMaxHealth() && f > 0.0f) {
-            this.heal(f);
-            bl = true;
+        if (this.getHealth() < this.getMaxHealth() && $$4 > 0.0f) {
+            this.heal($$4);
+            $$5 = true;
         }
-        if (this.isBaby() && n > 0) {
+        if (this.isBaby() && $$2 > 0) {
             this.level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.HAPPY_VILLAGER, this.getRandomX(1.0), this.getRandomY() + 0.5, this.getRandomZ(1.0), 0.0, 0.0, 0.0);
             if (!this.level.isClientSide) {
-                this.ageUp(n);
+                this.ageUp($$2);
             }
-            bl = true;
+            $$5 = true;
         }
-        if (n2 > 0 && (bl || !this.isTamed()) && this.getTemper() < this.getMaxTemper()) {
-            bl = true;
+        if ($$3 > 0 && ($$5 || !this.isTamed()) && this.getTemper() < this.getMaxTemper()) {
+            $$5 = true;
             if (!this.level.isClientSide) {
-                this.modifyTemper(n2);
+                this.modifyTemper($$3);
             }
         }
-        if (bl && !this.isSilent() && (soundEvent = this.getEatingSound()) != null) {
+        if ($$5 && !this.isSilent() && ($$6 = this.getEatingSound()) != null) {
             this.level.playSound(null, this.getX(), this.getY(), this.getZ(), this.getEatingSound(), this.getSoundSource(), 1.0f, 1.0f + (this.random.nextFloat() - this.random.nextFloat()) * 0.2f);
         }
-        return bl;
+        return $$5;
     }
 
     public boolean isImmobile() {
@@ -171,20 +171,24 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor level, net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.MobSpawnType reason, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData spawnData, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag dataTag) {
-        int n;
-        net.minecraft.util.RandomSource randomSource = level.getRandom();
-        this.setRandomStrength(randomSource);
-        if (spawnData instanceof net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData) {
-            n = ((net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData)spawnData).variant;
+    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.world.DifficultyInstance $$1, net.minecraft.world.entity.MobSpawnType $$2, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData $$3, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$4) {
+        int $$7;
+        net.minecraft.util.RandomSource $$5 = $$0.getRandom();
+        this.setRandomStrength($$5);
+        if ($$3 instanceof net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData) {
+            int $$6 = ((net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData)$$3).variant;
         } else {
-            n = randomSource.nextInt(4);
-            spawnData = new net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData(n);
+            $$7 = $$5.nextInt(4);
+            $$3 = new net.minecraft.world.entity.animal.horse.Llama$LlamaGroupData($$7);
         }
-        this.setVariant(n);
-        return super.finalizeSpawn(level, difficulty, reason, spawnData, dataTag);
+        this.setVariant($$7);
+        return super.finalizeSpawn($$0, $$1, $$2, $$3, $$4);
     }
 
+    public boolean canPerformRearing() {
+        return false;
+    }
+
     public net.minecraft.sounds.SoundEvent getAngrySound() {
         return net.minecraft.sounds.SoundEvents.LLAMA_ANGRY;
     }
@@ -193,7 +197,7 @@
         return net.minecraft.sounds.SoundEvents.LLAMA_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.LLAMA_HURT;
     }
 
@@ -206,7 +210,7 @@
         return net.minecraft.sounds.SoundEvents.LLAMA_EAT;
     }
 
-    public void playStepSound(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState block) {
+    public void playStepSound(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         this.playSound(net.minecraft.sounds.SoundEvents.LLAMA_STEP, 0.15f, 1.0f);
     }
 
@@ -214,13 +218,6 @@
         this.playSound(net.minecraft.sounds.SoundEvents.LLAMA_CHEST, 1.0f, (this.random.nextFloat() - this.random.nextFloat()) * 0.2f + 1.0f);
     }
 
-    public void makeMad() {
-        net.minecraft.sounds.SoundEvent $$0 = this.getAngrySound();
-        if ($$0 != null) {
-            this.playSound($$0, this.getSoundVolume(), this.getVoicePitch());
-        }
-    }
-
     public int getInventoryColumns() {
         return this.getStrength();
     }
@@ -233,19 +230,19 @@
         return !this.inventory.getItem(1).isEmpty();
     }
 
-    public boolean isArmor(net.minecraft.world.item.ItemStack stack) {
-        return stack.is(net.minecraft.tags.ItemTags.WOOL_CARPETS);
+    public boolean isArmor(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(net.minecraft.tags.ItemTags.WOOL_CARPETS);
     }
 
     public boolean isSaddleable() {
         return false;
     }
 
-    public void containerChanged(net.minecraft.world.Container invBasic) {
-        net.minecraft.world.item.DyeColor dyeColor = this.getSwag();
-        super.containerChanged(invBasic);
-        net.minecraft.world.item.DyeColor dyeColor2 = this.getSwag();
-        if (this.tickCount > 20 && dyeColor2 != null && dyeColor2 != dyeColor) {
+    public void containerChanged(net.minecraft.world.Container $$0) {
+        net.minecraft.world.item.DyeColor $$1 = this.getSwag();
+        super.containerChanged($$0);
+        net.minecraft.world.item.DyeColor $$2 = this.getSwag();
+        if (this.tickCount > 20 && $$2 != null && $$2 != $$1) {
             this.playSound(net.minecraft.sounds.SoundEvents.LLAMA_SWAG, 0.5f, 1.0f);
         }
     }
@@ -258,15 +255,15 @@
         this.setSwag(net.minecraft.world.entity.animal.horse.Llama.getDyeColor((net.minecraft.world.item.ItemStack)this.inventory.getItem(1)));
     }
 
-    public void setSwag(@javax.annotation.Nullable net.minecraft.world.item.DyeColor color) {
-        this.entityData.set(DATA_SWAG_ID, (java.lang.Object)(color == null ? -1 : color.getId()));
+    public void setSwag(@javax.annotation.Nullable net.minecraft.world.item.DyeColor $$0) {
+        this.entityData.set(DATA_SWAG_ID, (java.lang.Object)($$0 == null ? -1 : $$0.getId()));
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.item.DyeColor getDyeColor(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.world.level.block.Block block = net.minecraft.world.level.block.Block.byItem((net.minecraft.world.item.Item)stack.getItem());
-        if (block instanceof net.minecraft.world.level.block.WoolCarpetBlock) {
-            return ((net.minecraft.world.level.block.WoolCarpetBlock)block).getColor();
+    public static net.minecraft.world.item.DyeColor getDyeColor(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.world.level.block.Block $$1 = net.minecraft.world.level.block.Block.byItem((net.minecraft.world.item.Item)$$0.getItem());
+        if ($$1 instanceof net.minecraft.world.level.block.WoolCarpetBlock) {
+            return ((net.minecraft.world.level.block.WoolCarpetBlock)$$1).getColor();
         }
         return null;
     }
@@ -281,43 +278,47 @@
         return 30;
     }
 
-    public boolean canMate(net.minecraft.world.entity.animal.Animal otherAnimal) {
-        return otherAnimal != this && otherAnimal instanceof net.minecraft.world.entity.animal.horse.Llama && this.canParent() && ((net.minecraft.world.entity.animal.horse.Llama)otherAnimal).canParent();
+    public boolean canMate(net.minecraft.world.entity.animal.Animal $$0) {
+        return $$0 != this && $$0 instanceof net.minecraft.world.entity.animal.horse.Llama && this.canParent() && ((net.minecraft.world.entity.animal.horse.Llama)$$0).canParent();
     }
 
+    @javax.annotation.Nullable
     public net.minecraft.world.entity.animal.horse.Llama getBreedOffspring(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.AgeableMob $$1) {
-        net.minecraft.world.entity.animal.horse.Llama $$2 = this.makeBabyLlama();
-        this.setOffspringAttributes($$1, (net.minecraft.world.entity.animal.horse.AbstractHorse)$$2);
-        net.minecraft.world.entity.animal.horse.Llama $$3 = (net.minecraft.world.entity.animal.horse.Llama)$$1;
-        int $$4 = this.random.nextInt(java.lang.Math.max((int)this.getStrength(), (int)$$3.getStrength())) + 1;
-        if (this.random.nextFloat() < 0.03f) {
-            ++$$4;
+        net.minecraft.world.entity.animal.horse.Llama $$2 = this.makeNewLlama();
+        if ($$2 != null) {
+            this.setOffspringAttributes($$1, (net.minecraft.world.entity.animal.horse.AbstractHorse)$$2);
+            net.minecraft.world.entity.animal.horse.Llama $$3 = (net.minecraft.world.entity.animal.horse.Llama)$$1;
+            int $$4 = this.random.nextInt(java.lang.Math.max((int)this.getStrength(), (int)$$3.getStrength())) + 1;
+            if (this.random.nextFloat() < 0.03f) {
+                ++$$4;
+            }
+            $$2.setStrength($$4);
+            $$2.setVariant(this.random.nextBoolean() ? this.getVariant() : $$3.getVariant());
         }
-        $$2.setStrength($$4);
-        $$2.setVariant(this.random.nextBoolean() ? this.getVariant() : $$3.getVariant());
         return $$2;
     }
 
-    public net.minecraft.world.entity.animal.horse.Llama makeBabyLlama() {
+    @javax.annotation.Nullable
+    public net.minecraft.world.entity.animal.horse.Llama makeNewLlama() {
         return (net.minecraft.world.entity.animal.horse.Llama)net.minecraft.world.entity.EntityType.LLAMA.create(this.level);
     }
 
-    public void spit(net.minecraft.world.entity.LivingEntity target) {
-        net.minecraft.world.entity.projectile.LlamaSpit llamaSpit = new net.minecraft.world.entity.projectile.LlamaSpit(this.level, this);
-        double d = target.getX() - this.getX();
-        double d2 = target.getY(0.3333333333333333) - llamaSpit.getY();
-        double d3 = target.getZ() - this.getZ();
-        double d4 = java.lang.Math.sqrt((double)(d * d + d3 * d3)) * (double)0.2f;
-        llamaSpit.shoot(d, d2 + d4, d3, 1.5f, 10.0f);
+    public void spit(net.minecraft.world.entity.LivingEntity $$0) {
+        net.minecraft.world.entity.projectile.LlamaSpit $$1 = new net.minecraft.world.entity.projectile.LlamaSpit(this.level, this);
+        double $$2 = $$0.getX() - this.getX();
+        double $$3 = $$0.getY(0.3333333333333333) - $$1.getY();
+        double $$4 = $$0.getZ() - this.getZ();
+        double $$5 = java.lang.Math.sqrt((double)($$2 * $$2 + $$4 * $$4)) * (double)0.2f;
+        $$1.shoot($$2, $$3 + $$5, $$4, 1.5f, 10.0f);
         if (!this.isSilent()) {
             this.level.playSound(null, this.getX(), this.getY(), this.getZ(), net.minecraft.sounds.SoundEvents.LLAMA_SPIT, this.getSoundSource(), 1.0f, 1.0f + (this.random.nextFloat() - this.random.nextFloat()) * 0.2f);
         }
-        this.level.addFreshEntity((net.minecraft.world.entity.Entity)llamaSpit);
+        this.level.addFreshEntity((net.minecraft.world.entity.Entity)$$1);
         this.didSpit = true;
     }
 
-    public void setDidSpit(boolean didSpit) {
-        this.didSpit = didSpit;
+    public void setDidSpit(boolean $$0) {
+        this.didSpit = $$0;
     }
 
     public boolean causeFallDamage(float $$0, float $$1, net.minecraft.world.damagesource.DamageSource $$2) {
@@ -344,8 +345,8 @@
         this.caravanHead = null;
     }
 
-    public void joinCaravan(net.minecraft.world.entity.animal.horse.Llama caravanHead) {
-        this.caravanHead = caravanHead;
+    public void joinCaravan(net.minecraft.world.entity.animal.horse.Llama $$0) {
+        this.caravanHead = $$0;
         this.caravanHead.caravanTail = this;
     }
 
@@ -376,14 +377,15 @@
         return false;
     }
 
-    public void performRangedAttack(net.minecraft.world.entity.LivingEntity target, float distanceFactor) {
-        this.spit(target);
+    public void performRangedAttack(net.minecraft.world.entity.LivingEntity $$0, float $$1) {
+        this.spit($$0);
     }
 
     public net.minecraft.world.phys.Vec3 getLeashOffset() {
         return new net.minecraft.world.phys.Vec3(0.0, 0.75 * (double)this.getEyeHeight(), (double)this.getBbWidth() * 0.5);
     }
 
+    @javax.annotation.Nullable
     public /* synthetic */ net.minecraft.world.entity.AgeableMob getBreedOffspring(net.minecraft.server.level.ServerLevel serverLevel, net.minecraft.world.entity.AgeableMob ageableMob) {
         return this.getBreedOffspring(serverLevel, ageableMob);
     }