--- net/minecraft/world/entity/animal/horse/Markings.java
+++ net/minecraft/world/entity/animal/horse/Markings.java
@@ -23,17 +23,17 @@
         return (net.minecraft.world.entity.animal.horse.Markings)java.lang.Enum.valueOf(net.minecraft.world.entity.animal.horse.Markings.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.animal.horse.Markings(int id) {
+    public net.minecraft.world.entity.animal.horse.Markings(int $$0) {
         super(string, n);
-        this.id = id;
+        this.id = $$0;
     }
 
     public int getId() {
         return this.id;
     }
 
-    public static net.minecraft.world.entity.animal.horse.Markings byId(int id) {
-        return BY_ID[id % BY_ID.length];
+    public static net.minecraft.world.entity.animal.horse.Markings byId(int $$0) {
+        return BY_ID[$$0 % BY_ID.length];
     }
 
     public static /* synthetic */ net.minecraft.world.entity.animal.horse.Markings[] lambda$static$0(int $$0) {