--- net/minecraft/world/entity/animal/horse/Mule.java
+++ net/minecraft/world/entity/animal/horse/Mule.java
@@ -11,17 +11,14 @@
     }
 
     public net.minecraft.sounds.SoundEvent getAmbientSound() {
-        super.getAmbientSound();
         return net.minecraft.sounds.SoundEvents.MULE_AMBIENT;
     }
 
     public net.minecraft.sounds.SoundEvent getAngrySound() {
-        super.getAngrySound();
         return net.minecraft.sounds.SoundEvents.MULE_ANGRY;
     }
 
     public net.minecraft.sounds.SoundEvent getDeathSound() {
-        super.getDeathSound();
         return net.minecraft.sounds.SoundEvents.MULE_DEATH;
     }
 
@@ -30,8 +27,7 @@
         return net.minecraft.sounds.SoundEvents.MULE_EAT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
-        super.getHurtSound(damageSource);
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.MULE_HURT;
     }
 
@@ -39,6 +35,7 @@
         this.playSound(net.minecraft.sounds.SoundEvents.MULE_CHEST, 1.0f, (this.random.nextFloat() - this.random.nextFloat()) * 0.2f + 1.0f);
     }
 
+    @javax.annotation.Nullable
     public net.minecraft.world.entity.AgeableMob getBreedOffspring(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.AgeableMob $$1) {
         return (net.minecraft.world.entity.AgeableMob)net.minecraft.world.entity.EntityType.MULE.create((net.minecraft.world.level.Level)$$0);
     }