--- net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
+++ net/minecraft/world/entity/animal/horse/SkeletonTrapGoal.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.entity.animal.horse.SkeletonHorse horse;
 
-    public net.minecraft.world.entity.animal.horse.SkeletonTrapGoal(net.minecraft.world.entity.animal.horse.SkeletonHorse horse) {
-        this.horse = horse;
+    public net.minecraft.world.entity.animal.horse.SkeletonTrapGoal(net.minecraft.world.entity.animal.horse.SkeletonHorse $$0) {
+        this.horse = $$0;
     }
 
     public boolean canUse() {
@@ -23,49 +23,62 @@
         this.horse.setTamed(true);
         this.horse.setAge(0);
         net.minecraft.world.entity.LightningBolt $$2 = (net.minecraft.world.entity.LightningBolt)net.minecraft.world.entity.EntityType.LIGHTNING_BOLT.create((net.minecraft.world.level.Level)$$0);
+        if ($$2 == null) {
+            return;
+        }
         $$2.moveTo(this.horse.getX(), this.horse.getY(), this.horse.getZ());
         $$2.setVisualOnly(true);
         $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$2);
         net.minecraft.world.entity.monster.Skeleton $$3 = this.createSkeleton($$1, (net.minecraft.world.entity.animal.horse.AbstractHorse)this.horse);
+        if ($$3 == null) {
+            return;
+        }
         $$3.startRiding((net.minecraft.world.entity.Entity)this.horse);
         $$0.addFreshEntityWithPassengers((net.minecraft.world.entity.Entity)$$3);
         for (int $$4 = 0; $$4 < 3; ++$$4) {
+            net.minecraft.world.entity.monster.Skeleton $$6;
             net.minecraft.world.entity.animal.horse.AbstractHorse $$5 = this.createHorse($$1);
-            net.minecraft.world.entity.monster.Skeleton $$6 = this.createSkeleton($$1, $$5);
+            if ($$5 == null || ($$6 = this.createSkeleton($$1, $$5)) == null) continue;
             $$6.startRiding((net.minecraft.world.entity.Entity)$$5);
             $$5.push(this.horse.getRandom().triangle(0.0, 1.1485), 0.0, this.horse.getRandom().triangle(0.0, 1.1485));
             $$0.addFreshEntityWithPassengers((net.minecraft.world.entity.Entity)$$5);
         }
     }
 
-    public net.minecraft.world.entity.animal.horse.AbstractHorse createHorse(net.minecraft.world.DifficultyInstance difficulty) {
-        net.minecraft.world.entity.animal.horse.SkeletonHorse skeletonHorse = (net.minecraft.world.entity.animal.horse.SkeletonHorse)net.minecraft.world.entity.EntityType.SKELETON_HORSE.create(this.horse.level);
-        skeletonHorse.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)((net.minecraft.server.level.ServerLevel)this.horse.level), difficulty, net.minecraft.world.entity.MobSpawnType.TRIGGERED, null, null);
-        skeletonHorse.setPos(this.horse.getX(), this.horse.getY(), this.horse.getZ());
-        skeletonHorse.invulnerableTime = 60;
-        skeletonHorse.setPersistenceRequired();
-        skeletonHorse.setTamed(true);
-        skeletonHorse.setAge(0);
-        return skeletonHorse;
+    @javax.annotation.Nullable
+    public net.minecraft.world.entity.animal.horse.AbstractHorse createHorse(net.minecraft.world.DifficultyInstance $$0) {
+        net.minecraft.world.entity.animal.horse.SkeletonHorse $$1 = (net.minecraft.world.entity.animal.horse.SkeletonHorse)net.minecraft.world.entity.EntityType.SKELETON_HORSE.create(this.horse.level);
+        if ($$1 != null) {
+            $$1.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)((net.minecraft.server.level.ServerLevel)this.horse.level), $$0, net.minecraft.world.entity.MobSpawnType.TRIGGERED, null, null);
+            $$1.setPos(this.horse.getX(), this.horse.getY(), this.horse.getZ());
+            $$1.invulnerableTime = 60;
+            $$1.setPersistenceRequired();
+            $$1.setTamed(true);
+            $$1.setAge(0);
+        }
+        return $$1;
     }
 
-    public net.minecraft.world.entity.monster.Skeleton createSkeleton(net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.animal.horse.AbstractHorse horse) {
-        net.minecraft.world.entity.monster.Skeleton skeleton = (net.minecraft.world.entity.monster.Skeleton)net.minecraft.world.entity.EntityType.SKELETON.create(horse.level);
-        skeleton.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)((net.minecraft.server.level.ServerLevel)horse.level), difficulty, net.minecraft.world.entity.MobSpawnType.TRIGGERED, null, null);
-        skeleton.setPos(horse.getX(), horse.getY(), horse.getZ());
-        skeleton.invulnerableTime = 60;
-        skeleton.setPersistenceRequired();
-        if (skeleton.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD).isEmpty()) {
-            skeleton.setItemSlot(net.minecraft.world.entity.EquipmentSlot.HEAD, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_HELMET));
+    @javax.annotation.Nullable
+    public net.minecraft.world.entity.monster.Skeleton createSkeleton(net.minecraft.world.DifficultyInstance $$0, net.minecraft.world.entity.animal.horse.AbstractHorse $$1) {
+        net.minecraft.world.entity.monster.Skeleton $$2 = (net.minecraft.world.entity.monster.Skeleton)net.minecraft.world.entity.EntityType.SKELETON.create($$1.level);
+        if ($$2 != null) {
+            $$2.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)((net.minecraft.server.level.ServerLevel)$$1.level), $$0, net.minecraft.world.entity.MobSpawnType.TRIGGERED, null, null);
+            $$2.setPos($$1.getX(), $$1.getY(), $$1.getZ());
+            $$2.invulnerableTime = 60;
+            $$2.setPersistenceRequired();
+            if ($$2.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD).isEmpty()) {
+                $$2.setItemSlot(net.minecraft.world.entity.EquipmentSlot.HEAD, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.IRON_HELMET));
+            }
+            $$2.setItemSlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND, net.minecraft.world.item.enchantment.EnchantmentHelper.enchantItem((net.minecraft.util.RandomSource)$$2.getRandom(), (net.minecraft.world.item.ItemStack)this.disenchant($$2.getMainHandItem()), (int)((int)(5.0f + $$0.getSpecialMultiplier() * (float)$$2.getRandom().nextInt(18))), (boolean)false));
+            $$2.setItemSlot(net.minecraft.world.entity.EquipmentSlot.HEAD, net.minecraft.world.item.enchantment.EnchantmentHelper.enchantItem((net.minecraft.util.RandomSource)$$2.getRandom(), (net.minecraft.world.item.ItemStack)this.disenchant($$2.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD)), (int)((int)(5.0f + $$0.getSpecialMultiplier() * (float)$$2.getRandom().nextInt(18))), (boolean)false));
         }
-        skeleton.setItemSlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND, net.minecraft.world.item.enchantment.EnchantmentHelper.enchantItem((net.minecraft.util.RandomSource)skeleton.getRandom(), (net.minecraft.world.item.ItemStack)this.disenchant(skeleton.getMainHandItem()), (int)((int)(5.0f + difficulty.getSpecialMultiplier() * (float)skeleton.getRandom().nextInt(18))), (boolean)false));
-        skeleton.setItemSlot(net.minecraft.world.entity.EquipmentSlot.HEAD, net.minecraft.world.item.enchantment.EnchantmentHelper.enchantItem((net.minecraft.util.RandomSource)skeleton.getRandom(), (net.minecraft.world.item.ItemStack)this.disenchant(skeleton.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD)), (int)((int)(5.0f + difficulty.getSpecialMultiplier() * (float)skeleton.getRandom().nextInt(18))), (boolean)false));
-        return skeleton;
+        return $$2;
     }
 
-    public net.minecraft.world.item.ItemStack disenchant(net.minecraft.world.item.ItemStack stack) {
-        stack.removeTagKey("Enchantments");
-        return stack;
+    public net.minecraft.world.item.ItemStack disenchant(net.minecraft.world.item.ItemStack $$0) {
+        $$0.removeTagKey("Enchantments");
+        return $$0;
     }
 }
 