--- net/minecraft/world/entity/animal/horse/ZombieHorse.java
+++ net/minecraft/world/entity/animal/horse/ZombieHorse.java
@@ -23,17 +23,14 @@
     }
 
     public net.minecraft.sounds.SoundEvent getAmbientSound() {
-        super.getAmbientSound();
         return net.minecraft.sounds.SoundEvents.ZOMBIE_HORSE_AMBIENT;
     }
 
     public net.minecraft.sounds.SoundEvent getDeathSound() {
-        super.getDeathSound();
         return net.minecraft.sounds.SoundEvents.ZOMBIE_HORSE_DEATH;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
-        super.getHurtSound(damageSource);
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.ZOMBIE_HORSE_HURT;
     }
 
@@ -42,33 +39,11 @@
         return (net.minecraft.world.entity.AgeableMob)net.minecraft.world.entity.EntityType.ZOMBIE_HORSE.create((net.minecraft.world.level.Level)$$0);
     }
 
-    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
+    public net.minecraft.world.InteractionResult mobInteract(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$1) {
         if (!this.isTamed()) {
             return net.minecraft.world.InteractionResult.PASS;
         }
-        if (this.isBaby()) {
-            return super.mobInteract(player, hand);
-        }
-        if (player.isSecondaryUseActive()) {
-            this.openCustomInventoryScreen(player);
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)this.level.isClientSide);
-        }
-        if (this.isVehicle()) {
-            return super.mobInteract(player, hand);
-        }
-        if (!itemStack.isEmpty()) {
-            if (itemStack.is(net.minecraft.world.item.Items.SADDLE) && !this.isSaddled()) {
-                this.openCustomInventoryScreen(player);
-                return net.minecraft.world.InteractionResult.sidedSuccess((boolean)this.level.isClientSide);
-            }
-            net.minecraft.world.InteractionResult interactionResult = itemStack.interactLivingEntity(player, (net.minecraft.world.entity.LivingEntity)this, hand);
-            if (interactionResult.consumesAction()) {
-                return interactionResult;
-            }
-        }
-        this.doPlayerRide(player);
-        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)this.level.isClientSide);
+        return super.mobInteract($$0, $$1);
     }
 
     public void addBehaviourGoals() {