--- net/minecraft/world/level/block/state/predicate/BlockMaterialPredicate.java
+++ net/minecraft/world/level/block/state/predicate/BlockMaterialPredicate.java
@@ -9,12 +9,12 @@
     public static net.minecraft.world.level.block.state.predicate.BlockMaterialPredicate AIR;
     public net.minecraft.world.level.material.Material material;
 
-    public net.minecraft.world.level.block.state.predicate.BlockMaterialPredicate(net.minecraft.world.level.material.Material material) {
-        this.material = material;
+    public net.minecraft.world.level.block.state.predicate.BlockMaterialPredicate(net.minecraft.world.level.material.Material $$0) {
+        this.material = $$0;
     }
 
-    public static net.minecraft.world.level.block.state.predicate.BlockMaterialPredicate forMaterial(net.minecraft.world.level.material.Material material) {
-        return material == net.minecraft.world.level.material.Material.AIR ? AIR : new net.minecraft.world.level.block.state.predicate.BlockMaterialPredicate(material);
+    public static net.minecraft.world.level.block.state.predicate.BlockMaterialPredicate forMaterial(net.minecraft.world.level.material.Material $$0) {
+        return $$0 == net.minecraft.world.level.material.Material.AIR ? AIR : new net.minecraft.world.level.block.state.predicate.BlockMaterialPredicate($$0);
     }
 
     public boolean test(@javax.annotation.Nullable net.minecraft.world.level.block.state.BlockState $$0) {