--- net/minecraft/world/level/levelgen/Aquifer$NoiseBasedAquifer.java
+++ net/minecraft/world/level/levelgen/Aquifer$NoiseBasedAquifer.java
@@ -298,7 +298,7 @@
         double $$16;
         double $$15;
         net.minecraft.world.level.levelgen.DensityFunction$SinglePointContext $$6 = new net.minecraft.world.level.levelgen.DensityFunction$SinglePointContext($$0, $$1, $$2);
-        if (net.minecraft.world.level.biome.OverworldBiomeBuilder.isDeepDarkRegion((double)this.erosion.compute((net.minecraft.world.level.levelgen.DensityFunction$FunctionContext)$$6), (double)this.depth.compute((net.minecraft.world.level.levelgen.DensityFunction$FunctionContext)$$6))) {
+        if (net.minecraft.world.level.biome.OverworldBiomeBuilder.isDeepDarkRegion((net.minecraft.world.level.levelgen.DensityFunction)this.erosion, (net.minecraft.world.level.levelgen.DensityFunction)this.depth, (net.minecraft.world.level.levelgen.DensityFunction$FunctionContext)$$6)) {
             double $$7 = -1.0;
             double $$8 = -1.0;
         } else {
@@ -353,7 +353,7 @@
 
     static {
         FLOWING_UPDATE_SIMULARITY = net.minecraft.world.level.levelgen.Aquifer$NoiseBasedAquifer.similarity((int)net.minecraft.util.Mth.square((int)10), (int)net.minecraft.util.Mth.square((int)12));
-        SURFACE_SAMPLING_OFFSETS_IN_CHUNKS = new int[][]{{-2, -1}, {-1, -1}, {0, -1}, {1, -1}, {-3, 0}, {-2, 0}, {-1, 0}, {0, 0}, {1, 0}, {-2, 1}, {-1, 1}, {0, 1}, {1, 1}};
+        SURFACE_SAMPLING_OFFSETS_IN_CHUNKS = new int[][]{{0, 0}, {-2, -1}, {-1, -1}, {0, -1}, {1, -1}, {-3, 0}, {-2, 0}, {-1, 0}, {1, 0}, {-2, 1}, {-1, 1}, {0, 1}, {1, 1}};
     }
 }
 