--- net/minecraft/world/level/levelgen/DensityFunctions$Spline$Coordinate.java
+++ net/minecraft/world/level/levelgen/DensityFunctions$Spline$Coordinate.java
@@ -39,11 +39,11 @@
     }
 
     public float minValue() {
-        return (float)((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).minValue();
+        return this.function.isBound() ? (float)((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).minValue() : java.lang.Float.NEGATIVE_INFINITY;
     }
 
     public float maxValue() {
-        return (float)((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).maxValue();
+        return this.function.isBound() ? (float)((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).maxValue() : java.lang.Float.POSITIVE_INFINITY;
     }
 
     public net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate mapAll(net.minecraft.world.level.levelgen.DensityFunction$Visitor $$0) {