--- net/minecraft/world/level/levelgen/DensityFunctions$HolderHolder.java
+++ net/minecraft/world/level/levelgen/DensityFunctions$HolderHolder.java
@@ -27,11 +27,11 @@
     }
 
     public double minValue() {
-        return ((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).minValue();
+        return this.function.isBound() ? ((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).minValue() : java.lang.Double.NEGATIVE_INFINITY;
     }
 
     public double maxValue() {
-        return ((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).maxValue();
+        return this.function.isBound() ? ((net.minecraft.world.level.levelgen.DensityFunction)this.function.value()).maxValue() : java.lang.Double.POSITIVE_INFINITY;
     }
 
     public net.minecraft.util.KeyDispatchDataCodec<? extends net.minecraft.world.level.levelgen.DensityFunction> codec() {