--- net/minecraft/world/level/levelgen/Heightmap$Types.java
+++ net/minecraft/world/level/levelgen/Heightmap$Types.java
@@ -27,11 +27,11 @@
         return (net.minecraft.world.level.levelgen.Heightmap$Types)java.lang.Enum.valueOf(net.minecraft.world.level.levelgen.Heightmap$Types.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.levelgen.Heightmap$Types(java.lang.String serializationKey, net.minecraft.world.level.levelgen.Heightmap$Usage usage, java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> isOpaque) {
+    public net.minecraft.world.level.levelgen.Heightmap$Types(java.lang.String $$0, net.minecraft.world.level.levelgen.Heightmap$Usage $$1, java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> $$2) {
         super(string, n);
-        this.serializationKey = serializationKey;
-        this.usage = usage;
-        this.isOpaque = isOpaque;
+        this.serializationKey = $$0;
+        this.usage = $$1;
+        this.isOpaque = $$2;
     }
 
     public java.lang.String getSerializationKey() {