--- net/minecraft/world/level/levelgen/Noises.java
+++ net/minecraft/world/level/levelgen/Noises.java
@@ -74,7 +74,7 @@
     }
 
     public static net.minecraft.world.level.levelgen.synth.NormalNoise instantiate(net.minecraft.core.Registry<net.minecraft.world.level.levelgen.synth.NormalNoise$NoiseParameters> $$0, net.minecraft.world.level.levelgen.PositionalRandomFactory $$1, net.minecraft.resources.ResourceKey<net.minecraft.world.level.levelgen.synth.NormalNoise$NoiseParameters> $$2) {
-        net.minecraft.core.Holder $$3 = $$0.getHolderOrThrow($$2);
+        net.minecraft.core.Holder$Reference $$3 = $$0.getHolderOrThrow($$2);
         return net.minecraft.world.level.levelgen.synth.NormalNoise.create((net.minecraft.util.RandomSource)$$1.fromHashOf(((net.minecraft.resources.ResourceKey)$$3.unwrapKey().orElseThrow()).location()), (net.minecraft.world.level.levelgen.synth.NormalNoise$NoiseParameters)((net.minecraft.world.level.levelgen.synth.NormalNoise$NoiseParameters)$$3.value()));
     }
 