--- net/minecraft/world/level/levelgen/WorldDimensions$1Entry.java
+++ net/minecraft/world/level/levelgen/WorldDimensions$1Entry.java
@@ -1,0 +1,41 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level.levelgen;
+
+public class net.minecraft.world.level.levelgen.WorldDimensions$1Entry
+extends java.lang.Record
+ {
+    public net.minecraft.resources.ResourceKey<net.minecraft.world.level.dimension.LevelStem> key;
+    public net.minecraft.world.level.dimension.LevelStem value;
+
+    public net.minecraft.world.level.levelgen.WorldDimensions$1Entry(net.minecraft.resources.ResourceKey<net.minecraft.world.level.dimension.LevelStem> $$0, net.minecraft.world.level.dimension.LevelStem $$1) {
+        this.key = $$0;
+        this.value = $$1;
+    }
+
+    public com.mojang.serialization.Lifecycle lifecycle() {
+        return net.minecraft.world.level.levelgen.WorldDimensions.checkStability((net.minecraft.resources.ResourceKey)this.key, (net.minecraft.world.level.dimension.LevelStem)this.value);
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.levelgen.WorldDimensions$1Entry.class, "key;value", "key", "value"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.levelgen.WorldDimensions$1Entry.class, "key;value", "key", "value"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.levelgen.WorldDimensions$1Entry.class, "key;value", "key", "value"}, this, $$0);
+    }
+
+    public net.minecraft.resources.ResourceKey<net.minecraft.world.level.dimension.LevelStem> key() {
+        return this.key;
+    }
+
+    public net.minecraft.world.level.dimension.LevelStem value() {
+        return this.value;
+    }
+}
+