--- net/minecraft/world/effect/AbsoptionMobEffect.java
+++ net/minecraft/world/effect/AbsoptionMobEffect.java
@@ -10,14 +10,14 @@
         super($$0, $$1);
     }
 
-    public void removeAttributeModifiers(net.minecraft.world.entity.LivingEntity livingEntity, net.minecraft.world.entity.ai.attributes.AttributeMap attributeMap, int amplifier) {
-        livingEntity.setAbsorptionAmount(livingEntity.getAbsorptionAmount() - (float)(4 * (amplifier + 1)));
-        super.removeAttributeModifiers(livingEntity, attributeMap, amplifier);
+    public void removeAttributeModifiers(net.minecraft.world.entity.LivingEntity $$0, net.minecraft.world.entity.ai.attributes.AttributeMap $$1, int $$2) {
+        $$0.setAbsorptionAmount($$0.getAbsorptionAmount() - (float)(4 * ($$2 + 1)));
+        super.removeAttributeModifiers($$0, $$1, $$2);
     }
 
-    public void addAttributeModifiers(net.minecraft.world.entity.LivingEntity livingEntity, net.minecraft.world.entity.ai.attributes.AttributeMap attributeMap, int amplifier) {
-        livingEntity.setAbsorptionAmount(livingEntity.getAbsorptionAmount() + (float)(4 * (amplifier + 1)));
-        super.addAttributeModifiers(livingEntity, attributeMap, amplifier);
+    public void addAttributeModifiers(net.minecraft.world.entity.LivingEntity $$0, net.minecraft.world.entity.ai.attributes.AttributeMap $$1, int $$2) {
+        $$0.setAbsorptionAmount($$0.getAbsorptionAmount() + (float)(4 * ($$2 + 1)));
+        super.addAttributeModifiers($$0, $$1, $$2);
     }
 }
 