--- net/minecraft/world/effect/HealthBoostMobEffect.java
+++ net/minecraft/world/effect/HealthBoostMobEffect.java
@@ -10,10 +10,10 @@
         super($$0, $$1);
     }
 
-    public void removeAttributeModifiers(net.minecraft.world.entity.LivingEntity livingEntity, net.minecraft.world.entity.ai.attributes.AttributeMap attributeMap, int amplifier) {
-        super.removeAttributeModifiers(livingEntity, attributeMap, amplifier);
-        if (livingEntity.getHealth() > livingEntity.getMaxHealth()) {
-            livingEntity.setHealth(livingEntity.getMaxHealth());
+    public void removeAttributeModifiers(net.minecraft.world.entity.LivingEntity $$0, net.minecraft.world.entity.ai.attributes.AttributeMap $$1, int $$2) {
+        super.removeAttributeModifiers($$0, $$1, $$2);
+        if ($$0.getHealth() > $$0.getMaxHealth()) {
+            $$0.setHealth($$0.getMaxHealth());
         }
     }
 }