--- net/minecraft/world/effect/MobEffectCategory.java
+++ net/minecraft/world/effect/MobEffectCategory.java
@@ -20,9 +20,9 @@
         return (net.minecraft.world.effect.MobEffectCategory)java.lang.Enum.valueOf(net.minecraft.world.effect.MobEffectCategory.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.effect.MobEffectCategory(net.minecraft.ChatFormatting tooltipFormatting) {
+    public net.minecraft.world.effect.MobEffectCategory(net.minecraft.ChatFormatting $$0) {
         super(string, n);
-        this.tooltipFormatting = tooltipFormatting;
+        this.tooltipFormatting = $$0;
     }
 
     public net.minecraft.ChatFormatting getTooltipFormatting() {