--- net/minecraft/world/effect/MobEffects.java
+++ net/minecraft/world/effect/MobEffects.java
@@ -43,8 +43,8 @@
     public net.minecraft.world.effect.MobEffects() {
     }
 
-    public static net.minecraft.world.effect.MobEffect register(int id, java.lang.String key, net.minecraft.world.effect.MobEffect effect) {
-        return (net.minecraft.world.effect.MobEffect)net.minecraft.core.Registry.registerMapping((net.minecraft.core.Registry)net.minecraft.core.Registry.MOB_EFFECT, (int)id, (java.lang.String)key, (java.lang.Object)effect);
+    public static net.minecraft.world.effect.MobEffect register(int $$0, java.lang.String $$1, net.minecraft.world.effect.MobEffect $$2) {
+        return (net.minecraft.world.effect.MobEffect)net.minecraft.core.Registry.registerMapping((net.minecraft.core.Registry)net.minecraft.core.Registry.MOB_EFFECT, (int)$$0, (java.lang.String)$$1, (java.lang.Object)$$2);
     }
 
     public static /* synthetic */ net.minecraft.world.effect.MobEffectInstance$FactorData lambda$static$0() {