--- net/minecraft/server/Bootstrap$1.java
+++ net/minecraft/server/Bootstrap$1.java
@@ -14,9 +14,9 @@
         this.val$missing = set;
     }
 
-    public <T extends net.minecraft.world.level.GameRules$Value<T>> void visit(net.minecraft.world.level.GameRules$Key<T> key, net.minecraft.world.level.GameRules$Type<T> type) {
-        if (!this.val$language.has(key.getDescriptionId())) {
-            this.val$missing.add((java.lang.Object)key.getId());
+    public <T extends net.minecraft.world.level.GameRules$Value<T>> void visit(net.minecraft.world.level.GameRules$Key<T> $$0, net.minecraft.world.level.GameRules$Type<T> $$1) {
+        if (!this.val$language.has($$0.getDescriptionId())) {
+            this.val$missing.add((java.lang.Object)$$0.getId());
         }
     }
 }