--- net/minecraft/server/Bootstrap.java
+++ net/minecraft/server/Bootstrap.java
@@ -30,21 +30,22 @@
         net.minecraft.core.dispenser.DispenseItemBehavior.bootStrap();
         net.minecraft.core.cauldron.CauldronInteraction.bootStrap();
         net.minecraft.core.Registry.freezeBuiltins();
+        net.minecraft.core.Registry.checkRegistry((net.minecraft.core.Registry)net.minecraft.core.Registry.REGISTRY);
         net.minecraft.server.Bootstrap.wrapStreams();
     }
 
-    public static <T> void checkTranslations(java.lang.Iterable<T> objects, java.util.function.Function<T, java.lang.String> objectToKeyFunction, java.util.Set<java.lang.String> translationSet) {
-        net.minecraft.locale.Language language = net.minecraft.locale.Language.getInstance();
-        objects.forEach($$3 -> {
-            java.lang.String $$4 = (java.lang.String)objectToKeyFunction.apply($$3);
-            if (!language.has($$4)) {
-                translationSet.add((java.lang.Object)$$4);
+    public static <T> void checkTranslations(java.lang.Iterable<T> $$0, java.util.function.Function<T, java.lang.String> $$1, java.util.Set<java.lang.String> $$2) {
+        net.minecraft.locale.Language $$32 = net.minecraft.locale.Language.getInstance();
+        $$0.forEach($$3 -> {
+            java.lang.String $$4 = (java.lang.String)$$1.apply($$3);
+            if (!$$32.has($$4)) {
+                $$2.add((java.lang.Object)$$4);
             }
         });
     }
 
-    public static void checkGameruleTranslations(java.util.Set<java.lang.String> translations) {
-        net.minecraft.locale.Language language = net.minecraft.locale.Language.getInstance();
+    public static void checkGameruleTranslations(java.util.Set<java.lang.String> $$0) {
+        net.minecraft.locale.Language $$1 = net.minecraft.locale.Language.getInstance();
         net.minecraft.world.level.GameRules.visitGameRuleTypes((net.minecraft.world.level.GameRules$GameRuleTypeVisitor)new /* Unavailable Anonymous Inner Class!! */);
     }
 
@@ -110,8 +111,8 @@
         }
     }
 
-    public static void realStdoutPrintln(java.lang.String message) {
-        STDOUT.println(message);
+    public static void realStdoutPrintln(java.lang.String $$0) {
+        STDOUT.println($$0);
     }
 
     static {