--- net/minecraft/server/ChainedJsonException.java
+++ net/minecraft/server/ChainedJsonException.java
@@ -22,12 +22,12 @@
         this.message = $$0;
     }
 
-    public void prependJsonKey(java.lang.String key) {
-        ((net.minecraft.server.ChainedJsonException$Entry)this.entries.get(0)).addJsonKey(key);
+    public void prependJsonKey(java.lang.String $$0) {
+        ((net.minecraft.server.ChainedJsonException$Entry)this.entries.get(0)).addJsonKey($$0);
     }
 
-    public void setFilenameAndFlush(java.lang.String filename) {
-        ((net.minecraft.server.ChainedJsonException$Entry)this.entries.get((int)0)).filename = filename;
+    public void setFilenameAndFlush(java.lang.String $$0) {
+        ((net.minecraft.server.ChainedJsonException$Entry)this.entries.get((int)0)).filename = $$0;
         this.entries.add(0, (java.lang.Object)new net.minecraft.server.ChainedJsonException$Entry());
     }
 
@@ -35,15 +35,15 @@
         return "Invalid " + this.entries.get(this.entries.size() - 1) + ": " + this.message;
     }
 
-    public static net.minecraft.server.ChainedJsonException forException(java.lang.Exception exception) {
-        if (exception instanceof net.minecraft.server.ChainedJsonException) {
-            return (net.minecraft.server.ChainedJsonException)exception;
+    public static net.minecraft.server.ChainedJsonException forException(java.lang.Exception $$0) {
+        if ($$0 instanceof net.minecraft.server.ChainedJsonException) {
+            return (net.minecraft.server.ChainedJsonException)$$0;
         }
-        java.lang.String string = exception.getMessage();
-        if (exception instanceof java.io.FileNotFoundException) {
-            string = "File not found";
+        java.lang.String $$1 = $$0.getMessage();
+        if ($$0 instanceof java.io.FileNotFoundException) {
+            $$1 = "File not found";
         }
-        return new net.minecraft.server.ChainedJsonException(string, (java.lang.Throwable)exception);
+        return new net.minecraft.server.ChainedJsonException($$1, (java.lang.Throwable)$$0);
     }
 }
 