--- net/minecraft/server/ChainedJsonException$Entry.java
+++ net/minecraft/server/ChainedJsonException$Entry.java
@@ -13,8 +13,8 @@
         this.jsonKeys = com.google.common.collect.Lists.newArrayList();
     }
 
-    public void addJsonKey(java.lang.String key) {
-        this.jsonKeys.add(0, (java.lang.Object)key);
+    public void addJsonKey(java.lang.String $$0) {
+        this.jsonKeys.add(0, (java.lang.Object)$$0);
     }
 
     @javax.annotation.Nullable