--- net/minecraft/server/DebugLoggedPrintStream.java
+++ net/minecraft/server/DebugLoggedPrintStream.java
@@ -12,10 +12,10 @@
         super($$0, $$1);
     }
 
-    public void logLine(java.lang.String string) {
-        java.lang.StackTraceElement[] stackTraceElementArray = java.lang.Thread.currentThread().getStackTrace();
-        java.lang.StackTraceElement stackTraceElement = stackTraceElementArray[java.lang.Math.min((int)3, (int)stackTraceElementArray.length)];
-        LOGGER.info("[{}]@.({}:{}): {}", new java.lang.Object[]{this.name, stackTraceElement.getFileName(), stackTraceElement.getLineNumber(), string});
+    public void logLine(java.lang.String $$0) {
+        java.lang.StackTraceElement[] $$1 = java.lang.Thread.currentThread().getStackTrace();
+        java.lang.StackTraceElement $$2 = $$1[java.lang.Math.min((int)3, (int)$$1.length)];
+        LOGGER.info("[{}]@.({}:{}): {}", new java.lang.Object[]{this.name, $$2.getFileName(), $$2.getLineNumber(), $$0});
     }
 
     static {