--- net/minecraft/server/Eula.java
+++ net/minecraft/server/Eula.java
@@ -9,8 +9,8 @@
     public java.nio.file.Path file;
     public boolean agreed;
 
-    public net.minecraft.server.Eula(java.nio.file.Path file) {
-        this.file = file;
+    public net.minecraft.server.Eula(java.nio.file.Path $$0) {
+        this.file = $$0;
         this.agreed = net.minecraft.SharedConstants.IS_RUNNING_IN_IDE || this.readFile();
     }
 