--- net/minecraft/server/LoggedPrintStream.java
+++ net/minecraft/server/LoggedPrintStream.java
@@ -22,8 +22,8 @@
         this.logLine(java.lang.String.valueOf((java.lang.Object)$$0));
     }
 
-    public void logLine(@javax.annotation.Nullable java.lang.String string) {
-        LOGGER.info("[{}]: {}", (java.lang.Object)this.name, (java.lang.Object)string);
+    public void logLine(@javax.annotation.Nullable java.lang.String $$0) {
+        LOGGER.info("[{}]: {}", (java.lang.Object)this.name, (java.lang.Object)$$0);
     }
 
     static {