--- net/minecraft/server/RegistryLayer.java
+++ net/minecraft/server/RegistryLayer.java
@@ -1,0 +1,47 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server;
+
+public class net.minecraft.server.RegistryLayer
+extends java.lang.Enum<net.minecraft.server.RegistryLayer>
+ {
+    public static /* enum */ net.minecraft.server.RegistryLayer STATIC;
+    public static /* enum */ net.minecraft.server.RegistryLayer WORLDGEN;
+    public static /* enum */ net.minecraft.server.RegistryLayer DIMENSIONS;
+    public static /* enum */ net.minecraft.server.RegistryLayer RELOADABLE;
+    public static java.util.List<net.minecraft.server.RegistryLayer> VALUES;
+    public static net.minecraft.core.RegistryAccess$Frozen STATIC_ACCESS;
+    public static /* synthetic */ net.minecraft.server.RegistryLayer[] $VALUES;
+
+    public static net.minecraft.server.RegistryLayer[] values() {
+        return (net.minecraft.server.RegistryLayer[])$VALUES.clone();
+    }
+
+    public static net.minecraft.server.RegistryLayer valueOf(java.lang.String $$0) {
+        return (net.minecraft.server.RegistryLayer)java.lang.Enum.valueOf(net.minecraft.server.RegistryLayer.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.server.RegistryLayer() {
+        super(string, n);
+    }
+
+    public static net.minecraft.core.LayeredRegistryAccess<net.minecraft.server.RegistryLayer> createRegistryAccess() {
+        return new net.minecraft.core.LayeredRegistryAccess(VALUES).replaceFrom((java.lang.Object)STATIC, new net.minecraft.core.RegistryAccess$Frozen[]{STATIC_ACCESS});
+    }
+
+    public static /* synthetic */ net.minecraft.server.RegistryLayer[] $values() {
+        return new net.minecraft.server.RegistryLayer[]{STATIC, WORLDGEN, DIMENSIONS, RELOADABLE};
+    }
+
+    static {
+        STATIC = new net.minecraft.server.RegistryLayer("STATIC", 0);
+        WORLDGEN = new net.minecraft.server.RegistryLayer("WORLDGEN", 1);
+        DIMENSIONS = new net.minecraft.server.RegistryLayer("DIMENSIONS", 2);
+        RELOADABLE = new net.minecraft.server.RegistryLayer("RELOADABLE", 3);
+        $VALUES = net.minecraft.server.RegistryLayer.$values();
+        VALUES = java.util.List.of((java.lang.Object[])net.minecraft.server.RegistryLayer.values());
+        STATIC_ACCESS = net.minecraft.core.RegistryAccess.fromRegistryOfRegistries((net.minecraft.core.Registry)net.minecraft.core.Registry.REGISTRY);
+    }
+}
+