--- net/minecraft/server/ServerAdvancementManager.java
+++ net/minecraft/server/ServerAdvancementManager.java
@@ -17,21 +17,21 @@
         this.predicateManager = $$0;
     }
 
-    public void apply(java.util.Map<net.minecraft.resources.ResourceLocation, com.google.gson.JsonElement> object, net.minecraft.server.packs.resources.ResourceManager resourceManager, net.minecraft.util.profiling.ProfilerFiller profiler) {
-        java.util.HashMap hashMap = com.google.common.collect.Maps.newHashMap();
-        object.forEach((arg_0, arg_1) -> this.lambda$apply$0((java.util.Map)hashMap, arg_0, arg_1));
-        net.minecraft.advancements.AdvancementList advancementList = new net.minecraft.advancements.AdvancementList();
-        advancementList.add((java.util.Map)hashMap);
-        for (net.minecraft.advancements.Advancement advancement : advancementList.getRoots()) {
-            if (advancement.getDisplay() == null) continue;
-            net.minecraft.advancements.TreeNodePosition.run((net.minecraft.advancements.Advancement)advancement);
+    public void apply(java.util.Map<net.minecraft.resources.ResourceLocation, com.google.gson.JsonElement> $$0, net.minecraft.server.packs.resources.ResourceManager $$1, net.minecraft.util.profiling.ProfilerFiller $$2) {
+        java.util.HashMap $$3 = com.google.common.collect.Maps.newHashMap();
+        $$0.forEach((arg_0, arg_1) -> this.lambda$apply$0((java.util.Map)$$3, arg_0, arg_1));
+        net.minecraft.advancements.AdvancementList $$4 = new net.minecraft.advancements.AdvancementList();
+        $$4.add((java.util.Map)$$3);
+        for (net.minecraft.advancements.Advancement $$5 : $$4.getRoots()) {
+            if ($$5.getDisplay() == null) continue;
+            net.minecraft.advancements.TreeNodePosition.run((net.minecraft.advancements.Advancement)$$5);
         }
-        this.advancements = advancementList;
+        this.advancements = $$4;
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.advancements.Advancement getAdvancement(net.minecraft.resources.ResourceLocation id) {
-        return this.advancements.get(id);
+    public net.minecraft.advancements.Advancement getAdvancement(net.minecraft.resources.ResourceLocation $$0) {
+        return this.advancements.get($$0);
     }
 
     public java.util.Collection<net.minecraft.advancements.Advancement> getAllAdvancements() {