--- net/minecraft/server/WorldLoader$DataLoadOutput.java
+++ net/minecraft/server/WorldLoader$DataLoadOutput.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server;
+
+public class net.minecraft.server.WorldLoader$DataLoadOutput<D>
+extends java.lang.Record
+ {
+    public D cookie;
+    public net.minecraft.core.RegistryAccess$Frozen finalDimensions;
+
+    public net.minecraft.server.WorldLoader$DataLoadOutput(D $$0, net.minecraft.core.RegistryAccess$Frozen $$1) {
+        this.cookie = $$0;
+        this.finalDimensions = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.server.WorldLoader$DataLoadOutput.class, "cookie;finalDimensions", "cookie", "finalDimensions"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.server.WorldLoader$DataLoadOutput.class, "cookie;finalDimensions", "cookie", "finalDimensions"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.server.WorldLoader$DataLoadOutput.class, "cookie;finalDimensions", "cookie", "finalDimensions"}, this, $$0);
+    }
+
+    public D cookie() {
+        return (D)this.cookie;
+    }
+
+    public net.minecraft.core.RegistryAccess$Frozen finalDimensions() {
+        return this.finalDimensions;
+    }
+}
+