--- net/minecraft/world/level/levelgen/feature/featuresize/FeatureSizeType.java
+++ net/minecraft/world/level/levelgen/feature/featuresize/FeatureSizeType.java
@@ -9,12 +9,12 @@
     public static net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType<net.minecraft.world.level.levelgen.feature.featuresize.ThreeLayersFeatureSize> THREE_LAYERS_FEATURE_SIZE;
     public com.mojang.serialization.Codec<P> codec;
 
-    public static <P extends net.minecraft.world.level.levelgen.feature.featuresize.FeatureSize> net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType<P> register(java.lang.String name, com.mojang.serialization.Codec<P> codec) {
-        return (net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.FEATURE_SIZE_TYPES, (java.lang.String)name, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType(codec));
+    public static <P extends net.minecraft.world.level.levelgen.feature.featuresize.FeatureSize> net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType<P> register(java.lang.String $$0, com.mojang.serialization.Codec<P> $$1) {
+        return (net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.FEATURE_SIZE_TYPES, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType($$1));
     }
 
-    public net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType(com.mojang.serialization.Codec<P> codec) {
-        this.codec = codec;
+    public net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType(com.mojang.serialization.Codec<P> $$0) {
+        this.codec = $$0;
     }
 
     public com.mojang.serialization.Codec<P> codec() {