--- net/minecraft/world/level/levelgen/feature/featuresize/ThreeLayersFeatureSize.java
+++ net/minecraft/world/level/levelgen/feature/featuresize/ThreeLayersFeatureSize.java
@@ -13,24 +13,24 @@
     public int middleSize;
     public int upperSize;
 
-    public net.minecraft.world.level.levelgen.feature.featuresize.ThreeLayersFeatureSize(int limit, int upperLimit, int lowerSize, int middleSize, int upperSize, java.util.OptionalInt minClippedHeight) {
-        super(minClippedHeight);
-        this.limit = limit;
-        this.upperLimit = upperLimit;
-        this.lowerSize = lowerSize;
-        this.middleSize = middleSize;
-        this.upperSize = upperSize;
+    public net.minecraft.world.level.levelgen.feature.featuresize.ThreeLayersFeatureSize(int $$0, int $$1, int $$2, int $$3, int $$4, java.util.OptionalInt $$5) {
+        super($$5);
+        this.limit = $$0;
+        this.upperLimit = $$1;
+        this.lowerSize = $$2;
+        this.middleSize = $$3;
+        this.upperSize = $$4;
     }
 
     public net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType<?> type() {
         return net.minecraft.world.level.levelgen.feature.featuresize.FeatureSizeType.THREE_LAYERS_FEATURE_SIZE;
     }
 
-    public int getSizeAtHeight(int height, int midpoint) {
-        if (midpoint < this.limit) {
+    public int getSizeAtHeight(int $$0, int $$1) {
+        if ($$1 < this.limit) {
             return this.lowerSize;
         }
-        if (midpoint >= height - this.upperLimit) {
+        if ($$1 >= $$0 - this.upperLimit) {
             return this.upperSize;
         }
         return this.middleSize;