--- net/minecraft/data/loot/packs/UpdateOneTwentyLootTableProvider.java
+++ net/minecraft/data/loot/packs/UpdateOneTwentyLootTableProvider.java
@@ -1,0 +1,15 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data.loot.packs;
+
+public class net.minecraft.data.loot.packs.UpdateOneTwentyLootTableProvider
+ {
+    public net.minecraft.data.loot.packs.UpdateOneTwentyLootTableProvider() {
+    }
+
+    public static net.minecraft.data.loot.LootTableProvider create(net.minecraft.data.PackOutput $$0) {
+        return new net.minecraft.data.loot.LootTableProvider("1.20 Loot Tables", $$0, java.util.Set.of(), java.util.List.of((java.lang.Object)new net.minecraft.data.loot.LootTableProvider$SubProviderEntry(net.minecraft.data.loot.packs.UpdateOneTwentyBlockLoot::new, net.minecraft.world.level.storage.loot.parameters.LootContextParamSets.BLOCK)));
+    }
+}
+