--- net/minecraft/data/loot/packs/UpdateOneTwentyBlockLoot.java
+++ net/minecraft/data/loot/packs/UpdateOneTwentyBlockLoot.java
@@ -1,0 +1,40 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data.loot.packs;
+
+public class net.minecraft.data.loot.packs.UpdateOneTwentyBlockLoot
+extends net.minecraft.data.loot.BlockLootSubProvider
+ {
+    public net.minecraft.data.loot.packs.UpdateOneTwentyBlockLoot() {
+        super(java.util.Set.of(), net.minecraft.world.flag.FeatureFlagSet.of((net.minecraft.world.flag.FeatureFlag)net.minecraft.world.flag.FeatureFlags.UPDATE_1_20));
+    }
+
+    public void generate() {
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_PLANKS);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_STAIRS);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC_STAIRS);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.OAK_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.SPRUCE_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BIRCH_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.ACACIA_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.JUNGLE_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.DARK_OAK_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.MANGROVE_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.CRIMSON_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.WARPED_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_HANGING_SIGN);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_PRESSURE_PLATE);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_FENCE);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_TRAPDOOR);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_FENCE_GATE);
+        this.dropSelf(net.minecraft.world.level.block.Blocks.BAMBOO_BUTTON);
+        this.add(net.minecraft.world.level.block.Blocks.BAMBOO_SLAB, $$1 -> this.createSlabItemTable($$1));
+        this.add(net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC_SLAB, $$1 -> this.createSlabItemTable($$1));
+        this.add(net.minecraft.world.level.block.Blocks.BAMBOO_DOOR, $$1 -> this.createDoorTable($$1));
+        this.dropWhenSilkTouch(net.minecraft.world.level.block.Blocks.CHISELED_BOOKSHELF);
+    }
+}
+