--- net/minecraft/network/protocol/login/ClientboundGameProfilePacket.java
+++ net/minecraft/network/protocol/login/ClientboundGameProfilePacket.java
@@ -8,20 +8,20 @@
  {
     public com.mojang.authlib.GameProfile gameProfile;
 
-    public net.minecraft.network.protocol.login.ClientboundGameProfilePacket(com.mojang.authlib.GameProfile gameProfile) {
-        this.gameProfile = gameProfile;
+    public net.minecraft.network.protocol.login.ClientboundGameProfilePacket(com.mojang.authlib.GameProfile $$0) {
+        this.gameProfile = $$0;
     }
 
     public net.minecraft.network.protocol.login.ClientboundGameProfilePacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.gameProfile = $$0.readGameProfile();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeGameProfile(this.gameProfile);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeGameProfile(this.gameProfile);
     }
 
-    public void handle(net.minecraft.network.protocol.login.ClientLoginPacketListener handler) {
-        handler.handleGameProfile(this);
+    public void handle(net.minecraft.network.protocol.login.ClientLoginPacketListener $$0) {
+        $$0.handleGameProfile(this);
     }
 
     public com.mojang.authlib.GameProfile getGameProfile() {