--- net/minecraft/network/protocol/login/ClientboundLoginCompressionPacket.java
+++ net/minecraft/network/protocol/login/ClientboundLoginCompressionPacket.java
@@ -8,20 +8,20 @@
  {
     public int compressionThreshold;
 
-    public net.minecraft.network.protocol.login.ClientboundLoginCompressionPacket(int compressionThreshold) {
-        this.compressionThreshold = compressionThreshold;
+    public net.minecraft.network.protocol.login.ClientboundLoginCompressionPacket(int $$0) {
+        this.compressionThreshold = $$0;
     }
 
     public net.minecraft.network.protocol.login.ClientboundLoginCompressionPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.compressionThreshold = $$0.readVarInt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.compressionThreshold);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.compressionThreshold);
     }
 
-    public void handle(net.minecraft.network.protocol.login.ClientLoginPacketListener handler) {
-        handler.handleCompression(this);
+    public void handle(net.minecraft.network.protocol.login.ClientLoginPacketListener $$0) {
+        $$0.handleCompression(this);
     }
 
     public int getCompressionThreshold() {