--- net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
+++ net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
@@ -8,20 +8,20 @@
  {
     public net.minecraft.network.chat.Component reason;
 
-    public net.minecraft.network.protocol.login.ClientboundLoginDisconnectPacket(net.minecraft.network.chat.Component reason) {
-        this.reason = reason;
+    public net.minecraft.network.protocol.login.ClientboundLoginDisconnectPacket(net.minecraft.network.chat.Component $$0) {
+        this.reason = $$0;
     }
 
     public net.minecraft.network.protocol.login.ClientboundLoginDisconnectPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.reason = net.minecraft.network.chat.Component$Serializer.fromJsonLenient((java.lang.String)$$0.readUtf(262144));
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeComponent(this.reason);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeComponent(this.reason);
     }
 
-    public void handle(net.minecraft.network.protocol.login.ClientLoginPacketListener handler) {
-        handler.handleDisconnect(this);
+    public void handle(net.minecraft.network.protocol.login.ClientLoginPacketListener $$0) {
+        $$0.handleDisconnect(this);
     }
 
     public net.minecraft.network.chat.Component getReason() {