--- net/minecraft/network/protocol/login/ServerboundCustomQueryPacket.java
+++ net/minecraft/network/protocol/login/ServerboundCustomQueryPacket.java
@@ -11,9 +11,9 @@
     @javax.annotation.Nullable
     public net.minecraft.network.FriendlyByteBuf data;
 
-    public net.minecraft.network.protocol.login.ServerboundCustomQueryPacket(int transactionId, @javax.annotation.Nullable net.minecraft.network.FriendlyByteBuf data) {
-        this.transactionId = transactionId;
-        this.data = data;
+    public net.minecraft.network.protocol.login.ServerboundCustomQueryPacket(int $$0, @javax.annotation.Nullable net.minecraft.network.FriendlyByteBuf $$1) {
+        this.transactionId = $$0;
+        this.data = $$1;
     }
 
     public net.minecraft.network.protocol.login.ServerboundCustomQueryPacket(net.minecraft.network.FriendlyByteBuf $$02) {
@@ -27,13 +27,13 @@
         });
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.transactionId);
-        buffer.writeNullable((java.lang.Object)this.data, ($$0, $$1) -> $$0.writeBytes($$1.slice()));
+    public void write(net.minecraft.network.FriendlyByteBuf $$02) {
+        $$02.writeVarInt(this.transactionId);
+        $$02.writeNullable((java.lang.Object)this.data, ($$0, $$1) -> $$0.writeBytes($$1.slice()));
     }
 
-    public void handle(net.minecraft.network.protocol.login.ServerLoginPacketListener handler) {
-        handler.handleCustomQueryPacket(this);
+    public void handle(net.minecraft.network.protocol.login.ServerLoginPacketListener $$0) {
+        $$0.handleCustomQueryPacket(this);
     }
 
     public int getTransactionId() {