--- net/minecraft/client/gui/screens/achievement/StatsScreen$GeneralStatisticsList$Entry.java
+++ net/minecraft/client/gui/screens/achievement/StatsScreen$GeneralStatisticsList$Entry.java
@@ -20,10 +20,10 @@
         return this.stat.format(this.this$1.this$0.stats.getValue(this.stat));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.achievement.StatsScreen.access$000((net.minecraft.client.gui.screens.achievement.StatsScreen)this.this$1.this$0), (net.minecraft.network.chat.Component)this.statDisplay, (int)(left + 2), (int)(top + 1), (int)(index % 2 == 0 ? 0xFFFFFF : 0x909090));
-        java.lang.String string = this.getValueText();
-        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.achievement.StatsScreen.access$100((net.minecraft.client.gui.screens.achievement.StatsScreen)this.this$1.this$0), (java.lang.String)string, (int)(left + 2 + 213 - net.minecraft.client.gui.screens.achievement.StatsScreen.access$200((net.minecraft.client.gui.screens.achievement.StatsScreen)this.this$1.this$0).width(string)), (int)(top + 1), (int)(index % 2 == 0 ? 0xFFFFFF : 0x909090));
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.achievement.StatsScreen.access$000((net.minecraft.client.gui.screens.achievement.StatsScreen)this.this$1.this$0), (net.minecraft.network.chat.Component)this.statDisplay, (int)($$3 + 2), (int)($$2 + 1), (int)($$1 % 2 == 0 ? 0xFFFFFF : 0x909090));
+        java.lang.String $$10 = this.getValueText();
+        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.achievement.StatsScreen.access$100((net.minecraft.client.gui.screens.achievement.StatsScreen)this.this$1.this$0), (java.lang.String)$$10, (int)($$3 + 2 + 213 - net.minecraft.client.gui.screens.achievement.StatsScreen.access$200((net.minecraft.client.gui.screens.achievement.StatsScreen)this.this$1.this$0).width($$10)), (int)($$2 + 1), (int)($$1 % 2 == 0 ? 0xFFFFFF : 0x909090));
     }
 
     public net.minecraft.network.chat.Component getNarration() {