--- net/minecraft/client/gui/screens/achievement/StatsScreen.java
+++ net/minecraft/client/gui/screens/achievement/StatsScreen.java
@@ -29,11 +29,11 @@
     public static int SORT_DOWN = -1;
     public static int SORT_UP = 1;
 
-    public net.minecraft.client.gui.screens.achievement.StatsScreen(net.minecraft.client.gui.screens.Screen lastScreen, net.minecraft.stats.StatsCounter stats) {
+    public net.minecraft.client.gui.screens.achievement.StatsScreen(net.minecraft.client.gui.screens.Screen $$0, net.minecraft.stats.StatsCounter $$1) {
         super((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"gui.stats"));
         this.isLoading = true;
-        this.lastScreen = lastScreen;
-        this.stats = stats;
+        this.lastScreen = $$0;
+        this.stats = $$1;
     }
 
     public void init() {
@@ -60,19 +60,19 @@
         }
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         if (this.isLoading) {
-            this.renderBackground(matrixStack);
-            net.minecraft.client.gui.screens.achievement.StatsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)PENDING_TEXT, (int)(this.width / 2), (int)(this.height / 2), (int)0xFFFFFF);
+            this.renderBackground($$0);
+            net.minecraft.client.gui.screens.achievement.StatsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)PENDING_TEXT, (int)(this.width / 2), (int)(this.height / 2), (int)0xFFFFFF);
             java.lang.String string = LOADING_SYMBOLS[(int)(net.minecraft.Util.getMillis() / 150L % (long)LOADING_SYMBOLS.length)];
             int n = this.width / 2;
             int n2 = this.height / 2;
             java.util.Objects.requireNonNull((java.lang.Object)this.font);
-            net.minecraft.client.gui.screens.achievement.StatsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (java.lang.String)string, (int)n, (int)(n2 + 9 * 2), (int)0xFFFFFF);
+            net.minecraft.client.gui.screens.achievement.StatsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (java.lang.String)string, (int)n, (int)(n2 + 9 * 2), (int)0xFFFFFF);
         } else {
-            this.getActiveList().render(matrixStack, mouseX, mouseY, partialTicks);
-            net.minecraft.client.gui.screens.achievement.StatsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
-            super.render(matrixStack, mouseX, mouseY, partialTicks);
+            this.getActiveList().render($$0, $$1, $$2, $$3);
+            net.minecraft.client.gui.screens.achievement.StatsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
+            super.render($$0, $$1, $$2, $$3);
         }
     }
 