--- net/minecraft/client/gui/screens/achievement/StatsScreen$ItemStatisticsList.java
+++ net/minecraft/client/gui/screens/achievement/StatsScreen$ItemStatisticsList.java
@@ -51,25 +51,21 @@
         }
     }
 
-    public void renderHeader(com.mojang.blaze3d.vertex.PoseStack matrixStack, int x, int y, com.mojang.blaze3d.vertex.Tesselator tessellator) {
-        int n;
-        int n2;
+    public void renderHeader(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, com.mojang.blaze3d.vertex.Tesselator $$3) {
         if (!this.minecraft.mouseHandler.isLeftPressed()) {
             this.headerPressed = -1;
         }
-        for (n2 = 0;
- n2 < this.iconOffsets.length; ++n2) {
-            this.this$0.blitSlotIcon(matrixStack, x + this.this$0.getColumnX(n2) - 18, y + 1, 0, this.headerPressed == n2 ? 0 : 18);
+        for (int $$4 = 0; $$4 < this.iconOffsets.length; ++$$4) {
+            this.this$0.blitSlotIcon($$0, $$1 + this.this$0.getColumnX($$4) - 18, $$2 + 1, 0, this.headerPressed == $$4 ? 0 : 18);
         }
         if (this.sortColumn != null) {
-            n2 = this.this$0.getColumnX(this.getColumnIndex(this.sortColumn)) - 36;
-            n = this.sortOrder == 1 ? 2 : 1;
-            this.this$0.blitSlotIcon(matrixStack, x + n2, y + 1, 18 * n, 0);
+            int $$5 = this.this$0.getColumnX(this.getColumnIndex(this.sortColumn)) - 36;
+            int $$6 = this.sortOrder == 1 ? 2 : 1;
+            this.this$0.blitSlotIcon($$0, $$1 + $$5, $$2 + 1, 18 * $$6, 0);
         }
-        for (n2 = 0;
- n2 < this.iconOffsets.length; ++n2) {
-            n = this.headerPressed == n2 ? 1 : 0;
-            this.this$0.blitSlotIcon(matrixStack, x + this.this$0.getColumnX(n2) - 18 + n, y + 1 + n, 18 * this.iconOffsets[n2], 18);
+        for (int $$7 = 0; $$7 < this.iconOffsets.length; ++$$7) {
+            int $$8 = this.headerPressed == $$7 ? 1 : 0;
+            this.this$0.blitSlotIcon($$0, $$1 + this.this$0.getColumnX($$7) - 18 + $$8, $$2 + 1 + $$8, 18 * this.iconOffsets[$$7], 18);
         }
     }
 
@@ -81,8 +77,8 @@
         return this.width / 2 + 140;
     }
 
-    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        this.this$0.renderBackground(matrixStack);
+    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        this.this$0.renderBackground($$0);
     }
 
     public void clickedHeader(int $$0, int $$1) {
@@ -115,28 +111,28 @@
         return -1;
     }
 
-    public void renderDecorations(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY) {
-        if (mouseY < this.y0 || mouseY > this.y1) {
+    public void renderDecorations(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
+        if ($$2 < this.y0 || $$2 > this.y1) {
             return;
         }
-        net.minecraft.client.gui.screens.achievement.StatsScreen$ItemStatisticsList$ItemRow itemRow = (net.minecraft.client.gui.screens.achievement.StatsScreen$ItemStatisticsList$ItemRow)this.getHovered();
-        int n = (this.width - this.getRowWidth()) / 2;
-        if (itemRow != null) {
-            if (mouseX < n + 40 || mouseX > n + 40 + 20) {
+        net.minecraft.client.gui.screens.achievement.StatsScreen$ItemStatisticsList$ItemRow $$3 = (net.minecraft.client.gui.screens.achievement.StatsScreen$ItemStatisticsList$ItemRow)this.getHovered();
+        int $$4 = (this.width - this.getRowWidth()) / 2;
+        if ($$3 != null) {
+            if ($$1 < $$4 + 40 || $$1 > $$4 + 40 + 20) {
                 return;
             }
-            net.minecraft.world.item.Item item = itemRow.getItem();
-            this.renderMousehoverTooltip(matrixStack, this.getString(item), mouseX, mouseY);
+            net.minecraft.world.item.Item $$5 = $$3.getItem();
+            this.renderMousehoverTooltip($$0, this.getString($$5), $$1, $$2);
         } else {
-            net.minecraft.network.chat.Component component = null;
-            int n2 = mouseX - n;
-            for (int i = 0; i < this.iconOffsets.length; ++i) {
-                int n3 = this.this$0.getColumnX(i);
-                if (n2 < n3 - 18 || n2 > n3) continue;
-                component = this.getColumn(i).getDisplayName();
+            net.minecraft.network.chat.Component $$6 = null;
+            int $$7 = $$1 - $$4;
+            for (int $$8 = 0; $$8 < this.iconOffsets.length; ++$$8) {
+                int $$9 = this.this$0.getColumnX($$8);
+                if ($$7 < $$9 - 18 || $$7 > $$9) continue;
+                $$6 = this.getColumn($$8).getDisplayName();
                 break;
             }
-            this.renderMousehoverTooltip(matrixStack, component, mouseX, mouseY);
+            this.renderMousehoverTooltip($$0, $$6, $$1, $$2);
         }
     }
 