--- net/minecraft/client/gui/screens/achievement/StatsScreen$GeneralStatisticsList.java
+++ net/minecraft/client/gui/screens/achievement/StatsScreen$GeneralStatisticsList.java
@@ -18,8 +18,8 @@
         }
     }
 
-    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        this.this$0.renderBackground(matrixStack);
+    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        this.this$0.renderBackground($$0);
     }
 }
 