--- net/minecraft/world/entity/schedule/Activity.java
+++ net/minecraft/world/entity/schedule/Activity.java
@@ -34,28 +34,28 @@
     public java.lang.String name;
     public int hashCode;
 
-    public net.minecraft.world.entity.schedule.Activity(java.lang.String name) {
-        this.name = name;
-        this.hashCode = name.hashCode();
+    public net.minecraft.world.entity.schedule.Activity(java.lang.String $$0) {
+        this.name = $$0;
+        this.hashCode = $$0.hashCode();
     }
 
     public java.lang.String getName() {
         return this.name;
     }
 
-    public static net.minecraft.world.entity.schedule.Activity register(java.lang.String key) {
-        return (net.minecraft.world.entity.schedule.Activity)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.ACTIVITY, (java.lang.String)key, (java.lang.Object)new net.minecraft.world.entity.schedule.Activity(key));
+    public static net.minecraft.world.entity.schedule.Activity register(java.lang.String $$0) {
+        return (net.minecraft.world.entity.schedule.Activity)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.ACTIVITY, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.world.entity.schedule.Activity($$0));
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other == null || this.getClass() != other.getClass()) {
+        if ($$0 == null || this.getClass() != $$0.getClass()) {
             return false;
         }
-        net.minecraft.world.entity.schedule.Activity activity = (net.minecraft.world.entity.schedule.Activity)other;
-        return this.name.equals((java.lang.Object)activity.name);
+        net.minecraft.world.entity.schedule.Activity $$1 = (net.minecraft.world.entity.schedule.Activity)$$0;
+        return this.name.equals((java.lang.Object)$$1.name);
     }
 
     public int hashCode() {