--- net/minecraft/world/entity/schedule/Keyframe.java
+++ net/minecraft/world/entity/schedule/Keyframe.java
@@ -8,9 +8,9 @@
     public int timeStamp;
     public float value;
 
-    public net.minecraft.world.entity.schedule.Keyframe(int timeStamp, float value) {
-        this.timeStamp = timeStamp;
-        this.value = value;
+    public net.minecraft.world.entity.schedule.Keyframe(int $$0, float $$1) {
+        this.timeStamp = $$0;
+        this.value = $$1;
     }
 
     public int getTimeStamp() {