--- net/minecraft/world/entity/schedule/ScheduleBuilder.java
+++ net/minecraft/world/entity/schedule/ScheduleBuilder.java
@@ -8,22 +8,22 @@
     public net.minecraft.world.entity.schedule.Schedule schedule;
     public java.util.List<net.minecraft.world.entity.schedule.ScheduleBuilder$ActivityTransition> transitions;
 
-    public net.minecraft.world.entity.schedule.ScheduleBuilder(net.minecraft.world.entity.schedule.Schedule schedule) {
+    public net.minecraft.world.entity.schedule.ScheduleBuilder(net.minecraft.world.entity.schedule.Schedule $$0) {
         this.transitions = com.google.common.collect.Lists.newArrayList();
-        this.schedule = schedule;
+        this.schedule = $$0;
     }
 
-    public net.minecraft.world.entity.schedule.ScheduleBuilder changeActivityAt(int duration, net.minecraft.world.entity.schedule.Activity activity) {
-        this.transitions.add((java.lang.Object)new net.minecraft.world.entity.schedule.ScheduleBuilder$ActivityTransition(duration, activity));
+    public net.minecraft.world.entity.schedule.ScheduleBuilder changeActivityAt(int $$0, net.minecraft.world.entity.schedule.Activity $$1) {
+        this.transitions.add((java.lang.Object)new net.minecraft.world.entity.schedule.ScheduleBuilder$ActivityTransition($$0, $$1));
         return this;
     }
 
     public net.minecraft.world.entity.schedule.Schedule build() {
         ((java.util.Set)this.transitions.stream().map(net.minecraft.world.entity.schedule.ScheduleBuilder$ActivityTransition::getActivity).collect(java.util.stream.Collectors.toSet())).forEach(arg_0 -> ((net.minecraft.world.entity.schedule.Schedule)this.schedule).ensureTimelineExistsFor(arg_0));
-        this.transitions.forEach(transition -> {
-            net.minecraft.world.entity.schedule.Activity activity = transition.getActivity();
-            this.schedule.getAllTimelinesExceptFor(activity).forEach($$1 -> $$1.addKeyframe(transition.getTime(), 0.0f));
-            this.schedule.getTimelineFor(activity).addKeyframe(transition.getTime(), 1.0f);
+        this.transitions.forEach($$0 -> {
+            net.minecraft.world.entity.schedule.Activity $$12 = $$0.getActivity();
+            this.schedule.getAllTimelinesExceptFor($$12).forEach($$1 -> $$1.addKeyframe($$0.getTime(), 0.0f));
+            this.schedule.getTimelineFor($$12).addKeyframe($$0.getTime(), 1.0f);
         });
         return this.schedule;
     }