--- net/minecraft/world/entity/schedule/ScheduleBuilder$ActivityTransition.java
+++ net/minecraft/world/entity/schedule/ScheduleBuilder$ActivityTransition.java
@@ -8,9 +8,9 @@
     public int time;
     public net.minecraft.world.entity.schedule.Activity activity;
 
-    public net.minecraft.world.entity.schedule.ScheduleBuilder$ActivityTransition(int time, net.minecraft.world.entity.schedule.Activity activity) {
-        this.time = time;
-        this.activity = activity;
+    public net.minecraft.world.entity.schedule.ScheduleBuilder$ActivityTransition(int $$0, net.minecraft.world.entity.schedule.Activity $$1) {
+        this.time = $$0;
+        this.activity = $$1;
     }
 
     public int getTime() {