--- net/minecraft/world/entity/schedule/Timeline.java
+++ net/minecraft/world/entity/schedule/Timeline.java
@@ -16,8 +16,8 @@
         return com.google.common.collect.ImmutableList.copyOf((java.util.Collection)this.keyframes);
     }
 
-    public net.minecraft.world.entity.schedule.Timeline addKeyframe(int duration, float active) {
-        this.keyframes.add((java.lang.Object)new net.minecraft.world.entity.schedule.Keyframe(duration, active));
+    public net.minecraft.world.entity.schedule.Timeline addKeyframe(int $$0, float $$1) {
+        this.keyframes.add((java.lang.Object)new net.minecraft.world.entity.schedule.Keyframe($$0, $$1));
         this.sortAndDeduplicateKeyframes();
         return this;
     }
@@ -36,26 +36,26 @@
         this.previousIndex = 0;
     }
 
-    public float getValueAt(int dayTime) {
-        net.minecraft.world.entity.schedule.Keyframe keyframe;
+    public float getValueAt(int $$0) {
+        net.minecraft.world.entity.schedule.Keyframe $$7;
         if (this.keyframes.size() <= 0) {
             return 0.0f;
         }
-        net.minecraft.world.entity.schedule.Keyframe keyframe2 = (net.minecraft.world.entity.schedule.Keyframe)this.keyframes.get(this.previousIndex);
-        net.minecraft.world.entity.schedule.Keyframe keyframe3 = (net.minecraft.world.entity.schedule.Keyframe)this.keyframes.get(this.keyframes.size() - 1);
-        boolean bl = dayTime < keyframe2.getTimeStamp();
-        int n = bl ? 0 : this.previousIndex;
-        float f = bl ? keyframe3.getValue() : keyframe2.getValue();
-        int n2 = n;
-        while (n2 < this.keyframes.size() && (keyframe = (net.minecraft.world.entity.schedule.Keyframe)this.keyframes.get(n2)).getTimeStamp() <= dayTime) {
-            this.previousIndex = n2++;
-            f = keyframe.getValue();
+        net.minecraft.world.entity.schedule.Keyframe $$1 = (net.minecraft.world.entity.schedule.Keyframe)this.keyframes.get(this.previousIndex);
+        net.minecraft.world.entity.schedule.Keyframe $$2 = (net.minecraft.world.entity.schedule.Keyframe)this.keyframes.get(this.keyframes.size() - 1);
+        boolean $$3 = $$0 < $$1.getTimeStamp();
+        int $$4 = $$3 ? 0 : this.previousIndex;
+        float $$5 = $$3 ? $$2.getValue() : $$1.getValue();
+        int $$6 = $$4;
+        while ($$6 < this.keyframes.size() && ($$7 = (net.minecraft.world.entity.schedule.Keyframe)this.keyframes.get($$6)).getTimeStamp() <= $$0) {
+            this.previousIndex = $$6++;
+            $$5 = $$7.getValue();
         }
-        return f;
+        return $$5;
     }
 
-    public static /* synthetic */ void lambda$sortAndDeduplicateKeyframes$0(it.unimi.dsi.fastutil.ints.Int2ObjectSortedMap keyframe, net.minecraft.world.entity.schedule.Keyframe keyframe2) {
-        keyframe.put(keyframe2.getTimeStamp(), (java.lang.Object)keyframe2);
+    public static /* synthetic */ void lambda$sortAndDeduplicateKeyframes$0(it.unimi.dsi.fastutil.ints.Int2ObjectSortedMap $$0, net.minecraft.world.entity.schedule.Keyframe $$1) {
+        $$0.put($$1.getTimeStamp(), (java.lang.Object)$$1);
     }
 }
 