--- net/minecraft/world/level/storage/loot/functions/ApplyBonusCount$UniformBonusCount.java
+++ net/minecraft/world/level/storage/loot/functions/ApplyBonusCount$UniformBonusCount.java
@@ -9,21 +9,21 @@
     public static net.minecraft.resources.ResourceLocation TYPE;
     public int bonusMultiplier;
 
-    public net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$UniformBonusCount(int bonusMultiplier) {
-        this.bonusMultiplier = bonusMultiplier;
+    public net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$UniformBonusCount(int $$0) {
+        this.bonusMultiplier = $$0;
     }
 
     public int calculateNewCount(net.minecraft.util.RandomSource $$0, int $$1, int $$2) {
         return $$1 + $$0.nextInt(this.bonusMultiplier * $$2 + 1);
     }
 
-    public void serializeParams(com.google.gson.JsonObject json, com.google.gson.JsonSerializationContext serializationContext) {
-        json.addProperty("bonusMultiplier", (java.lang.Number)java.lang.Integer.valueOf((int)this.bonusMultiplier));
+    public void serializeParams(com.google.gson.JsonObject $$0, com.google.gson.JsonSerializationContext $$1) {
+        $$0.addProperty("bonusMultiplier", (java.lang.Number)java.lang.Integer.valueOf((int)this.bonusMultiplier));
     }
 
-    public static net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$Formula deserialize(com.google.gson.JsonObject json, com.google.gson.JsonDeserializationContext deserializationContext) {
-        int n = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)json, (java.lang.String)"bonusMultiplier");
-        return new net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$UniformBonusCount(n);
+    public static net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$Formula deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1) {
+        int $$2 = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)$$0, (java.lang.String)"bonusMultiplier");
+        return new net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$UniformBonusCount($$2);
     }
 
     public net.minecraft.resources.ResourceLocation getType() {