--- net/minecraft/world/level/storage/loot/functions/ApplyBonusCount$BinomialWithBonusCount.java
+++ net/minecraft/world/level/storage/loot/functions/ApplyBonusCount$BinomialWithBonusCount.java
@@ -10,9 +10,9 @@
     public int extraRounds;
     public float probability;
 
-    public net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$BinomialWithBonusCount(int extraRounds, float probability) {
-        this.extraRounds = extraRounds;
-        this.probability = probability;
+    public net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$BinomialWithBonusCount(int $$0, float $$1) {
+        this.extraRounds = $$0;
+        this.probability = $$1;
     }
 
     public int calculateNewCount(net.minecraft.util.RandomSource $$0, int $$1, int $$2) {
@@ -23,15 +23,15 @@
         return $$1;
     }
 
-    public void serializeParams(com.google.gson.JsonObject json, com.google.gson.JsonSerializationContext serializationContext) {
-        json.addProperty("extra", (java.lang.Number)java.lang.Integer.valueOf((int)this.extraRounds));
-        json.addProperty("probability", (java.lang.Number)java.lang.Float.valueOf((float)this.probability));
+    public void serializeParams(com.google.gson.JsonObject $$0, com.google.gson.JsonSerializationContext $$1) {
+        $$0.addProperty("extra", (java.lang.Number)java.lang.Integer.valueOf((int)this.extraRounds));
+        $$0.addProperty("probability", (java.lang.Number)java.lang.Float.valueOf((float)this.probability));
     }
 
-    public static net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$Formula deserialize(com.google.gson.JsonObject json, com.google.gson.JsonDeserializationContext deserializationContext) {
-        int n = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)json, (java.lang.String)"extra");
-        float f = net.minecraft.util.GsonHelper.getAsFloat((com.google.gson.JsonObject)json, (java.lang.String)"probability");
-        return new net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$BinomialWithBonusCount(n, f);
+    public static net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$Formula deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1) {
+        int $$2 = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)$$0, (java.lang.String)"extra");
+        float $$3 = net.minecraft.util.GsonHelper.getAsFloat((com.google.gson.JsonObject)$$0, (java.lang.String)"probability");
+        return new net.minecraft.world.level.storage.loot.functions.ApplyBonusCount$BinomialWithBonusCount($$2, $$3);
     }
 
     public net.minecraft.resources.ResourceLocation getType() {