--- net/minecraft/world/level/storage/loot/functions/ApplyExplosionDecay.java
+++ net/minecraft/world/level/storage/loot/functions/ApplyExplosionDecay.java
@@ -14,20 +14,20 @@
         return net.minecraft.world.level.storage.loot.functions.LootItemFunctions.EXPLOSION_DECAY;
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        java.lang.Float f = (java.lang.Float)context.getParamOrNull(net.minecraft.world.level.storage.loot.parameters.LootContextParams.EXPLOSION_RADIUS);
-        if (f != null) {
-            net.minecraft.util.RandomSource randomSource = context.getRandom();
-            float f2 = 1.0f / f.floatValue();
-            int n = stack.getCount();
-            int n2 = 0;
-            for (int i = 0; i < n; ++i) {
-                if (!(randomSource.nextFloat() <= f2)) continue;
-                ++n2;
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        java.lang.Float $$2 = (java.lang.Float)$$1.getParamOrNull(net.minecraft.world.level.storage.loot.parameters.LootContextParams.EXPLOSION_RADIUS);
+        if ($$2 != null) {
+            net.minecraft.util.RandomSource $$3 = $$1.getRandom();
+            float $$4 = 1.0f / $$2.floatValue();
+            int $$5 = $$0.getCount();
+            int $$6 = 0;
+            for (int $$7 = 0; $$7 < $$5; ++$$7) {
+                if (!($$3.nextFloat() <= $$4)) continue;
+                ++$$6;
             }
-            stack.setCount(n2);
+            $$0.setCount($$6);
         }
-        return stack;
+        return $$0;
     }
 
     public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> explosionDecay() {