--- net/minecraft/world/level/storage/loot/functions/CopyNbtFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/CopyNbtFunction$Serializer.java
@@ -9,23 +9,23 @@
     public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.CopyNbtFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        json.add("source", serializationContext.serialize((java.lang.Object)value.source));
-        com.google.gson.JsonArray jsonArray = new com.google.gson.JsonArray();
-        value.operations.stream().map(net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$CopyOperation::toJson).forEach(arg_0 -> ((com.google.gson.JsonArray)jsonArray).add(arg_0));
-        json.add("ops", (com.google.gson.JsonElement)jsonArray);
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.CopyNbtFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        $$0.add("source", $$2.serialize((java.lang.Object)$$1.source));
+        com.google.gson.JsonArray $$3 = new com.google.gson.JsonArray();
+        $$1.operations.stream().map(net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$CopyOperation::toJson).forEach(arg_0 -> ((com.google.gson.JsonArray)$$3).add(arg_0));
+        $$0.add("ops", (com.google.gson.JsonElement)$$3);
     }
 
-    public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        net.minecraft.world.level.storage.loot.providers.nbt.NbtProvider nbtProvider = (net.minecraft.world.level.storage.loot.providers.nbt.NbtProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"source", (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.providers.nbt.NbtProvider.class);
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)object, (java.lang.String)"ops");
-        for (com.google.gson.JsonElement jsonElement : jsonArray) {
-            com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)jsonElement, (java.lang.String)"op");
-            arrayList.add((java.lang.Object)net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$CopyOperation.fromJson((com.google.gson.JsonObject)jsonObject));
+    public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        net.minecraft.world.level.storage.loot.providers.nbt.NbtProvider $$3 = (net.minecraft.world.level.storage.loot.providers.nbt.NbtProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"source", (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.providers.nbt.NbtProvider.class);
+        java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayList();
+        com.google.gson.JsonArray $$5 = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)$$0, (java.lang.String)"ops");
+        for (com.google.gson.JsonElement $$6 : $$5) {
+            com.google.gson.JsonObject $$7 = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$6, (java.lang.String)"op");
+            $$4.add((java.lang.Object)net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$CopyOperation.fromJson((com.google.gson.JsonObject)$$7));
         }
-        return new net.minecraft.world.level.storage.loot.functions.CopyNbtFunction(conditions, nbtProvider, (java.util.List)arrayList);
+        return new net.minecraft.world.level.storage.loot.functions.CopyNbtFunction($$2, $$3, (java.util.List)$$4);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {