--- net/minecraft/world/level/storage/loot/functions/CopyNameFunction.java
+++ net/minecraft/world/level/storage/loot/functions/CopyNameFunction.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource source;
 
-    public net.minecraft.world.level.storage.loot.functions.CopyNameFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource nameSource) {
-        super(conditions);
-        this.source = nameSource;
+    public net.minecraft.world.level.storage.loot.functions.CopyNameFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$0, net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource $$1) {
+        super($$0);
+        this.source = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.functions.LootItemFunctionType getType() {
@@ -21,17 +21,17 @@
         return com.google.common.collect.ImmutableSet.of((java.lang.Object)this.source.param);
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        net.minecraft.world.Nameable nameable;
-        java.lang.Object object = context.getParamOrNull(this.source.param);
-        if (object instanceof net.minecraft.world.Nameable && (nameable = (net.minecraft.world.Nameable)object).hasCustomName()) {
-            stack.setHoverName(nameable.getDisplayName());
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        net.minecraft.world.Nameable $$3;
+        java.lang.Object $$2 = $$1.getParamOrNull(this.source.param);
+        if ($$2 instanceof net.minecraft.world.Nameable && ($$3 = (net.minecraft.world.Nameable)$$2).hasCustomName()) {
+            $$0.setHoverName($$3.getDisplayName());
         }
-        return stack;
+        return $$0;
     }
 
-    public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> copyName(net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource source) {
-        return net.minecraft.world.level.storage.loot.functions.CopyNameFunction.simpleBuilder($$1 -> new net.minecraft.world.level.storage.loot.functions.CopyNameFunction($$1, source));
+    public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> copyName(net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource $$0) {
+        return net.minecraft.world.level.storage.loot.functions.CopyNameFunction.simpleBuilder($$1 -> new net.minecraft.world.level.storage.loot.functions.CopyNameFunction($$1, $$0));
     }
 }
 