--- net/minecraft/world/level/storage/loot/functions/CopyNameFunction$NameSource.java
+++ net/minecraft/world/level/storage/loot/functions/CopyNameFunction$NameSource.java
@@ -22,18 +22,18 @@
         return (net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource)java.lang.Enum.valueOf(net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource(java.lang.String name, net.minecraft.world.level.storage.loot.parameters.LootContextParam<?> param) {
+    public net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource(java.lang.String $$0, net.minecraft.world.level.storage.loot.parameters.LootContextParam<?> $$1) {
         super(string, n);
-        this.name = name;
-        this.param = param;
+        this.name = $$0;
+        this.param = $$1;
     }
 
-    public static net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource getByName(java.lang.String name) {
-        for (net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource nameSource : net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource.values()) {
-            if (!nameSource.name.equals((java.lang.Object)name)) continue;
-            return nameSource;
+    public static net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource getByName(java.lang.String $$0) {
+        for (net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource $$1 : net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource.values()) {
+            if (!$$1.name.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
-        throw new java.lang.IllegalArgumentException("Invalid name source " + name);
+        throw new java.lang.IllegalArgumentException("Invalid name source " + $$0);
     }
 
     public static /* synthetic */ net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource[] $values() {