--- net/minecraft/world/level/storage/loot/functions/CopyNameFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/CopyNameFunction$Serializer.java
@@ -9,14 +9,14 @@
     public net.minecraft.world.level.storage.loot.functions.CopyNameFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.CopyNameFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        json.addProperty("source", value.source.name);
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.CopyNameFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        $$0.addProperty("source", $$1.source.name);
     }
 
-    public net.minecraft.world.level.storage.loot.functions.CopyNameFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource nameSource = net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource.getByName((java.lang.String)net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)object, (java.lang.String)"source"));
-        return new net.minecraft.world.level.storage.loot.functions.CopyNameFunction(conditions, nameSource);
+    public net.minecraft.world.level.storage.loot.functions.CopyNameFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource $$3 = net.minecraft.world.level.storage.loot.functions.CopyNameFunction$NameSource.getByName((java.lang.String)net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"source"));
+        return new net.minecraft.world.level.storage.loot.functions.CopyNameFunction($$2, $$3);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {