--- net/minecraft/world/level/storage/loot/functions/CopyNbtFunction$Builder.java
+++ net/minecraft/world/level/storage/loot/functions/CopyNbtFunction$Builder.java
@@ -14,13 +14,13 @@
         this.source = $$0;
     }
 
-    public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$Builder copy(java.lang.String sourcePath, java.lang.String targetPath, net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$MergeStrategy copyAction) {
-        this.ops.add((java.lang.Object)new net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$CopyOperation(sourcePath, targetPath, copyAction));
+    public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$Builder copy(java.lang.String $$0, java.lang.String $$1, net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$MergeStrategy $$2) {
+        this.ops.add((java.lang.Object)new net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$CopyOperation($$0, $$1, $$2));
         return this;
     }
 
-    public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$Builder copy(java.lang.String sourcePath, java.lang.String targetPath) {
-        return this.copy(sourcePath, targetPath, net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$MergeStrategy.REPLACE);
+    public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$Builder copy(java.lang.String $$0, java.lang.String $$1) {
+        return this.copy($$0, $$1, net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$MergeStrategy.REPLACE);
     }
 
     public net.minecraft.world.level.storage.loot.functions.CopyNbtFunction$Builder getThis() {