--- net/minecraft/world/level/storage/loot/functions/CopyBlockState$Builder.java
+++ net/minecraft/world/level/storage/loot/functions/CopyBlockState$Builder.java
@@ -9,16 +9,16 @@
     public net.minecraft.world.level.block.Block block;
     public java.util.Set<net.minecraft.world.level.block.state.properties.Property<?>> properties;
 
-    public net.minecraft.world.level.storage.loot.functions.CopyBlockState$Builder(net.minecraft.world.level.block.Block block) {
+    public net.minecraft.world.level.storage.loot.functions.CopyBlockState$Builder(net.minecraft.world.level.block.Block $$0) {
         this.properties = com.google.common.collect.Sets.newHashSet();
-        this.block = block;
+        this.block = $$0;
     }
 
-    public net.minecraft.world.level.storage.loot.functions.CopyBlockState$Builder copy(net.minecraft.world.level.block.state.properties.Property<?> property) {
-        if (!this.block.getStateDefinition().getProperties().contains(property)) {
-            throw new java.lang.IllegalStateException("Property " + property + " is not present on block " + this.block);
+    public net.minecraft.world.level.storage.loot.functions.CopyBlockState$Builder copy(net.minecraft.world.level.block.state.properties.Property<?> $$0) {
+        if (!this.block.getStateDefinition().getProperties().contains($$0)) {
+            throw new java.lang.IllegalStateException("Property " + $$0 + " is not present on block " + this.block);
         }
-        this.properties.add(property);
+        this.properties.add($$0);
         return this;
     }
 