--- net/minecraft/world/level/storage/loot/functions/EnchantWithLevelsFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/EnchantWithLevelsFunction$Serializer.java
@@ -9,16 +9,16 @@
     public net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        json.add("levels", serializationContext.serialize((java.lang.Object)value.levels));
-        json.addProperty("treasure", java.lang.Boolean.valueOf((boolean)value.treasure));
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        $$0.add("levels", $$2.serialize((java.lang.Object)$$1.levels));
+        $$0.addProperty("treasure", java.lang.Boolean.valueOf((boolean)$$1.treasure));
     }
 
-    public net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        net.minecraft.world.level.storage.loot.providers.number.NumberProvider numberProvider = (net.minecraft.world.level.storage.loot.providers.number.NumberProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"levels", (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.providers.number.NumberProvider.class);
-        boolean bl = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)object, (java.lang.String)"treasure", (boolean)false);
-        return new net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction(conditions, numberProvider, bl);
+    public net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        net.minecraft.world.level.storage.loot.providers.number.NumberProvider $$3 = (net.minecraft.world.level.storage.loot.providers.number.NumberProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"levels", (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.providers.number.NumberProvider.class);
+        boolean $$4 = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)$$0, (java.lang.String)"treasure", (boolean)false);
+        return new net.minecraft.world.level.storage.loot.functions.EnchantWithLevelsFunction($$2, $$3, $$4);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {