--- net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction$Builder.java
+++ net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction$Builder.java
@@ -16,8 +16,8 @@
         return this;
     }
 
-    public net.minecraft.world.level.storage.loot.functions.EnchantRandomlyFunction$Builder withEnchantment(net.minecraft.world.item.enchantment.Enchantment enchantment) {
-        this.enchantments.add((java.lang.Object)enchantment);
+    public net.minecraft.world.level.storage.loot.functions.EnchantRandomlyFunction$Builder withEnchantment(net.minecraft.world.item.enchantment.Enchantment $$0) {
+        this.enchantments.add((java.lang.Object)$$0);
         return this;
     }
 