--- net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction.java
+++ net/minecraft/world/level/storage/loot/functions/EnchantRandomlyFunction.java
@@ -9,30 +9,30 @@
     public static org.slf4j.Logger LOGGER;
     public java.util.List<net.minecraft.world.item.enchantment.Enchantment> enchantments;
 
-    public net.minecraft.world.level.storage.loot.functions.EnchantRandomlyFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, java.util.Collection<net.minecraft.world.item.enchantment.Enchantment> possibleEnchantments) {
-        super(conditions);
-        this.enchantments = com.google.common.collect.ImmutableList.copyOf(possibleEnchantments);
+    public net.minecraft.world.level.storage.loot.functions.EnchantRandomlyFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$0, java.util.Collection<net.minecraft.world.item.enchantment.Enchantment> $$1) {
+        super($$0);
+        this.enchantments = com.google.common.collect.ImmutableList.copyOf($$1);
     }
 
     public net.minecraft.world.level.storage.loot.functions.LootItemFunctionType getType() {
         return net.minecraft.world.level.storage.loot.functions.LootItemFunctions.ENCHANT_RANDOMLY;
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        net.minecraft.world.item.enchantment.Enchantment enchantment;
-        net.minecraft.util.RandomSource randomSource = context.getRandom();
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        net.minecraft.world.item.enchantment.Enchantment $$6;
+        net.minecraft.util.RandomSource $$22 = $$1.getRandom();
         if (this.enchantments.isEmpty()) {
-            boolean bl = stack.is(net.minecraft.world.item.Items.BOOK);
-            java.util.List list = (java.util.List)net.minecraft.core.Registry.ENCHANTMENT.stream().filter(net.minecraft.world.item.enchantment.Enchantment::isDiscoverable).filter($$2 -> bl || $$2.canEnchant(stack)).collect(java.util.stream.Collectors.toList());
-            if (list.isEmpty()) {
-                LOGGER.warn("Couldn't find a compatible enchantment for {}", (java.lang.Object)stack);
-                return stack;
+            boolean $$3 = $$0.is(net.minecraft.world.item.Items.BOOK);
+            java.util.List $$4 = (java.util.List)net.minecraft.core.Registry.ENCHANTMENT.stream().filter(net.minecraft.world.item.enchantment.Enchantment::isDiscoverable).filter($$2 -> $$3 || $$2.canEnchant($$0)).collect(java.util.stream.Collectors.toList());
+            if ($$4.isEmpty()) {
+                LOGGER.warn("Couldn't find a compatible enchantment for {}", (java.lang.Object)$$0);
+                return $$0;
             }
-            enchantment = (net.minecraft.world.item.enchantment.Enchantment)list.get(randomSource.nextInt(list.size()));
+            net.minecraft.world.item.enchantment.Enchantment $$5 = (net.minecraft.world.item.enchantment.Enchantment)$$4.get($$22.nextInt($$4.size()));
         } else {
-            enchantment = (net.minecraft.world.item.enchantment.Enchantment)this.enchantments.get(randomSource.nextInt(this.enchantments.size()));
+            $$6 = (net.minecraft.world.item.enchantment.Enchantment)this.enchantments.get($$22.nextInt(this.enchantments.size()));
         }
-        return net.minecraft.world.level.storage.loot.functions.EnchantRandomlyFunction.enchantItem((net.minecraft.world.item.ItemStack)stack, (net.minecraft.world.item.enchantment.Enchantment)enchantment, (net.minecraft.util.RandomSource)randomSource);
+        return net.minecraft.world.level.storage.loot.functions.EnchantRandomlyFunction.enchantItem((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.item.enchantment.Enchantment)$$6, (net.minecraft.util.RandomSource)$$22);
     }
 
     public static net.minecraft.world.item.ItemStack enchantItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.enchantment.Enchantment $$1, net.minecraft.util.RandomSource $$2) {