--- net/minecraft/world/level/storage/loot/functions/FillPlayerHead$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/FillPlayerHead$Serializer.java
@@ -9,14 +9,14 @@
     public net.minecraft.world.level.storage.loot.functions.FillPlayerHead$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.FillPlayerHead value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        json.add("entity", serializationContext.serialize((java.lang.Object)value.entityTarget));
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.FillPlayerHead $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        $$0.add("entity", $$2.serialize((java.lang.Object)$$1.entityTarget));
     }
 
-    public net.minecraft.world.level.storage.loot.functions.FillPlayerHead deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget = (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"entity", (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.LootContext$EntityTarget.class);
-        return new net.minecraft.world.level.storage.loot.functions.FillPlayerHead(conditions, entityTarget);
+    public net.minecraft.world.level.storage.loot.functions.FillPlayerHead deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$3 = (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"entity", (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.LootContext$EntityTarget.class);
+        return new net.minecraft.world.level.storage.loot.functions.FillPlayerHead($$2, $$3);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {