--- net/minecraft/world/level/storage/loot/functions/FillPlayerHead.java
+++ net/minecraft/world/level/storage/loot/functions/FillPlayerHead.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget;
 
-    public net.minecraft.world.level.storage.loot.functions.FillPlayerHead(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget) {
-        super(conditions);
-        this.entityTarget = entityTarget;
+    public net.minecraft.world.level.storage.loot.functions.FillPlayerHead(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$0, net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$1) {
+        super($$0);
+        this.entityTarget = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.functions.LootItemFunctionType getType() {
@@ -21,13 +21,13 @@
         return com.google.common.collect.ImmutableSet.of((java.lang.Object)this.entityTarget.getParam());
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        net.minecraft.world.entity.Entity entity;
-        if (stack.is(net.minecraft.world.item.Items.PLAYER_HEAD) && (entity = (net.minecraft.world.entity.Entity)context.getParamOrNull(this.entityTarget.getParam())) instanceof net.minecraft.world.entity.player.Player) {
-            com.mojang.authlib.GameProfile gameProfile = ((net.minecraft.world.entity.player.Player)entity).getGameProfile();
-            stack.getOrCreateTag().put("SkullOwner", (net.minecraft.nbt.Tag)net.minecraft.nbt.NbtUtils.writeGameProfile((net.minecraft.nbt.CompoundTag)new net.minecraft.nbt.CompoundTag(), (com.mojang.authlib.GameProfile)gameProfile));
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        net.minecraft.world.entity.Entity $$2;
+        if ($$0.is(net.minecraft.world.item.Items.PLAYER_HEAD) && ($$2 = (net.minecraft.world.entity.Entity)$$1.getParamOrNull(this.entityTarget.getParam())) instanceof net.minecraft.world.entity.player.Player) {
+            com.mojang.authlib.GameProfile $$3 = ((net.minecraft.world.entity.player.Player)$$2).getGameProfile();
+            $$0.getOrCreateTag().put("SkullOwner", (net.minecraft.nbt.Tag)net.minecraft.nbt.NbtUtils.writeGameProfile((net.minecraft.nbt.CompoundTag)new net.minecraft.nbt.CompoundTag(), (com.mojang.authlib.GameProfile)$$3));
         }
-        return stack;
+        return $$0;
     }
 
     public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> fillPlayerHead(net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$0) {