--- net/minecraft/world/level/storage/loot/functions/LimitCount.java
+++ net/minecraft/world/level/storage/loot/functions/LimitCount.java
@@ -21,10 +21,10 @@
         return this.limiter.getReferencedContextParams();
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        int n = this.limiter.clamp(context, stack.getCount());
-        stack.setCount(n);
-        return stack;
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        int $$2 = this.limiter.clamp($$1, $$0.getCount());
+        $$0.setCount($$2);
+        return $$0;
     }
 
     public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> limitCount(net.minecraft.world.level.storage.loot.IntRange $$0) {