--- net/minecraft/world/level/storage/loot/functions/LootingEnchantFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/LootingEnchantFunction$Serializer.java
@@ -9,17 +9,17 @@
     public net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        json.add("count", serializationContext.serialize((java.lang.Object)value.value));
-        if (value.hasLimit()) {
-            json.add("limit", serializationContext.serialize((java.lang.Object)value.limit));
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        $$0.add("count", $$2.serialize((java.lang.Object)$$1.value));
+        if ($$1.hasLimit()) {
+            $$0.add("limit", $$2.serialize((java.lang.Object)$$1.limit));
         }
     }
 
-    public net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        int n = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)object, (java.lang.String)"limit", (int)0);
-        return new net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction(conditions, (net.minecraft.world.level.storage.loot.providers.number.NumberProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"count", (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.providers.number.NumberProvider.class), n);
+    public net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        int $$3 = net.minecraft.util.GsonHelper.getAsInt((com.google.gson.JsonObject)$$0, (java.lang.String)"limit", (int)0);
+        return new net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction($$2, (net.minecraft.world.level.storage.loot.providers.number.NumberProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"count", (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.providers.number.NumberProvider.class), $$3);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {