--- net/minecraft/world/level/storage/loot/functions/LootingEnchantFunction.java
+++ net/minecraft/world/level/storage/loot/functions/LootingEnchantFunction.java
@@ -28,20 +28,20 @@
         return this.limit > 0;
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        net.minecraft.world.entity.Entity entity = (net.minecraft.world.entity.Entity)context.getParamOrNull(net.minecraft.world.level.storage.loot.parameters.LootContextParams.KILLER_ENTITY);
-        if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-            int n = net.minecraft.world.item.enchantment.EnchantmentHelper.getMobLooting((net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)entity));
-            if (n == 0) {
-                return stack;
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        net.minecraft.world.entity.Entity $$2 = (net.minecraft.world.entity.Entity)$$1.getParamOrNull(net.minecraft.world.level.storage.loot.parameters.LootContextParams.KILLER_ENTITY);
+        if ($$2 instanceof net.minecraft.world.entity.LivingEntity) {
+            int $$3 = net.minecraft.world.item.enchantment.EnchantmentHelper.getMobLooting((net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)$$2));
+            if ($$3 == 0) {
+                return $$0;
             }
-            float f = (float)n * this.value.getFloat(context);
-            stack.grow(java.lang.Math.round((float)f));
-            if (this.hasLimit() && stack.getCount() > this.limit) {
-                stack.setCount(this.limit);
+            float $$4 = (float)$$3 * this.value.getFloat($$1);
+            $$0.grow(java.lang.Math.round((float)$$4));
+            if (this.hasLimit() && $$0.getCount() > this.limit) {
+                $$0.setCount(this.limit);
             }
         }
-        return stack;
+        return $$0;
     }
 
     public static net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction$Builder lootingMultiplier(net.minecraft.world.level.storage.loot.providers.number.NumberProvider $$0) {