--- net/minecraft/world/level/storage/loot/functions/LootItemConditionalFunction$DummyBuilder.java
+++ net/minecraft/world/level/storage/loot/functions/LootItemConditionalFunction$DummyBuilder.java
@@ -8,8 +8,8 @@
  {
     public java.util.function.Function<net.minecraft.world.level.storage.loot.predicates.LootItemCondition[], net.minecraft.world.level.storage.loot.functions.LootItemFunction> constructor;
 
-    public net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$DummyBuilder(java.util.function.Function<net.minecraft.world.level.storage.loot.predicates.LootItemCondition[], net.minecraft.world.level.storage.loot.functions.LootItemFunction> constructor) {
-        this.constructor = constructor;
+    public net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$DummyBuilder(java.util.function.Function<net.minecraft.world.level.storage.loot.predicates.LootItemCondition[], net.minecraft.world.level.storage.loot.functions.LootItemFunction> $$0) {
+        this.constructor = $$0;
     }
 
     public net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$DummyBuilder getThis() {