--- net/minecraft/world/level/storage/loot/functions/LootingEnchantFunction$Builder.java
+++ net/minecraft/world/level/storage/loot/functions/LootingEnchantFunction$Builder.java
@@ -18,8 +18,8 @@
         return this;
     }
 
-    public net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction$Builder setLimit(int limit) {
-        this.limit = limit;
+    public net.minecraft.world.level.storage.loot.functions.LootingEnchantFunction$Builder setLimit(int $$0) {
+        this.limit = $$0;
         return this;
     }
 