--- net/minecraft/world/level/storage/loot/functions/SetAttributesFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/SetAttributesFunction$Serializer.java
@@ -9,25 +9,25 @@
     public net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.SetAttributesFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        com.google.gson.JsonArray jsonArray = new com.google.gson.JsonArray();
-        for (net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier modifier : value.modifiers) {
-            jsonArray.add((com.google.gson.JsonElement)modifier.serialize(serializationContext));
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.SetAttributesFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        com.google.gson.JsonArray $$3 = new com.google.gson.JsonArray();
+        for (net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier $$4 : $$1.modifiers) {
+            $$3.add((com.google.gson.JsonElement)$$4.serialize($$2));
         }
-        json.add("modifiers", (com.google.gson.JsonElement)jsonArray);
+        $$0.add("modifiers", (com.google.gson.JsonElement)$$3);
     }
 
-    public net.minecraft.world.level.storage.loot.functions.SetAttributesFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)object, (java.lang.String)"modifiers");
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayListWithExpectedSize((int)jsonArray.size());
-        for (com.google.gson.JsonElement jsonElement : jsonArray) {
-            arrayList.add((java.lang.Object)net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier.deserialize((com.google.gson.JsonObject)net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)jsonElement, (java.lang.String)"modifier"), (com.google.gson.JsonDeserializationContext)deserializationContext));
+    public net.minecraft.world.level.storage.loot.functions.SetAttributesFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        com.google.gson.JsonArray $$3 = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)$$0, (java.lang.String)"modifiers");
+        java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayListWithExpectedSize((int)$$3.size());
+        for (com.google.gson.JsonElement $$5 : $$3) {
+            $$4.add((java.lang.Object)net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier.deserialize((com.google.gson.JsonObject)net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$5, (java.lang.String)"modifier"), (com.google.gson.JsonDeserializationContext)$$1));
         }
-        if (arrayList.isEmpty()) {
+        if ($$4.isEmpty()) {
             throw new com.google.gson.JsonSyntaxException("Invalid attribute modifiers array; cannot be empty");
         }
-        return new net.minecraft.world.level.storage.loot.functions.SetAttributesFunction(conditions, (java.util.List)arrayList);
+        return new net.minecraft.world.level.storage.loot.functions.SetAttributesFunction($$2, (java.util.List)$$4);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {