--- net/minecraft/world/level/storage/loot/functions/SetAttributesFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetAttributesFunction.java
@@ -8,9 +8,9 @@
  {
     public java.util.List<net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier> modifiers;
 
-    public net.minecraft.world.level.storage.loot.functions.SetAttributesFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, java.util.List<net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier> modifiers) {
-        super(conditions);
-        this.modifiers = com.google.common.collect.ImmutableList.copyOf(modifiers);
+    public net.minecraft.world.level.storage.loot.functions.SetAttributesFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$0, java.util.List<net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier> $$1) {
+        super($$0);
+        this.modifiers = com.google.common.collect.ImmutableList.copyOf($$1);
     }
 
     public net.minecraft.world.level.storage.loot.functions.LootItemFunctionType getType() {
@@ -21,17 +21,17 @@
         return (java.util.Set)this.modifiers.stream().flatMap($$0 -> $$0.amount.getReferencedContextParams().stream()).collect(com.google.common.collect.ImmutableSet.toImmutableSet());
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        net.minecraft.util.RandomSource randomSource = context.getRandom();
-        for (net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier modifier : this.modifiers) {
-            java.util.UUID uUID = modifier.id;
-            if (uUID == null) {
-                uUID = java.util.UUID.randomUUID();
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        net.minecraft.util.RandomSource $$2 = $$1.getRandom();
+        for (net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$Modifier $$3 : this.modifiers) {
+            java.util.UUID $$4 = $$3.id;
+            if ($$4 == null) {
+                $$4 = java.util.UUID.randomUUID();
             }
-            net.minecraft.world.entity.EquipmentSlot equipmentSlot = (net.minecraft.world.entity.EquipmentSlot)net.minecraft.Util.getRandom((java.lang.Object[])modifier.slots, (net.minecraft.util.RandomSource)randomSource);
-            stack.addAttributeModifier(modifier.attribute, new net.minecraft.world.entity.ai.attributes.AttributeModifier(uUID, modifier.name, (double)modifier.amount.getFloat(context), modifier.operation), equipmentSlot);
+            net.minecraft.world.entity.EquipmentSlot $$5 = (net.minecraft.world.entity.EquipmentSlot)net.minecraft.Util.getRandom((java.lang.Object[])$$3.slots, (net.minecraft.util.RandomSource)$$2);
+            $$0.addAttributeModifier($$3.attribute, new net.minecraft.world.entity.ai.attributes.AttributeModifier($$4, $$3.name, (double)$$3.amount.getFloat($$1), $$3.operation), $$5);
         }
-        return stack;
+        return $$0;
     }
 
     public static net.minecraft.world.level.storage.loot.functions.SetAttributesFunction$ModifierBuilder modifier(java.lang.String $$0, net.minecraft.world.entity.ai.attributes.Attribute $$1, net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation $$2, net.minecraft.world.level.storage.loot.providers.number.NumberProvider $$3) {