--- net/minecraft/world/level/storage/loot/functions/SetBannerPatternFunction$Builder.java
+++ net/minecraft/world/level/storage/loot/functions/SetBannerPatternFunction$Builder.java
@@ -23,7 +23,7 @@
     }
 
     public net.minecraft.world.level.storage.loot.functions.SetBannerPatternFunction$Builder addPattern(net.minecraft.resources.ResourceKey<net.minecraft.world.level.block.entity.BannerPattern> $$0, net.minecraft.world.item.DyeColor $$1) {
-        return this.addPattern(net.minecraft.core.Registry.BANNER_PATTERN.getHolderOrThrow($$0), $$1);
+        return this.addPattern((net.minecraft.core.Holder)net.minecraft.core.Registry.BANNER_PATTERN.getHolderOrThrow($$0), $$1);
     }
 
     public net.minecraft.world.level.storage.loot.functions.SetBannerPatternFunction$Builder addPattern(net.minecraft.core.Holder<net.minecraft.world.level.block.entity.BannerPattern> $$0, net.minecraft.world.item.DyeColor $$1) {