--- net/minecraft/world/level/storage/loot/functions/SetContainerContents$Builder.java
+++ net/minecraft/world/level/storage/loot/functions/SetContainerContents$Builder.java
@@ -18,8 +18,8 @@
         return this;
     }
 
-    public net.minecraft.world.level.storage.loot.functions.SetContainerContents$Builder withEntry(net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer$Builder<?> lootEntryBuilder) {
-        this.entries.add((java.lang.Object)lootEntryBuilder.build());
+    public net.minecraft.world.level.storage.loot.functions.SetContainerContents$Builder withEntry(net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer$Builder<?> $$0) {
+        this.entries.add((java.lang.Object)$$0.build());
         return this;
     }
 