--- net/minecraft/world/level/storage/loot/functions/SetContainerContents$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/SetContainerContents$Serializer.java
@@ -9,17 +9,17 @@
     public net.minecraft.world.level.storage.loot.functions.SetContainerContents$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.SetContainerContents value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        json.addProperty("type", net.minecraft.core.Registry.BLOCK_ENTITY_TYPE.getKey((java.lang.Object)value.type).toString());
-        json.add("entries", serializationContext.serialize((java.lang.Object)value.entries));
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.SetContainerContents $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        $$0.addProperty("type", net.minecraft.core.Registry.BLOCK_ENTITY_TYPE.getKey((java.lang.Object)$$1.type).toString());
+        $$0.add("entries", $$2.serialize((java.lang.Object)$$1.entries));
     }
 
-    public net.minecraft.world.level.storage.loot.functions.SetContainerContents deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        java.lang.Object[] objectArray = (net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[])net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"entries", (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[].class);
-        net.minecraft.resources.ResourceLocation resourceLocation = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)object, (java.lang.String)"type"));
-        net.minecraft.world.level.block.entity.BlockEntityType blockEntityType = (net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.core.Registry.BLOCK_ENTITY_TYPE.getOptional(resourceLocation).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown block entity type id '" + resourceLocation + "'"));
-        return new net.minecraft.world.level.storage.loot.functions.SetContainerContents(conditions, blockEntityType, java.util.Arrays.asList((java.lang.Object[])objectArray));
+    public net.minecraft.world.level.storage.loot.functions.SetContainerContents deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        java.lang.Object[] $$3 = (net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[])net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"entries", (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[].class);
+        net.minecraft.resources.ResourceLocation $$4 = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"type"));
+        net.minecraft.world.level.block.entity.BlockEntityType $$5 = (net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.core.Registry.BLOCK_ENTITY_TYPE.getOptional($$4).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown block entity type id '" + $$4 + "'"));
+        return new net.minecraft.world.level.storage.loot.functions.SetContainerContents($$2, $$5, java.util.Arrays.asList((java.lang.Object[])$$3));
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {