--- net/minecraft/world/level/storage/loot/functions/SetContainerContents.java
+++ net/minecraft/world/level/storage/loot/functions/SetContainerContents.java
@@ -19,28 +19,28 @@
         return net.minecraft.world.level.storage.loot.functions.LootItemFunctions.SET_CONTENTS;
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        if (stack.isEmpty()) {
-            return stack;
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        if ($$0.isEmpty()) {
+            return $$0;
         }
-        net.minecraft.core.NonNullList nonNullList = net.minecraft.core.NonNullList.create();
-        this.entries.forEach($$22 -> $$22.expand(context, $$2 -> $$2.createItemStack(net.minecraft.world.level.storage.loot.LootTable.createStackSplitter(arg_0 -> ((net.minecraft.core.NonNullList)nonNullList).add(arg_0)), context)));
-        net.minecraft.nbt.CompoundTag compoundTag = new net.minecraft.nbt.CompoundTag();
-        net.minecraft.world.ContainerHelper.saveAllItems((net.minecraft.nbt.CompoundTag)compoundTag, (net.minecraft.core.NonNullList)nonNullList);
-        net.minecraft.nbt.CompoundTag compoundTag2 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack);
-        if (compoundTag2 == null) {
-            compoundTag2 = compoundTag;
+        net.minecraft.core.NonNullList $$2 = net.minecraft.core.NonNullList.create();
+        this.entries.forEach($$22 -> $$22.expand($$1, $$2 -> $$2.createItemStack(net.minecraft.world.level.storage.loot.LootTable.createStackSplitter((net.minecraft.world.level.storage.loot.LootContext)$$1, arg_0 -> ((net.minecraft.core.NonNullList)$$2).add(arg_0)), $$1)));
+        net.minecraft.nbt.CompoundTag $$3 = new net.minecraft.nbt.CompoundTag();
+        net.minecraft.world.ContainerHelper.saveAllItems((net.minecraft.nbt.CompoundTag)$$3, (net.minecraft.core.NonNullList)$$2);
+        net.minecraft.nbt.CompoundTag $$4 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$0);
+        if ($$4 == null) {
+            $$4 = $$3;
         } else {
-            compoundTag2.merge(compoundTag);
+            $$4.merge($$3);
         }
-        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)stack, (net.minecraft.world.level.block.entity.BlockEntityType)this.type, (net.minecraft.nbt.CompoundTag)compoundTag2);
-        return stack;
+        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.level.block.entity.BlockEntityType)this.type, (net.minecraft.nbt.CompoundTag)$$4);
+        return $$0;
     }
 
-    public void validate(net.minecraft.world.level.storage.loot.ValidationContext context) {
-        super.validate(context);
-        for (int i = 0; i < this.entries.size(); ++i) {
-            ((net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer)this.entries.get(i)).validate(context.forChild(".entry[" + i + "]"));
+    public void validate(net.minecraft.world.level.storage.loot.ValidationContext $$0) {
+        super.validate($$0);
+        for (int $$1 = 0; $$1 < this.entries.size(); ++$$1) {
+            ((net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer)this.entries.get($$1)).validate($$0.forChild(".entry[" + $$1 + "]"));
         }
     }
 