--- net/minecraft/world/level/storage/loot/functions/SetContainerLootTable.java
+++ net/minecraft/world/level/storage/loot/functions/SetContainerLootTable.java
@@ -21,33 +21,33 @@
         return net.minecraft.world.level.storage.loot.functions.LootItemFunctions.SET_LOOT_TABLE;
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        if (stack.isEmpty()) {
-            return stack;
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        if ($$0.isEmpty()) {
+            return $$0;
         }
-        net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack);
-        if (compoundTag == null) {
-            compoundTag = new net.minecraft.nbt.CompoundTag();
+        net.minecraft.nbt.CompoundTag $$2 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$0);
+        if ($$2 == null) {
+            $$2 = new net.minecraft.nbt.CompoundTag();
         }
-        compoundTag.putString("LootTable", this.name.toString());
+        $$2.putString("LootTable", this.name.toString());
         if (this.seed != 0L) {
-            compoundTag.putLong("LootTableSeed", this.seed);
+            $$2.putLong("LootTableSeed", this.seed);
         }
-        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)stack, (net.minecraft.world.level.block.entity.BlockEntityType)this.type, (net.minecraft.nbt.CompoundTag)compoundTag);
-        return stack;
+        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.level.block.entity.BlockEntityType)this.type, (net.minecraft.nbt.CompoundTag)$$2);
+        return $$0;
     }
 
-    public void validate(net.minecraft.world.level.storage.loot.ValidationContext context) {
-        if (context.hasVisitedTable(this.name)) {
-            context.reportProblem("Table " + this.name + " is recursively called");
+    public void validate(net.minecraft.world.level.storage.loot.ValidationContext $$0) {
+        if ($$0.hasVisitedTable(this.name)) {
+            $$0.reportProblem("Table " + this.name + " is recursively called");
             return;
         }
-        super.validate(context);
-        net.minecraft.world.level.storage.loot.LootTable lootTable = context.resolveLootTable(this.name);
-        if (lootTable == null) {
-            context.reportProblem("Unknown loot table called " + this.name);
+        super.validate($$0);
+        net.minecraft.world.level.storage.loot.LootTable $$1 = $$0.resolveLootTable(this.name);
+        if ($$1 == null) {
+            $$0.reportProblem("Unknown loot table called " + this.name);
         } else {
-            lootTable.validate(context.enterTable("->{" + this.name + "}", this.name));
+            $$1.validate($$0.enterTable("->{" + this.name + "}", this.name));
         }
     }
 