--- net/minecraft/world/level/storage/loot/functions/SetLoreFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/SetLoreFunction$Serializer.java
@@ -9,24 +9,24 @@
     public net.minecraft.world.level.storage.loot.functions.SetLoreFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.SetLoreFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        json.addProperty("replace", java.lang.Boolean.valueOf((boolean)value.replace));
-        com.google.gson.JsonArray jsonArray = new com.google.gson.JsonArray();
-        for (net.minecraft.network.chat.Component component : value.lore) {
-            jsonArray.add(net.minecraft.network.chat.Component$Serializer.toJsonTree((net.minecraft.network.chat.Component)component));
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.SetLoreFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        $$0.addProperty("replace", java.lang.Boolean.valueOf((boolean)$$1.replace));
+        com.google.gson.JsonArray $$3 = new com.google.gson.JsonArray();
+        for (net.minecraft.network.chat.Component $$4 : $$1.lore) {
+            $$3.add(net.minecraft.network.chat.Component$Serializer.toJsonTree((net.minecraft.network.chat.Component)$$4));
         }
-        json.add("lore", (com.google.gson.JsonElement)jsonArray);
-        if (value.resolutionContext != null) {
-            json.add("entity", serializationContext.serialize((java.lang.Object)value.resolutionContext));
+        $$0.add("lore", (com.google.gson.JsonElement)$$3);
+        if ($$1.resolutionContext != null) {
+            $$0.add("entity", $$2.serialize((java.lang.Object)$$1.resolutionContext));
         }
     }
 
-    public net.minecraft.world.level.storage.loot.functions.SetLoreFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        boolean bl = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)object, (java.lang.String)"replace", (boolean)false);
-        java.util.List list = (java.util.List)com.google.common.collect.Streams.stream((java.lang.Iterable)net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)object, (java.lang.String)"lore")).map(net.minecraft.network.chat.Component$Serializer::fromJson).collect(com.google.common.collect.ImmutableList.toImmutableList());
-        net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget = (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"entity", null, (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.LootContext$EntityTarget.class);
-        return new net.minecraft.world.level.storage.loot.functions.SetLoreFunction(conditions, bl, list, entityTarget);
+    public net.minecraft.world.level.storage.loot.functions.SetLoreFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        boolean $$3 = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)$$0, (java.lang.String)"replace", (boolean)false);
+        java.util.List $$4 = (java.util.List)com.google.common.collect.Streams.stream((java.lang.Iterable)net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)$$0, (java.lang.String)"lore")).map(net.minecraft.network.chat.Component$Serializer::fromJson).collect(com.google.common.collect.ImmutableList.toImmutableList());
+        net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$5 = (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"entity", null, (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.LootContext$EntityTarget.class);
+        return new net.minecraft.world.level.storage.loot.functions.SetLoreFunction($$2, $$3, $$4, $$5);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {