--- net/minecraft/world/level/storage/loot/functions/SetNameFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetNameFunction.java
@@ -11,10 +11,10 @@
     @javax.annotation.Nullable
     public net.minecraft.world.level.storage.loot.LootContext$EntityTarget resolutionContext;
 
-    public net.minecraft.world.level.storage.loot.functions.SetNameFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, @javax.annotation.Nullable net.minecraft.network.chat.Component name, @javax.annotation.Nullable net.minecraft.world.level.storage.loot.LootContext$EntityTarget resolutionContext) {
-        super(conditions);
-        this.name = name;
-        this.resolutionContext = resolutionContext;
+    public net.minecraft.world.level.storage.loot.functions.SetNameFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$0, @javax.annotation.Nullable net.minecraft.network.chat.Component $$1, @javax.annotation.Nullable net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$2) {
+        super($$0);
+        this.name = $$1;
+        this.resolutionContext = $$2;
     }
 
     public net.minecraft.world.level.storage.loot.functions.LootItemFunctionType getType() {
@@ -25,13 +25,13 @@
         return this.resolutionContext != null ? com.google.common.collect.ImmutableSet.of((java.lang.Object)this.resolutionContext.getParam()) : com.google.common.collect.ImmutableSet.of();
     }
 
-    public static java.util.function.UnaryOperator<net.minecraft.network.chat.Component> createResolver(net.minecraft.world.level.storage.loot.LootContext lootContext, @javax.annotation.Nullable net.minecraft.world.level.storage.loot.LootContext$EntityTarget resolutionContext) {
-        net.minecraft.world.entity.Entity entity;
-        if (resolutionContext != null && (entity = (net.minecraft.world.entity.Entity)lootContext.getParamOrNull(resolutionContext.getParam())) != null) {
-            net.minecraft.commands.CommandSourceStack commandSourceStack = entity.createCommandSourceStack().withPermission(2);
+    public static java.util.function.UnaryOperator<net.minecraft.network.chat.Component> createResolver(net.minecraft.world.level.storage.loot.LootContext $$02, @javax.annotation.Nullable net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$1) {
+        net.minecraft.world.entity.Entity $$22;
+        if ($$1 != null && ($$22 = (net.minecraft.world.entity.Entity)$$02.getParamOrNull($$1.getParam())) != null) {
+            net.minecraft.commands.CommandSourceStack $$3 = $$22.createCommandSourceStack().withPermission(2);
             return $$2 -> {
                 try {
-                    return net.minecraft.network.chat.ComponentUtils.updateForEntity((net.minecraft.commands.CommandSourceStack)commandSourceStack, (net.minecraft.network.chat.Component)$$2, (net.minecraft.world.entity.Entity)entity, (int)0);
+                    return net.minecraft.network.chat.ComponentUtils.updateForEntity((net.minecraft.commands.CommandSourceStack)$$3, (net.minecraft.network.chat.Component)$$2, (net.minecraft.world.entity.Entity)$$22, (int)0);
                 }
                 catch (com.mojang.brigadier.exceptions.CommandSyntaxException $$3) {
                     LOGGER.warn("Failed to resolve text component", (java.lang.Throwable)$$3);
@@ -42,11 +42,11 @@
         return $$0 -> $$0;
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
         if (this.name != null) {
-            stack.setHoverName((net.minecraft.network.chat.Component)net.minecraft.world.level.storage.loot.functions.SetNameFunction.createResolver((net.minecraft.world.level.storage.loot.LootContext)context, (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)this.resolutionContext).apply((java.lang.Object)this.name));
+            $$0.setHoverName((net.minecraft.network.chat.Component)net.minecraft.world.level.storage.loot.functions.SetNameFunction.createResolver((net.minecraft.world.level.storage.loot.LootContext)$$1, (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)this.resolutionContext).apply((java.lang.Object)this.name));
         }
-        return stack;
+        return $$0;
     }
 
     public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> setName(net.minecraft.network.chat.Component $$0) {