--- net/minecraft/world/level/storage/loot/functions/SetNameFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/SetNameFunction$Serializer.java
@@ -9,20 +9,20 @@
     public net.minecraft.world.level.storage.loot.functions.SetNameFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.SetNameFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        if (value.name != null) {
-            json.add("name", net.minecraft.network.chat.Component$Serializer.toJsonTree((net.minecraft.network.chat.Component)value.name));
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.SetNameFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        if ($$1.name != null) {
+            $$0.add("name", net.minecraft.network.chat.Component$Serializer.toJsonTree((net.minecraft.network.chat.Component)$$1.name));
         }
-        if (value.resolutionContext != null) {
-            json.add("entity", serializationContext.serialize((java.lang.Object)value.resolutionContext));
+        if ($$1.resolutionContext != null) {
+            $$0.add("entity", $$2.serialize((java.lang.Object)$$1.resolutionContext));
         }
     }
 
-    public net.minecraft.world.level.storage.loot.functions.SetNameFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)object.get("name"));
-        net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget = (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"entity", null, (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.LootContext$EntityTarget.class);
-        return new net.minecraft.world.level.storage.loot.functions.SetNameFunction(conditions, (net.minecraft.network.chat.Component)mutableComponent, entityTarget);
+    public net.minecraft.world.level.storage.loot.functions.SetNameFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        net.minecraft.network.chat.MutableComponent $$3 = net.minecraft.network.chat.Component$Serializer.fromJson((com.google.gson.JsonElement)$$0.get("name"));
+        net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$4 = (net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"entity", null, (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.LootContext$EntityTarget.class);
+        return new net.minecraft.world.level.storage.loot.functions.SetNameFunction($$2, (net.minecraft.network.chat.Component)$$3, $$4);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {