--- net/minecraft/world/level/storage/loot/functions/SetNbtFunction.java
+++ net/minecraft/world/level/storage/loot/functions/SetNbtFunction.java
@@ -8,23 +8,23 @@
  {
     public net.minecraft.nbt.CompoundTag tag;
 
-    public net.minecraft.world.level.storage.loot.functions.SetNbtFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, net.minecraft.nbt.CompoundTag tag) {
-        super(conditions);
-        this.tag = tag;
+    public net.minecraft.world.level.storage.loot.functions.SetNbtFunction(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$0, net.minecraft.nbt.CompoundTag $$1) {
+        super($$0);
+        this.tag = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.functions.LootItemFunctionType getType() {
         return net.minecraft.world.level.storage.loot.functions.LootItemFunctions.SET_NBT;
     }
 
-    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack stack, net.minecraft.world.level.storage.loot.LootContext context) {
-        stack.getOrCreateTag().merge(this.tag);
-        return stack;
+    public net.minecraft.world.item.ItemStack run(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        $$0.getOrCreateTag().merge(this.tag);
+        return $$0;
     }
 
     @java.lang.Deprecated
-    public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> setTag(net.minecraft.nbt.CompoundTag tag) {
-        return net.minecraft.world.level.storage.loot.functions.SetNbtFunction.simpleBuilder($$1 -> new net.minecraft.world.level.storage.loot.functions.SetNbtFunction($$1, tag));
+    public static net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction$Builder<?> setTag(net.minecraft.nbt.CompoundTag $$0) {
+        return net.minecraft.world.level.storage.loot.functions.SetNbtFunction.simpleBuilder($$1 -> new net.minecraft.world.level.storage.loot.functions.SetNbtFunction($$1, $$0));
     }
 }
 