--- net/minecraft/world/level/storage/loot/functions/SetStewEffectFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/SetStewEffectFunction$Serializer.java
@@ -9,36 +9,36 @@
     public net.minecraft.world.level.storage.loot.functions.SetStewEffectFunction$Serializer() {
     }
 
-    public void serialize(com.google.gson.JsonObject json, net.minecraft.world.level.storage.loot.functions.SetStewEffectFunction value, com.google.gson.JsonSerializationContext serializationContext) {
-        super.serialize(json, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)value, serializationContext);
-        if (!value.effectDurationMap.isEmpty()) {
-            com.google.gson.JsonArray jsonArray = new com.google.gson.JsonArray();
-            for (net.minecraft.world.effect.MobEffect mobEffect : value.effectDurationMap.keySet()) {
-                com.google.gson.JsonObject jsonObject = new com.google.gson.JsonObject();
-                net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.core.Registry.MOB_EFFECT.getKey((java.lang.Object)mobEffect);
-                if (resourceLocation == null) {
-                    throw new java.lang.IllegalArgumentException("Don't know how to serialize mob effect " + mobEffect);
+    public void serialize(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.functions.SetStewEffectFunction $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serialize($$0, (net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction)$$1, $$2);
+        if (!$$1.effectDurationMap.isEmpty()) {
+            com.google.gson.JsonArray $$3 = new com.google.gson.JsonArray();
+            for (net.minecraft.world.effect.MobEffect $$4 : $$1.effectDurationMap.keySet()) {
+                com.google.gson.JsonObject $$5 = new com.google.gson.JsonObject();
+                net.minecraft.resources.ResourceLocation $$6 = net.minecraft.core.Registry.MOB_EFFECT.getKey((java.lang.Object)$$4);
+                if ($$6 == null) {
+                    throw new java.lang.IllegalArgumentException("Don't know how to serialize mob effect " + $$4);
                 }
-                jsonObject.add("type", (com.google.gson.JsonElement)new com.google.gson.JsonPrimitive(resourceLocation.toString()));
-                jsonObject.add("duration", serializationContext.serialize(value.effectDurationMap.get((java.lang.Object)mobEffect)));
-                jsonArray.add((com.google.gson.JsonElement)jsonObject);
+                $$5.add("type", (com.google.gson.JsonElement)new com.google.gson.JsonPrimitive($$6.toString()));
+                $$5.add("duration", $$2.serialize($$1.effectDurationMap.get((java.lang.Object)$$4)));
+                $$3.add((com.google.gson.JsonElement)$$5);
             }
-            json.add("effects", (com.google.gson.JsonElement)jsonArray);
+            $$0.add("effects", (com.google.gson.JsonElement)$$3);
         }
     }
 
-    public net.minecraft.world.level.storage.loot.functions.SetStewEffectFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        java.util.LinkedHashMap linkedHashMap = com.google.common.collect.Maps.newLinkedHashMap();
-        if (object.has("effects")) {
-            com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)object, (java.lang.String)"effects");
-            for (com.google.gson.JsonElement jsonElement : jsonArray) {
-                java.lang.String string = net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)jsonElement.getAsJsonObject(), (java.lang.String)"type");
-                net.minecraft.world.effect.MobEffect mobEffect = (net.minecraft.world.effect.MobEffect)net.minecraft.core.Registry.MOB_EFFECT.getOptional(new net.minecraft.resources.ResourceLocation(string)).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown mob effect '" + string + "'"));
-                net.minecraft.world.level.storage.loot.providers.number.NumberProvider numberProvider = (net.minecraft.world.level.storage.loot.providers.number.NumberProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)jsonElement.getAsJsonObject(), (java.lang.String)"duration", (com.google.gson.JsonDeserializationContext)deserializationContext, net.minecraft.world.level.storage.loot.providers.number.NumberProvider.class);
-                linkedHashMap.put((java.lang.Object)mobEffect, (java.lang.Object)numberProvider);
+    public net.minecraft.world.level.storage.loot.functions.SetStewEffectFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        java.util.LinkedHashMap $$3 = com.google.common.collect.Maps.newLinkedHashMap();
+        if ($$0.has("effects")) {
+            com.google.gson.JsonArray $$4 = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)$$0, (java.lang.String)"effects");
+            for (com.google.gson.JsonElement $$5 : $$4) {
+                java.lang.String $$6 = net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$5.getAsJsonObject(), (java.lang.String)"type");
+                net.minecraft.world.effect.MobEffect $$7 = (net.minecraft.world.effect.MobEffect)net.minecraft.core.Registry.MOB_EFFECT.getOptional(new net.minecraft.resources.ResourceLocation($$6)).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown mob effect '" + $$6 + "'"));
+                net.minecraft.world.level.storage.loot.providers.number.NumberProvider $$8 = (net.minecraft.world.level.storage.loot.providers.number.NumberProvider)net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$5.getAsJsonObject(), (java.lang.String)"duration", (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.providers.number.NumberProvider.class);
+                $$3.put((java.lang.Object)$$7, (java.lang.Object)$$8);
             }
         }
-        return new net.minecraft.world.level.storage.loot.functions.SetStewEffectFunction(conditions, (java.util.Map)linkedHashMap);
+        return new net.minecraft.world.level.storage.loot.functions.SetStewEffectFunction($$2, (java.util.Map)$$3);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {