--- net/minecraft/world/level/storage/loot/functions/SmeltItemFunction$Serializer.java
+++ net/minecraft/world/level/storage/loot/functions/SmeltItemFunction$Serializer.java
@@ -9,8 +9,8 @@
     public net.minecraft.world.level.storage.loot.functions.SmeltItemFunction$Serializer() {
     }
 
-    public net.minecraft.world.level.storage.loot.functions.SmeltItemFunction deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext deserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        return new net.minecraft.world.level.storage.loot.functions.SmeltItemFunction(conditions);
+    public net.minecraft.world.level.storage.loot.functions.SmeltItemFunction deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        return new net.minecraft.world.level.storage.loot.functions.SmeltItemFunction($$2);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {