--- net/minecraft/client/player/AbstractClientPlayer.java
+++ net/minecraft/client/player/AbstractClientPlayer.java
@@ -14,18 +14,18 @@
     public float elytraRotZ;
     public net.minecraft.client.multiplayer.ClientLevel clientLevel;
 
-    public net.minecraft.client.player.AbstractClientPlayer(net.minecraft.client.multiplayer.ClientLevel $$0, com.mojang.authlib.GameProfile $$1, @javax.annotation.Nullable net.minecraft.world.entity.player.ProfilePublicKey $$2) {
-        super((net.minecraft.world.level.Level)$$0, $$0.getSharedSpawnPos(), $$0.getSharedSpawnAngle(), $$1, $$2);
+    public net.minecraft.client.player.AbstractClientPlayer(net.minecraft.client.multiplayer.ClientLevel $$0, com.mojang.authlib.GameProfile $$1) {
+        super((net.minecraft.world.level.Level)$$0, $$0.getSharedSpawnPos(), $$0.getSharedSpawnAngle(), $$1);
         this.clientLevel = $$0;
     }
 
     public boolean isSpectator() {
-        net.minecraft.client.multiplayer.PlayerInfo $$0 = net.minecraft.client.Minecraft.getInstance().getConnection().getPlayerInfo(this.getGameProfile().getId());
+        net.minecraft.client.multiplayer.PlayerInfo $$0 = this.getPlayerInfo();
         return $$0 != null && $$0.getGameMode() == net.minecraft.world.level.GameType.SPECTATOR;
     }
 
     public boolean isCreative() {
-        net.minecraft.client.multiplayer.PlayerInfo $$0 = net.minecraft.client.Minecraft.getInstance().getConnection().getPlayerInfo(this.getGameProfile().getId());
+        net.minecraft.client.multiplayer.PlayerInfo $$0 = this.getPlayerInfo();
         return $$0 != null && $$0.getGameMode() == net.minecraft.world.level.GameType.CREATIVE;
     }
 
@@ -76,8 +76,8 @@
         }
     }
 
-    public static net.minecraft.resources.ResourceLocation getSkinLocation(java.lang.String username) {
-        return new net.minecraft.resources.ResourceLocation("skins/" + com.google.common.hash.Hashing.sha1().hashUnencodedChars((java.lang.CharSequence)net.minecraft.util.StringUtil.stripColor((java.lang.String)username)));
+    public static net.minecraft.resources.ResourceLocation getSkinLocation(java.lang.String $$0) {
+        return new net.minecraft.resources.ResourceLocation("skins/" + com.google.common.hash.Hashing.sha1().hashUnencodedChars((java.lang.CharSequence)net.minecraft.util.StringUtil.stripColor((java.lang.String)$$0)));
     }
 
     public java.lang.String getModelName() {