--- net/minecraft/client/player/KeyboardInput.java
+++ net/minecraft/client/player/KeyboardInput.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.client.Options options;
 
-    public net.minecraft.client.player.KeyboardInput(net.minecraft.client.Options options) {
-        this.options = options;
+    public net.minecraft.client.player.KeyboardInput(net.minecraft.client.Options $$0) {
+        this.options = $$0;
     }
 
     public static float calculateImpulse(boolean $$0, boolean $$1) {