--- net/minecraft/client/player/RemotePlayer.java
+++ net/minecraft/client/player/RemotePlayer.java
@@ -6,21 +6,21 @@
 public class net.minecraft.client.player.RemotePlayer
 extends net.minecraft.client.player.AbstractClientPlayer
  {
-    public net.minecraft.client.player.RemotePlayer(net.minecraft.client.multiplayer.ClientLevel $$0, com.mojang.authlib.GameProfile $$1, @javax.annotation.Nullable net.minecraft.world.entity.player.ProfilePublicKey $$2) {
-        super($$0, $$1, $$2);
+    public net.minecraft.client.player.RemotePlayer(net.minecraft.client.multiplayer.ClientLevel $$0, com.mojang.authlib.GameProfile $$1) {
+        super($$0, $$1);
         this.maxUpStep = 1.0f;
         this.noPhysics = true;
     }
 
-    public boolean shouldRenderAtSqrDistance(double distance) {
-        double d = this.getBoundingBox().getSize() * 10.0;
-        if (java.lang.Double.isNaN((double)d)) {
-            d = 1.0;
+    public boolean shouldRenderAtSqrDistance(double $$0) {
+        double $$1 = this.getBoundingBox().getSize() * 10.0;
+        if (java.lang.Double.isNaN((double)$$1)) {
+            $$1 = 1.0;
         }
-        return distance < (d *= 64.0 * net.minecraft.client.player.RemotePlayer.getViewScale()) * d;
+        return $$0 < ($$1 *= 64.0 * net.minecraft.client.player.RemotePlayer.getViewScale()) * $$1;
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
         return true;
     }
 