--- net/minecraft/world/entity/animal/goat/Goat.java
+++ net/minecraft/world/entity/animal/goat/Goat.java
@@ -92,12 +92,15 @@
         return net.minecraft.sounds.SoundEvents.GOAT_MILK;
     }
 
+    @javax.annotation.Nullable
     public net.minecraft.world.entity.animal.goat.Goat getBreedOffspring(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.AgeableMob $$1) {
         net.minecraft.world.entity.animal.goat.Goat $$2 = (net.minecraft.world.entity.animal.goat.Goat)net.minecraft.world.entity.EntityType.GOAT.create((net.minecraft.world.level.Level)$$0);
         if ($$2 != null) {
+            net.minecraft.world.entity.animal.goat.Goat $$4;
             net.minecraft.world.entity.animal.goat.GoatAi.initMemories((net.minecraft.world.entity.animal.goat.Goat)$$2, (net.minecraft.util.RandomSource)$$0.getRandom());
-            boolean $$3 = $$1 instanceof net.minecraft.world.entity.animal.goat.Goat && ((net.minecraft.world.entity.animal.goat.Goat)$$1).isScreamingGoat();
-            $$2.setScreamingGoat($$3 || $$0.getRandom().nextDouble() < 0.02);
+            net.minecraft.world.entity.animal.goat.Goat $$3 = $$0.getRandom().nextBoolean() ? this : $$1;
+            boolean $$5 = $$3 instanceof net.minecraft.world.entity.animal.goat.Goat && ($$4 = $$3).isScreamingGoat() || $$0.getRandom().nextDouble() < 0.02;
+            $$2.setScreamingGoat($$5);
         }
         return $$2;
     }
@@ -263,6 +266,7 @@
         return $$1.getBlockState($$3.below()).is(net.minecraft.tags.BlockTags.GOATS_SPAWNABLE_ON) && net.minecraft.world.entity.animal.goat.Goat.isBrightEnoughToSpawn((net.minecraft.world.level.BlockAndTintGetter)$$1, (net.minecraft.core.BlockPos)$$3);
     }
 
+    @javax.annotation.Nullable
     public /* synthetic */ net.minecraft.world.entity.AgeableMob getBreedOffspring(net.minecraft.server.level.ServerLevel serverLevel, net.minecraft.world.entity.AgeableMob ageableMob) {
         return this.getBreedOffspring(serverLevel, ageableMob);
     }