--- net/minecraft/data/tags/BiomeTagsProvider.java
+++ net/minecraft/data/tags/BiomeTagsProvider.java
@@ -6,7 +6,7 @@
 public class net.minecraft.data.tags.BiomeTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.level.biome.Biome>
  {
-    public net.minecraft.data.tags.BiomeTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.BiomeTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, net.minecraft.data.BuiltinRegistries.BIOME);
     }
 