--- net/minecraft/data/tags/CatVariantTagsProvider.java
+++ net/minecraft/data/tags/CatVariantTagsProvider.java
@@ -6,7 +6,7 @@
 public class net.minecraft.data.tags.CatVariantTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.entity.animal.CatVariant>
  {
-    public net.minecraft.data.tags.CatVariantTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.CatVariantTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, net.minecraft.core.Registry.CAT_VARIANT);
     }
 