--- net/minecraft/data/tags/EntityTypeTagsProvider.java
+++ net/minecraft/data/tags/EntityTypeTagsProvider.java
@@ -6,8 +6,8 @@
 public class net.minecraft.data.tags.EntityTypeTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.entity.EntityType<?>>
  {
-    public net.minecraft.data.tags.EntityTypeTagsProvider(net.minecraft.data.DataGenerator generator) {
-        super(generator, (net.minecraft.core.Registry)net.minecraft.core.Registry.ENTITY_TYPE);
+    public net.minecraft.data.tags.EntityTypeTagsProvider(net.minecraft.data.PackOutput $$0) {
+        super($$0, (net.minecraft.core.Registry)net.minecraft.core.Registry.ENTITY_TYPE);
     }
 
     public void addTags() {