--- net/minecraft/data/tags/FlatLevelGeneratorPresetTagsProvider.java
+++ net/minecraft/data/tags/FlatLevelGeneratorPresetTagsProvider.java
@@ -6,7 +6,7 @@
 public class net.minecraft.data.tags.FlatLevelGeneratorPresetTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.level.levelgen.flat.FlatLevelGeneratorPreset>
  {
-    public net.minecraft.data.tags.FlatLevelGeneratorPresetTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.FlatLevelGeneratorPresetTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, net.minecraft.data.BuiltinRegistries.FLAT_LEVEL_GENERATOR_PRESET);
     }
 