--- net/minecraft/data/tags/FluidTagsProvider.java
+++ net/minecraft/data/tags/FluidTagsProvider.java
@@ -6,8 +6,8 @@
 public class net.minecraft.data.tags.FluidTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.level.material.Fluid>
  {
-    public net.minecraft.data.tags.FluidTagsProvider(net.minecraft.data.DataGenerator generator) {
-        super(generator, (net.minecraft.core.Registry)net.minecraft.core.Registry.FLUID);
+    public net.minecraft.data.tags.FluidTagsProvider(net.minecraft.data.PackOutput $$0) {
+        super($$0, (net.minecraft.core.Registry)net.minecraft.core.Registry.FLUID);
     }
 
     public void addTags() {