--- net/minecraft/data/tags/GameEventTagsProvider.java
+++ net/minecraft/data/tags/GameEventTagsProvider.java
@@ -9,7 +9,7 @@
     @com.google.common.annotations.VisibleForTesting
     public static net.minecraft.world.level.gameevent.GameEvent[] VIBRATIONS_EXCEPT_FLAP;
 
-    public net.minecraft.data.tags.GameEventTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.GameEventTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, (net.minecraft.core.Registry)net.minecraft.core.Registry.GAME_EVENT);
     }
 