--- net/minecraft/data/tags/InstrumentTagsProvider.java
+++ net/minecraft/data/tags/InstrumentTagsProvider.java
@@ -6,7 +6,7 @@
 public class net.minecraft.data.tags.InstrumentTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.item.Instrument>
  {
-    public net.minecraft.data.tags.InstrumentTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.InstrumentTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, net.minecraft.core.Registry.INSTRUMENT);
     }
 