--- net/minecraft/data/tags/PaintingVariantTagsProvider.java
+++ net/minecraft/data/tags/PaintingVariantTagsProvider.java
@@ -6,7 +6,7 @@
 public class net.minecraft.data.tags.PaintingVariantTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.entity.decoration.PaintingVariant>
  {
-    public net.minecraft.data.tags.PaintingVariantTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.PaintingVariantTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, (net.minecraft.core.Registry)net.minecraft.core.Registry.PAINTING_VARIANT);
     }
 