--- net/minecraft/data/tags/PoiTypeTagsProvider.java
+++ net/minecraft/data/tags/PoiTypeTagsProvider.java
@@ -6,7 +6,7 @@
 public class net.minecraft.data.tags.PoiTypeTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.entity.ai.village.poi.PoiType>
  {
-    public net.minecraft.data.tags.PoiTypeTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.PoiTypeTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, net.minecraft.core.Registry.POINT_OF_INTEREST_TYPE);
     }
 