--- net/minecraft/data/tags/TagsProvider$TagAppender.java
+++ net/minecraft/data/tags/TagsProvider$TagAppender.java
@@ -13,8 +13,8 @@
         this.registry = $$1;
     }
 
-    public net.minecraft.data.tags.TagsProvider$TagAppender<T> add(T item) {
-        this.builder.addElement(this.registry.getKey(item));
+    public net.minecraft.data.tags.TagsProvider$TagAppender<T> add(T $$0) {
+        this.builder.addElement(this.registry.getKey($$0));
         return this;
     }
 
@@ -42,8 +42,8 @@
     }
 
     @java.lang.SafeVarargs
-    public net.minecraft.data.tags.TagsProvider$TagAppender<T> add(T ... toAdd) {
-        java.util.stream.Stream.of((java.lang.Object[])toAdd).map(arg_0 -> ((net.minecraft.core.Registry)this.registry).getKey(arg_0)).forEach($$0 -> this.builder.addElement($$0));
+    public net.minecraft.data.tags.TagsProvider$TagAppender<T> add(T ... $$02) {
+        java.util.stream.Stream.of((java.lang.Object[])$$02).map(arg_0 -> ((net.minecraft.core.Registry)this.registry).getKey(arg_0)).forEach($$0 -> this.builder.addElement($$0));
         return this;
     }
 }