--- net/minecraft/data/tags/UpdateOneTwentyItemTagsProvider.java
+++ net/minecraft/data/tags/UpdateOneTwentyItemTagsProvider.java
@@ -1,0 +1,39 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data.tags;
+
+public class net.minecraft.data.tags.UpdateOneTwentyItemTagsProvider
+extends net.minecraft.data.tags.ItemTagsProvider
+ {
+    public net.minecraft.data.tags.UpdateOneTwentyItemTagsProvider(net.minecraft.data.PackOutput $$0, net.minecraft.data.tags.TagsProvider<net.minecraft.world.level.block.Block> $$1) {
+        super($$0, $$1);
+    }
+
+    public void addTags() {
+        this.copy(net.minecraft.tags.BlockTags.PLANKS, net.minecraft.tags.ItemTags.PLANKS);
+        this.copy(net.minecraft.tags.BlockTags.WOODEN_BUTTONS, net.minecraft.tags.ItemTags.WOODEN_BUTTONS);
+        this.copy(net.minecraft.tags.BlockTags.BUTTONS, net.minecraft.tags.ItemTags.BUTTONS);
+        this.copy(net.minecraft.tags.BlockTags.WOODEN_DOORS, net.minecraft.tags.ItemTags.WOODEN_DOORS);
+        this.copy(net.minecraft.tags.BlockTags.WOODEN_STAIRS, net.minecraft.tags.ItemTags.WOODEN_STAIRS);
+        this.copy(net.minecraft.tags.BlockTags.WOODEN_SLABS, net.minecraft.tags.ItemTags.WOODEN_SLABS);
+        this.copy(net.minecraft.tags.BlockTags.WOODEN_FENCES, net.minecraft.tags.ItemTags.WOODEN_FENCES);
+        this.copy(net.minecraft.tags.BlockTags.WOODEN_PRESSURE_PLATES, net.minecraft.tags.ItemTags.WOODEN_PRESSURE_PLATES);
+        this.copy(net.minecraft.tags.BlockTags.DOORS, net.minecraft.tags.ItemTags.DOORS);
+        this.copy(net.minecraft.tags.BlockTags.SLABS, net.minecraft.tags.ItemTags.SLABS);
+        this.copy(net.minecraft.tags.BlockTags.STAIRS, net.minecraft.tags.ItemTags.STAIRS);
+        this.copy(net.minecraft.tags.BlockTags.WOODEN_TRAPDOORS, net.minecraft.tags.ItemTags.WOODEN_TRAPDOORS);
+        this.copy(net.minecraft.tags.BlockTags.TRAPDOORS, net.minecraft.tags.ItemTags.TRAPDOORS);
+        this.copy(net.minecraft.tags.BlockTags.FENCES, net.minecraft.tags.ItemTags.FENCES);
+        this.copy(net.minecraft.tags.BlockTags.STANDING_SIGNS, net.minecraft.tags.ItemTags.SIGNS);
+        this.copy(net.minecraft.tags.BlockTags.CEILING_HANGING_SIGNS, net.minecraft.tags.ItemTags.HANGING_SIGNS);
+        this.tag(net.minecraft.tags.ItemTags.CHEST_BOATS).add((java.lang.Object)net.minecraft.world.item.Items.BAMBOO_CHEST_RAFT);
+        this.tag(net.minecraft.tags.ItemTags.BOATS).add((java.lang.Object)net.minecraft.world.item.Items.BAMBOO_RAFT);
+        this.tag(net.minecraft.tags.ItemTags.BOOKSHELF_BOOKS).add((java.lang.Object[])new net.minecraft.world.item.Item[]{net.minecraft.world.item.Items.BOOK, net.minecraft.world.item.Items.WRITTEN_BOOK, net.minecraft.world.item.Items.ENCHANTED_BOOK, net.minecraft.world.item.Items.WRITABLE_BOOK});
+    }
+
+    public java.lang.String getName() {
+        return super.getName() + "@1.20";
+    }
+}
+