--- net/minecraft/data/tags/UpdateOneTwentyBlockTagsProvider.java
+++ net/minecraft/data/tags/UpdateOneTwentyBlockTagsProvider.java
@@ -1,0 +1,35 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data.tags;
+
+public class net.minecraft.data.tags.UpdateOneTwentyBlockTagsProvider
+extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.level.block.Block>
+ {
+    public net.minecraft.data.tags.UpdateOneTwentyBlockTagsProvider(net.minecraft.data.PackOutput $$0) {
+        super($$0, (net.minecraft.core.Registry)net.minecraft.core.Registry.BLOCK);
+    }
+
+    public void addTags() {
+        this.tag(net.minecraft.tags.BlockTags.PLANKS).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_PLANKS);
+        this.tag(net.minecraft.tags.BlockTags.WOODEN_BUTTONS).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_BUTTON);
+        this.tag(net.minecraft.tags.BlockTags.WOODEN_DOORS).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_DOOR);
+        this.tag(net.minecraft.tags.BlockTags.WOODEN_STAIRS).add((java.lang.Object[])new net.minecraft.world.level.block.Block[]{net.minecraft.world.level.block.Blocks.BAMBOO_STAIRS, net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC_STAIRS});
+        this.tag(net.minecraft.tags.BlockTags.WOODEN_SLABS).add((java.lang.Object[])new net.minecraft.world.level.block.Block[]{net.minecraft.world.level.block.Blocks.BAMBOO_SLAB, net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC_SLAB});
+        this.tag(net.minecraft.tags.BlockTags.WOODEN_FENCES).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_FENCE);
+        this.tag(net.minecraft.tags.BlockTags.WOODEN_PRESSURE_PLATES).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_PRESSURE_PLATE);
+        this.tag(net.minecraft.tags.BlockTags.WOODEN_TRAPDOORS).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_TRAPDOOR);
+        this.tag(net.minecraft.tags.BlockTags.STANDING_SIGNS).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_SIGN);
+        this.tag(net.minecraft.tags.BlockTags.WALL_SIGNS).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_WALL_SIGN);
+        this.tag(net.minecraft.tags.BlockTags.FENCE_GATES).add((java.lang.Object)net.minecraft.world.level.block.Blocks.BAMBOO_FENCE_GATE);
+        this.tag(net.minecraft.tags.BlockTags.CEILING_HANGING_SIGNS).add((java.lang.Object[])new net.minecraft.world.level.block.Block[]{net.minecraft.world.level.block.Blocks.OAK_HANGING_SIGN, net.minecraft.world.level.block.Blocks.SPRUCE_HANGING_SIGN, net.minecraft.world.level.block.Blocks.BIRCH_HANGING_SIGN, net.minecraft.world.level.block.Blocks.ACACIA_HANGING_SIGN, net.minecraft.world.level.block.Blocks.JUNGLE_HANGING_SIGN, net.minecraft.world.level.block.Blocks.DARK_OAK_HANGING_SIGN, net.minecraft.world.level.block.Blocks.CRIMSON_HANGING_SIGN, net.minecraft.world.level.block.Blocks.WARPED_HANGING_SIGN, net.minecraft.world.level.block.Blocks.MANGROVE_HANGING_SIGN, net.minecraft.world.level.block.Blocks.BAMBOO_HANGING_SIGN});
+        this.tag(net.minecraft.tags.BlockTags.WALL_HANGING_SIGNS).add((java.lang.Object[])new net.minecraft.world.level.block.Block[]{net.minecraft.world.level.block.Blocks.OAK_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.SPRUCE_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.BIRCH_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.ACACIA_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.JUNGLE_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.DARK_OAK_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.CRIMSON_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.WARPED_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.MANGROVE_WALL_HANGING_SIGN, net.minecraft.world.level.block.Blocks.BAMBOO_WALL_HANGING_SIGN});
+        this.tag(net.minecraft.tags.BlockTags.ALL_HANGING_SIGNS).addTag(net.minecraft.tags.BlockTags.CEILING_HANGING_SIGNS).addTag(net.minecraft.tags.BlockTags.WALL_HANGING_SIGNS);
+        this.tag(net.minecraft.tags.BlockTags.MINEABLE_WITH_AXE).addTag(net.minecraft.tags.BlockTags.ALL_HANGING_SIGNS).add((java.lang.Object[])new net.minecraft.world.level.block.Block[]{net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC, net.minecraft.world.level.block.Blocks.CHISELED_BOOKSHELF});
+    }
+
+    public java.lang.String getName() {
+        return super.getName() + "@1.20";
+    }
+}
+