--- net/minecraft/data/tags/WorldPresetTagsProvider.java
+++ net/minecraft/data/tags/WorldPresetTagsProvider.java
@@ -6,7 +6,7 @@
 public class net.minecraft.data.tags.WorldPresetTagsProvider
 extends net.minecraft.data.tags.TagsProvider<net.minecraft.world.level.levelgen.presets.WorldPreset>
  {
-    public net.minecraft.data.tags.WorldPresetTagsProvider(net.minecraft.data.DataGenerator $$0) {
+    public net.minecraft.data.tags.WorldPresetTagsProvider(net.minecraft.data.PackOutput $$0) {
         super($$0, net.minecraft.data.BuiltinRegistries.WORLD_PRESET);
     }
 