--- net/minecraft/world/level/levelgen/feature/trunkplacers/FancyTrunkPlacer.java
+++ net/minecraft/world/level/levelgen/feature/trunkplacers/FancyTrunkPlacer.java
@@ -84,11 +84,11 @@
         return true;
     }
 
-    public int getSteps(net.minecraft.core.BlockPos pos) {
-        int n = net.minecraft.util.Mth.abs((int)pos.getX());
-        int n2 = net.minecraft.util.Mth.abs((int)pos.getY());
-        int n3 = net.minecraft.util.Mth.abs((int)pos.getZ());
-        return java.lang.Math.max((int)n, (int)java.lang.Math.max((int)n2, (int)n3));
+    public int getSteps(net.minecraft.core.BlockPos $$0) {
+        int $$1 = net.minecraft.util.Mth.abs((int)$$0.getX());
+        int $$2 = net.minecraft.util.Mth.abs((int)$$0.getY());
+        int $$3 = net.minecraft.util.Mth.abs((int)$$0.getZ());
+        return java.lang.Math.max((int)$$1, (int)java.lang.Math.max((int)$$2, (int)$$3));
     }
 
     public net.minecraft.core.Direction$Axis getLogAxis(net.minecraft.core.BlockPos $$0, net.minecraft.core.BlockPos $$1) {