--- net/minecraft/world/level/levelgen/feature/trunkplacers/FancyTrunkPlacer$FoliageCoords.java
+++ net/minecraft/world/level/levelgen/feature/trunkplacers/FancyTrunkPlacer$FoliageCoords.java
@@ -8,9 +8,9 @@
     public net.minecraft.world.level.levelgen.feature.foliageplacers.FoliagePlacer$FoliageAttachment attachment;
     public int branchBase;
 
-    public net.minecraft.world.level.levelgen.feature.trunkplacers.FancyTrunkPlacer$FoliageCoords(net.minecraft.core.BlockPos pos, int branchBase) {
-        this.attachment = new net.minecraft.world.level.levelgen.feature.foliageplacers.FoliagePlacer$FoliageAttachment(pos, 0, false);
-        this.branchBase = branchBase;
+    public net.minecraft.world.level.levelgen.feature.trunkplacers.FancyTrunkPlacer$FoliageCoords(net.minecraft.core.BlockPos $$0, int $$1) {
+        this.attachment = new net.minecraft.world.level.levelgen.feature.foliageplacers.FoliagePlacer$FoliageAttachment($$0, 0, false);
+        this.branchBase = $$1;
     }
 
     public int getBranchBase() {