--- net/minecraft/world/level/levelgen/feature/trunkplacers/TrunkPlacerType.java
+++ net/minecraft/world/level/levelgen/feature/trunkplacers/TrunkPlacerType.java
@@ -15,12 +15,12 @@
     public static net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType<net.minecraft.world.level.levelgen.feature.trunkplacers.UpwardsBranchingTrunkPlacer> UPWARDS_BRANCHING_TRUNK_PLACER;
     public com.mojang.serialization.Codec<P> codec;
 
-    public static <P extends net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacer> net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType<P> register(java.lang.String key, com.mojang.serialization.Codec<P> codec) {
-        return (net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.TRUNK_PLACER_TYPES, (java.lang.String)key, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType(codec));
+    public static <P extends net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacer> net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType<P> register(java.lang.String $$0, com.mojang.serialization.Codec<P> $$1) {
+        return (net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.TRUNK_PLACER_TYPES, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType($$1));
     }
 
-    public net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType(com.mojang.serialization.Codec<P> codec) {
-        this.codec = codec;
+    public net.minecraft.world.level.levelgen.feature.trunkplacers.TrunkPlacerType(com.mojang.serialization.Codec<P> $$0) {
+        this.codec = $$0;
     }
 
     public com.mojang.serialization.Codec<P> codec() {