--- net/minecraft/util/ExceptionCollector.java
+++ net/minecraft/util/ExceptionCollector.java
@@ -11,11 +11,11 @@
     public net.minecraft.util.ExceptionCollector() {
     }
 
-    public void add(T exception) {
+    public void add(T $$0) {
         if (this.result == null) {
-            this.result = exception;
+            this.result = $$0;
         } else {
-            this.result.addSuppressed(exception);
+            this.result.addSuppressed($$0);
         }
     }
 