--- net/minecraft/util/FastColor$ARGB32.java
+++ net/minecraft/util/FastColor$ARGB32.java
@@ -8,28 +8,28 @@
     public net.minecraft.util.FastColor$ARGB32() {
     }
 
-    public static int alpha(int packedColor) {
-        return packedColor >>> 24;
+    public static int alpha(int $$0) {
+        return $$0 >>> 24;
     }
 
-    public static int red(int packedColor) {
-        return packedColor >> 16 & 0xFF;
+    public static int red(int $$0) {
+        return $$0 >> 16 & 0xFF;
     }
 
-    public static int green(int packedColor) {
-        return packedColor >> 8 & 0xFF;
+    public static int green(int $$0) {
+        return $$0 >> 8 & 0xFF;
     }
 
-    public static int blue(int packedColor) {
-        return packedColor & 0xFF;
+    public static int blue(int $$0) {
+        return $$0 & 0xFF;
     }
 
-    public static int color(int alpha, int red, int green, int blue) {
-        return alpha << 24 | red << 16 | green << 8 | blue;
+    public static int color(int $$0, int $$1, int $$2, int $$3) {
+        return $$0 << 24 | $$1 << 16 | $$2 << 8 | $$3;
     }
 
-    public static int multiply(int packedColourOne, int packedColorTwo) {
-        return net.minecraft.util.FastColor$ARGB32.color((int)(net.minecraft.util.FastColor$ARGB32.alpha((int)packedColourOne) * net.minecraft.util.FastColor$ARGB32.alpha((int)packedColorTwo) / 255), (int)(net.minecraft.util.FastColor$ARGB32.red((int)packedColourOne) * net.minecraft.util.FastColor$ARGB32.red((int)packedColorTwo) / 255), (int)(net.minecraft.util.FastColor$ARGB32.green((int)packedColourOne) * net.minecraft.util.FastColor$ARGB32.green((int)packedColorTwo) / 255), (int)(net.minecraft.util.FastColor$ARGB32.blue((int)packedColourOne) * net.minecraft.util.FastColor$ARGB32.blue((int)packedColorTwo) / 255));
+    public static int multiply(int $$0, int $$1) {
+        return net.minecraft.util.FastColor$ARGB32.color((int)(net.minecraft.util.FastColor$ARGB32.alpha((int)$$0) * net.minecraft.util.FastColor$ARGB32.alpha((int)$$1) / 255), (int)(net.minecraft.util.FastColor$ARGB32.red((int)$$0) * net.minecraft.util.FastColor$ARGB32.red((int)$$1) / 255), (int)(net.minecraft.util.FastColor$ARGB32.green((int)$$0) * net.minecraft.util.FastColor$ARGB32.green((int)$$1) / 255), (int)(net.minecraft.util.FastColor$ARGB32.blue((int)$$0) * net.minecraft.util.FastColor$ARGB32.blue((int)$$1) / 255));
     }
 }
 