--- net/minecraft/util/FrameTimer.java
+++ net/minecraft/util/FrameTimer.java
@@ -15,8 +15,8 @@
         this.loggedTimes = new long[240];
     }
 
-    public void logFrameDuration(long runningTime) {
-        this.loggedTimes[this.logEnd] = runningTime;
+    public void logFrameDuration(long $$0) {
+        this.loggedTimes[this.logEnd] = $$0;
         ++this.logEnd;
         if (this.logEnd == 240) {
             this.logEnd = 0;
@@ -42,9 +42,9 @@
         return this.scaleSampleTo(this.getAverageDuration($$0), $$1, 60);
     }
 
-    public int scaleSampleTo(long value, int scale, int divisor) {
-        double d = (double)value / (double)(1000000000L / (long)divisor);
-        return (int)(d * (double)scale);
+    public int scaleSampleTo(long $$0, int $$1, int $$2) {
+        double $$3 = (double)$$0 / (double)(1000000000L / (long)$$2);
+        return (int)($$3 * (double)$$1);
     }
 
     public int getLogStart() {
@@ -55,8 +55,8 @@
         return this.logEnd;
     }
 
-    public int wrapIndex(int rawIndex) {
-        return rawIndex % 240;
+    public int wrapIndex(int $$0) {
+        return $$0 % 240;
     }
 
     public long[] getLog() {