--- net/minecraft/util/LinearCongruentialGenerator.java
+++ net/minecraft/util/LinearCongruentialGenerator.java
@@ -11,9 +11,9 @@
     public net.minecraft.util.LinearCongruentialGenerator() {
     }
 
-    public static long next(long left, long right) {
-        left *= left * 6364136223846793005L + 1442695040888963407L;
-        return left += right;
+    public static long next(long $$0, long $$1) {
+        $$0 *= $$0 * 6364136223846793005L + 1442695040888963407L;
+        return $$0 += $$1;
     }
 }
 