--- net/minecraft/util/LowerCaseEnumTypeAdapterFactory.java
+++ net/minecraft/util/LowerCaseEnumTypeAdapterFactory.java
@@ -22,11 +22,11 @@
         return new /* Unavailable Anonymous Inner Class!! */;
     }
 
-    public java.lang.String toLowercase(java.lang.Object object) {
-        if (object instanceof java.lang.Enum) {
-            return ((java.lang.Enum)object).name().toLowerCase(java.util.Locale.ROOT);
+    public java.lang.String toLowercase(java.lang.Object $$0) {
+        if ($$0 instanceof java.lang.Enum) {
+            return ((java.lang.Enum)$$0).name().toLowerCase(java.util.Locale.ROOT);
         }
-        return object.toString().toLowerCase(java.util.Locale.ROOT);
+        return $$0.toString().toLowerCase(java.util.Locale.ROOT);
     }
 }
 