--- net/minecraft/util/SortedArraySet.java
+++ net/minecraft/util/SortedArraySet.java
@@ -23,8 +23,8 @@
         return net.minecraft.util.SortedArraySet.create((int)10);
     }
 
-    public static <T extends java.lang.Comparable<T>> net.minecraft.util.SortedArraySet<T> create(int initialCapacity) {
-        return new net.minecraft.util.SortedArraySet(initialCapacity, java.util.Comparator.naturalOrder());
+    public static <T extends java.lang.Comparable<T>> net.minecraft.util.SortedArraySet<T> create(int $$0) {
+        return new net.minecraft.util.SortedArraySet($$0, java.util.Comparator.naturalOrder());
     }
 
     public static <T> net.minecraft.util.SortedArraySet<T> create(java.util.Comparator<T> $$0) {
@@ -35,12 +35,12 @@
         return new net.minecraft.util.SortedArraySet($$1, $$0);
     }
 
-    public static <T> T[] castRawArray(java.lang.Object[] array) {
-        return array;
+    public static <T> T[] castRawArray(java.lang.Object[] $$0) {
+        return $$0;
     }
 
-    public int findIndex(T object) {
-        return java.util.Arrays.binarySearch((java.lang.Object[])this.contents, (int)0, (int)this.size, object, (java.util.Comparator)this.comparator);
+    public int findIndex(T $$0) {
+        return java.util.Arrays.binarySearch((java.lang.Object[])this.contents, (int)0, (int)this.size, $$0, (java.util.Comparator)this.comparator);
     }
 
     public static int getInsertionPosition(int $$0) {