--- net/minecraft/util/TaskChainer$DelayedTask.java
+++ net/minecraft/util/TaskChainer$DelayedTask.java
@@ -1,10 +1,1 @@
-/*
- * Decompiled with CFR 0.151.
- */
-package net.minecraft.util;
-
-public interface net.minecraft.util.TaskChainer$DelayedTask
-extends java.util.function.Supplier<java.util.concurrent.CompletableFuture<?>>
- {
-}
-
+// Failed to decompile net/minecraft/util/TaskChainer$DelayedTask.class