--- net/minecraft/data/models/blockstates/Condition$CompositeCondition.java
+++ net/minecraft/data/models/blockstates/Condition$CompositeCondition.java
@@ -9,13 +9,13 @@
     public net.minecraft.data.models.blockstates.Condition$Operation operation;
     public java.util.List<net.minecraft.data.models.blockstates.Condition> subconditions;
 
-    public net.minecraft.data.models.blockstates.Condition$CompositeCondition(net.minecraft.data.models.blockstates.Condition$Operation operation, java.util.List<net.minecraft.data.models.blockstates.Condition> subconditions) {
-        this.operation = operation;
-        this.subconditions = subconditions;
+    public net.minecraft.data.models.blockstates.Condition$CompositeCondition(net.minecraft.data.models.blockstates.Condition$Operation $$0, java.util.List<net.minecraft.data.models.blockstates.Condition> $$1) {
+        this.operation = $$0;
+        this.subconditions = $$1;
     }
 
-    public void validate(net.minecraft.world.level.block.state.StateDefinition<?, ?> stateDefinition) {
-        this.subconditions.forEach($$1 -> $$1.validate(stateDefinition));
+    public void validate(net.minecraft.world.level.block.state.StateDefinition<?, ?> $$0) {
+        this.subconditions.forEach($$1 -> $$1.validate($$0));
     }
 
     public com.google.gson.JsonElement get() {