--- net/minecraft/data/models/blockstates/MultiPartGenerator$Entry.java
+++ net/minecraft/data/models/blockstates/MultiPartGenerator$Entry.java
@@ -8,14 +8,14 @@
  {
     public java.util.List<net.minecraft.data.models.blockstates.Variant> variants;
 
-    public net.minecraft.data.models.blockstates.MultiPartGenerator$Entry(java.util.List<net.minecraft.data.models.blockstates.Variant> variants) {
-        this.variants = variants;
+    public net.minecraft.data.models.blockstates.MultiPartGenerator$Entry(java.util.List<net.minecraft.data.models.blockstates.Variant> $$0) {
+        this.variants = $$0;
     }
 
-    public void validate(net.minecraft.world.level.block.state.StateDefinition<?, ?> stateDefinition) {
+    public void validate(net.minecraft.world.level.block.state.StateDefinition<?, ?> $$0) {
     }
 
-    public void decorate(com.google.gson.JsonObject jsonObject) {
+    public void decorate(com.google.gson.JsonObject $$0) {
     }
 
     public com.google.gson.JsonElement get() {