--- net/minecraft/data/models/blockstates/MultiPartGenerator$ConditionalEntry.java
+++ net/minecraft/data/models/blockstates/MultiPartGenerator$ConditionalEntry.java
@@ -8,17 +8,17 @@
  {
     public net.minecraft.data.models.blockstates.Condition condition;
 
-    public net.minecraft.data.models.blockstates.MultiPartGenerator$ConditionalEntry(net.minecraft.data.models.blockstates.Condition condition, java.util.List<net.minecraft.data.models.blockstates.Variant> variants) {
-        super(variants);
-        this.condition = condition;
+    public net.minecraft.data.models.blockstates.MultiPartGenerator$ConditionalEntry(net.minecraft.data.models.blockstates.Condition $$0, java.util.List<net.minecraft.data.models.blockstates.Variant> $$1) {
+        super($$1);
+        this.condition = $$0;
     }
 
-    public void validate(net.minecraft.world.level.block.state.StateDefinition<?, ?> stateDefinition) {
-        this.condition.validate(stateDefinition);
+    public void validate(net.minecraft.world.level.block.state.StateDefinition<?, ?> $$0) {
+        this.condition.validate($$0);
     }
 
-    public void decorate(com.google.gson.JsonObject jsonObject) {
-        jsonObject.add("when", (com.google.gson.JsonElement)this.condition.get());
+    public void decorate(com.google.gson.JsonObject $$0) {
+        $$0.add("when", (com.google.gson.JsonElement)this.condition.get());
     }
 }
 