--- net/minecraft/data/models/blockstates/PropertyDispatch$C3.java
+++ net/minecraft/data/models/blockstates/PropertyDispatch$C3.java
@@ -10,28 +10,28 @@
     public net.minecraft.world.level.block.state.properties.Property<T2> property2;
     public net.minecraft.world.level.block.state.properties.Property<T3> property3;
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C3(net.minecraft.world.level.block.state.properties.Property<T1> property1, net.minecraft.world.level.block.state.properties.Property<T2> property2, net.minecraft.world.level.block.state.properties.Property<T3> property3) {
-        this.property1 = property1;
-        this.property2 = property2;
-        this.property3 = property3;
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C3(net.minecraft.world.level.block.state.properties.Property<T1> $$0, net.minecraft.world.level.block.state.properties.Property<T2> $$1, net.minecraft.world.level.block.state.properties.Property<T3> $$2) {
+        this.property1 = $$0;
+        this.property2 = $$1;
+        this.property3 = $$2;
     }
 
     public java.util.List<net.minecraft.world.level.block.state.properties.Property<?>> getDefinedProperties() {
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.property1, (java.lang.Object)this.property2, (java.lang.Object)this.property3);
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C3<T1, T2, T3> select(T1 property1Value, T2 property2Value, T3 property3Value, java.util.List<net.minecraft.data.models.blockstates.Variant> variants) {
-        net.minecraft.data.models.blockstates.Selector selector = net.minecraft.data.models.blockstates.Selector.of((net.minecraft.world.level.block.state.properties.Property$Value[])new net.minecraft.world.level.block.state.properties.Property$Value[]{this.property1.value(property1Value), this.property2.value(property2Value), this.property3.value(property3Value)});
-        this.putValue(selector, variants);
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C3<T1, T2, T3> select(T1 $$0, T2 $$1, T3 $$2, java.util.List<net.minecraft.data.models.blockstates.Variant> $$3) {
+        net.minecraft.data.models.blockstates.Selector $$4 = net.minecraft.data.models.blockstates.Selector.of((net.minecraft.world.level.block.state.properties.Property$Value[])new net.minecraft.world.level.block.state.properties.Property$Value[]{this.property1.value($$0), this.property2.value($$1), this.property3.value($$2)});
+        this.putValue($$4, $$3);
         return this;
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C3<T1, T2, T3> select(T1 property1Value, T2 property2Value, T3 property3Value, net.minecraft.data.models.blockstates.Variant variant) {
-        return this.select(property1Value, property2Value, property3Value, java.util.Collections.singletonList((java.lang.Object)variant));
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C3<T1, T2, T3> select(T1 $$0, T2 $$1, T3 $$2, net.minecraft.data.models.blockstates.Variant $$3) {
+        return this.select($$0, $$1, $$2, java.util.Collections.singletonList((java.lang.Object)$$3));
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch generate(net.minecraft.data.models.blockstates.PropertyDispatch$TriFunction<T1, T2, T3, net.minecraft.data.models.blockstates.Variant> propertyValuesToVariantMapper) {
-        this.property1.getPossibleValues().forEach($$1 -> this.property2.getPossibleValues().forEach($$2 -> this.property3.getPossibleValues().forEach($$3 -> this.select($$1, $$2, $$3, (net.minecraft.data.models.blockstates.Variant)propertyValuesToVariantMapper.apply($$1, $$2, $$3)))));
+    public net.minecraft.data.models.blockstates.PropertyDispatch generate(net.minecraft.data.models.blockstates.PropertyDispatch$TriFunction<T1, T2, T3, net.minecraft.data.models.blockstates.Variant> $$0) {
+        this.property1.getPossibleValues().forEach($$1 -> this.property2.getPossibleValues().forEach($$2 -> this.property3.getPossibleValues().forEach($$3 -> this.select($$1, $$2, $$3, (net.minecraft.data.models.blockstates.Variant)$$0.apply($$1, $$2, $$3)))));
         return this;
     }
 