--- net/minecraft/data/models/blockstates/PropertyDispatch$C1.java
+++ net/minecraft/data/models/blockstates/PropertyDispatch$C1.java
@@ -8,26 +8,26 @@
  {
     public net.minecraft.world.level.block.state.properties.Property<T1> property1;
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C1(net.minecraft.world.level.block.state.properties.Property<T1> property1) {
-        this.property1 = property1;
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C1(net.minecraft.world.level.block.state.properties.Property<T1> $$0) {
+        this.property1 = $$0;
     }
 
     public java.util.List<net.minecraft.world.level.block.state.properties.Property<?>> getDefinedProperties() {
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.property1);
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C1<T1> select(T1 propertyValue, java.util.List<net.minecraft.data.models.blockstates.Variant> variants) {
-        net.minecraft.data.models.blockstates.Selector selector = net.minecraft.data.models.blockstates.Selector.of((net.minecraft.world.level.block.state.properties.Property$Value[])new net.minecraft.world.level.block.state.properties.Property$Value[]{this.property1.value(propertyValue)});
-        this.putValue(selector, variants);
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C1<T1> select(T1 $$0, java.util.List<net.minecraft.data.models.blockstates.Variant> $$1) {
+        net.minecraft.data.models.blockstates.Selector $$2 = net.minecraft.data.models.blockstates.Selector.of((net.minecraft.world.level.block.state.properties.Property$Value[])new net.minecraft.world.level.block.state.properties.Property$Value[]{this.property1.value($$0)});
+        this.putValue($$2, $$1);
         return this;
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C1<T1> select(T1 propertyValue, net.minecraft.data.models.blockstates.Variant variant) {
-        return this.select(propertyValue, java.util.Collections.singletonList((java.lang.Object)variant));
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C1<T1> select(T1 $$0, net.minecraft.data.models.blockstates.Variant $$1) {
+        return this.select($$0, java.util.Collections.singletonList((java.lang.Object)$$1));
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch generate(java.util.function.Function<T1, net.minecraft.data.models.blockstates.Variant> propertyValueToVariantMapper) {
-        this.property1.getPossibleValues().forEach($$1 -> this.select($$1, (net.minecraft.data.models.blockstates.Variant)propertyValueToVariantMapper.apply($$1)));
+    public net.minecraft.data.models.blockstates.PropertyDispatch generate(java.util.function.Function<T1, net.minecraft.data.models.blockstates.Variant> $$0) {
+        this.property1.getPossibleValues().forEach($$1 -> this.select($$1, (net.minecraft.data.models.blockstates.Variant)$$0.apply($$1)));
         return this;
     }
 