--- net/minecraft/data/models/blockstates/PropertyDispatch$C4.java
+++ net/minecraft/data/models/blockstates/PropertyDispatch$C4.java
@@ -11,25 +11,25 @@
     public net.minecraft.world.level.block.state.properties.Property<T3> property3;
     public net.minecraft.world.level.block.state.properties.Property<T4> property4;
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C4(net.minecraft.world.level.block.state.properties.Property<T1> property1, net.minecraft.world.level.block.state.properties.Property<T2> property2, net.minecraft.world.level.block.state.properties.Property<T3> property3, net.minecraft.world.level.block.state.properties.Property<T4> property4) {
-        this.property1 = property1;
-        this.property2 = property2;
-        this.property3 = property3;
-        this.property4 = property4;
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C4(net.minecraft.world.level.block.state.properties.Property<T1> $$0, net.minecraft.world.level.block.state.properties.Property<T2> $$1, net.minecraft.world.level.block.state.properties.Property<T3> $$2, net.minecraft.world.level.block.state.properties.Property<T4> $$3) {
+        this.property1 = $$0;
+        this.property2 = $$1;
+        this.property3 = $$2;
+        this.property4 = $$3;
     }
 
     public java.util.List<net.minecraft.world.level.block.state.properties.Property<?>> getDefinedProperties() {
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.property1, (java.lang.Object)this.property2, (java.lang.Object)this.property3, (java.lang.Object)this.property4);
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C4<T1, T2, T3, T4> select(T1 property1Value, T2 property2Value, T3 property3Value, T4 property4Value, java.util.List<net.minecraft.data.models.blockstates.Variant> variants) {
-        net.minecraft.data.models.blockstates.Selector selector = net.minecraft.data.models.blockstates.Selector.of((net.minecraft.world.level.block.state.properties.Property$Value[])new net.minecraft.world.level.block.state.properties.Property$Value[]{this.property1.value(property1Value), this.property2.value(property2Value), this.property3.value(property3Value), this.property4.value(property4Value)});
-        this.putValue(selector, variants);
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C4<T1, T2, T3, T4> select(T1 $$0, T2 $$1, T3 $$2, T4 $$3, java.util.List<net.minecraft.data.models.blockstates.Variant> $$4) {
+        net.minecraft.data.models.blockstates.Selector $$5 = net.minecraft.data.models.blockstates.Selector.of((net.minecraft.world.level.block.state.properties.Property$Value[])new net.minecraft.world.level.block.state.properties.Property$Value[]{this.property1.value($$0), this.property2.value($$1), this.property3.value($$2), this.property4.value($$3)});
+        this.putValue($$5, $$4);
         return this;
     }
 
-    public net.minecraft.data.models.blockstates.PropertyDispatch$C4<T1, T2, T3, T4> select(T1 property1Value, T2 property2Value, T3 property3Value, T4 property4Value, net.minecraft.data.models.blockstates.Variant variant) {
-        return this.select(property1Value, property2Value, property3Value, property4Value, java.util.Collections.singletonList((java.lang.Object)variant));
+    public net.minecraft.data.models.blockstates.PropertyDispatch$C4<T1, T2, T3, T4> select(T1 $$0, T2 $$1, T3 $$2, T4 $$3, net.minecraft.data.models.blockstates.Variant $$4) {
+        return this.select($$0, $$1, $$2, $$3, java.util.Collections.singletonList((java.lang.Object)$$4));
     }
 
     public net.minecraft.data.models.blockstates.PropertyDispatch generate(net.minecraft.data.models.blockstates.PropertyDispatch$QuadFunction<T1, T2, T3, T4, net.minecraft.data.models.blockstates.Variant> $$0) {