--- net/minecraft/data/models/blockstates/VariantProperty$Value.java
+++ net/minecraft/data/models/blockstates/VariantProperty$Value.java
@@ -8,17 +8,17 @@
     public T value;
     public /* synthetic */ net.minecraft.data.models.blockstates.VariantProperty this$0;
 
-    public net.minecraft.data.models.blockstates.VariantProperty$Value(net.minecraft.data.models.blockstates.VariantProperty variantProperty, T value) {
-        this.this$0 = variantProperty;
-        this.value = value;
+    public net.minecraft.data.models.blockstates.VariantProperty$Value(net.minecraft.data.models.blockstates.VariantProperty $$0, T $$1) {
+        this.this$0 = $$0;
+        this.value = $$1;
     }
 
     public net.minecraft.data.models.blockstates.VariantProperty<T> getKey() {
         return this.this$0;
     }
 
-    public void addToVariant(com.google.gson.JsonObject jsonObject) {
-        jsonObject.add(this.this$0.key, (com.google.gson.JsonElement)this.this$0.serializer.apply(this.value));
+    public void addToVariant(com.google.gson.JsonObject $$0) {
+        $$0.add(this.this$0.key, (com.google.gson.JsonElement)this.this$0.serializer.apply(this.value));
     }
 
     public java.lang.String toString() {