--- net/minecraft/data/models/blockstates/VariantProperty.java
+++ net/minecraft/data/models/blockstates/VariantProperty.java
@@ -8,13 +8,13 @@
     public java.lang.String key;
     public java.util.function.Function<T, com.google.gson.JsonElement> serializer;
 
-    public net.minecraft.data.models.blockstates.VariantProperty(java.lang.String key, java.util.function.Function<T, com.google.gson.JsonElement> serializer) {
-        this.key = key;
-        this.serializer = serializer;
+    public net.minecraft.data.models.blockstates.VariantProperty(java.lang.String $$0, java.util.function.Function<T, com.google.gson.JsonElement> $$1) {
+        this.key = $$0;
+        this.serializer = $$1;
     }
 
-    public net.minecraft.data.models.blockstates.VariantProperty$Value withValue(T value) {
-        return new net.minecraft.data.models.blockstates.VariantProperty$Value(this, value);
+    public net.minecraft.data.models.blockstates.VariantProperty$Value withValue(T $$0) {
+        return new net.minecraft.data.models.blockstates.VariantProperty$Value(this, $$0);
     }
 
     public java.lang.String toString() {