--- net/minecraft/world/level/block/AbstractBannerBlock.java
+++ net/minecraft/world/level/block/AbstractBannerBlock.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.item.DyeColor color;
 
-    public net.minecraft.world.level.block.AbstractBannerBlock(net.minecraft.world.item.DyeColor color, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
-        this.color = color;
+    public net.minecraft.world.level.block.AbstractBannerBlock(net.minecraft.world.item.DyeColor $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$1);
+        this.color = $$0;
     }
 
     public boolean isPossibleToRespawnInThis() {
@@ -21,20 +21,20 @@
         return new net.minecraft.world.level.block.entity.BannerBlockEntity($$0, $$1, this.color);
     }
 
-    public void setPlacedBy(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity placer, net.minecraft.world.item.ItemStack stack) {
-        if (level.isClientSide) {
-            level.getBlockEntity(pos, net.minecraft.world.level.block.entity.BlockEntityType.BANNER).ifPresent($$1 -> $$1.fromItem(stack));
-        } else if (stack.hasCustomHoverName()) {
-            level.getBlockEntity(pos, net.minecraft.world.level.block.entity.BlockEntityType.BANNER).ifPresent($$1 -> $$1.setCustomName(stack.getHoverName()));
+    public void setPlacedBy(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$12, net.minecraft.world.level.block.state.BlockState $$2, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$3, net.minecraft.world.item.ItemStack $$4) {
+        if ($$0.isClientSide) {
+            $$0.getBlockEntity($$12, net.minecraft.world.level.block.entity.BlockEntityType.BANNER).ifPresent($$1 -> $$1.fromItem($$4));
+        } else if ($$4.hasCustomHoverName()) {
+            $$0.getBlockEntity($$12, net.minecraft.world.level.block.entity.BlockEntityType.BANNER).ifPresent($$1 -> $$1.setCustomName($$4.getHoverName()));
         }
     }
 
-    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state) {
-        net.minecraft.world.level.block.entity.BlockEntity blockEntity = level.getBlockEntity(pos);
-        if (blockEntity instanceof net.minecraft.world.level.block.entity.BannerBlockEntity) {
-            return ((net.minecraft.world.level.block.entity.BannerBlockEntity)blockEntity).getItem();
+    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2) {
+        net.minecraft.world.level.block.entity.BlockEntity $$3 = $$0.getBlockEntity($$1);
+        if ($$3 instanceof net.minecraft.world.level.block.entity.BannerBlockEntity) {
+            return ((net.minecraft.world.level.block.entity.BannerBlockEntity)$$3).getItem();
         }
-        return super.getCloneItemStack(level, pos, state);
+        return super.getCloneItemStack($$0, $$1, $$2);
     }
 
     public net.minecraft.world.item.DyeColor getColor() {