--- net/minecraft/world/level/block/BambooSaplingBlock.java
+++ net/minecraft/world/level/block/BambooSaplingBlock.java
@@ -14,9 +14,9 @@
         super($$0);
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
-        net.minecraft.world.phys.Vec3 vec3 = state.getOffset(level, pos);
-        return SAPLING_SHAPE.move(vec3.x, vec3.y, vec3.z);
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
+        net.minecraft.world.phys.Vec3 $$4 = $$0.getOffset($$1, $$2);
+        return SAPLING_SHAPE.move($$4.x, $$4.y, $$4.z);
     }
 
     public void randomTick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.server.level.ServerLevel $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
@@ -25,26 +25,26 @@
         }
     }
 
-    public boolean canSurvive(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        return level.getBlockState(pos.below()).is(net.minecraft.tags.BlockTags.BAMBOO_PLANTABLE_ON);
+    public boolean canSurvive(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.LevelReader $$1, net.minecraft.core.BlockPos $$2) {
+        return $$1.getBlockState($$2.below()).is(net.minecraft.tags.BlockTags.BAMBOO_PLANTABLE_ON);
     }
 
-    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.core.Direction facing, net.minecraft.world.level.block.state.BlockState facingState, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos currentPos, net.minecraft.core.BlockPos facingPos) {
-        if (!state.canSurvive((net.minecraft.world.level.LevelReader)level, currentPos)) {
+    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.core.Direction $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.LevelAccessor $$3, net.minecraft.core.BlockPos $$4, net.minecraft.core.BlockPos $$5) {
+        if (!$$0.canSurvive((net.minecraft.world.level.LevelReader)$$3, $$4)) {
             return net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
         }
-        if (facing == net.minecraft.core.Direction.UP && facingState.is(net.minecraft.world.level.block.Blocks.BAMBOO)) {
-            level.setBlock(currentPos, net.minecraft.world.level.block.Blocks.BAMBOO.defaultBlockState(), 2);
+        if ($$1 == net.minecraft.core.Direction.UP && $$2.is(net.minecraft.world.level.block.Blocks.BAMBOO)) {
+            $$3.setBlock($$4, net.minecraft.world.level.block.Blocks.BAMBOO.defaultBlockState(), 2);
         }
-        return super.updateShape(state, facing, facingState, level, currentPos, facingPos);
+        return super.updateShape($$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2) {
         return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.BAMBOO);
     }
 
-    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, boolean isClient) {
-        return level.getBlockState(pos.above()).isAir();
+    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, boolean $$3) {
+        return $$0.getBlockState($$1.above()).isAir();
     }
 
     public boolean isBonemealSuccess(net.minecraft.world.level.Level $$0, net.minecraft.util.RandomSource $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3) {
@@ -55,15 +55,15 @@
         this.growBamboo((net.minecraft.world.level.Level)$$0, $$2);
     }
 
-    public float getDestroyProgress(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.entity.player.Player player, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        if (player.getMainHandItem().getItem() instanceof net.minecraft.world.item.SwordItem) {
+    public float getDestroyProgress(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.world.level.BlockGetter $$2, net.minecraft.core.BlockPos $$3) {
+        if ($$1.getMainHandItem().getItem() instanceof net.minecraft.world.item.SwordItem) {
             return 1.0f;
         }
-        return super.getDestroyProgress(state, player, level, pos);
+        return super.getDestroyProgress($$0, $$1, $$2, $$3);
     }
 
-    public void growBamboo(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos state) {
-        level.setBlock(state.above(), (net.minecraft.world.level.block.state.BlockState)net.minecraft.world.level.block.Blocks.BAMBOO.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BambooBlock.LEAVES, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.BambooLeaves.SMALL), 3);
+    public void growBamboo(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1) {
+        $$0.setBlock($$1.above(), (net.minecraft.world.level.block.state.BlockState)net.minecraft.world.level.block.Blocks.BAMBOO.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BambooBlock.LEAVES, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.BambooLeaves.SMALL), 3);
     }
 
     static {