--- net/minecraft/world/level/block/BlastFurnaceBlock.java
+++ net/minecraft/world/level/block/BlastFurnaceBlock.java
@@ -19,11 +19,11 @@
         return net.minecraft.world.level.block.BlastFurnaceBlock.createFurnaceTicker((net.minecraft.world.level.Level)$$0, $$2, (net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.BLAST_FURNACE);
     }
 
-    public void openContainer(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.player.Player player) {
-        net.minecraft.world.level.block.entity.BlockEntity blockEntity = level.getBlockEntity(pos);
-        if (blockEntity instanceof net.minecraft.world.level.block.entity.BlastFurnaceBlockEntity) {
-            player.openMenu((net.minecraft.world.MenuProvider)blockEntity);
-            player.awardStat(net.minecraft.stats.Stats.INTERACT_WITH_BLAST_FURNACE);
+    public void openContainer(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.entity.player.Player $$2) {
+        net.minecraft.world.level.block.entity.BlockEntity $$3 = $$0.getBlockEntity($$1);
+        if ($$3 instanceof net.minecraft.world.level.block.entity.BlastFurnaceBlockEntity) {
+            $$2.openMenu((net.minecraft.world.MenuProvider)$$3);
+            $$2.awardStat(net.minecraft.stats.Stats.INTERACT_WITH_BLAST_FURNACE);
         }
     }
 