--- net/minecraft/world/level/block/Block$BlockStatePairKey.java
+++ net/minecraft/world/level/block/Block$BlockStatePairKey.java
@@ -9,21 +9,21 @@
     public net.minecraft.world.level.block.state.BlockState second;
     public net.minecraft.core.Direction direction;
 
-    public net.minecraft.world.level.block.Block$BlockStatePairKey(net.minecraft.world.level.block.state.BlockState first, net.minecraft.world.level.block.state.BlockState second, net.minecraft.core.Direction direction) {
-        this.first = first;
-        this.second = second;
-        this.direction = direction;
+    public net.minecraft.world.level.block.Block$BlockStatePairKey(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.state.BlockState $$1, net.minecraft.core.Direction $$2) {
+        this.first = $$0;
+        this.second = $$1;
+        this.direction = $$2;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (!(other instanceof net.minecraft.world.level.block.Block$BlockStatePairKey)) {
+        if (!($$0 instanceof net.minecraft.world.level.block.Block$BlockStatePairKey)) {
             return false;
         }
-        net.minecraft.world.level.block.Block$BlockStatePairKey blockStatePairKey = (net.minecraft.world.level.block.Block$BlockStatePairKey)other;
-        return this.first == blockStatePairKey.first && this.second == blockStatePairKey.second && this.direction == blockStatePairKey.direction;
+        net.minecraft.world.level.block.Block$BlockStatePairKey $$1 = (net.minecraft.world.level.block.Block$BlockStatePairKey)$$0;
+        return this.first == $$1.first && this.second == $$1.second && this.direction == $$1.direction;
     }
 
     public int hashCode() {