--- net/minecraft/world/level/block/ButtonBlock$1.java
+++ net/minecraft/world/level/block/ButtonBlock$1.java
@@ -53,6 +53,18 @@
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
+        try {
+            net.minecraft.world.level.block.ButtonBlock$1.$SwitchMap$net$minecraft$core$Direction[net.minecraft.core.Direction.UP.ordinal()] = 5;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.world.level.block.ButtonBlock$1.$SwitchMap$net$minecraft$core$Direction[net.minecraft.core.Direction.DOWN.ordinal()] = 6;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
     }
 }
 