--- net/minecraft/world/level/block/ButtonBlock.java
+++ net/minecraft/world/level/block/ButtonBlock.java
@@ -1,1 +1,192 @@
-// Failed to decompile net/minecraft/world/level/block/ButtonBlock.class
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level.block;
+
+public class net.minecraft.world.level.block.ButtonBlock
+extends net.minecraft.world.level.block.FaceAttachedHorizontalDirectionalBlock
+ {
+    public static net.minecraft.world.level.block.state.properties.BooleanProperty POWERED;
+    public static int PRESSED_DEPTH = 1;
+    public static int UNPRESSED_DEPTH = 2;
+    public static int HALF_AABB_HEIGHT = 2;
+    public static int HALF_AABB_WIDTH = 3;
+    public static net.minecraft.world.phys.shapes.VoxelShape CEILING_AABB_X;
+    public static net.minecraft.world.phys.shapes.VoxelShape CEILING_AABB_Z;
+    public static net.minecraft.world.phys.shapes.VoxelShape FLOOR_AABB_X;
+    public static net.minecraft.world.phys.shapes.VoxelShape FLOOR_AABB_Z;
+    public static net.minecraft.world.phys.shapes.VoxelShape NORTH_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape SOUTH_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape WEST_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape EAST_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_CEILING_AABB_X;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_CEILING_AABB_Z;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_FLOOR_AABB_X;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_FLOOR_AABB_Z;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_NORTH_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_SOUTH_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_WEST_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape PRESSED_EAST_AABB;
+    public net.minecraft.sounds.SoundEvent soundOff;
+    public net.minecraft.sounds.SoundEvent soundOn;
+    public int ticksToStayPressed;
+    public boolean arrowsCanPress;
+
+    public net.minecraft.world.level.block.ButtonBlock(net.minecraft.world.level.block.state.BlockBehaviour$Properties $$0, int $$1, boolean $$2, net.minecraft.sounds.SoundEvent $$3, net.minecraft.sounds.SoundEvent $$4) {
+        super($$0);
+        this.registerDefaultState((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)this.stateDefinition.any()).setValue((net.minecraft.world.level.block.state.properties.Property)FACING, (java.lang.Comparable)net.minecraft.core.Direction.NORTH)).setValue((net.minecraft.world.level.block.state.properties.Property)POWERED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)false))).setValue((net.minecraft.world.level.block.state.properties.Property)FACE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.AttachFace.WALL));
+        this.ticksToStayPressed = $$1;
+        this.arrowsCanPress = $$2;
+        this.soundOff = $$3;
+        this.soundOn = $$4;
+    }
+
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
+        net.minecraft.core.Direction $$4 = (net.minecraft.core.Direction)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)FACING);
+        boolean $$5 = (java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED);
+        switch (net.minecraft.world.level.block.ButtonBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$AttachFace[((net.minecraft.world.level.block.state.properties.AttachFace)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)FACE)).ordinal()]) {
+            case 1: {
+                if ($$4.getAxis() == net.minecraft.core.Direction$Axis.X) {
+                    return $$5 ? PRESSED_FLOOR_AABB_X : FLOOR_AABB_X;
+                }
+                return $$5 ? PRESSED_FLOOR_AABB_Z : FLOOR_AABB_Z;
+            }
+            case 2: {
+                return switch (net.minecraft.world.level.block.ButtonBlock$1.$SwitchMap$net$minecraft$core$Direction[$$4.ordinal()]) {
+                    default -> throw new java.lang.IncompatibleClassChangeError();
+                    case 1 -> {
+                        if ($$5) {
+                            yield PRESSED_EAST_AABB;
+                        }
+                        yield EAST_AABB;
+                    }
+                    case 2 -> {
+                        if ($$5) {
+                            yield PRESSED_WEST_AABB;
+                        }
+                        yield WEST_AABB;
+                    }
+                    case 3 -> {
+                        if ($$5) {
+                            yield PRESSED_SOUTH_AABB;
+                        }
+                        yield SOUTH_AABB;
+                    }
+                    case 4, 5, 6 -> $$5 ? PRESSED_NORTH_AABB : NORTH_AABB;
+                };
+            }
+        }
+        if ($$4.getAxis() == net.minecraft.core.Direction$Axis.X) {
+            return $$5 ? PRESSED_CEILING_AABB_X : CEILING_AABB_X;
+        }
+        return $$5 ? PRESSED_CEILING_AABB_Z : CEILING_AABB_Z;
+    }
+
+    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.player.Player $$3, net.minecraft.world.InteractionHand $$4, net.minecraft.world.phys.BlockHitResult $$5) {
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue()) {
+            return net.minecraft.world.InteractionResult.CONSUME;
+        }
+        this.press($$0, $$1, $$2);
+        this.playSound($$3, (net.minecraft.world.level.LevelAccessor)$$1, $$2, true);
+        $$1.gameEvent((net.minecraft.world.entity.Entity)$$3, net.minecraft.world.level.gameevent.GameEvent.BLOCK_ACTIVATE, $$2);
+        return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
+    }
+
+    public void press(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2) {
+        $$1.setBlock($$2, (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)POWERED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)), 3);
+        this.updateNeighbours($$0, $$1, $$2);
+        $$1.scheduleTick($$2, (net.minecraft.world.level.block.Block)this, this.ticksToStayPressed);
+    }
+
+    public void playSound(@javax.annotation.Nullable net.minecraft.world.entity.player.Player $$0, net.minecraft.world.level.LevelAccessor $$1, net.minecraft.core.BlockPos $$2, boolean $$3) {
+        $$1.playSound((net.minecraft.world.entity.player.Player)($$3 ? $$0 : null), $$2, this.getSound($$3), net.minecraft.sounds.SoundSource.BLOCKS);
+    }
+
+    public net.minecraft.sounds.SoundEvent getSound(boolean $$0) {
+        return $$0 ? this.soundOn : this.soundOff;
+    }
+
+    public void onRemove(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, boolean $$4) {
+        if ($$4 || $$0.is($$3.getBlock())) {
+            return;
+        }
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue()) {
+            this.updateNeighbours($$0, $$1, $$2);
+        }
+        super.onRemove($$0, $$1, $$2, $$3, $$4);
+    }
+
+    public int getSignal(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.core.Direction $$3) {
+        return (java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED) != false ? 15 : 0;
+    }
+
+    public int getDirectSignal(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.core.Direction $$3) {
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue() && net.minecraft.world.level.block.ButtonBlock.getConnectedDirection((net.minecraft.world.level.block.state.BlockState)$$0) == $$3) {
+            return 15;
+        }
+        return 0;
+    }
+
+    public boolean isSignalSource(net.minecraft.world.level.block.state.BlockState $$0) {
+        return true;
+    }
+
+    public void tick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.server.level.ServerLevel $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
+        if (!((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue()) {
+            return;
+        }
+        this.checkPressed($$0, (net.minecraft.world.level.Level)$$1, $$2);
+    }
+
+    public void entityInside(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.Entity $$3) {
+        if ($$1.isClientSide || !this.arrowsCanPress || ((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue()) {
+            return;
+        }
+        this.checkPressed($$0, $$1, $$2);
+    }
+
+    public void checkPressed(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2) {
+        boolean $$5;
+        net.minecraft.world.entity.projectile.AbstractArrow $$3 = this.arrowsCanPress ? (net.minecraft.world.entity.projectile.AbstractArrow)$$1.getEntitiesOfClass(net.minecraft.world.entity.projectile.AbstractArrow.class, $$0.getShape((net.minecraft.world.level.BlockGetter)$$1, $$2).bounds().move($$2)).stream().findFirst().orElse(null) : null;
+        boolean $$4 = $$3 != null;
+        if ($$4 != ($$5 = ((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue())) {
+            $$1.setBlock($$2, (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)POWERED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)$$4)), 3);
+            this.updateNeighbours($$0, $$1, $$2);
+            this.playSound(null, (net.minecraft.world.level.LevelAccessor)$$1, $$2, $$4);
+            $$1.gameEvent((net.minecraft.world.entity.Entity)$$3, $$4 ? net.minecraft.world.level.gameevent.GameEvent.BLOCK_ACTIVATE : net.minecraft.world.level.gameevent.GameEvent.BLOCK_DEACTIVATE, $$2);
+        }
+        if ($$4) {
+            $$1.scheduleTick(new net.minecraft.core.BlockPos((net.minecraft.core.Vec3i)$$2), (net.minecraft.world.level.block.Block)this, this.ticksToStayPressed);
+        }
+    }
+
+    public void updateNeighbours(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2) {
+        $$1.updateNeighborsAt($$2, (net.minecraft.world.level.block.Block)this);
+        $$1.updateNeighborsAt($$2.relative(net.minecraft.world.level.block.ButtonBlock.getConnectedDirection((net.minecraft.world.level.block.state.BlockState)$$0).getOpposite()), (net.minecraft.world.level.block.Block)this);
+    }
+
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{FACING, POWERED, FACE});
+    }
+
+    static {
+        POWERED = net.minecraft.world.level.block.state.properties.BlockStateProperties.POWERED;
+        CEILING_AABB_X = net.minecraft.world.level.block.Block.box((double)6.0, (double)14.0, (double)5.0, (double)10.0, (double)16.0, (double)11.0);
+        CEILING_AABB_Z = net.minecraft.world.level.block.Block.box((double)5.0, (double)14.0, (double)6.0, (double)11.0, (double)16.0, (double)10.0);
+        FLOOR_AABB_X = net.minecraft.world.level.block.Block.box((double)6.0, (double)0.0, (double)5.0, (double)10.0, (double)2.0, (double)11.0);
+        FLOOR_AABB_Z = net.minecraft.world.level.block.Block.box((double)5.0, (double)0.0, (double)6.0, (double)11.0, (double)2.0, (double)10.0);
+        NORTH_AABB = net.minecraft.world.level.block.Block.box((double)5.0, (double)6.0, (double)14.0, (double)11.0, (double)10.0, (double)16.0);
+        SOUTH_AABB = net.minecraft.world.level.block.Block.box((double)5.0, (double)6.0, (double)0.0, (double)11.0, (double)10.0, (double)2.0);
+        WEST_AABB = net.minecraft.world.level.block.Block.box((double)14.0, (double)6.0, (double)5.0, (double)16.0, (double)10.0, (double)11.0);
+        EAST_AABB = net.minecraft.world.level.block.Block.box((double)0.0, (double)6.0, (double)5.0, (double)2.0, (double)10.0, (double)11.0);
+        PRESSED_CEILING_AABB_X = net.minecraft.world.level.block.Block.box((double)6.0, (double)15.0, (double)5.0, (double)10.0, (double)16.0, (double)11.0);
+        PRESSED_CEILING_AABB_Z = net.minecraft.world.level.block.Block.box((double)5.0, (double)15.0, (double)6.0, (double)11.0, (double)16.0, (double)10.0);
+        PRESSED_FLOOR_AABB_X = net.minecraft.world.level.block.Block.box((double)6.0, (double)0.0, (double)5.0, (double)10.0, (double)1.0, (double)11.0);
+        PRESSED_FLOOR_AABB_Z = net.minecraft.world.level.block.Block.box((double)5.0, (double)0.0, (double)6.0, (double)11.0, (double)1.0, (double)10.0);
+        PRESSED_NORTH_AABB = net.minecraft.world.level.block.Block.box((double)5.0, (double)6.0, (double)15.0, (double)11.0, (double)10.0, (double)16.0);
+        PRESSED_SOUTH_AABB = net.minecraft.world.level.block.Block.box((double)5.0, (double)6.0, (double)0.0, (double)11.0, (double)10.0, (double)1.0);
+        PRESSED_WEST_AABB = net.minecraft.world.level.block.Block.box((double)15.0, (double)6.0, (double)5.0, (double)16.0, (double)10.0, (double)11.0);
+        PRESSED_EAST_AABB = net.minecraft.world.level.block.Block.box((double)0.0, (double)6.0, (double)5.0, (double)1.0, (double)10.0, (double)11.0);
+    }
+}
+