--- net/minecraft/world/level/block/ComposterBlock$EmptyContainer.java
+++ net/minecraft/world/level/block/ComposterBlock$EmptyContainer.java
@@ -11,15 +11,15 @@
         super(0);
     }
 
-    public int[] getSlotsForFace(net.minecraft.core.Direction side) {
+    public int[] getSlotsForFace(net.minecraft.core.Direction $$0) {
         return new int[0];
     }
 
-    public boolean canPlaceItemThroughFace(int index, net.minecraft.world.item.ItemStack itemStack, @javax.annotation.Nullable net.minecraft.core.Direction direction) {
+    public boolean canPlaceItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, @javax.annotation.Nullable net.minecraft.core.Direction $$2) {
         return false;
     }
 
-    public boolean canTakeItemThroughFace(int index, net.minecraft.world.item.ItemStack stack, net.minecraft.core.Direction direction) {
+    public boolean canTakeItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.core.Direction $$2) {
         return false;
     }
 }