--- net/minecraft/world/level/block/ComposterBlock$OutputContainer.java
+++ net/minecraft/world/level/block/ComposterBlock$OutputContainer.java
@@ -12,20 +12,20 @@
     public net.minecraft.core.BlockPos pos;
     public boolean changed;
 
-    public net.minecraft.world.level.block.ComposterBlock$OutputContainer(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos, net.minecraft.world.item.ItemStack itemStack) {
-        super(new net.minecraft.world.item.ItemStack[]{itemStack});
-        this.state = state;
-        this.level = level;
-        this.pos = pos;
+    public net.minecraft.world.level.block.ComposterBlock$OutputContainer(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.LevelAccessor $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.item.ItemStack $$3) {
+        super(new net.minecraft.world.item.ItemStack[]{$$3});
+        this.state = $$0;
+        this.level = $$1;
+        this.pos = $$2;
     }
 
     public int getMaxStackSize() {
         return 1;
     }
 
-    public int[] getSlotsForFace(net.minecraft.core.Direction side) {
+    public int[] getSlotsForFace(net.minecraft.core.Direction $$0) {
         int[] nArray;
-        if (side == net.minecraft.core.Direction.DOWN) {
+        if ($$0 == net.minecraft.core.Direction.DOWN) {
             int[] nArray2 = new int[1];
             nArray = nArray2;
             nArray2[0] = 0;
@@ -35,12 +35,12 @@
         return nArray;
     }
 
-    public boolean canPlaceItemThroughFace(int index, net.minecraft.world.item.ItemStack itemStack, @javax.annotation.Nullable net.minecraft.core.Direction direction) {
+    public boolean canPlaceItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, @javax.annotation.Nullable net.minecraft.core.Direction $$2) {
         return false;
     }
 
-    public boolean canTakeItemThroughFace(int index, net.minecraft.world.item.ItemStack stack, net.minecraft.core.Direction direction) {
-        return !this.changed && direction == net.minecraft.core.Direction.DOWN && stack.is(net.minecraft.world.item.Items.BONE_MEAL);
+    public boolean canTakeItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.core.Direction $$2) {
+        return !this.changed && $$2 == net.minecraft.core.Direction.DOWN && $$1.is(net.minecraft.world.item.Items.BONE_MEAL);
     }
 
     public void setChanged() {