--- net/minecraft/world/level/block/ComposterBlock$InputContainer.java
+++ net/minecraft/world/level/block/ComposterBlock$InputContainer.java
@@ -12,20 +12,20 @@
     public net.minecraft.core.BlockPos pos;
     public boolean changed;
 
-    public net.minecraft.world.level.block.ComposterBlock$InputContainer(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos) {
+    public net.minecraft.world.level.block.ComposterBlock$InputContainer(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.LevelAccessor $$1, net.minecraft.core.BlockPos $$2) {
         super(1);
-        this.state = state;
-        this.level = level;
-        this.pos = pos;
+        this.state = $$0;
+        this.level = $$1;
+        this.pos = $$2;
     }
 
     public int getMaxStackSize() {
         return 1;
     }
 
-    public int[] getSlotsForFace(net.minecraft.core.Direction side) {
+    public int[] getSlotsForFace(net.minecraft.core.Direction $$0) {
         int[] nArray;
-        if (side == net.minecraft.core.Direction.UP) {
+        if ($$0 == net.minecraft.core.Direction.UP) {
             int[] nArray2 = new int[1];
             nArray = nArray2;
             nArray2[0] = 0;
@@ -35,11 +35,11 @@
         return nArray;
     }
 
-    public boolean canPlaceItemThroughFace(int index, net.minecraft.world.item.ItemStack itemStack, @javax.annotation.Nullable net.minecraft.core.Direction direction) {
-        return !this.changed && direction == net.minecraft.core.Direction.UP && net.minecraft.world.level.block.ComposterBlock.COMPOSTABLES.containsKey((java.lang.Object)itemStack.getItem());
+    public boolean canPlaceItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, @javax.annotation.Nullable net.minecraft.core.Direction $$2) {
+        return !this.changed && $$2 == net.minecraft.core.Direction.UP && net.minecraft.world.level.block.ComposterBlock.COMPOSTABLES.containsKey((java.lang.Object)$$1.getItem());
     }
 
-    public boolean canTakeItemThroughFace(int index, net.minecraft.world.item.ItemStack stack, net.minecraft.core.Direction direction) {
+    public boolean canTakeItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.core.Direction $$2) {
         return false;
     }
 