--- net/minecraft/world/level/block/ConcretePowderBlock.java
+++ net/minecraft/world/level/block/ConcretePowderBlock.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.level.block.state.BlockState concrete;
 
-    public net.minecraft.world.level.block.ConcretePowderBlock(net.minecraft.world.level.block.Block concrete, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
-        this.concrete = concrete.defaultBlockState();
+    public net.minecraft.world.level.block.ConcretePowderBlock(net.minecraft.world.level.block.Block $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$1);
+        this.concrete = $$0.defaultBlockState();
     }
 
     public void onLand(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.block.state.BlockState $$3, net.minecraft.world.entity.item.FallingBlockEntity $$4) {
@@ -19,48 +19,48 @@
         }
     }
 
-    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext context) {
-        net.minecraft.world.level.block.state.BlockState blockState;
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.world.level.Level level = context.getLevel();
-        if (net.minecraft.world.level.block.ConcretePowderBlock.shouldSolidify((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)(blockPos = context.getClickedPos()), (net.minecraft.world.level.block.state.BlockState)(blockState = level.getBlockState(blockPos)))) {
+    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext $$0) {
+        net.minecraft.world.level.block.state.BlockState $$3;
+        net.minecraft.core.BlockPos $$2;
+        net.minecraft.world.level.Level $$1 = $$0.getLevel();
+        if (net.minecraft.world.level.block.ConcretePowderBlock.shouldSolidify((net.minecraft.world.level.BlockGetter)$$1, (net.minecraft.core.BlockPos)($$2 = $$0.getClickedPos()), (net.minecraft.world.level.block.state.BlockState)($$3 = $$1.getBlockState($$2)))) {
             return this.concrete;
         }
-        return super.getStateForPlacement(context);
+        return super.getStateForPlacement($$0);
     }
 
-    public static boolean shouldSolidify(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state) {
-        return net.minecraft.world.level.block.ConcretePowderBlock.canSolidify((net.minecraft.world.level.block.state.BlockState)state) || net.minecraft.world.level.block.ConcretePowderBlock.touchesLiquid((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)pos);
+    public static boolean shouldSolidify(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2) {
+        return net.minecraft.world.level.block.ConcretePowderBlock.canSolidify((net.minecraft.world.level.block.state.BlockState)$$2) || net.minecraft.world.level.block.ConcretePowderBlock.touchesLiquid((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$1);
     }
 
-    public static boolean touchesLiquid(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        boolean bl = false;
-        net.minecraft.core.BlockPos$MutableBlockPos mutableBlockPos = pos.mutable();
-        for (net.minecraft.core.Direction direction : net.minecraft.core.Direction.values()) {
-            net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState((net.minecraft.core.BlockPos)mutableBlockPos);
-            if (direction == net.minecraft.core.Direction.DOWN && !net.minecraft.world.level.block.ConcretePowderBlock.canSolidify((net.minecraft.world.level.block.state.BlockState)blockState)) continue;
-            mutableBlockPos.setWithOffset((net.minecraft.core.Vec3i)pos, direction);
-            blockState = level.getBlockState((net.minecraft.core.BlockPos)mutableBlockPos);
-            if (!net.minecraft.world.level.block.ConcretePowderBlock.canSolidify((net.minecraft.world.level.block.state.BlockState)blockState) || blockState.isFaceSturdy(level, pos, direction.getOpposite())) continue;
-            bl = true;
+    public static boolean touchesLiquid(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1) {
+        boolean $$2 = false;
+        net.minecraft.core.BlockPos$MutableBlockPos $$3 = $$1.mutable();
+        for (net.minecraft.core.Direction $$4 : net.minecraft.core.Direction.values()) {
+            net.minecraft.world.level.block.state.BlockState $$5 = $$0.getBlockState((net.minecraft.core.BlockPos)$$3);
+            if ($$4 == net.minecraft.core.Direction.DOWN && !net.minecraft.world.level.block.ConcretePowderBlock.canSolidify((net.minecraft.world.level.block.state.BlockState)$$5)) continue;
+            $$3.setWithOffset((net.minecraft.core.Vec3i)$$1, $$4);
+            $$5 = $$0.getBlockState((net.minecraft.core.BlockPos)$$3);
+            if (!net.minecraft.world.level.block.ConcretePowderBlock.canSolidify((net.minecraft.world.level.block.state.BlockState)$$5) || $$5.isFaceSturdy($$0, $$1, $$4.getOpposite())) continue;
+            $$2 = true;
             break;
         }
-        return bl;
+        return $$2;
     }
 
-    public static boolean canSolidify(net.minecraft.world.level.block.state.BlockState state) {
-        return state.getFluidState().is(net.minecraft.tags.FluidTags.WATER);
+    public static boolean canSolidify(net.minecraft.world.level.block.state.BlockState $$0) {
+        return $$0.getFluidState().is(net.minecraft.tags.FluidTags.WATER);
     }
 
-    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.core.Direction facing, net.minecraft.world.level.block.state.BlockState facingState, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos currentPos, net.minecraft.core.BlockPos facingPos) {
-        if (net.minecraft.world.level.block.ConcretePowderBlock.touchesLiquid((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)currentPos)) {
+    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.core.Direction $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.LevelAccessor $$3, net.minecraft.core.BlockPos $$4, net.minecraft.core.BlockPos $$5) {
+        if (net.minecraft.world.level.block.ConcretePowderBlock.touchesLiquid((net.minecraft.world.level.BlockGetter)$$3, (net.minecraft.core.BlockPos)$$4)) {
             return this.concrete;
         }
-        return super.updateShape(state, facing, facingState, level, currentPos, facingPos);
+        return super.updateShape($$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public int getDustColor(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter reader, net.minecraft.core.BlockPos pos) {
-        return state.getMapColor((net.minecraft.world.level.BlockGetter)reader, (net.minecraft.core.BlockPos)pos).col;
+    public int getDustColor(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2) {
+        return $$0.getMapColor((net.minecraft.world.level.BlockGetter)$$1, (net.minecraft.core.BlockPos)$$2).col;
     }
 }
 