--- net/minecraft/world/level/block/DoubleBlockCombiner.java
+++ net/minecraft/world/level/block/DoubleBlockCombiner.java
@@ -8,36 +8,36 @@
     public net.minecraft.world.level.block.DoubleBlockCombiner() {
     }
 
-    public static <S extends net.minecraft.world.level.block.entity.BlockEntity> net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult<S> combineWithNeigbour(net.minecraft.world.level.block.entity.BlockEntityType<S> blockEntityType, java.util.function.Function<net.minecraft.world.level.block.state.BlockState, net.minecraft.world.level.block.DoubleBlockCombiner$BlockType> function, java.util.function.Function<net.minecraft.world.level.block.state.BlockState, net.minecraft.core.Direction> function2, net.minecraft.world.level.block.state.properties.DirectionProperty directionProperty, net.minecraft.world.level.block.state.BlockState blockState, net.minecraft.world.level.LevelAccessor levelAccessor, net.minecraft.core.BlockPos blockPos, java.util.function.BiPredicate<net.minecraft.world.level.LevelAccessor, net.minecraft.core.BlockPos> biPredicate) {
-        net.minecraft.world.level.block.DoubleBlockCombiner$BlockType blockType;
-        boolean bl;
-        net.minecraft.world.level.block.entity.BlockEntity blockEntity = blockEntityType.getBlockEntity((net.minecraft.world.level.BlockGetter)levelAccessor, blockPos);
-        if (blockEntity == null) {
+    public static <S extends net.minecraft.world.level.block.entity.BlockEntity> net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult<S> combineWithNeigbour(net.minecraft.world.level.block.entity.BlockEntityType<S> $$0, java.util.function.Function<net.minecraft.world.level.block.state.BlockState, net.minecraft.world.level.block.DoubleBlockCombiner$BlockType> $$1, java.util.function.Function<net.minecraft.world.level.block.state.BlockState, net.minecraft.core.Direction> $$2, net.minecraft.world.level.block.state.properties.DirectionProperty $$3, net.minecraft.world.level.block.state.BlockState $$4, net.minecraft.world.level.LevelAccessor $$5, net.minecraft.core.BlockPos $$6, java.util.function.BiPredicate<net.minecraft.world.level.LevelAccessor, net.minecraft.core.BlockPos> $$7) {
+        net.minecraft.world.level.block.DoubleBlockCombiner$BlockType $$14;
+        boolean $$11;
+        net.minecraft.world.level.block.entity.BlockEntity $$8 = $$0.getBlockEntity((net.minecraft.world.level.BlockGetter)$$5, $$6);
+        if ($$8 == null) {
             return net.minecraft.world.level.block.DoubleBlockCombiner$Combiner::acceptNone;
         }
-        if (biPredicate.test((java.lang.Object)levelAccessor, (java.lang.Object)blockPos)) {
+        if ($$7.test((java.lang.Object)$$5, (java.lang.Object)$$6)) {
             return net.minecraft.world.level.block.DoubleBlockCombiner$Combiner::acceptNone;
         }
-        net.minecraft.world.level.block.DoubleBlockCombiner$BlockType blockType2 = (net.minecraft.world.level.block.DoubleBlockCombiner$BlockType)function.apply((java.lang.Object)blockState);
-        boolean bl2 = blockType2 == net.minecraft.world.level.block.DoubleBlockCombiner$BlockType.SINGLE;
-        boolean bl3 = bl = blockType2 == net.minecraft.world.level.block.DoubleBlockCombiner$BlockType.FIRST;
-        if (bl2) {
-            return new net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult$Single((java.lang.Object)blockEntity);
+        net.minecraft.world.level.block.DoubleBlockCombiner$BlockType $$9 = (net.minecraft.world.level.block.DoubleBlockCombiner$BlockType)$$1.apply((java.lang.Object)$$4);
+        boolean $$10 = $$9 == net.minecraft.world.level.block.DoubleBlockCombiner$BlockType.SINGLE;
+        boolean bl = $$11 = $$9 == net.minecraft.world.level.block.DoubleBlockCombiner$BlockType.FIRST;
+        if ($$10) {
+            return new net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult$Single((java.lang.Object)$$8);
         }
-        net.minecraft.core.BlockPos blockPos2 = blockPos.relative((net.minecraft.core.Direction)function2.apply((java.lang.Object)blockState));
-        net.minecraft.world.level.block.state.BlockState blockState2 = levelAccessor.getBlockState(blockPos2);
-        if (blockState2.is(blockState.getBlock()) && (blockType = (net.minecraft.world.level.block.DoubleBlockCombiner$BlockType)function.apply((java.lang.Object)blockState2)) != net.minecraft.world.level.block.DoubleBlockCombiner$BlockType.SINGLE && blockType2 != blockType && blockState2.getValue((net.minecraft.world.level.block.state.properties.Property)directionProperty) == blockState.getValue((net.minecraft.world.level.block.state.properties.Property)directionProperty)) {
-            if (biPredicate.test((java.lang.Object)levelAccessor, (java.lang.Object)blockPos2)) {
+        net.minecraft.core.BlockPos $$12 = $$6.relative((net.minecraft.core.Direction)$$2.apply((java.lang.Object)$$4));
+        net.minecraft.world.level.block.state.BlockState $$13 = $$5.getBlockState($$12);
+        if ($$13.is($$4.getBlock()) && ($$14 = (net.minecraft.world.level.block.DoubleBlockCombiner$BlockType)$$1.apply((java.lang.Object)$$13)) != net.minecraft.world.level.block.DoubleBlockCombiner$BlockType.SINGLE && $$9 != $$14 && $$13.getValue((net.minecraft.world.level.block.state.properties.Property)$$3) == $$4.getValue((net.minecraft.world.level.block.state.properties.Property)$$3)) {
+            if ($$7.test((java.lang.Object)$$5, (java.lang.Object)$$12)) {
                 return net.minecraft.world.level.block.DoubleBlockCombiner$Combiner::acceptNone;
             }
-            net.minecraft.world.level.block.entity.BlockEntity blockEntity2 = blockEntityType.getBlockEntity((net.minecraft.world.level.BlockGetter)levelAccessor, blockPos2);
-            if (blockEntity2 != null) {
-                net.minecraft.world.level.block.entity.BlockEntity blockEntity3 = bl ? blockEntity : blockEntity2;
-                net.minecraft.world.level.block.entity.BlockEntity blockEntity4 = bl ? blockEntity2 : blockEntity;
-                return new net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult$Double((java.lang.Object)blockEntity3, (java.lang.Object)blockEntity4);
+            net.minecraft.world.level.block.entity.BlockEntity $$15 = $$0.getBlockEntity((net.minecraft.world.level.BlockGetter)$$5, $$12);
+            if ($$15 != null) {
+                net.minecraft.world.level.block.entity.BlockEntity $$16 = $$11 ? $$8 : $$15;
+                net.minecraft.world.level.block.entity.BlockEntity $$17 = $$11 ? $$15 : $$8;
+                return new net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult$Double((java.lang.Object)$$16, (java.lang.Object)$$17);
             }
         }
-        return new net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult$Single((java.lang.Object)blockEntity);
+        return new net.minecraft.world.level.block.DoubleBlockCombiner$NeighborCombineResult$Single((java.lang.Object)$$8);
     }
 }
 