--- net/minecraft/world/level/block/EndPortalBlock.java
+++ net/minecraft/world/level/block/EndPortalBlock.java
@@ -16,18 +16,18 @@
         return new net.minecraft.world.level.block.entity.TheEndPortalBlockEntity($$0, $$1);
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
         return SHAPE;
     }
 
-    public void entityInside(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.Entity entity) {
-        if (level instanceof net.minecraft.server.level.ServerLevel && !entity.isPassenger() && !entity.isVehicle() && entity.canChangeDimensions() && net.minecraft.world.phys.shapes.Shapes.joinIsNotEmpty((net.minecraft.world.phys.shapes.VoxelShape)net.minecraft.world.phys.shapes.Shapes.create((net.minecraft.world.phys.AABB)entity.getBoundingBox().move((double)(-pos.getX()), (double)(-pos.getY()), (double)(-pos.getZ()))), (net.minecraft.world.phys.shapes.VoxelShape)state.getShape((net.minecraft.world.level.BlockGetter)level, pos), (net.minecraft.world.phys.shapes.BooleanOp)net.minecraft.world.phys.shapes.BooleanOp.AND)) {
-            net.minecraft.resources.ResourceKey resourceKey = level.dimension() == net.minecraft.world.level.Level.END ? net.minecraft.world.level.Level.OVERWORLD : net.minecraft.world.level.Level.END;
-            net.minecraft.server.level.ServerLevel serverLevel = ((net.minecraft.server.level.ServerLevel)level).getServer().getLevel(resourceKey);
-            if (serverLevel == null) {
+    public void entityInside(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.Entity $$3) {
+        if ($$1 instanceof net.minecraft.server.level.ServerLevel && !$$3.isPassenger() && !$$3.isVehicle() && $$3.canChangeDimensions() && net.minecraft.world.phys.shapes.Shapes.joinIsNotEmpty((net.minecraft.world.phys.shapes.VoxelShape)net.minecraft.world.phys.shapes.Shapes.create((net.minecraft.world.phys.AABB)$$3.getBoundingBox().move((double)(-$$2.getX()), (double)(-$$2.getY()), (double)(-$$2.getZ()))), (net.minecraft.world.phys.shapes.VoxelShape)$$0.getShape((net.minecraft.world.level.BlockGetter)$$1, $$2), (net.minecraft.world.phys.shapes.BooleanOp)net.minecraft.world.phys.shapes.BooleanOp.AND)) {
+            net.minecraft.resources.ResourceKey $$4 = $$1.dimension() == net.minecraft.world.level.Level.END ? net.minecraft.world.level.Level.OVERWORLD : net.minecraft.world.level.Level.END;
+            net.minecraft.server.level.ServerLevel $$5 = ((net.minecraft.server.level.ServerLevel)$$1).getServer().getLevel($$4);
+            if ($$5 == null) {
                 return;
             }
-            entity.changeDimension(serverLevel);
+            $$3.changeDimension($$5);
         }
     }
 
@@ -38,11 +38,11 @@
         $$1.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.SMOKE, $$4, $$5, $$6, 0.0, 0.0, 0.0);
     }
 
-    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2) {
         return net.minecraft.world.item.ItemStack.EMPTY;
     }
 
-    public boolean canBeReplaced(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.material.Fluid fluid) {
+    public boolean canBeReplaced(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.material.Fluid $$1) {
         return false;
     }
 