--- net/minecraft/world/level/block/FallingBlock.java
+++ net/minecraft/world/level/block/FallingBlock.java
@@ -11,13 +11,13 @@
         super($$0);
     }
 
-    public void onPlace(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState oldState, boolean isMoving) {
-        level.scheduleTick(pos, (net.minecraft.world.level.block.Block)this, this.getDelayAfterPlace());
+    public void onPlace(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, boolean $$4) {
+        $$1.scheduleTick($$2, (net.minecraft.world.level.block.Block)this, this.getDelayAfterPlace());
     }
 
-    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.core.Direction facing, net.minecraft.world.level.block.state.BlockState facingState, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos currentPos, net.minecraft.core.BlockPos facingPos) {
-        level.scheduleTick(currentPos, (net.minecraft.world.level.block.Block)this, this.getDelayAfterPlace());
-        return super.updateShape(state, facing, facingState, level, currentPos, facingPos);
+    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.core.Direction $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.LevelAccessor $$3, net.minecraft.core.BlockPos $$4, net.minecraft.core.BlockPos $$5) {
+        $$3.scheduleTick($$4, (net.minecraft.world.level.block.Block)this, this.getDelayAfterPlace());
+        return super.updateShape($$0, $$1, $$2, $$3, $$4, $$5);
     }
 
     public void tick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.server.level.ServerLevel $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
@@ -28,16 +28,16 @@
         this.falling($$4);
     }
 
-    public void falling(net.minecraft.world.entity.item.FallingBlockEntity entity) {
+    public void falling(net.minecraft.world.entity.item.FallingBlockEntity $$0) {
     }
 
     public int getDelayAfterPlace() {
         return 2;
     }
 
-    public static boolean isFree(net.minecraft.world.level.block.state.BlockState state) {
-        net.minecraft.world.level.material.Material material = state.getMaterial();
-        return state.isAir() || state.is(net.minecraft.tags.BlockTags.FIRE) || material.isLiquid() || material.isReplaceable();
+    public static boolean isFree(net.minecraft.world.level.block.state.BlockState $$0) {
+        net.minecraft.world.level.material.Material $$1 = $$0.getMaterial();
+        return $$0.isAir() || $$0.is(net.minecraft.tags.BlockTags.FIRE) || $$1.isLiquid() || $$0.canBeReplaced();
     }
 
     public void animateTick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
@@ -50,7 +50,7 @@
         }
     }
 
-    public int getDustColor(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
+    public int getDustColor(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2) {
         return -16777216;
     }
 }