--- net/minecraft/world/level/block/FireBlock.java
+++ net/minecraft/world/level/block/FireBlock.java
@@ -39,59 +39,59 @@
         this.shapesCache = com.google.common.collect.ImmutableMap.copyOf((java.util.Map)((java.util.Map)this.stateDefinition.getPossibleStates().stream().filter($$0 -> (java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) == 0).collect(java.util.stream.Collectors.toMap((java.util.function.Function)java.util.function.Function.identity(), net.minecraft.world.level.block.FireBlock::calculateShape))));
     }
 
-    public static net.minecraft.world.phys.shapes.VoxelShape calculateShape(net.minecraft.world.level.block.state.BlockState state) {
-        net.minecraft.world.phys.shapes.VoxelShape voxelShape = net.minecraft.world.phys.shapes.Shapes.empty();
-        if (((java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)UP)).booleanValue()) {
-            voxelShape = UP_AABB;
+    public static net.minecraft.world.phys.shapes.VoxelShape calculateShape(net.minecraft.world.level.block.state.BlockState $$0) {
+        net.minecraft.world.phys.shapes.VoxelShape $$1 = net.minecraft.world.phys.shapes.Shapes.empty();
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)UP)).booleanValue()) {
+            $$1 = UP_AABB;
         }
-        if (((java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)NORTH)).booleanValue()) {
-            voxelShape = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)voxelShape, (net.minecraft.world.phys.shapes.VoxelShape)NORTH_AABB);
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)NORTH)).booleanValue()) {
+            $$1 = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)$$1, (net.minecraft.world.phys.shapes.VoxelShape)NORTH_AABB);
         }
-        if (((java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)SOUTH)).booleanValue()) {
-            voxelShape = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)voxelShape, (net.minecraft.world.phys.shapes.VoxelShape)SOUTH_AABB);
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SOUTH)).booleanValue()) {
+            $$1 = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)$$1, (net.minecraft.world.phys.shapes.VoxelShape)SOUTH_AABB);
         }
-        if (((java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)EAST)).booleanValue()) {
-            voxelShape = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)voxelShape, (net.minecraft.world.phys.shapes.VoxelShape)EAST_AABB);
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)EAST)).booleanValue()) {
+            $$1 = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)$$1, (net.minecraft.world.phys.shapes.VoxelShape)EAST_AABB);
         }
-        if (((java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)WEST)).booleanValue()) {
-            voxelShape = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)voxelShape, (net.minecraft.world.phys.shapes.VoxelShape)WEST_AABB);
+        if (((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)WEST)).booleanValue()) {
+            $$1 = net.minecraft.world.phys.shapes.Shapes.or((net.minecraft.world.phys.shapes.VoxelShape)$$1, (net.minecraft.world.phys.shapes.VoxelShape)WEST_AABB);
         }
-        return voxelShape.isEmpty() ? DOWN_AABB : voxelShape;
+        return $$1.isEmpty() ? DOWN_AABB : $$1;
     }
 
-    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.core.Direction facing, net.minecraft.world.level.block.state.BlockState facingState, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos currentPos, net.minecraft.core.BlockPos facingPos) {
-        if (this.canSurvive(state, (net.minecraft.world.level.LevelReader)level, currentPos)) {
-            return this.getStateWithAge(level, currentPos, ((java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)AGE)).intValue());
+    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.core.Direction $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.LevelAccessor $$3, net.minecraft.core.BlockPos $$4, net.minecraft.core.BlockPos $$5) {
+        if (this.canSurvive($$0, (net.minecraft.world.level.LevelReader)$$3, $$4)) {
+            return this.getStateWithAge($$3, $$4, ((java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)AGE)).intValue());
         }
         return net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
-        return (net.minecraft.world.phys.shapes.VoxelShape)this.shapesCache.get(state.setValue((net.minecraft.world.level.block.state.properties.Property)AGE, (java.lang.Comparable)java.lang.Integer.valueOf((int)0)));
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
+        return (net.minecraft.world.phys.shapes.VoxelShape)this.shapesCache.get($$0.setValue((net.minecraft.world.level.block.state.properties.Property)AGE, (java.lang.Comparable)java.lang.Integer.valueOf((int)0)));
     }
 
-    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext context) {
-        return this.getStateForPlacement((net.minecraft.world.level.BlockGetter)context.getLevel(), context.getClickedPos());
+    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext $$0) {
+        return this.getStateForPlacement((net.minecraft.world.level.BlockGetter)$$0.getLevel(), $$0.getClickedPos());
     }
 
-    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.core.BlockPos blockPos = pos.below();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(blockPos);
-        if (this.canBurn(blockState) || blockState.isFaceSturdy(level, blockPos, net.minecraft.core.Direction.UP)) {
+    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1) {
+        net.minecraft.core.BlockPos $$2 = $$1.below();
+        net.minecraft.world.level.block.state.BlockState $$3 = $$0.getBlockState($$2);
+        if (this.canBurn($$3) || $$3.isFaceSturdy($$0, $$2, net.minecraft.core.Direction.UP)) {
             return this.defaultBlockState();
         }
-        net.minecraft.world.level.block.state.BlockState blockState2 = this.defaultBlockState();
-        for (net.minecraft.core.Direction direction : net.minecraft.core.Direction.values()) {
-            net.minecraft.world.level.block.state.properties.BooleanProperty booleanProperty = (net.minecraft.world.level.block.state.properties.BooleanProperty)PROPERTY_BY_DIRECTION.get((java.lang.Object)direction);
-            if (booleanProperty == null) continue;
-            blockState2 = (net.minecraft.world.level.block.state.BlockState)blockState2.setValue((net.minecraft.world.level.block.state.properties.Property)booleanProperty, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)this.canBurn(level.getBlockState(pos.relative(direction)))));
+        net.minecraft.world.level.block.state.BlockState $$4 = this.defaultBlockState();
+        for (net.minecraft.core.Direction $$5 : net.minecraft.core.Direction.values()) {
+            net.minecraft.world.level.block.state.properties.BooleanProperty $$6 = (net.minecraft.world.level.block.state.properties.BooleanProperty)PROPERTY_BY_DIRECTION.get((java.lang.Object)$$5);
+            if ($$6 == null) continue;
+            $$4 = (net.minecraft.world.level.block.state.BlockState)$$4.setValue((net.minecraft.world.level.block.state.properties.Property)$$6, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)this.canBurn($$0.getBlockState($$1.relative($$5)))));
         }
-        return blockState2;
+        return $$4;
     }
 
-    public boolean canSurvive(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.core.BlockPos blockPos = pos.below();
-        return level.getBlockState(blockPos).isFaceSturdy((net.minecraft.world.level.BlockGetter)level, blockPos, net.minecraft.core.Direction.UP) || this.isValidFireLocation((net.minecraft.world.level.BlockGetter)level, pos);
+    public boolean canSurvive(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.LevelReader $$1, net.minecraft.core.BlockPos $$2) {
+        net.minecraft.core.BlockPos $$3 = $$2.below();
+        return $$1.getBlockState($$3).isFaceSturdy((net.minecraft.world.level.BlockGetter)$$1, $$3, net.minecraft.core.Direction.UP) || this.isValidFireLocation((net.minecraft.world.level.BlockGetter)$$1, $$2);
     }
 
     public void tick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.server.level.ServerLevel $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
@@ -159,8 +159,8 @@
         }
     }
 
-    public boolean isNearRain(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        return level.isRainingAt(pos) || level.isRainingAt(pos.west()) || level.isRainingAt(pos.east()) || level.isRainingAt(pos.north()) || level.isRainingAt(pos.south());
+    public boolean isNearRain(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1) {
+        return $$0.isRainingAt($$1) || $$0.isRainingAt($$1.west()) || $$0.isRainingAt($$1.east()) || $$0.isRainingAt($$1.north()) || $$0.isRainingAt($$1.south());
     }
 
     public int getBurnOdds(net.minecraft.world.level.block.state.BlockState $$0) {
@@ -194,17 +194,17 @@
         }
     }
 
-    public net.minecraft.world.level.block.state.BlockState getStateWithAge(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos, int age) {
-        net.minecraft.world.level.block.state.BlockState blockState = net.minecraft.world.level.block.FireBlock.getState((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)pos);
-        if (blockState.is(net.minecraft.world.level.block.Blocks.FIRE)) {
-            return (net.minecraft.world.level.block.state.BlockState)blockState.setValue((net.minecraft.world.level.block.state.properties.Property)AGE, (java.lang.Comparable)java.lang.Integer.valueOf((int)age));
+    public net.minecraft.world.level.block.state.BlockState getStateWithAge(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos $$1, int $$2) {
+        net.minecraft.world.level.block.state.BlockState $$3 = net.minecraft.world.level.block.FireBlock.getState((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$1);
+        if ($$3.is(net.minecraft.world.level.block.Blocks.FIRE)) {
+            return (net.minecraft.world.level.block.state.BlockState)$$3.setValue((net.minecraft.world.level.block.state.properties.Property)AGE, (java.lang.Comparable)java.lang.Integer.valueOf((int)$$2));
         }
-        return blockState;
+        return $$3;
     }
 
-    public boolean isValidFireLocation(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        for (net.minecraft.core.Direction direction : net.minecraft.core.Direction.values()) {
-            if (!this.canBurn(level.getBlockState(pos.relative(direction)))) continue;
+    public boolean isValidFireLocation(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1) {
+        for (net.minecraft.core.Direction $$2 : net.minecraft.core.Direction.values()) {
+            if (!this.canBurn($$0.getBlockState($$1.relative($$2)))) continue;
             return true;
         }
         return false;
@@ -222,26 +222,26 @@
         return $$2;
     }
 
-    public boolean canBurn(net.minecraft.world.level.block.state.BlockState state) {
-        return this.getIgniteOdds(state) > 0;
+    public boolean canBurn(net.minecraft.world.level.block.state.BlockState $$0) {
+        return this.getIgniteOdds($$0) > 0;
     }
 
-    public void onPlace(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState oldState, boolean isMoving) {
-        super.onPlace(state, level, pos, oldState, isMoving);
-        level.scheduleTick(pos, (net.minecraft.world.level.block.Block)this, net.minecraft.world.level.block.FireBlock.getFireTickDelay((net.minecraft.util.RandomSource)level.random));
+    public void onPlace(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, boolean $$4) {
+        super.onPlace($$0, $$1, $$2, $$3, $$4);
+        $$1.scheduleTick($$2, (net.minecraft.world.level.block.Block)this, net.minecraft.world.level.block.FireBlock.getFireTickDelay((net.minecraft.util.RandomSource)$$1.random));
     }
 
     public static int getFireTickDelay(net.minecraft.util.RandomSource $$0) {
         return 30 + $$0.nextInt(10);
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{AGE, NORTH, EAST, SOUTH, WEST, UP});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{AGE, NORTH, EAST, SOUTH, WEST, UP});
     }
 
-    public void setFlammable(net.minecraft.world.level.block.Block block, int encouragement, int flammability) {
-        this.igniteOdds.put((java.lang.Object)block, encouragement);
-        this.burnOdds.put((java.lang.Object)block, flammability);
+    public void setFlammable(net.minecraft.world.level.block.Block $$0, int $$1, int $$2) {
+        this.igniteOdds.put((java.lang.Object)$$0, $$1);
+        this.burnOdds.put((java.lang.Object)$$0, $$2);
     }
 
     public static void bootStrap() {
@@ -253,6 +253,8 @@
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.ACACIA_PLANKS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.DARK_OAK_PLANKS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.MANGROVE_PLANKS, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_PLANKS, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.OAK_SLAB, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.SPRUCE_SLAB, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.BIRCH_SLAB, 5, 20);
@@ -260,6 +262,8 @@
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.ACACIA_SLAB, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.DARK_OAK_SLAB, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.MANGROVE_SLAB, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_SLAB, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC_STAIRS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.OAK_FENCE_GATE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.SPRUCE_FENCE_GATE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.BIRCH_FENCE_GATE, 5, 20);
@@ -267,6 +271,7 @@
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.ACACIA_FENCE_GATE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.DARK_OAK_FENCE_GATE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.MANGROVE_FENCE_GATE, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_FENCE_GATE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.OAK_FENCE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.SPRUCE_FENCE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.BIRCH_FENCE, 5, 20);
@@ -274,6 +279,7 @@
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.ACACIA_FENCE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.DARK_OAK_FENCE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.MANGROVE_FENCE, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_FENCE, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.OAK_STAIRS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.BIRCH_STAIRS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.SPRUCE_STAIRS, 5, 20);
@@ -281,6 +287,8 @@
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.ACACIA_STAIRS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.DARK_OAK_STAIRS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.MANGROVE_STAIRS, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_STAIRS, 5, 20);
+        $$0.setFlammable(net.minecraft.world.level.block.Blocks.BAMBOO_MOSAIC_STAIRS, 5, 20);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.OAK_LOG, 5, 5);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.SPRUCE_LOG, 5, 5);
         $$0.setFlammable(net.minecraft.world.level.block.Blocks.BIRCH_LOG, 5, 5);