--- net/minecraft/world/level/block/FlowerBlock.java
+++ net/minecraft/world/level/block/FlowerBlock.java
@@ -11,15 +11,15 @@
     public net.minecraft.world.effect.MobEffect suspiciousStewEffect;
     public int effectDuration;
 
-    public net.minecraft.world.level.block.FlowerBlock(net.minecraft.world.effect.MobEffect suspiciousStewEffect, int effectDuration, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
-        this.suspiciousStewEffect = suspiciousStewEffect;
-        this.effectDuration = suspiciousStewEffect.isInstantenous() ? effectDuration : effectDuration * 20;
+    public net.minecraft.world.level.block.FlowerBlock(net.minecraft.world.effect.MobEffect $$0, int $$1, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$2) {
+        super($$2);
+        this.suspiciousStewEffect = $$0;
+        this.effectDuration = $$0.isInstantenous() ? $$1 : $$1 * 20;
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
-        net.minecraft.world.phys.Vec3 vec3 = state.getOffset(level, pos);
-        return SHAPE.move(vec3.x, vec3.y, vec3.z);
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
+        net.minecraft.world.phys.Vec3 $$4 = $$0.getOffset($$1, $$2);
+        return SHAPE.move($$4.x, $$4.y, $$4.z);
     }
 
     public net.minecraft.world.effect.MobEffect getSuspiciousStewEffect() {