--- net/minecraft/world/level/block/FlowerPotBlock.java
+++ net/minecraft/world/level/block/FlowerPotBlock.java
@@ -11,51 +11,51 @@
     public static net.minecraft.world.phys.shapes.VoxelShape SHAPE;
     public net.minecraft.world.level.block.Block content;
 
-    public net.minecraft.world.level.block.FlowerPotBlock(net.minecraft.world.level.block.Block content, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
-        this.content = content;
-        POTTED_BY_CONTENT.put((java.lang.Object)content, (java.lang.Object)this);
+    public net.minecraft.world.level.block.FlowerPotBlock(net.minecraft.world.level.block.Block $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$1);
+        this.content = $$0;
+        POTTED_BY_CONTENT.put((java.lang.Object)$$0, (java.lang.Object)this);
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
         return SHAPE;
     }
 
-    public net.minecraft.world.level.block.RenderShape getRenderShape(net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.level.block.RenderShape getRenderShape(net.minecraft.world.level.block.state.BlockState $$0) {
         return net.minecraft.world.level.block.RenderShape.MODEL;
     }
 
-    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand, net.minecraft.world.phys.BlockHitResult hit) {
-        boolean bl;
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        net.minecraft.world.item.Item item = itemStack.getItem();
-        net.minecraft.world.level.block.state.BlockState blockState = (item instanceof net.minecraft.world.item.BlockItem ? (net.minecraft.world.level.block.Block)POTTED_BY_CONTENT.getOrDefault((java.lang.Object)((net.minecraft.world.item.BlockItem)item).getBlock(), (java.lang.Object)net.minecraft.world.level.block.Blocks.AIR) : net.minecraft.world.level.block.Blocks.AIR).defaultBlockState();
-        boolean bl2 = blockState.is(net.minecraft.world.level.block.Blocks.AIR);
-        if (bl2 != (bl = this.isEmpty())) {
-            if (bl) {
-                level.setBlock(pos, blockState, 3);
-                player.awardStat(net.minecraft.stats.Stats.POT_FLOWER);
-                if (!player.getAbilities().instabuild) {
-                    itemStack.shrink(1);
+    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.player.Player $$3, net.minecraft.world.InteractionHand $$4, net.minecraft.world.phys.BlockHitResult $$5) {
+        boolean $$10;
+        net.minecraft.world.item.ItemStack $$6 = $$3.getItemInHand($$4);
+        net.minecraft.world.item.Item $$7 = $$6.getItem();
+        net.minecraft.world.level.block.state.BlockState $$8 = ($$7 instanceof net.minecraft.world.item.BlockItem ? (net.minecraft.world.level.block.Block)POTTED_BY_CONTENT.getOrDefault((java.lang.Object)((net.minecraft.world.item.BlockItem)$$7).getBlock(), (java.lang.Object)net.minecraft.world.level.block.Blocks.AIR) : net.minecraft.world.level.block.Blocks.AIR).defaultBlockState();
+        boolean $$9 = $$8.is(net.minecraft.world.level.block.Blocks.AIR);
+        if ($$9 != ($$10 = this.isEmpty())) {
+            if ($$10) {
+                $$1.setBlock($$2, $$8, 3);
+                $$3.awardStat(net.minecraft.stats.Stats.POT_FLOWER);
+                if (!$$3.getAbilities().instabuild) {
+                    $$6.shrink(1);
                 }
             } else {
-                net.minecraft.world.item.ItemStack itemStack2 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)this.content);
-                if (itemStack.isEmpty()) {
-                    player.setItemInHand(hand, itemStack2);
-                } else if (!player.addItem(itemStack2)) {
-                    player.drop(itemStack2, false);
+                net.minecraft.world.item.ItemStack $$11 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)this.content);
+                if ($$6.isEmpty()) {
+                    $$3.setItemInHand($$4, $$11);
+                } else if (!$$3.addItem($$11)) {
+                    $$3.drop($$11, false);
                 }
-                level.setBlock(pos, net.minecraft.world.level.block.Blocks.FLOWER_POT.defaultBlockState(), 3);
+                $$1.setBlock($$2, net.minecraft.world.level.block.Blocks.FLOWER_POT.defaultBlockState(), 3);
             }
-            level.gameEvent((net.minecraft.world.entity.Entity)player, net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, pos);
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+            $$1.gameEvent((net.minecraft.world.entity.Entity)$$3, net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, $$2);
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
         }
         return net.minecraft.world.InteractionResult.CONSUME;
     }
 
-    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2) {
         if (this.isEmpty()) {
-            return super.getCloneItemStack(level, pos, state);
+            return super.getCloneItemStack($$0, $$1, $$2);
         }
         return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)this.content);
     }
@@ -64,18 +64,18 @@
         return this.content == net.minecraft.world.level.block.Blocks.AIR;
     }
 
-    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.core.Direction facing, net.minecraft.world.level.block.state.BlockState facingState, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos currentPos, net.minecraft.core.BlockPos facingPos) {
-        if (facing == net.minecraft.core.Direction.DOWN && !state.canSurvive((net.minecraft.world.level.LevelReader)level, currentPos)) {
+    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.core.Direction $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.LevelAccessor $$3, net.minecraft.core.BlockPos $$4, net.minecraft.core.BlockPos $$5) {
+        if ($$1 == net.minecraft.core.Direction.DOWN && !$$0.canSurvive((net.minecraft.world.level.LevelReader)$$3, $$4)) {
             return net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
         }
-        return super.updateShape(state, facing, facingState, level, currentPos, facingPos);
+        return super.updateShape($$0, $$1, $$2, $$3, $$4, $$5);
     }
 
     public net.minecraft.world.level.block.Block getContent() {
         return this.content;
     }
 
-    public boolean isPathfindable(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.pathfinder.PathComputationType type) {
+    public boolean isPathfindable(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.pathfinder.PathComputationType $$3) {
         return false;
     }
 