--- net/minecraft/world/level/block/FrogspawnBlock.java
+++ net/minecraft/world/level/block/FrogspawnBlock.java
@@ -75,6 +75,7 @@
         int $$3 = $$2.nextInt(2, 6);
         for (int $$4 = 1; $$4 <= $$3; ++$$4) {
             net.minecraft.world.entity.animal.frog.Tadpole $$5 = (net.minecraft.world.entity.animal.frog.Tadpole)net.minecraft.world.entity.EntityType.TADPOLE.create((net.minecraft.world.level.Level)$$0);
+            if ($$5 == null) continue;
             double $$6 = (double)$$1.getX() + this.getRandomTadpolePositionOffset($$2);
             double $$7 = (double)$$1.getZ() + this.getRandomTadpolePositionOffset($$2);
             int $$8 = $$2.nextInt(1, 361);