--- net/minecraft/world/level/block/FungusBlock.java
+++ net/minecraft/world/level/block/FungusBlock.java
@@ -16,18 +16,18 @@
         this.feature = $$1;
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
         return SHAPE;
     }
 
-    public boolean mayPlaceOn(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        return state.is(net.minecraft.tags.BlockTags.NYLIUM) || state.is(net.minecraft.world.level.block.Blocks.MYCELIUM) || state.is(net.minecraft.world.level.block.Blocks.SOUL_SOIL) || super.mayPlaceOn(state, level, pos);
+    public boolean mayPlaceOn(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2) {
+        return $$0.is(net.minecraft.tags.BlockTags.NYLIUM) || $$0.is(net.minecraft.world.level.block.Blocks.MYCELIUM) || $$0.is(net.minecraft.world.level.block.Blocks.SOUL_SOIL) || super.mayPlaceOn($$0, $$1, $$2);
     }
 
-    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, boolean isClient) {
-        net.minecraft.world.level.block.Block block = ((net.minecraft.world.level.levelgen.feature.HugeFungusConfiguration)((net.minecraft.world.level.levelgen.feature.ConfiguredFeature)((net.minecraft.core.Holder)this.feature.get()).value()).config()).validBaseState.getBlock();
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(pos.below());
-        return blockState.is(block);
+    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, boolean $$3) {
+        net.minecraft.world.level.block.Block $$4 = ((net.minecraft.world.level.levelgen.feature.HugeFungusConfiguration)((net.minecraft.world.level.levelgen.feature.ConfiguredFeature)((net.minecraft.core.Holder)this.feature.get()).value()).config()).validBaseState.getBlock();
+        net.minecraft.world.level.block.state.BlockState $$5 = $$0.getBlockState($$1.below());
+        return $$5.is($$4);
     }
 
     public boolean isBonemealSuccess(net.minecraft.world.level.Level $$0, net.minecraft.util.RandomSource $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3) {