--- net/minecraft/world/level/block/HalfTransparentBlock.java
+++ net/minecraft/world/level/block/HalfTransparentBlock.java
@@ -10,11 +10,11 @@
         super($$0);
     }
 
-    public boolean skipRendering(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.state.BlockState adjacentBlockState, net.minecraft.core.Direction side) {
-        if (adjacentBlockState.is((net.minecraft.world.level.block.Block)this)) {
+    public boolean skipRendering(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.state.BlockState $$1, net.minecraft.core.Direction $$2) {
+        if ($$1.is((net.minecraft.world.level.block.Block)this)) {
             return true;
         }
-        return super.skipRendering(state, adjacentBlockState, side);
+        return super.skipRendering($$0, $$1, $$2);
     }
 }
 