--- net/minecraft/world/level/block/IceBlock.java
+++ net/minecraft/world/level/block/IceBlock.java
@@ -10,16 +10,16 @@
         super($$0);
     }
 
-    public void playerDestroy(net.minecraft.world.level.Level level, net.minecraft.world.entity.player.Player player, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, @javax.annotation.Nullable net.minecraft.world.level.block.entity.BlockEntity te, net.minecraft.world.item.ItemStack stack) {
-        super.playerDestroy(level, player, pos, state, te, stack);
-        if (net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.SILK_TOUCH, (net.minecraft.world.item.ItemStack)stack) == 0) {
-            if (level.dimensionType().ultraWarm()) {
-                level.removeBlock(pos, false);
+    public void playerDestroy(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.player.Player $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, @javax.annotation.Nullable net.minecraft.world.level.block.entity.BlockEntity $$4, net.minecraft.world.item.ItemStack $$5) {
+        super.playerDestroy($$0, $$1, $$2, $$3, $$4, $$5);
+        if (net.minecraft.world.item.enchantment.EnchantmentHelper.getItemEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.SILK_TOUCH, (net.minecraft.world.item.ItemStack)$$5) == 0) {
+            if ($$0.dimensionType().ultraWarm()) {
+                $$0.removeBlock($$2, false);
                 return;
             }
-            net.minecraft.world.level.material.Material material = level.getBlockState(pos.below()).getMaterial();
-            if (material.blocksMotion() || material.isLiquid()) {
-                level.setBlockAndUpdate(pos, net.minecraft.world.level.block.Blocks.WATER.defaultBlockState());
+            net.minecraft.world.level.material.Material $$6 = $$0.getBlockState($$2.below()).getMaterial();
+            if ($$6.blocksMotion() || $$6.isLiquid()) {
+                $$0.setBlockAndUpdate($$2, net.minecraft.world.level.block.Blocks.WATER.defaultBlockState());
             }
         }
     }
@@ -30,16 +30,16 @@
         }
     }
 
-    public void melt(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        if (level.dimensionType().ultraWarm()) {
-            level.removeBlock(pos, false);
+    public void melt(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2) {
+        if ($$1.dimensionType().ultraWarm()) {
+            $$1.removeBlock($$2, false);
             return;
         }
-        level.setBlockAndUpdate(pos, net.minecraft.world.level.block.Blocks.WATER.defaultBlockState());
-        level.neighborChanged(pos, net.minecraft.world.level.block.Blocks.WATER, pos);
+        $$1.setBlockAndUpdate($$2, net.minecraft.world.level.block.Blocks.WATER.defaultBlockState());
+        $$1.neighborChanged($$2, net.minecraft.world.level.block.Blocks.WATER, $$2);
     }
 
-    public net.minecraft.world.level.material.PushReaction getPistonPushReaction(net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.level.material.PushReaction getPistonPushReaction(net.minecraft.world.level.block.state.BlockState $$0) {
         return net.minecraft.world.level.material.PushReaction.NORMAL;
     }
 }