--- net/minecraft/world/level/block/InfestedBlock.java
+++ net/minecraft/world/level/block/InfestedBlock.java
@@ -11,25 +11,27 @@
     public static java.util.Map<net.minecraft.world.level.block.state.BlockState, net.minecraft.world.level.block.state.BlockState> HOST_TO_INFESTED_STATES;
     public static java.util.Map<net.minecraft.world.level.block.state.BlockState, net.minecraft.world.level.block.state.BlockState> INFESTED_TO_HOST_STATES;
 
-    public net.minecraft.world.level.block.InfestedBlock(net.minecraft.world.level.block.Block hostBlock, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties.destroyTime(hostBlock.defaultDestroyTime() / 2.0f).explosionResistance(0.75f));
-        this.hostBlock = hostBlock;
-        BLOCK_BY_HOST_BLOCK.put((java.lang.Object)hostBlock, (java.lang.Object)this);
+    public net.minecraft.world.level.block.InfestedBlock(net.minecraft.world.level.block.Block $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$1.destroyTime($$0.defaultDestroyTime() / 2.0f).explosionResistance(0.75f));
+        this.hostBlock = $$0;
+        BLOCK_BY_HOST_BLOCK.put((java.lang.Object)$$0, (java.lang.Object)this);
     }
 
     public net.minecraft.world.level.block.Block getHostBlock() {
         return this.hostBlock;
     }
 
-    public static boolean isCompatibleHostBlock(net.minecraft.world.level.block.state.BlockState state) {
-        return BLOCK_BY_HOST_BLOCK.containsKey((java.lang.Object)state.getBlock());
+    public static boolean isCompatibleHostBlock(net.minecraft.world.level.block.state.BlockState $$0) {
+        return BLOCK_BY_HOST_BLOCK.containsKey((java.lang.Object)$$0.getBlock());
     }
 
-    public void spawnInfestation(net.minecraft.server.level.ServerLevel level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.entity.monster.Silverfish silverfish = (net.minecraft.world.entity.monster.Silverfish)net.minecraft.world.entity.EntityType.SILVERFISH.create((net.minecraft.world.level.Level)level);
-        silverfish.moveTo((double)pos.getX() + 0.5, (double)pos.getY(), (double)pos.getZ() + 0.5, 0.0f, 0.0f);
-        level.addFreshEntity((net.minecraft.world.entity.Entity)silverfish);
-        silverfish.spawnAnim();
+    public void spawnInfestation(net.minecraft.server.level.ServerLevel $$0, net.minecraft.core.BlockPos $$1) {
+        net.minecraft.world.entity.monster.Silverfish $$2 = (net.minecraft.world.entity.monster.Silverfish)net.minecraft.world.entity.EntityType.SILVERFISH.create((net.minecraft.world.level.Level)$$0);
+        if ($$2 != null) {
+            $$2.moveTo((double)$$1.getX() + 0.5, (double)$$1.getY(), (double)$$1.getZ() + 0.5, 0.0f, 0.0f);
+            $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$2);
+            $$2.spawnAnim();
+        }
     }
 
     public void spawnAfterBreak(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.server.level.ServerLevel $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.item.ItemStack $$3, boolean $$4) {