--- net/minecraft/world/level/block/KelpPlantBlock.java
+++ net/minecraft/world/level/block/KelpPlantBlock.java
@@ -15,7 +15,7 @@
         return (net.minecraft.world.level.block.GrowingPlantHeadBlock)net.minecraft.world.level.block.Blocks.KELP;
     }
 
-    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.world.level.block.state.BlockState $$0) {
         return net.minecraft.world.level.material.Fluids.WATER.getSource(false);
     }
 
@@ -23,11 +23,11 @@
         return this.getHeadBlock().canAttachTo($$0);
     }
 
-    public boolean canPlaceLiquid(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.material.Fluid fluid) {
+    public boolean canPlaceLiquid(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.material.Fluid $$3) {
         return false;
     }
 
-    public boolean placeLiquid(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.material.FluidState fluidState) {
+    public boolean placeLiquid(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.material.FluidState $$3) {
         return false;
     }
 }