--- net/minecraft/world/level/block/KelpBlock.java
+++ net/minecraft/world/level/block/KelpBlock.java
@@ -14,8 +14,8 @@
         super($$0, net.minecraft.core.Direction.UP, SHAPE, true, 0.14);
     }
 
-    public boolean canGrowInto(net.minecraft.world.level.block.state.BlockState state) {
-        return state.is(net.minecraft.world.level.block.Blocks.WATER);
+    public boolean canGrowInto(net.minecraft.world.level.block.state.BlockState $$0) {
+        return $$0.is(net.minecraft.world.level.block.Blocks.WATER);
     }
 
     public net.minecraft.world.level.block.Block getBodyBlock() {
@@ -26,11 +26,11 @@
         return !$$0.is(net.minecraft.world.level.block.Blocks.MAGMA_BLOCK);
     }
 
-    public boolean canPlaceLiquid(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.material.Fluid fluid) {
+    public boolean canPlaceLiquid(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.material.Fluid $$3) {
         return false;
     }
 
-    public boolean placeLiquid(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.material.FluidState fluidState) {
+    public boolean placeLiquid(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.material.FluidState $$3) {
         return false;
     }
 
@@ -39,15 +39,15 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext context) {
-        net.minecraft.world.level.material.FluidState fluidState = context.getLevel().getFluidState(context.getClickedPos());
-        if (fluidState.is(net.minecraft.tags.FluidTags.WATER) && fluidState.getAmount() == 8) {
-            return super.getStateForPlacement(context);
+    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext $$0) {
+        net.minecraft.world.level.material.FluidState $$1 = $$0.getLevel().getFluidState($$0.getClickedPos());
+        if ($$1.is(net.minecraft.tags.FluidTags.WATER) && $$1.getAmount() == 8) {
+            return super.getStateForPlacement($$0);
         }
         return null;
     }
 
-    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.world.level.block.state.BlockState $$0) {
         return net.minecraft.world.level.material.Fluids.WATER.getSource(false);
     }
 