--- net/minecraft/world/level/block/LightningRodBlock.java
+++ net/minecraft/world/level/block/LightningRodBlock.java
@@ -95,10 +95,12 @@
         net.minecraft.core.BlockPos $$4;
         if ($$0.isThundering() && $$3 instanceof net.minecraft.world.entity.projectile.ThrownTrident && ((net.minecraft.world.entity.projectile.ThrownTrident)$$3).isChanneling() && $$0.canSeeSky($$4 = $$2.getBlockPos())) {
             net.minecraft.world.entity.LightningBolt $$5 = (net.minecraft.world.entity.LightningBolt)net.minecraft.world.entity.EntityType.LIGHTNING_BOLT.create($$0);
-            $$5.moveTo(net.minecraft.world.phys.Vec3.atBottomCenterOf((net.minecraft.core.Vec3i)$$4.above()));
-            net.minecraft.world.entity.Entity $$6 = $$3.getOwner();
-            $$5.setCause($$6 instanceof net.minecraft.server.level.ServerPlayer ? (net.minecraft.server.level.ServerPlayer)$$6 : null);
-            $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$5);
+            if ($$5 != null) {
+                $$5.moveTo(net.minecraft.world.phys.Vec3.atBottomCenterOf((net.minecraft.core.Vec3i)$$4.above()));
+                net.minecraft.world.entity.Entity $$6 = $$3.getOwner();
+                $$5.setCause($$6 instanceof net.minecraft.server.level.ServerPlayer ? (net.minecraft.server.level.ServerPlayer)$$6 : null);
+                $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$5);
+            }
             $$0.playSound(null, $$4, net.minecraft.sounds.SoundEvents.TRIDENT_THUNDER, net.minecraft.sounds.SoundSource.WEATHER, 5.0f, 1.0f);
         }
     }