--- net/minecraft/world/level/block/Mirror.java
+++ net/minecraft/world/level/block/Mirror.java
@@ -31,33 +31,33 @@
         this.rotation = $$1;
     }
 
-    public int mirror(int rotation, int rotationCount) {
-        int n = rotationCount / 2;
-        int n2 = rotation > n ? rotation - rotationCount : rotation;
+    public int mirror(int $$0, int $$1) {
+        int $$2 = $$1 / 2;
+        int $$3 = $$0 > $$2 ? $$0 - $$1 : $$0;
         switch (net.minecraft.world.level.block.Mirror$1.$SwitchMap$net$minecraft$world$level$block$Mirror[this.ordinal()]) {
             case 1: {
-                return (rotationCount - n2) % rotationCount;
+                return ($$1 - $$3) % $$1;
             }
             case 2: {
-                return (n - n2 + rotationCount) % rotationCount;
+                return ($$2 - $$3 + $$1) % $$1;
             }
         }
-        return rotation;
+        return $$0;
     }
 
-    public net.minecraft.world.level.block.Rotation getRotation(net.minecraft.core.Direction facing) {
-        net.minecraft.core.Direction$Axis axis = facing.getAxis();
-        return this == LEFT_RIGHT && axis == net.minecraft.core.Direction$Axis.Z || this == FRONT_BACK && axis == net.minecraft.core.Direction$Axis.X ? net.minecraft.world.level.block.Rotation.CLOCKWISE_180 : net.minecraft.world.level.block.Rotation.NONE;
+    public net.minecraft.world.level.block.Rotation getRotation(net.minecraft.core.Direction $$0) {
+        net.minecraft.core.Direction$Axis $$1 = $$0.getAxis();
+        return this == LEFT_RIGHT && $$1 == net.minecraft.core.Direction$Axis.Z || this == FRONT_BACK && $$1 == net.minecraft.core.Direction$Axis.X ? net.minecraft.world.level.block.Rotation.CLOCKWISE_180 : net.minecraft.world.level.block.Rotation.NONE;
     }
 
-    public net.minecraft.core.Direction mirror(net.minecraft.core.Direction facing) {
-        if (this == FRONT_BACK && facing.getAxis() == net.minecraft.core.Direction$Axis.X) {
-            return facing.getOpposite();
+    public net.minecraft.core.Direction mirror(net.minecraft.core.Direction $$0) {
+        if (this == FRONT_BACK && $$0.getAxis() == net.minecraft.core.Direction$Axis.X) {
+            return $$0.getOpposite();
         }
-        if (this == LEFT_RIGHT && facing.getAxis() == net.minecraft.core.Direction$Axis.Z) {
-            return facing.getOpposite();
+        if (this == LEFT_RIGHT && $$0.getAxis() == net.minecraft.core.Direction$Axis.Z) {
+            return $$0.getOpposite();
         }
-        return facing;
+        return $$0;
     }
 
     public com.mojang.math.OctahedralGroup rotation() {