--- net/minecraft/world/level/block/NetherrackBlock.java
+++ net/minecraft/world/level/block/NetherrackBlock.java
@@ -11,12 +11,12 @@
         super($$0);
     }
 
-    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, boolean isClient) {
-        if (!level.getBlockState(pos.above()).propagatesSkylightDown(level, pos)) {
+    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, boolean $$3) {
+        if (!$$0.getBlockState($$1.above()).propagatesSkylightDown($$0, $$1)) {
             return false;
         }
-        for (net.minecraft.core.BlockPos blockPos : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)pos.offset(-1, -1, -1), (net.minecraft.core.BlockPos)pos.offset(1, 1, 1))) {
-            if (!level.getBlockState(blockPos).is(net.minecraft.tags.BlockTags.NYLIUM)) continue;
+        for (net.minecraft.core.BlockPos $$4 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)$$1.offset(-1, -1, -1), (net.minecraft.core.BlockPos)$$1.offset(1, 1, 1))) {
+            if (!$$0.getBlockState($$4).is(net.minecraft.tags.BlockTags.NYLIUM)) continue;
             return true;
         }
         return false;