--- net/minecraft/world/level/block/PlayerHeadBlock.java
+++ net/minecraft/world/level/block/PlayerHeadBlock.java
@@ -10,21 +10,21 @@
         super((net.minecraft.world.level.block.SkullBlock$Type)net.minecraft.world.level.block.SkullBlock$Types.PLAYER, $$0);
     }
 
-    public void setPlacedBy(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity placer, net.minecraft.world.item.ItemStack stack) {
-        super.setPlacedBy(level, pos, state, placer, stack);
-        net.minecraft.world.level.block.entity.BlockEntity blockEntity = level.getBlockEntity(pos);
-        if (blockEntity instanceof net.minecraft.world.level.block.entity.SkullBlockEntity) {
-            net.minecraft.world.level.block.entity.SkullBlockEntity skullBlockEntity = (net.minecraft.world.level.block.entity.SkullBlockEntity)blockEntity;
-            com.mojang.authlib.GameProfile gameProfile = null;
-            if (stack.hasTag()) {
-                net.minecraft.nbt.CompoundTag compoundTag = stack.getTag();
-                if (compoundTag.contains("SkullOwner", 10)) {
-                    gameProfile = net.minecraft.nbt.NbtUtils.readGameProfile((net.minecraft.nbt.CompoundTag)compoundTag.getCompound("SkullOwner"));
-                } else if (compoundTag.contains("SkullOwner", 8) && !org.apache.commons.lang3.StringUtils.isBlank((java.lang.CharSequence)compoundTag.getString("SkullOwner"))) {
-                    gameProfile = new com.mojang.authlib.GameProfile(null, compoundTag.getString("SkullOwner"));
+    public void setPlacedBy(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$3, net.minecraft.world.item.ItemStack $$4) {
+        super.setPlacedBy($$0, $$1, $$2, $$3, $$4);
+        net.minecraft.world.level.block.entity.BlockEntity $$5 = $$0.getBlockEntity($$1);
+        if ($$5 instanceof net.minecraft.world.level.block.entity.SkullBlockEntity) {
+            net.minecraft.world.level.block.entity.SkullBlockEntity $$6 = (net.minecraft.world.level.block.entity.SkullBlockEntity)$$5;
+            com.mojang.authlib.GameProfile $$7 = null;
+            if ($$4.hasTag()) {
+                net.minecraft.nbt.CompoundTag $$8 = $$4.getTag();
+                if ($$8.contains("SkullOwner", 10)) {
+                    $$7 = net.minecraft.nbt.NbtUtils.readGameProfile((net.minecraft.nbt.CompoundTag)$$8.getCompound("SkullOwner"));
+                } else if ($$8.contains("SkullOwner", 8) && !org.apache.commons.lang3.StringUtils.isBlank((java.lang.CharSequence)$$8.getString("SkullOwner"))) {
+                    $$7 = new com.mojang.authlib.GameProfile(null, $$8.getString("SkullOwner"));
                 }
             }
-            skullBlockEntity.setOwner(gameProfile);
+            $$6.setOwner($$7);
         }
     }
 }