--- net/minecraft/world/level/block/PlayerWallHeadBlock.java
+++ net/minecraft/world/level/block/PlayerWallHeadBlock.java
@@ -10,12 +10,12 @@
         super((net.minecraft.world.level.block.SkullBlock$Type)net.minecraft.world.level.block.SkullBlock$Types.PLAYER, $$0);
     }
 
-    public void setPlacedBy(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity placer, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.world.level.block.Blocks.PLAYER_HEAD.setPlacedBy(level, pos, state, placer, stack);
+    public void setPlacedBy(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$3, net.minecraft.world.item.ItemStack $$4) {
+        net.minecraft.world.level.block.Blocks.PLAYER_HEAD.setPlacedBy($$0, $$1, $$2, $$3, $$4);
     }
 
-    public java.util.List<net.minecraft.world.item.ItemStack> getDrops(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.storage.loot.LootContext$Builder builder) {
-        return net.minecraft.world.level.block.Blocks.PLAYER_HEAD.getDrops(state, builder);
+    public java.util.List<net.minecraft.world.item.ItemStack> getDrops(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.storage.loot.LootContext$Builder $$1) {
+        return net.minecraft.world.level.block.Blocks.PLAYER_HEAD.getDrops($$0, $$1);
     }
 }
 