--- net/minecraft/world/level/block/PotatoBlock.java
+++ net/minecraft/world/level/block/PotatoBlock.java
@@ -16,8 +16,8 @@
         return net.minecraft.world.item.Items.POTATO;
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
-        return SHAPE_BY_AGE[(java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)this.getAgeProperty())];
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
+        return SHAPE_BY_AGE[(java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)this.getAgeProperty())];
     }
 
     static {