--- net/minecraft/world/level/block/PoweredRailBlock.java
+++ net/minecraft/world/level/block/PoweredRailBlock.java
@@ -14,112 +14,112 @@
         this.registerDefaultState((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)this.stateDefinition.any()).setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH)).setValue((net.minecraft.world.level.block.state.properties.Property)POWERED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)false))).setValue((net.minecraft.world.level.block.state.properties.Property)WATERLOGGED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)false)));
     }
 
-    public boolean findPoweredRailSignal(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, boolean searchForward, int recursionCount) {
-        if (recursionCount >= 8) {
+    public boolean findPoweredRailSignal(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, boolean $$3, int $$4) {
+        if ($$4 >= 8) {
             return false;
         }
-        int n = pos.getX();
-        int n2 = pos.getY();
-        int n3 = pos.getZ();
-        boolean bl = true;
-        net.minecraft.world.level.block.state.properties.RailShape railShape = (net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
-        switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[railShape.ordinal()]) {
+        int $$5 = $$1.getX();
+        int $$6 = $$1.getY();
+        int $$7 = $$1.getZ();
+        boolean $$8 = true;
+        net.minecraft.world.level.block.state.properties.RailShape $$9 = (net.minecraft.world.level.block.state.properties.RailShape)$$2.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
+        switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[$$9.ordinal()]) {
             case 1: {
-                if (searchForward) {
-                    ++n3;
+                if ($$3) {
+                    ++$$7;
                     break;
                 }
-                --n3;
+                --$$7;
                 break;
             }
             case 2: {
-                if (searchForward) {
-                    --n;
+                if ($$3) {
+                    --$$5;
                     break;
                 }
-                ++n;
+                ++$$5;
                 break;
             }
             case 3: {
-                if (searchForward) {
-                    --n;
+                if ($$3) {
+                    --$$5;
                 } else {
-                    ++n;
-                    ++n2;
-                    bl = false;
+                    ++$$5;
+                    ++$$6;
+                    $$8 = false;
                 }
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
                 break;
             }
             case 4: {
-                if (searchForward) {
-                    --n;
-                    ++n2;
-                    bl = false;
+                if ($$3) {
+                    --$$5;
+                    ++$$6;
+                    $$8 = false;
                 } else {
-                    ++n;
+                    ++$$5;
                 }
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
                 break;
             }
             case 5: {
-                if (searchForward) {
-                    ++n3;
+                if ($$3) {
+                    ++$$7;
                 } else {
-                    --n3;
-                    ++n2;
-                    bl = false;
+                    --$$7;
+                    ++$$6;
+                    $$8 = false;
                 }
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
                 break;
             }
             case 6: {
-                if (searchForward) {
-                    ++n3;
-                    ++n2;
-                    bl = false;
+                if ($$3) {
+                    ++$$7;
+                    ++$$6;
+                    $$8 = false;
                 } else {
-                    --n3;
+                    --$$7;
                 }
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
             }
         }
-        if (this.isSameRailWithPower(level, new net.minecraft.core.BlockPos(n, n2, n3), searchForward, recursionCount, railShape)) {
+        if (this.isSameRailWithPower($$0, new net.minecraft.core.BlockPos($$5, $$6, $$7), $$3, $$4, $$9)) {
             return true;
         }
-        return bl && this.isSameRailWithPower(level, new net.minecraft.core.BlockPos(n, n2 - 1, n3), searchForward, recursionCount, railShape);
+        return $$8 && this.isSameRailWithPower($$0, new net.minecraft.core.BlockPos($$5, $$6 - 1, $$7), $$3, $$4, $$9);
     }
 
-    public boolean isSameRailWithPower(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos state, boolean searchForward, int recursionCount, net.minecraft.world.level.block.state.properties.RailShape shape) {
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(state);
-        if (!blockState.is((net.minecraft.world.level.block.Block)this)) {
+    public boolean isSameRailWithPower(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, boolean $$2, int $$3, net.minecraft.world.level.block.state.properties.RailShape $$4) {
+        net.minecraft.world.level.block.state.BlockState $$5 = $$0.getBlockState($$1);
+        if (!$$5.is((net.minecraft.world.level.block.Block)this)) {
             return false;
         }
-        net.minecraft.world.level.block.state.properties.RailShape railShape = (net.minecraft.world.level.block.state.properties.RailShape)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
-        if (shape == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST && (railShape == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH || railShape == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH || railShape == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH)) {
+        net.minecraft.world.level.block.state.properties.RailShape $$6 = (net.minecraft.world.level.block.state.properties.RailShape)$$5.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
+        if ($$4 == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST && ($$6 == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH || $$6 == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH || $$6 == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH)) {
             return false;
         }
-        if (shape == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH && (railShape == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST || railShape == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST || railShape == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST)) {
+        if ($$4 == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH && ($$6 == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST || $$6 == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST || $$6 == net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST)) {
             return false;
         }
-        if (((java.lang.Boolean)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue()) {
-            if (level.hasNeighborSignal(state)) {
+        if (((java.lang.Boolean)$$5.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED)).booleanValue()) {
+            if ($$0.hasNeighborSignal($$1)) {
                 return true;
             }
-            return this.findPoweredRailSignal(level, state, blockState, searchForward, recursionCount + 1);
+            return this.findPoweredRailSignal($$0, $$1, $$5, $$2, $$3 + 1);
         }
         return false;
     }
 
-    public void updateState(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.Block block) {
-        boolean bl;
-        boolean bl2 = (java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED);
-        boolean bl3 = bl = level.hasNeighborSignal(pos) || this.findPoweredRailSignal(level, pos, state, true, 0) || this.findPoweredRailSignal(level, pos, state, false, 0);
-        if (bl != bl2) {
-            level.setBlock(pos, (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)POWERED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)bl)), 3);
-            level.updateNeighborsAt(pos.below(), (net.minecraft.world.level.block.Block)this);
-            if (((net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).isAscending()) {
-                level.updateNeighborsAt(pos.above(), (net.minecraft.world.level.block.Block)this);
+    public void updateState(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.Block $$3) {
+        boolean $$5;
+        boolean $$4 = (java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWERED);
+        boolean bl = $$5 = $$1.hasNeighborSignal($$2) || this.findPoweredRailSignal($$1, $$2, $$0, true, 0) || this.findPoweredRailSignal($$1, $$2, $$0, false, 0);
+        if ($$5 != $$4) {
+            $$1.setBlock($$2, (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)POWERED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)$$5)), 3);
+            $$1.updateNeighborsAt($$2.below(), (net.minecraft.world.level.block.Block)this);
+            if (((net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).isAscending()) {
+                $$1.updateNeighborsAt($$2.above(), (net.minecraft.world.level.block.Block)this);
             }
         }
     }
@@ -128,163 +128,163 @@
         return SHAPE;
     }
 
-    public net.minecraft.world.level.block.state.BlockState rotate(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.Rotation rot) {
-        switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$Rotation[rot.ordinal()]) {
+    public net.minecraft.world.level.block.state.BlockState rotate(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.Rotation $$1) {
+        switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$Rotation[$$1.ordinal()]) {
             case 1: {
-                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
+                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 9: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                     case 10: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                 }
             }
             case 2: {
-                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
+                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
                     case 1: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
                     }
                     case 2: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
                     }
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                     case 9: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 10: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                 }
             }
             case 3: {
-                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
+                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
                     case 1: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
                     }
                     case 2: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
                     }
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                     case 9: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 10: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                 }
             }
         }
-        return state;
+        return $$0;
     }
 
-    public net.minecraft.world.level.block.state.BlockState mirror(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.Mirror mirror) {
-        net.minecraft.world.level.block.state.properties.RailShape railShape = (net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
-        switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$Mirror[mirror.ordinal()]) {
+    public net.minecraft.world.level.block.state.BlockState mirror(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.Mirror $$1) {
+        net.minecraft.world.level.block.state.properties.RailShape $$2 = (net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
+        switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$Mirror[$$1.ordinal()]) {
             case 1: {
-                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[railShape.ordinal()]) {
+                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[$$2.ordinal()]) {
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                     case 9: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 10: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                 }
                 break;
             }
             case 2: {
-                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[railShape.ordinal()]) {
+                switch (net.minecraft.world.level.block.PoweredRailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[$$2.ordinal()]) {
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                     case 9: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 10: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                 }
                 break;
             }
         }
-        return super.mirror(state, mirror);
+        return super.mirror($$0, $$1);
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{SHAPE, POWERED, WATERLOGGED});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{SHAPE, POWERED, WATERLOGGED});
     }
 
     static {