--- net/minecraft/world/level/block/PumpkinBlock.java
+++ net/minecraft/world/level/block/PumpkinBlock.java
@@ -10,24 +10,24 @@
         super($$0);
     }
 
-    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand, net.minecraft.world.phys.BlockHitResult hit) {
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        if (itemStack.is(net.minecraft.world.item.Items.SHEARS)) {
-            if (!level.isClientSide) {
-                net.minecraft.core.Direction direction = hit.getDirection();
-                net.minecraft.core.Direction direction2 = direction.getAxis() == net.minecraft.core.Direction$Axis.Y ? player.getDirection().getOpposite() : direction;
-                level.playSound(null, pos, net.minecraft.sounds.SoundEvents.PUMPKIN_CARVE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
-                level.setBlock(pos, (net.minecraft.world.level.block.state.BlockState)net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.CarvedPumpkinBlock.FACING, (java.lang.Comparable)direction2), 11);
-                net.minecraft.world.entity.item.ItemEntity itemEntity = new net.minecraft.world.entity.item.ItemEntity(level, (double)pos.getX() + 0.5 + (double)direction2.getStepX() * 0.65, (double)pos.getY() + 0.1, (double)pos.getZ() + 0.5 + (double)direction2.getStepZ() * 0.65, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PUMPKIN_SEEDS, 4));
-                itemEntity.setDeltaMovement(0.05 * (double)direction2.getStepX() + level.random.nextDouble() * 0.02, 0.05, 0.05 * (double)direction2.getStepZ() + level.random.nextDouble() * 0.02);
-                level.addFreshEntity((net.minecraft.world.entity.Entity)itemEntity);
-                itemStack.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)player, $$1 -> $$1.broadcastBreakEvent(hand));
-                level.gameEvent((net.minecraft.world.entity.Entity)player, net.minecraft.world.level.gameevent.GameEvent.SHEAR, pos);
-                player.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)net.minecraft.world.item.Items.SHEARS));
+    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$12, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.player.Player $$3, net.minecraft.world.InteractionHand $$4, net.minecraft.world.phys.BlockHitResult $$5) {
+        net.minecraft.world.item.ItemStack $$6 = $$3.getItemInHand($$4);
+        if ($$6.is(net.minecraft.world.item.Items.SHEARS)) {
+            if (!$$12.isClientSide) {
+                net.minecraft.core.Direction $$7 = $$5.getDirection();
+                net.minecraft.core.Direction $$8 = $$7.getAxis() == net.minecraft.core.Direction$Axis.Y ? $$3.getDirection().getOpposite() : $$7;
+                $$12.playSound(null, $$2, net.minecraft.sounds.SoundEvents.PUMPKIN_CARVE, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
+                $$12.setBlock($$2, (net.minecraft.world.level.block.state.BlockState)net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.CarvedPumpkinBlock.FACING, (java.lang.Comparable)$$8), 11);
+                net.minecraft.world.entity.item.ItemEntity $$9 = new net.minecraft.world.entity.item.ItemEntity($$12, (double)$$2.getX() + 0.5 + (double)$$8.getStepX() * 0.65, (double)$$2.getY() + 0.1, (double)$$2.getZ() + 0.5 + (double)$$8.getStepZ() * 0.65, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.PUMPKIN_SEEDS, 4));
+                $$9.setDeltaMovement(0.05 * (double)$$8.getStepX() + $$12.random.nextDouble() * 0.02, 0.05, 0.05 * (double)$$8.getStepZ() + $$12.random.nextDouble() * 0.02);
+                $$12.addFreshEntity((net.minecraft.world.entity.Entity)$$9);
+                $$6.hurtAndBreak(1, (net.minecraft.world.entity.LivingEntity)$$3, $$1 -> $$1.broadcastBreakEvent($$4));
+                $$12.gameEvent((net.minecraft.world.entity.Entity)$$3, net.minecraft.world.level.gameevent.GameEvent.SHEAR, $$2);
+                $$3.awardStat(net.minecraft.stats.Stats.ITEM_USED.get((java.lang.Object)net.minecraft.world.item.Items.SHEARS));
             }
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$12.isClientSide);
         }
-        return super.use(state, level, pos, player, hand, hit);
+        return super.use($$0, $$12, $$2, $$3, $$4, $$5);
     }
 
     public net.minecraft.world.level.block.StemBlock getStem() {