--- net/minecraft/world/level/block/RailBlock.java
+++ net/minecraft/world/level/block/RailBlock.java
@@ -13,9 +13,9 @@
         this.registerDefaultState((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)this.stateDefinition.any()).setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH)).setValue((net.minecraft.world.level.block.state.properties.Property)WATERLOGGED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)false)));
     }
 
-    public void updateState(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.Block block) {
-        if (block.defaultBlockState().isSignalSource() && new net.minecraft.world.level.block.RailState(level, pos, state).countPotentialConnections() == 3) {
-            this.updateDir(level, pos, state, false);
+    public void updateState(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.Block $$3) {
+        if ($$3.defaultBlockState().isSignalSource() && new net.minecraft.world.level.block.RailState($$1, $$2, $$0).countPotentialConnections() == 3) {
+            this.updateDir($$1, $$2, $$0, false);
         }
     }
 
@@ -23,163 +23,163 @@
         return SHAPE;
     }
 
-    public net.minecraft.world.level.block.state.BlockState rotate(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.Rotation rot) {
-        switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$Rotation[rot.ordinal()]) {
+    public net.minecraft.world.level.block.state.BlockState rotate(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.Rotation $$1) {
+        switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$Rotation[$$1.ordinal()]) {
             case 1: {
-                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
+                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
                     case 1: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 2: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                 }
             }
             case 2: {
-                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
+                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
                     case 9: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
                     }
                     case 10: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
                     }
                     case 1: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 2: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                 }
             }
             case 3: {
-                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
+                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[((net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE)).ordinal()]) {
                     case 9: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST);
                     }
                     case 10: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH);
                     }
                     case 1: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 2: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                 }
             }
         }
-        return state;
+        return $$0;
     }
 
-    public net.minecraft.world.level.block.state.BlockState mirror(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.Mirror mirror) {
-        net.minecraft.world.level.block.state.properties.RailShape railShape = (net.minecraft.world.level.block.state.properties.RailShape)state.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
-        switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$Mirror[mirror.ordinal()]) {
+    public net.minecraft.world.level.block.state.BlockState mirror(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.Mirror $$1) {
+        net.minecraft.world.level.block.state.properties.RailShape $$2 = (net.minecraft.world.level.block.state.properties.RailShape)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)SHAPE);
+        switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$Mirror[$$1.ordinal()]) {
             case 1: {
-                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[railShape.ordinal()]) {
+                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[$$2.ordinal()]) {
                     case 3: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH);
                     }
                     case 4: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                 }
                 break;
             }
             case 2: {
-                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[railShape.ordinal()]) {
+                switch (net.minecraft.world.level.block.RailBlock$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[$$2.ordinal()]) {
                     case 1: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST);
                     }
                     case 2: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST);
                     }
                     case 5: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST);
                     }
                     case 6: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST);
                     }
                     case 7: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST);
                     }
                     case 8: {
-                        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
+                        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)SHAPE, (java.lang.Comparable)net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST);
                     }
                 }
                 break;
             }
         }
-        return super.mirror(state, mirror);
+        return super.mirror($$0, $$1);
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{SHAPE, WATERLOGGED});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{SHAPE, WATERLOGGED});
     }
 
     static {