--- net/minecraft/world/level/block/RailState.java
+++ net/minecraft/world/level/block/RailState.java
@@ -12,24 +12,24 @@
     public boolean isStraight;
     public java.util.List<net.minecraft.core.BlockPos> connections;
 
-    public net.minecraft.world.level.block.RailState(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.level.block.RailState(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2) {
         this.connections = com.google.common.collect.Lists.newArrayList();
-        this.level = level;
-        this.pos = pos;
-        this.state = state;
-        this.block = (net.minecraft.world.level.block.BaseRailBlock)state.getBlock();
-        net.minecraft.world.level.block.state.properties.RailShape railShape = (net.minecraft.world.level.block.state.properties.RailShape)state.getValue(this.block.getShapeProperty());
+        this.level = $$0;
+        this.pos = $$1;
+        this.state = $$2;
+        this.block = (net.minecraft.world.level.block.BaseRailBlock)$$2.getBlock();
+        net.minecraft.world.level.block.state.properties.RailShape $$3 = (net.minecraft.world.level.block.state.properties.RailShape)$$2.getValue(this.block.getShapeProperty());
         this.isStraight = this.block.isStraight();
-        this.updateConnections(railShape);
+        this.updateConnections($$3);
     }
 
     public java.util.List<net.minecraft.core.BlockPos> getConnections() {
         return this.connections;
     }
 
-    public void updateConnections(net.minecraft.world.level.block.state.properties.RailShape shape) {
+    public void updateConnections(net.minecraft.world.level.block.state.properties.RailShape $$0) {
         this.connections.clear();
-        switch (net.minecraft.world.level.block.RailState$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[shape.ordinal()]) {
+        switch (net.minecraft.world.level.block.RailState$1.$SwitchMap$net$minecraft$world$level$block$state$properties$RailShape[$$0.ordinal()]) {
             case 1: {
                 this.connections.add((java.lang.Object)this.pos.north());
                 this.connections.add((java.lang.Object)this.pos.south());
@@ -93,38 +93,38 @@
         }
     }
 
-    public boolean hasRail(net.minecraft.core.BlockPos pos) {
-        return net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)pos) || net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)pos.above()) || net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)pos.below());
+    public boolean hasRail(net.minecraft.core.BlockPos $$0) {
+        return net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$0) || net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$0.above()) || net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$0.below());
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.block.RailState getRail(net.minecraft.core.BlockPos pos) {
-        net.minecraft.core.BlockPos blockPos = pos;
-        net.minecraft.world.level.block.state.BlockState blockState = this.level.getBlockState(blockPos);
-        if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.block.state.BlockState)blockState)) {
-            return new net.minecraft.world.level.block.RailState(this.level, blockPos, blockState);
+    public net.minecraft.world.level.block.RailState getRail(net.minecraft.core.BlockPos $$0) {
+        net.minecraft.core.BlockPos $$1 = $$0;
+        net.minecraft.world.level.block.state.BlockState $$2 = this.level.getBlockState($$1);
+        if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.block.state.BlockState)$$2)) {
+            return new net.minecraft.world.level.block.RailState(this.level, $$1, $$2);
         }
-        blockPos = pos.above();
-        blockState = this.level.getBlockState(blockPos);
-        if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.block.state.BlockState)blockState)) {
-            return new net.minecraft.world.level.block.RailState(this.level, blockPos, blockState);
+        $$1 = $$0.above();
+        $$2 = this.level.getBlockState($$1);
+        if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.block.state.BlockState)$$2)) {
+            return new net.minecraft.world.level.block.RailState(this.level, $$1, $$2);
         }
-        blockPos = pos.below();
-        blockState = this.level.getBlockState(blockPos);
-        if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.block.state.BlockState)blockState)) {
-            return new net.minecraft.world.level.block.RailState(this.level, blockPos, blockState);
+        $$1 = $$0.below();
+        $$2 = this.level.getBlockState($$1);
+        if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.block.state.BlockState)$$2)) {
+            return new net.minecraft.world.level.block.RailState(this.level, $$1, $$2);
         }
         return null;
     }
 
-    public boolean connectsTo(net.minecraft.world.level.block.RailState state) {
-        return this.hasConnection(state.pos);
+    public boolean connectsTo(net.minecraft.world.level.block.RailState $$0) {
+        return this.hasConnection($$0.pos);
     }
 
-    public boolean hasConnection(net.minecraft.core.BlockPos pos) {
-        for (int i = 0; i < this.connections.size(); ++i) {
-            net.minecraft.core.BlockPos blockPos = (net.minecraft.core.BlockPos)this.connections.get(i);
-            if (blockPos.getX() != pos.getX() || blockPos.getZ() != pos.getZ()) continue;
+    public boolean hasConnection(net.minecraft.core.BlockPos $$0) {
+        for (int $$1 = 0; $$1 < this.connections.size(); ++$$1) {
+            net.minecraft.core.BlockPos $$2 = (net.minecraft.core.BlockPos)this.connections.get($$1);
+            if ($$2.getX() != $$0.getX() || $$2.getZ() != $$0.getZ()) continue;
             return true;
         }
         return false;
@@ -139,178 +139,178 @@
         return $$0;
     }
 
-    public boolean canConnectTo(net.minecraft.world.level.block.RailState state) {
-        return this.connectsTo(state) || this.connections.size() != 2;
+    public boolean canConnectTo(net.minecraft.world.level.block.RailState $$0) {
+        return this.connectsTo($$0) || this.connections.size() != 2;
     }
 
-    public void connectTo(net.minecraft.world.level.block.RailState state) {
-        this.connections.add((java.lang.Object)state.pos);
-        net.minecraft.core.BlockPos blockPos = this.pos.north();
-        net.minecraft.core.BlockPos blockPos2 = this.pos.south();
-        net.minecraft.core.BlockPos blockPos3 = this.pos.west();
-        net.minecraft.core.BlockPos blockPos4 = this.pos.east();
-        boolean bl = this.hasConnection(blockPos);
-        boolean bl2 = this.hasConnection(blockPos2);
-        boolean bl3 = this.hasConnection(blockPos3);
-        boolean bl4 = this.hasConnection(blockPos4);
-        net.minecraft.world.level.block.state.properties.RailShape railShape = null;
-        if (bl || bl2) {
-            railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+    public void connectTo(net.minecraft.world.level.block.RailState $$0) {
+        this.connections.add((java.lang.Object)$$0.pos);
+        net.minecraft.core.BlockPos $$1 = this.pos.north();
+        net.minecraft.core.BlockPos $$2 = this.pos.south();
+        net.minecraft.core.BlockPos $$3 = this.pos.west();
+        net.minecraft.core.BlockPos $$4 = this.pos.east();
+        boolean $$5 = this.hasConnection($$1);
+        boolean $$6 = this.hasConnection($$2);
+        boolean $$7 = this.hasConnection($$3);
+        boolean $$8 = this.hasConnection($$4);
+        net.minecraft.world.level.block.state.properties.RailShape $$9 = null;
+        if ($$5 || $$6) {
+            $$9 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
         }
-        if (bl3 || bl4) {
-            railShape = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
+        if ($$7 || $$8) {
+            $$9 = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
         }
         if (!this.isStraight) {
-            if (bl2 && bl4 && !bl && !bl3) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
+            if ($$6 && $$8 && !$$5 && !$$7) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
             }
-            if (bl2 && bl3 && !bl && !bl4) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
+            if ($$6 && $$7 && !$$5 && !$$8) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
             }
-            if (bl && bl3 && !bl2 && !bl4) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
+            if ($$5 && $$7 && !$$6 && !$$8) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
             }
-            if (bl && bl4 && !bl2 && !bl3) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
+            if ($$5 && $$8 && !$$6 && !$$7) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
             }
         }
-        if (railShape == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH) {
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH;
+        if ($$9 == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH) {
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$1.above())) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH;
             }
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos2.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH;
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$2.above())) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH;
             }
         }
-        if (railShape == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST) {
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos4.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST;
+        if ($$9 == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST) {
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$4.above())) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST;
             }
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos3.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST;
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$3.above())) {
+                $$9 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST;
             }
         }
-        if (railShape == null) {
-            railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+        if ($$9 == null) {
+            $$9 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
         }
-        this.state = (net.minecraft.world.level.block.state.BlockState)this.state.setValue(this.block.getShapeProperty(), (java.lang.Comparable)railShape);
+        this.state = (net.minecraft.world.level.block.state.BlockState)this.state.setValue(this.block.getShapeProperty(), (java.lang.Comparable)$$9);
         this.level.setBlock(this.pos, this.state, 3);
     }
 
-    public boolean hasNeighborRail(net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.level.block.RailState railState = this.getRail(pos);
-        if (railState == null) {
+    public boolean hasNeighborRail(net.minecraft.core.BlockPos $$0) {
+        net.minecraft.world.level.block.RailState $$1 = this.getRail($$0);
+        if ($$1 == null) {
             return false;
         }
-        railState.removeSoftConnections();
-        return railState.canConnectTo(this);
+        $$1.removeSoftConnections();
+        return $$1.canConnectTo(this);
     }
 
-    public net.minecraft.world.level.block.RailState place(boolean powered, boolean placeBlock, net.minecraft.world.level.block.state.properties.RailShape shape) {
-        boolean bl;
-        boolean bl2;
-        net.minecraft.core.BlockPos blockPos = this.pos.north();
-        net.minecraft.core.BlockPos blockPos2 = this.pos.south();
-        net.minecraft.core.BlockPos blockPos3 = this.pos.west();
-        net.minecraft.core.BlockPos blockPos4 = this.pos.east();
-        boolean bl3 = this.hasNeighborRail(blockPos);
-        boolean bl4 = this.hasNeighborRail(blockPos2);
-        boolean bl5 = this.hasNeighborRail(blockPos3);
-        boolean bl6 = this.hasNeighborRail(blockPos4);
-        net.minecraft.world.level.block.state.properties.RailShape railShape = null;
-        boolean bl7 = bl3 || bl4;
-        boolean bl8 = bl2 = bl5 || bl6;
-        if (bl7 && !bl2) {
-            railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+    public net.minecraft.world.level.block.RailState place(boolean $$0, boolean $$1, net.minecraft.world.level.block.state.properties.RailShape $$2) {
+        boolean $$17;
+        boolean $$13;
+        net.minecraft.core.BlockPos $$3 = this.pos.north();
+        net.minecraft.core.BlockPos $$4 = this.pos.south();
+        net.minecraft.core.BlockPos $$5 = this.pos.west();
+        net.minecraft.core.BlockPos $$6 = this.pos.east();
+        boolean $$7 = this.hasNeighborRail($$3);
+        boolean $$8 = this.hasNeighborRail($$4);
+        boolean $$9 = this.hasNeighborRail($$5);
+        boolean $$10 = this.hasNeighborRail($$6);
+        net.minecraft.world.level.block.state.properties.RailShape $$11 = null;
+        boolean $$12 = $$7 || $$8;
+        boolean bl = $$13 = $$9 || $$10;
+        if ($$12 && !$$13) {
+            $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
         }
-        if (bl2 && !bl7) {
-            railShape = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
+        if ($$13 && !$$12) {
+            $$11 = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
         }
-        boolean bl9 = bl4 && bl6;
-        boolean bl10 = bl4 && bl5;
-        boolean bl11 = bl3 && bl6;
-        boolean bl12 = bl = bl3 && bl5;
+        boolean $$14 = $$8 && $$10;
+        boolean $$15 = $$8 && $$9;
+        boolean $$16 = $$7 && $$10;
+        boolean bl2 = $$17 = $$7 && $$9;
         if (!this.isStraight) {
-            if (bl9 && !bl3 && !bl5) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
+            if ($$14 && !$$7 && !$$9) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
             }
-            if (bl10 && !bl3 && !bl6) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
+            if ($$15 && !$$7 && !$$10) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
             }
-            if (bl && !bl4 && !bl6) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
+            if ($$17 && !$$8 && !$$10) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
             }
-            if (bl11 && !bl4 && !bl5) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
+            if ($$16 && !$$8 && !$$9) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
             }
         }
-        if (railShape == null) {
-            if (bl7 && bl2) {
-                railShape = shape;
-            } else if (bl7) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
-            } else if (bl2) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
+        if ($$11 == null) {
+            if ($$12 && $$13) {
+                $$11 = $$2;
+            } else if ($$12) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH;
+            } else if ($$13) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST;
             }
             if (!this.isStraight) {
-                if (powered) {
-                    if (bl9) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
+                if ($$0) {
+                    if ($$14) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
                     }
-                    if (bl10) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
+                    if ($$15) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
                     }
-                    if (bl11) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
+                    if ($$16) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
                     }
-                    if (bl) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
+                    if ($$17) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
                     }
                 } else {
-                    if (bl) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
+                    if ($$17) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_WEST;
                     }
-                    if (bl11) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
+                    if ($$16) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.NORTH_EAST;
                     }
-                    if (bl10) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
+                    if ($$15) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_WEST;
                     }
-                    if (bl9) {
-                        railShape = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
+                    if ($$14) {
+                        $$11 = net.minecraft.world.level.block.state.properties.RailShape.SOUTH_EAST;
                     }
                 }
             }
         }
-        if (railShape == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH) {
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH;
+        if ($$11 == net.minecraft.world.level.block.state.properties.RailShape.NORTH_SOUTH) {
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$3.above())) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_NORTH;
             }
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos2.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH;
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$4.above())) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_SOUTH;
             }
         }
-        if (railShape == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST) {
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos4.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST;
+        if ($$11 == net.minecraft.world.level.block.state.properties.RailShape.EAST_WEST) {
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$6.above())) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_EAST;
             }
-            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)blockPos3.above())) {
-                railShape = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST;
+            if (net.minecraft.world.level.block.BaseRailBlock.isRail((net.minecraft.world.level.Level)this.level, (net.minecraft.core.BlockPos)$$5.above())) {
+                $$11 = net.minecraft.world.level.block.state.properties.RailShape.ASCENDING_WEST;
             }
         }
-        if (railShape == null) {
-            railShape = shape;
+        if ($$11 == null) {
+            $$11 = $$2;
         }
-        this.updateConnections(railShape);
-        this.state = (net.minecraft.world.level.block.state.BlockState)this.state.setValue(this.block.getShapeProperty(), (java.lang.Comparable)railShape);
-        if (placeBlock || this.level.getBlockState(this.pos) != this.state) {
+        this.updateConnections($$11);
+        this.state = (net.minecraft.world.level.block.state.BlockState)this.state.setValue(this.block.getShapeProperty(), (java.lang.Comparable)$$11);
+        if ($$1 || this.level.getBlockState(this.pos) != this.state) {
             this.level.setBlock(this.pos, this.state, 3);
-            for (int i = 0; i < this.connections.size(); ++i) {
-                net.minecraft.world.level.block.RailState railState = this.getRail((net.minecraft.core.BlockPos)this.connections.get(i));
-                if (railState == null) continue;
-                railState.removeSoftConnections();
-                if (!railState.canConnectTo(this)) continue;
-                railState.connectTo(this);
+            for (int $$18 = 0; $$18 < this.connections.size(); ++$$18) {
+                net.minecraft.world.level.block.RailState $$19 = this.getRail((net.minecraft.core.BlockPos)this.connections.get($$18));
+                if ($$19 == null) continue;
+                $$19.removeSoftConnections();
+                if (!$$19.canConnectTo(this)) continue;
+                $$19.connectTo(this);
             }
         }
         return this;