--- net/minecraft/world/level/block/RedstoneTorchBlock$Toggle.java
+++ net/minecraft/world/level/block/RedstoneTorchBlock$Toggle.java
@@ -8,9 +8,9 @@
     public net.minecraft.core.BlockPos pos;
     public long when;
 
-    public net.minecraft.world.level.block.RedstoneTorchBlock$Toggle(net.minecraft.core.BlockPos pos, long when) {
-        this.pos = pos;
-        this.when = when;
+    public net.minecraft.world.level.block.RedstoneTorchBlock$Toggle(net.minecraft.core.BlockPos $$0, long $$1) {
+        this.pos = $$0;
+        this.when = $$1;
     }
 }
 