--- net/minecraft/world/level/block/RedstoneLampBlock.java
+++ net/minecraft/world/level/block/RedstoneLampBlock.java
@@ -14,20 +14,20 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext context) {
-        return (net.minecraft.world.level.block.state.BlockState)this.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)context.getLevel().hasNeighborSignal(context.getClickedPos())));
+    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext $$0) {
+        return (net.minecraft.world.level.block.state.BlockState)this.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)$$0.getLevel().hasNeighborSignal($$0.getClickedPos())));
     }
 
-    public void neighborChanged(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.Block block, net.minecraft.core.BlockPos fromPos, boolean isMoving) {
-        if (level.isClientSide) {
+    public void neighborChanged(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.Block $$3, net.minecraft.core.BlockPos $$4, boolean $$5) {
+        if ($$1.isClientSide) {
             return;
         }
-        boolean bl = (java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)LIT);
-        if (bl != level.hasNeighborSignal(pos)) {
-            if (bl) {
-                level.scheduleTick(pos, (net.minecraft.world.level.block.Block)this, 4);
+        boolean $$6 = (java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)LIT);
+        if ($$6 != $$1.hasNeighborSignal($$2)) {
+            if ($$6) {
+                $$1.scheduleTick($$2, (net.minecraft.world.level.block.Block)this, 4);
             } else {
-                level.setBlock(pos, (net.minecraft.world.level.block.state.BlockState)state.cycle((net.minecraft.world.level.block.state.properties.Property)LIT), 2);
+                $$1.setBlock($$2, (net.minecraft.world.level.block.state.BlockState)$$0.cycle((net.minecraft.world.level.block.state.properties.Property)LIT), 2);
             }
         }
     }
@@ -38,8 +38,8 @@
         }
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{LIT});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{LIT});
     }
 
     static {