--- net/minecraft/world/level/block/RedStoneOreBlock.java
+++ net/minecraft/world/level/block/RedStoneOreBlock.java
@@ -13,9 +13,9 @@
         this.registerDefaultState((net.minecraft.world.level.block.state.BlockState)this.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)false)));
     }
 
-    public void attack(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.player.Player player) {
-        net.minecraft.world.level.block.RedStoneOreBlock.interact((net.minecraft.world.level.block.state.BlockState)state, (net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)pos);
-        super.attack(state, level, pos, player);
+    public void attack(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.player.Player $$3) {
+        net.minecraft.world.level.block.RedStoneOreBlock.interact((net.minecraft.world.level.block.state.BlockState)$$0, (net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2);
+        super.attack($$0, $$1, $$2, $$3);
     }
 
     public void stepOn(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.entity.Entity $$3) {
@@ -25,28 +25,28 @@
         super.stepOn($$0, $$1, $$2, $$3);
     }
 
-    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand, net.minecraft.world.phys.BlockHitResult hit) {
-        if (level.isClientSide) {
-            net.minecraft.world.level.block.RedStoneOreBlock.spawnParticles((net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)pos);
+    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.player.Player $$3, net.minecraft.world.InteractionHand $$4, net.minecraft.world.phys.BlockHitResult $$5) {
+        if ($$1.isClientSide) {
+            net.minecraft.world.level.block.RedStoneOreBlock.spawnParticles((net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2);
         } else {
-            net.minecraft.world.level.block.RedStoneOreBlock.interact((net.minecraft.world.level.block.state.BlockState)state, (net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)pos);
+            net.minecraft.world.level.block.RedStoneOreBlock.interact((net.minecraft.world.level.block.state.BlockState)$$0, (net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2);
         }
-        net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(hand);
-        if (itemStack.getItem() instanceof net.minecraft.world.item.BlockItem && new net.minecraft.world.item.context.BlockPlaceContext(player, hand, itemStack, hit).canPlace()) {
+        net.minecraft.world.item.ItemStack $$6 = $$3.getItemInHand($$4);
+        if ($$6.getItem() instanceof net.minecraft.world.item.BlockItem && new net.minecraft.world.item.context.BlockPlaceContext($$3, $$4, $$6, $$5).canPlace()) {
             return net.minecraft.world.InteractionResult.PASS;
         }
         return net.minecraft.world.InteractionResult.SUCCESS;
     }
 
-    public static void interact(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.level.block.RedStoneOreBlock.spawnParticles((net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)pos);
-        if (!((java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)LIT)).booleanValue()) {
-            level.setBlock(pos, (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)), 3);
+    public static void interact(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2) {
+        net.minecraft.world.level.block.RedStoneOreBlock.spawnParticles((net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2);
+        if (!((java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)LIT)).booleanValue()) {
+            $$1.setBlock($$2, (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)), 3);
         }
     }
 
-    public boolean isRandomlyTicking(net.minecraft.world.level.block.state.BlockState state) {
-        return (java.lang.Boolean)state.getValue((net.minecraft.world.level.block.state.properties.Property)LIT);
+    public boolean isRandomlyTicking(net.minecraft.world.level.block.state.BlockState $$0) {
+        return (java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)LIT);
     }
 
     public void randomTick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.server.level.ServerLevel $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
@@ -69,22 +69,22 @@
         }
     }
 
-    public static void spawnParticles(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        double d = 0.5625;
-        net.minecraft.util.RandomSource randomSource = level.random;
-        for (net.minecraft.core.Direction direction : net.minecraft.core.Direction.values()) {
-            net.minecraft.core.BlockPos blockPos = pos.relative(direction);
-            if (level.getBlockState(blockPos).isSolidRender((net.minecraft.world.level.BlockGetter)level, blockPos)) continue;
-            net.minecraft.core.Direction$Axis axis = direction.getAxis();
-            double d2 = axis == net.minecraft.core.Direction$Axis.X ? 0.5 + 0.5625 * (double)direction.getStepX() : (double)randomSource.nextFloat();
-            double d3 = axis == net.minecraft.core.Direction$Axis.Y ? 0.5 + 0.5625 * (double)direction.getStepY() : (double)randomSource.nextFloat();
-            double d4 = axis == net.minecraft.core.Direction$Axis.Z ? 0.5 + 0.5625 * (double)direction.getStepZ() : (double)randomSource.nextFloat();
-            level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.DustParticleOptions.REDSTONE, (double)pos.getX() + d2, (double)pos.getY() + d3, (double)pos.getZ() + d4, 0.0, 0.0, 0.0);
+    public static void spawnParticles(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1) {
+        double $$2 = 0.5625;
+        net.minecraft.util.RandomSource $$3 = $$0.random;
+        for (net.minecraft.core.Direction $$4 : net.minecraft.core.Direction.values()) {
+            net.minecraft.core.BlockPos $$5 = $$1.relative($$4);
+            if ($$0.getBlockState($$5).isSolidRender((net.minecraft.world.level.BlockGetter)$$0, $$5)) continue;
+            net.minecraft.core.Direction$Axis $$6 = $$4.getAxis();
+            double $$7 = $$6 == net.minecraft.core.Direction$Axis.X ? 0.5 + 0.5625 * (double)$$4.getStepX() : (double)$$3.nextFloat();
+            double $$8 = $$6 == net.minecraft.core.Direction$Axis.Y ? 0.5 + 0.5625 * (double)$$4.getStepY() : (double)$$3.nextFloat();
+            double $$9 = $$6 == net.minecraft.core.Direction$Axis.Z ? 0.5 + 0.5625 * (double)$$4.getStepZ() : (double)$$3.nextFloat();
+            $$0.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.DustParticleOptions.REDSTONE, (double)$$1.getX() + $$7, (double)$$1.getY() + $$8, (double)$$1.getZ() + $$9, 0.0, 0.0, 0.0);
         }
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{LIT});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{LIT});
     }
 
     static {