--- net/minecraft/world/level/block/Rotation.java
+++ net/minecraft/world/level/block/Rotation.java
@@ -30,8 +30,8 @@
         this.rotation = $$1;
     }
 
-    public net.minecraft.world.level.block.Rotation getRotated(net.minecraft.world.level.block.Rotation rotation) {
-        switch (net.minecraft.world.level.block.Rotation$1.$SwitchMap$net$minecraft$world$level$block$Rotation[rotation.ordinal()]) {
+    public net.minecraft.world.level.block.Rotation getRotated(net.minecraft.world.level.block.Rotation $$0) {
+        switch (net.minecraft.world.level.block.Rotation$1.$SwitchMap$net$minecraft$world$level$block$Rotation[$$0.ordinal()]) {
             case 3: {
                 switch (net.minecraft.world.level.block.Rotation$1.$SwitchMap$net$minecraft$world$level$block$Rotation[this.ordinal()]) {
                     case 1: {
@@ -88,37 +88,37 @@
         return this.rotation;
     }
 
-    public net.minecraft.core.Direction rotate(net.minecraft.core.Direction facing) {
-        if (facing.getAxis() == net.minecraft.core.Direction$Axis.Y) {
-            return facing;
+    public net.minecraft.core.Direction rotate(net.minecraft.core.Direction $$0) {
+        if ($$0.getAxis() == net.minecraft.core.Direction$Axis.Y) {
+            return $$0;
         }
         switch (net.minecraft.world.level.block.Rotation$1.$SwitchMap$net$minecraft$world$level$block$Rotation[this.ordinal()]) {
             case 3: {
-                return facing.getOpposite();
+                return $$0.getOpposite();
             }
             case 4: {
-                return facing.getCounterClockWise();
+                return $$0.getCounterClockWise();
             }
             case 2: {
-                return facing.getClockWise();
+                return $$0.getClockWise();
             }
         }
-        return facing;
+        return $$0;
     }
 
-    public int rotate(int rotation, int positionCount) {
+    public int rotate(int $$0, int $$1) {
         switch (net.minecraft.world.level.block.Rotation$1.$SwitchMap$net$minecraft$world$level$block$Rotation[this.ordinal()]) {
             case 3: {
-                return (rotation + positionCount / 2) % positionCount;
+                return ($$0 + $$1 / 2) % $$1;
             }
             case 4: {
-                return (rotation + positionCount * 3 / 4) % positionCount;
+                return ($$0 + $$1 * 3 / 4) % $$1;
             }
             case 2: {
-                return (rotation + positionCount / 4) % positionCount;
+                return ($$0 + $$1 / 4) % $$1;
             }
         }
-        return rotation;
+        return $$0;
     }
 
     public static net.minecraft.world.level.block.Rotation getRandom(net.minecraft.util.RandomSource $$0) {