--- net/minecraft/world/level/block/RotatedPillarBlock.java
+++ net/minecraft/world/level/block/RotatedPillarBlock.java
@@ -13,8 +13,8 @@
         this.registerDefaultState((net.minecraft.world.level.block.state.BlockState)this.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)AXIS, (java.lang.Comparable)net.minecraft.core.Direction$Axis.Y));
     }
 
-    public net.minecraft.world.level.block.state.BlockState rotate(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.Rotation rot) {
-        return net.minecraft.world.level.block.RotatedPillarBlock.rotatePillar((net.minecraft.world.level.block.state.BlockState)state, (net.minecraft.world.level.block.Rotation)rot);
+    public net.minecraft.world.level.block.state.BlockState rotate(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.Rotation $$1) {
+        return net.minecraft.world.level.block.RotatedPillarBlock.rotatePillar((net.minecraft.world.level.block.state.BlockState)$$0, (net.minecraft.world.level.block.Rotation)$$1);
     }
 
     public static net.minecraft.world.level.block.state.BlockState rotatePillar(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.Rotation $$1) {
@@ -35,12 +35,12 @@
         return $$0;
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{AXIS});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{AXIS});
     }
 
-    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext context) {
-        return (net.minecraft.world.level.block.state.BlockState)this.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)AXIS, (java.lang.Comparable)context.getClickedFace().getAxis());
+    public net.minecraft.world.level.block.state.BlockState getStateForPlacement(net.minecraft.world.item.context.BlockPlaceContext $$0) {
+        return (net.minecraft.world.level.block.state.BlockState)this.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)AXIS, (java.lang.Comparable)$$0.getClickedFace().getAxis());
     }
 
     static {