--- net/minecraft/world/level/block/SandBlock.java
+++ net/minecraft/world/level/block/SandBlock.java
@@ -8,12 +8,12 @@
  {
     public int dustColor;
 
-    public net.minecraft.world.level.block.SandBlock(int dustColor, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
-        this.dustColor = dustColor;
+    public net.minecraft.world.level.block.SandBlock(int $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$1);
+        this.dustColor = $$0;
     }
 
-    public int getDustColor(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter reader, net.minecraft.core.BlockPos pos) {
+    public int getDustColor(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2) {
         return this.dustColor;
     }
 }