--- net/minecraft/world/level/block/SculkVeinBlock$SculkVeinSpreaderConfig.java
+++ net/minecraft/world/level/block/SculkVeinBlock$SculkVeinSpreaderConfig.java
@@ -32,7 +32,7 @@
         if ($$8 == net.minecraft.world.level.material.Material.FIRE) {
             return false;
         }
-        return $$8.isReplaceable() || super.stateCanBeReplaced($$0, $$1, $$2, $$3, $$4);
+        return $$4.canBeReplaced() || super.stateCanBeReplaced($$0, $$1, $$2, $$3, $$4);
     }
 
     public net.minecraft.world.level.block.MultifaceSpreader$SpreadType[] getSpreadTypes() {