--- net/minecraft/world/level/block/SlimeBlock.java
+++ net/minecraft/world/level/block/SlimeBlock.java
@@ -18,19 +18,19 @@
         }
     }
 
-    public void updateEntityAfterFallOn(net.minecraft.world.level.BlockGetter level, net.minecraft.world.entity.Entity entity) {
-        if (entity.isSuppressingBounce()) {
-            super.updateEntityAfterFallOn(level, entity);
+    public void updateEntityAfterFallOn(net.minecraft.world.level.BlockGetter $$0, net.minecraft.world.entity.Entity $$1) {
+        if ($$1.isSuppressingBounce()) {
+            super.updateEntityAfterFallOn($$0, $$1);
         } else {
-            this.bounceUp(entity);
+            this.bounceUp($$1);
         }
     }
 
-    public void bounceUp(net.minecraft.world.entity.Entity entity) {
-        net.minecraft.world.phys.Vec3 vec3 = entity.getDeltaMovement();
-        if (vec3.y < 0.0) {
-            double d = entity instanceof net.minecraft.world.entity.LivingEntity ? 1.0 : 0.8;
-            entity.setDeltaMovement(vec3.x, -vec3.y * d, vec3.z);
+    public void bounceUp(net.minecraft.world.entity.Entity $$0) {
+        net.minecraft.world.phys.Vec3 $$1 = $$0.getDeltaMovement();
+        if ($$1.y < 0.0) {
+            double $$2 = $$0 instanceof net.minecraft.world.entity.LivingEntity ? 1.0 : 0.8;
+            $$0.setDeltaMovement($$1.x, -$$1.y * $$2, $$1.z);
         }
     }
 