--- net/minecraft/world/level/block/SmithingTableBlock.java
+++ net/minecraft/world/level/block/SmithingTableBlock.java
@@ -12,16 +12,16 @@
         super($$0);
     }
 
-    public net.minecraft.world.MenuProvider getMenuProvider(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        return new net.minecraft.world.SimpleMenuProvider((containerId, inventory, player) -> new net.minecraft.world.inventory.SmithingMenu(containerId, inventory, net.minecraft.world.inventory.ContainerLevelAccess.create((net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)pos)), CONTAINER_TITLE);
+    public net.minecraft.world.MenuProvider getMenuProvider(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$22) {
+        return new net.minecraft.world.SimpleMenuProvider(($$2, $$3, $$4) -> new net.minecraft.world.inventory.SmithingMenu($$2, $$3, net.minecraft.world.inventory.ContainerLevelAccess.create((net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$22)), CONTAINER_TITLE);
     }
 
-    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand, net.minecraft.world.phys.BlockHitResult hit) {
-        if (level.isClientSide) {
+    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.player.Player $$3, net.minecraft.world.InteractionHand $$4, net.minecraft.world.phys.BlockHitResult $$5) {
+        if ($$1.isClientSide) {
             return net.minecraft.world.InteractionResult.SUCCESS;
         }
-        player.openMenu(state.getMenuProvider(level, pos));
-        player.awardStat(net.minecraft.stats.Stats.INTERACT_WITH_SMITHING_TABLE);
+        $$3.openMenu($$0.getMenuProvider($$1, $$2));
+        $$3.awardStat(net.minecraft.stats.Stats.INTERACT_WITH_SMITHING_TABLE);
         return net.minecraft.world.InteractionResult.CONSUME;
     }
 