--- net/minecraft/world/level/block/SoulFireBlock.java
+++ net/minecraft/world/level/block/SoulFireBlock.java
@@ -10,22 +10,22 @@
         super($$0, 2.0f);
     }
 
-    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.core.Direction facing, net.minecraft.world.level.block.state.BlockState facingState, net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos currentPos, net.minecraft.core.BlockPos facingPos) {
-        if (this.canSurvive(state, (net.minecraft.world.level.LevelReader)level, currentPos)) {
+    public net.minecraft.world.level.block.state.BlockState updateShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.core.Direction $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.LevelAccessor $$3, net.minecraft.core.BlockPos $$4, net.minecraft.core.BlockPos $$5) {
+        if (this.canSurvive($$0, (net.minecraft.world.level.LevelReader)$$3, $$4)) {
             return this.defaultBlockState();
         }
         return net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
     }
 
-    public boolean canSurvive(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        return net.minecraft.world.level.block.SoulFireBlock.canSurviveOnBlock((net.minecraft.world.level.block.state.BlockState)level.getBlockState(pos.below()));
+    public boolean canSurvive(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.LevelReader $$1, net.minecraft.core.BlockPos $$2) {
+        return net.minecraft.world.level.block.SoulFireBlock.canSurviveOnBlock((net.minecraft.world.level.block.state.BlockState)$$1.getBlockState($$2.below()));
     }
 
     public static boolean canSurviveOnBlock(net.minecraft.world.level.block.state.BlockState $$0) {
         return $$0.is(net.minecraft.tags.BlockTags.SOUL_FIRE_BASE_BLOCKS);
     }
 
-    public boolean canBurn(net.minecraft.world.level.block.state.BlockState state) {
+    public boolean canBurn(net.minecraft.world.level.block.state.BlockState $$0) {
         return true;
     }
 }