--- net/minecraft/world/level/block/SoundType.java
+++ net/minecraft/world/level/block/SoundType.java
@@ -92,6 +92,9 @@
     public static net.minecraft.world.level.block.SoundType MUD;
     public static net.minecraft.world.level.block.SoundType MUD_BRICKS;
     public static net.minecraft.world.level.block.SoundType PACKED_MUD;
+    public static net.minecraft.world.level.block.SoundType HANGING_SIGN;
+    public static net.minecraft.world.level.block.SoundType BAMBOO_WOOD;
+    public static net.minecraft.world.level.block.SoundType NETHER_WOOD;
     public float volume;
     public float pitch;
     public net.minecraft.sounds.SoundEvent breakSound;
@@ -100,14 +103,14 @@
     public net.minecraft.sounds.SoundEvent hitSound;
     public net.minecraft.sounds.SoundEvent fallSound;
 
-    public net.minecraft.world.level.block.SoundType(float volume, float pitch, net.minecraft.sounds.SoundEvent breakSound, net.minecraft.sounds.SoundEvent stepSound, net.minecraft.sounds.SoundEvent placeSound, net.minecraft.sounds.SoundEvent hitSound, net.minecraft.sounds.SoundEvent fallSound) {
-        this.volume = volume;
-        this.pitch = pitch;
-        this.breakSound = breakSound;
-        this.stepSound = stepSound;
-        this.placeSound = placeSound;
-        this.hitSound = hitSound;
-        this.fallSound = fallSound;
+    public net.minecraft.world.level.block.SoundType(float $$0, float $$1, net.minecraft.sounds.SoundEvent $$2, net.minecraft.sounds.SoundEvent $$3, net.minecraft.sounds.SoundEvent $$4, net.minecraft.sounds.SoundEvent $$5, net.minecraft.sounds.SoundEvent $$6) {
+        this.volume = $$0;
+        this.pitch = $$1;
+        this.breakSound = $$2;
+        this.stepSound = $$3;
+        this.placeSound = $$4;
+        this.hitSound = $$5;
+        this.fallSound = $$6;
     }
 
     public float getVolume() {
@@ -226,6 +229,9 @@
         MUD = new net.minecraft.world.level.block.SoundType(1.0f, 1.0f, net.minecraft.sounds.SoundEvents.MUD_BREAK, net.minecraft.sounds.SoundEvents.MUD_STEP, net.minecraft.sounds.SoundEvents.MUD_PLACE, net.minecraft.sounds.SoundEvents.MUD_HIT, net.minecraft.sounds.SoundEvents.MUD_FALL);
         MUD_BRICKS = new net.minecraft.world.level.block.SoundType(1.0f, 1.0f, net.minecraft.sounds.SoundEvents.MUD_BRICKS_BREAK, net.minecraft.sounds.SoundEvents.MUD_BRICKS_STEP, net.minecraft.sounds.SoundEvents.MUD_BRICKS_PLACE, net.minecraft.sounds.SoundEvents.MUD_BRICKS_HIT, net.minecraft.sounds.SoundEvents.MUD_BRICKS_FALL);
         PACKED_MUD = new net.minecraft.world.level.block.SoundType(1.0f, 1.0f, net.minecraft.sounds.SoundEvents.PACKED_MUD_BREAK, net.minecraft.sounds.SoundEvents.PACKED_MUD_STEP, net.minecraft.sounds.SoundEvents.PACKED_MUD_PLACE, net.minecraft.sounds.SoundEvents.PACKED_MUD_HIT, net.minecraft.sounds.SoundEvents.PACKED_MUD_FALL);
+        HANGING_SIGN = new net.minecraft.world.level.block.SoundType(1.0f, 1.0f, net.minecraft.sounds.SoundEvents.HANGING_SIGN_BREAK, net.minecraft.sounds.SoundEvents.HANGING_SIGN_STEP, net.minecraft.sounds.SoundEvents.HANGING_SIGN_PLACE, net.minecraft.sounds.SoundEvents.HANGING_SIGN_HIT, net.minecraft.sounds.SoundEvents.HANGING_SIGN_FALL);
+        BAMBOO_WOOD = new net.minecraft.world.level.block.SoundType(1.0f, 1.0f, net.minecraft.sounds.SoundEvents.BAMBOO_WOOD_BREAK, net.minecraft.sounds.SoundEvents.BAMBOO_WOOD_STEP, net.minecraft.sounds.SoundEvents.BAMBOO_WOOD_PLACE, net.minecraft.sounds.SoundEvents.BAMBOO_WOOD_HIT, net.minecraft.sounds.SoundEvents.BAMBOO_WOOD_FALL);
+        NETHER_WOOD = new net.minecraft.world.level.block.SoundType(1.0f, 1.0f, net.minecraft.sounds.SoundEvents.NETHER_WOOD_BREAK, net.minecraft.sounds.SoundEvents.NETHER_WOOD_STEP, net.minecraft.sounds.SoundEvents.NETHER_WOOD_PLACE, net.minecraft.sounds.SoundEvents.NETHER_WOOD_HIT, net.minecraft.sounds.SoundEvents.NETHER_WOOD_FALL);
     }
 }
 