--- net/minecraft/world/level/block/StainedGlassBlock.java
+++ net/minecraft/world/level/block/StainedGlassBlock.java
@@ -9,9 +9,9 @@
  {
     public net.minecraft.world.item.DyeColor color;
 
-    public net.minecraft.world.level.block.StainedGlassBlock(net.minecraft.world.item.DyeColor dyeColor, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
-        this.color = dyeColor;
+    public net.minecraft.world.level.block.StainedGlassBlock(net.minecraft.world.item.DyeColor $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$1);
+        this.color = $$0;
     }
 
     public net.minecraft.world.item.DyeColor getColor() {