--- net/minecraft/world/level/block/SweetBerryBushBlock.java
+++ net/minecraft/world/level/block/SweetBerryBushBlock.java
@@ -18,22 +18,22 @@
         this.registerDefaultState((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)this.stateDefinition.any()).setValue((net.minecraft.world.level.block.state.properties.Property)AGE, (java.lang.Comparable)java.lang.Integer.valueOf((int)0)));
     }
 
-    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state) {
+    public net.minecraft.world.item.ItemStack getCloneItemStack(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2) {
         return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SWEET_BERRIES);
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
-        if ((java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) == 0) {
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
+        if ((java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) == 0) {
             return SAPLING_SHAPE;
         }
-        if ((java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) < 3) {
+        if ((java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) < 3) {
             return MID_GROWTH_SHAPE;
         }
-        return super.getShape(state, level, pos, context);
+        return super.getShape($$0, $$1, $$2, $$3);
     }
 
-    public boolean isRandomlyTicking(net.minecraft.world.level.block.state.BlockState state) {
-        return (java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) < 3;
+    public boolean isRandomlyTicking(net.minecraft.world.level.block.state.BlockState $$0) {
+        return (java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) < 3;
     }
 
     public void randomTick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.server.level.ServerLevel $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
@@ -45,45 +45,45 @@
         }
     }
 
-    public void entityInside(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.Entity entity) {
-        if (!(entity instanceof net.minecraft.world.entity.LivingEntity) || entity.getType() == net.minecraft.world.entity.EntityType.FOX || entity.getType() == net.minecraft.world.entity.EntityType.BEE) {
+    public void entityInside(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.Entity $$3) {
+        if (!($$3 instanceof net.minecraft.world.entity.LivingEntity) || $$3.getType() == net.minecraft.world.entity.EntityType.FOX || $$3.getType() == net.minecraft.world.entity.EntityType.BEE) {
             return;
         }
-        entity.makeStuckInBlock(state, new net.minecraft.world.phys.Vec3((double)0.8f, 0.75, (double)0.8f));
-        if (!(level.isClientSide || (java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) <= 0 || entity.xOld == entity.getX() && entity.zOld == entity.getZ())) {
-            double d = java.lang.Math.abs((double)(entity.getX() - entity.xOld));
-            double d2 = java.lang.Math.abs((double)(entity.getZ() - entity.zOld));
-            if (d >= (double)0.003f || d2 >= (double)0.003f) {
-                entity.hurt(net.minecraft.world.damagesource.DamageSource.SWEET_BERRY_BUSH, 1.0f);
+        $$3.makeStuckInBlock($$0, new net.minecraft.world.phys.Vec3((double)0.8f, 0.75, (double)0.8f));
+        if (!($$1.isClientSide || (java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) <= 0 || $$3.xOld == $$3.getX() && $$3.zOld == $$3.getZ())) {
+            double $$4 = java.lang.Math.abs((double)($$3.getX() - $$3.xOld));
+            double $$5 = java.lang.Math.abs((double)($$3.getZ() - $$3.zOld));
+            if ($$4 >= (double)0.003f || $$5 >= (double)0.003f) {
+                $$3.hurt(net.minecraft.world.damagesource.DamageSource.SWEET_BERRY_BUSH, 1.0f);
             }
         }
     }
 
-    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand, net.minecraft.world.phys.BlockHitResult hit) {
-        boolean bl;
-        int n = (java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)AGE);
-        boolean bl2 = bl = n == 3;
-        if (!bl && player.getItemInHand(hand).is(net.minecraft.world.item.Items.BONE_MEAL)) {
+    public net.minecraft.world.InteractionResult use(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.player.Player $$3, net.minecraft.world.InteractionHand $$4, net.minecraft.world.phys.BlockHitResult $$5) {
+        boolean $$7;
+        int $$6 = (java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)AGE);
+        boolean bl = $$7 = $$6 == 3;
+        if (!$$7 && $$3.getItemInHand($$4).is(net.minecraft.world.item.Items.BONE_MEAL)) {
             return net.minecraft.world.InteractionResult.PASS;
         }
-        if (n > 1) {
-            int n2 = 1 + level.random.nextInt(2);
-            net.minecraft.world.level.block.SweetBerryBushBlock.popResource((net.minecraft.world.level.Level)level, (net.minecraft.core.BlockPos)pos, (net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SWEET_BERRIES, n2 + (bl ? 1 : 0)));
-            level.playSound(null, pos, net.minecraft.sounds.SoundEvents.SWEET_BERRY_BUSH_PICK_BERRIES, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 0.8f + level.random.nextFloat() * 0.4f);
-            net.minecraft.world.level.block.state.BlockState blockState = (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)AGE, (java.lang.Comparable)java.lang.Integer.valueOf((int)1));
-            level.setBlock(pos, blockState, 2);
-            level.gameEvent(net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, pos, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)player, (net.minecraft.world.level.block.state.BlockState)blockState));
-            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)level.isClientSide);
+        if ($$6 > 1) {
+            int $$8 = 1 + $$1.random.nextInt(2);
+            net.minecraft.world.level.block.SweetBerryBushBlock.popResource((net.minecraft.world.level.Level)$$1, (net.minecraft.core.BlockPos)$$2, (net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SWEET_BERRIES, $$8 + ($$7 ? 1 : 0)));
+            $$1.playSound(null, $$2, net.minecraft.sounds.SoundEvents.SWEET_BERRY_BUSH_PICK_BERRIES, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 0.8f + $$1.random.nextFloat() * 0.4f);
+            net.minecraft.world.level.block.state.BlockState $$9 = (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)AGE, (java.lang.Comparable)java.lang.Integer.valueOf((int)1));
+            $$1.setBlock($$2, $$9, 2);
+            $$1.gameEvent(net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, $$2, net.minecraft.world.level.gameevent.GameEvent$Context.of((net.minecraft.world.entity.Entity)$$3, (net.minecraft.world.level.block.state.BlockState)$$9));
+            return net.minecraft.world.InteractionResult.sidedSuccess((boolean)$$1.isClientSide);
         }
-        return super.use(state, level, pos, player, hand, hit);
+        return super.use($$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{AGE});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{AGE});
     }
 
-    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, boolean isClient) {
-        return (java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) < 3;
+    public boolean isValidBonemealTarget(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, boolean $$3) {
+        return (java.lang.Integer)$$2.getValue((net.minecraft.world.level.block.state.properties.Property)AGE) < 3;
     }
 
     public boolean isBonemealSuccess(net.minecraft.world.level.Level $$0, net.minecraft.util.RandomSource $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3) {