--- net/minecraft/world/level/block/TrappedChestBlock.java
+++ net/minecraft/world/level/block/TrappedChestBlock.java
@@ -18,17 +18,17 @@
         return net.minecraft.stats.Stats.CUSTOM.get((java.lang.Object)net.minecraft.stats.Stats.TRIGGER_TRAPPED_CHEST);
     }
 
-    public boolean isSignalSource(net.minecraft.world.level.block.state.BlockState state) {
+    public boolean isSignalSource(net.minecraft.world.level.block.state.BlockState $$0) {
         return true;
     }
 
-    public int getSignal(net.minecraft.world.level.block.state.BlockState blockState, net.minecraft.world.level.BlockGetter blockAccess, net.minecraft.core.BlockPos pos, net.minecraft.core.Direction side) {
-        return net.minecraft.util.Mth.clamp((int)net.minecraft.world.level.block.entity.ChestBlockEntity.getOpenCount((net.minecraft.world.level.BlockGetter)blockAccess, (net.minecraft.core.BlockPos)pos), (int)0, (int)15);
+    public int getSignal(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.core.Direction $$3) {
+        return net.minecraft.util.Mth.clamp((int)net.minecraft.world.level.block.entity.ChestBlockEntity.getOpenCount((net.minecraft.world.level.BlockGetter)$$1, (net.minecraft.core.BlockPos)$$2), (int)0, (int)15);
     }
 
-    public int getDirectSignal(net.minecraft.world.level.block.state.BlockState blockState, net.minecraft.world.level.BlockGetter blockAccess, net.minecraft.core.BlockPos pos, net.minecraft.core.Direction side) {
-        if (side == net.minecraft.core.Direction.UP) {
-            return blockState.getSignal(blockAccess, pos, side);
+    public int getDirectSignal(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.core.Direction $$3) {
+        if ($$3 == net.minecraft.core.Direction.UP) {
+            return $$0.getSignal($$1, $$2, $$3);
         }
         return 0;
     }