--- net/minecraft/world/level/block/WallHangingSignBlock$1.java
+++ net/minecraft/world/level/block/WallHangingSignBlock$1.java
@@ -1,0 +1,26 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level.block;
+
+public class net.minecraft.world.level.block.WallHangingSignBlock$1
+ {
+    public static /* synthetic */ int[] $SwitchMap$net$minecraft$core$Direction;
+
+    static {
+        $SwitchMap$net$minecraft$core$Direction = new int[net.minecraft.core.Direction.values().length];
+        try {
+            net.minecraft.world.level.block.WallHangingSignBlock$1.$SwitchMap$net$minecraft$core$Direction[net.minecraft.core.Direction.EAST.ordinal()] = 1;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.world.level.block.WallHangingSignBlock$1.$SwitchMap$net$minecraft$core$Direction[net.minecraft.core.Direction.WEST.ordinal()] = 2;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+    }
+}
+