--- net/minecraft/world/level/block/WaterlilyBlock.java
+++ net/minecraft/world/level/block/WaterlilyBlock.java
@@ -12,21 +12,21 @@
         super($$0);
     }
 
-    public void entityInside(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.Entity entity) {
-        super.entityInside(state, level, pos, entity);
-        if (level instanceof net.minecraft.server.level.ServerLevel && entity instanceof net.minecraft.world.entity.vehicle.Boat) {
-            level.destroyBlock(new net.minecraft.core.BlockPos((net.minecraft.core.Vec3i)pos), true, entity);
+    public void entityInside(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.Entity $$3) {
+        super.entityInside($$0, $$1, $$2, $$3);
+        if ($$1 instanceof net.minecraft.server.level.ServerLevel && $$3 instanceof net.minecraft.world.entity.vehicle.Boat) {
+            $$1.destroyBlock(new net.minecraft.core.BlockPos((net.minecraft.core.Vec3i)$$2), true, $$3);
         }
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext context) {
+    public net.minecraft.world.phys.shapes.VoxelShape getShape(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
         return AABB;
     }
 
-    public boolean mayPlaceOn(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.level.material.FluidState fluidState = level.getFluidState(pos);
-        net.minecraft.world.level.material.FluidState fluidState2 = level.getFluidState(pos.above());
-        return (fluidState.getType() == net.minecraft.world.level.material.Fluids.WATER || state.getMaterial() == net.minecraft.world.level.material.Material.ICE) && fluidState2.getType() == net.minecraft.world.level.material.Fluids.EMPTY;
+    public boolean mayPlaceOn(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2) {
+        net.minecraft.world.level.material.FluidState $$3 = $$1.getFluidState($$2);
+        net.minecraft.world.level.material.FluidState $$4 = $$1.getFluidState($$2.above());
+        return ($$3.getType() == net.minecraft.world.level.material.Fluids.WATER || $$0.getMaterial() == net.minecraft.world.level.material.Material.ICE) && $$4.getType() == net.minecraft.world.level.material.Fluids.EMPTY;
     }
 
     static {