--- net/minecraft/world/level/block/WeightedPressurePlateBlock.java
+++ net/minecraft/world/level/block/WeightedPressurePlateBlock.java
@@ -8,44 +8,48 @@
  {
     public static net.minecraft.world.level.block.state.properties.IntegerProperty POWER;
     public int maxWeight;
+    public net.minecraft.sounds.SoundEvent soundOff;
+    public net.minecraft.sounds.SoundEvent soundOn;
 
-    public net.minecraft.world.level.block.WeightedPressurePlateBlock(int maxWeight, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
+    public net.minecraft.world.level.block.WeightedPressurePlateBlock(int $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1, net.minecraft.sounds.SoundEvent $$2, net.minecraft.sounds.SoundEvent $$3) {
+        super($$1);
         this.registerDefaultState((net.minecraft.world.level.block.state.BlockState)((net.minecraft.world.level.block.state.BlockState)this.stateDefinition.any()).setValue((net.minecraft.world.level.block.state.properties.Property)POWER, (java.lang.Comparable)java.lang.Integer.valueOf((int)0)));
-        this.maxWeight = maxWeight;
+        this.maxWeight = $$0;
+        this.soundOff = $$2;
+        this.soundOn = $$3;
     }
 
-    public int getSignalStrength(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        int n = java.lang.Math.min((int)level.getEntitiesOfClass(net.minecraft.world.entity.Entity.class, TOUCH_AABB.move(pos)).size(), (int)this.maxWeight);
-        if (n > 0) {
-            float f = (float)java.lang.Math.min((int)this.maxWeight, (int)n) / (float)this.maxWeight;
-            return net.minecraft.util.Mth.ceil((float)(f * 15.0f));
+    public int getSignalStrength(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1) {
+        int $$2 = java.lang.Math.min((int)$$0.getEntitiesOfClass(net.minecraft.world.entity.Entity.class, TOUCH_AABB.move($$1)).size(), (int)this.maxWeight);
+        if ($$2 > 0) {
+            float $$3 = (float)java.lang.Math.min((int)this.maxWeight, (int)$$2) / (float)this.maxWeight;
+            return net.minecraft.util.Mth.ceil((float)($$3 * 15.0f));
         }
         return 0;
     }
 
-    public void playOnSound(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos) {
-        level.playSound(null, pos, net.minecraft.sounds.SoundEvents.METAL_PRESSURE_PLATE_CLICK_ON, net.minecraft.sounds.SoundSource.BLOCKS, 0.3f, 0.90000004f);
+    public void playOnSound(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos $$1) {
+        $$0.playSound(null, $$1, this.soundOn, net.minecraft.sounds.SoundSource.BLOCKS);
     }
 
-    public void playOffSound(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos) {
-        level.playSound(null, pos, net.minecraft.sounds.SoundEvents.METAL_PRESSURE_PLATE_CLICK_OFF, net.minecraft.sounds.SoundSource.BLOCKS, 0.3f, 0.75f);
+    public void playOffSound(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos $$1) {
+        $$0.playSound(null, $$1, this.soundOff, net.minecraft.sounds.SoundSource.BLOCKS);
     }
 
-    public int getSignalForState(net.minecraft.world.level.block.state.BlockState state) {
-        return (java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)POWER);
+    public int getSignalForState(net.minecraft.world.level.block.state.BlockState $$0) {
+        return (java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)POWER);
     }
 
-    public net.minecraft.world.level.block.state.BlockState setSignalForState(net.minecraft.world.level.block.state.BlockState state, int strength) {
-        return (net.minecraft.world.level.block.state.BlockState)state.setValue((net.minecraft.world.level.block.state.properties.Property)POWER, (java.lang.Comparable)java.lang.Integer.valueOf((int)strength));
+    public net.minecraft.world.level.block.state.BlockState setSignalForState(net.minecraft.world.level.block.state.BlockState $$0, int $$1) {
+        return (net.minecraft.world.level.block.state.BlockState)$$0.setValue((net.minecraft.world.level.block.state.properties.Property)POWER, (java.lang.Comparable)java.lang.Integer.valueOf((int)$$1));
     }
 
     public int getPressedTime() {
         return 10;
     }
 
-    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> builder) {
-        builder.add(new net.minecraft.world.level.block.state.properties.Property[]{POWER});
+    public void createBlockStateDefinition(net.minecraft.world.level.block.state.StateDefinition$Builder<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        $$0.add(new net.minecraft.world.level.block.state.properties.Property[]{POWER});
     }
 
     static {