--- net/minecraft/world/level/block/WetSpongeBlock.java
+++ net/minecraft/world/level/block/WetSpongeBlock.java
@@ -10,11 +10,11 @@
         super($$0);
     }
 
-    public void onPlace(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState oldState, boolean isMoving) {
-        if (level.dimensionType().ultraWarm()) {
-            level.setBlock(pos, net.minecraft.world.level.block.Blocks.SPONGE.defaultBlockState(), 3);
-            level.levelEvent(2009, pos, 0);
-            level.playSound(null, pos, net.minecraft.sounds.SoundEvents.FIRE_EXTINGUISH, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, (1.0f + level.getRandom().nextFloat() * 0.2f) * 0.7f);
+    public void onPlace(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, boolean $$4) {
+        if ($$1.dimensionType().ultraWarm()) {
+            $$1.setBlock($$2, net.minecraft.world.level.block.Blocks.SPONGE.defaultBlockState(), 3);
+            $$1.levelEvent(2009, $$2, 0);
+            $$1.playSound(null, $$2, net.minecraft.sounds.SoundEvents.FIRE_EXTINGUISH, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, (1.0f + $$1.getRandom().nextFloat() * 0.2f) * 0.7f);
         }
     }
 