--- net/minecraft/world/level/block/WitherWallSkullBlock.java
+++ net/minecraft/world/level/block/WitherWallSkullBlock.java
@@ -10,8 +10,8 @@
         super((net.minecraft.world.level.block.SkullBlock$Type)net.minecraft.world.level.block.SkullBlock$Types.WITHER_SKELETON, $$0);
     }
 
-    public void setPlacedBy(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity placer, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.world.level.block.Blocks.WITHER_SKELETON_SKULL.setPlacedBy(level, pos, state, placer, stack);
+    public void setPlacedBy(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.state.BlockState $$2, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$3, net.minecraft.world.item.ItemStack $$4) {
+        net.minecraft.world.level.block.Blocks.WITHER_SKELETON_SKULL.setPlacedBy($$0, $$1, $$2, $$3, $$4);
     }
 }
 