--- net/minecraft/world/level/block/WitherRoseBlock.java
+++ net/minecraft/world/level/block/WitherRoseBlock.java
@@ -6,12 +6,12 @@
 public class net.minecraft.world.level.block.WitherRoseBlock
 extends net.minecraft.world.level.block.FlowerBlock
  {
-    public net.minecraft.world.level.block.WitherRoseBlock(net.minecraft.world.effect.MobEffect suspiciousStewEffect, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(suspiciousStewEffect, 8, properties);
+    public net.minecraft.world.level.block.WitherRoseBlock(net.minecraft.world.effect.MobEffect $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$0, 8, $$1);
     }
 
-    public boolean mayPlaceOn(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        return super.mayPlaceOn(state, level, pos) || state.is(net.minecraft.world.level.block.Blocks.NETHERRACK) || state.is(net.minecraft.world.level.block.Blocks.SOUL_SAND) || state.is(net.minecraft.world.level.block.Blocks.SOUL_SOIL);
+    public boolean mayPlaceOn(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2) {
+        return super.mayPlaceOn($$0, $$1, $$2) || $$0.is(net.minecraft.world.level.block.Blocks.NETHERRACK) || $$0.is(net.minecraft.world.level.block.Blocks.SOUL_SAND) || $$0.is(net.minecraft.world.level.block.Blocks.SOUL_SOIL);
     }
 
     public void animateTick(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.util.RandomSource $$3) {
@@ -25,13 +25,13 @@
         }
     }
 
-    public void entityInside(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos, net.minecraft.world.entity.Entity entity) {
-        net.minecraft.world.entity.LivingEntity livingEntity;
-        if (level.isClientSide || level.getDifficulty() == net.minecraft.world.Difficulty.PEACEFUL) {
+    public void entityInside(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.entity.Entity $$3) {
+        net.minecraft.world.entity.LivingEntity $$4;
+        if ($$1.isClientSide || $$1.getDifficulty() == net.minecraft.world.Difficulty.PEACEFUL) {
             return;
         }
-        if (entity instanceof net.minecraft.world.entity.LivingEntity && !(livingEntity = (net.minecraft.world.entity.LivingEntity)entity).isInvulnerableTo(net.minecraft.world.damagesource.DamageSource.WITHER)) {
-            livingEntity.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.WITHER, 40));
+        if ($$3 instanceof net.minecraft.world.entity.LivingEntity && !($$4 = (net.minecraft.world.entity.LivingEntity)$$3).isInvulnerableTo(net.minecraft.world.damagesource.DamageSource.WITHER)) {
+            $$4.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.WITHER, 40));
         }
     }
 }