--- net/minecraft/world/level/block/WoodButtonBlock.java
+++ net/minecraft/world/level/block/WoodButtonBlock.java
@@ -1,17 +1,0 @@
-/*
- * Decompiled with CFR 0.151.
- */
-package net.minecraft.world.level.block;
-
-public class net.minecraft.world.level.block.WoodButtonBlock
-extends net.minecraft.world.level.block.ButtonBlock
- {
-    public net.minecraft.world.level.block.WoodButtonBlock(net.minecraft.world.level.block.state.BlockBehaviour$Properties $$0) {
-        super(true, $$0);
-    }
-
-    public net.minecraft.sounds.SoundEvent getSound(boolean isOn) {
-        return isOn ? net.minecraft.sounds.SoundEvents.WOODEN_BUTTON_CLICK_ON : net.minecraft.sounds.SoundEvents.WOODEN_BUTTON_CLICK_OFF;
-    }
-}
-