--- net/minecraft/world/level/block/WoolCarpetBlock.java
+++ net/minecraft/world/level/block/WoolCarpetBlock.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.item.DyeColor color;
 
-    public net.minecraft.world.level.block.WoolCarpetBlock(net.minecraft.world.item.DyeColor color, net.minecraft.world.level.block.state.BlockBehaviour$Properties properties) {
-        super(properties);
-        this.color = color;
+    public net.minecraft.world.level.block.WoolCarpetBlock(net.minecraft.world.item.DyeColor $$0, net.minecraft.world.level.block.state.BlockBehaviour$Properties $$1) {
+        super($$1);
+        this.color = $$0;
     }
 
     public net.minecraft.world.item.DyeColor getColor() {