--- net/minecraft/world/level/storage/loot/parameters/LootContextParamSet$Builder.java
+++ net/minecraft/world/level/storage/loot/parameters/LootContextParamSet$Builder.java
@@ -13,19 +13,19 @@
         this.optional = com.google.common.collect.Sets.newIdentityHashSet();
     }
 
-    public net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder required(net.minecraft.world.level.storage.loot.parameters.LootContextParam<?> parameter) {
-        if (this.optional.contains(parameter)) {
-            throw new java.lang.IllegalArgumentException("Parameter " + parameter.getName() + " is already optional");
+    public net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder required(net.minecraft.world.level.storage.loot.parameters.LootContextParam<?> $$0) {
+        if (this.optional.contains($$0)) {
+            throw new java.lang.IllegalArgumentException("Parameter " + $$0.getName() + " is already optional");
         }
-        this.required.add(parameter);
+        this.required.add($$0);
         return this;
     }
 
-    public net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder optional(net.minecraft.world.level.storage.loot.parameters.LootContextParam<?> parameter) {
-        if (this.required.contains(parameter)) {
-            throw new java.lang.IllegalArgumentException("Parameter " + parameter.getName() + " is already required");
+    public net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder optional(net.minecraft.world.level.storage.loot.parameters.LootContextParam<?> $$0) {
+        if (this.required.contains($$0)) {
+            throw new java.lang.IllegalArgumentException("Parameter " + $$0.getName() + " is already required");
         }
-        this.optional.add(parameter);
+        this.optional.add($$0);
         return this;
     }
 