--- net/minecraft/world/level/storage/loot/parameters/LootContextParams.java
+++ net/minecraft/world/level/storage/loot/parameters/LootContextParams.java
@@ -19,8 +19,8 @@
     public net.minecraft.world.level.storage.loot.parameters.LootContextParams() {
     }
 
-    public static <T> net.minecraft.world.level.storage.loot.parameters.LootContextParam<T> create(java.lang.String id) {
-        return new net.minecraft.world.level.storage.loot.parameters.LootContextParam(new net.minecraft.resources.ResourceLocation(id));
+    public static <T> net.minecraft.world.level.storage.loot.parameters.LootContextParam<T> create(java.lang.String $$0) {
+        return new net.minecraft.world.level.storage.loot.parameters.LootContextParam(new net.minecraft.resources.ResourceLocation($$0));
     }
 
     static {