--- net/minecraft/world/level/storage/loot/parameters/LootContextParamSets.java
+++ net/minecraft/world/level/storage/loot/parameters/LootContextParamSets.java
@@ -22,26 +22,26 @@
     public net.minecraft.world.level.storage.loot.parameters.LootContextParamSets() {
     }
 
-    public static net.minecraft.world.level.storage.loot.parameters.LootContextParamSet register(java.lang.String registryName, java.util.function.Consumer<net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder> builderConsumer) {
-        net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder builder = new net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder();
-        builderConsumer.accept((java.lang.Object)builder);
-        net.minecraft.world.level.storage.loot.parameters.LootContextParamSet lootContextParamSet = builder.build();
-        net.minecraft.resources.ResourceLocation resourceLocation = new net.minecraft.resources.ResourceLocation(registryName);
-        net.minecraft.world.level.storage.loot.parameters.LootContextParamSet lootContextParamSet2 = (net.minecraft.world.level.storage.loot.parameters.LootContextParamSet)REGISTRY.put((java.lang.Object)resourceLocation, (java.lang.Object)lootContextParamSet);
-        if (lootContextParamSet2 != null) {
-            throw new java.lang.IllegalStateException("Loot table parameter set " + resourceLocation + " is already registered");
+    public static net.minecraft.world.level.storage.loot.parameters.LootContextParamSet register(java.lang.String $$0, java.util.function.Consumer<net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder> $$1) {
+        net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder $$2 = new net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder();
+        $$1.accept((java.lang.Object)$$2);
+        net.minecraft.world.level.storage.loot.parameters.LootContextParamSet $$3 = $$2.build();
+        net.minecraft.resources.ResourceLocation $$4 = new net.minecraft.resources.ResourceLocation($$0);
+        net.minecraft.world.level.storage.loot.parameters.LootContextParamSet $$5 = (net.minecraft.world.level.storage.loot.parameters.LootContextParamSet)REGISTRY.put((java.lang.Object)$$4, (java.lang.Object)$$3);
+        if ($$5 != null) {
+            throw new java.lang.IllegalStateException("Loot table parameter set " + $$4 + " is already registered");
         }
-        return lootContextParamSet;
+        return $$3;
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.level.storage.loot.parameters.LootContextParamSet get(net.minecraft.resources.ResourceLocation registryName) {
-        return (net.minecraft.world.level.storage.loot.parameters.LootContextParamSet)REGISTRY.get((java.lang.Object)registryName);
+    public static net.minecraft.world.level.storage.loot.parameters.LootContextParamSet get(net.minecraft.resources.ResourceLocation $$0) {
+        return (net.minecraft.world.level.storage.loot.parameters.LootContextParamSet)REGISTRY.get((java.lang.Object)$$0);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.resources.ResourceLocation getKey(net.minecraft.world.level.storage.loot.parameters.LootContextParamSet parameterSet) {
-        return (net.minecraft.resources.ResourceLocation)REGISTRY.inverse().get((java.lang.Object)parameterSet);
+    public static net.minecraft.resources.ResourceLocation getKey(net.minecraft.world.level.storage.loot.parameters.LootContextParamSet $$0) {
+        return (net.minecraft.resources.ResourceLocation)REGISTRY.inverse().get((java.lang.Object)$$0);
     }
 
     public static /* synthetic */ void lambda$static$11(net.minecraft.world.level.storage.loot.parameters.LootContextParamSet$Builder $$0) {