--- net/minecraft/world/level/storage/loot/parameters/LootContextParam.java
+++ net/minecraft/world/level/storage/loot/parameters/LootContextParam.java
@@ -7,8 +7,8 @@
  {
     public net.minecraft.resources.ResourceLocation name;
 
-    public net.minecraft.world.level.storage.loot.parameters.LootContextParam(net.minecraft.resources.ResourceLocation name) {
-        this.name = name;
+    public net.minecraft.world.level.storage.loot.parameters.LootContextParam(net.minecraft.resources.ResourceLocation $$0) {
+        this.name = $$0;
     }
 
     public net.minecraft.resources.ResourceLocation getName() {