--- net/minecraft/world/level/storage/loot/parameters/LootContextParamSet.java
+++ net/minecraft/world/level/storage/loot/parameters/LootContextParamSet.java
@@ -8,9 +8,9 @@
     public java.util.Set<net.minecraft.world.level.storage.loot.parameters.LootContextParam<?>> required;
     public java.util.Set<net.minecraft.world.level.storage.loot.parameters.LootContextParam<?>> all;
 
-    public net.minecraft.world.level.storage.loot.parameters.LootContextParamSet(java.util.Set<net.minecraft.world.level.storage.loot.parameters.LootContextParam<?>> required, java.util.Set<net.minecraft.world.level.storage.loot.parameters.LootContextParam<?>> optional) {
-        this.required = com.google.common.collect.ImmutableSet.copyOf(required);
-        this.all = com.google.common.collect.ImmutableSet.copyOf((java.util.Collection)com.google.common.collect.Sets.union(required, optional));
+    public net.minecraft.world.level.storage.loot.parameters.LootContextParamSet(java.util.Set<net.minecraft.world.level.storage.loot.parameters.LootContextParam<?>> $$0, java.util.Set<net.minecraft.world.level.storage.loot.parameters.LootContextParam<?>> $$1) {
+        this.required = com.google.common.collect.ImmutableSet.copyOf($$0);
+        this.all = com.google.common.collect.ImmutableSet.copyOf((java.util.Collection)com.google.common.collect.Sets.union($$0, $$1));
     }
 
     public boolean isAllowed(net.minecraft.world.level.storage.loot.parameters.LootContextParam<?> $$0) {
@@ -29,11 +29,11 @@
         return "[" + com.google.common.base.Joiner.on((java.lang.String)", ").join(this.all.stream().map($$0 -> (this.required.contains($$0) ? "!" : "") + $$0.getName()).iterator()) + "]";
     }
 
-    public void validateUser(net.minecraft.world.level.storage.loot.ValidationContext validationContext, net.minecraft.world.level.storage.loot.LootContextUser lootContextUser) {
-        java.util.Set set = lootContextUser.getReferencedContextParams();
-        com.google.common.collect.Sets$SetView setView = com.google.common.collect.Sets.difference((java.util.Set)set, (java.util.Set)this.all);
-        if (!setView.isEmpty()) {
-            validationContext.reportProblem("Parameters " + (java.util.Set)setView + " are not provided in this context");
+    public void validateUser(net.minecraft.world.level.storage.loot.ValidationContext $$0, net.minecraft.world.level.storage.loot.LootContextUser $$1) {
+        java.util.Set $$2 = $$1.getReferencedContextParams();
+        com.google.common.collect.Sets$SetView $$3 = com.google.common.collect.Sets.difference((java.util.Set)$$2, (java.util.Set)this.all);
+        if (!$$3.isEmpty()) {
+            $$0.reportProblem("Parameters " + (java.util.Set)$$3 + " are not provided in this context");
         }
     }
 