--- net/minecraft/server/players/GameProfileCache.java
+++ net/minecraft/server/players/GameProfileCache.java
@@ -55,21 +55,21 @@
         return java.util.Optional.ofNullable((java.lang.Object)$$4);
     }
 
-    public static void setUsesAuthentication(boolean onlineMode) {
-        usesAuthentication = onlineMode;
+    public static void setUsesAuthentication(boolean $$0) {
+        usesAuthentication = $$0;
     }
 
     public static boolean usesAuthentication() {
         return usesAuthentication;
     }
 
-    public void add(com.mojang.authlib.GameProfile gameProfile) {
-        java.util.Calendar calendar = java.util.Calendar.getInstance();
-        calendar.setTime(new java.util.Date());
-        calendar.add(2, 1);
-        java.util.Date date = calendar.getTime();
-        net.minecraft.server.players.GameProfileCache$GameProfileInfo gameProfileInfo = new net.minecraft.server.players.GameProfileCache$GameProfileInfo(gameProfile, date);
-        this.safeAdd(gameProfileInfo);
+    public void add(com.mojang.authlib.GameProfile $$0) {
+        java.util.Calendar $$1 = java.util.Calendar.getInstance();
+        $$1.setTime(new java.util.Date());
+        $$1.add(2, 1);
+        java.util.Date $$2 = $$1.getTime();
+        net.minecraft.server.players.GameProfileCache$GameProfileInfo $$3 = new net.minecraft.server.players.GameProfileCache$GameProfileInfo($$0, $$2);
+        this.safeAdd($$3);
         this.save();
     }
 