--- net/minecraft/server/players/IpBanList.java
+++ net/minecraft/server/players/IpBanList.java
@@ -10,13 +10,13 @@
         super($$0);
     }
 
-    public net.minecraft.server.players.StoredUserEntry<java.lang.String> createEntry(com.google.gson.JsonObject entryData) {
-        return new net.minecraft.server.players.IpBanListEntry(entryData);
+    public net.minecraft.server.players.StoredUserEntry<java.lang.String> createEntry(com.google.gson.JsonObject $$0) {
+        return new net.minecraft.server.players.IpBanListEntry($$0);
     }
 
-    public boolean isBanned(java.net.SocketAddress address) {
-        java.lang.String string = this.getIpFromAddress(address);
-        return this.contains((java.lang.Object)string);
+    public boolean isBanned(java.net.SocketAddress $$0) {
+        java.lang.String $$1 = this.getIpFromAddress($$0);
+        return this.contains((java.lang.Object)$$1);
     }
 
     public boolean isBanned(java.lang.String $$0) {
@@ -24,20 +24,20 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.server.players.IpBanListEntry get(java.net.SocketAddress address) {
-        java.lang.String string = this.getIpFromAddress(address);
-        return (net.minecraft.server.players.IpBanListEntry)this.get((java.lang.Object)string);
+    public net.minecraft.server.players.IpBanListEntry get(java.net.SocketAddress $$0) {
+        java.lang.String $$1 = this.getIpFromAddress($$0);
+        return (net.minecraft.server.players.IpBanListEntry)this.get((java.lang.Object)$$1);
     }
 
-    public java.lang.String getIpFromAddress(java.net.SocketAddress address) {
-        java.lang.String string = address.toString();
-        if (string.contains((java.lang.CharSequence)"/")) {
-            string = string.substring(string.indexOf(47) + 1);
+    public java.lang.String getIpFromAddress(java.net.SocketAddress $$0) {
+        java.lang.String $$1 = $$0.toString();
+        if ($$1.contains((java.lang.CharSequence)"/")) {
+            $$1 = $$1.substring($$1.indexOf(47) + 1);
         }
-        if (string.contains((java.lang.CharSequence)":")) {
-            string = string.substring(0, string.indexOf(58));
+        if ($$1.contains((java.lang.CharSequence)":")) {
+            $$1 = $$1.substring(0, $$1.indexOf(58));
         }
-        return string;
+        return $$1;
     }
 }
 