--- net/minecraft/server/players/IpBanListEntry.java
+++ net/minecraft/server/players/IpBanListEntry.java
@@ -22,16 +22,16 @@
         super((java.lang.Object)net.minecraft.server.players.IpBanListEntry.createIpInfo((com.google.gson.JsonObject)$$0), $$0);
     }
 
-    public static java.lang.String createIpInfo(com.google.gson.JsonObject json) {
-        return json.has("ip") ? json.get("ip").getAsString() : null;
+    public static java.lang.String createIpInfo(com.google.gson.JsonObject $$0) {
+        return $$0.has("ip") ? $$0.get("ip").getAsString() : null;
     }
 
-    public void serialize(com.google.gson.JsonObject data) {
+    public void serialize(com.google.gson.JsonObject $$0) {
         if (this.getUser() == null) {
             return;
         }
-        data.addProperty("ip", (java.lang.String)this.getUser());
-        super.serialize(data);
+        $$0.addProperty("ip", (java.lang.String)this.getUser());
+        super.serialize($$0);
     }
 }
 