--- net/minecraft/server/players/PlayerList$1.java
+++ net/minecraft/server/players/PlayerList$1.java
@@ -12,30 +12,30 @@
         this.this$0 = $$0;
     }
 
-    public void onBorderSizeSet(net.minecraft.world.level.border.WorldBorder border, double newSize) {
-        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderSizePacket(border));
+    public void onBorderSizeSet(net.minecraft.world.level.border.WorldBorder $$0, double $$1) {
+        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderSizePacket($$0));
     }
 
-    public void onBorderSizeLerping(net.minecraft.world.level.border.WorldBorder border, double oldSize, double newSize, long time) {
-        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderLerpSizePacket(border));
+    public void onBorderSizeLerping(net.minecraft.world.level.border.WorldBorder $$0, double $$1, double $$2, long $$3) {
+        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderLerpSizePacket($$0));
     }
 
-    public void onBorderCenterSet(net.minecraft.world.level.border.WorldBorder border, double x, double z) {
-        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderCenterPacket(border));
+    public void onBorderCenterSet(net.minecraft.world.level.border.WorldBorder $$0, double $$1, double $$2) {
+        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderCenterPacket($$0));
     }
 
-    public void onBorderSetWarningTime(net.minecraft.world.level.border.WorldBorder border, int newTime) {
-        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderWarningDelayPacket(border));
+    public void onBorderSetWarningTime(net.minecraft.world.level.border.WorldBorder $$0, int $$1) {
+        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderWarningDelayPacket($$0));
     }
 
-    public void onBorderSetWarningBlocks(net.minecraft.world.level.border.WorldBorder border, int newDistance) {
-        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderWarningDistancePacket(border));
+    public void onBorderSetWarningBlocks(net.minecraft.world.level.border.WorldBorder $$0, int $$1) {
+        this.this$0.broadcastAll((net.minecraft.network.protocol.Packet)new net.minecraft.network.protocol.game.ClientboundSetBorderWarningDistancePacket($$0));
     }
 
-    public void onBorderSetDamagePerBlock(net.minecraft.world.level.border.WorldBorder border, double newAmount) {
+    public void onBorderSetDamagePerBlock(net.minecraft.world.level.border.WorldBorder $$0, double $$1) {
     }
 
-    public void onBorderSetDamageSafeZOne(net.minecraft.world.level.border.WorldBorder border, double newSize) {
+    public void onBorderSetDamageSafeZOne(net.minecraft.world.level.border.WorldBorder $$0, double $$1) {
     }
 }
 