--- net/minecraft/server/players/ServerOpList.java
+++ net/minecraft/server/players/ServerOpList.java
@@ -10,24 +10,24 @@
         super($$0);
     }
 
-    public net.minecraft.server.players.StoredUserEntry<com.mojang.authlib.GameProfile> createEntry(com.google.gson.JsonObject entryData) {
-        return new net.minecraft.server.players.ServerOpListEntry(entryData);
+    public net.minecraft.server.players.StoredUserEntry<com.mojang.authlib.GameProfile> createEntry(com.google.gson.JsonObject $$0) {
+        return new net.minecraft.server.players.ServerOpListEntry($$0);
     }
 
     public java.lang.String[] getUserList() {
         return (java.lang.String[])this.getEntries().stream().map(net.minecraft.server.players.StoredUserEntry::getUser).filter(java.util.Objects::nonNull).map(com.mojang.authlib.GameProfile::getName).toArray(java.lang.String[]::new);
     }
 
-    public boolean canBypassPlayerLimit(com.mojang.authlib.GameProfile profile) {
-        net.minecraft.server.players.ServerOpListEntry serverOpListEntry = (net.minecraft.server.players.ServerOpListEntry)this.get((java.lang.Object)profile);
-        if (serverOpListEntry != null) {
-            return serverOpListEntry.getBypassesPlayerLimit();
+    public boolean canBypassPlayerLimit(com.mojang.authlib.GameProfile $$0) {
+        net.minecraft.server.players.ServerOpListEntry $$1 = (net.minecraft.server.players.ServerOpListEntry)this.get((java.lang.Object)$$0);
+        if ($$1 != null) {
+            return $$1.getBypassesPlayerLimit();
         }
         return false;
     }
 
-    public java.lang.String getKeyForUser(com.mojang.authlib.GameProfile obj) {
-        return obj.getId().toString();
+    public java.lang.String getKeyForUser(com.mojang.authlib.GameProfile $$0) {
+        return $$0.getId().toString();
     }
 
     public /* synthetic */ java.lang.String getKeyForUser(java.lang.Object object) {