--- net/minecraft/server/players/UserBanList.java
+++ net/minecraft/server/players/UserBanList.java
@@ -10,20 +10,20 @@
         super($$0);
     }
 
-    public net.minecraft.server.players.StoredUserEntry<com.mojang.authlib.GameProfile> createEntry(com.google.gson.JsonObject entryData) {
-        return new net.minecraft.server.players.UserBanListEntry(entryData);
+    public net.minecraft.server.players.StoredUserEntry<com.mojang.authlib.GameProfile> createEntry(com.google.gson.JsonObject $$0) {
+        return new net.minecraft.server.players.UserBanListEntry($$0);
     }
 
-    public boolean isBanned(com.mojang.authlib.GameProfile profile) {
-        return this.contains((java.lang.Object)profile);
+    public boolean isBanned(com.mojang.authlib.GameProfile $$0) {
+        return this.contains((java.lang.Object)$$0);
     }
 
     public java.lang.String[] getUserList() {
         return (java.lang.String[])this.getEntries().stream().map(net.minecraft.server.players.StoredUserEntry::getUser).filter(java.util.Objects::nonNull).map(com.mojang.authlib.GameProfile::getName).toArray(java.lang.String[]::new);
     }
 
-    public java.lang.String getKeyForUser(com.mojang.authlib.GameProfile obj) {
-        return obj.getId().toString();
+    public java.lang.String getKeyForUser(com.mojang.authlib.GameProfile $$0) {
+        return $$0.getId().toString();
     }
 
     public /* synthetic */ java.lang.String getKeyForUser(java.lang.Object object) {