--- net/minecraft/data/models/BlockModelGenerators$WoodProvider.java
+++ net/minecraft/data/models/BlockModelGenerators$WoodProvider.java
@@ -8,28 +8,28 @@
     public net.minecraft.data.models.model.TextureMapping logMapping;
     public /* synthetic */ net.minecraft.data.models.BlockModelGenerators this$0;
 
-    public net.minecraft.data.models.BlockModelGenerators$WoodProvider(net.minecraft.data.models.BlockModelGenerators blockModelGenerators, net.minecraft.data.models.model.TextureMapping logMapping) {
+    public net.minecraft.data.models.BlockModelGenerators$WoodProvider(net.minecraft.data.models.BlockModelGenerators blockModelGenerators, net.minecraft.data.models.model.TextureMapping $$0) {
         this.this$0 = blockModelGenerators;
-        this.logMapping = logMapping;
+        this.logMapping = $$0;
     }
 
-    public net.minecraft.data.models.BlockModelGenerators$WoodProvider wood(net.minecraft.world.level.block.Block woodBlock) {
-        net.minecraft.data.models.model.TextureMapping textureMapping = this.logMapping.copyAndUpdate(net.minecraft.data.models.model.TextureSlot.END, this.logMapping.get(net.minecraft.data.models.model.TextureSlot.SIDE));
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN.create(woodBlock, textureMapping, this.this$0.modelOutput);
-        this.this$0.blockStateOutput.accept((java.lang.Object)net.minecraft.data.models.BlockModelGenerators.createAxisAlignedPillarBlock((net.minecraft.world.level.block.Block)woodBlock, (net.minecraft.resources.ResourceLocation)resourceLocation));
+    public net.minecraft.data.models.BlockModelGenerators$WoodProvider wood(net.minecraft.world.level.block.Block $$0) {
+        net.minecraft.data.models.model.TextureMapping $$1 = this.logMapping.copyAndUpdate(net.minecraft.data.models.model.TextureSlot.END, this.logMapping.get(net.minecraft.data.models.model.TextureSlot.SIDE));
+        net.minecraft.resources.ResourceLocation $$2 = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN.create($$0, $$1, this.this$0.modelOutput);
+        this.this$0.blockStateOutput.accept((java.lang.Object)net.minecraft.data.models.BlockModelGenerators.createAxisAlignedPillarBlock((net.minecraft.world.level.block.Block)$$0, (net.minecraft.resources.ResourceLocation)$$2));
         return this;
     }
 
-    public net.minecraft.data.models.BlockModelGenerators$WoodProvider log(net.minecraft.world.level.block.Block logBlock) {
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN.create(logBlock, this.logMapping, this.this$0.modelOutput);
-        this.this$0.blockStateOutput.accept((java.lang.Object)net.minecraft.data.models.BlockModelGenerators.createAxisAlignedPillarBlock((net.minecraft.world.level.block.Block)logBlock, (net.minecraft.resources.ResourceLocation)resourceLocation));
+    public net.minecraft.data.models.BlockModelGenerators$WoodProvider log(net.minecraft.world.level.block.Block $$0) {
+        net.minecraft.resources.ResourceLocation $$1 = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN.create($$0, this.logMapping, this.this$0.modelOutput);
+        this.this$0.blockStateOutput.accept((java.lang.Object)net.minecraft.data.models.BlockModelGenerators.createAxisAlignedPillarBlock((net.minecraft.world.level.block.Block)$$0, (net.minecraft.resources.ResourceLocation)$$1));
         return this;
     }
 
-    public net.minecraft.data.models.BlockModelGenerators$WoodProvider logWithHorizontal(net.minecraft.world.level.block.Block logBlock) {
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN.create(logBlock, this.logMapping, this.this$0.modelOutput);
-        net.minecraft.resources.ResourceLocation resourceLocation2 = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN_HORIZONTAL.create(logBlock, this.logMapping, this.this$0.modelOutput);
-        this.this$0.blockStateOutput.accept((java.lang.Object)net.minecraft.data.models.BlockModelGenerators.createRotatedPillarWithHorizontalVariant((net.minecraft.world.level.block.Block)logBlock, (net.minecraft.resources.ResourceLocation)resourceLocation, (net.minecraft.resources.ResourceLocation)resourceLocation2));
+    public net.minecraft.data.models.BlockModelGenerators$WoodProvider logWithHorizontal(net.minecraft.world.level.block.Block $$0) {
+        net.minecraft.resources.ResourceLocation $$1 = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN.create($$0, this.logMapping, this.this$0.modelOutput);
+        net.minecraft.resources.ResourceLocation $$2 = net.minecraft.data.models.model.ModelTemplates.CUBE_COLUMN_HORIZONTAL.create($$0, this.logMapping, this.this$0.modelOutput);
+        this.this$0.blockStateOutput.accept((java.lang.Object)net.minecraft.data.models.BlockModelGenerators.createRotatedPillarWithHorizontalVariant((net.minecraft.world.level.block.Block)$$0, (net.minecraft.resources.ResourceLocation)$$1, (net.minecraft.resources.ResourceLocation)$$2));
         return this;
     }
 }