--- net/minecraft/data/models/ItemModelGenerators.java
+++ net/minecraft/data/models/ItemModelGenerators.java
@@ -7,20 +7,20 @@
  {
     public java.util.function.BiConsumer<net.minecraft.resources.ResourceLocation, java.util.function.Supplier<com.google.gson.JsonElement>> output;
 
-    public net.minecraft.data.models.ItemModelGenerators(java.util.function.BiConsumer<net.minecraft.resources.ResourceLocation, java.util.function.Supplier<com.google.gson.JsonElement>> output) {
-        this.output = output;
+    public net.minecraft.data.models.ItemModelGenerators(java.util.function.BiConsumer<net.minecraft.resources.ResourceLocation, java.util.function.Supplier<com.google.gson.JsonElement>> $$0) {
+        this.output = $$0;
     }
 
-    public void generateFlatItem(net.minecraft.world.item.Item item, net.minecraft.data.models.model.ModelTemplate modelTemplate) {
-        modelTemplate.create(net.minecraft.data.models.model.ModelLocationUtils.getModelLocation((net.minecraft.world.item.Item)item), net.minecraft.data.models.model.TextureMapping.layer0((net.minecraft.world.item.Item)item), this.output);
+    public void generateFlatItem(net.minecraft.world.item.Item $$0, net.minecraft.data.models.model.ModelTemplate $$1) {
+        $$1.create(net.minecraft.data.models.model.ModelLocationUtils.getModelLocation((net.minecraft.world.item.Item)$$0), net.minecraft.data.models.model.TextureMapping.layer0((net.minecraft.world.item.Item)$$0), this.output);
     }
 
-    public void generateFlatItem(net.minecraft.world.item.Item item, java.lang.String modelLocationSuffix, net.minecraft.data.models.model.ModelTemplate modelTemplate) {
-        modelTemplate.create(net.minecraft.data.models.model.ModelLocationUtils.getModelLocation((net.minecraft.world.item.Item)item, (java.lang.String)modelLocationSuffix), net.minecraft.data.models.model.TextureMapping.layer0((net.minecraft.resources.ResourceLocation)net.minecraft.data.models.model.TextureMapping.getItemTexture((net.minecraft.world.item.Item)item, (java.lang.String)modelLocationSuffix)), this.output);
+    public void generateFlatItem(net.minecraft.world.item.Item $$0, java.lang.String $$1, net.minecraft.data.models.model.ModelTemplate $$2) {
+        $$2.create(net.minecraft.data.models.model.ModelLocationUtils.getModelLocation((net.minecraft.world.item.Item)$$0, (java.lang.String)$$1), net.minecraft.data.models.model.TextureMapping.layer0((net.minecraft.resources.ResourceLocation)net.minecraft.data.models.model.TextureMapping.getItemTexture((net.minecraft.world.item.Item)$$0, (java.lang.String)$$1)), this.output);
     }
 
-    public void generateFlatItem(net.minecraft.world.item.Item item, net.minecraft.world.item.Item layerZeroItem, net.minecraft.data.models.model.ModelTemplate modelTemplate) {
-        modelTemplate.create(net.minecraft.data.models.model.ModelLocationUtils.getModelLocation((net.minecraft.world.item.Item)item), net.minecraft.data.models.model.TextureMapping.layer0((net.minecraft.world.item.Item)layerZeroItem), this.output);
+    public void generateFlatItem(net.minecraft.world.item.Item $$0, net.minecraft.world.item.Item $$1, net.minecraft.data.models.model.ModelTemplate $$2) {
+        $$2.create(net.minecraft.data.models.model.ModelLocationUtils.getModelLocation((net.minecraft.world.item.Item)$$0), net.minecraft.data.models.model.TextureMapping.layer0((net.minecraft.world.item.Item)$$1), this.output);
     }
 
     public void generateCompassItem(net.minecraft.world.item.Item $$0) {
@@ -178,6 +178,8 @@
         this.generateFlatItem(net.minecraft.world.item.Items.MAGMA_CREAM, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);
         this.generateFlatItem(net.minecraft.world.item.Items.MANGROVE_BOAT, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);
         this.generateFlatItem(net.minecraft.world.item.Items.MANGROVE_CHEST_BOAT, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);
+        this.generateFlatItem(net.minecraft.world.item.Items.BAMBOO_RAFT, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);
+        this.generateFlatItem(net.minecraft.world.item.Items.BAMBOO_CHEST_RAFT, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);
         this.generateFlatItem(net.minecraft.world.item.Items.MAP, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);
         this.generateFlatItem(net.minecraft.world.item.Items.MELON_SLICE, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);
         this.generateFlatItem(net.minecraft.world.item.Items.MILK_BUCKET, net.minecraft.data.models.model.ModelTemplates.FLAT_ITEM);