--- net/minecraft/world/entity/ai/control/LookControl.java
+++ net/minecraft/world/entity/ai/control/LookControl.java
@@ -18,28 +18,28 @@
         this.mob = $$0;
     }
 
-    public void setLookAt(net.minecraft.world.phys.Vec3 lookVector) {
-        this.setLookAt(lookVector.x, lookVector.y, lookVector.z);
+    public void setLookAt(net.minecraft.world.phys.Vec3 $$0) {
+        this.setLookAt($$0.x, $$0.y, $$0.z);
     }
 
     public void setLookAt(net.minecraft.world.entity.Entity $$0) {
         this.setLookAt($$0.getX(), net.minecraft.world.entity.ai.control.LookControl.getWantedY((net.minecraft.world.entity.Entity)$$0), $$0.getZ());
     }
 
-    public void setLookAt(net.minecraft.world.entity.Entity entity, float deltaYaw, float deltaPitch) {
-        this.setLookAt(entity.getX(), net.minecraft.world.entity.ai.control.LookControl.getWantedY((net.minecraft.world.entity.Entity)entity), entity.getZ(), deltaYaw, deltaPitch);
+    public void setLookAt(net.minecraft.world.entity.Entity $$0, float $$1, float $$2) {
+        this.setLookAt($$0.getX(), net.minecraft.world.entity.ai.control.LookControl.getWantedY((net.minecraft.world.entity.Entity)$$0), $$0.getZ(), $$1, $$2);
     }
 
-    public void setLookAt(double x, double y, double z) {
-        this.setLookAt(x, y, z, (float)this.mob.getHeadRotSpeed(), (float)this.mob.getMaxHeadXRot());
+    public void setLookAt(double $$0, double $$1, double $$2) {
+        this.setLookAt($$0, $$1, $$2, (float)this.mob.getHeadRotSpeed(), (float)this.mob.getMaxHeadXRot());
     }
 
-    public void setLookAt(double x, double y, double z, float deltaYaw, float deltaPitch) {
-        this.wantedX = x;
-        this.wantedY = y;
-        this.wantedZ = z;
-        this.yMaxRotSpeed = deltaYaw;
-        this.xMaxRotAngle = deltaPitch;
+    public void setLookAt(double $$0, double $$1, double $$2, float $$3, float $$4) {
+        this.wantedX = $$0;
+        this.wantedY = $$1;
+        this.wantedZ = $$2;
+        this.yMaxRotSpeed = $$3;
+        this.xMaxRotAngle = $$4;
         this.lookAtCooldown = 2;
     }
 
@@ -99,17 +99,17 @@
         return java.lang.Math.abs((double)$$1) > (double)1.0E-5f || java.lang.Math.abs((double)$$0) > (double)1.0E-5f ? java.util.Optional.of((java.lang.Object)java.lang.Float.valueOf((float)((float)(net.minecraft.util.Mth.atan2((double)$$1, (double)$$0) * 57.2957763671875) - 90.0f))) : java.util.Optional.empty();
     }
 
-    public float rotateTowards(float from, float to, float maxDelta) {
-        float f = net.minecraft.util.Mth.degreesDifference((float)from, (float)to);
-        float f2 = net.minecraft.util.Mth.clamp((float)f, (float)(-maxDelta), (float)maxDelta);
-        return from + f2;
+    public float rotateTowards(float $$0, float $$1, float $$2) {
+        float $$3 = net.minecraft.util.Mth.degreesDifference((float)$$0, (float)$$1);
+        float $$4 = net.minecraft.util.Mth.clamp((float)$$3, (float)(-$$2), (float)$$2);
+        return $$0 + $$4;
     }
 
-    public static double getWantedY(net.minecraft.world.entity.Entity entity) {
-        if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-            return entity.getEyeY();
+    public static double getWantedY(net.minecraft.world.entity.Entity $$0) {
+        if ($$0 instanceof net.minecraft.world.entity.LivingEntity) {
+            return $$0.getEyeY();
         }
-        return (entity.getBoundingBox().minY + entity.getBoundingBox().maxY) / 2.0;
+        return ($$0.getBoundingBox().minY + $$0.getBoundingBox().maxY) / 2.0;
     }
 }
 