--- net/minecraft/client/gui/screens/social/PlayerEntry.java
+++ net/minecraft/client/gui/screens/social/PlayerEntry.java
@@ -92,41 +92,41 @@
         return $$0 ? net.minecraft.network.chat.Component.translatable((java.lang.String)"gui.socialInteractions.narration.report", (java.lang.Object[])new java.lang.Object[]{this.playerName}) : REPORT_PLAYER_TOOLTIP;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        int n;
-        int n2 = left + 4;
-        int n3 = top + (height - 24) / 2;
-        int n4 = n2 + 24 + 4;
-        net.minecraft.network.chat.Component component = this.getStatusComponent();
-        if (component == net.minecraft.network.chat.CommonComponents.EMPTY) {
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (int)(left + width), (int)(top + height), (int)BG_FILL);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        int $$15;
+        int $$10 = $$3 + 4;
+        int $$11 = $$2 + ($$5 - 24) / 2;
+        int $$12 = $$10 + 24 + 4;
+        net.minecraft.network.chat.Component $$13 = this.getStatusComponent();
+        if ($$13 == net.minecraft.network.chat.CommonComponents.EMPTY) {
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (int)($$3 + $$4), (int)($$2 + $$5), (int)BG_FILL);
             java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-            n = top + (height - 9) / 2;
+            int $$14 = $$2 + ($$5 - 9) / 2;
         } else {
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (int)(left + width), (int)(top + height), (int)BG_FILL_REMOVED);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (int)($$3 + $$4), (int)($$2 + $$5), (int)BG_FILL_REMOVED);
             java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
             java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-            n = top + (height - (9 + 9)) / 2;
-            this.minecraft.font.draw(matrixStack, component, (float)n4, (float)(n + 12), PLAYER_STATUS_COLOR);
+            $$15 = $$2 + ($$5 - (9 + 9)) / 2;
+            this.minecraft.font.draw($$0, $$13, (float)$$12, (float)($$15 + 12), PLAYER_STATUS_COLOR);
         }
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)((net.minecraft.resources.ResourceLocation)this.skinGetter.get()));
-        net.minecraft.client.gui.components.PlayerFaceRenderer.draw((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)n2, (int)n3, (int)24);
-        this.minecraft.font.draw(matrixStack, this.playerName, (float)n4, (float)n, PLAYERNAME_COLOR);
+        net.minecraft.client.gui.components.PlayerFaceRenderer.draw((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$10, (int)$$11, (int)24);
+        this.minecraft.font.draw($$0, this.playerName, (float)$$12, (float)$$15, PLAYERNAME_COLOR);
         if (this.isRemoved) {
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)n2, (int)n3, (int)(n2 + 24), (int)(n3 + 24), (int)SKIN_SHADE);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$10, (int)$$11, (int)($$10 + 24), (int)($$11 + 24), (int)SKIN_SHADE);
         }
         if (this.hideButton != null && this.showButton != null && this.reportButton != null) {
-            float f = this.tooltipHoverTime;
-            this.hideButton.x = left + (width - this.hideButton.getWidth() - 4) - 20 - 4;
-            this.hideButton.y = top + (height - this.hideButton.getHeight()) / 2;
-            this.hideButton.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.showButton.x = left + (width - this.showButton.getWidth() - 4) - 20 - 4;
-            this.showButton.y = top + (height - this.showButton.getHeight()) / 2;
-            this.showButton.render(matrixStack, mouseX, mouseY, partialTicks);
-            this.reportButton.x = left + (width - this.showButton.getWidth() - 4);
-            this.reportButton.y = top + (height - this.showButton.getHeight()) / 2;
-            this.reportButton.render(matrixStack, mouseX, mouseY, partialTicks);
-            if (f == this.tooltipHoverTime) {
+            float $$16 = this.tooltipHoverTime;
+            this.hideButton.x = $$3 + ($$4 - this.hideButton.getWidth() - 4) - 20 - 4;
+            this.hideButton.y = $$2 + ($$5 - this.hideButton.getHeight()) / 2;
+            this.hideButton.render($$0, $$6, $$7, $$9);
+            this.showButton.x = $$3 + ($$4 - this.showButton.getWidth() - 4) - 20 - 4;
+            this.showButton.y = $$2 + ($$5 - this.showButton.getHeight()) / 2;
+            this.showButton.render($$0, $$6, $$7, $$9);
+            this.reportButton.x = $$3 + ($$4 - this.showButton.getWidth() - 4);
+            this.reportButton.y = $$2 + ($$5 - this.showButton.getHeight()) / 2;
+            this.reportButton.render($$0, $$6, $$7, $$9);
+            if ($$16 == this.tooltipHoverTime) {
                 this.tooltipHoverTime = 0.0f;
             }
         }