--- net/minecraft/network/protocol/game/ClientboundOpenBookPacket.java
+++ net/minecraft/network/protocol/game/ClientboundOpenBookPacket.java
@@ -8,20 +8,20 @@
  {
     public net.minecraft.world.InteractionHand hand;
 
-    public net.minecraft.network.protocol.game.ClientboundOpenBookPacket(net.minecraft.world.InteractionHand hand) {
-        this.hand = hand;
+    public net.minecraft.network.protocol.game.ClientboundOpenBookPacket(net.minecraft.world.InteractionHand $$0) {
+        this.hand = $$0;
     }
 
     public net.minecraft.network.protocol.game.ClientboundOpenBookPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.hand = (net.minecraft.world.InteractionHand)$$0.readEnum(net.minecraft.world.InteractionHand.class);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeEnum((java.lang.Enum)this.hand);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeEnum((java.lang.Enum)this.hand);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleOpenBook(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleOpenBook(this);
     }
 
     public net.minecraft.world.InteractionHand getHand() {