--- net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
+++ net/minecraft/network/protocol/game/ClientboundRotateHeadPacket.java
@@ -9,9 +9,9 @@
     public int entityId;
     public byte yHeadRot;
 
-    public net.minecraft.network.protocol.game.ClientboundRotateHeadPacket(net.minecraft.world.entity.Entity entity, byte yHeadRot) {
-        this.entityId = entity.getId();
-        this.yHeadRot = yHeadRot;
+    public net.minecraft.network.protocol.game.ClientboundRotateHeadPacket(net.minecraft.world.entity.Entity $$0, byte $$1) {
+        this.entityId = $$0.getId();
+        this.yHeadRot = $$1;
     }
 
     public net.minecraft.network.protocol.game.ClientboundRotateHeadPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,17 +19,17 @@
         this.yHeadRot = $$0.readByte();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entityId);
-        buffer.writeByte((int)this.yHeadRot);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.entityId);
+        $$0.writeByte((int)this.yHeadRot);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleRotateMob(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleRotateMob(this);
     }
 
-    public net.minecraft.world.entity.Entity getEntity(net.minecraft.world.level.Level level) {
-        return level.getEntity(this.entityId);
+    public net.minecraft.world.entity.Entity getEntity(net.minecraft.world.level.Level $$0) {
+        return $$0.getEntity(this.entityId);
     }
 
     public byte getYHeadRot() {