--- net/minecraft/network/protocol/game/ClientboundLoginPacket.java
+++ net/minecraft/network/protocol/game/ClientboundLoginPacket.java
@@ -27,7 +27,7 @@
     public java.util.Optional<net.minecraft.core.GlobalPos> lastDeathLocation;
 
     public net.minecraft.network.protocol.game.ClientboundLoginPacket(net.minecraft.network.FriendlyByteBuf $$02) {
-        this($$02.readInt(), $$02.readBoolean(), net.minecraft.world.level.GameType.byId((int)$$02.readByte()), net.minecraft.world.level.GameType.byNullableId((int)$$02.readByte()), (java.util.Set)$$02.readCollection(com.google.common.collect.Sets::newHashSetWithExpectedSize, $$0 -> $$0.readResourceKey(net.minecraft.core.Registry.DIMENSION_REGISTRY)), ((net.minecraft.core.RegistryAccess)$$02.readWithCodec(net.minecraft.core.RegistryAccess.NETWORK_CODEC)).freeze(), $$02.readResourceKey(net.minecraft.core.Registry.DIMENSION_TYPE_REGISTRY), $$02.readResourceKey(net.minecraft.core.Registry.DIMENSION_REGISTRY), $$02.readLong(), $$02.readVarInt(), $$02.readVarInt(), $$02.readVarInt(), $$02.readBoolean(), $$02.readBoolean(), $$02.readBoolean(), $$02.readBoolean(), $$02.readOptional(net.minecraft.network.FriendlyByteBuf::readGlobalPos));
+        this($$02.readInt(), $$02.readBoolean(), net.minecraft.world.level.GameType.byId((int)$$02.readByte()), net.minecraft.world.level.GameType.byNullableId((int)$$02.readByte()), (java.util.Set)$$02.readCollection(com.google.common.collect.Sets::newHashSetWithExpectedSize, $$0 -> $$0.readResourceKey(net.minecraft.core.Registry.DIMENSION_REGISTRY)), ((net.minecraft.core.RegistryAccess)$$02.readWithCodec(net.minecraft.core.RegistrySynchronization.NETWORK_CODEC)).freeze(), $$02.readResourceKey(net.minecraft.core.Registry.DIMENSION_TYPE_REGISTRY), $$02.readResourceKey(net.minecraft.core.Registry.DIMENSION_REGISTRY), $$02.readLong(), $$02.readVarInt(), $$02.readVarInt(), $$02.readVarInt(), $$02.readBoolean(), $$02.readBoolean(), $$02.readBoolean(), $$02.readBoolean(), $$02.readOptional(net.minecraft.network.FriendlyByteBuf::readGlobalPos));
     }
 
     public net.minecraft.network.protocol.game.ClientboundLoginPacket(int $$0, boolean $$1, net.minecraft.world.level.GameType $$2, @javax.annotation.Nullable net.minecraft.world.level.GameType $$3, java.util.Set<net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level>> $$4, net.minecraft.core.RegistryAccess$Frozen $$5, net.minecraft.resources.ResourceKey<net.minecraft.world.level.dimension.DimensionType> $$6, net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$7, long $$8, int $$9, int $$10, int $$11, boolean $$12, boolean $$13, boolean $$14, boolean $$15, java.util.Optional<net.minecraft.core.GlobalPos> $$16) {
@@ -50,28 +50,28 @@
         this.lastDeathLocation = $$16;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeInt(this.playerId);
-        buffer.writeBoolean(this.hardcore);
-        buffer.writeByte(this.gameType.getId());
-        buffer.writeByte(net.minecraft.world.level.GameType.getNullableId((net.minecraft.world.level.GameType)this.previousGameType));
-        buffer.writeCollection((java.util.Collection)this.levels, net.minecraft.network.FriendlyByteBuf::writeResourceKey);
-        buffer.writeWithCodec(net.minecraft.core.RegistryAccess.NETWORK_CODEC, (java.lang.Object)this.registryHolder);
-        buffer.writeResourceKey(this.dimensionType);
-        buffer.writeResourceKey(this.dimension);
-        buffer.writeLong(this.seed);
-        buffer.writeVarInt(this.maxPlayers);
-        buffer.writeVarInt(this.chunkRadius);
-        buffer.writeVarInt(this.simulationDistance);
-        buffer.writeBoolean(this.reducedDebugInfo);
-        buffer.writeBoolean(this.showDeathScreen);
-        buffer.writeBoolean(this.isDebug);
-        buffer.writeBoolean(this.isFlat);
-        buffer.writeOptional(this.lastDeathLocation, net.minecraft.network.FriendlyByteBuf::writeGlobalPos);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeInt(this.playerId);
+        $$0.writeBoolean(this.hardcore);
+        $$0.writeByte(this.gameType.getId());
+        $$0.writeByte(net.minecraft.world.level.GameType.getNullableId((net.minecraft.world.level.GameType)this.previousGameType));
+        $$0.writeCollection((java.util.Collection)this.levels, net.minecraft.network.FriendlyByteBuf::writeResourceKey);
+        $$0.writeWithCodec(net.minecraft.core.RegistrySynchronization.NETWORK_CODEC, (java.lang.Object)this.registryHolder);
+        $$0.writeResourceKey(this.dimensionType);
+        $$0.writeResourceKey(this.dimension);
+        $$0.writeLong(this.seed);
+        $$0.writeVarInt(this.maxPlayers);
+        $$0.writeVarInt(this.chunkRadius);
+        $$0.writeVarInt(this.simulationDistance);
+        $$0.writeBoolean(this.reducedDebugInfo);
+        $$0.writeBoolean(this.showDeathScreen);
+        $$0.writeBoolean(this.isDebug);
+        $$0.writeBoolean(this.isFlat);
+        $$0.writeOptional(this.lastDeathLocation, net.minecraft.network.FriendlyByteBuf::writeGlobalPos);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleLogin(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleLogin(this);
     }
 
     public java.lang.String toString() {