--- net/minecraft/network/protocol/game/ClientboundSoundPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSoundPacket.java
@@ -39,15 +39,15 @@
         this.seed = $$0.readLong();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.SOUND_EVENT, (java.lang.Object)this.sound);
-        buffer.writeEnum((java.lang.Enum)this.source);
-        buffer.writeInt(this.x);
-        buffer.writeInt(this.y);
-        buffer.writeInt(this.z);
-        buffer.writeFloat(this.volume);
-        buffer.writeFloat(this.pitch);
-        buffer.writeLong(this.seed);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.SOUND_EVENT, (java.lang.Object)this.sound);
+        $$0.writeEnum((java.lang.Enum)this.source);
+        $$0.writeInt(this.x);
+        $$0.writeInt(this.y);
+        $$0.writeInt(this.z);
+        $$0.writeFloat(this.volume);
+        $$0.writeFloat(this.pitch);
+        $$0.writeLong(this.seed);
     }
 
     public net.minecraft.sounds.SoundEvent getSound() {
@@ -82,8 +82,8 @@
         return this.seed;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSoundEvent(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSoundEvent(this);
     }
 }
 