--- net/minecraft/network/protocol/game/ClientboundSetHealthPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSetHealthPacket.java
@@ -10,10 +10,10 @@
     public int food;
     public float saturation;
 
-    public net.minecraft.network.protocol.game.ClientboundSetHealthPacket(float health, int food, float saturation) {
-        this.health = health;
-        this.food = food;
-        this.saturation = saturation;
+    public net.minecraft.network.protocol.game.ClientboundSetHealthPacket(float $$0, int $$1, float $$2) {
+        this.health = $$0;
+        this.food = $$1;
+        this.saturation = $$2;
     }
 
     public net.minecraft.network.protocol.game.ClientboundSetHealthPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -22,14 +22,14 @@
         this.saturation = $$0.readFloat();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeFloat(this.health);
-        buffer.writeVarInt(this.food);
-        buffer.writeFloat(this.saturation);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeFloat(this.health);
+        $$0.writeVarInt(this.food);
+        $$0.writeFloat(this.saturation);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSetHealth(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSetHealth(this);
     }
 
     public float getHealth() {