--- net/minecraft/network/protocol/game/ClientboundDeleteChatPacket.java
+++ net/minecraft/network/protocol/game/ClientboundDeleteChatPacket.java
@@ -7,18 +7,18 @@
 extends java.lang.Record
 implements net.minecraft.network.protocol.Packet<net.minecraft.network.protocol.game.ClientGamePacketListener>
  {
-    public net.minecraft.network.chat.MessageSignature messageSignature;
+    public net.minecraft.network.chat.MessageSignature$Packed messageSignature;
 
     public net.minecraft.network.protocol.game.ClientboundDeleteChatPacket(net.minecraft.network.FriendlyByteBuf $$0) {
-        this(new net.minecraft.network.chat.MessageSignature($$0));
+        this(net.minecraft.network.chat.MessageSignature$Packed.read((net.minecraft.network.FriendlyByteBuf)$$0));
     }
 
-    public net.minecraft.network.protocol.game.ClientboundDeleteChatPacket(net.minecraft.network.chat.MessageSignature $$0) {
+    public net.minecraft.network.protocol.game.ClientboundDeleteChatPacket(net.minecraft.network.chat.MessageSignature$Packed $$0) {
         this.messageSignature = $$0;
     }
 
     public void write(net.minecraft.network.FriendlyByteBuf $$0) {
-        this.messageSignature.write($$0);
+        net.minecraft.network.chat.MessageSignature$Packed.write((net.minecraft.network.FriendlyByteBuf)$$0, (net.minecraft.network.chat.MessageSignature$Packed)this.messageSignature);
     }
 
     public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
@@ -37,7 +37,7 @@
         return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ClientboundDeleteChatPacket.class, "messageSignature", "messageSignature"}, this, $$0);
     }
 
-    public net.minecraft.network.chat.MessageSignature messageSignature() {
+    public net.minecraft.network.chat.MessageSignature$Packed messageSignature() {
         return this.messageSignature;
     }
 }