--- net/minecraft/network/protocol/game/ClientboundSetPlayerTeamPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSetPlayerTeamPacket.java
@@ -44,14 +44,14 @@
         this.players = net.minecraft.network.protocol.game.ClientboundSetPlayerTeamPacket.shouldHavePlayerList((int)this.method) ? $$0.readList(net.minecraft.network.FriendlyByteBuf::readUtf) : com.google.common.collect.ImmutableList.of();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUtf(this.name);
-        buffer.writeByte(this.method);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUtf(this.name);
+        $$0.writeByte(this.method);
         if (net.minecraft.network.protocol.game.ClientboundSetPlayerTeamPacket.shouldHaveParameters((int)this.method)) {
-            ((net.minecraft.network.protocol.game.ClientboundSetPlayerTeamPacket$Parameters)this.parameters.orElseThrow(() -> new java.lang.IllegalStateException("Parameters not present, but method is" + this.method))).write(buffer);
+            ((net.minecraft.network.protocol.game.ClientboundSetPlayerTeamPacket$Parameters)this.parameters.orElseThrow(() -> new java.lang.IllegalStateException("Parameters not present, but method is" + this.method))).write($$0);
         }
         if (net.minecraft.network.protocol.game.ClientboundSetPlayerTeamPacket.shouldHavePlayerList((int)this.method)) {
-            buffer.writeCollection(this.players, net.minecraft.network.FriendlyByteBuf::writeUtf);
+            $$0.writeCollection(this.players, net.minecraft.network.FriendlyByteBuf::writeUtf);
         }
     }
 
@@ -90,8 +90,8 @@
         return null;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSetPlayerTeamPacket(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSetPlayerTeamPacket(this);
     }
 
     public java.lang.String getName() {