--- net/minecraft/network/protocol/game/ClientboundUpdateEnabledFeaturesPacket.java
+++ net/minecraft/network/protocol/game/ClientboundUpdateEnabledFeaturesPacket.java
@@ -1,0 +1,44 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.protocol.game;
+
+public class net.minecraft.network.protocol.game.ClientboundUpdateEnabledFeaturesPacket
+extends java.lang.Record
+implements net.minecraft.network.protocol.Packet<net.minecraft.network.protocol.game.ClientGamePacketListener>
+ {
+    public java.util.Set<net.minecraft.resources.ResourceLocation> features;
+
+    public net.minecraft.network.protocol.game.ClientboundUpdateEnabledFeaturesPacket(net.minecraft.network.FriendlyByteBuf $$0) {
+        this((java.util.Set)$$0.readCollection(java.util.HashSet::new, net.minecraft.network.FriendlyByteBuf::readResourceLocation));
+    }
+
+    public net.minecraft.network.protocol.game.ClientboundUpdateEnabledFeaturesPacket(java.util.Set<net.minecraft.resources.ResourceLocation> $$0) {
+        this.features = $$0;
+    }
+
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeCollection((java.util.Collection)this.features, net.minecraft.network.FriendlyByteBuf::writeResourceLocation);
+    }
+
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleEnabledFeatures(this);
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ClientboundUpdateEnabledFeaturesPacket.class, "features", "features"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ClientboundUpdateEnabledFeaturesPacket.class, "features", "features"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ClientboundUpdateEnabledFeaturesPacket.class, "features", "features"}, this, $$0);
+    }
+
+    public java.util.Set<net.minecraft.resources.ResourceLocation> features() {
+        return this.features;
+    }
+}
+