--- net/minecraft/network/protocol/game/ClientboundBossEventPacket.java
+++ net/minecraft/network/protocol/game/ClientboundBossEventPacket.java
@@ -48,10 +48,10 @@
         return new net.minecraft.network.protocol.game.ClientboundBossEventPacket($$0.getId(), (net.minecraft.network.protocol.game.ClientboundBossEventPacket$Operation)new net.minecraft.network.protocol.game.ClientboundBossEventPacket$UpdatePropertiesOperation($$0.shouldDarkenScreen(), $$0.shouldPlayBossMusic(), $$0.shouldCreateWorldFog()));
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUUID(this.id);
-        buffer.writeEnum((java.lang.Enum)this.operation.getType());
-        this.operation.write(buffer);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUUID(this.id);
+        $$0.writeEnum((java.lang.Enum)this.operation.getType());
+        this.operation.write($$0);
     }
 
     public static int encodeProperties(boolean $$0, boolean $$1, boolean $$2) {
@@ -68,8 +68,8 @@
         return $$3;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleBossUpdate(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleBossUpdate(this);
     }
 
     public void dispatch(net.minecraft.network.protocol.game.ClientboundBossEventPacket$Handler $$0) {