--- net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
+++ net/minecraft/network/protocol/game/ClientboundPlayerAbilitiesPacket.java
@@ -17,13 +17,13 @@
     public float flyingSpeed;
     public float walkingSpeed;
 
-    public net.minecraft.network.protocol.game.ClientboundPlayerAbilitiesPacket(net.minecraft.world.entity.player.Abilities abilities) {
-        this.invulnerable = abilities.invulnerable;
-        this.isFlying = abilities.flying;
-        this.canFly = abilities.mayfly;
-        this.instabuild = abilities.instabuild;
-        this.flyingSpeed = abilities.getFlyingSpeed();
-        this.walkingSpeed = abilities.getWalkingSpeed();
+    public net.minecraft.network.protocol.game.ClientboundPlayerAbilitiesPacket(net.minecraft.world.entity.player.Abilities $$0) {
+        this.invulnerable = $$0.invulnerable;
+        this.isFlying = $$0.flying;
+        this.canFly = $$0.mayfly;
+        this.instabuild = $$0.instabuild;
+        this.flyingSpeed = $$0.getFlyingSpeed();
+        this.walkingSpeed = $$0.getWalkingSpeed();
     }
 
     public net.minecraft.network.protocol.game.ClientboundPlayerAbilitiesPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -36,27 +36,27 @@
         this.walkingSpeed = $$0.readFloat();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        byte by = 0;
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        byte $$1 = 0;
         if (this.invulnerable) {
-            by = (byte)(by | 1);
+            $$1 = (byte)($$1 | 1);
         }
         if (this.isFlying) {
-            by = (byte)(by | 2);
+            $$1 = (byte)($$1 | 2);
         }
         if (this.canFly) {
-            by = (byte)(by | 4);
+            $$1 = (byte)($$1 | 4);
         }
         if (this.instabuild) {
-            by = (byte)(by | 8);
+            $$1 = (byte)($$1 | 8);
         }
-        buffer.writeByte((int)by);
-        buffer.writeFloat(this.flyingSpeed);
-        buffer.writeFloat(this.walkingSpeed);
+        $$0.writeByte((int)$$1);
+        $$0.writeFloat(this.flyingSpeed);
+        $$0.writeFloat(this.walkingSpeed);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handlePlayerAbilities(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handlePlayerAbilities(this);
     }
 
     public boolean isInvulnerable() {