--- net/minecraft/network/protocol/game/ClientboundOpenSignEditorPacket.java
+++ net/minecraft/network/protocol/game/ClientboundOpenSignEditorPacket.java
@@ -8,20 +8,20 @@
  {
     public net.minecraft.core.BlockPos pos;
 
-    public net.minecraft.network.protocol.game.ClientboundOpenSignEditorPacket(net.minecraft.core.BlockPos pos) {
-        this.pos = pos;
+    public net.minecraft.network.protocol.game.ClientboundOpenSignEditorPacket(net.minecraft.core.BlockPos $$0) {
+        this.pos = $$0;
     }
 
     public net.minecraft.network.protocol.game.ClientboundOpenSignEditorPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.pos = $$0.readBlockPos();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeBlockPos(this.pos);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeBlockPos(this.pos);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleOpenSignEditor(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleOpenSignEditor(this);
     }
 
     public net.minecraft.core.BlockPos getPos() {