--- net/minecraft/network/protocol/game/ClientboundResourcePackPacket.java
+++ net/minecraft/network/protocol/game/ClientboundResourcePackPacket.java
@@ -30,15 +30,15 @@
         this.prompt = (net.minecraft.network.chat.Component)$$0.readNullable(net.minecraft.network.FriendlyByteBuf::readComponent);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUtf(this.url);
-        buffer.writeUtf(this.hash);
-        buffer.writeBoolean(this.required);
-        buffer.writeNullable((java.lang.Object)this.prompt, net.minecraft.network.FriendlyByteBuf::writeComponent);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUtf(this.url);
+        $$0.writeUtf(this.hash);
+        $$0.writeBoolean(this.required);
+        $$0.writeNullable((java.lang.Object)this.prompt, net.minecraft.network.FriendlyByteBuf::writeComponent);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleResourcePack(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleResourcePack(this);
     }
 
     public java.lang.String getUrl() {