--- net/minecraft/network/protocol/game/ClientboundBlockEntityDataPacket.java
+++ net/minecraft/network/protocol/game/ClientboundBlockEntityDataPacket.java
@@ -31,14 +31,14 @@
         this.tag = $$0.readNbt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeBlockPos(this.pos);
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.BLOCK_ENTITY_TYPE, (java.lang.Object)this.type);
-        buffer.writeNbt(this.tag);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeBlockPos(this.pos);
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.BLOCK_ENTITY_TYPE, (java.lang.Object)this.type);
+        $$0.writeNbt(this.tag);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleBlockEntityData(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleBlockEntityData(this);
     }
 
     public net.minecraft.core.BlockPos getPos() {