--- net/minecraft/network/protocol/game/ClientboundCustomPayloadPacket.java
+++ net/minecraft/network/protocol/game/ClientboundCustomPayloadPacket.java
@@ -28,10 +28,10 @@
     public net.minecraft.resources.ResourceLocation identifier;
     public net.minecraft.network.FriendlyByteBuf data;
 
-    public net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket(net.minecraft.resources.ResourceLocation identifier, net.minecraft.network.FriendlyByteBuf data) {
-        this.identifier = identifier;
-        this.data = data;
-        if (data.writerIndex() > 0x100000) {
+    public net.minecraft.network.protocol.game.ClientboundCustomPayloadPacket(net.minecraft.resources.ResourceLocation $$0, net.minecraft.network.FriendlyByteBuf $$1) {
+        this.identifier = $$0;
+        this.data = $$1;
+        if ($$1.writerIndex() > 0x100000) {
             throw new java.lang.IllegalArgumentException("Payload may not be larger than 1048576 bytes");
         }
     }
@@ -45,13 +45,13 @@
         this.data = new net.minecraft.network.FriendlyByteBuf($$0.readBytes($$1));
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeResourceLocation(this.identifier);
-        buffer.writeBytes(this.data.copy());
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeResourceLocation(this.identifier);
+        $$0.writeBytes(this.data.copy());
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleCustomPayload(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleCustomPayload(this);
     }
 
     public net.minecraft.resources.ResourceLocation getIdentifier() {