--- net/minecraft/network/protocol/game/ClientboundChangeDifficultyPacket.java
+++ net/minecraft/network/protocol/game/ClientboundChangeDifficultyPacket.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.Difficulty difficulty;
     public boolean locked;
 
-    public net.minecraft.network.protocol.game.ClientboundChangeDifficultyPacket(net.minecraft.world.Difficulty difficulty, boolean locked) {
-        this.difficulty = difficulty;
-        this.locked = locked;
+    public net.minecraft.network.protocol.game.ClientboundChangeDifficultyPacket(net.minecraft.world.Difficulty $$0, boolean $$1) {
+        this.difficulty = $$0;
+        this.locked = $$1;
     }
 
     public net.minecraft.network.protocol.game.ClientboundChangeDifficultyPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,13 +19,13 @@
         this.locked = $$0.readBoolean();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeByte(this.difficulty.getId());
-        buffer.writeBoolean(this.locked);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeByte(this.difficulty.getId());
+        $$0.writeBoolean(this.locked);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleChangeDifficulty(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleChangeDifficulty(this);
     }
 
     public boolean isLocked() {