--- net/minecraft/network/protocol/game/ClientboundServerDataPacket.java
+++ net/minecraft/network/protocol/game/ClientboundServerDataPacket.java
@@ -8,27 +8,23 @@
  {
     public java.util.Optional<net.minecraft.network.chat.Component> motd;
     public java.util.Optional<java.lang.String> iconBase64;
-    public boolean previewsChat;
     public boolean enforcesSecureChat;
 
-    public net.minecraft.network.protocol.game.ClientboundServerDataPacket(@javax.annotation.Nullable net.minecraft.network.chat.Component $$0, @javax.annotation.Nullable java.lang.String $$1, boolean $$2, boolean $$3) {
+    public net.minecraft.network.protocol.game.ClientboundServerDataPacket(@javax.annotation.Nullable net.minecraft.network.chat.Component $$0, @javax.annotation.Nullable java.lang.String $$1, boolean $$2) {
         this.motd = java.util.Optional.ofNullable((java.lang.Object)$$0);
         this.iconBase64 = java.util.Optional.ofNullable((java.lang.Object)$$1);
-        this.previewsChat = $$2;
-        this.enforcesSecureChat = $$3;
+        this.enforcesSecureChat = $$2;
     }
 
     public net.minecraft.network.protocol.game.ClientboundServerDataPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.motd = $$0.readOptional(net.minecraft.network.FriendlyByteBuf::readComponent);
         this.iconBase64 = $$0.readOptional(net.minecraft.network.FriendlyByteBuf::readUtf);
-        this.previewsChat = $$0.readBoolean();
         this.enforcesSecureChat = $$0.readBoolean();
     }
 
     public void write(net.minecraft.network.FriendlyByteBuf $$0) {
         $$0.writeOptional(this.motd, net.minecraft.network.FriendlyByteBuf::writeComponent);
         $$0.writeOptional(this.iconBase64, net.minecraft.network.FriendlyByteBuf::writeUtf);
-        $$0.writeBoolean(this.previewsChat);
         $$0.writeBoolean(this.enforcesSecureChat);
     }
 
@@ -44,10 +40,6 @@
         return this.iconBase64;
     }
 
-    public boolean previewsChat() {
-        return this.previewsChat;
-    }
-
     public boolean enforcesSecureChat() {
         return this.enforcesSecureChat;
     }