--- net/minecraft/network/protocol/game/ClientboundSetCameraPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSetCameraPacket.java
@@ -8,25 +8,25 @@
  {
     public int cameraId;
 
-    public net.minecraft.network.protocol.game.ClientboundSetCameraPacket(net.minecraft.world.entity.Entity cameraEntity) {
-        this.cameraId = cameraEntity.getId();
+    public net.minecraft.network.protocol.game.ClientboundSetCameraPacket(net.minecraft.world.entity.Entity $$0) {
+        this.cameraId = $$0.getId();
     }
 
     public net.minecraft.network.protocol.game.ClientboundSetCameraPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.cameraId = $$0.readVarInt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.cameraId);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.cameraId);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSetCamera(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSetCamera(this);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.Entity getEntity(net.minecraft.world.level.Level level) {
-        return level.getEntity(this.cameraId);
+    public net.minecraft.world.entity.Entity getEntity(net.minecraft.world.level.Level $$0) {
+        return $$0.getEntity(this.cameraId);
     }
 }
 