--- net/minecraft/network/protocol/game/ClientboundAddEntityPacket.java
+++ net/minecraft/network/protocol/game/ClientboundAddEntityPacket.java
@@ -30,12 +30,12 @@
         this($$0.getId(), $$0.getUUID(), $$0.getX(), $$0.getY(), $$0.getZ(), $$0.getXRot(), $$0.getYRot(), $$0.getType(), $$1, $$0.getDeltaMovement(), (double)$$0.yHeadRot);
     }
 
-    public net.minecraft.network.protocol.game.ClientboundAddEntityPacket(net.minecraft.world.entity.Entity entity) {
-        this(entity, 0);
+    public net.minecraft.network.protocol.game.ClientboundAddEntityPacket(net.minecraft.world.entity.Entity $$0) {
+        this($$0, 0);
     }
 
-    public net.minecraft.network.protocol.game.ClientboundAddEntityPacket(net.minecraft.world.entity.Entity entity, int data) {
-        this(entity.getId(), entity.getUUID(), entity.getX(), entity.getY(), entity.getZ(), entity.getXRot(), entity.getYRot(), entity.getType(), data, entity.getDeltaMovement(), 0.0);
+    public net.minecraft.network.protocol.game.ClientboundAddEntityPacket(net.minecraft.world.entity.Entity $$0, int $$1) {
+        this($$0.getId(), $$0.getUUID(), $$0.getX(), $$0.getY(), $$0.getZ(), $$0.getXRot(), $$0.getYRot(), $$0.getType(), $$1, $$0.getDeltaMovement(), 0.0);
     }
 
     public net.minecraft.network.protocol.game.ClientboundAddEntityPacket(net.minecraft.world.entity.Entity $$0, int $$1, net.minecraft.core.BlockPos $$2) {
@@ -74,24 +74,24 @@
         this.za = $$0.readShort();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.id);
-        buffer.writeUUID(this.uuid);
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.ENTITY_TYPE, (java.lang.Object)this.type);
-        buffer.writeDouble(this.x);
-        buffer.writeDouble(this.y);
-        buffer.writeDouble(this.z);
-        buffer.writeByte((int)this.xRot);
-        buffer.writeByte((int)this.yRot);
-        buffer.writeByte((int)this.yHeadRot);
-        buffer.writeVarInt(this.data);
-        buffer.writeShort(this.xa);
-        buffer.writeShort(this.ya);
-        buffer.writeShort(this.za);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.id);
+        $$0.writeUUID(this.uuid);
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.ENTITY_TYPE, (java.lang.Object)this.type);
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeByte((int)this.xRot);
+        $$0.writeByte((int)this.yRot);
+        $$0.writeByte((int)this.yHeadRot);
+        $$0.writeVarInt(this.data);
+        $$0.writeShort(this.xa);
+        $$0.writeShort(this.ya);
+        $$0.writeShort(this.za);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleAddEntity(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleAddEntity(this);
     }
 
     public int getId() {