--- net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSetEquipmentPacket.java
@@ -10,9 +10,9 @@
     public int entity;
     public java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.world.entity.EquipmentSlot, net.minecraft.world.item.ItemStack>> slots;
 
-    public net.minecraft.network.protocol.game.ClientboundSetEquipmentPacket(int entity, java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.world.entity.EquipmentSlot, net.minecraft.world.item.ItemStack>> slots) {
-        this.entity = entity;
-        this.slots = slots;
+    public net.minecraft.network.protocol.game.ClientboundSetEquipmentPacket(int $$0, java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.world.entity.EquipmentSlot, net.minecraft.world.item.ItemStack>> $$1) {
+        this.entity = $$0;
+        this.slots = $$1;
     }
 
     public net.minecraft.network.protocol.game.ClientboundSetEquipmentPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -28,21 +28,21 @@
         } while (($$2 & 0xFFFFFF80) != 0);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entity);
-        int n = this.slots.size();
-        for (int i = 0; i < n; ++i) {
-            com.mojang.datafixers.util.Pair pair = (com.mojang.datafixers.util.Pair)this.slots.get(i);
-            net.minecraft.world.entity.EquipmentSlot equipmentSlot = (net.minecraft.world.entity.EquipmentSlot)pair.getFirst();
-            boolean bl = i != n - 1;
-            int n2 = equipmentSlot.ordinal();
-            buffer.writeByte(bl ? n2 | 0xFFFFFF80 : n2);
-            buffer.writeItem((net.minecraft.world.item.ItemStack)pair.getSecond());
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.entity);
+        int $$1 = this.slots.size();
+        for (int $$2 = 0; $$2 < $$1; ++$$2) {
+            com.mojang.datafixers.util.Pair $$3 = (com.mojang.datafixers.util.Pair)this.slots.get($$2);
+            net.minecraft.world.entity.EquipmentSlot $$4 = (net.minecraft.world.entity.EquipmentSlot)$$3.getFirst();
+            boolean $$5 = $$2 != $$1 - 1;
+            int $$6 = $$4.ordinal();
+            $$0.writeByte($$5 ? $$6 | 0xFFFFFF80 : $$6);
+            $$0.writeItem((net.minecraft.world.item.ItemStack)$$3.getSecond());
         }
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSetEquipment(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSetEquipment(this);
     }
 
     public int getEntity() {