--- net/minecraft/network/protocol/game/ClientboundBlockUpdatePacket.java
+++ net/minecraft/network/protocol/game/ClientboundBlockUpdatePacket.java
@@ -9,13 +9,13 @@
     public net.minecraft.core.BlockPos pos;
     public net.minecraft.world.level.block.state.BlockState blockState;
 
-    public net.minecraft.network.protocol.game.ClientboundBlockUpdatePacket(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState blockState) {
-        this.pos = pos;
-        this.blockState = blockState;
+    public net.minecraft.network.protocol.game.ClientboundBlockUpdatePacket(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
+        this.pos = $$0;
+        this.blockState = $$1;
     }
 
-    public net.minecraft.network.protocol.game.ClientboundBlockUpdatePacket(net.minecraft.world.level.BlockGetter blockGetter, net.minecraft.core.BlockPos pos) {
-        this(pos, blockGetter.getBlockState(pos));
+    public net.minecraft.network.protocol.game.ClientboundBlockUpdatePacket(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1) {
+        this($$1, $$0.getBlockState($$1));
     }
 
     public net.minecraft.network.protocol.game.ClientboundBlockUpdatePacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -23,13 +23,13 @@
         this.blockState = (net.minecraft.world.level.block.state.BlockState)$$0.readById((net.minecraft.core.IdMap)net.minecraft.world.level.block.Block.BLOCK_STATE_REGISTRY);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeBlockPos(this.pos);
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.world.level.block.Block.BLOCK_STATE_REGISTRY, (java.lang.Object)this.blockState);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeBlockPos(this.pos);
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.world.level.block.Block.BLOCK_STATE_REGISTRY, (java.lang.Object)this.blockState);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleBlockUpdate(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleBlockUpdate(this);
     }
 
     public net.minecraft.world.level.block.state.BlockState getBlockState() {