--- net/minecraft/network/protocol/game/ClientboundMerchantOffersPacket.java
+++ net/minecraft/network/protocol/game/ClientboundMerchantOffersPacket.java
@@ -13,13 +13,13 @@
     public boolean showProgress;
     public boolean canRestock;
 
-    public net.minecraft.network.protocol.game.ClientboundMerchantOffersPacket(int containerId, net.minecraft.world.item.trading.MerchantOffers offers, int villagerLevel, int villagerXp, boolean showProgress, boolean canRestock) {
-        this.containerId = containerId;
-        this.offers = offers;
-        this.villagerLevel = villagerLevel;
-        this.villagerXp = villagerXp;
-        this.showProgress = showProgress;
-        this.canRestock = canRestock;
+    public net.minecraft.network.protocol.game.ClientboundMerchantOffersPacket(int $$0, net.minecraft.world.item.trading.MerchantOffers $$1, int $$2, int $$3, boolean $$4, boolean $$5) {
+        this.containerId = $$0;
+        this.offers = $$1;
+        this.villagerLevel = $$2;
+        this.villagerXp = $$3;
+        this.showProgress = $$4;
+        this.canRestock = $$5;
     }
 
     public net.minecraft.network.protocol.game.ClientboundMerchantOffersPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -31,17 +31,17 @@
         this.canRestock = $$0.readBoolean();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.containerId);
-        this.offers.writeToStream(buffer);
-        buffer.writeVarInt(this.villagerLevel);
-        buffer.writeVarInt(this.villagerXp);
-        buffer.writeBoolean(this.showProgress);
-        buffer.writeBoolean(this.canRestock);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.containerId);
+        this.offers.writeToStream($$0);
+        $$0.writeVarInt(this.villagerLevel);
+        $$0.writeVarInt(this.villagerXp);
+        $$0.writeBoolean(this.showProgress);
+        $$0.writeBoolean(this.canRestock);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleMerchantOffers(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleMerchantOffers(this);
     }
 
     public int getContainerId() {