--- net/minecraft/network/protocol/game/ClientboundAwardStatsPacket.java
+++ net/minecraft/network/protocol/game/ClientboundAwardStatsPacket.java
@@ -8,8 +8,8 @@
  {
     public it.unimi.dsi.fastutil.objects.Object2IntMap<net.minecraft.stats.Stat<?>> stats;
 
-    public net.minecraft.network.protocol.game.ClientboundAwardStatsPacket(it.unimi.dsi.fastutil.objects.Object2IntMap<net.minecraft.stats.Stat<?>> stats) {
-        this.stats = stats;
+    public net.minecraft.network.protocol.game.ClientboundAwardStatsPacket(it.unimi.dsi.fastutil.objects.Object2IntMap<net.minecraft.stats.Stat<?>> $$0) {
+        this.stats = $$0;
     }
 
     public net.minecraft.network.protocol.game.ClientboundAwardStatsPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -23,12 +23,12 @@
         return $$1.get($$0.readById((net.minecraft.core.IdMap)$$1.getRegistry()));
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleAwardStats(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleAwardStats(this);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeMap((java.util.Map)this.stats, net.minecraft.network.protocol.game.ClientboundAwardStatsPacket::writeStatCap, net.minecraft.network.FriendlyByteBuf::writeVarInt);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeMap((java.util.Map)this.stats, net.minecraft.network.protocol.game.ClientboundAwardStatsPacket::writeStatCap, net.minecraft.network.FriendlyByteBuf::writeVarInt);
     }
 
     public static <T> void writeStatCap(net.minecraft.network.FriendlyByteBuf $$0, net.minecraft.stats.Stat<T> $$1) {