--- net/minecraft/network/protocol/game/ClientboundTeleportEntityPacket.java
+++ net/minecraft/network/protocol/game/ClientboundTeleportEntityPacket.java
@@ -14,15 +14,15 @@
     public byte xRot;
     public boolean onGround;
 
-    public net.minecraft.network.protocol.game.ClientboundTeleportEntityPacket(net.minecraft.world.entity.Entity entity) {
-        this.id = entity.getId();
-        net.minecraft.world.phys.Vec3 vec3 = entity.trackingPosition();
-        this.x = vec3.x;
-        this.y = vec3.y;
-        this.z = vec3.z;
-        this.yRot = (byte)(entity.getYRot() * 256.0f / 360.0f);
-        this.xRot = (byte)(entity.getXRot() * 256.0f / 360.0f);
-        this.onGround = entity.isOnGround();
+    public net.minecraft.network.protocol.game.ClientboundTeleportEntityPacket(net.minecraft.world.entity.Entity $$0) {
+        this.id = $$0.getId();
+        net.minecraft.world.phys.Vec3 $$1 = $$0.trackingPosition();
+        this.x = $$1.x;
+        this.y = $$1.y;
+        this.z = $$1.z;
+        this.yRot = (byte)($$0.getYRot() * 256.0f / 360.0f);
+        this.xRot = (byte)($$0.getXRot() * 256.0f / 360.0f);
+        this.onGround = $$0.isOnGround();
     }
 
     public net.minecraft.network.protocol.game.ClientboundTeleportEntityPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -35,18 +35,18 @@
         this.onGround = $$0.readBoolean();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.id);
-        buffer.writeDouble(this.x);
-        buffer.writeDouble(this.y);
-        buffer.writeDouble(this.z);
-        buffer.writeByte((int)this.yRot);
-        buffer.writeByte((int)this.xRot);
-        buffer.writeBoolean(this.onGround);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.id);
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeByte((int)this.yRot);
+        $$0.writeByte((int)this.xRot);
+        $$0.writeBoolean(this.onGround);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleTeleportEntity(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleTeleportEntity(this);
     }
 
     public int getId() {