--- net/minecraft/network/protocol/game/ClientboundBlockEventPacket.java
+++ net/minecraft/network/protocol/game/ClientboundBlockEventPacket.java
@@ -11,11 +11,11 @@
     public int b1;
     public net.minecraft.world.level.block.Block block;
 
-    public net.minecraft.network.protocol.game.ClientboundBlockEventPacket(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.Block block, int b0, int b1) {
-        this.pos = pos;
-        this.block = block;
-        this.b0 = b0;
-        this.b1 = b1;
+    public net.minecraft.network.protocol.game.ClientboundBlockEventPacket(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.Block $$1, int $$2, int $$3) {
+        this.pos = $$0;
+        this.block = $$1;
+        this.b0 = $$2;
+        this.b1 = $$3;
     }
 
     public net.minecraft.network.protocol.game.ClientboundBlockEventPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -25,15 +25,15 @@
         this.block = (net.minecraft.world.level.block.Block)$$0.readById((net.minecraft.core.IdMap)net.minecraft.core.Registry.BLOCK);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeBlockPos(this.pos);
-        buffer.writeByte(this.b0);
-        buffer.writeByte(this.b1);
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.BLOCK, (java.lang.Object)this.block);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeBlockPos(this.pos);
+        $$0.writeByte(this.b0);
+        $$0.writeByte(this.b1);
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.BLOCK, (java.lang.Object)this.block);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleBlockEvent(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleBlockEvent(this);
     }
 
     public net.minecraft.core.BlockPos getPos() {