--- net/minecraft/network/protocol/game/ClientboundGameEventPacket$Type.java
+++ net/minecraft/network/protocol/game/ClientboundGameEventPacket$Type.java
@@ -8,9 +8,9 @@
     public static it.unimi.dsi.fastutil.ints.Int2ObjectMap<net.minecraft.network.protocol.game.ClientboundGameEventPacket$Type> TYPES;
     public int id;
 
-    public net.minecraft.network.protocol.game.ClientboundGameEventPacket$Type(int id) {
-        this.id = id;
-        TYPES.put(id, (java.lang.Object)this);
+    public net.minecraft.network.protocol.game.ClientboundGameEventPacket$Type(int $$0) {
+        this.id = $$0;
+        TYPES.put($$0, (java.lang.Object)this);
     }
 
     static {