--- net/minecraft/network/protocol/game/ClientboundExplodePacket.java
+++ net/minecraft/network/protocol/game/ClientboundExplodePacket.java
@@ -15,16 +15,16 @@
     public float knockbackY;
     public float knockbackZ;
 
-    public net.minecraft.network.protocol.game.ClientboundExplodePacket(double x, double y, double z, float power, java.util.List<net.minecraft.core.BlockPos> toBlow, @javax.annotation.Nullable net.minecraft.world.phys.Vec3 knockback) {
-        this.x = x;
-        this.y = y;
-        this.z = z;
-        this.power = power;
-        this.toBlow = com.google.common.collect.Lists.newArrayList(toBlow);
-        if (knockback != null) {
-            this.knockbackX = (float)knockback.x;
-            this.knockbackY = (float)knockback.y;
-            this.knockbackZ = (float)knockback.z;
+    public net.minecraft.network.protocol.game.ClientboundExplodePacket(double $$0, double $$1, double $$2, float $$3, java.util.List<net.minecraft.core.BlockPos> $$4, @javax.annotation.Nullable net.minecraft.world.phys.Vec3 $$5) {
+        this.x = $$0;
+        this.y = $$1;
+        this.z = $$2;
+        this.power = $$3;
+        this.toBlow = com.google.common.collect.Lists.newArrayList($$4);
+        if ($$5 != null) {
+            this.knockbackX = (float)$$5.x;
+            this.knockbackY = (float)$$5.y;
+            this.knockbackZ = (float)$$5.z;
         } else {
             this.knockbackX = 0.0f;
             this.knockbackY = 0.0f;
@@ -33,9 +33,9 @@
     }
 
     public net.minecraft.network.protocol.game.ClientboundExplodePacket(net.minecraft.network.FriendlyByteBuf $$0) {
-        this.x = $$0.readFloat();
-        this.y = $$0.readFloat();
-        this.z = $$0.readFloat();
+        this.x = $$0.readDouble();
+        this.y = $$0.readDouble();
+        this.z = $$0.readDouble();
         this.power = $$0.readFloat();
         int $$1 = net.minecraft.util.Mth.floor((double)this.x);
         int $$2 = net.minecraft.util.Mth.floor((double)this.y);
@@ -51,29 +51,29 @@
         this.knockbackZ = $$0.readFloat();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeFloat((float)this.x);
-        buffer.writeFloat((float)this.y);
-        buffer.writeFloat((float)this.z);
-        buffer.writeFloat(this.power);
-        int n = net.minecraft.util.Mth.floor((double)this.x);
-        int n2 = net.minecraft.util.Mth.floor((double)this.y);
-        int n3 = net.minecraft.util.Mth.floor((double)this.z);
-        buffer.writeCollection((java.util.Collection)this.toBlow, ($$3, $$4) -> {
-            int $$5 = $$4.getX() - n;
-            int $$6 = $$4.getY() - n2;
-            int $$7 = $$4.getZ() - n3;
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeFloat(this.power);
+        int $$1 = net.minecraft.util.Mth.floor((double)this.x);
+        int $$2 = net.minecraft.util.Mth.floor((double)this.y);
+        int $$32 = net.minecraft.util.Mth.floor((double)this.z);
+        $$0.writeCollection((java.util.Collection)this.toBlow, ($$3, $$4) -> {
+            int $$5 = $$4.getX() - $$1;
+            int $$6 = $$4.getY() - $$2;
+            int $$7 = $$4.getZ() - $$32;
             $$3.writeByte($$5);
             $$3.writeByte($$6);
             $$3.writeByte($$7);
         });
-        buffer.writeFloat(this.knockbackX);
-        buffer.writeFloat(this.knockbackY);
-        buffer.writeFloat(this.knockbackZ);
+        $$0.writeFloat(this.knockbackX);
+        $$0.writeFloat(this.knockbackY);
+        $$0.writeFloat(this.knockbackZ);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleExplosion(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleExplosion(this);
     }
 
     public float getKnockbackX() {