--- net/minecraft/network/protocol/game/ClientboundUpdateMobEffectPacket.java
+++ net/minecraft/network/protocol/game/ClientboundUpdateMobEffectPacket.java
@@ -18,23 +18,23 @@
     @javax.annotation.Nullable
     public net.minecraft.world.effect.MobEffectInstance$FactorData factorData;
 
-    public net.minecraft.network.protocol.game.ClientboundUpdateMobEffectPacket(int entityId, net.minecraft.world.effect.MobEffectInstance effectInstance) {
-        this.entityId = entityId;
-        this.effect = effectInstance.getEffect();
-        this.effectAmplifier = (byte)(effectInstance.getAmplifier() & 0xFF);
-        this.effectDurationTicks = effectInstance.getDuration();
-        byte by = 0;
-        if (effectInstance.isAmbient()) {
-            by = (byte)(by | 1);
+    public net.minecraft.network.protocol.game.ClientboundUpdateMobEffectPacket(int $$0, net.minecraft.world.effect.MobEffectInstance $$1) {
+        this.entityId = $$0;
+        this.effect = $$1.getEffect();
+        this.effectAmplifier = (byte)($$1.getAmplifier() & 0xFF);
+        this.effectDurationTicks = $$1.getDuration();
+        byte $$2 = 0;
+        if ($$1.isAmbient()) {
+            $$2 = (byte)($$2 | 1);
         }
-        if (effectInstance.isVisible()) {
-            by = (byte)(by | 2);
+        if ($$1.isVisible()) {
+            $$2 = (byte)($$2 | 2);
         }
-        if (effectInstance.showIcon()) {
-            by = (byte)(by | 4);
+        if ($$1.showIcon()) {
+            $$2 = (byte)($$2 | 4);
         }
-        this.flags = by;
-        this.factorData = (net.minecraft.world.effect.MobEffectInstance$FactorData)effectInstance.getFactorData().orElse(null);
+        this.flags = $$2;
+        this.factorData = (net.minecraft.world.effect.MobEffectInstance$FactorData)$$1.getFactorData().orElse(null);
     }
 
     public net.minecraft.network.protocol.game.ClientboundUpdateMobEffectPacket(net.minecraft.network.FriendlyByteBuf $$02) {
@@ -46,21 +46,21 @@
         this.factorData = (net.minecraft.world.effect.MobEffectInstance$FactorData)$$02.readNullable($$0 -> (net.minecraft.world.effect.MobEffectInstance$FactorData)$$0.readWithCodec(net.minecraft.world.effect.MobEffectInstance$FactorData.CODEC));
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entityId);
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MOB_EFFECT, (java.lang.Object)this.effect);
-        buffer.writeByte((int)this.effectAmplifier);
-        buffer.writeVarInt(this.effectDurationTicks);
-        buffer.writeByte((int)this.flags);
-        buffer.writeNullable((java.lang.Object)this.factorData, ($$0, $$1) -> $$0.writeWithCodec(net.minecraft.world.effect.MobEffectInstance$FactorData.CODEC, $$1));
+    public void write(net.minecraft.network.FriendlyByteBuf $$02) {
+        $$02.writeVarInt(this.entityId);
+        $$02.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MOB_EFFECT, (java.lang.Object)this.effect);
+        $$02.writeByte((int)this.effectAmplifier);
+        $$02.writeVarInt(this.effectDurationTicks);
+        $$02.writeByte((int)this.flags);
+        $$02.writeNullable((java.lang.Object)this.factorData, ($$0, $$1) -> $$0.writeWithCodec(net.minecraft.world.effect.MobEffectInstance$FactorData.CODEC, $$1));
     }
 
     public boolean isSuperLongDuration() {
         return this.effectDurationTicks >= java.lang.Short.MAX_VALUE;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleUpdateMobEffect(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleUpdateMobEffect(this);
     }
 
     public int getEntityId() {