--- net/minecraft/network/protocol/game/ClientboundCommandSuggestionsPacket.java
+++ net/minecraft/network/protocol/game/ClientboundCommandSuggestionsPacket.java
@@ -9,9 +9,9 @@
     public int id;
     public com.mojang.brigadier.suggestion.Suggestions suggestions;
 
-    public net.minecraft.network.protocol.game.ClientboundCommandSuggestionsPacket(int id, com.mojang.brigadier.suggestion.Suggestions suggestions) {
-        this.id = id;
-        this.suggestions = suggestions;
+    public net.minecraft.network.protocol.game.ClientboundCommandSuggestionsPacket(int $$0, com.mojang.brigadier.suggestion.Suggestions $$1) {
+        this.id = $$0;
+        this.suggestions = $$1;
     }
 
     public net.minecraft.network.protocol.game.ClientboundCommandSuggestionsPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -27,18 +27,18 @@
         this.suggestions = new com.mojang.brigadier.suggestion.Suggestions($$3, $$4);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.id);
-        buffer.writeVarInt(this.suggestions.getRange().getStart());
-        buffer.writeVarInt(this.suggestions.getRange().getLength());
-        buffer.writeCollection((java.util.Collection)this.suggestions.getList(), ($$02, $$12) -> {
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.id);
+        $$0.writeVarInt(this.suggestions.getRange().getStart());
+        $$0.writeVarInt(this.suggestions.getRange().getLength());
+        $$0.writeCollection((java.util.Collection)this.suggestions.getList(), ($$02, $$12) -> {
             $$02.writeUtf($$12.getText());
             $$02.writeNullable((java.lang.Object)$$12.getTooltip(), ($$0, $$1) -> $$0.writeComponent(net.minecraft.network.chat.ComponentUtils.fromMessage((com.mojang.brigadier.Message)$$1)));
         });
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleCommandSuggestions(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleCommandSuggestions(this);
     }
 
     public int getId() {