--- net/minecraft/network/protocol/game/ClientboundOpenScreenPacket.java
+++ net/minecraft/network/protocol/game/ClientboundOpenScreenPacket.java
@@ -10,10 +10,10 @@
     public net.minecraft.world.inventory.MenuType<?> type;
     public net.minecraft.network.chat.Component title;
 
-    public net.minecraft.network.protocol.game.ClientboundOpenScreenPacket(int containerId, net.minecraft.world.inventory.MenuType<?> menuType, net.minecraft.network.chat.Component title) {
-        this.containerId = containerId;
-        this.type = menuType;
-        this.title = title;
+    public net.minecraft.network.protocol.game.ClientboundOpenScreenPacket(int $$0, net.minecraft.world.inventory.MenuType<?> $$1, net.minecraft.network.chat.Component $$2) {
+        this.containerId = $$0;
+        this.type = $$1;
+        this.title = $$2;
     }
 
     public net.minecraft.network.protocol.game.ClientboundOpenScreenPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -22,14 +22,14 @@
         this.title = $$0.readComponent();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.containerId);
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MENU, (java.lang.Object)this.type);
-        buffer.writeComponent(this.title);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.containerId);
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MENU, (java.lang.Object)this.type);
+        $$0.writeComponent(this.title);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleOpenScreen(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleOpenScreen(this);
     }
 
     public int getContainerId() {