--- net/minecraft/network/protocol/game/ClientboundPlayerPositionPacket.java
+++ net/minecraft/network/protocol/game/ClientboundPlayerPositionPacket.java
@@ -37,19 +37,19 @@
         this.dismountVehicle = $$0.readBoolean();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeDouble(this.x);
-        buffer.writeDouble(this.y);
-        buffer.writeDouble(this.z);
-        buffer.writeFloat(this.yRot);
-        buffer.writeFloat(this.xRot);
-        buffer.writeByte(net.minecraft.network.protocol.game.ClientboundPlayerPositionPacket$RelativeArgument.pack((java.util.Set)this.relativeArguments));
-        buffer.writeVarInt(this.id);
-        buffer.writeBoolean(this.dismountVehicle);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeFloat(this.yRot);
+        $$0.writeFloat(this.xRot);
+        $$0.writeByte(net.minecraft.network.protocol.game.ClientboundPlayerPositionPacket$RelativeArgument.pack((java.util.Set)this.relativeArguments));
+        $$0.writeVarInt(this.id);
+        $$0.writeBoolean(this.dismountVehicle);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleMovePlayer(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleMovePlayer(this);
     }
 
     public double getX() {