--- net/minecraft/network/protocol/game/ClientboundCooldownPacket.java
+++ net/minecraft/network/protocol/game/ClientboundCooldownPacket.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.item.Item item;
     public int duration;
 
-    public net.minecraft.network.protocol.game.ClientboundCooldownPacket(net.minecraft.world.item.Item item, int duration) {
-        this.item = item;
-        this.duration = duration;
+    public net.minecraft.network.protocol.game.ClientboundCooldownPacket(net.minecraft.world.item.Item $$0, int $$1) {
+        this.item = $$0;
+        this.duration = $$1;
     }
 
     public net.minecraft.network.protocol.game.ClientboundCooldownPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,13 +19,13 @@
         this.duration = $$0.readVarInt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.ITEM, (java.lang.Object)this.item);
-        buffer.writeVarInt(this.duration);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.ITEM, (java.lang.Object)this.item);
+        $$0.writeVarInt(this.duration);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleItemCooldown(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleItemCooldown(this);
     }
 
     public net.minecraft.world.item.Item getItem() {