--- net/minecraft/network/protocol/game/ClientboundEntityEventPacket.java
+++ net/minecraft/network/protocol/game/ClientboundEntityEventPacket.java
@@ -9,9 +9,9 @@
     public int entityId;
     public byte eventId;
 
-    public net.minecraft.network.protocol.game.ClientboundEntityEventPacket(net.minecraft.world.entity.Entity entity, byte eventId) {
-        this.entityId = entity.getId();
-        this.eventId = eventId;
+    public net.minecraft.network.protocol.game.ClientboundEntityEventPacket(net.minecraft.world.entity.Entity $$0, byte $$1) {
+        this.entityId = $$0.getId();
+        this.eventId = $$1;
     }
 
     public net.minecraft.network.protocol.game.ClientboundEntityEventPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,18 +19,18 @@
         this.eventId = $$0.readByte();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeInt(this.entityId);
-        buffer.writeByte((int)this.eventId);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeInt(this.entityId);
+        $$0.writeByte((int)this.eventId);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleEntityEvent(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleEntityEvent(this);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.Entity getEntity(net.minecraft.world.level.Level level) {
-        return level.getEntity(this.entityId);
+    public net.minecraft.world.entity.Entity getEntity(net.minecraft.world.level.Level $$0) {
+        return $$0.getEntity(this.entityId);
     }
 
     public byte getEventId() {