--- net/minecraft/network/protocol/game/ClientboundUpdateAttributesPacket.java
+++ net/minecraft/network/protocol/game/ClientboundUpdateAttributesPacket.java
@@ -9,11 +9,11 @@
     public int entityId;
     public java.util.List<net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket$AttributeSnapshot> attributes;
 
-    public net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket(int entityId, java.util.Collection<net.minecraft.world.entity.ai.attributes.AttributeInstance> attributes) {
-        this.entityId = entityId;
+    public net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket(int $$0, java.util.Collection<net.minecraft.world.entity.ai.attributes.AttributeInstance> $$1) {
+        this.entityId = $$0;
         this.attributes = com.google.common.collect.Lists.newArrayList();
-        for (net.minecraft.world.entity.ai.attributes.AttributeInstance attributeInstance : attributes) {
-            this.attributes.add((java.lang.Object)new net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket$AttributeSnapshot(attributeInstance.getAttribute(), attributeInstance.getBaseValue(), (java.util.Collection)attributeInstance.getModifiers()));
+        for (net.minecraft.world.entity.ai.attributes.AttributeInstance $$2 : $$1) {
+            this.attributes.add((java.lang.Object)new net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket$AttributeSnapshot($$2.getAttribute(), $$2.getBaseValue(), (java.util.Collection)$$2.getModifiers()));
         }
     }
 
@@ -28,9 +28,9 @@
         });
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entityId);
-        buffer.writeCollection((java.util.Collection)this.attributes, ($$02, $$12) -> {
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.entityId);
+        $$0.writeCollection((java.util.Collection)this.attributes, ($$02, $$12) -> {
             $$02.writeResourceLocation(net.minecraft.core.Registry.ATTRIBUTE.getKey((java.lang.Object)$$12.getAttribute()));
             $$02.writeDouble($$12.getBase());
             $$02.writeCollection($$12.getModifiers(), ($$0, $$1) -> {
@@ -41,8 +41,8 @@
         });
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleUpdateAttributes(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleUpdateAttributes(this);
     }
 
     public int getEntityId() {