--- net/minecraft/network/protocol/game/ClientboundSetScorePacket.java
+++ net/minecraft/network/protocol/game/ClientboundSetScorePacket.java
@@ -12,14 +12,14 @@
     public int score;
     public net.minecraft.server.ServerScoreboard$Method method;
 
-    public net.minecraft.network.protocol.game.ClientboundSetScorePacket(net.minecraft.server.ServerScoreboard$Method method, @javax.annotation.Nullable java.lang.String objectiveName, java.lang.String owner, int score) {
-        if (method != net.minecraft.server.ServerScoreboard$Method.REMOVE && objectiveName == null) {
+    public net.minecraft.network.protocol.game.ClientboundSetScorePacket(net.minecraft.server.ServerScoreboard$Method $$0, @javax.annotation.Nullable java.lang.String $$1, java.lang.String $$2, int $$3) {
+        if ($$0 != net.minecraft.server.ServerScoreboard$Method.REMOVE && $$1 == null) {
             throw new java.lang.IllegalArgumentException("Need an objective name");
         }
-        this.owner = owner;
-        this.objectiveName = objectiveName;
-        this.score = score;
-        this.method = method;
+        this.owner = $$2;
+        this.objectiveName = $$1;
+        this.score = $$3;
+        this.method = $$0;
     }
 
     public net.minecraft.network.protocol.game.ClientboundSetScorePacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -30,17 +30,17 @@
         this.score = this.method != net.minecraft.server.ServerScoreboard$Method.REMOVE ? $$0.readVarInt() : 0;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUtf(this.owner);
-        buffer.writeEnum((java.lang.Enum)this.method);
-        buffer.writeUtf(this.objectiveName == null ? "" : this.objectiveName);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUtf(this.owner);
+        $$0.writeEnum((java.lang.Enum)this.method);
+        $$0.writeUtf(this.objectiveName == null ? "" : this.objectiveName);
         if (this.method != net.minecraft.server.ServerScoreboard$Method.REMOVE) {
-            buffer.writeVarInt(this.score);
+            $$0.writeVarInt(this.score);
         }
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSetScore(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSetScore(this);
     }
 
     public java.lang.String getOwner() {