--- net/minecraft/network/protocol/game/ClientboundAnimatePacket.java
+++ net/minecraft/network/protocol/game/ClientboundAnimatePacket.java
@@ -15,9 +15,9 @@
     public int id;
     public int action;
 
-    public net.minecraft.network.protocol.game.ClientboundAnimatePacket(net.minecraft.world.entity.Entity entity, int action) {
-        this.id = entity.getId();
-        this.action = action;
+    public net.minecraft.network.protocol.game.ClientboundAnimatePacket(net.minecraft.world.entity.Entity $$0, int $$1) {
+        this.id = $$0.getId();
+        this.action = $$1;
     }
 
     public net.minecraft.network.protocol.game.ClientboundAnimatePacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -25,13 +25,13 @@
         this.action = $$0.readUnsignedByte();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.id);
-        buffer.writeByte(this.action);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.id);
+        $$0.writeByte(this.action);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleAnimate(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleAnimate(this);
     }
 
     public int getId() {