--- net/minecraft/network/protocol/game/ClientboundMoveEntityPacket$Pos.java
+++ net/minecraft/network/protocol/game/ClientboundMoveEntityPacket$Pos.java
@@ -6,8 +6,8 @@
 public class net.minecraft.network.protocol.game.ClientboundMoveEntityPacket$Pos
 extends net.minecraft.network.protocol.game.ClientboundMoveEntityPacket
  {
-    public net.minecraft.network.protocol.game.ClientboundMoveEntityPacket$Pos(int entityId, short xa, short ya, short za, boolean onGround) {
-        super(entityId, xa, ya, za, (byte)0, (byte)0, onGround, false, true);
+    public net.minecraft.network.protocol.game.ClientboundMoveEntityPacket$Pos(int $$0, short $$1, short $$2, short $$3, boolean $$4) {
+        super($$0, $$1, $$2, $$3, (byte)0, (byte)0, $$4, false, true);
     }
 
     public static net.minecraft.network.protocol.game.ClientboundMoveEntityPacket$Pos read(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,12 +19,12 @@
         return new net.minecraft.network.protocol.game.ClientboundMoveEntityPacket$Pos($$1, $$2, $$3, $$4, $$5);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entityId);
-        buffer.writeShort((int)this.xa);
-        buffer.writeShort((int)this.ya);
-        buffer.writeShort((int)this.za);
-        buffer.writeBoolean(this.onGround);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.entityId);
+        $$0.writeShort((int)this.xa);
+        $$0.writeShort((int)this.ya);
+        $$0.writeShort((int)this.za);
+        $$0.writeBoolean(this.onGround);
     }
 }
 