--- net/minecraft/network/protocol/game/ClientboundRespawnPacket.java
+++ net/minecraft/network/protocol/game/ClientboundRespawnPacket.java
@@ -41,20 +41,20 @@
         this.lastDeathLocation = $$0.readOptional(net.minecraft.network.FriendlyByteBuf::readGlobalPos);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeResourceKey(this.dimensionType);
-        buffer.writeResourceKey(this.dimension);
-        buffer.writeLong(this.seed);
-        buffer.writeByte(this.playerGameType.getId());
-        buffer.writeByte(net.minecraft.world.level.GameType.getNullableId((net.minecraft.world.level.GameType)this.previousPlayerGameType));
-        buffer.writeBoolean(this.isDebug);
-        buffer.writeBoolean(this.isFlat);
-        buffer.writeBoolean(this.keepAllPlayerData);
-        buffer.writeOptional(this.lastDeathLocation, net.minecraft.network.FriendlyByteBuf::writeGlobalPos);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeResourceKey(this.dimensionType);
+        $$0.writeResourceKey(this.dimension);
+        $$0.writeLong(this.seed);
+        $$0.writeByte(this.playerGameType.getId());
+        $$0.writeByte(net.minecraft.world.level.GameType.getNullableId((net.minecraft.world.level.GameType)this.previousPlayerGameType));
+        $$0.writeBoolean(this.isDebug);
+        $$0.writeBoolean(this.isFlat);
+        $$0.writeBoolean(this.keepAllPlayerData);
+        $$0.writeOptional(this.lastDeathLocation, net.minecraft.network.FriendlyByteBuf::writeGlobalPos);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleRespawn(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleRespawn(this);
     }
 
     public net.minecraft.resources.ResourceKey<net.minecraft.world.level.dimension.DimensionType> getDimensionType() {