--- net/minecraft/network/protocol/game/ClientboundAddPlayerPacket.java
+++ net/minecraft/network/protocol/game/ClientboundAddPlayerPacket.java
@@ -14,14 +14,14 @@
     public byte yRot;
     public byte xRot;
 
-    public net.minecraft.network.protocol.game.ClientboundAddPlayerPacket(net.minecraft.world.entity.player.Player player) {
-        this.entityId = player.getId();
-        this.playerId = player.getGameProfile().getId();
-        this.x = player.getX();
-        this.y = player.getY();
-        this.z = player.getZ();
-        this.yRot = (byte)(player.getYRot() * 256.0f / 360.0f);
-        this.xRot = (byte)(player.getXRot() * 256.0f / 360.0f);
+    public net.minecraft.network.protocol.game.ClientboundAddPlayerPacket(net.minecraft.world.entity.player.Player $$0) {
+        this.entityId = $$0.getId();
+        this.playerId = $$0.getGameProfile().getId();
+        this.x = $$0.getX();
+        this.y = $$0.getY();
+        this.z = $$0.getZ();
+        this.yRot = (byte)($$0.getYRot() * 256.0f / 360.0f);
+        this.xRot = (byte)($$0.getXRot() * 256.0f / 360.0f);
     }
 
     public net.minecraft.network.protocol.game.ClientboundAddPlayerPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -34,18 +34,18 @@
         this.xRot = $$0.readByte();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entityId);
-        buffer.writeUUID(this.playerId);
-        buffer.writeDouble(this.x);
-        buffer.writeDouble(this.y);
-        buffer.writeDouble(this.z);
-        buffer.writeByte((int)this.yRot);
-        buffer.writeByte((int)this.xRot);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.entityId);
+        $$0.writeUUID(this.playerId);
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeByte((int)this.yRot);
+        $$0.writeByte((int)this.xRot);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleAddPlayer(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleAddPlayer(this);
     }
 
     public int getEntityId() {