--- net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket$Entry.java
+++ net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket$Entry.java
@@ -1,0 +1,73 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.protocol.game;
+
+public class net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry
+extends java.lang.Record
+ {
+    public java.util.UUID profileId;
+    public com.mojang.authlib.GameProfile profile;
+    public boolean listed;
+    public int latency;
+    public net.minecraft.world.level.GameType gameMode;
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.Component displayName;
+    public net.minecraft.network.chat.RemoteChatSession$Data chatSession;
+
+    public net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry(net.minecraft.server.level.ServerPlayer $$0) {
+        this($$0.getUUID(), $$0.getGameProfile(), true, $$0.latency, $$0.gameMode.getGameModeForPlayer(), $$0.getTabListDisplayName(), $$0.getChatSession().asData());
+    }
+
+    public net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry(java.util.UUID $$0, com.mojang.authlib.GameProfile $$1, boolean $$2, int $$3, net.minecraft.world.level.GameType $$4, @javax.annotation.Nullable net.minecraft.network.chat.Component $$5, net.minecraft.network.chat.RemoteChatSession$Data $$6) {
+        this.profileId = $$0;
+        this.profile = $$1;
+        this.listed = $$2;
+        this.latency = $$3;
+        this.gameMode = $$4;
+        this.displayName = $$5;
+        this.chatSession = $$6;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry.class, "profileId;profile;listed;latency;gameMode;displayName;chatSession", "profileId", "profile", "listed", "latency", "gameMode", "displayName", "chatSession"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry.class, "profileId;profile;listed;latency;gameMode;displayName;chatSession", "profileId", "profile", "listed", "latency", "gameMode", "displayName", "chatSession"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry.class, "profileId;profile;listed;latency;gameMode;displayName;chatSession", "profileId", "profile", "listed", "latency", "gameMode", "displayName", "chatSession"}, this, $$0);
+    }
+
+    public java.util.UUID profileId() {
+        return this.profileId;
+    }
+
+    public com.mojang.authlib.GameProfile profile() {
+        return this.profile;
+    }
+
+    public boolean listed() {
+        return this.listed;
+    }
+
+    public int latency() {
+        return this.latency;
+    }
+
+    public net.minecraft.world.level.GameType gameMode() {
+        return this.gameMode;
+    }
+
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.Component displayName() {
+        return this.displayName;
+    }
+
+    public net.minecraft.network.chat.RemoteChatSession$Data chatSession() {
+        return this.chatSession;
+    }
+}
+