--- net/minecraft/network/protocol/game/ClientboundLevelParticlesPacket.java
+++ net/minecraft/network/protocol/game/ClientboundLevelParticlesPacket.java
@@ -17,17 +17,17 @@
     public boolean overrideLimiter;
     public net.minecraft.core.particles.ParticleOptions particle;
 
-    public <T extends net.minecraft.core.particles.ParticleOptions> net.minecraft.network.protocol.game.ClientboundLevelParticlesPacket(T particle, boolean overrideLimiter, double x, double y, double z, float xDist, float yDist, float zDist, float maxSpeed, int count) {
-        this.particle = particle;
-        this.overrideLimiter = overrideLimiter;
-        this.x = x;
-        this.y = y;
-        this.z = z;
-        this.xDist = xDist;
-        this.yDist = yDist;
-        this.zDist = zDist;
-        this.maxSpeed = maxSpeed;
-        this.count = count;
+    public <T extends net.minecraft.core.particles.ParticleOptions> net.minecraft.network.protocol.game.ClientboundLevelParticlesPacket(T $$0, boolean $$1, double $$2, double $$3, double $$4, float $$5, float $$6, float $$7, float $$8, int $$9) {
+        this.particle = $$0;
+        this.overrideLimiter = $$1;
+        this.x = $$2;
+        this.y = $$3;
+        this.z = $$4;
+        this.xDist = $$5;
+        this.yDist = $$6;
+        this.zDist = $$7;
+        this.maxSpeed = $$8;
+        this.count = $$9;
     }
 
     public net.minecraft.network.protocol.game.ClientboundLevelParticlesPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -44,22 +44,22 @@
         this.particle = this.readParticle($$0, $$1);
     }
 
-    public <T extends net.minecraft.core.particles.ParticleOptions> T readParticle(net.minecraft.network.FriendlyByteBuf buffer, net.minecraft.core.particles.ParticleType<T> particleType) {
-        return (T)particleType.getDeserializer().fromNetwork(particleType, buffer);
+    public <T extends net.minecraft.core.particles.ParticleOptions> T readParticle(net.minecraft.network.FriendlyByteBuf $$0, net.minecraft.core.particles.ParticleType<T> $$1) {
+        return (T)$$1.getDeserializer().fromNetwork($$1, $$0);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.PARTICLE_TYPE, (java.lang.Object)this.particle.getType());
-        buffer.writeBoolean(this.overrideLimiter);
-        buffer.writeDouble(this.x);
-        buffer.writeDouble(this.y);
-        buffer.writeDouble(this.z);
-        buffer.writeFloat(this.xDist);
-        buffer.writeFloat(this.yDist);
-        buffer.writeFloat(this.zDist);
-        buffer.writeFloat(this.maxSpeed);
-        buffer.writeInt(this.count);
-        this.particle.writeToNetwork(buffer);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.PARTICLE_TYPE, (java.lang.Object)this.particle.getType());
+        $$0.writeBoolean(this.overrideLimiter);
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeFloat(this.xDist);
+        $$0.writeFloat(this.yDist);
+        $$0.writeFloat(this.zDist);
+        $$0.writeFloat(this.maxSpeed);
+        $$0.writeInt(this.count);
+        this.particle.writeToNetwork($$0);
     }
 
     public boolean isOverrideLimiter() {
@@ -102,8 +102,8 @@
         return this.particle;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleParticleEvent(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleParticleEvent(this);
     }
 }
 