--- net/minecraft/network/protocol/game/ClientboundSetEntityLinkPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSetEntityLinkPacket.java
@@ -9,9 +9,9 @@
     public int sourceId;
     public int destId;
 
-    public net.minecraft.network.protocol.game.ClientboundSetEntityLinkPacket(net.minecraft.world.entity.Entity source, @javax.annotation.Nullable net.minecraft.world.entity.Entity destination) {
-        this.sourceId = source.getId();
-        this.destId = destination != null ? destination.getId() : 0;
+    public net.minecraft.network.protocol.game.ClientboundSetEntityLinkPacket(net.minecraft.world.entity.Entity $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1) {
+        this.sourceId = $$0.getId();
+        this.destId = $$1 != null ? $$1.getId() : 0;
     }
 
     public net.minecraft.network.protocol.game.ClientboundSetEntityLinkPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,13 +19,13 @@
         this.destId = $$0.readInt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeInt(this.sourceId);
-        buffer.writeInt(this.destId);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeInt(this.sourceId);
+        $$0.writeInt(this.destId);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleEntityLinkPacket(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleEntityLinkPacket(this);
     }
 
     public int getSourceId() {