--- net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket$EntryBuilder.java
+++ net/minecraft/network/protocol/game/ClientboundPlayerInfoUpdatePacket$EntryBuilder.java
@@ -1,0 +1,27 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.network.protocol.game;
+
+public class net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$EntryBuilder
+ {
+    public java.util.UUID profileId;
+    public com.mojang.authlib.GameProfile profile;
+    public boolean listed;
+    public int latency;
+    public net.minecraft.world.level.GameType gameMode;
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.Component displayName;
+    public net.minecraft.network.chat.RemoteChatSession$Data chatSession;
+
+    public net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$EntryBuilder(java.util.UUID $$0) {
+        this.gameMode = net.minecraft.world.level.GameType.DEFAULT_MODE;
+        this.profileId = $$0;
+        this.profile = new com.mojang.authlib.GameProfile($$0, null);
+    }
+
+    public net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry build() {
+        return new net.minecraft.network.protocol.game.ClientboundPlayerInfoUpdatePacket$Entry(this.profileId, this.profile, this.listed, this.latency, this.gameMode, this.displayName, this.chatSession);
+    }
+}
+