--- net/minecraft/network/protocol/game/ClientboundUpdateTagsPacket.java
+++ net/minecraft/network/protocol/game/ClientboundUpdateTagsPacket.java
@@ -16,12 +16,12 @@
         this.tags = $$02.readMap($$0 -> net.minecraft.resources.ResourceKey.createRegistryKey((net.minecraft.resources.ResourceLocation)$$0.readResourceLocation()), net.minecraft.tags.TagNetworkSerialization$NetworkPayload::read);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeMap(this.tags, ($$0, $$1) -> $$0.writeResourceLocation($$1.location()), ($$0, $$1) -> $$1.write($$0));
+    public void write(net.minecraft.network.FriendlyByteBuf $$02) {
+        $$02.writeMap(this.tags, ($$0, $$1) -> $$0.writeResourceLocation($$1.location()), ($$0, $$1) -> $$1.write($$0));
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleUpdateTags(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleUpdateTags(this);
     }
 
     public java.util.Map<net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<?>>, net.minecraft.tags.TagNetworkSerialization$NetworkPayload> getTags() {