--- net/minecraft/network/protocol/game/ClientboundSelectAdvancementsTabPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSelectAdvancementsTabPacket.java
@@ -9,20 +9,20 @@
     @javax.annotation.Nullable
     public net.minecraft.resources.ResourceLocation tab;
 
-    public net.minecraft.network.protocol.game.ClientboundSelectAdvancementsTabPacket(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation tab) {
-        this.tab = tab;
+    public net.minecraft.network.protocol.game.ClientboundSelectAdvancementsTabPacket(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation $$0) {
+        this.tab = $$0;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSelectAdvancementsTab(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSelectAdvancementsTab(this);
     }
 
     public net.minecraft.network.protocol.game.ClientboundSelectAdvancementsTabPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.tab = (net.minecraft.resources.ResourceLocation)$$0.readNullable(net.minecraft.network.FriendlyByteBuf::readResourceLocation);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeNullable((java.lang.Object)this.tab, net.minecraft.network.FriendlyByteBuf::writeResourceLocation);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeNullable((java.lang.Object)this.tab, net.minecraft.network.FriendlyByteBuf::writeResourceLocation);
     }
 
     @javax.annotation.Nullable