--- net/minecraft/network/protocol/game/ClientboundSetEntityMotionPacket.java
+++ net/minecraft/network/protocol/game/ClientboundSetEntityMotionPacket.java
@@ -11,19 +11,19 @@
     public int ya;
     public int za;
 
-    public net.minecraft.network.protocol.game.ClientboundSetEntityMotionPacket(net.minecraft.world.entity.Entity entity) {
-        this(entity.getId(), entity.getDeltaMovement());
+    public net.minecraft.network.protocol.game.ClientboundSetEntityMotionPacket(net.minecraft.world.entity.Entity $$0) {
+        this($$0.getId(), $$0.getDeltaMovement());
     }
 
-    public net.minecraft.network.protocol.game.ClientboundSetEntityMotionPacket(int id, net.minecraft.world.phys.Vec3 deltaMovement) {
-        this.id = id;
-        double d = 3.9;
-        double d2 = net.minecraft.util.Mth.clamp((double)deltaMovement.x, (double)-3.9, (double)3.9);
-        double d3 = net.minecraft.util.Mth.clamp((double)deltaMovement.y, (double)-3.9, (double)3.9);
-        double d4 = net.minecraft.util.Mth.clamp((double)deltaMovement.z, (double)-3.9, (double)3.9);
-        this.xa = (int)(d2 * 8000.0);
-        this.ya = (int)(d3 * 8000.0);
-        this.za = (int)(d4 * 8000.0);
+    public net.minecraft.network.protocol.game.ClientboundSetEntityMotionPacket(int $$0, net.minecraft.world.phys.Vec3 $$1) {
+        this.id = $$0;
+        double $$2 = 3.9;
+        double $$3 = net.minecraft.util.Mth.clamp((double)$$1.x, (double)-3.9, (double)3.9);
+        double $$4 = net.minecraft.util.Mth.clamp((double)$$1.y, (double)-3.9, (double)3.9);
+        double $$5 = net.minecraft.util.Mth.clamp((double)$$1.z, (double)-3.9, (double)3.9);
+        this.xa = (int)($$3 * 8000.0);
+        this.ya = (int)($$4 * 8000.0);
+        this.za = (int)($$5 * 8000.0);
     }
 
     public net.minecraft.network.protocol.game.ClientboundSetEntityMotionPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -33,15 +33,15 @@
         this.za = $$0.readShort();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.id);
-        buffer.writeShort(this.xa);
-        buffer.writeShort(this.ya);
-        buffer.writeShort(this.za);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.id);
+        $$0.writeShort(this.xa);
+        $$0.writeShort(this.ya);
+        $$0.writeShort(this.za);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleSetEntityMotion(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleSetEntityMotion(this);
     }
 
     public int getId() {