--- net/minecraft/network/protocol/game/ClientboundMapItemDataPacket.java
+++ net/minecraft/network/protocol/game/ClientboundMapItemDataPacket.java
@@ -46,11 +46,11 @@
         }
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.mapId);
-        buffer.writeByte((int)this.scale);
-        buffer.writeBoolean(this.locked);
-        buffer.writeNullable((java.lang.Object)this.decorations, ($$02, $$12) -> $$02.writeCollection((java.util.Collection)$$12, ($$0, $$1) -> {
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.mapId);
+        $$0.writeByte((int)this.scale);
+        $$0.writeBoolean(this.locked);
+        $$0.writeNullable((java.lang.Object)this.decorations, ($$02, $$12) -> $$02.writeCollection((java.util.Collection)$$12, ($$0, $$1) -> {
             $$0.writeEnum((java.lang.Enum)$$1.getType());
             $$0.writeByte((int)$$1.getX());
             $$0.writeByte((int)$$1.getY());
@@ -58,30 +58,30 @@
             $$0.writeNullable((java.lang.Object)$$1.getName(), net.minecraft.network.FriendlyByteBuf::writeComponent);
         }));
         if (this.colorPatch != null) {
-            buffer.writeByte(this.colorPatch.width);
-            buffer.writeByte(this.colorPatch.height);
-            buffer.writeByte(this.colorPatch.startX);
-            buffer.writeByte(this.colorPatch.startY);
-            buffer.writeByteArray(this.colorPatch.mapColors);
+            $$0.writeByte(this.colorPatch.width);
+            $$0.writeByte(this.colorPatch.height);
+            $$0.writeByte(this.colorPatch.startX);
+            $$0.writeByte(this.colorPatch.startY);
+            $$0.writeByteArray(this.colorPatch.mapColors);
         } else {
-            buffer.writeByte(0);
+            $$0.writeByte(0);
         }
     }
 
-    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener handler) {
-        handler.handleMapItemData(this);
+    public void handle(net.minecraft.network.protocol.game.ClientGamePacketListener $$0) {
+        $$0.handleMapItemData(this);
     }
 
     public int getMapId() {
         return this.mapId;
     }
 
-    public void applyToMap(net.minecraft.world.level.saveddata.maps.MapItemSavedData mapdata) {
+    public void applyToMap(net.minecraft.world.level.saveddata.maps.MapItemSavedData $$0) {
         if (this.decorations != null) {
-            mapdata.addClientSideDecorations(this.decorations);
+            $$0.addClientSideDecorations(this.decorations);
         }
         if (this.colorPatch != null) {
-            this.colorPatch.applyToMap(mapdata);
+            this.colorPatch.applyToMap($$0);
         }
     }
 