--- net/minecraft/network/protocol/game/ServerboundRecipeBookSeenRecipePacket.java
+++ net/minecraft/network/protocol/game/ServerboundRecipeBookSeenRecipePacket.java
@@ -8,20 +8,20 @@
  {
     public net.minecraft.resources.ResourceLocation recipe;
 
-    public net.minecraft.network.protocol.game.ServerboundRecipeBookSeenRecipePacket(net.minecraft.world.item.crafting.Recipe<?> recipe) {
-        this.recipe = recipe.getId();
+    public net.minecraft.network.protocol.game.ServerboundRecipeBookSeenRecipePacket(net.minecraft.world.item.crafting.Recipe<?> $$0) {
+        this.recipe = $$0.getId();
     }
 
     public net.minecraft.network.protocol.game.ServerboundRecipeBookSeenRecipePacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.recipe = $$0.readResourceLocation();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeResourceLocation(this.recipe);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeResourceLocation(this.recipe);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleRecipeBookSeenRecipePacket(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleRecipeBookSeenRecipePacket(this);
     }
 
     public net.minecraft.resources.ResourceLocation getRecipe() {