--- net/minecraft/network/protocol/game/ServerboundPlayerActionPacket.java
+++ net/minecraft/network/protocol/game/ServerboundPlayerActionPacket.java
@@ -18,8 +18,8 @@
         this.sequence = $$3;
     }
 
-    public net.minecraft.network.protocol.game.ServerboundPlayerActionPacket(net.minecraft.network.protocol.game.ServerboundPlayerActionPacket$Action action, net.minecraft.core.BlockPos pos, net.minecraft.core.Direction direction) {
-        this(action, pos, direction, 0);
+    public net.minecraft.network.protocol.game.ServerboundPlayerActionPacket(net.minecraft.network.protocol.game.ServerboundPlayerActionPacket$Action $$0, net.minecraft.core.BlockPos $$1, net.minecraft.core.Direction $$2) {
+        this($$0, $$1, $$2, 0);
     }
 
     public net.minecraft.network.protocol.game.ServerboundPlayerActionPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -29,15 +29,15 @@
         this.sequence = $$0.readVarInt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeEnum((java.lang.Enum)this.action);
-        buffer.writeBlockPos(this.pos);
-        buffer.writeByte(this.direction.get3DDataValue());
-        buffer.writeVarInt(this.sequence);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeEnum((java.lang.Enum)this.action);
+        $$0.writeBlockPos(this.pos);
+        $$0.writeByte(this.direction.get3DDataValue());
+        $$0.writeVarInt(this.sequence);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handlePlayerAction(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handlePlayerAction(this);
     }
 
     public net.minecraft.core.BlockPos getPos() {