--- net/minecraft/network/protocol/game/ServerboundContainerClickPacket.java
+++ net/minecraft/network/protocol/game/ServerboundContainerClickPacket.java
@@ -36,18 +36,18 @@
         this.carriedItem = $$02.readItem();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeByte(this.containerId);
-        buffer.writeVarInt(this.stateId);
-        buffer.writeShort(this.slotNum);
-        buffer.writeByte(this.buttonNum);
-        buffer.writeEnum((java.lang.Enum)this.clickType);
-        buffer.writeMap((java.util.Map)this.changedSlots, net.minecraft.network.FriendlyByteBuf::writeShort, net.minecraft.network.FriendlyByteBuf::writeItem);
-        buffer.writeItem(this.carriedItem);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeByte(this.containerId);
+        $$0.writeVarInt(this.stateId);
+        $$0.writeShort(this.slotNum);
+        $$0.writeByte(this.buttonNum);
+        $$0.writeEnum((java.lang.Enum)this.clickType);
+        $$0.writeMap((java.util.Map)this.changedSlots, net.minecraft.network.FriendlyByteBuf::writeShort, net.minecraft.network.FriendlyByteBuf::writeItem);
+        $$0.writeItem(this.carriedItem);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleContainerClick(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleContainerClick(this);
     }
 
     public int getContainerId() {