--- net/minecraft/network/protocol/game/ServerboundSetBeaconPacket.java
+++ net/minecraft/network/protocol/game/ServerboundSetBeaconPacket.java
@@ -19,13 +19,13 @@
         this.secondary = $$02.readOptional($$0 -> (net.minecraft.world.effect.MobEffect)$$0.readById((net.minecraft.core.IdMap)net.minecraft.core.Registry.MOB_EFFECT));
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeOptional(this.primary, ($$0, $$1) -> $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MOB_EFFECT, $$1));
-        buffer.writeOptional(this.secondary, ($$0, $$1) -> $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MOB_EFFECT, $$1));
+    public void write(net.minecraft.network.FriendlyByteBuf $$02) {
+        $$02.writeOptional(this.primary, ($$0, $$1) -> $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MOB_EFFECT, $$1));
+        $$02.writeOptional(this.secondary, ($$0, $$1) -> $$0.writeId((net.minecraft.core.IdMap)net.minecraft.core.Registry.MOB_EFFECT, $$1));
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleSetBeaconPacket(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleSetBeaconPacket(this);
     }
 
     public java.util.Optional<net.minecraft.world.effect.MobEffect> getPrimary() {