--- net/minecraft/network/protocol/game/ServerboundSeenAdvancementsPacket.java
+++ net/minecraft/network/protocol/game/ServerboundSeenAdvancementsPacket.java
@@ -10,13 +10,13 @@
     @javax.annotation.Nullable
     public net.minecraft.resources.ResourceLocation tab;
 
-    public net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket(net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket$Action action, @javax.annotation.Nullable net.minecraft.resources.ResourceLocation tab) {
-        this.action = action;
-        this.tab = tab;
+    public net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket(net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket$Action $$0, @javax.annotation.Nullable net.minecraft.resources.ResourceLocation $$1) {
+        this.action = $$0;
+        this.tab = $$1;
     }
 
-    public static net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket openedTab(net.minecraft.advancements.Advancement advancement) {
-        return new net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket(net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket$Action.OPENED_TAB, advancement.getId());
+    public static net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket openedTab(net.minecraft.advancements.Advancement $$0) {
+        return new net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket(net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket$Action.OPENED_TAB, $$0.getId());
     }
 
     public static net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket closedScreen() {
@@ -28,15 +28,15 @@
         this.tab = this.action == net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket$Action.OPENED_TAB ? $$0.readResourceLocation() : null;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeEnum((java.lang.Enum)this.action);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeEnum((java.lang.Enum)this.action);
         if (this.action == net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket$Action.OPENED_TAB) {
-            buffer.writeResourceLocation(this.tab);
+            $$0.writeResourceLocation(this.tab);
         }
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleSeenAdvancements(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleSeenAdvancements(this);
     }
 
     public net.minecraft.network.protocol.game.ServerboundSeenAdvancementsPacket$Action getAction() {