--- net/minecraft/network/protocol/game/ServerboundEditBookPacket.java
+++ net/minecraft/network/protocol/game/ServerboundEditBookPacket.java
@@ -26,14 +26,14 @@
         this.title = $$02.readOptional($$0 -> $$0.readUtf(128));
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.slot);
-        buffer.writeCollection((java.util.Collection)this.pages, ($$0, $$1) -> $$0.writeUtf($$1, 8192));
-        buffer.writeOptional(this.title, ($$0, $$1) -> $$0.writeUtf($$1, 128));
+    public void write(net.minecraft.network.FriendlyByteBuf $$02) {
+        $$02.writeVarInt(this.slot);
+        $$02.writeCollection((java.util.Collection)this.pages, ($$0, $$1) -> $$0.writeUtf($$1, 8192));
+        $$02.writeOptional(this.title, ($$0, $$1) -> $$0.writeUtf($$1, 128));
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleEditBook(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleEditBook(this);
     }
 
     public java.util.List<java.lang.String> getPages() {