--- net/minecraft/network/protocol/game/ServerboundContainerButtonClickPacket.java
+++ net/minecraft/network/protocol/game/ServerboundContainerButtonClickPacket.java
@@ -9,13 +9,13 @@
     public int containerId;
     public int buttonId;
 
-    public net.minecraft.network.protocol.game.ServerboundContainerButtonClickPacket(int containerId, int buttonId) {
-        this.containerId = containerId;
-        this.buttonId = buttonId;
+    public net.minecraft.network.protocol.game.ServerboundContainerButtonClickPacket(int $$0, int $$1) {
+        this.containerId = $$0;
+        this.buttonId = $$1;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleContainerButtonClick(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleContainerButtonClick(this);
     }
 
     public net.minecraft.network.protocol.game.ServerboundContainerButtonClickPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -23,9 +23,9 @@
         this.buttonId = $$0.readByte();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeByte(this.containerId);
-        buffer.writeByte(this.buttonId);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeByte(this.containerId);
+        $$0.writeByte(this.buttonId);
     }
 
     public int getContainerId() {