--- net/minecraft/network/protocol/game/ServerboundTeleportToEntityPacket.java
+++ net/minecraft/network/protocol/game/ServerboundTeleportToEntityPacket.java
@@ -8,25 +8,25 @@
  {
     public java.util.UUID uuid;
 
-    public net.minecraft.network.protocol.game.ServerboundTeleportToEntityPacket(java.util.UUID uuid) {
-        this.uuid = uuid;
+    public net.minecraft.network.protocol.game.ServerboundTeleportToEntityPacket(java.util.UUID $$0) {
+        this.uuid = $$0;
     }
 
     public net.minecraft.network.protocol.game.ServerboundTeleportToEntityPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.uuid = $$0.readUUID();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUUID(this.uuid);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUUID(this.uuid);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleTeleportToEntityPacket(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleTeleportToEntityPacket(this);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.Entity getEntity(net.minecraft.server.level.ServerLevel level) {
-        return level.getEntity(this.uuid);
+    public net.minecraft.world.entity.Entity getEntity(net.minecraft.server.level.ServerLevel $$0) {
+        return $$0.getEntity(this.uuid);
     }
 }
 