--- net/minecraft/network/protocol/game/ServerboundPlayerAbilitiesPacket.java
+++ net/minecraft/network/protocol/game/ServerboundPlayerAbilitiesPacket.java
@@ -9,8 +9,8 @@
     public static int FLAG_FLYING = 2;
     public boolean isFlying;
 
-    public net.minecraft.network.protocol.game.ServerboundPlayerAbilitiesPacket(net.minecraft.world.entity.player.Abilities abilities) {
-        this.isFlying = abilities.flying;
+    public net.minecraft.network.protocol.game.ServerboundPlayerAbilitiesPacket(net.minecraft.world.entity.player.Abilities $$0) {
+        this.isFlying = $$0.flying;
     }
 
     public net.minecraft.network.protocol.game.ServerboundPlayerAbilitiesPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -18,16 +18,16 @@
         this.isFlying = ($$1 & 2) != 0;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        int n = 0;
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        int $$1 = 0;
         if (this.isFlying) {
-            n = (byte)(n | 2);
+            $$1 = (byte)($$1 | 2);
         }
-        buffer.writeByte(n);
+        $$0.writeByte($$1);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handlePlayerAbilities(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handlePlayerAbilities(this);
     }
 
     public boolean isFlying() {