--- net/minecraft/network/protocol/game/ServerboundRenameItemPacket.java
+++ net/minecraft/network/protocol/game/ServerboundRenameItemPacket.java
@@ -8,20 +8,20 @@
  {
     public java.lang.String name;
 
-    public net.minecraft.network.protocol.game.ServerboundRenameItemPacket(java.lang.String name) {
-        this.name = name;
+    public net.minecraft.network.protocol.game.ServerboundRenameItemPacket(java.lang.String $$0) {
+        this.name = $$0;
     }
 
     public net.minecraft.network.protocol.game.ServerboundRenameItemPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.name = $$0.readUtf();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUtf(this.name);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUtf(this.name);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleRenameItem(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleRenameItem(this);
     }
 
     public java.lang.String getName() {