--- net/minecraft/network/protocol/game/ServerboundBlockEntityTagQuery.java
+++ net/minecraft/network/protocol/game/ServerboundBlockEntityTagQuery.java
@@ -9,9 +9,9 @@
     public int transactionId;
     public net.minecraft.core.BlockPos pos;
 
-    public net.minecraft.network.protocol.game.ServerboundBlockEntityTagQuery(int transactionId, net.minecraft.core.BlockPos pos) {
-        this.transactionId = transactionId;
-        this.pos = pos;
+    public net.minecraft.network.protocol.game.ServerboundBlockEntityTagQuery(int $$0, net.minecraft.core.BlockPos $$1) {
+        this.transactionId = $$0;
+        this.pos = $$1;
     }
 
     public net.minecraft.network.protocol.game.ServerboundBlockEntityTagQuery(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,13 +19,13 @@
         this.pos = $$0.readBlockPos();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.transactionId);
-        buffer.writeBlockPos(this.pos);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.transactionId);
+        $$0.writeBlockPos(this.pos);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleBlockEntityTagQuery(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleBlockEntityTagQuery(this);
     }
 
     public int getTransactionId() {