--- net/minecraft/network/protocol/game/ServerboundClientInformationPacket.java
+++ net/minecraft/network/protocol/game/ServerboundClientInformationPacket.java
@@ -32,19 +32,19 @@
         this.allowsListing = $$7;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeUtf(this.language);
-        buffer.writeByte(this.viewDistance);
-        buffer.writeEnum((java.lang.Enum)this.chatVisibility);
-        buffer.writeBoolean(this.chatColors);
-        buffer.writeByte(this.modelCustomisation);
-        buffer.writeEnum((java.lang.Enum)this.mainHand);
-        buffer.writeBoolean(this.textFilteringEnabled);
-        buffer.writeBoolean(this.allowsListing);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeUtf(this.language);
+        $$0.writeByte(this.viewDistance);
+        $$0.writeEnum((java.lang.Enum)this.chatVisibility);
+        $$0.writeBoolean(this.chatColors);
+        $$0.writeByte(this.modelCustomisation);
+        $$0.writeEnum((java.lang.Enum)this.mainHand);
+        $$0.writeBoolean(this.textFilteringEnabled);
+        $$0.writeBoolean(this.allowsListing);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleClientInformation(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleClientInformation(this);
     }
 
     public java.lang.String toString() {