--- net/minecraft/network/protocol/game/ServerboundSetCarriedItemPacket.java
+++ net/minecraft/network/protocol/game/ServerboundSetCarriedItemPacket.java
@@ -8,20 +8,20 @@
  {
     public int slot;
 
-    public net.minecraft.network.protocol.game.ServerboundSetCarriedItemPacket(int slot) {
-        this.slot = slot;
+    public net.minecraft.network.protocol.game.ServerboundSetCarriedItemPacket(int $$0) {
+        this.slot = $$0;
     }
 
     public net.minecraft.network.protocol.game.ServerboundSetCarriedItemPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.slot = $$0.readShort();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeShort(this.slot);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeShort(this.slot);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleSetCarriedItem(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleSetCarriedItem(this);
     }
 
     public int getSlot() {