--- net/minecraft/network/protocol/game/ServerboundAcceptTeleportationPacket.java
+++ net/minecraft/network/protocol/game/ServerboundAcceptTeleportationPacket.java
@@ -8,20 +8,20 @@
  {
     public int id;
 
-    public net.minecraft.network.protocol.game.ServerboundAcceptTeleportationPacket(int id) {
-        this.id = id;
+    public net.minecraft.network.protocol.game.ServerboundAcceptTeleportationPacket(int $$0) {
+        this.id = $$0;
     }
 
     public net.minecraft.network.protocol.game.ServerboundAcceptTeleportationPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.id = $$0.readVarInt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.id);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.id);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleAcceptTeleportPacket(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleAcceptTeleportPacket(this);
     }
 
     public int getId() {