--- net/minecraft/network/protocol/game/ServerboundPlaceRecipePacket.java
+++ net/minecraft/network/protocol/game/ServerboundPlaceRecipePacket.java
@@ -10,10 +10,10 @@
     public net.minecraft.resources.ResourceLocation recipe;
     public boolean shiftDown;
 
-    public net.minecraft.network.protocol.game.ServerboundPlaceRecipePacket(int containerId, net.minecraft.world.item.crafting.Recipe<?> recipe, boolean shiftDown) {
-        this.containerId = containerId;
-        this.recipe = recipe.getId();
-        this.shiftDown = shiftDown;
+    public net.minecraft.network.protocol.game.ServerboundPlaceRecipePacket(int $$0, net.minecraft.world.item.crafting.Recipe<?> $$1, boolean $$2) {
+        this.containerId = $$0;
+        this.recipe = $$1.getId();
+        this.shiftDown = $$2;
     }
 
     public net.minecraft.network.protocol.game.ServerboundPlaceRecipePacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -22,14 +22,14 @@
         this.shiftDown = $$0.readBoolean();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeByte(this.containerId);
-        buffer.writeResourceLocation(this.recipe);
-        buffer.writeBoolean(this.shiftDown);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeByte(this.containerId);
+        $$0.writeResourceLocation(this.recipe);
+        $$0.writeBoolean(this.shiftDown);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handlePlaceRecipe(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handlePlaceRecipe(this);
     }
 
     public int getContainerId() {