--- net/minecraft/network/protocol/game/ServerboundMovePlayerPacket$Pos.java
+++ net/minecraft/network/protocol/game/ServerboundMovePlayerPacket$Pos.java
@@ -6,8 +6,8 @@
 public class net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos
 extends net.minecraft.network.protocol.game.ServerboundMovePlayerPacket
  {
-    public net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos(double x, double y, double z, boolean onGround) {
-        super(x, y, z, 0.0f, 0.0f, onGround, true, false);
+    public net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos(double $$0, double $$1, double $$2, boolean $$3) {
+        super($$0, $$1, $$2, 0.0f, 0.0f, $$3, true, false);
     }
 
     public static net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos read(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -18,11 +18,11 @@
         return new net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos($$1, $$2, $$3, $$4);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeDouble(this.x);
-        buffer.writeDouble(this.y);
-        buffer.writeDouble(this.z);
-        buffer.writeByte(this.onGround ? 1 : 0);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeByte(this.onGround ? 1 : 0);
     }
 }
 