--- net/minecraft/network/protocol/game/ServerboundSetJigsawBlockPacket.java
+++ net/minecraft/network/protocol/game/ServerboundSetJigsawBlockPacket.java
@@ -13,13 +13,13 @@
     public java.lang.String finalState;
     public net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType joint;
 
-    public net.minecraft.network.protocol.game.ServerboundSetJigsawBlockPacket(net.minecraft.core.BlockPos pos, net.minecraft.resources.ResourceLocation name, net.minecraft.resources.ResourceLocation target, net.minecraft.resources.ResourceLocation pool, java.lang.String finalState, net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType joint) {
-        this.pos = pos;
-        this.name = name;
-        this.target = target;
-        this.pool = pool;
-        this.finalState = finalState;
-        this.joint = joint;
+    public net.minecraft.network.protocol.game.ServerboundSetJigsawBlockPacket(net.minecraft.core.BlockPos $$0, net.minecraft.resources.ResourceLocation $$1, net.minecraft.resources.ResourceLocation $$2, net.minecraft.resources.ResourceLocation $$3, java.lang.String $$4, net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType $$5) {
+        this.pos = $$0;
+        this.name = $$1;
+        this.target = $$2;
+        this.pool = $$3;
+        this.finalState = $$4;
+        this.joint = $$5;
     }
 
     public net.minecraft.network.protocol.game.ServerboundSetJigsawBlockPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -31,17 +31,17 @@
         this.joint = (net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType)net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType.byName((java.lang.String)$$0.readUtf()).orElse((java.lang.Object)net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType.ALIGNED);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeBlockPos(this.pos);
-        buffer.writeResourceLocation(this.name);
-        buffer.writeResourceLocation(this.target);
-        buffer.writeResourceLocation(this.pool);
-        buffer.writeUtf(this.finalState);
-        buffer.writeUtf(this.joint.getSerializedName());
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeBlockPos(this.pos);
+        $$0.writeResourceLocation(this.name);
+        $$0.writeResourceLocation(this.target);
+        $$0.writeResourceLocation(this.pool);
+        $$0.writeUtf(this.finalState);
+        $$0.writeUtf(this.joint.getSerializedName());
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleSetJigsawBlock(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleSetJigsawBlock(this);
     }
 
     public net.minecraft.core.BlockPos getPos() {