--- net/minecraft/network/protocol/game/ServerboundResourcePackPacket.java
+++ net/minecraft/network/protocol/game/ServerboundResourcePackPacket.java
@@ -8,20 +8,20 @@
  {
     public net.minecraft.network.protocol.game.ServerboundResourcePackPacket$Action action;
 
-    public net.minecraft.network.protocol.game.ServerboundResourcePackPacket(net.minecraft.network.protocol.game.ServerboundResourcePackPacket$Action action) {
-        this.action = action;
+    public net.minecraft.network.protocol.game.ServerboundResourcePackPacket(net.minecraft.network.protocol.game.ServerboundResourcePackPacket$Action $$0) {
+        this.action = $$0;
     }
 
     public net.minecraft.network.protocol.game.ServerboundResourcePackPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.action = (net.minecraft.network.protocol.game.ServerboundResourcePackPacket$Action)$$0.readEnum(net.minecraft.network.protocol.game.ServerboundResourcePackPacket$Action.class);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeEnum((java.lang.Enum)this.action);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeEnum((java.lang.Enum)this.action);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleResourcePackResponse(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleResourcePackResponse(this);
     }
 
     public net.minecraft.network.protocol.game.ServerboundResourcePackPacket$Action getAction() {