--- net/minecraft/network/protocol/game/ServerboundPlayerInputPacket.java
+++ net/minecraft/network/protocol/game/ServerboundPlayerInputPacket.java
@@ -13,11 +13,11 @@
     public boolean isJumping;
     public boolean isShiftKeyDown;
 
-    public net.minecraft.network.protocol.game.ServerboundPlayerInputPacket(float xxa, float zza, boolean isJumping, boolean isShiftKeyDown) {
-        this.xxa = xxa;
-        this.zza = zza;
-        this.isJumping = isJumping;
-        this.isShiftKeyDown = isShiftKeyDown;
+    public net.minecraft.network.protocol.game.ServerboundPlayerInputPacket(float $$0, float $$1, boolean $$2, boolean $$3) {
+        this.xxa = $$0;
+        this.zza = $$1;
+        this.isJumping = $$2;
+        this.isShiftKeyDown = $$3;
     }
 
     public net.minecraft.network.protocol.game.ServerboundPlayerInputPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -28,21 +28,21 @@
         this.isShiftKeyDown = ($$1 & 2) > 0;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeFloat(this.xxa);
-        buffer.writeFloat(this.zza);
-        byte by = 0;
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeFloat(this.xxa);
+        $$0.writeFloat(this.zza);
+        byte $$1 = 0;
         if (this.isJumping) {
-            by = (byte)(by | 1);
+            $$1 = (byte)($$1 | 1);
         }
         if (this.isShiftKeyDown) {
-            by = (byte)(by | 2);
+            $$1 = (byte)($$1 | 2);
         }
-        buffer.writeByte((int)by);
+        $$0.writeByte((int)$$1);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handlePlayerInput(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handlePlayerInput(this);
     }
 
     public float getXxa() {