--- net/minecraft/network/protocol/game/ServerboundInteractPacket.java
+++ net/minecraft/network/protocol/game/ServerboundInteractPacket.java
@@ -36,15 +36,15 @@
         this.usingSecondaryAction = $$0.readBoolean();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entityId);
-        buffer.writeEnum((java.lang.Enum)this.action.getType());
-        this.action.write(buffer);
-        buffer.writeBoolean(this.usingSecondaryAction);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.entityId);
+        $$0.writeEnum((java.lang.Enum)this.action.getType());
+        this.action.write($$0);
+        $$0.writeBoolean(this.usingSecondaryAction);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleInteract(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleInteract(this);
     }
 
     @javax.annotation.Nullable