--- net/minecraft/network/protocol/game/ServerboundMovePlayerPacket$PosRot.java
+++ net/minecraft/network/protocol/game/ServerboundMovePlayerPacket$PosRot.java
@@ -6,8 +6,8 @@
 public class net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$PosRot
 extends net.minecraft.network.protocol.game.ServerboundMovePlayerPacket
  {
-    public net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$PosRot(double x, double y, double z, float yRot, float xRot, boolean onGround) {
-        super(x, y, z, yRot, xRot, onGround, true, true);
+    public net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$PosRot(double $$0, double $$1, double $$2, float $$3, float $$4, boolean $$5) {
+        super($$0, $$1, $$2, $$3, $$4, $$5, true, true);
     }
 
     public static net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$PosRot read(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -20,13 +20,13 @@
         return new net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$PosRot($$1, $$2, $$3, $$4, $$5, $$6);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeDouble(this.x);
-        buffer.writeDouble(this.y);
-        buffer.writeDouble(this.z);
-        buffer.writeFloat(this.yRot);
-        buffer.writeFloat(this.xRot);
-        buffer.writeByte(this.onGround ? 1 : 0);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeDouble(this.x);
+        $$0.writeDouble(this.y);
+        $$0.writeDouble(this.z);
+        $$0.writeFloat(this.yRot);
+        $$0.writeFloat(this.xRot);
+        $$0.writeByte(this.onGround ? 1 : 0);
     }
 }
 