--- net/minecraft/network/protocol/game/ServerboundMovePlayerPacket.java
+++ net/minecraft/network/protocol/game/ServerboundMovePlayerPacket.java
@@ -26,28 +26,28 @@
         this.hasRot = $$7;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleMovePlayer(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleMovePlayer(this);
     }
 
-    public double getX(double defaultValue) {
-        return this.hasPos ? this.x : defaultValue;
+    public double getX(double $$0) {
+        return this.hasPos ? this.x : $$0;
     }
 
-    public double getY(double defaultValue) {
-        return this.hasPos ? this.y : defaultValue;
+    public double getY(double $$0) {
+        return this.hasPos ? this.y : $$0;
     }
 
-    public double getZ(double defaultValue) {
-        return this.hasPos ? this.z : defaultValue;
+    public double getZ(double $$0) {
+        return this.hasPos ? this.z : $$0;
     }
 
-    public float getYRot(float defaultValue) {
-        return this.hasRot ? this.yRot : defaultValue;
+    public float getYRot(float $$0) {
+        return this.hasRot ? this.yRot : $$0;
     }
 
-    public float getXRot(float defaultValue) {
-        return this.hasRot ? this.xRot : defaultValue;
+    public float getXRot(float $$0) {
+        return this.hasRot ? this.xRot : $$0;
     }
 
     public boolean isOnGround() {