--- net/minecraft/network/protocol/game/ServerboundChangeDifficultyPacket.java
+++ net/minecraft/network/protocol/game/ServerboundChangeDifficultyPacket.java
@@ -8,20 +8,20 @@
  {
     public net.minecraft.world.Difficulty difficulty;
 
-    public net.minecraft.network.protocol.game.ServerboundChangeDifficultyPacket(net.minecraft.world.Difficulty difficulty) {
-        this.difficulty = difficulty;
+    public net.minecraft.network.protocol.game.ServerboundChangeDifficultyPacket(net.minecraft.world.Difficulty $$0) {
+        this.difficulty = $$0;
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleChangeDifficulty(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleChangeDifficulty(this);
     }
 
     public net.minecraft.network.protocol.game.ServerboundChangeDifficultyPacket(net.minecraft.network.FriendlyByteBuf $$0) {
         this.difficulty = net.minecraft.world.Difficulty.byId((int)$$0.readUnsignedByte());
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeByte(this.difficulty.getId());
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeByte(this.difficulty.getId());
     }
 
     public net.minecraft.world.Difficulty getDifficulty() {