--- net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
+++ net/minecraft/network/protocol/game/ServerboundChatCommandPacket.java
@@ -11,28 +11,25 @@
     public java.time.Instant timeStamp;
     public long salt;
     public net.minecraft.commands.arguments.ArgumentSignatures argumentSignatures;
-    public boolean signedPreview;
     public net.minecraft.network.chat.LastSeenMessages$Update lastSeenMessages;
 
-    public net.minecraft.network.protocol.game.ServerboundChatCommandPacket(java.lang.String $$0, java.time.Instant $$1, long $$2, net.minecraft.commands.arguments.ArgumentSignatures $$3, boolean $$4, net.minecraft.network.chat.LastSeenMessages$Update $$5) {
-        this.command = $$0 = net.minecraft.util.StringUtil.trimChatMessage((java.lang.String)$$0);
+    public net.minecraft.network.protocol.game.ServerboundChatCommandPacket(net.minecraft.network.FriendlyByteBuf $$0) {
+        this($$0.readUtf(256), $$0.readInstant(), $$0.readLong(), new net.minecraft.commands.arguments.ArgumentSignatures($$0), new net.minecraft.network.chat.LastSeenMessages$Update($$0));
+    }
+
+    public net.minecraft.network.protocol.game.ServerboundChatCommandPacket(java.lang.String $$0, java.time.Instant $$1, long $$2, net.minecraft.commands.arguments.ArgumentSignatures $$3, net.minecraft.network.chat.LastSeenMessages$Update $$4) {
+        this.command = $$0;
         this.timeStamp = $$1;
         this.salt = $$2;
         this.argumentSignatures = $$3;
-        this.signedPreview = $$4;
-        this.lastSeenMessages = $$5;
+        this.lastSeenMessages = $$4;
     }
 
-    public net.minecraft.network.protocol.game.ServerboundChatCommandPacket(net.minecraft.network.FriendlyByteBuf $$0) {
-        this($$0.readUtf(256), $$0.readInstant(), $$0.readLong(), new net.minecraft.commands.arguments.ArgumentSignatures($$0), $$0.readBoolean(), new net.minecraft.network.chat.LastSeenMessages$Update($$0));
-    }
-
     public void write(net.minecraft.network.FriendlyByteBuf $$0) {
         $$0.writeUtf(this.command, 256);
         $$0.writeInstant(this.timeStamp);
         $$0.writeLong(this.salt);
         this.argumentSignatures.write($$0);
-        $$0.writeBoolean(this.signedPreview);
         this.lastSeenMessages.write($$0);
     }
 
@@ -41,15 +38,15 @@
     }
 
     public java.lang.String toString() {
-        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ServerboundChatCommandPacket.class, "command;timeStamp;salt;argumentSignatures;signedPreview;lastSeenMessages", "command", "timeStamp", "salt", "argumentSignatures", "signedPreview", "lastSeenMessages"}, this);
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ServerboundChatCommandPacket.class, "command;timeStamp;salt;argumentSignatures;lastSeenMessages", "command", "timeStamp", "salt", "argumentSignatures", "lastSeenMessages"}, this);
     }
 
     public int hashCode() {
-        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ServerboundChatCommandPacket.class, "command;timeStamp;salt;argumentSignatures;signedPreview;lastSeenMessages", "command", "timeStamp", "salt", "argumentSignatures", "signedPreview", "lastSeenMessages"}, this);
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ServerboundChatCommandPacket.class, "command;timeStamp;salt;argumentSignatures;lastSeenMessages", "command", "timeStamp", "salt", "argumentSignatures", "lastSeenMessages"}, this);
     }
 
     public boolean equals(java.lang.Object $$0) {
-        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ServerboundChatCommandPacket.class, "command;timeStamp;salt;argumentSignatures;signedPreview;lastSeenMessages", "command", "timeStamp", "salt", "argumentSignatures", "signedPreview", "lastSeenMessages"}, this, $$0);
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.network.protocol.game.ServerboundChatCommandPacket.class, "command;timeStamp;salt;argumentSignatures;lastSeenMessages", "command", "timeStamp", "salt", "argumentSignatures", "lastSeenMessages"}, this, $$0);
     }
 
     public java.lang.String command() {
@@ -68,10 +65,6 @@
         return this.argumentSignatures;
     }
 
-    public boolean signedPreview() {
-        return this.signedPreview;
-    }
-
     public net.minecraft.network.chat.LastSeenMessages$Update lastSeenMessages() {
         return this.lastSeenMessages;
     }