--- net/minecraft/network/protocol/game/ServerboundUseItemOnPacket.java
+++ net/minecraft/network/protocol/game/ServerboundUseItemOnPacket.java
@@ -22,14 +22,14 @@
         this.sequence = $$0.readVarInt();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeEnum((java.lang.Enum)this.hand);
-        buffer.writeBlockHitResult(this.blockHit);
-        buffer.writeVarInt(this.sequence);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeEnum((java.lang.Enum)this.hand);
+        $$0.writeBlockHitResult(this.blockHit);
+        $$0.writeVarInt(this.sequence);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleUseItemOn(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleUseItemOn(this);
     }
 
     public net.minecraft.world.InteractionHand getHand() {