--- net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
+++ net/minecraft/network/protocol/game/ServerboundSetCreativeModeSlotPacket.java
@@ -9,13 +9,13 @@
     public int slotNum;
     public net.minecraft.world.item.ItemStack itemStack;
 
-    public net.minecraft.network.protocol.game.ServerboundSetCreativeModeSlotPacket(int slotNum, net.minecraft.world.item.ItemStack itemStack) {
-        this.slotNum = slotNum;
-        this.itemStack = itemStack.copy();
+    public net.minecraft.network.protocol.game.ServerboundSetCreativeModeSlotPacket(int $$0, net.minecraft.world.item.ItemStack $$1) {
+        this.slotNum = $$0;
+        this.itemStack = $$1.copy();
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleSetCreativeModeSlot(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleSetCreativeModeSlot(this);
     }
 
     public net.minecraft.network.protocol.game.ServerboundSetCreativeModeSlotPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -23,9 +23,9 @@
         this.itemStack = $$0.readItem();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeShort(this.slotNum);
-        buffer.writeItem(this.itemStack);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeShort(this.slotNum);
+        $$0.writeItem(this.itemStack);
     }
 
     public int getSlotNum() {