--- net/minecraft/network/protocol/game/ServerboundSetCommandMinecartPacket.java
+++ net/minecraft/network/protocol/game/ServerboundSetCommandMinecartPacket.java
@@ -10,10 +10,10 @@
     public java.lang.String command;
     public boolean trackOutput;
 
-    public net.minecraft.network.protocol.game.ServerboundSetCommandMinecartPacket(int entity, java.lang.String command, boolean trackOutput) {
-        this.entity = entity;
-        this.command = command;
-        this.trackOutput = trackOutput;
+    public net.minecraft.network.protocol.game.ServerboundSetCommandMinecartPacket(int $$0, java.lang.String $$1, boolean $$2) {
+        this.entity = $$0;
+        this.command = $$1;
+        this.trackOutput = $$2;
     }
 
     public net.minecraft.network.protocol.game.ServerboundSetCommandMinecartPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -22,21 +22,21 @@
         this.trackOutput = $$0.readBoolean();
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.entity);
-        buffer.writeUtf(this.command);
-        buffer.writeBoolean(this.trackOutput);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.entity);
+        $$0.writeUtf(this.command);
+        $$0.writeBoolean(this.trackOutput);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleSetCommandMinecart(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleSetCommandMinecart(this);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.BaseCommandBlock getCommandBlock(net.minecraft.world.level.Level level) {
-        net.minecraft.world.entity.Entity entity = level.getEntity(this.entity);
-        if (entity instanceof net.minecraft.world.entity.vehicle.MinecartCommandBlock) {
-            return ((net.minecraft.world.entity.vehicle.MinecartCommandBlock)entity).getCommandBlock();
+    public net.minecraft.world.level.BaseCommandBlock getCommandBlock(net.minecraft.world.level.Level $$0) {
+        net.minecraft.world.entity.Entity $$1 = $$0.getEntity(this.entity);
+        if ($$1 instanceof net.minecraft.world.entity.vehicle.MinecartCommandBlock) {
+            return ((net.minecraft.world.entity.vehicle.MinecartCommandBlock)$$1).getCommandBlock();
         }
         return null;
     }