--- net/minecraft/network/protocol/game/ServerboundCommandSuggestionPacket.java
+++ net/minecraft/network/protocol/game/ServerboundCommandSuggestionPacket.java
@@ -9,9 +9,9 @@
     public int id;
     public java.lang.String command;
 
-    public net.minecraft.network.protocol.game.ServerboundCommandSuggestionPacket(int id, java.lang.String command) {
-        this.id = id;
-        this.command = command;
+    public net.minecraft.network.protocol.game.ServerboundCommandSuggestionPacket(int $$0, java.lang.String $$1) {
+        this.id = $$0;
+        this.command = $$1;
     }
 
     public net.minecraft.network.protocol.game.ServerboundCommandSuggestionPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -19,13 +19,13 @@
         this.command = $$0.readUtf(32500);
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.id);
-        buffer.writeUtf(this.command, 32500);
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.id);
+        $$0.writeUtf(this.command, 32500);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleCustomCommandSuggestions(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleCustomCommandSuggestions(this);
     }
 
     public int getId() {