--- net/minecraft/network/protocol/game/ServerboundSetCommandBlockPacket.java
+++ net/minecraft/network/protocol/game/ServerboundSetCommandBlockPacket.java
@@ -16,13 +16,13 @@
     public boolean automatic;
     public net.minecraft.world.level.block.entity.CommandBlockEntity$Mode mode;
 
-    public net.minecraft.network.protocol.game.ServerboundSetCommandBlockPacket(net.minecraft.core.BlockPos pos, java.lang.String command, net.minecraft.world.level.block.entity.CommandBlockEntity$Mode mode, boolean trackOutput, boolean conditional, boolean automatic) {
-        this.pos = pos;
-        this.command = command;
-        this.trackOutput = trackOutput;
-        this.conditional = conditional;
-        this.automatic = automatic;
-        this.mode = mode;
+    public net.minecraft.network.protocol.game.ServerboundSetCommandBlockPacket(net.minecraft.core.BlockPos $$0, java.lang.String $$1, net.minecraft.world.level.block.entity.CommandBlockEntity$Mode $$2, boolean $$3, boolean $$4, boolean $$5) {
+        this.pos = $$0;
+        this.command = $$1;
+        this.trackOutput = $$3;
+        this.conditional = $$4;
+        this.automatic = $$5;
+        this.mode = $$2;
     }
 
     public net.minecraft.network.protocol.game.ServerboundSetCommandBlockPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -35,25 +35,25 @@
         this.automatic = ($$1 & 4) != 0;
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeBlockPos(this.pos);
-        buffer.writeUtf(this.command);
-        buffer.writeEnum((java.lang.Enum)this.mode);
-        int n = 0;
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeBlockPos(this.pos);
+        $$0.writeUtf(this.command);
+        $$0.writeEnum((java.lang.Enum)this.mode);
+        int $$1 = 0;
         if (this.trackOutput) {
-            n |= 1;
+            $$1 |= 1;
         }
         if (this.conditional) {
-            n |= 2;
+            $$1 |= 2;
         }
         if (this.automatic) {
-            n |= 4;
+            $$1 |= 4;
         }
-        buffer.writeByte(n);
+        $$0.writeByte($$1);
     }
 
-    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener handler) {
-        handler.handleSetCommandBlock(this);
+    public void handle(net.minecraft.network.protocol.game.ServerGamePacketListener $$0) {
+        $$0.handleSetCommandBlock(this);
     }
 
     public net.minecraft.core.BlockPos getPos() {