--- net/minecraft/client/gui/screens/recipebook/OverlayRecipeComponent.java
+++ net/minecraft/client/gui/screens/recipebook/OverlayRecipeComponent.java
@@ -74,7 +74,7 @@
         this.lastRecipeClicked = null;
     }
 
-    public boolean changeFocus(boolean focus) {
+    public boolean changeFocus(boolean $$0) {
         return false;
     }
 
@@ -87,45 +87,45 @@
         return this.lastRecipeClicked;
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (button != 0) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if ($$2 != 0) {
             return false;
         }
-        for (net.minecraft.client.gui.screens.recipebook.OverlayRecipeComponent$OverlayRecipeButton overlayRecipeButton : this.recipeButtons) {
-            if (!overlayRecipeButton.mouseClicked(mouseX, mouseY, button)) continue;
-            this.lastRecipeClicked = overlayRecipeButton.recipe;
+        for (net.minecraft.client.gui.screens.recipebook.OverlayRecipeComponent$OverlayRecipeButton $$3 : this.recipeButtons) {
+            if (!$$3.mouseClicked($$0, $$1, $$2)) continue;
+            this.lastRecipeClicked = $$3.recipe;
             return true;
         }
         return false;
     }
 
-    public boolean isMouseOver(double mouseX, double mouseY) {
+    public boolean isMouseOver(double $$0, double $$1) {
         return false;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         if (!this.isVisible) {
             return;
         }
-        this.time += partialTicks;
+        this.time += $$3;
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)RECIPE_BOOK_LOCATION);
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.0, 170.0);
-        int n = this.recipeButtons.size() <= 16 ? 4 : 5;
-        int n2 = java.lang.Math.min((int)this.recipeButtons.size(), (int)n);
-        int n3 = net.minecraft.util.Mth.ceil((float)((float)this.recipeButtons.size() / (float)n));
-        int n4 = 24;
-        int n5 = 4;
-        int n6 = 82;
-        int n7 = 208;
-        this.nineInchSprite(matrixStack, n2, n3, 24, 4, 82, 208);
+        $$0.pushPose();
+        $$0.translate(0.0, 0.0, 170.0);
+        int $$4 = this.recipeButtons.size() <= 16 ? 4 : 5;
+        int $$5 = java.lang.Math.min((int)this.recipeButtons.size(), (int)$$4);
+        int $$6 = net.minecraft.util.Mth.ceil((float)((float)this.recipeButtons.size() / (float)$$4));
+        int $$7 = 24;
+        int $$8 = 4;
+        int $$9 = 82;
+        int $$10 = 208;
+        this.nineInchSprite($$0, $$5, $$6, 24, 4, 82, 208);
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();
-        for (net.minecraft.client.gui.screens.recipebook.OverlayRecipeComponent$OverlayRecipeButton overlayRecipeButton : this.recipeButtons) {
-            overlayRecipeButton.render(matrixStack, mouseX, mouseY, partialTicks);
+        for (net.minecraft.client.gui.screens.recipebook.OverlayRecipeComponent$OverlayRecipeButton $$11 : this.recipeButtons) {
+            $$11.render($$0, $$1, $$2, $$3);
         }
-        matrixStack.popPose();
+        $$0.popPose();
     }
 
     public void nineInchSprite(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6) {