--- net/minecraft/client/gui/screens/recipebook/RecipeButton.java
+++ net/minecraft/client/gui/screens/recipebook/RecipeButton.java
@@ -44,44 +44,44 @@
         this.y = $$1;
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         if (!net.minecraft.client.gui.screens.Screen.hasControlDown()) {
-            this.time += partialTicks;
+            this.time += $$3;
         }
-        net.minecraft.client.Minecraft minecraft = net.minecraft.client.Minecraft.getInstance();
+        net.minecraft.client.Minecraft $$4 = net.minecraft.client.Minecraft.getInstance();
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)RECIPE_BOOK_LOCATION);
-        int n = 29;
+        int $$5 = 29;
         if (!this.collection.hasCraftable()) {
-            n += 25;
+            $$5 += 25;
         }
-        int n2 = 206;
+        int $$6 = 206;
         if (this.collection.getRecipes(this.book.isFiltering(this.menu)).size() > 1) {
-            n2 += 25;
+            $$6 += 25;
         }
-        boolean bl = this.animationTime > 0.0f;
-        com.mojang.blaze3d.vertex.PoseStack poseStack = com.mojang.blaze3d.systems.RenderSystem.getModelViewStack();
-        if (bl) {
-            float f = 1.0f + 0.1f * (float)java.lang.Math.sin((double)(this.animationTime / 15.0f * (float)java.lang.Math.PI));
-            poseStack.pushPose();
-            poseStack.translate((double)(this.x + 8), (double)(this.y + 12), 0.0);
-            poseStack.scale(f, f, 1.0f);
-            poseStack.translate((double)(-(this.x + 8)), (double)(-(this.y + 12)), 0.0);
+        boolean $$7 = this.animationTime > 0.0f;
+        com.mojang.blaze3d.vertex.PoseStack $$8 = com.mojang.blaze3d.systems.RenderSystem.getModelViewStack();
+        if ($$7) {
+            float $$9 = 1.0f + 0.1f * (float)java.lang.Math.sin((double)(this.animationTime / 15.0f * (float)java.lang.Math.PI));
+            $$8.pushPose();
+            $$8.translate((double)(this.x + 8), (double)(this.y + 12), 0.0);
+            $$8.scale($$9, $$9, 1.0f);
+            $$8.translate((double)(-(this.x + 8)), (double)(-(this.y + 12)), 0.0);
             com.mojang.blaze3d.systems.RenderSystem.applyModelViewMatrix();
-            this.animationTime -= partialTicks;
+            this.animationTime -= $$3;
         }
-        this.blit(matrixStack, this.x, this.y, n, n2, this.width, this.height);
-        java.util.List list = this.getOrderedRecipes();
-        this.currentIndex = net.minecraft.util.Mth.floor((float)(this.time / 30.0f)) % list.size();
-        net.minecraft.world.item.ItemStack itemStack = ((net.minecraft.world.item.crafting.Recipe)list.get(this.currentIndex)).getResultItem();
-        int n3 = 4;
+        this.blit($$0, this.x, this.y, $$5, $$6, this.width, this.height);
+        java.util.List $$10 = this.getOrderedRecipes();
+        this.currentIndex = net.minecraft.util.Mth.floor((float)(this.time / 30.0f)) % $$10.size();
+        net.minecraft.world.item.ItemStack $$11 = ((net.minecraft.world.item.crafting.Recipe)$$10.get(this.currentIndex)).getResultItem();
+        int $$12 = 4;
         if (this.collection.hasSingleResultItem() && this.getOrderedRecipes().size() > 1) {
-            minecraft.getItemRenderer().renderAndDecorateItem(itemStack, this.x + n3 + 1, this.y + n3 + 1, 0, 10);
-            --n3;
+            $$4.getItemRenderer().renderAndDecorateItem($$11, this.x + $$12 + 1, this.y + $$12 + 1, 0, 10);
+            --$$12;
         }
-        minecraft.getItemRenderer().renderAndDecorateFakeItem(itemStack, this.x + n3, this.y + n3);
-        if (bl) {
-            poseStack.popPose();
+        $$4.getItemRenderer().renderAndDecorateFakeItem($$11, this.x + $$12, this.y + $$12);
+        if ($$7) {
+            $$8.popPose();
             com.mojang.blaze3d.systems.RenderSystem.applyModelViewMatrix();
         }
     }
@@ -126,8 +126,8 @@
         return 25;
     }
 
-    public boolean isValidClickButton(int button) {
-        return button == 0 || button == 1;
+    public boolean isValidClickButton(int $$0) {
+        return $$0 == 0 || $$0 == 1;
     }
 
     static {