--- net/minecraft/world/level/levelgen/blockpredicates/ReplaceablePredicate.java
+++ net/minecraft/world/level/levelgen/blockpredicates/ReplaceablePredicate.java
@@ -13,7 +13,7 @@
     }
 
     public boolean test(net.minecraft.world.level.block.state.BlockState $$0) {
-        return $$0.getMaterial().isReplaceable();
+        return $$0.canBeReplaced();
     }
 
     public net.minecraft.world.level.levelgen.blockpredicates.BlockPredicateType<?> type() {