--- net/minecraft/world/level/dimension/end/DragonRespawnAnimation$3.java
+++ net/minecraft/world/level/dimension/end/DragonRespawnAnimation$3.java
@@ -10,31 +10,31 @@
         super(string, n);
     }
 
-    public void tick(net.minecraft.server.level.ServerLevel level, net.minecraft.world.level.dimension.end.EndDragonFight manager, java.util.List<net.minecraft.world.entity.boss.enderdragon.EndCrystal> crystals, int ticks, net.minecraft.core.BlockPos pos) {
-        boolean bl;
-        int n = 40;
-        boolean bl2 = ticks % 40 == 0;
-        boolean bl3 = bl = ticks % 40 == 39;
-        if (bl2 || bl) {
-            int n2 = ticks / 40;
-            java.util.List list = net.minecraft.world.level.levelgen.feature.SpikeFeature.getSpikesForLevel((net.minecraft.world.level.WorldGenLevel)level);
-            if (n2 < list.size()) {
-                net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike endSpike = (net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike)list.get(n2);
-                if (bl2) {
-                    for (net.minecraft.world.entity.boss.enderdragon.EndCrystal endCrystal : crystals) {
-                        endCrystal.setBeamTarget(new net.minecraft.core.BlockPos(endSpike.getCenterX(), endSpike.getHeight() + 1, endSpike.getCenterZ()));
+    public void tick(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.dimension.end.EndDragonFight $$1, java.util.List<net.minecraft.world.entity.boss.enderdragon.EndCrystal> $$2, int $$3, net.minecraft.core.BlockPos $$4) {
+        boolean $$7;
+        int $$5 = 40;
+        boolean $$6 = $$3 % 40 == 0;
+        boolean bl = $$7 = $$3 % 40 == 39;
+        if ($$6 || $$7) {
+            int $$9 = $$3 / 40;
+            java.util.List $$8 = net.minecraft.world.level.levelgen.feature.SpikeFeature.getSpikesForLevel((net.minecraft.world.level.WorldGenLevel)$$0);
+            if ($$9 < $$8.size()) {
+                net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike $$10 = (net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike)$$8.get($$9);
+                if ($$6) {
+                    for (net.minecraft.world.entity.boss.enderdragon.EndCrystal $$11 : $$2) {
+                        $$11.setBeamTarget(new net.minecraft.core.BlockPos($$10.getCenterX(), $$10.getHeight() + 1, $$10.getCenterZ()));
                     }
                 } else {
-                    int n3 = 10;
-                    for (net.minecraft.core.BlockPos blockPos : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)new net.minecraft.core.BlockPos(endSpike.getCenterX() - 10, endSpike.getHeight() - 10, endSpike.getCenterZ() - 10), (net.minecraft.core.BlockPos)new net.minecraft.core.BlockPos(endSpike.getCenterX() + 10, endSpike.getHeight() + 10, endSpike.getCenterZ() + 10))) {
-                        level.removeBlock(blockPos, false);
+                    int $$12 = 10;
+                    for (net.minecraft.core.BlockPos $$13 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)new net.minecraft.core.BlockPos($$10.getCenterX() - 10, $$10.getHeight() - 10, $$10.getCenterZ() - 10), (net.minecraft.core.BlockPos)new net.minecraft.core.BlockPos($$10.getCenterX() + 10, $$10.getHeight() + 10, $$10.getCenterZ() + 10))) {
+                        $$0.removeBlock($$13, false);
                     }
-                    level.explode(null, (double)((float)endSpike.getCenterX() + 0.5f), (double)endSpike.getHeight(), (double)((float)endSpike.getCenterZ() + 0.5f), 5.0f, net.minecraft.world.level.Explosion$BlockInteraction.DESTROY);
-                    java.util.Iterator iterator = new net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration(true, (java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)endSpike), new net.minecraft.core.BlockPos(0, 128, 0));
-                    net.minecraft.world.level.levelgen.feature.Feature.END_SPIKE.place((net.minecraft.world.level.levelgen.feature.configurations.FeatureConfiguration)iterator, (net.minecraft.world.level.WorldGenLevel)level, level.getChunkSource().getGenerator(), net.minecraft.util.RandomSource.create(), new net.minecraft.core.BlockPos(endSpike.getCenterX(), 45, endSpike.getCenterZ()));
+                    $$0.explode(null, (double)((float)$$10.getCenterX() + 0.5f), (double)$$10.getHeight(), (double)((float)$$10.getCenterZ() + 0.5f), 5.0f, net.minecraft.world.level.Explosion$BlockInteraction.DESTROY);
+                    net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration $$14 = new net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration(true, (java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)$$10), new net.minecraft.core.BlockPos(0, 128, 0));
+                    net.minecraft.world.level.levelgen.feature.Feature.END_SPIKE.place((net.minecraft.world.level.levelgen.feature.configurations.FeatureConfiguration)$$14, (net.minecraft.world.level.WorldGenLevel)$$0, $$0.getChunkSource().getGenerator(), net.minecraft.util.RandomSource.create(), new net.minecraft.core.BlockPos($$10.getCenterX(), 45, $$10.getCenterZ()));
                 }
-            } else if (bl2) {
-                manager.setRespawnStage(SUMMONING_DRAGON);
+            } else if ($$6) {
+                $$1.setRespawnStage(SUMMONING_DRAGON);
             }
         }
     }