--- net/minecraft/world/level/dimension/end/DragonRespawnAnimation$5.java
+++ net/minecraft/world/level/dimension/end/DragonRespawnAnimation$5.java
@@ -10,7 +10,7 @@
         super(string, n);
     }
 
-    public void tick(net.minecraft.server.level.ServerLevel level, net.minecraft.world.level.dimension.end.EndDragonFight manager, java.util.List<net.minecraft.world.entity.boss.enderdragon.EndCrystal> crystals, int ticks, net.minecraft.core.BlockPos pos) {
+    public void tick(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.dimension.end.EndDragonFight $$1, java.util.List<net.minecraft.world.entity.boss.enderdragon.EndCrystal> $$2, int $$3, net.minecraft.core.BlockPos $$4) {
     }
 }
 