--- net/minecraft/world/level/block/state/properties/BellAttachType.java
+++ net/minecraft/world/level/block/state/properties/BellAttachType.java
@@ -22,9 +22,9 @@
         return (net.minecraft.world.level.block.state.properties.BellAttachType)java.lang.Enum.valueOf(net.minecraft.world.level.block.state.properties.BellAttachType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.block.state.properties.BellAttachType(java.lang.String name) {
+    public net.minecraft.world.level.block.state.properties.BellAttachType(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
+        this.name = $$0;
     }
 
     public java.lang.String getSerializedName() {