--- net/minecraft/world/level/block/state/properties/BlockStateProperties.java
+++ net/minecraft/world/level/block/state/properties/BlockStateProperties.java
@@ -109,7 +109,6 @@
     public static int MIN_RESPAWN_ANCHOR_CHARGES = 0;
     public static int MAX_RESPAWN_ANCHOR_CHARGES = 4;
     public static net.minecraft.world.level.block.state.properties.IntegerProperty RESPAWN_ANCHOR_CHARGES;
-    public static int MAX_ROTATIONS_16 = 15;
     public static net.minecraft.world.level.block.state.properties.IntegerProperty ROTATION_16;
     public static net.minecraft.world.level.block.state.properties.EnumProperty<net.minecraft.world.level.block.state.properties.BedPart> BED_PART;
     public static net.minecraft.world.level.block.state.properties.EnumProperty<net.minecraft.world.level.block.state.properties.ChestType> CHEST_TYPE;
@@ -125,6 +124,8 @@
     public static net.minecraft.world.level.block.state.properties.DirectionProperty VERTICAL_DIRECTION;
     public static net.minecraft.world.level.block.state.properties.EnumProperty<net.minecraft.world.level.block.state.properties.DripstoneThickness> DRIPSTONE_THICKNESS;
     public static net.minecraft.world.level.block.state.properties.EnumProperty<net.minecraft.world.level.block.state.properties.SculkSensorPhase> SCULK_SENSOR_PHASE;
+    public static net.minecraft.world.level.block.state.properties.IntegerProperty BOOKS_STORED;
+    public static net.minecraft.world.level.block.state.properties.IntegerProperty LAST_INTERACTION_BOOK_SLOT;
 
     public net.minecraft.world.level.block.state.properties.BlockStateProperties() {
     }
@@ -225,7 +226,7 @@
         STAGE = net.minecraft.world.level.block.state.properties.IntegerProperty.create((java.lang.String)"stage", (int)0, (int)1);
         STABILITY_DISTANCE = net.minecraft.world.level.block.state.properties.IntegerProperty.create((java.lang.String)"distance", (int)0, (int)7);
         RESPAWN_ANCHOR_CHARGES = net.minecraft.world.level.block.state.properties.IntegerProperty.create((java.lang.String)"charges", (int)0, (int)4);
-        ROTATION_16 = net.minecraft.world.level.block.state.properties.IntegerProperty.create((java.lang.String)"rotation", (int)0, (int)15);
+        ROTATION_16 = net.minecraft.world.level.block.state.properties.IntegerProperty.create((java.lang.String)"rotation", (int)0, (int)net.minecraft.world.level.block.state.properties.RotationSegment.getMaxSegmentIndex());
         BED_PART = net.minecraft.world.level.block.state.properties.EnumProperty.create((java.lang.String)"part", net.minecraft.world.level.block.state.properties.BedPart.class);
         CHEST_TYPE = net.minecraft.world.level.block.state.properties.EnumProperty.create((java.lang.String)"type", net.minecraft.world.level.block.state.properties.ChestType.class);
         MODE_COMPARATOR = net.minecraft.world.level.block.state.properties.EnumProperty.create((java.lang.String)"mode", net.minecraft.world.level.block.state.properties.ComparatorMode.class);
@@ -240,6 +241,8 @@
         VERTICAL_DIRECTION = net.minecraft.world.level.block.state.properties.DirectionProperty.create((java.lang.String)"vertical_direction", (net.minecraft.core.Direction[])new net.minecraft.core.Direction[]{net.minecraft.core.Direction.UP, net.minecraft.core.Direction.DOWN});
         DRIPSTONE_THICKNESS = net.minecraft.world.level.block.state.properties.EnumProperty.create((java.lang.String)"thickness", net.minecraft.world.level.block.state.properties.DripstoneThickness.class);
         SCULK_SENSOR_PHASE = net.minecraft.world.level.block.state.properties.EnumProperty.create((java.lang.String)"sculk_sensor_phase", net.minecraft.world.level.block.state.properties.SculkSensorPhase.class);
+        BOOKS_STORED = net.minecraft.world.level.block.state.properties.IntegerProperty.create((java.lang.String)"books_stored", (int)0, (int)6);
+        LAST_INTERACTION_BOOK_SLOT = net.minecraft.world.level.block.state.properties.IntegerProperty.create((java.lang.String)"last_interaction_book_slot", (int)0, (int)6);
     }
 }
 