--- net/minecraft/world/level/block/state/properties/BooleanProperty.java
+++ net/minecraft/world/level/block/state/properties/BooleanProperty.java
@@ -8,8 +8,8 @@
  {
     public com.google.common.collect.ImmutableSet<java.lang.Boolean> values;
 
-    public net.minecraft.world.level.block.state.properties.BooleanProperty(java.lang.String name) {
-        super(name, java.lang.Boolean.class);
+    public net.minecraft.world.level.block.state.properties.BooleanProperty(java.lang.String $$0) {
+        super($$0, java.lang.Boolean.class);
         this.values = com.google.common.collect.ImmutableSet.of((java.lang.Object)true, (java.lang.Object)false);
     }
 
@@ -17,28 +17,28 @@
         return this.values;
     }
 
-    public static net.minecraft.world.level.block.state.properties.BooleanProperty create(java.lang.String name) {
-        return new net.minecraft.world.level.block.state.properties.BooleanProperty(name);
+    public static net.minecraft.world.level.block.state.properties.BooleanProperty create(java.lang.String $$0) {
+        return new net.minecraft.world.level.block.state.properties.BooleanProperty($$0);
     }
 
-    public java.util.Optional<java.lang.Boolean> getValue(java.lang.String value) {
-        if ("true".equals((java.lang.Object)value) || "false".equals((java.lang.Object)value)) {
-            return java.util.Optional.of((java.lang.Object)java.lang.Boolean.valueOf((java.lang.String)value));
+    public java.util.Optional<java.lang.Boolean> getValue(java.lang.String $$0) {
+        if ("true".equals((java.lang.Object)$$0) || "false".equals((java.lang.Object)$$0)) {
+            return java.util.Optional.of((java.lang.Object)java.lang.Boolean.valueOf((java.lang.String)$$0));
         }
         return java.util.Optional.empty();
     }
 
-    public java.lang.String getName(java.lang.Boolean value) {
-        return value.toString();
+    public java.lang.String getName(java.lang.Boolean $$0) {
+        return $$0.toString();
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other instanceof net.minecraft.world.level.block.state.properties.BooleanProperty && super.equals(other)) {
-            net.minecraft.world.level.block.state.properties.BooleanProperty booleanProperty = (net.minecraft.world.level.block.state.properties.BooleanProperty)other;
-            return this.values.equals((java.lang.Object)booleanProperty.values);
+        if ($$0 instanceof net.minecraft.world.level.block.state.properties.BooleanProperty && super.equals($$0)) {
+            net.minecraft.world.level.block.state.properties.BooleanProperty $$1 = (net.minecraft.world.level.block.state.properties.BooleanProperty)$$0;
+            return this.values.equals((java.lang.Object)$$1.values);
         }
         return false;
     }