--- net/minecraft/world/level/block/state/properties/ChestType.java
+++ net/minecraft/world/level/block/state/properties/ChestType.java
@@ -23,10 +23,10 @@
         return (net.minecraft.world.level.block.state.properties.ChestType)java.lang.Enum.valueOf(net.minecraft.world.level.block.state.properties.ChestType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.block.state.properties.ChestType(java.lang.String name, int opposite) {
+    public net.minecraft.world.level.block.state.properties.ChestType(java.lang.String $$0, int $$1) {
         super(string, n);
-        this.name = name;
-        this.opposite = opposite;
+        this.name = $$0;
+        this.opposite = $$1;
     }
 
     public java.lang.String getSerializedName() {