--- net/minecraft/world/level/block/state/properties/DirectionProperty.java
+++ net/minecraft/world/level/block/state/properties/DirectionProperty.java
@@ -6,24 +6,24 @@
 public class net.minecraft.world.level.block.state.properties.DirectionProperty
 extends net.minecraft.world.level.block.state.properties.EnumProperty<net.minecraft.core.Direction>
  {
-    public net.minecraft.world.level.block.state.properties.DirectionProperty(java.lang.String name, java.util.Collection<net.minecraft.core.Direction> values) {
-        super(name, net.minecraft.core.Direction.class, values);
+    public net.minecraft.world.level.block.state.properties.DirectionProperty(java.lang.String $$0, java.util.Collection<net.minecraft.core.Direction> $$1) {
+        super($$0, net.minecraft.core.Direction.class, $$1);
     }
 
     public static net.minecraft.world.level.block.state.properties.DirectionProperty create(java.lang.String $$02) {
         return net.minecraft.world.level.block.state.properties.DirectionProperty.create((java.lang.String)$$02, $$0 -> true);
     }
 
-    public static net.minecraft.world.level.block.state.properties.DirectionProperty create(java.lang.String name, java.util.function.Predicate<net.minecraft.core.Direction> filter) {
-        return net.minecraft.world.level.block.state.properties.DirectionProperty.create((java.lang.String)name, (java.util.Collection)((java.util.Collection)java.util.Arrays.stream((java.lang.Object[])net.minecraft.core.Direction.values()).filter(filter).collect(java.util.stream.Collectors.toList())));
+    public static net.minecraft.world.level.block.state.properties.DirectionProperty create(java.lang.String $$0, java.util.function.Predicate<net.minecraft.core.Direction> $$1) {
+        return net.minecraft.world.level.block.state.properties.DirectionProperty.create((java.lang.String)$$0, (java.util.Collection)((java.util.Collection)java.util.Arrays.stream((java.lang.Object[])net.minecraft.core.Direction.values()).filter($$1).collect(java.util.stream.Collectors.toList())));
     }
 
-    public static net.minecraft.world.level.block.state.properties.DirectionProperty create(java.lang.String name, net.minecraft.core.Direction ... values) {
-        return net.minecraft.world.level.block.state.properties.DirectionProperty.create((java.lang.String)name, (java.util.Collection)com.google.common.collect.Lists.newArrayList((java.lang.Object[])values));
+    public static net.minecraft.world.level.block.state.properties.DirectionProperty create(java.lang.String $$0, net.minecraft.core.Direction ... $$1) {
+        return net.minecraft.world.level.block.state.properties.DirectionProperty.create((java.lang.String)$$0, (java.util.Collection)com.google.common.collect.Lists.newArrayList((java.lang.Object[])$$1));
     }
 
-    public static net.minecraft.world.level.block.state.properties.DirectionProperty create(java.lang.String name, java.util.Collection<net.minecraft.core.Direction> values) {
-        return new net.minecraft.world.level.block.state.properties.DirectionProperty(name, values);
+    public static net.minecraft.world.level.block.state.properties.DirectionProperty create(java.lang.String $$0, java.util.Collection<net.minecraft.core.Direction> $$1) {
+        return new net.minecraft.world.level.block.state.properties.DirectionProperty($$0, $$1);
     }
 }
 