--- net/minecraft/world/level/block/state/properties/Half.java
+++ net/minecraft/world/level/block/state/properties/Half.java
@@ -20,9 +20,9 @@
         return (net.minecraft.world.level.block.state.properties.Half)java.lang.Enum.valueOf(net.minecraft.world.level.block.state.properties.Half.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.block.state.properties.Half(java.lang.String name) {
+    public net.minecraft.world.level.block.state.properties.Half(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
+        this.name = $$0;
     }
 
     public java.lang.String toString() {