--- net/minecraft/world/level/block/state/properties/NoteBlockInstrument.java
+++ net/minecraft/world/level/block/state/properties/NoteBlockInstrument.java
@@ -35,10 +35,10 @@
         return (net.minecraft.world.level.block.state.properties.NoteBlockInstrument)java.lang.Enum.valueOf(net.minecraft.world.level.block.state.properties.NoteBlockInstrument.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.block.state.properties.NoteBlockInstrument(java.lang.String name, net.minecraft.sounds.SoundEvent soundEvent) {
+    public net.minecraft.world.level.block.state.properties.NoteBlockInstrument(java.lang.String $$0, net.minecraft.sounds.SoundEvent $$1) {
         super(string, n);
-        this.name = name;
-        this.soundEvent = soundEvent;
+        this.name = $$0;
+        this.soundEvent = $$1;
     }
 
     public java.lang.String getSerializedName() {
@@ -49,51 +49,51 @@
         return this.soundEvent;
     }
 
-    public static net.minecraft.world.level.block.state.properties.NoteBlockInstrument byState(net.minecraft.world.level.block.state.BlockState state) {
-        if (state.is(net.minecraft.world.level.block.Blocks.CLAY)) {
+    public static net.minecraft.world.level.block.state.properties.NoteBlockInstrument byState(net.minecraft.world.level.block.state.BlockState $$0) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.CLAY)) {
             return FLUTE;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.GOLD_BLOCK)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.GOLD_BLOCK)) {
             return BELL;
         }
-        if (state.is(net.minecraft.tags.BlockTags.WOOL)) {
+        if ($$0.is(net.minecraft.tags.BlockTags.WOOL)) {
             return GUITAR;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.PACKED_ICE)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.PACKED_ICE)) {
             return CHIME;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.BONE_BLOCK)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.BONE_BLOCK)) {
             return XYLOPHONE;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.IRON_BLOCK)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.IRON_BLOCK)) {
             return IRON_XYLOPHONE;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.SOUL_SAND)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.SOUL_SAND)) {
             return COW_BELL;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.PUMPKIN)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.PUMPKIN)) {
             return DIDGERIDOO;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.EMERALD_BLOCK)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.EMERALD_BLOCK)) {
             return BIT;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.HAY_BLOCK)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.HAY_BLOCK)) {
             return BANJO;
         }
-        if (state.is(net.minecraft.world.level.block.Blocks.GLOWSTONE)) {
+        if ($$0.is(net.minecraft.world.level.block.Blocks.GLOWSTONE)) {
             return PLING;
         }
-        net.minecraft.world.level.material.Material material = state.getMaterial();
-        if (material == net.minecraft.world.level.material.Material.STONE) {
+        net.minecraft.world.level.material.Material $$1 = $$0.getMaterial();
+        if ($$1 == net.minecraft.world.level.material.Material.STONE) {
             return BASEDRUM;
         }
-        if (material == net.minecraft.world.level.material.Material.SAND) {
+        if ($$1 == net.minecraft.world.level.material.Material.SAND) {
             return SNARE;
         }
-        if (material == net.minecraft.world.level.material.Material.GLASS) {
+        if ($$1 == net.minecraft.world.level.material.Material.GLASS) {
             return HAT;
         }
-        if (material == net.minecraft.world.level.material.Material.WOOD || material == net.minecraft.world.level.material.Material.NETHER_WOOD) {
+        if ($$1 == net.minecraft.world.level.material.Material.WOOD || $$1 == net.minecraft.world.level.material.Material.NETHER_WOOD) {
             return BASS;
         }
         return HARP;