--- net/minecraft/world/level/block/state/properties/Property$Value.java
+++ net/minecraft/world/level/block/state/properties/Property$Value.java
@@ -9,12 +9,12 @@
     public net.minecraft.world.level.block.state.properties.Property<T> property;
     public T value;
 
-    public net.minecraft.world.level.block.state.properties.Property$Value(net.minecraft.world.level.block.state.properties.Property<T> property, T value) {
-        if (!property.getPossibleValues().contains(value)) {
-            throw new java.lang.IllegalArgumentException("Value " + value + " does not belong to property " + property);
+    public net.minecraft.world.level.block.state.properties.Property$Value(net.minecraft.world.level.block.state.properties.Property<T> $$0, T $$1) {
+        if (!$$0.getPossibleValues().contains($$1)) {
+            throw new java.lang.IllegalArgumentException("Value " + $$1 + " does not belong to property " + $$0);
         }
-        this.property = property;
-        this.value = value;
+        this.property = $$0;
+        this.value = $$1;
     }
 
     public java.lang.String toString() {
@@ -25,8 +25,8 @@
         return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.block.state.properties.Property$Value.class, "property;value", "property", "value"}, this);
     }
 
-    public boolean equals(java.lang.Object other) {
-        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.block.state.properties.Property$Value.class, "property;value", "property", "value"}, this, other);
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.block.state.properties.Property$Value.class, "property;value", "property", "value"}, this, $$0);
     }
 
     public net.minecraft.world.level.block.state.properties.Property<T> property() {