--- net/minecraft/world/level/block/state/properties/RotationSegment.java
+++ net/minecraft/world/level/block/state/properties/RotationSegment.java
@@ -1,0 +1,44 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level.block.state.properties;
+
+public class net.minecraft.world.level.block.state.properties.RotationSegment
+ {
+    public static int MAX_SEGMENT_INDEX = 15;
+    public static int NORTH_0 = 0;
+    public static int EAST_90 = 4;
+    public static int SOUTH_180 = 8;
+    public static int WEST_270 = 12;
+
+    public net.minecraft.world.level.block.state.properties.RotationSegment() {
+    }
+
+    public static int getMaxSegmentIndex() {
+        return 15;
+    }
+
+    public static int convertToSegment(net.minecraft.core.Direction $$0) {
+        return $$0.getAxis().isVertical() ? 0 : $$0.getOpposite().get2DDataValue() * 4;
+    }
+
+    public static int convertToSegment(float $$0) {
+        return net.minecraft.util.Mth.floor((double)((double)((180.0f + $$0) * 16.0f / 360.0f) + 0.5)) & 0xF;
+    }
+
+    public static java.util.Optional<net.minecraft.core.Direction> convertToDirection(int $$0) {
+        net.minecraft.core.Direction $$1 = switch ($$0) {
+            case 0 -> net.minecraft.core.Direction.NORTH;
+            case 4 -> net.minecraft.core.Direction.EAST;
+            case 8 -> net.minecraft.core.Direction.SOUTH;
+            case 12 -> net.minecraft.core.Direction.WEST;
+            default -> null;
+        };
+        return java.util.Optional.ofNullable((java.lang.Object)$$1);
+    }
+
+    public static float convertToDegrees(int $$0) {
+        return (float)$$0 * 22.5f;
+    }
+}
+