--- net/minecraft/world/level/block/state/properties/StairsShape.java
+++ net/minecraft/world/level/block/state/properties/StairsShape.java
@@ -23,9 +23,9 @@
         return (net.minecraft.world.level.block.state.properties.StairsShape)java.lang.Enum.valueOf(net.minecraft.world.level.block.state.properties.StairsShape.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.block.state.properties.StairsShape(java.lang.String name) {
+    public net.minecraft.world.level.block.state.properties.StairsShape(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
+        this.name = $$0;
     }
 
     public java.lang.String toString() {