--- net/minecraft/world/level/block/state/properties/StructureMode.java
+++ net/minecraft/world/level/block/state/properties/StructureMode.java
@@ -23,10 +23,10 @@
         return (net.minecraft.world.level.block.state.properties.StructureMode)java.lang.Enum.valueOf(net.minecraft.world.level.block.state.properties.StructureMode.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.block.state.properties.StructureMode(java.lang.String name) {
+    public net.minecraft.world.level.block.state.properties.StructureMode(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
-        this.displayName = net.minecraft.network.chat.Component.translatable((java.lang.String)("structure_block.mode_info." + name));
+        this.name = $$0;
+        this.displayName = net.minecraft.network.chat.Component.translatable((java.lang.String)("structure_block.mode_info." + $$0));
     }
 
     public java.lang.String getSerializedName() {