--- net/minecraft/world/level/block/state/properties/WoodType.java
+++ net/minecraft/world/level/block/state/properties/WoodType.java
@@ -15,15 +15,16 @@
     public static net.minecraft.world.level.block.state.properties.WoodType CRIMSON;
     public static net.minecraft.world.level.block.state.properties.WoodType WARPED;
     public static net.minecraft.world.level.block.state.properties.WoodType MANGROVE;
+    public static net.minecraft.world.level.block.state.properties.WoodType BAMBOO;
     public java.lang.String name;
 
-    public net.minecraft.world.level.block.state.properties.WoodType(java.lang.String name) {
-        this.name = name;
+    public net.minecraft.world.level.block.state.properties.WoodType(java.lang.String $$0) {
+        this.name = $$0;
     }
 
-    public static net.minecraft.world.level.block.state.properties.WoodType register(net.minecraft.world.level.block.state.properties.WoodType woodType) {
-        VALUES.add((java.lang.Object)woodType);
-        return woodType;
+    public static net.minecraft.world.level.block.state.properties.WoodType register(net.minecraft.world.level.block.state.properties.WoodType $$0) {
+        VALUES.add((java.lang.Object)$$0);
+        return $$0;
     }
 
     public static java.util.stream.Stream<net.minecraft.world.level.block.state.properties.WoodType> values() {
@@ -45,6 +46,7 @@
         CRIMSON = net.minecraft.world.level.block.state.properties.WoodType.register((net.minecraft.world.level.block.state.properties.WoodType)new net.minecraft.world.level.block.state.properties.WoodType("crimson"));
         WARPED = net.minecraft.world.level.block.state.properties.WoodType.register((net.minecraft.world.level.block.state.properties.WoodType)new net.minecraft.world.level.block.state.properties.WoodType("warped"));
         MANGROVE = net.minecraft.world.level.block.state.properties.WoodType.register((net.minecraft.world.level.block.state.properties.WoodType)new net.minecraft.world.level.block.state.properties.WoodType("mangrove"));
+        BAMBOO = net.minecraft.world.level.block.state.properties.WoodType.register((net.minecraft.world.level.block.state.properties.WoodType)new net.minecraft.world.level.block.state.properties.WoodType("bamboo"));
     }
 }
 