--- net/minecraft/world/phys/shapes/CubeVoxelShape.java
+++ net/minecraft/world/phys/shapes/CubeVoxelShape.java
@@ -10,13 +10,13 @@
         super($$0);
     }
 
-    public it.unimi.dsi.fastutil.doubles.DoubleList getCoords(net.minecraft.core.Direction$Axis axis) {
-        return new net.minecraft.world.phys.shapes.CubePointRange(this.shape.getSize(axis));
+    public it.unimi.dsi.fastutil.doubles.DoubleList getCoords(net.minecraft.core.Direction$Axis $$0) {
+        return new net.minecraft.world.phys.shapes.CubePointRange(this.shape.getSize($$0));
     }
 
-    public int findIndex(net.minecraft.core.Direction$Axis axis, double position) {
-        int n = this.shape.getSize(axis);
-        return net.minecraft.util.Mth.floor((double)net.minecraft.util.Mth.clamp((double)(position * (double)n), (double)-1.0, (double)n));
+    public int findIndex(net.minecraft.core.Direction$Axis $$0, double $$1) {
+        int $$2 = this.shape.getSize($$0);
+        return net.minecraft.util.Mth.floor((double)net.minecraft.util.Mth.clamp((double)($$1 * (double)$$2), (double)-1.0, (double)$$2));
     }
 }
 