--- net/minecraft/world/phys/shapes/CubePointRange.java
+++ net/minecraft/world/phys/shapes/CubePointRange.java
@@ -8,15 +8,15 @@
  {
     public int parts;
 
-    public net.minecraft.world.phys.shapes.CubePointRange(int parts) {
-        if (parts <= 0) {
+    public net.minecraft.world.phys.shapes.CubePointRange(int $$0) {
+        if ($$0 <= 0) {
             throw new java.lang.IllegalArgumentException("Need at least 1 part");
         }
-        this.parts = parts;
+        this.parts = $$0;
     }
 
-    public double getDouble(int value) {
-        return (double)value / (double)this.parts;
+    public double getDouble(int $$0) {
+        return (double)$$0 / (double)this.parts;
     }
 
     public int size() {