--- net/minecraft/world/phys/shapes/EntityCollisionContext.java
+++ net/minecraft/world/phys/shapes/EntityCollisionContext.java
@@ -23,12 +23,12 @@
     }
 
     @java.lang.Deprecated
-    public net.minecraft.world.phys.shapes.EntityCollisionContext(net.minecraft.world.entity.Entity entity) {
-        this(entity.isDescending(), entity.getY(), entity instanceof net.minecraft.world.entity.LivingEntity ? ((net.minecraft.world.entity.LivingEntity)entity).getMainHandItem() : net.minecraft.world.item.ItemStack.EMPTY, entity instanceof net.minecraft.world.entity.LivingEntity ? arg_0 -> ((net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)entity)).canStandOnFluid(arg_0) : $$0 -> false, entity);
+    public net.minecraft.world.phys.shapes.EntityCollisionContext(net.minecraft.world.entity.Entity $$02) {
+        this($$02.isDescending(), $$02.getY(), $$02 instanceof net.minecraft.world.entity.LivingEntity ? ((net.minecraft.world.entity.LivingEntity)$$02).getMainHandItem() : net.minecraft.world.item.ItemStack.EMPTY, $$02 instanceof net.minecraft.world.entity.LivingEntity ? arg_0 -> ((net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)$$02)).canStandOnFluid(arg_0) : $$0 -> false, $$02);
     }
 
-    public boolean isHoldingItem(net.minecraft.world.item.Item item) {
-        return this.heldItem.is(item);
+    public boolean isHoldingItem(net.minecraft.world.item.Item $$0) {
+        return this.heldItem.is($$0);
     }
 
     public boolean canStandOnFluid(net.minecraft.world.level.material.FluidState $$0, net.minecraft.world.level.material.FluidState $$1) {
@@ -39,8 +39,8 @@
         return this.descending;
     }
 
-    public boolean isAbove(net.minecraft.world.phys.shapes.VoxelShape shape, net.minecraft.core.BlockPos pos, boolean canAscend) {
-        return this.entityBottom > (double)pos.getY() + shape.max(net.minecraft.core.Direction$Axis.Y) - (double)1.0E-5f;
+    public boolean isAbove(net.minecraft.world.phys.shapes.VoxelShape $$0, net.minecraft.core.BlockPos $$1, boolean $$2) {
+        return this.entityBottom > (double)$$1.getY() + $$0.max(net.minecraft.core.Direction$Axis.Y) - (double)1.0E-5f;
     }
 
     @javax.annotation.Nullable