--- net/minecraft/world/phys/shapes/OffsetDoubleList.java
+++ net/minecraft/world/phys/shapes/OffsetDoubleList.java
@@ -9,13 +9,13 @@
     public it.unimi.dsi.fastutil.doubles.DoubleList delegate;
     public double offset;
 
-    public net.minecraft.world.phys.shapes.OffsetDoubleList(it.unimi.dsi.fastutil.doubles.DoubleList delegate, double offset) {
-        this.delegate = delegate;
-        this.offset = offset;
+    public net.minecraft.world.phys.shapes.OffsetDoubleList(it.unimi.dsi.fastutil.doubles.DoubleList $$0, double $$1) {
+        this.delegate = $$0;
+        this.offset = $$1;
     }
 
-    public double getDouble(int value) {
-        return this.delegate.getDouble(value) + this.offset;
+    public double getDouble(int $$0) {
+        return this.delegate.getDouble($$0) + this.offset;
     }
 
     public int size() {