--- net/minecraft/world/scores/Objective.java
+++ net/minecraft/world/scores/Objective.java
@@ -12,13 +12,13 @@
     public net.minecraft.network.chat.Component formattedDisplayName;
     public net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType renderType;
 
-    public net.minecraft.world.scores.Objective(net.minecraft.world.scores.Scoreboard scoreboard, java.lang.String name, net.minecraft.world.scores.criteria.ObjectiveCriteria criteria, net.minecraft.network.chat.Component displayName, net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType renderType) {
-        this.scoreboard = scoreboard;
-        this.name = name;
-        this.criteria = criteria;
-        this.displayName = displayName;
+    public net.minecraft.world.scores.Objective(net.minecraft.world.scores.Scoreboard $$0, java.lang.String $$1, net.minecraft.world.scores.criteria.ObjectiveCriteria $$2, net.minecraft.network.chat.Component $$3, net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType $$4) {
+        this.scoreboard = $$0;
+        this.name = $$1;
+        this.criteria = $$2;
+        this.displayName = $$3;
         this.formattedDisplayName = this.createFormattedDisplayName();
-        this.renderType = renderType;
+        this.renderType = $$4;
     }
 
     public net.minecraft.world.scores.Scoreboard getScoreboard() {
@@ -38,15 +38,15 @@
     }
 
     public net.minecraft.network.chat.Component createFormattedDisplayName() {
-        return net.minecraft.network.chat.ComponentUtils.wrapInSquareBrackets((net.minecraft.network.chat.Component)this.displayName.copy().withStyle(style -> style.withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)this.name)))));
+        return net.minecraft.network.chat.ComponentUtils.wrapInSquareBrackets((net.minecraft.network.chat.Component)this.displayName.copy().withStyle($$0 -> $$0.withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)this.name)))));
     }
 
     public net.minecraft.network.chat.Component getFormattedDisplayName() {
         return this.formattedDisplayName;
     }
 
-    public void setDisplayName(net.minecraft.network.chat.Component displayName) {
-        this.displayName = displayName;
+    public void setDisplayName(net.minecraft.network.chat.Component $$0) {
+        this.displayName = $$0;
         this.formattedDisplayName = this.createFormattedDisplayName();
         this.scoreboard.onObjectiveChanged(this);
     }
@@ -55,8 +55,8 @@
         return this.renderType;
     }
 
-    public void setRenderType(net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType renderType) {
-        this.renderType = renderType;
+    public void setRenderType(net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType $$0) {
+        this.renderType = $$0;
         this.scoreboard.onObjectiveChanged(this);
     }
 }