--- net/minecraft/world/scores/PlayerTeam.java
+++ net/minecraft/world/scores/PlayerTeam.java
@@ -22,7 +22,7 @@
     public net.minecraft.world.scores.Team$CollisionRule collisionRule;
     public net.minecraft.network.chat.Style displayNameStyle;
 
-    public net.minecraft.world.scores.PlayerTeam(net.minecraft.world.scores.Scoreboard scoreboard, java.lang.String name) {
+    public net.minecraft.world.scores.PlayerTeam(net.minecraft.world.scores.Scoreboard $$0, java.lang.String $$1) {
         this.players = com.google.common.collect.Sets.newHashSet();
         this.playerPrefix = net.minecraft.network.chat.CommonComponents.EMPTY;
         this.playerSuffix = net.minecraft.network.chat.CommonComponents.EMPTY;
@@ -32,10 +32,10 @@
         this.deathMessageVisibility = net.minecraft.world.scores.Team$Visibility.ALWAYS;
         this.color = net.minecraft.ChatFormatting.RESET;
         this.collisionRule = net.minecraft.world.scores.Team$CollisionRule.ALWAYS;
-        this.scoreboard = scoreboard;
-        this.name = name;
-        this.displayName = net.minecraft.network.chat.Component.literal((java.lang.String)name);
-        this.displayNameStyle = net.minecraft.network.chat.Style.EMPTY.withInsertion(name).withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)name)));
+        this.scoreboard = $$0;
+        this.name = $$1;
+        this.displayName = net.minecraft.network.chat.Component.literal((java.lang.String)$$1);
+        this.displayNameStyle = net.minecraft.network.chat.Style.EMPTY.withInsertion($$1).withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)$$1)));
     }
 
     public net.minecraft.world.scores.Scoreboard getScoreboard() {
@@ -59,16 +59,16 @@
         return $$0;
     }
 
-    public void setDisplayName(net.minecraft.network.chat.Component name) {
-        if (name == null) {
+    public void setDisplayName(net.minecraft.network.chat.Component $$0) {
+        if ($$0 == null) {
             throw new java.lang.IllegalArgumentException("Name cannot be null");
         }
-        this.displayName = name;
+        this.displayName = $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
-    public void setPlayerPrefix(@javax.annotation.Nullable net.minecraft.network.chat.Component playerPrefix) {
-        this.playerPrefix = playerPrefix == null ? net.minecraft.network.chat.CommonComponents.EMPTY : playerPrefix;
+    public void setPlayerPrefix(@javax.annotation.Nullable net.minecraft.network.chat.Component $$0) {
+        this.playerPrefix = $$0 == null ? net.minecraft.network.chat.CommonComponents.EMPTY : $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
@@ -76,8 +76,8 @@
         return this.playerPrefix;
     }
 
-    public void setPlayerSuffix(@javax.annotation.Nullable net.minecraft.network.chat.Component playerSuffix) {
-        this.playerSuffix = playerSuffix == null ? net.minecraft.network.chat.CommonComponents.EMPTY : playerSuffix;
+    public void setPlayerSuffix(@javax.annotation.Nullable net.minecraft.network.chat.Component $$0) {
+        this.playerSuffix = $$0 == null ? net.minecraft.network.chat.CommonComponents.EMPTY : $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
@@ -89,28 +89,28 @@
         return this.players;
     }
 
-    public net.minecraft.network.chat.MutableComponent getFormattedName(net.minecraft.network.chat.Component formattedName) {
-        net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component.empty().append(this.playerPrefix).append(formattedName).append(this.playerSuffix);
-        net.minecraft.ChatFormatting chatFormatting = this.getColor();
-        if (chatFormatting != net.minecraft.ChatFormatting.RESET) {
-            mutableComponent.withStyle(chatFormatting);
+    public net.minecraft.network.chat.MutableComponent getFormattedName(net.minecraft.network.chat.Component $$0) {
+        net.minecraft.network.chat.MutableComponent $$1 = net.minecraft.network.chat.Component.empty().append(this.playerPrefix).append($$0).append(this.playerSuffix);
+        net.minecraft.ChatFormatting $$2 = this.getColor();
+        if ($$2 != net.minecraft.ChatFormatting.RESET) {
+            $$1.withStyle($$2);
         }
-        return mutableComponent;
+        return $$1;
     }
 
-    public static net.minecraft.network.chat.MutableComponent formatNameForTeam(@javax.annotation.Nullable net.minecraft.world.scores.Team playerTeam, net.minecraft.network.chat.Component playerName) {
-        if (playerTeam == null) {
-            return playerName.copy();
+    public static net.minecraft.network.chat.MutableComponent formatNameForTeam(@javax.annotation.Nullable net.minecraft.world.scores.Team $$0, net.minecraft.network.chat.Component $$1) {
+        if ($$0 == null) {
+            return $$1.copy();
         }
-        return playerTeam.getFormattedName(playerName);
+        return $$0.getFormattedName($$1);
     }
 
     public boolean isAllowFriendlyFire() {
         return this.allowFriendlyFire;
     }
 
-    public void setAllowFriendlyFire(boolean friendlyFire) {
-        this.allowFriendlyFire = friendlyFire;
+    public void setAllowFriendlyFire(boolean $$0) {
+        this.allowFriendlyFire = $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
@@ -118,8 +118,8 @@
         return this.seeFriendlyInvisibles;
     }
 
-    public void setSeeFriendlyInvisibles(boolean friendlyInvisibles) {
-        this.seeFriendlyInvisibles = friendlyInvisibles;
+    public void setSeeFriendlyInvisibles(boolean $$0) {
+        this.seeFriendlyInvisibles = $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
@@ -131,13 +131,13 @@
         return this.deathMessageVisibility;
     }
 
-    public void setNameTagVisibility(net.minecraft.world.scores.Team$Visibility visibility) {
-        this.nameTagVisibility = visibility;
+    public void setNameTagVisibility(net.minecraft.world.scores.Team$Visibility $$0) {
+        this.nameTagVisibility = $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
-    public void setDeathMessageVisibility(net.minecraft.world.scores.Team$Visibility visibility) {
-        this.deathMessageVisibility = visibility;
+    public void setDeathMessageVisibility(net.minecraft.world.scores.Team$Visibility $$0) {
+        this.deathMessageVisibility = $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
@@ -145,8 +145,8 @@
         return this.collisionRule;
     }
 
-    public void setCollisionRule(net.minecraft.world.scores.Team$CollisionRule rule) {
-        this.collisionRule = rule;
+    public void setCollisionRule(net.minecraft.world.scores.Team$CollisionRule $$0) {
+        this.collisionRule = $$0;
         this.scoreboard.onTeamChanged(this);
     }
 
@@ -161,13 +161,13 @@
         return $$0;
     }
 
-    public void unpackOptions(int flags) {
-        this.setAllowFriendlyFire((flags & 1) > 0);
-        this.setSeeFriendlyInvisibles((flags & 2) > 0);
+    public void unpackOptions(int $$0) {
+        this.setAllowFriendlyFire(($$0 & 1) > 0);
+        this.setSeeFriendlyInvisibles(($$0 & 2) > 0);
     }
 
-    public void setColor(net.minecraft.ChatFormatting color) {
-        this.color = color;
+    public void setColor(net.minecraft.ChatFormatting $$0) {
+        this.color = $$0;
         this.scoreboard.onTeamChanged(this);
     }
 