--- net/minecraft/world/scores/Team$Visibility.java
+++ net/minecraft/world/scores/Team$Visibility.java
@@ -28,14 +28,14 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.scores.Team$Visibility byName(java.lang.String name) {
-        return (net.minecraft.world.scores.Team$Visibility)BY_NAME.get((java.lang.Object)name);
+    public static net.minecraft.world.scores.Team$Visibility byName(java.lang.String $$0) {
+        return (net.minecraft.world.scores.Team$Visibility)BY_NAME.get((java.lang.Object)$$0);
     }
 
-    public net.minecraft.world.scores.Team$Visibility(java.lang.String name, int id) {
+    public net.minecraft.world.scores.Team$Visibility(java.lang.String $$0, int $$1) {
         super(string, n);
-        this.name = name;
-        this.id = id;
+        this.name = $$0;
+        this.id = $$1;
     }
 
     public net.minecraft.network.chat.Component getDisplayName() {