--- net/minecraft/server/level/BlockDestructionProgress.java
+++ net/minecraft/server/level/BlockDestructionProgress.java
@@ -24,19 +24,19 @@
         return this.pos;
     }
 
-    public void setProgress(int damage) {
-        if (damage > 10) {
-            damage = 10;
+    public void setProgress(int $$0) {
+        if ($$0 > 10) {
+            $$0 = 10;
         }
-        this.progress = damage;
+        this.progress = $$0;
     }
 
     public int getProgress() {
         return this.progress;
     }
 
-    public void updateTick(int createdAtCloudUpdateTick) {
-        this.updatedRenderTick = createdAtCloudUpdateTick;
+    public void updateTick(int $$0) {
+        this.updatedRenderTick = $$0;
     }
 
     public int getUpdatedRenderTick() {