--- net/minecraft/server/level/ChunkMap$DistanceManager.java
+++ net/minecraft/server/level/ChunkMap$DistanceManager.java
@@ -18,13 +18,13 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.server.level.ChunkHolder getChunk(long chunkPos) {
-        return this.this$0.getUpdatingChunkIfPresent(chunkPos);
+    public net.minecraft.server.level.ChunkHolder getChunk(long $$0) {
+        return this.this$0.getUpdatingChunkIfPresent($$0);
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.server.level.ChunkHolder updateChunkScheduling(long chunkPos, int newLevel, @javax.annotation.Nullable net.minecraft.server.level.ChunkHolder holder, int oldLevel) {
-        return this.this$0.updateChunkScheduling(chunkPos, newLevel, holder, oldLevel);
+    public net.minecraft.server.level.ChunkHolder updateChunkScheduling(long $$0, int $$1, @javax.annotation.Nullable net.minecraft.server.level.ChunkHolder $$2, int $$3) {
+        return this.this$0.updateChunkScheduling($$0, $$1, $$2, $$3);
     }
 }
 