--- net/minecraft/server/level/ChunkHolder$FullChunkStatus.java
+++ net/minecraft/server/level/ChunkHolder$FullChunkStatus.java
@@ -24,8 +24,8 @@
         super(string, n);
     }
 
-    public boolean isOrAfter(net.minecraft.server.level.ChunkHolder$FullChunkStatus type) {
-        return this.ordinal() >= type.ordinal();
+    public boolean isOrAfter(net.minecraft.server.level.ChunkHolder$FullChunkStatus $$0) {
+        return this.ordinal() >= $$0.ordinal();
     }
 
     public static /* synthetic */ net.minecraft.server.level.ChunkHolder$FullChunkStatus[] $values() {