--- net/minecraft/server/level/DemoMode.java
+++ net/minecraft/server/level/DemoMode.java
@@ -66,12 +66,12 @@
         super.handleBlockBreakAction($$0, $$1, $$2, $$3, $$4);
     }
 
-    public net.minecraft.world.InteractionResult useItem(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.level.Level level, net.minecraft.world.item.ItemStack stack, net.minecraft.world.InteractionHand hand) {
+    public net.minecraft.world.InteractionResult useItem(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.level.Level $$1, net.minecraft.world.item.ItemStack $$2, net.minecraft.world.InteractionHand $$3) {
         if (this.demoHasEnded) {
             this.outputDemoReminder();
             return net.minecraft.world.InteractionResult.PASS;
         }
-        return super.useItem(player, level, stack, hand);
+        return super.useItem($$0, $$1, $$2, $$3);
     }
 
     public net.minecraft.world.InteractionResult useItemOn(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.level.Level $$1, net.minecraft.world.item.ItemStack $$2, net.minecraft.world.InteractionHand $$3, net.minecraft.world.phys.BlockHitResult $$4) {