--- net/minecraft/server/level/PlayerMap.java
+++ net/minecraft/server/level/PlayerMap.java
@@ -15,31 +15,31 @@
         return this.players.keySet();
     }
 
-    public void addPlayer(long chunkPos, net.minecraft.server.level.ServerPlayer player, boolean canGenerateChunks) {
-        this.players.put((java.lang.Object)player, canGenerateChunks);
+    public void addPlayer(long $$0, net.minecraft.server.level.ServerPlayer $$1, boolean $$2) {
+        this.players.put((java.lang.Object)$$1, $$2);
     }
 
-    public void removePlayer(long chunkPos, net.minecraft.server.level.ServerPlayer player) {
-        this.players.removeBoolean((java.lang.Object)player);
+    public void removePlayer(long $$0, net.minecraft.server.level.ServerPlayer $$1) {
+        this.players.removeBoolean((java.lang.Object)$$1);
     }
 
-    public void ignorePlayer(net.minecraft.server.level.ServerPlayer player) {
-        this.players.replace((java.lang.Object)player, true);
+    public void ignorePlayer(net.minecraft.server.level.ServerPlayer $$0) {
+        this.players.replace((java.lang.Object)$$0, true);
     }
 
-    public void unIgnorePlayer(net.minecraft.server.level.ServerPlayer player) {
-        this.players.replace((java.lang.Object)player, false);
+    public void unIgnorePlayer(net.minecraft.server.level.ServerPlayer $$0) {
+        this.players.replace((java.lang.Object)$$0, false);
     }
 
-    public boolean ignoredOrUnknown(net.minecraft.server.level.ServerPlayer player) {
-        return this.players.getOrDefault((java.lang.Object)player, true);
+    public boolean ignoredOrUnknown(net.minecraft.server.level.ServerPlayer $$0) {
+        return this.players.getOrDefault((java.lang.Object)$$0, true);
     }
 
-    public boolean ignored(net.minecraft.server.level.ServerPlayer player) {
-        return this.players.getBoolean((java.lang.Object)player);
+    public boolean ignored(net.minecraft.server.level.ServerPlayer $$0) {
+        return this.players.getBoolean((java.lang.Object)$$0);
     }
 
-    public void updatePlayer(long oldChunkPos, long newChunkPos, net.minecraft.server.level.ServerPlayer player) {
+    public void updatePlayer(long $$0, long $$1, net.minecraft.server.level.ServerPlayer $$2) {
     }
 }
 