--- net/minecraft/server/level/ServerBossEvent.java
+++ net/minecraft/server/level/ServerBossEvent.java
@@ -24,47 +24,47 @@
         }
     }
 
-    public void setColor(net.minecraft.world.BossEvent$BossBarColor color) {
-        if (color != this.color) {
-            super.setColor(color);
+    public void setColor(net.minecraft.world.BossEvent$BossBarColor $$0) {
+        if ($$0 != this.color) {
+            super.setColor($$0);
             this.broadcast(net.minecraft.network.protocol.game.ClientboundBossEventPacket::createUpdateStylePacket);
         }
     }
 
-    public void setOverlay(net.minecraft.world.BossEvent$BossBarOverlay overlay) {
-        if (overlay != this.overlay) {
-            super.setOverlay(overlay);
+    public void setOverlay(net.minecraft.world.BossEvent$BossBarOverlay $$0) {
+        if ($$0 != this.overlay) {
+            super.setOverlay($$0);
             this.broadcast(net.minecraft.network.protocol.game.ClientboundBossEventPacket::createUpdateStylePacket);
         }
     }
 
-    public net.minecraft.world.BossEvent setDarkenScreen(boolean darkenSky) {
-        if (darkenSky != this.darkenScreen) {
-            super.setDarkenScreen(darkenSky);
+    public net.minecraft.world.BossEvent setDarkenScreen(boolean $$0) {
+        if ($$0 != this.darkenScreen) {
+            super.setDarkenScreen($$0);
             this.broadcast(net.minecraft.network.protocol.game.ClientboundBossEventPacket::createUpdatePropertiesPacket);
         }
         return this;
     }
 
-    public net.minecraft.world.BossEvent setPlayBossMusic(boolean playEndBossMusic) {
-        if (playEndBossMusic != this.playBossMusic) {
-            super.setPlayBossMusic(playEndBossMusic);
+    public net.minecraft.world.BossEvent setPlayBossMusic(boolean $$0) {
+        if ($$0 != this.playBossMusic) {
+            super.setPlayBossMusic($$0);
             this.broadcast(net.minecraft.network.protocol.game.ClientboundBossEventPacket::createUpdatePropertiesPacket);
         }
         return this;
     }
 
-    public net.minecraft.world.BossEvent setCreateWorldFog(boolean createFog) {
-        if (createFog != this.createWorldFog) {
-            super.setCreateWorldFog(createFog);
+    public net.minecraft.world.BossEvent setCreateWorldFog(boolean $$0) {
+        if ($$0 != this.createWorldFog) {
+            super.setCreateWorldFog($$0);
             this.broadcast(net.minecraft.network.protocol.game.ClientboundBossEventPacket::createUpdatePropertiesPacket);
         }
         return this;
     }
 
-    public void setName(net.minecraft.network.chat.Component name) {
-        if (!com.google.common.base.Objects.equal((java.lang.Object)name, (java.lang.Object)this.name)) {
-            super.setName(name);
+    public void setName(net.minecraft.network.chat.Component $$0) {
+        if (!com.google.common.base.Objects.equal((java.lang.Object)$$0, (java.lang.Object)this.name)) {
+            super.setName($$0);
             this.broadcast(net.minecraft.network.protocol.game.ClientboundBossEventPacket::createUpdateNamePacket);
         }
     }
@@ -78,15 +78,15 @@
         }
     }
 
-    public void addPlayer(net.minecraft.server.level.ServerPlayer player) {
-        if (this.players.add((java.lang.Object)player) && this.visible) {
-            player.connection.send((net.minecraft.network.protocol.Packet)net.minecraft.network.protocol.game.ClientboundBossEventPacket.createAddPacket((net.minecraft.world.BossEvent)this));
+    public void addPlayer(net.minecraft.server.level.ServerPlayer $$0) {
+        if (this.players.add((java.lang.Object)$$0) && this.visible) {
+            $$0.connection.send((net.minecraft.network.protocol.Packet)net.minecraft.network.protocol.game.ClientboundBossEventPacket.createAddPacket((net.minecraft.world.BossEvent)this));
         }
     }
 
-    public void removePlayer(net.minecraft.server.level.ServerPlayer player) {
-        if (this.players.remove((java.lang.Object)player) && this.visible) {
-            player.connection.send((net.minecraft.network.protocol.Packet)net.minecraft.network.protocol.game.ClientboundBossEventPacket.createRemovePacket((java.util.UUID)this.getId()));
+    public void removePlayer(net.minecraft.server.level.ServerPlayer $$0) {
+        if (this.players.remove((java.lang.Object)$$0) && this.visible) {
+            $$0.connection.send((net.minecraft.network.protocol.Packet)net.minecraft.network.protocol.game.ClientboundBossEventPacket.createRemovePacket((java.util.UUID)this.getId()));
         }
     }
 
@@ -102,11 +102,11 @@
         return this.visible;
     }
 
-    public void setVisible(boolean visible) {
-        if (visible != this.visible) {
-            this.visible = visible;
-            for (net.minecraft.server.level.ServerPlayer serverPlayer : this.players) {
-                serverPlayer.connection.send((net.minecraft.network.protocol.Packet)(visible ? net.minecraft.network.protocol.game.ClientboundBossEventPacket.createAddPacket((net.minecraft.world.BossEvent)this) : net.minecraft.network.protocol.game.ClientboundBossEventPacket.createRemovePacket((java.util.UUID)this.getId())));
+    public void setVisible(boolean $$0) {
+        if ($$0 != this.visible) {
+            this.visible = $$0;
+            for (net.minecraft.server.level.ServerPlayer $$1 : this.players) {
+                $$1.connection.send((net.minecraft.network.protocol.Packet)($$0 ? net.minecraft.network.protocol.game.ClientboundBossEventPacket.createAddPacket((net.minecraft.world.BossEvent)this) : net.minecraft.network.protocol.game.ClientboundBossEventPacket.createRemovePacket((java.util.UUID)this.getId())));
             }
         }
     }