--- net/minecraft/server/level/ServerChunkCache$MainThreadExecutor.java
+++ net/minecraft/server/level/ServerChunkCache$MainThreadExecutor.java
@@ -13,11 +13,11 @@
         super("Chunk source main thread executor for " + $$0.dimension().location());
     }
 
-    public java.lang.Runnable wrapRunnable(java.lang.Runnable runnable) {
-        return runnable;
+    public java.lang.Runnable wrapRunnable(java.lang.Runnable $$0) {
+        return $$0;
     }
 
-    public boolean shouldRun(java.lang.Runnable runnable) {
+    public boolean shouldRun(java.lang.Runnable $$0) {
         return true;
     }
 
@@ -29,9 +29,9 @@
         return this.this$0.mainThread;
     }
 
-    public void doRunTask(java.lang.Runnable task) {
+    public void doRunTask(java.lang.Runnable $$0) {
         this.this$0.level.getProfiler().incrementCounter("runTask");
-        super.doRunTask(task);
+        super.doRunTask($$0);
     }
 
     public boolean pollTask() {