--- net/minecraft/server/level/Ticket.java
+++ net/minecraft/server/level/Ticket.java
@@ -56,13 +56,13 @@
         return this.ticketLevel;
     }
 
-    public void setCreatedTick(long timestamp) {
-        this.createdTick = timestamp;
+    public void setCreatedTick(long $$0) {
+        this.createdTick = $$0;
     }
 
-    public boolean timedOut(long currentTime) {
-        long l = this.type.timeout();
-        return l != 0L && currentTime - this.createdTick > l;
+    public boolean timedOut(long $$0) {
+        long $$1 = this.type.timeout();
+        return $$1 != 0L && $$0 - this.createdTick > $$1;
     }
 
     public /* synthetic */ int compareTo(java.lang.Object object) {