--- net/minecraft/server/level/TicketType.java
+++ net/minecraft/server/level/TicketType.java
@@ -17,12 +17,12 @@
     public static net.minecraft.server.level.TicketType<java.lang.Integer> POST_TELEPORT;
     public static net.minecraft.server.level.TicketType<net.minecraft.world.level.ChunkPos> UNKNOWN;
 
-    public static <T> net.minecraft.server.level.TicketType<T> create(java.lang.String name, java.util.Comparator<T> comparator) {
-        return new net.minecraft.server.level.TicketType(name, comparator, 0L);
+    public static <T> net.minecraft.server.level.TicketType<T> create(java.lang.String $$0, java.util.Comparator<T> $$1) {
+        return new net.minecraft.server.level.TicketType($$0, $$1, 0L);
     }
 
-    public static <T> net.minecraft.server.level.TicketType<T> create(java.lang.String name, java.util.Comparator<T> comparator, int lifespan) {
-        return new net.minecraft.server.level.TicketType(name, comparator, (long)lifespan);
+    public static <T> net.minecraft.server.level.TicketType<T> create(java.lang.String $$0, java.util.Comparator<T> $$1, int $$2) {
+        return new net.minecraft.server.level.TicketType($$0, $$1, (long)$$2);
     }
 
     public net.minecraft.server.level.TicketType(java.lang.String $$0, java.util.Comparator<T> $$1, long $$2) {