--- net/minecraft/nbt/ByteTag.java
+++ net/minecraft/nbt/ByteTag.java
@@ -12,20 +12,20 @@
     public static net.minecraft.nbt.ByteTag ONE;
     public byte data;
 
-    public net.minecraft.nbt.ByteTag(byte data) {
-        this.data = data;
+    public net.minecraft.nbt.ByteTag(byte $$0) {
+        this.data = $$0;
     }
 
-    public static net.minecraft.nbt.ByteTag valueOf(byte data) {
-        return net.minecraft.nbt.ByteTag$Cache.cache[128 + data];
+    public static net.minecraft.nbt.ByteTag valueOf(byte $$0) {
+        return net.minecraft.nbt.ByteTag$Cache.cache[128 + $$0];
     }
 
-    public static net.minecraft.nbt.ByteTag valueOf(boolean data) {
-        return data ? ONE : ZERO;
+    public static net.minecraft.nbt.ByteTag valueOf(boolean $$0) {
+        return $$0 ? ONE : ZERO;
     }
 
-    public void write(java.io.DataOutput output) throws java.io.IOException {
-        output.writeByte((int)this.data);
+    public void write(java.io.DataOutput $$0) throws java.io.IOException {
+        $$0.writeByte((int)this.data);
     }
 
     public byte getId() {
@@ -40,11 +40,11 @@
         return this;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        return other instanceof net.minecraft.nbt.ByteTag && this.data == ((net.minecraft.nbt.ByteTag)other).data;
+        return $$0 instanceof net.minecraft.nbt.ByteTag && this.data == ((net.minecraft.nbt.ByteTag)$$0).data;
     }
 
     public int hashCode() {