--- net/minecraft/nbt/ByteTag$1.java
+++ net/minecraft/nbt/ByteTag$1.java
@@ -9,9 +9,9 @@
     public net.minecraft.nbt.ByteTag$1() {
     }
 
-    public net.minecraft.nbt.ByteTag load(java.io.DataInput input, int depth, net.minecraft.nbt.NbtAccounter accounter) throws java.io.IOException {
-        accounter.accountBits(72L);
-        return net.minecraft.nbt.ByteTag.valueOf((byte)input.readByte());
+    public net.minecraft.nbt.ByteTag load(java.io.DataInput $$0, int $$1, net.minecraft.nbt.NbtAccounter $$2) throws java.io.IOException {
+        $$2.accountBits(72L);
+        return net.minecraft.nbt.ByteTag.valueOf((byte)$$0.readByte());
     }
 
     public net.minecraft.nbt.StreamTagVisitor$ValueResult parse(java.io.DataInput $$0, net.minecraft.nbt.StreamTagVisitor $$1) throws java.io.IOException {