--- net/minecraft/nbt/DoubleTag.java
+++ net/minecraft/nbt/DoubleTag.java
@@ -11,19 +11,19 @@
     public static net.minecraft.nbt.TagType<net.minecraft.nbt.DoubleTag> TYPE;
     public double data;
 
-    public net.minecraft.nbt.DoubleTag(double data) {
-        this.data = data;
+    public net.minecraft.nbt.DoubleTag(double $$0) {
+        this.data = $$0;
     }
 
-    public static net.minecraft.nbt.DoubleTag valueOf(double data) {
-        if (data == 0.0) {
+    public static net.minecraft.nbt.DoubleTag valueOf(double $$0) {
+        if ($$0 == 0.0) {
             return ZERO;
         }
-        return new net.minecraft.nbt.DoubleTag(data);
+        return new net.minecraft.nbt.DoubleTag($$0);
     }
 
-    public void write(java.io.DataOutput output) throws java.io.IOException {
-        output.writeDouble(this.data);
+    public void write(java.io.DataOutput $$0) throws java.io.IOException {
+        $$0.writeDouble(this.data);
     }
 
     public byte getId() {
@@ -38,11 +38,11 @@
         return this;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        return other instanceof net.minecraft.nbt.DoubleTag && this.data == ((net.minecraft.nbt.DoubleTag)other).data;
+        return $$0 instanceof net.minecraft.nbt.DoubleTag && this.data == ((net.minecraft.nbt.DoubleTag)$$0).data;
     }
 
     public int hashCode() {