--- net/minecraft/nbt/FloatTag.java
+++ net/minecraft/nbt/FloatTag.java
@@ -11,19 +11,19 @@
     public static net.minecraft.nbt.TagType<net.minecraft.nbt.FloatTag> TYPE;
     public float data;
 
-    public net.minecraft.nbt.FloatTag(float data) {
-        this.data = data;
+    public net.minecraft.nbt.FloatTag(float $$0) {
+        this.data = $$0;
     }
 
-    public static net.minecraft.nbt.FloatTag valueOf(float data) {
-        if (data == 0.0f) {
+    public static net.minecraft.nbt.FloatTag valueOf(float $$0) {
+        if ($$0 == 0.0f) {
             return ZERO;
         }
-        return new net.minecraft.nbt.FloatTag(data);
+        return new net.minecraft.nbt.FloatTag($$0);
     }
 
-    public void write(java.io.DataOutput output) throws java.io.IOException {
-        output.writeFloat(this.data);
+    public void write(java.io.DataOutput $$0) throws java.io.IOException {
+        $$0.writeFloat(this.data);
     }
 
     public byte getId() {
@@ -38,11 +38,11 @@
         return this;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        return other instanceof net.minecraft.nbt.FloatTag && this.data == ((net.minecraft.nbt.FloatTag)other).data;
+        return $$0 instanceof net.minecraft.nbt.FloatTag && this.data == ((net.minecraft.nbt.FloatTag)$$0).data;
     }
 
     public int hashCode() {