--- net/minecraft/nbt/ListTag.java
+++ net/minecraft/nbt/ListTag.java
@@ -11,21 +11,21 @@
     public java.util.List<net.minecraft.nbt.Tag> list;
     public byte type;
 
-    public net.minecraft.nbt.ListTag(java.util.List<net.minecraft.nbt.Tag> list, byte type) {
-        this.list = list;
-        this.type = type;
+    public net.minecraft.nbt.ListTag(java.util.List<net.minecraft.nbt.Tag> $$0, byte $$1) {
+        this.list = $$0;
+        this.type = $$1;
     }
 
     public net.minecraft.nbt.ListTag() {
         this((java.util.List)com.google.common.collect.Lists.newArrayList(), 0);
     }
 
-    public void write(java.io.DataOutput output) throws java.io.IOException {
+    public void write(java.io.DataOutput $$0) throws java.io.IOException {
         this.type = this.list.isEmpty() ? (byte)0 : ((net.minecraft.nbt.Tag)this.list.get(0)).getId();
-        output.writeByte((int)this.type);
-        output.writeInt(this.list.size());
-        for (net.minecraft.nbt.Tag tag : this.list) {
-            tag.write(output);
+        $$0.writeByte((int)this.type);
+        $$0.writeInt(this.list.size());
+        for (net.minecraft.nbt.Tag $$1 : this.list) {
+            $$1.write($$0);
         }
     }
 
@@ -57,42 +57,42 @@
         return this.list.isEmpty();
     }
 
-    public net.minecraft.nbt.CompoundTag getCompound(int index) {
-        net.minecraft.nbt.Tag tag;
-        if (index >= 0 && index < this.list.size() && (tag = (net.minecraft.nbt.Tag)this.list.get(index)).getId() == 10) {
-            return (net.minecraft.nbt.CompoundTag)tag;
+    public net.minecraft.nbt.CompoundTag getCompound(int $$0) {
+        net.minecraft.nbt.Tag $$1;
+        if ($$0 >= 0 && $$0 < this.list.size() && ($$1 = (net.minecraft.nbt.Tag)this.list.get($$0)).getId() == 10) {
+            return (net.minecraft.nbt.CompoundTag)$$1;
         }
         return new net.minecraft.nbt.CompoundTag();
     }
 
-    public net.minecraft.nbt.ListTag getList(int index) {
-        net.minecraft.nbt.Tag tag;
-        if (index >= 0 && index < this.list.size() && (tag = (net.minecraft.nbt.Tag)this.list.get(index)).getId() == 9) {
-            return (net.minecraft.nbt.ListTag)tag;
+    public net.minecraft.nbt.ListTag getList(int $$0) {
+        net.minecraft.nbt.Tag $$1;
+        if ($$0 >= 0 && $$0 < this.list.size() && ($$1 = (net.minecraft.nbt.Tag)this.list.get($$0)).getId() == 9) {
+            return (net.minecraft.nbt.ListTag)$$1;
         }
         return new net.minecraft.nbt.ListTag();
     }
 
-    public short getShort(int index) {
-        net.minecraft.nbt.Tag tag;
-        if (index >= 0 && index < this.list.size() && (tag = (net.minecraft.nbt.Tag)this.list.get(index)).getId() == 2) {
-            return ((net.minecraft.nbt.ShortTag)tag).getAsShort();
+    public short getShort(int $$0) {
+        net.minecraft.nbt.Tag $$1;
+        if ($$0 >= 0 && $$0 < this.list.size() && ($$1 = (net.minecraft.nbt.Tag)this.list.get($$0)).getId() == 2) {
+            return ((net.minecraft.nbt.ShortTag)$$1).getAsShort();
         }
         return 0;
     }
 
-    public int getInt(int index) {
-        net.minecraft.nbt.Tag tag;
-        if (index >= 0 && index < this.list.size() && (tag = (net.minecraft.nbt.Tag)this.list.get(index)).getId() == 3) {
-            return ((net.minecraft.nbt.IntTag)tag).getAsInt();
+    public int getInt(int $$0) {
+        net.minecraft.nbt.Tag $$1;
+        if ($$0 >= 0 && $$0 < this.list.size() && ($$1 = (net.minecraft.nbt.Tag)this.list.get($$0)).getId() == 3) {
+            return ((net.minecraft.nbt.IntTag)$$1).getAsInt();
         }
         return 0;
     }
 
-    public int[] getIntArray(int index) {
-        net.minecraft.nbt.Tag tag;
-        if (index >= 0 && index < this.list.size() && (tag = (net.minecraft.nbt.Tag)this.list.get(index)).getId() == 11) {
-            return ((net.minecraft.nbt.IntArrayTag)tag).getAsIntArray();
+    public int[] getIntArray(int $$0) {
+        net.minecraft.nbt.Tag $$1;
+        if ($$0 >= 0 && $$0 < this.list.size() && ($$1 = (net.minecraft.nbt.Tag)this.list.get($$0)).getId() == 11) {
+            return ((net.minecraft.nbt.IntArrayTag)$$1).getAsIntArray();
         }
         return new int[0];
     }
@@ -105,31 +105,31 @@
         return new long[0];
     }
 
-    public double getDouble(int index) {
-        net.minecraft.nbt.Tag tag;
-        if (index >= 0 && index < this.list.size() && (tag = (net.minecraft.nbt.Tag)this.list.get(index)).getId() == 6) {
-            return ((net.minecraft.nbt.DoubleTag)tag).getAsDouble();
+    public double getDouble(int $$0) {
+        net.minecraft.nbt.Tag $$1;
+        if ($$0 >= 0 && $$0 < this.list.size() && ($$1 = (net.minecraft.nbt.Tag)this.list.get($$0)).getId() == 6) {
+            return ((net.minecraft.nbt.DoubleTag)$$1).getAsDouble();
         }
         return 0.0;
     }
 
-    public float getFloat(int index) {
-        net.minecraft.nbt.Tag tag;
-        if (index >= 0 && index < this.list.size() && (tag = (net.minecraft.nbt.Tag)this.list.get(index)).getId() == 5) {
-            return ((net.minecraft.nbt.FloatTag)tag).getAsFloat();
+    public float getFloat(int $$0) {
+        net.minecraft.nbt.Tag $$1;
+        if ($$0 >= 0 && $$0 < this.list.size() && ($$1 = (net.minecraft.nbt.Tag)this.list.get($$0)).getId() == 5) {
+            return ((net.minecraft.nbt.FloatTag)$$1).getAsFloat();
         }
         return 0.0f;
     }
 
-    public java.lang.String getString(int index) {
-        if (index < 0 || index >= this.list.size()) {
+    public java.lang.String getString(int $$0) {
+        if ($$0 < 0 || $$0 >= this.list.size()) {
             return "";
         }
-        net.minecraft.nbt.Tag tag = (net.minecraft.nbt.Tag)this.list.get(index);
-        if (tag.getId() == 8) {
-            return tag.getAsString();
+        net.minecraft.nbt.Tag $$1 = (net.minecraft.nbt.Tag)this.list.get($$0);
+        if ($$1.getId() == 8) {
+            return $$1.getAsString();
         }
-        return tag.toString();
+        return $$1.toString();
     }
 
     public int size() {
@@ -154,31 +154,31 @@
         }
     }
 
-    public boolean setTag(int index, net.minecraft.nbt.Tag nbt) {
-        if (this.updateType(nbt)) {
-            this.list.set(index, (java.lang.Object)nbt);
+    public boolean setTag(int $$0, net.minecraft.nbt.Tag $$1) {
+        if (this.updateType($$1)) {
+            this.list.set($$0, (java.lang.Object)$$1);
             return true;
         }
         return false;
     }
 
-    public boolean addTag(int index, net.minecraft.nbt.Tag nbt) {
-        if (this.updateType(nbt)) {
-            this.list.add(index, (java.lang.Object)nbt);
+    public boolean addTag(int $$0, net.minecraft.nbt.Tag $$1) {
+        if (this.updateType($$1)) {
+            this.list.add($$0, (java.lang.Object)$$1);
             return true;
         }
         return false;
     }
 
-    public boolean updateType(net.minecraft.nbt.Tag tag) {
-        if (tag.getId() == 0) {
+    public boolean updateType(net.minecraft.nbt.Tag $$0) {
+        if ($$0.getId() == 0) {
             return false;
         }
         if (this.type == 0) {
-            this.type = tag.getId();
+            this.type = $$0.getId();
             return true;
         }
-        return this.type == tag.getId();
+        return this.type == $$0.getId();
     }
 
     public net.minecraft.nbt.ListTag copy() {
@@ -187,11 +187,11 @@
         return new net.minecraft.nbt.ListTag((java.util.List)$$1, this.type);
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        return other instanceof net.minecraft.nbt.ListTag && java.util.Objects.equals((java.lang.Object)this.list, (java.lang.Object)((net.minecraft.nbt.ListTag)other).list);
+        return $$0 instanceof net.minecraft.nbt.ListTag && java.util.Objects.equals((java.lang.Object)this.list, (java.lang.Object)((net.minecraft.nbt.ListTag)$$0).list);
     }
 
     public int hashCode() {