--- net/minecraft/nbt/NbtAccounter.java
+++ net/minecraft/nbt/NbtAccounter.java
@@ -9,12 +9,12 @@
     public long quota;
     public long usage;
 
-    public net.minecraft.nbt.NbtAccounter(long quota) {
-        this.quota = quota;
+    public net.minecraft.nbt.NbtAccounter(long $$0) {
+        this.quota = $$0;
     }
 
-    public void accountBits(long bits) {
-        this.usage += bits / 8L;
+    public void accountBits(long $$0) {
+        this.usage += $$0 / 8L;
         if (this.usage > this.quota) {
             throw new java.lang.RuntimeException("Tried to read NBT tag that was too big; tried to allocate: " + this.usage + "bytes where max allowed: " + this.quota);
         }