--- net/minecraft/nbt/NbtIo.java
+++ net/minecraft/nbt/NbtIo.java
@@ -8,10 +8,10 @@
     public net.minecraft.nbt.NbtIo() {
     }
 
-    public static net.minecraft.nbt.CompoundTag readCompressed(java.io.File file) throws java.io.IOException {
-        try (java.io.FileInputStream fileInputStream = new java.io.FileInputStream(file);
+    public static net.minecraft.nbt.CompoundTag readCompressed(java.io.File $$0) throws java.io.IOException {
+        try (java.io.FileInputStream $$1 = new java.io.FileInputStream($$0);
 ){
-            net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.nbt.NbtIo.readCompressed((java.io.InputStream)fileInputStream);
+            net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.nbt.NbtIo.readCompressed((java.io.InputStream)$$1);
             return compoundTag;
         }
     }
@@ -20,10 +20,10 @@
         return new java.io.DataInputStream((java.io.InputStream)new net.minecraft.util.FastBufferedInputStream((java.io.InputStream)new java.util.zip.GZIPInputStream($$0)));
     }
 
-    public static net.minecraft.nbt.CompoundTag readCompressed(java.io.InputStream zippedStream) throws java.io.IOException {
-        try (java.io.DataInputStream dataInputStream = net.minecraft.nbt.NbtIo.createDecompressorStream((java.io.InputStream)zippedStream);
+    public static net.minecraft.nbt.CompoundTag readCompressed(java.io.InputStream $$0) throws java.io.IOException {
+        try (java.io.DataInputStream $$1 = net.minecraft.nbt.NbtIo.createDecompressorStream((java.io.InputStream)$$0);
 ){
-            net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.nbt.NbtIo.read((java.io.DataInput)dataInputStream, (net.minecraft.nbt.NbtAccounter)net.minecraft.nbt.NbtAccounter.UNLIMITED);
+            net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.nbt.NbtIo.read((java.io.DataInput)$$1, (net.minecraft.nbt.NbtAccounter)net.minecraft.nbt.NbtAccounter.UNLIMITED);
             return compoundTag;
         }
     }
@@ -42,58 +42,58 @@
         }
     }
 
-    public static void writeCompressed(net.minecraft.nbt.CompoundTag compoundTag, java.io.File file) throws java.io.IOException {
-        try (java.io.FileOutputStream fileOutputStream = new java.io.FileOutputStream(file);
+    public static void writeCompressed(net.minecraft.nbt.CompoundTag $$0, java.io.File $$1) throws java.io.IOException {
+        try (java.io.FileOutputStream $$2 = new java.io.FileOutputStream($$1);
 ){
-            net.minecraft.nbt.NbtIo.writeCompressed((net.minecraft.nbt.CompoundTag)compoundTag, (java.io.OutputStream)fileOutputStream);
+            net.minecraft.nbt.NbtIo.writeCompressed((net.minecraft.nbt.CompoundTag)$$0, (java.io.OutputStream)$$2);
         }
     }
 
-    public static void writeCompressed(net.minecraft.nbt.CompoundTag compoundTag, java.io.OutputStream outputStream) throws java.io.IOException {
-        try (java.io.DataOutputStream dataOutputStream = new java.io.DataOutputStream((java.io.OutputStream)new java.io.BufferedOutputStream((java.io.OutputStream)new java.util.zip.GZIPOutputStream(outputStream)));
+    public static void writeCompressed(net.minecraft.nbt.CompoundTag $$0, java.io.OutputStream $$1) throws java.io.IOException {
+        try (java.io.DataOutputStream $$2 = new java.io.DataOutputStream((java.io.OutputStream)new java.io.BufferedOutputStream((java.io.OutputStream)new java.util.zip.GZIPOutputStream($$1)));
 ){
-            net.minecraft.nbt.NbtIo.write((net.minecraft.nbt.CompoundTag)compoundTag, (java.io.DataOutput)dataOutputStream);
+            net.minecraft.nbt.NbtIo.write((net.minecraft.nbt.CompoundTag)$$0, (java.io.DataOutput)$$2);
         }
     }
 
-    public static void write(net.minecraft.nbt.CompoundTag compoundTag, java.io.File file) throws java.io.IOException {
-        try (java.io.FileOutputStream fileOutputStream = new java.io.FileOutputStream(file);
-             java.io.DataOutputStream dataOutputStream = new java.io.DataOutputStream((java.io.OutputStream)fileOutputStream);
+    public static void write(net.minecraft.nbt.CompoundTag $$0, java.io.File $$1) throws java.io.IOException {
+        try (java.io.FileOutputStream $$2 = new java.io.FileOutputStream($$1);
+             java.io.DataOutputStream $$3 = new java.io.DataOutputStream((java.io.OutputStream)$$2);
 ){
-            net.minecraft.nbt.NbtIo.write((net.minecraft.nbt.CompoundTag)compoundTag, (java.io.DataOutput)dataOutputStream);
+            net.minecraft.nbt.NbtIo.write((net.minecraft.nbt.CompoundTag)$$0, (java.io.DataOutput)$$3);
         }
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.nbt.CompoundTag read(java.io.File file) throws java.io.IOException {
-        if (!file.exists()) {
+    public static net.minecraft.nbt.CompoundTag read(java.io.File $$0) throws java.io.IOException {
+        if (!$$0.exists()) {
             return null;
         }
-        try (java.io.FileInputStream fileInputStream = new java.io.FileInputStream(file);
+        try (java.io.FileInputStream $$1 = new java.io.FileInputStream($$0);
 ){
             net.minecraft.nbt.CompoundTag compoundTag;
-            try (java.io.DataInputStream dataInputStream = new java.io.DataInputStream((java.io.InputStream)fileInputStream);
+            try (java.io.DataInputStream $$2 = new java.io.DataInputStream((java.io.InputStream)$$1);
 ){
-                compoundTag = net.minecraft.nbt.NbtIo.read((java.io.DataInput)dataInputStream, (net.minecraft.nbt.NbtAccounter)net.minecraft.nbt.NbtAccounter.UNLIMITED);
+                compoundTag = net.minecraft.nbt.NbtIo.read((java.io.DataInput)$$2, (net.minecraft.nbt.NbtAccounter)net.minecraft.nbt.NbtAccounter.UNLIMITED);
             }
             return compoundTag;
         }
     }
 
-    public static net.minecraft.nbt.CompoundTag read(java.io.DataInput input) throws java.io.IOException {
-        return net.minecraft.nbt.NbtIo.read((java.io.DataInput)input, (net.minecraft.nbt.NbtAccounter)net.minecraft.nbt.NbtAccounter.UNLIMITED);
+    public static net.minecraft.nbt.CompoundTag read(java.io.DataInput $$0) throws java.io.IOException {
+        return net.minecraft.nbt.NbtIo.read((java.io.DataInput)$$0, (net.minecraft.nbt.NbtAccounter)net.minecraft.nbt.NbtAccounter.UNLIMITED);
     }
 
-    public static net.minecraft.nbt.CompoundTag read(java.io.DataInput input, net.minecraft.nbt.NbtAccounter accounter) throws java.io.IOException {
-        net.minecraft.nbt.Tag tag = net.minecraft.nbt.NbtIo.readUnnamedTag((java.io.DataInput)input, (int)0, (net.minecraft.nbt.NbtAccounter)accounter);
-        if (tag instanceof net.minecraft.nbt.CompoundTag) {
-            return (net.minecraft.nbt.CompoundTag)tag;
+    public static net.minecraft.nbt.CompoundTag read(java.io.DataInput $$0, net.minecraft.nbt.NbtAccounter $$1) throws java.io.IOException {
+        net.minecraft.nbt.Tag $$2 = net.minecraft.nbt.NbtIo.readUnnamedTag((java.io.DataInput)$$0, (int)0, (net.minecraft.nbt.NbtAccounter)$$1);
+        if ($$2 instanceof net.minecraft.nbt.CompoundTag) {
+            return (net.minecraft.nbt.CompoundTag)$$2;
         }
         throw new java.io.IOException("Root tag must be a named compound tag");
     }
 
-    public static void write(net.minecraft.nbt.CompoundTag compoundTag, java.io.DataOutput output) throws java.io.IOException {
-        net.minecraft.nbt.NbtIo.writeUnnamedTag((net.minecraft.nbt.Tag)compoundTag, (java.io.DataOutput)output);
+    public static void write(net.minecraft.nbt.CompoundTag $$0, java.io.DataOutput $$1) throws java.io.IOException {
+        net.minecraft.nbt.NbtIo.writeUnnamedTag((net.minecraft.nbt.Tag)$$0, (java.io.DataOutput)$$1);
     }
 
     public static void parse(java.io.DataInput $$0, net.minecraft.nbt.StreamTagVisitor $$1) throws java.io.IOException {
@@ -120,29 +120,29 @@
         }
     }
 
-    public static void writeUnnamedTag(net.minecraft.nbt.Tag tag, java.io.DataOutput output) throws java.io.IOException {
-        output.writeByte((int)tag.getId());
-        if (tag.getId() == 0) {
+    public static void writeUnnamedTag(net.minecraft.nbt.Tag $$0, java.io.DataOutput $$1) throws java.io.IOException {
+        $$1.writeByte((int)$$0.getId());
+        if ($$0.getId() == 0) {
             return;
         }
-        output.writeUTF("");
-        tag.write(output);
+        $$1.writeUTF("");
+        $$0.write($$1);
     }
 
-    public static net.minecraft.nbt.Tag readUnnamedTag(java.io.DataInput input, int depth, net.minecraft.nbt.NbtAccounter accounter) throws java.io.IOException {
-        byte by = input.readByte();
-        if (by == 0) {
+    public static net.minecraft.nbt.Tag readUnnamedTag(java.io.DataInput $$0, int $$1, net.minecraft.nbt.NbtAccounter $$2) throws java.io.IOException {
+        byte $$3 = $$0.readByte();
+        if ($$3 == 0) {
             return net.minecraft.nbt.EndTag.INSTANCE;
         }
-        net.minecraft.nbt.StringTag.skipString((java.io.DataInput)input);
+        net.minecraft.nbt.StringTag.skipString((java.io.DataInput)$$0);
         try {
-            return net.minecraft.nbt.TagTypes.getType((int)by).load(input, depth, accounter);
+            return net.minecraft.nbt.TagTypes.getType((int)$$3).load($$0, $$1, $$2);
         }
-        catch (java.io.IOException iOException) {
-            net.minecraft.CrashReport crashReport = net.minecraft.CrashReport.forThrowable((java.lang.Throwable)iOException, (java.lang.String)"Loading NBT data");
-            net.minecraft.CrashReportCategory crashReportCategory = crashReport.addCategory("NBT Tag");
-            crashReportCategory.setDetail("Tag type", (java.lang.Object)by);
-            throw new net.minecraft.ReportedException(crashReport);
+        catch (java.io.IOException $$4) {
+            net.minecraft.CrashReport $$5 = net.minecraft.CrashReport.forThrowable((java.lang.Throwable)$$4, (java.lang.String)"Loading NBT data");
+            net.minecraft.CrashReportCategory $$6 = $$5.addCategory("NBT Tag");
+            $$6.setDetail("Tag type", (java.lang.Object)$$3);
+            throw new net.minecraft.ReportedException($$5);
         }
     }
 }