--- net/minecraft/nbt/NbtOps$LongListCollector.java
+++ net/minecraft/nbt/NbtOps$LongListCollector.java
@@ -1,0 +1,34 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.nbt;
+
+public class net.minecraft.nbt.NbtOps$LongListCollector
+implements net.minecraft.nbt.NbtOps$ListCollector
+ {
+    public it.unimi.dsi.fastutil.longs.LongArrayList values;
+
+    public net.minecraft.nbt.NbtOps$LongListCollector(long $$0) {
+        this.values = new it.unimi.dsi.fastutil.longs.LongArrayList();
+        this.values.add($$0);
+    }
+
+    public net.minecraft.nbt.NbtOps$LongListCollector(long[] $$0) {
+        this.values = new it.unimi.dsi.fastutil.longs.LongArrayList();
+        this.values.addElements(0, $$0);
+    }
+
+    public net.minecraft.nbt.NbtOps$ListCollector accept(net.minecraft.nbt.Tag $$0) {
+        if ($$0 instanceof net.minecraft.nbt.LongTag) {
+            net.minecraft.nbt.LongTag $$1 = (net.minecraft.nbt.LongTag)$$0;
+            this.values.add($$1.getAsLong());
+            return this;
+        }
+        return new net.minecraft.nbt.NbtOps$HeterogenousListCollector(this.values).accept($$0);
+    }
+
+    public net.minecraft.nbt.Tag result() {
+        return new net.minecraft.nbt.LongArrayTag(this.values.toLongArray());
+    }
+}
+