--- net/minecraft/nbt/NbtOps$InitialListCollector.java
+++ net/minecraft/nbt/NbtOps$InitialListCollector.java
@@ -1,0 +1,42 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.nbt;
+
+public class net.minecraft.nbt.NbtOps$InitialListCollector
+implements net.minecraft.nbt.NbtOps$ListCollector
+ {
+    public static net.minecraft.nbt.NbtOps$InitialListCollector INSTANCE;
+
+    public net.minecraft.nbt.NbtOps$InitialListCollector() {
+    }
+
+    public net.minecraft.nbt.NbtOps$ListCollector accept(net.minecraft.nbt.Tag $$0) {
+        if ($$0 instanceof net.minecraft.nbt.CompoundTag) {
+            net.minecraft.nbt.CompoundTag $$1 = (net.minecraft.nbt.CompoundTag)$$0;
+            return new net.minecraft.nbt.NbtOps$HeterogenousListCollector().accept((net.minecraft.nbt.Tag)$$1);
+        }
+        if ($$0 instanceof net.minecraft.nbt.ByteTag) {
+            net.minecraft.nbt.ByteTag $$2 = (net.minecraft.nbt.ByteTag)$$0;
+            return new net.minecraft.nbt.NbtOps$ByteListCollector($$2.getAsByte());
+        }
+        if ($$0 instanceof net.minecraft.nbt.IntTag) {
+            net.minecraft.nbt.IntTag $$3 = (net.minecraft.nbt.IntTag)$$0;
+            return new net.minecraft.nbt.NbtOps$IntListCollector($$3.getAsInt());
+        }
+        if ($$0 instanceof net.minecraft.nbt.LongTag) {
+            net.minecraft.nbt.LongTag $$4 = (net.minecraft.nbt.LongTag)$$0;
+            return new net.minecraft.nbt.NbtOps$LongListCollector($$4.getAsLong());
+        }
+        return new net.minecraft.nbt.NbtOps$HomogenousListCollector($$0);
+    }
+
+    public net.minecraft.nbt.Tag result() {
+        return new net.minecraft.nbt.ListTag();
+    }
+
+    static {
+        INSTANCE = new net.minecraft.nbt.NbtOps$InitialListCollector();
+    }
+}
+