--- net/minecraft/nbt/NbtOps$HeterogenousListCollector.java
+++ net/minecraft/nbt/NbtOps$HeterogenousListCollector.java
@@ -1,0 +1,62 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.nbt;
+
+public class net.minecraft.nbt.NbtOps$HeterogenousListCollector
+implements net.minecraft.nbt.NbtOps$ListCollector
+ {
+    public net.minecraft.nbt.ListTag result;
+
+    public net.minecraft.nbt.NbtOps$HeterogenousListCollector() {
+        this.result = new net.minecraft.nbt.ListTag();
+    }
+
+    public net.minecraft.nbt.NbtOps$HeterogenousListCollector(java.util.Collection<net.minecraft.nbt.Tag> $$0) {
+        this.result = new net.minecraft.nbt.ListTag();
+        this.result.addAll($$0);
+    }
+
+    public net.minecraft.nbt.NbtOps$HeterogenousListCollector(it.unimi.dsi.fastutil.ints.IntArrayList $$02) {
+        this.result = new net.minecraft.nbt.ListTag();
+        $$02.forEach($$0 -> this.result.add((java.lang.Object)net.minecraft.nbt.NbtOps$HeterogenousListCollector.wrapElement((net.minecraft.nbt.Tag)net.minecraft.nbt.IntTag.valueOf((int)$$0))));
+    }
+
+    public net.minecraft.nbt.NbtOps$HeterogenousListCollector(it.unimi.dsi.fastutil.bytes.ByteArrayList $$02) {
+        this.result = new net.minecraft.nbt.ListTag();
+        $$02.forEach($$0 -> this.result.add((java.lang.Object)net.minecraft.nbt.NbtOps$HeterogenousListCollector.wrapElement((net.minecraft.nbt.Tag)net.minecraft.nbt.ByteTag.valueOf((byte)$$0))));
+    }
+
+    public net.minecraft.nbt.NbtOps$HeterogenousListCollector(it.unimi.dsi.fastutil.longs.LongArrayList $$02) {
+        this.result = new net.minecraft.nbt.ListTag();
+        $$02.forEach($$0 -> this.result.add((java.lang.Object)net.minecraft.nbt.NbtOps$HeterogenousListCollector.wrapElement((net.minecraft.nbt.Tag)net.minecraft.nbt.LongTag.valueOf((long)$$0))));
+    }
+
+    public static boolean isWrapper(net.minecraft.nbt.CompoundTag $$0) {
+        return $$0.size() == 1 && $$0.contains("");
+    }
+
+    public static net.minecraft.nbt.Tag wrapIfNeeded(net.minecraft.nbt.Tag $$0) {
+        net.minecraft.nbt.CompoundTag $$1;
+        if ($$0 instanceof net.minecraft.nbt.CompoundTag && !net.minecraft.nbt.NbtOps$HeterogenousListCollector.isWrapper((net.minecraft.nbt.CompoundTag)($$1 = (net.minecraft.nbt.CompoundTag)$$0))) {
+            return $$1;
+        }
+        return net.minecraft.nbt.NbtOps$HeterogenousListCollector.wrapElement((net.minecraft.nbt.Tag)$$0);
+    }
+
+    public static net.minecraft.nbt.CompoundTag wrapElement(net.minecraft.nbt.Tag $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = new net.minecraft.nbt.CompoundTag();
+        $$1.put("", $$0);
+        return $$1;
+    }
+
+    public net.minecraft.nbt.NbtOps$ListCollector accept(net.minecraft.nbt.Tag $$0) {
+        this.result.add((java.lang.Object)net.minecraft.nbt.NbtOps$HeterogenousListCollector.wrapIfNeeded((net.minecraft.nbt.Tag)$$0));
+        return this;
+    }
+
+    public net.minecraft.nbt.Tag result() {
+        return this.result;
+    }
+}
+