--- net/minecraft/nbt/NbtOps$HomogenousListCollector.java
+++ net/minecraft/nbt/NbtOps$HomogenousListCollector.java
@@ -1,0 +1,33 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.nbt;
+
+public class net.minecraft.nbt.NbtOps$HomogenousListCollector
+implements net.minecraft.nbt.NbtOps$ListCollector
+ {
+    public net.minecraft.nbt.ListTag result;
+
+    public net.minecraft.nbt.NbtOps$HomogenousListCollector(net.minecraft.nbt.Tag $$0) {
+        this.result = new net.minecraft.nbt.ListTag();
+        this.result.add((java.lang.Object)$$0);
+    }
+
+    public net.minecraft.nbt.NbtOps$HomogenousListCollector(net.minecraft.nbt.ListTag $$0) {
+        this.result = new net.minecraft.nbt.ListTag();
+        this.result.addAll((java.util.Collection)$$0);
+    }
+
+    public net.minecraft.nbt.NbtOps$ListCollector accept(net.minecraft.nbt.Tag $$0) {
+        if ($$0.getId() != this.result.getElementType()) {
+            return new net.minecraft.nbt.NbtOps$HeterogenousListCollector().acceptAll((java.lang.Iterable)this.result).accept($$0);
+        }
+        this.result.add((java.lang.Object)$$0);
+        return this;
+    }
+
+    public net.minecraft.nbt.Tag result() {
+        return this.result;
+    }
+}
+