--- net/minecraft/nbt/NbtOps$ByteListCollector.java
+++ net/minecraft/nbt/NbtOps$ByteListCollector.java
@@ -1,0 +1,34 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.nbt;
+
+public class net.minecraft.nbt.NbtOps$ByteListCollector
+implements net.minecraft.nbt.NbtOps$ListCollector
+ {
+    public it.unimi.dsi.fastutil.bytes.ByteArrayList values;
+
+    public net.minecraft.nbt.NbtOps$ByteListCollector(byte $$0) {
+        this.values = new it.unimi.dsi.fastutil.bytes.ByteArrayList();
+        this.values.add($$0);
+    }
+
+    public net.minecraft.nbt.NbtOps$ByteListCollector(byte[] $$0) {
+        this.values = new it.unimi.dsi.fastutil.bytes.ByteArrayList();
+        this.values.addElements(0, $$0);
+    }
+
+    public net.minecraft.nbt.NbtOps$ListCollector accept(net.minecraft.nbt.Tag $$0) {
+        if ($$0 instanceof net.minecraft.nbt.ByteTag) {
+            net.minecraft.nbt.ByteTag $$1 = (net.minecraft.nbt.ByteTag)$$0;
+            this.values.add($$1.getAsByte());
+            return this;
+        }
+        return new net.minecraft.nbt.NbtOps$HeterogenousListCollector(this.values).accept($$0);
+    }
+
+    public net.minecraft.nbt.Tag result() {
+        return new net.minecraft.nbt.ByteArrayTag(this.values.toByteArray());
+    }
+}
+