--- net/minecraft/nbt/NbtOps$1.java
+++ net/minecraft/nbt/NbtOps$1.java
@@ -9,7 +9,7 @@
     public /* synthetic */ net.minecraft.nbt.CompoundTag val$tag;
     public /* synthetic */ net.minecraft.nbt.NbtOps this$0;
 
-    public net.minecraft.nbt.NbtOps$1() {
+    public net.minecraft.nbt.NbtOps$1(net.minecraft.nbt.NbtOps $$0, net.minecraft.nbt.CompoundTag compoundTag) {
         this.this$0 = $$0;
         this.val$tag = compoundTag;
     }
@@ -25,7 +25,7 @@
     }
 
     public java.util.stream.Stream<com.mojang.datafixers.util.Pair<net.minecraft.nbt.Tag, net.minecraft.nbt.Tag>> entries() {
-        return this.val$tag.getAllKeys().stream().map(key -> com.mojang.datafixers.util.Pair.of((java.lang.Object)this.this$0.createString(key), (java.lang.Object)this.val$tag.get(key)));
+        return this.val$tag.getAllKeys().stream().map($$1 -> com.mojang.datafixers.util.Pair.of((java.lang.Object)this.this$0.createString($$1), (java.lang.Object)this.val$tag.get($$1)));
     }
 
     public java.lang.String toString() {