--- net/minecraft/nbt/ShortTag.java
+++ net/minecraft/nbt/ShortTag.java
@@ -10,19 +10,19 @@
     public static net.minecraft.nbt.TagType<net.minecraft.nbt.ShortTag> TYPE;
     public short data;
 
-    public net.minecraft.nbt.ShortTag(short data) {
-        this.data = data;
+    public net.minecraft.nbt.ShortTag(short $$0) {
+        this.data = $$0;
     }
 
-    public static net.minecraft.nbt.ShortTag valueOf(short data) {
-        if (data >= -128 && data <= 1024) {
-            return net.minecraft.nbt.ShortTag$Cache.cache[data - -128];
+    public static net.minecraft.nbt.ShortTag valueOf(short $$0) {
+        if ($$0 >= -128 && $$0 <= 1024) {
+            return net.minecraft.nbt.ShortTag$Cache.cache[$$0 - -128];
         }
-        return new net.minecraft.nbt.ShortTag(data);
+        return new net.minecraft.nbt.ShortTag($$0);
     }
 
-    public void write(java.io.DataOutput output) throws java.io.IOException {
-        output.writeShort((int)this.data);
+    public void write(java.io.DataOutput $$0) throws java.io.IOException {
+        $$0.writeShort((int)this.data);
     }
 
     public byte getId() {
@@ -37,11 +37,11 @@
         return this;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        return other instanceof net.minecraft.nbt.ShortTag && this.data == ((net.minecraft.nbt.ShortTag)other).data;
+        return $$0 instanceof net.minecraft.nbt.ShortTag && this.data == ((net.minecraft.nbt.ShortTag)$$0).data;
     }
 
     public int hashCode() {