--- net/minecraft/nbt/TagType$1.java
+++ net/minecraft/nbt/TagType$1.java
@@ -16,7 +16,7 @@
         return new java.io.IOException("Invalid tag id: " + this.val$id);
     }
 
-    public net.minecraft.nbt.EndTag load(java.io.DataInput input, int depth, net.minecraft.nbt.NbtAccounter accounter) throws java.io.IOException {
+    public net.minecraft.nbt.EndTag load(java.io.DataInput $$0, int $$1, net.minecraft.nbt.NbtAccounter $$2) throws java.io.IOException {
         throw this.createException();
     }
 