--- net/minecraft/nbt/TagTypes.java
+++ net/minecraft/nbt/TagTypes.java
@@ -10,11 +10,11 @@
     public net.minecraft.nbt.TagTypes() {
     }
 
-    public static net.minecraft.nbt.TagType<?> getType(int id) {
-        if (id < 0 || id >= TYPES.length) {
-            return net.minecraft.nbt.TagType.createInvalid((int)id);
+    public static net.minecraft.nbt.TagType<?> getType(int $$0) {
+        if ($$0 < 0 || $$0 >= TYPES.length) {
+            return net.minecraft.nbt.TagType.createInvalid((int)$$0);
         }
-        return TYPES[id];
+        return TYPES[$$0];
     }
 
     static {