--- net/minecraft/client/multiplayer/ClientRegistryLayer.java
+++ net/minecraft/client/multiplayer/ClientRegistryLayer.java
@@ -1,0 +1,43 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.multiplayer;
+
+public class net.minecraft.client.multiplayer.ClientRegistryLayer
+extends java.lang.Enum<net.minecraft.client.multiplayer.ClientRegistryLayer>
+ {
+    public static /* enum */ net.minecraft.client.multiplayer.ClientRegistryLayer STATIC;
+    public static /* enum */ net.minecraft.client.multiplayer.ClientRegistryLayer REMOTE;
+    public static java.util.List<net.minecraft.client.multiplayer.ClientRegistryLayer> VALUES;
+    public static net.minecraft.core.RegistryAccess$Frozen STATIC_ACCESS;
+    public static /* synthetic */ net.minecraft.client.multiplayer.ClientRegistryLayer[] $VALUES;
+
+    public static net.minecraft.client.multiplayer.ClientRegistryLayer[] values() {
+        return (net.minecraft.client.multiplayer.ClientRegistryLayer[])$VALUES.clone();
+    }
+
+    public static net.minecraft.client.multiplayer.ClientRegistryLayer valueOf(java.lang.String $$0) {
+        return (net.minecraft.client.multiplayer.ClientRegistryLayer)java.lang.Enum.valueOf(net.minecraft.client.multiplayer.ClientRegistryLayer.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.client.multiplayer.ClientRegistryLayer() {
+        super(string, n);
+    }
+
+    public static net.minecraft.core.LayeredRegistryAccess<net.minecraft.client.multiplayer.ClientRegistryLayer> createRegistryAccess() {
+        return new net.minecraft.core.LayeredRegistryAccess(VALUES).replaceFrom((java.lang.Object)STATIC, new net.minecraft.core.RegistryAccess$Frozen[]{STATIC_ACCESS});
+    }
+
+    public static /* synthetic */ net.minecraft.client.multiplayer.ClientRegistryLayer[] $values() {
+        return new net.minecraft.client.multiplayer.ClientRegistryLayer[]{STATIC, REMOTE};
+    }
+
+    static {
+        STATIC = new net.minecraft.client.multiplayer.ClientRegistryLayer("STATIC", 0);
+        REMOTE = new net.minecraft.client.multiplayer.ClientRegistryLayer("REMOTE", 1);
+        $VALUES = net.minecraft.client.multiplayer.ClientRegistryLayer.$values();
+        VALUES = java.util.List.of((java.lang.Object[])net.minecraft.client.multiplayer.ClientRegistryLayer.values());
+        STATIC_ACCESS = net.minecraft.core.RegistryAccess.fromRegistryOfRegistries((net.minecraft.core.Registry)net.minecraft.core.Registry.REGISTRY);
+    }
+}
+