--- net/minecraft/client/multiplayer/ClientLevel$ClientLevelData.java
+++ net/minecraft/client/multiplayer/ClientLevel$ClientLevelData.java
@@ -50,35 +50,35 @@
         return this.dayTime;
     }
 
-    public void setXSpawn(int x) {
-        this.xSpawn = x;
+    public void setXSpawn(int $$0) {
+        this.xSpawn = $$0;
     }
 
-    public void setYSpawn(int y) {
-        this.ySpawn = y;
+    public void setYSpawn(int $$0) {
+        this.ySpawn = $$0;
     }
 
-    public void setZSpawn(int z) {
-        this.zSpawn = z;
+    public void setZSpawn(int $$0) {
+        this.zSpawn = $$0;
     }
 
-    public void setSpawnAngle(float angle) {
-        this.spawnAngle = angle;
+    public void setSpawnAngle(float $$0) {
+        this.spawnAngle = $$0;
     }
 
-    public void setGameTime(long time) {
-        this.gameTime = time;
+    public void setGameTime(long $$0) {
+        this.gameTime = $$0;
     }
 
-    public void setDayTime(long time) {
-        this.dayTime = time;
+    public void setDayTime(long $$0) {
+        this.dayTime = $$0;
     }
 
-    public void setSpawn(net.minecraft.core.BlockPos spawnPoint, float angle) {
-        this.xSpawn = spawnPoint.getX();
-        this.ySpawn = spawnPoint.getY();
-        this.zSpawn = spawnPoint.getZ();
-        this.spawnAngle = angle;
+    public void setSpawn(net.minecraft.core.BlockPos $$0, float $$1) {
+        this.xSpawn = $$0.getX();
+        this.ySpawn = $$0.getY();
+        this.zSpawn = $$0.getZ();
+        this.spawnAngle = $$1;
     }
 
     public boolean isThundering() {
@@ -89,8 +89,8 @@
         return this.raining;
     }
 
-    public void setRaining(boolean isRaining) {
-        this.raining = isRaining;
+    public void setRaining(boolean $$0) {
+        this.raining = $$0;
     }
 
     public boolean isHardcore() {
@@ -113,12 +113,12 @@
         super.fillCrashReportCategory($$0, $$1);
     }
 
-    public void setDifficulty(net.minecraft.world.Difficulty difficulty) {
-        this.difficulty = difficulty;
+    public void setDifficulty(net.minecraft.world.Difficulty $$0) {
+        this.difficulty = $$0;
     }
 
-    public void setDifficultyLocked(boolean difficultyLocked) {
-        this.difficultyLocked = difficultyLocked;
+    public void setDifficultyLocked(boolean $$0) {
+        this.difficultyLocked = $$0;
     }
 
     public double getHorizonHeight(net.minecraft.world.level.LevelHeightAccessor $$0) {