--- net/minecraft/client/multiplayer/ClientSuggestionProvider.java
+++ net/minecraft/client/multiplayer/ClientSuggestionProvider.java
@@ -56,9 +56,9 @@
         return this.connection.getRecipeManager().getRecipeIds();
     }
 
-    public boolean hasPermission(int level) {
-        net.minecraft.client.player.LocalPlayer localPlayer = this.minecraft.player;
-        return localPlayer != null ? localPlayer.hasPermissions(level) : level == 0;
+    public boolean hasPermission(int $$0) {
+        net.minecraft.client.player.LocalPlayer $$1 = this.minecraft.player;
+        return $$1 != null ? $$1.hasPermissions($$0) : $$0 == 0;
     }
 
     public java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> suggestRegistryElements(net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<?>> $$0, net.minecraft.commands.SharedSuggestionProvider$ElementSuggestionType $$1, com.mojang.brigadier.suggestion.SuggestionsBuilder $$22, com.mojang.brigadier.context.CommandContext<?> $$3) {
@@ -78,12 +78,12 @@
         return this.pendingSuggestionsFuture;
     }
 
-    public static java.lang.String prettyPrint(double doubleValue) {
-        return java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"%.2f", (java.lang.Object[])new java.lang.Object[]{doubleValue});
+    public static java.lang.String prettyPrint(double $$0) {
+        return java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"%.2f", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
-    public static java.lang.String prettyPrint(int intValue) {
-        return java.lang.Integer.toString((int)intValue);
+    public static java.lang.String prettyPrint(int $$0) {
+        return java.lang.Integer.toString((int)$$0);
     }
 
     public java.util.Collection<net.minecraft.commands.SharedSuggestionProvider$TextCoordinates> getRelevantCoordinates() {
@@ -112,9 +112,13 @@
         return this.connection.registryAccess();
     }
 
-    public void completeCustomSuggestions(int transaction, com.mojang.brigadier.suggestion.Suggestions result) {
-        if (transaction == this.pendingSuggestionsId) {
-            this.pendingSuggestionsFuture.complete((java.lang.Object)result);
+    public net.minecraft.world.flag.FeatureFlagSet enabledFeatures() {
+        return this.connection.enabledFeatures();
+    }
+
+    public void completeCustomSuggestions(int $$0, com.mojang.brigadier.suggestion.Suggestions $$1) {
+        if ($$0 == this.pendingSuggestionsId) {
+            this.pendingSuggestionsFuture.complete((java.lang.Object)$$1);
             this.pendingSuggestionsFuture = null;
             this.pendingSuggestionsId = -1;
         }