--- net/minecraft/client/multiplayer/ServerList.java
+++ net/minecraft/client/multiplayer/ServerList.java
@@ -69,8 +69,8 @@
         }
     }
 
-    public net.minecraft.client.multiplayer.ServerData get(int index) {
-        return (net.minecraft.client.multiplayer.ServerData)this.serverList.get(index);
+    public net.minecraft.client.multiplayer.ServerData get(int $$0) {
+        return (net.minecraft.client.multiplayer.ServerData)this.serverList.get($$0);
     }
 
     @javax.annotation.Nullable
@@ -98,9 +98,9 @@
         return null;
     }
 
-    public void remove(net.minecraft.client.multiplayer.ServerData serverData) {
-        if (!this.serverList.remove((java.lang.Object)serverData)) {
-            this.hiddenServerList.remove((java.lang.Object)serverData);
+    public void remove(net.minecraft.client.multiplayer.ServerData $$0) {
+        if (!this.serverList.remove((java.lang.Object)$$0)) {
+            this.hiddenServerList.remove((java.lang.Object)$$0);
         }
     }
 
@@ -119,15 +119,15 @@
         return this.serverList.size();
     }
 
-    public void swap(int pos1, int pos2) {
-        net.minecraft.client.multiplayer.ServerData serverData = this.get(pos1);
-        this.serverList.set(pos1, (java.lang.Object)this.get(pos2));
-        this.serverList.set(pos2, (java.lang.Object)serverData);
+    public void swap(int $$0, int $$1) {
+        net.minecraft.client.multiplayer.ServerData $$2 = this.get($$0);
+        this.serverList.set($$0, (java.lang.Object)this.get($$1));
+        this.serverList.set($$1, (java.lang.Object)$$2);
         this.save();
     }
 
-    public void replace(int index, net.minecraft.client.multiplayer.ServerData server) {
-        this.serverList.set(index, (java.lang.Object)server);
+    public void replace(int $$0, net.minecraft.client.multiplayer.ServerData $$1) {
+        this.serverList.set($$0, (java.lang.Object)$$1);
     }
 
     public static boolean set(net.minecraft.client.multiplayer.ServerData $$0, java.util.List<net.minecraft.client.multiplayer.ServerData> $$1) {
@@ -140,12 +140,12 @@
         return false;
     }
 
-    public static void saveSingleServer(net.minecraft.client.multiplayer.ServerData server) {
+    public static void saveSingleServer(net.minecraft.client.multiplayer.ServerData $$0) {
         IO_MAILBOX.tell(() -> {
             net.minecraft.client.multiplayer.ServerList $$1 = new net.minecraft.client.multiplayer.ServerList(net.minecraft.client.Minecraft.getInstance());
             $$1.load();
-            if (!net.minecraft.client.multiplayer.ServerList.set((net.minecraft.client.multiplayer.ServerData)server, (java.util.List)$$1.serverList)) {
-                net.minecraft.client.multiplayer.ServerList.set((net.minecraft.client.multiplayer.ServerData)server, (java.util.List)$$1.hiddenServerList);
+            if (!net.minecraft.client.multiplayer.ServerList.set((net.minecraft.client.multiplayer.ServerData)$$0, (java.util.List)$$1.serverList)) {
+                net.minecraft.client.multiplayer.ServerList.set((net.minecraft.client.multiplayer.ServerData)$$0, (java.util.List)$$1.hiddenServerList);
             }
             $$1.save();
         });