--- net/minecraft/BlockUtil$FoundRectangle.java
+++ net/minecraft/BlockUtil$FoundRectangle.java
@@ -9,10 +9,10 @@
     public int axis1Size;
     public int axis2Size;
 
-    public net.minecraft.BlockUtil$FoundRectangle(net.minecraft.core.BlockPos minCorner, int axis1Size, int axis2Size) {
-        this.minCorner = minCorner;
-        this.axis1Size = axis1Size;
-        this.axis2Size = axis2Size;
+    public net.minecraft.BlockUtil$FoundRectangle(net.minecraft.core.BlockPos $$0, int $$1, int $$2) {
+        this.minCorner = $$0;
+        this.axis1Size = $$1;
+        this.axis2Size = $$2;
     }
 }
 