--- net/minecraft/BlockUtil$IntBounds.java
+++ net/minecraft/BlockUtil$IntBounds.java
@@ -8,9 +8,9 @@
     public int min;
     public int max;
 
-    public net.minecraft.BlockUtil$IntBounds(int min, int max) {
-        this.min = min;
-        this.max = max;
+    public net.minecraft.BlockUtil$IntBounds(int $$0, int $$1) {
+        this.min = $$0;
+        this.max = $$1;
     }
 
     public java.lang.String toString() {