--- net/minecraft/DefaultUncaughtExceptionHandler.java
+++ net/minecraft/DefaultUncaughtExceptionHandler.java
@@ -12,8 +12,8 @@
         this.logger = $$0;
     }
 
-    public void uncaughtException(java.lang.Thread thread, java.lang.Throwable exception) {
-        this.logger.error("Caught previously unhandled exception :", exception);
+    public void uncaughtException(java.lang.Thread $$0, java.lang.Throwable $$1) {
+        this.logger.error("Caught previously unhandled exception :", $$1);
     }
 }
 