--- net/minecraft/ReportedException.java
+++ net/minecraft/ReportedException.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.CrashReport report;
 
-    public net.minecraft.ReportedException(net.minecraft.CrashReport report) {
-        this.report = report;
+    public net.minecraft.ReportedException(net.minecraft.CrashReport $$0) {
+        this.report = $$0;
     }
 
     public net.minecraft.CrashReport getReport() {