--- net/minecraft/ResourceLocationException.java
+++ net/minecraft/ResourceLocationException.java
@@ -6,12 +6,12 @@
 public class net.minecraft.ResourceLocationException
 extends java.lang.RuntimeException
  {
-    public net.minecraft.ResourceLocationException(java.lang.String error) {
-        super(error);
+    public net.minecraft.ResourceLocationException(java.lang.String $$0) {
+        super($$0);
     }
 
-    public net.minecraft.ResourceLocationException(java.lang.String message, java.lang.Throwable cause) {
-        super(message, cause);
+    public net.minecraft.ResourceLocationException(java.lang.String $$0, java.lang.Throwable $$1) {
+        super($$0, $$1);
     }
 }
 