--- net/minecraft/SharedConstants.java
+++ net/minecraft/SharedConstants.java
@@ -6,28 +6,28 @@
 public class net.minecraft.SharedConstants
  {
     @java.lang.Deprecated
-    public static boolean SNAPSHOT = false;
+    public static boolean SNAPSHOT = true;
     @java.lang.Deprecated
-    public static int WORLD_VERSION = 3120;
+    public static int WORLD_VERSION = 3205;
     @java.lang.Deprecated
     public static java.lang.String SERIES = "main";
     @java.lang.Deprecated
-    public static java.lang.String VERSION_STRING = "1.19.2";
+    public static java.lang.String VERSION_STRING = "22w42a";
     @java.lang.Deprecated
-    public static java.lang.String RELEASE_TARGET = "1.19.2";
+    public static int RELEASE_NETWORK_PROTOCOL_VERSION = 761;
     @java.lang.Deprecated
-    public static int RELEASE_NETWORK_PROTOCOL_VERSION = 760;
-    @java.lang.Deprecated
-    public static int SNAPSHOT_NETWORK_PROTOCOL_VERSION = 103;
-    public static int SNBT_NAG_VERSION = 3075;
+    public static int SNAPSHOT_NETWORK_PROTOCOL_VERSION = 104;
+    public static int SNBT_NAG_VERSION = 3200;
     public static int SNAPSHOT_PROTOCOL_BIT = 30;
-    public static boolean THROW_ON_TASK_FAILURE = false;
+    public static boolean THROW_ON_TASK_FAILURE = true;
     @java.lang.Deprecated
-    public static int RESOURCE_PACK_FORMAT = 9;
+    public static int RESOURCE_PACK_FORMAT = 11;
     @java.lang.Deprecated
     public static int DATA_PACK_FORMAT = 10;
+    @java.lang.Deprecated
+    public static int LANGUAGE_FORMAT = 1;
+    public static int REPORT_FORMAT_VERSION = 1;
     public static java.lang.String DATA_VERSION_TAG = "DataVersion";
-    public static boolean CNC_PART_2_ITEMS_AND_BLOCKS = false;
     public static boolean USE_NEW_RENDERSYSTEM = false;
     public static boolean MULTITHREADED_RENDERING = false;
     public static boolean FIX_TNT_DUPE = false;
@@ -74,6 +74,7 @@
     public static boolean DEBUG_SCULK_CATALYST = false;
     public static boolean DEBUG_BYPASS_REALMS_VERSION_CHECK = false;
     public static boolean DEBUG_SOCIAL_INTERACTIONS = false;
+    public static boolean DEBUG_VALIDATE_RESOURCE_PATH_CASE = false;
     public static boolean DEBUG_IGNORE_LOCAL_MOB_CAP = false;
     public static boolean DEBUG_SMALL_SPAWN = false;
     public static boolean DEBUG_DISABLE_LIQUID_SPREADING = false;
@@ -102,6 +103,7 @@
     public static boolean DEBUG_SHOW_SERVER_DEBUG_VALUES = false;
     public static boolean DEBUG_STORE_CHUNK_STACKTRACES = false;
     public static boolean DEBUG_FEATURE_COUNT = false;
+    public static boolean DEBUG_RESOURCE_GENERATION_OVERRIDE = false;
     public static long MAXIMUM_TICK_TIME_NANOS;
     public static boolean CHECK_DATA_FIXER_SCHEMA;
     public static boolean IS_RUNNING_IN_IDE;
@@ -124,12 +126,12 @@
     public net.minecraft.SharedConstants() {
     }
 
-    public static boolean isAllowedChatCharacter(char character) {
-        return character != '§' && character >= ' ' && character != '';
+    public static boolean isAllowedChatCharacter(char $$0) {
+        return $$0 != '§' && $$0 >= ' ' && $$0 != '';
     }
 
-    public static java.lang.String filterText(java.lang.String input) {
-        return net.minecraft.SharedConstants.filterText((java.lang.String)input, (boolean)false);
+    public static java.lang.String filterText(java.lang.String $$0) {
+        return net.minecraft.SharedConstants.filterText((java.lang.String)$$0, (boolean)false);
     }
 
     public static java.lang.String filterText(java.lang.String $$0, boolean $$1) {
@@ -167,7 +169,7 @@
     }
 
     public static int getProtocolVersion() {
-        return 760;
+        return 1073741928;
     }
 
     public static boolean debugVoidTerrain(net.minecraft.world.level.ChunkPos $$0) {