--- net/minecraft/Util$OS.java
+++ net/minecraft/Util$OS.java
@@ -27,50 +27,50 @@
         this.telemetryName = $$0;
     }
 
-    public void openUrl(java.net.URL url) {
+    public void openUrl(java.net.URL $$0) {
         try {
-            java.lang.Process process = (java.lang.Process)java.security.AccessController.doPrivileged(() -> java.lang.Runtime.getRuntime().exec(this.getOpenUrlArguments(url)));
-            process.getInputStream().close();
-            process.getErrorStream().close();
-            process.getOutputStream().close();
+            java.lang.Process $$1 = (java.lang.Process)java.security.AccessController.doPrivileged(() -> java.lang.Runtime.getRuntime().exec(this.getOpenUrlArguments($$0)));
+            $$1.getInputStream().close();
+            $$1.getErrorStream().close();
+            $$1.getOutputStream().close();
         }
-        catch (java.io.IOException | java.security.PrivilegedActionException throwable) {
-            net.minecraft.Util.LOGGER.error("Couldn't open url '{}'", (java.lang.Object)url, (java.lang.Object)throwable);
+        catch (java.io.IOException | java.security.PrivilegedActionException $$2) {
+            net.minecraft.Util.LOGGER.error("Couldn't open url '{}'", (java.lang.Object)$$0, (java.lang.Object)$$2);
         }
     }
 
-    public void openUri(java.net.URI uri) {
+    public void openUri(java.net.URI $$0) {
         try {
-            this.openUrl(uri.toURL());
+            this.openUrl($$0.toURL());
         }
-        catch (java.net.MalformedURLException malformedURLException) {
-            net.minecraft.Util.LOGGER.error("Couldn't open uri '{}'", (java.lang.Object)uri, (java.lang.Object)((java.lang.Object)malformedURLException));
+        catch (java.net.MalformedURLException $$1) {
+            net.minecraft.Util.LOGGER.error("Couldn't open uri '{}'", (java.lang.Object)$$0, (java.lang.Object)((java.lang.Object)$$1));
         }
     }
 
-    public void openFile(java.io.File file) {
+    public void openFile(java.io.File $$0) {
         try {
-            this.openUrl(file.toURI().toURL());
+            this.openUrl($$0.toURI().toURL());
         }
-        catch (java.net.MalformedURLException malformedURLException) {
-            net.minecraft.Util.LOGGER.error("Couldn't open file '{}'", (java.lang.Object)file, (java.lang.Object)((java.lang.Object)malformedURLException));
+        catch (java.net.MalformedURLException $$1) {
+            net.minecraft.Util.LOGGER.error("Couldn't open file '{}'", (java.lang.Object)$$0, (java.lang.Object)((java.lang.Object)$$1));
         }
     }
 
-    public java.lang.String[] getOpenUrlArguments(java.net.URL url) {
-        java.lang.String string = url.toString();
-        if ("file".equals((java.lang.Object)url.getProtocol())) {
-            string = string.replace((java.lang.CharSequence)"file:", (java.lang.CharSequence)"file://");
+    public java.lang.String[] getOpenUrlArguments(java.net.URL $$0) {
+        java.lang.String $$1 = $$0.toString();
+        if ("file".equals((java.lang.Object)$$0.getProtocol())) {
+            $$1 = $$1.replace((java.lang.CharSequence)"file:", (java.lang.CharSequence)"file://");
         }
-        return new java.lang.String[]{"xdg-open", string};
+        return new java.lang.String[]{"xdg-open", $$1};
     }
 
-    public void openUri(java.lang.String uri) {
+    public void openUri(java.lang.String $$0) {
         try {
-            this.openUrl(new java.net.URI(uri).toURL());
+            this.openUrl(new java.net.URI($$0).toURL());
         }
-        catch (java.lang.IllegalArgumentException | java.net.MalformedURLException | java.net.URISyntaxException throwable) {
-            net.minecraft.Util.LOGGER.error("Couldn't open uri '{}'", (java.lang.Object)uri, (java.lang.Object)throwable);
+        catch (java.lang.IllegalArgumentException | java.net.MalformedURLException | java.net.URISyntaxException $$1) {
+            net.minecraft.Util.LOGGER.error("Couldn't open uri '{}'", (java.lang.Object)$$0, (java.lang.Object)$$1);
         }
     }
 