--- net/minecraft/Util$OS$2.java
+++ net/minecraft/Util$OS$2.java
@@ -10,8 +10,8 @@
         super(string, n, $$0);
     }
 
-    public java.lang.String[] getOpenUrlArguments(java.net.URL url) {
-        return new java.lang.String[]{"open", url.toString()};
+    public java.lang.String[] getOpenUrlArguments(java.net.URL $$0) {
+        return new java.lang.String[]{"open", $$0.toString()};
     }
 }
 