--- net/minecraft/world/level/levelgen/feature/ConfiguredFeature.java
+++ net/minecraft/world/level/levelgen/feature/ConfiguredFeature.java
@@ -12,9 +12,9 @@
     public static com.mojang.serialization.Codec<net.minecraft.core.Holder<net.minecraft.world.level.levelgen.feature.ConfiguredFeature<?, ?>>> CODEC;
     public static com.mojang.serialization.Codec<net.minecraft.core.HolderSet<net.minecraft.world.level.levelgen.feature.ConfiguredFeature<?, ?>>> LIST_CODEC;
 
-    public net.minecraft.world.level.levelgen.feature.ConfiguredFeature(F feature, FC config) {
-        this.feature = feature;
-        this.config = config;
+    public net.minecraft.world.level.levelgen.feature.ConfiguredFeature(F $$0, FC $$1) {
+        this.feature = $$0;
+        this.config = $$1;
     }
 
     public boolean place(net.minecraft.world.level.WorldGenLevel $$0, net.minecraft.world.level.chunk.ChunkGenerator $$1, net.minecraft.util.RandomSource $$2, net.minecraft.core.BlockPos $$3) {
@@ -45,12 +45,12 @@
         return (FC)this.config;
     }
 
-    public static /* synthetic */ net.minecraft.world.level.levelgen.feature.Feature lambda$static$0(net.minecraft.world.level.levelgen.feature.ConfiguredFeature configuredFeature) {
-        return configuredFeature.feature;
+    public static /* synthetic */ net.minecraft.world.level.levelgen.feature.Feature lambda$static$0(net.minecraft.world.level.levelgen.feature.ConfiguredFeature $$0) {
+        return $$0.feature;
     }
 
     static {
-        DIRECT_CODEC = net.minecraft.core.Registry.FEATURE.byNameCodec().dispatch(configuredFeature -> configuredFeature.feature, net.minecraft.world.level.levelgen.feature.Feature::configuredCodec);
+        DIRECT_CODEC = net.minecraft.core.Registry.FEATURE.byNameCodec().dispatch($$0 -> $$0.feature, net.minecraft.world.level.levelgen.feature.Feature::configuredCodec);
         CODEC = net.minecraft.resources.RegistryFileCodec.create((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.CONFIGURED_FEATURE_REGISTRY, (com.mojang.serialization.Codec)DIRECT_CODEC);
         LIST_CODEC = net.minecraft.core.RegistryCodecs.homogeneousList((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.CONFIGURED_FEATURE_REGISTRY, (com.mojang.serialization.Codec)DIRECT_CODEC);
     }