--- net/minecraft/world/level/levelgen/feature/EndGatewayFeature.java
+++ net/minecraft/world/level/levelgen/feature/EndGatewayFeature.java
@@ -13,7 +13,7 @@
     public boolean place(net.minecraft.world.level.levelgen.feature.FeaturePlaceContext<net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration> $$0) {
         net.minecraft.core.BlockPos $$1 = $$0.origin();
         net.minecraft.world.level.WorldGenLevel $$2 = $$0.level();
-        net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration $$3 = (net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration)$$0.config();
+        net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration $$32 = (net.minecraft.world.level.levelgen.feature.configurations.EndGatewayConfiguration)$$0.config();
         for (net.minecraft.core.BlockPos $$4 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)$$1.offset(-1, -2, -1), (net.minecraft.core.BlockPos)$$1.offset(1, 2, 1))) {
             boolean $$8;
             boolean $$5 = $$4.getX() == $$1.getX();
@@ -23,12 +23,12 @@
             if ($$5 && $$6 && $$7) {
                 net.minecraft.core.BlockPos $$9 = $$4.immutable();
                 this.setBlock((net.minecraft.world.level.LevelWriter)$$2, $$9, net.minecraft.world.level.block.Blocks.END_GATEWAY.defaultBlockState());
-                $$3.getExit().ifPresent(pos -> {
-                    net.minecraft.world.level.block.entity.BlockEntity blockEntity = $$2.getBlockEntity($$9);
-                    if (blockEntity instanceof net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity) {
-                        net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity theEndGatewayBlockEntity = (net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity)blockEntity;
-                        theEndGatewayBlockEntity.setExitPosition(pos, $$3.isExitExact());
-                        blockEntity.setChanged();
+                $$32.getExit().ifPresent($$3 -> {
+                    net.minecraft.world.level.block.entity.BlockEntity $$4 = $$2.getBlockEntity($$9);
+                    if ($$4 instanceof net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity) {
+                        net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity $$5 = (net.minecraft.world.level.block.entity.TheEndGatewayBlockEntity)$$4;
+                        $$5.setExitPosition($$3, $$32.isExitExact());
+                        $$4.setChanged();
                     }
                 });
                 continue;