--- net/minecraft/world/level/levelgen/feature/EndPodiumFeature.java
+++ net/minecraft/world/level/levelgen/feature/EndPodiumFeature.java
@@ -13,9 +13,9 @@
     public static net.minecraft.core.BlockPos END_PODIUM_LOCATION;
     public boolean active;
 
-    public net.minecraft.world.level.levelgen.feature.EndPodiumFeature(boolean active) {
+    public net.minecraft.world.level.levelgen.feature.EndPodiumFeature(boolean $$0) {
         super(net.minecraft.world.level.levelgen.feature.configurations.NoneFeatureConfiguration.CODEC);
-        this.active = active;
+        this.active = $$0;
     }
 
     public boolean place(net.minecraft.world.level.levelgen.feature.FeaturePlaceContext<net.minecraft.world.level.levelgen.feature.configurations.NoneFeatureConfiguration> $$0) {