--- net/minecraft/world/level/levelgen/feature/HugeFungusConfiguration.java
+++ net/minecraft/world/level/levelgen/feature/HugeFungusConfiguration.java
@@ -13,12 +13,12 @@
     public net.minecraft.world.level.block.state.BlockState decorState;
     public boolean planted;
 
-    public net.minecraft.world.level.levelgen.feature.HugeFungusConfiguration(net.minecraft.world.level.block.state.BlockState validBaseState, net.minecraft.world.level.block.state.BlockState stemState, net.minecraft.world.level.block.state.BlockState hatState, net.minecraft.world.level.block.state.BlockState decorState, boolean planted) {
-        this.validBaseState = validBaseState;
-        this.stemState = stemState;
-        this.hatState = hatState;
-        this.decorState = decorState;
-        this.planted = planted;
+    public net.minecraft.world.level.levelgen.feature.HugeFungusConfiguration(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.state.BlockState $$1, net.minecraft.world.level.block.state.BlockState $$2, net.minecraft.world.level.block.state.BlockState $$3, boolean $$4) {
+        this.validBaseState = $$0;
+        this.stemState = $$1;
+        this.hatState = $$2;
+        this.decorState = $$3;
+        this.planted = $$4;
     }
 
     public static /* synthetic */ com.mojang.datafixers.kinds.App lambda$static$5(com.mojang.serialization.codecs.RecordCodecBuilder$Instance $$02) {