--- net/minecraft/world/level/levelgen/feature/ReplaceBlobsFeature.java
+++ net/minecraft/world/level/levelgen/feature/ReplaceBlobsFeature.java
@@ -36,13 +36,13 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.core.BlockPos findTarget(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos$MutableBlockPos topPos, net.minecraft.world.level.block.Block block) {
-        while (topPos.getY() > level.getMinBuildHeight() + 1) {
-            net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState((net.minecraft.core.BlockPos)topPos);
-            if (blockState.is(block)) {
-                return topPos;
+    public static net.minecraft.core.BlockPos findTarget(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos$MutableBlockPos $$1, net.minecraft.world.level.block.Block $$2) {
+        while ($$1.getY() > $$0.getMinBuildHeight() + 1) {
+            net.minecraft.world.level.block.state.BlockState $$3 = $$0.getBlockState((net.minecraft.core.BlockPos)$$1);
+            if ($$3.is($$2)) {
+                return $$1;
             }
-            topPos.move(net.minecraft.core.Direction.DOWN);
+            $$1.move(net.minecraft.core.Direction.DOWN);
         }
         return null;
     }