--- net/minecraft/world/level/levelgen/feature/SpikeFeature.java
+++ net/minecraft/world/level/levelgen/feature/SpikeFeature.java
@@ -14,10 +14,10 @@
         super($$0);
     }
 
-    public static java.util.List<net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike> getSpikesForLevel(net.minecraft.world.level.WorldGenLevel level) {
-        net.minecraft.util.RandomSource randomSource = net.minecraft.util.RandomSource.create((long)level.getSeed());
-        long l = randomSource.nextLong() & 0xFFFFL;
-        return (java.util.List)SPIKE_CACHE.getUnchecked((java.lang.Object)l);
+    public static java.util.List<net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike> getSpikesForLevel(net.minecraft.world.level.WorldGenLevel $$0) {
+        net.minecraft.util.RandomSource $$1 = net.minecraft.util.RandomSource.create((long)$$0.getSeed());
+        long $$2 = $$1.nextLong() & 0xFFFFL;
+        return (java.util.List)SPIKE_CACHE.getUnchecked((java.lang.Object)$$2);
     }
 
     public boolean place(net.minecraft.world.level.levelgen.feature.FeaturePlaceContext<net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration> $$0) {
@@ -37,6 +37,7 @@
     }
 
     public void placeSpike(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.util.RandomSource $$1, net.minecraft.world.level.levelgen.feature.configurations.SpikeConfiguration $$2, net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike $$3) {
+        net.minecraft.world.entity.boss.enderdragon.EndCrystal $$19;
         int $$4 = $$3.getRadius();
         for (net.minecraft.core.BlockPos $$5 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)new net.minecraft.core.BlockPos($$3.getCenterX() - $$4, $$0.getMinBuildHeight(), $$3.getCenterZ() - $$4), (net.minecraft.core.BlockPos)new net.minecraft.core.BlockPos($$3.getCenterX() + $$4, $$3.getHeight() + 10, $$3.getCenterZ() + $$4))) {
             if ($$5.distToLowCornerSqr((double)$$3.getCenterX(), (double)$$5.getY(), (double)$$3.getCenterZ()) <= (double)($$4 * $$4 + 1) && $$5.getY() < $$3.getHeight()) {
@@ -67,12 +68,13 @@
                 }
             }
         }
-        net.minecraft.world.entity.boss.enderdragon.EndCrystal $$19 = (net.minecraft.world.entity.boss.enderdragon.EndCrystal)net.minecraft.world.entity.EntityType.END_CRYSTAL.create((net.minecraft.world.level.Level)$$0.getLevel());
-        $$19.setBeamTarget($$2.getCrystalBeamTarget());
-        $$19.setInvulnerable($$2.isCrystalInvulnerable());
-        $$19.moveTo((double)$$3.getCenterX() + 0.5, (double)($$3.getHeight() + 1), (double)$$3.getCenterZ() + 0.5, $$1.nextFloat() * 360.0f, 0.0f);
-        $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$19);
-        this.setBlock((net.minecraft.world.level.LevelWriter)$$0, new net.minecraft.core.BlockPos($$3.getCenterX(), $$3.getHeight(), $$3.getCenterZ()), net.minecraft.world.level.block.Blocks.BEDROCK.defaultBlockState());
+        if (($$19 = (net.minecraft.world.entity.boss.enderdragon.EndCrystal)net.minecraft.world.entity.EntityType.END_CRYSTAL.create((net.minecraft.world.level.Level)$$0.getLevel())) != null) {
+            $$19.setBeamTarget($$2.getCrystalBeamTarget());
+            $$19.setInvulnerable($$2.isCrystalInvulnerable());
+            $$19.moveTo((double)$$3.getCenterX() + 0.5, (double)($$3.getHeight() + 1), (double)$$3.getCenterZ() + 0.5, $$1.nextFloat() * 360.0f, 0.0f);
+            $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$19);
+            this.setBlock((net.minecraft.world.level.LevelWriter)$$0, new net.minecraft.core.BlockPos($$3.getCenterX(), $$3.getHeight(), $$3.getCenterZ()), net.minecraft.world.level.block.Blocks.BEDROCK.defaultBlockState());
+        }
     }
 
     static {