--- net/minecraft/world/level/levelgen/feature/SpikeFeature$EndSpike.java
+++ net/minecraft/world/level/levelgen/feature/SpikeFeature$EndSpike.java
@@ -13,17 +13,17 @@
     public boolean guarded;
     public net.minecraft.world.phys.AABB topBoundingBox;
 
-    public net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike(int centerX, int centerZ, int radius, int height, boolean guarded) {
-        this.centerX = centerX;
-        this.centerZ = centerZ;
-        this.radius = radius;
-        this.height = height;
-        this.guarded = guarded;
-        this.topBoundingBox = new net.minecraft.world.phys.AABB((double)(centerX - radius), (double)net.minecraft.world.level.dimension.DimensionType.MIN_Y, (double)(centerZ - radius), (double)(centerX + radius), (double)net.minecraft.world.level.dimension.DimensionType.MAX_Y, (double)(centerZ + radius));
+    public net.minecraft.world.level.levelgen.feature.SpikeFeature$EndSpike(int $$0, int $$1, int $$2, int $$3, boolean $$4) {
+        this.centerX = $$0;
+        this.centerZ = $$1;
+        this.radius = $$2;
+        this.height = $$3;
+        this.guarded = $$4;
+        this.topBoundingBox = new net.minecraft.world.phys.AABB((double)($$0 - $$2), (double)net.minecraft.world.level.dimension.DimensionType.MIN_Y, (double)($$1 - $$2), (double)($$0 + $$2), (double)net.minecraft.world.level.dimension.DimensionType.MAX_Y, (double)($$1 + $$2));
     }
 
-    public boolean isCenterWithinChunk(net.minecraft.core.BlockPos pos) {
-        return net.minecraft.core.SectionPos.blockToSectionCoord((int)pos.getX()) == net.minecraft.core.SectionPos.blockToSectionCoord((int)this.centerX) && net.minecraft.core.SectionPos.blockToSectionCoord((int)pos.getZ()) == net.minecraft.core.SectionPos.blockToSectionCoord((int)this.centerZ);
+    public boolean isCenterWithinChunk(net.minecraft.core.BlockPos $$0) {
+        return net.minecraft.core.SectionPos.blockToSectionCoord((int)$$0.getX()) == net.minecraft.core.SectionPos.blockToSectionCoord((int)this.centerX) && net.minecraft.core.SectionPos.blockToSectionCoord((int)$$0.getZ()) == net.minecraft.core.SectionPos.blockToSectionCoord((int)this.centerZ);
     }
 
     public int getCenterX() {