--- net/minecraft/world/level/levelgen/feature/TwistingVinesFeature.java
+++ net/minecraft/world/level/levelgen/feature/TwistingVinesFeature.java
@@ -39,13 +39,13 @@
         return true;
     }
 
-    public static boolean findFirstAirBlockAboveGround(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos$MutableBlockPos pos) {
+    public static boolean findFirstAirBlockAboveGround(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos$MutableBlockPos $$1) {
         do {
-            pos.move(0, -1, 0);
-            if (!level.isOutsideBuildHeight((net.minecraft.core.BlockPos)pos)) continue;
+            $$1.move(0, -1, 0);
+            if (!$$0.isOutsideBuildHeight((net.minecraft.core.BlockPos)$$1)) continue;
             return false;
-        } while (level.getBlockState((net.minecraft.core.BlockPos)pos).isAir());
-        pos.move(0, 1, 0);
+        } while ($$0.getBlockState((net.minecraft.core.BlockPos)$$1).isAir());
+        $$1.move(0, 1, 0);
         return true;
     }
 
@@ -62,12 +62,12 @@
         }
     }
 
-    public static boolean isInvalidPlacementLocation(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos) {
-        if (!level.isEmptyBlock(pos)) {
+    public static boolean isInvalidPlacementLocation(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos $$1) {
+        if (!$$0.isEmptyBlock($$1)) {
             return true;
         }
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(pos.below());
-        return !blockState.is(net.minecraft.world.level.block.Blocks.NETHERRACK) && !blockState.is(net.minecraft.world.level.block.Blocks.WARPED_NYLIUM) && !blockState.is(net.minecraft.world.level.block.Blocks.WARPED_WART_BLOCK);
+        net.minecraft.world.level.block.state.BlockState $$2 = $$0.getBlockState($$1.below());
+        return !$$2.is(net.minecraft.world.level.block.Blocks.NETHERRACK) && !$$2.is(net.minecraft.world.level.block.Blocks.WARPED_NYLIUM) && !$$2.is(net.minecraft.world.level.block.Blocks.WARPED_WART_BLOCK);
     }
 }
 