--- net/minecraft/world/level/levelgen/feature/VoidStartPlatformFeature.java
+++ net/minecraft/world/level/levelgen/feature/VoidStartPlatformFeature.java
@@ -15,8 +15,8 @@
         super($$0);
     }
 
-    public static int checkerboardDistance(int firstX, int firstZ, int secondX, int secondZ) {
-        return java.lang.Math.max((int)java.lang.Math.abs((int)(firstX - secondX)), (int)java.lang.Math.abs((int)(firstZ - secondZ)));
+    public static int checkerboardDistance(int $$0, int $$1, int $$2, int $$3) {
+        return java.lang.Math.max((int)java.lang.Math.abs((int)($$0 - $$2)), (int)java.lang.Math.abs((int)($$1 - $$3)));
     }
 
     public boolean place(net.minecraft.world.level.levelgen.feature.FeaturePlaceContext<net.minecraft.world.level.levelgen.feature.configurations.NoneFeatureConfiguration> $$0) {