--- net/minecraft/commands/CommandBuildContext.java
+++ net/minecraft/commands/CommandBuildContext.java
@@ -6,11 +6,13 @@
 public class net.minecraft.commands.CommandBuildContext
  {
     public net.minecraft.core.RegistryAccess registryAccess;
+    public net.minecraft.world.flag.FeatureFlagSet enabledFeatures;
     public net.minecraft.commands.CommandBuildContext$MissingTagAccessPolicy missingTagAccessPolicy;
 
-    public net.minecraft.commands.CommandBuildContext(net.minecraft.core.RegistryAccess $$0) {
+    public net.minecraft.commands.CommandBuildContext(net.minecraft.core.RegistryAccess $$0, net.minecraft.world.flag.FeatureFlagSet $$1) {
         this.missingTagAccessPolicy = net.minecraft.commands.CommandBuildContext$MissingTagAccessPolicy.FAIL;
         this.registryAccess = $$0;
+        this.enabledFeatures = $$1;
     }
 
     public void missingTagAccessPolicy(net.minecraft.commands.CommandBuildContext$MissingTagAccessPolicy $$0) {
@@ -18,7 +20,8 @@
     }
 
     public <T> net.minecraft.core.HolderLookup<T> holderLookup(net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<T>> $$0) {
-        return new /* Unavailable Anonymous Inner Class!! */;
+        net.minecraft.commands.CommandBuildContext$1 $$1 = new /* Unavailable Anonymous Inner Class!! */;
+        return $$1.filterFeatures(this.enabledFeatures);
     }
 }
 