--- net/minecraft/commands/Commands$CommandSelection.java
+++ net/minecraft/commands/Commands$CommandSelection.java
@@ -21,10 +21,10 @@
         return (net.minecraft.commands.Commands$CommandSelection)java.lang.Enum.valueOf(net.minecraft.commands.Commands$CommandSelection.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.commands.Commands$CommandSelection(boolean includeIntegrated, boolean includeDedicated) {
+    public net.minecraft.commands.Commands$CommandSelection(boolean $$0, boolean $$1) {
         super(string, n);
-        this.includeIntegrated = includeIntegrated;
-        this.includeDedicated = includeDedicated;
+        this.includeIntegrated = $$0;
+        this.includeDedicated = $$1;
     }
 
     public static /* synthetic */ net.minecraft.commands.Commands$CommandSelection[] $values() {