--- net/minecraft/commands/CommandFunction$CacheableFunction.java
+++ net/minecraft/commands/CommandFunction$CacheableFunction.java
@@ -11,22 +11,22 @@
     public boolean resolved;
     public java.util.Optional<net.minecraft.commands.CommandFunction> function;
 
-    public net.minecraft.commands.CommandFunction$CacheableFunction(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation id) {
+    public net.minecraft.commands.CommandFunction$CacheableFunction(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation $$0) {
         this.function = java.util.Optional.empty();
-        this.id = id;
+        this.id = $$0;
     }
 
-    public net.minecraft.commands.CommandFunction$CacheableFunction(net.minecraft.commands.CommandFunction function) {
+    public net.minecraft.commands.CommandFunction$CacheableFunction(net.minecraft.commands.CommandFunction $$0) {
         this.function = java.util.Optional.empty();
         this.resolved = true;
         this.id = null;
-        this.function = java.util.Optional.of((java.lang.Object)function);
+        this.function = java.util.Optional.of((java.lang.Object)$$0);
     }
 
-    public java.util.Optional<net.minecraft.commands.CommandFunction> get(net.minecraft.server.ServerFunctionManager functionManager) {
+    public java.util.Optional<net.minecraft.commands.CommandFunction> get(net.minecraft.server.ServerFunctionManager $$0) {
         if (!this.resolved) {
             if (this.id != null) {
-                this.function = functionManager.get(this.id);
+                this.function = $$0.get(this.id);
             }
             this.resolved = true;
         }
@@ -35,7 +35,7 @@
 
     @javax.annotation.Nullable
     public net.minecraft.resources.ResourceLocation getId() {
-        return (net.minecraft.resources.ResourceLocation)this.function.map(function -> function.id).orElse((java.lang.Object)this.id);
+        return (net.minecraft.resources.ResourceLocation)this.function.map($$0 -> $$0.id).orElse((java.lang.Object)this.id);
     }
 
     static {