--- net/minecraft/commands/CommandFunction$CommandEntry.java
+++ net/minecraft/commands/CommandFunction$CommandEntry.java
@@ -8,8 +8,8 @@
  {
     public com.mojang.brigadier.ParseResults<net.minecraft.commands.CommandSourceStack> parse;
 
-    public net.minecraft.commands.CommandFunction$CommandEntry(com.mojang.brigadier.ParseResults<net.minecraft.commands.CommandSourceStack> parse) {
-        this.parse = parse;
+    public net.minecraft.commands.CommandFunction$CommandEntry(com.mojang.brigadier.ParseResults<net.minecraft.commands.CommandSourceStack> $$0) {
+        this.parse = $$0;
     }
 
     public void execute(net.minecraft.server.ServerFunctionManager $$0, net.minecraft.commands.CommandSourceStack $$1, java.util.Deque<net.minecraft.server.ServerFunctionManager$QueuedCommand> $$2, int $$3, int $$4, @javax.annotation.Nullable net.minecraft.server.ServerFunctionManager$TraceCallbacks $$5) throws com.mojang.brigadier.exceptions.CommandSyntaxException {