--- net/minecraft/commands/CommandFunction$FunctionEntry.java
+++ net/minecraft/commands/CommandFunction$FunctionEntry.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.commands.CommandFunction$CacheableFunction function;
 
-    public net.minecraft.commands.CommandFunction$FunctionEntry(net.minecraft.commands.CommandFunction function) {
-        this.function = new net.minecraft.commands.CommandFunction$CacheableFunction(function);
+    public net.minecraft.commands.CommandFunction$FunctionEntry(net.minecraft.commands.CommandFunction $$0) {
+        this.function = new net.minecraft.commands.CommandFunction$CacheableFunction($$0);
     }
 
     public void execute(net.minecraft.server.ServerFunctionManager $$0, net.minecraft.commands.CommandSourceStack $$1, java.util.Deque<net.minecraft.server.ServerFunctionManager$QueuedCommand> $$2, int $$3, int $$4, @javax.annotation.Nullable net.minecraft.server.ServerFunctionManager$TraceCallbacks $$52) {