--- net/minecraft/world/item/context/UseOnContext.java
+++ net/minecraft/world/item/context/UseOnContext.java
@@ -12,16 +12,16 @@
     public net.minecraft.world.level.Level level;
     public net.minecraft.world.item.ItemStack itemStack;
 
-    public net.minecraft.world.item.context.UseOnContext(net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand, net.minecraft.world.phys.BlockHitResult hitResult) {
-        this(player.level, player, hand, player.getItemInHand(hand), hitResult);
+    public net.minecraft.world.item.context.UseOnContext(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.InteractionHand $$1, net.minecraft.world.phys.BlockHitResult $$2) {
+        this($$0.level, $$0, $$1, $$0.getItemInHand($$1), $$2);
     }
 
-    public net.minecraft.world.item.context.UseOnContext(net.minecraft.world.level.Level level, @javax.annotation.Nullable net.minecraft.world.entity.player.Player player, net.minecraft.world.InteractionHand hand, net.minecraft.world.item.ItemStack itemStack, net.minecraft.world.phys.BlockHitResult hitResult) {
-        this.player = player;
-        this.hand = hand;
-        this.hitResult = hitResult;
-        this.itemStack = itemStack;
-        this.level = level;
+    public net.minecraft.world.item.context.UseOnContext(net.minecraft.world.level.Level $$0, @javax.annotation.Nullable net.minecraft.world.entity.player.Player $$1, net.minecraft.world.InteractionHand $$2, net.minecraft.world.item.ItemStack $$3, net.minecraft.world.phys.BlockHitResult $$4) {
+        this.player = $$1;
+        this.hand = $$2;
+        this.hitResult = $$4;
+        this.itemStack = $$3;
+        this.level = $$0;
     }
 
     public net.minecraft.world.phys.BlockHitResult getHitResult() {