--- net/minecraft/world/entity/ai/goal/BegGoal.java
+++ net/minecraft/world/entity/ai/goal/BegGoal.java
@@ -55,13 +55,13 @@
         --this.lookTime;
     }
 
-    public boolean playerHoldingInteresting(net.minecraft.world.entity.player.Player player) {
-        for (net.minecraft.world.InteractionHand interactionHand : net.minecraft.world.InteractionHand.values()) {
-            net.minecraft.world.item.ItemStack itemStack = player.getItemInHand(interactionHand);
-            if (this.wolf.isTame() && itemStack.is(net.minecraft.world.item.Items.BONE)) {
+    public boolean playerHoldingInteresting(net.minecraft.world.entity.player.Player $$0) {
+        for (net.minecraft.world.InteractionHand $$1 : net.minecraft.world.InteractionHand.values()) {
+            net.minecraft.world.item.ItemStack $$2 = $$0.getItemInHand($$1);
+            if (this.wolf.isTame() && $$2.is(net.minecraft.world.item.Items.BONE)) {
                 return true;
             }
-            if (!this.wolf.isFood(itemStack)) continue;
+            if (!this.wolf.isFood($$2)) continue;
             return true;
         }
         return false;