--- net/minecraft/world/entity/ai/goal/CatLieOnBedGoal.java
+++ net/minecraft/world/entity/ai/goal/CatLieOnBedGoal.java
@@ -24,7 +24,7 @@
         this.cat.setInSittingPose(false);
     }
 
-    public int nextStartTick(net.minecraft.world.entity.PathfinderMob creature) {
+    public int nextStartTick(net.minecraft.world.entity.PathfinderMob $$0) {
         return 40;
     }
 
@@ -43,8 +43,8 @@
         }
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        return level.isEmptyBlock(pos.above()) && level.getBlockState(pos).is(net.minecraft.tags.BlockTags.BEDS);
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        return $$0.isEmptyBlock($$1.above()) && $$0.getBlockState($$1).is(net.minecraft.tags.BlockTags.BEDS);
     }
 }
 