--- net/minecraft/world/entity/ai/goal/CatSitOnBlockGoal.java
+++ net/minecraft/world/entity/ai/goal/CatSitOnBlockGoal.java
@@ -32,18 +32,18 @@
         this.cat.setInSittingPose(this.isReachedTarget());
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        if (!level.isEmptyBlock(pos.above())) {
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        if (!$$0.isEmptyBlock($$1.above())) {
             return false;
         }
-        net.minecraft.world.level.block.state.BlockState blockState = level.getBlockState(pos);
-        if (blockState.is(net.minecraft.world.level.block.Blocks.CHEST)) {
-            return net.minecraft.world.level.block.entity.ChestBlockEntity.getOpenCount((net.minecraft.world.level.BlockGetter)level, (net.minecraft.core.BlockPos)pos) < 1;
+        net.minecraft.world.level.block.state.BlockState $$2 = $$0.getBlockState($$1);
+        if ($$2.is(net.minecraft.world.level.block.Blocks.CHEST)) {
+            return net.minecraft.world.level.block.entity.ChestBlockEntity.getOpenCount((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$1) < 1;
         }
-        if (blockState.is(net.minecraft.world.level.block.Blocks.FURNACE) && ((java.lang.Boolean)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.FurnaceBlock.LIT)).booleanValue()) {
+        if ($$2.is(net.minecraft.world.level.block.Blocks.FURNACE) && ((java.lang.Boolean)$$2.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.FurnaceBlock.LIT)).booleanValue()) {
             return true;
         }
-        return blockState.is(net.minecraft.tags.BlockTags.BEDS, $$02 -> (java.lang.Boolean)$$02.getOptionalValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BedBlock.PART).map($$0 -> $$0 != net.minecraft.world.level.block.state.properties.BedPart.HEAD).orElse((java.lang.Object)true));
+        return $$2.is(net.minecraft.tags.BlockTags.BEDS, $$02 -> (java.lang.Boolean)$$02.getOptionalValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BedBlock.PART).map($$0 -> $$0 != net.minecraft.world.level.block.state.properties.BedPart.HEAD).orElse((java.lang.Object)true));
     }
 }
 