--- net/minecraft/world/entity/ai/goal/DolphinJumpGoal.java
+++ net/minecraft/world/entity/ai/goal/DolphinJumpGoal.java
@@ -31,13 +31,13 @@
         return true;
     }
 
-    public boolean waterIsClear(net.minecraft.core.BlockPos pos, int dx, int dz, int scale) {
-        net.minecraft.core.BlockPos blockPos = pos.offset(dx * scale, 0, dz * scale);
-        return this.dolphin.level.getFluidState(blockPos).is(net.minecraft.tags.FluidTags.WATER) && !this.dolphin.level.getBlockState(blockPos).getMaterial().blocksMotion();
+    public boolean waterIsClear(net.minecraft.core.BlockPos $$0, int $$1, int $$2, int $$3) {
+        net.minecraft.core.BlockPos $$4 = $$0.offset($$1 * $$3, 0, $$2 * $$3);
+        return this.dolphin.level.getFluidState($$4).is(net.minecraft.tags.FluidTags.WATER) && !this.dolphin.level.getBlockState($$4).getMaterial().blocksMotion();
     }
 
-    public boolean surfaceIsClear(net.minecraft.core.BlockPos pos, int dx, int dz, int scale) {
-        return this.dolphin.level.getBlockState(pos.offset(dx * scale, 1, dz * scale)).isAir() && this.dolphin.level.getBlockState(pos.offset(dx * scale, 2, dz * scale)).isAir();
+    public boolean surfaceIsClear(net.minecraft.core.BlockPos $$0, int $$1, int $$2, int $$3) {
+        return this.dolphin.level.getBlockState($$0.offset($$1 * $$3, 1, $$2 * $$3)).isAir() && this.dolphin.level.getBlockState($$0.offset($$1 * $$3, 2, $$2 * $$3)).isAir();
     }
 
     public boolean canContinueToUse() {