--- net/minecraft/world/entity/ai/goal/DoorInteractGoal.java
+++ net/minecraft/world/entity/ai/goal/DoorInteractGoal.java
@@ -13,10 +13,10 @@
     public float doorOpenDirX;
     public float doorOpenDirZ;
 
-    public net.minecraft.world.entity.ai.goal.DoorInteractGoal(net.minecraft.world.entity.Mob mob) {
+    public net.minecraft.world.entity.ai.goal.DoorInteractGoal(net.minecraft.world.entity.Mob $$0) {
         this.doorPos = net.minecraft.core.BlockPos.ZERO;
-        this.mob = mob;
-        if (!net.minecraft.world.entity.ai.util.GoalUtils.hasGroundPathNavigation((net.minecraft.world.entity.Mob)mob)) {
+        this.mob = $$0;
+        if (!net.minecraft.world.entity.ai.util.GoalUtils.hasGroundPathNavigation((net.minecraft.world.entity.Mob)$$0)) {
             throw new java.lang.IllegalArgumentException("Unsupported mob type for DoorInteractGoal");
         }
     }
@@ -33,10 +33,10 @@
         return (java.lang.Boolean)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DoorBlock.OPEN);
     }
 
-    public void setOpen(boolean open) {
-        net.minecraft.world.level.block.state.BlockState blockState;
-        if (this.hasDoor && (blockState = this.mob.level.getBlockState(this.doorPos)).getBlock() instanceof net.minecraft.world.level.block.DoorBlock) {
-            ((net.minecraft.world.level.block.DoorBlock)blockState.getBlock()).setOpen((net.minecraft.world.entity.Entity)this.mob, this.mob.level, blockState, this.doorPos, open);
+    public void setOpen(boolean $$0) {
+        net.minecraft.world.level.block.state.BlockState $$1;
+        if (this.hasDoor && ($$1 = this.mob.level.getBlockState(this.doorPos)).getBlock() instanceof net.minecraft.world.level.block.DoorBlock) {
+            ((net.minecraft.world.level.block.DoorBlock)$$1.getBlock()).setOpen((net.minecraft.world.entity.Entity)this.mob, this.mob.level, $$1, this.doorPos, $$0);
         }
     }
 