--- net/minecraft/world/entity/ai/goal/FollowFlockLeaderGoal.java
+++ net/minecraft/world/entity/ai/goal/FollowFlockLeaderGoal.java
@@ -16,8 +16,8 @@
         this.nextStartTick = this.nextStartTick($$0);
     }
 
-    public int nextStartTick(net.minecraft.world.entity.animal.AbstractSchoolingFish taskOwner) {
-        return net.minecraft.world.entity.ai.goal.FollowFlockLeaderGoal.reducedTickDelay((int)(200 + taskOwner.getRandom().nextInt(200) % 20));
+    public int nextStartTick(net.minecraft.world.entity.animal.AbstractSchoolingFish $$0) {
+        return net.minecraft.world.entity.ai.goal.FollowFlockLeaderGoal.reducedTickDelay((int)(200 + $$0.getRandom().nextInt(200) % 20));
     }
 
     public boolean canUse() {