--- net/minecraft/world/entity/ai/goal/GoalSelector.java
+++ net/minecraft/world/entity/ai/goal/GoalSelector.java
@@ -22,8 +22,8 @@
         this.profiler = $$0;
     }
 
-    public void addGoal(int priority, net.minecraft.world.entity.ai.goal.Goal task) {
-        this.availableGoals.add((java.lang.Object)new net.minecraft.world.entity.ai.goal.WrappedGoal(priority, task));
+    public void addGoal(int $$0, net.minecraft.world.entity.ai.goal.Goal $$1) {
+        this.availableGoals.add((java.lang.Object)new net.minecraft.world.entity.ai.goal.WrappedGoal($$0, $$1));
     }
 
     @com.google.common.annotations.VisibleForTesting
@@ -31,9 +31,9 @@
         this.availableGoals.clear();
     }
 
-    public void removeGoal(net.minecraft.world.entity.ai.goal.Goal task) {
-        this.availableGoals.stream().filter($$1 -> $$1.getGoal() == task).filter(net.minecraft.world.entity.ai.goal.WrappedGoal::isRunning).forEach(net.minecraft.world.entity.ai.goal.WrappedGoal::stop);
-        this.availableGoals.removeIf($$1 -> $$1.getGoal() == task);
+    public void removeGoal(net.minecraft.world.entity.ai.goal.Goal $$0) {
+        this.availableGoals.stream().filter($$1 -> $$1.getGoal() == $$0).filter(net.minecraft.world.entity.ai.goal.WrappedGoal::isRunning).forEach(net.minecraft.world.entity.ai.goal.WrappedGoal::stop);
+        this.availableGoals.removeIf($$1 -> $$1.getGoal() == $$0);
     }
 
     public static boolean goalContainsAnyFlags(net.minecraft.world.entity.ai.goal.WrappedGoal $$0, java.util.EnumSet<net.minecraft.world.entity.ai.goal.Goal$Flag> $$1) {
@@ -102,19 +102,19 @@
         this.newGoalRate = $$0;
     }
 
-    public void disableControlFlag(net.minecraft.world.entity.ai.goal.Goal$Flag flag) {
-        this.disabledFlags.add((java.lang.Object)flag);
+    public void disableControlFlag(net.minecraft.world.entity.ai.goal.Goal$Flag $$0) {
+        this.disabledFlags.add((java.lang.Object)$$0);
     }
 
-    public void enableControlFlag(net.minecraft.world.entity.ai.goal.Goal$Flag flag) {
-        this.disabledFlags.remove((java.lang.Object)flag);
+    public void enableControlFlag(net.minecraft.world.entity.ai.goal.Goal$Flag $$0) {
+        this.disabledFlags.remove((java.lang.Object)$$0);
     }
 
-    public void setControlFlag(net.minecraft.world.entity.ai.goal.Goal$Flag flag, boolean enabled) {
-        if (enabled) {
-            this.enableControlFlag(flag);
+    public void setControlFlag(net.minecraft.world.entity.ai.goal.Goal$Flag $$0, boolean $$1) {
+        if ($$1) {
+            this.enableControlFlag($$0);
         } else {
-            this.disableControlFlag(flag);
+            this.disableControlFlag($$0);
         }
     }
 