--- net/minecraft/world/entity/ai/goal/GolemRandomStrollInVillageGoal.java
+++ net/minecraft/world/entity/ai/goal/GolemRandomStrollInVillageGoal.java
@@ -77,18 +77,18 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.core.BlockPos getRandomPoiWithinSection(net.minecraft.core.SectionPos sectionPos) {
-        net.minecraft.server.level.ServerLevel serverLevel = (net.minecraft.server.level.ServerLevel)this.mob.level;
-        net.minecraft.world.entity.ai.village.poi.PoiManager poiManager = serverLevel.getPoiManager();
-        java.util.List list = (java.util.List)poiManager.getInRange($$0 -> true, sectionPos.center(), 8, net.minecraft.world.entity.ai.village.poi.PoiManager$Occupancy.IS_OCCUPIED).map(net.minecraft.world.entity.ai.village.poi.PoiRecord::getPos).collect(java.util.stream.Collectors.toList());
-        if (list.isEmpty()) {
+    public net.minecraft.core.BlockPos getRandomPoiWithinSection(net.minecraft.core.SectionPos $$02) {
+        net.minecraft.server.level.ServerLevel $$1 = (net.minecraft.server.level.ServerLevel)this.mob.level;
+        net.minecraft.world.entity.ai.village.poi.PoiManager $$2 = $$1.getPoiManager();
+        java.util.List $$3 = (java.util.List)$$2.getInRange($$0 -> true, $$02.center(), 8, net.minecraft.world.entity.ai.village.poi.PoiManager$Occupancy.IS_OCCUPIED).map(net.minecraft.world.entity.ai.village.poi.PoiRecord::getPos).collect(java.util.stream.Collectors.toList());
+        if ($$3.isEmpty()) {
             return null;
         }
-        return (net.minecraft.core.BlockPos)list.get(serverLevel.random.nextInt(list.size()));
+        return (net.minecraft.core.BlockPos)$$3.get($$1.random.nextInt($$3.size()));
     }
 
-    public boolean doesVillagerWantGolem(net.minecraft.world.entity.npc.Villager villager) {
-        return villager.wantsToSpawnGolem(this.mob.level.getGameTime());
+    public boolean doesVillagerWantGolem(net.minecraft.world.entity.npc.Villager $$0) {
+        return $$0.wantsToSpawnGolem(this.mob.level.getGameTime());
     }
 }
 