--- net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
+++ net/minecraft/world/entity/ai/goal/LlamaFollowCaravanGoal.java
@@ -96,15 +96,15 @@
         this.llama.getNavigation().moveTo(this.llama.getX() + $$3.x, this.llama.getY() + $$3.y, this.llama.getZ() + $$3.z, this.speedModifier);
     }
 
-    public boolean firstIsLeashed(net.minecraft.world.entity.animal.horse.Llama llama, int leashedQueuePosition) {
-        if (leashedQueuePosition > 8) {
+    public boolean firstIsLeashed(net.minecraft.world.entity.animal.horse.Llama $$0, int $$1) {
+        if ($$1 > 8) {
             return false;
         }
-        if (llama.inCaravan()) {
-            if (llama.getCaravanHead().isLeashed()) {
+        if ($$0.inCaravan()) {
+            if ($$0.getCaravanHead().isLeashed()) {
                 return true;
             }
-            return this.firstIsLeashed(llama.getCaravanHead(), ++leashedQueuePosition);
+            return this.firstIsLeashed($$0.getCaravanHead(), ++$$1);
         }
         return false;
     }