--- net/minecraft/world/entity/ai/goal/MeleeAttackGoal.java
+++ net/minecraft/world/entity/ai/goal/MeleeAttackGoal.java
@@ -111,12 +111,12 @@
         this.checkAndPerformAttack($$0, $$1);
     }
 
-    public void checkAndPerformAttack(net.minecraft.world.entity.LivingEntity enemy, double distToEnemySqr) {
-        double d = this.getAttackReachSqr(enemy);
-        if (distToEnemySqr <= d && this.ticksUntilNextAttack <= 0) {
+    public void checkAndPerformAttack(net.minecraft.world.entity.LivingEntity $$0, double $$1) {
+        double $$2 = this.getAttackReachSqr($$0);
+        if ($$1 <= $$2 && this.ticksUntilNextAttack <= 0) {
             this.resetAttackCooldown();
             this.mob.swing(net.minecraft.world.InteractionHand.MAIN_HAND);
-            this.mob.doHurtTarget((net.minecraft.world.entity.Entity)enemy);
+            this.mob.doHurtTarget((net.minecraft.world.entity.Entity)$$0);
         }
     }
 
@@ -136,8 +136,8 @@
         return this.adjustedTickDelay(20);
     }
 
-    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity attackTarget) {
-        return this.mob.getBbWidth() * 2.0f * (this.mob.getBbWidth() * 2.0f) + attackTarget.getBbWidth();
+    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity $$0) {
+        return this.mob.getBbWidth() * 2.0f * (this.mob.getBbWidth() * 2.0f) + $$0.getBbWidth();
     }
 }
 