--- net/minecraft/world/entity/ai/goal/MoveThroughVillageGoal.java
+++ net/minecraft/world/entity/ai/goal/MoveThroughVillageGoal.java
@@ -101,9 +101,9 @@
         }
     }
 
-    public boolean hasNotVisited(net.minecraft.core.BlockPos pos) {
-        for (net.minecraft.core.BlockPos blockPos : this.visited) {
-            if (!java.util.Objects.equals((java.lang.Object)pos, (java.lang.Object)blockPos)) continue;
+    public boolean hasNotVisited(net.minecraft.core.BlockPos $$0) {
+        for (net.minecraft.core.BlockPos $$1 : this.visited) {
+            if (!java.util.Objects.equals((java.lang.Object)$$0, (java.lang.Object)$$1)) continue;
             return false;
         }
         return true;