--- net/minecraft/world/entity/ai/goal/PathfindToRaidGoal.java
+++ net/minecraft/world/entity/ai/goal/PathfindToRaidGoal.java
@@ -38,13 +38,13 @@
         }
     }
 
-    public void recruitNearby(net.minecraft.world.entity.raid.Raid raid) {
-        if (raid.isActive()) {
-            java.util.HashSet hashSet = com.google.common.collect.Sets.newHashSet();
-            java.util.List list = this.mob.level.getEntitiesOfClass(net.minecraft.world.entity.raid.Raider.class, this.mob.getBoundingBox().inflate(16.0), $$1 -> !$$1.hasActiveRaid() && net.minecraft.world.entity.raid.Raids.canJoinRaid((net.minecraft.world.entity.raid.Raider)$$1, (net.minecraft.world.entity.raid.Raid)raid));
-            hashSet.addAll((java.util.Collection)list);
-            for (net.minecraft.world.entity.raid.Raider raider : hashSet) {
-                raid.joinRaid(raid.getGroupsSpawned(), raider, null, true);
+    public void recruitNearby(net.minecraft.world.entity.raid.Raid $$0) {
+        if ($$0.isActive()) {
+            java.util.HashSet $$12 = com.google.common.collect.Sets.newHashSet();
+            java.util.List $$2 = this.mob.level.getEntitiesOfClass(net.minecraft.world.entity.raid.Raider.class, this.mob.getBoundingBox().inflate(16.0), $$1 -> !$$1.hasActiveRaid() && net.minecraft.world.entity.raid.Raids.canJoinRaid((net.minecraft.world.entity.raid.Raider)$$1, (net.minecraft.world.entity.raid.Raid)$$0));
+            $$12.addAll((java.util.Collection)$$2);
+            for (net.minecraft.world.entity.raid.Raider $$3 : $$12) {
+                $$0.joinRaid($$0.getGroupsSpawned(), $$3, null, true);
             }
         }
     }