--- net/minecraft/world/entity/ai/goal/RangedBowAttackGoal.java
+++ net/minecraft/world/entity/ai/goal/RangedBowAttackGoal.java
@@ -26,8 +26,8 @@
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE, (java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.LOOK));
     }
 
-    public void setMinAttackInterval(int attackCooldown) {
-        this.attackIntervalMin = attackCooldown;
+    public void setMinAttackInterval(int $$0) {
+        this.attackIntervalMin = $$0;
     }
 
     public boolean canUse() {