--- net/minecraft/world/entity/ai/goal/RangedAttackGoal.java
+++ net/minecraft/world/entity/ai/goal/RangedAttackGoal.java
@@ -18,8 +18,8 @@
     public float attackRadius;
     public float attackRadiusSqr;
 
-    public net.minecraft.world.entity.ai.goal.RangedAttackGoal(net.minecraft.world.entity.monster.RangedAttackMob rangedAttackMob, double speedModifier, int attackInterval, float attackRadius) {
-        this(rangedAttackMob, speedModifier, attackInterval, attackInterval, attackRadius);
+    public net.minecraft.world.entity.ai.goal.RangedAttackGoal(net.minecraft.world.entity.monster.RangedAttackMob $$0, double $$1, int $$2, float $$3) {
+        this($$0, $$1, $$2, $$2, $$3);
     }
 
     public net.minecraft.world.entity.ai.goal.RangedAttackGoal(net.minecraft.world.entity.monster.RangedAttackMob $$0, double $$1, int $$2, int $$3, float $$4) {