--- net/minecraft/world/entity/ai/goal/WrappedGoal.java
+++ net/minecraft/world/entity/ai/goal/WrappedGoal.java
@@ -15,8 +15,8 @@
         this.goal = $$1;
     }
 
-    public boolean canBeReplacedBy(net.minecraft.world.entity.ai.goal.WrappedGoal other) {
-        return this.isInterruptable() && other.getPriority() < this.getPriority();
+    public boolean canBeReplacedBy(net.minecraft.world.entity.ai.goal.WrappedGoal $$0) {
+        return this.isInterruptable() && $$0.getPriority() < this.getPriority();
     }
 
     public boolean canUse() {
@@ -59,8 +59,8 @@
         this.goal.tick();
     }
 
-    public void setFlags(java.util.EnumSet<net.minecraft.world.entity.ai.goal.Goal$Flag> flagSet) {
-        this.goal.setFlags(flagSet);
+    public void setFlags(java.util.EnumSet<net.minecraft.world.entity.ai.goal.Goal$Flag> $$0) {
+        this.goal.setFlags($$0);
     }
 
     public java.util.EnumSet<net.minecraft.world.entity.ai.goal.Goal$Flag> getFlags() {