--- net/minecraft/client/model/geom/LayerDefinitions.java
+++ net/minecraft/client/model/geom/LayerDefinitions.java
@@ -47,6 +47,7 @@
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.BOOK, (java.lang.Object)net.minecraft.client.model.BookModel.createBodyLayer());
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.CAT, (java.lang.Object)$$10);
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.CAT_COLLAR, (java.lang.Object)net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)net.minecraft.client.model.OcelotModel.createBodyMesh((net.minecraft.client.model.geom.builders.CubeDeformation)new net.minecraft.client.model.geom.builders.CubeDeformation(0.01f)), (int)64, (int)32));
+        $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.CAMEL, (java.lang.Object)net.minecraft.client.model.CamelModel.createBodyLayer());
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.CAVE_SPIDER, (java.lang.Object)$$18);
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.CHEST, (java.lang.Object)net.minecraft.client.renderer.blockentity.ChestRenderer.createSingleBodyLayer());
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.DOUBLE_CHEST_LEFT, (java.lang.Object)net.minecraft.client.renderer.blockentity.ChestRenderer.createDoubleBodyLeftLayer());
@@ -191,20 +192,29 @@
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.ZOMBIFIED_PIGLIN, (java.lang.Object)$$11);
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.ZOMBIFIED_PIGLIN_INNER_ARMOR, (java.lang.Object)$$4);
         $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.ZOMBIFIED_PIGLIN_OUTER_ARMOR, (java.lang.Object)$$3);
-        net.minecraft.client.model.geom.builders.LayerDefinition $$19 = net.minecraft.client.model.BoatModel.createBodyModel((boolean)false);
-        net.minecraft.client.model.geom.builders.LayerDefinition $$20 = net.minecraft.client.model.BoatModel.createBodyModel((boolean)true);
-        for (net.minecraft.world.entity.vehicle.Boat$Type $$21 : net.minecraft.world.entity.vehicle.Boat$Type.values()) {
-            $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createBoatModelName((net.minecraft.world.entity.vehicle.Boat$Type)$$21), (java.lang.Object)$$19);
-            $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createChestBoatModelName((net.minecraft.world.entity.vehicle.Boat$Type)$$21), (java.lang.Object)$$20);
+        net.minecraft.client.model.geom.builders.LayerDefinition $$19 = net.minecraft.client.model.BoatModel.createBodyModel();
+        net.minecraft.client.model.geom.builders.LayerDefinition $$20 = net.minecraft.client.model.ChestBoatModel.createBodyModel();
+        net.minecraft.client.model.geom.builders.LayerDefinition $$21 = net.minecraft.client.model.RaftModel.createBodyModel();
+        net.minecraft.client.model.geom.builders.LayerDefinition $$222 = net.minecraft.client.model.ChestRaftModel.createBodyModel();
+        for (net.minecraft.world.entity.vehicle.Boat$Type $$23 : net.minecraft.world.entity.vehicle.Boat$Type.values()) {
+            if ($$23 == net.minecraft.world.entity.vehicle.Boat$Type.BAMBOO) {
+                $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createBoatModelName((net.minecraft.world.entity.vehicle.Boat$Type)$$23), (java.lang.Object)$$21);
+                $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createChestBoatModelName((net.minecraft.world.entity.vehicle.Boat$Type)$$23), (java.lang.Object)$$222);
+                continue;
+            }
+            $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createBoatModelName((net.minecraft.world.entity.vehicle.Boat$Type)$$23), (java.lang.Object)$$19);
+            $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createChestBoatModelName((net.minecraft.world.entity.vehicle.Boat$Type)$$23), (java.lang.Object)$$20);
         }
-        net.minecraft.client.model.geom.builders.LayerDefinition $$222 = net.minecraft.client.renderer.blockentity.SignRenderer.createSignLayer();
-        net.minecraft.world.level.block.state.properties.WoodType.values().forEach($$2 -> $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createSignModelName((net.minecraft.world.level.block.state.properties.WoodType)$$2), (java.lang.Object)$$222));
-        com.google.common.collect.ImmutableMap $$23 = $$0.build();
-        java.util.List $$24 = (java.util.List)net.minecraft.client.model.geom.ModelLayers.getKnownLocations().filter($$1 -> !$$23.containsKey($$1)).collect(java.util.stream.Collectors.toList());
-        if (!$$24.isEmpty()) {
-            throw new java.lang.IllegalStateException("Missing layer definitions: " + $$24);
+        net.minecraft.client.model.geom.builders.LayerDefinition $$24 = net.minecraft.client.renderer.blockentity.SignRenderer.createSignLayer();
+        net.minecraft.world.level.block.state.properties.WoodType.values().forEach($$2 -> $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createSignModelName((net.minecraft.world.level.block.state.properties.WoodType)$$2), (java.lang.Object)$$24));
+        net.minecraft.client.model.geom.builders.LayerDefinition $$25 = net.minecraft.client.renderer.blockentity.HangingSignRenderer.createHangingSignLayer();
+        net.minecraft.world.level.block.state.properties.WoodType.values().forEach($$2 -> $$0.put((java.lang.Object)net.minecraft.client.model.geom.ModelLayers.createHangingSignModelName((net.minecraft.world.level.block.state.properties.WoodType)$$2), (java.lang.Object)$$25));
+        com.google.common.collect.ImmutableMap $$26 = $$0.build();
+        java.util.List $$27 = (java.util.List)net.minecraft.client.model.geom.ModelLayers.getKnownLocations().filter($$1 -> !$$26.containsKey($$1)).collect(java.util.stream.Collectors.toList());
+        if (!$$27.isEmpty()) {
+            throw new java.lang.IllegalStateException("Missing layer definitions: " + $$27);
         }
-        return $$23;
+        return $$26;
     }
 
     static {