--- net/minecraft/client/model/geom/ModelPart$Cube.java
+++ net/minecraft/client/model/geom/ModelPart$Cube.java
@@ -13,51 +13,51 @@
     public float maxY;
     public float maxZ;
 
-    public net.minecraft.client.model.geom.ModelPart$Cube(int texCoordU, int texCoordV, float originX, float originY, float originZ, float dimensionX, float dimensionY, float dimensionZ, float growX, float growY, float growZ, boolean mirror, float texWidthScaled, float texHeightScaled) {
-        this.minX = originX;
-        this.minY = originY;
-        this.minZ = originZ;
-        this.maxX = originX + dimensionX;
-        this.maxY = originY + dimensionY;
-        this.maxZ = originZ + dimensionZ;
+    public net.minecraft.client.model.geom.ModelPart$Cube(int $$0, int $$1, float $$2, float $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9, float $$10, boolean $$11, float $$12, float $$13) {
+        this.minX = $$2;
+        this.minY = $$3;
+        this.minZ = $$4;
+        this.maxX = $$2 + $$5;
+        this.maxY = $$3 + $$6;
+        this.maxZ = $$4 + $$7;
         this.polygons = new net.minecraft.client.model.geom.ModelPart$Polygon[6];
-        float f = originX + dimensionX;
-        float f2 = originY + dimensionY;
-        float f3 = originZ + dimensionZ;
-        originX -= growX;
-        originY -= growY;
-        originZ -= growZ;
-        f += growX;
-        f2 += growY;
-        f3 += growZ;
-        if (mirror) {
-            float f4 = f;
-            f = originX;
-            originX = f4;
+        float $$14 = $$2 + $$5;
+        float $$15 = $$3 + $$6;
+        float $$16 = $$4 + $$7;
+        $$2 -= $$8;
+        $$3 -= $$9;
+        $$4 -= $$10;
+        $$14 += $$8;
+        $$15 += $$9;
+        $$16 += $$10;
+        if ($$11) {
+            float $$17 = $$14;
+            $$14 = $$2;
+            $$2 = $$17;
         }
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex = new net.minecraft.client.model.geom.ModelPart$Vertex(originX, originY, originZ, 0.0f, 0.0f);
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex2 = new net.minecraft.client.model.geom.ModelPart$Vertex(f, originY, originZ, 0.0f, 8.0f);
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex3 = new net.minecraft.client.model.geom.ModelPart$Vertex(f, f2, originZ, 8.0f, 8.0f);
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex4 = new net.minecraft.client.model.geom.ModelPart$Vertex(originX, f2, originZ, 8.0f, 0.0f);
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex5 = new net.minecraft.client.model.geom.ModelPart$Vertex(originX, originY, f3, 0.0f, 0.0f);
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex6 = new net.minecraft.client.model.geom.ModelPart$Vertex(f, originY, f3, 0.0f, 8.0f);
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex7 = new net.minecraft.client.model.geom.ModelPart$Vertex(f, f2, f3, 8.0f, 8.0f);
-        net.minecraft.client.model.geom.ModelPart$Vertex vertex8 = new net.minecraft.client.model.geom.ModelPart$Vertex(originX, f2, f3, 8.0f, 0.0f);
-        float f5 = texCoordU;
-        float f6 = (float)texCoordU + dimensionZ;
-        float f7 = (float)texCoordU + dimensionZ + dimensionX;
-        float f8 = (float)texCoordU + dimensionZ + dimensionX + dimensionX;
-        float f9 = (float)texCoordU + dimensionZ + dimensionX + dimensionZ;
-        float f10 = (float)texCoordU + dimensionZ + dimensionX + dimensionZ + dimensionX;
-        float f11 = texCoordV;
-        float f12 = (float)texCoordV + dimensionZ;
-        float f13 = (float)texCoordV + dimensionZ + dimensionY;
-        this.polygons[2] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{vertex6, vertex5, vertex, vertex2}, f6, f11, f7, f12, texWidthScaled, texHeightScaled, mirror, net.minecraft.core.Direction.DOWN);
-        this.polygons[3] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{vertex3, vertex4, vertex8, vertex7}, f7, f12, f8, f11, texWidthScaled, texHeightScaled, mirror, net.minecraft.core.Direction.UP);
-        this.polygons[1] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{vertex, vertex5, vertex8, vertex4}, f5, f12, f6, f13, texWidthScaled, texHeightScaled, mirror, net.minecraft.core.Direction.WEST);
-        this.polygons[4] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{vertex2, vertex, vertex4, vertex3}, f6, f12, f7, f13, texWidthScaled, texHeightScaled, mirror, net.minecraft.core.Direction.NORTH);
-        this.polygons[0] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{vertex6, vertex2, vertex3, vertex7}, f7, f12, f9, f13, texWidthScaled, texHeightScaled, mirror, net.minecraft.core.Direction.EAST);
-        this.polygons[5] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{vertex5, vertex6, vertex7, vertex8}, f9, f12, f10, f13, texWidthScaled, texHeightScaled, mirror, net.minecraft.core.Direction.SOUTH);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$18 = new net.minecraft.client.model.geom.ModelPart$Vertex($$2, $$3, $$4, 0.0f, 0.0f);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$19 = new net.minecraft.client.model.geom.ModelPart$Vertex($$14, $$3, $$4, 0.0f, 8.0f);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$20 = new net.minecraft.client.model.geom.ModelPart$Vertex($$14, $$15, $$4, 8.0f, 8.0f);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$21 = new net.minecraft.client.model.geom.ModelPart$Vertex($$2, $$15, $$4, 8.0f, 0.0f);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$22 = new net.minecraft.client.model.geom.ModelPart$Vertex($$2, $$3, $$16, 0.0f, 0.0f);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$23 = new net.minecraft.client.model.geom.ModelPart$Vertex($$14, $$3, $$16, 0.0f, 8.0f);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$24 = new net.minecraft.client.model.geom.ModelPart$Vertex($$14, $$15, $$16, 8.0f, 8.0f);
+        net.minecraft.client.model.geom.ModelPart$Vertex $$25 = new net.minecraft.client.model.geom.ModelPart$Vertex($$2, $$15, $$16, 8.0f, 0.0f);
+        float $$26 = $$0;
+        float $$27 = (float)$$0 + $$7;
+        float $$28 = (float)$$0 + $$7 + $$5;
+        float $$29 = (float)$$0 + $$7 + $$5 + $$5;
+        float $$30 = (float)$$0 + $$7 + $$5 + $$7;
+        float $$31 = (float)$$0 + $$7 + $$5 + $$7 + $$5;
+        float $$32 = $$1;
+        float $$33 = (float)$$1 + $$7;
+        float $$34 = (float)$$1 + $$7 + $$6;
+        this.polygons[2] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{$$23, $$22, $$18, $$19}, $$27, $$32, $$28, $$33, $$12, $$13, $$11, net.minecraft.core.Direction.DOWN);
+        this.polygons[3] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{$$20, $$21, $$25, $$24}, $$28, $$33, $$29, $$32, $$12, $$13, $$11, net.minecraft.core.Direction.UP);
+        this.polygons[1] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{$$18, $$22, $$25, $$21}, $$26, $$33, $$27, $$34, $$12, $$13, $$11, net.minecraft.core.Direction.WEST);
+        this.polygons[4] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{$$19, $$18, $$21, $$20}, $$27, $$33, $$28, $$34, $$12, $$13, $$11, net.minecraft.core.Direction.NORTH);
+        this.polygons[0] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{$$23, $$19, $$20, $$24}, $$28, $$33, $$30, $$34, $$12, $$13, $$11, net.minecraft.core.Direction.EAST);
+        this.polygons[5] = new net.minecraft.client.model.geom.ModelPart$Polygon(new net.minecraft.client.model.geom.ModelPart$Vertex[]{$$22, $$23, $$24, $$25}, $$30, $$33, $$31, $$34, $$12, $$13, $$11, net.minecraft.core.Direction.SOUTH);
     }
 
     public void compile(com.mojang.blaze3d.vertex.PoseStack$Pose $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7) {