--- net/minecraft/stats/StatType.java
+++ net/minecraft/stats/StatType.java
@@ -11,17 +11,17 @@
     @javax.annotation.Nullable
     public net.minecraft.network.chat.Component displayName;
 
-    public net.minecraft.stats.StatType(net.minecraft.core.Registry<T> registry) {
+    public net.minecraft.stats.StatType(net.minecraft.core.Registry<T> $$0) {
         this.map = new java.util.IdentityHashMap();
-        this.registry = registry;
+        this.registry = $$0;
     }
 
-    public boolean contains(T stat) {
-        return this.map.containsKey(stat);
+    public boolean contains(T $$0) {
+        return this.map.containsKey($$0);
     }
 
-    public net.minecraft.stats.Stat<T> get(T stat2, net.minecraft.stats.StatFormatter formatter) {
-        return (net.minecraft.stats.Stat)this.map.computeIfAbsent(stat2, stat -> new net.minecraft.stats.Stat(this, stat, formatter));
+    public net.minecraft.stats.Stat<T> get(T $$0, net.minecraft.stats.StatFormatter $$12) {
+        return (net.minecraft.stats.Stat)this.map.computeIfAbsent($$0, $$1 -> new net.minecraft.stats.Stat(this, $$1, $$12));
     }
 
     public net.minecraft.core.Registry<T> getRegistry() {
@@ -32,8 +32,8 @@
         return this.map.values().iterator();
     }
 
-    public net.minecraft.stats.Stat<T> get(T stat) {
-        return this.get(stat, net.minecraft.stats.StatFormatter.DEFAULT);
+    public net.minecraft.stats.Stat<T> get(T $$0) {
+        return this.get($$0, net.minecraft.stats.StatFormatter.DEFAULT);
     }
 
     public java.lang.String getTranslationKey() {