--- net/minecraft/stats/StatsCounter.java
+++ net/minecraft/stats/StatsCounter.java
@@ -12,21 +12,21 @@
         this.stats.defaultReturnValue(0);
     }
 
-    public void increment(net.minecraft.world.entity.player.Player player, net.minecraft.stats.Stat<?> stat, int amount) {
-        int n = (int)java.lang.Math.min((long)((long)this.getValue(stat) + (long)amount), (long)java.lang.Integer.MAX_VALUE);
-        this.setValue(player, stat, n);
+    public void increment(net.minecraft.world.entity.player.Player $$0, net.minecraft.stats.Stat<?> $$1, int $$2) {
+        int $$3 = (int)java.lang.Math.min((long)((long)this.getValue($$1) + (long)$$2), (long)java.lang.Integer.MAX_VALUE);
+        this.setValue($$0, $$1, $$3);
     }
 
-    public void setValue(net.minecraft.world.entity.player.Player player, net.minecraft.stats.Stat<?> stat, int value) {
-        this.stats.put(stat, value);
+    public void setValue(net.minecraft.world.entity.player.Player $$0, net.minecraft.stats.Stat<?> $$1, int $$2) {
+        this.stats.put($$1, $$2);
     }
 
-    public <T> int getValue(net.minecraft.stats.StatType<T> type, T t) {
-        return type.contains(t) ? this.getValue(type.get(t)) : 0;
+    public <T> int getValue(net.minecraft.stats.StatType<T> $$0, T $$1) {
+        return $$0.contains($$1) ? this.getValue($$0.get($$1)) : 0;
     }
 
-    public int getValue(net.minecraft.stats.Stat<?> stat) {
-        return this.stats.getInt(stat);
+    public int getValue(net.minecraft.stats.Stat<?> $$0) {
+        return this.stats.getInt($$0);
     }
 }
 