--- net/minecraft/stats/Stat.java
+++ net/minecraft/stats/Stat.java
@@ -10,19 +10,19 @@
     public T value;
     public net.minecraft.stats.StatType<T> type;
 
-    public net.minecraft.stats.Stat(net.minecraft.stats.StatType<T> type, T value, net.minecraft.stats.StatFormatter formatter) {
-        super(net.minecraft.stats.Stat.buildName(type, value));
-        this.type = type;
-        this.formatter = formatter;
-        this.value = value;
+    public net.minecraft.stats.Stat(net.minecraft.stats.StatType<T> $$0, T $$1, net.minecraft.stats.StatFormatter $$2) {
+        super(net.minecraft.stats.Stat.buildName($$0, $$1));
+        this.type = $$0;
+        this.formatter = $$2;
+        this.value = $$1;
     }
 
-    public static <T> java.lang.String buildName(net.minecraft.stats.StatType<T> type, T value) {
-        return net.minecraft.stats.Stat.locationToKey((net.minecraft.resources.ResourceLocation)net.minecraft.core.Registry.STAT_TYPE.getKey(type)) + ":" + net.minecraft.stats.Stat.locationToKey((net.minecraft.resources.ResourceLocation)type.getRegistry().getKey(value));
+    public static <T> java.lang.String buildName(net.minecraft.stats.StatType<T> $$0, T $$1) {
+        return net.minecraft.stats.Stat.locationToKey((net.minecraft.resources.ResourceLocation)net.minecraft.core.Registry.STAT_TYPE.getKey($$0)) + ":" + net.minecraft.stats.Stat.locationToKey((net.minecraft.resources.ResourceLocation)$$0.getRegistry().getKey($$1));
     }
 
-    public static <T> java.lang.String locationToKey(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation id) {
-        return id.toString().replace(':', '.');
+    public static <T> java.lang.String locationToKey(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation $$0) {
+        return $$0.toString().replace(':', '.');
     }
 
     public net.minecraft.stats.StatType<T> getType() {
@@ -33,12 +33,12 @@
         return (T)this.value;
     }
 
-    public java.lang.String format(int number) {
-        return this.formatter.format(number);
+    public java.lang.String format(int $$0) {
+        return this.formatter.format($$0);
     }
 
-    public boolean equals(java.lang.Object other) {
-        return this == other || other instanceof net.minecraft.stats.Stat && java.util.Objects.equals((java.lang.Object)this.getName(), (java.lang.Object)((net.minecraft.stats.Stat)other).getName());
+    public boolean equals(java.lang.Object $$0) {
+        return this == $$0 || $$0 instanceof net.minecraft.stats.Stat && java.util.Objects.equals((java.lang.Object)this.getName(), (java.lang.Object)((net.minecraft.stats.Stat)$$0).getName());
     }
 
     public int hashCode() {