--- net/minecraft/stats/Stats.java
+++ net/minecraft/stats/Stats.java
@@ -93,15 +93,15 @@
     public net.minecraft.stats.Stats() {
     }
 
-    public static net.minecraft.resources.ResourceLocation makeCustomStat(java.lang.String key, net.minecraft.stats.StatFormatter formatter) {
-        net.minecraft.resources.ResourceLocation resourceLocation = new net.minecraft.resources.ResourceLocation(key);
-        net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.CUSTOM_STAT, (java.lang.String)key, (java.lang.Object)resourceLocation);
-        CUSTOM.get((java.lang.Object)resourceLocation, formatter);
-        return resourceLocation;
+    public static net.minecraft.resources.ResourceLocation makeCustomStat(java.lang.String $$0, net.minecraft.stats.StatFormatter $$1) {
+        net.minecraft.resources.ResourceLocation $$2 = new net.minecraft.resources.ResourceLocation($$0);
+        net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.CUSTOM_STAT, (java.lang.String)$$0, (java.lang.Object)$$2);
+        CUSTOM.get((java.lang.Object)$$2, $$1);
+        return $$2;
     }
 
-    public static <T> net.minecraft.stats.StatType<T> makeRegistryStatType(java.lang.String key, net.minecraft.core.Registry<T> registry) {
-        return (net.minecraft.stats.StatType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.STAT_TYPE, (java.lang.String)key, (java.lang.Object)new net.minecraft.stats.StatType(registry));
+    public static <T> net.minecraft.stats.StatType<T> makeRegistryStatType(java.lang.String $$0, net.minecraft.core.Registry<T> $$1) {
+        return (net.minecraft.stats.StatType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.STAT_TYPE, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.stats.StatType($$1));
     }
 
     static {