--- net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
+++ net/minecraft/world/entity/projectile/AbstractHurtingProjectile.java
@@ -14,33 +14,33 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.AbstractHurtingProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.AbstractHurtingProjectile> entityType, double x, double y, double z, double offsetX, double offsetY, double offsetZ, net.minecraft.world.level.Level level) {
-        this(entityType, level);
-        this.moveTo(x, y, z, this.getYRot(), this.getXRot());
+    public net.minecraft.world.entity.projectile.AbstractHurtingProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.AbstractHurtingProjectile> $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.world.level.Level $$7) {
+        this($$0, $$7);
+        this.moveTo($$1, $$2, $$3, this.getYRot(), this.getXRot());
         this.reapplyPosition();
-        double d = java.lang.Math.sqrt((double)(offsetX * offsetX + offsetY * offsetY + offsetZ * offsetZ));
-        if (d != 0.0) {
-            this.xPower = offsetX / d * 0.1;
-            this.yPower = offsetY / d * 0.1;
-            this.zPower = offsetZ / d * 0.1;
+        double $$8 = java.lang.Math.sqrt((double)($$4 * $$4 + $$5 * $$5 + $$6 * $$6));
+        if ($$8 != 0.0) {
+            this.xPower = $$4 / $$8 * 0.1;
+            this.yPower = $$5 / $$8 * 0.1;
+            this.zPower = $$6 / $$8 * 0.1;
         }
     }
 
-    public net.minecraft.world.entity.projectile.AbstractHurtingProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.AbstractHurtingProjectile> entityType, net.minecraft.world.entity.LivingEntity shooter, double offsetX, double offsetY, double offsetZ, net.minecraft.world.level.Level level) {
-        this(entityType, shooter.getX(), shooter.getY(), shooter.getZ(), offsetX, offsetY, offsetZ, level);
-        this.setOwner((net.minecraft.world.entity.Entity)shooter);
-        this.setRot(shooter.getYRot(), shooter.getXRot());
+    public net.minecraft.world.entity.projectile.AbstractHurtingProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.AbstractHurtingProjectile> $$0, net.minecraft.world.entity.LivingEntity $$1, double $$2, double $$3, double $$4, net.minecraft.world.level.Level $$5) {
+        this($$0, $$1.getX(), $$1.getY(), $$1.getZ(), $$2, $$3, $$4, $$5);
+        this.setOwner((net.minecraft.world.entity.Entity)$$1);
+        this.setRot($$1.getYRot(), $$1.getXRot());
     }
 
     public void defineSynchedData() {
     }
 
-    public boolean shouldRenderAtSqrDistance(double distance) {
-        double d = this.getBoundingBox().getSize() * 4.0;
-        if (java.lang.Double.isNaN((double)d)) {
-            d = 4.0;
+    public boolean shouldRenderAtSqrDistance(double $$0) {
+        double $$1 = this.getBoundingBox().getSize() * 4.0;
+        if (java.lang.Double.isNaN((double)$$1)) {
+            $$1 = 4.0;
         }
-        return distance < (d *= 64.0) * d;
+        return $$0 < ($$1 *= 64.0) * $$1;
     }
 
     public void tick() {
@@ -76,8 +76,8 @@
         this.setPos($$3, $$4, $$5);
     }
 
-    public boolean canHitEntity(net.minecraft.world.entity.Entity target) {
-        return super.canHitEntity(target) && !target.noPhysics;
+    public boolean canHitEntity(net.minecraft.world.entity.Entity $$0) {
+        return super.canHitEntity($$0) && !$$0.noPhysics;
     }
 
     public boolean shouldBurn() {
@@ -92,18 +92,18 @@
         return 0.95f;
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.put("power", (net.minecraft.nbt.Tag)this.newDoubleList(new double[]{this.xPower, this.yPower, this.zPower}));
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.put("power", (net.minecraft.nbt.Tag)this.newDoubleList(new double[]{this.xPower, this.yPower, this.zPower}));
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        net.minecraft.nbt.ListTag listTag;
-        super.readAdditionalSaveData(compound);
-        if (compound.contains("power", 9) && (listTag = compound.getList("power", 6)).size() == 3) {
-            this.xPower = listTag.getDouble(0);
-            this.yPower = listTag.getDouble(1);
-            this.zPower = listTag.getDouble(2);
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        net.minecraft.nbt.ListTag $$1;
+        super.readAdditionalSaveData($$0);
+        if ($$0.contains("power", 9) && ($$1 = $$0.getList("power", 6)).size() == 3) {
+            this.xPower = $$1.getDouble(0);
+            this.yPower = $$1.getDouble(1);
+            this.zPower = $$1.getDouble(2);
         }
     }
 
@@ -115,20 +115,20 @@
         return 1.0f;
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
-        if (this.isInvulnerableTo(source)) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
+        if (this.isInvulnerableTo($$0)) {
             return false;
         }
         this.markHurt();
-        net.minecraft.world.entity.Entity entity = source.getEntity();
-        if (entity != null) {
+        net.minecraft.world.entity.Entity $$2 = $$0.getEntity();
+        if ($$2 != null) {
             if (!this.level.isClientSide) {
-                net.minecraft.world.phys.Vec3 vec3 = entity.getLookAngle();
-                this.setDeltaMovement(vec3);
-                this.xPower = vec3.x * 0.1;
-                this.yPower = vec3.y * 0.1;
-                this.zPower = vec3.z * 0.1;
-                this.setOwner(entity);
+                net.minecraft.world.phys.Vec3 $$3 = $$2.getLookAngle();
+                this.setDeltaMovement($$3);
+                this.xPower = $$3.x * 0.1;
+                this.yPower = $$3.y * 0.1;
+                this.zPower = $$3.z * 0.1;
+                this.setOwner($$2);
             }
             return true;
         }
@@ -139,7 +139,7 @@
         return 1.0f;
     }
 
-    public net.minecraft.network.protocol.Packet<?> getAddEntityPacket() {
+    public net.minecraft.network.protocol.Packet<net.minecraft.network.protocol.game.ClientGamePacketListener> getAddEntityPacket() {
         net.minecraft.world.entity.Entity $$0 = this.getOwner();
         int $$1 = $$0 == null ? 0 : $$0.getId();
         return new net.minecraft.network.protocol.game.ClientboundAddEntityPacket(this.getId(), this.getUUID(), this.getX(), this.getY(), this.getZ(), this.getXRot(), this.getYRot(), this.getType(), $$1, new net.minecraft.world.phys.Vec3(this.xPower, this.yPower, this.zPower), 0.0);