--- net/minecraft/world/entity/projectile/Arrow.java
+++ net/minecraft/world/entity/projectile/Arrow.java
@@ -20,44 +20,44 @@
         this.effects = com.google.common.collect.Sets.newHashSet();
     }
 
-    public net.minecraft.world.entity.projectile.Arrow(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.ARROW, x, y, z, level);
+    public net.minecraft.world.entity.projectile.Arrow(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.ARROW, $$1, $$2, $$3, $$0);
         this.potion = net.minecraft.world.item.alchemy.Potions.EMPTY;
         this.effects = com.google.common.collect.Sets.newHashSet();
     }
 
-    public net.minecraft.world.entity.projectile.Arrow(net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity shooter) {
-        super(net.minecraft.world.entity.EntityType.ARROW, shooter, level);
+    public net.minecraft.world.entity.projectile.Arrow(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        super(net.minecraft.world.entity.EntityType.ARROW, $$1, $$0);
         this.potion = net.minecraft.world.item.alchemy.Potions.EMPTY;
         this.effects = com.google.common.collect.Sets.newHashSet();
     }
 
-    public void setEffectsFromItem(net.minecraft.world.item.ItemStack stack) {
-        if (stack.is(net.minecraft.world.item.Items.TIPPED_ARROW)) {
-            int n;
-            this.potion = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)stack);
-            java.util.List list = net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.world.item.ItemStack)stack);
-            if (!list.isEmpty()) {
-                for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : list) {
-                    this.effects.add((java.lang.Object)new net.minecraft.world.effect.MobEffectInstance(mobEffectInstance));
+    public void setEffectsFromItem(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0.is(net.minecraft.world.item.Items.TIPPED_ARROW)) {
+            int $$3;
+            this.potion = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)$$0);
+            java.util.List $$1 = net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.world.item.ItemStack)$$0);
+            if (!$$1.isEmpty()) {
+                for (net.minecraft.world.effect.MobEffectInstance $$2 : $$1) {
+                    this.effects.add((java.lang.Object)new net.minecraft.world.effect.MobEffectInstance($$2));
                 }
             }
-            if ((n = net.minecraft.world.entity.projectile.Arrow.getCustomColor((net.minecraft.world.item.ItemStack)stack)) == -1) {
+            if (($$3 = net.minecraft.world.entity.projectile.Arrow.getCustomColor((net.minecraft.world.item.ItemStack)$$0)) == -1) {
                 this.updateColor();
             } else {
-                this.setFixedColor(n);
+                this.setFixedColor($$3);
             }
-        } else if (stack.is(net.minecraft.world.item.Items.ARROW)) {
+        } else if ($$0.is(net.minecraft.world.item.Items.ARROW)) {
             this.potion = net.minecraft.world.item.alchemy.Potions.EMPTY;
             this.effects.clear();
             this.entityData.set(ID_EFFECT_COLOR, (java.lang.Object)-1);
         }
     }
 
-    public static int getCustomColor(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTag();
-        if (compoundTag != null && compoundTag.contains("CustomPotionColor", 99)) {
-            return compoundTag.getInt("CustomPotionColor");
+    public static int getCustomColor(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTag();
+        if ($$1 != null && $$1.contains("CustomPotionColor", 99)) {
+            return $$1.getInt("CustomPotionColor");
         }
         return -1;
     }
@@ -71,8 +71,8 @@
         }
     }
 
-    public void addEffect(net.minecraft.world.effect.MobEffectInstance effectInstance) {
-        this.effects.add((java.lang.Object)effectInstance);
+    public void addEffect(net.minecraft.world.effect.MobEffectInstance $$0) {
+        this.effects.add((java.lang.Object)$$0);
         this.getEntityData().set(ID_EFFECT_COLOR, (java.lang.Object)net.minecraft.world.item.alchemy.PotionUtils.getColor((java.util.Collection)net.minecraft.world.item.alchemy.PotionUtils.getAllEffects((net.minecraft.world.item.alchemy.Potion)this.potion, (java.util.Collection)this.effects)));
     }
 
@@ -99,16 +99,16 @@
         }
     }
 
-    public void makeParticle(int particleAmount) {
-        int n = this.getColor();
-        if (n == -1 || particleAmount <= 0) {
+    public void makeParticle(int $$0) {
+        int $$1 = this.getColor();
+        if ($$1 == -1 || $$0 <= 0) {
             return;
         }
-        double d = (double)(n >> 16 & 0xFF) / 255.0;
-        double d2 = (double)(n >> 8 & 0xFF) / 255.0;
-        double d3 = (double)(n >> 0 & 0xFF) / 255.0;
-        for (int i = 0; i < particleAmount; ++i) {
-            this.level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.ENTITY_EFFECT, this.getRandomX(0.5), this.getRandomY(), this.getRandomZ(0.5), d, d2, d3);
+        double $$2 = (double)($$1 >> 16 & 0xFF) / 255.0;
+        double $$3 = (double)($$1 >> 8 & 0xFF) / 255.0;
+        double $$4 = (double)($$1 >> 0 & 0xFF) / 255.0;
+        for (int $$5 = 0; $$5 < $$0; ++$$5) {
+            this.level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.ENTITY_EFFECT, this.getRandomX(0.5), this.getRandomY(), this.getRandomZ(0.5), $$2, $$3, $$4);
         }
     }
 
@@ -116,52 +116,52 @@
         return (java.lang.Integer)this.entityData.get(ID_EFFECT_COLOR);
     }
 
-    public void setFixedColor(int fixedColor) {
+    public void setFixedColor(int $$0) {
         this.fixedColor = true;
-        this.entityData.set(ID_EFFECT_COLOR, (java.lang.Object)fixedColor);
+        this.entityData.set(ID_EFFECT_COLOR, (java.lang.Object)$$0);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
         if (this.potion != net.minecraft.world.item.alchemy.Potions.EMPTY) {
-            compound.putString("Potion", net.minecraft.core.Registry.POTION.getKey((java.lang.Object)this.potion).toString());
+            $$0.putString("Potion", net.minecraft.core.Registry.POTION.getKey((java.lang.Object)this.potion).toString());
         }
         if (this.fixedColor) {
-            compound.putInt("Color", this.getColor());
+            $$0.putInt("Color", this.getColor());
         }
         if (!this.effects.isEmpty()) {
-            net.minecraft.nbt.ListTag listTag = new net.minecraft.nbt.ListTag();
-            for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : this.effects) {
-                listTag.add((java.lang.Object)mobEffectInstance.save(new net.minecraft.nbt.CompoundTag()));
+            net.minecraft.nbt.ListTag $$1 = new net.minecraft.nbt.ListTag();
+            for (net.minecraft.world.effect.MobEffectInstance $$2 : this.effects) {
+                $$1.add((java.lang.Object)$$2.save(new net.minecraft.nbt.CompoundTag()));
             }
-            compound.put("CustomPotionEffects", (net.minecraft.nbt.Tag)listTag);
+            $$0.put("CustomPotionEffects", (net.minecraft.nbt.Tag)$$1);
         }
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        if (compound.contains("Potion", 8)) {
-            this.potion = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.nbt.CompoundTag)compound);
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        if ($$0.contains("Potion", 8)) {
+            this.potion = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.nbt.CompoundTag)$$0);
         }
-        for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)compound)) {
-            this.addEffect(mobEffectInstance);
+        for (net.minecraft.world.effect.MobEffectInstance $$1 : net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)$$0)) {
+            this.addEffect($$1);
         }
-        if (compound.contains("Color", 99)) {
-            this.setFixedColor(compound.getInt("Color"));
+        if ($$0.contains("Color", 99)) {
+            this.setFixedColor($$0.getInt("Color"));
         } else {
             this.updateColor();
         }
     }
 
-    public void doPostHurtEffects(net.minecraft.world.entity.LivingEntity living) {
-        super.doPostHurtEffects(living);
-        net.minecraft.world.entity.Entity entity = this.getEffectSource();
-        for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : this.potion.getEffects()) {
-            living.addEffect(new net.minecraft.world.effect.MobEffectInstance(mobEffectInstance.getEffect(), java.lang.Math.max((int)(mobEffectInstance.getDuration() / 8), (int)1), mobEffectInstance.getAmplifier(), mobEffectInstance.isAmbient(), mobEffectInstance.isVisible()), entity);
+    public void doPostHurtEffects(net.minecraft.world.entity.LivingEntity $$0) {
+        super.doPostHurtEffects($$0);
+        net.minecraft.world.entity.Entity $$1 = this.getEffectSource();
+        for (net.minecraft.world.effect.MobEffectInstance $$2 : this.potion.getEffects()) {
+            $$0.addEffect(new net.minecraft.world.effect.MobEffectInstance($$2.getEffect(), java.lang.Math.max((int)($$2.getDuration() / 8), (int)1), $$2.getAmplifier(), $$2.isAmbient(), $$2.isVisible()), $$1);
         }
         if (!this.effects.isEmpty()) {
-            for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : this.effects) {
-                living.addEffect(mobEffectInstance, entity);
+            for (net.minecraft.world.effect.MobEffectInstance $$3 : this.effects) {
+                $$0.addEffect($$3, $$1);
             }
         }
     }
@@ -179,19 +179,19 @@
         return $$0;
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 0) {
-            int n = this.getColor();
-            if (n != -1) {
-                double d = (double)(n >> 16 & 0xFF) / 255.0;
-                double d2 = (double)(n >> 8 & 0xFF) / 255.0;
-                double d3 = (double)(n >> 0 & 0xFF) / 255.0;
-                for (int i = 0; i < 20; ++i) {
-                    this.level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.ENTITY_EFFECT, this.getRandomX(0.5), this.getRandomY(), this.getRandomZ(0.5), d, d2, d3);
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 0) {
+            int $$1 = this.getColor();
+            if ($$1 != -1) {
+                double $$2 = (double)($$1 >> 16 & 0xFF) / 255.0;
+                double $$3 = (double)($$1 >> 8 & 0xFF) / 255.0;
+                double $$4 = (double)($$1 >> 0 & 0xFF) / 255.0;
+                for (int $$5 = 0; $$5 < 20; ++$$5) {
+                    this.level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.ENTITY_EFFECT, this.getRandomX(0.5), this.getRandomY(), this.getRandomZ(0.5), $$2, $$3, $$4);
                 }
             }
         } else {
-            super.handleEntityEvent(id);
+            super.handleEntityEvent($$0);
         }
     }
 