--- net/minecraft/world/entity/projectile/DragonFireball.java
+++ net/minecraft/world/entity/projectile/DragonFireball.java
@@ -12,37 +12,37 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.DragonFireball(net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity shooter, double offsetX, double offsetY, double offsetZ) {
-        super(net.minecraft.world.entity.EntityType.DRAGON_FIREBALL, shooter, offsetX, offsetY, offsetZ, level);
+    public net.minecraft.world.entity.projectile.DragonFireball(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.LivingEntity $$1, double $$2, double $$3, double $$4) {
+        super(net.minecraft.world.entity.EntityType.DRAGON_FIREBALL, $$1, $$2, $$3, $$4, $$0);
     }
 
-    public void onHit(net.minecraft.world.phys.HitResult result) {
-        super.onHit(result);
-        if (result.getType() == net.minecraft.world.phys.HitResult$Type.ENTITY && this.ownedBy(((net.minecraft.world.phys.EntityHitResult)result).getEntity())) {
+    public void onHit(net.minecraft.world.phys.HitResult $$0) {
+        super.onHit($$0);
+        if ($$0.getType() == net.minecraft.world.phys.HitResult$Type.ENTITY && this.ownedBy(((net.minecraft.world.phys.EntityHitResult)$$0).getEntity())) {
             return;
         }
         if (!this.level.isClientSide) {
-            java.util.List list = this.level.getEntitiesOfClass(net.minecraft.world.entity.LivingEntity.class, this.getBoundingBox().inflate(4.0, 2.0, 4.0));
-            net.minecraft.world.entity.AreaEffectCloud areaEffectCloud = new net.minecraft.world.entity.AreaEffectCloud(this.level, this.getX(), this.getY(), this.getZ());
-            net.minecraft.world.entity.Entity entity = this.getOwner();
-            if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-                areaEffectCloud.setOwner((net.minecraft.world.entity.LivingEntity)entity);
+            java.util.List $$1 = this.level.getEntitiesOfClass(net.minecraft.world.entity.LivingEntity.class, this.getBoundingBox().inflate(4.0, 2.0, 4.0));
+            net.minecraft.world.entity.AreaEffectCloud $$2 = new net.minecraft.world.entity.AreaEffectCloud(this.level, this.getX(), this.getY(), this.getZ());
+            net.minecraft.world.entity.Entity $$3 = this.getOwner();
+            if ($$3 instanceof net.minecraft.world.entity.LivingEntity) {
+                $$2.setOwner((net.minecraft.world.entity.LivingEntity)$$3);
             }
-            areaEffectCloud.setParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.DRAGON_BREATH);
-            areaEffectCloud.setRadius(3.0f);
-            areaEffectCloud.setDuration(600);
-            areaEffectCloud.setRadiusPerTick((7.0f - areaEffectCloud.getRadius()) / (float)areaEffectCloud.getDuration());
-            areaEffectCloud.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.HARM, 1, 1));
-            if (!list.isEmpty()) {
-                for (net.minecraft.world.entity.LivingEntity livingEntity : list) {
-                    double d = this.distanceToSqr((net.minecraft.world.entity.Entity)livingEntity);
-                    if (!(d < 16.0)) continue;
-                    areaEffectCloud.setPos(livingEntity.getX(), livingEntity.getY(), livingEntity.getZ());
+            $$2.setParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.DRAGON_BREATH);
+            $$2.setRadius(3.0f);
+            $$2.setDuration(600);
+            $$2.setRadiusPerTick((7.0f - $$2.getRadius()) / (float)$$2.getDuration());
+            $$2.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.HARM, 1, 1));
+            if (!$$1.isEmpty()) {
+                for (net.minecraft.world.entity.LivingEntity $$4 : $$1) {
+                    double $$5 = this.distanceToSqr((net.minecraft.world.entity.Entity)$$4);
+                    if (!($$5 < 16.0)) continue;
+                    $$2.setPos($$4.getX(), $$4.getY(), $$4.getZ());
                     break;
                 }
             }
             this.level.levelEvent(2006, this.blockPosition(), this.isSilent() ? -1 : 1);
-            this.level.addFreshEntity((net.minecraft.world.entity.Entity)areaEffectCloud);
+            this.level.addFreshEntity((net.minecraft.world.entity.Entity)$$2);
             this.discard();
         }
     }
@@ -51,7 +51,7 @@
         return false;
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
         return false;
     }
 