--- net/minecraft/world/entity/projectile/EyeOfEnder.java
+++ net/minecraft/world/entity/projectile/EyeOfEnder.java
@@ -18,14 +18,14 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.EyeOfEnder(net.minecraft.world.level.Level level, double x, double y, double z) {
-        this(net.minecraft.world.entity.EntityType.EYE_OF_ENDER, level);
-        this.setPos(x, y, z);
+    public net.minecraft.world.entity.projectile.EyeOfEnder(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        this(net.minecraft.world.entity.EntityType.EYE_OF_ENDER, $$0);
+        this.setPos($$1, $$2, $$3);
     }
 
-    public void setItem(net.minecraft.world.item.ItemStack stack) {
-        if (!stack.is(net.minecraft.world.item.Items.ENDER_EYE) || stack.hasTag()) {
-            this.getEntityData().set(DATA_ITEM_STACK, (java.lang.Object)((net.minecraft.world.item.ItemStack)net.minecraft.Util.make((java.lang.Object)stack.copy(), copy -> copy.setCount(1))));
+    public void setItem(net.minecraft.world.item.ItemStack $$02) {
+        if (!$$02.is(net.minecraft.world.item.Items.ENDER_EYE) || $$02.hasTag()) {
+            this.getEntityData().set(DATA_ITEM_STACK, (java.lang.Object)((net.minecraft.world.item.ItemStack)net.minecraft.Util.make((java.lang.Object)$$02.copy(), $$0 -> $$0.setCount(1))));
         }
     }
 
@@ -42,40 +42,40 @@
         this.getEntityData().define(DATA_ITEM_STACK, (java.lang.Object)net.minecraft.world.item.ItemStack.EMPTY);
     }
 
-    public boolean shouldRenderAtSqrDistance(double distance) {
-        double d = this.getBoundingBox().getSize() * 4.0;
-        if (java.lang.Double.isNaN((double)d)) {
-            d = 4.0;
+    public boolean shouldRenderAtSqrDistance(double $$0) {
+        double $$1 = this.getBoundingBox().getSize() * 4.0;
+        if (java.lang.Double.isNaN((double)$$1)) {
+            $$1 = 4.0;
         }
-        return distance < (d *= 64.0) * d;
+        return $$0 < ($$1 *= 64.0) * $$1;
     }
 
-    public void signalTo(net.minecraft.core.BlockPos pos) {
-        double d;
-        double d2 = pos.getX();
-        int n = pos.getY();
-        double d3 = pos.getZ();
-        double d4 = d2 - this.getX();
-        double d5 = java.lang.Math.sqrt((double)(d4 * d4 + (d = d3 - this.getZ()) * d));
-        if (d5 > 12.0) {
-            this.tx = this.getX() + d4 / d5 * 12.0;
-            this.tz = this.getZ() + d / d5 * 12.0;
+    public void signalTo(net.minecraft.core.BlockPos $$0) {
+        double $$5;
+        double $$1 = $$0.getX();
+        int $$2 = $$0.getY();
+        double $$3 = $$0.getZ();
+        double $$4 = $$1 - this.getX();
+        double $$6 = java.lang.Math.sqrt((double)($$4 * $$4 + ($$5 = $$3 - this.getZ()) * $$5));
+        if ($$6 > 12.0) {
+            this.tx = this.getX() + $$4 / $$6 * 12.0;
+            this.tz = this.getZ() + $$5 / $$6 * 12.0;
             this.ty = this.getY() + 8.0;
         } else {
-            this.tx = d2;
-            this.ty = n;
-            this.tz = d3;
+            this.tx = $$1;
+            this.ty = $$2;
+            this.tz = $$3;
         }
         this.life = 0;
         this.surviveAfterDeath = this.random.nextInt(5) > 0;
     }
 
-    public void lerpMotion(double x, double y, double z) {
-        this.setDeltaMovement(x, y, z);
+    public void lerpMotion(double $$0, double $$1, double $$2) {
+        this.setDeltaMovement($$0, $$1, $$2);
         if (this.xRotO == 0.0f && this.yRotO == 0.0f) {
-            double d = java.lang.Math.sqrt((double)(x * x + z * z));
-            this.setYRot((float)(net.minecraft.util.Mth.atan2((double)x, (double)z) * 57.2957763671875));
-            this.setXRot((float)(net.minecraft.util.Mth.atan2((double)y, (double)d) * 57.2957763671875));
+            double $$3 = java.lang.Math.sqrt((double)($$0 * $$0 + $$2 * $$2));
+            this.setYRot((float)(net.minecraft.util.Mth.atan2((double)$$0, (double)$$2) * 57.2957763671875));
+            this.setXRot((float)(net.minecraft.util.Mth.atan2((double)$$1, (double)$$3) * 57.2957763671875));
             this.yRotO = this.getYRot();
             this.xRotO = this.getXRot();
         }
@@ -130,16 +130,16 @@
         }
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        net.minecraft.world.item.ItemStack itemStack = this.getItemRaw();
-        if (!itemStack.isEmpty()) {
-            compound.put("Item", (net.minecraft.nbt.Tag)itemStack.save(new net.minecraft.nbt.CompoundTag()));
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        net.minecraft.world.item.ItemStack $$1 = this.getItemRaw();
+        if (!$$1.isEmpty()) {
+            $$0.put("Item", (net.minecraft.nbt.Tag)$$1.save(new net.minecraft.nbt.CompoundTag()));
         }
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.of((net.minecraft.nbt.CompoundTag)compound.getCompound("Item"));
-        this.setItem(itemStack);
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        net.minecraft.world.item.ItemStack $$1 = net.minecraft.world.item.ItemStack.of((net.minecraft.nbt.CompoundTag)$$0.getCompound("Item"));
+        this.setItem($$1);
     }
 
     public float getLightLevelDependentMagicValue() {
@@ -150,10 +150,6 @@
         return false;
     }
 
-    public net.minecraft.network.protocol.Packet<?> getAddEntityPacket() {
-        return new net.minecraft.network.protocol.game.ClientboundAddEntityPacket((net.minecraft.world.entity.Entity)this);
-    }
-
     static {
         DATA_ITEM_STACK = net.minecraft.network.syncher.SynchedEntityData.defineId(net.minecraft.world.entity.projectile.EyeOfEnder.class, (net.minecraft.network.syncher.EntityDataSerializer)net.minecraft.network.syncher.EntityDataSerializers.ITEM_STACK);
     }