--- net/minecraft/world/entity/projectile/FireworkRocketEntity.java
+++ net/minecraft/world/entity/projectile/FireworkRocketEntity.java
@@ -19,38 +19,38 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level level, double x, double y, double z, net.minecraft.world.item.ItemStack stack) {
-        super(net.minecraft.world.entity.EntityType.FIREWORK_ROCKET, level);
+    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3, net.minecraft.world.item.ItemStack $$4) {
+        super(net.minecraft.world.entity.EntityType.FIREWORK_ROCKET, $$0);
         this.life = 0;
-        this.setPos(x, y, z);
-        int n = 1;
-        if (!stack.isEmpty() && stack.hasTag()) {
-            this.entityData.set(DATA_ID_FIREWORKS_ITEM, (java.lang.Object)stack.copy());
-            n += stack.getOrCreateTagElement("Fireworks").getByte("Flight");
+        this.setPos($$1, $$2, $$3);
+        int $$5 = 1;
+        if (!$$4.isEmpty() && $$4.hasTag()) {
+            this.entityData.set(DATA_ID_FIREWORKS_ITEM, (java.lang.Object)$$4.copy());
+            $$5 += $$4.getOrCreateTagElement("Fireworks").getByte("Flight");
         }
         this.setDeltaMovement(this.random.triangle(0.0, 0.002297), 0.05, this.random.triangle(0.0, 0.002297));
-        this.lifetime = 10 * n + this.random.nextInt(6) + this.random.nextInt(7);
+        this.lifetime = 10 * $$5 + this.random.nextInt(6) + this.random.nextInt(7);
     }
 
-    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level level, @javax.annotation.Nullable net.minecraft.world.entity.Entity shooter, double x, double y, double z, net.minecraft.world.item.ItemStack stack) {
-        this(level, x, y, z, stack);
-        this.setOwner(shooter);
+    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1, double $$2, double $$3, double $$4, net.minecraft.world.item.ItemStack $$5) {
+        this($$0, $$2, $$3, $$4, $$5);
+        this.setOwner($$1);
     }
 
-    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level level, net.minecraft.world.item.ItemStack stack, net.minecraft.world.entity.LivingEntity shooter) {
-        this(level, (net.minecraft.world.entity.Entity)shooter, shooter.getX(), shooter.getY(), shooter.getZ(), stack);
-        this.entityData.set(DATA_ATTACHED_TO_TARGET, (java.lang.Object)java.util.OptionalInt.of((int)shooter.getId()));
-        this.attachedToEntity = shooter;
+    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.world.entity.LivingEntity $$2) {
+        this($$0, (net.minecraft.world.entity.Entity)$$2, $$2.getX(), $$2.getY(), $$2.getZ(), $$1);
+        this.entityData.set(DATA_ATTACHED_TO_TARGET, (java.lang.Object)java.util.OptionalInt.of((int)$$2.getId()));
+        this.attachedToEntity = $$2;
     }
 
-    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level level, net.minecraft.world.item.ItemStack stack, double x, double y, double z, boolean shotAtAngle) {
-        this(level, x, y, z, stack);
-        this.entityData.set(DATA_SHOT_AT_ANGLE, (java.lang.Object)shotAtAngle);
+    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level $$0, net.minecraft.world.item.ItemStack $$1, double $$2, double $$3, double $$4, boolean $$5) {
+        this($$0, $$2, $$3, $$4, $$1);
+        this.entityData.set(DATA_SHOT_AT_ANGLE, (java.lang.Object)$$5);
     }
 
-    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level level, net.minecraft.world.item.ItemStack stack, net.minecraft.world.entity.Entity shooter, double x, double y, double z, boolean shotAtAngle) {
-        this(level, stack, x, y, z, shotAtAngle);
-        this.setOwner(shooter);
+    public net.minecraft.world.entity.projectile.FireworkRocketEntity(net.minecraft.world.level.Level $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.world.entity.Entity $$2, double $$3, double $$4, double $$5, boolean $$6) {
+        this($$0, $$1, $$3, $$4, $$5, $$6);
+        this.setOwner($$2);
     }
 
     public void defineSynchedData() {
@@ -59,33 +59,33 @@
         this.entityData.define(DATA_SHOT_AT_ANGLE, (java.lang.Object)false);
     }
 
-    public boolean shouldRenderAtSqrDistance(double distance) {
-        return distance < 4096.0 && !this.isAttachedToEntity();
+    public boolean shouldRenderAtSqrDistance(double $$0) {
+        return $$0 < 4096.0 && !this.isAttachedToEntity();
     }
 
-    public boolean shouldRender(double x, double y, double z) {
-        return super.shouldRender(x, y, z) && !this.isAttachedToEntity();
+    public boolean shouldRender(double $$0, double $$1, double $$2) {
+        return super.shouldRender($$0, $$1, $$2) && !this.isAttachedToEntity();
     }
 
     public void tick() {
         super.tick();
         if (this.isAttachedToEntity()) {
             if (this.attachedToEntity == null) {
-                ((java.util.OptionalInt)this.entityData.get(DATA_ATTACHED_TO_TARGET)).ifPresent(target -> {
-                    net.minecraft.world.entity.Entity entity = this.level.getEntity(target);
-                    if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-                        this.attachedToEntity = (net.minecraft.world.entity.LivingEntity)entity;
+                ((java.util.OptionalInt)this.entityData.get(DATA_ATTACHED_TO_TARGET)).ifPresent($$0 -> {
+                    net.minecraft.world.entity.Entity $$1 = this.level.getEntity($$0);
+                    if ($$1 instanceof net.minecraft.world.entity.LivingEntity) {
+                        this.attachedToEntity = (net.minecraft.world.entity.LivingEntity)$$1;
                     }
                 });
             }
             if (this.attachedToEntity != null) {
                 net.minecraft.world.phys.Vec3 $$5;
                 if (this.attachedToEntity.isFallFlying()) {
-                    net.minecraft.world.phys.Vec3 $$0 = this.attachedToEntity.getLookAngle();
+                    net.minecraft.world.phys.Vec3 $$02 = this.attachedToEntity.getLookAngle();
                     double $$1 = 1.5;
                     double $$2 = 0.1;
                     net.minecraft.world.phys.Vec3 $$3 = this.attachedToEntity.getDeltaMovement();
-                    this.attachedToEntity.setDeltaMovement($$3.add($$0.x * 0.1 + ($$0.x * 1.5 - $$3.x) * 0.5, $$0.y * 0.1 + ($$0.y * 1.5 - $$3.y) * 0.5, $$0.z * 0.1 + ($$0.z * 1.5 - $$3.z) * 0.5));
+                    this.attachedToEntity.setDeltaMovement($$3.add($$02.x * 0.1 + ($$02.x * 1.5 - $$3.x) * 0.5, $$02.y * 0.1 + ($$02.y * 1.5 - $$3.y) * 0.5, $$02.z * 0.1 + ($$02.z * 1.5 - $$3.z) * 0.5));
                     net.minecraft.world.phys.Vec3 $$4 = this.attachedToEntity.getHandHoldingItemAngle(net.minecraft.world.item.Items.FIREWORK_ROCKET);
                 } else {
                     $$5 = net.minecraft.world.phys.Vec3.ZERO;
@@ -127,21 +127,21 @@
         this.discard();
     }
 
-    public void onHitEntity(net.minecraft.world.phys.EntityHitResult result) {
-        super.onHitEntity(result);
+    public void onHitEntity(net.minecraft.world.phys.EntityHitResult $$0) {
+        super.onHitEntity($$0);
         if (this.level.isClientSide) {
             return;
         }
         this.explode();
     }
 
-    public void onHitBlock(net.minecraft.world.phys.BlockHitResult result) {
-        net.minecraft.core.BlockPos blockPos = new net.minecraft.core.BlockPos((net.minecraft.core.Vec3i)result.getBlockPos());
-        this.level.getBlockState(blockPos).entityInside(this.level, blockPos, (net.minecraft.world.entity.Entity)this);
+    public void onHitBlock(net.minecraft.world.phys.BlockHitResult $$0) {
+        net.minecraft.core.BlockPos $$1 = new net.minecraft.core.BlockPos((net.minecraft.core.Vec3i)$$0.getBlockPos());
+        this.level.getBlockState($$1).entityInside(this.level, $$1, (net.minecraft.world.entity.Entity)this);
         if (!this.level.isClientSide() && this.hasExplosion()) {
             this.explode();
         }
-        super.onHitBlock(result);
+        super.onHitBlock($$0);
     }
 
     public boolean hasExplosion() {
@@ -192,43 +192,43 @@
         return (java.lang.Boolean)this.entityData.get(DATA_SHOT_AT_ANGLE);
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 17 && this.level.isClientSide) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 17 && this.level.isClientSide) {
             if (!this.hasExplosion()) {
-                for (int i = 0; i < this.random.nextInt(3) + 2; ++i) {
+                for (int $$1 = 0; $$1 < this.random.nextInt(3) + 2; ++$$1) {
                     this.level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.POOF, this.getX(), this.getY(), this.getZ(), this.random.nextGaussian() * 0.05, 0.005, this.random.nextGaussian() * 0.05);
                 }
             } else {
-                net.minecraft.world.item.ItemStack itemStack = (net.minecraft.world.item.ItemStack)this.entityData.get(DATA_ID_FIREWORKS_ITEM);
-                net.minecraft.nbt.CompoundTag compoundTag = itemStack.isEmpty() ? null : itemStack.getTagElement("Fireworks");
-                net.minecraft.world.phys.Vec3 vec3 = this.getDeltaMovement();
-                this.level.createFireworks(this.getX(), this.getY(), this.getZ(), vec3.x, vec3.y, vec3.z, compoundTag);
+                net.minecraft.world.item.ItemStack $$2 = (net.minecraft.world.item.ItemStack)this.entityData.get(DATA_ID_FIREWORKS_ITEM);
+                net.minecraft.nbt.CompoundTag $$3 = $$2.isEmpty() ? null : $$2.getTagElement("Fireworks");
+                net.minecraft.world.phys.Vec3 $$4 = this.getDeltaMovement();
+                this.level.createFireworks(this.getX(), this.getY(), this.getZ(), $$4.x, $$4.y, $$4.z, $$3);
             }
         }
-        super.handleEntityEvent(id);
+        super.handleEntityEvent($$0);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("Life", this.life);
-        compound.putInt("LifeTime", this.lifetime);
-        net.minecraft.world.item.ItemStack itemStack = (net.minecraft.world.item.ItemStack)this.entityData.get(DATA_ID_FIREWORKS_ITEM);
-        if (!itemStack.isEmpty()) {
-            compound.put("FireworksItem", (net.minecraft.nbt.Tag)itemStack.save(new net.minecraft.nbt.CompoundTag()));
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("Life", this.life);
+        $$0.putInt("LifeTime", this.lifetime);
+        net.minecraft.world.item.ItemStack $$1 = (net.minecraft.world.item.ItemStack)this.entityData.get(DATA_ID_FIREWORKS_ITEM);
+        if (!$$1.isEmpty()) {
+            $$0.put("FireworksItem", (net.minecraft.nbt.Tag)$$1.save(new net.minecraft.nbt.CompoundTag()));
         }
-        compound.putBoolean("ShotAtAngle", ((java.lang.Boolean)this.entityData.get(DATA_SHOT_AT_ANGLE)).booleanValue());
+        $$0.putBoolean("ShotAtAngle", ((java.lang.Boolean)this.entityData.get(DATA_SHOT_AT_ANGLE)).booleanValue());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.life = compound.getInt("Life");
-        this.lifetime = compound.getInt("LifeTime");
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.of((net.minecraft.nbt.CompoundTag)compound.getCompound("FireworksItem"));
-        if (!itemStack.isEmpty()) {
-            this.entityData.set(DATA_ID_FIREWORKS_ITEM, (java.lang.Object)itemStack);
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.life = $$0.getInt("Life");
+        this.lifetime = $$0.getInt("LifeTime");
+        net.minecraft.world.item.ItemStack $$1 = net.minecraft.world.item.ItemStack.of((net.minecraft.nbt.CompoundTag)$$0.getCompound("FireworksItem"));
+        if (!$$1.isEmpty()) {
+            this.entityData.set(DATA_ID_FIREWORKS_ITEM, (java.lang.Object)$$1);
         }
-        if (compound.contains("ShotAtAngle")) {
-            this.entityData.set(DATA_SHOT_AT_ANGLE, (java.lang.Object)compound.getBoolean("ShotAtAngle"));
+        if ($$0.contains("ShotAtAngle")) {
+            this.entityData.set(DATA_SHOT_AT_ANGLE, (java.lang.Object)$$0.getBoolean("ShotAtAngle"));
         }
     }
 