--- net/minecraft/world/entity/projectile/LargeFireball.java
+++ net/minecraft/world/entity/projectile/LargeFireball.java
@@ -19,37 +19,37 @@
         this.explosionPower = $$5;
     }
 
-    public void onHit(net.minecraft.world.phys.HitResult result) {
-        super.onHit(result);
+    public void onHit(net.minecraft.world.phys.HitResult $$0) {
+        super.onHit($$0);
         if (!this.level.isClientSide) {
-            boolean bl = this.level.getGameRules().getBoolean(net.minecraft.world.level.GameRules.RULE_MOBGRIEFING);
-            this.level.explode(null, this.getX(), this.getY(), this.getZ(), (float)this.explosionPower, bl, bl ? net.minecraft.world.level.Explosion$BlockInteraction.DESTROY : net.minecraft.world.level.Explosion$BlockInteraction.NONE);
+            boolean $$1 = this.level.getGameRules().getBoolean(net.minecraft.world.level.GameRules.RULE_MOBGRIEFING);
+            this.level.explode(null, this.getX(), this.getY(), this.getZ(), (float)this.explosionPower, $$1, $$1 ? net.minecraft.world.level.Explosion$BlockInteraction.DESTROY : net.minecraft.world.level.Explosion$BlockInteraction.NONE);
             this.discard();
         }
     }
 
-    public void onHitEntity(net.minecraft.world.phys.EntityHitResult result) {
-        super.onHitEntity(result);
+    public void onHitEntity(net.minecraft.world.phys.EntityHitResult $$0) {
+        super.onHitEntity($$0);
         if (this.level.isClientSide) {
             return;
         }
-        net.minecraft.world.entity.Entity entity = result.getEntity();
-        net.minecraft.world.entity.Entity entity2 = this.getOwner();
-        entity.hurt(net.minecraft.world.damagesource.DamageSource.fireball((net.minecraft.world.entity.projectile.Fireball)this, (net.minecraft.world.entity.Entity)entity2), 6.0f);
-        if (entity2 instanceof net.minecraft.world.entity.LivingEntity) {
-            this.doEnchantDamageEffects((net.minecraft.world.entity.LivingEntity)entity2, entity);
+        net.minecraft.world.entity.Entity $$1 = $$0.getEntity();
+        net.minecraft.world.entity.Entity $$2 = this.getOwner();
+        $$1.hurt(net.minecraft.world.damagesource.DamageSource.fireball((net.minecraft.world.entity.projectile.Fireball)this, (net.minecraft.world.entity.Entity)$$2), 6.0f);
+        if ($$2 instanceof net.minecraft.world.entity.LivingEntity) {
+            this.doEnchantDamageEffects((net.minecraft.world.entity.LivingEntity)$$2, $$1);
         }
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putByte("ExplosionPower", (byte)this.explosionPower);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putByte("ExplosionPower", (byte)this.explosionPower);
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        if (compound.contains("ExplosionPower", 99)) {
-            this.explosionPower = compound.getByte("ExplosionPower");
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        if ($$0.contains("ExplosionPower", 99)) {
+            this.explosionPower = $$0.getByte("ExplosionPower");
         }
     }
 }