--- net/minecraft/world/entity/projectile/LlamaSpit.java
+++ net/minecraft/world/entity/projectile/LlamaSpit.java
@@ -10,10 +10,10 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.LlamaSpit(net.minecraft.world.level.Level level, net.minecraft.world.entity.animal.horse.Llama spitter) {
-        this(net.minecraft.world.entity.EntityType.LLAMA_SPIT, level);
-        this.setOwner((net.minecraft.world.entity.Entity)spitter);
-        this.setPos(spitter.getX() - (double)(spitter.getBbWidth() + 1.0f) * 0.5 * (double)net.minecraft.util.Mth.sin((float)(spitter.yBodyRot * ((float)java.lang.Math.PI / 180))), spitter.getEyeY() - (double)0.1f, spitter.getZ() + (double)(spitter.getBbWidth() + 1.0f) * 0.5 * (double)net.minecraft.util.Mth.cos((float)(spitter.yBodyRot * ((float)java.lang.Math.PI / 180))));
+    public net.minecraft.world.entity.projectile.LlamaSpit(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.animal.horse.Llama $$1) {
+        this(net.minecraft.world.entity.EntityType.LLAMA_SPIT, $$0);
+        this.setOwner((net.minecraft.world.entity.Entity)$$1);
+        this.setPos($$1.getX() - (double)($$1.getBbWidth() + 1.0f) * 0.5 * (double)net.minecraft.util.Mth.sin((float)($$1.yBodyRot * ((float)java.lang.Math.PI / 180))), $$1.getEyeY() - (double)0.1f, $$1.getZ() + (double)($$1.getBbWidth() + 1.0f) * 0.5 * (double)net.minecraft.util.Mth.cos((float)($$1.yBodyRot * ((float)java.lang.Math.PI / 180))));
     }
 
     public void tick() {
@@ -42,16 +42,16 @@
         this.setPos($$2, $$3, $$4);
     }
 
-    public void onHitEntity(net.minecraft.world.phys.EntityHitResult result) {
-        super.onHitEntity(result);
-        net.minecraft.world.entity.Entity entity = this.getOwner();
-        if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-            result.getEntity().hurt(net.minecraft.world.damagesource.DamageSource.indirectMobAttack((net.minecraft.world.entity.Entity)this, (net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)entity)).setProjectile(), 1.0f);
+    public void onHitEntity(net.minecraft.world.phys.EntityHitResult $$0) {
+        super.onHitEntity($$0);
+        net.minecraft.world.entity.Entity $$1 = this.getOwner();
+        if ($$1 instanceof net.minecraft.world.entity.LivingEntity) {
+            $$0.getEntity().hurt(net.minecraft.world.damagesource.DamageSource.indirectMobAttack((net.minecraft.world.entity.Entity)this, (net.minecraft.world.entity.LivingEntity)((net.minecraft.world.entity.LivingEntity)$$1)).setProjectile(), 1.0f);
         }
     }
 
-    public void onHitBlock(net.minecraft.world.phys.BlockHitResult result) {
-        super.onHitBlock(result);
+    public void onHitBlock(net.minecraft.world.phys.BlockHitResult $$0) {
+        super.onHitBlock($$0);
         if (!this.level.isClientSide) {
             this.discard();
         }