--- net/minecraft/world/entity/projectile/Snowball.java
+++ net/minecraft/world/entity/projectile/Snowball.java
@@ -10,12 +10,12 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.Snowball(net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity shooter) {
-        super(net.minecraft.world.entity.EntityType.SNOWBALL, shooter, level);
+    public net.minecraft.world.entity.projectile.Snowball(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        super(net.minecraft.world.entity.EntityType.SNOWBALL, $$1, $$0);
     }
 
-    public net.minecraft.world.entity.projectile.Snowball(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.SNOWBALL, x, y, z, level);
+    public net.minecraft.world.entity.projectile.Snowball(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.SNOWBALL, $$1, $$2, $$3, $$0);
     }
 
     public net.minecraft.world.item.Item getDefaultItem() {
@@ -27,24 +27,24 @@
         return $$0.isEmpty() ? net.minecraft.core.particles.ParticleTypes.ITEM_SNOWBALL : new net.minecraft.core.particles.ItemParticleOption(net.minecraft.core.particles.ParticleTypes.ITEM, $$0);
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 3) {
-            net.minecraft.core.particles.ParticleOptions particleOptions = this.getParticle();
-            for (int i = 0; i < 8; ++i) {
-                this.level.addParticle(particleOptions, this.getX(), this.getY(), this.getZ(), 0.0, 0.0, 0.0);
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 3) {
+            net.minecraft.core.particles.ParticleOptions $$1 = this.getParticle();
+            for (int $$2 = 0; $$2 < 8; ++$$2) {
+                this.level.addParticle($$1, this.getX(), this.getY(), this.getZ(), 0.0, 0.0, 0.0);
             }
         }
     }
 
-    public void onHitEntity(net.minecraft.world.phys.EntityHitResult result) {
-        super.onHitEntity(result);
-        net.minecraft.world.entity.Entity entity = result.getEntity();
-        int n = entity instanceof net.minecraft.world.entity.monster.Blaze ? 3 : 0;
-        entity.hurt(net.minecraft.world.damagesource.DamageSource.thrown((net.minecraft.world.entity.Entity)this, (net.minecraft.world.entity.Entity)this.getOwner()), (float)n);
+    public void onHitEntity(net.minecraft.world.phys.EntityHitResult $$0) {
+        super.onHitEntity($$0);
+        net.minecraft.world.entity.Entity $$1 = $$0.getEntity();
+        int $$2 = $$1 instanceof net.minecraft.world.entity.monster.Blaze ? 3 : 0;
+        $$1.hurt(net.minecraft.world.damagesource.DamageSource.thrown((net.minecraft.world.entity.Entity)this, (net.minecraft.world.entity.Entity)this.getOwner()), (float)$$2);
     }
 
-    public void onHit(net.minecraft.world.phys.HitResult result) {
-        super.onHit(result);
+    public void onHit(net.minecraft.world.phys.HitResult $$0) {
+        super.onHit($$0);
         if (!this.level.isClientSide) {
             this.level.broadcastEntityEvent((net.minecraft.world.entity.Entity)this, (byte)3);
             this.discard();