--- net/minecraft/world/entity/projectile/SpectralArrow.java
+++ net/minecraft/world/entity/projectile/SpectralArrow.java
@@ -13,13 +13,13 @@
         this.duration = 200;
     }
 
-    public net.minecraft.world.entity.projectile.SpectralArrow(net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity shooter) {
-        super(net.minecraft.world.entity.EntityType.SPECTRAL_ARROW, shooter, level);
+    public net.minecraft.world.entity.projectile.SpectralArrow(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        super(net.minecraft.world.entity.EntityType.SPECTRAL_ARROW, $$1, $$0);
         this.duration = 200;
     }
 
-    public net.minecraft.world.entity.projectile.SpectralArrow(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.SPECTRAL_ARROW, x, y, z, level);
+    public net.minecraft.world.entity.projectile.SpectralArrow(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.SPECTRAL_ARROW, $$1, $$2, $$3, $$0);
         this.duration = 200;
     }
 
@@ -34,22 +34,22 @@
         return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPECTRAL_ARROW);
     }
 
-    public void doPostHurtEffects(net.minecraft.world.entity.LivingEntity living) {
-        super.doPostHurtEffects(living);
-        net.minecraft.world.effect.MobEffectInstance mobEffectInstance = new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.GLOWING, this.duration, 0);
-        living.addEffect(mobEffectInstance, this.getEffectSource());
+    public void doPostHurtEffects(net.minecraft.world.entity.LivingEntity $$0) {
+        super.doPostHurtEffects($$0);
+        net.minecraft.world.effect.MobEffectInstance $$1 = new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.GLOWING, this.duration, 0);
+        $$0.addEffect($$1, this.getEffectSource());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        if (compound.contains("Duration")) {
-            this.duration = compound.getInt("Duration");
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        if ($$0.contains("Duration")) {
+            this.duration = $$0.getInt("Duration");
         }
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("Duration", this.duration);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("Duration", this.duration);
     }
 }
 