--- net/minecraft/world/entity/projectile/ThrowableProjectile.java
+++ net/minecraft/world/entity/projectile/ThrowableProjectile.java
@@ -10,22 +10,22 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.ThrowableProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.ThrowableProjectile> entityType, double x, double y, double z, net.minecraft.world.level.Level level) {
-        this(entityType, level);
-        this.setPos(x, y, z);
+    public net.minecraft.world.entity.projectile.ThrowableProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.ThrowableProjectile> $$0, double $$1, double $$2, double $$3, net.minecraft.world.level.Level $$4) {
+        this($$0, $$4);
+        this.setPos($$1, $$2, $$3);
     }
 
-    public net.minecraft.world.entity.projectile.ThrowableProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.ThrowableProjectile> entityType, net.minecraft.world.entity.LivingEntity shooter, net.minecraft.world.level.Level level) {
-        this(entityType, shooter.getX(), shooter.getEyeY() - (double)0.1f, shooter.getZ(), level);
-        this.setOwner((net.minecraft.world.entity.Entity)shooter);
+    public net.minecraft.world.entity.projectile.ThrowableProjectile(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.projectile.ThrowableProjectile> $$0, net.minecraft.world.entity.LivingEntity $$1, net.minecraft.world.level.Level $$2) {
+        this($$0, $$1.getX(), $$1.getEyeY() - (double)0.1f, $$1.getZ(), $$2);
+        this.setOwner((net.minecraft.world.entity.Entity)$$1);
     }
 
-    public boolean shouldRenderAtSqrDistance(double distance) {
-        double d = this.getBoundingBox().getSize() * 4.0;
-        if (java.lang.Double.isNaN((double)d)) {
-            d = 4.0;
+    public boolean shouldRenderAtSqrDistance(double $$0) {
+        double $$1 = this.getBoundingBox().getSize() * 4.0;
+        if (java.lang.Double.isNaN((double)$$1)) {
+            $$1 = 4.0;
         }
-        return distance < (d *= 64.0) * d;
+        return $$0 < ($$1 *= 64.0) * $$1;
     }
 
     public void tick() {