--- net/minecraft/world/entity/projectile/ThrownEgg.java
+++ net/minecraft/world/entity/projectile/ThrownEgg.java
@@ -10,41 +10,42 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.ThrownEgg(net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity shooter) {
-        super(net.minecraft.world.entity.EntityType.EGG, shooter, level);
+    public net.minecraft.world.entity.projectile.ThrownEgg(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        super(net.minecraft.world.entity.EntityType.EGG, $$1, $$0);
     }
 
-    public net.minecraft.world.entity.projectile.ThrownEgg(net.minecraft.world.level.Level level, double x, double y, double z) {
-        super(net.minecraft.world.entity.EntityType.EGG, x, y, z, level);
+    public net.minecraft.world.entity.projectile.ThrownEgg(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3) {
+        super(net.minecraft.world.entity.EntityType.EGG, $$1, $$2, $$3, $$0);
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 3) {
-            double d = 0.08;
-            for (int i = 0; i < 8; ++i) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 3) {
+            double $$1 = 0.08;
+            for (int $$2 = 0; $$2 < 8; ++$$2) {
                 this.level.addParticle((net.minecraft.core.particles.ParticleOptions)new net.minecraft.core.particles.ItemParticleOption(net.minecraft.core.particles.ParticleTypes.ITEM, this.getItem()), this.getX(), this.getY(), this.getZ(), ((double)this.random.nextFloat() - 0.5) * 0.08, ((double)this.random.nextFloat() - 0.5) * 0.08, ((double)this.random.nextFloat() - 0.5) * 0.08);
             }
         }
     }
 
-    public void onHitEntity(net.minecraft.world.phys.EntityHitResult result) {
-        super.onHitEntity(result);
-        result.getEntity().hurt(net.minecraft.world.damagesource.DamageSource.thrown((net.minecraft.world.entity.Entity)this, (net.minecraft.world.entity.Entity)this.getOwner()), 0.0f);
+    public void onHitEntity(net.minecraft.world.phys.EntityHitResult $$0) {
+        super.onHitEntity($$0);
+        $$0.getEntity().hurt(net.minecraft.world.damagesource.DamageSource.thrown((net.minecraft.world.entity.Entity)this, (net.minecraft.world.entity.Entity)this.getOwner()), 0.0f);
     }
 
-    public void onHit(net.minecraft.world.phys.HitResult result) {
-        super.onHit(result);
+    public void onHit(net.minecraft.world.phys.HitResult $$0) {
+        super.onHit($$0);
         if (!this.level.isClientSide) {
             if (this.random.nextInt(8) == 0) {
-                int n = 1;
+                int $$1 = 1;
                 if (this.random.nextInt(32) == 0) {
-                    n = 4;
+                    $$1 = 4;
                 }
-                for (int i = 0; i < n; ++i) {
-                    net.minecraft.world.entity.animal.Chicken chicken = (net.minecraft.world.entity.animal.Chicken)net.minecraft.world.entity.EntityType.CHICKEN.create(this.level);
-                    chicken.setAge(-24000);
-                    chicken.moveTo(this.getX(), this.getY(), this.getZ(), this.getYRot(), 0.0f);
-                    this.level.addFreshEntity((net.minecraft.world.entity.Entity)chicken);
+                for (int $$2 = 0; $$2 < $$1; ++$$2) {
+                    net.minecraft.world.entity.animal.Chicken $$3 = (net.minecraft.world.entity.animal.Chicken)net.minecraft.world.entity.EntityType.CHICKEN.create(this.level);
+                    if ($$3 == null) continue;
+                    $$3.setAge(-24000);
+                    $$3.moveTo(this.getX(), this.getY(), this.getZ(), this.getYRot(), 0.0f);
+                    this.level.addFreshEntity((net.minecraft.world.entity.Entity)$$3);
                 }
             }
             this.level.broadcastEntityEvent((net.minecraft.world.entity.Entity)this, (byte)3);