--- net/minecraft/world/entity/projectile/WitherSkull.java
+++ net/minecraft/world/entity/projectile/WitherSkull.java
@@ -12,8 +12,8 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.entity.projectile.WitherSkull(net.minecraft.world.level.Level level, net.minecraft.world.entity.LivingEntity shooter, double offsetX, double offsetY, double offsetZ) {
-        super(net.minecraft.world.entity.EntityType.WITHER_SKULL, shooter, offsetX, offsetY, offsetZ, level);
+    public net.minecraft.world.entity.projectile.WitherSkull(net.minecraft.world.level.Level $$0, net.minecraft.world.entity.LivingEntity $$1, double $$2, double $$3, double $$4) {
+        super(net.minecraft.world.entity.EntityType.WITHER_SKULL, $$1, $$2, $$3, $$4, $$0);
     }
 
     public float getInertia() {
@@ -24,52 +24,52 @@
         return false;
     }
 
-    public float getBlockExplosionResistance(net.minecraft.world.level.Explosion explosion, net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState blockState, net.minecraft.world.level.material.FluidState fluidState, float explosionPower) {
-        if (this.isDangerous() && net.minecraft.world.entity.boss.wither.WitherBoss.canDestroy((net.minecraft.world.level.block.state.BlockState)blockState)) {
-            return java.lang.Math.min((float)0.8f, (float)explosionPower);
+    public float getBlockExplosionResistance(net.minecraft.world.level.Explosion $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, net.minecraft.world.level.material.FluidState $$4, float $$5) {
+        if (this.isDangerous() && net.minecraft.world.entity.boss.wither.WitherBoss.canDestroy((net.minecraft.world.level.block.state.BlockState)$$3)) {
+            return java.lang.Math.min((float)0.8f, (float)$$5);
         }
-        return explosionPower;
+        return $$5;
     }
 
-    public void onHitEntity(net.minecraft.world.phys.EntityHitResult result) {
-        boolean bl;
-        super.onHitEntity(result);
+    public void onHitEntity(net.minecraft.world.phys.EntityHitResult $$0) {
+        boolean $$5;
+        super.onHitEntity($$0);
         if (this.level.isClientSide) {
             return;
         }
-        net.minecraft.world.entity.Entity entity = result.getEntity();
-        net.minecraft.world.entity.Entity entity2 = this.getOwner();
-        if (entity2 instanceof net.minecraft.world.entity.LivingEntity) {
-            net.minecraft.world.entity.LivingEntity livingEntity = (net.minecraft.world.entity.LivingEntity)entity2;
-            bl = entity.hurt(net.minecraft.world.damagesource.DamageSource.witherSkull((net.minecraft.world.entity.projectile.WitherSkull)this, (net.minecraft.world.entity.Entity)livingEntity), 8.0f);
-            if (bl) {
-                if (entity.isAlive()) {
-                    this.doEnchantDamageEffects(livingEntity, entity);
+        net.minecraft.world.entity.Entity $$1 = $$0.getEntity();
+        net.minecraft.world.entity.Entity $$2 = this.getOwner();
+        if ($$2 instanceof net.minecraft.world.entity.LivingEntity) {
+            net.minecraft.world.entity.LivingEntity $$3 = (net.minecraft.world.entity.LivingEntity)$$2;
+            boolean $$4 = $$1.hurt(net.minecraft.world.damagesource.DamageSource.witherSkull((net.minecraft.world.entity.projectile.WitherSkull)this, (net.minecraft.world.entity.Entity)$$3), 8.0f);
+            if ($$4) {
+                if ($$1.isAlive()) {
+                    this.doEnchantDamageEffects($$3, $$1);
                 } else {
-                    livingEntity.heal(5.0f);
+                    $$3.heal(5.0f);
                 }
             }
         } else {
-            bl = entity.hurt(net.minecraft.world.damagesource.DamageSource.MAGIC, 5.0f);
+            $$5 = $$1.hurt(net.minecraft.world.damagesource.DamageSource.MAGIC, 5.0f);
         }
-        if (bl && entity instanceof net.minecraft.world.entity.LivingEntity) {
-            int n = 0;
+        if ($$5 && $$1 instanceof net.minecraft.world.entity.LivingEntity) {
+            int $$6 = 0;
             if (this.level.getDifficulty() == net.minecraft.world.Difficulty.NORMAL) {
-                n = 10;
+                $$6 = 10;
             } else if (this.level.getDifficulty() == net.minecraft.world.Difficulty.HARD) {
-                n = 40;
+                $$6 = 40;
             }
-            if (n > 0) {
-                ((net.minecraft.world.entity.LivingEntity)entity).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.WITHER, 20 * n, 1), this.getEffectSource());
+            if ($$6 > 0) {
+                ((net.minecraft.world.entity.LivingEntity)$$1).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.WITHER, 20 * $$6, 1), this.getEffectSource());
             }
         }
     }
 
-    public void onHit(net.minecraft.world.phys.HitResult result) {
-        super.onHit(result);
+    public void onHit(net.minecraft.world.phys.HitResult $$0) {
+        super.onHit($$0);
         if (!this.level.isClientSide) {
-            net.minecraft.world.level.Explosion$BlockInteraction blockInteraction = this.level.getGameRules().getBoolean(net.minecraft.world.level.GameRules.RULE_MOBGRIEFING) ? net.minecraft.world.level.Explosion$BlockInteraction.DESTROY : net.minecraft.world.level.Explosion$BlockInteraction.NONE;
-            this.level.explode((net.minecraft.world.entity.Entity)this, this.getX(), this.getY(), this.getZ(), 1.0f, false, blockInteraction);
+            net.minecraft.world.level.Explosion$BlockInteraction $$1 = this.level.getGameRules().getBoolean(net.minecraft.world.level.GameRules.RULE_MOBGRIEFING) ? net.minecraft.world.level.Explosion$BlockInteraction.DESTROY : net.minecraft.world.level.Explosion$BlockInteraction.NONE;
+            this.level.explode((net.minecraft.world.entity.Entity)this, this.getX(), this.getY(), this.getZ(), 1.0f, false, $$1);
             this.discard();
         }
     }
@@ -78,7 +78,7 @@
         return false;
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
         return false;
     }
 
@@ -90,8 +90,8 @@
         return (java.lang.Boolean)this.entityData.get(DATA_DANGEROUS);
     }
 
-    public void setDangerous(boolean invulnerable) {
-        this.entityData.set(DATA_DANGEROUS, (java.lang.Object)invulnerable);
+    public void setDangerous(boolean $$0) {
+        this.entityData.set(DATA_DANGEROUS, (java.lang.Object)$$0);
     }
 
     public boolean shouldBurn() {