--- net/minecraft/world/inventory/AbstractFurnaceMenu.java
+++ net/minecraft/world/inventory/AbstractFurnaceMenu.java
@@ -21,39 +21,36 @@
     public net.minecraft.world.item.crafting.RecipeType<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> recipeType;
     public net.minecraft.world.inventory.RecipeBookType recipeBookType;
 
-    public net.minecraft.world.inventory.AbstractFurnaceMenu(net.minecraft.world.inventory.MenuType<?> menuType, net.minecraft.world.item.crafting.RecipeType<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> recipeType, net.minecraft.world.inventory.RecipeBookType recipeBookType, int containerId, net.minecraft.world.entity.player.Inventory playerInventory) {
-        this(menuType, recipeType, recipeBookType, containerId, playerInventory, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(3), (net.minecraft.world.inventory.ContainerData)new net.minecraft.world.inventory.SimpleContainerData(4));
+    public net.minecraft.world.inventory.AbstractFurnaceMenu(net.minecraft.world.inventory.MenuType<?> $$0, net.minecraft.world.item.crafting.RecipeType<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> $$1, net.minecraft.world.inventory.RecipeBookType $$2, int $$3, net.minecraft.world.entity.player.Inventory $$4) {
+        this($$0, $$1, $$2, $$3, $$4, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(3), (net.minecraft.world.inventory.ContainerData)new net.minecraft.world.inventory.SimpleContainerData(4));
     }
 
-    public net.minecraft.world.inventory.AbstractFurnaceMenu(net.minecraft.world.inventory.MenuType<?> menuType, net.minecraft.world.item.crafting.RecipeType<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> recipeType, net.minecraft.world.inventory.RecipeBookType recipeBookType, int containerId, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.world.Container container, net.minecraft.world.inventory.ContainerData data) {
-        super(menuType, containerId);
-        int n;
-        this.recipeType = recipeType;
-        this.recipeBookType = recipeBookType;
-        net.minecraft.world.inventory.AbstractFurnaceMenu.checkContainerSize((net.minecraft.world.Container)container, (int)3);
-        net.minecraft.world.inventory.AbstractFurnaceMenu.checkContainerDataCount((net.minecraft.world.inventory.ContainerData)data, (int)4);
-        this.container = container;
-        this.data = data;
-        this.level = playerInventory.player.level;
-        this.addSlot(new net.minecraft.world.inventory.Slot(container, 0, 56, 17));
-        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.FurnaceFuelSlot(this, container, 1, 56, 53));
-        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.FurnaceResultSlot(playerInventory.player, container, 2, 116, 35));
-        for (n = 0;
- n < 3; ++n) {
-            for (int i = 0; i < 9; ++i) {
-                this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)playerInventory, i + n * 9 + 9, 8 + i * 18, 84 + n * 18));
+    public net.minecraft.world.inventory.AbstractFurnaceMenu(net.minecraft.world.inventory.MenuType<?> $$0, net.minecraft.world.item.crafting.RecipeType<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> $$1, net.minecraft.world.inventory.RecipeBookType $$2, int $$3, net.minecraft.world.entity.player.Inventory $$4, net.minecraft.world.Container $$5, net.minecraft.world.inventory.ContainerData $$6) {
+        super($$0, $$3);
+        this.recipeType = $$1;
+        this.recipeBookType = $$2;
+        net.minecraft.world.inventory.AbstractFurnaceMenu.checkContainerSize((net.minecraft.world.Container)$$5, (int)3);
+        net.minecraft.world.inventory.AbstractFurnaceMenu.checkContainerDataCount((net.minecraft.world.inventory.ContainerData)$$6, (int)4);
+        this.container = $$5;
+        this.data = $$6;
+        this.level = $$4.player.level;
+        this.addSlot(new net.minecraft.world.inventory.Slot($$5, 0, 56, 17));
+        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.FurnaceFuelSlot(this, $$5, 1, 56, 53));
+        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.FurnaceResultSlot($$4.player, $$5, 2, 116, 35));
+        for (int $$7 = 0; $$7 < 3; ++$$7) {
+            for (int $$8 = 0; $$8 < 9; ++$$8) {
+                this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)$$4, $$8 + $$7 * 9 + 9, 8 + $$8 * 18, 84 + $$7 * 18));
             }
         }
-        for (n = 0;
- n < 9; ++n) {
-            this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)playerInventory, n, 8 + n * 18, 142));
+        for (int $$9 = 0; $$9 < 9; ++$$9) {
+            this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)$$4, $$9, 8 + $$9 * 18, 142));
         }
-        this.addDataSlots(data);
+        this.addDataSlots($$6);
     }
 
-    public void fillCraftSlotsStackedContents(net.minecraft.world.entity.player.StackedContents itemHelper) {
+    public void fillCraftSlotsStackedContents(net.minecraft.world.entity.player.StackedContents $$0) {
         if (this.container instanceof net.minecraft.world.inventory.StackedContentsCompatible) {
-            ((net.minecraft.world.inventory.StackedContentsCompatible)this.container).fillStackedContents(itemHelper);
+            ((net.minecraft.world.inventory.StackedContentsCompatible)this.container).fillStackedContents($$0);
         }
     }
 
@@ -62,8 +59,8 @@
         this.getSlot(2).set(net.minecraft.world.item.ItemStack.EMPTY);
     }
 
-    public boolean recipeMatches(net.minecraft.world.item.crafting.Recipe<? super net.minecraft.world.Container> recipe) {
-        return recipe.matches(this.container, this.level);
+    public boolean recipeMatches(net.minecraft.world.item.crafting.Recipe<? super net.minecraft.world.Container> $$0) {
+        return $$0.matches(this.container, this.level);
     }
 
     public int getResultSlotIndex() {
@@ -82,43 +79,43 @@
         return 3;
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
-        return this.container.stillValid(player);
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
+        return this.container.stillValid($$0);
     }
 
-    public net.minecraft.world.item.ItemStack quickMoveStack(net.minecraft.world.entity.player.Player player, int index) {
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.EMPTY;
-        net.minecraft.world.inventory.Slot slot = (net.minecraft.world.inventory.Slot)this.slots.get(index);
-        if (slot != null && slot.hasItem()) {
-            net.minecraft.world.item.ItemStack itemStack2 = slot.getItem();
-            itemStack = itemStack2.copy();
-            if (index == 2) {
-                if (!this.moveItemStackTo(itemStack2, 3, 39, true)) {
+    public net.minecraft.world.item.ItemStack quickMoveStack(net.minecraft.world.entity.player.Player $$0, int $$1) {
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        net.minecraft.world.inventory.Slot $$3 = (net.minecraft.world.inventory.Slot)this.slots.get($$1);
+        if ($$3 != null && $$3.hasItem()) {
+            net.minecraft.world.item.ItemStack $$4 = $$3.getItem();
+            $$2 = $$4.copy();
+            if ($$1 == 2) {
+                if (!this.moveItemStackTo($$4, 3, 39, true)) {
                     return net.minecraft.world.item.ItemStack.EMPTY;
                 }
-                slot.onQuickCraft(itemStack2, itemStack);
-            } else if (index == 1 || index == 0 ? !this.moveItemStackTo(itemStack2, 3, 39, false) : (this.canSmelt(itemStack2) ? !this.moveItemStackTo(itemStack2, 0, 1, false) : (this.isFuel(itemStack2) ? !this.moveItemStackTo(itemStack2, 1, 2, false) : (index >= 3 && index < 30 ? !this.moveItemStackTo(itemStack2, 30, 39, false) : index >= 30 && index < 39 && !this.moveItemStackTo(itemStack2, 3, 30, false))))) {
+                $$3.onQuickCraft($$4, $$2);
+            } else if ($$1 == 1 || $$1 == 0 ? !this.moveItemStackTo($$4, 3, 39, false) : (this.canSmelt($$4) ? !this.moveItemStackTo($$4, 0, 1, false) : (this.isFuel($$4) ? !this.moveItemStackTo($$4, 1, 2, false) : ($$1 >= 3 && $$1 < 30 ? !this.moveItemStackTo($$4, 30, 39, false) : $$1 >= 30 && $$1 < 39 && !this.moveItemStackTo($$4, 3, 30, false))))) {
                 return net.minecraft.world.item.ItemStack.EMPTY;
             }
-            if (itemStack2.isEmpty()) {
-                slot.set(net.minecraft.world.item.ItemStack.EMPTY);
+            if ($$4.isEmpty()) {
+                $$3.set(net.minecraft.world.item.ItemStack.EMPTY);
             } else {
-                slot.setChanged();
+                $$3.setChanged();
             }
-            if (itemStack2.getCount() == itemStack.getCount()) {
+            if ($$4.getCount() == $$2.getCount()) {
                 return net.minecraft.world.item.ItemStack.EMPTY;
             }
-            slot.onTake(player, itemStack2);
+            $$3.onTake($$0, $$4);
         }
-        return itemStack;
+        return $$2;
     }
 
-    public boolean canSmelt(net.minecraft.world.item.ItemStack stack) {
-        return this.level.getRecipeManager().getRecipeFor(this.recipeType, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(new net.minecraft.world.item.ItemStack[]{stack}), this.level).isPresent();
+    public boolean canSmelt(net.minecraft.world.item.ItemStack $$0) {
+        return this.level.getRecipeManager().getRecipeFor(this.recipeType, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(new net.minecraft.world.item.ItemStack[]{$$0}), this.level).isPresent();
     }
 
-    public boolean isFuel(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity.isFuel((net.minecraft.world.item.ItemStack)stack);
+    public boolean isFuel(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity.isFuel((net.minecraft.world.item.ItemStack)$$0);
     }
 
     public int getBurnProgress() {