--- net/minecraft/world/inventory/BlastFurnaceMenu.java
+++ net/minecraft/world/inventory/BlastFurnaceMenu.java
@@ -6,12 +6,12 @@
 public class net.minecraft.world.inventory.BlastFurnaceMenu
 extends net.minecraft.world.inventory.AbstractFurnaceMenu
  {
-    public net.minecraft.world.inventory.BlastFurnaceMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory) {
-        super(net.minecraft.world.inventory.MenuType.BLAST_FURNACE, net.minecraft.world.item.crafting.RecipeType.BLASTING, net.minecraft.world.inventory.RecipeBookType.BLAST_FURNACE, containerId, playerInventory);
+    public net.minecraft.world.inventory.BlastFurnaceMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        super(net.minecraft.world.inventory.MenuType.BLAST_FURNACE, net.minecraft.world.item.crafting.RecipeType.BLASTING, net.minecraft.world.inventory.RecipeBookType.BLAST_FURNACE, $$0, $$1);
     }
 
-    public net.minecraft.world.inventory.BlastFurnaceMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.world.Container blastFurnaceContainer, net.minecraft.world.inventory.ContainerData blastFurnaceData) {
-        super(net.minecraft.world.inventory.MenuType.BLAST_FURNACE, net.minecraft.world.item.crafting.RecipeType.BLASTING, net.minecraft.world.inventory.RecipeBookType.BLAST_FURNACE, containerId, playerInventory, blastFurnaceContainer, blastFurnaceData);
+    public net.minecraft.world.inventory.BlastFurnaceMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.world.Container $$2, net.minecraft.world.inventory.ContainerData $$3) {
+        super(net.minecraft.world.inventory.MenuType.BLAST_FURNACE, net.minecraft.world.item.crafting.RecipeType.BLASTING, net.minecraft.world.inventory.RecipeBookType.BLAST_FURNACE, $$0, $$1, $$2, $$3);
     }
 }
 