--- net/minecraft/world/inventory/BrewingStandMenu.java
+++ net/minecraft/world/inventory/BrewingStandMenu.java
@@ -20,64 +20,61 @@
     public net.minecraft.world.inventory.ContainerData brewingStandData;
     public net.minecraft.world.inventory.Slot ingredientSlot;
 
-    public net.minecraft.world.inventory.BrewingStandMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory) {
-        this(containerId, playerInventory, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(5), (net.minecraft.world.inventory.ContainerData)new net.minecraft.world.inventory.SimpleContainerData(2));
+    public net.minecraft.world.inventory.BrewingStandMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        this($$0, $$1, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(5), (net.minecraft.world.inventory.ContainerData)new net.minecraft.world.inventory.SimpleContainerData(2));
     }
 
-    public net.minecraft.world.inventory.BrewingStandMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.world.Container brewingStandContainer, net.minecraft.world.inventory.ContainerData brewingStandData) {
-        super(net.minecraft.world.inventory.MenuType.BREWING_STAND, containerId);
-        int n;
-        net.minecraft.world.inventory.BrewingStandMenu.checkContainerSize((net.minecraft.world.Container)brewingStandContainer, (int)5);
-        net.minecraft.world.inventory.BrewingStandMenu.checkContainerDataCount((net.minecraft.world.inventory.ContainerData)brewingStandData, (int)2);
-        this.brewingStand = brewingStandContainer;
-        this.brewingStandData = brewingStandData;
-        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$PotionSlot(brewingStandContainer, 0, 56, 51));
-        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$PotionSlot(brewingStandContainer, 1, 79, 58));
-        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$PotionSlot(brewingStandContainer, 2, 102, 51));
-        this.ingredientSlot = this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$IngredientsSlot(brewingStandContainer, 3, 79, 17));
-        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$FuelSlot(brewingStandContainer, 4, 17, 17));
-        this.addDataSlots(brewingStandData);
-        for (n = 0;
- n < 3; ++n) {
-            for (int i = 0; i < 9; ++i) {
-                this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)playerInventory, i + n * 9 + 9, 8 + i * 18, 84 + n * 18));
+    public net.minecraft.world.inventory.BrewingStandMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.world.Container $$2, net.minecraft.world.inventory.ContainerData $$3) {
+        super(net.minecraft.world.inventory.MenuType.BREWING_STAND, $$0);
+        net.minecraft.world.inventory.BrewingStandMenu.checkContainerSize((net.minecraft.world.Container)$$2, (int)5);
+        net.minecraft.world.inventory.BrewingStandMenu.checkContainerDataCount((net.minecraft.world.inventory.ContainerData)$$3, (int)2);
+        this.brewingStand = $$2;
+        this.brewingStandData = $$3;
+        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$PotionSlot($$2, 0, 56, 51));
+        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$PotionSlot($$2, 1, 79, 58));
+        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$PotionSlot($$2, 2, 102, 51));
+        this.ingredientSlot = this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$IngredientsSlot($$2, 3, 79, 17));
+        this.addSlot((net.minecraft.world.inventory.Slot)new net.minecraft.world.inventory.BrewingStandMenu$FuelSlot($$2, 4, 17, 17));
+        this.addDataSlots($$3);
+        for (int $$4 = 0; $$4 < 3; ++$$4) {
+            for (int $$5 = 0; $$5 < 9; ++$$5) {
+                this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)$$1, $$5 + $$4 * 9 + 9, 8 + $$5 * 18, 84 + $$4 * 18));
             }
         }
-        for (n = 0;
- n < 9; ++n) {
-            this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)playerInventory, n, 8 + n * 18, 142));
+        for (int $$6 = 0; $$6 < 9; ++$$6) {
+            this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)$$1, $$6, 8 + $$6 * 18, 142));
         }
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
-        return this.brewingStand.stillValid(player);
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
+        return this.brewingStand.stillValid($$0);
     }
 
-    public net.minecraft.world.item.ItemStack quickMoveStack(net.minecraft.world.entity.player.Player player, int index) {
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.EMPTY;
-        net.minecraft.world.inventory.Slot slot = (net.minecraft.world.inventory.Slot)this.slots.get(index);
-        if (slot != null && slot.hasItem()) {
-            net.minecraft.world.item.ItemStack itemStack2 = slot.getItem();
-            itemStack = itemStack2.copy();
-            if (index >= 0 && index <= 2 || index == 3 || index == 4) {
-                if (!this.moveItemStackTo(itemStack2, 5, 41, true)) {
+    public net.minecraft.world.item.ItemStack quickMoveStack(net.minecraft.world.entity.player.Player $$0, int $$1) {
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        net.minecraft.world.inventory.Slot $$3 = (net.minecraft.world.inventory.Slot)this.slots.get($$1);
+        if ($$3 != null && $$3.hasItem()) {
+            net.minecraft.world.item.ItemStack $$4 = $$3.getItem();
+            $$2 = $$4.copy();
+            if ($$1 >= 0 && $$1 <= 2 || $$1 == 3 || $$1 == 4) {
+                if (!this.moveItemStackTo($$4, 5, 41, true)) {
                     return net.minecraft.world.item.ItemStack.EMPTY;
                 }
-                slot.onQuickCraft(itemStack2, itemStack);
-            } else if (net.minecraft.world.inventory.BrewingStandMenu$FuelSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)itemStack) ? this.moveItemStackTo(itemStack2, 4, 5, false) || this.ingredientSlot.mayPlace(itemStack2) && !this.moveItemStackTo(itemStack2, 3, 4, false) : (this.ingredientSlot.mayPlace(itemStack2) ? !this.moveItemStackTo(itemStack2, 3, 4, false) : (net.minecraft.world.inventory.BrewingStandMenu$PotionSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)itemStack) && itemStack.getCount() == 1 ? !this.moveItemStackTo(itemStack2, 0, 3, false) : (index >= 5 && index < 32 ? !this.moveItemStackTo(itemStack2, 32, 41, false) : (index >= 32 && index < 41 ? !this.moveItemStackTo(itemStack2, 5, 32, false) : !this.moveItemStackTo(itemStack2, 5, 41, false)))))) {
+                $$3.onQuickCraft($$4, $$2);
+            } else if (net.minecraft.world.inventory.BrewingStandMenu$FuelSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)$$2) ? this.moveItemStackTo($$4, 4, 5, false) || this.ingredientSlot.mayPlace($$4) && !this.moveItemStackTo($$4, 3, 4, false) : (this.ingredientSlot.mayPlace($$4) ? !this.moveItemStackTo($$4, 3, 4, false) : (net.minecraft.world.inventory.BrewingStandMenu$PotionSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)$$2) && $$2.getCount() == 1 ? !this.moveItemStackTo($$4, 0, 3, false) : ($$1 >= 5 && $$1 < 32 ? !this.moveItemStackTo($$4, 32, 41, false) : ($$1 >= 32 && $$1 < 41 ? !this.moveItemStackTo($$4, 5, 32, false) : !this.moveItemStackTo($$4, 5, 41, false)))))) {
                 return net.minecraft.world.item.ItemStack.EMPTY;
             }
-            if (itemStack2.isEmpty()) {
-                slot.set(net.minecraft.world.item.ItemStack.EMPTY);
+            if ($$4.isEmpty()) {
+                $$3.set(net.minecraft.world.item.ItemStack.EMPTY);
             } else {
-                slot.setChanged();
+                $$3.setChanged();
             }
-            if (itemStack2.getCount() == itemStack.getCount()) {
+            if ($$4.getCount() == $$2.getCount()) {
                 return net.minecraft.world.item.ItemStack.EMPTY;
             }
-            slot.onTake(player, itemStack2);
+            $$3.onTake($$0, $$4);
         }
-        return itemStack;
+        return $$2;
     }
 
     public int getFuel() {