--- net/minecraft/world/inventory/BrewingStandMenu$PotionSlot.java
+++ net/minecraft/world/inventory/BrewingStandMenu$PotionSlot.java
@@ -10,8 +10,8 @@
         super($$0, $$1, $$2, $$3);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.inventory.BrewingStandMenu$PotionSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)stack);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.inventory.BrewingStandMenu$PotionSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)$$0);
     }
 
     public int getMaxStackSize() {
@@ -26,8 +26,8 @@
         super.onTake($$0, $$1);
     }
 
-    public static boolean mayPlaceItem(net.minecraft.world.item.ItemStack stack) {
-        return stack.is(net.minecraft.world.item.Items.POTION) || stack.is(net.minecraft.world.item.Items.SPLASH_POTION) || stack.is(net.minecraft.world.item.Items.LINGERING_POTION) || stack.is(net.minecraft.world.item.Items.GLASS_BOTTLE);
+    public static boolean mayPlaceItem(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(net.minecraft.world.item.Items.POTION) || $$0.is(net.minecraft.world.item.Items.SPLASH_POTION) || $$0.is(net.minecraft.world.item.Items.LINGERING_POTION) || $$0.is(net.minecraft.world.item.Items.GLASS_BOTTLE);
     }
 }
 