--- net/minecraft/world/inventory/BrewingStandMenu$IngredientsSlot.java
+++ net/minecraft/world/inventory/BrewingStandMenu$IngredientsSlot.java
@@ -10,8 +10,8 @@
         super($$0, $$1, $$2, $$3);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.item.alchemy.PotionBrewing.isIngredient((net.minecraft.world.item.ItemStack)stack);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.item.alchemy.PotionBrewing.isIngredient((net.minecraft.world.item.ItemStack)$$0);
     }
 
     public int getMaxStackSize() {