--- net/minecraft/world/inventory/BrewingStandMenu$FuelSlot.java
+++ net/minecraft/world/inventory/BrewingStandMenu$FuelSlot.java
@@ -10,12 +10,12 @@
         super($$0, $$1, $$2, $$3);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.inventory.BrewingStandMenu$FuelSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)stack);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.inventory.BrewingStandMenu$FuelSlot.mayPlaceItem((net.minecraft.world.item.ItemStack)$$0);
     }
 
-    public static boolean mayPlaceItem(net.minecraft.world.item.ItemStack itemStack) {
-        return itemStack.is(net.minecraft.world.item.Items.BLAZE_POWDER);
+    public static boolean mayPlaceItem(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(net.minecraft.world.item.Items.BLAZE_POWDER);
     }
 
     public int getMaxStackSize() {