--- net/minecraft/world/inventory/CartographyTableMenu$4.java
+++ net/minecraft/world/inventory/CartographyTableMenu$4.java
@@ -13,8 +13,8 @@
         super($$1, $$2, $$3, $$4);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return stack.is(net.minecraft.world.item.Items.PAPER) || stack.is(net.minecraft.world.item.Items.MAP) || stack.is(net.minecraft.world.item.Items.GLASS_PANE);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(net.minecraft.world.item.Items.PAPER) || $$0.is(net.minecraft.world.item.Items.MAP) || $$0.is(net.minecraft.world.item.Items.GLASS_PANE);
     }
 }
 