--- net/minecraft/world/inventory/ChestMenu.java
+++ net/minecraft/world/inventory/ChestMenu.java
@@ -10,96 +10,89 @@
     public net.minecraft.world.Container container;
     public int containerRows;
 
-    public net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType<?> type, int containerId, net.minecraft.world.entity.player.Inventory playerInventory, int rows) {
-        this(type, containerId, playerInventory, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(9 * rows), rows);
+    public net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType<?> $$0, int $$1, net.minecraft.world.entity.player.Inventory $$2, int $$3) {
+        this($$0, $$1, $$2, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(9 * $$3), $$3);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu oneRow(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x1, id, player, 1);
+    public static net.minecraft.world.inventory.ChestMenu oneRow(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x1, $$0, $$1, 1);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu twoRows(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x2, id, player, 2);
+    public static net.minecraft.world.inventory.ChestMenu twoRows(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x2, $$0, $$1, 2);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu threeRows(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x3, id, player, 3);
+    public static net.minecraft.world.inventory.ChestMenu threeRows(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x3, $$0, $$1, 3);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu fourRows(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x4, id, player, 4);
+    public static net.minecraft.world.inventory.ChestMenu fourRows(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x4, $$0, $$1, 4);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu fiveRows(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x5, id, player, 5);
+    public static net.minecraft.world.inventory.ChestMenu fiveRows(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x5, $$0, $$1, 5);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu sixRows(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x6, id, player, 6);
+    public static net.minecraft.world.inventory.ChestMenu sixRows(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x6, $$0, $$1, 6);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu threeRows(int id, net.minecraft.world.entity.player.Inventory player, net.minecraft.world.Container blockEntity) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x3, id, player, blockEntity, 3);
+    public static net.minecraft.world.inventory.ChestMenu threeRows(int $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.world.Container $$2) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x3, $$0, $$1, $$2, 3);
     }
 
-    public static net.minecraft.world.inventory.ChestMenu sixRows(int id, net.minecraft.world.entity.player.Inventory player, net.minecraft.world.Container blockEntity) {
-        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x6, id, player, blockEntity, 6);
+    public static net.minecraft.world.inventory.ChestMenu sixRows(int $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.world.Container $$2) {
+        return new net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType.GENERIC_9x6, $$0, $$1, $$2, 6);
     }
 
-    public net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType<?> type, int containerId, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.world.Container container, int rows) {
-        super(type, containerId);
-        int n;
-        int n2;
-        net.minecraft.world.inventory.ChestMenu.checkContainerSize((net.minecraft.world.Container)container, (int)(rows * 9));
-        this.container = container;
-        this.containerRows = rows;
-        container.startOpen(playerInventory.player);
-        int n3 = (this.containerRows - 4) * 18;
-        for (n2 = 0;
- n2 < this.containerRows; ++n2) {
-            for (n = 0;
- n < 9; ++n) {
-                this.addSlot(new net.minecraft.world.inventory.Slot(container, n + n2 * 9, 8 + n * 18, 18 + n2 * 18));
+    public net.minecraft.world.inventory.ChestMenu(net.minecraft.world.inventory.MenuType<?> $$0, int $$1, net.minecraft.world.entity.player.Inventory $$2, net.minecraft.world.Container $$3, int $$4) {
+        super($$0, $$1);
+        net.minecraft.world.inventory.ChestMenu.checkContainerSize((net.minecraft.world.Container)$$3, (int)($$4 * 9));
+        this.container = $$3;
+        this.containerRows = $$4;
+        $$3.startOpen($$2.player);
+        int $$5 = (this.containerRows - 4) * 18;
+        for (int $$6 = 0; $$6 < this.containerRows; ++$$6) {
+            for (int $$7 = 0; $$7 < 9; ++$$7) {
+                this.addSlot(new net.minecraft.world.inventory.Slot($$3, $$7 + $$6 * 9, 8 + $$7 * 18, 18 + $$6 * 18));
             }
         }
-        for (n2 = 0;
- n2 < 3; ++n2) {
-            for (n = 0;
- n < 9; ++n) {
-                this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)playerInventory, n + n2 * 9 + 9, 8 + n * 18, 103 + n2 * 18 + n3));
+        for (int $$8 = 0; $$8 < 3; ++$$8) {
+            for (int $$9 = 0; $$9 < 9; ++$$9) {
+                this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)$$2, $$9 + $$8 * 9 + 9, 8 + $$9 * 18, 103 + $$8 * 18 + $$5));
             }
         }
-        for (n2 = 0;
- n2 < 9; ++n2) {
-            this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)playerInventory, n2, 8 + n2 * 18, 161 + n3));
+        for (int $$10 = 0; $$10 < 9; ++$$10) {
+            this.addSlot(new net.minecraft.world.inventory.Slot((net.minecraft.world.Container)$$2, $$10, 8 + $$10 * 18, 161 + $$5));
         }
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
-        return this.container.stillValid(player);
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
+        return this.container.stillValid($$0);
     }
 
-    public net.minecraft.world.item.ItemStack quickMoveStack(net.minecraft.world.entity.player.Player player, int index) {
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.item.ItemStack.EMPTY;
-        net.minecraft.world.inventory.Slot slot = (net.minecraft.world.inventory.Slot)this.slots.get(index);
-        if (slot != null && slot.hasItem()) {
-            net.minecraft.world.item.ItemStack itemStack2 = slot.getItem();
-            itemStack = itemStack2.copy();
-            if (index < this.containerRows * 9 ? !this.moveItemStackTo(itemStack2, this.containerRows * 9, this.slots.size(), true) : !this.moveItemStackTo(itemStack2, 0, this.containerRows * 9, false)) {
+    public net.minecraft.world.item.ItemStack quickMoveStack(net.minecraft.world.entity.player.Player $$0, int $$1) {
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.item.ItemStack.EMPTY;
+        net.minecraft.world.inventory.Slot $$3 = (net.minecraft.world.inventory.Slot)this.slots.get($$1);
+        if ($$3 != null && $$3.hasItem()) {
+            net.minecraft.world.item.ItemStack $$4 = $$3.getItem();
+            $$2 = $$4.copy();
+            if ($$1 < this.containerRows * 9 ? !this.moveItemStackTo($$4, this.containerRows * 9, this.slots.size(), true) : !this.moveItemStackTo($$4, 0, this.containerRows * 9, false)) {
                 return net.minecraft.world.item.ItemStack.EMPTY;
             }
-            if (itemStack2.isEmpty()) {
-                slot.set(net.minecraft.world.item.ItemStack.EMPTY);
+            if ($$4.isEmpty()) {
+                $$3.set(net.minecraft.world.item.ItemStack.EMPTY);
             } else {
-                slot.setChanged();
+                $$3.setChanged();
             }
         }
-        return itemStack;
+        return $$2;
     }
 
-    public void removed(net.minecraft.world.entity.player.Player player) {
-        super.removed(player);
-        this.container.stopOpen(player);
+    public void removed(net.minecraft.world.entity.player.Player $$0) {
+        super.removed($$0);
+        this.container.stopOpen($$0);
     }
 
     public net.minecraft.world.Container getContainer() {