--- net/minecraft/world/inventory/ContainerLevelAccess$2.java
+++ net/minecraft/world/inventory/ContainerLevelAccess$2.java
@@ -14,8 +14,8 @@
         this.val$pos = blockPos;
     }
 
-    public <T> java.util.Optional<T> evaluate(java.util.function.BiFunction<net.minecraft.world.level.Level, net.minecraft.core.BlockPos, T> levelPosConsumer) {
-        return java.util.Optional.of((java.lang.Object)levelPosConsumer.apply((java.lang.Object)this.val$level, (java.lang.Object)this.val$pos));
+    public <T> java.util.Optional<T> evaluate(java.util.function.BiFunction<net.minecraft.world.level.Level, net.minecraft.core.BlockPos, T> $$0) {
+        return java.util.Optional.of((java.lang.Object)$$0.apply((java.lang.Object)this.val$level, (java.lang.Object)this.val$pos));
     }
 }
 