--- net/minecraft/world/inventory/CraftingContainer.java
+++ net/minecraft/world/inventory/CraftingContainer.java
@@ -12,11 +12,11 @@
     public int height;
     public net.minecraft.world.inventory.AbstractContainerMenu menu;
 
-    public net.minecraft.world.inventory.CraftingContainer(net.minecraft.world.inventory.AbstractContainerMenu menu, int width, int height) {
-        this.items = net.minecraft.core.NonNullList.withSize((int)(width * height), (java.lang.Object)net.minecraft.world.item.ItemStack.EMPTY);
-        this.menu = menu;
-        this.width = width;
-        this.height = height;
+    public net.minecraft.world.inventory.CraftingContainer(net.minecraft.world.inventory.AbstractContainerMenu $$0, int $$1, int $$2) {
+        this.items = net.minecraft.core.NonNullList.withSize((int)($$1 * $$2), (java.lang.Object)net.minecraft.world.item.ItemStack.EMPTY);
+        this.menu = $$0;
+        this.width = $$1;
+        this.height = $$2;
     }
 
     public int getContainerSize() {
@@ -31,34 +31,34 @@
         return true;
     }
 
-    public net.minecraft.world.item.ItemStack getItem(int index) {
-        if (index >= this.getContainerSize()) {
+    public net.minecraft.world.item.ItemStack getItem(int $$0) {
+        if ($$0 >= this.getContainerSize()) {
             return net.minecraft.world.item.ItemStack.EMPTY;
         }
-        return (net.minecraft.world.item.ItemStack)this.items.get(index);
+        return (net.minecraft.world.item.ItemStack)this.items.get($$0);
     }
 
-    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int index) {
-        return net.minecraft.world.ContainerHelper.takeItem((java.util.List)this.items, (int)index);
+    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int $$0) {
+        return net.minecraft.world.ContainerHelper.takeItem((java.util.List)this.items, (int)$$0);
     }
 
-    public net.minecraft.world.item.ItemStack removeItem(int index, int count) {
-        net.minecraft.world.item.ItemStack itemStack = net.minecraft.world.ContainerHelper.removeItem((java.util.List)this.items, (int)index, (int)count);
-        if (!itemStack.isEmpty()) {
+    public net.minecraft.world.item.ItemStack removeItem(int $$0, int $$1) {
+        net.minecraft.world.item.ItemStack $$2 = net.minecraft.world.ContainerHelper.removeItem((java.util.List)this.items, (int)$$0, (int)$$1);
+        if (!$$2.isEmpty()) {
             this.menu.slotsChanged((net.minecraft.world.Container)this);
         }
-        return itemStack;
+        return $$2;
     }
 
-    public void setItem(int index, net.minecraft.world.item.ItemStack stack) {
-        this.items.set(index, (java.lang.Object)stack);
+    public void setItem(int $$0, net.minecraft.world.item.ItemStack $$1) {
+        this.items.set($$0, (java.lang.Object)$$1);
         this.menu.slotsChanged((net.minecraft.world.Container)this);
     }
 
     public void setChanged() {
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
         return true;
     }
 
@@ -74,9 +74,9 @@
         return this.width;
     }
 
-    public void fillStackedContents(net.minecraft.world.entity.player.StackedContents helper) {
-        for (net.minecraft.world.item.ItemStack itemStack : this.items) {
-            helper.accountSimpleStack(itemStack);
+    public void fillStackedContents(net.minecraft.world.entity.player.StackedContents $$0) {
+        for (net.minecraft.world.item.ItemStack $$1 : this.items) {
+            $$0.accountSimpleStack($$1);
         }
     }
 }