--- net/minecraft/world/inventory/EnchantmentMenu$3.java
+++ net/minecraft/world/inventory/EnchantmentMenu$3.java
@@ -13,8 +13,8 @@
         super($$1, $$2, $$3, $$4);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return stack.is(net.minecraft.world.item.Items.LAPIS_LAZULI);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(net.minecraft.world.item.Items.LAPIS_LAZULI);
     }
 }
 