--- net/minecraft/world/inventory/FurnaceMenu.java
+++ net/minecraft/world/inventory/FurnaceMenu.java
@@ -6,12 +6,12 @@
 public class net.minecraft.world.inventory.FurnaceMenu
 extends net.minecraft.world.inventory.AbstractFurnaceMenu
  {
-    public net.minecraft.world.inventory.FurnaceMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory) {
-        super(net.minecraft.world.inventory.MenuType.FURNACE, net.minecraft.world.item.crafting.RecipeType.SMELTING, net.minecraft.world.inventory.RecipeBookType.FURNACE, containerId, playerInventory);
+    public net.minecraft.world.inventory.FurnaceMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        super(net.minecraft.world.inventory.MenuType.FURNACE, net.minecraft.world.item.crafting.RecipeType.SMELTING, net.minecraft.world.inventory.RecipeBookType.FURNACE, $$0, $$1);
     }
 
-    public net.minecraft.world.inventory.FurnaceMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.world.Container furnaceContainer, net.minecraft.world.inventory.ContainerData furnaceData) {
-        super(net.minecraft.world.inventory.MenuType.FURNACE, net.minecraft.world.item.crafting.RecipeType.SMELTING, net.minecraft.world.inventory.RecipeBookType.FURNACE, containerId, playerInventory, furnaceContainer, furnaceData);
+    public net.minecraft.world.inventory.FurnaceMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.world.Container $$2, net.minecraft.world.inventory.ContainerData $$3) {
+        super(net.minecraft.world.inventory.MenuType.FURNACE, net.minecraft.world.item.crafting.RecipeType.SMELTING, net.minecraft.world.inventory.RecipeBookType.FURNACE, $$0, $$1, $$2, $$3);
     }
 }
 