--- net/minecraft/world/inventory/FurnaceResultSlot.java
+++ net/minecraft/world/inventory/FurnaceResultSlot.java
@@ -9,20 +9,20 @@
     public net.minecraft.world.entity.player.Player player;
     public int removeCount;
 
-    public net.minecraft.world.inventory.FurnaceResultSlot(net.minecraft.world.entity.player.Player player, net.minecraft.world.Container container, int slot, int xPosition, int yPosition) {
-        super(container, slot, xPosition, yPosition);
-        this.player = player;
+    public net.minecraft.world.inventory.FurnaceResultSlot(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.Container $$1, int $$2, int $$3, int $$4) {
+        super($$1, $$2, $$3, $$4);
+        this.player = $$0;
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
         return false;
     }
 
-    public net.minecraft.world.item.ItemStack remove(int amount) {
+    public net.minecraft.world.item.ItemStack remove(int $$0) {
         if (this.hasItem()) {
-            this.removeCount += java.lang.Math.min((int)amount, (int)this.getItem().getCount());
+            this.removeCount += java.lang.Math.min((int)$$0, (int)this.getItem().getCount());
         }
-        return super.remove(amount);
+        return super.remove($$0);
     }
 
     public void onTake(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.item.ItemStack $$1) {
@@ -30,13 +30,13 @@
         super.onTake($$0, $$1);
     }
 
-    public void onQuickCraft(net.minecraft.world.item.ItemStack stack, int amount) {
-        this.removeCount += amount;
-        this.checkTakeAchievements(stack);
+    public void onQuickCraft(net.minecraft.world.item.ItemStack $$0, int $$1) {
+        this.removeCount += $$1;
+        this.checkTakeAchievements($$0);
     }
 
-    public void checkTakeAchievements(net.minecraft.world.item.ItemStack stack) {
-        stack.onCraftedBy(this.player.level, this.player, this.removeCount);
+    public void checkTakeAchievements(net.minecraft.world.item.ItemStack $$0) {
+        $$0.onCraftedBy(this.player.level, this.player, this.removeCount);
         if (this.player instanceof net.minecraft.server.level.ServerPlayer && this.container instanceof net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity) {
             ((net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity)this.container).awardUsedRecipesAndPopExperience((net.minecraft.server.level.ServerPlayer)this.player);
         }