--- net/minecraft/world/inventory/FurnaceFuelSlot.java
+++ net/minecraft/world/inventory/FurnaceFuelSlot.java
@@ -8,21 +8,21 @@
  {
     public net.minecraft.world.inventory.AbstractFurnaceMenu menu;
 
-    public net.minecraft.world.inventory.FurnaceFuelSlot(net.minecraft.world.inventory.AbstractFurnaceMenu furnaceMenu, net.minecraft.world.Container furnaceContainer, int slot, int xPosition, int yPosition) {
-        super(furnaceContainer, slot, xPosition, yPosition);
-        this.menu = furnaceMenu;
+    public net.minecraft.world.inventory.FurnaceFuelSlot(net.minecraft.world.inventory.AbstractFurnaceMenu $$0, net.minecraft.world.Container $$1, int $$2, int $$3, int $$4) {
+        super($$1, $$2, $$3, $$4);
+        this.menu = $$0;
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return this.menu.isFuel(stack) || net.minecraft.world.inventory.FurnaceFuelSlot.isBucket((net.minecraft.world.item.ItemStack)stack);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return this.menu.isFuel($$0) || net.minecraft.world.inventory.FurnaceFuelSlot.isBucket((net.minecraft.world.item.ItemStack)$$0);
     }
 
-    public int getMaxStackSize(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.inventory.FurnaceFuelSlot.isBucket((net.minecraft.world.item.ItemStack)stack) ? 1 : super.getMaxStackSize(stack);
+    public int getMaxStackSize(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.inventory.FurnaceFuelSlot.isBucket((net.minecraft.world.item.ItemStack)$$0) ? 1 : super.getMaxStackSize($$0);
     }
 
-    public static boolean isBucket(net.minecraft.world.item.ItemStack stack) {
-        return stack.is(net.minecraft.world.item.Items.BUCKET);
+    public static boolean isBucket(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.is(net.minecraft.world.item.Items.BUCKET);
     }
 }
 