--- net/minecraft/world/inventory/GrindstoneMenu$2.java
+++ net/minecraft/world/inventory/GrindstoneMenu$2.java
@@ -13,8 +13,8 @@
         super($$1, $$2, $$3, $$4);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return stack.isDamageableItem() || stack.is(net.minecraft.world.item.Items.ENCHANTED_BOOK) || stack.isEnchanted();
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.isDamageableItem() || $$0.is(net.minecraft.world.item.Items.ENCHANTED_BOOK) || $$0.isEnchanted();
     }
 }
 