--- net/minecraft/world/inventory/HorseInventoryMenu$2.java
+++ net/minecraft/world/inventory/HorseInventoryMenu$2.java
@@ -15,8 +15,8 @@
         super($$1, $$2, $$3, $$4);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return this.val$horse.isArmor(stack);
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return this.val$horse.isArmor($$0);
     }
 
     public boolean isActive() {