--- net/minecraft/world/inventory/LecternMenu.java
+++ net/minecraft/world/inventory/LecternMenu.java
@@ -15,45 +15,45 @@
     public net.minecraft.world.Container lectern;
     public net.minecraft.world.inventory.ContainerData lecternData;
 
-    public net.minecraft.world.inventory.LecternMenu(int containerId) {
-        this(containerId, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(1), (net.minecraft.world.inventory.ContainerData)new net.minecraft.world.inventory.SimpleContainerData(1));
+    public net.minecraft.world.inventory.LecternMenu(int $$0) {
+        this($$0, (net.minecraft.world.Container)new net.minecraft.world.SimpleContainer(1), (net.minecraft.world.inventory.ContainerData)new net.minecraft.world.inventory.SimpleContainerData(1));
     }
 
-    public net.minecraft.world.inventory.LecternMenu(int containerId, net.minecraft.world.Container lectern, net.minecraft.world.inventory.ContainerData lecternData) {
-        super(net.minecraft.world.inventory.MenuType.LECTERN, containerId);
-        net.minecraft.world.inventory.LecternMenu.checkContainerSize((net.minecraft.world.Container)lectern, (int)1);
-        net.minecraft.world.inventory.LecternMenu.checkContainerDataCount((net.minecraft.world.inventory.ContainerData)lecternData, (int)1);
-        this.lectern = lectern;
-        this.lecternData = lecternData;
+    public net.minecraft.world.inventory.LecternMenu(int $$0, net.minecraft.world.Container $$1, net.minecraft.world.inventory.ContainerData $$2) {
+        super(net.minecraft.world.inventory.MenuType.LECTERN, $$0);
+        net.minecraft.world.inventory.LecternMenu.checkContainerSize((net.minecraft.world.Container)$$1, (int)1);
+        net.minecraft.world.inventory.LecternMenu.checkContainerDataCount((net.minecraft.world.inventory.ContainerData)$$2, (int)1);
+        this.lectern = $$1;
+        this.lecternData = $$2;
         this.addSlot((net.minecraft.world.inventory.Slot)new /* Unavailable Anonymous Inner Class!! */);
-        this.addDataSlots(lecternData);
+        this.addDataSlots($$2);
     }
 
-    public boolean clickMenuButton(net.minecraft.world.entity.player.Player player, int id) {
-        if (id >= 100) {
-            int n = id - 100;
-            this.setData(0, n);
+    public boolean clickMenuButton(net.minecraft.world.entity.player.Player $$0, int $$1) {
+        if ($$1 >= 100) {
+            int $$2 = $$1 - 100;
+            this.setData(0, $$2);
             return true;
         }
-        switch (id) {
+        switch ($$1) {
             case 2: {
-                int n = this.lecternData.get(0);
-                this.setData(0, n + 1);
+                int $$3 = this.lecternData.get(0);
+                this.setData(0, $$3 + 1);
                 return true;
             }
             case 1: {
-                int n = this.lecternData.get(0);
-                this.setData(0, n - 1);
+                int $$4 = this.lecternData.get(0);
+                this.setData(0, $$4 - 1);
                 return true;
             }
             case 3: {
-                if (!player.mayBuild()) {
+                if (!$$0.mayBuild()) {
                     return false;
                 }
-                net.minecraft.world.item.ItemStack itemStack = this.lectern.removeItemNoUpdate(0);
+                net.minecraft.world.item.ItemStack $$5 = this.lectern.removeItemNoUpdate(0);
                 this.lectern.setChanged();
-                if (!player.getInventory().add(itemStack)) {
-                    player.drop(itemStack, false);
+                if (!$$0.getInventory().add($$5)) {
+                    $$0.drop($$5, false);
                 }
                 return true;
             }
@@ -65,13 +65,13 @@
         return net.minecraft.world.item.ItemStack.EMPTY;
     }
 
-    public void setData(int id, int data) {
-        super.setData(id, data);
+    public void setData(int $$0, int $$1) {
+        super.setData($$0, $$1);
         this.broadcastChanges();
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
-        return this.lectern.stillValid(player);
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
+        return this.lectern.stillValid($$0);
     }
 
     public net.minecraft.world.item.ItemStack getBook() {