--- net/minecraft/world/inventory/LoomMenu$6.java
+++ net/minecraft/world/inventory/LoomMenu$6.java
@@ -15,24 +15,24 @@
         super($$1, $$2, $$3, $$4);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
         return false;
     }
 
-    public void onTake(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.item.ItemStack $$1) {
+    public void onTake(net.minecraft.world.entity.player.Player $$02, net.minecraft.world.item.ItemStack $$12) {
         this.this$0.bannerSlot.remove(1);
         this.this$0.dyeSlot.remove(1);
         if (!this.this$0.bannerSlot.hasItem() || !this.this$0.dyeSlot.hasItem()) {
             this.this$0.selectedBannerPatternIndex.set(-1);
         }
-        this.val$access.execute((level, blockPos) -> {
-            long l = level.getGameTime();
-            if (this.this$0.lastSoundTime != l) {
-                level.playSound(null, blockPos, net.minecraft.sounds.SoundEvents.UI_LOOM_TAKE_RESULT, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
-                this.this$0.lastSoundTime = l;
+        this.val$access.execute(($$0, $$1) -> {
+            long $$2 = $$0.getGameTime();
+            if (this.this$0.lastSoundTime != $$2) {
+                $$0.playSound(null, $$1, net.minecraft.sounds.SoundEvents.UI_LOOM_TAKE_RESULT, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
+                this.this$0.lastSoundTime = $$2;
             }
         });
-        super.onTake($$0, $$1);
+        super.onTake($$02, $$12);
     }
 }
 