--- net/minecraft/world/inventory/LoomMenu$3.java
+++ net/minecraft/world/inventory/LoomMenu$3.java
@@ -13,8 +13,8 @@
         super($$1, $$2, $$3, $$4);
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
-        return stack.getItem() instanceof net.minecraft.world.item.BannerItem;
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
+        return $$0.getItem() instanceof net.minecraft.world.item.BannerItem;
     }
 }
 