--- net/minecraft/world/inventory/MerchantResultSlot.java
+++ net/minecraft/world/inventory/MerchantResultSlot.java
@@ -11,31 +11,31 @@
     public int removeCount;
     public net.minecraft.world.item.trading.Merchant merchant;
 
-    public net.minecraft.world.inventory.MerchantResultSlot(net.minecraft.world.entity.player.Player player, net.minecraft.world.item.trading.Merchant merchant, net.minecraft.world.inventory.MerchantContainer slots, int slot, int xPosition, int yPosition) {
-        super((net.minecraft.world.Container)slots, slot, xPosition, yPosition);
-        this.player = player;
-        this.merchant = merchant;
-        this.slots = slots;
+    public net.minecraft.world.inventory.MerchantResultSlot(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.item.trading.Merchant $$1, net.minecraft.world.inventory.MerchantContainer $$2, int $$3, int $$4, int $$5) {
+        super((net.minecraft.world.Container)$$2, $$3, $$4, $$5);
+        this.player = $$0;
+        this.merchant = $$1;
+        this.slots = $$2;
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
         return false;
     }
 
-    public net.minecraft.world.item.ItemStack remove(int amount) {
+    public net.minecraft.world.item.ItemStack remove(int $$0) {
         if (this.hasItem()) {
-            this.removeCount += java.lang.Math.min((int)amount, (int)this.getItem().getCount());
+            this.removeCount += java.lang.Math.min((int)$$0, (int)this.getItem().getCount());
         }
-        return super.remove(amount);
+        return super.remove($$0);
     }
 
-    public void onQuickCraft(net.minecraft.world.item.ItemStack stack, int amount) {
-        this.removeCount += amount;
-        this.checkTakeAchievements(stack);
+    public void onQuickCraft(net.minecraft.world.item.ItemStack $$0, int $$1) {
+        this.removeCount += $$1;
+        this.checkTakeAchievements($$0);
     }
 
-    public void checkTakeAchievements(net.minecraft.world.item.ItemStack stack) {
-        stack.onCraftedBy(this.player.level, this.player, this.removeCount);
+    public void checkTakeAchievements(net.minecraft.world.item.ItemStack $$0) {
+        $$0.onCraftedBy(this.player.level, this.player, this.removeCount);
         this.removeCount = 0;
     }
 