--- net/minecraft/world/inventory/ResultSlot.java
+++ net/minecraft/world/inventory/ResultSlot.java
@@ -10,35 +10,35 @@
     public net.minecraft.world.entity.player.Player player;
     public int removeCount;
 
-    public net.minecraft.world.inventory.ResultSlot(net.minecraft.world.entity.player.Player player, net.minecraft.world.inventory.CraftingContainer craftSlots, net.minecraft.world.Container container, int slot, int xPosition, int yPosition) {
-        super(container, slot, xPosition, yPosition);
-        this.player = player;
-        this.craftSlots = craftSlots;
+    public net.minecraft.world.inventory.ResultSlot(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.inventory.CraftingContainer $$1, net.minecraft.world.Container $$2, int $$3, int $$4, int $$5) {
+        super($$2, $$3, $$4, $$5);
+        this.player = $$0;
+        this.craftSlots = $$1;
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
         return false;
     }
 
-    public net.minecraft.world.item.ItemStack remove(int amount) {
+    public net.minecraft.world.item.ItemStack remove(int $$0) {
         if (this.hasItem()) {
-            this.removeCount += java.lang.Math.min((int)amount, (int)this.getItem().getCount());
+            this.removeCount += java.lang.Math.min((int)$$0, (int)this.getItem().getCount());
         }
-        return super.remove(amount);
+        return super.remove($$0);
     }
 
-    public void onQuickCraft(net.minecraft.world.item.ItemStack stack, int amount) {
-        this.removeCount += amount;
-        this.checkTakeAchievements(stack);
+    public void onQuickCraft(net.minecraft.world.item.ItemStack $$0, int $$1) {
+        this.removeCount += $$1;
+        this.checkTakeAchievements($$0);
     }
 
-    public void onSwapCraft(int numItemsCrafted) {
-        this.removeCount += numItemsCrafted;
+    public void onSwapCraft(int $$0) {
+        this.removeCount += $$0;
     }
 
-    public void checkTakeAchievements(net.minecraft.world.item.ItemStack stack) {
+    public void checkTakeAchievements(net.minecraft.world.item.ItemStack $$0) {
         if (this.removeCount > 0) {
-            stack.onCraftedBy(this.player.level, this.player, this.removeCount);
+            $$0.onCraftedBy(this.player.level, this.player, this.removeCount);
         }
         if (this.container instanceof net.minecraft.world.inventory.RecipeHolder) {
             ((net.minecraft.world.inventory.RecipeHolder)this.container).awardUsedRecipes(this.player);