--- net/minecraft/world/inventory/Slot.java
+++ net/minecraft/world/inventory/Slot.java
@@ -11,34 +11,34 @@
     public int x;
     public int y;
 
-    public net.minecraft.world.inventory.Slot(net.minecraft.world.Container container, int index, int x, int y) {
-        this.container = container;
-        this.slot = index;
-        this.x = x;
-        this.y = y;
+    public net.minecraft.world.inventory.Slot(net.minecraft.world.Container $$0, int $$1, int $$2, int $$3) {
+        this.container = $$0;
+        this.slot = $$1;
+        this.x = $$2;
+        this.y = $$3;
     }
 
-    public void onQuickCraft(net.minecraft.world.item.ItemStack oldStack, net.minecraft.world.item.ItemStack newStack) {
-        int n = newStack.getCount() - oldStack.getCount();
-        if (n > 0) {
-            this.onQuickCraft(newStack, n);
+    public void onQuickCraft(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        int $$2 = $$1.getCount() - $$0.getCount();
+        if ($$2 > 0) {
+            this.onQuickCraft($$1, $$2);
         }
     }
 
-    public void onQuickCraft(net.minecraft.world.item.ItemStack stack, int amount) {
+    public void onQuickCraft(net.minecraft.world.item.ItemStack $$0, int $$1) {
     }
 
-    public void onSwapCraft(int numItemsCrafted) {
+    public void onSwapCraft(int $$0) {
     }
 
-    public void checkTakeAchievements(net.minecraft.world.item.ItemStack stack) {
+    public void checkTakeAchievements(net.minecraft.world.item.ItemStack $$0) {
     }
 
     public void onTake(net.minecraft.world.entity.player.Player $$0, net.minecraft.world.item.ItemStack $$1) {
         this.setChanged();
     }
 
-    public boolean mayPlace(net.minecraft.world.item.ItemStack stack) {
+    public boolean mayPlace(net.minecraft.world.item.ItemStack $$0) {
         return true;
     }
 
@@ -50,8 +50,8 @@
         return !this.getItem().isEmpty();
     }
 
-    public void set(net.minecraft.world.item.ItemStack stack) {
-        this.container.setItem(this.slot, stack);
+    public void set(net.minecraft.world.item.ItemStack $$0) {
+        this.container.setItem(this.slot, $$0);
         this.setChanged();
     }
 
@@ -68,8 +68,8 @@
         return this.container.getMaxStackSize();
     }
 
-    public int getMaxStackSize(net.minecraft.world.item.ItemStack stack) {
-        return java.lang.Math.min((int)this.getMaxStackSize(), (int)stack.getMaxStackSize());
+    public int getMaxStackSize(net.minecraft.world.item.ItemStack $$0) {
+        return java.lang.Math.min((int)this.getMaxStackSize(), (int)$$0.getMaxStackSize());
     }
 
     @javax.annotation.Nullable
@@ -77,11 +77,11 @@
         return null;
     }
 
-    public net.minecraft.world.item.ItemStack remove(int amount) {
-        return this.container.removeItem(this.slot, amount);
+    public net.minecraft.world.item.ItemStack remove(int $$0) {
+        return this.container.removeItem(this.slot, $$0);
     }
 
-    public boolean mayPickup(net.minecraft.world.entity.player.Player player) {
+    public boolean mayPickup(net.minecraft.world.entity.player.Player $$0) {
         return true;
     }
 