--- net/minecraft/world/inventory/SmokerMenu.java
+++ net/minecraft/world/inventory/SmokerMenu.java
@@ -6,12 +6,12 @@
 public class net.minecraft.world.inventory.SmokerMenu
 extends net.minecraft.world.inventory.AbstractFurnaceMenu
  {
-    public net.minecraft.world.inventory.SmokerMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory) {
-        super(net.minecraft.world.inventory.MenuType.SMOKER, net.minecraft.world.item.crafting.RecipeType.SMOKING, net.minecraft.world.inventory.RecipeBookType.SMOKER, containerId, playerInventory);
+    public net.minecraft.world.inventory.SmokerMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        super(net.minecraft.world.inventory.MenuType.SMOKER, net.minecraft.world.item.crafting.RecipeType.SMOKING, net.minecraft.world.inventory.RecipeBookType.SMOKER, $$0, $$1);
     }
 
-    public net.minecraft.world.inventory.SmokerMenu(int containerId, net.minecraft.world.entity.player.Inventory playerInventory, net.minecraft.world.Container smokerContainer, net.minecraft.world.inventory.ContainerData smokerData) {
-        super(net.minecraft.world.inventory.MenuType.SMOKER, net.minecraft.world.item.crafting.RecipeType.SMOKING, net.minecraft.world.inventory.RecipeBookType.SMOKER, containerId, playerInventory, smokerContainer, smokerData);
+    public net.minecraft.world.inventory.SmokerMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.world.Container $$2, net.minecraft.world.inventory.ContainerData $$3) {
+        super(net.minecraft.world.inventory.MenuType.SMOKER, net.minecraft.world.item.crafting.RecipeType.SMOKING, net.minecraft.world.inventory.RecipeBookType.SMOKER, $$0, $$1, $$2, $$3);
     }
 }
 