--- com/mojang/realmsclient/util/task/ConnectTask.java
+++ com/mojang/realmsclient/util/task/ConnectTask.java
@@ -23,7 +23,7 @@
 
     public void abortTask() {
         this.realmsConnect.abort();
-        net.minecraft.client.Minecraft.getInstance().getClientPackSource().clearServerPack();
+        net.minecraft.client.Minecraft.getInstance().getDownloadedPackSource().clearServerPack();
     }
 
     public void tick() {