--- net/minecraft/client/renderer/block/LiquidBlockRenderer.java
+++ net/minecraft/client/renderer/block/LiquidBlockRenderer.java
@@ -151,121 +151,119 @@
             }
             void $$75 = ($$67 + $$69 + $$71 + $$73) / 4.0f;
             void $$76 = ($$68 + $$70 + $$72 + $$74) / 4.0f;
-            float $$77 = (float)$$6[0].getWidth() / ($$6[0].getU1() - $$6[0].getU0());
-            float $$78 = (float)$$6[0].getHeight() / ($$6[0].getV1() - $$6[0].getV0());
-            float $$79 = 4.0f / java.lang.Math.max((float)$$78, (float)$$77);
-            $$67 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$67, (float)$$75);
-            $$69 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$69, (float)$$75);
-            $$71 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$71, (float)$$75);
-            $$73 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$73, (float)$$75);
-            $$68 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$68, (float)$$76);
-            $$70 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$70, (float)$$76);
-            $$72 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$72, (float)$$76);
-            $$74 = net.minecraft.util.Mth.lerp((float)$$79, (float)$$74, (float)$$76);
-            int $$80 = this.getLightColor($$0, $$1);
-            float $$81 = $$30 * $$8;
-            float $$82 = $$30 * $$9;
-            float $$83 = $$30 * $$10;
-            this.vertex($$2, $$47 + 0.0, $$48 + (double)$$44, $$49 + 0.0, $$81, $$82, $$83, $$67, $$68, $$80);
-            this.vertex($$2, $$47 + 0.0, $$48 + (double)$$46, $$49 + 1.0, $$81, $$82, $$83, $$69, $$70, $$80);
-            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$45, $$49 + 1.0, $$81, $$82, $$83, $$71, $$72, $$80);
-            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$43, $$49 + 0.0, $$81, $$82, $$83, $$73, $$74, $$80);
+            float $$77 = $$6[0].uvShrinkRatio();
+            $$67 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$67, (float)$$75);
+            $$69 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$69, (float)$$75);
+            $$71 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$71, (float)$$75);
+            $$73 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$73, (float)$$75);
+            $$68 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$68, (float)$$76);
+            $$70 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$70, (float)$$76);
+            $$72 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$72, (float)$$76);
+            $$74 = net.minecraft.util.Mth.lerp((float)$$77, (float)$$74, (float)$$76);
+            int $$78 = this.getLightColor($$0, $$1);
+            float $$79 = $$30 * $$8;
+            float $$80 = $$30 * $$9;
+            float $$81 = $$30 * $$10;
+            this.vertex($$2, $$47 + 0.0, $$48 + (double)$$44, $$49 + 0.0, $$79, $$80, $$81, $$67, $$68, $$78);
+            this.vertex($$2, $$47 + 0.0, $$48 + (double)$$46, $$49 + 1.0, $$79, $$80, $$81, $$69, $$70, $$78);
+            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$45, $$49 + 1.0, $$79, $$80, $$81, $$71, $$72, $$78);
+            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$43, $$49 + 0.0, $$79, $$80, $$81, $$73, $$74, $$78);
             if ($$4.shouldRenderBackwardUpFace((net.minecraft.world.level.BlockGetter)$$0, $$1.above())) {
-                this.vertex($$2, $$47 + 0.0, $$48 + (double)$$44, $$49 + 0.0, $$81, $$82, $$83, $$67, $$68, $$80);
-                this.vertex($$2, $$47 + 1.0, $$48 + (double)$$43, $$49 + 0.0, $$81, $$82, $$83, $$73, $$74, $$80);
-                this.vertex($$2, $$47 + 1.0, $$48 + (double)$$45, $$49 + 1.0, $$81, $$82, $$83, $$71, $$72, $$80);
-                this.vertex($$2, $$47 + 0.0, $$48 + (double)$$46, $$49 + 1.0, $$81, $$82, $$83, $$69, $$70, $$80);
+                this.vertex($$2, $$47 + 0.0, $$48 + (double)$$44, $$49 + 0.0, $$79, $$80, $$81, $$67, $$68, $$78);
+                this.vertex($$2, $$47 + 1.0, $$48 + (double)$$43, $$49 + 0.0, $$79, $$80, $$81, $$73, $$74, $$78);
+                this.vertex($$2, $$47 + 1.0, $$48 + (double)$$45, $$49 + 1.0, $$79, $$80, $$81, $$71, $$72, $$78);
+                this.vertex($$2, $$47 + 0.0, $$48 + (double)$$46, $$49 + 1.0, $$79, $$80, $$81, $$69, $$70, $$78);
             }
         }
         if ($$24) {
-            float $$84 = $$6[0].getU0();
-            float $$85 = $$6[0].getU1();
-            float $$86 = $$6[0].getV0();
-            float $$87 = $$6[0].getV1();
-            int $$88 = this.getLightColor($$0, $$1.below());
-            float $$89 = $$29 * $$8;
-            float $$90 = $$29 * $$9;
-            float $$91 = $$29 * $$10;
-            this.vertex($$2, $$47, $$48 + (double)$$51, $$49 + 1.0, $$89, $$90, $$91, $$84, $$87, $$88);
-            this.vertex($$2, $$47, $$48 + (double)$$51, $$49, $$89, $$90, $$91, $$84, $$86, $$88);
-            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$51, $$49, $$89, $$90, $$91, $$85, $$86, $$88);
-            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$51, $$49 + 1.0, $$89, $$90, $$91, $$85, $$87, $$88);
+            float $$82 = $$6[0].getU0();
+            float $$83 = $$6[0].getU1();
+            float $$84 = $$6[0].getV0();
+            float $$85 = $$6[0].getV1();
+            int $$86 = this.getLightColor($$0, $$1.below());
+            float $$87 = $$29 * $$8;
+            float $$88 = $$29 * $$9;
+            float $$89 = $$29 * $$10;
+            this.vertex($$2, $$47, $$48 + (double)$$51, $$49 + 1.0, $$87, $$88, $$89, $$82, $$85, $$86);
+            this.vertex($$2, $$47, $$48 + (double)$$51, $$49, $$87, $$88, $$89, $$82, $$84, $$86);
+            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$51, $$49, $$87, $$88, $$89, $$83, $$84, $$86);
+            this.vertex($$2, $$47 + 1.0, $$48 + (double)$$51, $$49 + 1.0, $$87, $$88, $$89, $$83, $$85, $$86);
         }
-        int $$92 = this.getLightColor($$0, $$1);
-        for (net.minecraft.core.Direction $$93 : net.minecraft.core.Direction$Plane.HORIZONTAL) {
-            net.minecraft.world.level.block.Block $$124;
-            boolean $$121;
-            double $$120;
-            double $$119;
+        int $$90 = this.getLightColor($$0, $$1);
+        for (net.minecraft.core.Direction $$91 : net.minecraft.core.Direction$Plane.HORIZONTAL) {
+            net.minecraft.world.level.block.Block $$122;
+            boolean $$119;
             double $$118;
             double $$117;
-            void $$116;
-            float $$115;
-            switch (net.minecraft.client.renderer.block.LiquidBlockRenderer$1.$SwitchMap$net$minecraft$core$Direction[$$93.ordinal()]) {
+            double $$116;
+            double $$115;
+            void $$114;
+            float $$113;
+            switch (net.minecraft.client.renderer.block.LiquidBlockRenderer$1.$SwitchMap$net$minecraft$core$Direction[$$91.ordinal()]) {
                 case 1: {
-                    void $$94 = $$44;
-                    float $$95 = $$43;
-                    double $$96 = $$47;
-                    double $$97 = $$47 + 1.0;
-                    double $$98 = $$49 + (double)0.001f;
-                    double $$99 = $$49 + (double)0.001f;
-                    boolean $$100 = $$25;
+                    void $$92 = $$44;
+                    float $$93 = $$43;
+                    double $$94 = $$47;
+                    double $$95 = $$47 + 1.0;
+                    double $$96 = $$49 + (double)0.001f;
+                    double $$97 = $$49 + (double)0.001f;
+                    boolean $$98 = $$25;
                     break;
                 }
                 case 2: {
-                    void $$101 = $$45;
-                    void $$102 = $$46;
-                    double $$103 = $$47 + 1.0;
-                    double $$104 = $$47;
-                    double $$105 = $$49 + 1.0 - (double)0.001f;
-                    double $$106 = $$49 + 1.0 - (double)0.001f;
-                    boolean $$107 = $$26;
+                    void $$99 = $$45;
+                    void $$100 = $$46;
+                    double $$101 = $$47 + 1.0;
+                    double $$102 = $$47;
+                    double $$103 = $$49 + 1.0 - (double)0.001f;
+                    double $$104 = $$49 + 1.0 - (double)0.001f;
+                    boolean $$105 = $$26;
                     break;
                 }
                 case 3: {
-                    void $$108 = $$46;
-                    void $$109 = $$44;
-                    double $$110 = $$47 + (double)0.001f;
-                    double $$111 = $$47 + (double)0.001f;
-                    double $$112 = $$49 + 1.0;
-                    double $$113 = $$49;
-                    boolean $$114 = $$27;
+                    void $$106 = $$46;
+                    void $$107 = $$44;
+                    double $$108 = $$47 + (double)0.001f;
+                    double $$109 = $$47 + (double)0.001f;
+                    double $$110 = $$49 + 1.0;
+                    double $$111 = $$49;
+                    boolean $$112 = $$27;
                     break;
                 }
                 default: {
-                    $$115 = $$43;
-                    $$116 = $$45;
-                    $$117 = $$47 + 1.0 - (double)0.001f;
-                    $$118 = $$47 + 1.0 - (double)0.001f;
-                    $$119 = $$49;
-                    $$120 = $$49 + 1.0;
-                    $$121 = $$28;
+                    $$113 = $$43;
+                    $$114 = $$45;
+                    $$115 = $$47 + 1.0 - (double)0.001f;
+                    $$116 = $$47 + 1.0 - (double)0.001f;
+                    $$117 = $$49;
+                    $$118 = $$49 + 1.0;
+                    $$119 = $$28;
                 }
             }
-            if (!$$121 || net.minecraft.client.renderer.block.LiquidBlockRenderer.isFaceOccludedByNeighbor((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$1, (net.minecraft.core.Direction)$$93, (float)java.lang.Math.max((float)$$115, (float)$$116), (net.minecraft.world.level.block.state.BlockState)$$0.getBlockState($$1.relative($$93)))) continue;
-            net.minecraft.core.BlockPos $$122 = $$1.relative($$93);
-            net.minecraft.client.renderer.texture.TextureAtlasSprite $$123 = $$6[1];
-            if (!$$5 && (($$124 = $$0.getBlockState($$122).getBlock()) instanceof net.minecraft.world.level.block.HalfTransparentBlock || $$124 instanceof net.minecraft.world.level.block.LeavesBlock)) {
-                $$123 = this.waterOverlay;
+            if (!$$119 || net.minecraft.client.renderer.block.LiquidBlockRenderer.isFaceOccludedByNeighbor((net.minecraft.world.level.BlockGetter)$$0, (net.minecraft.core.BlockPos)$$1, (net.minecraft.core.Direction)$$91, (float)java.lang.Math.max((float)$$113, (float)$$114), (net.minecraft.world.level.block.state.BlockState)$$0.getBlockState($$1.relative($$91)))) continue;
+            net.minecraft.core.BlockPos $$120 = $$1.relative($$91);
+            net.minecraft.client.renderer.texture.TextureAtlasSprite $$121 = $$6[1];
+            if (!$$5 && (($$122 = $$0.getBlockState($$120).getBlock()) instanceof net.minecraft.world.level.block.HalfTransparentBlock || $$122 instanceof net.minecraft.world.level.block.LeavesBlock)) {
+                $$121 = this.waterOverlay;
             }
-            float $$125 = $$123.getU(0.0);
-            float $$126 = $$123.getU(8.0);
-            float $$127 = $$123.getV((double)((1.0f - $$115) * 16.0f * 0.5f));
-            float $$128 = $$123.getV((double)((1.0f - $$116) * 16.0f * 0.5f));
-            float $$129 = $$123.getV(8.0);
-            float $$130 = $$93.getAxis() == net.minecraft.core.Direction$Axis.Z ? $$31 : $$32;
-            float $$131 = $$30 * $$130 * $$8;
-            float $$132 = $$30 * $$130 * $$9;
-            float $$133 = $$30 * $$130 * $$10;
-            this.vertex($$2, $$117, $$48 + (double)$$115, $$119, $$131, $$132, $$133, $$125, $$127, $$92);
-            this.vertex($$2, $$118, $$48 + (double)$$116, $$120, $$131, $$132, $$133, $$126, $$128, $$92);
-            this.vertex($$2, $$118, $$48 + (double)$$51, $$120, $$131, $$132, $$133, $$126, $$129, $$92);
-            this.vertex($$2, $$117, $$48 + (double)$$51, $$119, $$131, $$132, $$133, $$125, $$129, $$92);
-            if ($$123 == this.waterOverlay) continue;
-            this.vertex($$2, $$117, $$48 + (double)$$51, $$119, $$131, $$132, $$133, $$125, $$129, $$92);
-            this.vertex($$2, $$118, $$48 + (double)$$51, $$120, $$131, $$132, $$133, $$126, $$129, $$92);
-            this.vertex($$2, $$118, $$48 + (double)$$116, $$120, $$131, $$132, $$133, $$126, $$128, $$92);
-            this.vertex($$2, $$117, $$48 + (double)$$115, $$119, $$131, $$132, $$133, $$125, $$127, $$92);
+            float $$123 = $$121.getU(0.0);
+            float $$124 = $$121.getU(8.0);
+            float $$125 = $$121.getV((double)((1.0f - $$113) * 16.0f * 0.5f));
+            float $$126 = $$121.getV((double)((1.0f - $$114) * 16.0f * 0.5f));
+            float $$127 = $$121.getV(8.0);
+            float $$128 = $$91.getAxis() == net.minecraft.core.Direction$Axis.Z ? $$31 : $$32;
+            float $$129 = $$30 * $$128 * $$8;
+            float $$130 = $$30 * $$128 * $$9;
+            float $$131 = $$30 * $$128 * $$10;
+            this.vertex($$2, $$115, $$48 + (double)$$113, $$117, $$129, $$130, $$131, $$123, $$125, $$90);
+            this.vertex($$2, $$116, $$48 + (double)$$114, $$118, $$129, $$130, $$131, $$124, $$126, $$90);
+            this.vertex($$2, $$116, $$48 + (double)$$51, $$118, $$129, $$130, $$131, $$124, $$127, $$90);
+            this.vertex($$2, $$115, $$48 + (double)$$51, $$117, $$129, $$130, $$131, $$123, $$127, $$90);
+            if ($$121 == this.waterOverlay) continue;
+            this.vertex($$2, $$115, $$48 + (double)$$51, $$117, $$129, $$130, $$131, $$123, $$127, $$90);
+            this.vertex($$2, $$116, $$48 + (double)$$51, $$118, $$129, $$130, $$131, $$124, $$127, $$90);
+            this.vertex($$2, $$116, $$48 + (double)$$114, $$118, $$129, $$130, $$131, $$124, $$126, $$90);
+            this.vertex($$2, $$115, $$48 + (double)$$113, $$117, $$129, $$130, $$131, $$123, $$125, $$90);
         }
     }
 
@@ -316,18 +314,18 @@
         return -1.0f;
     }
 
-    public void vertex(com.mojang.blaze3d.vertex.VertexConsumer vertexBuilder, double x, double y, double z, float red, float green, float blue, float u, float v, int packedLight) {
-        vertexBuilder.vertex(x, y, z).color(red, green, blue, 1.0f).uv(u, v).uv2(packedLight).normal(0.0f, 1.0f, 0.0f).endVertex();
+    public void vertex(com.mojang.blaze3d.vertex.VertexConsumer $$0, double $$1, double $$2, double $$3, float $$4, float $$5, float $$6, float $$7, float $$8, int $$9) {
+        $$0.vertex($$1, $$2, $$3).color($$4, $$5, $$6, 1.0f).uv($$7, $$8).uv2($$9).normal(0.0f, 1.0f, 0.0f).endVertex();
     }
 
-    public int getLightColor(net.minecraft.world.level.BlockAndTintGetter lightReader, net.minecraft.core.BlockPos pos) {
-        int n = net.minecraft.client.renderer.LevelRenderer.getLightColor((net.minecraft.world.level.BlockAndTintGetter)lightReader, (net.minecraft.core.BlockPos)pos);
-        int n2 = net.minecraft.client.renderer.LevelRenderer.getLightColor((net.minecraft.world.level.BlockAndTintGetter)lightReader, (net.minecraft.core.BlockPos)pos.above());
-        int n3 = n & 0xFF;
-        int n4 = n2 & 0xFF;
-        int n5 = n >> 16 & 0xFF;
-        int n6 = n2 >> 16 & 0xFF;
-        return (n3 > n4 ? n3 : n4) | (n5 > n6 ? n5 : n6) << 16;
+    public int getLightColor(net.minecraft.world.level.BlockAndTintGetter $$0, net.minecraft.core.BlockPos $$1) {
+        int $$2 = net.minecraft.client.renderer.LevelRenderer.getLightColor((net.minecraft.world.level.BlockAndTintGetter)$$0, (net.minecraft.core.BlockPos)$$1);
+        int $$3 = net.minecraft.client.renderer.LevelRenderer.getLightColor((net.minecraft.world.level.BlockAndTintGetter)$$0, (net.minecraft.core.BlockPos)$$1.above());
+        int $$4 = $$2 & 0xFF;
+        int $$5 = $$3 & 0xFF;
+        int $$6 = $$2 >> 16 & 0xFF;
+        int $$7 = $$3 >> 16 & 0xFF;
+        return ($$4 > $$5 ? $$4 : $$5) | ($$6 > $$7 ? $$6 : $$7) << 16;
     }
 }
 