--- net/minecraft/client/renderer/block/ModelBlockRenderer$AdjacencyInfo.java
+++ net/minecraft/client/renderer/block/ModelBlockRenderer$AdjacencyInfo.java
@@ -39,8 +39,8 @@
         this.vert3Weights = $$6;
     }
 
-    public static net.minecraft.client.renderer.block.ModelBlockRenderer$AdjacencyInfo fromFacing(net.minecraft.core.Direction facing) {
-        return BY_FACING[facing.get3DDataValue()];
+    public static net.minecraft.client.renderer.block.ModelBlockRenderer$AdjacencyInfo fromFacing(net.minecraft.core.Direction $$0) {
+        return BY_FACING[$$0.get3DDataValue()];
     }
 
     public static /* synthetic */ void lambda$static$0(net.minecraft.client.renderer.block.ModelBlockRenderer$AdjacencyInfo[] $$0) {