--- net/minecraft/client/renderer/block/ModelBlockRenderer$Cache.java
+++ net/minecraft/client/renderer/block/ModelBlockRenderer$Cache.java
@@ -32,36 +32,36 @@
         this.brightnessCache.clear();
     }
 
-    public int getLightColor(net.minecraft.world.level.block.state.BlockState blockState, net.minecraft.world.level.BlockAndTintGetter lightReader, net.minecraft.core.BlockPos blockPos) {
-        int n;
-        long l = blockPos.asLong();
-        if (this.enabled && (n = this.colorCache.get(l)) != java.lang.Integer.MAX_VALUE) {
-            return n;
+    public int getLightColor(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockAndTintGetter $$1, net.minecraft.core.BlockPos $$2) {
+        int $$4;
+        long $$3 = $$2.asLong();
+        if (this.enabled && ($$4 = this.colorCache.get($$3)) != java.lang.Integer.MAX_VALUE) {
+            return $$4;
         }
-        n = net.minecraft.client.renderer.LevelRenderer.getLightColor((net.minecraft.world.level.BlockAndTintGetter)lightReader, (net.minecraft.world.level.block.state.BlockState)blockState, (net.minecraft.core.BlockPos)blockPos);
+        int $$5 = net.minecraft.client.renderer.LevelRenderer.getLightColor((net.minecraft.world.level.BlockAndTintGetter)$$1, (net.minecraft.world.level.block.state.BlockState)$$0, (net.minecraft.core.BlockPos)$$2);
         if (this.enabled) {
             if (this.colorCache.size() == 100) {
                 this.colorCache.removeFirstInt();
             }
-            this.colorCache.put(l, n);
+            this.colorCache.put($$3, $$5);
         }
-        return n;
+        return $$5;
     }
 
-    public float getShadeBrightness(net.minecraft.world.level.block.state.BlockState blockState, net.minecraft.world.level.BlockAndTintGetter lightReader, net.minecraft.core.BlockPos blockPos) {
-        float f;
-        long l = blockPos.asLong();
-        if (this.enabled && !java.lang.Float.isNaN((float)(f = this.brightnessCache.get(l)))) {
-            return f;
+    public float getShadeBrightness(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockAndTintGetter $$1, net.minecraft.core.BlockPos $$2) {
+        float $$4;
+        long $$3 = $$2.asLong();
+        if (this.enabled && !java.lang.Float.isNaN((float)($$4 = this.brightnessCache.get($$3)))) {
+            return $$4;
         }
-        f = blockState.getShadeBrightness((net.minecraft.world.level.BlockGetter)lightReader, blockPos);
+        float $$5 = $$0.getShadeBrightness((net.minecraft.world.level.BlockGetter)$$1, $$2);
         if (this.enabled) {
             if (this.brightnessCache.size() == 100) {
                 this.brightnessCache.removeFirstFloat();
             }
-            this.brightnessCache.put(l, f);
+            this.brightnessCache.put($$3, $$5);
         }
-        return f;
+        return $$5;
     }
 }
 