--- net/minecraft/client/renderer/block/ModelBlockRenderer$AmbientVertexRemap.java
+++ net/minecraft/client/renderer/block/ModelBlockRenderer$AmbientVertexRemap.java
@@ -35,8 +35,8 @@
         this.vert3 = $$3;
     }
 
-    public static net.minecraft.client.renderer.block.ModelBlockRenderer$AmbientVertexRemap fromFacing(net.minecraft.core.Direction facing) {
-        return BY_FACING[facing.get3DDataValue()];
+    public static net.minecraft.client.renderer.block.ModelBlockRenderer$AmbientVertexRemap fromFacing(net.minecraft.core.Direction $$0) {
+        return BY_FACING[$$0.get3DDataValue()];
     }
 
     public static /* synthetic */ void lambda$static$0(net.minecraft.client.renderer.block.ModelBlockRenderer$AmbientVertexRemap[] $$0) {