--- com/mojang/realmsclient/RealmsMainScreen$RealmSelectionList.java
+++ com/mojang/realmsclient/RealmsMainScreen$RealmSelectionList.java
@@ -17,36 +17,36 @@
         return this.this$0.getFocused() == this;
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 257 || keyCode == 32 || keyCode == 335) {
-            com.mojang.realmsclient.RealmsMainScreen$Entry entry = (com.mojang.realmsclient.RealmsMainScreen$Entry)this.getSelected();
-            if (entry == null) {
-                return super.keyPressed(keyCode, scanCode, modifiers);
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 257 || $$0 == 32 || $$0 == 335) {
+            com.mojang.realmsclient.RealmsMainScreen$Entry $$3 = (com.mojang.realmsclient.RealmsMainScreen$Entry)this.getSelected();
+            if ($$3 == null) {
+                return super.keyPressed($$0, $$1, $$2);
             }
-            return entry.mouseClicked(0.0, 0.0, 0);
+            return $$3.mouseClicked(0.0, 0.0, 0);
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (button == 0 && mouseX < (double)this.getScrollbarPosition() && mouseY >= (double)this.y0 && mouseY <= (double)this.y1) {
-            int n = this.this$0.realmSelectionList.getRowLeft();
-            int n2 = this.getScrollbarPosition();
-            int n3 = (int)java.lang.Math.floor((double)(mouseY - (double)this.y0)) - this.headerHeight + (int)this.getScrollAmount() - 4;
-            int n4 = n3 / this.itemHeight;
-            if (mouseX >= (double)n && mouseX <= (double)n2 && n4 >= 0 && n3 >= 0 && n4 < this.getItemCount()) {
-                this.itemClicked(n3, n4, mouseX, mouseY, this.width);
-                this.selectItem(n4);
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if ($$2 == 0 && $$0 < (double)this.getScrollbarPosition() && $$1 >= (double)this.y0 && $$1 <= (double)this.y1) {
+            int $$3 = this.this$0.realmSelectionList.getRowLeft();
+            int $$4 = this.getScrollbarPosition();
+            int $$5 = (int)java.lang.Math.floor((double)($$1 - (double)this.y0)) - this.headerHeight + (int)this.getScrollAmount() - 4;
+            int $$6 = $$5 / this.itemHeight;
+            if ($$0 >= (double)$$3 && $$0 <= (double)$$4 && $$6 >= 0 && $$5 >= 0 && $$6 < this.getItemCount()) {
+                this.itemClicked($$5, $$6, $$0, $$1, this.width);
+                this.selectItem($$6);
             }
             return true;
         }
-        return super.mouseClicked(mouseX, mouseY, button);
+        return super.mouseClicked($$0, $$1, $$2);
     }
 
-    public void setSelected(@javax.annotation.Nullable com.mojang.realmsclient.RealmsMainScreen$Entry entry) {
-        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)entry);
-        if (entry != null) {
-            this.this$0.updateButtonStates(entry.getServer());
+    public void setSelected(@javax.annotation.Nullable com.mojang.realmsclient.RealmsMainScreen$Entry $$0) {
+        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)$$0);
+        if ($$0 != null) {
+            this.this$0.updateButtonStates($$0.getServer());
         } else {
             this.this$0.updateButtonStates(null);
         }