--- com/mojang/realmsclient/RealmsMainScreen$CloseButton.java
+++ com/mojang/realmsclient/RealmsMainScreen$CloseButton.java
@@ -13,12 +13,12 @@
         super(realmsMainScreen.popupX0() + 4, realmsMainScreen.popupY0() + 4, 12, 12, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"mco.selectServer.close"), $$1 -> realmsMainScreen.onClosePopup());
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)com.mojang.realmsclient.RealmsMainScreen.CROSS_ICON_LOCATION);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        float f = this.isHoveredOrFocused() ? 12.0f : 0.0f;
-        com.mojang.realmsclient.RealmsMainScreen$CloseButton.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)this.x, (int)this.y, (float)0.0f, (float)f, (int)12, (int)12, (int)12, (int)24);
-        if (this.isMouseOver((double)mouseX, (double)mouseY)) {
+        float $$4 = this.isHoveredOrFocused() ? 12.0f : 0.0f;
+        com.mojang.realmsclient.RealmsMainScreen$CloseButton.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)this.x, (int)this.y, (float)0.0f, (float)$$4, (int)12, (int)12, (int)12, (int)24);
+        if (this.isMouseOver((double)$$1, (double)$$2)) {
             this.this$0.setTooltip(new net.minecraft.network.chat.Component[]{this.getMessage()});
         }
     }