--- com/mojang/realmsclient/RealmsMainScreen$NewsButton.java
+++ com/mojang/realmsclient/RealmsMainScreen$NewsButton.java
@@ -10,11 +10,16 @@
 
     public com.mojang.realmsclient.RealmsMainScreen$NewsButton(com.mojang.realmsclient.RealmsMainScreen realmsMainScreen) {
         this.this$0 = realmsMainScreen;
-        super(realmsMainScreen.width - 115, 6, 20, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"mco.news"), $$1 -> {
+        super(realmsMainScreen.width - 115, 6, 20, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"mco.news"), $$12 -> {
             if ($$0.newsLink == null) {
                 return;
             }
-            net.minecraft.Util.getPlatform().openUri($$0.newsLink);
+            com.mojang.realmsclient.RealmsMainScreen.access$2900((com.mojang.realmsclient.RealmsMainScreen)realmsMainScreen).setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.ConfirmLinkScreen($$1 -> {
+                if ($$1) {
+                    net.minecraft.Util.getPlatform().openUri($$0.newsLink);
+                }
+                com.mojang.realmsclient.RealmsMainScreen.access$3000((com.mojang.realmsclient.RealmsMainScreen)realmsMainScreen).setScreen((net.minecraft.client.gui.screens.Screen)realmsMainScreen);
+            }, $$0.newsLink, true));
             if ($$0.hasUnreadNews) {
                 com.mojang.realmsclient.util.RealmsPersistence$RealmsPersistenceData $$2 = com.mojang.realmsclient.util.RealmsPersistence.readFile();
                 $$2.hasUnreadNews = false;
@@ -24,8 +29,8 @@
         });
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.this$0.renderNews(matrixStack, mouseX, mouseY, this.this$0.hasUnreadNews, this.x, this.y, this.isHoveredOrFocused(), this.active);
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.this$0.renderNews($$0, $$1, $$2, this.this$0.hasUnreadNews, this.x, this.y, this.isHoveredOrFocused(), this.active);
     }
 }
 