--- com/mojang/realmsclient/RealmsMainScreen$ServerEntry.java
+++ com/mojang/realmsclient/RealmsMainScreen$ServerEntry.java
@@ -16,11 +16,11 @@
         this.serverData = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.renderMcoServerItem(this.serverData, matrixStack, left, top, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        this.renderMcoServerItem(this.serverData, $$0, $$3, $$2, $$6, $$7);
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
         if (this.serverData.state == com.mojang.realmsclient.dto.RealmsServer$State.UNINITIALIZED) {
             com.mojang.realmsclient.RealmsMainScreen.access$1400((com.mojang.realmsclient.RealmsMainScreen)this.this$0).setScreen((net.minecraft.client.gui.screens.Screen)new com.mojang.realmsclient.gui.screens.RealmsCreateRealmScreen(this.serverData, this.this$0));
         }