--- com/mojang/realmsclient/RealmsMainScreen.java
+++ com/mojang/realmsclient/RealmsMainScreen.java
@@ -478,13 +478,13 @@
         }
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.keyCombos.forEach(com.mojang.realmsclient.KeyCombo::reset);
             this.onClosePopup();
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
     public void onClosePopup() {
@@ -495,25 +495,25 @@
         }
     }
 
-    public boolean charTyped(char codePoint, int modifiers) {
-        this.keyCombos.forEach($$1 -> $$1.keyPressed(codePoint));
+    public boolean charTyped(char $$0, int $$12) {
+        this.keyCombos.forEach($$1 -> $$1.keyPressed($$0));
         return true;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         this.hoveredElement = com.mojang.realmsclient.RealmsMainScreen$HoveredElement.NONE;
         this.toolTip = null;
-        this.renderBackground(matrixStack);
-        this.realmSelectionList.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.drawRealmsLogo(matrixStack, this.width / 2 - 50, 7);
+        this.renderBackground($$0);
+        this.realmSelectionList.render($$0, $$1, $$2, $$3);
+        this.drawRealmsLogo($$0, this.width / 2 - 50, 7);
         if (com.mojang.realmsclient.client.RealmsClient.currentEnvironment == com.mojang.realmsclient.client.RealmsClient$Environment.STAGE) {
-            this.renderStage(matrixStack);
+            this.renderStage($$0);
         }
         if (com.mojang.realmsclient.client.RealmsClient.currentEnvironment == com.mojang.realmsclient.client.RealmsClient$Environment.LOCAL) {
-            this.renderLocal(matrixStack);
+            this.renderLocal($$0);
         }
         if (this.shouldShowPopup()) {
-            this.drawPopup(matrixStack);
+            this.drawPopup($$0);
         } else {
             if (this.showingPopup) {
                 this.updateButtonStates(null);
@@ -525,20 +525,20 @@
             }
             this.showingPopup = false;
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
         if (this.toolTip != null) {
-            this.renderMousehoverTooltip(matrixStack, this.toolTip, mouseX, mouseY);
+            this.renderMousehoverTooltip($$0, this.toolTip, $$1, $$2);
         }
         if (this.trialsAvailable && !this.createdTrial && this.shouldShowPopup()) {
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)TRIAL_ICON_LOCATION);
             com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-            int n = 8;
-            int n2 = 8;
-            int n3 = 0;
+            int $$4 = 8;
+            int $$5 = 8;
+            int $$6 = 0;
             if ((net.minecraft.Util.getMillis() / 800L & 1L) == 1L) {
-                n3 = 8;
+                $$6 = 8;
             }
-            net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)(this.createTrialButton.x + this.createTrialButton.getWidth() - 8 - 4), (int)(this.createTrialButton.y + this.createTrialButton.getHeight() / 2 - 4), (float)0.0f, (float)n3, (int)8, (int)8, (int)8, (int)16);
+            net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)(this.createTrialButton.x + this.createTrialButton.getWidth() - 8 - 4), (int)(this.createTrialButton.y + this.createTrialButton.getHeight() / 2 - 4), (float)0.0f, (float)$$6, (int)8, (int)8, (int)8, (int)16);
         }
     }
 
@@ -552,13 +552,13 @@
         $$0.popPose();
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (this.isOutsidePopup(mouseX, mouseY) && this.popupOpenedByUser) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if (this.isOutsidePopup($$0, $$1) && this.popupOpenedByUser) {
             this.popupOpenedByUser = false;
             this.justClosedPopup = true;
             return true;
         }
-        return super.mouseClicked(mouseX, mouseY, button);
+        return super.mouseClicked($$0, $$1, $$2);
     }
 
     public boolean isOutsidePopup(double $$0, double $$1) {
@@ -982,6 +982,14 @@
         return $$0.font;
     }
 
+    public static /* synthetic */ net.minecraft.client.Minecraft access$2900(com.mojang.realmsclient.RealmsMainScreen $$0) {
+        return $$0.minecraft;
+    }
+
+    public static /* synthetic */ net.minecraft.client.Minecraft access$3000(com.mojang.realmsclient.RealmsMainScreen $$0) {
+        return $$0.minecraft;
+    }
+
     static {
         LOGGER = com.mojang.logging.LogUtils.getLogger();
         ON_ICON_LOCATION = new net.minecraft.resources.ResourceLocation("realms", "textures/gui/realms/on_icon.png");