--- com/mojang/realmsclient/RealmsMainScreen$PendingInvitesButton.java
+++ com/mojang/realmsclient/RealmsMainScreen$PendingInvitesButton.java
@@ -17,8 +17,8 @@
         this.setMessage(this.this$0.numberOfPendingInvites == 0 ? com.mojang.realmsclient.RealmsMainScreen.NO_PENDING_INVITES_TEXT : com.mojang.realmsclient.RealmsMainScreen.PENDING_INVITES_TEXT);
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.this$0.drawInvitationPendingIcon(matrixStack, mouseX, mouseY, this.x, this.y, this.isHoveredOrFocused(), this.active);
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.this$0.drawInvitationPendingIcon($$0, $$1, $$2, this.x, this.y, this.isHoveredOrFocused(), this.active);
     }
 }
 