--- net/minecraft/client/renderer/item/ItemProperties.java
+++ net/minecraft/client/renderer/item/ItemProperties.java
@@ -25,29 +25,29 @@
         GENERIC_PROPERTIES.put((java.lang.Object)new net.minecraft.resources.ResourceLocation("custom_model_data"), (java.lang.Object)$$0);
     }
 
-    public static void register(net.minecraft.world.item.Item $$0, net.minecraft.resources.ResourceLocation $$1, net.minecraft.client.renderer.item.ClampedItemPropertyFunction $$2) {
-        ((java.util.Map)PROPERTIES.computeIfAbsent((java.lang.Object)$$0, function -> com.google.common.collect.Maps.newHashMap())).put((java.lang.Object)$$1, (java.lang.Object)$$2);
+    public static void register(net.minecraft.world.item.Item $$02, net.minecraft.resources.ResourceLocation $$1, net.minecraft.client.renderer.item.ClampedItemPropertyFunction $$2) {
+        ((java.util.Map)PROPERTIES.computeIfAbsent((java.lang.Object)$$02, $$0 -> com.google.common.collect.Maps.newHashMap())).put((java.lang.Object)$$1, (java.lang.Object)$$2);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.client.renderer.item.ItemPropertyFunction getProperty(net.minecraft.world.item.Item item, net.minecraft.resources.ResourceLocation name) {
-        net.minecraft.client.renderer.item.ItemPropertyFunction itemPropertyFunction;
-        if (item.getMaxDamage() > 0) {
-            if (DAMAGE.equals((java.lang.Object)name)) {
+    public static net.minecraft.client.renderer.item.ItemPropertyFunction getProperty(net.minecraft.world.item.Item $$0, net.minecraft.resources.ResourceLocation $$1) {
+        net.minecraft.client.renderer.item.ItemPropertyFunction $$2;
+        if ($$0.getMaxDamage() > 0) {
+            if (DAMAGE.equals((java.lang.Object)$$1)) {
                 return PROPERTY_DAMAGE;
             }
-            if (DAMAGED.equals((java.lang.Object)name)) {
+            if (DAMAGED.equals((java.lang.Object)$$1)) {
                 return PROPERTY_DAMAGED;
             }
         }
-        if ((itemPropertyFunction = (net.minecraft.client.renderer.item.ItemPropertyFunction)GENERIC_PROPERTIES.get((java.lang.Object)name)) != null) {
-            return itemPropertyFunction;
+        if (($$2 = (net.minecraft.client.renderer.item.ItemPropertyFunction)GENERIC_PROPERTIES.get((java.lang.Object)$$1)) != null) {
+            return $$2;
         }
-        java.util.Map map = (java.util.Map)PROPERTIES.get((java.lang.Object)item);
-        if (map == null) {
+        java.util.Map $$3 = (java.util.Map)PROPERTIES.get((java.lang.Object)$$0);
+        if ($$3 == null) {
             return null;
         }
-        return (net.minecraft.client.renderer.item.ItemPropertyFunction)map.get((java.lang.Object)name);
+        return (net.minecraft.client.renderer.item.ItemPropertyFunction)$$3.get((java.lang.Object)$$1);
     }
 
     public static /* synthetic */ float lambda$static$20(net.minecraft.world.item.ItemStack $$0, net.minecraft.client.multiplayer.ClientLevel $$1, net.minecraft.world.entity.LivingEntity $$2, int $$3) {