--- net/minecraft/client/renderer/item/ItemProperties$1.java
+++ net/minecraft/client/renderer/item/ItemProperties$1.java
@@ -35,12 +35,12 @@
         return (float)$$6;
     }
 
-    public double wobble(net.minecraft.world.level.Level level, double d) {
-        if (level.getGameTime() != this.lastUpdateTick) {
-            this.lastUpdateTick = level.getGameTime();
-            double d2 = d - this.rotation;
-            d2 = net.minecraft.util.Mth.positiveModulo((double)(d2 + 0.5), (double)1.0) - 0.5;
-            this.rota += d2 * 0.1;
+    public double wobble(net.minecraft.world.level.Level $$0, double $$1) {
+        if ($$0.getGameTime() != this.lastUpdateTick) {
+            this.lastUpdateTick = $$0.getGameTime();
+            double $$2 = $$1 - this.rotation;
+            $$2 = net.minecraft.util.Mth.positiveModulo((double)($$2 + 0.5), (double)1.0) - 0.5;
+            this.rota += $$2 * 0.1;
             this.rota *= 0.9;
             this.rotation = net.minecraft.util.Mth.positiveModulo((double)(this.rotation + this.rota), (double)1.0);
         }