--- net/minecraft/world/item/alchemy/Potion.java
+++ net/minecraft/world/item/alchemy/Potion.java
@@ -9,21 +9,21 @@
     public java.lang.String name;
     public com.google.common.collect.ImmutableList<net.minecraft.world.effect.MobEffectInstance> effects;
 
-    public static net.minecraft.world.item.alchemy.Potion byName(java.lang.String name) {
-        return (net.minecraft.world.item.alchemy.Potion)net.minecraft.core.Registry.POTION.get(net.minecraft.resources.ResourceLocation.tryParse((java.lang.String)name));
+    public static net.minecraft.world.item.alchemy.Potion byName(java.lang.String $$0) {
+        return (net.minecraft.world.item.alchemy.Potion)net.minecraft.core.Registry.POTION.get(net.minecraft.resources.ResourceLocation.tryParse((java.lang.String)$$0));
     }
 
-    public net.minecraft.world.item.alchemy.Potion(net.minecraft.world.effect.MobEffectInstance ... effects) {
-        this(null, effects);
+    public net.minecraft.world.item.alchemy.Potion(net.minecraft.world.effect.MobEffectInstance ... $$0) {
+        this(null, $$0);
     }
 
-    public net.minecraft.world.item.alchemy.Potion(@javax.annotation.Nullable java.lang.String name, net.minecraft.world.effect.MobEffectInstance ... effects) {
-        this.name = name;
-        this.effects = com.google.common.collect.ImmutableList.copyOf((java.lang.Object[])effects);
+    public net.minecraft.world.item.alchemy.Potion(@javax.annotation.Nullable java.lang.String $$0, net.minecraft.world.effect.MobEffectInstance ... $$1) {
+        this.name = $$0;
+        this.effects = com.google.common.collect.ImmutableList.copyOf((java.lang.Object[])$$1);
     }
 
-    public java.lang.String getName(java.lang.String prefix) {
-        return prefix + (this.name == null ? net.minecraft.core.Registry.POTION.getKey((java.lang.Object)this).getPath() : this.name);
+    public java.lang.String getName(java.lang.String $$0) {
+        return $$0 + (this.name == null ? net.minecraft.core.Registry.POTION.getKey((java.lang.Object)this).getPath() : this.name);
     }
 
     public java.util.List<net.minecraft.world.effect.MobEffectInstance> getEffects() {