--- net/minecraft/world/item/alchemy/Potions.java
+++ net/minecraft/world/item/alchemy/Potions.java
@@ -52,8 +52,8 @@
     public net.minecraft.world.item.alchemy.Potions() {
     }
 
-    public static net.minecraft.world.item.alchemy.Potion register(java.lang.String key, net.minecraft.world.item.alchemy.Potion potion) {
-        return (net.minecraft.world.item.alchemy.Potion)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.POTION, (java.lang.String)key, (java.lang.Object)potion);
+    public static net.minecraft.world.item.alchemy.Potion register(java.lang.String $$0, net.minecraft.world.item.alchemy.Potion $$1) {
+        return (net.minecraft.world.item.alchemy.Potion)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.POTION, (java.lang.String)$$0, (java.lang.Object)$$1);
     }
 
     static {