--- net/minecraft/world/item/alchemy/PotionBrewing$Mix.java
+++ net/minecraft/world/item/alchemy/PotionBrewing$Mix.java
@@ -9,10 +9,10 @@
     public net.minecraft.world.item.crafting.Ingredient ingredient;
     public T to;
 
-    public net.minecraft.world.item.alchemy.PotionBrewing$Mix(T from, net.minecraft.world.item.crafting.Ingredient ingredient, T to) {
-        this.from = from;
-        this.ingredient = ingredient;
-        this.to = to;
+    public net.minecraft.world.item.alchemy.PotionBrewing$Mix(T $$0, net.minecraft.world.item.crafting.Ingredient $$1, T $$2) {
+        this.from = $$0;
+        this.ingredient = $$1;
+        this.to = $$2;
     }
 }
 