--- net/minecraft/world/item/alchemy/PotionBrewing.java
+++ net/minecraft/world/item/alchemy/PotionBrewing.java
@@ -14,88 +14,84 @@
     public net.minecraft.world.item.alchemy.PotionBrewing() {
     }
 
-    public static boolean isIngredient(net.minecraft.world.item.ItemStack input) {
-        return net.minecraft.world.item.alchemy.PotionBrewing.isContainerIngredient((net.minecraft.world.item.ItemStack)input) || net.minecraft.world.item.alchemy.PotionBrewing.isPotionIngredient((net.minecraft.world.item.ItemStack)input);
+    public static boolean isIngredient(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.item.alchemy.PotionBrewing.isContainerIngredient((net.minecraft.world.item.ItemStack)$$0) || net.minecraft.world.item.alchemy.PotionBrewing.isPotionIngredient((net.minecraft.world.item.ItemStack)$$0);
     }
 
-    public static boolean isContainerIngredient(net.minecraft.world.item.ItemStack input) {
-        int n = CONTAINER_MIXES.size();
-        for (int i = 0; i < n; ++i) {
-            if (!((net.minecraft.world.item.alchemy.PotionBrewing$Mix)net.minecraft.world.item.alchemy.PotionBrewing.CONTAINER_MIXES.get((int)i)).ingredient.test(input)) continue;
+    public static boolean isContainerIngredient(net.minecraft.world.item.ItemStack $$0) {
+        int $$2 = CONTAINER_MIXES.size();
+        for (int $$1 = 0; $$1 < $$2; ++$$1) {
+            if (!((net.minecraft.world.item.alchemy.PotionBrewing$Mix)net.minecraft.world.item.alchemy.PotionBrewing.CONTAINER_MIXES.get((int)$$1)).ingredient.test($$0)) continue;
             return true;
         }
         return false;
     }
 
-    public static boolean isPotionIngredient(net.minecraft.world.item.ItemStack input) {
-        int n = POTION_MIXES.size();
-        for (int i = 0; i < n; ++i) {
-            if (!((net.minecraft.world.item.alchemy.PotionBrewing$Mix)net.minecraft.world.item.alchemy.PotionBrewing.POTION_MIXES.get((int)i)).ingredient.test(input)) continue;
+    public static boolean isPotionIngredient(net.minecraft.world.item.ItemStack $$0) {
+        int $$2 = POTION_MIXES.size();
+        for (int $$1 = 0; $$1 < $$2; ++$$1) {
+            if (!((net.minecraft.world.item.alchemy.PotionBrewing$Mix)net.minecraft.world.item.alchemy.PotionBrewing.POTION_MIXES.get((int)$$1)).ingredient.test($$0)) continue;
             return true;
         }
         return false;
     }
 
-    public static boolean isBrewablePotion(net.minecraft.world.item.alchemy.Potion potion) {
-        int n = POTION_MIXES.size();
-        for (int i = 0; i < n; ++i) {
-            if (((net.minecraft.world.item.alchemy.PotionBrewing$Mix)net.minecraft.world.item.alchemy.PotionBrewing.POTION_MIXES.get((int)i)).to != potion) continue;
+    public static boolean isBrewablePotion(net.minecraft.world.item.alchemy.Potion $$0) {
+        int $$2 = POTION_MIXES.size();
+        for (int $$1 = 0; $$1 < $$2; ++$$1) {
+            if (((net.minecraft.world.item.alchemy.PotionBrewing$Mix)net.minecraft.world.item.alchemy.PotionBrewing.POTION_MIXES.get((int)$$1)).to != $$0) continue;
             return true;
         }
         return false;
     }
 
-    public static boolean hasMix(net.minecraft.world.item.ItemStack input, net.minecraft.world.item.ItemStack reagent) {
-        if (!ALLOWED_CONTAINER.test((java.lang.Object)input)) {
+    public static boolean hasMix(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        if (!ALLOWED_CONTAINER.test((java.lang.Object)$$0)) {
             return false;
         }
-        return net.minecraft.world.item.alchemy.PotionBrewing.hasContainerMix((net.minecraft.world.item.ItemStack)input, (net.minecraft.world.item.ItemStack)reagent) || net.minecraft.world.item.alchemy.PotionBrewing.hasPotionMix((net.minecraft.world.item.ItemStack)input, (net.minecraft.world.item.ItemStack)reagent);
+        return net.minecraft.world.item.alchemy.PotionBrewing.hasContainerMix((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.item.ItemStack)$$1) || net.minecraft.world.item.alchemy.PotionBrewing.hasPotionMix((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.item.ItemStack)$$1);
     }
 
-    public static boolean hasContainerMix(net.minecraft.world.item.ItemStack input, net.minecraft.world.item.ItemStack reagent) {
-        net.minecraft.world.item.Item item = input.getItem();
-        int n = CONTAINER_MIXES.size();
-        for (int i = 0; i < n; ++i) {
-            net.minecraft.world.item.alchemy.PotionBrewing$Mix mix = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)CONTAINER_MIXES.get(i);
-            if (mix.from != item || !mix.ingredient.test(reagent)) continue;
+    public static boolean hasContainerMix(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.world.item.Item $$2 = $$0.getItem();
+        int $$4 = CONTAINER_MIXES.size();
+        for (int $$3 = 0; $$3 < $$4; ++$$3) {
+            net.minecraft.world.item.alchemy.PotionBrewing$Mix $$5 = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)CONTAINER_MIXES.get($$3);
+            if ($$5.from != $$2 || !$$5.ingredient.test($$1)) continue;
             return true;
         }
         return false;
     }
 
-    public static boolean hasPotionMix(net.minecraft.world.item.ItemStack input, net.minecraft.world.item.ItemStack reagent) {
-        net.minecraft.world.item.alchemy.Potion potion = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)input);
-        int n = POTION_MIXES.size();
-        for (int i = 0; i < n; ++i) {
-            net.minecraft.world.item.alchemy.PotionBrewing$Mix mix = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)POTION_MIXES.get(i);
-            if (mix.from != potion || !mix.ingredient.test(reagent)) continue;
+    public static boolean hasPotionMix(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.world.item.alchemy.Potion $$2 = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)$$0);
+        int $$4 = POTION_MIXES.size();
+        for (int $$3 = 0; $$3 < $$4; ++$$3) {
+            net.minecraft.world.item.alchemy.PotionBrewing$Mix $$5 = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)POTION_MIXES.get($$3);
+            if ($$5.from != $$2 || !$$5.ingredient.test($$1)) continue;
             return true;
         }
         return false;
     }
 
-    public static net.minecraft.world.item.ItemStack mix(net.minecraft.world.item.ItemStack reagent, net.minecraft.world.item.ItemStack potion) {
-        if (!potion.isEmpty()) {
-            net.minecraft.world.item.alchemy.PotionBrewing$Mix mix;
-            int n;
-            net.minecraft.world.item.alchemy.Potion potion2 = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)potion);
-            net.minecraft.world.item.Item item = potion.getItem();
-            int n2 = CONTAINER_MIXES.size();
-            for (n = 0;
- n < n2; ++n) {
-                mix = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)CONTAINER_MIXES.get(n);
-                if (mix.from != item || !mix.ingredient.test(reagent)) continue;
-                return net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)mix.to), (net.minecraft.world.item.alchemy.Potion)potion2);
+    public static net.minecraft.world.item.ItemStack mix(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.ItemStack $$1) {
+        if (!$$1.isEmpty()) {
+            net.minecraft.world.item.alchemy.Potion $$2 = net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)$$1);
+            net.minecraft.world.item.Item $$3 = $$1.getItem();
+            int $$5 = CONTAINER_MIXES.size();
+            for (int $$4 = 0; $$4 < $$5; ++$$4) {
+                net.minecraft.world.item.alchemy.PotionBrewing$Mix $$6 = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)CONTAINER_MIXES.get($$4);
+                if ($$6.from != $$3 || !$$6.ingredient.test($$0)) continue;
+                return net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$6.to), (net.minecraft.world.item.alchemy.Potion)$$2);
             }
-            n2 = POTION_MIXES.size();
-            for (n = 0;
- n < n2; ++n) {
-                mix = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)POTION_MIXES.get(n);
-                if (mix.from != potion2 || !mix.ingredient.test(reagent)) continue;
-                return net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)item), (net.minecraft.world.item.alchemy.Potion)((net.minecraft.world.item.alchemy.Potion)mix.to));
+            int $$8 = POTION_MIXES.size();
+            for (int $$7 = 0; $$7 < $$8; ++$$7) {
+                net.minecraft.world.item.alchemy.PotionBrewing$Mix $$9 = (net.minecraft.world.item.alchemy.PotionBrewing$Mix)POTION_MIXES.get($$7);
+                if ($$9.from != $$2 || !$$9.ingredient.test($$0)) continue;
+                return net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$3), (net.minecraft.world.item.alchemy.Potion)((net.minecraft.world.item.alchemy.Potion)$$9.to));
             }
         }
-        return potion;
+        return $$1;
     }
 
     public static void bootStrap() {
@@ -161,30 +157,30 @@
         net.minecraft.world.item.alchemy.PotionBrewing.addMix((net.minecraft.world.item.alchemy.Potion)net.minecraft.world.item.alchemy.Potions.SLOW_FALLING, (net.minecraft.world.item.Item)net.minecraft.world.item.Items.REDSTONE, (net.minecraft.world.item.alchemy.Potion)net.minecraft.world.item.alchemy.Potions.LONG_SLOW_FALLING);
     }
 
-    public static void addContainerRecipe(net.minecraft.world.item.Item from, net.minecraft.world.item.Item ingredient, net.minecraft.world.item.Item to) {
-        if (!(from instanceof net.minecraft.world.item.PotionItem)) {
-            throw new java.lang.IllegalArgumentException("Expected a potion, got: " + net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)from));
+    public static void addContainerRecipe(net.minecraft.world.item.Item $$0, net.minecraft.world.item.Item $$1, net.minecraft.world.item.Item $$2) {
+        if (!($$0 instanceof net.minecraft.world.item.PotionItem)) {
+            throw new java.lang.IllegalArgumentException("Expected a potion, got: " + net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)$$0));
         }
-        if (!(to instanceof net.minecraft.world.item.PotionItem)) {
-            throw new java.lang.IllegalArgumentException("Expected a potion, got: " + net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)to));
+        if (!($$2 instanceof net.minecraft.world.item.PotionItem)) {
+            throw new java.lang.IllegalArgumentException("Expected a potion, got: " + net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)$$2));
         }
-        CONTAINER_MIXES.add((java.lang.Object)new net.minecraft.world.item.alchemy.PotionBrewing$Mix((java.lang.Object)from, net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{ingredient}), (java.lang.Object)to));
+        CONTAINER_MIXES.add((java.lang.Object)new net.minecraft.world.item.alchemy.PotionBrewing$Mix((java.lang.Object)$$0, net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{$$1}), (java.lang.Object)$$2));
     }
 
-    public static void addContainer(net.minecraft.world.item.Item container) {
-        if (!(container instanceof net.minecraft.world.item.PotionItem)) {
-            throw new java.lang.IllegalArgumentException("Expected a potion, got: " + net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)container));
+    public static void addContainer(net.minecraft.world.item.Item $$0) {
+        if (!($$0 instanceof net.minecraft.world.item.PotionItem)) {
+            throw new java.lang.IllegalArgumentException("Expected a potion, got: " + net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)$$0));
         }
-        ALLOWED_CONTAINERS.add((java.lang.Object)net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{container}));
+        ALLOWED_CONTAINERS.add((java.lang.Object)net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{$$0}));
     }
 
-    public static void addMix(net.minecraft.world.item.alchemy.Potion potionEntry, net.minecraft.world.item.Item potionIngredient, net.minecraft.world.item.alchemy.Potion potionResult) {
-        POTION_MIXES.add((java.lang.Object)new net.minecraft.world.item.alchemy.PotionBrewing$Mix((java.lang.Object)potionEntry, net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{potionIngredient}), (java.lang.Object)potionResult));
+    public static void addMix(net.minecraft.world.item.alchemy.Potion $$0, net.minecraft.world.item.Item $$1, net.minecraft.world.item.alchemy.Potion $$2) {
+        POTION_MIXES.add((java.lang.Object)new net.minecraft.world.item.alchemy.PotionBrewing$Mix((java.lang.Object)$$0, net.minecraft.world.item.crafting.Ingredient.of((net.minecraft.world.level.ItemLike[])new net.minecraft.world.level.ItemLike[]{$$1}), (java.lang.Object)$$2));
     }
 
-    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.item.ItemStack input) {
-        for (net.minecraft.world.item.crafting.Ingredient ingredient : ALLOWED_CONTAINERS) {
-            if (!ingredient.test(input)) continue;
+    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.item.ItemStack $$0) {
+        for (net.minecraft.world.item.crafting.Ingredient $$1 : ALLOWED_CONTAINERS) {
+            if (!$$1.test($$0)) continue;
             return true;
         }
         return false;
@@ -194,9 +190,9 @@
         POTION_MIXES = com.google.common.collect.Lists.newArrayList();
         CONTAINER_MIXES = com.google.common.collect.Lists.newArrayList();
         ALLOWED_CONTAINERS = com.google.common.collect.Lists.newArrayList();
-        ALLOWED_CONTAINER = input -> {
-            for (net.minecraft.world.item.crafting.Ingredient ingredient : ALLOWED_CONTAINERS) {
-                if (!ingredient.test(input)) continue;
+        ALLOWED_CONTAINER = $$0 -> {
+            for (net.minecraft.world.item.crafting.Ingredient $$1 : ALLOWED_CONTAINERS) {
+                if (!$$1.test($$0)) continue;
                 return true;
             }
             return false;