--- net/minecraft/world/item/alchemy/PotionUtils.java
+++ net/minecraft/world/item/alchemy/PotionUtils.java
@@ -14,159 +14,163 @@
     public net.minecraft.world.item.alchemy.PotionUtils() {
     }
 
-    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getMobEffects(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.item.alchemy.PotionUtils.getAllEffects((net.minecraft.nbt.CompoundTag)stack.getTag());
+    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getMobEffects(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.item.alchemy.PotionUtils.getAllEffects((net.minecraft.nbt.CompoundTag)$$0.getTag());
     }
 
-    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getAllEffects(net.minecraft.world.item.alchemy.Potion potion, java.util.Collection<net.minecraft.world.effect.MobEffectInstance> effects) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        arrayList.addAll((java.util.Collection)potion.getEffects());
-        arrayList.addAll(effects);
-        return arrayList;
+    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getAllEffects(net.minecraft.world.item.alchemy.Potion $$0, java.util.Collection<net.minecraft.world.effect.MobEffectInstance> $$1) {
+        java.util.ArrayList $$2 = com.google.common.collect.Lists.newArrayList();
+        $$2.addAll((java.util.Collection)$$0.getEffects());
+        $$2.addAll($$1);
+        return $$2;
     }
 
-    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getAllEffects(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag compoundTag) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        arrayList.addAll((java.util.Collection)net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.nbt.CompoundTag)compoundTag).getEffects());
-        net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)compoundTag, (java.util.List)arrayList);
-        return arrayList;
+    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getAllEffects(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$0) {
+        java.util.ArrayList $$1 = com.google.common.collect.Lists.newArrayList();
+        $$1.addAll((java.util.Collection)net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.nbt.CompoundTag)$$0).getEffects());
+        net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)$$0, (java.util.List)$$1);
+        return $$1;
     }
 
-    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getCustomEffects(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)stack.getTag());
+    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getCustomEffects(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)$$0.getTag());
     }
 
-    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getCustomEffects(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag compoundTag) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)compoundTag, (java.util.List)arrayList);
-        return arrayList;
+    public static java.util.List<net.minecraft.world.effect.MobEffectInstance> getCustomEffects(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$0) {
+        java.util.ArrayList $$1 = com.google.common.collect.Lists.newArrayList();
+        net.minecraft.world.item.alchemy.PotionUtils.getCustomEffects((net.minecraft.nbt.CompoundTag)$$0, (java.util.List)$$1);
+        return $$1;
     }
 
-    public static void getCustomEffects(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag compoundTag, java.util.List<net.minecraft.world.effect.MobEffectInstance> effectList) {
-        if (compoundTag != null && compoundTag.contains("CustomPotionEffects", 9)) {
-            net.minecraft.nbt.ListTag listTag = compoundTag.getList("CustomPotionEffects", 10);
-            for (int i = 0; i < listTag.size(); ++i) {
-                net.minecraft.nbt.CompoundTag compoundTag2 = listTag.getCompound(i);
-                net.minecraft.world.effect.MobEffectInstance mobEffectInstance = net.minecraft.world.effect.MobEffectInstance.load((net.minecraft.nbt.CompoundTag)compoundTag2);
-                if (mobEffectInstance == null) continue;
-                effectList.add((java.lang.Object)mobEffectInstance);
+    public static void getCustomEffects(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$0, java.util.List<net.minecraft.world.effect.MobEffectInstance> $$1) {
+        if ($$0 != null && $$0.contains("CustomPotionEffects", 9)) {
+            net.minecraft.nbt.ListTag $$2 = $$0.getList("CustomPotionEffects", 10);
+            for (int $$3 = 0; $$3 < $$2.size(); ++$$3) {
+                net.minecraft.nbt.CompoundTag $$4 = $$2.getCompound($$3);
+                net.minecraft.world.effect.MobEffectInstance $$5 = net.minecraft.world.effect.MobEffectInstance.load((net.minecraft.nbt.CompoundTag)$$4);
+                if ($$5 == null) continue;
+                $$1.add((java.lang.Object)$$5);
             }
         }
     }
 
-    public static int getColor(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getTag();
-        if (compoundTag != null && compoundTag.contains("CustomPotionColor", 99)) {
-            return compoundTag.getInt("CustomPotionColor");
+    public static int getColor(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = $$0.getTag();
+        if ($$1 != null && $$1.contains("CustomPotionColor", 99)) {
+            return $$1.getInt("CustomPotionColor");
         }
-        return net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)stack) == net.minecraft.world.item.alchemy.Potions.EMPTY ? 0xF800F8 : net.minecraft.world.item.alchemy.PotionUtils.getColor((java.util.Collection)net.minecraft.world.item.alchemy.PotionUtils.getMobEffects((net.minecraft.world.item.ItemStack)stack));
+        return net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)$$0) == net.minecraft.world.item.alchemy.Potions.EMPTY ? 0xF800F8 : net.minecraft.world.item.alchemy.PotionUtils.getColor((java.util.Collection)net.minecraft.world.item.alchemy.PotionUtils.getMobEffects((net.minecraft.world.item.ItemStack)$$0));
     }
 
-    public static int getColor(net.minecraft.world.item.alchemy.Potion potion) {
-        return potion == net.minecraft.world.item.alchemy.Potions.EMPTY ? 0xF800F8 : net.minecraft.world.item.alchemy.PotionUtils.getColor((java.util.Collection)potion.getEffects());
+    public static int getColor(net.minecraft.world.item.alchemy.Potion $$0) {
+        return $$0 == net.minecraft.world.item.alchemy.Potions.EMPTY ? 0xF800F8 : net.minecraft.world.item.alchemy.PotionUtils.getColor((java.util.Collection)$$0.getEffects());
     }
 
-    public static int getColor(java.util.Collection<net.minecraft.world.effect.MobEffectInstance> effects) {
-        int n = 3694022;
-        if (effects.isEmpty()) {
+    public static int getColor(java.util.Collection<net.minecraft.world.effect.MobEffectInstance> $$0) {
+        int $$1 = 3694022;
+        if ($$0.isEmpty()) {
             return 3694022;
         }
-        float f = 0.0f;
-        float f2 = 0.0f;
-        float f3 = 0.0f;
-        int n2 = 0;
-        for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : effects) {
-            if (!mobEffectInstance.isVisible()) continue;
-            int n3 = mobEffectInstance.getEffect().getColor();
-            int n4 = mobEffectInstance.getAmplifier() + 1;
-            f += (float)(n4 * (n3 >> 16 & 0xFF)) / 255.0f;
-            f2 += (float)(n4 * (n3 >> 8 & 0xFF)) / 255.0f;
-            f3 += (float)(n4 * (n3 >> 0 & 0xFF)) / 255.0f;
-            n2 += n4;
+        float $$2 = 0.0f;
+        float $$3 = 0.0f;
+        float $$4 = 0.0f;
+        int $$5 = 0;
+        for (net.minecraft.world.effect.MobEffectInstance $$6 : $$0) {
+            if (!$$6.isVisible()) continue;
+            int $$7 = $$6.getEffect().getColor();
+            int $$8 = $$6.getAmplifier() + 1;
+            $$2 += (float)($$8 * ($$7 >> 16 & 0xFF)) / 255.0f;
+            $$3 += (float)($$8 * ($$7 >> 8 & 0xFF)) / 255.0f;
+            $$4 += (float)($$8 * ($$7 >> 0 & 0xFF)) / 255.0f;
+            $$5 += $$8;
         }
-        if (n2 == 0) {
+        if ($$5 == 0) {
             return 0;
         }
-        f = f / (float)n2 * 255.0f;
-        f2 = f2 / (float)n2 * 255.0f;
-        f3 = f3 / (float)n2 * 255.0f;
-        return (int)f << 16 | (int)f2 << 8 | (int)f3;
+        $$2 = $$2 / (float)$$5 * 255.0f;
+        $$3 = $$3 / (float)$$5 * 255.0f;
+        $$4 = $$4 / (float)$$5 * 255.0f;
+        return (int)$$2 << 16 | (int)$$3 << 8 | (int)$$4;
     }
 
-    public static net.minecraft.world.item.alchemy.Potion getPotion(net.minecraft.world.item.ItemStack stack) {
-        return net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.nbt.CompoundTag)stack.getTag());
+    public static net.minecraft.world.item.alchemy.Potion getPotion(net.minecraft.world.item.ItemStack $$0) {
+        return net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.nbt.CompoundTag)$$0.getTag());
     }
 
-    public static net.minecraft.world.item.alchemy.Potion getPotion(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag compoundTag) {
-        if (compoundTag == null) {
+    public static net.minecraft.world.item.alchemy.Potion getPotion(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$0) {
+        if ($$0 == null) {
             return net.minecraft.world.item.alchemy.Potions.EMPTY;
         }
-        return net.minecraft.world.item.alchemy.Potion.byName((java.lang.String)compoundTag.getString("Potion"));
+        return net.minecraft.world.item.alchemy.Potion.byName((java.lang.String)$$0.getString("Potion"));
     }
 
-    public static net.minecraft.world.item.ItemStack setPotion(net.minecraft.world.item.ItemStack stack, net.minecraft.world.item.alchemy.Potion potion) {
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.core.Registry.POTION.getKey((java.lang.Object)potion);
-        if (potion == net.minecraft.world.item.alchemy.Potions.EMPTY) {
-            stack.removeTagKey("Potion");
+    public static net.minecraft.world.item.ItemStack setPotion(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.alchemy.Potion $$1) {
+        net.minecraft.resources.ResourceLocation $$2 = net.minecraft.core.Registry.POTION.getKey((java.lang.Object)$$1);
+        if ($$1 == net.minecraft.world.item.alchemy.Potions.EMPTY) {
+            $$0.removeTagKey("Potion");
         } else {
-            stack.getOrCreateTag().putString("Potion", resourceLocation.toString());
+            $$0.getOrCreateTag().putString("Potion", $$2.toString());
         }
-        return stack;
+        return $$0;
     }
 
-    public static net.minecraft.world.item.ItemStack setCustomEffects(net.minecraft.world.item.ItemStack stack, java.util.Collection<net.minecraft.world.effect.MobEffectInstance> effects) {
-        if (effects.isEmpty()) {
-            return stack;
+    public static net.minecraft.world.item.ItemStack setCustomEffects(net.minecraft.world.item.ItemStack $$0, java.util.Collection<net.minecraft.world.effect.MobEffectInstance> $$1) {
+        if ($$1.isEmpty()) {
+            return $$0;
         }
-        net.minecraft.nbt.CompoundTag compoundTag = stack.getOrCreateTag();
-        net.minecraft.nbt.ListTag listTag = compoundTag.getList("CustomPotionEffects", 9);
-        for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : effects) {
-            listTag.add((java.lang.Object)mobEffectInstance.save(new net.minecraft.nbt.CompoundTag()));
+        net.minecraft.nbt.CompoundTag $$2 = $$0.getOrCreateTag();
+        net.minecraft.nbt.ListTag $$3 = $$2.getList("CustomPotionEffects", 9);
+        for (net.minecraft.world.effect.MobEffectInstance $$4 : $$1) {
+            $$3.add((java.lang.Object)$$4.save(new net.minecraft.nbt.CompoundTag()));
         }
-        compoundTag.put("CustomPotionEffects", (net.minecraft.nbt.Tag)listTag);
-        return stack;
+        $$2.put("CustomPotionEffects", (net.minecraft.nbt.Tag)$$3);
+        return $$0;
     }
 
-    public static void addPotionTooltip(net.minecraft.world.item.ItemStack stack, java.util.List<net.minecraft.network.chat.Component> tooltips, float durationFactor) {
-        net.minecraft.network.chat.MutableComponent mutableComponent;
-        java.util.List list = net.minecraft.world.item.alchemy.PotionUtils.getMobEffects((net.minecraft.world.item.ItemStack)stack);
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        if (list.isEmpty()) {
-            tooltips.add((java.lang.Object)NO_EFFECT);
+    public static void addPotionTooltip(net.minecraft.world.item.ItemStack $$0, java.util.List<net.minecraft.network.chat.Component> $$1, float $$2) {
+        java.util.List $$3 = net.minecraft.world.item.alchemy.PotionUtils.getMobEffects((net.minecraft.world.item.ItemStack)$$0);
+        java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayList();
+        if ($$3.isEmpty()) {
+            $$1.add((java.lang.Object)NO_EFFECT);
         } else {
-            for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : list) {
-                mutableComponent = net.minecraft.network.chat.Component.translatable((java.lang.String)mobEffectInstance.getDescriptionId());
-                net.minecraft.world.effect.MobEffect mobEffect = mobEffectInstance.getEffect();
-                java.util.Map map = mobEffect.getAttributeModifiers();
-                if (!map.isEmpty()) {
-                    for (java.util.Map$Entry entry : map.entrySet()) {
-                        net.minecraft.world.entity.ai.attributes.AttributeModifier attributeModifier = (net.minecraft.world.entity.ai.attributes.AttributeModifier)entry.getValue();
-                        net.minecraft.world.entity.ai.attributes.AttributeModifier attributeModifier2 = new net.minecraft.world.entity.ai.attributes.AttributeModifier(attributeModifier.getName(), mobEffect.getAttributeModifierValue(mobEffectInstance.getAmplifier(), attributeModifier), attributeModifier.getOperation());
-                        arrayList.add((java.lang.Object)new com.mojang.datafixers.util.Pair((java.lang.Object)((net.minecraft.world.entity.ai.attributes.Attribute)entry.getKey()), (java.lang.Object)attributeModifier2));
+            for (net.minecraft.world.effect.MobEffectInstance $$5 : $$3) {
+                net.minecraft.network.chat.MutableComponent $$6 = net.minecraft.network.chat.Component.translatable((java.lang.String)$$5.getDescriptionId());
+                net.minecraft.world.effect.MobEffect $$7 = $$5.getEffect();
+                java.util.Map $$8 = $$7.getAttributeModifiers();
+                if (!$$8.isEmpty()) {
+                    for (java.util.Map$Entry $$9 : $$8.entrySet()) {
+                        net.minecraft.world.entity.ai.attributes.AttributeModifier $$10 = (net.minecraft.world.entity.ai.attributes.AttributeModifier)$$9.getValue();
+                        net.minecraft.world.entity.ai.attributes.AttributeModifier $$11 = new net.minecraft.world.entity.ai.attributes.AttributeModifier($$10.getName(), $$7.getAttributeModifierValue($$5.getAmplifier(), $$10), $$10.getOperation());
+                        $$4.add((java.lang.Object)new com.mojang.datafixers.util.Pair((java.lang.Object)((net.minecraft.world.entity.ai.attributes.Attribute)$$9.getKey()), (java.lang.Object)$$11));
                     }
                 }
-                if (mobEffectInstance.getAmplifier() > 0) {
-                    mutableComponent = net.minecraft.network.chat.Component.translatable((java.lang.String)"potion.withAmplifier", (java.lang.Object[])new java.lang.Object[]{mutableComponent, net.minecraft.network.chat.Component.translatable((java.lang.String)("potion.potency." + mobEffectInstance.getAmplifier()))});
+                if ($$5.getAmplifier() > 0) {
+                    $$6 = net.minecraft.network.chat.Component.translatable((java.lang.String)"potion.withAmplifier", (java.lang.Object[])new java.lang.Object[]{$$6, net.minecraft.network.chat.Component.translatable((java.lang.String)("potion.potency." + $$5.getAmplifier()))});
                 }
-                if (mobEffectInstance.getDuration() > 20) {
-                    mutableComponent = net.minecraft.network.chat.Component.translatable((java.lang.String)"potion.withDuration", (java.lang.Object[])new java.lang.Object[]{mutableComponent, net.minecraft.world.effect.MobEffectUtil.formatDuration((net.minecraft.world.effect.MobEffectInstance)mobEffectInstance, (float)durationFactor)});
+                if ($$5.getDuration() > 20) {
+                    $$6 = net.minecraft.network.chat.Component.translatable((java.lang.String)"potion.withDuration", (java.lang.Object[])new java.lang.Object[]{$$6, net.minecraft.world.effect.MobEffectUtil.formatDuration((net.minecraft.world.effect.MobEffectInstance)$$5, (float)$$2)});
                 }
-                tooltips.add((java.lang.Object)mutableComponent.withStyle(mobEffect.getCategory().getTooltipFormatting()));
+                $$1.add((java.lang.Object)$$6.withStyle($$7.getCategory().getTooltipFormatting()));
             }
         }
-        if (!arrayList.isEmpty()) {
-            tooltips.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.EMPTY);
-            tooltips.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"potion.whenDrank").withStyle(net.minecraft.ChatFormatting.DARK_PURPLE));
-            for (net.minecraft.world.effect.MobEffectInstance mobEffectInstance : arrayList) {
-                mutableComponent = (net.minecraft.world.entity.ai.attributes.AttributeModifier)mobEffectInstance.getSecond();
-                double d = mutableComponent.getAmount();
-                double d2 = mutableComponent.getOperation() == net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation.MULTIPLY_BASE || mutableComponent.getOperation() == net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation.MULTIPLY_TOTAL ? mutableComponent.getAmount() * 100.0 : mutableComponent.getAmount();
-                if (d > 0.0) {
-                    tooltips.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("attribute.modifier.plus." + mutableComponent.getOperation().toValue()), (java.lang.Object[])new java.lang.Object[]{net.minecraft.world.item.ItemStack.ATTRIBUTE_MODIFIER_FORMAT.format(d2), net.minecraft.network.chat.Component.translatable((java.lang.String)((net.minecraft.world.entity.ai.attributes.Attribute)mobEffectInstance.getFirst()).getDescriptionId())}).withStyle(net.minecraft.ChatFormatting.BLUE));
+        if (!$$4.isEmpty()) {
+            $$1.add((java.lang.Object)net.minecraft.network.chat.CommonComponents.EMPTY);
+            $$1.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)"potion.whenDrank").withStyle(net.minecraft.ChatFormatting.DARK_PURPLE));
+            for (com.mojang.datafixers.util.Pair $$12 : $$4) {
+                double $$16;
+                net.minecraft.world.entity.ai.attributes.AttributeModifier $$13 = (net.minecraft.world.entity.ai.attributes.AttributeModifier)$$12.getSecond();
+                double $$14 = $$13.getAmount();
+                if ($$13.getOperation() == net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation.MULTIPLY_BASE || $$13.getOperation() == net.minecraft.world.entity.ai.attributes.AttributeModifier$Operation.MULTIPLY_TOTAL) {
+                    double $$15 = $$13.getAmount() * 100.0;
+                } else {
+                    $$16 = $$13.getAmount();
+                }
+                if ($$14 > 0.0) {
+                    $$1.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("attribute.modifier.plus." + $$13.getOperation().toValue()), (java.lang.Object[])new java.lang.Object[]{net.minecraft.world.item.ItemStack.ATTRIBUTE_MODIFIER_FORMAT.format($$16), net.minecraft.network.chat.Component.translatable((java.lang.String)((net.minecraft.world.entity.ai.attributes.Attribute)$$12.getFirst()).getDescriptionId())}).withStyle(net.minecraft.ChatFormatting.BLUE));
                     continue;
                 }
-                if (!(d < 0.0)) continue;
-                tooltips.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("attribute.modifier.take." + mutableComponent.getOperation().toValue()), (java.lang.Object[])new java.lang.Object[]{net.minecraft.world.item.ItemStack.ATTRIBUTE_MODIFIER_FORMAT.format(d2 *= -1.0), net.minecraft.network.chat.Component.translatable((java.lang.String)((net.minecraft.world.entity.ai.attributes.Attribute)mobEffectInstance.getFirst()).getDescriptionId())}).withStyle(net.minecraft.ChatFormatting.RED));
+                if (!($$14 < 0.0)) continue;
+                $$1.add((java.lang.Object)net.minecraft.network.chat.Component.translatable((java.lang.String)("attribute.modifier.take." + $$13.getOperation().toValue()), (java.lang.Object[])new java.lang.Object[]{net.minecraft.world.item.ItemStack.ATTRIBUTE_MODIFIER_FORMAT.format($$16 *= -1.0), net.minecraft.network.chat.Component.translatable((java.lang.String)((net.minecraft.world.entity.ai.attributes.Attribute)$$12.getFirst()).getDescriptionId())}).withStyle(net.minecraft.ChatFormatting.RED));
             }
         }
     }