--- net/minecraft/world/entity/AnimationState.java
+++ net/minecraft/world/entity/AnimationState.java
@@ -24,6 +24,14 @@
         }
     }
 
+    public void animateWhen(boolean $$0, int $$1) {
+        if ($$0) {
+            this.startIfStopped($$1);
+        } else {
+            this.stop();
+        }
+    }
+
     public void stop() {
         this.lastTime = java.lang.Long.MAX_VALUE;
     }