--- net/minecraft/world/entity/EntitySelector$MobCanWearArmorEntitySelector.java
+++ net/minecraft/world/entity/EntitySelector$MobCanWearArmorEntitySelector.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.item.ItemStack itemStack;
 
-    public net.minecraft.world.entity.EntitySelector$MobCanWearArmorEntitySelector(net.minecraft.world.item.ItemStack stack) {
-        this.itemStack = stack;
+    public net.minecraft.world.entity.EntitySelector$MobCanWearArmorEntitySelector(net.minecraft.world.item.ItemStack $$0) {
+        this.itemStack = $$0;
     }
 
     public boolean test(@javax.annotation.Nullable net.minecraft.world.entity.Entity $$0) {