--- net/minecraft/world/entity/EntityDimensions.java
+++ net/minecraft/world/entity/EntityDimensions.java
@@ -9,39 +9,39 @@
     public float height;
     public boolean fixed;
 
-    public net.minecraft.world.entity.EntityDimensions(float width, float height, boolean fixed) {
-        this.width = width;
-        this.height = height;
-        this.fixed = fixed;
+    public net.minecraft.world.entity.EntityDimensions(float $$0, float $$1, boolean $$2) {
+        this.width = $$0;
+        this.height = $$1;
+        this.fixed = $$2;
     }
 
-    public net.minecraft.world.phys.AABB makeBoundingBox(net.minecraft.world.phys.Vec3 pos) {
-        return this.makeBoundingBox(pos.x, pos.y, pos.z);
+    public net.minecraft.world.phys.AABB makeBoundingBox(net.minecraft.world.phys.Vec3 $$0) {
+        return this.makeBoundingBox($$0.x, $$0.y, $$0.z);
     }
 
-    public net.minecraft.world.phys.AABB makeBoundingBox(double x, double y, double z) {
-        float f = this.width / 2.0f;
-        float f2 = this.height;
-        return new net.minecraft.world.phys.AABB(x - (double)f, y, z - (double)f, x + (double)f, y + (double)f2, z + (double)f);
+    public net.minecraft.world.phys.AABB makeBoundingBox(double $$0, double $$1, double $$2) {
+        float $$3 = this.width / 2.0f;
+        float $$4 = this.height;
+        return new net.minecraft.world.phys.AABB($$0 - (double)$$3, $$1, $$2 - (double)$$3, $$0 + (double)$$3, $$1 + (double)$$4, $$2 + (double)$$3);
     }
 
-    public net.minecraft.world.entity.EntityDimensions scale(float factor) {
-        return this.scale(factor, factor);
+    public net.minecraft.world.entity.EntityDimensions scale(float $$0) {
+        return this.scale($$0, $$0);
     }
 
-    public net.minecraft.world.entity.EntityDimensions scale(float widthFactor, float heightFactor) {
-        if (this.fixed || widthFactor == 1.0f && heightFactor == 1.0f) {
+    public net.minecraft.world.entity.EntityDimensions scale(float $$0, float $$1) {
+        if (this.fixed || $$0 == 1.0f && $$1 == 1.0f) {
             return this;
         }
-        return net.minecraft.world.entity.EntityDimensions.scalable((float)(this.width * widthFactor), (float)(this.height * heightFactor));
+        return net.minecraft.world.entity.EntityDimensions.scalable((float)(this.width * $$0), (float)(this.height * $$1));
     }
 
-    public static net.minecraft.world.entity.EntityDimensions scalable(float width, float height) {
-        return new net.minecraft.world.entity.EntityDimensions(width, height, false);
+    public static net.minecraft.world.entity.EntityDimensions scalable(float $$0, float $$1) {
+        return new net.minecraft.world.entity.EntityDimensions($$0, $$1, false);
     }
 
-    public static net.minecraft.world.entity.EntityDimensions fixed(float width, float height) {
-        return new net.minecraft.world.entity.EntityDimensions(width, height, true);
+    public static net.minecraft.world.entity.EntityDimensions fixed(float $$0, float $$1) {
+        return new net.minecraft.world.entity.EntityDimensions($$0, $$1, true);
     }
 
     public java.lang.String toString() {