--- net/minecraft/world/entity/EquipmentSlot.java
+++ net/minecraft/world/entity/EquipmentSlot.java
@@ -26,12 +26,12 @@
         return (net.minecraft.world.entity.EquipmentSlot)java.lang.Enum.valueOf(net.minecraft.world.entity.EquipmentSlot.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.EquipmentSlot(net.minecraft.world.entity.EquipmentSlot$Type type, int index, int filterFlag, java.lang.String name) {
+    public net.minecraft.world.entity.EquipmentSlot(net.minecraft.world.entity.EquipmentSlot$Type $$0, int $$1, int $$2, java.lang.String $$3) {
         super(string, n);
-        this.type = type;
-        this.index = index;
-        this.filterFlag = filterFlag;
-        this.name = name;
+        this.type = $$0;
+        this.index = $$1;
+        this.filterFlag = $$2;
+        this.name = $$3;
     }
 
     public net.minecraft.world.entity.EquipmentSlot$Type getType() {
@@ -54,20 +54,20 @@
         return this.name;
     }
 
-    public static net.minecraft.world.entity.EquipmentSlot byName(java.lang.String targetName) {
-        for (net.minecraft.world.entity.EquipmentSlot equipmentSlot : net.minecraft.world.entity.EquipmentSlot.values()) {
-            if (!equipmentSlot.getName().equals((java.lang.Object)targetName)) continue;
-            return equipmentSlot;
+    public static net.minecraft.world.entity.EquipmentSlot byName(java.lang.String $$0) {
+        for (net.minecraft.world.entity.EquipmentSlot $$1 : net.minecraft.world.entity.EquipmentSlot.values()) {
+            if (!$$1.getName().equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
-        throw new java.lang.IllegalArgumentException("Invalid slot '" + targetName + "'");
+        throw new java.lang.IllegalArgumentException("Invalid slot '" + $$0 + "'");
     }
 
-    public static net.minecraft.world.entity.EquipmentSlot byTypeAndIndex(net.minecraft.world.entity.EquipmentSlot$Type slotType, int slotIndex) {
-        for (net.minecraft.world.entity.EquipmentSlot equipmentSlot : net.minecraft.world.entity.EquipmentSlot.values()) {
-            if (equipmentSlot.getType() != slotType || equipmentSlot.getIndex() != slotIndex) continue;
-            return equipmentSlot;
+    public static net.minecraft.world.entity.EquipmentSlot byTypeAndIndex(net.minecraft.world.entity.EquipmentSlot$Type $$0, int $$1) {
+        for (net.minecraft.world.entity.EquipmentSlot $$2 : net.minecraft.world.entity.EquipmentSlot.values()) {
+            if ($$2.getType() != $$0 || $$2.getIndex() != $$1) continue;
+            return $$2;
         }
-        throw new java.lang.IllegalArgumentException("Invalid slot '" + slotType + "': " + slotIndex);
+        throw new java.lang.IllegalArgumentException("Invalid slot '" + $$0 + "': " + $$1);
     }
 
     public static /* synthetic */ net.minecraft.world.entity.EquipmentSlot[] $values() {