--- net/minecraft/world/entity/ItemBasedSteering.java
+++ net/minecraft/world/entity/ItemBasedSteering.java
@@ -14,10 +14,10 @@
     public int boostTime;
     public int boostTimeTotal;
 
-    public net.minecraft.world.entity.ItemBasedSteering(net.minecraft.network.syncher.SynchedEntityData entityData, net.minecraft.network.syncher.EntityDataAccessor<java.lang.Integer> boostTimeAccessor, net.minecraft.network.syncher.EntityDataAccessor<java.lang.Boolean> hasSaddleAccessor) {
-        this.entityData = entityData;
-        this.boostTimeAccessor = boostTimeAccessor;
-        this.hasSaddleAccessor = hasSaddleAccessor;
+    public net.minecraft.world.entity.ItemBasedSteering(net.minecraft.network.syncher.SynchedEntityData $$0, net.minecraft.network.syncher.EntityDataAccessor<java.lang.Integer> $$1, net.minecraft.network.syncher.EntityDataAccessor<java.lang.Boolean> $$2) {
+        this.entityData = $$0;
+        this.boostTimeAccessor = $$1;
+        this.hasSaddleAccessor = $$2;
     }
 
     public void onSynced() {
@@ -37,16 +37,16 @@
         return true;
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag nbt) {
-        nbt.putBoolean("Saddle", this.hasSaddle());
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        $$0.putBoolean("Saddle", this.hasSaddle());
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag nbt) {
-        this.setSaddle(nbt.getBoolean("Saddle"));
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        this.setSaddle($$0.getBoolean("Saddle"));
     }
 
-    public void setSaddle(boolean saddled) {
-        this.entityData.set(this.hasSaddleAccessor, (java.lang.Object)saddled);
+    public void setSaddle(boolean $$0) {
+        this.entityData.set(this.hasSaddleAccessor, (java.lang.Object)$$0);
     }
 
     public boolean hasSaddle() {