--- net/minecraft/world/entity/LightningBolt.java
+++ net/minecraft/world/entity/LightningBolt.java
@@ -27,8 +27,8 @@
         this.flashes = this.random.nextInt(3) + 1;
     }
 
-    public void setVisualOnly(boolean visualOnly) {
-        this.visualOnly = visualOnly;
+    public void setVisualOnly(boolean $$0) {
+        this.visualOnly = $$0;
     }
 
     public net.minecraft.sounds.SoundSource getSoundSource() {
@@ -40,8 +40,8 @@
         return this.cause;
     }
 
-    public void setCause(@javax.annotation.Nullable net.minecraft.server.level.ServerPlayer cause) {
-        this.cause = cause;
+    public void setCause(@javax.annotation.Nullable net.minecraft.server.level.ServerPlayer $$0) {
+        this.cause = $$0;
     }
 
     public void powerLightningRod() {
@@ -106,21 +106,21 @@
         return new net.minecraft.core.BlockPos($$0.x, $$0.y - 1.0E-6, $$0.z);
     }
 
-    public void spawnFire(int extraIgnitions) {
+    public void spawnFire(int $$0) {
         if (this.visualOnly || this.level.isClientSide || !this.level.getGameRules().getBoolean(net.minecraft.world.level.GameRules.RULE_DOFIRETICK)) {
             return;
         }
-        net.minecraft.core.BlockPos blockPos = this.blockPosition();
-        net.minecraft.world.level.block.state.BlockState blockState = net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)this.level, (net.minecraft.core.BlockPos)blockPos);
-        if (this.level.getBlockState(blockPos).isAir() && blockState.canSurvive((net.minecraft.world.level.LevelReader)this.level, blockPos)) {
-            this.level.setBlockAndUpdate(blockPos, blockState);
+        net.minecraft.core.BlockPos $$1 = this.blockPosition();
+        net.minecraft.world.level.block.state.BlockState $$2 = net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)this.level, (net.minecraft.core.BlockPos)$$1);
+        if (this.level.getBlockState($$1).isAir() && $$2.canSurvive((net.minecraft.world.level.LevelReader)this.level, $$1)) {
+            this.level.setBlockAndUpdate($$1, $$2);
             ++this.blocksSetOnFire;
         }
-        for (int i = 0; i < extraIgnitions; ++i) {
-            net.minecraft.core.BlockPos blockPos2 = blockPos.offset(this.random.nextInt(3) - 1, this.random.nextInt(3) - 1, this.random.nextInt(3) - 1);
-            blockState = net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)this.level, (net.minecraft.core.BlockPos)blockPos2);
-            if (!this.level.getBlockState(blockPos2).isAir() || !blockState.canSurvive((net.minecraft.world.level.LevelReader)this.level, blockPos2)) continue;
-            this.level.setBlockAndUpdate(blockPos2, blockState);
+        for (int $$3 = 0; $$3 < $$0; ++$$3) {
+            net.minecraft.core.BlockPos $$4 = $$1.offset(this.random.nextInt(3) - 1, this.random.nextInt(3) - 1, this.random.nextInt(3) - 1);
+            $$2 = net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)this.level, (net.minecraft.core.BlockPos)$$4);
+            if (!this.level.getBlockState($$4).isAir() || !$$2.canSurvive((net.minecraft.world.level.LevelReader)this.level, $$4)) continue;
+            this.level.setBlockAndUpdate($$4, $$2);
             ++this.blocksSetOnFire;
         }
     }
@@ -167,24 +167,20 @@
         return java.util.Optional.empty();
     }
 
-    public boolean shouldRenderAtSqrDistance(double distance) {
-        double d = 64.0 * net.minecraft.world.entity.LightningBolt.getViewScale();
-        return distance < d * d;
+    public boolean shouldRenderAtSqrDistance(double $$0) {
+        double $$1 = 64.0 * net.minecraft.world.entity.LightningBolt.getViewScale();
+        return $$0 < $$1 * $$1;
     }
 
     public void defineSynchedData() {
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
     }
 
-    public net.minecraft.network.protocol.Packet<?> getAddEntityPacket() {
-        return new net.minecraft.network.protocol.game.ClientboundAddEntityPacket((net.minecraft.world.entity.Entity)this);
-    }
-
     public int getBlocksSetOnFire() {
         return this.blocksSetOnFire;
     }