--- net/minecraft/world/entity/MobCategory.java
+++ net/minecraft/world/entity/MobCategory.java
@@ -32,14 +32,14 @@
         return (net.minecraft.world.entity.MobCategory)java.lang.Enum.valueOf(net.minecraft.world.entity.MobCategory.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.MobCategory(java.lang.String name, int max, boolean isFriendly, boolean isPersistent, int despawnDistance) {
+    public net.minecraft.world.entity.MobCategory(java.lang.String $$0, int $$1, boolean $$2, boolean $$3, int $$4) {
         super(string, n);
         this.noDespawnDistance = 32;
-        this.name = name;
-        this.max = max;
-        this.isFriendly = isFriendly;
-        this.isPersistent = isPersistent;
-        this.despawnDistance = despawnDistance;
+        this.name = $$0;
+        this.max = $$1;
+        this.isFriendly = $$2;
+        this.isPersistent = $$3;
+        this.despawnDistance = $$4;
     }
 
     public java.lang.String getName() {