--- net/minecraft/world/entity/PathfinderMob.java
+++ net/minecraft/world/entity/PathfinderMob.java
@@ -12,16 +12,16 @@
         super($$0, $$1);
     }
 
-    public float getWalkTargetValue(net.minecraft.core.BlockPos pos) {
-        return this.getWalkTargetValue(pos, (net.minecraft.world.level.LevelReader)this.level);
+    public float getWalkTargetValue(net.minecraft.core.BlockPos $$0) {
+        return this.getWalkTargetValue($$0, (net.minecraft.world.level.LevelReader)this.level);
     }
 
-    public float getWalkTargetValue(net.minecraft.core.BlockPos pos, net.minecraft.world.level.LevelReader level) {
+    public float getWalkTargetValue(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.LevelReader $$1) {
         return 0.0f;
     }
 
-    public boolean checkSpawnRules(net.minecraft.world.level.LevelAccessor level, net.minecraft.world.entity.MobSpawnType spawnReason) {
-        return this.getWalkTargetValue(this.blockPosition(), (net.minecraft.world.level.LevelReader)level) >= 0.0f;
+    public boolean checkSpawnRules(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.world.entity.MobSpawnType $$1) {
+        return this.getWalkTargetValue(this.blockPosition(), (net.minecraft.world.level.LevelReader)$$0) >= 0.0f;
     }
 
     public boolean isPathFinding() {
@@ -49,6 +49,7 @@
                 double $$3 = ($$0.getY() - this.getY()) / (double)$$1;
                 double $$4 = ($$0.getZ() - this.getZ()) / (double)$$1;
                 this.setDeltaMovement(this.getDeltaMovement().add(java.lang.Math.copySign((double)($$2 * $$2 * 0.4), (double)$$2), java.lang.Math.copySign((double)($$3 * $$3 * 0.4), (double)$$3), java.lang.Math.copySign((double)($$4 * $$4 * 0.4), (double)$$4)));
+                this.checkSlowFallDistance();
             } else if (this.shouldStayCloseToLeashHolder()) {
                 this.goalSelector.enableControlFlag(net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE);
                 float $$5 = 2.0f;
@@ -66,7 +67,7 @@
         return 1.0;
     }
 
-    public void onLeashDistance(float distance) {
+    public void onLeashDistance(float $$0) {
     }
 }
 