--- net/minecraft/world/entity/Pose.java
+++ net/minecraft/world/entity/Pose.java
@@ -16,6 +16,7 @@
     public static /* enum */ net.minecraft.world.entity.Pose DYING;
     public static /* enum */ net.minecraft.world.entity.Pose CROAKING;
     public static /* enum */ net.minecraft.world.entity.Pose USING_TONGUE;
+    public static /* enum */ net.minecraft.world.entity.Pose SITTING;
     public static /* enum */ net.minecraft.world.entity.Pose ROARING;
     public static /* enum */ net.minecraft.world.entity.Pose SNIFFING;
     public static /* enum */ net.minecraft.world.entity.Pose EMERGING;
@@ -35,7 +36,7 @@
     }
 
     public static /* synthetic */ net.minecraft.world.entity.Pose[] $values() {
-        return new net.minecraft.world.entity.Pose[]{STANDING, FALL_FLYING, SLEEPING, SWIMMING, SPIN_ATTACK, CROUCHING, LONG_JUMPING, DYING, CROAKING, USING_TONGUE, ROARING, SNIFFING, EMERGING, DIGGING};
+        return new net.minecraft.world.entity.Pose[]{STANDING, FALL_FLYING, SLEEPING, SWIMMING, SPIN_ATTACK, CROUCHING, LONG_JUMPING, DYING, CROAKING, USING_TONGUE, SITTING, ROARING, SNIFFING, EMERGING, DIGGING};
     }
 
     static {
@@ -49,10 +50,11 @@
         DYING = new net.minecraft.world.entity.Pose("DYING", 7);
         CROAKING = new net.minecraft.world.entity.Pose("CROAKING", 8);
         USING_TONGUE = new net.minecraft.world.entity.Pose("USING_TONGUE", 9);
-        ROARING = new net.minecraft.world.entity.Pose("ROARING", 10);
-        SNIFFING = new net.minecraft.world.entity.Pose("SNIFFING", 11);
-        EMERGING = new net.minecraft.world.entity.Pose("EMERGING", 12);
-        DIGGING = new net.minecraft.world.entity.Pose("DIGGING", 13);
+        SITTING = new net.minecraft.world.entity.Pose("SITTING", 10);
+        ROARING = new net.minecraft.world.entity.Pose("ROARING", 11);
+        SNIFFING = new net.minecraft.world.entity.Pose("SNIFFING", 12);
+        EMERGING = new net.minecraft.world.entity.Pose("EMERGING", 13);
+        DIGGING = new net.minecraft.world.entity.Pose("DIGGING", 14);
         $VALUES = net.minecraft.world.entity.Pose.$values();
     }
 }