--- net/minecraft/tags/BlockTags.java
+++ net/minecraft/tags/BlockTags.java
@@ -7,6 +7,7 @@
  {
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> WOOL;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> PLANKS;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> STRIPPED_LOGS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> STONE_BRICKS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> WOODEN_BUTTONS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> BUTTONS;
@@ -76,6 +77,9 @@
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> STANDING_SIGNS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> WALL_SIGNS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> SIGNS;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> CEILING_HANGING_SIGNS;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> WALL_HANGING_SIGNS;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> ALL_HANGING_SIGNS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> DRAGON_IMMUNE;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> DRAGON_TRANSPARENT;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> WITHER_IMMUNE;
@@ -157,6 +161,7 @@
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> SNAPS_GOAT_HORN;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> SNOW_LAYER_CANNOT_SURVIVE_ON;
     public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> SNOW_LAYER_CAN_SURVIVE_ON;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.level.block.Block> INVALID_SPAWN_INSIDE;
 
     public net.minecraft.tags.BlockTags() {
     }
@@ -168,6 +173,7 @@
     static {
         WOOL = net.minecraft.tags.BlockTags.create((java.lang.String)"wool");
         PLANKS = net.minecraft.tags.BlockTags.create((java.lang.String)"planks");
+        STRIPPED_LOGS = net.minecraft.tags.BlockTags.create((java.lang.String)"stripped_logs");
         STONE_BRICKS = net.minecraft.tags.BlockTags.create((java.lang.String)"stone_bricks");
         WOODEN_BUTTONS = net.minecraft.tags.BlockTags.create((java.lang.String)"wooden_buttons");
         BUTTONS = net.minecraft.tags.BlockTags.create((java.lang.String)"buttons");
@@ -237,6 +243,9 @@
         STANDING_SIGNS = net.minecraft.tags.BlockTags.create((java.lang.String)"standing_signs");
         WALL_SIGNS = net.minecraft.tags.BlockTags.create((java.lang.String)"wall_signs");
         SIGNS = net.minecraft.tags.BlockTags.create((java.lang.String)"signs");
+        CEILING_HANGING_SIGNS = net.minecraft.tags.BlockTags.create((java.lang.String)"ceiling_hanging_signs");
+        WALL_HANGING_SIGNS = net.minecraft.tags.BlockTags.create((java.lang.String)"wall_hanging_signs");
+        ALL_HANGING_SIGNS = net.minecraft.tags.BlockTags.create((java.lang.String)"all_hanging_signs");
         DRAGON_IMMUNE = net.minecraft.tags.BlockTags.create((java.lang.String)"dragon_immune");
         DRAGON_TRANSPARENT = net.minecraft.tags.BlockTags.create((java.lang.String)"dragon_transparent");
         WITHER_IMMUNE = net.minecraft.tags.BlockTags.create((java.lang.String)"wither_immune");
@@ -318,6 +327,7 @@
         SNAPS_GOAT_HORN = net.minecraft.tags.BlockTags.create((java.lang.String)"snaps_goat_horn");
         SNOW_LAYER_CANNOT_SURVIVE_ON = net.minecraft.tags.BlockTags.create((java.lang.String)"snow_layer_cannot_survive_on");
         SNOW_LAYER_CAN_SURVIVE_ON = net.minecraft.tags.BlockTags.create((java.lang.String)"snow_layer_can_survive_on");
+        INVALID_SPAWN_INSIDE = net.minecraft.tags.BlockTags.create((java.lang.String)"invalid_spawn_inside");
     }
 }
 