--- net/minecraft/tags/ItemTags.java
+++ net/minecraft/tags/ItemTags.java
@@ -7,6 +7,7 @@
  {
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> WOOL;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> PLANKS;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> STRIPPED_LOGS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> STONE_BRICKS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> WOODEN_BUTTONS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> BUTTONS;
@@ -74,6 +75,7 @@
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> COALS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> ARROWS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> LECTERN_BOOKS;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> BOOKSHELF_BOOKS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> BEACON_PAYMENT_ITEMS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> STONE_TOOL_MATERIALS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> STONE_CRAFTING_MATERIALS;
@@ -82,6 +84,7 @@
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> DAMPENS_VIBRATIONS;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> CLUSTER_MAX_HARVESTABLES;
     public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> COMPASSES;
+    public static net.minecraft.tags.TagKey<net.minecraft.world.item.Item> HANGING_SIGNS;
 
     public net.minecraft.tags.ItemTags() {
     }
@@ -93,6 +96,7 @@
     static {
         WOOL = net.minecraft.tags.ItemTags.bind((java.lang.String)"wool");
         PLANKS = net.minecraft.tags.ItemTags.bind((java.lang.String)"planks");
+        STRIPPED_LOGS = net.minecraft.tags.ItemTags.bind((java.lang.String)"stripped_logs");
         STONE_BRICKS = net.minecraft.tags.ItemTags.bind((java.lang.String)"stone_bricks");
         WOODEN_BUTTONS = net.minecraft.tags.ItemTags.bind((java.lang.String)"wooden_buttons");
         BUTTONS = net.minecraft.tags.ItemTags.bind((java.lang.String)"buttons");
@@ -160,6 +164,7 @@
         COALS = net.minecraft.tags.ItemTags.bind((java.lang.String)"coals");
         ARROWS = net.minecraft.tags.ItemTags.bind((java.lang.String)"arrows");
         LECTERN_BOOKS = net.minecraft.tags.ItemTags.bind((java.lang.String)"lectern_books");
+        BOOKSHELF_BOOKS = net.minecraft.tags.ItemTags.bind((java.lang.String)"bookshelf_books");
         BEACON_PAYMENT_ITEMS = net.minecraft.tags.ItemTags.bind((java.lang.String)"beacon_payment_items");
         STONE_TOOL_MATERIALS = net.minecraft.tags.ItemTags.bind((java.lang.String)"stone_tool_materials");
         STONE_CRAFTING_MATERIALS = net.minecraft.tags.ItemTags.bind((java.lang.String)"stone_crafting_materials");
@@ -168,6 +173,7 @@
         DAMPENS_VIBRATIONS = net.minecraft.tags.ItemTags.bind((java.lang.String)"dampens_vibrations");
         CLUSTER_MAX_HARVESTABLES = net.minecraft.tags.ItemTags.bind((java.lang.String)"cluster_max_harvestables");
         COMPASSES = net.minecraft.tags.ItemTags.bind((java.lang.String)"compasses");
+        HANGING_SIGNS = net.minecraft.tags.ItemTags.bind((java.lang.String)"hanging_signs");
     }
 }
 