--- net/minecraft/world/level/timers/FunctionCallback.java
+++ net/minecraft/world/level/timers/FunctionCallback.java
@@ -12,9 +12,9 @@
         this.functionId = $$0;
     }
 
-    public void handle(net.minecraft.server.MinecraftServer obj, net.minecraft.world.level.timers.TimerQueue<net.minecraft.server.MinecraftServer> manager, long gameTime) {
-        net.minecraft.server.ServerFunctionManager serverFunctionManager = obj.getFunctions();
-        serverFunctionManager.get(this.functionId).ifPresent($$1 -> serverFunctionManager.execute($$1, serverFunctionManager.getGameLoopSender()));
+    public void handle(net.minecraft.server.MinecraftServer $$0, net.minecraft.world.level.timers.TimerQueue<net.minecraft.server.MinecraftServer> $$12, long $$2) {
+        net.minecraft.server.ServerFunctionManager $$3 = $$0.getFunctions();
+        $$3.get(this.functionId).ifPresent($$1 -> $$3.execute($$1, $$3.getGameLoopSender()));
     }
 
     public /* synthetic */ void handle(java.lang.Object object, net.minecraft.world.level.timers.TimerQueue timerQueue, long l) {