--- net/minecraft/world/level/timers/TimerQueue.java
+++ net/minecraft/world/level/timers/TimerQueue.java
@@ -39,12 +39,12 @@
         this.callbacksRegistry = $$0;
     }
 
-    public void tick(T t, long gameTime) {
-        net.minecraft.world.level.timers.TimerQueue$Event event;
-        while ((event = (net.minecraft.world.level.timers.TimerQueue$Event)this.queue.peek()) != null && event.triggerTime <= gameTime) {
+    public void tick(T $$0, long $$1) {
+        net.minecraft.world.level.timers.TimerQueue$Event $$2;
+        while (($$2 = (net.minecraft.world.level.timers.TimerQueue$Event)this.queue.peek()) != null && $$2.triggerTime <= $$1) {
             this.queue.remove();
-            this.events.remove((java.lang.Object)event.id, (java.lang.Object)gameTime);
-            event.callback.handle(t, this, gameTime);
+            this.events.remove((java.lang.Object)$$2.id, (java.lang.Object)$$1);
+            $$2.callback.handle($$0, this, $$1);
         }
     }
 