--- net/minecraft/client/model/AllayModel.java
+++ net/minecraft/client/model/AllayModel.java
@@ -14,11 +14,12 @@
     public net.minecraft.client.model.geom.ModelPart left_arm;
     public net.minecraft.client.model.geom.ModelPart right_wing;
     public net.minecraft.client.model.geom.ModelPart left_wing;
-    public static float FLYING_ANIMATION_X_ROT = 0.6981317f;
-    public static float MAX_HAND_HOLDING_ITEM_X_ROT_RAD = -0.7853982f;
+    public static float FLYING_ANIMATION_X_ROT = 0.7853982f;
+    public static float MAX_HAND_HOLDING_ITEM_X_ROT_RAD = -1.134464f;
     public static float MIN_HAND_HOLDING_ITEM_X_ROT_RAD = -1.0471976f;
 
     public net.minecraft.client.model.AllayModel(net.minecraft.client.model.geom.ModelPart $$0) {
+        super(net.minecraft.client.renderer.RenderType::entityTranslucent);
         this.root = $$0.getChild("root");
         this.head = this.root.getChild("head");
         this.body = this.root.getChild("body");
@@ -40,16 +41,15 @@
         net.minecraft.client.model.geom.builders.PartDefinition $$3 = $$2.addOrReplaceChild("body", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 10).addBox(-1.5f, 0.0f, -1.0f, 3.0f, 4.0f, 2.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(0.0f)).texOffs(0, 16).addBox(-1.5f, 0.0f, -1.0f, 3.0f, 5.0f, 2.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(-0.2f)), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)-4.0f, (float)0.0f));
         $$3.addOrReplaceChild("right_arm", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(23, 0).addBox(-0.75f, -0.5f, -1.0f, 1.0f, 4.0f, 2.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(-0.01f)), net.minecraft.client.model.geom.PartPose.offset((float)-1.75f, (float)0.5f, (float)0.0f));
         $$3.addOrReplaceChild("left_arm", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(23, 6).addBox(-0.25f, -0.5f, -1.0f, 1.0f, 4.0f, 2.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(-0.01f)), net.minecraft.client.model.geom.PartPose.offset((float)1.75f, (float)0.5f, (float)0.0f));
-        $$3.addOrReplaceChild("right_wing", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(16, 14).addBox(0.0f, 1.0f, 0.0f, 0.0f, 5.0f, 8.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(0.0f)), net.minecraft.client.model.geom.PartPose.offset((float)-0.5f, (float)0.0f, (float)0.65f));
-        $$3.addOrReplaceChild("left_wing", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(16, 14).addBox(0.0f, 1.0f, 0.0f, 0.0f, 5.0f, 8.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(0.0f)), net.minecraft.client.model.geom.PartPose.offset((float)0.5f, (float)0.0f, (float)0.65f));
+        $$3.addOrReplaceChild("right_wing", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(16, 14).addBox(0.0f, 1.0f, 0.0f, 0.0f, 5.0f, 8.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(0.0f)), net.minecraft.client.model.geom.PartPose.offset((float)-0.5f, (float)0.0f, (float)0.6f));
+        $$3.addOrReplaceChild("left_wing", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(16, 14).addBox(0.0f, 1.0f, 0.0f, 0.0f, 5.0f, 8.0f, new net.minecraft.client.model.geom.builders.CubeDeformation(0.0f)), net.minecraft.client.model.geom.PartPose.offset((float)0.5f, (float)0.0f, (float)0.6f));
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)32, (int)32);
     }
 
     public void setupAnim(net.minecraft.world.entity.animal.allay.Allay $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
-        float $$18;
         this.root().getAllParts().forEach(net.minecraft.client.model.geom.ModelPart::resetPose);
-        float $$6 = $$3 * 20.0f * ((float)java.lang.Math.PI / 180) + $$2;
-        float $$7 = net.minecraft.util.Mth.cos((float)$$6) * (float)java.lang.Math.PI * 0.15f;
+        float $$6 = $$3 * 20.0f * ((float)java.lang.Math.PI / 180) + $$1;
+        float $$7 = net.minecraft.util.Mth.cos((float)$$6) * (float)java.lang.Math.PI * 0.15f + $$2;
         float $$8 = $$3 - (float)$$0.tickCount;
         float $$9 = $$3 * 9.0f * ((float)java.lang.Math.PI / 180);
         float $$10 = java.lang.Math.min((float)($$2 / 0.3f), (float)1.0f);
@@ -69,34 +69,30 @@
             this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
             this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
         }
-        this.right_wing.xRot = 0.43633232f;
-        this.right_wing.yRot = -0.61086524f + $$7;
-        this.left_wing.xRot = 0.43633232f;
-        this.left_wing.yRot = 0.61086524f - $$7;
-        this.body.xRot = $$18 = $$10 * 0.6981317f;
-        float $$19 = net.minecraft.util.Mth.lerp((float)$$12, (float)$$18, (float)net.minecraft.util.Mth.lerp((float)$$10, (float)-1.0471976f, (float)-0.7853982f));
+        this.right_wing.xRot = 0.43633232f * (1.0f - $$10);
+        this.right_wing.yRot = -0.7853982f + $$7;
+        this.left_wing.xRot = 0.43633232f * (1.0f - $$10);
+        this.left_wing.yRot = 0.7853982f - $$7;
+        this.body.xRot = $$10 * 0.7853982f;
+        float $$18 = $$12 * net.minecraft.util.Mth.lerp((float)$$10, (float)-1.0471976f, (float)-1.134464f);
         this.root.y += (float)java.lang.Math.cos((double)$$9) * 0.25f * $$11;
-        this.right_arm.xRot = $$19;
-        this.left_arm.xRot = $$19;
-        float $$20 = $$11 * (1.0f - $$12);
-        float $$21 = 0.43633232f - net.minecraft.util.Mth.cos((float)($$9 + 4.712389f)) * (float)java.lang.Math.PI * 0.075f * $$20;
-        this.left_arm.zRot = -$$21;
-        this.right_arm.zRot = $$21;
+        this.right_arm.xRot = $$18;
+        this.left_arm.xRot = $$18;
+        float $$19 = $$11 * (1.0f - $$12);
+        float $$20 = 0.43633232f - net.minecraft.util.Mth.cos((float)($$9 + 4.712389f)) * (float)java.lang.Math.PI * 0.075f * $$19;
+        this.left_arm.zRot = -$$20;
+        this.right_arm.zRot = $$20;
         this.right_arm.yRot = 0.27925268f * $$12;
         this.left_arm.yRot = -0.27925268f * $$12;
     }
 
-    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7) {
-        this.root.render($$0, $$1, $$2, $$3);
-    }
-
     public void translateToHand(net.minecraft.world.entity.HumanoidArm $$0, com.mojang.blaze3d.vertex.PoseStack $$1) {
-        float $$2 = -1.5f;
-        float $$3 = 1.5f;
+        float $$2 = 1.0f;
+        float $$3 = 3.0f;
         this.root.translateAndRotate($$1);
         this.body.translateAndRotate($$1);
-        $$1.translate(0.0, -0.09375, 0.09375);
-        $$1.mulPose(com.mojang.math.Vector3f.XP.rotation(this.right_arm.xRot + 0.43633232f));
+        $$1.translate(0.0, 0.0625, 0.1875);
+        $$1.mulPose(com.mojang.math.Vector3f.XP.rotation(this.right_arm.xRot));
         $$1.scale(0.7f, 0.7f, 0.7f);
         $$1.translate(0.0625, 0.0, 0.0);
     }