--- net/minecraft/client/model/ArmorStandModel.java
+++ net/minecraft/client/model/ArmorStandModel.java
@@ -40,38 +40,38 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)64);
     }
 
-    public void prepareMobModel(net.minecraft.world.entity.decoration.ArmorStand entity, float limbSwing, float limbSwingAmount, float partialTick) {
+    public void prepareMobModel(net.minecraft.world.entity.decoration.ArmorStand $$0, float $$1, float $$2, float $$3) {
         this.basePlate.xRot = 0.0f;
-        this.basePlate.yRot = (float)java.lang.Math.PI / 180 * -net.minecraft.util.Mth.rotLerp((float)partialTick, (float)entity.yRotO, (float)entity.getYRot());
+        this.basePlate.yRot = (float)java.lang.Math.PI / 180 * -net.minecraft.util.Mth.rotLerp((float)$$3, (float)$$0.yRotO, (float)$$0.getYRot());
         this.basePlate.zRot = 0.0f;
     }
 
-    public void setupAnim(net.minecraft.world.entity.decoration.ArmorStand entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        this.leftArm.visible = entity.isShowArms();
-        this.rightArm.visible = entity.isShowArms();
-        this.basePlate.visible = !entity.isNoBasePlate();
-        this.rightBodyStick.xRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getX();
-        this.rightBodyStick.yRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getY();
-        this.rightBodyStick.zRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getZ();
-        this.leftBodyStick.xRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getX();
-        this.leftBodyStick.yRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getY();
-        this.leftBodyStick.zRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getZ();
-        this.shoulderStick.xRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getX();
-        this.shoulderStick.yRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getY();
-        this.shoulderStick.zRot = (float)java.lang.Math.PI / 180 * entity.getBodyPose().getZ();
+    public void setupAnim(net.minecraft.world.entity.decoration.ArmorStand $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
+        this.leftArm.visible = $$0.isShowArms();
+        this.rightArm.visible = $$0.isShowArms();
+        this.basePlate.visible = !$$0.isNoBasePlate();
+        this.rightBodyStick.xRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getX();
+        this.rightBodyStick.yRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getY();
+        this.rightBodyStick.zRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getZ();
+        this.leftBodyStick.xRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getX();
+        this.leftBodyStick.yRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getY();
+        this.leftBodyStick.zRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getZ();
+        this.shoulderStick.xRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getX();
+        this.shoulderStick.yRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getY();
+        this.shoulderStick.zRot = (float)java.lang.Math.PI / 180 * $$0.getBodyPose().getZ();
     }
 
     public java.lang.Iterable<net.minecraft.client.model.geom.ModelPart> bodyParts() {
         return com.google.common.collect.Iterables.concat((java.lang.Iterable)super.bodyParts(), (java.lang.Iterable)com.google.common.collect.ImmutableList.of((java.lang.Object)this.rightBodyStick, (java.lang.Object)this.leftBodyStick, (java.lang.Object)this.shoulderStick, (java.lang.Object)this.basePlate));
     }
 
-    public void translateToHand(net.minecraft.world.entity.HumanoidArm side, com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        net.minecraft.client.model.geom.ModelPart modelPart = this.getArm(side);
-        boolean bl = modelPart.visible;
-        modelPart.visible = true;
-        super.translateToHand(side, matrixStack);
-        modelPart.visible = bl;
+    public void translateToHand(net.minecraft.world.entity.HumanoidArm $$0, com.mojang.blaze3d.vertex.PoseStack $$1) {
+        net.minecraft.client.model.geom.ModelPart $$2 = this.getArm($$0);
+        boolean $$3 = $$2.visible;
+        $$2.visible = true;
+        super.translateToHand($$0, $$1);
+        $$2.visible = $$3;
     }
 }
 