--- net/minecraft/client/model/BatModel.java
+++ net/minecraft/client/model/BatModel.java
@@ -42,10 +42,10 @@
         return this.root;
     }
 
-    public void setupAnim(net.minecraft.world.entity.ambient.Bat entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (entity.isResting()) {
-            this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-            this.head.yRot = (float)java.lang.Math.PI - netHeadYaw * ((float)java.lang.Math.PI / 180);
+    public void setupAnim(net.minecraft.world.entity.ambient.Bat $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        if ($$0.isResting()) {
+            this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+            this.head.yRot = (float)java.lang.Math.PI - $$4 * ((float)java.lang.Math.PI / 180);
             this.head.zRot = (float)java.lang.Math.PI;
             this.head.setPos(0.0f, -2.0f, 0.0f);
             this.rightWing.setPos(-3.0f, 0.0f, 3.0f);
@@ -58,15 +58,15 @@
             this.leftWing.yRot = -this.rightWing.yRot;
             this.leftWingTip.yRot = -this.rightWingTip.yRot;
         } else {
-            this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-            this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
+            this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+            this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
             this.head.zRot = 0.0f;
             this.head.setPos(0.0f, 0.0f, 0.0f);
             this.rightWing.setPos(0.0f, 0.0f, 0.0f);
             this.leftWing.setPos(0.0f, 0.0f, 0.0f);
-            this.body.xRot = 0.7853982f + net.minecraft.util.Mth.cos((float)(ageInTicks * 0.1f)) * 0.15f;
+            this.body.xRot = 0.7853982f + net.minecraft.util.Mth.cos((float)($$3 * 0.1f)) * 0.15f;
             this.body.yRot = 0.0f;
-            this.rightWing.yRot = net.minecraft.util.Mth.cos((float)(ageInTicks * 74.48451f * ((float)java.lang.Math.PI / 180))) * (float)java.lang.Math.PI * 0.25f;
+            this.rightWing.yRot = net.minecraft.util.Mth.cos((float)($$3 * 74.48451f * ((float)java.lang.Math.PI / 180))) * (float)java.lang.Math.PI * 0.25f;
             this.leftWing.yRot = -this.rightWing.yRot;
             this.rightWingTip.yRot = this.rightWing.yRot * 0.5f;
             this.leftWingTip.yRot = -this.rightWing.yRot * 0.5f;