--- net/minecraft/client/model/BeeModel.java
+++ net/minecraft/client/model/BeeModel.java
@@ -57,21 +57,20 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$1, (int)64, (int)64);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        super.prepareMobModel(entity, limbSwing, limbSwingAmount, partialTick);
-        this.rollAmount = entity.getRollAmount(partialTick);
-        this.stinger.visible = !entity.hasStung();
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        super.prepareMobModel($$0, $$1, $$2, $$3);
+        this.rollAmount = $$0.getRollAmount($$3);
+        this.stinger.visible = !$$0.hasStung();
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f;
-        boolean bl;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        boolean $$6;
         this.rightWing.xRot = 0.0f;
         this.leftAntenna.xRot = 0.0f;
         this.rightAntenna.xRot = 0.0f;
         this.bone.xRot = 0.0f;
-        boolean bl2 = bl = entity.isOnGround() && entity.getDeltaMovement().lengthSqr() < 1.0E-7;
-        if (bl) {
+        boolean bl = $$6 = $$0.isOnGround() && $$0.getDeltaMovement().lengthSqr() < 1.0E-7;
+        if ($$6) {
             this.rightWing.yRot = -0.2618f;
             this.rightWing.zRot = 0.0f;
             this.leftWing.xRot = 0.0f;
@@ -81,9 +80,9 @@
             this.midLeg.xRot = 0.0f;
             this.backLeg.xRot = 0.0f;
         } else {
-            f = ageInTicks * 120.32113f * ((float)java.lang.Math.PI / 180);
+            float $$7 = $$3 * 120.32113f * ((float)java.lang.Math.PI / 180);
             this.rightWing.yRot = 0.0f;
-            this.rightWing.zRot = net.minecraft.util.Mth.cos((float)f) * (float)java.lang.Math.PI * 0.15f;
+            this.rightWing.zRot = net.minecraft.util.Mth.cos((float)$$7) * (float)java.lang.Math.PI * 0.15f;
             this.leftWing.xRot = this.rightWing.xRot;
             this.leftWing.yRot = this.rightWing.yRot;
             this.leftWing.zRot = -this.rightWing.zRot;
@@ -94,18 +93,18 @@
             this.bone.yRot = 0.0f;
             this.bone.zRot = 0.0f;
         }
-        if (!entity.isAngry()) {
+        if (!$$0.isAngry()) {
             this.bone.xRot = 0.0f;
             this.bone.yRot = 0.0f;
             this.bone.zRot = 0.0f;
-            if (!bl) {
-                f = net.minecraft.util.Mth.cos((float)(ageInTicks * 0.18f));
-                this.bone.xRot = 0.1f + f * (float)java.lang.Math.PI * 0.025f;
-                this.leftAntenna.xRot = f * (float)java.lang.Math.PI * 0.03f;
-                this.rightAntenna.xRot = f * (float)java.lang.Math.PI * 0.03f;
-                this.frontLeg.xRot = -f * (float)java.lang.Math.PI * 0.1f + 0.3926991f;
-                this.backLeg.xRot = -f * (float)java.lang.Math.PI * 0.05f + 0.7853982f;
-                this.bone.y = 19.0f - net.minecraft.util.Mth.cos((float)(ageInTicks * 0.18f)) * 0.9f;
+            if (!$$6) {
+                float $$8 = net.minecraft.util.Mth.cos((float)($$3 * 0.18f));
+                this.bone.xRot = 0.1f + $$8 * (float)java.lang.Math.PI * 0.025f;
+                this.leftAntenna.xRot = $$8 * (float)java.lang.Math.PI * 0.03f;
+                this.rightAntenna.xRot = $$8 * (float)java.lang.Math.PI * 0.03f;
+                this.frontLeg.xRot = -$$8 * (float)java.lang.Math.PI * 0.1f + 0.3926991f;
+                this.backLeg.xRot = -$$8 * (float)java.lang.Math.PI * 0.05f + 0.7853982f;
+                this.bone.y = 19.0f - net.minecraft.util.Mth.cos((float)($$3 * 0.18f)) * 0.9f;
             }
         }
         if (this.rollAmount > 0.0f) {