--- net/minecraft/client/model/BlazeModel.java
+++ net/minecraft/client/model/BlazeModel.java
@@ -57,34 +57,30 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        int n;
-        float f = ageInTicks * (float)java.lang.Math.PI * -0.1f;
-        for (n = 0;
- n < 4; ++n) {
-            this.upperBodyParts[n].y = -2.0f + net.minecraft.util.Mth.cos((float)(((float)(n * 2) + ageInTicks) * 0.25f));
-            this.upperBodyParts[n].x = net.minecraft.util.Mth.cos((float)f) * 9.0f;
-            this.upperBodyParts[n].z = net.minecraft.util.Mth.sin((float)f) * 9.0f;
-            f += 1.5707964f;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        float $$6 = $$3 * (float)java.lang.Math.PI * -0.1f;
+        for (int $$7 = 0; $$7 < 4; ++$$7) {
+            this.upperBodyParts[$$7].y = -2.0f + net.minecraft.util.Mth.cos((float)(((float)($$7 * 2) + $$3) * 0.25f));
+            this.upperBodyParts[$$7].x = net.minecraft.util.Mth.cos((float)$$6) * 9.0f;
+            this.upperBodyParts[$$7].z = net.minecraft.util.Mth.sin((float)$$6) * 9.0f;
+            $$6 += 1.5707964f;
         }
-        f = 0.7853982f + ageInTicks * (float)java.lang.Math.PI * 0.03f;
-        for (n = 4;
- n < 8; ++n) {
-            this.upperBodyParts[n].y = 2.0f + net.minecraft.util.Mth.cos((float)(((float)(n * 2) + ageInTicks) * 0.25f));
-            this.upperBodyParts[n].x = net.minecraft.util.Mth.cos((float)f) * 7.0f;
-            this.upperBodyParts[n].z = net.minecraft.util.Mth.sin((float)f) * 7.0f;
-            f += 1.5707964f;
+        $$6 = 0.7853982f + $$3 * (float)java.lang.Math.PI * 0.03f;
+        for (int $$8 = 4; $$8 < 8; ++$$8) {
+            this.upperBodyParts[$$8].y = 2.0f + net.minecraft.util.Mth.cos((float)(((float)($$8 * 2) + $$3) * 0.25f));
+            this.upperBodyParts[$$8].x = net.minecraft.util.Mth.cos((float)$$6) * 7.0f;
+            this.upperBodyParts[$$8].z = net.minecraft.util.Mth.sin((float)$$6) * 7.0f;
+            $$6 += 1.5707964f;
         }
-        f = 0.47123894f + ageInTicks * (float)java.lang.Math.PI * -0.05f;
-        for (n = 8;
- n < 12; ++n) {
-            this.upperBodyParts[n].y = 11.0f + net.minecraft.util.Mth.cos((float)(((float)n * 1.5f + ageInTicks) * 0.5f));
-            this.upperBodyParts[n].x = net.minecraft.util.Mth.cos((float)f) * 5.0f;
-            this.upperBodyParts[n].z = net.minecraft.util.Mth.sin((float)f) * 5.0f;
-            f += 1.5707964f;
+        $$6 = 0.47123894f + $$3 * (float)java.lang.Math.PI * -0.05f;
+        for (int $$9 = 8; $$9 < 12; ++$$9) {
+            this.upperBodyParts[$$9].y = 11.0f + net.minecraft.util.Mth.cos((float)(((float)$$9 * 1.5f + $$3) * 0.5f));
+            this.upperBodyParts[$$9].x = net.minecraft.util.Mth.cos((float)$$6) * 5.0f;
+            this.upperBodyParts[$$9].z = net.minecraft.util.Mth.sin((float)$$6) * 5.0f;
+            $$6 += 1.5707964f;
         }
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
     }
 }
 