--- net/minecraft/client/model/BoatModel.java
+++ net/minecraft/client/model/BoatModel.java
@@ -5,6 +5,7 @@
 
 public class net.minecraft.client.model.BoatModel
 extends net.minecraft.client.model.ListModel<net.minecraft.world.entity.vehicle.Boat>
+implements net.minecraft.client.model.WaterPatchModel
  {
     public static java.lang.String LEFT_PADDLE = "left_paddle";
     public static java.lang.String RIGHT_PADDLE = "right_paddle";
@@ -14,61 +15,56 @@
     public static java.lang.String FRONT = "front";
     public static java.lang.String RIGHT = "right";
     public static java.lang.String LEFT = "left";
-    public static java.lang.String CHEST_BOTTOM = "chest_bottom";
-    public static java.lang.String CHEST_LID = "chest_lid";
-    public static java.lang.String CHEST_LOCK = "chest_lock";
     public net.minecraft.client.model.geom.ModelPart leftPaddle;
     public net.minecraft.client.model.geom.ModelPart rightPaddle;
     public net.minecraft.client.model.geom.ModelPart waterPatch;
     public com.google.common.collect.ImmutableList<net.minecraft.client.model.geom.ModelPart> parts;
 
-    public net.minecraft.client.model.BoatModel(net.minecraft.client.model.geom.ModelPart $$0, boolean $$1) {
+    public net.minecraft.client.model.BoatModel(net.minecraft.client.model.geom.ModelPart $$0) {
         this.leftPaddle = $$0.getChild("left_paddle");
         this.rightPaddle = $$0.getChild("right_paddle");
         this.waterPatch = $$0.getChild("water_patch");
-        com.google.common.collect.ImmutableList$Builder $$2 = new com.google.common.collect.ImmutableList$Builder();
-        $$2.add((java.lang.Object[])new net.minecraft.client.model.geom.ModelPart[]{$$0.getChild("bottom"), $$0.getChild("back"), $$0.getChild("front"), $$0.getChild("right"), $$0.getChild("left"), this.leftPaddle, this.rightPaddle});
-        if ($$1) {
-            $$2.add((java.lang.Object)$$0.getChild("chest_bottom"));
-            $$2.add((java.lang.Object)$$0.getChild("chest_lid"));
-            $$2.add((java.lang.Object)$$0.getChild("chest_lock"));
-        }
-        this.parts = $$2.build();
+        this.parts = this.createPartsBuilder($$0).build();
     }
 
-    public static net.minecraft.client.model.geom.builders.LayerDefinition createBodyModel(boolean $$0) {
-        net.minecraft.client.model.geom.builders.MeshDefinition $$1 = new net.minecraft.client.model.geom.builders.MeshDefinition();
-        net.minecraft.client.model.geom.builders.PartDefinition $$2 = $$1.getRoot();
-        int $$3 = 32;
-        int $$4 = 6;
-        int $$5 = 20;
-        int $$6 = 4;
-        int $$7 = 28;
-        $$2.addOrReplaceChild("bottom", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 0).addBox(-14.0f, -9.0f, -3.0f, 28.0f, 16.0f, 3.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)0.0f, (float)3.0f, (float)1.0f, (float)1.5707964f, (float)0.0f, (float)0.0f));
-        $$2.addOrReplaceChild("back", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 19).addBox(-13.0f, -7.0f, -1.0f, 18.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-15.0f, (float)4.0f, (float)4.0f, (float)0.0f, (float)4.712389f, (float)0.0f));
-        $$2.addOrReplaceChild("front", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 27).addBox(-8.0f, -7.0f, -1.0f, 16.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)15.0f, (float)4.0f, (float)0.0f, (float)0.0f, (float)1.5707964f, (float)0.0f));
-        $$2.addOrReplaceChild("right", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 35).addBox(-14.0f, -7.0f, -1.0f, 28.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)0.0f, (float)4.0f, (float)-9.0f, (float)0.0f, (float)((float)java.lang.Math.PI), (float)0.0f));
-        $$2.addOrReplaceChild("left", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 43).addBox(-14.0f, -7.0f, -1.0f, 28.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)4.0f, (float)9.0f));
-        if ($$0) {
-            $$2.addOrReplaceChild("chest_bottom", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 76).addBox(0.0f, 0.0f, 0.0f, 12.0f, 8.0f, 12.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-2.0f, (float)-5.0f, (float)-6.0f, (float)0.0f, (float)-1.5707964f, (float)0.0f));
-            $$2.addOrReplaceChild("chest_lid", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 59).addBox(0.0f, 0.0f, 0.0f, 12.0f, 4.0f, 12.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-2.0f, (float)-9.0f, (float)-6.0f, (float)0.0f, (float)-1.5707964f, (float)0.0f));
-            $$2.addOrReplaceChild("chest_lock", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 59).addBox(0.0f, 0.0f, 0.0f, 2.0f, 4.0f, 1.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-1.0f, (float)-6.0f, (float)-1.0f, (float)0.0f, (float)-1.5707964f, (float)0.0f));
-        }
-        int $$8 = 20;
-        int $$9 = 7;
-        int $$10 = 6;
-        float $$11 = -5.0f;
-        $$2.addOrReplaceChild("left_paddle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(62, 0).addBox(-1.0f, 0.0f, -5.0f, 2.0f, 2.0f, 18.0f).addBox(-1.001f, -3.0f, 8.0f, 1.0f, 6.0f, 7.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)3.0f, (float)-5.0f, (float)9.0f, (float)0.0f, (float)0.0f, (float)0.19634955f));
-        $$2.addOrReplaceChild("right_paddle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(62, 20).addBox(-1.0f, 0.0f, -5.0f, 2.0f, 2.0f, 18.0f).addBox(0.001f, -3.0f, 8.0f, 1.0f, 6.0f, 7.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)3.0f, (float)-5.0f, (float)-9.0f, (float)0.0f, (float)((float)java.lang.Math.PI), (float)0.19634955f));
-        $$2.addOrReplaceChild("water_patch", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 0).addBox(-14.0f, -9.0f, -3.0f, 28.0f, 16.0f, 3.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)0.0f, (float)-3.0f, (float)1.0f, (float)1.5707964f, (float)0.0f, (float)0.0f));
-        return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$1, (int)128, (int)($$0 ? 128 : 64));
+    public com.google.common.collect.ImmutableList$Builder<net.minecraft.client.model.geom.ModelPart> createPartsBuilder(net.minecraft.client.model.geom.ModelPart $$0) {
+        com.google.common.collect.ImmutableList$Builder $$1 = new com.google.common.collect.ImmutableList$Builder();
+        $$1.add((java.lang.Object[])new net.minecraft.client.model.geom.ModelPart[]{$$0.getChild("bottom"), $$0.getChild("back"), $$0.getChild("front"), $$0.getChild("right"), $$0.getChild("left"), this.leftPaddle, this.rightPaddle});
+        return $$1;
     }
 
-    public void setupAnim(net.minecraft.world.entity.vehicle.Boat entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        net.minecraft.client.model.BoatModel.animatePaddle((net.minecraft.world.entity.vehicle.Boat)entity, (int)0, (net.minecraft.client.model.geom.ModelPart)this.leftPaddle, (float)limbSwing);
-        net.minecraft.client.model.BoatModel.animatePaddle((net.minecraft.world.entity.vehicle.Boat)entity, (int)1, (net.minecraft.client.model.geom.ModelPart)this.rightPaddle, (float)limbSwing);
+    public static void createChildren(net.minecraft.client.model.geom.builders.PartDefinition $$0) {
+        int $$1 = 32;
+        int $$2 = 6;
+        int $$3 = 20;
+        int $$4 = 4;
+        int $$5 = 28;
+        $$0.addOrReplaceChild("bottom", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 0).addBox(-14.0f, -9.0f, -3.0f, 28.0f, 16.0f, 3.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)0.0f, (float)3.0f, (float)1.0f, (float)1.5707964f, (float)0.0f, (float)0.0f));
+        $$0.addOrReplaceChild("back", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 19).addBox(-13.0f, -7.0f, -1.0f, 18.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-15.0f, (float)4.0f, (float)4.0f, (float)0.0f, (float)4.712389f, (float)0.0f));
+        $$0.addOrReplaceChild("front", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 27).addBox(-8.0f, -7.0f, -1.0f, 16.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)15.0f, (float)4.0f, (float)0.0f, (float)0.0f, (float)1.5707964f, (float)0.0f));
+        $$0.addOrReplaceChild("right", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 35).addBox(-14.0f, -7.0f, -1.0f, 28.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)0.0f, (float)4.0f, (float)-9.0f, (float)0.0f, (float)((float)java.lang.Math.PI), (float)0.0f));
+        $$0.addOrReplaceChild("left", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 43).addBox(-14.0f, -7.0f, -1.0f, 28.0f, 6.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)4.0f, (float)9.0f));
+        int $$6 = 20;
+        int $$7 = 7;
+        int $$8 = 6;
+        float $$9 = -5.0f;
+        $$0.addOrReplaceChild("left_paddle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(62, 0).addBox(-1.0f, 0.0f, -5.0f, 2.0f, 2.0f, 18.0f).addBox(-1.001f, -3.0f, 8.0f, 1.0f, 6.0f, 7.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)3.0f, (float)-5.0f, (float)9.0f, (float)0.0f, (float)0.0f, (float)0.19634955f));
+        $$0.addOrReplaceChild("right_paddle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(62, 20).addBox(-1.0f, 0.0f, -5.0f, 2.0f, 2.0f, 18.0f).addBox(0.001f, -3.0f, 8.0f, 1.0f, 6.0f, 7.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)3.0f, (float)-5.0f, (float)-9.0f, (float)0.0f, (float)((float)java.lang.Math.PI), (float)0.19634955f));
+        $$0.addOrReplaceChild("water_patch", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 0).addBox(-14.0f, -9.0f, -3.0f, 28.0f, 16.0f, 3.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)0.0f, (float)-3.0f, (float)1.0f, (float)1.5707964f, (float)0.0f, (float)0.0f));
     }
 
+    public static net.minecraft.client.model.geom.builders.LayerDefinition createBodyModel() {
+        net.minecraft.client.model.geom.builders.MeshDefinition $$0 = new net.minecraft.client.model.geom.builders.MeshDefinition();
+        net.minecraft.client.model.geom.builders.PartDefinition $$1 = $$0.getRoot();
+        net.minecraft.client.model.BoatModel.createChildren((net.minecraft.client.model.geom.builders.PartDefinition)$$1);
+        return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)128, (int)64);
+    }
+
+    public void setupAnim(net.minecraft.world.entity.vehicle.Boat $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        net.minecraft.client.model.BoatModel.animatePaddle((net.minecraft.world.entity.vehicle.Boat)$$0, (int)0, (net.minecraft.client.model.geom.ModelPart)this.leftPaddle, (float)$$1);
+        net.minecraft.client.model.BoatModel.animatePaddle((net.minecraft.world.entity.vehicle.Boat)$$0, (int)1, (net.minecraft.client.model.geom.ModelPart)this.rightPaddle, (float)$$1);
+    }
+
     public com.google.common.collect.ImmutableList<net.minecraft.client.model.geom.ModelPart> parts() {
         return this.parts;
     }