--- net/minecraft/client/model/CamelModel.java
+++ net/minecraft/client/model/CamelModel.java
@@ -1,0 +1,100 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.model;
+
+public class net.minecraft.client.model.CamelModel<T extends net.minecraft.world.entity.animal.camel.Camel>
+extends net.minecraft.client.model.HierarchicalModel<T>
+ {
+    public static float WALK_ANIMATION_SPEED_FACTOR = 75.0f;
+    public static float MAX_WALK_ANIMATION_SPEED = 2.0f;
+    public static java.lang.String SADDLE = "saddle";
+    public static java.lang.String BRIDLE = "bridle";
+    public static java.lang.String REINS = "reins";
+    public net.minecraft.client.model.geom.ModelPart root;
+    public net.minecraft.client.model.geom.ModelPart head;
+    public net.minecraft.client.model.geom.ModelPart[] saddleParts;
+    public net.minecraft.client.model.geom.ModelPart[] ridingParts;
+
+    public net.minecraft.client.model.CamelModel(net.minecraft.client.model.geom.ModelPart $$0) {
+        this.root = $$0;
+        net.minecraft.client.model.geom.ModelPart $$1 = $$0.getChild("body");
+        this.head = $$1.getChild("head");
+        this.saddleParts = new net.minecraft.client.model.geom.ModelPart[]{$$1.getChild("saddle"), this.head.getChild("bridle")};
+        this.ridingParts = new net.minecraft.client.model.geom.ModelPart[]{this.head.getChild("reins")};
+    }
+
+    public static net.minecraft.client.model.geom.builders.LayerDefinition createBodyLayer() {
+        net.minecraft.client.model.geom.builders.MeshDefinition $$0 = new net.minecraft.client.model.geom.builders.MeshDefinition();
+        net.minecraft.client.model.geom.builders.PartDefinition $$1 = $$0.getRoot();
+        net.minecraft.client.model.geom.builders.CubeDeformation $$2 = new net.minecraft.client.model.geom.builders.CubeDeformation(0.1f);
+        net.minecraft.client.model.geom.builders.PartDefinition $$3 = $$1.addOrReplaceChild("body", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 25).addBox(-7.5f, -12.0f, -23.5f, 15.0f, 12.0f, 27.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)4.0f, (float)9.5f));
+        $$3.addOrReplaceChild("hump", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(74, 0).addBox(-4.5f, -5.0f, -5.5f, 9.0f, 5.0f, 11.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)-12.0f, (float)-10.0f));
+        $$3.addOrReplaceChild("tail", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(122, 0).addBox(-1.5f, 0.0f, 0.0f, 3.0f, 14.0f, 0.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)-9.0f, (float)3.5f));
+        net.minecraft.client.model.geom.builders.PartDefinition $$4 = $$3.addOrReplaceChild("head", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(60, 24).addBox(-3.5f, -7.0f, -15.0f, 7.0f, 8.0f, 19.0f).texOffs(21, 0).addBox(-3.5f, -21.0f, -15.0f, 7.0f, 14.0f, 7.0f).texOffs(50, 0).addBox(-2.5f, -21.0f, -21.0f, 5.0f, 5.0f, 6.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)-3.0f, (float)-19.5f));
+        $$4.addOrReplaceChild("left_ear", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(45, 0).addBox(-0.5f, 0.5f, -1.0f, 3.0f, 1.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offset((float)3.0f, (float)-21.0f, (float)-9.5f));
+        $$4.addOrReplaceChild("right_ear", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(67, 0).addBox(-2.5f, 0.5f, -1.0f, 3.0f, 1.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offset((float)-3.0f, (float)-21.0f, (float)-9.5f));
+        $$1.addOrReplaceChild("left_hind_leg", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(58, 16).addBox(-2.5f, 2.0f, -2.5f, 5.0f, 21.0f, 5.0f), net.minecraft.client.model.geom.PartPose.offset((float)4.9f, (float)1.0f, (float)9.5f));
+        $$1.addOrReplaceChild("right_hind_leg", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(94, 16).addBox(-2.5f, 2.0f, -2.5f, 5.0f, 21.0f, 5.0f), net.minecraft.client.model.geom.PartPose.offset((float)-4.9f, (float)1.0f, (float)9.5f));
+        $$1.addOrReplaceChild("left_front_leg", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 0).addBox(-2.5f, 2.0f, -2.5f, 5.0f, 21.0f, 5.0f), net.minecraft.client.model.geom.PartPose.offset((float)4.9f, (float)1.0f, (float)-10.5f));
+        $$1.addOrReplaceChild("right_front_leg", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 26).addBox(-2.5f, 2.0f, -2.5f, 5.0f, 21.0f, 5.0f), net.minecraft.client.model.geom.PartPose.offset((float)-4.9f, (float)1.0f, (float)-10.5f));
+        $$3.addOrReplaceChild("saddle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(74, 64).addBox(-4.5f, -17.0f, -15.5f, 9.0f, 5.0f, 11.0f, $$2).texOffs(92, 114).addBox(-3.5f, -20.0f, -15.5f, 7.0f, 3.0f, 11.0f, $$2).texOffs(0, 89).addBox(-7.5f, -12.0f, -23.5f, 15.0f, 12.0f, 27.0f, $$2), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)0.0f, (float)0.0f));
+        $$4.addOrReplaceChild("reins", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(98, 42).addBox(3.51f, -18.0f, -17.0f, 0.0f, 7.0f, 15.0f).texOffs(84, 57).addBox(-3.5f, -18.0f, -2.0f, 7.0f, 7.0f, 0.0f).texOffs(98, 42).addBox(-3.51f, -18.0f, -17.0f, 0.0f, 7.0f, 15.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)0.0f, (float)0.0f));
+        $$4.addOrReplaceChild("bridle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(60, 87).addBox(-3.5f, -7.0f, -15.0f, 7.0f, 8.0f, 19.0f, $$2).texOffs(21, 64).addBox(-3.5f, -21.0f, -15.0f, 7.0f, 14.0f, 7.0f, $$2).texOffs(50, 64).addBox(-2.5f, -21.0f, -21.0f, 5.0f, 5.0f, 6.0f, $$2).texOffs(74, 70).addBox(2.5f, -19.0f, -18.0f, 1.0f, 2.0f, 2.0f).texOffs(74, 70).mirror().addBox(-3.5f, -19.0f, -18.0f, 1.0f, 2.0f, 2.0f), net.minecraft.client.model.geom.PartPose.offset((float)0.0f, (float)0.0f, (float)0.0f));
+        return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)128, (int)128);
+    }
+
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.root().getAllParts().forEach(net.minecraft.client.model.geom.ModelPart::resetPose);
+        this.applyHeadRotation($$0, $$4, $$5, $$3);
+        this.toggleInvisibleParts($$0);
+        float $$6 = java.lang.Math.min((float)((float)$$0.getDeltaMovement().lengthSqr() * 75.0f), (float)2.0f);
+        this.animate(((net.minecraft.world.entity.animal.camel.Camel)$$0).walkAnimationState, net.minecraft.client.animation.definitions.CamelAnimation.CAMEL_WALK, $$3, $$6);
+        this.animate(((net.minecraft.world.entity.animal.camel.Camel)$$0).sitAnimationState, net.minecraft.client.animation.definitions.CamelAnimation.CAMEL_SIT, $$3, 1.0f);
+        this.animate(((net.minecraft.world.entity.animal.camel.Camel)$$0).sitUpAnimationState, net.minecraft.client.animation.definitions.CamelAnimation.CAMEL_STANDUP, $$3, 1.0f);
+        this.animate(((net.minecraft.world.entity.animal.camel.Camel)$$0).idleAnimationState, net.minecraft.client.animation.definitions.CamelAnimation.CAMEL_IDLE, $$3, 1.0f);
+        this.animate(((net.minecraft.world.entity.animal.camel.Camel)$$0).dashAnimationState, net.minecraft.client.animation.definitions.CamelAnimation.CAMEL_DASH, $$3, 1.0f);
+    }
+
+    public void applyHeadRotation(T $$0, float $$1, float $$2, float $$3) {
+        $$1 = net.minecraft.util.Mth.clamp((float)$$1, (float)-30.0f, (float)30.0f);
+        $$2 = net.minecraft.util.Mth.clamp((float)$$2, (float)-25.0f, (float)45.0f);
+        if ($$0.getJumpCooldown() > 0) {
+            float $$4 = $$3 - (float)((net.minecraft.world.entity.animal.camel.Camel)$$0).tickCount;
+            float $$5 = 45.0f * ((float)$$0.getJumpCooldown() - $$4) / 55.0f;
+            $$2 = net.minecraft.util.Mth.clamp((float)($$2 + $$5), (float)-25.0f, (float)70.0f);
+        }
+        this.head.yRot = $$1 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$2 * ((float)java.lang.Math.PI / 180);
+    }
+
+    public void toggleInvisibleParts(T $$0) {
+        boolean $$1 = $$0.isSaddled();
+        boolean $$2 = $$0.isVehicle();
+        for (net.minecraft.client.model.geom.ModelPart $$3 : this.saddleParts) {
+            $$3.visible = $$1;
+        }
+        for (net.minecraft.client.model.geom.ModelPart $$4 : this.ridingParts) {
+            $$4.visible = $$2 && $$1;
+        }
+    }
+
+    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7) {
+        if (this.young) {
+            float $$8 = 2.0f;
+            float $$9 = 1.1f;
+            $$0.pushPose();
+            $$0.scale(0.45454544f, 0.41322312f, 0.45454544f);
+            $$0.translate(0.0, 2.0625, 0.0);
+            this.root().render($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7);
+            $$0.popPose();
+        } else {
+            this.root().render($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7);
+        }
+    }
+
+    public net.minecraft.client.model.geom.ModelPart root() {
+        return this.root;
+    }
+}
+