--- net/minecraft/client/model/CatModel.java
+++ net/minecraft/client/model/CatModel.java
@@ -14,10 +14,10 @@
         super($$0);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        this.lieDownAmount = entity.getLieDownAmount(partialTick);
-        this.lieDownAmountTail = entity.getLieDownAmountTail(partialTick);
-        this.relaxStateOneAmount = entity.getRelaxStateOneAmount(partialTick);
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        this.lieDownAmount = $$0.getLieDownAmount($$3);
+        this.lieDownAmountTail = $$0.getLieDownAmountTail($$3);
+        this.relaxStateOneAmount = $$0.getRelaxStateOneAmount($$3);
         if (this.lieDownAmount <= 0.0f) {
             this.head.xRot = 0.0f;
             this.head.zRot = 0.0f;
@@ -32,8 +32,8 @@
             this.rightHindLeg.x = -1.1f;
             this.rightHindLeg.y = 18.0f;
         }
-        super.prepareMobModel(entity, limbSwing, limbSwingAmount, partialTick);
-        if (entity.isInSittingPose()) {
+        super.prepareMobModel($$0, $$1, $$2, $$3);
+        if ($$0.isInSittingPose()) {
             this.body.xRot = 0.7853982f;
             this.body.y += -4.0f;
             this.body.z += 5.0f;
@@ -61,8 +61,8 @@
         }
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
         if (this.lieDownAmount > 0.0f) {
             this.head.zRot = net.minecraft.client.model.ModelUtils.rotlerpRad((float)this.head.zRot, (float)-1.2707963f, (float)this.lieDownAmount);
             this.head.yRot = net.minecraft.client.model.ModelUtils.rotlerpRad((float)this.head.yRot, (float)1.2707963f, (float)this.lieDownAmount);