--- net/minecraft/client/model/ChestBoatModel.java
+++ net/minecraft/client/model/ChestBoatModel.java
@@ -1,0 +1,35 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.model;
+
+public class net.minecraft.client.model.ChestBoatModel
+extends net.minecraft.client.model.BoatModel
+ {
+    public static java.lang.String CHEST_BOTTOM = "chest_bottom";
+    public static java.lang.String CHEST_LID = "chest_lid";
+    public static java.lang.String CHEST_LOCK = "chest_lock";
+
+    public net.minecraft.client.model.ChestBoatModel(net.minecraft.client.model.geom.ModelPart $$0) {
+        super($$0);
+    }
+
+    public com.google.common.collect.ImmutableList$Builder<net.minecraft.client.model.geom.ModelPart> createPartsBuilder(net.minecraft.client.model.geom.ModelPart $$0) {
+        com.google.common.collect.ImmutableList$Builder $$1 = super.createPartsBuilder($$0);
+        $$1.add((java.lang.Object)$$0.getChild("chest_bottom"));
+        $$1.add((java.lang.Object)$$0.getChild("chest_lid"));
+        $$1.add((java.lang.Object)$$0.getChild("chest_lock"));
+        return $$1;
+    }
+
+    public static net.minecraft.client.model.geom.builders.LayerDefinition createBodyModel() {
+        net.minecraft.client.model.geom.builders.MeshDefinition $$0 = new net.minecraft.client.model.geom.builders.MeshDefinition();
+        net.minecraft.client.model.geom.builders.PartDefinition $$1 = $$0.getRoot();
+        net.minecraft.client.model.BoatModel.createChildren((net.minecraft.client.model.geom.builders.PartDefinition)$$1);
+        $$1.addOrReplaceChild("chest_bottom", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 76).addBox(0.0f, 0.0f, 0.0f, 12.0f, 8.0f, 12.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-2.0f, (float)-5.0f, (float)-6.0f, (float)0.0f, (float)-1.5707964f, (float)0.0f));
+        $$1.addOrReplaceChild("chest_lid", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 59).addBox(0.0f, 0.0f, 0.0f, 12.0f, 4.0f, 12.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-2.0f, (float)-9.0f, (float)-6.0f, (float)0.0f, (float)-1.5707964f, (float)0.0f));
+        $$1.addOrReplaceChild("chest_lock", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 59).addBox(0.0f, 0.0f, 0.0f, 2.0f, 4.0f, 1.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)-1.0f, (float)-6.0f, (float)-1.0f, (float)0.0f, (float)-1.5707964f, (float)0.0f));
+        return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)128, (int)128);
+    }
+}
+