--- net/minecraft/client/model/ChickenModel.java
+++ net/minecraft/client/model/ChickenModel.java
@@ -51,17 +51,17 @@
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.body, (java.lang.Object)this.rightLeg, (java.lang.Object)this.leftLeg, (java.lang.Object)this.rightWing, (java.lang.Object)this.leftWing);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
         this.beak.xRot = this.head.xRot;
         this.beak.yRot = this.head.yRot;
         this.redThing.xRot = this.head.xRot;
         this.redThing.yRot = this.head.yRot;
-        this.rightLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * 1.4f * limbSwingAmount;
-        this.leftLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * limbSwingAmount;
-        this.rightWing.zRot = ageInTicks;
-        this.leftWing.zRot = -ageInTicks;
+        this.rightLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * 1.4f * $$2;
+        this.leftLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * $$2;
+        this.rightWing.zRot = $$3;
+        this.leftWing.zRot = -$$3;
     }
 }
 