--- net/minecraft/client/model/DolphinModel.java
+++ net/minecraft/client/model/DolphinModel.java
@@ -38,13 +38,13 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.body.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        this.body.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        if (entity.getDeltaMovement().horizontalDistanceSqr() > 1.0E-7) {
-            this.body.xRot += -0.05f - 0.05f * net.minecraft.util.Mth.cos((float)(ageInTicks * 0.3f));
-            this.tail.xRot = -0.1f * net.minecraft.util.Mth.cos((float)(ageInTicks * 0.3f));
-            this.tailFin.xRot = -0.2f * net.minecraft.util.Mth.cos((float)(ageInTicks * 0.3f));
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.body.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        this.body.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        if ($$0.getDeltaMovement().horizontalDistanceSqr() > 1.0E-7) {
+            this.body.xRot += -0.05f - 0.05f * net.minecraft.util.Mth.cos((float)($$3 * 0.3f));
+            this.tail.xRot = -0.1f * net.minecraft.util.Mth.cos((float)($$3 * 0.3f));
+            this.tailFin.xRot = -0.2f * net.minecraft.util.Mth.cos((float)($$3 * 0.3f));
         }
     }
 }