--- net/minecraft/client/model/DrownedModel.java
+++ net/minecraft/client/model/DrownedModel.java
@@ -18,22 +18,22 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$1, (int)64, (int)64);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
         this.rightArmPose = net.minecraft.client.model.HumanoidModel$ArmPose.EMPTY;
         this.leftArmPose = net.minecraft.client.model.HumanoidModel$ArmPose.EMPTY;
-        net.minecraft.world.item.ItemStack itemStack = entity.getItemInHand(net.minecraft.world.InteractionHand.MAIN_HAND);
-        if (itemStack.is(net.minecraft.world.item.Items.TRIDENT) && entity.isAggressive()) {
-            if (entity.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT) {
+        net.minecraft.world.item.ItemStack $$4 = $$0.getItemInHand(net.minecraft.world.InteractionHand.MAIN_HAND);
+        if ($$4.is(net.minecraft.world.item.Items.TRIDENT) && $$0.isAggressive()) {
+            if ($$0.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT) {
                 this.rightArmPose = net.minecraft.client.model.HumanoidModel$ArmPose.THROW_SPEAR;
             } else {
                 this.leftArmPose = net.minecraft.client.model.HumanoidModel$ArmPose.THROW_SPEAR;
             }
         }
-        super.prepareMobModel(entity, limbSwing, limbSwingAmount, partialTick);
+        super.prepareMobModel($$0, $$1, $$2, $$3);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
         if (this.leftArmPose == net.minecraft.client.model.HumanoidModel$ArmPose.THROW_SPEAR) {
             this.leftArm.xRot = this.leftArm.xRot * 0.5f - (float)java.lang.Math.PI;
             this.leftArm.yRot = 0.0f;
@@ -43,12 +43,12 @@
             this.rightArm.yRot = 0.0f;
         }
         if (this.swimAmount > 0.0f) {
-            this.rightArm.xRot = this.rotlerpRad(this.swimAmount, this.rightArm.xRot, -2.5132742f) + this.swimAmount * 0.35f * net.minecraft.util.Mth.sin((float)(0.1f * ageInTicks));
-            this.leftArm.xRot = this.rotlerpRad(this.swimAmount, this.leftArm.xRot, -2.5132742f) - this.swimAmount * 0.35f * net.minecraft.util.Mth.sin((float)(0.1f * ageInTicks));
+            this.rightArm.xRot = this.rotlerpRad(this.swimAmount, this.rightArm.xRot, -2.5132742f) + this.swimAmount * 0.35f * net.minecraft.util.Mth.sin((float)(0.1f * $$3));
+            this.leftArm.xRot = this.rotlerpRad(this.swimAmount, this.leftArm.xRot, -2.5132742f) - this.swimAmount * 0.35f * net.minecraft.util.Mth.sin((float)(0.1f * $$3));
             this.rightArm.zRot = this.rotlerpRad(this.swimAmount, this.rightArm.zRot, -0.15f);
             this.leftArm.zRot = this.rotlerpRad(this.swimAmount, this.leftArm.zRot, 0.15f);
-            this.leftLeg.xRot -= this.swimAmount * 0.55f * net.minecraft.util.Mth.sin((float)(0.1f * ageInTicks));
-            this.rightLeg.xRot += this.swimAmount * 0.55f * net.minecraft.util.Mth.sin((float)(0.1f * ageInTicks));
+            this.leftLeg.xRot -= this.swimAmount * 0.55f * net.minecraft.util.Mth.sin((float)(0.1f * $$3));
+            this.rightLeg.xRot += this.swimAmount * 0.55f * net.minecraft.util.Mth.sin((float)(0.1f * $$3));
             this.head.xRot = 0.0f;
         }
     }