--- net/minecraft/client/model/ElytraModel.java
+++ net/minecraft/client/model/ElytraModel.java
@@ -31,39 +31,39 @@
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.leftWing, (java.lang.Object)this.rightWing);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f = 0.2617994f;
-        float f2 = -0.2617994f;
-        float f3 = 0.0f;
-        float f4 = 0.0f;
-        if (entity.isFallFlying()) {
-            float f5 = 1.0f;
-            net.minecraft.world.phys.Vec3 vec3 = entity.getDeltaMovement();
-            if (vec3.y < 0.0) {
-                net.minecraft.world.phys.Vec3 vec32 = vec3.normalize();
-                f5 = 1.0f - (float)java.lang.Math.pow((double)(-vec32.y), (double)1.5);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        float $$6 = 0.2617994f;
+        float $$7 = -0.2617994f;
+        float $$8 = 0.0f;
+        float $$9 = 0.0f;
+        if ($$0.isFallFlying()) {
+            float $$10 = 1.0f;
+            net.minecraft.world.phys.Vec3 $$11 = $$0.getDeltaMovement();
+            if ($$11.y < 0.0) {
+                net.minecraft.world.phys.Vec3 $$12 = $$11.normalize();
+                $$10 = 1.0f - (float)java.lang.Math.pow((double)(-$$12.y), (double)1.5);
             }
-            f = f5 * 0.34906584f + (1.0f - f5) * f;
-            f2 = f5 * -1.5707964f + (1.0f - f5) * f2;
-        } else if (entity.isCrouching()) {
-            f = 0.6981317f;
-            f2 = -0.7853982f;
-            f3 = 3.0f;
-            f4 = 0.08726646f;
+            $$6 = $$10 * 0.34906584f + (1.0f - $$10) * $$6;
+            $$7 = $$10 * -1.5707964f + (1.0f - $$10) * $$7;
+        } else if ($$0.isCrouching()) {
+            $$6 = 0.6981317f;
+            $$7 = -0.7853982f;
+            $$8 = 3.0f;
+            $$9 = 0.08726646f;
         }
-        this.leftWing.y = f3;
-        if (entity instanceof net.minecraft.client.player.AbstractClientPlayer) {
-            net.minecraft.client.player.AbstractClientPlayer abstractClientPlayer = (net.minecraft.client.player.AbstractClientPlayer)entity;
-            abstractClientPlayer.elytraRotX += (f - abstractClientPlayer.elytraRotX) * 0.1f;
-            abstractClientPlayer.elytraRotY += (f4 - abstractClientPlayer.elytraRotY) * 0.1f;
-            abstractClientPlayer.elytraRotZ += (f2 - abstractClientPlayer.elytraRotZ) * 0.1f;
-            this.leftWing.xRot = abstractClientPlayer.elytraRotX;
-            this.leftWing.yRot = abstractClientPlayer.elytraRotY;
-            this.leftWing.zRot = abstractClientPlayer.elytraRotZ;
+        this.leftWing.y = $$8;
+        if ($$0 instanceof net.minecraft.client.player.AbstractClientPlayer) {
+            net.minecraft.client.player.AbstractClientPlayer $$13 = (net.minecraft.client.player.AbstractClientPlayer)$$0;
+            $$13.elytraRotX += ($$6 - $$13.elytraRotX) * 0.1f;
+            $$13.elytraRotY += ($$9 - $$13.elytraRotY) * 0.1f;
+            $$13.elytraRotZ += ($$7 - $$13.elytraRotZ) * 0.1f;
+            this.leftWing.xRot = $$13.elytraRotX;
+            this.leftWing.yRot = $$13.elytraRotY;
+            this.leftWing.zRot = $$13.elytraRotZ;
         } else {
-            this.leftWing.xRot = f;
-            this.leftWing.zRot = f2;
-            this.leftWing.yRot = f4;
+            this.leftWing.xRot = $$6;
+            this.leftWing.zRot = $$7;
+            this.leftWing.yRot = $$9;
         }
         this.rightWing.yRot = -this.leftWing.yRot;
         this.rightWing.y = this.leftWing.y;