--- net/minecraft/client/model/GhastModel.java
+++ net/minecraft/client/model/GhastModel.java
@@ -35,9 +35,9 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)32);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        for (int i = 0; i < this.tentacles.length; ++i) {
-            this.tentacles[i].xRot = 0.2f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.3f + (float)i)) + 0.4f;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        for (int $$6 = 0; $$6 < this.tentacles.length; ++$$6) {
+            this.tentacles[$$6].xRot = 0.2f * net.minecraft.util.Mth.sin((float)($$3 * 0.3f + (float)$$6)) + 0.4f;
         }
     }
 