--- net/minecraft/client/model/GuardianModel.java
+++ net/minecraft/client/model/GuardianModel.java
@@ -65,32 +65,32 @@
         return this.root;
     }
 
-    public void setupAnim(net.minecraft.world.entity.monster.Guardian entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f = ageInTicks - (float)entity.tickCount;
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        float f2 = (1.0f - entity.getSpikesAnimation(f)) * 0.55f;
-        this.setupSpikes(ageInTicks, f2);
-        net.minecraft.world.entity.Entity entity2 = net.minecraft.client.Minecraft.getInstance().getCameraEntity();
-        if (entity.hasActiveAttackTarget()) {
-            entity2 = entity.getActiveAttackTarget();
+    public void setupAnim(net.minecraft.world.entity.monster.Guardian $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        float $$6 = $$3 - (float)$$0.tickCount;
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        float $$7 = (1.0f - $$0.getSpikesAnimation($$6)) * 0.55f;
+        this.setupSpikes($$3, $$7);
+        net.minecraft.world.entity.Entity $$8 = net.minecraft.client.Minecraft.getInstance().getCameraEntity();
+        if ($$0.hasActiveAttackTarget()) {
+            $$8 = $$0.getActiveAttackTarget();
         }
-        if (entity2 != null) {
-            net.minecraft.world.phys.Vec3 vec3 = entity2.getEyePosition(0.0f);
-            net.minecraft.world.phys.Vec3 vec32 = entity.getEyePosition(0.0f);
-            double d = vec3.y - vec32.y;
-            this.eye.y = d > 0.0 ? 0.0f : 1.0f;
-            net.minecraft.world.phys.Vec3 vec33 = entity.getViewVector(0.0f);
-            vec33 = new net.minecraft.world.phys.Vec3(vec33.x, 0.0, vec33.z);
-            net.minecraft.world.phys.Vec3 vec34 = new net.minecraft.world.phys.Vec3(vec32.x - vec3.x, 0.0, vec32.z - vec3.z).normalize().yRot(1.5707964f);
-            double d2 = vec33.dot(vec34);
-            this.eye.x = net.minecraft.util.Mth.sqrt((float)((float)java.lang.Math.abs((double)d2))) * 2.0f * (float)java.lang.Math.signum((double)d2);
+        if ($$8 != null) {
+            net.minecraft.world.phys.Vec3 $$9 = $$8.getEyePosition(0.0f);
+            net.minecraft.world.phys.Vec3 $$10 = $$0.getEyePosition(0.0f);
+            double $$11 = $$9.y - $$10.y;
+            this.eye.y = $$11 > 0.0 ? 0.0f : 1.0f;
+            net.minecraft.world.phys.Vec3 $$12 = $$0.getViewVector(0.0f);
+            $$12 = new net.minecraft.world.phys.Vec3($$12.x, 0.0, $$12.z);
+            net.minecraft.world.phys.Vec3 $$13 = new net.minecraft.world.phys.Vec3($$10.x - $$9.x, 0.0, $$10.z - $$9.z).normalize().yRot(1.5707964f);
+            double $$14 = $$12.dot($$13);
+            this.eye.x = net.minecraft.util.Mth.sqrt((float)((float)java.lang.Math.abs((double)$$14))) * 2.0f * (float)java.lang.Math.signum((double)$$14);
         }
         this.eye.visible = true;
-        float f3 = entity.getTailAnimation(f);
-        this.tailParts[0].yRot = net.minecraft.util.Mth.sin((float)f3) * (float)java.lang.Math.PI * 0.05f;
-        this.tailParts[1].yRot = net.minecraft.util.Mth.sin((float)f3) * (float)java.lang.Math.PI * 0.1f;
-        this.tailParts[2].yRot = net.minecraft.util.Mth.sin((float)f3) * (float)java.lang.Math.PI * 0.15f;
+        float $$15 = $$0.getTailAnimation($$6);
+        this.tailParts[0].yRot = net.minecraft.util.Mth.sin((float)$$15) * (float)java.lang.Math.PI * 0.05f;
+        this.tailParts[1].yRot = net.minecraft.util.Mth.sin((float)$$15) * (float)java.lang.Math.PI * 0.1f;
+        this.tailParts[2].yRot = net.minecraft.util.Mth.sin((float)$$15) * (float)java.lang.Math.PI * 0.15f;
     }
 
     public void setupSpikes(float $$0, float $$1) {