--- net/minecraft/client/model/HoglinModel.java
+++ net/minecraft/client/model/HoglinModel.java
@@ -58,23 +58,23 @@
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.body, (java.lang.Object)this.rightFrontLeg, (java.lang.Object)this.leftFrontLeg, (java.lang.Object)this.rightHindLeg, (java.lang.Object)this.leftHindLeg);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.rightEar.zRot = -0.6981317f - limbSwingAmount * net.minecraft.util.Mth.sin((float)limbSwing);
-        this.leftEar.zRot = 0.6981317f + limbSwingAmount * net.minecraft.util.Mth.sin((float)limbSwing);
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        int n = ((net.minecraft.world.entity.monster.hoglin.HoglinBase)entity).getAttackAnimationRemainingTicks();
-        float f = 1.0f - (float)net.minecraft.util.Mth.abs((int)(10 - 2 * n)) / 10.0f;
-        this.head.xRot = net.minecraft.util.Mth.lerp((float)f, (float)0.87266463f, (float)-0.34906584f);
-        if (entity.isBaby()) {
-            this.head.y = net.minecraft.util.Mth.lerp((float)f, (float)2.0f, (float)5.0f);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.rightEar.zRot = -0.6981317f - $$2 * net.minecraft.util.Mth.sin((float)$$1);
+        this.leftEar.zRot = 0.6981317f + $$2 * net.minecraft.util.Mth.sin((float)$$1);
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        int $$6 = ((net.minecraft.world.entity.monster.hoglin.HoglinBase)$$0).getAttackAnimationRemainingTicks();
+        float $$7 = 1.0f - (float)net.minecraft.util.Mth.abs((int)(10 - 2 * $$6)) / 10.0f;
+        this.head.xRot = net.minecraft.util.Mth.lerp((float)$$7, (float)0.87266463f, (float)-0.34906584f);
+        if ($$0.isBaby()) {
+            this.head.y = net.minecraft.util.Mth.lerp((float)$$7, (float)2.0f, (float)5.0f);
             this.mane.z = -3.0f;
         } else {
             this.head.y = 2.0f;
             this.mane.z = -7.0f;
         }
-        float f2 = 1.2f;
-        this.rightFrontLeg.xRot = net.minecraft.util.Mth.cos((float)limbSwing) * 1.2f * limbSwingAmount;
-        this.rightHindLeg.xRot = this.leftFrontLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing + (float)java.lang.Math.PI)) * 1.2f * limbSwingAmount;
+        float $$8 = 1.2f;
+        this.rightFrontLeg.xRot = net.minecraft.util.Mth.cos((float)$$1) * 1.2f * $$2;
+        this.rightHindLeg.xRot = this.leftFrontLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 + (float)java.lang.Math.PI)) * 1.2f * $$2;
         this.leftHindLeg.xRot = this.rightFrontLeg.xRot;
     }
 }