--- net/minecraft/client/model/HorseModel.java
+++ net/minecraft/client/model/HorseModel.java
@@ -91,14 +91,14 @@
         return $$1;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        boolean bl = entity.isSaddled();
-        boolean bl2 = entity.isVehicle();
-        for (net.minecraft.client.model.geom.ModelPart modelPart : this.saddleParts) {
-            modelPart.visible = bl;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        boolean $$6 = $$0.isSaddled();
+        boolean $$7 = $$0.isVehicle();
+        for (net.minecraft.client.model.geom.ModelPart $$8 : this.saddleParts) {
+            $$8.visible = $$6;
         }
-        for (net.minecraft.client.model.geom.ModelPart modelPart : this.ridingParts) {
-            modelPart.visible = bl2 && bl;
+        for (net.minecraft.client.model.geom.ModelPart $$9 : this.ridingParts) {
+            $$9.visible = $$7 && $$6;
         }
         this.body.y = 11.0f;
     }
@@ -111,58 +111,58 @@
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.body, (java.lang.Object)this.rightHindLeg, (java.lang.Object)this.leftHindLeg, (java.lang.Object)this.rightFrontLeg, (java.lang.Object)this.leftFrontLeg, (java.lang.Object)this.rightHindBabyLeg, (java.lang.Object)this.leftHindBabyLeg, (java.lang.Object)this.rightFrontBabyLeg, (java.lang.Object)this.leftFrontBabyLeg);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        super.prepareMobModel(entity, limbSwing, limbSwingAmount, partialTick);
-        float f = net.minecraft.util.Mth.rotlerp((float)((net.minecraft.world.entity.animal.horse.AbstractHorse)entity).yBodyRotO, (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)entity).yBodyRot, (float)partialTick);
-        float f2 = net.minecraft.util.Mth.rotlerp((float)((net.minecraft.world.entity.animal.horse.AbstractHorse)entity).yHeadRotO, (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)entity).yHeadRot, (float)partialTick);
-        float f3 = net.minecraft.util.Mth.lerp((float)partialTick, (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)entity).xRotO, (float)entity.getXRot());
-        float f4 = f2 - f;
-        float f5 = f3 * ((float)java.lang.Math.PI / 180);
-        if (f4 > 20.0f) {
-            f4 = 20.0f;
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        super.prepareMobModel($$0, $$1, $$2, $$3);
+        float $$4 = net.minecraft.util.Mth.rotlerp((float)((net.minecraft.world.entity.animal.horse.AbstractHorse)$$0).yBodyRotO, (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)$$0).yBodyRot, (float)$$3);
+        float $$5 = net.minecraft.util.Mth.rotlerp((float)((net.minecraft.world.entity.animal.horse.AbstractHorse)$$0).yHeadRotO, (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)$$0).yHeadRot, (float)$$3);
+        float $$6 = net.minecraft.util.Mth.lerp((float)$$3, (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)$$0).xRotO, (float)$$0.getXRot());
+        float $$7 = $$5 - $$4;
+        float $$8 = $$6 * ((float)java.lang.Math.PI / 180);
+        if ($$7 > 20.0f) {
+            $$7 = 20.0f;
         }
-        if (f4 < -20.0f) {
-            f4 = -20.0f;
+        if ($$7 < -20.0f) {
+            $$7 = -20.0f;
         }
-        if (limbSwingAmount > 0.2f) {
-            f5 += net.minecraft.util.Mth.cos((float)(limbSwing * 0.4f)) * 0.15f * limbSwingAmount;
+        if ($$2 > 0.2f) {
+            $$8 += net.minecraft.util.Mth.cos((float)($$1 * 0.4f)) * 0.15f * $$2;
         }
-        float f6 = entity.getEatAnim(partialTick);
-        float f7 = entity.getStandAnim(partialTick);
-        float f8 = 1.0f - f7;
-        float f9 = entity.getMouthAnim(partialTick);
-        boolean bl = ((net.minecraft.world.entity.animal.horse.AbstractHorse)entity).tailCounter != 0;
-        float f10 = (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)entity).tickCount + partialTick;
+        float $$9 = $$0.getEatAnim($$3);
+        float $$10 = $$0.getStandAnim($$3);
+        float $$11 = 1.0f - $$10;
+        float $$12 = $$0.getMouthAnim($$3);
+        boolean $$13 = ((net.minecraft.world.entity.animal.horse.AbstractHorse)$$0).tailCounter != 0;
+        float $$14 = (float)((net.minecraft.world.entity.animal.horse.AbstractHorse)$$0).tickCount + $$3;
         this.headParts.y = 4.0f;
         this.headParts.z = -12.0f;
         this.body.xRot = 0.0f;
-        this.headParts.xRot = 0.5235988f + f5;
-        this.headParts.yRot = f4 * ((float)java.lang.Math.PI / 180);
-        float f11 = entity.isInWater() ? 0.2f : 1.0f;
-        float f12 = net.minecraft.util.Mth.cos((float)(f11 * limbSwing * 0.6662f + (float)java.lang.Math.PI));
-        float f13 = f12 * 0.8f * limbSwingAmount;
-        float f14 = (1.0f - java.lang.Math.max((float)f7, (float)f6)) * (0.5235988f + f5 + f9 * net.minecraft.util.Mth.sin((float)f10) * 0.05f);
-        this.headParts.xRot = f7 * (0.2617994f + f5) + f6 * (2.1816616f + net.minecraft.util.Mth.sin((float)f10) * 0.05f) + f14;
-        this.headParts.yRot = f7 * f4 * ((float)java.lang.Math.PI / 180) + (1.0f - java.lang.Math.max((float)f7, (float)f6)) * this.headParts.yRot;
-        this.headParts.y = f7 * -4.0f + f6 * 11.0f + (1.0f - java.lang.Math.max((float)f7, (float)f6)) * this.headParts.y;
-        this.headParts.z = f7 * -4.0f + f6 * -12.0f + (1.0f - java.lang.Math.max((float)f7, (float)f6)) * this.headParts.z;
-        this.body.xRot = f7 * -0.7853982f + f8 * this.body.xRot;
-        float f15 = 0.2617994f * f7;
-        float f16 = net.minecraft.util.Mth.cos((float)(f10 * 0.6f + (float)java.lang.Math.PI));
-        this.leftFrontLeg.y = 2.0f * f7 + 14.0f * f8;
-        this.leftFrontLeg.z = -6.0f * f7 - 10.0f * f8;
+        this.headParts.xRot = 0.5235988f + $$8;
+        this.headParts.yRot = $$7 * ((float)java.lang.Math.PI / 180);
+        float $$15 = $$0.isInWater() ? 0.2f : 1.0f;
+        float $$16 = net.minecraft.util.Mth.cos((float)($$15 * $$1 * 0.6662f + (float)java.lang.Math.PI));
+        float $$17 = $$16 * 0.8f * $$2;
+        float $$18 = (1.0f - java.lang.Math.max((float)$$10, (float)$$9)) * (0.5235988f + $$8 + $$12 * net.minecraft.util.Mth.sin((float)$$14) * 0.05f);
+        this.headParts.xRot = $$10 * (0.2617994f + $$8) + $$9 * (2.1816616f + net.minecraft.util.Mth.sin((float)$$14) * 0.05f) + $$18;
+        this.headParts.yRot = $$10 * $$7 * ((float)java.lang.Math.PI / 180) + (1.0f - java.lang.Math.max((float)$$10, (float)$$9)) * this.headParts.yRot;
+        this.headParts.y = $$10 * -4.0f + $$9 * 11.0f + (1.0f - java.lang.Math.max((float)$$10, (float)$$9)) * this.headParts.y;
+        this.headParts.z = $$10 * -4.0f + $$9 * -12.0f + (1.0f - java.lang.Math.max((float)$$10, (float)$$9)) * this.headParts.z;
+        this.body.xRot = $$10 * -0.7853982f + $$11 * this.body.xRot;
+        float $$19 = 0.2617994f * $$10;
+        float $$20 = net.minecraft.util.Mth.cos((float)($$14 * 0.6f + (float)java.lang.Math.PI));
+        this.leftFrontLeg.y = 2.0f * $$10 + 14.0f * $$11;
+        this.leftFrontLeg.z = -6.0f * $$10 - 10.0f * $$11;
         this.rightFrontLeg.y = this.leftFrontLeg.y;
         this.rightFrontLeg.z = this.leftFrontLeg.z;
-        float f17 = (-1.0471976f + f16) * f7 + f13 * f8;
-        float f18 = (-1.0471976f - f16) * f7 - f13 * f8;
-        this.leftHindLeg.xRot = f15 - f12 * 0.5f * limbSwingAmount * f8;
-        this.rightHindLeg.xRot = f15 + f12 * 0.5f * limbSwingAmount * f8;
-        this.leftFrontLeg.xRot = f17;
-        this.rightFrontLeg.xRot = f18;
-        this.tail.xRot = 0.5235988f + limbSwingAmount * 0.75f;
-        this.tail.y = -5.0f + limbSwingAmount;
-        this.tail.z = 2.0f + limbSwingAmount * 2.0f;
-        this.tail.yRot = bl ? net.minecraft.util.Mth.cos((float)(f10 * 0.7f)) : 0.0f;
+        float $$21 = (-1.0471976f + $$20) * $$10 + $$17 * $$11;
+        float $$22 = (-1.0471976f - $$20) * $$10 - $$17 * $$11;
+        this.leftHindLeg.xRot = $$19 - $$16 * 0.5f * $$2 * $$11;
+        this.rightHindLeg.xRot = $$19 + $$16 * 0.5f * $$2 * $$11;
+        this.leftFrontLeg.xRot = $$21;
+        this.rightFrontLeg.xRot = $$22;
+        this.tail.xRot = 0.5235988f + $$2 * 0.75f;
+        this.tail.y = -5.0f + $$2;
+        this.tail.z = 2.0f + $$2 * 2.0f;
+        this.tail.yRot = $$13 ? net.minecraft.util.Mth.cos((float)($$14 * 0.7f)) : 0.0f;
         this.rightHindBabyLeg.y = this.rightHindLeg.y;
         this.rightHindBabyLeg.z = this.rightHindLeg.z;
         this.rightHindBabyLeg.xRot = this.rightHindLeg.xRot;
@@ -175,16 +175,16 @@
         this.leftFrontBabyLeg.y = this.leftFrontLeg.y;
         this.leftFrontBabyLeg.z = this.leftFrontLeg.z;
         this.leftFrontBabyLeg.xRot = this.leftFrontLeg.xRot;
-        boolean bl2 = entity.isBaby();
-        this.rightHindLeg.visible = !bl2;
-        this.leftHindLeg.visible = !bl2;
-        this.rightFrontLeg.visible = !bl2;
-        this.leftFrontLeg.visible = !bl2;
-        this.rightHindBabyLeg.visible = bl2;
-        this.leftHindBabyLeg.visible = bl2;
-        this.rightFrontBabyLeg.visible = bl2;
-        this.leftFrontBabyLeg.visible = bl2;
-        this.body.y = bl2 ? 10.8f : 0.0f;
+        boolean $$23 = $$0.isBaby();
+        this.rightHindLeg.visible = !$$23;
+        this.leftHindLeg.visible = !$$23;
+        this.rightFrontLeg.visible = !$$23;
+        this.leftFrontLeg.visible = !$$23;
+        this.rightHindBabyLeg.visible = $$23;
+        this.leftHindBabyLeg.visible = $$23;
+        this.rightFrontBabyLeg.visible = $$23;
+        this.leftFrontBabyLeg.visible = $$23;
+        this.body.y = $$23 ? 10.8f : 0.0f;
     }
 }
 