--- net/minecraft/client/model/IllagerModel.java
+++ net/minecraft/client/model/IllagerModel.java
@@ -49,10 +49,10 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        boolean bl;
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        boolean $$7;
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
         if (this.riding) {
             this.rightArm.xRot = -0.62831855f;
             this.rightArm.yRot = 0.0f;
@@ -67,62 +67,62 @@
             this.leftLeg.yRot = -0.31415927f;
             this.leftLeg.zRot = -0.07853982f;
         } else {
-            this.rightArm.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * 2.0f * limbSwingAmount * 0.5f;
+            this.rightArm.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * 2.0f * $$2 * 0.5f;
             this.rightArm.yRot = 0.0f;
             this.rightArm.zRot = 0.0f;
-            this.leftArm.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * 2.0f * limbSwingAmount * 0.5f;
+            this.leftArm.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * 2.0f * $$2 * 0.5f;
             this.leftArm.yRot = 0.0f;
             this.leftArm.zRot = 0.0f;
-            this.rightLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * 1.4f * limbSwingAmount * 0.5f;
+            this.rightLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * 1.4f * $$2 * 0.5f;
             this.rightLeg.yRot = 0.0f;
             this.rightLeg.zRot = 0.0f;
-            this.leftLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * limbSwingAmount * 0.5f;
+            this.leftLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * $$2 * 0.5f;
             this.leftLeg.yRot = 0.0f;
             this.leftLeg.zRot = 0.0f;
         }
-        net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose illagerArmPose = entity.getArmPose();
-        if (illagerArmPose == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.ATTACKING) {
-            if (entity.getMainHandItem().isEmpty()) {
-                net.minecraft.client.model.AnimationUtils.animateZombieArms((net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.rightArm, (boolean)true, (float)this.attackTime, (float)ageInTicks);
+        net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose $$6 = $$0.getArmPose();
+        if ($$6 == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.ATTACKING) {
+            if ($$0.getMainHandItem().isEmpty()) {
+                net.minecraft.client.model.AnimationUtils.animateZombieArms((net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.rightArm, (boolean)true, (float)this.attackTime, (float)$$3);
             } else {
-                net.minecraft.client.model.AnimationUtils.swingWeaponDown((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, entity, (float)this.attackTime, (float)ageInTicks);
+                net.minecraft.client.model.AnimationUtils.swingWeaponDown((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, $$0, (float)this.attackTime, (float)$$3);
             }
-        } else if (illagerArmPose == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.SPELLCASTING) {
+        } else if ($$6 == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.SPELLCASTING) {
             this.rightArm.z = 0.0f;
             this.rightArm.x = -5.0f;
             this.leftArm.z = 0.0f;
             this.leftArm.x = 5.0f;
-            this.rightArm.xRot = net.minecraft.util.Mth.cos((float)(ageInTicks * 0.6662f)) * 0.25f;
-            this.leftArm.xRot = net.minecraft.util.Mth.cos((float)(ageInTicks * 0.6662f)) * 0.25f;
+            this.rightArm.xRot = net.minecraft.util.Mth.cos((float)($$3 * 0.6662f)) * 0.25f;
+            this.leftArm.xRot = net.minecraft.util.Mth.cos((float)($$3 * 0.6662f)) * 0.25f;
             this.rightArm.zRot = 2.3561945f;
             this.leftArm.zRot = -2.3561945f;
             this.rightArm.yRot = 0.0f;
             this.leftArm.yRot = 0.0f;
-        } else if (illagerArmPose == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.BOW_AND_ARROW) {
+        } else if ($$6 == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.BOW_AND_ARROW) {
             this.rightArm.yRot = -0.1f + this.head.yRot;
             this.rightArm.xRot = -1.5707964f + this.head.xRot;
             this.leftArm.xRot = -0.9424779f + this.head.xRot;
             this.leftArm.yRot = this.head.yRot - 0.4f;
             this.leftArm.zRot = 1.5707964f;
-        } else if (illagerArmPose == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CROSSBOW_HOLD) {
+        } else if ($$6 == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CROSSBOW_HOLD) {
             net.minecraft.client.model.AnimationUtils.animateCrossbowHold((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.head, (boolean)true);
-        } else if (illagerArmPose == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CROSSBOW_CHARGE) {
-            net.minecraft.client.model.AnimationUtils.animateCrossbowCharge((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, entity, (boolean)true);
-        } else if (illagerArmPose == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CELEBRATING) {
+        } else if ($$6 == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CROSSBOW_CHARGE) {
+            net.minecraft.client.model.AnimationUtils.animateCrossbowCharge((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, $$0, (boolean)true);
+        } else if ($$6 == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CELEBRATING) {
             this.rightArm.z = 0.0f;
             this.rightArm.x = -5.0f;
-            this.rightArm.xRot = net.minecraft.util.Mth.cos((float)(ageInTicks * 0.6662f)) * 0.05f;
+            this.rightArm.xRot = net.minecraft.util.Mth.cos((float)($$3 * 0.6662f)) * 0.05f;
             this.rightArm.zRot = 2.670354f;
             this.rightArm.yRot = 0.0f;
             this.leftArm.z = 0.0f;
             this.leftArm.x = 5.0f;
-            this.leftArm.xRot = net.minecraft.util.Mth.cos((float)(ageInTicks * 0.6662f)) * 0.05f;
+            this.leftArm.xRot = net.minecraft.util.Mth.cos((float)($$3 * 0.6662f)) * 0.05f;
             this.leftArm.zRot = -2.3561945f;
             this.leftArm.yRot = 0.0f;
         }
-        this.arms.visible = bl = illagerArmPose == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CROSSED;
-        this.leftArm.visible = !bl;
-        this.rightArm.visible = !bl;
+        this.arms.visible = $$7 = $$6 == net.minecraft.world.entity.monster.AbstractIllager$IllagerArmPose.CROSSED;
+        this.leftArm.visible = !$$7;
+        this.rightArm.visible = !$$7;
     }
 
     public net.minecraft.client.model.geom.ModelPart getArm(net.minecraft.world.entity.HumanoidArm $$0) {
@@ -140,8 +140,8 @@
         return this.head;
     }
 
-    public void translateToHand(net.minecraft.world.entity.HumanoidArm side, com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        this.getArm(side).translateAndRotate(matrixStack);
+    public void translateToHand(net.minecraft.world.entity.HumanoidArm $$0, com.mojang.blaze3d.vertex.PoseStack $$1) {
+        this.getArm($$0).translateAndRotate($$1);
     }
 }
 