--- net/minecraft/client/model/IronGolemModel.java
+++ net/minecraft/client/model/IronGolemModel.java
@@ -38,28 +38,28 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        this.rightLeg.xRot = -1.5f * net.minecraft.util.Mth.triangleWave((float)limbSwing, (float)13.0f) * limbSwingAmount;
-        this.leftLeg.xRot = 1.5f * net.minecraft.util.Mth.triangleWave((float)limbSwing, (float)13.0f) * limbSwingAmount;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        this.rightLeg.xRot = -1.5f * net.minecraft.util.Mth.triangleWave((float)$$1, (float)13.0f) * $$2;
+        this.leftLeg.xRot = 1.5f * net.minecraft.util.Mth.triangleWave((float)$$1, (float)13.0f) * $$2;
         this.rightLeg.yRot = 0.0f;
         this.leftLeg.yRot = 0.0f;
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        int n = entity.getAttackAnimationTick();
-        if (n > 0) {
-            this.rightArm.xRot = -2.0f + 1.5f * net.minecraft.util.Mth.triangleWave((float)((float)n - partialTick), (float)10.0f);
-            this.leftArm.xRot = -2.0f + 1.5f * net.minecraft.util.Mth.triangleWave((float)((float)n - partialTick), (float)10.0f);
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        int $$4 = $$0.getAttackAnimationTick();
+        if ($$4 > 0) {
+            this.rightArm.xRot = -2.0f + 1.5f * net.minecraft.util.Mth.triangleWave((float)((float)$$4 - $$3), (float)10.0f);
+            this.leftArm.xRot = -2.0f + 1.5f * net.minecraft.util.Mth.triangleWave((float)((float)$$4 - $$3), (float)10.0f);
         } else {
-            int n2 = entity.getOfferFlowerTick();
-            if (n2 > 0) {
-                this.rightArm.xRot = -0.8f + 0.025f * net.minecraft.util.Mth.triangleWave((float)n2, (float)70.0f);
+            int $$5 = $$0.getOfferFlowerTick();
+            if ($$5 > 0) {
+                this.rightArm.xRot = -0.8f + 0.025f * net.minecraft.util.Mth.triangleWave((float)$$5, (float)70.0f);
                 this.leftArm.xRot = 0.0f;
             } else {
-                this.rightArm.xRot = (-0.2f + 1.5f * net.minecraft.util.Mth.triangleWave((float)limbSwing, (float)13.0f)) * limbSwingAmount;
-                this.leftArm.xRot = (-0.2f - 1.5f * net.minecraft.util.Mth.triangleWave((float)limbSwing, (float)13.0f)) * limbSwingAmount;
+                this.rightArm.xRot = (-0.2f + 1.5f * net.minecraft.util.Mth.triangleWave((float)$$1, (float)13.0f)) * $$2;
+                this.leftArm.xRot = (-0.2f - 1.5f * net.minecraft.util.Mth.triangleWave((float)$$1, (float)13.0f)) * $$2;
             }
         }
     }