--- net/minecraft/client/model/LavaSlimeModel.java
+++ net/minecraft/client/model/LavaSlimeModel.java
@@ -39,16 +39,16 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)32);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        float f = net.minecraft.util.Mth.lerp((float)partialTick, (float)((net.minecraft.world.entity.monster.Slime)entity).oSquish, (float)((net.minecraft.world.entity.monster.Slime)entity).squish);
-        if (f < 0.0f) {
-            f = 0.0f;
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        float $$4 = net.minecraft.util.Mth.lerp((float)$$3, (float)((net.minecraft.world.entity.monster.Slime)$$0).oSquish, (float)((net.minecraft.world.entity.monster.Slime)$$0).squish);
+        if ($$4 < 0.0f) {
+            $$4 = 0.0f;
         }
-        for (int i = 0; i < this.bodyCubes.length; ++i) {
-            this.bodyCubes[i].y = (float)(-(4 - i)) * f * 1.7f;
+        for (int $$5 = 0; $$5 < this.bodyCubes.length; ++$$5) {
+            this.bodyCubes[$$5].y = (float)(-(4 - $$5)) * $$4 * 1.7f;
         }
     }
 