--- net/minecraft/client/model/LeashKnotModel.java
+++ net/minecraft/client/model/LeashKnotModel.java
@@ -26,9 +26,9 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.knot.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.knot.xRot = headPitch * ((float)java.lang.Math.PI / 180);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.knot.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.knot.xRot = $$5 * ((float)java.lang.Math.PI / 180);
     }
 }
 