--- net/minecraft/client/model/PandaModel.java
+++ net/minecraft/client/model/PandaModel.java
@@ -29,34 +29,34 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)64);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        super.prepareMobModel(entity, limbSwing, limbSwingAmount, partialTick);
-        this.sitAmount = entity.getSitAmount(partialTick);
-        this.lieOnBackAmount = entity.getLieOnBackAmount(partialTick);
-        this.rollAmount = entity.isBaby() ? 0.0f : entity.getRollAmount(partialTick);
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        super.prepareMobModel($$0, $$1, $$2, $$3);
+        this.sitAmount = $$0.getSitAmount($$3);
+        this.lieOnBackAmount = $$0.getLieOnBackAmount($$3);
+        this.rollAmount = $$0.isBaby() ? 0.0f : $$0.getRollAmount($$3);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        boolean bl = entity.getUnhappyCounter() > 0;
-        boolean bl2 = entity.isSneezing();
-        int n = entity.getSneezeCounter();
-        boolean bl3 = entity.isEating();
-        boolean bl4 = entity.isScared();
-        if (bl) {
-            this.head.yRot = 0.35f * net.minecraft.util.Mth.sin((float)(0.6f * ageInTicks));
-            this.head.zRot = 0.35f * net.minecraft.util.Mth.sin((float)(0.6f * ageInTicks));
-            this.rightFrontLeg.xRot = -0.75f * net.minecraft.util.Mth.sin((float)(0.3f * ageInTicks));
-            this.leftFrontLeg.xRot = 0.75f * net.minecraft.util.Mth.sin((float)(0.3f * ageInTicks));
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
+        boolean $$6 = $$0.getUnhappyCounter() > 0;
+        boolean $$7 = $$0.isSneezing();
+        int $$8 = $$0.getSneezeCounter();
+        boolean $$9 = $$0.isEating();
+        boolean $$10 = $$0.isScared();
+        if ($$6) {
+            this.head.yRot = 0.35f * net.minecraft.util.Mth.sin((float)(0.6f * $$3));
+            this.head.zRot = 0.35f * net.minecraft.util.Mth.sin((float)(0.6f * $$3));
+            this.rightFrontLeg.xRot = -0.75f * net.minecraft.util.Mth.sin((float)(0.3f * $$3));
+            this.leftFrontLeg.xRot = 0.75f * net.minecraft.util.Mth.sin((float)(0.3f * $$3));
         } else {
             this.head.zRot = 0.0f;
         }
-        if (bl2) {
-            if (n < 15) {
-                this.head.xRot = -0.7853982f * (float)n / 14.0f;
-            } else if (n < 20) {
-                float f = (n - 15) / 5;
-                this.head.xRot = -0.7853982f + 0.7853982f * f;
+        if ($$7) {
+            if ($$8 < 15) {
+                this.head.xRot = -0.7853982f * (float)$$8 / 14.0f;
+            } else if ($$8 < 20) {
+                float $$11 = ($$8 - 15) / 5;
+                this.head.xRot = -0.7853982f + 0.7853982f * $$11;
             }
         }
         if (this.sitAmount > 0.0f) {
@@ -66,12 +66,12 @@
             this.leftFrontLeg.zRot = 0.27079642f;
             this.rightHindLeg.zRot = 0.5707964f;
             this.leftHindLeg.zRot = -0.5707964f;
-            if (bl3) {
-                this.head.xRot = 1.5707964f + 0.2f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.6f));
-                this.rightFrontLeg.xRot = -0.4f - 0.2f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.6f));
-                this.leftFrontLeg.xRot = -0.4f - 0.2f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.6f));
+            if ($$9) {
+                this.head.xRot = 1.5707964f + 0.2f * net.minecraft.util.Mth.sin((float)($$3 * 0.6f));
+                this.rightFrontLeg.xRot = -0.4f - 0.2f * net.minecraft.util.Mth.sin((float)($$3 * 0.6f));
+                this.leftFrontLeg.xRot = -0.4f - 0.2f * net.minecraft.util.Mth.sin((float)($$3 * 0.6f));
             }
-            if (bl4) {
+            if ($$10) {
                 this.head.xRot = 2.1707964f;
                 this.rightFrontLeg.xRot = -0.9f;
                 this.leftFrontLeg.xRot = -0.9f;
@@ -83,18 +83,18 @@
             this.leftFrontLeg.zRot = 0.0f;
         }
         if (this.lieOnBackAmount > 0.0f) {
-            this.rightHindLeg.xRot = -0.6f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.15f));
-            this.leftHindLeg.xRot = 0.6f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.15f));
-            this.rightFrontLeg.xRot = 0.3f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.25f));
-            this.leftFrontLeg.xRot = -0.3f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.25f));
+            this.rightHindLeg.xRot = -0.6f * net.minecraft.util.Mth.sin((float)($$3 * 0.15f));
+            this.leftHindLeg.xRot = 0.6f * net.minecraft.util.Mth.sin((float)($$3 * 0.15f));
+            this.rightFrontLeg.xRot = 0.3f * net.minecraft.util.Mth.sin((float)($$3 * 0.25f));
+            this.leftFrontLeg.xRot = -0.3f * net.minecraft.util.Mth.sin((float)($$3 * 0.25f));
             this.head.xRot = net.minecraft.client.model.ModelUtils.rotlerpRad((float)this.head.xRot, (float)1.5707964f, (float)this.lieOnBackAmount);
         }
         if (this.rollAmount > 0.0f) {
             this.head.xRot = net.minecraft.client.model.ModelUtils.rotlerpRad((float)this.head.xRot, (float)2.0561945f, (float)this.rollAmount);
-            this.rightHindLeg.xRot = -0.5f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.5f));
-            this.leftHindLeg.xRot = 0.5f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.5f));
-            this.rightFrontLeg.xRot = 0.5f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.5f));
-            this.leftFrontLeg.xRot = -0.5f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.5f));
+            this.rightHindLeg.xRot = -0.5f * net.minecraft.util.Mth.sin((float)($$3 * 0.5f));
+            this.leftHindLeg.xRot = 0.5f * net.minecraft.util.Mth.sin((float)($$3 * 0.5f));
+            this.rightFrontLeg.xRot = 0.5f * net.minecraft.util.Mth.sin((float)($$3 * 0.5f));
+            this.leftFrontLeg.xRot = -0.5f * net.minecraft.util.Mth.sin((float)($$3 * 0.5f));
         }
     }
 }