--- net/minecraft/client/model/ParrotModel.java
+++ net/minecraft/client/model/ParrotModel.java
@@ -51,18 +51,18 @@
         return this.root;
     }
 
-    public void setupAnim(net.minecraft.world.entity.animal.Parrot entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.setupAnim(net.minecraft.client.model.ParrotModel.getState((net.minecraft.world.entity.animal.Parrot)entity), entity.tickCount, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
+    public void setupAnim(net.minecraft.world.entity.animal.Parrot $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.setupAnim(net.minecraft.client.model.ParrotModel.getState((net.minecraft.world.entity.animal.Parrot)$$0), $$0.tickCount, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public void prepareMobModel(net.minecraft.world.entity.animal.Parrot entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        this.prepare(net.minecraft.client.model.ParrotModel.getState((net.minecraft.world.entity.animal.Parrot)entity));
+    public void prepareMobModel(net.minecraft.world.entity.animal.Parrot $$0, float $$1, float $$2, float $$3) {
+        this.prepare(net.minecraft.client.model.ParrotModel.getState((net.minecraft.world.entity.animal.Parrot)$$0));
     }
 
-    public void renderOnShoulder(com.mojang.blaze3d.vertex.PoseStack matrixStack, com.mojang.blaze3d.vertex.VertexConsumer buffer, int packedLight, int packedOverlay, float f, float f2, float f3, float f4, int n) {
+    public void renderOnShoulder(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7, int $$8) {
         this.prepare(net.minecraft.client.model.ParrotModel$State.ON_SHOULDER);
-        this.setupAnim(net.minecraft.client.model.ParrotModel$State.ON_SHOULDER, n, f, f2, 0.0f, f3, f4);
-        this.root.render(matrixStack, buffer, packedLight, packedOverlay);
+        this.setupAnim(net.minecraft.client.model.ParrotModel$State.ON_SHOULDER, $$8, $$4, $$5, 0.0f, $$6, $$7);
+        this.root.render($$0, $$1, $$2, $$3);
     }
 
     public void setupAnim(net.minecraft.client.model.ParrotModel$State $$0, int $$1, float $$2, float $$3, float $$4, float $$5, float $$6) {