--- net/minecraft/client/model/PiglinModel.java
+++ net/minecraft/client/model/PiglinModel.java
@@ -34,41 +34,41 @@
         return $$1;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
         this.body.loadPose(this.bodyDefault);
         this.head.loadPose(this.headDefault);
         this.leftArm.loadPose(this.leftArmDefault);
         this.rightArm.loadPose(this.rightArmDefault);
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        float f = 0.5235988f;
-        float f2 = ageInTicks * 0.1f + limbSwing * 0.5f;
-        float f3 = 0.08f + limbSwingAmount * 0.4f;
-        this.leftEar.zRot = -0.5235988f - net.minecraft.util.Mth.cos((float)(f2 * 1.2f)) * f3;
-        this.rightEar.zRot = 0.5235988f + net.minecraft.util.Mth.cos((float)f2) * f3;
-        if (entity instanceof net.minecraft.world.entity.monster.piglin.AbstractPiglin) {
-            net.minecraft.world.entity.monster.piglin.AbstractPiglin abstractPiglin = (net.minecraft.world.entity.monster.piglin.AbstractPiglin)entity;
-            net.minecraft.world.entity.monster.piglin.PiglinArmPose piglinArmPose = abstractPiglin.getArmPose();
-            if (piglinArmPose == net.minecraft.world.entity.monster.piglin.PiglinArmPose.DANCING) {
-                float f4 = ageInTicks / 60.0f;
-                this.rightEar.zRot = 0.5235988f + (float)java.lang.Math.PI / 180 * net.minecraft.util.Mth.sin((float)(f4 * 30.0f)) * 10.0f;
-                this.leftEar.zRot = -0.5235988f - (float)java.lang.Math.PI / 180 * net.minecraft.util.Mth.cos((float)(f4 * 30.0f)) * 10.0f;
-                this.head.x = net.minecraft.util.Mth.sin((float)(f4 * 10.0f));
-                this.head.y = net.minecraft.util.Mth.sin((float)(f4 * 40.0f)) + 0.4f;
-                this.rightArm.zRot = (float)java.lang.Math.PI / 180 * (70.0f + net.minecraft.util.Mth.cos((float)(f4 * 40.0f)) * 10.0f);
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
+        float $$6 = 0.5235988f;
+        float $$7 = $$3 * 0.1f + $$1 * 0.5f;
+        float $$8 = 0.08f + $$2 * 0.4f;
+        this.leftEar.zRot = -0.5235988f - net.minecraft.util.Mth.cos((float)($$7 * 1.2f)) * $$8;
+        this.rightEar.zRot = 0.5235988f + net.minecraft.util.Mth.cos((float)$$7) * $$8;
+        if ($$0 instanceof net.minecraft.world.entity.monster.piglin.AbstractPiglin) {
+            net.minecraft.world.entity.monster.piglin.AbstractPiglin $$9 = (net.minecraft.world.entity.monster.piglin.AbstractPiglin)$$0;
+            net.minecraft.world.entity.monster.piglin.PiglinArmPose $$10 = $$9.getArmPose();
+            if ($$10 == net.minecraft.world.entity.monster.piglin.PiglinArmPose.DANCING) {
+                float $$11 = $$3 / 60.0f;
+                this.rightEar.zRot = 0.5235988f + (float)java.lang.Math.PI / 180 * net.minecraft.util.Mth.sin((float)($$11 * 30.0f)) * 10.0f;
+                this.leftEar.zRot = -0.5235988f - (float)java.lang.Math.PI / 180 * net.minecraft.util.Mth.cos((float)($$11 * 30.0f)) * 10.0f;
+                this.head.x = net.minecraft.util.Mth.sin((float)($$11 * 10.0f));
+                this.head.y = net.minecraft.util.Mth.sin((float)($$11 * 40.0f)) + 0.4f;
+                this.rightArm.zRot = (float)java.lang.Math.PI / 180 * (70.0f + net.minecraft.util.Mth.cos((float)($$11 * 40.0f)) * 10.0f);
                 this.leftArm.zRot = this.rightArm.zRot * -1.0f;
-                this.rightArm.y = net.minecraft.util.Mth.sin((float)(f4 * 40.0f)) * 0.5f + 1.5f;
-                this.leftArm.y = net.minecraft.util.Mth.sin((float)(f4 * 40.0f)) * 0.5f + 1.5f;
-                this.body.y = net.minecraft.util.Mth.sin((float)(f4 * 40.0f)) * 0.35f;
-            } else if (piglinArmPose == net.minecraft.world.entity.monster.piglin.PiglinArmPose.ATTACKING_WITH_MELEE_WEAPON && this.attackTime == 0.0f) {
-                this.holdWeaponHigh(entity);
-            } else if (piglinArmPose == net.minecraft.world.entity.monster.piglin.PiglinArmPose.CROSSBOW_HOLD) {
-                net.minecraft.client.model.AnimationUtils.animateCrossbowHold((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.head, (!entity.isLeftHanded() ? 1 : 0) != 0);
-            } else if (piglinArmPose == net.minecraft.world.entity.monster.piglin.PiglinArmPose.CROSSBOW_CHARGE) {
-                net.minecraft.client.model.AnimationUtils.animateCrossbowCharge((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, entity, (!entity.isLeftHanded() ? 1 : 0) != 0);
-            } else if (piglinArmPose == net.minecraft.world.entity.monster.piglin.PiglinArmPose.ADMIRING_ITEM) {
+                this.rightArm.y = net.minecraft.util.Mth.sin((float)($$11 * 40.0f)) * 0.5f + 1.5f;
+                this.leftArm.y = net.minecraft.util.Mth.sin((float)($$11 * 40.0f)) * 0.5f + 1.5f;
+                this.body.y = net.minecraft.util.Mth.sin((float)($$11 * 40.0f)) * 0.35f;
+            } else if ($$10 == net.minecraft.world.entity.monster.piglin.PiglinArmPose.ATTACKING_WITH_MELEE_WEAPON && this.attackTime == 0.0f) {
+                this.holdWeaponHigh($$0);
+            } else if ($$10 == net.minecraft.world.entity.monster.piglin.PiglinArmPose.CROSSBOW_HOLD) {
+                net.minecraft.client.model.AnimationUtils.animateCrossbowHold((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.head, (!$$0.isLeftHanded() ? 1 : 0) != 0);
+            } else if ($$10 == net.minecraft.world.entity.monster.piglin.PiglinArmPose.CROSSBOW_CHARGE) {
+                net.minecraft.client.model.AnimationUtils.animateCrossbowCharge((net.minecraft.client.model.geom.ModelPart)this.rightArm, (net.minecraft.client.model.geom.ModelPart)this.leftArm, $$0, (!$$0.isLeftHanded() ? 1 : 0) != 0);
+            } else if ($$10 == net.minecraft.world.entity.monster.piglin.PiglinArmPose.ADMIRING_ITEM) {
                 this.head.xRot = 0.5f;
                 this.head.yRot = 0.0f;
-                if (entity.isLeftHanded()) {
+                if ($$0.isLeftHanded()) {
                     this.rightArm.yRot = -0.5f;
                     this.rightArm.xRot = -0.9f;
                 } else {
@@ -76,8 +76,8 @@
                     this.leftArm.xRot = -0.9f;
                 }
             }
-        } else if (entity.getType() == net.minecraft.world.entity.EntityType.ZOMBIFIED_PIGLIN) {
-            net.minecraft.client.model.AnimationUtils.animateZombieArms((net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.rightArm, (boolean)entity.isAggressive(), (float)this.attackTime, (float)ageInTicks);
+        } else if ($$0.getType() == net.minecraft.world.entity.EntityType.ZOMBIFIED_PIGLIN) {
+            net.minecraft.client.model.AnimationUtils.animateZombieArms((net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.rightArm, (boolean)$$0.isAggressive(), (float)this.attackTime, (float)$$3);
         }
         this.leftPants.copyFrom(this.leftLeg);
         this.rightPants.copyFrom(this.rightLeg);