--- net/minecraft/client/model/PufferfishSmallModel.java
+++ net/minecraft/client/model/PufferfishSmallModel.java
@@ -33,9 +33,9 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.rightFin.zRot = -0.2f + 0.4f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.2f));
-        this.leftFin.zRot = 0.2f - 0.4f * net.minecraft.util.Mth.sin((float)(ageInTicks * 0.2f));
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.rightFin.zRot = -0.2f + 0.4f * net.minecraft.util.Mth.sin((float)($$3 * 0.2f));
+        this.leftFin.zRot = 0.2f - 0.4f * net.minecraft.util.Mth.sin((float)($$3 * 0.2f));
     }
 }
 