--- net/minecraft/client/model/RaftModel.java
+++ net/minecraft/client/model/RaftModel.java
@@ -1,0 +1,67 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.model;
+
+public class net.minecraft.client.model.RaftModel
+extends net.minecraft.client.model.ListModel<net.minecraft.world.entity.vehicle.Boat>
+ {
+    public static java.lang.String LEFT_PADDLE = "left_paddle";
+    public static java.lang.String RIGHT_PADDLE = "right_paddle";
+    public static java.lang.String BOTTOM = "bottom";
+    public net.minecraft.client.model.geom.ModelPart leftPaddle;
+    public net.minecraft.client.model.geom.ModelPart rightPaddle;
+    public com.google.common.collect.ImmutableList<net.minecraft.client.model.geom.ModelPart> parts;
+
+    public net.minecraft.client.model.RaftModel(net.minecraft.client.model.geom.ModelPart $$0) {
+        this.leftPaddle = $$0.getChild("left_paddle");
+        this.rightPaddle = $$0.getChild("right_paddle");
+        this.parts = this.createPartsBuilder($$0).build();
+    }
+
+    public com.google.common.collect.ImmutableList$Builder<net.minecraft.client.model.geom.ModelPart> createPartsBuilder(net.minecraft.client.model.geom.ModelPart $$0) {
+        com.google.common.collect.ImmutableList$Builder $$1 = new com.google.common.collect.ImmutableList$Builder();
+        $$1.add((java.lang.Object[])new net.minecraft.client.model.geom.ModelPart[]{$$0.getChild("bottom"), this.leftPaddle, this.rightPaddle});
+        return $$1;
+    }
+
+    public static void createChildren(net.minecraft.client.model.geom.builders.PartDefinition $$0) {
+        $$0.addOrReplaceChild("bottom", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 0).addBox(-14.0f, -11.0f, -4.0f, 28.0f, 20.0f, 4.0f).texOffs(0, 0).addBox(-14.0f, -9.0f, -8.0f, 28.0f, 16.0f, 4.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)0.0f, (float)-3.0f, (float)1.0f, (float)1.5708f, (float)0.0f, (float)0.0f));
+        int $$1 = 20;
+        int $$2 = 7;
+        int $$3 = 6;
+        float $$4 = -5.0f;
+        $$0.addOrReplaceChild("left_paddle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(0, 24).addBox(-1.0f, 0.0f, -5.0f, 2.0f, 2.0f, 18.0f).addBox(-1.001f, -3.0f, 8.0f, 1.0f, 6.0f, 7.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)3.0f, (float)-5.0f, (float)9.0f, (float)0.0f, (float)0.0f, (float)0.19634955f));
+        $$0.addOrReplaceChild("right_paddle", net.minecraft.client.model.geom.builders.CubeListBuilder.create().texOffs(40, 24).addBox(-1.0f, 0.0f, -5.0f, 2.0f, 2.0f, 18.0f).addBox(0.001f, -3.0f, 8.0f, 1.0f, 6.0f, 7.0f), net.minecraft.client.model.geom.PartPose.offsetAndRotation((float)3.0f, (float)-5.0f, (float)-9.0f, (float)0.0f, (float)((float)java.lang.Math.PI), (float)0.19634955f));
+    }
+
+    public static net.minecraft.client.model.geom.builders.LayerDefinition createBodyModel() {
+        net.minecraft.client.model.geom.builders.MeshDefinition $$0 = new net.minecraft.client.model.geom.builders.MeshDefinition();
+        net.minecraft.client.model.geom.builders.PartDefinition $$1 = $$0.getRoot();
+        net.minecraft.client.model.RaftModel.createChildren((net.minecraft.client.model.geom.builders.PartDefinition)$$1);
+        return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)128, (int)64);
+    }
+
+    public void setupAnim(net.minecraft.world.entity.vehicle.Boat $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        net.minecraft.client.model.RaftModel.animatePaddle((net.minecraft.world.entity.vehicle.Boat)$$0, (int)0, (net.minecraft.client.model.geom.ModelPart)this.leftPaddle, (float)$$1);
+        net.minecraft.client.model.RaftModel.animatePaddle((net.minecraft.world.entity.vehicle.Boat)$$0, (int)1, (net.minecraft.client.model.geom.ModelPart)this.rightPaddle, (float)$$1);
+    }
+
+    public com.google.common.collect.ImmutableList<net.minecraft.client.model.geom.ModelPart> parts() {
+        return this.parts;
+    }
+
+    public static void animatePaddle(net.minecraft.world.entity.vehicle.Boat $$0, int $$1, net.minecraft.client.model.geom.ModelPart $$2, float $$3) {
+        float $$4 = $$0.getRowingTime($$1, $$3);
+        $$2.xRot = net.minecraft.util.Mth.clampedLerp((float)-1.0471976f, (float)-0.2617994f, (float)((net.minecraft.util.Mth.sin((float)(-$$4)) + 1.0f) / 2.0f));
+        $$2.yRot = net.minecraft.util.Mth.clampedLerp((float)-0.7853982f, (float)0.7853982f, (float)((net.minecraft.util.Mth.sin((float)(-$$4 + 1.0f)) + 1.0f) / 2.0f));
+        if ($$1 == 1) {
+            $$2.yRot = (float)java.lang.Math.PI - $$2.yRot;
+        }
+    }
+
+    public /* synthetic */ java.lang.Iterable parts() {
+        return this.parts();
+    }
+}
+