--- net/minecraft/client/model/RavagerModel.java
+++ net/minecraft/client/model/RavagerModel.java
@@ -47,48 +47,48 @@
         return this.root;
     }
 
-    public void setupAnim(net.minecraft.world.entity.monster.Ravager entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        float f = 0.4f * limbSwingAmount;
-        this.rightHindLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * f;
-        this.leftHindLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * f;
-        this.rightFrontLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * f;
-        this.leftFrontLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * f;
+    public void setupAnim(net.minecraft.world.entity.monster.Ravager $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        float $$6 = 0.4f * $$2;
+        this.rightHindLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * $$6;
+        this.leftHindLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * $$6;
+        this.rightFrontLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * $$6;
+        this.leftFrontLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * $$6;
     }
 
-    public void prepareMobModel(net.minecraft.world.entity.monster.Ravager entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        super.prepareMobModel((net.minecraft.world.entity.Entity)entity, limbSwing, limbSwingAmount, partialTick);
-        int n = entity.getStunnedTick();
-        int n2 = entity.getRoarTick();
-        int n3 = 20;
-        int n4 = entity.getAttackTick();
-        int n5 = 10;
-        if (n4 > 0) {
-            float f = net.minecraft.util.Mth.triangleWave((float)((float)n4 - partialTick), (float)10.0f);
-            float f2 = (1.0f + f) * 0.5f;
-            float f3 = f2 * f2 * f2 * 12.0f;
-            float f4 = f3 * net.minecraft.util.Mth.sin((float)this.neck.xRot);
-            this.neck.z = -6.5f + f3;
-            this.neck.y = -7.0f - f4;
-            float f5 = net.minecraft.util.Mth.sin((float)(((float)n4 - partialTick) / 10.0f * (float)java.lang.Math.PI * 0.25f));
-            this.mouth.xRot = 1.5707964f * f5;
-            this.mouth.xRot = n4 > 5 ? net.minecraft.util.Mth.sin((float)(((float)(-4 + n4) - partialTick) / 4.0f)) * (float)java.lang.Math.PI * 0.4f : 0.15707964f * net.minecraft.util.Mth.sin((float)((float)java.lang.Math.PI * ((float)n4 - partialTick) / 10.0f));
+    public void prepareMobModel(net.minecraft.world.entity.monster.Ravager $$0, float $$1, float $$2, float $$3) {
+        super.prepareMobModel((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3);
+        int $$4 = $$0.getStunnedTick();
+        int $$5 = $$0.getRoarTick();
+        int $$6 = 20;
+        int $$7 = $$0.getAttackTick();
+        int $$8 = 10;
+        if ($$7 > 0) {
+            float $$9 = net.minecraft.util.Mth.triangleWave((float)((float)$$7 - $$3), (float)10.0f);
+            float $$10 = (1.0f + $$9) * 0.5f;
+            float $$11 = $$10 * $$10 * $$10 * 12.0f;
+            float $$12 = $$11 * net.minecraft.util.Mth.sin((float)this.neck.xRot);
+            this.neck.z = -6.5f + $$11;
+            this.neck.y = -7.0f - $$12;
+            float $$13 = net.minecraft.util.Mth.sin((float)(((float)$$7 - $$3) / 10.0f * (float)java.lang.Math.PI * 0.25f));
+            this.mouth.xRot = 1.5707964f * $$13;
+            this.mouth.xRot = $$7 > 5 ? net.minecraft.util.Mth.sin((float)(((float)(-4 + $$7) - $$3) / 4.0f)) * (float)java.lang.Math.PI * 0.4f : 0.15707964f * net.minecraft.util.Mth.sin((float)((float)java.lang.Math.PI * ((float)$$7 - $$3) / 10.0f));
         } else {
-            float f = -1.0f;
-            float f6 = -1.0f * net.minecraft.util.Mth.sin((float)this.neck.xRot);
+            float $$14 = -1.0f;
+            float $$15 = -1.0f * net.minecraft.util.Mth.sin((float)this.neck.xRot);
             this.neck.x = 0.0f;
-            this.neck.y = -7.0f - f6;
+            this.neck.y = -7.0f - $$15;
             this.neck.z = 5.5f;
-            boolean bl = n > 0;
-            this.neck.xRot = bl ? 0.21991149f : 0.0f;
-            this.mouth.xRot = (float)java.lang.Math.PI * (bl ? 0.05f : 0.01f);
-            if (bl) {
-                double d = (double)n / 40.0;
-                this.neck.x = (float)java.lang.Math.sin((double)(d * 10.0)) * 3.0f;
-            } else if (n2 > 0) {
-                float f7 = net.minecraft.util.Mth.sin((float)(((float)(20 - n2) - partialTick) / 20.0f * (float)java.lang.Math.PI * 0.25f));
-                this.mouth.xRot = 1.5707964f * f7;
+            boolean $$16 = $$4 > 0;
+            this.neck.xRot = $$16 ? 0.21991149f : 0.0f;
+            this.mouth.xRot = (float)java.lang.Math.PI * ($$16 ? 0.05f : 0.01f);
+            if ($$16) {
+                double $$17 = (double)$$4 / 40.0;
+                this.neck.x = (float)java.lang.Math.sin((double)($$17 * 10.0)) * 3.0f;
+            } else if ($$5 > 0) {
+                float $$18 = net.minecraft.util.Mth.sin((float)(((float)(20 - $$5) - $$3) / 20.0f * (float)java.lang.Math.PI * 0.25f));
+                this.mouth.xRot = 1.5707964f * $$18;
             }
         }
     }