--- net/minecraft/client/model/SalmonModel.java
+++ net/minecraft/client/model/SalmonModel.java
@@ -35,14 +35,14 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f = 1.0f;
-        float f2 = 1.0f;
-        if (!entity.isInWater()) {
-            f = 1.3f;
-            f2 = 1.7f;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        float $$6 = 1.0f;
+        float $$7 = 1.0f;
+        if (!$$0.isInWater()) {
+            $$6 = 1.3f;
+            $$7 = 1.7f;
         }
-        this.bodyBack.yRot = -f * 0.25f * net.minecraft.util.Mth.sin((float)(f2 * 0.6f * ageInTicks));
+        this.bodyBack.yRot = -$$6 * 0.25f * net.minecraft.util.Mth.sin((float)($$7 * 0.6f * $$3));
     }
 }
 