--- net/minecraft/client/model/SheepModel.java
+++ net/minecraft/client/model/SheepModel.java
@@ -20,14 +20,14 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)32);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        super.prepareMobModel(entity, limbSwing, limbSwingAmount, partialTick);
-        this.head.y = 6.0f + entity.getHeadEatPositionScale(partialTick) * 9.0f;
-        this.headXRot = entity.getHeadEatAngleScale(partialTick);
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        super.prepareMobModel($$0, $$1, $$2, $$3);
+        this.head.y = 6.0f + $$0.getHeadEatPositionScale($$3) * 9.0f;
+        this.headXRot = $$0.getHeadEatAngleScale($$3);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
         this.head.xRot = this.headXRot;
     }
 }