--- net/minecraft/client/model/ShulkerModel.java
+++ net/minecraft/client/model/ShulkerModel.java
@@ -28,18 +28,18 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)64);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f = ageInTicks - (float)((net.minecraft.world.entity.monster.Shulker)entity).tickCount;
-        float f2 = (0.5f + entity.getClientPeekAmount(f)) * (float)java.lang.Math.PI;
-        float f3 = -1.0f + net.minecraft.util.Mth.sin((float)f2);
-        float f4 = 0.0f;
-        if (f2 > (float)java.lang.Math.PI) {
-            f4 = net.minecraft.util.Mth.sin((float)(ageInTicks * 0.1f)) * 0.7f;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        float $$6 = $$3 - (float)((net.minecraft.world.entity.monster.Shulker)$$0).tickCount;
+        float $$7 = (0.5f + $$0.getClientPeekAmount($$6)) * (float)java.lang.Math.PI;
+        float $$8 = -1.0f + net.minecraft.util.Mth.sin((float)$$7);
+        float $$9 = 0.0f;
+        if ($$7 > (float)java.lang.Math.PI) {
+            $$9 = net.minecraft.util.Mth.sin((float)($$3 * 0.1f)) * 0.7f;
         }
-        this.lid.setPos(0.0f, 16.0f + net.minecraft.util.Mth.sin((float)f2) * 8.0f + f4, 0.0f);
-        this.lid.yRot = entity.getClientPeekAmount(f) > 0.3f ? f3 * f3 * f3 * f3 * (float)java.lang.Math.PI * 0.125f : 0.0f;
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        this.head.yRot = (((net.minecraft.world.entity.monster.Shulker)entity).yHeadRot - 180.0f - ((net.minecraft.world.entity.monster.Shulker)entity).yBodyRot) * ((float)java.lang.Math.PI / 180);
+        this.lid.setPos(0.0f, 16.0f + net.minecraft.util.Mth.sin((float)$$7) * 8.0f + $$9, 0.0f);
+        this.lid.yRot = $$0.getClientPeekAmount($$6) > 0.3f ? $$8 * $$8 * $$8 * $$8 * (float)java.lang.Math.PI * 0.125f : 0.0f;
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        this.head.yRot = (((net.minecraft.world.entity.monster.Shulker)$$0).yHeadRot - 180.0f - ((net.minecraft.world.entity.monster.Shulker)$$0).yBodyRot) * ((float)java.lang.Math.PI / 180);
     }
 
     public java.lang.Iterable<net.minecraft.client.model.geom.ModelPart> parts() {