--- net/minecraft/client/model/SilverfishModel.java
+++ net/minecraft/client/model/SilverfishModel.java
@@ -50,10 +50,10 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        for (int i = 0; i < this.bodyParts.length; ++i) {
-            this.bodyParts[i].yRot = net.minecraft.util.Mth.cos((float)(ageInTicks * 0.9f + (float)i * 0.15f * (float)java.lang.Math.PI)) * (float)java.lang.Math.PI * 0.05f * (float)(1 + java.lang.Math.abs((int)(i - 2)));
-            this.bodyParts[i].x = net.minecraft.util.Mth.sin((float)(ageInTicks * 0.9f + (float)i * 0.15f * (float)java.lang.Math.PI)) * (float)java.lang.Math.PI * 0.2f * (float)java.lang.Math.abs((int)(i - 2));
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        for (int $$6 = 0; $$6 < this.bodyParts.length; ++$$6) {
+            this.bodyParts[$$6].yRot = net.minecraft.util.Mth.cos((float)($$3 * 0.9f + (float)$$6 * 0.15f * (float)java.lang.Math.PI)) * (float)java.lang.Math.PI * 0.05f * (float)(1 + java.lang.Math.abs((int)($$6 - 2)));
+            this.bodyParts[$$6].x = net.minecraft.util.Mth.sin((float)($$3 * 0.9f + (float)$$6 * 0.15f * (float)java.lang.Math.PI)) * (float)java.lang.Math.PI * 0.2f * (float)java.lang.Math.abs((int)($$6 - 2));
         }
         this.bodyLayers[0].yRot = this.bodyParts[2].yRot;
         this.bodyLayers[1].yRot = this.bodyParts[4].yRot;