--- net/minecraft/client/model/SkullModel.java
+++ net/minecraft/client/model/SkullModel.java
@@ -38,8 +38,8 @@
         this.head.xRot = $$2 * ((float)java.lang.Math.PI / 180);
     }
 
-    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack matrixStack, com.mojang.blaze3d.vertex.VertexConsumer buffer, int packedLight, int packedOverlay, float red, float green, float blue, float alpha) {
-        this.root.render(matrixStack, buffer, packedLight, packedOverlay, red, green, blue, alpha);
+    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7) {
+        this.root.render($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7);
     }
 }
 