--- net/minecraft/client/model/SnowGolemModel.java
+++ net/minecraft/client/model/SnowGolemModel.java
@@ -35,18 +35,18 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)64);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        this.upperBody.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180) * 0.25f;
-        float f = net.minecraft.util.Mth.sin((float)this.upperBody.yRot);
-        float f2 = net.minecraft.util.Mth.cos((float)this.upperBody.yRot);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        this.upperBody.yRot = $$4 * ((float)java.lang.Math.PI / 180) * 0.25f;
+        float $$6 = net.minecraft.util.Mth.sin((float)this.upperBody.yRot);
+        float $$7 = net.minecraft.util.Mth.cos((float)this.upperBody.yRot);
         this.leftArm.yRot = this.upperBody.yRot;
         this.rightArm.yRot = this.upperBody.yRot + (float)java.lang.Math.PI;
-        this.leftArm.x = f2 * 5.0f;
-        this.leftArm.z = -f * 5.0f;
-        this.rightArm.x = -f2 * 5.0f;
-        this.rightArm.z = f * 5.0f;
+        this.leftArm.x = $$7 * 5.0f;
+        this.leftArm.z = -$$6 * 5.0f;
+        this.rightArm.x = -$$7 * 5.0f;
+        this.rightArm.z = $$6 * 5.0f;
     }
 
     public net.minecraft.client.model.geom.ModelPart root() {