--- net/minecraft/client/model/SpiderModel.java
+++ net/minecraft/client/model/SpiderModel.java
@@ -60,10 +60,10 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        float f = 0.7853982f;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        float $$6 = 0.7853982f;
         this.rightHindLeg.zRot = -0.7853982f;
         this.leftHindLeg.zRot = 0.7853982f;
         this.rightMiddleHindLeg.zRot = -0.58119464f;
@@ -72,8 +72,8 @@
         this.leftMiddleFrontLeg.zRot = 0.58119464f;
         this.rightFrontLeg.zRot = -0.7853982f;
         this.leftFrontLeg.zRot = 0.7853982f;
-        float f2 = -0.0f;
-        float f3 = 0.3926991f;
+        float $$7 = -0.0f;
+        float $$8 = 0.3926991f;
         this.rightHindLeg.yRot = 0.7853982f;
         this.leftHindLeg.yRot = -0.7853982f;
         this.rightMiddleHindLeg.yRot = 0.3926991f;
@@ -82,30 +82,30 @@
         this.leftMiddleFrontLeg.yRot = 0.3926991f;
         this.rightFrontLeg.yRot = -0.7853982f;
         this.leftFrontLeg.yRot = 0.7853982f;
-        float f4 = -(net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 2.0f + 0.0f)) * 0.4f) * limbSwingAmount;
-        float f5 = -(net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 2.0f + (float)java.lang.Math.PI)) * 0.4f) * limbSwingAmount;
-        float f6 = -(net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 2.0f + 1.5707964f)) * 0.4f) * limbSwingAmount;
-        float f7 = -(net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 2.0f + 4.712389f)) * 0.4f) * limbSwingAmount;
-        float f8 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)(limbSwing * 0.6662f + 0.0f)) * 0.4f)) * limbSwingAmount;
-        float f9 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * 0.4f)) * limbSwingAmount;
-        float f10 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)(limbSwing * 0.6662f + 1.5707964f)) * 0.4f)) * limbSwingAmount;
-        float f11 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)(limbSwing * 0.6662f + 4.712389f)) * 0.4f)) * limbSwingAmount;
-        this.rightHindLeg.yRot += f4;
-        this.leftHindLeg.yRot += -f4;
-        this.rightMiddleHindLeg.yRot += f5;
-        this.leftMiddleHindLeg.yRot += -f5;
-        this.rightMiddleFrontLeg.yRot += f6;
-        this.leftMiddleFrontLeg.yRot += -f6;
-        this.rightFrontLeg.yRot += f7;
-        this.leftFrontLeg.yRot += -f7;
-        this.rightHindLeg.zRot += f8;
-        this.leftHindLeg.zRot += -f8;
-        this.rightMiddleHindLeg.zRot += f9;
-        this.leftMiddleHindLeg.zRot += -f9;
-        this.rightMiddleFrontLeg.zRot += f10;
-        this.leftMiddleFrontLeg.zRot += -f10;
-        this.rightFrontLeg.zRot += f11;
-        this.leftFrontLeg.zRot += -f11;
+        float $$9 = -(net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 2.0f + 0.0f)) * 0.4f) * $$2;
+        float $$10 = -(net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 2.0f + (float)java.lang.Math.PI)) * 0.4f) * $$2;
+        float $$11 = -(net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 2.0f + 1.5707964f)) * 0.4f) * $$2;
+        float $$12 = -(net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 2.0f + 4.712389f)) * 0.4f) * $$2;
+        float $$13 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)($$1 * 0.6662f + 0.0f)) * 0.4f)) * $$2;
+        float $$14 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * 0.4f)) * $$2;
+        float $$15 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)($$1 * 0.6662f + 1.5707964f)) * 0.4f)) * $$2;
+        float $$16 = java.lang.Math.abs((float)(net.minecraft.util.Mth.sin((float)($$1 * 0.6662f + 4.712389f)) * 0.4f)) * $$2;
+        this.rightHindLeg.yRot += $$9;
+        this.leftHindLeg.yRot += -$$9;
+        this.rightMiddleHindLeg.yRot += $$10;
+        this.leftMiddleHindLeg.yRot += -$$10;
+        this.rightMiddleFrontLeg.yRot += $$11;
+        this.leftMiddleFrontLeg.yRot += -$$11;
+        this.rightFrontLeg.yRot += $$12;
+        this.leftFrontLeg.yRot += -$$12;
+        this.rightHindLeg.zRot += $$13;
+        this.leftHindLeg.zRot += -$$13;
+        this.rightMiddleHindLeg.zRot += $$14;
+        this.leftMiddleHindLeg.zRot += -$$14;
+        this.rightMiddleFrontLeg.zRot += $$15;
+        this.leftMiddleFrontLeg.zRot += -$$15;
+        this.rightFrontLeg.zRot += $$16;
+        this.leftFrontLeg.zRot += -$$16;
     }
 }
 