--- net/minecraft/client/model/StriderModel.java
+++ net/minecraft/client/model/StriderModel.java
@@ -51,46 +51,46 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)128);
     }
 
-    public void setupAnim(net.minecraft.world.entity.monster.Strider entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        limbSwingAmount = java.lang.Math.min((float)0.25f, (float)limbSwingAmount);
-        if (!entity.isVehicle()) {
-            this.body.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-            this.body.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
+    public void setupAnim(net.minecraft.world.entity.monster.Strider $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        $$2 = java.lang.Math.min((float)0.25f, (float)$$2);
+        if (!$$0.isVehicle()) {
+            this.body.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+            this.body.yRot = $$4 * ((float)java.lang.Math.PI / 180);
         } else {
             this.body.xRot = 0.0f;
             this.body.yRot = 0.0f;
         }
-        float f = 1.5f;
-        this.body.zRot = 0.1f * net.minecraft.util.Mth.sin((float)(limbSwing * 1.5f)) * 4.0f * limbSwingAmount;
+        float $$6 = 1.5f;
+        this.body.zRot = 0.1f * net.minecraft.util.Mth.sin((float)($$1 * 1.5f)) * 4.0f * $$2;
         this.body.y = 2.0f;
-        this.body.y -= 2.0f * net.minecraft.util.Mth.cos((float)(limbSwing * 1.5f)) * 2.0f * limbSwingAmount;
-        this.leftLeg.xRot = net.minecraft.util.Mth.sin((float)(limbSwing * 1.5f * 0.5f)) * 2.0f * limbSwingAmount;
-        this.rightLeg.xRot = net.minecraft.util.Mth.sin((float)(limbSwing * 1.5f * 0.5f + (float)java.lang.Math.PI)) * 2.0f * limbSwingAmount;
-        this.leftLeg.zRot = 0.17453292f * net.minecraft.util.Mth.cos((float)(limbSwing * 1.5f * 0.5f)) * limbSwingAmount;
-        this.rightLeg.zRot = 0.17453292f * net.minecraft.util.Mth.cos((float)(limbSwing * 1.5f * 0.5f + (float)java.lang.Math.PI)) * limbSwingAmount;
-        this.leftLeg.y = 8.0f + 2.0f * net.minecraft.util.Mth.sin((float)(limbSwing * 1.5f * 0.5f + (float)java.lang.Math.PI)) * 2.0f * limbSwingAmount;
-        this.rightLeg.y = 8.0f + 2.0f * net.minecraft.util.Mth.sin((float)(limbSwing * 1.5f * 0.5f)) * 2.0f * limbSwingAmount;
+        this.body.y -= 2.0f * net.minecraft.util.Mth.cos((float)($$1 * 1.5f)) * 2.0f * $$2;
+        this.leftLeg.xRot = net.minecraft.util.Mth.sin((float)($$1 * 1.5f * 0.5f)) * 2.0f * $$2;
+        this.rightLeg.xRot = net.minecraft.util.Mth.sin((float)($$1 * 1.5f * 0.5f + (float)java.lang.Math.PI)) * 2.0f * $$2;
+        this.leftLeg.zRot = 0.17453292f * net.minecraft.util.Mth.cos((float)($$1 * 1.5f * 0.5f)) * $$2;
+        this.rightLeg.zRot = 0.17453292f * net.minecraft.util.Mth.cos((float)($$1 * 1.5f * 0.5f + (float)java.lang.Math.PI)) * $$2;
+        this.leftLeg.y = 8.0f + 2.0f * net.minecraft.util.Mth.sin((float)($$1 * 1.5f * 0.5f + (float)java.lang.Math.PI)) * 2.0f * $$2;
+        this.rightLeg.y = 8.0f + 2.0f * net.minecraft.util.Mth.sin((float)($$1 * 1.5f * 0.5f)) * 2.0f * $$2;
         this.rightBottomBristle.zRot = -1.2217305f;
         this.rightMiddleBristle.zRot = -1.134464f;
         this.rightTopBristle.zRot = -0.87266463f;
         this.leftTopBristle.zRot = 0.87266463f;
         this.leftMiddleBristle.zRot = 1.134464f;
         this.leftBottomBristle.zRot = 1.2217305f;
-        float f2 = net.minecraft.util.Mth.cos((float)(limbSwing * 1.5f + (float)java.lang.Math.PI)) * limbSwingAmount;
-        this.rightBottomBristle.zRot += f2 * 1.3f;
-        this.rightMiddleBristle.zRot += f2 * 1.2f;
-        this.rightTopBristle.zRot += f2 * 0.6f;
-        this.leftTopBristle.zRot += f2 * 0.6f;
-        this.leftMiddleBristle.zRot += f2 * 1.2f;
-        this.leftBottomBristle.zRot += f2 * 1.3f;
-        float f3 = 1.0f;
-        float f4 = 1.0f;
-        this.rightBottomBristle.zRot += 0.05f * net.minecraft.util.Mth.sin((float)(ageInTicks * 1.0f * -0.4f));
-        this.rightMiddleBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)(ageInTicks * 1.0f * 0.2f));
-        this.rightTopBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)(ageInTicks * 1.0f * 0.4f));
-        this.leftTopBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)(ageInTicks * 1.0f * 0.4f));
-        this.leftMiddleBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)(ageInTicks * 1.0f * 0.2f));
-        this.leftBottomBristle.zRot += 0.05f * net.minecraft.util.Mth.sin((float)(ageInTicks * 1.0f * -0.4f));
+        float $$7 = net.minecraft.util.Mth.cos((float)($$1 * 1.5f + (float)java.lang.Math.PI)) * $$2;
+        this.rightBottomBristle.zRot += $$7 * 1.3f;
+        this.rightMiddleBristle.zRot += $$7 * 1.2f;
+        this.rightTopBristle.zRot += $$7 * 0.6f;
+        this.leftTopBristle.zRot += $$7 * 0.6f;
+        this.leftMiddleBristle.zRot += $$7 * 1.2f;
+        this.leftBottomBristle.zRot += $$7 * 1.3f;
+        float $$8 = 1.0f;
+        float $$9 = 1.0f;
+        this.rightBottomBristle.zRot += 0.05f * net.minecraft.util.Mth.sin((float)($$3 * 1.0f * -0.4f));
+        this.rightMiddleBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)($$3 * 1.0f * 0.2f));
+        this.rightTopBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)($$3 * 1.0f * 0.4f));
+        this.leftTopBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)($$3 * 1.0f * 0.4f));
+        this.leftMiddleBristle.zRot += 0.1f * net.minecraft.util.Mth.sin((float)($$3 * 1.0f * 0.2f));
+        this.leftBottomBristle.zRot += 0.05f * net.minecraft.util.Mth.sin((float)($$3 * 1.0f * -0.4f));
     }
 
     public net.minecraft.client.model.geom.ModelPart root() {