--- net/minecraft/client/model/TropicalFishModelA.java
+++ net/minecraft/client/model/TropicalFishModelA.java
@@ -30,12 +30,12 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f = 1.0f;
-        if (!entity.isInWater()) {
-            f = 1.5f;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        float $$6 = 1.0f;
+        if (!$$0.isInWater()) {
+            $$6 = 1.5f;
         }
-        this.tail.yRot = -f * 0.45f * net.minecraft.util.Mth.sin((float)(0.6f * ageInTicks));
+        this.tail.yRot = -$$6 * 0.45f * net.minecraft.util.Mth.sin((float)(0.6f * $$3));
     }
 }
 