--- net/minecraft/client/model/TurtleModel.java
+++ net/minecraft/client/model/TurtleModel.java
@@ -32,43 +32,43 @@
         return com.google.common.collect.Iterables.concat((java.lang.Iterable)super.bodyParts(), (java.lang.Iterable)com.google.common.collect.ImmutableList.of((java.lang.Object)this.eggBelly));
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        this.rightHindLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 0.6f)) * 0.5f * limbSwingAmount;
-        this.leftHindLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 0.6f + (float)java.lang.Math.PI)) * 0.5f * limbSwingAmount;
-        this.rightFrontLeg.zRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 0.6f + (float)java.lang.Math.PI)) * 0.5f * limbSwingAmount;
-        this.leftFrontLeg.zRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f * 0.6f)) * 0.5f * limbSwingAmount;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
+        this.rightHindLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 0.6f)) * 0.5f * $$2;
+        this.leftHindLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 0.6f + (float)java.lang.Math.PI)) * 0.5f * $$2;
+        this.rightFrontLeg.zRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 0.6f + (float)java.lang.Math.PI)) * 0.5f * $$2;
+        this.leftFrontLeg.zRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f * 0.6f)) * 0.5f * $$2;
         this.rightFrontLeg.xRot = 0.0f;
         this.leftFrontLeg.xRot = 0.0f;
         this.rightFrontLeg.yRot = 0.0f;
         this.leftFrontLeg.yRot = 0.0f;
         this.rightHindLeg.yRot = 0.0f;
         this.leftHindLeg.yRot = 0.0f;
-        if (!entity.isInWater() && entity.isOnGround()) {
-            float f = entity.isLayingEgg() ? 4.0f : 1.0f;
-            float f2 = entity.isLayingEgg() ? 2.0f : 1.0f;
-            float f3 = 5.0f;
-            this.rightFrontLeg.yRot = net.minecraft.util.Mth.cos((float)(f * limbSwing * 5.0f + (float)java.lang.Math.PI)) * 8.0f * limbSwingAmount * f2;
+        if (!$$0.isInWater() && $$0.isOnGround()) {
+            float $$6 = $$0.isLayingEgg() ? 4.0f : 1.0f;
+            float $$7 = $$0.isLayingEgg() ? 2.0f : 1.0f;
+            float $$8 = 5.0f;
+            this.rightFrontLeg.yRot = net.minecraft.util.Mth.cos((float)($$6 * $$1 * 5.0f + (float)java.lang.Math.PI)) * 8.0f * $$2 * $$7;
             this.rightFrontLeg.zRot = 0.0f;
-            this.leftFrontLeg.yRot = net.minecraft.util.Mth.cos((float)(f * limbSwing * 5.0f)) * 8.0f * limbSwingAmount * f2;
+            this.leftFrontLeg.yRot = net.minecraft.util.Mth.cos((float)($$6 * $$1 * 5.0f)) * 8.0f * $$2 * $$7;
             this.leftFrontLeg.zRot = 0.0f;
-            this.rightHindLeg.yRot = net.minecraft.util.Mth.cos((float)(limbSwing * 5.0f + (float)java.lang.Math.PI)) * 3.0f * limbSwingAmount;
+            this.rightHindLeg.yRot = net.minecraft.util.Mth.cos((float)($$1 * 5.0f + (float)java.lang.Math.PI)) * 3.0f * $$2;
             this.rightHindLeg.xRot = 0.0f;
-            this.leftHindLeg.yRot = net.minecraft.util.Mth.cos((float)(limbSwing * 5.0f)) * 3.0f * limbSwingAmount;
+            this.leftHindLeg.yRot = net.minecraft.util.Mth.cos((float)($$1 * 5.0f)) * 3.0f * $$2;
             this.leftHindLeg.xRot = 0.0f;
         }
-        this.eggBelly.visible = !this.young && entity.hasEgg();
+        this.eggBelly.visible = !this.young && $$0.hasEgg();
     }
 
-    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack matrixStack, com.mojang.blaze3d.vertex.VertexConsumer buffer, int packedLight, int packedOverlay, float red, float green, float blue, float alpha) {
-        boolean bl = this.eggBelly.visible;
-        if (bl) {
-            matrixStack.pushPose();
-            matrixStack.translate(0.0, (double)-0.08f, 0.0);
+    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7) {
+        boolean $$8 = this.eggBelly.visible;
+        if ($$8) {
+            $$0.pushPose();
+            $$0.translate(0.0, (double)-0.08f, 0.0);
         }
-        super.renderToBuffer(matrixStack, buffer, packedLight, packedOverlay, red, green, blue, alpha);
-        if (bl) {
-            matrixStack.popPose();
+        super.renderToBuffer($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7);
+        if ($$8) {
+            $$0.popPose();
         }
     }
 }