--- net/minecraft/client/model/VexModel.java
+++ net/minecraft/client/model/VexModel.java
@@ -30,13 +30,13 @@
         return com.google.common.collect.Iterables.concat((java.lang.Iterable)super.bodyParts(), (java.lang.Iterable)com.google.common.collect.ImmutableList.of((java.lang.Object)this.rightWing, (java.lang.Object)this.leftWing));
     }
 
-    public void setupAnim(net.minecraft.world.entity.monster.Vex entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim((net.minecraft.world.entity.LivingEntity)entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        if (entity.isCharging()) {
-            if (entity.getMainHandItem().isEmpty()) {
+    public void setupAnim(net.minecraft.world.entity.monster.Vex $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4, $$5);
+        if ($$0.isCharging()) {
+            if ($$0.getMainHandItem().isEmpty()) {
                 this.rightArm.xRot = 4.712389f;
                 this.leftArm.xRot = 4.712389f;
-            } else if (entity.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT) {
+            } else if ($$0.getMainArm() == net.minecraft.world.entity.HumanoidArm.RIGHT) {
                 this.rightArm.xRot = 3.7699115f;
             } else {
                 this.leftArm.xRot = 3.7699115f;
@@ -47,7 +47,7 @@
         this.leftWing.z = 2.0f;
         this.rightWing.y = 1.0f;
         this.leftWing.y = 1.0f;
-        this.rightWing.yRot = 0.47123894f + net.minecraft.util.Mth.cos((float)(ageInTicks * 45.836624f * ((float)java.lang.Math.PI / 180))) * (float)java.lang.Math.PI * 0.05f;
+        this.rightWing.yRot = 0.47123894f + net.minecraft.util.Mth.cos((float)($$3 * 45.836624f * ((float)java.lang.Math.PI / 180))) * (float)java.lang.Math.PI * 0.05f;
         this.leftWing.yRot = -this.rightWing.yRot;
         this.leftWing.zRot = -0.47123894f;
         this.leftWing.xRot = 0.47123894f;