--- net/minecraft/client/model/VillagerModel.java
+++ net/minecraft/client/model/VillagerModel.java
@@ -46,21 +46,21 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        boolean bl = false;
-        if (entity instanceof net.minecraft.world.entity.npc.AbstractVillager) {
-            bl = ((net.minecraft.world.entity.npc.AbstractVillager)entity).getUnhappyCounter() > 0;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        boolean $$6 = false;
+        if ($$0 instanceof net.minecraft.world.entity.npc.AbstractVillager) {
+            $$6 = ((net.minecraft.world.entity.npc.AbstractVillager)$$0).getUnhappyCounter() > 0;
         }
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        if (bl) {
-            this.head.zRot = 0.3f * net.minecraft.util.Mth.sin((float)(0.45f * ageInTicks));
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        if ($$6) {
+            this.head.zRot = 0.3f * net.minecraft.util.Mth.sin((float)(0.45f * $$3));
             this.head.xRot = 0.4f;
         } else {
             this.head.zRot = 0.0f;
         }
-        this.rightLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * 1.4f * limbSwingAmount * 0.5f;
-        this.leftLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * limbSwingAmount * 0.5f;
+        this.rightLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * 1.4f * $$2 * 0.5f;
+        this.leftLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * $$2 * 0.5f;
         this.rightLeg.yRot = 0.0f;
         this.leftLeg.yRot = 0.0f;
     }