--- net/minecraft/client/model/WitherBossModel.java
+++ net/minecraft/client/model/WitherBossModel.java
@@ -46,18 +46,18 @@
         return this.root;
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        float f = net.minecraft.util.Mth.cos((float)(ageInTicks * 0.1f));
-        this.ribcage.xRot = (0.065f + 0.05f * f) * (float)java.lang.Math.PI;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        float $$6 = net.minecraft.util.Mth.cos((float)($$3 * 0.1f));
+        this.ribcage.xRot = (0.065f + 0.05f * $$6) * (float)java.lang.Math.PI;
         this.tail.setPos(-2.0f, 6.9f + net.minecraft.util.Mth.cos((float)this.ribcage.xRot) * 10.0f, -0.5f + net.minecraft.util.Mth.sin((float)this.ribcage.xRot) * 10.0f);
-        this.tail.xRot = (0.265f + 0.1f * f) * (float)java.lang.Math.PI;
-        this.centerHead.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.centerHead.xRot = headPitch * ((float)java.lang.Math.PI / 180);
+        this.tail.xRot = (0.265f + 0.1f * $$6) * (float)java.lang.Math.PI;
+        this.centerHead.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.centerHead.xRot = $$5 * ((float)java.lang.Math.PI / 180);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        net.minecraft.client.model.WitherBossModel.setupHeadRotation(entity, (net.minecraft.client.model.geom.ModelPart)this.rightHead, (int)0);
-        net.minecraft.client.model.WitherBossModel.setupHeadRotation(entity, (net.minecraft.client.model.geom.ModelPart)this.leftHead, (int)1);
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        net.minecraft.client.model.WitherBossModel.setupHeadRotation($$0, (net.minecraft.client.model.geom.ModelPart)this.rightHead, (int)0);
+        net.minecraft.client.model.WitherBossModel.setupHeadRotation($$0, (net.minecraft.client.model.geom.ModelPart)this.leftHead, (int)1);
     }
 
     public static <T extends net.minecraft.world.entity.boss.wither.WitherBoss> void setupHeadRotation(T $$0, net.minecraft.client.model.geom.ModelPart $$1, int $$2) {