--- net/minecraft/client/model/WolfModel.java
+++ net/minecraft/client/model/WolfModel.java
@@ -60,9 +60,9 @@
         return com.google.common.collect.ImmutableList.of((java.lang.Object)this.body, (java.lang.Object)this.rightHindLeg, (java.lang.Object)this.leftHindLeg, (java.lang.Object)this.rightFrontLeg, (java.lang.Object)this.leftFrontLeg, (java.lang.Object)this.tail, (java.lang.Object)this.upperBody);
     }
 
-    public void prepareMobModel(T entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        this.tail.yRot = entity.isAngry() ? 0.0f : net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * 1.4f * limbSwingAmount;
-        if (entity.isInSittingPose()) {
+    public void prepareMobModel(T $$0, float $$1, float $$2, float $$3) {
+        this.tail.yRot = $$0.isAngry() ? 0.0f : net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * 1.4f * $$2;
+        if ($$0.isInSittingPose()) {
             this.upperBody.setPos(-1.0f, 16.0f, -3.0f);
             this.upperBody.xRot = 1.2566371f;
             this.upperBody.yRot = 0.0f;
@@ -87,21 +87,21 @@
             this.leftHindLeg.setPos(0.5f, 16.0f, 7.0f);
             this.rightFrontLeg.setPos(-2.5f, 16.0f, -4.0f);
             this.leftFrontLeg.setPos(0.5f, 16.0f, -4.0f);
-            this.rightHindLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * 1.4f * limbSwingAmount;
-            this.leftHindLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * limbSwingAmount;
-            this.rightFrontLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * limbSwingAmount;
-            this.leftFrontLeg.xRot = net.minecraft.util.Mth.cos((float)(limbSwing * 0.6662f)) * 1.4f * limbSwingAmount;
+            this.rightHindLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * 1.4f * $$2;
+            this.leftHindLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * $$2;
+            this.rightFrontLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f + (float)java.lang.Math.PI)) * 1.4f * $$2;
+            this.leftFrontLeg.xRot = net.minecraft.util.Mth.cos((float)($$1 * 0.6662f)) * 1.4f * $$2;
         }
-        this.realHead.zRot = entity.getHeadRollAngle(partialTick) + entity.getBodyRollAngle(partialTick, 0.0f);
-        this.upperBody.zRot = entity.getBodyRollAngle(partialTick, -0.08f);
-        this.body.zRot = entity.getBodyRollAngle(partialTick, -0.16f);
-        this.realTail.zRot = entity.getBodyRollAngle(partialTick, -0.2f);
+        this.realHead.zRot = $$0.getHeadRollAngle($$3) + $$0.getBodyRollAngle($$3, 0.0f);
+        this.upperBody.zRot = $$0.getBodyRollAngle($$3, -0.08f);
+        this.body.zRot = $$0.getBodyRollAngle($$3, -0.16f);
+        this.realTail.zRot = $$0.getBodyRollAngle($$3, -0.2f);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        this.head.xRot = headPitch * ((float)java.lang.Math.PI / 180);
-        this.head.yRot = netHeadYaw * ((float)java.lang.Math.PI / 180);
-        this.tail.xRot = ageInTicks;
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        this.head.xRot = $$5 * ((float)java.lang.Math.PI / 180);
+        this.head.yRot = $$4 * ((float)java.lang.Math.PI / 180);
+        this.tail.xRot = $$3;
     }
 }
 