--- net/minecraft/client/model/ZombieVillagerModel.java
+++ net/minecraft/client/model/ZombieVillagerModel.java
@@ -39,9 +39,9 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$1, (int)64, (int)32);
     }
 
-    public void setupAnim(T entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
-        super.setupAnim(entity, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch);
-        net.minecraft.client.model.AnimationUtils.animateZombieArms((net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.rightArm, (boolean)entity.isAggressive(), (float)this.attackTime, (float)ageInTicks);
+    public void setupAnim(T $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
+        super.setupAnim($$0, $$1, $$2, $$3, $$4, $$5);
+        net.minecraft.client.model.AnimationUtils.animateZombieArms((net.minecraft.client.model.geom.ModelPart)this.leftArm, (net.minecraft.client.model.geom.ModelPart)this.rightArm, (boolean)$$0.isAggressive(), (float)this.attackTime, (float)$$3);
     }
 
     public void hatVisible(boolean $$0) {