--- net/minecraft/world/flag/FeatureFlagUniverse.java
+++ net/minecraft/world/flag/FeatureFlagUniverse.java
@@ -1,0 +1,18 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.flag;
+
+public class net.minecraft.world.flag.FeatureFlagUniverse
+ {
+    public java.lang.String id;
+
+    public net.minecraft.world.flag.FeatureFlagUniverse(java.lang.String $$0) {
+        this.id = $$0;
+    }
+
+    public java.lang.String toString() {
+        return this.id;
+    }
+}
+