--- net/minecraft/world/flag/FeatureFlags.java
+++ net/minecraft/world/flag/FeatureFlags.java
@@ -1,0 +1,43 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.flag;
+
+public class net.minecraft.world.flag.FeatureFlags
+ {
+    public static net.minecraft.world.flag.FeatureFlag VANILLA;
+    public static net.minecraft.world.flag.FeatureFlag BUNDLE;
+    public static net.minecraft.world.flag.FeatureFlag UPDATE_1_20;
+    public static net.minecraft.world.flag.FeatureFlagRegistry REGISTRY;
+    public static com.mojang.serialization.Codec<net.minecraft.world.flag.FeatureFlagSet> CODEC;
+    public static net.minecraft.world.flag.FeatureFlagSet VANILLA_SET;
+    public static net.minecraft.world.flag.FeatureFlagSet DEFAULT_FLAGS;
+
+    public net.minecraft.world.flag.FeatureFlags() {
+    }
+
+    public static java.lang.String printMissingFlags(net.minecraft.world.flag.FeatureFlagSet $$0, net.minecraft.world.flag.FeatureFlagSet $$1) {
+        return net.minecraft.world.flag.FeatureFlags.printMissingFlags((net.minecraft.world.flag.FeatureFlagRegistry)REGISTRY, (net.minecraft.world.flag.FeatureFlagSet)$$0, (net.minecraft.world.flag.FeatureFlagSet)$$1);
+    }
+
+    public static java.lang.String printMissingFlags(net.minecraft.world.flag.FeatureFlagRegistry $$0, net.minecraft.world.flag.FeatureFlagSet $$12, net.minecraft.world.flag.FeatureFlagSet $$2) {
+        java.util.Set $$3 = $$0.toNames($$2);
+        java.util.Set $$4 = $$0.toNames($$12);
+        return (java.lang.String)$$3.stream().filter($$1 -> !$$4.contains($$1)).map(net.minecraft.resources.ResourceLocation::toString).collect(java.util.stream.Collectors.joining((java.lang.CharSequence)", "));
+    }
+
+    public static boolean isExperimental(net.minecraft.world.flag.FeatureFlagSet $$0) {
+        return !$$0.isSubsetOf(VANILLA_SET);
+    }
+
+    static {
+        net.minecraft.world.flag.FeatureFlagRegistry$Builder $$0 = new net.minecraft.world.flag.FeatureFlagRegistry$Builder("main");
+        VANILLA = $$0.createVanilla("vanilla");
+        BUNDLE = $$0.createVanilla("bundle");
+        UPDATE_1_20 = $$0.createVanilla("update_1_20");
+        REGISTRY = $$0.build();
+        CODEC = REGISTRY.codec();
+        DEFAULT_FLAGS = VANILLA_SET = net.minecraft.world.flag.FeatureFlagSet.of((net.minecraft.world.flag.FeatureFlag)VANILLA);
+    }
+}
+