--- net/minecraft/world/flag/FeatureFlag.java
+++ net/minecraft/world/flag/FeatureFlag.java
@@ -1,0 +1,16 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.flag;
+
+public class net.minecraft.world.flag.FeatureFlag
+ {
+    public net.minecraft.world.flag.FeatureFlagUniverse universe;
+    public long mask;
+
+    public net.minecraft.world.flag.FeatureFlag(net.minecraft.world.flag.FeatureFlagUniverse $$0, int $$1) {
+        this.universe = $$0;
+        this.mask = 1L << $$1;
+    }
+}
+