--- net/minecraft/world/flag/FeatureFlagRegistry$Builder.java
+++ net/minecraft/world/flag/FeatureFlagRegistry$Builder.java
@@ -1,0 +1,38 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.flag;
+
+public class net.minecraft.world.flag.FeatureFlagRegistry$Builder
+ {
+    public net.minecraft.world.flag.FeatureFlagUniverse universe;
+    public int id;
+    public java.util.Map<net.minecraft.resources.ResourceLocation, net.minecraft.world.flag.FeatureFlag> flags;
+
+    public net.minecraft.world.flag.FeatureFlagRegistry$Builder(java.lang.String $$0) {
+        this.flags = new java.util.LinkedHashMap();
+        this.universe = new net.minecraft.world.flag.FeatureFlagUniverse($$0);
+    }
+
+    public net.minecraft.world.flag.FeatureFlag createVanilla(java.lang.String $$0) {
+        return this.create(new net.minecraft.resources.ResourceLocation("minecraft", $$0));
+    }
+
+    public net.minecraft.world.flag.FeatureFlag create(net.minecraft.resources.ResourceLocation $$0) {
+        net.minecraft.world.flag.FeatureFlag $$1;
+        net.minecraft.world.flag.FeatureFlag $$2;
+        if (this.id >= 64) {
+            throw new java.lang.IllegalStateException("Too many feature flags");
+        }
+        if (($$2 = (net.minecraft.world.flag.FeatureFlag)this.flags.put((java.lang.Object)$$0, (java.lang.Object)($$1 = new net.minecraft.world.flag.FeatureFlag(this.universe, this.id++)))) != null) {
+            throw new java.lang.IllegalStateException("Duplicate feature flag " + $$0);
+        }
+        return $$1;
+    }
+
+    public net.minecraft.world.flag.FeatureFlagRegistry build() {
+        net.minecraft.world.flag.FeatureFlagSet $$0 = net.minecraft.world.flag.FeatureFlagSet.create((net.minecraft.world.flag.FeatureFlagUniverse)this.universe, (java.util.Collection)this.flags.values());
+        return new net.minecraft.world.flag.FeatureFlagRegistry(this.universe, $$0, java.util.Map.copyOf((java.util.Map)this.flags));
+    }
+}
+