--- net/minecraft/server/packs/BuiltInMetadata.java
+++ net/minecraft/server/packs/BuiltInMetadata.java
@@ -1,0 +1,35 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs;
+
+public class net.minecraft.server.packs.BuiltInMetadata
+ {
+    public static net.minecraft.server.packs.BuiltInMetadata EMPTY;
+    public java.util.Map<net.minecraft.server.packs.metadata.MetadataSectionSerializer<?>, ?> values;
+
+    public net.minecraft.server.packs.BuiltInMetadata(java.util.Map<net.minecraft.server.packs.metadata.MetadataSectionSerializer<?>, ?> $$0) {
+        this.values = $$0;
+    }
+
+    public <T> T get(net.minecraft.server.packs.metadata.MetadataSectionSerializer<T> $$0) {
+        return (T)this.values.get($$0);
+    }
+
+    public static net.minecraft.server.packs.BuiltInMetadata of() {
+        return EMPTY;
+    }
+
+    public static <T> net.minecraft.server.packs.BuiltInMetadata of(net.minecraft.server.packs.metadata.MetadataSectionSerializer<T> $$0, T $$1) {
+        return new net.minecraft.server.packs.BuiltInMetadata(java.util.Map.of($$0, $$1));
+    }
+
+    public static <T1, T2> net.minecraft.server.packs.BuiltInMetadata of(net.minecraft.server.packs.metadata.MetadataSectionSerializer<T1> $$0, T1 $$1, net.minecraft.server.packs.metadata.MetadataSectionSerializer<T2> $$2, T2 $$3) {
+        return new net.minecraft.server.packs.BuiltInMetadata(java.util.Map.of($$0, $$1, $$2, $$3));
+    }
+
+    static {
+        EMPTY = new net.minecraft.server.packs.BuiltInMetadata(java.util.Map.of());
+    }
+}
+