--- net/minecraft/server/packs/FeatureFlagsMetadataSection.java
+++ net/minecraft/server/packs/FeatureFlagsMetadataSection.java
@@ -1,0 +1,42 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs;
+
+public class net.minecraft.server.packs.FeatureFlagsMetadataSection
+extends java.lang.Record
+ {
+    public net.minecraft.world.flag.FeatureFlagSet flags;
+    public static com.mojang.serialization.Codec<net.minecraft.server.packs.FeatureFlagsMetadataSection> CODEC;
+    public static net.minecraft.server.packs.metadata.MetadataSectionType<net.minecraft.server.packs.FeatureFlagsMetadataSection> TYPE;
+
+    public net.minecraft.server.packs.FeatureFlagsMetadataSection(net.minecraft.world.flag.FeatureFlagSet $$0) {
+        this.flags = $$0;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.FeatureFlagsMetadataSection.class, "flags", "flags"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.FeatureFlagsMetadataSection.class, "flags", "flags"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.FeatureFlagsMetadataSection.class, "flags", "flags"}, this, $$0);
+    }
+
+    public net.minecraft.world.flag.FeatureFlagSet flags() {
+        return this.flags;
+    }
+
+    public static /* synthetic */ com.mojang.datafixers.kinds.App lambda$static$0(com.mojang.serialization.codecs.RecordCodecBuilder$Instance $$0) {
+        return $$0.group((com.mojang.datafixers.kinds.App)net.minecraft.world.flag.FeatureFlags.CODEC.fieldOf("enabled").forGetter(net.minecraft.server.packs.FeatureFlagsMetadataSection::flags)).apply((com.mojang.datafixers.kinds.Applicative)$$0, net.minecraft.server.packs.FeatureFlagsMetadataSection::new);
+    }
+
+    static {
+        CODEC = com.mojang.serialization.codecs.RecordCodecBuilder.create($$0 -> $$0.group((com.mojang.datafixers.kinds.App)net.minecraft.world.flag.FeatureFlags.CODEC.fieldOf("enabled").forGetter(net.minecraft.server.packs.FeatureFlagsMetadataSection::flags)).apply((com.mojang.datafixers.kinds.Applicative)$$0, net.minecraft.server.packs.FeatureFlagsMetadataSection::new));
+        TYPE = net.minecraft.server.packs.metadata.MetadataSectionType.fromCodec((java.lang.String)"features", (com.mojang.serialization.Codec)CODEC);
+    }
+}
+