--- net/minecraft/world/level/levelgen/carver/ConfiguredWorldCarver.java
+++ net/minecraft/world/level/levelgen/carver/ConfiguredWorldCarver.java
@@ -12,9 +12,9 @@
     public static com.mojang.serialization.Codec<net.minecraft.core.Holder<net.minecraft.world.level.levelgen.carver.ConfiguredWorldCarver<?>>> CODEC;
     public static com.mojang.serialization.Codec<net.minecraft.core.HolderSet<net.minecraft.world.level.levelgen.carver.ConfiguredWorldCarver<?>>> LIST_CODEC;
 
-    public net.minecraft.world.level.levelgen.carver.ConfiguredWorldCarver(net.minecraft.world.level.levelgen.carver.WorldCarver<WC> worldCarver, WC config) {
-        this.worldCarver = worldCarver;
-        this.config = config;
+    public net.minecraft.world.level.levelgen.carver.ConfiguredWorldCarver(net.minecraft.world.level.levelgen.carver.WorldCarver<WC> $$0, WC $$1) {
+        this.worldCarver = $$0;
+        this.config = $$1;
     }
 
     public boolean isStartChunk(net.minecraft.util.RandomSource $$0) {