--- net/minecraft/network/protocol/handshake/ClientIntentionPacket.java
+++ net/minecraft/network/protocol/handshake/ClientIntentionPacket.java
@@ -12,11 +12,11 @@
     public int port;
     public net.minecraft.network.ConnectionProtocol intention;
 
-    public net.minecraft.network.protocol.handshake.ClientIntentionPacket(java.lang.String hostName, int port, net.minecraft.network.ConnectionProtocol intention) {
+    public net.minecraft.network.protocol.handshake.ClientIntentionPacket(java.lang.String $$0, int $$1, net.minecraft.network.ConnectionProtocol $$2) {
         this.protocolVersion = net.minecraft.SharedConstants.getCurrentVersion().getProtocolVersion();
-        this.hostName = hostName;
-        this.port = port;
-        this.intention = intention;
+        this.hostName = $$0;
+        this.port = $$1;
+        this.intention = $$2;
     }
 
     public net.minecraft.network.protocol.handshake.ClientIntentionPacket(net.minecraft.network.FriendlyByteBuf $$0) {
@@ -26,15 +26,15 @@
         this.intention = net.minecraft.network.ConnectionProtocol.getById((int)$$0.readVarInt());
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buffer) {
-        buffer.writeVarInt(this.protocolVersion);
-        buffer.writeUtf(this.hostName);
-        buffer.writeShort(this.port);
-        buffer.writeVarInt(this.intention.getId());
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
+        $$0.writeVarInt(this.protocolVersion);
+        $$0.writeUtf(this.hostName);
+        $$0.writeShort(this.port);
+        $$0.writeVarInt(this.intention.getId());
     }
 
-    public void handle(net.minecraft.network.protocol.handshake.ServerHandshakePacketListener handler) {
-        handler.handleIntention(this);
+    public void handle(net.minecraft.network.protocol.handshake.ServerHandshakePacketListener $$0) {
+        $$0.handleIntention(this);
     }
 
     public net.minecraft.network.ConnectionProtocol getIntention() {