--- net/minecraft/client/renderer/texture/OverlayTexture.java
+++ net/minecraft/client/renderer/texture/OverlayTexture.java
@@ -40,20 +40,20 @@
         com.mojang.blaze3d.systems.RenderSystem.setupOverlayColor(() -> ((net.minecraft.client.renderer.texture.DynamicTexture)this.texture).getId(), (int)16);
     }
 
-    public static int u(float u) {
-        return (int)(u * 15.0f);
+    public static int u(float $$0) {
+        return (int)($$0 * 15.0f);
     }
 
-    public static int v(boolean hurt) {
-        return hurt ? 3 : 10;
+    public static int v(boolean $$0) {
+        return $$0 ? 3 : 10;
     }
 
-    public static int pack(int u, int v) {
-        return u | v << 16;
+    public static int pack(int $$0, int $$1) {
+        return $$0 | $$1 << 16;
     }
 
-    public static int pack(float u, boolean hurt) {
-        return net.minecraft.client.renderer.texture.OverlayTexture.pack((int)net.minecraft.client.renderer.texture.OverlayTexture.u((float)u), (int)net.minecraft.client.renderer.texture.OverlayTexture.v((boolean)hurt));
+    public static int pack(float $$0, boolean $$1) {
+        return net.minecraft.client.renderer.texture.OverlayTexture.pack((int)net.minecraft.client.renderer.texture.OverlayTexture.u((float)$$0), (int)net.minecraft.client.renderer.texture.OverlayTexture.v((boolean)$$1));
     }
 
     public void teardownOverlayColor() {