--- net/minecraft/client/renderer/texture/SimpleTexture$TextureImage.java
+++ net/minecraft/client/renderer/texture/SimpleTexture$TextureImage.java
@@ -13,37 +13,40 @@
     @javax.annotation.Nullable
     public java.io.IOException exception;
 
-    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(java.io.IOException exception) {
-        this.exception = exception;
+    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(java.io.IOException $$0) {
+        this.exception = $$0;
         this.metadata = null;
         this.image = null;
     }
 
-    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(@javax.annotation.Nullable net.minecraft.client.resources.metadata.texture.TextureMetadataSection metadata, com.mojang.blaze3d.platform.NativeImage image) {
+    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(@javax.annotation.Nullable net.minecraft.client.resources.metadata.texture.TextureMetadataSection $$0, com.mojang.blaze3d.platform.NativeImage $$1) {
         this.exception = null;
-        this.metadata = metadata;
-        this.image = image;
+        this.metadata = $$0;
+        this.image = $$1;
     }
 
-    public static net.minecraft.client.renderer.texture.SimpleTexture$TextureImage load(net.minecraft.server.packs.resources.ResourceManager resourceManager, net.minecraft.resources.ResourceLocation location) {
+    /*
+     * WARNING - void declaration
+     */
+    public static net.minecraft.client.renderer.texture.SimpleTexture$TextureImage load(net.minecraft.server.packs.resources.ResourceManager $$0, net.minecraft.resources.ResourceLocation $$1) {
         try {
-            com.mojang.blaze3d.platform.NativeImage nativeImage;
-            net.minecraft.server.packs.resources.Resource resource = resourceManager.getResourceOrThrow(location);
-            try (java.io.InputStream inputStream = resource.open();
+            void $$5;
+            net.minecraft.server.packs.resources.Resource $$2 = $$0.getResourceOrThrow($$1);
+            try (java.io.InputStream $$3 = $$2.open();
 ){
-                nativeImage = com.mojang.blaze3d.platform.NativeImage.read((java.io.InputStream)inputStream);
+                com.mojang.blaze3d.platform.NativeImage $$4 = com.mojang.blaze3d.platform.NativeImage.read((java.io.InputStream)$$3);
             }
-            inputStream = null;
+            net.minecraft.client.resources.metadata.texture.TextureMetadataSection $$6 = null;
             try {
-                inputStream = (net.minecraft.client.resources.metadata.texture.TextureMetadataSection)resource.metadata().getSection((net.minecraft.server.packs.metadata.MetadataSectionSerializer)net.minecraft.client.resources.metadata.texture.TextureMetadataSection.SERIALIZER).orElse(null);
+                $$6 = (net.minecraft.client.resources.metadata.texture.TextureMetadataSection)$$2.metadata().getSection((net.minecraft.server.packs.metadata.MetadataSectionSerializer)net.minecraft.client.resources.metadata.texture.TextureMetadataSection.SERIALIZER).orElse(null);
             }
-            catch (java.lang.RuntimeException runtimeException) {
-                net.minecraft.client.renderer.texture.SimpleTexture.LOGGER.warn("Failed reading metadata of: {}", (java.lang.Object)location, (java.lang.Object)((java.lang.Object)runtimeException));
+            catch (java.lang.RuntimeException $$7) {
+                net.minecraft.client.renderer.texture.SimpleTexture.LOGGER.warn("Failed reading metadata of: {}", (java.lang.Object)$$1, (java.lang.Object)((java.lang.Object)$$7));
             }
-            return new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage((net.minecraft.client.resources.metadata.texture.TextureMetadataSection)inputStream, nativeImage);
+            return new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage($$6, (com.mojang.blaze3d.platform.NativeImage)$$5);
         }
-        catch (java.io.IOException iOException) {
-            return new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage(iOException);
+        catch (java.io.IOException $$8) {
+            return new net.minecraft.client.renderer.texture.SimpleTexture$TextureImage($$8);
         }
     }
 