--- net/minecraft/client/renderer/texture/SimpleTexture.java
+++ net/minecraft/client/renderer/texture/SimpleTexture.java
@@ -9,38 +9,38 @@
     public static org.slf4j.Logger LOGGER;
     public net.minecraft.resources.ResourceLocation location;
 
-    public net.minecraft.client.renderer.texture.SimpleTexture(net.minecraft.resources.ResourceLocation location) {
-        this.location = location;
+    public net.minecraft.client.renderer.texture.SimpleTexture(net.minecraft.resources.ResourceLocation $$0) {
+        this.location = $$0;
     }
 
-    public void load(net.minecraft.server.packs.resources.ResourceManager manager) throws java.io.IOException {
-        boolean bl;
-        boolean bl2;
-        net.minecraft.client.renderer.texture.SimpleTexture$TextureImage textureImage = this.getTextureImage(manager);
-        textureImage.throwIfError();
-        net.minecraft.client.resources.metadata.texture.TextureMetadataSection textureMetadataSection = textureImage.getTextureMetadata();
-        if (textureMetadataSection != null) {
-            bl2 = textureMetadataSection.isBlur();
-            bl = textureMetadataSection.isClamp();
+    public void load(net.minecraft.server.packs.resources.ResourceManager $$0) throws java.io.IOException {
+        boolean $$6;
+        boolean $$5;
+        net.minecraft.client.renderer.texture.SimpleTexture$TextureImage $$1 = this.getTextureImage($$0);
+        $$1.throwIfError();
+        net.minecraft.client.resources.metadata.texture.TextureMetadataSection $$2 = $$1.getTextureMetadata();
+        if ($$2 != null) {
+            boolean $$3 = $$2.isBlur();
+            boolean $$4 = $$2.isClamp();
         } else {
-            bl2 = false;
-            bl = false;
+            $$5 = false;
+            $$6 = false;
         }
-        com.mojang.blaze3d.platform.NativeImage nativeImage = textureImage.getImage();
+        com.mojang.blaze3d.platform.NativeImage $$7 = $$1.getImage();
         if (!com.mojang.blaze3d.systems.RenderSystem.isOnRenderThreadOrInit()) {
-            com.mojang.blaze3d.systems.RenderSystem.recordRenderCall(() -> this.doLoad(nativeImage, bl2, bl));
+            com.mojang.blaze3d.systems.RenderSystem.recordRenderCall(() -> this.doLoad($$7, $$5, $$6));
         } else {
-            this.doLoad(nativeImage, bl2, bl);
+            this.doLoad($$7, $$5, $$6);
         }
     }
 
-    public void doLoad(com.mojang.blaze3d.platform.NativeImage image, boolean blur, boolean clamp) {
-        com.mojang.blaze3d.platform.TextureUtil.prepareImage((int)this.getId(), (int)0, (int)image.getWidth(), (int)image.getHeight());
-        image.upload(0, 0, 0, 0, 0, image.getWidth(), image.getHeight(), blur, clamp, false, true);
+    public void doLoad(com.mojang.blaze3d.platform.NativeImage $$0, boolean $$1, boolean $$2) {
+        com.mojang.blaze3d.platform.TextureUtil.prepareImage((int)this.getId(), (int)0, (int)$$0.getWidth(), (int)$$0.getHeight());
+        $$0.upload(0, 0, 0, 0, 0, $$0.getWidth(), $$0.getHeight(), $$1, $$2, false, true);
     }
 
-    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage getTextureImage(net.minecraft.server.packs.resources.ResourceManager resourceManager) {
-        return net.minecraft.client.renderer.texture.SimpleTexture$TextureImage.load((net.minecraft.server.packs.resources.ResourceManager)resourceManager, (net.minecraft.resources.ResourceLocation)this.location);
+    public net.minecraft.client.renderer.texture.SimpleTexture$TextureImage getTextureImage(net.minecraft.server.packs.resources.ResourceManager $$0) {
+        return net.minecraft.client.renderer.texture.SimpleTexture$TextureImage.load((net.minecraft.server.packs.resources.ResourceManager)$$0, (net.minecraft.resources.ResourceLocation)this.location);
     }
 
     static {