--- net/minecraft/client/renderer/texture/SpriteContents$FrameInfo.java
+++ net/minecraft/client/renderer/texture/SpriteContents$FrameInfo.java
@@ -1,0 +1,16 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.renderer.texture;
+
+public class net.minecraft.client.renderer.texture.SpriteContents$FrameInfo
+ {
+    public int index;
+    public int time;
+
+    public net.minecraft.client.renderer.texture.SpriteContents$FrameInfo(int $$0, int $$1) {
+        this.index = $$0;
+        this.time = $$1;
+    }
+}
+