--- net/minecraft/client/renderer/texture/TextureAtlasSprite$1.java
+++ net/minecraft/client/renderer/texture/TextureAtlasSprite$1.java
@@ -1,0 +1,25 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.renderer.texture;
+
+public class net.minecraft.client.renderer.texture.TextureAtlasSprite$1
+implements net.minecraft.client.renderer.texture.TextureAtlasSprite$Ticker
+ {
+    public /* synthetic */ net.minecraft.client.renderer.texture.SpriteTicker val$ticker;
+    public /* synthetic */ net.minecraft.client.renderer.texture.TextureAtlasSprite this$0;
+
+    public net.minecraft.client.renderer.texture.TextureAtlasSprite$1() {
+        this.this$0 = $$0;
+        this.val$ticker = spriteTicker;
+    }
+
+    public void tickAndUpload() {
+        this.val$ticker.tickAndUpload(this.this$0.x, this.this$0.y);
+    }
+
+    public void close() {
+        this.val$ticker.close();
+    }
+}
+