--- net/minecraft/client/renderer/texture/TextureAtlasSprite$FrameInfo.java
+++ net/minecraft/client/renderer/texture/TextureAtlasSprite$FrameInfo.java
@@ -1,16 +1,0 @@
-/*
- * Decompiled with CFR 0.151.
- */
-package net.minecraft.client.renderer.texture;
-
-public class net.minecraft.client.renderer.texture.TextureAtlasSprite$FrameInfo
- {
-    public int index;
-    public int time;
-
-    public net.minecraft.client.renderer.texture.TextureAtlasSprite$FrameInfo(int $$0, int $$1) {
-        this.index = $$0;
-        this.time = $$1;
-    }
-}
-