--- net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
+++ net/minecraft/world/item/enchantment/ArrowDamageEnchantment.java
@@ -6,16 +6,16 @@
 public class net.minecraft.world.item.enchantment.ArrowDamageEnchantment
 extends net.minecraft.world.item.enchantment.Enchantment
  {
-    public net.minecraft.world.item.enchantment.ArrowDamageEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.BOW, applicableSlots);
+    public net.minecraft.world.item.enchantment.ArrowDamageEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.entity.EquipmentSlot ... $$1) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.BOW, $$1);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 1 + (enchantmentLevel - 1) * 10;
+    public int getMinCost(int $$0) {
+        return 1 + ($$0 - 1) * 10;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
-        return this.getMinCost(enchantmentLevel) + 15;
+    public int getMaxCost(int $$0) {
+        return this.getMinCost($$0) + 15;
     }
 
     public int getMaxLevel() {