--- net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
+++ net/minecraft/world/item/enchantment/ArrowPiercingEnchantment.java
@@ -6,15 +6,15 @@
 public class net.minecraft.world.item.enchantment.ArrowPiercingEnchantment
 extends net.minecraft.world.item.enchantment.Enchantment
  {
-    public net.minecraft.world.item.enchantment.ArrowPiercingEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.CROSSBOW, applicableSlots);
+    public net.minecraft.world.item.enchantment.ArrowPiercingEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.entity.EquipmentSlot ... $$1) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.CROSSBOW, $$1);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 1 + (enchantmentLevel - 1) * 10;
+    public int getMinCost(int $$0) {
+        return 1 + ($$0 - 1) * 10;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
+    public int getMaxCost(int $$0) {
         return 50;
     }
 
@@ -22,8 +22,8 @@
         return 4;
     }
 
-    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment ench) {
-        return super.checkCompatibility(ench) && ench != net.minecraft.world.item.enchantment.Enchantments.MULTISHOT;
+    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment $$0) {
+        return super.checkCompatibility($$0) && $$0 != net.minecraft.world.item.enchantment.Enchantments.MULTISHOT;
     }
 }
 