--- net/minecraft/world/item/enchantment/DamageEnchantment.java
+++ net/minecraft/world/item/enchantment/DamageEnchantment.java
@@ -15,53 +15,53 @@
     public static int[] LEVEL_COST_SPAN;
     public int type;
 
-    public net.minecraft.world.item.enchantment.DamageEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, int type, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.WEAPON, applicableSlots);
-        this.type = type;
+    public net.minecraft.world.item.enchantment.DamageEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, int $$1, net.minecraft.world.entity.EquipmentSlot ... $$2) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.WEAPON, $$2);
+        this.type = $$1;
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return MIN_COST[this.type] + (enchantmentLevel - 1) * LEVEL_COST[this.type];
+    public int getMinCost(int $$0) {
+        return MIN_COST[this.type] + ($$0 - 1) * LEVEL_COST[this.type];
     }
 
-    public int getMaxCost(int enchantmentLevel) {
-        return this.getMinCost(enchantmentLevel) + LEVEL_COST_SPAN[this.type];
+    public int getMaxCost(int $$0) {
+        return this.getMinCost($$0) + LEVEL_COST_SPAN[this.type];
     }
 
     public int getMaxLevel() {
         return 5;
     }
 
-    public float getDamageBonus(int level, net.minecraft.world.entity.MobType creatureType) {
+    public float getDamageBonus(int $$0, net.minecraft.world.entity.MobType $$1) {
         if (this.type == 0) {
-            return 1.0f + (float)java.lang.Math.max((int)0, (int)(level - 1)) * 0.5f;
+            return 1.0f + (float)java.lang.Math.max((int)0, (int)($$0 - 1)) * 0.5f;
         }
-        if (this.type == 1 && creatureType == net.minecraft.world.entity.MobType.UNDEAD) {
-            return (float)level * 2.5f;
+        if (this.type == 1 && $$1 == net.minecraft.world.entity.MobType.UNDEAD) {
+            return (float)$$0 * 2.5f;
         }
-        if (this.type == 2 && creatureType == net.minecraft.world.entity.MobType.ARTHROPOD) {
-            return (float)level * 2.5f;
+        if (this.type == 2 && $$1 == net.minecraft.world.entity.MobType.ARTHROPOD) {
+            return (float)$$0 * 2.5f;
         }
         return 0.0f;
     }
 
-    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment ench) {
-        return !(ench instanceof net.minecraft.world.item.enchantment.DamageEnchantment);
+    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment $$0) {
+        return !($$0 instanceof net.minecraft.world.item.enchantment.DamageEnchantment);
     }
 
-    public boolean canEnchant(net.minecraft.world.item.ItemStack stack) {
-        if (stack.getItem() instanceof net.minecraft.world.item.AxeItem) {
+    public boolean canEnchant(net.minecraft.world.item.ItemStack $$0) {
+        if ($$0.getItem() instanceof net.minecraft.world.item.AxeItem) {
             return true;
         }
-        return super.canEnchant(stack);
+        return super.canEnchant($$0);
     }
 
-    public void doPostAttack(net.minecraft.world.entity.LivingEntity user, net.minecraft.world.entity.Entity target, int level) {
-        if (target instanceof net.minecraft.world.entity.LivingEntity) {
-            net.minecraft.world.entity.LivingEntity livingEntity = (net.minecraft.world.entity.LivingEntity)target;
-            if (this.type == 2 && level > 0 && livingEntity.getMobType() == net.minecraft.world.entity.MobType.ARTHROPOD) {
-                int n = 20 + user.getRandom().nextInt(10 * level);
-                livingEntity.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.MOVEMENT_SLOWDOWN, n, 3));
+    public void doPostAttack(net.minecraft.world.entity.LivingEntity $$0, net.minecraft.world.entity.Entity $$1, int $$2) {
+        if ($$1 instanceof net.minecraft.world.entity.LivingEntity) {
+            net.minecraft.world.entity.LivingEntity $$3 = (net.minecraft.world.entity.LivingEntity)$$1;
+            if (this.type == 2 && $$2 > 0 && $$3.getMobType() == net.minecraft.world.entity.MobType.ARTHROPOD) {
+                int $$4 = 20 + $$0.getRandom().nextInt(10 * $$2);
+                $$3.addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.MOVEMENT_SLOWDOWN, $$4, 3));
             }
         }
     }