--- net/minecraft/world/item/enchantment/EnchantmentInstance.java
+++ net/minecraft/world/item/enchantment/EnchantmentInstance.java
@@ -9,10 +9,10 @@
     public net.minecraft.world.item.enchantment.Enchantment enchantment;
     public int level;
 
-    public net.minecraft.world.item.enchantment.EnchantmentInstance(net.minecraft.world.item.enchantment.Enchantment enchantment, int level) {
-        super(enchantment.getRarity().getWeight());
-        this.enchantment = enchantment;
-        this.level = level;
+    public net.minecraft.world.item.enchantment.EnchantmentInstance(net.minecraft.world.item.enchantment.Enchantment $$0, int $$1) {
+        super($$0.getRarity().getWeight());
+        this.enchantment = $$0;
+        this.level = $$1;
     }
 }
 