--- net/minecraft/world/item/enchantment/EnchantmentCategory$4.java
+++ net/minecraft/world/item/enchantment/EnchantmentCategory$4.java
@@ -10,8 +10,8 @@
         super(string, n);
     }
 
-    public boolean canEnchant(net.minecraft.world.item.Item item) {
-        return item instanceof net.minecraft.world.item.Wearable || net.minecraft.world.level.block.Block.byItem((net.minecraft.world.item.Item)item) instanceof net.minecraft.world.item.Wearable;
+    public boolean canEnchant(net.minecraft.world.item.Item $$0) {
+        return $$0 instanceof net.minecraft.world.item.Wearable || net.minecraft.world.level.block.Block.byItem((net.minecraft.world.item.Item)$$0) instanceof net.minecraft.world.item.Wearable;
     }
 }
 