--- net/minecraft/world/item/enchantment/Enchantment$Rarity.java
+++ net/minecraft/world/item/enchantment/Enchantment$Rarity.java
@@ -21,9 +21,9 @@
         return (net.minecraft.world.item.enchantment.Enchantment$Rarity)java.lang.Enum.valueOf(net.minecraft.world.item.enchantment.Enchantment$Rarity.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.item.enchantment.Enchantment$Rarity(int weight) {
+    public net.minecraft.world.item.enchantment.Enchantment$Rarity(int $$0) {
         super(string, n);
-        this.weight = weight;
+        this.weight = $$0;
     }
 
     public int getWeight() {