--- net/minecraft/world/item/enchantment/Enchantments.java
+++ net/minecraft/world/item/enchantment/Enchantments.java
@@ -49,8 +49,8 @@
     public net.minecraft.world.item.enchantment.Enchantments() {
     }
 
-    public static net.minecraft.world.item.enchantment.Enchantment register(java.lang.String identifier, net.minecraft.world.item.enchantment.Enchantment enchantment) {
-        return (net.minecraft.world.item.enchantment.Enchantment)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.ENCHANTMENT, (java.lang.String)identifier, (java.lang.Object)enchantment);
+    public static net.minecraft.world.item.enchantment.Enchantment register(java.lang.String $$0, net.minecraft.world.item.enchantment.Enchantment $$1) {
+        return (net.minecraft.world.item.enchantment.Enchantment)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.ENCHANTMENT, (java.lang.String)$$0, (java.lang.Object)$$1);
     }
 
     static {