--- net/minecraft/world/item/enchantment/EnchantmentCategory$10.java
+++ net/minecraft/world/item/enchantment/EnchantmentCategory$10.java
@@ -10,8 +10,8 @@
         super(string, n);
     }
 
-    public boolean canEnchant(net.minecraft.world.item.Item item) {
-        return item instanceof net.minecraft.world.item.ArmorItem && ((net.minecraft.world.item.ArmorItem)item).getSlot() == net.minecraft.world.entity.EquipmentSlot.HEAD;
+    public boolean canEnchant(net.minecraft.world.item.Item $$0) {
+        return $$0 instanceof net.minecraft.world.item.ArmorItem && ((net.minecraft.world.item.ArmorItem)$$0).getSlot() == net.minecraft.world.entity.EquipmentSlot.HEAD;
     }
 }
 