--- net/minecraft/world/item/enchantment/FireAspectEnchantment.java
+++ net/minecraft/world/item/enchantment/FireAspectEnchantment.java
@@ -6,16 +6,16 @@
 public class net.minecraft.world.item.enchantment.FireAspectEnchantment
 extends net.minecraft.world.item.enchantment.Enchantment
  {
-    public net.minecraft.world.item.enchantment.FireAspectEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.WEAPON, applicableSlots);
+    public net.minecraft.world.item.enchantment.FireAspectEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.entity.EquipmentSlot ... $$1) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.WEAPON, $$1);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 10 + 20 * (enchantmentLevel - 1);
+    public int getMinCost(int $$0) {
+        return 10 + 20 * ($$0 - 1);
     }
 
-    public int getMaxCost(int enchantmentLevel) {
-        return super.getMinCost(enchantmentLevel) + 50;
+    public int getMaxCost(int $$0) {
+        return super.getMinCost($$0) + 50;
     }
 
     public int getMaxLevel() {