--- net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
+++ net/minecraft/world/item/enchantment/FishingSpeedEnchantment.java
@@ -10,12 +10,12 @@
         super($$0, $$1, $$2);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 15 + (enchantmentLevel - 1) * 9;
+    public int getMinCost(int $$0) {
+        return 15 + ($$0 - 1) * 9;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
-        return super.getMinCost(enchantmentLevel) + 50;
+    public int getMaxCost(int $$0) {
+        return super.getMinCost($$0) + 50;
     }
 
     public int getMaxLevel() {