--- net/minecraft/world/item/enchantment/LootBonusEnchantment.java
+++ net/minecraft/world/item/enchantment/LootBonusEnchantment.java
@@ -10,20 +10,20 @@
         super($$0, $$1, $$2);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 15 + (enchantmentLevel - 1) * 9;
+    public int getMinCost(int $$0) {
+        return 15 + ($$0 - 1) * 9;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
-        return super.getMinCost(enchantmentLevel) + 50;
+    public int getMaxCost(int $$0) {
+        return super.getMinCost($$0) + 50;
     }
 
     public int getMaxLevel() {
         return 3;
     }
 
-    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment ench) {
-        return super.checkCompatibility(ench) && ench != net.minecraft.world.item.enchantment.Enchantments.SILK_TOUCH;
+    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment $$0) {
+        return super.checkCompatibility($$0) && $$0 != net.minecraft.world.item.enchantment.Enchantments.SILK_TOUCH;
     }
 }
 