--- net/minecraft/world/item/enchantment/OxygenEnchantment.java
+++ net/minecraft/world/item/enchantment/OxygenEnchantment.java
@@ -6,16 +6,16 @@
 public class net.minecraft.world.item.enchantment.OxygenEnchantment
 extends net.minecraft.world.item.enchantment.Enchantment
  {
-    public net.minecraft.world.item.enchantment.OxygenEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.ARMOR_HEAD, applicableSlots);
+    public net.minecraft.world.item.enchantment.OxygenEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.entity.EquipmentSlot ... $$1) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.ARMOR_HEAD, $$1);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 10 * enchantmentLevel;
+    public int getMinCost(int $$0) {
+        return 10 * $$0;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
-        return this.getMinCost(enchantmentLevel) + 30;
+    public int getMaxCost(int $$0) {
+        return this.getMinCost($$0) + 30;
     }
 
     public int getMaxLevel() {