--- net/minecraft/world/item/enchantment/ProtectionEnchantment.java
+++ net/minecraft/world/item/enchantment/ProtectionEnchantment.java
@@ -8,70 +8,70 @@
  {
     public net.minecraft.world.item.enchantment.ProtectionEnchantment$Type type;
 
-    public net.minecraft.world.item.enchantment.ProtectionEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.item.enchantment.ProtectionEnchantment$Type type, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL ? net.minecraft.world.item.enchantment.EnchantmentCategory.ARMOR_FEET : net.minecraft.world.item.enchantment.EnchantmentCategory.ARMOR, applicableSlots);
-        this.type = type;
+    public net.minecraft.world.item.enchantment.ProtectionEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.item.enchantment.ProtectionEnchantment$Type $$1, net.minecraft.world.entity.EquipmentSlot ... $$2) {
+        super($$0, $$1 == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL ? net.minecraft.world.item.enchantment.EnchantmentCategory.ARMOR_FEET : net.minecraft.world.item.enchantment.EnchantmentCategory.ARMOR, $$2);
+        this.type = $$1;
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return this.type.getMinCost() + (enchantmentLevel - 1) * this.type.getLevelCost();
+    public int getMinCost(int $$0) {
+        return this.type.getMinCost() + ($$0 - 1) * this.type.getLevelCost();
     }
 
-    public int getMaxCost(int enchantmentLevel) {
-        return this.getMinCost(enchantmentLevel) + this.type.getLevelCost();
+    public int getMaxCost(int $$0) {
+        return this.getMinCost($$0) + this.type.getLevelCost();
     }
 
     public int getMaxLevel() {
         return 4;
     }
 
-    public int getDamageProtection(int level, net.minecraft.world.damagesource.DamageSource source) {
-        if (source.isBypassInvul()) {
+    public int getDamageProtection(int $$0, net.minecraft.world.damagesource.DamageSource $$1) {
+        if ($$1.isBypassInvul()) {
             return 0;
         }
         if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.ALL) {
-            return level;
+            return $$0;
         }
-        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FIRE && source.isFire()) {
-            return level * 2;
+        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FIRE && $$1.isFire()) {
+            return $$0 * 2;
         }
-        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL && source.isFall()) {
-            return level * 3;
+        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL && $$1.isFall()) {
+            return $$0 * 3;
         }
-        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.EXPLOSION && source.isExplosion()) {
-            return level * 2;
+        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.EXPLOSION && $$1.isExplosion()) {
+            return $$0 * 2;
         }
-        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.PROJECTILE && source.isProjectile()) {
-            return level * 2;
+        if (this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.PROJECTILE && $$1.isProjectile()) {
+            return $$0 * 2;
         }
         return 0;
     }
 
-    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment ench) {
-        if (ench instanceof net.minecraft.world.item.enchantment.ProtectionEnchantment) {
-            net.minecraft.world.item.enchantment.ProtectionEnchantment protectionEnchantment = (net.minecraft.world.item.enchantment.ProtectionEnchantment)ench;
-            if (this.type == protectionEnchantment.type) {
+    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment $$0) {
+        if ($$0 instanceof net.minecraft.world.item.enchantment.ProtectionEnchantment) {
+            net.minecraft.world.item.enchantment.ProtectionEnchantment $$1 = (net.minecraft.world.item.enchantment.ProtectionEnchantment)$$0;
+            if (this.type == $$1.type) {
                 return false;
             }
-            return this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL || protectionEnchantment.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL;
+            return this.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL || $$1.type == net.minecraft.world.item.enchantment.ProtectionEnchantment$Type.FALL;
         }
-        return super.checkCompatibility(ench);
+        return super.checkCompatibility($$0);
     }
 
-    public static int getFireAfterDampener(net.minecraft.world.entity.LivingEntity livingEntity, int level) {
-        int n = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.FIRE_PROTECTION, (net.minecraft.world.entity.LivingEntity)livingEntity);
-        if (n > 0) {
-            level -= net.minecraft.util.Mth.floor((float)((float)level * ((float)n * 0.15f)));
+    public static int getFireAfterDampener(net.minecraft.world.entity.LivingEntity $$0, int $$1) {
+        int $$2 = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.FIRE_PROTECTION, (net.minecraft.world.entity.LivingEntity)$$0);
+        if ($$2 > 0) {
+            $$1 -= net.minecraft.util.Mth.floor((float)((float)$$1 * ((float)$$2 * 0.15f)));
         }
-        return level;
+        return $$1;
     }
 
-    public static double getExplosionKnockbackAfterDampener(net.minecraft.world.entity.LivingEntity livingEntity, double damage) {
-        int n = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.BLAST_PROTECTION, (net.minecraft.world.entity.LivingEntity)livingEntity);
-        if (n > 0) {
-            damage -= (double)net.minecraft.util.Mth.floor((double)(damage * (double)((float)n * 0.15f)));
+    public static double getExplosionKnockbackAfterDampener(net.minecraft.world.entity.LivingEntity $$0, double $$1) {
+        int $$2 = net.minecraft.world.item.enchantment.EnchantmentHelper.getEnchantmentLevel((net.minecraft.world.item.enchantment.Enchantment)net.minecraft.world.item.enchantment.Enchantments.BLAST_PROTECTION, (net.minecraft.world.entity.LivingEntity)$$0);
+        if ($$2 > 0) {
+            $$1 -= (double)net.minecraft.util.Mth.floor((double)($$1 * (double)((float)$$2 * 0.15f)));
         }
-        return damage;
+        return $$1;
     }
 }
 