--- net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
+++ net/minecraft/world/item/enchantment/TridentRiptideEnchantment.java
@@ -6,15 +6,15 @@
 public class net.minecraft.world.item.enchantment.TridentRiptideEnchantment
 extends net.minecraft.world.item.enchantment.Enchantment
  {
-    public net.minecraft.world.item.enchantment.TridentRiptideEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.TRIDENT, applicableSlots);
+    public net.minecraft.world.item.enchantment.TridentRiptideEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.entity.EquipmentSlot ... $$1) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.TRIDENT, $$1);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 10 + enchantmentLevel * 7;
+    public int getMinCost(int $$0) {
+        return 10 + $$0 * 7;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
+    public int getMaxCost(int $$0) {
         return 50;
     }
 
@@ -22,8 +22,8 @@
         return 3;
     }
 
-    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment ench) {
-        return super.checkCompatibility(ench) && ench != net.minecraft.world.item.enchantment.Enchantments.LOYALTY && ench != net.minecraft.world.item.enchantment.Enchantments.CHANNELING;
+    public boolean checkCompatibility(net.minecraft.world.item.enchantment.Enchantment $$0) {
+        return super.checkCompatibility($$0) && $$0 != net.minecraft.world.item.enchantment.Enchantments.LOYALTY && $$0 != net.minecraft.world.item.enchantment.Enchantments.CHANNELING;
     }
 }
 