--- net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
+++ net/minecraft/world/item/enchantment/TridentLoyaltyEnchantment.java
@@ -6,15 +6,15 @@
 public class net.minecraft.world.item.enchantment.TridentLoyaltyEnchantment
 extends net.minecraft.world.item.enchantment.Enchantment
  {
-    public net.minecraft.world.item.enchantment.TridentLoyaltyEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.TRIDENT, applicableSlots);
+    public net.minecraft.world.item.enchantment.TridentLoyaltyEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.entity.EquipmentSlot ... $$1) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.TRIDENT, $$1);
     }
 
-    public int getMinCost(int enchantmentLevel) {
-        return 5 + enchantmentLevel * 7;
+    public int getMinCost(int $$0) {
+        return 5 + $$0 * 7;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
+    public int getMaxCost(int $$0) {
         return 50;
     }
 