--- net/minecraft/world/item/enchantment/VanishingCurseEnchantment.java
+++ net/minecraft/world/item/enchantment/VanishingCurseEnchantment.java
@@ -6,15 +6,15 @@
 public class net.minecraft.world.item.enchantment.VanishingCurseEnchantment
 extends net.minecraft.world.item.enchantment.Enchantment
  {
-    public net.minecraft.world.item.enchantment.VanishingCurseEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity rarity, net.minecraft.world.entity.EquipmentSlot ... applicableSlots) {
-        super(rarity, net.minecraft.world.item.enchantment.EnchantmentCategory.VANISHABLE, applicableSlots);
+    public net.minecraft.world.item.enchantment.VanishingCurseEnchantment(net.minecraft.world.item.enchantment.Enchantment$Rarity $$0, net.minecraft.world.entity.EquipmentSlot ... $$1) {
+        super($$0, net.minecraft.world.item.enchantment.EnchantmentCategory.VANISHABLE, $$1);
     }
 
-    public int getMinCost(int enchantmentLevel) {
+    public int getMinCost(int $$0) {
         return 25;
     }
 
-    public int getMaxCost(int enchantmentLevel) {
+    public int getMaxCost(int $$0) {
         return 50;
     }
 