--- net/minecraft/world/scores/criteria/ObjectiveCriteria$RenderType.java
+++ net/minecraft/world/scores/criteria/ObjectiveCriteria$RenderType.java
@@ -20,17 +20,17 @@
         return (net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType)java.lang.Enum.valueOf(net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType(java.lang.String id) {
+    public net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType(java.lang.String $$0) {
         super(string, n);
-        this.id = id;
+        this.id = $$0;
     }
 
     public java.lang.String getId() {
         return this.id;
     }
 
-    public static net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType byId(java.lang.String renderType) {
-        return (net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType)BY_ID.getOrDefault((java.lang.Object)renderType, (java.lang.Object)INTEGER);
+    public static net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType byId(java.lang.String $$0) {
+        return (net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType)BY_ID.getOrDefault((java.lang.Object)$$0, (java.lang.Object)INTEGER);
     }
 
     public static /* synthetic */ net.minecraft.world.scores.criteria.ObjectiveCriteria$RenderType[] $values() {