--- net/minecraft/server/packs/repository/Pack$Info.java
+++ net/minecraft/server/packs/repository/Pack$Info.java
@@ -1,0 +1,47 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.repository;
+
+public class net.minecraft.server.packs.repository.Pack$Info
+extends java.lang.Record
+ {
+    public net.minecraft.network.chat.Component description;
+    public int format;
+    public net.minecraft.world.flag.FeatureFlagSet requestedFeatures;
+
+    public net.minecraft.server.packs.repository.Pack$Info(net.minecraft.network.chat.Component $$0, int $$1, net.minecraft.world.flag.FeatureFlagSet $$2) {
+        this.description = $$0;
+        this.format = $$1;
+        this.requestedFeatures = $$2;
+    }
+
+    public net.minecraft.server.packs.repository.PackCompatibility compatibility(net.minecraft.server.packs.PackType $$0) {
+        return net.minecraft.server.packs.repository.PackCompatibility.forFormat((int)this.format, (net.minecraft.server.packs.PackType)$$0);
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.repository.Pack$Info.class, "description;format;requestedFeatures", "description", "format", "requestedFeatures"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.repository.Pack$Info.class, "description;format;requestedFeatures", "description", "format", "requestedFeatures"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.server.packs.repository.Pack$Info.class, "description;format;requestedFeatures", "description", "format", "requestedFeatures"}, this, $$0);
+    }
+
+    public net.minecraft.network.chat.Component description() {
+        return this.description;
+    }
+
+    public int format() {
+        return this.format;
+    }
+
+    public net.minecraft.world.flag.FeatureFlagSet requestedFeatures() {
+        return this.requestedFeatures;
+    }
+}
+