--- net/minecraft/server/packs/repository/PackCompatibility.java
+++ net/minecraft/server/packs/repository/PackCompatibility.java
@@ -21,10 +21,10 @@
         return (net.minecraft.server.packs.repository.PackCompatibility)java.lang.Enum.valueOf(net.minecraft.server.packs.repository.PackCompatibility.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.server.packs.repository.PackCompatibility(java.lang.String type) {
+    public net.minecraft.server.packs.repository.PackCompatibility(java.lang.String $$0) {
         super(string, n);
-        this.description = net.minecraft.network.chat.Component.translatable((java.lang.String)("pack.incompatible." + type)).withStyle(net.minecraft.ChatFormatting.GRAY);
-        this.confirmation = net.minecraft.network.chat.Component.translatable((java.lang.String)("pack.incompatible.confirm." + type));
+        this.description = net.minecraft.network.chat.Component.translatable((java.lang.String)("pack.incompatible." + $$0)).withStyle(net.minecraft.ChatFormatting.GRAY);
+        this.confirmation = net.minecraft.network.chat.Component.translatable((java.lang.String)("pack.incompatible.confirm." + $$0));
     }
 
     public boolean isCompatible() {
@@ -42,10 +42,6 @@
         return COMPATIBLE;
     }
 
-    public static net.minecraft.server.packs.repository.PackCompatibility forMetadata(net.minecraft.server.packs.metadata.pack.PackMetadataSection $$0, net.minecraft.server.packs.PackType $$1) {
-        return net.minecraft.server.packs.repository.PackCompatibility.forFormat((int)$$0.getPackFormat(), (net.minecraft.server.packs.PackType)$$1);
-    }
-
     public net.minecraft.network.chat.Component getDescription() {
         return this.description;
     }