--- net/minecraft/server/packs/repository/PackSource$1.java
+++ net/minecraft/server/packs/repository/PackSource$1.java
@@ -1,0 +1,25 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.server.packs.repository;
+
+public class net.minecraft.server.packs.repository.PackSource$1
+implements net.minecraft.server.packs.repository.PackSource
+ {
+    public /* synthetic */ java.util.function.UnaryOperator val$decorator;
+    public /* synthetic */ boolean val$addAutomatically;
+
+    public net.minecraft.server.packs.repository.PackSource$1() {
+        this.val$decorator = unaryOperator;
+        this.val$addAutomatically = bl;
+    }
+
+    public net.minecraft.network.chat.Component decorate(net.minecraft.network.chat.Component $$0) {
+        return (net.minecraft.network.chat.Component)this.val$decorator.apply((java.lang.Object)$$0);
+    }
+
+    public boolean shouldAddAutomatically() {
+        return this.val$addAutomatically;
+    }
+}
+