--- net/minecraft/sounds/Music.java
+++ net/minecraft/sounds/Music.java
@@ -11,11 +11,11 @@
     public int maxDelay;
     public boolean replaceCurrentMusic;
 
-    public net.minecraft.sounds.Music(net.minecraft.sounds.SoundEvent event, int minDelay, int maxDelay, boolean replaceCurrentMusic) {
-        this.event = event;
-        this.minDelay = minDelay;
-        this.maxDelay = maxDelay;
-        this.replaceCurrentMusic = replaceCurrentMusic;
+    public net.minecraft.sounds.Music(net.minecraft.sounds.SoundEvent $$0, int $$1, int $$2, boolean $$3) {
+        this.event = $$0;
+        this.minDelay = $$1;
+        this.maxDelay = $$2;
+        this.replaceCurrentMusic = $$3;
     }
 
     public net.minecraft.sounds.SoundEvent getEvent() {