--- net/minecraft/sounds/SoundEvent.java
+++ net/minecraft/sounds/SoundEvent.java
@@ -10,8 +10,8 @@
     public float range;
     public boolean newSystem;
 
-    public net.minecraft.sounds.SoundEvent(net.minecraft.resources.ResourceLocation location) {
-        this(location, 16.0f, false);
+    public net.minecraft.sounds.SoundEvent(net.minecraft.resources.ResourceLocation $$0) {
+        this($$0, 16.0f, false);
     }
 
     public net.minecraft.sounds.SoundEvent(net.minecraft.resources.ResourceLocation $$0, float $$1) {