--- net/minecraft/sounds/SoundEvents.java
+++ net/minecraft/sounds/SoundEvents.java
@@ -102,6 +102,21 @@
     public static net.minecraft.sounds.SoundEvent BAMBOO_SAPLING_BREAK;
     public static net.minecraft.sounds.SoundEvent BAMBOO_SAPLING_HIT;
     public static net.minecraft.sounds.SoundEvent BAMBOO_SAPLING_PLACE;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_BREAK;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_FALL;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_HIT;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_PLACE;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_STEP;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_DOOR_CLOSE;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_DOOR_OPEN;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_TRAPDOOR_CLOSE;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_TRAPDOOR_OPEN;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_BUTTON_CLICK_OFF;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_BUTTON_CLICK_ON;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_PRESSURE_PLATE_CLICK_OFF;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_PRESSURE_PLATE_CLICK_ON;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_FENCE_GATE_CLOSE;
+    public static net.minecraft.sounds.SoundEvent BAMBOO_WOOD_FENCE_GATE_OPEN;
     public static net.minecraft.sounds.SoundEvent BARREL_CLOSE;
     public static net.minecraft.sounds.SoundEvent BARREL_OPEN;
     public static net.minecraft.sounds.SoundEvent BASALT_BREAK;
@@ -182,6 +197,17 @@
     public static net.minecraft.sounds.SoundEvent CALCITE_PLACE;
     public static net.minecraft.sounds.SoundEvent CALCITE_HIT;
     public static net.minecraft.sounds.SoundEvent CALCITE_FALL;
+    public static net.minecraft.sounds.SoundEvent CAMEL_AMBIENT;
+    public static net.minecraft.sounds.SoundEvent CAMEL_DASH;
+    public static net.minecraft.sounds.SoundEvent CAMEL_DASH_READY;
+    public static net.minecraft.sounds.SoundEvent CAMEL_DEATH;
+    public static net.minecraft.sounds.SoundEvent CAMEL_EAT;
+    public static net.minecraft.sounds.SoundEvent CAMEL_HURT;
+    public static net.minecraft.sounds.SoundEvent CAMEL_SADDLE;
+    public static net.minecraft.sounds.SoundEvent CAMEL_SIT;
+    public static net.minecraft.sounds.SoundEvent CAMEL_STAND;
+    public static net.minecraft.sounds.SoundEvent CAMEL_STEP;
+    public static net.minecraft.sounds.SoundEvent CAMEL_STEP_SAND;
     public static net.minecraft.sounds.SoundEvent CAMPFIRE_CRACKLE;
     public static net.minecraft.sounds.SoundEvent CANDLE_AMBIENT;
     public static net.minecraft.sounds.SoundEvent CANDLE_BREAK;
@@ -512,6 +538,11 @@
     public static net.minecraft.sounds.SoundEvent HANGING_ROOTS_HIT;
     public static net.minecraft.sounds.SoundEvent HANGING_ROOTS_PLACE;
     public static net.minecraft.sounds.SoundEvent HANGING_ROOTS_STEP;
+    public static net.minecraft.sounds.SoundEvent HANGING_SIGN_STEP;
+    public static net.minecraft.sounds.SoundEvent HANGING_SIGN_BREAK;
+    public static net.minecraft.sounds.SoundEvent HANGING_SIGN_FALL;
+    public static net.minecraft.sounds.SoundEvent HANGING_SIGN_HIT;
+    public static net.minecraft.sounds.SoundEvent HANGING_SIGN_PLACE;
     public static net.minecraft.sounds.SoundEvent HOE_TILL;
     public static net.minecraft.sounds.SoundEvent HOGLIN_AMBIENT;
     public static net.minecraft.sounds.SoundEvent HOGLIN_ANGRY;
@@ -722,6 +753,21 @@
     public static net.minecraft.sounds.SoundEvent NETHER_BRICKS_FALL;
     public static net.minecraft.sounds.SoundEvent NETHER_WART_BREAK;
     public static net.minecraft.sounds.SoundEvent NETHER_WART_PLANTED;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_BREAK;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_FALL;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_HIT;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_PLACE;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_STEP;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_DOOR_CLOSE;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_DOOR_OPEN;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_TRAPDOOR_CLOSE;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_TRAPDOOR_OPEN;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_BUTTON_CLICK_OFF;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_BUTTON_CLICK_ON;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_PRESSURE_PLATE_CLICK_OFF;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_PRESSURE_PLATE_CLICK_ON;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_FENCE_GATE_CLOSE;
+    public static net.minecraft.sounds.SoundEvent NETHER_WOOD_FENCE_GATE_OPEN;
     public static net.minecraft.sounds.SoundEvent PACKED_MUD_BREAK;
     public static net.minecraft.sounds.SoundEvent PACKED_MUD_FALL;
     public static net.minecraft.sounds.SoundEvent PACKED_MUD_HIT;
@@ -1328,8 +1374,8 @@
         return (net.minecraft.sounds.SoundEvent)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.SOUND_EVENT, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.sounds.SoundEvent(new net.minecraft.resources.ResourceLocation($$0), $$1));
     }
 
-    public static net.minecraft.sounds.SoundEvent register(java.lang.String key) {
-        return (net.minecraft.sounds.SoundEvent)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.SOUND_EVENT, (java.lang.String)key, (java.lang.Object)new net.minecraft.sounds.SoundEvent(new net.minecraft.resources.ResourceLocation(key)));
+    public static net.minecraft.sounds.SoundEvent register(java.lang.String $$0) {
+        return (net.minecraft.sounds.SoundEvent)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.SOUND_EVENT, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.sounds.SoundEvent(new net.minecraft.resources.ResourceLocation($$0)));
     }
 
     public static com.google.common.collect.ImmutableList<net.minecraft.sounds.SoundEvent> registerGoatHornSoundVariants() {
@@ -1434,6 +1480,21 @@
         BAMBOO_SAPLING_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_sapling.break");
         BAMBOO_SAPLING_HIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_sapling.hit");
         BAMBOO_SAPLING_PLACE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_sapling.place");
+        BAMBOO_WOOD_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood.break");
+        BAMBOO_WOOD_FALL = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood.fall");
+        BAMBOO_WOOD_HIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood.hit");
+        BAMBOO_WOOD_PLACE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood.place");
+        BAMBOO_WOOD_STEP = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood.step");
+        BAMBOO_WOOD_DOOR_CLOSE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_door.close");
+        BAMBOO_WOOD_DOOR_OPEN = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_door.open");
+        BAMBOO_WOOD_TRAPDOOR_CLOSE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_trapdoor.close");
+        BAMBOO_WOOD_TRAPDOOR_OPEN = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_trapdoor.open");
+        BAMBOO_WOOD_BUTTON_CLICK_OFF = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_button.click_off");
+        BAMBOO_WOOD_BUTTON_CLICK_ON = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_button.click_on");
+        BAMBOO_WOOD_PRESSURE_PLATE_CLICK_OFF = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_pressure_plate.click_off");
+        BAMBOO_WOOD_PRESSURE_PLATE_CLICK_ON = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_pressure_plate.click_on");
+        BAMBOO_WOOD_FENCE_GATE_CLOSE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_fence_gate.close");
+        BAMBOO_WOOD_FENCE_GATE_OPEN = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.bamboo_wood_fence_gate.open");
         BARREL_CLOSE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.barrel.close");
         BARREL_OPEN = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.barrel.open");
         BASALT_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.basalt.break");
@@ -1514,6 +1575,17 @@
         CALCITE_PLACE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.calcite.place");
         CALCITE_HIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.calcite.hit");
         CALCITE_FALL = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.calcite.fall");
+        CAMEL_AMBIENT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.ambient");
+        CAMEL_DASH = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.dash");
+        CAMEL_DASH_READY = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.dash_ready");
+        CAMEL_DEATH = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.death");
+        CAMEL_EAT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.eat");
+        CAMEL_HURT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.hurt");
+        CAMEL_SADDLE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.saddle");
+        CAMEL_SIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.sit");
+        CAMEL_STAND = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.stand");
+        CAMEL_STEP = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.step");
+        CAMEL_STEP_SAND = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.camel.step_sand");
         CAMPFIRE_CRACKLE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.campfire.crackle");
         CANDLE_AMBIENT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.candle.ambient");
         CANDLE_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.candle.break");
@@ -1844,6 +1916,11 @@
         HANGING_ROOTS_HIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_roots.hit");
         HANGING_ROOTS_PLACE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_roots.place");
         HANGING_ROOTS_STEP = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_roots.step");
+        HANGING_SIGN_STEP = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_sign.step");
+        HANGING_SIGN_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_sign.break");
+        HANGING_SIGN_FALL = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_sign.fall");
+        HANGING_SIGN_HIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_sign.hit");
+        HANGING_SIGN_PLACE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.hanging_sign.place");
         HOE_TILL = net.minecraft.sounds.SoundEvents.register((java.lang.String)"item.hoe.till");
         HOGLIN_AMBIENT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.hoglin.ambient");
         HOGLIN_ANGRY = net.minecraft.sounds.SoundEvents.register((java.lang.String)"entity.hoglin.angry");
@@ -2053,6 +2130,21 @@
         NETHER_BRICKS_FALL = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_bricks.fall");
         NETHER_WART_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wart.break");
         NETHER_WART_PLANTED = net.minecraft.sounds.SoundEvents.register((java.lang.String)"item.nether_wart.plant");
+        NETHER_WOOD_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood.break");
+        NETHER_WOOD_FALL = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood.fall");
+        NETHER_WOOD_HIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood.hit");
+        NETHER_WOOD_PLACE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood.place");
+        NETHER_WOOD_STEP = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood.step");
+        NETHER_WOOD_DOOR_CLOSE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_door.close");
+        NETHER_WOOD_DOOR_OPEN = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_door.open");
+        NETHER_WOOD_TRAPDOOR_CLOSE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_trapdoor.close");
+        NETHER_WOOD_TRAPDOOR_OPEN = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_trapdoor.open");
+        NETHER_WOOD_BUTTON_CLICK_OFF = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_button.click_off");
+        NETHER_WOOD_BUTTON_CLICK_ON = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_button.click_on");
+        NETHER_WOOD_PRESSURE_PLATE_CLICK_OFF = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_pressure_plate.click_off");
+        NETHER_WOOD_PRESSURE_PLATE_CLICK_ON = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_pressure_plate.click_on");
+        NETHER_WOOD_FENCE_GATE_CLOSE = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_fence_gate.close");
+        NETHER_WOOD_FENCE_GATE_OPEN = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.nether_wood_fence_gate.open");
         PACKED_MUD_BREAK = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.packed_mud.break");
         PACKED_MUD_FALL = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.packed_mud.fall");
         PACKED_MUD_HIT = net.minecraft.sounds.SoundEvents.register((java.lang.String)"block.packed_mud.hit");