--- com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen$PendingInvitationSelectionList.java
+++ com/mojang/realmsclient/gui/screens/RealmsPendingInvitesScreen$PendingInvitationSelectionList.java
@@ -29,13 +29,13 @@
         return this.this$0.getFocused() == this;
     }
 
-    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        this.this$0.renderBackground(matrixStack);
+    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        this.this$0.renderBackground($$0);
     }
 
-    public void selectItem(int index) {
-        super.selectItem(index);
-        this.selectInviteListItem(index);
+    public void selectItem(int $$0) {
+        super.selectItem($$0);
+        this.selectInviteListItem($$0);
     }
 
     public void selectInviteListItem(int $$0) {
@@ -43,9 +43,9 @@
         this.this$0.updateButtonStates();
     }
 
-    public void setSelected(@javax.annotation.Nullable com.mojang.realmsclient.gui.screens.RealmsPendingInvitesScreen$Entry entry) {
-        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)entry);
-        this.this$0.selectedInvite = this.children().indexOf((java.lang.Object)entry);
+    public void setSelected(@javax.annotation.Nullable com.mojang.realmsclient.gui.screens.RealmsPendingInvitesScreen$Entry $$0) {
+        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)$$0);
+        this.this$0.selectedInvite = this.children().indexOf((java.lang.Object)$$0);
         this.this$0.updateButtonStates();
     }
 }