--- com/mojang/realmsclient/gui/screens/RealmsConfigureWorldScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsConfigureWorldScreen.java
@@ -139,28 +139,28 @@
         this.slotButtonList.forEach(com.mojang.realmsclient.gui.RealmsWorldSlotButton::tick);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         this.toolTip = null;
-        this.renderBackground(matrixStack);
-        com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)WORLD_LIST_TITLE, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.row((int)4), (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        this.renderBackground($$0);
+        com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)WORLD_LIST_TITLE, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.row((int)4), (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
         if (this.serverData == null) {
-            com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)17, (int)0xFFFFFF);
+            com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)17, (int)0xFFFFFF);
             return;
         }
-        java.lang.String string = this.serverData.getName();
-        int n = this.font.width(string);
-        int n2 = this.serverData.state == com.mojang.realmsclient.dto.RealmsServer$State.CLOSED ? 0xA0A0A0 : 0x7FFF7F;
-        int n3 = this.font.width((net.minecraft.network.chat.FormattedText)this.title);
-        com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)12, (int)0xFFFFFF);
-        com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (java.lang.String)string, (int)(this.width / 2), (int)24, (int)n2);
-        int n4 = java.lang.Math.min((int)(this.centerButton(2, 3) + 80 - 11), (int)(this.width / 2 + n / 2 + n3 / 2 + 10));
-        this.drawServerStatus(matrixStack, n4, 7, mouseX, mouseY);
+        java.lang.String $$4 = this.serverData.getName();
+        int $$5 = this.font.width($$4);
+        int $$6 = this.serverData.state == com.mojang.realmsclient.dto.RealmsServer$State.CLOSED ? 0xA0A0A0 : 0x7FFF7F;
+        int $$7 = this.font.width((net.minecraft.network.chat.FormattedText)this.title);
+        com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)12, (int)0xFFFFFF);
+        com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (java.lang.String)$$4, (int)(this.width / 2), (int)24, (int)$$6);
+        int $$8 = java.lang.Math.min((int)(this.centerButton(2, 3) + 80 - 11), (int)(this.width / 2 + $$5 / 2 + $$7 / 2 + 10));
+        this.drawServerStatus($$0, $$8, 7, $$1, $$2);
         if (this.isMinigame()) {
-            this.font.draw(matrixStack, (net.minecraft.network.chat.Component)MINIGAME_PREFIX.copy().append(this.serverData.getMinigameName()), (float)(this.leftX + 80 + 20 + 10), (float)com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.row((int)13), 0xFFFFFF);
+            this.font.draw($$0, (net.minecraft.network.chat.Component)MINIGAME_PREFIX.copy().append(this.serverData.getMinigameName()), (float)(this.leftX + 80 + 20 + 10), (float)com.mojang.realmsclient.gui.screens.RealmsConfigureWorldScreen.row((int)13), 0xFFFFFF);
         }
         if (this.toolTip != null) {
-            this.renderMousehoverTooltip(matrixStack, this.toolTip, mouseX, mouseY);
+            this.renderMousehoverTooltip($$0, this.toolTip, $$1, $$2);
         }
     }
 
@@ -172,12 +172,12 @@
         this.minecraft.keyboardHandler.setSendRepeatsToGui(false);
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.backButtonClicked();
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
     public void backButtonClicked() {