--- com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen$WorldTemplateObjectSelectionList.java
+++ com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen$WorldTemplateObjectSelectionList.java
@@ -22,19 +22,19 @@
         this.addEntry((net.minecraft.client.gui.components.ObjectSelectionList$Entry)new com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen$Entry(this.this$0, $$0));
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        if (button == 0 && mouseY >= (double)this.y0 && mouseY <= (double)this.y1) {
-            int n = this.width / 2 - 150;
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        if ($$2 == 0 && $$1 >= (double)this.y0 && $$1 <= (double)this.y1) {
+            int $$3 = this.width / 2 - 150;
             if (this.this$0.currentLink != null) {
                 net.minecraft.Util.getPlatform().openUri(this.this$0.currentLink);
             }
-            int n2 = (int)java.lang.Math.floor((double)(mouseY - (double)this.y0)) - this.headerHeight + (int)this.getScrollAmount() - 4;
-            int n3 = n2 / this.itemHeight;
-            if (mouseX >= (double)n && mouseX < (double)this.getScrollbarPosition() && n3 >= 0 && n2 >= 0 && n3 < this.getItemCount()) {
-                this.selectItem(n3);
-                this.itemClicked(n2, n3, mouseX, mouseY, this.width);
-                if (n3 >= this.this$0.worldTemplateObjectSelectionList.getItemCount()) {
-                    return super.mouseClicked(mouseX, mouseY, button);
+            int $$4 = (int)java.lang.Math.floor((double)($$1 - (double)this.y0)) - this.headerHeight + (int)this.getScrollAmount() - 4;
+            int $$5 = $$4 / this.itemHeight;
+            if ($$0 >= (double)$$3 && $$0 < (double)this.getScrollbarPosition() && $$5 >= 0 && $$4 >= 0 && $$5 < this.getItemCount()) {
+                this.selectItem($$5);
+                this.itemClicked($$4, $$5, $$0, $$1, this.width);
+                if ($$5 >= this.this$0.worldTemplateObjectSelectionList.getItemCount()) {
+                    return super.mouseClicked($$0, $$1, $$2);
                 }
                 this.this$0.clicks += 7;
                 if (this.this$0.clicks >= 10) {
@@ -43,12 +43,12 @@
                 return true;
             }
         }
-        return super.mouseClicked(mouseX, mouseY, button);
+        return super.mouseClicked($$0, $$1, $$2);
     }
 
-    public void setSelected(@javax.annotation.Nullable com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen$Entry entry) {
-        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)entry);
-        this.this$0.selectedTemplate = this.children().indexOf((java.lang.Object)entry);
+    public void setSelected(@javax.annotation.Nullable com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen$Entry $$0) {
+        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)$$0);
+        this.this$0.selectedTemplate = this.children().indexOf((java.lang.Object)$$0);
         this.this$0.updateButtonStates();
     }
 
@@ -60,8 +60,8 @@
         return 300;
     }
 
-    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack matrixStack) {
-        this.this$0.renderBackground(matrixStack);
+    public void renderBackground(com.mojang.blaze3d.vertex.PoseStack $$0) {
+        this.this$0.renderBackground($$0);
     }
 
     public boolean isFocused() {