--- com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
@@ -80,8 +80,8 @@
         this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)this.lastScreen);
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             if (this.showDots) {
                 this.onCancel();
             } else {
@@ -89,29 +89,29 @@
             }
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
         if (!this.uploadFinished && this.uploadStatus.bytesWritten != 0L && this.uploadStatus.bytesWritten == this.uploadStatus.totalBytes) {
             this.status = VERIFYING_TEXT;
             this.cancelButton.active = false;
         }
-        com.mojang.realmsclient.gui.screens.RealmsUploadScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.status, (int)(this.width / 2), (int)50, (int)0xFFFFFF);
+        com.mojang.realmsclient.gui.screens.RealmsUploadScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.status, (int)(this.width / 2), (int)50, (int)0xFFFFFF);
         if (this.showDots) {
-            this.drawDots(matrixStack);
+            this.drawDots($$0);
         }
         if (this.uploadStatus.bytesWritten != 0L && !this.cancelled) {
-            this.drawProgressBar(matrixStack);
-            this.drawUploadSpeed(matrixStack);
+            this.drawProgressBar($$0);
+            this.drawUploadSpeed($$0);
         }
         if (this.errorMessage != null) {
-            for (int i = 0; i < this.errorMessage.length; ++i) {
-                com.mojang.realmsclient.gui.screens.RealmsUploadScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.errorMessage[i], (int)(this.width / 2), (int)(110 + 12 * i), (int)0xFF0000);
+            for (int $$4 = 0; $$4 < this.errorMessage.length; ++$$4) {
+                com.mojang.realmsclient.gui.screens.RealmsUploadScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.errorMessage[$$4], (int)(this.width / 2), (int)(110 + 12 * $$4), (int)0xFF0000);
             }
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     public void drawDots(com.mojang.blaze3d.vertex.PoseStack $$0) {