--- com/mojang/realmsclient/gui/screens/RealmsPlayerScreen$Entry.java
+++ com/mojang/realmsclient/gui/screens/RealmsPlayerScreen$Entry.java
@@ -14,8 +14,8 @@
         this.playerInfo = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.renderInvitedItem(matrixStack, this.playerInfo, left, top, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        this.renderInvitedItem($$0, this.playerInfo, $$3, $$2, $$6, $$7);
     }
 
     public void renderInvitedItem(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.realmsclient.dto.PlayerInfo $$1, int $$2, int $$3, int $$4, int $$5) {