--- com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsGenericErrorScreen.java
@@ -56,13 +56,13 @@
         return net.minecraft.network.chat.Component.empty().append((net.minecraft.network.chat.Component)this.lines.getFirst()).append(": ").append((net.minecraft.network.chat.Component)this.lines.getSecond());
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        com.mojang.realmsclient.gui.screens.RealmsGenericErrorScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)((net.minecraft.network.chat.Component)this.lines.getFirst()), (int)(this.width / 2), (int)80, (int)0xFFFFFF);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        com.mojang.realmsclient.gui.screens.RealmsGenericErrorScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)((net.minecraft.network.chat.Component)this.lines.getFirst()), (int)(this.width / 2), (int)80, (int)0xFFFFFF);
         int n = this.width / 2;
         java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-        this.line2Split.renderCentered(matrixStack, n, 100, 9, 0xFF0000);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        this.line2Split.renderCentered($$0, n, 100, 9, 0xFF0000);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 