--- com/mojang/realmsclient/gui/screens/RealmsPlayerScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsPlayerScreen.java
@@ -77,12 +77,12 @@
         this.minecraft.keyboardHandler.setSendRepeatsToGui(false);
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.backButtonClicked();
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
     public void backButtonClicked() {
@@ -153,37 +153,37 @@
         this.serverData.players.remove($$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         this.toolTip = null;
         this.hoveredUserAction = com.mojang.realmsclient.gui.screens.RealmsPlayerScreen$UserAction.NONE;
-        this.renderBackground(matrixStack);
+        this.renderBackground($$0);
         if (this.invitedObjectSelectionList != null) {
-            this.invitedObjectSelectionList.render(matrixStack, mouseX, mouseY, partialTicks);
+            this.invitedObjectSelectionList.render($$0, $$1, $$2, $$3);
         }
-        com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)17, (int)0xFFFFFF);
-        int n = com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.row((int)12) + 20;
-        com.mojang.blaze3d.vertex.Tesselator tesselator = com.mojang.blaze3d.vertex.Tesselator.getInstance();
-        com.mojang.blaze3d.vertex.BufferBuilder bufferBuilder = tesselator.getBuilder();
+        com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)17, (int)0xFFFFFF);
+        int $$4 = com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.row((int)12) + 20;
+        com.mojang.blaze3d.vertex.Tesselator $$5 = com.mojang.blaze3d.vertex.Tesselator.getInstance();
+        com.mojang.blaze3d.vertex.BufferBuilder $$6 = $$5.getBuilder();
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexColorShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)OPTIONS_BACKGROUND);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        float f = 32.0f;
-        bufferBuilder.begin(com.mojang.blaze3d.vertex.VertexFormat$Mode.QUADS, com.mojang.blaze3d.vertex.DefaultVertexFormat.POSITION_TEX_COLOR);
-        bufferBuilder.vertex(0.0, (double)this.height, 0.0).uv(0.0f, (float)(this.height - n) / 32.0f + 0.0f).color(64, 64, 64, 255).endVertex();
-        bufferBuilder.vertex((double)this.width, (double)this.height, 0.0).uv((float)this.width / 32.0f, (float)(this.height - n) / 32.0f + 0.0f).color(64, 64, 64, 255).endVertex();
-        bufferBuilder.vertex((double)this.width, (double)n, 0.0).uv((float)this.width / 32.0f, 0.0f).color(64, 64, 64, 255).endVertex();
-        bufferBuilder.vertex(0.0, (double)n, 0.0).uv(0.0f, 0.0f).color(64, 64, 64, 255).endVertex();
-        tesselator.end();
+        float $$7 = 32.0f;
+        $$6.begin(com.mojang.blaze3d.vertex.VertexFormat$Mode.QUADS, com.mojang.blaze3d.vertex.DefaultVertexFormat.POSITION_TEX_COLOR);
+        $$6.vertex(0.0, (double)this.height, 0.0).uv(0.0f, (float)(this.height - $$4) / 32.0f + 0.0f).color(64, 64, 64, 255).endVertex();
+        $$6.vertex((double)this.width, (double)this.height, 0.0).uv((float)this.width / 32.0f, (float)(this.height - $$4) / 32.0f + 0.0f).color(64, 64, 64, 255).endVertex();
+        $$6.vertex((double)this.width, (double)$$4, 0.0).uv((float)this.width / 32.0f, 0.0f).color(64, 64, 64, 255).endVertex();
+        $$6.vertex(0.0, (double)$$4, 0.0).uv(0.0f, 0.0f).color(64, 64, 64, 255).endVertex();
+        $$5.end();
         if (this.serverData != null && this.serverData.players != null) {
-            this.font.draw(matrixStack, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.empty().append(INVITED_LABEL).append(" (").append(java.lang.Integer.toString((int)this.serverData.players.size())).append(")"), (float)this.column1X, (float)com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.row((int)0), 0xA0A0A0);
+            this.font.draw($$0, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.empty().append(INVITED_LABEL).append(" (").append(java.lang.Integer.toString((int)this.serverData.players.size())).append(")"), (float)this.column1X, (float)com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.row((int)0), 0xA0A0A0);
         } else {
-            this.font.draw(matrixStack, INVITED_LABEL, (float)this.column1X, (float)com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.row((int)0), 0xA0A0A0);
+            this.font.draw($$0, INVITED_LABEL, (float)this.column1X, (float)com.mojang.realmsclient.gui.screens.RealmsPlayerScreen.row((int)0), 0xA0A0A0);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
         if (this.serverData == null) {
             return;
         }
-        this.renderMousehoverTooltip(matrixStack, this.toolTip, mouseX, mouseY);
+        this.renderMousehoverTooltip($$0, this.toolTip, $$1, $$2);
     }
 
     public void renderMousehoverTooltip(com.mojang.blaze3d.vertex.PoseStack $$0, @javax.annotation.Nullable net.minecraft.network.chat.Component $$1, int $$2, int $$3) {