--- com/mojang/realmsclient/gui/screens/RealmsLongConfirmationScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsLongConfirmationScreen.java
@@ -34,20 +34,20 @@
         return net.minecraft.network.chat.CommonComponents.joinLines((net.minecraft.network.chat.Component[])new net.minecraft.network.chat.Component[]{this.type.text, this.line2, this.line3});
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.callback.accept(false);
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.type.text, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.row((int)2), (int)this.type.colorCode);
-        com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.line2, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.row((int)4), (int)0xFFFFFF);
-        com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.line3, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.row((int)6), (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.type.text, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.row((int)2), (int)this.type.colorCode);
+        com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.line2, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.row((int)4), (int)0xFFFFFF);
+        com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.line3, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongConfirmationScreen.row((int)6), (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 