--- com/mojang/realmsclient/gui/screens/RealmsBackupInfoScreen$BackupInfoListEntry.java
+++ com/mojang/realmsclient/gui/screens/RealmsBackupInfoScreen$BackupInfoListEntry.java
@@ -16,10 +16,10 @@
         this.value = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        net.minecraft.client.gui.Font font = com.mojang.realmsclient.gui.screens.RealmsBackupInfoScreen.access$000((com.mojang.realmsclient.gui.screens.RealmsBackupInfoScreen)this.this$0).font;
-        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)font, (java.lang.String)this.key, (int)left, (int)top, (int)0xA0A0A0);
-        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)font, (net.minecraft.network.chat.Component)this.this$0.checkForSpecificMetadata(this.key, this.value), (int)left, (int)(top + 12), (int)0xFFFFFF);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        net.minecraft.client.gui.Font $$10 = com.mojang.realmsclient.gui.screens.RealmsBackupInfoScreen.access$000((com.mojang.realmsclient.gui.screens.RealmsBackupInfoScreen)this.this$0).font;
+        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)$$10, (java.lang.String)this.key, (int)$$3, (int)$$2, (int)0xA0A0A0);
+        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)$$10, (net.minecraft.network.chat.Component)this.this$0.checkForSpecificMetadata(this.key, this.value), (int)$$3, (int)($$2 + 12), (int)0xFFFFFF);
     }
 
     public net.minecraft.network.chat.Component getNarration() {