--- com/mojang/realmsclient/gui/screens/RealmsLongRunningMcoTaskScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsLongRunningMcoTaskScreen.java
@@ -39,12 +39,12 @@
         this.task.tick();
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.cancelOrBackButtonClicked();
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
     public void init() {
@@ -58,16 +58,16 @@
         this.minecraft.setScreen(this.lastScreen);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.row((int)3), (int)0xFFFFFF);
-        net.minecraft.network.chat.Component component = this.errorMessage;
-        if (component == null) {
-            com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (java.lang.String)SYMBOLS[this.animTicks % SYMBOLS.length], (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.row((int)8), (int)0x808080);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.row((int)3), (int)0xFFFFFF);
+        net.minecraft.network.chat.Component $$4 = this.errorMessage;
+        if ($$4 == null) {
+            com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (java.lang.String)SYMBOLS[this.animTicks % SYMBOLS.length], (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.row((int)8), (int)0x808080);
         } else {
-            com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)component, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.row((int)8), (int)0xFF0000);
+            com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)$$4, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsLongRunningMcoTaskScreen.row((int)8), (int)0xFF0000);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     public void error(net.minecraft.network.chat.Component $$0) {