--- com/mojang/realmsclient/gui/screens/RealmsParentalConsentScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsParentalConsentScreen.java
@@ -32,10 +32,10 @@
         return MESSAGE;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        this.messageLines.renderCentered(matrixStack, this.width / 2, 15, 15, 0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        this.messageLines.renderCentered($$0, this.width / 2, 15, 15, 0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {