--- com/mojang/realmsclient/gui/screens/RealmsNotificationsScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsNotificationsScreen.java
@@ -72,11 +72,11 @@
         }
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         if (validClient) {
-            this.drawIcons(matrixStack, mouseX, mouseY);
+            this.drawIcons($$0, $$1, $$2);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     public void drawIcons(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {