--- com/mojang/realmsclient/gui/screens/RealmsSlotOptionsScreen$SettingsSlider.java
+++ com/mojang/realmsclient/gui/screens/RealmsSlotOptionsScreen$SettingsSlider.java
@@ -30,10 +30,10 @@
         this.setMessage((net.minecraft.network.chat.Component)net.minecraft.network.chat.CommonComponents.optionNameValue((net.minecraft.network.chat.Component)com.mojang.realmsclient.gui.screens.RealmsSlotOptionsScreen.SPAWN_PROTECTION_TEXT, (net.minecraft.network.chat.Component)(this.this$0.spawnProtection == 0 ? net.minecraft.network.chat.CommonComponents.OPTION_OFF : net.minecraft.network.chat.Component.literal((java.lang.String)java.lang.String.valueOf((int)this.this$0.spawnProtection)))));
     }
 
-    public void onClick(double mouseX, double mouseY) {
+    public void onClick(double $$0, double $$1) {
     }
 
-    public void onRelease(double mouseX, double mouseY) {
+    public void onRelease(double $$0, double $$1) {
     }
 }
 