--- com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
@@ -55,12 +55,12 @@
         this.displayWarning = true;
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
         if (this.hoverWarning && this.warningURL != null) {
             net.minecraft.Util.getPlatform().openUri("https://www.minecraft.net/realms/adventure-maps-in-1-9");
             return true;
         }
-        return super.mouseClicked(mouseX, mouseY, button);
+        return super.mouseClicked($$0, $$1, $$2);
     }
 
     public void init() {
@@ -165,47 +165,43 @@
         }
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         this.toolTip = null;
         this.currentLink = null;
         this.hoverWarning = false;
-        this.renderBackground(matrixStack);
-        this.worldTemplateObjectSelectionList.render(matrixStack, mouseX, mouseY, partialTicks);
+        this.renderBackground($$0);
+        this.worldTemplateObjectSelectionList.render($$0, $$1, $$2, $$3);
         if (this.noTemplatesMessage != null) {
-            this.renderMultilineMessage(matrixStack, mouseX, mouseY, this.noTemplatesMessage);
+            this.renderMultilineMessage($$0, $$1, $$2, this.noTemplatesMessage);
         }
-        com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)13, (int)0xFFFFFF);
+        com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)13, (int)0xFFFFFF);
         if (this.displayWarning) {
-            int n;
-            int n2;
-            net.minecraft.network.chat.Component[] componentArray = this.warning;
-            for (n2 = 0;
- n2 < componentArray.length; ++n2) {
-                int n3 = this.font.width((net.minecraft.network.chat.FormattedText)componentArray[n2]);
-                n = this.width / 2 - n3 / 2;
-                int n4 = com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.row((int)(-1 + n2));
-                if (mouseX < n || mouseX > n + n3 || mouseY < n4) continue;
+            net.minecraft.network.chat.Component[] $$4 = this.warning;
+            for (int $$5 = 0; $$5 < $$4.length; ++$$5) {
+                int $$6 = this.font.width((net.minecraft.network.chat.FormattedText)$$4[$$5]);
+                int $$7 = this.width / 2 - $$6 / 2;
+                int $$8 = com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.row((int)(-1 + $$5));
+                if ($$1 < $$7 || $$1 > $$7 + $$6 || $$2 < $$8) continue;
                 java.util.Objects.requireNonNull((java.lang.Object)this.font);
-                if (mouseY > n4 + 9) continue;
+                if ($$2 > $$8 + 9) continue;
                 this.hoverWarning = true;
             }
-            for (n2 = 0;
- n2 < componentArray.length; ++n2) {
-                net.minecraft.network.chat.Component component = componentArray[n2];
-                n = 0xA0A0A0;
+            for (int $$9 = 0; $$9 < $$4.length; ++$$9) {
+                net.minecraft.network.chat.Component $$10 = $$4[$$9];
+                int $$11 = 0xA0A0A0;
                 if (this.warningURL != null) {
                     if (this.hoverWarning) {
-                        n = 7107012;
-                        component = component.copy().withStyle(net.minecraft.ChatFormatting.STRIKETHROUGH);
+                        $$11 = 7107012;
+                        $$10 = $$10.copy().withStyle(net.minecraft.ChatFormatting.STRIKETHROUGH);
                     } else {
-                        n = 0x3366BB;
+                        $$11 = 0x3366BB;
                     }
                 }
-                com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)component, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.row((int)(-1 + n2)), (int)n);
+                com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)$$10, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsSelectWorldTemplateScreen.row((int)(-1 + $$9)), (int)$$11);
             }
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        this.renderMousehoverTooltip(matrixStack, this.toolTip, mouseX, mouseY);
+        super.render($$0, $$1, $$2, $$3);
+        this.renderMousehoverTooltip($$0, this.toolTip, $$1, $$2);
     }
 
     public void renderMultilineMessage(com.mojang.blaze3d.vertex.PoseStack $$02, int $$1, int $$2, java.util.List<com.mojang.realmsclient.util.TextRenderingUtils$Line> $$3) {