--- com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsInviteScreen.java
@@ -66,22 +66,22 @@
         this.minecraft.getNarrator().sayNow($$0);
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.minecraft.setScreen(this.lastScreen);
             return true;
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        this.font.draw(matrixStack, NAME_LABEL, (float)(this.width / 2 - 100), (float)com.mojang.realmsclient.gui.screens.RealmsInviteScreen.row((int)1), 0xA0A0A0);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        this.font.draw($$0, NAME_LABEL, (float)(this.width / 2 - 100), (float)com.mojang.realmsclient.gui.screens.RealmsInviteScreen.row((int)1), 0xA0A0A0);
         if (this.errorMsg != null) {
-            com.mojang.realmsclient.gui.screens.RealmsInviteScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.errorMsg, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsInviteScreen.row((int)5), (int)0xFF0000);
+            com.mojang.realmsclient.gui.screens.RealmsInviteScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.errorMsg, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsInviteScreen.row((int)5), (int)0xFF0000);
         }
-        this.profileName.render(matrixStack, mouseX, mouseY, partialTicks);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        this.profileName.render($$0, $$1, $$2, $$3);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {