--- com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsCreateRealmScreen.java
@@ -45,20 +45,20 @@
         this.minecraft.keyboardHandler.setSendRepeatsToGui(false);
     }
 
-    public boolean charTyped(char codePoint, int modifiers) {
-        boolean bl = super.charTyped(codePoint, modifiers);
+    public boolean charTyped(char $$0, int $$1) {
+        boolean $$2 = super.charTyped($$0, $$1);
         this.createButton.active = this.valid();
-        return bl;
+        return $$2;
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 256) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 256) {
             this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)this.lastScreen);
             return true;
         }
-        boolean bl = super.keyPressed(keyCode, scanCode, modifiers);
+        boolean $$3 = super.keyPressed($$0, $$1, $$2);
         this.createButton.active = this.valid();
-        return bl;
+        return $$3;
     }
 
     public void createWorld() {
@@ -73,18 +73,18 @@
         return !this.nameBox.getValue().trim().isEmpty();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        com.mojang.realmsclient.gui.screens.RealmsCreateRealmScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)11, (int)0xFFFFFF);
-        this.font.draw(matrixStack, NAME_LABEL, (float)(this.width / 2 - 100), 52.0f, 0xA0A0A0);
-        this.font.draw(matrixStack, DESCRIPTION_LABEL, (float)(this.width / 2 - 100), 102.0f, 0xA0A0A0);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        com.mojang.realmsclient.gui.screens.RealmsCreateRealmScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)11, (int)0xFFFFFF);
+        this.font.draw($$0, NAME_LABEL, (float)(this.width / 2 - 100), 52.0f, 0xA0A0A0);
+        this.font.draw($$0, DESCRIPTION_LABEL, (float)(this.width / 2 - 100), 102.0f, 0xA0A0A0);
         if (this.nameBox != null) {
-            this.nameBox.render(matrixStack, mouseX, mouseY, partialTicks);
+            this.nameBox.render($$0, $$1, $$2, $$3);
         }
         if (this.descriptionBox != null) {
-            this.descriptionBox.render(matrixStack, mouseX, mouseY, partialTicks);
+            this.descriptionBox.render($$0, $$1, $$2, $$3);
         }
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        super.render($$0, $$1, $$2, $$3);
     }
 
     static {