--- com/mojang/realmsclient/gui/screens/RealmsBackupScreen$Entry.java
+++ com/mojang/realmsclient/gui/screens/RealmsBackupScreen$Entry.java
@@ -14,8 +14,8 @@
         this.backup = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.renderBackupItem(matrixStack, this.backup, left - 40, top, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        this.renderBackupItem($$0, this.backup, $$3 - 40, $$2, $$6, $$7);
     }
 
     public void renderBackupItem(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.realmsclient.dto.Backup $$1, int $$2, int $$3, int $$4, int $$5) {