--- com/mojang/realmsclient/gui/screens/RealmsResetWorldScreen$FrameButton.java
+++ com/mojang/realmsclient/gui/screens/RealmsResetWorldScreen$FrameButton.java
@@ -15,8 +15,8 @@
         this.image = $$3;
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.this$0.drawFrame(matrixStack, this.x, this.y, this.getMessage(), this.image, this.isHoveredOrFocused(), this.isMouseOver((double)mouseX, (double)mouseY));
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.this$0.drawFrame($$0, this.x, this.y, this.getMessage(), this.image, this.isHoveredOrFocused(), this.isMouseOver((double)$$1, (double)$$2));
     }
 }
 