--- com/mojang/realmsclient/gui/screens/RealmsConfirmScreen.java
+++ com/mojang/realmsclient/gui/screens/RealmsConfirmScreen.java
@@ -22,11 +22,11 @@
         this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 + 5, com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.row((int)9), 100, 20, net.minecraft.network.chat.CommonComponents.GUI_NO, $$0 -> this.callback.accept(false)));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title1, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.row((int)3), (int)0xFFFFFF);
-        com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title2, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.row((int)5), (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title1, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.row((int)3), (int)0xFFFFFF);
+        com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title2, (int)(this.width / 2), (int)com.mojang.realmsclient.gui.screens.RealmsConfirmScreen.row((int)5), (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 