--- com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen$Entry.java
+++ com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen$Entry.java
@@ -14,8 +14,8 @@
         this.template = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.renderWorldTemplateItem(matrixStack, this.template, left, top, mouseX, mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        this.renderWorldTemplateItem($$0, this.template, $$3, $$2, $$6, $$7);
     }
 
     public void renderWorldTemplateItem(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.realmsclient.dto.WorldTemplate $$1, int $$2, int $$3, int $$4, int $$5) {