--- net/minecraft/world/level/storage/loot/predicates/BonusLevelTableCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/BonusLevelTableCondition.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.item.enchantment.Enchantment enchantment;
     public float[] values;
 
-    public net.minecraft.world.level.storage.loot.predicates.BonusLevelTableCondition(net.minecraft.world.item.enchantment.Enchantment enchantment, float[] chances) {
-        this.enchantment = enchantment;
-        this.values = chances;
+    public net.minecraft.world.level.storage.loot.predicates.BonusLevelTableCondition(net.minecraft.world.item.enchantment.Enchantment $$0, float[] $$1) {
+        this.enchantment = $$0;
+        this.values = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {
@@ -29,8 +29,8 @@
         return $$0.getRandom().nextFloat() < $$3;
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder bonusLevelFlatChance(net.minecraft.world.item.enchantment.Enchantment enchantment, float ... chances) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.BonusLevelTableCondition(enchantment, chances);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder bonusLevelFlatChance(net.minecraft.world.item.enchantment.Enchantment $$0, float ... $$1) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.BonusLevelTableCondition($$0, $$1);
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {