--- net/minecraft/world/level/storage/loot/predicates/DamageSourceCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/DamageSourceCondition.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.advancements.critereon.DamageSourcePredicate predicate;
 
-    public net.minecraft.world.level.storage.loot.predicates.DamageSourceCondition(net.minecraft.advancements.critereon.DamageSourcePredicate damageSourcePredicate) {
-        this.predicate = damageSourcePredicate;
+    public net.minecraft.world.level.storage.loot.predicates.DamageSourceCondition(net.minecraft.advancements.critereon.DamageSourcePredicate $$0) {
+        this.predicate = $$0;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {
@@ -26,8 +26,8 @@
         return $$2 != null && $$1 != null && this.predicate.matches($$0.getLevel(), $$2, $$1);
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder hasDamageSource(net.minecraft.advancements.critereon.DamageSourcePredicate$Builder builder) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.DamageSourceCondition(builder.build());
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder hasDamageSource(net.minecraft.advancements.critereon.DamageSourcePredicate$Builder $$0) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.DamageSourceCondition($$0.build());
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {