--- net/minecraft/world/level/storage/loot/predicates/EntityHasScoreCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/EntityHasScoreCondition.java
@@ -9,9 +9,9 @@
     public java.util.Map<java.lang.String, net.minecraft.world.level.storage.loot.IntRange> scores;
     public net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget;
 
-    public net.minecraft.world.level.storage.loot.predicates.EntityHasScoreCondition(java.util.Map<java.lang.String, net.minecraft.world.level.storage.loot.IntRange> scoreRanges, net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget) {
-        this.scores = com.google.common.collect.ImmutableMap.copyOf(scoreRanges);
-        this.entityTarget = entityTarget;
+    public net.minecraft.world.level.storage.loot.predicates.EntityHasScoreCondition(java.util.Map<java.lang.String, net.minecraft.world.level.storage.loot.IntRange> $$0, net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$1) {
+        this.scores = com.google.common.collect.ImmutableMap.copyOf($$0);
+        this.entityTarget = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {