--- net/minecraft/world/level/storage/loot/predicates/InvertedLootItemCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/InvertedLootItemCondition.java
@@ -24,14 +24,14 @@
         return this.term.getReferencedContextParams();
     }
 
-    public void validate(net.minecraft.world.level.storage.loot.ValidationContext context) {
-        super.validate(context);
-        this.term.validate(context);
+    public void validate(net.minecraft.world.level.storage.loot.ValidationContext $$0) {
+        super.validate($$0);
+        this.term.validate($$0);
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder invert(net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder toInvert) {
-        net.minecraft.world.level.storage.loot.predicates.InvertedLootItemCondition invertedLootItemCondition = new net.minecraft.world.level.storage.loot.predicates.InvertedLootItemCondition(toInvert.build());
-        return () -> invertedLootItemCondition;
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder invert(net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder $$0) {
+        net.minecraft.world.level.storage.loot.predicates.InvertedLootItemCondition $$1 = new net.minecraft.world.level.storage.loot.predicates.InvertedLootItemCondition($$0.build());
+        return () -> $$1;
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {