--- net/minecraft/world/level/storage/loot/predicates/LocationCheck.java
+++ net/minecraft/world/level/storage/loot/predicates/LocationCheck.java
@@ -9,9 +9,9 @@
     public net.minecraft.advancements.critereon.LocationPredicate predicate;
     public net.minecraft.core.BlockPos offset;
 
-    public net.minecraft.world.level.storage.loot.predicates.LocationCheck(net.minecraft.advancements.critereon.LocationPredicate locationPredicate, net.minecraft.core.BlockPos offset) {
-        this.predicate = locationPredicate;
-        this.offset = offset;
+    public net.minecraft.world.level.storage.loot.predicates.LocationCheck(net.minecraft.advancements.critereon.LocationPredicate $$0, net.minecraft.core.BlockPos $$1) {
+        this.predicate = $$0;
+        this.offset = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {
@@ -23,12 +23,12 @@
         return $$1 != null && this.predicate.matches($$0.getLevel(), $$1.x() + (double)this.offset.getX(), $$1.y() + (double)this.offset.getY(), $$1.z() + (double)this.offset.getZ());
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder checkLocation(net.minecraft.advancements.critereon.LocationPredicate$Builder locationPredicateBuilder) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.LocationCheck(locationPredicateBuilder.build(), net.minecraft.core.BlockPos.ZERO);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder checkLocation(net.minecraft.advancements.critereon.LocationPredicate$Builder $$0) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.LocationCheck($$0.build(), net.minecraft.core.BlockPos.ZERO);
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder checkLocation(net.minecraft.advancements.critereon.LocationPredicate$Builder locationPredicateBuilder, net.minecraft.core.BlockPos offset) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.LocationCheck(locationPredicateBuilder.build(), offset);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder checkLocation(net.minecraft.advancements.critereon.LocationPredicate$Builder $$0, net.minecraft.core.BlockPos $$1) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.LocationCheck($$0.build(), $$1);
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {