--- net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.level.block.Block block;
     public net.minecraft.advancements.critereon.StatePropertiesPredicate properties;
 
-    public net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition(net.minecraft.world.level.block.Block block, net.minecraft.advancements.critereon.StatePropertiesPredicate statePredicate) {
-        this.block = block;
-        this.properties = statePredicate;
+    public net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition(net.minecraft.world.level.block.Block $$0, net.minecraft.advancements.critereon.StatePropertiesPredicate $$1) {
+        this.block = $$0;
+        this.properties = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {
@@ -27,8 +27,8 @@
         return $$1 != null && $$1.is(this.block) && this.properties.matches($$1);
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder hasBlockStateProperties(net.minecraft.world.level.block.Block block) {
-        return new net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder(block);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder hasBlockStateProperties(net.minecraft.world.level.block.Block $$0) {
+        return new net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder($$0);
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {