--- net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition$Builder.java
+++ net/minecraft/world/level/storage/loot/predicates/LootItemBlockStatePropertyCondition$Builder.java
@@ -9,13 +9,13 @@
     public net.minecraft.world.level.block.Block block;
     public net.minecraft.advancements.critereon.StatePropertiesPredicate properties;
 
-    public net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder(net.minecraft.world.level.block.Block block) {
+    public net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder(net.minecraft.world.level.block.Block $$0) {
         this.properties = net.minecraft.advancements.critereon.StatePropertiesPredicate.ANY;
-        this.block = block;
+        this.block = $$0;
     }
 
-    public net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder setProperties(net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder statePredicateBuilder) {
-        this.properties = statePredicateBuilder.build();
+    public net.minecraft.world.level.storage.loot.predicates.LootItemBlockStatePropertyCondition$Builder setProperties(net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder $$0) {
+        this.properties = $$0.build();
         return this;
     }
 