--- net/minecraft/world/level/storage/loot/predicates/LootItemRandomChanceCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/LootItemRandomChanceCondition.java
@@ -8,8 +8,8 @@
  {
     public float probability;
 
-    public net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceCondition(float probability) {
-        this.probability = probability;
+    public net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceCondition(float $$0) {
+        this.probability = $$0;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {
@@ -20,8 +20,8 @@
         return $$0.getRandom().nextFloat() < this.probability;
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder randomChance(float probability) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceCondition(probability);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder randomChance(float $$0) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.LootItemRandomChanceCondition($$0);
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {