--- net/minecraft/world/level/storage/loot/predicates/LootItemEntityPropertyCondition.java
+++ net/minecraft/world/level/storage/loot/predicates/LootItemEntityPropertyCondition.java
@@ -9,9 +9,9 @@
     public net.minecraft.advancements.critereon.EntityPredicate predicate;
     public net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget;
 
-    public net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition(net.minecraft.advancements.critereon.EntityPredicate entityPredicate, net.minecraft.world.level.storage.loot.LootContext$EntityTarget entityTarget) {
-        this.predicate = entityPredicate;
-        this.entityTarget = entityTarget;
+    public net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition(net.minecraft.advancements.critereon.EntityPredicate $$0, net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$1) {
+        this.predicate = $$0;
+        this.entityTarget = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {
@@ -28,16 +28,16 @@
         return this.predicate.matches($$0.getLevel(), $$2, $$1);
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder entityPresent(net.minecraft.world.level.storage.loot.LootContext$EntityTarget target) {
-        return net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition.hasProperties((net.minecraft.world.level.storage.loot.LootContext$EntityTarget)target, (net.minecraft.advancements.critereon.EntityPredicate$Builder)net.minecraft.advancements.critereon.EntityPredicate$Builder.entity());
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder entityPresent(net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$0) {
+        return net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition.hasProperties((net.minecraft.world.level.storage.loot.LootContext$EntityTarget)$$0, (net.minecraft.advancements.critereon.EntityPredicate$Builder)net.minecraft.advancements.critereon.EntityPredicate$Builder.entity());
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder hasProperties(net.minecraft.world.level.storage.loot.LootContext$EntityTarget target, net.minecraft.advancements.critereon.EntityPredicate$Builder predicateBuilder) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition(predicateBuilder.build(), target);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder hasProperties(net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$0, net.minecraft.advancements.critereon.EntityPredicate$Builder $$1) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition($$1.build(), $$0);
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder hasProperties(net.minecraft.world.level.storage.loot.LootContext$EntityTarget target, net.minecraft.advancements.critereon.EntityPredicate entityPredicate) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition(entityPredicate, target);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder hasProperties(net.minecraft.world.level.storage.loot.LootContext$EntityTarget $$0, net.minecraft.advancements.critereon.EntityPredicate $$1) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition($$1, $$0);
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {