--- net/minecraft/world/level/storage/loot/predicates/MatchTool.java
+++ net/minecraft/world/level/storage/loot/predicates/MatchTool.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.advancements.critereon.ItemPredicate predicate;
 
-    public net.minecraft.world.level.storage.loot.predicates.MatchTool(net.minecraft.advancements.critereon.ItemPredicate toolPredicate) {
-        this.predicate = toolPredicate;
+    public net.minecraft.world.level.storage.loot.predicates.MatchTool(net.minecraft.advancements.critereon.ItemPredicate $$0) {
+        this.predicate = $$0;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {
@@ -25,8 +25,8 @@
         return $$1 != null && this.predicate.matches($$1);
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder toolMatches(net.minecraft.advancements.critereon.ItemPredicate$Builder toolPredicateBuilder) {
-        return () -> new net.minecraft.world.level.storage.loot.predicates.MatchTool(toolPredicateBuilder.build());
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition$Builder toolMatches(net.minecraft.advancements.critereon.ItemPredicate$Builder $$0) {
+        return () -> new net.minecraft.world.level.storage.loot.predicates.MatchTool($$0.build());
     }
 
     public /* synthetic */ boolean test(java.lang.Object object) {