--- net/minecraft/world/level/storage/loot/predicates/WeatherCheck.java
+++ net/minecraft/world/level/storage/loot/predicates/WeatherCheck.java
@@ -11,9 +11,9 @@
     @javax.annotation.Nullable
     public java.lang.Boolean isThundering;
 
-    public net.minecraft.world.level.storage.loot.predicates.WeatherCheck(@javax.annotation.Nullable java.lang.Boolean isRaining, @javax.annotation.Nullable java.lang.Boolean isThundering) {
-        this.isRaining = isRaining;
-        this.isThundering = isThundering;
+    public net.minecraft.world.level.storage.loot.predicates.WeatherCheck(@javax.annotation.Nullable java.lang.Boolean $$0, @javax.annotation.Nullable java.lang.Boolean $$1) {
+        this.isRaining = $$0;
+        this.isThundering = $$1;
     }
 
     public net.minecraft.world.level.storage.loot.predicates.LootItemConditionType getType() {