--- net/minecraft/client/sounds/ChannelAccess$ChannelHandle.java
+++ net/minecraft/client/sounds/ChannelAccess$ChannelHandle.java
@@ -14,15 +14,15 @@
         return this.stopped;
     }
 
-    public net.minecraft.client.sounds.ChannelAccess$ChannelHandle(net.minecraft.client.sounds.ChannelAccess channelAccess, com.mojang.blaze3d.audio.Channel channel) {
-        this.this$0 = channelAccess;
-        this.channel = channel;
+    public net.minecraft.client.sounds.ChannelAccess$ChannelHandle(net.minecraft.client.sounds.ChannelAccess $$0, com.mojang.blaze3d.audio.Channel $$1) {
+        this.this$0 = $$0;
+        this.channel = $$1;
     }
 
-    public void execute(java.util.function.Consumer<com.mojang.blaze3d.audio.Channel> soundConsumer) {
+    public void execute(java.util.function.Consumer<com.mojang.blaze3d.audio.Channel> $$0) {
         this.this$0.executor.execute(() -> {
             if (this.channel != null) {
-                soundConsumer.accept((java.lang.Object)this.channel);
+                $$0.accept((java.lang.Object)this.channel);
             }
         });
     }