--- net/minecraft/client/sounds/SoundEngineExecutor.java
+++ net/minecraft/client/sounds/SoundEngineExecutor.java
@@ -22,11 +22,11 @@
         return $$0;
     }
 
-    public java.lang.Runnable wrapRunnable(java.lang.Runnable runnable) {
-        return runnable;
+    public java.lang.Runnable wrapRunnable(java.lang.Runnable $$0) {
+        return $$0;
     }
 
-    public boolean shouldRun(java.lang.Runnable runnable) {
+    public boolean shouldRun(java.lang.Runnable $$0) {
         return !this.shutdown;
     }
 