--- net/minecraft/client/sounds/WeighedSoundEvents.java
+++ net/minecraft/client/sounds/WeighedSoundEvents.java
@@ -12,11 +12,11 @@
     @javax.annotation.Nullable
     public net.minecraft.network.chat.Component subtitle;
 
-    public net.minecraft.client.sounds.WeighedSoundEvents(net.minecraft.resources.ResourceLocation location, @javax.annotation.Nullable java.lang.String subtitle) {
+    public net.minecraft.client.sounds.WeighedSoundEvents(net.minecraft.resources.ResourceLocation $$0, @javax.annotation.Nullable java.lang.String $$1) {
         this.list = com.google.common.collect.Lists.newArrayList();
         this.random = net.minecraft.util.RandomSource.create();
-        this.location = location;
-        this.subtitle = subtitle == null ? null : net.minecraft.network.chat.Component.translatable((java.lang.String)subtitle);
+        this.location = $$0;
+        this.subtitle = $$1 == null ? null : net.minecraft.network.chat.Component.translatable((java.lang.String)$$1);
     }
 
     public int getWeight() {
@@ -40,8 +40,8 @@
         return net.minecraft.client.sounds.SoundManager.EMPTY_SOUND;
     }
 
-    public void addSound(net.minecraft.client.sounds.Weighted<net.minecraft.client.resources.sounds.Sound> accessor) {
-        this.list.add(accessor);
+    public void addSound(net.minecraft.client.sounds.Weighted<net.minecraft.client.resources.sounds.Sound> $$0) {
+        this.list.add($$0);
     }
 
     public net.minecraft.resources.ResourceLocation getResourceLocation() {
@@ -53,9 +53,9 @@
         return this.subtitle;
     }
 
-    public void preloadIfRequired(net.minecraft.client.sounds.SoundEngine engine) {
-        for (net.minecraft.client.sounds.Weighted weighted : this.list) {
-            weighted.preloadIfRequired(engine);
+    public void preloadIfRequired(net.minecraft.client.sounds.SoundEngine $$0) {
+        for (net.minecraft.client.sounds.Weighted $$1 : this.list) {
+            $$1.preloadIfRequired($$0);
         }
     }
 