--- net/minecraft/world/entity/monster/AbstractIllager$RaiderOpenDoorGoal.java
+++ net/minecraft/world/entity/monster/AbstractIllager$RaiderOpenDoorGoal.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.monster.AbstractIllager this$0;
 
-    public net.minecraft.world.entity.monster.AbstractIllager$RaiderOpenDoorGoal(net.minecraft.world.entity.monster.AbstractIllager abstractIllager, net.minecraft.world.entity.raid.Raider raider) {
-        this.this$0 = abstractIllager;
-        super((net.minecraft.world.entity.Mob)raider, false);
+    public net.minecraft.world.entity.monster.AbstractIllager$RaiderOpenDoorGoal(net.minecraft.world.entity.monster.AbstractIllager $$0, net.minecraft.world.entity.raid.Raider $$1) {
+        this.this$0 = $$0;
+        super((net.minecraft.world.entity.Mob)$$1, false);
     }
 
     public boolean canUse() {