--- net/minecraft/world/entity/monster/Blaze$BlazeAttackGoal.java
+++ net/minecraft/world/entity/monster/Blaze$BlazeAttackGoal.java
@@ -11,8 +11,8 @@
     public int attackTime;
     public int lastSeen;
 
-    public net.minecraft.world.entity.monster.Blaze$BlazeAttackGoal(net.minecraft.world.entity.monster.Blaze blaze) {
-        this.blaze = blaze;
+    public net.minecraft.world.entity.monster.Blaze$BlazeAttackGoal(net.minecraft.world.entity.monster.Blaze $$0) {
+        this.blaze = $$0;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE, (java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.LOOK));
     }
 