--- net/minecraft/world/entity/monster/Blaze.java
+++ net/minecraft/world/entity/monster/Blaze.java
@@ -43,7 +43,7 @@
         return net.minecraft.sounds.SoundEvents.BLAZE_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.BLAZE_HURT;
     }
 
@@ -101,10 +101,10 @@
         return ((java.lang.Byte)this.entityData.get(DATA_FLAGS_ID) & 1) != 0;
     }
 
-    public void setCharged(boolean charged) {
-        byte by = (java.lang.Byte)this.entityData.get(DATA_FLAGS_ID);
-        by = charged ? (byte)(by | 1) : (byte)(by & 0xFFFFFFFE);
-        this.entityData.set(DATA_FLAGS_ID, (java.lang.Object)by);
+    public void setCharged(boolean $$0) {
+        byte $$1 = (java.lang.Byte)this.entityData.get(DATA_FLAGS_ID);
+        $$1 = $$0 ? (byte)($$1 | 1) : (byte)($$1 & 0xFFFFFFFE);
+        this.entityData.set(DATA_FLAGS_ID, (java.lang.Object)$$1);
     }
 
     static {