--- net/minecraft/world/entity/monster/CaveSpider.java
+++ net/minecraft/world/entity/monster/CaveSpider.java
@@ -14,17 +14,17 @@
         return net.minecraft.world.entity.monster.Spider.createAttributes().add(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, 12.0);
     }
 
-    public boolean doHurtTarget(net.minecraft.world.entity.Entity entity) {
-        if (super.doHurtTarget(entity)) {
-            if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-                int n = 0;
+    public boolean doHurtTarget(net.minecraft.world.entity.Entity $$0) {
+        if (super.doHurtTarget($$0)) {
+            if ($$0 instanceof net.minecraft.world.entity.LivingEntity) {
+                int $$1 = 0;
                 if (this.level.getDifficulty() == net.minecraft.world.Difficulty.NORMAL) {
-                    n = 7;
+                    $$1 = 7;
                 } else if (this.level.getDifficulty() == net.minecraft.world.Difficulty.HARD) {
-                    n = 15;
+                    $$1 = 15;
                 }
-                if (n > 0) {
-                    ((net.minecraft.world.entity.LivingEntity)entity).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.POISON, n * 20, 0), (net.minecraft.world.entity.Entity)this);
+                if ($$1 > 0) {
+                    ((net.minecraft.world.entity.LivingEntity)$$0).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.POISON, $$1 * 20, 0), (net.minecraft.world.entity.Entity)this);
                 }
             }
             return true;
@@ -33,11 +33,11 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor level, net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.MobSpawnType reason, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData spawnData, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag dataTag) {
-        return spawnData;
+    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.world.DifficultyInstance $$1, net.minecraft.world.entity.MobSpawnType $$2, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData $$3, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$4) {
+        return $$3;
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
         return 0.45f;
     }
 }