--- net/minecraft/world/entity/monster/Drowned$DrownedAttackGoal.java
+++ net/minecraft/world/entity/monster/Drowned$DrownedAttackGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.monster.Drowned drowned;
 
-    public net.minecraft.world.entity.monster.Drowned$DrownedAttackGoal(net.minecraft.world.entity.monster.Drowned drowned, double speedModifier, boolean followingTargetEvenIfNotSeen) {
-        super((net.minecraft.world.entity.monster.Zombie)drowned, speedModifier, followingTargetEvenIfNotSeen);
-        this.drowned = drowned;
+    public net.minecraft.world.entity.monster.Drowned$DrownedAttackGoal(net.minecraft.world.entity.monster.Drowned $$0, double $$1, boolean $$2) {
+        super((net.minecraft.world.entity.monster.Zombie)$$0, $$1, $$2);
+        this.drowned = $$0;
     }
 
     public boolean canUse() {