--- net/minecraft/world/entity/monster/Drowned$DrownedGoToWaterGoal.java
+++ net/minecraft/world/entity/monster/Drowned$DrownedGoToWaterGoal.java
@@ -13,10 +13,10 @@
     public double speedModifier;
     public net.minecraft.world.level.Level level;
 
-    public net.minecraft.world.entity.monster.Drowned$DrownedGoToWaterGoal(net.minecraft.world.entity.PathfinderMob mob, double speedModifier) {
-        this.mob = mob;
-        this.speedModifier = speedModifier;
-        this.level = mob.level;
+    public net.minecraft.world.entity.monster.Drowned$DrownedGoToWaterGoal(net.minecraft.world.entity.PathfinderMob $$0, double $$1) {
+        this.mob = $$0;
+        this.speedModifier = $$1;
+        this.level = $$0.level;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE));
     }
 