--- net/minecraft/world/entity/monster/Drowned$DrownedGoToBeachGoal.java
+++ net/minecraft/world/entity/monster/Drowned$DrownedGoToBeachGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.monster.Drowned drowned;
 
-    public net.minecraft.world.entity.monster.Drowned$DrownedGoToBeachGoal(net.minecraft.world.entity.monster.Drowned drowned, double speedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)drowned, speedModifier, 8, 2);
-        this.drowned = drowned;
+    public net.minecraft.world.entity.monster.Drowned$DrownedGoToBeachGoal(net.minecraft.world.entity.monster.Drowned $$0, double $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, 8, 2);
+        this.drowned = $$0;
     }
 
     public boolean canUse() {
@@ -21,12 +21,12 @@
         return super.canContinueToUse();
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.core.BlockPos blockPos = pos.above();
-        if (!level.isEmptyBlock(blockPos) || !level.isEmptyBlock(blockPos.above())) {
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        net.minecraft.core.BlockPos $$2 = $$1.above();
+        if (!$$0.isEmptyBlock($$2) || !$$0.isEmptyBlock($$2.above())) {
             return false;
         }
-        return level.getBlockState(pos).entityCanStandOn((net.minecraft.world.level.BlockGetter)level, pos, (net.minecraft.world.entity.Entity)this.drowned);
+        return $$0.getBlockState($$1).entityCanStandOn((net.minecraft.world.level.BlockGetter)$$0, $$1, (net.minecraft.world.entity.Entity)this.drowned);
     }
 
     public void start() {