--- net/minecraft/world/entity/monster/Drowned$DrownedSwimUpGoal.java
+++ net/minecraft/world/entity/monster/Drowned$DrownedSwimUpGoal.java
@@ -11,10 +11,10 @@
     public int seaLevel;
     public boolean stuck;
 
-    public net.minecraft.world.entity.monster.Drowned$DrownedSwimUpGoal(net.minecraft.world.entity.monster.Drowned drowned, double speedModifier, int seaLevel) {
-        this.drowned = drowned;
-        this.speedModifier = speedModifier;
-        this.seaLevel = seaLevel;
+    public net.minecraft.world.entity.monster.Drowned$DrownedSwimUpGoal(net.minecraft.world.entity.monster.Drowned $$0, double $$1, int $$2) {
+        this.drowned = $$0;
+        this.speedModifier = $$1;
+        this.seaLevel = $$2;
     }
 
     public boolean canUse() {