--- net/minecraft/world/entity/monster/Drowned$DrownedMoveControl.java
+++ net/minecraft/world/entity/monster/Drowned$DrownedMoveControl.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.monster.Drowned drowned;
 
-    public net.minecraft.world.entity.monster.Drowned$DrownedMoveControl(net.minecraft.world.entity.monster.Drowned drowned) {
-        super((net.minecraft.world.entity.Mob)drowned);
-        this.drowned = drowned;
+    public net.minecraft.world.entity.monster.Drowned$DrownedMoveControl(net.minecraft.world.entity.monster.Drowned $$0) {
+        super((net.minecraft.world.entity.Mob)$$0);
+        this.drowned = $$0;
     }
 
     public void tick() {