--- net/minecraft/world/entity/monster/ElderGuardian.java
+++ net/minecraft/world/entity/monster/ElderGuardian.java
@@ -33,7 +33,7 @@
         return this.isInWaterOrBubble() ? net.minecraft.sounds.SoundEvents.ELDER_GUARDIAN_AMBIENT : net.minecraft.sounds.SoundEvents.ELDER_GUARDIAN_AMBIENT_LAND;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return this.isInWaterOrBubble() ? net.minecraft.sounds.SoundEvents.ELDER_GUARDIAN_HURT : net.minecraft.sounds.SoundEvents.ELDER_GUARDIAN_HURT_LAND;
     }
 