--- net/minecraft/world/entity/monster/Endermite.java
+++ net/minecraft/world/entity/monster/Endermite.java
@@ -25,7 +25,7 @@
         this.targetSelector.addGoal(2, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal((net.minecraft.world.entity.Mob)this, net.minecraft.world.entity.player.Player.class, true));
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
         return 0.13f;
     }
 
@@ -41,7 +41,7 @@
         return net.minecraft.sounds.SoundEvents.ENDERMITE_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.ENDERMITE_HURT;
     }
 
@@ -49,18 +49,18 @@
         return net.minecraft.sounds.SoundEvents.ENDERMITE_DEATH;
     }
 
-    public void playStepSound(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState block) {
+    public void playStepSound(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         this.playSound(net.minecraft.sounds.SoundEvents.ENDERMITE_STEP, 0.15f, 1.0f);
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.life = compound.getInt("Lifetime");
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.life = $$0.getInt("Lifetime");
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("Lifetime", this.life);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("Lifetime", this.life);
     }
 
     public void tick() {
@@ -68,9 +68,9 @@
         super.tick();
     }
 
-    public void setYBodyRot(float offset) {
-        this.setYRot(offset);
-        super.setYBodyRot(offset);
+    public void setYBodyRot(float $$0) {
+        this.setYRot($$0);
+        super.setYBodyRot($$0);
     }
 
     public double getMyRidingOffset() {