--- net/minecraft/world/entity/monster/EnderMan$EndermanLookForPlayerGoal.java
+++ net/minecraft/world/entity/monster/EnderMan$EndermanLookForPlayerGoal.java
@@ -14,11 +14,11 @@
     public net.minecraft.world.entity.ai.targeting.TargetingConditions startAggroTargetConditions;
     public net.minecraft.world.entity.ai.targeting.TargetingConditions continueAggroTargetConditions;
 
-    public net.minecraft.world.entity.monster.EnderMan$EndermanLookForPlayerGoal(net.minecraft.world.entity.monster.EnderMan enderman, @javax.annotation.Nullable java.util.function.Predicate<net.minecraft.world.entity.LivingEntity> selectionPredicate) {
-        super((net.minecraft.world.entity.Mob)enderman, net.minecraft.world.entity.player.Player.class, 10, false, false, selectionPredicate);
+    public net.minecraft.world.entity.monster.EnderMan$EndermanLookForPlayerGoal(net.minecraft.world.entity.monster.EnderMan $$0, @javax.annotation.Nullable java.util.function.Predicate<net.minecraft.world.entity.LivingEntity> $$12) {
+        super((net.minecraft.world.entity.Mob)$$0, net.minecraft.world.entity.player.Player.class, 10, false, false, $$12);
         this.continueAggroTargetConditions = net.minecraft.world.entity.ai.targeting.TargetingConditions.forCombat().ignoreLineOfSight();
-        this.enderman = enderman;
-        this.startAggroTargetConditions = net.minecraft.world.entity.ai.targeting.TargetingConditions.forCombat().range(this.getFollowDistance()).selector(entity -> enderman.isLookingAtMe((net.minecraft.world.entity.player.Player)entity));
+        this.enderman = $$0;
+        this.startAggroTargetConditions = net.minecraft.world.entity.ai.targeting.TargetingConditions.forCombat().range(this.getFollowDistance()).selector($$1 -> $$0.isLookingAtMe((net.minecraft.world.entity.player.Player)$$1));
     }
 
     public boolean canUse() {