--- net/minecraft/world/entity/monster/Evoker$EvokerAttackSpellGoal.java
+++ net/minecraft/world/entity/monster/Evoker$EvokerAttackSpellGoal.java
@@ -44,24 +44,24 @@
         }
     }
 
-    public void createSpellEntity(double x, double z, double minY, double maxY, float yRot, int warmupDelay) {
-        net.minecraft.core.BlockPos blockPos = new net.minecraft.core.BlockPos(x, maxY, z);
-        boolean bl = false;
-        double d = 0.0;
+    public void createSpellEntity(double $$0, double $$1, double $$2, double $$3, float $$4, int $$5) {
+        net.minecraft.core.BlockPos $$6 = new net.minecraft.core.BlockPos($$0, $$3, $$1);
+        boolean $$7 = false;
+        double $$8 = 0.0;
         do {
-            net.minecraft.world.level.block.state.BlockState blockState;
-            net.minecraft.world.phys.shapes.VoxelShape voxelShape;
-            net.minecraft.core.BlockPos blockPos2;
-            net.minecraft.world.level.block.state.BlockState blockState2;
-            if (!(blockState2 = this.this$0.level.getBlockState(blockPos2 = blockPos.below())).isFaceSturdy((net.minecraft.world.level.BlockGetter)this.this$0.level, blockPos2, net.minecraft.core.Direction.UP)) continue;
-            if (!this.this$0.level.isEmptyBlock(blockPos) && !(voxelShape = (blockState = this.this$0.level.getBlockState(blockPos)).getCollisionShape((net.minecraft.world.level.BlockGetter)this.this$0.level, blockPos)).isEmpty()) {
-                d = voxelShape.max(net.minecraft.core.Direction$Axis.Y);
+            net.minecraft.world.level.block.state.BlockState $$11;
+            net.minecraft.world.phys.shapes.VoxelShape $$12;
+            net.minecraft.core.BlockPos $$9;
+            net.minecraft.world.level.block.state.BlockState $$10;
+            if (!($$10 = this.this$0.level.getBlockState($$9 = $$6.below())).isFaceSturdy((net.minecraft.world.level.BlockGetter)this.this$0.level, $$9, net.minecraft.core.Direction.UP)) continue;
+            if (!this.this$0.level.isEmptyBlock($$6) && !($$12 = ($$11 = this.this$0.level.getBlockState($$6)).getCollisionShape((net.minecraft.world.level.BlockGetter)this.this$0.level, $$6)).isEmpty()) {
+                $$8 = $$12.max(net.minecraft.core.Direction$Axis.Y);
             }
-            bl = true;
+            $$7 = true;
             break;
-        } while ((blockPos = blockPos.below()).getY() >= net.minecraft.util.Mth.floor((double)minY) - 1);
-        if (bl) {
-            this.this$0.level.addFreshEntity((net.minecraft.world.entity.Entity)new net.minecraft.world.entity.projectile.EvokerFangs(this.this$0.level, x, (double)blockPos.getY() + d, z, yRot, warmupDelay, (net.minecraft.world.entity.LivingEntity)this.this$0));
+        } while (($$6 = $$6.below()).getY() >= net.minecraft.util.Mth.floor((double)$$2) - 1);
+        if ($$7) {
+            this.this$0.level.addFreshEntity((net.minecraft.world.entity.Entity)new net.minecraft.world.entity.projectile.EvokerFangs(this.this$0.level, $$0, (double)$$6.getY() + $$8, $$1, $$4, $$5, (net.minecraft.world.entity.LivingEntity)this.this$0));
         }
     }
 