--- net/minecraft/world/entity/monster/Evoker.java
+++ net/minecraft/world/entity/monster/Evoker.java
@@ -39,37 +39,37 @@
         super.defineSynchedData();
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
     }
 
     public net.minecraft.sounds.SoundEvent getCelebrateSound() {
         return net.minecraft.sounds.SoundEvents.EVOKER_CELEBRATE;
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
     }
 
     public void customServerAiStep() {
         super.customServerAiStep();
     }
 
-    public boolean isAlliedTo(net.minecraft.world.entity.Entity entity) {
-        if (entity == null) {
+    public boolean isAlliedTo(net.minecraft.world.entity.Entity $$0) {
+        if ($$0 == null) {
             return false;
         }
-        if (entity == this) {
+        if ($$0 == this) {
             return true;
         }
-        if (super.isAlliedTo(entity)) {
+        if (super.isAlliedTo($$0)) {
             return true;
         }
-        if (entity instanceof net.minecraft.world.entity.monster.Vex) {
-            return this.isAlliedTo((net.minecraft.world.entity.Entity)((net.minecraft.world.entity.monster.Vex)entity).getOwner());
+        if ($$0 instanceof net.minecraft.world.entity.monster.Vex) {
+            return this.isAlliedTo((net.minecraft.world.entity.Entity)((net.minecraft.world.entity.monster.Vex)$$0).getOwner());
         }
-        if (entity instanceof net.minecraft.world.entity.LivingEntity && ((net.minecraft.world.entity.LivingEntity)entity).getMobType() == net.minecraft.world.entity.MobType.ILLAGER) {
-            return this.getTeam() == null && entity.getTeam() == null;
+        if ($$0 instanceof net.minecraft.world.entity.LivingEntity && ((net.minecraft.world.entity.LivingEntity)$$0).getMobType() == net.minecraft.world.entity.MobType.ILLAGER) {
+            return this.getTeam() == null && $$0.getTeam() == null;
         }
         return false;
     }
@@ -82,12 +82,12 @@
         return net.minecraft.sounds.SoundEvents.EVOKER_DEATH;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.EVOKER_HURT;
     }
 
-    public void setWololoTarget(@javax.annotation.Nullable net.minecraft.world.entity.animal.Sheep wololoTarget) {
-        this.wololoTarget = wololoTarget;
+    public void setWololoTarget(@javax.annotation.Nullable net.minecraft.world.entity.animal.Sheep $$0) {
+        this.wololoTarget = $$0;
     }
 
     @javax.annotation.Nullable
@@ -99,7 +99,7 @@
         return net.minecraft.sounds.SoundEvents.EVOKER_CAST_SPELL;
     }
 
-    public void applyRaidBuffs(int wave, boolean unusedFalse) {
+    public void applyRaidBuffs(int $$0, boolean $$1) {
     }
 
     public static /* synthetic */ net.minecraft.util.RandomSource access$000(net.minecraft.world.entity.monster.Evoker $$0) {