--- net/minecraft/world/entity/monster/Evoker$EvokerSummonSpellGoal.java
+++ net/minecraft/world/entity/monster/Evoker$EvokerSummonSpellGoal.java
@@ -36,6 +36,7 @@
         for (int $$1 = 0; $$1 < 3; ++$$1) {
             net.minecraft.core.BlockPos $$2 = this.this$0.blockPosition().offset(-2 + net.minecraft.world.entity.monster.Evoker.access$100((net.minecraft.world.entity.monster.Evoker)this.this$0).nextInt(5), 1, -2 + net.minecraft.world.entity.monster.Evoker.access$200((net.minecraft.world.entity.monster.Evoker)this.this$0).nextInt(5));
             net.minecraft.world.entity.monster.Vex $$3 = (net.minecraft.world.entity.monster.Vex)net.minecraft.world.entity.EntityType.VEX.create(this.this$0.level);
+            if ($$3 == null) continue;
             $$3.moveTo($$2, 0.0f, 0.0f);
             $$3.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)$$0, this.this$0.level.getCurrentDifficultyAt($$2), net.minecraft.world.entity.MobSpawnType.MOB_SUMMONED, null, null);
             $$3.setOwner((net.minecraft.world.entity.Mob)this.this$0);