--- net/minecraft/world/entity/monster/Ghast$GhastLookGoal.java
+++ net/minecraft/world/entity/monster/Ghast$GhastLookGoal.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.entity.monster.Ghast ghast;
 
-    public net.minecraft.world.entity.monster.Ghast$GhastLookGoal(net.minecraft.world.entity.monster.Ghast ghast) {
-        this.ghast = ghast;
+    public net.minecraft.world.entity.monster.Ghast$GhastLookGoal(net.minecraft.world.entity.monster.Ghast $$0) {
+        this.ghast = $$0;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.LOOK));
     }
 