--- net/minecraft/world/entity/monster/Ghast$GhastMoveControl.java
+++ net/minecraft/world/entity/monster/Ghast$GhastMoveControl.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.entity.monster.Ghast ghast;
     public int floatDuration;
 
-    public net.minecraft.world.entity.monster.Ghast$GhastMoveControl(net.minecraft.world.entity.monster.Ghast ghast) {
-        super((net.minecraft.world.entity.Mob)ghast);
-        this.ghast = ghast;
+    public net.minecraft.world.entity.monster.Ghast$GhastMoveControl(net.minecraft.world.entity.monster.Ghast $$0) {
+        super((net.minecraft.world.entity.Mob)$$0);
+        this.ghast = $$0;
     }
 
     public void tick() {
@@ -30,10 +30,10 @@
         }
     }
 
-    public boolean canReach(net.minecraft.world.phys.Vec3 pos, int length) {
-        net.minecraft.world.phys.AABB aABB = this.ghast.getBoundingBox();
-        for (int i = 1; i < length; ++i) {
-            if (this.ghast.level.noCollision((net.minecraft.world.entity.Entity)this.ghast, aABB = aABB.move(pos))) continue;
+    public boolean canReach(net.minecraft.world.phys.Vec3 $$0, int $$1) {
+        net.minecraft.world.phys.AABB $$2 = this.ghast.getBoundingBox();
+        for (int $$3 = 1; $$3 < $$1; ++$$3) {
+            if (this.ghast.level.noCollision((net.minecraft.world.entity.Entity)this.ghast, $$2 = $$2.move($$0))) continue;
             return false;
         }
         return true;