--- net/minecraft/world/entity/monster/Giant.java
+++ net/minecraft/world/entity/monster/Giant.java
@@ -10,7 +10,7 @@
         super($$0, $$1);
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
         return 10.440001f;
     }
 
@@ -18,8 +18,8 @@
         return net.minecraft.world.entity.monster.Monster.createMonsterAttributes().add(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, 100.0).add(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED, 0.5).add(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE, 50.0);
     }
 
-    public float getWalkTargetValue(net.minecraft.core.BlockPos pos, net.minecraft.world.level.LevelReader level) {
-        return level.getPathfindingCostFromLightLevels(pos);
+    public float getWalkTargetValue(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.LevelReader $$1) {
+        return $$1.getPathfindingCostFromLightLevels($$0);
     }
 }
 