--- net/minecraft/world/entity/monster/Guardian$GuardianAttackSelector.java
+++ net/minecraft/world/entity/monster/Guardian$GuardianAttackSelector.java
@@ -8,8 +8,8 @@
  {
     public net.minecraft.world.entity.monster.Guardian guardian;
 
-    public net.minecraft.world.entity.monster.Guardian$GuardianAttackSelector(net.minecraft.world.entity.monster.Guardian guardian) {
-        this.guardian = guardian;
+    public net.minecraft.world.entity.monster.Guardian$GuardianAttackSelector(net.minecraft.world.entity.monster.Guardian $$0) {
+        this.guardian = $$0;
     }
 
     public boolean test(@javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$0) {