--- net/minecraft/world/entity/monster/Guardian$GuardianAttackGoal.java
+++ net/minecraft/world/entity/monster/Guardian$GuardianAttackGoal.java
@@ -10,9 +10,9 @@
     public int attackTime;
     public boolean elder;
 
-    public net.minecraft.world.entity.monster.Guardian$GuardianAttackGoal(net.minecraft.world.entity.monster.Guardian guardian) {
-        this.guardian = guardian;
-        this.elder = guardian instanceof net.minecraft.world.entity.monster.ElderGuardian;
+    public net.minecraft.world.entity.monster.Guardian$GuardianAttackGoal(net.minecraft.world.entity.monster.Guardian $$0) {
+        this.guardian = $$0;
+        this.elder = $$0 instanceof net.minecraft.world.entity.monster.ElderGuardian;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE, (java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.LOOK));
     }
 