--- net/minecraft/world/entity/monster/Husk.java
+++ net/minecraft/world/entity/monster/Husk.java
@@ -22,7 +22,7 @@
         return net.minecraft.sounds.SoundEvents.HUSK_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.HUSK_HURT;
     }
 
@@ -34,13 +34,13 @@
         return net.minecraft.sounds.SoundEvents.HUSK_STEP;
     }
 
-    public boolean doHurtTarget(net.minecraft.world.entity.Entity entity) {
-        boolean bl = super.doHurtTarget(entity);
-        if (bl && this.getMainHandItem().isEmpty() && entity instanceof net.minecraft.world.entity.LivingEntity) {
-            float f = this.level.getCurrentDifficultyAt(this.blockPosition()).getEffectiveDifficulty();
-            ((net.minecraft.world.entity.LivingEntity)entity).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.HUNGER, 140 * (int)f), (net.minecraft.world.entity.Entity)this);
+    public boolean doHurtTarget(net.minecraft.world.entity.Entity $$0) {
+        boolean $$1 = super.doHurtTarget($$0);
+        if ($$1 && this.getMainHandItem().isEmpty() && $$0 instanceof net.minecraft.world.entity.LivingEntity) {
+            float $$2 = this.level.getCurrentDifficultyAt(this.blockPosition()).getEffectiveDifficulty();
+            ((net.minecraft.world.entity.LivingEntity)$$0).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.HUNGER, 140 * (int)$$2), (net.minecraft.world.entity.Entity)this);
         }
-        return bl;
+        return $$1;
     }
 
     public boolean convertsInWater() {