--- net/minecraft/world/entity/monster/Phantom$PhantomLookControl.java
+++ net/minecraft/world/entity/monster/Phantom$PhantomLookControl.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.monster.Phantom this$0;
 
-    public net.minecraft.world.entity.monster.Phantom$PhantomLookControl(net.minecraft.world.entity.monster.Phantom phantom, net.minecraft.world.entity.Mob mob) {
+    public net.minecraft.world.entity.monster.Phantom$PhantomLookControl(net.minecraft.world.entity.monster.Phantom phantom, net.minecraft.world.entity.Mob $$0) {
         this.this$0 = phantom;
-        super(mob);
+        super($$0);
     }
 
     public void tick() {