--- net/minecraft/world/entity/monster/Ravager$RavagerNodeEvaluator.java
+++ net/minecraft/world/entity/monster/Ravager$RavagerNodeEvaluator.java
@@ -9,11 +9,11 @@
     public net.minecraft.world.entity.monster.Ravager$RavagerNodeEvaluator() {
     }
 
-    public net.minecraft.world.level.pathfinder.BlockPathTypes evaluateBlockPathType(net.minecraft.world.level.BlockGetter level, boolean canOpenDoors, boolean canEnterDoors, net.minecraft.core.BlockPos pos, net.minecraft.world.level.pathfinder.BlockPathTypes nodeType) {
-        if (nodeType == net.minecraft.world.level.pathfinder.BlockPathTypes.LEAVES) {
+    public net.minecraft.world.level.pathfinder.BlockPathTypes evaluateBlockPathType(net.minecraft.world.level.BlockGetter $$0, boolean $$1, boolean $$2, net.minecraft.core.BlockPos $$3, net.minecraft.world.level.pathfinder.BlockPathTypes $$4) {
+        if ($$4 == net.minecraft.world.level.pathfinder.BlockPathTypes.LEAVES) {
             return net.minecraft.world.level.pathfinder.BlockPathTypes.OPEN;
         }
-        return super.evaluateBlockPathType(level, canOpenDoors, canEnterDoors, pos, nodeType);
+        return super.evaluateBlockPathType($$0, $$1, $$2, $$3, $$4);
     }
 }
 