--- net/minecraft/world/entity/monster/Ravager.java
+++ net/minecraft/world/entity/monster/Ravager.java
@@ -51,26 +51,26 @@
         return net.minecraft.world.entity.monster.Monster.createMonsterAttributes().add(net.minecraft.world.entity.ai.attributes.Attributes.MAX_HEALTH, 100.0).add(net.minecraft.world.entity.ai.attributes.Attributes.MOVEMENT_SPEED, 0.3).add(net.minecraft.world.entity.ai.attributes.Attributes.KNOCKBACK_RESISTANCE, 0.75).add(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE, 12.0).add(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_KNOCKBACK, 1.5).add(net.minecraft.world.entity.ai.attributes.Attributes.FOLLOW_RANGE, 32.0);
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.addAdditionalSaveData(compound);
-        compound.putInt("AttackTick", this.attackTick);
-        compound.putInt("StunTick", this.stunnedTick);
-        compound.putInt("RoarTick", this.roarTick);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.addAdditionalSaveData($$0);
+        $$0.putInt("AttackTick", this.attackTick);
+        $$0.putInt("StunTick", this.stunnedTick);
+        $$0.putInt("RoarTick", this.roarTick);
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        super.readAdditionalSaveData(compound);
-        this.attackTick = compound.getInt("AttackTick");
-        this.stunnedTick = compound.getInt("StunTick");
-        this.roarTick = compound.getInt("RoarTick");
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        super.readAdditionalSaveData($$0);
+        this.attackTick = $$0.getInt("AttackTick");
+        this.stunnedTick = $$0.getInt("StunTick");
+        this.roarTick = $$0.getInt("RoarTick");
     }
 
     public net.minecraft.sounds.SoundEvent getCelebrateSound() {
         return net.minecraft.sounds.SoundEvents.RAVAGER_CELEBRATE;
     }
 
-    public net.minecraft.world.entity.ai.navigation.PathNavigation createNavigation(net.minecraft.world.level.Level level) {
-        return new net.minecraft.world.entity.monster.Ravager$RavagerNavigation((net.minecraft.world.entity.Mob)this, level);
+    public net.minecraft.world.entity.ai.navigation.PathNavigation createNavigation(net.minecraft.world.level.Level $$0) {
+        return new net.minecraft.world.entity.monster.Ravager$RavagerNavigation((net.minecraft.world.entity.Mob)this, $$0);
     }
 
     public int getMaxHeadYRot() {
@@ -155,17 +155,17 @@
         return super.hasLineOfSight($$0);
     }
 
-    public void blockedByShield(net.minecraft.world.entity.LivingEntity entity) {
+    public void blockedByShield(net.minecraft.world.entity.LivingEntity $$0) {
         if (this.roarTick == 0) {
             if (this.random.nextDouble() < 0.5) {
                 this.stunnedTick = 40;
                 this.playSound(net.minecraft.sounds.SoundEvents.RAVAGER_STUNNED, 1.0f, 1.0f);
                 this.level.broadcastEntityEvent((net.minecraft.world.entity.Entity)this, (byte)39);
-                entity.push((net.minecraft.world.entity.Entity)this);
+                $$0.push((net.minecraft.world.entity.Entity)this);
             } else {
-                this.strongKnockback((net.minecraft.world.entity.Entity)entity);
+                this.strongKnockback((net.minecraft.world.entity.Entity)$$0);
             }
-            entity.hurtMarked = true;
+            $$0.hurtMarked = true;
         }
     }
 
@@ -189,21 +189,21 @@
         }
     }
 
-    public void strongKnockback(net.minecraft.world.entity.Entity entity) {
-        double d = entity.getX() - this.getX();
-        double d2 = entity.getZ() - this.getZ();
-        double d3 = java.lang.Math.max((double)(d * d + d2 * d2), (double)0.001);
-        entity.push(d / d3 * 4.0, 0.2, d2 / d3 * 4.0);
+    public void strongKnockback(net.minecraft.world.entity.Entity $$0) {
+        double $$1 = $$0.getX() - this.getX();
+        double $$2 = $$0.getZ() - this.getZ();
+        double $$3 = java.lang.Math.max((double)($$1 * $$1 + $$2 * $$2), (double)0.001);
+        $$0.push($$1 / $$3 * 4.0, 0.2, $$2 / $$3 * 4.0);
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 4) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 4) {
             this.attackTick = 10;
             this.playSound(net.minecraft.sounds.SoundEvents.RAVAGER_ATTACK, 1.0f, 1.0f);
-        } else if (id == 39) {
+        } else if ($$0 == 39) {
             this.stunnedTick = 40;
         }
-        super.handleEntityEvent(id);
+        super.handleEntityEvent($$0);
     }
 
     public int getAttackTick() {
@@ -218,11 +218,11 @@
         return this.roarTick;
     }
 
-    public boolean doHurtTarget(net.minecraft.world.entity.Entity entity) {
+    public boolean doHurtTarget(net.minecraft.world.entity.Entity $$0) {
         this.attackTick = 10;
         this.level.broadcastEntityEvent((net.minecraft.world.entity.Entity)this, (byte)4);
         this.playSound(net.minecraft.sounds.SoundEvents.RAVAGER_ATTACK, 1.0f, 1.0f);
-        return super.doHurtTarget(entity);
+        return super.doHurtTarget($$0);
     }
 
     @javax.annotation.Nullable
@@ -230,7 +230,7 @@
         return net.minecraft.sounds.SoundEvents.RAVAGER_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.RAVAGER_HURT;
     }
 
@@ -238,27 +238,27 @@
         return net.minecraft.sounds.SoundEvents.RAVAGER_DEATH;
     }
 
-    public void playStepSound(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState block) {
+    public void playStepSound(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         this.playSound(net.minecraft.sounds.SoundEvents.RAVAGER_STEP, 0.15f, 1.0f);
     }
 
-    public boolean checkSpawnObstruction(net.minecraft.world.level.LevelReader level) {
-        return !level.containsAnyLiquid(this.getBoundingBox());
+    public boolean checkSpawnObstruction(net.minecraft.world.level.LevelReader $$0) {
+        return !$$0.containsAnyLiquid(this.getBoundingBox());
     }
 
-    public void applyRaidBuffs(int wave, boolean unusedFalse) {
+    public void applyRaidBuffs(int $$0, boolean $$1) {
     }
 
     public boolean canBeLeader() {
         return false;
     }
 
-    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.entity.Entity entity) {
-        return entity.isAlive() && !(entity instanceof net.minecraft.world.entity.monster.Ravager);
+    public static /* synthetic */ boolean lambda$static$0(net.minecraft.world.entity.Entity $$0) {
+        return $$0.isAlive() && !($$0 instanceof net.minecraft.world.entity.monster.Ravager);
     }
 
     static {
-        NO_RAVAGER_AND_ALIVE = entity -> entity.isAlive() && !(entity instanceof net.minecraft.world.entity.monster.Ravager);
+        NO_RAVAGER_AND_ALIVE = $$0 -> $$0.isAlive() && !($$0 instanceof net.minecraft.world.entity.monster.Ravager);
     }
 }
 