--- net/minecraft/world/entity/monster/Ravager$RavagerMeleeAttackGoal.java
+++ net/minecraft/world/entity/monster/Ravager$RavagerMeleeAttackGoal.java
@@ -13,9 +13,9 @@
         super((net.minecraft.world.entity.PathfinderMob)ravager, 1.0, true);
     }
 
-    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity attackTarget) {
-        float f = this.this$0.getBbWidth() - 0.1f;
-        return f * 2.0f * (f * 2.0f) + attackTarget.getBbWidth();
+    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity $$0) {
+        float $$1 = this.this$0.getBbWidth() - 0.1f;
+        return $$1 * 2.0f * ($$1 * 2.0f) + $$0.getBbWidth();
     }
 }
 