--- net/minecraft/world/entity/monster/Shulker$ShulkerNearestAttackGoal.java
+++ net/minecraft/world/entity/monster/Shulker$ShulkerNearestAttackGoal.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.monster.Shulker this$0;
 
-    public net.minecraft.world.entity.monster.Shulker$ShulkerNearestAttackGoal(net.minecraft.world.entity.monster.Shulker shulker, net.minecraft.world.entity.monster.Shulker shulker2) {
+    public net.minecraft.world.entity.monster.Shulker$ShulkerNearestAttackGoal(net.minecraft.world.entity.monster.Shulker shulker, net.minecraft.world.entity.monster.Shulker $$0) {
         this.this$0 = shulker;
-        super((net.minecraft.world.entity.Mob)shulker2, net.minecraft.world.entity.player.Player.class, true);
+        super((net.minecraft.world.entity.Mob)$$0, net.minecraft.world.entity.player.Player.class, true);
     }
 
     public boolean canUse() {
@@ -20,15 +20,15 @@
         return super.canUse();
     }
 
-    public net.minecraft.world.phys.AABB getTargetSearchArea(double targetDistance) {
-        net.minecraft.core.Direction direction = ((net.minecraft.world.entity.monster.Shulker)this.mob).getAttachFace();
-        if (direction.getAxis() == net.minecraft.core.Direction$Axis.X) {
-            return this.mob.getBoundingBox().inflate(4.0, targetDistance, targetDistance);
+    public net.minecraft.world.phys.AABB getTargetSearchArea(double $$0) {
+        net.minecraft.core.Direction $$1 = ((net.minecraft.world.entity.monster.Shulker)this.mob).getAttachFace();
+        if ($$1.getAxis() == net.minecraft.core.Direction$Axis.X) {
+            return this.mob.getBoundingBox().inflate(4.0, $$0, $$0);
         }
-        if (direction.getAxis() == net.minecraft.core.Direction$Axis.Z) {
-            return this.mob.getBoundingBox().inflate(targetDistance, targetDistance, 4.0);
+        if ($$1.getAxis() == net.minecraft.core.Direction$Axis.Z) {
+            return this.mob.getBoundingBox().inflate($$0, $$0, 4.0);
         }
-        return this.mob.getBoundingBox().inflate(targetDistance, 4.0, targetDistance);
+        return this.mob.getBoundingBox().inflate($$0, 4.0, $$0);
     }
 }
 