--- net/minecraft/world/entity/monster/Shulker$ShulkerDefenseAttackGoal.java
+++ net/minecraft/world/entity/monster/Shulker$ShulkerDefenseAttackGoal.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.entity.monster.Shulker$ShulkerDefenseAttackGoal
 extends net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal<net.minecraft.world.entity.LivingEntity>
  {
-    public net.minecraft.world.entity.monster.Shulker$ShulkerDefenseAttackGoal(net.minecraft.world.entity.monster.Shulker shulker) {
-        super((net.minecraft.world.entity.Mob)shulker, net.minecraft.world.entity.LivingEntity.class, 10, true, false, entity -> entity instanceof net.minecraft.world.entity.monster.Enemy);
+    public net.minecraft.world.entity.monster.Shulker$ShulkerDefenseAttackGoal(net.minecraft.world.entity.monster.Shulker $$02) {
+        super((net.minecraft.world.entity.Mob)$$02, net.minecraft.world.entity.LivingEntity.class, 10, true, false, $$0 -> $$0 instanceof net.minecraft.world.entity.monster.Enemy);
     }
 
     public boolean canUse() {
@@ -17,15 +17,15 @@
         return super.canUse();
     }
 
-    public net.minecraft.world.phys.AABB getTargetSearchArea(double targetDistance) {
-        net.minecraft.core.Direction direction = ((net.minecraft.world.entity.monster.Shulker)this.mob).getAttachFace();
-        if (direction.getAxis() == net.minecraft.core.Direction$Axis.X) {
-            return this.mob.getBoundingBox().inflate(4.0, targetDistance, targetDistance);
+    public net.minecraft.world.phys.AABB getTargetSearchArea(double $$0) {
+        net.minecraft.core.Direction $$1 = ((net.minecraft.world.entity.monster.Shulker)this.mob).getAttachFace();
+        if ($$1.getAxis() == net.minecraft.core.Direction$Axis.X) {
+            return this.mob.getBoundingBox().inflate(4.0, $$0, $$0);
         }
-        if (direction.getAxis() == net.minecraft.core.Direction$Axis.Z) {
-            return this.mob.getBoundingBox().inflate(targetDistance, targetDistance, 4.0);
+        if ($$1.getAxis() == net.minecraft.core.Direction$Axis.Z) {
+            return this.mob.getBoundingBox().inflate($$0, $$0, 4.0);
         }
-        return this.mob.getBoundingBox().inflate(targetDistance, 4.0, targetDistance);
+        return this.mob.getBoundingBox().inflate($$0, 4.0, $$0);
     }
 }
 