--- net/minecraft/world/entity/monster/Silverfish$SilverfishWakeUpFriendsGoal.java
+++ net/minecraft/world/entity/monster/Silverfish$SilverfishWakeUpFriendsGoal.java
@@ -9,8 +9,8 @@
     public net.minecraft.world.entity.monster.Silverfish silverfish;
     public int lookForFriends;
 
-    public net.minecraft.world.entity.monster.Silverfish$SilverfishWakeUpFriendsGoal(net.minecraft.world.entity.monster.Silverfish silverfish) {
-        this.silverfish = silverfish;
+    public net.minecraft.world.entity.monster.Silverfish$SilverfishWakeUpFriendsGoal(net.minecraft.world.entity.monster.Silverfish $$0) {
+        this.silverfish = $$0;
     }
 
     public void notifyHurt() {