--- net/minecraft/world/entity/monster/Silverfish.java
+++ net/minecraft/world/entity/monster/Silverfish.java
@@ -28,7 +28,7 @@
         return 0.1;
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
         return 0.13f;
     }
 
@@ -44,7 +44,7 @@
         return net.minecraft.sounds.SoundEvents.SILVERFISH_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.SILVERFISH_HURT;
     }
 
@@ -52,18 +52,18 @@
         return net.minecraft.sounds.SoundEvents.SILVERFISH_DEATH;
     }
 
-    public void playStepSound(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState block) {
+    public void playStepSound(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         this.playSound(net.minecraft.sounds.SoundEvents.SILVERFISH_STEP, 0.15f, 1.0f);
     }
 
-    public boolean hurt(net.minecraft.world.damagesource.DamageSource source, float amount) {
-        if (this.isInvulnerableTo(source)) {
+    public boolean hurt(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
+        if (this.isInvulnerableTo($$0)) {
             return false;
         }
-        if ((source instanceof net.minecraft.world.damagesource.EntityDamageSource || source == net.minecraft.world.damagesource.DamageSource.MAGIC) && this.friendsGoal != null) {
+        if (($$0 instanceof net.minecraft.world.damagesource.EntityDamageSource || $$0 == net.minecraft.world.damagesource.DamageSource.MAGIC) && this.friendsGoal != null) {
             this.friendsGoal.notifyHurt();
         }
-        return super.hurt(source, amount);
+        return super.hurt($$0, $$1);
     }
 
     public void tick() {
@@ -71,16 +71,16 @@
         super.tick();
     }
 
-    public void setYBodyRot(float offset) {
-        this.setYRot(offset);
-        super.setYBodyRot(offset);
+    public void setYBodyRot(float $$0) {
+        this.setYRot($$0);
+        super.setYBodyRot($$0);
     }
 
-    public float getWalkTargetValue(net.minecraft.core.BlockPos pos, net.minecraft.world.level.LevelReader level) {
-        if (net.minecraft.world.level.block.InfestedBlock.isCompatibleHostBlock((net.minecraft.world.level.block.state.BlockState)level.getBlockState(pos.below()))) {
+    public float getWalkTargetValue(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.LevelReader $$1) {
+        if (net.minecraft.world.level.block.InfestedBlock.isCompatibleHostBlock((net.minecraft.world.level.block.state.BlockState)$$1.getBlockState($$0.below()))) {
             return 10.0f;
         }
-        return super.getWalkTargetValue(pos, level);
+        return super.getWalkTargetValue($$0, $$1);
     }
 
     public static boolean checkSilverfishSpawnRules(net.minecraft.world.entity.EntityType<net.minecraft.world.entity.monster.Silverfish> $$0, net.minecraft.world.level.LevelAccessor $$1, net.minecraft.world.entity.MobSpawnType $$2, net.minecraft.core.BlockPos $$3, net.minecraft.util.RandomSource $$4) {