--- net/minecraft/world/entity/monster/Skeleton.java
+++ net/minecraft/world/entity/monster/Skeleton.java
@@ -83,7 +83,7 @@
         return net.minecraft.sounds.SoundEvents.SKELETON_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.SKELETON_HURT;
     }
 
@@ -95,12 +95,12 @@
         return net.minecraft.sounds.SoundEvents.SKELETON_STEP;
     }
 
-    public void dropCustomDeathLoot(net.minecraft.world.damagesource.DamageSource source, int looting, boolean recentlyHit) {
-        net.minecraft.world.entity.monster.Creeper creeper;
-        super.dropCustomDeathLoot(source, looting, recentlyHit);
-        net.minecraft.world.entity.Entity entity = source.getEntity();
-        if (entity instanceof net.minecraft.world.entity.monster.Creeper && (creeper = (net.minecraft.world.entity.monster.Creeper)entity).canDropMobsSkull()) {
-            creeper.increaseDroppedSkulls();
+    public void dropCustomDeathLoot(net.minecraft.world.damagesource.DamageSource $$0, int $$1, boolean $$2) {
+        net.minecraft.world.entity.monster.Creeper $$4;
+        super.dropCustomDeathLoot($$0, $$1, $$2);
+        net.minecraft.world.entity.Entity $$3 = $$0.getEntity();
+        if ($$3 instanceof net.minecraft.world.entity.monster.Creeper && ($$4 = (net.minecraft.world.entity.monster.Creeper)$$3).canDropMobsSkull()) {
+            $$4.increaseDroppedSkulls();
             this.spawnAtLocation((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SKELETON_SKULL);
         }
     }