--- net/minecraft/world/entity/monster/Slime$SlimeMoveControl.java
+++ net/minecraft/world/entity/monster/Slime$SlimeMoveControl.java
@@ -11,19 +11,19 @@
     public net.minecraft.world.entity.monster.Slime slime;
     public boolean isAggressive;
 
-    public net.minecraft.world.entity.monster.Slime$SlimeMoveControl(net.minecraft.world.entity.monster.Slime slime) {
-        super((net.minecraft.world.entity.Mob)slime);
-        this.slime = slime;
-        this.yRot = 180.0f * slime.getYRot() / (float)java.lang.Math.PI;
+    public net.minecraft.world.entity.monster.Slime$SlimeMoveControl(net.minecraft.world.entity.monster.Slime $$0) {
+        super((net.minecraft.world.entity.Mob)$$0);
+        this.slime = $$0;
+        this.yRot = 180.0f * $$0.getYRot() / (float)java.lang.Math.PI;
     }
 
-    public void setDirection(float yRot, boolean aggressive) {
-        this.yRot = yRot;
-        this.isAggressive = aggressive;
+    public void setDirection(float $$0, boolean $$1) {
+        this.yRot = $$0;
+        this.isAggressive = $$1;
     }
 
-    public void setWantedMovement(double speed) {
-        this.speedModifier = speed;
+    public void setWantedMovement(double $$0) {
+        this.speedModifier = $$0;
         this.operation = net.minecraft.world.entity.ai.control.MoveControl$Operation.MOVE_TO;
     }
 