--- net/minecraft/world/entity/monster/Slime$SlimeAttackGoal.java
+++ net/minecraft/world/entity/monster/Slime$SlimeAttackGoal.java
@@ -9,8 +9,8 @@
     public net.minecraft.world.entity.monster.Slime slime;
     public int growTiredTimer;
 
-    public net.minecraft.world.entity.monster.Slime$SlimeAttackGoal(net.minecraft.world.entity.monster.Slime slime) {
-        this.slime = slime;
+    public net.minecraft.world.entity.monster.Slime$SlimeAttackGoal(net.minecraft.world.entity.monster.Slime $$0) {
+        this.slime = $$0;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.LOOK));
     }
 