--- net/minecraft/world/entity/monster/SpellcasterIllager$IllagerSpell.java
+++ net/minecraft/world/entity/monster/SpellcasterIllager$IllagerSpell.java
@@ -24,16 +24,16 @@
         return (net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell)java.lang.Enum.valueOf(net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell(int id, double red, double green, double blue) {
+    public net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell(int $$0, double $$1, double $$2, double $$3) {
         super(string, n);
-        this.id = id;
-        this.spellColor = new double[]{red, green, blue};
+        this.id = $$0;
+        this.spellColor = new double[]{$$1, $$2, $$3};
     }
 
-    public static net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell byId(int id) {
-        for (net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell illagerSpell : net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell.values()) {
-            if (id != illagerSpell.id) continue;
-            return illagerSpell;
+    public static net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell byId(int $$0) {
+        for (net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell $$1 : net.minecraft.world.entity.monster.SpellcasterIllager$IllagerSpell.values()) {
+            if ($$0 != $$1.id) continue;
+            return $$1;
         }
         return NONE;
     }