--- net/minecraft/world/entity/monster/Spider$SpiderTargetGoal.java
+++ net/minecraft/world/entity/monster/Spider$SpiderTargetGoal.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.entity.monster.Spider$SpiderTargetGoal<T extends net.minecraft.world.entity.LivingEntity>
 extends net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal<T>
  {
-    public net.minecraft.world.entity.monster.Spider$SpiderTargetGoal(net.minecraft.world.entity.monster.Spider spider, java.lang.Class<T> entityTypeToTarget) {
-        super((net.minecraft.world.entity.Mob)spider, entityTypeToTarget, true);
+    public net.minecraft.world.entity.monster.Spider$SpiderTargetGoal(net.minecraft.world.entity.monster.Spider $$0, java.lang.Class<T> $$1) {
+        super((net.minecraft.world.entity.Mob)$$0, $$1, true);
     }
 
     public boolean canUse() {