--- net/minecraft/world/entity/monster/Stray.java
+++ net/minecraft/world/entity/monster/Stray.java
@@ -21,7 +21,7 @@
         return net.minecraft.sounds.SoundEvents.STRAY_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.STRAY_HURT;
     }
 
@@ -33,12 +33,12 @@
         return net.minecraft.sounds.SoundEvents.STRAY_STEP;
     }
 
-    public net.minecraft.world.entity.projectile.AbstractArrow getArrow(net.minecraft.world.item.ItemStack arrowStack, float distanceFactor) {
-        net.minecraft.world.entity.projectile.AbstractArrow abstractArrow = super.getArrow(arrowStack, distanceFactor);
-        if (abstractArrow instanceof net.minecraft.world.entity.projectile.Arrow) {
-            ((net.minecraft.world.entity.projectile.Arrow)abstractArrow).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.MOVEMENT_SLOWDOWN, 600));
+    public net.minecraft.world.entity.projectile.AbstractArrow getArrow(net.minecraft.world.item.ItemStack $$0, float $$1) {
+        net.minecraft.world.entity.projectile.AbstractArrow $$2 = super.getArrow($$0, $$1);
+        if ($$2 instanceof net.minecraft.world.entity.projectile.Arrow) {
+            ((net.minecraft.world.entity.projectile.Arrow)$$2).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.MOVEMENT_SLOWDOWN, 600));
         }
-        return abstractArrow;
+        return $$2;
     }
 }
 