--- net/minecraft/world/entity/monster/Strider$StriderGoToLavaGoal.java
+++ net/minecraft/world/entity/monster/Strider$StriderGoToLavaGoal.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.world.entity.monster.Strider strider;
 
-    public net.minecraft.world.entity.monster.Strider$StriderGoToLavaGoal(net.minecraft.world.entity.monster.Strider strider, double speedModifier) {
-        super((net.minecraft.world.entity.PathfinderMob)strider, speedModifier, 8, 2);
-        this.strider = strider;
+    public net.minecraft.world.entity.monster.Strider$StriderGoToLavaGoal(net.minecraft.world.entity.monster.Strider $$0, double $$1) {
+        super((net.minecraft.world.entity.PathfinderMob)$$0, $$1, 8, 2);
+        this.strider = $$0;
     }
 
     public net.minecraft.core.BlockPos getMoveToTarget() {
@@ -29,8 +29,8 @@
         return this.tryTicks % 20 == 0;
     }
 
-    public boolean isValidTarget(net.minecraft.world.level.LevelReader level, net.minecraft.core.BlockPos pos) {
-        return level.getBlockState(pos).is(net.minecraft.world.level.block.Blocks.LAVA) && level.getBlockState(pos.above()).isPathfindable((net.minecraft.world.level.BlockGetter)level, pos, net.minecraft.world.level.pathfinder.PathComputationType.LAND);
+    public boolean isValidTarget(net.minecraft.world.level.LevelReader $$0, net.minecraft.core.BlockPos $$1) {
+        return $$0.getBlockState($$1).is(net.minecraft.world.level.block.Blocks.LAVA) && $$0.getBlockState($$1.above()).isPathfindable((net.minecraft.world.level.BlockGetter)$$0, $$1, net.minecraft.world.level.pathfinder.PathComputationType.LAND);
     }
 }
 