--- net/minecraft/world/entity/monster/Strider$StriderPathNavigation.java
+++ net/minecraft/world/entity/monster/Strider$StriderPathNavigation.java
@@ -6,25 +6,25 @@
 public class net.minecraft.world.entity.monster.Strider$StriderPathNavigation
 extends net.minecraft.world.entity.ai.navigation.GroundPathNavigation
  {
-    public net.minecraft.world.entity.monster.Strider$StriderPathNavigation(net.minecraft.world.entity.monster.Strider strider, net.minecraft.world.level.Level level) {
-        super((net.minecraft.world.entity.Mob)strider, level);
+    public net.minecraft.world.entity.monster.Strider$StriderPathNavigation(net.minecraft.world.entity.monster.Strider $$0, net.minecraft.world.level.Level $$1) {
+        super((net.minecraft.world.entity.Mob)$$0, $$1);
     }
 
-    public net.minecraft.world.level.pathfinder.PathFinder createPathFinder(int maxVisitedNodes) {
+    public net.minecraft.world.level.pathfinder.PathFinder createPathFinder(int $$0) {
         this.nodeEvaluator = new net.minecraft.world.level.pathfinder.WalkNodeEvaluator();
         this.nodeEvaluator.setCanPassDoors(true);
-        return new net.minecraft.world.level.pathfinder.PathFinder(this.nodeEvaluator, maxVisitedNodes);
+        return new net.minecraft.world.level.pathfinder.PathFinder(this.nodeEvaluator, $$0);
     }
 
-    public boolean hasValidPathType(net.minecraft.world.level.pathfinder.BlockPathTypes pathType) {
-        if (pathType == net.minecraft.world.level.pathfinder.BlockPathTypes.LAVA || pathType == net.minecraft.world.level.pathfinder.BlockPathTypes.DAMAGE_FIRE || pathType == net.minecraft.world.level.pathfinder.BlockPathTypes.DANGER_FIRE) {
+    public boolean hasValidPathType(net.minecraft.world.level.pathfinder.BlockPathTypes $$0) {
+        if ($$0 == net.minecraft.world.level.pathfinder.BlockPathTypes.LAVA || $$0 == net.minecraft.world.level.pathfinder.BlockPathTypes.DAMAGE_FIRE || $$0 == net.minecraft.world.level.pathfinder.BlockPathTypes.DANGER_FIRE) {
             return true;
         }
-        return super.hasValidPathType(pathType);
+        return super.hasValidPathType($$0);
     }
 
-    public boolean isStableDestination(net.minecraft.core.BlockPos pos) {
-        return this.level.getBlockState(pos).is(net.minecraft.world.level.block.Blocks.LAVA) || super.isStableDestination(pos);
+    public boolean isStableDestination(net.minecraft.core.BlockPos $$0) {
+        return this.level.getBlockState($$0).is(net.minecraft.world.level.block.Blocks.LAVA) || super.isStableDestination($$0);
     }
 }
 