--- net/minecraft/world/entity/monster/Vex$VexMoveControl.java
+++ net/minecraft/world/entity/monster/Vex$VexMoveControl.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.world.entity.monster.Vex this$0;
 
-    public net.minecraft.world.entity.monster.Vex$VexMoveControl(net.minecraft.world.entity.monster.Vex vex, net.minecraft.world.entity.monster.Vex vex2) {
+    public net.minecraft.world.entity.monster.Vex$VexMoveControl(net.minecraft.world.entity.monster.Vex vex, net.minecraft.world.entity.monster.Vex $$0) {
         this.this$0 = vex;
-        super((net.minecraft.world.entity.Mob)vex2);
+        super((net.minecraft.world.entity.Mob)$$0);
     }
 
     public void tick() {