--- net/minecraft/world/entity/monster/Vex$VexCopyOwnerTargetGoal.java
+++ net/minecraft/world/entity/monster/Vex$VexCopyOwnerTargetGoal.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.entity.ai.targeting.TargetingConditions copyOwnerTargeting;
     public /* synthetic */ net.minecraft.world.entity.monster.Vex this$0;
 
-    public net.minecraft.world.entity.monster.Vex$VexCopyOwnerTargetGoal(net.minecraft.world.entity.monster.Vex vex, net.minecraft.world.entity.PathfinderMob mob) {
+    public net.minecraft.world.entity.monster.Vex$VexCopyOwnerTargetGoal(net.minecraft.world.entity.monster.Vex vex, net.minecraft.world.entity.PathfinderMob $$0) {
         this.this$0 = vex;
-        super((net.minecraft.world.entity.Mob)mob, false);
+        super((net.minecraft.world.entity.Mob)$$0, false);
         this.copyOwnerTargeting = net.minecraft.world.entity.ai.targeting.TargetingConditions.forNonCombat().ignoreLineOfSight().ignoreInvisibilityTesting();
     }
 