--- net/minecraft/world/entity/monster/Vindicator$VindicatorJohnnyAttackGoal.java
+++ net/minecraft/world/entity/monster/Vindicator$VindicatorJohnnyAttackGoal.java
@@ -6,8 +6,8 @@
 public class net.minecraft.world.entity.monster.Vindicator$VindicatorJohnnyAttackGoal
 extends net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal<net.minecraft.world.entity.LivingEntity>
  {
-    public net.minecraft.world.entity.monster.Vindicator$VindicatorJohnnyAttackGoal(net.minecraft.world.entity.monster.Vindicator vindicator) {
-        super((net.minecraft.world.entity.Mob)vindicator, net.minecraft.world.entity.LivingEntity.class, 0, true, true, net.minecraft.world.entity.LivingEntity::attackable);
+    public net.minecraft.world.entity.monster.Vindicator$VindicatorJohnnyAttackGoal(net.minecraft.world.entity.monster.Vindicator $$0) {
+        super((net.minecraft.world.entity.Mob)$$0, net.minecraft.world.entity.LivingEntity.class, 0, true, true, net.minecraft.world.entity.LivingEntity::attackable);
     }
 
     public boolean canUse() {