--- net/minecraft/world/entity/monster/Vindicator$VindicatorMeleeAttackGoal.java
+++ net/minecraft/world/entity/monster/Vindicator$VindicatorMeleeAttackGoal.java
@@ -8,17 +8,17 @@
  {
     public /* synthetic */ net.minecraft.world.entity.monster.Vindicator this$0;
 
-    public net.minecraft.world.entity.monster.Vindicator$VindicatorMeleeAttackGoal(net.minecraft.world.entity.monster.Vindicator vindicator, net.minecraft.world.entity.monster.Vindicator vindicator2) {
+    public net.minecraft.world.entity.monster.Vindicator$VindicatorMeleeAttackGoal(net.minecraft.world.entity.monster.Vindicator vindicator, net.minecraft.world.entity.monster.Vindicator $$0) {
         this.this$0 = vindicator;
-        super((net.minecraft.world.entity.PathfinderMob)vindicator2, 1.0, false);
+        super((net.minecraft.world.entity.PathfinderMob)$$0, 1.0, false);
     }
 
-    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity attackTarget) {
+    public double getAttackReachSqr(net.minecraft.world.entity.LivingEntity $$0) {
         if (this.mob.getVehicle() instanceof net.minecraft.world.entity.monster.Ravager) {
-            float f = this.mob.getVehicle().getBbWidth() - 0.1f;
-            return f * 2.0f * (f * 2.0f) + attackTarget.getBbWidth();
+            float $$1 = this.mob.getVehicle().getBbWidth() - 0.1f;
+            return $$1 * 2.0f * ($$1 * 2.0f) + $$0.getBbWidth();
         }
-        return super.getAttackReachSqr(attackTarget);
+        return super.getAttackReachSqr($$0);
     }
 }
 