--- net/minecraft/world/entity/monster/Witch.java
+++ net/minecraft/world/entity/monster/Witch.java
@@ -20,7 +20,7 @@
 
     public void registerGoals() {
         super.registerGoals();
-        this.healRaidersGoal = new net.minecraft.world.entity.ai.goal.target.NearestHealableRaiderTargetGoal((net.minecraft.world.entity.raid.Raider)this, net.minecraft.world.entity.raid.Raider.class, true, entity -> entity != null && this.hasActiveRaid() && entity.getType() != net.minecraft.world.entity.EntityType.WITCH);
+        this.healRaidersGoal = new net.minecraft.world.entity.ai.goal.target.NearestHealableRaiderTargetGoal((net.minecraft.world.entity.raid.Raider)this, net.minecraft.world.entity.raid.Raider.class, true, $$0 -> $$0 != null && this.hasActiveRaid() && $$0.getType() != net.minecraft.world.entity.EntityType.WITCH);
         this.attackPlayersGoal = new net.minecraft.world.entity.ai.goal.target.NearestAttackableWitchTargetGoal((net.minecraft.world.entity.raid.Raider)this, net.minecraft.world.entity.player.Player.class, 10, true, false, null);
         this.goalSelector.addGoal(1, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.FloatGoal((net.minecraft.world.entity.Mob)this));
         this.goalSelector.addGoal(2, (net.minecraft.world.entity.ai.goal.Goal)new net.minecraft.world.entity.ai.goal.RangedAttackGoal((net.minecraft.world.entity.monster.RangedAttackMob)this, 1.0, 60, 10.0f));
@@ -41,7 +41,7 @@
         return net.minecraft.sounds.SoundEvents.WITCH_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.WITCH_HURT;
     }
 
@@ -49,8 +49,8 @@
         return net.minecraft.sounds.SoundEvents.WITCH_DEATH;
     }
 
-    public void setUsingItem(boolean usingItem) {
-        this.getEntityData().set(DATA_USING_ITEM, (java.lang.Object)usingItem);
+    public void setUsingItem(boolean $$0) {
+        this.getEntityData().set(DATA_USING_ITEM, (java.lang.Object)$$0);
     }
 
     public boolean isDrinkingPotion() {
@@ -116,62 +116,62 @@
         return net.minecraft.sounds.SoundEvents.WITCH_CELEBRATE;
     }
 
-    public void handleEntityEvent(byte id) {
-        if (id == 15) {
-            for (int i = 0; i < this.random.nextInt(35) + 10; ++i) {
+    public void handleEntityEvent(byte $$0) {
+        if ($$0 == 15) {
+            for (int $$1 = 0; $$1 < this.random.nextInt(35) + 10; ++$$1) {
                 this.level.addParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.WITCH, this.getX() + this.random.nextGaussian() * (double)0.13f, this.getBoundingBox().maxY + 0.5 + this.random.nextGaussian() * (double)0.13f, this.getZ() + this.random.nextGaussian() * (double)0.13f, 0.0, 0.0, 0.0);
             }
         } else {
-            super.handleEntityEvent(id);
+            super.handleEntityEvent($$0);
         }
     }
 
-    public float getDamageAfterMagicAbsorb(net.minecraft.world.damagesource.DamageSource source, float damage) {
-        damage = super.getDamageAfterMagicAbsorb(source, damage);
-        if (source.getEntity() == this) {
-            damage = 0.0f;
+    public float getDamageAfterMagicAbsorb(net.minecraft.world.damagesource.DamageSource $$0, float $$1) {
+        $$1 = super.getDamageAfterMagicAbsorb($$0, $$1);
+        if ($$0.getEntity() == this) {
+            $$1 = 0.0f;
         }
-        if (source.isMagic()) {
-            damage *= 0.15f;
+        if ($$0.isMagic()) {
+            $$1 *= 0.15f;
         }
-        return damage;
+        return $$1;
     }
 
-    public void performRangedAttack(net.minecraft.world.entity.LivingEntity target, float distanceFactor) {
+    public void performRangedAttack(net.minecraft.world.entity.LivingEntity $$0, float $$1) {
         if (this.isDrinkingPotion()) {
             return;
         }
-        net.minecraft.world.phys.Vec3 vec3 = target.getDeltaMovement();
-        double d = target.getX() + vec3.x - this.getX();
-        double d2 = target.getEyeY() - (double)1.1f - this.getY();
-        double d3 = target.getZ() + vec3.z - this.getZ();
-        double d4 = java.lang.Math.sqrt((double)(d * d + d3 * d3));
-        net.minecraft.world.item.alchemy.Potion potion = net.minecraft.world.item.alchemy.Potions.HARMING;
-        if (target instanceof net.minecraft.world.entity.raid.Raider) {
-            potion = target.getHealth() <= 4.0f ? net.minecraft.world.item.alchemy.Potions.HEALING : net.minecraft.world.item.alchemy.Potions.REGENERATION;
+        net.minecraft.world.phys.Vec3 $$2 = $$0.getDeltaMovement();
+        double $$3 = $$0.getX() + $$2.x - this.getX();
+        double $$4 = $$0.getEyeY() - (double)1.1f - this.getY();
+        double $$5 = $$0.getZ() + $$2.z - this.getZ();
+        double $$6 = java.lang.Math.sqrt((double)($$3 * $$3 + $$5 * $$5));
+        net.minecraft.world.item.alchemy.Potion $$7 = net.minecraft.world.item.alchemy.Potions.HARMING;
+        if ($$0 instanceof net.minecraft.world.entity.raid.Raider) {
+            $$7 = $$0.getHealth() <= 4.0f ? net.minecraft.world.item.alchemy.Potions.HEALING : net.minecraft.world.item.alchemy.Potions.REGENERATION;
             this.setTarget(null);
-        } else if (d4 >= 8.0 && !target.hasEffect(net.minecraft.world.effect.MobEffects.MOVEMENT_SLOWDOWN)) {
-            potion = net.minecraft.world.item.alchemy.Potions.SLOWNESS;
-        } else if (target.getHealth() >= 8.0f && !target.hasEffect(net.minecraft.world.effect.MobEffects.POISON)) {
-            potion = net.minecraft.world.item.alchemy.Potions.POISON;
-        } else if (d4 <= 3.0 && !target.hasEffect(net.minecraft.world.effect.MobEffects.WEAKNESS) && this.random.nextFloat() < 0.25f) {
-            potion = net.minecraft.world.item.alchemy.Potions.WEAKNESS;
+        } else if ($$6 >= 8.0 && !$$0.hasEffect(net.minecraft.world.effect.MobEffects.MOVEMENT_SLOWDOWN)) {
+            $$7 = net.minecraft.world.item.alchemy.Potions.SLOWNESS;
+        } else if ($$0.getHealth() >= 8.0f && !$$0.hasEffect(net.minecraft.world.effect.MobEffects.POISON)) {
+            $$7 = net.minecraft.world.item.alchemy.Potions.POISON;
+        } else if ($$6 <= 3.0 && !$$0.hasEffect(net.minecraft.world.effect.MobEffects.WEAKNESS) && this.random.nextFloat() < 0.25f) {
+            $$7 = net.minecraft.world.item.alchemy.Potions.WEAKNESS;
         }
-        net.minecraft.world.entity.projectile.ThrownPotion thrownPotion = new net.minecraft.world.entity.projectile.ThrownPotion(this.level, (net.minecraft.world.entity.LivingEntity)this);
-        thrownPotion.setItem(net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPLASH_POTION), (net.minecraft.world.item.alchemy.Potion)potion));
-        thrownPotion.setXRot(thrownPotion.getXRot() - -20.0f);
-        thrownPotion.shoot(d, d2 + d4 * 0.2, d3, 0.75f, 8.0f);
+        net.minecraft.world.entity.projectile.ThrownPotion $$8 = new net.minecraft.world.entity.projectile.ThrownPotion(this.level, (net.minecraft.world.entity.LivingEntity)this);
+        $$8.setItem(net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SPLASH_POTION), (net.minecraft.world.item.alchemy.Potion)$$7));
+        $$8.setXRot($$8.getXRot() - -20.0f);
+        $$8.shoot($$3, $$4 + $$6 * 0.2, $$5, 0.75f, 8.0f);
         if (!this.isSilent()) {
             this.level.playSound(null, this.getX(), this.getY(), this.getZ(), net.minecraft.sounds.SoundEvents.WITCH_THROW, this.getSoundSource(), 1.0f, 0.8f + this.random.nextFloat() * 0.4f);
         }
-        this.level.addFreshEntity((net.minecraft.world.entity.Entity)thrownPotion);
+        this.level.addFreshEntity((net.minecraft.world.entity.Entity)$$8);
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
         return 1.62f;
     }
 
-    public void applyRaidBuffs(int wave, boolean unusedFalse) {
+    public void applyRaidBuffs(int $$0, boolean $$1) {
     }
 
     public boolean canBeLeader() {