--- net/minecraft/world/entity/monster/WitherSkeleton.java
+++ net/minecraft/world/entity/monster/WitherSkeleton.java
@@ -20,7 +20,7 @@
         return net.minecraft.sounds.SoundEvents.WITHER_SKELETON_AMBIENT;
     }
 
-    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource damageSource) {
+    public net.minecraft.sounds.SoundEvent getHurtSound(net.minecraft.world.damagesource.DamageSource $$0) {
         return net.minecraft.sounds.SoundEvents.WITHER_SKELETON_HURT;
     }
 
@@ -32,12 +32,12 @@
         return net.minecraft.sounds.SoundEvents.WITHER_SKELETON_STEP;
     }
 
-    public void dropCustomDeathLoot(net.minecraft.world.damagesource.DamageSource source, int looting, boolean recentlyHit) {
-        net.minecraft.world.entity.monster.Creeper creeper;
-        super.dropCustomDeathLoot(source, looting, recentlyHit);
-        net.minecraft.world.entity.Entity entity = source.getEntity();
-        if (entity instanceof net.minecraft.world.entity.monster.Creeper && (creeper = (net.minecraft.world.entity.monster.Creeper)entity).canDropMobsSkull()) {
-            creeper.increaseDroppedSkulls();
+    public void dropCustomDeathLoot(net.minecraft.world.damagesource.DamageSource $$0, int $$1, boolean $$2) {
+        net.minecraft.world.entity.monster.Creeper $$4;
+        super.dropCustomDeathLoot($$0, $$1, $$2);
+        net.minecraft.world.entity.Entity $$3 = $$0.getEntity();
+        if ($$3 instanceof net.minecraft.world.entity.monster.Creeper && ($$4 = (net.minecraft.world.entity.monster.Creeper)$$3).canDropMobsSkull()) {
+            $$4.increaseDroppedSkulls();
             this.spawnAtLocation((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.WITHER_SKELETON_SKULL);
         }
     }
@@ -50,38 +50,38 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor level, net.minecraft.world.DifficultyInstance difficulty, net.minecraft.world.entity.MobSpawnType reason, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData spawnData, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag dataTag) {
-        net.minecraft.world.entity.SpawnGroupData spawnGroupData = super.finalizeSpawn(level, difficulty, reason, spawnData, dataTag);
+    public net.minecraft.world.entity.SpawnGroupData finalizeSpawn(net.minecraft.world.level.ServerLevelAccessor $$0, net.minecraft.world.DifficultyInstance $$1, net.minecraft.world.entity.MobSpawnType $$2, @javax.annotation.Nullable net.minecraft.world.entity.SpawnGroupData $$3, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$4) {
+        net.minecraft.world.entity.SpawnGroupData $$5 = super.finalizeSpawn($$0, $$1, $$2, $$3, $$4);
         this.getAttribute(net.minecraft.world.entity.ai.attributes.Attributes.ATTACK_DAMAGE).setBaseValue(4.0);
         this.reassessWeaponGoal();
-        return spawnGroupData;
+        return $$5;
     }
 
-    public float getStandingEyeHeight(net.minecraft.world.entity.Pose pose, net.minecraft.world.entity.EntityDimensions size) {
+    public float getStandingEyeHeight(net.minecraft.world.entity.Pose $$0, net.minecraft.world.entity.EntityDimensions $$1) {
         return 2.1f;
     }
 
-    public boolean doHurtTarget(net.minecraft.world.entity.Entity entity) {
-        if (!super.doHurtTarget(entity)) {
+    public boolean doHurtTarget(net.minecraft.world.entity.Entity $$0) {
+        if (!super.doHurtTarget($$0)) {
             return false;
         }
-        if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-            ((net.minecraft.world.entity.LivingEntity)entity).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.WITHER, 200), (net.minecraft.world.entity.Entity)this);
+        if ($$0 instanceof net.minecraft.world.entity.LivingEntity) {
+            ((net.minecraft.world.entity.LivingEntity)$$0).addEffect(new net.minecraft.world.effect.MobEffectInstance(net.minecraft.world.effect.MobEffects.WITHER, 200), (net.minecraft.world.entity.Entity)this);
         }
         return true;
     }
 
-    public net.minecraft.world.entity.projectile.AbstractArrow getArrow(net.minecraft.world.item.ItemStack arrowStack, float distanceFactor) {
-        net.minecraft.world.entity.projectile.AbstractArrow abstractArrow = super.getArrow(arrowStack, distanceFactor);
-        abstractArrow.setSecondsOnFire(100);
-        return abstractArrow;
+    public net.minecraft.world.entity.projectile.AbstractArrow getArrow(net.minecraft.world.item.ItemStack $$0, float $$1) {
+        net.minecraft.world.entity.projectile.AbstractArrow $$2 = super.getArrow($$0, $$1);
+        $$2.setSecondsOnFire(100);
+        return $$2;
     }
 
-    public boolean canBeAffected(net.minecraft.world.effect.MobEffectInstance potioneffect) {
-        if (potioneffect.getEffect() == net.minecraft.world.effect.MobEffects.WITHER) {
+    public boolean canBeAffected(net.minecraft.world.effect.MobEffectInstance $$0) {
+        if ($$0.getEffect() == net.minecraft.world.effect.MobEffects.WITHER) {
             return false;
         }
-        return super.canBeAffected(potioneffect);
+        return super.canBeAffected($$0);
     }
 }
 