--- net/minecraft/world/entity/monster/Zombie$ZombieGroupData.java
+++ net/minecraft/world/entity/monster/Zombie$ZombieGroupData.java
@@ -9,9 +9,9 @@
     public boolean isBaby;
     public boolean canSpawnJockey;
 
-    public net.minecraft.world.entity.monster.Zombie$ZombieGroupData(boolean isBaby, boolean canSpawnJockey) {
-        this.isBaby = isBaby;
-        this.canSpawnJockey = canSpawnJockey;
+    public net.minecraft.world.entity.monster.Zombie$ZombieGroupData(boolean $$0, boolean $$1) {
+        this.isBaby = $$0;
+        this.canSpawnJockey = $$1;
     }
 }
 