--- net/minecraft/world/entity/monster/Zombie$ZombieAttackTurtleEggGoal.java
+++ net/minecraft/world/entity/monster/Zombie$ZombieAttackTurtleEggGoal.java
@@ -8,17 +8,17 @@
  {
     public /* synthetic */ net.minecraft.world.entity.monster.Zombie this$0;
 
-    public net.minecraft.world.entity.monster.Zombie$ZombieAttackTurtleEggGoal(net.minecraft.world.entity.monster.Zombie zombie, net.minecraft.world.entity.PathfinderMob mob, double speedModifier, int verticalSearchRange) {
+    public net.minecraft.world.entity.monster.Zombie$ZombieAttackTurtleEggGoal(net.minecraft.world.entity.monster.Zombie zombie, net.minecraft.world.entity.PathfinderMob $$0, double $$1, int $$2) {
         this.this$0 = zombie;
-        super(net.minecraft.world.level.block.Blocks.TURTLE_EGG, mob, speedModifier, verticalSearchRange);
+        super(net.minecraft.world.level.block.Blocks.TURTLE_EGG, $$0, $$1, $$2);
     }
 
-    public void playDestroyProgressSound(net.minecraft.world.level.LevelAccessor level, net.minecraft.core.BlockPos pos) {
-        level.playSound(null, pos, net.minecraft.sounds.SoundEvents.ZOMBIE_DESTROY_EGG, net.minecraft.sounds.SoundSource.HOSTILE, 0.5f, 0.9f + net.minecraft.world.entity.monster.Zombie.access$000((net.minecraft.world.entity.monster.Zombie)this.this$0).nextFloat() * 0.2f);
+    public void playDestroyProgressSound(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.core.BlockPos $$1) {
+        $$0.playSound(null, $$1, net.minecraft.sounds.SoundEvents.ZOMBIE_DESTROY_EGG, net.minecraft.sounds.SoundSource.HOSTILE, 0.5f, 0.9f + net.minecraft.world.entity.monster.Zombie.access$000((net.minecraft.world.entity.monster.Zombie)this.this$0).nextFloat() * 0.2f);
     }
 
-    public void playBreakSound(net.minecraft.world.level.Level level, net.minecraft.core.BlockPos pos) {
-        level.playSound(null, pos, net.minecraft.sounds.SoundEvents.TURTLE_EGG_BREAK, net.minecraft.sounds.SoundSource.BLOCKS, 0.7f, 0.9f + level.random.nextFloat() * 0.2f);
+    public void playBreakSound(net.minecraft.world.level.Level $$0, net.minecraft.core.BlockPos $$1) {
+        $$0.playSound(null, $$1, net.minecraft.sounds.SoundEvents.TURTLE_EGG_BREAK, net.minecraft.sounds.SoundSource.BLOCKS, 0.7f, 0.9f + $$0.random.nextFloat() * 0.2f);
     }
 
     public double acceptedDistance() {