--- net/minecraft/world/level/lighting/LayerLightEventListener$DummyLightLayerEventListener.java
+++ net/minecraft/world/level/lighting/LayerLightEventListener$DummyLightLayerEventListener.java
@@ -27,7 +27,7 @@
         return null;
     }
 
-    public int getLightValue(net.minecraft.core.BlockPos levelPos) {
+    public int getLightValue(net.minecraft.core.BlockPos $$0) {
         return 0;
     }
 
@@ -45,7 +45,7 @@
         return $$0;
     }
 
-    public void updateSectionStatus(net.minecraft.core.SectionPos pos, boolean isEmpty) {
+    public void updateSectionStatus(net.minecraft.core.SectionPos $$0, boolean $$1) {
     }
 
     public void enableLightSources(net.minecraft.world.level.ChunkPos $$0, boolean $$1) {