--- net/minecraft/world/level/lighting/LevelLightEngine.java
+++ net/minecraft/world/level/lighting/LevelLightEngine.java
@@ -62,12 +62,12 @@
         return $$0;
     }
 
-    public void updateSectionStatus(net.minecraft.core.SectionPos pos, boolean isEmpty) {
+    public void updateSectionStatus(net.minecraft.core.SectionPos $$0, boolean $$1) {
         if (this.blockEngine != null) {
-            this.blockEngine.updateSectionStatus(pos, isEmpty);
+            this.blockEngine.updateSectionStatus($$0, $$1);
         }
         if (this.skyEngine != null) {
-            this.skyEngine.updateSectionStatus(pos, isEmpty);
+            this.skyEngine.updateSectionStatus($$0, $$1);
         }
     }
 
@@ -80,8 +80,8 @@
         }
     }
 
-    public net.minecraft.world.level.lighting.LayerLightEventListener getLayerListener(net.minecraft.world.level.LightLayer type) {
-        if (type == net.minecraft.world.level.LightLayer.BLOCK) {
+    public net.minecraft.world.level.lighting.LayerLightEventListener getLayerListener(net.minecraft.world.level.LightLayer $$0) {
+        if ($$0 == net.minecraft.world.level.LightLayer.BLOCK) {
             if (this.blockEngine == null) {
                 return net.minecraft.world.level.lighting.LayerLightEventListener$DummyLightLayerEventListener.INSTANCE;
             }
@@ -104,29 +104,29 @@
         return "n/a";
     }
 
-    public void queueSectionData(net.minecraft.world.level.LightLayer type, net.minecraft.core.SectionPos pos, @javax.annotation.Nullable net.minecraft.world.level.chunk.DataLayer array, boolean bl) {
-        if (type == net.minecraft.world.level.LightLayer.BLOCK) {
+    public void queueSectionData(net.minecraft.world.level.LightLayer $$0, net.minecraft.core.SectionPos $$1, @javax.annotation.Nullable net.minecraft.world.level.chunk.DataLayer $$2, boolean $$3) {
+        if ($$0 == net.minecraft.world.level.LightLayer.BLOCK) {
             if (this.blockEngine != null) {
-                this.blockEngine.queueSectionData(pos.asLong(), array, bl);
+                this.blockEngine.queueSectionData($$1.asLong(), $$2, $$3);
             }
         } else if (this.skyEngine != null) {
-            this.skyEngine.queueSectionData(pos.asLong(), array, bl);
+            this.skyEngine.queueSectionData($$1.asLong(), $$2, $$3);
         }
     }
 
-    public void retainData(net.minecraft.world.level.ChunkPos pos, boolean retain) {
+    public void retainData(net.minecraft.world.level.ChunkPos $$0, boolean $$1) {
         if (this.blockEngine != null) {
-            this.blockEngine.retainData(pos, retain);
+            this.blockEngine.retainData($$0, $$1);
         }
         if (this.skyEngine != null) {
-            this.skyEngine.retainData(pos, retain);
+            this.skyEngine.retainData($$0, $$1);
         }
     }
 
-    public int getRawBrightness(net.minecraft.core.BlockPos blockPos, int amount) {
-        int n = this.skyEngine == null ? 0 : this.skyEngine.getLightValue(blockPos) - amount;
-        int n2 = this.blockEngine == null ? 0 : this.blockEngine.getLightValue(blockPos);
-        return java.lang.Math.max((int)n2, (int)n);
+    public int getRawBrightness(net.minecraft.core.BlockPos $$0, int $$1) {
+        int $$2 = this.skyEngine == null ? 0 : this.skyEngine.getLightValue($$0) - $$1;
+        int $$3 = this.blockEngine == null ? 0 : this.blockEngine.getLightValue($$0);
+        return java.lang.Math.max((int)$$3, (int)$$2);
     }
 
     public int getLightSectionCount() {