--- net/minecraft/client/renderer/BiomeColors.java
+++ net/minecraft/client/renderer/BiomeColors.java
@@ -12,20 +12,20 @@
     public net.minecraft.client.renderer.BiomeColors() {
     }
 
-    public static int getAverageColor(net.minecraft.world.level.BlockAndTintGetter level, net.minecraft.core.BlockPos blockPos, net.minecraft.world.level.ColorResolver colorResolver) {
-        return level.getBlockTint(blockPos, colorResolver);
+    public static int getAverageColor(net.minecraft.world.level.BlockAndTintGetter $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.ColorResolver $$2) {
+        return $$0.getBlockTint($$1, $$2);
     }
 
-    public static int getAverageGrassColor(net.minecraft.world.level.BlockAndTintGetter level, net.minecraft.core.BlockPos blockPos) {
-        return net.minecraft.client.renderer.BiomeColors.getAverageColor((net.minecraft.world.level.BlockAndTintGetter)level, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.world.level.ColorResolver)GRASS_COLOR_RESOLVER);
+    public static int getAverageGrassColor(net.minecraft.world.level.BlockAndTintGetter $$0, net.minecraft.core.BlockPos $$1) {
+        return net.minecraft.client.renderer.BiomeColors.getAverageColor((net.minecraft.world.level.BlockAndTintGetter)$$0, (net.minecraft.core.BlockPos)$$1, (net.minecraft.world.level.ColorResolver)GRASS_COLOR_RESOLVER);
     }
 
-    public static int getAverageFoliageColor(net.minecraft.world.level.BlockAndTintGetter level, net.minecraft.core.BlockPos blockPos) {
-        return net.minecraft.client.renderer.BiomeColors.getAverageColor((net.minecraft.world.level.BlockAndTintGetter)level, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.world.level.ColorResolver)FOLIAGE_COLOR_RESOLVER);
+    public static int getAverageFoliageColor(net.minecraft.world.level.BlockAndTintGetter $$0, net.minecraft.core.BlockPos $$1) {
+        return net.minecraft.client.renderer.BiomeColors.getAverageColor((net.minecraft.world.level.BlockAndTintGetter)$$0, (net.minecraft.core.BlockPos)$$1, (net.minecraft.world.level.ColorResolver)FOLIAGE_COLOR_RESOLVER);
     }
 
-    public static int getAverageWaterColor(net.minecraft.world.level.BlockAndTintGetter level, net.minecraft.core.BlockPos blockPos) {
-        return net.minecraft.client.renderer.BiomeColors.getAverageColor((net.minecraft.world.level.BlockAndTintGetter)level, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.world.level.ColorResolver)WATER_COLOR_RESOLVER);
+    public static int getAverageWaterColor(net.minecraft.world.level.BlockAndTintGetter $$0, net.minecraft.core.BlockPos $$1) {
+        return net.minecraft.client.renderer.BiomeColors.getAverageColor((net.minecraft.world.level.BlockAndTintGetter)$$0, (net.minecraft.core.BlockPos)$$1, (net.minecraft.world.level.ColorResolver)WATER_COLOR_RESOLVER);
     }
 
     public static /* synthetic */ int lambda$static$1(net.minecraft.world.level.biome.Biome $$0, double $$1, double $$2) {