--- net/minecraft/client/renderer/ChunkBufferBuilderPack.java
+++ net/minecraft/client/renderer/ChunkBufferBuilderPack.java
@@ -11,8 +11,8 @@
         this.builders = (java.util.Map)net.minecraft.client.renderer.RenderType.chunkBufferLayers().stream().collect(java.util.stream.Collectors.toMap($$0 -> $$0, $$0 -> new com.mojang.blaze3d.vertex.BufferBuilder($$0.bufferSize())));
     }
 
-    public com.mojang.blaze3d.vertex.BufferBuilder builder(net.minecraft.client.renderer.RenderType renderType) {
-        return (com.mojang.blaze3d.vertex.BufferBuilder)this.builders.get((java.lang.Object)renderType);
+    public com.mojang.blaze3d.vertex.BufferBuilder builder(net.minecraft.client.renderer.RenderType $$0) {
+        return (com.mojang.blaze3d.vertex.BufferBuilder)this.builders.get((java.lang.Object)$$0);
     }
 
     public void clearAll() {