--- net/minecraft/client/renderer/FaceInfo.java
+++ net/minecraft/client/renderer/FaceInfo.java
@@ -24,17 +24,17 @@
         return (net.minecraft.client.renderer.FaceInfo)java.lang.Enum.valueOf(net.minecraft.client.renderer.FaceInfo.class, (java.lang.String)$$0);
     }
 
-    public static net.minecraft.client.renderer.FaceInfo fromFacing(net.minecraft.core.Direction facing) {
-        return BY_FACING[facing.get3DDataValue()];
+    public static net.minecraft.client.renderer.FaceInfo fromFacing(net.minecraft.core.Direction $$0) {
+        return BY_FACING[$$0.get3DDataValue()];
     }
 
-    public net.minecraft.client.renderer.FaceInfo(net.minecraft.client.renderer.FaceInfo$VertexInfo ... infos) {
+    public net.minecraft.client.renderer.FaceInfo(net.minecraft.client.renderer.FaceInfo$VertexInfo ... $$0) {
         super(string, n);
-        this.infos = infos;
+        this.infos = $$0;
     }
 
-    public net.minecraft.client.renderer.FaceInfo$VertexInfo getVertexInfo(int index) {
-        return this.infos[index];
+    public net.minecraft.client.renderer.FaceInfo$VertexInfo getVertexInfo(int $$0) {
+        return this.infos[$$0];
     }
 
     public static /* synthetic */ void lambda$static$0(net.minecraft.client.renderer.FaceInfo[] $$0) {