--- net/minecraft/client/renderer/ItemModelShaper.java
+++ net/minecraft/client/renderer/ItemModelShaper.java
@@ -15,22 +15,22 @@
         this.modelManager = $$0;
     }
 
-    public net.minecraft.client.resources.model.BakedModel getItemModel(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.client.resources.model.BakedModel bakedModel = this.getItemModel(stack.getItem());
-        return bakedModel == null ? this.modelManager.getMissingModel() : bakedModel;
+    public net.minecraft.client.resources.model.BakedModel getItemModel(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.client.resources.model.BakedModel $$1 = this.getItemModel($$0.getItem());
+        return $$1 == null ? this.modelManager.getMissingModel() : $$1;
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.client.resources.model.BakedModel getItemModel(net.minecraft.world.item.Item item) {
-        return (net.minecraft.client.resources.model.BakedModel)this.shapesCache.get(net.minecraft.client.renderer.ItemModelShaper.getIndex((net.minecraft.world.item.Item)item));
+    public net.minecraft.client.resources.model.BakedModel getItemModel(net.minecraft.world.item.Item $$0) {
+        return (net.minecraft.client.resources.model.BakedModel)this.shapesCache.get(net.minecraft.client.renderer.ItemModelShaper.getIndex((net.minecraft.world.item.Item)$$0));
     }
 
-    public static int getIndex(net.minecraft.world.item.Item item) {
-        return net.minecraft.world.item.Item.getId((net.minecraft.world.item.Item)item);
+    public static int getIndex(net.minecraft.world.item.Item $$0) {
+        return net.minecraft.world.item.Item.getId((net.minecraft.world.item.Item)$$0);
     }
 
-    public void register(net.minecraft.world.item.Item item, net.minecraft.client.resources.model.ModelResourceLocation modelLocation) {
-        this.shapes.put(net.minecraft.client.renderer.ItemModelShaper.getIndex((net.minecraft.world.item.Item)item), (java.lang.Object)modelLocation);
+    public void register(net.minecraft.world.item.Item $$0, net.minecraft.client.resources.model.ModelResourceLocation $$1) {
+        this.shapes.put(net.minecraft.client.renderer.ItemModelShaper.getIndex((net.minecraft.world.item.Item)$$0), (java.lang.Object)$$1);
     }
 
     public net.minecraft.client.resources.model.ModelManager getModelManager() {