--- net/minecraft/client/renderer/LevelRenderer$RenderChunkInfo.java
+++ net/minecraft/client/renderer/LevelRenderer$RenderChunkInfo.java
@@ -18,12 +18,12 @@
         this.step = $$2;
     }
 
-    public void setDirections(byte dir, net.minecraft.core.Direction facing) {
-        this.directions = (byte)(this.directions | (dir | 1 << facing.ordinal()));
+    public void setDirections(byte $$0, net.minecraft.core.Direction $$1) {
+        this.directions = (byte)(this.directions | ($$0 | 1 << $$1.ordinal()));
     }
 
-    public boolean hasDirection(net.minecraft.core.Direction facing) {
-        return (this.directions & 1 << facing.ordinal()) > 0;
+    public boolean hasDirection(net.minecraft.core.Direction $$0) {
+        return (this.directions & 1 << $$0.ordinal()) > 0;
     }
 
     public void addSourceDirection(net.minecraft.core.Direction $$0) {