--- net/minecraft/client/renderer/OutlineBufferSource.java
+++ net/minecraft/client/renderer/OutlineBufferSource.java
@@ -37,11 +37,11 @@
         return $$2;
     }
 
-    public void setColor(int red, int green, int blue, int alpha) {
-        this.teamR = red;
-        this.teamG = green;
-        this.teamB = blue;
-        this.teamA = alpha;
+    public void setColor(int $$0, int $$1, int $$2, int $$3) {
+        this.teamR = $$0;
+        this.teamG = $$1;
+        this.teamB = $$2;
+        this.teamA = $$3;
     }
 
     public void endOutlineBatch() {