--- net/minecraft/client/renderer/Rect2i.java
+++ net/minecraft/client/renderer/Rect2i.java
@@ -10,11 +10,11 @@
     public int width;
     public int height;
 
-    public net.minecraft.client.renderer.Rect2i(int xPos, int yPos, int width, int height) {
-        this.xPos = xPos;
-        this.yPos = yPos;
-        this.width = width;
-        this.height = height;
+    public net.minecraft.client.renderer.Rect2i(int $$0, int $$1, int $$2, int $$3) {
+        this.xPos = $$0;
+        this.yPos = $$1;
+        this.width = $$2;
+        this.height = $$3;
     }
 
     public net.minecraft.client.renderer.Rect2i intersect(net.minecraft.client.renderer.Rect2i $$0) {
@@ -70,8 +70,8 @@
         this.yPos = $$1;
     }
 
-    public boolean contains(int x, int y) {
-        return x >= this.xPos && x <= this.xPos + this.width && y >= this.yPos && y <= this.yPos + this.height;
+    public boolean contains(int $$0, int $$1) {
+        return $$0 >= this.xPos && $$0 <= this.xPos + this.width && $$1 >= this.yPos && $$1 <= this.yPos + this.height;
     }
 }
 