--- net/minecraft/client/renderer/RenderBuffers.java
+++ net/minecraft/client/renderer/RenderBuffers.java
@@ -22,6 +22,7 @@
             net.minecraft.client.renderer.RenderBuffers.put((it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap)$$0, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.Sheets.bedSheet());
             net.minecraft.client.renderer.RenderBuffers.put((it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap)$$0, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.Sheets.shulkerBoxSheet());
             net.minecraft.client.renderer.RenderBuffers.put((it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap)$$0, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.Sheets.signSheet());
+            net.minecraft.client.renderer.RenderBuffers.put((it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap)$$0, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.Sheets.hangingSignSheet());
             net.minecraft.client.renderer.RenderBuffers.put((it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap)$$0, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.Sheets.chestSheet());
             net.minecraft.client.renderer.RenderBuffers.put((it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap)$$0, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.RenderType.translucentNoCrumbling());
             net.minecraft.client.renderer.RenderBuffers.put((it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap)$$0, (net.minecraft.client.renderer.RenderType)net.minecraft.client.renderer.RenderType.armorGlint());
@@ -39,8 +40,8 @@
         this.outlineBufferSource = new net.minecraft.client.renderer.OutlineBufferSource(this.bufferSource);
     }
 
-    public static void put(it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<net.minecraft.client.renderer.RenderType, com.mojang.blaze3d.vertex.BufferBuilder> mapBuilders, net.minecraft.client.renderer.RenderType renderType) {
-        mapBuilders.put((java.lang.Object)renderType, (java.lang.Object)new com.mojang.blaze3d.vertex.BufferBuilder(renderType.bufferSize()));
+    public static void put(it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<net.minecraft.client.renderer.RenderType, com.mojang.blaze3d.vertex.BufferBuilder> $$0, net.minecraft.client.renderer.RenderType $$1) {
+        $$0.put((java.lang.Object)$$1, (java.lang.Object)new com.mojang.blaze3d.vertex.BufferBuilder($$1.bufferSize()));
     }
 
     public net.minecraft.client.renderer.ChunkBufferBuilderPack fixedBufferPack() {