--- net/minecraft/client/renderer/RenderStateShard.java
+++ net/minecraft/client/renderer/RenderStateShard.java
@@ -115,14 +115,14 @@
         return this.name;
     }
 
-    public static void setupGlintTexturing(float scale) {
-        long l = net.minecraft.Util.getMillis() * 8L;
-        float f = (float)(l % 110000L) / 110000.0f;
-        float f2 = (float)(l % 30000L) / 30000.0f;
-        com.mojang.math.Matrix4f matrix4f = com.mojang.math.Matrix4f.createTranslateMatrix((float)(-f), (float)f2, (float)0.0f);
-        matrix4f.multiply(com.mojang.math.Vector3f.ZP.rotationDegrees(10.0f));
-        matrix4f.multiply(com.mojang.math.Matrix4f.createScaleMatrix((float)scale, (float)scale, (float)scale));
-        com.mojang.blaze3d.systems.RenderSystem.setTextureMatrix((com.mojang.math.Matrix4f)matrix4f);
+    public static void setupGlintTexturing(float $$0) {
+        long $$1 = net.minecraft.Util.getMillis() * 8L;
+        float $$2 = (float)($$1 % 110000L) / 110000.0f;
+        float $$3 = (float)($$1 % 30000L) / 30000.0f;
+        com.mojang.math.Matrix4f $$4 = com.mojang.math.Matrix4f.createTranslateMatrix((float)(-$$2), (float)$$3, (float)0.0f);
+        $$4.multiply(com.mojang.math.Vector3f.ZP.rotationDegrees(10.0f));
+        $$4.multiply(com.mojang.math.Matrix4f.createScaleMatrix((float)$$0, (float)$$0, (float)$$0));
+        com.mojang.blaze3d.systems.RenderSystem.setTextureMatrix((com.mojang.math.Matrix4f)$$4);
     }
 
     public static /* synthetic */ void lambda$static$37() {