--- net/minecraft/client/renderer/VirtualScreen.java
+++ net/minecraft/client/renderer/VirtualScreen.java
@@ -9,13 +9,13 @@
     public net.minecraft.client.Minecraft minecraft;
     public com.mojang.blaze3d.platform.ScreenManager screenManager;
 
-    public net.minecraft.client.renderer.VirtualScreen(net.minecraft.client.Minecraft minecraft) {
-        this.minecraft = minecraft;
+    public net.minecraft.client.renderer.VirtualScreen(net.minecraft.client.Minecraft $$0) {
+        this.minecraft = $$0;
         this.screenManager = new com.mojang.blaze3d.platform.ScreenManager(com.mojang.blaze3d.platform.Monitor::new);
     }
 
-    public com.mojang.blaze3d.platform.Window newWindow(com.mojang.blaze3d.platform.DisplayData screenSize, @javax.annotation.Nullable java.lang.String videoModeName, java.lang.String title) {
-        return new com.mojang.blaze3d.platform.Window((com.mojang.blaze3d.platform.WindowEventHandler)this.minecraft, this.screenManager, screenSize, videoModeName, title);
+    public com.mojang.blaze3d.platform.Window newWindow(com.mojang.blaze3d.platform.DisplayData $$0, @javax.annotation.Nullable java.lang.String $$1, java.lang.String $$2) {
+        return new com.mojang.blaze3d.platform.Window((com.mojang.blaze3d.platform.WindowEventHandler)this.minecraft, this.screenManager, $$0, $$1, $$2);
     }
 
     public void close() {