--- net/minecraft/world/level/pathfinder/PathFinder.java
+++ net/minecraft/world/level/pathfinder/PathFinder.java
@@ -20,17 +20,17 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.pathfinder.Path findPath(net.minecraft.world.level.PathNavigationRegion region, net.minecraft.world.entity.Mob mob, java.util.Set<net.minecraft.core.BlockPos> targetPositions, float maxRange, int accuracy, float searchDepthMultiplier) {
+    public net.minecraft.world.level.pathfinder.Path findPath(net.minecraft.world.level.PathNavigationRegion $$02, net.minecraft.world.entity.Mob $$1, java.util.Set<net.minecraft.core.BlockPos> $$2, float $$3, int $$4, float $$5) {
         this.openSet.clear();
-        this.nodeEvaluator.prepare(region, mob);
-        net.minecraft.world.level.pathfinder.Node node = this.nodeEvaluator.getStart();
-        if (node == null) {
+        this.nodeEvaluator.prepare($$02, $$1);
+        net.minecraft.world.level.pathfinder.Node $$6 = this.nodeEvaluator.getStart();
+        if ($$6 == null) {
             return null;
         }
-        java.util.Map map = (java.util.Map)targetPositions.stream().collect(java.util.stream.Collectors.toMap($$0 -> this.nodeEvaluator.getGoal((double)$$0.getX(), (double)$$0.getY(), (double)$$0.getZ()), (java.util.function.Function)java.util.function.Function.identity()));
-        net.minecraft.world.level.pathfinder.Path path = this.findPath(region.getProfiler(), node, map, maxRange, accuracy, searchDepthMultiplier);
+        java.util.Map $$7 = (java.util.Map)$$2.stream().collect(java.util.stream.Collectors.toMap($$0 -> this.nodeEvaluator.getGoal((double)$$0.getX(), (double)$$0.getY(), (double)$$0.getZ()), (java.util.function.Function)java.util.function.Function.identity()));
+        net.minecraft.world.level.pathfinder.Path $$8 = this.findPath($$02.getProfiler(), $$6, $$7, $$3, $$4, $$5);
         this.nodeEvaluator.done();
-        return path;
+        return $$8;
     }
 
     @javax.annotation.Nullable
@@ -97,15 +97,15 @@
         return $$2;
     }
 
-    public net.minecraft.world.level.pathfinder.Path reconstructPath(net.minecraft.world.level.pathfinder.Node point, net.minecraft.core.BlockPos targetPos, boolean reachesTarget) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        net.minecraft.world.level.pathfinder.Node node = point;
-        arrayList.add(0, (java.lang.Object)node);
-        while (node.cameFrom != null) {
-            node = node.cameFrom;
-            arrayList.add(0, (java.lang.Object)node);
+    public net.minecraft.world.level.pathfinder.Path reconstructPath(net.minecraft.world.level.pathfinder.Node $$0, net.minecraft.core.BlockPos $$1, boolean $$2) {
+        java.util.ArrayList $$3 = com.google.common.collect.Lists.newArrayList();
+        net.minecraft.world.level.pathfinder.Node $$4 = $$0;
+        $$3.add(0, (java.lang.Object)$$4);
+        while ($$4.cameFrom != null) {
+            $$4 = $$4.cameFrom;
+            $$3.add(0, (java.lang.Object)$$4);
         }
-        return new net.minecraft.world.level.pathfinder.Path((java.util.List)arrayList, targetPos, reachesTarget);
+        return new net.minecraft.world.level.pathfinder.Path((java.util.List)$$3, $$1, $$2);
     }
 }
 