--- net/minecraft/server/level/progress/ProcessorChunkProgressListener.java
+++ net/minecraft/server/level/progress/ProcessorChunkProgressListener.java
@@ -20,12 +20,12 @@
         return $$2;
     }
 
-    public void updateSpawnPos(net.minecraft.world.level.ChunkPos center) {
-        this.mailbox.tell(() -> this.delegate.updateSpawnPos(center));
+    public void updateSpawnPos(net.minecraft.world.level.ChunkPos $$0) {
+        this.mailbox.tell(() -> this.delegate.updateSpawnPos($$0));
     }
 
-    public void onStatusChange(net.minecraft.world.level.ChunkPos chunkPosition, @javax.annotation.Nullable net.minecraft.world.level.chunk.ChunkStatus newStatus) {
-        this.mailbox.tell(() -> this.delegate.onStatusChange(chunkPosition, newStatus));
+    public void onStatusChange(net.minecraft.world.level.ChunkPos $$0, @javax.annotation.Nullable net.minecraft.world.level.chunk.ChunkStatus $$1) {
+        this.mailbox.tell(() -> this.delegate.onStatusChange($$0, $$1));
     }
 
     public void start() {