--- net/minecraft/server/level/progress/StoringChunkProgressListener.java
+++ net/minecraft/server/level/progress/StoringChunkProgressListener.java
@@ -23,23 +23,23 @@
         this.statuses = new it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap();
     }
 
-    public void updateSpawnPos(net.minecraft.world.level.ChunkPos center) {
+    public void updateSpawnPos(net.minecraft.world.level.ChunkPos $$0) {
         if (!this.started) {
             return;
         }
-        this.delegate.updateSpawnPos(center);
-        this.spawnPos = center;
+        this.delegate.updateSpawnPos($$0);
+        this.spawnPos = $$0;
     }
 
-    public void onStatusChange(net.minecraft.world.level.ChunkPos chunkPosition, @javax.annotation.Nullable net.minecraft.world.level.chunk.ChunkStatus newStatus) {
+    public void onStatusChange(net.minecraft.world.level.ChunkPos $$0, @javax.annotation.Nullable net.minecraft.world.level.chunk.ChunkStatus $$1) {
         if (!this.started) {
             return;
         }
-        this.delegate.onStatusChange(chunkPosition, newStatus);
-        if (newStatus == null) {
-            this.statuses.remove(chunkPosition.toLong());
+        this.delegate.onStatusChange($$0, $$1);
+        if ($$1 == null) {
+            this.statuses.remove($$0.toLong());
         } else {
-            this.statuses.put(chunkPosition.toLong(), (java.lang.Object)newStatus);
+            this.statuses.put($$0.toLong(), (java.lang.Object)$$1);
         }
     }
 
@@ -67,8 +67,8 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.chunk.ChunkStatus getStatus(int x, int z) {
-        return (net.minecraft.world.level.chunk.ChunkStatus)this.statuses.get(net.minecraft.world.level.ChunkPos.asLong((int)(x + this.spawnPos.x - this.radius), (int)(z + this.spawnPos.z - this.radius)));
+    public net.minecraft.world.level.chunk.ChunkStatus getStatus(int $$0, int $$1) {
+        return (net.minecraft.world.level.chunk.ChunkStatus)this.statuses.get(net.minecraft.world.level.ChunkPos.asLong((int)($$0 + this.spawnPos.x - this.radius), (int)($$1 + this.spawnPos.z - this.radius)));
     }
 }
 