--- net/minecraft/client/gui/screens/debug/GameModeSwitcherScreen$GameModeSlot.java
+++ net/minecraft/client/gui/screens/debug/GameModeSwitcherScreen$GameModeSlot.java
@@ -16,12 +16,12 @@
         this.icon = $$1;
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        net.minecraft.client.Minecraft minecraft = net.minecraft.client.Minecraft.getInstance();
-        this.drawSlot(matrixStack, minecraft.getTextureManager());
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        net.minecraft.client.Minecraft $$4 = net.minecraft.client.Minecraft.getInstance();
+        this.drawSlot($$0, $$4.getTextureManager());
         this.icon.drawIcon(net.minecraft.client.gui.screens.debug.GameModeSwitcherScreen.access$000((net.minecraft.client.gui.screens.debug.GameModeSwitcherScreen)this.this$0), this.x + 5, this.y + 5);
         if (this.isSelected) {
-            this.drawSelection(matrixStack, minecraft.getTextureManager());
+            this.drawSelection($$0, $$4.getTextureManager());
         }
     }
 