--- net/minecraft/world/entity/ai/gossip/GossipContainer$EntityGossips.java
+++ net/minecraft/world/entity/ai/gossip/GossipContainer$EntityGossips.java
@@ -11,12 +11,12 @@
         this.entries = new it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap();
     }
 
-    public int weightedValue(java.util.function.Predicate<net.minecraft.world.entity.ai.gossip.GossipType> gossipType) {
-        return this.entries.object2IntEntrySet().stream().filter($$1 -> gossipType.test((java.lang.Object)((net.minecraft.world.entity.ai.gossip.GossipType)$$1.getKey()))).mapToInt($$0 -> $$0.getIntValue() * ((net.minecraft.world.entity.ai.gossip.GossipType)$$0.getKey()).weight).sum();
+    public int weightedValue(java.util.function.Predicate<net.minecraft.world.entity.ai.gossip.GossipType> $$02) {
+        return this.entries.object2IntEntrySet().stream().filter($$1 -> $$02.test((java.lang.Object)((net.minecraft.world.entity.ai.gossip.GossipType)$$1.getKey()))).mapToInt($$0 -> $$0.getIntValue() * ((net.minecraft.world.entity.ai.gossip.GossipType)$$0.getKey()).weight).sum();
     }
 
-    public java.util.stream.Stream<net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry> unpack(java.util.UUID identifier) {
-        return this.entries.object2IntEntrySet().stream().map($$1 -> new net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry(identifier, (net.minecraft.world.entity.ai.gossip.GossipType)$$1.getKey(), $$1.getIntValue()));
+    public java.util.stream.Stream<net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry> unpack(java.util.UUID $$0) {
+        return this.entries.object2IntEntrySet().stream().map($$1 -> new net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry($$0, (net.minecraft.world.entity.ai.gossip.GossipType)$$1.getKey(), $$1.getIntValue()));
     }
 
     public void decay() {
@@ -36,18 +36,18 @@
         return this.entries.isEmpty();
     }
 
-    public void makeSureValueIsntTooLowOrTooHigh(net.minecraft.world.entity.ai.gossip.GossipType gossipType) {
-        int n = this.entries.getInt((java.lang.Object)gossipType);
-        if (n > gossipType.max) {
-            this.entries.put((java.lang.Object)gossipType, gossipType.max);
+    public void makeSureValueIsntTooLowOrTooHigh(net.minecraft.world.entity.ai.gossip.GossipType $$0) {
+        int $$1 = this.entries.getInt((java.lang.Object)$$0);
+        if ($$1 > $$0.max) {
+            this.entries.put((java.lang.Object)$$0, $$0.max);
         }
-        if (n < 2) {
-            this.remove(gossipType);
+        if ($$1 < 2) {
+            this.remove($$0);
         }
     }
 
-    public void remove(net.minecraft.world.entity.ai.gossip.GossipType gossipType) {
-        this.entries.removeInt((java.lang.Object)gossipType);
+    public void remove(net.minecraft.world.entity.ai.gossip.GossipType $$0) {
+        this.entries.removeInt((java.lang.Object)$$0);
     }
 }
 