--- net/minecraft/world/entity/ai/gossip/GossipContainer$GossipEntry.java
+++ net/minecraft/world/entity/ai/gossip/GossipContainer$GossipEntry.java
@@ -26,12 +26,12 @@
         return "GossipEntry{target=" + this.target + ", type=" + this.type + ", value=" + this.value + "}";
     }
 
-    public <T> com.mojang.serialization.Dynamic<T> store(com.mojang.serialization.DynamicOps<T> dynamic) {
-        return new com.mojang.serialization.Dynamic(dynamic, dynamic.createMap((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)dynamic.createString("Target"), (java.lang.Object)net.minecraft.core.UUIDUtil.CODEC.encodeStart(dynamic, (java.lang.Object)this.target).result().orElseThrow(java.lang.RuntimeException::new), (java.lang.Object)dynamic.createString("Type"), (java.lang.Object)dynamic.createString(this.type.id), (java.lang.Object)dynamic.createString("Value"), (java.lang.Object)dynamic.createInt(this.value))));
+    public <T> com.mojang.serialization.Dynamic<T> store(com.mojang.serialization.DynamicOps<T> $$0) {
+        return new com.mojang.serialization.Dynamic($$0, $$0.createMap((java.util.Map)com.google.common.collect.ImmutableMap.of((java.lang.Object)$$0.createString("Target"), (java.lang.Object)net.minecraft.core.UUIDUtil.CODEC.encodeStart($$0, (java.lang.Object)this.target).result().orElseThrow(java.lang.RuntimeException::new), (java.lang.Object)$$0.createString("Type"), (java.lang.Object)$$0.createString(this.type.id), (java.lang.Object)$$0.createString("Value"), (java.lang.Object)$$0.createInt(this.value))));
     }
 
-    public static com.mojang.serialization.DataResult<net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry> load(com.mojang.serialization.Dynamic<?> dynamic) {
-        return com.mojang.serialization.DataResult.unbox((com.mojang.datafixers.kinds.App)com.mojang.serialization.DataResult.instance().group((com.mojang.datafixers.kinds.App)dynamic.get("Target").read((com.mojang.serialization.Decoder)net.minecraft.core.UUIDUtil.CODEC), (com.mojang.datafixers.kinds.App)dynamic.get("Type").asString().map(net.minecraft.world.entity.ai.gossip.GossipType::byId), (com.mojang.datafixers.kinds.App)dynamic.get("Value").asNumber().map(java.lang.Number::intValue)).apply((com.mojang.datafixers.kinds.Applicative)com.mojang.serialization.DataResult.instance(), net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry::new));
+    public static com.mojang.serialization.DataResult<net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry> load(com.mojang.serialization.Dynamic<?> $$0) {
+        return com.mojang.serialization.DataResult.unbox((com.mojang.datafixers.kinds.App)com.mojang.serialization.DataResult.instance().group((com.mojang.datafixers.kinds.App)$$0.get("Target").read((com.mojang.serialization.Decoder)net.minecraft.core.UUIDUtil.CODEC), (com.mojang.datafixers.kinds.App)$$0.get("Type").asString().map(net.minecraft.world.entity.ai.gossip.GossipType::byId), (com.mojang.datafixers.kinds.App)$$0.get("Value").read((com.mojang.serialization.Decoder)net.minecraft.util.ExtraCodecs.POSITIVE_INT)).apply((com.mojang.datafixers.kinds.Applicative)com.mojang.serialization.DataResult.instance(), net.minecraft.world.entity.ai.gossip.GossipContainer$GossipEntry::new));
     }
 }
 