--- net/minecraft/world/entity/ai/gossip/GossipType.java
+++ net/minecraft/world/entity/ai/gossip/GossipType.java
@@ -40,8 +40,8 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.entity.ai.gossip.GossipType byId(java.lang.String id) {
-        return (net.minecraft.world.entity.ai.gossip.GossipType)BY_ID.get((java.lang.Object)id);
+    public static net.minecraft.world.entity.ai.gossip.GossipType byId(java.lang.String $$0) {
+        return (net.minecraft.world.entity.ai.gossip.GossipType)BY_ID.get((java.lang.Object)$$0);
     }
 
     public static /* synthetic */ java.lang.String lambda$static$0(net.minecraft.world.entity.ai.gossip.GossipType $$0) {