--- net/minecraft/world/level/levelgen/feature/stateproviders/BlockStateProviderType.java
+++ net/minecraft/world/level/levelgen/feature/stateproviders/BlockStateProviderType.java
@@ -14,12 +14,12 @@
     public static net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType<net.minecraft.world.level.levelgen.feature.stateproviders.RandomizedIntStateProvider> RANDOMIZED_INT_STATE_PROVIDER;
     public com.mojang.serialization.Codec<P> codec;
 
-    public static <P extends net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider> net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType<P> register(java.lang.String name, com.mojang.serialization.Codec<P> codec) {
-        return (net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.BLOCKSTATE_PROVIDER_TYPES, (java.lang.String)name, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType(codec));
+    public static <P extends net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider> net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType<P> register(java.lang.String $$0, com.mojang.serialization.Codec<P> $$1) {
+        return (net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.BLOCKSTATE_PROVIDER_TYPES, (java.lang.String)$$0, (java.lang.Object)new net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType($$1));
     }
 
-    public net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType(com.mojang.serialization.Codec<P> codec) {
-        this.codec = codec;
+    public net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType(com.mojang.serialization.Codec<P> $$0) {
+        this.codec = $$0;
     }
 
     public com.mojang.serialization.Codec<P> codec() {