--- net/minecraft/world/level/levelgen/feature/stateproviders/SimpleStateProvider.java
+++ net/minecraft/world/level/levelgen/feature/stateproviders/SimpleStateProvider.java
@@ -9,8 +9,8 @@
     public static com.mojang.serialization.Codec<net.minecraft.world.level.levelgen.feature.stateproviders.SimpleStateProvider> CODEC;
     public net.minecraft.world.level.block.state.BlockState state;
 
-    public net.minecraft.world.level.levelgen.feature.stateproviders.SimpleStateProvider(net.minecraft.world.level.block.state.BlockState state) {
-        this.state = state;
+    public net.minecraft.world.level.levelgen.feature.stateproviders.SimpleStateProvider(net.minecraft.world.level.block.state.BlockState $$0) {
+        this.state = $$0;
     }
 
     public net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProviderType<?> type() {