--- net/minecraft/client/gui/screens/multiplayer/JoinMultiplayerScreen.java
+++ net/minecraft/client/gui/screens/multiplayer/JoinMultiplayerScreen.java
@@ -85,9 +85,8 @@
 
     public void tick() {
         super.tick();
-        if (this.lanServerList.isDirty()) {
-            java.util.List $$0 = this.lanServerList.getServers();
-            this.lanServerList.markClean();
+        java.util.List $$0 = this.lanServerList.takeDirtyServers();
+        if ($$0 != null) {
             this.serverSelectionList.updateNetworkServers($$0);
         }
         this.pinger.tick();
@@ -161,32 +160,32 @@
         }
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (super.keyPressed(keyCode, scanCode, modifiers)) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if (super.keyPressed($$0, $$1, $$2)) {
             return true;
         }
-        if (keyCode == 294) {
+        if ($$0 == 294) {
             this.refreshServerList();
             return true;
         }
         if (this.serverSelectionList.getSelected() != null) {
-            if (keyCode == 257 || keyCode == 335) {
+            if ($$0 == 257 || $$0 == 335) {
                 this.joinSelectedServer();
                 return true;
             }
-            return this.serverSelectionList.keyPressed(keyCode, scanCode, modifiers);
+            return this.serverSelectionList.keyPressed($$0, $$1, $$2);
         }
         return false;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         this.toolTip = null;
-        this.renderBackground(matrixStack);
-        this.serverSelectionList.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.multiplayer.JoinMultiplayerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        this.renderBackground($$0);
+        this.serverSelectionList.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.multiplayer.JoinMultiplayerScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
         if (this.toolTip != null) {
-            this.renderComponentTooltip(matrixStack, this.toolTip, mouseX, mouseY);
+            this.renderComponentTooltip($$0, this.toolTip, $$1, $$2);
         }
     }
 
@@ -200,8 +199,8 @@
         }
     }
 
-    public void join(net.minecraft.client.multiplayer.ServerData server) {
-        net.minecraft.client.gui.screens.ConnectScreen.startConnecting((net.minecraft.client.gui.screens.Screen)this, (net.minecraft.client.Minecraft)this.minecraft, (net.minecraft.client.multiplayer.resolver.ServerAddress)net.minecraft.client.multiplayer.resolver.ServerAddress.parseString((java.lang.String)server.ip), (net.minecraft.client.multiplayer.ServerData)server);
+    public void join(net.minecraft.client.multiplayer.ServerData $$0) {
+        net.minecraft.client.gui.screens.ConnectScreen.startConnecting((net.minecraft.client.gui.screens.Screen)this, (net.minecraft.client.Minecraft)this.minecraft, (net.minecraft.client.multiplayer.resolver.ServerAddress)net.minecraft.client.multiplayer.resolver.ServerAddress.parseString((java.lang.String)$$0.ip), (net.minecraft.client.multiplayer.ServerData)$$0);
     }
 
     public void setSelected(net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry $$0) {