--- net/minecraft/client/gui/screens/multiplayer/ServerSelectionList.java
+++ net/minecraft/client/gui/screens/multiplayer/ServerSelectionList.java
@@ -36,18 +36,18 @@
         this.networkServers.forEach($$1 -> this.addEntry($$1));
     }
 
-    public void setSelected(@javax.annotation.Nullable net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry entry) {
-        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)entry);
+    public void setSelected(@javax.annotation.Nullable net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry $$0) {
+        super.setSelected((net.minecraft.client.gui.components.AbstractSelectionList$Entry)$$0);
         this.screen.onSelectedChange();
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry entry = (net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry)this.getSelected();
-        return entry != null && entry.keyPressed(keyCode, scanCode, modifiers) || super.keyPressed(keyCode, scanCode, modifiers);
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry $$3 = (net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry)this.getSelected();
+        return $$3 != null && $$3.keyPressed($$0, $$1, $$2) || super.keyPressed($$0, $$1, $$2);
     }
 
-    public void moveSelection(net.minecraft.client.gui.components.AbstractSelectionList$SelectionDirection ordering) {
-        this.moveSelection(ordering, $$0 -> !($$0 instanceof net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$LANHeader));
+    public void moveSelection(net.minecraft.client.gui.components.AbstractSelectionList$SelectionDirection $$02) {
+        this.moveSelection($$02, $$0 -> !($$0 instanceof net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$LANHeader));
     }
 
     public void updateOnlineServers(net.minecraft.client.multiplayer.ServerList $$0) {