--- net/minecraft/client/gui/screens/multiplayer/ServerSelectionList$LANHeader.java
+++ net/minecraft/client/gui/screens/multiplayer/ServerSelectionList$LANHeader.java
@@ -12,15 +12,15 @@
         this.minecraft = net.minecraft.client.Minecraft.getInstance();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        int n = top + height / 2;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        int n = $$2 + $$5 / 2;
         java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-        int n2 = n - 9 / 2;
-        this.minecraft.font.draw(matrixStack, net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.SCANNING_LABEL, (float)(this.minecraft.screen.width / 2 - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.SCANNING_LABEL) / 2), (float)n2, 0xFFFFFF);
-        java.lang.String string = net.minecraft.client.gui.screens.LoadingDotsText.get((long)net.minecraft.Util.getMillis());
-        float f = this.minecraft.screen.width / 2 - this.minecraft.font.width(string) / 2;
+        int $$10 = n - 9 / 2;
+        this.minecraft.font.draw($$0, net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.SCANNING_LABEL, (float)(this.minecraft.screen.width / 2 - this.minecraft.font.width((net.minecraft.network.chat.FormattedText)net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.SCANNING_LABEL) / 2), (float)$$10, 0xFFFFFF);
+        java.lang.String $$11 = net.minecraft.client.gui.screens.LoadingDotsText.get((long)net.minecraft.Util.getMillis());
+        float f = this.minecraft.screen.width / 2 - this.minecraft.font.width($$11) / 2;
         java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-        this.minecraft.font.draw(matrixStack, string, f, (float)(n2 + 9), 0x808080);
+        this.minecraft.font.draw($$0, $$11, f, (float)($$10 + 9), 0x808080);
     }
 
     public net.minecraft.network.chat.Component getNarration() {