--- net/minecraft/client/gui/screens/multiplayer/ServerSelectionList$NetworkServerEntry.java
+++ net/minecraft/client/gui/screens/multiplayer/ServerSelectionList$NetworkServerEntry.java
@@ -20,17 +20,17 @@
         this.minecraft = net.minecraft.client.Minecraft.getInstance();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.minecraft.font.draw(matrixStack, LAN_SERVER_HEADER, (float)(left + 32 + 3), (float)(top + 1), 0xFFFFFF);
-        this.minecraft.font.draw(matrixStack, this.serverData.getMotd(), (float)(left + 32 + 3), (float)(top + 12), 0x808080);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        this.minecraft.font.draw($$0, LAN_SERVER_HEADER, (float)($$3 + 32 + 3), (float)($$2 + 1), 0xFFFFFF);
+        this.minecraft.font.draw($$0, this.serverData.getMotd(), (float)($$3 + 32 + 3), (float)($$2 + 12), 0x808080);
         if (this.minecraft.options.hideServerAddress) {
-            this.minecraft.font.draw(matrixStack, HIDDEN_ADDRESS_TEXT, (float)(left + 32 + 3), (float)(top + 12 + 11), 0x303030);
+            this.minecraft.font.draw($$0, HIDDEN_ADDRESS_TEXT, (float)($$3 + 32 + 3), (float)($$2 + 12 + 11), 0x303030);
         } else {
-            this.minecraft.font.draw(matrixStack, this.serverData.getAddress(), (float)(left + 32 + 3), (float)(top + 12 + 11), 0x303030);
+            this.minecraft.font.draw($$0, this.serverData.getAddress(), (float)($$3 + 32 + 3), (float)($$2 + 12 + 11), 0x303030);
         }
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
         this.screen.setSelected((net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry)this);
         if (net.minecraft.Util.getMillis() - this.lastClickTime < 250L) {
             this.screen.joinSelectedServer();