--- net/minecraft/client/gui/screens/multiplayer/ServerSelectionList$OnlineServerEntry.java
+++ net/minecraft/client/gui/screens/multiplayer/ServerSelectionList$OnlineServerEntry.java
@@ -37,10 +37,10 @@
         }
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        java.util.List list;
-        net.minecraft.network.chat.Component component;
-        int n;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        java.util.List $$31;
+        net.minecraft.network.chat.Component $$30;
+        int $$29;
         if (!this.serverData.pinged) {
             this.serverData.pinged = true;
             this.serverData.ping = -2L;
@@ -60,92 +60,104 @@
                 }
             });
         }
-        boolean bl = this.serverData.protocol != net.minecraft.SharedConstants.getCurrentVersion().getProtocolVersion();
-        this.minecraft.font.draw(matrixStack, this.serverData.name, (float)(left + 32 + 3), (float)(top + 1), 0xFFFFFF);
-        java.util.List list2 = this.minecraft.font.split((net.minecraft.network.chat.FormattedText)this.serverData.motd, width - 32 - 2);
-        for (int i = 0; i < java.lang.Math.min((int)list2.size(), (int)2); ++i) {
-            net.minecraft.util.FormattedCharSequence formattedCharSequence = (net.minecraft.util.FormattedCharSequence)list2.get(i);
-            float f = left + 32 + 3;
+        boolean $$10 = this.serverData.protocol != net.minecraft.SharedConstants.getCurrentVersion().getProtocolVersion();
+        this.minecraft.font.draw($$0, this.serverData.name, (float)($$3 + 32 + 3), (float)($$2 + 1), 0xFFFFFF);
+        java.util.List $$11 = this.minecraft.font.split((net.minecraft.network.chat.FormattedText)this.serverData.motd, $$4 - 32 - 2);
+        for (int $$12 = 0; $$12 < java.lang.Math.min((int)$$11.size(), (int)2); ++$$12) {
+            net.minecraft.util.FormattedCharSequence formattedCharSequence = (net.minecraft.util.FormattedCharSequence)$$11.get($$12);
+            float f = $$3 + 32 + 3;
             java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-            this.minecraft.font.draw(matrixStack, formattedCharSequence, f, (float)(top + 12 + 9 * i), 0x808080);
+            this.minecraft.font.draw($$0, formattedCharSequence, f, (float)($$2 + 12 + 9 * $$12), 0x808080);
         }
-        net.minecraft.network.chat.Component component2 = bl ? this.serverData.version.copy().withStyle(net.minecraft.ChatFormatting.RED) : this.serverData.status;
-        int n2 = this.minecraft.font.width((net.minecraft.network.chat.FormattedText)component2);
-        this.minecraft.font.draw(matrixStack, component2, (float)(left + width - n2 - 15 - 2), (float)(top + 1), 0x808080);
-        int n3 = 0;
-        if (bl) {
-            n = 5;
-            component = net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.INCOMPATIBLE_TOOLTIP;
-            list = this.serverData.playerList;
+        net.minecraft.network.chat.Component $$13 = $$10 ? this.serverData.version.copy().withStyle(net.minecraft.ChatFormatting.RED) : this.serverData.status;
+        int $$14 = this.minecraft.font.width((net.minecraft.network.chat.FormattedText)$$13);
+        this.minecraft.font.draw($$0, $$13, (float)($$3 + $$4 - $$14 - 15 - 2), (float)($$2 + 1), 0x808080);
+        int $$15 = 0;
+        if ($$10) {
+            int $$16 = 5;
+            net.minecraft.network.chat.Component $$17 = net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.INCOMPATIBLE_TOOLTIP;
+            java.util.List $$18 = this.serverData.playerList;
         } else if (this.serverData.pinged && this.serverData.ping != -2L) {
-            n = this.serverData.ping < 0L ? 5 : (this.serverData.ping < 150L ? 0 : (this.serverData.ping < 300L ? 1 : (this.serverData.ping < 600L ? 2 : (this.serverData.ping < 1000L ? 3 : 4))));
             if (this.serverData.ping < 0L) {
-                component = net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.NO_CONNECTION_TOOLTIP;
-                list = java.util.Collections.emptyList();
+                int $$19 = 5;
+            } else if (this.serverData.ping < 150L) {
+                boolean $$20 = false;
+            } else if (this.serverData.ping < 300L) {
+                boolean $$21 = true;
+            } else if (this.serverData.ping < 600L) {
+                int $$22 = 2;
+            } else if (this.serverData.ping < 1000L) {
+                int $$23 = 3;
             } else {
-                component = net.minecraft.network.chat.Component.translatable((java.lang.String)"multiplayer.status.ping", (java.lang.Object[])new java.lang.Object[]{this.serverData.ping});
-                list = this.serverData.playerList;
+                int $$24 = 4;
             }
+            if (this.serverData.ping < 0L) {
+                net.minecraft.network.chat.Component $$25 = net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.NO_CONNECTION_TOOLTIP;
+                java.util.List $$26 = java.util.Collections.emptyList();
+            } else {
+                net.minecraft.network.chat.MutableComponent $$27 = net.minecraft.network.chat.Component.translatable((java.lang.String)"multiplayer.status.ping", (java.lang.Object[])new java.lang.Object[]{this.serverData.ping});
+                java.util.List $$28 = this.serverData.playerList;
+            }
         } else {
-            n3 = 1;
-            n = (int)(net.minecraft.Util.getMillis() / 100L + (long)(index * 2) & 7L);
-            if (n > 4) {
-                n = 8 - n;
+            $$15 = 1;
+            $$29 = (int)(net.minecraft.Util.getMillis() / 100L + (long)($$1 * 2) & 7L);
+            if ($$29 > 4) {
+                $$29 = 8 - $$29;
             }
-            component = net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.PINGING_TOOLTIP;
-            list = java.util.Collections.emptyList();
+            $$30 = net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.PINGING_TOOLTIP;
+            $$31 = java.util.Collections.emptyList();
         }
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)net.minecraft.client.gui.GuiComponent.GUI_ICONS_LOCATION);
-        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)(left + width - 15), (int)top, (float)(n3 * 10), (float)(176 + n * 8), (int)10, (int)8, (int)256, (int)256);
-        java.lang.String string = this.serverData.getIconB64();
-        if (!java.util.Objects.equals((java.lang.Object)string, (java.lang.Object)this.lastIconB64)) {
-            if (this.uploadServerIcon(string)) {
-                this.lastIconB64 = string;
+        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$3 + $$4 - 15), (int)$$2, (float)($$15 * 10), (float)(176 + $$29 * 8), (int)10, (int)8, (int)256, (int)256);
+        java.lang.String $$32 = this.serverData.getIconB64();
+        if (!java.util.Objects.equals((java.lang.Object)$$32, (java.lang.Object)this.lastIconB64)) {
+            if (this.uploadServerIcon($$32)) {
+                this.lastIconB64 = $$32;
             } else {
                 this.serverData.setIconB64(null);
                 this.updateServerList();
             }
         }
         if (this.icon == null) {
-            this.drawIcon(matrixStack, left, top, net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.ICON_MISSING);
+            this.drawIcon($$0, $$3, $$2, net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.ICON_MISSING);
         } else {
-            this.drawIcon(matrixStack, left, top, this.iconLocation);
+            this.drawIcon($$0, $$3, $$2, this.iconLocation);
         }
-        int n4 = mouseX - left;
-        int n5 = mouseY - top;
-        if (n4 >= width - 15 && n4 <= width - 5 && n5 >= 0 && n5 <= 8) {
-            this.screen.setToolTip(java.util.Collections.singletonList((java.lang.Object)component));
-        } else if (n4 >= width - n2 - 15 - 2 && n4 <= width - 15 - 2 && n5 >= 0 && n5 <= 8) {
-            this.screen.setToolTip(list);
+        int $$33 = $$6 - $$3;
+        int $$34 = $$7 - $$2;
+        if ($$33 >= $$4 - 15 && $$33 <= $$4 - 5 && $$34 >= 0 && $$34 <= 8) {
+            this.screen.setToolTip(java.util.Collections.singletonList((java.lang.Object)$$30));
+        } else if ($$33 >= $$4 - $$14 - 15 - 2 && $$33 <= $$4 - 15 - 2 && $$34 >= 0 && $$34 <= 8) {
+            this.screen.setToolTip($$31);
         }
-        if (((java.lang.Boolean)this.minecraft.options.touchscreen().get()).booleanValue() || isMouseOver) {
+        if (((java.lang.Boolean)this.minecraft.options.touchscreen().get()).booleanValue() || $$8) {
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.ICON_OVERLAY_LOCATION);
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (int)(left + 32), (int)(top + 32), (int)-1601138544);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (int)($$3 + 32), (int)($$2 + 32), (int)-1601138544);
             com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
             com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-            int n6 = mouseX - left;
-            int n7 = mouseY - top;
+            int $$35 = $$6 - $$3;
+            int $$36 = $$7 - $$2;
             if (this.canJoin()) {
-                if (n6 < 32 && n6 > 16) {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)0.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
+                if ($$35 < 32 && $$35 > 16) {
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)0.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
                 } else {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
                 }
             }
-            if (index > 0) {
-                if (n6 < 16 && n7 < 16) {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)96.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
+            if ($$1 > 0) {
+                if ($$35 < 16 && $$36 < 16) {
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)96.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
                 } else {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)96.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)96.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
                 }
             }
-            if (index < this.screen.getServers().size() - 1) {
-                if (n6 < 16 && n7 > 16) {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)64.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
+            if ($$1 < this.screen.getServers().size() - 1) {
+                if ($$35 < 16 && $$36 > 16) {
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)64.0f, (float)32.0f, (int)32, (int)32, (int)256, (int)256);
                 } else {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)64.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)64.0f, (float)0.0f, (int)32, (int)32, (int)256, (int)256);
                 }
             }
         }
@@ -194,19 +206,19 @@
         return true;
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
         if (net.minecraft.client.gui.screens.Screen.hasShiftDown()) {
-            net.minecraft.client.gui.screens.multiplayer.ServerSelectionList serverSelectionList = this.screen.serverSelectionList;
-            int n = serverSelectionList.children().indexOf((java.lang.Object)this);
-            if (n == -1) {
+            net.minecraft.client.gui.screens.multiplayer.ServerSelectionList $$3 = this.screen.serverSelectionList;
+            int $$4 = $$3.children().indexOf((java.lang.Object)this);
+            if ($$4 == -1) {
                 return true;
             }
-            if (keyCode == 264 && n < this.screen.getServers().size() - 1 || keyCode == 265 && n > 0) {
-                this.swap(n, keyCode == 264 ? n + 1 : n - 1);
+            if ($$0 == 264 && $$4 < this.screen.getServers().size() - 1 || $$0 == 265 && $$4 > 0) {
+                this.swap($$4, $$0 == 264 ? $$4 + 1 : $$4 - 1);
                 return true;
             }
         }
-        return super.keyPressed(keyCode, scanCode, modifiers);
+        return super.keyPressed($$0, $$1, $$2);
     }
 
     public void swap(int $$0, int $$1) {
@@ -217,22 +229,22 @@
         net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.access$000((net.minecraft.client.gui.screens.multiplayer.ServerSelectionList)this.this$0, (net.minecraft.client.gui.components.AbstractSelectionList$Entry)$$2);
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
-        double d = mouseX - (double)this.this$0.getRowLeft();
-        double d2 = mouseY - (double)net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.access$100((net.minecraft.client.gui.screens.multiplayer.ServerSelectionList)this.this$0, (int)this.this$0.children().indexOf((java.lang.Object)this));
-        if (d <= 32.0) {
-            if (d < 32.0 && d > 16.0 && this.canJoin()) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
+        double $$3 = $$0 - (double)this.this$0.getRowLeft();
+        double $$4 = $$1 - (double)net.minecraft.client.gui.screens.multiplayer.ServerSelectionList.access$100((net.minecraft.client.gui.screens.multiplayer.ServerSelectionList)this.this$0, (int)this.this$0.children().indexOf((java.lang.Object)this));
+        if ($$3 <= 32.0) {
+            if ($$3 < 32.0 && $$3 > 16.0 && this.canJoin()) {
                 this.screen.setSelected((net.minecraft.client.gui.screens.multiplayer.ServerSelectionList$Entry)this);
                 this.screen.joinSelectedServer();
                 return true;
             }
-            int n = this.screen.serverSelectionList.children().indexOf((java.lang.Object)this);
-            if (d < 16.0 && d2 < 16.0 && n > 0) {
-                this.swap(n, n - 1);
+            int $$5 = this.screen.serverSelectionList.children().indexOf((java.lang.Object)this);
+            if ($$3 < 16.0 && $$4 < 16.0 && $$5 > 0) {
+                this.swap($$5, $$5 - 1);
                 return true;
             }
-            if (d < 16.0 && d2 > 16.0 && n < this.screen.getServers().size() - 1) {
-                this.swap(n, n + 1);
+            if ($$3 < 16.0 && $$4 > 16.0 && $$5 < this.screen.getServers().size() - 1) {
+                this.swap($$5, $$5 + 1);
                 return true;
             }
         }