--- net/minecraft/client/renderer/entity/AbstractZombieRenderer.java
+++ net/minecraft/client/renderer/entity/AbstractZombieRenderer.java
@@ -13,7 +13,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$2, $$3));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zombie entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zombie $$0) {
         return ZOMBIE_LOCATION;
     }
 