--- net/minecraft/client/renderer/entity/ArmorStandRenderer.java
+++ net/minecraft/client/renderer/entity/ArmorStandRenderer.java
@@ -16,26 +16,26 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.CustomHeadLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet(), $$0.getItemInHandRenderer()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.decoration.ArmorStand entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.decoration.ArmorStand $$0) {
         return DEFAULT_SKIN_LOCATION;
     }
 
-    public void setupRotations(net.minecraft.world.entity.decoration.ArmorStand entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - rotationYaw));
-        float f = (float)(entityLiving.level.getGameTime() - entityLiving.lastHit) + partialTicks;
-        if (f < 5.0f) {
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(net.minecraft.util.Mth.sin((float)(f / 1.5f * (float)java.lang.Math.PI)) * 3.0f));
+    public void setupRotations(net.minecraft.world.entity.decoration.ArmorStand $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - $$3));
+        float $$5 = (float)($$0.level.getGameTime() - $$0.lastHit) + $$4;
+        if ($$5 < 5.0f) {
+            $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(net.minecraft.util.Mth.sin((float)($$5 / 1.5f * (float)java.lang.Math.PI)) * 3.0f));
         }
     }
 
-    public boolean shouldShowName(net.minecraft.world.entity.decoration.ArmorStand entity) {
-        float f;
-        double d = this.entityRenderDispatcher.distanceToSqr((net.minecraft.world.entity.Entity)entity);
-        float f2 = f = entity.isCrouching() ? 32.0f : 64.0f;
-        if (d >= (double)(f * f)) {
+    public boolean shouldShowName(net.minecraft.world.entity.decoration.ArmorStand $$0) {
+        float $$2;
+        double $$1 = this.entityRenderDispatcher.distanceToSqr((net.minecraft.world.entity.Entity)$$0);
+        float f = $$2 = $$0.isCrouching() ? 32.0f : 64.0f;
+        if ($$1 >= (double)($$2 * $$2)) {
             return false;
         }
-        return entity.isCustomNameVisible();
+        return $$0.isCustomNameVisible();
     }
 
     @javax.annotation.Nullable