--- net/minecraft/client/renderer/entity/BatRenderer.java
+++ net/minecraft/client/renderer/entity/BatRenderer.java
@@ -12,21 +12,21 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.BatModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.BAT)), 0.25f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.ambient.Bat entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.ambient.Bat $$0) {
         return BAT_LOCATION;
     }
 
-    public void scale(net.minecraft.world.entity.ambient.Bat livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        matrixStack.scale(0.35f, 0.35f, 0.35f);
+    public void scale(net.minecraft.world.entity.ambient.Bat $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        $$1.scale(0.35f, 0.35f, 0.35f);
     }
 
-    public void setupRotations(net.minecraft.world.entity.ambient.Bat entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        if (entityLiving.isResting()) {
-            matrixStack.translate(0.0, (double)-0.1f, 0.0);
+    public void setupRotations(net.minecraft.world.entity.ambient.Bat $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        if ($$0.isResting()) {
+            $$1.translate(0.0, (double)-0.1f, 0.0);
         } else {
-            matrixStack.translate(0.0, (double)(net.minecraft.util.Mth.cos((float)(ageInTicks * 0.3f)) * 0.1f), 0.0);
+            $$1.translate(0.0, (double)(net.minecraft.util.Mth.cos((float)($$2 * 0.3f)) * 0.1f), 0.0);
         }
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
     }
 
     static {