--- net/minecraft/client/renderer/entity/BeeRenderer.java
+++ net/minecraft/client/renderer/entity/BeeRenderer.java
@@ -15,14 +15,14 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.BeeModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.BEE)), 0.4f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Bee entity) {
-        if (entity.isAngry()) {
-            if (entity.hasNectar()) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Bee $$0) {
+        if ($$0.isAngry()) {
+            if ($$0.hasNectar()) {
                 return ANGRY_NECTAR_BEE_TEXTURE;
             }
             return ANGRY_BEE_TEXTURE;
         }
-        if (entity.hasNectar()) {
+        if ($$0.hasNectar()) {
             return NECTAR_BEE_TEXTURE;
         }
         return BEE_TEXTURE;