--- net/minecraft/client/renderer/entity/CamelRenderer.java
+++ net/minecraft/client/renderer/entity/CamelRenderer.java
@@ -1,0 +1,23 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.renderer.entity;
+
+public class net.minecraft.client.renderer.entity.CamelRenderer
+extends net.minecraft.client.renderer.entity.MobRenderer<net.minecraft.world.entity.animal.camel.Camel, net.minecraft.client.model.CamelModel<net.minecraft.world.entity.animal.camel.Camel>>
+ {
+    public static net.minecraft.resources.ResourceLocation CAMEL_LOCATION;
+
+    public net.minecraft.client.renderer.entity.CamelRenderer(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0, net.minecraft.client.model.geom.ModelLayerLocation $$1) {
+        super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.CamelModel($$0.bakeLayer($$1)), 0.7f);
+    }
+
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.camel.Camel $$0) {
+        return CAMEL_LOCATION;
+    }
+
+    static {
+        CAMEL_LOCATION = new net.minecraft.resources.ResourceLocation("textures/entity/camel/camel.png");
+    }
+}
+