--- net/minecraft/client/renderer/entity/CatRenderer.java
+++ net/minecraft/client/renderer/entity/CatRenderer.java
@@ -11,26 +11,26 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.CatCollarLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Cat entity) {
-        return entity.getResourceLocation();
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Cat $$0) {
+        return $$0.getResourceLocation();
     }
 
-    public void scale(net.minecraft.world.entity.animal.Cat livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        super.scale((net.minecraft.world.entity.LivingEntity)livingEntity, matrixStack, partialTickTime);
-        matrixStack.scale(0.8f, 0.8f, 0.8f);
+    public void scale(net.minecraft.world.entity.animal.Cat $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        super.scale((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2);
+        $$1.scale(0.8f, 0.8f, 0.8f);
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.Cat entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        float f = entityLiving.getLieDownAmount(partialTicks);
-        if (f > 0.0f) {
-            matrixStack.translate((double)(0.4f * f), (double)(0.15f * f), (double)(0.1f * f));
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(net.minecraft.util.Mth.rotLerp((float)f, (float)0.0f, (float)90.0f)));
-            net.minecraft.core.BlockPos blockPos = entityLiving.blockPosition();
-            java.util.List list = entityLiving.level.getEntitiesOfClass(net.minecraft.world.entity.player.Player.class, new net.minecraft.world.phys.AABB(blockPos).inflate(2.0, 2.0, 2.0));
-            for (net.minecraft.world.entity.player.Player player : list) {
-                if (!player.isSleeping()) continue;
-                matrixStack.translate((double)(0.15f * f), 0.0, 0.0);
+    public void setupRotations(net.minecraft.world.entity.animal.Cat $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        float $$5 = $$0.getLieDownAmount($$4);
+        if ($$5 > 0.0f) {
+            $$1.translate((double)(0.4f * $$5), (double)(0.15f * $$5), (double)(0.1f * $$5));
+            $$1.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(net.minecraft.util.Mth.rotLerp((float)$$5, (float)0.0f, (float)90.0f)));
+            net.minecraft.core.BlockPos $$6 = $$0.blockPosition();
+            java.util.List $$7 = $$0.level.getEntitiesOfClass(net.minecraft.world.entity.player.Player.class, new net.minecraft.world.phys.AABB($$6).inflate(2.0, 2.0, 2.0));
+            for (net.minecraft.world.entity.player.Player $$8 : $$7) {
+                if (!$$8.isSleeping()) continue;
+                $$1.translate((double)(0.15f * $$5), 0.0, 0.0);
                 break;
             }
         }