--- net/minecraft/client/renderer/entity/ChestedHorseRenderer.java
+++ net/minecraft/client/renderer/entity/ChestedHorseRenderer.java
@@ -12,8 +12,8 @@
         super($$0, (net.minecraft.client.model.HorseModel)new net.minecraft.client.model.ChestedHorseModel($$0.bakeLayer($$2)), $$1);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(T entity) {
-        return (net.minecraft.resources.ResourceLocation)MAP.get((java.lang.Object)entity.getType());
+    public net.minecraft.resources.ResourceLocation getTextureLocation(T $$0) {
+        return (net.minecraft.resources.ResourceLocation)MAP.get((java.lang.Object)$$0.getType());
     }
 
     static {