--- net/minecraft/client/renderer/entity/CodRenderer.java
+++ net/minecraft/client/renderer/entity/CodRenderer.java
@@ -12,17 +12,17 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.CodModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.COD)), 0.3f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Cod entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Cod $$0) {
         return COD_LOCATION;
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.Cod entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        float f = 4.3f * net.minecraft.util.Mth.sin((float)(0.6f * ageInTicks));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f));
-        if (!entityLiving.isInWater()) {
-            matrixStack.translate((double)0.1f, (double)0.1f, (double)-0.1f);
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
+    public void setupRotations(net.minecraft.world.entity.animal.Cod $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        float $$5 = 4.3f * net.minecraft.util.Mth.sin((float)(0.6f * $$2));
+        $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$5));
+        if (!$$0.isInWater()) {
+            $$1.translate((double)0.1f, (double)0.1f, (double)-0.1f);
+            $$1.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
         }
     }
 