--- net/minecraft/client/renderer/entity/CowRenderer.java
+++ net/minecraft/client/renderer/entity/CowRenderer.java
@@ -12,7 +12,7 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.CowModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.COW)), 0.7f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Cow entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Cow $$0) {
         return COW_LOCATION;
     }
 