--- net/minecraft/client/renderer/entity/CreeperRenderer.java
+++ net/minecraft/client/renderer/entity/CreeperRenderer.java
@@ -13,26 +13,26 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.CreeperPowerLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public void scale(net.minecraft.world.entity.monster.Creeper livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        float f = livingEntity.getSwelling(partialTickTime);
-        float f2 = 1.0f + net.minecraft.util.Mth.sin((float)(f * 100.0f)) * f * 0.01f;
-        f = net.minecraft.util.Mth.clamp((float)f, (float)0.0f, (float)1.0f);
-        f *= f;
-        f *= f;
-        float f3 = (1.0f + f * 0.4f) * f2;
-        float f4 = (1.0f + f * 0.1f) / f2;
-        matrixStack.scale(f3, f4, f3);
+    public void scale(net.minecraft.world.entity.monster.Creeper $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        float $$3 = $$0.getSwelling($$2);
+        float $$4 = 1.0f + net.minecraft.util.Mth.sin((float)($$3 * 100.0f)) * $$3 * 0.01f;
+        $$3 = net.minecraft.util.Mth.clamp((float)$$3, (float)0.0f, (float)1.0f);
+        $$3 *= $$3;
+        $$3 *= $$3;
+        float $$5 = (1.0f + $$3 * 0.4f) * $$4;
+        float $$6 = (1.0f + $$3 * 0.1f) / $$4;
+        $$1.scale($$5, $$6, $$5);
     }
 
-    public float getWhiteOverlayProgress(net.minecraft.world.entity.monster.Creeper livingEntity, float partialTicks) {
-        float f = livingEntity.getSwelling(partialTicks);
-        if ((int)(f * 10.0f) % 2 == 0) {
+    public float getWhiteOverlayProgress(net.minecraft.world.entity.monster.Creeper $$0, float $$1) {
+        float $$2 = $$0.getSwelling($$1);
+        if ((int)($$2 * 10.0f) % 2 == 0) {
             return 0.0f;
         }
-        return net.minecraft.util.Mth.clamp((float)f, (float)0.5f, (float)1.0f);
+        return net.minecraft.util.Mth.clamp((float)$$2, (float)0.5f, (float)1.0f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Creeper entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Creeper $$0) {
         return CREEPER_LOCATION;
     }
 