--- net/minecraft/client/renderer/entity/DolphinRenderer.java
+++ net/minecraft/client/renderer/entity/DolphinRenderer.java
@@ -13,7 +13,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.DolphinCarryingItemLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getItemInHandRenderer()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Dolphin entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Dolphin $$0) {
         return DOLPHIN_LOCATION;
     }
 