--- net/minecraft/client/renderer/entity/DragonFireballRenderer.java
+++ net/minecraft/client/renderer/entity/DragonFireballRenderer.java
@@ -13,32 +13,32 @@
         super($$0);
     }
 
-    public int getBlockLightLevel(net.minecraft.world.entity.projectile.DragonFireball entity, net.minecraft.core.BlockPos pos) {
+    public int getBlockLightLevel(net.minecraft.world.entity.projectile.DragonFireball $$0, net.minecraft.core.BlockPos $$1) {
         return 15;
     }
 
-    public void render(net.minecraft.world.entity.projectile.DragonFireball entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        matrixStack.scale(2.0f, 2.0f, 2.0f);
-        matrixStack.mulPose(this.entityRenderDispatcher.cameraOrientation());
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
-        com.mojang.blaze3d.vertex.PoseStack$Pose pose = matrixStack.last();
-        com.mojang.math.Matrix4f matrix4f = pose.pose();
-        com.mojang.math.Matrix3f matrix3f = pose.normal();
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(RENDER_TYPE);
-        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (int)packedLight, (float)0.0f, (int)0, (int)0, (int)1);
-        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (int)packedLight, (float)1.0f, (int)0, (int)1, (int)1);
-        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (int)packedLight, (float)1.0f, (int)1, (int)1, (int)0);
-        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (int)packedLight, (float)0.0f, (int)1, (int)0, (int)0);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.projectile.DragonFireball $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        $$3.scale(2.0f, 2.0f, 2.0f);
+        $$3.mulPose(this.entityRenderDispatcher.cameraOrientation());
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
+        com.mojang.blaze3d.vertex.PoseStack$Pose $$6 = $$3.last();
+        com.mojang.math.Matrix4f $$7 = $$6.pose();
+        com.mojang.math.Matrix3f $$8 = $$6.normal();
+        com.mojang.blaze3d.vertex.VertexConsumer $$9 = $$4.getBuffer(RENDER_TYPE);
+        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$9, (com.mojang.math.Matrix4f)$$7, (com.mojang.math.Matrix3f)$$8, (int)$$5, (float)0.0f, (int)0, (int)0, (int)1);
+        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$9, (com.mojang.math.Matrix4f)$$7, (com.mojang.math.Matrix3f)$$8, (int)$$5, (float)1.0f, (int)0, (int)1, (int)1);
+        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$9, (com.mojang.math.Matrix4f)$$7, (com.mojang.math.Matrix3f)$$8, (int)$$5, (float)1.0f, (int)1, (int)1, (int)0);
+        net.minecraft.client.renderer.entity.DragonFireballRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$9, (com.mojang.math.Matrix4f)$$7, (com.mojang.math.Matrix3f)$$8, (int)$$5, (float)0.0f, (int)1, (int)0, (int)0);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
     public static void vertex(com.mojang.blaze3d.vertex.VertexConsumer $$0, com.mojang.math.Matrix4f $$1, com.mojang.math.Matrix3f $$2, int $$3, float $$4, int $$5, int $$6, int $$7) {
         $$0.vertex($$1, $$4 - 0.5f, (float)$$5 - 0.25f, 0.0f).color(255, 255, 255, 255).uv((float)$$6, (float)$$7).overlayCoords(net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY).uv2($$3).normal($$2, 0.0f, 1.0f, 0.0f).endVertex();
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.DragonFireball entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.DragonFireball $$0) {
         return TEXTURE_LOCATION;
     }
 