--- net/minecraft/client/renderer/entity/DrownedRenderer.java
+++ net/minecraft/client/renderer/entity/DrownedRenderer.java
@@ -13,15 +13,15 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.DrownedOuterLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zombie entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zombie $$0) {
         return DROWNED_LOCATION;
     }
 
-    public void setupRotations(net.minecraft.world.entity.monster.Drowned entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        float f = entityLiving.getSwimAmount(partialTicks);
-        if (f > 0.0f) {
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.lerp((float)f, (float)entityLiving.getXRot(), (float)(-10.0f - entityLiving.getXRot()))));
+    public void setupRotations(net.minecraft.world.entity.monster.Drowned $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        float $$5 = $$0.getSwimAmount($$4);
+        if ($$5 > 0.0f) {
+            $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.lerp((float)$$5, (float)$$0.getXRot(), (float)(-10.0f - $$0.getXRot()))));
         }
     }
 