--- net/minecraft/client/renderer/entity/EndCrystalRenderer.java
+++ net/minecraft/client/renderer/entity/EndCrystalRenderer.java
@@ -33,45 +33,45 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)32);
     }
 
-    public void render(net.minecraft.world.entity.boss.enderdragon.EndCrystal entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        float f = net.minecraft.client.renderer.entity.EndCrystalRenderer.getY((net.minecraft.world.entity.boss.enderdragon.EndCrystal)entity, (float)partialTicks);
-        float f2 = ((float)entity.time + partialTicks) * 3.0f;
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(RENDER_TYPE);
-        matrixStack.pushPose();
-        matrixStack.scale(2.0f, 2.0f, 2.0f);
-        matrixStack.translate(0.0, -0.5, 0.0);
-        int n = net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY;
-        if (entity.showsBottom()) {
-            this.base.render(matrixStack, vertexConsumer, packedLight, n);
+    public void render(net.minecraft.world.entity.boss.enderdragon.EndCrystal $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        float $$6 = net.minecraft.client.renderer.entity.EndCrystalRenderer.getY((net.minecraft.world.entity.boss.enderdragon.EndCrystal)$$0, (float)$$2);
+        float $$7 = ((float)$$0.time + $$2) * 3.0f;
+        com.mojang.blaze3d.vertex.VertexConsumer $$8 = $$4.getBuffer(RENDER_TYPE);
+        $$3.pushPose();
+        $$3.scale(2.0f, 2.0f, 2.0f);
+        $$3.translate(0.0, -0.5, 0.0);
+        int $$9 = net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY;
+        if ($$0.showsBottom()) {
+            this.base.render($$3, $$8, $$5, $$9);
         }
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f2));
-        matrixStack.translate(0.0, (double)(1.5f + f / 2.0f), 0.0);
-        matrixStack.mulPose(new com.mojang.math.Quaternion(new com.mojang.math.Vector3f(SIN_45, 0.0f, SIN_45), 60.0f, true));
-        this.glass.render(matrixStack, vertexConsumer, packedLight, n);
-        float f3 = 0.875f;
-        matrixStack.scale(0.875f, 0.875f, 0.875f);
-        matrixStack.mulPose(new com.mojang.math.Quaternion(new com.mojang.math.Vector3f(SIN_45, 0.0f, SIN_45), 60.0f, true));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f2));
-        this.glass.render(matrixStack, vertexConsumer, packedLight, n);
-        matrixStack.scale(0.875f, 0.875f, 0.875f);
-        matrixStack.mulPose(new com.mojang.math.Quaternion(new com.mojang.math.Vector3f(SIN_45, 0.0f, SIN_45), 60.0f, true));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f2));
-        this.cube.render(matrixStack, vertexConsumer, packedLight, n);
-        matrixStack.popPose();
-        matrixStack.popPose();
-        net.minecraft.core.BlockPos blockPos = entity.getBeamTarget();
-        if (blockPos != null) {
-            float f4 = (float)blockPos.getX() + 0.5f;
-            float f5 = (float)blockPos.getY() + 0.5f;
-            float f6 = (float)blockPos.getZ() + 0.5f;
-            float f7 = (float)((double)f4 - entity.getX());
-            float f8 = (float)((double)f5 - entity.getY());
-            float f9 = (float)((double)f6 - entity.getZ());
-            matrixStack.translate((double)f7, (double)f8, (double)f9);
-            net.minecraft.client.renderer.entity.EnderDragonRenderer.renderCrystalBeams((float)(-f7), (float)(-f8 + f), (float)(-f9), (float)partialTicks, (int)entity.time, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight);
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$7));
+        $$3.translate(0.0, (double)(1.5f + $$6 / 2.0f), 0.0);
+        $$3.mulPose(new com.mojang.math.Quaternion(new com.mojang.math.Vector3f(SIN_45, 0.0f, SIN_45), 60.0f, true));
+        this.glass.render($$3, $$8, $$5, $$9);
+        float $$10 = 0.875f;
+        $$3.scale(0.875f, 0.875f, 0.875f);
+        $$3.mulPose(new com.mojang.math.Quaternion(new com.mojang.math.Vector3f(SIN_45, 0.0f, SIN_45), 60.0f, true));
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$7));
+        this.glass.render($$3, $$8, $$5, $$9);
+        $$3.scale(0.875f, 0.875f, 0.875f);
+        $$3.mulPose(new com.mojang.math.Quaternion(new com.mojang.math.Vector3f(SIN_45, 0.0f, SIN_45), 60.0f, true));
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$7));
+        this.cube.render($$3, $$8, $$5, $$9);
+        $$3.popPose();
+        $$3.popPose();
+        net.minecraft.core.BlockPos $$11 = $$0.getBeamTarget();
+        if ($$11 != null) {
+            float $$12 = (float)$$11.getX() + 0.5f;
+            float $$13 = (float)$$11.getY() + 0.5f;
+            float $$14 = (float)$$11.getZ() + 0.5f;
+            float $$15 = (float)((double)$$12 - $$0.getX());
+            float $$16 = (float)((double)$$13 - $$0.getY());
+            float $$17 = (float)((double)$$14 - $$0.getZ());
+            $$3.translate((double)$$15, (double)$$16, (double)$$17);
+            net.minecraft.client.renderer.entity.EnderDragonRenderer.renderCrystalBeams((float)(-$$15), (float)(-$$16 + $$6), (float)(-$$17), (float)$$2, (int)$$0.time, (com.mojang.blaze3d.vertex.PoseStack)$$3, (net.minecraft.client.renderer.MultiBufferSource)$$4, (int)$$5);
         }
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
     public static float getY(net.minecraft.world.entity.boss.enderdragon.EndCrystal $$0, float $$1) {
@@ -81,12 +81,12 @@
         return $$3 - 1.4f;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.boss.enderdragon.EndCrystal entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.boss.enderdragon.EndCrystal $$0) {
         return END_CRYSTAL_LOCATION;
     }
 
-    public boolean shouldRender(net.minecraft.world.entity.boss.enderdragon.EndCrystal livingEntity, net.minecraft.client.renderer.culling.Frustum camera, double camX, double camY, double camZ) {
-        return super.shouldRender((net.minecraft.world.entity.Entity)livingEntity, camera, camX, camY, camZ) || livingEntity.getBeamTarget() != null;
+    public boolean shouldRender(net.minecraft.world.entity.boss.enderdragon.EndCrystal $$0, net.minecraft.client.renderer.culling.Frustum $$1, double $$2, double $$3, double $$4) {
+        return super.shouldRender((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4) || $$0.getBeamTarget() != null;
     }
 
     static {