--- net/minecraft/client/renderer/entity/EndermanRenderer.java
+++ net/minecraft/client/renderer/entity/EndermanRenderer.java
@@ -16,23 +16,23 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.CarriedBlockLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getBlockRenderDispatcher()));
     }
 
-    public void render(net.minecraft.world.entity.monster.EnderMan entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        net.minecraft.world.level.block.state.BlockState blockState = entity.getCarriedBlock();
-        net.minecraft.client.model.EndermanModel endermanModel = (net.minecraft.client.model.EndermanModel)this.getModel();
-        endermanModel.carrying = blockState != null;
-        endermanModel.creepy = entity.isCreepy();
-        super.render((net.minecraft.world.entity.Mob)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.monster.EnderMan $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        net.minecraft.world.level.block.state.BlockState $$6 = $$0.getCarriedBlock();
+        net.minecraft.client.model.EndermanModel $$7 = (net.minecraft.client.model.EndermanModel)this.getModel();
+        $$7.carrying = $$6 != null;
+        $$7.creepy = $$0.isCreepy();
+        super.render((net.minecraft.world.entity.Mob)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.world.phys.Vec3 getRenderOffset(net.minecraft.world.entity.monster.EnderMan entity, float partialTicks) {
-        if (entity.isCreepy()) {
-            double d = 0.02;
+    public net.minecraft.world.phys.Vec3 getRenderOffset(net.minecraft.world.entity.monster.EnderMan $$0, float $$1) {
+        if ($$0.isCreepy()) {
+            double $$2 = 0.02;
             return new net.minecraft.world.phys.Vec3(this.random.nextGaussian() * 0.02, 0.0, this.random.nextGaussian() * 0.02);
         }
-        return super.getRenderOffset((net.minecraft.world.entity.Entity)entity, partialTicks);
+        return super.getRenderOffset((net.minecraft.world.entity.Entity)$$0, $$1);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.EnderMan entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.EnderMan $$0) {
         return ENDERMAN_LOCATION;
     }
 