--- net/minecraft/client/renderer/entity/EnderDragonRenderer$DragonModel.java
+++ net/minecraft/client/renderer/entity/EnderDragonRenderer$DragonModel.java
@@ -53,91 +53,90 @@
         this.rightRearFoot = this.rightRearLegTip.getChild("right_hind_foot");
     }
 
-    public void prepareMobModel(net.minecraft.world.entity.boss.enderdragon.EnderDragon entity, float limbSwing, float limbSwingAmount, float partialTick) {
-        this.entity = entity;
-        this.a = partialTick;
+    public void prepareMobModel(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0, float $$1, float $$2, float $$3) {
+        this.entity = $$0;
+        this.a = $$3;
     }
 
-    public void setupAnim(net.minecraft.world.entity.boss.enderdragon.EnderDragon entity, float limbSwing, float limbSwingAmount, float ageInTicks, float netHeadYaw, float headPitch) {
+    public void setupAnim(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0, float $$1, float $$2, float $$3, float $$4, float $$5) {
     }
 
-    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack matrixStack, com.mojang.blaze3d.vertex.VertexConsumer buffer, int packedLight, int packedOverlay, float red, float green, float blue, float alpha) {
-        float f;
-        matrixStack.pushPose();
-        float f2 = net.minecraft.util.Mth.lerp((float)this.a, (float)this.entity.oFlapTime, (float)this.entity.flapTime);
-        this.jaw.xRot = (float)(java.lang.Math.sin((double)(f2 * ((float)java.lang.Math.PI * 2))) + 1.0) * 0.2f;
-        float f3 = (float)(java.lang.Math.sin((double)(f2 * ((float)java.lang.Math.PI * 2) - 1.0f)) + 1.0);
-        f3 = (f3 * f3 + f3 * 2.0f) * 0.05f;
-        matrixStack.translate(0.0, (double)(f3 - 2.0f), -3.0);
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(f3 * 2.0f));
-        float f4 = 0.0f;
-        float f5 = 20.0f;
-        float f6 = -12.0f;
-        float f7 = 1.5f;
-        double[] dArray = this.entity.getLatencyPos(6, this.a);
-        float f8 = net.minecraft.util.Mth.rotWrap((double)(this.entity.getLatencyPos(5, this.a)[0] - this.entity.getLatencyPos(10, this.a)[0]));
-        float f9 = net.minecraft.util.Mth.rotWrap((double)(this.entity.getLatencyPos(5, this.a)[0] + (double)(f8 / 2.0f)));
-        float f10 = f2 * ((float)java.lang.Math.PI * 2);
-        for (int i = 0; i < 5; ++i) {
-            double[] dArray2 = this.entity.getLatencyPos(5 - i, this.a);
-            f = (float)java.lang.Math.cos((double)((float)i * 0.45f + f10)) * 0.15f;
-            this.neck.yRot = net.minecraft.util.Mth.rotWrap((double)(dArray2[0] - dArray[0])) * ((float)java.lang.Math.PI / 180) * 1.5f;
-            this.neck.xRot = f + this.entity.getHeadPartYOffset(i, dArray, dArray2) * ((float)java.lang.Math.PI / 180) * 1.5f * 5.0f;
-            this.neck.zRot = -net.minecraft.util.Mth.rotWrap((double)(dArray2[0] - (double)f9)) * ((float)java.lang.Math.PI / 180) * 1.5f;
-            this.neck.y = f5;
-            this.neck.z = f6;
-            this.neck.x = f4;
-            f5 += net.minecraft.util.Mth.sin((float)this.neck.xRot) * 10.0f;
-            f6 -= net.minecraft.util.Mth.cos((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
-            f4 -= net.minecraft.util.Mth.sin((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
-            this.neck.render(matrixStack, buffer, packedLight, packedOverlay, 1.0f, 1.0f, 1.0f, alpha);
+    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7) {
+        $$0.pushPose();
+        float $$8 = net.minecraft.util.Mth.lerp((float)this.a, (float)this.entity.oFlapTime, (float)this.entity.flapTime);
+        this.jaw.xRot = (float)(java.lang.Math.sin((double)($$8 * ((float)java.lang.Math.PI * 2))) + 1.0) * 0.2f;
+        float $$9 = (float)(java.lang.Math.sin((double)($$8 * ((float)java.lang.Math.PI * 2) - 1.0f)) + 1.0);
+        $$9 = ($$9 * $$9 + $$9 * 2.0f) * 0.05f;
+        $$0.translate(0.0, (double)($$9 - 2.0f), -3.0);
+        $$0.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$9 * 2.0f));
+        float $$10 = 0.0f;
+        float $$11 = 20.0f;
+        float $$12 = -12.0f;
+        float $$13 = 1.5f;
+        double[] $$14 = this.entity.getLatencyPos(6, this.a);
+        float $$15 = net.minecraft.util.Mth.rotWrap((double)(this.entity.getLatencyPos(5, this.a)[0] - this.entity.getLatencyPos(10, this.a)[0]));
+        float $$16 = net.minecraft.util.Mth.rotWrap((double)(this.entity.getLatencyPos(5, this.a)[0] + (double)($$15 / 2.0f)));
+        float $$17 = $$8 * ((float)java.lang.Math.PI * 2);
+        for (int $$18 = 0; $$18 < 5; ++$$18) {
+            double[] $$19 = this.entity.getLatencyPos(5 - $$18, this.a);
+            float $$20 = (float)java.lang.Math.cos((double)((float)$$18 * 0.45f + $$17)) * 0.15f;
+            this.neck.yRot = net.minecraft.util.Mth.rotWrap((double)($$19[0] - $$14[0])) * ((float)java.lang.Math.PI / 180) * 1.5f;
+            this.neck.xRot = $$20 + this.entity.getHeadPartYOffset($$18, $$14, $$19) * ((float)java.lang.Math.PI / 180) * 1.5f * 5.0f;
+            this.neck.zRot = -net.minecraft.util.Mth.rotWrap((double)($$19[0] - (double)$$16)) * ((float)java.lang.Math.PI / 180) * 1.5f;
+            this.neck.y = $$11;
+            this.neck.z = $$12;
+            this.neck.x = $$10;
+            $$11 += net.minecraft.util.Mth.sin((float)this.neck.xRot) * 10.0f;
+            $$12 -= net.minecraft.util.Mth.cos((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
+            $$10 -= net.minecraft.util.Mth.sin((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
+            this.neck.render($$0, $$1, $$2, $$3, 1.0f, 1.0f, 1.0f, $$7);
         }
-        this.head.y = f5;
-        this.head.z = f6;
-        this.head.x = f4;
-        double[] dArray3 = this.entity.getLatencyPos(0, this.a);
-        this.head.yRot = net.minecraft.util.Mth.rotWrap((double)(dArray3[0] - dArray[0])) * ((float)java.lang.Math.PI / 180);
-        this.head.xRot = net.minecraft.util.Mth.rotWrap((double)this.entity.getHeadPartYOffset(6, dArray, dArray3)) * ((float)java.lang.Math.PI / 180) * 1.5f * 5.0f;
-        this.head.zRot = -net.minecraft.util.Mth.rotWrap((double)(dArray3[0] - (double)f9)) * ((float)java.lang.Math.PI / 180);
-        this.head.render(matrixStack, buffer, packedLight, packedOverlay, 1.0f, 1.0f, 1.0f, alpha);
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 1.0, 0.0);
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(-f8 * 1.5f));
-        matrixStack.translate(0.0, -1.0, 0.0);
+        this.head.y = $$11;
+        this.head.z = $$12;
+        this.head.x = $$10;
+        double[] $$21 = this.entity.getLatencyPos(0, this.a);
+        this.head.yRot = net.minecraft.util.Mth.rotWrap((double)($$21[0] - $$14[0])) * ((float)java.lang.Math.PI / 180);
+        this.head.xRot = net.minecraft.util.Mth.rotWrap((double)this.entity.getHeadPartYOffset(6, $$14, $$21)) * ((float)java.lang.Math.PI / 180) * 1.5f * 5.0f;
+        this.head.zRot = -net.minecraft.util.Mth.rotWrap((double)($$21[0] - (double)$$16)) * ((float)java.lang.Math.PI / 180);
+        this.head.render($$0, $$1, $$2, $$3, 1.0f, 1.0f, 1.0f, $$7);
+        $$0.pushPose();
+        $$0.translate(0.0, 1.0, 0.0);
+        $$0.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(-$$15 * 1.5f));
+        $$0.translate(0.0, -1.0, 0.0);
         this.body.zRot = 0.0f;
-        this.body.render(matrixStack, buffer, packedLight, packedOverlay, 1.0f, 1.0f, 1.0f, alpha);
-        float f11 = f2 * ((float)java.lang.Math.PI * 2);
-        this.leftWing.xRot = 0.125f - (float)java.lang.Math.cos((double)f11) * 0.2f;
+        this.body.render($$0, $$1, $$2, $$3, 1.0f, 1.0f, 1.0f, $$7);
+        float $$22 = $$8 * ((float)java.lang.Math.PI * 2);
+        this.leftWing.xRot = 0.125f - (float)java.lang.Math.cos((double)$$22) * 0.2f;
         this.leftWing.yRot = -0.25f;
-        this.leftWing.zRot = -((float)(java.lang.Math.sin((double)f11) + 0.125)) * 0.8f;
-        this.leftWingTip.zRot = (float)(java.lang.Math.sin((double)(f11 + 2.0f)) + 0.5) * 0.75f;
+        this.leftWing.zRot = -((float)(java.lang.Math.sin((double)$$22) + 0.125)) * 0.8f;
+        this.leftWingTip.zRot = (float)(java.lang.Math.sin((double)($$22 + 2.0f)) + 0.5) * 0.75f;
         this.rightWing.xRot = this.leftWing.xRot;
         this.rightWing.yRot = -this.leftWing.yRot;
         this.rightWing.zRot = -this.leftWing.zRot;
         this.rightWingTip.zRot = -this.leftWingTip.zRot;
-        this.renderSide(matrixStack, buffer, packedLight, packedOverlay, f3, this.leftWing, this.leftFrontLeg, this.leftFrontLegTip, this.leftFrontFoot, this.leftRearLeg, this.leftRearLegTip, this.leftRearFoot, alpha);
-        this.renderSide(matrixStack, buffer, packedLight, packedOverlay, f3, this.rightWing, this.rightFrontLeg, this.rightFrontLegTip, this.rightFrontFoot, this.rightRearLeg, this.rightRearLegTip, this.rightRearFoot, alpha);
-        matrixStack.popPose();
-        f = -net.minecraft.util.Mth.sin((float)(f2 * ((float)java.lang.Math.PI * 2))) * 0.0f;
-        f10 = f2 * ((float)java.lang.Math.PI * 2);
-        f5 = 10.0f;
-        f6 = 60.0f;
-        f4 = 0.0f;
-        dArray = this.entity.getLatencyPos(11, this.a);
-        for (int i = 0; i < 12; ++i) {
-            dArray3 = this.entity.getLatencyPos(12 + i, this.a);
-            this.neck.yRot = (net.minecraft.util.Mth.rotWrap((double)(dArray3[0] - dArray[0])) * 1.5f + 180.0f) * ((float)java.lang.Math.PI / 180);
-            this.neck.xRot = (f += net.minecraft.util.Mth.sin((float)((float)i * 0.45f + f10)) * 0.05f) + (float)(dArray3[1] - dArray[1]) * ((float)java.lang.Math.PI / 180) * 1.5f * 5.0f;
-            this.neck.zRot = net.minecraft.util.Mth.rotWrap((double)(dArray3[0] - (double)f9)) * ((float)java.lang.Math.PI / 180) * 1.5f;
-            this.neck.y = f5;
-            this.neck.z = f6;
-            this.neck.x = f4;
-            f5 += net.minecraft.util.Mth.sin((float)this.neck.xRot) * 10.0f;
-            f6 -= net.minecraft.util.Mth.cos((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
-            f4 -= net.minecraft.util.Mth.sin((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
-            this.neck.render(matrixStack, buffer, packedLight, packedOverlay, 1.0f, 1.0f, 1.0f, alpha);
+        this.renderSide($$0, $$1, $$2, $$3, $$9, this.leftWing, this.leftFrontLeg, this.leftFrontLegTip, this.leftFrontFoot, this.leftRearLeg, this.leftRearLegTip, this.leftRearFoot, $$7);
+        this.renderSide($$0, $$1, $$2, $$3, $$9, this.rightWing, this.rightFrontLeg, this.rightFrontLegTip, this.rightFrontFoot, this.rightRearLeg, this.rightRearLegTip, this.rightRearFoot, $$7);
+        $$0.popPose();
+        float $$23 = -net.minecraft.util.Mth.sin((float)($$8 * ((float)java.lang.Math.PI * 2))) * 0.0f;
+        $$17 = $$8 * ((float)java.lang.Math.PI * 2);
+        $$11 = 10.0f;
+        $$12 = 60.0f;
+        $$10 = 0.0f;
+        $$14 = this.entity.getLatencyPos(11, this.a);
+        for (int $$24 = 0; $$24 < 12; ++$$24) {
+            $$21 = this.entity.getLatencyPos(12 + $$24, this.a);
+            this.neck.yRot = (net.minecraft.util.Mth.rotWrap((double)($$21[0] - $$14[0])) * 1.5f + 180.0f) * ((float)java.lang.Math.PI / 180);
+            this.neck.xRot = ($$23 += net.minecraft.util.Mth.sin((float)((float)$$24 * 0.45f + $$17)) * 0.05f) + (float)($$21[1] - $$14[1]) * ((float)java.lang.Math.PI / 180) * 1.5f * 5.0f;
+            this.neck.zRot = net.minecraft.util.Mth.rotWrap((double)($$21[0] - (double)$$16)) * ((float)java.lang.Math.PI / 180) * 1.5f;
+            this.neck.y = $$11;
+            this.neck.z = $$12;
+            this.neck.x = $$10;
+            $$11 += net.minecraft.util.Mth.sin((float)this.neck.xRot) * 10.0f;
+            $$12 -= net.minecraft.util.Mth.cos((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
+            $$10 -= net.minecraft.util.Mth.sin((float)this.neck.yRot) * net.minecraft.util.Mth.cos((float)this.neck.xRot) * 10.0f;
+            this.neck.render($$0, $$1, $$2, $$3, 1.0f, 1.0f, 1.0f, $$7);
         }
-        matrixStack.popPose();
+        $$0.popPose();
     }
 
     public void renderSide(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, net.minecraft.client.model.geom.ModelPart $$5, net.minecraft.client.model.geom.ModelPart $$6, net.minecraft.client.model.geom.ModelPart $$7, net.minecraft.client.model.geom.ModelPart $$8, net.minecraft.client.model.geom.ModelPart $$9, net.minecraft.client.model.geom.ModelPart $$10, net.minecraft.client.model.geom.ModelPart $$11, float $$12) {