--- net/minecraft/client/renderer/entity/EnderDragonRenderer.java
+++ net/minecraft/client/renderer/entity/EnderDragonRenderer.java
@@ -23,71 +23,71 @@
         this.model = new net.minecraft.client.renderer.entity.EnderDragonRenderer$DragonModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.ENDER_DRAGON));
     }
 
-    public void render(net.minecraft.world.entity.boss.enderdragon.EnderDragon entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        float f = (float)entity.getLatencyPos(7, partialTicks)[0];
-        float f2 = (float)(entity.getLatencyPos(5, partialTicks)[1] - entity.getLatencyPos(10, partialTicks)[1]);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(f2 * 10.0f));
-        matrixStack.translate(0.0, 0.0, 1.0);
-        matrixStack.scale(-1.0f, -1.0f, 1.0f);
-        matrixStack.translate(0.0, (double)-1.501f, 0.0);
-        boolean bl = entity.hurtTime > 0;
-        this.model.prepareMobModel(entity, 0.0f, 0.0f, partialTicks);
-        if (entity.dragonDeathTime > 0) {
-            float f3 = (float)entity.dragonDeathTime / 200.0f;
-            com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.dragonExplosionAlpha((net.minecraft.resources.ResourceLocation)DRAGON_EXPLODING_LOCATION));
-            this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, f3);
-            com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer2 = buffer.getBuffer(DECAL);
-            this.model.renderToBuffer(matrixStack, vertexConsumer2, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.pack((float)0.0f, (boolean)bl), 1.0f, 1.0f, 1.0f, 1.0f);
+    public void render(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        float $$6 = (float)$$0.getLatencyPos(7, $$2)[0];
+        float $$7 = (float)($$0.getLatencyPos(5, $$2)[1] - $$0.getLatencyPos(10, $$2)[1]);
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-$$6));
+        $$3.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$7 * 10.0f));
+        $$3.translate(0.0, 0.0, 1.0);
+        $$3.scale(-1.0f, -1.0f, 1.0f);
+        $$3.translate(0.0, (double)-1.501f, 0.0);
+        boolean $$8 = $$0.hurtTime > 0;
+        this.model.prepareMobModel($$0, 0.0f, 0.0f, $$2);
+        if ($$0.dragonDeathTime > 0) {
+            float $$9 = (float)$$0.dragonDeathTime / 200.0f;
+            com.mojang.blaze3d.vertex.VertexConsumer $$10 = $$4.getBuffer(net.minecraft.client.renderer.RenderType.dragonExplosionAlpha((net.minecraft.resources.ResourceLocation)DRAGON_EXPLODING_LOCATION));
+            this.model.renderToBuffer($$3, $$10, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, $$9);
+            com.mojang.blaze3d.vertex.VertexConsumer $$11 = $$4.getBuffer(DECAL);
+            this.model.renderToBuffer($$3, $$11, $$5, net.minecraft.client.renderer.texture.OverlayTexture.pack((float)0.0f, (boolean)$$8), 1.0f, 1.0f, 1.0f, 1.0f);
         } else {
-            com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(RENDER_TYPE);
-            this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.pack((float)0.0f, (boolean)bl), 1.0f, 1.0f, 1.0f, 1.0f);
+            com.mojang.blaze3d.vertex.VertexConsumer $$12 = $$4.getBuffer(RENDER_TYPE);
+            this.model.renderToBuffer($$3, $$12, $$5, net.minecraft.client.renderer.texture.OverlayTexture.pack((float)0.0f, (boolean)$$8), 1.0f, 1.0f, 1.0f, 1.0f);
         }
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(EYES);
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        if (entity.dragonDeathTime > 0) {
-            float f4 = ((float)entity.dragonDeathTime + partialTicks) / 200.0f;
-            float f5 = java.lang.Math.min((float)(f4 > 0.8f ? (f4 - 0.8f) / 0.2f : 0.0f), (float)1.0f);
-            net.minecraft.util.RandomSource randomSource = net.minecraft.util.RandomSource.create((long)432L);
-            com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer3 = buffer.getBuffer(net.minecraft.client.renderer.RenderType.lightning());
-            matrixStack.pushPose();
-            matrixStack.translate(0.0, -1.0, -2.0);
-            int n = 0;
-            while ((float)n < (f4 + f4 * f4) / 2.0f * 60.0f) {
-                matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(randomSource.nextFloat() * 360.0f));
-                matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(randomSource.nextFloat() * 360.0f));
-                matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(randomSource.nextFloat() * 360.0f));
-                matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(randomSource.nextFloat() * 360.0f));
-                matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(randomSource.nextFloat() * 360.0f));
-                matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(randomSource.nextFloat() * 360.0f + f4 * 90.0f));
-                float f6 = randomSource.nextFloat() * 20.0f + 5.0f + f5 * 10.0f;
-                float f7 = randomSource.nextFloat() * 2.0f + 1.0f + f5 * 2.0f;
-                com.mojang.math.Matrix4f matrix4f = matrixStack.last().pose();
-                int n2 = (int)(255.0f * (1.0f - f5));
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex01((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (int)n2);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex2((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (float)f6, (float)f7);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex3((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (float)f6, (float)f7);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex01((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (int)n2);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex3((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (float)f6, (float)f7);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex4((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (float)f6, (float)f7);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex01((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (int)n2);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex4((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (float)f6, (float)f7);
-                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex2((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer3, (com.mojang.math.Matrix4f)matrix4f, (float)f6, (float)f7);
-                ++n;
+        com.mojang.blaze3d.vertex.VertexConsumer $$13 = $$4.getBuffer(EYES);
+        this.model.renderToBuffer($$3, $$13, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        if ($$0.dragonDeathTime > 0) {
+            float $$14 = ((float)$$0.dragonDeathTime + $$2) / 200.0f;
+            float $$15 = java.lang.Math.min((float)($$14 > 0.8f ? ($$14 - 0.8f) / 0.2f : 0.0f), (float)1.0f);
+            net.minecraft.util.RandomSource $$16 = net.minecraft.util.RandomSource.create((long)432L);
+            com.mojang.blaze3d.vertex.VertexConsumer $$17 = $$4.getBuffer(net.minecraft.client.renderer.RenderType.lightning());
+            $$3.pushPose();
+            $$3.translate(0.0, -1.0, -2.0);
+            int $$18 = 0;
+            while ((float)$$18 < ($$14 + $$14 * $$14) / 2.0f * 60.0f) {
+                $$3.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$16.nextFloat() * 360.0f));
+                $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$16.nextFloat() * 360.0f));
+                $$3.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees($$16.nextFloat() * 360.0f));
+                $$3.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$16.nextFloat() * 360.0f));
+                $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$16.nextFloat() * 360.0f));
+                $$3.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees($$16.nextFloat() * 360.0f + $$14 * 90.0f));
+                float $$19 = $$16.nextFloat() * 20.0f + 5.0f + $$15 * 10.0f;
+                float $$20 = $$16.nextFloat() * 2.0f + 1.0f + $$15 * 2.0f;
+                com.mojang.math.Matrix4f $$21 = $$3.last().pose();
+                int $$22 = (int)(255.0f * (1.0f - $$15));
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex01((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (int)$$22);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex2((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (float)$$19, (float)$$20);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex3((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (float)$$19, (float)$$20);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex01((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (int)$$22);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex3((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (float)$$19, (float)$$20);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex4((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (float)$$19, (float)$$20);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex01((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (int)$$22);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex4((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (float)$$19, (float)$$20);
+                net.minecraft.client.renderer.entity.EnderDragonRenderer.vertex2((com.mojang.blaze3d.vertex.VertexConsumer)$$17, (com.mojang.math.Matrix4f)$$21, (float)$$19, (float)$$20);
+                ++$$18;
             }
-            matrixStack.popPose();
+            $$3.popPose();
         }
-        matrixStack.popPose();
-        if (entity.nearestCrystal != null) {
-            matrixStack.pushPose();
-            float f8 = (float)(entity.nearestCrystal.getX() - net.minecraft.util.Mth.lerp((double)partialTicks, (double)entity.xo, (double)entity.getX()));
-            float f9 = (float)(entity.nearestCrystal.getY() - net.minecraft.util.Mth.lerp((double)partialTicks, (double)entity.yo, (double)entity.getY()));
-            float f10 = (float)(entity.nearestCrystal.getZ() - net.minecraft.util.Mth.lerp((double)partialTicks, (double)entity.zo, (double)entity.getZ()));
-            net.minecraft.client.renderer.entity.EnderDragonRenderer.renderCrystalBeams((float)f8, (float)(f9 + net.minecraft.client.renderer.entity.EndCrystalRenderer.getY((net.minecraft.world.entity.boss.enderdragon.EndCrystal)entity.nearestCrystal, (float)partialTicks)), (float)f10, (float)partialTicks, (int)entity.tickCount, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight);
-            matrixStack.popPose();
+        $$3.popPose();
+        if ($$0.nearestCrystal != null) {
+            $$3.pushPose();
+            float $$23 = (float)($$0.nearestCrystal.getX() - net.minecraft.util.Mth.lerp((double)$$2, (double)$$0.xo, (double)$$0.getX()));
+            float $$24 = (float)($$0.nearestCrystal.getY() - net.minecraft.util.Mth.lerp((double)$$2, (double)$$0.yo, (double)$$0.getY()));
+            float $$25 = (float)($$0.nearestCrystal.getZ() - net.minecraft.util.Mth.lerp((double)$$2, (double)$$0.zo, (double)$$0.getZ()));
+            net.minecraft.client.renderer.entity.EnderDragonRenderer.renderCrystalBeams((float)$$23, (float)($$24 + net.minecraft.client.renderer.entity.EndCrystalRenderer.getY((net.minecraft.world.entity.boss.enderdragon.EndCrystal)$$0.nearestCrystal, (float)$$2)), (float)$$25, (float)$$2, (int)$$0.tickCount, (com.mojang.blaze3d.vertex.PoseStack)$$3, (net.minecraft.client.renderer.MultiBufferSource)$$4, (int)$$5);
+            $$3.popPose();
         }
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
     public static void vertex01(com.mojang.blaze3d.vertex.VertexConsumer $$0, com.mojang.math.Matrix4f $$1, int $$2) {
@@ -138,7 +138,7 @@
         $$5.popPose();
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.boss.enderdragon.EnderDragon entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.boss.enderdragon.EnderDragon $$0) {
         return DRAGON_LOCATION;
     }
 