--- net/minecraft/client/renderer/entity/EntityRenderers.java
+++ net/minecraft/client/renderer/entity/EntityRenderers.java
@@ -53,94 +53,98 @@
         return !$$0;
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$24(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$25(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.PiglinRenderer($$0, net.minecraft.client.model.geom.ModelLayers.ZOMBIFIED_PIGLIN, net.minecraft.client.model.geom.ModelLayers.ZOMBIFIED_PIGLIN_INNER_ARMOR, net.minecraft.client.model.geom.ModelLayers.ZOMBIFIED_PIGLIN_OUTER_ARMOR, true);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$23(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$24(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.UndeadHorseRenderer($$0, net.minecraft.client.model.geom.ModelLayers.ZOMBIE_HORSE);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$22(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$23(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.LlamaRenderer($$0, net.minecraft.client.model.geom.ModelLayers.TRADER_LLAMA);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$21(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$22(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.SquidRenderer($$0, new net.minecraft.client.model.SquidModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.SQUID)));
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$20(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$21(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.MinecartRenderer($$0, net.minecraft.client.model.geom.ModelLayers.SPAWNER_MINECART);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$19(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$20(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.ThrownItemRenderer($$0, 0.75f, true);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$18(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$19(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.UndeadHorseRenderer($$0, net.minecraft.client.model.geom.ModelLayers.SKELETON_HORSE);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$17(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$18(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.PiglinRenderer($$0, net.minecraft.client.model.geom.ModelLayers.PIGLIN_BRUTE, net.minecraft.client.model.geom.ModelLayers.PIGLIN_BRUTE_INNER_ARMOR, net.minecraft.client.model.geom.ModelLayers.PIGLIN_BRUTE_OUTER_ARMOR, false);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$16(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$17(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.PiglinRenderer($$0, net.minecraft.client.model.geom.ModelLayers.PIGLIN, net.minecraft.client.model.geom.ModelLayers.PIGLIN_INNER_ARMOR, net.minecraft.client.model.geom.ModelLayers.PIGLIN_OUTER_ARMOR, false);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$15(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$16(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.ChestedHorseRenderer($$0, 0.92f, net.minecraft.client.model.geom.ModelLayers.MULE);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$14(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$15(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.MinecartRenderer($$0, net.minecraft.client.model.geom.ModelLayers.MINECART);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$13(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$14(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.LlamaRenderer($$0, net.minecraft.client.model.geom.ModelLayers.LLAMA);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$12(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$13(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.MinecartRenderer($$0, net.minecraft.client.model.geom.ModelLayers.HOPPER_MINECART);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$11(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$12(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.GlowSquidRenderer($$0, new net.minecraft.client.model.SquidModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.GLOW_SQUID)));
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$10(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$11(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.GiantMobRenderer($$0, 6.0f);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$9(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$10(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.MinecartRenderer($$0, net.minecraft.client.model.geom.ModelLayers.FURNACE_MINECART);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$8(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$9(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.ThrownItemRenderer($$0, 3.0f, true);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$7(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$8(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.ThrownItemRenderer($$0, 1.0f, true);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$6(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$7(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.ChestedHorseRenderer($$0, 0.87f, net.minecraft.client.model.geom.ModelLayers.DONKEY);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$5(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$6(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.MinecartRenderer($$0, net.minecraft.client.model.geom.ModelLayers.COMMAND_BLOCK_MINECART);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$4(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$5(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.MinecartRenderer($$0, net.minecraft.client.model.geom.ModelLayers.CHEST_MINECART);
     }
 
-    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$3(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$4(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.BoatRenderer($$0, true);
     }
 
+    public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$3(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
+        return new net.minecraft.client.renderer.entity.CamelRenderer($$0, net.minecraft.client.model.geom.ModelLayers.CAMEL);
+    }
+
     public static /* synthetic */ net.minecraft.client.renderer.entity.EntityRenderer lambda$static$2(net.minecraft.client.renderer.entity.EntityRendererProvider$Context $$0) {
         return new net.minecraft.client.renderer.entity.BoatRenderer($$0, false);
     }
@@ -167,6 +171,7 @@
         net.minecraft.client.renderer.entity.EntityRenderers.register((net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.BLAZE, net.minecraft.client.renderer.entity.BlazeRenderer::new);
         net.minecraft.client.renderer.entity.EntityRenderers.register((net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.BOAT, $$0 -> new net.minecraft.client.renderer.entity.BoatRenderer($$0, false));
         net.minecraft.client.renderer.entity.EntityRenderers.register((net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.CAT, net.minecraft.client.renderer.entity.CatRenderer::new);
+        net.minecraft.client.renderer.entity.EntityRenderers.register((net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.CAMEL, $$0 -> new net.minecraft.client.renderer.entity.CamelRenderer($$0, net.minecraft.client.model.geom.ModelLayers.CAMEL));
         net.minecraft.client.renderer.entity.EntityRenderers.register((net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.CAVE_SPIDER, net.minecraft.client.renderer.entity.CaveSpiderRenderer::new);
         net.minecraft.client.renderer.entity.EntityRenderers.register((net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.CHEST_BOAT, $$0 -> new net.minecraft.client.renderer.entity.BoatRenderer($$0, true));
         net.minecraft.client.renderer.entity.EntityRenderers.register((net.minecraft.world.entity.EntityType)net.minecraft.world.entity.EntityType.CHEST_MINECART, $$0 -> new net.minecraft.client.renderer.entity.MinecartRenderer($$0, net.minecraft.client.model.geom.ModelLayers.CHEST_MINECART));