--- net/minecraft/client/renderer/entity/EvokerFangsRenderer.java
+++ net/minecraft/client/renderer/entity/EvokerFangsRenderer.java
@@ -14,29 +14,29 @@
         this.model = new net.minecraft.client.model.EvokerFangsModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.EVOKER_FANGS));
     }
 
-    public void render(net.minecraft.world.entity.projectile.EvokerFangs entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        float f = entity.getAnimationProgress(partialTicks);
-        if (f == 0.0f) {
+    public void render(net.minecraft.world.entity.projectile.EvokerFangs $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        float $$6 = $$0.getAnimationProgress($$2);
+        if ($$6 == 0.0f) {
             return;
         }
-        float f2 = 2.0f;
-        if (f > 0.9f) {
-            f2 *= (1.0f - f) / 0.1f;
+        float $$7 = 2.0f;
+        if ($$6 > 0.9f) {
+            $$7 *= (1.0f - $$6) / 0.1f;
         }
-        matrixStack.pushPose();
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f - entity.getYRot()));
-        matrixStack.scale(-f2, -f2, f2);
-        float f3 = 0.03125f;
-        matrixStack.translate(0.0, -0.626, 0.0);
-        matrixStack.scale(0.5f, 0.5f, 0.5f);
-        this.model.setupAnim((net.minecraft.world.entity.Entity)entity, f, 0.0f, 0.0f, entity.getYRot(), entity.getXRot());
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(this.model.renderType(TEXTURE_LOCATION));
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        $$3.pushPose();
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f - $$0.getYRot()));
+        $$3.scale(-$$7, -$$7, $$7);
+        float $$8 = 0.03125f;
+        $$3.translate(0.0, -0.626, 0.0);
+        $$3.scale(0.5f, 0.5f, 0.5f);
+        this.model.setupAnim((net.minecraft.world.entity.Entity)$$0, $$6, 0.0f, 0.0f, $$0.getYRot(), $$0.getXRot());
+        com.mojang.blaze3d.vertex.VertexConsumer $$9 = $$4.getBuffer(this.model.renderType(TEXTURE_LOCATION));
+        this.model.renderToBuffer($$3, $$9, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.EvokerFangs entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.EvokerFangs $$0) {
         return TEXTURE_LOCATION;
     }
 