--- net/minecraft/client/renderer/entity/EvokerRenderer.java
+++ net/minecraft/client/renderer/entity/EvokerRenderer.java
@@ -13,7 +13,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new /* Unavailable Anonymous Inner Class!! */);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(T entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(T $$0) {
         return EVOKER_ILLAGER;
     }
 