--- net/minecraft/client/renderer/entity/ExperienceOrbRenderer.java
+++ net/minecraft/client/renderer/entity/ExperienceOrbRenderer.java
@@ -15,47 +15,47 @@
         this.shadowStrength = 0.75f;
     }
 
-    public int getBlockLightLevel(net.minecraft.world.entity.ExperienceOrb entity, net.minecraft.core.BlockPos pos) {
-        return net.minecraft.util.Mth.clamp((int)(super.getBlockLightLevel((net.minecraft.world.entity.Entity)entity, pos) + 7), (int)0, (int)15);
+    public int getBlockLightLevel(net.minecraft.world.entity.ExperienceOrb $$0, net.minecraft.core.BlockPos $$1) {
+        return net.minecraft.util.Mth.clamp((int)(super.getBlockLightLevel((net.minecraft.world.entity.Entity)$$0, $$1) + 7), (int)0, (int)15);
     }
 
-    public void render(net.minecraft.world.entity.ExperienceOrb entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        int n = entity.getIcon();
-        float f = (float)(n % 4 * 16 + 0) / 64.0f;
-        float f2 = (float)(n % 4 * 16 + 16) / 64.0f;
-        float f3 = (float)(n / 4 * 16 + 0) / 64.0f;
-        float f4 = (float)(n / 4 * 16 + 16) / 64.0f;
-        float f5 = 1.0f;
-        float f6 = 0.5f;
-        float f7 = 0.25f;
-        float f8 = 255.0f;
-        float f9 = ((float)entity.tickCount + partialTicks) / 2.0f;
-        int n2 = (int)((net.minecraft.util.Mth.sin((float)(f9 + 0.0f)) + 1.0f) * 0.5f * 255.0f);
-        int n3 = 255;
-        int n4 = (int)((net.minecraft.util.Mth.sin((float)(f9 + 4.1887903f)) + 1.0f) * 0.1f * 255.0f);
-        matrixStack.translate(0.0, (double)0.1f, 0.0);
-        matrixStack.mulPose(this.entityRenderDispatcher.cameraOrientation());
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
-        float f10 = 0.3f;
-        matrixStack.scale(0.3f, 0.3f, 0.3f);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(RENDER_TYPE);
-        com.mojang.blaze3d.vertex.PoseStack$Pose pose = matrixStack.last();
-        com.mojang.math.Matrix4f matrix4f = pose.pose();
-        com.mojang.math.Matrix3f matrix3f = pose.normal();
-        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (float)-0.5f, (float)-0.25f, (int)n2, (int)255, (int)n4, (float)f, (float)f4, (int)packedLight);
-        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (float)0.5f, (float)-0.25f, (int)n2, (int)255, (int)n4, (float)f2, (float)f4, (int)packedLight);
-        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (float)0.5f, (float)0.75f, (int)n2, (int)255, (int)n4, (float)f2, (float)f3, (int)packedLight);
-        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (com.mojang.math.Matrix4f)matrix4f, (com.mojang.math.Matrix3f)matrix3f, (float)-0.5f, (float)0.75f, (int)n2, (int)255, (int)n4, (float)f, (float)f3, (int)packedLight);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.ExperienceOrb $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        int $$6 = $$0.getIcon();
+        float $$7 = (float)($$6 % 4 * 16 + 0) / 64.0f;
+        float $$8 = (float)($$6 % 4 * 16 + 16) / 64.0f;
+        float $$9 = (float)($$6 / 4 * 16 + 0) / 64.0f;
+        float $$10 = (float)($$6 / 4 * 16 + 16) / 64.0f;
+        float $$11 = 1.0f;
+        float $$12 = 0.5f;
+        float $$13 = 0.25f;
+        float $$14 = 255.0f;
+        float $$15 = ((float)$$0.tickCount + $$2) / 2.0f;
+        int $$16 = (int)((net.minecraft.util.Mth.sin((float)($$15 + 0.0f)) + 1.0f) * 0.5f * 255.0f);
+        int $$17 = 255;
+        int $$18 = (int)((net.minecraft.util.Mth.sin((float)($$15 + 4.1887903f)) + 1.0f) * 0.1f * 255.0f);
+        $$3.translate(0.0, (double)0.1f, 0.0);
+        $$3.mulPose(this.entityRenderDispatcher.cameraOrientation());
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
+        float $$19 = 0.3f;
+        $$3.scale(0.3f, 0.3f, 0.3f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$20 = $$4.getBuffer(RENDER_TYPE);
+        com.mojang.blaze3d.vertex.PoseStack$Pose $$21 = $$3.last();
+        com.mojang.math.Matrix4f $$22 = $$21.pose();
+        com.mojang.math.Matrix3f $$23 = $$21.normal();
+        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$20, (com.mojang.math.Matrix4f)$$22, (com.mojang.math.Matrix3f)$$23, (float)-0.5f, (float)-0.25f, (int)$$16, (int)255, (int)$$18, (float)$$7, (float)$$10, (int)$$5);
+        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$20, (com.mojang.math.Matrix4f)$$22, (com.mojang.math.Matrix3f)$$23, (float)0.5f, (float)-0.25f, (int)$$16, (int)255, (int)$$18, (float)$$8, (float)$$10, (int)$$5);
+        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$20, (com.mojang.math.Matrix4f)$$22, (com.mojang.math.Matrix3f)$$23, (float)0.5f, (float)0.75f, (int)$$16, (int)255, (int)$$18, (float)$$8, (float)$$9, (int)$$5);
+        net.minecraft.client.renderer.entity.ExperienceOrbRenderer.vertex((com.mojang.blaze3d.vertex.VertexConsumer)$$20, (com.mojang.math.Matrix4f)$$22, (com.mojang.math.Matrix3f)$$23, (float)-0.5f, (float)0.75f, (int)$$16, (int)255, (int)$$18, (float)$$7, (float)$$9, (int)$$5);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public static void vertex(com.mojang.blaze3d.vertex.VertexConsumer buffer, com.mojang.math.Matrix4f matrix, com.mojang.math.Matrix3f matrixNormal, float x, float y, int red, int green, int blue, float texU, float texV, int packedLight) {
-        buffer.vertex(matrix, x, y, 0.0f).color(red, green, blue, 128).uv(texU, texV).overlayCoords(net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY).uv2(packedLight).normal(matrixNormal, 0.0f, 1.0f, 0.0f).endVertex();
+    public static void vertex(com.mojang.blaze3d.vertex.VertexConsumer $$0, com.mojang.math.Matrix4f $$1, com.mojang.math.Matrix3f $$2, float $$3, float $$4, int $$5, int $$6, int $$7, float $$8, float $$9, int $$10) {
+        $$0.vertex($$1, $$3, $$4, 0.0f).color($$5, $$6, $$7, 128).uv($$8, $$9).overlayCoords(net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY).uv2($$10).normal($$2, 0.0f, 1.0f, 0.0f).endVertex();
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.ExperienceOrb entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.ExperienceOrb $$0) {
         return EXPERIENCE_ORB_LOCATION;
     }
 