--- net/minecraft/client/renderer/entity/FallingBlockRenderer.java
+++ net/minecraft/client/renderer/entity/FallingBlockRenderer.java
@@ -14,24 +14,24 @@
         this.dispatcher = $$0.getBlockRenderDispatcher();
     }
 
-    public void render(net.minecraft.world.entity.item.FallingBlockEntity entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        net.minecraft.world.level.block.state.BlockState blockState = entity.getBlockState();
-        if (blockState.getRenderShape() != net.minecraft.world.level.block.RenderShape.MODEL) {
+    public void render(net.minecraft.world.entity.item.FallingBlockEntity $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        net.minecraft.world.level.block.state.BlockState $$6 = $$0.getBlockState();
+        if ($$6.getRenderShape() != net.minecraft.world.level.block.RenderShape.MODEL) {
             return;
         }
-        net.minecraft.world.level.Level level = entity.getLevel();
-        if (blockState == level.getBlockState(entity.blockPosition()) || blockState.getRenderShape() == net.minecraft.world.level.block.RenderShape.INVISIBLE) {
+        net.minecraft.world.level.Level $$7 = $$0.getLevel();
+        if ($$6 == $$7.getBlockState($$0.blockPosition()) || $$6.getRenderShape() == net.minecraft.world.level.block.RenderShape.INVISIBLE) {
             return;
         }
-        matrixStack.pushPose();
-        net.minecraft.core.BlockPos blockPos = new net.minecraft.core.BlockPos(entity.getX(), entity.getBoundingBox().maxY, entity.getZ());
-        matrixStack.translate(-0.5, 0.0, -0.5);
-        this.dispatcher.getModelRenderer().tesselateBlock((net.minecraft.world.level.BlockAndTintGetter)level, this.dispatcher.getBlockModel(blockState), blockState, blockPos, matrixStack, buffer.getBuffer(net.minecraft.client.renderer.ItemBlockRenderTypes.getMovingBlockRenderType((net.minecraft.world.level.block.state.BlockState)blockState)), false, net.minecraft.util.RandomSource.create(), blockState.getSeed(entity.getStartPos()), net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        $$3.pushPose();
+        net.minecraft.core.BlockPos $$8 = new net.minecraft.core.BlockPos($$0.getX(), $$0.getBoundingBox().maxY, $$0.getZ());
+        $$3.translate(-0.5, 0.0, -0.5);
+        this.dispatcher.getModelRenderer().tesselateBlock((net.minecraft.world.level.BlockAndTintGetter)$$7, this.dispatcher.getBlockModel($$6), $$6, $$8, $$3, $$4.getBuffer(net.minecraft.client.renderer.ItemBlockRenderTypes.getMovingBlockRenderType((net.minecraft.world.level.block.state.BlockState)$$6)), false, net.minecraft.util.RandomSource.create(), $$6.getSeed($$0.getStartPos()), net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.item.FallingBlockEntity entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.item.FallingBlockEntity $$0) {
         return net.minecraft.client.renderer.texture.TextureAtlas.LOCATION_BLOCKS;
     }
 }