--- net/minecraft/client/renderer/entity/FireworkEntityRenderer.java
+++ net/minecraft/client/renderer/entity/FireworkEntityRenderer.java
@@ -13,21 +13,21 @@
         this.itemRenderer = $$0.getItemRenderer();
     }
 
-    public void render(net.minecraft.world.entity.projectile.FireworkRocketEntity entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        matrixStack.mulPose(this.entityRenderDispatcher.cameraOrientation());
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
-        if (entity.isShotAtAngle()) {
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
+    public void render(net.minecraft.world.entity.projectile.FireworkRocketEntity $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        $$3.mulPose(this.entityRenderDispatcher.cameraOrientation());
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
+        if ($$0.isShotAtAngle()) {
+            $$3.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(180.0f));
+            $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f));
+            $$3.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(90.0f));
         }
-        this.itemRenderer.renderStatic(entity.getItem(), net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, matrixStack, buffer, entity.getId());
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        this.itemRenderer.renderStatic($$0.getItem(), net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, $$3, $$4, $$0.getId());
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.FireworkRocketEntity entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.FireworkRocketEntity $$0) {
         return net.minecraft.client.renderer.texture.TextureAtlas.LOCATION_BLOCKS;
     }
 }