--- net/minecraft/client/renderer/entity/FoxRenderer.java
+++ net/minecraft/client/renderer/entity/FoxRenderer.java
@@ -16,19 +16,19 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.FoxHeldItemLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getItemInHandRenderer()));
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.Fox entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        if (entityLiving.isPouncing() || entityLiving.isFaceplanted()) {
-            float f = -net.minecraft.util.Mth.lerp((float)partialTicks, (float)entityLiving.xRotO, (float)entityLiving.getXRot());
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(f));
+    public void setupRotations(net.minecraft.world.entity.animal.Fox $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        if ($$0.isPouncing() || $$0.isFaceplanted()) {
+            float $$5 = -net.minecraft.util.Mth.lerp((float)$$4, (float)$$0.xRotO, (float)$$0.getXRot());
+            $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$5));
         }
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Fox entity) {
-        if (entity.getFoxType() == net.minecraft.world.entity.animal.Fox$Type.RED) {
-            return entity.isSleeping() ? RED_FOX_SLEEP_TEXTURE : RED_FOX_TEXTURE;
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Fox $$0) {
+        if ($$0.getFoxType() == net.minecraft.world.entity.animal.Fox$Type.RED) {
+            return $$0.isSleeping() ? RED_FOX_SLEEP_TEXTURE : RED_FOX_TEXTURE;
         }
-        return entity.isSleeping() ? SNOW_FOX_SLEEP_TEXTURE : SNOW_FOX_TEXTURE;
+        return $$0.isSleeping() ? SNOW_FOX_SLEEP_TEXTURE : SNOW_FOX_TEXTURE;
     }
 
     static {