--- net/minecraft/client/renderer/entity/GiantMobRenderer.java
+++ net/minecraft/client/renderer/entity/GiantMobRenderer.java
@@ -16,11 +16,11 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, (net.minecraft.client.model.HumanoidModel)new net.minecraft.client.model.GiantZombieModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.GIANT_INNER_ARMOR)), (net.minecraft.client.model.HumanoidModel)new net.minecraft.client.model.GiantZombieModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.GIANT_OUTER_ARMOR))));
     }
 
-    public void scale(net.minecraft.world.entity.monster.Giant livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        matrixStack.scale(this.scale, this.scale, this.scale);
+    public void scale(net.minecraft.world.entity.monster.Giant $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        $$1.scale(this.scale, this.scale, this.scale);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Giant entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Giant $$0) {
         return ZOMBIE_LOCATION;
     }
 