--- net/minecraft/client/renderer/entity/HoglinRenderer.java
+++ net/minecraft/client/renderer/entity/HoglinRenderer.java
@@ -12,7 +12,7 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.HoglinModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.HOGLIN)), 0.7f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.hoglin.Hoglin entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.hoglin.Hoglin $$0) {
         return HOGLIN_LOCATION;
     }
 