--- net/minecraft/client/renderer/entity/HorseRenderer.java
+++ net/minecraft/client/renderer/entity/HorseRenderer.java
@@ -14,8 +14,8 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.HorseArmorLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.horse.Horse entity) {
-        return (net.minecraft.resources.ResourceLocation)LOCATION_BY_VARIANT.get((java.lang.Object)entity.getVariant());
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.horse.Horse $$0) {
+        return (net.minecraft.resources.ResourceLocation)LOCATION_BY_VARIANT.get((java.lang.Object)$$0.getVariant());
     }
 
     public static /* synthetic */ void lambda$static$0(java.util.EnumMap $$0) {