--- net/minecraft/client/renderer/entity/HumanoidMobRenderer.java
+++ net/minecraft/client/renderer/entity/HumanoidMobRenderer.java
@@ -19,7 +19,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.ItemInHandLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getItemInHandRenderer()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(T entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(T $$0) {
         return DEFAULT_LOCATION;
     }
 