--- net/minecraft/client/renderer/entity/HuskRenderer.java
+++ net/minecraft/client/renderer/entity/HuskRenderer.java
@@ -12,13 +12,13 @@
         super($$0, net.minecraft.client.model.geom.ModelLayers.HUSK, net.minecraft.client.model.geom.ModelLayers.HUSK_INNER_ARMOR, net.minecraft.client.model.geom.ModelLayers.HUSK_OUTER_ARMOR);
     }
 
-    public void scale(net.minecraft.world.entity.monster.Zombie livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        float f = 1.0625f;
-        matrixStack.scale(1.0625f, 1.0625f, 1.0625f);
-        super.scale((net.minecraft.world.entity.LivingEntity)livingEntity, matrixStack, partialTickTime);
+    public void scale(net.minecraft.world.entity.monster.Zombie $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        float $$3 = 1.0625f;
+        $$1.scale(1.0625f, 1.0625f, 1.0625f);
+        super.scale((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zombie entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zombie $$0) {
         return HUSK_LOCATION;
     }
 