--- net/minecraft/client/renderer/entity/IllagerRenderer.java
+++ net/minecraft/client/renderer/entity/IllagerRenderer.java
@@ -11,9 +11,9 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.CustomHeadLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet(), $$0.getItemInHandRenderer()));
     }
 
-    public void scale(T livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        float f = 0.9375f;
-        matrixStack.scale(0.9375f, 0.9375f, 0.9375f);
+    public void scale(T $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        float $$3 = 0.9375f;
+        $$1.scale(0.9375f, 0.9375f, 0.9375f);
     }
 }
 