--- net/minecraft/client/renderer/entity/IllusionerRenderer$1.java
+++ net/minecraft/client/renderer/entity/IllusionerRenderer$1.java
@@ -13,9 +13,9 @@
         super($$1, $$2);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight, net.minecraft.world.entity.monster.Illusioner livingEntity, float limbSwing, float limbSwingAmount, float partialTicks, float ageInTicks, float netHeadYaw, float headPitch) {
-        if (livingEntity.isCastingSpell() || livingEntity.isAggressive()) {
-            super.render(matrixStack, buffer, packedLight, (net.minecraft.world.entity.LivingEntity)livingEntity, limbSwing, limbSwingAmount, partialTicks, ageInTicks, netHeadYaw, headPitch);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, int $$2, net.minecraft.world.entity.monster.Illusioner $$3, float $$4, float $$5, float $$6, float $$7, float $$8, float $$9) {
+        if ($$3.isCastingSpell() || $$3.isAggressive()) {
+            super.render($$0, $$1, $$2, (net.minecraft.world.entity.LivingEntity)$$3, $$4, $$5, $$6, $$7, $$8, $$9);
         }
     }
 }