--- net/minecraft/client/renderer/entity/IllusionerRenderer.java
+++ net/minecraft/client/renderer/entity/IllusionerRenderer.java
@@ -14,26 +14,26 @@
         ((net.minecraft.client.model.IllagerModel)this.model).getHat().visible = true;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Illusioner entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Illusioner $$0) {
         return ILLUSIONER;
     }
 
-    public void render(net.minecraft.world.entity.monster.Illusioner entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        if (entity.isInvisible()) {
-            net.minecraft.world.phys.Vec3[] vec3Array = entity.getIllusionOffsets(partialTicks);
-            float f = this.getBob((net.minecraft.world.entity.LivingEntity)entity, partialTicks);
-            for (int i = 0; i < vec3Array.length; ++i) {
-                matrixStack.pushPose();
-                matrixStack.translate(vec3Array[i].x + (double)net.minecraft.util.Mth.cos((float)((float)i + f * 0.5f)) * 0.025, vec3Array[i].y + (double)net.minecraft.util.Mth.cos((float)((float)i + f * 0.75f)) * 0.0125, vec3Array[i].z + (double)net.minecraft.util.Mth.cos((float)((float)i + f * 0.7f)) * 0.025);
-                super.render((net.minecraft.world.entity.Mob)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
-                matrixStack.popPose();
+    public void render(net.minecraft.world.entity.monster.Illusioner $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        if ($$0.isInvisible()) {
+            net.minecraft.world.phys.Vec3[] $$6 = $$0.getIllusionOffsets($$2);
+            float $$7 = this.getBob((net.minecraft.world.entity.LivingEntity)$$0, $$2);
+            for (int $$8 = 0; $$8 < $$6.length; ++$$8) {
+                $$3.pushPose();
+                $$3.translate($$6[$$8].x + (double)net.minecraft.util.Mth.cos((float)((float)$$8 + $$7 * 0.5f)) * 0.025, $$6[$$8].y + (double)net.minecraft.util.Mth.cos((float)((float)$$8 + $$7 * 0.75f)) * 0.0125, $$6[$$8].z + (double)net.minecraft.util.Mth.cos((float)((float)$$8 + $$7 * 0.7f)) * 0.025);
+                super.render((net.minecraft.world.entity.Mob)$$0, $$1, $$2, $$3, $$4, $$5);
+                $$3.popPose();
             }
         } else {
-            super.render((net.minecraft.world.entity.Mob)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+            super.render((net.minecraft.world.entity.Mob)$$0, $$1, $$2, $$3, $$4, $$5);
         }
     }
 
-    public boolean isBodyVisible(net.minecraft.world.entity.monster.Illusioner livingEntity) {
+    public boolean isBodyVisible(net.minecraft.world.entity.monster.Illusioner $$0) {
         return true;
     }
 