--- net/minecraft/client/renderer/entity/IronGolemRenderer.java
+++ net/minecraft/client/renderer/entity/IronGolemRenderer.java
@@ -14,19 +14,19 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.IronGolemFlowerLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getBlockRenderDispatcher()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.IronGolem entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.IronGolem $$0) {
         return GOLEM_LOCATION;
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.IronGolem entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        if ((double)entityLiving.animationSpeed < 0.01) {
+    public void setupRotations(net.minecraft.world.entity.animal.IronGolem $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        if ((double)$$0.animationSpeed < 0.01) {
             return;
         }
-        float f = 13.0f;
-        float f2 = entityLiving.animationPosition - entityLiving.animationSpeed * (1.0f - partialTicks) + 6.0f;
-        float f3 = (java.lang.Math.abs((float)(f2 % 13.0f - 6.5f)) - 3.25f) / 3.25f;
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(6.5f * f3));
+        float $$5 = 13.0f;
+        float $$6 = $$0.animationPosition - $$0.animationSpeed * (1.0f - $$4) + 6.0f;
+        float $$7 = (java.lang.Math.abs((float)($$6 % 13.0f - 6.5f)) - 3.25f) / 3.25f;
+        $$1.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(6.5f * $$7));
     }
 
     static {