--- net/minecraft/client/renderer/entity/ItemEntityRenderer.java
+++ net/minecraft/client/renderer/entity/ItemEntityRenderer.java
@@ -25,69 +25,67 @@
         this.shadowStrength = 0.75f;
     }
 
-    public int getRenderAmount(net.minecraft.world.item.ItemStack stack) {
-        int n = 1;
-        if (stack.getCount() > 48) {
-            n = 5;
-        } else if (stack.getCount() > 32) {
-            n = 4;
-        } else if (stack.getCount() > 16) {
-            n = 3;
-        } else if (stack.getCount() > 1) {
-            n = 2;
+    public int getRenderAmount(net.minecraft.world.item.ItemStack $$0) {
+        int $$1 = 1;
+        if ($$0.getCount() > 48) {
+            $$1 = 5;
+        } else if ($$0.getCount() > 32) {
+            $$1 = 4;
+        } else if ($$0.getCount() > 16) {
+            $$1 = 3;
+        } else if ($$0.getCount() > 1) {
+            $$1 = 2;
         }
-        return n;
+        return $$1;
     }
 
-    public void render(net.minecraft.world.entity.item.ItemEntity entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        float f;
-        float f2;
-        matrixStack.pushPose();
-        net.minecraft.world.item.ItemStack itemStack = entity.getItem();
-        int n = itemStack.isEmpty() ? 187 : net.minecraft.world.item.Item.getId((net.minecraft.world.item.Item)itemStack.getItem()) + itemStack.getDamageValue();
-        this.random.setSeed((long)n);
-        net.minecraft.client.resources.model.BakedModel bakedModel = this.itemRenderer.getModel(itemStack, entity.level, null, entity.getId());
-        boolean bl = bakedModel.isGui3d();
-        int n2 = this.getRenderAmount(itemStack);
-        float f3 = 0.25f;
-        float f4 = net.minecraft.util.Mth.sin((float)(((float)entity.getAge() + partialTicks) / 10.0f + entity.bobOffs)) * 0.1f + 0.1f;
-        float f5 = bakedModel.getTransforms().getTransform((net.minecraft.client.renderer.block.model.ItemTransforms$TransformType)net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND).scale.y();
-        matrixStack.translate(0.0, (double)(f4 + 0.25f * f5), 0.0);
-        float f6 = entity.getSpin(partialTicks);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotation(f6));
-        float f7 = bakedModel.getTransforms().ground.scale.x();
-        float f8 = bakedModel.getTransforms().ground.scale.y();
-        float f9 = bakedModel.getTransforms().ground.scale.z();
-        if (!bl) {
-            float f10 = -0.0f * (float)(n2 - 1) * 0.5f * f7;
-            f2 = -0.0f * (float)(n2 - 1) * 0.5f * f8;
-            f = -0.09375f * (float)(n2 - 1) * 0.5f * f9;
-            matrixStack.translate((double)f10, (double)f2, (double)f);
+    public void render(net.minecraft.world.entity.item.ItemEntity $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        net.minecraft.world.item.ItemStack $$6 = $$0.getItem();
+        int $$7 = $$6.isEmpty() ? 187 : net.minecraft.world.item.Item.getId((net.minecraft.world.item.Item)$$6.getItem()) + $$6.getDamageValue();
+        this.random.setSeed((long)$$7);
+        net.minecraft.client.resources.model.BakedModel $$8 = this.itemRenderer.getModel($$6, $$0.level, null, $$0.getId());
+        boolean $$9 = $$8.isGui3d();
+        int $$10 = this.getRenderAmount($$6);
+        float $$11 = 0.25f;
+        float $$12 = net.minecraft.util.Mth.sin((float)(((float)$$0.getAge() + $$2) / 10.0f + $$0.bobOffs)) * 0.1f + 0.1f;
+        float $$13 = $$8.getTransforms().getTransform((net.minecraft.client.renderer.block.model.ItemTransforms$TransformType)net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND).scale.y();
+        $$3.translate(0.0, (double)($$12 + 0.25f * $$13), 0.0);
+        float $$14 = $$0.getSpin($$2);
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotation($$14));
+        float $$15 = $$8.getTransforms().ground.scale.x();
+        float $$16 = $$8.getTransforms().ground.scale.y();
+        float $$17 = $$8.getTransforms().ground.scale.z();
+        if (!$$9) {
+            float $$18 = -0.0f * (float)($$10 - 1) * 0.5f * $$15;
+            float $$19 = -0.0f * (float)($$10 - 1) * 0.5f * $$16;
+            float $$20 = -0.09375f * (float)($$10 - 1) * 0.5f * $$17;
+            $$3.translate((double)$$18, (double)$$19, (double)$$20);
         }
-        for (int i = 0; i < n2; ++i) {
-            matrixStack.pushPose();
-            if (i > 0) {
-                if (bl) {
-                    f2 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f;
-                    f = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f;
-                    float f11 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f;
-                    matrixStack.translate((double)f2, (double)f, (double)f11);
+        for (int $$21 = 0; $$21 < $$10; ++$$21) {
+            $$3.pushPose();
+            if ($$21 > 0) {
+                if ($$9) {
+                    float $$22 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f;
+                    float $$23 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f;
+                    float $$24 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f;
+                    $$3.translate((double)$$22, (double)$$23, (double)$$24);
                 } else {
-                    f2 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f * 0.5f;
-                    f = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f * 0.5f;
-                    matrixStack.translate((double)f2, (double)f, 0.0);
+                    float $$25 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f * 0.5f;
+                    float $$26 = (this.random.nextFloat() * 2.0f - 1.0f) * 0.15f * 0.5f;
+                    $$3.translate((double)$$25, (double)$$26, 0.0);
                 }
             }
-            this.itemRenderer.render(itemStack, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, matrixStack, buffer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, bakedModel);
-            matrixStack.popPose();
-            if (bl) continue;
-            matrixStack.translate((double)(0.0f * f7), (double)(0.0f * f8), (double)(0.09375f * f9));
+            this.itemRenderer.render($$6, net.minecraft.client.renderer.block.model.ItemTransforms$TransformType.GROUND, false, $$3, $$4, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, $$8);
+            $$3.popPose();
+            if ($$9) continue;
+            $$3.translate((double)(0.0f * $$15), (double)(0.0f * $$16), (double)(0.09375f * $$17));
         }
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.item.ItemEntity entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.item.ItemEntity $$0) {
         return net.minecraft.client.renderer.texture.TextureAtlas.LOCATION_BLOCKS;
     }
 }