--- net/minecraft/client/renderer/entity/LeashKnotRenderer.java
+++ net/minecraft/client/renderer/entity/LeashKnotRenderer.java
@@ -14,17 +14,17 @@
         this.model = new net.minecraft.client.model.LeashKnotModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.LEASH_KNOT));
     }
 
-    public void render(net.minecraft.world.entity.decoration.LeashFenceKnotEntity entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        matrixStack.scale(-1.0f, -1.0f, 1.0f);
-        this.model.setupAnim((net.minecraft.world.entity.Entity)entity, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(this.model.renderType(KNOT_LOCATION));
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.decoration.LeashFenceKnotEntity $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        $$3.scale(-1.0f, -1.0f, 1.0f);
+        this.model.setupAnim((net.minecraft.world.entity.Entity)$$0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$6 = $$4.getBuffer(this.model.renderType(KNOT_LOCATION));
+        this.model.renderToBuffer($$3, $$6, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.decoration.LeashFenceKnotEntity entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.decoration.LeashFenceKnotEntity $$0) {
         return KNOT_LOCATION;
     }
 