--- net/minecraft/client/renderer/entity/LlamaRenderer.java
+++ net/minecraft/client/renderer/entity/LlamaRenderer.java
@@ -13,8 +13,8 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.LlamaDecorLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.horse.Llama entity) {
-        return LLAMA_LOCATIONS[entity.getVariant()];
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.horse.Llama $$0) {
+        return LLAMA_LOCATIONS[$$0.getVariant()];
     }
 
     static {