--- net/minecraft/client/renderer/entity/MagmaCubeRenderer.java
+++ net/minecraft/client/renderer/entity/MagmaCubeRenderer.java
@@ -12,19 +12,19 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.LavaSlimeModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.MAGMA_CUBE)), 0.25f);
     }
 
-    public int getBlockLightLevel(net.minecraft.world.entity.monster.MagmaCube entity, net.minecraft.core.BlockPos pos) {
+    public int getBlockLightLevel(net.minecraft.world.entity.monster.MagmaCube $$0, net.minecraft.core.BlockPos $$1) {
         return 15;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.MagmaCube entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.MagmaCube $$0) {
         return MAGMACUBE_LOCATION;
     }
 
-    public void scale(net.minecraft.world.entity.monster.MagmaCube livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        int n = livingEntity.getSize();
-        float f = net.minecraft.util.Mth.lerp((float)partialTickTime, (float)livingEntity.oSquish, (float)livingEntity.squish) / ((float)n * 0.5f + 1.0f);
-        float f2 = 1.0f / (f + 1.0f);
-        matrixStack.scale(f2 * (float)n, 1.0f / f2 * (float)n, f2 * (float)n);
+    public void scale(net.minecraft.world.entity.monster.MagmaCube $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        int $$3 = $$0.getSize();
+        float $$4 = net.minecraft.util.Mth.lerp((float)$$2, (float)$$0.oSquish, (float)$$0.squish) / ((float)$$3 * 0.5f + 1.0f);
+        float $$5 = 1.0f / ($$4 + 1.0f);
+        $$1.scale($$5 * (float)$$3, 1.0f / $$5 * (float)$$3, $$5 * (float)$$3);
     }
 
     static {