--- net/minecraft/client/renderer/entity/MinecartRenderer.java
+++ net/minecraft/client/renderer/entity/MinecartRenderer.java
@@ -17,73 +17,73 @@
         this.blockRenderer = $$0.getBlockRenderDispatcher();
     }
 
-    public void render(T entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        super.render(entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
-        matrixStack.pushPose();
-        long l = (long)entity.getId() * 493286711L;
-        l = l * l * 4392167121L + l * 98761L;
-        float f = (((float)(l >> 16 & 7L) + 0.5f) / 8.0f - 0.5f) * 0.004f;
-        float f2 = (((float)(l >> 20 & 7L) + 0.5f) / 8.0f - 0.5f) * 0.004f;
-        float f3 = (((float)(l >> 24 & 7L) + 0.5f) / 8.0f - 0.5f) * 0.004f;
-        matrixStack.translate((double)f, (double)f2, (double)f3);
-        double d = net.minecraft.util.Mth.lerp((double)partialTicks, (double)((net.minecraft.world.entity.vehicle.AbstractMinecart)entity).xOld, (double)entity.getX());
-        double d2 = net.minecraft.util.Mth.lerp((double)partialTicks, (double)((net.minecraft.world.entity.vehicle.AbstractMinecart)entity).yOld, (double)entity.getY());
-        double d3 = net.minecraft.util.Mth.lerp((double)partialTicks, (double)((net.minecraft.world.entity.vehicle.AbstractMinecart)entity).zOld, (double)entity.getZ());
-        double d4 = (double)0.3f;
-        net.minecraft.world.phys.Vec3 vec3 = entity.getPos(d, d2, d3);
-        float f4 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)((net.minecraft.world.entity.vehicle.AbstractMinecart)entity).xRotO, (float)entity.getXRot());
-        if (vec3 != null) {
-            net.minecraft.world.phys.Vec3 vec32 = entity.getPosOffs(d, d2, d3, (double)0.3f);
-            net.minecraft.world.phys.Vec3 vec33 = entity.getPosOffs(d, d2, d3, (double)-0.3f);
-            if (vec32 == null) {
-                vec32 = vec3;
+    public void render(T $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        super.render($$0, $$1, $$2, $$3, $$4, $$5);
+        $$3.pushPose();
+        long $$6 = (long)$$0.getId() * 493286711L;
+        $$6 = $$6 * $$6 * 4392167121L + $$6 * 98761L;
+        float $$7 = (((float)($$6 >> 16 & 7L) + 0.5f) / 8.0f - 0.5f) * 0.004f;
+        float $$8 = (((float)($$6 >> 20 & 7L) + 0.5f) / 8.0f - 0.5f) * 0.004f;
+        float $$9 = (((float)($$6 >> 24 & 7L) + 0.5f) / 8.0f - 0.5f) * 0.004f;
+        $$3.translate((double)$$7, (double)$$8, (double)$$9);
+        double $$10 = net.minecraft.util.Mth.lerp((double)$$2, (double)((net.minecraft.world.entity.vehicle.AbstractMinecart)$$0).xOld, (double)$$0.getX());
+        double $$11 = net.minecraft.util.Mth.lerp((double)$$2, (double)((net.minecraft.world.entity.vehicle.AbstractMinecart)$$0).yOld, (double)$$0.getY());
+        double $$12 = net.minecraft.util.Mth.lerp((double)$$2, (double)((net.minecraft.world.entity.vehicle.AbstractMinecart)$$0).zOld, (double)$$0.getZ());
+        double $$13 = (double)0.3f;
+        net.minecraft.world.phys.Vec3 $$14 = $$0.getPos($$10, $$11, $$12);
+        float $$15 = net.minecraft.util.Mth.lerp((float)$$2, (float)((net.minecraft.world.entity.vehicle.AbstractMinecart)$$0).xRotO, (float)$$0.getXRot());
+        if ($$14 != null) {
+            net.minecraft.world.phys.Vec3 $$16 = $$0.getPosOffs($$10, $$11, $$12, (double)0.3f);
+            net.minecraft.world.phys.Vec3 $$17 = $$0.getPosOffs($$10, $$11, $$12, (double)-0.3f);
+            if ($$16 == null) {
+                $$16 = $$14;
             }
-            if (vec33 == null) {
-                vec33 = vec3;
+            if ($$17 == null) {
+                $$17 = $$14;
             }
-            matrixStack.translate(vec3.x - d, (vec32.y + vec33.y) / 2.0 - d2, vec3.z - d3);
-            net.minecraft.world.phys.Vec3 vec34 = vec33.add(-vec32.x, -vec32.y, -vec32.z);
-            if (vec34.length() != 0.0) {
-                vec34 = vec34.normalize();
-                entityYaw = (float)(java.lang.Math.atan2((double)vec34.z, (double)vec34.x) * 180.0 / java.lang.Math.PI);
-                f4 = (float)(java.lang.Math.atan((double)vec34.y) * 73.0);
+            $$3.translate($$14.x - $$10, ($$16.y + $$17.y) / 2.0 - $$11, $$14.z - $$12);
+            net.minecraft.world.phys.Vec3 $$18 = $$17.add(-$$16.x, -$$16.y, -$$16.z);
+            if ($$18.length() != 0.0) {
+                $$18 = $$18.normalize();
+                $$1 = (float)(java.lang.Math.atan2((double)$$18.z, (double)$$18.x) * 180.0 / java.lang.Math.PI);
+                $$15 = (float)(java.lang.Math.atan((double)$$18.y) * 73.0);
             }
         }
-        matrixStack.translate(0.0, 0.375, 0.0);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - entityYaw));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(-f4));
-        float f5 = (float)entity.getHurtTime() - partialTicks;
-        float f6 = entity.getDamage() - partialTicks;
-        if (f6 < 0.0f) {
-            f6 = 0.0f;
+        $$3.translate(0.0, 0.375, 0.0);
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - $$1));
+        $$3.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(-$$15));
+        float $$19 = (float)$$0.getHurtTime() - $$2;
+        float $$20 = $$0.getDamage() - $$2;
+        if ($$20 < 0.0f) {
+            $$20 = 0.0f;
         }
-        if (f5 > 0.0f) {
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.sin((float)f5) * f5 * f6 / 10.0f * (float)entity.getHurtDir()));
+        if ($$19 > 0.0f) {
+            $$3.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.sin((float)$$19) * $$19 * $$20 / 10.0f * (float)$$0.getHurtDir()));
         }
-        int n = entity.getDisplayOffset();
-        net.minecraft.world.level.block.state.BlockState blockState = entity.getDisplayBlockState();
-        if (blockState.getRenderShape() != net.minecraft.world.level.block.RenderShape.INVISIBLE) {
-            matrixStack.pushPose();
-            float f7 = 0.75f;
-            matrixStack.scale(0.75f, 0.75f, 0.75f);
-            matrixStack.translate(-0.5, (double)((float)(n - 8) / 16.0f), 0.5);
-            matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f));
-            this.renderMinecartContents(entity, partialTicks, blockState, matrixStack, buffer, packedLight);
-            matrixStack.popPose();
+        int $$21 = $$0.getDisplayOffset();
+        net.minecraft.world.level.block.state.BlockState $$22 = $$0.getDisplayBlockState();
+        if ($$22.getRenderShape() != net.minecraft.world.level.block.RenderShape.INVISIBLE) {
+            $$3.pushPose();
+            float $$23 = 0.75f;
+            $$3.scale(0.75f, 0.75f, 0.75f);
+            $$3.translate(-0.5, (double)((float)($$21 - 8) / 16.0f), 0.5);
+            $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f));
+            this.renderMinecartContents($$0, $$2, $$22, $$3, $$4, $$5);
+            $$3.popPose();
         }
-        matrixStack.scale(-1.0f, -1.0f, 1.0f);
-        this.model.setupAnim(entity, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(this.model.renderType(this.getTextureLocation(entity)));
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
+        $$3.scale(-1.0f, -1.0f, 1.0f);
+        this.model.setupAnim($$0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$24 = $$4.getBuffer(this.model.renderType(this.getTextureLocation($$0)));
+        this.model.renderToBuffer($$3, $$24, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$3.popPose();
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(T entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(T $$0) {
         return MINECART_LOCATION;
     }
 
-    public void renderMinecartContents(T entity, float partialTicks, net.minecraft.world.level.block.state.BlockState state, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        this.blockRenderer.renderSingleBlock(state, matrixStack, buffer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
+    public void renderMinecartContents(T $$0, float $$1, net.minecraft.world.level.block.state.BlockState $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        this.blockRenderer.renderSingleBlock($$2, $$3, $$4, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY);
     }
 
     static {