--- net/minecraft/client/renderer/entity/MushroomCowRenderer.java
+++ net/minecraft/client/renderer/entity/MushroomCowRenderer.java
@@ -13,8 +13,8 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.MushroomCowMushroomLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getBlockRenderDispatcher()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.MushroomCow entity) {
-        return (net.minecraft.resources.ResourceLocation)TEXTURES.get((java.lang.Object)entity.getMushroomType());
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.MushroomCow $$0) {
+        return (net.minecraft.resources.ResourceLocation)TEXTURES.get((java.lang.Object)$$0.getMushroomType());
     }
 
     public static /* synthetic */ void lambda$static$0(java.util.HashMap $$0) {