--- net/minecraft/client/renderer/entity/OcelotRenderer.java
+++ net/minecraft/client/renderer/entity/OcelotRenderer.java
@@ -12,7 +12,7 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.OcelotModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.OCELOT)), 0.4f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Ocelot entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Ocelot $$0) {
         return CAT_OCELOT_LOCATION;
     }
 