--- net/minecraft/client/renderer/entity/PaintingRenderer.java
+++ net/minecraft/client/renderer/entity/PaintingRenderer.java
@@ -10,21 +10,21 @@
         super($$0);
     }
 
-    public void render(net.minecraft.world.entity.decoration.Painting entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - entityYaw));
-        net.minecraft.world.entity.decoration.PaintingVariant paintingVariant = (net.minecraft.world.entity.decoration.PaintingVariant)entity.getVariant().value();
-        float f = 0.0625f;
-        matrixStack.scale(0.0625f, 0.0625f, 0.0625f);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)this.getTextureLocation(entity)));
-        net.minecraft.client.resources.PaintingTextureManager paintingTextureManager = net.minecraft.client.Minecraft.getInstance().getPaintingTextures();
-        this.renderPainting(matrixStack, vertexConsumer, entity, paintingVariant.getWidth(), paintingVariant.getHeight(), paintingTextureManager.get(paintingVariant), paintingTextureManager.getBackSprite());
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.decoration.Painting $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - $$1));
+        net.minecraft.world.entity.decoration.PaintingVariant $$6 = (net.minecraft.world.entity.decoration.PaintingVariant)$$0.getVariant().value();
+        float $$7 = 0.0625f;
+        $$3.scale(0.0625f, 0.0625f, 0.0625f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$8 = $$4.getBuffer(net.minecraft.client.renderer.RenderType.entitySolid((net.minecraft.resources.ResourceLocation)this.getTextureLocation($$0)));
+        net.minecraft.client.resources.PaintingTextureManager $$9 = net.minecraft.client.Minecraft.getInstance().getPaintingTextures();
+        this.renderPainting($$3, $$8, $$0, $$6.getWidth(), $$6.getHeight(), $$9.get($$6), $$9.getBackSprite());
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.decoration.Painting entity) {
-        return net.minecraft.client.Minecraft.getInstance().getPaintingTextures().getBackSprite().atlas().location();
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.decoration.Painting $$0) {
+        return net.minecraft.client.Minecraft.getInstance().getPaintingTextures().getBackSprite().atlasLocation();
     }
 
     public void renderPainting(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, net.minecraft.world.entity.decoration.Painting $$2, int $$3, int $$4, net.minecraft.client.renderer.texture.TextureAtlasSprite $$5, net.minecraft.client.renderer.texture.TextureAtlasSprite $$6) {