--- net/minecraft/client/renderer/entity/PandaRenderer.java
+++ net/minecraft/client/renderer/entity/PandaRenderer.java
@@ -13,66 +13,65 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.PandaHoldsItemLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getItemInHandRenderer()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Panda entity) {
-        return (net.minecraft.resources.ResourceLocation)TEXTURES.getOrDefault((java.lang.Object)entity.getVariant(), (java.lang.Object)((net.minecraft.resources.ResourceLocation)TEXTURES.get((java.lang.Object)net.minecraft.world.entity.animal.Panda$Gene.NORMAL)));
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Panda $$0) {
+        return (net.minecraft.resources.ResourceLocation)TEXTURES.getOrDefault((java.lang.Object)$$0.getVariant(), (java.lang.Object)((net.minecraft.resources.ResourceLocation)TEXTURES.get((java.lang.Object)net.minecraft.world.entity.animal.Panda$Gene.NORMAL)));
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.Panda entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        float f;
-        float f2;
-        float f3;
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        if (entityLiving.rollCounter > 0) {
-            float f4;
-            int n = entityLiving.rollCounter;
-            int n2 = n + 1;
-            f3 = 7.0f;
-            float f5 = f4 = entityLiving.isBaby() ? 0.3f : 0.8f;
-            if (n < 8) {
-                float f6 = (float)(90 * n) / 7.0f;
-                float f7 = (float)(90 * n2) / 7.0f;
-                float f8 = this.getAngle(f6, f7, n2, partialTicks, 8.0f);
-                matrixStack.translate(0.0, (double)((f4 + 0.2f) * (f8 / 90.0f)), 0.0);
-                matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-f8));
-            } else if (n < 16) {
-                float f9 = ((float)n - 8.0f) / 7.0f;
-                float f10 = 90.0f + 90.0f * f9;
-                float f11 = 90.0f + 90.0f * ((float)n2 - 8.0f) / 7.0f;
-                float f12 = this.getAngle(f10, f11, n2, partialTicks, 16.0f);
-                matrixStack.translate(0.0, (double)(f4 + 0.2f + (f4 - 0.2f) * (f12 - 90.0f) / 90.0f), 0.0);
-                matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-f12));
-            } else if ((float)n < 24.0f) {
-                float f13 = ((float)n - 16.0f) / 7.0f;
-                float f14 = 180.0f + 90.0f * f13;
-                float f15 = 180.0f + 90.0f * ((float)n2 - 16.0f) / 7.0f;
-                float f16 = this.getAngle(f14, f15, n2, partialTicks, 24.0f);
-                matrixStack.translate(0.0, (double)(f4 + f4 * (270.0f - f16) / 90.0f), 0.0);
-                matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-f16));
-            } else if (n < 32) {
-                float f17 = ((float)n - 24.0f) / 7.0f;
-                float f18 = 270.0f + 90.0f * f17;
-                float f19 = 270.0f + 90.0f * ((float)n2 - 24.0f) / 7.0f;
-                float f20 = this.getAngle(f18, f19, n2, partialTicks, 32.0f);
-                matrixStack.translate(0.0, (double)(f4 * ((360.0f - f20) / 90.0f)), 0.0);
-                matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-f20));
+    public void setupRotations(net.minecraft.world.entity.animal.Panda $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        float $$26;
+        float $$24;
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        if ($$0.rollCounter > 0) {
+            float $$8;
+            int $$5 = $$0.rollCounter;
+            int $$6 = $$5 + 1;
+            float $$7 = 7.0f;
+            float f = $$8 = $$0.isBaby() ? 0.3f : 0.8f;
+            if ($$5 < 8) {
+                float $$9 = (float)(90 * $$5) / 7.0f;
+                float $$10 = (float)(90 * $$6) / 7.0f;
+                float $$11 = this.getAngle($$9, $$10, $$6, $$4, 8.0f);
+                $$1.translate(0.0, (double)(($$8 + 0.2f) * ($$11 / 90.0f)), 0.0);
+                $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-$$11));
+            } else if ($$5 < 16) {
+                float $$12 = ((float)$$5 - 8.0f) / 7.0f;
+                float $$13 = 90.0f + 90.0f * $$12;
+                float $$14 = 90.0f + 90.0f * ((float)$$6 - 8.0f) / 7.0f;
+                float $$15 = this.getAngle($$13, $$14, $$6, $$4, 16.0f);
+                $$1.translate(0.0, (double)($$8 + 0.2f + ($$8 - 0.2f) * ($$15 - 90.0f) / 90.0f), 0.0);
+                $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-$$15));
+            } else if ((float)$$5 < 24.0f) {
+                float $$16 = ((float)$$5 - 16.0f) / 7.0f;
+                float $$17 = 180.0f + 90.0f * $$16;
+                float $$18 = 180.0f + 90.0f * ((float)$$6 - 16.0f) / 7.0f;
+                float $$19 = this.getAngle($$17, $$18, $$6, $$4, 24.0f);
+                $$1.translate(0.0, (double)($$8 + $$8 * (270.0f - $$19) / 90.0f), 0.0);
+                $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-$$19));
+            } else if ($$5 < 32) {
+                float $$20 = ((float)$$5 - 24.0f) / 7.0f;
+                float $$21 = 270.0f + 90.0f * $$20;
+                float $$22 = 270.0f + 90.0f * ((float)$$6 - 24.0f) / 7.0f;
+                float $$23 = this.getAngle($$21, $$22, $$6, $$4, 32.0f);
+                $$1.translate(0.0, (double)($$8 * ((360.0f - $$23) / 90.0f)), 0.0);
+                $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(-$$23));
             }
         }
-        if ((f2 = entityLiving.getSitAmount(partialTicks)) > 0.0f) {
-            matrixStack.translate(0.0, (double)(0.8f * f2), 0.0);
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.lerp((float)f2, (float)entityLiving.getXRot(), (float)(entityLiving.getXRot() + 90.0f))));
-            matrixStack.translate(0.0, (double)(-1.0f * f2), 0.0);
-            if (entityLiving.isScared()) {
-                float f21 = (float)(java.lang.Math.cos((double)((double)entityLiving.tickCount * 1.25)) * java.lang.Math.PI * (double)0.05f);
-                matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f21));
-                if (entityLiving.isBaby()) {
-                    matrixStack.translate(0.0, (double)0.8f, (double)0.55f);
+        if (($$24 = $$0.getSitAmount($$4)) > 0.0f) {
+            $$1.translate(0.0, (double)(0.8f * $$24), 0.0);
+            $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.lerp((float)$$24, (float)$$0.getXRot(), (float)($$0.getXRot() + 90.0f))));
+            $$1.translate(0.0, (double)(-1.0f * $$24), 0.0);
+            if ($$0.isScared()) {
+                float $$25 = (float)(java.lang.Math.cos((double)((double)$$0.tickCount * 1.25)) * java.lang.Math.PI * (double)0.05f);
+                $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$25));
+                if ($$0.isBaby()) {
+                    $$1.translate(0.0, (double)0.8f, (double)0.55f);
                 }
             }
         }
-        if ((f = entityLiving.getLieOnBackAmount(partialTicks)) > 0.0f) {
-            f3 = entityLiving.isBaby() ? 0.5f : 1.3f;
-            matrixStack.translate(0.0, (double)(f3 * f), 0.0);
-            matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.lerp((float)f, (float)entityLiving.getXRot(), (float)(entityLiving.getXRot() + 180.0f))));
+        if (($$26 = $$0.getLieOnBackAmount($$4)) > 0.0f) {
+            float $$27 = $$0.isBaby() ? 0.5f : 1.3f;
+            $$1.translate(0.0, (double)($$27 * $$26), 0.0);
+            $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.lerp((float)$$26, (float)$$0.getXRot(), (float)($$0.getXRot() + 180.0f))));
         }
     }
 