--- net/minecraft/client/renderer/entity/PhantomRenderer.java
+++ net/minecraft/client/renderer/entity/PhantomRenderer.java
@@ -13,20 +13,20 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.PhantomEyesLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Phantom entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Phantom $$0) {
         return PHANTOM_LOCATION;
     }
 
-    public void scale(net.minecraft.world.entity.monster.Phantom livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        int n = livingEntity.getPhantomSize();
-        float f = 1.0f + 0.15f * (float)n;
-        matrixStack.scale(f, f, f);
-        matrixStack.translate(0.0, 1.3125, 0.1875);
+    public void scale(net.minecraft.world.entity.monster.Phantom $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        int $$3 = $$0.getPhantomSize();
+        float $$4 = 1.0f + 0.15f * (float)$$3;
+        $$1.scale($$4, $$4, $$4);
+        $$1.translate(0.0, 1.3125, 0.1875);
     }
 
-    public void setupRotations(net.minecraft.world.entity.monster.Phantom entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(entityLiving.getXRot()));
+    public void setupRotations(net.minecraft.world.entity.monster.Phantom $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$0.getXRot()));
     }
 
     static {