--- net/minecraft/client/renderer/entity/PiglinRenderer.java
+++ net/minecraft/client/renderer/entity/PiglinRenderer.java
@@ -22,12 +22,12 @@
         return $$3;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.Mob entity) {
-        net.minecraft.resources.ResourceLocation resourceLocation = (net.minecraft.resources.ResourceLocation)TEXTURES.get((java.lang.Object)entity.getType());
-        if (resourceLocation == null) {
-            throw new java.lang.IllegalArgumentException("I don't know what texture to use for " + entity.getType());
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.Mob $$0) {
+        net.minecraft.resources.ResourceLocation $$1 = (net.minecraft.resources.ResourceLocation)TEXTURES.get((java.lang.Object)$$0.getType());
+        if ($$1 == null) {
+            throw new java.lang.IllegalArgumentException("I don't know what texture to use for " + $$0.getType());
         }
-        return resourceLocation;
+        return $$1;
     }
 
     public boolean isShaking(net.minecraft.world.entity.Mob $$0) {