--- net/minecraft/client/renderer/entity/PigRenderer.java
+++ net/minecraft/client/renderer/entity/PigRenderer.java
@@ -13,7 +13,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.SaddleLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.PigModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.PIG_SADDLE)), new net.minecraft.resources.ResourceLocation("textures/entity/pig/pig_saddle.png")));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Pig entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Pig $$0) {
         return PIG_LOCATION;
     }
 