--- net/minecraft/client/renderer/entity/PufferfishRenderer.java
+++ net/minecraft/client/renderer/entity/PufferfishRenderer.java
@@ -20,23 +20,23 @@
         this.small = new net.minecraft.client.model.PufferfishSmallModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.PUFFERFISH_SMALL));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Pufferfish entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Pufferfish $$0) {
         return PUFFER_LOCATION;
     }
 
-    public void render(net.minecraft.world.entity.animal.Pufferfish entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        int n = entity.getPuffState();
-        if (n != this.puffStateO) {
-            this.model = n == 0 ? this.small : (n == 1 ? this.mid : this.big);
+    public void render(net.minecraft.world.entity.animal.Pufferfish $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        int $$6 = $$0.getPuffState();
+        if ($$6 != this.puffStateO) {
+            this.model = $$6 == 0 ? this.small : ($$6 == 1 ? this.mid : this.big);
         }
-        this.puffStateO = n;
-        this.shadowRadius = 0.1f + 0.1f * (float)n;
-        super.render((net.minecraft.world.entity.Mob)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        this.puffStateO = $$6;
+        this.shadowRadius = 0.1f + 0.1f * (float)$$6;
+        super.render((net.minecraft.world.entity.Mob)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.Pufferfish entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        matrixStack.translate(0.0, (double)(net.minecraft.util.Mth.cos((float)(ageInTicks * 0.05f)) * 0.08f), 0.0);
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
+    public void setupRotations(net.minecraft.world.entity.animal.Pufferfish $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        $$1.translate(0.0, (double)(net.minecraft.util.Mth.cos((float)($$2 * 0.05f)) * 0.08f), 0.0);
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
     }
 
     static {