--- net/minecraft/client/renderer/entity/RavagerRenderer.java
+++ net/minecraft/client/renderer/entity/RavagerRenderer.java
@@ -12,7 +12,7 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.RavagerModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.RAVAGER)), 1.1f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Ravager entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Ravager $$0) {
         return TEXTURE_LOCATION;
     }
 