--- net/minecraft/client/renderer/entity/SalmonRenderer.java
+++ net/minecraft/client/renderer/entity/SalmonRenderer.java
@@ -12,24 +12,24 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.SalmonModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.SALMON)), 0.4f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Salmon entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Salmon $$0) {
         return SALMON_LOCATION;
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.Salmon entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        float f = 1.0f;
-        float f2 = 1.0f;
-        if (!entityLiving.isInWater()) {
-            f = 1.3f;
-            f2 = 1.7f;
+    public void setupRotations(net.minecraft.world.entity.animal.Salmon $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        float $$5 = 1.0f;
+        float $$6 = 1.0f;
+        if (!$$0.isInWater()) {
+            $$5 = 1.3f;
+            $$6 = 1.7f;
         }
-        float f3 = f * 4.3f * net.minecraft.util.Mth.sin((float)(f2 * 0.6f * ageInTicks));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f3));
-        matrixStack.translate(0.0, 0.0, (double)-0.4f);
-        if (!entityLiving.isInWater()) {
-            matrixStack.translate((double)0.2f, (double)0.1f, 0.0);
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
+        float $$7 = $$5 * 4.3f * net.minecraft.util.Mth.sin((float)($$6 * 0.6f * $$2));
+        $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$7));
+        $$1.translate(0.0, 0.0, (double)-0.4f);
+        if (!$$0.isInWater()) {
+            $$1.translate((double)0.2f, (double)0.1f, 0.0);
+            $$1.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
         }
     }
 