--- net/minecraft/client/renderer/entity/SheepRenderer.java
+++ net/minecraft/client/renderer/entity/SheepRenderer.java
@@ -13,7 +13,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.SheepFurLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Sheep entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Sheep $$0) {
         return SHEEP_LOCATION;
     }
 