--- net/minecraft/client/renderer/entity/ShulkerRenderer.java
+++ net/minecraft/client/renderer/entity/ShulkerRenderer.java
@@ -14,25 +14,25 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.ShulkerHeadLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this));
     }
 
-    public net.minecraft.world.phys.Vec3 getRenderOffset(net.minecraft.world.entity.monster.Shulker entity, float partialTicks) {
-        return (net.minecraft.world.phys.Vec3)entity.getRenderPosition(partialTicks).orElse((java.lang.Object)super.getRenderOffset((net.minecraft.world.entity.Entity)entity, partialTicks));
+    public net.minecraft.world.phys.Vec3 getRenderOffset(net.minecraft.world.entity.monster.Shulker $$0, float $$1) {
+        return (net.minecraft.world.phys.Vec3)$$0.getRenderPosition($$1).orElse((java.lang.Object)super.getRenderOffset((net.minecraft.world.entity.Entity)$$0, $$1));
     }
 
-    public boolean shouldRender(net.minecraft.world.entity.monster.Shulker livingEntity, net.minecraft.client.renderer.culling.Frustum camera, double camX, double camY, double camZ) {
-        if (super.shouldRender((net.minecraft.world.entity.Mob)livingEntity, camera, camX, camY, camZ)) {
+    public boolean shouldRender(net.minecraft.world.entity.monster.Shulker $$0, net.minecraft.client.renderer.culling.Frustum $$1, double $$22, double $$3, double $$4) {
+        if (super.shouldRender((net.minecraft.world.entity.Mob)$$0, $$1, $$22, $$3, $$4)) {
             return true;
         }
-        return livingEntity.getRenderPosition(0.0f).filter($$2 -> {
-            net.minecraft.world.entity.EntityType $$3 = livingEntity.getType();
+        return $$0.getRenderPosition(0.0f).filter($$2 -> {
+            net.minecraft.world.entity.EntityType $$3 = $$0.getType();
             float $$4 = $$3.getHeight() / 2.0f;
             float $$5 = $$3.getWidth() / 2.0f;
-            net.minecraft.world.phys.Vec3 $$6 = net.minecraft.world.phys.Vec3.atBottomCenterOf((net.minecraft.core.Vec3i)livingEntity.blockPosition());
-            return camera.isVisible(new net.minecraft.world.phys.AABB($$2.x, $$2.y + (double)$$4, $$2.z, $$6.x, $$6.y + (double)$$4, $$6.z).inflate((double)$$5, (double)$$4, (double)$$5));
+            net.minecraft.world.phys.Vec3 $$6 = net.minecraft.world.phys.Vec3.atBottomCenterOf((net.minecraft.core.Vec3i)$$0.blockPosition());
+            return $$1.isVisible(new net.minecraft.world.phys.AABB($$2.x, $$2.y + (double)$$4, $$2.z, $$6.x, $$6.y + (double)$$4, $$6.z).inflate((double)$$5, (double)$$4, (double)$$5));
         }).isPresent();
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Shulker entity) {
-        return net.minecraft.client.renderer.entity.ShulkerRenderer.getTextureLocation((net.minecraft.world.item.DyeColor)entity.getColor());
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Shulker $$0) {
+        return net.minecraft.client.renderer.entity.ShulkerRenderer.getTextureLocation((net.minecraft.world.item.DyeColor)$$0.getColor());
     }
 
     public static net.minecraft.resources.ResourceLocation getTextureLocation(@javax.annotation.Nullable net.minecraft.world.item.DyeColor $$0) {
@@ -42,11 +42,11 @@
         return TEXTURE_LOCATION[$$0.getId()];
     }
 
-    public void setupRotations(net.minecraft.world.entity.monster.Shulker entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw + 180.0f, partialTicks);
-        matrixStack.translate(0.0, 0.5, 0.0);
-        matrixStack.mulPose(entityLiving.getAttachFace().getOpposite().getRotation());
-        matrixStack.translate(0.0, -0.5, 0.0);
+    public void setupRotations(net.minecraft.world.entity.monster.Shulker $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3 + 180.0f, $$4);
+        $$1.translate(0.0, 0.5, 0.0);
+        $$1.mulPose($$0.getAttachFace().getOpposite().getRotation());
+        $$1.translate(0.0, -0.5, 0.0);
     }
 
     public static /* synthetic */ net.minecraft.resources.ResourceLocation[] lambda$static$1(int $$0) {