--- net/minecraft/client/renderer/entity/ShulkerBulletRenderer.java
+++ net/minecraft/client/renderer/entity/ShulkerBulletRenderer.java
@@ -15,31 +15,31 @@
         this.model = new net.minecraft.client.model.ShulkerBulletModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.SHULKER_BULLET));
     }
 
-    public int getBlockLightLevel(net.minecraft.world.entity.projectile.ShulkerBullet entity, net.minecraft.core.BlockPos pos) {
+    public int getBlockLightLevel(net.minecraft.world.entity.projectile.ShulkerBullet $$0, net.minecraft.core.BlockPos $$1) {
         return 15;
     }
 
-    public void render(net.minecraft.world.entity.projectile.ShulkerBullet entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        float f = net.minecraft.util.Mth.rotlerp((float)entity.yRotO, (float)entity.getYRot(), (float)partialTicks);
-        float f2 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)entity.xRotO, (float)entity.getXRot());
-        float f3 = (float)entity.tickCount + partialTicks;
-        matrixStack.translate(0.0, (double)0.15f, 0.0);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(net.minecraft.util.Mth.sin((float)(f3 * 0.1f)) * 180.0f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.cos((float)(f3 * 0.1f)) * 180.0f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(net.minecraft.util.Mth.sin((float)(f3 * 0.15f)) * 360.0f));
-        matrixStack.scale(-0.5f, -0.5f, 0.5f);
-        this.model.setupAnim((net.minecraft.world.entity.Entity)entity, 0.0f, 0.0f, 0.0f, f, f2);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(this.model.renderType(TEXTURE_LOCATION));
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.scale(1.5f, 1.5f, 1.5f);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer2 = buffer.getBuffer(RENDER_TYPE);
-        this.model.renderToBuffer(matrixStack, vertexConsumer2, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 0.15f);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.projectile.ShulkerBullet $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        float $$6 = net.minecraft.util.Mth.rotlerp((float)$$0.yRotO, (float)$$0.getYRot(), (float)$$2);
+        float $$7 = net.minecraft.util.Mth.lerp((float)$$2, (float)$$0.xRotO, (float)$$0.getXRot());
+        float $$8 = (float)$$0.tickCount + $$2;
+        $$3.translate(0.0, (double)0.15f, 0.0);
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(net.minecraft.util.Mth.sin((float)($$8 * 0.1f)) * 180.0f));
+        $$3.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(net.minecraft.util.Mth.cos((float)($$8 * 0.1f)) * 180.0f));
+        $$3.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(net.minecraft.util.Mth.sin((float)($$8 * 0.15f)) * 360.0f));
+        $$3.scale(-0.5f, -0.5f, 0.5f);
+        this.model.setupAnim((net.minecraft.world.entity.Entity)$$0, 0.0f, 0.0f, 0.0f, $$6, $$7);
+        com.mojang.blaze3d.vertex.VertexConsumer $$9 = $$4.getBuffer(this.model.renderType(TEXTURE_LOCATION));
+        this.model.renderToBuffer($$3, $$9, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$3.scale(1.5f, 1.5f, 1.5f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$10 = $$4.getBuffer(RENDER_TYPE);
+        this.model.renderToBuffer($$3, $$10, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 0.15f);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.ShulkerBullet entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.ShulkerBullet $$0) {
         return TEXTURE_LOCATION;
     }
 