--- net/minecraft/client/renderer/entity/SilverfishRenderer.java
+++ net/minecraft/client/renderer/entity/SilverfishRenderer.java
@@ -12,11 +12,11 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.SilverfishModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.SILVERFISH)), 0.3f);
     }
 
-    public float getFlipDegrees(net.minecraft.world.entity.monster.Silverfish livingEntity) {
+    public float getFlipDegrees(net.minecraft.world.entity.monster.Silverfish $$0) {
         return 180.0f;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Silverfish entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Silverfish $$0) {
         return SILVERFISH_LOCATION;
     }
 