--- net/minecraft/client/renderer/entity/SkeletonRenderer.java
+++ net/minecraft/client/renderer/entity/SkeletonRenderer.java
@@ -17,7 +17,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, (net.minecraft.client.model.HumanoidModel)new net.minecraft.client.model.SkeletonModel($$0.bakeLayer($$2)), (net.minecraft.client.model.HumanoidModel)new net.minecraft.client.model.SkeletonModel($$0.bakeLayer($$3))));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.AbstractSkeleton entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.AbstractSkeleton $$0) {
         return SKELETON_LOCATION;
     }
 