--- net/minecraft/client/renderer/entity/SlimeRenderer.java
+++ net/minecraft/client/renderer/entity/SlimeRenderer.java
@@ -13,22 +13,22 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.SlimeOuterLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public void render(net.minecraft.world.entity.monster.Slime entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        this.shadowRadius = 0.25f * (float)entity.getSize();
-        super.render((net.minecraft.world.entity.Mob)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.monster.Slime $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        this.shadowRadius = 0.25f * (float)$$0.getSize();
+        super.render((net.minecraft.world.entity.Mob)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public void scale(net.minecraft.world.entity.monster.Slime livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        float f = 0.999f;
-        matrixStack.scale(0.999f, 0.999f, 0.999f);
-        matrixStack.translate(0.0, (double)0.001f, 0.0);
-        float f2 = livingEntity.getSize();
-        float f3 = net.minecraft.util.Mth.lerp((float)partialTickTime, (float)livingEntity.oSquish, (float)livingEntity.squish) / (f2 * 0.5f + 1.0f);
-        float f4 = 1.0f / (f3 + 1.0f);
-        matrixStack.scale(f4 * f2, 1.0f / f4 * f2, f4 * f2);
+    public void scale(net.minecraft.world.entity.monster.Slime $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        float $$3 = 0.999f;
+        $$1.scale(0.999f, 0.999f, 0.999f);
+        $$1.translate(0.0, (double)0.001f, 0.0);
+        float $$4 = $$0.getSize();
+        float $$5 = net.minecraft.util.Mth.lerp((float)$$2, (float)$$0.oSquish, (float)$$0.squish) / ($$4 * 0.5f + 1.0f);
+        float $$6 = 1.0f / ($$5 + 1.0f);
+        $$1.scale($$6 * $$4, 1.0f / $$6 * $$4, $$6 * $$4);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Slime entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Slime $$0) {
         return SLIME_LOCATION;
     }
 