--- net/minecraft/client/renderer/entity/SpiderRenderer.java
+++ net/minecraft/client/renderer/entity/SpiderRenderer.java
@@ -17,11 +17,11 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.SpiderEyesLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this));
     }
 
-    public float getFlipDegrees(T livingEntity) {
+    public float getFlipDegrees(T $$0) {
         return 180.0f;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(T entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(T $$0) {
         return SPIDER_LOCATION;
     }
 