--- net/minecraft/client/renderer/entity/SquidRenderer.java
+++ net/minecraft/client/renderer/entity/SquidRenderer.java
@@ -12,22 +12,22 @@
         super($$0, $$1, 0.7f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(T entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(T $$0) {
         return SQUID_LOCATION;
     }
 
-    public void setupRotations(T entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        float f = net.minecraft.util.Mth.lerp((float)partialTicks, (float)((net.minecraft.world.entity.animal.Squid)entityLiving).xBodyRotO, (float)((net.minecraft.world.entity.animal.Squid)entityLiving).xBodyRot);
-        float f2 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)((net.minecraft.world.entity.animal.Squid)entityLiving).zBodyRotO, (float)((net.minecraft.world.entity.animal.Squid)entityLiving).zBodyRot);
-        matrixStack.translate(0.0, 0.5, 0.0);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - rotationYaw));
-        matrixStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f2));
-        matrixStack.translate(0.0, (double)-1.2f, 0.0);
+    public void setupRotations(T $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        float $$5 = net.minecraft.util.Mth.lerp((float)$$4, (float)((net.minecraft.world.entity.animal.Squid)$$0).xBodyRotO, (float)((net.minecraft.world.entity.animal.Squid)$$0).xBodyRot);
+        float $$6 = net.minecraft.util.Mth.lerp((float)$$4, (float)((net.minecraft.world.entity.animal.Squid)$$0).zBodyRotO, (float)((net.minecraft.world.entity.animal.Squid)$$0).zBodyRot);
+        $$1.translate(0.0, 0.5, 0.0);
+        $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(180.0f - $$3));
+        $$1.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees($$5));
+        $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$6));
+        $$1.translate(0.0, (double)-1.2f, 0.0);
     }
 
-    public float getBob(T livingBase, float partialTicks) {
-        return net.minecraft.util.Mth.lerp((float)partialTicks, (float)((net.minecraft.world.entity.animal.Squid)livingBase).oldTentacleAngle, (float)((net.minecraft.world.entity.animal.Squid)livingBase).tentacleAngle);
+    public float getBob(T $$0, float $$1) {
+        return net.minecraft.util.Mth.lerp((float)$$1, (float)((net.minecraft.world.entity.animal.Squid)$$0).oldTentacleAngle, (float)((net.minecraft.world.entity.animal.Squid)$$0).tentacleAngle);
     }
 
     static {