--- net/minecraft/client/renderer/entity/StriderRenderer.java
+++ net/minecraft/client/renderer/entity/StriderRenderer.java
@@ -14,13 +14,13 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.SaddleLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.StriderModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.STRIDER_SADDLE)), new net.minecraft.resources.ResourceLocation("textures/entity/strider/strider_saddle.png")));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Strider entity) {
-        return entity.isSuffocating() ? COLD_LOCATION : STRIDER_LOCATION;
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Strider $$0) {
+        return $$0.isSuffocating() ? COLD_LOCATION : STRIDER_LOCATION;
     }
 
-    public void scale(net.minecraft.world.entity.monster.Strider livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        if (livingEntity.isBaby()) {
-            matrixStack.scale(0.5f, 0.5f, 0.5f);
+    public void scale(net.minecraft.world.entity.monster.Strider $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        if ($$0.isBaby()) {
+            $$1.scale(0.5f, 0.5f, 0.5f);
             this.shadowRadius = 0.25f;
         } else {
             this.shadowRadius = 0.5f;