--- net/minecraft/client/renderer/entity/ThrownTridentRenderer.java
+++ net/minecraft/client/renderer/entity/ThrownTridentRenderer.java
@@ -14,17 +14,17 @@
         this.model = new net.minecraft.client.model.TridentModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.TRIDENT));
     }
 
-    public void render(net.minecraft.world.entity.projectile.ThrownTrident entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(net.minecraft.util.Mth.lerp((float)partialTicks, (float)entity.yRotO, (float)entity.getYRot()) - 90.0f));
-        matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(net.minecraft.util.Mth.lerp((float)partialTicks, (float)entity.xRotO, (float)entity.getXRot()) + 90.0f));
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = net.minecraft.client.renderer.entity.ItemRenderer.getFoilBufferDirect((net.minecraft.client.renderer.MultiBufferSource)buffer, (net.minecraft.client.renderer.RenderType)this.model.renderType(this.getTextureLocation(entity)), (boolean)false, (boolean)entity.isFoil());
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.projectile.ThrownTrident $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(net.minecraft.util.Mth.lerp((float)$$2, (float)$$0.yRotO, (float)$$0.getYRot()) - 90.0f));
+        $$3.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(net.minecraft.util.Mth.lerp((float)$$2, (float)$$0.xRotO, (float)$$0.getXRot()) + 90.0f));
+        com.mojang.blaze3d.vertex.VertexConsumer $$6 = net.minecraft.client.renderer.entity.ItemRenderer.getFoilBufferDirect((net.minecraft.client.renderer.MultiBufferSource)$$4, (net.minecraft.client.renderer.RenderType)this.model.renderType(this.getTextureLocation($$0)), (boolean)false, (boolean)$$0.isFoil());
+        this.model.renderToBuffer($$3, $$6, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.ThrownTrident entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.ThrownTrident $$0) {
         return TRIDENT_LOCATION;
     }
 