--- net/minecraft/client/renderer/entity/TntMinecartRenderer.java
+++ net/minecraft/client/renderer/entity/TntMinecartRenderer.java
@@ -13,17 +13,17 @@
         this.blockRenderer = $$0.getBlockRenderDispatcher();
     }
 
-    public void renderMinecartContents(net.minecraft.world.entity.vehicle.MinecartTNT entity, float partialTicks, net.minecraft.world.level.block.state.BlockState state, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        int n = entity.getFuse();
-        if (n > -1 && (float)n - partialTicks + 1.0f < 10.0f) {
-            float f = 1.0f - ((float)n - partialTicks + 1.0f) / 10.0f;
-            f = net.minecraft.util.Mth.clamp((float)f, (float)0.0f, (float)1.0f);
-            f *= f;
-            f *= f;
-            float f2 = 1.0f + f * 0.3f;
-            matrixStack.scale(f2, f2, f2);
+    public void renderMinecartContents(net.minecraft.world.entity.vehicle.MinecartTNT $$0, float $$1, net.minecraft.world.level.block.state.BlockState $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        int $$6 = $$0.getFuse();
+        if ($$6 > -1 && (float)$$6 - $$1 + 1.0f < 10.0f) {
+            float $$7 = 1.0f - ((float)$$6 - $$1 + 1.0f) / 10.0f;
+            $$7 = net.minecraft.util.Mth.clamp((float)$$7, (float)0.0f, (float)1.0f);
+            $$7 *= $$7;
+            $$7 *= $$7;
+            float $$8 = 1.0f + $$7 * 0.3f;
+            $$3.scale($$8, $$8, $$8);
         }
-        net.minecraft.client.renderer.entity.TntMinecartRenderer.renderWhiteSolidBlock((net.minecraft.client.renderer.block.BlockRenderDispatcher)this.blockRenderer, (net.minecraft.world.level.block.state.BlockState)state, (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight, (n > -1 && n / 5 % 2 == 0 ? 1 : 0) != 0);
+        net.minecraft.client.renderer.entity.TntMinecartRenderer.renderWhiteSolidBlock((net.minecraft.client.renderer.block.BlockRenderDispatcher)this.blockRenderer, (net.minecraft.world.level.block.state.BlockState)$$2, (com.mojang.blaze3d.vertex.PoseStack)$$3, (net.minecraft.client.renderer.MultiBufferSource)$$4, (int)$$5, ($$6 > -1 && $$6 / 5 % 2 == 0 ? 1 : 0) != 0);
     }
 
     public static void renderWhiteSolidBlock(net.minecraft.client.renderer.block.BlockRenderDispatcher $$0, net.minecraft.world.level.block.state.BlockState $$1, com.mojang.blaze3d.vertex.PoseStack $$2, net.minecraft.client.renderer.MultiBufferSource $$3, int $$4, boolean $$5) {