--- net/minecraft/client/renderer/entity/TntRenderer.java
+++ net/minecraft/client/renderer/entity/TntRenderer.java
@@ -14,27 +14,27 @@
         this.blockRenderer = $$0.getBlockRenderDispatcher();
     }
 
-    public void render(net.minecraft.world.entity.item.PrimedTnt entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.5, 0.0);
-        int n = entity.getFuse();
-        if ((float)n - partialTicks + 1.0f < 10.0f) {
-            float f = 1.0f - ((float)n - partialTicks + 1.0f) / 10.0f;
-            f = net.minecraft.util.Mth.clamp((float)f, (float)0.0f, (float)1.0f);
-            f *= f;
-            f *= f;
-            float f2 = 1.0f + f * 0.3f;
-            matrixStack.scale(f2, f2, f2);
+    public void render(net.minecraft.world.entity.item.PrimedTnt $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        $$3.translate(0.0, 0.5, 0.0);
+        int $$6 = $$0.getFuse();
+        if ((float)$$6 - $$2 + 1.0f < 10.0f) {
+            float $$7 = 1.0f - ((float)$$6 - $$2 + 1.0f) / 10.0f;
+            $$7 = net.minecraft.util.Mth.clamp((float)$$7, (float)0.0f, (float)1.0f);
+            $$7 *= $$7;
+            $$7 *= $$7;
+            float $$8 = 1.0f + $$7 * 0.3f;
+            $$3.scale($$8, $$8, $$8);
         }
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-90.0f));
-        matrixStack.translate(-0.5, -0.5, 0.5);
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f));
-        net.minecraft.client.renderer.entity.TntMinecartRenderer.renderWhiteSolidBlock((net.minecraft.client.renderer.block.BlockRenderDispatcher)this.blockRenderer, (net.minecraft.world.level.block.state.BlockState)net.minecraft.world.level.block.Blocks.TNT.defaultBlockState(), (com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.renderer.MultiBufferSource)buffer, (int)packedLight, (n / 5 % 2 == 0 ? 1 : 0) != 0);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(-90.0f));
+        $$3.translate(-0.5, -0.5, 0.5);
+        $$3.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(90.0f));
+        net.minecraft.client.renderer.entity.TntMinecartRenderer.renderWhiteSolidBlock((net.minecraft.client.renderer.block.BlockRenderDispatcher)this.blockRenderer, (net.minecraft.world.level.block.state.BlockState)net.minecraft.world.level.block.Blocks.TNT.defaultBlockState(), (com.mojang.blaze3d.vertex.PoseStack)$$3, (net.minecraft.client.renderer.MultiBufferSource)$$4, (int)$$5, ($$6 / 5 % 2 == 0 ? 1 : 0) != 0);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.item.PrimedTnt entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.item.PrimedTnt $$0) {
         return net.minecraft.client.renderer.texture.TextureAtlas.LOCATION_BLOCKS;
     }
 }