--- net/minecraft/client/renderer/entity/TropicalFishRenderer.java
+++ net/minecraft/client/renderer/entity/TropicalFishRenderer.java
@@ -16,26 +16,26 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.TropicalFishPatternLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.TropicalFish entity) {
-        return entity.getBaseTextureLocation();
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.TropicalFish $$0) {
+        return $$0.getBaseTextureLocation();
     }
 
-    public void render(net.minecraft.world.entity.animal.TropicalFish entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        net.minecraft.client.model.ColorableHierarchicalModel colorableHierarchicalModel = entity.getBaseVariant() == 0 ? this.modelA : this.modelB;
-        this.model = colorableHierarchicalModel;
-        float[] fArray = entity.getBaseColor();
-        colorableHierarchicalModel.setColor(fArray[0], fArray[1], fArray[2]);
-        super.render((net.minecraft.world.entity.Mob)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
-        colorableHierarchicalModel.setColor(1.0f, 1.0f, 1.0f);
+    public void render(net.minecraft.world.entity.animal.TropicalFish $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        net.minecraft.client.model.ColorableHierarchicalModel $$6 = $$0.getBaseVariant() == 0 ? this.modelA : this.modelB;
+        this.model = $$6;
+        float[] $$7 = $$0.getBaseColor();
+        $$6.setColor($$7[0], $$7[1], $$7[2]);
+        super.render((net.minecraft.world.entity.Mob)$$0, $$1, $$2, $$3, $$4, $$5);
+        $$6.setColor(1.0f, 1.0f, 1.0f);
     }
 
-    public void setupRotations(net.minecraft.world.entity.animal.TropicalFish entityLiving, com.mojang.blaze3d.vertex.PoseStack matrixStack, float ageInTicks, float rotationYaw, float partialTicks) {
-        super.setupRotations((net.minecraft.world.entity.LivingEntity)entityLiving, matrixStack, ageInTicks, rotationYaw, partialTicks);
-        float f = 4.3f * net.minecraft.util.Mth.sin((float)(0.6f * ageInTicks));
-        matrixStack.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees(f));
-        if (!entityLiving.isInWater()) {
-            matrixStack.translate((double)0.2f, (double)0.1f, 0.0);
-            matrixStack.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
+    public void setupRotations(net.minecraft.world.entity.animal.TropicalFish $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2, float $$3, float $$4) {
+        super.setupRotations((net.minecraft.world.entity.LivingEntity)$$0, $$1, $$2, $$3, $$4);
+        float $$5 = 4.3f * net.minecraft.util.Mth.sin((float)(0.6f * $$2));
+        $$1.mulPose(com.mojang.math.Vector3f.YP.rotationDegrees($$5));
+        if (!$$0.isInWater()) {
+            $$1.translate((double)0.2f, (double)0.1f, 0.0);
+            $$1.mulPose(com.mojang.math.Vector3f.ZP.rotationDegrees(90.0f));
         }
     }
 }