--- net/minecraft/client/renderer/entity/TurtleRenderer.java
+++ net/minecraft/client/renderer/entity/TurtleRenderer.java
@@ -12,14 +12,14 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.TurtleModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.TURTLE)), 0.7f);
     }
 
-    public void render(net.minecraft.world.entity.animal.Turtle entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        if (entity.isBaby()) {
+    public void render(net.minecraft.world.entity.animal.Turtle $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        if ($$0.isBaby()) {
             this.shadowRadius *= 0.5f;
         }
-        super.render((net.minecraft.world.entity.Mob)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+        super.render((net.minecraft.world.entity.Mob)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Turtle entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.Turtle $$0) {
         return TURTLE_LOCATION;
     }
 