--- net/minecraft/client/renderer/entity/UndeadHorseRenderer.java
+++ net/minecraft/client/renderer/entity/UndeadHorseRenderer.java
@@ -12,8 +12,8 @@
         super($$0, new net.minecraft.client.model.HorseModel($$0.bakeLayer($$1)), 1.0f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.horse.AbstractHorse entity) {
-        return (net.minecraft.resources.ResourceLocation)MAP.get((java.lang.Object)entity.getType());
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.animal.horse.AbstractHorse $$0) {
+        return (net.minecraft.resources.ResourceLocation)MAP.get((java.lang.Object)$$0.getType());
     }
 
     static {