--- net/minecraft/client/renderer/entity/VexRenderer.java
+++ net/minecraft/client/renderer/entity/VexRenderer.java
@@ -13,19 +13,19 @@
         super($$0, (net.minecraft.client.model.HumanoidModel)new net.minecraft.client.model.VexModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.VEX)), 0.3f);
     }
 
-    public int getBlockLightLevel(net.minecraft.world.entity.monster.Vex entity, net.minecraft.core.BlockPos pos) {
+    public int getBlockLightLevel(net.minecraft.world.entity.monster.Vex $$0, net.minecraft.core.BlockPos $$1) {
         return 15;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Vex entity) {
-        if (entity.isCharging()) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Vex $$0) {
+        if ($$0.isCharging()) {
             return VEX_CHARGING_LOCATION;
         }
         return VEX_LOCATION;
     }
 
-    public void scale(net.minecraft.world.entity.monster.Vex livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        matrixStack.scale(0.4f, 0.4f, 0.4f);
+    public void scale(net.minecraft.world.entity.monster.Vex $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        $$1.scale(0.4f, 0.4f, 0.4f);
     }
 
     static {