--- net/minecraft/client/renderer/entity/VillagerRenderer.java
+++ net/minecraft/client/renderer/entity/VillagerRenderer.java
@@ -15,19 +15,19 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.CrossedArmsItemLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getItemInHandRenderer()));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.npc.Villager entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.npc.Villager $$0) {
         return VILLAGER_BASE_SKIN;
     }
 
-    public void scale(net.minecraft.world.entity.npc.Villager livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        float f = 0.9375f;
-        if (livingEntity.isBaby()) {
-            f *= 0.5f;
+    public void scale(net.minecraft.world.entity.npc.Villager $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        float $$3 = 0.9375f;
+        if ($$0.isBaby()) {
+            $$3 *= 0.5f;
             this.shadowRadius = 0.25f;
         } else {
             this.shadowRadius = 0.5f;
         }
-        matrixStack.scale(f, f, f);
+        $$1.scale($$3, $$3, $$3);
     }
 
     static {