--- net/minecraft/client/renderer/entity/VindicatorRenderer.java
+++ net/minecraft/client/renderer/entity/VindicatorRenderer.java
@@ -13,7 +13,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new /* Unavailable Anonymous Inner Class!! */);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Vindicator entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Vindicator $$0) {
         return VINDICATOR;
     }
 