--- net/minecraft/client/renderer/entity/WitherSkullRenderer.java
+++ net/minecraft/client/renderer/entity/WitherSkullRenderer.java
@@ -22,24 +22,24 @@
         return net.minecraft.client.model.geom.builders.LayerDefinition.create((net.minecraft.client.model.geom.builders.MeshDefinition)$$0, (int)64, (int)64);
     }
 
-    public int getBlockLightLevel(net.minecraft.world.entity.projectile.WitherSkull entity, net.minecraft.core.BlockPos pos) {
+    public int getBlockLightLevel(net.minecraft.world.entity.projectile.WitherSkull $$0, net.minecraft.core.BlockPos $$1) {
         return 15;
     }
 
-    public void render(net.minecraft.world.entity.projectile.WitherSkull entity, float entityYaw, float partialTicks, com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, int packedLight) {
-        matrixStack.pushPose();
-        matrixStack.scale(-1.0f, -1.0f, 1.0f);
-        float f = net.minecraft.util.Mth.rotlerp((float)entity.yRotO, (float)entity.getYRot(), (float)partialTicks);
-        float f2 = net.minecraft.util.Mth.lerp((float)partialTicks, (float)entity.xRotO, (float)entity.getXRot());
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(this.model.renderType(this.getTextureLocation(entity)));
-        this.model.setupAnim(0.0f, f, f2);
-        this.model.renderToBuffer(matrixStack, vertexConsumer, packedLight, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
-        matrixStack.popPose();
-        super.render((net.minecraft.world.entity.Entity)entity, entityYaw, partialTicks, matrixStack, buffer, packedLight);
+    public void render(net.minecraft.world.entity.projectile.WitherSkull $$0, float $$1, float $$2, com.mojang.blaze3d.vertex.PoseStack $$3, net.minecraft.client.renderer.MultiBufferSource $$4, int $$5) {
+        $$3.pushPose();
+        $$3.scale(-1.0f, -1.0f, 1.0f);
+        float $$6 = net.minecraft.util.Mth.rotlerp((float)$$0.yRotO, (float)$$0.getYRot(), (float)$$2);
+        float $$7 = net.minecraft.util.Mth.lerp((float)$$2, (float)$$0.xRotO, (float)$$0.getXRot());
+        com.mojang.blaze3d.vertex.VertexConsumer $$8 = $$4.getBuffer(this.model.renderType(this.getTextureLocation($$0)));
+        this.model.setupAnim(0.0f, $$6, $$7);
+        this.model.renderToBuffer($$3, $$8, $$5, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, 1.0f);
+        $$3.popPose();
+        super.render((net.minecraft.world.entity.Entity)$$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.WitherSkull entity) {
-        return entity.isDangerous() ? WITHER_INVULNERABLE_LOCATION : WITHER_LOCATION;
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.projectile.WitherSkull $$0) {
+        return $$0.isDangerous() ? WITHER_INVULNERABLE_LOCATION : WITHER_LOCATION;
     }
 
     static {