--- net/minecraft/client/renderer/entity/WitherBossRenderer.java
+++ net/minecraft/client/renderer/entity/WitherBossRenderer.java
@@ -14,25 +14,25 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.WitherArmorLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getModelSet()));
     }
 
-    public int getBlockLightLevel(net.minecraft.world.entity.boss.wither.WitherBoss entity, net.minecraft.core.BlockPos pos) {
+    public int getBlockLightLevel(net.minecraft.world.entity.boss.wither.WitherBoss $$0, net.minecraft.core.BlockPos $$1) {
         return 15;
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.boss.wither.WitherBoss entity) {
-        int n = entity.getInvulnerableTicks();
-        if (n <= 0 || n <= 80 && n / 5 % 2 == 1) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.boss.wither.WitherBoss $$0) {
+        int $$1 = $$0.getInvulnerableTicks();
+        if ($$1 <= 0 || $$1 <= 80 && $$1 / 5 % 2 == 1) {
             return WITHER_LOCATION;
         }
         return WITHER_INVULNERABLE_LOCATION;
     }
 
-    public void scale(net.minecraft.world.entity.boss.wither.WitherBoss livingEntity, com.mojang.blaze3d.vertex.PoseStack matrixStack, float partialTickTime) {
-        float f = 2.0f;
-        int n = livingEntity.getInvulnerableTicks();
-        if (n > 0) {
-            f -= ((float)n - partialTickTime) / 220.0f * 0.5f;
+    public void scale(net.minecraft.world.entity.boss.wither.WitherBoss $$0, com.mojang.blaze3d.vertex.PoseStack $$1, float $$2) {
+        float $$3 = 2.0f;
+        int $$4 = $$0.getInvulnerableTicks();
+        if ($$4 > 0) {
+            $$3 -= ((float)$$4 - $$2) / 220.0f * 0.5f;
         }
-        matrixStack.scale(f, f, f);
+        $$1.scale($$3, $$3, $$3);
     }
 
     static {