--- net/minecraft/client/renderer/entity/ZoglinRenderer.java
+++ net/minecraft/client/renderer/entity/ZoglinRenderer.java
@@ -12,7 +12,7 @@
         super($$0, (net.minecraft.client.model.EntityModel)new net.minecraft.client.model.HoglinModel($$0.bakeLayer(net.minecraft.client.model.geom.ModelLayers.ZOGLIN)), 0.7f);
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zoglin entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.Zoglin $$0) {
         return ZOGLIN_LOCATION;
     }
 