--- net/minecraft/client/renderer/entity/ZombieVillagerRenderer.java
+++ net/minecraft/client/renderer/entity/ZombieVillagerRenderer.java
@@ -14,7 +14,7 @@
         this.addLayer((net.minecraft.client.renderer.entity.layers.RenderLayer)new net.minecraft.client.renderer.entity.layers.VillagerProfessionLayer((net.minecraft.client.renderer.entity.RenderLayerParent)this, $$0.getResourceManager(), "zombie_villager"));
     }
 
-    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.ZombieVillager entity) {
+    public net.minecraft.resources.ResourceLocation getTextureLocation(net.minecraft.world.entity.monster.ZombieVillager $$0) {
         return ZOMBIE_VILLAGER_LOCATION;
     }
 