--- net/minecraft/client/gui/components/AbstractOptionSliderButton.java
+++ net/minecraft/client/gui/components/AbstractOptionSliderButton.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.client.Options options;
 
-    public net.minecraft.client.gui.components.AbstractOptionSliderButton(net.minecraft.client.Options options, int x, int y, int width, int height, double value) {
-        super(x, y, width, height, net.minecraft.network.chat.CommonComponents.EMPTY, value);
-        this.options = options;
+    public net.minecraft.client.gui.components.AbstractOptionSliderButton(net.minecraft.client.Options $$0, int $$1, int $$2, int $$3, int $$4, double $$5) {
+        super($$1, $$2, $$3, $$4, net.minecraft.network.chat.CommonComponents.EMPTY, $$5);
+        this.options = $$0;
     }
 }
 