--- net/minecraft/client/gui/components/BossHealthOverlay.java
+++ net/minecraft/client/gui/components/BossHealthOverlay.java
@@ -13,39 +13,39 @@
     public net.minecraft.client.Minecraft minecraft;
     public java.util.Map<java.util.UUID, net.minecraft.client.gui.components.LerpingBossEvent> events;
 
-    public net.minecraft.client.gui.components.BossHealthOverlay(net.minecraft.client.Minecraft minecraft) {
+    public net.minecraft.client.gui.components.BossHealthOverlay(net.minecraft.client.Minecraft $$0) {
         this.events = com.google.common.collect.Maps.newLinkedHashMap();
-        this.minecraft = minecraft;
+        this.minecraft = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack poseStack) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0) {
         if (this.events.isEmpty()) {
             return;
         }
-        int n = this.minecraft.getWindow().getGuiScaledWidth();
-        int n2 = 12;
-        for (net.minecraft.client.gui.components.LerpingBossEvent lerpingBossEvent : this.events.values()) {
-            int n3 = n / 2 - 91;
-            int n4 = n2;
+        int $$1 = this.minecraft.getWindow().getGuiScaledWidth();
+        int $$2 = 12;
+        for (net.minecraft.client.gui.components.LerpingBossEvent $$3 : this.events.values()) {
+            int $$4 = $$1 / 2 - 91;
+            int $$5 = $$2;
             com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)GUI_BARS_LOCATION);
-            this.drawBar(poseStack, n3, n4, (net.minecraft.world.BossEvent)lerpingBossEvent);
-            net.minecraft.network.chat.Component component = lerpingBossEvent.getName();
-            int n5 = this.minecraft.font.width((net.minecraft.network.chat.FormattedText)component);
-            int n6 = n / 2 - n5 / 2;
-            int n7 = n4 - 9;
-            this.minecraft.font.drawShadow(poseStack, component, (float)n6, (float)n7, 0xFFFFFF);
+            this.drawBar($$0, $$4, $$5, (net.minecraft.world.BossEvent)$$3);
+            net.minecraft.network.chat.Component $$6 = $$3.getName();
+            int $$7 = this.minecraft.font.width((net.minecraft.network.chat.FormattedText)$$6);
+            int $$8 = $$1 / 2 - $$7 / 2;
+            int $$9 = $$5 - 9;
+            this.minecraft.font.drawShadow($$0, $$6, (float)$$8, (float)$$9, 0xFFFFFF);
             java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-            if ((n2 += 10 + 9) < this.minecraft.getWindow().getGuiScaledHeight() / 3) continue;
+            if (($$2 += 10 + 9) < this.minecraft.getWindow().getGuiScaledHeight() / 3) continue;
             break;
         }
     }
 
-    public void drawBar(com.mojang.blaze3d.vertex.PoseStack poseStack, int x, int y, net.minecraft.world.BossEvent bossEvent) {
-        this.drawBar(poseStack, x, y, bossEvent, 182, 0);
-        int n = (int)(bossEvent.getProgress() * 183.0f);
-        if (n > 0) {
-            this.drawBar(poseStack, x, y, bossEvent, n, 5);
+    public void drawBar(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, net.minecraft.world.BossEvent $$3) {
+        this.drawBar($$0, $$1, $$2, $$3, 182, 0);
+        int $$4 = (int)($$3.getProgress() * 183.0f);
+        if ($$4 > 0) {
+            this.drawBar($$0, $$1, $$2, $$3, $$4, 5);
         }
     }
 
@@ -59,8 +59,8 @@
         }
     }
 
-    public void update(net.minecraft.network.protocol.game.ClientboundBossEventPacket packet) {
-        packet.dispatch((net.minecraft.network.protocol.game.ClientboundBossEventPacket$Handler)new /* Unavailable Anonymous Inner Class!! */);
+    public void update(net.minecraft.network.protocol.game.ClientboundBossEventPacket $$0) {
+        $$0.dispatch((net.minecraft.network.protocol.game.ClientboundBossEventPacket$Handler)new /* Unavailable Anonymous Inner Class!! */);
     }
 
     public void reset() {