--- net/minecraft/client/gui/components/Button.java
+++ net/minecraft/client/gui/components/Button.java
@@ -13,29 +13,29 @@
     public net.minecraft.client.gui.components.Button$OnPress onPress;
     public net.minecraft.client.gui.components.Button$OnTooltip onTooltip;
 
-    public net.minecraft.client.gui.components.Button(int x, int y, int width, int height, net.minecraft.network.chat.Component message, net.minecraft.client.gui.components.Button$OnPress onPress) {
-        this(x, y, width, height, message, onPress, NO_TOOLTIP);
+    public net.minecraft.client.gui.components.Button(int $$0, int $$1, int $$2, int $$3, net.minecraft.network.chat.Component $$4, net.minecraft.client.gui.components.Button$OnPress $$5) {
+        this($$0, $$1, $$2, $$3, $$4, $$5, NO_TOOLTIP);
     }
 
-    public net.minecraft.client.gui.components.Button(int x, int y, int width, int height, net.minecraft.network.chat.Component message, net.minecraft.client.gui.components.Button$OnPress onPress, net.minecraft.client.gui.components.Button$OnTooltip onTooltip) {
-        super(x, y, width, height, message);
-        this.onPress = onPress;
-        this.onTooltip = onTooltip;
+    public net.minecraft.client.gui.components.Button(int $$0, int $$1, int $$2, int $$3, net.minecraft.network.chat.Component $$4, net.minecraft.client.gui.components.Button$OnPress $$5, net.minecraft.client.gui.components.Button$OnTooltip $$6) {
+        super($$0, $$1, $$2, $$3, $$4);
+        this.onPress = $$5;
+        this.onTooltip = $$6;
     }
 
     public void onPress() {
         this.onPress.onPress(this);
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        super.renderButton(matrixStack, mouseX, mouseY, partialTicks);
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        super.renderButton($$0, $$1, $$2, $$3);
         if (this.isHoveredOrFocused()) {
-            this.renderToolTip(matrixStack, mouseX, mouseY);
+            this.renderToolTip($$0, $$1, $$2);
         }
     }
 
-    public void renderToolTip(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY) {
-        this.onTooltip.onTooltip(this, matrixStack, mouseX, mouseY);
+    public void renderToolTip(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
+        this.onTooltip.onTooltip(this, $$0, $$1, $$2);
     }
 
     public void updateNarration(net.minecraft.client.gui.narration.NarrationElementOutput $$0) {
@@ -43,11 +43,11 @@
         this.onTooltip.narrateTooltip($$1 -> $$0.add(net.minecraft.client.gui.narration.NarratedElementType.HINT, $$1));
     }
 
-    public static /* synthetic */ void lambda$static$0(net.minecraft.client.gui.components.Button button, com.mojang.blaze3d.vertex.PoseStack poseStack, int mouseX, int mouseY) {
+    public static /* synthetic */ void lambda$static$0(net.minecraft.client.gui.components.Button $$0, com.mojang.blaze3d.vertex.PoseStack $$1, int $$2, int $$3) {
     }
 
     static {
-        NO_TOOLTIP = (button, poseStack, mouseX, mouseY) -> {}
+        NO_TOOLTIP = ($$0, $$1, $$2, $$3) -> {}
 ;
     }
 }