--- net/minecraft/client/gui/components/Checkbox.java
+++ net/minecraft/client/gui/components/Checkbox.java
@@ -11,14 +11,14 @@
     public boolean selected;
     public boolean showLabel;
 
-    public net.minecraft.client.gui.components.Checkbox(int x, int y, int width, int height, net.minecraft.network.chat.Component message, boolean selected) {
-        this(x, y, width, height, message, selected, true);
+    public net.minecraft.client.gui.components.Checkbox(int $$0, int $$1, int $$2, int $$3, net.minecraft.network.chat.Component $$4, boolean $$5) {
+        this($$0, $$1, $$2, $$3, $$4, $$5, true);
     }
 
-    public net.minecraft.client.gui.components.Checkbox(int x, int y, int width, int height, net.minecraft.network.chat.Component message, boolean selected, boolean showLabel) {
-        super(x, y, width, height, message);
-        this.selected = selected;
-        this.showLabel = showLabel;
+    public net.minecraft.client.gui.components.Checkbox(int $$0, int $$1, int $$2, int $$3, net.minecraft.network.chat.Component $$4, boolean $$5, boolean $$6) {
+        super($$0, $$1, $$2, $$3, $$4);
+        this.selected = $$5;
+        this.showLabel = $$6;
     }
 
     public void onPress() {
@@ -40,19 +40,19 @@
         }
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        net.minecraft.client.Minecraft minecraft = net.minecraft.client.Minecraft.getInstance();
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        net.minecraft.client.Minecraft $$4 = net.minecraft.client.Minecraft.getInstance();
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)TEXTURE);
         com.mojang.blaze3d.systems.RenderSystem.enableDepthTest();
-        net.minecraft.client.gui.Font font = minecraft.font;
+        net.minecraft.client.gui.Font $$5 = $$4.font;
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)this.alpha);
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
         com.mojang.blaze3d.systems.RenderSystem.blendFunc((com.mojang.blaze3d.platform.GlStateManager$SourceFactor)com.mojang.blaze3d.platform.GlStateManager$SourceFactor.SRC_ALPHA, (com.mojang.blaze3d.platform.GlStateManager$DestFactor)com.mojang.blaze3d.platform.GlStateManager$DestFactor.ONE_MINUS_SRC_ALPHA);
-        net.minecraft.client.gui.components.Checkbox.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)this.x, (int)this.y, (float)(this.isFocused() ? 20.0f : 0.0f), (float)(this.selected ? 20.0f : 0.0f), (int)20, (int)this.height, (int)64, (int)64);
-        this.renderBg(matrixStack, minecraft, mouseX, mouseY);
+        net.minecraft.client.gui.components.Checkbox.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)this.x, (int)this.y, (float)(this.isFocused() ? 20.0f : 0.0f), (float)(this.selected ? 20.0f : 0.0f), (int)20, (int)this.height, (int)64, (int)64);
+        this.renderBg($$0, $$4, $$1, $$2);
         if (this.showLabel) {
-            net.minecraft.client.gui.components.Checkbox.drawString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)font, (net.minecraft.network.chat.Component)this.getMessage(), (int)(this.x + 24), (int)(this.y + (this.height - 8) / 2), (int)(0xE0E0E0 | net.minecraft.util.Mth.ceil((float)(this.alpha * 255.0f)) << 24));
+            net.minecraft.client.gui.components.Checkbox.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)$$5, (net.minecraft.network.chat.Component)this.getMessage(), (int)(this.x + 24), (int)(this.y + (this.height - 8) / 2), (int)(0xE0E0E0 | net.minecraft.util.Mth.ceil((float)(this.alpha * 255.0f)) << 24));
         }
     }
 