--- net/minecraft/client/gui/components/CommandSuggestions$SuggestionsList.java
+++ net/minecraft/client/gui/components/CommandSuggestions$SuggestionsList.java
@@ -27,122 +27,119 @@
         this.select(0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack poseStack, int mouseX, int mouseY) {
-        com.mojang.brigadier.Message message;
-        int n;
-        boolean bl;
-        int n2 = java.lang.Math.min((int)this.suggestionList.size(), (int)this.this$0.suggestionLineLimit);
-        int n3 = -5592406;
-        boolean bl2 = this.offset > 0;
-        boolean bl3 = this.suggestionList.size() > this.offset + n2;
-        boolean bl4 = bl2 || bl3;
-        boolean bl5 = bl = this.lastMouse.x != (float)mouseX || this.lastMouse.y != (float)mouseY;
-        if (bl) {
-            this.lastMouse = new net.minecraft.world.phys.Vec2((float)mouseX, (float)mouseY);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2) {
+        com.mojang.brigadier.Message $$14;
+        boolean $$8;
+        int $$3 = java.lang.Math.min((int)this.suggestionList.size(), (int)this.this$0.suggestionLineLimit);
+        int $$4 = -5592406;
+        boolean $$5 = this.offset > 0;
+        boolean $$6 = this.suggestionList.size() > this.offset + $$3;
+        boolean $$7 = $$5 || $$6;
+        boolean bl = $$8 = this.lastMouse.x != (float)$$1 || this.lastMouse.y != (float)$$2;
+        if ($$8) {
+            this.lastMouse = new net.minecraft.world.phys.Vec2((float)$$1, (float)$$2);
         }
-        if (bl4) {
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)this.rect.getX(), (int)(this.rect.getY() - 1), (int)(this.rect.getX() + this.rect.getWidth()), (int)this.rect.getY(), (int)this.this$0.fillColor);
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)this.rect.getX(), (int)(this.rect.getY() + this.rect.getHeight()), (int)(this.rect.getX() + this.rect.getWidth()), (int)(this.rect.getY() + this.rect.getHeight() + 1), (int)this.this$0.fillColor);
-            if (bl2) {
-                for (n = 0;
- n < this.rect.getWidth(); ++n) {
-                    if (n % 2 != 0) continue;
-                    net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(this.rect.getX() + n), (int)(this.rect.getY() - 1), (int)(this.rect.getX() + n + 1), (int)this.rect.getY(), (int)-1);
+        if ($$7) {
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)this.rect.getX(), (int)(this.rect.getY() - 1), (int)(this.rect.getX() + this.rect.getWidth()), (int)this.rect.getY(), (int)this.this$0.fillColor);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)this.rect.getX(), (int)(this.rect.getY() + this.rect.getHeight()), (int)(this.rect.getX() + this.rect.getWidth()), (int)(this.rect.getY() + this.rect.getHeight() + 1), (int)this.this$0.fillColor);
+            if ($$5) {
+                for (int $$9 = 0; $$9 < this.rect.getWidth(); ++$$9) {
+                    if ($$9 % 2 != 0) continue;
+                    net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)(this.rect.getX() + $$9), (int)(this.rect.getY() - 1), (int)(this.rect.getX() + $$9 + 1), (int)this.rect.getY(), (int)-1);
                 }
             }
-            if (bl3) {
-                for (n = 0;
- n < this.rect.getWidth(); ++n) {
-                    if (n % 2 != 0) continue;
-                    net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(this.rect.getX() + n), (int)(this.rect.getY() + this.rect.getHeight()), (int)(this.rect.getX() + n + 1), (int)(this.rect.getY() + this.rect.getHeight() + 1), (int)-1);
+            if ($$6) {
+                for (int $$10 = 0; $$10 < this.rect.getWidth(); ++$$10) {
+                    if ($$10 % 2 != 0) continue;
+                    net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)(this.rect.getX() + $$10), (int)(this.rect.getY() + this.rect.getHeight()), (int)(this.rect.getX() + $$10 + 1), (int)(this.rect.getY() + this.rect.getHeight() + 1), (int)-1);
                 }
             }
         }
-        n = 0;
-        for (int i = 0; i < n2; ++i) {
-            com.mojang.brigadier.suggestion.Suggestion suggestion = (com.mojang.brigadier.suggestion.Suggestion)this.suggestionList.get(i + this.offset);
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)this.rect.getX(), (int)(this.rect.getY() + 12 * i), (int)(this.rect.getX() + this.rect.getWidth()), (int)(this.rect.getY() + 12 * i + 12), (int)this.this$0.fillColor);
-            if (mouseX > this.rect.getX() && mouseX < this.rect.getX() + this.rect.getWidth() && mouseY > this.rect.getY() + 12 * i && mouseY < this.rect.getY() + 12 * i + 12) {
-                if (bl) {
-                    this.select(i + this.offset);
+        boolean $$11 = false;
+        for (int $$12 = 0; $$12 < $$3; ++$$12) {
+            com.mojang.brigadier.suggestion.Suggestion $$13 = (com.mojang.brigadier.suggestion.Suggestion)this.suggestionList.get($$12 + this.offset);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)this.rect.getX(), (int)(this.rect.getY() + 12 * $$12), (int)(this.rect.getX() + this.rect.getWidth()), (int)(this.rect.getY() + 12 * $$12 + 12), (int)this.this$0.fillColor);
+            if ($$1 > this.rect.getX() && $$1 < this.rect.getX() + this.rect.getWidth() && $$2 > this.rect.getY() + 12 * $$12 && $$2 < this.rect.getY() + 12 * $$12 + 12) {
+                if ($$8) {
+                    this.select($$12 + this.offset);
                 }
-                n = 1;
+                $$11 = true;
             }
-            this.this$0.font.drawShadow(poseStack, suggestion.getText(), (float)(this.rect.getX() + 1), (float)(this.rect.getY() + 2 + 12 * i), i + this.offset == this.current ? -256 : -5592406);
+            this.this$0.font.drawShadow($$0, $$13.getText(), (float)(this.rect.getX() + 1), (float)(this.rect.getY() + 2 + 12 * $$12), $$12 + this.offset == this.current ? -256 : -5592406);
         }
-        if (n != 0 && (message = ((com.mojang.brigadier.suggestion.Suggestion)this.suggestionList.get(this.current)).getTooltip()) != null) {
-            this.this$0.screen.renderTooltip(poseStack, net.minecraft.network.chat.ComponentUtils.fromMessage((com.mojang.brigadier.Message)message), mouseX, mouseY);
+        if ($$11 && ($$14 = ((com.mojang.brigadier.suggestion.Suggestion)this.suggestionList.get(this.current)).getTooltip()) != null) {
+            this.this$0.screen.renderTooltip($$0, net.minecraft.network.chat.ComponentUtils.fromMessage((com.mojang.brigadier.Message)$$14), $$1, $$2);
         }
     }
 
-    public boolean mouseClicked(int mouseX, int mouseY, int mouseButton) {
-        if (!this.rect.contains(mouseX, mouseY)) {
+    public boolean mouseClicked(int $$0, int $$1, int $$2) {
+        if (!this.rect.contains($$0, $$1)) {
             return false;
         }
-        int n = (mouseY - this.rect.getY()) / 12 + this.offset;
-        if (n >= 0 && n < this.suggestionList.size()) {
-            this.select(n);
+        int $$3 = ($$1 - this.rect.getY()) / 12 + this.offset;
+        if ($$3 >= 0 && $$3 < this.suggestionList.size()) {
+            this.select($$3);
             this.useSuggestion();
         }
         return true;
     }
 
-    public boolean mouseScrolled(double delta) {
-        int n;
-        int n2 = (int)(this.this$0.minecraft.mouseHandler.xpos() * (double)this.this$0.minecraft.getWindow().getGuiScaledWidth() / (double)this.this$0.minecraft.getWindow().getScreenWidth());
-        if (this.rect.contains(n2, n = (int)(this.this$0.minecraft.mouseHandler.ypos() * (double)this.this$0.minecraft.getWindow().getGuiScaledHeight() / (double)this.this$0.minecraft.getWindow().getScreenHeight()))) {
-            this.offset = net.minecraft.util.Mth.clamp((int)((int)((double)this.offset - delta)), (int)0, (int)java.lang.Math.max((int)(this.suggestionList.size() - this.this$0.suggestionLineLimit), (int)0));
+    public boolean mouseScrolled(double $$0) {
+        int $$2;
+        int $$1 = (int)(this.this$0.minecraft.mouseHandler.xpos() * (double)this.this$0.minecraft.getWindow().getGuiScaledWidth() / (double)this.this$0.minecraft.getWindow().getScreenWidth());
+        if (this.rect.contains($$1, $$2 = (int)(this.this$0.minecraft.mouseHandler.ypos() * (double)this.this$0.minecraft.getWindow().getGuiScaledHeight() / (double)this.this$0.minecraft.getWindow().getScreenHeight()))) {
+            this.offset = net.minecraft.util.Mth.clamp((int)((int)((double)this.offset - $$0)), (int)0, (int)java.lang.Math.max((int)(this.suggestionList.size() - this.this$0.suggestionLineLimit), (int)0));
             return true;
         }
         return false;
     }
 
-    public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
-        if (keyCode == 265) {
+    public boolean keyPressed(int $$0, int $$1, int $$2) {
+        if ($$0 == 265) {
             this.cycle(-1);
             this.tabCycles = false;
             return true;
         }
-        if (keyCode == 264) {
+        if ($$0 == 264) {
             this.cycle(1);
             this.tabCycles = false;
             return true;
         }
-        if (keyCode == 258) {
+        if ($$0 == 258) {
             if (this.tabCycles) {
                 this.cycle(net.minecraft.client.gui.screens.Screen.hasShiftDown() ? -1 : 1);
             }
             this.useSuggestion();
             return true;
         }
-        if (keyCode == 256) {
+        if ($$0 == 256) {
             this.this$0.hide();
             return true;
         }
         return false;
     }
 
-    public void cycle(int change) {
-        this.select(this.current + change);
-        int n = this.offset;
-        int n2 = this.offset + this.this$0.suggestionLineLimit - 1;
-        if (this.current < n) {
+    public void cycle(int $$0) {
+        this.select(this.current + $$0);
+        int $$1 = this.offset;
+        int $$2 = this.offset + this.this$0.suggestionLineLimit - 1;
+        if (this.current < $$1) {
             this.offset = net.minecraft.util.Mth.clamp((int)this.current, (int)0, (int)java.lang.Math.max((int)(this.suggestionList.size() - this.this$0.suggestionLineLimit), (int)0));
-        } else if (this.current > n2) {
+        } else if (this.current > $$2) {
             this.offset = net.minecraft.util.Mth.clamp((int)(this.current + this.this$0.lineStartOffset - this.this$0.suggestionLineLimit), (int)0, (int)java.lang.Math.max((int)(this.suggestionList.size() - this.this$0.suggestionLineLimit), (int)0));
         }
     }
 
-    public void select(int index) {
-        this.current = index;
+    public void select(int $$0) {
+        this.current = $$0;
         if (this.current < 0) {
             this.current += this.suggestionList.size();
         }
         if (this.current >= this.suggestionList.size()) {
             this.current -= this.suggestionList.size();
         }
-        com.mojang.brigadier.suggestion.Suggestion suggestion = (com.mojang.brigadier.suggestion.Suggestion)this.suggestionList.get(this.current);
-        this.this$0.input.setSuggestion(net.minecraft.client.gui.components.CommandSuggestions.calculateSuggestionSuffix((java.lang.String)this.this$0.input.getValue(), (java.lang.String)suggestion.apply(this.originalContents)));
+        com.mojang.brigadier.suggestion.Suggestion $$1 = (com.mojang.brigadier.suggestion.Suggestion)this.suggestionList.get(this.current);
+        this.this$0.input.setSuggestion(net.minecraft.client.gui.components.CommandSuggestions.calculateSuggestionSuffix((java.lang.String)this.this$0.input.getValue(), (java.lang.String)$$1.apply(this.originalContents)));
         if (this.lastNarratedEntry != this.current) {
             this.this$0.minecraft.getNarrator().sayNow(this.getNarrationMessage());
         }