--- net/minecraft/client/gui/components/ComponentRenderUtils.java
+++ net/minecraft/client/gui/components/ComponentRenderUtils.java
@@ -10,8 +10,8 @@
     public net.minecraft.client.gui.components.ComponentRenderUtils() {
     }
 
-    public static java.lang.String stripColor(java.lang.String text) {
-        return (java.lang.Boolean)net.minecraft.client.Minecraft.getInstance().options.chatColors().get() != false ? text : net.minecraft.ChatFormatting.stripFormatting((java.lang.String)text);
+    public static java.lang.String stripColor(java.lang.String $$0) {
+        return (java.lang.Boolean)net.minecraft.client.Minecraft.getInstance().options.chatColors().get() != false ? $$0 : net.minecraft.ChatFormatting.stripFormatting((java.lang.String)$$0);
     }
 
     public static java.util.List<net.minecraft.util.FormattedCharSequence> wrapComponents(net.minecraft.network.chat.FormattedText $$0, int $$12, net.minecraft.client.gui.Font $$22) {