--- net/minecraft/client/gui/components/ContainerObjectSelectionList.java
+++ net/minecraft/client/gui/components/ContainerObjectSelectionList.java
@@ -12,8 +12,8 @@
         super($$0, $$1, $$2, $$3, $$4, $$5);
     }
 
-    public boolean changeFocus(boolean focus) {
-        this.hasFocus = super.changeFocus(focus);
+    public boolean changeFocus(boolean $$0) {
+        this.hasFocus = super.changeFocus($$0);
         if (this.hasFocus) {
             this.ensureVisible((net.minecraft.client.gui.components.AbstractSelectionList$Entry)((net.minecraft.client.gui.components.ContainerObjectSelectionList$Entry)this.getFocused()));
         }
@@ -27,7 +27,7 @@
         return super.narrationPriority();
     }
 
-    public boolean isSelectedItem(int index) {
+    public boolean isSelectedItem(int $$0) {
         return false;
     }
 