--- net/minecraft/client/gui/components/ImageButton.java
+++ net/minecraft/client/gui/components/ImageButton.java
@@ -17,46 +17,46 @@
         this($$0, $$1, $$2, $$3, $$4, $$5, $$3, $$6, 256, 256, $$7);
     }
 
-    public net.minecraft.client.gui.components.ImageButton(int x, int y, int width, int height, int xTexStart, int yTexStart, int yDiffTex, net.minecraft.resources.ResourceLocation resourceLocation, net.minecraft.client.gui.components.Button$OnPress onPress) {
-        this(x, y, width, height, xTexStart, yTexStart, yDiffTex, resourceLocation, 256, 256, onPress);
+    public net.minecraft.client.gui.components.ImageButton(int $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, net.minecraft.resources.ResourceLocation $$7, net.minecraft.client.gui.components.Button$OnPress $$8) {
+        this($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7, 256, 256, $$8);
     }
 
-    public net.minecraft.client.gui.components.ImageButton(int x, int y, int width, int height, int xTexStart, int yTexStart, int yDiffTex, net.minecraft.resources.ResourceLocation resourceLocation, int textureWidth, int textureHeight, net.minecraft.client.gui.components.Button$OnPress onPress) {
-        this(x, y, width, height, xTexStart, yTexStart, yDiffTex, resourceLocation, textureWidth, textureHeight, onPress, net.minecraft.network.chat.CommonComponents.EMPTY);
+    public net.minecraft.client.gui.components.ImageButton(int $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, net.minecraft.resources.ResourceLocation $$7, int $$8, int $$9, net.minecraft.client.gui.components.Button$OnPress $$10) {
+        this($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7, $$8, $$9, $$10, net.minecraft.network.chat.CommonComponents.EMPTY);
     }
 
-    public net.minecraft.client.gui.components.ImageButton(int x, int y, int width, int height, int xTexStart, int yTexStart, int yDiffTex, net.minecraft.resources.ResourceLocation resourceLocation, int textureWidth, int textureHeight, net.minecraft.client.gui.components.Button$OnPress onPress, net.minecraft.network.chat.Component message) {
-        this(x, y, width, height, xTexStart, yTexStart, yDiffTex, resourceLocation, textureWidth, textureHeight, onPress, NO_TOOLTIP, message);
+    public net.minecraft.client.gui.components.ImageButton(int $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, net.minecraft.resources.ResourceLocation $$7, int $$8, int $$9, net.minecraft.client.gui.components.Button$OnPress $$10, net.minecraft.network.chat.Component $$11) {
+        this($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7, $$8, $$9, $$10, NO_TOOLTIP, $$11);
     }
 
-    public net.minecraft.client.gui.components.ImageButton(int x, int y, int width, int height, int xTexStart, int yTexStart, int yDiffTex, net.minecraft.resources.ResourceLocation resourceLocation, int textureWidth, int textureHeight, net.minecraft.client.gui.components.Button$OnPress onPress, net.minecraft.client.gui.components.Button$OnTooltip onTooltip, net.minecraft.network.chat.Component message) {
-        super(x, y, width, height, message, onPress, onTooltip);
-        this.textureWidth = textureWidth;
-        this.textureHeight = textureHeight;
-        this.xTexStart = xTexStart;
-        this.yTexStart = yTexStart;
-        this.yDiffTex = yDiffTex;
-        this.resourceLocation = resourceLocation;
+    public net.minecraft.client.gui.components.ImageButton(int $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, net.minecraft.resources.ResourceLocation $$7, int $$8, int $$9, net.minecraft.client.gui.components.Button$OnPress $$10, net.minecraft.client.gui.components.Button$OnTooltip $$11, net.minecraft.network.chat.Component $$12) {
+        super($$0, $$1, $$2, $$3, $$12, $$10, $$11);
+        this.textureWidth = $$8;
+        this.textureHeight = $$9;
+        this.xTexStart = $$4;
+        this.yTexStart = $$5;
+        this.yDiffTex = $$6;
+        this.resourceLocation = $$7;
     }
 
-    public void setPosition(int x, int y) {
-        this.x = x;
-        this.y = y;
+    public void setPosition(int $$0, int $$1) {
+        this.x = $$0;
+        this.y = $$1;
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)this.resourceLocation);
-        int n = this.yTexStart;
+        int $$4 = this.yTexStart;
         if (!this.isActive()) {
-            n += this.yDiffTex * 2;
+            $$4 += this.yDiffTex * 2;
         } else if (this.isHoveredOrFocused()) {
-            n += this.yDiffTex;
+            $$4 += this.yDiffTex;
         }
         com.mojang.blaze3d.systems.RenderSystem.enableDepthTest();
-        net.minecraft.client.gui.components.ImageButton.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)this.x, (int)this.y, (float)this.xTexStart, (float)n, (int)this.width, (int)this.height, (int)this.textureWidth, (int)this.textureHeight);
+        net.minecraft.client.gui.components.ImageButton.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)this.x, (int)this.y, (float)this.xTexStart, (float)$$4, (int)this.width, (int)this.height, (int)this.textureWidth, (int)this.textureHeight);
         if (this.isHovered) {
-            this.renderToolTip(matrixStack, mouseX, mouseY);
+            this.renderToolTip($$0, $$1, $$2);
         }
     }
 }