--- net/minecraft/client/gui/components/LockIconButton$Icon.java
+++ net/minecraft/client/gui/components/LockIconButton$Icon.java
@@ -24,10 +24,10 @@
         return (net.minecraft.client.gui.components.LockIconButton$Icon)java.lang.Enum.valueOf(net.minecraft.client.gui.components.LockIconButton$Icon.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.client.gui.components.LockIconButton$Icon(int x, int y) {
+    public net.minecraft.client.gui.components.LockIconButton$Icon(int $$0, int $$1) {
         super(string, n);
-        this.x = x;
-        this.y = y;
+        this.x = $$0;
+        this.y = $$1;
     }
 
     public int getX() {