--- net/minecraft/client/gui/components/LockIconButton.java
+++ net/minecraft/client/gui/components/LockIconButton.java
@@ -8,8 +8,8 @@
  {
     public boolean locked;
 
-    public net.minecraft.client.gui.components.LockIconButton(int x, int y, net.minecraft.client.gui.components.Button$OnPress onPress) {
-        super(x, y, 20, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"narrator.button.difficulty_lock"), onPress);
+    public net.minecraft.client.gui.components.LockIconButton(int $$0, int $$1, net.minecraft.client.gui.components.Button$OnPress $$2) {
+        super($$0, $$1, 20, 20, (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"narrator.button.difficulty_lock"), $$2);
     }
 
     public net.minecraft.network.chat.MutableComponent createNarrationMessage() {
@@ -20,16 +20,23 @@
         return this.locked;
     }
 
-    public void setLocked(boolean locked) {
-        this.locked = locked;
+    public void setLocked(boolean $$0) {
+        this.locked = $$0;
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        net.minecraft.client.gui.components.LockIconButton$Icon $$6;
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)net.minecraft.client.gui.components.Button.WIDGETS_LOCATION);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-        net.minecraft.client.gui.components.LockIconButton$Icon icon = !this.active ? (this.locked ? net.minecraft.client.gui.components.LockIconButton$Icon.LOCKED_DISABLED : net.minecraft.client.gui.components.LockIconButton$Icon.UNLOCKED_DISABLED) : (this.isHoveredOrFocused() ? (this.locked ? net.minecraft.client.gui.components.LockIconButton$Icon.LOCKED_HOVER : net.minecraft.client.gui.components.LockIconButton$Icon.UNLOCKED_HOVER) : (this.locked ? net.minecraft.client.gui.components.LockIconButton$Icon.LOCKED : net.minecraft.client.gui.components.LockIconButton$Icon.UNLOCKED));
-        this.blit(matrixStack, this.x, this.y, icon.getX(), icon.getY(), this.width, this.height);
+        if (!this.active) {
+            net.minecraft.client.gui.components.LockIconButton$Icon $$4 = this.locked ? net.minecraft.client.gui.components.LockIconButton$Icon.LOCKED_DISABLED : net.minecraft.client.gui.components.LockIconButton$Icon.UNLOCKED_DISABLED;
+        } else if (this.isHoveredOrFocused()) {
+            net.minecraft.client.gui.components.LockIconButton$Icon $$5 = this.locked ? net.minecraft.client.gui.components.LockIconButton$Icon.LOCKED_HOVER : net.minecraft.client.gui.components.LockIconButton$Icon.UNLOCKED_HOVER;
+        } else {
+            $$6 = this.locked ? net.minecraft.client.gui.components.LockIconButton$Icon.LOCKED : net.minecraft.client.gui.components.LockIconButton$Icon.UNLOCKED;
+        }
+        this.blit($$0, this.x, this.y, $$6.getX(), $$6.getY(), this.width, this.height);
     }
 }
 