--- net/minecraft/client/gui/components/OptionsList$Entry.java
+++ net/minecraft/client/gui/components/OptionsList$Entry.java
@@ -26,10 +26,10 @@
         return new net.minecraft.client.gui.components.OptionsList$Entry((java.util.Map)com.google.common.collect.ImmutableMap.of($$2, (java.lang.Object)$$4, $$3, (java.lang.Object)$$3.createButton($$0, $$1 / 2 - 155 + 160, 0, 150)));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.children.forEach(widget -> {
-            widget.y = top;
-            widget.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$52, int $$6, int $$7, boolean $$8, float $$9) {
+        this.children.forEach($$5 -> {
+            $$5.y = $$2;
+            $$5.render($$0, $$6, $$7, $$9);
         });
     }
 