--- net/minecraft/client/gui/components/OptionsList.java
+++ net/minecraft/client/gui/components/OptionsList.java
@@ -43,11 +43,11 @@
         return null;
     }
 
-    public java.util.Optional<net.minecraft.client.gui.components.AbstractWidget> getMouseOver(double mouseX, double mouseY) {
-        for (net.minecraft.client.gui.components.OptionsList$Entry entry : this.children()) {
-            for (net.minecraft.client.gui.components.AbstractWidget abstractWidget : entry.children) {
-                if (!abstractWidget.isMouseOver(mouseX, mouseY)) continue;
-                return java.util.Optional.of((java.lang.Object)abstractWidget);
+    public java.util.Optional<net.minecraft.client.gui.components.AbstractWidget> getMouseOver(double $$0, double $$1) {
+        for (net.minecraft.client.gui.components.OptionsList$Entry $$2 : this.children()) {
+            for (net.minecraft.client.gui.components.AbstractWidget $$3 : $$2.children) {
+                if (!$$3.isMouseOver($$0, $$1)) continue;
+                return java.util.Optional.of((java.lang.Object)$$3);
             }
         }
         return java.util.Optional.empty();