--- net/minecraft/client/gui/components/PlayerTabOverlay$HealthState.java
+++ net/minecraft/client/gui/components/PlayerTabOverlay$HealthState.java
@@ -1,0 +1,41 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.gui.components;
+
+public class net.minecraft.client.gui.components.PlayerTabOverlay$HealthState
+ {
+    public static long DISPLAY_UPDATE_DELAY = 20L;
+    public static long DECREASE_BLINK_DURATION = 20L;
+    public static long INCREASE_BLINK_DURATION = 10L;
+    public int lastValue;
+    public int displayedValue;
+    public long lastUpdateTick;
+    public long blinkUntilTick;
+
+    public net.minecraft.client.gui.components.PlayerTabOverlay$HealthState(int $$0) {
+        this.displayedValue = $$0;
+        this.lastValue = $$0;
+    }
+
+    public void update(int $$0, long $$1) {
+        if ($$0 != this.lastValue) {
+            long $$2 = $$0 < this.lastValue ? 20L : 10L;
+            this.blinkUntilTick = $$1 + $$2;
+            this.lastValue = $$0;
+            this.lastUpdateTick = $$1;
+        }
+        if ($$1 - this.lastUpdateTick > 20L) {
+            this.displayedValue = $$0;
+        }
+    }
+
+    public int displayedValue() {
+        return this.displayedValue;
+    }
+
+    public boolean isBlinking(long $$0) {
+        return this.blinkUntilTick > $$0 && (this.blinkUntilTick - $$0) % 6L >= 3L;
+    }
+}
+