--- net/minecraft/client/gui/components/StateSwitchingButton.java
+++ net/minecraft/client/gui/components/StateSwitchingButton.java
@@ -13,49 +13,49 @@
     public int xDiffTex;
     public int yDiffTex;
 
-    public net.minecraft.client.gui.components.StateSwitchingButton(int x, int y, int width, int height, boolean initialState) {
-        super(x, y, width, height, net.minecraft.network.chat.CommonComponents.EMPTY);
-        this.isStateTriggered = initialState;
+    public net.minecraft.client.gui.components.StateSwitchingButton(int $$0, int $$1, int $$2, int $$3, boolean $$4) {
+        super($$0, $$1, $$2, $$3, net.minecraft.network.chat.CommonComponents.EMPTY);
+        this.isStateTriggered = $$4;
     }
 
-    public void initTextureValues(int xTexStart, int yTexStart, int xDiffTex, int yDiffTex, net.minecraft.resources.ResourceLocation resourceLocation) {
-        this.xTexStart = xTexStart;
-        this.yTexStart = yTexStart;
-        this.xDiffTex = xDiffTex;
-        this.yDiffTex = yDiffTex;
-        this.resourceLocation = resourceLocation;
+    public void initTextureValues(int $$0, int $$1, int $$2, int $$3, net.minecraft.resources.ResourceLocation $$4) {
+        this.xTexStart = $$0;
+        this.yTexStart = $$1;
+        this.xDiffTex = $$2;
+        this.yDiffTex = $$3;
+        this.resourceLocation = $$4;
     }
 
-    public void setStateTriggered(boolean triggered) {
-        this.isStateTriggered = triggered;
+    public void setStateTriggered(boolean $$0) {
+        this.isStateTriggered = $$0;
     }
 
     public boolean isStateTriggered() {
         return this.isStateTriggered;
     }
 
-    public void setPosition(int x, int y) {
-        this.x = x;
-        this.y = y;
+    public void setPosition(int $$0, int $$1) {
+        this.x = $$0;
+        this.y = $$1;
     }
 
     public void updateNarration(net.minecraft.client.gui.narration.NarrationElementOutput $$0) {
         this.defaultButtonNarrationText($$0);
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)this.resourceLocation);
         com.mojang.blaze3d.systems.RenderSystem.disableDepthTest();
-        int n = this.xTexStart;
-        int n2 = this.yTexStart;
+        int $$4 = this.xTexStart;
+        int $$5 = this.yTexStart;
         if (this.isStateTriggered) {
-            n += this.xDiffTex;
+            $$4 += this.xDiffTex;
         }
         if (this.isHoveredOrFocused()) {
-            n2 += this.yDiffTex;
+            $$5 += this.yDiffTex;
         }
-        this.blit(matrixStack, this.x, this.y, n, n2, this.width, this.height);
+        this.blit($$0, this.x, this.y, $$4, $$5, this.width, this.height);
         com.mojang.blaze3d.systems.RenderSystem.enableDepthTest();
     }
 }