--- net/minecraft/client/gui/components/SubtitleOverlay$Subtitle.java
+++ net/minecraft/client/gui/components/SubtitleOverlay$Subtitle.java
@@ -27,8 +27,8 @@
         return this.location;
     }
 
-    public void refresh(net.minecraft.world.phys.Vec3 location) {
-        this.location = location;
+    public void refresh(net.minecraft.world.phys.Vec3 $$0) {
+        this.location = $$0;
         this.time = net.minecraft.Util.getMillis();
     }
 }