--- net/minecraft/client/gui/components/SubtitleOverlay.java
+++ net/minecraft/client/gui/components/SubtitleOverlay.java
@@ -12,12 +12,12 @@
     public java.util.List<net.minecraft.client.gui.components.SubtitleOverlay$Subtitle> subtitles;
     public boolean isListening;
 
-    public net.minecraft.client.gui.components.SubtitleOverlay(net.minecraft.client.Minecraft minecraft) {
+    public net.minecraft.client.gui.components.SubtitleOverlay(net.minecraft.client.Minecraft $$0) {
         this.subtitles = com.google.common.collect.Lists.newArrayList();
-        this.minecraft = minecraft;
+        this.minecraft = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack poseStack) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0) {
         if (!this.isListening && ((java.lang.Boolean)this.minecraft.options.showSubtitles().get()).booleanValue()) {
             this.minecraft.getSoundManager().addListener((net.minecraft.client.sounds.SoundEventListener)this);
             this.isListening = true;
@@ -30,69 +30,69 @@
         }
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
-        net.minecraft.world.phys.Vec3 vec3 = new net.minecraft.world.phys.Vec3(this.minecraft.player.getX(), this.minecraft.player.getEyeY(), this.minecraft.player.getZ());
-        net.minecraft.world.phys.Vec3 vec32 = new net.minecraft.world.phys.Vec3(0.0, 0.0, -1.0).xRot(-this.minecraft.player.getXRot() * ((float)java.lang.Math.PI / 180)).yRot(-this.minecraft.player.getYRot() * ((float)java.lang.Math.PI / 180));
-        net.minecraft.world.phys.Vec3 vec33 = new net.minecraft.world.phys.Vec3(0.0, 1.0, 0.0).xRot(-this.minecraft.player.getXRot() * ((float)java.lang.Math.PI / 180)).yRot(-this.minecraft.player.getYRot() * ((float)java.lang.Math.PI / 180));
-        net.minecraft.world.phys.Vec3 vec34 = vec32.cross(vec33);
-        int n = 0;
-        int n2 = 0;
-        java.util.Iterator iterator = this.subtitles.iterator();
-        while (iterator.hasNext()) {
-            net.minecraft.client.gui.components.SubtitleOverlay$Subtitle subtitle = (net.minecraft.client.gui.components.SubtitleOverlay$Subtitle)iterator.next();
-            if (subtitle.getTime() + 3000L <= net.minecraft.Util.getMillis()) {
-                iterator.remove();
+        net.minecraft.world.phys.Vec3 $$1 = new net.minecraft.world.phys.Vec3(this.minecraft.player.getX(), this.minecraft.player.getEyeY(), this.minecraft.player.getZ());
+        net.minecraft.world.phys.Vec3 $$2 = new net.minecraft.world.phys.Vec3(0.0, 0.0, -1.0).xRot(-this.minecraft.player.getXRot() * ((float)java.lang.Math.PI / 180)).yRot(-this.minecraft.player.getYRot() * ((float)java.lang.Math.PI / 180));
+        net.minecraft.world.phys.Vec3 $$3 = new net.minecraft.world.phys.Vec3(0.0, 1.0, 0.0).xRot(-this.minecraft.player.getXRot() * ((float)java.lang.Math.PI / 180)).yRot(-this.minecraft.player.getYRot() * ((float)java.lang.Math.PI / 180));
+        net.minecraft.world.phys.Vec3 $$4 = $$2.cross($$3);
+        int $$5 = 0;
+        int $$6 = 0;
+        java.util.Iterator $$7 = this.subtitles.iterator();
+        while ($$7.hasNext()) {
+            net.minecraft.client.gui.components.SubtitleOverlay$Subtitle $$8 = (net.minecraft.client.gui.components.SubtitleOverlay$Subtitle)$$7.next();
+            if ($$8.getTime() + 3000L <= net.minecraft.Util.getMillis()) {
+                $$7.remove();
                 continue;
             }
-            n2 = java.lang.Math.max((int)n2, (int)this.minecraft.font.width((net.minecraft.network.chat.FormattedText)subtitle.getText()));
+            $$6 = java.lang.Math.max((int)$$6, (int)this.minecraft.font.width((net.minecraft.network.chat.FormattedText)$$8.getText()));
         }
-        n2 += this.minecraft.font.width("<") + this.minecraft.font.width(" ") + this.minecraft.font.width(">") + this.minecraft.font.width(" ");
-        for (net.minecraft.client.gui.components.SubtitleOverlay$Subtitle subtitle : this.subtitles) {
-            int n3 = 255;
-            net.minecraft.network.chat.Component component = subtitle.getText();
-            net.minecraft.world.phys.Vec3 vec35 = subtitle.getLocation().subtract(vec3).normalize();
-            double d = -vec34.dot(vec35);
-            double d2 = -vec32.dot(vec35);
-            boolean bl = d2 > 0.5;
-            int n4 = n2 / 2;
+        $$6 += this.minecraft.font.width("<") + this.minecraft.font.width(" ") + this.minecraft.font.width(">") + this.minecraft.font.width(" ");
+        for (net.minecraft.client.gui.components.SubtitleOverlay$Subtitle $$9 : this.subtitles) {
+            int $$10 = 255;
+            net.minecraft.network.chat.Component $$11 = $$9.getText();
+            net.minecraft.world.phys.Vec3 $$12 = $$9.getLocation().subtract($$1).normalize();
+            double $$13 = -$$4.dot($$12);
+            double $$14 = -$$2.dot($$12);
+            boolean $$15 = $$14 > 0.5;
+            int $$16 = $$6 / 2;
             java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-            int n5 = 9;
-            int n6 = n5 / 2;
-            float f = 1.0f;
-            int n7 = this.minecraft.font.width((net.minecraft.network.chat.FormattedText)component);
-            int n8 = net.minecraft.util.Mth.floor((float)net.minecraft.util.Mth.clampedLerp((float)255.0f, (float)75.0f, (float)((float)(net.minecraft.Util.getMillis() - subtitle.getTime()) / 3000.0f)));
-            int n9 = n8 << 16 | n8 << 8 | n8;
-            poseStack.pushPose();
-            poseStack.translate((double)((float)this.minecraft.getWindow().getGuiScaledWidth() - (float)n4 * 1.0f - 2.0f), (double)((float)(this.minecraft.getWindow().getGuiScaledHeight() - 35) - (float)(n * (n5 + 1)) * 1.0f), 0.0);
-            poseStack.scale(1.0f, 1.0f, 1.0f);
-            net.minecraft.client.gui.components.SubtitleOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)poseStack, (int)(-n4 - 1), (int)(-n6 - 1), (int)(n4 + 1), (int)(n6 + 1), (int)this.minecraft.options.getBackgroundColor(0.8f));
+            int $$17 = 9;
+            int $$18 = $$17 / 2;
+            float $$19 = 1.0f;
+            int $$20 = this.minecraft.font.width((net.minecraft.network.chat.FormattedText)$$11);
+            int $$21 = net.minecraft.util.Mth.floor((float)net.minecraft.util.Mth.clampedLerp((float)255.0f, (float)75.0f, (float)((float)(net.minecraft.Util.getMillis() - $$9.getTime()) / 3000.0f)));
+            int $$22 = $$21 << 16 | $$21 << 8 | $$21;
+            $$0.pushPose();
+            $$0.translate((double)((float)this.minecraft.getWindow().getGuiScaledWidth() - (float)$$16 * 1.0f - 2.0f), (double)((float)(this.minecraft.getWindow().getGuiScaledHeight() - 35) - (float)($$5 * ($$17 + 1)) * 1.0f), 0.0);
+            $$0.scale(1.0f, 1.0f, 1.0f);
+            net.minecraft.client.gui.components.SubtitleOverlay.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)(-$$16 - 1), (int)(-$$18 - 1), (int)($$16 + 1), (int)($$18 + 1), (int)this.minecraft.options.getBackgroundColor(0.8f));
             com.mojang.blaze3d.systems.RenderSystem.enableBlend();
-            if (!bl) {
-                if (d > 0.0) {
-                    this.minecraft.font.draw(poseStack, ">", (float)(n4 - this.minecraft.font.width(">")), (float)(-n6), n9 + -16777216);
-                } else if (d < 0.0) {
-                    this.minecraft.font.draw(poseStack, "<", (float)(-n4), (float)(-n6), n9 + -16777216);
+            if (!$$15) {
+                if ($$13 > 0.0) {
+                    this.minecraft.font.draw($$0, ">", (float)($$16 - this.minecraft.font.width(">")), (float)(-$$18), $$22 + -16777216);
+                } else if ($$13 < 0.0) {
+                    this.minecraft.font.draw($$0, "<", (float)(-$$16), (float)(-$$18), $$22 + -16777216);
                 }
             }
-            this.minecraft.font.draw(poseStack, component, (float)(-n7 / 2), (float)(-n6), n9 + -16777216);
-            poseStack.popPose();
-            ++n;
+            this.minecraft.font.draw($$0, $$11, (float)(-$$20 / 2), (float)(-$$18), $$22 + -16777216);
+            $$0.popPose();
+            ++$$5;
         }
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();
     }
 
-    public void onPlaySound(net.minecraft.client.resources.sounds.SoundInstance sound, net.minecraft.client.sounds.WeighedSoundEvents accessor) {
-        if (accessor.getSubtitle() == null) {
+    public void onPlaySound(net.minecraft.client.resources.sounds.SoundInstance $$0, net.minecraft.client.sounds.WeighedSoundEvents $$1) {
+        if ($$1.getSubtitle() == null) {
             return;
         }
-        net.minecraft.network.chat.Component component = accessor.getSubtitle();
+        net.minecraft.network.chat.Component $$2 = $$1.getSubtitle();
         if (!this.subtitles.isEmpty()) {
-            for (net.minecraft.client.gui.components.SubtitleOverlay$Subtitle subtitle : this.subtitles) {
-                if (!subtitle.getText().equals((java.lang.Object)component)) continue;
-                subtitle.refresh(new net.minecraft.world.phys.Vec3(sound.getX(), sound.getY(), sound.getZ()));
+            for (net.minecraft.client.gui.components.SubtitleOverlay$Subtitle $$3 : this.subtitles) {
+                if (!$$3.getText().equals((java.lang.Object)$$2)) continue;
+                $$3.refresh(new net.minecraft.world.phys.Vec3($$0.getX(), $$0.getY(), $$0.getZ()));
                 return;
             }
         }
-        this.subtitles.add((java.lang.Object)new net.minecraft.client.gui.components.SubtitleOverlay$Subtitle(component, new net.minecraft.world.phys.Vec3(sound.getX(), sound.getY(), sound.getZ())));
+        this.subtitles.add((java.lang.Object)new net.minecraft.client.gui.components.SubtitleOverlay$Subtitle($$2, new net.minecraft.world.phys.Vec3($$0.getX(), $$0.getY(), $$0.getZ())));
     }
 }
 