--- net/minecraft/network/syncher/EntityDataSerializers.java
+++ net/minecraft/network/syncher/EntityDataSerializers.java
@@ -8,6 +8,7 @@
     public static net.minecraft.util.CrudeIncrementalIntIdentityHashBiMap<net.minecraft.network.syncher.EntityDataSerializer<?>> SERIALIZERS;
     public static net.minecraft.network.syncher.EntityDataSerializer<java.lang.Byte> BYTE;
     public static net.minecraft.network.syncher.EntityDataSerializer<java.lang.Integer> INT;
+    public static net.minecraft.network.syncher.EntityDataSerializer<java.lang.Long> LONG;
     public static net.minecraft.network.syncher.EntityDataSerializer<java.lang.Float> FLOAT;
     public static net.minecraft.network.syncher.EntityDataSerializer<java.lang.String> STRING;
     public static net.minecraft.network.syncher.EntityDataSerializer<net.minecraft.network.chat.Component> COMPONENT;
@@ -30,17 +31,17 @@
     public static net.minecraft.network.syncher.EntityDataSerializer<net.minecraft.world.entity.animal.FrogVariant> FROG_VARIANT;
     public static net.minecraft.network.syncher.EntityDataSerializer<net.minecraft.core.Holder<net.minecraft.world.entity.decoration.PaintingVariant>> PAINTING_VARIANT;
 
-    public static void registerSerializer(net.minecraft.network.syncher.EntityDataSerializer<?> serializer) {
-        SERIALIZERS.add(serializer);
+    public static void registerSerializer(net.minecraft.network.syncher.EntityDataSerializer<?> $$0) {
+        SERIALIZERS.add($$0);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.network.syncher.EntityDataSerializer<?> getSerializer(int id) {
-        return (net.minecraft.network.syncher.EntityDataSerializer)SERIALIZERS.byId(id);
+    public static net.minecraft.network.syncher.EntityDataSerializer<?> getSerializer(int $$0) {
+        return (net.minecraft.network.syncher.EntityDataSerializer)SERIALIZERS.byId($$0);
     }
 
-    public static int getSerializedId(net.minecraft.network.syncher.EntityDataSerializer<?> serializer) {
-        return SERIALIZERS.getId(serializer);
+    public static int getSerializedId(net.minecraft.network.syncher.EntityDataSerializer<?> $$0) {
+        return SERIALIZERS.getId($$0);
     }
 
     public net.minecraft.network.syncher.EntityDataSerializers() {
@@ -54,6 +55,7 @@
         SERIALIZERS = net.minecraft.util.CrudeIncrementalIntIdentityHashBiMap.create((int)16);
         BYTE = net.minecraft.network.syncher.EntityDataSerializer.simple(($$0, $$1) -> $$0.writeByte((int)$$1.byteValue()), net.minecraft.network.FriendlyByteBuf::readByte);
         INT = net.minecraft.network.syncher.EntityDataSerializer.simple(net.minecraft.network.FriendlyByteBuf::writeVarInt, net.minecraft.network.FriendlyByteBuf::readVarInt);
+        LONG = net.minecraft.network.syncher.EntityDataSerializer.simple(net.minecraft.network.FriendlyByteBuf::writeVarLong, net.minecraft.network.FriendlyByteBuf::readVarLong);
         FLOAT = net.minecraft.network.syncher.EntityDataSerializer.simple(net.minecraft.network.FriendlyByteBuf::writeFloat, net.minecraft.network.FriendlyByteBuf::readFloat);
         STRING = net.minecraft.network.syncher.EntityDataSerializer.simple(net.minecraft.network.FriendlyByteBuf::writeUtf, net.minecraft.network.FriendlyByteBuf::readUtf);
         COMPONENT = net.minecraft.network.syncher.EntityDataSerializer.simple(net.minecraft.network.FriendlyByteBuf::writeComponent, net.minecraft.network.FriendlyByteBuf::readComponent);
@@ -77,6 +79,7 @@
         PAINTING_VARIANT = net.minecraft.network.syncher.EntityDataSerializer.simpleId((net.minecraft.core.IdMap)net.minecraft.core.Registry.PAINTING_VARIANT.asHolderIdMap());
         net.minecraft.network.syncher.EntityDataSerializers.registerSerializer((net.minecraft.network.syncher.EntityDataSerializer)BYTE);
         net.minecraft.network.syncher.EntityDataSerializers.registerSerializer((net.minecraft.network.syncher.EntityDataSerializer)INT);
+        net.minecraft.network.syncher.EntityDataSerializers.registerSerializer((net.minecraft.network.syncher.EntityDataSerializer)LONG);
         net.minecraft.network.syncher.EntityDataSerializers.registerSerializer((net.minecraft.network.syncher.EntityDataSerializer)FLOAT);
         net.minecraft.network.syncher.EntityDataSerializers.registerSerializer((net.minecraft.network.syncher.EntityDataSerializer)STRING);
         net.minecraft.network.syncher.EntityDataSerializers.registerSerializer((net.minecraft.network.syncher.EntityDataSerializer)COMPONENT);