--- net/minecraft/client/model/dragon/DragonHeadModel.java
+++ net/minecraft/client/model/dragon/DragonHeadModel.java
@@ -29,12 +29,12 @@
         this.head.xRot = $$2 * ((float)java.lang.Math.PI / 180);
     }
 
-    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack matrixStack, com.mojang.blaze3d.vertex.VertexConsumer buffer, int packedLight, int packedOverlay, float red, float green, float blue, float alpha) {
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, (double)-0.374375f, 0.0);
-        matrixStack.scale(0.75f, 0.75f, 0.75f);
-        this.head.render(matrixStack, buffer, packedLight, packedOverlay, red, green, blue, alpha);
-        matrixStack.popPose();
+    public void renderToBuffer(com.mojang.blaze3d.vertex.PoseStack $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1, int $$2, int $$3, float $$4, float $$5, float $$6, float $$7) {
+        $$0.pushPose();
+        $$0.translate(0.0, (double)-0.374375f, 0.0);
+        $$0.scale(0.75f, 0.75f, 0.75f);
+        this.head.render($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7);
+        $$0.popPose();
     }
 }
 