--- net/minecraft/client/gui/font/FontTexture.java
+++ net/minecraft/client/gui/font/FontTexture.java
@@ -24,7 +24,7 @@
         this.polygonOffsetType = $$1 ? net.minecraft.client.renderer.RenderType.textPolygonOffset((net.minecraft.resources.ResourceLocation)$$0) : net.minecraft.client.renderer.RenderType.textIntensityPolygonOffset((net.minecraft.resources.ResourceLocation)$$0);
     }
 
-    public void load(net.minecraft.server.packs.resources.ResourceManager manager) {
+    public void load(net.minecraft.server.packs.resources.ResourceManager $$0) {
     }
 
     public void close() {