--- net/minecraft/client/gui/font/FontManager.java
+++ net/minecraft/client/gui/font/FontManager.java
@@ -9,6 +9,7 @@
     public static org.slf4j.Logger LOGGER;
     public static java.lang.String FONTS_PATH = "fonts.json";
     public static net.minecraft.resources.ResourceLocation MISSING_FONT;
+    public static net.minecraft.resources.FileToIdConverter FONT_DEFINITIONS;
     public net.minecraft.client.gui.font.FontSet missingFontSet;
     public java.util.Map<net.minecraft.resources.ResourceLocation, net.minecraft.client.gui.font.FontSet> fontSets;
     public net.minecraft.client.renderer.texture.TextureManager textureManager;
@@ -23,8 +24,8 @@
         this.missingFontSet = (net.minecraft.client.gui.font.FontSet)net.minecraft.Util.make((java.lang.Object)new net.minecraft.client.gui.font.FontSet($$02, MISSING_FONT), $$0 -> $$0.reload((java.util.List)com.google.common.collect.Lists.newArrayList((java.lang.Object[])new com.mojang.blaze3d.font.GlyphProvider[]{new net.minecraft.client.gui.font.AllMissingGlyphProvider()})));
     }
 
-    public void setRenames(java.util.Map<net.minecraft.resources.ResourceLocation, net.minecraft.resources.ResourceLocation> unicodeForcedMap) {
-        this.renames = unicodeForcedMap;
+    public void setRenames(java.util.Map<net.minecraft.resources.ResourceLocation, net.minecraft.resources.ResourceLocation> $$0) {
+        this.renames = $$0;
     }
 
     public net.minecraft.client.gui.Font createFont() {
@@ -47,6 +48,7 @@
     static {
         LOGGER = com.mojang.logging.LogUtils.getLogger();
         MISSING_FONT = new net.minecraft.resources.ResourceLocation("minecraft", "missing");
+        FONT_DEFINITIONS = net.minecraft.resources.FileToIdConverter.json((java.lang.String)"font");
     }
 }
 