--- net/minecraft/advancements/critereon/AbstractCriterionTriggerInstance.java
+++ net/minecraft/advancements/critereon/AbstractCriterionTriggerInstance.java
@@ -9,9 +9,9 @@
     public net.minecraft.resources.ResourceLocation criterion;
     public net.minecraft.advancements.critereon.EntityPredicate$Composite player;
 
-    public net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance(net.minecraft.resources.ResourceLocation criterion, net.minecraft.advancements.critereon.EntityPredicate$Composite player) {
-        this.criterion = criterion;
-        this.player = player;
+    public net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance(net.minecraft.resources.ResourceLocation $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1) {
+        this.criterion = $$0;
+        this.player = $$1;
     }
 
     public net.minecraft.resources.ResourceLocation getCriterion() {
@@ -22,10 +22,10 @@
         return this.player;
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = new com.google.gson.JsonObject();
-        jsonObject.add("player", this.player.toJson(conditions));
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = new com.google.gson.JsonObject();
+        $$1.add("player", this.player.toJson($$0));
+        return $$1;
     }
 
     public java.lang.String toString() {