--- net/minecraft/advancements/critereon/BeeNestDestroyedTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/BeeNestDestroyedTrigger$TriggerInstance.java
@@ -11,15 +11,15 @@
     public net.minecraft.advancements.critereon.ItemPredicate item;
     public net.minecraft.advancements.critereon.MinMaxBounds$Ints numBees;
 
-    public net.minecraft.advancements.critereon.BeeNestDestroyedTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite player, @javax.annotation.Nullable net.minecraft.world.level.block.Block block, net.minecraft.advancements.critereon.ItemPredicate item, net.minecraft.advancements.critereon.MinMaxBounds$Ints numBees) {
-        super(net.minecraft.advancements.critereon.BeeNestDestroyedTrigger.ID, player);
-        this.block = block;
-        this.item = item;
-        this.numBees = numBees;
+    public net.minecraft.advancements.critereon.BeeNestDestroyedTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite $$0, @javax.annotation.Nullable net.minecraft.world.level.block.Block $$1, net.minecraft.advancements.critereon.ItemPredicate $$2, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$3) {
+        super(net.minecraft.advancements.critereon.BeeNestDestroyedTrigger.ID, $$0);
+        this.block = $$1;
+        this.item = $$2;
+        this.numBees = $$3;
     }
 
-    public static net.minecraft.advancements.critereon.BeeNestDestroyedTrigger$TriggerInstance destroyedBeeNest(net.minecraft.world.level.block.Block block, net.minecraft.advancements.critereon.ItemPredicate$Builder itemConditionBuilder, net.minecraft.advancements.critereon.MinMaxBounds$Ints beesContained) {
-        return new net.minecraft.advancements.critereon.BeeNestDestroyedTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, block, itemConditionBuilder.build(), beesContained);
+    public static net.minecraft.advancements.critereon.BeeNestDestroyedTrigger$TriggerInstance destroyedBeeNest(net.minecraft.world.level.block.Block $$0, net.minecraft.advancements.critereon.ItemPredicate$Builder $$1, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$2) {
+        return new net.minecraft.advancements.critereon.BeeNestDestroyedTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0, $$1.build(), $$2);
     }
 
     public boolean matches(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.item.ItemStack $$1, int $$2) {
@@ -32,14 +32,14 @@
         return this.numBees.matches($$2);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
         if (this.block != null) {
-            jsonObject.addProperty("block", net.minecraft.core.Registry.BLOCK.getKey((java.lang.Object)this.block).toString());
+            $$1.addProperty("block", net.minecraft.core.Registry.BLOCK.getKey((java.lang.Object)this.block).toString());
         }
-        jsonObject.add("item", this.item.serializeToJson());
-        jsonObject.add("num_bees_inside", this.numBees.serializeToJson());
-        return jsonObject;
+        $$1.add("item", this.item.serializeToJson());
+        $$1.add("num_bees_inside", this.numBees.serializeToJson());
+        return $$1;
     }
 }
 