--- net/minecraft/advancements/critereon/BlockPredicate$Builder.java
+++ net/minecraft/advancements/critereon/BlockPredicate$Builder.java
@@ -41,8 +41,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.BlockPredicate$Builder setProperties(net.minecraft.advancements.critereon.StatePropertiesPredicate statePredicate) {
-        this.properties = statePredicate;
+    public net.minecraft.advancements.critereon.BlockPredicate$Builder setProperties(net.minecraft.advancements.critereon.StatePropertiesPredicate $$0) {
+        this.properties = $$0;
         return this;
     }
 