--- net/minecraft/advancements/critereon/BredAnimalsTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/BredAnimalsTrigger$TriggerInstance.java
@@ -10,38 +10,38 @@
     public net.minecraft.advancements.critereon.EntityPredicate$Composite partner;
     public net.minecraft.advancements.critereon.EntityPredicate$Composite child;
 
-    public net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite player, net.minecraft.advancements.critereon.EntityPredicate$Composite parent, net.minecraft.advancements.critereon.EntityPredicate$Composite partner, net.minecraft.advancements.critereon.EntityPredicate$Composite child) {
-        super(net.minecraft.advancements.critereon.BredAnimalsTrigger.ID, player);
-        this.parent = parent;
-        this.partner = partner;
-        this.child = child;
+    public net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1, net.minecraft.advancements.critereon.EntityPredicate$Composite $$2, net.minecraft.advancements.critereon.EntityPredicate$Composite $$3) {
+        super(net.minecraft.advancements.critereon.BredAnimalsTrigger.ID, $$0);
+        this.parent = $$1;
+        this.partner = $$2;
+        this.child = $$3;
     }
 
     public static net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance bredAnimals() {
         return new net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY);
     }
 
-    public static net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance bredAnimals(net.minecraft.advancements.critereon.EntityPredicate$Builder builder) {
-        return new net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)builder.build()));
+    public static net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance bredAnimals(net.minecraft.advancements.critereon.EntityPredicate$Builder $$0) {
+        return new net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$0.build()));
     }
 
-    public static net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance bredAnimals(net.minecraft.advancements.critereon.EntityPredicate parent, net.minecraft.advancements.critereon.EntityPredicate partner, net.minecraft.advancements.critereon.EntityPredicate child) {
-        return new net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)parent), net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)partner), net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)child));
+    public static net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance bredAnimals(net.minecraft.advancements.critereon.EntityPredicate $$0, net.minecraft.advancements.critereon.EntityPredicate $$1, net.minecraft.advancements.critereon.EntityPredicate $$2) {
+        return new net.minecraft.advancements.critereon.BredAnimalsTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$0), net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$1), net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$2));
     }
 
-    public boolean matches(net.minecraft.world.level.storage.loot.LootContext parentContext, net.minecraft.world.level.storage.loot.LootContext partnerContext, @javax.annotation.Nullable net.minecraft.world.level.storage.loot.LootContext childContext) {
-        if (!(this.child == net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY || childContext != null && this.child.matches(childContext))) {
+    public boolean matches(net.minecraft.world.level.storage.loot.LootContext $$0, net.minecraft.world.level.storage.loot.LootContext $$1, @javax.annotation.Nullable net.minecraft.world.level.storage.loot.LootContext $$2) {
+        if (!(this.child == net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY || $$2 != null && this.child.matches($$2))) {
             return false;
         }
-        return this.parent.matches(parentContext) && this.partner.matches(partnerContext) || this.parent.matches(partnerContext) && this.partner.matches(parentContext);
+        return this.parent.matches($$0) && this.partner.matches($$1) || this.parent.matches($$1) && this.partner.matches($$0);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("parent", this.parent.toJson(conditions));
-        jsonObject.add("partner", this.partner.toJson(conditions));
-        jsonObject.add("child", this.child.toJson(conditions));
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("parent", this.parent.toJson($$0));
+        $$1.add("partner", this.partner.toJson($$0));
+        $$1.add("child", this.child.toJson($$0));
+        return $$1;
     }
 }
 