--- net/minecraft/advancements/critereon/BrewedPotionTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/BrewedPotionTrigger$TriggerInstance.java
@@ -18,16 +18,16 @@
         return new net.minecraft.advancements.critereon.BrewedPotionTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, null);
     }
 
-    public boolean matches(net.minecraft.world.item.alchemy.Potion potion) {
-        return this.potion == null || this.potion == potion;
+    public boolean matches(net.minecraft.world.item.alchemy.Potion $$0) {
+        return this.potion == null || this.potion == $$0;
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
         if (this.potion != null) {
-            jsonObject.addProperty("potion", net.minecraft.core.Registry.POTION.getKey((java.lang.Object)this.potion).toString());
+            $$1.addProperty("potion", net.minecraft.core.Registry.POTION.getKey((java.lang.Object)this.potion).toString());
         }
-        return jsonObject;
+        return $$1;
     }
 }
 