--- net/minecraft/advancements/critereon/ChangeDimensionTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/ChangeDimensionTrigger$TriggerInstance.java
@@ -25,30 +25,30 @@
         return new net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0, $$1);
     }
 
-    public static net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance changedDimensionTo(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> toLevel) {
-        return new net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, null, toLevel);
+    public static net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance changedDimensionTo(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$0) {
+        return new net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, null, $$0);
     }
 
     public static net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance changedDimensionFrom(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$0) {
         return new net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0, null);
     }
 
-    public boolean matches(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> fromLevel, net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> toLevel) {
-        if (this.from != null && this.from != fromLevel) {
+    public boolean matches(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$0, net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$1) {
+        if (this.from != null && this.from != $$0) {
             return false;
         }
-        return this.to == null || this.to == toLevel;
+        return this.to == null || this.to == $$1;
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
         if (this.from != null) {
-            jsonObject.addProperty("from", this.from.location().toString());
+            $$1.addProperty("from", this.from.location().toString());
         }
         if (this.to != null) {
-            jsonObject.addProperty("to", this.to.location().toString());
+            $$1.addProperty("to", this.to.location().toString());
         }
-        return jsonObject;
+        return $$1;
     }
 }
 