--- net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
+++ net/minecraft/advancements/critereon/ChangeDimensionTrigger.java
@@ -15,14 +15,14 @@
         return ID;
     }
 
-    public net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance createInstance(com.google.gson.JsonObject json, net.minecraft.advancements.critereon.EntityPredicate$Composite entityPredicate, net.minecraft.advancements.critereon.DeserializationContext conditionsParser) {
-        net.minecraft.resources.ResourceKey resourceKey = json.has("from") ? net.minecraft.resources.ResourceKey.create((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.DIMENSION_REGISTRY, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)json, (java.lang.String)"from"))) : null;
-        net.minecraft.resources.ResourceKey resourceKey2 = json.has("to") ? net.minecraft.resources.ResourceKey.create((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.DIMENSION_REGISTRY, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)json, (java.lang.String)"to"))) : null;
-        return new net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance(entityPredicate, resourceKey, resourceKey2);
+    public net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance createInstance(com.google.gson.JsonObject $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        net.minecraft.resources.ResourceKey $$3 = $$0.has("from") ? net.minecraft.resources.ResourceKey.create((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.DIMENSION_REGISTRY, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"from"))) : null;
+        net.minecraft.resources.ResourceKey $$4 = $$0.has("to") ? net.minecraft.resources.ResourceKey.create((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.DIMENSION_REGISTRY, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"to"))) : null;
+        return new net.minecraft.advancements.critereon.ChangeDimensionTrigger$TriggerInstance($$1, $$3, $$4);
     }
 
-    public void trigger(net.minecraft.server.level.ServerPlayer player, net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> fromLevel, net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> toLevel) {
-        this.trigger(player, $$2 -> $$2.matches(fromLevel, toLevel));
+    public void trigger(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$1, net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$22) {
+        this.trigger($$0, $$2 -> $$2.matches($$1, $$22));
     }
 
     public /* synthetic */ net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance createInstance(com.google.gson.JsonObject jsonObject, net.minecraft.advancements.critereon.EntityPredicate$Composite composite, net.minecraft.advancements.critereon.DeserializationContext deserializationContext) {