--- net/minecraft/advancements/critereon/ConsumeItemTrigger.java
+++ net/minecraft/advancements/critereon/ConsumeItemTrigger.java
@@ -15,12 +15,12 @@
         return ID;
     }
 
-    public net.minecraft.advancements.critereon.ConsumeItemTrigger$TriggerInstance createInstance(com.google.gson.JsonObject json, net.minecraft.advancements.critereon.EntityPredicate$Composite entityPredicate, net.minecraft.advancements.critereon.DeserializationContext conditionsParser) {
-        return new net.minecraft.advancements.critereon.ConsumeItemTrigger$TriggerInstance(entityPredicate, net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)json.get("item")));
+    public net.minecraft.advancements.critereon.ConsumeItemTrigger$TriggerInstance createInstance(com.google.gson.JsonObject $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        return new net.minecraft.advancements.critereon.ConsumeItemTrigger$TriggerInstance($$1, net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$0.get("item")));
     }
 
-    public void trigger(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.item.ItemStack item) {
-        this.trigger(player, $$1 -> $$1.matches(item));
+    public void trigger(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.item.ItemStack $$12) {
+        this.trigger($$0, $$1 -> $$1.matches($$12));
     }
 
     public /* synthetic */ net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance createInstance(com.google.gson.JsonObject jsonObject, net.minecraft.advancements.critereon.EntityPredicate$Composite composite, net.minecraft.advancements.critereon.DeserializationContext deserializationContext) {