--- net/minecraft/advancements/critereon/CuredZombieVillagerTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/CuredZombieVillagerTrigger$TriggerInstance.java
@@ -19,18 +19,18 @@
         return new net.minecraft.advancements.critereon.CuredZombieVillagerTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY);
     }
 
-    public boolean matches(net.minecraft.world.level.storage.loot.LootContext zombie, net.minecraft.world.level.storage.loot.LootContext villager) {
-        if (!this.zombie.matches(zombie)) {
+    public boolean matches(net.minecraft.world.level.storage.loot.LootContext $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        if (!this.zombie.matches($$0)) {
             return false;
         }
-        return this.villager.matches(villager);
+        return this.villager.matches($$1);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("zombie", this.zombie.toJson(conditions));
-        jsonObject.add("villager", this.villager.toJson(conditions));
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("zombie", this.zombie.toJson($$0));
+        $$1.add("villager", this.villager.toJson($$0));
+        return $$1;
     }
 }
 