--- net/minecraft/advancements/critereon/DamagePredicate.java
+++ net/minecraft/advancements/critereon/DamagePredicate.java
@@ -29,36 +29,36 @@
         this.type = $$4;
     }
 
-    public boolean matches(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.damagesource.DamageSource source, float dealt, float taken, boolean blocked) {
+    public boolean matches(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.damagesource.DamageSource $$1, float $$2, float $$3, boolean $$4) {
         if (this == ANY) {
             return true;
         }
-        if (!this.dealtDamage.matches((double)dealt)) {
+        if (!this.dealtDamage.matches((double)$$2)) {
             return false;
         }
-        if (!this.takenDamage.matches((double)taken)) {
+        if (!this.takenDamage.matches((double)$$3)) {
             return false;
         }
-        if (!this.sourceEntity.matches(player, source.getEntity())) {
+        if (!this.sourceEntity.matches($$0, $$1.getEntity())) {
             return false;
         }
-        if (this.blocked != null && this.blocked != blocked) {
+        if (this.blocked != null && this.blocked != $$4) {
             return false;
         }
-        return this.type.matches(player, source);
+        return this.type.matches($$0, $$1);
     }
 
-    public static net.minecraft.advancements.critereon.DamagePredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement element) {
-        if (element == null || element.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.DamagePredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement $$0) {
+        if ($$0 == null || $$0.isJsonNull()) {
             return ANY;
         }
-        com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)element, (java.lang.String)"damage");
-        net.minecraft.advancements.critereon.MinMaxBounds$Doubles doubles = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)jsonObject.get("dealt"));
-        net.minecraft.advancements.critereon.MinMaxBounds$Doubles doubles2 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)jsonObject.get("taken"));
-        java.lang.Boolean bl = jsonObject.has("blocked") ? java.lang.Boolean.valueOf((boolean)net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)jsonObject, (java.lang.String)"blocked")) : null;
-        net.minecraft.advancements.critereon.EntityPredicate entityPredicate = net.minecraft.advancements.critereon.EntityPredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("source_entity"));
-        net.minecraft.advancements.critereon.DamageSourcePredicate damageSourcePredicate = net.minecraft.advancements.critereon.DamageSourcePredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("type"));
-        return new net.minecraft.advancements.critereon.DamagePredicate(doubles, doubles2, entityPredicate, bl, damageSourcePredicate);
+        com.google.gson.JsonObject $$1 = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$0, (java.lang.String)"damage");
+        net.minecraft.advancements.critereon.MinMaxBounds$Doubles $$2 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)$$1.get("dealt"));
+        net.minecraft.advancements.critereon.MinMaxBounds$Doubles $$3 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)$$1.get("taken"));
+        java.lang.Boolean $$4 = $$1.has("blocked") ? java.lang.Boolean.valueOf((boolean)net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)$$1, (java.lang.String)"blocked")) : null;
+        net.minecraft.advancements.critereon.EntityPredicate $$5 = net.minecraft.advancements.critereon.EntityPredicate.fromJson((com.google.gson.JsonElement)$$1.get("source_entity"));
+        net.minecraft.advancements.critereon.DamageSourcePredicate $$6 = net.minecraft.advancements.critereon.DamageSourcePredicate.fromJson((com.google.gson.JsonElement)$$1.get("type"));
+        return new net.minecraft.advancements.critereon.DamagePredicate($$2, $$3, $$5, $$4, $$6);
     }
 
     public com.google.gson.JsonElement serializeToJson() {