--- net/minecraft/advancements/critereon/DamageSourcePredicate$Builder.java
+++ net/minecraft/advancements/critereon/DamageSourcePredicate$Builder.java
@@ -33,8 +33,8 @@
         return new net.minecraft.advancements.critereon.DamageSourcePredicate$Builder();
     }
 
-    public net.minecraft.advancements.critereon.DamageSourcePredicate$Builder isProjectile(java.lang.Boolean isProjectile) {
-        this.isProjectile = isProjectile;
+    public net.minecraft.advancements.critereon.DamageSourcePredicate$Builder isProjectile(java.lang.Boolean $$0) {
+        this.isProjectile = $$0;
         return this;
     }
 
@@ -68,8 +68,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.DamageSourcePredicate$Builder isLightning(java.lang.Boolean isLightning) {
-        this.isLightning = isLightning;
+    public net.minecraft.advancements.critereon.DamageSourcePredicate$Builder isLightning(java.lang.Boolean $$0) {
+        this.isLightning = $$0;
         return this;
     }
 
@@ -78,8 +78,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.DamageSourcePredicate$Builder direct(net.minecraft.advancements.critereon.EntityPredicate$Builder directEntity) {
-        this.directEntity = directEntity.build();
+    public net.minecraft.advancements.critereon.DamageSourcePredicate$Builder direct(net.minecraft.advancements.critereon.EntityPredicate$Builder $$0) {
+        this.directEntity = $$0.build();
         return this;
     }
 