--- net/minecraft/advancements/critereon/DamagePredicate$Builder.java
+++ net/minecraft/advancements/critereon/DamagePredicate$Builder.java
@@ -38,8 +38,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.DamagePredicate$Builder blocked(java.lang.Boolean blocked) {
-        this.blocked = blocked;
+    public net.minecraft.advancements.critereon.DamagePredicate$Builder blocked(java.lang.Boolean $$0) {
+        this.blocked = $$0;
         return this;
     }
 
@@ -48,8 +48,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.DamagePredicate$Builder type(net.minecraft.advancements.critereon.DamageSourcePredicate$Builder damageType) {
-        this.type = damageType.build();
+    public net.minecraft.advancements.critereon.DamagePredicate$Builder type(net.minecraft.advancements.critereon.DamageSourcePredicate$Builder $$0) {
+        this.type = $$0.build();
         return this;
     }
 