--- net/minecraft/advancements/critereon/DeserializationContext.java
+++ net/minecraft/advancements/critereon/DeserializationContext.java
@@ -16,14 +16,14 @@
         this.predicateManager = $$1;
     }
 
-    public net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] deserializeConditions(com.google.gson.JsonArray json, java.lang.String string, net.minecraft.world.level.storage.loot.parameters.LootContextParamSet parameterSet) {
-        net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray = (net.minecraft.world.level.storage.loot.predicates.LootItemCondition[])this.predicateGson.fromJson((com.google.gson.JsonElement)json, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[].class);
-        net.minecraft.world.level.storage.loot.ValidationContext validationContext = new net.minecraft.world.level.storage.loot.ValidationContext(parameterSet, arg_0 -> ((net.minecraft.world.level.storage.loot.PredicateManager)this.predicateManager).get(arg_0), $$0 -> null);
-        for (net.minecraft.world.level.storage.loot.predicates.LootItemCondition lootItemCondition : lootItemConditionArray) {
-            lootItemCondition.validate(validationContext);
-            validationContext.getProblems().forEach(($$1, $$2) -> LOGGER.warn("Found validation problem in advancement trigger {}/{}: {}", new java.lang.Object[]{string, $$1, $$2}));
+    public net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] deserializeConditions(com.google.gson.JsonArray $$02, java.lang.String $$12, net.minecraft.world.level.storage.loot.parameters.LootContextParamSet $$22) {
+        net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$3 = (net.minecraft.world.level.storage.loot.predicates.LootItemCondition[])this.predicateGson.fromJson((com.google.gson.JsonElement)$$02, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[].class);
+        net.minecraft.world.level.storage.loot.ValidationContext $$4 = new net.minecraft.world.level.storage.loot.ValidationContext($$22, arg_0 -> ((net.minecraft.world.level.storage.loot.PredicateManager)this.predicateManager).get(arg_0), $$0 -> null);
+        for (net.minecraft.world.level.storage.loot.predicates.LootItemCondition $$5 : $$3) {
+            $$5.validate($$4);
+            $$4.getProblems().forEach(($$1, $$2) -> LOGGER.warn("Found validation problem in advancement trigger {}/{}: {}", new java.lang.Object[]{$$12, $$1, $$2}));
         }
-        return lootItemConditionArray;
+        return $$3;
     }
 
     public net.minecraft.resources.ResourceLocation getAdvancementId() {