--- net/minecraft/advancements/critereon/DistancePredicate.java
+++ net/minecraft/advancements/critereon/DistancePredicate.java
@@ -32,30 +32,30 @@
         return new net.minecraft.advancements.critereon.DistancePredicate(net.minecraft.advancements.critereon.MinMaxBounds$Doubles.ANY, net.minecraft.advancements.critereon.MinMaxBounds$Doubles.ANY, net.minecraft.advancements.critereon.MinMaxBounds$Doubles.ANY, net.minecraft.advancements.critereon.MinMaxBounds$Doubles.ANY, $$0);
     }
 
-    public boolean matches(double x1, double y1, double z1, double x2, double y2, double z2) {
-        float f = (float)(x1 - x2);
-        float f2 = (float)(y1 - y2);
-        float f3 = (float)(z1 - z2);
-        if (!(this.x.matches((double)net.minecraft.util.Mth.abs((float)f)) && this.y.matches((double)net.minecraft.util.Mth.abs((float)f2)) && this.z.matches((double)net.minecraft.util.Mth.abs((float)f3)))) {
+    public boolean matches(double $$0, double $$1, double $$2, double $$3, double $$4, double $$5) {
+        float $$6 = (float)($$0 - $$3);
+        float $$7 = (float)($$1 - $$4);
+        float $$8 = (float)($$2 - $$5);
+        if (!(this.x.matches((double)net.minecraft.util.Mth.abs((float)$$6)) && this.y.matches((double)net.minecraft.util.Mth.abs((float)$$7)) && this.z.matches((double)net.minecraft.util.Mth.abs((float)$$8)))) {
             return false;
         }
-        if (!this.horizontal.matchesSqr((double)(f * f + f3 * f3))) {
+        if (!this.horizontal.matchesSqr((double)($$6 * $$6 + $$8 * $$8))) {
             return false;
         }
-        return this.absolute.matchesSqr((double)(f * f + f2 * f2 + f3 * f3));
+        return this.absolute.matchesSqr((double)($$6 * $$6 + $$7 * $$7 + $$8 * $$8));
     }
 
-    public static net.minecraft.advancements.critereon.DistancePredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement element) {
-        if (element == null || element.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.DistancePredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement $$0) {
+        if ($$0 == null || $$0.isJsonNull()) {
             return ANY;
         }
-        com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)element, (java.lang.String)"distance");
-        net.minecraft.advancements.critereon.MinMaxBounds$Doubles doubles = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)jsonObject.get("x"));
-        net.minecraft.advancements.critereon.MinMaxBounds$Doubles doubles2 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)jsonObject.get("y"));
-        net.minecraft.advancements.critereon.MinMaxBounds$Doubles doubles3 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)jsonObject.get("z"));
-        net.minecraft.advancements.critereon.MinMaxBounds$Doubles doubles4 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)jsonObject.get("horizontal"));
-        net.minecraft.advancements.critereon.MinMaxBounds$Doubles doubles5 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)jsonObject.get("absolute"));
-        return new net.minecraft.advancements.critereon.DistancePredicate(doubles, doubles2, doubles3, doubles4, doubles5);
+        com.google.gson.JsonObject $$1 = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$0, (java.lang.String)"distance");
+        net.minecraft.advancements.critereon.MinMaxBounds$Doubles $$2 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)$$1.get("x"));
+        net.minecraft.advancements.critereon.MinMaxBounds$Doubles $$3 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)$$1.get("y"));
+        net.minecraft.advancements.critereon.MinMaxBounds$Doubles $$4 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)$$1.get("z"));
+        net.minecraft.advancements.critereon.MinMaxBounds$Doubles $$5 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)$$1.get("horizontal"));
+        net.minecraft.advancements.critereon.MinMaxBounds$Doubles $$6 = net.minecraft.advancements.critereon.MinMaxBounds$Doubles.fromJson((com.google.gson.JsonElement)$$1.get("absolute"));
+        return new net.minecraft.advancements.critereon.DistancePredicate($$2, $$3, $$4, $$5, $$6);
     }
 
     public com.google.gson.JsonElement serializeToJson() {