--- net/minecraft/advancements/critereon/EnchantedItemTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/EnchantedItemTrigger$TriggerInstance.java
@@ -9,28 +9,28 @@
     public net.minecraft.advancements.critereon.ItemPredicate item;
     public net.minecraft.advancements.critereon.MinMaxBounds$Ints levels;
 
-    public net.minecraft.advancements.critereon.EnchantedItemTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite player, net.minecraft.advancements.critereon.ItemPredicate item, net.minecraft.advancements.critereon.MinMaxBounds$Ints levels) {
-        super(net.minecraft.advancements.critereon.EnchantedItemTrigger.ID, player);
-        this.item = item;
-        this.levels = levels;
+    public net.minecraft.advancements.critereon.EnchantedItemTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite $$0, net.minecraft.advancements.critereon.ItemPredicate $$1, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$2) {
+        super(net.minecraft.advancements.critereon.EnchantedItemTrigger.ID, $$0);
+        this.item = $$1;
+        this.levels = $$2;
     }
 
     public static net.minecraft.advancements.critereon.EnchantedItemTrigger$TriggerInstance enchantedItem() {
         return new net.minecraft.advancements.critereon.EnchantedItemTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.ItemPredicate.ANY, net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY);
     }
 
-    public boolean matches(net.minecraft.world.item.ItemStack item, int levels) {
-        if (!this.item.matches(item)) {
+    public boolean matches(net.minecraft.world.item.ItemStack $$0, int $$1) {
+        if (!this.item.matches($$0)) {
             return false;
         }
-        return this.levels.matches(levels);
+        return this.levels.matches($$1);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("item", this.item.serializeToJson());
-        jsonObject.add("levels", this.levels.serializeToJson());
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("item", this.item.serializeToJson());
+        $$1.add("levels", this.levels.serializeToJson());
+        return $$1;
     }
 }
 