--- net/minecraft/advancements/critereon/EnchantmentPredicate.java
+++ net/minecraft/advancements/critereon/EnchantmentPredicate.java
@@ -16,23 +16,23 @@
         this.level = net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY;
     }
 
-    public net.minecraft.advancements.critereon.EnchantmentPredicate(@javax.annotation.Nullable net.minecraft.world.item.enchantment.Enchantment enchantment, net.minecraft.advancements.critereon.MinMaxBounds$Ints level) {
-        this.enchantment = enchantment;
-        this.level = level;
+    public net.minecraft.advancements.critereon.EnchantmentPredicate(@javax.annotation.Nullable net.minecraft.world.item.enchantment.Enchantment $$0, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$1) {
+        this.enchantment = $$0;
+        this.level = $$1;
     }
 
-    public boolean containedIn(java.util.Map<net.minecraft.world.item.enchantment.Enchantment, java.lang.Integer> enchantments) {
+    public boolean containedIn(java.util.Map<net.minecraft.world.item.enchantment.Enchantment, java.lang.Integer> $$0) {
         if (this.enchantment != null) {
-            if (!enchantments.containsKey((java.lang.Object)this.enchantment)) {
+            if (!$$0.containsKey((java.lang.Object)this.enchantment)) {
                 return false;
             }
-            int n = (java.lang.Integer)enchantments.get((java.lang.Object)this.enchantment);
-            if (this.level != net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY && !this.level.matches(n)) {
+            int $$1 = (java.lang.Integer)$$0.get((java.lang.Object)this.enchantment);
+            if (this.level != net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY && !this.level.matches($$1)) {
                 return false;
             }
         } else if (this.level != net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY) {
-            for (java.lang.Integer n : enchantments.values()) {
-                if (!this.level.matches(n.intValue())) continue;
+            for (java.lang.Integer $$2 : $$0.values()) {
+                if (!this.level.matches($$2.intValue())) continue;
                 return true;
             }
             return false;
@@ -52,37 +52,32 @@
         return $$0;
     }
 
-    public static net.minecraft.advancements.critereon.EnchantmentPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement element) {
-        net.minecraft.advancements.critereon.MinMaxBounds$Ints ints;
-        if (element == null || element.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.EnchantmentPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement $$0) {
+        if ($$0 == null || $$0.isJsonNull()) {
             return ANY;
         }
-        com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)element, (java.lang.String)"enchantment");
-        net.minecraft.world.item.enchantment.Enchantment enchantment = null;
-        if (jsonObject.has("enchantment")) {
-            ints = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)jsonObject, (java.lang.String)"enchantment"));
-            enchantment = (net.minecraft.world.item.enchantment.Enchantment)net.minecraft.core.Registry.ENCHANTMENT.getOptional((net.minecraft.resources.ResourceLocation)ints).orElseThrow(() -> net.minecraft.advancements.critereon.EnchantmentPredicate.lambda$fromJson$0((net.minecraft.resources.ResourceLocation)ints));
+        com.google.gson.JsonObject $$1 = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$0, (java.lang.String)"enchantment");
+        net.minecraft.world.item.enchantment.Enchantment $$2 = null;
+        if ($$1.has("enchantment")) {
+            net.minecraft.resources.ResourceLocation $$3 = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$1, (java.lang.String)"enchantment"));
+            $$2 = (net.minecraft.world.item.enchantment.Enchantment)net.minecraft.core.Registry.ENCHANTMENT.getOptional($$3).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown enchantment '" + $$3 + "'"));
         }
-        ints = net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromJson((com.google.gson.JsonElement)jsonObject.get("levels"));
-        return new net.minecraft.advancements.critereon.EnchantmentPredicate(enchantment, ints);
+        net.minecraft.advancements.critereon.MinMaxBounds$Ints $$4 = net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromJson((com.google.gson.JsonElement)$$1.get("levels"));
+        return new net.minecraft.advancements.critereon.EnchantmentPredicate($$2, $$4);
     }
 
-    public static net.minecraft.advancements.critereon.EnchantmentPredicate[] fromJsonArray(@javax.annotation.Nullable com.google.gson.JsonElement element) {
-        if (element == null || element.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.EnchantmentPredicate[] fromJsonArray(@javax.annotation.Nullable com.google.gson.JsonElement $$0) {
+        if ($$0 == null || $$0.isJsonNull()) {
             return NONE;
         }
-        com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.convertToJsonArray((com.google.gson.JsonElement)element, (java.lang.String)"enchantments");
-        net.minecraft.advancements.critereon.EnchantmentPredicate[] enchantmentPredicateArray = new net.minecraft.advancements.critereon.EnchantmentPredicate[jsonArray.size()];
-        for (int i = 0; i < enchantmentPredicateArray.length; ++i) {
-            enchantmentPredicateArray[i] = net.minecraft.advancements.critereon.EnchantmentPredicate.fromJson((com.google.gson.JsonElement)jsonArray.get(i));
+        com.google.gson.JsonArray $$1 = net.minecraft.util.GsonHelper.convertToJsonArray((com.google.gson.JsonElement)$$0, (java.lang.String)"enchantments");
+        net.minecraft.advancements.critereon.EnchantmentPredicate[] $$2 = new net.minecraft.advancements.critereon.EnchantmentPredicate[$$1.size()];
+        for (int $$3 = 0; $$3 < $$2.length; ++$$3) {
+            $$2[$$3] = net.minecraft.advancements.critereon.EnchantmentPredicate.fromJson((com.google.gson.JsonElement)$$1.get($$3));
         }
-        return enchantmentPredicateArray;
+        return $$2;
     }
 
-    public static /* synthetic */ com.google.gson.JsonSyntaxException lambda$fromJson$0(net.minecraft.resources.ResourceLocation $$0) {
-        return new com.google.gson.JsonSyntaxException("Unknown enchantment '" + $$0 + "'");
-    }
-
     static {
         ANY = new net.minecraft.advancements.critereon.EnchantmentPredicate();
         NONE = new net.minecraft.advancements.critereon.EnchantmentPredicate[0];