--- net/minecraft/advancements/critereon/EntityTypePredicate$TagPredicate.java
+++ net/minecraft/advancements/critereon/EntityTypePredicate$TagPredicate.java
@@ -12,8 +12,8 @@
         this.tag = $$0;
     }
 
-    public boolean matches(net.minecraft.world.entity.EntityType<?> type) {
-        return type.is(this.tag);
+    public boolean matches(net.minecraft.world.entity.EntityType<?> $$0) {
+        return $$0.is(this.tag);
     }
 
     public com.google.gson.JsonElement serializeToJson() {