--- net/minecraft/advancements/critereon/EntityHurtPlayerTrigger.java
+++ net/minecraft/advancements/critereon/EntityHurtPlayerTrigger.java
@@ -15,13 +15,13 @@
         return ID;
     }
 
-    public net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance createInstance(com.google.gson.JsonObject json, net.minecraft.advancements.critereon.EntityPredicate$Composite entityPredicate, net.minecraft.advancements.critereon.DeserializationContext conditionsParser) {
-        net.minecraft.advancements.critereon.DamagePredicate damagePredicate = net.minecraft.advancements.critereon.DamagePredicate.fromJson((com.google.gson.JsonElement)json.get("damage"));
-        return new net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance(entityPredicate, damagePredicate);
+    public net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance createInstance(com.google.gson.JsonObject $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        net.minecraft.advancements.critereon.DamagePredicate $$3 = net.minecraft.advancements.critereon.DamagePredicate.fromJson((com.google.gson.JsonElement)$$0.get("damage"));
+        return new net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance($$1, $$3);
     }
 
-    public void trigger(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.damagesource.DamageSource source, float amountDealt, float amountTaken, boolean wasBlocked) {
-        this.trigger(player, $$5 -> $$5.matches(player, source, amountDealt, amountTaken, wasBlocked));
+    public void trigger(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.damagesource.DamageSource $$1, float $$2, float $$3, boolean $$4) {
+        this.trigger($$0, $$5 -> $$5.matches($$0, $$1, $$2, $$3, $$4));
     }
 
     public /* synthetic */ net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance createInstance(com.google.gson.JsonObject jsonObject, net.minecraft.advancements.critereon.EntityPredicate$Composite composite, net.minecraft.advancements.critereon.DeserializationContext deserializationContext) {