--- net/minecraft/advancements/critereon/EntityTypePredicate$TypePredicate.java
+++ net/minecraft/advancements/critereon/EntityTypePredicate$TypePredicate.java
@@ -12,8 +12,8 @@
         this.type = $$0;
     }
 
-    public boolean matches(net.minecraft.world.entity.EntityType<?> type) {
-        return this.type == type;
+    public boolean matches(net.minecraft.world.entity.EntityType<?> $$0) {
+        return this.type == $$0;
     }
 
     public com.google.gson.JsonElement serializeToJson() {