--- net/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder.java
+++ net/minecraft/advancements/critereon/EntityEquipmentPredicate$Builder.java
@@ -25,23 +25,23 @@
         return new net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder();
     }
 
-    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder head(net.minecraft.advancements.critereon.ItemPredicate condition) {
-        this.head = condition;
+    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder head(net.minecraft.advancements.critereon.ItemPredicate $$0) {
+        this.head = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder chest(net.minecraft.advancements.critereon.ItemPredicate condition) {
-        this.chest = condition;
+    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder chest(net.minecraft.advancements.critereon.ItemPredicate $$0) {
+        this.chest = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder legs(net.minecraft.advancements.critereon.ItemPredicate condition) {
-        this.legs = condition;
+    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder legs(net.minecraft.advancements.critereon.ItemPredicate $$0) {
+        this.legs = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder feet(net.minecraft.advancements.critereon.ItemPredicate condition) {
-        this.feet = condition;
+    public net.minecraft.advancements.critereon.EntityEquipmentPredicate$Builder feet(net.minecraft.advancements.critereon.ItemPredicate $$0) {
+        this.feet = $$0;
         return this;
     }
 