--- net/minecraft/advancements/critereon/EntityPredicate$Builder.java
+++ net/minecraft/advancements/critereon/EntityPredicate$Builder.java
@@ -39,8 +39,8 @@
         return new net.minecraft.advancements.critereon.EntityPredicate$Builder();
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder of(net.minecraft.world.entity.EntityType<?> type) {
-        this.entityType = net.minecraft.advancements.critereon.EntityTypePredicate.of(type);
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder of(net.minecraft.world.entity.EntityType<?> $$0) {
+        this.entityType = net.minecraft.advancements.critereon.EntityTypePredicate.of($$0);
         return this;
     }
 
@@ -49,18 +49,18 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder entityType(net.minecraft.advancements.critereon.EntityTypePredicate type) {
-        this.entityType = type;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder entityType(net.minecraft.advancements.critereon.EntityTypePredicate $$0) {
+        this.entityType = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder distance(net.minecraft.advancements.critereon.DistancePredicate distance) {
-        this.distanceToPlayer = distance;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder distance(net.minecraft.advancements.critereon.DistancePredicate $$0) {
+        this.distanceToPlayer = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder located(net.minecraft.advancements.critereon.LocationPredicate location) {
-        this.location = location;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder located(net.minecraft.advancements.critereon.LocationPredicate $$0) {
+        this.location = $$0;
         return this;
     }
 
@@ -69,23 +69,23 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder effects(net.minecraft.advancements.critereon.MobEffectsPredicate effects) {
-        this.effects = effects;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder effects(net.minecraft.advancements.critereon.MobEffectsPredicate $$0) {
+        this.effects = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder nbt(net.minecraft.advancements.critereon.NbtPredicate nbt) {
-        this.nbt = nbt;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder nbt(net.minecraft.advancements.critereon.NbtPredicate $$0) {
+        this.nbt = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder flags(net.minecraft.advancements.critereon.EntityFlagsPredicate flags) {
-        this.flags = flags;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder flags(net.minecraft.advancements.critereon.EntityFlagsPredicate $$0) {
+        this.flags = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder equipment(net.minecraft.advancements.critereon.EntityEquipmentPredicate equipment) {
-        this.equipment = equipment;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder equipment(net.minecraft.advancements.critereon.EntityEquipmentPredicate $$0) {
+        this.equipment = $$0;
         return this;
     }
 
@@ -94,8 +94,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder vehicle(net.minecraft.advancements.critereon.EntityPredicate mount) {
-        this.vehicle = mount;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder vehicle(net.minecraft.advancements.critereon.EntityPredicate $$0) {
+        this.vehicle = $$0;
         return this;
     }
 
@@ -104,13 +104,13 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder targetedEntity(net.minecraft.advancements.critereon.EntityPredicate target) {
-        this.targetedEntity = target;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder targetedEntity(net.minecraft.advancements.critereon.EntityPredicate $$0) {
+        this.targetedEntity = $$0;
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityPredicate$Builder team(@javax.annotation.Nullable java.lang.String team) {
-        this.team = team;
+    public net.minecraft.advancements.critereon.EntityPredicate$Builder team(@javax.annotation.Nullable java.lang.String $$0) {
+        this.team = $$0;
         return this;
     }
 