--- net/minecraft/advancements/critereon/EntityFlagsPredicate$Builder.java
+++ net/minecraft/advancements/critereon/EntityFlagsPredicate$Builder.java
@@ -23,8 +23,8 @@
         return new net.minecraft.advancements.critereon.EntityFlagsPredicate$Builder();
     }
 
-    public net.minecraft.advancements.critereon.EntityFlagsPredicate$Builder setOnFire(@javax.annotation.Nullable java.lang.Boolean onFire) {
-        this.isOnFire = onFire;
+    public net.minecraft.advancements.critereon.EntityFlagsPredicate$Builder setOnFire(@javax.annotation.Nullable java.lang.Boolean $$0) {
+        this.isOnFire = $$0;
         return this;
     }
 
@@ -43,8 +43,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.EntityFlagsPredicate$Builder setIsBaby(@javax.annotation.Nullable java.lang.Boolean baby) {
-        this.isBaby = baby;
+    public net.minecraft.advancements.critereon.EntityFlagsPredicate$Builder setIsBaby(@javax.annotation.Nullable java.lang.Boolean $$0) {
+        this.isBaby = $$0;
         return this;
     }
 