--- net/minecraft/advancements/critereon/EntityHurtPlayerTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/EntityHurtPlayerTrigger$TriggerInstance.java
@@ -21,18 +21,18 @@
         return new net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0);
     }
 
-    public static net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance entityHurtPlayer(net.minecraft.advancements.critereon.DamagePredicate$Builder damageConditionBuilder) {
-        return new net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, damageConditionBuilder.build());
+    public static net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance entityHurtPlayer(net.minecraft.advancements.critereon.DamagePredicate$Builder $$0) {
+        return new net.minecraft.advancements.critereon.EntityHurtPlayerTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0.build());
     }
 
-    public boolean matches(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.damagesource.DamageSource source, float amountDealt, float amountTaken, boolean wasBlocked) {
-        return this.damage.matches(player, source, amountDealt, amountTaken, wasBlocked);
+    public boolean matches(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.damagesource.DamageSource $$1, float $$2, float $$3, boolean $$4) {
+        return this.damage.matches($$0, $$1, $$2, $$3, $$4);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("damage", this.damage.serializeToJson());
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("damage", this.damage.serializeToJson());
+        return $$1;
     }
 }
 