--- net/minecraft/advancements/critereon/EntityEquipmentPredicate.java
+++ net/minecraft/advancements/critereon/EntityEquipmentPredicate.java
@@ -23,44 +23,44 @@
         this.offhand = $$5;
     }
 
-    public boolean matches(@javax.annotation.Nullable net.minecraft.world.entity.Entity entity) {
+    public boolean matches(@javax.annotation.Nullable net.minecraft.world.entity.Entity $$0) {
         if (this == ANY) {
             return true;
         }
-        if (!(entity instanceof net.minecraft.world.entity.LivingEntity)) {
+        if (!($$0 instanceof net.minecraft.world.entity.LivingEntity)) {
             return false;
         }
-        net.minecraft.world.entity.LivingEntity livingEntity = (net.minecraft.world.entity.LivingEntity)entity;
-        if (!this.head.matches(livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD))) {
+        net.minecraft.world.entity.LivingEntity $$1 = (net.minecraft.world.entity.LivingEntity)$$0;
+        if (!this.head.matches($$1.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.HEAD))) {
             return false;
         }
-        if (!this.chest.matches(livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.CHEST))) {
+        if (!this.chest.matches($$1.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.CHEST))) {
             return false;
         }
-        if (!this.legs.matches(livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.LEGS))) {
+        if (!this.legs.matches($$1.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.LEGS))) {
             return false;
         }
-        if (!this.feet.matches(livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.FEET))) {
+        if (!this.feet.matches($$1.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.FEET))) {
             return false;
         }
-        if (!this.mainhand.matches(livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND))) {
+        if (!this.mainhand.matches($$1.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.MAINHAND))) {
             return false;
         }
-        return this.offhand.matches(livingEntity.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.OFFHAND));
+        return this.offhand.matches($$1.getItemBySlot(net.minecraft.world.entity.EquipmentSlot.OFFHAND));
     }
 
-    public static net.minecraft.advancements.critereon.EntityEquipmentPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement element) {
-        if (element == null || element.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.EntityEquipmentPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement $$0) {
+        if ($$0 == null || $$0.isJsonNull()) {
             return ANY;
         }
-        com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)element, (java.lang.String)"equipment");
-        net.minecraft.advancements.critereon.ItemPredicate itemPredicate = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("head"));
-        net.minecraft.advancements.critereon.ItemPredicate itemPredicate2 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("chest"));
-        net.minecraft.advancements.critereon.ItemPredicate itemPredicate3 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("legs"));
-        net.minecraft.advancements.critereon.ItemPredicate itemPredicate4 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("feet"));
-        net.minecraft.advancements.critereon.ItemPredicate itemPredicate5 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("mainhand"));
-        net.minecraft.advancements.critereon.ItemPredicate itemPredicate6 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)jsonObject.get("offhand"));
-        return new net.minecraft.advancements.critereon.EntityEquipmentPredicate(itemPredicate, itemPredicate2, itemPredicate3, itemPredicate4, itemPredicate5, itemPredicate6);
+        com.google.gson.JsonObject $$1 = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$0, (java.lang.String)"equipment");
+        net.minecraft.advancements.critereon.ItemPredicate $$2 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$1.get("head"));
+        net.minecraft.advancements.critereon.ItemPredicate $$3 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$1.get("chest"));
+        net.minecraft.advancements.critereon.ItemPredicate $$4 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$1.get("legs"));
+        net.minecraft.advancements.critereon.ItemPredicate $$5 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$1.get("feet"));
+        net.minecraft.advancements.critereon.ItemPredicate $$6 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$1.get("mainhand"));
+        net.minecraft.advancements.critereon.ItemPredicate $$7 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$1.get("offhand"));
+        return new net.minecraft.advancements.critereon.EntityEquipmentPredicate($$2, $$3, $$4, $$5, $$6, $$7);
     }
 
     public com.google.gson.JsonElement serializeToJson() {