--- net/minecraft/advancements/critereon/EntityPredicate$Composite.java
+++ net/minecraft/advancements/critereon/EntityPredicate$Composite.java
@@ -14,65 +14,65 @@
         this.compositePredicates = net.minecraft.world.level.storage.loot.predicates.LootItemConditions.andConditions((java.util.function.Predicate[])$$0);
     }
 
-    public static net.minecraft.advancements.critereon.EntityPredicate$Composite create(net.minecraft.world.level.storage.loot.predicates.LootItemCondition ... conditions) {
-        return new net.minecraft.advancements.critereon.EntityPredicate$Composite(conditions);
+    public static net.minecraft.advancements.critereon.EntityPredicate$Composite create(net.minecraft.world.level.storage.loot.predicates.LootItemCondition ... $$0) {
+        return new net.minecraft.advancements.critereon.EntityPredicate$Composite($$0);
     }
 
-    public static net.minecraft.advancements.critereon.EntityPredicate$Composite fromJson(com.google.gson.JsonObject jsonObject, java.lang.String name, net.minecraft.advancements.critereon.DeserializationContext conditions) {
-        com.google.gson.JsonElement jsonElement = jsonObject.get(name);
-        return net.minecraft.advancements.critereon.EntityPredicate$Composite.fromElement((java.lang.String)name, (net.minecraft.advancements.critereon.DeserializationContext)conditions, (com.google.gson.JsonElement)jsonElement);
+    public static net.minecraft.advancements.critereon.EntityPredicate$Composite fromJson(com.google.gson.JsonObject $$0, java.lang.String $$1, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        com.google.gson.JsonElement $$3 = $$0.get($$1);
+        return net.minecraft.advancements.critereon.EntityPredicate$Composite.fromElement((java.lang.String)$$1, (net.minecraft.advancements.critereon.DeserializationContext)$$2, (com.google.gson.JsonElement)$$3);
     }
 
-    public static net.minecraft.advancements.critereon.EntityPredicate$Composite[] fromJsonArray(com.google.gson.JsonObject jsonObject, java.lang.String name, net.minecraft.advancements.critereon.DeserializationContext conditions) {
-        com.google.gson.JsonElement jsonElement = jsonObject.get(name);
-        if (jsonElement == null || jsonElement.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.EntityPredicate$Composite[] fromJsonArray(com.google.gson.JsonObject $$0, java.lang.String $$1, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        com.google.gson.JsonElement $$3 = $$0.get($$1);
+        if ($$3 == null || $$3.isJsonNull()) {
             return new net.minecraft.advancements.critereon.EntityPredicate$Composite[0];
         }
-        com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.convertToJsonArray((com.google.gson.JsonElement)jsonElement, (java.lang.String)name);
-        net.minecraft.advancements.critereon.EntityPredicate$Composite[] compositeArray = new net.minecraft.advancements.critereon.EntityPredicate$Composite[jsonArray.size()];
-        for (int i = 0; i < jsonArray.size(); ++i) {
-            compositeArray[i] = net.minecraft.advancements.critereon.EntityPredicate$Composite.fromElement((java.lang.String)(name + "[" + i + "]"), (net.minecraft.advancements.critereon.DeserializationContext)conditions, (com.google.gson.JsonElement)jsonArray.get(i));
+        com.google.gson.JsonArray $$4 = net.minecraft.util.GsonHelper.convertToJsonArray((com.google.gson.JsonElement)$$3, (java.lang.String)$$1);
+        net.minecraft.advancements.critereon.EntityPredicate$Composite[] $$5 = new net.minecraft.advancements.critereon.EntityPredicate$Composite[$$4.size()];
+        for (int $$6 = 0; $$6 < $$4.size(); ++$$6) {
+            $$5[$$6] = net.minecraft.advancements.critereon.EntityPredicate$Composite.fromElement((java.lang.String)($$1 + "[" + $$6 + "]"), (net.minecraft.advancements.critereon.DeserializationContext)$$2, (com.google.gson.JsonElement)$$4.get($$6));
         }
-        return compositeArray;
+        return $$5;
     }
 
-    public static net.minecraft.advancements.critereon.EntityPredicate$Composite fromElement(java.lang.String name, net.minecraft.advancements.critereon.DeserializationContext conditions, @javax.annotation.Nullable com.google.gson.JsonElement element) {
-        if (element != null && element.isJsonArray()) {
-            net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray = conditions.deserializeConditions(element.getAsJsonArray(), conditions.getAdvancementId() + "/" + name, net.minecraft.world.level.storage.loot.parameters.LootContextParamSets.ADVANCEMENT_ENTITY);
-            return new net.minecraft.advancements.critereon.EntityPredicate$Composite(lootItemConditionArray);
+    public static net.minecraft.advancements.critereon.EntityPredicate$Composite fromElement(java.lang.String $$0, net.minecraft.advancements.critereon.DeserializationContext $$1, @javax.annotation.Nullable com.google.gson.JsonElement $$2) {
+        if ($$2 != null && $$2.isJsonArray()) {
+            net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$3 = $$1.deserializeConditions($$2.getAsJsonArray(), $$1.getAdvancementId() + "/" + $$0, net.minecraft.world.level.storage.loot.parameters.LootContextParamSets.ADVANCEMENT_ENTITY);
+            return new net.minecraft.advancements.critereon.EntityPredicate$Composite($$3);
         }
-        net.minecraft.advancements.critereon.EntityPredicate entityPredicate = net.minecraft.advancements.critereon.EntityPredicate.fromJson((com.google.gson.JsonElement)element);
-        return net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)entityPredicate);
+        net.minecraft.advancements.critereon.EntityPredicate $$4 = net.minecraft.advancements.critereon.EntityPredicate.fromJson((com.google.gson.JsonElement)$$2);
+        return net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$4);
     }
 
-    public static net.minecraft.advancements.critereon.EntityPredicate$Composite wrap(net.minecraft.advancements.critereon.EntityPredicate entityCondition) {
-        if (entityCondition == net.minecraft.advancements.critereon.EntityPredicate.ANY) {
+    public static net.minecraft.advancements.critereon.EntityPredicate$Composite wrap(net.minecraft.advancements.critereon.EntityPredicate $$0) {
+        if ($$0 == net.minecraft.advancements.critereon.EntityPredicate.ANY) {
             return ANY;
         }
-        net.minecraft.world.level.storage.loot.predicates.LootItemCondition lootItemCondition = net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition.hasProperties((net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.world.level.storage.loot.LootContext$EntityTarget.THIS, (net.minecraft.advancements.critereon.EntityPredicate)entityCondition).build();
-        return new net.minecraft.advancements.critereon.EntityPredicate$Composite(new net.minecraft.world.level.storage.loot.predicates.LootItemCondition[]{lootItemCondition});
+        net.minecraft.world.level.storage.loot.predicates.LootItemCondition $$1 = net.minecraft.world.level.storage.loot.predicates.LootItemEntityPropertyCondition.hasProperties((net.minecraft.world.level.storage.loot.LootContext$EntityTarget)net.minecraft.world.level.storage.loot.LootContext$EntityTarget.THIS, (net.minecraft.advancements.critereon.EntityPredicate)$$0).build();
+        return new net.minecraft.advancements.critereon.EntityPredicate$Composite(new net.minecraft.world.level.storage.loot.predicates.LootItemCondition[]{$$1});
     }
 
-    public boolean matches(net.minecraft.world.level.storage.loot.LootContext context) {
-        return this.compositePredicates.test((java.lang.Object)context);
+    public boolean matches(net.minecraft.world.level.storage.loot.LootContext $$0) {
+        return this.compositePredicates.test((java.lang.Object)$$0);
     }
 
-    public com.google.gson.JsonElement toJson(net.minecraft.advancements.critereon.SerializationContext serializer) {
+    public com.google.gson.JsonElement toJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
         if (this.conditions.length == 0) {
             return com.google.gson.JsonNull.INSTANCE;
         }
-        return serializer.serializeConditions(this.conditions);
+        return $$0.serializeConditions(this.conditions);
     }
 
-    public static com.google.gson.JsonElement toJson(net.minecraft.advancements.critereon.EntityPredicate$Composite[] predicates, net.minecraft.advancements.critereon.SerializationContext serializer) {
-        if (predicates.length == 0) {
+    public static com.google.gson.JsonElement toJson(net.minecraft.advancements.critereon.EntityPredicate$Composite[] $$0, net.minecraft.advancements.critereon.SerializationContext $$1) {
+        if ($$0.length == 0) {
             return com.google.gson.JsonNull.INSTANCE;
         }
-        com.google.gson.JsonArray jsonArray = new com.google.gson.JsonArray();
-        for (net.minecraft.advancements.critereon.EntityPredicate$Composite composite : predicates) {
-            jsonArray.add(composite.toJson(serializer));
+        com.google.gson.JsonArray $$2 = new com.google.gson.JsonArray();
+        for (net.minecraft.advancements.critereon.EntityPredicate$Composite $$3 : $$0) {
+            $$2.add($$3.toJson($$1));
         }
-        return jsonArray;
+        return $$2;
     }
 
     static {