--- net/minecraft/advancements/critereon/EntityTypePredicate$1.java
+++ net/minecraft/advancements/critereon/EntityTypePredicate$1.java
@@ -9,7 +9,7 @@
     public net.minecraft.advancements.critereon.EntityTypePredicate$1() {
     }
 
-    public boolean matches(net.minecraft.world.entity.EntityType<?> type) {
+    public boolean matches(net.minecraft.world.entity.EntityType<?> $$0) {
         return true;
     }
 