--- net/minecraft/advancements/critereon/ImpossibleTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/ImpossibleTrigger$TriggerInstance.java
@@ -13,7 +13,7 @@
         return net.minecraft.advancements.critereon.ImpossibleTrigger.ID;
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
         return new com.google.gson.JsonObject();
     }
 }