--- net/minecraft/advancements/critereon/ImpossibleTrigger.java
+++ net/minecraft/advancements/critereon/ImpossibleTrigger.java
@@ -15,16 +15,16 @@
         return ID;
     }
 
-    public void addPlayerListener(net.minecraft.server.PlayerAdvancements playerAdvancements, net.minecraft.advancements.CriterionTrigger$Listener<net.minecraft.advancements.critereon.ImpossibleTrigger$TriggerInstance> listener) {
+    public void addPlayerListener(net.minecraft.server.PlayerAdvancements $$0, net.minecraft.advancements.CriterionTrigger$Listener<net.minecraft.advancements.critereon.ImpossibleTrigger$TriggerInstance> $$1) {
     }
 
-    public void removePlayerListener(net.minecraft.server.PlayerAdvancements playerAdvancements, net.minecraft.advancements.CriterionTrigger$Listener<net.minecraft.advancements.critereon.ImpossibleTrigger$TriggerInstance> listener) {
+    public void removePlayerListener(net.minecraft.server.PlayerAdvancements $$0, net.minecraft.advancements.CriterionTrigger$Listener<net.minecraft.advancements.critereon.ImpossibleTrigger$TriggerInstance> $$1) {
     }
 
-    public void removePlayerListeners(net.minecraft.server.PlayerAdvancements playerAdvancements) {
+    public void removePlayerListeners(net.minecraft.server.PlayerAdvancements $$0) {
     }
 
-    public net.minecraft.advancements.critereon.ImpossibleTrigger$TriggerInstance createInstance(com.google.gson.JsonObject object, net.minecraft.advancements.critereon.DeserializationContext conditions) {
+    public net.minecraft.advancements.critereon.ImpossibleTrigger$TriggerInstance createInstance(com.google.gson.JsonObject $$0, net.minecraft.advancements.critereon.DeserializationContext $$1) {
         return new net.minecraft.advancements.critereon.ImpossibleTrigger$TriggerInstance();
     }
 