--- net/minecraft/advancements/critereon/ItemPredicate$Builder.java
+++ net/minecraft/advancements/critereon/ItemPredicate$Builder.java
@@ -54,13 +54,13 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.ItemPredicate$Builder hasNbt(net.minecraft.nbt.CompoundTag nbt) {
-        this.nbt = new net.minecraft.advancements.critereon.NbtPredicate(nbt);
+    public net.minecraft.advancements.critereon.ItemPredicate$Builder hasNbt(net.minecraft.nbt.CompoundTag $$0) {
+        this.nbt = new net.minecraft.advancements.critereon.NbtPredicate($$0);
         return this;
     }
 
-    public net.minecraft.advancements.critereon.ItemPredicate$Builder hasEnchantment(net.minecraft.advancements.critereon.EnchantmentPredicate enchantmentCondition) {
-        this.enchantments.add((java.lang.Object)enchantmentCondition);
+    public net.minecraft.advancements.critereon.ItemPredicate$Builder hasEnchantment(net.minecraft.advancements.critereon.EnchantmentPredicate $$0) {
+        this.enchantments.add((java.lang.Object)$$0);
         return this;
     }
 