--- net/minecraft/advancements/critereon/ItemDurabilityTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/ItemDurabilityTrigger$TriggerInstance.java
@@ -10,37 +10,37 @@
     public net.minecraft.advancements.critereon.MinMaxBounds$Ints durability;
     public net.minecraft.advancements.critereon.MinMaxBounds$Ints delta;
 
-    public net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite player, net.minecraft.advancements.critereon.ItemPredicate item, net.minecraft.advancements.critereon.MinMaxBounds$Ints durability, net.minecraft.advancements.critereon.MinMaxBounds$Ints delta) {
-        super(net.minecraft.advancements.critereon.ItemDurabilityTrigger.ID, player);
-        this.item = item;
-        this.durability = durability;
-        this.delta = delta;
+    public net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite $$0, net.minecraft.advancements.critereon.ItemPredicate $$1, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$2, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$3) {
+        super(net.minecraft.advancements.critereon.ItemDurabilityTrigger.ID, $$0);
+        this.item = $$1;
+        this.durability = $$2;
+        this.delta = $$3;
     }
 
     public static net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance changedDurability(net.minecraft.advancements.critereon.ItemPredicate $$0, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$1) {
         return net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance.changedDurability((net.minecraft.advancements.critereon.EntityPredicate$Composite)net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, (net.minecraft.advancements.critereon.ItemPredicate)$$0, (net.minecraft.advancements.critereon.MinMaxBounds$Ints)$$1);
     }
 
-    public static net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance changedDurability(net.minecraft.advancements.critereon.EntityPredicate$Composite player, net.minecraft.advancements.critereon.ItemPredicate item, net.minecraft.advancements.critereon.MinMaxBounds$Ints durability) {
-        return new net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance(player, item, durability, net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY);
+    public static net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance changedDurability(net.minecraft.advancements.critereon.EntityPredicate$Composite $$0, net.minecraft.advancements.critereon.ItemPredicate $$1, net.minecraft.advancements.critereon.MinMaxBounds$Ints $$2) {
+        return new net.minecraft.advancements.critereon.ItemDurabilityTrigger$TriggerInstance($$0, $$1, $$2, net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY);
     }
 
-    public boolean matches(net.minecraft.world.item.ItemStack item, int durability) {
-        if (!this.item.matches(item)) {
+    public boolean matches(net.minecraft.world.item.ItemStack $$0, int $$1) {
+        if (!this.item.matches($$0)) {
             return false;
         }
-        if (!this.durability.matches(item.getMaxDamage() - durability)) {
+        if (!this.durability.matches($$0.getMaxDamage() - $$1)) {
             return false;
         }
-        return this.delta.matches(item.getDamageValue() - durability);
+        return this.delta.matches($$0.getDamageValue() - $$1);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("item", this.item.serializeToJson());
-        jsonObject.add("durability", this.durability.serializeToJson());
-        jsonObject.add("delta", this.delta.serializeToJson());
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("item", this.item.serializeToJson());
+        $$1.add("durability", this.durability.serializeToJson());
+        $$1.add("delta", this.delta.serializeToJson());
+        return $$1;
     }
 }
 