--- net/minecraft/advancements/critereon/KilledTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/KilledTrigger$TriggerInstance.java
@@ -19,8 +19,8 @@
         return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(net.minecraft.advancements.CriteriaTriggers.PLAYER_KILLED_ENTITY.id, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$0), net.minecraft.advancements.critereon.DamageSourcePredicate.ANY);
     }
 
-    public static net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance playerKilledEntity(net.minecraft.advancements.critereon.EntityPredicate$Builder builder) {
-        return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(net.minecraft.advancements.CriteriaTriggers.PLAYER_KILLED_ENTITY.id, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)builder.build()), net.minecraft.advancements.critereon.DamageSourcePredicate.ANY);
+    public static net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance playerKilledEntity(net.minecraft.advancements.critereon.EntityPredicate$Builder $$0) {
+        return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(net.minecraft.advancements.CriteriaTriggers.PLAYER_KILLED_ENTITY.id, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$0.build()), net.minecraft.advancements.critereon.DamageSourcePredicate.ANY);
     }
 
     public static net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance playerKilledEntity() {
@@ -39,8 +39,8 @@
         return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(net.minecraft.advancements.CriteriaTriggers.PLAYER_KILLED_ENTITY.id, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$0), $$1.build());
     }
 
-    public static net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance playerKilledEntity(net.minecraft.advancements.critereon.EntityPredicate$Builder entityBuilder, net.minecraft.advancements.critereon.DamageSourcePredicate$Builder sourceBuilder) {
-        return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(net.minecraft.advancements.CriteriaTriggers.PLAYER_KILLED_ENTITY.id, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)entityBuilder.build()), sourceBuilder.build());
+    public static net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance playerKilledEntity(net.minecraft.advancements.critereon.EntityPredicate$Builder $$0, net.minecraft.advancements.critereon.DamageSourcePredicate$Builder $$1) {
+        return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(net.minecraft.advancements.CriteriaTriggers.PLAYER_KILLED_ENTITY.id, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$0.build()), $$1.build());
     }
 
     public static net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance playerKilledEntityNearSculkCatalyst() {
@@ -75,18 +75,18 @@
         return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(net.minecraft.advancements.CriteriaTriggers.ENTITY_KILLED_PLAYER.id, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$0.build()), $$1.build());
     }
 
-    public boolean matches(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.level.storage.loot.LootContext context, net.minecraft.world.damagesource.DamageSource source) {
-        if (!this.killingBlow.matches(player, source)) {
+    public boolean matches(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.level.storage.loot.LootContext $$1, net.minecraft.world.damagesource.DamageSource $$2) {
+        if (!this.killingBlow.matches($$0, $$2)) {
             return false;
         }
-        return this.entityPredicate.matches(context);
+        return this.entityPredicate.matches($$1);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("entity", this.entityPredicate.toJson(conditions));
-        jsonObject.add("killing_blow", this.killingBlow.serializeToJson());
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("entity", this.entityPredicate.toJson($$0));
+        $$1.add("killing_blow", this.killingBlow.serializeToJson());
+        return $$1;
     }
 }
 