--- net/minecraft/advancements/critereon/KilledTrigger.java
+++ net/minecraft/advancements/critereon/KilledTrigger.java
@@ -16,13 +16,13 @@
         return this.id;
     }
 
-    public net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance createInstance(com.google.gson.JsonObject json, net.minecraft.advancements.critereon.EntityPredicate$Composite entityPredicate, net.minecraft.advancements.critereon.DeserializationContext conditionsParser) {
-        return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(this.id, entityPredicate, net.minecraft.advancements.critereon.EntityPredicate$Composite.fromJson((com.google.gson.JsonObject)json, (java.lang.String)"entity", (net.minecraft.advancements.critereon.DeserializationContext)conditionsParser), net.minecraft.advancements.critereon.DamageSourcePredicate.fromJson((com.google.gson.JsonElement)json.get("killing_blow")));
+    public net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance createInstance(com.google.gson.JsonObject $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        return new net.minecraft.advancements.critereon.KilledTrigger$TriggerInstance(this.id, $$1, net.minecraft.advancements.critereon.EntityPredicate$Composite.fromJson((com.google.gson.JsonObject)$$0, (java.lang.String)"entity", (net.minecraft.advancements.critereon.DeserializationContext)$$2), net.minecraft.advancements.critereon.DamageSourcePredicate.fromJson((com.google.gson.JsonElement)$$0.get("killing_blow")));
     }
 
-    public void trigger(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.entity.Entity entity, net.minecraft.world.damagesource.DamageSource source) {
-        net.minecraft.world.level.storage.loot.LootContext lootContext = net.minecraft.advancements.critereon.EntityPredicate.createContext((net.minecraft.server.level.ServerPlayer)player, (net.minecraft.world.entity.Entity)entity);
-        this.trigger(player, $$3 -> $$3.matches(player, lootContext, source));
+    public void trigger(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.entity.Entity $$1, net.minecraft.world.damagesource.DamageSource $$2) {
+        net.minecraft.world.level.storage.loot.LootContext $$32 = net.minecraft.advancements.critereon.EntityPredicate.createContext((net.minecraft.server.level.ServerPlayer)$$0, (net.minecraft.world.entity.Entity)$$1);
+        this.trigger($$0, $$3 -> $$3.matches($$0, $$32, $$2));
     }
 
     public /* synthetic */ net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance createInstance(com.google.gson.JsonObject jsonObject, net.minecraft.advancements.critereon.EntityPredicate$Composite composite, net.minecraft.advancements.critereon.DeserializationContext deserializationContext) {