--- net/minecraft/advancements/critereon/KilledByCrossbowTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/KilledByCrossbowTrigger$TriggerInstance.java
@@ -15,45 +15,45 @@
         this.uniqueEntityTypes = $$2;
     }
 
-    public static net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance crossbowKilled(net.minecraft.advancements.critereon.EntityPredicate$Builder ... builders) {
-        net.minecraft.advancements.critereon.EntityPredicate$Composite[] compositeArray = new net.minecraft.advancements.critereon.EntityPredicate$Composite[builders.length];
-        for (int i = 0; i < builders.length; ++i) {
-            net.minecraft.advancements.critereon.EntityPredicate$Builder builder = builders[i];
-            compositeArray[i] = net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)builder.build());
+    public static net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance crossbowKilled(net.minecraft.advancements.critereon.EntityPredicate$Builder ... $$0) {
+        net.minecraft.advancements.critereon.EntityPredicate$Composite[] $$1 = new net.minecraft.advancements.critereon.EntityPredicate$Composite[$$0.length];
+        for (int $$2 = 0; $$2 < $$0.length; ++$$2) {
+            net.minecraft.advancements.critereon.EntityPredicate$Builder $$3 = $$0[$$2];
+            $$1[$$2] = net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$3.build());
         }
-        return new net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, compositeArray, net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY);
+        return new net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$1, net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY);
     }
 
-    public static net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance crossbowKilled(net.minecraft.advancements.critereon.MinMaxBounds$Ints bounds) {
-        net.minecraft.advancements.critereon.EntityPredicate$Composite[] compositeArray = new net.minecraft.advancements.critereon.EntityPredicate$Composite[]{};
-        return new net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, compositeArray, bounds);
+    public static net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance crossbowKilled(net.minecraft.advancements.critereon.MinMaxBounds$Ints $$0) {
+        net.minecraft.advancements.critereon.EntityPredicate$Composite[] $$1 = new net.minecraft.advancements.critereon.EntityPredicate$Composite[]{};
+        return new net.minecraft.advancements.critereon.KilledByCrossbowTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$1, $$0);
     }
 
-    public boolean matches(java.util.Collection<net.minecraft.world.level.storage.loot.LootContext> contexts, int bounds) {
+    public boolean matches(java.util.Collection<net.minecraft.world.level.storage.loot.LootContext> $$0, int $$1) {
         if (this.victims.length > 0) {
-            java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList(contexts);
-            for (net.minecraft.advancements.critereon.EntityPredicate$Composite composite : this.victims) {
-                boolean bl = false;
-                java.util.Iterator iterator = arrayList.iterator();
-                while (iterator.hasNext()) {
-                    net.minecraft.world.level.storage.loot.LootContext lootContext = (net.minecraft.world.level.storage.loot.LootContext)iterator.next();
-                    if (!composite.matches(lootContext)) continue;
-                    iterator.remove();
-                    bl = true;
+            java.util.ArrayList $$2 = com.google.common.collect.Lists.newArrayList($$0);
+            for (net.minecraft.advancements.critereon.EntityPredicate$Composite $$3 : this.victims) {
+                boolean $$4 = false;
+                java.util.Iterator $$5 = $$2.iterator();
+                while ($$5.hasNext()) {
+                    net.minecraft.world.level.storage.loot.LootContext $$6 = (net.minecraft.world.level.storage.loot.LootContext)$$5.next();
+                    if (!$$3.matches($$6)) continue;
+                    $$5.remove();
+                    $$4 = true;
                     break;
                 }
-                if (bl) continue;
+                if ($$4) continue;
                 return false;
             }
         }
-        return this.uniqueEntityTypes.matches(bounds);
+        return this.uniqueEntityTypes.matches($$1);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("victims", net.minecraft.advancements.critereon.EntityPredicate$Composite.toJson((net.minecraft.advancements.critereon.EntityPredicate$Composite[])this.victims, (net.minecraft.advancements.critereon.SerializationContext)conditions));
-        jsonObject.add("unique_entity_types", this.uniqueEntityTypes.serializeToJson());
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("victims", net.minecraft.advancements.critereon.EntityPredicate$Composite.toJson((net.minecraft.advancements.critereon.EntityPredicate$Composite[])this.victims, (net.minecraft.advancements.critereon.SerializationContext)$$0));
+        $$1.add("unique_entity_types", this.uniqueEntityTypes.serializeToJson());
+        return $$1;
     }
 }
 