--- net/minecraft/advancements/critereon/LevitationTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/LevitationTrigger$TriggerInstance.java
@@ -15,22 +15,22 @@
         this.duration = $$2;
     }
 
-    public static net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance levitated(net.minecraft.advancements.critereon.DistancePredicate distance) {
-        return new net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, distance, net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY);
+    public static net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance levitated(net.minecraft.advancements.critereon.DistancePredicate $$0) {
+        return new net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0, net.minecraft.advancements.critereon.MinMaxBounds$Ints.ANY);
     }
 
-    public boolean matches(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.phys.Vec3 startPos, int duration) {
-        if (!this.distance.matches(startPos.x, startPos.y, startPos.z, player.getX(), player.getY(), player.getZ())) {
+    public boolean matches(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.phys.Vec3 $$1, int $$2) {
+        if (!this.distance.matches($$1.x, $$1.y, $$1.z, $$0.getX(), $$0.getY(), $$0.getZ())) {
             return false;
         }
-        return this.duration.matches(duration);
+        return this.duration.matches($$2);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("distance", this.distance.serializeToJson());
-        jsonObject.add("duration", this.duration.serializeToJson());
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("distance", this.distance.serializeToJson());
+        $$1.add("duration", this.duration.serializeToJson());
+        return $$1;
     }
 }
 