--- net/minecraft/advancements/critereon/LevitationTrigger.java
+++ net/minecraft/advancements/critereon/LevitationTrigger.java
@@ -15,14 +15,14 @@
         return ID;
     }
 
-    public net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance createInstance(com.google.gson.JsonObject json, net.minecraft.advancements.critereon.EntityPredicate$Composite entityPredicate, net.minecraft.advancements.critereon.DeserializationContext conditionsParser) {
-        net.minecraft.advancements.critereon.DistancePredicate distancePredicate = net.minecraft.advancements.critereon.DistancePredicate.fromJson((com.google.gson.JsonElement)json.get("distance"));
-        net.minecraft.advancements.critereon.MinMaxBounds$Ints ints = net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromJson((com.google.gson.JsonElement)json.get("duration"));
-        return new net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance(entityPredicate, distancePredicate, ints);
+    public net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance createInstance(com.google.gson.JsonObject $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        net.minecraft.advancements.critereon.DistancePredicate $$3 = net.minecraft.advancements.critereon.DistancePredicate.fromJson((com.google.gson.JsonElement)$$0.get("distance"));
+        net.minecraft.advancements.critereon.MinMaxBounds$Ints $$4 = net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromJson((com.google.gson.JsonElement)$$0.get("duration"));
+        return new net.minecraft.advancements.critereon.LevitationTrigger$TriggerInstance($$1, $$3, $$4);
     }
 
-    public void trigger(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.phys.Vec3 startPos, int duration) {
-        this.trigger(player, $$3 -> $$3.matches(player, startPos, duration));
+    public void trigger(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.phys.Vec3 $$1, int $$2) {
+        this.trigger($$0, $$3 -> $$3.matches($$0, $$1, $$2));
     }
 
     public /* synthetic */ net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance createInstance(com.google.gson.JsonObject jsonObject, net.minecraft.advancements.critereon.EntityPredicate$Composite composite, net.minecraft.advancements.critereon.DeserializationContext deserializationContext) {