--- net/minecraft/advancements/critereon/LightPredicate.java
+++ net/minecraft/advancements/critereon/LightPredicate.java
@@ -12,14 +12,14 @@
         this.composite = $$0;
     }
 
-    public boolean matches(net.minecraft.server.level.ServerLevel level, net.minecraft.core.BlockPos pos) {
+    public boolean matches(net.minecraft.server.level.ServerLevel $$0, net.minecraft.core.BlockPos $$1) {
         if (this == ANY) {
             return true;
         }
-        if (!level.isLoaded(pos)) {
+        if (!$$0.isLoaded($$1)) {
             return false;
         }
-        return this.composite.matches(level.getMaxLocalRawBrightness(pos));
+        return this.composite.matches($$0.getMaxLocalRawBrightness($$1));
     }
 
     public com.google.gson.JsonElement serializeToJson() {
@@ -31,13 +31,13 @@
         return $$0;
     }
 
-    public static net.minecraft.advancements.critereon.LightPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement element) {
-        if (element == null || element.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.LightPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement $$0) {
+        if ($$0 == null || $$0.isJsonNull()) {
             return ANY;
         }
-        com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)element, (java.lang.String)"light");
-        net.minecraft.advancements.critereon.MinMaxBounds$Ints ints = net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromJson((com.google.gson.JsonElement)jsonObject.get("light"));
-        return new net.minecraft.advancements.critereon.LightPredicate(ints);
+        com.google.gson.JsonObject $$1 = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$0, (java.lang.String)"light");
+        net.minecraft.advancements.critereon.MinMaxBounds$Ints $$2 = net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromJson((com.google.gson.JsonElement)$$1.get("light"));
+        return new net.minecraft.advancements.critereon.LightPredicate($$2);
     }
 
     static {