--- net/minecraft/advancements/critereon/LocationPredicate$Builder.java
+++ net/minecraft/advancements/critereon/LocationPredicate$Builder.java
@@ -48,8 +48,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.LocationPredicate$Builder setBiome(@javax.annotation.Nullable net.minecraft.resources.ResourceKey<net.minecraft.world.level.biome.Biome> biome) {
-        this.biome = biome;
+    public net.minecraft.advancements.critereon.LocationPredicate$Builder setBiome(@javax.annotation.Nullable net.minecraft.resources.ResourceKey<net.minecraft.world.level.biome.Biome> $$0) {
+        this.biome = $$0;
         return this;
     }
 
@@ -68,8 +68,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.LocationPredicate$Builder setBlock(net.minecraft.advancements.critereon.BlockPredicate block) {
-        this.block = block;
+    public net.minecraft.advancements.critereon.LocationPredicate$Builder setBlock(net.minecraft.advancements.critereon.BlockPredicate $$0) {
+        this.block = $$0;
         return this;
     }
 
@@ -78,8 +78,8 @@
         return this;
     }
 
-    public net.minecraft.advancements.critereon.LocationPredicate$Builder setSmokey(java.lang.Boolean smokey) {
-        this.smokey = smokey;
+    public net.minecraft.advancements.critereon.LocationPredicate$Builder setSmokey(java.lang.Boolean $$0) {
+        this.smokey = $$0;
         return this;
     }
 