--- net/minecraft/advancements/critereon/MobEffectsPredicate.java
+++ net/minecraft/advancements/critereon/MobEffectsPredicate.java
@@ -16,8 +16,8 @@
         return new net.minecraft.advancements.critereon.MobEffectsPredicate((java.util.Map)com.google.common.collect.Maps.newLinkedHashMap());
     }
 
-    public net.minecraft.advancements.critereon.MobEffectsPredicate and(net.minecraft.world.effect.MobEffect effect) {
-        this.effects.put((java.lang.Object)effect, (java.lang.Object)new net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate());
+    public net.minecraft.advancements.critereon.MobEffectsPredicate and(net.minecraft.world.effect.MobEffect $$0) {
+        this.effects.put((java.lang.Object)$$0, (java.lang.Object)new net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate());
         return this;
     }
 
@@ -26,48 +26,48 @@
         return this;
     }
 
-    public boolean matches(net.minecraft.world.entity.Entity entity) {
+    public boolean matches(net.minecraft.world.entity.Entity $$0) {
         if (this == ANY) {
             return true;
         }
-        if (entity instanceof net.minecraft.world.entity.LivingEntity) {
-            return this.matches(((net.minecraft.world.entity.LivingEntity)entity).getActiveEffectsMap());
+        if ($$0 instanceof net.minecraft.world.entity.LivingEntity) {
+            return this.matches(((net.minecraft.world.entity.LivingEntity)$$0).getActiveEffectsMap());
         }
         return false;
     }
 
-    public boolean matches(net.minecraft.world.entity.LivingEntity entity) {
+    public boolean matches(net.minecraft.world.entity.LivingEntity $$0) {
         if (this == ANY) {
             return true;
         }
-        return this.matches(entity.getActiveEffectsMap());
+        return this.matches($$0.getActiveEffectsMap());
     }
 
-    public boolean matches(java.util.Map<net.minecraft.world.effect.MobEffect, net.minecraft.world.effect.MobEffectInstance> potions) {
+    public boolean matches(java.util.Map<net.minecraft.world.effect.MobEffect, net.minecraft.world.effect.MobEffectInstance> $$0) {
         if (this == ANY) {
             return true;
         }
-        for (java.util.Map$Entry entry : this.effects.entrySet()) {
-            net.minecraft.world.effect.MobEffectInstance mobEffectInstance = (net.minecraft.world.effect.MobEffectInstance)potions.get(entry.getKey());
-            if (((net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate)entry.getValue()).matches(mobEffectInstance)) continue;
+        for (java.util.Map$Entry $$1 : this.effects.entrySet()) {
+            net.minecraft.world.effect.MobEffectInstance $$2 = (net.minecraft.world.effect.MobEffectInstance)$$0.get($$1.getKey());
+            if (((net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate)$$1.getValue()).matches($$2)) continue;
             return false;
         }
         return true;
     }
 
-    public static net.minecraft.advancements.critereon.MobEffectsPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement element) {
-        if (element == null || element.isJsonNull()) {
+    public static net.minecraft.advancements.critereon.MobEffectsPredicate fromJson(@javax.annotation.Nullable com.google.gson.JsonElement $$0) {
+        if ($$0 == null || $$0.isJsonNull()) {
             return ANY;
         }
-        com.google.gson.JsonObject jsonObject = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)element, (java.lang.String)"effects");
-        java.util.LinkedHashMap linkedHashMap = com.google.common.collect.Maps.newLinkedHashMap();
-        for (java.util.Map$Entry entry : jsonObject.entrySet()) {
-            net.minecraft.resources.ResourceLocation resourceLocation = new net.minecraft.resources.ResourceLocation((java.lang.String)entry.getKey());
-            net.minecraft.world.effect.MobEffect mobEffect = (net.minecraft.world.effect.MobEffect)net.minecraft.core.Registry.MOB_EFFECT.getOptional(resourceLocation).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown effect '" + resourceLocation + "'"));
-            net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate mobEffectInstancePredicate = net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate.fromJson((com.google.gson.JsonObject)net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)((com.google.gson.JsonElement)entry.getValue()), (java.lang.String)((java.lang.String)entry.getKey())));
-            linkedHashMap.put((java.lang.Object)mobEffect, (java.lang.Object)mobEffectInstancePredicate);
+        com.google.gson.JsonObject $$1 = net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)$$0, (java.lang.String)"effects");
+        java.util.LinkedHashMap $$2 = com.google.common.collect.Maps.newLinkedHashMap();
+        for (java.util.Map$Entry $$3 : $$1.entrySet()) {
+            net.minecraft.resources.ResourceLocation $$4 = new net.minecraft.resources.ResourceLocation((java.lang.String)$$3.getKey());
+            net.minecraft.world.effect.MobEffect $$5 = (net.minecraft.world.effect.MobEffect)net.minecraft.core.Registry.MOB_EFFECT.getOptional($$4).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown effect '" + $$4 + "'"));
+            net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate $$6 = net.minecraft.advancements.critereon.MobEffectsPredicate$MobEffectInstancePredicate.fromJson((com.google.gson.JsonObject)net.minecraft.util.GsonHelper.convertToJsonObject((com.google.gson.JsonElement)((com.google.gson.JsonElement)$$3.getValue()), (java.lang.String)((java.lang.String)$$3.getKey())));
+            $$2.put((java.lang.Object)$$5, (java.lang.Object)$$6);
         }
-        return new net.minecraft.advancements.critereon.MobEffectsPredicate((java.util.Map)linkedHashMap);
+        return new net.minecraft.advancements.critereon.MobEffectsPredicate((java.util.Map)$$2);
     }
 
     public com.google.gson.JsonElement serializeToJson() {