--- net/minecraft/advancements/critereon/PlacedBlockTrigger.java
+++ net/minecraft/advancements/critereon/PlacedBlockTrigger.java
@@ -15,31 +15,31 @@
         return ID;
     }
 
-    public net.minecraft.advancements.critereon.PlacedBlockTrigger$TriggerInstance createInstance(com.google.gson.JsonObject json, net.minecraft.advancements.critereon.EntityPredicate$Composite entityPredicate, net.minecraft.advancements.critereon.DeserializationContext conditionsParser) {
-        net.minecraft.world.level.block.Block block = net.minecraft.advancements.critereon.PlacedBlockTrigger.deserializeBlock((com.google.gson.JsonObject)json);
-        net.minecraft.advancements.critereon.StatePropertiesPredicate statePropertiesPredicate = net.minecraft.advancements.critereon.StatePropertiesPredicate.fromJson((com.google.gson.JsonElement)json.get("state"));
-        if (block != null) {
-            statePropertiesPredicate.checkState(block.getStateDefinition(), $$1 -> {
-                throw new com.google.gson.JsonSyntaxException("Block " + block + " has no property " + $$1 + ":");
+    public net.minecraft.advancements.critereon.PlacedBlockTrigger$TriggerInstance createInstance(com.google.gson.JsonObject $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$12, net.minecraft.advancements.critereon.DeserializationContext $$2) {
+        net.minecraft.world.level.block.Block $$3 = net.minecraft.advancements.critereon.PlacedBlockTrigger.deserializeBlock((com.google.gson.JsonObject)$$0);
+        net.minecraft.advancements.critereon.StatePropertiesPredicate $$4 = net.minecraft.advancements.critereon.StatePropertiesPredicate.fromJson((com.google.gson.JsonElement)$$0.get("state"));
+        if ($$3 != null) {
+            $$4.checkState($$3.getStateDefinition(), $$1 -> {
+                throw new com.google.gson.JsonSyntaxException("Block " + $$3 + " has no property " + $$1 + ":");
             });
         }
-        net.minecraft.advancements.critereon.LocationPredicate locationPredicate = net.minecraft.advancements.critereon.LocationPredicate.fromJson((com.google.gson.JsonElement)json.get("location"));
-        net.minecraft.advancements.critereon.ItemPredicate itemPredicate = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)json.get("item"));
-        return new net.minecraft.advancements.critereon.PlacedBlockTrigger$TriggerInstance(entityPredicate, block, statePropertiesPredicate, locationPredicate, itemPredicate);
+        net.minecraft.advancements.critereon.LocationPredicate $$5 = net.minecraft.advancements.critereon.LocationPredicate.fromJson((com.google.gson.JsonElement)$$0.get("location"));
+        net.minecraft.advancements.critereon.ItemPredicate $$6 = net.minecraft.advancements.critereon.ItemPredicate.fromJson((com.google.gson.JsonElement)$$0.get("item"));
+        return new net.minecraft.advancements.critereon.PlacedBlockTrigger$TriggerInstance($$12, $$3, $$4, $$5, $$6);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.level.block.Block deserializeBlock(com.google.gson.JsonObject object) {
-        if (object.has("block")) {
-            net.minecraft.resources.ResourceLocation resourceLocation = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)object, (java.lang.String)"block"));
-            return (net.minecraft.world.level.block.Block)net.minecraft.core.Registry.BLOCK.getOptional(resourceLocation).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown block type '" + resourceLocation + "'"));
+    public static net.minecraft.world.level.block.Block deserializeBlock(com.google.gson.JsonObject $$0) {
+        if ($$0.has("block")) {
+            net.minecraft.resources.ResourceLocation $$1 = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"block"));
+            return (net.minecraft.world.level.block.Block)net.minecraft.core.Registry.BLOCK.getOptional($$1).orElseThrow(() -> new com.google.gson.JsonSyntaxException("Unknown block type '" + $$1 + "'"));
         }
         return null;
     }
 
-    public void trigger(net.minecraft.server.level.ServerPlayer player, net.minecraft.core.BlockPos pos, net.minecraft.world.item.ItemStack item) {
-        net.minecraft.world.level.block.state.BlockState blockState = player.getLevel().getBlockState(pos);
-        this.trigger(player, $$4 -> $$4.matches(blockState, pos, player.getLevel(), item));
+    public void trigger(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.item.ItemStack $$2) {
+        net.minecraft.world.level.block.state.BlockState $$3 = $$0.getLevel().getBlockState($$1);
+        this.trigger($$0, $$4 -> $$4.matches($$3, $$1, $$0.getLevel(), $$2));
     }
 
     public /* synthetic */ net.minecraft.advancements.critereon.AbstractCriterionTriggerInstance createInstance(com.google.gson.JsonObject jsonObject, net.minecraft.advancements.critereon.EntityPredicate$Composite composite, net.minecraft.advancements.critereon.DeserializationContext deserializationContext) {