--- net/minecraft/advancements/critereon/PlayerHurtEntityTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/PlayerHurtEntityTrigger$TriggerInstance.java
@@ -23,8 +23,8 @@
         return new net.minecraft.advancements.critereon.PlayerHurtEntityTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY);
     }
 
-    public static net.minecraft.advancements.critereon.PlayerHurtEntityTrigger$TriggerInstance playerHurtEntity(net.minecraft.advancements.critereon.DamagePredicate$Builder builder) {
-        return new net.minecraft.advancements.critereon.PlayerHurtEntityTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, builder.build(), net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY);
+    public static net.minecraft.advancements.critereon.PlayerHurtEntityTrigger$TriggerInstance playerHurtEntity(net.minecraft.advancements.critereon.DamagePredicate$Builder $$0) {
+        return new net.minecraft.advancements.critereon.PlayerHurtEntityTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0.build(), net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY);
     }
 
     public static net.minecraft.advancements.critereon.PlayerHurtEntityTrigger$TriggerInstance playerHurtEntity(net.minecraft.advancements.critereon.EntityPredicate $$0) {
@@ -39,18 +39,18 @@
         return new net.minecraft.advancements.critereon.PlayerHurtEntityTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0.build(), net.minecraft.advancements.critereon.EntityPredicate$Composite.wrap((net.minecraft.advancements.critereon.EntityPredicate)$$1));
     }
 
-    public boolean matches(net.minecraft.server.level.ServerPlayer player, net.minecraft.world.level.storage.loot.LootContext context, net.minecraft.world.damagesource.DamageSource damage, float dealt, float taken, boolean blocked) {
-        if (!this.damage.matches(player, damage, dealt, taken, blocked)) {
+    public boolean matches(net.minecraft.server.level.ServerPlayer $$0, net.minecraft.world.level.storage.loot.LootContext $$1, net.minecraft.world.damagesource.DamageSource $$2, float $$3, float $$4, boolean $$5) {
+        if (!this.damage.matches($$0, $$2, $$3, $$4, $$5)) {
             return false;
         }
-        return this.entity.matches(context);
+        return this.entity.matches($$1);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("damage", this.damage.serializeToJson());
-        jsonObject.add("entity", this.entity.toJson(conditions));
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("damage", this.damage.serializeToJson());
+        $$1.add("entity", this.entity.toJson($$0));
+        return $$1;
     }
 }
 