--- net/minecraft/advancements/critereon/SerializationContext.java
+++ net/minecraft/advancements/critereon/SerializationContext.java
@@ -12,8 +12,8 @@
         this.predicateGson = net.minecraft.world.level.storage.loot.Deserializers.createConditionSerializer().create();
     }
 
-    public com.google.gson.JsonElement serializeConditions(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        return this.predicateGson.toJsonTree((java.lang.Object)conditions);
+    public com.google.gson.JsonElement serializeConditions(net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$0) {
+        return this.predicateGson.toJsonTree((java.lang.Object)$$0);
     }
 
     static {