--- net/minecraft/advancements/critereon/StatePropertiesPredicate$Builder.java
+++ net/minecraft/advancements/critereon/StatePropertiesPredicate$Builder.java
@@ -15,21 +15,21 @@
         return new net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder();
     }
 
-    public net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<?> property, java.lang.String value) {
-        this.matchers.add((java.lang.Object)new net.minecraft.advancements.critereon.StatePropertiesPredicate$ExactPropertyMatcher(property.getName(), value));
+    public net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<?> $$0, java.lang.String $$1) {
+        this.matchers.add((java.lang.Object)new net.minecraft.advancements.critereon.StatePropertiesPredicate$ExactPropertyMatcher($$0.getName(), $$1));
         return this;
     }
 
-    public net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<java.lang.Integer> intProp, int value) {
-        return this.hasProperty(intProp, java.lang.Integer.toString((int)value));
+    public net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<java.lang.Integer> $$0, int $$1) {
+        return this.hasProperty($$0, java.lang.Integer.toString((int)$$1));
     }
 
-    public net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<java.lang.Boolean> boolProp, boolean value) {
-        return this.hasProperty(boolProp, java.lang.Boolean.toString((boolean)value));
+    public net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<java.lang.Boolean> $$0, boolean $$1) {
+        return this.hasProperty($$0, java.lang.Boolean.toString((boolean)$$1));
     }
 
-    public <T extends java.lang.Comparable<T> & net.minecraft.util.StringRepresentable> net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<T> prop, T value) {
-        return this.hasProperty(prop, ((net.minecraft.util.StringRepresentable)value).getSerializedName());
+    public <T extends java.lang.Comparable<T> & net.minecraft.util.StringRepresentable> net.minecraft.advancements.critereon.StatePropertiesPredicate$Builder hasProperty(net.minecraft.world.level.block.state.properties.Property<T> $$0, T $$1) {
+        return this.hasProperty($$0, ((net.minecraft.util.StringRepresentable)$$1).getSerializedName());
     }
 
     public net.minecraft.advancements.critereon.StatePropertiesPredicate build() {