--- net/minecraft/advancements/critereon/TargetBlockTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/TargetBlockTrigger$TriggerInstance.java
@@ -15,22 +15,22 @@
         this.projectile = $$2;
     }
 
-    public static net.minecraft.advancements.critereon.TargetBlockTrigger$TriggerInstance targetHit(net.minecraft.advancements.critereon.MinMaxBounds$Ints signalStrength, net.minecraft.advancements.critereon.EntityPredicate$Composite projectile) {
-        return new net.minecraft.advancements.critereon.TargetBlockTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, signalStrength, projectile);
+    public static net.minecraft.advancements.critereon.TargetBlockTrigger$TriggerInstance targetHit(net.minecraft.advancements.critereon.MinMaxBounds$Ints $$0, net.minecraft.advancements.critereon.EntityPredicate$Composite $$1) {
+        return new net.minecraft.advancements.critereon.TargetBlockTrigger$TriggerInstance(net.minecraft.advancements.critereon.EntityPredicate$Composite.ANY, $$0, $$1);
     }
 
-    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext conditions) {
-        com.google.gson.JsonObject jsonObject = super.serializeToJson(conditions);
-        jsonObject.add("signal_strength", this.signalStrength.serializeToJson());
-        jsonObject.add("projectile", this.projectile.toJson(conditions));
-        return jsonObject;
+    public com.google.gson.JsonObject serializeToJson(net.minecraft.advancements.critereon.SerializationContext $$0) {
+        com.google.gson.JsonObject $$1 = super.serializeToJson($$0);
+        $$1.add("signal_strength", this.signalStrength.serializeToJson());
+        $$1.add("projectile", this.projectile.toJson($$0));
+        return $$1;
     }
 
-    public boolean matches(net.minecraft.world.level.storage.loot.LootContext context, net.minecraft.world.phys.Vec3 vector, int signalStrength) {
-        if (!this.signalStrength.matches(signalStrength)) {
+    public boolean matches(net.minecraft.world.level.storage.loot.LootContext $$0, net.minecraft.world.phys.Vec3 $$1, int $$2) {
+        if (!this.signalStrength.matches($$2)) {
             return false;
         }
-        return this.projectile.matches(context);
+        return this.projectile.matches($$0);
     }
 }
 