--- net/minecraft/advancements/critereon/UsedEnderEyeTrigger$TriggerInstance.java
+++ net/minecraft/advancements/critereon/UsedEnderEyeTrigger$TriggerInstance.java
@@ -13,8 +13,8 @@
         this.level = $$1;
     }
 
-    public boolean matches(double distanceSq) {
-        return this.level.matchesSqr(distanceSq);
+    public boolean matches(double $$0) {
+        return this.level.matchesSqr($$0);
     }
 }
 