--- net/minecraft/world/level/gameevent/DynamicGameEventListener.java
+++ net/minecraft/world/level/gameevent/DynamicGameEventListener.java
@@ -48,13 +48,13 @@
         });
     }
 
-    public static void ifChunkExists(net.minecraft.world.level.LevelReader $$0, @javax.annotation.Nullable net.minecraft.core.SectionPos $$1, java.util.function.Consumer<net.minecraft.world.level.gameevent.GameEventDispatcher> $$2) {
+    public static void ifChunkExists(net.minecraft.world.level.LevelReader $$0, @javax.annotation.Nullable net.minecraft.core.SectionPos $$1, java.util.function.Consumer<net.minecraft.world.level.gameevent.GameEventListenerRegistry> $$2) {
         if ($$1 == null) {
             return;
         }
         net.minecraft.world.level.chunk.ChunkAccess $$3 = $$0.getChunk($$1.x(), $$1.z(), net.minecraft.world.level.chunk.ChunkStatus.FULL, false);
         if ($$3 != null) {
-            $$2.accept((java.lang.Object)$$3.getEventDispatcher($$1.y()));
+            $$2.accept((java.lang.Object)$$3.getListenerRegistry($$1.y()));
         }
     }
 }