--- net/minecraft/world/level/gameevent/GameEventListener$DeliveryMode.java
+++ net/minecraft/world/level/gameevent/GameEventListener$DeliveryMode.java
@@ -1,0 +1,35 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level.gameevent;
+
+public class net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode
+extends java.lang.Enum<net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode>
+ {
+    public static /* enum */ net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode UNSPECIFIED;
+    public static /* enum */ net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode BY_DISTANCE;
+    public static /* synthetic */ net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode[] $VALUES;
+
+    public static net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode[] values() {
+        return (net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode[])$VALUES.clone();
+    }
+
+    public static net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode valueOf(java.lang.String $$0) {
+        return (net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode)java.lang.Enum.valueOf(net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode() {
+        super(string, n);
+    }
+
+    public static /* synthetic */ net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode[] $values() {
+        return new net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode[]{UNSPECIFIED, BY_DISTANCE};
+    }
+
+    static {
+        UNSPECIFIED = new net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode("UNSPECIFIED", 0);
+        BY_DISTANCE = new net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode("BY_DISTANCE", 1);
+        $VALUES = net.minecraft.world.level.gameevent.GameEventListener$DeliveryMode.$values();
+    }
+}
+