--- net/minecraft/world/level/levelgen/flat/FlatLayerInfo.java
+++ net/minecraft/world/level/levelgen/flat/FlatLayerInfo.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.level.block.Block block;
     public int height;
 
-    public net.minecraft.world.level.levelgen.flat.FlatLayerInfo(int height, net.minecraft.world.level.block.Block block) {
-        this.height = height;
-        this.block = block;
+    public net.minecraft.world.level.levelgen.flat.FlatLayerInfo(int $$0, net.minecraft.world.level.block.Block $$1) {
+        this.height = $$0;
+        this.block = $$1;
     }
 
     public int getHeight() {