--- net/minecraft/world/BossEvent.java
+++ net/minecraft/world/BossEvent.java
@@ -14,11 +14,11 @@
     public boolean playBossMusic;
     public boolean createWorldFog;
 
-    public net.minecraft.world.BossEvent(java.util.UUID id, net.minecraft.network.chat.Component name, net.minecraft.world.BossEvent$BossBarColor color, net.minecraft.world.BossEvent$BossBarOverlay overlay) {
-        this.id = id;
-        this.name = name;
-        this.color = color;
-        this.overlay = overlay;
+    public net.minecraft.world.BossEvent(java.util.UUID $$0, net.minecraft.network.chat.Component $$1, net.minecraft.world.BossEvent$BossBarColor $$2, net.minecraft.world.BossEvent$BossBarOverlay $$3) {
+        this.id = $$0;
+        this.name = $$1;
+        this.color = $$2;
+        this.overlay = $$3;
         this.progress = 1.0f;
     }
 
@@ -30,8 +30,8 @@
         return this.name;
     }
 
-    public void setName(net.minecraft.network.chat.Component name) {
-        this.name = name;
+    public void setName(net.minecraft.network.chat.Component $$0) {
+        this.name = $$0;
     }
 
     public float getProgress() {
@@ -46,24 +46,24 @@
         return this.color;
     }
 
-    public void setColor(net.minecraft.world.BossEvent$BossBarColor color) {
-        this.color = color;
+    public void setColor(net.minecraft.world.BossEvent$BossBarColor $$0) {
+        this.color = $$0;
     }
 
     public net.minecraft.world.BossEvent$BossBarOverlay getOverlay() {
         return this.overlay;
     }
 
-    public void setOverlay(net.minecraft.world.BossEvent$BossBarOverlay overlay) {
-        this.overlay = overlay;
+    public void setOverlay(net.minecraft.world.BossEvent$BossBarOverlay $$0) {
+        this.overlay = $$0;
     }
 
     public boolean shouldDarkenScreen() {
         return this.darkenScreen;
     }
 
-    public net.minecraft.world.BossEvent setDarkenScreen(boolean darkenSky) {
-        this.darkenScreen = darkenSky;
+    public net.minecraft.world.BossEvent setDarkenScreen(boolean $$0) {
+        this.darkenScreen = $$0;
         return this;
     }
 
@@ -71,13 +71,13 @@
         return this.playBossMusic;
     }
 
-    public net.minecraft.world.BossEvent setPlayBossMusic(boolean playEndBossMusic) {
-        this.playBossMusic = playEndBossMusic;
+    public net.minecraft.world.BossEvent setPlayBossMusic(boolean $$0) {
+        this.playBossMusic = $$0;
         return this;
     }
 
-    public net.minecraft.world.BossEvent setCreateWorldFog(boolean createFog) {
-        this.createWorldFog = createFog;
+    public net.minecraft.world.BossEvent setCreateWorldFog(boolean $$0) {
+        this.createWorldFog = $$0;
         return this;
     }
 