--- net/minecraft/world/BossEvent$BossBarColor.java
+++ net/minecraft/world/BossEvent$BossBarColor.java
@@ -25,10 +25,10 @@
         return (net.minecraft.world.BossEvent$BossBarColor)java.lang.Enum.valueOf(net.minecraft.world.BossEvent$BossBarColor.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.BossEvent$BossBarColor(java.lang.String name, net.minecraft.ChatFormatting formatting) {
+    public net.minecraft.world.BossEvent$BossBarColor(java.lang.String $$0, net.minecraft.ChatFormatting $$1) {
         super(string, n);
-        this.name = name;
-        this.formatting = formatting;
+        this.name = $$0;
+        this.formatting = $$1;
     }
 
     public net.minecraft.ChatFormatting getFormatting() {
@@ -39,10 +39,10 @@
         return this.name;
     }
 
-    public static net.minecraft.world.BossEvent$BossBarColor byName(java.lang.String name) {
-        for (net.minecraft.world.BossEvent$BossBarColor bossBarColor : net.minecraft.world.BossEvent$BossBarColor.values()) {
-            if (!bossBarColor.name.equals((java.lang.Object)name)) continue;
-            return bossBarColor;
+    public static net.minecraft.world.BossEvent$BossBarColor byName(java.lang.String $$0) {
+        for (net.minecraft.world.BossEvent$BossBarColor $$1 : net.minecraft.world.BossEvent$BossBarColor.values()) {
+            if (!$$1.name.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
         return WHITE;
     }