--- net/minecraft/world/CompoundContainer.java
+++ net/minecraft/world/CompoundContainer.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.Container container1;
     public net.minecraft.world.Container container2;
 
-    public net.minecraft.world.CompoundContainer(net.minecraft.world.Container container1, net.minecraft.world.Container container2) {
-        this.container1 = container1;
-        this.container2 = container2;
+    public net.minecraft.world.CompoundContainer(net.minecraft.world.Container $$0, net.minecraft.world.Container $$1) {
+        this.container1 = $$0;
+        this.container2 = $$1;
     }
 
     public int getContainerSize() {
@@ -22,36 +22,36 @@
         return this.container1.isEmpty() && this.container2.isEmpty();
     }
 
-    public boolean contains(net.minecraft.world.Container inventory) {
-        return this.container1 == inventory || this.container2 == inventory;
+    public boolean contains(net.minecraft.world.Container $$0) {
+        return this.container1 == $$0 || this.container2 == $$0;
     }
 
-    public net.minecraft.world.item.ItemStack getItem(int index) {
-        if (index >= this.container1.getContainerSize()) {
-            return this.container2.getItem(index - this.container1.getContainerSize());
+    public net.minecraft.world.item.ItemStack getItem(int $$0) {
+        if ($$0 >= this.container1.getContainerSize()) {
+            return this.container2.getItem($$0 - this.container1.getContainerSize());
         }
-        return this.container1.getItem(index);
+        return this.container1.getItem($$0);
     }
 
-    public net.minecraft.world.item.ItemStack removeItem(int index, int count) {
-        if (index >= this.container1.getContainerSize()) {
-            return this.container2.removeItem(index - this.container1.getContainerSize(), count);
+    public net.minecraft.world.item.ItemStack removeItem(int $$0, int $$1) {
+        if ($$0 >= this.container1.getContainerSize()) {
+            return this.container2.removeItem($$0 - this.container1.getContainerSize(), $$1);
         }
-        return this.container1.removeItem(index, count);
+        return this.container1.removeItem($$0, $$1);
     }
 
-    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int index) {
-        if (index >= this.container1.getContainerSize()) {
-            return this.container2.removeItemNoUpdate(index - this.container1.getContainerSize());
+    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int $$0) {
+        if ($$0 >= this.container1.getContainerSize()) {
+            return this.container2.removeItemNoUpdate($$0 - this.container1.getContainerSize());
         }
-        return this.container1.removeItemNoUpdate(index);
+        return this.container1.removeItemNoUpdate($$0);
     }
 
-    public void setItem(int index, net.minecraft.world.item.ItemStack stack) {
-        if (index >= this.container1.getContainerSize()) {
-            this.container2.setItem(index - this.container1.getContainerSize(), stack);
+    public void setItem(int $$0, net.minecraft.world.item.ItemStack $$1) {
+        if ($$0 >= this.container1.getContainerSize()) {
+            this.container2.setItem($$0 - this.container1.getContainerSize(), $$1);
         } else {
-            this.container1.setItem(index, stack);
+            this.container1.setItem($$0, $$1);
         }
     }
 
@@ -64,25 +64,25 @@
         this.container2.setChanged();
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
-        return this.container1.stillValid(player) && this.container2.stillValid(player);
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
+        return this.container1.stillValid($$0) && this.container2.stillValid($$0);
     }
 
-    public void startOpen(net.minecraft.world.entity.player.Player player) {
-        this.container1.startOpen(player);
-        this.container2.startOpen(player);
+    public void startOpen(net.minecraft.world.entity.player.Player $$0) {
+        this.container1.startOpen($$0);
+        this.container2.startOpen($$0);
     }
 
-    public void stopOpen(net.minecraft.world.entity.player.Player player) {
-        this.container1.stopOpen(player);
-        this.container2.stopOpen(player);
+    public void stopOpen(net.minecraft.world.entity.player.Player $$0) {
+        this.container1.stopOpen($$0);
+        this.container2.stopOpen($$0);
     }
 
-    public boolean canPlaceItem(int index, net.minecraft.world.item.ItemStack stack) {
-        if (index >= this.container1.getContainerSize()) {
-            return this.container2.canPlaceItem(index - this.container1.getContainerSize(), stack);
+    public boolean canPlaceItem(int $$0, net.minecraft.world.item.ItemStack $$1) {
+        if ($$0 >= this.container1.getContainerSize()) {
+            return this.container2.canPlaceItem($$0 - this.container1.getContainerSize(), $$1);
         }
-        return this.container1.canPlaceItem(index, stack);
+        return this.container1.canPlaceItem($$0, $$1);
     }
 
     public void clearContent() {