--- net/minecraft/world/DifficultyInstance.java
+++ net/minecraft/world/DifficultyInstance.java
@@ -12,9 +12,9 @@
     public net.minecraft.world.Difficulty base;
     public float effectiveDifficulty;
 
-    public net.minecraft.world.DifficultyInstance(net.minecraft.world.Difficulty base, long levelTime, long chunkInhabitedTime, float moonPhaseFactor) {
-        this.base = base;
-        this.effectiveDifficulty = this.calculateDifficulty(base, levelTime, chunkInhabitedTime, moonPhaseFactor);
+    public net.minecraft.world.DifficultyInstance(net.minecraft.world.Difficulty $$0, long $$1, long $$2, float $$3) {
+        this.base = $$0;
+        this.effectiveDifficulty = this.calculateDifficulty($$0, $$1, $$2, $$3);
     }
 
     public net.minecraft.world.Difficulty getDifficulty() {
@@ -29,8 +29,8 @@
         return this.effectiveDifficulty >= (float)net.minecraft.world.Difficulty.HARD.ordinal();
     }
 
-    public boolean isHarderThan(float difficulty) {
-        return this.effectiveDifficulty > difficulty;
+    public boolean isHarderThan(float $$0) {
+        return this.effectiveDifficulty > $$0;
     }
 
     public float getSpecialMultiplier() {
@@ -43,21 +43,21 @@
         return (this.effectiveDifficulty - 2.0f) / 2.0f;
     }
 
-    public float calculateDifficulty(net.minecraft.world.Difficulty difficulty, long levelTime, long chunkInhabitedTime, float moonPhaseFactor) {
-        if (difficulty == net.minecraft.world.Difficulty.PEACEFUL) {
+    public float calculateDifficulty(net.minecraft.world.Difficulty $$0, long $$1, long $$2, float $$3) {
+        if ($$0 == net.minecraft.world.Difficulty.PEACEFUL) {
             return 0.0f;
         }
-        boolean bl = difficulty == net.minecraft.world.Difficulty.HARD;
-        float f = 0.75f;
-        float f2 = net.minecraft.util.Mth.clamp((float)(((float)levelTime + -72000.0f) / 1440000.0f), (float)0.0f, (float)1.0f) * 0.25f;
-        f += f2;
-        float f3 = 0.0f;
-        f3 += net.minecraft.util.Mth.clamp((float)((float)chunkInhabitedTime / 3600000.0f), (float)0.0f, (float)1.0f) * (bl ? 1.0f : 0.75f);
-        f3 += net.minecraft.util.Mth.clamp((float)(moonPhaseFactor * 0.25f), (float)0.0f, (float)f2);
-        if (difficulty == net.minecraft.world.Difficulty.EASY) {
-            f3 *= 0.5f;
+        boolean $$4 = $$0 == net.minecraft.world.Difficulty.HARD;
+        float $$5 = 0.75f;
+        float $$6 = net.minecraft.util.Mth.clamp((float)(((float)$$1 + -72000.0f) / 1440000.0f), (float)0.0f, (float)1.0f) * 0.25f;
+        $$5 += $$6;
+        float $$7 = 0.0f;
+        $$7 += net.minecraft.util.Mth.clamp((float)((float)$$2 / 3600000.0f), (float)0.0f, (float)1.0f) * ($$4 ? 1.0f : 0.75f);
+        $$7 += net.minecraft.util.Mth.clamp((float)($$3 * 0.25f), (float)0.0f, (float)$$6);
+        if ($$0 == net.minecraft.world.Difficulty.EASY) {
+            $$7 *= 0.5f;
         }
-        return (float)difficulty.getId() * (f += f3);
+        return (float)$$0.getId() * ($$5 += $$7);
     }
 }
 