--- net/minecraft/world/InteractionResult.java
+++ net/minecraft/world/InteractionResult.java
@@ -37,8 +37,8 @@
         return this == SUCCESS || this == CONSUME;
     }
 
-    public static net.minecraft.world.InteractionResult sidedSuccess(boolean isClientSide) {
-        return isClientSide ? SUCCESS : CONSUME;
+    public static net.minecraft.world.InteractionResult sidedSuccess(boolean $$0) {
+        return $$0 ? SUCCESS : CONSUME;
     }
 
     public static /* synthetic */ net.minecraft.world.InteractionResult[] $values() {