--- net/minecraft/world/LockCode.java
+++ net/minecraft/world/LockCode.java
@@ -10,23 +10,23 @@
     public static java.lang.String TAG_LOCK = "Lock";
     public java.lang.String key;
 
-    public net.minecraft.world.LockCode(java.lang.String key) {
-        this.key = key;
+    public net.minecraft.world.LockCode(java.lang.String $$0) {
+        this.key = $$0;
     }
 
-    public boolean unlocksWith(net.minecraft.world.item.ItemStack stack) {
-        return this.key.isEmpty() || !stack.isEmpty() && stack.hasCustomHoverName() && this.key.equals((java.lang.Object)stack.getHoverName().getString());
+    public boolean unlocksWith(net.minecraft.world.item.ItemStack $$0) {
+        return this.key.isEmpty() || !$$0.isEmpty() && $$0.hasCustomHoverName() && this.key.equals((java.lang.Object)$$0.getHoverName().getString());
     }
 
-    public void addToTag(net.minecraft.nbt.CompoundTag nbt) {
+    public void addToTag(net.minecraft.nbt.CompoundTag $$0) {
         if (!this.key.isEmpty()) {
-            nbt.putString("Lock", this.key);
+            $$0.putString("Lock", this.key);
         }
     }
 
-    public static net.minecraft.world.LockCode fromTag(net.minecraft.nbt.CompoundTag nbt) {
-        if (nbt.contains("Lock", 8)) {
-            return new net.minecraft.world.LockCode(nbt.getString("Lock"));
+    public static net.minecraft.world.LockCode fromTag(net.minecraft.nbt.CompoundTag $$0) {
+        if ($$0.contains("Lock", 8)) {
+            return new net.minecraft.world.LockCode($$0.getString("Lock"));
         }
         return NO_LOCK;
     }