--- net/minecraft/world/SimpleMenuProvider.java
+++ net/minecraft/world/SimpleMenuProvider.java
@@ -9,17 +9,17 @@
     public net.minecraft.network.chat.Component title;
     public net.minecraft.world.inventory.MenuConstructor menuConstructor;
 
-    public net.minecraft.world.SimpleMenuProvider(net.minecraft.world.inventory.MenuConstructor menuConstructor, net.minecraft.network.chat.Component title) {
-        this.menuConstructor = menuConstructor;
-        this.title = title;
+    public net.minecraft.world.SimpleMenuProvider(net.minecraft.world.inventory.MenuConstructor $$0, net.minecraft.network.chat.Component $$1) {
+        this.menuConstructor = $$0;
+        this.title = $$1;
     }
 
     public net.minecraft.network.chat.Component getDisplayName() {
         return this.title;
     }
 
-    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int containerId, net.minecraft.world.entity.player.Inventory inventory, net.minecraft.world.entity.player.Player player) {
-        return this.menuConstructor.createMenu(containerId, inventory, player);
+    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1, net.minecraft.world.entity.player.Player $$2) {
+        return this.menuConstructor.createMenu($$0, $$1, $$2);
     }
 }
 