--- net/minecraft/world/level/BlockEventData.java
+++ net/minecraft/world/level/BlockEventData.java
@@ -11,11 +11,11 @@
     public int paramA;
     public int paramB;
 
-    public net.minecraft.world.level.BlockEventData(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.Block block, int paramA, int paramB) {
-        this.pos = pos;
-        this.block = block;
-        this.paramA = paramA;
-        this.paramB = paramB;
+    public net.minecraft.world.level.BlockEventData(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.Block $$1, int $$2, int $$3) {
+        this.pos = $$0;
+        this.block = $$1;
+        this.paramA = $$2;
+        this.paramB = $$3;
     }
 
     public java.lang.String toString() {
@@ -26,8 +26,8 @@
         return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.BlockEventData.class, "pos;block;paramA;paramB", "pos", "block", "paramA", "paramB"}, this);
     }
 
-    public boolean equals(java.lang.Object other) {
-        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.BlockEventData.class, "pos;block;paramA;paramB", "pos", "block", "paramA", "paramB"}, this, other);
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.world.level.BlockEventData.class, "pos;block;paramA;paramB", "pos", "block", "paramA", "paramB"}, this, $$0);
     }
 
     public net.minecraft.core.BlockPos pos() {