--- net/minecraft/world/level/ChunkPos$1.java
+++ net/minecraft/world/level/ChunkPos$1.java
@@ -21,22 +21,22 @@
         super($$0, $$1);
     }
 
-    public boolean tryAdvance(java.util.function.Consumer<? super net.minecraft.world.level.ChunkPos> consumer) {
+    public boolean tryAdvance(java.util.function.Consumer<? super net.minecraft.world.level.ChunkPos> $$0) {
         if (this.pos == null) {
             this.pos = this.val$from;
         } else {
-            int n = this.pos.x;
-            int n2 = this.pos.z;
-            if (n == this.val$to.x) {
-                if (n2 == this.val$to.z) {
+            int $$1 = this.pos.x;
+            int $$2 = this.pos.z;
+            if ($$1 == this.val$to.x) {
+                if ($$2 == this.val$to.z) {
                     return false;
                 }
-                this.pos = new net.minecraft.world.level.ChunkPos(this.val$from.x, n2 + this.val$zDiff);
+                this.pos = new net.minecraft.world.level.ChunkPos(this.val$from.x, $$2 + this.val$zDiff);
             } else {
-                this.pos = new net.minecraft.world.level.ChunkPos(n + this.val$xDiff, n2);
+                this.pos = new net.minecraft.world.level.ChunkPos($$1 + this.val$xDiff, $$2);
             }
         }
-        consumer.accept((java.lang.Object)this.pos);
+        $$0.accept((java.lang.Object)this.pos);
         return true;
     }
 }