--- net/minecraft/world/level/ClipContext$Fluid.java
+++ net/minecraft/world/level/ClipContext$Fluid.java
@@ -21,13 +21,13 @@
         return (net.minecraft.world.level.ClipContext$Fluid)java.lang.Enum.valueOf(net.minecraft.world.level.ClipContext$Fluid.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.ClipContext$Fluid(java.util.function.Predicate<net.minecraft.world.level.material.FluidState> canPick) {
+    public net.minecraft.world.level.ClipContext$Fluid(java.util.function.Predicate<net.minecraft.world.level.material.FluidState> $$0) {
         super(string, n);
-        this.canPick = canPick;
+        this.canPick = $$0;
     }
 
-    public boolean canPick(net.minecraft.world.level.material.FluidState state) {
-        return this.canPick.test((java.lang.Object)state);
+    public boolean canPick(net.minecraft.world.level.material.FluidState $$0) {
+        return this.canPick.test((java.lang.Object)$$0);
     }
 
     public static /* synthetic */ boolean lambda$static$2(net.minecraft.world.level.material.FluidState $$0) {