--- net/minecraft/world/level/ClipContext$Block.java
+++ net/minecraft/world/level/ClipContext$Block.java
@@ -22,13 +22,13 @@
         return (net.minecraft.world.level.ClipContext$Block)java.lang.Enum.valueOf(net.minecraft.world.level.ClipContext$Block.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.ClipContext$Block(net.minecraft.world.level.ClipContext$ShapeGetter shapeGetter) {
+    public net.minecraft.world.level.ClipContext$Block(net.minecraft.world.level.ClipContext$ShapeGetter $$0) {
         super(string, n);
-        this.shapeGetter = shapeGetter;
+        this.shapeGetter = $$0;
     }
 
-    public net.minecraft.world.phys.shapes.VoxelShape get(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.BlockGetter block, net.minecraft.core.BlockPos pos, net.minecraft.world.phys.shapes.CollisionContext collisionContext) {
-        return this.shapeGetter.get(state, block, pos, collisionContext);
+    public net.minecraft.world.phys.shapes.VoxelShape get(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {
+        return this.shapeGetter.get($$0, $$1, $$2, $$3);
     }
 
     public static /* synthetic */ net.minecraft.world.phys.shapes.VoxelShape lambda$static$0(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.phys.shapes.CollisionContext $$3) {