--- net/minecraft/world/level/CommonLevelAccessor.java
+++ net/minecraft/world/level/CommonLevelAccessor.java
@@ -1,1 +1,27 @@
-// Failed to decompile net/minecraft/world/level/CommonLevelAccessor.class
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level;
+
+public interface net.minecraft.world.level.CommonLevelAccessor
+extends net.minecraft.world.level.EntityGetter,
+net.minecraft.world.level.LevelReader,
+net.minecraft.world.level.LevelSimulatedRW
+ {
+    default public <T extends net.minecraft.world.level.block.entity.BlockEntity> java.util.Optional<T> getBlockEntity(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.entity.BlockEntityType<T> $$1) {
+        return super.getBlockEntity($$0, $$1);
+    }
+
+    default public java.util.List<net.minecraft.world.phys.shapes.VoxelShape> getEntityCollisions(@javax.annotation.Nullable net.minecraft.world.entity.Entity $$0, net.minecraft.world.phys.AABB $$1) {
+        return super.getEntityCollisions($$0, $$1);
+    }
+
+    default public boolean isUnobstructed(@javax.annotation.Nullable net.minecraft.world.entity.Entity $$0, net.minecraft.world.phys.shapes.VoxelShape $$1) {
+        return super.isUnobstructed($$0, $$1);
+    }
+
+    default public net.minecraft.core.BlockPos getHeightmapPos(net.minecraft.world.level.levelgen.Heightmap$Types $$0, net.minecraft.core.BlockPos $$1) {
+        return super.getHeightmapPos($$0, $$1);
+    }
+}
+