--- net/minecraft/world/level/DataPackConfig.java
+++ net/minecraft/world/level/DataPackConfig.java
@@ -10,9 +10,9 @@
     public java.util.List<java.lang.String> enabled;
     public java.util.List<java.lang.String> disabled;
 
-    public net.minecraft.world.level.DataPackConfig(java.util.List<java.lang.String> enabled, java.util.List<java.lang.String> disabled) {
-        this.enabled = com.google.common.collect.ImmutableList.copyOf(enabled);
-        this.disabled = com.google.common.collect.ImmutableList.copyOf(disabled);
+    public net.minecraft.world.level.DataPackConfig(java.util.List<java.lang.String> $$0, java.util.List<java.lang.String> $$1) {
+        this.enabled = com.google.common.collect.ImmutableList.copyOf($$0);
+        this.disabled = com.google.common.collect.ImmutableList.copyOf($$1);
     }
 
     public java.util.List<java.lang.String> getEnabled() {