--- net/minecraft/world/level/EmptyBlockGetter.java
+++ net/minecraft/world/level/EmptyBlockGetter.java
@@ -23,15 +23,15 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.block.entity.BlockEntity getBlockEntity(net.minecraft.core.BlockPos pos) {
+    public net.minecraft.world.level.block.entity.BlockEntity getBlockEntity(net.minecraft.core.BlockPos $$0) {
         return null;
     }
 
-    public net.minecraft.world.level.block.state.BlockState getBlockState(net.minecraft.core.BlockPos pos) {
+    public net.minecraft.world.level.block.state.BlockState getBlockState(net.minecraft.core.BlockPos $$0) {
         return net.minecraft.world.level.block.Blocks.AIR.defaultBlockState();
     }
 
-    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.core.BlockPos pos) {
+    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.core.BlockPos $$0) {
         return net.minecraft.world.level.material.Fluids.EMPTY.defaultFluidState();
     }
 