--- net/minecraft/world/level/EntityBasedExplosionDamageCalculator.java
+++ net/minecraft/world/level/EntityBasedExplosionDamageCalculator.java
@@ -8,16 +8,16 @@
  {
     public net.minecraft.world.entity.Entity source;
 
-    public net.minecraft.world.level.EntityBasedExplosionDamageCalculator(net.minecraft.world.entity.Entity souce) {
-        this.source = souce;
+    public net.minecraft.world.level.EntityBasedExplosionDamageCalculator(net.minecraft.world.entity.Entity $$0) {
+        this.source = $$0;
     }
 
-    public java.util.Optional<java.lang.Float> getBlockExplosionResistance(net.minecraft.world.level.Explosion explosion, net.minecraft.world.level.BlockGetter reader, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.material.FluidState fluid) {
-        return super.getBlockExplosionResistance(explosion, reader, pos, state, fluid).map($$5 -> java.lang.Float.valueOf((float)this.source.getBlockExplosionResistance(explosion, reader, pos, state, fluid, $$5.floatValue())));
+    public java.util.Optional<java.lang.Float> getBlockExplosionResistance(net.minecraft.world.level.Explosion $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, net.minecraft.world.level.material.FluidState $$4) {
+        return super.getBlockExplosionResistance($$0, $$1, $$2, $$3, $$4).map($$5 -> java.lang.Float.valueOf((float)this.source.getBlockExplosionResistance($$0, $$1, $$2, $$3, $$4, $$5.floatValue())));
     }
 
-    public boolean shouldBlockExplode(net.minecraft.world.level.Explosion explosion, net.minecraft.world.level.BlockGetter reader, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, float power) {
-        return this.source.shouldBlockExplode(explosion, reader, pos, state, power);
+    public boolean shouldBlockExplode(net.minecraft.world.level.Explosion $$0, net.minecraft.world.level.BlockGetter $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.block.state.BlockState $$3, float $$4) {
+        return this.source.shouldBlockExplode($$0, $$1, $$2, $$3, $$4);
     }
 }
 