--- net/minecraft/world/level/ForcedChunksSavedData.java
+++ net/minecraft/world/level/ForcedChunksSavedData.java
@@ -22,9 +22,9 @@
         return new net.minecraft.world.level.ForcedChunksSavedData((it.unimi.dsi.fastutil.longs.LongSet)new it.unimi.dsi.fastutil.longs.LongOpenHashSet($$0.getLongArray("Forced")));
     }
 
-    public net.minecraft.nbt.CompoundTag save(net.minecraft.nbt.CompoundTag compound) {
-        compound.putLongArray("Forced", this.chunks.toLongArray());
-        return compound;
+    public net.minecraft.nbt.CompoundTag save(net.minecraft.nbt.CompoundTag $$0) {
+        $$0.putLongArray("Forced", this.chunks.toLongArray());
+        return $$0;
     }
 
     public it.unimi.dsi.fastutil.longs.LongSet getChunks() {