--- net/minecraft/world/level/GameRules$GameRuleTypeVisitor.java
+++ net/minecraft/world/level/GameRules$GameRuleTypeVisitor.java
@@ -5,13 +5,13 @@
 
 public interface net.minecraft.world.level.GameRules$GameRuleTypeVisitor
  {
-    default public <T extends net.minecraft.world.level.GameRules$Value<T>> void visit(net.minecraft.world.level.GameRules$Key<T> key, net.minecraft.world.level.GameRules$Type<T> type) {
+    default public <T extends net.minecraft.world.level.GameRules$Value<T>> void visit(net.minecraft.world.level.GameRules$Key<T> $$0, net.minecraft.world.level.GameRules$Type<T> $$1) {
     }
 
-    default public void visitBoolean(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$BooleanValue> key, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> type) {
+    default public void visitBoolean(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$BooleanValue> $$0, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> $$1) {
     }
 
-    default public void visitInteger(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$IntegerValue> key, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$IntegerValue> type) {
+    default public void visitInteger(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$IntegerValue> $$0, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$IntegerValue> $$1) {
     }
 }
 