--- net/minecraft/world/level/GameRules$Category.java
+++ net/minecraft/world/level/GameRules$Category.java
@@ -24,9 +24,9 @@
         return (net.minecraft.world.level.GameRules$Category)java.lang.Enum.valueOf(net.minecraft.world.level.GameRules$Category.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.GameRules$Category(java.lang.String descriptionId) {
+    public net.minecraft.world.level.GameRules$Category(java.lang.String $$0) {
         super(string, n);
-        this.descriptionId = descriptionId;
+        this.descriptionId = $$0;
     }
 
     public java.lang.String getDescriptionId() {