--- net/minecraft/world/level/GameRules$Key.java
+++ net/minecraft/world/level/GameRules$Key.java
@@ -8,20 +8,20 @@
     public java.lang.String id;
     public net.minecraft.world.level.GameRules$Category category;
 
-    public net.minecraft.world.level.GameRules$Key(java.lang.String id, net.minecraft.world.level.GameRules$Category category) {
-        this.id = id;
-        this.category = category;
+    public net.minecraft.world.level.GameRules$Key(java.lang.String $$0, net.minecraft.world.level.GameRules$Category $$1) {
+        this.id = $$0;
+        this.category = $$1;
     }
 
     public java.lang.String toString() {
         return this.id;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        return other instanceof net.minecraft.world.level.GameRules$Key && ((net.minecraft.world.level.GameRules$Key)other).id.equals((java.lang.Object)this.id);
+        return $$0 instanceof net.minecraft.world.level.GameRules$Key && ((net.minecraft.world.level.GameRules$Key)$$0).id.equals((java.lang.Object)this.id);
     }
 
     public int hashCode() {