--- net/minecraft/world/level/GameRules$BooleanValue.java
+++ net/minecraft/world/level/GameRules$BooleanValue.java
@@ -8,39 +8,39 @@
  {
     public boolean value;
 
-    public static net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> create(boolean defaultValue, java.util.function.BiConsumer<net.minecraft.server.MinecraftServer, net.minecraft.world.level.GameRules$BooleanValue> changeListener) {
-        return new net.minecraft.world.level.GameRules$Type(com.mojang.brigadier.arguments.BoolArgumentType::bool, $$1 -> new net.minecraft.world.level.GameRules$BooleanValue($$1, defaultValue), changeListener, net.minecraft.world.level.GameRules$GameRuleTypeVisitor::visitBoolean);
+    public static net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> create(boolean $$0, java.util.function.BiConsumer<net.minecraft.server.MinecraftServer, net.minecraft.world.level.GameRules$BooleanValue> $$12) {
+        return new net.minecraft.world.level.GameRules$Type(com.mojang.brigadier.arguments.BoolArgumentType::bool, $$1 -> new net.minecraft.world.level.GameRules$BooleanValue($$1, $$0), $$12, net.minecraft.world.level.GameRules$GameRuleTypeVisitor::visitBoolean);
     }
 
-    public static net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> create(boolean defaultValue) {
-        return net.minecraft.world.level.GameRules$BooleanValue.create((boolean)defaultValue, ($$0, $$1) -> {}
+    public static net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> create(boolean $$02) {
+        return net.minecraft.world.level.GameRules$BooleanValue.create((boolean)$$02, ($$0, $$1) -> {}
 );
     }
 
-    public net.minecraft.world.level.GameRules$BooleanValue(net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> type, boolean value) {
-        super(type);
-        this.value = value;
+    public net.minecraft.world.level.GameRules$BooleanValue(net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> $$0, boolean $$1) {
+        super($$0);
+        this.value = $$1;
     }
 
-    public void updateFromArgument(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String paramName) {
-        this.value = com.mojang.brigadier.arguments.BoolArgumentType.getBool(context, (java.lang.String)paramName);
+    public void updateFromArgument(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        this.value = com.mojang.brigadier.arguments.BoolArgumentType.getBool($$0, (java.lang.String)$$1);
     }
 
     public boolean get() {
         return this.value;
     }
 
-    public void set(boolean value, @javax.annotation.Nullable net.minecraft.server.MinecraftServer server) {
-        this.value = value;
-        this.onChanged(server);
+    public void set(boolean $$0, @javax.annotation.Nullable net.minecraft.server.MinecraftServer $$1) {
+        this.value = $$0;
+        this.onChanged($$1);
     }
 
     public java.lang.String serialize() {
         return java.lang.Boolean.toString((boolean)this.value);
     }
 
-    public void deserialize(java.lang.String value) {
-        this.value = java.lang.Boolean.parseBoolean((java.lang.String)value);
+    public void deserialize(java.lang.String $$0) {
+        this.value = java.lang.Boolean.parseBoolean((java.lang.String)$$0);
     }
 
     public int getCommandResult() {
@@ -55,9 +55,9 @@
         return new net.minecraft.world.level.GameRules$BooleanValue(this.type, this.value);
     }
 
-    public void setFrom(net.minecraft.world.level.GameRules$BooleanValue value, @javax.annotation.Nullable net.minecraft.server.MinecraftServer server) {
-        this.value = value.value;
-        this.onChanged(server);
+    public void setFrom(net.minecraft.world.level.GameRules$BooleanValue $$0, @javax.annotation.Nullable net.minecraft.server.MinecraftServer $$1) {
+        this.value = $$0.value;
+        this.onChanged($$1);
     }
 
     public /* synthetic */ net.minecraft.world.level.GameRules$Value copy() {