--- net/minecraft/world/level/GrassColor.java
+++ net/minecraft/world/level/GrassColor.java
@@ -10,18 +10,18 @@
     public net.minecraft.world.level.GrassColor() {
     }
 
-    public static void init(int[] grassBuffer) {
-        pixels = grassBuffer;
+    public static void init(int[] $$0) {
+        pixels = $$0;
     }
 
-    public static int get(double temperature, double humidity) {
-        int n = (int)((1.0 - (humidity *= temperature)) * 255.0);
-        int n2 = (int)((1.0 - temperature) * 255.0);
-        int n3 = n << 8 | n2;
-        if (n3 >= pixels.length) {
+    public static int get(double $$0, double $$1) {
+        int $$3 = (int)((1.0 - ($$1 *= $$0)) * 255.0);
+        int $$2 = (int)((1.0 - $$0) * 255.0);
+        int $$4 = $$3 << 8 | $$2;
+        if ($$4 >= pixels.length) {
             return -65281;
         }
-        return pixels[n3];
+        return pixels[$$4];
     }
 
     static {