--- net/minecraft/world/level/LightLayer.java
+++ net/minecraft/world/level/LightLayer.java
@@ -19,9 +19,9 @@
         return (net.minecraft.world.level.LightLayer)java.lang.Enum.valueOf(net.minecraft.world.level.LightLayer.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.LightLayer(int surrounding) {
+    public net.minecraft.world.level.LightLayer(int $$0) {
         super(string, n);
-        this.surrounding = surrounding;
+        this.surrounding = $$0;
     }
 
     public static /* synthetic */ net.minecraft.world.level.LightLayer[] $values() {