--- net/minecraft/world/level/PotentialCalculator.java
+++ net/minecraft/world/level/PotentialCalculator.java
@@ -11,21 +11,21 @@
         this.charges = com.google.common.collect.Lists.newArrayList();
     }
 
-    public void addCharge(net.minecraft.core.BlockPos pos, double charge) {
-        if (charge != 0.0) {
-            this.charges.add((java.lang.Object)new net.minecraft.world.level.PotentialCalculator$PointCharge(pos, charge));
+    public void addCharge(net.minecraft.core.BlockPos $$0, double $$1) {
+        if ($$1 != 0.0) {
+            this.charges.add((java.lang.Object)new net.minecraft.world.level.PotentialCalculator$PointCharge($$0, $$1));
         }
     }
 
-    public double getPotentialEnergyChange(net.minecraft.core.BlockPos pos, double charge) {
-        if (charge == 0.0) {
+    public double getPotentialEnergyChange(net.minecraft.core.BlockPos $$0, double $$1) {
+        if ($$1 == 0.0) {
             return 0.0;
         }
-        double d = 0.0;
-        for (net.minecraft.world.level.PotentialCalculator$PointCharge pointCharge : this.charges) {
-            d += pointCharge.getPotentialChange(pos);
+        double $$2 = 0.0;
+        for (net.minecraft.world.level.PotentialCalculator$PointCharge $$3 : this.charges) {
+            $$2 += $$3.getPotentialChange($$0);
         }
-        return d * charge;
+        return $$2 * $$1;
     }
 }
 