--- net/minecraft/world/level/PotentialCalculator$PointCharge.java
+++ net/minecraft/world/level/PotentialCalculator$PointCharge.java
@@ -8,17 +8,17 @@
     public net.minecraft.core.BlockPos pos;
     public double charge;
 
-    public net.minecraft.world.level.PotentialCalculator$PointCharge(net.minecraft.core.BlockPos pos, double charge) {
-        this.pos = pos;
-        this.charge = charge;
+    public net.minecraft.world.level.PotentialCalculator$PointCharge(net.minecraft.core.BlockPos $$0, double $$1) {
+        this.pos = $$0;
+        this.charge = $$1;
     }
 
-    public double getPotentialChange(net.minecraft.core.BlockPos pos) {
-        double d = this.pos.distSqr((net.minecraft.core.Vec3i)pos);
-        if (d == 0.0) {
+    public double getPotentialChange(net.minecraft.core.BlockPos $$0) {
+        double $$1 = this.pos.distSqr((net.minecraft.core.Vec3i)$$0);
+        if ($$1 == 0.0) {
             return java.lang.Double.POSITIVE_INFINITY;
         }
-        return this.charge / java.lang.Math.sqrt((double)d);
+        return this.charge / java.lang.Math.sqrt((double)$$1);
     }
 }
 