--- net/minecraft/world/level/StructureManager.java
+++ net/minecraft/world/level/StructureManager.java
@@ -6,12 +6,12 @@
 public class net.minecraft.world.level.StructureManager
  {
     public net.minecraft.world.level.LevelAccessor level;
-    public net.minecraft.world.level.levelgen.WorldGenSettings worldGenSettings;
+    public net.minecraft.world.level.levelgen.WorldOptions worldOptions;
     public net.minecraft.world.level.levelgen.structure.StructureCheck structureCheck;
 
-    public net.minecraft.world.level.StructureManager(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.world.level.levelgen.WorldGenSettings $$1, net.minecraft.world.level.levelgen.structure.StructureCheck $$2) {
+    public net.minecraft.world.level.StructureManager(net.minecraft.world.level.LevelAccessor $$0, net.minecraft.world.level.levelgen.WorldOptions $$1, net.minecraft.world.level.levelgen.structure.StructureCheck $$2) {
         this.level = $$0;
-        this.worldGenSettings = $$1;
+        this.worldOptions = $$1;
         this.structureCheck = $$2;
     }
 
@@ -19,7 +19,7 @@
         if ($$0.getLevel() != this.level) {
             throw new java.lang.IllegalStateException("Using invalid structure manager (source level: " + $$0.getLevel() + ", region: " + $$0);
         }
-        return new net.minecraft.world.level.StructureManager((net.minecraft.world.level.LevelAccessor)$$0, this.worldGenSettings, this.structureCheck);
+        return new net.minecraft.world.level.StructureManager((net.minecraft.world.level.LevelAccessor)$$0, this.worldOptions, this.structureCheck);
     }
 
     public java.util.List<net.minecraft.world.level.levelgen.structure.StructureStart> startsForStructure(net.minecraft.world.level.ChunkPos $$0, java.util.function.Predicate<net.minecraft.world.level.levelgen.structure.Structure> $$1) {
@@ -65,7 +65,7 @@
     }
 
     public boolean shouldGenerateStructures() {
-        return this.worldGenSettings.generateStructures();
+        return this.worldOptions.generateStructures();
     }
 
     public net.minecraft.world.level.levelgen.structure.StructureStart getStructureAt(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.levelgen.structure.Structure $$1) {