--- net/minecraft/client/multiplayer/chat/ChatLog.java
+++ net/minecraft/client/multiplayer/chat/ChatLog.java
@@ -1,1 +1,36 @@
-// Failed to decompile net/minecraft/client/multiplayer/chat/ChatLog.class
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.multiplayer.chat;
+
+public class net.minecraft.client.multiplayer.chat.ChatLog
+ {
+    public net.minecraft.client.multiplayer.chat.LoggedChatEvent[] buffer;
+    public int nextId;
+
+    public net.minecraft.client.multiplayer.chat.ChatLog(int $$0) {
+        this.buffer = new net.minecraft.client.multiplayer.chat.LoggedChatEvent[$$0];
+    }
+
+    public void push(net.minecraft.client.multiplayer.chat.LoggedChatEvent $$0) {
+        this.buffer[this.index((int)this.nextId++)] = $$0;
+    }
+
+    @javax.annotation.Nullable
+    public net.minecraft.client.multiplayer.chat.LoggedChatEvent lookup(int $$0) {
+        return $$0 >= this.start() && $$0 <= this.end() ? this.buffer[this.index($$0)] : null;
+    }
+
+    public int index(int $$0) {
+        return $$0 % this.buffer.length;
+    }
+
+    public int start() {
+        return java.lang.Math.max((int)(this.nextId - this.buffer.length), (int)0);
+    }
+
+    public int end() {
+        return this.nextId - 1;
+    }
+}
+