--- net/minecraft/client/multiplayer/chat/ChatListener$Message.java
+++ net/minecraft/client/multiplayer/chat/ChatListener$Message.java
@@ -1,1 +1,43 @@
-// Failed to decompile net/minecraft/client/multiplayer/chat/ChatListener$Message.class
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.multiplayer.chat;
+
+public class net.minecraft.client.multiplayer.chat.ChatListener$Message
+extends java.lang.Record
+ {
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.MessageSignature signature;
+    public java.util.function.BooleanSupplier handler;
+
+    public net.minecraft.client.multiplayer.chat.ChatListener$Message(@javax.annotation.Nullable net.minecraft.network.chat.MessageSignature $$0, java.util.function.BooleanSupplier $$1) {
+        this.signature = $$0;
+        this.handler = $$1;
+    }
+
+    public boolean accept() {
+        return this.handler.getAsBoolean();
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.multiplayer.chat.ChatListener$Message.class, "signature;handler", "signature", "handler"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.multiplayer.chat.ChatListener$Message.class, "signature;handler", "signature", "handler"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.multiplayer.chat.ChatListener$Message.class, "signature;handler", "signature", "handler"}, this, $$0);
+    }
+
+    @javax.annotation.Nullable
+    public net.minecraft.network.chat.MessageSignature signature() {
+        return this.signature;
+    }
+
+    public java.util.function.BooleanSupplier handler() {
+        return this.handler;
+    }
+}
+