--- net/minecraft/client/multiplayer/chat/ChatTrustLevel$1.java
+++ net/minecraft/client/multiplayer/chat/ChatTrustLevel$1.java
@@ -16,17 +16,11 @@
             // empty catch block
         }
         try {
-            net.minecraft.client.multiplayer.chat.ChatTrustLevel$1.$SwitchMap$net$minecraft$client$multiplayer$chat$ChatTrustLevel[net.minecraft.client.multiplayer.chat.ChatTrustLevel.FILTERED.ordinal()] = 2;
+            net.minecraft.client.multiplayer.chat.ChatTrustLevel$1.$SwitchMap$net$minecraft$client$multiplayer$chat$ChatTrustLevel[net.minecraft.client.multiplayer.chat.ChatTrustLevel.NOT_SECURE.ordinal()] = 2;
         }
         catch (java.lang.NoSuchFieldError noSuchFieldError) {
             // empty catch block
         }
-        try {
-            net.minecraft.client.multiplayer.chat.ChatTrustLevel$1.$SwitchMap$net$minecraft$client$multiplayer$chat$ChatTrustLevel[net.minecraft.client.multiplayer.chat.ChatTrustLevel.NOT_SECURE.ordinal()] = 3;
-        }
-        catch (java.lang.NoSuchFieldError noSuchFieldError) {
-            // empty catch block
-        }
     }
 }
 