--- net/minecraft/world/level/levelgen/structure/structures/EndCityPieces$EndCityPiece.java
+++ net/minecraft/world/level/levelgen/structure/structures/EndCityPieces$EndCityPiece.java
@@ -42,8 +42,10 @@
         } else if ($$4.isInside((net.minecraft.core.Vec3i)$$1) && net.minecraft.world.level.Level.isInSpawnableBounds((net.minecraft.core.BlockPos)$$1)) {
             if ($$0.startsWith("Sentry")) {
                 net.minecraft.world.entity.monster.Shulker $$6 = (net.minecraft.world.entity.monster.Shulker)net.minecraft.world.entity.EntityType.SHULKER.create((net.minecraft.world.level.Level)$$2.getLevel());
-                $$6.setPos((double)$$1.getX() + 0.5, (double)$$1.getY(), (double)$$1.getZ() + 0.5);
-                $$2.addFreshEntity((net.minecraft.world.entity.Entity)$$6);
+                if ($$6 != null) {
+                    $$6.setPos((double)$$1.getX() + 0.5, (double)$$1.getY(), (double)$$1.getZ() + 0.5);
+                    $$2.addFreshEntity((net.minecraft.world.entity.Entity)$$6);
+                }
             } else if ($$0.startsWith("Elytra")) {
                 net.minecraft.world.entity.decoration.ItemFrame $$7 = new net.minecraft.world.entity.decoration.ItemFrame((net.minecraft.world.level.Level)$$2.getLevel(), $$1, this.placeSettings.getRotation().rotate(net.minecraft.core.Direction.SOUTH));
                 $$7.setItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.ELYTRA), false);