--- net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces$OceanRuinPiece.java
+++ net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces$OceanRuinPiece.java
@@ -37,14 +37,14 @@
     }
 
     public void handleDataMarker(java.lang.String $$0, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.ServerLevelAccessor $$2, net.minecraft.util.RandomSource $$3, net.minecraft.world.level.levelgen.structure.BoundingBox $$4) {
+        net.minecraft.world.entity.monster.Drowned $$6;
         if ("chest".equals((java.lang.Object)$$0)) {
             $$2.setBlock($$1, (net.minecraft.world.level.block.state.BlockState)net.minecraft.world.level.block.Blocks.CHEST.defaultBlockState().setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.ChestBlock.WATERLOGGED, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)$$2.getFluidState($$1).is(net.minecraft.tags.FluidTags.WATER))), 2);
             net.minecraft.world.level.block.entity.BlockEntity $$5 = $$2.getBlockEntity($$1);
             if ($$5 instanceof net.minecraft.world.level.block.entity.ChestBlockEntity) {
                 ((net.minecraft.world.level.block.entity.ChestBlockEntity)$$5).setLootTable(this.isLarge ? net.minecraft.world.level.storage.loot.BuiltInLootTables.UNDERWATER_RUIN_BIG : net.minecraft.world.level.storage.loot.BuiltInLootTables.UNDERWATER_RUIN_SMALL, $$3.nextLong());
             }
-        } else if ("drowned".equals((java.lang.Object)$$0)) {
-            net.minecraft.world.entity.monster.Drowned $$6 = (net.minecraft.world.entity.monster.Drowned)net.minecraft.world.entity.EntityType.DROWNED.create((net.minecraft.world.level.Level)$$2.getLevel());
+        } else if ("drowned".equals((java.lang.Object)$$0) && ($$6 = (net.minecraft.world.entity.monster.Drowned)net.minecraft.world.entity.EntityType.DROWNED.create((net.minecraft.world.level.Level)$$2.getLevel())) != null) {
             $$6.setPersistenceRequired();
             $$6.moveTo($$1, 0.0f, 0.0f);
             $$6.finalizeSpawn($$2, $$2.getCurrentDifficultyAt($$1), net.minecraft.world.entity.MobSpawnType.STRUCTURE, null, null);