--- net/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces$OceanMonumentPiece.java
+++ net/minecraft/world/level/levelgen/structure/structures/OceanMonumentPieces$OceanMonumentPiece.java
@@ -129,17 +129,15 @@
         return $$0.intersects(java.lang.Math.min((int)$$5, (int)$$7), java.lang.Math.min((int)$$6, (int)$$8), java.lang.Math.max((int)$$5, (int)$$7), java.lang.Math.max((int)$$6, (int)$$8));
     }
 
-    public boolean spawnElder(net.minecraft.world.level.WorldGenLevel $$0, net.minecraft.world.level.levelgen.structure.BoundingBox $$1, int $$2, int $$3, int $$4) {
+    public void spawnElder(net.minecraft.world.level.WorldGenLevel $$0, net.minecraft.world.level.levelgen.structure.BoundingBox $$1, int $$2, int $$3, int $$4) {
+        net.minecraft.world.entity.monster.ElderGuardian $$6;
         net.minecraft.core.BlockPos$MutableBlockPos $$5 = this.getWorldPos($$2, $$3, $$4);
-        if ($$1.isInside((net.minecraft.core.Vec3i)$$5)) {
-            net.minecraft.world.entity.monster.ElderGuardian $$6 = (net.minecraft.world.entity.monster.ElderGuardian)net.minecraft.world.entity.EntityType.ELDER_GUARDIAN.create((net.minecraft.world.level.Level)$$0.getLevel());
+        if ($$1.isInside((net.minecraft.core.Vec3i)$$5) && ($$6 = (net.minecraft.world.entity.monster.ElderGuardian)net.minecraft.world.entity.EntityType.ELDER_GUARDIAN.create((net.minecraft.world.level.Level)$$0.getLevel())) != null) {
             $$6.heal($$6.getMaxHealth());
             $$6.moveTo((double)$$5.getX() + 0.5, (double)$$5.getY(), (double)$$5.getZ() + 0.5, 0.0f, 0.0f);
             $$6.finalizeSpawn((net.minecraft.world.level.ServerLevelAccessor)$$0, $$0.getCurrentDifficultyAt($$6.blockPosition()), net.minecraft.world.entity.MobSpawnType.STRUCTURE, null, null);
             $$0.addFreshEntityWithPassengers((net.minecraft.world.entity.Entity)$$6);
-            return true;
         }
-        return false;
     }
 
     static {