--- net/minecraft/world/entity/player/Abilities.java
+++ net/minecraft/world/entity/player/Abilities.java
@@ -19,31 +19,31 @@
         this.walkingSpeed = 0.1f;
     }
 
-    public void addSaveData(net.minecraft.nbt.CompoundTag compound) {
-        net.minecraft.nbt.CompoundTag compoundTag = new net.minecraft.nbt.CompoundTag();
-        compoundTag.putBoolean("invulnerable", this.invulnerable);
-        compoundTag.putBoolean("flying", this.flying);
-        compoundTag.putBoolean("mayfly", this.mayfly);
-        compoundTag.putBoolean("instabuild", this.instabuild);
-        compoundTag.putBoolean("mayBuild", this.mayBuild);
-        compoundTag.putFloat("flySpeed", this.flyingSpeed);
-        compoundTag.putFloat("walkSpeed", this.walkingSpeed);
-        compound.put("abilities", (net.minecraft.nbt.Tag)compoundTag);
+    public void addSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = new net.minecraft.nbt.CompoundTag();
+        $$1.putBoolean("invulnerable", this.invulnerable);
+        $$1.putBoolean("flying", this.flying);
+        $$1.putBoolean("mayfly", this.mayfly);
+        $$1.putBoolean("instabuild", this.instabuild);
+        $$1.putBoolean("mayBuild", this.mayBuild);
+        $$1.putFloat("flySpeed", this.flyingSpeed);
+        $$1.putFloat("walkSpeed", this.walkingSpeed);
+        $$0.put("abilities", (net.minecraft.nbt.Tag)$$1);
     }
 
-    public void loadSaveData(net.minecraft.nbt.CompoundTag compound) {
-        if (compound.contains("abilities", 10)) {
-            net.minecraft.nbt.CompoundTag compoundTag = compound.getCompound("abilities");
-            this.invulnerable = compoundTag.getBoolean("invulnerable");
-            this.flying = compoundTag.getBoolean("flying");
-            this.mayfly = compoundTag.getBoolean("mayfly");
-            this.instabuild = compoundTag.getBoolean("instabuild");
-            if (compoundTag.contains("flySpeed", 99)) {
-                this.flyingSpeed = compoundTag.getFloat("flySpeed");
-                this.walkingSpeed = compoundTag.getFloat("walkSpeed");
+    public void loadSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        if ($$0.contains("abilities", 10)) {
+            net.minecraft.nbt.CompoundTag $$1 = $$0.getCompound("abilities");
+            this.invulnerable = $$1.getBoolean("invulnerable");
+            this.flying = $$1.getBoolean("flying");
+            this.mayfly = $$1.getBoolean("mayfly");
+            this.instabuild = $$1.getBoolean("instabuild");
+            if ($$1.contains("flySpeed", 99)) {
+                this.flyingSpeed = $$1.getFloat("flySpeed");
+                this.walkingSpeed = $$1.getFloat("walkSpeed");
             }
-            if (compoundTag.contains("mayBuild", 1)) {
-                this.mayBuild = compoundTag.getBoolean("mayBuild");
+            if ($$1.contains("mayBuild", 1)) {
+                this.mayBuild = $$1.getBoolean("mayBuild");
             }
         }
     }
@@ -52,16 +52,16 @@
         return this.flyingSpeed;
     }
 
-    public void setFlyingSpeed(float flyingSpeed) {
-        this.flyingSpeed = flyingSpeed;
+    public void setFlyingSpeed(float $$0) {
+        this.flyingSpeed = $$0;
     }
 
     public float getWalkingSpeed() {
         return this.walkingSpeed;
     }
 
-    public void setWalkingSpeed(float walkingSpeed) {
-        this.walkingSpeed = walkingSpeed;
+    public void setWalkingSpeed(float $$0) {
+        this.walkingSpeed = $$0;
     }
 }
 