--- net/minecraft/world/entity/player/ChatVisiblity.java
+++ net/minecraft/world/entity/player/ChatVisiblity.java
@@ -23,10 +23,10 @@
         return (net.minecraft.world.entity.player.ChatVisiblity)java.lang.Enum.valueOf(net.minecraft.world.entity.player.ChatVisiblity.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.player.ChatVisiblity(int id, java.lang.String key) {
+    public net.minecraft.world.entity.player.ChatVisiblity(int $$0, java.lang.String $$1) {
         super(string, n);
-        this.id = id;
-        this.key = key;
+        this.id = $$0;
+        this.key = $$1;
     }
 
     public int getId() {
@@ -37,8 +37,8 @@
         return this.key;
     }
 
-    public static net.minecraft.world.entity.player.ChatVisiblity byId(int id) {
-        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)id, (int)BY_ID.length)];
+    public static net.minecraft.world.entity.player.ChatVisiblity byId(int $$0) {
+        return BY_ID[net.minecraft.util.Mth.positiveModulo((int)$$0, (int)BY_ID.length)];
     }
 
     public static /* synthetic */ net.minecraft.world.entity.player.ChatVisiblity[] lambda$static$0(int $$0) {