--- net/minecraft/world/entity/player/Player$BedSleepingProblem.java
+++ net/minecraft/world/entity/player/Player$BedSleepingProblem.java
@@ -29,9 +29,9 @@
         this.message = null;
     }
 
-    public net.minecraft.world.entity.player.Player$BedSleepingProblem(net.minecraft.network.chat.Component message) {
+    public net.minecraft.world.entity.player.Player$BedSleepingProblem(net.minecraft.network.chat.Component $$0) {
         super(string, n);
-        this.message = message;
+        this.message = $$0;
     }
 
     @javax.annotation.Nullable