--- net/minecraft/world/entity/player/PlayerModelPart.java
+++ net/minecraft/world/entity/player/PlayerModelPart.java
@@ -27,12 +27,12 @@
         return (net.minecraft.world.entity.player.PlayerModelPart)java.lang.Enum.valueOf(net.minecraft.world.entity.player.PlayerModelPart.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.player.PlayerModelPart(int bit, java.lang.String id) {
+    public net.minecraft.world.entity.player.PlayerModelPart(int $$0, java.lang.String $$1) {
         super(string, n);
-        this.bit = bit;
-        this.mask = 1 << bit;
-        this.id = id;
-        this.name = net.minecraft.network.chat.Component.translatable((java.lang.String)("options.modelPart." + id));
+        this.bit = $$0;
+        this.mask = 1 << $$0;
+        this.id = $$1;
+        this.name = net.minecraft.network.chat.Component.translatable((java.lang.String)("options.modelPart." + $$1));
     }
 
     public int getMask() {