--- net/minecraft/world/entity/ai/memory/MemoryModuleType.java
+++ net/minecraft/world/entity/ai/memory/MemoryModuleType.java
@@ -46,6 +46,7 @@
     public static net.minecraft.world.entity.ai.memory.MemoryModuleType<java.lang.Integer> PLAY_DEAD_TICKS;
     public static net.minecraft.world.entity.ai.memory.MemoryModuleType<net.minecraft.world.entity.player.Player> TEMPTING_PLAYER;
     public static net.minecraft.world.entity.ai.memory.MemoryModuleType<java.lang.Integer> TEMPTATION_COOLDOWN_TICKS;
+    public static net.minecraft.world.entity.ai.memory.MemoryModuleType<java.lang.Integer> GAZE_COOLDOWN_TICKS;
     public static net.minecraft.world.entity.ai.memory.MemoryModuleType<java.lang.Boolean> IS_TEMPTED;
     public static net.minecraft.world.entity.ai.memory.MemoryModuleType<java.lang.Integer> LONG_JUMP_COOLDOWN_TICKS;
     public static net.minecraft.world.entity.ai.memory.MemoryModuleType<java.lang.Boolean> LONG_JUMP_MID_JUMP;
@@ -112,12 +113,12 @@
         return this.codec;
     }
 
-    public static <U> net.minecraft.world.entity.ai.memory.MemoryModuleType<U> register(java.lang.String identifier, com.mojang.serialization.Codec<U> codec) {
-        return (net.minecraft.world.entity.ai.memory.MemoryModuleType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.MEMORY_MODULE_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(identifier), (java.lang.Object)new net.minecraft.world.entity.ai.memory.MemoryModuleType(java.util.Optional.of(codec)));
+    public static <U> net.minecraft.world.entity.ai.memory.MemoryModuleType<U> register(java.lang.String $$0, com.mojang.serialization.Codec<U> $$1) {
+        return (net.minecraft.world.entity.ai.memory.MemoryModuleType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.MEMORY_MODULE_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0), (java.lang.Object)new net.minecraft.world.entity.ai.memory.MemoryModuleType(java.util.Optional.of($$1)));
     }
 
-    public static <U> net.minecraft.world.entity.ai.memory.MemoryModuleType<U> register(java.lang.String identifier) {
-        return (net.minecraft.world.entity.ai.memory.MemoryModuleType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.MEMORY_MODULE_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(identifier), (java.lang.Object)new net.minecraft.world.entity.ai.memory.MemoryModuleType(java.util.Optional.empty()));
+    public static <U> net.minecraft.world.entity.ai.memory.MemoryModuleType<U> register(java.lang.String $$0) {
+        return (net.minecraft.world.entity.ai.memory.MemoryModuleType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.MEMORY_MODULE_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0), (java.lang.Object)new net.minecraft.world.entity.ai.memory.MemoryModuleType(java.util.Optional.empty()));
     }
 
     static {
@@ -162,6 +163,7 @@
         PLAY_DEAD_TICKS = net.minecraft.world.entity.ai.memory.MemoryModuleType.register((java.lang.String)"play_dead_ticks", (com.mojang.serialization.Codec)com.mojang.serialization.Codec.INT);
         TEMPTING_PLAYER = net.minecraft.world.entity.ai.memory.MemoryModuleType.register((java.lang.String)"tempting_player");
         TEMPTATION_COOLDOWN_TICKS = net.minecraft.world.entity.ai.memory.MemoryModuleType.register((java.lang.String)"temptation_cooldown_ticks", (com.mojang.serialization.Codec)com.mojang.serialization.Codec.INT);
+        GAZE_COOLDOWN_TICKS = net.minecraft.world.entity.ai.memory.MemoryModuleType.register((java.lang.String)"gaze_cooldown_ticks", (com.mojang.serialization.Codec)com.mojang.serialization.Codec.INT);
         IS_TEMPTED = net.minecraft.world.entity.ai.memory.MemoryModuleType.register((java.lang.String)"is_tempted", (com.mojang.serialization.Codec)com.mojang.serialization.Codec.BOOL);
         LONG_JUMP_COOLDOWN_TICKS = net.minecraft.world.entity.ai.memory.MemoryModuleType.register((java.lang.String)"long_jump_cooling_down", (com.mojang.serialization.Codec)com.mojang.serialization.Codec.INT);
         LONG_JUMP_MID_JUMP = net.minecraft.world.entity.ai.memory.MemoryModuleType.register((java.lang.String)"long_jump_mid_jump");