--- net/minecraft/commands/arguments/AngleArgument$SingleAngle.java
+++ net/minecraft/commands/arguments/AngleArgument$SingleAngle.java
@@ -8,13 +8,13 @@
     public float angle;
     public boolean isRelative;
 
-    public net.minecraft.commands.arguments.AngleArgument$SingleAngle(float angle, boolean isRelative) {
-        this.angle = angle;
-        this.isRelative = isRelative;
+    public net.minecraft.commands.arguments.AngleArgument$SingleAngle(float $$0, boolean $$1) {
+        this.angle = $$0;
+        this.isRelative = $$1;
     }
 
-    public float getAngle(net.minecraft.commands.CommandSourceStack source) {
-        return net.minecraft.util.Mth.wrapDegrees((float)(this.isRelative ? this.angle + source.getRotation().y : this.angle));
+    public float getAngle(net.minecraft.commands.CommandSourceStack $$0) {
+        return net.minecraft.util.Mth.wrapDegrees((float)(this.isRelative ? this.angle + $$0.getRotation().y : this.angle));
     }
 }
 