--- net/minecraft/commands/arguments/AngleArgument.java
+++ net/minecraft/commands/arguments/AngleArgument.java
@@ -17,21 +17,21 @@
         return new net.minecraft.commands.arguments.AngleArgument();
     }
 
-    public static float getAngle(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return ((net.minecraft.commands.arguments.AngleArgument$SingleAngle)context.getArgument(name, net.minecraft.commands.arguments.AngleArgument$SingleAngle.class)).getAngle((net.minecraft.commands.CommandSourceStack)context.getSource());
+    public static float getAngle(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return ((net.minecraft.commands.arguments.AngleArgument$SingleAngle)$$0.getArgument($$1, net.minecraft.commands.arguments.AngleArgument$SingleAngle.class)).getAngle((net.minecraft.commands.CommandSourceStack)$$0.getSource());
     }
 
-    public net.minecraft.commands.arguments.AngleArgument$SingleAngle parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        float f;
-        if (!reader.canRead()) {
-            throw ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+    public net.minecraft.commands.arguments.AngleArgument$SingleAngle parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        float $$2;
+        if (!$$0.canRead()) {
+            throw ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        boolean bl = net.minecraft.commands.arguments.coordinates.WorldCoordinate.isRelative((com.mojang.brigadier.StringReader)reader);
-        float f2 = f = reader.canRead() && reader.peek() != ' ' ? reader.readFloat() : 0.0f;
-        if (java.lang.Float.isNaN((float)f) || java.lang.Float.isInfinite((float)f)) {
-            throw ERROR_INVALID_ANGLE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+        boolean $$1 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.isRelative((com.mojang.brigadier.StringReader)$$0);
+        float f = $$2 = $$0.canRead() && $$0.peek() != ' ' ? $$0.readFloat() : 0.0f;
+        if (java.lang.Float.isNaN((float)$$2) || java.lang.Float.isInfinite((float)$$2)) {
+            throw ERROR_INVALID_ANGLE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        return new net.minecraft.commands.arguments.AngleArgument$SingleAngle(f, bl);
+        return new net.minecraft.commands.arguments.AngleArgument$SingleAngle($$2, $$1);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {