--- net/minecraft/commands/arguments/CompoundTagArgument.java
+++ net/minecraft/commands/arguments/CompoundTagArgument.java
@@ -15,12 +15,12 @@
         return new net.minecraft.commands.arguments.CompoundTagArgument();
     }
 
-    public static <S> net.minecraft.nbt.CompoundTag getCompoundTag(com.mojang.brigadier.context.CommandContext<S> context, java.lang.String name) {
-        return (net.minecraft.nbt.CompoundTag)context.getArgument(name, net.minecraft.nbt.CompoundTag.class);
+    public static <S> net.minecraft.nbt.CompoundTag getCompoundTag(com.mojang.brigadier.context.CommandContext<S> $$0, java.lang.String $$1) {
+        return (net.minecraft.nbt.CompoundTag)$$0.getArgument($$1, net.minecraft.nbt.CompoundTag.class);
     }
 
-    public net.minecraft.nbt.CompoundTag parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return new net.minecraft.nbt.TagParser(reader).readStruct();
+    public net.minecraft.nbt.CompoundTag parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return new net.minecraft.nbt.TagParser($$0).readStruct();
     }
 
     public java.util.Collection<java.lang.String> getExamples() {