--- net/minecraft/commands/arguments/ComponentArgument.java
+++ net/minecraft/commands/arguments/ComponentArgument.java
@@ -12,25 +12,25 @@
     public net.minecraft.commands.arguments.ComponentArgument() {
     }
 
-    public static net.minecraft.network.chat.Component getComponent(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (net.minecraft.network.chat.Component)context.getArgument(name, net.minecraft.network.chat.Component.class);
+    public static net.minecraft.network.chat.Component getComponent(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (net.minecraft.network.chat.Component)$$0.getArgument($$1, net.minecraft.network.chat.Component.class);
     }
 
     public static net.minecraft.commands.arguments.ComponentArgument textComponent() {
         return new net.minecraft.commands.arguments.ComponentArgument();
     }
 
-    public net.minecraft.network.chat.Component parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+    public net.minecraft.network.chat.Component parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
         try {
-            net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.Component$Serializer.fromJson((com.mojang.brigadier.StringReader)reader);
-            if (mutableComponent == null) {
-                throw ERROR_INVALID_JSON.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader, (java.lang.Object)"empty");
+            net.minecraft.network.chat.MutableComponent $$1 = net.minecraft.network.chat.Component$Serializer.fromJson((com.mojang.brigadier.StringReader)$$0);
+            if ($$1 == null) {
+                throw ERROR_INVALID_JSON.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0, (java.lang.Object)"empty");
             }
-            return mutableComponent;
+            return $$1;
         }
-        catch (java.lang.Exception exception) {
-            java.lang.String string = exception.getCause() != null ? exception.getCause().getMessage() : exception.getMessage();
-            throw ERROR_INVALID_JSON.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader, (java.lang.Object)string);
+        catch (java.lang.Exception $$2) {
+            java.lang.String $$3 = $$2.getCause() != null ? $$2.getCause().getMessage() : $$2.getMessage();
+            throw ERROR_INVALID_JSON.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0, (java.lang.Object)$$3);
         }
     }
 
@@ -42,13 +42,13 @@
         return this.parse(stringReader);
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object component) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.component.invalid", (java.lang.Object[])new java.lang.Object[]{component});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.component.invalid", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
     static {
         EXAMPLES = java.util.Arrays.asList((java.lang.Object[])new java.lang.String[]{"\"hello world\"", "\"\"", "\"{\"text\":\"hello world\"}", "[\"\"]"});
-        ERROR_INVALID_JSON = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(component -> net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.component.invalid", (java.lang.Object[])new java.lang.Object[]{component}));
+        ERROR_INVALID_JSON = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.component.invalid", (java.lang.Object[])new java.lang.Object[]{$$0}));
     }
 }
 