--- net/minecraft/commands/arguments/EntityAnchorArgument.java
+++ net/minecraft/commands/arguments/EntityAnchorArgument.java
@@ -12,8 +12,8 @@
     public net.minecraft.commands.arguments.EntityAnchorArgument() {
     }
 
-    public static net.minecraft.commands.arguments.EntityAnchorArgument$Anchor getAnchor(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor)context.getArgument(name, net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.class);
+    public static net.minecraft.commands.arguments.EntityAnchorArgument$Anchor getAnchor(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor)$$0.getArgument($$1, net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.class);
     }
 
     public static net.minecraft.commands.arguments.EntityAnchorArgument anchor() {
@@ -31,8 +31,8 @@
         return $$3;
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.BY_NAME.keySet(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder);
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.BY_NAME.keySet(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {
@@ -43,13 +43,13 @@
         return this.parse(stringReader);
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object anchor) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.anchor.invalid", (java.lang.Object[])new java.lang.Object[]{anchor});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.anchor.invalid", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
     static {
         EXAMPLES = java.util.Arrays.asList((java.lang.Object[])new java.lang.String[]{"eyes", "feet"});
-        ERROR_INVALID = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(anchor -> net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.anchor.invalid", (java.lang.Object[])new java.lang.Object[]{anchor}));
+        ERROR_INVALID = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.anchor.invalid", (java.lang.Object[])new java.lang.Object[]{$$0}));
     }
 }
 