--- net/minecraft/commands/arguments/EntityAnchorArgument$Anchor.java
+++ net/minecraft/commands/arguments/EntityAnchorArgument$Anchor.java
@@ -21,41 +21,41 @@
         return (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor)java.lang.Enum.valueOf(net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.commands.arguments.EntityAnchorArgument$Anchor(java.lang.String name, java.util.function.BiFunction<net.minecraft.world.phys.Vec3, net.minecraft.world.entity.Entity, net.minecraft.world.phys.Vec3> transform) {
+    public net.minecraft.commands.arguments.EntityAnchorArgument$Anchor(java.lang.String $$0, java.util.function.BiFunction<net.minecraft.world.phys.Vec3, net.minecraft.world.entity.Entity, net.minecraft.world.phys.Vec3> $$1) {
         super(string, n);
-        this.name = name;
-        this.transform = transform;
+        this.name = $$0;
+        this.transform = $$1;
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.commands.arguments.EntityAnchorArgument$Anchor getByName(java.lang.String name) {
-        return (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor)BY_NAME.get((java.lang.Object)name);
+    public static net.minecraft.commands.arguments.EntityAnchorArgument$Anchor getByName(java.lang.String $$0) {
+        return (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor)BY_NAME.get((java.lang.Object)$$0);
     }
 
-    public net.minecraft.world.phys.Vec3 apply(net.minecraft.world.entity.Entity entity) {
-        return (net.minecraft.world.phys.Vec3)this.transform.apply((java.lang.Object)entity.position(), (java.lang.Object)entity);
+    public net.minecraft.world.phys.Vec3 apply(net.minecraft.world.entity.Entity $$0) {
+        return (net.minecraft.world.phys.Vec3)this.transform.apply((java.lang.Object)$$0.position(), (java.lang.Object)$$0);
     }
 
-    public net.minecraft.world.phys.Vec3 apply(net.minecraft.commands.CommandSourceStack source) {
-        net.minecraft.world.entity.Entity entity = source.getEntity();
-        if (entity == null) {
-            return source.getPosition();
+    public net.minecraft.world.phys.Vec3 apply(net.minecraft.commands.CommandSourceStack $$0) {
+        net.minecraft.world.entity.Entity $$1 = $$0.getEntity();
+        if ($$1 == null) {
+            return $$0.getPosition();
         }
-        return (net.minecraft.world.phys.Vec3)this.transform.apply((java.lang.Object)source.getPosition(), (java.lang.Object)entity);
+        return (net.minecraft.world.phys.Vec3)this.transform.apply((java.lang.Object)$$0.getPosition(), (java.lang.Object)$$1);
     }
 
-    public static /* synthetic */ void lambda$static$2(java.util.HashMap map) {
-        for (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor anchor : net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.values()) {
-            map.put((java.lang.Object)anchor.name, (java.lang.Object)anchor);
+    public static /* synthetic */ void lambda$static$2(java.util.HashMap $$0) {
+        for (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor $$1 : net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.values()) {
+            $$0.put((java.lang.Object)$$1.name, (java.lang.Object)$$1);
         }
     }
 
-    public static /* synthetic */ net.minecraft.world.phys.Vec3 lambda$static$1(net.minecraft.world.phys.Vec3 pos, net.minecraft.world.entity.Entity entity) {
-        return new net.minecraft.world.phys.Vec3(pos.x, pos.y + (double)entity.getEyeHeight(), pos.z);
+    public static /* synthetic */ net.minecraft.world.phys.Vec3 lambda$static$1(net.minecraft.world.phys.Vec3 $$0, net.minecraft.world.entity.Entity $$1) {
+        return new net.minecraft.world.phys.Vec3($$0.x, $$0.y + (double)$$1.getEyeHeight(), $$0.z);
     }
 
-    public static /* synthetic */ net.minecraft.world.phys.Vec3 lambda$static$0(net.minecraft.world.phys.Vec3 pos, net.minecraft.world.entity.Entity entity) {
-        return pos;
+    public static /* synthetic */ net.minecraft.world.phys.Vec3 lambda$static$0(net.minecraft.world.phys.Vec3 $$0, net.minecraft.world.entity.Entity $$1) {
+        return $$0;
     }
 
     public static /* synthetic */ net.minecraft.commands.arguments.EntityAnchorArgument$Anchor[] $values() {
@@ -63,12 +63,12 @@
     }
 
     static {
-        FEET = new net.minecraft.commands.arguments.EntityAnchorArgument$Anchor("FEET", 0, "feet", (pos, entity) -> pos);
-        EYES = new net.minecraft.commands.arguments.EntityAnchorArgument$Anchor("EYES", 1, "eyes", (pos, entity) -> new net.minecraft.world.phys.Vec3(pos.x, pos.y + (double)entity.getEyeHeight(), pos.z));
+        FEET = new net.minecraft.commands.arguments.EntityAnchorArgument$Anchor("FEET", 0, "feet", ($$0, $$1) -> $$0);
+        EYES = new net.minecraft.commands.arguments.EntityAnchorArgument$Anchor("EYES", 1, "eyes", ($$0, $$1) -> new net.minecraft.world.phys.Vec3($$0.x, $$0.y + (double)$$1.getEyeHeight(), $$0.z));
         $VALUES = net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.$values();
-        BY_NAME = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), map -> {
-            for (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor anchor : net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.values()) {
-                map.put((java.lang.Object)anchor.name, (java.lang.Object)anchor);
+        BY_NAME = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), $$0 -> {
+            for (net.minecraft.commands.arguments.EntityAnchorArgument$Anchor $$1 : net.minecraft.commands.arguments.EntityAnchorArgument$Anchor.values()) {
+                $$0.put((java.lang.Object)$$1.name, (java.lang.Object)$$1);
             }
         });
     }