--- net/minecraft/commands/arguments/MessageArgument$Part.java
+++ net/minecraft/commands/arguments/MessageArgument$Part.java
@@ -9,10 +9,10 @@
     public int end;
     public net.minecraft.commands.arguments.selector.EntitySelector selector;
 
-    public net.minecraft.commands.arguments.MessageArgument$Part(int start, int end, net.minecraft.commands.arguments.selector.EntitySelector selector) {
-        this.start = start;
-        this.end = end;
-        this.selector = selector;
+    public net.minecraft.commands.arguments.MessageArgument$Part(int $$0, int $$1, net.minecraft.commands.arguments.selector.EntitySelector $$2) {
+        this.start = $$0;
+        this.end = $$1;
+        this.selector = $$2;
     }
 
     public int getStart() {
@@ -28,8 +28,8 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.network.chat.Component toComponent(net.minecraft.commands.CommandSourceStack source) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return net.minecraft.commands.arguments.selector.EntitySelector.joinNames((java.util.List)this.selector.findEntities(source));
+    public net.minecraft.network.chat.Component toComponent(net.minecraft.commands.CommandSourceStack $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return net.minecraft.commands.arguments.selector.EntitySelector.joinNames((java.util.List)this.selector.findEntities($$0));
     }
 }
 