--- net/minecraft/commands/arguments/NbtPathArgument$IndexedElementNode.java
+++ net/minecraft/commands/arguments/NbtPathArgument$IndexedElementNode.java
@@ -8,40 +8,40 @@
  {
     public int index;
 
-    public net.minecraft.commands.arguments.NbtPathArgument$IndexedElementNode(int index) {
-        this.index = index;
+    public net.minecraft.commands.arguments.NbtPathArgument$IndexedElementNode(int $$0) {
+        this.index = $$0;
     }
 
-    public void getTag(net.minecraft.nbt.Tag tag, java.util.List<net.minecraft.nbt.Tag> tags) {
-        if (tag instanceof net.minecraft.nbt.CollectionTag) {
-            int n;
-            net.minecraft.nbt.CollectionTag collectionTag = (net.minecraft.nbt.CollectionTag)tag;
-            int n2 = collectionTag.size();
-            int n3 = n = this.index < 0 ? n2 + this.index : this.index;
-            if (0 <= n && n < n2) {
-                tags.add((java.lang.Object)((net.minecraft.nbt.Tag)collectionTag.get(n)));
+    public void getTag(net.minecraft.nbt.Tag $$0, java.util.List<net.minecraft.nbt.Tag> $$1) {
+        if ($$0 instanceof net.minecraft.nbt.CollectionTag) {
+            int $$4;
+            net.minecraft.nbt.CollectionTag $$2 = (net.minecraft.nbt.CollectionTag)$$0;
+            int $$3 = $$2.size();
+            int n = $$4 = this.index < 0 ? $$3 + this.index : this.index;
+            if (0 <= $$4 && $$4 < $$3) {
+                $$1.add((java.lang.Object)((net.minecraft.nbt.Tag)$$2.get($$4)));
             }
         }
     }
 
-    public void getOrCreateTag(net.minecraft.nbt.Tag tag, java.util.function.Supplier<net.minecraft.nbt.Tag> supplier, java.util.List<net.minecraft.nbt.Tag> tags) {
-        this.getTag(tag, tags);
+    public void getOrCreateTag(net.minecraft.nbt.Tag $$0, java.util.function.Supplier<net.minecraft.nbt.Tag> $$1, java.util.List<net.minecraft.nbt.Tag> $$2) {
+        this.getTag($$0, $$2);
     }
 
     public net.minecraft.nbt.Tag createPreferredParentTag() {
         return new net.minecraft.nbt.ListTag();
     }
 
-    public int setTag(net.minecraft.nbt.Tag tag, java.util.function.Supplier<net.minecraft.nbt.Tag> supplier) {
-        if (tag instanceof net.minecraft.nbt.CollectionTag) {
-            int n;
-            net.minecraft.nbt.CollectionTag collectionTag = (net.minecraft.nbt.CollectionTag)tag;
-            int n2 = collectionTag.size();
-            int n3 = n = this.index < 0 ? n2 + this.index : this.index;
-            if (0 <= n && n < n2) {
-                net.minecraft.nbt.Tag tag2 = (net.minecraft.nbt.Tag)collectionTag.get(n);
-                net.minecraft.nbt.Tag tag3 = (net.minecraft.nbt.Tag)supplier.get();
-                if (!tag3.equals((java.lang.Object)tag2) && collectionTag.setTag(n, tag3)) {
+    public int setTag(net.minecraft.nbt.Tag $$0, java.util.function.Supplier<net.minecraft.nbt.Tag> $$1) {
+        if ($$0 instanceof net.minecraft.nbt.CollectionTag) {
+            int $$4;
+            net.minecraft.nbt.CollectionTag $$2 = (net.minecraft.nbt.CollectionTag)$$0;
+            int $$3 = $$2.size();
+            int n = $$4 = this.index < 0 ? $$3 + this.index : this.index;
+            if (0 <= $$4 && $$4 < $$3) {
+                net.minecraft.nbt.Tag $$5 = (net.minecraft.nbt.Tag)$$2.get($$4);
+                net.minecraft.nbt.Tag $$6 = (net.minecraft.nbt.Tag)$$1.get();
+                if (!$$6.equals((java.lang.Object)$$5) && $$2.setTag($$4, $$6)) {
                     return 1;
                 }
             }
@@ -49,14 +49,14 @@
         return 0;
     }
 
-    public int removeTag(net.minecraft.nbt.Tag tag) {
-        if (tag instanceof net.minecraft.nbt.CollectionTag) {
-            int n;
-            net.minecraft.nbt.CollectionTag collectionTag = (net.minecraft.nbt.CollectionTag)tag;
-            int n2 = collectionTag.size();
-            int n3 = n = this.index < 0 ? n2 + this.index : this.index;
-            if (0 <= n && n < n2) {
-                collectionTag.remove(n);
+    public int removeTag(net.minecraft.nbt.Tag $$0) {
+        if ($$0 instanceof net.minecraft.nbt.CollectionTag) {
+            int $$3;
+            net.minecraft.nbt.CollectionTag $$1 = (net.minecraft.nbt.CollectionTag)$$0;
+            int $$2 = $$1.size();
+            int n = $$3 = this.index < 0 ? $$2 + this.index : this.index;
+            if (0 <= $$3 && $$3 < $$2) {
+                $$1.remove($$3);
                 return 1;
             }
         }