--- net/minecraft/commands/arguments/NbtPathArgument$AllElementsNode.java
+++ net/minecraft/commands/arguments/NbtPathArgument$AllElementsNode.java
@@ -11,22 +11,22 @@
     public net.minecraft.commands.arguments.NbtPathArgument$AllElementsNode() {
     }
 
-    public void getTag(net.minecraft.nbt.Tag tag, java.util.List<net.minecraft.nbt.Tag> tags) {
-        if (tag instanceof net.minecraft.nbt.CollectionTag) {
-            tags.addAll((java.util.Collection)((net.minecraft.nbt.CollectionTag)tag));
+    public void getTag(net.minecraft.nbt.Tag $$0, java.util.List<net.minecraft.nbt.Tag> $$1) {
+        if ($$0 instanceof net.minecraft.nbt.CollectionTag) {
+            $$1.addAll((java.util.Collection)((net.minecraft.nbt.CollectionTag)$$0));
         }
     }
 
-    public void getOrCreateTag(net.minecraft.nbt.Tag tag, java.util.function.Supplier<net.minecraft.nbt.Tag> supplier, java.util.List<net.minecraft.nbt.Tag> tags) {
-        if (tag instanceof net.minecraft.nbt.CollectionTag) {
-            net.minecraft.nbt.CollectionTag collectionTag = (net.minecraft.nbt.CollectionTag)tag;
-            if (collectionTag.isEmpty()) {
-                net.minecraft.nbt.Tag tag2 = (net.minecraft.nbt.Tag)supplier.get();
-                if (collectionTag.addTag(0, tag2)) {
-                    tags.add((java.lang.Object)tag2);
+    public void getOrCreateTag(net.minecraft.nbt.Tag $$0, java.util.function.Supplier<net.minecraft.nbt.Tag> $$1, java.util.List<net.minecraft.nbt.Tag> $$2) {
+        if ($$0 instanceof net.minecraft.nbt.CollectionTag) {
+            net.minecraft.nbt.CollectionTag $$3 = (net.minecraft.nbt.CollectionTag)$$0;
+            if ($$3.isEmpty()) {
+                net.minecraft.nbt.Tag $$4 = (net.minecraft.nbt.Tag)$$1.get();
+                if ($$3.addTag(0, $$4)) {
+                    $$2.add((java.lang.Object)$$4);
                 }
             } else {
-                tags.addAll((java.util.Collection)collectionTag);
+                $$2.addAll((java.util.Collection)$$3);
             }
         }
     }
@@ -35,37 +35,37 @@
         return new net.minecraft.nbt.ListTag();
     }
 
-    public int setTag(net.minecraft.nbt.Tag tag, java.util.function.Supplier<net.minecraft.nbt.Tag> supplier) {
-        if (tag instanceof net.minecraft.nbt.CollectionTag) {
-            net.minecraft.nbt.CollectionTag collectionTag = (net.minecraft.nbt.CollectionTag)tag;
-            int n = collectionTag.size();
-            if (n == 0) {
-                collectionTag.addTag(0, (net.minecraft.nbt.Tag)supplier.get());
+    public int setTag(net.minecraft.nbt.Tag $$0, java.util.function.Supplier<net.minecraft.nbt.Tag> $$1) {
+        if ($$0 instanceof net.minecraft.nbt.CollectionTag) {
+            net.minecraft.nbt.CollectionTag $$2 = (net.minecraft.nbt.CollectionTag)$$0;
+            int $$3 = $$2.size();
+            if ($$3 == 0) {
+                $$2.addTag(0, (net.minecraft.nbt.Tag)$$1.get());
                 return 1;
             }
-            net.minecraft.nbt.Tag tag2 = (net.minecraft.nbt.Tag)supplier.get();
-            int n2 = n - (int)collectionTag.stream().filter(arg_0 -> ((net.minecraft.nbt.Tag)tag2).equals(arg_0)).count();
-            if (n2 == 0) {
+            net.minecraft.nbt.Tag $$4 = (net.minecraft.nbt.Tag)$$1.get();
+            int $$5 = $$3 - (int)$$2.stream().filter(arg_0 -> ((net.minecraft.nbt.Tag)$$4).equals(arg_0)).count();
+            if ($$5 == 0) {
                 return 0;
             }
-            collectionTag.clear();
-            if (!collectionTag.addTag(0, tag2)) {
+            $$2.clear();
+            if (!$$2.addTag(0, $$4)) {
                 return 0;
             }
-            for (int i = 1; i < n; ++i) {
-                collectionTag.addTag(i, (net.minecraft.nbt.Tag)supplier.get());
+            for (int $$6 = 1; $$6 < $$3; ++$$6) {
+                $$2.addTag($$6, (net.minecraft.nbt.Tag)$$1.get());
             }
-            return n2;
+            return $$5;
         }
         return 0;
     }
 
-    public int removeTag(net.minecraft.nbt.Tag tag) {
-        net.minecraft.nbt.CollectionTag collectionTag;
-        int n;
-        if (tag instanceof net.minecraft.nbt.CollectionTag && (n = (collectionTag = (net.minecraft.nbt.CollectionTag)tag).size()) > 0) {
-            collectionTag.clear();
-            return n;
+    public int removeTag(net.minecraft.nbt.Tag $$0) {
+        net.minecraft.nbt.CollectionTag $$1;
+        int $$2;
+        if ($$0 instanceof net.minecraft.nbt.CollectionTag && ($$2 = ($$1 = (net.minecraft.nbt.CollectionTag)$$0).size()) > 0) {
+            $$1.clear();
+            return $$2;
         }
         return 0;
     }