--- net/minecraft/commands/arguments/NbtPathArgument$MatchRootObjectNode.java
+++ net/minecraft/commands/arguments/NbtPathArgument$MatchRootObjectNode.java
@@ -8,29 +8,29 @@
  {
     public java.util.function.Predicate<net.minecraft.nbt.Tag> predicate;
 
-    public net.minecraft.commands.arguments.NbtPathArgument$MatchRootObjectNode(net.minecraft.nbt.CompoundTag tag) {
-        this.predicate = net.minecraft.commands.arguments.NbtPathArgument.createTagPredicate((net.minecraft.nbt.CompoundTag)tag);
+    public net.minecraft.commands.arguments.NbtPathArgument$MatchRootObjectNode(net.minecraft.nbt.CompoundTag $$0) {
+        this.predicate = net.minecraft.commands.arguments.NbtPathArgument.createTagPredicate((net.minecraft.nbt.CompoundTag)$$0);
     }
 
-    public void getTag(net.minecraft.nbt.Tag tag, java.util.List<net.minecraft.nbt.Tag> tags) {
-        if (tag instanceof net.minecraft.nbt.CompoundTag && this.predicate.test((java.lang.Object)tag)) {
-            tags.add((java.lang.Object)tag);
+    public void getTag(net.minecraft.nbt.Tag $$0, java.util.List<net.minecraft.nbt.Tag> $$1) {
+        if ($$0 instanceof net.minecraft.nbt.CompoundTag && this.predicate.test((java.lang.Object)$$0)) {
+            $$1.add((java.lang.Object)$$0);
         }
     }
 
-    public void getOrCreateTag(net.minecraft.nbt.Tag tag, java.util.function.Supplier<net.minecraft.nbt.Tag> supplier, java.util.List<net.minecraft.nbt.Tag> tags) {
-        this.getTag(tag, tags);
+    public void getOrCreateTag(net.minecraft.nbt.Tag $$0, java.util.function.Supplier<net.minecraft.nbt.Tag> $$1, java.util.List<net.minecraft.nbt.Tag> $$2) {
+        this.getTag($$0, $$2);
     }
 
     public net.minecraft.nbt.Tag createPreferredParentTag() {
         return new net.minecraft.nbt.CompoundTag();
     }
 
-    public int setTag(net.minecraft.nbt.Tag tag, java.util.function.Supplier<net.minecraft.nbt.Tag> supplier) {
+    public int setTag(net.minecraft.nbt.Tag $$0, java.util.function.Supplier<net.minecraft.nbt.Tag> $$1) {
         return 0;
     }
 
-    public int removeTag(net.minecraft.nbt.Tag tag) {
+    public int removeTag(net.minecraft.nbt.Tag $$0) {
         return 0;
     }
 }