--- net/minecraft/commands/arguments/RangeArgument$Ints.java
+++ net/minecraft/commands/arguments/RangeArgument$Ints.java
@@ -11,12 +11,12 @@
     public net.minecraft.commands.arguments.RangeArgument$Ints() {
     }
 
-    public static net.minecraft.advancements.critereon.MinMaxBounds$Ints getRange(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (net.minecraft.advancements.critereon.MinMaxBounds$Ints)context.getArgument(name, net.minecraft.advancements.critereon.MinMaxBounds$Ints.class);
+    public static net.minecraft.advancements.critereon.MinMaxBounds$Ints getRange(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (net.minecraft.advancements.critereon.MinMaxBounds$Ints)$$0.getArgument($$1, net.minecraft.advancements.critereon.MinMaxBounds$Ints.class);
     }
 
-    public net.minecraft.advancements.critereon.MinMaxBounds$Ints parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromReader((com.mojang.brigadier.StringReader)reader);
+    public net.minecraft.advancements.critereon.MinMaxBounds$Ints parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return net.minecraft.advancements.critereon.MinMaxBounds$Ints.fromReader((com.mojang.brigadier.StringReader)$$0);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {