--- net/minecraft/commands/arguments/ResourceLocationArgument.java
+++ net/minecraft/commands/arguments/ResourceLocationArgument.java
@@ -19,29 +19,29 @@
         return new net.minecraft.commands.arguments.ResourceLocationArgument();
     }
 
-    public static net.minecraft.advancements.Advancement getAdvancement(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.commands.arguments.ResourceLocationArgument.getId(context, (java.lang.String)name);
-        net.minecraft.advancements.Advancement advancement = ((net.minecraft.commands.CommandSourceStack)context.getSource()).getServer().getAdvancements().getAdvancement(resourceLocation);
-        if (advancement == null) {
-            throw ERROR_UNKNOWN_ADVANCEMENT.create((java.lang.Object)resourceLocation);
+    public static net.minecraft.advancements.Advancement getAdvancement(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        net.minecraft.resources.ResourceLocation $$2 = net.minecraft.commands.arguments.ResourceLocationArgument.getId($$0, (java.lang.String)$$1);
+        net.minecraft.advancements.Advancement $$3 = ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().getAdvancements().getAdvancement($$2);
+        if ($$3 == null) {
+            throw ERROR_UNKNOWN_ADVANCEMENT.create((java.lang.Object)$$2);
         }
-        return advancement;
+        return $$3;
     }
 
-    public static net.minecraft.world.item.crafting.Recipe<?> getRecipe(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        net.minecraft.world.item.crafting.RecipeManager recipeManager = ((net.minecraft.commands.CommandSourceStack)context.getSource()).getServer().getRecipeManager();
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.commands.arguments.ResourceLocationArgument.getId(context, (java.lang.String)name);
-        return (net.minecraft.world.item.crafting.Recipe)recipeManager.byKey(resourceLocation).orElseThrow(() -> ERROR_UNKNOWN_RECIPE.create((java.lang.Object)resourceLocation));
+    public static net.minecraft.world.item.crafting.Recipe<?> getRecipe(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        net.minecraft.world.item.crafting.RecipeManager $$2 = ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().getRecipeManager();
+        net.minecraft.resources.ResourceLocation $$3 = net.minecraft.commands.arguments.ResourceLocationArgument.getId($$0, (java.lang.String)$$1);
+        return (net.minecraft.world.item.crafting.Recipe)$$2.byKey($$3).orElseThrow(() -> ERROR_UNKNOWN_RECIPE.create((java.lang.Object)$$3));
     }
 
-    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition getPredicate(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        net.minecraft.resources.ResourceLocation resourceLocation = net.minecraft.commands.arguments.ResourceLocationArgument.getId(context, (java.lang.String)name);
-        net.minecraft.world.level.storage.loot.PredicateManager predicateManager = ((net.minecraft.commands.CommandSourceStack)context.getSource()).getServer().getPredicateManager();
-        net.minecraft.world.level.storage.loot.predicates.LootItemCondition lootItemCondition = predicateManager.get(resourceLocation);
-        if (lootItemCondition == null) {
-            throw ERROR_UNKNOWN_PREDICATE.create((java.lang.Object)resourceLocation);
+    public static net.minecraft.world.level.storage.loot.predicates.LootItemCondition getPredicate(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        net.minecraft.resources.ResourceLocation $$2 = net.minecraft.commands.arguments.ResourceLocationArgument.getId($$0, (java.lang.String)$$1);
+        net.minecraft.world.level.storage.loot.PredicateManager $$3 = ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().getPredicateManager();
+        net.minecraft.world.level.storage.loot.predicates.LootItemCondition $$4 = $$3.get($$2);
+        if ($$4 == null) {
+            throw ERROR_UNKNOWN_PREDICATE.create((java.lang.Object)$$2);
         }
-        return lootItemCondition;
+        return $$4;
     }
 
     public static net.minecraft.world.level.storage.loot.functions.LootItemFunction getItemModifier(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
@@ -54,12 +54,12 @@
         return $$4;
     }
 
-    public static net.minecraft.resources.ResourceLocation getId(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (net.minecraft.resources.ResourceLocation)context.getArgument(name, net.minecraft.resources.ResourceLocation.class);
+    public static net.minecraft.resources.ResourceLocation getId(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (net.minecraft.resources.ResourceLocation)$$0.getArgument($$1, net.minecraft.resources.ResourceLocation.class);
     }
 
-    public net.minecraft.resources.ResourceLocation parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return net.minecraft.resources.ResourceLocation.read((com.mojang.brigadier.StringReader)reader);
+    public net.minecraft.resources.ResourceLocation parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return net.minecraft.resources.ResourceLocation.read((com.mojang.brigadier.StringReader)$$0);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {
@@ -70,28 +70,28 @@
         return this.parse(stringReader);
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$3(java.lang.Object attribute) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"item_modifier.unknown", (java.lang.Object[])new java.lang.Object[]{attribute});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$3(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"item_modifier.unknown", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$2(java.lang.Object predicate) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"predicate.unknown", (java.lang.Object[])new java.lang.Object[]{predicate});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$2(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"predicate.unknown", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$1(java.lang.Object recipe) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"recipe.notFound", (java.lang.Object[])new java.lang.Object[]{recipe});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$1(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"recipe.notFound", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object advancement) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"advancement.advancementNotFound", (java.lang.Object[])new java.lang.Object[]{advancement});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"advancement.advancementNotFound", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
     static {
         EXAMPLES = java.util.Arrays.asList((java.lang.Object[])new java.lang.String[]{"foo", "foo:bar", "012"});
-        ERROR_UNKNOWN_ADVANCEMENT = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(advancement -> net.minecraft.network.chat.Component.translatable((java.lang.String)"advancement.advancementNotFound", (java.lang.Object[])new java.lang.Object[]{advancement}));
-        ERROR_UNKNOWN_RECIPE = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(recipe -> net.minecraft.network.chat.Component.translatable((java.lang.String)"recipe.notFound", (java.lang.Object[])new java.lang.Object[]{recipe}));
-        ERROR_UNKNOWN_PREDICATE = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(predicate -> net.minecraft.network.chat.Component.translatable((java.lang.String)"predicate.unknown", (java.lang.Object[])new java.lang.Object[]{predicate}));
-        ERROR_UNKNOWN_ITEM_MODIFIER = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(attribute -> net.minecraft.network.chat.Component.translatable((java.lang.String)"item_modifier.unknown", (java.lang.Object[])new java.lang.Object[]{attribute}));
+        ERROR_UNKNOWN_ADVANCEMENT = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"advancement.advancementNotFound", (java.lang.Object[])new java.lang.Object[]{$$0}));
+        ERROR_UNKNOWN_RECIPE = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"recipe.notFound", (java.lang.Object[])new java.lang.Object[]{$$0}));
+        ERROR_UNKNOWN_PREDICATE = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"predicate.unknown", (java.lang.Object[])new java.lang.Object[]{$$0}));
+        ERROR_UNKNOWN_ITEM_MODIFIER = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"item_modifier.unknown", (java.lang.Object[])new java.lang.Object[]{$$0}));
     }
 }
 