--- net/minecraft/commands/arguments/ResourceOrTagArgument$Info.java
+++ net/minecraft/commands/arguments/ResourceOrTagArgument$Info.java
@@ -1,0 +1,33 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.commands.arguments;
+
+public class net.minecraft.commands.arguments.ResourceOrTagArgument$Info<T>
+implements net.minecraft.commands.synchronization.ArgumentTypeInfo<net.minecraft.commands.arguments.ResourceOrTagArgument<T>, net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template>
+ {
+    public net.minecraft.commands.arguments.ResourceOrTagArgument$Info() {
+    }
+
+    public void serializeToNetwork(net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template $$0, net.minecraft.network.FriendlyByteBuf $$1) {
+        $$1.writeResourceLocation($$0.registryKey.location());
+    }
+
+    public net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template deserializeFromNetwork(net.minecraft.network.FriendlyByteBuf $$0) {
+        net.minecraft.resources.ResourceLocation $$1 = $$0.readResourceLocation();
+        return new net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template(this, net.minecraft.resources.ResourceKey.createRegistryKey((net.minecraft.resources.ResourceLocation)$$1));
+    }
+
+    public void serializeToJson(net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template $$0, com.google.gson.JsonObject $$1) {
+        $$1.addProperty("registry", $$0.registryKey.location().toString());
+    }
+
+    public net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template unpack(net.minecraft.commands.arguments.ResourceOrTagArgument<T> $$0) {
+        return new net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template(this, $$0.registryKey);
+    }
+
+    public /* synthetic */ net.minecraft.commands.synchronization.ArgumentTypeInfo$Template deserializeFromNetwork(net.minecraft.network.FriendlyByteBuf friendlyByteBuf) {
+        return this.deserializeFromNetwork(friendlyByteBuf);
+    }
+}
+