--- net/minecraft/commands/arguments/ResourceOrTagArgument$TagResult.java
+++ net/minecraft/commands/arguments/ResourceOrTagArgument$TagResult.java
@@ -1,0 +1,52 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.commands.arguments;
+
+public class net.minecraft.commands.arguments.ResourceOrTagArgument$TagResult<T>
+extends java.lang.Record
+implements net.minecraft.commands.arguments.ResourceOrTagArgument$Result<T>
+ {
+    public net.minecraft.core.HolderSet$Named<T> tag;
+
+    public net.minecraft.commands.arguments.ResourceOrTagArgument$TagResult(net.minecraft.core.HolderSet$Named<T> $$0) {
+        this.tag = $$0;
+    }
+
+    public com.mojang.datafixers.util.Either<net.minecraft.core.Holder$Reference<T>, net.minecraft.core.HolderSet$Named<T>> unwrap() {
+        return com.mojang.datafixers.util.Either.right((java.lang.Object)this.tag);
+    }
+
+    public <E> java.util.Optional<net.minecraft.commands.arguments.ResourceOrTagArgument$Result<E>> cast(net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<E>> $$0) {
+        return this.tag.key().isFor($$0) ? java.util.Optional.of((java.lang.Object)this) : java.util.Optional.empty();
+    }
+
+    public boolean test(net.minecraft.core.Holder<T> $$0) {
+        return this.tag.contains($$0);
+    }
+
+    public java.lang.String asPrintable() {
+        return "#" + this.tag.key().location();
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.commands.arguments.ResourceOrTagArgument$TagResult.class, "tag", "tag"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.commands.arguments.ResourceOrTagArgument$TagResult.class, "tag", "tag"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.commands.arguments.ResourceOrTagArgument$TagResult.class, "tag", "tag"}, this, $$0);
+    }
+
+    public net.minecraft.core.HolderSet$Named<T> tag() {
+        return this.tag;
+    }
+
+    public /* synthetic */ boolean test(java.lang.Object object) {
+        return this.test((net.minecraft.core.Holder)object);
+    }
+}
+