--- net/minecraft/commands/arguments/ResourceOrTagArgument$Info$Template.java
+++ net/minecraft/commands/arguments/ResourceOrTagArgument$Info$Template.java
@@ -1,0 +1,29 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.commands.arguments;
+
+public class net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template
+implements net.minecraft.commands.synchronization.ArgumentTypeInfo$Template<net.minecraft.commands.arguments.ResourceOrTagArgument<T>>
+ {
+    public net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<T>> registryKey;
+    public /* synthetic */ net.minecraft.commands.arguments.ResourceOrTagArgument$Info this$0;
+
+    public net.minecraft.commands.arguments.ResourceOrTagArgument$Info$Template(net.minecraft.commands.arguments.ResourceOrTagArgument$Info $$0, net.minecraft.resources.ResourceKey<? extends net.minecraft.core.Registry<T>> $$1) {
+        this.this$0 = $$0;
+        this.registryKey = $$1;
+    }
+
+    public net.minecraft.commands.arguments.ResourceOrTagArgument<T> instantiate(net.minecraft.commands.CommandBuildContext $$0) {
+        return new net.minecraft.commands.arguments.ResourceOrTagArgument($$0, this.registryKey);
+    }
+
+    public net.minecraft.commands.synchronization.ArgumentTypeInfo<net.minecraft.commands.arguments.ResourceOrTagArgument<T>, ?> type() {
+        return this.this$0;
+    }
+
+    public /* synthetic */ com.mojang.brigadier.arguments.ArgumentType instantiate(net.minecraft.commands.CommandBuildContext commandBuildContext) {
+        return this.instantiate(commandBuildContext);
+    }
+}
+