--- net/minecraft/commands/arguments/ScoreHolderArgument$SelectorResult.java
+++ net/minecraft/commands/arguments/ScoreHolderArgument$SelectorResult.java
@@ -8,20 +8,20 @@
  {
     public net.minecraft.commands.arguments.selector.EntitySelector selector;
 
-    public net.minecraft.commands.arguments.ScoreHolderArgument$SelectorResult(net.minecraft.commands.arguments.selector.EntitySelector selector) {
-        this.selector = selector;
+    public net.minecraft.commands.arguments.ScoreHolderArgument$SelectorResult(net.minecraft.commands.arguments.selector.EntitySelector $$0) {
+        this.selector = $$0;
     }
 
-    public java.util.Collection<java.lang.String> getNames(net.minecraft.commands.CommandSourceStack source, java.util.function.Supplier<java.util.Collection<java.lang.String>> objectives) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        java.util.List list = this.selector.findEntities(source);
-        if (list.isEmpty()) {
+    public java.util.Collection<java.lang.String> getNames(net.minecraft.commands.CommandSourceStack $$0, java.util.function.Supplier<java.util.Collection<java.lang.String>> $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        java.util.List $$2 = this.selector.findEntities($$0);
+        if ($$2.isEmpty()) {
             throw net.minecraft.commands.arguments.EntityArgument.NO_ENTITIES_FOUND.create();
         }
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        for (net.minecraft.world.entity.Entity entity : list) {
-            arrayList.add((java.lang.Object)entity.getScoreboardName());
+        java.util.ArrayList $$3 = com.google.common.collect.Lists.newArrayList();
+        for (net.minecraft.world.entity.Entity $$4 : $$2) {
+            $$3.add((java.lang.Object)$$4.getScoreboardName());
         }
-        return arrayList;
+        return $$3;
     }
 }
 