--- net/minecraft/commands/arguments/SignedArgument.java
+++ net/minecraft/commands/arguments/SignedArgument.java
@@ -1,1 +1,10 @@
-// Failed to decompile net/minecraft/commands/arguments/SignedArgument.class
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.commands.arguments;
+
+public interface net.minecraft.commands.arguments.SignedArgument<T>
+extends com.mojang.brigadier.arguments.ArgumentType<T>
+ {
+}
+