--- net/minecraft/commands/arguments/SlotArgument.java
+++ net/minecraft/commands/arguments/SlotArgument.java
@@ -17,20 +17,20 @@
         return new net.minecraft.commands.arguments.SlotArgument();
     }
 
-    public static int getSlot(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (java.lang.Integer)context.getArgument(name, java.lang.Integer.class);
+    public static int getSlot(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (java.lang.Integer)$$0.getArgument($$1, java.lang.Integer.class);
     }
 
-    public java.lang.Integer parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        java.lang.String string = reader.readUnquotedString();
-        if (!SLOTS.containsKey((java.lang.Object)string)) {
-            throw ERROR_UNKNOWN_SLOT.create((java.lang.Object)string);
+    public java.lang.Integer parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        java.lang.String $$1 = $$0.readUnquotedString();
+        if (!SLOTS.containsKey((java.lang.Object)$$1)) {
+            throw ERROR_UNKNOWN_SLOT.create((java.lang.Object)$$1);
         }
-        return (java.lang.Integer)SLOTS.get((java.lang.Object)string);
+        return (java.lang.Integer)SLOTS.get((java.lang.Object)$$1);
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)SLOTS.keySet(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder);
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)SLOTS.keySet(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {
@@ -41,87 +41,73 @@
         return this.parse(stringReader);
     }
 
-    public static /* synthetic */ void lambda$static$1(java.util.HashMap slots) {
-        int n;
-        for (n = 0;
- n < 54; ++n) {
-            slots.put((java.lang.Object)("container." + n), (java.lang.Object)n);
+    public static /* synthetic */ void lambda$static$1(java.util.HashMap $$0) {
+        for (int $$1 = 0; $$1 < 54; ++$$1) {
+            $$0.put((java.lang.Object)("container." + $$1), (java.lang.Object)$$1);
         }
-        for (n = 0;
- n < 9; ++n) {
-            slots.put((java.lang.Object)("hotbar." + n), (java.lang.Object)n);
+        for (int $$2 = 0; $$2 < 9; ++$$2) {
+            $$0.put((java.lang.Object)("hotbar." + $$2), (java.lang.Object)$$2);
         }
-        for (n = 0;
- n < 27; ++n) {
-            slots.put((java.lang.Object)("inventory." + n), (java.lang.Object)(9 + n));
+        for (int $$3 = 0; $$3 < 27; ++$$3) {
+            $$0.put((java.lang.Object)("inventory." + $$3), (java.lang.Object)(9 + $$3));
         }
-        for (n = 0;
- n < 27; ++n) {
-            slots.put((java.lang.Object)("enderchest." + n), (java.lang.Object)(200 + n));
+        for (int $$4 = 0; $$4 < 27; ++$$4) {
+            $$0.put((java.lang.Object)("enderchest." + $$4), (java.lang.Object)(200 + $$4));
         }
-        for (n = 0;
- n < 8; ++n) {
-            slots.put((java.lang.Object)("villager." + n), (java.lang.Object)(300 + n));
+        for (int $$5 = 0; $$5 < 8; ++$$5) {
+            $$0.put((java.lang.Object)("villager." + $$5), (java.lang.Object)(300 + $$5));
         }
-        for (n = 0;
- n < 15; ++n) {
-            slots.put((java.lang.Object)("horse." + n), (java.lang.Object)(500 + n));
+        for (int $$6 = 0; $$6 < 15; ++$$6) {
+            $$0.put((java.lang.Object)("horse." + $$6), (java.lang.Object)(500 + $$6));
         }
-        slots.put((java.lang.Object)"weapon", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
-        slots.put((java.lang.Object)"weapon.mainhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
-        slots.put((java.lang.Object)"weapon.offhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.OFFHAND.getIndex(98));
-        slots.put((java.lang.Object)"armor.head", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.HEAD.getIndex(100));
-        slots.put((java.lang.Object)"armor.chest", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.CHEST.getIndex(100));
-        slots.put((java.lang.Object)"armor.legs", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.LEGS.getIndex(100));
-        slots.put((java.lang.Object)"armor.feet", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.FEET.getIndex(100));
-        slots.put((java.lang.Object)"horse.saddle", (java.lang.Object)400);
-        slots.put((java.lang.Object)"horse.armor", (java.lang.Object)401);
-        slots.put((java.lang.Object)"horse.chest", (java.lang.Object)499);
+        $$0.put((java.lang.Object)"weapon", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
+        $$0.put((java.lang.Object)"weapon.mainhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
+        $$0.put((java.lang.Object)"weapon.offhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.OFFHAND.getIndex(98));
+        $$0.put((java.lang.Object)"armor.head", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.HEAD.getIndex(100));
+        $$0.put((java.lang.Object)"armor.chest", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.CHEST.getIndex(100));
+        $$0.put((java.lang.Object)"armor.legs", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.LEGS.getIndex(100));
+        $$0.put((java.lang.Object)"armor.feet", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.FEET.getIndex(100));
+        $$0.put((java.lang.Object)"horse.saddle", (java.lang.Object)400);
+        $$0.put((java.lang.Object)"horse.armor", (java.lang.Object)401);
+        $$0.put((java.lang.Object)"horse.chest", (java.lang.Object)499);
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object slot) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"slot.unknown", (java.lang.Object[])new java.lang.Object[]{slot});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"slot.unknown", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
     static {
         EXAMPLES = java.util.Arrays.asList((java.lang.Object[])new java.lang.String[]{"container.5", "12", "weapon"});
-        ERROR_UNKNOWN_SLOT = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(slot -> net.minecraft.network.chat.Component.translatable((java.lang.String)"slot.unknown", (java.lang.Object[])new java.lang.Object[]{slot}));
-        SLOTS = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), slots -> {
-            int n;
-            for (n = 0;
- n < 54; ++n) {
-                slots.put((java.lang.Object)("container." + n), (java.lang.Object)n);
+        ERROR_UNKNOWN_SLOT = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"slot.unknown", (java.lang.Object[])new java.lang.Object[]{$$0}));
+        SLOTS = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), $$0 -> {
+            for (int $$1 = 0; $$1 < 54; ++$$1) {
+                $$0.put((java.lang.Object)("container." + $$1), (java.lang.Object)$$1);
             }
-            for (n = 0;
- n < 9; ++n) {
-                slots.put((java.lang.Object)("hotbar." + n), (java.lang.Object)n);
+            for (int $$2 = 0; $$2 < 9; ++$$2) {
+                $$0.put((java.lang.Object)("hotbar." + $$2), (java.lang.Object)$$2);
             }
-            for (n = 0;
- n < 27; ++n) {
-                slots.put((java.lang.Object)("inventory." + n), (java.lang.Object)(9 + n));
+            for (int $$3 = 0; $$3 < 27; ++$$3) {
+                $$0.put((java.lang.Object)("inventory." + $$3), (java.lang.Object)(9 + $$3));
             }
-            for (n = 0;
- n < 27; ++n) {
-                slots.put((java.lang.Object)("enderchest." + n), (java.lang.Object)(200 + n));
+            for (int $$4 = 0; $$4 < 27; ++$$4) {
+                $$0.put((java.lang.Object)("enderchest." + $$4), (java.lang.Object)(200 + $$4));
             }
-            for (n = 0;
- n < 8; ++n) {
-                slots.put((java.lang.Object)("villager." + n), (java.lang.Object)(300 + n));
+            for (int $$5 = 0; $$5 < 8; ++$$5) {
+                $$0.put((java.lang.Object)("villager." + $$5), (java.lang.Object)(300 + $$5));
             }
-            for (n = 0;
- n < 15; ++n) {
-                slots.put((java.lang.Object)("horse." + n), (java.lang.Object)(500 + n));
+            for (int $$6 = 0; $$6 < 15; ++$$6) {
+                $$0.put((java.lang.Object)("horse." + $$6), (java.lang.Object)(500 + $$6));
             }
-            slots.put((java.lang.Object)"weapon", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
-            slots.put((java.lang.Object)"weapon.mainhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
-            slots.put((java.lang.Object)"weapon.offhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.OFFHAND.getIndex(98));
-            slots.put((java.lang.Object)"armor.head", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.HEAD.getIndex(100));
-            slots.put((java.lang.Object)"armor.chest", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.CHEST.getIndex(100));
-            slots.put((java.lang.Object)"armor.legs", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.LEGS.getIndex(100));
-            slots.put((java.lang.Object)"armor.feet", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.FEET.getIndex(100));
-            slots.put((java.lang.Object)"horse.saddle", (java.lang.Object)400);
-            slots.put((java.lang.Object)"horse.armor", (java.lang.Object)401);
-            slots.put((java.lang.Object)"horse.chest", (java.lang.Object)499);
+            $$0.put((java.lang.Object)"weapon", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
+            $$0.put((java.lang.Object)"weapon.mainhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.MAINHAND.getIndex(98));
+            $$0.put((java.lang.Object)"weapon.offhand", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.OFFHAND.getIndex(98));
+            $$0.put((java.lang.Object)"armor.head", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.HEAD.getIndex(100));
+            $$0.put((java.lang.Object)"armor.chest", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.CHEST.getIndex(100));
+            $$0.put((java.lang.Object)"armor.legs", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.LEGS.getIndex(100));
+            $$0.put((java.lang.Object)"armor.feet", (java.lang.Object)net.minecraft.world.entity.EquipmentSlot.FEET.getIndex(100));
+            $$0.put((java.lang.Object)"horse.saddle", (java.lang.Object)400);
+            $$0.put((java.lang.Object)"horse.armor", (java.lang.Object)401);
+            $$0.put((java.lang.Object)"horse.chest", (java.lang.Object)499);
         });
     }
 }