--- net/minecraft/commands/arguments/TeamArgument.java
+++ net/minecraft/commands/arguments/TeamArgument.java
@@ -16,23 +16,23 @@
         return new net.minecraft.commands.arguments.TeamArgument();
     }
 
-    public static net.minecraft.world.scores.PlayerTeam getTeam(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        java.lang.String string = (java.lang.String)context.getArgument(name, java.lang.String.class);
-        net.minecraft.server.ServerScoreboard serverScoreboard = ((net.minecraft.commands.CommandSourceStack)context.getSource()).getServer().getScoreboard();
-        net.minecraft.world.scores.PlayerTeam playerTeam = serverScoreboard.getPlayerTeam(string);
-        if (playerTeam == null) {
-            throw ERROR_TEAM_NOT_FOUND.create((java.lang.Object)string);
+    public static net.minecraft.world.scores.PlayerTeam getTeam(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        java.lang.String $$2 = (java.lang.String)$$0.getArgument($$1, java.lang.String.class);
+        net.minecraft.server.ServerScoreboard $$3 = ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getServer().getScoreboard();
+        net.minecraft.world.scores.PlayerTeam $$4 = $$3.getPlayerTeam($$2);
+        if ($$4 == null) {
+            throw ERROR_TEAM_NOT_FOUND.create((java.lang.Object)$$2);
         }
-        return playerTeam;
+        return $$4;
     }
 
-    public java.lang.String parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return reader.readUnquotedString();
+    public java.lang.String parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return $$0.readUnquotedString();
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        if (context.getSource() instanceof net.minecraft.commands.SharedSuggestionProvider) {
-            return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)((net.minecraft.commands.SharedSuggestionProvider)context.getSource()).getAllTeams(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder);
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        if ($$0.getSource() instanceof net.minecraft.commands.SharedSuggestionProvider) {
+            return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)((net.minecraft.commands.SharedSuggestionProvider)$$0.getSource()).getAllTeams(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1);
         }
         return com.mojang.brigadier.suggestion.Suggestions.empty();
     }
@@ -45,13 +45,13 @@
         return this.parse(stringReader);
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object team) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"team.notFound", (java.lang.Object[])new java.lang.Object[]{team});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"team.notFound", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
     static {
         EXAMPLES = java.util.Arrays.asList((java.lang.Object[])new java.lang.String[]{"foo", "123"});
-        ERROR_TEAM_NOT_FOUND = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(team -> net.minecraft.network.chat.Component.translatable((java.lang.String)"team.notFound", (java.lang.Object[])new java.lang.Object[]{team}));
+        ERROR_TEAM_NOT_FOUND = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"team.notFound", (java.lang.Object[])new java.lang.Object[]{$$0}));
     }
 }
 