--- net/minecraft/commands/arguments/TimeArgument.java
+++ net/minecraft/commands/arguments/TimeArgument.java
@@ -18,29 +18,29 @@
         return new net.minecraft.commands.arguments.TimeArgument();
     }
 
-    public java.lang.Integer parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        float f = reader.readFloat();
-        java.lang.String string = reader.readUnquotedString();
-        int n = UNITS.getOrDefault((java.lang.Object)string, 0);
-        if (n == 0) {
+    public java.lang.Integer parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        float $$1 = $$0.readFloat();
+        java.lang.String $$2 = $$0.readUnquotedString();
+        int $$3 = UNITS.getOrDefault((java.lang.Object)$$2, 0);
+        if ($$3 == 0) {
             throw ERROR_INVALID_UNIT.create();
         }
-        int n2 = java.lang.Math.round((float)(f * (float)n));
-        if (n2 < 0) {
-            throw ERROR_INVALID_TICK_COUNT.create((java.lang.Object)n2);
+        int $$4 = java.lang.Math.round((float)($$1 * (float)$$3));
+        if ($$4 < 0) {
+            throw ERROR_INVALID_TICK_COUNT.create((java.lang.Object)$$4);
         }
-        return n2;
+        return $$4;
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        com.mojang.brigadier.StringReader stringReader = new com.mojang.brigadier.StringReader(builder.getRemaining());
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        com.mojang.brigadier.StringReader $$2 = new com.mojang.brigadier.StringReader($$1.getRemaining());
         try {
-            stringReader.readFloat();
+            $$2.readFloat();
         }
-        catch (com.mojang.brigadier.exceptions.CommandSyntaxException commandSyntaxException) {
-            return builder.buildFuture();
+        catch (com.mojang.brigadier.exceptions.CommandSyntaxException $$3) {
+            return $$1.buildFuture();
         }
-        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)UNITS.keySet(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder.createOffset(builder.getStart() + stringReader.getCursor()));
+        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.lang.Iterable)UNITS.keySet(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1.createOffset($$1.getStart() + $$2.getCursor()));
     }
 
     public java.util.Collection<java.lang.String> getExamples() {
@@ -51,14 +51,14 @@
         return this.parse(stringReader);
     }
 
-    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object tickCount) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.time.invalid_tick_count", (java.lang.Object[])new java.lang.Object[]{tickCount});
+    public static /* synthetic */ com.mojang.brigadier.Message lambda$static$0(java.lang.Object $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.time.invalid_tick_count", (java.lang.Object[])new java.lang.Object[]{$$0});
     }
 
     static {
         EXAMPLES = java.util.Arrays.asList((java.lang.Object[])new java.lang.String[]{"0d", "0s", "0t", "0"});
         ERROR_INVALID_UNIT = new com.mojang.brigadier.exceptions.SimpleCommandExceptionType((com.mojang.brigadier.Message)net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.time.invalid_unit"));
-        ERROR_INVALID_TICK_COUNT = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType(tickCount -> net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.time.invalid_tick_count", (java.lang.Object[])new java.lang.Object[]{tickCount}));
+        ERROR_INVALID_TICK_COUNT = new com.mojang.brigadier.exceptions.DynamicCommandExceptionType($$0 -> net.minecraft.network.chat.Component.translatable((java.lang.String)"argument.time.invalid_tick_count", (java.lang.Object[])new java.lang.Object[]{$$0}));
         UNITS = new it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap();
         UNITS.put((java.lang.Object)"d", 24000);
         UNITS.put((java.lang.Object)"s", 20);