--- net/minecraft/commands/arguments/UuidArgument.java
+++ net/minecraft/commands/arguments/UuidArgument.java
@@ -13,23 +13,23 @@
     public net.minecraft.commands.arguments.UuidArgument() {
     }
 
-    public static java.util.UUID getUuid(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (java.util.UUID)context.getArgument(name, java.util.UUID.class);
+    public static java.util.UUID getUuid(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (java.util.UUID)$$0.getArgument($$1, java.util.UUID.class);
     }
 
     public static net.minecraft.commands.arguments.UuidArgument uuid() {
         return new net.minecraft.commands.arguments.UuidArgument();
     }
 
-    public java.util.UUID parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        java.lang.String string = reader.getRemaining();
-        java.util.regex.Matcher matcher = ALLOWED_CHARACTERS.matcher((java.lang.CharSequence)string);
-        if (matcher.find()) {
-            java.lang.String string2 = matcher.group(1);
+    public java.util.UUID parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        java.lang.String $$1 = $$0.getRemaining();
+        java.util.regex.Matcher $$2 = ALLOWED_CHARACTERS.matcher((java.lang.CharSequence)$$1);
+        if ($$2.find()) {
+            java.lang.String $$3 = $$2.group(1);
             try {
-                java.util.UUID uUID = java.util.UUID.fromString((java.lang.String)string2);
-                reader.setCursor(reader.getCursor() + string2.length());
-                return uUID;
+                java.util.UUID $$4 = java.util.UUID.fromString((java.lang.String)$$3);
+                $$0.setCursor($$0.getCursor() + $$3.length());
+                return $$4;
             }
             catch (java.lang.IllegalArgumentException illegalArgumentException) {
                 // empty catch block