--- net/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
+++ net/minecraft/commands/arguments/blocks/BlockPredicateArgument.java
@@ -17,20 +17,20 @@
         return new net.minecraft.commands.arguments.blocks.BlockPredicateArgument($$0);
     }
 
-    public net.minecraft.commands.arguments.blocks.BlockPredicateArgument$Result parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return net.minecraft.commands.arguments.blocks.BlockPredicateArgument.parse((net.minecraft.core.HolderLookup)this.blocks, (com.mojang.brigadier.StringReader)reader);
+    public net.minecraft.commands.arguments.blocks.BlockPredicateArgument$Result parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return net.minecraft.commands.arguments.blocks.BlockPredicateArgument.parse((net.minecraft.core.HolderLookup)this.blocks, (com.mojang.brigadier.StringReader)$$0);
     }
 
     public static net.minecraft.commands.arguments.blocks.BlockPredicateArgument$Result parse(net.minecraft.core.HolderLookup<net.minecraft.world.level.block.Block> $$02, com.mojang.brigadier.StringReader $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
         return (net.minecraft.commands.arguments.blocks.BlockPredicateArgument$Result)net.minecraft.commands.arguments.blocks.BlockStateParser.parseForTesting($$02, (com.mojang.brigadier.StringReader)$$1, (boolean)true).map($$0 -> new net.minecraft.commands.arguments.blocks.BlockPredicateArgument$BlockPredicate($$0.blockState(), $$0.properties().keySet(), $$0.nbt()), $$0 -> new net.minecraft.commands.arguments.blocks.BlockPredicateArgument$TagPredicate($$0.tag(), $$0.vagueProperties(), $$0.nbt()));
     }
 
-    public static java.util.function.Predicate<net.minecraft.world.level.block.state.pattern.BlockInWorld> getBlockPredicate(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return (java.util.function.Predicate)context.getArgument(name, net.minecraft.commands.arguments.blocks.BlockPredicateArgument$Result.class);
+    public static java.util.function.Predicate<net.minecraft.world.level.block.state.pattern.BlockInWorld> getBlockPredicate(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return (java.util.function.Predicate)$$0.getArgument($$1, net.minecraft.commands.arguments.blocks.BlockPredicateArgument$Result.class);
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        return net.minecraft.commands.arguments.blocks.BlockStateParser.fillSuggestions((net.minecraft.core.HolderLookup)this.blocks, (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder, (boolean)true, (boolean)true);
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        return net.minecraft.commands.arguments.blocks.BlockStateParser.fillSuggestions((net.minecraft.core.HolderLookup)this.blocks, (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1, (boolean)true, (boolean)true);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {