--- net/minecraft/world/entity/item/FallingBlockEntity.java
+++ net/minecraft/world/entity/item/FallingBlockEntity.java
@@ -25,15 +25,15 @@
         this.fallDamageMax = 40;
     }
 
-    public net.minecraft.world.entity.item.FallingBlockEntity(net.minecraft.world.level.Level level, double x, double y, double z, net.minecraft.world.level.block.state.BlockState state) {
-        this(net.minecraft.world.entity.EntityType.FALLING_BLOCK, level);
-        this.blockState = state;
+    public net.minecraft.world.entity.item.FallingBlockEntity(net.minecraft.world.level.Level $$0, double $$1, double $$2, double $$3, net.minecraft.world.level.block.state.BlockState $$4) {
+        this(net.minecraft.world.entity.EntityType.FALLING_BLOCK, $$0);
+        this.blockState = $$4;
         this.blocksBuilding = true;
-        this.setPos(x, y, z);
+        this.setPos($$1, $$2, $$3);
         this.setDeltaMovement(net.minecraft.world.phys.Vec3.ZERO);
-        this.xo = x;
-        this.yo = y;
-        this.zo = z;
+        this.xo = $$1;
+        this.yo = $$2;
+        this.zo = $$3;
         this.setStartPos(this.blockPosition());
     }
 
@@ -48,8 +48,8 @@
         return false;
     }
 
-    public void setStartPos(net.minecraft.core.BlockPos startPos) {
-        this.entityData.set(DATA_START_POS, (java.lang.Object)startPos);
+    public void setStartPos(net.minecraft.core.BlockPos $$0) {
+        this.entityData.set(DATA_START_POS, (java.lang.Object)$$0);
     }
 
     public net.minecraft.core.BlockPos getStartPos() {
@@ -187,33 +187,33 @@
         return false;
     }
 
-    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        compound.put("BlockState", (net.minecraft.nbt.Tag)net.minecraft.nbt.NbtUtils.writeBlockState((net.minecraft.world.level.block.state.BlockState)this.blockState));
-        compound.putInt("Time", this.time);
-        compound.putBoolean("DropItem", this.dropItem);
-        compound.putBoolean("HurtEntities", this.hurtEntities);
-        compound.putFloat("FallHurtAmount", this.fallDamagePerDistance);
-        compound.putInt("FallHurtMax", this.fallDamageMax);
+    public void addAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        $$0.put("BlockState", (net.minecraft.nbt.Tag)net.minecraft.nbt.NbtUtils.writeBlockState((net.minecraft.world.level.block.state.BlockState)this.blockState));
+        $$0.putInt("Time", this.time);
+        $$0.putBoolean("DropItem", this.dropItem);
+        $$0.putBoolean("HurtEntities", this.hurtEntities);
+        $$0.putFloat("FallHurtAmount", this.fallDamagePerDistance);
+        $$0.putInt("FallHurtMax", this.fallDamageMax);
         if (this.blockData != null) {
-            compound.put("TileEntityData", (net.minecraft.nbt.Tag)this.blockData);
+            $$0.put("TileEntityData", (net.minecraft.nbt.Tag)this.blockData);
         }
     }
 
-    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag compound) {
-        this.blockState = net.minecraft.nbt.NbtUtils.readBlockState((net.minecraft.nbt.CompoundTag)compound.getCompound("BlockState"));
-        this.time = compound.getInt("Time");
-        if (compound.contains("HurtEntities", 99)) {
-            this.hurtEntities = compound.getBoolean("HurtEntities");
-            this.fallDamagePerDistance = compound.getFloat("FallHurtAmount");
-            this.fallDamageMax = compound.getInt("FallHurtMax");
+    public void readAdditionalSaveData(net.minecraft.nbt.CompoundTag $$0) {
+        this.blockState = net.minecraft.nbt.NbtUtils.readBlockState((net.minecraft.core.HolderLookup)this.level.holderLookup(net.minecraft.core.Registry.BLOCK_REGISTRY), (net.minecraft.nbt.CompoundTag)$$0.getCompound("BlockState"));
+        this.time = $$0.getInt("Time");
+        if ($$0.contains("HurtEntities", 99)) {
+            this.hurtEntities = $$0.getBoolean("HurtEntities");
+            this.fallDamagePerDistance = $$0.getFloat("FallHurtAmount");
+            this.fallDamageMax = $$0.getInt("FallHurtMax");
         } else if (this.blockState.is(net.minecraft.tags.BlockTags.ANVIL)) {
             this.hurtEntities = true;
         }
-        if (compound.contains("DropItem", 99)) {
-            this.dropItem = compound.getBoolean("DropItem");
+        if ($$0.contains("DropItem", 99)) {
+            this.dropItem = $$0.getBoolean("DropItem");
         }
-        if (compound.contains("TileEntityData", 10)) {
-            this.blockData = compound.getCompound("TileEntityData");
+        if ($$0.contains("TileEntityData", 10)) {
+            this.blockData = $$0.getCompound("TileEntityData");
         }
         if (this.blockState.isAir()) {
             this.blockState = net.minecraft.world.level.block.Blocks.SAND.defaultBlockState();
@@ -230,9 +230,9 @@
         return false;
     }
 
-    public void fillCrashReportCategory(net.minecraft.CrashReportCategory category) {
-        super.fillCrashReportCategory(category);
-        category.setDetail("Immitating BlockState", (java.lang.Object)this.blockState.toString());
+    public void fillCrashReportCategory(net.minecraft.CrashReportCategory $$0) {
+        super.fillCrashReportCategory($$0);
+        $$0.setDetail("Immitating BlockState", (java.lang.Object)this.blockState.toString());
     }
 
     public net.minecraft.world.level.block.state.BlockState getBlockState() {
@@ -243,7 +243,7 @@
         return true;
     }
 
-    public net.minecraft.network.protocol.Packet<?> getAddEntityPacket() {
+    public net.minecraft.network.protocol.Packet<net.minecraft.network.protocol.game.ClientGamePacketListener> getAddEntityPacket() {
         return new net.minecraft.network.protocol.game.ClientboundAddEntityPacket((net.minecraft.world.entity.Entity)this, net.minecraft.world.level.block.Block.getId((net.minecraft.world.level.block.state.BlockState)this.getBlockState()));
     }
 