--- net/minecraft/world/level/dimension/DimensionType.java
+++ net/minecraft/world/level/dimension/DimensionType.java
@@ -67,27 +67,27 @@
     }
 
     @java.lang.Deprecated
-    public static com.mojang.serialization.DataResult<net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level>> parseLegacy(com.mojang.serialization.Dynamic<?> dynamic) {
-        java.util.Optional optional = dynamic.asNumber().result();
-        if (optional.isPresent()) {
-            int n = ((java.lang.Number)optional.get()).intValue();
-            if (n == -1) {
+    public static com.mojang.serialization.DataResult<net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level>> parseLegacy(com.mojang.serialization.Dynamic<?> $$0) {
+        java.util.Optional $$1 = $$0.asNumber().result();
+        if ($$1.isPresent()) {
+            int $$2 = ((java.lang.Number)$$1.get()).intValue();
+            if ($$2 == -1) {
                 return com.mojang.serialization.DataResult.success((java.lang.Object)net.minecraft.world.level.Level.NETHER);
             }
-            if (n == 0) {
+            if ($$2 == 0) {
                 return com.mojang.serialization.DataResult.success((java.lang.Object)net.minecraft.world.level.Level.OVERWORLD);
             }
-            if (n == 1) {
+            if ($$2 == 1) {
                 return com.mojang.serialization.DataResult.success((java.lang.Object)net.minecraft.world.level.Level.END);
             }
         }
-        return net.minecraft.world.level.Level.RESOURCE_KEY_CODEC.parse(dynamic);
+        return net.minecraft.world.level.Level.RESOURCE_KEY_CODEC.parse($$0);
     }
 
-    public static double getTeleportationScale(net.minecraft.world.level.dimension.DimensionType firstType, net.minecraft.world.level.dimension.DimensionType secondType) {
-        double d = firstType.coordinateScale();
-        double d2 = secondType.coordinateScale();
-        return d / d2;
+    public static double getTeleportationScale(net.minecraft.world.level.dimension.DimensionType $$0, net.minecraft.world.level.dimension.DimensionType $$1) {
+        double $$2 = $$0.coordinateScale();
+        double $$3 = $$1.coordinateScale();
+        return $$2 / $$3;
     }
 
     public static java.nio.file.Path getStorageFolder(net.minecraft.resources.ResourceKey<net.minecraft.world.level.Level> $$0, java.nio.file.Path $$1) {
@@ -107,14 +107,14 @@
         return this.fixedTime.isPresent();
     }
 
-    public float timeOfDay(long dayTime) {
-        double d = net.minecraft.util.Mth.frac((double)((double)this.fixedTime.orElse(dayTime) / 24000.0 - 0.25));
-        double d2 = 0.5 - java.lang.Math.cos((double)(d * java.lang.Math.PI)) / 2.0;
-        return (float)(d * 2.0 + d2) / 3.0f;
+    public float timeOfDay(long $$0) {
+        double $$1 = net.minecraft.util.Mth.frac((double)((double)this.fixedTime.orElse($$0) / 24000.0 - 0.25));
+        double $$2 = 0.5 - java.lang.Math.cos((double)($$1 * java.lang.Math.PI)) / 2.0;
+        return (float)($$1 * 2.0 + $$2) / 3.0f;
     }
 
-    public int moonPhase(long dayTime) {
-        return (int)(dayTime / 24000L % 8L + 8L) % 8;
+    public int moonPhase(long $$0) {
+        return (int)($$0 / 24000L % 8L + 8L) % 8;
     }
 
     public boolean piglinSafe() {