--- net/minecraft/world/damagesource/BadRespawnPointDamage.java
+++ net/minecraft/world/damagesource/BadRespawnPointDamage.java
@@ -12,9 +12,9 @@
         this.setExplosion();
     }
 
-    public net.minecraft.network.chat.Component getLocalizedDeathMessage(net.minecraft.world.entity.LivingEntity livingEntity) {
-        net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.ComponentUtils.wrapInSquareBrackets((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"death.attack.badRespawnPoint.link")).withStyle(componentStyle -> componentStyle.withClickEvent(new net.minecraft.network.chat.ClickEvent(net.minecraft.network.chat.ClickEvent$Action.OPEN_URL, "https://bugs.mojang.com/browse/MCPE-28723")).withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)"MCPE-28723"))));
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"death.attack.badRespawnPoint.message", (java.lang.Object[])new java.lang.Object[]{livingEntity.getDisplayName(), mutableComponent});
+    public net.minecraft.network.chat.Component getLocalizedDeathMessage(net.minecraft.world.entity.LivingEntity $$02) {
+        net.minecraft.network.chat.MutableComponent $$1 = net.minecraft.network.chat.ComponentUtils.wrapInSquareBrackets((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"death.attack.badRespawnPoint.link")).withStyle($$0 -> $$0.withClickEvent(new net.minecraft.network.chat.ClickEvent(net.minecraft.network.chat.ClickEvent$Action.OPEN_URL, "https://bugs.mojang.com/browse/MCPE-28723")).withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)net.minecraft.network.chat.Component.literal((java.lang.String)"MCPE-28723"))));
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"death.attack.badRespawnPoint.message", (java.lang.Object[])new java.lang.Object[]{$$02.getDisplayName(), $$1});
     }
 }
 