--- net/minecraft/world/damagesource/CombatTracker.java
+++ net/minecraft/world/damagesource/CombatTracker.java
@@ -17,9 +17,9 @@
     @javax.annotation.Nullable
     public java.lang.String nextLocation;
 
-    public net.minecraft.world.damagesource.CombatTracker(net.minecraft.world.entity.LivingEntity mob) {
+    public net.minecraft.world.damagesource.CombatTracker(net.minecraft.world.entity.LivingEntity $$0) {
         this.entries = com.google.common.collect.Lists.newArrayList();
-        this.mob = mob;
+        this.mob = $$0;
     }
 
     public void prepareForDamage() {
@@ -33,14 +33,14 @@
         }
     }
 
-    public void recordDamage(net.minecraft.world.damagesource.DamageSource damageSrc, float health, float damageAmount) {
+    public void recordDamage(net.minecraft.world.damagesource.DamageSource $$0, float $$1, float $$2) {
         this.recheckStatus();
         this.prepareForDamage();
-        net.minecraft.world.damagesource.CombatEntry combatEntry = new net.minecraft.world.damagesource.CombatEntry(damageSrc, this.mob.tickCount, health, damageAmount, this.nextLocation, this.mob.fallDistance);
-        this.entries.add((java.lang.Object)combatEntry);
+        net.minecraft.world.damagesource.CombatEntry $$3 = new net.minecraft.world.damagesource.CombatEntry($$0, this.mob.tickCount, $$1, $$2, this.nextLocation, this.mob.fallDistance);
+        this.entries.add((java.lang.Object)$$3);
         this.lastDamageTime = this.mob.tickCount;
         this.takingDamage = true;
-        if (combatEntry.isCombatRelated() && !this.inCombat && this.mob.isAlive()) {
+        if ($$3.isCombatRelated() && !this.inCombat && this.mob.isAlive()) {
             this.inCombat = true;
             this.combatEndTime = this.combatStartTime = this.mob.tickCount;
             this.mob.onEnterCombat();
@@ -134,8 +134,8 @@
         return null;
     }
 
-    public java.lang.String getFallLocation(net.minecraft.world.damagesource.CombatEntry entry) {
-        return entry.getLocation() == null ? "generic" : entry.getLocation();
+    public java.lang.String getFallLocation(net.minecraft.world.damagesource.CombatEntry $$0) {
+        return $$0.getLocation() == null ? "generic" : $$0.getLocation();
     }
 
     public boolean isTakingDamage() {