--- net/minecraft/world/damagesource/CombatRules.java
+++ net/minecraft/world/damagesource/CombatRules.java
@@ -14,15 +14,15 @@
     public net.minecraft.world.damagesource.CombatRules() {
     }
 
-    public static float getDamageAfterAbsorb(float damage, float totalArmor, float toughnessAttribute) {
-        float f = 2.0f + toughnessAttribute / 4.0f;
-        float f2 = net.minecraft.util.Mth.clamp((float)(totalArmor - damage / f), (float)(totalArmor * 0.2f), (float)20.0f);
-        return damage * (1.0f - f2 / 25.0f);
+    public static float getDamageAfterAbsorb(float $$0, float $$1, float $$2) {
+        float $$3 = 2.0f + $$2 / 4.0f;
+        float $$4 = net.minecraft.util.Mth.clamp((float)($$1 - $$0 / $$3), (float)($$1 * 0.2f), (float)20.0f);
+        return $$0 * (1.0f - $$4 / 25.0f);
     }
 
-    public static float getDamageAfterMagicAbsorb(float damage, float enchantModifiers) {
-        float f = net.minecraft.util.Mth.clamp((float)enchantModifiers, (float)0.0f, (float)20.0f);
-        return damage * (1.0f - f / 25.0f);
+    public static float getDamageAfterMagicAbsorb(float $$0, float $$1) {
+        float $$2 = net.minecraft.util.Mth.clamp((float)$$1, (float)0.0f, (float)20.0f);
+        return $$0 * (1.0f - $$2 / 25.0f);
     }
 }
 