--- net/minecraft/world/damagesource/DamageSource.java
+++ net/minecraft/world/damagesource/DamageSource.java
@@ -44,64 +44,64 @@
     public boolean noAggro;
     public java.lang.String msgId;
 
-    public static net.minecraft.world.damagesource.DamageSource sting(net.minecraft.world.entity.LivingEntity bee) {
-        return new net.minecraft.world.damagesource.EntityDamageSource("sting", (net.minecraft.world.entity.Entity)bee);
+    public static net.minecraft.world.damagesource.DamageSource sting(net.minecraft.world.entity.LivingEntity $$0) {
+        return new net.minecraft.world.damagesource.EntityDamageSource("sting", (net.minecraft.world.entity.Entity)$$0);
     }
 
-    public static net.minecraft.world.damagesource.DamageSource mobAttack(net.minecraft.world.entity.LivingEntity mob) {
-        return new net.minecraft.world.damagesource.EntityDamageSource("mob", (net.minecraft.world.entity.Entity)mob);
+    public static net.minecraft.world.damagesource.DamageSource mobAttack(net.minecraft.world.entity.LivingEntity $$0) {
+        return new net.minecraft.world.damagesource.EntityDamageSource("mob", (net.minecraft.world.entity.Entity)$$0);
     }
 
-    public static net.minecraft.world.damagesource.DamageSource indirectMobAttack(net.minecraft.world.entity.Entity source, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity indirectEntity) {
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("mob", source, (net.minecraft.world.entity.Entity)indirectEntity);
+    public static net.minecraft.world.damagesource.DamageSource indirectMobAttack(net.minecraft.world.entity.Entity $$0, @javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$1) {
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("mob", $$0, (net.minecraft.world.entity.Entity)$$1);
     }
 
-    public static net.minecraft.world.damagesource.DamageSource playerAttack(net.minecraft.world.entity.player.Player player) {
-        return new net.minecraft.world.damagesource.EntityDamageSource("player", (net.minecraft.world.entity.Entity)player);
+    public static net.minecraft.world.damagesource.DamageSource playerAttack(net.minecraft.world.entity.player.Player $$0) {
+        return new net.minecraft.world.damagesource.EntityDamageSource("player", (net.minecraft.world.entity.Entity)$$0);
     }
 
-    public static net.minecraft.world.damagesource.DamageSource arrow(net.minecraft.world.entity.projectile.AbstractArrow arrow, @javax.annotation.Nullable net.minecraft.world.entity.Entity indirectEntity) {
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("arrow", (net.minecraft.world.entity.Entity)arrow, indirectEntity).setProjectile();
+    public static net.minecraft.world.damagesource.DamageSource arrow(net.minecraft.world.entity.projectile.AbstractArrow $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1) {
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("arrow", (net.minecraft.world.entity.Entity)$$0, $$1).setProjectile();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource trident(net.minecraft.world.entity.Entity source, @javax.annotation.Nullable net.minecraft.world.entity.Entity indirectEntity) {
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("trident", source, indirectEntity).setProjectile();
+    public static net.minecraft.world.damagesource.DamageSource trident(net.minecraft.world.entity.Entity $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1) {
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("trident", $$0, $$1).setProjectile();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource fireworks(net.minecraft.world.entity.projectile.FireworkRocketEntity firework, @javax.annotation.Nullable net.minecraft.world.entity.Entity indirectEntity) {
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("fireworks", (net.minecraft.world.entity.Entity)firework, indirectEntity).setExplosion();
+    public static net.minecraft.world.damagesource.DamageSource fireworks(net.minecraft.world.entity.projectile.FireworkRocketEntity $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1) {
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("fireworks", (net.minecraft.world.entity.Entity)$$0, $$1).setExplosion();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource fireball(net.minecraft.world.entity.projectile.Fireball fireball, @javax.annotation.Nullable net.minecraft.world.entity.Entity indirectEntity) {
-        if (indirectEntity == null) {
-            return new net.minecraft.world.damagesource.IndirectEntityDamageSource("onFire", (net.minecraft.world.entity.Entity)fireball, (net.minecraft.world.entity.Entity)fireball).setIsFire().setProjectile();
+    public static net.minecraft.world.damagesource.DamageSource fireball(net.minecraft.world.entity.projectile.Fireball $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1) {
+        if ($$1 == null) {
+            return new net.minecraft.world.damagesource.IndirectEntityDamageSource("onFire", (net.minecraft.world.entity.Entity)$$0, (net.minecraft.world.entity.Entity)$$0).setIsFire().setProjectile();
         }
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("fireball", (net.minecraft.world.entity.Entity)fireball, indirectEntity).setIsFire().setProjectile();
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("fireball", (net.minecraft.world.entity.Entity)$$0, $$1).setIsFire().setProjectile();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource witherSkull(net.minecraft.world.entity.projectile.WitherSkull witherSkull, net.minecraft.world.entity.Entity indirectEntity) {
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("witherSkull", (net.minecraft.world.entity.Entity)witherSkull, indirectEntity).setProjectile();
+    public static net.minecraft.world.damagesource.DamageSource witherSkull(net.minecraft.world.entity.projectile.WitherSkull $$0, net.minecraft.world.entity.Entity $$1) {
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("witherSkull", (net.minecraft.world.entity.Entity)$$0, $$1).setProjectile();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource thrown(net.minecraft.world.entity.Entity source, @javax.annotation.Nullable net.minecraft.world.entity.Entity indirectEntity) {
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("thrown", source, indirectEntity).setProjectile();
+    public static net.minecraft.world.damagesource.DamageSource thrown(net.minecraft.world.entity.Entity $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1) {
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("thrown", $$0, $$1).setProjectile();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource indirectMagic(net.minecraft.world.entity.Entity source, @javax.annotation.Nullable net.minecraft.world.entity.Entity indirectEntity) {
-        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("indirectMagic", source, indirectEntity).bypassArmor().setMagic();
+    public static net.minecraft.world.damagesource.DamageSource indirectMagic(net.minecraft.world.entity.Entity $$0, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$1) {
+        return new net.minecraft.world.damagesource.IndirectEntityDamageSource("indirectMagic", $$0, $$1).bypassArmor().setMagic();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource thorns(net.minecraft.world.entity.Entity source) {
-        return new net.minecraft.world.damagesource.EntityDamageSource("thorns", source).setThorns().setMagic();
+    public static net.minecraft.world.damagesource.DamageSource thorns(net.minecraft.world.entity.Entity $$0) {
+        return new net.minecraft.world.damagesource.EntityDamageSource("thorns", $$0).setThorns().setMagic();
     }
 
-    public static net.minecraft.world.damagesource.DamageSource explosion(@javax.annotation.Nullable net.minecraft.world.level.Explosion explosion) {
-        return net.minecraft.world.damagesource.DamageSource.explosion((net.minecraft.world.entity.LivingEntity)(explosion != null ? explosion.getSourceMob() : null));
+    public static net.minecraft.world.damagesource.DamageSource explosion(@javax.annotation.Nullable net.minecraft.world.level.Explosion $$0) {
+        return net.minecraft.world.damagesource.DamageSource.explosion((net.minecraft.world.entity.LivingEntity)($$0 != null ? $$0.getSourceMob() : null));
     }
 
-    public static net.minecraft.world.damagesource.DamageSource explosion(@javax.annotation.Nullable net.minecraft.world.entity.LivingEntity livingEntity) {
-        if (livingEntity != null) {
-            return new net.minecraft.world.damagesource.EntityDamageSource("explosion.player", (net.minecraft.world.entity.Entity)livingEntity).setScalesWithDifficulty().setExplosion();
+    public static net.minecraft.world.damagesource.DamageSource explosion(@javax.annotation.Nullable net.minecraft.world.entity.LivingEntity $$0) {
+        if ($$0 != null) {
+            return new net.minecraft.world.damagesource.EntityDamageSource("explosion.player", (net.minecraft.world.entity.Entity)$$0).setScalesWithDifficulty().setExplosion();
         }
         return new net.minecraft.world.damagesource.DamageSource("explosion").setScalesWithDifficulty().setExplosion();
     }
@@ -160,9 +160,9 @@
         return this.bypassEnchantments;
     }
 
-    public net.minecraft.world.damagesource.DamageSource(java.lang.String messageId) {
+    public net.minecraft.world.damagesource.DamageSource(java.lang.String $$0) {
         this.exhaustion = 0.1f;
-        this.msgId = messageId;
+        this.msgId = $$0;
     }
 
     @javax.annotation.Nullable
@@ -212,14 +212,14 @@
         return this;
     }
 
-    public net.minecraft.network.chat.Component getLocalizedDeathMessage(net.minecraft.world.entity.LivingEntity livingEntity) {
-        net.minecraft.world.entity.LivingEntity livingEntity2 = livingEntity.getKillCredit();
-        java.lang.String string = "death.attack." + this.msgId;
-        java.lang.String string2 = string + ".player";
-        if (livingEntity2 != null) {
-            return net.minecraft.network.chat.Component.translatable((java.lang.String)string2, (java.lang.Object[])new java.lang.Object[]{livingEntity.getDisplayName(), livingEntity2.getDisplayName()});
+    public net.minecraft.network.chat.Component getLocalizedDeathMessage(net.minecraft.world.entity.LivingEntity $$0) {
+        net.minecraft.world.entity.LivingEntity $$1 = $$0.getKillCredit();
+        java.lang.String $$2 = "death.attack." + this.msgId;
+        java.lang.String $$3 = $$2 + ".player";
+        if ($$1 != null) {
+            return net.minecraft.network.chat.Component.translatable((java.lang.String)$$3, (java.lang.Object[])new java.lang.Object[]{$$0.getDisplayName(), $$1.getDisplayName()});
         }
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)string, (java.lang.Object[])new java.lang.Object[]{livingEntity.getDisplayName()});
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)$$2, (java.lang.Object[])new java.lang.Object[]{$$0.getDisplayName()});
     }
 
     public boolean isFire() {