--- net/minecraft/world/damagesource/EntityDamageSource.java
+++ net/minecraft/world/damagesource/EntityDamageSource.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.entity.Entity entity;
     public boolean isThorns;
 
-    public net.minecraft.world.damagesource.EntityDamageSource(java.lang.String damageTypeId, net.minecraft.world.entity.Entity entity) {
-        super(damageTypeId);
-        this.entity = entity;
+    public net.minecraft.world.damagesource.EntityDamageSource(java.lang.String $$0, net.minecraft.world.entity.Entity $$1) {
+        super($$0);
+        this.entity = $$1;
     }
 
     public net.minecraft.world.damagesource.EntityDamageSource setThorns() {
@@ -27,13 +27,13 @@
         return this.entity;
     }
 
-    public net.minecraft.network.chat.Component getLocalizedDeathMessage(net.minecraft.world.entity.LivingEntity livingEntity) {
-        net.minecraft.world.item.ItemStack itemStack = this.entity instanceof net.minecraft.world.entity.LivingEntity ? ((net.minecraft.world.entity.LivingEntity)this.entity).getMainHandItem() : net.minecraft.world.item.ItemStack.EMPTY;
-        java.lang.String string = "death.attack." + this.msgId;
-        if (!itemStack.isEmpty() && itemStack.hasCustomHoverName()) {
-            return net.minecraft.network.chat.Component.translatable((java.lang.String)(string + ".item"), (java.lang.Object[])new java.lang.Object[]{livingEntity.getDisplayName(), this.entity.getDisplayName(), itemStack.getDisplayName()});
+    public net.minecraft.network.chat.Component getLocalizedDeathMessage(net.minecraft.world.entity.LivingEntity $$0) {
+        net.minecraft.world.item.ItemStack $$1 = this.entity instanceof net.minecraft.world.entity.LivingEntity ? ((net.minecraft.world.entity.LivingEntity)this.entity).getMainHandItem() : net.minecraft.world.item.ItemStack.EMPTY;
+        java.lang.String $$2 = "death.attack." + this.msgId;
+        if (!$$1.isEmpty() && $$1.hasCustomHoverName()) {
+            return net.minecraft.network.chat.Component.translatable((java.lang.String)($$2 + ".item"), (java.lang.Object[])new java.lang.Object[]{$$0.getDisplayName(), this.entity.getDisplayName(), $$1.getDisplayName()});
         }
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)string, (java.lang.Object[])new java.lang.Object[]{livingEntity.getDisplayName(), this.entity.getDisplayName()});
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)$$2, (java.lang.Object[])new java.lang.Object[]{$$0.getDisplayName(), this.entity.getDisplayName()});
     }
 
     public boolean scalesWithDifficulty() {