--- net/minecraft/client/resources/ClientPackSource$2.java
+++ net/minecraft/client/resources/ClientPackSource$2.java
@@ -1,17 +1,0 @@
-/*
- * Decompiled with CFR 0.151.
- */
-package net.minecraft.client.resources;
-
-public class net.minecraft.client.resources.ClientPackSource$2
-extends net.minecraft.server.packs.FilePackResources
- {
-    public net.minecraft.client.resources.ClientPackSource$2(java.io.File $$0) {
-        super($$0);
-    }
-
-    public java.lang.String getName() {
-        return "Programmer Art";
-    }
-}
-