--- net/minecraft/client/resources/DefaultPlayerSkin.java
+++ net/minecraft/client/resources/DefaultPlayerSkin.java
@@ -17,22 +17,22 @@
         return STEVE_SKIN_LOCATION;
     }
 
-    public static net.minecraft.resources.ResourceLocation getDefaultSkin(java.util.UUID playerUUID) {
-        if (net.minecraft.client.resources.DefaultPlayerSkin.isAlexDefault((java.util.UUID)playerUUID)) {
+    public static net.minecraft.resources.ResourceLocation getDefaultSkin(java.util.UUID $$0) {
+        if (net.minecraft.client.resources.DefaultPlayerSkin.isAlexDefault((java.util.UUID)$$0)) {
             return ALEX_SKIN_LOCATION;
         }
         return STEVE_SKIN_LOCATION;
     }
 
-    public static java.lang.String getSkinModelName(java.util.UUID playerUUID) {
-        if (net.minecraft.client.resources.DefaultPlayerSkin.isAlexDefault((java.util.UUID)playerUUID)) {
+    public static java.lang.String getSkinModelName(java.util.UUID $$0) {
+        if (net.minecraft.client.resources.DefaultPlayerSkin.isAlexDefault((java.util.UUID)$$0)) {
             return "slim";
         }
         return "default";
     }
 
-    public static boolean isAlexDefault(java.util.UUID playerUUID) {
-        return (playerUUID.hashCode() & 1) == 1;
+    public static boolean isAlexDefault(java.util.UUID $$0) {
+        return ($$0.hashCode() & 1) == 1;
     }
 
     static {