--- net/minecraft/client/resources/GrassColorReloadListener.java
+++ net/minecraft/client/resources/GrassColorReloadListener.java
@@ -11,17 +11,17 @@
     public net.minecraft.client.resources.GrassColorReloadListener() {
     }
 
-    public int[] prepare(net.minecraft.server.packs.resources.ResourceManager resourceManager, net.minecraft.util.profiling.ProfilerFiller profiler) {
+    public int[] prepare(net.minecraft.server.packs.resources.ResourceManager $$0, net.minecraft.util.profiling.ProfilerFiller $$1) {
         try {
-            return net.minecraft.client.resources.LegacyStuffWrapper.getPixels((net.minecraft.server.packs.resources.ResourceManager)resourceManager, (net.minecraft.resources.ResourceLocation)LOCATION);
+            return net.minecraft.client.resources.LegacyStuffWrapper.getPixels((net.minecraft.server.packs.resources.ResourceManager)$$0, (net.minecraft.resources.ResourceLocation)LOCATION);
         }
-        catch (java.io.IOException iOException) {
-            throw new java.lang.IllegalStateException("Failed to load grass color texture", (java.lang.Throwable)iOException);
+        catch (java.io.IOException $$2) {
+            throw new java.lang.IllegalStateException("Failed to load grass color texture", (java.lang.Throwable)$$2);
         }
     }
 
-    public void apply(int[] object, net.minecraft.server.packs.resources.ResourceManager resourceManager, net.minecraft.util.profiling.ProfilerFiller profiler) {
-        net.minecraft.world.level.GrassColor.init((int[])object);
+    public void apply(int[] $$0, net.minecraft.server.packs.resources.ResourceManager $$1, net.minecraft.util.profiling.ProfilerFiller $$2) {
+        net.minecraft.world.level.GrassColor.init((int[])$$0);
     }
 
     public /* synthetic */ java.lang.Object prepare(net.minecraft.server.packs.resources.ResourceManager resourceManager, net.minecraft.util.profiling.ProfilerFiller profilerFiller) {