--- net/minecraft/client/resources/LegacyStuffWrapper.java
+++ net/minecraft/client/resources/LegacyStuffWrapper.java
@@ -9,20 +9,20 @@
     }
 
     @java.lang.Deprecated
-    public static int[] getPixels(net.minecraft.server.packs.resources.ResourceManager manager, net.minecraft.resources.ResourceLocation location) throws java.io.IOException {
-        try (java.io.InputStream inputStream = manager.open(location);
+    public static int[] getPixels(net.minecraft.server.packs.resources.ResourceManager $$0, net.minecraft.resources.ResourceLocation $$1) throws java.io.IOException {
+        try (java.io.InputStream $$2 = $$0.open($$1);
 ){
             int[] nArray;
 block12:            {
-                com.mojang.blaze3d.platform.NativeImage nativeImage = com.mojang.blaze3d.platform.NativeImage.read((java.io.InputStream)inputStream);
+                com.mojang.blaze3d.platform.NativeImage $$3 = com.mojang.blaze3d.platform.NativeImage.read((java.io.InputStream)$$2);
                 try {
-                    nArray = nativeImage.makePixelArray();
-                    if (nativeImage == null) break block12;
+                    nArray = $$3.makePixelArray();
+                    if ($$3 == null) break block12;
                 }
                 catch (java.lang.Throwable throwable) {
-                    if (nativeImage != null) {
+                    if ($$3 != null) {
                         try {
-                            nativeImage.close();
+                            $$3.close();
                         }
                         catch (java.lang.Throwable throwable2) {
                             throwable.addSuppressed(throwable2);
@@ -30,7 +30,7 @@
                     }
                     throw throwable;
                 }
-                nativeImage.close();
+                $$3.close();
             }
             return nArray;
         }