--- net/minecraft/client/resources/MobEffectTextureManager.java
+++ net/minecraft/client/resources/MobEffectTextureManager.java
@@ -6,16 +6,12 @@
 public class net.minecraft.client.resources.MobEffectTextureManager
 extends net.minecraft.client.resources.TextureAtlasHolder
  {
-    public net.minecraft.client.resources.MobEffectTextureManager(net.minecraft.client.renderer.texture.TextureManager textureManager) {
-        super(textureManager, new net.minecraft.resources.ResourceLocation("textures/atlas/mob_effects.png"), "mob_effect");
+    public net.minecraft.client.resources.MobEffectTextureManager(net.minecraft.client.renderer.texture.TextureManager $$0) {
+        super($$0, new net.minecraft.resources.ResourceLocation("textures/atlas/mob_effects.png"), "mob_effect");
     }
 
-    public java.util.stream.Stream<net.minecraft.resources.ResourceLocation> getResourcesToLoad() {
-        return net.minecraft.core.Registry.MOB_EFFECT.keySet().stream();
+    public net.minecraft.client.renderer.texture.TextureAtlasSprite get(net.minecraft.world.effect.MobEffect $$0) {
+        return this.getSprite(net.minecraft.core.Registry.MOB_EFFECT.getKey((java.lang.Object)$$0));
     }
-
-    public net.minecraft.client.renderer.texture.TextureAtlasSprite get(net.minecraft.world.effect.MobEffect effect) {
-        return this.getSprite(net.minecraft.core.Registry.MOB_EFFECT.getKey((java.lang.Object)effect));
-    }
 }
 