--- net/minecraft/client/resources/SplashManager.java
+++ net/minecraft/client/resources/SplashManager.java
@@ -11,24 +11,24 @@
     public java.util.List<java.lang.String> splashes;
     public net.minecraft.client.User user;
 
-    public net.minecraft.client.resources.SplashManager(net.minecraft.client.User user) {
+    public net.minecraft.client.resources.SplashManager(net.minecraft.client.User $$0) {
         this.splashes = com.google.common.collect.Lists.newArrayList();
-        this.user = user;
+        this.user = $$0;
     }
 
-    public java.util.List<java.lang.String> prepare(net.minecraft.server.packs.resources.ResourceManager resourceManager, net.minecraft.util.profiling.ProfilerFiller profiler) {
+    public java.util.List<java.lang.String> prepare(net.minecraft.server.packs.resources.ResourceManager $$02, net.minecraft.util.profiling.ProfilerFiller $$1) {
         java.util.List list;
 block8:        {
-            java.io.BufferedReader bufferedReader = net.minecraft.client.Minecraft.getInstance().getResourceManager().openAsReader(SPLASHES_LOCATION);
+            java.io.BufferedReader $$2 = net.minecraft.client.Minecraft.getInstance().getResourceManager().openAsReader(SPLASHES_LOCATION);
             try {
-                list = (java.util.List)bufferedReader.lines().map(java.lang.String::trim).filter($$0 -> $$0.hashCode() != 125780783).collect(java.util.stream.Collectors.toList());
-                if (bufferedReader == null) break block8;
+                list = (java.util.List)$$2.lines().map(java.lang.String::trim).filter($$0 -> $$0.hashCode() != 125780783).collect(java.util.stream.Collectors.toList());
+                if ($$2 == null) break block8;
             }
             catch (java.lang.Throwable throwable) {
                 try {
-                    if (bufferedReader != null) {
+                    if ($$2 != null) {
                         try {
-                            bufferedReader.close();
+                            $$2.close();
                         }
                         catch (java.lang.Throwable throwable2) {
                             throwable.addSuppressed(throwable2);
@@ -36,18 +36,18 @@
                     }
                     throw throwable;
                 }
-                catch (java.io.IOException iOException) {
+                catch (java.io.IOException $$3) {
                     return java.util.Collections.emptyList();
                 }
             }
-            bufferedReader.close();
+            $$2.close();
         }
         return list;
     }
 
-    public void apply(java.util.List<java.lang.String> object, net.minecraft.server.packs.resources.ResourceManager resourceManager, net.minecraft.util.profiling.ProfilerFiller profiler) {
+    public void apply(java.util.List<java.lang.String> $$0, net.minecraft.server.packs.resources.ResourceManager $$1, net.minecraft.util.profiling.ProfilerFiller $$2) {
         this.splashes.clear();
-        this.splashes.addAll(object);
+        this.splashes.addAll($$0);
     }
 
     @javax.annotation.Nullable