--- net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java
+++ net/minecraft/world/entity/ai/navigation/WallClimberNavigation.java
@@ -13,23 +13,23 @@
         super($$0, $$1);
     }
 
-    public net.minecraft.world.level.pathfinder.Path createPath(net.minecraft.core.BlockPos pos, int accuracy) {
-        this.pathToPosition = pos;
-        return super.createPath(pos, accuracy);
+    public net.minecraft.world.level.pathfinder.Path createPath(net.minecraft.core.BlockPos $$0, int $$1) {
+        this.pathToPosition = $$0;
+        return super.createPath($$0, $$1);
     }
 
-    public net.minecraft.world.level.pathfinder.Path createPath(net.minecraft.world.entity.Entity entity, int n) {
-        this.pathToPosition = entity.blockPosition();
-        return super.createPath(entity, n);
+    public net.minecraft.world.level.pathfinder.Path createPath(net.minecraft.world.entity.Entity $$0, int $$1) {
+        this.pathToPosition = $$0.blockPosition();
+        return super.createPath($$0, $$1);
     }
 
-    public boolean moveTo(net.minecraft.world.entity.Entity entity, double speed) {
-        net.minecraft.world.level.pathfinder.Path path = this.createPath(entity, 0);
-        if (path != null) {
-            return this.moveTo(path, speed);
+    public boolean moveTo(net.minecraft.world.entity.Entity $$0, double $$1) {
+        net.minecraft.world.level.pathfinder.Path $$2 = this.createPath($$0, 0);
+        if ($$2 != null) {
+            return this.moveTo($$2, $$1);
         }
-        this.pathToPosition = entity.blockPosition();
-        this.speedModifier = speed;
+        this.pathToPosition = $$0.blockPosition();
+        this.speedModifier = $$1;
         return true;
     }
 