--- net/minecraft/world/entity/monster/warden/Warden.java
+++ net/minecraft/world/entity/monster/warden/Warden.java
@@ -53,7 +53,7 @@
         this.attackAnimationState = new net.minecraft.world.entity.AnimationState();
         this.sonicBoomAnimationState = new net.minecraft.world.entity.AnimationState();
         this.angerManagement = new net.minecraft.world.entity.monster.warden.AngerManagement(arg_0 -> this.canTargetEntity(arg_0), java.util.Collections.emptyList());
-        this.dynamicGameEventListener = new net.minecraft.world.level.gameevent.DynamicGameEventListener((net.minecraft.world.level.gameevent.GameEventListener)new net.minecraft.world.level.gameevent.vibrations.VibrationListener((net.minecraft.world.level.gameevent.PositionSource)new net.minecraft.world.level.gameevent.EntityPositionSource((net.minecraft.world.entity.Entity)this, this.getEyeHeight()), 16, (net.minecraft.world.level.gameevent.vibrations.VibrationListener$VibrationListenerConfig)this, null, 0.0f, 0));
+        this.dynamicGameEventListener = new net.minecraft.world.level.gameevent.DynamicGameEventListener((net.minecraft.world.level.gameevent.GameEventListener)new net.minecraft.world.level.gameevent.vibrations.VibrationListener((net.minecraft.world.level.gameevent.PositionSource)new net.minecraft.world.level.gameevent.EntityPositionSource((net.minecraft.world.entity.Entity)this, this.getEyeHeight()), 16, (net.minecraft.world.level.gameevent.vibrations.VibrationListener$VibrationListenerConfig)this));
         this.xpReward = 5;
         this.getNavigation().setCanFloat(true);
         this.setPathfindingMalus(net.minecraft.world.level.pathfinder.BlockPathTypes.UNPASSABLE_RAIL, 0.0f);
@@ -64,7 +64,7 @@
         this.setPathfindingMalus(net.minecraft.world.level.pathfinder.BlockPathTypes.DANGER_FIRE, 0.0f);
     }
 
-    public net.minecraft.network.protocol.Packet<?> getAddEntityPacket() {
+    public net.minecraft.network.protocol.Packet<net.minecraft.network.protocol.game.ClientGamePacketListener> getAddEntityPacket() {
         return new net.minecraft.network.protocol.game.ClientboundAddEntityPacket((net.minecraft.world.entity.LivingEntity)this, this.hasPose(net.minecraft.world.entity.Pose.EMERGING) ? 1 : 0);
     }
 
@@ -457,7 +457,7 @@
 
     public boolean shouldListen(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.gameevent.GameEventListener $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.gameevent.GameEvent $$3, net.minecraft.world.level.gameevent.GameEvent$Context $$4) {
         net.minecraft.world.entity.LivingEntity $$5;
-        if (this.isNoAi() || this.isDeadOrDying() || this.getBrain().hasMemoryValue(net.minecraft.world.entity.ai.memory.MemoryModuleType.VIBRATION_COOLDOWN) || this.isDiggingOrEmerging() || !$$0.getWorldBorder().isWithinBounds($$2) || this.isRemoved() || this.level != $$0) {
+        if (this.isNoAi() || this.isDeadOrDying() || this.getBrain().hasMemoryValue(net.minecraft.world.entity.ai.memory.MemoryModuleType.VIBRATION_COOLDOWN) || this.isDiggingOrEmerging() || !$$0.getWorldBorder().isWithinBounds($$2)) {
             return false;
         }
         net.minecraft.world.entity.Entity entity = $$4.sourceEntity();