--- net/minecraft/client/renderer/chunk/ChunkRenderDispatcher$CompiledChunk.java
+++ net/minecraft/client/renderer/chunk/ChunkRenderDispatcher$CompiledChunk.java
@@ -22,16 +22,16 @@
         return this.hasBlocks.isEmpty();
     }
 
-    public boolean isEmpty(net.minecraft.client.renderer.RenderType renderType) {
-        return !this.hasBlocks.contains((java.lang.Object)renderType);
+    public boolean isEmpty(net.minecraft.client.renderer.RenderType $$0) {
+        return !this.hasBlocks.contains((java.lang.Object)$$0);
     }
 
     public java.util.List<net.minecraft.world.level.block.entity.BlockEntity> getRenderableBlockEntities() {
         return this.renderableBlockEntities;
     }
 
-    public boolean facesCanSeeEachother(net.minecraft.core.Direction facing, net.minecraft.core.Direction facing2) {
-        return this.visibilitySet.visibilityBetween(facing, facing2);
+    public boolean facesCanSeeEachother(net.minecraft.core.Direction $$0, net.minecraft.core.Direction $$1) {
+        return this.visibilitySet.visibilityBetween($$0, $$1);
     }
 
     static {