--- net/minecraft/client/renderer/chunk/ChunkRenderDispatcher$CompiledChunk$1.java
+++ net/minecraft/client/renderer/chunk/ChunkRenderDispatcher$CompiledChunk$1.java
@@ -9,7 +9,7 @@
     public net.minecraft.client.renderer.chunk.ChunkRenderDispatcher$CompiledChunk$1() {
     }
 
-    public boolean facesCanSeeEachother(net.minecraft.core.Direction facing, net.minecraft.core.Direction facing2) {
+    public boolean facesCanSeeEachother(net.minecraft.core.Direction $$0, net.minecraft.core.Direction $$1) {
         return false;
     }
 }