--- net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
+++ net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java
@@ -122,8 +122,8 @@
         return this.freeBufferCount;
     }
 
-    public void setCamera(net.minecraft.world.phys.Vec3 pos) {
-        this.camera = pos;
+    public void setCamera(net.minecraft.world.phys.Vec3 $$0) {
+        this.camera = $$0;
     }
 
     public net.minecraft.world.phys.Vec3 getCameraPosition() {
@@ -145,12 +145,12 @@
         this.clearBatchQueue();
     }
 
-    public void schedule(net.minecraft.client.renderer.chunk.ChunkRenderDispatcher$RenderChunk$ChunkCompileTask renderTask) {
+    public void schedule(net.minecraft.client.renderer.chunk.ChunkRenderDispatcher$RenderChunk$ChunkCompileTask $$0) {
         this.mailbox.tell(() -> {
             if ($$0.isHighPriority) {
-                this.toBatchHighPriority.offer((java.lang.Object)renderTask);
+                this.toBatchHighPriority.offer((java.lang.Object)$$0);
             } else {
-                this.toBatchLowPriority.offer((java.lang.Object)renderTask);
+                this.toBatchLowPriority.offer((java.lang.Object)$$0);
             }
             this.toBatchCount = this.toBatchHighPriority.size() + this.toBatchLowPriority.size();
             this.runTask();