--- net/minecraft/data/recipes/CraftingRecipeBuilder.java
+++ net/minecraft/data/recipes/CraftingRecipeBuilder.java
@@ -1,0 +1,20 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data.recipes;
+
+public abstract class net.minecraft.data.recipes.CraftingRecipeBuilder
+ {
+    public net.minecraft.data.recipes.CraftingRecipeBuilder() {
+    }
+
+    public static net.minecraft.world.item.crafting.CraftingBookCategory determineBookCategory(net.minecraft.data.recipes.RecipeCategory $$0) {
+        return switch (net.minecraft.data.recipes.CraftingRecipeBuilder$1.$SwitchMap$net$minecraft$data$recipes$RecipeCategory[$$0.ordinal()]) {
+            case 1 -> net.minecraft.world.item.crafting.CraftingBookCategory.BUILDING;
+            case 2, 3 -> net.minecraft.world.item.crafting.CraftingBookCategory.EQUIPMENT;
+            case 4 -> net.minecraft.world.item.crafting.CraftingBookCategory.REDSTONE;
+            default -> net.minecraft.world.item.crafting.CraftingBookCategory.MISC;
+        };
+    }
+}
+