--- net/minecraft/data/recipes/RecipeCategory.java
+++ net/minecraft/data/recipes/RecipeCategory.java
@@ -1,0 +1,55 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.data.recipes;
+
+public class net.minecraft.data.recipes.RecipeCategory
+extends java.lang.Enum<net.minecraft.data.recipes.RecipeCategory>
+ {
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory BUILDING_BLOCKS;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory DECORATIONS;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory REDSTONE;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory TRANSPORTATION;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory TOOLS;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory COMBAT;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory FOOD;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory BREWING;
+    public static /* enum */ net.minecraft.data.recipes.RecipeCategory MISC;
+    public java.lang.String recipeFolderName;
+    public static /* synthetic */ net.minecraft.data.recipes.RecipeCategory[] $VALUES;
+
+    public static net.minecraft.data.recipes.RecipeCategory[] values() {
+        return (net.minecraft.data.recipes.RecipeCategory[])$VALUES.clone();
+    }
+
+    public static net.minecraft.data.recipes.RecipeCategory valueOf(java.lang.String $$0) {
+        return (net.minecraft.data.recipes.RecipeCategory)java.lang.Enum.valueOf(net.minecraft.data.recipes.RecipeCategory.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.data.recipes.RecipeCategory(java.lang.String $$0) {
+        super(string, n);
+        this.recipeFolderName = $$0;
+    }
+
+    public java.lang.String getFolderName() {
+        return this.recipeFolderName;
+    }
+
+    public static /* synthetic */ net.minecraft.data.recipes.RecipeCategory[] $values() {
+        return new net.minecraft.data.recipes.RecipeCategory[]{BUILDING_BLOCKS, DECORATIONS, REDSTONE, TRANSPORTATION, TOOLS, COMBAT, FOOD, BREWING, MISC};
+    }
+
+    static {
+        BUILDING_BLOCKS = new net.minecraft.data.recipes.RecipeCategory("BUILDING_BLOCKS", 0, "building_blocks");
+        DECORATIONS = new net.minecraft.data.recipes.RecipeCategory("DECORATIONS", 1, "decorations");
+        REDSTONE = new net.minecraft.data.recipes.RecipeCategory("REDSTONE", 2, "redstone");
+        TRANSPORTATION = new net.minecraft.data.recipes.RecipeCategory("TRANSPORTATION", 3, "transportation");
+        TOOLS = new net.minecraft.data.recipes.RecipeCategory("TOOLS", 4, "tools");
+        COMBAT = new net.minecraft.data.recipes.RecipeCategory("COMBAT", 5, "combat");
+        FOOD = new net.minecraft.data.recipes.RecipeCategory("FOOD", 6, "food");
+        BREWING = new net.minecraft.data.recipes.RecipeCategory("BREWING", 7, "brewing");
+        MISC = new net.minecraft.data.recipes.RecipeCategory("MISC", 8, "misc");
+        $VALUES = net.minecraft.data.recipes.RecipeCategory.$values();
+    }
+}
+