--- net/minecraft/data/recipes/SimpleCookingRecipeBuilder$Result.java
+++ net/minecraft/data/recipes/SimpleCookingRecipeBuilder$Result.java
@@ -8,6 +8,7 @@
  {
     public net.minecraft.resources.ResourceLocation id;
     public java.lang.String group;
+    public net.minecraft.world.item.crafting.CookingBookCategory category;
     public net.minecraft.world.item.crafting.Ingredient ingredient;
     public net.minecraft.world.item.Item result;
     public float experience;
@@ -16,26 +17,28 @@
     public net.minecraft.resources.ResourceLocation advancementId;
     public net.minecraft.world.item.crafting.RecipeSerializer<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> serializer;
 
-    public net.minecraft.data.recipes.SimpleCookingRecipeBuilder$Result(net.minecraft.resources.ResourceLocation id, java.lang.String group, net.minecraft.world.item.crafting.Ingredient ingredient, net.minecraft.world.item.Item result, float experience, int cookingTime, net.minecraft.advancements.Advancement$Builder advancement, net.minecraft.resources.ResourceLocation advancementId, net.minecraft.world.item.crafting.RecipeSerializer<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> serializer) {
-        this.id = id;
-        this.group = group;
-        this.ingredient = ingredient;
-        this.result = result;
-        this.experience = experience;
-        this.cookingTime = cookingTime;
-        this.advancement = advancement;
-        this.advancementId = advancementId;
-        this.serializer = serializer;
+    public net.minecraft.data.recipes.SimpleCookingRecipeBuilder$Result(net.minecraft.resources.ResourceLocation $$0, java.lang.String $$1, net.minecraft.world.item.crafting.CookingBookCategory $$2, net.minecraft.world.item.crafting.Ingredient $$3, net.minecraft.world.item.Item $$4, float $$5, int $$6, net.minecraft.advancements.Advancement$Builder $$7, net.minecraft.resources.ResourceLocation $$8, net.minecraft.world.item.crafting.RecipeSerializer<? extends net.minecraft.world.item.crafting.AbstractCookingRecipe> $$9) {
+        this.id = $$0;
+        this.group = $$1;
+        this.category = $$2;
+        this.ingredient = $$3;
+        this.result = $$4;
+        this.experience = $$5;
+        this.cookingTime = $$6;
+        this.advancement = $$7;
+        this.advancementId = $$8;
+        this.serializer = $$9;
     }
 
-    public void serializeRecipeData(com.google.gson.JsonObject json) {
+    public void serializeRecipeData(com.google.gson.JsonObject $$0) {
         if (!this.group.isEmpty()) {
-            json.addProperty("group", this.group);
+            $$0.addProperty("group", this.group);
         }
-        json.add("ingredient", this.ingredient.toJson());
-        json.addProperty("result", net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)this.result).toString());
-        json.addProperty("experience", (java.lang.Number)java.lang.Float.valueOf((float)this.experience));
-        json.addProperty("cookingtime", (java.lang.Number)java.lang.Integer.valueOf((int)this.cookingTime));
+        $$0.addProperty("category", this.category.getSerializedName());
+        $$0.add("ingredient", this.ingredient.toJson());
+        $$0.addProperty("result", net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)this.result).toString());
+        $$0.addProperty("experience", (java.lang.Number)java.lang.Float.valueOf((float)this.experience));
+        $$0.addProperty("cookingtime", (java.lang.Number)java.lang.Integer.valueOf((int)this.cookingTime));
     }
 
     public net.minecraft.world.item.crafting.RecipeSerializer<?> getType() {