--- net/minecraft/data/recipes/SingleItemRecipeBuilder$Result.java
+++ net/minecraft/data/recipes/SingleItemRecipeBuilder$Result.java
@@ -15,24 +15,24 @@
     public net.minecraft.resources.ResourceLocation advancementId;
     public net.minecraft.world.item.crafting.RecipeSerializer<?> type;
 
-    public net.minecraft.data.recipes.SingleItemRecipeBuilder$Result(net.minecraft.resources.ResourceLocation id, net.minecraft.world.item.crafting.RecipeSerializer<?> type, java.lang.String group, net.minecraft.world.item.crafting.Ingredient ingredient, net.minecraft.world.item.Item result, int count, net.minecraft.advancements.Advancement$Builder advancement, net.minecraft.resources.ResourceLocation advancementId) {
-        this.id = id;
-        this.type = type;
-        this.group = group;
-        this.ingredient = ingredient;
-        this.result = result;
-        this.count = count;
-        this.advancement = advancement;
-        this.advancementId = advancementId;
+    public net.minecraft.data.recipes.SingleItemRecipeBuilder$Result(net.minecraft.resources.ResourceLocation $$0, net.minecraft.world.item.crafting.RecipeSerializer<?> $$1, java.lang.String $$2, net.minecraft.world.item.crafting.Ingredient $$3, net.minecraft.world.item.Item $$4, int $$5, net.minecraft.advancements.Advancement$Builder $$6, net.minecraft.resources.ResourceLocation $$7) {
+        this.id = $$0;
+        this.type = $$1;
+        this.group = $$2;
+        this.ingredient = $$3;
+        this.result = $$4;
+        this.count = $$5;
+        this.advancement = $$6;
+        this.advancementId = $$7;
     }
 
-    public void serializeRecipeData(com.google.gson.JsonObject json) {
+    public void serializeRecipeData(com.google.gson.JsonObject $$0) {
         if (!this.group.isEmpty()) {
-            json.addProperty("group", this.group);
+            $$0.addProperty("group", this.group);
         }
-        json.add("ingredient", this.ingredient.toJson());
-        json.addProperty("result", net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)this.result).toString());
-        json.addProperty("count", (java.lang.Number)java.lang.Integer.valueOf((int)this.count));
+        $$0.add("ingredient", this.ingredient.toJson());
+        $$0.addProperty("result", net.minecraft.core.Registry.ITEM.getKey((java.lang.Object)this.result).toString());
+        $$0.addProperty("count", (java.lang.Number)java.lang.Integer.valueOf((int)this.count));
     }
 
     public net.minecraft.resources.ResourceLocation getId() {