--- net/minecraft/data/recipes/SpecialRecipeBuilder.java
+++ net/minecraft/data/recipes/SpecialRecipeBuilder.java
@@ -4,19 +4,20 @@
 package net.minecraft.data.recipes;
 
 public class net.minecraft.data.recipes.SpecialRecipeBuilder
+extends net.minecraft.data.recipes.CraftingRecipeBuilder
  {
-    public net.minecraft.world.item.crafting.SimpleRecipeSerializer<?> serializer;
+    public net.minecraft.world.item.crafting.RecipeSerializer<?> serializer;
 
-    public net.minecraft.data.recipes.SpecialRecipeBuilder(net.minecraft.world.item.crafting.SimpleRecipeSerializer<?> serializer) {
-        this.serializer = serializer;
+    public net.minecraft.data.recipes.SpecialRecipeBuilder(net.minecraft.world.item.crafting.RecipeSerializer<?> $$0) {
+        this.serializer = $$0;
     }
 
-    public static net.minecraft.data.recipes.SpecialRecipeBuilder special(net.minecraft.world.item.crafting.SimpleRecipeSerializer<?> serializer) {
-        return new net.minecraft.data.recipes.SpecialRecipeBuilder(serializer);
+    public static net.minecraft.data.recipes.SpecialRecipeBuilder special(net.minecraft.world.item.crafting.RecipeSerializer<? extends net.minecraft.world.item.crafting.CraftingRecipe> $$0) {
+        return new net.minecraft.data.recipes.SpecialRecipeBuilder($$0);
     }
 
-    public void save(java.util.function.Consumer<net.minecraft.data.recipes.FinishedRecipe> finishedRecipeConsumer, java.lang.String id) {
-        finishedRecipeConsumer.accept((java.lang.Object)new /* Unavailable Anonymous Inner Class!! */);
+    public void save(java.util.function.Consumer<net.minecraft.data.recipes.FinishedRecipe> $$0, java.lang.String $$1) {
+        $$0.accept((java.lang.Object)new /* Unavailable Anonymous Inner Class!! */);
     }
 }
 