--- net/minecraft/data/recipes/SpecialRecipeBuilder$1.java
+++ net/minecraft/data/recipes/SpecialRecipeBuilder$1.java
@@ -4,19 +4,17 @@
 package net.minecraft.data.recipes;
 
 public class net.minecraft.data.recipes.SpecialRecipeBuilder$1
-implements net.minecraft.data.recipes.FinishedRecipe
+extends net.minecraft.data.recipes.CraftingRecipeBuilder$CraftingResult
  {
     public /* synthetic */ java.lang.String val$id;
     public /* synthetic */ net.minecraft.data.recipes.SpecialRecipeBuilder this$0;
 
-    public net.minecraft.data.recipes.SpecialRecipeBuilder$1() {
+    public net.minecraft.data.recipes.SpecialRecipeBuilder$1(net.minecraft.data.recipes.SpecialRecipeBuilder $$0, net.minecraft.world.item.crafting.CraftingBookCategory $$1, java.lang.String string) {
         this.this$0 = $$0;
         this.val$id = string;
+        super($$1);
     }
 
-    public void serializeRecipeData(com.google.gson.JsonObject json) {
-    }
-
     public net.minecraft.world.item.crafting.RecipeSerializer<?> getType() {
         return this.this$0.serializer;
     }