--- net/minecraft/world/entity/raid/Raider$HoldGroundAttackGoal.java
+++ net/minecraft/world/entity/raid/Raider$HoldGroundAttackGoal.java
@@ -11,11 +11,11 @@
     public net.minecraft.world.entity.ai.targeting.TargetingConditions shoutTargeting;
     public /* synthetic */ net.minecraft.world.entity.raid.Raider this$0;
 
-    public net.minecraft.world.entity.raid.Raider$HoldGroundAttackGoal(net.minecraft.world.entity.raid.Raider raider, net.minecraft.world.entity.monster.AbstractIllager mob, float hostileRadiusSquare) {
-        this.this$0 = raider;
+    public net.minecraft.world.entity.raid.Raider$HoldGroundAttackGoal(net.minecraft.world.entity.raid.Raider $$0, net.minecraft.world.entity.monster.AbstractIllager $$1, float $$2) {
+        this.this$0 = $$0;
         this.shoutTargeting = net.minecraft.world.entity.ai.targeting.TargetingConditions.forNonCombat().range(8.0).ignoreLineOfSight().ignoreInvisibilityTesting();
-        this.mob = mob;
-        this.hostileRadiusSqr = hostileRadiusSquare * hostileRadiusSquare;
+        this.mob = $$1;
+        this.hostileRadiusSqr = $$2 * $$2;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE, (java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.LOOK));
     }
 