--- net/minecraft/world/entity/raid/Raider$RaiderCelebration.java
+++ net/minecraft/world/entity/raid/Raider$RaiderCelebration.java
@@ -9,9 +9,9 @@
     public net.minecraft.world.entity.raid.Raider mob;
     public /* synthetic */ net.minecraft.world.entity.raid.Raider this$0;
 
-    public net.minecraft.world.entity.raid.Raider$RaiderCelebration(net.minecraft.world.entity.raid.Raider raider, net.minecraft.world.entity.raid.Raider mob) {
-        this.this$0 = raider;
-        this.mob = mob;
+    public net.minecraft.world.entity.raid.Raider$RaiderCelebration(net.minecraft.world.entity.raid.Raider $$0, net.minecraft.world.entity.raid.Raider $$1) {
+        this.this$0 = $$0;
+        this.mob = $$1;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE));
     }
 