--- net/minecraft/world/entity/raid/Raid$RaiderType.java
+++ net/minecraft/world/entity/raid/Raid$RaiderType.java
@@ -24,10 +24,10 @@
         return (net.minecraft.world.entity.raid.Raid$RaiderType)java.lang.Enum.valueOf(net.minecraft.world.entity.raid.Raid$RaiderType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.entity.raid.Raid$RaiderType(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.raid.Raider> entityType, int[] spawnsPerWaveBeforeBonus) {
+    public net.minecraft.world.entity.raid.Raid$RaiderType(net.minecraft.world.entity.EntityType<? extends net.minecraft.world.entity.raid.Raider> $$0, int[] $$1) {
         super(string, n);
-        this.entityType = entityType;
-        this.spawnsPerWaveBeforeBonus = spawnsPerWaveBeforeBonus;
+        this.entityType = $$0;
+        this.spawnsPerWaveBeforeBonus = $$1;
     }
 
     public static /* synthetic */ net.minecraft.world.entity.raid.Raid$RaiderType[] $values() {