--- net/minecraft/client/renderer/debug/BeeDebugRenderer.java
+++ net/minecraft/client/renderer/debug/BeeDebugRenderer.java
@@ -61,7 +61,7 @@
         this.beeInfosPerEntity.values().removeIf($$1 -> $$1.id == $$0);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, double camX, double camY, double camZ) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, double $$2, double $$3, double $$4) {
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
         com.mojang.blaze3d.systems.RenderSystem.disableTexture();