--- net/minecraft/client/renderer/debug/CollisionBoxRenderer.java
+++ net/minecraft/client/renderer/debug/CollisionBoxRenderer.java
@@ -16,17 +16,16 @@
         this.minecraft = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, double camX, double camY, double camZ) {
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer;
-        double d = net.minecraft.Util.getNanos();
-        if (d - this.lastUpdateTime > 1.0E8) {
-            this.lastUpdateTime = d;
-            vertexConsumer = this.minecraft.gameRenderer.getMainCamera().getEntity();
-            this.shapes = com.google.common.collect.ImmutableList.copyOf((java.lang.Iterable)vertexConsumer.level.getCollisions((net.minecraft.world.entity.Entity)vertexConsumer, vertexConsumer.getBoundingBox().inflate(6.0)));
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, double $$2, double $$3, double $$4) {
+        double $$5 = net.minecraft.Util.getNanos();
+        if ($$5 - this.lastUpdateTime > 1.0E8) {
+            this.lastUpdateTime = $$5;
+            net.minecraft.world.entity.Entity $$6 = this.minecraft.gameRenderer.getMainCamera().getEntity();
+            this.shapes = com.google.common.collect.ImmutableList.copyOf((java.lang.Iterable)$$6.level.getCollisions($$6, $$6.getBoundingBox().inflate(6.0)));
         }
-        vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.lines());
-        for (net.minecraft.world.phys.shapes.VoxelShape voxelShape : this.shapes) {
-            net.minecraft.client.renderer.LevelRenderer.renderVoxelShape((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (net.minecraft.world.phys.shapes.VoxelShape)voxelShape, (double)(-camX), (double)(-camY), (double)(-camZ), (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
+        com.mojang.blaze3d.vertex.VertexConsumer $$7 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.lines());
+        for (net.minecraft.world.phys.shapes.VoxelShape $$8 : this.shapes) {
+            net.minecraft.client.renderer.LevelRenderer.renderVoxelShape((com.mojang.blaze3d.vertex.PoseStack)$$0, (com.mojang.blaze3d.vertex.VertexConsumer)$$7, (net.minecraft.world.phys.shapes.VoxelShape)$$8, (double)(-$$2), (double)(-$$3), (double)(-$$4), (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         }
     }
 }