--- net/minecraft/client/renderer/debug/GameEventListenerRenderer$TrackedListener.java
+++ net/minecraft/client/renderer/debug/GameEventListenerRenderer$TrackedListener.java
@@ -30,7 +30,7 @@
         return this.listenerRange;
     }
 
-    public boolean handleGameEvent(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.gameevent.GameEvent$Message $$1) {
+    public boolean handleGameEvent(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.gameevent.GameEvent $$1, net.minecraft.world.level.gameevent.GameEvent$Context $$2, net.minecraft.world.phys.Vec3 $$3) {
         return false;
     }
 }