--- net/minecraft/client/renderer/debug/GameTestDebugRenderer.java
+++ net/minecraft/client/renderer/debug/GameTestDebugRenderer.java
@@ -21,9 +21,9 @@
         this.markers.clear();
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, double camX, double camY, double camZ) {
-        long l = net.minecraft.Util.getMillis();
-        this.markers.entrySet().removeIf($$1 -> l > ((net.minecraft.client.renderer.debug.GameTestDebugRenderer$Marker)$$1.getValue()).removeAtTime);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$12, double $$2, double $$3, double $$4) {
+        long $$5 = net.minecraft.Util.getMillis();
+        this.markers.entrySet().removeIf($$1 -> $$5 > ((net.minecraft.client.renderer.debug.GameTestDebugRenderer$Marker)$$1.getValue()).removeAtTime);
         this.markers.forEach((arg_0, arg_1) -> this.renderMarker(arg_0, arg_1));
     }
 