--- net/minecraft/client/renderer/debug/LightDebugRenderer.java
+++ net/minecraft/client/renderer/debug/LightDebugRenderer.java
@@ -13,23 +13,23 @@
         this.minecraft = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, double camX, double camY, double camZ) {
-        net.minecraft.client.multiplayer.ClientLevel clientLevel = this.minecraft.level;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, double $$2, double $$3, double $$4) {
+        net.minecraft.client.multiplayer.ClientLevel $$5 = this.minecraft.level;
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
         com.mojang.blaze3d.systems.RenderSystem.disableTexture();
-        net.minecraft.core.BlockPos blockPos = new net.minecraft.core.BlockPos(camX, camY, camZ);
-        it.unimi.dsi.fastutil.longs.LongOpenHashSet longOpenHashSet = new it.unimi.dsi.fastutil.longs.LongOpenHashSet();
-        for (net.minecraft.core.BlockPos blockPos2 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)blockPos.offset(-10, -10, -10), (net.minecraft.core.BlockPos)blockPos.offset(10, 10, 10))) {
-            int n = clientLevel.getBrightness(net.minecraft.world.level.LightLayer.SKY, blockPos2);
-            float f = (float)(15 - n) / 15.0f * 0.5f + 0.16f;
-            int n2 = net.minecraft.util.Mth.hsvToRgb((float)f, (float)0.9f, (float)0.9f);
-            long l = net.minecraft.core.SectionPos.blockToSection((long)blockPos2.asLong());
-            if (longOpenHashSet.add(l)) {
-                net.minecraft.client.renderer.debug.DebugRenderer.renderFloatingText((java.lang.String)clientLevel.getChunkSource().getLightEngine().getDebugData(net.minecraft.world.level.LightLayer.SKY, net.minecraft.core.SectionPos.of((long)l)), (double)net.minecraft.core.SectionPos.sectionToBlockCoord((int)net.minecraft.core.SectionPos.x((long)l), (int)8), (double)net.minecraft.core.SectionPos.sectionToBlockCoord((int)net.minecraft.core.SectionPos.y((long)l), (int)8), (double)net.minecraft.core.SectionPos.sectionToBlockCoord((int)net.minecraft.core.SectionPos.z((long)l), (int)8), (int)0xFF0000, (float)0.3f);
+        net.minecraft.core.BlockPos $$6 = new net.minecraft.core.BlockPos($$2, $$3, $$4);
+        it.unimi.dsi.fastutil.longs.LongOpenHashSet $$7 = new it.unimi.dsi.fastutil.longs.LongOpenHashSet();
+        for (net.minecraft.core.BlockPos $$8 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)$$6.offset(-10, -10, -10), (net.minecraft.core.BlockPos)$$6.offset(10, 10, 10))) {
+            int $$9 = $$5.getBrightness(net.minecraft.world.level.LightLayer.SKY, $$8);
+            float $$10 = (float)(15 - $$9) / 15.0f * 0.5f + 0.16f;
+            int $$11 = net.minecraft.util.Mth.hsvToRgb((float)$$10, (float)0.9f, (float)0.9f);
+            long $$12 = net.minecraft.core.SectionPos.blockToSection((long)$$8.asLong());
+            if ($$7.add($$12)) {
+                net.minecraft.client.renderer.debug.DebugRenderer.renderFloatingText((java.lang.String)$$5.getChunkSource().getLightEngine().getDebugData(net.minecraft.world.level.LightLayer.SKY, net.minecraft.core.SectionPos.of((long)$$12)), (double)net.minecraft.core.SectionPos.sectionToBlockCoord((int)net.minecraft.core.SectionPos.x((long)$$12), (int)8), (double)net.minecraft.core.SectionPos.sectionToBlockCoord((int)net.minecraft.core.SectionPos.y((long)$$12), (int)8), (double)net.minecraft.core.SectionPos.sectionToBlockCoord((int)net.minecraft.core.SectionPos.z((long)$$12), (int)8), (int)0xFF0000, (float)0.3f);
             }
-            if (n == 15) continue;
-            net.minecraft.client.renderer.debug.DebugRenderer.renderFloatingText((java.lang.String)java.lang.String.valueOf((int)n), (double)((double)blockPos2.getX() + 0.5), (double)((double)blockPos2.getY() + 0.25), (double)((double)blockPos2.getZ() + 0.5), (int)n2);
+            if ($$9 == 15) continue;
+            net.minecraft.client.renderer.debug.DebugRenderer.renderFloatingText((java.lang.String)java.lang.String.valueOf((int)$$9), (double)((double)$$8.getX() + 0.5), (double)((double)$$8.getY() + 0.25), (double)((double)$$8.getZ() + 0.5), (int)$$11);
         }
         com.mojang.blaze3d.systems.RenderSystem.enableTexture();
     }