--- net/minecraft/client/renderer/debug/PathfindingRenderer.java
+++ net/minecraft/client/renderer/debug/PathfindingRenderer.java
@@ -24,26 +24,26 @@
         this.creationMap = com.google.common.collect.Maps.newHashMap();
     }
 
-    public void addPath(int eid, net.minecraft.world.level.pathfinder.Path path, float distance) {
-        this.pathMap.put((java.lang.Object)eid, (java.lang.Object)path);
-        this.creationMap.put((java.lang.Object)eid, (java.lang.Object)net.minecraft.Util.getMillis());
-        this.pathMaxDist.put((java.lang.Object)eid, (java.lang.Object)java.lang.Float.valueOf((float)distance));
+    public void addPath(int $$0, net.minecraft.world.level.pathfinder.Path $$1, float $$2) {
+        this.pathMap.put((java.lang.Object)$$0, (java.lang.Object)$$1);
+        this.creationMap.put((java.lang.Object)$$0, (java.lang.Object)net.minecraft.Util.getMillis());
+        this.pathMaxDist.put((java.lang.Object)$$0, (java.lang.Object)java.lang.Float.valueOf((float)$$2));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, double camX, double camY, double camZ) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, double $$2, double $$3, double $$4) {
         if (this.pathMap.isEmpty()) {
             return;
         }
-        long l = net.minecraft.Util.getMillis();
-        for (java.lang.Integer n : this.pathMap.keySet()) {
-            net.minecraft.world.level.pathfinder.Path path = (net.minecraft.world.level.pathfinder.Path)this.pathMap.get((java.lang.Object)n);
-            float f = ((java.lang.Float)this.pathMaxDist.get((java.lang.Object)n)).floatValue();
-            net.minecraft.client.renderer.debug.PathfindingRenderer.renderPath((net.minecraft.world.level.pathfinder.Path)path, (float)f, (boolean)true, (boolean)true, (double)camX, (double)camY, (double)camZ);
+        long $$5 = net.minecraft.Util.getMillis();
+        for (java.lang.Integer $$6 : this.pathMap.keySet()) {
+            net.minecraft.world.level.pathfinder.Path $$7 = (net.minecraft.world.level.pathfinder.Path)this.pathMap.get((java.lang.Object)$$6);
+            float $$8 = ((java.lang.Float)this.pathMaxDist.get((java.lang.Object)$$6)).floatValue();
+            net.minecraft.client.renderer.debug.PathfindingRenderer.renderPath((net.minecraft.world.level.pathfinder.Path)$$7, (float)$$8, (boolean)true, (boolean)true, (double)$$2, (double)$$3, (double)$$4);
         }
-        for (java.lang.Integer n : (java.lang.Integer[])this.creationMap.keySet().toArray((java.lang.Object[])new java.lang.Integer[0])) {
-            if (l - (java.lang.Long)this.creationMap.get((java.lang.Object)n) <= 5000L) continue;
-            this.pathMap.remove((java.lang.Object)n);
-            this.creationMap.remove((java.lang.Object)n);
+        for (java.lang.Integer $$9 : (java.lang.Integer[])this.creationMap.keySet().toArray((java.lang.Object[])new java.lang.Integer[0])) {
+            if ($$5 - (java.lang.Long)this.creationMap.get((java.lang.Object)$$9) <= 5000L) continue;
+            this.pathMap.remove((java.lang.Object)$$9);
+            this.creationMap.remove((java.lang.Object)$$9);
         }
     }
 