--- net/minecraft/client/renderer/debug/StructureRenderer.java
+++ net/minecraft/client/renderer/debug/StructureRenderer.java
@@ -19,29 +19,29 @@
         this.minecraft = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, double camX, double camY, double camZ) {
-        net.minecraft.client.Camera camera = this.minecraft.gameRenderer.getMainCamera();
-        net.minecraft.client.multiplayer.ClientLevel clientLevel = this.minecraft.level;
-        net.minecraft.world.level.dimension.DimensionType dimensionType = clientLevel.dimensionType();
-        net.minecraft.core.BlockPos blockPos = new net.minecraft.core.BlockPos(camera.getPosition().x, 0.0, camera.getPosition().z);
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = buffer.getBuffer(net.minecraft.client.renderer.RenderType.lines());
-        if (this.postMainBoxes.containsKey((java.lang.Object)dimensionType)) {
-            for (net.minecraft.world.level.levelgen.structure.BoundingBox boundingBox : ((java.util.Map)this.postMainBoxes.get((java.lang.Object)dimensionType)).values()) {
-                if (!blockPos.closerThan((net.minecraft.core.Vec3i)boundingBox.getCenter(), 500.0)) continue;
-                net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (double)((double)boundingBox.minX() - camX), (double)((double)boundingBox.minY() - camY), (double)((double)boundingBox.minZ() - camZ), (double)((double)(boundingBox.maxX() + 1) - camX), (double)((double)(boundingBox.maxY() + 1) - camY), (double)((double)(boundingBox.maxZ() + 1) - camZ), (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, double $$2, double $$3, double $$4) {
+        net.minecraft.client.Camera $$5 = this.minecraft.gameRenderer.getMainCamera();
+        net.minecraft.client.multiplayer.ClientLevel $$6 = this.minecraft.level;
+        net.minecraft.world.level.dimension.DimensionType $$7 = $$6.dimensionType();
+        net.minecraft.core.BlockPos $$8 = new net.minecraft.core.BlockPos($$5.getPosition().x, 0.0, $$5.getPosition().z);
+        com.mojang.blaze3d.vertex.VertexConsumer $$9 = $$1.getBuffer(net.minecraft.client.renderer.RenderType.lines());
+        if (this.postMainBoxes.containsKey((java.lang.Object)$$7)) {
+            for (net.minecraft.world.level.levelgen.structure.BoundingBox $$10 : ((java.util.Map)this.postMainBoxes.get((java.lang.Object)$$7)).values()) {
+                if (!$$8.closerThan((net.minecraft.core.Vec3i)$$10.getCenter(), 500.0)) continue;
+                net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)$$0, (com.mojang.blaze3d.vertex.VertexConsumer)$$9, (double)((double)$$10.minX() - $$2), (double)((double)$$10.minY() - $$3), (double)((double)$$10.minZ() - $$4), (double)((double)($$10.maxX() + 1) - $$2), (double)((double)($$10.maxY() + 1) - $$3), (double)((double)($$10.maxZ() + 1) - $$4), (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
             }
         }
-        if (this.postPiecesBoxes.containsKey((java.lang.Object)dimensionType)) {
-            for (net.minecraft.world.level.levelgen.structure.BoundingBox boundingBox : ((java.util.Map)this.postPiecesBoxes.get((java.lang.Object)dimensionType)).entrySet()) {
-                java.lang.String string = (java.lang.String)boundingBox.getKey();
-                net.minecraft.world.level.levelgen.structure.BoundingBox boundingBox2 = (net.minecraft.world.level.levelgen.structure.BoundingBox)boundingBox.getValue();
-                java.lang.Boolean bl = (java.lang.Boolean)((java.util.Map)this.startPiecesMap.get((java.lang.Object)dimensionType)).get((java.lang.Object)string);
-                if (!blockPos.closerThan((net.minecraft.core.Vec3i)boundingBox2.getCenter(), 500.0)) continue;
-                if (bl.booleanValue()) {
-                    net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (double)((double)boundingBox2.minX() - camX), (double)((double)boundingBox2.minY() - camY), (double)((double)boundingBox2.minZ() - camZ), (double)((double)(boundingBox2.maxX() + 1) - camX), (double)((double)(boundingBox2.maxY() + 1) - camY), (double)((double)(boundingBox2.maxZ() + 1) - camZ), (float)0.0f, (float)1.0f, (float)0.0f, (float)1.0f, (float)0.0f, (float)1.0f, (float)0.0f);
+        if (this.postPiecesBoxes.containsKey((java.lang.Object)$$7)) {
+            for (java.util.Map$Entry $$11 : ((java.util.Map)this.postPiecesBoxes.get((java.lang.Object)$$7)).entrySet()) {
+                java.lang.String $$12 = (java.lang.String)$$11.getKey();
+                net.minecraft.world.level.levelgen.structure.BoundingBox $$13 = (net.minecraft.world.level.levelgen.structure.BoundingBox)$$11.getValue();
+                java.lang.Boolean $$14 = (java.lang.Boolean)((java.util.Map)this.startPiecesMap.get((java.lang.Object)$$7)).get((java.lang.Object)$$12);
+                if (!$$8.closerThan((net.minecraft.core.Vec3i)$$13.getCenter(), 500.0)) continue;
+                if ($$14.booleanValue()) {
+                    net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)$$0, (com.mojang.blaze3d.vertex.VertexConsumer)$$9, (double)((double)$$13.minX() - $$2), (double)((double)$$13.minY() - $$3), (double)((double)$$13.minZ() - $$4), (double)((double)($$13.maxX() + 1) - $$2), (double)((double)($$13.maxY() + 1) - $$3), (double)((double)($$13.maxZ() + 1) - $$4), (float)0.0f, (float)1.0f, (float)0.0f, (float)1.0f, (float)0.0f, (float)1.0f, (float)0.0f);
                     continue;
                 }
-                net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (com.mojang.blaze3d.vertex.VertexConsumer)vertexConsumer, (double)((double)boundingBox2.minX() - camX), (double)((double)boundingBox2.minY() - camY), (double)((double)boundingBox2.minZ() - camZ), (double)((double)(boundingBox2.maxX() + 1) - camX), (double)((double)(boundingBox2.maxY() + 1) - camY), (double)((double)(boundingBox2.maxZ() + 1) - camZ), (float)0.0f, (float)0.0f, (float)1.0f, (float)1.0f, (float)0.0f, (float)0.0f, (float)1.0f);
+                net.minecraft.client.renderer.LevelRenderer.renderLineBox((com.mojang.blaze3d.vertex.PoseStack)$$0, (com.mojang.blaze3d.vertex.VertexConsumer)$$9, (double)((double)$$13.minX() - $$2), (double)((double)$$13.minY() - $$3), (double)((double)$$13.minZ() - $$4), (double)((double)($$13.maxX() + 1) - $$2), (double)((double)($$13.maxY() + 1) - $$3), (double)((double)($$13.maxZ() + 1) - $$4), (float)0.0f, (float)0.0f, (float)1.0f, (float)1.0f, (float)0.0f, (float)0.0f, (float)1.0f);
             }
         }
     }