--- net/minecraft/client/renderer/debug/WaterDebugRenderer.java
+++ net/minecraft/client/renderer/debug/WaterDebugRenderer.java
@@ -12,25 +12,24 @@
         this.minecraft = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, net.minecraft.client.renderer.MultiBufferSource buffer, double camX, double camY, double camZ) {
-        net.minecraft.world.level.material.FluidState fluidState;
-        net.minecraft.core.BlockPos blockPos = this.minecraft.player.blockPosition();
-        net.minecraft.world.level.Level level = this.minecraft.player.level;
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, net.minecraft.client.renderer.MultiBufferSource $$1, double $$2, double $$3, double $$4) {
+        net.minecraft.core.BlockPos $$5 = this.minecraft.player.blockPosition();
+        net.minecraft.world.level.Level $$6 = this.minecraft.player.level;
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
         com.mojang.blaze3d.systems.RenderSystem.defaultBlendFunc();
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)0.0f, (float)1.0f, (float)0.0f, (float)0.75f);
         com.mojang.blaze3d.systems.RenderSystem.disableTexture();
         com.mojang.blaze3d.systems.RenderSystem.lineWidth((float)6.0f);
-        for (net.minecraft.core.BlockPos blockPos2 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)blockPos.offset(-10, -10, -10), (net.minecraft.core.BlockPos)blockPos.offset(10, 10, 10))) {
-            fluidState = level.getFluidState(blockPos2);
-            if (!fluidState.is(net.minecraft.tags.FluidTags.WATER)) continue;
-            double d = (float)blockPos2.getY() + fluidState.getHeight((net.minecraft.world.level.BlockGetter)level, blockPos2);
-            net.minecraft.client.renderer.debug.DebugRenderer.renderFilledBox((net.minecraft.world.phys.AABB)new net.minecraft.world.phys.AABB((double)((float)blockPos2.getX() + 0.01f), (double)((float)blockPos2.getY() + 0.01f), (double)((float)blockPos2.getZ() + 0.01f), (double)((float)blockPos2.getX() + 0.99f), d, (double)((float)blockPos2.getZ() + 0.99f)).move(-camX, -camY, -camZ), (float)1.0f, (float)1.0f, (float)1.0f, (float)0.2f);
+        for (net.minecraft.core.BlockPos $$7 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)$$5.offset(-10, -10, -10), (net.minecraft.core.BlockPos)$$5.offset(10, 10, 10))) {
+            net.minecraft.world.level.material.FluidState $$8 = $$6.getFluidState($$7);
+            if (!$$8.is(net.minecraft.tags.FluidTags.WATER)) continue;
+            double $$9 = (float)$$7.getY() + $$8.getHeight((net.minecraft.world.level.BlockGetter)$$6, $$7);
+            net.minecraft.client.renderer.debug.DebugRenderer.renderFilledBox((net.minecraft.world.phys.AABB)new net.minecraft.world.phys.AABB((double)((float)$$7.getX() + 0.01f), (double)((float)$$7.getY() + 0.01f), (double)((float)$$7.getZ() + 0.01f), (double)((float)$$7.getX() + 0.99f), $$9, (double)((float)$$7.getZ() + 0.99f)).move(-$$2, -$$3, -$$4), (float)1.0f, (float)1.0f, (float)1.0f, (float)0.2f);
         }
-        for (net.minecraft.core.BlockPos blockPos2 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)blockPos.offset(-10, -10, -10), (net.minecraft.core.BlockPos)blockPos.offset(10, 10, 10))) {
-            fluidState = level.getFluidState(blockPos2);
-            if (!fluidState.is(net.minecraft.tags.FluidTags.WATER)) continue;
-            net.minecraft.client.renderer.debug.DebugRenderer.renderFloatingText((java.lang.String)java.lang.String.valueOf((int)fluidState.getAmount()), (double)((double)blockPos2.getX() + 0.5), (double)((float)blockPos2.getY() + fluidState.getHeight((net.minecraft.world.level.BlockGetter)level, blockPos2)), (double)((double)blockPos2.getZ() + 0.5), (int)-16777216);
+        for (net.minecraft.core.BlockPos $$10 : net.minecraft.core.BlockPos.betweenClosed((net.minecraft.core.BlockPos)$$5.offset(-10, -10, -10), (net.minecraft.core.BlockPos)$$5.offset(10, 10, 10))) {
+            net.minecraft.world.level.material.FluidState $$11 = $$6.getFluidState($$10);
+            if (!$$11.is(net.minecraft.tags.FluidTags.WATER)) continue;
+            net.minecraft.client.renderer.debug.DebugRenderer.renderFloatingText((java.lang.String)java.lang.String.valueOf((int)$$11.getAmount()), (double)((double)$$10.getX() + 0.5), (double)((float)$$10.getY() + $$11.getHeight((net.minecraft.world.level.BlockGetter)$$6, $$10)), (double)((double)$$10.getZ() + 0.5), (int)-16777216);
         }
         com.mojang.blaze3d.systems.RenderSystem.enableTexture();
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();