--- net/minecraft/client/gui/screens/controls/ControlsScreen.java
+++ net/minecraft/client/gui/screens/controls/ControlsScreen.java
@@ -25,10 +25,10 @@
         this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, $$2 += 24, 200, 20, net.minecraft.network.chat.CommonComponents.GUI_DONE, $$0 -> this.minecraft.setScreen(this.lastScreen)));
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        this.renderBackground(matrixStack);
-        net.minecraft.client.gui.screens.controls.ControlsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)15, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        net.minecraft.client.gui.screens.controls.ControlsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)15, (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
     }
 }
 