--- net/minecraft/world/level/levelgen/synth/ImprovedNoise.java
+++ net/minecraft/world/level/levelgen/synth/ImprovedNoise.java
@@ -70,12 +70,12 @@
         return this.sampleWithDerivative($$7, $$8, $$9, $$10, $$11, $$12, $$3);
     }
 
-    public static double gradDot(int gradIndex, double xFactor, double yFactor, double zFactor) {
-        return net.minecraft.world.level.levelgen.synth.SimplexNoise.dot((int[])net.minecraft.world.level.levelgen.synth.SimplexNoise.GRADIENT[gradIndex & 0xF], (double)xFactor, (double)yFactor, (double)zFactor);
+    public static double gradDot(int $$0, double $$1, double $$2, double $$3) {
+        return net.minecraft.world.level.levelgen.synth.SimplexNoise.dot((int[])net.minecraft.world.level.levelgen.synth.SimplexNoise.GRADIENT[$$0 & 0xF], (double)$$1, (double)$$2, (double)$$3);
     }
 
-    public int p(int permutIndex) {
-        return this.p[permutIndex & 0xFF] & 0xFF;
+    public int p(int $$0) {
+        return this.p[$$0 & 0xFF] & 0xFF;
     }
 
     public double sampleAndLerp(int $$0, int $$1, int $$2, double $$3, double $$4, double $$5, double $$6) {