--- net/minecraft/world/level/levelgen/synth/NormalNoise.java
+++ net/minecraft/world/level/levelgen/synth/NormalNoise.java
@@ -59,11 +59,11 @@
         return 0.1 * (1.0 + 1.0 / (double)($$0 + 1));
     }
 
-    public double getValue(double x, double y, double z) {
-        double d = x * 1.0181268882175227;
-        double d2 = y * 1.0181268882175227;
-        double d3 = z * 1.0181268882175227;
-        return (this.first.getValue(x, y, z) + this.second.getValue(d, d2, d3)) * this.valueFactor;
+    public double getValue(double $$0, double $$1, double $$2) {
+        double $$3 = $$0 * 1.0181268882175227;
+        double $$4 = $$1 * 1.0181268882175227;
+        double $$5 = $$2 * 1.0181268882175227;
+        return (this.first.getValue($$0, $$1, $$2) + this.second.getValue($$3, $$4, $$5)) * this.valueFactor;
     }
 
     public net.minecraft.world.level.levelgen.synth.NormalNoise$NoiseParameters parameters() {