--- net/minecraft/world/level/levelgen/synth/PerlinSimplexNoise.java
+++ net/minecraft/world/level/levelgen/synth/PerlinSimplexNoise.java
@@ -51,18 +51,18 @@
         this.highestFreqValueFactor = 1.0 / (java.lang.Math.pow((double)2.0, (double)$$4) - 1.0);
     }
 
-    public double getValue(double x, double y, boolean useNoiseOffsets) {
-        double d = 0.0;
-        double d2 = this.highestFreqInputFactor;
-        double d3 = this.highestFreqValueFactor;
-        for (net.minecraft.world.level.levelgen.synth.SimplexNoise simplexNoise : this.noiseLevels) {
-            if (simplexNoise != null) {
-                d += simplexNoise.getValue(x * d2 + (useNoiseOffsets ? simplexNoise.xo : 0.0), y * d2 + (useNoiseOffsets ? simplexNoise.yo : 0.0)) * d3;
+    public double getValue(double $$0, double $$1, boolean $$2) {
+        double $$3 = 0.0;
+        double $$4 = this.highestFreqInputFactor;
+        double $$5 = this.highestFreqValueFactor;
+        for (net.minecraft.world.level.levelgen.synth.SimplexNoise $$6 : this.noiseLevels) {
+            if ($$6 != null) {
+                $$3 += $$6.getValue($$0 * $$4 + ($$2 ? $$6.xo : 0.0), $$1 * $$4 + ($$2 ? $$6.yo : 0.0)) * $$5;
             }
-            d2 /= 2.0;
-            d3 *= 2.0;
+            $$4 /= 2.0;
+            $$5 *= 2.0;
         }
-        return d;
+        return $$3;
     }
 }
 