--- net/minecraft/world/level/levelgen/synth/PerlinNoise.java
+++ net/minecraft/world/level/levelgen/synth/PerlinNoise.java
@@ -111,25 +111,25 @@
         $$0.consumeCount(262);
     }
 
-    public double getValue(double x, double y, double z) {
-        return this.getValue(x, y, z, 0.0, 0.0, false);
+    public double getValue(double $$0, double $$1, double $$2) {
+        return this.getValue($$0, $$1, $$2, 0.0, 0.0, false);
     }
 
     @java.lang.Deprecated
-    public double getValue(double x, double y, double z, double d, double d2, boolean bl) {
-        double d3 = 0.0;
-        double d4 = this.lowestFreqInputFactor;
-        double d5 = this.lowestFreqValueFactor;
-        for (int i = 0; i < this.noiseLevels.length; ++i) {
-            net.minecraft.world.level.levelgen.synth.ImprovedNoise improvedNoise = this.noiseLevels[i];
-            if (improvedNoise != null) {
-                double d6 = improvedNoise.noise(net.minecraft.world.level.levelgen.synth.PerlinNoise.wrap((double)(x * d4)), bl ? -improvedNoise.yo : net.minecraft.world.level.levelgen.synth.PerlinNoise.wrap((double)(y * d4)), net.minecraft.world.level.levelgen.synth.PerlinNoise.wrap((double)(z * d4)), d * d4, d2 * d4);
-                d3 += this.amplitudes.getDouble(i) * d6 * d5;
+    public double getValue(double $$0, double $$1, double $$2, double $$3, double $$4, boolean $$5) {
+        double $$6 = 0.0;
+        double $$7 = this.lowestFreqInputFactor;
+        double $$8 = this.lowestFreqValueFactor;
+        for (int $$9 = 0; $$9 < this.noiseLevels.length; ++$$9) {
+            net.minecraft.world.level.levelgen.synth.ImprovedNoise $$10 = this.noiseLevels[$$9];
+            if ($$10 != null) {
+                double $$11 = $$10.noise(net.minecraft.world.level.levelgen.synth.PerlinNoise.wrap((double)($$0 * $$7)), $$5 ? -$$10.yo : net.minecraft.world.level.levelgen.synth.PerlinNoise.wrap((double)($$1 * $$7)), net.minecraft.world.level.levelgen.synth.PerlinNoise.wrap((double)($$2 * $$7)), $$3 * $$7, $$4 * $$7);
+                $$6 += this.amplitudes.getDouble($$9) * $$11 * $$8;
             }
-            d4 *= 2.0;
-            d5 /= 2.0;
+            $$7 *= 2.0;
+            $$8 /= 2.0;
         }
-        return d3;
+        return $$6;
     }
 
     public double maxBrokenValue(double $$0) {
@@ -150,8 +150,8 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.levelgen.synth.ImprovedNoise getOctaveNoise(int octaveIndex) {
-        return this.noiseLevels[this.noiseLevels.length - 1 - octaveIndex];
+    public net.minecraft.world.level.levelgen.synth.ImprovedNoise getOctaveNoise(int $$0) {
+        return this.noiseLevels[this.noiseLevels.length - 1 - $$0];
     }
 
     public static double wrap(double $$0) {