--- net/minecraft/gametest/framework/GameTestHelper$2.java
+++ net/minecraft/gametest/framework/GameTestHelper$2.java
@@ -1,0 +1,28 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.gametest.framework;
+
+public class net.minecraft.gametest.framework.GameTestHelper$2
+extends net.minecraft.world.entity.player.Player
+ {
+    public /* synthetic */ net.minecraft.gametest.framework.GameTestHelper this$0;
+
+    public net.minecraft.gametest.framework.GameTestHelper$2(net.minecraft.gametest.framework.GameTestHelper $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, float $$3, com.mojang.authlib.GameProfile $$4) {
+        this.this$0 = $$0;
+        super($$1, $$2, $$3, $$4);
+    }
+
+    public boolean isSpectator() {
+        return false;
+    }
+
+    public boolean isCreative() {
+        return true;
+    }
+
+    public boolean isLocalPlayer() {
+        return true;
+    }
+}
+