--- net/minecraft/gametest/framework/GameTestBatchRunner.java
+++ net/minecraft/gametest/framework/GameTestBatchRunner.java
@@ -14,14 +14,14 @@
     public java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.gametest.framework.GameTestBatch, java.util.Collection<net.minecraft.gametest.framework.GameTestInfo>>> batches;
     public net.minecraft.core.BlockPos$MutableBlockPos nextTestNorthWestCorner;
 
-    public net.minecraft.gametest.framework.GameTestBatchRunner(java.util.Collection<net.minecraft.gametest.framework.GameTestBatch> testBatches, net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.Rotation rotation, net.minecraft.server.level.ServerLevel serverLevel, net.minecraft.gametest.framework.GameTestTicker testTicker, int testsPerRow) {
-        this.nextTestNorthWestCorner = pos.mutable();
-        this.firstTestNorthWestCorner = pos;
-        this.level = serverLevel;
-        this.testTicker = testTicker;
-        this.testsPerRow = testsPerRow;
-        this.batches = (java.util.List)testBatches.stream().map($$22 -> {
-            java.util.Collection $$3 = (java.util.Collection)$$22.getTestFunctions().stream().map($$2 -> new net.minecraft.gametest.framework.GameTestInfo($$2, rotation, serverLevel)).collect(com.google.common.collect.ImmutableList.toImmutableList());
+    public net.minecraft.gametest.framework.GameTestBatchRunner(java.util.Collection<net.minecraft.gametest.framework.GameTestBatch> $$02, net.minecraft.core.BlockPos $$1, net.minecraft.world.level.block.Rotation $$2, net.minecraft.server.level.ServerLevel $$3, net.minecraft.gametest.framework.GameTestTicker $$4, int $$5) {
+        this.nextTestNorthWestCorner = $$1.mutable();
+        this.firstTestNorthWestCorner = $$1;
+        this.level = $$3;
+        this.testTicker = $$4;
+        this.testsPerRow = $$5;
+        this.batches = (java.util.List)$$02.stream().map($$22 -> {
+            java.util.Collection $$3 = (java.util.Collection)$$22.getTestFunctions().stream().map($$2 -> new net.minecraft.gametest.framework.GameTestInfo($$2, $$2, $$3)).collect(com.google.common.collect.ImmutableList.toImmutableList());
             return com.mojang.datafixers.util.Pair.of((java.lang.Object)$$22, (java.lang.Object)$$3);
         }).collect(com.google.common.collect.ImmutableList.toImmutableList());
         this.allTestInfos = (java.util.List)this.batches.stream().flatMap($$0 -> ((java.util.Collection)$$0.getSecond()).stream()).collect(com.google.common.collect.ImmutableList.toImmutableList());
@@ -35,22 +35,22 @@
         this.runBatch(0);
     }
 
-    public void runBatch(int batchId) {
-        if (batchId >= this.batches.size()) {
+    public void runBatch(int $$0) {
+        if ($$0 >= this.batches.size()) {
             return;
         }
-        com.mojang.datafixers.util.Pair pair = (com.mojang.datafixers.util.Pair)this.batches.get(batchId);
-        net.minecraft.gametest.framework.GameTestBatch gameTestBatch = (net.minecraft.gametest.framework.GameTestBatch)pair.getFirst();
-        java.util.Collection collection = (java.util.Collection)pair.getSecond();
-        java.util.Map map = this.createStructuresForBatch(collection);
-        java.lang.String string = gameTestBatch.getName();
-        LOGGER.info("Running test batch '{}' ({} tests)...", (java.lang.Object)string, (java.lang.Object)collection.size());
-        gameTestBatch.runBeforeBatchFunction(this.level);
-        net.minecraft.gametest.framework.MultipleTestTracker multipleTestTracker = new net.minecraft.gametest.framework.MultipleTestTracker();
-        collection.forEach(arg_0 -> ((net.minecraft.gametest.framework.MultipleTestTracker)multipleTestTracker).addTestToTrack(arg_0));
-        multipleTestTracker.addListener((net.minecraft.gametest.framework.GameTestListener)new /* Unavailable Anonymous Inner Class!! */);
-        collection.forEach($$1 -> {
-            net.minecraft.core.BlockPos $$2 = (net.minecraft.core.BlockPos)map.get($$1);
+        com.mojang.datafixers.util.Pair $$12 = (com.mojang.datafixers.util.Pair)this.batches.get($$0);
+        net.minecraft.gametest.framework.GameTestBatch $$2 = (net.minecraft.gametest.framework.GameTestBatch)$$12.getFirst();
+        java.util.Collection $$3 = (java.util.Collection)$$12.getSecond();
+        java.util.Map $$4 = this.createStructuresForBatch($$3);
+        java.lang.String $$5 = $$2.getName();
+        LOGGER.info("Running test batch '{}' ({} tests)...", (java.lang.Object)$$5, (java.lang.Object)$$3.size());
+        $$2.runBeforeBatchFunction(this.level);
+        net.minecraft.gametest.framework.MultipleTestTracker $$6 = new net.minecraft.gametest.framework.MultipleTestTracker();
+        $$3.forEach(arg_0 -> ((net.minecraft.gametest.framework.MultipleTestTracker)$$6).addTestToTrack(arg_0));
+        $$6.addListener((net.minecraft.gametest.framework.GameTestListener)new /* Unavailable Anonymous Inner Class!! */);
+        $$3.forEach($$1 -> {
+            net.minecraft.core.BlockPos $$2 = (net.minecraft.core.BlockPos)$$4.get($$1);
             net.minecraft.gametest.framework.GameTestRunner.runTest((net.minecraft.gametest.framework.GameTestInfo)$$1, (net.minecraft.core.BlockPos)$$2, (net.minecraft.gametest.framework.GameTestTicker)this.testTicker);
         });
     }