--- net/minecraft/gametest/framework/GameTestSequence.java
+++ net/minecraft/gametest/framework/GameTestSequence.java
@@ -69,21 +69,21 @@
         return $$0;
     }
 
-    public void tickAndContinue(long tick) {
+    public void tickAndContinue(long $$0) {
         try {
-            this.tick(tick);
+            this.tick($$0);
         }
         catch (net.minecraft.gametest.framework.GameTestAssertException gameTestAssertException) {
             // empty catch block
         }
     }
 
-    public void tickAndFailIfNotComplete(long ticks) {
+    public void tickAndFailIfNotComplete(long $$0) {
         try {
-            this.tick(ticks);
+            this.tick($$0);
         }
-        catch (net.minecraft.gametest.framework.GameTestAssertException gameTestAssertException) {
-            this.parent.fail((java.lang.Throwable)gameTestAssertException);
+        catch (net.minecraft.gametest.framework.GameTestAssertException $$1) {
+            this.parent.fail((java.lang.Throwable)$$1);
         }
     }
 
@@ -96,17 +96,17 @@
         }
     }
 
-    public void tick(long tick) {
-        java.util.Iterator iterator = this.events.iterator();
-        while (iterator.hasNext()) {
-            net.minecraft.gametest.framework.GameTestEvent gameTestEvent = (net.minecraft.gametest.framework.GameTestEvent)iterator.next();
-            gameTestEvent.assertion.run();
-            iterator.remove();
-            long l = tick - this.lastTick;
-            long l2 = this.lastTick;
-            this.lastTick = tick;
-            if (gameTestEvent.expectedDelay == null || gameTestEvent.expectedDelay == l) continue;
-            this.parent.fail((java.lang.Throwable)new net.minecraft.gametest.framework.GameTestAssertException("Succeeded in invalid tick: expected " + (l2 + gameTestEvent.expectedDelay) + ", but current tick is " + tick));
+    public void tick(long $$0) {
+        java.util.Iterator $$1 = this.events.iterator();
+        while ($$1.hasNext()) {
+            net.minecraft.gametest.framework.GameTestEvent $$2 = (net.minecraft.gametest.framework.GameTestEvent)$$1.next();
+            $$2.assertion.run();
+            $$1.remove();
+            long $$3 = $$0 - this.lastTick;
+            long $$4 = this.lastTick;
+            this.lastTick = $$0;
+            if ($$2.expectedDelay == null || $$2.expectedDelay == $$3) continue;
+            this.parent.fail((java.lang.Throwable)new net.minecraft.gametest.framework.GameTestAssertException("Succeeded in invalid tick: expected " + ($$4 + $$2.expectedDelay) + ", but current tick is " + $$0));
             break;
         }
     }