--- net/minecraft/gametest/framework/GameTestBatch.java
+++ net/minecraft/gametest/framework/GameTestBatch.java
@@ -31,9 +31,9 @@
         return this.testFunctions;
     }
 
-    public void runBeforeBatchFunction(net.minecraft.server.level.ServerLevel serverLevel) {
+    public void runBeforeBatchFunction(net.minecraft.server.level.ServerLevel $$0) {
         if (this.beforeBatchFunction != null) {
-            this.beforeBatchFunction.accept((java.lang.Object)serverLevel);
+            this.beforeBatchFunction.accept((java.lang.Object)$$0);
         }
     }
 