--- net/minecraft/gametest/framework/GameTestHelper$1.java
+++ net/minecraft/gametest/framework/GameTestHelper$1.java
@@ -8,9 +8,9 @@
  {
     public /* synthetic */ net.minecraft.gametest.framework.GameTestHelper this$0;
 
-    public net.minecraft.gametest.framework.GameTestHelper$1(net.minecraft.gametest.framework.GameTestHelper $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, float $$3, com.mojang.authlib.GameProfile $$4, net.minecraft.world.entity.player.ProfilePublicKey $$5) {
+    public net.minecraft.gametest.framework.GameTestHelper$1(net.minecraft.gametest.framework.GameTestHelper $$0, net.minecraft.world.level.Level $$1, net.minecraft.core.BlockPos $$2, float $$3, com.mojang.authlib.GameProfile $$4) {
         this.this$0 = $$0;
-        super($$1, $$2, $$3, $$4, $$5);
+        super($$1, $$2, $$3, $$4);
     }
 
     public boolean isSpectator() {
@@ -18,7 +18,7 @@
     }
 
     public boolean isCreative() {
-        return true;
+        return false;
     }
 }
 