--- net/minecraft/gametest/framework/GameTestServer$1.java
+++ net/minecraft/gametest/framework/GameTestServer$1.java
@@ -8,7 +8,7 @@
  {
     public /* synthetic */ net.minecraft.gametest.framework.GameTestServer this$0;
 
-    public net.minecraft.gametest.framework.GameTestServer$1(net.minecraft.gametest.framework.GameTestServer $$0, net.minecraft.server.MinecraftServer $$1, net.minecraft.core.RegistryAccess$Frozen $$2, net.minecraft.world.level.storage.PlayerDataStorage $$3, int $$4) {
+    public net.minecraft.gametest.framework.GameTestServer$1(net.minecraft.gametest.framework.GameTestServer $$0, net.minecraft.server.MinecraftServer $$1, net.minecraft.core.LayeredRegistryAccess $$2, net.minecraft.world.level.storage.PlayerDataStorage $$3, int $$4) {
         this.this$0 = $$0;
         super($$1, $$2, $$3, $$4);
     }