--- net/minecraft/gametest/framework/GameTestAssertException.java
+++ net/minecraft/gametest/framework/GameTestAssertException.java
@@ -6,8 +6,8 @@
 public class net.minecraft.gametest.framework.GameTestAssertException
 extends java.lang.RuntimeException
  {
-    public net.minecraft.gametest.framework.GameTestAssertException(java.lang.String exceptionMessage) {
-        super(exceptionMessage);
+    public net.minecraft.gametest.framework.GameTestAssertException(java.lang.String $$0) {
+        super($$0);
     }
 }
 