--- net/minecraft/gametest/framework/GameTestRegistry.java
+++ net/minecraft/gametest/framework/GameTestRegistry.java
@@ -43,8 +43,8 @@
         }
     }
 
-    public static java.util.Collection<net.minecraft.gametest.framework.TestFunction> getTestFunctionsForClassName(java.lang.String className) {
-        return (java.util.Collection)TEST_FUNCTIONS.stream().filter(testFunction -> net.minecraft.gametest.framework.GameTestRegistry.isTestFunctionPartOfClass((net.minecraft.gametest.framework.TestFunction)testFunction, (java.lang.String)className)).collect(java.util.stream.Collectors.toList());
+    public static java.util.Collection<net.minecraft.gametest.framework.TestFunction> getTestFunctionsForClassName(java.lang.String $$0) {
+        return (java.util.Collection)TEST_FUNCTIONS.stream().filter($$1 -> net.minecraft.gametest.framework.GameTestRegistry.isTestFunctionPartOfClass((net.minecraft.gametest.framework.TestFunction)$$1, (java.lang.String)$$0)).collect(java.util.stream.Collectors.toList());
     }
 
     public static java.util.Collection<net.minecraft.gametest.framework.TestFunction> getAllTestFunctions() {
@@ -55,13 +55,13 @@
         return TEST_CLASS_NAMES;
     }
 
-    public static boolean isTestClass(java.lang.String className) {
-        return TEST_CLASS_NAMES.contains((java.lang.Object)className);
+    public static boolean isTestClass(java.lang.String $$0) {
+        return TEST_CLASS_NAMES.contains((java.lang.Object)$$0);
     }
 
     @javax.annotation.Nullable
-    public static java.util.function.Consumer<net.minecraft.server.level.ServerLevel> getBeforeBatchFunction(java.lang.String functionName) {
-        return (java.util.function.Consumer)BEFORE_BATCH_FUNCTIONS.get((java.lang.Object)functionName);
+    public static java.util.function.Consumer<net.minecraft.server.level.ServerLevel> getBeforeBatchFunction(java.lang.String $$0) {
+        return (java.util.function.Consumer)BEFORE_BATCH_FUNCTIONS.get((java.lang.Object)$$0);
     }
 
     @javax.annotation.Nullable
@@ -69,16 +69,16 @@
         return (java.util.function.Consumer)AFTER_BATCH_FUNCTIONS.get((java.lang.Object)$$0);
     }
 
-    public static java.util.Optional<net.minecraft.gametest.framework.TestFunction> findTestFunction(java.lang.String testName) {
-        return net.minecraft.gametest.framework.GameTestRegistry.getAllTestFunctions().stream().filter(testFunction -> testFunction.getTestName().equalsIgnoreCase(testName)).findFirst();
+    public static java.util.Optional<net.minecraft.gametest.framework.TestFunction> findTestFunction(java.lang.String $$0) {
+        return net.minecraft.gametest.framework.GameTestRegistry.getAllTestFunctions().stream().filter($$1 -> $$1.getTestName().equalsIgnoreCase($$0)).findFirst();
     }
 
-    public static net.minecraft.gametest.framework.TestFunction getTestFunction(java.lang.String testName) {
-        java.util.Optional optional = net.minecraft.gametest.framework.GameTestRegistry.findTestFunction((java.lang.String)testName);
-        if (!optional.isPresent()) {
-            throw new java.lang.IllegalArgumentException("Can't find the test function for " + testName);
+    public static net.minecraft.gametest.framework.TestFunction getTestFunction(java.lang.String $$0) {
+        java.util.Optional $$1 = net.minecraft.gametest.framework.GameTestRegistry.findTestFunction((java.lang.String)$$0);
+        if (!$$1.isPresent()) {
+            throw new java.lang.IllegalArgumentException("Can't find the test function for " + $$0);
         }
-        return (net.minecraft.gametest.framework.TestFunction)optional.get();
+        return (net.minecraft.gametest.framework.TestFunction)$$1.get();
     }
 
     public static java.util.Collection<net.minecraft.gametest.framework.TestFunction> useTestGeneratorMethod(java.lang.reflect.Method $$0) {
@@ -120,16 +120,16 @@
         };
     }
 
-    public static boolean isTestFunctionPartOfClass(net.minecraft.gametest.framework.TestFunction testFunction, java.lang.String className) {
-        return testFunction.getTestName().toLowerCase().startsWith(className.toLowerCase() + ".");
+    public static boolean isTestFunctionPartOfClass(net.minecraft.gametest.framework.TestFunction $$0, java.lang.String $$1) {
+        return $$0.getTestName().toLowerCase().startsWith($$1.toLowerCase() + ".");
     }
 
     public static java.util.Collection<net.minecraft.gametest.framework.TestFunction> getLastFailedTests() {
         return LAST_FAILED_TESTS;
     }
 
-    public static void rememberFailedTest(net.minecraft.gametest.framework.TestFunction testFunction) {
-        LAST_FAILED_TESTS.add((java.lang.Object)testFunction);
+    public static void rememberFailedTest(net.minecraft.gametest.framework.TestFunction $$0) {
+        LAST_FAILED_TESTS.add((java.lang.Object)$$0);
     }
 
     public static void forgetFailedTests() {