--- net/minecraft/gametest/framework/GameTestInfo.java
+++ net/minecraft/gametest/framework/GameTestInfo.java
@@ -24,19 +24,19 @@
     @javax.annotation.Nullable
     public net.minecraft.world.level.block.entity.StructureBlockEntity structureBlockEntity;
 
-    public net.minecraft.gametest.framework.GameTestInfo(net.minecraft.gametest.framework.TestFunction function, net.minecraft.world.level.block.Rotation rotation, net.minecraft.server.level.ServerLevel serverLevel) {
+    public net.minecraft.gametest.framework.GameTestInfo(net.minecraft.gametest.framework.TestFunction $$0, net.minecraft.world.level.block.Rotation $$1, net.minecraft.server.level.ServerLevel $$2) {
         this.listeners = com.google.common.collect.Lists.newArrayList();
         this.sequences = com.google.common.collect.Lists.newCopyOnWriteArrayList();
         this.runAtTickTimeMap = new it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap();
         this.timer = com.google.common.base.Stopwatch.createUnstarted();
-        this.testFunction = function;
-        this.level = serverLevel;
-        this.timeoutTicks = function.getMaxTicks();
-        this.rotation = function.getRotation().getRotated(rotation);
+        this.testFunction = $$0;
+        this.level = $$2;
+        this.timeoutTicks = $$0.getMaxTicks();
+        this.rotation = $$0.getRotation().getRotated($$1);
     }
 
-    public void setStructureBlockPos(net.minecraft.core.BlockPos pos) {
-        this.structureBlockPos = pos;
+    public void setStructureBlockPos(net.minecraft.core.BlockPos $$0) {
+        this.structureBlockPos = $$0;
     }
 
     public void startExecution() {
@@ -177,8 +177,8 @@
         }
     }
 
-    public void fail(java.lang.Throwable error) {
-        this.error = error;
+    public void fail(java.lang.Throwable $$0) {
+        this.error = $$0;
         this.finish();
     }
 
@@ -191,16 +191,16 @@
         return this.getTestName();
     }
 
-    public void addListener(net.minecraft.gametest.framework.GameTestListener listener) {
-        this.listeners.add((java.lang.Object)listener);
+    public void addListener(net.minecraft.gametest.framework.GameTestListener $$0) {
+        this.listeners.add((java.lang.Object)$$0);
     }
 
-    public void spawnStructure(net.minecraft.core.BlockPos pos, int n) {
-        this.structureBlockEntity = net.minecraft.gametest.framework.StructureUtils.spawnStructure((java.lang.String)this.getStructureName(), (net.minecraft.core.BlockPos)pos, (net.minecraft.world.level.block.Rotation)this.getRotation(), (int)n, (net.minecraft.server.level.ServerLevel)this.level, (boolean)false);
+    public void spawnStructure(net.minecraft.core.BlockPos $$02, int $$1) {
+        this.structureBlockEntity = net.minecraft.gametest.framework.StructureUtils.spawnStructure((java.lang.String)this.getStructureName(), (net.minecraft.core.BlockPos)$$02, (net.minecraft.world.level.block.Rotation)this.getRotation(), (int)$$1, (net.minecraft.server.level.ServerLevel)this.level, (boolean)false);
         this.structureBlockPos = this.structureBlockEntity.getBlockPos();
         this.structureBlockEntity.setStructureName(this.getTestName());
         net.minecraft.gametest.framework.StructureUtils.addCommandBlockAndButtonToStartTest((net.minecraft.core.BlockPos)this.structureBlockPos, (net.minecraft.core.BlockPos)new net.minecraft.core.BlockPos(1, 0, -1), (net.minecraft.world.level.block.Rotation)this.getRotation(), (net.minecraft.server.level.ServerLevel)this.level);
-        this.listeners.forEach(listener -> listener.testStructureLoaded(this));
+        this.listeners.forEach($$0 -> $$0.testStructureLoaded(this));
     }
 
     public void clearStructure() {