--- net/minecraft/gametest/framework/TestClassNameArgument.java
+++ net/minecraft/gametest/framework/TestClassNameArgument.java
@@ -24,12 +24,12 @@
         return new net.minecraft.gametest.framework.TestClassNameArgument();
     }
 
-    public static java.lang.String getTestClassName(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String argument) {
-        return (java.lang.String)context.getArgument(argument, java.lang.String.class);
+    public static java.lang.String getTestClassName(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (java.lang.String)$$0.getArgument($$1, java.lang.String.class);
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.util.stream.Stream)net.minecraft.gametest.framework.GameTestRegistry.getAllTestClassNames().stream(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder);
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.util.stream.Stream)net.minecraft.gametest.framework.GameTestRegistry.getAllTestClassNames().stream(), (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {