--- net/minecraft/gametest/framework/TestCommand$TestSummaryDisplayer.java
+++ net/minecraft/gametest/framework/TestCommand$TestSummaryDisplayer.java
@@ -9,19 +9,19 @@
     public net.minecraft.server.level.ServerLevel level;
     public net.minecraft.gametest.framework.MultipleTestTracker tracker;
 
-    public net.minecraft.gametest.framework.TestCommand$TestSummaryDisplayer(net.minecraft.server.level.ServerLevel serverLevel, net.minecraft.gametest.framework.MultipleTestTracker tracker) {
-        this.level = serverLevel;
-        this.tracker = tracker;
+    public net.minecraft.gametest.framework.TestCommand$TestSummaryDisplayer(net.minecraft.server.level.ServerLevel $$0, net.minecraft.gametest.framework.MultipleTestTracker $$1) {
+        this.level = $$0;
+        this.tracker = $$1;
     }
 
-    public void testStructureLoaded(net.minecraft.gametest.framework.GameTestInfo testInfo) {
+    public void testStructureLoaded(net.minecraft.gametest.framework.GameTestInfo $$0) {
     }
 
     public void testPassed(net.minecraft.gametest.framework.GameTestInfo $$0) {
         net.minecraft.gametest.framework.TestCommand.showTestSummaryIfAllDone((net.minecraft.server.level.ServerLevel)this.level, (net.minecraft.gametest.framework.MultipleTestTracker)this.tracker);
     }
 
-    public void testFailed(net.minecraft.gametest.framework.GameTestInfo testInfo) {
+    public void testFailed(net.minecraft.gametest.framework.GameTestInfo $$0) {
         net.minecraft.gametest.framework.TestCommand.showTestSummaryIfAllDone((net.minecraft.server.level.ServerLevel)this.level, (net.minecraft.gametest.framework.MultipleTestTracker)this.tracker);
     }
 }