--- net/minecraft/gametest/framework/TestFunctionArgument.java
+++ net/minecraft/gametest/framework/TestFunctionArgument.java
@@ -25,13 +25,13 @@
         return new net.minecraft.gametest.framework.TestFunctionArgument();
     }
 
-    public static net.minecraft.gametest.framework.TestFunction getTestFunction(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String argument) {
-        return (net.minecraft.gametest.framework.TestFunction)context.getArgument(argument, net.minecraft.gametest.framework.TestFunction.class);
+    public static net.minecraft.gametest.framework.TestFunction getTestFunction(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (net.minecraft.gametest.framework.TestFunction)$$0.getArgument($$1, net.minecraft.gametest.framework.TestFunction.class);
     }
 
-    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> context, com.mojang.brigadier.suggestion.SuggestionsBuilder builder) {
-        java.util.stream.Stream stream = net.minecraft.gametest.framework.GameTestRegistry.getAllTestFunctions().stream().map(net.minecraft.gametest.framework.TestFunction::getTestName);
-        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.util.stream.Stream)stream, (com.mojang.brigadier.suggestion.SuggestionsBuilder)builder);
+    public <S> java.util.concurrent.CompletableFuture<com.mojang.brigadier.suggestion.Suggestions> listSuggestions(com.mojang.brigadier.context.CommandContext<S> $$0, com.mojang.brigadier.suggestion.SuggestionsBuilder $$1) {
+        java.util.stream.Stream $$2 = net.minecraft.gametest.framework.GameTestRegistry.getAllTestFunctions().stream().map(net.minecraft.gametest.framework.TestFunction::getTestName);
+        return net.minecraft.commands.SharedSuggestionProvider.suggest((java.util.stream.Stream)$$2, (com.mojang.brigadier.suggestion.SuggestionsBuilder)$$1);
     }
 
     public java.util.Collection<java.lang.String> getExamples() {