--- net/minecraft/gametest/framework/TestFunction.java
+++ net/minecraft/gametest/framework/TestFunction.java
@@ -16,8 +16,8 @@
     public long setupTicks;
     public net.minecraft.world.level.block.Rotation rotation;
 
-    public net.minecraft.gametest.framework.TestFunction(java.lang.String batchName, java.lang.String testName, java.lang.String structureName, int maxTicks, long setupTicks, boolean required, java.util.function.Consumer<net.minecraft.gametest.framework.GameTestHelper> function) {
-        this(batchName, testName, structureName, net.minecraft.world.level.block.Rotation.NONE, maxTicks, setupTicks, required, 1, 1, function);
+    public net.minecraft.gametest.framework.TestFunction(java.lang.String $$0, java.lang.String $$1, java.lang.String $$2, int $$3, long $$4, boolean $$5, java.util.function.Consumer<net.minecraft.gametest.framework.GameTestHelper> $$6) {
+        this($$0, $$1, $$2, net.minecraft.world.level.block.Rotation.NONE, $$3, $$4, $$5, 1, 1, $$6);
     }
 
     public net.minecraft.gametest.framework.TestFunction(java.lang.String $$0, java.lang.String $$1, java.lang.String $$2, net.minecraft.world.level.block.Rotation $$3, int $$4, long $$5, boolean $$6, java.util.function.Consumer<net.minecraft.gametest.framework.GameTestHelper> $$7) {
@@ -37,8 +37,8 @@
         this.setupTicks = $$5;
     }
 
-    public void run(net.minecraft.gametest.framework.GameTestHelper gameTestHelper) {
-        this.function.accept((java.lang.Object)gameTestHelper);
+    public void run(net.minecraft.gametest.framework.GameTestHelper $$0) {
+        this.function.accept((java.lang.Object)$$0);
     }
 
     public java.lang.String getTestName() {