--- net/minecraft/world/level/levelgen/feature/foliageplacers/FoliagePlacer$FoliageAttachment.java
+++ net/minecraft/world/level/levelgen/feature/foliageplacers/FoliagePlacer$FoliageAttachment.java
@@ -9,10 +9,10 @@
     public int radiusOffset;
     public boolean doubleTrunk;
 
-    public net.minecraft.world.level.levelgen.feature.foliageplacers.FoliagePlacer$FoliageAttachment(net.minecraft.core.BlockPos pos, int radiusOffset, boolean doubleTrunk) {
-        this.pos = pos;
-        this.radiusOffset = radiusOffset;
-        this.doubleTrunk = doubleTrunk;
+    public net.minecraft.world.level.levelgen.feature.foliageplacers.FoliagePlacer$FoliageAttachment(net.minecraft.core.BlockPos $$0, int $$1, boolean $$2) {
+        this.pos = $$0;
+        this.radiusOffset = $$1;
+        this.doubleTrunk = $$2;
     }
 
     public net.minecraft.core.BlockPos pos() {