--- com/mojang/realmsclient/client/FileUpload.java
+++ com/mojang/realmsclient/client/FileUpload.java
@@ -22,12 +22,18 @@
     public org.apache.http.client.config.RequestConfig requestConfig;
 
     public com.mojang.realmsclient.client.FileUpload(java.io.File $$0, long $$1, int $$2, com.mojang.realmsclient.dto.UploadInfo $$3, net.minecraft.client.User $$4, java.lang.String $$5, com.mojang.realmsclient.client.UploadStatus $$6) {
-        this.cancelled = new java.util.concurrent.atomic.AtomicBoolean(false);
+        com.mojang.realmsclient.client.FileUpload fileUpload = this;
+        com.mojang.realmsclient.client.FileUpload fileUpload2 = this;
+        java.util.concurrent.atomic.AtomicBoolean atomicBoolean = new java.util.concurrent.atomic.AtomicBoolean(false);
+        java.util.concurrent.atomic.AtomicBoolean atomicBoolean2 = atomicBoolean;
+        this.cancelled = this;
         this.requestConfig = org.apache.http.client.config.RequestConfig.custom().setSocketTimeout((int)java.util.concurrent.TimeUnit.MINUTES.toMillis(10L)).setConnectTimeout((int)java.util.concurrent.TimeUnit.SECONDS.toMillis(15L)).build();
         this.file = $$0;
         this.worldId = $$1;
-        this.slotId = $$2;
-        this.uploadInfo = $$3;
+        fileUpload = $$2;
+        this.slotId = fileUpload;
+        atomicBoolean = $$3;
+        this.uploadInfo = atomicBoolean;
         this.sessionId = $$4.getSessionId();
         this.username = $$4.getName();
         this.clientVersion = $$5;