--- net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
+++ net/minecraft/core/dispenser/BoatDispenseItemBehavior.java
@@ -10,8 +10,8 @@
     public net.minecraft.world.entity.vehicle.Boat$Type type;
     public boolean isChestBoat;
 
-    public net.minecraft.core.dispenser.BoatDispenseItemBehavior(net.minecraft.world.entity.vehicle.Boat$Type type) {
-        this(type, false);
+    public net.minecraft.core.dispenser.BoatDispenseItemBehavior(net.minecraft.world.entity.vehicle.Boat$Type $$0) {
+        this($$0, false);
     }
 
     public net.minecraft.core.dispenser.BoatDispenseItemBehavior(net.minecraft.world.entity.vehicle.Boat$Type $$0, boolean $$1) {
@@ -20,31 +20,34 @@
         this.isChestBoat = $$1;
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        double d;
-        net.minecraft.core.Direction direction = (net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        double d2 = source.x() + (double)((float)direction.getStepX() * 1.125f);
-        double d3 = source.y() + (double)((float)direction.getStepY() * 1.125f);
-        double d4 = source.z() + (double)((float)direction.getStepZ() * 1.125f);
-        net.minecraft.core.BlockPos blockPos = source.getPos().relative(direction);
-        if (serverLevel.getFluidState(blockPos).is(net.minecraft.tags.FluidTags.WATER)) {
-            d = 1.0;
-        } else if (serverLevel.getBlockState(blockPos).isAir() && serverLevel.getFluidState(blockPos.below()).is(net.minecraft.tags.FluidTags.WATER)) {
-            d = 0.0;
+    /*
+     * WARNING - void declaration
+     */
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        void $$10;
+        net.minecraft.core.Direction $$2 = (net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
+        net.minecraft.server.level.ServerLevel $$3 = $$0.getLevel();
+        double $$4 = $$0.x() + (double)((float)$$2.getStepX() * 1.125f);
+        double $$5 = $$0.y() + (double)((float)$$2.getStepY() * 1.125f);
+        double $$6 = $$0.z() + (double)((float)$$2.getStepZ() * 1.125f);
+        net.minecraft.core.BlockPos $$7 = $$0.getPos().relative($$2);
+        if ($$3.getFluidState($$7).is(net.minecraft.tags.FluidTags.WATER)) {
+            double $$8 = 1.0;
+        } else if ($$3.getBlockState($$7).isAir() && $$3.getFluidState($$7.below()).is(net.minecraft.tags.FluidTags.WATER)) {
+            double $$9 = 0.0;
         } else {
-            return this.defaultDispenseItemBehavior.dispense(source, stack);
+            return this.defaultDispenseItemBehavior.dispense($$0, $$1);
         }
-        net.minecraft.world.entity.vehicle.ChestBoat chestBoat = this.isChestBoat ? new net.minecraft.world.entity.vehicle.ChestBoat((net.minecraft.world.level.Level)serverLevel, d2, d3 + d, d4) : new net.minecraft.world.entity.vehicle.Boat((net.minecraft.world.level.Level)serverLevel, d2, d3 + d, d4);
-        chestBoat.setType(this.type);
-        chestBoat.setYRot(direction.toYRot());
-        serverLevel.addFreshEntity((net.minecraft.world.entity.Entity)chestBoat);
-        stack.shrink(1);
-        return stack;
+        net.minecraft.world.entity.vehicle.ChestBoat $$11 = this.isChestBoat ? new net.minecraft.world.entity.vehicle.ChestBoat((net.minecraft.world.level.Level)$$3, $$4, $$5 + $$10, $$6) : new net.minecraft.world.entity.vehicle.Boat((net.minecraft.world.level.Level)$$3, $$4, $$5 + $$10, $$6);
+        $$11.setType(this.type);
+        $$11.setYRot($$2.toYRot());
+        $$3.addFreshEntity((net.minecraft.world.entity.Entity)$$11);
+        $$1.shrink(1);
+        return $$1;
     }
 
-    public void playSound(net.minecraft.core.BlockSource source) {
-        source.getLevel().levelEvent(1000, source.getPos(), 0);
+    public void playSound(net.minecraft.core.BlockSource $$0) {
+        $$0.getLevel().levelEvent(1000, $$0.getPos(), 0);
     }
 }
 