--- net/minecraft/core/dispenser/DefaultDispenseItemBehavior.java
+++ net/minecraft/core/dispenser/DefaultDispenseItemBehavior.java
@@ -9,38 +9,38 @@
     public net.minecraft.core.dispenser.DefaultDispenseItemBehavior() {
     }
 
-    public net.minecraft.world.item.ItemStack dispense(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.world.item.ItemStack itemStack = this.execute(source, stack);
-        this.playSound(source);
-        this.playAnimation(source, (net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
-        return itemStack;
+    public net.minecraft.world.item.ItemStack dispense(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.world.item.ItemStack $$2 = this.execute($$0, $$1);
+        this.playSound($$0);
+        this.playAnimation($$0, (net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
+        return $$2;
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.core.Direction direction = (net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
-        net.minecraft.core.Position position = net.minecraft.world.level.block.DispenserBlock.getDispensePosition((net.minecraft.core.BlockSource)source);
-        net.minecraft.world.item.ItemStack itemStack = stack.split(1);
-        net.minecraft.core.dispenser.DefaultDispenseItemBehavior.spawnItem((net.minecraft.world.level.Level)source.getLevel(), (net.minecraft.world.item.ItemStack)itemStack, (int)6, (net.minecraft.core.Direction)direction, (net.minecraft.core.Position)position);
-        return stack;
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.core.Direction $$2 = (net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
+        net.minecraft.core.Position $$3 = net.minecraft.world.level.block.DispenserBlock.getDispensePosition((net.minecraft.core.BlockSource)$$0);
+        net.minecraft.world.item.ItemStack $$4 = $$1.split(1);
+        net.minecraft.core.dispenser.DefaultDispenseItemBehavior.spawnItem((net.minecraft.world.level.Level)$$0.getLevel(), (net.minecraft.world.item.ItemStack)$$4, (int)6, (net.minecraft.core.Direction)$$2, (net.minecraft.core.Position)$$3);
+        return $$1;
     }
 
-    public static void spawnItem(net.minecraft.world.level.Level level, net.minecraft.world.item.ItemStack stack, int speed, net.minecraft.core.Direction facing, net.minecraft.core.Position position) {
-        double d = position.x();
-        double d2 = position.y();
-        double d3 = position.z();
-        d2 = facing.getAxis() == net.minecraft.core.Direction$Axis.Y ? (d2 -= 0.125) : (d2 -= 0.15625);
-        net.minecraft.world.entity.item.ItemEntity itemEntity = new net.minecraft.world.entity.item.ItemEntity(level, d, d2, d3, stack);
-        double d4 = level.random.nextDouble() * 0.1 + 0.2;
-        itemEntity.setDeltaMovement(level.random.triangle((double)facing.getStepX() * d4, 0.0172275 * (double)speed), level.random.triangle(0.2, 0.0172275 * (double)speed), level.random.triangle((double)facing.getStepZ() * d4, 0.0172275 * (double)speed));
-        level.addFreshEntity((net.minecraft.world.entity.Entity)itemEntity);
+    public static void spawnItem(net.minecraft.world.level.Level $$0, net.minecraft.world.item.ItemStack $$1, int $$2, net.minecraft.core.Direction $$3, net.minecraft.core.Position $$4) {
+        double $$5 = $$4.x();
+        double $$6 = $$4.y();
+        double $$7 = $$4.z();
+        $$6 = $$3.getAxis() == net.minecraft.core.Direction$Axis.Y ? ($$6 -= 0.125) : ($$6 -= 0.15625);
+        net.minecraft.world.entity.item.ItemEntity $$8 = new net.minecraft.world.entity.item.ItemEntity($$0, $$5, $$6, $$7, $$1);
+        double $$9 = $$0.random.nextDouble() * 0.1 + 0.2;
+        $$8.setDeltaMovement($$0.random.triangle((double)$$3.getStepX() * $$9, 0.0172275 * (double)$$2), $$0.random.triangle(0.2, 0.0172275 * (double)$$2), $$0.random.triangle((double)$$3.getStepZ() * $$9, 0.0172275 * (double)$$2));
+        $$0.addFreshEntity((net.minecraft.world.entity.Entity)$$8);
     }
 
-    public void playSound(net.minecraft.core.BlockSource source) {
-        source.getLevel().levelEvent(1000, source.getPos(), 0);
+    public void playSound(net.minecraft.core.BlockSource $$0) {
+        $$0.getLevel().levelEvent(1000, $$0.getPos(), 0);
     }
 
-    public void playAnimation(net.minecraft.core.BlockSource source, net.minecraft.core.Direction facing) {
-        source.getLevel().levelEvent(2000, source.getPos(), facing.get3DDataValue());
+    public void playAnimation(net.minecraft.core.BlockSource $$0, net.minecraft.core.Direction $$1) {
+        $$0.getLevel().levelEvent(2000, $$0.getPos(), $$1.get3DDataValue());
     }
 }
 