--- net/minecraft/core/dispenser/DispenseItemBehavior$9.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$9.java
@@ -12,30 +12,34 @@
         this.defaultDispenseItemBehavior = new net.minecraft.core.dispenser.DefaultDispenseItemBehavior();
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.world.item.ItemStack itemStack;
-        net.minecraft.core.BlockPos blockPos;
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = serverLevel.getBlockState(blockPos = source.getPos().relative((net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING)));
-        net.minecraft.world.level.block.Block block = blockState.getBlock();
-        if (block instanceof net.minecraft.world.level.block.BucketPickup) {
-            itemStack = ((net.minecraft.world.level.block.BucketPickup)block).pickupBlock((net.minecraft.world.level.LevelAccessor)serverLevel, blockPos, blockState);
-            if (itemStack.isEmpty()) {
-                return super.execute(source, stack);
+    /*
+     * WARNING - void declaration
+     */
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        void $$8;
+        net.minecraft.world.item.ItemStack $$6;
+        net.minecraft.core.BlockPos $$3;
+        net.minecraft.server.level.ServerLevel $$2 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$4 = $$2.getBlockState($$3 = $$0.getPos().relative((net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING)));
+        net.minecraft.world.level.block.Block $$5 = $$4.getBlock();
+        if ($$5 instanceof net.minecraft.world.level.block.BucketPickup) {
+            $$6 = ((net.minecraft.world.level.block.BucketPickup)$$5).pickupBlock((net.minecraft.world.level.LevelAccessor)$$2, $$3, $$4);
+            if ($$6.isEmpty()) {
+                return super.execute($$0, $$1);
             }
         } else {
-            return super.execute(source, stack);
+            return super.execute($$0, $$1);
         }
-        serverLevel.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.FLUID_PICKUP, blockPos);
-        net.minecraft.world.item.Item item = itemStack.getItem();
-        stack.shrink(1);
-        if (stack.isEmpty()) {
-            return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)item);
+        $$2.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.FLUID_PICKUP, $$3);
+        net.minecraft.world.item.Item $$7 = $$6.getItem();
+        $$1.shrink(1);
+        if ($$1.isEmpty()) {
+            return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$8);
         }
-        if (((net.minecraft.world.level.block.entity.DispenserBlockEntity)source.getEntity()).addItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)item)) < 0) {
-            this.defaultDispenseItemBehavior.dispense(source, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)item));
+        if (((net.minecraft.world.level.block.entity.DispenserBlockEntity)$$0.getEntity()).addItem(new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$8)) < 0) {
+            this.defaultDispenseItemBehavior.dispense($$0, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$8));
         }
-        return stack;
+        return $$1;
     }
 }
 