--- net/minecraft/core/dispenser/DispenseItemBehavior$18.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$18.java
@@ -9,22 +9,22 @@
     public net.minecraft.core.dispenser.DispenseItemBehavior$18() {
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.core.Direction direction = (net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
-        net.minecraft.core.BlockPos blockPos = source.getPos().relative(direction);
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        net.minecraft.world.level.block.state.BlockState blockState = serverLevel.getBlockState(blockPos);
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.core.Direction $$2 = (net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
+        net.minecraft.core.BlockPos $$3 = $$0.getPos().relative($$2);
+        net.minecraft.server.level.ServerLevel $$4 = $$0.getLevel();
+        net.minecraft.world.level.block.state.BlockState $$5 = $$4.getBlockState($$3);
         this.setSuccess(true);
-        if (blockState.is(net.minecraft.world.level.block.Blocks.RESPAWN_ANCHOR)) {
-            if ((java.lang.Integer)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.RespawnAnchorBlock.CHARGE) != 4) {
-                net.minecraft.world.level.block.RespawnAnchorBlock.charge((net.minecraft.world.level.Level)serverLevel, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.world.level.block.state.BlockState)blockState);
-                stack.shrink(1);
+        if ($$5.is(net.minecraft.world.level.block.Blocks.RESPAWN_ANCHOR)) {
+            if ((java.lang.Integer)$$5.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.RespawnAnchorBlock.CHARGE) != 4) {
+                net.minecraft.world.level.block.RespawnAnchorBlock.charge((net.minecraft.world.level.Level)$$4, (net.minecraft.core.BlockPos)$$3, (net.minecraft.world.level.block.state.BlockState)$$5);
+                $$1.shrink(1);
             } else {
                 this.setSuccess(false);
             }
-            return stack;
+            return $$1;
         }
-        return super.execute(source, stack);
+        return super.execute($$0, $$1);
     }
 }
 