--- net/minecraft/core/dispenser/DispenseItemBehavior$1.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$1.java
@@ -9,10 +9,10 @@
     public net.minecraft.core.dispenser.DispenseItemBehavior$1() {
     }
 
-    public net.minecraft.world.entity.projectile.Projectile getProjectile(net.minecraft.world.level.Level level, net.minecraft.core.Position position, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.world.entity.projectile.Arrow arrow = new net.minecraft.world.entity.projectile.Arrow(level, position.x(), position.y(), position.z());
-        arrow.pickup = net.minecraft.world.entity.projectile.AbstractArrow$Pickup.ALLOWED;
-        return arrow;
+    public net.minecraft.world.entity.projectile.Projectile getProjectile(net.minecraft.world.level.Level $$0, net.minecraft.core.Position $$1, net.minecraft.world.item.ItemStack $$2) {
+        net.minecraft.world.entity.projectile.Arrow $$3 = new net.minecraft.world.entity.projectile.Arrow($$0, $$1.x(), $$1.y(), $$1.z());
+        $$3.pickup = net.minecraft.world.entity.projectile.AbstractArrow$Pickup.ALLOWED;
+        return $$3;
     }
 }
 