--- net/minecraft/core/dispenser/DispenseItemBehavior$11.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$11.java
@@ -9,18 +9,18 @@
     public net.minecraft.core.dispenser.DispenseItemBehavior$11() {
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
         this.setSuccess(true);
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        net.minecraft.core.BlockPos blockPos = source.getPos().relative((net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
-        if (net.minecraft.world.item.BoneMealItem.growCrop((net.minecraft.world.item.ItemStack)stack, (net.minecraft.world.level.Level)serverLevel, (net.minecraft.core.BlockPos)blockPos) || net.minecraft.world.item.BoneMealItem.growWaterPlant((net.minecraft.world.item.ItemStack)stack, (net.minecraft.world.level.Level)serverLevel, (net.minecraft.core.BlockPos)blockPos, null)) {
-            if (!serverLevel.isClientSide) {
-                serverLevel.levelEvent(1505, blockPos, 0);
+        net.minecraft.server.level.ServerLevel $$2 = $$0.getLevel();
+        net.minecraft.core.BlockPos $$3 = $$0.getPos().relative((net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
+        if (net.minecraft.world.item.BoneMealItem.growCrop((net.minecraft.world.item.ItemStack)$$1, (net.minecraft.world.level.Level)$$2, (net.minecraft.core.BlockPos)$$3) || net.minecraft.world.item.BoneMealItem.growWaterPlant((net.minecraft.world.item.ItemStack)$$1, (net.minecraft.world.level.Level)$$2, (net.minecraft.core.BlockPos)$$3, null)) {
+            if (!$$2.isClientSide) {
+                $$2.levelEvent(1505, $$3, 0);
             }
         } else {
             this.setSuccess(false);
         }
-        return stack;
+        return $$1;
     }
 }
 