--- net/minecraft/core/dispenser/DispenseItemBehavior$10.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$10.java
@@ -9,28 +9,28 @@
     public net.minecraft.core.dispenser.DispenseItemBehavior$10() {
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.server.level.ServerLevel $$2 = $$0.getLevel();
         this.setSuccess(true);
-        net.minecraft.core.Direction direction = (net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
-        net.minecraft.core.BlockPos blockPos = source.getPos().relative(direction);
-        net.minecraft.world.level.block.state.BlockState blockState = serverLevel.getBlockState(blockPos);
-        if (net.minecraft.world.level.block.BaseFireBlock.canBePlacedAt((net.minecraft.world.level.Level)serverLevel, (net.minecraft.core.BlockPos)blockPos, (net.minecraft.core.Direction)direction)) {
-            serverLevel.setBlockAndUpdate(blockPos, net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)serverLevel, (net.minecraft.core.BlockPos)blockPos));
-            serverLevel.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, blockPos);
-        } else if (net.minecraft.world.level.block.CampfireBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState) || net.minecraft.world.level.block.CandleBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState) || net.minecraft.world.level.block.CandleCakeBlock.canLight((net.minecraft.world.level.block.state.BlockState)blockState)) {
-            serverLevel.setBlockAndUpdate(blockPos, (net.minecraft.world.level.block.state.BlockState)blockState.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.state.properties.BlockStateProperties.LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)));
-            serverLevel.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, blockPos);
-        } else if (blockState.getBlock() instanceof net.minecraft.world.level.block.TntBlock) {
-            net.minecraft.world.level.block.TntBlock.explode((net.minecraft.world.level.Level)serverLevel, (net.minecraft.core.BlockPos)blockPos);
-            serverLevel.removeBlock(blockPos, false);
+        net.minecraft.core.Direction $$3 = (net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING);
+        net.minecraft.core.BlockPos $$4 = $$0.getPos().relative($$3);
+        net.minecraft.world.level.block.state.BlockState $$5 = $$2.getBlockState($$4);
+        if (net.minecraft.world.level.block.BaseFireBlock.canBePlacedAt((net.minecraft.world.level.Level)$$2, (net.minecraft.core.BlockPos)$$4, (net.minecraft.core.Direction)$$3)) {
+            $$2.setBlockAndUpdate($$4, net.minecraft.world.level.block.BaseFireBlock.getState((net.minecraft.world.level.BlockGetter)$$2, (net.minecraft.core.BlockPos)$$4));
+            $$2.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, $$4);
+        } else if (net.minecraft.world.level.block.CampfireBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$5) || net.minecraft.world.level.block.CandleBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$5) || net.minecraft.world.level.block.CandleCakeBlock.canLight((net.minecraft.world.level.block.state.BlockState)$$5)) {
+            $$2.setBlockAndUpdate($$4, (net.minecraft.world.level.block.state.BlockState)$$5.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.state.properties.BlockStateProperties.LIT, (java.lang.Comparable)java.lang.Boolean.valueOf((boolean)true)));
+            $$2.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.BLOCK_CHANGE, $$4);
+        } else if ($$5.getBlock() instanceof net.minecraft.world.level.block.TntBlock) {
+            net.minecraft.world.level.block.TntBlock.explode((net.minecraft.world.level.Level)$$2, (net.minecraft.core.BlockPos)$$4);
+            $$2.removeBlock($$4, false);
         } else {
             this.setSuccess(false);
         }
-        if (this.isSuccess() && stack.hurt(1, serverLevel.random, null)) {
-            stack.setCount(0);
+        if (this.isSuccess() && $$1.hurt(1, $$2.random, null)) {
+            $$1.setCount(0);
         }
-        return stack;
+        return $$1;
     }
 }
 