--- net/minecraft/core/dispenser/DispenseItemBehavior$16.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$16.java
@@ -9,21 +9,21 @@
     public net.minecraft.core.dispenser.DispenseItemBehavior$16() {
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        net.minecraft.core.BlockPos blockPos = source.getPos().relative((net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
-        net.minecraft.world.level.block.CarvedPumpkinBlock carvedPumpkinBlock = (net.minecraft.world.level.block.CarvedPumpkinBlock)net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN;
-        if (serverLevel.isEmptyBlock(blockPos) && carvedPumpkinBlock.canSpawnGolem((net.minecraft.world.level.LevelReader)serverLevel, blockPos)) {
-            if (!serverLevel.isClientSide) {
-                serverLevel.setBlock(blockPos, carvedPumpkinBlock.defaultBlockState(), 3);
-                serverLevel.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, blockPos);
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        net.minecraft.server.level.ServerLevel $$2 = $$0.getLevel();
+        net.minecraft.core.BlockPos $$3 = $$0.getPos().relative((net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
+        net.minecraft.world.level.block.CarvedPumpkinBlock $$4 = (net.minecraft.world.level.block.CarvedPumpkinBlock)net.minecraft.world.level.block.Blocks.CARVED_PUMPKIN;
+        if ($$2.isEmptyBlock($$3) && $$4.canSpawnGolem((net.minecraft.world.level.LevelReader)$$2, $$3)) {
+            if (!$$2.isClientSide) {
+                $$2.setBlock($$3, $$4.defaultBlockState(), 3);
+                $$2.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE, $$3);
             }
-            stack.shrink(1);
+            $$1.shrink(1);
             this.setSuccess(true);
         } else {
-            this.setSuccess(net.minecraft.world.item.ArmorItem.dispenseArmor((net.minecraft.core.BlockSource)source, (net.minecraft.world.item.ItemStack)stack));
+            this.setSuccess(net.minecraft.world.item.ArmorItem.dispenseArmor((net.minecraft.core.BlockSource)$$0, (net.minecraft.world.item.ItemStack)$$1));
         }
-        return stack;
+        return $$1;
     }
 }
 