--- net/minecraft/core/dispenser/DispenseItemBehavior$20.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$20.java
@@ -12,25 +12,25 @@
         this.defaultDispenseItemBehavior = new net.minecraft.core.dispenser.DefaultDispenseItemBehavior();
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
-        if (net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)stack) != net.minecraft.world.item.alchemy.Potions.WATER) {
-            return this.defaultDispenseItemBehavior.dispense(source, stack);
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$0, net.minecraft.world.item.ItemStack $$1) {
+        if (net.minecraft.world.item.alchemy.PotionUtils.getPotion((net.minecraft.world.item.ItemStack)$$1) != net.minecraft.world.item.alchemy.Potions.WATER) {
+            return this.defaultDispenseItemBehavior.dispense($$0, $$1);
         }
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        net.minecraft.core.BlockPos blockPos = source.getPos();
-        net.minecraft.core.BlockPos blockPos2 = source.getPos().relative((net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
-        if (serverLevel.getBlockState(blockPos2).is(net.minecraft.tags.BlockTags.CONVERTABLE_TO_MUD)) {
-            if (!serverLevel.isClientSide) {
-                for (int i = 0; i < 5; ++i) {
-                    serverLevel.sendParticles((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.SPLASH, (double)blockPos.getX() + serverLevel.random.nextDouble(), (double)(blockPos.getY() + 1), (double)blockPos.getZ() + serverLevel.random.nextDouble(), 1, 0.0, 0.0, 0.0, 1.0);
+        net.minecraft.server.level.ServerLevel $$2 = $$0.getLevel();
+        net.minecraft.core.BlockPos $$3 = $$0.getPos();
+        net.minecraft.core.BlockPos $$4 = $$0.getPos().relative((net.minecraft.core.Direction)$$0.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
+        if ($$2.getBlockState($$4).is(net.minecraft.tags.BlockTags.CONVERTABLE_TO_MUD)) {
+            if (!$$2.isClientSide) {
+                for (int $$5 = 0; $$5 < 5; ++$$5) {
+                    $$2.sendParticles((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.SPLASH, (double)$$3.getX() + $$2.random.nextDouble(), (double)($$3.getY() + 1), (double)$$3.getZ() + $$2.random.nextDouble(), 1, 0.0, 0.0, 0.0, 1.0);
                 }
             }
-            serverLevel.playSound(null, blockPos, net.minecraft.sounds.SoundEvents.BOTTLE_EMPTY, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
-            serverLevel.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.FLUID_PLACE, blockPos);
-            serverLevel.setBlockAndUpdate(blockPos2, net.minecraft.world.level.block.Blocks.MUD.defaultBlockState());
+            $$2.playSound(null, $$3, net.minecraft.sounds.SoundEvents.BOTTLE_EMPTY, net.minecraft.sounds.SoundSource.BLOCKS, 1.0f, 1.0f);
+            $$2.gameEvent(null, net.minecraft.world.level.gameevent.GameEvent.FLUID_PLACE, $$3);
+            $$2.setBlockAndUpdate($$4, net.minecraft.world.level.block.Blocks.MUD.defaultBlockState());
             return new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.GLASS_BOTTLE);
         }
-        return this.defaultDispenseItemBehavior.dispense(source, stack);
+        return this.defaultDispenseItemBehavior.dispense($$0, $$1);
     }
 }
 