--- net/minecraft/core/dispenser/DispenseItemBehavior$17.java
+++ net/minecraft/core/dispenser/DispenseItemBehavior$17.java
@@ -24,21 +24,21 @@
         return $$1;
     }
 
-    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource source, net.minecraft.world.item.ItemStack stack) {
+    public net.minecraft.world.item.ItemStack execute(net.minecraft.core.BlockSource $$02, net.minecraft.world.item.ItemStack $$1) {
         this.setSuccess(false);
-        net.minecraft.server.level.ServerLevel serverLevel = source.getLevel();
-        net.minecraft.core.BlockPos blockPos = source.getPos().relative((net.minecraft.core.Direction)source.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
-        net.minecraft.world.level.block.state.BlockState blockState = serverLevel.getBlockState(blockPos);
-        if (blockState.is(net.minecraft.tags.BlockTags.BEEHIVES, $$0 -> $$0.hasProperty((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BeehiveBlock.HONEY_LEVEL) && $$0.getBlock() instanceof net.minecraft.world.level.block.BeehiveBlock) && (java.lang.Integer)blockState.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BeehiveBlock.HONEY_LEVEL) >= 5) {
-            ((net.minecraft.world.level.block.BeehiveBlock)blockState.getBlock()).releaseBeesAndResetHoneyLevel((net.minecraft.world.level.Level)serverLevel, blockState, blockPos, null, net.minecraft.world.level.block.entity.BeehiveBlockEntity$BeeReleaseStatus.BEE_RELEASED);
+        net.minecraft.server.level.ServerLevel $$2 = $$02.getLevel();
+        net.minecraft.core.BlockPos $$3 = $$02.getPos().relative((net.minecraft.core.Direction)$$02.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.DispenserBlock.FACING));
+        net.minecraft.world.level.block.state.BlockState $$4 = $$2.getBlockState($$3);
+        if ($$4.is(net.minecraft.tags.BlockTags.BEEHIVES, $$0 -> $$0.hasProperty((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BeehiveBlock.HONEY_LEVEL) && $$0.getBlock() instanceof net.minecraft.world.level.block.BeehiveBlock) && (java.lang.Integer)$$4.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BeehiveBlock.HONEY_LEVEL) >= 5) {
+            ((net.minecraft.world.level.block.BeehiveBlock)$$4.getBlock()).releaseBeesAndResetHoneyLevel((net.minecraft.world.level.Level)$$2, $$4, $$3, null, net.minecraft.world.level.block.entity.BeehiveBlockEntity$BeeReleaseStatus.BEE_RELEASED);
             this.setSuccess(true);
-            return this.takeLiquid(source, stack, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HONEY_BOTTLE));
+            return this.takeLiquid($$02, $$1, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.HONEY_BOTTLE));
         }
-        if (serverLevel.getFluidState(blockPos).is(net.minecraft.tags.FluidTags.WATER)) {
+        if ($$2.getFluidState($$3).is(net.minecraft.tags.FluidTags.WATER)) {
             this.setSuccess(true);
-            return this.takeLiquid(source, stack, net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POTION), (net.minecraft.world.item.alchemy.Potion)net.minecraft.world.item.alchemy.Potions.WATER));
+            return this.takeLiquid($$02, $$1, net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.POTION), (net.minecraft.world.item.alchemy.Potion)net.minecraft.world.item.alchemy.Potions.WATER));
         }
-        return super.execute(source, stack);
+        return super.execute($$02, $$1);
     }
 }
 