--- net/minecraft/core/dispenser/OptionalDispenseItemBehavior.java
+++ net/minecraft/core/dispenser/OptionalDispenseItemBehavior.java
@@ -16,12 +16,12 @@
         return this.success;
     }
 
-    public void setSuccess(boolean success) {
-        this.success = success;
+    public void setSuccess(boolean $$0) {
+        this.success = $$0;
     }
 
-    public void playSound(net.minecraft.core.BlockSource source) {
-        source.getLevel().levelEvent(this.isSuccess() ? 1000 : 1001, source.getPos(), 0);
+    public void playSound(net.minecraft.core.BlockSource $$0) {
+        $$0.getLevel().levelEvent(this.isSuccess() ? 1000 : 1001, $$0.getPos(), 0);
     }
 }
 