--- net/minecraft/world/level/storage/loot/entries/CompositeEntryBase$1.java
+++ net/minecraft/world/level/storage/loot/entries/CompositeEntryBase$1.java
@@ -12,13 +12,13 @@
         this.val$constructor = compositeEntryConstructor;
     }
 
-    public void serializeCustom(com.google.gson.JsonObject object, T context, com.google.gson.JsonSerializationContext conditions) {
-        object.add("children", conditions.serialize((java.lang.Object)((net.minecraft.world.level.storage.loot.entries.CompositeEntryBase)context).children));
+    public void serializeCustom(com.google.gson.JsonObject $$0, T $$1, com.google.gson.JsonSerializationContext $$2) {
+        $$0.add("children", $$2.serialize((java.lang.Object)((net.minecraft.world.level.storage.loot.entries.CompositeEntryBase)$$1).children));
     }
 
-    public T deserializeCustom(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext context, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions) {
-        net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[] lootPoolEntryContainerArray = (net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[])net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)object, (java.lang.String)"children", (com.google.gson.JsonDeserializationContext)context, net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[].class);
-        return (T)this.val$constructor.create(lootPoolEntryContainerArray, conditions);
+    public T deserializeCustom(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$2) {
+        net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[] $$3 = (net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[])net.minecraft.util.GsonHelper.getAsObject((com.google.gson.JsonObject)$$0, (java.lang.String)"children", (com.google.gson.JsonDeserializationContext)$$1, net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer[].class);
+        return (T)this.val$constructor.create($$3, $$2);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer deserializeCustom(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray) {