--- net/minecraft/world/level/storage/loot/entries/DynamicLoot.java
+++ net/minecraft/world/level/storage/loot/entries/DynamicLoot.java
@@ -8,21 +8,21 @@
  {
     public net.minecraft.resources.ResourceLocation name;
 
-    public net.minecraft.world.level.storage.loot.entries.DynamicLoot(net.minecraft.resources.ResourceLocation dynamicDropsName, int weight, int quality, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, net.minecraft.world.level.storage.loot.functions.LootItemFunction[] functions) {
-        super(weight, quality, conditions, functions);
-        this.name = dynamicDropsName;
+    public net.minecraft.world.level.storage.loot.entries.DynamicLoot(net.minecraft.resources.ResourceLocation $$0, int $$1, int $$2, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$3, net.minecraft.world.level.storage.loot.functions.LootItemFunction[] $$4) {
+        super($$1, $$2, $$3, $$4);
+        this.name = $$0;
     }
 
     public net.minecraft.world.level.storage.loot.entries.LootPoolEntryType getType() {
         return net.minecraft.world.level.storage.loot.entries.LootPoolEntries.DYNAMIC;
     }
 
-    public void createItemStack(java.util.function.Consumer<net.minecraft.world.item.ItemStack> stackConsumer, net.minecraft.world.level.storage.loot.LootContext lootContext) {
-        lootContext.addDynamicDrops(this.name, stackConsumer);
+    public void createItemStack(java.util.function.Consumer<net.minecraft.world.item.ItemStack> $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        $$1.addDynamicDrops(this.name, $$0);
     }
 
-    public static net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer$Builder<?> dynamicEntry(net.minecraft.resources.ResourceLocation dynamicDropsName) {
-        return net.minecraft.world.level.storage.loot.entries.DynamicLoot.simpleBuilder(($$1, $$2, $$3, $$4) -> new net.minecraft.world.level.storage.loot.entries.DynamicLoot(dynamicDropsName, $$1, $$2, $$3, $$4));
+    public static net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer$Builder<?> dynamicEntry(net.minecraft.resources.ResourceLocation $$0) {
+        return net.minecraft.world.level.storage.loot.entries.DynamicLoot.simpleBuilder(($$1, $$2, $$3, $$4) -> new net.minecraft.world.level.storage.loot.entries.DynamicLoot($$0, $$1, $$2, $$3, $$4));
     }
 }
 