--- net/minecraft/world/level/storage/loot/entries/EmptyLootItem.java
+++ net/minecraft/world/level/storage/loot/entries/EmptyLootItem.java
@@ -14,7 +14,7 @@
         return net.minecraft.world.level.storage.loot.entries.LootPoolEntries.EMPTY;
     }
 
-    public void createItemStack(java.util.function.Consumer<net.minecraft.world.item.ItemStack> stackConsumer, net.minecraft.world.level.storage.loot.LootContext lootContext) {
+    public void createItemStack(java.util.function.Consumer<net.minecraft.world.item.ItemStack> $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
     }
 
     public static net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer$Builder<?> emptyItem() {