--- net/minecraft/world/level/storage/loot/entries/LootTableReference$Serializer.java
+++ net/minecraft/world/level/storage/loot/entries/LootTableReference$Serializer.java
@@ -9,14 +9,14 @@
     public net.minecraft.world.level.storage.loot.entries.LootTableReference$Serializer() {
     }
 
-    public void serializeCustom(com.google.gson.JsonObject object, net.minecraft.world.level.storage.loot.entries.LootTableReference context, com.google.gson.JsonSerializationContext conditions) {
-        super.serializeCustom(object, (net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer)context, conditions);
-        object.addProperty("name", context.name.toString());
+    public void serializeCustom(com.google.gson.JsonObject $$0, net.minecraft.world.level.storage.loot.entries.LootTableReference $$1, com.google.gson.JsonSerializationContext $$2) {
+        super.serializeCustom($$0, (net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer)$$1, $$2);
+        $$0.addProperty("name", $$1.name.toString());
     }
 
-    public net.minecraft.world.level.storage.loot.entries.LootTableReference deserialize(com.google.gson.JsonObject object, com.google.gson.JsonDeserializationContext context, int weight, int quality, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, net.minecraft.world.level.storage.loot.functions.LootItemFunction[] functions) {
-        net.minecraft.resources.ResourceLocation resourceLocation = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)object, (java.lang.String)"name"));
-        return new net.minecraft.world.level.storage.loot.entries.LootTableReference(resourceLocation, weight, quality, conditions, functions);
+    public net.minecraft.world.level.storage.loot.entries.LootTableReference deserialize(com.google.gson.JsonObject $$0, com.google.gson.JsonDeserializationContext $$1, int $$2, int $$3, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$4, net.minecraft.world.level.storage.loot.functions.LootItemFunction[] $$5) {
+        net.minecraft.resources.ResourceLocation $$6 = new net.minecraft.resources.ResourceLocation(net.minecraft.util.GsonHelper.getAsString((com.google.gson.JsonObject)$$0, (java.lang.String)"name"));
+        return new net.minecraft.world.level.storage.loot.entries.LootTableReference($$6, $$2, $$3, $$4, $$5);
     }
 
     public /* synthetic */ net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer deserialize(com.google.gson.JsonObject jsonObject, com.google.gson.JsonDeserializationContext jsonDeserializationContext, int n, int n2, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] lootItemConditionArray, net.minecraft.world.level.storage.loot.functions.LootItemFunction[] lootItemFunctionArray) {