--- net/minecraft/world/level/storage/loot/entries/LootPoolSingletonContainer$EntryBase.java
+++ net/minecraft/world/level/storage/loot/entries/LootPoolSingletonContainer$EntryBase.java
@@ -12,8 +12,8 @@
         this.this$0 = $$0;
     }
 
-    public int getWeight(float luck) {
-        return java.lang.Math.max((int)net.minecraft.util.Mth.floor((float)((float)this.this$0.weight + (float)this.this$0.quality * luck)), (int)0);
+    public int getWeight(float $$0) {
+        return java.lang.Math.max((int)net.minecraft.util.Mth.floor((float)((float)this.this$0.weight + (float)this.this$0.quality * $$0)), (int)0);
     }
 }
 