--- net/minecraft/world/level/storage/loot/entries/LootItem.java
+++ net/minecraft/world/level/storage/loot/entries/LootItem.java
@@ -8,21 +8,21 @@
  {
     public net.minecraft.world.item.Item item;
 
-    public net.minecraft.world.level.storage.loot.entries.LootItem(net.minecraft.world.item.Item item, int weight, int quality, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] conditions, net.minecraft.world.level.storage.loot.functions.LootItemFunction[] functions) {
-        super(weight, quality, conditions, functions);
-        this.item = item;
+    public net.minecraft.world.level.storage.loot.entries.LootItem(net.minecraft.world.item.Item $$0, int $$1, int $$2, net.minecraft.world.level.storage.loot.predicates.LootItemCondition[] $$3, net.minecraft.world.level.storage.loot.functions.LootItemFunction[] $$4) {
+        super($$1, $$2, $$3, $$4);
+        this.item = $$0;
     }
 
     public net.minecraft.world.level.storage.loot.entries.LootPoolEntryType getType() {
         return net.minecraft.world.level.storage.loot.entries.LootPoolEntries.ITEM;
     }
 
-    public void createItemStack(java.util.function.Consumer<net.minecraft.world.item.ItemStack> stackConsumer, net.minecraft.world.level.storage.loot.LootContext lootContext) {
-        stackConsumer.accept((java.lang.Object)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)this.item));
+    public void createItemStack(java.util.function.Consumer<net.minecraft.world.item.ItemStack> $$0, net.minecraft.world.level.storage.loot.LootContext $$1) {
+        $$0.accept((java.lang.Object)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)this.item));
     }
 
-    public static net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer$Builder<?> lootTableItem(net.minecraft.world.level.ItemLike item) {
-        return net.minecraft.world.level.storage.loot.entries.LootItem.simpleBuilder(($$1, $$2, $$3, $$4) -> new net.minecraft.world.level.storage.loot.entries.LootItem(item.asItem(), $$1, $$2, $$3, $$4));
+    public static net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer$Builder<?> lootTableItem(net.minecraft.world.level.ItemLike $$0) {
+        return net.minecraft.world.level.storage.loot.entries.LootItem.simpleBuilder(($$1, $$2, $$3, $$4) -> new net.minecraft.world.level.storage.loot.entries.LootItem($$0.asItem(), $$1, $$2, $$3, $$4));
     }
 }
 