--- net/minecraft/world/level/storage/loot/entries/LootPoolEntries.java
+++ net/minecraft/world/level/storage/loot/entries/LootPoolEntries.java
@@ -17,8 +17,8 @@
     public net.minecraft.world.level.storage.loot.entries.LootPoolEntries() {
     }
 
-    public static net.minecraft.world.level.storage.loot.entries.LootPoolEntryType register(java.lang.String name, net.minecraft.world.level.storage.loot.Serializer<? extends net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer> serializer) {
-        return (net.minecraft.world.level.storage.loot.entries.LootPoolEntryType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.LOOT_POOL_ENTRY_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(name), (java.lang.Object)new net.minecraft.world.level.storage.loot.entries.LootPoolEntryType(serializer));
+    public static net.minecraft.world.level.storage.loot.entries.LootPoolEntryType register(java.lang.String $$0, net.minecraft.world.level.storage.loot.Serializer<? extends net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer> $$1) {
+        return (net.minecraft.world.level.storage.loot.entries.LootPoolEntryType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.LOOT_POOL_ENTRY_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0), (java.lang.Object)new net.minecraft.world.level.storage.loot.entries.LootPoolEntryType($$1));
     }
 
     public static java.lang.Object createGsonAdapter() {