--- net/minecraft/world/level/chunk/storage/ChunkStorage.java
+++ net/minecraft/world/level/chunk/storage/ChunkStorage.java
@@ -60,18 +60,18 @@
         $$0.put("__context", (net.minecraft.nbt.Tag)$$3);
     }
 
-    public static int getVersion(net.minecraft.nbt.CompoundTag chunkData) {
-        return chunkData.contains("DataVersion", 99) ? chunkData.getInt("DataVersion") : -1;
+    public static int getVersion(net.minecraft.nbt.CompoundTag $$0) {
+        return $$0.contains("DataVersion", 99) ? $$0.getInt("DataVersion") : -1;
     }
 
     public java.util.concurrent.CompletableFuture<java.util.Optional<net.minecraft.nbt.CompoundTag>> read(net.minecraft.world.level.ChunkPos $$0) {
         return this.worker.loadAsync($$0);
     }
 
-    public void write(net.minecraft.world.level.ChunkPos chunkPos, net.minecraft.nbt.CompoundTag chunkData) {
-        this.worker.store(chunkPos, chunkData);
+    public void write(net.minecraft.world.level.ChunkPos $$0, net.minecraft.nbt.CompoundTag $$1) {
+        this.worker.store($$0, $$1);
         if (this.legacyStructureHandler != null) {
-            this.legacyStructureHandler.removeIndex(chunkPos.toLong());
+            this.legacyStructureHandler.removeIndex($$0.toLong());
         }
     }
 