--- net/minecraft/world/level/chunk/storage/IOWorker$PendingStore.java
+++ net/minecraft/world/level/chunk/storage/IOWorker$PendingStore.java
@@ -9,9 +9,9 @@
     public net.minecraft.nbt.CompoundTag data;
     public java.util.concurrent.CompletableFuture<java.lang.Void> result;
 
-    public net.minecraft.world.level.chunk.storage.IOWorker$PendingStore(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag data) {
+    public net.minecraft.world.level.chunk.storage.IOWorker$PendingStore(@javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$0) {
         this.result = new java.util.concurrent.CompletableFuture();
-        this.data = data;
+        this.data = $$0;
     }
 }
 