--- net/minecraft/world/level/chunk/storage/RegionFile$ChunkBuffer.java
+++ net/minecraft/world/level/chunk/storage/RegionFile$ChunkBuffer.java
@@ -9,7 +9,7 @@
     public net.minecraft.world.level.ChunkPos pos;
     public /* synthetic */ net.minecraft.world.level.chunk.storage.RegionFile this$0;
 
-    public net.minecraft.world.level.chunk.storage.RegionFile$ChunkBuffer(net.minecraft.world.level.chunk.storage.RegionFile regionFile, net.minecraft.world.level.ChunkPos pos) {
+    public net.minecraft.world.level.chunk.storage.RegionFile$ChunkBuffer(net.minecraft.world.level.chunk.storage.RegionFile regionFile, net.minecraft.world.level.ChunkPos $$0) {
         this.this$0 = regionFile;
         super(8096);
         super.write(0);
@@ -17,7 +17,7 @@
         super.write(0);
         super.write(0);
         super.write(regionFile.version.getId());
-        this.pos = pos;
+        this.pos = $$0;
     }
 
     public void close() throws java.io.IOException {