--- net/minecraft/world/level/chunk/storage/RegionBitmap.java
+++ net/minecraft/world/level/chunk/storage/RegionBitmap.java
@@ -11,24 +11,24 @@
         this.used = new java.util.BitSet();
     }
 
-    public void force(int sectorOffset, int sectorCount) {
-        this.used.set(sectorOffset, sectorOffset + sectorCount);
+    public void force(int $$0, int $$1) {
+        this.used.set($$0, $$0 + $$1);
     }
 
-    public void free(int sectorOffset, int sectorCount) {
-        this.used.clear(sectorOffset, sectorOffset + sectorCount);
+    public void free(int $$0, int $$1) {
+        this.used.clear($$0, $$0 + $$1);
     }
 
-    public int allocate(int sectorCount) {
-        int n = 0;
+    public int allocate(int $$0) {
+        int $$1 = 0;
         while (true) {
-            int n2;
-            int n3;
-            if ((n3 = this.used.nextSetBit(n2 = this.used.nextClearBit(n))) == -1 || n3 - n2 >= sectorCount) {
-                this.force(n2, sectorCount);
-                return n2;
+            int $$2;
+            int $$3;
+            if (($$3 = this.used.nextSetBit($$2 = this.used.nextClearBit($$1))) == -1 || $$3 - $$2 >= $$0) {
+                this.force($$2, $$0);
+                return $$2;
             }
-            n = n3;
+            $$1 = $$3;
         }
     }
 