--- net/minecraft/world/level/chunk/storage/SectionStorage.java
+++ net/minecraft/world/level/chunk/storage/SectionStorage.java
@@ -155,13 +155,13 @@
     public void onSectionLoad(long $$0) {
     }
 
-    public void setDirty(long sectionPos) {
-        java.util.Optional optional = (java.util.Optional)this.storage.get(sectionPos);
-        if (optional == null || !optional.isPresent()) {
-            LOGGER.warn("No data for position: {}", (java.lang.Object)net.minecraft.core.SectionPos.of((long)sectionPos));
+    public void setDirty(long $$0) {
+        java.util.Optional $$1 = (java.util.Optional)this.storage.get($$0);
+        if ($$1 == null || !$$1.isPresent()) {
+            LOGGER.warn("No data for position: {}", (java.lang.Object)net.minecraft.core.SectionPos.of((long)$$0));
             return;
         }
-        this.dirty.add(sectionPos);
+        this.dirty.add($$0);
     }
 
     public static int getVersion(com.mojang.serialization.Dynamic<?> $$0) {