--- net/minecraft/world/entity/ai/sensing/DummySensor.java
+++ net/minecraft/world/entity/ai/sensing/DummySensor.java
@@ -9,7 +9,7 @@
     public net.minecraft.world.entity.ai.sensing.DummySensor() {
     }
 
-    public void doTick(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity) {
+    public void doTick(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
     }
 
     public java.util.Set<net.minecraft.world.entity.ai.memory.MemoryModuleType<?>> requires() {