--- net/minecraft/world/entity/ai/sensing/GolemSensor.java
+++ net/minecraft/world/entity/ai/sensing/GolemSensor.java
@@ -17,27 +17,27 @@
         super($$0);
     }
 
-    public void doTick(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity) {
-        net.minecraft.world.entity.ai.sensing.GolemSensor.checkForNearbyGolem((net.minecraft.world.entity.LivingEntity)entity);
+    public void doTick(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        net.minecraft.world.entity.ai.sensing.GolemSensor.checkForNearbyGolem((net.minecraft.world.entity.LivingEntity)$$1);
     }
 
     public java.util.Set<net.minecraft.world.entity.ai.memory.MemoryModuleType<?>> requires() {
         return com.google.common.collect.ImmutableSet.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_LIVING_ENTITIES);
     }
 
-    public static void checkForNearbyGolem(net.minecraft.world.entity.LivingEntity livingEntity) {
-        java.util.Optional optional = livingEntity.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_LIVING_ENTITIES);
-        if (!optional.isPresent()) {
+    public static void checkForNearbyGolem(net.minecraft.world.entity.LivingEntity $$02) {
+        java.util.Optional $$1 = $$02.getBrain().getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_LIVING_ENTITIES);
+        if (!$$1.isPresent()) {
             return;
         }
-        boolean bl = ((java.util.List)optional.get()).stream().anyMatch($$0 -> $$0.getType().equals((java.lang.Object)net.minecraft.world.entity.EntityType.IRON_GOLEM));
-        if (bl) {
-            net.minecraft.world.entity.ai.sensing.GolemSensor.golemDetected((net.minecraft.world.entity.LivingEntity)livingEntity);
+        boolean $$2 = ((java.util.List)$$1.get()).stream().anyMatch($$0 -> $$0.getType().equals((java.lang.Object)net.minecraft.world.entity.EntityType.IRON_GOLEM));
+        if ($$2) {
+            net.minecraft.world.entity.ai.sensing.GolemSensor.golemDetected((net.minecraft.world.entity.LivingEntity)$$02);
         }
     }
 
-    public static void golemDetected(net.minecraft.world.entity.LivingEntity livingEntity) {
-        livingEntity.getBrain().setMemoryWithExpiry(net.minecraft.world.entity.ai.memory.MemoryModuleType.GOLEM_DETECTED_RECENTLY, (java.lang.Object)true, 600L);
+    public static void golemDetected(net.minecraft.world.entity.LivingEntity $$0) {
+        $$0.getBrain().setMemoryWithExpiry(net.minecraft.world.entity.ai.memory.MemoryModuleType.GOLEM_DETECTED_RECENTLY, (java.lang.Object)true, 600L);
     }
 }
 