--- net/minecraft/world/entity/ai/sensing/HurtBySensor.java
+++ net/minecraft/world/entity/ai/sensing/HurtBySensor.java
@@ -13,21 +13,21 @@
         return com.google.common.collect.ImmutableSet.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY, (java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY);
     }
 
-    public void doTick(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity) {
-        net.minecraft.world.entity.ai.Brain brain = entity.getBrain();
-        net.minecraft.world.damagesource.DamageSource damageSource = entity.getLastDamageSource();
-        if (damageSource != null) {
-            brain.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY, (java.lang.Object)entity.getLastDamageSource());
-            net.minecraft.world.entity.Entity entity2 = damageSource.getEntity();
-            if (entity2 instanceof net.minecraft.world.entity.LivingEntity) {
-                brain.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY, (java.lang.Object)((net.minecraft.world.entity.LivingEntity)entity2));
+    public void doTick(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        net.minecraft.world.entity.ai.Brain $$22 = $$1.getBrain();
+        net.minecraft.world.damagesource.DamageSource $$3 = $$1.getLastDamageSource();
+        if ($$3 != null) {
+            $$22.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY, (java.lang.Object)$$1.getLastDamageSource());
+            net.minecraft.world.entity.Entity $$4 = $$3.getEntity();
+            if ($$4 instanceof net.minecraft.world.entity.LivingEntity) {
+                $$22.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY, (java.lang.Object)((net.minecraft.world.entity.LivingEntity)$$4));
             }
         } else {
-            brain.eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY);
+            $$22.eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY);
         }
-        brain.getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY).ifPresent($$2 -> {
-            if (!$$2.isAlive() || $$2.level != level) {
-                brain.eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY);
+        $$22.getMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY).ifPresent($$2 -> {
+            if (!$$2.isAlive() || $$2.level != $$0) {
+                $$22.eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.HURT_BY_ENTITY);
             }
         });
     }