--- net/minecraft/world/entity/ai/sensing/NearestLivingEntitySensor.java
+++ net/minecraft/world/entity/ai/sensing/NearestLivingEntitySensor.java
@@ -9,13 +9,13 @@
     public net.minecraft.world.entity.ai.sensing.NearestLivingEntitySensor() {
     }
 
-    public void doTick(net.minecraft.server.level.ServerLevel level, T entity) {
-        net.minecraft.world.phys.AABB aABB = entity.getBoundingBox().inflate((double)this.radiusXZ(), (double)this.radiusY(), (double)this.radiusXZ());
-        java.util.List list = level.getEntitiesOfClass(net.minecraft.world.entity.LivingEntity.class, aABB, $$1 -> $$1 != entity && $$1.isAlive());
-        list.sort(java.util.Comparator.comparingDouble(arg_0 -> entity.distanceToSqr(arg_0)));
-        net.minecraft.world.entity.ai.Brain brain = entity.getBrain();
-        brain.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_LIVING_ENTITIES, (java.lang.Object)list);
-        brain.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES, (java.lang.Object)new net.minecraft.world.entity.ai.memory.NearestVisibleLivingEntities(entity, list));
+    public void doTick(net.minecraft.server.level.ServerLevel $$0, T $$12) {
+        net.minecraft.world.phys.AABB $$2 = $$12.getBoundingBox().inflate((double)this.radiusXZ(), (double)this.radiusY(), (double)this.radiusXZ());
+        java.util.List $$3 = $$0.getEntitiesOfClass(net.minecraft.world.entity.LivingEntity.class, $$2, $$1 -> $$1 != $$12 && $$1.isAlive());
+        $$3.sort(java.util.Comparator.comparingDouble(arg_0 -> $$12.distanceToSqr(arg_0)));
+        net.minecraft.world.entity.ai.Brain $$4 = $$12.getBrain();
+        $$4.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_LIVING_ENTITIES, (java.lang.Object)$$3);
+        $$4.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.NEAREST_VISIBLE_LIVING_ENTITIES, (java.lang.Object)new net.minecraft.world.entity.ai.memory.NearestVisibleLivingEntities($$12, $$3));
     }
 
     public int radiusXZ() {