--- net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
+++ net/minecraft/world/entity/ai/sensing/SecondaryPoiSensor.java
@@ -12,25 +12,25 @@
         super(40);
     }
 
-    public void doTick(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.npc.Villager entity) {
-        net.minecraft.resources.ResourceKey resourceKey = level.dimension();
-        net.minecraft.core.BlockPos blockPos = entity.blockPosition();
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        int n = 4;
-        for (int i = -4; i <= 4; ++i) {
-            for (int j = -2; j <= 2; ++j) {
-                for (int k = -4; k <= 4; ++k) {
-                    net.minecraft.core.BlockPos blockPos2 = blockPos.offset(i, j, k);
-                    if (!entity.getVillagerData().getProfession().secondaryPoi().contains((java.lang.Object)level.getBlockState(blockPos2).getBlock())) continue;
-                    arrayList.add((java.lang.Object)net.minecraft.core.GlobalPos.of((net.minecraft.resources.ResourceKey)resourceKey, (net.minecraft.core.BlockPos)blockPos2));
+    public void doTick(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.npc.Villager $$1) {
+        net.minecraft.resources.ResourceKey $$2 = $$0.dimension();
+        net.minecraft.core.BlockPos $$3 = $$1.blockPosition();
+        java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayList();
+        int $$5 = 4;
+        for (int $$6 = -4; $$6 <= 4; ++$$6) {
+            for (int $$7 = -2; $$7 <= 2; ++$$7) {
+                for (int $$8 = -4; $$8 <= 4; ++$$8) {
+                    net.minecraft.core.BlockPos $$9 = $$3.offset($$6, $$7, $$8);
+                    if (!$$1.getVillagerData().getProfession().secondaryPoi().contains((java.lang.Object)$$0.getBlockState($$9).getBlock())) continue;
+                    $$4.add((java.lang.Object)net.minecraft.core.GlobalPos.of((net.minecraft.resources.ResourceKey)$$2, (net.minecraft.core.BlockPos)$$9));
                 }
             }
         }
-        net.minecraft.world.entity.ai.Brain brain = entity.getBrain();
-        if (!arrayList.isEmpty()) {
-            brain.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.SECONDARY_JOB_SITE, (java.lang.Object)arrayList);
+        net.minecraft.world.entity.ai.Brain $$10 = $$1.getBrain();
+        if (!$$4.isEmpty()) {
+            $$10.setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.SECONDARY_JOB_SITE, (java.lang.Object)$$4);
         } else {
-            brain.eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.SECONDARY_JOB_SITE);
+            $$10.eraseMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.SECONDARY_JOB_SITE);
         }
     }
 