--- net/minecraft/world/entity/ai/sensing/SensorType.java
+++ net/minecraft/world/entity/ai/sensing/SensorType.java
@@ -23,6 +23,7 @@
     public static net.minecraft.world.entity.ai.sensing.SensorType<net.minecraft.world.entity.ai.sensing.TemptingSensor> AXOLOTL_TEMPTATIONS;
     public static net.minecraft.world.entity.ai.sensing.SensorType<net.minecraft.world.entity.ai.sensing.TemptingSensor> GOAT_TEMPTATIONS;
     public static net.minecraft.world.entity.ai.sensing.SensorType<net.minecraft.world.entity.ai.sensing.TemptingSensor> FROG_TEMPTATIONS;
+    public static net.minecraft.world.entity.ai.sensing.SensorType<net.minecraft.world.entity.ai.sensing.TemptingSensor> CAMEL_TEMPTATIONS;
     public static net.minecraft.world.entity.ai.sensing.SensorType<net.minecraft.world.entity.ai.sensing.FrogAttackablesSensor> FROG_ATTACKABLES;
     public static net.minecraft.world.entity.ai.sensing.SensorType<net.minecraft.world.entity.ai.sensing.IsInWaterSensor> IS_IN_WATER;
     public static net.minecraft.world.entity.ai.sensing.SensorType<net.minecraft.world.entity.ai.sensing.WardenEntitySensor> WARDEN_ENTITY_SENSOR;
@@ -36,10 +37,14 @@
         return (U)((net.minecraft.world.entity.ai.sensing.Sensor)this.factory.get());
     }
 
-    public static <U extends net.minecraft.world.entity.ai.sensing.Sensor<?>> net.minecraft.world.entity.ai.sensing.SensorType<U> register(java.lang.String key, java.util.function.Supplier<U> sensorSupplier) {
-        return (net.minecraft.world.entity.ai.sensing.SensorType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.SENSOR_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(key), (java.lang.Object)new net.minecraft.world.entity.ai.sensing.SensorType(sensorSupplier));
+    public static <U extends net.minecraft.world.entity.ai.sensing.Sensor<?>> net.minecraft.world.entity.ai.sensing.SensorType<U> register(java.lang.String $$0, java.util.function.Supplier<U> $$1) {
+        return (net.minecraft.world.entity.ai.sensing.SensorType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.SENSOR_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0), (java.lang.Object)new net.minecraft.world.entity.ai.sensing.SensorType($$1));
     }
 
+    public static /* synthetic */ net.minecraft.world.entity.ai.sensing.TemptingSensor lambda$static$3() {
+        return new net.minecraft.world.entity.ai.sensing.TemptingSensor(net.minecraft.world.entity.animal.camel.CamelAi.getTemptations());
+    }
+
     public static /* synthetic */ net.minecraft.world.entity.ai.sensing.TemptingSensor lambda$static$2() {
         return new net.minecraft.world.entity.ai.sensing.TemptingSensor(net.minecraft.world.entity.animal.frog.FrogAi.getTemptations());
     }
@@ -71,6 +76,7 @@
         AXOLOTL_TEMPTATIONS = net.minecraft.world.entity.ai.sensing.SensorType.register((java.lang.String)"axolotl_temptations", () -> new net.minecraft.world.entity.ai.sensing.TemptingSensor(net.minecraft.world.entity.animal.axolotl.AxolotlAi.getTemptations()));
         GOAT_TEMPTATIONS = net.minecraft.world.entity.ai.sensing.SensorType.register((java.lang.String)"goat_temptations", () -> new net.minecraft.world.entity.ai.sensing.TemptingSensor(net.minecraft.world.entity.animal.goat.GoatAi.getTemptations()));
         FROG_TEMPTATIONS = net.minecraft.world.entity.ai.sensing.SensorType.register((java.lang.String)"frog_temptations", () -> new net.minecraft.world.entity.ai.sensing.TemptingSensor(net.minecraft.world.entity.animal.frog.FrogAi.getTemptations()));
+        CAMEL_TEMPTATIONS = net.minecraft.world.entity.ai.sensing.SensorType.register((java.lang.String)"camel_temptations", () -> new net.minecraft.world.entity.ai.sensing.TemptingSensor(net.minecraft.world.entity.animal.camel.CamelAi.getTemptations()));
         FROG_ATTACKABLES = net.minecraft.world.entity.ai.sensing.SensorType.register((java.lang.String)"frog_attackables", net.minecraft.world.entity.ai.sensing.FrogAttackablesSensor::new);
         IS_IN_WATER = net.minecraft.world.entity.ai.sensing.SensorType.register((java.lang.String)"is_in_water", net.minecraft.world.entity.ai.sensing.IsInWaterSensor::new);
         WARDEN_ENTITY_SENSOR = net.minecraft.world.entity.ai.sensing.SensorType.register((java.lang.String)"warden_entity_sensor", net.minecraft.world.entity.ai.sensing.WardenEntitySensor::new);