--- net/minecraft/world/entity/ai/sensing/VillagerBabiesSensor.java
+++ net/minecraft/world/entity/ai/sensing/VillagerBabiesSensor.java
@@ -13,16 +13,16 @@
         return com.google.common.collect.ImmutableSet.of((java.lang.Object)net.minecraft.world.entity.ai.memory.MemoryModuleType.VISIBLE_VILLAGER_BABIES);
     }
 
-    public void doTick(net.minecraft.server.level.ServerLevel level, net.minecraft.world.entity.LivingEntity entity) {
-        entity.getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.VISIBLE_VILLAGER_BABIES, (java.lang.Object)this.getNearestVillagerBabies(entity));
+    public void doTick(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.LivingEntity $$1) {
+        $$1.getBrain().setMemory(net.minecraft.world.entity.ai.memory.MemoryModuleType.VISIBLE_VILLAGER_BABIES, (java.lang.Object)this.getNearestVillagerBabies($$1));
     }
 
-    public java.util.List<net.minecraft.world.entity.LivingEntity> getNearestVillagerBabies(net.minecraft.world.entity.LivingEntity livingEntity) {
-        return com.google.common.collect.ImmutableList.copyOf((java.lang.Iterable)this.getVisibleEntities(livingEntity).findAll(arg_0 -> this.isVillagerBaby(arg_0)));
+    public java.util.List<net.minecraft.world.entity.LivingEntity> getNearestVillagerBabies(net.minecraft.world.entity.LivingEntity $$0) {
+        return com.google.common.collect.ImmutableList.copyOf((java.lang.Iterable)this.getVisibleEntities($$0).findAll(arg_0 -> this.isVillagerBaby(arg_0)));
     }
 
-    public boolean isVillagerBaby(net.minecraft.world.entity.LivingEntity livingEntity) {
-        return livingEntity.getType() == net.minecraft.world.entity.EntityType.VILLAGER && livingEntity.isBaby();
+    public boolean isVillagerBaby(net.minecraft.world.entity.LivingEntity $$0) {
+        return $$0.getType() == net.minecraft.world.entity.EntityType.VILLAGER && $$0.isBaby();
     }
 
     public net.minecraft.world.entity.ai.memory.NearestVisibleLivingEntities getVisibleEntities(net.minecraft.world.entity.LivingEntity $$0) {