--- net/minecraft/client/color/item/ItemColors.java
+++ net/minecraft/client/color/item/ItemColors.java
@@ -12,11 +12,11 @@
         this.itemColors = new net.minecraft.core.IdMapper(32);
     }
 
-    public static net.minecraft.client.color.item.ItemColors createDefault(net.minecraft.client.color.block.BlockColors colors) {
-        net.minecraft.client.color.item.ItemColors itemColors = new net.minecraft.client.color.item.ItemColors();
-        itemColors.register(($$0, $$1) -> $$1 > 0 ? -1 : ((net.minecraft.world.item.DyeableLeatherItem)$$0.getItem()).getColor($$0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.LEATHER_HELMET, net.minecraft.world.item.Items.LEATHER_CHESTPLATE, net.minecraft.world.item.Items.LEATHER_LEGGINGS, net.minecraft.world.item.Items.LEATHER_BOOTS, net.minecraft.world.item.Items.LEATHER_HORSE_ARMOR});
-        itemColors.register(($$0, $$1) -> net.minecraft.world.level.GrassColor.get((double)0.5, (double)1.0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.level.block.Blocks.TALL_GRASS, net.minecraft.world.level.block.Blocks.LARGE_FERN});
-        itemColors.register(($$0, $$1) -> {
+    public static net.minecraft.client.color.item.ItemColors createDefault(net.minecraft.client.color.block.BlockColors $$02) {
+        net.minecraft.client.color.item.ItemColors $$12 = new net.minecraft.client.color.item.ItemColors();
+        $$12.register(($$0, $$1) -> $$1 > 0 ? -1 : ((net.minecraft.world.item.DyeableLeatherItem)$$0.getItem()).getColor($$0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.LEATHER_HELMET, net.minecraft.world.item.Items.LEATHER_CHESTPLATE, net.minecraft.world.item.Items.LEATHER_LEGGINGS, net.minecraft.world.item.Items.LEATHER_BOOTS, net.minecraft.world.item.Items.LEATHER_HORSE_ARMOR});
+        $$12.register(($$0, $$1) -> net.minecraft.world.level.GrassColor.get((double)0.5, (double)1.0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.level.block.Blocks.TALL_GRASS, net.minecraft.world.level.block.Blocks.LARGE_FERN});
+        $$12.register(($$0, $$1) -> {
             int[] $$3;
             if ($$1 != 1) {
                 return -1;
@@ -39,28 +39,28 @@
             }
             return ($$4 /= $$3.length) << 16 | ($$5 /= $$3.length) << 8 | ($$6 /= $$3.length);
         }, new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.FIREWORK_STAR});
-        itemColors.register(($$0, $$1) -> $$1 > 0 ? -1 : net.minecraft.world.item.alchemy.PotionUtils.getColor((net.minecraft.world.item.ItemStack)$$0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.POTION, net.minecraft.world.item.Items.SPLASH_POTION, net.minecraft.world.item.Items.LINGERING_POTION});
-        for (net.minecraft.world.item.SpawnEggItem spawnEggItem : net.minecraft.world.item.SpawnEggItem.eggs()) {
-            itemColors.register(($$1, $$2) -> spawnEggItem.getColor($$2), new net.minecraft.world.level.ItemLike[]{spawnEggItem});
+        $$12.register(($$0, $$1) -> $$1 > 0 ? -1 : net.minecraft.world.item.alchemy.PotionUtils.getColor((net.minecraft.world.item.ItemStack)$$0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.POTION, net.minecraft.world.item.Items.SPLASH_POTION, net.minecraft.world.item.Items.LINGERING_POTION});
+        for (net.minecraft.world.item.SpawnEggItem $$22 : net.minecraft.world.item.SpawnEggItem.eggs()) {
+            $$12.register(($$1, $$2) -> $$22.getColor($$2), new net.minecraft.world.level.ItemLike[]{$$22});
         }
-        itemColors.register(($$1, $$2) -> {
+        $$12.register(($$1, $$2) -> {
             net.minecraft.world.level.block.state.BlockState $$3 = ((net.minecraft.world.item.BlockItem)$$1.getItem()).getBlock().defaultBlockState();
-            return colors.getColor($$3, null, null, $$2);
+            return $$02.getColor($$3, null, null, $$2);
         }, new net.minecraft.world.level.ItemLike[]{net.minecraft.world.level.block.Blocks.GRASS_BLOCK, net.minecraft.world.level.block.Blocks.GRASS, net.minecraft.world.level.block.Blocks.FERN, net.minecraft.world.level.block.Blocks.VINE, net.minecraft.world.level.block.Blocks.OAK_LEAVES, net.minecraft.world.level.block.Blocks.SPRUCE_LEAVES, net.minecraft.world.level.block.Blocks.BIRCH_LEAVES, net.minecraft.world.level.block.Blocks.JUNGLE_LEAVES, net.minecraft.world.level.block.Blocks.ACACIA_LEAVES, net.minecraft.world.level.block.Blocks.DARK_OAK_LEAVES, net.minecraft.world.level.block.Blocks.LILY_PAD});
-        itemColors.register(($$0, $$1) -> net.minecraft.world.level.FoliageColor.getMangroveColor(), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.level.block.Blocks.MANGROVE_LEAVES});
-        itemColors.register(($$0, $$1) -> $$1 == 0 ? net.minecraft.world.item.alchemy.PotionUtils.getColor((net.minecraft.world.item.ItemStack)$$0) : -1, new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.TIPPED_ARROW});
-        itemColors.register(($$0, $$1) -> $$1 == 0 ? -1 : net.minecraft.world.item.MapItem.getColor((net.minecraft.world.item.ItemStack)$$0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.FILLED_MAP});
-        return itemColors;
+        $$12.register(($$0, $$1) -> net.minecraft.world.level.FoliageColor.getMangroveColor(), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.level.block.Blocks.MANGROVE_LEAVES});
+        $$12.register(($$0, $$1) -> $$1 == 0 ? net.minecraft.world.item.alchemy.PotionUtils.getColor((net.minecraft.world.item.ItemStack)$$0) : -1, new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.TIPPED_ARROW});
+        $$12.register(($$0, $$1) -> $$1 == 0 ? -1 : net.minecraft.world.item.MapItem.getColor((net.minecraft.world.item.ItemStack)$$0), new net.minecraft.world.level.ItemLike[]{net.minecraft.world.item.Items.FILLED_MAP});
+        return $$12;
     }
 
-    public int getColor(net.minecraft.world.item.ItemStack stack, int tintIndex) {
-        net.minecraft.client.color.item.ItemColor itemColor = (net.minecraft.client.color.item.ItemColor)this.itemColors.byId(net.minecraft.core.Registry.ITEM.getId((java.lang.Object)stack.getItem()));
-        return itemColor == null ? -1 : itemColor.getColor(stack, tintIndex);
+    public int getColor(net.minecraft.world.item.ItemStack $$0, int $$1) {
+        net.minecraft.client.color.item.ItemColor $$2 = (net.minecraft.client.color.item.ItemColor)this.itemColors.byId(net.minecraft.core.Registry.ITEM.getId((java.lang.Object)$$0.getItem()));
+        return $$2 == null ? -1 : $$2.getColor($$0, $$1);
     }
 
-    public void register(net.minecraft.client.color.item.ItemColor itemColor, net.minecraft.world.level.ItemLike ... items) {
-        for (net.minecraft.world.level.ItemLike itemLike : items) {
-            this.itemColors.addMapping((java.lang.Object)itemColor, net.minecraft.world.item.Item.getId((net.minecraft.world.item.Item)itemLike.asItem()));
+    public void register(net.minecraft.client.color.item.ItemColor $$0, net.minecraft.world.level.ItemLike ... $$1) {
+        for (net.minecraft.world.level.ItemLike $$2 : $$1) {
+            this.itemColors.addMapping((java.lang.Object)$$0, net.minecraft.world.item.Item.getId((net.minecraft.world.item.Item)$$2.asItem()));
         }
     }
 }