--- net/minecraft/client/resources/metadata/texture/TextureMetadataSection.java
+++ net/minecraft/client/resources/metadata/texture/TextureMetadataSection.java
@@ -11,9 +11,9 @@
     public boolean blur;
     public boolean clamp;
 
-    public net.minecraft.client.resources.metadata.texture.TextureMetadataSection(boolean blur, boolean clamp) {
-        this.blur = blur;
-        this.clamp = clamp;
+    public net.minecraft.client.resources.metadata.texture.TextureMetadataSection(boolean $$0, boolean $$1) {
+        this.blur = $$0;
+        this.clamp = $$1;
     }
 
     public boolean isBlur() {