--- net/minecraft/client/resources/metadata/texture/TextureMetadataSectionSerializer.java
+++ net/minecraft/client/resources/metadata/texture/TextureMetadataSectionSerializer.java
@@ -9,10 +9,10 @@
     public net.minecraft.client.resources.metadata.texture.TextureMetadataSectionSerializer() {
     }
 
-    public net.minecraft.client.resources.metadata.texture.TextureMetadataSection fromJson(com.google.gson.JsonObject json) {
-        boolean bl = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)json, (java.lang.String)"blur", (boolean)false);
-        boolean bl2 = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)json, (java.lang.String)"clamp", (boolean)false);
-        return new net.minecraft.client.resources.metadata.texture.TextureMetadataSection(bl, bl2);
+    public net.minecraft.client.resources.metadata.texture.TextureMetadataSection fromJson(com.google.gson.JsonObject $$0) {
+        boolean $$1 = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)$$0, (java.lang.String)"blur", (boolean)false);
+        boolean $$2 = net.minecraft.util.GsonHelper.getAsBoolean((com.google.gson.JsonObject)$$0, (java.lang.String)"clamp", (boolean)false);
+        return new net.minecraft.client.resources.metadata.texture.TextureMetadataSection($$1, $$2);
     }
 
     public java.lang.String getMetadataSectionName() {