--- com/mojang/blaze3d/vertex/SheetedDecalTextureGenerator.java
+++ com/mojang/blaze3d/vertex/SheetedDecalTextureGenerator.java
@@ -19,11 +19,11 @@
     public float ny;
     public float nz;
 
-    public com.mojang.blaze3d.vertex.SheetedDecalTextureGenerator(com.mojang.blaze3d.vertex.VertexConsumer delegate, com.mojang.math.Matrix4f cameraPose, com.mojang.math.Matrix3f normalPose) {
-        this.delegate = delegate;
-        this.cameraInversePose = cameraPose.copy();
+    public com.mojang.blaze3d.vertex.SheetedDecalTextureGenerator(com.mojang.blaze3d.vertex.VertexConsumer $$0, com.mojang.math.Matrix4f $$1, com.mojang.math.Matrix3f $$2) {
+        this.delegate = $$0;
+        this.cameraInversePose = $$1.copy();
         this.cameraInversePose.invert();
-        this.normalInversePose = normalPose.copy();
+        this.normalInversePose = $$2.copy();
         this.normalInversePose.invert();
         this.resetState();
     }
@@ -55,36 +55,36 @@
         this.resetState();
     }
 
-    public com.mojang.blaze3d.vertex.VertexConsumer vertex(double x, double y, double z) {
-        this.x = (float)x;
-        this.y = (float)y;
-        this.z = (float)z;
+    public com.mojang.blaze3d.vertex.VertexConsumer vertex(double $$0, double $$1, double $$2) {
+        this.x = (float)$$0;
+        this.y = (float)$$1;
+        this.z = (float)$$2;
         return this;
     }
 
-    public com.mojang.blaze3d.vertex.VertexConsumer color(int red, int green, int blue, int alpha) {
+    public com.mojang.blaze3d.vertex.VertexConsumer color(int $$0, int $$1, int $$2, int $$3) {
         return this;
     }
 
-    public com.mojang.blaze3d.vertex.VertexConsumer uv(float u, float v) {
+    public com.mojang.blaze3d.vertex.VertexConsumer uv(float $$0, float $$1) {
         return this;
     }
 
-    public com.mojang.blaze3d.vertex.VertexConsumer overlayCoords(int u, int v) {
-        this.overlayU = u;
-        this.overlayV = v;
+    public com.mojang.blaze3d.vertex.VertexConsumer overlayCoords(int $$0, int $$1) {
+        this.overlayU = $$0;
+        this.overlayV = $$1;
         return this;
     }
 
-    public com.mojang.blaze3d.vertex.VertexConsumer uv2(int u, int v) {
-        this.lightCoords = u | v << 16;
+    public com.mojang.blaze3d.vertex.VertexConsumer uv2(int $$0, int $$1) {
+        this.lightCoords = $$0 | $$1 << 16;
         return this;
     }
 
-    public com.mojang.blaze3d.vertex.VertexConsumer normal(float x, float y, float z) {
-        this.nx = x;
-        this.ny = y;
-        this.nz = z;
+    public com.mojang.blaze3d.vertex.VertexConsumer normal(float $$0, float $$1, float $$2) {
+        this.nx = $$0;
+        this.ny = $$1;
+        this.nz = $$2;
         return this;
     }
 }