--- com/mojang/blaze3d/vertex/Tesselator.java
+++ com/mojang/blaze3d/vertex/Tesselator.java
@@ -15,8 +15,8 @@
         return INSTANCE;
     }
 
-    public com.mojang.blaze3d.vertex.Tesselator(int capacity) {
-        this.builder = new com.mojang.blaze3d.vertex.BufferBuilder(capacity);
+    public com.mojang.blaze3d.vertex.Tesselator(int $$0) {
+        this.builder = new com.mojang.blaze3d.vertex.BufferBuilder($$0);
     }
 
     public com.mojang.blaze3d.vertex.Tesselator() {