--- com/mojang/blaze3d/vertex/VertexFormatElement.java
+++ com/mojang/blaze3d/vertex/VertexFormatElement.java
@@ -11,19 +11,19 @@
     public int count;
     public int byteSize;
 
-    public com.mojang.blaze3d.vertex.VertexFormatElement(int index, com.mojang.blaze3d.vertex.VertexFormatElement$Type type, com.mojang.blaze3d.vertex.VertexFormatElement$Usage usage, int count) {
-        if (!this.supportsUsage(index, usage)) {
+    public com.mojang.blaze3d.vertex.VertexFormatElement(int $$0, com.mojang.blaze3d.vertex.VertexFormatElement$Type $$1, com.mojang.blaze3d.vertex.VertexFormatElement$Usage $$2, int $$3) {
+        if (!this.supportsUsage($$0, $$2)) {
             throw new java.lang.IllegalStateException("Multiple vertex elements of the same type other than UVs are not supported");
         }
-        this.usage = usage;
-        this.type = type;
-        this.index = index;
-        this.count = count;
-        this.byteSize = type.getSize() * this.count;
+        this.usage = $$2;
+        this.type = $$1;
+        this.index = $$0;
+        this.count = $$3;
+        this.byteSize = $$1.getSize() * this.count;
     }
 
-    public boolean supportsUsage(int index, com.mojang.blaze3d.vertex.VertexFormatElement$Usage usage) {
-        return index == 0 || usage == com.mojang.blaze3d.vertex.VertexFormatElement$Usage.UV;
+    public boolean supportsUsage(int $$0, com.mojang.blaze3d.vertex.VertexFormatElement$Usage $$1) {
+        return $$0 == 0 || $$1 == com.mojang.blaze3d.vertex.VertexFormatElement$Usage.UV;
     }
 
     public com.mojang.blaze3d.vertex.VertexFormatElement$Type getType() {
@@ -54,24 +54,24 @@
         return this.usage == com.mojang.blaze3d.vertex.VertexFormatElement$Usage.POSITION;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other == null || this.getClass() != other.getClass()) {
+        if ($$0 == null || this.getClass() != $$0.getClass()) {
             return false;
         }
-        com.mojang.blaze3d.vertex.VertexFormatElement vertexFormatElement = (com.mojang.blaze3d.vertex.VertexFormatElement)other;
-        if (this.count != vertexFormatElement.count) {
+        com.mojang.blaze3d.vertex.VertexFormatElement $$1 = (com.mojang.blaze3d.vertex.VertexFormatElement)$$0;
+        if (this.count != $$1.count) {
             return false;
         }
-        if (this.index != vertexFormatElement.index) {
+        if (this.index != $$1.index) {
             return false;
         }
-        if (this.type != vertexFormatElement.type) {
+        if (this.type != $$1.type) {
             return false;
         }
-        return this.usage == vertexFormatElement.usage;
+        return this.usage == $$1.usage;
     }
 
     public int hashCode() {