--- com/mojang/blaze3d/vertex/VertexFormatElement$Type.java
+++ com/mojang/blaze3d/vertex/VertexFormatElement$Type.java
@@ -26,11 +26,11 @@
         return (com.mojang.blaze3d.vertex.VertexFormatElement$Type)java.lang.Enum.valueOf(com.mojang.blaze3d.vertex.VertexFormatElement$Type.class, (java.lang.String)$$0);
     }
 
-    public com.mojang.blaze3d.vertex.VertexFormatElement$Type(int size, java.lang.String name, int glType) {
+    public com.mojang.blaze3d.vertex.VertexFormatElement$Type(int $$0, java.lang.String $$1, int $$2) {
         super(string, n);
-        this.size = size;
-        this.name = name;
-        this.glType = glType;
+        this.size = $$0;
+        this.name = $$1;
+        this.glType = $$2;
     }
 
     public int getSize() {