--- com/mojang/blaze3d/vertex/VertexMultiConsumer.java
+++ com/mojang/blaze3d/vertex/VertexMultiConsumer.java
@@ -16,8 +16,8 @@
         return $$0;
     }
 
-    public static com.mojang.blaze3d.vertex.VertexConsumer create(com.mojang.blaze3d.vertex.VertexConsumer first, com.mojang.blaze3d.vertex.VertexConsumer second) {
-        return new com.mojang.blaze3d.vertex.VertexMultiConsumer$Double(first, second);
+    public static com.mojang.blaze3d.vertex.VertexConsumer create(com.mojang.blaze3d.vertex.VertexConsumer $$0, com.mojang.blaze3d.vertex.VertexConsumer $$1) {
+        return new com.mojang.blaze3d.vertex.VertexMultiConsumer$Double($$0, $$1);
     }
 
     public static com.mojang.blaze3d.vertex.VertexConsumer create(com.mojang.blaze3d.vertex.VertexConsumer ... $$0) {