--- com/mojang/blaze3d/vertex/VertexFormat.java
+++ com/mojang/blaze3d/vertex/VertexFormat.java
@@ -44,18 +44,18 @@
         return this.elementMapping.keySet().asList();
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other == null || this.getClass() != other.getClass()) {
+        if ($$0 == null || this.getClass() != $$0.getClass()) {
             return false;
         }
-        com.mojang.blaze3d.vertex.VertexFormat vertexFormat = (com.mojang.blaze3d.vertex.VertexFormat)other;
-        if (this.vertexSize != vertexFormat.vertexSize) {
+        com.mojang.blaze3d.vertex.VertexFormat $$1 = (com.mojang.blaze3d.vertex.VertexFormat)$$0;
+        if (this.vertexSize != $$1.vertexSize) {
             return false;
         }
-        return this.elementMapping.equals((java.lang.Object)vertexFormat.elementMapping);
+        return this.elementMapping.equals((java.lang.Object)$$1.elementMapping);
     }
 
     public int hashCode() {