--- net/minecraft/world/level/saveddata/maps/MapBanner.java
+++ net/minecraft/world/level/saveddata/maps/MapBanner.java
@@ -10,27 +10,27 @@
     @javax.annotation.Nullable
     public net.minecraft.network.chat.Component name;
 
-    public net.minecraft.world.level.saveddata.maps.MapBanner(net.minecraft.core.BlockPos pos, net.minecraft.world.item.DyeColor color, @javax.annotation.Nullable net.minecraft.network.chat.Component name) {
-        this.pos = pos;
-        this.color = color;
-        this.name = name;
+    public net.minecraft.world.level.saveddata.maps.MapBanner(net.minecraft.core.BlockPos $$0, net.minecraft.world.item.DyeColor $$1, @javax.annotation.Nullable net.minecraft.network.chat.Component $$2) {
+        this.pos = $$0;
+        this.color = $$1;
+        this.name = $$2;
     }
 
-    public static net.minecraft.world.level.saveddata.maps.MapBanner load(net.minecraft.nbt.CompoundTag compoundTag) {
-        net.minecraft.core.BlockPos blockPos = net.minecraft.nbt.NbtUtils.readBlockPos((net.minecraft.nbt.CompoundTag)compoundTag.getCompound("Pos"));
-        net.minecraft.world.item.DyeColor dyeColor = net.minecraft.world.item.DyeColor.byName((java.lang.String)compoundTag.getString("Color"), (net.minecraft.world.item.DyeColor)net.minecraft.world.item.DyeColor.WHITE);
-        net.minecraft.network.chat.MutableComponent mutableComponent = compoundTag.contains("Name") ? net.minecraft.network.chat.Component$Serializer.fromJson((java.lang.String)compoundTag.getString("Name")) : null;
-        return new net.minecraft.world.level.saveddata.maps.MapBanner(blockPos, dyeColor, (net.minecraft.network.chat.Component)mutableComponent);
+    public static net.minecraft.world.level.saveddata.maps.MapBanner load(net.minecraft.nbt.CompoundTag $$0) {
+        net.minecraft.core.BlockPos $$1 = net.minecraft.nbt.NbtUtils.readBlockPos((net.minecraft.nbt.CompoundTag)$$0.getCompound("Pos"));
+        net.minecraft.world.item.DyeColor $$2 = net.minecraft.world.item.DyeColor.byName((java.lang.String)$$0.getString("Color"), (net.minecraft.world.item.DyeColor)net.minecraft.world.item.DyeColor.WHITE);
+        net.minecraft.network.chat.MutableComponent $$3 = $$0.contains("Name") ? net.minecraft.network.chat.Component$Serializer.fromJson((java.lang.String)$$0.getString("Name")) : null;
+        return new net.minecraft.world.level.saveddata.maps.MapBanner($$1, $$2, (net.minecraft.network.chat.Component)$$3);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.world.level.saveddata.maps.MapBanner fromWorld(net.minecraft.world.level.BlockGetter level, net.minecraft.core.BlockPos pos) {
-        net.minecraft.world.level.block.entity.BlockEntity blockEntity = level.getBlockEntity(pos);
-        if (blockEntity instanceof net.minecraft.world.level.block.entity.BannerBlockEntity) {
-            net.minecraft.world.level.block.entity.BannerBlockEntity bannerBlockEntity = (net.minecraft.world.level.block.entity.BannerBlockEntity)blockEntity;
-            net.minecraft.world.item.DyeColor dyeColor = bannerBlockEntity.getBaseColor();
-            net.minecraft.network.chat.Component component = bannerBlockEntity.hasCustomName() ? bannerBlockEntity.getCustomName() : null;
-            return new net.minecraft.world.level.saveddata.maps.MapBanner(pos, dyeColor, component);
+    public static net.minecraft.world.level.saveddata.maps.MapBanner fromWorld(net.minecraft.world.level.BlockGetter $$0, net.minecraft.core.BlockPos $$1) {
+        net.minecraft.world.level.block.entity.BlockEntity $$2 = $$0.getBlockEntity($$1);
+        if ($$2 instanceof net.minecraft.world.level.block.entity.BannerBlockEntity) {
+            net.minecraft.world.level.block.entity.BannerBlockEntity $$3 = (net.minecraft.world.level.block.entity.BannerBlockEntity)$$2;
+            net.minecraft.world.item.DyeColor $$4 = $$3.getBaseColor();
+            net.minecraft.network.chat.Component $$5 = $$3.hasCustomName() ? $$3.getCustomName() : null;
+            return new net.minecraft.world.level.saveddata.maps.MapBanner($$1, $$4, $$5);
         }
         return null;
     }