--- net/minecraft/world/level/saveddata/maps/MapDecoration$Type.java
+++ net/minecraft/world/level/saveddata/maps/MapDecoration$Type.java
@@ -75,8 +75,8 @@
         return this.mapColor;
     }
 
-    public static net.minecraft.world.level.saveddata.maps.MapDecoration$Type byIcon(byte iconId) {
-        return net.minecraft.world.level.saveddata.maps.MapDecoration$Type.values()[net.minecraft.util.Mth.clamp((int)iconId, (int)0, (int)(net.minecraft.world.level.saveddata.maps.MapDecoration$Type.values().length - 1))];
+    public static net.minecraft.world.level.saveddata.maps.MapDecoration$Type byIcon(byte $$0) {
+        return net.minecraft.world.level.saveddata.maps.MapDecoration$Type.values()[net.minecraft.util.Mth.clamp((int)$$0, (int)0, (int)(net.minecraft.world.level.saveddata.maps.MapDecoration$Type.values().length - 1))];
     }
 
     public boolean shouldTrackCount() {