--- net/minecraft/world/level/saveddata/maps/MapDecoration.java
+++ net/minecraft/world/level/saveddata/maps/MapDecoration.java
@@ -12,12 +12,12 @@
     @javax.annotation.Nullable
     public net.minecraft.network.chat.Component name;
 
-    public net.minecraft.world.level.saveddata.maps.MapDecoration(net.minecraft.world.level.saveddata.maps.MapDecoration$Type type, byte x, byte y, byte rot, @javax.annotation.Nullable net.minecraft.network.chat.Component name) {
-        this.type = type;
-        this.x = x;
-        this.y = y;
-        this.rot = rot;
-        this.name = name;
+    public net.minecraft.world.level.saveddata.maps.MapDecoration(net.minecraft.world.level.saveddata.maps.MapDecoration$Type $$0, byte $$1, byte $$2, byte $$3, @javax.annotation.Nullable net.minecraft.network.chat.Component $$4) {
+        this.type = $$0;
+        this.x = $$1;
+        this.y = $$2;
+        this.rot = $$3;
+        this.name = $$4;
     }
 
     public byte getImage() {