--- net/minecraft/world/level/saveddata/maps/MapFrame.java
+++ net/minecraft/world/level/saveddata/maps/MapFrame.java
@@ -9,17 +9,17 @@
     public int rotation;
     public int entityId;
 
-    public net.minecraft.world.level.saveddata.maps.MapFrame(net.minecraft.core.BlockPos pos, int rotation, int entityId) {
-        this.pos = pos;
-        this.rotation = rotation;
-        this.entityId = entityId;
+    public net.minecraft.world.level.saveddata.maps.MapFrame(net.minecraft.core.BlockPos $$0, int $$1, int $$2) {
+        this.pos = $$0;
+        this.rotation = $$1;
+        this.entityId = $$2;
     }
 
-    public static net.minecraft.world.level.saveddata.maps.MapFrame load(net.minecraft.nbt.CompoundTag compoundTag) {
-        net.minecraft.core.BlockPos blockPos = net.minecraft.nbt.NbtUtils.readBlockPos((net.minecraft.nbt.CompoundTag)compoundTag.getCompound("Pos"));
-        int n = compoundTag.getInt("Rotation");
-        int n2 = compoundTag.getInt("EntityId");
-        return new net.minecraft.world.level.saveddata.maps.MapFrame(blockPos, n, n2);
+    public static net.minecraft.world.level.saveddata.maps.MapFrame load(net.minecraft.nbt.CompoundTag $$0) {
+        net.minecraft.core.BlockPos $$1 = net.minecraft.nbt.NbtUtils.readBlockPos((net.minecraft.nbt.CompoundTag)$$0.getCompound("Pos"));
+        int $$2 = $$0.getInt("Rotation");
+        int $$3 = $$0.getInt("EntityId");
+        return new net.minecraft.world.level.saveddata.maps.MapFrame($$1, $$2, $$3);
     }
 
     public net.minecraft.nbt.CompoundTag save() {
@@ -46,8 +46,8 @@
         return net.minecraft.world.level.saveddata.maps.MapFrame.frameId((net.minecraft.core.BlockPos)this.pos);
     }
 
-    public static java.lang.String frameId(net.minecraft.core.BlockPos pos) {
-        return "frame-" + pos.getX() + "," + pos.getY() + "," + pos.getZ();
+    public static java.lang.String frameId(net.minecraft.core.BlockPos $$0) {
+        return "frame-" + $$0.getX() + "," + $$0.getY() + "," + $$0.getZ();
     }
 }
 