--- net/minecraft/world/level/saveddata/maps/MapItemSavedData$HoldingPlayer.java
+++ net/minecraft/world/level/saveddata/maps/MapItemSavedData$HoldingPlayer.java
@@ -16,13 +16,13 @@
     public int step;
     public /* synthetic */ net.minecraft.world.level.saveddata.maps.MapItemSavedData this$0;
 
-    public net.minecraft.world.level.saveddata.maps.MapItemSavedData$HoldingPlayer(net.minecraft.world.level.saveddata.maps.MapItemSavedData mapItemSavedData, net.minecraft.world.entity.player.Player player) {
-        this.this$0 = mapItemSavedData;
+    public net.minecraft.world.level.saveddata.maps.MapItemSavedData$HoldingPlayer(net.minecraft.world.level.saveddata.maps.MapItemSavedData $$0, net.minecraft.world.entity.player.Player $$1) {
+        this.this$0 = $$0;
         this.dirtyData = true;
         this.maxDirtyX = 127;
         this.maxDirtyY = 127;
         this.dirtyDecorations = true;
-        this.player = player;
+        this.player = $$1;
     }
 
     public net.minecraft.world.level.saveddata.maps.MapItemSavedData$MapPatch createPatch() {