--- net/minecraft/world/level/saveddata/maps/MapIndex.java
+++ net/minecraft/world/level/saveddata/maps/MapIndex.java
@@ -23,11 +23,11 @@
         return $$1;
     }
 
-    public net.minecraft.nbt.CompoundTag save(net.minecraft.nbt.CompoundTag compound) {
-        for (it.unimi.dsi.fastutil.objects.Object2IntMap$Entry entry : this.usedAuxIds.object2IntEntrySet()) {
-            compound.putInt((java.lang.String)entry.getKey(), entry.getIntValue());
+    public net.minecraft.nbt.CompoundTag save(net.minecraft.nbt.CompoundTag $$0) {
+        for (it.unimi.dsi.fastutil.objects.Object2IntMap$Entry $$1 : this.usedAuxIds.object2IntEntrySet()) {
+            $$0.putInt((java.lang.String)$$1.getKey(), $$1.getIntValue());
         }
-        return compound;
+        return $$0;
     }
 
     public int getFreeAuxValueForMap() {