--- net/minecraft/advancements/Advancement.java
+++ net/minecraft/advancements/Advancement.java
@@ -16,25 +16,25 @@
     public java.util.Set<net.minecraft.advancements.Advancement> children;
     public net.minecraft.network.chat.Component chatComponent;
 
-    public net.minecraft.advancements.Advancement(net.minecraft.resources.ResourceLocation id, @javax.annotation.Nullable net.minecraft.advancements.Advancement parent, @javax.annotation.Nullable net.minecraft.advancements.DisplayInfo display, net.minecraft.advancements.AdvancementRewards rewards, java.util.Map<java.lang.String, net.minecraft.advancements.Criterion> criteria, java.lang.String[][] requirements) {
+    public net.minecraft.advancements.Advancement(net.minecraft.resources.ResourceLocation $$0, @javax.annotation.Nullable net.minecraft.advancements.Advancement $$1, @javax.annotation.Nullable net.minecraft.advancements.DisplayInfo $$2, net.minecraft.advancements.AdvancementRewards $$3, java.util.Map<java.lang.String, net.minecraft.advancements.Criterion> $$4, java.lang.String[][] $$5) {
         this.children = com.google.common.collect.Sets.newLinkedHashSet();
-        this.id = id;
-        this.display = display;
-        this.criteria = com.google.common.collect.ImmutableMap.copyOf(criteria);
-        this.parent = parent;
-        this.rewards = rewards;
-        this.requirements = requirements;
-        if (parent != null) {
-            parent.addChild(this);
+        this.id = $$0;
+        this.display = $$2;
+        this.criteria = com.google.common.collect.ImmutableMap.copyOf($$4);
+        this.parent = $$1;
+        this.rewards = $$3;
+        this.requirements = $$5;
+        if ($$1 != null) {
+            $$1.addChild(this);
         }
-        if (display == null) {
-            this.chatComponent = net.minecraft.network.chat.Component.literal((java.lang.String)id.toString());
+        if ($$2 == null) {
+            this.chatComponent = net.minecraft.network.chat.Component.literal((java.lang.String)$$0.toString());
         } else {
-            net.minecraft.network.chat.Component component = display.getTitle();
-            net.minecraft.ChatFormatting chatFormatting = display.getFrame().getChatColor();
-            net.minecraft.network.chat.MutableComponent mutableComponent = net.minecraft.network.chat.ComponentUtils.mergeStyles((net.minecraft.network.chat.MutableComponent)component.copy(), (net.minecraft.network.chat.Style)net.minecraft.network.chat.Style.EMPTY.withColor(chatFormatting)).append("\n").append(display.getDescription());
-            net.minecraft.network.chat.MutableComponent mutableComponent2 = component.copy().withStyle(arg_0 -> net.minecraft.advancements.Advancement.lambda$new$0((net.minecraft.network.chat.Component)mutableComponent, arg_0));
-            this.chatComponent = net.minecraft.network.chat.ComponentUtils.wrapInSquareBrackets((net.minecraft.network.chat.Component)mutableComponent2).withStyle(chatFormatting);
+            net.minecraft.network.chat.Component $$6 = $$2.getTitle();
+            net.minecraft.ChatFormatting $$7 = $$2.getFrame().getChatColor();
+            net.minecraft.network.chat.MutableComponent $$8 = net.minecraft.network.chat.ComponentUtils.mergeStyles((net.minecraft.network.chat.MutableComponent)$$6.copy(), (net.minecraft.network.chat.Style)net.minecraft.network.chat.Style.EMPTY.withColor($$7)).append("\n").append($$2.getDescription());
+            net.minecraft.network.chat.MutableComponent $$9 = $$6.copy().withStyle(arg_0 -> net.minecraft.advancements.Advancement.lambda$new$0((net.minecraft.network.chat.Component)$$8, arg_0));
+            this.chatComponent = net.minecraft.network.chat.ComponentUtils.wrapInSquareBrackets((net.minecraft.network.chat.Component)$$9).withStyle($$7);
         }
     }
 
@@ -72,23 +72,23 @@
         return this.requirements.length;
     }
 
-    public void addChild(net.minecraft.advancements.Advancement advancement) {
-        this.children.add((java.lang.Object)advancement);
+    public void addChild(net.minecraft.advancements.Advancement $$0) {
+        this.children.add((java.lang.Object)$$0);
     }
 
     public net.minecraft.resources.ResourceLocation getId() {
         return this.id;
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (!(other instanceof net.minecraft.advancements.Advancement)) {
+        if (!($$0 instanceof net.minecraft.advancements.Advancement)) {
             return false;
         }
-        net.minecraft.advancements.Advancement advancement = (net.minecraft.advancements.Advancement)other;
-        return this.id.equals((java.lang.Object)advancement.id);
+        net.minecraft.advancements.Advancement $$1 = (net.minecraft.advancements.Advancement)$$0;
+        return this.id.equals((java.lang.Object)$$1.id);
     }
 
     public int hashCode() {
@@ -103,8 +103,8 @@
         return this.chatComponent;
     }
 
-    public static /* synthetic */ net.minecraft.network.chat.Style lambda$new$0(net.minecraft.network.chat.Component component, net.minecraft.network.chat.Style style) {
-        return style.withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)component));
+    public static /* synthetic */ net.minecraft.network.chat.Style lambda$new$0(net.minecraft.network.chat.Component $$0, net.minecraft.network.chat.Style $$1) {
+        return $$1.withHoverEvent(new net.minecraft.network.chat.HoverEvent(net.minecraft.network.chat.HoverEvent$Action.SHOW_TEXT, (java.lang.Object)$$0));
     }
 }
 