--- net/minecraft/advancements/CriterionTrigger$Listener.java
+++ net/minecraft/advancements/CriterionTrigger$Listener.java
@@ -9,35 +9,35 @@
     public net.minecraft.advancements.Advancement advancement;
     public java.lang.String criterion;
 
-    public net.minecraft.advancements.CriterionTrigger$Listener(T trigger, net.minecraft.advancements.Advancement advancement, java.lang.String criterion) {
-        this.trigger = trigger;
-        this.advancement = advancement;
-        this.criterion = criterion;
+    public net.minecraft.advancements.CriterionTrigger$Listener(T $$0, net.minecraft.advancements.Advancement $$1, java.lang.String $$2) {
+        this.trigger = $$0;
+        this.advancement = $$1;
+        this.criterion = $$2;
     }
 
     public T getTriggerInstance() {
         return (T)this.trigger;
     }
 
-    public void run(net.minecraft.server.PlayerAdvancements playerAdvancements) {
-        playerAdvancements.award(this.advancement, this.criterion);
+    public void run(net.minecraft.server.PlayerAdvancements $$0) {
+        $$0.award(this.advancement, this.criterion);
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (other == null || this.getClass() != other.getClass()) {
+        if ($$0 == null || this.getClass() != $$0.getClass()) {
             return false;
         }
-        net.minecraft.advancements.CriterionTrigger$Listener listener = (net.minecraft.advancements.CriterionTrigger$Listener)other;
-        if (!this.trigger.equals((java.lang.Object)listener.trigger)) {
+        net.minecraft.advancements.CriterionTrigger$Listener $$1 = (net.minecraft.advancements.CriterionTrigger$Listener)$$0;
+        if (!this.trigger.equals((java.lang.Object)$$1.trigger)) {
             return false;
         }
-        if (!this.advancement.equals((java.lang.Object)listener.advancement)) {
+        if (!this.advancement.equals((java.lang.Object)$$1.advancement)) {
             return false;
         }
-        return this.criterion.equals((java.lang.Object)listener.criterion);
+        return this.criterion.equals((java.lang.Object)$$1.criterion);
     }
 
     public int hashCode() {