--- net/minecraft/advancements/FrameType.java
+++ net/minecraft/advancements/FrameType.java
@@ -23,12 +23,12 @@
         return (net.minecraft.advancements.FrameType)java.lang.Enum.valueOf(net.minecraft.advancements.FrameType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.advancements.FrameType(java.lang.String name, int texture, net.minecraft.ChatFormatting chatColor) {
+    public net.minecraft.advancements.FrameType(java.lang.String $$0, int $$1, net.minecraft.ChatFormatting $$2) {
         super(string, n);
-        this.name = name;
-        this.texture = texture;
-        this.chatColor = chatColor;
-        this.displayName = net.minecraft.network.chat.Component.translatable((java.lang.String)("advancements.toast." + name));
+        this.name = $$0;
+        this.texture = $$1;
+        this.chatColor = $$2;
+        this.displayName = net.minecraft.network.chat.Component.translatable((java.lang.String)("advancements.toast." + $$0));
     }
 
     public java.lang.String getName() {
@@ -39,12 +39,12 @@
         return this.texture;
     }
 
-    public static net.minecraft.advancements.FrameType byName(java.lang.String name) {
-        for (net.minecraft.advancements.FrameType frameType : net.minecraft.advancements.FrameType.values()) {
-            if (!frameType.name.equals((java.lang.Object)name)) continue;
-            return frameType;
+    public static net.minecraft.advancements.FrameType byName(java.lang.String $$0) {
+        for (net.minecraft.advancements.FrameType $$1 : net.minecraft.advancements.FrameType.values()) {
+            if (!$$1.name.equals((java.lang.Object)$$0)) continue;
+            return $$1;
         }
-        throw new java.lang.IllegalArgumentException("Unknown frame type '" + name + "'");
+        throw new java.lang.IllegalArgumentException("Unknown frame type '" + $$0 + "'");
     }
 
     public net.minecraft.ChatFormatting getChatColor() {