--- net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity$1.java
+++ net/minecraft/world/level/block/entity/AbstractFurnaceBlockEntity$1.java
@@ -12,8 +12,8 @@
         this.this$0 = $$0;
     }
 
-    public int get(int index) {
-        switch (index) {
+    public int get(int $$0) {
+        switch ($$0) {
             case 0: {
                 return this.this$0.litTime;
             }
@@ -30,22 +30,22 @@
         return 0;
     }
 
-    public void set(int index, int value) {
-        switch (index) {
+    public void set(int $$0, int $$1) {
+        switch ($$0) {
             case 0: {
-                this.this$0.litTime = value;
+                this.this$0.litTime = $$1;
                 break;
             }
             case 1: {
-                this.this$0.litDuration = value;
+                this.this$0.litDuration = $$1;
                 break;
             }
             case 2: {
-                this.this$0.cookingProgress = value;
+                this.this$0.cookingProgress = $$1;
                 break;
             }
             case 3: {
-                this.this$0.cookingTotalTime = value;
+                this.this$0.cookingTotalTime = $$1;
                 break;
             }
         }