--- net/minecraft/world/level/block/entity/BannerPattern.java
+++ net/minecraft/world/level/block/entity/BannerPattern.java
@@ -13,8 +13,7 @@
 
     public static net.minecraft.resources.ResourceLocation location(net.minecraft.resources.ResourceKey<net.minecraft.world.level.block.entity.BannerPattern> $$0, boolean $$1) {
         java.lang.String $$2 = $$1 ? "banner" : "shield";
-        net.minecraft.resources.ResourceLocation $$3 = $$0.location();
-        return new net.minecraft.resources.ResourceLocation($$3.getNamespace(), "entity/" + $$2 + "/" + $$3.getPath());
+        return $$0.location().withPrefix("entity/" + $$2 + "/");
     }
 
     public java.lang.String getHashname() {