--- net/minecraft/world/level/block/entity/BannerBlockEntity.java
+++ net/minecraft/world/level/block/entity/BannerBlockEntity.java
@@ -30,18 +30,18 @@
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.nbt.ListTag getItemPatterns(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.ListTag listTag = null;
-        net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack);
-        if (compoundTag != null && compoundTag.contains("Patterns", 9)) {
-            listTag = compoundTag.getList("Patterns", 10).copy();
+    public static net.minecraft.nbt.ListTag getItemPatterns(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.ListTag $$1 = null;
+        net.minecraft.nbt.CompoundTag $$2 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$0);
+        if ($$2 != null && $$2.contains("Patterns", 9)) {
+            $$1 = $$2.getList("Patterns", 10).copy();
         }
-        return listTag;
+        return $$1;
     }
 
-    public void fromItem(net.minecraft.world.item.ItemStack stack, net.minecraft.world.item.DyeColor color) {
-        this.baseColor = color;
-        this.fromItem(stack);
+    public void fromItem(net.minecraft.world.item.ItemStack $$0, net.minecraft.world.item.DyeColor $$1) {
+        this.baseColor = $$1;
+        this.fromItem($$0);
     }
 
     public void fromItem(net.minecraft.world.item.ItemStack $$0) {
@@ -62,8 +62,8 @@
         return this.name;
     }
 
-    public void setCustomName(net.minecraft.network.chat.Component name) {
-        this.name = name;
+    public void setCustomName(net.minecraft.network.chat.Component $$0) {
+        this.name = $$0;
     }
 
     public void saveAdditional(net.minecraft.nbt.CompoundTag $$0) {
@@ -93,10 +93,10 @@
         return this.saveWithoutMetadata();
     }
 
-    public static int getPatternCount(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack);
-        if (compoundTag != null && compoundTag.contains("Patterns")) {
-            return compoundTag.getList("Patterns", 10).size();
+    public static int getPatternCount(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$0);
+        if ($$1 != null && $$1.contains("Patterns")) {
+            return $$1.getList("Patterns", 10).size();
         }
         return 0;
     }
@@ -108,35 +108,35 @@
         return this.patterns;
     }
 
-    public static java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.core.Holder<net.minecraft.world.level.block.entity.BannerPattern>, net.minecraft.world.item.DyeColor>> createPatterns(net.minecraft.world.item.DyeColor color, @javax.annotation.Nullable net.minecraft.nbt.ListTag listTag) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        arrayList.add((java.lang.Object)com.mojang.datafixers.util.Pair.of((java.lang.Object)net.minecraft.core.Registry.BANNER_PATTERN.getHolderOrThrow(net.minecraft.world.level.block.entity.BannerPatterns.BASE), (java.lang.Object)color));
-        if (listTag != null) {
-            for (int i = 0; i < listTag.size(); ++i) {
-                net.minecraft.nbt.CompoundTag compoundTag = listTag.getCompound(i);
-                net.minecraft.core.Holder holder = net.minecraft.world.level.block.entity.BannerPattern.byHash((java.lang.String)compoundTag.getString("Pattern"));
-                if (holder == null) continue;
-                int n = compoundTag.getInt("Color");
-                arrayList.add((java.lang.Object)com.mojang.datafixers.util.Pair.of((java.lang.Object)holder, (java.lang.Object)net.minecraft.world.item.DyeColor.byId((int)n)));
+    public static java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.core.Holder<net.minecraft.world.level.block.entity.BannerPattern>, net.minecraft.world.item.DyeColor>> createPatterns(net.minecraft.world.item.DyeColor $$0, @javax.annotation.Nullable net.minecraft.nbt.ListTag $$1) {
+        java.util.ArrayList $$2 = com.google.common.collect.Lists.newArrayList();
+        $$2.add((java.lang.Object)com.mojang.datafixers.util.Pair.of((java.lang.Object)net.minecraft.core.Registry.BANNER_PATTERN.getHolderOrThrow(net.minecraft.world.level.block.entity.BannerPatterns.BASE), (java.lang.Object)$$0));
+        if ($$1 != null) {
+            for (int $$3 = 0; $$3 < $$1.size(); ++$$3) {
+                net.minecraft.nbt.CompoundTag $$4 = $$1.getCompound($$3);
+                net.minecraft.core.Holder $$5 = net.minecraft.world.level.block.entity.BannerPattern.byHash((java.lang.String)$$4.getString("Pattern"));
+                if ($$5 == null) continue;
+                int $$6 = $$4.getInt("Color");
+                $$2.add((java.lang.Object)com.mojang.datafixers.util.Pair.of((java.lang.Object)$$5, (java.lang.Object)net.minecraft.world.item.DyeColor.byId((int)$$6)));
             }
         }
-        return arrayList;
+        return $$2;
     }
 
-    public static void removeLastPattern(net.minecraft.world.item.ItemStack stack) {
-        net.minecraft.nbt.CompoundTag compoundTag = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)stack);
-        if (compoundTag == null || !compoundTag.contains("Patterns", 9)) {
+    public static void removeLastPattern(net.minecraft.world.item.ItemStack $$0) {
+        net.minecraft.nbt.CompoundTag $$1 = net.minecraft.world.item.BlockItem.getBlockEntityData((net.minecraft.world.item.ItemStack)$$0);
+        if ($$1 == null || !$$1.contains("Patterns", 9)) {
             return;
         }
-        net.minecraft.nbt.ListTag listTag = compoundTag.getList("Patterns", 10);
-        if (listTag.isEmpty()) {
+        net.minecraft.nbt.ListTag $$2 = $$1.getList("Patterns", 10);
+        if ($$2.isEmpty()) {
             return;
         }
-        listTag.remove(listTag.size() - 1);
-        if (listTag.isEmpty()) {
-            compoundTag.remove("Patterns");
+        $$2.remove($$2.size() - 1);
+        if ($$2.isEmpty()) {
+            $$1.remove("Patterns");
         }
-        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)stack, (net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.BANNER, (net.minecraft.nbt.CompoundTag)compoundTag);
+        net.minecraft.world.item.BlockItem.setBlockEntityData((net.minecraft.world.item.ItemStack)$$0, (net.minecraft.world.level.block.entity.BlockEntityType)net.minecraft.world.level.block.entity.BlockEntityType.BANNER, (net.minecraft.nbt.CompoundTag)$$1);
     }
 
     public net.minecraft.world.item.ItemStack getItem() {