--- net/minecraft/world/level/block/entity/BeaconBlockEntity$BeaconBeamSection.java
+++ net/minecraft/world/level/block/entity/BeaconBlockEntity$BeaconBeamSection.java
@@ -8,8 +8,8 @@
     public float[] color;
     public int height;
 
-    public net.minecraft.world.level.block.entity.BeaconBlockEntity$BeaconBeamSection(float[] color) {
-        this.color = color;
+    public net.minecraft.world.level.block.entity.BeaconBlockEntity$BeaconBeamSection(float[] $$0) {
+        this.color = $$0;
         this.height = 1;
     }
 