--- net/minecraft/world/level/block/entity/BedBlockEntity.java
+++ net/minecraft/world/level/block/entity/BedBlockEntity.java
@@ -26,8 +26,8 @@
         return this.color;
     }
 
-    public void setColor(net.minecraft.world.item.DyeColor color) {
-        this.color = color;
+    public void setColor(net.minecraft.world.item.DyeColor $$0) {
+        this.color = $$0;
     }
 
     public /* synthetic */ net.minecraft.network.protocol.Packet getUpdatePacket() {