--- net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+++ net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
@@ -52,29 +52,29 @@
         return this.stored.size() == 3;
     }
 
-    public void emptyAllLivingFromHive(@javax.annotation.Nullable net.minecraft.world.entity.player.Player player, net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.entity.BeehiveBlockEntity$BeeReleaseStatus releaseStatus) {
-        java.util.List list = this.releaseAllOccupants(state, releaseStatus);
-        if (player != null) {
-            for (net.minecraft.world.entity.Entity entity : list) {
-                if (!(entity instanceof net.minecraft.world.entity.animal.Bee)) continue;
-                net.minecraft.world.entity.animal.Bee bee = (net.minecraft.world.entity.animal.Bee)entity;
-                if (!(player.position().distanceToSqr(entity.position()) <= 16.0)) continue;
+    public void emptyAllLivingFromHive(@javax.annotation.Nullable net.minecraft.world.entity.player.Player $$0, net.minecraft.world.level.block.state.BlockState $$1, net.minecraft.world.level.block.entity.BeehiveBlockEntity$BeeReleaseStatus $$2) {
+        java.util.List $$3 = this.releaseAllOccupants($$1, $$2);
+        if ($$0 != null) {
+            for (net.minecraft.world.entity.Entity $$4 : $$3) {
+                if (!($$4 instanceof net.minecraft.world.entity.animal.Bee)) continue;
+                net.minecraft.world.entity.animal.Bee $$5 = (net.minecraft.world.entity.animal.Bee)$$4;
+                if (!($$0.position().distanceToSqr($$4.position()) <= 16.0)) continue;
                 if (!this.isSedated()) {
-                    bee.setTarget((net.minecraft.world.entity.LivingEntity)player);
+                    $$5.setTarget((net.minecraft.world.entity.LivingEntity)$$0);
                     continue;
                 }
-                bee.setStayOutOfHiveCountdown(400);
+                $$5.setStayOutOfHiveCountdown(400);
             }
         }
     }
 
-    public java.util.List<net.minecraft.world.entity.Entity> releaseAllOccupants(net.minecraft.world.level.block.state.BlockState state, net.minecraft.world.level.block.entity.BeehiveBlockEntity$BeeReleaseStatus releaseStatus) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        this.stored.removeIf(arg_0 -> this.lambda$releaseAllOccupants$0(state, (java.util.List)arrayList, releaseStatus, arg_0));
-        if (!arrayList.isEmpty()) {
+    public java.util.List<net.minecraft.world.entity.Entity> releaseAllOccupants(net.minecraft.world.level.block.state.BlockState $$0, net.minecraft.world.level.block.entity.BeehiveBlockEntity$BeeReleaseStatus $$1) {
+        java.util.ArrayList $$2 = com.google.common.collect.Lists.newArrayList();
+        this.stored.removeIf(arg_0 -> this.lambda$releaseAllOccupants$0($$0, (java.util.List)$$2, $$1, arg_0));
+        if (!$$2.isEmpty()) {
             super.setChanged();
         }
-        return arrayList;
+        return $$2;
     }
 
     public void addOccupant(net.minecraft.world.entity.Entity $$0, boolean $$1) {
@@ -86,8 +86,8 @@
         return this.stored.size();
     }
 
-    public static int getHoneyLevel(net.minecraft.world.level.block.state.BlockState state) {
-        return (java.lang.Integer)state.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BeehiveBlock.HONEY_LEVEL);
+    public static int getHoneyLevel(net.minecraft.world.level.block.state.BlockState $$0) {
+        return (java.lang.Integer)$$0.getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.BeehiveBlock.HONEY_LEVEL);
     }
 
     @net.minecraft.util.VisibleForDebug