--- net/minecraft/world/level/block/entity/BlockEntity.java
+++ net/minecraft/world/level/block/entity/BlockEntity.java
@@ -156,17 +156,17 @@
         this.remove = false;
     }
 
-    public boolean triggerEvent(int id, int type) {
+    public boolean triggerEvent(int $$0, int $$1) {
         return false;
     }
 
-    public void fillCrashReportCategory(net.minecraft.CrashReportCategory reportCategory) {
-        reportCategory.setDetail("Name", () -> net.minecraft.core.Registry.BLOCK_ENTITY_TYPE.getKey((java.lang.Object)this.getType()) + " // " + this.getClass().getCanonicalName());
+    public void fillCrashReportCategory(net.minecraft.CrashReportCategory $$0) {
+        $$0.setDetail("Name", () -> net.minecraft.core.Registry.BLOCK_ENTITY_TYPE.getKey((java.lang.Object)this.getType()) + " // " + this.getClass().getCanonicalName());
         if (this.level == null) {
             return;
         }
-        net.minecraft.CrashReportCategory.populateBlockDetails((net.minecraft.CrashReportCategory)reportCategory, (net.minecraft.world.level.LevelHeightAccessor)this.level, (net.minecraft.core.BlockPos)this.worldPosition, (net.minecraft.world.level.block.state.BlockState)this.getBlockState());
-        net.minecraft.CrashReportCategory.populateBlockDetails((net.minecraft.CrashReportCategory)reportCategory, (net.minecraft.world.level.LevelHeightAccessor)this.level, (net.minecraft.core.BlockPos)this.worldPosition, (net.minecraft.world.level.block.state.BlockState)this.level.getBlockState(this.worldPosition));
+        net.minecraft.CrashReportCategory.populateBlockDetails((net.minecraft.CrashReportCategory)$$0, (net.minecraft.world.level.LevelHeightAccessor)this.level, (net.minecraft.core.BlockPos)this.worldPosition, (net.minecraft.world.level.block.state.BlockState)this.getBlockState());
+        net.minecraft.CrashReportCategory.populateBlockDetails((net.minecraft.CrashReportCategory)$$0, (net.minecraft.world.level.LevelHeightAccessor)this.level, (net.minecraft.core.BlockPos)this.worldPosition, (net.minecraft.world.level.block.state.BlockState)this.level.getBlockState(this.worldPosition));
     }
 
     public boolean onlyOpCanSetNbt() {