--- net/minecraft/world/level/block/entity/BlockEntityType$Builder.java
+++ net/minecraft/world/level/block/entity/BlockEntityType$Builder.java
@@ -17,8 +17,8 @@
         return new net.minecraft.world.level.block.entity.BlockEntityType$Builder($$0, (java.util.Set)com.google.common.collect.ImmutableSet.copyOf((java.lang.Object[])$$1));
     }
 
-    public net.minecraft.world.level.block.entity.BlockEntityType<T> build(com.mojang.datafixers.types.Type<?> dataType) {
-        return new net.minecraft.world.level.block.entity.BlockEntityType(this.factory, this.validBlocks, dataType);
+    public net.minecraft.world.level.block.entity.BlockEntityType<T> build(com.mojang.datafixers.types.Type<?> $$0) {
+        return new net.minecraft.world.level.block.entity.BlockEntityType(this.factory, this.validBlocks, $$0);
     }
 }
 