--- net/minecraft/world/level/block/entity/BrewingStandBlockEntity.java
+++ net/minecraft/world/level/block/entity/BrewingStandBlockEntity.java
@@ -144,61 +144,61 @@
         $$0.putByte("Fuel", (byte)this.fuel);
     }
 
-    public net.minecraft.world.item.ItemStack getItem(int index) {
-        if (index >= 0 && index < this.items.size()) {
-            return (net.minecraft.world.item.ItemStack)this.items.get(index);
+    public net.minecraft.world.item.ItemStack getItem(int $$0) {
+        if ($$0 >= 0 && $$0 < this.items.size()) {
+            return (net.minecraft.world.item.ItemStack)this.items.get($$0);
         }
         return net.minecraft.world.item.ItemStack.EMPTY;
     }
 
-    public net.minecraft.world.item.ItemStack removeItem(int index, int count) {
-        return net.minecraft.world.ContainerHelper.removeItem((java.util.List)this.items, (int)index, (int)count);
+    public net.minecraft.world.item.ItemStack removeItem(int $$0, int $$1) {
+        return net.minecraft.world.ContainerHelper.removeItem((java.util.List)this.items, (int)$$0, (int)$$1);
     }
 
-    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int index) {
-        return net.minecraft.world.ContainerHelper.takeItem((java.util.List)this.items, (int)index);
+    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int $$0) {
+        return net.minecraft.world.ContainerHelper.takeItem((java.util.List)this.items, (int)$$0);
     }
 
-    public void setItem(int index, net.minecraft.world.item.ItemStack stack) {
-        if (index >= 0 && index < this.items.size()) {
-            this.items.set(index, (java.lang.Object)stack);
+    public void setItem(int $$0, net.minecraft.world.item.ItemStack $$1) {
+        if ($$0 >= 0 && $$0 < this.items.size()) {
+            this.items.set($$0, (java.lang.Object)$$1);
         }
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
         if (this.level.getBlockEntity(this.worldPosition) != this) {
             return false;
         }
-        return !(player.distanceToSqr((double)this.worldPosition.getX() + 0.5, (double)this.worldPosition.getY() + 0.5, (double)this.worldPosition.getZ() + 0.5) > 64.0);
+        return !($$0.distanceToSqr((double)this.worldPosition.getX() + 0.5, (double)this.worldPosition.getY() + 0.5, (double)this.worldPosition.getZ() + 0.5) > 64.0);
     }
 
-    public boolean canPlaceItem(int index, net.minecraft.world.item.ItemStack stack) {
-        if (index == 3) {
-            return net.minecraft.world.item.alchemy.PotionBrewing.isIngredient((net.minecraft.world.item.ItemStack)stack);
+    public boolean canPlaceItem(int $$0, net.minecraft.world.item.ItemStack $$1) {
+        if ($$0 == 3) {
+            return net.minecraft.world.item.alchemy.PotionBrewing.isIngredient((net.minecraft.world.item.ItemStack)$$1);
         }
-        if (index == 4) {
-            return stack.is(net.minecraft.world.item.Items.BLAZE_POWDER);
+        if ($$0 == 4) {
+            return $$1.is(net.minecraft.world.item.Items.BLAZE_POWDER);
         }
-        return (stack.is(net.minecraft.world.item.Items.POTION) || stack.is(net.minecraft.world.item.Items.SPLASH_POTION) || stack.is(net.minecraft.world.item.Items.LINGERING_POTION) || stack.is(net.minecraft.world.item.Items.GLASS_BOTTLE)) && this.getItem(index).isEmpty();
+        return ($$1.is(net.minecraft.world.item.Items.POTION) || $$1.is(net.minecraft.world.item.Items.SPLASH_POTION) || $$1.is(net.minecraft.world.item.Items.LINGERING_POTION) || $$1.is(net.minecraft.world.item.Items.GLASS_BOTTLE)) && this.getItem($$0).isEmpty();
     }
 
-    public int[] getSlotsForFace(net.minecraft.core.Direction side) {
-        if (side == net.minecraft.core.Direction.UP) {
+    public int[] getSlotsForFace(net.minecraft.core.Direction $$0) {
+        if ($$0 == net.minecraft.core.Direction.UP) {
             return SLOTS_FOR_UP;
         }
-        if (side == net.minecraft.core.Direction.DOWN) {
+        if ($$0 == net.minecraft.core.Direction.DOWN) {
             return SLOTS_FOR_DOWN;
         }
         return SLOTS_FOR_SIDES;
     }
 
-    public boolean canPlaceItemThroughFace(int index, net.minecraft.world.item.ItemStack itemStack, @javax.annotation.Nullable net.minecraft.core.Direction direction) {
-        return this.canPlaceItem(index, itemStack);
+    public boolean canPlaceItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, @javax.annotation.Nullable net.minecraft.core.Direction $$2) {
+        return this.canPlaceItem($$0, $$1);
     }
 
-    public boolean canTakeItemThroughFace(int index, net.minecraft.world.item.ItemStack stack, net.minecraft.core.Direction direction) {
-        if (index == 3) {
-            return stack.is(net.minecraft.world.item.Items.GLASS_BOTTLE);
+    public boolean canTakeItemThroughFace(int $$0, net.minecraft.world.item.ItemStack $$1, net.minecraft.core.Direction $$2) {
+        if ($$0 == 3) {
+            return $$1.is(net.minecraft.world.item.Items.GLASS_BOTTLE);
         }
         return true;
     }
@@ -207,8 +207,8 @@
         this.items.clear();
     }
 
-    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.BrewingStandMenu(id, player, (net.minecraft.world.Container)this, this.dataAccess);
+    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.BrewingStandMenu($$0, $$1, (net.minecraft.world.Container)this, this.dataAccess);
     }
 
     static {