--- net/minecraft/world/level/block/entity/CommandBlockEntity.java
+++ net/minecraft/world/level/block/entity/CommandBlockEntity.java
@@ -40,8 +40,8 @@
         return this.commandBlock;
     }
 
-    public void setPowered(boolean powered) {
-        this.powered = powered;
+    public void setPowered(boolean $$0) {
+        this.powered = $$0;
     }
 
     public boolean isPowered() {
@@ -52,10 +52,10 @@
         return this.auto;
     }
 
-    public void setAutomatic(boolean auto) {
-        boolean bl = this.auto;
-        this.auto = auto;
-        if (!bl && auto && !this.powered && this.level != null && this.getMode() != net.minecraft.world.level.block.entity.CommandBlockEntity$Mode.SEQUENCE) {
+    public void setAutomatic(boolean $$0) {
+        boolean $$1 = this.auto;
+        this.auto = $$0;
+        if (!$$1 && $$0 && !this.powered && this.level != null && this.getMode() != net.minecraft.world.level.block.entity.CommandBlockEntity$Mode.SEQUENCE) {
             this.scheduleTick();
         }
     }