--- net/minecraft/world/level/block/entity/CommandBlockEntity$1.java
+++ net/minecraft/world/level/block/entity/CommandBlockEntity$1.java
@@ -12,8 +12,8 @@
         this.this$0 = $$0;
     }
 
-    public void setCommand(java.lang.String command) {
-        super.setCommand(command);
+    public void setCommand(java.lang.String $$0) {
+        super.setCommand($$0);
         this.this$0.setChanged();
     }
 