--- net/minecraft/world/level/block/entity/DispenserBlockEntity.java
+++ net/minecraft/world/level/block/entity/DispenserBlockEntity.java
@@ -33,11 +33,11 @@
         return $$1;
     }
 
-    public int addItem(net.minecraft.world.item.ItemStack stack) {
-        for (int i = 0; i < this.items.size(); ++i) {
-            if (!((net.minecraft.world.item.ItemStack)this.items.get(i)).isEmpty()) continue;
-            this.setItem(i, stack);
-            return i;
+    public int addItem(net.minecraft.world.item.ItemStack $$0) {
+        for (int $$1 = 0; $$1 < this.items.size(); ++$$1) {
+            if (!((net.minecraft.world.item.ItemStack)this.items.get($$1)).isEmpty()) continue;
+            this.setItem($$1, $$0);
+            return $$1;
         }
         return -1;
     }
@@ -65,12 +65,12 @@
         return this.items;
     }
 
-    public void setItems(net.minecraft.core.NonNullList<net.minecraft.world.item.ItemStack> items) {
-        this.items = items;
+    public void setItems(net.minecraft.core.NonNullList<net.minecraft.world.item.ItemStack> $$0) {
+        this.items = $$0;
     }
 
-    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.DispenserMenu(id, player, (net.minecraft.world.Container)this);
+    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.DispenserMenu($$0, $$1, (net.minecraft.world.Container)this);
     }
 }
 