--- net/minecraft/world/level/block/entity/EnchantmentTableBlockEntity.java
+++ net/minecraft/world/level/block/entity/EnchantmentTableBlockEntity.java
@@ -94,8 +94,8 @@
         return net.minecraft.network.chat.Component.translatable((java.lang.String)"container.enchant");
     }
 
-    public void setCustomName(@javax.annotation.Nullable net.minecraft.network.chat.Component name) {
-        this.name = name;
+    public void setCustomName(@javax.annotation.Nullable net.minecraft.network.chat.Component $$0) {
+        this.name = $$0;
     }
 
     @javax.annotation.Nullable