--- net/minecraft/world/level/block/entity/FurnaceBlockEntity.java
+++ net/minecraft/world/level/block/entity/FurnaceBlockEntity.java
@@ -14,8 +14,8 @@
         return net.minecraft.network.chat.Component.translatable((java.lang.String)"container.furnace");
     }
 
-    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int id, net.minecraft.world.entity.player.Inventory player) {
-        return new net.minecraft.world.inventory.FurnaceMenu(id, player, (net.minecraft.world.Container)this, this.dataAccess);
+    public net.minecraft.world.inventory.AbstractContainerMenu createMenu(int $$0, net.minecraft.world.entity.player.Inventory $$1) {
+        return new net.minecraft.world.inventory.FurnaceMenu($$0, $$1, (net.minecraft.world.Container)this, this.dataAccess);
     }
 }
 