--- net/minecraft/world/level/block/entity/HangingSignBlockEntity.java
+++ net/minecraft/world/level/block/entity/HangingSignBlockEntity.java
@@ -1,0 +1,24 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level.block.entity;
+
+public class net.minecraft.world.level.block.entity.HangingSignBlockEntity
+extends net.minecraft.world.level.block.entity.SignBlockEntity
+ {
+    public static int MAX_TEXT_LINE_WIDTH = 50;
+    public static int TEXT_LINE_HEIGHT = 8;
+
+    public net.minecraft.world.level.block.entity.HangingSignBlockEntity(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
+        super(net.minecraft.world.level.block.entity.BlockEntityType.HANGING_SIGN, $$0, $$1);
+    }
+
+    public int getTextLineHeight() {
+        return 8;
+    }
+
+    public int getMaxTextLineWidth() {
+        return 50;
+    }
+}
+