--- net/minecraft/world/level/block/entity/JigsawBlockEntity.java
+++ net/minecraft/world/level/block/entity/JigsawBlockEntity.java
@@ -46,24 +46,24 @@
         return this.joint;
     }
 
-    public void setName(net.minecraft.resources.ResourceLocation name) {
-        this.name = name;
+    public void setName(net.minecraft.resources.ResourceLocation $$0) {
+        this.name = $$0;
     }
 
-    public void setTarget(net.minecraft.resources.ResourceLocation target) {
-        this.target = target;
+    public void setTarget(net.minecraft.resources.ResourceLocation $$0) {
+        this.target = $$0;
     }
 
     public void setPool(net.minecraft.resources.ResourceKey<net.minecraft.world.level.levelgen.structure.pools.StructureTemplatePool> $$0) {
         this.pool = $$0;
     }
 
-    public void setFinalState(java.lang.String finalState) {
-        this.finalState = finalState;
+    public void setFinalState(java.lang.String $$0) {
+        this.finalState = $$0;
     }
 
-    public void setJoint(net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType joint) {
-        this.joint = joint;
+    public void setJoint(net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType $$0) {
+        this.joint = $$0;
     }
 
     public void saveAdditional(net.minecraft.nbt.CompoundTag $$0) {
@@ -92,11 +92,11 @@
         return this.saveWithoutMetadata();
     }
 
-    public void generate(net.minecraft.server.level.ServerLevel level, int n, boolean bl) {
-        net.minecraft.core.BlockPos blockPos = this.getBlockPos().relative(((net.minecraft.core.FrontAndTop)this.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.JigsawBlock.ORIENTATION)).front());
-        net.minecraft.core.Registry registry = level.registryAccess().registryOrThrow(net.minecraft.core.Registry.TEMPLATE_POOL_REGISTRY);
-        net.minecraft.core.Holder holder = registry.getHolderOrThrow(this.pool);
-        net.minecraft.world.level.levelgen.structure.pools.JigsawPlacement.generateJigsaw((net.minecraft.server.level.ServerLevel)level, (net.minecraft.core.Holder)holder, (net.minecraft.resources.ResourceLocation)this.target, (int)n, (net.minecraft.core.BlockPos)blockPos, (boolean)bl);
+    public void generate(net.minecraft.server.level.ServerLevel $$0, int $$1, boolean $$2) {
+        net.minecraft.core.BlockPos $$3 = this.getBlockPos().relative(((net.minecraft.core.FrontAndTop)this.getBlockState().getValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.JigsawBlock.ORIENTATION)).front());
+        net.minecraft.core.Registry $$4 = $$0.registryAccess().registryOrThrow(net.minecraft.core.Registry.TEMPLATE_POOL_REGISTRY);
+        net.minecraft.core.Holder$Reference $$5 = $$4.getHolderOrThrow(this.pool);
+        net.minecraft.world.level.levelgen.structure.pools.JigsawPlacement.generateJigsaw((net.minecraft.server.level.ServerLevel)$$0, (net.minecraft.core.Holder)$$5, (net.minecraft.resources.ResourceLocation)this.target, (int)$$1, (net.minecraft.core.BlockPos)$$3, (boolean)$$2);
     }
 
     public /* synthetic */ net.minecraft.network.protocol.Packet getUpdatePacket() {