--- net/minecraft/world/level/block/entity/JigsawBlockEntity$JointType.java
+++ net/minecraft/world/level/block/entity/JigsawBlockEntity$JointType.java
@@ -20,17 +20,17 @@
         return (net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType)java.lang.Enum.valueOf(net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType(java.lang.String name) {
+    public net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
+        this.name = $$0;
     }
 
     public java.lang.String getSerializedName() {
         return this.name;
     }
 
-    public static java.util.Optional<net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType> byName(java.lang.String name) {
-        return java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType.values()).filter(jointType -> jointType.getSerializedName().equals((java.lang.Object)name)).findFirst();
+    public static java.util.Optional<net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType> byName(java.lang.String $$0) {
+        return java.util.Arrays.stream((java.lang.Object[])net.minecraft.world.level.block.entity.JigsawBlockEntity$JointType.values()).filter($$1 -> $$1.getSerializedName().equals((java.lang.Object)$$0)).findFirst();
     }
 
     public net.minecraft.network.chat.Component getTranslatedName() {