--- net/minecraft/world/level/block/entity/JukeboxBlockEntity.java
+++ net/minecraft/world/level/block/entity/JukeboxBlockEntity.java
@@ -42,8 +42,8 @@
         return this.record;
     }
 
-    public void setRecord(net.minecraft.world.item.ItemStack record) {
-        this.record = record;
+    public void setRecord(net.minecraft.world.item.ItemStack $$0) {
+        this.record = $$0;
         this.setChanged();
     }
 