--- net/minecraft/world/level/block/entity/LecternBlockEntity$1.java
+++ net/minecraft/world/level/block/entity/LecternBlockEntity$1.java
@@ -20,32 +20,32 @@
         return this.this$0.book.isEmpty();
     }
 
-    public net.minecraft.world.item.ItemStack getItem(int index) {
-        return index == 0 ? this.this$0.book : net.minecraft.world.item.ItemStack.EMPTY;
+    public net.minecraft.world.item.ItemStack getItem(int $$0) {
+        return $$0 == 0 ? this.this$0.book : net.minecraft.world.item.ItemStack.EMPTY;
     }
 
-    public net.minecraft.world.item.ItemStack removeItem(int index, int count) {
-        if (index == 0) {
-            net.minecraft.world.item.ItemStack itemStack = this.this$0.book.split(count);
+    public net.minecraft.world.item.ItemStack removeItem(int $$0, int $$1) {
+        if ($$0 == 0) {
+            net.minecraft.world.item.ItemStack $$2 = this.this$0.book.split($$1);
             if (this.this$0.book.isEmpty()) {
                 this.this$0.onBookItemRemove();
             }
-            return itemStack;
+            return $$2;
         }
         return net.minecraft.world.item.ItemStack.EMPTY;
     }
 
-    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int index) {
-        if (index == 0) {
-            net.minecraft.world.item.ItemStack itemStack = this.this$0.book;
+    public net.minecraft.world.item.ItemStack removeItemNoUpdate(int $$0) {
+        if ($$0 == 0) {
+            net.minecraft.world.item.ItemStack $$1 = this.this$0.book;
             this.this$0.book = net.minecraft.world.item.ItemStack.EMPTY;
             this.this$0.onBookItemRemove();
-            return itemStack;
+            return $$1;
         }
         return net.minecraft.world.item.ItemStack.EMPTY;
     }
 
-    public void setItem(int index, net.minecraft.world.item.ItemStack stack) {
+    public void setItem(int $$0, net.minecraft.world.item.ItemStack $$1) {
     }
 
     public int getMaxStackSize() {
@@ -56,17 +56,17 @@
         this.this$0.setChanged();
     }
 
-    public boolean stillValid(net.minecraft.world.entity.player.Player player) {
+    public boolean stillValid(net.minecraft.world.entity.player.Player $$0) {
         if (this.this$0.level.getBlockEntity(this.this$0.worldPosition) != this.this$0) {
             return false;
         }
-        if (player.distanceToSqr((double)this.this$0.worldPosition.getX() + 0.5, (double)this.this$0.worldPosition.getY() + 0.5, (double)this.this$0.worldPosition.getZ() + 0.5) > 64.0) {
+        if ($$0.distanceToSqr((double)this.this$0.worldPosition.getX() + 0.5, (double)this.this$0.worldPosition.getY() + 0.5, (double)this.this$0.worldPosition.getZ() + 0.5) > 64.0) {
             return false;
         }
         return this.this$0.hasBook();
     }
 
-    public boolean canPlaceItem(int index, net.minecraft.world.item.ItemStack stack) {
+    public boolean canPlaceItem(int $$0, net.minecraft.world.item.ItemStack $$1) {
         return false;
     }
 