--- net/minecraft/world/level/block/entity/LecternBlockEntity$2.java
+++ net/minecraft/world/level/block/entity/LecternBlockEntity$2.java
@@ -12,13 +12,13 @@
         this.this$0 = $$0;
     }
 
-    public int get(int index) {
-        return index == 0 ? this.this$0.page : 0;
+    public int get(int $$0) {
+        return $$0 == 0 ? this.this$0.page : 0;
     }
 
-    public void set(int index, int value) {
-        if (index == 0) {
-            this.this$0.setPage(value);
+    public void set(int $$0, int $$1) {
+        if ($$0 == 0) {
+            this.this$0.setPage($$1);
         }
     }
 