--- net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java
+++ net/minecraft/world/level/block/entity/SculkSensorBlockEntity.java
@@ -13,7 +13,7 @@
 
     public net.minecraft.world.level.block.entity.SculkSensorBlockEntity(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1) {
         super(net.minecraft.world.level.block.entity.BlockEntityType.SCULK_SENSOR, $$0, $$1);
-        this.listener = new net.minecraft.world.level.gameevent.vibrations.VibrationListener((net.minecraft.world.level.gameevent.PositionSource)new net.minecraft.world.level.gameevent.BlockPositionSource(this.worldPosition), ((net.minecraft.world.level.block.SculkSensorBlock)$$1.getBlock()).getListenerRange(), (net.minecraft.world.level.gameevent.vibrations.VibrationListener$VibrationListenerConfig)this, null, 0.0f, 0);
+        this.listener = new net.minecraft.world.level.gameevent.vibrations.VibrationListener((net.minecraft.world.level.gameevent.PositionSource)new net.minecraft.world.level.gameevent.BlockPositionSource(this.worldPosition), ((net.minecraft.world.level.block.SculkSensorBlock)$$1.getBlock()).getListenerRange(), (net.minecraft.world.level.gameevent.vibrations.VibrationListener$VibrationListenerConfig)this);
     }
 
     public void load(net.minecraft.nbt.CompoundTag $$02) {
@@ -45,7 +45,7 @@
     }
 
     public boolean shouldListen(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.gameevent.GameEventListener $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.gameevent.GameEvent $$3, @javax.annotation.Nullable net.minecraft.world.level.gameevent.GameEvent$Context $$4) {
-        if (this.isRemoved() || $$2.equals((java.lang.Object)this.getBlockPos()) && ($$3 == net.minecraft.world.level.gameevent.GameEvent.BLOCK_DESTROY || $$3 == net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE)) {
+        if ($$2.equals((java.lang.Object)this.getBlockPos()) && ($$3 == net.minecraft.world.level.gameevent.GameEvent.BLOCK_DESTROY || $$3 == net.minecraft.world.level.gameevent.GameEvent.BLOCK_PLACE)) {
             return false;
         }
         return net.minecraft.world.level.block.SculkSensorBlock.canActivate((net.minecraft.world.level.block.state.BlockState)this.getBlockState());
@@ -54,7 +54,7 @@
     public void onSignalReceive(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.level.gameevent.GameEventListener $$1, net.minecraft.core.BlockPos $$2, net.minecraft.world.level.gameevent.GameEvent $$3, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$4, @javax.annotation.Nullable net.minecraft.world.entity.Entity $$5, float $$6) {
         net.minecraft.world.level.block.state.BlockState $$7 = this.getBlockState();
         if (net.minecraft.world.level.block.SculkSensorBlock.canActivate((net.minecraft.world.level.block.state.BlockState)$$7)) {
-            this.lastVibrationFrequency = net.minecraft.world.level.block.SculkSensorBlock.VIBRATION_FREQUENCY_FOR_EVENT.getInt((java.lang.Object)$$3);
+            this.lastVibrationFrequency = net.minecraft.world.level.gameevent.vibrations.VibrationListener.getGameEventFrequency((net.minecraft.world.level.gameevent.GameEvent)$$3);
             net.minecraft.world.level.block.SculkSensorBlock.activate((net.minecraft.world.entity.Entity)$$4, (net.minecraft.world.level.Level)$$0, (net.minecraft.core.BlockPos)this.worldPosition, (net.minecraft.world.level.block.state.BlockState)$$7, (int)net.minecraft.world.level.block.entity.SculkSensorBlockEntity.getRedstoneStrengthForDistance((float)$$6, (int)$$1.getListenerRadius()));
         }
     }