--- net/minecraft/world/level/block/entity/SkullBlockEntity.java
+++ net/minecraft/world/level/block/entity/SkullBlockEntity.java
@@ -62,9 +62,9 @@
         }
     }
 
-    public float getMouthAnimation(float partialTicks) {
+    public float getMouthAnimation(float $$0) {
         if (this.isMovingMouth) {
-            return (float)this.mouthTickCount + partialTicks;
+            return (float)this.mouthTickCount + $$0;
         }
         return this.mouthTickCount;
     }
@@ -85,10 +85,10 @@
     /*
      * WARNING - Removed try catching itself - possible behaviour change.
      */
-    public void setOwner(@javax.annotation.Nullable com.mojang.authlib.GameProfile owner) {
+    public void setOwner(@javax.annotation.Nullable com.mojang.authlib.GameProfile $$0) {
         net.minecraft.world.level.block.entity.SkullBlockEntity skullBlockEntity = this;
         synchronized (skullBlockEntity) {
-            this.owner = owner;
+            this.owner = $$0;
         }
         this.updateOwnerProfile();
     }