--- net/minecraft/world/level/block/entity/SpawnerBlockEntity.java
+++ net/minecraft/world/level/block/entity/SpawnerBlockEntity.java
@@ -41,11 +41,11 @@
         return $$0;
     }
 
-    public boolean triggerEvent(int id, int type) {
-        if (this.spawner.onEventTriggered(this.level, id)) {
+    public boolean triggerEvent(int $$0, int $$1) {
+        if (this.spawner.onEventTriggered(this.level, $$0)) {
             return true;
         }
-        return super.triggerEvent(id, type);
+        return super.triggerEvent($$0, $$1);
     }
 
     public boolean onlyOpCanSetNbt() {