--- net/minecraft/world/level/block/entity/StructureBlockEntity.java
+++ net/minecraft/world/level/block/entity/StructureBlockEntity.java
@@ -120,12 +120,12 @@
         return this.saveWithoutMetadata();
     }
 
-    public boolean usedBy(net.minecraft.world.entity.player.Player player) {
-        if (!player.canUseGameMasterBlocks()) {
+    public boolean usedBy(net.minecraft.world.entity.player.Player $$0) {
+        if (!$$0.canUseGameMasterBlocks()) {
             return false;
         }
-        if (player.getCommandSenderWorld().isClientSide) {
-            player.openStructureBlock(this);
+        if ($$0.getCommandSenderWorld().isClientSide) {
+            $$0.openStructureBlock(this);
         }
         return true;
     }
@@ -142,24 +142,24 @@
         return this.structureName != null;
     }
 
-    public void setStructureName(@javax.annotation.Nullable java.lang.String structureName) {
-        this.setStructureName(net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)structureName) ? null : net.minecraft.resources.ResourceLocation.tryParse((java.lang.String)structureName));
+    public void setStructureName(@javax.annotation.Nullable java.lang.String $$0) {
+        this.setStructureName(net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)$$0) ? null : net.minecraft.resources.ResourceLocation.tryParse((java.lang.String)$$0));
     }
 
-    public void setStructureName(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation structureName) {
-        this.structureName = structureName;
+    public void setStructureName(@javax.annotation.Nullable net.minecraft.resources.ResourceLocation $$0) {
+        this.structureName = $$0;
     }
 
-    public void createdBy(net.minecraft.world.entity.LivingEntity author) {
-        this.author = author.getName().getString();
+    public void createdBy(net.minecraft.world.entity.LivingEntity $$0) {
+        this.author = $$0.getName().getString();
     }
 
     public net.minecraft.core.BlockPos getStructurePos() {
         return this.structurePos;
     }
 
-    public void setStructurePos(net.minecraft.core.BlockPos structurePos) {
-        this.structurePos = structurePos;
+    public void setStructurePos(net.minecraft.core.BlockPos $$0) {
+        this.structurePos = $$0;
     }
 
     public net.minecraft.core.Vec3i getStructureSize() {
@@ -174,35 +174,35 @@
         return this.mirror;
     }
 
-    public void setMirror(net.minecraft.world.level.block.Mirror mirror) {
-        this.mirror = mirror;
+    public void setMirror(net.minecraft.world.level.block.Mirror $$0) {
+        this.mirror = $$0;
     }
 
     public net.minecraft.world.level.block.Rotation getRotation() {
         return this.rotation;
     }
 
-    public void setRotation(net.minecraft.world.level.block.Rotation rotation) {
-        this.rotation = rotation;
+    public void setRotation(net.minecraft.world.level.block.Rotation $$0) {
+        this.rotation = $$0;
     }
 
     public java.lang.String getMetaData() {
         return this.metaData;
     }
 
-    public void setMetaData(java.lang.String metaData) {
-        this.metaData = metaData;
+    public void setMetaData(java.lang.String $$0) {
+        this.metaData = $$0;
     }
 
     public net.minecraft.world.level.block.state.properties.StructureMode getMode() {
         return this.mode;
     }
 
-    public void setMode(net.minecraft.world.level.block.state.properties.StructureMode mode) {
-        this.mode = mode;
-        net.minecraft.world.level.block.state.BlockState blockState = this.level.getBlockState(this.getBlockPos());
-        if (blockState.is(net.minecraft.world.level.block.Blocks.STRUCTURE_BLOCK)) {
-            this.level.setBlock(this.getBlockPos(), (net.minecraft.world.level.block.state.BlockState)blockState.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.StructureBlock.MODE, (java.lang.Comparable)mode), 2);
+    public void setMode(net.minecraft.world.level.block.state.properties.StructureMode $$0) {
+        this.mode = $$0;
+        net.minecraft.world.level.block.state.BlockState $$1 = this.level.getBlockState(this.getBlockPos());
+        if ($$1.is(net.minecraft.world.level.block.Blocks.STRUCTURE_BLOCK)) {
+            this.level.setBlock(this.getBlockPos(), (net.minecraft.world.level.block.state.BlockState)$$1.setValue((net.minecraft.world.level.block.state.properties.Property)net.minecraft.world.level.block.StructureBlock.MODE, (java.lang.Comparable)$$0), 2);
         }
     }
 
@@ -210,24 +210,24 @@
         return this.ignoreEntities;
     }
 
-    public void setIgnoreEntities(boolean ignoreEntities) {
-        this.ignoreEntities = ignoreEntities;
+    public void setIgnoreEntities(boolean $$0) {
+        this.ignoreEntities = $$0;
     }
 
     public float getIntegrity() {
         return this.integrity;
     }
 
-    public void setIntegrity(float integrity) {
-        this.integrity = integrity;
+    public void setIntegrity(float $$0) {
+        this.integrity = $$0;
     }
 
     public long getSeed() {
         return this.seed;
     }
 
-    public void setSeed(long seed) {
-        this.seed = seed;
+    public void setSeed(long $$0) {
+        this.seed = $$0;
     }
 
     public boolean detectSize() {
@@ -278,35 +278,38 @@
         return this.saveStructure(true);
     }
 
-    public boolean saveStructure(boolean writeToDisk) {
-        net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate structureTemplate;
+    /*
+     * WARNING - void declaration
+     */
+    public boolean saveStructure(boolean $$0) {
+        void $$6;
         if (this.mode != net.minecraft.world.level.block.state.properties.StructureMode.SAVE || this.level.isClientSide || this.structureName == null) {
             return false;
         }
-        net.minecraft.core.BlockPos blockPos = this.getBlockPos().offset((net.minecraft.core.Vec3i)this.structurePos);
-        net.minecraft.server.level.ServerLevel serverLevel = (net.minecraft.server.level.ServerLevel)this.level;
-        net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager structureTemplateManager = serverLevel.getStructureManager();
+        net.minecraft.core.BlockPos $$1 = this.getBlockPos().offset((net.minecraft.core.Vec3i)this.structurePos);
+        net.minecraft.server.level.ServerLevel $$2 = (net.minecraft.server.level.ServerLevel)this.level;
+        net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager $$3 = $$2.getStructureManager();
         try {
-            structureTemplate = structureTemplateManager.getOrCreate(this.structureName);
+            net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate $$4 = $$3.getOrCreate(this.structureName);
         }
-        catch (net.minecraft.ResourceLocationException resourceLocationException) {
+        catch (net.minecraft.ResourceLocationException $$5) {
             return false;
         }
-        structureTemplate.fillFromWorld(this.level, blockPos, this.structureSize, !this.ignoreEntities, net.minecraft.world.level.block.Blocks.STRUCTURE_VOID);
-        structureTemplate.setAuthor(this.author);
-        if (writeToDisk) {
+        $$6.fillFromWorld(this.level, $$1, this.structureSize, !this.ignoreEntities, net.minecraft.world.level.block.Blocks.STRUCTURE_VOID);
+        $$6.setAuthor(this.author);
+        if ($$0) {
             try {
-                return structureTemplateManager.save(this.structureName);
+                return $$3.save(this.structureName);
             }
-            catch (net.minecraft.ResourceLocationException resourceLocationException) {
+            catch (net.minecraft.ResourceLocationException $$7) {
                 return false;
             }
         }
         return true;
     }
 
-    public boolean loadStructure(net.minecraft.server.level.ServerLevel level) {
-        return this.loadStructure(level, true);
+    public boolean loadStructure(net.minecraft.server.level.ServerLevel $$0) {
+        return this.loadStructure($$0, true);
     }
 
     public static net.minecraft.util.RandomSource createRandom(long $$0) {
@@ -316,45 +319,47 @@
         return net.minecraft.util.RandomSource.create((long)$$0);
     }
 
-    public boolean loadStructure(net.minecraft.server.level.ServerLevel level, boolean bl) {
-        java.util.Optional optional;
+    /*
+     * WARNING - void declaration
+     */
+    public boolean loadStructure(net.minecraft.server.level.ServerLevel $$0, boolean $$1) {
+        void $$5;
         if (this.mode != net.minecraft.world.level.block.state.properties.StructureMode.LOAD || this.structureName == null) {
             return false;
         }
-        net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager structureTemplateManager = level.getStructureManager();
+        net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager $$2 = $$0.getStructureManager();
         try {
-            optional = structureTemplateManager.get(this.structureName);
+            java.util.Optional $$3 = $$2.get(this.structureName);
         }
-        catch (net.minecraft.ResourceLocationException resourceLocationException) {
+        catch (net.minecraft.ResourceLocationException $$4) {
             return false;
         }
-        if (!optional.isPresent()) {
+        if (!$$5.isPresent()) {
             return false;
         }
-        return this.loadStructure(level, bl, (net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate)optional.get());
+        return this.loadStructure($$0, $$1, (net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate)$$5.get());
     }
 
-    public boolean loadStructure(net.minecraft.server.level.ServerLevel level, boolean bl, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate structureTemplate) {
-        net.minecraft.world.level.block.state.BlockState blockState;
-        net.minecraft.core.Vec3i vec3i;
-        boolean bl2;
-        net.minecraft.core.BlockPos blockPos = this.getBlockPos();
-        if (!net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)structureTemplate.getAuthor())) {
-            this.author = structureTemplate.getAuthor();
+    public boolean loadStructure(net.minecraft.server.level.ServerLevel $$0, boolean $$1, net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplate $$2) {
+        net.minecraft.core.Vec3i $$4;
+        boolean $$5;
+        net.minecraft.core.BlockPos $$3 = this.getBlockPos();
+        if (!net.minecraft.util.StringUtil.isNullOrEmpty((java.lang.String)$$2.getAuthor())) {
+            this.author = $$2.getAuthor();
         }
-        if (!(bl2 = this.structureSize.equals((java.lang.Object)(vec3i = structureTemplate.getSize())))) {
-            this.structureSize = vec3i;
+        if (!($$5 = this.structureSize.equals((java.lang.Object)($$4 = $$2.getSize())))) {
+            this.structureSize = $$4;
             this.setChanged();
-            blockState = level.getBlockState(blockPos);
-            level.sendBlockUpdated(blockPos, blockState, blockState, 3);
+            net.minecraft.world.level.block.state.BlockState $$6 = $$0.getBlockState($$3);
+            $$0.sendBlockUpdated($$3, $$6, $$6, 3);
         }
-        if (!bl || bl2) {
-            blockState = new net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings().setMirror(this.mirror).setRotation(this.rotation).setIgnoreEntities(this.ignoreEntities);
+        if (!$$1 || $$5) {
+            net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings $$7 = new net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings().setMirror(this.mirror).setRotation(this.rotation).setIgnoreEntities(this.ignoreEntities);
             if (this.integrity < 1.0f) {
-                blockState.clearProcessors().addProcessor((net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor)new net.minecraft.world.level.levelgen.structure.templatesystem.BlockRotProcessor(net.minecraft.util.Mth.clamp((float)this.integrity, (float)0.0f, (float)1.0f))).setRandom(net.minecraft.world.level.block.entity.StructureBlockEntity.createRandom((long)this.seed));
+                $$7.clearProcessors().addProcessor((net.minecraft.world.level.levelgen.structure.templatesystem.StructureProcessor)new net.minecraft.world.level.levelgen.structure.templatesystem.BlockRotProcessor(net.minecraft.util.Mth.clamp((float)this.integrity, (float)0.0f, (float)1.0f))).setRandom(net.minecraft.world.level.block.entity.StructureBlockEntity.createRandom((long)this.seed));
             }
-            net.minecraft.core.BlockPos blockPos2 = blockPos.offset((net.minecraft.core.Vec3i)this.structurePos);
-            structureTemplate.placeInWorld((net.minecraft.world.level.ServerLevelAccessor)level, blockPos2, blockPos2, (net.minecraft.world.level.levelgen.structure.templatesystem.StructurePlaceSettings)blockState, net.minecraft.world.level.block.entity.StructureBlockEntity.createRandom((long)this.seed), 2);
+            net.minecraft.core.BlockPos $$8 = $$3.offset((net.minecraft.core.Vec3i)this.structurePos);
+            $$2.placeInWorld((net.minecraft.world.level.ServerLevelAccessor)$$0, $$8, $$8, $$7, net.minecraft.world.level.block.entity.StructureBlockEntity.createRandom((long)this.seed), 2);
             return true;
         }
         return false;
@@ -387,24 +392,24 @@
         return this.powered;
     }
 
-    public void setPowered(boolean powered) {
-        this.powered = powered;
+    public void setPowered(boolean $$0) {
+        this.powered = $$0;
     }
 
     public boolean getShowAir() {
         return this.showAir;
     }
 
-    public void setShowAir(boolean showAir) {
-        this.showAir = showAir;
+    public void setShowAir(boolean $$0) {
+        this.showAir = $$0;
     }
 
     public boolean getShowBoundingBox() {
         return this.showBoundingBox;
     }
 
-    public void setShowBoundingBox(boolean showBoundingBox) {
-        this.showBoundingBox = showBoundingBox;
+    public void setShowBoundingBox(boolean $$0) {
+        this.showBoundingBox = $$0;
     }
 
     public /* synthetic */ net.minecraft.network.protocol.Packet getUpdatePacket() {