--- net/minecraft/world/level/block/entity/TheEndPortalBlockEntity.java
+++ net/minecraft/world/level/block/entity/TheEndPortalBlockEntity.java
@@ -14,8 +14,8 @@
         this(net.minecraft.world.level.block.entity.BlockEntityType.END_PORTAL, $$0, $$1);
     }
 
-    public boolean shouldRenderFace(net.minecraft.core.Direction face) {
-        return face.getAxis() == net.minecraft.core.Direction$Axis.Y;
+    public boolean shouldRenderFace(net.minecraft.core.Direction $$0) {
+        return $$0.getAxis() == net.minecraft.core.Direction$Axis.Y;
     }
 }
 