--- net/minecraft/commands/arguments/coordinates/RotationArgument.java
+++ net/minecraft/commands/arguments/coordinates/RotationArgument.java
@@ -16,23 +16,23 @@
         return new net.minecraft.commands.arguments.coordinates.RotationArgument();
     }
 
-    public static net.minecraft.commands.arguments.coordinates.Coordinates getRotation(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (net.minecraft.commands.arguments.coordinates.Coordinates)context.getArgument(name, net.minecraft.commands.arguments.coordinates.Coordinates.class);
+    public static net.minecraft.commands.arguments.coordinates.Coordinates getRotation(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (net.minecraft.commands.arguments.coordinates.Coordinates)$$0.getArgument($$1, net.minecraft.commands.arguments.coordinates.Coordinates.class);
     }
 
-    public net.minecraft.commands.arguments.coordinates.Coordinates parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        int n = reader.getCursor();
-        if (!reader.canRead()) {
-            throw ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+    public net.minecraft.commands.arguments.coordinates.Coordinates parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        int $$1 = $$0.getCursor();
+        if (!$$0.canRead()) {
+            throw ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)reader, (boolean)false);
-        if (!reader.canRead() || reader.peek() != ' ') {
-            reader.setCursor(n);
-            throw ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$2 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)$$0, (boolean)false);
+        if (!$$0.canRead() || $$0.peek() != ' ') {
+            $$0.setCursor($$1);
+            throw ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        reader.skip();
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate2 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)reader, (boolean)false);
-        return new net.minecraft.commands.arguments.coordinates.WorldCoordinates(worldCoordinate2, worldCoordinate, new net.minecraft.commands.arguments.coordinates.WorldCoordinate(true, 0.0));
+        $$0.skip();
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$3 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)$$0, (boolean)false);
+        return new net.minecraft.commands.arguments.coordinates.WorldCoordinates($$3, $$2, new net.minecraft.commands.arguments.coordinates.WorldCoordinate(true, 0.0));
     }
 
     public java.util.Collection<java.lang.String> getExamples() {