--- net/minecraft/commands/arguments/coordinates/SwizzleArgument.java
+++ net/minecraft/commands/arguments/coordinates/SwizzleArgument.java
@@ -16,26 +16,41 @@
         return new net.minecraft.commands.arguments.coordinates.SwizzleArgument();
     }
 
-    public static java.util.EnumSet<net.minecraft.core.Direction$Axis> getSwizzle(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context, java.lang.String name) {
-        return (java.util.EnumSet)context.getArgument(name, java.util.EnumSet.class);
+    public static java.util.EnumSet<net.minecraft.core.Direction$Axis> getSwizzle(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0, java.lang.String $$1) {
+        return (java.util.EnumSet)$$0.getArgument($$1, java.util.EnumSet.class);
     }
 
-    public java.util.EnumSet<net.minecraft.core.Direction$Axis> parse(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        java.util.EnumSet enumSet = java.util.EnumSet.noneOf(net.minecraft.core.Direction$Axis.class);
-        while (reader.canRead() && reader.peek() != ' ') {
-            char c = reader.read();
-            net.minecraft.core.Direction$Axis axis = switch (c) {
-                case 'x' -> net.minecraft.core.Direction$Axis.X;
-                case 'y' -> net.minecraft.core.Direction$Axis.Y;
-                case 'z' -> net.minecraft.core.Direction$Axis.Z;
-                default -> throw ERROR_INVALID.create();
-            };
-            if (enumSet.contains((java.lang.Object)axis)) {
+    /*
+     * WARNING - void declaration
+     */
+    public java.util.EnumSet<net.minecraft.core.Direction$Axis> parse(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        java.util.EnumSet $$1 = java.util.EnumSet.noneOf(net.minecraft.core.Direction$Axis.class);
+        while ($$0.canRead() && $$0.peek() != ' ') {
+            void $$6;
+            char $$2 = $$0.read();
+            switch ($$2) {
+                case 'x': {
+                    net.minecraft.core.Direction$Axis $$3 = net.minecraft.core.Direction$Axis.X;
+                    break;
+                }
+                case 'y': {
+                    net.minecraft.core.Direction$Axis $$4 = net.minecraft.core.Direction$Axis.Y;
+                    break;
+                }
+                case 'z': {
+                    net.minecraft.core.Direction$Axis $$5 = net.minecraft.core.Direction$Axis.Z;
+                    break;
+                }
+                default: {
+                    throw ERROR_INVALID.create();
+                }
+            }
+            if ($$1.contains((java.lang.Object)$$6)) {
                 throw ERROR_INVALID.create();
             }
-            enumSet.add((java.lang.Object)axis);
+            $$1.add((java.lang.Object)$$6);
         }
-        return enumSet;
+        return $$1;
     }
 
     public java.util.Collection<java.lang.String> getExamples() {