--- net/minecraft/commands/arguments/coordinates/WorldCoordinates.java
+++ net/minecraft/commands/arguments/coordinates/WorldCoordinates.java
@@ -10,20 +10,20 @@
     public net.minecraft.commands.arguments.coordinates.WorldCoordinate y;
     public net.minecraft.commands.arguments.coordinates.WorldCoordinate z;
 
-    public net.minecraft.commands.arguments.coordinates.WorldCoordinates(net.minecraft.commands.arguments.coordinates.WorldCoordinate x, net.minecraft.commands.arguments.coordinates.WorldCoordinate y, net.minecraft.commands.arguments.coordinates.WorldCoordinate z) {
-        this.x = x;
-        this.y = y;
-        this.z = z;
+    public net.minecraft.commands.arguments.coordinates.WorldCoordinates(net.minecraft.commands.arguments.coordinates.WorldCoordinate $$0, net.minecraft.commands.arguments.coordinates.WorldCoordinate $$1, net.minecraft.commands.arguments.coordinates.WorldCoordinate $$2) {
+        this.x = $$0;
+        this.y = $$1;
+        this.z = $$2;
     }
 
-    public net.minecraft.world.phys.Vec3 getPosition(net.minecraft.commands.CommandSourceStack source) {
-        net.minecraft.world.phys.Vec3 vec3 = source.getPosition();
-        return new net.minecraft.world.phys.Vec3(this.x.get(vec3.x), this.y.get(vec3.y), this.z.get(vec3.z));
+    public net.minecraft.world.phys.Vec3 getPosition(net.minecraft.commands.CommandSourceStack $$0) {
+        net.minecraft.world.phys.Vec3 $$1 = $$0.getPosition();
+        return new net.minecraft.world.phys.Vec3(this.x.get($$1.x), this.y.get($$1.y), this.z.get($$1.z));
     }
 
-    public net.minecraft.world.phys.Vec2 getRotation(net.minecraft.commands.CommandSourceStack source) {
-        net.minecraft.world.phys.Vec2 vec2 = source.getRotation();
-        return new net.minecraft.world.phys.Vec2((float)this.x.get((double)vec2.x), (float)this.y.get((double)vec2.y));
+    public net.minecraft.world.phys.Vec2 getRotation(net.minecraft.commands.CommandSourceStack $$0) {
+        net.minecraft.world.phys.Vec2 $$1 = $$0.getRotation();
+        return new net.minecraft.world.phys.Vec2((float)this.x.get((double)$$1.x), (float)this.y.get((double)$$1.y));
     }
 
     public boolean isXRelative() {
@@ -38,57 +38,57 @@
         return this.z.isRelative();
     }
 
-    public boolean equals(java.lang.Object other) {
-        if (this == other) {
+    public boolean equals(java.lang.Object $$0) {
+        if (this == $$0) {
             return true;
         }
-        if (!(other instanceof net.minecraft.commands.arguments.coordinates.WorldCoordinates)) {
+        if (!($$0 instanceof net.minecraft.commands.arguments.coordinates.WorldCoordinates)) {
             return false;
         }
-        net.minecraft.commands.arguments.coordinates.WorldCoordinates worldCoordinates = (net.minecraft.commands.arguments.coordinates.WorldCoordinates)other;
-        if (!this.x.equals((java.lang.Object)worldCoordinates.x)) {
+        net.minecraft.commands.arguments.coordinates.WorldCoordinates $$1 = (net.minecraft.commands.arguments.coordinates.WorldCoordinates)$$0;
+        if (!this.x.equals((java.lang.Object)$$1.x)) {
             return false;
         }
-        if (!this.y.equals((java.lang.Object)worldCoordinates.y)) {
+        if (!this.y.equals((java.lang.Object)$$1.y)) {
             return false;
         }
-        return this.z.equals((java.lang.Object)worldCoordinates.z);
+        return this.z.equals((java.lang.Object)$$1.z);
     }
 
-    public static net.minecraft.commands.arguments.coordinates.WorldCoordinates parseInt(com.mojang.brigadier.StringReader reader) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        int n = reader.getCursor();
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseInt((com.mojang.brigadier.StringReader)reader);
-        if (!reader.canRead() || reader.peek() != ' ') {
-            reader.setCursor(n);
-            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+    public static net.minecraft.commands.arguments.coordinates.WorldCoordinates parseInt(com.mojang.brigadier.StringReader $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        int $$1 = $$0.getCursor();
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$2 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseInt((com.mojang.brigadier.StringReader)$$0);
+        if (!$$0.canRead() || $$0.peek() != ' ') {
+            $$0.setCursor($$1);
+            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        reader.skip();
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate2 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseInt((com.mojang.brigadier.StringReader)reader);
-        if (!reader.canRead() || reader.peek() != ' ') {
-            reader.setCursor(n);
-            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+        $$0.skip();
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$3 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseInt((com.mojang.brigadier.StringReader)$$0);
+        if (!$$0.canRead() || $$0.peek() != ' ') {
+            $$0.setCursor($$1);
+            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        reader.skip();
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate3 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseInt((com.mojang.brigadier.StringReader)reader);
-        return new net.minecraft.commands.arguments.coordinates.WorldCoordinates(worldCoordinate, worldCoordinate2, worldCoordinate3);
+        $$0.skip();
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$4 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseInt((com.mojang.brigadier.StringReader)$$0);
+        return new net.minecraft.commands.arguments.coordinates.WorldCoordinates($$2, $$3, $$4);
     }
 
-    public static net.minecraft.commands.arguments.coordinates.WorldCoordinates parseDouble(com.mojang.brigadier.StringReader reader, boolean centerCorrect) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        int n = reader.getCursor();
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)reader, (boolean)centerCorrect);
-        if (!reader.canRead() || reader.peek() != ' ') {
-            reader.setCursor(n);
-            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+    public static net.minecraft.commands.arguments.coordinates.WorldCoordinates parseDouble(com.mojang.brigadier.StringReader $$0, boolean $$1) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        int $$2 = $$0.getCursor();
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$3 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)$$0, (boolean)$$1);
+        if (!$$0.canRead() || $$0.peek() != ' ') {
+            $$0.setCursor($$2);
+            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        reader.skip();
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate2 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)reader, (boolean)false);
-        if (!reader.canRead() || reader.peek() != ' ') {
-            reader.setCursor(n);
-            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)reader);
+        $$0.skip();
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$4 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)$$0, (boolean)false);
+        if (!$$0.canRead() || $$0.peek() != ' ') {
+            $$0.setCursor($$2);
+            throw net.minecraft.commands.arguments.coordinates.Vec3Argument.ERROR_NOT_COMPLETE.createWithContext((com.mojang.brigadier.ImmutableStringReader)$$0);
         }
-        reader.skip();
-        net.minecraft.commands.arguments.coordinates.WorldCoordinate worldCoordinate3 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)reader, (boolean)centerCorrect);
-        return new net.minecraft.commands.arguments.coordinates.WorldCoordinates(worldCoordinate, worldCoordinate2, worldCoordinate3);
+        $$0.skip();
+        net.minecraft.commands.arguments.coordinates.WorldCoordinate $$5 = net.minecraft.commands.arguments.coordinates.WorldCoordinate.parseDouble((com.mojang.brigadier.StringReader)$$0, (boolean)$$1);
+        return new net.minecraft.commands.arguments.coordinates.WorldCoordinates($$3, $$4, $$5);
     }
 
     public static net.minecraft.commands.arguments.coordinates.WorldCoordinates absolute(double $$0, double $$1, double $$2) {