--- net/minecraft/util/thread/ReentrantBlockableEventLoop.java
+++ net/minecraft/util/thread/ReentrantBlockableEventLoop.java
@@ -20,10 +20,10 @@
         return this.reentrantCount != 0;
     }
 
-    public void doRunTask(R task) {
+    public void doRunTask(R $$0) {
         ++this.reentrantCount;
         try {
-            super.doRunTask(task);
+            super.doRunTask($$0);
         }
         finally {
             --this.reentrantCount;