--- net/minecraft/world/level/biome/AmbientAdditionsSettings.java
+++ net/minecraft/world/level/biome/AmbientAdditionsSettings.java
@@ -9,9 +9,9 @@
     public net.minecraft.sounds.SoundEvent soundEvent;
     public double tickChance;
 
-    public net.minecraft.world.level.biome.AmbientAdditionsSettings(net.minecraft.sounds.SoundEvent soundEvent, double tickChance) {
-        this.soundEvent = soundEvent;
-        this.tickChance = tickChance;
+    public net.minecraft.world.level.biome.AmbientAdditionsSettings(net.minecraft.sounds.SoundEvent $$0, double $$1) {
+        this.soundEvent = $$0;
+        this.tickChance = $$1;
     }
 
     public net.minecraft.sounds.SoundEvent getSoundEvent() {