--- net/minecraft/world/level/biome/AmbientMoodSettings.java
+++ net/minecraft/world/level/biome/AmbientMoodSettings.java
@@ -12,11 +12,11 @@
     public int blockSearchExtent;
     public double soundPositionOffset;
 
-    public net.minecraft.world.level.biome.AmbientMoodSettings(net.minecraft.sounds.SoundEvent soundEvent, int tickDelay, int blockSearchExtent, double soundPositionOffset) {
-        this.soundEvent = soundEvent;
-        this.tickDelay = tickDelay;
-        this.blockSearchExtent = blockSearchExtent;
-        this.soundPositionOffset = soundPositionOffset;
+    public net.minecraft.world.level.biome.AmbientMoodSettings(net.minecraft.sounds.SoundEvent $$0, int $$1, int $$2, double $$3) {
+        this.soundEvent = $$0;
+        this.tickDelay = $$1;
+        this.blockSearchExtent = $$2;
+        this.soundPositionOffset = $$3;
     }
 
     public net.minecraft.sounds.SoundEvent getSoundEvent() {