--- net/minecraft/world/level/biome/Biome$TemperatureModifier$2.java
+++ net/minecraft/world/level/biome/Biome$TemperatureModifier$2.java
@@ -10,15 +10,15 @@
         super(string, n, $$0);
     }
 
-    public float modifyTemperature(net.minecraft.core.BlockPos pos, float temperature) {
-        double d;
-        double d2;
-        double d3 = net.minecraft.world.level.biome.Biome.FROZEN_TEMPERATURE_NOISE.getValue((double)pos.getX() * 0.05, (double)pos.getZ() * 0.05, false) * 7.0;
-        double d4 = d3 + (d2 = net.minecraft.world.level.biome.Biome.BIOME_INFO_NOISE.getValue((double)pos.getX() * 0.2, (double)pos.getZ() * 0.2, false));
-        if (d4 < 0.3 && (d = net.minecraft.world.level.biome.Biome.BIOME_INFO_NOISE.getValue((double)pos.getX() * 0.09, (double)pos.getZ() * 0.09, false)) < 0.8) {
+    public float modifyTemperature(net.minecraft.core.BlockPos $$0, float $$1) {
+        double $$5;
+        double $$3;
+        double $$2 = net.minecraft.world.level.biome.Biome.FROZEN_TEMPERATURE_NOISE.getValue((double)$$0.getX() * 0.05, (double)$$0.getZ() * 0.05, false) * 7.0;
+        double $$4 = $$2 + ($$3 = net.minecraft.world.level.biome.Biome.BIOME_INFO_NOISE.getValue((double)$$0.getX() * 0.2, (double)$$0.getZ() * 0.2, false));
+        if ($$4 < 0.3 && ($$5 = net.minecraft.world.level.biome.Biome.BIOME_INFO_NOISE.getValue((double)$$0.getX() * 0.09, (double)$$0.getZ() * 0.09, false)) < 0.8) {
             return 0.2f;
         }
-        return temperature;
+        return $$1;
     }
 }
 