--- net/minecraft/world/level/biome/Biome$TemperatureModifier$1.java
+++ net/minecraft/world/level/biome/Biome$TemperatureModifier$1.java
@@ -10,8 +10,8 @@
         super(string, n, $$0);
     }
 
-    public float modifyTemperature(net.minecraft.core.BlockPos pos, float temperature) {
-        return temperature;
+    public float modifyTemperature(net.minecraft.core.BlockPos $$0, float $$1) {
+        return $$1;
     }
 }
 