--- net/minecraft/world/level/biome/BiomeSpecialEffects$GrassColorModifier$1.java
+++ net/minecraft/world/level/biome/BiomeSpecialEffects$GrassColorModifier$1.java
@@ -10,8 +10,8 @@
         super(string, n, $$0);
     }
 
-    public int modifyColor(double x, double z, int grassColor) {
-        return grassColor;
+    public int modifyColor(double $$0, double $$1, int $$2) {
+        return $$2;
     }
 }
 