--- net/minecraft/world/level/biome/Biomes.java
+++ net/minecraft/world/level/biome/Biomes.java
@@ -72,8 +72,8 @@
     public net.minecraft.world.level.biome.Biomes() {
     }
 
-    public static net.minecraft.resources.ResourceKey<net.minecraft.world.level.biome.Biome> register(java.lang.String name) {
-        return net.minecraft.resources.ResourceKey.create((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.BIOME_REGISTRY, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(name));
+    public static net.minecraft.resources.ResourceKey<net.minecraft.world.level.biome.Biome> register(java.lang.String $$0) {
+        return net.minecraft.resources.ResourceKey.create((net.minecraft.resources.ResourceKey)net.minecraft.core.Registry.BIOME_REGISTRY, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0));
     }
 
     static {