--- net/minecraft/world/level/biome/BiomeManager.java
+++ net/minecraft/world/level/biome/BiomeManager.java
@@ -17,8 +17,8 @@
         this.biomeZoomSeed = $$1;
     }
 
-    public static long obfuscateSeed(long seed) {
-        return com.google.common.hash.Hashing.sha256().hashLong(seed).asLong();
+    public static long obfuscateSeed(long $$0) {
+        return com.google.common.hash.Hashing.sha256().hashLong($$0).asLong();
     }
 
     public net.minecraft.world.level.biome.BiomeManager withDifferentSource(net.minecraft.world.level.biome.BiomeManager$NoiseBiomeSource $$0) {