--- net/minecraft/world/level/biome/BiomeGenerationSettings$Builder.java
+++ net/minecraft/world/level/biome/BiomeGenerationSettings$Builder.java
@@ -28,8 +28,8 @@
         return this;
     }
 
-    public void addFeatureStepsUpTo(int stage) {
-        while (this.features.size() <= stage) {
+    public void addFeatureStepsUpTo(int $$0) {
+        while (this.features.size() <= $$0) {
             this.features.add((java.lang.Object)com.google.common.collect.Lists.newArrayList());
         }
     }