--- net/minecraft/world/level/biome/BiomeSpecialEffects.java
+++ net/minecraft/world/level/biome/BiomeSpecialEffects.java
@@ -19,19 +19,19 @@
     public java.util.Optional<net.minecraft.world.level.biome.AmbientAdditionsSettings> ambientAdditionsSettings;
     public java.util.Optional<net.minecraft.sounds.Music> backgroundMusic;
 
-    public net.minecraft.world.level.biome.BiomeSpecialEffects(int fogColor, int waterColor, int waterFogColor, int skyColor, java.util.Optional<java.lang.Integer> foliageColorOverride, java.util.Optional<java.lang.Integer> grassColorOverride, net.minecraft.world.level.biome.BiomeSpecialEffects$GrassColorModifier grassColorModifier, java.util.Optional<net.minecraft.world.level.biome.AmbientParticleSettings> ambientParticleSettings, java.util.Optional<net.minecraft.sounds.SoundEvent> ambientLoopSoundEvent, java.util.Optional<net.minecraft.world.level.biome.AmbientMoodSettings> ambientMoodSettings, java.util.Optional<net.minecraft.world.level.biome.AmbientAdditionsSettings> ambientAdditionsSettings, java.util.Optional<net.minecraft.sounds.Music> backgroundMusic) {
-        this.fogColor = fogColor;
-        this.waterColor = waterColor;
-        this.waterFogColor = waterFogColor;
-        this.skyColor = skyColor;
-        this.foliageColorOverride = foliageColorOverride;
-        this.grassColorOverride = grassColorOverride;
-        this.grassColorModifier = grassColorModifier;
-        this.ambientParticleSettings = ambientParticleSettings;
-        this.ambientLoopSoundEvent = ambientLoopSoundEvent;
-        this.ambientMoodSettings = ambientMoodSettings;
-        this.ambientAdditionsSettings = ambientAdditionsSettings;
-        this.backgroundMusic = backgroundMusic;
+    public net.minecraft.world.level.biome.BiomeSpecialEffects(int $$0, int $$1, int $$2, int $$3, java.util.Optional<java.lang.Integer> $$4, java.util.Optional<java.lang.Integer> $$5, net.minecraft.world.level.biome.BiomeSpecialEffects$GrassColorModifier $$6, java.util.Optional<net.minecraft.world.level.biome.AmbientParticleSettings> $$7, java.util.Optional<net.minecraft.sounds.SoundEvent> $$8, java.util.Optional<net.minecraft.world.level.biome.AmbientMoodSettings> $$9, java.util.Optional<net.minecraft.world.level.biome.AmbientAdditionsSettings> $$10, java.util.Optional<net.minecraft.sounds.Music> $$11) {
+        this.fogColor = $$0;
+        this.waterColor = $$1;
+        this.waterFogColor = $$2;
+        this.skyColor = $$3;
+        this.foliageColorOverride = $$4;
+        this.grassColorOverride = $$5;
+        this.grassColorModifier = $$6;
+        this.ambientParticleSettings = $$7;
+        this.ambientLoopSoundEvent = $$8;
+        this.ambientMoodSettings = $$9;
+        this.ambientAdditionsSettings = $$10;
+        this.backgroundMusic = $$11;
     }
 
     public int getFogColor() {