--- net/minecraft/world/level/biome/Biome$Precipitation.java
+++ net/minecraft/world/level/biome/Biome$Precipitation.java
@@ -22,9 +22,9 @@
         return (net.minecraft.world.level.biome.Biome$Precipitation)java.lang.Enum.valueOf(net.minecraft.world.level.biome.Biome$Precipitation.class, (java.lang.String)$$0);
     }
 
-    public net.minecraft.world.level.biome.Biome$Precipitation(java.lang.String name) {
+    public net.minecraft.world.level.biome.Biome$Precipitation(java.lang.String $$0) {
         super(string, n);
-        this.name = name;
+        this.name = $$0;
     }
 
     public java.lang.String getName() {