--- net/minecraft/world/level/biome/BiomeSpecialEffects$GrassColorModifier$3.java
+++ net/minecraft/world/level/biome/BiomeSpecialEffects$GrassColorModifier$3.java
@@ -10,9 +10,9 @@
         super(string, n, $$0);
     }
 
-    public int modifyColor(double x, double z, int grassColor) {
-        double d = net.minecraft.world.level.biome.Biome.BIOME_INFO_NOISE.getValue(x * 0.0225, z * 0.0225, false);
-        if (d < -0.1) {
+    public int modifyColor(double $$0, double $$1, int $$2) {
+        double $$3 = net.minecraft.world.level.biome.Biome.BIOME_INFO_NOISE.getValue($$0 * 0.0225, $$1 * 0.0225, false);
+        if ($$3 < -0.1) {
             return 5011004;
         }
         return 6975545;