--- net/minecraft/world/level/biome/Biome$BiomeBuilder.java
+++ net/minecraft/world/level/biome/Biome$BiomeBuilder.java
@@ -23,38 +23,38 @@
         this.temperatureModifier = net.minecraft.world.level.biome.Biome$TemperatureModifier.NONE;
     }
 
-    public net.minecraft.world.level.biome.Biome$BiomeBuilder precipitation(net.minecraft.world.level.biome.Biome$Precipitation precipitation) {
-        this.precipitation = precipitation;
+    public net.minecraft.world.level.biome.Biome$BiomeBuilder precipitation(net.minecraft.world.level.biome.Biome$Precipitation $$0) {
+        this.precipitation = $$0;
         return this;
     }
 
-    public net.minecraft.world.level.biome.Biome$BiomeBuilder temperature(float temperature) {
-        this.temperature = java.lang.Float.valueOf((float)temperature);
+    public net.minecraft.world.level.biome.Biome$BiomeBuilder temperature(float $$0) {
+        this.temperature = java.lang.Float.valueOf((float)$$0);
         return this;
     }
 
-    public net.minecraft.world.level.biome.Biome$BiomeBuilder downfall(float downfall) {
-        this.downfall = java.lang.Float.valueOf((float)downfall);
+    public net.minecraft.world.level.biome.Biome$BiomeBuilder downfall(float $$0) {
+        this.downfall = java.lang.Float.valueOf((float)$$0);
         return this;
     }
 
-    public net.minecraft.world.level.biome.Biome$BiomeBuilder specialEffects(net.minecraft.world.level.biome.BiomeSpecialEffects effects) {
-        this.specialEffects = effects;
+    public net.minecraft.world.level.biome.Biome$BiomeBuilder specialEffects(net.minecraft.world.level.biome.BiomeSpecialEffects $$0) {
+        this.specialEffects = $$0;
         return this;
     }
 
-    public net.minecraft.world.level.biome.Biome$BiomeBuilder mobSpawnSettings(net.minecraft.world.level.biome.MobSpawnSettings mobSpawnSettings) {
-        this.mobSpawnSettings = mobSpawnSettings;
+    public net.minecraft.world.level.biome.Biome$BiomeBuilder mobSpawnSettings(net.minecraft.world.level.biome.MobSpawnSettings $$0) {
+        this.mobSpawnSettings = $$0;
         return this;
     }
 
-    public net.minecraft.world.level.biome.Biome$BiomeBuilder generationSettings(net.minecraft.world.level.biome.BiomeGenerationSettings generationSettings) {
-        this.generationSettings = generationSettings;
+    public net.minecraft.world.level.biome.Biome$BiomeBuilder generationSettings(net.minecraft.world.level.biome.BiomeGenerationSettings $$0) {
+        this.generationSettings = $$0;
         return this;
     }
 
-    public net.minecraft.world.level.biome.Biome$BiomeBuilder temperatureAdjustment(net.minecraft.world.level.biome.Biome$TemperatureModifier temperatureSettings) {
-        this.temperatureModifier = temperatureSettings;
+    public net.minecraft.world.level.biome.Biome$BiomeBuilder temperatureAdjustment(net.minecraft.world.level.biome.Biome$TemperatureModifier $$0) {
+        this.temperatureModifier = $$0;
         return this;
     }
 