--- net/minecraft/world/level/biome/Biome$ClimateSettings.java
+++ net/minecraft/world/level/biome/Biome$ClimateSettings.java
@@ -12,11 +12,11 @@
     public float downfall;
     public static com.mojang.serialization.MapCodec<net.minecraft.world.level.biome.Biome$ClimateSettings> CODEC;
 
-    public net.minecraft.world.level.biome.Biome$ClimateSettings(net.minecraft.world.level.biome.Biome$Precipitation precipitation, float temperature, net.minecraft.world.level.biome.Biome$TemperatureModifier temperatureModifier, float downfall) {
-        this.precipitation = precipitation;
-        this.temperature = temperature;
-        this.temperatureModifier = temperatureModifier;
-        this.downfall = downfall;
+    public net.minecraft.world.level.biome.Biome$ClimateSettings(net.minecraft.world.level.biome.Biome$Precipitation $$0, float $$1, net.minecraft.world.level.biome.Biome$TemperatureModifier $$2, float $$3) {
+        this.precipitation = $$0;
+        this.temperature = $$1;
+        this.temperatureModifier = $$2;
+        this.downfall = $$3;
     }
 
     public java.lang.String toString() {