--- net/minecraft/world/level/biome/MobSpawnSettings.java
+++ net/minecraft/world/level/biome/MobSpawnSettings.java
@@ -25,8 +25,8 @@
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.biome.MobSpawnSettings$MobSpawnCost getMobSpawnCost(net.minecraft.world.entity.EntityType<?> entityType) {
-        return (net.minecraft.world.level.biome.MobSpawnSettings$MobSpawnCost)this.mobSpawnCosts.get(entityType);
+    public net.minecraft.world.level.biome.MobSpawnSettings$MobSpawnCost getMobSpawnCost(net.minecraft.world.entity.EntityType<?> $$0) {
+        return (net.minecraft.world.level.biome.MobSpawnSettings$MobSpawnCost)this.mobSpawnCosts.get($$0);
     }
 
     public float getCreatureProbability() {