--- net/minecraft/world/level/biome/MobSpawnSettings$SpawnerData.java
+++ net/minecraft/world/level/biome/MobSpawnSettings$SpawnerData.java
@@ -11,8 +11,8 @@
     public int minCount;
     public int maxCount;
 
-    public net.minecraft.world.level.biome.MobSpawnSettings$SpawnerData(net.minecraft.world.entity.EntityType<?> type, int weight, int minCount, int maxCount) {
-        this(type, net.minecraft.util.random.Weight.of((int)weight), minCount, maxCount);
+    public net.minecraft.world.level.biome.MobSpawnSettings$SpawnerData(net.minecraft.world.entity.EntityType<?> $$0, int $$1, int $$2, int $$3) {
+        this($$0, net.minecraft.util.random.Weight.of((int)$$1), $$2, $$3);
     }
 
     public net.minecraft.world.level.biome.MobSpawnSettings$SpawnerData(net.minecraft.world.entity.EntityType<?> $$0, net.minecraft.util.random.Weight $$1, int $$2, int $$3) {