--- net/minecraft/world/level/biome/MobSpawnSettings$MobSpawnCost.java
+++ net/minecraft/world/level/biome/MobSpawnSettings$MobSpawnCost.java
@@ -9,9 +9,9 @@
     public double energyBudget;
     public double charge;
 
-    public net.minecraft.world.level.biome.MobSpawnSettings$MobSpawnCost(double energyBudget, double charge) {
-        this.energyBudget = energyBudget;
-        this.charge = charge;
+    public net.minecraft.world.level.biome.MobSpawnSettings$MobSpawnCost(double $$0, double $$1) {
+        this.energyBudget = $$0;
+        this.charge = $$1;
     }
 
     public double getEnergyBudget() {