--- net/minecraft/world/level/biome/OverworldBiomeBuilder.java
+++ net/minecraft/world/level/biome/OverworldBiomeBuilder.java
@@ -72,9 +72,9 @@
     public void addBiomes(java.util.function.Consumer<com.mojang.datafixers.util.Pair<net.minecraft.world.level.biome.Climate$ParameterPoint, net.minecraft.resources.ResourceKey<net.minecraft.world.level.biome.Biome>>> $$0) {
         if (net.minecraft.SharedConstants.debugGenerateSquareTerrainWithoutNoise) {
             net.minecraft.util.CubicSpline $$8;
-            net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate $$1 = new net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate(net.minecraft.data.BuiltinRegistries.DENSITY_FUNCTION.getHolderOrThrow(net.minecraft.world.level.levelgen.NoiseRouterData.CONTINENTS));
-            net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate $$2 = new net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate(net.minecraft.data.BuiltinRegistries.DENSITY_FUNCTION.getHolderOrThrow(net.minecraft.world.level.levelgen.NoiseRouterData.EROSION));
-            net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate $$3 = new net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate(net.minecraft.data.BuiltinRegistries.DENSITY_FUNCTION.getHolderOrThrow(net.minecraft.world.level.levelgen.NoiseRouterData.RIDGES_FOLDED));
+            net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate $$1 = new net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate((net.minecraft.core.Holder)net.minecraft.data.BuiltinRegistries.DENSITY_FUNCTION.getHolderOrThrow(net.minecraft.world.level.levelgen.NoiseRouterData.CONTINENTS));
+            net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate $$2 = new net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate((net.minecraft.core.Holder)net.minecraft.data.BuiltinRegistries.DENSITY_FUNCTION.getHolderOrThrow(net.minecraft.world.level.levelgen.NoiseRouterData.EROSION));
+            net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate $$3 = new net.minecraft.world.level.levelgen.DensityFunctions$Spline$Coordinate((net.minecraft.core.Holder)net.minecraft.data.BuiltinRegistries.DENSITY_FUNCTION.getHolderOrThrow(net.minecraft.world.level.levelgen.NoiseRouterData.RIDGES_FOLDED));
             $$0.accept((java.lang.Object)com.mojang.datafixers.util.Pair.of((java.lang.Object)net.minecraft.world.level.biome.Climate.parameters((net.minecraft.world.level.biome.Climate$Parameter)this.FULL_RANGE, (net.minecraft.world.level.biome.Climate$Parameter)this.FULL_RANGE, (net.minecraft.world.level.biome.Climate$Parameter)this.FULL_RANGE, (net.minecraft.world.level.biome.Climate$Parameter)this.FULL_RANGE, (net.minecraft.world.level.biome.Climate$Parameter)net.minecraft.world.level.biome.Climate$Parameter.point((float)0.0f), (net.minecraft.world.level.biome.Climate$Parameter)this.FULL_RANGE, (float)0.01f), (java.lang.Object)net.minecraft.world.level.biome.Biomes.PLAINS));
             net.minecraft.util.CubicSpline $$4 = net.minecraft.data.worldgen.TerrainProvider.buildErosionOffsetSpline((net.minecraft.util.ToFloatFunction)$$2, (net.minecraft.util.ToFloatFunction)$$3, (float)-0.15f, (float)0.0f, (float)0.0f, (float)0.1f, (float)0.0f, (float)-0.03f, (boolean)false, (boolean)false, (net.minecraft.util.ToFloatFunction)net.minecraft.util.ToFloatFunction.IDENTITY);
             if ($$4 instanceof net.minecraft.util.CubicSpline$Multipoint) {
@@ -377,8 +377,8 @@
         $$0.accept((java.lang.Object)com.mojang.datafixers.util.Pair.of((java.lang.Object)net.minecraft.world.level.biome.Climate.parameters((net.minecraft.world.level.biome.Climate$Parameter)$$1, (net.minecraft.world.level.biome.Climate$Parameter)$$2, (net.minecraft.world.level.biome.Climate$Parameter)$$3, (net.minecraft.world.level.biome.Climate$Parameter)$$4, (net.minecraft.world.level.biome.Climate$Parameter)net.minecraft.world.level.biome.Climate$Parameter.point((float)1.1f), (net.minecraft.world.level.biome.Climate$Parameter)$$5, (float)$$6), $$7));
     }
 
-    public static boolean isDeepDarkRegion(double $$0, double $$1) {
-        return $$0 < (double)-0.225f && $$1 > (double)0.9f;
+    public static boolean isDeepDarkRegion(net.minecraft.world.level.levelgen.DensityFunction $$0, net.minecraft.world.level.levelgen.DensityFunction $$1, net.minecraft.world.level.levelgen.DensityFunction$FunctionContext $$2) {
+        return $$0.compute($$2) < (double)-0.225f && $$1.compute($$2) > (double)0.9f;
     }
 
     public static java.lang.String getDebugStringForPeaksAndValleys(double $$0) {