--- net/minecraft/world/level/biome/TheEndBiomeSource.java
+++ net/minecraft/world/level/biome/TheEndBiomeSource.java
@@ -14,7 +14,7 @@
     public net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> barrens;
 
     public net.minecraft.world.level.biome.TheEndBiomeSource(net.minecraft.core.Registry<net.minecraft.world.level.biome.Biome> $$0) {
-        this($$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.THE_END), $$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.END_HIGHLANDS), $$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.END_MIDLANDS), $$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.SMALL_END_ISLANDS), $$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.END_BARRENS));
+        this((net.minecraft.core.Holder)$$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.THE_END), (net.minecraft.core.Holder)$$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.END_HIGHLANDS), (net.minecraft.core.Holder)$$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.END_MIDLANDS), (net.minecraft.core.Holder)$$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.SMALL_END_ISLANDS), (net.minecraft.core.Holder)$$0.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.END_BARRENS));
     }
 
     public net.minecraft.world.level.biome.TheEndBiomeSource(net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> $$0, net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> $$1, net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> $$2, net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> $$3, net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> $$4) {