--- net/minecraft/server/commands/data/BlockDataAccessor$1.java
+++ net/minecraft/server/commands/data/BlockDataAccessor$1.java
@@ -12,17 +12,17 @@
         this.val$argPrefix = string;
     }
 
-    public net.minecraft.server.commands.data.DataAccessor access(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        net.minecraft.core.BlockPos blockPos = net.minecraft.commands.arguments.coordinates.BlockPosArgument.getLoadedBlockPos(context, (java.lang.String)(this.val$argPrefix + "Pos"));
-        net.minecraft.world.level.block.entity.BlockEntity blockEntity = ((net.minecraft.commands.CommandSourceStack)context.getSource()).getLevel().getBlockEntity(blockPos);
-        if (blockEntity == null) {
+    public net.minecraft.server.commands.data.DataAccessor access(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        net.minecraft.core.BlockPos $$1 = net.minecraft.commands.arguments.coordinates.BlockPosArgument.getLoadedBlockPos($$0, (java.lang.String)(this.val$argPrefix + "Pos"));
+        net.minecraft.world.level.block.entity.BlockEntity $$2 = ((net.minecraft.commands.CommandSourceStack)$$0.getSource()).getLevel().getBlockEntity($$1);
+        if ($$2 == null) {
             throw net.minecraft.server.commands.data.BlockDataAccessor.ERROR_NOT_A_BLOCK_ENTITY.create();
         }
-        return new net.minecraft.server.commands.data.BlockDataAccessor(blockEntity, blockPos);
+        return new net.minecraft.server.commands.data.BlockDataAccessor($$2, $$1);
     }
 
-    public com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> wrap(com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> builder, java.util.function.Function<com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>, com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>> action) {
-        return builder.then(net.minecraft.commands.Commands.literal((java.lang.String)"block").then((com.mojang.brigadier.builder.ArgumentBuilder)action.apply((java.lang.Object)net.minecraft.commands.Commands.argument((java.lang.String)(this.val$argPrefix + "Pos"), (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.coordinates.BlockPosArgument.blockPos()))));
+    public com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> wrap(com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> $$0, java.util.function.Function<com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>, com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>> $$1) {
+        return $$0.then(net.minecraft.commands.Commands.literal((java.lang.String)"block").then((com.mojang.brigadier.builder.ArgumentBuilder)$$1.apply((java.lang.Object)net.minecraft.commands.Commands.argument((java.lang.String)(this.val$argPrefix + "Pos"), (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.coordinates.BlockPosArgument.blockPos()))));
     }
 }
 