--- net/minecraft/server/commands/data/EntityDataAccessor$1.java
+++ net/minecraft/server/commands/data/EntityDataAccessor$1.java
@@ -12,12 +12,12 @@
         this.val$arg = string;
     }
 
-    public net.minecraft.server.commands.data.DataAccessor access(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
-        return new net.minecraft.server.commands.data.EntityDataAccessor(net.minecraft.commands.arguments.EntityArgument.getEntity(context, (java.lang.String)this.val$arg));
+    public net.minecraft.server.commands.data.DataAccessor access(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+        return new net.minecraft.server.commands.data.EntityDataAccessor(net.minecraft.commands.arguments.EntityArgument.getEntity($$0, (java.lang.String)this.val$arg));
     }
 
-    public com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> wrap(com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> builder, java.util.function.Function<com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>, com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>> action) {
-        return builder.then(net.minecraft.commands.Commands.literal((java.lang.String)"entity").then((com.mojang.brigadier.builder.ArgumentBuilder)action.apply((java.lang.Object)net.minecraft.commands.Commands.argument((java.lang.String)this.val$arg, (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.EntityArgument.entity()))));
+    public com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> wrap(com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> $$0, java.util.function.Function<com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>, com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>> $$1) {
+        return $$0.then(net.minecraft.commands.Commands.literal((java.lang.String)"entity").then((com.mojang.brigadier.builder.ArgumentBuilder)$$1.apply((java.lang.Object)net.minecraft.commands.Commands.argument((java.lang.String)this.val$arg, (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.EntityArgument.entity()))));
     }
 }
 