--- net/minecraft/server/commands/data/EntityDataAccessor.java
+++ net/minecraft/server/commands/data/EntityDataAccessor.java
@@ -14,13 +14,13 @@
         this.entity = $$0;
     }
 
-    public void setData(net.minecraft.nbt.CompoundTag other) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
+    public void setData(net.minecraft.nbt.CompoundTag $$0) throws com.mojang.brigadier.exceptions.CommandSyntaxException {
         if (this.entity instanceof net.minecraft.world.entity.player.Player) {
             throw ERROR_NO_PLAYERS.create();
         }
-        java.util.UUID uUID = this.entity.getUUID();
-        this.entity.load(other);
-        this.entity.setUUID(uUID);
+        java.util.UUID $$1 = this.entity.getUUID();
+        this.entity.load($$0);
+        this.entity.setUUID($$1);
     }
 
     public net.minecraft.nbt.CompoundTag getData() {
@@ -31,12 +31,12 @@
         return net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.data.entity.modified", (java.lang.Object[])new java.lang.Object[]{this.entity.getDisplayName()});
     }
 
-    public net.minecraft.network.chat.Component getPrintSuccess(net.minecraft.nbt.Tag nbt) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.data.entity.query", (java.lang.Object[])new java.lang.Object[]{this.entity.getDisplayName(), net.minecraft.nbt.NbtUtils.toPrettyComponent((net.minecraft.nbt.Tag)nbt)});
+    public net.minecraft.network.chat.Component getPrintSuccess(net.minecraft.nbt.Tag $$0) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.data.entity.query", (java.lang.Object[])new java.lang.Object[]{this.entity.getDisplayName(), net.minecraft.nbt.NbtUtils.toPrettyComponent((net.minecraft.nbt.Tag)$$0)});
     }
 
-    public net.minecraft.network.chat.Component getPrintSuccess(net.minecraft.commands.arguments.NbtPathArgument$NbtPath path, double scale, int value) {
-        return net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.data.entity.get", (java.lang.Object[])new java.lang.Object[]{path, this.entity.getDisplayName(), java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"%.2f", (java.lang.Object[])new java.lang.Object[]{scale}), value});
+    public net.minecraft.network.chat.Component getPrintSuccess(net.minecraft.commands.arguments.NbtPathArgument$NbtPath $$0, double $$1, int $$2) {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"commands.data.entity.get", (java.lang.Object[])new java.lang.Object[]{$$0, this.entity.getDisplayName(), java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"%.2f", (java.lang.Object[])new java.lang.Object[]{$$1}), $$2});
     }
 
     public static /* synthetic */ net.minecraft.server.commands.data.DataCommands$DataProvider lambda$static$0(java.lang.String $$0) {