--- net/minecraft/server/commands/data/StorageDataAccessor$1.java
+++ net/minecraft/server/commands/data/StorageDataAccessor$1.java
@@ -12,12 +12,12 @@
         this.val$arg = string;
     }
 
-    public net.minecraft.server.commands.data.DataAccessor access(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> context) {
-        return new net.minecraft.server.commands.data.StorageDataAccessor(net.minecraft.server.commands.data.StorageDataAccessor.getGlobalTags(context), net.minecraft.commands.arguments.ResourceLocationArgument.getId(context, (java.lang.String)this.val$arg));
+    public net.minecraft.server.commands.data.DataAccessor access(com.mojang.brigadier.context.CommandContext<net.minecraft.commands.CommandSourceStack> $$0) {
+        return new net.minecraft.server.commands.data.StorageDataAccessor(net.minecraft.server.commands.data.StorageDataAccessor.getGlobalTags($$0), net.minecraft.commands.arguments.ResourceLocationArgument.getId($$0, (java.lang.String)this.val$arg));
     }
 
-    public com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> wrap(com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> builder, java.util.function.Function<com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>, com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>> action) {
-        return builder.then(net.minecraft.commands.Commands.literal((java.lang.String)"storage").then((com.mojang.brigadier.builder.ArgumentBuilder)action.apply((java.lang.Object)net.minecraft.commands.Commands.argument((java.lang.String)this.val$arg, (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.ResourceLocationArgument.id()).suggests(net.minecraft.server.commands.data.StorageDataAccessor.SUGGEST_STORAGE))));
+    public com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> wrap(com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?> $$0, java.util.function.Function<com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>, com.mojang.brigadier.builder.ArgumentBuilder<net.minecraft.commands.CommandSourceStack, ?>> $$1) {
+        return $$0.then(net.minecraft.commands.Commands.literal((java.lang.String)"storage").then((com.mojang.brigadier.builder.ArgumentBuilder)$$1.apply((java.lang.Object)net.minecraft.commands.Commands.argument((java.lang.String)this.val$arg, (com.mojang.brigadier.arguments.ArgumentType)net.minecraft.commands.arguments.ResourceLocationArgument.id()).suggests(net.minecraft.server.commands.data.StorageDataAccessor.SUGGEST_STORAGE))));
     }
 }
 