--- net/minecraft/client/particle/BaseAshSmokeParticle.java
+++ net/minecraft/client/particle/BaseAshSmokeParticle.java
@@ -35,8 +35,8 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_OPAQUE;
     }
 
-    public float getQuadSize(float scaleFactor) {
-        return this.quadSize * net.minecraft.util.Mth.clamp((float)(((float)this.age + scaleFactor) / (float)this.lifetime * 32.0f), (float)0.0f, (float)1.0f);
+    public float getQuadSize(float $$0) {
+        return this.quadSize * net.minecraft.util.Mth.clamp((float)(((float)this.age + $$0) / (float)this.lifetime * 32.0f), (float)0.0f, (float)1.0f);
     }
 
     public void tick() {