--- net/minecraft/client/particle/BreakingItemParticle$SnowballProvider.java
+++ net/minecraft/client/particle/BreakingItemParticle$SnowballProvider.java
@@ -9,8 +9,8 @@
     public net.minecraft.client.particle.BreakingItemParticle$SnowballProvider() {
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        return new net.minecraft.client.particle.BreakingItemParticle(level, x, y, z, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SNOWBALL));
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        return new net.minecraft.client.particle.BreakingItemParticle($$1, $$2, $$3, $$4, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.SNOWBALL));
     }
 }
 