--- net/minecraft/client/particle/BreakingItemParticle.java
+++ net/minecraft/client/particle/BreakingItemParticle.java
@@ -9,23 +9,23 @@
     public float uo;
     public float vo;
 
-    public net.minecraft.client.particle.BreakingItemParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.world.item.ItemStack stack) {
-        this(level, x, y, z, stack);
+    public net.minecraft.client.particle.BreakingItemParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.world.item.ItemStack $$7) {
+        this($$0, $$1, $$2, $$3, $$7);
         this.xd *= (double)0.1f;
         this.yd *= (double)0.1f;
         this.zd *= (double)0.1f;
-        this.xd += xSpeed;
-        this.yd += ySpeed;
-        this.zd += zSpeed;
+        this.xd += $$4;
+        this.yd += $$5;
+        this.zd += $$6;
     }
 
     public net.minecraft.client.particle.ParticleRenderType getRenderType() {
         return net.minecraft.client.particle.ParticleRenderType.TERRAIN_SHEET;
     }
 
-    public net.minecraft.client.particle.BreakingItemParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, net.minecraft.world.item.ItemStack stack) {
-        super(level, x, y, z, 0.0, 0.0, 0.0);
-        this.setSprite(net.minecraft.client.Minecraft.getInstance().getItemRenderer().getModel(stack, (net.minecraft.world.level.Level)level, null, 0).getParticleIcon());
+    public net.minecraft.client.particle.BreakingItemParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, net.minecraft.world.item.ItemStack $$4) {
+        super($$0, $$1, $$2, $$3, 0.0, 0.0, 0.0);
+        this.setSprite(net.minecraft.client.Minecraft.getInstance().getItemRenderer().getModel($$4, (net.minecraft.world.level.Level)$$0, null, 0).getParticleIcon());
         this.gravity = 1.0f;
         this.quadSize /= 2.0f;
         this.uo = this.random.nextFloat() * 3.0f;