--- net/minecraft/client/particle/CampfireSmokeParticle.java
+++ net/minecraft/client/particle/CampfireSmokeParticle.java
@@ -6,15 +6,15 @@
 public class net.minecraft.client.particle.CampfireSmokeParticle
 extends net.minecraft.client.particle.TextureSheetParticle
  {
-    public net.minecraft.client.particle.CampfireSmokeParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, boolean signal) {
-        super(level, x, y, z);
+    public net.minecraft.client.particle.CampfireSmokeParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, boolean $$7) {
+        super($$0, $$1, $$2, $$3);
         this.scale(3.0f);
         this.setSize(0.25f, 0.25f);
-        this.lifetime = signal ? this.random.nextInt(50) + 280 : this.random.nextInt(50) + 80;
+        this.lifetime = $$7 ? this.random.nextInt(50) + 280 : this.random.nextInt(50) + 80;
         this.gravity = 3.0E-6f;
-        this.xd = xSpeed;
-        this.yd = ySpeed + (double)(this.random.nextFloat() / 500.0f);
-        this.zd = zSpeed;
+        this.xd = $$4;
+        this.yd = $$5 + (double)(this.random.nextFloat() / 500.0f);
+        this.zd = $$6;
     }
 
     public void tick() {