--- net/minecraft/client/particle/CritParticle.java
+++ net/minecraft/client/particle/CritParticle.java
@@ -26,8 +26,8 @@
         this.tick();
     }
 
-    public float getQuadSize(float scaleFactor) {
-        return this.quadSize * net.minecraft.util.Mth.clamp((float)(((float)this.age + scaleFactor) / (float)this.lifetime * 32.0f), (float)0.0f, (float)1.0f);
+    public float getQuadSize(float $$0) {
+        return this.quadSize * net.minecraft.util.Mth.clamp((float)(((float)this.age + $$0) / (float)this.lifetime * 32.0f), (float)0.0f, (float)1.0f);
     }
 
     public void tick() {