--- net/minecraft/client/particle/CritParticle$DamageIndicatorProvider.java
+++ net/minecraft/client/particle/CritParticle$DamageIndicatorProvider.java
@@ -8,15 +8,15 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.CritParticle$DamageIndicatorProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.CritParticle$DamageIndicatorProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.CritParticle critParticle = new net.minecraft.client.particle.CritParticle(level, x, y, z, xSpeed, ySpeed + 1.0, zSpeed);
-        critParticle.setLifetime(20);
-        critParticle.pickSprite(this.sprite);
-        return critParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.CritParticle $$8 = new net.minecraft.client.particle.CritParticle($$1, $$2, $$3, $$4, $$5, $$6 + 1.0, $$7);
+        $$8.setLifetime(20);
+        $$8.pickSprite(this.sprite);
+        return $$8;
     }
 }
 