--- net/minecraft/client/particle/DripParticle$WaterFallProvider.java
+++ net/minecraft/client/particle/DripParticle$WaterFallProvider.java
@@ -8,15 +8,15 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.DripParticle$WaterFallProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.DripParticle$WaterFallProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.DripParticle$FallAndLandParticle fallAndLandParticle = new net.minecraft.client.particle.DripParticle$FallAndLandParticle(level, x, y, z, (net.minecraft.world.level.material.Fluid)net.minecraft.world.level.material.Fluids.WATER, (net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.SPLASH);
-        fallAndLandParticle.setColor(0.2f, 0.3f, 1.0f);
-        fallAndLandParticle.pickSprite(this.sprite);
-        return fallAndLandParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.DripParticle$FallAndLandParticle $$8 = new net.minecraft.client.particle.DripParticle$FallAndLandParticle($$1, $$2, $$3, $$4, (net.minecraft.world.level.material.Fluid)net.minecraft.world.level.material.Fluids.WATER, (net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.SPLASH);
+        $$8.setColor(0.2f, 0.3f, 1.0f);
+        $$8.pickSprite(this.sprite);
+        return $$8;
     }
 }
 