--- net/minecraft/client/particle/DripParticle.java
+++ net/minecraft/client/particle/DripParticle.java
@@ -9,11 +9,11 @@
     public net.minecraft.world.level.material.Fluid type;
     public boolean isGlowing;
 
-    public net.minecraft.client.particle.DripParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, net.minecraft.world.level.material.Fluid type) {
-        super(level, x, y, z);
+    public net.minecraft.client.particle.DripParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, net.minecraft.world.level.material.Fluid $$4) {
+        super($$0, $$1, $$2, $$3);
         this.setSize(0.01f, 0.01f);
         this.gravity = 0.06f;
-        this.type = type;
+        this.type = $$4;
     }
 
     public net.minecraft.world.level.material.Fluid getType() {
@@ -24,11 +24,11 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_OPAQUE;
     }
 
-    public int getLightColor(float partialTick) {
+    public int getLightColor(float $$0) {
         if (this.isGlowing) {
             return 240;
         }
-        return super.getLightColor(partialTick);
+        return super.getLightColor($$0);
     }
 
     public void tick() {