--- net/minecraft/client/particle/DripParticle$HoneyFallProvider.java
+++ net/minecraft/client/particle/DripParticle$HoneyFallProvider.java
@@ -8,16 +8,16 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.DripParticle$HoneyFallProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.DripParticle$HoneyFallProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.DripParticle$HoneyFallAndLandParticle honeyFallAndLandParticle = new net.minecraft.client.particle.DripParticle$HoneyFallAndLandParticle(level, x, y, z, net.minecraft.world.level.material.Fluids.EMPTY, (net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.LANDING_HONEY);
-        honeyFallAndLandParticle.gravity = 0.01f;
-        honeyFallAndLandParticle.setColor(0.582f, 0.448f, 0.082f);
-        honeyFallAndLandParticle.pickSprite(this.sprite);
-        return honeyFallAndLandParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.DripParticle$HoneyFallAndLandParticle $$8 = new net.minecraft.client.particle.DripParticle$HoneyFallAndLandParticle($$1, $$2, $$3, $$4, net.minecraft.world.level.material.Fluids.EMPTY, (net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.LANDING_HONEY);
+        $$8.gravity = 0.01f;
+        $$8.setColor(0.582f, 0.448f, 0.082f);
+        $$8.pickSprite(this.sprite);
+        return $$8;
     }
 }
 