--- net/minecraft/client/particle/DripParticle$FallAndLandParticle.java
+++ net/minecraft/client/particle/DripParticle$FallAndLandParticle.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.core.particles.ParticleOptions landParticle;
 
-    public net.minecraft.client.particle.DripParticle$FallAndLandParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, net.minecraft.world.level.material.Fluid type, net.minecraft.core.particles.ParticleOptions landParticle) {
-        super(level, x, y, z, type);
-        this.landParticle = landParticle;
+    public net.minecraft.client.particle.DripParticle$FallAndLandParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, net.minecraft.world.level.material.Fluid $$4, net.minecraft.core.particles.ParticleOptions $$5) {
+        super($$0, $$1, $$2, $$3, $$4);
+        this.landParticle = $$5;
     }
 
     public void postMoveUpdate() {