--- net/minecraft/client/particle/DripParticle$HoneyLandProvider.java
+++ net/minecraft/client/particle/DripParticle$HoneyLandProvider.java
@@ -8,16 +8,16 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.DripParticle$HoneyLandProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.DripParticle$HoneyLandProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.DripParticle$DripLandParticle dripLandParticle = new net.minecraft.client.particle.DripParticle$DripLandParticle(level, x, y, z, net.minecraft.world.level.material.Fluids.EMPTY);
-        dripLandParticle.lifetime = (int)(128.0 / (java.lang.Math.random() * 0.8 + 0.2));
-        dripLandParticle.setColor(0.522f, 0.408f, 0.082f);
-        dripLandParticle.pickSprite(this.sprite);
-        return dripLandParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.DripParticle$DripLandParticle $$8 = new net.minecraft.client.particle.DripParticle$DripLandParticle($$1, $$2, $$3, $$4, net.minecraft.world.level.material.Fluids.EMPTY);
+        $$8.lifetime = (int)(128.0 / (java.lang.Math.random() * 0.8 + 0.2));
+        $$8.setColor(0.522f, 0.408f, 0.082f);
+        $$8.pickSprite(this.sprite);
+        return $$8;
     }
 }
 