--- net/minecraft/client/particle/DripParticle$ObsidianTearFallProvider.java
+++ net/minecraft/client/particle/DripParticle$ObsidianTearFallProvider.java
@@ -8,17 +8,17 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.DripParticle$ObsidianTearFallProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.DripParticle$ObsidianTearFallProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.DripParticle$FallAndLandParticle fallAndLandParticle = new net.minecraft.client.particle.DripParticle$FallAndLandParticle(level, x, y, z, net.minecraft.world.level.material.Fluids.EMPTY, (net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.LANDING_OBSIDIAN_TEAR);
-        fallAndLandParticle.isGlowing = true;
-        fallAndLandParticle.gravity = 0.01f;
-        fallAndLandParticle.setColor(0.51171875f, 0.03125f, 0.890625f);
-        fallAndLandParticle.pickSprite(this.sprite);
-        return fallAndLandParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.DripParticle$FallAndLandParticle $$8 = new net.minecraft.client.particle.DripParticle$FallAndLandParticle($$1, $$2, $$3, $$4, net.minecraft.world.level.material.Fluids.EMPTY, (net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.LANDING_OBSIDIAN_TEAR);
+        $$8.isGlowing = true;
+        $$8.gravity = 0.01f;
+        $$8.setColor(0.51171875f, 0.03125f, 0.890625f);
+        $$8.pickSprite(this.sprite);
+        return $$8;
     }
 }
 