--- net/minecraft/client/particle/DripParticle$DripHangParticle.java
+++ net/minecraft/client/particle/DripParticle$DripHangParticle.java
@@ -8,9 +8,9 @@
  {
     public net.minecraft.core.particles.ParticleOptions fallingParticle;
 
-    public net.minecraft.client.particle.DripParticle$DripHangParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, net.minecraft.world.level.material.Fluid type, net.minecraft.core.particles.ParticleOptions fallingParticle) {
-        super(level, x, y, z, type);
-        this.fallingParticle = fallingParticle;
+    public net.minecraft.client.particle.DripParticle$DripHangParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, net.minecraft.world.level.material.Fluid $$4, net.minecraft.core.particles.ParticleOptions $$5) {
+        super($$0, $$1, $$2, $$3, $$4);
+        this.fallingParticle = $$5;
         this.gravity *= 0.02f;
         this.lifetime = 40;
     }