--- net/minecraft/client/particle/DustParticle.java
+++ net/minecraft/client/particle/DustParticle.java
@@ -6,8 +6,8 @@
 public class net.minecraft.client.particle.DustParticle
 extends net.minecraft.client.particle.DustParticleBase<net.minecraft.core.particles.DustParticleOptions>
  {
-    public net.minecraft.client.particle.DustParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.core.particles.DustParticleOptions options, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, xSpeed, ySpeed, zSpeed, (net.minecraft.core.particles.DustParticleOptionsBase)options, sprites);
+    public net.minecraft.client.particle.DustParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.core.particles.DustParticleOptions $$7, net.minecraft.client.particle.SpriteSet $$8) {
+        super($$0, $$1, $$2, $$3, $$4, $$5, $$6, (net.minecraft.core.particles.DustParticleOptionsBase)$$7, $$8);
     }
 }
 