--- net/minecraft/client/particle/EnchantmentTableParticle.java
+++ net/minecraft/client/particle/EnchantmentTableParticle.java
@@ -37,22 +37,22 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_OPAQUE;
     }
 
-    public void move(double x, double y, double z) {
-        this.setBoundingBox(this.getBoundingBox().move(x, y, z));
+    public void move(double $$0, double $$1, double $$2) {
+        this.setBoundingBox(this.getBoundingBox().move($$0, $$1, $$2));
         this.setLocationFromBoundingbox();
     }
 
-    public int getLightColor(float partialTick) {
-        int n = super.getLightColor(partialTick);
-        float f = (float)this.age / (float)this.lifetime;
-        f *= f;
-        f *= f;
-        int n2 = n & 0xFF;
-        int n3 = n >> 16 & 0xFF;
-        if ((n3 += (int)(f * 15.0f * 16.0f)) > 240) {
-            n3 = 240;
+    public int getLightColor(float $$0) {
+        int $$1 = super.getLightColor($$0);
+        float $$2 = (float)this.age / (float)this.lifetime;
+        $$2 *= $$2;
+        $$2 *= $$2;
+        int $$3 = $$1 & 0xFF;
+        int $$4 = $$1 >> 16 & 0xFF;
+        if (($$4 += (int)($$2 * 15.0f * 16.0f)) > 240) {
+            $$4 = 240;
         }
-        return n2 | n3 << 16;
+        return $$3 | $$4 << 16;
     }
 
     public void tick() {