--- net/minecraft/client/particle/EndRodParticle.java
+++ net/minecraft/client/particle/EndRodParticle.java
@@ -6,19 +6,19 @@
 public class net.minecraft.client.particle.EndRodParticle
 extends net.minecraft.client.particle.SimpleAnimatedParticle
  {
-    public net.minecraft.client.particle.EndRodParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, sprites, 0.0125f);
-        this.xd = xSpeed;
-        this.yd = ySpeed;
-        this.zd = zSpeed;
+    public net.minecraft.client.particle.EndRodParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.client.particle.SpriteSet $$7) {
+        super($$0, $$1, $$2, $$3, $$7, 0.0125f);
+        this.xd = $$4;
+        this.yd = $$5;
+        this.zd = $$6;
         this.quadSize *= 0.75f;
         this.lifetime = 60 + this.random.nextInt(12);
         this.setFadeColor(15916745);
-        this.setSpriteFromAge(sprites);
+        this.setSpriteFromAge($$7);
     }
 
-    public void move(double x, double y, double z) {
-        this.setBoundingBox(this.getBoundingBox().move(x, y, z));
+    public void move(double $$0, double $$1, double $$2) {
+        this.setBoundingBox(this.getBoundingBox().move($$0, $$1, $$2));
         this.setLocationFromBoundingbox();
     }
 }