--- net/minecraft/client/particle/ExplodeParticle.java
+++ net/minecraft/client/particle/ExplodeParticle.java
@@ -8,21 +8,21 @@
  {
     public net.minecraft.client.particle.SpriteSet sprites;
 
-    public net.minecraft.client.particle.ExplodeParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z);
-        float f;
+    public net.minecraft.client.particle.ExplodeParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.client.particle.SpriteSet $$7) {
+        super($$0, $$1, $$2, $$3);
+        float $$8;
         this.gravity = -0.1f;
         this.friction = 0.9f;
-        this.sprites = sprites;
-        this.xd = xSpeed + (java.lang.Math.random() * 2.0 - 1.0) * (double)0.05f;
-        this.yd = ySpeed + (java.lang.Math.random() * 2.0 - 1.0) * (double)0.05f;
-        this.zd = zSpeed + (java.lang.Math.random() * 2.0 - 1.0) * (double)0.05f;
-        this.rCol = f = this.random.nextFloat() * 0.3f + 0.7f;
-        this.gCol = f;
-        this.bCol = f;
+        this.sprites = $$7;
+        this.xd = $$4 + (java.lang.Math.random() * 2.0 - 1.0) * (double)0.05f;
+        this.yd = $$5 + (java.lang.Math.random() * 2.0 - 1.0) * (double)0.05f;
+        this.zd = $$6 + (java.lang.Math.random() * 2.0 - 1.0) * (double)0.05f;
+        this.rCol = $$8 = this.random.nextFloat() * 0.3f + 0.7f;
+        this.gCol = $$8;
+        this.bCol = $$8;
         this.quadSize = 0.1f * (this.random.nextFloat() * this.random.nextFloat() * 6.0f + 1.0f);
         this.lifetime = (int)(16.0 / ((double)this.random.nextFloat() * 0.8 + 0.2)) + 2;
-        this.setSpriteFromAge(sprites);
+        this.setSpriteFromAge($$7);
     }
 
     public net.minecraft.client.particle.ParticleRenderType getRenderType() {