--- net/minecraft/client/particle/FireworkParticles$SparkProvider.java
+++ net/minecraft/client/particle/FireworkParticles$SparkProvider.java
@@ -8,14 +8,14 @@
  {
     public net.minecraft.client.particle.SpriteSet sprites;
 
-    public net.minecraft.client.particle.FireworkParticles$SparkProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprites = sprites;
+    public net.minecraft.client.particle.FireworkParticles$SparkProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprites = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.FireworkParticles$SparkParticle sparkParticle = new net.minecraft.client.particle.FireworkParticles$SparkParticle(level, x, y, z, xSpeed, ySpeed, zSpeed, net.minecraft.client.Minecraft.getInstance().particleEngine, this.sprites);
-        sparkParticle.setAlpha(0.99f);
-        return sparkParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.FireworkParticles$SparkParticle $$8 = new net.minecraft.client.particle.FireworkParticles$SparkParticle($$1, $$2, $$3, $$4, $$5, $$6, $$7, net.minecraft.client.Minecraft.getInstance().particleEngine, this.sprites);
+        $$8.setAlpha(0.99f);
+        return $$8;
     }
 }
 