--- net/minecraft/client/particle/FireworkParticles$SparkParticle.java
+++ net/minecraft/client/particle/FireworkParticles$SparkParticle.java
@@ -14,28 +14,28 @@
     public float fadeB;
     public boolean hasFade;
 
-    public net.minecraft.client.particle.FireworkParticles$SparkParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.client.particle.ParticleEngine engine, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, sprites, 0.1f);
-        this.xd = xSpeed;
-        this.yd = ySpeed;
-        this.zd = zSpeed;
-        this.engine = engine;
+    public net.minecraft.client.particle.FireworkParticles$SparkParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.client.particle.ParticleEngine $$7, net.minecraft.client.particle.SpriteSet $$8) {
+        super($$0, $$1, $$2, $$3, $$8, 0.1f);
+        this.xd = $$4;
+        this.yd = $$5;
+        this.zd = $$6;
+        this.engine = $$7;
         this.quadSize *= 0.75f;
         this.lifetime = 48 + this.random.nextInt(12);
-        this.setSpriteFromAge(sprites);
+        this.setSpriteFromAge($$8);
     }
 
-    public void setTrail(boolean trail) {
-        this.trail = trail;
+    public void setTrail(boolean $$0) {
+        this.trail = $$0;
     }
 
-    public void setFlicker(boolean twinkle) {
-        this.flicker = twinkle;
+    public void setFlicker(boolean $$0) {
+        this.flicker = $$0;
     }
 
-    public void render(com.mojang.blaze3d.vertex.VertexConsumer buffer, net.minecraft.client.Camera renderInfo, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.VertexConsumer $$0, net.minecraft.client.Camera $$1, float $$2) {
         if (!this.flicker || this.age < this.lifetime / 3 || (this.age + this.lifetime) / 3 % 2 == 0) {
-            super.render(buffer, renderInfo, partialTicks);
+            super.render($$0, $$1, $$2);
         }
     }
 