--- net/minecraft/client/particle/FireworkParticles$Starter.java
+++ net/minecraft/client/particle/FireworkParticles$Starter.java
@@ -11,22 +11,22 @@
     public net.minecraft.nbt.ListTag explosions;
     public boolean twinkleDelay;
 
-    public net.minecraft.client.particle.FireworkParticles$Starter(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.client.particle.ParticleEngine engine, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag tag) {
-        super(level, x, y, z);
-        this.xd = xSpeed;
-        this.yd = ySpeed;
-        this.zd = zSpeed;
-        this.engine = engine;
+    public net.minecraft.client.particle.FireworkParticles$Starter(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.client.particle.ParticleEngine $$7, @javax.annotation.Nullable net.minecraft.nbt.CompoundTag $$8) {
+        super($$0, $$1, $$2, $$3);
+        this.xd = $$4;
+        this.yd = $$5;
+        this.zd = $$6;
+        this.engine = $$7;
         this.lifetime = 8;
-        if (tag != null) {
-            this.explosions = tag.getList("Explosions", 10);
+        if ($$8 != null) {
+            this.explosions = $$8.getList("Explosions", 10);
             if (this.explosions.isEmpty()) {
                 this.explosions = null;
             } else {
                 this.lifetime = this.explosions.size() * 2 - 1;
-                for (int i = 0; i < this.explosions.size(); ++i) {
-                    net.minecraft.nbt.CompoundTag compoundTag = this.explosions.getCompound(i);
-                    if (!compoundTag.getBoolean("Flicker")) continue;
+                for (int $$9 = 0; $$9 < this.explosions.size(); ++$$9) {
+                    net.minecraft.nbt.CompoundTag $$10 = this.explosions.getCompound($$9);
+                    if (!$$10.getBoolean("Flicker")) continue;
                     this.twinkleDelay = true;
                     this.lifetime += 15;
                     break;
@@ -112,73 +112,73 @@
         return $$0.gameRenderer.getMainCamera().getPosition().distanceToSqr(this.x, this.y, this.z) >= 256.0;
     }
 
-    public void createParticle(double x, double y, double z, double motionX, double motionY, double motionZ, int[] sparkColors, int[] sparkColorFades, boolean hasTrail, boolean hasTwinkle) {
-        net.minecraft.client.particle.FireworkParticles$SparkParticle sparkParticle = (net.minecraft.client.particle.FireworkParticles$SparkParticle)this.engine.createParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.FIREWORK, x, y, z, motionX, motionY, motionZ);
-        sparkParticle.setTrail(hasTrail);
-        sparkParticle.setFlicker(hasTwinkle);
-        sparkParticle.setAlpha(0.99f);
-        int n = this.random.nextInt(sparkColors.length);
-        sparkParticle.setColor(sparkColors[n]);
-        if (sparkColorFades.length > 0) {
-            sparkParticle.setFadeColor(net.minecraft.Util.getRandom((int[])sparkColorFades, (net.minecraft.util.RandomSource)this.random));
+    public void createParticle(double $$0, double $$1, double $$2, double $$3, double $$4, double $$5, int[] $$6, int[] $$7, boolean $$8, boolean $$9) {
+        net.minecraft.client.particle.FireworkParticles$SparkParticle $$10 = (net.minecraft.client.particle.FireworkParticles$SparkParticle)this.engine.createParticle((net.minecraft.core.particles.ParticleOptions)net.minecraft.core.particles.ParticleTypes.FIREWORK, $$0, $$1, $$2, $$3, $$4, $$5);
+        $$10.setTrail($$8);
+        $$10.setFlicker($$9);
+        $$10.setAlpha(0.99f);
+        int $$11 = this.random.nextInt($$6.length);
+        $$10.setColor($$6[$$11]);
+        if ($$7.length > 0) {
+            $$10.setFadeColor(net.minecraft.Util.getRandom((int[])$$7, (net.minecraft.util.RandomSource)this.random));
         }
     }
 
-    public void createParticleBall(double speed, int size, int[] colours, int[] fadeColours, boolean trail, boolean twinkle) {
-        double d = this.x;
-        double d2 = this.y;
-        double d3 = this.z;
-        for (int i = -size; i <= size; ++i) {
-            for (int j = -size; j <= size; ++j) {
-                for (int k = -size; k <= size; ++k) {
-                    double d4 = (double)j + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
-                    double d5 = (double)i + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
-                    double d6 = (double)k + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
-                    double d7 = java.lang.Math.sqrt((double)(d4 * d4 + d5 * d5 + d6 * d6)) / speed + this.random.nextGaussian() * 0.05;
-                    this.createParticle(d, d2, d3, d4 / d7, d5 / d7, d6 / d7, colours, fadeColours, trail, twinkle);
-                    if (i == -size || i == size || j == -size || j == size) continue;
-                    k += size * 2 - 1;
+    public void createParticleBall(double $$0, int $$1, int[] $$2, int[] $$3, boolean $$4, boolean $$5) {
+        double $$6 = this.x;
+        double $$7 = this.y;
+        double $$8 = this.z;
+        for (int $$9 = -$$1; $$9 <= $$1; ++$$9) {
+            for (int $$10 = -$$1; $$10 <= $$1; ++$$10) {
+                for (int $$11 = -$$1; $$11 <= $$1; ++$$11) {
+                    double $$12 = (double)$$10 + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
+                    double $$13 = (double)$$9 + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
+                    double $$14 = (double)$$11 + (this.random.nextDouble() - this.random.nextDouble()) * 0.5;
+                    double $$15 = java.lang.Math.sqrt((double)($$12 * $$12 + $$13 * $$13 + $$14 * $$14)) / $$0 + this.random.nextGaussian() * 0.05;
+                    this.createParticle($$6, $$7, $$8, $$12 / $$15, $$13 / $$15, $$14 / $$15, $$2, $$3, $$4, $$5);
+                    if ($$9 == -$$1 || $$9 == $$1 || $$10 == -$$1 || $$10 == $$1) continue;
+                    $$11 += $$1 * 2 - 1;
                 }
             }
         }
     }
 
-    public void createParticleShape(double speed, double[][] shape, int[] colours, int[] fadeColours, boolean trail, boolean twinkle, boolean creeper) {
-        double d = shape[0][0];
-        double d2 = shape[0][1];
-        this.createParticle(this.x, this.y, this.z, d * speed, d2 * speed, 0.0, colours, fadeColours, trail, twinkle);
-        float f = this.random.nextFloat() * (float)java.lang.Math.PI;
-        double d3 = creeper ? 0.034 : 0.34;
-        for (int i = 0; i < 3; ++i) {
-            double d4 = (double)f + (double)((float)i * (float)java.lang.Math.PI) * d3;
-            double d5 = d;
-            double d6 = d2;
-            for (int j = 1; j < shape.length; ++j) {
-                double d7 = shape[j][0];
-                double d8 = shape[j][1];
-                for (double d9 = 0.25; d9 <= 1.0; d9 += 0.25) {
-                    double d10 = net.minecraft.util.Mth.lerp((double)d9, (double)d5, (double)d7) * speed;
-                    double d11 = net.minecraft.util.Mth.lerp((double)d9, (double)d6, (double)d8) * speed;
-                    double d12 = d10 * java.lang.Math.sin((double)d4);
-                    d10 *= java.lang.Math.cos((double)d4);
-                    for (double d13 = -1.0; d13 <= 1.0; d13 += 2.0) {
-                        this.createParticle(this.x, this.y, this.z, d10 * d13, d11, d12 * d13, colours, fadeColours, trail, twinkle);
+    public void createParticleShape(double $$0, double[][] $$1, int[] $$2, int[] $$3, boolean $$4, boolean $$5, boolean $$6) {
+        double $$7 = $$1[0][0];
+        double $$8 = $$1[0][1];
+        this.createParticle(this.x, this.y, this.z, $$7 * $$0, $$8 * $$0, 0.0, $$2, $$3, $$4, $$5);
+        float $$9 = this.random.nextFloat() * (float)java.lang.Math.PI;
+        double $$10 = $$6 ? 0.034 : 0.34;
+        for (int $$11 = 0; $$11 < 3; ++$$11) {
+            double $$12 = (double)$$9 + (double)((float)$$11 * (float)java.lang.Math.PI) * $$10;
+            double $$13 = $$7;
+            double $$14 = $$8;
+            for (int $$15 = 1; $$15 < $$1.length; ++$$15) {
+                double $$16 = $$1[$$15][0];
+                double $$17 = $$1[$$15][1];
+                for (double $$18 = 0.25; $$18 <= 1.0; $$18 += 0.25) {
+                    double $$19 = net.minecraft.util.Mth.lerp((double)$$18, (double)$$13, (double)$$16) * $$0;
+                    double $$20 = net.minecraft.util.Mth.lerp((double)$$18, (double)$$14, (double)$$17) * $$0;
+                    double $$21 = $$19 * java.lang.Math.sin((double)$$12);
+                    $$19 *= java.lang.Math.cos((double)$$12);
+                    for (double $$22 = -1.0; $$22 <= 1.0; $$22 += 2.0) {
+                        this.createParticle(this.x, this.y, this.z, $$19 * $$22, $$20, $$21 * $$22, $$2, $$3, $$4, $$5);
                     }
                 }
-                d5 = d7;
-                d6 = d8;
+                $$13 = $$16;
+                $$14 = $$17;
             }
         }
     }
 
-    public void createParticleBurst(int[] colours, int[] fadeColours, boolean trail, boolean twinkle) {
-        double d = this.random.nextGaussian() * 0.05;
-        double d2 = this.random.nextGaussian() * 0.05;
-        for (int i = 0; i < 70; ++i) {
-            double d3 = this.xd * 0.5 + this.random.nextGaussian() * 0.15 + d;
-            double d4 = this.zd * 0.5 + this.random.nextGaussian() * 0.15 + d2;
-            double d5 = this.yd * 0.5 + this.random.nextDouble() * 0.5;
-            this.createParticle(this.x, this.y, this.z, d3, d5, d4, colours, fadeColours, trail, twinkle);
+    public void createParticleBurst(int[] $$0, int[] $$1, boolean $$2, boolean $$3) {
+        double $$4 = this.random.nextGaussian() * 0.05;
+        double $$5 = this.random.nextGaussian() * 0.05;
+        for (int $$6 = 0; $$6 < 70; ++$$6) {
+            double $$7 = this.xd * 0.5 + this.random.nextGaussian() * 0.15 + $$4;
+            double $$8 = this.zd * 0.5 + this.random.nextGaussian() * 0.15 + $$5;
+            double $$9 = this.yd * 0.5 + this.random.nextDouble() * 0.5;
+            this.createParticle(this.x, this.y, this.z, $$7, $$9, $$8, $$0, $$1, $$2, $$3);
         }
     }
 }