--- net/minecraft/client/particle/FireworkParticles$OverlayParticle.java
+++ net/minecraft/client/particle/FireworkParticles$OverlayParticle.java
@@ -15,13 +15,13 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_TRANSLUCENT;
     }
 
-    public void render(com.mojang.blaze3d.vertex.VertexConsumer buffer, net.minecraft.client.Camera renderInfo, float partialTicks) {
-        this.setAlpha(0.6f - ((float)this.age + partialTicks - 1.0f) * 0.25f * 0.5f);
-        super.render(buffer, renderInfo, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.VertexConsumer $$0, net.minecraft.client.Camera $$1, float $$2) {
+        this.setAlpha(0.6f - ((float)this.age + $$2 - 1.0f) * 0.25f * 0.5f);
+        super.render($$0, $$1, $$2);
     }
 
-    public float getQuadSize(float scaleFactor) {
-        return 7.1f * net.minecraft.util.Mth.sin((float)(((float)this.age + scaleFactor - 1.0f) * 0.25f * (float)java.lang.Math.PI));
+    public float getQuadSize(float $$0) {
+        return 7.1f * net.minecraft.util.Mth.sin((float)(((float)this.age + $$0 - 1.0f) * 0.25f * (float)java.lang.Math.PI));
     }
 }
 