--- net/minecraft/client/particle/FlameParticle.java
+++ net/minecraft/client/particle/FlameParticle.java
@@ -14,26 +14,26 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_OPAQUE;
     }
 
-    public void move(double x, double y, double z) {
-        this.setBoundingBox(this.getBoundingBox().move(x, y, z));
+    public void move(double $$0, double $$1, double $$2) {
+        this.setBoundingBox(this.getBoundingBox().move($$0, $$1, $$2));
         this.setLocationFromBoundingbox();
     }
 
-    public float getQuadSize(float scaleFactor) {
-        float f = ((float)this.age + scaleFactor) / (float)this.lifetime;
-        return this.quadSize * (1.0f - f * f * 0.5f);
+    public float getQuadSize(float $$0) {
+        float $$1 = ((float)this.age + $$0) / (float)this.lifetime;
+        return this.quadSize * (1.0f - $$1 * $$1 * 0.5f);
     }
 
-    public int getLightColor(float partialTick) {
-        float f = ((float)this.age + partialTick) / (float)this.lifetime;
-        f = net.minecraft.util.Mth.clamp((float)f, (float)0.0f, (float)1.0f);
-        int n = super.getLightColor(partialTick);
-        int n2 = n & 0xFF;
-        int n3 = n >> 16 & 0xFF;
-        if ((n2 += (int)(f * 15.0f * 16.0f)) > 240) {
-            n2 = 240;
+    public int getLightColor(float $$0) {
+        float $$1 = ((float)this.age + $$0) / (float)this.lifetime;
+        $$1 = net.minecraft.util.Mth.clamp((float)$$1, (float)0.0f, (float)1.0f);
+        int $$2 = super.getLightColor($$0);
+        int $$3 = $$2 & 0xFF;
+        int $$4 = $$2 >> 16 & 0xFF;
+        if (($$3 += (int)($$1 * 15.0f * 16.0f)) > 240) {
+            $$3 = 240;
         }
-        return n2 | n3 << 16;
+        return $$3 | $$4 << 16;
     }
 }
 