--- net/minecraft/client/particle/HeartParticle$AngryVillagerProvider.java
+++ net/minecraft/client/particle/HeartParticle$AngryVillagerProvider.java
@@ -8,15 +8,15 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.HeartParticle$AngryVillagerProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.HeartParticle$AngryVillagerProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.HeartParticle heartParticle = new net.minecraft.client.particle.HeartParticle(level, x, y + 0.5, z);
-        heartParticle.pickSprite(this.sprite);
-        heartParticle.setColor(1.0f, 1.0f, 1.0f);
-        return heartParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.HeartParticle $$8 = new net.minecraft.client.particle.HeartParticle($$1, $$2, $$3 + 0.5, $$4);
+        $$8.pickSprite(this.sprite);
+        $$8.setColor(1.0f, 1.0f, 1.0f);
+        return $$8;
     }
 }
 