--- net/minecraft/client/particle/HugeExplosionParticle.java
+++ net/minecraft/client/particle/HugeExplosionParticle.java
@@ -8,19 +8,19 @@
  {
     public net.minecraft.client.particle.SpriteSet sprites;
 
-    public net.minecraft.client.particle.HugeExplosionParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double quadSizeMulitiplier, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, 0.0, 0.0, 0.0);
-        float f;
+    public net.minecraft.client.particle.HugeExplosionParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, net.minecraft.client.particle.SpriteSet $$5) {
+        super($$0, $$1, $$2, $$3, 0.0, 0.0, 0.0);
+        float $$6;
         this.lifetime = 6 + this.random.nextInt(4);
-        this.rCol = f = this.random.nextFloat() * 0.6f + 0.4f;
-        this.gCol = f;
-        this.bCol = f;
-        this.quadSize = 2.0f * (1.0f - (float)quadSizeMulitiplier * 0.5f);
-        this.sprites = sprites;
-        this.setSpriteFromAge(sprites);
+        this.rCol = $$6 = this.random.nextFloat() * 0.6f + 0.4f;
+        this.gCol = $$6;
+        this.bCol = $$6;
+        this.quadSize = 2.0f * (1.0f - (float)$$4 * 0.5f);
+        this.sprites = $$5;
+        this.setSpriteFromAge($$5);
     }
 
-    public int getLightColor(float partialTick) {
+    public int getLightColor(float $$0) {
         return 0xF000F0;
     }
 