--- net/minecraft/client/particle/HugeExplosionSeedParticle$Provider.java
+++ net/minecraft/client/particle/HugeExplosionSeedParticle$Provider.java
@@ -9,8 +9,8 @@
     public net.minecraft.client.particle.HugeExplosionSeedParticle$Provider() {
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        return new net.minecraft.client.particle.HugeExplosionSeedParticle(level, x, y, z);
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        return new net.minecraft.client.particle.HugeExplosionSeedParticle($$1, $$2, $$3, $$4);
     }
 }
 