--- net/minecraft/client/particle/ItemPickupParticle.java
+++ net/minecraft/client/particle/ItemPickupParticle.java
@@ -13,42 +13,42 @@
     public int life;
     public net.minecraft.client.renderer.entity.EntityRenderDispatcher entityRenderDispatcher;
 
-    public net.minecraft.client.particle.ItemPickupParticle(net.minecraft.client.renderer.entity.EntityRenderDispatcher entityRenderDispatcher, net.minecraft.client.renderer.RenderBuffers buffers, net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.world.entity.Entity itemEntity, net.minecraft.world.entity.Entity target) {
-        this(entityRenderDispatcher, buffers, level, itemEntity, target, itemEntity.getDeltaMovement());
+    public net.minecraft.client.particle.ItemPickupParticle(net.minecraft.client.renderer.entity.EntityRenderDispatcher $$0, net.minecraft.client.renderer.RenderBuffers $$1, net.minecraft.client.multiplayer.ClientLevel $$2, net.minecraft.world.entity.Entity $$3, net.minecraft.world.entity.Entity $$4) {
+        this($$0, $$1, $$2, $$3, $$4, $$3.getDeltaMovement());
     }
 
-    public net.minecraft.client.particle.ItemPickupParticle(net.minecraft.client.renderer.entity.EntityRenderDispatcher entityRenderDispatcher, net.minecraft.client.renderer.RenderBuffers buffers, net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.world.entity.Entity itemEntity, net.minecraft.world.entity.Entity target, net.minecraft.world.phys.Vec3 speedVector) {
-        super(level, itemEntity.getX(), itemEntity.getY(), itemEntity.getZ(), speedVector.x, speedVector.y, speedVector.z);
-        this.renderBuffers = buffers;
-        this.itemEntity = this.getSafeCopy(itemEntity);
-        this.target = target;
-        this.entityRenderDispatcher = entityRenderDispatcher;
+    public net.minecraft.client.particle.ItemPickupParticle(net.minecraft.client.renderer.entity.EntityRenderDispatcher $$0, net.minecraft.client.renderer.RenderBuffers $$1, net.minecraft.client.multiplayer.ClientLevel $$2, net.minecraft.world.entity.Entity $$3, net.minecraft.world.entity.Entity $$4, net.minecraft.world.phys.Vec3 $$5) {
+        super($$2, $$3.getX(), $$3.getY(), $$3.getZ(), $$5.x, $$5.y, $$5.z);
+        this.renderBuffers = $$1;
+        this.itemEntity = this.getSafeCopy($$3);
+        this.target = $$4;
+        this.entityRenderDispatcher = $$0;
     }
 
-    public net.minecraft.world.entity.Entity getSafeCopy(net.minecraft.world.entity.Entity entity) {
-        if (!(entity instanceof net.minecraft.world.entity.item.ItemEntity)) {
-            return entity;
+    public net.minecraft.world.entity.Entity getSafeCopy(net.minecraft.world.entity.Entity $$0) {
+        if (!($$0 instanceof net.minecraft.world.entity.item.ItemEntity)) {
+            return $$0;
         }
-        return ((net.minecraft.world.entity.item.ItemEntity)entity).copy();
+        return ((net.minecraft.world.entity.item.ItemEntity)$$0).copy();
     }
 
     public net.minecraft.client.particle.ParticleRenderType getRenderType() {
         return net.minecraft.client.particle.ParticleRenderType.CUSTOM;
     }
 
-    public void render(com.mojang.blaze3d.vertex.VertexConsumer buffer, net.minecraft.client.Camera renderInfo, float partialTicks) {
-        float f = ((float)this.life + partialTicks) / 3.0f;
-        f *= f;
-        double d = net.minecraft.util.Mth.lerp((double)partialTicks, (double)this.target.xOld, (double)this.target.getX());
-        double d2 = net.minecraft.util.Mth.lerp((double)partialTicks, (double)this.target.yOld, (double)((this.target.getY() + this.target.getEyeY()) / 2.0));
-        double d3 = net.minecraft.util.Mth.lerp((double)partialTicks, (double)this.target.zOld, (double)this.target.getZ());
-        double d4 = net.minecraft.util.Mth.lerp((double)f, (double)this.itemEntity.getX(), (double)d);
-        double d5 = net.minecraft.util.Mth.lerp((double)f, (double)this.itemEntity.getY(), (double)d2);
-        double d6 = net.minecraft.util.Mth.lerp((double)f, (double)this.itemEntity.getZ(), (double)d3);
-        net.minecraft.client.renderer.MultiBufferSource$BufferSource bufferSource = this.renderBuffers.bufferSource();
-        net.minecraft.world.phys.Vec3 vec3 = renderInfo.getPosition();
-        this.entityRenderDispatcher.render(this.itemEntity, d4 - vec3.x(), d5 - vec3.y(), d6 - vec3.z(), this.itemEntity.getYRot(), partialTicks, new com.mojang.blaze3d.vertex.PoseStack(), (net.minecraft.client.renderer.MultiBufferSource)bufferSource, this.entityRenderDispatcher.getPackedLightCoords(this.itemEntity, partialTicks));
-        bufferSource.endBatch();
+    public void render(com.mojang.blaze3d.vertex.VertexConsumer $$0, net.minecraft.client.Camera $$1, float $$2) {
+        float $$3 = ((float)this.life + $$2) / 3.0f;
+        $$3 *= $$3;
+        double $$4 = net.minecraft.util.Mth.lerp((double)$$2, (double)this.target.xOld, (double)this.target.getX());
+        double $$5 = net.minecraft.util.Mth.lerp((double)$$2, (double)this.target.yOld, (double)((this.target.getY() + this.target.getEyeY()) / 2.0));
+        double $$6 = net.minecraft.util.Mth.lerp((double)$$2, (double)this.target.zOld, (double)this.target.getZ());
+        double $$7 = net.minecraft.util.Mth.lerp((double)$$3, (double)this.itemEntity.getX(), (double)$$4);
+        double $$8 = net.minecraft.util.Mth.lerp((double)$$3, (double)this.itemEntity.getY(), (double)$$5);
+        double $$9 = net.minecraft.util.Mth.lerp((double)$$3, (double)this.itemEntity.getZ(), (double)$$6);
+        net.minecraft.client.renderer.MultiBufferSource$BufferSource $$10 = this.renderBuffers.bufferSource();
+        net.minecraft.world.phys.Vec3 $$11 = $$1.getPosition();
+        this.entityRenderDispatcher.render(this.itemEntity, $$7 - $$11.x(), $$8 - $$11.y(), $$9 - $$11.z(), this.itemEntity.getYRot(), $$2, new com.mojang.blaze3d.vertex.PoseStack(), (net.minecraft.client.renderer.MultiBufferSource)$$10, this.entityRenderDispatcher.getPackedLightCoords(this.itemEntity, $$2));
+        $$10.endBatch();
     }
 
     public void tick() {