--- net/minecraft/client/particle/LavaParticle.java
+++ net/minecraft/client/particle/LavaParticle.java
@@ -22,16 +22,16 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_OPAQUE;
     }
 
-    public int getLightColor(float partialTick) {
-        int n = super.getLightColor(partialTick);
-        int n2 = 240;
-        int n3 = n >> 16 & 0xFF;
-        return 0xF0 | n3 << 16;
+    public int getLightColor(float $$0) {
+        int $$1 = super.getLightColor($$0);
+        int $$2 = 240;
+        int $$3 = $$1 >> 16 & 0xFF;
+        return 0xF0 | $$3 << 16;
     }
 
-    public float getQuadSize(float scaleFactor) {
-        float f = ((float)this.age + scaleFactor) / (float)this.lifetime;
-        return this.quadSize * (1.0f - f * f);
+    public float getQuadSize(float $$0) {
+        float $$1 = ((float)this.age + $$0) / (float)this.lifetime;
+        return this.quadSize * (1.0f - $$1 * $$1);
     }
 
     public void tick() {