--- net/minecraft/client/particle/MobAppearanceParticle.java
+++ net/minecraft/client/particle/MobAppearanceParticle.java
@@ -21,18 +21,18 @@
         return net.minecraft.client.particle.ParticleRenderType.CUSTOM;
     }
 
-    public void render(com.mojang.blaze3d.vertex.VertexConsumer buffer, net.minecraft.client.Camera renderInfo, float partialTicks) {
-        float f = ((float)this.age + partialTicks) / (float)this.lifetime;
-        float f2 = 0.05f + 0.5f * net.minecraft.util.Mth.sin((float)(f * (float)java.lang.Math.PI));
-        com.mojang.blaze3d.vertex.PoseStack poseStack = new com.mojang.blaze3d.vertex.PoseStack();
-        poseStack.mulPose(renderInfo.rotation());
-        poseStack.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(150.0f * f - 60.0f));
-        poseStack.scale(-1.0f, -1.0f, 1.0f);
-        poseStack.translate(0.0, (double)-1.101f, 1.5);
-        net.minecraft.client.renderer.MultiBufferSource$BufferSource bufferSource = net.minecraft.client.Minecraft.getInstance().renderBuffers().bufferSource();
-        com.mojang.blaze3d.vertex.VertexConsumer vertexConsumer = bufferSource.getBuffer(this.renderType);
-        this.model.renderToBuffer(poseStack, vertexConsumer, 0xF000F0, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, f2);
-        bufferSource.endBatch();
+    public void render(com.mojang.blaze3d.vertex.VertexConsumer $$0, net.minecraft.client.Camera $$1, float $$2) {
+        float $$3 = ((float)this.age + $$2) / (float)this.lifetime;
+        float $$4 = 0.05f + 0.5f * net.minecraft.util.Mth.sin((float)($$3 * (float)java.lang.Math.PI));
+        com.mojang.blaze3d.vertex.PoseStack $$5 = new com.mojang.blaze3d.vertex.PoseStack();
+        $$5.mulPose($$1.rotation());
+        $$5.mulPose(com.mojang.math.Vector3f.XP.rotationDegrees(150.0f * $$3 - 60.0f));
+        $$5.scale(-1.0f, -1.0f, 1.0f);
+        $$5.translate(0.0, (double)-1.101f, 1.5);
+        net.minecraft.client.renderer.MultiBufferSource$BufferSource $$6 = net.minecraft.client.Minecraft.getInstance().renderBuffers().bufferSource();
+        com.mojang.blaze3d.vertex.VertexConsumer $$7 = $$6.getBuffer(this.renderType);
+        this.model.renderToBuffer($$5, $$7, 0xF000F0, net.minecraft.client.renderer.texture.OverlayTexture.NO_OVERLAY, 1.0f, 1.0f, 1.0f, $$4);
+        $$6.endBatch();
     }
 }
 