--- net/minecraft/client/particle/NoteParticle.java
+++ net/minecraft/client/particle/NoteParticle.java
@@ -6,17 +6,17 @@
 public class net.minecraft.client.particle.NoteParticle
 extends net.minecraft.client.particle.TextureSheetParticle
  {
-    public net.minecraft.client.particle.NoteParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double d) {
-        super(level, x, y, z, 0.0, 0.0, 0.0);
+    public net.minecraft.client.particle.NoteParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4) {
+        super($$0, $$1, $$2, $$3, 0.0, 0.0, 0.0);
         this.friction = 0.66f;
         this.speedUpWhenYMotionIsBlocked = true;
         this.xd *= (double)0.01f;
         this.yd *= (double)0.01f;
         this.zd *= (double)0.01f;
         this.yd += 0.2;
-        this.rCol = java.lang.Math.max((float)0.0f, (float)(net.minecraft.util.Mth.sin((float)(((float)d + 0.0f) * ((float)java.lang.Math.PI * 2))) * 0.65f + 0.35f));
-        this.gCol = java.lang.Math.max((float)0.0f, (float)(net.minecraft.util.Mth.sin((float)(((float)d + 0.33333334f) * ((float)java.lang.Math.PI * 2))) * 0.65f + 0.35f));
-        this.bCol = java.lang.Math.max((float)0.0f, (float)(net.minecraft.util.Mth.sin((float)(((float)d + 0.6666667f) * ((float)java.lang.Math.PI * 2))) * 0.65f + 0.35f));
+        this.rCol = java.lang.Math.max((float)0.0f, (float)(net.minecraft.util.Mth.sin((float)(((float)$$4 + 0.0f) * ((float)java.lang.Math.PI * 2))) * 0.65f + 0.35f));
+        this.gCol = java.lang.Math.max((float)0.0f, (float)(net.minecraft.util.Mth.sin((float)(((float)$$4 + 0.33333334f) * ((float)java.lang.Math.PI * 2))) * 0.65f + 0.35f));
+        this.bCol = java.lang.Math.max((float)0.0f, (float)(net.minecraft.util.Mth.sin((float)(((float)$$4 + 0.6666667f) * ((float)java.lang.Math.PI * 2))) * 0.65f + 0.35f));
         this.quadSize *= 1.5f;
         this.lifetime = 6;
     }
@@ -25,8 +25,8 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_OPAQUE;
     }
 
-    public float getQuadSize(float scaleFactor) {
-        return this.quadSize * net.minecraft.util.Mth.clamp((float)(((float)this.age + scaleFactor) / (float)this.lifetime * 32.0f), (float)0.0f, (float)1.0f);
+    public float getQuadSize(float $$0) {
+        return this.quadSize * net.minecraft.util.Mth.clamp((float)(((float)this.age + $$0) / (float)this.lifetime * 32.0f), (float)0.0f, (float)1.0f);
     }
 }
 