--- net/minecraft/client/particle/ParticleEngine$1ParticleDefinition.java
+++ net/minecraft/client/particle/ParticleEngine$1ParticleDefinition.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.particle;
+
+public class net.minecraft.client.particle.ParticleEngine$1ParticleDefinition
+extends java.lang.Record
+ {
+    public net.minecraft.resources.ResourceLocation id;
+    public java.util.Optional<java.util.List<net.minecraft.resources.ResourceLocation>> sprites;
+
+    public net.minecraft.client.particle.ParticleEngine$1ParticleDefinition(net.minecraft.resources.ResourceLocation $$0, java.util.Optional<java.util.List<net.minecraft.resources.ResourceLocation>> $$1) {
+        this.id = $$0;
+        this.sprites = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.particle.ParticleEngine$1ParticleDefinition.class, "id;sprites", "id", "sprites"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.particle.ParticleEngine$1ParticleDefinition.class, "id;sprites", "id", "sprites"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.particle.ParticleEngine$1ParticleDefinition.class, "id;sprites", "id", "sprites"}, this, $$0);
+    }
+
+    public net.minecraft.resources.ResourceLocation id() {
+        return this.id;
+    }
+
+    public java.util.Optional<java.util.List<net.minecraft.resources.ResourceLocation>> sprites() {
+        return this.sprites;
+    }
+}
+