--- net/minecraft/client/particle/ParticleRenderType$6.java
+++ net/minecraft/client/particle/ParticleRenderType$6.java
@@ -9,10 +9,10 @@
     public net.minecraft.client.particle.ParticleRenderType$6() {
     }
 
-    public void begin(com.mojang.blaze3d.vertex.BufferBuilder bufferBuilder, net.minecraft.client.renderer.texture.TextureManager textureManager) {
+    public void begin(com.mojang.blaze3d.vertex.BufferBuilder $$0, net.minecraft.client.renderer.texture.TextureManager $$1) {
     }
 
-    public void end(com.mojang.blaze3d.vertex.Tesselator tesselator) {
+    public void end(com.mojang.blaze3d.vertex.Tesselator $$0) {
     }
 
     public java.lang.String toString() {