--- net/minecraft/client/particle/ParticleDescription.java
+++ net/minecraft/client/particle/ParticleDescription.java
@@ -8,8 +8,8 @@
     @javax.annotation.Nullable
     public java.util.List<net.minecraft.resources.ResourceLocation> textures;
 
-    public net.minecraft.client.particle.ParticleDescription(@javax.annotation.Nullable java.util.List<net.minecraft.resources.ResourceLocation> textures) {
-        this.textures = textures;
+    public net.minecraft.client.particle.ParticleDescription(@javax.annotation.Nullable java.util.List<net.minecraft.resources.ResourceLocation> $$0) {
+        this.textures = $$0;
     }
 
     @javax.annotation.Nullable
@@ -17,10 +17,15 @@
         return this.textures;
     }
 
-    public static net.minecraft.client.particle.ParticleDescription fromJson(com.google.gson.JsonObject json) {
-        com.google.gson.JsonArray jsonArray = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)json, (java.lang.String)"textures", null);
-        java.util.List list = jsonArray != null ? (java.util.List)com.google.common.collect.Streams.stream((java.lang.Iterable)jsonArray).map(jsonElement -> net.minecraft.util.GsonHelper.convertToString((com.google.gson.JsonElement)jsonElement, (java.lang.String)"texture")).map(net.minecraft.resources.ResourceLocation::new).collect(com.google.common.collect.ImmutableList.toImmutableList()) : null;
-        return new net.minecraft.client.particle.ParticleDescription(list);
+    public static net.minecraft.client.particle.ParticleDescription fromJson(com.google.gson.JsonObject $$02) {
+        java.util.List $$3;
+        com.google.gson.JsonArray $$1 = net.minecraft.util.GsonHelper.getAsJsonArray((com.google.gson.JsonObject)$$02, (java.lang.String)"textures", null);
+        if ($$1 != null) {
+            java.util.List $$2 = (java.util.List)com.google.common.collect.Streams.stream((java.lang.Iterable)$$1).map($$0 -> net.minecraft.util.GsonHelper.convertToString((com.google.gson.JsonElement)$$0, (java.lang.String)"texture")).map(net.minecraft.resources.ResourceLocation::new).collect(com.google.common.collect.ImmutableList.toImmutableList());
+        } else {
+            $$3 = null;
+        }
+        return new net.minecraft.client.particle.ParticleDescription($$3);
     }
 }
 