--- net/minecraft/client/particle/PlayerCloudParticle$Provider.java
+++ net/minecraft/client/particle/PlayerCloudParticle$Provider.java
@@ -8,12 +8,12 @@
  {
     public net.minecraft.client.particle.SpriteSet sprites;
 
-    public net.minecraft.client.particle.PlayerCloudParticle$Provider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprites = sprites;
+    public net.minecraft.client.particle.PlayerCloudParticle$Provider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprites = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        return new net.minecraft.client.particle.PlayerCloudParticle(level, x, y, z, xSpeed, ySpeed, zSpeed, this.sprites);
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        return new net.minecraft.client.particle.PlayerCloudParticle($$1, $$2, $$3, $$4, $$5, $$6, $$7, this.sprites);
     }
 }
 