--- net/minecraft/client/particle/ReversePortalParticle.java
+++ net/minecraft/client/particle/ReversePortalParticle.java
@@ -12,9 +12,9 @@
         this.lifetime = (int)(java.lang.Math.random() * 2.0) + 60;
     }
 
-    public float getQuadSize(float scaleFactor) {
-        float f = 1.0f - ((float)this.age + scaleFactor) / ((float)this.lifetime * 1.5f);
-        return this.quadSize * f;
+    public float getQuadSize(float $$0) {
+        float $$1 = 1.0f - ((float)this.age + $$0) / ((float)this.lifetime * 1.5f);
+        return this.quadSize * $$1;
     }
 
     public void tick() {