--- net/minecraft/client/particle/SimpleAnimatedParticle.java
+++ net/minecraft/client/particle/SimpleAnimatedParticle.java
@@ -12,25 +12,25 @@
     public float fadeB;
     public boolean hasFade;
 
-    public net.minecraft.client.particle.SimpleAnimatedParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, net.minecraft.client.particle.SpriteSet sprites, float gravity) {
-        super(level, x, y, z);
+    public net.minecraft.client.particle.SimpleAnimatedParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, net.minecraft.client.particle.SpriteSet $$4, float $$5) {
+        super($$0, $$1, $$2, $$3);
         this.friction = 0.91f;
-        this.gravity = gravity;
-        this.sprites = sprites;
+        this.gravity = $$5;
+        this.sprites = $$4;
     }
 
-    public void setColor(int color) {
-        float f = (float)((color & 0xFF0000) >> 16) / 255.0f;
-        float f2 = (float)((color & 0xFF00) >> 8) / 255.0f;
-        float f3 = (float)((color & 0xFF) >> 0) / 255.0f;
-        float f4 = 1.0f;
-        this.setColor(f * 1.0f, f2 * 1.0f, f3 * 1.0f);
+    public void setColor(int $$0) {
+        float $$1 = (float)(($$0 & 0xFF0000) >> 16) / 255.0f;
+        float $$2 = (float)(($$0 & 0xFF00) >> 8) / 255.0f;
+        float $$3 = (float)(($$0 & 0xFF) >> 0) / 255.0f;
+        float $$4 = 1.0f;
+        this.setColor($$1 * 1.0f, $$2 * 1.0f, $$3 * 1.0f);
     }
 
-    public void setFadeColor(int rgb) {
-        this.fadeR = (float)((rgb & 0xFF0000) >> 16) / 255.0f;
-        this.fadeG = (float)((rgb & 0xFF00) >> 8) / 255.0f;
-        this.fadeB = (float)((rgb & 0xFF) >> 0) / 255.0f;
+    public void setFadeColor(int $$0) {
+        this.fadeR = (float)(($$0 & 0xFF0000) >> 16) / 255.0f;
+        this.fadeG = (float)(($$0 & 0xFF00) >> 8) / 255.0f;
+        this.fadeB = (float)(($$0 & 0xFF) >> 0) / 255.0f;
         this.hasFade = true;
     }
 
@@ -51,7 +51,7 @@
         }
     }
 
-    public int getLightColor(float partialTick) {
+    public int getLightColor(float $$0) {
         return 0xF000F0;
     }
 }