--- net/minecraft/client/particle/SoulParticle.java
+++ net/minecraft/client/particle/SoulParticle.java
@@ -9,11 +9,11 @@
     public net.minecraft.client.particle.SpriteSet sprites;
     public boolean isGlowing;
 
-    public net.minecraft.client.particle.SoulParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, xSpeed, ySpeed, zSpeed);
-        this.sprites = sprites;
+    public net.minecraft.client.particle.SoulParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.client.particle.SpriteSet $$7) {
+        super($$0, $$1, $$2, $$3, $$4, $$5, $$6);
+        this.sprites = $$7;
         this.scale(1.5f);
-        this.setSpriteFromAge(sprites);
+        this.setSpriteFromAge($$7);
     }
 
     public int getLightColor(float $$0) {