--- net/minecraft/client/particle/SoulParticle$Provider.java
+++ net/minecraft/client/particle/SoulParticle$Provider.java
@@ -8,14 +8,14 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.SoulParticle$Provider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.SoulParticle$Provider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.SoulParticle soulParticle = new net.minecraft.client.particle.SoulParticle(level, x, y, z, xSpeed, ySpeed, zSpeed, this.sprite);
-        soulParticle.setAlpha(1.0f);
-        return soulParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.SoulParticle $$8 = new net.minecraft.client.particle.SoulParticle($$1, $$2, $$3, $$4, $$5, $$6, $$7, this.sprite);
+        $$8.setAlpha(1.0f);
+        return $$8;
     }
 }
 