--- net/minecraft/client/particle/SpellParticle.java
+++ net/minecraft/client/particle/SpellParticle.java
@@ -9,21 +9,21 @@
     public static net.minecraft.util.RandomSource RANDOM;
     public net.minecraft.client.particle.SpriteSet sprites;
 
-    public net.minecraft.client.particle.SpellParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, 0.5 - RANDOM.nextDouble(), ySpeed, 0.5 - RANDOM.nextDouble());
+    public net.minecraft.client.particle.SpellParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.client.particle.SpriteSet $$7) {
+        super($$0, $$1, $$2, $$3, 0.5 - RANDOM.nextDouble(), $$5, 0.5 - RANDOM.nextDouble());
         this.friction = 0.96f;
         this.gravity = -0.1f;
         this.speedUpWhenYMotionIsBlocked = true;
-        this.sprites = sprites;
+        this.sprites = $$7;
         this.yd *= (double)0.2f;
-        if (xSpeed == 0.0 && zSpeed == 0.0) {
+        if ($$4 == 0.0 && $$6 == 0.0) {
             this.xd *= (double)0.1f;
             this.zd *= (double)0.1f;
         }
         this.quadSize *= 0.75f;
         this.lifetime = (int)(8.0 / (java.lang.Math.random() * 0.8 + 0.2));
         this.hasPhysics = false;
-        this.setSpriteFromAge(sprites);
+        this.setSpriteFromAge($$7);
         if (this.isCloseToScopingPlayer()) {
             this.setAlpha(0.0f);
         }