--- net/minecraft/client/particle/SpellParticle$WitchProvider.java
+++ net/minecraft/client/particle/SpellParticle$WitchProvider.java
@@ -8,15 +8,15 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.SpellParticle$WitchProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.SpellParticle$WitchProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.SpellParticle spellParticle = new net.minecraft.client.particle.SpellParticle(level, x, y, z, xSpeed, ySpeed, zSpeed, this.sprite);
-        float f = level.random.nextFloat() * 0.5f + 0.35f;
-        spellParticle.setColor(1.0f * f, 0.0f * f, 1.0f * f);
-        return spellParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.SpellParticle $$8 = new net.minecraft.client.particle.SpellParticle($$1, $$2, $$3, $$4, $$5, $$6, $$7, this.sprite);
+        float $$9 = $$1.random.nextFloat() * 0.5f + 0.35f;
+        $$8.setColor(1.0f * $$9, 0.0f * $$9, 1.0f * $$9);
+        return $$8;
     }
 }
 