--- net/minecraft/client/particle/SpellParticle$AmbientMobProvider.java
+++ net/minecraft/client/particle/SpellParticle$AmbientMobProvider.java
@@ -8,15 +8,15 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.SpellParticle$AmbientMobProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.SpellParticle$AmbientMobProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.SpellParticle spellParticle = new net.minecraft.client.particle.SpellParticle(level, x, y, z, xSpeed, ySpeed, zSpeed, this.sprite);
-        spellParticle.setAlpha(0.15f);
-        spellParticle.setColor((float)xSpeed, (float)ySpeed, (float)zSpeed);
-        return spellParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.SpellParticle $$8 = new net.minecraft.client.particle.SpellParticle($$1, $$2, $$3, $$4, $$5, $$6, $$7, this.sprite);
+        $$8.setAlpha(0.15f);
+        $$8.setColor((float)$$5, (float)$$6, (float)$$7);
+        return $$8;
     }
 }
 