--- net/minecraft/client/particle/SuspendedTownParticle$ComposterFillProvider.java
+++ net/minecraft/client/particle/SuspendedTownParticle$ComposterFillProvider.java
@@ -8,16 +8,16 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.SuspendedTownParticle$ComposterFillProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.SuspendedTownParticle$ComposterFillProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.SuspendedTownParticle suspendedTownParticle = new net.minecraft.client.particle.SuspendedTownParticle(level, x, y, z, xSpeed, ySpeed, zSpeed);
-        suspendedTownParticle.pickSprite(this.sprite);
-        suspendedTownParticle.setColor(1.0f, 1.0f, 1.0f);
-        suspendedTownParticle.setLifetime(3 + level.getRandom().nextInt(5));
-        return suspendedTownParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.SuspendedTownParticle $$8 = new net.minecraft.client.particle.SuspendedTownParticle($$1, $$2, $$3, $$4, $$5, $$6, $$7);
+        $$8.pickSprite(this.sprite);
+        $$8.setColor(1.0f, 1.0f, 1.0f);
+        $$8.setLifetime(3 + $$1.getRandom().nextInt(5));
+        return $$8;
     }
 }
 