--- net/minecraft/client/particle/SuspendedParticle$CrimsonSporeProvider.java
+++ net/minecraft/client/particle/SuspendedParticle$CrimsonSporeProvider.java
@@ -8,18 +8,18 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.SuspendedParticle$CrimsonSporeProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.SuspendedParticle$CrimsonSporeProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.util.RandomSource randomSource = level.random;
-        double d = randomSource.nextGaussian() * (double)1.0E-6f;
-        double d2 = randomSource.nextGaussian() * (double)1.0E-4f;
-        double d3 = randomSource.nextGaussian() * (double)1.0E-6f;
-        net.minecraft.client.particle.SuspendedParticle suspendedParticle = new net.minecraft.client.particle.SuspendedParticle(level, this.sprite, x, y, z, d, d2, d3);
-        suspendedParticle.setColor(0.9f, 0.4f, 0.5f);
-        return suspendedParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.util.RandomSource $$8 = $$1.random;
+        double $$9 = $$8.nextGaussian() * (double)1.0E-6f;
+        double $$10 = $$8.nextGaussian() * (double)1.0E-4f;
+        double $$11 = $$8.nextGaussian() * (double)1.0E-6f;
+        net.minecraft.client.particle.SuspendedParticle $$12 = new net.minecraft.client.particle.SuspendedParticle($$1, this.sprite, $$2, $$3, $$4, $$9, $$10, $$11);
+        $$12.setColor(0.9f, 0.4f, 0.5f);
+        return $$12;
     }
 }
 