--- net/minecraft/client/particle/SuspendedTownParticle.java
+++ net/minecraft/client/particle/SuspendedTownParticle.java
@@ -24,8 +24,8 @@
         return net.minecraft.client.particle.ParticleRenderType.PARTICLE_SHEET_OPAQUE;
     }
 
-    public void move(double x, double y, double z) {
-        this.setBoundingBox(this.getBoundingBox().move(x, y, z));
+    public void move(double $$0, double $$1, double $$2) {
+        this.setBoundingBox(this.getBoundingBox().move($$0, $$1, $$2));
         this.setLocationFromBoundingbox();
     }
 