--- net/minecraft/client/particle/SuspendedParticle$WarpedSporeProvider.java
+++ net/minecraft/client/particle/SuspendedParticle$WarpedSporeProvider.java
@@ -8,16 +8,16 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.SuspendedParticle$WarpedSporeProvider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.SuspendedParticle$WarpedSporeProvider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        double d = (double)level.random.nextFloat() * -1.9 * (double)level.random.nextFloat() * 0.1;
-        net.minecraft.client.particle.SuspendedParticle suspendedParticle = new net.minecraft.client.particle.SuspendedParticle(level, this.sprite, x, y, z, 0.0, d, 0.0);
-        suspendedParticle.setColor(0.1f, 0.1f, 0.3f);
-        suspendedParticle.setSize(0.001f, 0.001f);
-        return suspendedParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        double $$8 = (double)$$1.random.nextFloat() * -1.9 * (double)$$1.random.nextFloat() * 0.1;
+        net.minecraft.client.particle.SuspendedParticle $$9 = new net.minecraft.client.particle.SuspendedParticle($$1, this.sprite, $$2, $$3, $$4, 0.0, $$8, 0.0);
+        $$9.setColor(0.1f, 0.1f, 0.3f);
+        $$9.setSize(0.001f, 0.001f);
+        return $$9;
     }
 }
 