--- net/minecraft/client/particle/TerrainParticle.java
+++ net/minecraft/client/particle/TerrainParticle.java
@@ -10,8 +10,8 @@
     public float uo;
     public float vo;
 
-    public net.minecraft.client.particle.TerrainParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.world.level.block.state.BlockState state) {
-        this(level, x, y, z, xSpeed, ySpeed, zSpeed, state, new net.minecraft.core.BlockPos(x, y, z));
+    public net.minecraft.client.particle.TerrainParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.world.level.block.state.BlockState $$7) {
+        this($$0, $$1, $$2, $$3, $$4, $$5, $$6, $$7, new net.minecraft.core.BlockPos($$1, $$2, $$3));
     }
 
     public net.minecraft.client.particle.TerrainParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.world.level.block.state.BlockState $$7, net.minecraft.core.BlockPos $$8) {
@@ -53,12 +53,12 @@
         return this.sprite.getV((double)((this.vo + 1.0f) / 4.0f * 16.0f));
     }
 
-    public int getLightColor(float partialTick) {
-        int n = super.getLightColor(partialTick);
-        if (n == 0 && this.level.hasChunkAt(this.pos)) {
+    public int getLightColor(float $$0) {
+        int $$1 = super.getLightColor($$0);
+        if ($$1 == 0 && this.level.hasChunkAt(this.pos)) {
             return net.minecraft.client.renderer.LevelRenderer.getLightColor((net.minecraft.world.level.BlockAndTintGetter)this.level, (net.minecraft.core.BlockPos)this.pos);
         }
-        return n;
+        return $$1;
     }
 }
 