--- net/minecraft/client/particle/TerrainParticle$Provider.java
+++ net/minecraft/client/particle/TerrainParticle$Provider.java
@@ -9,12 +9,12 @@
     public net.minecraft.client.particle.TerrainParticle$Provider() {
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.BlockParticleOption type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.world.level.block.state.BlockState blockState = type.getState();
-        if (blockState.isAir() || blockState.is(net.minecraft.world.level.block.Blocks.MOVING_PISTON)) {
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.BlockParticleOption $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.world.level.block.state.BlockState $$8 = $$0.getState();
+        if ($$8.isAir() || $$8.is(net.minecraft.world.level.block.Blocks.MOVING_PISTON)) {
             return null;
         }
-        return new net.minecraft.client.particle.TerrainParticle(level, x, y, z, xSpeed, ySpeed, zSpeed, blockState);
+        return new net.minecraft.client.particle.TerrainParticle($$1, $$2, $$3, $$4, $$5, $$6, $$7, $$8);
     }
 }
 