--- net/minecraft/client/particle/TextureSheetParticle.java
+++ net/minecraft/client/particle/TextureSheetParticle.java
@@ -16,8 +16,8 @@
         super($$0, $$1, $$2, $$3, $$4, $$5, $$6);
     }
 
-    public void setSprite(net.minecraft.client.renderer.texture.TextureAtlasSprite sprite) {
-        this.sprite = sprite;
+    public void setSprite(net.minecraft.client.renderer.texture.TextureAtlasSprite $$0) {
+        this.sprite = $$0;
     }
 
     public float getU0() {
@@ -36,13 +36,13 @@
         return this.sprite.getV1();
     }
 
-    public void pickSprite(net.minecraft.client.particle.SpriteSet sprite) {
-        this.setSprite(sprite.get(this.random));
+    public void pickSprite(net.minecraft.client.particle.SpriteSet $$0) {
+        this.setSprite($$0.get(this.random));
     }
 
-    public void setSpriteFromAge(net.minecraft.client.particle.SpriteSet sprite) {
+    public void setSpriteFromAge(net.minecraft.client.particle.SpriteSet $$0) {
         if (!this.removed) {
-            this.setSprite(sprite.get(this.age, this.lifetime));
+            this.setSprite($$0.get(this.age, this.lifetime));
         }
     }
 }