--- net/minecraft/client/particle/TrackingEmitter.java
+++ net/minecraft/client/particle/TrackingEmitter.java
@@ -11,19 +11,19 @@
     public int lifeTime;
     public net.minecraft.core.particles.ParticleOptions particleType;
 
-    public net.minecraft.client.particle.TrackingEmitter(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.world.entity.Entity entity, net.minecraft.core.particles.ParticleOptions particleType) {
-        this(level, entity, particleType, 3);
+    public net.minecraft.client.particle.TrackingEmitter(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.world.entity.Entity $$1, net.minecraft.core.particles.ParticleOptions $$2) {
+        this($$0, $$1, $$2, 3);
     }
 
-    public net.minecraft.client.particle.TrackingEmitter(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.world.entity.Entity entity, net.minecraft.core.particles.ParticleOptions particleType, int lifetime) {
-        this(level, entity, particleType, lifetime, entity.getDeltaMovement());
+    public net.minecraft.client.particle.TrackingEmitter(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.world.entity.Entity $$1, net.minecraft.core.particles.ParticleOptions $$2, int $$3) {
+        this($$0, $$1, $$2, $$3, $$1.getDeltaMovement());
     }
 
-    public net.minecraft.client.particle.TrackingEmitter(net.minecraft.client.multiplayer.ClientLevel level, net.minecraft.world.entity.Entity entity, net.minecraft.core.particles.ParticleOptions particleType, int lifetime, net.minecraft.world.phys.Vec3 speedVector) {
-        super(level, entity.getX(), entity.getY(0.5), entity.getZ(), speedVector.x, speedVector.y, speedVector.z);
-        this.entity = entity;
-        this.lifeTime = lifetime;
-        this.particleType = particleType;
+    public net.minecraft.client.particle.TrackingEmitter(net.minecraft.client.multiplayer.ClientLevel $$0, net.minecraft.world.entity.Entity $$1, net.minecraft.core.particles.ParticleOptions $$2, int $$3, net.minecraft.world.phys.Vec3 $$4) {
+        super($$0, $$1.getX(), $$1.getY(0.5), $$1.getZ(), $$4.x, $$4.y, $$4.z);
+        this.entity = $$1;
+        this.lifeTime = $$3;
+        this.particleType = $$2;
         this.tick();
     }
 