--- net/minecraft/client/particle/WakeParticle.java
+++ net/minecraft/client/particle/WakeParticle.java
@@ -8,19 +8,19 @@
  {
     public net.minecraft.client.particle.SpriteSet sprites;
 
-    public net.minecraft.client.particle.WakeParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, 0.0, 0.0, 0.0);
-        this.sprites = sprites;
+    public net.minecraft.client.particle.WakeParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, net.minecraft.client.particle.SpriteSet $$7) {
+        super($$0, $$1, $$2, $$3, 0.0, 0.0, 0.0);
+        this.sprites = $$7;
         this.xd *= (double)0.3f;
         this.yd = java.lang.Math.random() * (double)0.2f + (double)0.1f;
         this.zd *= (double)0.3f;
         this.setSize(0.01f, 0.01f);
         this.lifetime = (int)(8.0 / (java.lang.Math.random() * 0.8 + 0.2));
-        this.setSpriteFromAge(sprites);
+        this.setSpriteFromAge($$7);
         this.gravity = 0.0f;
-        this.xd = xSpeed;
-        this.yd = ySpeed;
-        this.zd = zSpeed;
+        this.xd = $$4;
+        this.yd = $$5;
+        this.zd = $$6;
     }
 
     public net.minecraft.client.particle.ParticleRenderType getRenderType() {