--- net/minecraft/client/particle/WaterDropParticle$Provider.java
+++ net/minecraft/client/particle/WaterDropParticle$Provider.java
@@ -8,14 +8,14 @@
  {
     public net.minecraft.client.particle.SpriteSet sprite;
 
-    public net.minecraft.client.particle.WaterDropParticle$Provider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprite = sprites;
+    public net.minecraft.client.particle.WaterDropParticle$Provider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprite = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.client.particle.WaterDropParticle waterDropParticle = new net.minecraft.client.particle.WaterDropParticle(level, x, y, z);
-        waterDropParticle.pickSprite(this.sprite);
-        return waterDropParticle;
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.client.particle.WaterDropParticle $$8 = new net.minecraft.client.particle.WaterDropParticle($$1, $$2, $$3, $$4);
+        $$8.pickSprite(this.sprite);
+        return $$8;
     }
 }
 