--- net/minecraft/client/particle/WhiteAshParticle$Provider.java
+++ net/minecraft/client/particle/WhiteAshParticle$Provider.java
@@ -8,16 +8,16 @@
  {
     public net.minecraft.client.particle.SpriteSet sprites;
 
-    public net.minecraft.client.particle.WhiteAshParticle$Provider(net.minecraft.client.particle.SpriteSet sprites) {
-        this.sprites = sprites;
+    public net.minecraft.client.particle.WhiteAshParticle$Provider(net.minecraft.client.particle.SpriteSet $$0) {
+        this.sprites = $$0;
     }
 
-    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType type, net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed) {
-        net.minecraft.util.RandomSource randomSource = level.random;
-        double d = (double)randomSource.nextFloat() * -1.9 * (double)randomSource.nextFloat() * 0.1;
-        double d2 = (double)randomSource.nextFloat() * -0.5 * (double)randomSource.nextFloat() * 0.1 * 5.0;
-        double d3 = (double)randomSource.nextFloat() * -1.9 * (double)randomSource.nextFloat() * 0.1;
-        return new net.minecraft.client.particle.WhiteAshParticle(level, x, y, z, d, d2, d3, 1.0f, this.sprites);
+    public net.minecraft.client.particle.Particle createParticle(net.minecraft.core.particles.SimpleParticleType $$0, net.minecraft.client.multiplayer.ClientLevel $$1, double $$2, double $$3, double $$4, double $$5, double $$6, double $$7) {
+        net.minecraft.util.RandomSource $$8 = $$1.random;
+        double $$9 = (double)$$8.nextFloat() * -1.9 * (double)$$8.nextFloat() * 0.1;
+        double $$10 = (double)$$8.nextFloat() * -0.5 * (double)$$8.nextFloat() * 0.1 * 5.0;
+        double $$11 = (double)$$8.nextFloat() * -1.9 * (double)$$8.nextFloat() * 0.1;
+        return new net.minecraft.client.particle.WhiteAshParticle($$1, $$2, $$3, $$4, $$9, $$10, $$11, 1.0f, this.sprites);
     }
 }
 