--- net/minecraft/client/particle/WhiteAshParticle.java
+++ net/minecraft/client/particle/WhiteAshParticle.java
@@ -8,8 +8,8 @@
  {
     public static int COLOR_RGB24 = 12235202;
 
-    public net.minecraft.client.particle.WhiteAshParticle(net.minecraft.client.multiplayer.ClientLevel level, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed, float quadSizeMultiplier, net.minecraft.client.particle.SpriteSet sprites) {
-        super(level, x, y, z, 0.1f, -0.1f, 0.1f, xSpeed, ySpeed, zSpeed, quadSizeMultiplier, sprites, 0.0f, 20, 0.0125f, false);
+    public net.minecraft.client.particle.WhiteAshParticle(net.minecraft.client.multiplayer.ClientLevel $$0, double $$1, double $$2, double $$3, double $$4, double $$5, double $$6, float $$7, net.minecraft.client.particle.SpriteSet $$8) {
+        super($$0, $$1, $$2, $$3, 0.1f, -0.1f, 0.1f, $$4, $$5, $$6, $$7, $$8, 0.0f, 20, 0.0125f, false);
         this.rCol = 0.7294118f;
         this.gCol = 0.69411767f;
         this.bCol = 0.7607843f;