--- net/minecraft/world/food/FoodProperties.java
+++ net/minecraft/world/food/FoodProperties.java
@@ -12,13 +12,13 @@
     public boolean fastFood;
     public java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.world.effect.MobEffectInstance, java.lang.Float>> effects;
 
-    public net.minecraft.world.food.FoodProperties(int nutrition, float saturationModifier, boolean isMeat, boolean canAlwaysEat, boolean fastFood, java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.world.effect.MobEffectInstance, java.lang.Float>> effects) {
-        this.nutrition = nutrition;
-        this.saturationModifier = saturationModifier;
-        this.isMeat = isMeat;
-        this.canAlwaysEat = canAlwaysEat;
-        this.fastFood = fastFood;
-        this.effects = effects;
+    public net.minecraft.world.food.FoodProperties(int $$0, float $$1, boolean $$2, boolean $$3, boolean $$4, java.util.List<com.mojang.datafixers.util.Pair<net.minecraft.world.effect.MobEffectInstance, java.lang.Float>> $$5) {
+        this.nutrition = $$0;
+        this.saturationModifier = $$1;
+        this.isMeat = $$2;
+        this.canAlwaysEat = $$3;
+        this.fastFood = $$4;
+        this.effects = $$5;
     }
 
     public int getNutrition() {