--- net/minecraft/world/entity/ai/Brain$Provider.java
+++ net/minecraft/world/entity/ai/Brain$Provider.java
@@ -15,8 +15,8 @@
         this.codec = net.minecraft.world.entity.ai.Brain.codec($$0, $$1);
     }
 
-    public net.minecraft.world.entity.ai.Brain<E> makeBrain(com.mojang.serialization.Dynamic<?> ops) {
-        return (net.minecraft.world.entity.ai.Brain)this.codec.parse(ops).resultOrPartial(arg_0 -> ((org.slf4j.Logger)net.minecraft.world.entity.ai.Brain.LOGGER).error(arg_0)).orElseGet(() -> new net.minecraft.world.entity.ai.Brain(this.memoryTypes, this.sensorTypes, com.google.common.collect.ImmutableList.of(), () -> this.codec));
+    public net.minecraft.world.entity.ai.Brain<E> makeBrain(com.mojang.serialization.Dynamic<?> $$0) {
+        return (net.minecraft.world.entity.ai.Brain)this.codec.parse($$0).resultOrPartial(arg_0 -> ((org.slf4j.Logger)net.minecraft.world.entity.ai.Brain.LOGGER).error(arg_0)).orElseGet(() -> new net.minecraft.world.entity.ai.Brain(this.memoryTypes, this.sensorTypes, com.google.common.collect.ImmutableList.of(), () -> this.codec));
     }
 }
 