--- net/minecraft/world/entity/ai/Brain$MemoryValue.java
+++ net/minecraft/world/entity/ai/Brain$MemoryValue.java
@@ -8,8 +8,8 @@
     public net.minecraft.world.entity.ai.memory.MemoryModuleType<U> type;
     public java.util.Optional<? extends net.minecraft.world.entity.ai.memory.ExpirableValue<U>> value;
 
-    public static <U> net.minecraft.world.entity.ai.Brain$MemoryValue<U> createUnchecked(net.minecraft.world.entity.ai.memory.MemoryModuleType<U> memoryType, java.util.Optional<? extends net.minecraft.world.entity.ai.memory.ExpirableValue<?>> memory) {
-        return new net.minecraft.world.entity.ai.Brain$MemoryValue(memoryType, memory);
+    public static <U> net.minecraft.world.entity.ai.Brain$MemoryValue<U> createUnchecked(net.minecraft.world.entity.ai.memory.MemoryModuleType<U> $$0, java.util.Optional<? extends net.minecraft.world.entity.ai.memory.ExpirableValue<?>> $$1) {
+        return new net.minecraft.world.entity.ai.Brain$MemoryValue($$0, $$1);
     }
 
     public net.minecraft.world.entity.ai.Brain$MemoryValue(net.minecraft.world.entity.ai.memory.MemoryModuleType<U> $$0, java.util.Optional<? extends net.minecraft.world.entity.ai.memory.ExpirableValue<U>> $$1) {
@@ -17,12 +17,12 @@
         this.value = $$1;
     }
 
-    public void setMemoryInternal(net.minecraft.world.entity.ai.Brain<?> brain) {
-        brain.setMemoryInternal(this.type, this.value);
+    public void setMemoryInternal(net.minecraft.world.entity.ai.Brain<?> $$0) {
+        $$0.setMemoryInternal(this.type, this.value);
     }
 
-    public <T> void serialize(com.mojang.serialization.DynamicOps<T> ops, com.mojang.serialization.RecordBuilder<T> builder) {
-        this.type.getCodec().ifPresent($$2 -> this.value.ifPresent($$3 -> builder.add(net.minecraft.core.Registry.MEMORY_MODULE_TYPE.byNameCodec().encodeStart(ops, (java.lang.Object)this.type), $$2.encodeStart(ops, $$3))));
+    public <T> void serialize(com.mojang.serialization.DynamicOps<T> $$0, com.mojang.serialization.RecordBuilder<T> $$1) {
+        this.type.getCodec().ifPresent($$2 -> this.value.ifPresent($$3 -> $$1.add(net.minecraft.core.Registry.MEMORY_MODULE_TYPE.byNameCodec().encodeStart($$0, (java.lang.Object)this.type), $$2.encodeStart($$0, $$3))));
     }
 }
 