--- net/minecraft/client/renderer/block/model/multipart/KeyValueCondition.java
+++ net/minecraft/client/renderer/block/model/multipart/KeyValueCondition.java
@@ -40,12 +40,12 @@
         return $$3 ? $$7.negate() : $$7;
     }
 
-    public java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> getBlockStatePredicate(net.minecraft.world.level.block.state.StateDefinition<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> container, net.minecraft.world.level.block.state.properties.Property<?> property, java.lang.String value) {
-        java.util.Optional optional = property.getValue(value);
-        if (!optional.isPresent()) {
-            throw new java.lang.RuntimeException(java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"Unknown value '%s' for property '%s' on '%s' in '%s'", (java.lang.Object[])new java.lang.Object[]{value, this.key, container.getOwner(), this.value}));
+    public java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> getBlockStatePredicate(net.minecraft.world.level.block.state.StateDefinition<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0, net.minecraft.world.level.block.state.properties.Property<?> $$1, java.lang.String $$22) {
+        java.util.Optional $$3 = $$1.getValue($$22);
+        if (!$$3.isPresent()) {
+            throw new java.lang.RuntimeException(java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"Unknown value '%s' for property '%s' on '%s' in '%s'", (java.lang.Object[])new java.lang.Object[]{$$22, this.key, $$0.getOwner(), this.value}));
         }
-        return $$2 -> $$2.getValue(property).equals(optional.get());
+        return $$2 -> $$2.getValue($$1).equals($$3.get());
     }
 
     public java.lang.String toString() {