--- net/minecraft/client/renderer/block/model/multipart/MultiPart$Deserializer.java
+++ net/minecraft/client/renderer/block/model/multipart/MultiPart$Deserializer.java
@@ -16,12 +16,12 @@
         return new net.minecraft.client.renderer.block.model.multipart.MultiPart(this.context.getDefinition(), this.getSelectors($$2, $$0.getAsJsonArray()));
     }
 
-    public java.util.List<net.minecraft.client.renderer.block.model.multipart.Selector> getSelectors(com.google.gson.JsonDeserializationContext context, com.google.gson.JsonArray elements) {
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        for (com.google.gson.JsonElement jsonElement : elements) {
-            arrayList.add((java.lang.Object)((net.minecraft.client.renderer.block.model.multipart.Selector)context.deserialize(jsonElement, net.minecraft.client.renderer.block.model.multipart.Selector.class)));
+    public java.util.List<net.minecraft.client.renderer.block.model.multipart.Selector> getSelectors(com.google.gson.JsonDeserializationContext $$0, com.google.gson.JsonArray $$1) {
+        java.util.ArrayList $$2 = com.google.common.collect.Lists.newArrayList();
+        for (com.google.gson.JsonElement $$3 : $$1) {
+            $$2.add((java.lang.Object)((net.minecraft.client.renderer.block.model.multipart.Selector)$$0.deserialize($$3, net.minecraft.client.renderer.block.model.multipart.Selector.class)));
         }
-        return arrayList;
+        return $$2;
     }
 
     public /* synthetic */ java.lang.Object deserialize(com.google.gson.JsonElement jsonElement, java.lang.reflect.Type type, com.google.gson.JsonDeserializationContext jsonDeserializationContext) throws com.google.gson.JsonParseException {