--- net/minecraft/client/renderer/block/model/multipart/Selector.java
+++ net/minecraft/client/renderer/block/model/multipart/Selector.java
@@ -23,8 +23,8 @@
         return this.variant;
     }
 
-    public java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> getPredicate(net.minecraft.world.level.block.state.StateDefinition<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> state) {
-        return this.condition.getPredicate(state);
+    public java.util.function.Predicate<net.minecraft.world.level.block.state.BlockState> getPredicate(net.minecraft.world.level.block.state.StateDefinition<net.minecraft.world.level.block.Block, net.minecraft.world.level.block.state.BlockState> $$0) {
+        return this.condition.getPredicate($$0);
     }
 
     public boolean equals(java.lang.Object $$0) {