--- net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen$DetailsScreen.java
+++ net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen$DetailsScreen.java
@@ -1,0 +1,47 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.gui.screens.worldselection;
+
+public class net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen
+extends net.minecraft.client.gui.screens.Screen
+ {
+    public net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackList packList;
+    public /* synthetic */ net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen this$0;
+
+    public net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen(net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen confirmExperimentalFeaturesScreen) {
+        this.this$0 = confirmExperimentalFeaturesScreen;
+        super((net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.experimental.details.title"));
+    }
+
+    public void onClose() {
+        this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)this.this$0);
+    }
+
+    public void init() {
+        super.init();
+        this.addRenderableWidget((net.minecraft.client.gui.components.events.GuiEventListener)new net.minecraft.client.gui.components.Button(this.width / 2 - 100, this.height / 4 + 120 + 24, 200, 20, net.minecraft.network.chat.CommonComponents.GUI_BACK, $$0 -> this.onClose()));
+        this.packList = new net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackList(this, this.minecraft, this.this$0.enabledPacks);
+        this.addWidget((net.minecraft.client.gui.components.events.GuiEventListener)this.packList);
+    }
+
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        this.renderBackground($$0);
+        this.packList.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)10, (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
+    }
+
+    public static /* synthetic */ net.minecraft.client.gui.Font access$000(net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen $$0) {
+        return $$0.font;
+    }
+
+    public static /* synthetic */ net.minecraft.client.Minecraft access$100(net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen $$0) {
+        return $$0.minecraft;
+    }
+
+    public static /* synthetic */ net.minecraft.client.gui.Font access$200(net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen $$0) {
+        return $$0.font;
+    }
+}
+