--- net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen$DetailsScreen$PackListEntry.java
+++ net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen$DetailsScreen$PackListEntry.java
@@ -1,0 +1,31 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.gui.screens.worldselection;
+
+public class net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackListEntry
+extends net.minecraft.client.gui.components.ObjectSelectionList$Entry<net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackListEntry>
+ {
+    public net.minecraft.network.chat.Component packId;
+    public net.minecraft.network.chat.Component message;
+    public net.minecraft.client.gui.components.MultiLineLabel splitMessage;
+    public /* synthetic */ net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen this$1;
+
+    public net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackListEntry(net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen detailsScreen, net.minecraft.network.chat.Component $$0, net.minecraft.network.chat.Component $$1, net.minecraft.client.gui.components.MultiLineLabel $$2) {
+        this.this$1 = detailsScreen;
+        this.packId = $$0;
+        this.message = $$1;
+        this.splitMessage = $$2;
+    }
+
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        net.minecraft.client.gui.GuiComponent.drawString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen.access$100((net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen)this.this$1).font, (net.minecraft.network.chat.Component)this.packId, (int)$$3, (int)$$2, (int)0xFFFFFF);
+        java.util.Objects.requireNonNull((java.lang.Object)net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen.access$200((net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen)this.this$1));
+        this.splitMessage.renderLeftAligned($$0, $$3, $$2 + 12, 9, 0xFFFFFF);
+    }
+
+    public net.minecraft.network.chat.Component getNarration() {
+        return net.minecraft.network.chat.Component.translatable((java.lang.String)"narrator.select", (java.lang.Object[])new java.lang.Object[]{net.minecraft.network.chat.CommonComponents.joinForNarration((net.minecraft.network.chat.Component)this.packId, (net.minecraft.network.chat.Component)this.message)});
+    }
+}
+