--- net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen$DetailsScreen$PackList.java
+++ net/minecraft/client/gui/screens/worldselection/ConfirmExperimentalFeaturesScreen$DetailsScreen$PackList.java
@@ -1,0 +1,32 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.gui.screens.worldselection;
+
+public class net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackList
+extends net.minecraft.client.gui.components.ObjectSelectionList<net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackListEntry>
+ {
+    public /* synthetic */ net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen this$1;
+
+    public net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackList(net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen detailsScreen, net.minecraft.client.Minecraft $$0, java.util.Collection<net.minecraft.server.packs.repository.Pack> $$1) {
+        this.this$1 = detailsScreen;
+        java.util.Objects.requireNonNull((java.lang.Object)$$0.font);
+        super($$0, detailsScreen.width, detailsScreen.height, 32, detailsScreen.height - 64, (9 + 2) * 3);
+        for (net.minecraft.server.packs.repository.Pack $$2 : $$1) {
+            java.lang.String $$3 = net.minecraft.world.flag.FeatureFlags.printMissingFlags((net.minecraft.world.flag.FeatureFlagSet)net.minecraft.world.flag.FeatureFlags.VANILLA_SET, (net.minecraft.world.flag.FeatureFlagSet)$$2.getRequestedFeatures());
+            if ($$3.isEmpty()) continue;
+            net.minecraft.network.chat.MutableComponent $$4 = net.minecraft.network.chat.ComponentUtils.mergeStyles((net.minecraft.network.chat.MutableComponent)$$2.getTitle().copy(), (net.minecraft.network.chat.Style)net.minecraft.network.chat.Style.EMPTY.withBold(java.lang.Boolean.valueOf((boolean)true)));
+            net.minecraft.network.chat.MutableComponent $$5 = net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.experimental.details.entry", (java.lang.Object[])new java.lang.Object[]{$$3});
+            this.addEntry((net.minecraft.client.gui.components.AbstractSelectionList$Entry)new net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen$PackListEntry(detailsScreen, (net.minecraft.network.chat.Component)$$4, (net.minecraft.network.chat.Component)$$5, net.minecraft.client.gui.components.MultiLineLabel.create((net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen.access$000((net.minecraft.client.gui.screens.worldselection.ConfirmExperimentalFeaturesScreen$DetailsScreen)detailsScreen), (net.minecraft.network.chat.FormattedText)$$5, (int)this.getRowWidth())));
+        }
+    }
+
+    public int getRowWidth() {
+        return this.width * 3 / 4;
+    }
+
+    public boolean isFocused() {
+        return this.this$1.getFocused() == this;
+    }
+}
+