--- net/minecraft/client/gui/screens/worldselection/CreateWorldScreen$DataPackReloadCookie.java
+++ net/minecraft/client/gui/screens/worldselection/CreateWorldScreen$DataPackReloadCookie.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.gui.screens.worldselection;
+
+public class net.minecraft.client.gui.screens.worldselection.CreateWorldScreen$DataPackReloadCookie
+extends java.lang.Record
+ {
+    public net.minecraft.world.level.levelgen.WorldGenSettings worldGenSettings;
+    public net.minecraft.world.level.WorldDataConfiguration dataConfiguration;
+
+    public net.minecraft.client.gui.screens.worldselection.CreateWorldScreen$DataPackReloadCookie(net.minecraft.world.level.levelgen.WorldGenSettings $$0, net.minecraft.world.level.WorldDataConfiguration $$1) {
+        this.worldGenSettings = $$0;
+        this.dataConfiguration = $$1;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.gui.screens.worldselection.CreateWorldScreen$DataPackReloadCookie.class, "worldGenSettings;dataConfiguration", "worldGenSettings", "dataConfiguration"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.gui.screens.worldselection.CreateWorldScreen$DataPackReloadCookie.class, "worldGenSettings;dataConfiguration", "worldGenSettings", "dataConfiguration"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.gui.screens.worldselection.CreateWorldScreen$DataPackReloadCookie.class, "worldGenSettings;dataConfiguration", "worldGenSettings", "dataConfiguration"}, this, $$0);
+    }
+
+    public net.minecraft.world.level.levelgen.WorldGenSettings worldGenSettings() {
+        return this.worldGenSettings;
+    }
+
+    public net.minecraft.world.level.WorldDataConfiguration dataConfiguration() {
+        return this.dataConfiguration;
+    }
+}
+