--- net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$RuleList.java
+++ net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$RuleList.java
@@ -19,11 +19,11 @@
         });
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$RuleEntry ruleEntry = (net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$RuleEntry)this.getHovered();
-        if (ruleEntry != null) {
-            this.this$0.setTooltip(ruleEntry.tooltip);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
+        super.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$RuleEntry $$4 = (net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$RuleEntry)this.getHovered();
+        if ($$4 != null) {
+            this.this$0.setTooltip($$4.tooltip);
         }
     }
 }