--- net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$BooleanRuleEntry.java
+++ net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$BooleanRuleEntry.java
@@ -16,11 +16,11 @@
         this.children.add((java.lang.Object)this.checkbox);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.renderLabel(matrixStack, top, left);
-        this.checkbox.x = left + width - 45;
-        this.checkbox.y = top;
-        this.checkbox.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        this.renderLabel($$0, $$2, $$3);
+        this.checkbox.x = $$3 + $$4 - 45;
+        this.checkbox.y = $$2;
+        this.checkbox.render($$0, $$6, $$7, $$9);
     }
 }
 