--- net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$RuleList$1.java
+++ net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$RuleList$1.java
@@ -18,12 +18,12 @@
         this.val$entries = map;
     }
 
-    public void visitBoolean(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$BooleanValue> value1, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> value2) {
-        this.addEntry(value1, ($$0, $$1, $$2, $$3) -> new net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$BooleanRuleEntry(this.this$1.this$0, $$0, $$1, $$2, $$3));
+    public void visitBoolean(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$BooleanValue> $$02, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$BooleanValue> $$12) {
+        this.addEntry($$02, ($$0, $$1, $$2, $$3) -> new net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$BooleanRuleEntry(this.this$1.this$0, $$0, $$1, $$2, $$3));
     }
 
-    public void visitInteger(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$IntegerValue> value1, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$IntegerValue> value2) {
-        this.addEntry(value1, ($$0, $$1, $$2, $$3) -> new net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$IntegerRuleEntry(this.this$1.this$0, $$0, $$1, $$2, $$3));
+    public void visitInteger(net.minecraft.world.level.GameRules$Key<net.minecraft.world.level.GameRules$IntegerValue> $$02, net.minecraft.world.level.GameRules$Type<net.minecraft.world.level.GameRules$IntegerValue> $$12) {
+        this.addEntry($$02, ($$0, $$1, $$2, $$3) -> new net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$IntegerRuleEntry(this.this$1.this$0, $$0, $$1, $$2, $$3));
     }
 
     public <T extends net.minecraft.world.level.GameRules$Value<T>> void addEntry(net.minecraft.world.level.GameRules$Key<T> $$02, net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen$EntryFactory<T> $$1) {