--- net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$IntegerRuleEntry.java
+++ net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$IntegerRuleEntry.java
@@ -26,11 +26,11 @@
         this.children.add((java.lang.Object)this.input);
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        this.renderLabel(matrixStack, top, left);
-        this.input.x = left + width - 44;
-        this.input.y = top;
-        this.input.render(matrixStack, mouseX, mouseY, partialTicks);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        this.renderLabel($$0, $$2, $$3);
+        this.input.x = $$3 + $$4 - 44;
+        this.input.y = $$2;
+        this.input.render($$0, $$6, $$7, $$9);
     }
 }
 