--- net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
+++ net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen.java
@@ -38,13 +38,13 @@
         this.exitCallback.accept((java.lang.Object)java.util.Optional.empty());
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         this.tooltip = null;
-        this.rules.render(matrixStack, mouseX, mouseY, partialTicks);
-        net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
-        super.render(matrixStack, mouseX, mouseY, partialTicks);
+        this.rules.render($$0, $$1, $$2, $$3);
+        net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)this.font, (net.minecraft.network.chat.Component)this.title, (int)(this.width / 2), (int)20, (int)0xFFFFFF);
+        super.render($$0, $$1, $$2, $$3);
         if (this.tooltip != null) {
-            this.renderTooltip(matrixStack, this.tooltip, mouseX, mouseY);
+            this.renderTooltip($$0, this.tooltip, $$1, $$2);
         }
     }
 