--- net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$CategoryRuleEntry.java
+++ net/minecraft/client/gui/screens/worldselection/EditGameRulesScreen$CategoryRuleEntry.java
@@ -15,8 +15,8 @@
         this.label = $$1;
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        net.minecraft.client.gui.GuiComponent.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen.access$000((net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen)this.this$0).font, (net.minecraft.network.chat.Component)this.label, (int)(left + width / 2), (int)(top + 5), (int)0xFFFFFF);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        net.minecraft.client.gui.GuiComponent.drawCenteredString((com.mojang.blaze3d.vertex.PoseStack)$$0, (net.minecraft.client.gui.Font)net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen.access$000((net.minecraft.client.gui.screens.worldselection.EditGameRulesScreen)this.this$0).font, (net.minecraft.network.chat.Component)this.label, (int)($$3 + $$4 / 2), (int)($$2 + 5), (int)0xFFFFFF);
     }
 
     public java.util.List<? extends net.minecraft.client.gui.components.events.GuiEventListener> children() {