--- net/minecraft/client/gui/screens/worldselection/WorldOpenFlows$1Data.java
+++ net/minecraft/client/gui/screens/worldselection/WorldOpenFlows$1Data.java
@@ -1,0 +1,43 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.client.gui.screens.worldselection;
+
+public class net.minecraft.client.gui.screens.worldselection.WorldOpenFlows$1Data
+extends java.lang.Record
+ {
+    public net.minecraft.world.level.LevelSettings levelSettings;
+    public net.minecraft.world.level.levelgen.WorldOptions options;
+    public net.minecraft.core.Registry<net.minecraft.world.level.dimension.LevelStem> existingDimensions;
+
+    public net.minecraft.client.gui.screens.worldselection.WorldOpenFlows$1Data(net.minecraft.world.level.LevelSettings $$0, net.minecraft.world.level.levelgen.WorldOptions $$1, net.minecraft.core.Registry<net.minecraft.world.level.dimension.LevelStem> $$2) {
+        this.levelSettings = $$0;
+        this.options = $$1;
+        this.existingDimensions = $$2;
+    }
+
+    public java.lang.String toString() {
+        return java.lang.runtime.ObjectMethods.bootstrap("toString", new java.lang.invoke.MethodHandle[]{net.minecraft.client.gui.screens.worldselection.WorldOpenFlows$1Data.class, "levelSettings;options;existingDimensions", "levelSettings", "options", "existingDimensions"}, this);
+    }
+
+    public int hashCode() {
+        return (int)java.lang.runtime.ObjectMethods.bootstrap("hashCode", new java.lang.invoke.MethodHandle[]{net.minecraft.client.gui.screens.worldselection.WorldOpenFlows$1Data.class, "levelSettings;options;existingDimensions", "levelSettings", "options", "existingDimensions"}, this);
+    }
+
+    public boolean equals(java.lang.Object $$0) {
+        return (boolean)java.lang.runtime.ObjectMethods.bootstrap("equals", new java.lang.invoke.MethodHandle[]{net.minecraft.client.gui.screens.worldselection.WorldOpenFlows$1Data.class, "levelSettings;options;existingDimensions", "levelSettings", "options", "existingDimensions"}, this, $$0);
+    }
+
+    public net.minecraft.world.level.LevelSettings levelSettings() {
+        return this.levelSettings;
+    }
+
+    public net.minecraft.world.level.levelgen.WorldOptions options() {
+        return this.options;
+    }
+
+    public net.minecraft.core.Registry<net.minecraft.world.level.dimension.LevelStem> existingDimensions() {
+        return this.existingDimensions;
+    }
+}
+