--- net/minecraft/client/gui/screens/worldselection/WorldCreationContext$Updater.java
+++ net/minecraft/client/gui/screens/worldselection/WorldCreationContext$Updater.java
@@ -1,11 +1,0 @@
-/*
- * Decompiled with CFR 0.151.
- */
-package net.minecraft.client.gui.screens.worldselection;
-
-@java.lang.FunctionalInterface
-public interface net.minecraft.client.gui.screens.worldselection.WorldCreationContext$Updater
-extends java.util.function.BiFunction<net.minecraft.core.RegistryAccess$Frozen, net.minecraft.world.level.levelgen.WorldGenSettings, net.minecraft.world.level.levelgen.WorldGenSettings>
- {
-}
-