--- net/minecraft/client/gui/screens/worldselection/WorldSelectionList$WorldListEntry.java
+++ net/minecraft/client/gui/screens/worldselection/WorldSelectionList$WorldListEntry.java
@@ -51,72 +51,72 @@
         return net.minecraft.network.chat.Component.translatable((java.lang.String)"narrator.select", (java.lang.Object[])new java.lang.Object[]{$$2});
     }
 
-    public void render(com.mojang.blaze3d.vertex.PoseStack matrixStack, int index, int top, int left, int width, int height, int mouseX, int mouseY, boolean isMouseOver, float partialTicks) {
-        java.lang.String string = this.summary.getLevelName();
-        java.lang.String string2 = this.summary.getLevelId() + " (" + net.minecraft.client.gui.screens.worldselection.WorldSelectionList.DATE_FORMAT.format(new java.util.Date(this.summary.getLastPlayed())) + ")";
-        if (org.apache.commons.lang3.StringUtils.isEmpty((java.lang.CharSequence)string)) {
-            string = net.minecraft.client.resources.language.I18n.get((java.lang.String)"selectWorld.world", (java.lang.Object[])new java.lang.Object[0]) + " " + (index + 1);
+    public void render(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, int $$5, int $$6, int $$7, boolean $$8, float $$9) {
+        java.lang.String $$10 = this.summary.getLevelName();
+        java.lang.String $$11 = this.summary.getLevelId() + " (" + net.minecraft.client.gui.screens.worldselection.WorldSelectionList.DATE_FORMAT.format(new java.util.Date(this.summary.getLastPlayed())) + ")";
+        if (org.apache.commons.lang3.StringUtils.isEmpty((java.lang.CharSequence)$$10)) {
+            $$10 = net.minecraft.client.resources.language.I18n.get((java.lang.String)"selectWorld.world", (java.lang.Object[])new java.lang.Object[0]) + " " + ($$1 + 1);
         }
-        net.minecraft.network.chat.Component component = this.summary.getInfo();
-        this.minecraft.font.draw(matrixStack, string, (float)(left + 32 + 3), (float)(top + 1), 0xFFFFFF);
-        float f = left + 32 + 3;
+        net.minecraft.network.chat.Component $$12 = this.summary.getInfo();
+        this.minecraft.font.draw($$0, $$10, (float)($$3 + 32 + 3), (float)($$2 + 1), 0xFFFFFF);
+        float f = $$3 + 32 + 3;
         java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-        this.minecraft.font.draw(matrixStack, string2, f, (float)(top + 9 + 3), 0x808080);
-        float f2 = left + 32 + 3;
+        this.minecraft.font.draw($$0, $$11, f, (float)($$2 + 9 + 3), 0x808080);
+        float f2 = $$3 + 32 + 3;
         java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
         java.util.Objects.requireNonNull((java.lang.Object)this.minecraft.font);
-        this.minecraft.font.draw(matrixStack, component, f2, (float)(top + 9 + 9 + 3), 0x808080);
+        this.minecraft.font.draw($$0, $$12, f2, (float)($$2 + 9 + 9 + 3), 0x808080);
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
         com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
         com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)(this.icon != null ? this.iconLocation : net.minecraft.client.gui.screens.worldselection.WorldSelectionList.ICON_MISSING));
         com.mojang.blaze3d.systems.RenderSystem.enableBlend();
-        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)32, (int)32);
+        net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)0.0f, (float)0.0f, (int)32, (int)32, (int)32, (int)32);
         com.mojang.blaze3d.systems.RenderSystem.disableBlend();
-        if (((java.lang.Boolean)this.minecraft.options.touchscreen().get()).booleanValue() || isMouseOver) {
-            int n;
+        if (((java.lang.Boolean)this.minecraft.options.touchscreen().get()).booleanValue() || $$8) {
+            int $$15;
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)net.minecraft.client.gui.screens.worldselection.WorldSelectionList.ICON_OVERLAY_LOCATION);
-            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (int)(left + 32), (int)(top + 32), (int)-1601138544);
+            net.minecraft.client.gui.GuiComponent.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (int)($$3 + 32), (int)($$2 + 32), (int)-1601138544);
             com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
             com.mojang.blaze3d.systems.RenderSystem.setShaderColor((float)1.0f, (float)1.0f, (float)1.0f, (float)1.0f);
-            int n2 = mouseX - left;
-            boolean bl = n2 < 32;
-            int n3 = n = bl ? 32 : 0;
+            int $$13 = $$6 - $$3;
+            boolean $$14 = $$13 < 32;
+            int n = $$15 = $$14 ? 32 : 0;
             if (this.summary.isLocked()) {
-                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)96.0f, (float)n, (int)32, (int)32, (int)256, (int)256);
-                if (bl) {
+                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)96.0f, (float)$$15, (int)32, (int)32, (int)256, (int)256);
+                if ($$14) {
                     this.screen.setToolTip(this.minecraft.font.split((net.minecraft.network.chat.FormattedText)net.minecraft.client.gui.screens.worldselection.WorldSelectionList.WORLD_LOCKED_TOOLTIP, 175));
                 }
             } else if (this.summary.requiresManualConversion()) {
-                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)96.0f, (float)n, (int)32, (int)32, (int)256, (int)256);
-                if (bl) {
+                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)96.0f, (float)$$15, (int)32, (int)32, (int)256, (int)256);
+                if ($$14) {
                     this.screen.setToolTip(this.minecraft.font.split((net.minecraft.network.chat.FormattedText)net.minecraft.client.gui.screens.worldselection.WorldSelectionList.WORLD_REQUIRES_CONVERSION, 175));
                 }
             } else if (this.summary.markVersionInList()) {
-                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)32.0f, (float)n, (int)32, (int)32, (int)256, (int)256);
+                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)32.0f, (float)$$15, (int)32, (int)32, (int)256, (int)256);
                 if (this.summary.askToOpenWorld()) {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)96.0f, (float)n, (int)32, (int)32, (int)256, (int)256);
-                    if (bl) {
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)96.0f, (float)$$15, (int)32, (int)32, (int)256, (int)256);
+                    if ($$14) {
                         this.screen.setToolTip((java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)net.minecraft.client.gui.screens.worldselection.WorldSelectionList.FROM_NEWER_TOOLTIP_1.getVisualOrderText(), (java.lang.Object)net.minecraft.client.gui.screens.worldselection.WorldSelectionList.FROM_NEWER_TOOLTIP_2.getVisualOrderText()));
                     }
                 } else if (!net.minecraft.SharedConstants.getCurrentVersion().isStable()) {
-                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)64.0f, (float)n, (int)32, (int)32, (int)256, (int)256);
-                    if (bl) {
+                    net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)64.0f, (float)$$15, (int)32, (int)32, (int)256, (int)256);
+                    if ($$14) {
                         this.screen.setToolTip((java.util.List)com.google.common.collect.ImmutableList.of((java.lang.Object)net.minecraft.client.gui.screens.worldselection.WorldSelectionList.SNAPSHOT_TOOLTIP_1.getVisualOrderText(), (java.lang.Object)net.minecraft.client.gui.screens.worldselection.WorldSelectionList.SNAPSHOT_TOOLTIP_2.getVisualOrderText()));
                     }
                 }
             } else {
-                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)left, (int)top, (float)0.0f, (float)n, (int)32, (int)32, (int)256, (int)256);
+                net.minecraft.client.gui.GuiComponent.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)$$3, (int)$$2, (float)0.0f, (float)$$15, (int)32, (int)32, (int)256, (int)256);
             }
         }
     }
 
-    public boolean mouseClicked(double mouseX, double mouseY, int button) {
+    public boolean mouseClicked(double $$0, double $$1, int $$2) {
         if (this.summary.isDisabled()) {
             return true;
         }
         this.this$0.setSelected((net.minecraft.client.gui.screens.worldselection.WorldSelectionList$Entry)this);
         this.screen.updateButtonStatus(this.this$0.getSelectedOpt().isPresent());
-        if (mouseX - (double)this.this$0.getRowLeft() <= 32.0) {
+        if ($$0 - (double)this.this$0.getRowLeft() <= 32.0) {
             this.joinWorld();
             return true;
         }
@@ -226,18 +226,19 @@
     public void recreateWorld() {
         this.queueLoadScreen();
         try (net.minecraft.world.level.storage.LevelStorageSource$LevelStorageAccess $$0 = this.minecraft.getLevelSource().createAccess(this.summary.getLevelId());
-             net.minecraft.server.WorldStem $$1 = this.minecraft.createWorldOpenFlows().loadWorldStem($$0, false);
 ){
-            net.minecraft.world.level.levelgen.WorldGenSettings $$22 = $$1.worldData().worldGenSettings();
-            java.nio.file.Path $$3 = net.minecraft.client.gui.screens.worldselection.CreateWorldScreen.createTempDataPackDirFromExistingWorld((java.nio.file.Path)$$0.getLevelPath(net.minecraft.world.level.storage.LevelResource.DATAPACK_DIR), (net.minecraft.client.Minecraft)this.minecraft);
-            if ($$22.isOldCustomizedWorld()) {
-                this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.ConfirmScreen($$2 -> this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)($$2 ? net.minecraft.client.gui.screens.worldselection.CreateWorldScreen.createFromExisting((net.minecraft.client.gui.screens.Screen)this.screen, (net.minecraft.server.WorldStem)$$1, (java.nio.file.Path)$$3) : this.screen)), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.recreate.customized.title"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.recreate.customized.text"), net.minecraft.network.chat.CommonComponents.GUI_PROCEED, net.minecraft.network.chat.CommonComponents.GUI_CANCEL));
+            com.mojang.datafixers.util.Pair $$1 = this.minecraft.createWorldOpenFlows().recreateWorldData($$0);
+            net.minecraft.world.level.LevelSettings $$2 = (net.minecraft.world.level.LevelSettings)$$1.getFirst();
+            net.minecraft.client.gui.screens.worldselection.WorldCreationContext $$32 = (net.minecraft.client.gui.screens.worldselection.WorldCreationContext)$$1.getSecond();
+            java.nio.file.Path $$4 = net.minecraft.client.gui.screens.worldselection.CreateWorldScreen.createTempDataPackDirFromExistingWorld((java.nio.file.Path)$$0.getLevelPath(net.minecraft.world.level.storage.LevelResource.DATAPACK_DIR), (net.minecraft.client.Minecraft)this.minecraft);
+            if ($$32.options().isOldCustomizedWorld()) {
+                this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.ConfirmScreen($$3 -> this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)($$3 ? net.minecraft.client.gui.screens.worldselection.CreateWorldScreen.createFromExisting((net.minecraft.client.gui.screens.Screen)this.screen, (net.minecraft.world.level.LevelSettings)$$2, (net.minecraft.client.gui.screens.worldselection.WorldCreationContext)$$32, (java.nio.file.Path)$$4) : this.screen)), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.recreate.customized.title"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.recreate.customized.text"), net.minecraft.network.chat.CommonComponents.GUI_PROCEED, net.minecraft.network.chat.CommonComponents.GUI_CANCEL));
             } else {
-                this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)net.minecraft.client.gui.screens.worldselection.CreateWorldScreen.createFromExisting((net.minecraft.client.gui.screens.Screen)this.screen, (net.minecraft.server.WorldStem)$$1, (java.nio.file.Path)$$3));
+                this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)net.minecraft.client.gui.screens.worldselection.CreateWorldScreen.createFromExisting((net.minecraft.client.gui.screens.Screen)this.screen, (net.minecraft.world.level.LevelSettings)$$2, (net.minecraft.client.gui.screens.worldselection.WorldCreationContext)$$32, (java.nio.file.Path)$$4));
             }
         }
-        catch (java.lang.Exception $$4) {
-            net.minecraft.client.gui.screens.worldselection.WorldSelectionList.LOGGER.error("Unable to recreate world", (java.lang.Throwable)$$4);
+        catch (java.lang.Exception $$5) {
+            net.minecraft.client.gui.screens.worldselection.WorldSelectionList.LOGGER.error("Unable to recreate world", (java.lang.Throwable)$$5);
             this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)new net.minecraft.client.gui.screens.AlertScreen(() -> this.minecraft.setScreen((net.minecraft.client.gui.screens.Screen)this.screen), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.recreate.error.title"), (net.minecraft.network.chat.Component)net.minecraft.network.chat.Component.translatable((java.lang.String)"selectWorld.recreate.error.text")));
         }
     }