--- net/minecraft/client/gui/screens/worldselection/WorldSelectionList.java
+++ net/minecraft/client/gui/screens/worldselection/WorldSelectionList.java
@@ -138,8 +138,8 @@
         this.screen.updateButtonStatus($$0 != null && $$0.isSelectable());
     }
 
-    public void moveSelection(net.minecraft.client.gui.components.AbstractSelectionList$SelectionDirection ordering) {
-        this.moveSelection(ordering, net.minecraft.client.gui.screens.worldselection.WorldSelectionList$Entry::isSelectable);
+    public void moveSelection(net.minecraft.client.gui.components.AbstractSelectionList$SelectionDirection $$0) {
+        this.moveSelection($$0, net.minecraft.client.gui.screens.worldselection.WorldSelectionList$Entry::isSelectable);
     }
 
     public java.util.Optional<net.minecraft.client.gui.screens.worldselection.WorldSelectionList$WorldListEntry> getSelectedOpt() {