--- net/minecraft/client/gui/screens/advancements/AdvancementTab.java
+++ net/minecraft/client/gui/screens/advancements/AdvancementTab.java
@@ -63,131 +63,131 @@
         return this.display;
     }
 
-    public void drawTab(com.mojang.blaze3d.vertex.PoseStack matrixStack, int offsetX, int offsetY, boolean isSelected) {
-        this.type.draw(matrixStack, (net.minecraft.client.gui.GuiComponent)this, offsetX, offsetY, isSelected, this.index);
+    public void drawTab(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, boolean $$3) {
+        this.type.draw($$0, (net.minecraft.client.gui.GuiComponent)this, $$1, $$2, $$3, this.index);
     }
 
-    public void drawIcon(int offsetX, int offsetY, net.minecraft.client.renderer.entity.ItemRenderer renderer) {
-        this.type.drawIcon(offsetX, offsetY, this.index, renderer, this.icon);
+    public void drawIcon(int $$0, int $$1, net.minecraft.client.renderer.entity.ItemRenderer $$2) {
+        this.type.drawIcon($$0, $$1, this.index, $$2, this.icon);
     }
 
-    public void drawContents(com.mojang.blaze3d.vertex.PoseStack matrixStack) {
+    public void drawContents(com.mojang.blaze3d.vertex.PoseStack $$0) {
         if (!this.centered) {
             this.scrollX = 117 - (this.maxX + this.minX) / 2;
             this.scrollY = 56 - (this.maxY + this.minY) / 2;
             this.centered = true;
         }
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.0, 950.0);
+        $$0.pushPose();
+        $$0.translate(0.0, 0.0, 950.0);
         com.mojang.blaze3d.systems.RenderSystem.enableDepthTest();
         com.mojang.blaze3d.systems.RenderSystem.colorMask((boolean)false, (boolean)false, (boolean)false, (boolean)false);
-        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)4680, (int)2260, (int)-4680, (int)-2260, (int)-16777216);
+        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)4680, (int)2260, (int)-4680, (int)-2260, (int)-16777216);
         com.mojang.blaze3d.systems.RenderSystem.colorMask((boolean)true, (boolean)true, (boolean)true, (boolean)true);
-        matrixStack.translate(0.0, 0.0, -950.0);
+        $$0.translate(0.0, 0.0, -950.0);
         com.mojang.blaze3d.systems.RenderSystem.depthFunc((int)518);
-        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)234, (int)113, (int)0, (int)0, (int)-16777216);
+        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)234, (int)113, (int)0, (int)0, (int)-16777216);
         com.mojang.blaze3d.systems.RenderSystem.depthFunc((int)515);
-        net.minecraft.resources.ResourceLocation resourceLocation = this.display.getBackground();
+        net.minecraft.resources.ResourceLocation $$1 = this.display.getBackground();
         com.mojang.blaze3d.systems.RenderSystem.setShader(net.minecraft.client.renderer.GameRenderer::getPositionTexShader);
-        if (resourceLocation != null) {
-            com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)resourceLocation);
+        if ($$1 != null) {
+            com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)$$1);
         } else {
             com.mojang.blaze3d.systems.RenderSystem.setShaderTexture((int)0, (net.minecraft.resources.ResourceLocation)net.minecraft.client.renderer.texture.TextureManager.INTENTIONAL_MISSING_TEXTURE);
         }
-        int n = net.minecraft.util.Mth.floor((double)this.scrollX);
-        int n2 = net.minecraft.util.Mth.floor((double)this.scrollY);
-        int n3 = n % 16;
-        int n4 = n2 % 16;
-        for (int i = -1; i <= 15; ++i) {
-            for (int j = -1; j <= 8; ++j) {
-                net.minecraft.client.gui.screens.advancements.AdvancementTab.blit((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)(n3 + 16 * i), (int)(n4 + 16 * j), (float)0.0f, (float)0.0f, (int)16, (int)16, (int)16, (int)16);
+        int $$2 = net.minecraft.util.Mth.floor((double)this.scrollX);
+        int $$3 = net.minecraft.util.Mth.floor((double)this.scrollY);
+        int $$4 = $$2 % 16;
+        int $$5 = $$3 % 16;
+        for (int $$6 = -1; $$6 <= 15; ++$$6) {
+            for (int $$7 = -1; $$7 <= 8; ++$$7) {
+                net.minecraft.client.gui.screens.advancements.AdvancementTab.blit((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)($$4 + 16 * $$6), (int)($$5 + 16 * $$7), (float)0.0f, (float)0.0f, (int)16, (int)16, (int)16, (int)16);
             }
         }
-        this.root.drawConnectivity(matrixStack, n, n2, true);
-        this.root.drawConnectivity(matrixStack, n, n2, false);
-        this.root.draw(matrixStack, n, n2);
+        this.root.drawConnectivity($$0, $$2, $$3, true);
+        this.root.drawConnectivity($$0, $$2, $$3, false);
+        this.root.draw($$0, $$2, $$3);
         com.mojang.blaze3d.systems.RenderSystem.depthFunc((int)518);
-        matrixStack.translate(0.0, 0.0, -950.0);
+        $$0.translate(0.0, 0.0, -950.0);
         com.mojang.blaze3d.systems.RenderSystem.colorMask((boolean)false, (boolean)false, (boolean)false, (boolean)false);
-        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)4680, (int)2260, (int)-4680, (int)-2260, (int)-16777216);
+        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)4680, (int)2260, (int)-4680, (int)-2260, (int)-16777216);
         com.mojang.blaze3d.systems.RenderSystem.colorMask((boolean)true, (boolean)true, (boolean)true, (boolean)true);
         com.mojang.blaze3d.systems.RenderSystem.depthFunc((int)515);
-        matrixStack.popPose();
+        $$0.popPose();
     }
 
-    public void drawTooltips(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, int width, int height) {
-        matrixStack.pushPose();
-        matrixStack.translate(0.0, 0.0, -200.0);
-        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)matrixStack, (int)0, (int)0, (int)234, (int)113, (int)(net.minecraft.util.Mth.floor((float)(this.fade * 255.0f)) << 24));
-        boolean bl = false;
-        int n = net.minecraft.util.Mth.floor((double)this.scrollX);
-        int n2 = net.minecraft.util.Mth.floor((double)this.scrollY);
-        if (mouseX > 0 && mouseX < 234 && mouseY > 0 && mouseY < 113) {
-            for (net.minecraft.client.gui.screens.advancements.AdvancementWidget advancementWidget : this.widgets.values()) {
-                if (!advancementWidget.isMouseOver(n, n2, mouseX, mouseY)) continue;
-                bl = true;
-                advancementWidget.drawHover(matrixStack, n, n2, this.fade, width, height);
+    public void drawTooltips(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4) {
+        $$0.pushPose();
+        $$0.translate(0.0, 0.0, -200.0);
+        net.minecraft.client.gui.screens.advancements.AdvancementTab.fill((com.mojang.blaze3d.vertex.PoseStack)$$0, (int)0, (int)0, (int)234, (int)113, (int)(net.minecraft.util.Mth.floor((float)(this.fade * 255.0f)) << 24));
+        boolean $$5 = false;
+        int $$6 = net.minecraft.util.Mth.floor((double)this.scrollX);
+        int $$7 = net.minecraft.util.Mth.floor((double)this.scrollY);
+        if ($$1 > 0 && $$1 < 234 && $$2 > 0 && $$2 < 113) {
+            for (net.minecraft.client.gui.screens.advancements.AdvancementWidget $$8 : this.widgets.values()) {
+                if (!$$8.isMouseOver($$6, $$7, $$1, $$2)) continue;
+                $$5 = true;
+                $$8.drawHover($$0, $$6, $$7, this.fade, $$3, $$4);
                 break;
             }
         }
-        matrixStack.popPose();
-        this.fade = bl ? net.minecraft.util.Mth.clamp((float)(this.fade + 0.02f), (float)0.0f, (float)0.3f) : net.minecraft.util.Mth.clamp((float)(this.fade - 0.04f), (float)0.0f, (float)1.0f);
+        $$0.popPose();
+        this.fade = $$5 ? net.minecraft.util.Mth.clamp((float)(this.fade + 0.02f), (float)0.0f, (float)0.3f) : net.minecraft.util.Mth.clamp((float)(this.fade - 0.04f), (float)0.0f, (float)1.0f);
     }
 
-    public boolean isMouseOver(int offsetX, int offsetY, double mouseX, double mouseY) {
-        return this.type.isMouseOver(offsetX, offsetY, this.index, mouseX, mouseY);
+    public boolean isMouseOver(int $$0, int $$1, double $$2, double $$3) {
+        return this.type.isMouseOver($$0, $$1, this.index, $$2, $$3);
     }
 
     @javax.annotation.Nullable
-    public static net.minecraft.client.gui.screens.advancements.AdvancementTab create(net.minecraft.client.Minecraft minecraft, net.minecraft.client.gui.screens.advancements.AdvancementsScreen screen, int tabIndex, net.minecraft.advancements.Advancement advancement) {
-        if (advancement.getDisplay() == null) {
+    public static net.minecraft.client.gui.screens.advancements.AdvancementTab create(net.minecraft.client.Minecraft $$0, net.minecraft.client.gui.screens.advancements.AdvancementsScreen $$1, int $$2, net.minecraft.advancements.Advancement $$3) {
+        if ($$3.getDisplay() == null) {
             return null;
         }
-        for (net.minecraft.client.gui.screens.advancements.AdvancementTabType advancementTabType : net.minecraft.client.gui.screens.advancements.AdvancementTabType.values()) {
-            if (tabIndex >= advancementTabType.getMax()) {
-                tabIndex -= advancementTabType.getMax();
+        for (net.minecraft.client.gui.screens.advancements.AdvancementTabType $$4 : net.minecraft.client.gui.screens.advancements.AdvancementTabType.values()) {
+            if ($$2 >= $$4.getMax()) {
+                $$2 -= $$4.getMax();
                 continue;
             }
-            return new net.minecraft.client.gui.screens.advancements.AdvancementTab(minecraft, screen, advancementTabType, tabIndex, advancement, advancement.getDisplay());
+            return new net.minecraft.client.gui.screens.advancements.AdvancementTab($$0, $$1, $$4, $$2, $$3, $$3.getDisplay());
         }
         return null;
     }
 
-    public void scroll(double dragX, double dragY) {
+    public void scroll(double $$0, double $$1) {
         if (this.maxX - this.minX > 234) {
-            this.scrollX = net.minecraft.util.Mth.clamp((double)(this.scrollX + dragX), (double)(-(this.maxX - 234)), (double)0.0);
+            this.scrollX = net.minecraft.util.Mth.clamp((double)(this.scrollX + $$0), (double)(-(this.maxX - 234)), (double)0.0);
         }
         if (this.maxY - this.minY > 113) {
-            this.scrollY = net.minecraft.util.Mth.clamp((double)(this.scrollY + dragY), (double)(-(this.maxY - 113)), (double)0.0);
+            this.scrollY = net.minecraft.util.Mth.clamp((double)(this.scrollY + $$1), (double)(-(this.maxY - 113)), (double)0.0);
         }
     }
 
-    public void addAdvancement(net.minecraft.advancements.Advancement advancement) {
-        if (advancement.getDisplay() == null) {
+    public void addAdvancement(net.minecraft.advancements.Advancement $$0) {
+        if ($$0.getDisplay() == null) {
             return;
         }
-        net.minecraft.client.gui.screens.advancements.AdvancementWidget advancementWidget = new net.minecraft.client.gui.screens.advancements.AdvancementWidget(this, this.minecraft, advancement, advancement.getDisplay());
-        this.addWidget(advancementWidget, advancement);
+        net.minecraft.client.gui.screens.advancements.AdvancementWidget $$1 = new net.minecraft.client.gui.screens.advancements.AdvancementWidget(this, this.minecraft, $$0, $$0.getDisplay());
+        this.addWidget($$1, $$0);
     }
 
-    public void addWidget(net.minecraft.client.gui.screens.advancements.AdvancementWidget gui, net.minecraft.advancements.Advancement advancement) {
-        this.widgets.put((java.lang.Object)advancement, (java.lang.Object)gui);
-        int n = gui.getX();
-        int n2 = n + 28;
-        int n3 = gui.getY();
-        int n4 = n3 + 27;
-        this.minX = java.lang.Math.min((int)this.minX, (int)n);
-        this.maxX = java.lang.Math.max((int)this.maxX, (int)n2);
-        this.minY = java.lang.Math.min((int)this.minY, (int)n3);
-        this.maxY = java.lang.Math.max((int)this.maxY, (int)n4);
-        for (net.minecraft.client.gui.screens.advancements.AdvancementWidget advancementWidget : this.widgets.values()) {
-            advancementWidget.attachToParent();
+    public void addWidget(net.minecraft.client.gui.screens.advancements.AdvancementWidget $$0, net.minecraft.advancements.Advancement $$1) {
+        this.widgets.put((java.lang.Object)$$1, (java.lang.Object)$$0);
+        int $$2 = $$0.getX();
+        int $$3 = $$2 + 28;
+        int $$4 = $$0.getY();
+        int $$5 = $$4 + 27;
+        this.minX = java.lang.Math.min((int)this.minX, (int)$$2);
+        this.maxX = java.lang.Math.max((int)this.maxX, (int)$$3);
+        this.minY = java.lang.Math.min((int)this.minY, (int)$$4);
+        this.maxY = java.lang.Math.max((int)this.maxY, (int)$$5);
+        for (net.minecraft.client.gui.screens.advancements.AdvancementWidget $$6 : this.widgets.values()) {
+            $$6.attachToParent();
         }
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.client.gui.screens.advancements.AdvancementWidget getWidget(net.minecraft.advancements.Advancement advancement) {
-        return (net.minecraft.client.gui.screens.advancements.AdvancementWidget)this.widgets.get((java.lang.Object)advancement);
+    public net.minecraft.client.gui.screens.advancements.AdvancementWidget getWidget(net.minecraft.advancements.Advancement $$0) {
+        return (net.minecraft.client.gui.screens.advancements.AdvancementWidget)this.widgets.get((java.lang.Object)$$0);
     }
 
     public net.minecraft.client.gui.screens.advancements.AdvancementsScreen getScreen() {