--- net/minecraft/world/level/storage/DerivedLevelData.java
+++ net/minecraft/world/level/storage/DerivedLevelData.java
@@ -46,7 +46,7 @@
         return this.wrapped.getClearWeatherTime();
     }
 
-    public void setClearWeatherTime(int time) {
+    public void setClearWeatherTime(int $$0) {
     }
 
     public boolean isThundering() {
@@ -69,40 +69,40 @@
         return this.worldData.getGameType();
     }
 
-    public void setXSpawn(int x) {
+    public void setXSpawn(int $$0) {
     }
 
-    public void setYSpawn(int y) {
+    public void setYSpawn(int $$0) {
     }
 
-    public void setZSpawn(int z) {
+    public void setZSpawn(int $$0) {
     }
 
-    public void setSpawnAngle(float angle) {
+    public void setSpawnAngle(float $$0) {
     }
 
-    public void setGameTime(long time) {
+    public void setGameTime(long $$0) {
     }
 
-    public void setDayTime(long time) {
+    public void setDayTime(long $$0) {
     }
 
-    public void setSpawn(net.minecraft.core.BlockPos spawnPoint, float angle) {
+    public void setSpawn(net.minecraft.core.BlockPos $$0, float $$1) {
     }
 
-    public void setThundering(boolean thundering) {
+    public void setThundering(boolean $$0) {
     }
 
-    public void setThunderTime(int time) {
+    public void setThunderTime(int $$0) {
     }
 
-    public void setRaining(boolean isRaining) {
+    public void setRaining(boolean $$0) {
     }
 
-    public void setRainTime(int time) {
+    public void setRainTime(int $$0) {
     }
 
-    public void setGameType(net.minecraft.world.level.GameType type) {
+    public void setGameType(net.minecraft.world.level.GameType $$0) {
     }
 
     public boolean isHardcore() {
@@ -117,7 +117,7 @@
         return this.wrapped.isInitialized();
     }
 
-    public void setInitialized(boolean initialized) {
+    public void setInitialized(boolean $$0) {
     }
 
     public net.minecraft.world.level.GameRules getGameRules() {
@@ -128,7 +128,7 @@
         return this.wrapped.getWorldBorder();
     }
 
-    public void setWorldBorder(net.minecraft.world.level.border.WorldBorder$Settings serializer) {
+    public void setWorldBorder(net.minecraft.world.level.border.WorldBorder$Settings $$0) {
     }
 
     public net.minecraft.world.Difficulty getDifficulty() {
@@ -147,21 +147,21 @@
         return 0;
     }
 
-    public void setWanderingTraderSpawnDelay(int delay) {
+    public void setWanderingTraderSpawnDelay(int $$0) {
     }
 
     public int getWanderingTraderSpawnChance() {
         return 0;
     }
 
-    public void setWanderingTraderSpawnChance(int chance) {
+    public void setWanderingTraderSpawnChance(int $$0) {
     }
 
     public java.util.UUID getWanderingTraderId() {
         return null;
     }
 
-    public void setWanderingTraderId(java.util.UUID id) {
+    public void setWanderingTraderId(java.util.UUID $$0) {
     }
 
     public void fillCrashReportCategory(net.minecraft.CrashReportCategory $$0, net.minecraft.world.level.LevelHeightAccessor $$1) {