--- net/minecraft/world/level/storage/LevelVersion.java
+++ net/minecraft/world/level/storage/LevelVersion.java
@@ -19,14 +19,14 @@
         this.snapshot = $$5;
     }
 
-    public static net.minecraft.world.level.storage.LevelVersion parse(com.mojang.serialization.Dynamic<?> nbt) {
-        int n = nbt.get("version").asInt(0);
-        long l = nbt.get("LastPlayed").asLong(0L);
-        com.mojang.serialization.OptionalDynamic optionalDynamic = nbt.get("Version");
-        if (optionalDynamic.result().isPresent()) {
-            return new net.minecraft.world.level.storage.LevelVersion(n, l, optionalDynamic.get("Name").asString(net.minecraft.SharedConstants.getCurrentVersion().getName()), optionalDynamic.get("Id").asInt(net.minecraft.SharedConstants.getCurrentVersion().getDataVersion().getVersion()), optionalDynamic.get("Series").asString(net.minecraft.world.level.storage.DataVersion.MAIN_SERIES), optionalDynamic.get("Snapshot").asBoolean(!net.minecraft.SharedConstants.getCurrentVersion().isStable()));
+    public static net.minecraft.world.level.storage.LevelVersion parse(com.mojang.serialization.Dynamic<?> $$0) {
+        int $$1 = $$0.get("version").asInt(0);
+        long $$2 = $$0.get("LastPlayed").asLong(0L);
+        com.mojang.serialization.OptionalDynamic $$3 = $$0.get("Version");
+        if ($$3.result().isPresent()) {
+            return new net.minecraft.world.level.storage.LevelVersion($$1, $$2, $$3.get("Name").asString(net.minecraft.SharedConstants.getCurrentVersion().getName()), $$3.get("Id").asInt(net.minecraft.SharedConstants.getCurrentVersion().getDataVersion().getVersion()), $$3.get("Series").asString(net.minecraft.world.level.storage.DataVersion.MAIN_SERIES), $$3.get("Snapshot").asBoolean(!net.minecraft.SharedConstants.getCurrentVersion().isStable()));
         }
-        return new net.minecraft.world.level.storage.LevelVersion(n, l, "", 0, net.minecraft.world.level.storage.DataVersion.MAIN_SERIES, false);
+        return new net.minecraft.world.level.storage.LevelVersion($$1, $$2, "", 0, net.minecraft.world.level.storage.DataVersion.MAIN_SERIES, false);
     }
 
     public int levelDataVersion() {