--- net/minecraft/world/level/storage/PrimaryLevelData$SpecialWorldProperty.java
+++ net/minecraft/world/level/storage/PrimaryLevelData$SpecialWorldProperty.java
@@ -1,0 +1,38 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.level.storage;
+
+@java.lang.Deprecated
+public class net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty
+extends java.lang.Enum<net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty>
+ {
+    public static /* enum */ net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty NONE;
+    public static /* enum */ net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty FLAT;
+    public static /* enum */ net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty DEBUG;
+    public static /* synthetic */ net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty[] $VALUES;
+
+    public static net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty[] values() {
+        return (net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty[])$VALUES.clone();
+    }
+
+    public static net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty valueOf(java.lang.String $$0) {
+        return (net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty)java.lang.Enum.valueOf(net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty.class, (java.lang.String)$$0);
+    }
+
+    public net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty() {
+        super(string, n);
+    }
+
+    public static /* synthetic */ net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty[] $values() {
+        return new net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty[]{NONE, FLAT, DEBUG};
+    }
+
+    static {
+        NONE = new net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty("NONE", 0);
+        FLAT = new net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty("FLAT", 1);
+        DEBUG = new net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty("DEBUG", 2);
+        $VALUES = net.minecraft.world.level.storage.PrimaryLevelData$SpecialWorldProperty.$values();
+    }
+}
+