--- net/minecraft/client/main/GameConfig$ServerData.java
+++ net/minecraft/client/main/GameConfig$ServerData.java
@@ -9,9 +9,9 @@
     public java.lang.String hostname;
     public int port;
 
-    public net.minecraft.client.main.GameConfig$ServerData(@javax.annotation.Nullable java.lang.String hostname, int port) {
-        this.hostname = hostname;
-        this.port = port;
+    public net.minecraft.client.main.GameConfig$ServerData(@javax.annotation.Nullable java.lang.String $$0, int $$1) {
+        this.hostname = $$0;
+        this.port = $$1;
     }
 }
 