--- net/minecraft/client/main/GameConfig$GameData.java
+++ net/minecraft/client/main/GameConfig$GameData.java
@@ -11,12 +11,12 @@
     public boolean disableMultiplayer;
     public boolean disableChat;
 
-    public net.minecraft.client.main.GameConfig$GameData(boolean demo, java.lang.String launchVersion, java.lang.String versionType, boolean disableMultiplayer, boolean disableChat) {
-        this.demo = demo;
-        this.launchVersion = launchVersion;
-        this.versionType = versionType;
-        this.disableMultiplayer = disableMultiplayer;
-        this.disableChat = disableChat;
+    public net.minecraft.client.main.GameConfig$GameData(boolean $$0, java.lang.String $$1, java.lang.String $$2, boolean $$3, boolean $$4) {
+        this.demo = $$0;
+        this.launchVersion = $$1;
+        this.versionType = $$2;
+        this.disableMultiplayer = $$3;
+        this.disableChat = $$4;
     }
 }
 