--- net/minecraft/client/main/GameConfig.java
+++ net/minecraft/client/main/GameConfig.java
@@ -11,12 +11,12 @@
     public net.minecraft.client.main.GameConfig$GameData game;
     public net.minecraft.client.main.GameConfig$ServerData server;
 
-    public net.minecraft.client.main.GameConfig(net.minecraft.client.main.GameConfig$UserData user, com.mojang.blaze3d.platform.DisplayData display, net.minecraft.client.main.GameConfig$FolderData location, net.minecraft.client.main.GameConfig$GameData game, net.minecraft.client.main.GameConfig$ServerData server) {
-        this.user = user;
-        this.display = display;
-        this.location = location;
-        this.game = game;
-        this.server = server;
+    public net.minecraft.client.main.GameConfig(net.minecraft.client.main.GameConfig$UserData $$0, com.mojang.blaze3d.platform.DisplayData $$1, net.minecraft.client.main.GameConfig$FolderData $$2, net.minecraft.client.main.GameConfig$GameData $$3, net.minecraft.client.main.GameConfig$ServerData $$4) {
+        this.user = $$0;
+        this.display = $$1;
+        this.location = $$2;
+        this.game = $$3;
+        this.server = $$4;
     }
 }
 