--- net/minecraft/client/main/GameConfig$UserData.java
+++ net/minecraft/client/main/GameConfig$UserData.java
@@ -10,11 +10,11 @@
     public com.mojang.authlib.properties.PropertyMap profileProperties;
     public java.net.Proxy proxy;
 
-    public net.minecraft.client.main.GameConfig$UserData(net.minecraft.client.User user, com.mojang.authlib.properties.PropertyMap userProperties, com.mojang.authlib.properties.PropertyMap profileProperties, java.net.Proxy proxy) {
-        this.user = user;
-        this.userProperties = userProperties;
-        this.profileProperties = profileProperties;
-        this.proxy = proxy;
+    public net.minecraft.client.main.GameConfig$UserData(net.minecraft.client.User $$0, com.mojang.authlib.properties.PropertyMap $$1, com.mojang.authlib.properties.PropertyMap $$2, java.net.Proxy $$3) {
+        this.user = $$0;
+        this.userProperties = $$1;
+        this.profileProperties = $$2;
+        this.proxy = $$3;
     }
 }
 