--- net/minecraft/client/main/GameConfig$FolderData.java
+++ net/minecraft/client/main/GameConfig$FolderData.java
@@ -11,15 +11,15 @@
     @javax.annotation.Nullable
     public java.lang.String assetIndex;
 
-    public net.minecraft.client.main.GameConfig$FolderData(java.io.File gameDirectory, java.io.File resourcePackDirectory, java.io.File assetDirectory, @javax.annotation.Nullable java.lang.String assetIndex) {
-        this.gameDirectory = gameDirectory;
-        this.resourcePackDirectory = resourcePackDirectory;
-        this.assetDirectory = assetDirectory;
-        this.assetIndex = assetIndex;
+    public net.minecraft.client.main.GameConfig$FolderData(java.io.File $$0, java.io.File $$1, java.io.File $$2, @javax.annotation.Nullable java.lang.String $$3) {
+        this.gameDirectory = $$0;
+        this.resourcePackDirectory = $$1;
+        this.assetDirectory = $$2;
+        this.assetIndex = $$3;
     }
 
-    public net.minecraft.client.resources.AssetIndex getAssetIndex() {
-        return this.assetIndex == null ? new net.minecraft.client.resources.DirectAssetIndex(this.assetDirectory) : new net.minecraft.client.resources.AssetIndex(this.assetDirectory, this.assetIndex);
+    public java.nio.file.Path getExternalAssetSource() {
+        return this.assetIndex == null ? this.assetDirectory.toPath() : net.minecraft.client.resources.IndexedAssetSource.createIndexFs((java.nio.file.Path)this.assetDirectory.toPath(), (java.lang.String)this.assetIndex);
     }
 }
 