--- net/minecraft/client/main/SilentInitException.java
+++ net/minecraft/client/main/SilentInitException.java
@@ -6,12 +6,12 @@
 public class net.minecraft.client.main.SilentInitException
 extends java.lang.RuntimeException
  {
-    public net.minecraft.client.main.SilentInitException(java.lang.String message) {
-        super(message);
+    public net.minecraft.client.main.SilentInitException(java.lang.String $$0) {
+        super($$0);
     }
 
-    public net.minecraft.client.main.SilentInitException(java.lang.String message, java.lang.Throwable cause) {
-        super(message, cause);
+    public net.minecraft.client.main.SilentInitException(java.lang.String $$0, java.lang.Throwable $$1) {
+        super($$0, $$1);
     }
 }
 