--- net/minecraft/world/entity/ai/village/poi/PoiManager$DistanceTracker.java
+++ net/minecraft/world/entity/ai/village/poi/PoiManager$DistanceTracker.java
@@ -16,19 +16,19 @@
         this.levels.defaultReturnValue((byte)7);
     }
 
-    public int getLevelFromSource(long pos) {
-        return this.this$0.isVillageCenter(pos) ? 0 : 7;
+    public int getLevelFromSource(long $$0) {
+        return this.this$0.isVillageCenter($$0) ? 0 : 7;
     }
 
-    public int getLevel(long sectionPos) {
-        return this.levels.get(sectionPos);
+    public int getLevel(long $$0) {
+        return this.levels.get($$0);
     }
 
-    public void setLevel(long sectionPos, int level) {
-        if (level > 6) {
-            this.levels.remove(sectionPos);
+    public void setLevel(long $$0, int $$1) {
+        if ($$1 > 6) {
+            this.levels.remove($$0);
         } else {
-            this.levels.put(sectionPos, (byte)level);
+            this.levels.put($$0, (byte)$$1);
         }
     }
 