--- net/minecraft/world/entity/ai/village/poi/PoiTypes.java
+++ net/minecraft/world/entity/ai/village/poi/PoiTypes.java
@@ -44,12 +44,12 @@
     public static net.minecraft.world.entity.ai.village.poi.PoiType register(net.minecraft.core.Registry<net.minecraft.world.entity.ai.village.poi.PoiType> $$0, net.minecraft.resources.ResourceKey<net.minecraft.world.entity.ai.village.poi.PoiType> $$1, java.util.Set<net.minecraft.world.level.block.state.BlockState> $$2, int $$3, int $$4) {
         net.minecraft.world.entity.ai.village.poi.PoiType $$5 = new net.minecraft.world.entity.ai.village.poi.PoiType($$2, $$3, $$4);
         net.minecraft.core.Registry.register($$0, $$1, (java.lang.Object)$$5);
-        net.minecraft.world.entity.ai.village.poi.PoiTypes.registerBlockStates((net.minecraft.core.Holder)$$0.getHolderOrThrow($$1));
+        net.minecraft.world.entity.ai.village.poi.PoiTypes.registerBlockStates((net.minecraft.core.Holder)$$0.getHolderOrThrow($$1), $$2);
         return $$5;
     }
 
-    public static void registerBlockStates(net.minecraft.core.Holder<net.minecraft.world.entity.ai.village.poi.PoiType> $$0) {
-        ((net.minecraft.world.entity.ai.village.poi.PoiType)$$0.value()).matchingStates().forEach($$1 -> {
+    public static void registerBlockStates(net.minecraft.core.Holder<net.minecraft.world.entity.ai.village.poi.PoiType> $$0, java.util.Set<net.minecraft.world.level.block.state.BlockState> $$12) {
+        $$12.forEach($$1 -> {
             net.minecraft.core.Holder $$2 = (net.minecraft.core.Holder)TYPE_BY_STATE.put($$1, (java.lang.Object)$$0);
             if ($$2 != null) {
                 throw (java.lang.IllegalStateException)net.minecraft.Util.pauseInIde((java.lang.Throwable)new java.lang.IllegalStateException(java.lang.String.format((java.util.Locale)java.util.Locale.ROOT, (java.lang.String)"%s is defined in more than one PoI type", (java.lang.Object[])new java.lang.Object[]{$$1})));