--- net/minecraft/world/level/levelgen/structure/pools/JigsawPlacement.java
+++ net/minecraft/world/level/levelgen/structure/pools/JigsawPlacement.java
@@ -10,14 +10,14 @@
     public net.minecraft.world.level.levelgen.structure.pools.JigsawPlacement() {
     }
 
-    public static java.util.Optional<net.minecraft.world.level.levelgen.structure.Structure$GenerationStub> addPieces(net.minecraft.world.level.levelgen.structure.Structure$GenerationContext $$0, net.minecraft.core.Holder<net.minecraft.world.level.levelgen.structure.pools.StructureTemplatePool> $$1, java.util.Optional<net.minecraft.resources.ResourceLocation> $$2, int $$3, net.minecraft.core.BlockPos $$4, boolean $$5, java.util.Optional<net.minecraft.world.level.levelgen.Heightmap$Types> $$6, int $$7) {
+    public static java.util.Optional<net.minecraft.world.level.levelgen.structure.Structure$GenerationStub> addPieces(net.minecraft.world.level.levelgen.structure.Structure$GenerationContext $$02, net.minecraft.core.Holder<net.minecraft.world.level.levelgen.structure.pools.StructureTemplatePool> $$1, java.util.Optional<net.minecraft.resources.ResourceLocation> $$2, int $$3, net.minecraft.core.BlockPos $$4, boolean $$5, java.util.Optional<net.minecraft.world.level.levelgen.Heightmap$Types> $$6, int $$7) {
         int $$28;
         net.minecraft.core.BlockPos $$20;
-        net.minecraft.core.RegistryAccess $$8 = $$0.registryAccess();
-        net.minecraft.world.level.chunk.ChunkGenerator $$9 = $$0.chunkGenerator();
-        net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager $$10 = $$0.structureTemplateManager();
-        net.minecraft.world.level.LevelHeightAccessor $$11 = $$0.heightAccessor();
-        net.minecraft.world.level.levelgen.WorldgenRandom $$12 = $$0.random();
+        net.minecraft.core.RegistryAccess $$8 = $$02.registryAccess();
+        net.minecraft.world.level.chunk.ChunkGenerator $$9 = $$02.chunkGenerator();
+        net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager $$10 = $$02.structureTemplateManager();
+        net.minecraft.world.level.LevelHeightAccessor $$11 = $$02.heightAccessor();
+        net.minecraft.world.level.levelgen.WorldgenRandom $$12 = $$02.random();
         net.minecraft.core.Registry $$13 = $$8.registryOrThrow(net.minecraft.core.Registry.TEMPLATE_POOL_REGISTRY);
         net.minecraft.world.level.block.Rotation $$142 = net.minecraft.world.level.block.Rotation.getRandom((net.minecraft.util.RandomSource)$$12);
         net.minecraft.world.level.levelgen.structure.pools.StructureTemplatePool $$15 = (net.minecraft.world.level.levelgen.structure.pools.StructureTemplatePool)$$1.value();
@@ -29,7 +29,7 @@
             net.minecraft.resources.ResourceLocation $$17 = (net.minecraft.resources.ResourceLocation)$$2.get();
             java.util.Optional $$18 = net.minecraft.world.level.levelgen.structure.pools.JigsawPlacement.getRandomNamedJigsaw((net.minecraft.world.level.levelgen.structure.pools.StructurePoolElement)$$16, (net.minecraft.resources.ResourceLocation)$$17, (net.minecraft.core.BlockPos)$$4, (net.minecraft.world.level.block.Rotation)$$142, (net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager)$$10, (net.minecraft.world.level.levelgen.WorldgenRandom)$$12);
             if ($$18.isEmpty()) {
-                LOGGER.error("No starting jigsaw {} found in start pool {}", (java.lang.Object)$$17, (java.lang.Object)((net.minecraft.resources.ResourceKey)$$1.unwrapKey().get()).location());
+                LOGGER.error("No starting jigsaw {} found in start pool {}", (java.lang.Object)$$17, $$1.unwrapKey().map($$0 -> $$0.location().toString()).orElse((java.lang.Object)"<unregistered>"));
                 return java.util.Optional.empty();
             }
             net.minecraft.core.BlockPos $$19 = (net.minecraft.core.BlockPos)$$18.get();
@@ -43,7 +43,7 @@
         int $$25 = ($$24.maxX() + $$24.minX()) / 2;
         int $$26 = ($$24.maxZ() + $$24.minZ()) / 2;
         if ($$6.isPresent()) {
-            int $$27 = $$4.getY() + $$9.getFirstFreeHeight($$25, $$26, (net.minecraft.world.level.levelgen.Heightmap$Types)$$6.get(), $$11, $$0.randomState());
+            int $$27 = $$4.getY() + $$9.getFirstFreeHeight($$25, $$26, (net.minecraft.world.level.levelgen.Heightmap$Types)$$6.get(), $$11, $$02.randomState());
         } else {
             $$28 = $$22.getY();
         }
@@ -58,7 +58,7 @@
             }
             net.minecraft.world.phys.AABB $$16 = new net.minecraft.world.phys.AABB((double)($$25 - $$7), (double)($$30 - $$7), (double)($$26 - $$7), (double)($$25 + $$7 + 1), (double)($$30 + $$7 + 1), (double)($$26 + $$7 + 1));
             net.minecraft.world.phys.shapes.VoxelShape $$17 = net.minecraft.world.phys.shapes.Shapes.join((net.minecraft.world.phys.shapes.VoxelShape)net.minecraft.world.phys.shapes.Shapes.create((net.minecraft.world.phys.AABB)$$16), (net.minecraft.world.phys.shapes.VoxelShape)net.minecraft.world.phys.shapes.Shapes.create((net.minecraft.world.phys.AABB)net.minecraft.world.phys.AABB.of((net.minecraft.world.level.levelgen.structure.BoundingBox)$$24)), (net.minecraft.world.phys.shapes.BooleanOp)net.minecraft.world.phys.shapes.BooleanOp.ONLY_FIRST);
-            net.minecraft.world.level.levelgen.structure.pools.JigsawPlacement.addPieces((net.minecraft.world.level.levelgen.RandomState)$$0.randomState(), (int)$$3, (boolean)$$5, (net.minecraft.world.level.chunk.ChunkGenerator)$$9, (net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager)$$10, (net.minecraft.world.level.LevelHeightAccessor)$$11, (net.minecraft.util.RandomSource)$$12, (net.minecraft.core.Registry)$$13, (net.minecraft.world.level.levelgen.structure.PoolElementStructurePiece)$$23, (java.util.List)$$15, (net.minecraft.world.phys.shapes.VoxelShape)$$17);
+            net.minecraft.world.level.levelgen.structure.pools.JigsawPlacement.addPieces((net.minecraft.world.level.levelgen.RandomState)$$02.randomState(), (int)$$3, (boolean)$$5, (net.minecraft.world.level.chunk.ChunkGenerator)$$9, (net.minecraft.world.level.levelgen.structure.templatesystem.StructureTemplateManager)$$10, (net.minecraft.world.level.LevelHeightAccessor)$$11, (net.minecraft.util.RandomSource)$$12, (net.minecraft.core.Registry)$$13, (net.minecraft.world.level.levelgen.structure.PoolElementStructurePiece)$$23, (java.util.List)$$15, (net.minecraft.world.phys.shapes.VoxelShape)$$17);
             $$15.forEach(arg_0 -> ((net.minecraft.world.level.levelgen.structure.pieces.StructurePiecesBuilder)$$14).addPiece(arg_0));
         }));
     }