--- net/minecraft/resources/FileToIdConverter.java
+++ net/minecraft/resources/FileToIdConverter.java
@@ -1,0 +1,37 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.resources;
+
+public class net.minecraft.resources.FileToIdConverter
+ {
+    public java.lang.String prefix;
+    public java.lang.String extension;
+
+    public net.minecraft.resources.FileToIdConverter(java.lang.String $$0, java.lang.String $$1) {
+        this.prefix = $$0;
+        this.extension = $$1;
+    }
+
+    public static net.minecraft.resources.FileToIdConverter json(java.lang.String $$0) {
+        return new net.minecraft.resources.FileToIdConverter($$0, ".json");
+    }
+
+    public net.minecraft.resources.ResourceLocation idToFile(net.minecraft.resources.ResourceLocation $$0) {
+        return $$0.withPath(this.prefix + "/" + $$0.getPath() + this.extension);
+    }
+
+    public net.minecraft.resources.ResourceLocation fileToId(net.minecraft.resources.ResourceLocation $$0) {
+        java.lang.String $$1 = $$0.getPath();
+        return $$0.withPath($$1.substring(this.prefix.length() + 1, $$1.length() - this.extension.length()));
+    }
+
+    public java.util.Map<net.minecraft.resources.ResourceLocation, net.minecraft.server.packs.resources.Resource> listMatchingResources(net.minecraft.server.packs.resources.ResourceManager $$02) {
+        return $$02.listResources(this.prefix, $$0 -> $$0.getPath().endsWith(this.extension));
+    }
+
+    public java.util.Map<net.minecraft.resources.ResourceLocation, java.util.List<net.minecraft.server.packs.resources.Resource>> listMatchingResourceStacks(net.minecraft.server.packs.resources.ResourceManager $$02) {
+        return $$02.listResourceStacks(this.prefix, $$0 -> $$0.getPath().endsWith(this.extension));
+    }
+}
+