--- net/minecraft/resources/RegistryLoader$ReadCache.java
+++ net/minecraft/resources/RegistryLoader$ReadCache.java
@@ -1,14 +1,0 @@
-/*
- * Decompiled with CFR 0.151.
- */
-package net.minecraft.resources;
-
-public class net.minecraft.resources.RegistryLoader$ReadCache<E>
- {
-    public java.util.Map<net.minecraft.resources.ResourceKey<E>, com.mojang.serialization.DataResult<net.minecraft.core.Holder<E>>> values;
-
-    public net.minecraft.resources.RegistryLoader$ReadCache() {
-        this.values = com.google.common.collect.Maps.newIdentityHashMap();
-    }
-}
-