--- net/minecraft/world/level/levelgen/presets/WorldPresets$Bootstrap.java
+++ net/minecraft/world/level/levelgen/presets/WorldPresets$Bootstrap.java
@@ -53,13 +53,13 @@
 
     public net.minecraft.core.Holder<net.minecraft.world.level.levelgen.presets.WorldPreset> run() {
         net.minecraft.world.level.biome.MultiNoiseBiomeSource $$0 = net.minecraft.world.level.biome.MultiNoiseBiomeSource$Preset.OVERWORLD.biomeSource(this.biomes);
-        net.minecraft.core.Holder $$1 = this.noiseSettings.getOrCreateHolderOrThrow(net.minecraft.world.level.levelgen.NoiseGeneratorSettings.OVERWORLD);
-        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.NORMAL, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)$$0, $$1));
-        net.minecraft.core.Holder $$2 = this.noiseSettings.getOrCreateHolderOrThrow(net.minecraft.world.level.levelgen.NoiseGeneratorSettings.LARGE_BIOMES);
-        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.LARGE_BIOMES, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)$$0, $$2));
-        net.minecraft.core.Holder $$3 = this.noiseSettings.getOrCreateHolderOrThrow(net.minecraft.world.level.levelgen.NoiseGeneratorSettings.AMPLIFIED);
-        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.AMPLIFIED, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)$$0, $$3));
-        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.SINGLE_BIOME_SURFACE, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)new net.minecraft.world.level.biome.FixedBiomeSource(this.biomes.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.PLAINS)), $$1));
+        net.minecraft.core.Holder$Reference $$1 = this.noiseSettings.getOrCreateHolderOrThrow(net.minecraft.world.level.levelgen.NoiseGeneratorSettings.OVERWORLD);
+        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.NORMAL, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)$$0, (net.minecraft.core.Holder)$$1));
+        net.minecraft.core.Holder$Reference $$2 = this.noiseSettings.getOrCreateHolderOrThrow(net.minecraft.world.level.levelgen.NoiseGeneratorSettings.LARGE_BIOMES);
+        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.LARGE_BIOMES, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)$$0, (net.minecraft.core.Holder)$$2));
+        net.minecraft.core.Holder$Reference $$3 = this.noiseSettings.getOrCreateHolderOrThrow(net.minecraft.world.level.levelgen.NoiseGeneratorSettings.AMPLIFIED);
+        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.AMPLIFIED, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)$$0, (net.minecraft.core.Holder)$$3));
+        this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.SINGLE_BIOME_SURFACE, this.makeNoiseBasedOverworld((net.minecraft.world.level.biome.BiomeSource)new net.minecraft.world.level.biome.FixedBiomeSource((net.minecraft.core.Holder)this.biomes.getOrCreateHolderOrThrow(net.minecraft.world.level.biome.Biomes.PLAINS)), (net.minecraft.core.Holder)$$1));
         this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.FLAT, this.makeOverworld((net.minecraft.world.level.chunk.ChunkGenerator)new net.minecraft.world.level.levelgen.FlatLevelSource(this.structureSets, net.minecraft.world.level.levelgen.flat.FlatLevelGeneratorSettings.getDefault((net.minecraft.core.Registry)this.biomes, (net.minecraft.core.Registry)this.structureSets))));
         return this.registerCustomOverworldPreset(net.minecraft.world.level.levelgen.presets.WorldPresets.DEBUG, this.makeOverworld((net.minecraft.world.level.chunk.ChunkGenerator)new net.minecraft.world.level.levelgen.DebugLevelSource(this.structureSets, this.biomes)));
     }