--- net/minecraft/network/CompressionEncoder.java
+++ net/minecraft/network/CompressionEncoder.java
@@ -10,9 +10,9 @@
     public java.util.zip.Deflater deflater;
     public int threshold;
 
-    public net.minecraft.network.CompressionEncoder(int threshold) {
+    public net.minecraft.network.CompressionEncoder(int $$0) {
         this.encodeBuf = new byte[8192];
-        this.threshold = threshold;
+        this.threshold = $$0;
         this.deflater = new java.util.zip.Deflater();
     }
 
@@ -40,8 +40,8 @@
         return this.threshold;
     }
 
-    public void setThreshold(int threshold) {
-        this.threshold = threshold;
+    public void setThreshold(int $$0) {
+        this.threshold = $$0;
     }
 
     public /* synthetic */ void encode(io.netty.channel.ChannelHandlerContext channelHandlerContext, java.lang.Object object, io.netty.buffer.ByteBuf byteBuf) throws java.lang.Exception {