--- net/minecraft/network/ConnectionProtocol$ProtocolBuilder.java
+++ net/minecraft/network/ConnectionProtocol$ProtocolBuilder.java
@@ -11,8 +11,8 @@
         this.flows = com.google.common.collect.Maps.newEnumMap(net.minecraft.network.protocol.PacketFlow.class);
     }
 
-    public <T extends net.minecraft.network.PacketListener> net.minecraft.network.ConnectionProtocol$ProtocolBuilder addFlow(net.minecraft.network.protocol.PacketFlow packetFlow, net.minecraft.network.ConnectionProtocol$PacketSet<T> packetSet) {
-        this.flows.put((java.lang.Object)packetFlow, packetSet);
+    public <T extends net.minecraft.network.PacketListener> net.minecraft.network.ConnectionProtocol$ProtocolBuilder addFlow(net.minecraft.network.protocol.PacketFlow $$0, net.minecraft.network.ConnectionProtocol$PacketSet<T> $$1) {
+        this.flows.put((java.lang.Object)$$0, $$1);
         return this;
     }
 }