--- net/minecraft/network/PacketEncoder.java
+++ net/minecraft/network/PacketEncoder.java
@@ -9,8 +9,8 @@
     public static org.slf4j.Logger LOGGER;
     public net.minecraft.network.protocol.PacketFlow flow;
 
-    public net.minecraft.network.PacketEncoder(net.minecraft.network.protocol.PacketFlow flow) {
-        this.flow = flow;
+    public net.minecraft.network.PacketEncoder(net.minecraft.network.protocol.PacketFlow $$0) {
+        this.flow = $$0;
     }
 
     public void encode(io.netty.channel.ChannelHandlerContext $$0, net.minecraft.network.protocol.Packet<?> $$1, io.netty.buffer.ByteBuf $$2) throws java.lang.Exception {