--- net/minecraft/network/RateKickingConnection.java
+++ net/minecraft/network/RateKickingConnection.java
@@ -10,9 +10,9 @@
     public static net.minecraft.network.chat.Component EXCEED_REASON;
     public int rateLimitPacketsPerSecond;
 
-    public net.minecraft.network.RateKickingConnection(int rateLimitPacketsPerSecond) {
+    public net.minecraft.network.RateKickingConnection(int $$0) {
         super(net.minecraft.network.protocol.PacketFlow.SERVERBOUND);
-        this.rateLimitPacketsPerSecond = rateLimitPacketsPerSecond;
+        this.rateLimitPacketsPerSecond = $$0;
     }
 
     public void tickSecond() {