--- net/minecraft/world/level/chunk/DataLayer.java
+++ net/minecraft/world/level/chunk/DataLayer.java
@@ -15,47 +15,47 @@
     public net.minecraft.world.level.chunk.DataLayer() {
     }
 
-    public net.minecraft.world.level.chunk.DataLayer(byte[] data) {
-        this.data = data;
-        if (data.length != 2048) {
-            throw (java.lang.IllegalArgumentException)net.minecraft.Util.pauseInIde((java.lang.Throwable)new java.lang.IllegalArgumentException("DataLayer should be 2048 bytes not: " + data.length));
+    public net.minecraft.world.level.chunk.DataLayer(byte[] $$0) {
+        this.data = $$0;
+        if ($$0.length != 2048) {
+            throw (java.lang.IllegalArgumentException)net.minecraft.Util.pauseInIde((java.lang.Throwable)new java.lang.IllegalArgumentException("DataLayer should be 2048 bytes not: " + $$0.length));
         }
     }
 
-    public net.minecraft.world.level.chunk.DataLayer(int size) {
-        this.data = new byte[size];
+    public net.minecraft.world.level.chunk.DataLayer(int $$0) {
+        this.data = new byte[$$0];
     }
 
-    public int get(int x, int y, int z) {
-        return this.get(net.minecraft.world.level.chunk.DataLayer.getIndex((int)x, (int)y, (int)z));
+    public int get(int $$0, int $$1, int $$2) {
+        return this.get(net.minecraft.world.level.chunk.DataLayer.getIndex((int)$$0, (int)$$1, (int)$$2));
     }
 
-    public void set(int x, int y, int z, int value) {
-        this.set(net.minecraft.world.level.chunk.DataLayer.getIndex((int)x, (int)y, (int)z), value);
+    public void set(int $$0, int $$1, int $$2, int $$3) {
+        this.set(net.minecraft.world.level.chunk.DataLayer.getIndex((int)$$0, (int)$$1, (int)$$2), $$3);
     }
 
-    public static int getIndex(int n, int x, int y) {
-        return x << 8 | y << 4 | n;
+    public static int getIndex(int $$0, int $$1, int $$2) {
+        return $$1 << 8 | $$2 << 4 | $$0;
     }
 
-    public int get(int index) {
+    public int get(int $$0) {
         if (this.data == null) {
             return 0;
         }
-        int n = net.minecraft.world.level.chunk.DataLayer.getByteIndex((int)index);
-        int n2 = net.minecraft.world.level.chunk.DataLayer.getNibbleIndex((int)index);
-        return this.data[n] >> 4 * n2 & 0xF;
+        int $$1 = net.minecraft.world.level.chunk.DataLayer.getByteIndex((int)$$0);
+        int $$2 = net.minecraft.world.level.chunk.DataLayer.getNibbleIndex((int)$$0);
+        return this.data[$$1] >> 4 * $$2 & 0xF;
     }
 
-    public void set(int index, int value) {
+    public void set(int $$0, int $$1) {
         if (this.data == null) {
             this.data = new byte[2048];
         }
-        int n = net.minecraft.world.level.chunk.DataLayer.getByteIndex((int)index);
-        int n2 = net.minecraft.world.level.chunk.DataLayer.getNibbleIndex((int)index);
-        int n3 = ~(15 << 4 * n2);
-        int n4 = (value & 0xF) << 4 * n2;
-        this.data[n] = (byte)(this.data[n] & n3 | n4);
+        int $$2 = net.minecraft.world.level.chunk.DataLayer.getByteIndex((int)$$0);
+        int $$3 = net.minecraft.world.level.chunk.DataLayer.getNibbleIndex((int)$$0);
+        int $$4 = ~(15 << 4 * $$3);
+        int $$5 = ($$1 & 0xF) << 4 * $$3;
+        this.data[$$2] = (byte)(this.data[$$2] & $$4 | $$5);
     }
 
     public static int getNibbleIndex(int $$0) {