--- net/minecraft/world/level/chunk/EmptyLevelChunk.java
+++ net/minecraft/world/level/chunk/EmptyLevelChunk.java
@@ -13,25 +13,25 @@
         this.biome = $$2;
     }
 
-    public net.minecraft.world.level.block.state.BlockState getBlockState(net.minecraft.core.BlockPos pos) {
+    public net.minecraft.world.level.block.state.BlockState getBlockState(net.minecraft.core.BlockPos $$0) {
         return net.minecraft.world.level.block.Blocks.VOID_AIR.defaultBlockState();
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.block.state.BlockState setBlockState(net.minecraft.core.BlockPos pos, net.minecraft.world.level.block.state.BlockState state, boolean isMoving) {
+    public net.minecraft.world.level.block.state.BlockState setBlockState(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.block.state.BlockState $$1, boolean $$2) {
         return null;
     }
 
-    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.core.BlockPos pos) {
+    public net.minecraft.world.level.material.FluidState getFluidState(net.minecraft.core.BlockPos $$0) {
         return net.minecraft.world.level.material.Fluids.EMPTY.defaultFluidState();
     }
 
-    public int getLightEmission(net.minecraft.core.BlockPos pos) {
+    public int getLightEmission(net.minecraft.core.BlockPos $$0) {
         return 0;
     }
 
     @javax.annotation.Nullable
-    public net.minecraft.world.level.block.entity.BlockEntity getBlockEntity(net.minecraft.core.BlockPos pos, net.minecraft.world.level.chunk.LevelChunk$EntityCreationType creationMode) {
+    public net.minecraft.world.level.block.entity.BlockEntity getBlockEntity(net.minecraft.core.BlockPos $$0, net.minecraft.world.level.chunk.LevelChunk$EntityCreationType $$1) {
         return null;
     }
 
@@ -41,14 +41,14 @@
     public void setBlockEntity(net.minecraft.world.level.block.entity.BlockEntity $$0) {
     }
 
-    public void removeBlockEntity(net.minecraft.core.BlockPos pos) {
+    public void removeBlockEntity(net.minecraft.core.BlockPos $$0) {
     }
 
     public boolean isEmpty() {
         return true;
     }
 
-    public boolean isYSpaceEmpty(int startY, int endY) {
+    public boolean isYSpaceEmpty(int $$0, int $$1) {
         return true;
     }
 