--- net/minecraft/world/level/chunk/GlobalPalette.java
+++ net/minecraft/world/level/chunk/GlobalPalette.java
@@ -16,27 +16,27 @@
         return new net.minecraft.world.level.chunk.GlobalPalette($$1);
     }
 
-    public int idFor(T state) {
-        int n = this.registry.getId(state);
-        return n == -1 ? 0 : n;
+    public int idFor(T $$0) {
+        int $$1 = this.registry.getId($$0);
+        return $$1 == -1 ? 0 : $$1;
     }
 
-    public boolean maybeHas(java.util.function.Predicate<T> filter) {
+    public boolean maybeHas(java.util.function.Predicate<T> $$0) {
         return true;
     }
 
-    public T valueFor(int indexKey) {
-        java.lang.Object object = this.registry.byId(indexKey);
-        if (object == null) {
-            throw new net.minecraft.world.level.chunk.MissingPaletteEntryException(indexKey);
+    public T valueFor(int $$0) {
+        java.lang.Object $$1 = this.registry.byId($$0);
+        if ($$1 == null) {
+            throw new net.minecraft.world.level.chunk.MissingPaletteEntryException($$0);
         }
-        return (T)object;
+        return (T)$$1;
     }
 
-    public void read(net.minecraft.network.FriendlyByteBuf buf) {
+    public void read(net.minecraft.network.FriendlyByteBuf $$0) {
     }
 
-    public void write(net.minecraft.network.FriendlyByteBuf buf) {
+    public void write(net.minecraft.network.FriendlyByteBuf $$0) {
     }
 
     public int getSerializedSize() {