--- net/minecraft/world/entity/npc/VillagerTrades$EmeraldForItems.java
+++ net/minecraft/world/entity/npc/VillagerTrades$EmeraldForItems.java
@@ -12,11 +12,11 @@
     public int villagerXp;
     public float priceMultiplier;
 
-    public net.minecraft.world.entity.npc.VillagerTrades$EmeraldForItems(net.minecraft.world.level.ItemLike item, int cost, int maxUses, int villagerXp) {
-        this.item = item.asItem();
-        this.cost = cost;
-        this.maxUses = maxUses;
-        this.villagerXp = villagerXp;
+    public net.minecraft.world.entity.npc.VillagerTrades$EmeraldForItems(net.minecraft.world.level.ItemLike $$0, int $$1, int $$2, int $$3) {
+        this.item = $$0.asItem();
+        this.cost = $$1;
+        this.maxUses = $$2;
+        this.villagerXp = $$3;
         this.priceMultiplier = 0.05f;
     }
 