--- net/minecraft/world/entity/npc/VillagerTrades$TippedArrowForItemsAndEmeralds.java
+++ net/minecraft/world/entity/npc/VillagerTrades$TippedArrowForItemsAndEmeralds.java
@@ -15,20 +15,20 @@
     public int fromCount;
     public float priceMultiplier;
 
-    public net.minecraft.world.entity.npc.VillagerTrades$TippedArrowForItemsAndEmeralds(net.minecraft.world.item.Item fromItem, int fromCount, net.minecraft.world.item.Item toItem, int toCount, int emeraldCost, int maxUses, int villagerXp) {
-        this.toItem = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)toItem);
-        this.emeraldCost = emeraldCost;
-        this.maxUses = maxUses;
-        this.villagerXp = villagerXp;
-        this.fromItem = fromItem;
-        this.fromCount = fromCount;
-        this.toCount = toCount;
+    public net.minecraft.world.entity.npc.VillagerTrades$TippedArrowForItemsAndEmeralds(net.minecraft.world.item.Item $$0, int $$1, net.minecraft.world.item.Item $$2, int $$3, int $$4, int $$5, int $$6) {
+        this.toItem = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$2);
+        this.emeraldCost = $$4;
+        this.maxUses = $$5;
+        this.villagerXp = $$6;
+        this.fromItem = $$0;
+        this.fromCount = $$1;
+        this.toCount = $$3;
         this.priceMultiplier = 0.05f;
     }
 
-    public net.minecraft.world.item.trading.MerchantOffer getOffer(net.minecraft.world.entity.Entity $$0, net.minecraft.util.RandomSource $$1) {
+    public net.minecraft.world.item.trading.MerchantOffer getOffer(net.minecraft.world.entity.Entity $$02, net.minecraft.util.RandomSource $$1) {
         net.minecraft.world.item.ItemStack $$2 = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)net.minecraft.world.item.Items.EMERALD, this.emeraldCost);
-        java.util.List $$3 = (java.util.List)net.minecraft.core.Registry.POTION.stream().filter(potion -> !potion.getEffects().isEmpty() && net.minecraft.world.item.alchemy.PotionBrewing.isBrewablePotion((net.minecraft.world.item.alchemy.Potion)potion)).collect(java.util.stream.Collectors.toList());
+        java.util.List $$3 = (java.util.List)net.minecraft.core.Registry.POTION.stream().filter($$0 -> !$$0.getEffects().isEmpty() && net.minecraft.world.item.alchemy.PotionBrewing.isBrewablePotion((net.minecraft.world.item.alchemy.Potion)$$0)).collect(java.util.stream.Collectors.toList());
         net.minecraft.world.item.alchemy.Potion $$4 = (net.minecraft.world.item.alchemy.Potion)$$3.get($$1.nextInt($$3.size()));
         net.minecraft.world.item.ItemStack $$5 = net.minecraft.world.item.alchemy.PotionUtils.setPotion((net.minecraft.world.item.ItemStack)new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)this.toItem.getItem(), this.toCount), (net.minecraft.world.item.alchemy.Potion)$$4);
         return new net.minecraft.world.item.trading.MerchantOffer($$2, new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)this.fromItem, this.fromCount), $$5, this.maxUses, this.villagerXp, this.priceMultiplier);