--- net/minecraft/world/entity/npc/VillagerTrades$ItemsAndEmeraldsToItems.java
+++ net/minecraft/world/entity/npc/VillagerTrades$ItemsAndEmeraldsToItems.java
@@ -15,18 +15,18 @@
     public int villagerXp;
     public float priceMultiplier;
 
-    public net.minecraft.world.entity.npc.VillagerTrades$ItemsAndEmeraldsToItems(net.minecraft.world.level.ItemLike fromItem, int fromCount, net.minecraft.world.item.Item toItem, int toCount, int maxUses, int villagerXp) {
-        this(fromItem, fromCount, 1, toItem, toCount, maxUses, villagerXp);
+    public net.minecraft.world.entity.npc.VillagerTrades$ItemsAndEmeraldsToItems(net.minecraft.world.level.ItemLike $$0, int $$1, net.minecraft.world.item.Item $$2, int $$3, int $$4, int $$5) {
+        this($$0, $$1, 1, $$2, $$3, $$4, $$5);
     }
 
-    public net.minecraft.world.entity.npc.VillagerTrades$ItemsAndEmeraldsToItems(net.minecraft.world.level.ItemLike fromItem, int fromCount, int emeraldCost, net.minecraft.world.item.Item toItem, int toCount, int maxUses, int villagerXp) {
-        this.fromItem = new net.minecraft.world.item.ItemStack(fromItem);
-        this.fromCount = fromCount;
-        this.emeraldCost = emeraldCost;
-        this.toItem = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)toItem);
-        this.toCount = toCount;
-        this.maxUses = maxUses;
-        this.villagerXp = villagerXp;
+    public net.minecraft.world.entity.npc.VillagerTrades$ItemsAndEmeraldsToItems(net.minecraft.world.level.ItemLike $$0, int $$1, int $$2, net.minecraft.world.item.Item $$3, int $$4, int $$5, int $$6) {
+        this.fromItem = new net.minecraft.world.item.ItemStack($$0);
+        this.fromCount = $$1;
+        this.emeraldCost = $$2;
+        this.toItem = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$3);
+        this.toCount = $$4;
+        this.maxUses = $$5;
+        this.villagerXp = $$6;
         this.priceMultiplier = 0.05f;
     }
 