--- net/minecraft/world/entity/npc/VillagerTrades$EnchantedItemForEmeralds.java
+++ net/minecraft/world/entity/npc/VillagerTrades$EnchantedItemForEmeralds.java
@@ -12,16 +12,16 @@
     public int villagerXp;
     public float priceMultiplier;
 
-    public net.minecraft.world.entity.npc.VillagerTrades$EnchantedItemForEmeralds(net.minecraft.world.item.Item item, int baseEmeraldCost, int maxUses, int villagerXp) {
-        this(item, baseEmeraldCost, maxUses, villagerXp, 0.05f);
+    public net.minecraft.world.entity.npc.VillagerTrades$EnchantedItemForEmeralds(net.minecraft.world.item.Item $$0, int $$1, int $$2, int $$3) {
+        this($$0, $$1, $$2, $$3, 0.05f);
     }
 
-    public net.minecraft.world.entity.npc.VillagerTrades$EnchantedItemForEmeralds(net.minecraft.world.item.Item item, int baseEmeraldCost, int maxUses, int villagerXp, float priceMultiplier) {
-        this.itemStack = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)item);
-        this.baseEmeraldCost = baseEmeraldCost;
-        this.maxUses = maxUses;
-        this.villagerXp = villagerXp;
-        this.priceMultiplier = priceMultiplier;
+    public net.minecraft.world.entity.npc.VillagerTrades$EnchantedItemForEmeralds(net.minecraft.world.item.Item $$0, int $$1, int $$2, int $$3, float $$4) {
+        this.itemStack = new net.minecraft.world.item.ItemStack((net.minecraft.world.level.ItemLike)$$0);
+        this.baseEmeraldCost = $$1;
+        this.maxUses = $$2;
+        this.villagerXp = $$3;
+        this.priceMultiplier = $$4;
     }
 
     public net.minecraft.world.item.trading.MerchantOffer getOffer(net.minecraft.world.entity.Entity $$0, net.minecraft.util.RandomSource $$1) {