--- net/minecraft/world/entity/npc/VillagerType.java
+++ net/minecraft/world/entity/npc/VillagerType.java
@@ -15,52 +15,52 @@
     public java.lang.String name;
     public static java.util.Map<net.minecraft.resources.ResourceKey<net.minecraft.world.level.biome.Biome>, net.minecraft.world.entity.npc.VillagerType> BY_BIOME;
 
-    public net.minecraft.world.entity.npc.VillagerType(java.lang.String name) {
-        this.name = name;
+    public net.minecraft.world.entity.npc.VillagerType(java.lang.String $$0) {
+        this.name = $$0;
     }
 
     public java.lang.String toString() {
         return this.name;
     }
 
-    public static net.minecraft.world.entity.npc.VillagerType register(java.lang.String key) {
-        return (net.minecraft.world.entity.npc.VillagerType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.VILLAGER_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation(key), (java.lang.Object)new net.minecraft.world.entity.npc.VillagerType(key));
+    public static net.minecraft.world.entity.npc.VillagerType register(java.lang.String $$0) {
+        return (net.minecraft.world.entity.npc.VillagerType)net.minecraft.core.Registry.register((net.minecraft.core.Registry)net.minecraft.core.Registry.VILLAGER_TYPE, (net.minecraft.resources.ResourceLocation)new net.minecraft.resources.ResourceLocation($$0), (java.lang.Object)new net.minecraft.world.entity.npc.VillagerType($$0));
     }
 
     public static net.minecraft.world.entity.npc.VillagerType byBiome(net.minecraft.core.Holder<net.minecraft.world.level.biome.Biome> $$0) {
         return (net.minecraft.world.entity.npc.VillagerType)$$0.unwrapKey().map(arg_0 -> ((java.util.Map)BY_BIOME).get(arg_0)).orElse((java.lang.Object)PLAINS);
     }
 
-    public static /* synthetic */ void lambda$static$0(java.util.HashMap map) {
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BADLANDS, (java.lang.Object)DESERT);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DESERT, (java.lang.Object)DESERT);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ERODED_BADLANDS, (java.lang.Object)DESERT);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WOODED_BADLANDS, (java.lang.Object)DESERT);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BAMBOO_JUNGLE, (java.lang.Object)JUNGLE);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JUNGLE, (java.lang.Object)JUNGLE);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SPARSE_JUNGLE, (java.lang.Object)JUNGLE);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA_PLATEAU, (java.lang.Object)SAVANNA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA, (java.lang.Object)SAVANNA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_SAVANNA, (java.lang.Object)SAVANNA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DEEP_FROZEN_OCEAN, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_OCEAN, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_RIVER, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ICE_SPIKES, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_BEACH, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_TAIGA, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_PLAINS, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.GROVE, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_SLOPES, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_PEAKS, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JAGGED_PEAKS, (java.lang.Object)SNOW);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SWAMP, (java.lang.Object)SWAMP);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.MANGROVE_SWAMP, (java.lang.Object)SWAMP);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_SPRUCE_TAIGA, (java.lang.Object)TAIGA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_PINE_TAIGA, (java.lang.Object)TAIGA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_GRAVELLY_HILLS, (java.lang.Object)TAIGA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_HILLS, (java.lang.Object)TAIGA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.TAIGA, (java.lang.Object)TAIGA);
-        map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_FOREST, (java.lang.Object)TAIGA);
+    public static /* synthetic */ void lambda$static$0(java.util.HashMap $$0) {
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BADLANDS, (java.lang.Object)DESERT);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DESERT, (java.lang.Object)DESERT);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ERODED_BADLANDS, (java.lang.Object)DESERT);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WOODED_BADLANDS, (java.lang.Object)DESERT);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BAMBOO_JUNGLE, (java.lang.Object)JUNGLE);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JUNGLE, (java.lang.Object)JUNGLE);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SPARSE_JUNGLE, (java.lang.Object)JUNGLE);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA_PLATEAU, (java.lang.Object)SAVANNA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA, (java.lang.Object)SAVANNA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_SAVANNA, (java.lang.Object)SAVANNA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DEEP_FROZEN_OCEAN, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_OCEAN, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_RIVER, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ICE_SPIKES, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_BEACH, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_TAIGA, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_PLAINS, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.GROVE, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_SLOPES, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_PEAKS, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JAGGED_PEAKS, (java.lang.Object)SNOW);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SWAMP, (java.lang.Object)SWAMP);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.MANGROVE_SWAMP, (java.lang.Object)SWAMP);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_SPRUCE_TAIGA, (java.lang.Object)TAIGA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_PINE_TAIGA, (java.lang.Object)TAIGA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_GRAVELLY_HILLS, (java.lang.Object)TAIGA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_HILLS, (java.lang.Object)TAIGA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.TAIGA, (java.lang.Object)TAIGA);
+        $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_FOREST, (java.lang.Object)TAIGA);
     }
 
     static {
@@ -71,36 +71,36 @@
         SNOW = net.minecraft.world.entity.npc.VillagerType.register((java.lang.String)"snow");
         SWAMP = net.minecraft.world.entity.npc.VillagerType.register((java.lang.String)"swamp");
         TAIGA = net.minecraft.world.entity.npc.VillagerType.register((java.lang.String)"taiga");
-        BY_BIOME = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), map -> {
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BADLANDS, (java.lang.Object)DESERT);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DESERT, (java.lang.Object)DESERT);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ERODED_BADLANDS, (java.lang.Object)DESERT);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WOODED_BADLANDS, (java.lang.Object)DESERT);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BAMBOO_JUNGLE, (java.lang.Object)JUNGLE);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JUNGLE, (java.lang.Object)JUNGLE);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SPARSE_JUNGLE, (java.lang.Object)JUNGLE);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA_PLATEAU, (java.lang.Object)SAVANNA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA, (java.lang.Object)SAVANNA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_SAVANNA, (java.lang.Object)SAVANNA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DEEP_FROZEN_OCEAN, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_OCEAN, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_RIVER, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ICE_SPIKES, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_BEACH, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_TAIGA, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_PLAINS, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.GROVE, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_SLOPES, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_PEAKS, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JAGGED_PEAKS, (java.lang.Object)SNOW);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SWAMP, (java.lang.Object)SWAMP);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.MANGROVE_SWAMP, (java.lang.Object)SWAMP);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_SPRUCE_TAIGA, (java.lang.Object)TAIGA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_PINE_TAIGA, (java.lang.Object)TAIGA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_GRAVELLY_HILLS, (java.lang.Object)TAIGA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_HILLS, (java.lang.Object)TAIGA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.TAIGA, (java.lang.Object)TAIGA);
-            map.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_FOREST, (java.lang.Object)TAIGA);
+        BY_BIOME = (java.util.Map)net.minecraft.Util.make((java.lang.Object)com.google.common.collect.Maps.newHashMap(), $$0 -> {
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BADLANDS, (java.lang.Object)DESERT);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DESERT, (java.lang.Object)DESERT);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ERODED_BADLANDS, (java.lang.Object)DESERT);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WOODED_BADLANDS, (java.lang.Object)DESERT);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.BAMBOO_JUNGLE, (java.lang.Object)JUNGLE);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JUNGLE, (java.lang.Object)JUNGLE);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SPARSE_JUNGLE, (java.lang.Object)JUNGLE);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA_PLATEAU, (java.lang.Object)SAVANNA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SAVANNA, (java.lang.Object)SAVANNA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_SAVANNA, (java.lang.Object)SAVANNA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.DEEP_FROZEN_OCEAN, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_OCEAN, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_RIVER, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.ICE_SPIKES, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_BEACH, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_TAIGA, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_PLAINS, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.GROVE, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SNOWY_SLOPES, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.FROZEN_PEAKS, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.JAGGED_PEAKS, (java.lang.Object)SNOW);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.SWAMP, (java.lang.Object)SWAMP);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.MANGROVE_SWAMP, (java.lang.Object)SWAMP);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_SPRUCE_TAIGA, (java.lang.Object)TAIGA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.OLD_GROWTH_PINE_TAIGA, (java.lang.Object)TAIGA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_GRAVELLY_HILLS, (java.lang.Object)TAIGA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_HILLS, (java.lang.Object)TAIGA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.TAIGA, (java.lang.Object)TAIGA);
+            $$0.put((java.lang.Object)net.minecraft.world.level.biome.Biomes.WINDSWEPT_FOREST, (java.lang.Object)TAIGA);
         });
     }
 }