--- net/minecraft/world/entity/npc/VillagerTrades$EmeraldsForVillagerTypeItem.java
+++ net/minecraft/world/entity/npc/VillagerTrades$EmeraldsForVillagerTypeItem.java
@@ -11,14 +11,14 @@
     public int maxUses;
     public int villagerXp;
 
-    public net.minecraft.world.entity.npc.VillagerTrades$EmeraldsForVillagerTypeItem(int cost, int maxUses, int villagerXp, java.util.Map<net.minecraft.world.entity.npc.VillagerType, net.minecraft.world.item.Item> trades) {
-        net.minecraft.core.Registry.VILLAGER_TYPE.stream().filter(type -> !trades.containsKey(type)).findAny().ifPresent(type -> {
-            throw new java.lang.IllegalStateException("Missing trade for villager type: " + net.minecraft.core.Registry.VILLAGER_TYPE.getKey(type));
+    public net.minecraft.world.entity.npc.VillagerTrades$EmeraldsForVillagerTypeItem(int $$02, int $$12, int $$2, java.util.Map<net.minecraft.world.entity.npc.VillagerType, net.minecraft.world.item.Item> $$3) {
+        net.minecraft.core.Registry.VILLAGER_TYPE.stream().filter($$1 -> !$$3.containsKey($$1)).findAny().ifPresent($$0 -> {
+            throw new java.lang.IllegalStateException("Missing trade for villager type: " + net.minecraft.core.Registry.VILLAGER_TYPE.getKey($$0));
         });
-        this.trades = trades;
-        this.cost = cost;
-        this.maxUses = maxUses;
-        this.villagerXp = villagerXp;
+        this.trades = $$3;
+        this.cost = $$02;
+        this.maxUses = $$12;
+        this.villagerXp = $$2;
     }
 
     @javax.annotation.Nullable