--- net/minecraft/world/entity/npc/WanderingTrader$WanderToPositionGoal.java
+++ net/minecraft/world/entity/npc/WanderingTrader$WanderToPositionGoal.java
@@ -11,11 +11,11 @@
     public double speedModifier;
     public /* synthetic */ net.minecraft.world.entity.npc.WanderingTrader this$0;
 
-    public net.minecraft.world.entity.npc.WanderingTrader$WanderToPositionGoal(net.minecraft.world.entity.npc.WanderingTrader wanderingTrader, net.minecraft.world.entity.npc.WanderingTrader trader, double stopDistance, double speedModifier) {
+    public net.minecraft.world.entity.npc.WanderingTrader$WanderToPositionGoal(net.minecraft.world.entity.npc.WanderingTrader wanderingTrader, net.minecraft.world.entity.npc.WanderingTrader $$0, double $$1, double $$2) {
         this.this$0 = wanderingTrader;
-        this.trader = trader;
-        this.stopDistance = stopDistance;
-        this.speedModifier = speedModifier;
+        this.trader = $$0;
+        this.stopDistance = $$1;
+        this.speedModifier = $$2;
         this.setFlags(java.util.EnumSet.of((java.lang.Enum)net.minecraft.world.entity.ai.goal.Goal$Flag.MOVE));
     }
 
@@ -42,8 +42,8 @@
         }
     }
 
-    public boolean isTooFarAway(net.minecraft.core.BlockPos pos, double distance) {
-        return !pos.closerToCenterThan((net.minecraft.core.Position)this.trader.position(), distance);
+    public boolean isTooFarAway(net.minecraft.core.BlockPos $$0, double $$1) {
+        return !$$0.closerToCenterThan((net.minecraft.core.Position)this.trader.position(), $$1);
     }
 }
 