--- net/minecraft/client/resources/language/FormattedBidiReorder.java
+++ net/minecraft/client/resources/language/FormattedBidiReorder.java
@@ -8,17 +8,17 @@
     public net.minecraft.client.resources.language.FormattedBidiReorder() {
     }
 
-    public static net.minecraft.util.FormattedCharSequence reorder(net.minecraft.network.chat.FormattedText text, boolean bl) {
-        net.minecraft.network.chat.SubStringSource subStringSource = net.minecraft.network.chat.SubStringSource.create((net.minecraft.network.chat.FormattedText)text, com.ibm.icu.lang.UCharacter::getMirror, net.minecraft.client.resources.language.FormattedBidiReorder::shape);
-        com.ibm.icu.text.Bidi bidi = new com.ibm.icu.text.Bidi(subStringSource.getPlainText(), bl ? 127 : 126);
-        bidi.setReorderingMode(0);
-        java.util.ArrayList arrayList = com.google.common.collect.Lists.newArrayList();
-        int n = bidi.countRuns();
-        for (int i = 0; i < n; ++i) {
-            com.ibm.icu.text.BidiRun bidiRun = bidi.getVisualRun(i);
-            arrayList.addAll((java.util.Collection)subStringSource.substring(bidiRun.getStart(), bidiRun.getLength(), bidiRun.isOddRun()));
+    public static net.minecraft.util.FormattedCharSequence reorder(net.minecraft.network.chat.FormattedText $$0, boolean $$1) {
+        net.minecraft.network.chat.SubStringSource $$2 = net.minecraft.network.chat.SubStringSource.create((net.minecraft.network.chat.FormattedText)$$0, com.ibm.icu.lang.UCharacter::getMirror, net.minecraft.client.resources.language.FormattedBidiReorder::shape);
+        com.ibm.icu.text.Bidi $$3 = new com.ibm.icu.text.Bidi($$2.getPlainText(), $$1 ? 127 : 126);
+        $$3.setReorderingMode(0);
+        java.util.ArrayList $$4 = com.google.common.collect.Lists.newArrayList();
+        int $$5 = $$3.countRuns();
+        for (int $$6 = 0; $$6 < $$5; ++$$6) {
+            com.ibm.icu.text.BidiRun $$7 = $$3.getVisualRun($$6);
+            $$4.addAll((java.util.Collection)$$2.substring($$7.getStart(), $$7.getLength(), $$7.isOddRun()));
         }
-        return net.minecraft.util.FormattedCharSequence.composite((java.util.List)arrayList);
+        return net.minecraft.util.FormattedCharSequence.composite((java.util.List)$$4);
     }
 
     public static java.lang.String shape(java.lang.String $$0) {