--- net/minecraft/client/resources/language/LanguageInfo.java
+++ net/minecraft/client/resources/language/LanguageInfo.java
@@ -12,11 +12,11 @@
     public java.lang.String name;
     public boolean bidirectional;
 
-    public net.minecraft.client.resources.language.LanguageInfo(java.lang.String code, java.lang.String region, java.lang.String name, boolean bidirectional) {
-        this.code = code;
-        this.region = region;
-        this.name = name;
-        this.bidirectional = bidirectional;
+    public net.minecraft.client.resources.language.LanguageInfo(java.lang.String $$0, java.lang.String $$1, java.lang.String $$2, boolean $$3) {
+        this.code = $$0;
+        this.region = $$1;
+        this.name = $$2;
+        this.bidirectional = $$3;
     }
 
     public java.lang.String getCode() {