--- net/minecraft/client/multiplayer/chat/report/AbuseReportSender$Services.java
+++ net/minecraft/client/multiplayer/chat/report/AbuseReportSender$Services.java
@@ -20,7 +20,7 @@
 
     public java.util.concurrent.CompletableFuture<com.mojang.datafixers.util.Unit> send(java.util.UUID $$0, com.mojang.authlib.minecraft.report.AbuseReport $$1) {
         return java.util.concurrent.CompletableFuture.supplyAsync(() -> {
-            com.mojang.authlib.yggdrasil.request.AbuseReportRequest $$2 = new com.mojang.authlib.yggdrasil.request.AbuseReportRequest($$0, $$1, this.environment.clientInfo(), this.environment.thirdPartyServerInfo(), this.environment.realmInfo());
+            com.mojang.authlib.yggdrasil.request.AbuseReportRequest $$2 = new com.mojang.authlib.yggdrasil.request.AbuseReportRequest(1, $$0, $$1, this.environment.clientInfo(), this.environment.thirdPartyServerInfo(), this.environment.realmInfo());
             try {
                 this.userApiService.reportAbuse($$2);
                 return com.mojang.datafixers.util.Unit.INSTANCE;