--- net/minecraft/client/multiplayer/chat/report/ReportingContext.java
+++ net/minecraft/client/multiplayer/chat/report/ReportingContext.java
@@ -18,9 +18,9 @@
     }
 
     public static net.minecraft.client.multiplayer.chat.report.ReportingContext create(net.minecraft.client.multiplayer.chat.report.ReportEnvironment $$0, com.mojang.authlib.minecraft.UserApiService $$1) {
-        net.minecraft.client.multiplayer.chat.RollingMemoryChatLog $$2 = new net.minecraft.client.multiplayer.chat.RollingMemoryChatLog(1024);
+        net.minecraft.client.multiplayer.chat.ChatLog $$2 = new net.minecraft.client.multiplayer.chat.ChatLog(1024);
         net.minecraft.client.multiplayer.chat.report.AbuseReportSender $$3 = net.minecraft.client.multiplayer.chat.report.AbuseReportSender.create((net.minecraft.client.multiplayer.chat.report.ReportEnvironment)$$0, (com.mojang.authlib.minecraft.UserApiService)$$1);
-        return new net.minecraft.client.multiplayer.chat.report.ReportingContext($$3, $$0, (net.minecraft.client.multiplayer.chat.ChatLog)$$2);
+        return new net.minecraft.client.multiplayer.chat.report.ReportingContext($$3, $$0, $$2);
     }
 
     public boolean matches(net.minecraft.client.multiplayer.chat.report.ReportEnvironment $$0) {