--- net/minecraft/world/entity/animal/camel/CamelAi$RandomSitting.java
+++ net/minecraft/world/entity/animal/camel/CamelAi$RandomSitting.java
@@ -1,0 +1,28 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.entity.animal.camel;
+
+public class net.minecraft.world.entity.animal.camel.CamelAi$RandomSitting
+extends net.minecraft.world.entity.ai.behavior.Behavior<net.minecraft.world.entity.animal.camel.Camel>
+ {
+    public int minimalPoseTicks;
+
+    public net.minecraft.world.entity.animal.camel.CamelAi$RandomSitting(int $$0) {
+        super((java.util.Map)com.google.common.collect.ImmutableMap.of());
+        this.minimalPoseTicks = $$0 * 20;
+    }
+
+    public boolean checkExtraStartConditions(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.animal.camel.Camel $$1) {
+        return !$$1.isInWater() && $$1.getPoseTime() >= (long)this.minimalPoseTicks && !$$1.isLeashed() && $$1.isOnGround() && !$$1.hasControllingPassenger();
+    }
+
+    public void start(net.minecraft.server.level.ServerLevel $$0, net.minecraft.world.entity.animal.camel.Camel $$1, long $$2) {
+        if ($$1.isPoseSitting()) {
+            $$1.standUp();
+        } else if (!$$1.isPanicking()) {
+            $$1.sitDown();
+        }
+    }
+}
+