--- net/minecraft/world/entity/animal/camel/Camel$1.java
+++ net/minecraft/world/entity/animal/camel/Camel$1.java
@@ -1,0 +1,26 @@
+/*
+ * Decompiled with CFR 0.151.
+ */
+package net.minecraft.world.entity.animal.camel;
+
+public class net.minecraft.world.entity.animal.camel.Camel$1
+ {
+    public static /* synthetic */ int[] $SwitchMap$net$minecraft$world$entity$Pose;
+
+    static {
+        $SwitchMap$net$minecraft$world$entity$Pose = new int[net.minecraft.world.entity.Pose.values().length];
+        try {
+            net.minecraft.world.entity.animal.camel.Camel$1.$SwitchMap$net$minecraft$world$entity$Pose[net.minecraft.world.entity.Pose.STANDING.ordinal()] = 1;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+        try {
+            net.minecraft.world.entity.animal.camel.Camel$1.$SwitchMap$net$minecraft$world$entity$Pose[net.minecraft.world.entity.Pose.SITTING.ordinal()] = 2;
+        }
+        catch (java.lang.NoSuchFieldError noSuchFieldError) {
+            // empty catch block
+        }
+    }
+}
+