--- net/minecraft/client/resources/metadata/language/LanguageMetadataSection.java
+++ net/minecraft/client/resources/metadata/language/LanguageMetadataSection.java
@@ -9,8 +9,8 @@
     public static boolean DEFAULT_BIDIRECTIONAL = false;
     public java.util.Collection<net.minecraft.client.resources.language.LanguageInfo> languages;
 
-    public net.minecraft.client.resources.metadata.language.LanguageMetadataSection(java.util.Collection<net.minecraft.client.resources.language.LanguageInfo> languages) {
-        this.languages = languages;
+    public net.minecraft.client.resources.metadata.language.LanguageMetadataSection(java.util.Collection<net.minecraft.client.resources.language.LanguageInfo> $$0) {
+        this.languages = $$0;
     }
 
     public java.util.Collection<net.minecraft.client.resources.language.LanguageInfo> getLanguages() {