--- com/mojang/realmsclient/gui/RealmsWorldSlotButton.java
+++ com/mojang/realmsclient/gui/RealmsWorldSlotButton.java
@@ -104,11 +104,11 @@
         return com.mojang.datafixers.util.Pair.of((java.lang.Object)$$9, (java.lang.Object)$$10);
     }
 
-    public void renderButton(com.mojang.blaze3d.vertex.PoseStack matrixStack, int mouseX, int mouseY, float partialTicks) {
+    public void renderButton(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, float $$3) {
         if (this.state == null) {
             return;
         }
-        this.drawSlotFrame(matrixStack, this.x, this.y, mouseX, mouseY, this.state.isCurrentlyActiveSlot, this.state.slotName, this.slotIndex, this.state.imageId, this.state.image, this.state.empty, this.state.minigame, this.state.action, this.state.actionPrompt);
+        this.drawSlotFrame($$0, this.x, this.y, $$1, $$2, this.state.isCurrentlyActiveSlot, this.state.slotName, this.slotIndex, this.state.imageId, this.state.image, this.state.empty, this.state.minigame, this.state.action, this.state.actionPrompt);
     }
 
     public void drawSlotFrame(com.mojang.blaze3d.vertex.PoseStack $$0, int $$1, int $$2, int $$3, int $$4, boolean $$5, java.lang.String $$6, int $$7, long $$8, @javax.annotation.Nullable java.lang.String $$9, boolean $$10, boolean $$11, com.mojang.realmsclient.gui.RealmsWorldSlotButton$Action $$12, @javax.annotation.Nullable net.minecraft.network.chat.Component $$13) {
@@ -165,7 +165,7 @@
     static {
         SLOT_FRAME_LOCATION = new net.minecraft.resources.ResourceLocation("realms", "textures/gui/realms/slot_frame.png");
         EMPTY_SLOT_LOCATION = new net.minecraft.resources.ResourceLocation("realms", "textures/gui/realms/empty_frame.png");
-        CHECK_MARK_LOCATION = new net.minecraft.resources.ResourceLocation("realms", "textures/gui/realms/checkmark.png");
+        CHECK_MARK_LOCATION = new net.minecraft.resources.ResourceLocation("minecraft", "textures/gui/checkmark.png");
         DEFAULT_WORLD_SLOT_1 = new net.minecraft.resources.ResourceLocation("minecraft", "textures/gui/title/background/panorama_0.png");
         DEFAULT_WORLD_SLOT_2 = new net.minecraft.resources.ResourceLocation("minecraft", "textures/gui/title/background/panorama_2.png");
         DEFAULT_WORLD_SLOT_3 = new net.minecraft.resources.ResourceLocation("minecraft", "textures/gui/title/background/panorama_3.png");